// memory-pager/index.js
J$.noInstrEval = false;
jalangiLabel13:
    while (true) {
        try {
            J$.Se(529593, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/memory-pager/index.js');
            function Pager(pageSize, opts) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(525777, arguments.callee, this, arguments);
                            arguments = J$.N(525785, 'arguments', arguments, true, false, false);
                            pageSize = J$.N(525793, 'pageSize', pageSize, true, false, false);
                            opts = J$.N(525801, 'opts', opts, true, false, false);
                            if (J$.C(29552, J$.U(73010, '!', J$.B(73002, 'instanceof', J$.R(525369, 'this', this, false, false), J$.R(525377, 'Pager', Pager, false, true)))))
                                return J$.Rt(525417, J$.F(525409, J$.R(525385, 'Pager', Pager, false, true), true)(J$.R(525393, 'pageSize', pageSize, false, false), J$.R(525401, 'opts', opts, false, false)));
                            J$.P(525441, J$.R(525425, 'this', this, false, false), 'length', J$.T(525433, 0, 22, false));
                            J$.P(525465, J$.R(525449, 'this', this, false, false), 'updates', J$.T(525457, [], 10, false));
                            J$.P(525505, J$.R(525473, 'this', this, false, false), 'path', J$.F(525497, J$.I(typeof Uint16Array === 'undefined' ? Uint16Array = J$.R(525481, 'Uint16Array', undefined, true, true) : Uint16Array = J$.R(525481, 'Uint16Array', Uint16Array, true, true)), true)(J$.T(525489, 4, 22, false)));
                            J$.P(525545, J$.R(525513, 'this', this, false, false), 'pages', J$.F(525537, J$.I(typeof Array === 'undefined' ? Array = J$.R(525521, 'Array', undefined, true, true) : Array = J$.R(525521, 'Array', Array, true, true)), true)(J$.T(525529, 32768, 22, false)));
                            J$.P(525585, J$.R(525553, 'this', this, false, false), 'maxPages', J$.G(525577, J$.G(525569, J$.R(525561, 'this', this, false, false), 'pages'), 'length'));
                            J$.P(525609, J$.R(525593, 'this', this, false, false), 'level', J$.T(525601, 0, 22, false));
                            J$.P(525641, J$.R(525617, 'this', this, false, false), 'pageSize', J$.C(29560, J$.R(525625, 'pageSize', pageSize, false, false)) ? J$._() : J$.T(525633, 1024, 22, false));
                            J$.P(525689, J$.R(525649, 'this', this, false, false), 'deduplicate', J$.C(29568, J$.R(525657, 'opts', opts, false, false)) ? J$.G(525673, J$.R(525665, 'opts', opts, false, false), 'deduplicate') : J$.T(525681, null, 25, false));
                            J$.P(525769, J$.R(525697, 'this', this, false, false), 'zeros', J$.C(29576, J$.G(525713, J$.R(525705, 'this', this, false, false), 'deduplicate')) ? J$.F(525753, J$.R(525721, 'alloc', alloc, false, true), false)(J$.G(525745, J$.G(525737, J$.R(525729, 'this', this, false, false), 'deduplicate'), 'length')) : J$.T(525761, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(529713, J$e);
                        } finally {
                            if (J$.Fr(529721))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function grow(pager, index) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(528633, arguments.callee, this, arguments);
                            arguments = J$.N(528641, 'arguments', arguments, true, false, false);
                            pager = J$.N(528649, 'pager', pager, true, false, false);
                            index = J$.N(528657, 'index', index, true, false, false);
                            J$.N(528665, 'old', old, false, false, false);
                            while (J$.C(29872, J$.B(73258, '<', J$.G(528473, J$.R(528465, 'pager', pager, false, false), 'maxPages'), J$.R(528481, 'index', index, false, false)))) {
                                var old = J$.W(528505, 'old', J$.G(528497, J$.R(528489, 'pager', pager, false, false), 'pages'), old, false, false);
                                J$.P(528545, J$.R(528513, 'pager', pager, false, false), 'pages', J$.F(528537, J$.I(typeof Array === 'undefined' ? Array = J$.R(528521, 'Array', undefined, true, true) : Array = J$.R(528521, 'Array', Array, true, true)), true)(J$.T(528529, 32768, 22, false)));
                                J$.P(528585, J$.G(528561, J$.R(528553, 'pager', pager, false, false), 'pages'), J$.T(528569, 0, 22, false), J$.R(528577, 'old', old, false, false));
                                J$.B(73266, '-', J$.A(528601, J$.R(528593, 'pager', pager, false, false), 'level', '+')(1), 1);
                                J$.A(528625, J$.R(528609, 'pager', pager, false, false), 'maxPages', '*')(J$.T(528617, 32768, 22, false));
                            }
                        } catch (J$e) {
                            J$.Ex(529825, J$e);
                        } finally {
                            if (J$.Fr(529833))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function truncate(buf, len) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(528849, arguments.callee, this, arguments);
                            arguments = J$.N(528857, 'arguments', arguments, true, false, false);
                            buf = J$.N(528865, 'buf', buf, true, false, false);
                            len = J$.N(528873, 'len', len, true, false, false);
                            J$.N(528881, 'cpy', cpy, false, false, false);
                            if (J$.C(29880, J$.B(73274, '===', J$.G(528681, J$.R(528673, 'buf', buf, false, false), 'length'), J$.R(528689, 'len', len, false, false))))
                                return J$.Rt(528705, J$.R(528697, 'buf', buf, false, false));
                            if (J$.C(29888, J$.B(73282, '>', J$.G(528721, J$.R(528713, 'buf', buf, false, false), 'length'), J$.R(528729, 'len', len, false, false))))
                                return J$.Rt(528769, J$.M(528761, J$.R(528737, 'buf', buf, false, false), 'slice', false)(J$.T(528745, 0, 22, false), J$.R(528753, 'len', len, false, false)));
                            var cpy = J$.W(528801, 'cpy', J$.F(528793, J$.R(528777, 'alloc', alloc, false, true), false)(J$.R(528785, 'len', len, false, false)), cpy, false, false);
                            J$.M(528825, J$.R(528809, 'buf', buf, false, false), 'copy', false)(J$.R(528817, 'cpy', cpy, false, false));
                            return J$.Rt(528841, J$.R(528833, 'cpy', cpy, false, false));
                        } catch (J$e) {
                            J$.Ex(529841, J$e);
                        } finally {
                            if (J$.Fr(529849))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function alloc(size) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(529009, arguments.callee, this, arguments);
                            arguments = J$.N(529017, 'arguments', arguments, true, false, false);
                            size = J$.N(529025, 'size', size, true, false, false);
                            J$.N(529033, 'buf', buf, false, false, false);
                            if (J$.C(29896, J$.G(528897, J$.I(typeof Buffer === 'undefined' ? Buffer = J$.R(528889, 'Buffer', undefined, true, true) : Buffer = J$.R(528889, 'Buffer', Buffer, true, true)), 'alloc')))
                                return J$.Rt(528929, J$.M(528921, J$.I(typeof Buffer === 'undefined' ? Buffer = J$.R(528905, 'Buffer', undefined, true, true) : Buffer = J$.R(528905, 'Buffer', Buffer, true, true)), 'alloc', false)(J$.R(528913, 'size', size, false, false)));
                            var buf = J$.W(528961, 'buf', J$.F(528953, J$.I(typeof Buffer === 'undefined' ? Buffer = J$.R(528937, 'Buffer', undefined, true, true) : Buffer = J$.R(528937, 'Buffer', Buffer, true, true)), true)(J$.R(528945, 'size', size, false, false)), buf, false, false);
                            J$.M(528985, J$.R(528969, 'buf', buf, false, false), 'fill', false)(J$.T(528977, 0, 22, false));
                            return J$.Rt(529001, J$.R(528993, 'buf', buf, false, false));
                        } catch (J$e) {
                            J$.Ex(529857, J$e);
                        } finally {
                            if (J$.Fr(529865))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function copy(buf) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(529169, arguments.callee, this, arguments);
                            arguments = J$.N(529177, 'arguments', arguments, true, false, false);
                            buf = J$.N(529185, 'buf', buf, true, false, false);
                            J$.N(529193, 'cpy', cpy, false, false, false);
                            var cpy = J$.W(529121, 'cpy', J$.C(29904, J$.G(529049, J$.I(typeof Buffer === 'undefined' ? Buffer = J$.R(529041, 'Buffer', undefined, true, true) : Buffer = J$.R(529041, 'Buffer', Buffer, true, true)), 'allocUnsafe')) ? J$.M(529081, J$.I(typeof Buffer === 'undefined' ? Buffer = J$.R(529057, 'Buffer', undefined, true, true) : Buffer = J$.R(529057, 'Buffer', Buffer, true, true)), 'allocUnsafe', false)(J$.G(529073, J$.R(529065, 'buf', buf, false, false), 'length')) : J$.F(529113, J$.I(typeof Buffer === 'undefined' ? Buffer = J$.R(529089, 'Buffer', undefined, true, true) : Buffer = J$.R(529089, 'Buffer', Buffer, true, true)), true)(J$.G(529105, J$.R(529097, 'buf', buf, false, false), 'length')), cpy, false, false);
                            J$.M(529145, J$.R(529129, 'buf', buf, false, false), 'copy', false)(J$.R(529137, 'cpy', cpy, false, false));
                            return J$.Rt(529161, J$.R(529153, 'cpy', cpy, false, false));
                        } catch (J$e) {
                            J$.Ex(529873, J$e);
                        } finally {
                            if (J$.Fr(529881))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function Page(i, buf) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(529313, arguments.callee, this, arguments);
                            arguments = J$.N(529321, 'arguments', arguments, true, false, false);
                            i = J$.N(529329, 'i', i, true, false, false);
                            buf = J$.N(529337, 'buf', buf, true, false, false);
                            J$.P(529233, J$.R(529201, 'this', this, false, false), 'offset', J$.B(73290, '*', J$.R(529209, 'i', i, false, false), J$.G(529225, J$.R(529217, 'buf', buf, false, false), 'length')));
                            J$.P(529257, J$.R(529241, 'this', this, false, false), 'buffer', J$.R(529249, 'buf', buf, false, false));
                            J$.P(529281, J$.R(529265, 'this', this, false, false), 'updated', J$.T(529273, false, 23, false));
                            J$.P(529305, J$.R(529289, 'this', this, false, false), 'deduplicate', J$.T(529297, 0, 22, false));
                        } catch (J$e) {
                            J$.Ex(529889, J$e);
                        } finally {
                            if (J$.Fr(529897))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function factor(n, out) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(529561, arguments.callee, this, arguments);
                            arguments = J$.N(529569, 'arguments', arguments, true, false, false);
                            n = J$.N(529577, 'n', n, true, false, false);
                            out = J$.N(529585, 'out', out, true, false, false);
                            n = J$.W(529401, 'n', J$.B(73314, '/', J$.B(73306, '-', J$.R(529345, 'n', n, false, false), J$.P(529385, J$.R(529353, 'out', out, false, false), J$.T(529361, 0, 22, false), J$.B(73298, '&', J$.R(529369, 'n', n, false, false), J$.T(529377, 32767, 22, false)))), J$.T(529393, 32768, 22, false)), n, false, false);
                            n = J$.W(529465, 'n', J$.B(73338, '/', J$.B(73330, '-', J$.R(529409, 'n', n, false, false), J$.P(529449, J$.R(529417, 'out', out, false, false), J$.T(529425, 1, 22, false), J$.B(73322, '&', J$.R(529433, 'n', n, false, false), J$.T(529441, 32767, 22, false)))), J$.T(529457, 32768, 22, false)), n, false, false);
                            J$.P(529553, J$.R(529473, 'out', out, false, false), J$.T(529481, 3, 22, false), J$.B(73370, '&', J$.B(73362, '/', J$.B(73354, '-', J$.R(529489, 'n', n, false, false), J$.P(529529, J$.R(529497, 'out', out, false, false), J$.T(529505, 2, 22, false), J$.B(73346, '&', J$.R(529513, 'n', n, false, false), J$.T(529521, 32767, 22, false)))), J$.T(529537, 32768, 22, false)), J$.T(529545, 32767, 22, false)));
                        } catch (J$e) {
                            J$.Ex(529905, J$e);
                        } finally {
                            if (J$.Fr(529913))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            Pager = J$.N(529609, 'Pager', J$.T(529601, Pager, 12, false), true, false, false);
            grow = J$.N(529625, 'grow', J$.T(529617, grow, 12, false), true, false, false);
            truncate = J$.N(529641, 'truncate', J$.T(529633, truncate, 12, false), true, false, false);
            alloc = J$.N(529657, 'alloc', J$.T(529649, alloc, 12, false), true, false, false);
            copy = J$.N(529673, 'copy', J$.T(529665, copy, 12, false), true, false, false);
            Page = J$.N(529689, 'Page', J$.T(529681, Page, 12, false), true, false, false);
            factor = J$.N(529705, 'factor', J$.T(529697, factor, 12, false), true, false, false);
            J$.T(525337, 'use strict', 21, false);
            J$.P(525361, J$.I(typeof module === 'undefined' ? module = J$.R(525345, 'module', undefined, true, true) : module = J$.R(525345, 'module', module, true, true)), 'exports', J$.R(525353, 'Pager', Pager, false, true));
            J$.P(526225, J$.G(525817, J$.R(525809, 'Pager', Pager, false, true), 'prototype'), 'updated', J$.T(526217, function (page) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(526193, arguments.callee, this, arguments);
                            arguments = J$.N(526201, 'arguments', arguments, true, false, false);
                            page = J$.N(526209, 'page', page, true, false, false);
                            while (J$.C(29616, J$.C(29584, J$.G(525833, J$.R(525825, 'this', this, false, false), 'deduplicate')) ? J$.B(73018, '===', J$.G(525873, J$.G(525849, J$.R(525841, 'page', page, false, false), 'buffer'), J$.G(525865, J$.R(525857, 'page', page, false, false), 'deduplicate')), J$.G(525913, J$.G(525889, J$.R(525881, 'this', this, false, false), 'deduplicate'), J$.G(525905, J$.R(525897, 'page', page, false, false), 'deduplicate'))) : J$._())) {
                                J$.B(73026, '-', J$.A(525929, J$.R(525921, 'page', page, false, false), 'deduplicate', '+')(1), 1);
                                if (J$.C(29608, J$.B(73034, '===', J$.G(525945, J$.R(525937, 'page', page, false, false), 'deduplicate'), J$.G(525969, J$.G(525961, J$.R(525953, 'this', this, false, false), 'deduplicate'), 'length')))) {
                                    J$.P(525993, J$.R(525977, 'page', page, false, false), 'deduplicate', J$.T(525985, 0, 22, false));
                                    if (J$.C(29600, J$.C(29592, J$.G(526017, J$.G(526009, J$.R(526001, 'page', page, false, false), 'buffer'), 'equals')) ? J$.M(526057, J$.G(526033, J$.R(526025, 'page', page, false, false), 'buffer'), 'equals', false)(J$.G(526049, J$.R(526041, 'this', this, false, false), 'deduplicate')) : J$._()))
                                        J$.P(526089, J$.R(526065, 'page', page, false, false), 'buffer', J$.G(526081, J$.R(526073, 'this', this, false, false), 'deduplicate'));
                                    break;
                                }
                            }
                            if (J$.C(29632, J$.C(29624, J$.G(526105, J$.R(526097, 'page', page, false, false), 'updated')) ? J$._() : J$.U(73042, '!', J$.G(526121, J$.R(526113, 'this', this, false, false), 'updates'))))
                                return J$.Rt(526129, undefined);
                            J$.P(526153, J$.R(526137, 'page', page, false, false), 'updated', J$.T(526145, true, 23, false));
                            J$.M(526185, J$.G(526169, J$.R(526161, 'this', this, false, false), 'updates'), 'push', false)(J$.R(526177, 'page', page, false, false));
                        } catch (J$e) {
                            J$.Ex(529729, J$e);
                        } finally {
                            if (J$.Fr(529737))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(526409, J$.G(526241, J$.R(526233, 'Pager', Pager, false, true), 'prototype'), 'lastUpdate', J$.T(526401, function () {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(526377, arguments.callee, this, arguments);
                            arguments = J$.N(526385, 'arguments', arguments, true, false, false);
                            J$.N(526393, 'page', page, false, false, false);
                            if (J$.C(29648, J$.C(29640, J$.U(73050, '!', J$.G(526257, J$.R(526249, 'this', this, false, false), 'updates'))) ? J$._() : J$.U(73058, '!', J$.G(526281, J$.G(526273, J$.R(526265, 'this', this, false, false), 'updates'), 'length'))))
                                return J$.Rt(526297, J$.T(526289, null, 25, false));
                            var page = J$.W(526329, 'page', J$.M(526321, J$.G(526313, J$.R(526305, 'this', this, false, false), 'updates'), 'pop', false)(), page, false, false);
                            J$.P(526353, J$.R(526337, 'page', page, false, false), 'updated', J$.T(526345, false, 23, false));
                            return J$.Rt(526369, J$.R(526361, 'page', page, false, false));
                        } catch (J$e) {
                            J$.Ex(529745, J$e);
                        } finally {
                            if (J$.Fr(529753))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(526881, J$.G(526425, J$.R(526417, 'Pager', Pager, false, true), 'prototype'), '_array', J$.T(526873, function (i, noAllocate) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(526809, arguments.callee, this, arguments);
                            arguments = J$.N(526817, 'arguments', arguments, true, false, false);
                            i = J$.N(526825, 'i', i, true, false, false);
                            noAllocate = J$.N(526833, 'noAllocate', noAllocate, true, false, false);
                            J$.N(526841, 'arr', arr, false, false, false);
                            J$.N(526849, 'j', j, false, false, false);
                            J$.N(526857, 'p', p, false, false, false);
                            J$.N(526865, 'next', next, false, false, false);
                            if (J$.C(29664, J$.B(73066, '>=', J$.R(526433, 'i', i, false, false), J$.G(526449, J$.R(526441, 'this', this, false, false), 'maxPages')))) {
                                if (J$.C(29656, J$.R(526457, 'noAllocate', noAllocate, false, false)))
                                    return J$.Rt(526465, undefined);
                                J$.F(526497, J$.R(526473, 'grow', grow, false, true), false)(J$.R(526481, 'this', this, false, false), J$.R(526489, 'i', i, false, false));
                            }
                            J$.F(526537, J$.R(526505, 'factor', factor, false, true), false)(J$.R(526513, 'i', i, false, false), J$.G(526529, J$.R(526521, 'this', this, false, false), 'path'));
                            var arr = J$.W(526561, 'arr', J$.G(526553, J$.R(526545, 'this', this, false, false), 'pages'), arr, false, false);
                            for (var j = J$.W(526585, 'j', J$.G(526577, J$.R(526569, 'this', this, false, false), 'level'), j, false, false); J$.C(29688, J$.B(73074, '>', J$.R(526593, 'j', j, false, false), J$.T(526601, 0, 22, false))); J$.B(73098, '+', j = J$.W(526617, 'j', J$.B(73090, '-', J$.U(73082, '+', J$.R(526609, 'j', j, false, false)), 1), j, false, false), 1)) {
                                var p = J$.W(526657, 'p', J$.G(526649, J$.G(526633, J$.R(526625, 'this', this, false, false), 'path'), J$.R(526641, 'j', j, false, false)), p, false, false);
                                var next = J$.W(526689, 'next', J$.G(526681, J$.R(526665, 'arr', arr, false, false), J$.R(526673, 'p', p, false, false)), next, false, false);
                                if (J$.C(29680, J$.U(73106, '!', J$.R(526697, 'next', next, false, false)))) {
                                    if (J$.C(29672, J$.R(526705, 'noAllocate', noAllocate, false, false)))
                                        return J$.Rt(526713, undefined);
                                    next = J$.W(526769, 'next', J$.P(526761, J$.R(526721, 'arr', arr, false, false), J$.R(526729, 'p', p, false, false), J$.F(526753, J$.I(typeof Array === 'undefined' ? Array = J$.R(526737, 'Array', undefined, true, true) : Array = J$.R(526737, 'Array', Array, true, true)), true)(J$.T(526745, 32768, 22, false))), next, false, false);
                                }
                                arr = J$.W(526785, 'arr', J$.R(526777, 'next', next, false, false), arr, false, false);
                            }
                            return J$.Rt(526801, J$.R(526793, 'arr', arr, false, false));
                        } catch (J$e) {
                            J$.Ex(529761, J$e);
                        } finally {
                            if (J$.Fr(529769))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(527401, J$.G(526897, J$.R(526889, 'Pager', Pager, false, true), 'prototype'), 'get', J$.T(527393, function (i, noAllocate) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(527337, arguments.callee, this, arguments);
                            arguments = J$.N(527345, 'arguments', arguments, true, false, false);
                            i = J$.N(527353, 'i', i, true, false, false);
                            noAllocate = J$.N(527361, 'noAllocate', noAllocate, true, false, false);
                            J$.N(527369, 'arr', arr, false, false, false);
                            J$.N(527377, 'first', first, false, false, false);
                            J$.N(527385, 'page', page, false, false, false);
                            var arr = J$.W(526937, 'arr', J$.M(526929, J$.R(526905, 'this', this, false, false), '_array', false)(J$.R(526913, 'i', i, false, false), J$.R(526921, 'noAllocate', noAllocate, false, false)), arr, false, false);
                            var first = J$.W(526977, 'first', J$.G(526969, J$.G(526953, J$.R(526945, 'this', this, false, false), 'path'), J$.T(526961, 0, 22, false)), first, false, false);
                            var page = J$.W(527017, 'page', J$.C(29696, J$.R(526985, 'arr', arr, false, false)) ? J$.G(527009, J$.R(526993, 'arr', arr, false, false), J$.R(527001, 'first', first, false, false)) : J$._(), page, false, false);
                            if (J$.C(29720, J$.C(29704, J$.U(73114, '!', J$.R(527025, 'page', page, false, false))) ? J$.U(73122, '!', J$.R(527033, 'noAllocate', noAllocate, false, false)) : J$._())) {
                                page = J$.W(527121, 'page', J$.P(527113, J$.R(527041, 'arr', arr, false, false), J$.R(527049, 'first', first, false, false), J$.F(527105, J$.R(527057, 'Page', Page, false, true), true)(J$.R(527065, 'i', i, false, false), J$.F(527097, J$.R(527073, 'alloc', alloc, false, true), false)(J$.G(527089, J$.R(527081, 'this', this, false, false), 'pageSize')))), page, false, false);
                                if (J$.C(29712, J$.B(73130, '>=', J$.R(527129, 'i', i, false, false), J$.G(527145, J$.R(527137, 'this', this, false, false), 'length'))))
                                    J$.P(527177, J$.R(527153, 'this', this, false, false), 'length', J$.B(73138, '+', J$.R(527161, 'i', i, false, false), J$.T(527169, 1, 22, false)));
                            }
                            if (J$.C(29752, J$.C(29744, J$.C(29736, J$.C(29728, J$.R(527185, 'page', page, false, false)) ? J$.B(73146, '===', J$.G(527201, J$.R(527193, 'page', page, false, false), 'buffer'), J$.G(527217, J$.R(527209, 'this', this, false, false), 'deduplicate')) : J$._()) ? J$.G(527233, J$.R(527225, 'this', this, false, false), 'deduplicate') : J$._()) ? J$.U(73154, '!', J$.R(527241, 'noAllocate', noAllocate, false, false)) : J$._())) {
                                J$.P(527289, J$.R(527249, 'page', page, false, false), 'buffer', J$.F(527281, J$.R(527257, 'copy', copy, false, true), false)(J$.G(527273, J$.R(527265, 'page', page, false, false), 'buffer')));
                                J$.P(527313, J$.R(527297, 'page', page, false, false), 'deduplicate', J$.T(527305, 0, 22, false));
                            }
                            return J$.Rt(527329, J$.R(527321, 'page', page, false, false));
                        } catch (J$e) {
                            J$.Ex(529777, J$e);
                        } finally {
                            if (J$.Fr(529785))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(528001, J$.G(527417, J$.R(527409, 'Pager', Pager, false, true), 'prototype'), 'set', J$.T(527993, function (i, buf) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(527929, arguments.callee, this, arguments);
                            arguments = J$.N(527937, 'arguments', arguments, true, false, false);
                            i = J$.N(527945, 'i', i, true, false, false);
                            buf = J$.N(527953, 'buf', buf, true, false, false);
                            J$.N(527961, 'arr', arr, false, false, false);
                            J$.N(527969, 'first', first, false, false, false);
                            J$.N(527977, 'page', page, false, false, false);
                            J$.N(527985, 'b', b, false, false, false);
                            var arr = J$.W(527457, 'arr', J$.M(527449, J$.R(527425, 'this', this, false, false), '_array', false)(J$.R(527433, 'i', i, false, false), J$.T(527441, false, 23, false)), arr, false, false);
                            var first = J$.W(527497, 'first', J$.G(527489, J$.G(527473, J$.R(527465, 'this', this, false, false), 'path'), J$.T(527481, 0, 22, false)), first, false, false);
                            if (J$.C(29760, J$.B(73162, '>=', J$.R(527505, 'i', i, false, false), J$.G(527521, J$.R(527513, 'this', this, false, false), 'length'))))
                                J$.P(527553, J$.R(527529, 'this', this, false, false), 'length', J$.B(73170, '+', J$.R(527537, 'i', i, false, false), J$.T(527545, 1, 22, false)));
                            if (J$.C(29792, J$.C(29784, J$.U(73178, '!', J$.R(527561, 'buf', buf, false, false))) ? J$._() : J$.C(29776, J$.C(29768, J$.G(527577, J$.R(527569, 'this', this, false, false), 'zeros')) ? J$.G(527593, J$.R(527585, 'buf', buf, false, false), 'equals') : J$._()) ? J$.M(527625, J$.R(527601, 'buf', buf, false, false), 'equals', false)(J$.G(527617, J$.R(527609, 'this', this, false, false), 'zeros')) : J$._())) {
                                J$.P(527657, J$.R(527633, 'arr', arr, false, false), J$.R(527641, 'first', first, false, false), J$.T(527649, undefined, 24, false));
                                return J$.Rt(527665, undefined);
                            }
                            if (J$.C(29816, J$.C(29808, J$.C(29800, J$.G(527681, J$.R(527673, 'this', this, false, false), 'deduplicate')) ? J$.G(527697, J$.R(527689, 'buf', buf, false, false), 'equals') : J$._()) ? J$.M(527729, J$.R(527705, 'buf', buf, false, false), 'equals', false)(J$.G(527721, J$.R(527713, 'this', this, false, false), 'deduplicate')) : J$._())) {
                                buf = J$.W(527753, 'buf', J$.G(527745, J$.R(527737, 'this', this, false, false), 'deduplicate'), buf, false, false);
                            }
                            var page = J$.W(527785, 'page', J$.G(527777, J$.R(527761, 'arr', arr, false, false), J$.R(527769, 'first', first, false, false)), page, false, false);
                            var b = J$.W(527833, 'b', J$.F(527825, J$.R(527793, 'truncate', truncate, false, true), false)(J$.R(527801, 'buf', buf, false, false), J$.G(527817, J$.R(527809, 'this', this, false, false), 'pageSize')), b, false, false);
                            if (J$.C(29824, J$.R(527841, 'page', page, false, false)))
                                J$.P(527865, J$.R(527849, 'page', page, false, false), 'buffer', J$.R(527857, 'b', b, false, false));
                            else
                                J$.P(527921, J$.R(527873, 'arr', arr, false, false), J$.R(527881, 'first', first, false, false), J$.F(527913, J$.R(527889, 'Page', Page, false, true), true)(J$.R(527897, 'i', i, false, false), J$.R(527905, 'b', b, false, false)));
                        } catch (J$e) {
                            J$.Ex(529793, J$e);
                        } finally {
                            if (J$.Fr(529801))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(528457, J$.G(528017, J$.R(528009, 'Pager', Pager, false, true), 'prototype'), 'toBuffer', J$.T(528449, function () {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(528393, arguments.callee, this, arguments);
                            arguments = J$.N(528401, 'arguments', arguments, true, false, false);
                            J$.N(528409, 'list', list, false, false, false);
                            J$.N(528417, 'empty', empty, false, false, false);
                            J$.N(528425, 'ptr', ptr, false, false, false);
                            J$.N(528433, 'arr', arr, false, false, false);
                            J$.N(528441, 'i', i, false, false, false);
                            var list = J$.W(528057, 'list', J$.F(528049, J$.I(typeof Array === 'undefined' ? Array = J$.R(528025, 'Array', undefined, true, true) : Array = J$.R(528025, 'Array', Array, true, true)), true)(J$.G(528041, J$.R(528033, 'this', this, false, false), 'length')), list, false, false);
                            var empty = J$.W(528097, 'empty', J$.F(528089, J$.R(528065, 'alloc', alloc, false, true), false)(J$.G(528081, J$.R(528073, 'this', this, false, false), 'pageSize')), empty, false, false);
                            var ptr = J$.W(528113, 'ptr', J$.T(528105, 0, 22, false), ptr, false, false);
                            while (J$.C(29864, J$.B(73186, '<', J$.R(528121, 'ptr', ptr, false, false), J$.G(528137, J$.R(528129, 'list', list, false, false), 'length')))) {
                                var arr = J$.W(528177, 'arr', J$.M(528169, J$.R(528145, 'this', this, false, false), '_array', false)(J$.R(528153, 'ptr', ptr, false, false), J$.T(528161, true, 23, false)), arr, false, false);
                                for (var i = J$.W(528193, 'i', J$.T(528185, 0, 22, false), i, false, false); J$.C(29856, J$.C(29832, J$.B(73194, '<', J$.R(528201, 'i', i, false, false), J$.T(528209, 32768, 22, false))) ? J$.B(73202, '<', J$.R(528217, 'ptr', ptr, false, false), J$.G(528233, J$.R(528225, 'list', list, false, false), 'length')) : J$._()); J$.B(73226, '-', i = J$.W(528249, 'i', J$.B(73218, '+', J$.U(73210, '+', J$.R(528241, 'i', i, false, false)), 1), i, false, false), 1)) {
                                    J$.P(528353, J$.R(528257, 'list', list, false, false), J$.B(73250, '-', ptr = J$.W(528273, 'ptr', J$.B(73242, '+', J$.U(73234, '+', J$.R(528265, 'ptr', ptr, false, false)), 1), ptr, false, false), 1), J$.C(29848, J$.C(29840, J$.R(528281, 'arr', arr, false, false)) ? J$.G(528305, J$.R(528289, 'arr', arr, false, false), J$.R(528297, 'i', i, false, false)) : J$._()) ? J$.G(528337, J$.G(528329, J$.R(528313, 'arr', arr, false, false), J$.R(528321, 'i', i, false, false)), 'buffer') : J$.R(528345, 'empty', empty, false, false));
                                }
                            }
                            return J$.Rt(528385, J$.M(528377, J$.I(typeof Buffer === 'undefined' ? Buffer = J$.R(528361, 'Buffer', undefined, true, true) : Buffer = J$.R(528361, 'Buffer', Buffer, true, true)), 'concat', false)(J$.R(528369, 'list', list, false, false)));
                        } catch (J$e) {
                            J$.Ex(529809, J$e);
                        } finally {
                            if (J$.Fr(529817))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(529921, J$e);
        } finally {
            if (J$.Sr(529929))
                continue jalangiLabel13;
            else
                break jalangiLabel13;
        }
    }
// JALANGI DO NOT INSTRUMENT
