// debug/karma.conf.js
J$.noInstrEval = false;
jalangiLabel1:
    while (true) {
        try {
            J$.Se(2329817, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestAnalytics/node_modules/debug/karma.conf.js');
            J$.P(2329809, J$.I(typeof module === 'undefined' ? module = J$.R(2329569, 'module', undefined, true, true) : module = J$.R(2329569, 'module', module, true, true)), 'exports', J$.T(2329801, function (config) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(2329777, arguments.callee, this, arguments);
                            arguments = J$.N(2329785, 'arguments', arguments, true, false, false);
                            config = J$.N(2329793, 'config', config, true, false, false);
                            J$.M(2329769, J$.R(2329577, 'config', config, false, false), 'set', false)(J$.T(2329761, {
                                basePath: J$.T(2329585, '', 21, false),
                                frameworks: J$.T(2329617, [
                                    J$.T(2329593, 'mocha', 21, false),
                                    J$.T(2329601, 'chai', 21, false),
                                    J$.T(2329609, 'sinon', 21, false)
                                ], 10, false),
                                files: J$.T(2329641, [
                                    J$.T(2329625, 'dist/debug.js', 21, false),
                                    J$.T(2329633, 'test/*spec.js', 21, false)
                                ], 10, false),
                                exclude: J$.T(2329657, [J$.T(2329649, 'src/node.js', 21, false)], 10, false),
                                preprocessors: J$.T(2329665, {}, 11, false),
                                reporters: J$.T(2329681, [J$.T(2329673, 'progress', 21, false)], 10, false),
                                port: J$.T(2329689, 9876, 22, false),
                                colors: J$.T(2329697, true, 23, false),
                                logLevel: J$.G(2329713, J$.R(2329705, 'config', config, false, false), 'LOG_INFO'),
                                autoWatch: J$.T(2329721, true, 23, false),
                                browsers: J$.T(2329737, [J$.T(2329729, 'PhantomJS', 21, false)], 10, false),
                                singleRun: J$.T(2329745, false, 23, false),
                                concurrency: J$.T(2329753, Infinity, 22, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(2329825, J$e);
                        } finally {
                            if (J$.Fr(2329833))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(2329841, J$e);
        } finally {
            if (J$.Sr(2329849))
                continue jalangiLabel1;
            else
                break jalangiLabel1;
        }
    }
// JALANGI DO NOT INSTRUMENT
