// mongodb/lib/core/sdam/server.js
J$.noInstrEval = false;
jalangiLabel49:
    while (true) {
        try {
            J$.Se(769321, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/sdam/server.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(759137, arguments.callee, this, arguments);
                            arguments = J$.N(759145, 'arguments', arguments, true, false, false);
                            obj = J$.N(759153, 'obj', obj, true, false, false);
                            if (J$.C(44000, J$.C(43960, J$.B(89282, '===', J$.U(89274, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(758873, 'Symbol', undefined, true, true) : Symbol = J$.R(758873, 'Symbol', Symbol, true, true))), J$.T(758881, 'function', 21, false))) ? J$.B(89298, '===', J$.U(89290, 'typeof', J$.G(758897, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(758889, 'Symbol', undefined, true, true) : Symbol = J$.R(758889, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(758905, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(758961, '_typeof2', J$.T(758953, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(758929, arguments.callee, this, arguments);
                                                arguments = J$.N(758937, 'arguments', arguments, true, false, false);
                                                obj = J$.N(758945, 'obj', obj, true, false, false);
                                                return J$.Rt(758921, J$.U(89306, 'typeof', J$.R(758913, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(769761, J$e);
                                            } finally {
                                                if (J$.Fr(769769))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(759097, '_typeof2', J$.T(759089, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(759065, arguments.callee, this, arguments);
                                                arguments = J$.N(759073, 'arguments', arguments, true, false, false);
                                                obj = J$.N(759081, 'obj', obj, true, false, false);
                                                return J$.Rt(759057, J$.C(43992, J$.C(43984, J$.C(43976, J$.C(43968, J$.R(758969, 'obj', obj, false, false)) ? J$.B(89322, '===', J$.U(89314, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(758977, 'Symbol', undefined, true, true) : Symbol = J$.R(758977, 'Symbol', Symbol, true, true))), J$.T(758985, 'function', 21, false)) : J$._()) ? J$.B(89330, '===', J$.G(759001, J$.R(758993, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(759009, 'Symbol', undefined, true, true) : Symbol = J$.R(759009, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(89338, '!==', J$.R(759017, 'obj', obj, false, false), J$.G(759033, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(759025, 'Symbol', undefined, true, true) : Symbol = J$.R(759025, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(759041, 'symbol', 21, false) : J$.U(89346, 'typeof', J$.R(759049, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(769777, J$e);
                                            } finally {
                                                if (J$.Fr(769785))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(759129, J$.F(759121, J$.R(759105, '_typeof2', _typeof2, false, true), false)(J$.R(759113, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(769793, J$e);
                        } finally {
                            if (J$.Fr(769801))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(759473, arguments.callee, this, arguments);
                            arguments = J$.N(759481, 'arguments', arguments, true, false, false);
                            obj = J$.N(759489, 'obj', obj, true, false, false);
                            if (J$.C(44048, J$.C(44008, J$.B(89362, '===', J$.U(89354, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(759161, 'Symbol', undefined, true, true) : Symbol = J$.R(759161, 'Symbol', Symbol, true, true))), J$.T(759169, 'function', 21, false))) ? J$.B(89370, '===', J$.F(759201, J$.R(759177, '_typeof2', _typeof2, false, true), false)(J$.G(759193, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(759185, 'Symbol', undefined, true, true) : Symbol = J$.R(759185, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(759209, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(759281, '_typeof', J$.T(759273, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(759249, arguments.callee, this, arguments);
                                                arguments = J$.N(759257, 'arguments', arguments, true, false, false);
                                                obj = J$.N(759265, 'obj', obj, true, false, false);
                                                return J$.Rt(759241, J$.F(759233, J$.R(759217, '_typeof2', _typeof2, false, true), false)(J$.R(759225, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(769809, J$e);
                                            } finally {
                                                if (J$.Fr(769817))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(759433, '_typeof', J$.T(759425, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(759401, arguments.callee, this, arguments);
                                                arguments = J$.N(759409, 'arguments', arguments, true, false, false);
                                                obj = J$.N(759417, 'obj', obj, true, false, false);
                                                return J$.Rt(759393, J$.C(44040, J$.C(44032, J$.C(44024, J$.C(44016, J$.R(759289, 'obj', obj, false, false)) ? J$.B(89386, '===', J$.U(89378, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(759297, 'Symbol', undefined, true, true) : Symbol = J$.R(759297, 'Symbol', Symbol, true, true))), J$.T(759305, 'function', 21, false)) : J$._()) ? J$.B(89394, '===', J$.G(759321, J$.R(759313, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(759329, 'Symbol', undefined, true, true) : Symbol = J$.R(759329, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(89402, '!==', J$.R(759337, 'obj', obj, false, false), J$.G(759353, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(759345, 'Symbol', undefined, true, true) : Symbol = J$.R(759345, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(759361, 'symbol', 21, false) : J$.F(759385, J$.R(759369, '_typeof2', _typeof2, false, true), false)(J$.R(759377, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(769825, J$e);
                                            } finally {
                                                if (J$.Fr(769833))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(759465, J$.F(759457, J$.R(759441, '_typeof', _typeof, false, true), false)(J$.R(759449, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(769841, J$e);
                        } finally {
                            if (J$.Fr(769849))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(759537, arguments.callee, this, arguments);
                            arguments = J$.N(759545, 'arguments', arguments, true, false, false);
                            instance = J$.N(759553, 'instance', instance, true, false, false);
                            Constructor = J$.N(759561, 'Constructor', Constructor, true, false, false);
                            if (J$.C(44056, J$.U(89418, '!', J$.B(89410, 'instanceof', J$.R(759497, 'instance', instance, false, false), J$.R(759505, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(759529, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(759513, 'TypeError', undefined, true, true) : TypeError = J$.R(759513, 'TypeError', TypeError, true, true)), true)(J$.T(759521, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(769857, J$e);
                        } finally {
                            if (J$.Fr(769865))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(759809, arguments.callee, this, arguments);
                            arguments = J$.N(759817, 'arguments', arguments, true, false, false);
                            target = J$.N(759825, 'target', target, true, false, false);
                            props = J$.N(759833, 'props', props, true, false, false);
                            J$.N(759841, 'i', i, false, false, false);
                            J$.N(759849, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(759577, 'i', J$.T(759569, 0, 22, false), i, false, false); J$.C(44080, J$.B(89426, '<', J$.R(759585, 'i', i, false, false), J$.G(759601, J$.R(759593, 'props', props, false, false), 'length'))); J$.B(89450, '-', i = J$.W(759617, 'i', J$.B(89442, '+', J$.U(89434, '+', J$.R(759609, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(759649, 'descriptor', J$.G(759641, J$.R(759625, 'props', props, false, false), J$.R(759633, 'i', i, false, false)), descriptor, false, false);
                                J$.P(759689, J$.R(759657, 'descriptor', descriptor, false, false), 'enumerable', J$.C(44064, J$.G(759673, J$.R(759665, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(759681, false, 23, false));
                                J$.P(759713, J$.R(759697, 'descriptor', descriptor, false, false), 'configurable', J$.T(759705, true, 23, false));
                                if (J$.C(44072, J$.B(89458, 'in', J$.T(759721, 'value', 21, false), J$.R(759729, 'descriptor', descriptor, false, false))))
                                    J$.P(759753, J$.R(759737, 'descriptor', descriptor, false, false), 'writable', J$.T(759745, true, 23, false));
                                J$.M(759801, J$.I(typeof Object === 'undefined' ? Object = J$.R(759761, 'Object', undefined, true, true) : Object = J$.R(759761, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(759769, 'target', target, false, false), J$.G(759785, J$.R(759777, 'descriptor', descriptor, false, false), 'key'), J$.R(759793, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(769873, J$e);
                        } finally {
                            if (J$.Fr(769881))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(759961, arguments.callee, this, arguments);
                            arguments = J$.N(759969, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(759977, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(759985, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(759993, 'staticProps', staticProps, true, false, false);
                            if (J$.C(44088, J$.R(759857, 'protoProps', protoProps, false, false)))
                                J$.F(759897, J$.R(759865, '_defineProperties', _defineProperties, false, true), false)(J$.G(759881, J$.R(759873, 'Constructor', Constructor, false, false), 'prototype'), J$.R(759889, 'protoProps', protoProps, false, false));
                            if (J$.C(44096, J$.R(759905, 'staticProps', staticProps, false, false)))
                                J$.F(759937, J$.R(759913, '_defineProperties', _defineProperties, false, true), false)(J$.R(759921, 'Constructor', Constructor, false, false), J$.R(759929, 'staticProps', staticProps, false, false));
                            return J$.Rt(759953, J$.R(759945, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(769889, J$e);
                        } finally {
                            if (J$.Fr(769897))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(760105, arguments.callee, this, arguments);
                            arguments = J$.N(760113, 'arguments', arguments, true, false, false);
                            self = J$.N(760121, 'self', self, true, false, false);
                            call = J$.N(760129, 'call', call, true, false, false);
                            if (J$.C(44120, J$.C(44112, J$.R(760001, 'call', call, false, false)) ? J$.C(44104, J$.B(89466, '===', J$.F(760025, J$.R(760009, '_typeof', _typeof, false, true), false)(J$.R(760017, 'call', call, false, false)), J$.T(760033, 'object', 21, false))) ? J$._() : J$.B(89482, '===', J$.U(89474, 'typeof', J$.R(760041, 'call', call, false, false)), J$.T(760049, 'function', 21, false)) : J$._())) {
                                return J$.Rt(760065, J$.R(760057, 'call', call, false, false));
                            }
                            return J$.Rt(760097, J$.F(760089, J$.R(760073, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(760081, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(769905, J$e);
                        } finally {
                            if (J$.Fr(769913))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(760193, arguments.callee, this, arguments);
                            arguments = J$.N(760201, 'arguments', arguments, true, false, false);
                            self = J$.N(760209, 'self', self, true, false, false);
                            if (J$.C(44128, J$.B(89490, '===', J$.R(760137, 'self', self, false, false), void J$.T(760145, 0, 22, false)))) {
                                throw J$.F(760169, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(760153, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(760153, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(760161, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(760185, J$.R(760177, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(769921, J$e);
                        } finally {
                            if (J$.Fr(769929))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(760369, arguments.callee, this, arguments);
                            arguments = J$.N(760377, 'arguments', arguments, true, false, false);
                            o = J$.N(760385, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(760329, '_getPrototypeOf', J$.C(44144, J$.G(760225, J$.I(typeof Object === 'undefined' ? Object = J$.R(760217, 'Object', undefined, true, true) : Object = J$.R(760217, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(760241, J$.I(typeof Object === 'undefined' ? Object = J$.R(760233, 'Object', undefined, true, true) : Object = J$.R(760233, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(760321, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(760297, arguments.callee, this, arguments);
                                            arguments = J$.N(760305, 'arguments', arguments, true, false, false);
                                            o = J$.N(760313, 'o', o, true, false, false);
                                            return J$.Rt(760289, J$.C(44136, J$.G(760257, J$.R(760249, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(760281, J$.I(typeof Object === 'undefined' ? Object = J$.R(760265, 'Object', undefined, true, true) : Object = J$.R(760265, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(760273, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(769937, J$e);
                                        } finally {
                                            if (J$.Fr(769945))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(760361, J$.F(760353, J$.R(760337, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(760345, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(769953, J$e);
                        } finally {
                            if (J$.Fr(769961))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(760585, arguments.callee, this, arguments);
                            arguments = J$.N(760593, 'arguments', arguments, true, false, false);
                            subClass = J$.N(760601, 'subClass', subClass, true, false, false);
                            superClass = J$.N(760609, 'superClass', superClass, true, false, false);
                            if (J$.C(44160, J$.C(44152, J$.B(89506, '!==', J$.U(89498, 'typeof', J$.R(760393, 'superClass', superClass, false, false)), J$.T(760401, 'function', 21, false))) ? J$.B(89514, '!==', J$.R(760409, 'superClass', superClass, false, false), J$.T(760417, null, 25, false)) : J$._())) {
                                throw J$.F(760441, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(760425, 'TypeError', undefined, true, true) : TypeError = J$.R(760425, 'TypeError', TypeError, true, true)), true)(J$.T(760433, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(760537, J$.R(760449, 'subClass', subClass, false, false), 'prototype', J$.M(760529, J$.I(typeof Object === 'undefined' ? Object = J$.R(760457, 'Object', undefined, true, true) : Object = J$.R(760457, 'Object', Object, true, true)), 'create', false)(J$.C(44168, J$.R(760465, 'superClass', superClass, false, false)) ? J$.G(760481, J$.R(760473, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(760521, {
                                constructor: J$.T(760513, {
                                    value: J$.R(760489, 'subClass', subClass, false, false),
                                    writable: J$.T(760497, true, 23, false),
                                    configurable: J$.T(760505, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(44176, J$.R(760545, 'superClass', superClass, false, false)))
                                J$.F(760577, J$.R(760553, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(760561, 'subClass', subClass, false, false), J$.R(760569, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(769969, J$e);
                        } finally {
                            if (J$.Fr(769977))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(760761, arguments.callee, this, arguments);
                            arguments = J$.N(760769, 'arguments', arguments, true, false, false);
                            o = J$.N(760777, 'o', o, true, false, false);
                            p = J$.N(760785, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(760713, '_setPrototypeOf', J$.C(44184, J$.G(760625, J$.I(typeof Object === 'undefined' ? Object = J$.R(760617, 'Object', undefined, true, true) : Object = J$.R(760617, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(760705, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(760673, arguments.callee, this, arguments);
                                            arguments = J$.N(760681, 'arguments', arguments, true, false, false);
                                            o = J$.N(760689, 'o', o, true, false, false);
                                            p = J$.N(760697, 'p', p, true, false, false);
                                            J$.P(760649, J$.R(760633, 'o', o, false, false), '__proto__', J$.R(760641, 'p', p, false, false));
                                            return J$.Rt(760665, J$.R(760657, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(769985, J$e);
                                        } finally {
                                            if (J$.Fr(769993))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(760753, J$.F(760745, J$.R(760721, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(760729, 'o', o, false, false), J$.R(760737, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(770001, J$e);
                        } finally {
                            if (J$.Fr(770009))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function basicWriteValidations(server) {
                jalangiLabel39:
                    while (true) {
                        try {
                            J$.Fe(767361, arguments.callee, this, arguments);
                            arguments = J$.N(767369, 'arguments', arguments, true, false, false);
                            server = J$.N(767377, 'server', server, true, false, false);
                            if (J$.C(44488, J$.U(89642, '!', J$.G(767241, J$.G(767233, J$.R(767225, 'server', server, false, false), 's'), 'pool')))) {
                                return J$.Rt(767273, J$.F(767265, J$.R(767249, 'MongoError', MongoError, false, true), true)(J$.T(767257, 'server instance is not connected', 21, false)));
                            }
                            if (J$.C(44496, J$.M(767305, J$.G(767297, J$.G(767289, J$.R(767281, 'server', server, false, false), 's'), 'pool'), 'isDestroyed', false)())) {
                                return J$.Rt(767337, J$.F(767329, J$.R(767313, 'MongoError', MongoError, false, true), true)(J$.T(767321, 'server instance pool was destroyed', 21, false)));
                            }
                            return J$.Rt(767353, J$.T(767345, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(770385, J$e);
                        } finally {
                            if (J$.Fr(770393))
                                continue jalangiLabel39;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function basicReadValidations(server, options) {
                jalangiLabel40:
                    while (true) {
                        try {
                            J$.Fe(767521, arguments.callee, this, arguments);
                            arguments = J$.N(767529, 'arguments', arguments, true, false, false);
                            server = J$.N(767537, 'server', server, true, false, false);
                            options = J$.N(767545, 'options', options, true, false, false);
                            J$.N(767553, 'error', error, false, false, false);
                            var error = J$.W(767417, 'error', J$.F(767409, J$.R(767385, 'basicWriteValidations', basicWriteValidations, false, true), false)(J$.R(767393, 'server', server, false, false), J$.R(767401, 'options', options, false, false)), error, false, false);
                            if (J$.C(44504, J$.R(767425, 'error', error, false, false))) {
                                return J$.Rt(767441, J$.R(767433, 'error', error, false, false));
                            }
                            if (J$.C(44520, J$.C(44512, J$.G(767457, J$.R(767449, 'options', options, false, false), 'readPreference')) ? J$.U(89658, '!', J$.B(89650, 'instanceof', J$.G(767473, J$.R(767465, 'options', options, false, false), 'readPreference'), J$.R(767481, 'ReadPreference', ReadPreference, false, true))) : J$._())) {
                                return J$.Rt(767513, J$.F(767505, J$.R(767489, 'MongoError', MongoError, false, true), true)(J$.T(767497, 'readPreference must be an instance of ReadPreference', 21, false)));
                            }
                        } catch (J$e) {
                            J$.Ex(770401, J$e);
                        } finally {
                            if (J$.Fr(770409))
                                continue jalangiLabel40;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function executeWriteOperation(args, options, callback) {
                jalangiLabel42:
                    while (true) {
                        try {
                            J$.Fe(768265, arguments.callee, this, arguments);
                            arguments = J$.N(768273, 'arguments', arguments, true, false, false);
                            args = J$.N(768281, 'args', args, true, false, false);
                            options = J$.N(768289, 'options', options, true, false, false);
                            callback = J$.N(768297, 'callback', callback, true, false, false);
                            J$.N(768305, 'server', server, false, false, false);
                            J$.N(768313, 'op', op, false, false, false);
                            J$.N(768321, 'ns', ns, false, false, false);
                            J$.N(768329, 'ops', ops, false, false, false);
                            J$.N(768337, 'error', error, false, false, false);
                            if (J$.C(44528, J$.B(89674, '===', J$.U(89666, 'typeof', J$.R(767561, 'options', options, false, false)), J$.T(767569, 'function', 21, false))))
                                callback = J$.W(767585, 'callback', J$.R(767577, 'options', options, false, false), callback, false, false), options = J$.W(767601, 'options', J$.T(767593, {}, 11, false), options, false, false);
                            options = J$.W(767625, 'options', J$.C(44536, J$.R(767609, 'options', options, false, false)) ? J$._() : J$.T(767617, {}, 11, false), options, false, false);
                            var server = J$.W(767649, 'server', J$.G(767641, J$.R(767633, 'args', args, false, false), 'server'), server, false, false);
                            var op = J$.W(767673, 'op', J$.G(767665, J$.R(767657, 'args', args, false, false), 'op'), op, false, false);
                            var ns = J$.W(767697, 'ns', J$.G(767689, J$.R(767681, 'args', args, false, false), 'ns'), ns, false, false);
                            var ops = J$.W(767777, 'ops', J$.C(44544, J$.M(767729, J$.I(typeof Array === 'undefined' ? Array = J$.R(767705, 'Array', undefined, true, true) : Array = J$.R(767705, 'Array', Array, true, true)), 'isArray', false)(J$.G(767721, J$.R(767713, 'args', args, false, false), 'ops'))) ? J$.G(767745, J$.R(767737, 'args', args, false, false), 'ops') : J$.T(767769, [J$.G(767761, J$.R(767753, 'args', args, false, false), 'ops')], 10, false), ops, false, false);
                            var error = J$.W(767817, 'error', J$.F(767809, J$.R(767785, 'basicWriteValidations', basicWriteValidations, false, true), false)(J$.R(767793, 'server', server, false, false), J$.R(767801, 'options', options, false, false)), error, false, false);
                            if (J$.C(44552, J$.R(767825, 'error', error, false, false))) {
                                J$.F(767857, J$.R(767833, 'callback', callback, false, false), false)(J$.R(767841, 'error', error, false, false), J$.T(767849, null, 25, false));
                                return J$.Rt(767865, undefined);
                            }
                            if (J$.C(44560, J$.F(767897, J$.R(767873, 'collationNotSupported', collationNotSupported, false, true), false)(J$.R(767881, 'server', server, false, false), J$.R(767889, 'options', options, false, false)))) {
                                J$.F(767969, J$.R(767905, 'callback', callback, false, false), false)(J$.F(767961, J$.R(767913, 'MongoError', MongoError, false, true), true)(J$.M(767953, J$.T(767921, 'server ', 21, false), 'concat', false)(J$.G(767937, J$.R(767929, 'server', server, false, false), 'name'), J$.T(767945, ' does not support collation', 21, false))));
                                return J$.Rt(767977, undefined);
                            }
                            return J$.Rt(768257, J$.M(768249, J$.R(767985, 'wireProtocol', wireProtocol, false, true), J$.R(767993, 'op', op, false, false), false)(J$.R(768001, 'server', server, false, false), J$.R(768009, 'ns', ns, false, false), J$.R(768017, 'ops', ops, false, false), J$.R(768025, 'options', options, false, false), J$.T(768241, function (err, result) {
                                jalangiLabel41:
                                    while (true) {
                                        try {
                                            J$.Fe(768209, arguments.callee, this, arguments);
                                            arguments = J$.N(768217, 'arguments', arguments, true, false, false);
                                            err = J$.N(768225, 'err', err, true, false, false);
                                            result = J$.N(768233, 'result', result, true, false, false);
                                            if (J$.C(44592, J$.R(768033, 'err', err, false, false))) {
                                                if (J$.C(44576, J$.C(44568, J$.G(768049, J$.R(768041, 'options', options, false, false), 'session')) ? J$.B(89682, 'instanceof', J$.R(768057, 'err', err, false, false), J$.R(768065, 'MongoNetworkError', MongoNetworkError, false, true)) : J$._())) {
                                                    J$.P(768105, J$.G(768089, J$.G(768081, J$.R(768073, 'options', options, false, false), 'session'), 'serverSession'), 'isDirty', J$.T(768097, true, 23, false));
                                                }
                                                if (J$.C(44584, J$.F(768137, J$.R(768113, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true), false)(J$.R(768121, 'err', err, false, false), J$.R(768129, 'server', server, false, false)))) {
                                                    J$.M(768169, J$.R(768145, 'server', server, false, false), 'emit', false)(J$.T(768153, 'error', 21, false), J$.R(768161, 'err', err, false, false));
                                                }
                                            }
                                            J$.F(768201, J$.R(768177, 'callback', callback, false, false), false)(J$.R(768185, 'err', err, false, false), J$.R(768193, 'result', result, false, false));
                                        } catch (J$e) {
                                            J$.Ex(770417, J$e);
                                        } finally {
                                            if (J$.Fr(770425))
                                                continue jalangiLabel41;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(770433, J$e);
                        } finally {
                            if (J$.Fr(770441))
                                continue jalangiLabel42;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function connectEventHandler(server) {
                jalangiLabel44:
                    while (true) {
                        try {
                            J$.Fe(768985, arguments.callee, this, arguments);
                            arguments = J$.N(768993, 'arguments', arguments, true, false, false);
                            server = J$.N(769001, 'server', server, true, false, false);
                            return J$.Rt(768977, J$.T(768969, function (pool, conn) {
                                jalangiLabel43:
                                    while (true) {
                                        try {
                                            J$.Fe(768921, arguments.callee, this, arguments);
                                            arguments = J$.N(768929, 'arguments', arguments, true, false, false);
                                            pool = J$.N(768937, 'pool', pool, true, false, false);
                                            conn = J$.N(768945, 'conn', conn, true, false, false);
                                            J$.N(768953, 'ismaster', ismaster, false, false, false);
                                            J$.N(768961, '$clusterTime', $clusterTime, false, false, false);
                                            var ismaster = J$.W(768361, 'ismaster', J$.G(768353, J$.R(768345, 'conn', conn, false, false), 'ismaster'), ismaster, false, false);
                                            J$.P(768401, J$.G(768377, J$.R(768369, 'server', server, false, false), 's'), 'lastIsMasterMS', J$.G(768393, J$.R(768385, 'conn', conn, false, false), 'lastIsMasterMS'));
                                            if (J$.C(44600, J$.G(768417, J$.R(768409, 'conn', conn, false, false), 'agreedCompressor'))) {
                                                J$.P(768473, J$.G(768449, J$.G(768441, J$.G(768433, J$.R(768425, 'server', server, false, false), 's'), 'pool'), 'options'), 'agreedCompressor', J$.G(768465, J$.R(768457, 'conn', conn, false, false), 'agreedCompressor'));
                                            }
                                            if (J$.C(44608, J$.G(768489, J$.R(768481, 'conn', conn, false, false), 'zlibCompressionLevel'))) {
                                                J$.P(768545, J$.G(768521, J$.G(768513, J$.G(768505, J$.R(768497, 'server', server, false, false), 's'), 'pool'), 'options'), 'zlibCompressionLevel', J$.G(768537, J$.R(768529, 'conn', conn, false, false), 'zlibCompressionLevel'));
                                            }
                                            if (J$.C(44616, J$.G(768569, J$.G(768561, J$.R(768553, 'conn', conn, false, false), 'ismaster'), '$clusterTime'))) {
                                                var $clusterTime = J$.W(768601, '$clusterTime', J$.G(768593, J$.G(768585, J$.R(768577, 'conn', conn, false, false), 'ismaster'), '$clusterTime'), $clusterTime, false, false);
                                                J$.P(768633, J$.G(768617, J$.R(768609, 'server', server, false, false), 's'), 'sclusterTime', J$.R(768625, '$clusterTime', $clusterTime, false, false));
                                            }
                                            if (J$.C(44624, J$.M(768665, J$.G(768657, J$.G(768649, J$.R(768641, 'server', server, false, false), 's'), 'logger'), 'isInfo', false)())) {
                                                J$.M(768777, J$.G(768689, J$.G(768681, J$.R(768673, 'server', server, false, false), 's'), 'logger'), 'info', false)(J$.M(768769, J$.M(768729, J$.T(768697, 'server ', 21, false), 'concat', false)(J$.G(768713, J$.R(768705, 'server', server, false, false), 'name'), J$.T(768721, ' connected with ismaster [', 21, false)), 'concat', false)(J$.M(768753, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(768737, 'JSON', undefined, true, true) : JSON = J$.R(768737, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(768745, 'ismaster', ismaster, false, false)), J$.T(768761, ']', 21, false)));
                                            }
                                            J$.M(768849, J$.R(768785, 'server', server, false, false), 'emit', false)(J$.T(768793, 'descriptionReceived', 21, false), J$.F(768841, J$.R(768801, 'ServerDescription', ServerDescription, false, true), true)(J$.G(768825, J$.G(768817, J$.R(768809, 'server', server, false, false), 'description'), 'address'), J$.R(768833, 'ismaster', ismaster, false, false)));
                                            J$.P(768881, J$.G(768865, J$.R(768857, 'server', server, false, false), 's'), 'state', J$.R(768873, 'STATE_CONNECTED', STATE_CONNECTED, false, true));
                                            J$.M(768913, J$.R(768889, 'server', server, false, false), 'emit', false)(J$.T(768897, 'connect', 21, false), J$.R(768905, 'server', server, false, false));
                                        } catch (J$e) {
                                            J$.Ex(770449, J$e);
                                        } finally {
                                            if (J$.Fr(770457))
                                                continue jalangiLabel43;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(770465, J$e);
                        } finally {
                            if (J$.Fr(770473))
                                continue jalangiLabel44;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function errorEventHandler(server) {
                jalangiLabel46:
                    while (true) {
                        try {
                            J$.Fe(769129, arguments.callee, this, arguments);
                            arguments = J$.N(769137, 'arguments', arguments, true, false, false);
                            server = J$.N(769145, 'server', server, true, false, false);
                            return J$.Rt(769121, J$.T(769113, function (err) {
                                jalangiLabel45:
                                    while (true) {
                                        try {
                                            J$.Fe(769089, arguments.callee, this, arguments);
                                            arguments = J$.N(769097, 'arguments', arguments, true, false, false);
                                            err = J$.N(769105, 'err', err, true, false, false);
                                            if (J$.C(44632, J$.R(769009, 'err', err, false, false))) {
                                                J$.M(769057, J$.R(769017, 'server', server, false, false), 'emit', false)(J$.T(769025, 'error', 21, false), J$.F(769049, J$.R(769033, 'MongoNetworkError', MongoNetworkError, false, true), true)(J$.R(769041, 'err', err, false, false)));
                                            }
                                            J$.M(769081, J$.R(769065, 'server', server, false, false), 'emit', false)(J$.T(769073, 'close', 21, false));
                                        } catch (J$e) {
                                            J$.Ex(770481, J$e);
                                        } finally {
                                            if (J$.Fr(770489))
                                                continue jalangiLabel45;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(770497, J$e);
                        } finally {
                            if (J$.Fr(770505))
                                continue jalangiLabel46;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function parseErrorEventHandler(server) {
                jalangiLabel48:
                    while (true) {
                        try {
                            J$.Fe(769273, arguments.callee, this, arguments);
                            arguments = J$.N(769281, 'arguments', arguments, true, false, false);
                            server = J$.N(769289, 'server', server, true, false, false);
                            return J$.Rt(769265, J$.T(769257, function (err) {
                                jalangiLabel47:
                                    while (true) {
                                        try {
                                            J$.Fe(769233, arguments.callee, this, arguments);
                                            arguments = J$.N(769241, 'arguments', arguments, true, false, false);
                                            err = J$.N(769249, 'err', err, true, false, false);
                                            J$.P(769177, J$.G(769161, J$.R(769153, 'server', server, false, false), 's'), 'state', J$.R(769169, 'STATE_DISCONNECTED', STATE_DISCONNECTED, false, true));
                                            J$.M(769225, J$.R(769185, 'server', server, false, false), 'emit', false)(J$.T(769193, 'error', 21, false), J$.F(769217, J$.R(769201, 'MongoParseError', MongoParseError, false, true), true)(J$.R(769209, 'err', err, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(770513, J$e);
                                        } finally {
                                            if (J$.Fr(770521))
                                                continue jalangiLabel47;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(770529, J$e);
                        } finally {
                            if (J$.Fr(770537))
                                continue jalangiLabel48;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(769337, '_typeof2', J$.T(769329, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(769353, '_typeof', J$.T(769345, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(769369, '_classCallCheck', J$.T(769361, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(769385, '_defineProperties', J$.T(769377, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(769401, '_createClass', J$.T(769393, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(769417, '_possibleConstructorReturn', J$.T(769409, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(769433, '_assertThisInitialized', J$.T(769425, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(769449, '_getPrototypeOf', J$.T(769441, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(769465, '_inherits', J$.T(769457, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(769481, '_setPrototypeOf', J$.T(769473, _setPrototypeOf, 12, false), true, false, false);
            J$.N(769489, 'EventEmitter', EventEmitter, false, false, false);
            J$.N(769497, 'MongoError', MongoError, false, false, false);
            J$.N(769505, 'Pool', Pool, false, false, false);
            J$.N(769513, 'relayEvents', relayEvents, false, false, false);
            J$.N(769521, 'wireProtocol', wireProtocol, false, false, false);
            J$.N(769529, 'BSON', BSON, false, false, false);
            J$.N(769537, 'createClientInfo', createClientInfo, false, false, false);
            J$.N(769545, 'Logger', Logger, false, false, false);
            J$.N(769553, 'ServerDescription', ServerDescription, false, false, false);
            J$.N(769561, 'ReadPreference', ReadPreference, false, false, false);
            J$.N(769569, 'monitorServer', monitorServer, false, false, false);
            J$.N(769577, 'MongoParseError', MongoParseError, false, false, false);
            J$.N(769585, 'MongoNetworkError', MongoNetworkError, false, false, false);
            J$.N(769593, 'collationNotSupported', collationNotSupported, false, false, false);
            J$.N(769601, 'debugOptions', debugOptions, false, false, false);
            J$.N(769609, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, false, false);
            J$.N(769617, 'DEBUG_FIELDS', DEBUG_FIELDS, false, false, false);
            J$.N(769625, 'STATE_DISCONNECTING', STATE_DISCONNECTING, false, false, false);
            J$.N(769633, 'STATE_DISCONNECTED', STATE_DISCONNECTED, false, false, false);
            J$.N(769641, 'STATE_CONNECTING', STATE_CONNECTING, false, false, false);
            J$.N(769649, 'STATE_CONNECTED', STATE_CONNECTED, false, false, false);
            J$.N(769657, 'Server', Server, false, false, false);
            basicWriteValidations = J$.N(769673, 'basicWriteValidations', J$.T(769665, basicWriteValidations, 12, false), true, false, false);
            basicReadValidations = J$.N(769689, 'basicReadValidations', J$.T(769681, basicReadValidations, 12, false), true, false, false);
            executeWriteOperation = J$.N(769705, 'executeWriteOperation', J$.T(769697, executeWriteOperation, 12, false), true, false, false);
            connectEventHandler = J$.N(769721, 'connectEventHandler', J$.T(769713, connectEventHandler, 12, false), true, false, false);
            errorEventHandler = J$.N(769737, 'errorEventHandler', J$.T(769729, errorEventHandler, 12, false), true, false, false);
            parseErrorEventHandler = J$.N(769753, 'parseErrorEventHandler', J$.T(769745, parseErrorEventHandler, 12, false), true, false, false);
            J$.T(758865, 'use strict', 21, false);
            var EventEmitter = J$.W(760817, 'EventEmitter', J$.F(760809, J$.I(typeof require === 'undefined' ? require = J$.R(760793, 'require', undefined, true, true) : require = J$.R(760793, 'require', require, true, true)), false)(J$.T(760801, 'events', 21, false)), EventEmitter, false, true);
            var MongoError = J$.W(760857, 'MongoError', J$.G(760849, J$.F(760841, J$.I(typeof require === 'undefined' ? require = J$.R(760825, 'require', undefined, true, true) : require = J$.R(760825, 'require', require, true, true)), false)(J$.T(760833, '../error', 21, false)), 'MongoError'), MongoError, false, true);
            var Pool = J$.W(760889, 'Pool', J$.F(760881, J$.I(typeof require === 'undefined' ? require = J$.R(760865, 'require', undefined, true, true) : require = J$.R(760865, 'require', require, true, true)), false)(J$.T(760873, '../connection/pool', 21, false)), Pool, false, true);
            var relayEvents = J$.W(760929, 'relayEvents', J$.G(760921, J$.F(760913, J$.I(typeof require === 'undefined' ? require = J$.R(760897, 'require', undefined, true, true) : require = J$.R(760897, 'require', require, true, true)), false)(J$.T(760905, '../utils', 21, false)), 'relayEvents'), relayEvents, false, true);
            var wireProtocol = J$.W(760961, 'wireProtocol', J$.F(760953, J$.I(typeof require === 'undefined' ? require = J$.R(760937, 'require', undefined, true, true) : require = J$.R(760937, 'require', require, true, true)), false)(J$.T(760945, '../wireprotocol', 21, false)), wireProtocol, false, true);
            var BSON = J$.W(761001, 'BSON', J$.M(760993, J$.F(760985, J$.I(typeof require === 'undefined' ? require = J$.R(760969, 'require', undefined, true, true) : require = J$.R(760969, 'require', require, true, true)), false)(J$.T(760977, '../connection/utils', 21, false)), 'retrieveBSON', false)(), BSON, false, true);
            var createClientInfo = J$.W(761041, 'createClientInfo', J$.G(761033, J$.F(761025, J$.I(typeof require === 'undefined' ? require = J$.R(761009, 'require', undefined, true, true) : require = J$.R(761009, 'require', require, true, true)), false)(J$.T(761017, '../topologies/shared', 21, false)), 'createClientInfo'), createClientInfo, false, true);
            var Logger = J$.W(761073, 'Logger', J$.F(761065, J$.I(typeof require === 'undefined' ? require = J$.R(761049, 'require', undefined, true, true) : require = J$.R(761049, 'require', require, true, true)), false)(J$.T(761057, '../connection/logger', 21, false)), Logger, false, true);
            var ServerDescription = J$.W(761113, 'ServerDescription', J$.G(761105, J$.F(761097, J$.I(typeof require === 'undefined' ? require = J$.R(761081, 'require', undefined, true, true) : require = J$.R(761081, 'require', require, true, true)), false)(J$.T(761089, './server_description', 21, false)), 'ServerDescription'), ServerDescription, false, true);
            var ReadPreference = J$.W(761145, 'ReadPreference', J$.F(761137, J$.I(typeof require === 'undefined' ? require = J$.R(761121, 'require', undefined, true, true) : require = J$.R(761121, 'require', require, true, true)), false)(J$.T(761129, '../topologies/read_preference', 21, false)), ReadPreference, false, true);
            var monitorServer = J$.W(761185, 'monitorServer', J$.G(761177, J$.F(761169, J$.I(typeof require === 'undefined' ? require = J$.R(761153, 'require', undefined, true, true) : require = J$.R(761153, 'require', require, true, true)), false)(J$.T(761161, './monitoring', 21, false)), 'monitorServer'), monitorServer, false, true);
            var MongoParseError = J$.W(761225, 'MongoParseError', J$.G(761217, J$.F(761209, J$.I(typeof require === 'undefined' ? require = J$.R(761193, 'require', undefined, true, true) : require = J$.R(761193, 'require', require, true, true)), false)(J$.T(761201, '../error', 21, false)), 'MongoParseError'), MongoParseError, false, true);
            var MongoNetworkError = J$.W(761265, 'MongoNetworkError', J$.G(761257, J$.F(761249, J$.I(typeof require === 'undefined' ? require = J$.R(761233, 'require', undefined, true, true) : require = J$.R(761233, 'require', require, true, true)), false)(J$.T(761241, '../error', 21, false)), 'MongoNetworkError'), MongoNetworkError, false, true);
            var collationNotSupported = J$.W(761305, 'collationNotSupported', J$.G(761297, J$.F(761289, J$.I(typeof require === 'undefined' ? require = J$.R(761273, 'require', undefined, true, true) : require = J$.R(761273, 'require', require, true, true)), false)(J$.T(761281, '../utils', 21, false)), 'collationNotSupported'), collationNotSupported, false, true);
            var debugOptions = J$.W(761345, 'debugOptions', J$.G(761337, J$.F(761329, J$.I(typeof require === 'undefined' ? require = J$.R(761313, 'require', undefined, true, true) : require = J$.R(761313, 'require', require, true, true)), false)(J$.T(761321, '../connection/utils', 21, false)), 'debugOptions'), debugOptions, false, true);
            var isSDAMUnrecoverableError = J$.W(761385, 'isSDAMUnrecoverableError', J$.G(761377, J$.F(761369, J$.I(typeof require === 'undefined' ? require = J$.R(761353, 'require', undefined, true, true) : require = J$.R(761353, 'require', require, true, true)), false)(J$.T(761361, '../error', 21, false)), 'isSDAMUnrecoverableError'), isSDAMUnrecoverableError, false, true);
            var DEBUG_FIELDS = J$.W(761593, 'DEBUG_FIELDS', J$.T(761585, [
                J$.T(761393, 'reconnect', 21, false),
                J$.T(761401, 'reconnectTries', 21, false),
                J$.T(761409, 'reconnectInterval', 21, false),
                J$.T(761417, 'emitError', 21, false),
                J$.T(761425, 'cursorFactory', 21, false),
                J$.T(761433, 'host', 21, false),
                J$.T(761441, 'port', 21, false),
                J$.T(761449, 'size', 21, false),
                J$.T(761457, 'keepAlive', 21, false),
                J$.T(761465, 'keepAliveInitialDelay', 21, false),
                J$.T(761473, 'noDelay', 21, false),
                J$.T(761481, 'connectionTimeout', 21, false),
                J$.T(761489, 'checkServerIdentity', 21, false),
                J$.T(761497, 'socketTimeout', 21, false),
                J$.T(761505, 'ssl', 21, false),
                J$.T(761513, 'ca', 21, false),
                J$.T(761521, 'crl', 21, false),
                J$.T(761529, 'cert', 21, false),
                J$.T(761537, 'key', 21, false),
                J$.T(761545, 'rejectUnauthorized', 21, false),
                J$.T(761553, 'promoteLongs', 21, false),
                J$.T(761561, 'promoteValues', 21, false),
                J$.T(761569, 'promoteBuffers', 21, false),
                J$.T(761577, 'servername', 21, false)
            ], 10, false), DEBUG_FIELDS, false, true);
            var STATE_DISCONNECTING = J$.W(761609, 'STATE_DISCONNECTING', J$.T(761601, 'disconnecting', 21, false), STATE_DISCONNECTING, false, true);
            var STATE_DISCONNECTED = J$.W(761625, 'STATE_DISCONNECTED', J$.T(761617, 'disconnected', 21, false), STATE_DISCONNECTED, false, true);
            var STATE_CONNECTING = J$.W(761641, 'STATE_CONNECTING', J$.T(761633, 'connecting', 21, false), STATE_CONNECTING, false, true);
            var STATE_CONNECTED = J$.W(761657, 'STATE_CONNECTED', J$.T(761649, 'connected', 21, false), STATE_CONNECTED, false, true);
            var Server = J$.W(767033, 'Server', J$.F(767025, J$.T(767009, function (_EventEmitter) {
                jalangiLabel36:
                    while (true) {
                        try {
                            J$.Fe(766969, arguments.callee, this, arguments);
                            function Server(description, options, topology) {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(762249, arguments.callee, this, arguments);
                                            arguments = J$.N(762257, 'arguments', arguments, true, false, false);
                                            description = J$.N(762265, 'description', description, true, false, false);
                                            options = J$.N(762273, 'options', options, true, false, false);
                                            topology = J$.N(762281, 'topology', topology, true, false, false);
                                            J$.N(762289, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(761721, J$.R(761697, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(761705, 'this', this, false, false), J$.R(761713, 'Server', Server, false, false));
                                            _this = J$.W(761793, '_this', J$.F(761785, J$.R(761729, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(761737, 'this', this, false, false), J$.M(761777, J$.F(761761, J$.R(761745, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(761753, 'Server', Server, false, false)), 'call', false)(J$.R(761769, 'this', this, false, false))), _this, false, false);
                                            J$.P(762225, J$.R(761801, '_this', _this, false, false), 's', J$.T(762217, {
                                                description: J$.R(761809, 'description', description, false, false),
                                                options: J$.R(761817, 'options', options, false, false),
                                                logger: J$.F(761849, J$.R(761825, 'Logger', Logger, false, true), false)(J$.T(761833, 'Server', 21, false), J$.R(761841, 'options', options, false, false)),
                                                bson: J$.C(44192, J$.G(761865, J$.R(761857, 'options', options, false, false), 'bson')) ? J$._() : J$.F(762113, J$.R(761873, 'BSON', BSON, false, true), true)(J$.T(762105, [
                                                    J$.G(761889, J$.R(761881, 'BSON', BSON, false, true), 'Binary'),
                                                    J$.G(761905, J$.R(761897, 'BSON', BSON, false, true), 'Code'),
                                                    J$.G(761921, J$.R(761913, 'BSON', BSON, false, true), 'DBRef'),
                                                    J$.G(761937, J$.R(761929, 'BSON', BSON, false, true), 'Decimal128'),
                                                    J$.G(761953, J$.R(761945, 'BSON', BSON, false, true), 'Double'),
                                                    J$.G(761969, J$.R(761961, 'BSON', BSON, false, true), 'Int32'),
                                                    J$.G(761985, J$.R(761977, 'BSON', BSON, false, true), 'Long'),
                                                    J$.G(762001, J$.R(761993, 'BSON', BSON, false, true), 'Map'),
                                                    J$.G(762017, J$.R(762009, 'BSON', BSON, false, true), 'MaxKey'),
                                                    J$.G(762033, J$.R(762025, 'BSON', BSON, false, true), 'MinKey'),
                                                    J$.G(762049, J$.R(762041, 'BSON', BSON, false, true), 'ObjectId'),
                                                    J$.G(762065, J$.R(762057, 'BSON', BSON, false, true), 'BSONRegExp'),
                                                    J$.G(762081, J$.R(762073, 'BSON', BSON, false, true), 'Symbol'),
                                                    J$.G(762097, J$.R(762089, 'BSON', BSON, false, true), 'Timestamp')
                                                ], 10, false)),
                                                clientInfo: J$.F(762137, J$.R(762121, 'createClientInfo', createClientInfo, false, true), false)(J$.R(762129, 'options', options, false, false)),
                                                monitoring: J$.T(762145, false, 23, false),
                                                monitorFunction: J$.C(44200, J$.G(762161, J$.R(762153, 'options', options, false, false), 'monitorFunction')) ? J$._() : J$.R(762169, 'monitorServer', monitorServer, false, true),
                                                pool: J$.T(762177, null, 25, false),
                                                state: J$.R(762185, 'STATE_DISCONNECTED', STATE_DISCONNECTED, false, true),
                                                credentials: J$.G(762201, J$.R(762193, 'options', options, false, false), 'credentials'),
                                                topology: J$.R(762209, 'topology', topology, false, false)
                                            }, 11, false));
                                            return J$.Rt(762241, J$.R(762233, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(770017, J$e);
                                        } finally {
                                            if (J$.Fr(770025))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(766977, 'arguments', arguments, true, false, false);
                            _EventEmitter = J$.N(766985, '_EventEmitter', _EventEmitter, true, false, false);
                            Server = J$.N(767001, 'Server', J$.T(766993, Server, 12, false), true, false, false);
                            J$.F(761689, J$.R(761665, '_inherits', _inherits, false, true), false)(J$.R(761673, 'Server', Server, false, false), J$.R(761681, '_EventEmitter', _EventEmitter, false, false));
                            J$.F(766945, J$.R(762297, '_createClass', _createClass, false, true), false)(J$.R(762305, 'Server', Server, false, false), J$.T(766937, [
                                J$.T(763369, {
                                    key: J$.T(762313, 'connect', 21, false),
                                    value: J$.T(763361, function connect(options) {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(763321, arguments.callee, this, arguments);
                                                    arguments = J$.N(763329, 'arguments', arguments, true, false, false);
                                                    options = J$.N(763337, 'options', options, true, false, false);
                                                    J$.N(763345, 'addressParts', addressParts, false, false, false);
                                                    J$.N(763353, 'poolOptions', poolOptions, false, false, false);
                                                    options = J$.W(762337, 'options', J$.C(44208, J$.R(762321, 'options', options, false, false)) ? J$._() : J$.T(762329, {}, 11, false), options, false, false);
                                                    if (J$.C(44232, J$.C(44224, J$.C(44216, J$.G(762361, J$.G(762353, J$.R(762345, 'this', this, false, false), 's'), 'pool')) ? J$.U(89522, '!', J$.M(762393, J$.G(762385, J$.G(762377, J$.R(762369, 'this', this, false, false), 's'), 'pool'), 'isDisconnected', false)()) : J$._()) ? J$.U(89530, '!', J$.M(762425, J$.G(762417, J$.G(762409, J$.R(762401, 'this', this, false, false), 's'), 'pool'), 'isDestroyed', false)()) : J$._())) {
                                                        throw J$.F(762489, J$.R(762433, 'MongoError', MongoError, false, true), true)(J$.M(762481, J$.T(762441, 'Server instance in invalid state ', 21, false), 'concat', false)(J$.G(762473, J$.G(762465, J$.G(762457, J$.R(762449, 'this', this, false, false), 's'), 'pool'), 'state')));
                                                    }
                                                    var addressParts = J$.W(762537, 'addressParts', J$.M(762529, J$.G(762513, J$.G(762505, J$.R(762497, 'this', this, false, false), 'description'), 'address'), 'split', false)(J$.T(762521, ':', 21, false)), addressParts, false, false);
                                                    var poolOptions = J$.W(762705, 'poolOptions', J$.M(762697, J$.I(typeof Object === 'undefined' ? Object = J$.R(762545, 'Object', undefined, true, true) : Object = J$.R(762545, 'Object', Object, true, true)), 'assign', false)(J$.T(762625, {
                                                        host: J$.G(762569, J$.R(762553, 'addressParts', addressParts, false, false), J$.T(762561, 0, 22, false)),
                                                        port: J$.F(762617, J$.I(typeof parseInt === 'undefined' ? parseInt = J$.R(762577, 'parseInt', undefined, true, true) : parseInt = J$.R(762577, 'parseInt', parseInt, true, true)), false)(J$.G(762601, J$.R(762585, 'addressParts', addressParts, false, false), J$.T(762593, 1, 22, false)), J$.T(762609, 10, 22, false))
                                                    }, 11, false), J$.G(762649, J$.G(762641, J$.R(762633, 'this', this, false, false), 's'), 'options'), J$.R(762657, 'options', options, false, false), J$.T(762689, { bson: J$.G(762681, J$.G(762673, J$.R(762665, 'this', this, false, false), 's'), 'bson') }, 11, false)), poolOptions, false, false);
                                                    J$.P(762729, J$.R(762713, 'poolOptions', poolOptions, false, false), 'reconnect', J$.T(762721, true, 23, false));
                                                    J$.P(762753, J$.R(762737, 'poolOptions', poolOptions, false, false), 'legacyCompatMode', J$.T(762745, false, 23, false));
                                                    J$.P(762809, J$.G(762769, J$.R(762761, 'this', this, false, false), 's'), 'pool', J$.F(762801, J$.R(762777, 'Pool', Pool, false, true), true)(J$.R(762785, 'this', this, false, false), J$.R(762793, 'poolOptions', poolOptions, false, false)));
                                                    J$.M(762873, J$.G(762833, J$.G(762825, J$.R(762817, 'this', this, false, false), 's'), 'pool'), 'on', false)(J$.T(762841, 'connect', 21, false), J$.F(762865, J$.R(762849, 'connectEventHandler', connectEventHandler, false, true), false)(J$.R(762857, 'this', this, false, false)));
                                                    J$.M(762937, J$.G(762897, J$.G(762889, J$.R(762881, 'this', this, false, false), 's'), 'pool'), 'on', false)(J$.T(762905, 'close', 21, false), J$.F(762929, J$.R(762913, 'errorEventHandler', errorEventHandler, false, true), false)(J$.R(762921, 'this', this, false, false)));
                                                    J$.M(763001, J$.G(762961, J$.G(762953, J$.R(762945, 'this', this, false, false), 's'), 'pool'), 'on', false)(J$.T(762969, 'error', 21, false), J$.F(762993, J$.R(762977, 'errorEventHandler', errorEventHandler, false, true), false)(J$.R(762985, 'this', this, false, false)));
                                                    J$.M(763065, J$.G(763025, J$.G(763017, J$.R(763009, 'this', this, false, false), 's'), 'pool'), 'on', false)(J$.T(763033, 'parseError', 21, false), J$.F(763057, J$.R(763041, 'parseErrorEventHandler', parseErrorEventHandler, false, true), false)(J$.R(763049, 'this', this, false, false)));
                                                    J$.F(763145, J$.R(763073, 'relayEvents', relayEvents, false, true), false)(J$.G(763097, J$.G(763089, J$.R(763081, 'this', this, false, false), 's'), 'pool'), J$.R(763105, 'this', this, false, false), J$.T(763137, [
                                                        J$.T(763113, 'commandStarted', 21, false),
                                                        J$.T(763121, 'commandSucceeded', 21, false),
                                                        J$.T(763129, 'commandFailed', 21, false)
                                                    ], 10, false));
                                                    J$.P(763177, J$.G(763161, J$.R(763153, 'this', this, false, false), 's'), 'state', J$.R(763169, 'STATE_CONNECTING', STATE_CONNECTING, false, true));
                                                    if (J$.C(44240, J$.G(763193, J$.R(763185, 'options', options, false, false), 'auth'))) {
                                                        J$.M(763273, J$.G(763225, J$.G(763217, J$.G(763209, J$.R(763201, 'this', this, false, false), 's'), 'pool'), 'connect'), 'apply', false)(J$.G(763249, J$.G(763241, J$.R(763233, 'this', this, false, false), 's'), 'pool'), J$.G(763265, J$.R(763257, 'options', options, false, false), 'auth'));
                                                        return J$.Rt(763281, undefined);
                                                    }
                                                    J$.M(763313, J$.G(763305, J$.G(763297, J$.R(763289, 'this', this, false, false), 's'), 'pool'), 'connect', false)();
                                                } catch (J$e) {
                                                    J$.Ex(770033, J$e);
                                                } finally {
                                                    if (J$.Fr(770041))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(764033, {
                                    key: J$.T(763377, 'destroy', 21, false),
                                    value: J$.T(764025, function destroy(options, callback) {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(763977, arguments.callee, this, arguments);
                                                    arguments = J$.N(763985, 'arguments', arguments, true, false, false);
                                                    options = J$.N(763993, 'options', options, true, false, false);
                                                    callback = J$.N(764001, 'callback', callback, true, false, false);
                                                    J$.N(764009, '_this2', _this2, false, false, false);
                                                    J$.N(764017, 'done', done, false, false, false);
                                                    var _this2 = J$.W(763393, '_this2', J$.R(763385, 'this', this, false, false), _this2, false, false);
                                                    if (J$.C(44248, J$.B(89546, '===', J$.U(89538, 'typeof', J$.R(763401, 'options', options, false, false)), J$.T(763409, 'function', 21, false))))
                                                        callback = J$.W(763425, 'callback', J$.R(763417, 'options', options, false, false), callback, false, false), options = J$.W(763441, 'options', J$.T(763433, {}, 11, false), options, false, false);
                                                    options = J$.W(763497, 'options', J$.M(763489, J$.I(typeof Object === 'undefined' ? Object = J$.R(763449, 'Object', undefined, true, true) : Object = J$.R(763449, 'Object', Object, true, true)), 'assign', false)(J$.T(763457, {}, 11, false), J$.T(763473, { force: J$.T(763465, false, 23, false) }, 11, false), J$.R(763481, 'options', options, false, false)), options, false, false);
                                                    J$.P(763529, J$.G(763513, J$.R(763505, 'this', this, false, false), 's'), 'state', J$.R(763521, 'STATE_DISCONNECTING', STATE_DISCONNECTING, false, true));
                                                    var done = J$.W(763673, 'done', J$.T(763665, function done(err) {
                                                        jalangiLabel18:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(763641, arguments.callee, this, arguments);
                                                                    arguments = J$.N(763649, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(763657, 'err', err, true, false, false);
                                                                    J$.M(763553, J$.R(763537, '_this2', _this2, false, false), 'emit', false)(J$.T(763545, 'closed', 21, false));
                                                                    J$.P(763585, J$.G(763569, J$.R(763561, '_this2', _this2, false, false), 's'), 'state', J$.R(763577, 'STATE_DISCONNECTED', STATE_DISCONNECTED, false, true));
                                                                    if (J$.C(44256, J$.B(89562, '===', J$.U(89554, 'typeof', J$.R(763593, 'callback', callback, false, false)), J$.T(763601, 'function', 21, false)))) {
                                                                        J$.F(763633, J$.R(763609, 'callback', callback, false, false), false)(J$.R(763617, 'err', err, false, false), J$.T(763625, null, 25, false));
                                                                    }
                                                                } catch (J$e) {
                                                                    J$.Ex(770049, J$e);
                                                                } finally {
                                                                    if (J$.Fr(770057))
                                                                        continue jalangiLabel18;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false), done, false, false);
                                                    if (J$.C(44264, J$.U(89570, '!', J$.G(763697, J$.G(763689, J$.R(763681, 'this', this, false, false), 's'), 'pool')))) {
                                                        return J$.Rt(763721, J$.F(763713, J$.R(763705, 'done', done, false, false), false)());
                                                    }
                                                    J$.M(763849, J$.T(763769, [
                                                        J$.T(763729, 'close', 21, false),
                                                        J$.T(763737, 'error', 21, false),
                                                        J$.T(763745, 'timeout', 21, false),
                                                        J$.T(763753, 'parseError', 21, false),
                                                        J$.T(763761, 'connect', 21, false)
                                                    ], 10, false), 'forEach', false)(J$.T(763841, function (event) {
                                                        jalangiLabel19:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(763817, arguments.callee, this, arguments);
                                                                    arguments = J$.N(763825, 'arguments', arguments, true, false, false);
                                                                    event = J$.N(763833, 'event', event, true, false, false);
                                                                    J$.M(763809, J$.G(763793, J$.G(763785, J$.R(763777, '_this2', _this2, false, false), 's'), 'pool'), 'removeAllListeners', false)(J$.R(763801, 'event', event, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(770065, J$e);
                                                                } finally {
                                                                    if (J$.Fr(770073))
                                                                        continue jalangiLabel19;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                    if (J$.C(44272, J$.G(763873, J$.G(763865, J$.R(763857, 'this', this, false, false), 's'), 'monitorId'))) {
                                                        J$.F(763913, J$.I(typeof clearTimeout === 'undefined' ? clearTimeout = J$.R(763881, 'clearTimeout', undefined, true, true) : clearTimeout = J$.R(763881, 'clearTimeout', clearTimeout, true, true)), false)(J$.G(763905, J$.G(763897, J$.R(763889, 'this', this, false, false), 's'), 'monitorId'));
                                                    }
                                                    J$.M(763969, J$.G(763937, J$.G(763929, J$.R(763921, 'this', this, false, false), 's'), 'pool'), 'destroy', false)(J$.G(763953, J$.R(763945, 'options', options, false, false), 'force'), J$.R(763961, 'done', done, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(770081, J$e);
                                                } finally {
                                                    if (J$.Fr(770089))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(764273, {
                                    key: J$.T(764041, 'monitor', 21, false),
                                    value: J$.T(764265, function monitor(options) {
                                        jalangiLabel21:
                                            while (true) {
                                                try {
                                                    J$.Fe(764241, arguments.callee, this, arguments);
                                                    arguments = J$.N(764249, 'arguments', arguments, true, false, false);
                                                    options = J$.N(764257, 'options', options, true, false, false);
                                                    options = J$.W(764065, 'options', J$.C(44280, J$.R(764049, 'options', options, false, false)) ? J$._() : J$.T(764057, {}, 11, false), options, false, false);
                                                    if (J$.C(44296, J$.C(44288, J$.B(89578, '!==', J$.G(764089, J$.G(764081, J$.R(764073, 'this', this, false, false), 's'), 'state'), J$.R(764097, 'STATE_CONNECTED', STATE_CONNECTED, false, true))) ? J$._() : J$.G(764121, J$.G(764113, J$.R(764105, 'this', this, false, false), 's'), 'monitoring')))
                                                        return J$.Rt(764129, undefined);
                                                    if (J$.C(44304, J$.G(764153, J$.G(764145, J$.R(764137, 'this', this, false, false), 's'), 'monitorId')))
                                                        J$.F(764193, J$.I(typeof clearTimeout === 'undefined' ? clearTimeout = J$.R(764161, 'clearTimeout', undefined, true, true) : clearTimeout = J$.R(764161, 'clearTimeout', clearTimeout, true, true)), false)(J$.G(764185, J$.G(764177, J$.R(764169, 'this', this, false, false), 's'), 'monitorId'));
                                                    J$.M(764233, J$.G(764209, J$.R(764201, 'this', this, false, false), 's'), 'monitorFunction', false)(J$.R(764217, 'this', this, false, false), J$.R(764225, 'options', options, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(770097, J$e);
                                                } finally {
                                                    if (J$.Fr(770105))
                                                        continue jalangiLabel21;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(765153, {
                                    key: J$.T(764281, 'command', 21, false),
                                    value: J$.T(765145, function command(ns, cmd, options, callback) {
                                        jalangiLabel23:
                                            while (true) {
                                                try {
                                                    J$.Fe(765081, arguments.callee, this, arguments);
                                                    arguments = J$.N(765089, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(765097, 'ns', ns, true, false, false);
                                                    cmd = J$.N(765105, 'cmd', cmd, true, false, false);
                                                    options = J$.N(765113, 'options', options, true, false, false);
                                                    callback = J$.N(765121, 'callback', callback, true, false, false);
                                                    J$.N(765129, '_this3', _this3, false, false, false);
                                                    J$.N(765137, 'error', error, false, false, false);
                                                    var _this3 = J$.W(764297, '_this3', J$.R(764289, 'this', this, false, false), _this3, false, false);
                                                    if (J$.C(44320, J$.B(89594, '===', J$.U(89586, 'typeof', J$.R(764305, 'options', options, false, false)), J$.T(764313, 'function', 21, false)))) {
                                                        callback = J$.W(764329, 'callback', J$.R(764321, 'options', options, false, false), callback, false, false), options = J$.W(764345, 'options', J$.T(764337, {}, 11, false), options, false, false), options = J$.W(764369, 'options', J$.C(44312, J$.R(764353, 'options', options, false, false)) ? J$._() : J$.T(764361, {}, 11, false), options, false, false);
                                                    }
                                                    var error = J$.W(764409, 'error', J$.F(764401, J$.R(764377, 'basicReadValidations', basicReadValidations, false, true), false)(J$.R(764385, 'this', this, false, false), J$.R(764393, 'options', options, false, false)), error, false, false);
                                                    if (J$.C(44328, J$.R(764417, 'error', error, false, false))) {
                                                        return J$.Rt(764457, J$.F(764449, J$.R(764425, 'callback', callback, false, false), false)(J$.R(764433, 'error', error, false, false), J$.T(764441, null, 25, false)));
                                                    }
                                                    options = J$.W(764513, 'options', J$.M(764505, J$.I(typeof Object === 'undefined' ? Object = J$.R(764465, 'Object', undefined, true, true) : Object = J$.R(764465, 'Object', Object, true, true)), 'assign', false)(J$.T(764473, {}, 11, false), J$.R(764481, 'options', options, false, false), J$.T(764497, { wireProtocolCommand: J$.T(764489, false, 23, false) }, 11, false)), options, false, false);
                                                    if (J$.C(44336, J$.M(764545, J$.G(764537, J$.G(764529, J$.R(764521, 'this', this, false, false), 's'), 'logger'), 'isDebug', false)())) {
                                                        J$.M(764697, J$.G(764569, J$.G(764561, J$.R(764553, 'this', this, false, false), 's'), 'logger'), 'debug', false)(J$.M(764689, J$.M(764665, J$.T(764577, 'executing command [', 21, false), 'concat', false)(J$.M(764649, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(764585, 'JSON', undefined, true, true) : JSON = J$.R(764585, 'JSON', JSON, true, true)), 'stringify', false)(J$.T(764641, {
                                                            ns: J$.R(764593, 'ns', ns, false, false),
                                                            cmd: J$.R(764601, 'cmd', cmd, false, false),
                                                            options: J$.F(764633, J$.R(764609, 'debugOptions', debugOptions, false, true), false)(J$.R(764617, 'DEBUG_FIELDS', DEBUG_FIELDS, false, true), J$.R(764625, 'options', options, false, false))
                                                        }, 11, false)), J$.T(764657, '] against ', 21, false)), 'concat', false)(J$.G(764681, J$.R(764673, 'this', this, false, false), 'name')));
                                                    }
                                                    if (J$.C(44344, J$.F(764729, J$.R(764705, 'collationNotSupported', collationNotSupported, false, true), false)(J$.R(764713, 'this', this, false, false), J$.R(764721, 'cmd', cmd, false, false)))) {
                                                        J$.F(764801, J$.R(764737, 'callback', callback, false, false), false)(J$.F(764793, J$.R(764745, 'MongoError', MongoError, false, true), true)(J$.M(764785, J$.T(764753, 'server ', 21, false), 'concat', false)(J$.G(764769, J$.R(764761, 'this', this, false, false), 'name'), J$.T(764777, ' does not support collation', 21, false))));
                                                        return J$.Rt(764809, undefined);
                                                    }
                                                    J$.M(765073, J$.R(764817, 'wireProtocol', wireProtocol, false, true), 'command', false)(J$.R(764825, 'this', this, false, false), J$.R(764833, 'ns', ns, false, false), J$.R(764841, 'cmd', cmd, false, false), J$.R(764849, 'options', options, false, false), J$.T(765065, function (err, result) {
                                                        jalangiLabel22:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(765033, arguments.callee, this, arguments);
                                                                    arguments = J$.N(765041, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(765049, 'err', err, true, false, false);
                                                                    result = J$.N(765057, 'result', result, true, false, false);
                                                                    if (J$.C(44376, J$.R(764857, 'err', err, false, false))) {
                                                                        if (J$.C(44360, J$.C(44352, J$.G(764873, J$.R(764865, 'options', options, false, false), 'session')) ? J$.B(89602, 'instanceof', J$.R(764881, 'err', err, false, false), J$.R(764889, 'MongoNetworkError', MongoNetworkError, false, true)) : J$._())) {
                                                                            J$.P(764929, J$.G(764913, J$.G(764905, J$.R(764897, 'options', options, false, false), 'session'), 'serverSession'), 'isDirty', J$.T(764921, true, 23, false));
                                                                        }
                                                                        if (J$.C(44368, J$.F(764961, J$.R(764937, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true), false)(J$.R(764945, 'err', err, false, false), J$.R(764953, '_this3', _this3, false, false)))) {
                                                                            J$.M(764993, J$.R(764969, '_this3', _this3, false, false), 'emit', false)(J$.T(764977, 'error', 21, false), J$.R(764985, 'err', err, false, false));
                                                                        }
                                                                    }
                                                                    J$.F(765025, J$.R(765001, 'callback', callback, false, false), false)(J$.R(765009, 'err', err, false, false), J$.R(765017, 'result', result, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(770113, J$e);
                                                                } finally {
                                                                    if (J$.Fr(770121))
                                                                        continue jalangiLabel22;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(770129, J$e);
                                                } finally {
                                                    if (J$.Fr(770137))
                                                        continue jalangiLabel23;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(765529, {
                                    key: J$.T(765161, 'query', 21, false),
                                    value: J$.T(765521, function query(ns, cmd, cursorState, options, callback) {
                                        jalangiLabel25:
                                            while (true) {
                                                try {
                                                    J$.Fe(765457, arguments.callee, this, arguments);
                                                    arguments = J$.N(765465, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(765473, 'ns', ns, true, false, false);
                                                    cmd = J$.N(765481, 'cmd', cmd, true, false, false);
                                                    cursorState = J$.N(765489, 'cursorState', cursorState, true, false, false);
                                                    options = J$.N(765497, 'options', options, true, false, false);
                                                    callback = J$.N(765505, 'callback', callback, true, false, false);
                                                    J$.N(765513, '_this4', _this4, false, false, false);
                                                    var _this4 = J$.W(765177, '_this4', J$.R(765169, 'this', this, false, false), _this4, false, false);
                                                    J$.M(765449, J$.R(765185, 'wireProtocol', wireProtocol, false, true), 'query', false)(J$.R(765193, 'this', this, false, false), J$.R(765201, 'ns', ns, false, false), J$.R(765209, 'cmd', cmd, false, false), J$.R(765217, 'cursorState', cursorState, false, false), J$.R(765225, 'options', options, false, false), J$.T(765441, function (err, result) {
                                                        jalangiLabel24:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(765409, arguments.callee, this, arguments);
                                                                    arguments = J$.N(765417, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(765425, 'err', err, true, false, false);
                                                                    result = J$.N(765433, 'result', result, true, false, false);
                                                                    if (J$.C(44408, J$.R(765233, 'err', err, false, false))) {
                                                                        if (J$.C(44392, J$.C(44384, J$.G(765249, J$.R(765241, 'options', options, false, false), 'session')) ? J$.B(89610, 'instanceof', J$.R(765257, 'err', err, false, false), J$.R(765265, 'MongoNetworkError', MongoNetworkError, false, true)) : J$._())) {
                                                                            J$.P(765305, J$.G(765289, J$.G(765281, J$.R(765273, 'options', options, false, false), 'session'), 'serverSession'), 'isDirty', J$.T(765297, true, 23, false));
                                                                        }
                                                                        if (J$.C(44400, J$.F(765337, J$.R(765313, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true), false)(J$.R(765321, 'err', err, false, false), J$.R(765329, '_this4', _this4, false, false)))) {
                                                                            J$.M(765369, J$.R(765345, '_this4', _this4, false, false), 'emit', false)(J$.T(765353, 'error', 21, false), J$.R(765361, 'err', err, false, false));
                                                                        }
                                                                    }
                                                                    J$.F(765401, J$.R(765377, 'callback', callback, false, false), false)(J$.R(765385, 'err', err, false, false), J$.R(765393, 'result', result, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(770145, J$e);
                                                                } finally {
                                                                    if (J$.Fr(770153))
                                                                        continue jalangiLabel24;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(770161, J$e);
                                                } finally {
                                                    if (J$.Fr(770169))
                                                        continue jalangiLabel25;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(765905, {
                                    key: J$.T(765537, 'getMore', 21, false),
                                    value: J$.T(765897, function getMore(ns, cursorState, batchSize, options, callback) {
                                        jalangiLabel27:
                                            while (true) {
                                                try {
                                                    J$.Fe(765833, arguments.callee, this, arguments);
                                                    arguments = J$.N(765841, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(765849, 'ns', ns, true, false, false);
                                                    cursorState = J$.N(765857, 'cursorState', cursorState, true, false, false);
                                                    batchSize = J$.N(765865, 'batchSize', batchSize, true, false, false);
                                                    options = J$.N(765873, 'options', options, true, false, false);
                                                    callback = J$.N(765881, 'callback', callback, true, false, false);
                                                    J$.N(765889, '_this5', _this5, false, false, false);
                                                    var _this5 = J$.W(765553, '_this5', J$.R(765545, 'this', this, false, false), _this5, false, false);
                                                    J$.M(765825, J$.R(765561, 'wireProtocol', wireProtocol, false, true), 'getMore', false)(J$.R(765569, 'this', this, false, false), J$.R(765577, 'ns', ns, false, false), J$.R(765585, 'cursorState', cursorState, false, false), J$.R(765593, 'batchSize', batchSize, false, false), J$.R(765601, 'options', options, false, false), J$.T(765817, function (err, result) {
                                                        jalangiLabel26:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(765785, arguments.callee, this, arguments);
                                                                    arguments = J$.N(765793, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(765801, 'err', err, true, false, false);
                                                                    result = J$.N(765809, 'result', result, true, false, false);
                                                                    if (J$.C(44440, J$.R(765609, 'err', err, false, false))) {
                                                                        if (J$.C(44424, J$.C(44416, J$.G(765625, J$.R(765617, 'options', options, false, false), 'session')) ? J$.B(89618, 'instanceof', J$.R(765633, 'err', err, false, false), J$.R(765641, 'MongoNetworkError', MongoNetworkError, false, true)) : J$._())) {
                                                                            J$.P(765681, J$.G(765665, J$.G(765657, J$.R(765649, 'options', options, false, false), 'session'), 'serverSession'), 'isDirty', J$.T(765673, true, 23, false));
                                                                        }
                                                                        if (J$.C(44432, J$.F(765713, J$.R(765689, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true), false)(J$.R(765697, 'err', err, false, false), J$.R(765705, '_this5', _this5, false, false)))) {
                                                                            J$.M(765745, J$.R(765721, '_this5', _this5, false, false), 'emit', false)(J$.T(765729, 'error', 21, false), J$.R(765737, 'err', err, false, false));
                                                                        }
                                                                    }
                                                                    J$.F(765777, J$.R(765753, 'callback', callback, false, false), false)(J$.R(765761, 'err', err, false, false), J$.R(765769, 'result', result, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(770177, J$e);
                                                                } finally {
                                                                    if (J$.Fr(770185))
                                                                        continue jalangiLabel26;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(770193, J$e);
                                                } finally {
                                                    if (J$.Fr(770201))
                                                        continue jalangiLabel27;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(766193, {
                                    key: J$.T(765913, 'killCursors', 21, false),
                                    value: J$.T(766185, function killCursors(ns, cursorState, callback) {
                                        jalangiLabel29:
                                            while (true) {
                                                try {
                                                    J$.Fe(766137, arguments.callee, this, arguments);
                                                    arguments = J$.N(766145, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(766153, 'ns', ns, true, false, false);
                                                    cursorState = J$.N(766161, 'cursorState', cursorState, true, false, false);
                                                    callback = J$.N(766169, 'callback', callback, true, false, false);
                                                    J$.N(766177, '_this6', _this6, false, false, false);
                                                    var _this6 = J$.W(765929, '_this6', J$.R(765921, 'this', this, false, false), _this6, false, false);
                                                    J$.M(766129, J$.R(765937, 'wireProtocol', wireProtocol, false, true), 'killCursors', false)(J$.R(765945, 'this', this, false, false), J$.R(765953, 'ns', ns, false, false), J$.R(765961, 'cursorState', cursorState, false, false), J$.T(766121, function (err, result) {
                                                        jalangiLabel28:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(766089, arguments.callee, this, arguments);
                                                                    arguments = J$.N(766097, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(766105, 'err', err, true, false, false);
                                                                    result = J$.N(766113, 'result', result, true, false, false);
                                                                    if (J$.C(44456, J$.C(44448, J$.R(765969, 'err', err, false, false)) ? J$.F(766001, J$.R(765977, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true), false)(J$.R(765985, 'err', err, false, false), J$.R(765993, '_this6', _this6, false, false)) : J$._())) {
                                                                        J$.M(766033, J$.R(766009, '_this6', _this6, false, false), 'emit', false)(J$.T(766017, 'error', 21, false), J$.R(766025, 'err', err, false, false));
                                                                    }
                                                                    if (J$.C(44464, J$.B(89634, '===', J$.U(89626, 'typeof', J$.R(766041, 'callback', callback, false, false)), J$.T(766049, 'function', 21, false)))) {
                                                                        J$.F(766081, J$.R(766057, 'callback', callback, false, false), false)(J$.R(766065, 'err', err, false, false), J$.R(766073, 'result', result, false, false));
                                                                    }
                                                                } catch (J$e) {
                                                                    J$.Ex(770209, J$e);
                                                                } finally {
                                                                    if (J$.Fr(770217))
                                                                        continue jalangiLabel28;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(770225, J$e);
                                                } finally {
                                                    if (J$.Fr(770233))
                                                        continue jalangiLabel29;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(766337, {
                                    key: J$.T(766201, 'insert', 21, false),
                                    value: J$.T(766329, function insert(ns, ops, options, callback) {
                                        jalangiLabel30:
                                            while (true) {
                                                try {
                                                    J$.Fe(766281, arguments.callee, this, arguments);
                                                    arguments = J$.N(766289, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(766297, 'ns', ns, true, false, false);
                                                    ops = J$.N(766305, 'ops', ops, true, false, false);
                                                    options = J$.N(766313, 'options', options, true, false, false);
                                                    callback = J$.N(766321, 'callback', callback, true, false, false);
                                                    J$.F(766273, J$.R(766209, 'executeWriteOperation', executeWriteOperation, false, true), false)(J$.T(766249, {
                                                        server: J$.R(766217, 'this', this, false, false),
                                                        op: J$.T(766225, 'insert', 21, false),
                                                        ns: J$.R(766233, 'ns', ns, false, false),
                                                        ops: J$.R(766241, 'ops', ops, false, false)
                                                    }, 11, false), J$.R(766257, 'options', options, false, false), J$.R(766265, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(770241, J$e);
                                                } finally {
                                                    if (J$.Fr(770249))
                                                        continue jalangiLabel30;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(766481, {
                                    key: J$.T(766345, 'update', 21, false),
                                    value: J$.T(766473, function update(ns, ops, options, callback) {
                                        jalangiLabel31:
                                            while (true) {
                                                try {
                                                    J$.Fe(766425, arguments.callee, this, arguments);
                                                    arguments = J$.N(766433, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(766441, 'ns', ns, true, false, false);
                                                    ops = J$.N(766449, 'ops', ops, true, false, false);
                                                    options = J$.N(766457, 'options', options, true, false, false);
                                                    callback = J$.N(766465, 'callback', callback, true, false, false);
                                                    J$.F(766417, J$.R(766353, 'executeWriteOperation', executeWriteOperation, false, true), false)(J$.T(766393, {
                                                        server: J$.R(766361, 'this', this, false, false),
                                                        op: J$.T(766369, 'update', 21, false),
                                                        ns: J$.R(766377, 'ns', ns, false, false),
                                                        ops: J$.R(766385, 'ops', ops, false, false)
                                                    }, 11, false), J$.R(766401, 'options', options, false, false), J$.R(766409, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(770257, J$e);
                                                } finally {
                                                    if (J$.Fr(770265))
                                                        continue jalangiLabel31;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(766625, {
                                    key: J$.T(766489, 'remove', 21, false),
                                    value: J$.T(766617, function remove(ns, ops, options, callback) {
                                        jalangiLabel32:
                                            while (true) {
                                                try {
                                                    J$.Fe(766569, arguments.callee, this, arguments);
                                                    arguments = J$.N(766577, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(766585, 'ns', ns, true, false, false);
                                                    ops = J$.N(766593, 'ops', ops, true, false, false);
                                                    options = J$.N(766601, 'options', options, true, false, false);
                                                    callback = J$.N(766609, 'callback', callback, true, false, false);
                                                    J$.F(766561, J$.R(766497, 'executeWriteOperation', executeWriteOperation, false, true), false)(J$.T(766537, {
                                                        server: J$.R(766505, 'this', this, false, false),
                                                        op: J$.T(766513, 'remove', 21, false),
                                                        ns: J$.R(766521, 'ns', ns, false, false),
                                                        ops: J$.R(766529, 'ops', ops, false, false)
                                                    }, 11, false), J$.R(766545, 'options', options, false, false), J$.R(766553, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(770273, J$e);
                                                } finally {
                                                    if (J$.Fr(770281))
                                                        continue jalangiLabel32;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(766697, {
                                    key: J$.T(766633, 'description', 21, false),
                                    get: J$.T(766689, function get() {
                                        jalangiLabel33:
                                            while (true) {
                                                try {
                                                    J$.Fe(766673, arguments.callee, this, arguments);
                                                    arguments = J$.N(766681, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(766665, J$.G(766657, J$.G(766649, J$.R(766641, 'this', this, false, false), 's'), 'description'));
                                                } catch (J$e) {
                                                    J$.Ex(770289, J$e);
                                                } finally {
                                                    if (J$.Fr(770297))
                                                        continue jalangiLabel33;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(766777, {
                                    key: J$.T(766705, 'name', 21, false),
                                    get: J$.T(766769, function get() {
                                        jalangiLabel34:
                                            while (true) {
                                                try {
                                                    J$.Fe(766753, arguments.callee, this, arguments);
                                                    arguments = J$.N(766761, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(766745, J$.G(766737, J$.G(766729, J$.G(766721, J$.R(766713, 'this', this, false, false), 's'), 'description'), 'address'));
                                                } catch (J$e) {
                                                    J$.Ex(770305, J$e);
                                                } finally {
                                                    if (J$.Fr(770313))
                                                        continue jalangiLabel34;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(766929, {
                                    key: J$.T(766785, 'autoEncrypter', 21, false),
                                    get: J$.T(766921, function get() {
                                        jalangiLabel35:
                                            while (true) {
                                                try {
                                                    J$.Fe(766905, arguments.callee, this, arguments);
                                                    arguments = J$.N(766913, 'arguments', arguments, true, false, false);
                                                    if (J$.C(44480, J$.C(44472, J$.G(766809, J$.G(766801, J$.R(766793, 'this', this, false, false), 's'), 'options')) ? J$.G(766841, J$.G(766833, J$.G(766825, J$.R(766817, 'this', this, false, false), 's'), 'options'), 'autoEncrypter') : J$._())) {
                                                        return J$.Rt(766881, J$.G(766873, J$.G(766865, J$.G(766857, J$.R(766849, 'this', this, false, false), 's'), 'options'), 'autoEncrypter'));
                                                    }
                                                    return J$.Rt(766897, J$.T(766889, null, 25, false));
                                                } catch (J$e) {
                                                    J$.Ex(770321, J$e);
                                                } finally {
                                                    if (J$.Fr(770329))
                                                        continue jalangiLabel35;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(766961, J$.R(766953, 'Server', Server, false, false));
                        } catch (J$e) {
                            J$.Ex(770337, J$e);
                        } finally {
                            if (J$.Fr(770345))
                                continue jalangiLabel36;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(767017, 'EventEmitter', EventEmitter, false, true)), Server, false, true);
            J$.M(767217, J$.I(typeof Object === 'undefined' ? Object = J$.R(767041, 'Object', undefined, true, true) : Object = J$.R(767041, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(767057, J$.R(767049, 'Server', Server, false, true), 'prototype'), J$.T(767065, 'clusterTime', 21, false), J$.T(767209, {
                get: J$.T(767129, function get() {
                    jalangiLabel37:
                        while (true) {
                            try {
                                J$.Fe(767113, arguments.callee, this, arguments);
                                arguments = J$.N(767121, 'arguments', arguments, true, false, false);
                                return J$.Rt(767105, J$.G(767097, J$.G(767089, J$.G(767081, J$.R(767073, 'this', this, false, false), 's'), 'topology'), 'clusterTime'));
                            } catch (J$e) {
                                J$.Ex(770353, J$e);
                            } finally {
                                if (J$.Fr(770361))
                                    continue jalangiLabel37;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                set: J$.T(767201, function set(clusterTime) {
                    jalangiLabel38:
                        while (true) {
                            try {
                                J$.Fe(767177, arguments.callee, this, arguments);
                                arguments = J$.N(767185, 'arguments', arguments, true, false, false);
                                clusterTime = J$.N(767193, 'clusterTime', clusterTime, true, false, false);
                                J$.P(767169, J$.G(767153, J$.G(767145, J$.R(767137, 'this', this, false, false), 's'), 'topology'), 'clusterTime', J$.R(767161, 'clusterTime', clusterTime, false, false));
                            } catch (J$e) {
                                J$.Ex(770369, J$e);
                            } finally {
                                if (J$.Fr(770377))
                                    continue jalangiLabel38;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.P(769313, J$.I(typeof module === 'undefined' ? module = J$.R(769297, 'module', undefined, true, true) : module = J$.R(769297, 'module', module, true, true)), 'exports', J$.R(769305, 'Server', Server, false, true));
        } catch (J$e) {
            J$.Ex(770545, J$e);
        } finally {
            if (J$.Sr(770553))
                continue jalangiLabel49;
            else
                break jalangiLabel49;
        }
    }
// JALANGI DO NOT INSTRUMENT
