// mongodb/lib/core/sdam/server_selectors.js
J$.noInstrEval = false;
jalangiLabel26:
    while (true) {
        try {
            J$.Se(778649, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/sdam/server_selectors.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(774025, arguments.callee, this, arguments);
                            arguments = J$.N(774033, 'arguments', arguments, true, false, false);
                            obj = J$.N(774041, 'obj', obj, true, false, false);
                            if (J$.C(44872, J$.C(44832, J$.B(89802, '===', J$.U(89794, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(773761, 'Symbol', undefined, true, true) : Symbol = J$.R(773761, 'Symbol', Symbol, true, true))), J$.T(773769, 'function', 21, false))) ? J$.B(89818, '===', J$.U(89810, 'typeof', J$.G(773785, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(773777, 'Symbol', undefined, true, true) : Symbol = J$.R(773777, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(773793, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(773849, '_typeof2', J$.T(773841, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(773817, arguments.callee, this, arguments);
                                                arguments = J$.N(773825, 'arguments', arguments, true, false, false);
                                                obj = J$.N(773833, 'obj', obj, true, false, false);
                                                return J$.Rt(773809, J$.U(89826, 'typeof', J$.R(773801, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(778897, J$e);
                                            } finally {
                                                if (J$.Fr(778905))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(773985, '_typeof2', J$.T(773977, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(773953, arguments.callee, this, arguments);
                                                arguments = J$.N(773961, 'arguments', arguments, true, false, false);
                                                obj = J$.N(773969, 'obj', obj, true, false, false);
                                                return J$.Rt(773945, J$.C(44864, J$.C(44856, J$.C(44848, J$.C(44840, J$.R(773857, 'obj', obj, false, false)) ? J$.B(89842, '===', J$.U(89834, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(773865, 'Symbol', undefined, true, true) : Symbol = J$.R(773865, 'Symbol', Symbol, true, true))), J$.T(773873, 'function', 21, false)) : J$._()) ? J$.B(89850, '===', J$.G(773889, J$.R(773881, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(773897, 'Symbol', undefined, true, true) : Symbol = J$.R(773897, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(89858, '!==', J$.R(773905, 'obj', obj, false, false), J$.G(773921, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(773913, 'Symbol', undefined, true, true) : Symbol = J$.R(773913, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(773929, 'symbol', 21, false) : J$.U(89866, 'typeof', J$.R(773937, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(778913, J$e);
                                            } finally {
                                                if (J$.Fr(778921))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(774017, J$.F(774009, J$.R(773993, '_typeof2', _typeof2, false, true), false)(J$.R(774001, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(778929, J$e);
                        } finally {
                            if (J$.Fr(778937))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(774361, arguments.callee, this, arguments);
                            arguments = J$.N(774369, 'arguments', arguments, true, false, false);
                            obj = J$.N(774377, 'obj', obj, true, false, false);
                            if (J$.C(44920, J$.C(44880, J$.B(89882, '===', J$.U(89874, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(774049, 'Symbol', undefined, true, true) : Symbol = J$.R(774049, 'Symbol', Symbol, true, true))), J$.T(774057, 'function', 21, false))) ? J$.B(89890, '===', J$.F(774089, J$.R(774065, '_typeof2', _typeof2, false, true), false)(J$.G(774081, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(774073, 'Symbol', undefined, true, true) : Symbol = J$.R(774073, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(774097, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(774169, '_typeof', J$.T(774161, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(774137, arguments.callee, this, arguments);
                                                arguments = J$.N(774145, 'arguments', arguments, true, false, false);
                                                obj = J$.N(774153, 'obj', obj, true, false, false);
                                                return J$.Rt(774129, J$.F(774121, J$.R(774105, '_typeof2', _typeof2, false, true), false)(J$.R(774113, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(778945, J$e);
                                            } finally {
                                                if (J$.Fr(778953))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(774321, '_typeof', J$.T(774313, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(774289, arguments.callee, this, arguments);
                                                arguments = J$.N(774297, 'arguments', arguments, true, false, false);
                                                obj = J$.N(774305, 'obj', obj, true, false, false);
                                                return J$.Rt(774281, J$.C(44912, J$.C(44904, J$.C(44896, J$.C(44888, J$.R(774177, 'obj', obj, false, false)) ? J$.B(89906, '===', J$.U(89898, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(774185, 'Symbol', undefined, true, true) : Symbol = J$.R(774185, 'Symbol', Symbol, true, true))), J$.T(774193, 'function', 21, false)) : J$._()) ? J$.B(89914, '===', J$.G(774209, J$.R(774201, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(774217, 'Symbol', undefined, true, true) : Symbol = J$.R(774217, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(89922, '!==', J$.R(774225, 'obj', obj, false, false), J$.G(774241, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(774233, 'Symbol', undefined, true, true) : Symbol = J$.R(774233, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(774249, 'symbol', 21, false) : J$.F(774273, J$.R(774257, '_typeof2', _typeof2, false, true), false)(J$.R(774265, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(778961, J$e);
                                            } finally {
                                                if (J$.Fr(778969))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(774353, J$.F(774345, J$.R(774329, '_typeof', _typeof, false, true), false)(J$.R(774337, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(778977, J$e);
                        } finally {
                            if (J$.Fr(778985))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function writableServerSelector() {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(774721, arguments.callee, this, arguments);
                            arguments = J$.N(774729, 'arguments', arguments, true, false, false);
                            return J$.Rt(774713, J$.T(774705, function (topologyDescription, servers) {
                                jalangiLabel7:
                                    while (true) {
                                        try {
                                            J$.Fe(774673, arguments.callee, this, arguments);
                                            arguments = J$.N(774681, 'arguments', arguments, true, false, false);
                                            topologyDescription = J$.N(774689, 'topologyDescription', topologyDescription, true, false, false);
                                            servers = J$.N(774697, 'servers', servers, true, false, false);
                                            return J$.Rt(774665, J$.F(774657, J$.R(774569, 'latencyWindowReducer', latencyWindowReducer, false, true), false)(J$.R(774577, 'topologyDescription', topologyDescription, false, false), J$.M(774649, J$.R(774585, 'servers', servers, false, false), 'filter', false)(J$.T(774641, function (s) {
                                                jalangiLabel6:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(774617, arguments.callee, this, arguments);
                                                            arguments = J$.N(774625, 'arguments', arguments, true, false, false);
                                                            s = J$.N(774633, 's', s, true, false, false);
                                                            return J$.Rt(774609, J$.G(774601, J$.R(774593, 's', s, false, false), 'isWritable'));
                                                        } catch (J$e) {
                                                            J$.Ex(778993, J$e);
                                                        } finally {
                                                            if (J$.Fr(779001))
                                                                continue jalangiLabel6;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false))));
                                        } catch (J$e) {
                                            J$.Ex(779009, J$e);
                                        } finally {
                                            if (J$.Fr(779017))
                                                continue jalangiLabel7;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(779025, J$e);
                        } finally {
                            if (J$.Fr(779033))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function maxStalenessReducer(readPreference, topologyDescription, servers) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(775737, arguments.callee, this, arguments);
                            arguments = J$.N(775745, 'arguments', arguments, true, false, false);
                            readPreference = J$.N(775753, 'readPreference', readPreference, true, false, false);
                            topologyDescription = J$.N(775761, 'topologyDescription', topologyDescription, true, false, false);
                            servers = J$.N(775769, 'servers', servers, true, false, false);
                            J$.N(775777, 'maxStaleness', maxStaleness, false, false, false);
                            J$.N(775785, 'maxStalenessVariance', maxStalenessVariance, false, false, false);
                            J$.N(775793, 'primary', primary, false, false, false);
                            J$.N(775801, 'sMax', sMax, false, false, false);
                            if (J$.C(44936, J$.C(44928, J$.B(89930, '==', J$.G(774745, J$.R(774737, 'readPreference', readPreference, false, false), 'maxStalenessSeconds'), J$.T(774753, null, 25, false))) ? J$._() : J$.B(89938, '<', J$.G(774769, J$.R(774761, 'readPreference', readPreference, false, false), 'maxStalenessSeconds'), J$.T(774777, 0, 22, false)))) {
                                return J$.Rt(774793, J$.R(774785, 'servers', servers, false, false));
                            }
                            var maxStaleness = J$.W(774817, 'maxStaleness', J$.G(774809, J$.R(774801, 'readPreference', readPreference, false, false), 'maxStalenessSeconds'), maxStaleness, false, false);
                            var maxStalenessVariance = J$.W(774857, 'maxStalenessVariance', J$.B(89954, '/', J$.B(89946, '+', J$.G(774833, J$.R(774825, 'topologyDescription', topologyDescription, false, false), 'heartbeatFrequencyMS'), J$.R(774841, 'IDLE_WRITE_PERIOD', IDLE_WRITE_PERIOD, false, true)), J$.T(774849, 1000, 22, false)), maxStalenessVariance, false, false);
                            if (J$.C(44944, J$.B(89962, '<', J$.R(774865, 'maxStaleness', maxStaleness, false, false), J$.R(774873, 'maxStalenessVariance', maxStalenessVariance, false, false)))) {
                                throw J$.F(774921, J$.R(774881, 'MongoError', MongoError, false, true), true)(J$.M(774913, J$.T(774889, 'maxStalenessSeconds must be at least ', 21, false), 'concat', false)(J$.R(774897, 'maxStalenessVariance', maxStalenessVariance, false, false), J$.T(774905, ' seconds', 21, false)));
                            }
                            if (J$.C(44952, J$.B(89970, '<', J$.R(774929, 'maxStaleness', maxStaleness, false, false), J$.R(774937, 'SMALLEST_MAX_STALENESS_SECONDS', SMALLEST_MAX_STALENESS_SECONDS, false, true)))) {
                                throw J$.F(774985, J$.R(774945, 'MongoError', MongoError, false, true), true)(J$.M(774977, J$.T(774953, 'maxStalenessSeconds must be at least ', 21, false), 'concat', false)(J$.R(774961, 'SMALLEST_MAX_STALENESS_SECONDS', SMALLEST_MAX_STALENESS_SECONDS, false, true), J$.T(774969, ' seconds', 21, false)));
                            }
                            if (J$.C(44992, J$.B(89978, '===', J$.G(775001, J$.R(774993, 'topologyDescription', topologyDescription, false, false), 'type'), J$.G(775017, J$.R(775009, 'TopologyType', TopologyType, false, true), 'ReplicaSetWithPrimary')))) {
                                var primary = J$.W(775065, 'primary', J$.G(775057, J$.M(775041, J$.R(775025, 'servers', servers, false, false), 'filter', false)(J$.R(775033, 'primaryFilter', primaryFilter, false, true)), J$.T(775049, 0, 22, false)), primary, false, false);
                                return J$.Rt(775329, J$.M(775321, J$.R(775073, 'servers', servers, false, false), 'reduce', false)(J$.T(775305, function (result, server) {
                                    jalangiLabel9:
                                        while (true) {
                                            try {
                                                J$.Fe(775257, arguments.callee, this, arguments);
                                                arguments = J$.N(775265, 'arguments', arguments, true, false, false);
                                                result = J$.N(775273, 'result', result, true, false, false);
                                                server = J$.N(775281, 'server', server, true, false, false);
                                                J$.N(775289, 'stalenessMS', stalenessMS, false, false, false);
                                                J$.N(775297, 'staleness', staleness, false, false, false);
                                                var stalenessMS = J$.W(775161, 'stalenessMS', J$.B(90010, '+', J$.B(90002, '-', J$.B(89986, '-', J$.G(775089, J$.R(775081, 'server', server, false, false), 'lastUpdateTime'), J$.G(775105, J$.R(775097, 'server', server, false, false), 'lastWriteDate')), J$.B(89994, '-', J$.G(775121, J$.R(775113, 'primary', primary, false, false), 'lastUpdateTime'), J$.G(775137, J$.R(775129, 'primary', primary, false, false), 'lastWriteDate'))), J$.G(775153, J$.R(775145, 'topologyDescription', topologyDescription, false, false), 'heartbeatFrequencyMS')), stalenessMS, false, false);
                                                var staleness = J$.W(775185, 'staleness', J$.B(90018, '/', J$.R(775169, 'stalenessMS', stalenessMS, false, false), J$.T(775177, 1000, 22, false)), staleness, false, false);
                                                if (J$.C(44960, J$.B(90026, '<=', J$.R(775193, 'staleness', staleness, false, false), J$.G(775209, J$.R(775201, 'readPreference', readPreference, false, false), 'maxStalenessSeconds'))))
                                                    J$.M(775233, J$.R(775217, 'result', result, false, false), 'push', false)(J$.R(775225, 'server', server, false, false));
                                                return J$.Rt(775249, J$.R(775241, 'result', result, false, false));
                                            } catch (J$e) {
                                                J$.Ex(779041, J$e);
                                            } finally {
                                                if (J$.Fr(779049))
                                                    continue jalangiLabel9;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), J$.T(775313, [], 10, false)));
                            } else if (J$.C(44984, J$.B(90034, '===', J$.G(775345, J$.R(775337, 'topologyDescription', topologyDescription, false, false), 'type'), J$.G(775361, J$.R(775353, 'TopologyType', TopologyType, false, true), 'ReplicaSetNoPrimary')))) {
                                var sMax = J$.W(775481, 'sMax', J$.M(775473, J$.R(775369, 'servers', servers, false, false), 'reduce', false)(J$.T(775465, function (max, s) {
                                    jalangiLabel10:
                                        while (true) {
                                            try {
                                                J$.Fe(775433, arguments.callee, this, arguments);
                                                arguments = J$.N(775441, 'arguments', arguments, true, false, false);
                                                max = J$.N(775449, 'max', max, true, false, false);
                                                s = J$.N(775457, 's', s, true, false, false);
                                                return J$.Rt(775425, J$.C(44968, J$.B(90042, '>', J$.G(775385, J$.R(775377, 's', s, false, false), 'lastWriteDate'), J$.G(775401, J$.R(775393, 'max', max, false, false), 'lastWriteDate'))) ? J$.R(775409, 's', s, false, false) : J$.R(775417, 'max', max, false, false));
                                            } catch (J$e) {
                                                J$.Ex(779057, J$e);
                                            } finally {
                                                if (J$.Fr(779065))
                                                    continue jalangiLabel10;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)), sMax, false, false);
                                return J$.Rt(775713, J$.M(775705, J$.R(775489, 'servers', servers, false, false), 'reduce', false)(J$.T(775689, function (result, server) {
                                    jalangiLabel11:
                                        while (true) {
                                            try {
                                                J$.Fe(775641, arguments.callee, this, arguments);
                                                arguments = J$.N(775649, 'arguments', arguments, true, false, false);
                                                result = J$.N(775657, 'result', result, true, false, false);
                                                server = J$.N(775665, 'server', server, true, false, false);
                                                J$.N(775673, 'stalenessMS', stalenessMS, false, false, false);
                                                J$.N(775681, 'staleness', staleness, false, false, false);
                                                var stalenessMS = J$.W(775545, 'stalenessMS', J$.B(90058, '+', J$.B(90050, '-', J$.G(775505, J$.R(775497, 'sMax', sMax, false, false), 'lastWriteDate'), J$.G(775521, J$.R(775513, 'server', server, false, false), 'lastWriteDate')), J$.G(775537, J$.R(775529, 'topologyDescription', topologyDescription, false, false), 'heartbeatFrequencyMS')), stalenessMS, false, false);
                                                var staleness = J$.W(775569, 'staleness', J$.B(90066, '/', J$.R(775553, 'stalenessMS', stalenessMS, false, false), J$.T(775561, 1000, 22, false)), staleness, false, false);
                                                if (J$.C(44976, J$.B(90074, '<=', J$.R(775577, 'staleness', staleness, false, false), J$.G(775593, J$.R(775585, 'readPreference', readPreference, false, false), 'maxStalenessSeconds'))))
                                                    J$.M(775617, J$.R(775601, 'result', result, false, false), 'push', false)(J$.R(775609, 'server', server, false, false));
                                                return J$.Rt(775633, J$.R(775625, 'result', result, false, false));
                                            } catch (J$e) {
                                                J$.Ex(779073, J$e);
                                            } finally {
                                                if (J$.Fr(779081))
                                                    continue jalangiLabel11;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), J$.T(775697, [], 10, false)));
                            }
                            return J$.Rt(775729, J$.R(775721, 'servers', servers, false, false));
                        } catch (J$e) {
                            J$.Ex(779089, J$e);
                        } finally {
                            if (J$.Fr(779097))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function tagSetMatch(tagSet, serverTags) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(776073, arguments.callee, this, arguments);
                            arguments = J$.N(776081, 'arguments', arguments, true, false, false);
                            tagSet = J$.N(776089, 'tagSet', tagSet, true, false, false);
                            serverTags = J$.N(776097, 'serverTags', serverTags, true, false, false);
                            J$.N(776105, 'keys', keys, false, false, false);
                            J$.N(776113, 'serverTagKeys', serverTagKeys, false, false, false);
                            J$.N(776121, 'i', i, false, false, false);
                            J$.N(776129, 'key', key, false, false, false);
                            var keys = J$.W(775833, 'keys', J$.M(775825, J$.I(typeof Object === 'undefined' ? Object = J$.R(775809, 'Object', undefined, true, true) : Object = J$.R(775809, 'Object', Object, true, true)), 'keys', false)(J$.R(775817, 'tagSet', tagSet, false, false)), keys, false, false);
                            var serverTagKeys = J$.W(775865, 'serverTagKeys', J$.M(775857, J$.I(typeof Object === 'undefined' ? Object = J$.R(775841, 'Object', undefined, true, true) : Object = J$.R(775841, 'Object', Object, true, true)), 'keys', false)(J$.R(775849, 'serverTags', serverTags, false, false)), serverTagKeys, false, false);
                            for (var i = J$.W(775881, 'i', J$.T(775873, 0, 22, false), i, false, false); J$.C(45016, J$.B(90082, '<', J$.R(775889, 'i', i, false, false), J$.G(775905, J$.R(775897, 'keys', keys, false, false), 'length'))); i = J$.W(775921, 'i', J$.B(90098, '+', J$.U(90090, '+', J$.R(775913, 'i', i, false, false)), 1), i, false, false)) {
                                var key = J$.W(775953, 'key', J$.G(775945, J$.R(775929, 'keys', keys, false, false), J$.R(775937, 'i', i, false, false)), key, false, false);
                                if (J$.C(45008, J$.C(45000, J$.B(90114, '===', J$.M(775977, J$.R(775961, 'serverTagKeys', serverTagKeys, false, false), 'indexOf', false)(J$.R(775969, 'key', key, false, false)), J$.U(90106, '-', J$.T(775985, 1, 22, false)))) ? J$._() : J$.B(90122, '!==', J$.G(776009, J$.R(775993, 'serverTags', serverTags, false, false), J$.R(776001, 'key', key, false, false)), J$.G(776033, J$.R(776017, 'tagSet', tagSet, false, false), J$.R(776025, 'key', key, false, false))))) {
                                    return J$.Rt(776049, J$.T(776041, false, 23, false));
                                }
                            }
                            return J$.Rt(776065, J$.T(776057, true, 23, false));
                        } catch (J$e) {
                            J$.Ex(779105, J$e);
                        } finally {
                            if (J$.Fr(779113))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function tagSetReducer(readPreference, servers) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(776697, arguments.callee, this, arguments);
                            arguments = J$.N(776705, 'arguments', arguments, true, false, false);
                            readPreference = J$.N(776713, 'readPreference', readPreference, true, false, false);
                            servers = J$.N(776721, 'servers', servers, true, false, false);
                            J$.N(776729, '_loop', _loop, false, false, false);
                            J$.N(776737, 'i', i, false, false, false);
                            J$.N(776745, '_ret', _ret, false, false, false);
                            if (J$.C(45040, J$.C(45032, J$.B(90130, '==', J$.G(776145, J$.R(776137, 'readPreference', readPreference, false, false), 'tags'), J$.T(776153, null, 25, false))) ? J$._() : J$.C(45024, J$.M(776185, J$.I(typeof Array === 'undefined' ? Array = J$.R(776161, 'Array', undefined, true, true) : Array = J$.R(776161, 'Array', Array, true, true)), 'isArray', false)(J$.G(776177, J$.R(776169, 'readPreference', readPreference, false, false), 'tags'))) ? J$.B(90138, '===', J$.G(776209, J$.G(776201, J$.R(776193, 'readPreference', readPreference, false, false), 'tags'), 'length'), J$.T(776217, 0, 22, false)) : J$._())) {
                                return J$.Rt(776233, J$.R(776225, 'servers', servers, false, false));
                            }
                            var _loop = J$.W(776521, '_loop', J$.T(776513, function _loop(i) {
                                jalangiLabel15:
                                    while (true) {
                                        try {
                                            J$.Fe(776473, arguments.callee, this, arguments);
                                            arguments = J$.N(776481, 'arguments', arguments, true, false, false);
                                            i = J$.N(776489, 'i', i, true, false, false);
                                            J$.N(776497, 'tagSet', tagSet, false, false, false);
                                            J$.N(776505, 'serversMatchingTagset', serversMatchingTagset, false, false, false);
                                            var tagSet = J$.W(776273, 'tagSet', J$.G(776265, J$.G(776249, J$.R(776241, 'readPreference', readPreference, false, false), 'tags'), J$.R(776257, 'i', i, false, false)), tagSet, false, false);
                                            var serversMatchingTagset = J$.W(776425, 'serversMatchingTagset', J$.M(776417, J$.R(776281, 'servers', servers, false, false), 'reduce', false)(J$.T(776401, function (matched, server) {
                                                jalangiLabel14:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(776369, arguments.callee, this, arguments);
                                                            arguments = J$.N(776377, 'arguments', arguments, true, false, false);
                                                            matched = J$.N(776385, 'matched', matched, true, false, false);
                                                            server = J$.N(776393, 'server', server, true, false, false);
                                                            if (J$.C(45048, J$.F(776321, J$.R(776289, 'tagSetMatch', tagSetMatch, false, true), false)(J$.R(776297, 'tagSet', tagSet, false, false), J$.G(776313, J$.R(776305, 'server', server, false, false), 'tags'))))
                                                                J$.M(776345, J$.R(776329, 'matched', matched, false, false), 'push', false)(J$.R(776337, 'server', server, false, false));
                                                            return J$.Rt(776361, J$.R(776353, 'matched', matched, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(779121, J$e);
                                                        } finally {
                                                            if (J$.Fr(779129))
                                                                continue jalangiLabel14;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false), J$.T(776409, [], 10, false)), serversMatchingTagset, false, false);
                                            if (J$.C(45056, J$.G(776441, J$.R(776433, 'serversMatchingTagset', serversMatchingTagset, false, false), 'length'))) {
                                                return J$.Rt(776465, J$.T(776457, { v: J$.R(776449, 'serversMatchingTagset', serversMatchingTagset, false, false) }, 11, false));
                                            }
                                        } catch (J$e) {
                                            J$.Ex(779137, J$e);
                                        } finally {
                                            if (J$.Fr(779145))
                                                continue jalangiLabel15;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _loop, false, false);
                            for (var i = J$.W(776537, 'i', J$.T(776529, 0, 22, false), i, false, false); J$.C(45072, J$.B(90146, '<', J$.R(776545, 'i', i, false, false), J$.G(776569, J$.G(776561, J$.R(776553, 'readPreference', readPreference, false, false), 'tags'), 'length'))); i = J$.W(776585, 'i', J$.B(90162, '+', J$.U(90154, '+', J$.R(776577, 'i', i, false, false)), 1), i, false, false)) {
                                var _ret = J$.W(776617, '_ret', J$.F(776609, J$.R(776593, '_loop', _loop, false, false), false)(J$.R(776601, 'i', i, false, false)), _ret, false, false);
                                if (J$.C(45064, J$.B(90170, '===', J$.F(776641, J$.R(776625, '_typeof', _typeof, false, true), false)(J$.R(776633, '_ret', _ret, false, false)), J$.T(776649, 'object', 21, false))))
                                    return J$.Rt(776673, J$.G(776665, J$.R(776657, '_ret', _ret, false, false), 'v'));
                            }
                            return J$.Rt(776689, J$.T(776681, [], 10, false));
                        } catch (J$e) {
                            J$.Ex(779153, J$e);
                        } finally {
                            if (J$.Fr(779161))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function latencyWindowReducer(topologyDescription, servers) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(777097, arguments.callee, this, arguments);
                            arguments = J$.N(777105, 'arguments', arguments, true, false, false);
                            topologyDescription = J$.N(777113, 'topologyDescription', topologyDescription, true, false, false);
                            servers = J$.N(777121, 'servers', servers, true, false, false);
                            J$.N(777129, 'low', low, false, false, false);
                            J$.N(777137, 'high', high, false, false, false);
                            var low = J$.W(776897, 'low', J$.M(776889, J$.R(776753, 'servers', servers, false, false), 'reduce', false)(J$.T(776873, function (min, server) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(776841, arguments.callee, this, arguments);
                                            arguments = J$.N(776849, 'arguments', arguments, true, false, false);
                                            min = J$.N(776857, 'min', min, true, false, false);
                                            server = J$.N(776865, 'server', server, true, false, false);
                                            return J$.Rt(776833, J$.C(45080, J$.B(90186, '===', J$.R(776761, 'min', min, false, false), J$.U(90178, '-', J$.T(776769, 1, 22, false)))) ? J$.G(776785, J$.R(776777, 'server', server, false, false), 'roundTripTime') : J$.M(776825, J$.I(typeof Math === 'undefined' ? Math = J$.R(776793, 'Math', undefined, true, true) : Math = J$.R(776793, 'Math', Math, true, true)), 'min', false)(J$.G(776809, J$.R(776801, 'server', server, false, false), 'roundTripTime'), J$.R(776817, 'min', min, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(779169, J$e);
                                        } finally {
                                            if (J$.Fr(779177))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), J$.U(90194, '-', J$.T(776881, 1, 22, false))), low, false, false);
                            var high = J$.W(776929, 'high', J$.B(90202, '+', J$.R(776905, 'low', low, false, false), J$.G(776921, J$.R(776913, 'topologyDescription', topologyDescription, false, false), 'localThresholdMS')), high, false, false);
                            return J$.Rt(777089, J$.M(777081, J$.R(776937, 'servers', servers, false, false), 'reduce', false)(J$.T(777065, function (result, server) {
                                jalangiLabel18:
                                    while (true) {
                                        try {
                                            J$.Fe(777033, arguments.callee, this, arguments);
                                            arguments = J$.N(777041, 'arguments', arguments, true, false, false);
                                            result = J$.N(777049, 'result', result, true, false, false);
                                            server = J$.N(777057, 'server', server, true, false, false);
                                            if (J$.C(45096, J$.C(45088, J$.B(90210, '<=', J$.G(776953, J$.R(776945, 'server', server, false, false), 'roundTripTime'), J$.R(776961, 'high', high, false, false))) ? J$.B(90218, '>=', J$.G(776977, J$.R(776969, 'server', server, false, false), 'roundTripTime'), J$.R(776985, 'low', low, false, false)) : J$._()))
                                                J$.M(777009, J$.R(776993, 'result', result, false, false), 'push', false)(J$.R(777001, 'server', server, false, false));
                                            return J$.Rt(777025, J$.R(777017, 'result', result, false, false));
                                        } catch (J$e) {
                                            J$.Ex(779185, J$e);
                                        } finally {
                                            if (J$.Fr(779193))
                                                continue jalangiLabel18;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), J$.T(777073, [], 10, false)));
                        } catch (J$e) {
                            J$.Ex(779201, J$e);
                        } finally {
                            if (J$.Fr(779209))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function primaryFilter(server) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(777185, arguments.callee, this, arguments);
                            arguments = J$.N(777193, 'arguments', arguments, true, false, false);
                            server = J$.N(777201, 'server', server, true, false, false);
                            return J$.Rt(777177, J$.B(90226, '===', J$.G(777153, J$.R(777145, 'server', server, false, false), 'type'), J$.G(777169, J$.R(777161, 'ServerType', ServerType, false, true), 'RSPrimary')));
                        } catch (J$e) {
                            J$.Ex(779217, J$e);
                        } finally {
                            if (J$.Fr(779225))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secondaryFilter(server) {
                jalangiLabel21:
                    while (true) {
                        try {
                            J$.Fe(777249, arguments.callee, this, arguments);
                            arguments = J$.N(777257, 'arguments', arguments, true, false, false);
                            server = J$.N(777265, 'server', server, true, false, false);
                            return J$.Rt(777241, J$.B(90234, '===', J$.G(777217, J$.R(777209, 'server', server, false, false), 'type'), J$.G(777233, J$.R(777225, 'ServerType', ServerType, false, true), 'RSSecondary')));
                        } catch (J$e) {
                            J$.Ex(779233, J$e);
                        } finally {
                            if (J$.Fr(779241))
                                continue jalangiLabel21;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function nearestFilter(server) {
                jalangiLabel22:
                    while (true) {
                        try {
                            J$.Fe(777345, arguments.callee, this, arguments);
                            arguments = J$.N(777353, 'arguments', arguments, true, false, false);
                            server = J$.N(777361, 'server', server, true, false, false);
                            return J$.Rt(777337, J$.C(45104, J$.B(90242, '===', J$.G(777281, J$.R(777273, 'server', server, false, false), 'type'), J$.G(777297, J$.R(777289, 'ServerType', ServerType, false, true), 'RSSecondary'))) ? J$._() : J$.B(90250, '===', J$.G(777313, J$.R(777305, 'server', server, false, false), 'type'), J$.G(777329, J$.R(777321, 'ServerType', ServerType, false, true), 'RSPrimary')));
                        } catch (J$e) {
                            J$.Ex(779249, J$e);
                        } finally {
                            if (J$.Fr(779257))
                                continue jalangiLabel22;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function knownFilter(server) {
                jalangiLabel23:
                    while (true) {
                        try {
                            J$.Fe(777409, arguments.callee, this, arguments);
                            arguments = J$.N(777417, 'arguments', arguments, true, false, false);
                            server = J$.N(777425, 'server', server, true, false, false);
                            return J$.Rt(777401, J$.B(90258, '!==', J$.G(777377, J$.R(777369, 'server', server, false, false), 'type'), J$.G(777393, J$.R(777385, 'ServerType', ServerType, false, true), 'Unknown')));
                        } catch (J$e) {
                            J$.Ex(779265, J$e);
                        } finally {
                            if (J$.Fr(779273))
                                continue jalangiLabel23;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function readPreferenceServerSelector(readPreference) {
                jalangiLabel25:
                    while (true) {
                        try {
                            J$.Fe(778585, arguments.callee, this, arguments);
                            arguments = J$.N(778593, 'arguments', arguments, true, false, false);
                            readPreference = J$.N(778601, 'readPreference', readPreference, true, false, false);
                            if (J$.C(45112, J$.U(90266, '!', J$.M(777441, J$.R(777433, 'readPreference', readPreference, false, false), 'isValid', false)()))) {
                                throw J$.F(777465, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(777449, 'TypeError', undefined, true, true) : TypeError = J$.R(777449, 'TypeError', TypeError, true, true)), true)(J$.T(777457, 'Invalid read preference specified', 21, false));
                            }
                            return J$.Rt(778577, J$.T(778569, function (topologyDescription, servers) {
                                jalangiLabel24:
                                    while (true) {
                                        try {
                                            J$.Fe(778513, arguments.callee, this, arguments);
                                            arguments = J$.N(778521, 'arguments', arguments, true, false, false);
                                            topologyDescription = J$.N(778529, 'topologyDescription', topologyDescription, true, false, false);
                                            servers = J$.N(778537, 'servers', servers, true, false, false);
                                            J$.N(778545, 'commonWireVersion', commonWireVersion, false, false, false);
                                            J$.N(778553, 'result', result, false, false, false);
                                            J$.N(778561, '_result', _result, false, false, false);
                                            var commonWireVersion = J$.W(777489, 'commonWireVersion', J$.G(777481, J$.R(777473, 'topologyDescription', topologyDescription, false, false), 'commonWireVersion'), commonWireVersion, false, false);
                                            if (J$.C(45136, J$.C(45128, J$.C(45120, J$.R(777497, 'commonWireVersion', commonWireVersion, false, false)) ? J$.G(777513, J$.R(777505, 'readPreference', readPreference, false, false), 'minWireVersion') : J$._()) ? J$.B(90274, '>', J$.G(777529, J$.R(777521, 'readPreference', readPreference, false, false), 'minWireVersion'), J$.R(777537, 'commonWireVersion', commonWireVersion, false, false)) : J$._())) {
                                                throw J$.F(777617, J$.R(777545, 'MongoError', MongoError, false, true), true)(J$.M(777609, J$.M(777585, J$.T(777553, 'Minimum wire version \'', 21, false), 'concat', false)(J$.G(777569, J$.R(777561, 'readPreference', readPreference, false, false), 'minWireVersion'), J$.T(777577, '\' required, but found \'', 21, false)), 'concat', false)(J$.R(777593, 'commonWireVersion', commonWireVersion, false, false), J$.T(777601, '\'', 21, false)));
                                            }
                                            if (J$.C(45152, J$.C(45144, J$.B(90282, '===', J$.G(777633, J$.R(777625, 'topologyDescription', topologyDescription, false, false), 'type'), J$.G(777649, J$.R(777641, 'TopologyType', TopologyType, false, true), 'Single'))) ? J$._() : J$.B(90290, '===', J$.G(777665, J$.R(777657, 'topologyDescription', topologyDescription, false, false), 'type'), J$.G(777681, J$.R(777673, 'TopologyType', TopologyType, false, true), 'Sharded')))) {
                                                return J$.Rt(777737, J$.F(777729, J$.R(777689, 'latencyWindowReducer', latencyWindowReducer, false, true), false)(J$.R(777697, 'topologyDescription', topologyDescription, false, false), J$.M(777721, J$.R(777705, 'servers', servers, false, false), 'filter', false)(J$.R(777713, 'knownFilter', knownFilter, false, true))));
                                            }
                                            if (J$.C(45160, J$.B(90298, '===', J$.G(777753, J$.R(777745, 'readPreference', readPreference, false, false), 'mode'), J$.G(777769, J$.R(777761, 'ReadPreference', ReadPreference, false, true), 'PRIMARY')))) {
                                                return J$.Rt(777801, J$.M(777793, J$.R(777777, 'servers', servers, false, false), 'filter', false)(J$.R(777785, 'primaryFilter', primaryFilter, false, true)));
                                            }
                                            if (J$.C(45208, J$.B(90306, '===', J$.G(777817, J$.R(777809, 'readPreference', readPreference, false, false), 'mode'), J$.G(777833, J$.R(777825, 'ReadPreference', ReadPreference, false, true), 'SECONDARY')))) {
                                                return J$.Rt(777945, J$.M(777937, J$.F(777921, J$.R(777841, 'latencyWindowReducer', latencyWindowReducer, false, true), false)(J$.R(777849, 'topologyDescription', topologyDescription, false, false), J$.F(777913, J$.R(777857, 'tagSetReducer', tagSetReducer, false, true), false)(J$.R(777865, 'readPreference', readPreference, false, false), J$.F(777905, J$.R(777873, 'maxStalenessReducer', maxStalenessReducer, false, true), false)(J$.R(777881, 'readPreference', readPreference, false, false), J$.R(777889, 'topologyDescription', topologyDescription, false, false), J$.R(777897, 'servers', servers, false, false)))), 'filter', false)(J$.R(777929, 'secondaryFilter', secondaryFilter, false, true)));
                                            } else if (J$.C(45200, J$.B(90314, '===', J$.G(777961, J$.R(777953, 'readPreference', readPreference, false, false), 'mode'), J$.G(777977, J$.R(777969, 'ReadPreference', ReadPreference, false, true), 'NEAREST')))) {
                                                return J$.Rt(778089, J$.M(778081, J$.F(778065, J$.R(777985, 'latencyWindowReducer', latencyWindowReducer, false, true), false)(J$.R(777993, 'topologyDescription', topologyDescription, false, false), J$.F(778057, J$.R(778001, 'tagSetReducer', tagSetReducer, false, true), false)(J$.R(778009, 'readPreference', readPreference, false, false), J$.F(778049, J$.R(778017, 'maxStalenessReducer', maxStalenessReducer, false, true), false)(J$.R(778025, 'readPreference', readPreference, false, false), J$.R(778033, 'topologyDescription', topologyDescription, false, false), J$.R(778041, 'servers', servers, false, false)))), 'filter', false)(J$.R(778073, 'nearestFilter', nearestFilter, false, true)));
                                            } else if (J$.C(45192, J$.B(90322, '===', J$.G(778105, J$.R(778097, 'readPreference', readPreference, false, false), 'mode'), J$.G(778121, J$.R(778113, 'ReadPreference', ReadPreference, false, true), 'SECONDARY_PREFERRED')))) {
                                                var result = J$.W(778233, 'result', J$.M(778225, J$.F(778209, J$.R(778129, 'latencyWindowReducer', latencyWindowReducer, false, true), false)(J$.R(778137, 'topologyDescription', topologyDescription, false, false), J$.F(778201, J$.R(778145, 'tagSetReducer', tagSetReducer, false, true), false)(J$.R(778153, 'readPreference', readPreference, false, false), J$.F(778193, J$.R(778161, 'maxStalenessReducer', maxStalenessReducer, false, true), false)(J$.R(778169, 'readPreference', readPreference, false, false), J$.R(778177, 'topologyDescription', topologyDescription, false, false), J$.R(778185, 'servers', servers, false, false)))), 'filter', false)(J$.R(778217, 'secondaryFilter', secondaryFilter, false, true)), result, false, false);
                                                return J$.Rt(778297, J$.C(45168, J$.B(90330, '===', J$.G(778249, J$.R(778241, 'result', result, false, false), 'length'), J$.T(778257, 0, 22, false))) ? J$.M(778281, J$.R(778265, 'servers', servers, false, false), 'filter', false)(J$.R(778273, 'primaryFilter', primaryFilter, false, true)) : J$.R(778289, 'result', result, false, false));
                                            } else if (J$.C(45184, J$.B(90338, '===', J$.G(778313, J$.R(778305, 'readPreference', readPreference, false, false), 'mode'), J$.G(778329, J$.R(778321, 'ReadPreference', ReadPreference, false, true), 'PRIMARY_PREFERRED')))) {
                                                var _result = J$.W(778361, '_result', J$.M(778353, J$.R(778337, 'servers', servers, false, false), 'filter', false)(J$.R(778345, 'primaryFilter', primaryFilter, false, true)), _result, false, false);
                                                if (J$.C(45176, J$.G(778377, J$.R(778369, '_result', _result, false, false), 'length'))) {
                                                    return J$.Rt(778393, J$.R(778385, '_result', _result, false, false));
                                                }
                                                return J$.Rt(778505, J$.M(778497, J$.F(778481, J$.R(778401, 'latencyWindowReducer', latencyWindowReducer, false, true), false)(J$.R(778409, 'topologyDescription', topologyDescription, false, false), J$.F(778473, J$.R(778417, 'tagSetReducer', tagSetReducer, false, true), false)(J$.R(778425, 'readPreference', readPreference, false, false), J$.F(778465, J$.R(778433, 'maxStalenessReducer', maxStalenessReducer, false, true), false)(J$.R(778441, 'readPreference', readPreference, false, false), J$.R(778449, 'topologyDescription', topologyDescription, false, false), J$.R(778457, 'servers', servers, false, false)))), 'filter', false)(J$.R(778489, 'secondaryFilter', secondaryFilter, false, true)));
                                            }
                                        } catch (J$e) {
                                            J$.Ex(779281, J$e);
                                        } finally {
                                            if (J$.Fr(779289))
                                                continue jalangiLabel24;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(779297, J$e);
                        } finally {
                            if (J$.Fr(779305))
                                continue jalangiLabel25;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(778665, '_typeof2', J$.T(778657, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(778681, '_typeof', J$.T(778673, _typeof, 12, false), true, false, false);
            J$.N(778689, 'ServerType', ServerType, false, false, false);
            J$.N(778697, 'TopologyType', TopologyType, false, false, false);
            J$.N(778705, 'ReadPreference', ReadPreference, false, false, false);
            J$.N(778713, 'MongoError', MongoError, false, false, false);
            J$.N(778721, 'IDLE_WRITE_PERIOD', IDLE_WRITE_PERIOD, false, false, false);
            J$.N(778729, 'SMALLEST_MAX_STALENESS_SECONDS', SMALLEST_MAX_STALENESS_SECONDS, false, false, false);
            writableServerSelector = J$.N(778745, 'writableServerSelector', J$.T(778737, writableServerSelector, 12, false), true, false, false);
            maxStalenessReducer = J$.N(778761, 'maxStalenessReducer', J$.T(778753, maxStalenessReducer, 12, false), true, false, false);
            tagSetMatch = J$.N(778777, 'tagSetMatch', J$.T(778769, tagSetMatch, 12, false), true, false, false);
            tagSetReducer = J$.N(778793, 'tagSetReducer', J$.T(778785, tagSetReducer, 12, false), true, false, false);
            latencyWindowReducer = J$.N(778809, 'latencyWindowReducer', J$.T(778801, latencyWindowReducer, 12, false), true, false, false);
            primaryFilter = J$.N(778825, 'primaryFilter', J$.T(778817, primaryFilter, 12, false), true, false, false);
            secondaryFilter = J$.N(778841, 'secondaryFilter', J$.T(778833, secondaryFilter, 12, false), true, false, false);
            nearestFilter = J$.N(778857, 'nearestFilter', J$.T(778849, nearestFilter, 12, false), true, false, false);
            knownFilter = J$.N(778873, 'knownFilter', J$.T(778865, knownFilter, 12, false), true, false, false);
            readPreferenceServerSelector = J$.N(778889, 'readPreferenceServerSelector', J$.T(778881, readPreferenceServerSelector, 12, false), true, false, false);
            J$.T(773753, 'use strict', 21, false);
            var ServerType = J$.W(774417, 'ServerType', J$.G(774409, J$.F(774401, J$.I(typeof require === 'undefined' ? require = J$.R(774385, 'require', undefined, true, true) : require = J$.R(774385, 'require', require, true, true)), false)(J$.T(774393, './server_description', 21, false)), 'ServerType'), ServerType, false, true);
            var TopologyType = J$.W(774457, 'TopologyType', J$.G(774449, J$.F(774441, J$.I(typeof require === 'undefined' ? require = J$.R(774425, 'require', undefined, true, true) : require = J$.R(774425, 'require', require, true, true)), false)(J$.T(774433, './topology_description', 21, false)), 'TopologyType'), TopologyType, false, true);
            var ReadPreference = J$.W(774489, 'ReadPreference', J$.F(774481, J$.I(typeof require === 'undefined' ? require = J$.R(774465, 'require', undefined, true, true) : require = J$.R(774465, 'require', require, true, true)), false)(J$.T(774473, '../topologies/read_preference', 21, false)), ReadPreference, false, true);
            var MongoError = J$.W(774529, 'MongoError', J$.G(774521, J$.F(774513, J$.I(typeof require === 'undefined' ? require = J$.R(774497, 'require', undefined, true, true) : require = J$.R(774497, 'require', require, true, true)), false)(J$.T(774505, '../error', 21, false)), 'MongoError'), MongoError, false, true);
            var IDLE_WRITE_PERIOD = J$.W(774545, 'IDLE_WRITE_PERIOD', J$.T(774537, 10000, 22, false), IDLE_WRITE_PERIOD, false, true);
            var SMALLEST_MAX_STALENESS_SECONDS = J$.W(774561, 'SMALLEST_MAX_STALENESS_SECONDS', J$.T(774553, 90, 22, false), SMALLEST_MAX_STALENESS_SECONDS, false, true);
            J$.P(778641, J$.I(typeof module === 'undefined' ? module = J$.R(778609, 'module', undefined, true, true) : module = J$.R(778609, 'module', module, true, true)), 'exports', J$.T(778633, {
                writableServerSelector: J$.R(778617, 'writableServerSelector', writableServerSelector, false, true),
                readPreferenceServerSelector: J$.R(778625, 'readPreferenceServerSelector', readPreferenceServerSelector, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(779313, J$e);
        } finally {
            if (J$.Sr(779321))
                continue jalangiLabel26;
            else
                break jalangiLabel26;
        }
    }
// JALANGI DO NOT INSTRUMENT
