// mongodb/lib/core/sdam/srv_polling.js
J$.noInstrEval = false;
jalangiLabel35:
    while (true) {
        try {
            J$.Se(784073, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/sdam/srv_polling.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(779601, arguments.callee, this, arguments);
                            arguments = J$.N(779609, 'arguments', arguments, true, false, false);
                            obj = J$.N(779617, 'obj', obj, true, false, false);
                            if (J$.C(45256, J$.C(45216, J$.B(90354, '===', J$.U(90346, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(779337, 'Symbol', undefined, true, true) : Symbol = J$.R(779337, 'Symbol', Symbol, true, true))), J$.T(779345, 'function', 21, false))) ? J$.B(90370, '===', J$.U(90362, 'typeof', J$.G(779361, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(779353, 'Symbol', undefined, true, true) : Symbol = J$.R(779353, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(779369, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(779425, '_typeof2', J$.T(779417, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(779393, arguments.callee, this, arguments);
                                                arguments = J$.N(779401, 'arguments', arguments, true, false, false);
                                                obj = J$.N(779409, 'obj', obj, true, false, false);
                                                return J$.Rt(779385, J$.U(90378, 'typeof', J$.R(779377, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(784297, J$e);
                                            } finally {
                                                if (J$.Fr(784305))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(779561, '_typeof2', J$.T(779553, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(779529, arguments.callee, this, arguments);
                                                arguments = J$.N(779537, 'arguments', arguments, true, false, false);
                                                obj = J$.N(779545, 'obj', obj, true, false, false);
                                                return J$.Rt(779521, J$.C(45248, J$.C(45240, J$.C(45232, J$.C(45224, J$.R(779433, 'obj', obj, false, false)) ? J$.B(90394, '===', J$.U(90386, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(779441, 'Symbol', undefined, true, true) : Symbol = J$.R(779441, 'Symbol', Symbol, true, true))), J$.T(779449, 'function', 21, false)) : J$._()) ? J$.B(90402, '===', J$.G(779465, J$.R(779457, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(779473, 'Symbol', undefined, true, true) : Symbol = J$.R(779473, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(90410, '!==', J$.R(779481, 'obj', obj, false, false), J$.G(779497, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(779489, 'Symbol', undefined, true, true) : Symbol = J$.R(779489, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(779505, 'symbol', 21, false) : J$.U(90418, 'typeof', J$.R(779513, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(784313, J$e);
                                            } finally {
                                                if (J$.Fr(784321))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(779593, J$.F(779585, J$.R(779569, '_typeof2', _typeof2, false, true), false)(J$.R(779577, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(784329, J$e);
                        } finally {
                            if (J$.Fr(784337))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(779937, arguments.callee, this, arguments);
                            arguments = J$.N(779945, 'arguments', arguments, true, false, false);
                            obj = J$.N(779953, 'obj', obj, true, false, false);
                            if (J$.C(45304, J$.C(45264, J$.B(90434, '===', J$.U(90426, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(779625, 'Symbol', undefined, true, true) : Symbol = J$.R(779625, 'Symbol', Symbol, true, true))), J$.T(779633, 'function', 21, false))) ? J$.B(90442, '===', J$.F(779665, J$.R(779641, '_typeof2', _typeof2, false, true), false)(J$.G(779657, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(779649, 'Symbol', undefined, true, true) : Symbol = J$.R(779649, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(779673, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(779745, '_typeof', J$.T(779737, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(779713, arguments.callee, this, arguments);
                                                arguments = J$.N(779721, 'arguments', arguments, true, false, false);
                                                obj = J$.N(779729, 'obj', obj, true, false, false);
                                                return J$.Rt(779705, J$.F(779697, J$.R(779681, '_typeof2', _typeof2, false, true), false)(J$.R(779689, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(784345, J$e);
                                            } finally {
                                                if (J$.Fr(784353))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(779897, '_typeof', J$.T(779889, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(779865, arguments.callee, this, arguments);
                                                arguments = J$.N(779873, 'arguments', arguments, true, false, false);
                                                obj = J$.N(779881, 'obj', obj, true, false, false);
                                                return J$.Rt(779857, J$.C(45296, J$.C(45288, J$.C(45280, J$.C(45272, J$.R(779753, 'obj', obj, false, false)) ? J$.B(90458, '===', J$.U(90450, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(779761, 'Symbol', undefined, true, true) : Symbol = J$.R(779761, 'Symbol', Symbol, true, true))), J$.T(779769, 'function', 21, false)) : J$._()) ? J$.B(90466, '===', J$.G(779785, J$.R(779777, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(779793, 'Symbol', undefined, true, true) : Symbol = J$.R(779793, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(90474, '!==', J$.R(779801, 'obj', obj, false, false), J$.G(779817, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(779809, 'Symbol', undefined, true, true) : Symbol = J$.R(779809, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(779825, 'symbol', 21, false) : J$.F(779849, J$.R(779833, '_typeof2', _typeof2, false, true), false)(J$.R(779841, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(784361, J$e);
                                            } finally {
                                                if (J$.Fr(784369))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(779929, J$.F(779921, J$.R(779905, '_typeof', _typeof, false, true), false)(J$.R(779913, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(784377, J$e);
                        } finally {
                            if (J$.Fr(784385))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(780065, arguments.callee, this, arguments);
                            arguments = J$.N(780073, 'arguments', arguments, true, false, false);
                            self = J$.N(780081, 'self', self, true, false, false);
                            call = J$.N(780089, 'call', call, true, false, false);
                            if (J$.C(45328, J$.C(45320, J$.R(779961, 'call', call, false, false)) ? J$.C(45312, J$.B(90482, '===', J$.F(779985, J$.R(779969, '_typeof', _typeof, false, true), false)(J$.R(779977, 'call', call, false, false)), J$.T(779993, 'object', 21, false))) ? J$._() : J$.B(90498, '===', J$.U(90490, 'typeof', J$.R(780001, 'call', call, false, false)), J$.T(780009, 'function', 21, false)) : J$._())) {
                                return J$.Rt(780025, J$.R(780017, 'call', call, false, false));
                            }
                            return J$.Rt(780057, J$.F(780049, J$.R(780033, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(780041, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(784393, J$e);
                        } finally {
                            if (J$.Fr(784401))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(780153, arguments.callee, this, arguments);
                            arguments = J$.N(780161, 'arguments', arguments, true, false, false);
                            self = J$.N(780169, 'self', self, true, false, false);
                            if (J$.C(45336, J$.B(90506, '===', J$.R(780097, 'self', self, false, false), void J$.T(780105, 0, 22, false)))) {
                                throw J$.F(780129, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(780113, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(780113, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(780121, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(780145, J$.R(780137, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(784409, J$e);
                        } finally {
                            if (J$.Fr(784417))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(780329, arguments.callee, this, arguments);
                            arguments = J$.N(780337, 'arguments', arguments, true, false, false);
                            o = J$.N(780345, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(780289, '_getPrototypeOf', J$.C(45352, J$.G(780185, J$.I(typeof Object === 'undefined' ? Object = J$.R(780177, 'Object', undefined, true, true) : Object = J$.R(780177, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(780201, J$.I(typeof Object === 'undefined' ? Object = J$.R(780193, 'Object', undefined, true, true) : Object = J$.R(780193, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(780281, function _getPrototypeOf(o) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(780257, arguments.callee, this, arguments);
                                            arguments = J$.N(780265, 'arguments', arguments, true, false, false);
                                            o = J$.N(780273, 'o', o, true, false, false);
                                            return J$.Rt(780249, J$.C(45344, J$.G(780217, J$.R(780209, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(780241, J$.I(typeof Object === 'undefined' ? Object = J$.R(780225, 'Object', undefined, true, true) : Object = J$.R(780225, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(780233, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(784425, J$e);
                                        } finally {
                                            if (J$.Fr(784433))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(780321, J$.F(780313, J$.R(780297, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(780305, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(784441, J$e);
                        } finally {
                            if (J$.Fr(784449))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(780545, arguments.callee, this, arguments);
                            arguments = J$.N(780553, 'arguments', arguments, true, false, false);
                            subClass = J$.N(780561, 'subClass', subClass, true, false, false);
                            superClass = J$.N(780569, 'superClass', superClass, true, false, false);
                            if (J$.C(45368, J$.C(45360, J$.B(90522, '!==', J$.U(90514, 'typeof', J$.R(780353, 'superClass', superClass, false, false)), J$.T(780361, 'function', 21, false))) ? J$.B(90530, '!==', J$.R(780369, 'superClass', superClass, false, false), J$.T(780377, null, 25, false)) : J$._())) {
                                throw J$.F(780401, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(780385, 'TypeError', undefined, true, true) : TypeError = J$.R(780385, 'TypeError', TypeError, true, true)), true)(J$.T(780393, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(780497, J$.R(780409, 'subClass', subClass, false, false), 'prototype', J$.M(780489, J$.I(typeof Object === 'undefined' ? Object = J$.R(780417, 'Object', undefined, true, true) : Object = J$.R(780417, 'Object', Object, true, true)), 'create', false)(J$.C(45376, J$.R(780425, 'superClass', superClass, false, false)) ? J$.G(780441, J$.R(780433, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(780481, {
                                constructor: J$.T(780473, {
                                    value: J$.R(780449, 'subClass', subClass, false, false),
                                    writable: J$.T(780457, true, 23, false),
                                    configurable: J$.T(780465, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(45384, J$.R(780505, 'superClass', superClass, false, false)))
                                J$.F(780537, J$.R(780513, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(780521, 'subClass', subClass, false, false), J$.R(780529, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(784457, J$e);
                        } finally {
                            if (J$.Fr(784465))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(780721, arguments.callee, this, arguments);
                            arguments = J$.N(780729, 'arguments', arguments, true, false, false);
                            o = J$.N(780737, 'o', o, true, false, false);
                            p = J$.N(780745, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(780673, '_setPrototypeOf', J$.C(45392, J$.G(780585, J$.I(typeof Object === 'undefined' ? Object = J$.R(780577, 'Object', undefined, true, true) : Object = J$.R(780577, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(780665, function _setPrototypeOf(o, p) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(780633, arguments.callee, this, arguments);
                                            arguments = J$.N(780641, 'arguments', arguments, true, false, false);
                                            o = J$.N(780649, 'o', o, true, false, false);
                                            p = J$.N(780657, 'p', p, true, false, false);
                                            J$.P(780609, J$.R(780593, 'o', o, false, false), '__proto__', J$.R(780601, 'p', p, false, false));
                                            return J$.Rt(780625, J$.R(780617, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(784473, J$e);
                                        } finally {
                                            if (J$.Fr(784481))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(780713, J$.F(780705, J$.R(780681, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(780689, 'o', o, false, false), J$.R(780697, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(784489, J$e);
                        } finally {
                            if (J$.Fr(784497))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(780793, arguments.callee, this, arguments);
                            arguments = J$.N(780801, 'arguments', arguments, true, false, false);
                            instance = J$.N(780809, 'instance', instance, true, false, false);
                            Constructor = J$.N(780817, 'Constructor', Constructor, true, false, false);
                            if (J$.C(45400, J$.U(90546, '!', J$.B(90538, 'instanceof', J$.R(780753, 'instance', instance, false, false), J$.R(780761, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(780785, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(780769, 'TypeError', undefined, true, true) : TypeError = J$.R(780769, 'TypeError', TypeError, true, true)), true)(J$.T(780777, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(784505, J$e);
                        } finally {
                            if (J$.Fr(784513))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(781065, arguments.callee, this, arguments);
                            arguments = J$.N(781073, 'arguments', arguments, true, false, false);
                            target = J$.N(781081, 'target', target, true, false, false);
                            props = J$.N(781089, 'props', props, true, false, false);
                            J$.N(781097, 'i', i, false, false, false);
                            J$.N(781105, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(780833, 'i', J$.T(780825, 0, 22, false), i, false, false); J$.C(45424, J$.B(90554, '<', J$.R(780841, 'i', i, false, false), J$.G(780857, J$.R(780849, 'props', props, false, false), 'length'))); J$.B(90578, '-', i = J$.W(780873, 'i', J$.B(90570, '+', J$.U(90562, '+', J$.R(780865, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(780905, 'descriptor', J$.G(780897, J$.R(780881, 'props', props, false, false), J$.R(780889, 'i', i, false, false)), descriptor, false, false);
                                J$.P(780945, J$.R(780913, 'descriptor', descriptor, false, false), 'enumerable', J$.C(45408, J$.G(780929, J$.R(780921, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(780937, false, 23, false));
                                J$.P(780969, J$.R(780953, 'descriptor', descriptor, false, false), 'configurable', J$.T(780961, true, 23, false));
                                if (J$.C(45416, J$.B(90586, 'in', J$.T(780977, 'value', 21, false), J$.R(780985, 'descriptor', descriptor, false, false))))
                                    J$.P(781009, J$.R(780993, 'descriptor', descriptor, false, false), 'writable', J$.T(781001, true, 23, false));
                                J$.M(781057, J$.I(typeof Object === 'undefined' ? Object = J$.R(781017, 'Object', undefined, true, true) : Object = J$.R(781017, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(781025, 'target', target, false, false), J$.G(781041, J$.R(781033, 'descriptor', descriptor, false, false), 'key'), J$.R(781049, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(784521, J$e);
                        } finally {
                            if (J$.Fr(784529))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(781217, arguments.callee, this, arguments);
                            arguments = J$.N(781225, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(781233, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(781241, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(781249, 'staticProps', staticProps, true, false, false);
                            if (J$.C(45432, J$.R(781113, 'protoProps', protoProps, false, false)))
                                J$.F(781153, J$.R(781121, '_defineProperties', _defineProperties, false, true), false)(J$.G(781137, J$.R(781129, 'Constructor', Constructor, false, false), 'prototype'), J$.R(781145, 'protoProps', protoProps, false, false));
                            if (J$.C(45440, J$.R(781161, 'staticProps', staticProps, false, false)))
                                J$.F(781193, J$.R(781169, '_defineProperties', _defineProperties, false, true), false)(J$.R(781177, 'Constructor', Constructor, false, false), J$.R(781185, 'staticProps', staticProps, false, false));
                            return J$.Rt(781209, J$.R(781201, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(784537, J$e);
                        } finally {
                            if (J$.Fr(784545))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function matchesParentDomain(srvAddress, parentDomain) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(781521, arguments.callee, this, arguments);
                            arguments = J$.N(781529, 'arguments', arguments, true, false, false);
                            srvAddress = J$.N(781537, 'srvAddress', srvAddress, true, false, false);
                            parentDomain = J$.N(781545, 'parentDomain', parentDomain, true, false, false);
                            J$.N(781553, 'regex', regex, false, false, false);
                            J$.N(781561, 'srv', srv, false, false, false);
                            J$.N(781569, 'parent', parent, false, false, false);
                            var regex = J$.W(781369, 'regex', J$.T(781361, /^.*?\./, 14, false), regex, false, false);
                            var srv = J$.W(781425, 'srv', J$.M(781417, J$.T(781377, '.', 21, false), 'concat', false)(J$.M(781409, J$.R(781385, 'srvAddress', srvAddress, false, false), 'replace', false)(J$.R(781393, 'regex', regex, false, false), J$.T(781401, '', 21, false))), srv, false, false);
                            var parent = J$.W(781481, 'parent', J$.M(781473, J$.T(781433, '.', 21, false), 'concat', false)(J$.M(781465, J$.R(781441, 'parentDomain', parentDomain, false, false), 'replace', false)(J$.R(781449, 'regex', regex, false, false), J$.T(781457, '', 21, false))), parent, false, false);
                            return J$.Rt(781513, J$.M(781505, J$.R(781489, 'srv', srv, false, false), 'endsWith', false)(J$.R(781497, 'parent', parent, false, false)));
                        } catch (J$e) {
                            J$.Ex(784553, J$e);
                        } finally {
                            if (J$.Fr(784561))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(784089, '_typeof2', J$.T(784081, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(784105, '_typeof', J$.T(784097, _typeof, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(784121, '_possibleConstructorReturn', J$.T(784113, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(784137, '_assertThisInitialized', J$.T(784129, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(784153, '_getPrototypeOf', J$.T(784145, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(784169, '_inherits', J$.T(784161, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(784185, '_setPrototypeOf', J$.T(784177, _setPrototypeOf, 12, false), true, false, false);
            _classCallCheck = J$.N(784201, '_classCallCheck', J$.T(784193, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(784217, '_defineProperties', J$.T(784209, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(784233, '_createClass', J$.T(784225, _createClass, 12, false), true, false, false);
            J$.N(784241, 'Logger', Logger, false, false, false);
            J$.N(784249, 'EventEmitter', EventEmitter, false, false, false);
            J$.N(784257, 'dns', dns, false, false, false);
            matchesParentDomain = J$.N(784273, 'matchesParentDomain', J$.T(784265, matchesParentDomain, 12, false), true, false, false);
            J$.N(784281, 'SrvPollingEvent', SrvPollingEvent, false, false, false);
            J$.N(784289, 'SrvPoller', SrvPoller, false, false, false);
            J$.T(779329, 'use strict', 21, false);
            var Logger = J$.W(781281, 'Logger', J$.F(781273, J$.I(typeof require === 'undefined' ? require = J$.R(781257, 'require', undefined, true, true) : require = J$.R(781257, 'require', require, true, true)), false)(J$.T(781265, '../connection/logger', 21, false)), Logger, false, true);
            var EventEmitter = J$.W(781321, 'EventEmitter', J$.G(781313, J$.F(781305, J$.I(typeof require === 'undefined' ? require = J$.R(781289, 'require', undefined, true, true) : require = J$.R(781289, 'require', require, true, true)), false)(J$.T(781297, 'events', 21, false)), 'EventEmitter'), EventEmitter, false, true);
            var dns = J$.W(781353, 'dns', J$.F(781345, J$.I(typeof require === 'undefined' ? require = J$.R(781329, 'require', undefined, true, true) : require = J$.R(781329, 'require', require, true, true)), false)(J$.T(781337, 'dns', 21, false)), dns, false, true);
            var SrvPollingEvent = J$.W(781945, 'SrvPollingEvent', J$.F(781937, J$.T(781929, function () {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(781897, arguments.callee, this, arguments);
                            function SrvPollingEvent(srvRecords) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(781633, arguments.callee, this, arguments);
                                            arguments = J$.N(781641, 'arguments', arguments, true, false, false);
                                            srvRecords = J$.N(781649, 'srvRecords', srvRecords, true, false, false);
                                            J$.F(781601, J$.R(781577, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(781585, 'this', this, false, false), J$.R(781593, 'SrvPollingEvent', SrvPollingEvent, false, false));
                                            J$.P(781625, J$.R(781609, 'this', this, false, false), 'srvRecords', J$.R(781617, 'srvRecords', srvRecords, false, false));
                                        } catch (J$e) {
                                            J$.Ex(784569, J$e);
                                        } finally {
                                            if (J$.Fr(784577))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(781905, 'arguments', arguments, true, false, false);
                            SrvPollingEvent = J$.N(781921, 'SrvPollingEvent', J$.T(781913, SrvPollingEvent, 12, false), true, false, false);
                            J$.F(781873, J$.R(781657, '_createClass', _createClass, false, true), false)(J$.R(781665, 'SrvPollingEvent', SrvPollingEvent, false, false), J$.T(781865, [J$.T(781857, {
                                    key: J$.T(781673, 'addresses', 21, false),
                                    value: J$.T(781849, function addresses() {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(781833, arguments.callee, this, arguments);
                                                    arguments = J$.N(781841, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(781825, J$.F(781817, J$.I(typeof Set === 'undefined' ? Set = J$.R(781681, 'Set', undefined, true, true) : Set = J$.R(781681, 'Set', Set, true, true)), true)(J$.M(781809, J$.G(781697, J$.R(781689, 'this', this, false, false), 'srvRecords'), 'map', false)(J$.T(781801, function (record) {
                                                        jalangiLabel18:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(781777, arguments.callee, this, arguments);
                                                                    arguments = J$.N(781785, 'arguments', arguments, true, false, false);
                                                                    record = J$.N(781793, 'record', record, true, false, false);
                                                                    return J$.Rt(781769, J$.M(781761, J$.M(781737, J$.T(781705, '', 21, false), 'concat', false)(J$.G(781721, J$.R(781713, 'record', record, false, false), 'name'), J$.T(781729, ':', 21, false)), 'concat', false)(J$.G(781753, J$.R(781745, 'record', record, false, false), 'port')));
                                                                } catch (J$e) {
                                                                    J$.Ex(784585, J$e);
                                                                } finally {
                                                                    if (J$.Fr(784593))
                                                                        continue jalangiLabel18;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false))));
                                                } catch (J$e) {
                                                    J$.Ex(784601, J$e);
                                                } finally {
                                                    if (J$.Fr(784609))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(781889, J$.R(781881, 'SrvPollingEvent', SrvPollingEvent, false, false));
                        } catch (J$e) {
                            J$.Ex(784617, J$e);
                        } finally {
                            if (J$.Fr(784625))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), SrvPollingEvent, false, true);
            var SrvPoller = J$.W(784001, 'SrvPoller', J$.F(783993, J$.T(783977, function (_EventEmitter) {
                jalangiLabel34:
                    while (true) {
                        try {
                            J$.Fe(783937, arguments.callee, this, arguments);
                            function SrvPoller(options) {
                                jalangiLabel21:
                                    while (true) {
                                        try {
                                            J$.Fe(782369, arguments.callee, this, arguments);
                                            arguments = J$.N(782377, 'arguments', arguments, true, false, false);
                                            options = J$.N(782385, 'options', options, true, false, false);
                                            J$.N(782393, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(782009, J$.R(781985, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(781993, 'this', this, false, false), J$.R(782001, 'SrvPoller', SrvPoller, false, false));
                                            _this = J$.W(782081, '_this', J$.F(782073, J$.R(782017, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(782025, 'this', this, false, false), J$.M(782065, J$.F(782049, J$.R(782033, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(782041, 'SrvPoller', SrvPoller, false, false)), 'call', false)(J$.R(782057, 'this', this, false, false))), _this, false, false);
                                            if (J$.C(45456, J$.C(45448, J$.U(90594, '!', J$.R(782089, 'options', options, false, false))) ? J$._() : J$.U(90602, '!', J$.G(782105, J$.R(782097, 'options', options, false, false), 'srvHost')))) {
                                                throw J$.F(782129, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(782113, 'TypeError', undefined, true, true) : TypeError = J$.R(782113, 'TypeError', TypeError, true, true)), true)(J$.T(782121, 'options for SrvPoller must exist and include srvHost', 21, false));
                                            }
                                            J$.P(782161, J$.R(782137, '_this', _this, false, false), 'srvHost', J$.G(782153, J$.R(782145, 'options', options, false, false), 'srvHost'));
                                            J$.P(782185, J$.R(782169, '_this', _this, false, false), 'rescanSrvIntervalMS', J$.T(782177, 60000, 22, false));
                                            J$.P(782225, J$.R(782193, '_this', _this, false, false), 'heartbeatFrequencyMS', J$.C(45464, J$.G(782209, J$.R(782201, 'options', options, false, false), 'heartbeatFrequencyMS')) ? J$._() : J$.T(782217, 10000, 22, false));
                                            J$.P(782273, J$.R(782233, '_this', _this, false, false), 'logger', J$.F(782265, J$.R(782241, 'Logger', Logger, false, true), false)(J$.T(782249, 'srvPoller', 21, false), J$.R(782257, 'options', options, false, false)));
                                            J$.P(782297, J$.R(782281, '_this', _this, false, false), 'haMode', J$.T(782289, false, 23, false));
                                            J$.P(782321, J$.R(782305, '_this', _this, false, false), 'generation', J$.T(782313, 0, 22, false));
                                            J$.P(782345, J$.R(782329, '_this', _this, false, false), '_timeout', J$.T(782337, null, 25, false));
                                            return J$.Rt(782361, J$.R(782353, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(784633, J$e);
                                        } finally {
                                            if (J$.Fr(784641))
                                                continue jalangiLabel21;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(783945, 'arguments', arguments, true, false, false);
                            _EventEmitter = J$.N(783953, '_EventEmitter', _EventEmitter, true, false, false);
                            SrvPoller = J$.N(783969, 'SrvPoller', J$.T(783961, SrvPoller, 12, false), true, false, false);
                            J$.F(781977, J$.R(781953, '_inherits', _inherits, false, true), false)(J$.R(781961, 'SrvPoller', SrvPoller, false, false), J$.R(781969, '_EventEmitter', _EventEmitter, false, false));
                            J$.F(783913, J$.R(782401, '_createClass', _createClass, false, true), false)(J$.R(782409, 'SrvPoller', SrvPoller, false, false), J$.T(783905, [
                                J$.T(782481, {
                                    key: J$.T(782417, 'start', 21, false),
                                    value: J$.T(782473, function start() {
                                        jalangiLabel22:
                                            while (true) {
                                                try {
                                                    J$.Fe(782457, arguments.callee, this, arguments);
                                                    arguments = J$.N(782465, 'arguments', arguments, true, false, false);
                                                    if (J$.C(45472, J$.U(90610, '!', J$.G(782433, J$.R(782425, 'this', this, false, false), '_timeout')))) {
                                                        J$.M(782449, J$.R(782441, 'this', this, false, false), 'schedule', false)();
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(784649, J$e);
                                                } finally {
                                                    if (J$.Fr(784657))
                                                        continue jalangiLabel22;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(782617, {
                                    key: J$.T(782489, 'stop', 21, false),
                                    value: J$.T(782609, function stop() {
                                        jalangiLabel23:
                                            while (true) {
                                                try {
                                                    J$.Fe(782593, arguments.callee, this, arguments);
                                                    arguments = J$.N(782601, 'arguments', arguments, true, false, false);
                                                    if (J$.C(45480, J$.G(782505, J$.R(782497, 'this', this, false, false), '_timeout'))) {
                                                        J$.F(782537, J$.I(typeof clearTimeout === 'undefined' ? clearTimeout = J$.R(782513, 'clearTimeout', undefined, true, true) : clearTimeout = J$.R(782513, 'clearTimeout', clearTimeout, true, true)), false)(J$.G(782529, J$.R(782521, 'this', this, false, false), '_timeout'));
                                                        J$.A(782561, J$.R(782545, 'this', this, false, false), 'generation', '+')(J$.T(782553, 1, 22, false));
                                                        J$.P(782585, J$.R(782569, 'this', this, false, false), '_timeout', J$.T(782577, null, 25, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(784665, J$e);
                                                } finally {
                                                    if (J$.Fr(784673))
                                                        continue jalangiLabel23;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(782809, {
                                    key: J$.T(782625, 'schedule', 21, false),
                                    value: J$.T(782801, function schedule() {
                                        jalangiLabel25:
                                            while (true) {
                                                try {
                                                    J$.Fe(782777, arguments.callee, this, arguments);
                                                    arguments = J$.N(782785, 'arguments', arguments, true, false, false);
                                                    J$.N(782793, '_this2', _this2, false, false, false);
                                                    var _this2 = J$.W(782641, '_this2', J$.R(782633, 'this', this, false, false), _this2, false, false);
                                                    J$.F(782673, J$.I(typeof clearTimeout === 'undefined' ? clearTimeout = J$.R(782649, 'clearTimeout', undefined, true, true) : clearTimeout = J$.R(782649, 'clearTimeout', clearTimeout, true, true)), false)(J$.G(782665, J$.R(782657, 'this', this, false, false), '_timeout'));
                                                    J$.P(782769, J$.R(782681, 'this', this, false, false), '_timeout', J$.F(782761, J$.I(typeof setTimeout === 'undefined' ? setTimeout = J$.R(782689, 'setTimeout', undefined, true, true) : setTimeout = J$.R(782689, 'setTimeout', setTimeout, true, true)), false)(J$.T(782737, function () {
                                                        jalangiLabel24:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(782721, arguments.callee, this, arguments);
                                                                    arguments = J$.N(782729, 'arguments', arguments, true, false, false);
                                                                    return J$.Rt(782713, J$.M(782705, J$.R(782697, '_this2', _this2, false, false), '_poll', false)());
                                                                } catch (J$e) {
                                                                    J$.Ex(784681, J$e);
                                                                } finally {
                                                                    if (J$.Fr(784689))
                                                                        continue jalangiLabel24;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false), J$.G(782753, J$.R(782745, 'this', this, false, false), 'intervalMS')));
                                                } catch (J$e) {
                                                    J$.Ex(784697, J$e);
                                                } finally {
                                                    if (J$.Fr(784705))
                                                        continue jalangiLabel25;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(782945, {
                                    key: J$.T(782817, 'success', 21, false),
                                    value: J$.T(782937, function success(srvRecords) {
                                        jalangiLabel26:
                                            while (true) {
                                                try {
                                                    J$.Fe(782913, arguments.callee, this, arguments);
                                                    arguments = J$.N(782921, 'arguments', arguments, true, false, false);
                                                    srvRecords = J$.N(782929, 'srvRecords', srvRecords, true, false, false);
                                                    J$.P(782841, J$.R(782825, 'this', this, false, false), 'haMode', J$.T(782833, false, 23, false));
                                                    J$.M(782857, J$.R(782849, 'this', this, false, false), 'schedule', false)();
                                                    J$.M(782905, J$.R(782865, 'this', this, false, false), 'emit', false)(J$.T(782873, 'srvRecordDiscovery', 21, false), J$.F(782897, J$.R(782881, 'SrvPollingEvent', SrvPollingEvent, false, true), true)(J$.R(782889, 'srvRecords', srvRecords, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(784713, J$e);
                                                } finally {
                                                    if (J$.Fr(784721))
                                                        continue jalangiLabel26;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(783081, {
                                    key: J$.T(782953, 'failure', 21, false),
                                    value: J$.T(783073, function failure(message, obj) {
                                        jalangiLabel27:
                                            while (true) {
                                                try {
                                                    J$.Fe(783041, arguments.callee, this, arguments);
                                                    arguments = J$.N(783049, 'arguments', arguments, true, false, false);
                                                    message = J$.N(783057, 'message', message, true, false, false);
                                                    obj = J$.N(783065, 'obj', obj, true, false, false);
                                                    J$.M(782993, J$.G(782969, J$.R(782961, 'this', this, false, false), 'logger'), 'warn', false)(J$.R(782977, 'message', message, false, false), J$.R(782985, 'obj', obj, false, false));
                                                    J$.P(783017, J$.R(783001, 'this', this, false, false), 'haMode', J$.T(783009, true, 23, false));
                                                    J$.M(783033, J$.R(783025, 'this', this, false, false), 'schedule', false)();
                                                } catch (J$e) {
                                                    J$.Ex(784729, J$e);
                                                } finally {
                                                    if (J$.Fr(784737))
                                                        continue jalangiLabel27;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(783233, {
                                    key: J$.T(783089, 'parentDomainMismatch', 21, false),
                                    value: J$.T(783225, function parentDomainMismatch(srvRecord) {
                                        jalangiLabel28:
                                            while (true) {
                                                try {
                                                    J$.Fe(783201, arguments.callee, this, arguments);
                                                    arguments = J$.N(783209, 'arguments', arguments, true, false, false);
                                                    srvRecord = J$.N(783217, 'srvRecord', srvRecord, true, false, false);
                                                    J$.M(783193, J$.G(783105, J$.R(783097, 'this', this, false, false), 'logger'), 'warn', false)(J$.M(783177, J$.M(783145, J$.T(783113, 'parent domain mismatch on SRV record (', 21, false), 'concat', false)(J$.G(783129, J$.R(783121, 'srvRecord', srvRecord, false, false), 'name'), J$.T(783137, ':', 21, false)), 'concat', false)(J$.G(783161, J$.R(783153, 'srvRecord', srvRecord, false, false), 'port'), J$.T(783169, ')', 21, false)), J$.R(783185, 'srvRecord', srvRecord, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(784745, J$e);
                                                } finally {
                                                    if (J$.Fr(784753))
                                                        continue jalangiLabel28;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(783721, {
                                    key: J$.T(783241, '_poll', 21, false),
                                    value: J$.T(783713, function _poll() {
                                        jalangiLabel31:
                                            while (true) {
                                                try {
                                                    J$.Fe(783681, arguments.callee, this, arguments);
                                                    arguments = J$.N(783689, 'arguments', arguments, true, false, false);
                                                    J$.N(783697, '_this3', _this3, false, false, false);
                                                    J$.N(783705, 'generation', generation, false, false, false);
                                                    var _this3 = J$.W(783257, '_this3', J$.R(783249, 'this', this, false, false), _this3, false, false);
                                                    var generation = J$.W(783281, 'generation', J$.G(783273, J$.R(783265, 'this', this, false, false), 'generation'), generation, false, false);
                                                    J$.M(783673, J$.R(783289, 'dns', dns, false, true), 'resolveSrv', false)(J$.G(783305, J$.R(783297, 'this', this, false, false), 'srvAddress'), J$.T(783665, function (err, srvRecords) {
                                                        jalangiLabel30:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(783625, arguments.callee, this, arguments);
                                                                    arguments = J$.N(783633, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(783641, 'err', err, true, false, false);
                                                                    srvRecords = J$.N(783649, 'srvRecords', srvRecords, true, false, false);
                                                                    J$.N(783657, 'finalAddresses', finalAddresses, false, false, false);
                                                                    if (J$.C(45488, J$.B(90618, '!==', J$.R(783313, 'generation', generation, false, false), J$.G(783329, J$.R(783321, '_this3', _this3, false, false), 'generation')))) {
                                                                        return J$.Rt(783337, undefined);
                                                                    }
                                                                    if (J$.C(45496, J$.R(783345, 'err', err, false, false))) {
                                                                        J$.M(783377, J$.R(783353, '_this3', _this3, false, false), 'failure', false)(J$.T(783361, 'DNS error', 21, false), J$.R(783369, 'err', err, false, false));
                                                                        return J$.Rt(783385, undefined);
                                                                    }
                                                                    var finalAddresses = J$.W(783401, 'finalAddresses', J$.T(783393, [], 10, false), finalAddresses, false, false);
                                                                    J$.M(783545, J$.R(783409, 'srvRecords', srvRecords, false, false), 'forEach', false)(J$.T(783537, function (record) {
                                                                        jalangiLabel29:
                                                                            while (true) {
                                                                                try {
                                                                                    J$.Fe(783513, arguments.callee, this, arguments);
                                                                                    arguments = J$.N(783521, 'arguments', arguments, true, false, false);
                                                                                    record = J$.N(783529, 'record', record, true, false, false);
                                                                                    if (J$.C(45504, J$.F(783457, J$.R(783417, 'matchesParentDomain', matchesParentDomain, false, true), false)(J$.G(783433, J$.R(783425, 'record', record, false, false), 'name'), J$.G(783449, J$.R(783441, '_this3', _this3, false, false), 'srvHost')))) {
                                                                                        J$.M(783481, J$.R(783465, 'finalAddresses', finalAddresses, false, false), 'push', false)(J$.R(783473, 'record', record, false, false));
                                                                                    } else {
                                                                                        J$.M(783505, J$.R(783489, '_this3', _this3, false, false), 'parentDomainMismatch', false)(J$.R(783497, 'record', record, false, false));
                                                                                    }
                                                                                } catch (J$e) {
                                                                                    J$.Ex(784761, J$e);
                                                                                } finally {
                                                                                    if (J$.Fr(784769))
                                                                                        continue jalangiLabel29;
                                                                                    else
                                                                                        return J$.Ra();
                                                                                }
                                                                            }
                                                                    }, 12, false));
                                                                    if (J$.C(45512, J$.U(90626, '!', J$.G(783561, J$.R(783553, 'finalAddresses', finalAddresses, false, false), 'length')))) {
                                                                        J$.M(783585, J$.R(783569, '_this3', _this3, false, false), 'failure', false)(J$.T(783577, 'No valid addresses found at host', 21, false));
                                                                        return J$.Rt(783593, undefined);
                                                                    }
                                                                    J$.M(783617, J$.R(783601, '_this3', _this3, false, false), 'success', false)(J$.R(783609, 'finalAddresses', finalAddresses, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(784777, J$e);
                                                                } finally {
                                                                    if (J$.Fr(784785))
                                                                        continue jalangiLabel30;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(784793, J$e);
                                                } finally {
                                                    if (J$.Fr(784801))
                                                        continue jalangiLabel31;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(783801, {
                                    key: J$.T(783729, 'srvAddress', 21, false),
                                    get: J$.T(783793, function get() {
                                        jalangiLabel32:
                                            while (true) {
                                                try {
                                                    J$.Fe(783777, arguments.callee, this, arguments);
                                                    arguments = J$.N(783785, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(783769, J$.M(783761, J$.T(783737, '_mongodb._tcp.', 21, false), 'concat', false)(J$.G(783753, J$.R(783745, 'this', this, false, false), 'srvHost')));
                                                } catch (J$e) {
                                                    J$.Ex(784809, J$e);
                                                } finally {
                                                    if (J$.Fr(784817))
                                                        continue jalangiLabel32;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(783897, {
                                    key: J$.T(783809, 'intervalMS', 21, false),
                                    get: J$.T(783889, function get() {
                                        jalangiLabel33:
                                            while (true) {
                                                try {
                                                    J$.Fe(783873, arguments.callee, this, arguments);
                                                    arguments = J$.N(783881, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(783865, J$.C(45520, J$.G(783825, J$.R(783817, 'this', this, false, false), 'haMode')) ? J$.G(783841, J$.R(783833, 'this', this, false, false), 'heartbeatFrequencyMS') : J$.G(783857, J$.R(783849, 'this', this, false, false), 'rescanSrvIntervalMs'));
                                                } catch (J$e) {
                                                    J$.Ex(784825, J$e);
                                                } finally {
                                                    if (J$.Fr(784833))
                                                        continue jalangiLabel33;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(783929, J$.R(783921, 'SrvPoller', SrvPoller, false, false));
                        } catch (J$e) {
                            J$.Ex(784841, J$e);
                        } finally {
                            if (J$.Fr(784849))
                                continue jalangiLabel34;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(783985, 'EventEmitter', EventEmitter, false, true)), SrvPoller, false, true);
            J$.P(784033, J$.G(784017, J$.I(typeof module === 'undefined' ? module = J$.R(784009, 'module', undefined, true, true) : module = J$.R(784009, 'module', module, true, true)), 'exports'), 'SrvPollingEvent', J$.R(784025, 'SrvPollingEvent', SrvPollingEvent, false, true));
            J$.P(784065, J$.G(784049, J$.I(typeof module === 'undefined' ? module = J$.R(784041, 'module', undefined, true, true) : module = J$.R(784041, 'module', module, true, true)), 'exports'), 'SrvPoller', J$.R(784057, 'SrvPoller', SrvPoller, false, true));
        } catch (J$e) {
            J$.Ex(784857, J$e);
        } finally {
            if (J$.Sr(784865))
                continue jalangiLabel35;
            else
                break jalangiLabel35;
        }
    }
// JALANGI DO NOT INSTRUMENT
