// schema-inspector/test/generator_test.js
J$.noInstrEval = false;
jalangiLabel13:
    while (true) {
        try {
            J$.Se(752249, '/mnt/data/fxiao/HiPar/outputs/target_cache/TestSchema-inspector/node_modules/schema-inspector/test/generator_test.js');
            function testRandomCandidates(schema) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(748073, arguments.callee, this, arguments);
                            arguments = J$.N(748081, 'arguments', arguments, true, false, false);
                            schema = J$.N(748089, 'schema', schema, true, false, false);
                            J$.N(748097, 'i', i, false, false, false);
                            for (var i = J$.W(747633, 'i', J$.T(747625, 1, 22, false), i, false, false); J$.C(46800, J$.B(64442, '<=', J$.R(747641, 'i', i, false, false), J$.R(747649, 'NB_TEST', NB_TEST, false, true))); J$.B(64466, '-', i = J$.W(747665, 'i', J$.B(64458, '+', J$.U(64450, '+', J$.R(747657, 'i', i, false, false)), 1), i, false, false), 1)) {
                                J$.F(748065, J$.I(typeof test === 'undefined' ? test = J$.R(747673, 'test', undefined, true, true) : test = J$.R(747673, 'test', test, true, true)), false)(J$.B(64474, '+', J$.T(747681, 'Random candidate #', 21, false), J$.R(747689, 'i', i, false, false)), J$.T(748057, function () {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(748025, arguments.callee, this, arguments);
                                                arguments = J$.N(748033, 'arguments', arguments, true, false, false);
                                                J$.N(748041, 'candidate', candidate, false, false, false);
                                                J$.N(748049, 'result', result, false, false, false);
                                                var candidate = J$.W(747721, 'candidate', J$.M(747713, J$.R(747697, 'si', si, false, true), 'generate', false)(J$.R(747705, 'schema', schema, false, false)), candidate, false, false);
                                                var result = J$.W(747761, 'result', J$.M(747753, J$.R(747729, 'si', si, false, true), 'validate', false)(J$.R(747737, 'schema', schema, false, false), J$.R(747745, 'candidate', candidate, false, false)), result, false, false);
                                                J$.G(747801, J$.G(747793, J$.G(747785, J$.G(747777, J$.R(747769, 'result', result, false, false), 'should'), 'be'), 'an'), 'Object');
                                                try {
                                                    J$.M(747865, J$.G(747849, J$.M(747841, J$.G(747825, J$.G(747817, J$.R(747809, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(747833, 'valid', 21, false)), 'with'), 'equal', false)(J$.T(747857, true, 23, false));
                                                    J$.M(747977, J$.G(747961, J$.G(747953, J$.M(747945, J$.G(747929, J$.G(747921, J$.G(747913, J$.M(747905, J$.G(747889, J$.G(747881, J$.R(747873, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(747897, 'error', 21, false)), 'with'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(747937, 'Array', undefined, true, true) : Array = J$.R(747937, 'Array', Array, true, true))), 'and'), 'be'), 'lengthOf', false)(J$.T(747969, 0, 22, false));
                                                } catch (e) {
                                                    J$.N(748017, 'e', e, false, false, true);
                                                    throw J$.F(748009, J$.I(typeof Error === 'undefined' ? Error = J$.R(747985, 'Error', undefined, true, true) : Error = J$.R(747985, 'Error', Error, true, true)), true)(J$.M(748001, J$.R(747993, 'result', result, false, false), 'format', false)());
                                                }
                                            } catch (J$e) {
                                                J$.Ex(752297, J$e);
                                            } finally {
                                                if (J$.Fr(752305))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            }
                        } catch (J$e) {
                            J$.Ex(752313, J$e);
                        } finally {
                            if (J$.Fr(752321))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(752257, 'should', should, false, false, false);
            J$.N(752265, 'si', si, false, false, false);
            J$.N(752273, 'NB_TEST', NB_TEST, false, false, false);
            testRandomCandidates = J$.N(752289, 'testRandomCandidates', J$.T(752281, testRandomCandidates, 12, false), true, false, false);
            var should = J$.W(747569, 'should', J$.F(747561, J$.I(typeof require === 'undefined' ? require = J$.R(747545, 'require', undefined, true, true) : require = J$.R(747545, 'require', require, true, true)), false)(J$.T(747553, 'should', 21, false)), should, false, true);
            var si = J$.W(747601, 'si', J$.F(747593, J$.I(typeof require === 'undefined' ? require = J$.R(747577, 'require', undefined, true, true) : require = J$.R(747577, 'require', require, true, true)), false)(J$.T(747585, '../', 21, false)), si, false, true);
            const NB_TEST = J$.W(747617, 'NB_TEST', J$.T(747609, 25, 22, false), NB_TEST, false, true);
            J$.P(752241, J$.I(typeof exports === 'undefined' ? exports = J$.R(748105, 'exports', undefined, true, true) : exports = J$.R(748105, 'exports', exports, true, true)), 'generator', J$.T(752233, function () {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(752217, arguments.callee, this, arguments);
                            arguments = J$.N(752225, 'arguments', arguments, true, false, false);
                            J$.F(748393, J$.I(typeof suite === 'undefined' ? suite = J$.R(748113, 'suite', undefined, true, true) : suite = J$.R(748113, 'suite', suite, true, true)), false)(J$.T(748121, 'Schema #1 (Basics schemas with nested object, only considering type)', 21, false), J$.T(748385, function () {
                                jalangiLabel2:
                                    while (true) {
                                        try {
                                            J$.Fe(748361, arguments.callee, this, arguments);
                                            arguments = J$.N(748369, 'arguments', arguments, true, false, false);
                                            J$.N(748377, 'schema', schema, false, false, false);
                                            var schema = J$.W(748329, 'schema', J$.T(748321, {
                                                type: J$.T(748129, 'object', 21, false),
                                                properties: J$.T(748313, {
                                                    lorem: J$.T(748305, {
                                                        type: J$.T(748137, 'object', 21, false),
                                                        properties: J$.T(748297, {
                                                            ipsum: J$.T(748169, {
                                                                type: J$.T(748161, [
                                                                    J$.T(748145, 'string', 21, false),
                                                                    J$.T(748153, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(748185, { type: J$.T(748177, 'number', 21, false) }, 11, false),
                                                            sit: J$.T(748201, { type: J$.T(748193, 'any', 21, false) }, 11, false),
                                                            amet: J$.T(748233, {
                                                                type: J$.T(748209, 'array', 21, false),
                                                                items: J$.T(748225, { type: J$.T(748217, 'string', 21, false) }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(748289, {
                                                                type: J$.T(748241, 'array', 21, false),
                                                                items: J$.T(748281, [
                                                                    J$.T(748257, { type: J$.T(748249, 'integer', 21, false) }, 11, false),
                                                                    J$.T(748273, { type: J$.T(748265, 'number', 21, false) }, 11, false)
                                                                ], 10, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(748353, J$.R(748337, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(748345, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(752329, J$e);
                                        } finally {
                                            if (J$.Fr(752337))
                                                continue jalangiLabel2;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(748705, J$.I(typeof suite === 'undefined' ? suite = J$.R(748401, 'suite', undefined, true, true) : suite = J$.R(748401, 'suite', suite, true, true)), false)(J$.T(748409, 'Schema #2 (Basic schemas, considering *Length)', 21, false), J$.T(748697, function () {
                                jalangiLabel3:
                                    while (true) {
                                        try {
                                            J$.Fe(748673, arguments.callee, this, arguments);
                                            arguments = J$.N(748681, 'arguments', arguments, true, false, false);
                                            J$.N(748689, 'schema', schema, false, false, false);
                                            var schema = J$.W(748641, 'schema', J$.T(748633, {
                                                type: J$.T(748417, 'object', 21, false),
                                                properties: J$.T(748625, {
                                                    lorem: J$.T(748617, {
                                                        type: J$.T(748425, 'object', 21, false),
                                                        properties: J$.T(748609, {
                                                            ipsum: J$.T(748457, {
                                                                type: J$.T(748449, [
                                                                    J$.T(748433, 'string', 21, false),
                                                                    J$.T(748441, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(748473, { type: J$.T(748465, 'number', 21, false) }, 11, false),
                                                            sit: J$.T(748489, { type: J$.T(748481, 'any', 21, false) }, 11, false),
                                                            amet: J$.T(748537, {
                                                                type: J$.T(748497, 'array', 21, false),
                                                                exactLength: J$.T(748505, 6, 22, false),
                                                                items: J$.T(748529, {
                                                                    type: J$.T(748513, 'string', 21, false),
                                                                    exactLength: J$.T(748521, 10, 22, false)
                                                                }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(748601, {
                                                                type: J$.T(748545, 'array', 21, false),
                                                                minLength: J$.T(748553, 4, 22, false),
                                                                maxLength: J$.T(748561, 8, 22, false),
                                                                items: J$.T(748593, {
                                                                    type: J$.T(748569, 'string', 21, false),
                                                                    minLength: J$.T(748577, 6, 22, false),
                                                                    maxLength: J$.T(748585, 12, 22, false)
                                                                }, 11, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(748665, J$.R(748649, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(748657, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(752345, J$e);
                                        } finally {
                                            if (J$.Fr(752353))
                                                continue jalangiLabel3;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(749433, J$.I(typeof suite === 'undefined' ? suite = J$.R(748713, 'suite', undefined, true, true) : suite = J$.R(748713, 'suite', suite, true, true)), false)(J$.T(748721, 'Schema #3 (Intermediate schemas, considering *Length, lt, lte, gt, gte, eq and ne)', 21, false), J$.T(749425, function () {
                                jalangiLabel4:
                                    while (true) {
                                        try {
                                            J$.Fe(749401, arguments.callee, this, arguments);
                                            arguments = J$.N(749409, 'arguments', arguments, true, false, false);
                                            J$.N(749417, 'schema', schema, false, false, false);
                                            var schema = J$.W(749369, 'schema', J$.T(749361, {
                                                type: J$.T(748729, 'object', 21, false),
                                                properties: J$.T(749353, {
                                                    lorem: J$.T(749345, {
                                                        type: J$.T(748737, 'object', 21, false),
                                                        properties: J$.T(749337, {
                                                            ipsum: J$.T(748769, {
                                                                type: J$.T(748761, [
                                                                    J$.T(748745, 'string', 21, false),
                                                                    J$.T(748753, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(748785, { type: J$.T(748777, 'number', 21, false) }, 11, false),
                                                            sit: J$.T(748801, { type: J$.T(748793, 'any', 21, false) }, 11, false),
                                                            amet: J$.T(748849, {
                                                                type: J$.T(748809, 'array', 21, false),
                                                                exactLength: J$.T(748817, 6, 22, false),
                                                                items: J$.T(748841, {
                                                                    type: J$.T(748825, 'string', 21, false),
                                                                    exactLength: J$.T(748833, 10, 22, false)
                                                                }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(749329, {
                                                                type: J$.T(748857, 'array', 21, false),
                                                                maxLength: J$.T(748865, 16, 22, false),
                                                                items: J$.T(749321, [
                                                                    J$.T(748889, {
                                                                        type: J$.T(748873, 'number', 21, false),
                                                                        gt: J$.T(748881, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(748913, {
                                                                        type: J$.T(748897, 'number', 21, false),
                                                                        lt: J$.T(748905, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(748937, {
                                                                        type: J$.T(748921, 'number', 21, false),
                                                                        gte: J$.T(748929, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(748961, {
                                                                        type: J$.T(748945, 'number', 21, false),
                                                                        lte: J$.T(748953, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(748993, {
                                                                        type: J$.T(748969, 'number', 21, false),
                                                                        gt: J$.T(748977, 5, 22, false),
                                                                        lt: J$.T(748985, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749025, {
                                                                        type: J$.T(749001, 'number', 21, false),
                                                                        gte: J$.T(749009, 5, 22, false),
                                                                        lte: J$.T(749017, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749049, {
                                                                        type: J$.T(749033, 'number', 21, false),
                                                                        eq: J$.T(749041, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749089, {
                                                                        type: J$.T(749057, 'number', 21, false),
                                                                        gte: J$.T(749065, 0, 22, false),
                                                                        lte: J$.T(749073, 1, 22, false),
                                                                        ne: J$.T(749081, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749113, {
                                                                        type: J$.T(749097, 'integer', 21, false),
                                                                        gt: J$.T(749105, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749137, {
                                                                        type: J$.T(749121, 'integer', 21, false),
                                                                        lt: J$.T(749129, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749161, {
                                                                        type: J$.T(749145, 'integer', 21, false),
                                                                        gte: J$.T(749153, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749185, {
                                                                        type: J$.T(749169, 'integer', 21, false),
                                                                        lte: J$.T(749177, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749217, {
                                                                        type: J$.T(749193, 'integer', 21, false),
                                                                        gt: J$.T(749201, 5, 22, false),
                                                                        lt: J$.T(749209, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749249, {
                                                                        type: J$.T(749225, 'integer', 21, false),
                                                                        gte: J$.T(749233, 5, 22, false),
                                                                        lte: J$.T(749241, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749273, {
                                                                        type: J$.T(749257, 'integer', 21, false),
                                                                        eq: J$.T(749265, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749313, {
                                                                        type: J$.T(749281, 'integer', 21, false),
                                                                        gte: J$.T(749289, 0, 22, false),
                                                                        lte: J$.T(749297, 1, 22, false),
                                                                        ne: J$.T(749305, 1, 22, false)
                                                                    }, 11, false)
                                                                ], 10, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(749393, J$.R(749377, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(749385, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(752361, J$e);
                                        } finally {
                                            if (J$.Fr(752369))
                                                continue jalangiLabel4;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(750369, J$.I(typeof suite === 'undefined' ? suite = J$.R(749441, 'suite', undefined, true, true) : suite = J$.R(749441, 'suite', suite, true, true)), false)(J$.T(749449, 'Schema #4 (Advanced schemas, considering *Length, lt, lte, gt, gte, eq, array of ne and optional fields)', 21, false), J$.T(750361, function () {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(750337, arguments.callee, this, arguments);
                                            arguments = J$.N(750345, 'arguments', arguments, true, false, false);
                                            J$.N(750353, 'schema', schema, false, false, false);
                                            var schema = J$.W(750305, 'schema', J$.T(750297, {
                                                type: J$.T(749457, 'object', 21, false),
                                                properties: J$.T(750289, {
                                                    lorem: J$.T(750281, {
                                                        type: J$.T(749465, 'object', 21, false),
                                                        properties: J$.T(750273, {
                                                            ipsum: J$.T(749497, {
                                                                type: J$.T(749489, [
                                                                    J$.T(749473, 'string', 21, false),
                                                                    J$.T(749481, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(749521, {
                                                                type: J$.T(749505, 'number', 21, false),
                                                                optional: J$.T(749513, true, 23, false)
                                                            }, 11, false),
                                                            sit: J$.T(749545, {
                                                                type: J$.T(749529, 'any', 21, false),
                                                                optional: J$.T(749537, true, 23, false)
                                                            }, 11, false),
                                                            amet: J$.T(749609, {
                                                                type: J$.T(749553, 'array', 21, false),
                                                                exactLength: J$.T(749561, 6, 22, false),
                                                                optional: J$.T(749569, true, 23, false),
                                                                items: J$.T(749601, {
                                                                    type: J$.T(749577, 'string', 21, false),
                                                                    optional: J$.T(749585, true, 23, false),
                                                                    exactLength: J$.T(749593, 10, 22, false)
                                                                }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(750265, {
                                                                type: J$.T(749617, 'array', 21, false),
                                                                maxLength: J$.T(749625, 18, 22, false),
                                                                optional: J$.T(749633, true, 23, false),
                                                                items: J$.T(750257, [
                                                                    J$.T(749657, {
                                                                        type: J$.T(749641, 'number', 21, false),
                                                                        gt: J$.T(749649, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749681, {
                                                                        type: J$.T(749665, 'number', 21, false),
                                                                        lt: J$.T(749673, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749705, {
                                                                        type: J$.T(749689, 'number', 21, false),
                                                                        gte: J$.T(749697, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749729, {
                                                                        type: J$.T(749713, 'number', 21, false),
                                                                        lte: J$.T(749721, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749761, {
                                                                        type: J$.T(749737, 'number', 21, false),
                                                                        gt: J$.T(749745, 5, 22, false),
                                                                        lt: J$.T(749753, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749793, {
                                                                        type: J$.T(749769, 'number', 21, false),
                                                                        gte: J$.T(749777, 5, 22, false),
                                                                        lte: J$.T(749785, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749817, {
                                                                        type: J$.T(749801, 'number', 21, false),
                                                                        eq: J$.T(749809, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749873, {
                                                                        type: J$.T(749825, 'number', 21, false),
                                                                        gte: J$.T(749833, 0, 22, false),
                                                                        lte: J$.T(749841, 2, 22, false),
                                                                        ne: J$.T(749865, [
                                                                            J$.T(749849, 0, 22, false),
                                                                            J$.T(749857, 1, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false),
                                                                    J$.T(749897, {
                                                                        type: J$.T(749881, 'integer', 21, false),
                                                                        gt: J$.T(749889, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749921, {
                                                                        type: J$.T(749905, 'integer', 21, false),
                                                                        lt: J$.T(749913, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749945, {
                                                                        type: J$.T(749929, 'integer', 21, false),
                                                                        gte: J$.T(749937, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(749969, {
                                                                        type: J$.T(749953, 'integer', 21, false),
                                                                        lte: J$.T(749961, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(750001, {
                                                                        type: J$.T(749977, 'integer', 21, false),
                                                                        gt: J$.T(749985, 5, 22, false),
                                                                        lt: J$.T(749993, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(750033, {
                                                                        type: J$.T(750009, 'integer', 21, false),
                                                                        gte: J$.T(750017, 5, 22, false),
                                                                        lte: J$.T(750025, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(750057, {
                                                                        type: J$.T(750041, 'integer', 21, false),
                                                                        eq: J$.T(750049, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(750113, {
                                                                        type: J$.T(750065, 'integer', 21, false),
                                                                        gte: J$.T(750073, 0, 22, false),
                                                                        lte: J$.T(750081, 2, 22, false),
                                                                        ne: J$.T(750105, [
                                                                            J$.T(750089, 0, 22, false),
                                                                            J$.T(750097, 1, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false),
                                                                    J$.T(750177, {
                                                                        type: J$.T(750121, 'integer', 21, false),
                                                                        gte: J$.T(750129, 0, 22, false),
                                                                        lte: J$.T(750137, 3, 22, false),
                                                                        ne: J$.T(750169, [
                                                                            J$.T(750145, 0, 22, false),
                                                                            J$.T(750153, 1, 22, false),
                                                                            J$.T(750161, 2, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false),
                                                                    J$.T(750249, {
                                                                        type: J$.T(750185, 'integer', 21, false),
                                                                        gte: J$.T(750193, 0, 22, false),
                                                                        lte: J$.T(750201, 4, 22, false),
                                                                        ne: J$.T(750241, [
                                                                            J$.T(750209, 0, 22, false),
                                                                            J$.T(750217, 1, 22, false),
                                                                            J$.T(750225, 2, 22, false),
                                                                            J$.T(750233, 3, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false)
                                                                ], 10, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(750329, J$.R(750313, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(750321, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(752377, J$e);
                                        } finally {
                                            if (J$.Fr(752385))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(750825, J$.I(typeof suite === 'undefined' ? suite = J$.R(750377, 'suite', undefined, true, true) : suite = J$.R(750377, 'suite', suite, true, true)), false)(J$.T(750385, 'Schema #5 (Advanced schemas, considering *Length, eq and format fields)', 21, false), J$.T(750817, function () {
                                jalangiLabel6:
                                    while (true) {
                                        try {
                                            J$.Fe(750793, arguments.callee, this, arguments);
                                            arguments = J$.N(750801, 'arguments', arguments, true, false, false);
                                            J$.N(750809, 'schema', schema, false, false, false);
                                            var schema = J$.W(750761, 'schema', J$.T(750753, {
                                                type: J$.T(750393, 'object', 21, false),
                                                properties: J$.T(750745, {
                                                    lorem: J$.T(750737, {
                                                        type: J$.T(750401, 'array', 21, false),
                                                        items: J$.T(750729, [
                                                            J$.T(750425, {
                                                                type: J$.T(750409, 'string', 21, false),
                                                                pattern: J$.T(750417, 'date-time', 21, false)
                                                            }, 11, false),
                                                            J$.T(750449, {
                                                                type: J$.T(750433, 'string', 21, false),
                                                                pattern: J$.T(750441, 'date-time', 21, false)
                                                            }, 11, false),
                                                            J$.T(750473, {
                                                                type: J$.T(750457, 'string', 21, false),
                                                                pattern: J$.T(750465, 'date', 21, false)
                                                            }, 11, false),
                                                            J$.T(750497, {
                                                                type: J$.T(750481, 'string', 21, false),
                                                                pattern: J$.T(750489, 'time', 21, false)
                                                            }, 11, false),
                                                            J$.T(750521, {
                                                                type: J$.T(750505, 'string', 21, false),
                                                                pattern: J$.T(750513, 'color', 21, false)
                                                            }, 11, false),
                                                            J$.T(750553, {
                                                                type: J$.T(750529, 'string', 21, false),
                                                                pattern: J$.T(750537, 'color', 21, false),
                                                                eq: J$.T(750545, '#FFffFF', 21, false)
                                                            }, 11, false),
                                                            J$.T(750577, {
                                                                type: J$.T(750561, 'string', 21, false),
                                                                pattern: J$.T(750569, 'numeric', 21, false)
                                                            }, 11, false),
                                                            J$.T(750601, {
                                                                type: J$.T(750585, 'string', 21, false),
                                                                pattern: J$.T(750593, 'integer', 21, false)
                                                            }, 11, false),
                                                            J$.T(750625, {
                                                                type: J$.T(750609, 'string', 21, false),
                                                                pattern: J$.T(750617, 'decimal', 21, false)
                                                            }, 11, false),
                                                            J$.T(750649, {
                                                                type: J$.T(750633, 'string', 21, false),
                                                                pattern: J$.T(750641, 'alpha', 21, false)
                                                            }, 11, false),
                                                            J$.T(750673, {
                                                                type: J$.T(750657, 'string', 21, false),
                                                                pattern: J$.T(750665, 'alphaNumeric', 21, false)
                                                            }, 11, false),
                                                            J$.T(750697, {
                                                                type: J$.T(750681, 'string', 21, false),
                                                                pattern: J$.T(750689, 'alphaDash', 21, false)
                                                            }, 11, false),
                                                            J$.T(750721, {
                                                                type: J$.T(750705, 'string', 21, false),
                                                                pattern: J$.T(750713, 'javascript', 21, false)
                                                            }, 11, false)
                                                        ], 10, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(750785, J$.R(750769, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(750777, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(752393, J$e);
                                        } finally {
                                            if (J$.Fr(752401))
                                                continue jalangiLabel6;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(751473, J$.I(typeof suite === 'undefined' ? suite = J$.R(750833, 'suite', undefined, true, true) : suite = J$.R(750833, 'suite', suite, true, true)), false)(J$.T(750841, 'Schema #6 (Generate several candidate with one call)', 21, false), J$.T(751465, function () {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(751433, arguments.callee, this, arguments);
                                            arguments = J$.N(751441, 'arguments', arguments, true, false, false);
                                            J$.N(751449, 'schema', schema, false, false, false);
                                            J$.N(751457, 'candidates', candidates, false, false, false);
                                            var schema = J$.W(750889, 'schema', J$.T(750881, {
                                                type: J$.T(750849, 'object', 21, false),
                                                properties: J$.T(750873, { lorem: J$.T(750865, { type: J$.T(750857, 'integer', 21, false) }, 11, false) }, 11, false)
                                            }, 11, false), schema, false, false);
                                            var candidates = J$.W(750929, 'candidates', J$.M(750921, J$.R(750897, 'si', si, false, true), 'generate', false)(J$.R(750905, 'schema', schema, false, false), J$.R(750913, 'NB_TEST', NB_TEST, false, true)), candidates, false, false);
                                            J$.M(751001, J$.G(750985, J$.M(750977, J$.G(750961, J$.G(750953, J$.G(750945, J$.R(750937, 'candidates', candidates, false, false), 'should'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(750969, 'Array', undefined, true, true) : Array = J$.R(750969, 'Array', Array, true, true))), 'with'), 'lengthOf', false)(J$.R(750993, 'NB_TEST', NB_TEST, false, true));
                                            J$.M(751425, J$.R(751009, 'candidates', candidates, false, false), 'forEach', false)(J$.T(751417, function (candidate, i) {
                                                jalangiLabel8:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(751385, arguments.callee, this, arguments);
                                                            arguments = J$.N(751393, 'arguments', arguments, true, false, false);
                                                            candidate = J$.N(751401, 'candidate', candidate, true, false, false);
                                                            i = J$.N(751409, 'i', i, true, false, false);
                                                            J$.F(751377, J$.I(typeof test === 'undefined' ? test = J$.R(751017, 'test', undefined, true, true) : test = J$.R(751017, 'test', test, true, true)), false)(J$.B(64490, '+', J$.T(751025, 'Random candidate #', 21, false), J$.B(64482, '+', J$.R(751033, 'i', i, false, false), J$.T(751041, 1, 22, false))), J$.T(751369, function () {
                                                                jalangiLabel7:
                                                                    while (true) {
                                                                        try {
                                                                            J$.Fe(751345, arguments.callee, this, arguments);
                                                                            arguments = J$.N(751353, 'arguments', arguments, true, false, false);
                                                                            J$.N(751361, 'result', result, false, false, false);
                                                                            var result = J$.W(751081, 'result', J$.M(751073, J$.R(751049, 'si', si, false, true), 'validate', false)(J$.R(751057, 'schema', schema, false, false), J$.R(751065, 'candidate', candidate, false, false)), result, false, false);
                                                                            J$.G(751121, J$.G(751113, J$.G(751105, J$.G(751097, J$.R(751089, 'result', result, false, false), 'should'), 'be'), 'an'), 'Object');
                                                                            try {
                                                                                J$.M(751185, J$.G(751169, J$.M(751161, J$.G(751145, J$.G(751137, J$.R(751129, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(751153, 'valid', 21, false)), 'with'), 'equal', false)(J$.T(751177, true, 23, false));
                                                                                J$.M(751297, J$.G(751281, J$.G(751273, J$.M(751265, J$.G(751249, J$.G(751241, J$.G(751233, J$.M(751225, J$.G(751209, J$.G(751201, J$.R(751193, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(751217, 'error', 21, false)), 'with'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(751257, 'Array', undefined, true, true) : Array = J$.R(751257, 'Array', Array, true, true))), 'and'), 'be'), 'lengthOf', false)(J$.T(751289, 0, 22, false));
                                                                            } catch (e) {
                                                                                J$.N(751337, 'e', e, false, false, true);
                                                                                throw J$.F(751329, J$.I(typeof Error === 'undefined' ? Error = J$.R(751305, 'Error', undefined, true, true) : Error = J$.R(751305, 'Error', Error, true, true)), true)(J$.M(751321, J$.R(751313, 'result', result, false, false), 'format', false)());
                                                                            }
                                                                        } catch (J$e) {
                                                                            J$.Ex(752409, J$e);
                                                                        } finally {
                                                                            if (J$.Fr(752417))
                                                                                continue jalangiLabel7;
                                                                            else
                                                                                return J$.Ra();
                                                                        }
                                                                    }
                                                            }, 12, false));
                                                        } catch (J$e) {
                                                            J$.Ex(752425, J$e);
                                                        } finally {
                                                            if (J$.Fr(752433))
                                                                continue jalangiLabel8;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(752441, J$e);
                                        } finally {
                                            if (J$.Fr(752449))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(752209, J$.I(typeof suite === 'undefined' ? suite = J$.R(751481, 'suite', undefined, true, true) : suite = J$.R(751481, 'suite', suite, true, true)), false)(J$.T(751489, 'Schema #7 (Globing for Object keys)', 21, false), J$.T(752201, function () {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(752169, arguments.callee, this, arguments);
                                            arguments = J$.N(752177, 'arguments', arguments, true, false, false);
                                            J$.N(752185, 'schema', schema, false, false, false);
                                            J$.N(752193, 'i', i, false, false, false);
                                            var schema = J$.W(751601, 'schema', J$.T(751593, {
                                                type: J$.T(751497, 'object', 21, false),
                                                properties: J$.T(751585, {
                                                    globString: J$.T(751537, {
                                                        type: J$.T(751505, 'object', 21, false),
                                                        properties: J$.T(751529, { '*': J$.T(751521, { type: J$.T(751513, 'string', 21, false) }, 11, false) }, 11, false)
                                                    }, 11, false),
                                                    globInteger: J$.T(751577, {
                                                        type: J$.T(751545, 'object', 21, false),
                                                        properties: J$.T(751569, { '*': J$.T(751561, { type: J$.T(751553, 'integer', 21, false) }, 11, false) }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            for (var i = J$.W(751617, 'i', J$.T(751609, 1, 22, false), i, false, false); J$.C(46808, J$.B(64498, '<=', J$.R(751625, 'i', i, false, false), J$.R(751633, 'NB_TEST', NB_TEST, false, true))); J$.B(64522, '-', i = J$.W(751649, 'i', J$.B(64514, '+', J$.U(64506, '+', J$.R(751641, 'i', i, false, false)), 1), i, false, false), 1)) {
                                                J$.F(752161, J$.I(typeof test === 'undefined' ? test = J$.R(751657, 'test', undefined, true, true) : test = J$.R(751657, 'test', test, true, true)), false)(J$.B(64530, '+', J$.T(751665, 'Random candidate #', 21, false), J$.R(751673, 'i', i, false, false)), J$.T(752153, function () {
                                                    jalangiLabel10:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(752121, arguments.callee, this, arguments);
                                                                arguments = J$.N(752129, 'arguments', arguments, true, false, false);
                                                                J$.N(752137, 'candidate', candidate, false, false, false);
                                                                J$.N(752145, 'result', result, false, false, false);
                                                                var candidate = J$.W(751705, 'candidate', J$.M(751697, J$.R(751681, 'si', si, false, true), 'generate', false)(J$.R(751689, 'schema', schema, false, false)), candidate, false, false);
                                                                var result = J$.W(751745, 'result', J$.M(751737, J$.R(751713, 'si', si, false, true), 'validate', false)(J$.R(751721, 'schema', schema, false, false), J$.R(751729, 'candidate', candidate, false, false)), result, false, false);
                                                                J$.G(751785, J$.G(751777, J$.G(751769, J$.G(751761, J$.R(751753, 'result', result, false, false), 'should'), 'be'), 'an'), 'Object');
                                                                J$.M(751841, J$.G(751825, J$.G(751817, J$.G(751809, J$.G(751801, J$.R(751793, 'candidate', candidate, false, false), 'globString'), 'should'), 'not'), 'have'), 'property', false)(J$.T(751833, '*', 21, false));
                                                                J$.M(751897, J$.G(751881, J$.G(751873, J$.G(751865, J$.G(751857, J$.R(751849, 'candidate', candidate, false, false), 'globInteger'), 'should'), 'not'), 'have'), 'property', false)(J$.T(751889, '*', 21, false));
                                                                try {
                                                                    J$.M(751961, J$.G(751945, J$.M(751937, J$.G(751921, J$.G(751913, J$.R(751905, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(751929, 'valid', 21, false)), 'with'), 'equal', false)(J$.T(751953, true, 23, false));
                                                                    J$.M(752073, J$.G(752057, J$.G(752049, J$.M(752041, J$.G(752025, J$.G(752017, J$.G(752009, J$.M(752001, J$.G(751985, J$.G(751977, J$.R(751969, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(751993, 'error', 21, false)), 'with'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(752033, 'Array', undefined, true, true) : Array = J$.R(752033, 'Array', Array, true, true))), 'and'), 'be'), 'lengthOf', false)(J$.T(752065, 0, 22, false));
                                                                } catch (e) {
                                                                    J$.N(752113, 'e', e, false, false, true);
                                                                    throw J$.F(752105, J$.I(typeof Error === 'undefined' ? Error = J$.R(752081, 'Error', undefined, true, true) : Error = J$.R(752081, 'Error', Error, true, true)), true)(J$.M(752097, J$.R(752089, 'result', result, false, false), 'format', false)());
                                                                }
                                                            } catch (J$e) {
                                                                J$.Ex(752457, J$e);
                                                            } finally {
                                                                if (J$.Fr(752465))
                                                                    continue jalangiLabel10;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false));
                                            }
                                        } catch (J$e) {
                                            J$.Ex(752473, J$e);
                                        } finally {
                                            if (J$.Fr(752481))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(752489, J$e);
                        } finally {
                            if (J$.Fr(752497))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(752505, J$e);
        } finally {
            if (J$.Sr(752513))
                continue jalangiLabel13;
            else
                break jalangiLabel13;
        }
    }
// JALANGI DO NOT INSTRUMENT
