// schema-inspector/test/generator_test.js
J$.noInstrEval = false;
jalangiLabel13:
    while (true) {
        try {
            J$.Se(4728465, '/home/james/nodejs/HiPar/outputs/target_cache/TestSchema-inspector/node_modules/schema-inspector/test/generator_test.js');
            function testRandomCandidates(schema) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(4724289, arguments.callee, this, arguments);
                            arguments = J$.N(4724297, 'arguments', arguments, true, false, false);
                            schema = J$.N(4724305, 'schema', schema, true, false, false);
                            J$.N(4724313, 'i', i, false, false, false);
                            for (var i = J$.W(4723849, 'i', J$.T(4723841, 1, 22, false), i, false, false); J$.C(255680, J$.B(451354, '<=', J$.R(4723857, 'i', i, false, false), J$.R(4723865, 'NB_TEST', NB_TEST, false, true))); J$.B(451378, '-', i = J$.W(4723881, 'i', J$.B(451370, '+', J$.U(451362, '+', J$.R(4723873, 'i', i, false, false)), 1), i, false, false), 1)) {
                                J$.F(4724281, J$.I(typeof test === 'undefined' ? test = J$.R(4723889, 'test', undefined, true, true) : test = J$.R(4723889, 'test', test, true, true)), false)(J$.B(451386, '+', J$.T(4723897, 'Random candidate #', 21, false), J$.R(4723905, 'i', i, false, false)), J$.T(4724273, function () {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(4724241, arguments.callee, this, arguments);
                                                arguments = J$.N(4724249, 'arguments', arguments, true, false, false);
                                                J$.N(4724257, 'candidate', candidate, false, false, false);
                                                J$.N(4724265, 'result', result, false, false, false);
                                                var candidate = J$.W(4723937, 'candidate', J$.M(4723929, J$.R(4723913, 'si', si, false, true), 'generate', false)(J$.R(4723921, 'schema', schema, false, false)), candidate, false, false);
                                                var result = J$.W(4723977, 'result', J$.M(4723969, J$.R(4723945, 'si', si, false, true), 'validate', false)(J$.R(4723953, 'schema', schema, false, false), J$.R(4723961, 'candidate', candidate, false, false)), result, false, false);
                                                J$.G(4724017, J$.G(4724009, J$.G(4724001, J$.G(4723993, J$.R(4723985, 'result', result, false, false), 'should'), 'be'), 'an'), 'Object');
                                                try {
                                                    J$.M(4724081, J$.G(4724065, J$.M(4724057, J$.G(4724041, J$.G(4724033, J$.R(4724025, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(4724049, 'valid', 21, false)), 'with'), 'equal', false)(J$.T(4724073, true, 23, false));
                                                    J$.M(4724193, J$.G(4724177, J$.G(4724169, J$.M(4724161, J$.G(4724145, J$.G(4724137, J$.G(4724129, J$.M(4724121, J$.G(4724105, J$.G(4724097, J$.R(4724089, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(4724113, 'error', 21, false)), 'with'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(4724153, 'Array', undefined, true, true) : Array = J$.R(4724153, 'Array', Array, true, true))), 'and'), 'be'), 'lengthOf', false)(J$.T(4724185, 0, 22, false));
                                                } catch (e) {
                                                    J$.N(4724233, 'e', e, false, false, true);
                                                    throw J$.F(4724225, J$.I(typeof Error === 'undefined' ? Error = J$.R(4724201, 'Error', undefined, true, true) : Error = J$.R(4724201, 'Error', Error, true, true)), true)(J$.M(4724217, J$.R(4724209, 'result', result, false, false), 'format', false)());
                                                }
                                            } catch (J$e) {
                                                J$.Ex(4728513, J$e);
                                            } finally {
                                                if (J$.Fr(4728521))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            }
                        } catch (J$e) {
                            J$.Ex(4728529, J$e);
                        } finally {
                            if (J$.Fr(4728537))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(4728473, 'should', should, false, false, false);
            J$.N(4728481, 'si', si, false, false, false);
            J$.N(4728489, 'NB_TEST', NB_TEST, false, false, false);
            testRandomCandidates = J$.N(4728505, 'testRandomCandidates', J$.T(4728497, testRandomCandidates, 12, false), true, false, false);
            var should = J$.W(4723785, 'should', J$.F(4723777, J$.I(typeof require === 'undefined' ? require = J$.R(4723761, 'require', undefined, true, true) : require = J$.R(4723761, 'require', require, true, true)), false)(J$.T(4723769, 'should', 21, false)), should, false, true);
            var si = J$.W(4723817, 'si', J$.F(4723809, J$.I(typeof require === 'undefined' ? require = J$.R(4723793, 'require', undefined, true, true) : require = J$.R(4723793, 'require', require, true, true)), false)(J$.T(4723801, '../', 21, false)), si, false, true);
            const NB_TEST = J$.W(4723833, 'NB_TEST', J$.T(4723825, 25, 22, false), NB_TEST, false, true);
            J$.P(4728457, J$.I(typeof exports === 'undefined' ? exports = J$.R(4724321, 'exports', undefined, true, true) : exports = J$.R(4724321, 'exports', exports, true, true)), 'generator', J$.T(4728449, function () {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(4728433, arguments.callee, this, arguments);
                            arguments = J$.N(4728441, 'arguments', arguments, true, false, false);
                            J$.F(4724609, J$.I(typeof suite === 'undefined' ? suite = J$.R(4724329, 'suite', undefined, true, true) : suite = J$.R(4724329, 'suite', suite, true, true)), false)(J$.T(4724337, 'Schema #1 (Basics schemas with nested object, only considering type)', 21, false), J$.T(4724601, function () {
                                jalangiLabel2:
                                    while (true) {
                                        try {
                                            J$.Fe(4724577, arguments.callee, this, arguments);
                                            arguments = J$.N(4724585, 'arguments', arguments, true, false, false);
                                            J$.N(4724593, 'schema', schema, false, false, false);
                                            var schema = J$.W(4724545, 'schema', J$.T(4724537, {
                                                type: J$.T(4724345, 'object', 21, false),
                                                properties: J$.T(4724529, {
                                                    lorem: J$.T(4724521, {
                                                        type: J$.T(4724353, 'object', 21, false),
                                                        properties: J$.T(4724513, {
                                                            ipsum: J$.T(4724385, {
                                                                type: J$.T(4724377, [
                                                                    J$.T(4724361, 'string', 21, false),
                                                                    J$.T(4724369, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(4724401, { type: J$.T(4724393, 'number', 21, false) }, 11, false),
                                                            sit: J$.T(4724417, { type: J$.T(4724409, 'any', 21, false) }, 11, false),
                                                            amet: J$.T(4724449, {
                                                                type: J$.T(4724425, 'array', 21, false),
                                                                items: J$.T(4724441, { type: J$.T(4724433, 'string', 21, false) }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(4724505, {
                                                                type: J$.T(4724457, 'array', 21, false),
                                                                items: J$.T(4724497, [
                                                                    J$.T(4724473, { type: J$.T(4724465, 'integer', 21, false) }, 11, false),
                                                                    J$.T(4724489, { type: J$.T(4724481, 'number', 21, false) }, 11, false)
                                                                ], 10, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(4724569, J$.R(4724553, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(4724561, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4728545, J$e);
                                        } finally {
                                            if (J$.Fr(4728553))
                                                continue jalangiLabel2;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(4724921, J$.I(typeof suite === 'undefined' ? suite = J$.R(4724617, 'suite', undefined, true, true) : suite = J$.R(4724617, 'suite', suite, true, true)), false)(J$.T(4724625, 'Schema #2 (Basic schemas, considering *Length)', 21, false), J$.T(4724913, function () {
                                jalangiLabel3:
                                    while (true) {
                                        try {
                                            J$.Fe(4724889, arguments.callee, this, arguments);
                                            arguments = J$.N(4724897, 'arguments', arguments, true, false, false);
                                            J$.N(4724905, 'schema', schema, false, false, false);
                                            var schema = J$.W(4724857, 'schema', J$.T(4724849, {
                                                type: J$.T(4724633, 'object', 21, false),
                                                properties: J$.T(4724841, {
                                                    lorem: J$.T(4724833, {
                                                        type: J$.T(4724641, 'object', 21, false),
                                                        properties: J$.T(4724825, {
                                                            ipsum: J$.T(4724673, {
                                                                type: J$.T(4724665, [
                                                                    J$.T(4724649, 'string', 21, false),
                                                                    J$.T(4724657, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(4724689, { type: J$.T(4724681, 'number', 21, false) }, 11, false),
                                                            sit: J$.T(4724705, { type: J$.T(4724697, 'any', 21, false) }, 11, false),
                                                            amet: J$.T(4724753, {
                                                                type: J$.T(4724713, 'array', 21, false),
                                                                exactLength: J$.T(4724721, 6, 22, false),
                                                                items: J$.T(4724745, {
                                                                    type: J$.T(4724729, 'string', 21, false),
                                                                    exactLength: J$.T(4724737, 10, 22, false)
                                                                }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(4724817, {
                                                                type: J$.T(4724761, 'array', 21, false),
                                                                minLength: J$.T(4724769, 4, 22, false),
                                                                maxLength: J$.T(4724777, 8, 22, false),
                                                                items: J$.T(4724809, {
                                                                    type: J$.T(4724785, 'string', 21, false),
                                                                    minLength: J$.T(4724793, 6, 22, false),
                                                                    maxLength: J$.T(4724801, 12, 22, false)
                                                                }, 11, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(4724881, J$.R(4724865, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(4724873, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4728561, J$e);
                                        } finally {
                                            if (J$.Fr(4728569))
                                                continue jalangiLabel3;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(4725649, J$.I(typeof suite === 'undefined' ? suite = J$.R(4724929, 'suite', undefined, true, true) : suite = J$.R(4724929, 'suite', suite, true, true)), false)(J$.T(4724937, 'Schema #3 (Intermediate schemas, considering *Length, lt, lte, gt, gte, eq and ne)', 21, false), J$.T(4725641, function () {
                                jalangiLabel4:
                                    while (true) {
                                        try {
                                            J$.Fe(4725617, arguments.callee, this, arguments);
                                            arguments = J$.N(4725625, 'arguments', arguments, true, false, false);
                                            J$.N(4725633, 'schema', schema, false, false, false);
                                            var schema = J$.W(4725585, 'schema', J$.T(4725577, {
                                                type: J$.T(4724945, 'object', 21, false),
                                                properties: J$.T(4725569, {
                                                    lorem: J$.T(4725561, {
                                                        type: J$.T(4724953, 'object', 21, false),
                                                        properties: J$.T(4725553, {
                                                            ipsum: J$.T(4724985, {
                                                                type: J$.T(4724977, [
                                                                    J$.T(4724961, 'string', 21, false),
                                                                    J$.T(4724969, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(4725001, { type: J$.T(4724993, 'number', 21, false) }, 11, false),
                                                            sit: J$.T(4725017, { type: J$.T(4725009, 'any', 21, false) }, 11, false),
                                                            amet: J$.T(4725065, {
                                                                type: J$.T(4725025, 'array', 21, false),
                                                                exactLength: J$.T(4725033, 6, 22, false),
                                                                items: J$.T(4725057, {
                                                                    type: J$.T(4725041, 'string', 21, false),
                                                                    exactLength: J$.T(4725049, 10, 22, false)
                                                                }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(4725545, {
                                                                type: J$.T(4725073, 'array', 21, false),
                                                                maxLength: J$.T(4725081, 16, 22, false),
                                                                items: J$.T(4725537, [
                                                                    J$.T(4725105, {
                                                                        type: J$.T(4725089, 'number', 21, false),
                                                                        gt: J$.T(4725097, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725129, {
                                                                        type: J$.T(4725113, 'number', 21, false),
                                                                        lt: J$.T(4725121, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725153, {
                                                                        type: J$.T(4725137, 'number', 21, false),
                                                                        gte: J$.T(4725145, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725177, {
                                                                        type: J$.T(4725161, 'number', 21, false),
                                                                        lte: J$.T(4725169, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725209, {
                                                                        type: J$.T(4725185, 'number', 21, false),
                                                                        gt: J$.T(4725193, 5, 22, false),
                                                                        lt: J$.T(4725201, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725241, {
                                                                        type: J$.T(4725217, 'number', 21, false),
                                                                        gte: J$.T(4725225, 5, 22, false),
                                                                        lte: J$.T(4725233, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725265, {
                                                                        type: J$.T(4725249, 'number', 21, false),
                                                                        eq: J$.T(4725257, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725305, {
                                                                        type: J$.T(4725273, 'number', 21, false),
                                                                        gte: J$.T(4725281, 0, 22, false),
                                                                        lte: J$.T(4725289, 1, 22, false),
                                                                        ne: J$.T(4725297, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725329, {
                                                                        type: J$.T(4725313, 'integer', 21, false),
                                                                        gt: J$.T(4725321, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725353, {
                                                                        type: J$.T(4725337, 'integer', 21, false),
                                                                        lt: J$.T(4725345, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725377, {
                                                                        type: J$.T(4725361, 'integer', 21, false),
                                                                        gte: J$.T(4725369, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725401, {
                                                                        type: J$.T(4725385, 'integer', 21, false),
                                                                        lte: J$.T(4725393, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725433, {
                                                                        type: J$.T(4725409, 'integer', 21, false),
                                                                        gt: J$.T(4725417, 5, 22, false),
                                                                        lt: J$.T(4725425, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725465, {
                                                                        type: J$.T(4725441, 'integer', 21, false),
                                                                        gte: J$.T(4725449, 5, 22, false),
                                                                        lte: J$.T(4725457, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725489, {
                                                                        type: J$.T(4725473, 'integer', 21, false),
                                                                        eq: J$.T(4725481, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725529, {
                                                                        type: J$.T(4725497, 'integer', 21, false),
                                                                        gte: J$.T(4725505, 0, 22, false),
                                                                        lte: J$.T(4725513, 1, 22, false),
                                                                        ne: J$.T(4725521, 1, 22, false)
                                                                    }, 11, false)
                                                                ], 10, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(4725609, J$.R(4725593, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(4725601, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4728577, J$e);
                                        } finally {
                                            if (J$.Fr(4728585))
                                                continue jalangiLabel4;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(4726585, J$.I(typeof suite === 'undefined' ? suite = J$.R(4725657, 'suite', undefined, true, true) : suite = J$.R(4725657, 'suite', suite, true, true)), false)(J$.T(4725665, 'Schema #4 (Advanced schemas, considering *Length, lt, lte, gt, gte, eq, array of ne and optional fields)', 21, false), J$.T(4726577, function () {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(4726553, arguments.callee, this, arguments);
                                            arguments = J$.N(4726561, 'arguments', arguments, true, false, false);
                                            J$.N(4726569, 'schema', schema, false, false, false);
                                            var schema = J$.W(4726521, 'schema', J$.T(4726513, {
                                                type: J$.T(4725673, 'object', 21, false),
                                                properties: J$.T(4726505, {
                                                    lorem: J$.T(4726497, {
                                                        type: J$.T(4725681, 'object', 21, false),
                                                        properties: J$.T(4726489, {
                                                            ipsum: J$.T(4725713, {
                                                                type: J$.T(4725705, [
                                                                    J$.T(4725689, 'string', 21, false),
                                                                    J$.T(4725697, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(4725737, {
                                                                type: J$.T(4725721, 'number', 21, false),
                                                                optional: J$.T(4725729, true, 23, false)
                                                            }, 11, false),
                                                            sit: J$.T(4725761, {
                                                                type: J$.T(4725745, 'any', 21, false),
                                                                optional: J$.T(4725753, true, 23, false)
                                                            }, 11, false),
                                                            amet: J$.T(4725825, {
                                                                type: J$.T(4725769, 'array', 21, false),
                                                                exactLength: J$.T(4725777, 6, 22, false),
                                                                optional: J$.T(4725785, true, 23, false),
                                                                items: J$.T(4725817, {
                                                                    type: J$.T(4725793, 'string', 21, false),
                                                                    optional: J$.T(4725801, true, 23, false),
                                                                    exactLength: J$.T(4725809, 10, 22, false)
                                                                }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(4726481, {
                                                                type: J$.T(4725833, 'array', 21, false),
                                                                maxLength: J$.T(4725841, 18, 22, false),
                                                                optional: J$.T(4725849, true, 23, false),
                                                                items: J$.T(4726473, [
                                                                    J$.T(4725873, {
                                                                        type: J$.T(4725857, 'number', 21, false),
                                                                        gt: J$.T(4725865, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725897, {
                                                                        type: J$.T(4725881, 'number', 21, false),
                                                                        lt: J$.T(4725889, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725921, {
                                                                        type: J$.T(4725905, 'number', 21, false),
                                                                        gte: J$.T(4725913, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725945, {
                                                                        type: J$.T(4725929, 'number', 21, false),
                                                                        lte: J$.T(4725937, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4725977, {
                                                                        type: J$.T(4725953, 'number', 21, false),
                                                                        gt: J$.T(4725961, 5, 22, false),
                                                                        lt: J$.T(4725969, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4726009, {
                                                                        type: J$.T(4725985, 'number', 21, false),
                                                                        gte: J$.T(4725993, 5, 22, false),
                                                                        lte: J$.T(4726001, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4726033, {
                                                                        type: J$.T(4726017, 'number', 21, false),
                                                                        eq: J$.T(4726025, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4726089, {
                                                                        type: J$.T(4726041, 'number', 21, false),
                                                                        gte: J$.T(4726049, 0, 22, false),
                                                                        lte: J$.T(4726057, 2, 22, false),
                                                                        ne: J$.T(4726081, [
                                                                            J$.T(4726065, 0, 22, false),
                                                                            J$.T(4726073, 1, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false),
                                                                    J$.T(4726113, {
                                                                        type: J$.T(4726097, 'integer', 21, false),
                                                                        gt: J$.T(4726105, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4726137, {
                                                                        type: J$.T(4726121, 'integer', 21, false),
                                                                        lt: J$.T(4726129, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4726161, {
                                                                        type: J$.T(4726145, 'integer', 21, false),
                                                                        gte: J$.T(4726153, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4726185, {
                                                                        type: J$.T(4726169, 'integer', 21, false),
                                                                        lte: J$.T(4726177, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4726217, {
                                                                        type: J$.T(4726193, 'integer', 21, false),
                                                                        gt: J$.T(4726201, 5, 22, false),
                                                                        lt: J$.T(4726209, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4726249, {
                                                                        type: J$.T(4726225, 'integer', 21, false),
                                                                        gte: J$.T(4726233, 5, 22, false),
                                                                        lte: J$.T(4726241, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4726273, {
                                                                        type: J$.T(4726257, 'integer', 21, false),
                                                                        eq: J$.T(4726265, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(4726329, {
                                                                        type: J$.T(4726281, 'integer', 21, false),
                                                                        gte: J$.T(4726289, 0, 22, false),
                                                                        lte: J$.T(4726297, 2, 22, false),
                                                                        ne: J$.T(4726321, [
                                                                            J$.T(4726305, 0, 22, false),
                                                                            J$.T(4726313, 1, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false),
                                                                    J$.T(4726393, {
                                                                        type: J$.T(4726337, 'integer', 21, false),
                                                                        gte: J$.T(4726345, 0, 22, false),
                                                                        lte: J$.T(4726353, 3, 22, false),
                                                                        ne: J$.T(4726385, [
                                                                            J$.T(4726361, 0, 22, false),
                                                                            J$.T(4726369, 1, 22, false),
                                                                            J$.T(4726377, 2, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false),
                                                                    J$.T(4726465, {
                                                                        type: J$.T(4726401, 'integer', 21, false),
                                                                        gte: J$.T(4726409, 0, 22, false),
                                                                        lte: J$.T(4726417, 4, 22, false),
                                                                        ne: J$.T(4726457, [
                                                                            J$.T(4726425, 0, 22, false),
                                                                            J$.T(4726433, 1, 22, false),
                                                                            J$.T(4726441, 2, 22, false),
                                                                            J$.T(4726449, 3, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false)
                                                                ], 10, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(4726545, J$.R(4726529, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(4726537, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4728593, J$e);
                                        } finally {
                                            if (J$.Fr(4728601))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(4727041, J$.I(typeof suite === 'undefined' ? suite = J$.R(4726593, 'suite', undefined, true, true) : suite = J$.R(4726593, 'suite', suite, true, true)), false)(J$.T(4726601, 'Schema #5 (Advanced schemas, considering *Length, eq and format fields)', 21, false), J$.T(4727033, function () {
                                jalangiLabel6:
                                    while (true) {
                                        try {
                                            J$.Fe(4727009, arguments.callee, this, arguments);
                                            arguments = J$.N(4727017, 'arguments', arguments, true, false, false);
                                            J$.N(4727025, 'schema', schema, false, false, false);
                                            var schema = J$.W(4726977, 'schema', J$.T(4726969, {
                                                type: J$.T(4726609, 'object', 21, false),
                                                properties: J$.T(4726961, {
                                                    lorem: J$.T(4726953, {
                                                        type: J$.T(4726617, 'array', 21, false),
                                                        items: J$.T(4726945, [
                                                            J$.T(4726641, {
                                                                type: J$.T(4726625, 'string', 21, false),
                                                                pattern: J$.T(4726633, 'date-time', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726665, {
                                                                type: J$.T(4726649, 'string', 21, false),
                                                                pattern: J$.T(4726657, 'date-time', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726689, {
                                                                type: J$.T(4726673, 'string', 21, false),
                                                                pattern: J$.T(4726681, 'date', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726713, {
                                                                type: J$.T(4726697, 'string', 21, false),
                                                                pattern: J$.T(4726705, 'time', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726737, {
                                                                type: J$.T(4726721, 'string', 21, false),
                                                                pattern: J$.T(4726729, 'color', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726769, {
                                                                type: J$.T(4726745, 'string', 21, false),
                                                                pattern: J$.T(4726753, 'color', 21, false),
                                                                eq: J$.T(4726761, '#FFffFF', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726793, {
                                                                type: J$.T(4726777, 'string', 21, false),
                                                                pattern: J$.T(4726785, 'numeric', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726817, {
                                                                type: J$.T(4726801, 'string', 21, false),
                                                                pattern: J$.T(4726809, 'integer', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726841, {
                                                                type: J$.T(4726825, 'string', 21, false),
                                                                pattern: J$.T(4726833, 'decimal', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726865, {
                                                                type: J$.T(4726849, 'string', 21, false),
                                                                pattern: J$.T(4726857, 'alpha', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726889, {
                                                                type: J$.T(4726873, 'string', 21, false),
                                                                pattern: J$.T(4726881, 'alphaNumeric', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726913, {
                                                                type: J$.T(4726897, 'string', 21, false),
                                                                pattern: J$.T(4726905, 'alphaDash', 21, false)
                                                            }, 11, false),
                                                            J$.T(4726937, {
                                                                type: J$.T(4726921, 'string', 21, false),
                                                                pattern: J$.T(4726929, 'javascript', 21, false)
                                                            }, 11, false)
                                                        ], 10, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(4727001, J$.R(4726985, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(4726993, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4728609, J$e);
                                        } finally {
                                            if (J$.Fr(4728617))
                                                continue jalangiLabel6;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(4727689, J$.I(typeof suite === 'undefined' ? suite = J$.R(4727049, 'suite', undefined, true, true) : suite = J$.R(4727049, 'suite', suite, true, true)), false)(J$.T(4727057, 'Schema #6 (Generate several candidate with one call)', 21, false), J$.T(4727681, function () {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(4727649, arguments.callee, this, arguments);
                                            arguments = J$.N(4727657, 'arguments', arguments, true, false, false);
                                            J$.N(4727665, 'schema', schema, false, false, false);
                                            J$.N(4727673, 'candidates', candidates, false, false, false);
                                            var schema = J$.W(4727105, 'schema', J$.T(4727097, {
                                                type: J$.T(4727065, 'object', 21, false),
                                                properties: J$.T(4727089, { lorem: J$.T(4727081, { type: J$.T(4727073, 'integer', 21, false) }, 11, false) }, 11, false)
                                            }, 11, false), schema, false, false);
                                            var candidates = J$.W(4727145, 'candidates', J$.M(4727137, J$.R(4727113, 'si', si, false, true), 'generate', false)(J$.R(4727121, 'schema', schema, false, false), J$.R(4727129, 'NB_TEST', NB_TEST, false, true)), candidates, false, false);
                                            J$.M(4727217, J$.G(4727201, J$.M(4727193, J$.G(4727177, J$.G(4727169, J$.G(4727161, J$.R(4727153, 'candidates', candidates, false, false), 'should'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(4727185, 'Array', undefined, true, true) : Array = J$.R(4727185, 'Array', Array, true, true))), 'with'), 'lengthOf', false)(J$.R(4727209, 'NB_TEST', NB_TEST, false, true));
                                            J$.M(4727641, J$.R(4727225, 'candidates', candidates, false, false), 'forEach', false)(J$.T(4727633, function (candidate, i) {
                                                jalangiLabel8:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(4727601, arguments.callee, this, arguments);
                                                            arguments = J$.N(4727609, 'arguments', arguments, true, false, false);
                                                            candidate = J$.N(4727617, 'candidate', candidate, true, false, false);
                                                            i = J$.N(4727625, 'i', i, true, false, false);
                                                            J$.F(4727593, J$.I(typeof test === 'undefined' ? test = J$.R(4727233, 'test', undefined, true, true) : test = J$.R(4727233, 'test', test, true, true)), false)(J$.B(451402, '+', J$.T(4727241, 'Random candidate #', 21, false), J$.B(451394, '+', J$.R(4727249, 'i', i, false, false), J$.T(4727257, 1, 22, false))), J$.T(4727585, function () {
                                                                jalangiLabel7:
                                                                    while (true) {
                                                                        try {
                                                                            J$.Fe(4727561, arguments.callee, this, arguments);
                                                                            arguments = J$.N(4727569, 'arguments', arguments, true, false, false);
                                                                            J$.N(4727577, 'result', result, false, false, false);
                                                                            var result = J$.W(4727297, 'result', J$.M(4727289, J$.R(4727265, 'si', si, false, true), 'validate', false)(J$.R(4727273, 'schema', schema, false, false), J$.R(4727281, 'candidate', candidate, false, false)), result, false, false);
                                                                            J$.G(4727337, J$.G(4727329, J$.G(4727321, J$.G(4727313, J$.R(4727305, 'result', result, false, false), 'should'), 'be'), 'an'), 'Object');
                                                                            try {
                                                                                J$.M(4727401, J$.G(4727385, J$.M(4727377, J$.G(4727361, J$.G(4727353, J$.R(4727345, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(4727369, 'valid', 21, false)), 'with'), 'equal', false)(J$.T(4727393, true, 23, false));
                                                                                J$.M(4727513, J$.G(4727497, J$.G(4727489, J$.M(4727481, J$.G(4727465, J$.G(4727457, J$.G(4727449, J$.M(4727441, J$.G(4727425, J$.G(4727417, J$.R(4727409, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(4727433, 'error', 21, false)), 'with'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(4727473, 'Array', undefined, true, true) : Array = J$.R(4727473, 'Array', Array, true, true))), 'and'), 'be'), 'lengthOf', false)(J$.T(4727505, 0, 22, false));
                                                                            } catch (e) {
                                                                                J$.N(4727553, 'e', e, false, false, true);
                                                                                throw J$.F(4727545, J$.I(typeof Error === 'undefined' ? Error = J$.R(4727521, 'Error', undefined, true, true) : Error = J$.R(4727521, 'Error', Error, true, true)), true)(J$.M(4727537, J$.R(4727529, 'result', result, false, false), 'format', false)());
                                                                            }
                                                                        } catch (J$e) {
                                                                            J$.Ex(4728625, J$e);
                                                                        } finally {
                                                                            if (J$.Fr(4728633))
                                                                                continue jalangiLabel7;
                                                                            else
                                                                                return J$.Ra();
                                                                        }
                                                                    }
                                                            }, 12, false));
                                                        } catch (J$e) {
                                                            J$.Ex(4728641, J$e);
                                                        } finally {
                                                            if (J$.Fr(4728649))
                                                                continue jalangiLabel8;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(4728657, J$e);
                                        } finally {
                                            if (J$.Fr(4728665))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(4728425, J$.I(typeof suite === 'undefined' ? suite = J$.R(4727697, 'suite', undefined, true, true) : suite = J$.R(4727697, 'suite', suite, true, true)), false)(J$.T(4727705, 'Schema #7 (Globing for Object keys)', 21, false), J$.T(4728417, function () {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(4728385, arguments.callee, this, arguments);
                                            arguments = J$.N(4728393, 'arguments', arguments, true, false, false);
                                            J$.N(4728401, 'schema', schema, false, false, false);
                                            J$.N(4728409, 'i', i, false, false, false);
                                            var schema = J$.W(4727817, 'schema', J$.T(4727809, {
                                                type: J$.T(4727713, 'object', 21, false),
                                                properties: J$.T(4727801, {
                                                    globString: J$.T(4727753, {
                                                        type: J$.T(4727721, 'object', 21, false),
                                                        properties: J$.T(4727745, { '*': J$.T(4727737, { type: J$.T(4727729, 'string', 21, false) }, 11, false) }, 11, false)
                                                    }, 11, false),
                                                    globInteger: J$.T(4727793, {
                                                        type: J$.T(4727761, 'object', 21, false),
                                                        properties: J$.T(4727785, { '*': J$.T(4727777, { type: J$.T(4727769, 'integer', 21, false) }, 11, false) }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            for (var i = J$.W(4727833, 'i', J$.T(4727825, 1, 22, false), i, false, false); J$.C(255688, J$.B(451410, '<=', J$.R(4727841, 'i', i, false, false), J$.R(4727849, 'NB_TEST', NB_TEST, false, true))); J$.B(451434, '-', i = J$.W(4727865, 'i', J$.B(451426, '+', J$.U(451418, '+', J$.R(4727857, 'i', i, false, false)), 1), i, false, false), 1)) {
                                                J$.F(4728377, J$.I(typeof test === 'undefined' ? test = J$.R(4727873, 'test', undefined, true, true) : test = J$.R(4727873, 'test', test, true, true)), false)(J$.B(451442, '+', J$.T(4727881, 'Random candidate #', 21, false), J$.R(4727889, 'i', i, false, false)), J$.T(4728369, function () {
                                                    jalangiLabel10:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(4728337, arguments.callee, this, arguments);
                                                                arguments = J$.N(4728345, 'arguments', arguments, true, false, false);
                                                                J$.N(4728353, 'candidate', candidate, false, false, false);
                                                                J$.N(4728361, 'result', result, false, false, false);
                                                                var candidate = J$.W(4727921, 'candidate', J$.M(4727913, J$.R(4727897, 'si', si, false, true), 'generate', false)(J$.R(4727905, 'schema', schema, false, false)), candidate, false, false);
                                                                var result = J$.W(4727961, 'result', J$.M(4727953, J$.R(4727929, 'si', si, false, true), 'validate', false)(J$.R(4727937, 'schema', schema, false, false), J$.R(4727945, 'candidate', candidate, false, false)), result, false, false);
                                                                J$.G(4728001, J$.G(4727993, J$.G(4727985, J$.G(4727977, J$.R(4727969, 'result', result, false, false), 'should'), 'be'), 'an'), 'Object');
                                                                J$.M(4728057, J$.G(4728041, J$.G(4728033, J$.G(4728025, J$.G(4728017, J$.R(4728009, 'candidate', candidate, false, false), 'globString'), 'should'), 'not'), 'have'), 'property', false)(J$.T(4728049, '*', 21, false));
                                                                J$.M(4728113, J$.G(4728097, J$.G(4728089, J$.G(4728081, J$.G(4728073, J$.R(4728065, 'candidate', candidate, false, false), 'globInteger'), 'should'), 'not'), 'have'), 'property', false)(J$.T(4728105, '*', 21, false));
                                                                try {
                                                                    J$.M(4728177, J$.G(4728161, J$.M(4728153, J$.G(4728137, J$.G(4728129, J$.R(4728121, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(4728145, 'valid', 21, false)), 'with'), 'equal', false)(J$.T(4728169, true, 23, false));
                                                                    J$.M(4728289, J$.G(4728273, J$.G(4728265, J$.M(4728257, J$.G(4728241, J$.G(4728233, J$.G(4728225, J$.M(4728217, J$.G(4728201, J$.G(4728193, J$.R(4728185, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(4728209, 'error', 21, false)), 'with'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(4728249, 'Array', undefined, true, true) : Array = J$.R(4728249, 'Array', Array, true, true))), 'and'), 'be'), 'lengthOf', false)(J$.T(4728281, 0, 22, false));
                                                                } catch (e) {
                                                                    J$.N(4728329, 'e', e, false, false, true);
                                                                    throw J$.F(4728321, J$.I(typeof Error === 'undefined' ? Error = J$.R(4728297, 'Error', undefined, true, true) : Error = J$.R(4728297, 'Error', Error, true, true)), true)(J$.M(4728313, J$.R(4728305, 'result', result, false, false), 'format', false)());
                                                                }
                                                            } catch (J$e) {
                                                                J$.Ex(4728673, J$e);
                                                            } finally {
                                                                if (J$.Fr(4728681))
                                                                    continue jalangiLabel10;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false));
                                            }
                                        } catch (J$e) {
                                            J$.Ex(4728689, J$e);
                                        } finally {
                                            if (J$.Fr(4728697))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(4728705, J$e);
                        } finally {
                            if (J$.Fr(4728713))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(4728721, J$e);
        } finally {
            if (J$.Sr(4728729))
                continue jalangiLabel13;
            else
                break jalangiLabel13;
        }
    }
// JALANGI DO NOT INSTRUMENT
