// schema-inspector/test/generator_test.js
J$.noInstrEval = false;
jalangiLabel13:
    while (true) {
        try {
            J$.Se(795121, '/mnt/data/fxiao/HiPar/outputs/target_cache/TestSchema-inspector/node_modules/schema-inspector/test/generator_test.js');
            function testRandomCandidates(schema) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(790945, arguments.callee, this, arguments);
                            arguments = J$.N(790953, 'arguments', arguments, true, false, false);
                            schema = J$.N(790961, 'schema', schema, true, false, false);
                            J$.N(790969, 'i', i, false, false, false);
                            for (var i = J$.W(790505, 'i', J$.T(790497, 1, 22, false), i, false, false); J$.C(49072, J$.B(67562, '<=', J$.R(790513, 'i', i, false, false), J$.R(790521, 'NB_TEST', NB_TEST, false, true))); J$.B(67586, '-', i = J$.W(790537, 'i', J$.B(67578, '+', J$.U(67570, '+', J$.R(790529, 'i', i, false, false)), 1), i, false, false), 1)) {
                                J$.F(790937, J$.I(typeof test === 'undefined' ? test = J$.R(790545, 'test', undefined, true, true) : test = J$.R(790545, 'test', test, true, true)), false)(J$.B(67594, '+', J$.T(790553, 'Random candidate #', 21, false), J$.R(790561, 'i', i, false, false)), J$.T(790929, function () {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(790897, arguments.callee, this, arguments);
                                                arguments = J$.N(790905, 'arguments', arguments, true, false, false);
                                                J$.N(790913, 'candidate', candidate, false, false, false);
                                                J$.N(790921, 'result', result, false, false, false);
                                                var candidate = J$.W(790593, 'candidate', J$.M(790585, J$.R(790569, 'si', si, false, true), 'generate', false)(J$.R(790577, 'schema', schema, false, false)), candidate, false, false);
                                                var result = J$.W(790633, 'result', J$.M(790625, J$.R(790601, 'si', si, false, true), 'validate', false)(J$.R(790609, 'schema', schema, false, false), J$.R(790617, 'candidate', candidate, false, false)), result, false, false);
                                                J$.G(790673, J$.G(790665, J$.G(790657, J$.G(790649, J$.R(790641, 'result', result, false, false), 'should'), 'be'), 'an'), 'Object');
                                                try {
                                                    J$.M(790737, J$.G(790721, J$.M(790713, J$.G(790697, J$.G(790689, J$.R(790681, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(790705, 'valid', 21, false)), 'with'), 'equal', false)(J$.T(790729, true, 23, false));
                                                    J$.M(790849, J$.G(790833, J$.G(790825, J$.M(790817, J$.G(790801, J$.G(790793, J$.G(790785, J$.M(790777, J$.G(790761, J$.G(790753, J$.R(790745, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(790769, 'error', 21, false)), 'with'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(790809, 'Array', undefined, true, true) : Array = J$.R(790809, 'Array', Array, true, true))), 'and'), 'be'), 'lengthOf', false)(J$.T(790841, 0, 22, false));
                                                } catch (e) {
                                                    J$.N(790889, 'e', e, false, false, true);
                                                    throw J$.F(790881, J$.I(typeof Error === 'undefined' ? Error = J$.R(790857, 'Error', undefined, true, true) : Error = J$.R(790857, 'Error', Error, true, true)), true)(J$.M(790873, J$.R(790865, 'result', result, false, false), 'format', false)());
                                                }
                                            } catch (J$e) {
                                                J$.Ex(795169, J$e);
                                            } finally {
                                                if (J$.Fr(795177))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            }
                        } catch (J$e) {
                            J$.Ex(795185, J$e);
                        } finally {
                            if (J$.Fr(795193))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(795129, 'should', should, false, false, false);
            J$.N(795137, 'si', si, false, false, false);
            J$.N(795145, 'NB_TEST', NB_TEST, false, false, false);
            testRandomCandidates = J$.N(795161, 'testRandomCandidates', J$.T(795153, testRandomCandidates, 12, false), true, false, false);
            var should = J$.W(790441, 'should', J$.F(790433, J$.I(typeof require === 'undefined' ? require = J$.R(790417, 'require', undefined, true, true) : require = J$.R(790417, 'require', require, true, true)), false)(J$.T(790425, 'should', 21, false)), should, false, true);
            var si = J$.W(790473, 'si', J$.F(790465, J$.I(typeof require === 'undefined' ? require = J$.R(790449, 'require', undefined, true, true) : require = J$.R(790449, 'require', require, true, true)), false)(J$.T(790457, '../', 21, false)), si, false, true);
            const NB_TEST = J$.W(790489, 'NB_TEST', J$.T(790481, 25, 22, false), NB_TEST, false, true);
            J$.P(795113, J$.I(typeof exports === 'undefined' ? exports = J$.R(790977, 'exports', undefined, true, true) : exports = J$.R(790977, 'exports', exports, true, true)), 'generator', J$.T(795105, function () {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(795089, arguments.callee, this, arguments);
                            arguments = J$.N(795097, 'arguments', arguments, true, false, false);
                            J$.F(791265, J$.I(typeof suite === 'undefined' ? suite = J$.R(790985, 'suite', undefined, true, true) : suite = J$.R(790985, 'suite', suite, true, true)), false)(J$.T(790993, 'Schema #1 (Basics schemas with nested object, only considering type)', 21, false), J$.T(791257, function () {
                                jalangiLabel2:
                                    while (true) {
                                        try {
                                            J$.Fe(791233, arguments.callee, this, arguments);
                                            arguments = J$.N(791241, 'arguments', arguments, true, false, false);
                                            J$.N(791249, 'schema', schema, false, false, false);
                                            var schema = J$.W(791201, 'schema', J$.T(791193, {
                                                type: J$.T(791001, 'object', 21, false),
                                                properties: J$.T(791185, {
                                                    lorem: J$.T(791177, {
                                                        type: J$.T(791009, 'object', 21, false),
                                                        properties: J$.T(791169, {
                                                            ipsum: J$.T(791041, {
                                                                type: J$.T(791033, [
                                                                    J$.T(791017, 'string', 21, false),
                                                                    J$.T(791025, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(791057, { type: J$.T(791049, 'number', 21, false) }, 11, false),
                                                            sit: J$.T(791073, { type: J$.T(791065, 'any', 21, false) }, 11, false),
                                                            amet: J$.T(791105, {
                                                                type: J$.T(791081, 'array', 21, false),
                                                                items: J$.T(791097, { type: J$.T(791089, 'string', 21, false) }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(791161, {
                                                                type: J$.T(791113, 'array', 21, false),
                                                                items: J$.T(791153, [
                                                                    J$.T(791129, { type: J$.T(791121, 'integer', 21, false) }, 11, false),
                                                                    J$.T(791145, { type: J$.T(791137, 'number', 21, false) }, 11, false)
                                                                ], 10, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(791225, J$.R(791209, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(791217, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(795201, J$e);
                                        } finally {
                                            if (J$.Fr(795209))
                                                continue jalangiLabel2;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(791577, J$.I(typeof suite === 'undefined' ? suite = J$.R(791273, 'suite', undefined, true, true) : suite = J$.R(791273, 'suite', suite, true, true)), false)(J$.T(791281, 'Schema #2 (Basic schemas, considering *Length)', 21, false), J$.T(791569, function () {
                                jalangiLabel3:
                                    while (true) {
                                        try {
                                            J$.Fe(791545, arguments.callee, this, arguments);
                                            arguments = J$.N(791553, 'arguments', arguments, true, false, false);
                                            J$.N(791561, 'schema', schema, false, false, false);
                                            var schema = J$.W(791513, 'schema', J$.T(791505, {
                                                type: J$.T(791289, 'object', 21, false),
                                                properties: J$.T(791497, {
                                                    lorem: J$.T(791489, {
                                                        type: J$.T(791297, 'object', 21, false),
                                                        properties: J$.T(791481, {
                                                            ipsum: J$.T(791329, {
                                                                type: J$.T(791321, [
                                                                    J$.T(791305, 'string', 21, false),
                                                                    J$.T(791313, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(791345, { type: J$.T(791337, 'number', 21, false) }, 11, false),
                                                            sit: J$.T(791361, { type: J$.T(791353, 'any', 21, false) }, 11, false),
                                                            amet: J$.T(791409, {
                                                                type: J$.T(791369, 'array', 21, false),
                                                                exactLength: J$.T(791377, 6, 22, false),
                                                                items: J$.T(791401, {
                                                                    type: J$.T(791385, 'string', 21, false),
                                                                    exactLength: J$.T(791393, 10, 22, false)
                                                                }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(791473, {
                                                                type: J$.T(791417, 'array', 21, false),
                                                                minLength: J$.T(791425, 4, 22, false),
                                                                maxLength: J$.T(791433, 8, 22, false),
                                                                items: J$.T(791465, {
                                                                    type: J$.T(791441, 'string', 21, false),
                                                                    minLength: J$.T(791449, 6, 22, false),
                                                                    maxLength: J$.T(791457, 12, 22, false)
                                                                }, 11, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(791537, J$.R(791521, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(791529, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(795217, J$e);
                                        } finally {
                                            if (J$.Fr(795225))
                                                continue jalangiLabel3;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(792305, J$.I(typeof suite === 'undefined' ? suite = J$.R(791585, 'suite', undefined, true, true) : suite = J$.R(791585, 'suite', suite, true, true)), false)(J$.T(791593, 'Schema #3 (Intermediate schemas, considering *Length, lt, lte, gt, gte, eq and ne)', 21, false), J$.T(792297, function () {
                                jalangiLabel4:
                                    while (true) {
                                        try {
                                            J$.Fe(792273, arguments.callee, this, arguments);
                                            arguments = J$.N(792281, 'arguments', arguments, true, false, false);
                                            J$.N(792289, 'schema', schema, false, false, false);
                                            var schema = J$.W(792241, 'schema', J$.T(792233, {
                                                type: J$.T(791601, 'object', 21, false),
                                                properties: J$.T(792225, {
                                                    lorem: J$.T(792217, {
                                                        type: J$.T(791609, 'object', 21, false),
                                                        properties: J$.T(792209, {
                                                            ipsum: J$.T(791641, {
                                                                type: J$.T(791633, [
                                                                    J$.T(791617, 'string', 21, false),
                                                                    J$.T(791625, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(791657, { type: J$.T(791649, 'number', 21, false) }, 11, false),
                                                            sit: J$.T(791673, { type: J$.T(791665, 'any', 21, false) }, 11, false),
                                                            amet: J$.T(791721, {
                                                                type: J$.T(791681, 'array', 21, false),
                                                                exactLength: J$.T(791689, 6, 22, false),
                                                                items: J$.T(791713, {
                                                                    type: J$.T(791697, 'string', 21, false),
                                                                    exactLength: J$.T(791705, 10, 22, false)
                                                                }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(792201, {
                                                                type: J$.T(791729, 'array', 21, false),
                                                                maxLength: J$.T(791737, 16, 22, false),
                                                                items: J$.T(792193, [
                                                                    J$.T(791761, {
                                                                        type: J$.T(791745, 'number', 21, false),
                                                                        gt: J$.T(791753, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(791785, {
                                                                        type: J$.T(791769, 'number', 21, false),
                                                                        lt: J$.T(791777, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(791809, {
                                                                        type: J$.T(791793, 'number', 21, false),
                                                                        gte: J$.T(791801, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(791833, {
                                                                        type: J$.T(791817, 'number', 21, false),
                                                                        lte: J$.T(791825, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(791865, {
                                                                        type: J$.T(791841, 'number', 21, false),
                                                                        gt: J$.T(791849, 5, 22, false),
                                                                        lt: J$.T(791857, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(791897, {
                                                                        type: J$.T(791873, 'number', 21, false),
                                                                        gte: J$.T(791881, 5, 22, false),
                                                                        lte: J$.T(791889, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(791921, {
                                                                        type: J$.T(791905, 'number', 21, false),
                                                                        eq: J$.T(791913, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(791961, {
                                                                        type: J$.T(791929, 'number', 21, false),
                                                                        gte: J$.T(791937, 0, 22, false),
                                                                        lte: J$.T(791945, 1, 22, false),
                                                                        ne: J$.T(791953, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(791985, {
                                                                        type: J$.T(791969, 'integer', 21, false),
                                                                        gt: J$.T(791977, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792009, {
                                                                        type: J$.T(791993, 'integer', 21, false),
                                                                        lt: J$.T(792001, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792033, {
                                                                        type: J$.T(792017, 'integer', 21, false),
                                                                        gte: J$.T(792025, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792057, {
                                                                        type: J$.T(792041, 'integer', 21, false),
                                                                        lte: J$.T(792049, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792089, {
                                                                        type: J$.T(792065, 'integer', 21, false),
                                                                        gt: J$.T(792073, 5, 22, false),
                                                                        lt: J$.T(792081, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792121, {
                                                                        type: J$.T(792097, 'integer', 21, false),
                                                                        gte: J$.T(792105, 5, 22, false),
                                                                        lte: J$.T(792113, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792145, {
                                                                        type: J$.T(792129, 'integer', 21, false),
                                                                        eq: J$.T(792137, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792185, {
                                                                        type: J$.T(792153, 'integer', 21, false),
                                                                        gte: J$.T(792161, 0, 22, false),
                                                                        lte: J$.T(792169, 1, 22, false),
                                                                        ne: J$.T(792177, 1, 22, false)
                                                                    }, 11, false)
                                                                ], 10, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(792265, J$.R(792249, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(792257, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(795233, J$e);
                                        } finally {
                                            if (J$.Fr(795241))
                                                continue jalangiLabel4;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(793241, J$.I(typeof suite === 'undefined' ? suite = J$.R(792313, 'suite', undefined, true, true) : suite = J$.R(792313, 'suite', suite, true, true)), false)(J$.T(792321, 'Schema #4 (Advanced schemas, considering *Length, lt, lte, gt, gte, eq, array of ne and optional fields)', 21, false), J$.T(793233, function () {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(793209, arguments.callee, this, arguments);
                                            arguments = J$.N(793217, 'arguments', arguments, true, false, false);
                                            J$.N(793225, 'schema', schema, false, false, false);
                                            var schema = J$.W(793177, 'schema', J$.T(793169, {
                                                type: J$.T(792329, 'object', 21, false),
                                                properties: J$.T(793161, {
                                                    lorem: J$.T(793153, {
                                                        type: J$.T(792337, 'object', 21, false),
                                                        properties: J$.T(793145, {
                                                            ipsum: J$.T(792369, {
                                                                type: J$.T(792361, [
                                                                    J$.T(792345, 'string', 21, false),
                                                                    J$.T(792353, 'integer', 21, false)
                                                                ], 10, false)
                                                            }, 11, false),
                                                            dolor: J$.T(792393, {
                                                                type: J$.T(792377, 'number', 21, false),
                                                                optional: J$.T(792385, true, 23, false)
                                                            }, 11, false),
                                                            sit: J$.T(792417, {
                                                                type: J$.T(792401, 'any', 21, false),
                                                                optional: J$.T(792409, true, 23, false)
                                                            }, 11, false),
                                                            amet: J$.T(792481, {
                                                                type: J$.T(792425, 'array', 21, false),
                                                                exactLength: J$.T(792433, 6, 22, false),
                                                                optional: J$.T(792441, true, 23, false),
                                                                items: J$.T(792473, {
                                                                    type: J$.T(792449, 'string', 21, false),
                                                                    optional: J$.T(792457, true, 23, false),
                                                                    exactLength: J$.T(792465, 10, 22, false)
                                                                }, 11, false)
                                                            }, 11, false),
                                                            consectetur: J$.T(793137, {
                                                                type: J$.T(792489, 'array', 21, false),
                                                                maxLength: J$.T(792497, 18, 22, false),
                                                                optional: J$.T(792505, true, 23, false),
                                                                items: J$.T(793129, [
                                                                    J$.T(792529, {
                                                                        type: J$.T(792513, 'number', 21, false),
                                                                        gt: J$.T(792521, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792553, {
                                                                        type: J$.T(792537, 'number', 21, false),
                                                                        lt: J$.T(792545, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792577, {
                                                                        type: J$.T(792561, 'number', 21, false),
                                                                        gte: J$.T(792569, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792601, {
                                                                        type: J$.T(792585, 'number', 21, false),
                                                                        lte: J$.T(792593, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792633, {
                                                                        type: J$.T(792609, 'number', 21, false),
                                                                        gt: J$.T(792617, 5, 22, false),
                                                                        lt: J$.T(792625, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792665, {
                                                                        type: J$.T(792641, 'number', 21, false),
                                                                        gte: J$.T(792649, 5, 22, false),
                                                                        lte: J$.T(792657, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792689, {
                                                                        type: J$.T(792673, 'number', 21, false),
                                                                        eq: J$.T(792681, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792745, {
                                                                        type: J$.T(792697, 'number', 21, false),
                                                                        gte: J$.T(792705, 0, 22, false),
                                                                        lte: J$.T(792713, 2, 22, false),
                                                                        ne: J$.T(792737, [
                                                                            J$.T(792721, 0, 22, false),
                                                                            J$.T(792729, 1, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false),
                                                                    J$.T(792769, {
                                                                        type: J$.T(792753, 'integer', 21, false),
                                                                        gt: J$.T(792761, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792793, {
                                                                        type: J$.T(792777, 'integer', 21, false),
                                                                        lt: J$.T(792785, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792817, {
                                                                        type: J$.T(792801, 'integer', 21, false),
                                                                        gte: J$.T(792809, 50, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792841, {
                                                                        type: J$.T(792825, 'integer', 21, false),
                                                                        lte: J$.T(792833, 1, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792873, {
                                                                        type: J$.T(792849, 'integer', 21, false),
                                                                        gt: J$.T(792857, 5, 22, false),
                                                                        lt: J$.T(792865, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792905, {
                                                                        type: J$.T(792881, 'integer', 21, false),
                                                                        gte: J$.T(792889, 5, 22, false),
                                                                        lte: J$.T(792897, 10, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792929, {
                                                                        type: J$.T(792913, 'integer', 21, false),
                                                                        eq: J$.T(792921, 7, 22, false)
                                                                    }, 11, false),
                                                                    J$.T(792985, {
                                                                        type: J$.T(792937, 'integer', 21, false),
                                                                        gte: J$.T(792945, 0, 22, false),
                                                                        lte: J$.T(792953, 2, 22, false),
                                                                        ne: J$.T(792977, [
                                                                            J$.T(792961, 0, 22, false),
                                                                            J$.T(792969, 1, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false),
                                                                    J$.T(793049, {
                                                                        type: J$.T(792993, 'integer', 21, false),
                                                                        gte: J$.T(793001, 0, 22, false),
                                                                        lte: J$.T(793009, 3, 22, false),
                                                                        ne: J$.T(793041, [
                                                                            J$.T(793017, 0, 22, false),
                                                                            J$.T(793025, 1, 22, false),
                                                                            J$.T(793033, 2, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false),
                                                                    J$.T(793121, {
                                                                        type: J$.T(793057, 'integer', 21, false),
                                                                        gte: J$.T(793065, 0, 22, false),
                                                                        lte: J$.T(793073, 4, 22, false),
                                                                        ne: J$.T(793113, [
                                                                            J$.T(793081, 0, 22, false),
                                                                            J$.T(793089, 1, 22, false),
                                                                            J$.T(793097, 2, 22, false),
                                                                            J$.T(793105, 3, 22, false)
                                                                        ], 10, false)
                                                                    }, 11, false)
                                                                ], 10, false)
                                                            }, 11, false)
                                                        }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(793201, J$.R(793185, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(793193, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(795249, J$e);
                                        } finally {
                                            if (J$.Fr(795257))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(793697, J$.I(typeof suite === 'undefined' ? suite = J$.R(793249, 'suite', undefined, true, true) : suite = J$.R(793249, 'suite', suite, true, true)), false)(J$.T(793257, 'Schema #5 (Advanced schemas, considering *Length, eq and format fields)', 21, false), J$.T(793689, function () {
                                jalangiLabel6:
                                    while (true) {
                                        try {
                                            J$.Fe(793665, arguments.callee, this, arguments);
                                            arguments = J$.N(793673, 'arguments', arguments, true, false, false);
                                            J$.N(793681, 'schema', schema, false, false, false);
                                            var schema = J$.W(793633, 'schema', J$.T(793625, {
                                                type: J$.T(793265, 'object', 21, false),
                                                properties: J$.T(793617, {
                                                    lorem: J$.T(793609, {
                                                        type: J$.T(793273, 'array', 21, false),
                                                        items: J$.T(793601, [
                                                            J$.T(793297, {
                                                                type: J$.T(793281, 'string', 21, false),
                                                                pattern: J$.T(793289, 'date-time', 21, false)
                                                            }, 11, false),
                                                            J$.T(793321, {
                                                                type: J$.T(793305, 'string', 21, false),
                                                                pattern: J$.T(793313, 'date-time', 21, false)
                                                            }, 11, false),
                                                            J$.T(793345, {
                                                                type: J$.T(793329, 'string', 21, false),
                                                                pattern: J$.T(793337, 'date', 21, false)
                                                            }, 11, false),
                                                            J$.T(793369, {
                                                                type: J$.T(793353, 'string', 21, false),
                                                                pattern: J$.T(793361, 'time', 21, false)
                                                            }, 11, false),
                                                            J$.T(793393, {
                                                                type: J$.T(793377, 'string', 21, false),
                                                                pattern: J$.T(793385, 'color', 21, false)
                                                            }, 11, false),
                                                            J$.T(793425, {
                                                                type: J$.T(793401, 'string', 21, false),
                                                                pattern: J$.T(793409, 'color', 21, false),
                                                                eq: J$.T(793417, '#FFffFF', 21, false)
                                                            }, 11, false),
                                                            J$.T(793449, {
                                                                type: J$.T(793433, 'string', 21, false),
                                                                pattern: J$.T(793441, 'numeric', 21, false)
                                                            }, 11, false),
                                                            J$.T(793473, {
                                                                type: J$.T(793457, 'string', 21, false),
                                                                pattern: J$.T(793465, 'integer', 21, false)
                                                            }, 11, false),
                                                            J$.T(793497, {
                                                                type: J$.T(793481, 'string', 21, false),
                                                                pattern: J$.T(793489, 'decimal', 21, false)
                                                            }, 11, false),
                                                            J$.T(793521, {
                                                                type: J$.T(793505, 'string', 21, false),
                                                                pattern: J$.T(793513, 'alpha', 21, false)
                                                            }, 11, false),
                                                            J$.T(793545, {
                                                                type: J$.T(793529, 'string', 21, false),
                                                                pattern: J$.T(793537, 'alphaNumeric', 21, false)
                                                            }, 11, false),
                                                            J$.T(793569, {
                                                                type: J$.T(793553, 'string', 21, false),
                                                                pattern: J$.T(793561, 'alphaDash', 21, false)
                                                            }, 11, false),
                                                            J$.T(793593, {
                                                                type: J$.T(793577, 'string', 21, false),
                                                                pattern: J$.T(793585, 'javascript', 21, false)
                                                            }, 11, false)
                                                        ], 10, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            J$.F(793657, J$.R(793641, 'testRandomCandidates', testRandomCandidates, false, true), false)(J$.R(793649, 'schema', schema, false, false));
                                        } catch (J$e) {
                                            J$.Ex(795265, J$e);
                                        } finally {
                                            if (J$.Fr(795273))
                                                continue jalangiLabel6;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(794345, J$.I(typeof suite === 'undefined' ? suite = J$.R(793705, 'suite', undefined, true, true) : suite = J$.R(793705, 'suite', suite, true, true)), false)(J$.T(793713, 'Schema #6 (Generate several candidate with one call)', 21, false), J$.T(794337, function () {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(794305, arguments.callee, this, arguments);
                                            arguments = J$.N(794313, 'arguments', arguments, true, false, false);
                                            J$.N(794321, 'schema', schema, false, false, false);
                                            J$.N(794329, 'candidates', candidates, false, false, false);
                                            var schema = J$.W(793761, 'schema', J$.T(793753, {
                                                type: J$.T(793721, 'object', 21, false),
                                                properties: J$.T(793745, { lorem: J$.T(793737, { type: J$.T(793729, 'integer', 21, false) }, 11, false) }, 11, false)
                                            }, 11, false), schema, false, false);
                                            var candidates = J$.W(793801, 'candidates', J$.M(793793, J$.R(793769, 'si', si, false, true), 'generate', false)(J$.R(793777, 'schema', schema, false, false), J$.R(793785, 'NB_TEST', NB_TEST, false, true)), candidates, false, false);
                                            J$.M(793873, J$.G(793857, J$.M(793849, J$.G(793833, J$.G(793825, J$.G(793817, J$.R(793809, 'candidates', candidates, false, false), 'should'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(793841, 'Array', undefined, true, true) : Array = J$.R(793841, 'Array', Array, true, true))), 'with'), 'lengthOf', false)(J$.R(793865, 'NB_TEST', NB_TEST, false, true));
                                            J$.M(794297, J$.R(793881, 'candidates', candidates, false, false), 'forEach', false)(J$.T(794289, function (candidate, i) {
                                                jalangiLabel8:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(794257, arguments.callee, this, arguments);
                                                            arguments = J$.N(794265, 'arguments', arguments, true, false, false);
                                                            candidate = J$.N(794273, 'candidate', candidate, true, false, false);
                                                            i = J$.N(794281, 'i', i, true, false, false);
                                                            J$.F(794249, J$.I(typeof test === 'undefined' ? test = J$.R(793889, 'test', undefined, true, true) : test = J$.R(793889, 'test', test, true, true)), false)(J$.B(67610, '+', J$.T(793897, 'Random candidate #', 21, false), J$.B(67602, '+', J$.R(793905, 'i', i, false, false), J$.T(793913, 1, 22, false))), J$.T(794241, function () {
                                                                jalangiLabel7:
                                                                    while (true) {
                                                                        try {
                                                                            J$.Fe(794217, arguments.callee, this, arguments);
                                                                            arguments = J$.N(794225, 'arguments', arguments, true, false, false);
                                                                            J$.N(794233, 'result', result, false, false, false);
                                                                            var result = J$.W(793953, 'result', J$.M(793945, J$.R(793921, 'si', si, false, true), 'validate', false)(J$.R(793929, 'schema', schema, false, false), J$.R(793937, 'candidate', candidate, false, false)), result, false, false);
                                                                            J$.G(793993, J$.G(793985, J$.G(793977, J$.G(793969, J$.R(793961, 'result', result, false, false), 'should'), 'be'), 'an'), 'Object');
                                                                            try {
                                                                                J$.M(794057, J$.G(794041, J$.M(794033, J$.G(794017, J$.G(794009, J$.R(794001, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(794025, 'valid', 21, false)), 'with'), 'equal', false)(J$.T(794049, true, 23, false));
                                                                                J$.M(794169, J$.G(794153, J$.G(794145, J$.M(794137, J$.G(794121, J$.G(794113, J$.G(794105, J$.M(794097, J$.G(794081, J$.G(794073, J$.R(794065, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(794089, 'error', 21, false)), 'with'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(794129, 'Array', undefined, true, true) : Array = J$.R(794129, 'Array', Array, true, true))), 'and'), 'be'), 'lengthOf', false)(J$.T(794161, 0, 22, false));
                                                                            } catch (e) {
                                                                                J$.N(794209, 'e', e, false, false, true);
                                                                                throw J$.F(794201, J$.I(typeof Error === 'undefined' ? Error = J$.R(794177, 'Error', undefined, true, true) : Error = J$.R(794177, 'Error', Error, true, true)), true)(J$.M(794193, J$.R(794185, 'result', result, false, false), 'format', false)());
                                                                            }
                                                                        } catch (J$e) {
                                                                            J$.Ex(795281, J$e);
                                                                        } finally {
                                                                            if (J$.Fr(795289))
                                                                                continue jalangiLabel7;
                                                                            else
                                                                                return J$.Ra();
                                                                        }
                                                                    }
                                                            }, 12, false));
                                                        } catch (J$e) {
                                                            J$.Ex(795297, J$e);
                                                        } finally {
                                                            if (J$.Fr(795305))
                                                                continue jalangiLabel8;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(795313, J$e);
                                        } finally {
                                            if (J$.Fr(795321))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(795081, J$.I(typeof suite === 'undefined' ? suite = J$.R(794353, 'suite', undefined, true, true) : suite = J$.R(794353, 'suite', suite, true, true)), false)(J$.T(794361, 'Schema #7 (Globing for Object keys)', 21, false), J$.T(795073, function () {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(795041, arguments.callee, this, arguments);
                                            arguments = J$.N(795049, 'arguments', arguments, true, false, false);
                                            J$.N(795057, 'schema', schema, false, false, false);
                                            J$.N(795065, 'i', i, false, false, false);
                                            var schema = J$.W(794473, 'schema', J$.T(794465, {
                                                type: J$.T(794369, 'object', 21, false),
                                                properties: J$.T(794457, {
                                                    globString: J$.T(794409, {
                                                        type: J$.T(794377, 'object', 21, false),
                                                        properties: J$.T(794401, { '*': J$.T(794393, { type: J$.T(794385, 'string', 21, false) }, 11, false) }, 11, false)
                                                    }, 11, false),
                                                    globInteger: J$.T(794449, {
                                                        type: J$.T(794417, 'object', 21, false),
                                                        properties: J$.T(794441, { '*': J$.T(794433, { type: J$.T(794425, 'integer', 21, false) }, 11, false) }, 11, false)
                                                    }, 11, false)
                                                }, 11, false)
                                            }, 11, false), schema, false, false);
                                            for (var i = J$.W(794489, 'i', J$.T(794481, 1, 22, false), i, false, false); J$.C(49080, J$.B(67618, '<=', J$.R(794497, 'i', i, false, false), J$.R(794505, 'NB_TEST', NB_TEST, false, true))); J$.B(67642, '-', i = J$.W(794521, 'i', J$.B(67634, '+', J$.U(67626, '+', J$.R(794513, 'i', i, false, false)), 1), i, false, false), 1)) {
                                                J$.F(795033, J$.I(typeof test === 'undefined' ? test = J$.R(794529, 'test', undefined, true, true) : test = J$.R(794529, 'test', test, true, true)), false)(J$.B(67650, '+', J$.T(794537, 'Random candidate #', 21, false), J$.R(794545, 'i', i, false, false)), J$.T(795025, function () {
                                                    jalangiLabel10:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(794993, arguments.callee, this, arguments);
                                                                arguments = J$.N(795001, 'arguments', arguments, true, false, false);
                                                                J$.N(795009, 'candidate', candidate, false, false, false);
                                                                J$.N(795017, 'result', result, false, false, false);
                                                                var candidate = J$.W(794577, 'candidate', J$.M(794569, J$.R(794553, 'si', si, false, true), 'generate', false)(J$.R(794561, 'schema', schema, false, false)), candidate, false, false);
                                                                var result = J$.W(794617, 'result', J$.M(794609, J$.R(794585, 'si', si, false, true), 'validate', false)(J$.R(794593, 'schema', schema, false, false), J$.R(794601, 'candidate', candidate, false, false)), result, false, false);
                                                                J$.G(794657, J$.G(794649, J$.G(794641, J$.G(794633, J$.R(794625, 'result', result, false, false), 'should'), 'be'), 'an'), 'Object');
                                                                J$.M(794713, J$.G(794697, J$.G(794689, J$.G(794681, J$.G(794673, J$.R(794665, 'candidate', candidate, false, false), 'globString'), 'should'), 'not'), 'have'), 'property', false)(J$.T(794705, '*', 21, false));
                                                                J$.M(794769, J$.G(794753, J$.G(794745, J$.G(794737, J$.G(794729, J$.R(794721, 'candidate', candidate, false, false), 'globInteger'), 'should'), 'not'), 'have'), 'property', false)(J$.T(794761, '*', 21, false));
                                                                try {
                                                                    J$.M(794833, J$.G(794817, J$.M(794809, J$.G(794793, J$.G(794785, J$.R(794777, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(794801, 'valid', 21, false)), 'with'), 'equal', false)(J$.T(794825, true, 23, false));
                                                                    J$.M(794945, J$.G(794929, J$.G(794921, J$.M(794913, J$.G(794897, J$.G(794889, J$.G(794881, J$.M(794873, J$.G(794857, J$.G(794849, J$.R(794841, 'result', result, false, false), 'should'), 'have'), 'property', false)(J$.T(794865, 'error', 21, false)), 'with'), 'be'), 'an'), 'instanceof', false)(J$.I(typeof Array === 'undefined' ? Array = J$.R(794905, 'Array', undefined, true, true) : Array = J$.R(794905, 'Array', Array, true, true))), 'and'), 'be'), 'lengthOf', false)(J$.T(794937, 0, 22, false));
                                                                } catch (e) {
                                                                    J$.N(794985, 'e', e, false, false, true);
                                                                    throw J$.F(794977, J$.I(typeof Error === 'undefined' ? Error = J$.R(794953, 'Error', undefined, true, true) : Error = J$.R(794953, 'Error', Error, true, true)), true)(J$.M(794969, J$.R(794961, 'result', result, false, false), 'format', false)());
                                                                }
                                                            } catch (J$e) {
                                                                J$.Ex(795329, J$e);
                                                            } finally {
                                                                if (J$.Fr(795337))
                                                                    continue jalangiLabel10;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false));
                                            }
                                        } catch (J$e) {
                                            J$.Ex(795345, J$e);
                                        } finally {
                                            if (J$.Fr(795353))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(795361, J$e);
                        } finally {
                            if (J$.Fr(795369))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(795377, J$e);
        } finally {
            if (J$.Sr(795385))
                continue jalangiLabel13;
            else
                break jalangiLabel13;
        }
    }
// JALANGI DO NOT INSTRUMENT
