// bintrees/test/test_api.js
J$.noInstrEval = false;
jalangiLabel32:
    while (true) {
        try {
            J$.Se(717705, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCoinBasePro/node_modules/bintrees/test/test_api.js');
            function clear(assert, tree_class) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(709881, arguments.callee, this, arguments);
                            arguments = J$.N(709889, 'arguments', arguments, true, false, false);
                            assert = J$.N(709897, 'assert', assert, true, false, false);
                            tree_class = J$.N(709905, 'tree_class', tree_class, true, false, false);
                            J$.N(709913, 'inserts', inserts, false, false, false);
                            J$.N(709921, 'tree', tree, false, false, false);
                            var inserts = J$.W(709721, 'inserts', J$.M(709713, J$.R(709681, 'loader', loader, false, true), 'get_inserts', false)(J$.M(709705, J$.R(709689, 'loader', loader, false, true), 'load', false)(J$.R(709697, 'SAMPLE_FILE', SAMPLE_FILE, false, true))), inserts, false, false);
                            var tree = J$.W(709761, 'tree', J$.M(709753, J$.R(709729, 'loader', loader, false, true), 'build_tree', false)(J$.R(709737, 'tree_class', tree_class, false, false), J$.R(709745, 'inserts', inserts, false, false)), tree, false, false);
                            J$.M(709777, J$.R(709769, 'tree', tree, false, false), 'clear', false)();
                            J$.M(709873, J$.R(709785, 'inserts', inserts, false, false), 'forEach', false)(J$.T(709865, function (data) {
                                jalangiLabel0:
                                    while (true) {
                                        try {
                                            J$.Fe(709841, arguments.callee, this, arguments);
                                            arguments = J$.N(709849, 'arguments', arguments, true, false, false);
                                            data = J$.N(709857, 'data', data, true, false, false);
                                            J$.M(709833, J$.R(709793, 'assert', assert, false, false), 'equal', false)(J$.M(709817, J$.R(709801, 'tree', tree, false, false), 'find', false)(J$.R(709809, 'data', data, false, false)), J$.T(709825, null, 25, false));
                                        } catch (J$e) {
                                            J$.Ex(717985, J$e);
                                        } finally {
                                            if (J$.Fr(717993))
                                                continue jalangiLabel0;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(718001, J$e);
                        } finally {
                            if (J$.Fr(718009))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function dup(assert, tree_class) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(710265, arguments.callee, this, arguments);
                            arguments = J$.N(710273, 'arguments', arguments, true, false, false);
                            assert = J$.N(710281, 'assert', assert, true, false, false);
                            tree_class = J$.N(710289, 'tree_class', tree_class, true, false, false);
                            J$.N(710297, 'tree', tree, false, false, false);
                            var tree = J$.W(709953, 'tree', J$.M(709945, J$.R(709929, 'loader', loader, false, true), 'new_tree', false)(J$.R(709937, 'tree_class', tree_class, false, false)), tree, false, false);
                            J$.M(709993, J$.R(709961, 'assert', assert, false, false), 'ok', false)(J$.M(709985, J$.R(709969, 'tree', tree, false, false), 'insert', false)(J$.T(709977, 100, 22, false)));
                            J$.M(710033, J$.R(710001, 'assert', assert, false, false), 'ok', false)(J$.M(710025, J$.R(710009, 'tree', tree, false, false), 'insert', false)(J$.T(710017, 101, 22, false)));
                            J$.M(710073, J$.R(710041, 'assert', assert, false, false), 'ok', false)(J$.U(136922, '!', J$.M(710065, J$.R(710049, 'tree', tree, false, false), 'insert', false)(J$.T(710057, 101, 22, false))));
                            J$.M(710113, J$.R(710081, 'assert', assert, false, false), 'ok', false)(J$.U(136930, '!', J$.M(710105, J$.R(710089, 'tree', tree, false, false), 'insert', false)(J$.T(710097, 100, 22, false))));
                            J$.M(710137, J$.R(710121, 'tree', tree, false, false), 'remove', false)(J$.T(710129, 100, 22, false));
                            J$.M(710177, J$.R(710145, 'assert', assert, false, false), 'ok', false)(J$.U(136938, '!', J$.M(710169, J$.R(710153, 'tree', tree, false, false), 'insert', false)(J$.T(710161, 101, 22, false))));
                            J$.M(710217, J$.R(710185, 'assert', assert, false, false), 'ok', false)(J$.M(710209, J$.R(710193, 'tree', tree, false, false), 'insert', false)(J$.T(710201, 100, 22, false)));
                            J$.M(710257, J$.R(710225, 'assert', assert, false, false), 'ok', false)(J$.U(136946, '!', J$.M(710249, J$.R(710233, 'tree', tree, false, false), 'insert', false)(J$.T(710241, 100, 22, false))));
                        } catch (J$e) {
                            J$.Ex(718017, J$e);
                        } finally {
                            if (J$.Fr(718025))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function nonexist(assert, tree_class) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(710481, arguments.callee, this, arguments);
                            arguments = J$.N(710489, 'arguments', arguments, true, false, false);
                            assert = J$.N(710497, 'assert', assert, true, false, false);
                            tree_class = J$.N(710505, 'tree_class', tree_class, true, false, false);
                            J$.N(710513, 'tree', tree, false, false, false);
                            var tree = J$.W(710329, 'tree', J$.M(710321, J$.R(710305, 'loader', loader, false, true), 'new_tree', false)(J$.R(710313, 'tree_class', tree_class, false, false)), tree, false, false);
                            J$.M(710369, J$.R(710337, 'assert', assert, false, false), 'ok', false)(J$.U(136954, '!', J$.M(710361, J$.R(710345, 'tree', tree, false, false), 'remove', false)(J$.T(710353, 100, 22, false))));
                            J$.M(710393, J$.R(710377, 'tree', tree, false, false), 'insert', false)(J$.T(710385, 100, 22, false));
                            J$.M(710433, J$.R(710401, 'assert', assert, false, false), 'ok', false)(J$.U(136962, '!', J$.M(710425, J$.R(710409, 'tree', tree, false, false), 'remove', false)(J$.T(710417, 101, 22, false))));
                            J$.M(710473, J$.R(710441, 'assert', assert, false, false), 'ok', false)(J$.M(710465, J$.R(710449, 'tree', tree, false, false), 'remove', false)(J$.T(710457, 100, 22, false)));
                        } catch (J$e) {
                            J$.Ex(718033, J$e);
                        } finally {
                            if (J$.Fr(718041))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function minmax(assert, tree_class) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(710833, arguments.callee, this, arguments);
                            arguments = J$.N(710841, 'arguments', arguments, true, false, false);
                            assert = J$.N(710849, 'assert', assert, true, false, false);
                            tree_class = J$.N(710857, 'tree_class', tree_class, true, false, false);
                            J$.N(710865, 'tree', tree, false, false, false);
                            J$.N(710873, 'inserts', inserts, false, false, false);
                            var tree = J$.W(710545, 'tree', J$.M(710537, J$.R(710521, 'loader', loader, false, true), 'new_tree', false)(J$.R(710529, 'tree_class', tree_class, false, false)), tree, false, false);
                            J$.M(710585, J$.R(710553, 'assert', assert, false, false), 'equal', false)(J$.M(710569, J$.R(710561, 'tree', tree, false, false), 'min', false)(), J$.T(710577, null, 25, false));
                            J$.M(710625, J$.R(710593, 'assert', assert, false, false), 'equal', false)(J$.M(710609, J$.R(710601, 'tree', tree, false, false), 'max', false)(), J$.T(710617, null, 25, false));
                            var inserts = J$.W(710673, 'inserts', J$.M(710665, J$.R(710633, 'loader', loader, false, true), 'get_inserts', false)(J$.M(710657, J$.R(710641, 'loader', loader, false, true), 'load', false)(J$.R(710649, 'SAMPLE_FILE', SAMPLE_FILE, false, true))), inserts, false, false);
                            tree = J$.W(710713, 'tree', J$.M(710705, J$.R(710681, 'loader', loader, false, true), 'build_tree', false)(J$.R(710689, 'tree_class', tree_class, false, false), J$.R(710697, 'inserts', inserts, false, false)), tree, false, false);
                            J$.M(710769, J$.R(710721, 'assert', assert, false, false), 'equal', false)(J$.M(710737, J$.R(710729, 'tree', tree, false, false), 'min', false)(), J$.M(710761, J$.R(710745, '_', _, false, true), 'min', false)(J$.R(710753, 'inserts', inserts, false, false)));
                            J$.M(710825, J$.R(710777, 'assert', assert, false, false), 'equal', false)(J$.M(710793, J$.R(710785, 'tree', tree, false, false), 'max', false)(), J$.M(710817, J$.R(710801, '_', _, false, true), 'max', false)(J$.R(710809, 'inserts', inserts, false, false)));
                        } catch (J$e) {
                            J$.Ex(718049, J$e);
                        } finally {
                            if (J$.Fr(718057))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function forward_it(assert, tree_class) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(711297, arguments.callee, this, arguments);
                            arguments = J$.N(711305, 'arguments', arguments, true, false, false);
                            assert = J$.N(711313, 'assert', assert, true, false, false);
                            tree_class = J$.N(711321, 'tree_class', tree_class, true, false, false);
                            J$.N(711329, 'inserts', inserts, false, false, false);
                            J$.N(711337, 'tree', tree, false, false, false);
                            J$.N(711345, 'items', items, false, false, false);
                            J$.N(711353, 'it', it, false, false, false);
                            J$.N(711361, 'data', data, false, false, false);
                            var inserts = J$.W(710921, 'inserts', J$.M(710913, J$.R(710881, 'loader', loader, false, true), 'get_inserts', false)(J$.M(710905, J$.R(710889, 'loader', loader, false, true), 'load', false)(J$.R(710897, 'SAMPLE_FILE', SAMPLE_FILE, false, true))), inserts, false, false);
                            var tree = J$.W(710961, 'tree', J$.M(710953, J$.R(710929, 'loader', loader, false, true), 'build_tree', false)(J$.R(710937, 'tree_class', tree_class, false, false), J$.R(710945, 'inserts', inserts, false, false)), tree, false, false);
                            var items = J$.W(710977, 'items', J$.T(710969, [], 10, false), items, false, false);
                            var it = J$.W(711001, 'it', J$.M(710993, J$.R(710985, 'tree', tree, false, false), 'iterator', false)(), it, false, false), data;
                            while (J$.C(46968, J$.B(136970, '!==', data = J$.W(711025, 'data', J$.M(711017, J$.R(711009, 'it', it, false, false), 'next', false)(), data, false, false), J$.T(711033, null, 25, false)))) {
                                J$.M(711057, J$.R(711041, 'items', items, false, false), 'push', false)(J$.R(711049, 'data', data, false, false));
                            }
                            J$.M(711137, J$.R(711065, 'inserts', inserts, false, false), 'sort', false)(J$.T(711129, function (a, b) {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(711097, arguments.callee, this, arguments);
                                            arguments = J$.N(711105, 'arguments', arguments, true, false, false);
                                            a = J$.N(711113, 'a', a, true, false, false);
                                            b = J$.N(711121, 'b', b, true, false, false);
                                            return J$.Rt(711089, J$.B(136978, '-', J$.R(711073, 'a', a, false, false), J$.R(711081, 'b', b, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(718065, J$e);
                                        } finally {
                                            if (J$.Fr(718073))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.M(711169, J$.R(711145, 'assert', assert, false, false), 'deepEqual', false)(J$.R(711153, 'items', items, false, false), J$.R(711161, 'inserts', inserts, false, false));
                            items = J$.W(711185, 'items', J$.T(711177, [], 10, false), items, false, false);
                            J$.M(711257, J$.R(711193, 'tree', tree, false, false), 'each', false)(J$.T(711249, function (data) {
                                jalangiLabel6:
                                    while (true) {
                                        try {
                                            J$.Fe(711225, arguments.callee, this, arguments);
                                            arguments = J$.N(711233, 'arguments', arguments, true, false, false);
                                            data = J$.N(711241, 'data', data, true, false, false);
                                            J$.M(711217, J$.R(711201, 'items', items, false, false), 'push', false)(J$.R(711209, 'data', data, false, false));
                                        } catch (J$e) {
                                            J$.Ex(718081, J$e);
                                        } finally {
                                            if (J$.Fr(718089))
                                                continue jalangiLabel6;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.M(711289, J$.R(711265, 'assert', assert, false, false), 'deepEqual', false)(J$.R(711273, 'items', items, false, false), J$.R(711281, 'inserts', inserts, false, false));
                        } catch (J$e) {
                            J$.Ex(718097, J$e);
                        } finally {
                            if (J$.Fr(718105))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function forward_it_break(assert, tree_class) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(711857, arguments.callee, this, arguments);
                            arguments = J$.N(711865, 'arguments', arguments, true, false, false);
                            assert = J$.N(711873, 'assert', assert, true, false, false);
                            tree_class = J$.N(711881, 'tree_class', tree_class, true, false, false);
                            J$.N(711889, 'inserts', inserts, false, false, false);
                            J$.N(711897, 'tree', tree, false, false, false);
                            J$.N(711905, 'items', items, false, false, false);
                            J$.N(711913, 'it', it, false, false, false);
                            J$.N(711921, 'data', data, false, false, false);
                            J$.N(711929, 'i', i, false, false, false);
                            var inserts = J$.W(711409, 'inserts', J$.M(711401, J$.R(711369, 'loader', loader, false, true), 'get_inserts', false)(J$.M(711393, J$.R(711377, 'loader', loader, false, true), 'load', false)(J$.R(711385, 'SAMPLE_FILE', SAMPLE_FILE, false, true))), inserts, false, false);
                            var tree = J$.W(711449, 'tree', J$.M(711441, J$.R(711417, 'loader', loader, false, true), 'build_tree', false)(J$.R(711425, 'tree_class', tree_class, false, false), J$.R(711433, 'inserts', inserts, false, false)), tree, false, false);
                            var items = J$.W(711465, 'items', J$.T(711457, [], 10, false), items, false, false);
                            var it = J$.W(711489, 'it', J$.M(711481, J$.R(711473, 'tree', tree, false, false), 'iterator', false)(), it, false, false), data;
                            while (J$.C(46976, J$.B(136986, '!==', data = J$.W(711513, 'data', J$.M(711505, J$.R(711497, 'it', it, false, false), 'next', false)(), data, false, false), J$.T(711521, null, 25, false)))) {
                                J$.M(711545, J$.R(711529, 'items', items, false, false), 'push', false)(J$.R(711537, 'data', data, false, false));
                            }
                            J$.M(711625, J$.R(711553, 'inserts', inserts, false, false), 'sort', false)(J$.T(711617, function (a, b) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(711585, arguments.callee, this, arguments);
                                            arguments = J$.N(711593, 'arguments', arguments, true, false, false);
                                            a = J$.N(711601, 'a', a, true, false, false);
                                            b = J$.N(711609, 'b', b, true, false, false);
                                            return J$.Rt(711577, J$.B(136994, '-', J$.R(711561, 'a', a, false, false), J$.R(711569, 'b', b, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(718113, J$e);
                                        } finally {
                                            if (J$.Fr(718121))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.M(711657, J$.R(711633, 'assert', assert, false, false), 'deepEqual', false)(J$.R(711641, 'items', items, false, false), J$.R(711649, 'inserts', inserts, false, false));
                            items = J$.W(711673, 'items', J$.T(711665, [], 10, false), items, false, false);
                            var i = J$.W(711689, 'i', J$.T(711681, 0, 22, false), i, false, false);
                            J$.M(711809, J$.R(711697, 'tree', tree, false, false), 'each', false)(J$.T(711801, function (data) {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(711777, arguments.callee, this, arguments);
                                            arguments = J$.N(711785, 'arguments', arguments, true, false, false);
                                            data = J$.N(711793, 'data', data, true, false, false);
                                            J$.M(711721, J$.R(711705, 'items', items, false, false), 'push', false)(J$.R(711713, 'data', data, false, false));
                                            if (J$.C(46984, J$.B(137002, '===', J$.R(711729, 'i', i, false, false), J$.T(711737, 3, 22, false)))) {
                                                return J$.Rt(711753, J$.T(711745, false, 23, false));
                                            }
                                            J$.B(137026, '-', i = J$.W(711769, 'i', J$.B(137018, '+', J$.U(137010, '+', J$.R(711761, 'i', i, false, false)), 1), i, false, false), 1);
                                        } catch (J$e) {
                                            J$.Ex(718129, J$e);
                                        } finally {
                                            if (J$.Fr(718137))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.M(711849, J$.R(711817, 'assert', assert, false, false), 'equal', false)(J$.G(711833, J$.R(711825, 'items', items, false, false), 'length'), J$.T(711841, 4, 22, false));
                        } catch (J$e) {
                            J$.Ex(718145, J$e);
                        } finally {
                            if (J$.Fr(718153))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function reverse_it(assert, tree_class) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(712353, arguments.callee, this, arguments);
                            arguments = J$.N(712361, 'arguments', arguments, true, false, false);
                            assert = J$.N(712369, 'assert', assert, true, false, false);
                            tree_class = J$.N(712377, 'tree_class', tree_class, true, false, false);
                            J$.N(712385, 'inserts', inserts, false, false, false);
                            J$.N(712393, 'tree', tree, false, false, false);
                            J$.N(712401, 'items', items, false, false, false);
                            J$.N(712409, 'it', it, false, false, false);
                            J$.N(712417, 'data', data, false, false, false);
                            var inserts = J$.W(711977, 'inserts', J$.M(711969, J$.R(711937, 'loader', loader, false, true), 'get_inserts', false)(J$.M(711961, J$.R(711945, 'loader', loader, false, true), 'load', false)(J$.R(711953, 'SAMPLE_FILE', SAMPLE_FILE, false, true))), inserts, false, false);
                            var tree = J$.W(712017, 'tree', J$.M(712009, J$.R(711985, 'loader', loader, false, true), 'build_tree', false)(J$.R(711993, 'tree_class', tree_class, false, false), J$.R(712001, 'inserts', inserts, false, false)), tree, false, false);
                            var items = J$.W(712033, 'items', J$.T(712025, [], 10, false), items, false, false);
                            var it = J$.W(712057, 'it', J$.M(712049, J$.R(712041, 'tree', tree, false, false), 'iterator', false)(), it, false, false), data;
                            while (J$.C(46992, J$.B(137034, '!==', data = J$.W(712081, 'data', J$.M(712073, J$.R(712065, 'it', it, false, false), 'prev', false)(), data, false, false), J$.T(712089, null, 25, false)))) {
                                J$.M(712113, J$.R(712097, 'items', items, false, false), 'push', false)(J$.R(712105, 'data', data, false, false));
                            }
                            J$.M(712193, J$.R(712121, 'inserts', inserts, false, false), 'sort', false)(J$.T(712185, function (a, b) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(712153, arguments.callee, this, arguments);
                                            arguments = J$.N(712161, 'arguments', arguments, true, false, false);
                                            a = J$.N(712169, 'a', a, true, false, false);
                                            b = J$.N(712177, 'b', b, true, false, false);
                                            return J$.Rt(712145, J$.B(137042, '-', J$.R(712129, 'b', b, false, false), J$.R(712137, 'a', a, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(718161, J$e);
                                        } finally {
                                            if (J$.Fr(718169))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.M(712225, J$.R(712201, 'assert', assert, false, false), 'deepEqual', false)(J$.R(712209, 'items', items, false, false), J$.R(712217, 'inserts', inserts, false, false));
                            items = J$.W(712241, 'items', J$.T(712233, [], 10, false), items, false, false);
                            J$.M(712313, J$.R(712249, 'tree', tree, false, false), 'reach', false)(J$.T(712305, function (data) {
                                jalangiLabel12:
                                    while (true) {
                                        try {
                                            J$.Fe(712281, arguments.callee, this, arguments);
                                            arguments = J$.N(712289, 'arguments', arguments, true, false, false);
                                            data = J$.N(712297, 'data', data, true, false, false);
                                            J$.M(712273, J$.R(712257, 'items', items, false, false), 'push', false)(J$.R(712265, 'data', data, false, false));
                                        } catch (J$e) {
                                            J$.Ex(718177, J$e);
                                        } finally {
                                            if (J$.Fr(718185))
                                                continue jalangiLabel12;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.M(712345, J$.R(712321, 'assert', assert, false, false), 'deepEqual', false)(J$.R(712329, 'items', items, false, false), J$.R(712337, 'inserts', inserts, false, false));
                        } catch (J$e) {
                            J$.Ex(718193, J$e);
                        } finally {
                            if (J$.Fr(718201))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function reverse_it_break(assert, tree_class) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(712913, arguments.callee, this, arguments);
                            arguments = J$.N(712921, 'arguments', arguments, true, false, false);
                            assert = J$.N(712929, 'assert', assert, true, false, false);
                            tree_class = J$.N(712937, 'tree_class', tree_class, true, false, false);
                            J$.N(712945, 'inserts', inserts, false, false, false);
                            J$.N(712953, 'tree', tree, false, false, false);
                            J$.N(712961, 'items', items, false, false, false);
                            J$.N(712969, 'it', it, false, false, false);
                            J$.N(712977, 'data', data, false, false, false);
                            J$.N(712985, 'i', i, false, false, false);
                            var inserts = J$.W(712465, 'inserts', J$.M(712457, J$.R(712425, 'loader', loader, false, true), 'get_inserts', false)(J$.M(712449, J$.R(712433, 'loader', loader, false, true), 'load', false)(J$.R(712441, 'SAMPLE_FILE', SAMPLE_FILE, false, true))), inserts, false, false);
                            var tree = J$.W(712505, 'tree', J$.M(712497, J$.R(712473, 'loader', loader, false, true), 'build_tree', false)(J$.R(712481, 'tree_class', tree_class, false, false), J$.R(712489, 'inserts', inserts, false, false)), tree, false, false);
                            var items = J$.W(712521, 'items', J$.T(712513, [], 10, false), items, false, false);
                            var it = J$.W(712545, 'it', J$.M(712537, J$.R(712529, 'tree', tree, false, false), 'iterator', false)(), it, false, false), data;
                            while (J$.C(47000, J$.B(137050, '!==', data = J$.W(712569, 'data', J$.M(712561, J$.R(712553, 'it', it, false, false), 'prev', false)(), data, false, false), J$.T(712577, null, 25, false)))) {
                                J$.M(712601, J$.R(712585, 'items', items, false, false), 'push', false)(J$.R(712593, 'data', data, false, false));
                            }
                            J$.M(712681, J$.R(712609, 'inserts', inserts, false, false), 'sort', false)(J$.T(712673, function (a, b) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(712641, arguments.callee, this, arguments);
                                            arguments = J$.N(712649, 'arguments', arguments, true, false, false);
                                            a = J$.N(712657, 'a', a, true, false, false);
                                            b = J$.N(712665, 'b', b, true, false, false);
                                            return J$.Rt(712633, J$.B(137058, '-', J$.R(712617, 'b', b, false, false), J$.R(712625, 'a', a, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(718209, J$e);
                                        } finally {
                                            if (J$.Fr(718217))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.M(712713, J$.R(712689, 'assert', assert, false, false), 'deepEqual', false)(J$.R(712697, 'items', items, false, false), J$.R(712705, 'inserts', inserts, false, false));
                            items = J$.W(712729, 'items', J$.T(712721, [], 10, false), items, false, false);
                            var i = J$.W(712745, 'i', J$.T(712737, 0, 22, false), i, false, false);
                            J$.M(712865, J$.R(712753, 'tree', tree, false, false), 'reach', false)(J$.T(712857, function (data) {
                                jalangiLabel15:
                                    while (true) {
                                        try {
                                            J$.Fe(712833, arguments.callee, this, arguments);
                                            arguments = J$.N(712841, 'arguments', arguments, true, false, false);
                                            data = J$.N(712849, 'data', data, true, false, false);
                                            J$.M(712777, J$.R(712761, 'items', items, false, false), 'push', false)(J$.R(712769, 'data', data, false, false));
                                            if (J$.C(47008, J$.B(137066, '===', J$.R(712785, 'i', i, false, false), J$.T(712793, 3, 22, false)))) {
                                                return J$.Rt(712809, J$.T(712801, false, 23, false));
                                            }
                                            J$.B(137090, '-', i = J$.W(712825, 'i', J$.B(137082, '+', J$.U(137074, '+', J$.R(712817, 'i', i, false, false)), 1), i, false, false), 1);
                                        } catch (J$e) {
                                            J$.Ex(718225, J$e);
                                        } finally {
                                            if (J$.Fr(718233))
                                                continue jalangiLabel15;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.M(712905, J$.R(712873, 'assert', assert, false, false), 'equal', false)(J$.G(712889, J$.R(712881, 'items', items, false, false), 'length'), J$.T(712897, 4, 22, false));
                        } catch (J$e) {
                            J$.Ex(718241, J$e);
                        } finally {
                            if (J$.Fr(718249))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function switch_it(assert, tree_class) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(713665, arguments.callee, this, arguments);
                            function do_switch(after) {
                                jalangiLabel18:
                                    while (true) {
                                        try {
                                            J$.Fe(713497, arguments.callee, this, arguments);
                                            arguments = J$.N(713505, 'arguments', arguments, true, false, false);
                                            after = J$.N(713513, 'after', after, true, false, false);
                                            J$.N(713521, 'items', items, false, false, false);
                                            J$.N(713529, 'it', it, false, false, false);
                                            J$.N(713537, 'i', i, false, false, false);
                                            J$.N(713545, 'forward', forward, false, false, false);
                                            J$.N(713553, 'reverse', reverse, false, false, false);
                                            J$.N(713561, 'all', all, false, false, false);
                                            var items = J$.W(713169, 'items', J$.T(713161, [], 10, false), items, false, false);
                                            var it = J$.W(713193, 'it', J$.M(713185, J$.R(713177, 'tree', tree, false, false), 'iterator', false)(), it, false, false);
                                            for (var i = J$.W(713209, 'i', J$.T(713201, 0, 22, false), i, false, false); J$.C(47016, J$.B(137106, '<', J$.R(713217, 'i', i, false, false), J$.R(713225, 'after', after, false, false))); J$.B(137130, '-', i = J$.W(713241, 'i', J$.B(137122, '+', J$.U(137114, '+', J$.R(713233, 'i', i, false, false)), 1), i, false, false), 1)) {
                                                J$.M(713273, J$.R(713249, 'items', items, false, false), 'push', false)(J$.M(713265, J$.R(713257, 'it', it, false, false), 'next', false)());
                                            }
                                            while (J$.C(47024, J$.B(137138, '!==', data = J$.W(713297, 'data', J$.M(713289, J$.R(713281, 'it', it, false, false), 'prev', false)(), J$.I(typeof data === 'undefined' ? undefined : data), true, true), J$.T(713305, null, 25, false)))) {
                                                J$.M(713329, J$.R(713313, 'items', items, false, false), 'push', false)(J$.I(typeof data === 'undefined' ? data = J$.R(713321, 'data', undefined, true, true) : data = J$.R(713321, 'data', data, true, true)));
                                            }
                                            var forward = J$.W(713369, 'forward', J$.M(713361, J$.R(713337, 'inserts', inserts, false, false), 'slice', false)(J$.T(713345, 0, 22, false), J$.R(713353, 'after', after, false, false)), forward, false, false);
                                            var reverse = J$.W(713425, 'reverse', J$.M(713417, J$.M(713409, J$.R(713377, 'inserts', inserts, false, false), 'slice', false)(J$.T(713385, 0, 22, false), J$.B(137146, '-', J$.R(713393, 'after', after, false, false), J$.T(713401, 1, 22, false))), 'reverse', false)(), reverse, false, false);
                                            var all = J$.W(713457, 'all', J$.M(713449, J$.R(713433, 'forward', forward, false, false), 'concat', false)(J$.R(713441, 'reverse', reverse, false, false)), all, false, false);
                                            J$.M(713489, J$.R(713465, 'assert', assert, false, false), 'deepEqual', false)(J$.R(713473, 'items', items, false, false), J$.R(713481, 'all', all, false, false));
                                        } catch (J$e) {
                                            J$.Ex(718273, J$e);
                                        } finally {
                                            if (J$.Fr(718281))
                                                continue jalangiLabel18;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(713673, 'arguments', arguments, true, false, false);
                            assert = J$.N(713681, 'assert', assert, true, false, false);
                            tree_class = J$.N(713689, 'tree_class', tree_class, true, false, false);
                            J$.N(713697, 'inserts', inserts, false, false, false);
                            J$.N(713705, 'tree', tree, false, false, false);
                            do_switch = J$.N(713721, 'do_switch', J$.T(713713, do_switch, 12, false), true, false, false);
                            var inserts = J$.W(713033, 'inserts', J$.M(713025, J$.R(712993, 'loader', loader, false, true), 'get_inserts', false)(J$.M(713017, J$.R(713001, 'loader', loader, false, true), 'load', false)(J$.R(713009, 'SAMPLE_FILE', SAMPLE_FILE, false, true))), inserts, false, false);
                            var tree = J$.W(713073, 'tree', J$.M(713065, J$.R(713041, 'loader', loader, false, true), 'build_tree', false)(J$.R(713049, 'tree_class', tree_class, false, false), J$.R(713057, 'inserts', inserts, false, false)), tree, false, false);
                            J$.M(713153, J$.R(713081, 'inserts', inserts, false, false), 'sort', false)(J$.T(713145, function (a, b) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(713113, arguments.callee, this, arguments);
                                            arguments = J$.N(713121, 'arguments', arguments, true, false, false);
                                            a = J$.N(713129, 'a', a, true, false, false);
                                            b = J$.N(713137, 'b', b, true, false, false);
                                            return J$.Rt(713105, J$.B(137098, '-', J$.R(713089, 'a', a, false, false), J$.R(713097, 'b', b, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(718257, J$e);
                                        } finally {
                                            if (J$.Fr(718265))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(713585, J$.R(713569, 'do_switch', do_switch, false, false), false)(J$.T(713577, 1, 22, false));
                            J$.F(713609, J$.R(713593, 'do_switch', do_switch, false, false), false)(J$.T(713601, 10, 22, false));
                            J$.F(713633, J$.R(713617, 'do_switch', do_switch, false, false), false)(J$.T(713625, 1000, 22, false));
                            J$.F(713657, J$.R(713641, 'do_switch', do_switch, false, false), false)(J$.T(713649, 9000, 22, false));
                        } catch (J$e) {
                            J$.Ex(718289, J$e);
                        } finally {
                            if (J$.Fr(718297))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function empty_it(assert, tree_class) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(713889, arguments.callee, this, arguments);
                            arguments = J$.N(713897, 'arguments', arguments, true, false, false);
                            assert = J$.N(713905, 'assert', assert, true, false, false);
                            tree_class = J$.N(713913, 'tree_class', tree_class, true, false, false);
                            J$.N(713921, 'tree', tree, false, false, false);
                            J$.N(713929, 'it', it, false, false, false);
                            var tree = J$.W(713753, 'tree', J$.M(713745, J$.R(713729, 'loader', loader, false, true), 'new_tree', false)(J$.R(713737, 'tree_class', tree_class, false, false)), tree, false, false);
                            var it = J$.W(713777, 'it', J$.M(713769, J$.R(713761, 'tree', tree, false, false), 'iterator', false)(), it, false, false);
                            J$.M(713817, J$.R(713785, 'assert', assert, false, false), 'equal', false)(J$.M(713801, J$.R(713793, 'it', it, false, false), 'next', false)(), J$.T(713809, null, 25, false));
                            it = J$.W(713841, 'it', J$.M(713833, J$.R(713825, 'tree', tree, false, false), 'iterator', false)(), it, false, false);
                            J$.M(713881, J$.R(713849, 'assert', assert, false, false), 'equal', false)(J$.M(713865, J$.R(713857, 'it', it, false, false), 'prev', false)(), J$.T(713873, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(718305, J$e);
                        } finally {
                            if (J$.Fr(718313))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function lower_bound(assert, tree_class) {
                jalangiLabel22:
                    while (true) {
                        try {
                            J$.Fe(714905, arguments.callee, this, arguments);
                            arguments = J$.N(714913, 'arguments', arguments, true, false, false);
                            assert = J$.N(714921, 'assert', assert, true, false, false);
                            tree_class = J$.N(714929, 'tree_class', tree_class, true, false, false);
                            J$.N(714937, 'inserts', inserts, false, false, false);
                            J$.N(714945, 'tree', tree, false, false, false);
                            J$.N(714953, 'i', i, false, false, false);
                            J$.N(714961, 'item', item, false, false, false);
                            J$.N(714969, 'iter', iter, false, false, false);
                            J$.N(714977, 'prev', prev, false, false, false);
                            J$.N(714985, 'next', next, false, false, false);
                            J$.N(714993, 'last', last, false, false, false);
                            var inserts = J$.W(713977, 'inserts', J$.M(713969, J$.R(713937, 'loader', loader, false, true), 'get_inserts', false)(J$.M(713961, J$.R(713945, 'loader', loader, false, true), 'load', false)(J$.R(713953, 'SAMPLE_FILE', SAMPLE_FILE, false, true))), inserts, false, false);
                            var tree = J$.W(714017, 'tree', J$.M(714009, J$.R(713985, 'loader', loader, false, true), 'build_tree', false)(J$.R(713993, 'tree_class', tree_class, false, false), J$.R(714001, 'inserts', inserts, false, false)), tree, false, false);
                            J$.M(714097, J$.R(714025, 'inserts', inserts, false, false), 'sort', false)(J$.T(714089, function (a, b) {
                                jalangiLabel21:
                                    while (true) {
                                        try {
                                            J$.Fe(714057, arguments.callee, this, arguments);
                                            arguments = J$.N(714065, 'arguments', arguments, true, false, false);
                                            a = J$.N(714073, 'a', a, true, false, false);
                                            b = J$.N(714081, 'b', b, true, false, false);
                                            return J$.Rt(714049, J$.B(137154, '-', J$.R(714033, 'a', a, false, false), J$.R(714041, 'b', b, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(718321, J$e);
                                        } finally {
                                            if (J$.Fr(718329))
                                                continue jalangiLabel21;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            for (var i = J$.W(714113, 'i', J$.T(714105, 1, 22, false), i, false, false); J$.C(47032, J$.B(137170, '<', J$.R(714121, 'i', i, false, false), J$.B(137162, '-', J$.G(714137, J$.R(714129, 'inserts', inserts, false, false), 'length'), J$.T(714145, 1, 22, false)))); i = J$.W(714161, 'i', J$.B(137186, '+', J$.U(137178, '+', J$.R(714153, 'i', i, false, false)), 1), i, false, false)) {
                                var item = J$.W(714193, 'item', J$.G(714185, J$.R(714169, 'inserts', inserts, false, false), J$.R(714177, 'i', i, false, false)), item, false, false);
                                var iter = J$.W(714225, 'iter', J$.M(714217, J$.R(714201, 'tree', tree, false, false), 'lowerBound', false)(J$.R(714209, 'item', item, false, false)), iter, false, false);
                                J$.M(714265, J$.R(714233, 'assert', assert, false, false), 'equal', false)(J$.M(714249, J$.R(714241, 'iter', iter, false, false), 'data', false)(), J$.R(714257, 'item', item, false, false));
                                J$.M(714329, J$.R(714273, 'assert', assert, false, false), 'equal', false)(J$.M(714289, J$.R(714281, 'iter', iter, false, false), 'prev', false)(), J$.G(714321, J$.R(714297, 'inserts', inserts, false, false), J$.B(137194, '-', J$.R(714305, 'i', i, false, false), J$.T(714313, 1, 22, false))));
                                J$.M(714345, J$.R(714337, 'iter', iter, false, false), 'next', false)();
                                J$.M(714409, J$.R(714353, 'assert', assert, false, false), 'equal', false)(J$.M(714369, J$.R(714361, 'iter', iter, false, false), 'next', false)(), J$.G(714401, J$.R(714377, 'inserts', inserts, false, false), J$.B(137202, '+', J$.R(714385, 'i', i, false, false), J$.T(714393, 1, 22, false))));
                                var prev = J$.W(714449, 'prev', J$.M(714441, J$.R(714417, 'tree', tree, false, false), 'lowerBound', false)(J$.B(137210, '-', J$.R(714425, 'item', item, false, false), J$.T(714433, 0.1, 22, false))), prev, false, false);
                                J$.M(714505, J$.R(714457, 'assert', assert, false, false), 'equal', false)(J$.M(714473, J$.R(714465, 'prev', prev, false, false), 'data', false)(), J$.G(714497, J$.R(714481, 'inserts', inserts, false, false), J$.R(714489, 'i', i, false, false)));
                                var next = J$.W(714545, 'next', J$.M(714537, J$.R(714513, 'tree', tree, false, false), 'lowerBound', false)(J$.B(137218, '+', J$.R(714521, 'item', item, false, false), J$.T(714529, 0.1, 22, false))), next, false, false);
                                J$.M(714609, J$.R(714553, 'assert', assert, false, false), 'equal', false)(J$.M(714569, J$.R(714561, 'next', next, false, false), 'data', false)(), J$.G(714601, J$.R(714577, 'inserts', inserts, false, false), J$.B(137226, '+', J$.R(714585, 'i', i, false, false), J$.T(714593, 1, 22, false))));
                            }
                            var iter = J$.W(714641, 'iter', J$.M(714633, J$.R(714617, 'tree', tree, false, false), 'lowerBound', false)(J$.U(137234, '-', J$.T(714625, 1, 22, false))), iter, false, false);
                            J$.M(714697, J$.R(714649, 'assert', assert, false, false), 'equal', false)(J$.M(714665, J$.R(714657, 'iter', iter, false, false), 'data', false)(), J$.G(714689, J$.R(714673, 'inserts', inserts, false, false), J$.T(714681, 0, 22, false)));
                            var last = J$.W(714745, 'last', J$.G(714737, J$.R(714705, 'inserts', inserts, false, false), J$.B(137242, '-', J$.G(714721, J$.R(714713, 'inserts', inserts, false, false), 'length'), J$.T(714729, 1, 22, false))), last, false, false);
                            iter = J$.W(714777, 'iter', J$.M(714769, J$.R(714753, 'tree', tree, false, false), 'lowerBound', false)(J$.R(714761, 'last', last, false, false)), iter, false, false);
                            J$.M(714817, J$.R(714785, 'assert', assert, false, false), 'equal', false)(J$.M(714801, J$.R(714793, 'iter', iter, false, false), 'data', false)(), J$.R(714809, 'last', last, false, false));
                            iter = J$.W(714857, 'iter', J$.M(714849, J$.R(714825, 'tree', tree, false, false), 'lowerBound', false)(J$.B(137250, '+', J$.R(714833, 'last', last, false, false), J$.T(714841, 1, 22, false))), iter, false, false);
                            J$.M(714897, J$.R(714865, 'assert', assert, false, false), 'equal', false)(J$.M(714881, J$.R(714873, 'iter', iter, false, false), 'data', false)(), J$.T(714889, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(718337, J$e);
                        } finally {
                            if (J$.Fr(718345))
                                continue jalangiLabel22;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function upper_bound(assert, tree_class) {
                jalangiLabel25:
                    while (true) {
                        try {
                            J$.Fe(716169, arguments.callee, this, arguments);
                            arguments = J$.N(716177, 'arguments', arguments, true, false, false);
                            assert = J$.N(716185, 'assert', assert, true, false, false);
                            tree_class = J$.N(716193, 'tree_class', tree_class, true, false, false);
                            J$.N(716201, 'inserts', inserts, false, false, false);
                            J$.N(716209, 'tree', tree, false, false, false);
                            J$.N(716217, 'i', i, false, false, false);
                            J$.N(716225, 'item', item, false, false, false);
                            J$.N(716233, 'iter', iter, false, false, false);
                            J$.N(716241, 'prev', prev, false, false, false);
                            J$.N(716249, 'next', next, false, false, false);
                            J$.N(716257, 'last', last, false, false, false);
                            J$.N(716265, 'empty', empty, false, false, false);
                            var inserts = J$.W(715041, 'inserts', J$.M(715033, J$.R(715001, 'loader', loader, false, true), 'get_inserts', false)(J$.M(715025, J$.R(715009, 'loader', loader, false, true), 'load', false)(J$.R(715017, 'SAMPLE_FILE', SAMPLE_FILE, false, true))), inserts, false, false);
                            var tree = J$.W(715081, 'tree', J$.M(715073, J$.R(715049, 'loader', loader, false, true), 'build_tree', false)(J$.R(715057, 'tree_class', tree_class, false, false), J$.R(715065, 'inserts', inserts, false, false)), tree, false, false);
                            J$.M(715161, J$.R(715089, 'inserts', inserts, false, false), 'sort', false)(J$.T(715153, function (a, b) {
                                jalangiLabel23:
                                    while (true) {
                                        try {
                                            J$.Fe(715121, arguments.callee, this, arguments);
                                            arguments = J$.N(715129, 'arguments', arguments, true, false, false);
                                            a = J$.N(715137, 'a', a, true, false, false);
                                            b = J$.N(715145, 'b', b, true, false, false);
                                            return J$.Rt(715113, J$.B(137258, '-', J$.R(715097, 'a', a, false, false), J$.R(715105, 'b', b, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(718353, J$e);
                                        } finally {
                                            if (J$.Fr(718361))
                                                continue jalangiLabel23;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            for (var i = J$.W(715177, 'i', J$.T(715169, 0, 22, false), i, false, false); J$.C(47040, J$.B(137274, '<', J$.R(715185, 'i', i, false, false), J$.B(137266, '-', J$.G(715201, J$.R(715193, 'inserts', inserts, false, false), 'length'), J$.T(715209, 2, 22, false)))); i = J$.W(715225, 'i', J$.B(137290, '+', J$.U(137282, '+', J$.R(715217, 'i', i, false, false)), 1), i, false, false)) {
                                var item = J$.W(715257, 'item', J$.G(715249, J$.R(715233, 'inserts', inserts, false, false), J$.R(715241, 'i', i, false, false)), item, false, false);
                                var iter = J$.W(715289, 'iter', J$.M(715281, J$.R(715265, 'tree', tree, false, false), 'upperBound', false)(J$.R(715273, 'item', item, false, false)), iter, false, false);
                                J$.M(715353, J$.R(715297, 'assert', assert, false, false), 'equal', false)(J$.M(715313, J$.R(715305, 'iter', iter, false, false), 'data', false)(), J$.G(715345, J$.R(715321, 'inserts', inserts, false, false), J$.B(137298, '+', J$.R(715329, 'i', i, false, false), J$.T(715337, 1, 22, false))));
                                J$.M(715409, J$.R(715361, 'assert', assert, false, false), 'equal', false)(J$.M(715377, J$.R(715369, 'iter', iter, false, false), 'prev', false)(), J$.G(715401, J$.R(715385, 'inserts', inserts, false, false), J$.R(715393, 'i', i, false, false)));
                                J$.M(715425, J$.R(715417, 'iter', iter, false, false), 'next', false)();
                                J$.M(715489, J$.R(715433, 'assert', assert, false, false), 'equal', false)(J$.M(715449, J$.R(715441, 'iter', iter, false, false), 'next', false)(), J$.G(715481, J$.R(715457, 'inserts', inserts, false, false), J$.B(137306, '+', J$.R(715465, 'i', i, false, false), J$.T(715473, 2, 22, false))));
                                var prev = J$.W(715529, 'prev', J$.M(715521, J$.R(715497, 'tree', tree, false, false), 'upperBound', false)(J$.B(137314, '-', J$.R(715505, 'item', item, false, false), J$.T(715513, 0.1, 22, false))), prev, false, false);
                                J$.M(715585, J$.R(715537, 'assert', assert, false, false), 'equal', false)(J$.M(715553, J$.R(715545, 'prev', prev, false, false), 'data', false)(), J$.G(715577, J$.R(715561, 'inserts', inserts, false, false), J$.R(715569, 'i', i, false, false)));
                                var next = J$.W(715625, 'next', J$.M(715617, J$.R(715593, 'tree', tree, false, false), 'upperBound', false)(J$.B(137322, '+', J$.R(715601, 'item', item, false, false), J$.T(715609, 0.1, 22, false))), next, false, false);
                                J$.M(715689, J$.R(715633, 'assert', assert, false, false), 'equal', false)(J$.M(715649, J$.R(715641, 'next', next, false, false), 'data', false)(), J$.G(715681, J$.R(715657, 'inserts', inserts, false, false), J$.B(137330, '+', J$.R(715665, 'i', i, false, false), J$.T(715673, 1, 22, false))));
                            }
                            var iter = J$.W(715721, 'iter', J$.M(715713, J$.R(715697, 'tree', tree, false, false), 'upperBound', false)(J$.U(137338, '-', J$.T(715705, 1, 22, false))), iter, false, false);
                            J$.M(715777, J$.R(715729, 'assert', assert, false, false), 'equal', false)(J$.M(715745, J$.R(715737, 'iter', iter, false, false), 'data', false)(), J$.G(715769, J$.R(715753, 'inserts', inserts, false, false), J$.T(715761, 0, 22, false)));
                            var last = J$.W(715825, 'last', J$.G(715817, J$.R(715785, 'inserts', inserts, false, false), J$.B(137346, '-', J$.G(715801, J$.R(715793, 'inserts', inserts, false, false), 'length'), J$.T(715809, 1, 22, false))), last, false, false);
                            iter = J$.W(715857, 'iter', J$.M(715849, J$.R(715833, 'tree', tree, false, false), 'upperBound', false)(J$.R(715841, 'last', last, false, false)), iter, false, false);
                            J$.M(715897, J$.R(715865, 'assert', assert, false, false), 'equal', false)(J$.M(715881, J$.R(715873, 'iter', iter, false, false), 'data', false)(), J$.T(715889, null, 25, false));
                            iter = J$.W(715937, 'iter', J$.M(715929, J$.R(715905, 'tree', tree, false, false), 'upperBound', false)(J$.B(137354, '+', J$.R(715913, 'last', last, false, false), J$.T(715921, 1, 22, false))), iter, false, false);
                            J$.M(715977, J$.R(715945, 'assert', assert, false, false), 'equal', false)(J$.M(715961, J$.R(715953, 'iter', iter, false, false), 'data', false)(), J$.T(715969, null, 25, false));
                            var empty = J$.W(716081, 'empty', J$.F(716073, J$.R(715985, 'tree_class', tree_class, false, false), true)(J$.T(716065, function (a, b) {
                                jalangiLabel24:
                                    while (true) {
                                        try {
                                            J$.Fe(716033, arguments.callee, this, arguments);
                                            arguments = J$.N(716041, 'arguments', arguments, true, false, false);
                                            a = J$.N(716049, 'a', a, true, false, false);
                                            b = J$.N(716057, 'b', b, true, false, false);
                                            return J$.Rt(716025, J$.B(137362, '-', J$.G(716001, J$.R(715993, 'a', a, false, false), 'val'), J$.G(716017, J$.R(716009, 'b', b, false, false), 'val')));
                                        } catch (J$e) {
                                            J$.Ex(718369, J$e);
                                        } finally {
                                            if (J$.Fr(718377))
                                                continue jalangiLabel24;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), empty, false, false);
                            var iter = J$.W(716121, 'iter', J$.M(716113, J$.R(716089, 'empty', empty, false, false), 'upperBound', false)(J$.T(716105, { val: J$.T(716097, 0, 22, false) }, 11, false)), iter, false, false);
                            J$.M(716161, J$.R(716129, 'assert', assert, false, false), 'equal', false)(J$.M(716145, J$.R(716137, 'iter', iter, false, false), 'data', false)(), J$.T(716153, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(718385, J$e);
                        } finally {
                            if (J$.Fr(718393))
                                continue jalangiLabel25;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function find(assert, tree_class) {
                jalangiLabel26:
                    while (true) {
                        try {
                            J$.Fe(716561, arguments.callee, this, arguments);
                            arguments = J$.N(716569, 'arguments', arguments, true, false, false);
                            assert = J$.N(716577, 'assert', assert, true, false, false);
                            tree_class = J$.N(716585, 'tree_class', tree_class, true, false, false);
                            J$.N(716593, 'inserts', inserts, false, false, false);
                            J$.N(716601, 'tree', tree, false, false, false);
                            J$.N(716609, 'i', i, false, false, false);
                            J$.N(716617, 'item', item, false, false, false);
                            var inserts = J$.W(716313, 'inserts', J$.M(716305, J$.R(716273, 'loader', loader, false, true), 'get_inserts', false)(J$.M(716297, J$.R(716281, 'loader', loader, false, true), 'load', false)(J$.R(716289, 'SAMPLE_FILE', SAMPLE_FILE, false, true))), inserts, false, false);
                            var tree = J$.W(716353, 'tree', J$.M(716345, J$.R(716321, 'loader', loader, false, true), 'build_tree', false)(J$.R(716329, 'tree_class', tree_class, false, false), J$.R(716337, 'inserts', inserts, false, false)), tree, false, false);
                            for (var i = J$.W(716369, 'i', J$.T(716361, 1, 22, false), i, false, false); J$.C(47048, J$.B(137378, '<', J$.R(716377, 'i', i, false, false), J$.B(137370, '-', J$.G(716393, J$.R(716385, 'inserts', inserts, false, false), 'length'), J$.T(716401, 1, 22, false)))); i = J$.W(716417, 'i', J$.B(137394, '+', J$.U(137386, '+', J$.R(716409, 'i', i, false, false)), 1), i, false, false)) {
                                var item = J$.W(716449, 'item', J$.G(716441, J$.R(716425, 'inserts', inserts, false, false), J$.R(716433, 'i', i, false, false)), item, false, false);
                                J$.M(716497, J$.R(716457, 'assert', assert, false, false), 'equal', false)(J$.M(716481, J$.R(716465, 'tree', tree, false, false), 'find', false)(J$.R(716473, 'item', item, false, false)), J$.R(716489, 'item', item, false, false));
                                J$.M(716553, J$.R(716505, 'assert', assert, false, false), 'equal', false)(J$.M(716537, J$.R(716513, 'tree', tree, false, false), 'find', false)(J$.B(137402, '+', J$.R(716521, 'item', item, false, false), J$.T(716529, 0.1, 22, false))), J$.T(716545, null, 25, false));
                            }
                        } catch (J$e) {
                            J$.Ex(718401, J$e);
                        } finally {
                            if (J$.Fr(718409))
                                continue jalangiLabel26;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function find_iter(assert, tree_class) {
                jalangiLabel28:
                    while (true) {
                        try {
                            J$.Fe(717161, arguments.callee, this, arguments);
                            arguments = J$.N(717169, 'arguments', arguments, true, false, false);
                            assert = J$.N(717177, 'assert', assert, true, false, false);
                            tree_class = J$.N(717185, 'tree_class', tree_class, true, false, false);
                            J$.N(717193, 'inserts', inserts, false, false, false);
                            J$.N(717201, 'tree', tree, false, false, false);
                            J$.N(717209, 'i', i, false, false, false);
                            J$.N(717217, 'item', item, false, false, false);
                            J$.N(717225, 'iter', iter, false, false, false);
                            var inserts = J$.W(716665, 'inserts', J$.M(716657, J$.R(716625, 'loader', loader, false, true), 'get_inserts', false)(J$.M(716649, J$.R(716633, 'loader', loader, false, true), 'load', false)(J$.R(716641, 'SAMPLE_FILE', SAMPLE_FILE, false, true))), inserts, false, false);
                            var tree = J$.W(716705, 'tree', J$.M(716697, J$.R(716673, 'loader', loader, false, true), 'build_tree', false)(J$.R(716681, 'tree_class', tree_class, false, false), J$.R(716689, 'inserts', inserts, false, false)), tree, false, false);
                            J$.M(716785, J$.R(716713, 'inserts', inserts, false, false), 'sort', false)(J$.T(716777, function (a, b) {
                                jalangiLabel27:
                                    while (true) {
                                        try {
                                            J$.Fe(716745, arguments.callee, this, arguments);
                                            arguments = J$.N(716753, 'arguments', arguments, true, false, false);
                                            a = J$.N(716761, 'a', a, true, false, false);
                                            b = J$.N(716769, 'b', b, true, false, false);
                                            return J$.Rt(716737, J$.B(137410, '-', J$.R(716721, 'a', a, false, false), J$.R(716729, 'b', b, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(718417, J$e);
                                        } finally {
                                            if (J$.Fr(718425))
                                                continue jalangiLabel27;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            for (var i = J$.W(716801, 'i', J$.T(716793, 1, 22, false), i, false, false); J$.C(47056, J$.B(137426, '<', J$.R(716809, 'i', i, false, false), J$.B(137418, '-', J$.G(716825, J$.R(716817, 'inserts', inserts, false, false), 'length'), J$.T(716833, 1, 22, false)))); i = J$.W(716849, 'i', J$.B(137442, '+', J$.U(137434, '+', J$.R(716841, 'i', i, false, false)), 1), i, false, false)) {
                                var item = J$.W(716881, 'item', J$.G(716873, J$.R(716857, 'inserts', inserts, false, false), J$.R(716865, 'i', i, false, false)), item, false, false);
                                var iter = J$.W(716913, 'iter', J$.M(716905, J$.R(716889, 'tree', tree, false, false), 'findIter', false)(J$.R(716897, 'item', item, false, false)), iter, false, false);
                                J$.M(716953, J$.R(716921, 'assert', assert, false, false), 'equal', false)(J$.M(716937, J$.R(716929, 'iter', iter, false, false), 'data', false)(), J$.R(716945, 'item', item, false, false));
                                J$.M(717017, J$.R(716961, 'assert', assert, false, false), 'equal', false)(J$.M(716977, J$.R(716969, 'iter', iter, false, false), 'prev', false)(), J$.G(717009, J$.R(716985, 'inserts', inserts, false, false), J$.B(137450, '-', J$.R(716993, 'i', i, false, false), J$.T(717001, 1, 22, false))));
                                J$.M(717033, J$.R(717025, 'iter', iter, false, false), 'next', false)();
                                J$.M(717097, J$.R(717041, 'assert', assert, false, false), 'equal', false)(J$.M(717057, J$.R(717049, 'iter', iter, false, false), 'next', false)(), J$.G(717089, J$.R(717065, 'inserts', inserts, false, false), J$.B(137458, '+', J$.R(717073, 'i', i, false, false), J$.T(717081, 1, 22, false))));
                                J$.M(717153, J$.R(717105, 'assert', assert, false, false), 'equal', false)(J$.M(717137, J$.R(717113, 'tree', tree, false, false), 'findIter', false)(J$.B(137466, '+', J$.R(717121, 'item', item, false, false), J$.T(717129, 0.1, 22, false))), J$.T(717145, null, 25, false));
                            }
                        } catch (J$e) {
                            J$.Ex(718433, J$e);
                        } finally {
                            if (J$.Fr(718441))
                                continue jalangiLabel28;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(717713, '_', _, false, false, false);
            J$.N(717721, 'loader', loader, false, false, false);
            J$.N(717729, 'SAMPLE_FILE', SAMPLE_FILE, false, false, false);
            J$.N(717737, 'TREES', TREES, false, false, false);
            clear = J$.N(717753, 'clear', J$.T(717745, clear, 12, false), true, false, false);
            dup = J$.N(717769, 'dup', J$.T(717761, dup, 12, false), true, false, false);
            nonexist = J$.N(717785, 'nonexist', J$.T(717777, nonexist, 12, false), true, false, false);
            minmax = J$.N(717801, 'minmax', J$.T(717793, minmax, 12, false), true, false, false);
            forward_it = J$.N(717817, 'forward_it', J$.T(717809, forward_it, 12, false), true, false, false);
            forward_it_break = J$.N(717833, 'forward_it_break', J$.T(717825, forward_it_break, 12, false), true, false, false);
            reverse_it = J$.N(717849, 'reverse_it', J$.T(717841, reverse_it, 12, false), true, false, false);
            reverse_it_break = J$.N(717865, 'reverse_it_break', J$.T(717857, reverse_it_break, 12, false), true, false, false);
            switch_it = J$.N(717881, 'switch_it', J$.T(717873, switch_it, 12, false), true, false, false);
            empty_it = J$.N(717897, 'empty_it', J$.T(717889, empty_it, 12, false), true, false, false);
            lower_bound = J$.N(717913, 'lower_bound', J$.T(717905, lower_bound, 12, false), true, false, false);
            upper_bound = J$.N(717929, 'upper_bound', J$.T(717921, upper_bound, 12, false), true, false, false);
            find = J$.N(717945, 'find', J$.T(717937, find, 12, false), true, false, false);
            find_iter = J$.N(717961, 'find_iter', J$.T(717953, find_iter, 12, false), true, false, false);
            J$.N(717969, 'TESTS', TESTS, false, false, false);
            J$.N(717977, 'test_funcs', test_funcs, false, false, false);
            J$.T(709385, 'use strict', 21, false);
            J$.F(709409, J$.I(typeof require === 'undefined' ? require = J$.R(709393, 'require', undefined, true, true) : require = J$.R(709393, 'require', require, true, true)), false)(J$.T(709401, 'core-js/modules/es.array.concat', 21, false));
            J$.F(709433, J$.I(typeof require === 'undefined' ? require = J$.R(709417, 'require', undefined, true, true) : require = J$.R(709417, 'require', require, true, true)), false)(J$.T(709425, 'core-js/modules/es.array.find', 21, false));
            J$.F(709457, J$.I(typeof require === 'undefined' ? require = J$.R(709441, 'require', undefined, true, true) : require = J$.R(709441, 'require', require, true, true)), false)(J$.T(709449, 'core-js/modules/es.array.for-each', 21, false));
            J$.F(709481, J$.I(typeof require === 'undefined' ? require = J$.R(709465, 'require', undefined, true, true) : require = J$.R(709465, 'require', require, true, true)), false)(J$.T(709473, 'core-js/modules/es.array.reverse', 21, false));
            J$.F(709505, J$.I(typeof require === 'undefined' ? require = J$.R(709489, 'require', undefined, true, true) : require = J$.R(709489, 'require', require, true, true)), false)(J$.T(709497, 'core-js/modules/es.array.slice', 21, false));
            J$.F(709529, J$.I(typeof require === 'undefined' ? require = J$.R(709513, 'require', undefined, true, true) : require = J$.R(709513, 'require', require, true, true)), false)(J$.T(709521, 'core-js/modules/es.array.sort', 21, false));
            J$.F(709553, J$.I(typeof require === 'undefined' ? require = J$.R(709537, 'require', undefined, true, true) : require = J$.R(709537, 'require', require, true, true)), false)(J$.T(709545, 'core-js/modules/web.dom-collections.for-each', 21, false));
            var _ = J$.W(709585, '_', J$.F(709577, J$.I(typeof require === 'undefined' ? require = J$.R(709561, 'require', undefined, true, true) : require = J$.R(709561, 'require', require, true, true)), false)(J$.T(709569, 'underscore', 21, false)), _, false, true);
            var loader = J$.W(709617, 'loader', J$.F(709609, J$.I(typeof require === 'undefined' ? require = J$.R(709593, 'require', undefined, true, true) : require = J$.R(709593, 'require', require, true, true)), false)(J$.T(709601, './loader', 21, false)), loader, false, true);
            var SAMPLE_FILE = J$.W(709641, 'SAMPLE_FILE', J$.B(136914, '+', J$.I(typeof __dirname === 'undefined' ? __dirname = J$.R(709625, '__dirname', undefined, true, true) : __dirname = J$.R(709625, '__dirname', __dirname, true, true)), J$.T(709633, '/samples/10k', 21, false)), SAMPLE_FILE, false, true);
            var TREES = J$.W(709673, 'TREES', J$.T(709665, [
                J$.T(709649, 'rbtree', 21, false),
                J$.T(709657, 'bintree', 21, false)
            ], 10, false), TREES, false, true);
            var TESTS = J$.W(717353, 'TESTS', J$.T(717345, {
                clear: J$.R(717233, 'clear', clear, false, true),
                dup: J$.R(717241, 'dup', dup, false, true),
                nonexist: J$.R(717249, 'nonexist', nonexist, false, true),
                minmax: J$.R(717257, 'minmax', minmax, false, true),
                forward_it: J$.R(717265, 'forward_it', forward_it, false, true),
                forward_it_break: J$.R(717273, 'forward_it_break', forward_it_break, false, true),
                reverse_it: J$.R(717281, 'reverse_it', reverse_it, false, true),
                reverse_it_break: J$.R(717289, 'reverse_it_break', reverse_it_break, false, true),
                switch_it: J$.R(717297, 'switch_it', switch_it, false, true),
                empty_it: J$.R(717305, 'empty_it', empty_it, false, true),
                lower_bound: J$.R(717313, 'lower_bound', lower_bound, false, true),
                upper_bound: J$.R(717321, 'upper_bound', upper_bound, false, true),
                find: J$.R(717329, 'find', find, false, true),
                find_iter: J$.R(717337, 'find_iter', find_iter, false, true)
            }, 11, false), TESTS, false, true);
            var test_funcs = J$.W(717369, 'test_funcs', J$.T(717361, {}, 11, false), test_funcs, false, true);
            J$.M(717673, J$.R(717377, 'TREES', TREES, false, true), 'forEach', false)(J$.T(717665, function (tree) {
                jalangiLabel31:
                    while (true) {
                        try {
                            J$.Fe(717625, arguments.callee, this, arguments);
                            arguments = J$.N(717633, 'arguments', arguments, true, false, false);
                            tree = J$.N(717641, 'tree', tree, true, false, false);
                            J$.N(717649, 'tree_class', tree_class, false, false, false);
                            J$.N(717657, 'test', test, false, false, false);
                            var tree_class = J$.W(717417, 'tree_class', J$.F(717409, J$.I(typeof require === 'undefined' ? require = J$.R(717385, 'require', undefined, true, true) : require = J$.R(717385, 'require', require, true, true)), false)(J$.B(137474, '+', J$.T(717393, '../lib/', 21, false), J$.R(717401, 'tree', tree, false, false))), tree_class, false, false);
                            for (var test in J$.H(717609, J$.R(717425, 'TESTS', TESTS, false, true))) {
                                J$.N(717617, 'test', test, false, true, false);
                                {
                                    {
                                        J$.F(717601, J$.T(717585, function (test) {
                                            jalangiLabel30:
                                                while (true) {
                                                    try {
                                                        J$.Fe(717561, arguments.callee, this, arguments);
                                                        arguments = J$.N(717569, 'arguments', arguments, true, false, false);
                                                        test = J$.N(717577, 'test', test, true, false, false);
                                                        J$.P(717553, J$.R(717433, 'test_funcs', test_funcs, false, true), J$.B(137490, '+', J$.B(137482, '+', J$.R(717441, 'tree', tree, false, false), J$.T(717449, '_', 21, false)), J$.R(717457, 'test', test, false, false)), J$.T(717545, function (assert) {
                                                            jalangiLabel29:
                                                                while (true) {
                                                                    try {
                                                                        J$.Fe(717521, arguments.callee, this, arguments);
                                                                        arguments = J$.N(717529, 'arguments', arguments, true, false, false);
                                                                        assert = J$.N(717537, 'assert', assert, true, false, false);
                                                                        J$.M(717497, J$.R(717465, 'TESTS', TESTS, false, true), J$.R(717473, 'test', test, false, false), false)(J$.R(717481, 'assert', assert, false, false), J$.R(717489, 'tree_class', tree_class, false, false));
                                                                        J$.M(717513, J$.R(717505, 'assert', assert, false, false), 'done', false)();
                                                                    } catch (J$e) {
                                                                        J$.Ex(718449, J$e);
                                                                    } finally {
                                                                        if (J$.Fr(718457))
                                                                            continue jalangiLabel29;
                                                                        else
                                                                            return J$.Ra();
                                                                    }
                                                                }
                                                        }, 12, false));
                                                    } catch (J$e) {
                                                        J$.Ex(718465, J$e);
                                                    } finally {
                                                        if (J$.Fr(718473))
                                                            continue jalangiLabel30;
                                                        else
                                                            return J$.Ra();
                                                    }
                                                }
                                        }, 12, false), false)(J$.R(717593, 'test', test, false, false));
                                    }
                                }
                            }
                        } catch (J$e) {
                            J$.Ex(718481, J$e);
                        } finally {
                            if (J$.Fr(718489))
                                continue jalangiLabel31;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(717697, J$.I(typeof exports === 'undefined' ? exports = J$.R(717681, 'exports', undefined, true, true) : exports = J$.R(717681, 'exports', exports, true, true)), 'api', J$.R(717689, 'test_funcs', test_funcs, false, true));
        } catch (J$e) {
            J$.Ex(718497, J$e);
        } finally {
            if (J$.Sr(718505))
                continue jalangiLabel32;
            else
                break jalangiLabel32;
        }
    }
// JALANGI DO NOT INSTRUMENT
