// mongodb/lib/core/auth/auth_provider.js
J$.noInstrEval = false;
jalangiLabel17:
    while (true) {
        try {
            J$.Se(99633, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/core/auth/auth_provider.js');
            function _classCallCheck(instance, Constructor) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(97153, arguments.callee, this, arguments);
                            arguments = J$.N(97161, 'arguments', arguments, true, false, false);
                            instance = J$.N(97169, 'instance', instance, true, false, false);
                            Constructor = J$.N(97177, 'Constructor', Constructor, true, false, false);
                            if (J$.C(5512, J$.U(5154, '!', J$.B(5146, 'instanceof', J$.R(97113, 'instance', instance, false, false), J$.R(97121, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(97145, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(97129, 'TypeError', undefined, true, true) : TypeError = J$.R(97129, 'TypeError', TypeError, true, true)), true)(J$.T(97137, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(99705, J$e);
                        } finally {
                            if (J$.Fr(99713))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(97425, arguments.callee, this, arguments);
                            arguments = J$.N(97433, 'arguments', arguments, true, false, false);
                            target = J$.N(97441, 'target', target, true, false, false);
                            props = J$.N(97449, 'props', props, true, false, false);
                            J$.N(97457, 'i', i, false, false, false);
                            J$.N(97465, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(97193, 'i', J$.T(97185, 0, 22, false), i, false, false); J$.C(5536, J$.B(5162, '<', J$.R(97201, 'i', i, false, false), J$.G(97217, J$.R(97209, 'props', props, false, false), 'length'))); J$.B(5186, '-', i = J$.W(97233, 'i', J$.B(5178, '+', J$.U(5170, '+', J$.R(97225, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(97265, 'descriptor', J$.G(97257, J$.R(97241, 'props', props, false, false), J$.R(97249, 'i', i, false, false)), descriptor, false, false);
                                J$.P(97305, J$.R(97273, 'descriptor', descriptor, false, false), 'enumerable', J$.C(5520, J$.G(97289, J$.R(97281, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(97297, false, 23, false));
                                J$.P(97329, J$.R(97313, 'descriptor', descriptor, false, false), 'configurable', J$.T(97321, true, 23, false));
                                if (J$.C(5528, J$.B(5194, 'in', J$.T(97337, 'value', 21, false), J$.R(97345, 'descriptor', descriptor, false, false))))
                                    J$.P(97369, J$.R(97353, 'descriptor', descriptor, false, false), 'writable', J$.T(97361, true, 23, false));
                                J$.M(97417, J$.I(typeof Object === 'undefined' ? Object = J$.R(97377, 'Object', undefined, true, true) : Object = J$.R(97377, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(97385, 'target', target, false, false), J$.G(97401, J$.R(97393, 'descriptor', descriptor, false, false), 'key'), J$.R(97409, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(99721, J$e);
                        } finally {
                            if (J$.Fr(99729))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(97577, arguments.callee, this, arguments);
                            arguments = J$.N(97585, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(97593, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(97601, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(97609, 'staticProps', staticProps, true, false, false);
                            if (J$.C(5544, J$.R(97473, 'protoProps', protoProps, false, false)))
                                J$.F(97513, J$.R(97481, '_defineProperties', _defineProperties, false, true), false)(J$.G(97497, J$.R(97489, 'Constructor', Constructor, false, false), 'prototype'), J$.R(97505, 'protoProps', protoProps, false, false));
                            if (J$.C(5552, J$.R(97521, 'staticProps', staticProps, false, false)))
                                J$.F(97553, J$.R(97529, '_defineProperties', _defineProperties, false, true), false)(J$.R(97537, 'Constructor', Constructor, false, false), J$.R(97545, 'staticProps', staticProps, false, false));
                            return J$.Rt(97569, J$.R(97561, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(99737, J$e);
                        } finally {
                            if (J$.Fr(99745))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _classCallCheck = J$.N(99649, '_classCallCheck', J$.T(99641, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(99665, '_defineProperties', J$.T(99657, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(99681, '_createClass', J$.T(99673, _createClass, 12, false), true, false, false);
            J$.N(99689, 'MongoError', MongoError, false, false, false);
            J$.N(99697, 'AuthProvider', AuthProvider, false, false, false);
            J$.T(97105, 'use strict', 21, false);
            var MongoError = J$.W(97649, 'MongoError', J$.G(97641, J$.F(97633, J$.I(typeof require === 'undefined' ? require = J$.R(97617, 'require', undefined, true, true) : require = J$.R(97617, 'require', require, true, true)), false)(J$.T(97625, '../error', 21, false)), 'MongoError'), MongoError, false, true);
            var AuthProvider = J$.W(99593, 'AuthProvider', J$.F(99585, J$.T(99577, function () {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(99545, arguments.callee, this, arguments);
                            function AuthProvider(bson) {
                                jalangiLabel3:
                                    while (true) {
                                        try {
                                            J$.Fe(97737, arguments.callee, this, arguments);
                                            arguments = J$.N(97745, 'arguments', arguments, true, false, false);
                                            bson = J$.N(97753, 'bson', bson, true, false, false);
                                            J$.F(97681, J$.R(97657, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(97665, 'this', this, false, false), J$.R(97673, 'AuthProvider', AuthProvider, false, false));
                                            J$.P(97705, J$.R(97689, 'this', this, false, false), 'bson', J$.R(97697, 'bson', bson, false, false));
                                            J$.P(97729, J$.R(97713, 'this', this, false, false), 'authStore', J$.T(97721, [], 10, false));
                                        } catch (J$e) {
                                            J$.Ex(99753, J$e);
                                        } finally {
                                            if (J$.Fr(99761))
                                                continue jalangiLabel3;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(99553, 'arguments', arguments, true, false, false);
                            AuthProvider = J$.N(99569, 'AuthProvider', J$.T(99561, AuthProvider, 12, false), true, false, false);
                            J$.F(99521, J$.R(97761, '_createClass', _createClass, false, true), false)(J$.R(97769, 'AuthProvider', AuthProvider, false, false), J$.T(99513, [
                                J$.T(98673, {
                                    key: J$.T(97777, 'auth', 21, false),
                                    value: J$.T(98665, function auth(sendAuthCommand, connections, credentials, callback) {
                                        jalangiLabel8:
                                            while (true) {
                                                try {
                                                    J$.Fe(98569, arguments.callee, this, arguments);
                                                    arguments = J$.N(98577, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(98585, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connections = J$.N(98593, 'connections', connections, true, false, false);
                                                    credentials = J$.N(98601, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(98609, 'callback', callback, true, false, false);
                                                    J$.N(98617, '_this', _this, false, false, false);
                                                    J$.N(98625, 'count', count, false, false, false);
                                                    J$.N(98633, 'numberOfValidConnections', numberOfValidConnections, false, false, false);
                                                    J$.N(98641, 'errorObject', errorObject, false, false, false);
                                                    J$.N(98649, 'execute', execute, false, false, false);
                                                    J$.N(98657, 'executeInNextTick', executeInNextTick, false, false, false);
                                                    var _this = J$.W(97793, '_this', J$.R(97785, 'this', this, false, false), _this, false, false);
                                                    var count = J$.W(97817, 'count', J$.G(97809, J$.R(97801, 'connections', connections, false, false), 'length'), count, false, false);
                                                    if (J$.C(5560, J$.B(5202, '===', J$.R(97825, 'count', count, false, false), J$.T(97833, 0, 22, false)))) {
                                                        J$.F(97865, J$.R(97841, 'callback', callback, false, false), false)(J$.T(97849, null, 25, false), J$.T(97857, null, 25, false));
                                                        return J$.Rt(97873, undefined);
                                                    }
                                                    var numberOfValidConnections = J$.W(97889, 'numberOfValidConnections', J$.T(97881, 0, 22, false), numberOfValidConnections, false, false);
                                                    var errorObject = J$.W(97905, 'errorObject', J$.T(97897, null, 25, false), errorObject, false, false);
                                                    var execute = J$.W(98385, 'execute', J$.T(98377, function execute(connection) {
                                                        jalangiLabel5:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(98353, arguments.callee, this, arguments);
                                                                    arguments = J$.N(98361, 'arguments', arguments, true, false, false);
                                                                    connection = J$.N(98369, 'connection', connection, true, false, false);
                                                                    J$.M(98345, J$.R(97913, '_this', _this, false, false), '_authenticateSingleConnection', false)(J$.R(97921, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(97929, 'connection', connection, false, false), J$.R(97937, 'credentials', credentials, false, false), J$.T(98337, function (err, r) {
                                                                        jalangiLabel4:
                                                                            while (true) {
                                                                                try {
                                                                                    J$.Fe(98305, arguments.callee, this, arguments);
                                                                                    arguments = J$.N(98313, 'arguments', arguments, true, false, false);
                                                                                    err = J$.N(98321, 'err', err, true, false, false);
                                                                                    r = J$.N(98329, 'r', r, true, false, false);
                                                                                    count = J$.W(97961, 'count', J$.B(5210, '-', J$.R(97945, 'count', count, false, false), J$.T(97953, 1, 22, false)), count, false, false);
                                                                                    if (J$.C(5592, J$.R(97969, 'err', err, false, false))) {
                                                                                        errorObject = J$.W(98001, 'errorObject', J$.F(97993, J$.R(97977, 'MongoError', MongoError, false, true), true)(J$.R(97985, 'err', err, false, false)), errorObject, false, false);
                                                                                    } else if (J$.C(5584, J$.C(5576, J$.R(98009, 'r', r, false, false)) ? J$.C(5568, J$.G(98025, J$.R(98017, 'r', r, false, false), '$err')) ? J$._() : J$.G(98041, J$.R(98033, 'r', r, false, false), 'errmsg') : J$._())) {
                                                                                        errorObject = J$.W(98073, 'errorObject', J$.F(98065, J$.R(98049, 'MongoError', MongoError, false, true), true)(J$.R(98057, 'r', r, false, false)), errorObject, false, false);
                                                                                    } else {
                                                                                        numberOfValidConnections = J$.W(98097, 'numberOfValidConnections', J$.B(5218, '+', J$.R(98081, 'numberOfValidConnections', numberOfValidConnections, false, false), J$.T(98089, 1, 22, false)), numberOfValidConnections, false, false);
                                                                                    }
                                                                                    if (J$.C(5600, J$.B(5226, '!==', J$.R(98105, 'count', count, false, false), J$.T(98113, 0, 22, false)))) {
                                                                                        return J$.Rt(98121, undefined);
                                                                                    }
                                                                                    if (J$.C(5616, J$.B(5234, '>', J$.R(98129, 'numberOfValidConnections', numberOfValidConnections, false, false), J$.T(98137, 0, 22, false)))) {
                                                                                        J$.M(98161, J$.R(98145, '_this', _this, false, false), 'addCredentials', false)(J$.R(98153, 'credentials', credentials, false, false));
                                                                                        J$.F(98193, J$.R(98169, 'callback', callback, false, false), false)(J$.T(98177, null, 25, false), J$.T(98185, true, 23, false));
                                                                                    } else {
                                                                                        if (J$.C(5608, J$.B(5242, '==', J$.R(98201, 'errorObject', errorObject, false, false), J$.T(98209, null, 25, false)))) {
                                                                                            errorObject = J$.W(98265, 'errorObject', J$.F(98257, J$.R(98217, 'MongoError', MongoError, false, true), true)(J$.M(98249, J$.T(98225, 'failed to authenticate using ', 21, false), 'concat', false)(J$.G(98241, J$.R(98233, 'credentials', credentials, false, false), 'mechanism'))), errorObject, false, false);
                                                                                        }
                                                                                        J$.F(98297, J$.R(98273, 'callback', callback, false, false), false)(J$.R(98281, 'errorObject', errorObject, false, false), J$.T(98289, false, 23, false));
                                                                                    }
                                                                                } catch (J$e) {
                                                                                    J$.Ex(99769, J$e);
                                                                                } finally {
                                                                                    if (J$.Fr(99777))
                                                                                        continue jalangiLabel4;
                                                                                    else
                                                                                        return J$.Ra();
                                                                                }
                                                                            }
                                                                    }, 12, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(99785, J$e);
                                                                } finally {
                                                                    if (J$.Fr(99793))
                                                                        continue jalangiLabel5;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false), execute, false, false);
                                                    var executeInNextTick = J$.W(98505, 'executeInNextTick', J$.T(98497, function executeInNextTick(_connection) {
                                                        jalangiLabel7:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(98473, arguments.callee, this, arguments);
                                                                    arguments = J$.N(98481, 'arguments', arguments, true, false, false);
                                                                    _connection = J$.N(98489, '_connection', _connection, true, false, false);
                                                                    return J$.Rt(98465, J$.M(98457, J$.I(typeof process === 'undefined' ? process = J$.R(98393, 'process', undefined, true, true) : process = J$.R(98393, 'process', process, true, true)), 'nextTick', false)(J$.T(98449, function () {
                                                                        jalangiLabel6:
                                                                            while (true) {
                                                                                try {
                                                                                    J$.Fe(98433, arguments.callee, this, arguments);
                                                                                    arguments = J$.N(98441, 'arguments', arguments, true, false, false);
                                                                                    return J$.Rt(98425, J$.F(98417, J$.R(98401, 'execute', execute, false, false), false)(J$.R(98409, '_connection', _connection, false, false)));
                                                                                } catch (J$e) {
                                                                                    J$.Ex(99801, J$e);
                                                                                } finally {
                                                                                    if (J$.Fr(99809))
                                                                                        continue jalangiLabel6;
                                                                                    else
                                                                                        return J$.Ra();
                                                                                }
                                                                            }
                                                                    }, 12, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(99817, J$e);
                                                                } finally {
                                                                    if (J$.Fr(99825))
                                                                        continue jalangiLabel7;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false), executeInNextTick, false, false);
                                                    while (J$.C(5624, J$.B(5250, '>', J$.G(98521, J$.R(98513, 'connections', connections, false, false), 'length'), J$.T(98529, 0, 22, false)))) {
                                                        J$.F(98561, J$.R(98537, 'executeInNextTick', executeInNextTick, false, false), false)(J$.M(98553, J$.R(98545, 'connections', connections, false, false), 'shift', false)());
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(99833, J$e);
                                                } finally {
                                                    if (J$.Fr(99841))
                                                        continue jalangiLabel8;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(98737, {
                                    key: J$.T(98681, '_authenticateSingleConnection', 21, false),
                                    value: J$.T(98729, function _authenticateSingleConnection() {
                                        jalangiLabel9:
                                            while (true) {
                                                try {
                                                    J$.Fe(98713, arguments.callee, this, arguments);
                                                    arguments = J$.N(98721, 'arguments', arguments, true, false, false);
                                                    throw J$.F(98705, J$.I(typeof Error === 'undefined' ? Error = J$.R(98689, 'Error', undefined, true, true) : Error = J$.R(98689, 'Error', Error, true, true)), true)(J$.T(98697, '_authenticateSingleConnection must be overridden', 21, false));
                                                } catch (J$e) {
                                                    J$.Ex(99849, J$e);
                                                } finally {
                                                    if (J$.Fr(99857))
                                                        continue jalangiLabel9;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(98929, {
                                    key: J$.T(98745, 'addCredentials', 21, false),
                                    value: J$.T(98921, function addCredentials(credentials) {
                                        jalangiLabel11:
                                            while (true) {
                                                try {
                                                    J$.Fe(98889, arguments.callee, this, arguments);
                                                    arguments = J$.N(98897, 'arguments', arguments, true, false, false);
                                                    credentials = J$.N(98905, 'credentials', credentials, true, false, false);
                                                    J$.N(98913, 'found', found, false, false, false);
                                                    var found = J$.W(98841, 'found', J$.M(98833, J$.G(98761, J$.R(98753, 'this', this, false, false), 'authStore'), 'some', false)(J$.T(98825, function (cred) {
                                                        jalangiLabel10:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(98801, arguments.callee, this, arguments);
                                                                    arguments = J$.N(98809, 'arguments', arguments, true, false, false);
                                                                    cred = J$.N(98817, 'cred', cred, true, false, false);
                                                                    return J$.Rt(98793, J$.M(98785, J$.R(98769, 'cred', cred, false, false), 'equals', false)(J$.R(98777, 'credentials', credentials, false, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(99865, J$e);
                                                                } finally {
                                                                    if (J$.Fr(99873))
                                                                        continue jalangiLabel10;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false)), found, false, false);
                                                    if (J$.C(5632, J$.U(5258, '!', J$.R(98849, 'found', found, false, false)))) {
                                                        J$.M(98881, J$.G(98865, J$.R(98857, 'this', this, false, false), 'authStore'), 'push', false)(J$.R(98873, 'credentials', credentials, false, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(99881, J$e);
                                                } finally {
                                                    if (J$.Fr(99889))
                                                        continue jalangiLabel11;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(99353, {
                                    key: J$.T(98937, 'reauthenticate', 21, false),
                                    value: J$.T(99345, function reauthenticate(sendAuthCommand, connections, callback) {
                                        jalangiLabel13:
                                            while (true) {
                                                try {
                                                    J$.Fe(99281, arguments.callee, this, arguments);
                                                    arguments = J$.N(99289, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(99297, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connections = J$.N(99305, 'connections', connections, true, false, false);
                                                    callback = J$.N(99313, 'callback', callback, true, false, false);
                                                    J$.N(99321, 'authStore', authStore, false, false, false);
                                                    J$.N(99329, 'count', count, false, false, false);
                                                    J$.N(99337, 'i', i, false, false, false);
                                                    var authStore = J$.W(98977, 'authStore', J$.M(98969, J$.G(98953, J$.R(98945, 'this', this, false, false), 'authStore'), 'slice', false)(J$.T(98961, 0, 22, false)), authStore, false, false);
                                                    var count = J$.W(99001, 'count', J$.G(98993, J$.R(98985, 'authStore', authStore, false, false), 'length'), count, false, false);
                                                    if (J$.C(5640, J$.B(5266, '===', J$.R(99009, 'count', count, false, false), J$.T(99017, 0, 22, false)))) {
                                                        return J$.Rt(99057, J$.F(99049, J$.R(99025, 'callback', callback, false, false), false)(J$.T(99033, null, 25, false), J$.T(99041, null, 25, false)));
                                                    }
                                                    for (var i = J$.W(99073, 'i', J$.T(99065, 0, 22, false), i, false, false); J$.C(5656, J$.B(5274, '<', J$.R(99081, 'i', i, false, false), J$.G(99097, J$.R(99089, 'authStore', authStore, false, false), 'length'))); J$.B(5298, '-', i = J$.W(99113, 'i', J$.B(5290, '+', J$.U(5282, '+', J$.R(99105, 'i', i, false, false)), 1), i, false, false), 1)) {
                                                        J$.M(99273, J$.R(99121, 'this', this, false, false), 'auth', false)(J$.R(99129, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(99137, 'connections', connections, false, false), J$.G(99161, J$.R(99145, 'authStore', authStore, false, false), J$.R(99153, 'i', i, false, false)), J$.T(99265, function (err) {
                                                            jalangiLabel12:
                                                                while (true) {
                                                                    try {
                                                                        J$.Fe(99241, arguments.callee, this, arguments);
                                                                        arguments = J$.N(99249, 'arguments', arguments, true, false, false);
                                                                        err = J$.N(99257, 'err', err, true, false, false);
                                                                        count = J$.W(99185, 'count', J$.B(5306, '-', J$.R(99169, 'count', count, false, false), J$.T(99177, 1, 22, false)), count, false, false);
                                                                        if (J$.C(5648, J$.B(5314, '===', J$.R(99193, 'count', count, false, false), J$.T(99201, 0, 22, false)))) {
                                                                            J$.F(99233, J$.R(99209, 'callback', callback, false, false), false)(J$.R(99217, 'err', err, false, false), J$.T(99225, null, 25, false));
                                                                        }
                                                                    } catch (J$e) {
                                                                        J$.Ex(99897, J$e);
                                                                    } finally {
                                                                        if (J$.Fr(99905))
                                                                            continue jalangiLabel12;
                                                                        else
                                                                            return J$.Ra();
                                                                    }
                                                                }
                                                        }, 12, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(99913, J$e);
                                                } finally {
                                                    if (J$.Fr(99921))
                                                        continue jalangiLabel13;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(99505, {
                                    key: J$.T(99361, 'logout', 21, false),
                                    value: J$.T(99497, function logout(source) {
                                        jalangiLabel15:
                                            while (true) {
                                                try {
                                                    J$.Fe(99473, arguments.callee, this, arguments);
                                                    arguments = J$.N(99481, 'arguments', arguments, true, false, false);
                                                    source = J$.N(99489, 'source', source, true, false, false);
                                                    J$.P(99465, J$.R(99369, 'this', this, false, false), 'authStore', J$.M(99457, J$.G(99385, J$.R(99377, 'this', this, false, false), 'authStore'), 'filter', false)(J$.T(99449, function (credentials) {
                                                        jalangiLabel14:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(99425, arguments.callee, this, arguments);
                                                                    arguments = J$.N(99433, 'arguments', arguments, true, false, false);
                                                                    credentials = J$.N(99441, 'credentials', credentials, true, false, false);
                                                                    return J$.Rt(99417, J$.B(5322, '!==', J$.G(99401, J$.R(99393, 'credentials', credentials, false, false), 'source'), J$.R(99409, 'source', source, false, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(99929, J$e);
                                                                } finally {
                                                                    if (J$.Fr(99937))
                                                                        continue jalangiLabel14;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false)));
                                                } catch (J$e) {
                                                    J$.Ex(99945, J$e);
                                                } finally {
                                                    if (J$.Fr(99953))
                                                        continue jalangiLabel15;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(99537, J$.R(99529, 'AuthProvider', AuthProvider, false, false));
                        } catch (J$e) {
                            J$.Ex(99961, J$e);
                        } finally {
                            if (J$.Fr(99969))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), AuthProvider, false, true);
            J$.P(99625, J$.I(typeof module === 'undefined' ? module = J$.R(99601, 'module', undefined, true, true) : module = J$.R(99601, 'module', module, true, true)), 'exports', J$.T(99617, { AuthProvider: J$.R(99609, 'AuthProvider', AuthProvider, false, true) }, 11, false));
        } catch (J$e) {
            J$.Ex(99977, J$e);
        } finally {
            if (J$.Sr(99985))
                continue jalangiLabel17;
            else
                break jalangiLabel17;
        }
    }
// JALANGI DO NOT INSTRUMENT
