// mongodb/lib/core/auth/gssapi.js
J$.noInstrEval = false;
jalangiLabel32:
    while (true) {
        try {
            J$.Se(105785, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/core/auth/gssapi.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(100865, arguments.callee, this, arguments);
                            arguments = J$.N(100873, 'arguments', arguments, true, false, false);
                            obj = J$.N(100881, 'obj', obj, true, false, false);
                            if (J$.C(5704, J$.C(5664, J$.B(5338, '===', J$.U(5330, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(100601, 'Symbol', undefined, true, true) : Symbol = J$.R(100601, 'Symbol', Symbol, true, true))), J$.T(100609, 'function', 21, false))) ? J$.B(5354, '===', J$.U(5346, 'typeof', J$.G(100625, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(100617, 'Symbol', undefined, true, true) : Symbol = J$.R(100617, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(100633, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(100689, '_typeof', J$.T(100681, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(100657, arguments.callee, this, arguments);
                                                arguments = J$.N(100665, 'arguments', arguments, true, false, false);
                                                obj = J$.N(100673, 'obj', obj, true, false, false);
                                                return J$.Rt(100649, J$.U(5362, 'typeof', J$.R(100641, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(106033, J$e);
                                            } finally {
                                                if (J$.Fr(106041))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(100825, '_typeof', J$.T(100817, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(100793, arguments.callee, this, arguments);
                                                arguments = J$.N(100801, 'arguments', arguments, true, false, false);
                                                obj = J$.N(100809, 'obj', obj, true, false, false);
                                                return J$.Rt(100785, J$.C(5696, J$.C(5688, J$.C(5680, J$.C(5672, J$.R(100697, 'obj', obj, false, false)) ? J$.B(5378, '===', J$.U(5370, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(100705, 'Symbol', undefined, true, true) : Symbol = J$.R(100705, 'Symbol', Symbol, true, true))), J$.T(100713, 'function', 21, false)) : J$._()) ? J$.B(5386, '===', J$.G(100729, J$.R(100721, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(100737, 'Symbol', undefined, true, true) : Symbol = J$.R(100737, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(5394, '!==', J$.R(100745, 'obj', obj, false, false), J$.G(100761, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(100753, 'Symbol', undefined, true, true) : Symbol = J$.R(100753, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(100769, 'symbol', 21, false) : J$.U(5402, 'typeof', J$.R(100777, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(106049, J$e);
                                            } finally {
                                                if (J$.Fr(106057))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(100857, J$.F(100849, J$.R(100833, '_typeof', _typeof, false, true), false)(J$.R(100841, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(106065, J$e);
                        } finally {
                            if (J$.Fr(106073))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(100929, arguments.callee, this, arguments);
                            arguments = J$.N(100937, 'arguments', arguments, true, false, false);
                            instance = J$.N(100945, 'instance', instance, true, false, false);
                            Constructor = J$.N(100953, 'Constructor', Constructor, true, false, false);
                            if (J$.C(5712, J$.U(5418, '!', J$.B(5410, 'instanceof', J$.R(100889, 'instance', instance, false, false), J$.R(100897, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(100921, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(100905, 'TypeError', undefined, true, true) : TypeError = J$.R(100905, 'TypeError', TypeError, true, true)), true)(J$.T(100913, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(106081, J$e);
                        } finally {
                            if (J$.Fr(106089))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(101201, arguments.callee, this, arguments);
                            arguments = J$.N(101209, 'arguments', arguments, true, false, false);
                            target = J$.N(101217, 'target', target, true, false, false);
                            props = J$.N(101225, 'props', props, true, false, false);
                            J$.N(101233, 'i', i, false, false, false);
                            J$.N(101241, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(100969, 'i', J$.T(100961, 0, 22, false), i, false, false); J$.C(5736, J$.B(5426, '<', J$.R(100977, 'i', i, false, false), J$.G(100993, J$.R(100985, 'props', props, false, false), 'length'))); J$.B(5450, '-', i = J$.W(101009, 'i', J$.B(5442, '+', J$.U(5434, '+', J$.R(101001, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(101041, 'descriptor', J$.G(101033, J$.R(101017, 'props', props, false, false), J$.R(101025, 'i', i, false, false)), descriptor, false, false);
                                J$.P(101081, J$.R(101049, 'descriptor', descriptor, false, false), 'enumerable', J$.C(5720, J$.G(101065, J$.R(101057, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(101073, false, 23, false));
                                J$.P(101105, J$.R(101089, 'descriptor', descriptor, false, false), 'configurable', J$.T(101097, true, 23, false));
                                if (J$.C(5728, J$.B(5458, 'in', J$.T(101113, 'value', 21, false), J$.R(101121, 'descriptor', descriptor, false, false))))
                                    J$.P(101145, J$.R(101129, 'descriptor', descriptor, false, false), 'writable', J$.T(101137, true, 23, false));
                                J$.M(101193, J$.I(typeof Object === 'undefined' ? Object = J$.R(101153, 'Object', undefined, true, true) : Object = J$.R(101153, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(101161, 'target', target, false, false), J$.G(101177, J$.R(101169, 'descriptor', descriptor, false, false), 'key'), J$.R(101185, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(106097, J$e);
                        } finally {
                            if (J$.Fr(106105))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(101353, arguments.callee, this, arguments);
                            arguments = J$.N(101361, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(101369, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(101377, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(101385, 'staticProps', staticProps, true, false, false);
                            if (J$.C(5744, J$.R(101249, 'protoProps', protoProps, false, false)))
                                J$.F(101289, J$.R(101257, '_defineProperties', _defineProperties, false, true), false)(J$.G(101273, J$.R(101265, 'Constructor', Constructor, false, false), 'prototype'), J$.R(101281, 'protoProps', protoProps, false, false));
                            if (J$.C(5752, J$.R(101297, 'staticProps', staticProps, false, false)))
                                J$.F(101329, J$.R(101305, '_defineProperties', _defineProperties, false, true), false)(J$.R(101313, 'Constructor', Constructor, false, false), J$.R(101321, 'staticProps', staticProps, false, false));
                            return J$.Rt(101345, J$.R(101337, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(106113, J$e);
                        } finally {
                            if (J$.Fr(106121))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(101497, arguments.callee, this, arguments);
                            arguments = J$.N(101505, 'arguments', arguments, true, false, false);
                            self = J$.N(101513, 'self', self, true, false, false);
                            call = J$.N(101521, 'call', call, true, false, false);
                            if (J$.C(5776, J$.C(5768, J$.R(101393, 'call', call, false, false)) ? J$.C(5760, J$.B(5466, '===', J$.F(101417, J$.R(101401, '_typeof', _typeof, false, true), false)(J$.R(101409, 'call', call, false, false)), J$.T(101425, 'object', 21, false))) ? J$._() : J$.B(5482, '===', J$.U(5474, 'typeof', J$.R(101433, 'call', call, false, false)), J$.T(101441, 'function', 21, false)) : J$._())) {
                                return J$.Rt(101457, J$.R(101449, 'call', call, false, false));
                            }
                            return J$.Rt(101489, J$.F(101481, J$.R(101465, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(101473, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(106129, J$e);
                        } finally {
                            if (J$.Fr(106137))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(101585, arguments.callee, this, arguments);
                            arguments = J$.N(101593, 'arguments', arguments, true, false, false);
                            self = J$.N(101601, 'self', self, true, false, false);
                            if (J$.C(5784, J$.B(5490, '===', J$.R(101529, 'self', self, false, false), void J$.T(101537, 0, 22, false)))) {
                                throw J$.F(101561, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(101545, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(101545, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(101553, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(101577, J$.R(101569, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(106145, J$e);
                        } finally {
                            if (J$.Fr(106153))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(101969, arguments.callee, this, arguments);
                            arguments = J$.N(101977, 'arguments', arguments, true, false, false);
                            target = J$.N(101985, 'target', target, true, false, false);
                            property = J$.N(101993, 'property', property, true, false, false);
                            receiver = J$.N(102001, 'receiver', receiver, true, false, false);
                            if (J$.C(5816, J$.C(5792, J$.B(5506, '!==', J$.U(5498, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(101609, 'Reflect', undefined, true, true) : Reflect = J$.R(101609, 'Reflect', Reflect, true, true))), J$.T(101617, 'undefined', 21, false))) ? J$.G(101633, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(101625, 'Reflect', undefined, true, true) : Reflect = J$.R(101625, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(101657, '_get', J$.G(101649, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(101641, 'Reflect', undefined, true, true) : Reflect = J$.R(101641, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(101905, '_get', J$.T(101897, function _get(target, property, receiver) {
                                    jalangiLabel8:
                                        while (true) {
                                            try {
                                                J$.Fe(101841, arguments.callee, this, arguments);
                                                arguments = J$.N(101849, 'arguments', arguments, true, false, false);
                                                target = J$.N(101857, 'target', target, true, false, false);
                                                property = J$.N(101865, 'property', property, true, false, false);
                                                receiver = J$.N(101873, 'receiver', receiver, true, false, false);
                                                J$.N(101881, 'base', base, false, false, false);
                                                J$.N(101889, 'desc', desc, false, false, false);
                                                var base = J$.W(101697, 'base', J$.F(101689, J$.R(101665, '_superPropBase', _superPropBase, false, true), false)(J$.R(101673, 'target', target, false, false), J$.R(101681, 'property', property, false, false)), base, false, false);
                                                if (J$.C(5800, J$.U(5514, '!', J$.R(101705, 'base', base, false, false))))
                                                    return J$.Rt(101713, undefined);
                                                var desc = J$.W(101753, 'desc', J$.M(101745, J$.I(typeof Object === 'undefined' ? Object = J$.R(101721, 'Object', undefined, true, true) : Object = J$.R(101721, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(101729, 'base', base, false, false), J$.R(101737, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(5808, J$.G(101769, J$.R(101761, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(101809, J$.M(101801, J$.G(101785, J$.R(101777, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(101793, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(101833, J$.G(101825, J$.R(101817, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(106161, J$e);
                                            } finally {
                                                if (J$.Fr(106169))
                                                    continue jalangiLabel8;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(101961, J$.F(101953, J$.R(101913, '_get', _get, false, true), false)(J$.R(101921, 'target', target, false, false), J$.R(101929, 'property', property, false, false), J$.C(5824, J$.R(101937, 'receiver', receiver, false, false)) ? J$._() : J$.R(101945, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(106177, J$e);
                        } finally {
                            if (J$.Fr(106185))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(102121, arguments.callee, this, arguments);
                            arguments = J$.N(102129, 'arguments', arguments, true, false, false);
                            object = J$.N(102137, 'object', object, true, false, false);
                            property = J$.N(102145, 'property', property, true, false, false);
                            while (J$.C(5840, J$.U(5522, '!', J$.M(102049, J$.G(102025, J$.G(102017, J$.I(typeof Object === 'undefined' ? Object = J$.R(102009, 'Object', undefined, true, true) : Object = J$.R(102009, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(102033, 'object', object, false, false), J$.R(102041, 'property', property, false, false))))) {
                                object = J$.W(102081, 'object', J$.F(102073, J$.R(102057, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(102065, 'object', object, false, false)), object, false, false);
                                if (J$.C(5832, J$.B(5530, '===', J$.R(102089, 'object', object, false, false), J$.T(102097, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(102113, J$.R(102105, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(106193, J$e);
                        } finally {
                            if (J$.Fr(106201))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(102305, arguments.callee, this, arguments);
                            arguments = J$.N(102313, 'arguments', arguments, true, false, false);
                            o = J$.N(102321, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(102265, '_getPrototypeOf', J$.C(5856, J$.G(102161, J$.I(typeof Object === 'undefined' ? Object = J$.R(102153, 'Object', undefined, true, true) : Object = J$.R(102153, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(102177, J$.I(typeof Object === 'undefined' ? Object = J$.R(102169, 'Object', undefined, true, true) : Object = J$.R(102169, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(102257, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(102233, arguments.callee, this, arguments);
                                            arguments = J$.N(102241, 'arguments', arguments, true, false, false);
                                            o = J$.N(102249, 'o', o, true, false, false);
                                            return J$.Rt(102225, J$.C(5848, J$.G(102193, J$.R(102185, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(102217, J$.I(typeof Object === 'undefined' ? Object = J$.R(102201, 'Object', undefined, true, true) : Object = J$.R(102201, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(102209, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(106209, J$e);
                                        } finally {
                                            if (J$.Fr(106217))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(102297, J$.F(102289, J$.R(102273, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(102281, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(106225, J$e);
                        } finally {
                            if (J$.Fr(106233))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(102521, arguments.callee, this, arguments);
                            arguments = J$.N(102529, 'arguments', arguments, true, false, false);
                            subClass = J$.N(102537, 'subClass', subClass, true, false, false);
                            superClass = J$.N(102545, 'superClass', superClass, true, false, false);
                            if (J$.C(5872, J$.C(5864, J$.B(5546, '!==', J$.U(5538, 'typeof', J$.R(102329, 'superClass', superClass, false, false)), J$.T(102337, 'function', 21, false))) ? J$.B(5554, '!==', J$.R(102345, 'superClass', superClass, false, false), J$.T(102353, null, 25, false)) : J$._())) {
                                throw J$.F(102377, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(102361, 'TypeError', undefined, true, true) : TypeError = J$.R(102361, 'TypeError', TypeError, true, true)), true)(J$.T(102369, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(102473, J$.R(102385, 'subClass', subClass, false, false), 'prototype', J$.M(102465, J$.I(typeof Object === 'undefined' ? Object = J$.R(102393, 'Object', undefined, true, true) : Object = J$.R(102393, 'Object', Object, true, true)), 'create', false)(J$.C(5880, J$.R(102401, 'superClass', superClass, false, false)) ? J$.G(102417, J$.R(102409, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(102457, {
                                constructor: J$.T(102449, {
                                    value: J$.R(102425, 'subClass', subClass, false, false),
                                    writable: J$.T(102433, true, 23, false),
                                    configurable: J$.T(102441, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(5888, J$.R(102481, 'superClass', superClass, false, false)))
                                J$.F(102513, J$.R(102489, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(102497, 'subClass', subClass, false, false), J$.R(102505, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(106241, J$e);
                        } finally {
                            if (J$.Fr(106249))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(102697, arguments.callee, this, arguments);
                            arguments = J$.N(102705, 'arguments', arguments, true, false, false);
                            o = J$.N(102713, 'o', o, true, false, false);
                            p = J$.N(102721, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(102649, '_setPrototypeOf', J$.C(5896, J$.G(102561, J$.I(typeof Object === 'undefined' ? Object = J$.R(102553, 'Object', undefined, true, true) : Object = J$.R(102553, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(102641, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(102609, arguments.callee, this, arguments);
                                            arguments = J$.N(102617, 'arguments', arguments, true, false, false);
                                            o = J$.N(102625, 'o', o, true, false, false);
                                            p = J$.N(102633, 'p', p, true, false, false);
                                            J$.P(102585, J$.R(102569, 'o', o, false, false), '__proto__', J$.R(102577, 'p', p, false, false));
                                            return J$.Rt(102601, J$.R(102593, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(106257, J$e);
                                        } finally {
                                            if (J$.Fr(106265))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(102689, J$.F(102681, J$.R(102657, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(102665, 'o', o, false, false), J$.R(102673, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(106273, J$e);
                        } finally {
                            if (J$.Fr(106281))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(105801, '_typeof', J$.T(105793, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(105817, '_classCallCheck', J$.T(105809, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(105833, '_defineProperties', J$.T(105825, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(105849, '_createClass', J$.T(105841, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(105865, '_possibleConstructorReturn', J$.T(105857, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(105881, '_assertThisInitialized', J$.T(105873, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(105897, '_get', J$.T(105889, _get, 12, false), true, false, false);
            _superPropBase = J$.N(105913, '_superPropBase', J$.T(105905, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(105929, '_getPrototypeOf', J$.T(105921, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(105945, '_inherits', J$.T(105937, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(105961, '_setPrototypeOf', J$.T(105953, _setPrototypeOf, 12, false), true, false, false);
            J$.N(105969, 'AuthProvider', AuthProvider, false, false, false);
            J$.N(105977, 'retrieveKerberos', retrieveKerberos, false, false, false);
            J$.N(105985, 'kerberos', kerberos, false, false, false);
            J$.N(105993, 'GSSAPI', GSSAPI, false, false, false);
            J$.N(106001, 'GSSAPIInitialize', GSSAPIInitialize, false, false, false);
            J$.N(106009, 'MongoDBGSSAPIFirstStep', MongoDBGSSAPIFirstStep, false, false, false);
            J$.N(106017, 'MongoDBGSSAPISecondStep', MongoDBGSSAPISecondStep, false, false, false);
            J$.N(106025, 'MongoDBGSSAPIThirdStep', MongoDBGSSAPIThirdStep, false, false, false);
            J$.T(100593, 'use strict', 21, false);
            var AuthProvider = J$.W(102761, 'AuthProvider', J$.G(102753, J$.F(102745, J$.I(typeof require === 'undefined' ? require = J$.R(102729, 'require', undefined, true, true) : require = J$.R(102729, 'require', require, true, true)), false)(J$.T(102737, './auth_provider', 21, false)), 'AuthProvider'), AuthProvider, false, true);
            var retrieveKerberos = J$.W(102801, 'retrieveKerberos', J$.G(102793, J$.F(102785, J$.I(typeof require === 'undefined' ? require = J$.R(102769, 'require', undefined, true, true) : require = J$.R(102769, 'require', require, true, true)), false)(J$.T(102777, '../utils', 21, false)), 'retrieveKerberos'), retrieveKerberos, false, true);
            var kerberos;
            var GSSAPI = J$.W(103745, 'GSSAPI', J$.F(103737, J$.T(103721, function (_AuthProvider) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(103681, arguments.callee, this, arguments);
                            function GSSAPI() {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(102953, arguments.callee, this, arguments);
                                            arguments = J$.N(102961, 'arguments', arguments, true, false, false);
                                            J$.F(102865, J$.R(102841, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(102849, 'this', this, false, false), J$.R(102857, 'GSSAPI', GSSAPI, false, false));
                                            return J$.Rt(102945, J$.F(102937, J$.R(102873, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(102881, 'this', this, false, false), J$.M(102929, J$.F(102905, J$.R(102889, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(102897, 'GSSAPI', GSSAPI, false, false)), 'apply', false)(J$.R(102913, 'this', this, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(102921, 'arguments', undefined, true, true) : arguments = J$.R(102921, 'arguments', arguments, true, true)))));
                                        } catch (J$e) {
                                            J$.Ex(106289, J$e);
                                        } finally {
                                            if (J$.Fr(106297))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(103689, 'arguments', arguments, true, false, false);
                            _AuthProvider = J$.N(103697, '_AuthProvider', _AuthProvider, true, false, false);
                            GSSAPI = J$.N(103713, 'GSSAPI', J$.T(103705, GSSAPI, 12, false), true, false, false);
                            J$.F(102833, J$.R(102809, '_inherits', _inherits, false, true), false)(J$.R(102817, 'GSSAPI', GSSAPI, false, false), J$.R(102825, '_AuthProvider', _AuthProvider, false, false));
                            J$.F(103657, J$.R(102969, '_createClass', _createClass, false, true), false)(J$.R(102977, 'GSSAPI', GSSAPI, false, false), J$.T(103649, [
                                J$.T(103369, {
                                    key: J$.T(102985, '_authenticateSingleConnection', 21, false),
                                    value: J$.T(103361, function _authenticateSingleConnection(sendAuthCommand, connection, credentials, callback) {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(103273, arguments.callee, this, arguments);
                                                    arguments = J$.N(103281, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(103289, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connection = J$.N(103297, 'connection', connection, true, false, false);
                                                    credentials = J$.N(103305, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(103313, 'callback', callback, true, false, false);
                                                    J$.N(103321, 'source', source, false, false, false);
                                                    J$.N(103329, 'username', username, false, false, false);
                                                    J$.N(103337, 'password', password, false, false, false);
                                                    J$.N(103345, 'mechanismProperties', mechanismProperties, false, false, false);
                                                    J$.N(103353, 'gssapiServiceName', gssapiServiceName, false, false, false);
                                                    var source = J$.W(103009, 'source', J$.G(103001, J$.R(102993, 'credentials', credentials, false, false), 'source'), source, false, false);
                                                    var username = J$.W(103033, 'username', J$.G(103025, J$.R(103017, 'credentials', credentials, false, false), 'username'), username, false, false);
                                                    var password = J$.W(103057, 'password', J$.G(103049, J$.R(103041, 'credentials', credentials, false, false), 'password'), password, false, false);
                                                    var mechanismProperties = J$.W(103081, 'mechanismProperties', J$.G(103073, J$.R(103065, 'credentials', credentials, false, false), 'mechanismProperties'), mechanismProperties, false, false);
                                                    var gssapiServiceName = J$.W(103145, 'gssapiServiceName', J$.C(5912, J$.C(5904, J$.G(103105, J$.R(103089, 'mechanismProperties', mechanismProperties, false, false), J$.T(103097, 'gssapiservicename', 21, false))) ? J$._() : J$.G(103129, J$.R(103113, 'mechanismProperties', mechanismProperties, false, false), J$.T(103121, 'gssapiServiceName', 21, false))) ? J$._() : J$.T(103137, 'mongodb', 21, false), gssapiServiceName, false, false);
                                                    J$.F(103265, J$.R(103153, 'GSSAPIInitialize', GSSAPIInitialize, false, true), false)(J$.R(103161, 'this', this, false, false), J$.G(103185, J$.G(103177, J$.R(103169, 'kerberos', kerberos, false, true), 'processes'), 'MongoAuthProcess'), J$.R(103193, 'source', source, false, false), J$.R(103201, 'username', username, false, false), J$.R(103209, 'password', password, false, false), J$.R(103217, 'source', source, false, false), J$.R(103225, 'gssapiServiceName', gssapiServiceName, false, false), J$.R(103233, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(103241, 'connection', connection, false, false), J$.R(103249, 'mechanismProperties', mechanismProperties, false, false), J$.R(103257, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(106305, J$e);
                                                } finally {
                                                    if (J$.Fr(106313))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(103641, {
                                    key: J$.T(103377, 'auth', 21, false),
                                    value: J$.T(103633, function auth(sendAuthCommand, connections, credentials, callback) {
                                        jalangiLabel18:
                                            while (true) {
                                                try {
                                                    J$.Fe(103585, arguments.callee, this, arguments);
                                                    arguments = J$.N(103593, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(103601, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connections = J$.N(103609, 'connections', connections, true, false, false);
                                                    credentials = J$.N(103617, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(103625, 'callback', callback, true, false, false);
                                                    if (J$.C(5920, J$.B(5562, '==', J$.R(103385, 'kerberos', kerberos, false, true), J$.T(103393, null, 25, false)))) {
                                                        try {
                                                            kerberos = J$.W(103417, 'kerberos', J$.F(103409, J$.R(103401, 'retrieveKerberos', retrieveKerberos, false, true), false)(), kerberos, false, true);
                                                        } catch (e) {
                                                            J$.N(103465, 'e', e, false, false, true);
                                                            return J$.Rt(103457, J$.F(103449, J$.R(103425, 'callback', callback, false, false), false)(J$.R(103433, 'e', e, false, false), J$.T(103441, null, 25, false)));
                                                        }
                                                    }
                                                    J$.M(103577, J$.F(103529, J$.R(103473, '_get', _get, false, true), false)(J$.F(103505, J$.R(103481, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(103497, J$.R(103489, 'GSSAPI', GSSAPI, false, false), 'prototype')), J$.T(103513, 'auth', 21, false), J$.R(103521, 'this', this, false, false)), 'call', false)(J$.R(103537, 'this', this, false, false), J$.R(103545, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(103553, 'connections', connections, false, false), J$.R(103561, 'credentials', credentials, false, false), J$.R(103569, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(106321, J$e);
                                                } finally {
                                                    if (J$.Fr(106329))
                                                        continue jalangiLabel18;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(103673, J$.R(103665, 'GSSAPI', GSSAPI, false, false));
                        } catch (J$e) {
                            J$.Ex(106337, J$e);
                        } finally {
                            if (J$.Fr(106345))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(103729, 'AuthProvider', AuthProvider, false, true)), GSSAPI, false, true);
            var GSSAPIInitialize = J$.W(104265, 'GSSAPIInitialize', J$.T(104257, function GSSAPIInitialize(self, MongoAuthProcess, db, username, password, authdb, gssapiServiceName, sendAuthCommand, connection, options, callback) {
                jalangiLabel22:
                    while (true) {
                        try {
                            J$.Fe(104145, arguments.callee, this, arguments);
                            arguments = J$.N(104153, 'arguments', arguments, true, false, false);
                            self = J$.N(104161, 'self', self, true, false, false);
                            MongoAuthProcess = J$.N(104169, 'MongoAuthProcess', MongoAuthProcess, true, false, false);
                            db = J$.N(104177, 'db', db, true, false, false);
                            username = J$.N(104185, 'username', username, true, false, false);
                            password = J$.N(104193, 'password', password, true, false, false);
                            authdb = J$.N(104201, 'authdb', authdb, true, false, false);
                            gssapiServiceName = J$.N(104209, 'gssapiServiceName', gssapiServiceName, true, false, false);
                            sendAuthCommand = J$.N(104217, 'sendAuthCommand', sendAuthCommand, true, false, false);
                            connection = J$.N(104225, 'connection', connection, true, false, false);
                            options = J$.N(104233, 'options', options, true, false, false);
                            callback = J$.N(104241, 'callback', callback, true, false, false);
                            J$.N(104249, 'mongo_auth_process', mongo_auth_process, false, false, false);
                            var mongo_auth_process = J$.W(103817, 'mongo_auth_process', J$.F(103809, J$.R(103753, 'MongoAuthProcess', MongoAuthProcess, false, false), true)(J$.G(103769, J$.R(103761, 'connection', connection, false, false), 'host'), J$.G(103785, J$.R(103777, 'connection', connection, false, false), 'port'), J$.R(103793, 'gssapiServiceName', gssapiServiceName, false, false), J$.R(103801, 'options', options, false, false)), mongo_auth_process, false, false);
                            J$.M(104137, J$.R(103825, 'mongo_auth_process', mongo_auth_process, false, false), 'init', false)(J$.R(103833, 'username', username, false, false), J$.R(103841, 'password', password, false, false), J$.T(104129, function (err) {
                                jalangiLabel21:
                                    while (true) {
                                        try {
                                            J$.Fe(104105, arguments.callee, this, arguments);
                                            arguments = J$.N(104113, 'arguments', arguments, true, false, false);
                                            err = J$.N(104121, 'err', err, true, false, false);
                                            if (J$.C(5928, J$.R(103849, 'err', err, false, false)))
                                                return J$.Rt(103889, J$.F(103881, J$.R(103857, 'callback', callback, false, false), false)(J$.R(103865, 'err', err, false, false), J$.T(103873, false, 23, false)));
                                            J$.M(104097, J$.R(103897, 'mongo_auth_process', mongo_auth_process, false, false), 'transition', false)(J$.T(103905, '', 21, false), J$.T(104089, function (err, payload) {
                                                jalangiLabel20:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(104057, arguments.callee, this, arguments);
                                                            arguments = J$.N(104065, 'arguments', arguments, true, false, false);
                                                            err = J$.N(104073, 'err', err, true, false, false);
                                                            payload = J$.N(104081, 'payload', payload, true, false, false);
                                                            if (J$.C(5936, J$.R(103913, 'err', err, false, false)))
                                                                return J$.Rt(103953, J$.F(103945, J$.R(103921, 'callback', callback, false, false), false)(J$.R(103929, 'err', err, false, false), J$.T(103937, false, 23, false)));
                                                            J$.F(104049, J$.R(103961, 'MongoDBGSSAPIFirstStep', MongoDBGSSAPIFirstStep, false, true), false)(J$.R(103969, 'self', self, false, false), J$.R(103977, 'mongo_auth_process', mongo_auth_process, false, false), J$.R(103985, 'payload', payload, false, false), J$.R(103993, 'db', db, false, false), J$.R(104001, 'username', username, false, false), J$.R(104009, 'password', password, false, false), J$.R(104017, 'authdb', authdb, false, false), J$.R(104025, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(104033, 'connection', connection, false, false), J$.R(104041, 'callback', callback, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(106353, J$e);
                                                        } finally {
                                                            if (J$.Fr(106361))
                                                                continue jalangiLabel20;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(106369, J$e);
                                        } finally {
                                            if (J$.Fr(106377))
                                                continue jalangiLabel21;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(106385, J$e);
                        } finally {
                            if (J$.Fr(106393))
                                continue jalangiLabel22;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), GSSAPIInitialize, false, true);
            var MongoDBGSSAPIFirstStep = J$.W(104785, 'MongoDBGSSAPIFirstStep', J$.T(104777, function MongoDBGSSAPIFirstStep(self, mongo_auth_process, payload, db, username, password, authdb, sendAuthCommand, connection, callback) {
                jalangiLabel25:
                    while (true) {
                        try {
                            J$.Fe(104673, arguments.callee, this, arguments);
                            arguments = J$.N(104681, 'arguments', arguments, true, false, false);
                            self = J$.N(104689, 'self', self, true, false, false);
                            mongo_auth_process = J$.N(104697, 'mongo_auth_process', mongo_auth_process, true, false, false);
                            payload = J$.N(104705, 'payload', payload, true, false, false);
                            db = J$.N(104713, 'db', db, true, false, false);
                            username = J$.N(104721, 'username', username, true, false, false);
                            password = J$.N(104729, 'password', password, true, false, false);
                            authdb = J$.N(104737, 'authdb', authdb, true, false, false);
                            sendAuthCommand = J$.N(104745, 'sendAuthCommand', sendAuthCommand, true, false, false);
                            connection = J$.N(104753, 'connection', connection, true, false, false);
                            callback = J$.N(104761, 'callback', callback, true, false, false);
                            J$.N(104769, 'command', command, false, false, false);
                            var command = J$.W(104313, 'command', J$.T(104305, {
                                saslStart: J$.T(104273, 1, 22, false),
                                mechanism: J$.T(104281, 'GSSAPI', 21, false),
                                payload: J$.R(104289, 'payload', payload, false, false),
                                autoAuthorize: J$.T(104297, 1, 22, false)
                            }, 11, false), command, false, false);
                            J$.F(104665, J$.R(104321, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(104329, 'connection', connection, false, false), J$.T(104337, '$external.$cmd', 21, false), J$.R(104345, 'command', command, false, false), J$.T(104657, function (err, doc) {
                                jalangiLabel24:
                                    while (true) {
                                        try {
                                            J$.Fe(104625, arguments.callee, this, arguments);
                                            arguments = J$.N(104633, 'arguments', arguments, true, false, false);
                                            err = J$.N(104641, 'err', err, true, false, false);
                                            doc = J$.N(104649, 'doc', doc, true, false, false);
                                            if (J$.C(5944, J$.R(104353, 'err', err, false, false)))
                                                return J$.Rt(104393, J$.F(104385, J$.R(104361, 'callback', callback, false, false), false)(J$.R(104369, 'err', err, false, false), J$.T(104377, false, 23, false)));
                                            J$.M(104617, J$.R(104401, 'mongo_auth_process', mongo_auth_process, false, false), 'transition', false)(J$.G(104417, J$.R(104409, 'doc', doc, false, false), 'payload'), J$.T(104609, function (err, payload) {
                                                jalangiLabel23:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(104577, arguments.callee, this, arguments);
                                                            arguments = J$.N(104585, 'arguments', arguments, true, false, false);
                                                            err = J$.N(104593, 'err', err, true, false, false);
                                                            payload = J$.N(104601, 'payload', payload, true, false, false);
                                                            if (J$.C(5952, J$.R(104425, 'err', err, false, false)))
                                                                return J$.Rt(104465, J$.F(104457, J$.R(104433, 'callback', callback, false, false), false)(J$.R(104441, 'err', err, false, false), J$.T(104449, false, 23, false)));
                                                            J$.F(104569, J$.R(104473, 'MongoDBGSSAPISecondStep', MongoDBGSSAPISecondStep, false, true), false)(J$.R(104481, 'self', self, false, false), J$.R(104489, 'mongo_auth_process', mongo_auth_process, false, false), J$.R(104497, 'payload', payload, false, false), J$.R(104505, 'doc', doc, false, false), J$.R(104513, 'db', db, false, false), J$.R(104521, 'username', username, false, false), J$.R(104529, 'password', password, false, false), J$.R(104537, 'authdb', authdb, false, false), J$.R(104545, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(104553, 'connection', connection, false, false), J$.R(104561, 'callback', callback, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(106401, J$e);
                                                        } finally {
                                                            if (J$.Fr(106409))
                                                                continue jalangiLabel23;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(106417, J$e);
                                        } finally {
                                            if (J$.Fr(106425))
                                                continue jalangiLabel24;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(106433, J$e);
                        } finally {
                            if (J$.Fr(106441))
                                continue jalangiLabel25;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), MongoDBGSSAPIFirstStep, false, true);
            var MongoDBGSSAPISecondStep = J$.W(105313, 'MongoDBGSSAPISecondStep', J$.T(105305, function MongoDBGSSAPISecondStep(self, mongo_auth_process, payload, doc, db, username, password, authdb, sendAuthCommand, connection, callback) {
                jalangiLabel28:
                    while (true) {
                        try {
                            J$.Fe(105193, arguments.callee, this, arguments);
                            arguments = J$.N(105201, 'arguments', arguments, true, false, false);
                            self = J$.N(105209, 'self', self, true, false, false);
                            mongo_auth_process = J$.N(105217, 'mongo_auth_process', mongo_auth_process, true, false, false);
                            payload = J$.N(105225, 'payload', payload, true, false, false);
                            doc = J$.N(105233, 'doc', doc, true, false, false);
                            db = J$.N(105241, 'db', db, true, false, false);
                            username = J$.N(105249, 'username', username, true, false, false);
                            password = J$.N(105257, 'password', password, true, false, false);
                            authdb = J$.N(105265, 'authdb', authdb, true, false, false);
                            sendAuthCommand = J$.N(105273, 'sendAuthCommand', sendAuthCommand, true, false, false);
                            connection = J$.N(105281, 'connection', connection, true, false, false);
                            callback = J$.N(105289, 'callback', callback, true, false, false);
                            J$.N(105297, 'command', command, false, false, false);
                            var command = J$.W(104833, 'command', J$.T(104825, {
                                saslContinue: J$.T(104793, 1, 22, false),
                                conversationId: J$.G(104809, J$.R(104801, 'doc', doc, false, false), 'conversationId'),
                                payload: J$.R(104817, 'payload', payload, false, false)
                            }, 11, false), command, false, false);
                            J$.F(105185, J$.R(104841, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(104849, 'connection', connection, false, false), J$.T(104857, '$external.$cmd', 21, false), J$.R(104865, 'command', command, false, false), J$.T(105177, function (err, doc) {
                                jalangiLabel27:
                                    while (true) {
                                        try {
                                            J$.Fe(105145, arguments.callee, this, arguments);
                                            arguments = J$.N(105153, 'arguments', arguments, true, false, false);
                                            err = J$.N(105161, 'err', err, true, false, false);
                                            doc = J$.N(105169, 'doc', doc, true, false, false);
                                            if (J$.C(5960, J$.R(104873, 'err', err, false, false)))
                                                return J$.Rt(104913, J$.F(104905, J$.R(104881, 'callback', callback, false, false), false)(J$.R(104889, 'err', err, false, false), J$.T(104897, false, 23, false)));
                                            J$.M(105137, J$.R(104921, 'mongo_auth_process', mongo_auth_process, false, false), 'transition', false)(J$.G(104937, J$.R(104929, 'doc', doc, false, false), 'payload'), J$.T(105129, function (err, payload) {
                                                jalangiLabel26:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(105097, arguments.callee, this, arguments);
                                                            arguments = J$.N(105105, 'arguments', arguments, true, false, false);
                                                            err = J$.N(105113, 'err', err, true, false, false);
                                                            payload = J$.N(105121, 'payload', payload, true, false, false);
                                                            if (J$.C(5968, J$.R(104945, 'err', err, false, false)))
                                                                return J$.Rt(104985, J$.F(104977, J$.R(104953, 'callback', callback, false, false), false)(J$.R(104961, 'err', err, false, false), J$.T(104969, false, 23, false)));
                                                            J$.F(105089, J$.R(104993, 'MongoDBGSSAPIThirdStep', MongoDBGSSAPIThirdStep, false, true), false)(J$.R(105001, 'self', self, false, false), J$.R(105009, 'mongo_auth_process', mongo_auth_process, false, false), J$.R(105017, 'payload', payload, false, false), J$.R(105025, 'doc', doc, false, false), J$.R(105033, 'db', db, false, false), J$.R(105041, 'username', username, false, false), J$.R(105049, 'password', password, false, false), J$.R(105057, 'authdb', authdb, false, false), J$.R(105065, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(105073, 'connection', connection, false, false), J$.R(105081, 'callback', callback, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(106449, J$e);
                                                        } finally {
                                                            if (J$.Fr(106457))
                                                                continue jalangiLabel26;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(106465, J$e);
                                        } finally {
                                            if (J$.Fr(106473))
                                                continue jalangiLabel27;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(106481, J$e);
                        } finally {
                            if (J$.Fr(106489))
                                continue jalangiLabel28;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), MongoDBGSSAPISecondStep, false, true);
            var MongoDBGSSAPIThirdStep = J$.W(105753, 'MongoDBGSSAPIThirdStep', J$.T(105745, function MongoDBGSSAPIThirdStep(self, mongo_auth_process, payload, doc, db, username, password, authdb, sendAuthCommand, connection, callback) {
                jalangiLabel31:
                    while (true) {
                        try {
                            J$.Fe(105633, arguments.callee, this, arguments);
                            arguments = J$.N(105641, 'arguments', arguments, true, false, false);
                            self = J$.N(105649, 'self', self, true, false, false);
                            mongo_auth_process = J$.N(105657, 'mongo_auth_process', mongo_auth_process, true, false, false);
                            payload = J$.N(105665, 'payload', payload, true, false, false);
                            doc = J$.N(105673, 'doc', doc, true, false, false);
                            db = J$.N(105681, 'db', db, true, false, false);
                            username = J$.N(105689, 'username', username, true, false, false);
                            password = J$.N(105697, 'password', password, true, false, false);
                            authdb = J$.N(105705, 'authdb', authdb, true, false, false);
                            sendAuthCommand = J$.N(105713, 'sendAuthCommand', sendAuthCommand, true, false, false);
                            connection = J$.N(105721, 'connection', connection, true, false, false);
                            callback = J$.N(105729, 'callback', callback, true, false, false);
                            J$.N(105737, 'command', command, false, false, false);
                            var command = J$.W(105361, 'command', J$.T(105353, {
                                saslContinue: J$.T(105321, 1, 22, false),
                                conversationId: J$.G(105337, J$.R(105329, 'doc', doc, false, false), 'conversationId'),
                                payload: J$.R(105345, 'payload', payload, false, false)
                            }, 11, false), command, false, false);
                            J$.F(105625, J$.R(105369, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(105377, 'connection', connection, false, false), J$.T(105385, '$external.$cmd', 21, false), J$.R(105393, 'command', command, false, false), J$.T(105617, function (err, r) {
                                jalangiLabel30:
                                    while (true) {
                                        try {
                                            J$.Fe(105585, arguments.callee, this, arguments);
                                            arguments = J$.N(105593, 'arguments', arguments, true, false, false);
                                            err = J$.N(105601, 'err', err, true, false, false);
                                            r = J$.N(105609, 'r', r, true, false, false);
                                            if (J$.C(5976, J$.R(105401, 'err', err, false, false)))
                                                return J$.Rt(105441, J$.F(105433, J$.R(105409, 'callback', callback, false, false), false)(J$.R(105417, 'err', err, false, false), J$.T(105425, false, 23, false)));
                                            J$.M(105577, J$.R(105449, 'mongo_auth_process', mongo_auth_process, false, false), 'transition', false)(J$.T(105457, null, 25, false), J$.T(105569, function (err) {
                                                jalangiLabel29:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(105545, arguments.callee, this, arguments);
                                                            arguments = J$.N(105553, 'arguments', arguments, true, false, false);
                                                            err = J$.N(105561, 'err', err, true, false, false);
                                                            if (J$.C(5984, J$.R(105465, 'err', err, false, false)))
                                                                return J$.Rt(105505, J$.F(105497, J$.R(105473, 'callback', callback, false, false), false)(J$.R(105481, 'err', err, false, false), J$.T(105489, null, 25, false)));
                                                            J$.F(105537, J$.R(105513, 'callback', callback, false, false), false)(J$.T(105521, null, 25, false), J$.R(105529, 'r', r, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(106497, J$e);
                                                        } finally {
                                                            if (J$.Fr(106505))
                                                                continue jalangiLabel29;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(106513, J$e);
                                        } finally {
                                            if (J$.Fr(106521))
                                                continue jalangiLabel30;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(106529, J$e);
                        } finally {
                            if (J$.Fr(106537))
                                continue jalangiLabel31;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), MongoDBGSSAPIThirdStep, false, true);
            J$.P(105777, J$.I(typeof module === 'undefined' ? module = J$.R(105761, 'module', undefined, true, true) : module = J$.R(105761, 'module', module, true, true)), 'exports', J$.R(105769, 'GSSAPI', GSSAPI, false, true));
        } catch (J$e) {
            J$.Ex(106545, J$e);
        } finally {
            if (J$.Sr(106553))
                continue jalangiLabel32;
            else
                break jalangiLabel32;
        }
    }
// JALANGI DO NOT INSTRUMENT
