// mongodb/lib/core/auth/plain.js
J$.noInstrEval = false;
jalangiLabel16:
    while (true) {
        try {
            J$.Se(113697, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/core/auth/plain.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(111625, arguments.callee, this, arguments);
                            arguments = J$.N(111633, 'arguments', arguments, true, false, false);
                            obj = J$.N(111641, 'obj', obj, true, false, false);
                            if (J$.C(6360, J$.C(6320, J$.B(5922, '===', J$.U(5914, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(111361, 'Symbol', undefined, true, true) : Symbol = J$.R(111361, 'Symbol', Symbol, true, true))), J$.T(111369, 'function', 21, false))) ? J$.B(5938, '===', J$.U(5930, 'typeof', J$.G(111385, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(111377, 'Symbol', undefined, true, true) : Symbol = J$.R(111377, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(111393, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(111449, '_typeof', J$.T(111441, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(111417, arguments.callee, this, arguments);
                                                arguments = J$.N(111425, 'arguments', arguments, true, false, false);
                                                obj = J$.N(111433, 'obj', obj, true, false, false);
                                                return J$.Rt(111409, J$.U(5946, 'typeof', J$.R(111401, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(113889, J$e);
                                            } finally {
                                                if (J$.Fr(113897))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(111585, '_typeof', J$.T(111577, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(111553, arguments.callee, this, arguments);
                                                arguments = J$.N(111561, 'arguments', arguments, true, false, false);
                                                obj = J$.N(111569, 'obj', obj, true, false, false);
                                                return J$.Rt(111545, J$.C(6352, J$.C(6344, J$.C(6336, J$.C(6328, J$.R(111457, 'obj', obj, false, false)) ? J$.B(5962, '===', J$.U(5954, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(111465, 'Symbol', undefined, true, true) : Symbol = J$.R(111465, 'Symbol', Symbol, true, true))), J$.T(111473, 'function', 21, false)) : J$._()) ? J$.B(5970, '===', J$.G(111489, J$.R(111481, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(111497, 'Symbol', undefined, true, true) : Symbol = J$.R(111497, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(5978, '!==', J$.R(111505, 'obj', obj, false, false), J$.G(111521, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(111513, 'Symbol', undefined, true, true) : Symbol = J$.R(111513, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(111529, 'symbol', 21, false) : J$.U(5986, 'typeof', J$.R(111537, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(113905, J$e);
                                            } finally {
                                                if (J$.Fr(113913))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(111617, J$.F(111609, J$.R(111593, '_typeof', _typeof, false, true), false)(J$.R(111601, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(113921, J$e);
                        } finally {
                            if (J$.Fr(113929))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(111689, arguments.callee, this, arguments);
                            arguments = J$.N(111697, 'arguments', arguments, true, false, false);
                            instance = J$.N(111705, 'instance', instance, true, false, false);
                            Constructor = J$.N(111713, 'Constructor', Constructor, true, false, false);
                            if (J$.C(6368, J$.U(6002, '!', J$.B(5994, 'instanceof', J$.R(111649, 'instance', instance, false, false), J$.R(111657, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(111681, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(111665, 'TypeError', undefined, true, true) : TypeError = J$.R(111665, 'TypeError', TypeError, true, true)), true)(J$.T(111673, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(113937, J$e);
                        } finally {
                            if (J$.Fr(113945))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(111961, arguments.callee, this, arguments);
                            arguments = J$.N(111969, 'arguments', arguments, true, false, false);
                            target = J$.N(111977, 'target', target, true, false, false);
                            props = J$.N(111985, 'props', props, true, false, false);
                            J$.N(111993, 'i', i, false, false, false);
                            J$.N(112001, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(111729, 'i', J$.T(111721, 0, 22, false), i, false, false); J$.C(6392, J$.B(6010, '<', J$.R(111737, 'i', i, false, false), J$.G(111753, J$.R(111745, 'props', props, false, false), 'length'))); J$.B(6034, '-', i = J$.W(111769, 'i', J$.B(6026, '+', J$.U(6018, '+', J$.R(111761, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(111801, 'descriptor', J$.G(111793, J$.R(111777, 'props', props, false, false), J$.R(111785, 'i', i, false, false)), descriptor, false, false);
                                J$.P(111841, J$.R(111809, 'descriptor', descriptor, false, false), 'enumerable', J$.C(6376, J$.G(111825, J$.R(111817, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(111833, false, 23, false));
                                J$.P(111865, J$.R(111849, 'descriptor', descriptor, false, false), 'configurable', J$.T(111857, true, 23, false));
                                if (J$.C(6384, J$.B(6042, 'in', J$.T(111873, 'value', 21, false), J$.R(111881, 'descriptor', descriptor, false, false))))
                                    J$.P(111905, J$.R(111889, 'descriptor', descriptor, false, false), 'writable', J$.T(111897, true, 23, false));
                                J$.M(111953, J$.I(typeof Object === 'undefined' ? Object = J$.R(111913, 'Object', undefined, true, true) : Object = J$.R(111913, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(111921, 'target', target, false, false), J$.G(111937, J$.R(111929, 'descriptor', descriptor, false, false), 'key'), J$.R(111945, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(113953, J$e);
                        } finally {
                            if (J$.Fr(113961))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(112113, arguments.callee, this, arguments);
                            arguments = J$.N(112121, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(112129, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(112137, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(112145, 'staticProps', staticProps, true, false, false);
                            if (J$.C(6400, J$.R(112009, 'protoProps', protoProps, false, false)))
                                J$.F(112049, J$.R(112017, '_defineProperties', _defineProperties, false, true), false)(J$.G(112033, J$.R(112025, 'Constructor', Constructor, false, false), 'prototype'), J$.R(112041, 'protoProps', protoProps, false, false));
                            if (J$.C(6408, J$.R(112057, 'staticProps', staticProps, false, false)))
                                J$.F(112089, J$.R(112065, '_defineProperties', _defineProperties, false, true), false)(J$.R(112073, 'Constructor', Constructor, false, false), J$.R(112081, 'staticProps', staticProps, false, false));
                            return J$.Rt(112105, J$.R(112097, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(113969, J$e);
                        } finally {
                            if (J$.Fr(113977))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(112257, arguments.callee, this, arguments);
                            arguments = J$.N(112265, 'arguments', arguments, true, false, false);
                            self = J$.N(112273, 'self', self, true, false, false);
                            call = J$.N(112281, 'call', call, true, false, false);
                            if (J$.C(6432, J$.C(6424, J$.R(112153, 'call', call, false, false)) ? J$.C(6416, J$.B(6050, '===', J$.F(112177, J$.R(112161, '_typeof', _typeof, false, true), false)(J$.R(112169, 'call', call, false, false)), J$.T(112185, 'object', 21, false))) ? J$._() : J$.B(6066, '===', J$.U(6058, 'typeof', J$.R(112193, 'call', call, false, false)), J$.T(112201, 'function', 21, false)) : J$._())) {
                                return J$.Rt(112217, J$.R(112209, 'call', call, false, false));
                            }
                            return J$.Rt(112249, J$.F(112241, J$.R(112225, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(112233, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(113985, J$e);
                        } finally {
                            if (J$.Fr(113993))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(112345, arguments.callee, this, arguments);
                            arguments = J$.N(112353, 'arguments', arguments, true, false, false);
                            self = J$.N(112361, 'self', self, true, false, false);
                            if (J$.C(6440, J$.B(6074, '===', J$.R(112289, 'self', self, false, false), void J$.T(112297, 0, 22, false)))) {
                                throw J$.F(112321, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(112305, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(112305, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(112313, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(112337, J$.R(112329, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(114001, J$e);
                        } finally {
                            if (J$.Fr(114009))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(112521, arguments.callee, this, arguments);
                            arguments = J$.N(112529, 'arguments', arguments, true, false, false);
                            o = J$.N(112537, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(112481, '_getPrototypeOf', J$.C(6456, J$.G(112377, J$.I(typeof Object === 'undefined' ? Object = J$.R(112369, 'Object', undefined, true, true) : Object = J$.R(112369, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(112393, J$.I(typeof Object === 'undefined' ? Object = J$.R(112385, 'Object', undefined, true, true) : Object = J$.R(112385, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(112473, function _getPrototypeOf(o) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(112449, arguments.callee, this, arguments);
                                            arguments = J$.N(112457, 'arguments', arguments, true, false, false);
                                            o = J$.N(112465, 'o', o, true, false, false);
                                            return J$.Rt(112441, J$.C(6448, J$.G(112409, J$.R(112401, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(112433, J$.I(typeof Object === 'undefined' ? Object = J$.R(112417, 'Object', undefined, true, true) : Object = J$.R(112417, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(112425, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(114017, J$e);
                                        } finally {
                                            if (J$.Fr(114025))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(112513, J$.F(112505, J$.R(112489, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(112497, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(114033, J$e);
                        } finally {
                            if (J$.Fr(114041))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(112737, arguments.callee, this, arguments);
                            arguments = J$.N(112745, 'arguments', arguments, true, false, false);
                            subClass = J$.N(112753, 'subClass', subClass, true, false, false);
                            superClass = J$.N(112761, 'superClass', superClass, true, false, false);
                            if (J$.C(6472, J$.C(6464, J$.B(6090, '!==', J$.U(6082, 'typeof', J$.R(112545, 'superClass', superClass, false, false)), J$.T(112553, 'function', 21, false))) ? J$.B(6098, '!==', J$.R(112561, 'superClass', superClass, false, false), J$.T(112569, null, 25, false)) : J$._())) {
                                throw J$.F(112593, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(112577, 'TypeError', undefined, true, true) : TypeError = J$.R(112577, 'TypeError', TypeError, true, true)), true)(J$.T(112585, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(112689, J$.R(112601, 'subClass', subClass, false, false), 'prototype', J$.M(112681, J$.I(typeof Object === 'undefined' ? Object = J$.R(112609, 'Object', undefined, true, true) : Object = J$.R(112609, 'Object', Object, true, true)), 'create', false)(J$.C(6480, J$.R(112617, 'superClass', superClass, false, false)) ? J$.G(112633, J$.R(112625, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(112673, {
                                constructor: J$.T(112665, {
                                    value: J$.R(112641, 'subClass', subClass, false, false),
                                    writable: J$.T(112649, true, 23, false),
                                    configurable: J$.T(112657, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(6488, J$.R(112697, 'superClass', superClass, false, false)))
                                J$.F(112729, J$.R(112705, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(112713, 'subClass', subClass, false, false), J$.R(112721, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(114049, J$e);
                        } finally {
                            if (J$.Fr(114057))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(112913, arguments.callee, this, arguments);
                            arguments = J$.N(112921, 'arguments', arguments, true, false, false);
                            o = J$.N(112929, 'o', o, true, false, false);
                            p = J$.N(112937, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(112865, '_setPrototypeOf', J$.C(6496, J$.G(112777, J$.I(typeof Object === 'undefined' ? Object = J$.R(112769, 'Object', undefined, true, true) : Object = J$.R(112769, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(112857, function _setPrototypeOf(o, p) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(112825, arguments.callee, this, arguments);
                                            arguments = J$.N(112833, 'arguments', arguments, true, false, false);
                                            o = J$.N(112841, 'o', o, true, false, false);
                                            p = J$.N(112849, 'p', p, true, false, false);
                                            J$.P(112801, J$.R(112785, 'o', o, false, false), '__proto__', J$.R(112793, 'p', p, false, false));
                                            return J$.Rt(112817, J$.R(112809, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(114065, J$e);
                                        } finally {
                                            if (J$.Fr(114073))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(112905, J$.F(112897, J$.R(112873, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(112881, 'o', o, false, false), J$.R(112889, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(114081, J$e);
                        } finally {
                            if (J$.Fr(114089))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(113713, '_typeof', J$.T(113705, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(113729, '_classCallCheck', J$.T(113721, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(113745, '_defineProperties', J$.T(113737, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(113761, '_createClass', J$.T(113753, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(113777, '_possibleConstructorReturn', J$.T(113769, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(113793, '_assertThisInitialized', J$.T(113785, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(113809, '_getPrototypeOf', J$.T(113801, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(113825, '_inherits', J$.T(113817, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(113841, '_setPrototypeOf', J$.T(113833, _setPrototypeOf, 12, false), true, false, false);
            J$.N(113849, 'retrieveBSON', retrieveBSON, false, false, false);
            J$.N(113857, 'AuthProvider', AuthProvider, false, false, false);
            J$.N(113865, 'BSON', BSON, false, false, false);
            J$.N(113873, 'Binary', Binary, false, false, false);
            J$.N(113881, 'Plain', Plain, false, false, false);
            J$.T(111353, 'use strict', 21, false);
            var retrieveBSON = J$.W(112977, 'retrieveBSON', J$.G(112969, J$.F(112961, J$.I(typeof require === 'undefined' ? require = J$.R(112945, 'require', undefined, true, true) : require = J$.R(112945, 'require', require, true, true)), false)(J$.T(112953, '../connection/utils', 21, false)), 'retrieveBSON'), retrieveBSON, false, true);
            var AuthProvider = J$.W(113017, 'AuthProvider', J$.G(113009, J$.F(113001, J$.I(typeof require === 'undefined' ? require = J$.R(112985, 'require', undefined, true, true) : require = J$.R(112985, 'require', require, true, true)), false)(J$.T(112993, './auth_provider', 21, false)), 'AuthProvider'), AuthProvider, false, true);
            var BSON = J$.W(113041, 'BSON', J$.F(113033, J$.R(113025, 'retrieveBSON', retrieveBSON, false, true), false)(), BSON, false, true);
            var Binary = J$.W(113065, 'Binary', J$.G(113057, J$.R(113049, 'BSON', BSON, false, true), 'Binary'), Binary, false, true);
            var Plain = J$.W(113665, 'Plain', J$.F(113657, J$.T(113641, function (_AuthProvider) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(113601, arguments.callee, this, arguments);
                            function Plain() {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(113217, arguments.callee, this, arguments);
                                            arguments = J$.N(113225, 'arguments', arguments, true, false, false);
                                            J$.F(113129, J$.R(113105, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(113113, 'this', this, false, false), J$.R(113121, 'Plain', Plain, false, false));
                                            return J$.Rt(113209, J$.F(113201, J$.R(113137, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(113145, 'this', this, false, false), J$.M(113193, J$.F(113169, J$.R(113153, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(113161, 'Plain', Plain, false, false)), 'apply', false)(J$.R(113177, 'this', this, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(113185, 'arguments', undefined, true, true) : arguments = J$.R(113185, 'arguments', arguments, true, true)))));
                                        } catch (J$e) {
                                            J$.Ex(114097, J$e);
                                        } finally {
                                            if (J$.Fr(114105))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(113609, 'arguments', arguments, true, false, false);
                            _AuthProvider = J$.N(113617, '_AuthProvider', _AuthProvider, true, false, false);
                            Plain = J$.N(113633, 'Plain', J$.T(113625, Plain, 12, false), true, false, false);
                            J$.F(113097, J$.R(113073, '_inherits', _inherits, false, true), false)(J$.R(113081, 'Plain', Plain, false, false), J$.R(113089, '_AuthProvider', _AuthProvider, false, false));
                            J$.F(113577, J$.R(113233, '_createClass', _createClass, false, true), false)(J$.R(113241, 'Plain', Plain, false, false), J$.T(113569, [J$.T(113561, {
                                    key: J$.T(113249, '_authenticateSingleConnection', 21, false),
                                    value: J$.T(113553, function _authenticateSingleConnection(sendAuthCommand, connection, credentials, callback) {
                                        jalangiLabel14:
                                            while (true) {
                                                try {
                                                    J$.Fe(113473, arguments.callee, this, arguments);
                                                    arguments = J$.N(113481, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(113489, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connection = J$.N(113497, 'connection', connection, true, false, false);
                                                    credentials = J$.N(113505, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(113513, 'callback', callback, true, false, false);
                                                    J$.N(113521, 'username', username, false, false, false);
                                                    J$.N(113529, 'password', password, false, false, false);
                                                    J$.N(113537, 'payload', payload, false, false, false);
                                                    J$.N(113545, 'command', command, false, false, false);
                                                    var username = J$.W(113273, 'username', J$.G(113265, J$.R(113257, 'credentials', credentials, false, false), 'username'), username, false, false);
                                                    var password = J$.W(113297, 'password', J$.G(113289, J$.R(113281, 'credentials', credentials, false, false), 'password'), password, false, false);
                                                    var payload = J$.W(113369, 'payload', J$.F(113361, J$.R(113305, 'Binary', Binary, false, true), true)(J$.M(113353, J$.M(113337, J$.T(113313, '\0', 21, false), 'concat', false)(J$.R(113321, 'username', username, false, false), J$.T(113329, '\0', 21, false)), 'concat', false)(J$.R(113345, 'password', password, false, false))), payload, false, false);
                                                    var command = J$.W(113417, 'command', J$.T(113409, {
                                                        saslStart: J$.T(113377, 1, 22, false),
                                                        mechanism: J$.T(113385, 'PLAIN', 21, false),
                                                        payload: J$.R(113393, 'payload', payload, false, false),
                                                        autoAuthorize: J$.T(113401, 1, 22, false)
                                                    }, 11, false), command, false, false);
                                                    J$.F(113465, J$.R(113425, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(113433, 'connection', connection, false, false), J$.T(113441, '$external.$cmd', 21, false), J$.R(113449, 'command', command, false, false), J$.R(113457, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(114113, J$e);
                                                } finally {
                                                    if (J$.Fr(114121))
                                                        continue jalangiLabel14;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(113593, J$.R(113585, 'Plain', Plain, false, false));
                        } catch (J$e) {
                            J$.Ex(114129, J$e);
                        } finally {
                            if (J$.Fr(114137))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(113649, 'AuthProvider', AuthProvider, false, true)), Plain, false, true);
            J$.P(113689, J$.I(typeof module === 'undefined' ? module = J$.R(113673, 'module', undefined, true, true) : module = J$.R(113673, 'module', module, true, true)), 'exports', J$.R(113681, 'Plain', Plain, false, true));
        } catch (J$e) {
            J$.Ex(114145, J$e);
        } finally {
            if (J$.Sr(114153))
                continue jalangiLabel16;
            else
                break jalangiLabel16;
        }
    }
// JALANGI DO NOT INSTRUMENT
