// mongodb/lib/core/auth/sspi.js
J$.noInstrEval = false;
jalangiLabel30:
    while (true) {
        try {
            J$.Se(127713, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/core/auth/sspi.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(123481, arguments.callee, this, arguments);
                            arguments = J$.N(123489, 'arguments', arguments, true, false, false);
                            obj = J$.N(123497, 'obj', obj, true, false, false);
                            if (J$.C(6976, J$.C(6936, J$.B(6546, '===', J$.U(6538, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(123217, 'Symbol', undefined, true, true) : Symbol = J$.R(123217, 'Symbol', Symbol, true, true))), J$.T(123225, 'function', 21, false))) ? J$.B(6562, '===', J$.U(6554, 'typeof', J$.G(123241, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(123233, 'Symbol', undefined, true, true) : Symbol = J$.R(123233, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(123249, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(123305, '_typeof', J$.T(123297, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(123273, arguments.callee, this, arguments);
                                                arguments = J$.N(123281, 'arguments', arguments, true, false, false);
                                                obj = J$.N(123289, 'obj', obj, true, false, false);
                                                return J$.Rt(123265, J$.U(6570, 'typeof', J$.R(123257, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(127945, J$e);
                                            } finally {
                                                if (J$.Fr(127953))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(123441, '_typeof', J$.T(123433, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(123409, arguments.callee, this, arguments);
                                                arguments = J$.N(123417, 'arguments', arguments, true, false, false);
                                                obj = J$.N(123425, 'obj', obj, true, false, false);
                                                return J$.Rt(123401, J$.C(6968, J$.C(6960, J$.C(6952, J$.C(6944, J$.R(123313, 'obj', obj, false, false)) ? J$.B(6586, '===', J$.U(6578, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(123321, 'Symbol', undefined, true, true) : Symbol = J$.R(123321, 'Symbol', Symbol, true, true))), J$.T(123329, 'function', 21, false)) : J$._()) ? J$.B(6594, '===', J$.G(123345, J$.R(123337, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(123353, 'Symbol', undefined, true, true) : Symbol = J$.R(123353, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(6602, '!==', J$.R(123361, 'obj', obj, false, false), J$.G(123377, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(123369, 'Symbol', undefined, true, true) : Symbol = J$.R(123369, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(123385, 'symbol', 21, false) : J$.U(6610, 'typeof', J$.R(123393, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(127961, J$e);
                                            } finally {
                                                if (J$.Fr(127969))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(123473, J$.F(123465, J$.R(123449, '_typeof', _typeof, false, true), false)(J$.R(123457, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(127977, J$e);
                        } finally {
                            if (J$.Fr(127985))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(123545, arguments.callee, this, arguments);
                            arguments = J$.N(123553, 'arguments', arguments, true, false, false);
                            instance = J$.N(123561, 'instance', instance, true, false, false);
                            Constructor = J$.N(123569, 'Constructor', Constructor, true, false, false);
                            if (J$.C(6984, J$.U(6626, '!', J$.B(6618, 'instanceof', J$.R(123505, 'instance', instance, false, false), J$.R(123513, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(123537, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(123521, 'TypeError', undefined, true, true) : TypeError = J$.R(123521, 'TypeError', TypeError, true, true)), true)(J$.T(123529, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(127993, J$e);
                        } finally {
                            if (J$.Fr(128001))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(123817, arguments.callee, this, arguments);
                            arguments = J$.N(123825, 'arguments', arguments, true, false, false);
                            target = J$.N(123833, 'target', target, true, false, false);
                            props = J$.N(123841, 'props', props, true, false, false);
                            J$.N(123849, 'i', i, false, false, false);
                            J$.N(123857, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(123585, 'i', J$.T(123577, 0, 22, false), i, false, false); J$.C(7008, J$.B(6634, '<', J$.R(123593, 'i', i, false, false), J$.G(123609, J$.R(123601, 'props', props, false, false), 'length'))); J$.B(6658, '-', i = J$.W(123625, 'i', J$.B(6650, '+', J$.U(6642, '+', J$.R(123617, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(123657, 'descriptor', J$.G(123649, J$.R(123633, 'props', props, false, false), J$.R(123641, 'i', i, false, false)), descriptor, false, false);
                                J$.P(123697, J$.R(123665, 'descriptor', descriptor, false, false), 'enumerable', J$.C(6992, J$.G(123681, J$.R(123673, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(123689, false, 23, false));
                                J$.P(123721, J$.R(123705, 'descriptor', descriptor, false, false), 'configurable', J$.T(123713, true, 23, false));
                                if (J$.C(7000, J$.B(6666, 'in', J$.T(123729, 'value', 21, false), J$.R(123737, 'descriptor', descriptor, false, false))))
                                    J$.P(123761, J$.R(123745, 'descriptor', descriptor, false, false), 'writable', J$.T(123753, true, 23, false));
                                J$.M(123809, J$.I(typeof Object === 'undefined' ? Object = J$.R(123769, 'Object', undefined, true, true) : Object = J$.R(123769, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(123777, 'target', target, false, false), J$.G(123793, J$.R(123785, 'descriptor', descriptor, false, false), 'key'), J$.R(123801, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(128009, J$e);
                        } finally {
                            if (J$.Fr(128017))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(123969, arguments.callee, this, arguments);
                            arguments = J$.N(123977, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(123985, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(123993, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(124001, 'staticProps', staticProps, true, false, false);
                            if (J$.C(7016, J$.R(123865, 'protoProps', protoProps, false, false)))
                                J$.F(123905, J$.R(123873, '_defineProperties', _defineProperties, false, true), false)(J$.G(123889, J$.R(123881, 'Constructor', Constructor, false, false), 'prototype'), J$.R(123897, 'protoProps', protoProps, false, false));
                            if (J$.C(7024, J$.R(123913, 'staticProps', staticProps, false, false)))
                                J$.F(123945, J$.R(123921, '_defineProperties', _defineProperties, false, true), false)(J$.R(123929, 'Constructor', Constructor, false, false), J$.R(123937, 'staticProps', staticProps, false, false));
                            return J$.Rt(123961, J$.R(123953, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(128025, J$e);
                        } finally {
                            if (J$.Fr(128033))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(124113, arguments.callee, this, arguments);
                            arguments = J$.N(124121, 'arguments', arguments, true, false, false);
                            self = J$.N(124129, 'self', self, true, false, false);
                            call = J$.N(124137, 'call', call, true, false, false);
                            if (J$.C(7048, J$.C(7040, J$.R(124009, 'call', call, false, false)) ? J$.C(7032, J$.B(6674, '===', J$.F(124033, J$.R(124017, '_typeof', _typeof, false, true), false)(J$.R(124025, 'call', call, false, false)), J$.T(124041, 'object', 21, false))) ? J$._() : J$.B(6690, '===', J$.U(6682, 'typeof', J$.R(124049, 'call', call, false, false)), J$.T(124057, 'function', 21, false)) : J$._())) {
                                return J$.Rt(124073, J$.R(124065, 'call', call, false, false));
                            }
                            return J$.Rt(124105, J$.F(124097, J$.R(124081, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(124089, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(128041, J$e);
                        } finally {
                            if (J$.Fr(128049))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(124201, arguments.callee, this, arguments);
                            arguments = J$.N(124209, 'arguments', arguments, true, false, false);
                            self = J$.N(124217, 'self', self, true, false, false);
                            if (J$.C(7056, J$.B(6698, '===', J$.R(124145, 'self', self, false, false), void J$.T(124153, 0, 22, false)))) {
                                throw J$.F(124177, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(124161, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(124161, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(124169, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(124193, J$.R(124185, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(128057, J$e);
                        } finally {
                            if (J$.Fr(128065))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(124585, arguments.callee, this, arguments);
                            arguments = J$.N(124593, 'arguments', arguments, true, false, false);
                            target = J$.N(124601, 'target', target, true, false, false);
                            property = J$.N(124609, 'property', property, true, false, false);
                            receiver = J$.N(124617, 'receiver', receiver, true, false, false);
                            if (J$.C(7088, J$.C(7064, J$.B(6714, '!==', J$.U(6706, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(124225, 'Reflect', undefined, true, true) : Reflect = J$.R(124225, 'Reflect', Reflect, true, true))), J$.T(124233, 'undefined', 21, false))) ? J$.G(124249, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(124241, 'Reflect', undefined, true, true) : Reflect = J$.R(124241, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(124273, '_get', J$.G(124265, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(124257, 'Reflect', undefined, true, true) : Reflect = J$.R(124257, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(124521, '_get', J$.T(124513, function _get(target, property, receiver) {
                                    jalangiLabel8:
                                        while (true) {
                                            try {
                                                J$.Fe(124457, arguments.callee, this, arguments);
                                                arguments = J$.N(124465, 'arguments', arguments, true, false, false);
                                                target = J$.N(124473, 'target', target, true, false, false);
                                                property = J$.N(124481, 'property', property, true, false, false);
                                                receiver = J$.N(124489, 'receiver', receiver, true, false, false);
                                                J$.N(124497, 'base', base, false, false, false);
                                                J$.N(124505, 'desc', desc, false, false, false);
                                                var base = J$.W(124313, 'base', J$.F(124305, J$.R(124281, '_superPropBase', _superPropBase, false, true), false)(J$.R(124289, 'target', target, false, false), J$.R(124297, 'property', property, false, false)), base, false, false);
                                                if (J$.C(7072, J$.U(6722, '!', J$.R(124321, 'base', base, false, false))))
                                                    return J$.Rt(124329, undefined);
                                                var desc = J$.W(124369, 'desc', J$.M(124361, J$.I(typeof Object === 'undefined' ? Object = J$.R(124337, 'Object', undefined, true, true) : Object = J$.R(124337, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(124345, 'base', base, false, false), J$.R(124353, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(7080, J$.G(124385, J$.R(124377, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(124425, J$.M(124417, J$.G(124401, J$.R(124393, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(124409, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(124449, J$.G(124441, J$.R(124433, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(128073, J$e);
                                            } finally {
                                                if (J$.Fr(128081))
                                                    continue jalangiLabel8;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(124577, J$.F(124569, J$.R(124529, '_get', _get, false, true), false)(J$.R(124537, 'target', target, false, false), J$.R(124545, 'property', property, false, false), J$.C(7096, J$.R(124553, 'receiver', receiver, false, false)) ? J$._() : J$.R(124561, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(128089, J$e);
                        } finally {
                            if (J$.Fr(128097))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(124737, arguments.callee, this, arguments);
                            arguments = J$.N(124745, 'arguments', arguments, true, false, false);
                            object = J$.N(124753, 'object', object, true, false, false);
                            property = J$.N(124761, 'property', property, true, false, false);
                            while (J$.C(7112, J$.U(6730, '!', J$.M(124665, J$.G(124641, J$.G(124633, J$.I(typeof Object === 'undefined' ? Object = J$.R(124625, 'Object', undefined, true, true) : Object = J$.R(124625, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(124649, 'object', object, false, false), J$.R(124657, 'property', property, false, false))))) {
                                object = J$.W(124697, 'object', J$.F(124689, J$.R(124673, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(124681, 'object', object, false, false)), object, false, false);
                                if (J$.C(7104, J$.B(6738, '===', J$.R(124705, 'object', object, false, false), J$.T(124713, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(124729, J$.R(124721, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(128105, J$e);
                        } finally {
                            if (J$.Fr(128113))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(124921, arguments.callee, this, arguments);
                            arguments = J$.N(124929, 'arguments', arguments, true, false, false);
                            o = J$.N(124937, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(124881, '_getPrototypeOf', J$.C(7128, J$.G(124777, J$.I(typeof Object === 'undefined' ? Object = J$.R(124769, 'Object', undefined, true, true) : Object = J$.R(124769, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(124793, J$.I(typeof Object === 'undefined' ? Object = J$.R(124785, 'Object', undefined, true, true) : Object = J$.R(124785, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(124873, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(124849, arguments.callee, this, arguments);
                                            arguments = J$.N(124857, 'arguments', arguments, true, false, false);
                                            o = J$.N(124865, 'o', o, true, false, false);
                                            return J$.Rt(124841, J$.C(7120, J$.G(124809, J$.R(124801, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(124833, J$.I(typeof Object === 'undefined' ? Object = J$.R(124817, 'Object', undefined, true, true) : Object = J$.R(124817, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(124825, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(128121, J$e);
                                        } finally {
                                            if (J$.Fr(128129))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(124913, J$.F(124905, J$.R(124889, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(124897, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(128137, J$e);
                        } finally {
                            if (J$.Fr(128145))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(125137, arguments.callee, this, arguments);
                            arguments = J$.N(125145, 'arguments', arguments, true, false, false);
                            subClass = J$.N(125153, 'subClass', subClass, true, false, false);
                            superClass = J$.N(125161, 'superClass', superClass, true, false, false);
                            if (J$.C(7144, J$.C(7136, J$.B(6754, '!==', J$.U(6746, 'typeof', J$.R(124945, 'superClass', superClass, false, false)), J$.T(124953, 'function', 21, false))) ? J$.B(6762, '!==', J$.R(124961, 'superClass', superClass, false, false), J$.T(124969, null, 25, false)) : J$._())) {
                                throw J$.F(124993, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(124977, 'TypeError', undefined, true, true) : TypeError = J$.R(124977, 'TypeError', TypeError, true, true)), true)(J$.T(124985, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(125089, J$.R(125001, 'subClass', subClass, false, false), 'prototype', J$.M(125081, J$.I(typeof Object === 'undefined' ? Object = J$.R(125009, 'Object', undefined, true, true) : Object = J$.R(125009, 'Object', Object, true, true)), 'create', false)(J$.C(7152, J$.R(125017, 'superClass', superClass, false, false)) ? J$.G(125033, J$.R(125025, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(125073, {
                                constructor: J$.T(125065, {
                                    value: J$.R(125041, 'subClass', subClass, false, false),
                                    writable: J$.T(125049, true, 23, false),
                                    configurable: J$.T(125057, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(7160, J$.R(125097, 'superClass', superClass, false, false)))
                                J$.F(125129, J$.R(125105, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(125113, 'subClass', subClass, false, false), J$.R(125121, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(128153, J$e);
                        } finally {
                            if (J$.Fr(128161))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(125313, arguments.callee, this, arguments);
                            arguments = J$.N(125321, 'arguments', arguments, true, false, false);
                            o = J$.N(125329, 'o', o, true, false, false);
                            p = J$.N(125337, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(125265, '_setPrototypeOf', J$.C(7168, J$.G(125177, J$.I(typeof Object === 'undefined' ? Object = J$.R(125169, 'Object', undefined, true, true) : Object = J$.R(125169, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(125257, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(125225, arguments.callee, this, arguments);
                                            arguments = J$.N(125233, 'arguments', arguments, true, false, false);
                                            o = J$.N(125241, 'o', o, true, false, false);
                                            p = J$.N(125249, 'p', p, true, false, false);
                                            J$.P(125201, J$.R(125185, 'o', o, false, false), '__proto__', J$.R(125193, 'p', p, false, false));
                                            return J$.Rt(125217, J$.R(125209, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(128169, J$e);
                                        } finally {
                                            if (J$.Fr(128177))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(125305, J$.F(125297, J$.R(125273, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(125281, 'o', o, false, false), J$.R(125289, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(128185, J$e);
                        } finally {
                            if (J$.Fr(128193))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function SSIPAuthenticate(self, MongoAuthProcess, username, password, gssapiServiceName, sendAuthCommand, connection, options, callback) {
                jalangiLabel29:
                    while (true) {
                        try {
                            J$.Fe(127577, arguments.callee, this, arguments);
                            function authCommand(command, authCb) {
                                jalangiLabel20:
                                    while (true) {
                                        try {
                                            J$.Fe(126441, arguments.callee, this, arguments);
                                            arguments = J$.N(126449, 'arguments', arguments, true, false, false);
                                            command = J$.N(126457, 'command', command, true, false, false);
                                            authCb = J$.N(126465, 'authCb', authCb, true, false, false);
                                            J$.F(126433, J$.R(126393, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(126401, 'connection', connection, false, false), J$.T(126409, '$external.$cmd', 21, false), J$.R(126417, 'command', command, false, false), J$.R(126425, 'authCb', authCb, false, false));
                                        } catch (J$e) {
                                            J$.Ex(128265, J$e);
                                        } finally {
                                            if (J$.Fr(128273))
                                                continue jalangiLabel20;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(127585, 'arguments', arguments, true, false, false);
                            self = J$.N(127593, 'self', self, true, false, false);
                            MongoAuthProcess = J$.N(127601, 'MongoAuthProcess', MongoAuthProcess, true, false, false);
                            username = J$.N(127609, 'username', username, true, false, false);
                            password = J$.N(127617, 'password', password, true, false, false);
                            gssapiServiceName = J$.N(127625, 'gssapiServiceName', gssapiServiceName, true, false, false);
                            sendAuthCommand = J$.N(127633, 'sendAuthCommand', sendAuthCommand, true, false, false);
                            connection = J$.N(127641, 'connection', connection, true, false, false);
                            options = J$.N(127649, 'options', options, true, false, false);
                            callback = J$.N(127657, 'callback', callback, true, false, false);
                            J$.N(127665, 'authProcess', authProcess, false, false, false);
                            authCommand = J$.N(127681, 'authCommand', J$.T(127673, authCommand, 12, false), true, false, false);
                            var authProcess = J$.W(126385, 'authProcess', J$.F(126377, J$.R(126321, 'MongoAuthProcess', MongoAuthProcess, false, false), true)(J$.G(126337, J$.R(126329, 'connection', connection, false, false), 'host'), J$.G(126353, J$.R(126345, 'connection', connection, false, false), 'port'), J$.R(126361, 'gssapiServiceName', gssapiServiceName, false, false), J$.R(126369, 'options', options, false, false)), authProcess, false, false);
                            J$.M(127569, J$.R(126473, 'authProcess', authProcess, false, false), 'init', false)(J$.R(126481, 'username', username, false, false), J$.R(126489, 'password', password, false, false), J$.T(127561, function (err) {
                                jalangiLabel28:
                                    while (true) {
                                        try {
                                            J$.Fe(127537, arguments.callee, this, arguments);
                                            arguments = J$.N(127545, 'arguments', arguments, true, false, false);
                                            err = J$.N(127553, 'err', err, true, false, false);
                                            if (J$.C(7200, J$.R(126497, 'err', err, false, false)))
                                                return J$.Rt(126537, J$.F(126529, J$.R(126505, 'callback', callback, false, false), false)(J$.R(126513, 'err', err, false, false), J$.T(126521, false, 23, false)));
                                            J$.M(127529, J$.R(126545, 'authProcess', authProcess, false, false), 'transition', false)(J$.T(126553, '', 21, false), J$.T(127521, function (err, payload) {
                                                jalangiLabel27:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(127481, arguments.callee, this, arguments);
                                                            arguments = J$.N(127489, 'arguments', arguments, true, false, false);
                                                            err = J$.N(127497, 'err', err, true, false, false);
                                                            payload = J$.N(127505, 'payload', payload, true, false, false);
                                                            J$.N(127513, 'command', command, false, false, false);
                                                            if (J$.C(7208, J$.R(126561, 'err', err, false, false)))
                                                                return J$.Rt(126601, J$.F(126593, J$.R(126569, 'callback', callback, false, false), false)(J$.R(126577, 'err', err, false, false), J$.T(126585, false, 23, false)));
                                                            var command = J$.W(126649, 'command', J$.T(126641, {
                                                                saslStart: J$.T(126609, 1, 22, false),
                                                                mechanism: J$.T(126617, 'GSSAPI', 21, false),
                                                                payload: J$.R(126625, 'payload', payload, false, false),
                                                                autoAuthorize: J$.T(126633, 1, 22, false)
                                                            }, 11, false), command, false, false);
                                                            J$.F(127473, J$.R(126657, 'authCommand', authCommand, false, false), false)(J$.R(126665, 'command', command, false, false), J$.T(127465, function (err, doc) {
                                                                jalangiLabel26:
                                                                    while (true) {
                                                                        try {
                                                                            J$.Fe(127433, arguments.callee, this, arguments);
                                                                            arguments = J$.N(127441, 'arguments', arguments, true, false, false);
                                                                            err = J$.N(127449, 'err', err, true, false, false);
                                                                            doc = J$.N(127457, 'doc', doc, true, false, false);
                                                                            if (J$.C(7216, J$.R(126673, 'err', err, false, false)))
                                                                                return J$.Rt(126713, J$.F(126705, J$.R(126681, 'callback', callback, false, false), false)(J$.R(126689, 'err', err, false, false), J$.T(126697, false, 23, false)));
                                                                            J$.M(127425, J$.R(126721, 'authProcess', authProcess, false, false), 'transition', false)(J$.G(126737, J$.R(126729, 'doc', doc, false, false), 'payload'), J$.T(127417, function (err, payload) {
                                                                                jalangiLabel25:
                                                                                    while (true) {
                                                                                        try {
                                                                                            J$.Fe(127377, arguments.callee, this, arguments);
                                                                                            arguments = J$.N(127385, 'arguments', arguments, true, false, false);
                                                                                            err = J$.N(127393, 'err', err, true, false, false);
                                                                                            payload = J$.N(127401, 'payload', payload, true, false, false);
                                                                                            J$.N(127409, 'command', command, false, false, false);
                                                                                            if (J$.C(7224, J$.R(126745, 'err', err, false, false)))
                                                                                                return J$.Rt(126785, J$.F(126777, J$.R(126753, 'callback', callback, false, false), false)(J$.R(126761, 'err', err, false, false), J$.T(126769, false, 23, false)));
                                                                                            var command = J$.W(126833, 'command', J$.T(126825, {
                                                                                                saslContinue: J$.T(126793, 1, 22, false),
                                                                                                conversationId: J$.G(126809, J$.R(126801, 'doc', doc, false, false), 'conversationId'),
                                                                                                payload: J$.R(126817, 'payload', payload, false, false)
                                                                                            }, 11, false), command, false, false);
                                                                                            J$.F(127369, J$.R(126841, 'authCommand', authCommand, false, false), false)(J$.R(126849, 'command', command, false, false), J$.T(127361, function (err, doc) {
                                                                                                jalangiLabel24:
                                                                                                    while (true) {
                                                                                                        try {
                                                                                                            J$.Fe(127329, arguments.callee, this, arguments);
                                                                                                            arguments = J$.N(127337, 'arguments', arguments, true, false, false);
                                                                                                            err = J$.N(127345, 'err', err, true, false, false);
                                                                                                            doc = J$.N(127353, 'doc', doc, true, false, false);
                                                                                                            if (J$.C(7232, J$.R(126857, 'err', err, false, false)))
                                                                                                                return J$.Rt(126897, J$.F(126889, J$.R(126865, 'callback', callback, false, false), false)(J$.R(126873, 'err', err, false, false), J$.T(126881, false, 23, false)));
                                                                                                            J$.M(127321, J$.R(126905, 'authProcess', authProcess, false, false), 'transition', false)(J$.G(126921, J$.R(126913, 'doc', doc, false, false), 'payload'), J$.T(127313, function (err, payload) {
                                                                                                                jalangiLabel23:
                                                                                                                    while (true) {
                                                                                                                        try {
                                                                                                                            J$.Fe(127273, arguments.callee, this, arguments);
                                                                                                                            arguments = J$.N(127281, 'arguments', arguments, true, false, false);
                                                                                                                            err = J$.N(127289, 'err', err, true, false, false);
                                                                                                                            payload = J$.N(127297, 'payload', payload, true, false, false);
                                                                                                                            J$.N(127305, 'command', command, false, false, false);
                                                                                                                            if (J$.C(7240, J$.R(126929, 'err', err, false, false)))
                                                                                                                                return J$.Rt(126969, J$.F(126961, J$.R(126937, 'callback', callback, false, false), false)(J$.R(126945, 'err', err, false, false), J$.T(126953, false, 23, false)));
                                                                                                                            var command = J$.W(127017, 'command', J$.T(127009, {
                                                                                                                                saslContinue: J$.T(126977, 1, 22, false),
                                                                                                                                conversationId: J$.G(126993, J$.R(126985, 'doc', doc, false, false), 'conversationId'),
                                                                                                                                payload: J$.R(127001, 'payload', payload, false, false)
                                                                                                                            }, 11, false), command, false, false);
                                                                                                                            J$.F(127265, J$.R(127025, 'authCommand', authCommand, false, false), false)(J$.R(127033, 'command', command, false, false), J$.T(127257, function (err, response) {
                                                                                                                                jalangiLabel22:
                                                                                                                                    while (true) {
                                                                                                                                        try {
                                                                                                                                            J$.Fe(127225, arguments.callee, this, arguments);
                                                                                                                                            arguments = J$.N(127233, 'arguments', arguments, true, false, false);
                                                                                                                                            err = J$.N(127241, 'err', err, true, false, false);
                                                                                                                                            response = J$.N(127249, 'response', response, true, false, false);
                                                                                                                                            if (J$.C(7248, J$.R(127041, 'err', err, false, false)))
                                                                                                                                                return J$.Rt(127081, J$.F(127073, J$.R(127049, 'callback', callback, false, false), false)(J$.R(127057, 'err', err, false, false), J$.T(127065, false, 23, false)));
                                                                                                                                            J$.M(127217, J$.R(127089, 'authProcess', authProcess, false, false), 'transition', false)(J$.T(127097, null, 25, false), J$.T(127209, function (err) {
                                                                                                                                                jalangiLabel21:
                                                                                                                                                    while (true) {
                                                                                                                                                        try {
                                                                                                                                                            J$.Fe(127185, arguments.callee, this, arguments);
                                                                                                                                                            arguments = J$.N(127193, 'arguments', arguments, true, false, false);
                                                                                                                                                            err = J$.N(127201, 'err', err, true, false, false);
                                                                                                                                                            if (J$.C(7256, J$.R(127105, 'err', err, false, false)))
                                                                                                                                                                return J$.Rt(127145, J$.F(127137, J$.R(127113, 'callback', callback, false, false), false)(J$.R(127121, 'err', err, false, false), J$.T(127129, null, 25, false)));
                                                                                                                                                            J$.F(127177, J$.R(127153, 'callback', callback, false, false), false)(J$.T(127161, null, 25, false), J$.R(127169, 'response', response, false, false));
                                                                                                                                                        } catch (J$e) {
                                                                                                                                                            J$.Ex(128281, J$e);
                                                                                                                                                        } finally {
                                                                                                                                                            if (J$.Fr(128289))
                                                                                                                                                                continue jalangiLabel21;
                                                                                                                                                            else
                                                                                                                                                                return J$.Ra();
                                                                                                                                                        }
                                                                                                                                                    }
                                                                                                                                            }, 12, false));
                                                                                                                                        } catch (J$e) {
                                                                                                                                            J$.Ex(128297, J$e);
                                                                                                                                        } finally {
                                                                                                                                            if (J$.Fr(128305))
                                                                                                                                                continue jalangiLabel22;
                                                                                                                                            else
                                                                                                                                                return J$.Ra();
                                                                                                                                        }
                                                                                                                                    }
                                                                                                                            }, 12, false));
                                                                                                                        } catch (J$e) {
                                                                                                                            J$.Ex(128313, J$e);
                                                                                                                        } finally {
                                                                                                                            if (J$.Fr(128321))
                                                                                                                                continue jalangiLabel23;
                                                                                                                            else
                                                                                                                                return J$.Ra();
                                                                                                                        }
                                                                                                                    }
                                                                                                            }, 12, false));
                                                                                                        } catch (J$e) {
                                                                                                            J$.Ex(128329, J$e);
                                                                                                        } finally {
                                                                                                            if (J$.Fr(128337))
                                                                                                                continue jalangiLabel24;
                                                                                                            else
                                                                                                                return J$.Ra();
                                                                                                        }
                                                                                                    }
                                                                                            }, 12, false));
                                                                                        } catch (J$e) {
                                                                                            J$.Ex(128345, J$e);
                                                                                        } finally {
                                                                                            if (J$.Fr(128353))
                                                                                                continue jalangiLabel25;
                                                                                            else
                                                                                                return J$.Ra();
                                                                                        }
                                                                                    }
                                                                            }, 12, false));
                                                                        } catch (J$e) {
                                                                            J$.Ex(128361, J$e);
                                                                        } finally {
                                                                            if (J$.Fr(128369))
                                                                                continue jalangiLabel26;
                                                                            else
                                                                                return J$.Ra();
                                                                        }
                                                                    }
                                                            }, 12, false));
                                                        } catch (J$e) {
                                                            J$.Ex(128377, J$e);
                                                        } finally {
                                                            if (J$.Fr(128385))
                                                                continue jalangiLabel27;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(128393, J$e);
                                        } finally {
                                            if (J$.Fr(128401))
                                                continue jalangiLabel28;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(128409, J$e);
                        } finally {
                            if (J$.Fr(128417))
                                continue jalangiLabel29;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(127729, '_typeof', J$.T(127721, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(127745, '_classCallCheck', J$.T(127737, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(127761, '_defineProperties', J$.T(127753, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(127777, '_createClass', J$.T(127769, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(127793, '_possibleConstructorReturn', J$.T(127785, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(127809, '_assertThisInitialized', J$.T(127801, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(127825, '_get', J$.T(127817, _get, 12, false), true, false, false);
            _superPropBase = J$.N(127841, '_superPropBase', J$.T(127833, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(127857, '_getPrototypeOf', J$.T(127849, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(127873, '_inherits', J$.T(127865, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(127889, '_setPrototypeOf', J$.T(127881, _setPrototypeOf, 12, false), true, false, false);
            J$.N(127897, 'AuthProvider', AuthProvider, false, false, false);
            J$.N(127905, 'retrieveKerberos', retrieveKerberos, false, false, false);
            J$.N(127913, 'kerberos', kerberos, false, false, false);
            J$.N(127921, 'SSPI', SSPI, false, false, false);
            SSIPAuthenticate = J$.N(127937, 'SSIPAuthenticate', J$.T(127929, SSIPAuthenticate, 12, false), true, false, false);
            J$.T(123209, 'use strict', 21, false);
            var AuthProvider = J$.W(125377, 'AuthProvider', J$.G(125369, J$.F(125361, J$.I(typeof require === 'undefined' ? require = J$.R(125345, 'require', undefined, true, true) : require = J$.R(125345, 'require', require, true, true)), false)(J$.T(125353, './auth_provider', 21, false)), 'AuthProvider'), AuthProvider, false, true);
            var retrieveKerberos = J$.W(125417, 'retrieveKerberos', J$.G(125409, J$.F(125401, J$.I(typeof require === 'undefined' ? require = J$.R(125385, 'require', undefined, true, true) : require = J$.R(125385, 'require', require, true, true)), false)(J$.T(125393, '../utils', 21, false)), 'retrieveKerberos'), retrieveKerberos, false, true);
            var kerberos;
            var SSPI = J$.W(126313, 'SSPI', J$.F(126305, J$.T(126289, function (_AuthProvider) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(126249, arguments.callee, this, arguments);
                            function SSPI() {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(125569, arguments.callee, this, arguments);
                                            arguments = J$.N(125577, 'arguments', arguments, true, false, false);
                                            J$.F(125481, J$.R(125457, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(125465, 'this', this, false, false), J$.R(125473, 'SSPI', SSPI, false, false));
                                            return J$.Rt(125561, J$.F(125553, J$.R(125489, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(125497, 'this', this, false, false), J$.M(125545, J$.F(125521, J$.R(125505, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(125513, 'SSPI', SSPI, false, false)), 'apply', false)(J$.R(125529, 'this', this, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(125537, 'arguments', undefined, true, true) : arguments = J$.R(125537, 'arguments', arguments, true, true)))));
                                        } catch (J$e) {
                                            J$.Ex(128201, J$e);
                                        } finally {
                                            if (J$.Fr(128209))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(126257, 'arguments', arguments, true, false, false);
                            _AuthProvider = J$.N(126265, '_AuthProvider', _AuthProvider, true, false, false);
                            SSPI = J$.N(126281, 'SSPI', J$.T(126273, SSPI, 12, false), true, false, false);
                            J$.F(125449, J$.R(125425, '_inherits', _inherits, false, true), false)(J$.R(125433, 'SSPI', SSPI, false, false), J$.R(125441, '_AuthProvider', _AuthProvider, false, false));
                            J$.F(126225, J$.R(125585, '_createClass', _createClass, false, true), false)(J$.R(125593, 'SSPI', SSPI, false, false), J$.T(126217, [
                                J$.T(125937, {
                                    key: J$.T(125601, '_authenticateSingleConnection', 21, false),
                                    value: J$.T(125929, function _authenticateSingleConnection(sendAuthCommand, connection, credentials, callback) {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(125849, arguments.callee, this, arguments);
                                                    arguments = J$.N(125857, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(125865, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connection = J$.N(125873, 'connection', connection, true, false, false);
                                                    credentials = J$.N(125881, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(125889, 'callback', callback, true, false, false);
                                                    J$.N(125897, 'username', username, false, false, false);
                                                    J$.N(125905, 'password', password, false, false, false);
                                                    J$.N(125913, 'mechanismProperties', mechanismProperties, false, false, false);
                                                    J$.N(125921, 'gssapiServiceName', gssapiServiceName, false, false, false);
                                                    var username = J$.W(125625, 'username', J$.G(125617, J$.R(125609, 'credentials', credentials, false, false), 'username'), username, false, false);
                                                    var password = J$.W(125649, 'password', J$.G(125641, J$.R(125633, 'credentials', credentials, false, false), 'password'), password, false, false);
                                                    var mechanismProperties = J$.W(125673, 'mechanismProperties', J$.G(125665, J$.R(125657, 'credentials', credentials, false, false), 'mechanismProperties'), mechanismProperties, false, false);
                                                    var gssapiServiceName = J$.W(125737, 'gssapiServiceName', J$.C(7184, J$.C(7176, J$.G(125697, J$.R(125681, 'mechanismProperties', mechanismProperties, false, false), J$.T(125689, 'gssapiservicename', 21, false))) ? J$._() : J$.G(125721, J$.R(125705, 'mechanismProperties', mechanismProperties, false, false), J$.T(125713, 'gssapiServiceName', 21, false))) ? J$._() : J$.T(125729, 'mongodb', 21, false), gssapiServiceName, false, false);
                                                    J$.F(125841, J$.R(125745, 'SSIPAuthenticate', SSIPAuthenticate, false, true), false)(J$.R(125753, 'this', this, false, false), J$.G(125777, J$.G(125769, J$.R(125761, 'kerberos', kerberos, false, true), 'processes'), 'MongoAuthProcess'), J$.R(125785, 'username', username, false, false), J$.R(125793, 'password', password, false, false), J$.R(125801, 'gssapiServiceName', gssapiServiceName, false, false), J$.R(125809, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(125817, 'connection', connection, false, false), J$.R(125825, 'mechanismProperties', mechanismProperties, false, false), J$.R(125833, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(128217, J$e);
                                                } finally {
                                                    if (J$.Fr(128225))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(126209, {
                                    key: J$.T(125945, 'auth', 21, false),
                                    value: J$.T(126201, function auth(sendAuthCommand, connections, credentials, callback) {
                                        jalangiLabel18:
                                            while (true) {
                                                try {
                                                    J$.Fe(126153, arguments.callee, this, arguments);
                                                    arguments = J$.N(126161, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(126169, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connections = J$.N(126177, 'connections', connections, true, false, false);
                                                    credentials = J$.N(126185, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(126193, 'callback', callback, true, false, false);
                                                    if (J$.C(7192, J$.B(6770, '==', J$.R(125953, 'kerberos', kerberos, false, true), J$.T(125961, null, 25, false)))) {
                                                        try {
                                                            kerberos = J$.W(125985, 'kerberos', J$.F(125977, J$.R(125969, 'retrieveKerberos', retrieveKerberos, false, true), false)(), kerberos, false, true);
                                                        } catch (e) {
                                                            J$.N(126033, 'e', e, false, false, true);
                                                            return J$.Rt(126025, J$.F(126017, J$.R(125993, 'callback', callback, false, false), false)(J$.R(126001, 'e', e, false, false), J$.T(126009, null, 25, false)));
                                                        }
                                                    }
                                                    J$.M(126145, J$.F(126097, J$.R(126041, '_get', _get, false, true), false)(J$.F(126073, J$.R(126049, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(126065, J$.R(126057, 'SSPI', SSPI, false, false), 'prototype')), J$.T(126081, 'auth', 21, false), J$.R(126089, 'this', this, false, false)), 'call', false)(J$.R(126105, 'this', this, false, false), J$.R(126113, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(126121, 'connections', connections, false, false), J$.R(126129, 'credentials', credentials, false, false), J$.R(126137, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(128233, J$e);
                                                } finally {
                                                    if (J$.Fr(128241))
                                                        continue jalangiLabel18;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(126241, J$.R(126233, 'SSPI', SSPI, false, false));
                        } catch (J$e) {
                            J$.Ex(128249, J$e);
                        } finally {
                            if (J$.Fr(128257))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(126297, 'AuthProvider', AuthProvider, false, true)), SSPI, false, true);
            J$.P(127705, J$.I(typeof module === 'undefined' ? module = J$.R(127689, 'module', undefined, true, true) : module = J$.R(127689, 'module', module, true, true)), 'exports', J$.R(127697, 'SSPI', SSPI, false, true));
        } catch (J$e) {
            J$.Ex(128425, J$e);
        } finally {
            if (J$.Sr(128433))
                continue jalangiLabel30;
            else
                break jalangiLabel30;
        }
    }
// JALANGI DO NOT INSTRUMENT
