// codemirror/src/edit/options.js
"use strict";

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.constructor");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.regexp.to-string");

require("core-js/modules/web.timers");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.constructor");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.regexp.to-string");

require("core-js/modules/web.timers");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.constructor");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.regexp.to-string");

require("core-js/modules/web.timers");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.constructor");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.regexp.to-string");

require("core-js/modules/web.timers");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.constructor");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.regexp.to-string");

require("core-js/modules/web.timers");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.constructor");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.regexp.to-string");

require("core-js/modules/web.timers");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.constructor");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.regexp.to-string");

require("core-js/modules/web.timers");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.constructor");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.regexp.to-string");

require("core-js/modules/web.timers");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.constructor");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.regexp.to-string");

require("core-js/modules/web.timers");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.regexp.constructor");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.regexp.to-string");

require("core-js/modules/web.timers");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.defineOptions = defineOptions;
exports.optionHandlers = exports.defaults = exports.Init = void 0;

var _focus = require("../display/focus.js");

var _gutters = require("../display/gutters.js");

var _mode_state = require("../display/mode_state.js");

var _scrollbars = require("../display/scrollbars.js");

var _selection = require("../display/selection.js");

var _view_tracking = require("../display/view_tracking.js");

var _keymap = require("../input/keymap.js");

var _line_data = require("../line/line_data.js");

var _pos = require("../line/pos.js");

var _spans = require("../line/spans.js");

var _position_measurement = require("../measurement/position_measurement.js");

var _changes = require("../model/changes.js");

var _browser = require("../util/browser.js");

var _dom = require("../util/dom.js");

var _event = require("../util/event.js");

var _utils = require("./utils.js");

var Init = {
  toString: function toString() {
    return "CodeMirror.Init";
  }
};
exports.Init = Init;
var defaults = {};
exports.defaults = defaults;
var optionHandlers = {};
exports.optionHandlers = optionHandlers;

function defineOptions(CodeMirror) {
  var optionHandlers = CodeMirror.optionHandlers;

  function option(name, deflt, handle, notOnInit) {
    CodeMirror.defaults[name] = deflt;
    if (handle) optionHandlers[name] = notOnInit ? function (cm, val, old) {
      if (old != Init) handle(cm, val, old);
    } : handle;
  }

  CodeMirror.defineOption = option; // Passed to option handlers when there is no old value.

  CodeMirror.Init = Init; // These two are, on init, called from the constructor because they
  // have to be initialized before the editor can start at all.

  option("value", "", function (cm, val) {
    return cm.setValue(val);
  }, true);
  option("mode", null, function (cm, val) {
    cm.doc.modeOption = val;
    (0, _mode_state.loadMode)(cm);
  }, true);
  option("indentUnit", 2, _mode_state.loadMode, true);
  option("indentWithTabs", false);
  option("smartIndent", true);
  option("tabSize", 4, function (cm) {
    (0, _mode_state.resetModeState)(cm);
    (0, _position_measurement.clearCaches)(cm);
    (0, _view_tracking.regChange)(cm);
  }, true);
  option("lineSeparator", null, function (cm, val) {
    cm.doc.lineSep = val;
    if (!val) return;
    var newBreaks = [],
        lineNo = cm.doc.first;
    cm.doc.iter(function (line) {
      for (var pos = 0;;) {
        var found = line.text.indexOf(val, pos);
        if (found == -1) break;
        pos = found + val.length;
        newBreaks.push((0, _pos.Pos)(lineNo, found));
      }

      lineNo++;
    });

    for (var i = newBreaks.length - 1; i >= 0; i--) {
      (0, _changes.replaceRange)(cm.doc, val, newBreaks[i], (0, _pos.Pos)(newBreaks[i].line, newBreaks[i].ch + val.length));
    }
  });
  option("specialChars", /[\u0000-\u001f\u007f-\u009f\u00ad\u061c\u200b-\u200f\u2028\u2029\ufeff\ufff9-\ufffc]/g, function (cm, val, old) {
    cm.state.specialChars = new RegExp(val.source + (val.test("\t") ? "" : "|\t"), "g");
    if (old != Init) cm.refresh();
  });
  option("specialCharPlaceholder", _line_data.defaultSpecialCharPlaceholder, function (cm) {
    return cm.refresh();
  }, true);
  option("electricChars", true);
  option("inputStyle", _browser.mobile ? "contenteditable" : "textarea", function () {
    throw new Error("inputStyle can not (yet) be changed in a running editor"); // FIXME
  }, true);
  option("spellcheck", false, function (cm, val) {
    return cm.getInputField().spellcheck = val;
  }, true);
  option("autocorrect", false, function (cm, val) {
    return cm.getInputField().autocorrect = val;
  }, true);
  option("autocapitalize", false, function (cm, val) {
    return cm.getInputField().autocapitalize = val;
  }, true);
  option("rtlMoveVisually", !_browser.windows);
  option("wholeLineUpdateBefore", true);
  option("theme", "default", function (cm) {
    (0, _utils.themeChanged)(cm);
    (0, _gutters.updateGutters)(cm);
  }, true);
  option("keyMap", "default", function (cm, val, old) {
    var next = (0, _keymap.getKeyMap)(val);
    var prev = old != Init && (0, _keymap.getKeyMap)(old);
    if (prev && prev.detach) prev.detach(cm, next);
    if (next.attach) next.attach(cm, prev || null);
  });
  option("extraKeys", null);
  option("configureMouse", null);
  option("lineWrapping", false, wrappingChanged, true);
  option("gutters", [], function (cm, val) {
    cm.display.gutterSpecs = (0, _gutters.getGutters)(val, cm.options.lineNumbers);
    (0, _gutters.updateGutters)(cm);
  }, true);
  option("fixedGutter", true, function (cm, val) {
    cm.display.gutters.style.left = val ? (0, _position_measurement.compensateForHScroll)(cm.display) + "px" : "0";
    cm.refresh();
  }, true);
  option("coverGutterNextToScrollbar", false, function (cm) {
    return (0, _scrollbars.updateScrollbars)(cm);
  }, true);
  option("scrollbarStyle", "native", function (cm) {
    (0, _scrollbars.initScrollbars)(cm);
    (0, _scrollbars.updateScrollbars)(cm);
    cm.display.scrollbars.setScrollTop(cm.doc.scrollTop);
    cm.display.scrollbars.setScrollLeft(cm.doc.scrollLeft);
  }, true);
  option("lineNumbers", false, function (cm, val) {
    cm.display.gutterSpecs = (0, _gutters.getGutters)(cm.options.gutters, val);
    (0, _gutters.updateGutters)(cm);
  }, true);
  option("firstLineNumber", 1, _gutters.updateGutters, true);
  option("lineNumberFormatter", function (integer) {
    return integer;
  }, _gutters.updateGutters, true);
  option("showCursorWhenSelecting", false, _selection.updateSelection, true);
  option("resetSelectionOnContextMenu", true);
  option("lineWiseCopyCut", true);
  option("pasteLinesPerSelection", true);
  option("selectionsMayTouch", false);
  option("readOnly", false, function (cm, val) {
    if (val == "nocursor") {
      (0, _focus.onBlur)(cm);
      cm.display.input.blur();
    }

    cm.display.input.readOnlyChanged(val);
  });
  option("disableInput", false, function (cm, val) {
    if (!val) cm.display.input.reset();
  }, true);
  option("dragDrop", true, dragDropChanged);
  option("allowDropFileTypes", null);
  option("cursorBlinkRate", 530);
  option("cursorScrollMargin", 0);
  option("cursorHeight", 1, _selection.updateSelection, true);
  option("singleCursorHeightPerLine", true, _selection.updateSelection, true);
  option("workTime", 100);
  option("workDelay", 100);
  option("flattenSpans", true, _mode_state.resetModeState, true);
  option("addModeClass", false, _mode_state.resetModeState, true);
  option("pollInterval", 100);
  option("undoDepth", 200, function (cm, val) {
    return cm.doc.history.undoDepth = val;
  });
  option("historyEventDelay", 1250);
  option("viewportMargin", 10, function (cm) {
    return cm.refresh();
  }, true);
  option("maxHighlightLength", 10000, _mode_state.resetModeState, true);
  option("moveInputWithCursor", true, function (cm, val) {
    if (!val) cm.display.input.resetPosition();
  });
  option("tabindex", null, function (cm, val) {
    return cm.display.input.getField().tabIndex = val || "";
  });
  option("autofocus", null);
  option("direction", "ltr", function (cm, val) {
    return cm.doc.setDirection(val);
  }, true);
  option("phrases", null);
}

function dragDropChanged(cm, value, old) {
  var wasOn = old && old != Init;

  if (!value != !wasOn) {
    var funcs = cm.display.dragFunctions;
    var toggle = value ? _event.on : _event.off;
    toggle(cm.display.scroller, "dragstart", funcs.start);
    toggle(cm.display.scroller, "dragenter", funcs.enter);
    toggle(cm.display.scroller, "dragover", funcs.over);
    toggle(cm.display.scroller, "dragleave", funcs.leave);
    toggle(cm.display.scroller, "drop", funcs.drop);
  }
}

function wrappingChanged(cm) {
  if (cm.options.lineWrapping) {
    (0, _dom.addClass)(cm.display.wrapper, "CodeMirror-wrap");
    cm.display.sizer.style.minWidth = "";
    cm.display.sizerWidth = null;
  } else {
    (0, _dom.rmClass)(cm.display.wrapper, "CodeMirror-wrap");
    (0, _spans.findMaxLine)(cm);
  }

  (0, _position_measurement.estimateLineHeights)(cm);
  (0, _view_tracking.regChange)(cm);
  (0, _position_measurement.clearCaches)(cm);
  setTimeout(function () {
    return (0, _scrollbars.updateScrollbars)(cm);
  }, 100);
}
