// common-shakeify/test/source-map/expected.js
"use strict";

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

(function () {
  function r(e, n, t) {
    function o(i, f) {
      if (!n[i]) {
        if (!e[i]) {
          var c = "function" == typeof require && require;
          if (!f && c) return c(i, !0);
          if (u) return u(i, !0);
          var a = new Error("Cannot find module '" + i + "'");
          throw a.code = "MODULE_NOT_FOUND", a;
        }

        var p = n[i] = {
          exports: {}
        };
        e[i][0].call(p.exports, function (r) {
          var n = e[i][1][r];
          return o(n || r);
        }, p, p.exports, r, e, n, t);
      }

      return n[i].exports;
    }

    for (var u = "function" == typeof require && require, i = 0; i < t.length; i++) {
      o(t[i]);
    }

    return o;
  }

  return r;
})()({
  1: [function (require, module, exports) {
    'use strict';

    var val = require('./dep').a;

    console.log(val);
  }, {
    "./dep": 2
  }],
  2: [function (require, module, exports) {
    'use strict';

    Object.defineProperty(exports, "__esModule", {
      value: true
    });
    var a = exports.a = something();
    var b =
    /* common-shake removed: exports.b = */
    whatever;

    function something() {
      return 'abc';
    }

    throw 'something';

    function whatever() {
      return 'xyz';
    }
  }, {}]
}, {}, [1]);
