// follow-redirects/index.js
J$.noInstrEval = false;
jalangiLabel20:
    while (true) {
        try {
            J$.Se(2347697, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestAnalytics/node_modules/follow-redirects/index.js');
            function RedirectableRequest(options, responseCallback) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(2343321, arguments.callee, this, arguments);
                            arguments = J$.N(2343329, 'arguments', arguments, true, false, false);
                            options = J$.N(2343337, 'options', options, true, false, false);
                            responseCallback = J$.N(2343345, 'responseCallback', responseCallback, true, false, false);
                            J$.N(2343353, 'self', self, false, false, false);
                            J$.N(2343361, 'searchPos', searchPos, false, false, false);
                            J$.M(2342713, J$.R(2342697, 'Writable', Writable, false, true), 'call', false)(J$.R(2342705, 'this', this, false, false));
                            J$.P(2342753, J$.R(2342721, 'options', options, false, false), 'headers', J$.C(110288, J$.G(2342737, J$.R(2342729, 'options', options, false, false), 'headers')) ? J$._() : J$.T(2342745, {}, 11, false));
                            J$.P(2342777, J$.R(2342761, 'this', this, false, false), '_options', J$.R(2342769, 'options', options, false, false));
                            J$.P(2342801, J$.R(2342785, 'this', this, false, false), '_redirectCount', J$.T(2342793, 0, 22, false));
                            J$.P(2342825, J$.R(2342809, 'this', this, false, false), '_redirects', J$.T(2342817, [], 10, false));
                            J$.P(2342849, J$.R(2342833, 'this', this, false, false), '_requestBodyLength', J$.T(2342841, 0, 22, false));
                            J$.P(2342873, J$.R(2342857, 'this', this, false, false), '_requestBodyBuffers', J$.T(2342865, [], 10, false));
                            if (J$.C(110304, J$.G(2342889, J$.R(2342881, 'options', options, false, false), 'host'))) {
                                if (J$.C(110296, J$.U(298802, '!', J$.G(2342905, J$.R(2342897, 'options', options, false, false), 'hostname')))) {
                                    J$.P(2342937, J$.R(2342913, 'options', options, false, false), 'hostname', J$.G(2342929, J$.R(2342921, 'options', options, false, false), 'host'));
                                }
                                J$.B(298810, 'delete', J$.R(2342945, 'options', options, false, false), 'host');
                            }
                            if (J$.C(110312, J$.R(2342953, 'responseCallback', responseCallback, false, false))) {
                                J$.M(2342985, J$.R(2342961, 'this', this, false, false), 'on', false)(J$.T(2342969, 'response', 21, false), J$.R(2342977, 'responseCallback', responseCallback, false, false));
                            }
                            var self = J$.W(2343001, 'self', J$.R(2342993, 'this', this, false, false), self, false, false);
                            J$.P(2343073, J$.R(2343009, 'this', this, false, false), '_onNativeResponse', J$.T(2343065, function (response) {
                                jalangiLabel2:
                                    while (true) {
                                        try {
                                            J$.Fe(2343041, arguments.callee, this, arguments);
                                            arguments = J$.N(2343049, 'arguments', arguments, true, false, false);
                                            response = J$.N(2343057, 'response', response, true, false, false);
                                            J$.M(2343033, J$.R(2343017, 'self', self, false, false), '_processResponse', false)(J$.R(2343025, 'response', response, false, false));
                                        } catch (J$e) {
                                            J$.Ex(2347833, J$e);
                                        } finally {
                                            if (J$.Fr(2347841))
                                                continue jalangiLabel2;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            if (J$.C(110336, J$.C(110320, J$.U(298818, '!', J$.G(2343089, J$.R(2343081, 'options', options, false, false), 'pathname'))) ? J$.G(2343105, J$.R(2343097, 'options', options, false, false), 'path') : J$._())) {
                                var searchPos = J$.W(2343145, 'searchPos', J$.M(2343137, J$.G(2343121, J$.R(2343113, 'options', options, false, false), 'path'), 'indexOf', false)(J$.T(2343129, '?', 21, false)), searchPos, false, false);
                                if (J$.C(110328, J$.B(298826, '<', J$.R(2343153, 'searchPos', searchPos, false, false), J$.T(2343161, 0, 22, false)))) {
                                    J$.P(2343193, J$.R(2343169, 'options', options, false, false), 'pathname', J$.G(2343185, J$.R(2343177, 'options', options, false, false), 'path'));
                                } else {
                                    J$.P(2343249, J$.R(2343201, 'options', options, false, false), 'pathname', J$.M(2343241, J$.G(2343217, J$.R(2343209, 'options', options, false, false), 'path'), 'substring', false)(J$.T(2343225, 0, 22, false), J$.R(2343233, 'searchPos', searchPos, false, false)));
                                    J$.P(2343297, J$.R(2343257, 'options', options, false, false), 'search', J$.M(2343289, J$.G(2343273, J$.R(2343265, 'options', options, false, false), 'path'), 'substring', false)(J$.R(2343281, 'searchPos', searchPos, false, false)));
                                }
                            }
                            J$.M(2343313, J$.R(2343305, 'this', this, false, false), '_performRequest', false)();
                        } catch (J$e) {
                            J$.Ex(2347849, J$e);
                        } finally {
                            if (J$.Fr(2347857))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function wrap(protocols) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(2347569, arguments.callee, this, arguments);
                            arguments = J$.N(2347577, 'arguments', arguments, true, false, false);
                            protocols = J$.N(2347585, 'protocols', protocols, true, false, false);
                            J$.N(2347593, 'exports', exports, false, false, false);
                            J$.N(2347601, 'nativeProtocols', nativeProtocols, false, false, false);
                            var exports = J$.W(2346809, 'exports', J$.T(2346801, {
                                maxRedirects: J$.T(2346769, 21, 22, false),
                                maxBodyLength: J$.B(299106, '*', J$.B(299098, '*', J$.T(2346777, 10, 22, false), J$.T(2346785, 1024, 22, false)), J$.T(2346793, 1024, 22, false))
                            }, 11, false), exports, false, false);
                            var nativeProtocols = J$.W(2346825, 'nativeProtocols', J$.T(2346817, {}, 11, false), nativeProtocols, false, false);
                            J$.M(2347545, J$.M(2346849, J$.I(typeof Object === 'undefined' ? Object = J$.R(2346833, 'Object', undefined, true, true) : Object = J$.R(2346833, 'Object', Object, true, true)), 'keys', false)(J$.R(2346841, 'protocols', protocols, false, false)), 'forEach', false)(J$.T(2347537, function (scheme) {
                                jalangiLabel18:
                                    while (true) {
                                        try {
                                            J$.Fe(2347489, arguments.callee, this, arguments);
                                            arguments = J$.N(2347497, 'arguments', arguments, true, false, false);
                                            scheme = J$.N(2347505, 'scheme', scheme, true, false, false);
                                            J$.N(2347513, 'protocol', protocol, false, false, false);
                                            J$.N(2347521, 'nativeProtocol', nativeProtocol, false, false, false);
                                            J$.N(2347529, 'wrappedProtocol', wrappedProtocol, false, false, false);
                                            var protocol = J$.W(2346873, 'protocol', J$.B(299114, '+', J$.R(2346857, 'scheme', scheme, false, false), J$.T(2346865, ':', 21, false)), protocol, false, false);
                                            var nativeProtocol = J$.W(2346929, 'nativeProtocol', J$.P(2346921, J$.R(2346881, 'nativeProtocols', nativeProtocols, false, false), J$.R(2346889, 'protocol', protocol, false, false), J$.G(2346913, J$.R(2346897, 'protocols', protocols, false, false), J$.R(2346905, 'scheme', scheme, false, false))), nativeProtocol, false, false);
                                            var wrappedProtocol = J$.W(2346985, 'wrappedProtocol', J$.P(2346977, J$.R(2346937, 'exports', exports, false, false), J$.R(2346945, 'scheme', scheme, false, false), J$.M(2346969, J$.I(typeof Object === 'undefined' ? Object = J$.R(2346953, 'Object', undefined, true, true) : Object = J$.R(2346953, 'Object', Object, true, true)), 'create', false)(J$.R(2346961, 'nativeProtocol', nativeProtocol, false, false))), wrappedProtocol, false, false);
                                            J$.P(2347345, J$.R(2346993, 'wrappedProtocol', wrappedProtocol, false, false), 'request', J$.T(2347337, function (options, callback) {
                                                jalangiLabel16:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(2347305, arguments.callee, this, arguments);
                                                            arguments = J$.N(2347313, 'arguments', arguments, true, false, false);
                                                            options = J$.N(2347321, 'options', options, true, false, false);
                                                            callback = J$.N(2347329, 'callback', callback, true, false, false);
                                                            if (J$.C(110552, J$.B(299130, '===', J$.U(299122, 'typeof', J$.R(2347001, 'options', options, false, false)), J$.T(2347009, 'string', 21, false)))) {
                                                                options = J$.W(2347041, 'options', J$.M(2347033, J$.R(2347017, 'url', url, false, true), 'parse', false)(J$.R(2347025, 'options', options, false, false)), options, false, false);
                                                                J$.P(2347073, J$.R(2347049, 'options', options, false, false), 'maxRedirects', J$.G(2347065, J$.R(2347057, 'exports', exports, false, false), 'maxRedirects'));
                                                            } else {
                                                                options = J$.W(2347153, 'options', J$.M(2347145, J$.I(typeof Object === 'undefined' ? Object = J$.R(2347081, 'Object', undefined, true, true) : Object = J$.R(2347081, 'Object', Object, true, true)), 'assign', false)(J$.T(2347129, {
                                                                    protocol: J$.R(2347089, 'protocol', protocol, false, false),
                                                                    maxRedirects: J$.G(2347105, J$.R(2347097, 'exports', exports, false, false), 'maxRedirects'),
                                                                    maxBodyLength: J$.G(2347121, J$.R(2347113, 'exports', exports, false, false), 'maxBodyLength')
                                                                }, 11, false), J$.R(2347137, 'options', options, false, false)), options, false, false);
                                                            }
                                                            J$.P(2347177, J$.R(2347161, 'options', options, false, false), 'nativeProtocols', J$.R(2347169, 'nativeProtocols', nativeProtocols, false, false));
                                                            J$.M(2347225, J$.R(2347185, 'assert', assert, false, true), 'equal', false)(J$.G(2347201, J$.R(2347193, 'options', options, false, false), 'protocol'), J$.R(2347209, 'protocol', protocol, false, false), J$.T(2347217, 'protocol mismatch', 21, false));
                                                            J$.F(2347257, J$.R(2347233, 'debug', debug, false, true), false)(J$.T(2347241, 'options', 21, false), J$.R(2347249, 'options', options, false, false));
                                                            return J$.Rt(2347297, J$.F(2347289, J$.R(2347265, 'RedirectableRequest', RedirectableRequest, false, true), true)(J$.R(2347273, 'options', options, false, false), J$.R(2347281, 'callback', callback, false, false)));
                                                        } catch (J$e) {
                                                            J$.Ex(2348057, J$e);
                                                        } finally {
                                                            if (J$.Fr(2348065))
                                                                continue jalangiLabel16;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                            J$.P(2347481, J$.R(2347353, 'wrappedProtocol', wrappedProtocol, false, false), 'get', J$.T(2347473, function (options, callback) {
                                                jalangiLabel17:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(2347433, arguments.callee, this, arguments);
                                                            arguments = J$.N(2347441, 'arguments', arguments, true, false, false);
                                                            options = J$.N(2347449, 'options', options, true, false, false);
                                                            callback = J$.N(2347457, 'callback', callback, true, false, false);
                                                            J$.N(2347465, 'request', request, false, false, false);
                                                            var request = J$.W(2347393, 'request', J$.M(2347385, J$.R(2347361, 'wrappedProtocol', wrappedProtocol, false, false), 'request', false)(J$.R(2347369, 'options', options, false, false), J$.R(2347377, 'callback', callback, false, false)), request, false, false);
                                                            J$.M(2347409, J$.R(2347401, 'request', request, false, false), 'end', false)();
                                                            return J$.Rt(2347425, J$.R(2347417, 'request', request, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(2348073, J$e);
                                                        } finally {
                                                            if (J$.Fr(2348081))
                                                                continue jalangiLabel17;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(2348089, J$e);
                                        } finally {
                                            if (J$.Fr(2348097))
                                                continue jalangiLabel18;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            return J$.Rt(2347561, J$.R(2347553, 'exports', exports, false, false));
                        } catch (J$e) {
                            J$.Ex(2348105, J$e);
                        } finally {
                            if (J$.Fr(2348113))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(2347705, 'url', url, false, false, false);
            J$.N(2347713, 'http', http, false, false, false);
            J$.N(2347721, 'https', https, false, false, false);
            J$.N(2347729, 'assert', assert, false, false, false);
            J$.N(2347737, 'Writable', Writable, false, false, false);
            J$.N(2347745, 'debug', debug, false, false, false);
            J$.N(2347753, 'SAFE_METHODS', SAFE_METHODS, false, false, false);
            J$.N(2347761, 'eventHandlers', eventHandlers, false, false, false);
            RedirectableRequest = J$.N(2347777, 'RedirectableRequest', J$.T(2347769, RedirectableRequest, 12, false), true, false, false);
            wrap = J$.N(2347793, 'wrap', J$.T(2347785, wrap, 12, false), true, false, false);
            var url = J$.W(2342241, 'url', J$.F(2342233, J$.I(typeof require === 'undefined' ? require = J$.R(2342217, 'require', undefined, true, true) : require = J$.R(2342217, 'require', require, true, true)), false)(J$.T(2342225, 'url', 21, false)), url, false, true);
            var http = J$.W(2342273, 'http', J$.F(2342265, J$.I(typeof require === 'undefined' ? require = J$.R(2342249, 'require', undefined, true, true) : require = J$.R(2342249, 'require', require, true, true)), false)(J$.T(2342257, 'http', 21, false)), http, false, true);
            var https = J$.W(2342305, 'https', J$.F(2342297, J$.I(typeof require === 'undefined' ? require = J$.R(2342281, 'require', undefined, true, true) : require = J$.R(2342281, 'require', require, true, true)), false)(J$.T(2342289, 'https', 21, false)), https, false, true);
            var assert = J$.W(2342337, 'assert', J$.F(2342329, J$.I(typeof require === 'undefined' ? require = J$.R(2342313, 'require', undefined, true, true) : require = J$.R(2342313, 'require', require, true, true)), false)(J$.T(2342321, 'assert', 21, false)), assert, false, true);
            var Writable = J$.W(2342377, 'Writable', J$.G(2342369, J$.F(2342361, J$.I(typeof require === 'undefined' ? require = J$.R(2342345, 'require', undefined, true, true) : require = J$.R(2342345, 'require', require, true, true)), false)(J$.T(2342353, 'stream', 21, false)), 'Writable'), Writable, false, true);
            var debug = J$.W(2342425, 'debug', J$.F(2342417, J$.F(2342401, J$.I(typeof require === 'undefined' ? require = J$.R(2342385, 'require', undefined, true, true) : require = J$.R(2342385, 'require', require, true, true)), false)(J$.T(2342393, 'debug', 21, false)), false)(J$.T(2342409, 'follow-redirects', 21, false)), debug, false, true);
            var SAFE_METHODS = J$.W(2342473, 'SAFE_METHODS', J$.T(2342465, {
                GET: J$.T(2342433, true, 23, false),
                HEAD: J$.T(2342441, true, 23, false),
                OPTIONS: J$.T(2342449, true, 23, false),
                TRACE: J$.T(2342457, true, 23, false)
            }, 11, false), SAFE_METHODS, false, true);
            var eventHandlers = J$.W(2342505, 'eventHandlers', J$.M(2342497, J$.I(typeof Object === 'undefined' ? Object = J$.R(2342481, 'Object', undefined, true, true) : Object = J$.R(2342481, 'Object', Object, true, true)), 'create', false)(J$.T(2342489, null, 25, false)), eventHandlers, false, true);
            J$.M(2342689, J$.T(2342553, [
                J$.T(2342513, 'abort', 21, false),
                J$.T(2342521, 'aborted', 21, false),
                J$.T(2342529, 'error', 21, false),
                J$.T(2342537, 'socket', 21, false),
                J$.T(2342545, 'timeout', 21, false)
            ], 10, false), 'forEach', false)(J$.T(2342681, function (event) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(2342657, arguments.callee, this, arguments);
                            arguments = J$.N(2342665, 'arguments', arguments, true, false, false);
                            event = J$.N(2342673, 'event', event, true, false, false);
                            J$.P(2342649, J$.R(2342561, 'eventHandlers', eventHandlers, false, true), J$.R(2342569, 'event', event, false, false), J$.T(2342641, function (arg) {
                                jalangiLabel0:
                                    while (true) {
                                        try {
                                            J$.Fe(2342617, arguments.callee, this, arguments);
                                            arguments = J$.N(2342625, 'arguments', arguments, true, false, false);
                                            arg = J$.N(2342633, 'arg', arg, true, false, false);
                                            J$.M(2342609, J$.G(2342585, J$.R(2342577, 'this', this, false, false), '_redirectable'), 'emit', false)(J$.R(2342593, 'event', event, false, false), J$.R(2342601, 'arg', arg, false, false));
                                        } catch (J$e) {
                                            J$.Ex(2347801, J$e);
                                        } finally {
                                            if (J$.Fr(2347809))
                                                continue jalangiLabel0;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(2347817, J$e);
                        } finally {
                            if (J$.Fr(2347825))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(2343409, J$.R(2343369, 'RedirectableRequest', RedirectableRequest, false, true), 'prototype', J$.M(2343401, J$.I(typeof Object === 'undefined' ? Object = J$.R(2343377, 'Object', undefined, true, true) : Object = J$.R(2343377, 'Object', Object, true, true)), 'create', false)(J$.G(2343393, J$.R(2343385, 'Writable', Writable, false, true), 'prototype')));
            J$.P(2343905, J$.G(2343425, J$.R(2343417, 'RedirectableRequest', RedirectableRequest, false, true), 'prototype'), 'write', J$.T(2343897, function (data, encoding, callback) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(2343857, arguments.callee, this, arguments);
                            arguments = J$.N(2343865, 'arguments', arguments, true, false, false);
                            data = J$.N(2343873, 'data', data, true, false, false);
                            encoding = J$.N(2343881, 'encoding', encoding, true, false, false);
                            callback = J$.N(2343889, 'callback', callback, true, false, false);
                            if (J$.C(110360, J$.U(298874, '!', J$.C(110352, J$.B(298842, '===', J$.U(298834, 'typeof', J$.R(2343433, 'data', data, false, false)), J$.T(2343441, 'string', 21, false))) ? J$._() : J$.C(110344, J$.B(298858, '===', J$.U(298850, 'typeof', J$.R(2343449, 'data', data, false, false)), J$.T(2343457, 'object', 21, false))) ? J$.B(298866, 'in', J$.T(2343465, 'length', 21, false), J$.R(2343473, 'data', data, false, false)) : J$._()))) {
                                throw J$.F(2343497, J$.I(typeof Error === 'undefined' ? Error = J$.R(2343481, 'Error', undefined, true, true) : Error = J$.R(2343481, 'Error', Error, true, true)), true)(J$.T(2343489, 'data should be a string, Buffer or Uint8Array', 21, false));
                            }
                            if (J$.C(110368, J$.B(298890, '===', J$.U(298882, 'typeof', J$.R(2343505, 'encoding', encoding, false, false)), J$.T(2343513, 'function', 21, false)))) {
                                callback = J$.W(2343529, 'callback', J$.R(2343521, 'encoding', encoding, false, false), callback, false, false);
                                encoding = J$.W(2343545, 'encoding', J$.T(2343537, null, 25, false), encoding, false, false);
                            }
                            if (J$.C(110384, J$.B(298898, '===', J$.G(2343561, J$.R(2343553, 'data', data, false, false), 'length'), J$.T(2343569, 0, 22, false)))) {
                                if (J$.C(110376, J$.R(2343577, 'callback', callback, false, false))) {
                                    J$.F(2343593, J$.R(2343585, 'callback', callback, false, false), false)();
                                }
                                return J$.Rt(2343601, undefined);
                            }
                            if (J$.C(110392, J$.B(298914, '<=', J$.B(298906, '+', J$.G(2343617, J$.R(2343609, 'this', this, false, false), '_requestBodyLength'), J$.G(2343633, J$.R(2343625, 'data', data, false, false), 'length')), J$.G(2343657, J$.G(2343649, J$.R(2343641, 'this', this, false, false), '_options'), 'maxBodyLength')))) {
                                J$.A(2343689, J$.R(2343665, 'this', this, false, false), '_requestBodyLength', '+')(J$.G(2343681, J$.R(2343673, 'data', data, false, false), 'length'));
                                J$.M(2343737, J$.G(2343705, J$.R(2343697, 'this', this, false, false), '_requestBodyBuffers'), 'push', false)(J$.T(2343729, {
                                    data: J$.R(2343713, 'data', data, false, false),
                                    encoding: J$.R(2343721, 'encoding', encoding, false, false)
                                }, 11, false));
                                J$.M(2343785, J$.G(2343753, J$.R(2343745, 'this', this, false, false), '_currentRequest'), 'write', false)(J$.R(2343761, 'data', data, false, false), J$.R(2343769, 'encoding', encoding, false, false), J$.R(2343777, 'callback', callback, false, false));
                            } else {
                                J$.M(2343833, J$.R(2343793, 'this', this, false, false), 'emit', false)(J$.T(2343801, 'error', 21, false), J$.F(2343825, J$.I(typeof Error === 'undefined' ? Error = J$.R(2343809, 'Error', undefined, true, true) : Error = J$.R(2343809, 'Error', Error, true, true)), true)(J$.T(2343817, 'Request body larger than maxBodyLength limit', 21, false)));
                                J$.M(2343849, J$.R(2343841, 'this', this, false, false), 'abort', false)();
                            }
                        } catch (J$e) {
                            J$.Ex(2347865, J$e);
                        } finally {
                            if (J$.Fr(2347873))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(2344217, J$.G(2343921, J$.R(2343913, 'RedirectableRequest', RedirectableRequest, false, true), 'prototype'), 'end', J$.T(2344209, function (data, encoding, callback) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(2344161, arguments.callee, this, arguments);
                            arguments = J$.N(2344169, 'arguments', arguments, true, false, false);
                            data = J$.N(2344177, 'data', data, true, false, false);
                            encoding = J$.N(2344185, 'encoding', encoding, true, false, false);
                            callback = J$.N(2344193, 'callback', callback, true, false, false);
                            J$.N(2344201, 'currentRequest', currentRequest, false, false, false);
                            if (J$.C(110408, J$.B(298930, '===', J$.U(298922, 'typeof', J$.R(2343929, 'data', data, false, false)), J$.T(2343937, 'function', 21, false)))) {
                                callback = J$.W(2343953, 'callback', J$.R(2343945, 'data', data, false, false), callback, false, false);
                                data = J$.W(2343977, 'data', encoding = J$.W(2343969, 'encoding', J$.T(2343961, null, 25, false), encoding, false, false), data, false, false);
                            } else if (J$.C(110400, J$.B(298946, '===', J$.U(298938, 'typeof', J$.R(2343985, 'encoding', encoding, false, false)), J$.T(2343993, 'function', 21, false)))) {
                                callback = J$.W(2344009, 'callback', J$.R(2344001, 'encoding', encoding, false, false), callback, false, false);
                                encoding = J$.W(2344025, 'encoding', J$.T(2344017, null, 25, false), encoding, false, false);
                            }
                            var currentRequest = J$.W(2344049, 'currentRequest', J$.G(2344041, J$.R(2344033, 'this', this, false, false), '_currentRequest'), currentRequest, false, false);
                            J$.M(2344153, J$.R(2344057, 'this', this, false, false), 'write', false)(J$.C(110416, J$.R(2344065, 'data', data, false, false)) ? J$._() : J$.T(2344073, '', 21, false), J$.R(2344081, 'encoding', encoding, false, false), J$.T(2344145, function () {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(2344129, arguments.callee, this, arguments);
                                            arguments = J$.N(2344137, 'arguments', arguments, true, false, false);
                                            J$.M(2344121, J$.R(2344089, 'currentRequest', currentRequest, false, false), 'end', false)(J$.T(2344097, null, 25, false), J$.T(2344105, null, 25, false), J$.R(2344113, 'callback', callback, false, false));
                                        } catch (J$e) {
                                            J$.Ex(2347881, J$e);
                                        } finally {
                                            if (J$.Fr(2347889))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(2347897, J$e);
                        } finally {
                            if (J$.Fr(2347905))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(2344369, J$.G(2344233, J$.R(2344225, 'RedirectableRequest', RedirectableRequest, false, true), 'prototype'), 'setHeader', J$.T(2344361, function (name, value) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(2344329, arguments.callee, this, arguments);
                            arguments = J$.N(2344337, 'arguments', arguments, true, false, false);
                            name = J$.N(2344345, 'name', name, true, false, false);
                            value = J$.N(2344353, 'value', value, true, false, false);
                            J$.P(2344281, J$.G(2344257, J$.G(2344249, J$.R(2344241, 'this', this, false, false), '_options'), 'headers'), J$.R(2344265, 'name', name, false, false), J$.R(2344273, 'value', value, false, false));
                            J$.M(2344321, J$.G(2344297, J$.R(2344289, 'this', this, false, false), '_currentRequest'), 'setHeader', false)(J$.R(2344305, 'name', name, false, false), J$.R(2344313, 'value', value, false, false));
                        } catch (J$e) {
                            J$.Ex(2347913, J$e);
                        } finally {
                            if (J$.Fr(2347921))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(2344489, J$.G(2344385, J$.R(2344377, 'RedirectableRequest', RedirectableRequest, false, true), 'prototype'), 'removeHeader', J$.T(2344481, function (name) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(2344457, arguments.callee, this, arguments);
                            arguments = J$.N(2344465, 'arguments', arguments, true, false, false);
                            name = J$.N(2344473, 'name', name, true, false, false);
                            J$.B(298954, 'delete', J$.G(2344409, J$.G(2344401, J$.R(2344393, 'this', this, false, false), '_options'), 'headers'), J$.R(2344417, 'name', name, false, false));
                            J$.M(2344449, J$.G(2344433, J$.R(2344425, 'this', this, false, false), '_currentRequest'), 'removeHeader', false)(J$.R(2344441, 'name', name, false, false));
                        } catch (J$e) {
                            J$.Ex(2347929, J$e);
                        } finally {
                            if (J$.Fr(2347937))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.M(2344713, J$.T(2344545, [
                J$.T(2344497, 'abort', 21, false),
                J$.T(2344505, 'flushHeaders', 21, false),
                J$.T(2344513, 'getHeader', 21, false),
                J$.T(2344521, 'setNoDelay', 21, false),
                J$.T(2344529, 'setSocketKeepAlive', 21, false),
                J$.T(2344537, 'setTimeout', 21, false)
            ], 10, false), 'forEach', false)(J$.T(2344705, function (method) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(2344681, arguments.callee, this, arguments);
                            arguments = J$.N(2344689, 'arguments', arguments, true, false, false);
                            method = J$.N(2344697, 'method', method, true, false, false);
                            J$.P(2344673, J$.G(2344561, J$.R(2344553, 'RedirectableRequest', RedirectableRequest, false, true), 'prototype'), J$.R(2344569, 'method', method, false, false), J$.T(2344665, function (a, b) {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(2344633, arguments.callee, this, arguments);
                                            arguments = J$.N(2344641, 'arguments', arguments, true, false, false);
                                            a = J$.N(2344649, 'a', a, true, false, false);
                                            b = J$.N(2344657, 'b', b, true, false, false);
                                            return J$.Rt(2344625, J$.M(2344617, J$.G(2344585, J$.R(2344577, 'this', this, false, false), '_currentRequest'), J$.R(2344593, 'method', method, false, false), false)(J$.R(2344601, 'a', a, false, false), J$.R(2344609, 'b', b, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(2347945, J$e);
                                        } finally {
                                            if (J$.Fr(2347953))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(2347961, J$e);
                        } finally {
                            if (J$.Fr(2347969))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.M(2344897, J$.T(2344745, [
                J$.T(2344721, 'aborted', 21, false),
                J$.T(2344729, 'connection', 21, false),
                J$.T(2344737, 'socket', 21, false)
            ], 10, false), 'forEach', false)(J$.T(2344889, function (property) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(2344865, arguments.callee, this, arguments);
                            arguments = J$.N(2344873, 'arguments', arguments, true, false, false);
                            property = J$.N(2344881, 'property', property, true, false, false);
                            J$.M(2344857, J$.I(typeof Object === 'undefined' ? Object = J$.R(2344753, 'Object', undefined, true, true) : Object = J$.R(2344753, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(2344769, J$.R(2344761, 'RedirectableRequest', RedirectableRequest, false, true), 'prototype'), J$.R(2344777, 'property', property, false, false), J$.T(2344849, {
                                get: J$.T(2344841, function () {
                                    jalangiLabel11:
                                        while (true) {
                                            try {
                                                J$.Fe(2344825, arguments.callee, this, arguments);
                                                arguments = J$.N(2344833, 'arguments', arguments, true, false, false);
                                                return J$.Rt(2344817, J$.G(2344809, J$.G(2344793, J$.R(2344785, 'this', this, false, false), '_currentRequest'), J$.R(2344801, 'property', property, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(2347977, J$e);
                                            } finally {
                                                if (J$.Fr(2347985))
                                                    continue jalangiLabel11;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(2347993, J$e);
                        } finally {
                            if (J$.Fr(2348001))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(2345753, J$.G(2344913, J$.R(2344905, 'RedirectableRequest', RedirectableRequest, false, true), 'prototype'), '_performRequest', J$.T(2345745, function () {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(2345673, arguments.callee, this, arguments);
                            arguments = J$.N(2345681, 'arguments', arguments, true, false, false);
                            J$.N(2345689, 'protocol', protocol, false, false, false);
                            J$.N(2345697, 'nativeProtocol', nativeProtocol, false, false, false);
                            J$.N(2345705, 'scheme', scheme, false, false, false);
                            J$.N(2345713, 'request', request, false, false, false);
                            J$.N(2345721, 'event', event, false, false, false);
                            J$.N(2345729, 'i', i, false, false, false);
                            J$.N(2345737, 'buffers', buffers, false, false, false);
                            var protocol = J$.W(2344945, 'protocol', J$.G(2344937, J$.G(2344929, J$.R(2344921, 'this', this, false, false), '_options'), 'protocol'), protocol, false, false);
                            var nativeProtocol = J$.W(2344993, 'nativeProtocol', J$.G(2344985, J$.G(2344969, J$.G(2344961, J$.R(2344953, 'this', this, false, false), '_options'), 'nativeProtocols'), J$.R(2344977, 'protocol', protocol, false, false)), nativeProtocol, false, false);
                            if (J$.C(110424, J$.U(298962, '!', J$.R(2345001, 'nativeProtocol', nativeProtocol, false, false)))) {
                                J$.M(2345057, J$.R(2345009, 'this', this, false, false), 'emit', false)(J$.T(2345017, 'error', 21, false), J$.F(2345049, J$.I(typeof Error === 'undefined' ? Error = J$.R(2345025, 'Error', undefined, true, true) : Error = J$.R(2345025, 'Error', Error, true, true)), true)(J$.B(298970, '+', J$.T(2345033, 'Unsupported protocol ', 21, false), J$.R(2345041, 'protocol', protocol, false, false))));
                                return J$.Rt(2345065, undefined);
                            }
                            if (J$.C(110432, J$.G(2345089, J$.G(2345081, J$.R(2345073, 'this', this, false, false), '_options'), 'agents'))) {
                                var scheme = J$.W(2345145, 'scheme', J$.M(2345137, J$.R(2345097, 'protocol', protocol, false, false), 'substr', false)(J$.T(2345105, 0, 22, false), J$.B(298978, '-', J$.G(2345121, J$.R(2345113, 'protocol', protocol, false, false), 'length'), J$.T(2345129, 1, 22, false))), scheme, false, false);
                                J$.P(2345209, J$.G(2345161, J$.R(2345153, 'this', this, false, false), '_options'), 'agent', J$.G(2345201, J$.G(2345185, J$.G(2345177, J$.R(2345169, 'this', this, false, false), '_options'), 'agents'), J$.R(2345193, 'scheme', scheme, false, false)));
                            }
                            var request = J$.W(2345281, 'request', J$.P(2345273, J$.R(2345217, 'this', this, false, false), '_currentRequest', J$.M(2345265, J$.R(2345225, 'nativeProtocol', nativeProtocol, false, false), 'request', false)(J$.G(2345241, J$.R(2345233, 'this', this, false, false), '_options'), J$.G(2345257, J$.R(2345249, 'this', this, false, false), '_onNativeResponse'))), request, false, false);
                            J$.P(2345329, J$.R(2345289, 'this', this, false, false), '_currentUrl', J$.M(2345321, J$.R(2345297, 'url', url, false, true), 'format', false)(J$.G(2345313, J$.R(2345305, 'this', this, false, false), '_options')));
                            J$.P(2345353, J$.R(2345337, 'request', request, false, false), '_redirectable', J$.R(2345345, 'this', this, false, false));
                            for (var event in J$.H(2345425, J$.R(2345361, 'eventHandlers', eventHandlers, false, true))) {
                                J$.N(2345433, 'event', event, false, true, false);
                                {
                                    {
                                        if (J$.C(110440, J$.R(2345369, 'event', event, false, false))) {
                                            J$.M(2345417, J$.R(2345377, 'request', request, false, false), 'on', false)(J$.R(2345385, 'event', event, false, false), J$.G(2345409, J$.R(2345393, 'eventHandlers', eventHandlers, false, true), J$.R(2345401, 'event', event, false, false)));
                                        }
                                    }
                                }
                            }
                            if (J$.C(110456, J$.G(2345449, J$.R(2345441, 'this', this, false, false), '_isRedirect'))) {
                                var i = J$.W(2345465, 'i', J$.T(2345457, 0, 22, false), i, false, false);
                                var buffers = J$.W(2345489, 'buffers', J$.G(2345481, J$.R(2345473, 'this', this, false, false), '_requestBodyBuffers'), buffers, false, false);
                                J$.F(2345665, J$.T(2345657, function writeNext() {
                                    jalangiLabel13:
                                        while (true) {
                                            try {
                                                J$.Fe(2345633, arguments.callee, this, arguments);
                                                arguments = J$.N(2345641, 'arguments', arguments, true, false, false);
                                                J$.N(2345649, 'buffer', buffer, false, false, false);
                                                if (J$.C(110448, J$.B(298986, '<', J$.R(2345497, 'i', i, false, false), J$.G(2345513, J$.R(2345505, 'buffers', buffers, false, false), 'length')))) {
                                                    var buffer = J$.W(2345553, 'buffer', J$.G(2345545, J$.R(2345521, 'buffers', buffers, false, false), J$.B(299010, '-', i = J$.W(2345537, 'i', J$.B(299002, '+', J$.U(298994, '+', J$.R(2345529, 'i', i, false, false)), 1), i, false, false), 1)), buffer, false, false);
                                                    J$.M(2345609, J$.R(2345561, 'request', request, false, false), 'write', false)(J$.G(2345577, J$.R(2345569, 'buffer', buffer, false, false), 'data'), J$.G(2345593, J$.R(2345585, 'buffer', buffer, false, false), 'encoding'), J$.R(2345601, 'writeNext', writeNext, false, false));
                                                } else {
                                                    J$.M(2345625, J$.R(2345617, 'request', request, false, false), 'end', false)();
                                                }
                                            } catch (J$e) {
                                                J$.Ex(2348009, J$e);
                                            } finally {
                                                if (J$.Fr(2348017))
                                                    continue jalangiLabel13;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), false)();
                            }
                        } catch (J$e) {
                            J$.Ex(2348025, J$e);
                        } finally {
                            if (J$.Fr(2348033))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(2346761, J$.G(2345769, J$.R(2345761, 'RedirectableRequest', RedirectableRequest, false, true), 'prototype'), '_processResponse', J$.T(2346753, function (response) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(2346697, arguments.callee, this, arguments);
                            arguments = J$.N(2346705, 'arguments', arguments, true, false, false);
                            response = J$.N(2346713, 'response', response, true, false, false);
                            J$.N(2346721, 'location', location, false, false, false);
                            J$.N(2346729, 'header', header, false, false, false);
                            J$.N(2346737, 'headers', headers, false, false, false);
                            J$.N(2346745, 'redirectUrl', redirectUrl, false, false, false);
                            if (J$.C(110464, J$.G(2345793, J$.G(2345785, J$.R(2345777, 'this', this, false, false), '_options'), 'trackRedirects'))) {
                                J$.M(2345873, J$.G(2345809, J$.R(2345801, 'this', this, false, false), '_redirects'), 'push', false)(J$.T(2345865, {
                                    url: J$.G(2345825, J$.R(2345817, 'this', this, false, false), '_currentUrl'),
                                    headers: J$.G(2345841, J$.R(2345833, 'response', response, false, false), 'headers'),
                                    statusCode: J$.G(2345857, J$.R(2345849, 'response', response, false, false), 'statusCode')
                                }, 11, false));
                            }
                            var location = J$.W(2345905, 'location', J$.G(2345897, J$.G(2345889, J$.R(2345881, 'response', response, false, false), 'headers'), 'location'), location, false, false);
                            if (J$.C(110544, J$.C(110488, J$.C(110480, J$.C(110472, J$.R(2345913, 'location', location, false, false)) ? J$.B(299018, '!==', J$.G(2345937, J$.G(2345929, J$.R(2345921, 'this', this, false, false), '_options'), 'followRedirects'), J$.T(2345945, false, 23, false)) : J$._()) ? J$.B(299026, '>=', J$.G(2345961, J$.R(2345953, 'response', response, false, false), 'statusCode'), J$.T(2345969, 300, 22, false)) : J$._()) ? J$.B(299034, '<', J$.G(2345985, J$.R(2345977, 'response', response, false, false), 'statusCode'), J$.T(2345993, 400, 22, false)) : J$._())) {
                                if (J$.C(110496, J$.B(299042, '>', J$.A(2346009, J$.R(2346001, 'this', this, false, false), '_redirectCount', '+')(1), J$.G(2346033, J$.G(2346025, J$.R(2346017, 'this', this, false, false), '_options'), 'maxRedirects')))) {
                                    J$.M(2346081, J$.R(2346041, 'this', this, false, false), 'emit', false)(J$.T(2346049, 'error', 21, false), J$.F(2346073, J$.I(typeof Error === 'undefined' ? Error = J$.R(2346057, 'Error', undefined, true, true) : Error = J$.R(2346057, 'Error', Error, true, true)), true)(J$.T(2346065, 'Max redirects exceeded.', 21, false)));
                                    return J$.Rt(2346089, undefined);
                                }
                                var header;
                                var headers = J$.W(2346121, 'headers', J$.G(2346113, J$.G(2346105, J$.R(2346097, 'this', this, false, false), '_options'), 'headers'), headers, false, false);
                                if (J$.C(110520, J$.C(110504, J$.B(299050, '!==', J$.G(2346137, J$.R(2346129, 'response', response, false, false), 'statusCode'), J$.T(2346145, 307, 22, false))) ? J$.U(299066, '!', J$.B(299058, 'in', J$.G(2346169, J$.G(2346161, J$.R(2346153, 'this', this, false, false), '_options'), 'method'), J$.R(2346177, 'SAFE_METHODS', SAFE_METHODS, false, true))) : J$._())) {
                                    J$.P(2346209, J$.G(2346193, J$.R(2346185, 'this', this, false, false), '_options'), 'method', J$.T(2346201, 'GET', 21, false));
                                    J$.P(2346233, J$.R(2346217, 'this', this, false, false), '_requestBodyBuffers', J$.T(2346225, [], 10, false));
                                    for (header in J$.H(2346289, J$.R(2346241, 'headers', headers, false, false))) {
                                        J$.N(2346297, 'header', header, false, true, false);
                                        {
                                            {
                                                if (J$.C(110512, J$.M(2346265, J$.T(2346249, /^content-/i, 14, false), 'test', false)(J$.R(2346257, 'header', header, false, false)))) {
                                                    J$.B(299074, 'delete', J$.R(2346273, 'headers', headers, false, false), J$.R(2346281, 'header', header, false, false));
                                                }
                                            }
                                        }
                                    }
                                }
                                if (J$.C(110536, J$.U(299082, '!', J$.G(2346313, J$.R(2346305, 'this', this, false, false), '_isRedirect')))) {
                                    for (header in J$.H(2346369, J$.R(2346321, 'headers', headers, false, false))) {
                                        J$.N(2346377, 'header', header, false, true, false);
                                        {
                                            {
                                                if (J$.C(110528, J$.M(2346345, J$.T(2346329, /^host$/i, 14, false), 'test', false)(J$.R(2346337, 'header', header, false, false)))) {
                                                    J$.B(299090, 'delete', J$.R(2346353, 'headers', headers, false, false), J$.R(2346361, 'header', header, false, false));
                                                }
                                            }
                                        }
                                    }
                                }
                                var redirectUrl = J$.W(2346425, 'redirectUrl', J$.M(2346417, J$.R(2346385, 'url', url, false, true), 'resolve', false)(J$.G(2346401, J$.R(2346393, 'this', this, false, false), '_currentUrl'), J$.R(2346409, 'location', location, false, false)), redirectUrl, false, false);
                                J$.F(2346457, J$.R(2346433, 'debug', debug, false, true), false)(J$.T(2346441, 'redirecting to', 21, false), J$.R(2346449, 'redirectUrl', redirectUrl, false, false));
                                J$.M(2346513, J$.I(typeof Object === 'undefined' ? Object = J$.R(2346465, 'Object', undefined, true, true) : Object = J$.R(2346465, 'Object', Object, true, true)), 'assign', false)(J$.G(2346481, J$.R(2346473, 'this', this, false, false), '_options'), J$.M(2346505, J$.R(2346489, 'url', url, false, true), 'parse', false)(J$.R(2346497, 'redirectUrl', redirectUrl, false, false)));
                                J$.P(2346537, J$.R(2346521, 'this', this, false, false), '_isRedirect', J$.T(2346529, true, 23, false));
                                J$.M(2346553, J$.R(2346545, 'this', this, false, false), '_performRequest', false)();
                                J$.M(2346569, J$.R(2346561, 'response', response, false, false), 'destroy', false)();
                            } else {
                                J$.P(2346601, J$.R(2346577, 'response', response, false, false), 'responseUrl', J$.G(2346593, J$.R(2346585, 'this', this, false, false), '_currentUrl'));
                                J$.P(2346633, J$.R(2346609, 'response', response, false, false), 'redirects', J$.G(2346625, J$.R(2346617, 'this', this, false, false), '_redirects'));
                                J$.M(2346665, J$.R(2346641, 'this', this, false, false), 'emit', false)(J$.T(2346649, 'response', 21, false), J$.R(2346657, 'response', response, false, false));
                                J$.P(2346689, J$.R(2346673, 'this', this, false, false), '_requestBodyBuffers', J$.T(2346681, [], 10, false));
                            }
                        } catch (J$e) {
                            J$.Ex(2348041, J$e);
                        } finally {
                            if (J$.Fr(2348049))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(2347657, J$.I(typeof module === 'undefined' ? module = J$.R(2347609, 'module', undefined, true, true) : module = J$.R(2347609, 'module', module, true, true)), 'exports', J$.F(2347649, J$.R(2347617, 'wrap', wrap, false, true), false)(J$.T(2347641, {
                http: J$.R(2347625, 'http', http, false, true),
                https: J$.R(2347633, 'https', https, false, true)
            }, 11, false)));
            J$.P(2347689, J$.G(2347673, J$.I(typeof module === 'undefined' ? module = J$.R(2347665, 'module', undefined, true, true) : module = J$.R(2347665, 'module', module, true, true)), 'exports'), 'wrap', J$.R(2347681, 'wrap', wrap, false, true));
        } catch (J$e) {
            J$.Ex(2348121, J$e);
        } finally {
            if (J$.Sr(2348129))
                continue jalangiLabel20;
            else
                break jalangiLabel20;
        }
    }
// JALANGI DO NOT INSTRUMENT
