// mongodb/lib/operations/add_user.js
J$.noInstrEval = false;
jalangiLabel21:
    while (true) {
        try {
            J$.Se(529713, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/operations/add_user.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(526001, arguments.callee, this, arguments);
                            arguments = J$.N(526009, 'arguments', arguments, true, false, false);
                            obj = J$.N(526017, 'obj', obj, true, false, false);
                            if (J$.C(33144, J$.C(33104, J$.B(32802, '===', J$.U(32794, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(525737, 'Symbol', undefined, true, true) : Symbol = J$.R(525737, 'Symbol', Symbol, true, true))), J$.T(525745, 'function', 21, false))) ? J$.B(32818, '===', J$.U(32810, 'typeof', J$.G(525761, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(525753, 'Symbol', undefined, true, true) : Symbol = J$.R(525753, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(525769, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(525825, '_typeof', J$.T(525817, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(525793, arguments.callee, this, arguments);
                                                arguments = J$.N(525801, 'arguments', arguments, true, false, false);
                                                obj = J$.N(525809, 'obj', obj, true, false, false);
                                                return J$.Rt(525785, J$.U(32826, 'typeof', J$.R(525777, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(529953, J$e);
                                            } finally {
                                                if (J$.Fr(529961))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(525961, '_typeof', J$.T(525953, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(525929, arguments.callee, this, arguments);
                                                arguments = J$.N(525937, 'arguments', arguments, true, false, false);
                                                obj = J$.N(525945, 'obj', obj, true, false, false);
                                                return J$.Rt(525921, J$.C(33136, J$.C(33128, J$.C(33120, J$.C(33112, J$.R(525833, 'obj', obj, false, false)) ? J$.B(32842, '===', J$.U(32834, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(525841, 'Symbol', undefined, true, true) : Symbol = J$.R(525841, 'Symbol', Symbol, true, true))), J$.T(525849, 'function', 21, false)) : J$._()) ? J$.B(32850, '===', J$.G(525865, J$.R(525857, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(525873, 'Symbol', undefined, true, true) : Symbol = J$.R(525873, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(32858, '!==', J$.R(525881, 'obj', obj, false, false), J$.G(525897, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(525889, 'Symbol', undefined, true, true) : Symbol = J$.R(525889, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(525905, 'symbol', 21, false) : J$.U(32866, 'typeof', J$.R(525913, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(529969, J$e);
                                            } finally {
                                                if (J$.Fr(529977))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(525993, J$.F(525985, J$.R(525969, '_typeof', _typeof, false, true), false)(J$.R(525977, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(529985, J$e);
                        } finally {
                            if (J$.Fr(529993))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(526065, arguments.callee, this, arguments);
                            arguments = J$.N(526073, 'arguments', arguments, true, false, false);
                            instance = J$.N(526081, 'instance', instance, true, false, false);
                            Constructor = J$.N(526089, 'Constructor', Constructor, true, false, false);
                            if (J$.C(33152, J$.U(32882, '!', J$.B(32874, 'instanceof', J$.R(526025, 'instance', instance, false, false), J$.R(526033, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(526057, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(526041, 'TypeError', undefined, true, true) : TypeError = J$.R(526041, 'TypeError', TypeError, true, true)), true)(J$.T(526049, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(530001, J$e);
                        } finally {
                            if (J$.Fr(530009))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(526337, arguments.callee, this, arguments);
                            arguments = J$.N(526345, 'arguments', arguments, true, false, false);
                            target = J$.N(526353, 'target', target, true, false, false);
                            props = J$.N(526361, 'props', props, true, false, false);
                            J$.N(526369, 'i', i, false, false, false);
                            J$.N(526377, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(526105, 'i', J$.T(526097, 0, 22, false), i, false, false); J$.C(33176, J$.B(32890, '<', J$.R(526113, 'i', i, false, false), J$.G(526129, J$.R(526121, 'props', props, false, false), 'length'))); J$.B(32914, '-', i = J$.W(526145, 'i', J$.B(32906, '+', J$.U(32898, '+', J$.R(526137, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(526177, 'descriptor', J$.G(526169, J$.R(526153, 'props', props, false, false), J$.R(526161, 'i', i, false, false)), descriptor, false, false);
                                J$.P(526217, J$.R(526185, 'descriptor', descriptor, false, false), 'enumerable', J$.C(33160, J$.G(526201, J$.R(526193, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(526209, false, 23, false));
                                J$.P(526241, J$.R(526225, 'descriptor', descriptor, false, false), 'configurable', J$.T(526233, true, 23, false));
                                if (J$.C(33168, J$.B(32922, 'in', J$.T(526249, 'value', 21, false), J$.R(526257, 'descriptor', descriptor, false, false))))
                                    J$.P(526281, J$.R(526265, 'descriptor', descriptor, false, false), 'writable', J$.T(526273, true, 23, false));
                                J$.M(526329, J$.I(typeof Object === 'undefined' ? Object = J$.R(526289, 'Object', undefined, true, true) : Object = J$.R(526289, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(526297, 'target', target, false, false), J$.G(526313, J$.R(526305, 'descriptor', descriptor, false, false), 'key'), J$.R(526321, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(530017, J$e);
                        } finally {
                            if (J$.Fr(530025))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(526489, arguments.callee, this, arguments);
                            arguments = J$.N(526497, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(526505, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(526513, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(526521, 'staticProps', staticProps, true, false, false);
                            if (J$.C(33184, J$.R(526385, 'protoProps', protoProps, false, false)))
                                J$.F(526425, J$.R(526393, '_defineProperties', _defineProperties, false, true), false)(J$.G(526409, J$.R(526401, 'Constructor', Constructor, false, false), 'prototype'), J$.R(526417, 'protoProps', protoProps, false, false));
                            if (J$.C(33192, J$.R(526433, 'staticProps', staticProps, false, false)))
                                J$.F(526465, J$.R(526441, '_defineProperties', _defineProperties, false, true), false)(J$.R(526449, 'Constructor', Constructor, false, false), J$.R(526457, 'staticProps', staticProps, false, false));
                            return J$.Rt(526481, J$.R(526473, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(530033, J$e);
                        } finally {
                            if (J$.Fr(530041))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(526633, arguments.callee, this, arguments);
                            arguments = J$.N(526641, 'arguments', arguments, true, false, false);
                            self = J$.N(526649, 'self', self, true, false, false);
                            call = J$.N(526657, 'call', call, true, false, false);
                            if (J$.C(33216, J$.C(33208, J$.R(526529, 'call', call, false, false)) ? J$.C(33200, J$.B(32930, '===', J$.F(526553, J$.R(526537, '_typeof', _typeof, false, true), false)(J$.R(526545, 'call', call, false, false)), J$.T(526561, 'object', 21, false))) ? J$._() : J$.B(32946, '===', J$.U(32938, 'typeof', J$.R(526569, 'call', call, false, false)), J$.T(526577, 'function', 21, false)) : J$._())) {
                                return J$.Rt(526593, J$.R(526585, 'call', call, false, false));
                            }
                            return J$.Rt(526625, J$.F(526617, J$.R(526601, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(526609, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(530049, J$e);
                        } finally {
                            if (J$.Fr(530057))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(526721, arguments.callee, this, arguments);
                            arguments = J$.N(526729, 'arguments', arguments, true, false, false);
                            self = J$.N(526737, 'self', self, true, false, false);
                            if (J$.C(33224, J$.B(32954, '===', J$.R(526665, 'self', self, false, false), void J$.T(526673, 0, 22, false)))) {
                                throw J$.F(526697, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(526681, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(526681, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(526689, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(526713, J$.R(526705, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(530065, J$e);
                        } finally {
                            if (J$.Fr(530073))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(527105, arguments.callee, this, arguments);
                            arguments = J$.N(527113, 'arguments', arguments, true, false, false);
                            target = J$.N(527121, 'target', target, true, false, false);
                            property = J$.N(527129, 'property', property, true, false, false);
                            receiver = J$.N(527137, 'receiver', receiver, true, false, false);
                            if (J$.C(33256, J$.C(33232, J$.B(32970, '!==', J$.U(32962, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(526745, 'Reflect', undefined, true, true) : Reflect = J$.R(526745, 'Reflect', Reflect, true, true))), J$.T(526753, 'undefined', 21, false))) ? J$.G(526769, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(526761, 'Reflect', undefined, true, true) : Reflect = J$.R(526761, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(526793, '_get', J$.G(526785, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(526777, 'Reflect', undefined, true, true) : Reflect = J$.R(526777, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(527041, '_get', J$.T(527033, function _get(target, property, receiver) {
                                    jalangiLabel8:
                                        while (true) {
                                            try {
                                                J$.Fe(526977, arguments.callee, this, arguments);
                                                arguments = J$.N(526985, 'arguments', arguments, true, false, false);
                                                target = J$.N(526993, 'target', target, true, false, false);
                                                property = J$.N(527001, 'property', property, true, false, false);
                                                receiver = J$.N(527009, 'receiver', receiver, true, false, false);
                                                J$.N(527017, 'base', base, false, false, false);
                                                J$.N(527025, 'desc', desc, false, false, false);
                                                var base = J$.W(526833, 'base', J$.F(526825, J$.R(526801, '_superPropBase', _superPropBase, false, true), false)(J$.R(526809, 'target', target, false, false), J$.R(526817, 'property', property, false, false)), base, false, false);
                                                if (J$.C(33240, J$.U(32978, '!', J$.R(526841, 'base', base, false, false))))
                                                    return J$.Rt(526849, undefined);
                                                var desc = J$.W(526889, 'desc', J$.M(526881, J$.I(typeof Object === 'undefined' ? Object = J$.R(526857, 'Object', undefined, true, true) : Object = J$.R(526857, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(526865, 'base', base, false, false), J$.R(526873, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(33248, J$.G(526905, J$.R(526897, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(526945, J$.M(526937, J$.G(526921, J$.R(526913, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(526929, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(526969, J$.G(526961, J$.R(526953, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(530081, J$e);
                                            } finally {
                                                if (J$.Fr(530089))
                                                    continue jalangiLabel8;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(527097, J$.F(527089, J$.R(527049, '_get', _get, false, true), false)(J$.R(527057, 'target', target, false, false), J$.R(527065, 'property', property, false, false), J$.C(33264, J$.R(527073, 'receiver', receiver, false, false)) ? J$._() : J$.R(527081, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(530097, J$e);
                        } finally {
                            if (J$.Fr(530105))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(527257, arguments.callee, this, arguments);
                            arguments = J$.N(527265, 'arguments', arguments, true, false, false);
                            object = J$.N(527273, 'object', object, true, false, false);
                            property = J$.N(527281, 'property', property, true, false, false);
                            while (J$.C(33280, J$.U(32986, '!', J$.M(527185, J$.G(527161, J$.G(527153, J$.I(typeof Object === 'undefined' ? Object = J$.R(527145, 'Object', undefined, true, true) : Object = J$.R(527145, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(527169, 'object', object, false, false), J$.R(527177, 'property', property, false, false))))) {
                                object = J$.W(527217, 'object', J$.F(527209, J$.R(527193, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(527201, 'object', object, false, false)), object, false, false);
                                if (J$.C(33272, J$.B(32994, '===', J$.R(527225, 'object', object, false, false), J$.T(527233, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(527249, J$.R(527241, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(530113, J$e);
                        } finally {
                            if (J$.Fr(530121))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(527441, arguments.callee, this, arguments);
                            arguments = J$.N(527449, 'arguments', arguments, true, false, false);
                            o = J$.N(527457, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(527401, '_getPrototypeOf', J$.C(33296, J$.G(527297, J$.I(typeof Object === 'undefined' ? Object = J$.R(527289, 'Object', undefined, true, true) : Object = J$.R(527289, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(527313, J$.I(typeof Object === 'undefined' ? Object = J$.R(527305, 'Object', undefined, true, true) : Object = J$.R(527305, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(527393, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(527369, arguments.callee, this, arguments);
                                            arguments = J$.N(527377, 'arguments', arguments, true, false, false);
                                            o = J$.N(527385, 'o', o, true, false, false);
                                            return J$.Rt(527361, J$.C(33288, J$.G(527329, J$.R(527321, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(527353, J$.I(typeof Object === 'undefined' ? Object = J$.R(527337, 'Object', undefined, true, true) : Object = J$.R(527337, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(527345, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(530129, J$e);
                                        } finally {
                                            if (J$.Fr(530137))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(527433, J$.F(527425, J$.R(527409, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(527417, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(530145, J$e);
                        } finally {
                            if (J$.Fr(530153))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(527657, arguments.callee, this, arguments);
                            arguments = J$.N(527665, 'arguments', arguments, true, false, false);
                            subClass = J$.N(527673, 'subClass', subClass, true, false, false);
                            superClass = J$.N(527681, 'superClass', superClass, true, false, false);
                            if (J$.C(33312, J$.C(33304, J$.B(33010, '!==', J$.U(33002, 'typeof', J$.R(527465, 'superClass', superClass, false, false)), J$.T(527473, 'function', 21, false))) ? J$.B(33018, '!==', J$.R(527481, 'superClass', superClass, false, false), J$.T(527489, null, 25, false)) : J$._())) {
                                throw J$.F(527513, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(527497, 'TypeError', undefined, true, true) : TypeError = J$.R(527497, 'TypeError', TypeError, true, true)), true)(J$.T(527505, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(527609, J$.R(527521, 'subClass', subClass, false, false), 'prototype', J$.M(527601, J$.I(typeof Object === 'undefined' ? Object = J$.R(527529, 'Object', undefined, true, true) : Object = J$.R(527529, 'Object', Object, true, true)), 'create', false)(J$.C(33320, J$.R(527537, 'superClass', superClass, false, false)) ? J$.G(527553, J$.R(527545, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(527593, {
                                constructor: J$.T(527585, {
                                    value: J$.R(527561, 'subClass', subClass, false, false),
                                    writable: J$.T(527569, true, 23, false),
                                    configurable: J$.T(527577, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(33328, J$.R(527617, 'superClass', superClass, false, false)))
                                J$.F(527649, J$.R(527625, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(527633, 'subClass', subClass, false, false), J$.R(527641, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(530161, J$e);
                        } finally {
                            if (J$.Fr(530169))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(527833, arguments.callee, this, arguments);
                            arguments = J$.N(527841, 'arguments', arguments, true, false, false);
                            o = J$.N(527849, 'o', o, true, false, false);
                            p = J$.N(527857, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(527785, '_setPrototypeOf', J$.C(33336, J$.G(527697, J$.I(typeof Object === 'undefined' ? Object = J$.R(527689, 'Object', undefined, true, true) : Object = J$.R(527689, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(527777, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(527745, arguments.callee, this, arguments);
                                            arguments = J$.N(527753, 'arguments', arguments, true, false, false);
                                            o = J$.N(527761, 'o', o, true, false, false);
                                            p = J$.N(527769, 'p', p, true, false, false);
                                            J$.P(527721, J$.R(527705, 'o', o, false, false), '__proto__', J$.R(527713, 'p', p, false, false));
                                            return J$.Rt(527737, J$.R(527729, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(530177, J$e);
                                        } finally {
                                            if (J$.Fr(530185))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(527825, J$.F(527817, J$.R(527793, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(527801, 'o', o, false, false), J$.R(527809, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(530193, J$e);
                        } finally {
                            if (J$.Fr(530201))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(529729, '_typeof', J$.T(529721, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(529745, '_classCallCheck', J$.T(529737, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(529761, '_defineProperties', J$.T(529753, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(529777, '_createClass', J$.T(529769, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(529793, '_possibleConstructorReturn', J$.T(529785, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(529809, '_assertThisInitialized', J$.T(529801, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(529825, '_get', J$.T(529817, _get, 12, false), true, false, false);
            _superPropBase = J$.N(529841, '_superPropBase', J$.T(529833, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(529857, '_getPrototypeOf', J$.T(529849, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(529873, '_inherits', J$.T(529865, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(529889, '_setPrototypeOf', J$.T(529881, _setPrototypeOf, 12, false), true, false, false);
            J$.N(529897, 'Aspect', Aspect, false, false, false);
            J$.N(529905, 'CommandOperation', CommandOperation, false, false, false);
            J$.N(529913, 'defineAspects', defineAspects, false, false, false);
            J$.N(529921, 'crypto', crypto, false, false, false);
            J$.N(529929, 'handleCallback', handleCallback, false, false, false);
            J$.N(529937, 'toError', toError, false, false, false);
            J$.N(529945, 'AddUserOperation', AddUserOperation, false, false, false);
            J$.T(525729, 'use strict', 21, false);
            var Aspect = J$.W(527897, 'Aspect', J$.G(527889, J$.F(527881, J$.I(typeof require === 'undefined' ? require = J$.R(527865, 'require', undefined, true, true) : require = J$.R(527865, 'require', require, true, true)), false)(J$.T(527873, './operation', 21, false)), 'Aspect'), Aspect, false, true);
            var CommandOperation = J$.W(527929, 'CommandOperation', J$.F(527921, J$.I(typeof require === 'undefined' ? require = J$.R(527905, 'require', undefined, true, true) : require = J$.R(527905, 'require', require, true, true)), false)(J$.T(527913, './command', 21, false)), CommandOperation, false, true);
            var defineAspects = J$.W(527969, 'defineAspects', J$.G(527961, J$.F(527953, J$.I(typeof require === 'undefined' ? require = J$.R(527937, 'require', undefined, true, true) : require = J$.R(527937, 'require', require, true, true)), false)(J$.T(527945, './operation', 21, false)), 'defineAspects'), defineAspects, false, true);
            var crypto = J$.W(528001, 'crypto', J$.F(527993, J$.I(typeof require === 'undefined' ? require = J$.R(527977, 'require', undefined, true, true) : require = J$.R(527977, 'require', require, true, true)), false)(J$.T(527985, 'crypto', 21, false)), crypto, false, true);
            var handleCallback = J$.W(528041, 'handleCallback', J$.G(528033, J$.F(528025, J$.I(typeof require === 'undefined' ? require = J$.R(528009, 'require', undefined, true, true) : require = J$.R(528009, 'require', require, true, true)), false)(J$.T(528017, '../utils', 21, false)), 'handleCallback'), handleCallback, false, true);
            var toError = J$.W(528081, 'toError', J$.G(528073, J$.F(528065, J$.I(typeof require === 'undefined' ? require = J$.R(528049, 'require', undefined, true, true) : require = J$.R(528049, 'require', require, true, true)), false)(J$.T(528057, '../utils', 21, false)), 'toError'), toError, false, true);
            var AddUserOperation = J$.W(529641, 'AddUserOperation', J$.F(529633, J$.T(529617, function (_CommandOperation) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(529577, arguments.callee, this, arguments);
                            function AddUserOperation(db, username, password, options) {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(528305, arguments.callee, this, arguments);
                                            arguments = J$.N(528313, 'arguments', arguments, true, false, false);
                                            db = J$.N(528321, 'db', db, true, false, false);
                                            username = J$.N(528329, 'username', username, true, false, false);
                                            password = J$.N(528337, 'password', password, true, false, false);
                                            options = J$.N(528345, 'options', options, true, false, false);
                                            J$.N(528353, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(528145, J$.R(528121, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(528129, 'this', this, false, false), J$.R(528137, 'AddUserOperation', AddUserOperation, false, false));
                                            _this = J$.W(528233, '_this', J$.F(528225, J$.R(528153, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(528161, 'this', this, false, false), J$.M(528217, J$.F(528185, J$.R(528169, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(528177, 'AddUserOperation', AddUserOperation, false, false)), 'call', false)(J$.R(528193, 'this', this, false, false), J$.R(528201, 'db', db, false, false), J$.R(528209, 'options', options, false, false))), _this, false, false);
                                            J$.P(528257, J$.R(528241, '_this', _this, false, false), 'username', J$.R(528249, 'username', username, false, false));
                                            J$.P(528281, J$.R(528265, '_this', _this, false, false), 'password', J$.R(528273, 'password', password, false, false));
                                            return J$.Rt(528297, J$.R(528289, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(530209, J$e);
                                        } finally {
                                            if (J$.Fr(530217))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(529585, 'arguments', arguments, true, false, false);
                            _CommandOperation = J$.N(529593, '_CommandOperation', _CommandOperation, true, false, false);
                            AddUserOperation = J$.N(529609, 'AddUserOperation', J$.T(529601, AddUserOperation, 12, false), true, false, false);
                            J$.F(528113, J$.R(528089, '_inherits', _inherits, false, true), false)(J$.R(528097, 'AddUserOperation', AddUserOperation, false, false), J$.R(528105, '_CommandOperation', _CommandOperation, false, false));
                            J$.F(529553, J$.R(528361, '_createClass', _createClass, false, true), false)(J$.R(528369, 'AddUserOperation', AddUserOperation, false, false), J$.T(529545, [
                                J$.T(529161, {
                                    key: J$.T(528377, '_buildCommand', 21, false),
                                    value: J$.T(529153, function _buildCommand() {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(529065, arguments.callee, this, arguments);
                                                    arguments = J$.N(529073, 'arguments', arguments, true, false, false);
                                                    J$.N(529081, 'db', db, false, false, false);
                                                    J$.N(529089, 'username', username, false, false, false);
                                                    J$.N(529097, 'password', password, false, false, false);
                                                    J$.N(529105, 'options', options, false, false, false);
                                                    J$.N(529113, 'roles', roles, false, false, false);
                                                    J$.N(529121, 'digestPassword', digestPassword, false, false, false);
                                                    J$.N(529129, 'userPassword', userPassword, false, false, false);
                                                    J$.N(529137, 'md5', md5, false, false, false);
                                                    J$.N(529145, 'command', command, false, false, false);
                                                    var db = J$.W(528401, 'db', J$.G(528393, J$.R(528385, 'this', this, false, false), 'db'), db, false, false);
                                                    var username = J$.W(528425, 'username', J$.G(528417, J$.R(528409, 'this', this, false, false), 'username'), username, false, false);
                                                    var password = J$.W(528449, 'password', J$.G(528441, J$.R(528433, 'this', this, false, false), 'password'), password, false, false);
                                                    var options = J$.W(528473, 'options', J$.G(528465, J$.R(528457, 'this', this, false, false), 'options'), options, false, false);
                                                    var roles = J$.W(528537, 'roles', J$.C(33344, J$.M(528505, J$.I(typeof Array === 'undefined' ? Array = J$.R(528481, 'Array', undefined, true, true) : Array = J$.R(528481, 'Array', Array, true, true)), 'isArray', false)(J$.G(528497, J$.R(528489, 'options', options, false, false), 'roles'))) ? J$.G(528521, J$.R(528513, 'options', options, false, false), 'roles') : J$.T(528529, [], 10, false), roles, false, false);
                                                    if (J$.C(33352, J$.B(33026, '===', J$.G(528553, J$.R(528545, 'roles', roles, false, false), 'length'), J$.T(528561, 0, 22, false)))) {
                                                        J$.M(528585, J$.I(typeof console === 'undefined' ? console = J$.R(528569, 'console', undefined, true, true) : console = J$.R(528569, 'console', console, true, true)), 'log', false)(J$.T(528577, 'Creating a user without roles is deprecated in MongoDB >= 2.6', 21, false));
                                                    }
                                                    if (J$.C(33384, J$.C(33368, J$.C(33360, J$.B(33034, '===', J$.M(528609, J$.G(528601, J$.R(528593, 'db', db, false, false), 'databaseName'), 'toLowerCase', false)(), J$.T(528617, 'admin', 21, false))) ? J$._() : J$.B(33042, '===', J$.G(528633, J$.R(528625, 'options', options, false, false), 'dbName'), J$.T(528641, 'admin', 21, false))) ? J$.U(33050, '!', J$.M(528673, J$.I(typeof Array === 'undefined' ? Array = J$.R(528649, 'Array', undefined, true, true) : Array = J$.R(528649, 'Array', Array, true, true)), 'isArray', false)(J$.G(528665, J$.R(528657, 'options', options, false, false), 'roles'))) : J$._())) {
                                                        roles = J$.W(528697, 'roles', J$.T(528689, [J$.T(528681, 'root', 21, false)], 10, false), roles, false, false);
                                                    } else if (J$.C(33376, J$.U(33058, '!', J$.M(528729, J$.I(typeof Array === 'undefined' ? Array = J$.R(528705, 'Array', undefined, true, true) : Array = J$.R(528705, 'Array', Array, true, true)), 'isArray', false)(J$.G(528721, J$.R(528713, 'options', options, false, false), 'roles'))))) {
                                                        roles = J$.W(528753, 'roles', J$.T(528745, [J$.T(528737, 'dbOwner', 21, false)], 10, false), roles, false, false);
                                                    }
                                                    var digestPassword = J$.W(528809, 'digestPassword', J$.B(33066, '>=', J$.G(528793, J$.M(528785, J$.G(528777, J$.G(528769, J$.R(528761, 'db', db, false, false), 's'), 'topology'), 'lastIsMaster', false)(), 'maxWireVersion'), J$.T(528801, 7, 22, false)), digestPassword, false, false);
                                                    var userPassword = J$.W(528825, 'userPassword', J$.R(528817, 'password', password, false, false), userPassword, false, false);
                                                    if (J$.C(33392, J$.U(33074, '!', J$.R(528833, 'digestPassword', digestPassword, false, false)))) {
                                                        var md5 = J$.W(528865, 'md5', J$.M(528857, J$.R(528841, 'crypto', crypto, false, true), 'createHash', false)(J$.T(528849, 'md5', 21, false)), md5, false, false);
                                                        J$.M(528905, J$.R(528873, 'md5', md5, false, false), 'update', false)(J$.B(33090, '+', J$.B(33082, '+', J$.R(528881, 'username', username, false, false), J$.T(528889, ':mongo:', 21, false)), J$.R(528897, 'password', password, false, false)));
                                                        userPassword = J$.W(528937, 'userPassword', J$.M(528929, J$.R(528913, 'md5', md5, false, false), 'digest', false)(J$.T(528921, 'hex', 21, false)), userPassword, false, false);
                                                    }
                                                    var command = J$.W(529001, 'command', J$.T(528993, {
                                                        createUser: J$.R(528945, 'username', username, false, false),
                                                        customData: J$.C(33400, J$.G(528961, J$.R(528953, 'options', options, false, false), 'customData')) ? J$._() : J$.T(528969, {}, 11, false),
                                                        roles: J$.R(528977, 'roles', roles, false, false),
                                                        digestPassword: J$.R(528985, 'digestPassword', digestPassword, false, false)
                                                    }, 11, false), command, false, false);
                                                    if (J$.C(33408, J$.B(33106, '===', J$.U(33098, 'typeof', J$.R(529009, 'password', password, false, false)), J$.T(529017, 'string', 21, false)))) {
                                                        J$.P(529041, J$.R(529025, 'command', command, false, false), 'pwd', J$.R(529033, 'userPassword', userPassword, false, false));
                                                    }
                                                    return J$.Rt(529057, J$.R(529049, 'command', command, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(530225, J$e);
                                                } finally {
                                                    if (J$.Fr(530233))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(529537, {
                                    key: J$.T(529169, 'execute', 21, false),
                                    value: J$.T(529529, function execute(callback) {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(529497, arguments.callee, this, arguments);
                                                    arguments = J$.N(529505, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(529513, 'callback', callback, true, false, false);
                                                    J$.N(529521, 'options', options, false, false, false);
                                                    var options = J$.W(529193, 'options', J$.G(529185, J$.R(529177, 'this', this, false, false), 'options'), options, false, false);
                                                    if (J$.C(33416, J$.B(33114, '!=', J$.G(529209, J$.R(529201, 'options', options, false, false), 'digestPassword'), J$.T(529217, null, 25, false)))) {
                                                        return J$.Rt(529265, J$.F(529257, J$.R(529225, 'callback', callback, false, false), false)(J$.F(529249, J$.R(529233, 'toError', toError, false, true), false)(J$.T(529241, 'The digestPassword option is not supported via add_user. Please use db.command(\'createUser\', ...) instead for this option.', 21, false))));
                                                    }
                                                    J$.M(529489, J$.F(529329, J$.R(529273, '_get', _get, false, true), false)(J$.F(529305, J$.R(529281, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(529297, J$.R(529289, 'AddUserOperation', AddUserOperation, false, false), 'prototype')), J$.T(529313, 'execute', 21, false), J$.R(529321, 'this', this, false, false)), 'call', false)(J$.R(529337, 'this', this, false, false), J$.T(529481, function (err, r) {
                                                        jalangiLabel18:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(529449, arguments.callee, this, arguments);
                                                                    arguments = J$.N(529457, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(529465, 'err', err, true, false, false);
                                                                    r = J$.N(529473, 'r', r, true, false, false);
                                                                    if (J$.C(33424, J$.U(33122, '!', J$.R(529345, 'err', err, false, false)))) {
                                                                        return J$.Rt(529393, J$.F(529385, J$.R(529353, 'handleCallback', handleCallback, false, true), false)(J$.R(529361, 'callback', callback, false, false), J$.R(529369, 'err', err, false, false), J$.R(529377, 'r', r, false, false)));
                                                                    }
                                                                    return J$.Rt(529441, J$.F(529433, J$.R(529401, 'handleCallback', handleCallback, false, true), false)(J$.R(529409, 'callback', callback, false, false), J$.R(529417, 'err', err, false, false), J$.T(529425, null, 25, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(530241, J$e);
                                                                } finally {
                                                                    if (J$.Fr(530249))
                                                                        continue jalangiLabel18;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(530257, J$e);
                                                } finally {
                                                    if (J$.Fr(530265))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(529569, J$.R(529561, 'AddUserOperation', AddUserOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(530273, J$e);
                        } finally {
                            if (J$.Fr(530281))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(529625, 'CommandOperation', CommandOperation, false, true)), AddUserOperation, false, true);
            J$.F(529681, J$.R(529649, 'defineAspects', defineAspects, false, true), false)(J$.R(529657, 'AddUserOperation', AddUserOperation, false, true), J$.G(529673, J$.R(529665, 'Aspect', Aspect, false, true), 'WRITE_OPERATION'));
            J$.P(529705, J$.I(typeof module === 'undefined' ? module = J$.R(529689, 'module', undefined, true, true) : module = J$.R(529689, 'module', module, true, true)), 'exports', J$.R(529697, 'AddUserOperation', AddUserOperation, false, true));
        } catch (J$e) {
            J$.Ex(530289, J$e);
        } finally {
            if (J$.Sr(530297))
                continue jalangiLabel21;
            else
                break jalangiLabel21;
        }
    }
// JALANGI DO NOT INSTRUMENT
