// mongodb/lib/operations/aggregate.js
J$.noInstrEval = false;
jalangiLabel21:
    while (true) {
        try {
            J$.Se(536169, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/operations/aggregate.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(532041, arguments.callee, this, arguments);
                            arguments = J$.N(532049, 'arguments', arguments, true, false, false);
                            obj = J$.N(532057, 'obj', obj, true, false, false);
                            if (J$.C(33560, J$.C(33520, J$.B(33258, '===', J$.U(33250, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(531777, 'Symbol', undefined, true, true) : Symbol = J$.R(531777, 'Symbol', Symbol, true, true))), J$.T(531785, 'function', 21, false))) ? J$.B(33274, '===', J$.U(33266, 'typeof', J$.G(531801, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(531793, 'Symbol', undefined, true, true) : Symbol = J$.R(531793, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(531809, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(531865, '_typeof', J$.T(531857, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(531833, arguments.callee, this, arguments);
                                                arguments = J$.N(531841, 'arguments', arguments, true, false, false);
                                                obj = J$.N(531849, 'obj', obj, true, false, false);
                                                return J$.Rt(531825, J$.U(33282, 'typeof', J$.R(531817, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(536425, J$e);
                                            } finally {
                                                if (J$.Fr(536433))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(532001, '_typeof', J$.T(531993, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(531969, arguments.callee, this, arguments);
                                                arguments = J$.N(531977, 'arguments', arguments, true, false, false);
                                                obj = J$.N(531985, 'obj', obj, true, false, false);
                                                return J$.Rt(531961, J$.C(33552, J$.C(33544, J$.C(33536, J$.C(33528, J$.R(531873, 'obj', obj, false, false)) ? J$.B(33298, '===', J$.U(33290, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(531881, 'Symbol', undefined, true, true) : Symbol = J$.R(531881, 'Symbol', Symbol, true, true))), J$.T(531889, 'function', 21, false)) : J$._()) ? J$.B(33306, '===', J$.G(531905, J$.R(531897, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(531913, 'Symbol', undefined, true, true) : Symbol = J$.R(531913, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(33314, '!==', J$.R(531921, 'obj', obj, false, false), J$.G(531937, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(531929, 'Symbol', undefined, true, true) : Symbol = J$.R(531929, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(531945, 'symbol', 21, false) : J$.U(33322, 'typeof', J$.R(531953, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(536441, J$e);
                                            } finally {
                                                if (J$.Fr(536449))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(532033, J$.F(532025, J$.R(532009, '_typeof', _typeof, false, true), false)(J$.R(532017, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(536457, J$e);
                        } finally {
                            if (J$.Fr(536465))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(532105, arguments.callee, this, arguments);
                            arguments = J$.N(532113, 'arguments', arguments, true, false, false);
                            instance = J$.N(532121, 'instance', instance, true, false, false);
                            Constructor = J$.N(532129, 'Constructor', Constructor, true, false, false);
                            if (J$.C(33568, J$.U(33338, '!', J$.B(33330, 'instanceof', J$.R(532065, 'instance', instance, false, false), J$.R(532073, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(532097, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(532081, 'TypeError', undefined, true, true) : TypeError = J$.R(532081, 'TypeError', TypeError, true, true)), true)(J$.T(532089, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(536473, J$e);
                        } finally {
                            if (J$.Fr(536481))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(532377, arguments.callee, this, arguments);
                            arguments = J$.N(532385, 'arguments', arguments, true, false, false);
                            target = J$.N(532393, 'target', target, true, false, false);
                            props = J$.N(532401, 'props', props, true, false, false);
                            J$.N(532409, 'i', i, false, false, false);
                            J$.N(532417, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(532145, 'i', J$.T(532137, 0, 22, false), i, false, false); J$.C(33592, J$.B(33346, '<', J$.R(532153, 'i', i, false, false), J$.G(532169, J$.R(532161, 'props', props, false, false), 'length'))); J$.B(33370, '-', i = J$.W(532185, 'i', J$.B(33362, '+', J$.U(33354, '+', J$.R(532177, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(532217, 'descriptor', J$.G(532209, J$.R(532193, 'props', props, false, false), J$.R(532201, 'i', i, false, false)), descriptor, false, false);
                                J$.P(532257, J$.R(532225, 'descriptor', descriptor, false, false), 'enumerable', J$.C(33576, J$.G(532241, J$.R(532233, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(532249, false, 23, false));
                                J$.P(532281, J$.R(532265, 'descriptor', descriptor, false, false), 'configurable', J$.T(532273, true, 23, false));
                                if (J$.C(33584, J$.B(33378, 'in', J$.T(532289, 'value', 21, false), J$.R(532297, 'descriptor', descriptor, false, false))))
                                    J$.P(532321, J$.R(532305, 'descriptor', descriptor, false, false), 'writable', J$.T(532313, true, 23, false));
                                J$.M(532369, J$.I(typeof Object === 'undefined' ? Object = J$.R(532329, 'Object', undefined, true, true) : Object = J$.R(532329, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(532337, 'target', target, false, false), J$.G(532353, J$.R(532345, 'descriptor', descriptor, false, false), 'key'), J$.R(532361, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(536489, J$e);
                        } finally {
                            if (J$.Fr(536497))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(532529, arguments.callee, this, arguments);
                            arguments = J$.N(532537, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(532545, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(532553, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(532561, 'staticProps', staticProps, true, false, false);
                            if (J$.C(33600, J$.R(532425, 'protoProps', protoProps, false, false)))
                                J$.F(532465, J$.R(532433, '_defineProperties', _defineProperties, false, true), false)(J$.G(532449, J$.R(532441, 'Constructor', Constructor, false, false), 'prototype'), J$.R(532457, 'protoProps', protoProps, false, false));
                            if (J$.C(33608, J$.R(532473, 'staticProps', staticProps, false, false)))
                                J$.F(532505, J$.R(532481, '_defineProperties', _defineProperties, false, true), false)(J$.R(532489, 'Constructor', Constructor, false, false), J$.R(532497, 'staticProps', staticProps, false, false));
                            return J$.Rt(532521, J$.R(532513, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(536505, J$e);
                        } finally {
                            if (J$.Fr(536513))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(532673, arguments.callee, this, arguments);
                            arguments = J$.N(532681, 'arguments', arguments, true, false, false);
                            self = J$.N(532689, 'self', self, true, false, false);
                            call = J$.N(532697, 'call', call, true, false, false);
                            if (J$.C(33632, J$.C(33624, J$.R(532569, 'call', call, false, false)) ? J$.C(33616, J$.B(33386, '===', J$.F(532593, J$.R(532577, '_typeof', _typeof, false, true), false)(J$.R(532585, 'call', call, false, false)), J$.T(532601, 'object', 21, false))) ? J$._() : J$.B(33402, '===', J$.U(33394, 'typeof', J$.R(532609, 'call', call, false, false)), J$.T(532617, 'function', 21, false)) : J$._())) {
                                return J$.Rt(532633, J$.R(532625, 'call', call, false, false));
                            }
                            return J$.Rt(532665, J$.F(532657, J$.R(532641, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(532649, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(536521, J$e);
                        } finally {
                            if (J$.Fr(536529))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(532761, arguments.callee, this, arguments);
                            arguments = J$.N(532769, 'arguments', arguments, true, false, false);
                            self = J$.N(532777, 'self', self, true, false, false);
                            if (J$.C(33640, J$.B(33410, '===', J$.R(532705, 'self', self, false, false), void J$.T(532713, 0, 22, false)))) {
                                throw J$.F(532737, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(532721, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(532721, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(532729, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(532753, J$.R(532745, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(536537, J$e);
                        } finally {
                            if (J$.Fr(536545))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(533145, arguments.callee, this, arguments);
                            arguments = J$.N(533153, 'arguments', arguments, true, false, false);
                            target = J$.N(533161, 'target', target, true, false, false);
                            property = J$.N(533169, 'property', property, true, false, false);
                            receiver = J$.N(533177, 'receiver', receiver, true, false, false);
                            if (J$.C(33672, J$.C(33648, J$.B(33426, '!==', J$.U(33418, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(532785, 'Reflect', undefined, true, true) : Reflect = J$.R(532785, 'Reflect', Reflect, true, true))), J$.T(532793, 'undefined', 21, false))) ? J$.G(532809, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(532801, 'Reflect', undefined, true, true) : Reflect = J$.R(532801, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(532833, '_get', J$.G(532825, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(532817, 'Reflect', undefined, true, true) : Reflect = J$.R(532817, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(533081, '_get', J$.T(533073, function _get(target, property, receiver) {
                                    jalangiLabel8:
                                        while (true) {
                                            try {
                                                J$.Fe(533017, arguments.callee, this, arguments);
                                                arguments = J$.N(533025, 'arguments', arguments, true, false, false);
                                                target = J$.N(533033, 'target', target, true, false, false);
                                                property = J$.N(533041, 'property', property, true, false, false);
                                                receiver = J$.N(533049, 'receiver', receiver, true, false, false);
                                                J$.N(533057, 'base', base, false, false, false);
                                                J$.N(533065, 'desc', desc, false, false, false);
                                                var base = J$.W(532873, 'base', J$.F(532865, J$.R(532841, '_superPropBase', _superPropBase, false, true), false)(J$.R(532849, 'target', target, false, false), J$.R(532857, 'property', property, false, false)), base, false, false);
                                                if (J$.C(33656, J$.U(33434, '!', J$.R(532881, 'base', base, false, false))))
                                                    return J$.Rt(532889, undefined);
                                                var desc = J$.W(532929, 'desc', J$.M(532921, J$.I(typeof Object === 'undefined' ? Object = J$.R(532897, 'Object', undefined, true, true) : Object = J$.R(532897, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(532905, 'base', base, false, false), J$.R(532913, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(33664, J$.G(532945, J$.R(532937, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(532985, J$.M(532977, J$.G(532961, J$.R(532953, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(532969, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(533009, J$.G(533001, J$.R(532993, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(536553, J$e);
                                            } finally {
                                                if (J$.Fr(536561))
                                                    continue jalangiLabel8;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(533137, J$.F(533129, J$.R(533089, '_get', _get, false, true), false)(J$.R(533097, 'target', target, false, false), J$.R(533105, 'property', property, false, false), J$.C(33680, J$.R(533113, 'receiver', receiver, false, false)) ? J$._() : J$.R(533121, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(536569, J$e);
                        } finally {
                            if (J$.Fr(536577))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(533297, arguments.callee, this, arguments);
                            arguments = J$.N(533305, 'arguments', arguments, true, false, false);
                            object = J$.N(533313, 'object', object, true, false, false);
                            property = J$.N(533321, 'property', property, true, false, false);
                            while (J$.C(33696, J$.U(33442, '!', J$.M(533225, J$.G(533201, J$.G(533193, J$.I(typeof Object === 'undefined' ? Object = J$.R(533185, 'Object', undefined, true, true) : Object = J$.R(533185, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(533209, 'object', object, false, false), J$.R(533217, 'property', property, false, false))))) {
                                object = J$.W(533257, 'object', J$.F(533249, J$.R(533233, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(533241, 'object', object, false, false)), object, false, false);
                                if (J$.C(33688, J$.B(33450, '===', J$.R(533265, 'object', object, false, false), J$.T(533273, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(533289, J$.R(533281, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(536585, J$e);
                        } finally {
                            if (J$.Fr(536593))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(533481, arguments.callee, this, arguments);
                            arguments = J$.N(533489, 'arguments', arguments, true, false, false);
                            o = J$.N(533497, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(533441, '_getPrototypeOf', J$.C(33712, J$.G(533337, J$.I(typeof Object === 'undefined' ? Object = J$.R(533329, 'Object', undefined, true, true) : Object = J$.R(533329, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(533353, J$.I(typeof Object === 'undefined' ? Object = J$.R(533345, 'Object', undefined, true, true) : Object = J$.R(533345, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(533433, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(533409, arguments.callee, this, arguments);
                                            arguments = J$.N(533417, 'arguments', arguments, true, false, false);
                                            o = J$.N(533425, 'o', o, true, false, false);
                                            return J$.Rt(533401, J$.C(33704, J$.G(533369, J$.R(533361, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(533393, J$.I(typeof Object === 'undefined' ? Object = J$.R(533377, 'Object', undefined, true, true) : Object = J$.R(533377, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(533385, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(536601, J$e);
                                        } finally {
                                            if (J$.Fr(536609))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(533473, J$.F(533465, J$.R(533449, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(533457, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(536617, J$e);
                        } finally {
                            if (J$.Fr(536625))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(533697, arguments.callee, this, arguments);
                            arguments = J$.N(533705, 'arguments', arguments, true, false, false);
                            subClass = J$.N(533713, 'subClass', subClass, true, false, false);
                            superClass = J$.N(533721, 'superClass', superClass, true, false, false);
                            if (J$.C(33728, J$.C(33720, J$.B(33466, '!==', J$.U(33458, 'typeof', J$.R(533505, 'superClass', superClass, false, false)), J$.T(533513, 'function', 21, false))) ? J$.B(33474, '!==', J$.R(533521, 'superClass', superClass, false, false), J$.T(533529, null, 25, false)) : J$._())) {
                                throw J$.F(533553, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(533537, 'TypeError', undefined, true, true) : TypeError = J$.R(533537, 'TypeError', TypeError, true, true)), true)(J$.T(533545, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(533649, J$.R(533561, 'subClass', subClass, false, false), 'prototype', J$.M(533641, J$.I(typeof Object === 'undefined' ? Object = J$.R(533569, 'Object', undefined, true, true) : Object = J$.R(533569, 'Object', Object, true, true)), 'create', false)(J$.C(33736, J$.R(533577, 'superClass', superClass, false, false)) ? J$.G(533593, J$.R(533585, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(533633, {
                                constructor: J$.T(533625, {
                                    value: J$.R(533601, 'subClass', subClass, false, false),
                                    writable: J$.T(533609, true, 23, false),
                                    configurable: J$.T(533617, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(33744, J$.R(533657, 'superClass', superClass, false, false)))
                                J$.F(533689, J$.R(533665, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(533673, 'subClass', subClass, false, false), J$.R(533681, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(536633, J$e);
                        } finally {
                            if (J$.Fr(536641))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(533873, arguments.callee, this, arguments);
                            arguments = J$.N(533881, 'arguments', arguments, true, false, false);
                            o = J$.N(533889, 'o', o, true, false, false);
                            p = J$.N(533897, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(533825, '_setPrototypeOf', J$.C(33752, J$.G(533737, J$.I(typeof Object === 'undefined' ? Object = J$.R(533729, 'Object', undefined, true, true) : Object = J$.R(533729, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(533817, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(533785, arguments.callee, this, arguments);
                                            arguments = J$.N(533793, 'arguments', arguments, true, false, false);
                                            o = J$.N(533801, 'o', o, true, false, false);
                                            p = J$.N(533809, 'p', p, true, false, false);
                                            J$.P(533761, J$.R(533745, 'o', o, false, false), '__proto__', J$.R(533753, 'p', p, false, false));
                                            return J$.Rt(533777, J$.R(533769, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(536649, J$e);
                                        } finally {
                                            if (J$.Fr(536657))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(533865, J$.F(533857, J$.R(533833, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(533841, 'o', o, false, false), J$.R(533849, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(536665, J$e);
                        } finally {
                            if (J$.Fr(536673))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(536185, '_typeof', J$.T(536177, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(536201, '_classCallCheck', J$.T(536193, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(536217, '_defineProperties', J$.T(536209, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(536233, '_createClass', J$.T(536225, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(536249, '_possibleConstructorReturn', J$.T(536241, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(536265, '_assertThisInitialized', J$.T(536257, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(536281, '_get', J$.T(536273, _get, 12, false), true, false, false);
            _superPropBase = J$.N(536297, '_superPropBase', J$.T(536289, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(536313, '_getPrototypeOf', J$.T(536305, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(536329, '_inherits', J$.T(536321, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(536345, '_setPrototypeOf', J$.T(536337, _setPrototypeOf, 12, false), true, false, false);
            J$.N(536353, 'CommandOperationV2', CommandOperationV2, false, false, false);
            J$.N(536361, 'MongoError', MongoError, false, false, false);
            J$.N(536369, 'maxWireVersion', maxWireVersion, false, false, false);
            J$.N(536377, 'ReadPreference', ReadPreference, false, false, false);
            J$.N(536385, 'Aspect', Aspect, false, false, false);
            J$.N(536393, 'defineAspects', defineAspects, false, false, false);
            J$.N(536401, 'DB_AGGREGATE_COLLECTION', DB_AGGREGATE_COLLECTION, false, false, false);
            J$.N(536409, 'MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT', MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT, false, false, false);
            J$.N(536417, 'AggregateOperation', AggregateOperation, false, false, false);
            J$.T(531769, 'use strict', 21, false);
            var CommandOperationV2 = J$.W(533929, 'CommandOperationV2', J$.F(533921, J$.I(typeof require === 'undefined' ? require = J$.R(533905, 'require', undefined, true, true) : require = J$.R(533905, 'require', require, true, true)), false)(J$.T(533913, './command_v2', 21, false)), CommandOperationV2, false, true);
            var MongoError = J$.W(533969, 'MongoError', J$.G(533961, J$.F(533953, J$.I(typeof require === 'undefined' ? require = J$.R(533937, 'require', undefined, true, true) : require = J$.R(533937, 'require', require, true, true)), false)(J$.T(533945, '../core', 21, false)), 'MongoError'), MongoError, false, true);
            var maxWireVersion = J$.W(534009, 'maxWireVersion', J$.G(534001, J$.F(533993, J$.I(typeof require === 'undefined' ? require = J$.R(533977, 'require', undefined, true, true) : require = J$.R(533977, 'require', require, true, true)), false)(J$.T(533985, '../core/utils', 21, false)), 'maxWireVersion'), maxWireVersion, false, true);
            var ReadPreference = J$.W(534049, 'ReadPreference', J$.G(534041, J$.F(534033, J$.I(typeof require === 'undefined' ? require = J$.R(534017, 'require', undefined, true, true) : require = J$.R(534017, 'require', require, true, true)), false)(J$.T(534025, '../core', 21, false)), 'ReadPreference'), ReadPreference, false, true);
            var Aspect = J$.W(534089, 'Aspect', J$.G(534081, J$.F(534073, J$.I(typeof require === 'undefined' ? require = J$.R(534057, 'require', undefined, true, true) : require = J$.R(534057, 'require', require, true, true)), false)(J$.T(534065, './operation', 21, false)), 'Aspect'), Aspect, false, true);
            var defineAspects = J$.W(534129, 'defineAspects', J$.G(534121, J$.F(534113, J$.I(typeof require === 'undefined' ? require = J$.R(534097, 'require', undefined, true, true) : require = J$.R(534097, 'require', require, true, true)), false)(J$.T(534105, './operation', 21, false)), 'defineAspects'), defineAspects, false, true);
            var DB_AGGREGATE_COLLECTION = J$.W(534145, 'DB_AGGREGATE_COLLECTION', J$.T(534137, 1, 22, false), DB_AGGREGATE_COLLECTION, false, true);
            var MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT = J$.W(534161, 'MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT', J$.T(534153, 8, 22, false), MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT, false, true);
            var AggregateOperation = J$.W(536057, 'AggregateOperation', J$.F(536049, J$.T(536033, function (_CommandOperationV) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(535993, arguments.callee, this, arguments);
                            function AggregateOperation(parent, pipeline, options) {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(534961, arguments.callee, this, arguments);
                                            arguments = J$.N(534969, 'arguments', arguments, true, false, false);
                                            parent = J$.N(534977, 'parent', parent, true, false, false);
                                            pipeline = J$.N(534985, 'pipeline', pipeline, true, false, false);
                                            options = J$.N(534993, 'options', options, true, false, false);
                                            J$.N(535001, '_this', _this, false, false, false);
                                            J$.N(535009, 'finalStage', finalStage, false, false, false);
                                            var _this;
                                            J$.F(534225, J$.R(534201, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(534209, 'this', this, false, false), J$.R(534217, 'AggregateOperation', AggregateOperation, false, false));
                                            _this = J$.W(534329, '_this', J$.F(534321, J$.R(534233, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(534241, 'this', this, false, false), J$.M(534313, J$.F(534265, J$.R(534249, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(534257, 'AggregateOperation', AggregateOperation, false, false)), 'call', false)(J$.R(534273, 'this', this, false, false), J$.R(534281, 'parent', parent, false, false), J$.R(534289, 'options', options, false, false), J$.T(534305, { fullResponse: J$.T(534297, true, 23, false) }, 11, false))), _this, false, false);
                                            J$.P(534441, J$.R(534337, '_this', _this, false, false), 'target', J$.C(33768, J$.C(33760, J$.G(534361, J$.G(534353, J$.R(534345, 'parent', parent, false, false), 's'), 'namespace')) ? J$.G(534393, J$.G(534385, J$.G(534377, J$.R(534369, 'parent', parent, false, false), 's'), 'namespace'), 'collection') : J$._()) ? J$.G(534425, J$.G(534417, J$.G(534409, J$.R(534401, 'parent', parent, false, false), 's'), 'namespace'), 'collection') : J$.R(534433, 'DB_AGGREGATE_COLLECTION', DB_AGGREGATE_COLLECTION, false, true));
                                            J$.P(534465, J$.R(534449, '_this', _this, false, false), 'pipeline', J$.R(534457, 'pipeline', pipeline, false, false));
                                            J$.P(534489, J$.R(534473, '_this', _this, false, false), 'hasWriteStage', J$.T(534481, false, 23, false));
                                            if (J$.C(33800, J$.B(33490, '===', J$.U(33482, 'typeof', J$.G(534505, J$.R(534497, 'options', options, false, false), 'out')), J$.T(534513, 'string', 21, false)))) {
                                                J$.P(534577, J$.R(534521, '_this', _this, false, false), 'pipeline', J$.M(534569, J$.G(534537, J$.R(534529, '_this', _this, false, false), 'pipeline'), 'concat', false)(J$.T(534561, { $out: J$.G(534553, J$.R(534545, 'options', options, false, false), 'out') }, 11, false)));
                                                J$.P(534601, J$.R(534585, '_this', _this, false, false), 'hasWriteStage', J$.T(534593, true, 23, false));
                                            } else if (J$.C(33792, J$.B(33498, '>', J$.G(534617, J$.R(534609, 'pipeline', pipeline, false, false), 'length'), J$.T(534625, 0, 22, false)))) {
                                                var finalStage = J$.W(534673, 'finalStage', J$.G(534665, J$.R(534633, 'pipeline', pipeline, false, false), J$.B(33506, '-', J$.G(534649, J$.R(534641, 'pipeline', pipeline, false, false), 'length'), J$.T(534657, 1, 22, false))), finalStage, false, false);
                                                if (J$.C(33784, J$.C(33776, J$.G(534689, J$.R(534681, 'finalStage', finalStage, false, false), '$out')) ? J$._() : J$.G(534705, J$.R(534697, 'finalStage', finalStage, false, false), '$merge'))) {
                                                    J$.P(534729, J$.R(534713, '_this', _this, false, false), 'hasWriteStage', J$.T(534721, true, 23, false));
                                                }
                                            }
                                            if (J$.C(33808, J$.G(534745, J$.R(534737, '_this', _this, false, false), 'hasWriteStage'))) {
                                                J$.P(534777, J$.R(534753, '_this', _this, false, false), 'readPreference', J$.G(534769, J$.R(534761, 'ReadPreference', ReadPreference, false, true), 'primary'));
                                            }
                                            if (J$.C(33832, J$.C(33824, J$.G(534793, J$.R(534785, 'options', options, false, false), 'explain')) ? J$.C(33816, J$.G(534809, J$.R(534801, '_this', _this, false, false), 'readConcern')) ? J$._() : J$.G(534825, J$.R(534817, '_this', _this, false, false), 'writeConcern') : J$._())) {
                                                throw J$.F(534849, J$.R(534833, 'MongoError', MongoError, false, true), true)(J$.T(534841, '"explain" cannot be used on an aggregate call with readConcern/writeConcern', 21, false));
                                            }
                                            if (J$.C(33848, J$.C(33840, J$.B(33514, '!=', J$.G(534865, J$.R(534857, 'options', options, false, false), 'cursor'), J$.T(534873, null, 25, false))) ? J$.B(33522, '!==', J$.F(534905, J$.R(534881, '_typeof', _typeof, false, true), false)(J$.G(534897, J$.R(534889, 'options', options, false, false), 'cursor')), J$.T(534913, 'object', 21, false)) : J$._())) {
                                                throw J$.F(534937, J$.R(534921, 'MongoError', MongoError, false, true), true)(J$.T(534929, 'cursor options must be an object', 21, false));
                                            }
                                            return J$.Rt(534953, J$.R(534945, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(536681, J$e);
                                        } finally {
                                            if (J$.Fr(536689))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(536001, 'arguments', arguments, true, false, false);
                            _CommandOperationV = J$.N(536009, '_CommandOperationV', _CommandOperationV, true, false, false);
                            AggregateOperation = J$.N(536025, 'AggregateOperation', J$.T(536017, AggregateOperation, 12, false), true, false, false);
                            J$.F(534193, J$.R(534169, '_inherits', _inherits, false, true), false)(J$.R(534177, 'AggregateOperation', AggregateOperation, false, false), J$.R(534185, '_CommandOperationV', _CommandOperationV, false, false));
                            J$.F(535969, J$.R(535017, '_createClass', _createClass, false, true), false)(J$.R(535025, 'AggregateOperation', AggregateOperation, false, false), J$.T(535961, [
                                J$.T(535105, {
                                    key: J$.T(535033, 'addToPipeline', 21, false),
                                    value: J$.T(535097, function addToPipeline(stage) {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(535073, arguments.callee, this, arguments);
                                                    arguments = J$.N(535081, 'arguments', arguments, true, false, false);
                                                    stage = J$.N(535089, 'stage', stage, true, false, false);
                                                    J$.M(535065, J$.G(535049, J$.R(535041, 'this', this, false, false), 'pipeline'), 'push', false)(J$.R(535057, 'stage', stage, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(536697, J$e);
                                                } finally {
                                                    if (J$.Fr(536705))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(535889, {
                                    key: J$.T(535113, 'execute', 21, false),
                                    value: J$.T(535881, function execute(server, callback) {
                                        jalangiLabel18:
                                            while (true) {
                                                try {
                                                    J$.Fe(535825, arguments.callee, this, arguments);
                                                    arguments = J$.N(535833, 'arguments', arguments, true, false, false);
                                                    server = J$.N(535841, 'server', server, true, false, false);
                                                    callback = J$.N(535849, 'callback', callback, true, false, false);
                                                    J$.N(535857, 'options', options, false, false, false);
                                                    J$.N(535865, 'serverWireVersion', serverWireVersion, false, false, false);
                                                    J$.N(535873, 'command', command, false, false, false);
                                                    var options = J$.W(535137, 'options', J$.G(535129, J$.R(535121, 'this', this, false, false), 'options'), options, false, false);
                                                    var serverWireVersion = J$.W(535169, 'serverWireVersion', J$.F(535161, J$.R(535145, 'maxWireVersion', maxWireVersion, false, true), false)(J$.R(535153, 'server', server, false, false)), serverWireVersion, false, false);
                                                    var command = J$.W(535217, 'command', J$.T(535209, {
                                                        aggregate: J$.G(535185, J$.R(535177, 'this', this, false, false), 'target'),
                                                        pipeline: J$.G(535201, J$.R(535193, 'this', this, false, false), 'pipeline')
                                                    }, 11, false), command, false, false);
                                                    if (J$.C(33864, J$.C(33856, J$.G(535233, J$.R(535225, 'this', this, false, false), 'hasWriteStage')) ? J$.B(33530, '<', J$.R(535241, 'serverWireVersion', serverWireVersion, false, false), J$.R(535249, 'MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT', MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT, false, true)) : J$._())) {
                                                        J$.P(535273, J$.R(535257, 'this', this, false, false), 'readConcern', J$.T(535265, null, 25, false));
                                                    }
                                                    if (J$.C(33888, J$.B(33538, '>=', J$.R(535281, 'serverWireVersion', serverWireVersion, false, false), J$.T(535289, 5, 22, false)))) {
                                                        if (J$.C(33880, J$.C(33872, J$.G(535305, J$.R(535297, 'this', this, false, false), 'hasWriteStage')) ? J$.G(535321, J$.R(535313, 'this', this, false, false), 'writeConcern') : J$._())) {
                                                            J$.M(535369, J$.I(typeof Object === 'undefined' ? Object = J$.R(535329, 'Object', undefined, true, true) : Object = J$.R(535329, 'Object', Object, true, true)), 'assign', false)(J$.R(535337, 'command', command, false, false), J$.T(535361, { writeConcern: J$.G(535353, J$.R(535345, 'this', this, false, false), 'writeConcern') }, 11, false));
                                                        }
                                                    }
                                                    if (J$.C(33896, J$.B(33546, '===', J$.G(535385, J$.R(535377, 'options', options, false, false), 'bypassDocumentValidation'), J$.T(535393, true, 23, false)))) {
                                                        J$.P(535425, J$.R(535401, 'command', command, false, false), 'bypassDocumentValidation', J$.G(535417, J$.R(535409, 'options', options, false, false), 'bypassDocumentValidation'));
                                                    }
                                                    if (J$.C(33904, J$.B(33562, '===', J$.U(33554, 'typeof', J$.G(535441, J$.R(535433, 'options', options, false, false), 'allowDiskUse')), J$.T(535449, 'boolean', 21, false)))) {
                                                        J$.P(535481, J$.R(535457, 'command', command, false, false), 'allowDiskUse', J$.G(535473, J$.R(535465, 'options', options, false, false), 'allowDiskUse'));
                                                    }
                                                    if (J$.C(33912, J$.G(535497, J$.R(535489, 'options', options, false, false), 'hint'))) {
                                                        J$.P(535529, J$.R(535505, 'command', command, false, false), 'hint', J$.G(535521, J$.R(535513, 'options', options, false, false), 'hint'));
                                                    }
                                                    if (J$.C(33920, J$.G(535545, J$.R(535537, 'options', options, false, false), 'explain'))) {
                                                        J$.P(535569, J$.R(535553, 'options', options, false, false), 'full', J$.T(535561, false, 23, false));
                                                        J$.P(535601, J$.R(535577, 'command', command, false, false), 'explain', J$.G(535593, J$.R(535585, 'options', options, false, false), 'explain'));
                                                    }
                                                    J$.P(535641, J$.R(535609, 'command', command, false, false), 'cursor', J$.C(33928, J$.G(535625, J$.R(535617, 'options', options, false, false), 'cursor')) ? J$._() : J$.T(535633, {}, 11, false));
                                                    if (J$.C(33944, J$.C(33936, J$.G(535657, J$.R(535649, 'options', options, false, false), 'batchSize')) ? J$.U(33570, '!', J$.G(535673, J$.R(535665, 'this', this, false, false), 'hasWriteStage')) : J$._())) {
                                                        J$.P(535713, J$.G(535689, J$.R(535681, 'command', command, false, false), 'cursor'), 'batchSize', J$.G(535705, J$.R(535697, 'options', options, false, false), 'batchSize'));
                                                    }
                                                    J$.M(535817, J$.F(535777, J$.R(535721, '_get', _get, false, true), false)(J$.F(535753, J$.R(535729, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(535745, J$.R(535737, 'AggregateOperation', AggregateOperation, false, false), 'prototype')), J$.T(535761, 'executeCommand', 21, false), J$.R(535769, 'this', this, false, false)), 'call', false)(J$.R(535785, 'this', this, false, false), J$.R(535793, 'server', server, false, false), J$.R(535801, 'command', command, false, false), J$.R(535809, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(536713, J$e);
                                                } finally {
                                                    if (J$.Fr(536721))
                                                        continue jalangiLabel18;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(535953, {
                                    key: J$.T(535897, 'canRetryRead', 21, false),
                                    get: J$.T(535945, function get() {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(535929, arguments.callee, this, arguments);
                                                    arguments = J$.N(535937, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(535921, J$.U(33578, '!', J$.G(535913, J$.R(535905, 'this', this, false, false), 'hasWriteStage')));
                                                } catch (J$e) {
                                                    J$.Ex(536729, J$e);
                                                } finally {
                                                    if (J$.Fr(536737))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(535985, J$.R(535977, 'AggregateOperation', AggregateOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(536745, J$e);
                        } finally {
                            if (J$.Fr(536753))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(536041, 'CommandOperationV2', CommandOperationV2, false, true)), AggregateOperation, false, true);
            J$.F(536137, J$.R(536065, 'defineAspects', defineAspects, false, true), false)(J$.R(536073, 'AggregateOperation', AggregateOperation, false, true), J$.T(536129, [
                J$.G(536089, J$.R(536081, 'Aspect', Aspect, false, true), 'READ_OPERATION'),
                J$.G(536105, J$.R(536097, 'Aspect', Aspect, false, true), 'RETRYABLE'),
                J$.G(536121, J$.R(536113, 'Aspect', Aspect, false, true), 'EXECUTE_WITH_SELECTION')
            ], 10, false));
            J$.P(536161, J$.I(typeof module === 'undefined' ? module = J$.R(536145, 'module', undefined, true, true) : module = J$.R(536145, 'module', module, true, true)), 'exports', J$.R(536153, 'AggregateOperation', AggregateOperation, false, true));
        } catch (J$e) {
            J$.Ex(536761, J$e);
        } finally {
            if (J$.Sr(536769))
                continue jalangiLabel21;
            else
                break jalangiLabel21;
        }
    }
// JALANGI DO NOT INSTRUMENT
