// mongodb/lib/operations/create_index.js
J$.noInstrEval = false;
jalangiLabel21:
    while (true) {
        try {
            J$.Se(605425, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/operations/create_index.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(601641, arguments.callee, this, arguments);
                            arguments = J$.N(601649, 'arguments', arguments, true, false, false);
                            obj = J$.N(601657, 'obj', obj, true, false, false);
                            if (J$.C(38312, J$.C(38272, J$.B(37266, '===', J$.U(37258, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(601377, 'Symbol', undefined, true, true) : Symbol = J$.R(601377, 'Symbol', Symbol, true, true))), J$.T(601385, 'function', 21, false))) ? J$.B(37282, '===', J$.U(37274, 'typeof', J$.G(601401, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(601393, 'Symbol', undefined, true, true) : Symbol = J$.R(601393, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(601409, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(601465, '_typeof', J$.T(601457, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(601433, arguments.callee, this, arguments);
                                                arguments = J$.N(601441, 'arguments', arguments, true, false, false);
                                                obj = J$.N(601449, 'obj', obj, true, false, false);
                                                return J$.Rt(601425, J$.U(37290, 'typeof', J$.R(601417, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(605673, J$e);
                                            } finally {
                                                if (J$.Fr(605681))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(601601, '_typeof', J$.T(601593, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(601569, arguments.callee, this, arguments);
                                                arguments = J$.N(601577, 'arguments', arguments, true, false, false);
                                                obj = J$.N(601585, 'obj', obj, true, false, false);
                                                return J$.Rt(601561, J$.C(38304, J$.C(38296, J$.C(38288, J$.C(38280, J$.R(601473, 'obj', obj, false, false)) ? J$.B(37306, '===', J$.U(37298, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(601481, 'Symbol', undefined, true, true) : Symbol = J$.R(601481, 'Symbol', Symbol, true, true))), J$.T(601489, 'function', 21, false)) : J$._()) ? J$.B(37314, '===', J$.G(601505, J$.R(601497, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(601513, 'Symbol', undefined, true, true) : Symbol = J$.R(601513, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(37322, '!==', J$.R(601521, 'obj', obj, false, false), J$.G(601537, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(601529, 'Symbol', undefined, true, true) : Symbol = J$.R(601529, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(601545, 'symbol', 21, false) : J$.U(37330, 'typeof', J$.R(601553, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(605689, J$e);
                                            } finally {
                                                if (J$.Fr(605697))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(601633, J$.F(601625, J$.R(601609, '_typeof', _typeof, false, true), false)(J$.R(601617, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(605705, J$e);
                        } finally {
                            if (J$.Fr(605713))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(601705, arguments.callee, this, arguments);
                            arguments = J$.N(601713, 'arguments', arguments, true, false, false);
                            instance = J$.N(601721, 'instance', instance, true, false, false);
                            Constructor = J$.N(601729, 'Constructor', Constructor, true, false, false);
                            if (J$.C(38320, J$.U(37346, '!', J$.B(37338, 'instanceof', J$.R(601665, 'instance', instance, false, false), J$.R(601673, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(601697, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(601681, 'TypeError', undefined, true, true) : TypeError = J$.R(601681, 'TypeError', TypeError, true, true)), true)(J$.T(601689, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(605721, J$e);
                        } finally {
                            if (J$.Fr(605729))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(601977, arguments.callee, this, arguments);
                            arguments = J$.N(601985, 'arguments', arguments, true, false, false);
                            target = J$.N(601993, 'target', target, true, false, false);
                            props = J$.N(602001, 'props', props, true, false, false);
                            J$.N(602009, 'i', i, false, false, false);
                            J$.N(602017, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(601745, 'i', J$.T(601737, 0, 22, false), i, false, false); J$.C(38344, J$.B(37354, '<', J$.R(601753, 'i', i, false, false), J$.G(601769, J$.R(601761, 'props', props, false, false), 'length'))); J$.B(37378, '-', i = J$.W(601785, 'i', J$.B(37370, '+', J$.U(37362, '+', J$.R(601777, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(601817, 'descriptor', J$.G(601809, J$.R(601793, 'props', props, false, false), J$.R(601801, 'i', i, false, false)), descriptor, false, false);
                                J$.P(601857, J$.R(601825, 'descriptor', descriptor, false, false), 'enumerable', J$.C(38328, J$.G(601841, J$.R(601833, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(601849, false, 23, false));
                                J$.P(601881, J$.R(601865, 'descriptor', descriptor, false, false), 'configurable', J$.T(601873, true, 23, false));
                                if (J$.C(38336, J$.B(37386, 'in', J$.T(601889, 'value', 21, false), J$.R(601897, 'descriptor', descriptor, false, false))))
                                    J$.P(601921, J$.R(601905, 'descriptor', descriptor, false, false), 'writable', J$.T(601913, true, 23, false));
                                J$.M(601969, J$.I(typeof Object === 'undefined' ? Object = J$.R(601929, 'Object', undefined, true, true) : Object = J$.R(601929, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(601937, 'target', target, false, false), J$.G(601953, J$.R(601945, 'descriptor', descriptor, false, false), 'key'), J$.R(601961, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(605737, J$e);
                        } finally {
                            if (J$.Fr(605745))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(602129, arguments.callee, this, arguments);
                            arguments = J$.N(602137, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(602145, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(602153, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(602161, 'staticProps', staticProps, true, false, false);
                            if (J$.C(38352, J$.R(602025, 'protoProps', protoProps, false, false)))
                                J$.F(602065, J$.R(602033, '_defineProperties', _defineProperties, false, true), false)(J$.G(602049, J$.R(602041, 'Constructor', Constructor, false, false), 'prototype'), J$.R(602057, 'protoProps', protoProps, false, false));
                            if (J$.C(38360, J$.R(602073, 'staticProps', staticProps, false, false)))
                                J$.F(602105, J$.R(602081, '_defineProperties', _defineProperties, false, true), false)(J$.R(602089, 'Constructor', Constructor, false, false), J$.R(602097, 'staticProps', staticProps, false, false));
                            return J$.Rt(602121, J$.R(602113, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(605753, J$e);
                        } finally {
                            if (J$.Fr(605761))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(602273, arguments.callee, this, arguments);
                            arguments = J$.N(602281, 'arguments', arguments, true, false, false);
                            self = J$.N(602289, 'self', self, true, false, false);
                            call = J$.N(602297, 'call', call, true, false, false);
                            if (J$.C(38384, J$.C(38376, J$.R(602169, 'call', call, false, false)) ? J$.C(38368, J$.B(37394, '===', J$.F(602193, J$.R(602177, '_typeof', _typeof, false, true), false)(J$.R(602185, 'call', call, false, false)), J$.T(602201, 'object', 21, false))) ? J$._() : J$.B(37410, '===', J$.U(37402, 'typeof', J$.R(602209, 'call', call, false, false)), J$.T(602217, 'function', 21, false)) : J$._())) {
                                return J$.Rt(602233, J$.R(602225, 'call', call, false, false));
                            }
                            return J$.Rt(602265, J$.F(602257, J$.R(602241, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(602249, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(605769, J$e);
                        } finally {
                            if (J$.Fr(605777))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(602361, arguments.callee, this, arguments);
                            arguments = J$.N(602369, 'arguments', arguments, true, false, false);
                            self = J$.N(602377, 'self', self, true, false, false);
                            if (J$.C(38392, J$.B(37418, '===', J$.R(602305, 'self', self, false, false), void J$.T(602313, 0, 22, false)))) {
                                throw J$.F(602337, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(602321, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(602321, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(602329, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(602353, J$.R(602345, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(605785, J$e);
                        } finally {
                            if (J$.Fr(605793))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(602745, arguments.callee, this, arguments);
                            arguments = J$.N(602753, 'arguments', arguments, true, false, false);
                            target = J$.N(602761, 'target', target, true, false, false);
                            property = J$.N(602769, 'property', property, true, false, false);
                            receiver = J$.N(602777, 'receiver', receiver, true, false, false);
                            if (J$.C(38424, J$.C(38400, J$.B(37434, '!==', J$.U(37426, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(602385, 'Reflect', undefined, true, true) : Reflect = J$.R(602385, 'Reflect', Reflect, true, true))), J$.T(602393, 'undefined', 21, false))) ? J$.G(602409, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(602401, 'Reflect', undefined, true, true) : Reflect = J$.R(602401, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(602433, '_get', J$.G(602425, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(602417, 'Reflect', undefined, true, true) : Reflect = J$.R(602417, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(602681, '_get', J$.T(602673, function _get(target, property, receiver) {
                                    jalangiLabel8:
                                        while (true) {
                                            try {
                                                J$.Fe(602617, arguments.callee, this, arguments);
                                                arguments = J$.N(602625, 'arguments', arguments, true, false, false);
                                                target = J$.N(602633, 'target', target, true, false, false);
                                                property = J$.N(602641, 'property', property, true, false, false);
                                                receiver = J$.N(602649, 'receiver', receiver, true, false, false);
                                                J$.N(602657, 'base', base, false, false, false);
                                                J$.N(602665, 'desc', desc, false, false, false);
                                                var base = J$.W(602473, 'base', J$.F(602465, J$.R(602441, '_superPropBase', _superPropBase, false, true), false)(J$.R(602449, 'target', target, false, false), J$.R(602457, 'property', property, false, false)), base, false, false);
                                                if (J$.C(38408, J$.U(37442, '!', J$.R(602481, 'base', base, false, false))))
                                                    return J$.Rt(602489, undefined);
                                                var desc = J$.W(602529, 'desc', J$.M(602521, J$.I(typeof Object === 'undefined' ? Object = J$.R(602497, 'Object', undefined, true, true) : Object = J$.R(602497, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(602505, 'base', base, false, false), J$.R(602513, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(38416, J$.G(602545, J$.R(602537, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(602585, J$.M(602577, J$.G(602561, J$.R(602553, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(602569, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(602609, J$.G(602601, J$.R(602593, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(605801, J$e);
                                            } finally {
                                                if (J$.Fr(605809))
                                                    continue jalangiLabel8;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(602737, J$.F(602729, J$.R(602689, '_get', _get, false, true), false)(J$.R(602697, 'target', target, false, false), J$.R(602705, 'property', property, false, false), J$.C(38432, J$.R(602713, 'receiver', receiver, false, false)) ? J$._() : J$.R(602721, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(605817, J$e);
                        } finally {
                            if (J$.Fr(605825))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(602897, arguments.callee, this, arguments);
                            arguments = J$.N(602905, 'arguments', arguments, true, false, false);
                            object = J$.N(602913, 'object', object, true, false, false);
                            property = J$.N(602921, 'property', property, true, false, false);
                            while (J$.C(38448, J$.U(37450, '!', J$.M(602825, J$.G(602801, J$.G(602793, J$.I(typeof Object === 'undefined' ? Object = J$.R(602785, 'Object', undefined, true, true) : Object = J$.R(602785, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(602809, 'object', object, false, false), J$.R(602817, 'property', property, false, false))))) {
                                object = J$.W(602857, 'object', J$.F(602849, J$.R(602833, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(602841, 'object', object, false, false)), object, false, false);
                                if (J$.C(38440, J$.B(37458, '===', J$.R(602865, 'object', object, false, false), J$.T(602873, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(602889, J$.R(602881, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(605833, J$e);
                        } finally {
                            if (J$.Fr(605841))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(603081, arguments.callee, this, arguments);
                            arguments = J$.N(603089, 'arguments', arguments, true, false, false);
                            o = J$.N(603097, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(603041, '_getPrototypeOf', J$.C(38464, J$.G(602937, J$.I(typeof Object === 'undefined' ? Object = J$.R(602929, 'Object', undefined, true, true) : Object = J$.R(602929, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(602953, J$.I(typeof Object === 'undefined' ? Object = J$.R(602945, 'Object', undefined, true, true) : Object = J$.R(602945, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(603033, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(603009, arguments.callee, this, arguments);
                                            arguments = J$.N(603017, 'arguments', arguments, true, false, false);
                                            o = J$.N(603025, 'o', o, true, false, false);
                                            return J$.Rt(603001, J$.C(38456, J$.G(602969, J$.R(602961, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(602993, J$.I(typeof Object === 'undefined' ? Object = J$.R(602977, 'Object', undefined, true, true) : Object = J$.R(602977, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(602985, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(605849, J$e);
                                        } finally {
                                            if (J$.Fr(605857))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(603073, J$.F(603065, J$.R(603049, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(603057, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(605865, J$e);
                        } finally {
                            if (J$.Fr(605873))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(603297, arguments.callee, this, arguments);
                            arguments = J$.N(603305, 'arguments', arguments, true, false, false);
                            subClass = J$.N(603313, 'subClass', subClass, true, false, false);
                            superClass = J$.N(603321, 'superClass', superClass, true, false, false);
                            if (J$.C(38480, J$.C(38472, J$.B(37474, '!==', J$.U(37466, 'typeof', J$.R(603105, 'superClass', superClass, false, false)), J$.T(603113, 'function', 21, false))) ? J$.B(37482, '!==', J$.R(603121, 'superClass', superClass, false, false), J$.T(603129, null, 25, false)) : J$._())) {
                                throw J$.F(603153, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(603137, 'TypeError', undefined, true, true) : TypeError = J$.R(603137, 'TypeError', TypeError, true, true)), true)(J$.T(603145, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(603249, J$.R(603161, 'subClass', subClass, false, false), 'prototype', J$.M(603241, J$.I(typeof Object === 'undefined' ? Object = J$.R(603169, 'Object', undefined, true, true) : Object = J$.R(603169, 'Object', Object, true, true)), 'create', false)(J$.C(38488, J$.R(603177, 'superClass', superClass, false, false)) ? J$.G(603193, J$.R(603185, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(603233, {
                                constructor: J$.T(603225, {
                                    value: J$.R(603201, 'subClass', subClass, false, false),
                                    writable: J$.T(603209, true, 23, false),
                                    configurable: J$.T(603217, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(38496, J$.R(603257, 'superClass', superClass, false, false)))
                                J$.F(603289, J$.R(603265, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(603273, 'subClass', subClass, false, false), J$.R(603281, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(605881, J$e);
                        } finally {
                            if (J$.Fr(605889))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(603473, arguments.callee, this, arguments);
                            arguments = J$.N(603481, 'arguments', arguments, true, false, false);
                            o = J$.N(603489, 'o', o, true, false, false);
                            p = J$.N(603497, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(603425, '_setPrototypeOf', J$.C(38504, J$.G(603337, J$.I(typeof Object === 'undefined' ? Object = J$.R(603329, 'Object', undefined, true, true) : Object = J$.R(603329, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(603417, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(603385, arguments.callee, this, arguments);
                                            arguments = J$.N(603393, 'arguments', arguments, true, false, false);
                                            o = J$.N(603401, 'o', o, true, false, false);
                                            p = J$.N(603409, 'p', p, true, false, false);
                                            J$.P(603361, J$.R(603345, 'o', o, false, false), '__proto__', J$.R(603353, 'p', p, false, false));
                                            return J$.Rt(603377, J$.R(603369, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(605897, J$e);
                                        } finally {
                                            if (J$.Fr(605905))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(603465, J$.F(603457, J$.R(603433, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(603441, 'o', o, false, false), J$.R(603449, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(605913, J$e);
                        } finally {
                            if (J$.Fr(605921))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(605441, '_typeof', J$.T(605433, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(605457, '_classCallCheck', J$.T(605449, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(605473, '_defineProperties', J$.T(605465, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(605489, '_createClass', J$.T(605481, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(605505, '_possibleConstructorReturn', J$.T(605497, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(605521, '_assertThisInitialized', J$.T(605513, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(605537, '_get', J$.T(605529, _get, 12, false), true, false, false);
            _superPropBase = J$.N(605553, '_superPropBase', J$.T(605545, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(605569, '_getPrototypeOf', J$.T(605561, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(605585, '_inherits', J$.T(605577, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(605601, '_setPrototypeOf', J$.T(605593, _setPrototypeOf, 12, false), true, false, false);
            J$.N(605609, 'Aspect', Aspect, false, false, false);
            J$.N(605617, 'CommandOperation', CommandOperation, false, false, false);
            J$.N(605625, 'defineAspects', defineAspects, false, false, false);
            J$.N(605633, 'handleCallback', handleCallback, false, false, false);
            J$.N(605641, 'MongoError', MongoError, false, false, false);
            J$.N(605649, 'parseIndexOptions', parseIndexOptions, false, false, false);
            J$.N(605657, 'keysToOmit', keysToOmit, false, false, false);
            J$.N(605665, 'CreateIndexOperation', CreateIndexOperation, false, false, false);
            J$.T(601369, 'use strict', 21, false);
            var Aspect = J$.W(603537, 'Aspect', J$.G(603529, J$.F(603521, J$.I(typeof require === 'undefined' ? require = J$.R(603505, 'require', undefined, true, true) : require = J$.R(603505, 'require', require, true, true)), false)(J$.T(603513, './operation', 21, false)), 'Aspect'), Aspect, false, true);
            var CommandOperation = J$.W(603569, 'CommandOperation', J$.F(603561, J$.I(typeof require === 'undefined' ? require = J$.R(603545, 'require', undefined, true, true) : require = J$.R(603545, 'require', require, true, true)), false)(J$.T(603553, './command', 21, false)), CommandOperation, false, true);
            var defineAspects = J$.W(603609, 'defineAspects', J$.G(603601, J$.F(603593, J$.I(typeof require === 'undefined' ? require = J$.R(603577, 'require', undefined, true, true) : require = J$.R(603577, 'require', require, true, true)), false)(J$.T(603585, './operation', 21, false)), 'defineAspects'), defineAspects, false, true);
            var handleCallback = J$.W(603649, 'handleCallback', J$.G(603641, J$.F(603633, J$.I(typeof require === 'undefined' ? require = J$.R(603617, 'require', undefined, true, true) : require = J$.R(603617, 'require', require, true, true)), false)(J$.T(603625, '../utils', 21, false)), 'handleCallback'), handleCallback, false, true);
            var MongoError = J$.W(603689, 'MongoError', J$.G(603681, J$.F(603673, J$.I(typeof require === 'undefined' ? require = J$.R(603657, 'require', undefined, true, true) : require = J$.R(603657, 'require', require, true, true)), false)(J$.T(603665, '../core', 21, false)), 'MongoError'), MongoError, false, true);
            var parseIndexOptions = J$.W(603729, 'parseIndexOptions', J$.G(603721, J$.F(603713, J$.I(typeof require === 'undefined' ? require = J$.R(603697, 'require', undefined, true, true) : require = J$.R(603697, 'require', require, true, true)), false)(J$.T(603705, '../utils', 21, false)), 'parseIndexOptions'), parseIndexOptions, false, true);
            var keysToOmit = J$.W(603833, 'keysToOmit', J$.F(603825, J$.I(typeof Set === 'undefined' ? Set = J$.R(603737, 'Set', undefined, true, true) : Set = J$.R(603737, 'Set', Set, true, true)), true)(J$.T(603817, [
                J$.T(603745, 'name', 21, false),
                J$.T(603753, 'key', 21, false),
                J$.T(603761, 'writeConcern', 21, false),
                J$.T(603769, 'w', 21, false),
                J$.T(603777, 'wtimeout', 21, false),
                J$.T(603785, 'j', 21, false),
                J$.T(603793, 'fsync', 21, false),
                J$.T(603801, 'readPreference', 21, false),
                J$.T(603809, 'session', 21, false)
            ], 10, false)), keysToOmit, false, true);
            var CreateIndexOperation = J$.W(605353, 'CreateIndexOperation', J$.F(605345, J$.T(605329, function (_CommandOperation) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(605289, arguments.callee, this, arguments);
                            function CreateIndexOperation(db, name, fieldOrSpec, options) {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(604217, arguments.callee, this, arguments);
                                            arguments = J$.N(604225, 'arguments', arguments, true, false, false);
                                            db = J$.N(604233, 'db', db, true, false, false);
                                            name = J$.N(604241, 'name', name, true, false, false);
                                            fieldOrSpec = J$.N(604249, 'fieldOrSpec', fieldOrSpec, true, false, false);
                                            options = J$.N(604257, 'options', options, true, false, false);
                                            J$.N(604265, '_this', _this, false, false, false);
                                            J$.N(604273, 'indexParameters', indexParameters, false, false, false);
                                            J$.N(604281, 'indexName', indexName, false, false, false);
                                            J$.N(604289, 'indexesObject', indexesObject, false, false, false);
                                            var _this;
                                            J$.F(603897, J$.R(603873, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(603881, 'this', this, false, false), J$.R(603889, 'CreateIndexOperation', CreateIndexOperation, false, false));
                                            _this = J$.W(603985, '_this', J$.F(603977, J$.R(603905, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(603913, 'this', this, false, false), J$.M(603969, J$.F(603937, J$.R(603921, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(603929, 'CreateIndexOperation', CreateIndexOperation, false, false)), 'call', false)(J$.R(603945, 'this', this, false, false), J$.R(603953, 'db', db, false, false), J$.R(603961, 'options', options, false, false))), _this, false, false);
                                            var indexParameters = J$.W(604017, 'indexParameters', J$.F(604009, J$.R(603993, 'parseIndexOptions', parseIndexOptions, false, true), false)(J$.R(604001, 'fieldOrSpec', fieldOrSpec, false, false)), indexParameters, false, false);
                                            var indexName = J$.W(604081, 'indexName', J$.C(38512, J$.B(37498, '===', J$.U(37490, 'typeof', J$.G(604033, J$.R(604025, 'options', options, false, false), 'name')), J$.T(604041, 'string', 21, false))) ? J$.G(604057, J$.R(604049, 'options', options, false, false), 'name') : J$.G(604073, J$.R(604065, 'indexParameters', indexParameters, false, false), 'name'), indexName, false, false);
                                            var indexesObject = J$.W(604121, 'indexesObject', J$.T(604113, {
                                                name: J$.R(604089, 'indexName', indexName, false, false),
                                                key: J$.G(604105, J$.R(604097, 'indexParameters', indexParameters, false, false), 'fieldHash')
                                            }, 11, false), indexesObject, false, false);
                                            J$.P(604145, J$.R(604129, '_this', _this, false, false), 'name', J$.R(604137, 'name', name, false, false));
                                            J$.P(604169, J$.R(604153, '_this', _this, false, false), 'fieldOrSpec', J$.R(604161, 'fieldOrSpec', fieldOrSpec, false, false));
                                            J$.P(604193, J$.R(604177, '_this', _this, false, false), 'indexes', J$.R(604185, 'indexesObject', indexesObject, false, false));
                                            return J$.Rt(604209, J$.R(604201, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(605929, J$e);
                                        } finally {
                                            if (J$.Fr(605937))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(605297, 'arguments', arguments, true, false, false);
                            _CommandOperation = J$.N(605305, '_CommandOperation', _CommandOperation, true, false, false);
                            CreateIndexOperation = J$.N(605321, 'CreateIndexOperation', J$.T(605313, CreateIndexOperation, 12, false), true, false, false);
                            J$.F(603865, J$.R(603841, '_inherits', _inherits, false, true), false)(J$.R(603849, 'CreateIndexOperation', CreateIndexOperation, false, false), J$.R(603857, '_CommandOperation', _CommandOperation, false, false));
                            J$.F(605265, J$.R(604297, '_createClass', _createClass, false, true), false)(J$.R(604305, 'CreateIndexOperation', CreateIndexOperation, false, false), J$.T(605257, [
                                J$.T(604609, {
                                    key: J$.T(604313, '_buildCommand', 21, false),
                                    value: J$.T(604601, function _buildCommand() {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(604545, arguments.callee, this, arguments);
                                                    arguments = J$.N(604553, 'arguments', arguments, true, false, false);
                                                    J$.N(604561, 'options', options, false, false, false);
                                                    J$.N(604569, 'name', name, false, false, false);
                                                    J$.N(604577, 'indexes', indexes, false, false, false);
                                                    J$.N(604585, 'optionName', optionName, false, false, false);
                                                    J$.N(604593, 'cmd', cmd, false, false, false);
                                                    var options = J$.W(604337, 'options', J$.G(604329, J$.R(604321, 'this', this, false, false), 'options'), options, false, false);
                                                    var name = J$.W(604361, 'name', J$.G(604353, J$.R(604345, 'this', this, false, false), 'name'), name, false, false);
                                                    var indexes = J$.W(604385, 'indexes', J$.G(604377, J$.R(604369, 'this', this, false, false), 'indexes'), indexes, false, false);
                                                    for (var optionName in J$.H(604473, J$.R(604393, 'options', options, false, false))) {
                                                        J$.N(604481, 'optionName', optionName, false, true, false);
                                                        {
                                                            {
                                                                if (J$.C(38520, J$.U(37506, '!', J$.M(604417, J$.R(604401, 'keysToOmit', keysToOmit, false, true), 'has', false)(J$.R(604409, 'optionName', optionName, false, false))))) {
                                                                    J$.P(604465, J$.R(604425, 'indexes', indexes, false, false), J$.R(604433, 'optionName', optionName, false, false), J$.G(604457, J$.R(604441, 'options', options, false, false), J$.R(604449, 'optionName', optionName, false, false)));
                                                                }
                                                            }
                                                        }
                                                    }
                                                    var cmd = J$.W(604521, 'cmd', J$.T(604513, {
                                                        createIndexes: J$.R(604489, 'name', name, false, false),
                                                        indexes: J$.T(604505, [J$.R(604497, 'indexes', indexes, false, false)], 10, false)
                                                    }, 11, false), cmd, false, false);
                                                    return J$.Rt(604537, J$.R(604529, 'cmd', cmd, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(605945, J$e);
                                                } finally {
                                                    if (J$.Fr(605953))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(605249, {
                                    key: J$.T(604617, 'execute', 21, false),
                                    value: J$.T(605241, function execute(callback) {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(605177, arguments.callee, this, arguments);
                                                    arguments = J$.N(605185, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(605193, 'callback', callback, true, false, false);
                                                    J$.N(605201, 'db', db, false, false, false);
                                                    J$.N(605209, 'options', options, false, false, false);
                                                    J$.N(605217, 'indexes', indexes, false, false, false);
                                                    J$.N(605225, 'capabilities', capabilities, false, false, false);
                                                    J$.N(605233, 'error', error, false, false, false);
                                                    var db = J$.W(604641, 'db', J$.G(604633, J$.R(604625, 'this', this, false, false), 'db'), db, false, false);
                                                    var options = J$.W(604665, 'options', J$.G(604657, J$.R(604649, 'this', this, false, false), 'options'), options, false, false);
                                                    var indexes = J$.W(604689, 'indexes', J$.G(604681, J$.R(604673, 'this', this, false, false), 'indexes'), indexes, false, false);
                                                    var capabilities = J$.W(604729, 'capabilities', J$.M(604721, J$.G(604713, J$.G(604705, J$.R(604697, 'db', db, false, false), 's'), 'topology'), 'capabilities', false)(), capabilities, false, false);
                                                    if (J$.C(38544, J$.C(38536, J$.C(38528, J$.G(604745, J$.R(604737, 'options', options, false, false), 'collation')) ? J$.R(604753, 'capabilities', capabilities, false, false) : J$._()) ? J$.U(37514, '!', J$.G(604769, J$.R(604761, 'capabilities', capabilities, false, false), 'commandsTakeCollation')) : J$._())) {
                                                        var error = J$.W(604801, 'error', J$.F(604793, J$.R(604777, 'MongoError', MongoError, false, true), true)(J$.T(604785, 'server/primary/mongos does not support collation', 21, false)), error, false, false);
                                                        J$.P(604825, J$.R(604809, 'error', error, false, false), 'code', J$.T(604817, 67, 22, false));
                                                        return J$.Rt(604857, J$.F(604849, J$.R(604833, 'callback', callback, false, false), false)(J$.R(604841, 'error', error, false, false)));
                                                    }
                                                    if (J$.C(38560, J$.C(38552, J$.G(604873, J$.R(604865, 'options', options, false, false), 'writeConcern')) ? J$.B(37530, '!==', J$.U(37522, 'typeof', J$.R(604881, 'callback', callback, false, false)), J$.T(604889, 'function', 21, false)) : J$._())) {
                                                        throw J$.M(604929, J$.R(604897, 'MongoError', MongoError, false, true), 'create', false)(J$.T(604921, {
                                                            message: J$.T(604905, 'Cannot use a writeConcern without a provided callback', 21, false),
                                                            driver: J$.T(604913, true, 23, false)
                                                        }, 11, false));
                                                    }
                                                    J$.M(605169, J$.F(604993, J$.R(604937, '_get', _get, false, true), false)(J$.F(604969, J$.R(604945, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(604961, J$.R(604953, 'CreateIndexOperation', CreateIndexOperation, false, false), 'prototype')), J$.T(604977, 'execute', 21, false), J$.R(604985, 'this', this, false, false)), 'call', false)(J$.R(605001, 'this', this, false, false), J$.T(605161, function (err, result) {
                                                        jalangiLabel18:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(605129, arguments.callee, this, arguments);
                                                                    arguments = J$.N(605137, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(605145, 'err', err, true, false, false);
                                                                    result = J$.N(605153, 'result', result, true, false, false);
                                                                    if (J$.C(38568, J$.B(37538, '==', J$.R(605009, 'err', err, false, false), J$.T(605017, null, 25, false))))
                                                                        return J$.Rt(605073, J$.F(605065, J$.R(605025, 'handleCallback', handleCallback, false, true), false)(J$.R(605033, 'callback', callback, false, false), J$.R(605041, 'err', err, false, false), J$.G(605057, J$.R(605049, 'indexes', indexes, false, false), 'name')));
                                                                    return J$.Rt(605121, J$.F(605113, J$.R(605081, 'handleCallback', handleCallback, false, true), false)(J$.R(605089, 'callback', callback, false, false), J$.R(605097, 'err', err, false, false), J$.R(605105, 'result', result, false, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(605961, J$e);
                                                                } finally {
                                                                    if (J$.Fr(605969))
                                                                        continue jalangiLabel18;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(605977, J$e);
                                                } finally {
                                                    if (J$.Fr(605985))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(605281, J$.R(605273, 'CreateIndexOperation', CreateIndexOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(605993, J$e);
                        } finally {
                            if (J$.Fr(606001))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(605337, 'CommandOperation', CommandOperation, false, true)), CreateIndexOperation, false, true);
            J$.F(605393, J$.R(605361, 'defineAspects', defineAspects, false, true), false)(J$.R(605369, 'CreateIndexOperation', CreateIndexOperation, false, true), J$.G(605385, J$.R(605377, 'Aspect', Aspect, false, true), 'WRITE_OPERATION'));
            J$.P(605417, J$.I(typeof module === 'undefined' ? module = J$.R(605401, 'module', undefined, true, true) : module = J$.R(605401, 'module', module, true, true)), 'exports', J$.R(605409, 'CreateIndexOperation', CreateIndexOperation, false, true));
        } catch (J$e) {
            J$.Ex(606009, J$e);
        } finally {
            if (J$.Sr(606017))
                continue jalangiLabel21;
            else
                break jalangiLabel21;
        }
    }
// JALANGI DO NOT INSTRUMENT
