// mongodb/lib/operations/drop_index.js
J$.noInstrEval = false;
jalangiLabel21:
    while (true) {
        try {
            J$.Se(648209, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/operations/drop_index.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(645113, arguments.callee, this, arguments);
                            arguments = J$.N(645121, 'arguments', arguments, true, false, false);
                            obj = J$.N(645129, 'obj', obj, true, false, false);
                            if (J$.C(41200, J$.C(41160, J$.B(39890, '===', J$.U(39882, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(644849, 'Symbol', undefined, true, true) : Symbol = J$.R(644849, 'Symbol', Symbol, true, true))), J$.T(644857, 'function', 21, false))) ? J$.B(39906, '===', J$.U(39898, 'typeof', J$.G(644873, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(644865, 'Symbol', undefined, true, true) : Symbol = J$.R(644865, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(644881, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(644937, '_typeof', J$.T(644929, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(644905, arguments.callee, this, arguments);
                                                arguments = J$.N(644913, 'arguments', arguments, true, false, false);
                                                obj = J$.N(644921, 'obj', obj, true, false, false);
                                                return J$.Rt(644897, J$.U(39914, 'typeof', J$.R(644889, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(648441, J$e);
                                            } finally {
                                                if (J$.Fr(648449))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(645073, '_typeof', J$.T(645065, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(645041, arguments.callee, this, arguments);
                                                arguments = J$.N(645049, 'arguments', arguments, true, false, false);
                                                obj = J$.N(645057, 'obj', obj, true, false, false);
                                                return J$.Rt(645033, J$.C(41192, J$.C(41184, J$.C(41176, J$.C(41168, J$.R(644945, 'obj', obj, false, false)) ? J$.B(39930, '===', J$.U(39922, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(644953, 'Symbol', undefined, true, true) : Symbol = J$.R(644953, 'Symbol', Symbol, true, true))), J$.T(644961, 'function', 21, false)) : J$._()) ? J$.B(39938, '===', J$.G(644977, J$.R(644969, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(644985, 'Symbol', undefined, true, true) : Symbol = J$.R(644985, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(39946, '!==', J$.R(644993, 'obj', obj, false, false), J$.G(645009, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(645001, 'Symbol', undefined, true, true) : Symbol = J$.R(645001, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(645017, 'symbol', 21, false) : J$.U(39954, 'typeof', J$.R(645025, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(648457, J$e);
                                            } finally {
                                                if (J$.Fr(648465))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(645105, J$.F(645097, J$.R(645081, '_typeof', _typeof, false, true), false)(J$.R(645089, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(648473, J$e);
                        } finally {
                            if (J$.Fr(648481))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(645177, arguments.callee, this, arguments);
                            arguments = J$.N(645185, 'arguments', arguments, true, false, false);
                            instance = J$.N(645193, 'instance', instance, true, false, false);
                            Constructor = J$.N(645201, 'Constructor', Constructor, true, false, false);
                            if (J$.C(41208, J$.U(39970, '!', J$.B(39962, 'instanceof', J$.R(645137, 'instance', instance, false, false), J$.R(645145, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(645169, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(645153, 'TypeError', undefined, true, true) : TypeError = J$.R(645153, 'TypeError', TypeError, true, true)), true)(J$.T(645161, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(648489, J$e);
                        } finally {
                            if (J$.Fr(648497))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(645449, arguments.callee, this, arguments);
                            arguments = J$.N(645457, 'arguments', arguments, true, false, false);
                            target = J$.N(645465, 'target', target, true, false, false);
                            props = J$.N(645473, 'props', props, true, false, false);
                            J$.N(645481, 'i', i, false, false, false);
                            J$.N(645489, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(645217, 'i', J$.T(645209, 0, 22, false), i, false, false); J$.C(41232, J$.B(39978, '<', J$.R(645225, 'i', i, false, false), J$.G(645241, J$.R(645233, 'props', props, false, false), 'length'))); J$.B(40002, '-', i = J$.W(645257, 'i', J$.B(39994, '+', J$.U(39986, '+', J$.R(645249, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(645289, 'descriptor', J$.G(645281, J$.R(645265, 'props', props, false, false), J$.R(645273, 'i', i, false, false)), descriptor, false, false);
                                J$.P(645329, J$.R(645297, 'descriptor', descriptor, false, false), 'enumerable', J$.C(41216, J$.G(645313, J$.R(645305, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(645321, false, 23, false));
                                J$.P(645353, J$.R(645337, 'descriptor', descriptor, false, false), 'configurable', J$.T(645345, true, 23, false));
                                if (J$.C(41224, J$.B(40010, 'in', J$.T(645361, 'value', 21, false), J$.R(645369, 'descriptor', descriptor, false, false))))
                                    J$.P(645393, J$.R(645377, 'descriptor', descriptor, false, false), 'writable', J$.T(645385, true, 23, false));
                                J$.M(645441, J$.I(typeof Object === 'undefined' ? Object = J$.R(645401, 'Object', undefined, true, true) : Object = J$.R(645401, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(645409, 'target', target, false, false), J$.G(645425, J$.R(645417, 'descriptor', descriptor, false, false), 'key'), J$.R(645433, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(648505, J$e);
                        } finally {
                            if (J$.Fr(648513))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(645601, arguments.callee, this, arguments);
                            arguments = J$.N(645609, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(645617, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(645625, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(645633, 'staticProps', staticProps, true, false, false);
                            if (J$.C(41240, J$.R(645497, 'protoProps', protoProps, false, false)))
                                J$.F(645537, J$.R(645505, '_defineProperties', _defineProperties, false, true), false)(J$.G(645521, J$.R(645513, 'Constructor', Constructor, false, false), 'prototype'), J$.R(645529, 'protoProps', protoProps, false, false));
                            if (J$.C(41248, J$.R(645545, 'staticProps', staticProps, false, false)))
                                J$.F(645577, J$.R(645553, '_defineProperties', _defineProperties, false, true), false)(J$.R(645561, 'Constructor', Constructor, false, false), J$.R(645569, 'staticProps', staticProps, false, false));
                            return J$.Rt(645593, J$.R(645585, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(648521, J$e);
                        } finally {
                            if (J$.Fr(648529))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(645745, arguments.callee, this, arguments);
                            arguments = J$.N(645753, 'arguments', arguments, true, false, false);
                            self = J$.N(645761, 'self', self, true, false, false);
                            call = J$.N(645769, 'call', call, true, false, false);
                            if (J$.C(41272, J$.C(41264, J$.R(645641, 'call', call, false, false)) ? J$.C(41256, J$.B(40018, '===', J$.F(645665, J$.R(645649, '_typeof', _typeof, false, true), false)(J$.R(645657, 'call', call, false, false)), J$.T(645673, 'object', 21, false))) ? J$._() : J$.B(40034, '===', J$.U(40026, 'typeof', J$.R(645681, 'call', call, false, false)), J$.T(645689, 'function', 21, false)) : J$._())) {
                                return J$.Rt(645705, J$.R(645697, 'call', call, false, false));
                            }
                            return J$.Rt(645737, J$.F(645729, J$.R(645713, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(645721, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(648537, J$e);
                        } finally {
                            if (J$.Fr(648545))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(645833, arguments.callee, this, arguments);
                            arguments = J$.N(645841, 'arguments', arguments, true, false, false);
                            self = J$.N(645849, 'self', self, true, false, false);
                            if (J$.C(41280, J$.B(40042, '===', J$.R(645777, 'self', self, false, false), void J$.T(645785, 0, 22, false)))) {
                                throw J$.F(645809, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(645793, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(645793, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(645801, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(645825, J$.R(645817, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(648553, J$e);
                        } finally {
                            if (J$.Fr(648561))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(646217, arguments.callee, this, arguments);
                            arguments = J$.N(646225, 'arguments', arguments, true, false, false);
                            target = J$.N(646233, 'target', target, true, false, false);
                            property = J$.N(646241, 'property', property, true, false, false);
                            receiver = J$.N(646249, 'receiver', receiver, true, false, false);
                            if (J$.C(41312, J$.C(41288, J$.B(40058, '!==', J$.U(40050, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(645857, 'Reflect', undefined, true, true) : Reflect = J$.R(645857, 'Reflect', Reflect, true, true))), J$.T(645865, 'undefined', 21, false))) ? J$.G(645881, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(645873, 'Reflect', undefined, true, true) : Reflect = J$.R(645873, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(645905, '_get', J$.G(645897, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(645889, 'Reflect', undefined, true, true) : Reflect = J$.R(645889, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(646153, '_get', J$.T(646145, function _get(target, property, receiver) {
                                    jalangiLabel8:
                                        while (true) {
                                            try {
                                                J$.Fe(646089, arguments.callee, this, arguments);
                                                arguments = J$.N(646097, 'arguments', arguments, true, false, false);
                                                target = J$.N(646105, 'target', target, true, false, false);
                                                property = J$.N(646113, 'property', property, true, false, false);
                                                receiver = J$.N(646121, 'receiver', receiver, true, false, false);
                                                J$.N(646129, 'base', base, false, false, false);
                                                J$.N(646137, 'desc', desc, false, false, false);
                                                var base = J$.W(645945, 'base', J$.F(645937, J$.R(645913, '_superPropBase', _superPropBase, false, true), false)(J$.R(645921, 'target', target, false, false), J$.R(645929, 'property', property, false, false)), base, false, false);
                                                if (J$.C(41296, J$.U(40066, '!', J$.R(645953, 'base', base, false, false))))
                                                    return J$.Rt(645961, undefined);
                                                var desc = J$.W(646001, 'desc', J$.M(645993, J$.I(typeof Object === 'undefined' ? Object = J$.R(645969, 'Object', undefined, true, true) : Object = J$.R(645969, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(645977, 'base', base, false, false), J$.R(645985, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(41304, J$.G(646017, J$.R(646009, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(646057, J$.M(646049, J$.G(646033, J$.R(646025, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(646041, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(646081, J$.G(646073, J$.R(646065, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(648569, J$e);
                                            } finally {
                                                if (J$.Fr(648577))
                                                    continue jalangiLabel8;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(646209, J$.F(646201, J$.R(646161, '_get', _get, false, true), false)(J$.R(646169, 'target', target, false, false), J$.R(646177, 'property', property, false, false), J$.C(41320, J$.R(646185, 'receiver', receiver, false, false)) ? J$._() : J$.R(646193, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(648585, J$e);
                        } finally {
                            if (J$.Fr(648593))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(646369, arguments.callee, this, arguments);
                            arguments = J$.N(646377, 'arguments', arguments, true, false, false);
                            object = J$.N(646385, 'object', object, true, false, false);
                            property = J$.N(646393, 'property', property, true, false, false);
                            while (J$.C(41336, J$.U(40074, '!', J$.M(646297, J$.G(646273, J$.G(646265, J$.I(typeof Object === 'undefined' ? Object = J$.R(646257, 'Object', undefined, true, true) : Object = J$.R(646257, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(646281, 'object', object, false, false), J$.R(646289, 'property', property, false, false))))) {
                                object = J$.W(646329, 'object', J$.F(646321, J$.R(646305, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(646313, 'object', object, false, false)), object, false, false);
                                if (J$.C(41328, J$.B(40082, '===', J$.R(646337, 'object', object, false, false), J$.T(646345, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(646361, J$.R(646353, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(648601, J$e);
                        } finally {
                            if (J$.Fr(648609))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(646553, arguments.callee, this, arguments);
                            arguments = J$.N(646561, 'arguments', arguments, true, false, false);
                            o = J$.N(646569, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(646513, '_getPrototypeOf', J$.C(41352, J$.G(646409, J$.I(typeof Object === 'undefined' ? Object = J$.R(646401, 'Object', undefined, true, true) : Object = J$.R(646401, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(646425, J$.I(typeof Object === 'undefined' ? Object = J$.R(646417, 'Object', undefined, true, true) : Object = J$.R(646417, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(646505, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(646481, arguments.callee, this, arguments);
                                            arguments = J$.N(646489, 'arguments', arguments, true, false, false);
                                            o = J$.N(646497, 'o', o, true, false, false);
                                            return J$.Rt(646473, J$.C(41344, J$.G(646441, J$.R(646433, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(646465, J$.I(typeof Object === 'undefined' ? Object = J$.R(646449, 'Object', undefined, true, true) : Object = J$.R(646449, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(646457, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(648617, J$e);
                                        } finally {
                                            if (J$.Fr(648625))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(646545, J$.F(646537, J$.R(646521, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(646529, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(648633, J$e);
                        } finally {
                            if (J$.Fr(648641))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(646769, arguments.callee, this, arguments);
                            arguments = J$.N(646777, 'arguments', arguments, true, false, false);
                            subClass = J$.N(646785, 'subClass', subClass, true, false, false);
                            superClass = J$.N(646793, 'superClass', superClass, true, false, false);
                            if (J$.C(41368, J$.C(41360, J$.B(40098, '!==', J$.U(40090, 'typeof', J$.R(646577, 'superClass', superClass, false, false)), J$.T(646585, 'function', 21, false))) ? J$.B(40106, '!==', J$.R(646593, 'superClass', superClass, false, false), J$.T(646601, null, 25, false)) : J$._())) {
                                throw J$.F(646625, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(646609, 'TypeError', undefined, true, true) : TypeError = J$.R(646609, 'TypeError', TypeError, true, true)), true)(J$.T(646617, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(646721, J$.R(646633, 'subClass', subClass, false, false), 'prototype', J$.M(646713, J$.I(typeof Object === 'undefined' ? Object = J$.R(646641, 'Object', undefined, true, true) : Object = J$.R(646641, 'Object', Object, true, true)), 'create', false)(J$.C(41376, J$.R(646649, 'superClass', superClass, false, false)) ? J$.G(646665, J$.R(646657, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(646705, {
                                constructor: J$.T(646697, {
                                    value: J$.R(646673, 'subClass', subClass, false, false),
                                    writable: J$.T(646681, true, 23, false),
                                    configurable: J$.T(646689, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(41384, J$.R(646729, 'superClass', superClass, false, false)))
                                J$.F(646761, J$.R(646737, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(646745, 'subClass', subClass, false, false), J$.R(646753, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(648649, J$e);
                        } finally {
                            if (J$.Fr(648657))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(646945, arguments.callee, this, arguments);
                            arguments = J$.N(646953, 'arguments', arguments, true, false, false);
                            o = J$.N(646961, 'o', o, true, false, false);
                            p = J$.N(646969, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(646897, '_setPrototypeOf', J$.C(41392, J$.G(646809, J$.I(typeof Object === 'undefined' ? Object = J$.R(646801, 'Object', undefined, true, true) : Object = J$.R(646801, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(646889, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(646857, arguments.callee, this, arguments);
                                            arguments = J$.N(646865, 'arguments', arguments, true, false, false);
                                            o = J$.N(646873, 'o', o, true, false, false);
                                            p = J$.N(646881, 'p', p, true, false, false);
                                            J$.P(646833, J$.R(646817, 'o', o, false, false), '__proto__', J$.R(646825, 'p', p, false, false));
                                            return J$.Rt(646849, J$.R(646841, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(648665, J$e);
                                        } finally {
                                            if (J$.Fr(648673))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(646937, J$.F(646929, J$.R(646905, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(646913, 'o', o, false, false), J$.R(646921, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(648681, J$e);
                        } finally {
                            if (J$.Fr(648689))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(648225, '_typeof', J$.T(648217, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(648241, '_classCallCheck', J$.T(648233, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(648257, '_defineProperties', J$.T(648249, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(648273, '_createClass', J$.T(648265, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(648289, '_possibleConstructorReturn', J$.T(648281, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(648305, '_assertThisInitialized', J$.T(648297, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(648321, '_get', J$.T(648313, _get, 12, false), true, false, false);
            _superPropBase = J$.N(648337, '_superPropBase', J$.T(648329, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(648353, '_getPrototypeOf', J$.T(648345, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(648369, '_inherits', J$.T(648361, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(648385, '_setPrototypeOf', J$.T(648377, _setPrototypeOf, 12, false), true, false, false);
            J$.N(648393, 'Aspect', Aspect, false, false, false);
            J$.N(648401, 'defineAspects', defineAspects, false, false, false);
            J$.N(648409, 'CommandOperation', CommandOperation, false, false, false);
            J$.N(648417, 'applyWriteConcern', applyWriteConcern, false, false, false);
            J$.N(648425, 'handleCallback', handleCallback, false, false, false);
            J$.N(648433, 'DropIndexOperation', DropIndexOperation, false, false, false);
            J$.T(644841, 'use strict', 21, false);
            var Aspect = J$.W(647009, 'Aspect', J$.G(647001, J$.F(646993, J$.I(typeof require === 'undefined' ? require = J$.R(646977, 'require', undefined, true, true) : require = J$.R(646977, 'require', require, true, true)), false)(J$.T(646985, './operation', 21, false)), 'Aspect'), Aspect, false, true);
            var defineAspects = J$.W(647049, 'defineAspects', J$.G(647041, J$.F(647033, J$.I(typeof require === 'undefined' ? require = J$.R(647017, 'require', undefined, true, true) : require = J$.R(647017, 'require', require, true, true)), false)(J$.T(647025, './operation', 21, false)), 'defineAspects'), defineAspects, false, true);
            var CommandOperation = J$.W(647081, 'CommandOperation', J$.F(647073, J$.I(typeof require === 'undefined' ? require = J$.R(647057, 'require', undefined, true, true) : require = J$.R(647057, 'require', require, true, true)), false)(J$.T(647065, './command', 21, false)), CommandOperation, false, true);
            var applyWriteConcern = J$.W(647121, 'applyWriteConcern', J$.G(647113, J$.F(647105, J$.I(typeof require === 'undefined' ? require = J$.R(647089, 'require', undefined, true, true) : require = J$.R(647089, 'require', require, true, true)), false)(J$.T(647097, '../utils', 21, false)), 'applyWriteConcern'), applyWriteConcern, false, true);
            var handleCallback = J$.W(647161, 'handleCallback', J$.G(647153, J$.F(647145, J$.I(typeof require === 'undefined' ? require = J$.R(647129, 'require', undefined, true, true) : require = J$.R(647129, 'require', require, true, true)), false)(J$.T(647137, '../utils', 21, false)), 'handleCallback'), handleCallback, false, true);
            var DropIndexOperation = J$.W(648137, 'DropIndexOperation', J$.F(648129, J$.T(648113, function (_CommandOperation) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(648073, arguments.callee, this, arguments);
                            function DropIndexOperation(collection, indexName, options) {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(647409, arguments.callee, this, arguments);
                                            arguments = J$.N(647417, 'arguments', arguments, true, false, false);
                                            collection = J$.N(647425, 'collection', collection, true, false, false);
                                            indexName = J$.N(647433, 'indexName', indexName, true, false, false);
                                            options = J$.N(647441, 'options', options, true, false, false);
                                            J$.N(647449, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(647225, J$.R(647201, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(647209, 'this', this, false, false), J$.R(647217, 'DropIndexOperation', DropIndexOperation, false, false));
                                            _this = J$.W(647337, '_this', J$.F(647329, J$.R(647233, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(647241, 'this', this, false, false), J$.M(647321, J$.F(647265, J$.R(647249, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(647257, 'DropIndexOperation', DropIndexOperation, false, false)), 'call', false)(J$.R(647273, 'this', this, false, false), J$.G(647297, J$.G(647289, J$.R(647281, 'collection', collection, false, false), 's'), 'db'), J$.R(647305, 'options', options, false, false), J$.R(647313, 'collection', collection, false, false))), _this, false, false);
                                            J$.P(647361, J$.R(647345, '_this', _this, false, false), 'collection', J$.R(647353, 'collection', collection, false, false));
                                            J$.P(647385, J$.R(647369, '_this', _this, false, false), 'indexName', J$.R(647377, 'indexName', indexName, false, false));
                                            return J$.Rt(647401, J$.R(647393, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(648697, J$e);
                                        } finally {
                                            if (J$.Fr(648705))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(648081, 'arguments', arguments, true, false, false);
                            _CommandOperation = J$.N(648089, '_CommandOperation', _CommandOperation, true, false, false);
                            DropIndexOperation = J$.N(648105, 'DropIndexOperation', J$.T(648097, DropIndexOperation, 12, false), true, false, false);
                            J$.F(647193, J$.R(647169, '_inherits', _inherits, false, true), false)(J$.R(647177, 'DropIndexOperation', DropIndexOperation, false, false), J$.R(647185, '_CommandOperation', _CommandOperation, false, false));
                            J$.F(648049, J$.R(647457, '_createClass', _createClass, false, true), false)(J$.R(647465, 'DropIndexOperation', DropIndexOperation, false, false), J$.T(648041, [
                                J$.T(647745, {
                                    key: J$.T(647473, '_buildCommand', 21, false),
                                    value: J$.T(647737, function _buildCommand() {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(647689, arguments.callee, this, arguments);
                                                    arguments = J$.N(647697, 'arguments', arguments, true, false, false);
                                                    J$.N(647705, 'collection', collection, false, false, false);
                                                    J$.N(647713, 'indexName', indexName, false, false, false);
                                                    J$.N(647721, 'options', options, false, false, false);
                                                    J$.N(647729, 'cmd', cmd, false, false, false);
                                                    var collection = J$.W(647497, 'collection', J$.G(647489, J$.R(647481, 'this', this, false, false), 'collection'), collection, false, false);
                                                    var indexName = J$.W(647521, 'indexName', J$.G(647513, J$.R(647505, 'this', this, false, false), 'indexName'), indexName, false, false);
                                                    var options = J$.W(647545, 'options', J$.G(647537, J$.R(647529, 'this', this, false, false), 'options'), options, false, false);
                                                    var cmd = J$.W(647585, 'cmd', J$.T(647577, {
                                                        dropIndexes: J$.G(647561, J$.R(647553, 'collection', collection, false, false), 'collectionName'),
                                                        index: J$.R(647569, 'indexName', indexName, false, false)
                                                    }, 11, false), cmd, false, false);
                                                    cmd = J$.W(647665, 'cmd', J$.F(647657, J$.R(647593, 'applyWriteConcern', applyWriteConcern, false, true), false)(J$.R(647601, 'cmd', cmd, false, false), J$.T(647641, {
                                                        db: J$.G(647625, J$.G(647617, J$.R(647609, 'collection', collection, false, false), 's'), 'db'),
                                                        collection: J$.R(647633, 'collection', collection, false, false)
                                                    }, 11, false), J$.R(647649, 'options', options, false, false)), cmd, false, false);
                                                    return J$.Rt(647681, J$.R(647673, 'cmd', cmd, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(648713, J$e);
                                                } finally {
                                                    if (J$.Fr(648721))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(648033, {
                                    key: J$.T(647753, 'execute', 21, false),
                                    value: J$.T(648025, function execute(callback) {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(648001, arguments.callee, this, arguments);
                                                    arguments = J$.N(648009, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(648017, 'callback', callback, true, false, false);
                                                    J$.M(647993, J$.F(647817, J$.R(647761, '_get', _get, false, true), false)(J$.F(647793, J$.R(647769, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(647785, J$.R(647777, 'DropIndexOperation', DropIndexOperation, false, false), 'prototype')), J$.T(647801, 'execute', 21, false), J$.R(647809, 'this', this, false, false)), 'call', false)(J$.R(647825, 'this', this, false, false), J$.T(647985, function (err, result) {
                                                        jalangiLabel18:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(647953, arguments.callee, this, arguments);
                                                                    arguments = J$.N(647961, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(647969, 'err', err, true, false, false);
                                                                    result = J$.N(647977, 'result', result, true, false, false);
                                                                    if (J$.C(41400, J$.B(40122, '!==', J$.U(40114, 'typeof', J$.R(647833, 'callback', callback, false, false)), J$.T(647841, 'function', 21, false))))
                                                                        return J$.Rt(647849, undefined);
                                                                    if (J$.C(41408, J$.R(647857, 'err', err, false, false)))
                                                                        return J$.Rt(647905, J$.F(647897, J$.R(647865, 'handleCallback', handleCallback, false, true), false)(J$.R(647873, 'callback', callback, false, false), J$.R(647881, 'err', err, false, false), J$.T(647889, null, 25, false)));
                                                                    J$.F(647945, J$.R(647913, 'handleCallback', handleCallback, false, true), false)(J$.R(647921, 'callback', callback, false, false), J$.T(647929, null, 25, false), J$.R(647937, 'result', result, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(648729, J$e);
                                                                } finally {
                                                                    if (J$.Fr(648737))
                                                                        continue jalangiLabel18;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(648745, J$e);
                                                } finally {
                                                    if (J$.Fr(648753))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(648065, J$.R(648057, 'DropIndexOperation', DropIndexOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(648761, J$e);
                        } finally {
                            if (J$.Fr(648769))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(648121, 'CommandOperation', CommandOperation, false, true)), DropIndexOperation, false, true);
            J$.F(648177, J$.R(648145, 'defineAspects', defineAspects, false, true), false)(J$.R(648153, 'DropIndexOperation', DropIndexOperation, false, true), J$.G(648169, J$.R(648161, 'Aspect', Aspect, false, true), 'WRITE_OPERATION'));
            J$.P(648201, J$.I(typeof module === 'undefined' ? module = J$.R(648185, 'module', undefined, true, true) : module = J$.R(648185, 'module', module, true, true)), 'exports', J$.R(648193, 'DropIndexOperation', DropIndexOperation, false, true));
        } catch (J$e) {
            J$.Ex(648777, J$e);
        } finally {
            if (J$.Sr(648785))
                continue jalangiLabel21;
            else
                break jalangiLabel21;
        }
    }
// JALANGI DO NOT INSTRUMENT
