// mongodb/lib/operations/insert_many.js
J$.noInstrEval = false;
jalangiLabel18:
    while (true) {
        try {
            J$.Se(700569, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/operations/insert_many.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(697889, arguments.callee, this, arguments);
                            arguments = J$.N(697897, 'arguments', arguments, true, false, false);
                            obj = J$.N(697905, 'obj', obj, true, false, false);
                            if (J$.C(44840, J$.C(44800, J$.B(43370, '===', J$.U(43362, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(697625, 'Symbol', undefined, true, true) : Symbol = J$.R(697625, 'Symbol', Symbol, true, true))), J$.T(697633, 'function', 21, false))) ? J$.B(43386, '===', J$.U(43378, 'typeof', J$.G(697649, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(697641, 'Symbol', undefined, true, true) : Symbol = J$.R(697641, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(697657, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(697713, '_typeof', J$.T(697705, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(697681, arguments.callee, this, arguments);
                                                arguments = J$.N(697689, 'arguments', arguments, true, false, false);
                                                obj = J$.N(697697, 'obj', obj, true, false, false);
                                                return J$.Rt(697673, J$.U(43394, 'typeof', J$.R(697665, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(700777, J$e);
                                            } finally {
                                                if (J$.Fr(700785))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(697849, '_typeof', J$.T(697841, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(697817, arguments.callee, this, arguments);
                                                arguments = J$.N(697825, 'arguments', arguments, true, false, false);
                                                obj = J$.N(697833, 'obj', obj, true, false, false);
                                                return J$.Rt(697809, J$.C(44832, J$.C(44824, J$.C(44816, J$.C(44808, J$.R(697721, 'obj', obj, false, false)) ? J$.B(43410, '===', J$.U(43402, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(697729, 'Symbol', undefined, true, true) : Symbol = J$.R(697729, 'Symbol', Symbol, true, true))), J$.T(697737, 'function', 21, false)) : J$._()) ? J$.B(43418, '===', J$.G(697753, J$.R(697745, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(697761, 'Symbol', undefined, true, true) : Symbol = J$.R(697761, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(43426, '!==', J$.R(697769, 'obj', obj, false, false), J$.G(697785, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(697777, 'Symbol', undefined, true, true) : Symbol = J$.R(697777, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(697793, 'symbol', 21, false) : J$.U(43434, 'typeof', J$.R(697801, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(700793, J$e);
                                            } finally {
                                                if (J$.Fr(700801))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(697881, J$.F(697873, J$.R(697857, '_typeof', _typeof, false, true), false)(J$.R(697865, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(700809, J$e);
                        } finally {
                            if (J$.Fr(700817))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(697953, arguments.callee, this, arguments);
                            arguments = J$.N(697961, 'arguments', arguments, true, false, false);
                            instance = J$.N(697969, 'instance', instance, true, false, false);
                            Constructor = J$.N(697977, 'Constructor', Constructor, true, false, false);
                            if (J$.C(44848, J$.U(43450, '!', J$.B(43442, 'instanceof', J$.R(697913, 'instance', instance, false, false), J$.R(697921, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(697945, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(697929, 'TypeError', undefined, true, true) : TypeError = J$.R(697929, 'TypeError', TypeError, true, true)), true)(J$.T(697937, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(700825, J$e);
                        } finally {
                            if (J$.Fr(700833))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(698225, arguments.callee, this, arguments);
                            arguments = J$.N(698233, 'arguments', arguments, true, false, false);
                            target = J$.N(698241, 'target', target, true, false, false);
                            props = J$.N(698249, 'props', props, true, false, false);
                            J$.N(698257, 'i', i, false, false, false);
                            J$.N(698265, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(697993, 'i', J$.T(697985, 0, 22, false), i, false, false); J$.C(44872, J$.B(43458, '<', J$.R(698001, 'i', i, false, false), J$.G(698017, J$.R(698009, 'props', props, false, false), 'length'))); J$.B(43482, '-', i = J$.W(698033, 'i', J$.B(43474, '+', J$.U(43466, '+', J$.R(698025, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(698065, 'descriptor', J$.G(698057, J$.R(698041, 'props', props, false, false), J$.R(698049, 'i', i, false, false)), descriptor, false, false);
                                J$.P(698105, J$.R(698073, 'descriptor', descriptor, false, false), 'enumerable', J$.C(44856, J$.G(698089, J$.R(698081, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(698097, false, 23, false));
                                J$.P(698129, J$.R(698113, 'descriptor', descriptor, false, false), 'configurable', J$.T(698121, true, 23, false));
                                if (J$.C(44864, J$.B(43490, 'in', J$.T(698137, 'value', 21, false), J$.R(698145, 'descriptor', descriptor, false, false))))
                                    J$.P(698169, J$.R(698153, 'descriptor', descriptor, false, false), 'writable', J$.T(698161, true, 23, false));
                                J$.M(698217, J$.I(typeof Object === 'undefined' ? Object = J$.R(698177, 'Object', undefined, true, true) : Object = J$.R(698177, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(698185, 'target', target, false, false), J$.G(698201, J$.R(698193, 'descriptor', descriptor, false, false), 'key'), J$.R(698209, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(700841, J$e);
                        } finally {
                            if (J$.Fr(700849))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(698377, arguments.callee, this, arguments);
                            arguments = J$.N(698385, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(698393, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(698401, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(698409, 'staticProps', staticProps, true, false, false);
                            if (J$.C(44880, J$.R(698273, 'protoProps', protoProps, false, false)))
                                J$.F(698313, J$.R(698281, '_defineProperties', _defineProperties, false, true), false)(J$.G(698297, J$.R(698289, 'Constructor', Constructor, false, false), 'prototype'), J$.R(698305, 'protoProps', protoProps, false, false));
                            if (J$.C(44888, J$.R(698321, 'staticProps', staticProps, false, false)))
                                J$.F(698353, J$.R(698329, '_defineProperties', _defineProperties, false, true), false)(J$.R(698337, 'Constructor', Constructor, false, false), J$.R(698345, 'staticProps', staticProps, false, false));
                            return J$.Rt(698369, J$.R(698361, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(700857, J$e);
                        } finally {
                            if (J$.Fr(700865))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(698521, arguments.callee, this, arguments);
                            arguments = J$.N(698529, 'arguments', arguments, true, false, false);
                            self = J$.N(698537, 'self', self, true, false, false);
                            call = J$.N(698545, 'call', call, true, false, false);
                            if (J$.C(44912, J$.C(44904, J$.R(698417, 'call', call, false, false)) ? J$.C(44896, J$.B(43498, '===', J$.F(698441, J$.R(698425, '_typeof', _typeof, false, true), false)(J$.R(698433, 'call', call, false, false)), J$.T(698449, 'object', 21, false))) ? J$._() : J$.B(43514, '===', J$.U(43506, 'typeof', J$.R(698457, 'call', call, false, false)), J$.T(698465, 'function', 21, false)) : J$._())) {
                                return J$.Rt(698481, J$.R(698473, 'call', call, false, false));
                            }
                            return J$.Rt(698513, J$.F(698505, J$.R(698489, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(698497, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(700873, J$e);
                        } finally {
                            if (J$.Fr(700881))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(698609, arguments.callee, this, arguments);
                            arguments = J$.N(698617, 'arguments', arguments, true, false, false);
                            self = J$.N(698625, 'self', self, true, false, false);
                            if (J$.C(44920, J$.B(43522, '===', J$.R(698553, 'self', self, false, false), void J$.T(698561, 0, 22, false)))) {
                                throw J$.F(698585, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(698569, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(698569, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(698577, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(698601, J$.R(698593, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(700889, J$e);
                        } finally {
                            if (J$.Fr(700897))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(698785, arguments.callee, this, arguments);
                            arguments = J$.N(698793, 'arguments', arguments, true, false, false);
                            o = J$.N(698801, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(698745, '_getPrototypeOf', J$.C(44936, J$.G(698641, J$.I(typeof Object === 'undefined' ? Object = J$.R(698633, 'Object', undefined, true, true) : Object = J$.R(698633, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(698657, J$.I(typeof Object === 'undefined' ? Object = J$.R(698649, 'Object', undefined, true, true) : Object = J$.R(698649, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(698737, function _getPrototypeOf(o) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(698713, arguments.callee, this, arguments);
                                            arguments = J$.N(698721, 'arguments', arguments, true, false, false);
                                            o = J$.N(698729, 'o', o, true, false, false);
                                            return J$.Rt(698705, J$.C(44928, J$.G(698673, J$.R(698665, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(698697, J$.I(typeof Object === 'undefined' ? Object = J$.R(698681, 'Object', undefined, true, true) : Object = J$.R(698681, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(698689, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(700905, J$e);
                                        } finally {
                                            if (J$.Fr(700913))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(698777, J$.F(698769, J$.R(698753, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(698761, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(700921, J$e);
                        } finally {
                            if (J$.Fr(700929))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(699001, arguments.callee, this, arguments);
                            arguments = J$.N(699009, 'arguments', arguments, true, false, false);
                            subClass = J$.N(699017, 'subClass', subClass, true, false, false);
                            superClass = J$.N(699025, 'superClass', superClass, true, false, false);
                            if (J$.C(44952, J$.C(44944, J$.B(43538, '!==', J$.U(43530, 'typeof', J$.R(698809, 'superClass', superClass, false, false)), J$.T(698817, 'function', 21, false))) ? J$.B(43546, '!==', J$.R(698825, 'superClass', superClass, false, false), J$.T(698833, null, 25, false)) : J$._())) {
                                throw J$.F(698857, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(698841, 'TypeError', undefined, true, true) : TypeError = J$.R(698841, 'TypeError', TypeError, true, true)), true)(J$.T(698849, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(698953, J$.R(698865, 'subClass', subClass, false, false), 'prototype', J$.M(698945, J$.I(typeof Object === 'undefined' ? Object = J$.R(698873, 'Object', undefined, true, true) : Object = J$.R(698873, 'Object', Object, true, true)), 'create', false)(J$.C(44960, J$.R(698881, 'superClass', superClass, false, false)) ? J$.G(698897, J$.R(698889, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(698937, {
                                constructor: J$.T(698929, {
                                    value: J$.R(698905, 'subClass', subClass, false, false),
                                    writable: J$.T(698913, true, 23, false),
                                    configurable: J$.T(698921, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(44968, J$.R(698961, 'superClass', superClass, false, false)))
                                J$.F(698993, J$.R(698969, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(698977, 'subClass', subClass, false, false), J$.R(698985, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(700937, J$e);
                        } finally {
                            if (J$.Fr(700945))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(699177, arguments.callee, this, arguments);
                            arguments = J$.N(699185, 'arguments', arguments, true, false, false);
                            o = J$.N(699193, 'o', o, true, false, false);
                            p = J$.N(699201, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(699129, '_setPrototypeOf', J$.C(44976, J$.G(699041, J$.I(typeof Object === 'undefined' ? Object = J$.R(699033, 'Object', undefined, true, true) : Object = J$.R(699033, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(699121, function _setPrototypeOf(o, p) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(699089, arguments.callee, this, arguments);
                                            arguments = J$.N(699097, 'arguments', arguments, true, false, false);
                                            o = J$.N(699105, 'o', o, true, false, false);
                                            p = J$.N(699113, 'p', p, true, false, false);
                                            J$.P(699065, J$.R(699049, 'o', o, false, false), '__proto__', J$.R(699057, 'p', p, false, false));
                                            return J$.Rt(699081, J$.R(699073, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(700953, J$e);
                                        } finally {
                                            if (J$.Fr(700961))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(699169, J$.F(699161, J$.R(699137, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(699145, 'o', o, false, false), J$.R(699153, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(700969, J$e);
                        } finally {
                            if (J$.Fr(700977))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function mapInsertManyResults(docs, r) {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(700505, arguments.callee, this, arguments);
                            arguments = J$.N(700513, 'arguments', arguments, true, false, false);
                            docs = J$.N(700521, 'docs', docs, true, false, false);
                            r = J$.N(700529, 'r', r, true, false, false);
                            J$.N(700537, 'finalResult', finalResult, false, false, false);
                            var finalResult = J$.W(700425, 'finalResult', J$.T(700417, {
                                result: J$.T(700369, {
                                    ok: J$.T(700345, 1, 22, false),
                                    n: J$.G(700361, J$.R(700353, 'r', r, false, false), 'insertedCount')
                                }, 11, false),
                                ops: J$.R(700377, 'docs', docs, false, false),
                                insertedCount: J$.G(700393, J$.R(700385, 'r', r, false, false), 'insertedCount'),
                                insertedIds: J$.G(700409, J$.R(700401, 'r', r, false, false), 'insertedIds')
                            }, 11, false), finalResult, false, false);
                            if (J$.C(45008, J$.M(700441, J$.R(700433, 'r', r, false, false), 'getLastOp', false)())) {
                                J$.P(700481, J$.G(700457, J$.R(700449, 'finalResult', finalResult, false, false), 'result'), 'opTime', J$.M(700473, J$.R(700465, 'r', r, false, false), 'getLastOp', false)());
                            }
                            return J$.Rt(700497, J$.R(700489, 'finalResult', finalResult, false, false));
                        } catch (J$e) {
                            J$.Ex(701049, J$e);
                        } finally {
                            if (J$.Fr(701057))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(700585, '_typeof', J$.T(700577, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(700601, '_classCallCheck', J$.T(700593, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(700617, '_defineProperties', J$.T(700609, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(700633, '_createClass', J$.T(700625, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(700649, '_possibleConstructorReturn', J$.T(700641, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(700665, '_assertThisInitialized', J$.T(700657, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(700681, '_getPrototypeOf', J$.T(700673, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(700697, '_inherits', J$.T(700689, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(700713, '_setPrototypeOf', J$.T(700705, _setPrototypeOf, 12, false), true, false, false);
            J$.N(700721, 'OperationBase', OperationBase, false, false, false);
            J$.N(700729, 'BulkWriteOperation', BulkWriteOperation, false, false, false);
            J$.N(700737, 'MongoError', MongoError, false, false, false);
            J$.N(700745, 'prepareDocs', prepareDocs, false, false, false);
            J$.N(700753, 'InsertManyOperation', InsertManyOperation, false, false, false);
            mapInsertManyResults = J$.N(700769, 'mapInsertManyResults', J$.T(700761, mapInsertManyResults, 12, false), true, false, false);
            J$.T(697617, 'use strict', 21, false);
            var OperationBase = J$.W(699241, 'OperationBase', J$.G(699233, J$.F(699225, J$.I(typeof require === 'undefined' ? require = J$.R(699209, 'require', undefined, true, true) : require = J$.R(699209, 'require', require, true, true)), false)(J$.T(699217, './operation', 21, false)), 'OperationBase'), OperationBase, false, true);
            var BulkWriteOperation = J$.W(699273, 'BulkWriteOperation', J$.F(699265, J$.I(typeof require === 'undefined' ? require = J$.R(699249, 'require', undefined, true, true) : require = J$.R(699249, 'require', require, true, true)), false)(J$.T(699257, './bulk_write', 21, false)), BulkWriteOperation, false, true);
            var MongoError = J$.W(699313, 'MongoError', J$.G(699305, J$.F(699297, J$.I(typeof require === 'undefined' ? require = J$.R(699281, 'require', undefined, true, true) : require = J$.R(699281, 'require', require, true, true)), false)(J$.T(699289, '../core', 21, false)), 'MongoError'), MongoError, false, true);
            var prepareDocs = J$.W(699353, 'prepareDocs', J$.G(699345, J$.F(699337, J$.I(typeof require === 'undefined' ? require = J$.R(699321, 'require', undefined, true, true) : require = J$.R(699321, 'require', require, true, true)), false)(J$.T(699329, './common_functions', 21, false)), 'prepareDocs'), prepareDocs, false, true);
            var InsertManyOperation = J$.W(700337, 'InsertManyOperation', J$.F(700329, J$.T(700313, function (_OperationBase) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(700273, arguments.callee, this, arguments);
                            function InsertManyOperation(collection, docs, options) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(699569, arguments.callee, this, arguments);
                                            arguments = J$.N(699577, 'arguments', arguments, true, false, false);
                                            collection = J$.N(699585, 'collection', collection, true, false, false);
                                            docs = J$.N(699593, 'docs', docs, true, false, false);
                                            options = J$.N(699601, 'options', options, true, false, false);
                                            J$.N(699609, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(699417, J$.R(699393, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(699401, 'this', this, false, false), J$.R(699409, 'InsertManyOperation', InsertManyOperation, false, false));
                                            _this = J$.W(699497, '_this', J$.F(699489, J$.R(699425, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(699433, 'this', this, false, false), J$.M(699481, J$.F(699457, J$.R(699441, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(699449, 'InsertManyOperation', InsertManyOperation, false, false)), 'call', false)(J$.R(699465, 'this', this, false, false), J$.R(699473, 'options', options, false, false))), _this, false, false);
                                            J$.P(699521, J$.R(699505, '_this', _this, false, false), 'collection', J$.R(699513, 'collection', collection, false, false));
                                            J$.P(699545, J$.R(699529, '_this', _this, false, false), 'docs', J$.R(699537, 'docs', docs, false, false));
                                            return J$.Rt(699561, J$.R(699553, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(700985, J$e);
                                        } finally {
                                            if (J$.Fr(700993))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(700281, 'arguments', arguments, true, false, false);
                            _OperationBase = J$.N(700289, '_OperationBase', _OperationBase, true, false, false);
                            InsertManyOperation = J$.N(700305, 'InsertManyOperation', J$.T(700297, InsertManyOperation, 12, false), true, false, false);
                            J$.F(699385, J$.R(699361, '_inherits', _inherits, false, true), false)(J$.R(699369, 'InsertManyOperation', InsertManyOperation, false, false), J$.R(699377, '_OperationBase', _OperationBase, false, false));
                            J$.F(700249, J$.R(699617, '_createClass', _createClass, false, true), false)(J$.R(699625, 'InsertManyOperation', InsertManyOperation, false, false), J$.T(700241, [J$.T(700233, {
                                    key: J$.T(699633, 'execute', 21, false),
                                    value: J$.T(700225, function execute(callback) {
                                        jalangiLabel15:
                                            while (true) {
                                                try {
                                                    J$.Fe(700161, arguments.callee, this, arguments);
                                                    arguments = J$.N(700169, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(700177, 'callback', callback, true, false, false);
                                                    J$.N(700185, 'coll', coll, false, false, false);
                                                    J$.N(700193, 'docs', docs, false, false, false);
                                                    J$.N(700201, 'options', options, false, false, false);
                                                    J$.N(700209, 'operations', operations, false, false, false);
                                                    J$.N(700217, 'bulkWriteOperation', bulkWriteOperation, false, false, false);
                                                    var coll = J$.W(699657, 'coll', J$.G(699649, J$.R(699641, 'this', this, false, false), 'collection'), coll, false, false);
                                                    var docs = J$.W(699681, 'docs', J$.G(699673, J$.R(699665, 'this', this, false, false), 'docs'), docs, false, false);
                                                    var options = J$.W(699705, 'options', J$.G(699697, J$.R(699689, 'this', this, false, false), 'options'), options, false, false);
                                                    if (J$.C(44984, J$.U(43554, '!', J$.M(699729, J$.I(typeof Array === 'undefined' ? Array = J$.R(699713, 'Array', undefined, true, true) : Array = J$.R(699713, 'Array', Array, true, true)), 'isArray', false)(J$.R(699721, 'docs', docs, false, false))))) {
                                                        return J$.Rt(699793, J$.F(699785, J$.R(699737, 'callback', callback, false, false), false)(J$.M(699777, J$.R(699745, 'MongoError', MongoError, false, true), 'create', false)(J$.T(699769, {
                                                            message: J$.T(699753, 'docs parameter must be an array of documents', 21, false),
                                                            driver: J$.T(699761, true, 23, false)
                                                        }, 11, false))));
                                                    }
                                                    J$.P(699865, J$.R(699801, 'options', options, false, false), J$.T(699809, 'serializeFunctions', 21, false), J$.C(44992, J$.G(699833, J$.R(699817, 'options', options, false, false), J$.T(699825, 'serializeFunctions', 21, false))) ? J$._() : J$.G(699857, J$.G(699849, J$.R(699841, 'coll', coll, false, false), 's'), 'serializeFunctions'));
                                                    docs = J$.W(699913, 'docs', J$.F(699905, J$.R(699873, 'prepareDocs', prepareDocs, false, true), false)(J$.R(699881, 'coll', coll, false, false), J$.R(699889, 'docs', docs, false, false), J$.R(699897, 'options', options, false, false)), docs, false, false);
                                                    var operations = J$.W(699945, 'operations', J$.T(699937, [J$.T(699929, { insertMany: J$.R(699921, 'docs', docs, false, false) }, 11, false)], 10, false), operations, false, false);
                                                    var bulkWriteOperation = J$.W(699993, 'bulkWriteOperation', J$.F(699985, J$.R(699953, 'BulkWriteOperation', BulkWriteOperation, false, true), true)(J$.R(699961, 'coll', coll, false, false), J$.R(699969, 'operations', operations, false, false), J$.R(699977, 'options', options, false, false)), bulkWriteOperation, false, false);
                                                    J$.M(700153, J$.R(700001, 'bulkWriteOperation', bulkWriteOperation, false, false), 'execute', false)(J$.T(700145, function (err, result) {
                                                        jalangiLabel14:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(700113, arguments.callee, this, arguments);
                                                                    arguments = J$.N(700121, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(700129, 'err', err, true, false, false);
                                                                    result = J$.N(700137, 'result', result, true, false, false);
                                                                    if (J$.C(45000, J$.R(700009, 'err', err, false, false)))
                                                                        return J$.Rt(700049, J$.F(700041, J$.R(700017, 'callback', callback, false, false), false)(J$.R(700025, 'err', err, false, false), J$.T(700033, null, 25, false)));
                                                                    J$.F(700105, J$.R(700057, 'callback', callback, false, false), false)(J$.T(700065, null, 25, false), J$.F(700097, J$.R(700073, 'mapInsertManyResults', mapInsertManyResults, false, true), false)(J$.R(700081, 'docs', docs, false, false), J$.R(700089, 'result', result, false, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(701001, J$e);
                                                                } finally {
                                                                    if (J$.Fr(701009))
                                                                        continue jalangiLabel14;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(701017, J$e);
                                                } finally {
                                                    if (J$.Fr(701025))
                                                        continue jalangiLabel15;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(700265, J$.R(700257, 'InsertManyOperation', InsertManyOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(701033, J$e);
                        } finally {
                            if (J$.Fr(701041))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(700321, 'OperationBase', OperationBase, false, true)), InsertManyOperation, false, true);
            J$.P(700561, J$.I(typeof module === 'undefined' ? module = J$.R(700545, 'module', undefined, true, true) : module = J$.R(700545, 'module', module, true, true)), 'exports', J$.R(700553, 'InsertManyOperation', InsertManyOperation, false, true));
        } catch (J$e) {
            J$.Ex(701065, J$e);
        } finally {
            if (J$.Sr(701073))
                continue jalangiLabel18;
            else
                break jalangiLabel18;
        }
    }
// JALANGI DO NOT INSTRUMENT
