// mongodb/lib/operations/list_collections.js
J$.noInstrEval = false;
jalangiLabel22:
    while (true) {
        try {
            J$.Se(712009, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/operations/list_collections.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(707865, arguments.callee, this, arguments);
                            arguments = J$.N(707873, 'arguments', arguments, true, false, false);
                            obj = J$.N(707881, 'obj', obj, true, false, false);
                            if (J$.C(45544, J$.C(45504, J$.B(44026, '===', J$.U(44018, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(707601, 'Symbol', undefined, true, true) : Symbol = J$.R(707601, 'Symbol', Symbol, true, true))), J$.T(707609, 'function', 21, false))) ? J$.B(44042, '===', J$.U(44034, 'typeof', J$.G(707625, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(707617, 'Symbol', undefined, true, true) : Symbol = J$.R(707617, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(707633, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(707689, '_typeof', J$.T(707681, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(707657, arguments.callee, this, arguments);
                                                arguments = J$.N(707665, 'arguments', arguments, true, false, false);
                                                obj = J$.N(707673, 'obj', obj, true, false, false);
                                                return J$.Rt(707649, J$.U(44050, 'typeof', J$.R(707641, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(712265, J$e);
                                            } finally {
                                                if (J$.Fr(712273))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(707825, '_typeof', J$.T(707817, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(707793, arguments.callee, this, arguments);
                                                arguments = J$.N(707801, 'arguments', arguments, true, false, false);
                                                obj = J$.N(707809, 'obj', obj, true, false, false);
                                                return J$.Rt(707785, J$.C(45536, J$.C(45528, J$.C(45520, J$.C(45512, J$.R(707697, 'obj', obj, false, false)) ? J$.B(44066, '===', J$.U(44058, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(707705, 'Symbol', undefined, true, true) : Symbol = J$.R(707705, 'Symbol', Symbol, true, true))), J$.T(707713, 'function', 21, false)) : J$._()) ? J$.B(44074, '===', J$.G(707729, J$.R(707721, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(707737, 'Symbol', undefined, true, true) : Symbol = J$.R(707737, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(44082, '!==', J$.R(707745, 'obj', obj, false, false), J$.G(707761, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(707753, 'Symbol', undefined, true, true) : Symbol = J$.R(707753, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(707769, 'symbol', 21, false) : J$.U(44090, 'typeof', J$.R(707777, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(712281, J$e);
                                            } finally {
                                                if (J$.Fr(712289))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(707857, J$.F(707849, J$.R(707833, '_typeof', _typeof, false, true), false)(J$.R(707841, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(712297, J$e);
                        } finally {
                            if (J$.Fr(712305))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(707929, arguments.callee, this, arguments);
                            arguments = J$.N(707937, 'arguments', arguments, true, false, false);
                            instance = J$.N(707945, 'instance', instance, true, false, false);
                            Constructor = J$.N(707953, 'Constructor', Constructor, true, false, false);
                            if (J$.C(45552, J$.U(44106, '!', J$.B(44098, 'instanceof', J$.R(707889, 'instance', instance, false, false), J$.R(707897, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(707921, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(707905, 'TypeError', undefined, true, true) : TypeError = J$.R(707905, 'TypeError', TypeError, true, true)), true)(J$.T(707913, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(712313, J$e);
                        } finally {
                            if (J$.Fr(712321))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(708201, arguments.callee, this, arguments);
                            arguments = J$.N(708209, 'arguments', arguments, true, false, false);
                            target = J$.N(708217, 'target', target, true, false, false);
                            props = J$.N(708225, 'props', props, true, false, false);
                            J$.N(708233, 'i', i, false, false, false);
                            J$.N(708241, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(707969, 'i', J$.T(707961, 0, 22, false), i, false, false); J$.C(45576, J$.B(44114, '<', J$.R(707977, 'i', i, false, false), J$.G(707993, J$.R(707985, 'props', props, false, false), 'length'))); J$.B(44138, '-', i = J$.W(708009, 'i', J$.B(44130, '+', J$.U(44122, '+', J$.R(708001, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(708041, 'descriptor', J$.G(708033, J$.R(708017, 'props', props, false, false), J$.R(708025, 'i', i, false, false)), descriptor, false, false);
                                J$.P(708081, J$.R(708049, 'descriptor', descriptor, false, false), 'enumerable', J$.C(45560, J$.G(708065, J$.R(708057, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(708073, false, 23, false));
                                J$.P(708105, J$.R(708089, 'descriptor', descriptor, false, false), 'configurable', J$.T(708097, true, 23, false));
                                if (J$.C(45568, J$.B(44146, 'in', J$.T(708113, 'value', 21, false), J$.R(708121, 'descriptor', descriptor, false, false))))
                                    J$.P(708145, J$.R(708129, 'descriptor', descriptor, false, false), 'writable', J$.T(708137, true, 23, false));
                                J$.M(708193, J$.I(typeof Object === 'undefined' ? Object = J$.R(708153, 'Object', undefined, true, true) : Object = J$.R(708153, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(708161, 'target', target, false, false), J$.G(708177, J$.R(708169, 'descriptor', descriptor, false, false), 'key'), J$.R(708185, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(712329, J$e);
                        } finally {
                            if (J$.Fr(712337))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(708353, arguments.callee, this, arguments);
                            arguments = J$.N(708361, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(708369, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(708377, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(708385, 'staticProps', staticProps, true, false, false);
                            if (J$.C(45584, J$.R(708249, 'protoProps', protoProps, false, false)))
                                J$.F(708289, J$.R(708257, '_defineProperties', _defineProperties, false, true), false)(J$.G(708273, J$.R(708265, 'Constructor', Constructor, false, false), 'prototype'), J$.R(708281, 'protoProps', protoProps, false, false));
                            if (J$.C(45592, J$.R(708297, 'staticProps', staticProps, false, false)))
                                J$.F(708329, J$.R(708305, '_defineProperties', _defineProperties, false, true), false)(J$.R(708313, 'Constructor', Constructor, false, false), J$.R(708321, 'staticProps', staticProps, false, false));
                            return J$.Rt(708345, J$.R(708337, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(712345, J$e);
                        } finally {
                            if (J$.Fr(712353))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(708497, arguments.callee, this, arguments);
                            arguments = J$.N(708505, 'arguments', arguments, true, false, false);
                            self = J$.N(708513, 'self', self, true, false, false);
                            call = J$.N(708521, 'call', call, true, false, false);
                            if (J$.C(45616, J$.C(45608, J$.R(708393, 'call', call, false, false)) ? J$.C(45600, J$.B(44154, '===', J$.F(708417, J$.R(708401, '_typeof', _typeof, false, true), false)(J$.R(708409, 'call', call, false, false)), J$.T(708425, 'object', 21, false))) ? J$._() : J$.B(44170, '===', J$.U(44162, 'typeof', J$.R(708433, 'call', call, false, false)), J$.T(708441, 'function', 21, false)) : J$._())) {
                                return J$.Rt(708457, J$.R(708449, 'call', call, false, false));
                            }
                            return J$.Rt(708489, J$.F(708481, J$.R(708465, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(708473, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(712361, J$e);
                        } finally {
                            if (J$.Fr(712369))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(708585, arguments.callee, this, arguments);
                            arguments = J$.N(708593, 'arguments', arguments, true, false, false);
                            self = J$.N(708601, 'self', self, true, false, false);
                            if (J$.C(45624, J$.B(44178, '===', J$.R(708529, 'self', self, false, false), void J$.T(708537, 0, 22, false)))) {
                                throw J$.F(708561, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(708545, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(708545, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(708553, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(708577, J$.R(708569, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(712377, J$e);
                        } finally {
                            if (J$.Fr(712385))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(708969, arguments.callee, this, arguments);
                            arguments = J$.N(708977, 'arguments', arguments, true, false, false);
                            target = J$.N(708985, 'target', target, true, false, false);
                            property = J$.N(708993, 'property', property, true, false, false);
                            receiver = J$.N(709001, 'receiver', receiver, true, false, false);
                            if (J$.C(45656, J$.C(45632, J$.B(44194, '!==', J$.U(44186, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(708609, 'Reflect', undefined, true, true) : Reflect = J$.R(708609, 'Reflect', Reflect, true, true))), J$.T(708617, 'undefined', 21, false))) ? J$.G(708633, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(708625, 'Reflect', undefined, true, true) : Reflect = J$.R(708625, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(708657, '_get', J$.G(708649, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(708641, 'Reflect', undefined, true, true) : Reflect = J$.R(708641, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(708905, '_get', J$.T(708897, function _get(target, property, receiver) {
                                    jalangiLabel8:
                                        while (true) {
                                            try {
                                                J$.Fe(708841, arguments.callee, this, arguments);
                                                arguments = J$.N(708849, 'arguments', arguments, true, false, false);
                                                target = J$.N(708857, 'target', target, true, false, false);
                                                property = J$.N(708865, 'property', property, true, false, false);
                                                receiver = J$.N(708873, 'receiver', receiver, true, false, false);
                                                J$.N(708881, 'base', base, false, false, false);
                                                J$.N(708889, 'desc', desc, false, false, false);
                                                var base = J$.W(708697, 'base', J$.F(708689, J$.R(708665, '_superPropBase', _superPropBase, false, true), false)(J$.R(708673, 'target', target, false, false), J$.R(708681, 'property', property, false, false)), base, false, false);
                                                if (J$.C(45640, J$.U(44202, '!', J$.R(708705, 'base', base, false, false))))
                                                    return J$.Rt(708713, undefined);
                                                var desc = J$.W(708753, 'desc', J$.M(708745, J$.I(typeof Object === 'undefined' ? Object = J$.R(708721, 'Object', undefined, true, true) : Object = J$.R(708721, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(708729, 'base', base, false, false), J$.R(708737, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(45648, J$.G(708769, J$.R(708761, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(708809, J$.M(708801, J$.G(708785, J$.R(708777, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(708793, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(708833, J$.G(708825, J$.R(708817, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(712393, J$e);
                                            } finally {
                                                if (J$.Fr(712401))
                                                    continue jalangiLabel8;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(708961, J$.F(708953, J$.R(708913, '_get', _get, false, true), false)(J$.R(708921, 'target', target, false, false), J$.R(708929, 'property', property, false, false), J$.C(45664, J$.R(708937, 'receiver', receiver, false, false)) ? J$._() : J$.R(708945, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(712409, J$e);
                        } finally {
                            if (J$.Fr(712417))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(709121, arguments.callee, this, arguments);
                            arguments = J$.N(709129, 'arguments', arguments, true, false, false);
                            object = J$.N(709137, 'object', object, true, false, false);
                            property = J$.N(709145, 'property', property, true, false, false);
                            while (J$.C(45680, J$.U(44210, '!', J$.M(709049, J$.G(709025, J$.G(709017, J$.I(typeof Object === 'undefined' ? Object = J$.R(709009, 'Object', undefined, true, true) : Object = J$.R(709009, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(709033, 'object', object, false, false), J$.R(709041, 'property', property, false, false))))) {
                                object = J$.W(709081, 'object', J$.F(709073, J$.R(709057, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(709065, 'object', object, false, false)), object, false, false);
                                if (J$.C(45672, J$.B(44218, '===', J$.R(709089, 'object', object, false, false), J$.T(709097, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(709113, J$.R(709105, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(712425, J$e);
                        } finally {
                            if (J$.Fr(712433))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(709305, arguments.callee, this, arguments);
                            arguments = J$.N(709313, 'arguments', arguments, true, false, false);
                            o = J$.N(709321, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(709265, '_getPrototypeOf', J$.C(45696, J$.G(709161, J$.I(typeof Object === 'undefined' ? Object = J$.R(709153, 'Object', undefined, true, true) : Object = J$.R(709153, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(709177, J$.I(typeof Object === 'undefined' ? Object = J$.R(709169, 'Object', undefined, true, true) : Object = J$.R(709169, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(709257, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(709233, arguments.callee, this, arguments);
                                            arguments = J$.N(709241, 'arguments', arguments, true, false, false);
                                            o = J$.N(709249, 'o', o, true, false, false);
                                            return J$.Rt(709225, J$.C(45688, J$.G(709193, J$.R(709185, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(709217, J$.I(typeof Object === 'undefined' ? Object = J$.R(709201, 'Object', undefined, true, true) : Object = J$.R(709201, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(709209, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(712441, J$e);
                                        } finally {
                                            if (J$.Fr(712449))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(709297, J$.F(709289, J$.R(709273, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(709281, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(712457, J$e);
                        } finally {
                            if (J$.Fr(712465))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(709521, arguments.callee, this, arguments);
                            arguments = J$.N(709529, 'arguments', arguments, true, false, false);
                            subClass = J$.N(709537, 'subClass', subClass, true, false, false);
                            superClass = J$.N(709545, 'superClass', superClass, true, false, false);
                            if (J$.C(45712, J$.C(45704, J$.B(44234, '!==', J$.U(44226, 'typeof', J$.R(709329, 'superClass', superClass, false, false)), J$.T(709337, 'function', 21, false))) ? J$.B(44242, '!==', J$.R(709345, 'superClass', superClass, false, false), J$.T(709353, null, 25, false)) : J$._())) {
                                throw J$.F(709377, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(709361, 'TypeError', undefined, true, true) : TypeError = J$.R(709361, 'TypeError', TypeError, true, true)), true)(J$.T(709369, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(709473, J$.R(709385, 'subClass', subClass, false, false), 'prototype', J$.M(709465, J$.I(typeof Object === 'undefined' ? Object = J$.R(709393, 'Object', undefined, true, true) : Object = J$.R(709393, 'Object', Object, true, true)), 'create', false)(J$.C(45720, J$.R(709401, 'superClass', superClass, false, false)) ? J$.G(709417, J$.R(709409, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(709457, {
                                constructor: J$.T(709449, {
                                    value: J$.R(709425, 'subClass', subClass, false, false),
                                    writable: J$.T(709433, true, 23, false),
                                    configurable: J$.T(709441, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(45728, J$.R(709481, 'superClass', superClass, false, false)))
                                J$.F(709513, J$.R(709489, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(709497, 'subClass', subClass, false, false), J$.R(709505, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(712473, J$e);
                        } finally {
                            if (J$.Fr(712481))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(709697, arguments.callee, this, arguments);
                            arguments = J$.N(709705, 'arguments', arguments, true, false, false);
                            o = J$.N(709713, 'o', o, true, false, false);
                            p = J$.N(709721, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(709649, '_setPrototypeOf', J$.C(45736, J$.G(709561, J$.I(typeof Object === 'undefined' ? Object = J$.R(709553, 'Object', undefined, true, true) : Object = J$.R(709553, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(709641, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(709609, arguments.callee, this, arguments);
                                            arguments = J$.N(709617, 'arguments', arguments, true, false, false);
                                            o = J$.N(709625, 'o', o, true, false, false);
                                            p = J$.N(709633, 'p', p, true, false, false);
                                            J$.P(709585, J$.R(709569, 'o', o, false, false), '__proto__', J$.R(709577, 'p', p, false, false));
                                            return J$.Rt(709601, J$.R(709593, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(712489, J$e);
                                        } finally {
                                            if (J$.Fr(712497))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(709689, J$.F(709681, J$.R(709657, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(709665, 'o', o, false, false), J$.R(709673, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(712505, J$e);
                        } finally {
                            if (J$.Fr(712513))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function listCollectionsTransforms(databaseName) {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(710177, arguments.callee, this, arguments);
                            arguments = J$.N(710185, 'arguments', arguments, true, false, false);
                            databaseName = J$.N(710193, 'databaseName', databaseName, true, false, false);
                            J$.N(710201, 'matching', matching, false, false, false);
                            var matching = J$.W(709961, 'matching', J$.M(709953, J$.T(709929, '', 21, false), 'concat', false)(J$.R(709937, 'databaseName', databaseName, false, false), J$.T(709945, '.', 21, false)), matching, false, false);
                            return J$.Rt(710169, J$.T(710161, {
                                doc: J$.T(710153, function doc(_doc) {
                                    jalangiLabel16:
                                        while (true) {
                                            try {
                                                J$.Fe(710121, arguments.callee, this, arguments);
                                                arguments = J$.N(710129, 'arguments', arguments, true, false, false);
                                                _doc = J$.N(710137, '_doc', _doc, true, false, false);
                                                J$.N(710145, 'index', index, false, false, false);
                                                var index = J$.W(710001, 'index', J$.M(709993, J$.G(709977, J$.R(709969, '_doc', _doc, false, false), 'name'), 'indexOf', false)(J$.R(709985, 'matching', matching, false, false)), index, false, false);
                                                if (J$.C(45752, J$.C(45744, J$.G(710017, J$.R(710009, '_doc', _doc, false, false), 'name')) ? J$.B(44250, '===', J$.R(710025, 'index', index, false, false), J$.T(710033, 0, 22, false)) : J$._())) {
                                                    J$.P(710097, J$.R(710041, '_doc', _doc, false, false), 'name', J$.M(710089, J$.G(710057, J$.R(710049, '_doc', _doc, false, false), 'name'), 'substr', false)(J$.B(44258, '+', J$.R(710065, 'index', index, false, false), J$.G(710081, J$.R(710073, 'matching', matching, false, false), 'length'))));
                                                }
                                                return J$.Rt(710113, J$.R(710105, '_doc', _doc, false, false));
                                            } catch (J$e) {
                                                J$.Ex(712521, J$e);
                                            } finally {
                                                if (J$.Fr(712529))
                                                    continue jalangiLabel16;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(712537, J$e);
                        } finally {
                            if (J$.Fr(712545))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(712025, '_typeof', J$.T(712017, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(712041, '_classCallCheck', J$.T(712033, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(712057, '_defineProperties', J$.T(712049, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(712073, '_createClass', J$.T(712065, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(712089, '_possibleConstructorReturn', J$.T(712081, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(712105, '_assertThisInitialized', J$.T(712097, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(712121, '_get', J$.T(712113, _get, 12, false), true, false, false);
            _superPropBase = J$.N(712137, '_superPropBase', J$.T(712129, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(712153, '_getPrototypeOf', J$.T(712145, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(712169, '_inherits', J$.T(712161, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(712185, '_setPrototypeOf', J$.T(712177, _setPrototypeOf, 12, false), true, false, false);
            J$.N(712193, 'CommandOperationV2', CommandOperationV2, false, false, false);
            J$.N(712201, 'Aspect', Aspect, false, false, false);
            J$.N(712209, 'defineAspects', defineAspects, false, false, false);
            J$.N(712217, 'maxWireVersion', maxWireVersion, false, false, false);
            J$.N(712225, 'CONSTANTS', CONSTANTS, false, false, false);
            J$.N(712233, 'LIST_COLLECTIONS_WIRE_VERSION', LIST_COLLECTIONS_WIRE_VERSION, false, false, false);
            listCollectionsTransforms = J$.N(712249, 'listCollectionsTransforms', J$.T(712241, listCollectionsTransforms, 12, false), true, false, false);
            J$.N(712257, 'ListCollectionsOperation', ListCollectionsOperation, false, false, false);
            J$.T(707593, 'use strict', 21, false);
            var CommandOperationV2 = J$.W(709753, 'CommandOperationV2', J$.F(709745, J$.I(typeof require === 'undefined' ? require = J$.R(709729, 'require', undefined, true, true) : require = J$.R(709729, 'require', require, true, true)), false)(J$.T(709737, './command_v2', 21, false)), CommandOperationV2, false, true);
            var Aspect = J$.W(709793, 'Aspect', J$.G(709785, J$.F(709777, J$.I(typeof require === 'undefined' ? require = J$.R(709761, 'require', undefined, true, true) : require = J$.R(709761, 'require', require, true, true)), false)(J$.T(709769, './operation', 21, false)), 'Aspect'), Aspect, false, true);
            var defineAspects = J$.W(709833, 'defineAspects', J$.G(709825, J$.F(709817, J$.I(typeof require === 'undefined' ? require = J$.R(709801, 'require', undefined, true, true) : require = J$.R(709801, 'require', require, true, true)), false)(J$.T(709809, './operation', 21, false)), 'defineAspects'), defineAspects, false, true);
            var maxWireVersion = J$.W(709873, 'maxWireVersion', J$.G(709865, J$.F(709857, J$.I(typeof require === 'undefined' ? require = J$.R(709841, 'require', undefined, true, true) : require = J$.R(709841, 'require', require, true, true)), false)(J$.T(709849, '../core/utils', 21, false)), 'maxWireVersion'), maxWireVersion, false, true);
            var CONSTANTS = J$.W(709905, 'CONSTANTS', J$.F(709897, J$.I(typeof require === 'undefined' ? require = J$.R(709881, 'require', undefined, true, true) : require = J$.R(709881, 'require', require, true, true)), false)(J$.T(709889, '../constants', 21, false)), CONSTANTS, false, true);
            var LIST_COLLECTIONS_WIRE_VERSION = J$.W(709921, 'LIST_COLLECTIONS_WIRE_VERSION', J$.T(709913, 3, 22, false), LIST_COLLECTIONS_WIRE_VERSION, false, true);
            var ListCollectionsOperation = J$.W(711897, 'ListCollectionsOperation', J$.F(711889, J$.T(711873, function (_CommandOperationV) {
                jalangiLabel21:
                    while (true) {
                        try {
                            J$.Fe(711833, arguments.callee, this, arguments);
                            function ListCollectionsOperation(db, filter, options) {
                                jalangiLabel18:
                                    while (true) {
                                        try {
                                            J$.Fe(710553, arguments.callee, this, arguments);
                                            arguments = J$.N(710561, 'arguments', arguments, true, false, false);
                                            db = J$.N(710569, 'db', db, true, false, false);
                                            filter = J$.N(710577, 'filter', filter, true, false, false);
                                            options = J$.N(710585, 'options', options, true, false, false);
                                            J$.N(710593, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(710265, J$.R(710241, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(710249, 'this', this, false, false), J$.R(710257, 'ListCollectionsOperation', ListCollectionsOperation, false, false));
                                            _this = J$.W(710369, '_this', J$.F(710361, J$.R(710273, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(710281, 'this', this, false, false), J$.M(710353, J$.F(710305, J$.R(710289, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(710297, 'ListCollectionsOperation', ListCollectionsOperation, false, false)), 'call', false)(J$.R(710313, 'this', this, false, false), J$.R(710321, 'db', db, false, false), J$.R(710329, 'options', options, false, false), J$.T(710345, { fullResponse: J$.T(710337, true, 23, false) }, 11, false))), _this, false, false);
                                            J$.P(710393, J$.R(710377, '_this', _this, false, false), 'db', J$.R(710385, 'db', db, false, false));
                                            J$.P(710417, J$.R(710401, '_this', _this, false, false), 'filter', J$.R(710409, 'filter', filter, false, false));
                                            J$.P(710457, J$.R(710425, '_this', _this, false, false), 'nameOnly', J$.U(44274, '!', J$.U(44266, '!', J$.G(710449, J$.G(710441, J$.R(710433, '_this', _this, false, false), 'options'), 'nameOnly'))));
                                            if (J$.C(45760, J$.B(44290, '===', J$.U(44282, 'typeof', J$.G(710481, J$.G(710473, J$.R(710465, '_this', _this, false, false), 'options'), 'batchSize')), J$.T(710489, 'number', 21, false)))) {
                                                J$.P(710529, J$.R(710497, '_this', _this, false, false), 'batchSize', J$.G(710521, J$.G(710513, J$.R(710505, '_this', _this, false, false), 'options'), 'batchSize'));
                                            }
                                            return J$.Rt(710545, J$.R(710537, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(712553, J$e);
                                        } finally {
                                            if (J$.Fr(712561))
                                                continue jalangiLabel18;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(711841, 'arguments', arguments, true, false, false);
                            _CommandOperationV = J$.N(711849, '_CommandOperationV', _CommandOperationV, true, false, false);
                            ListCollectionsOperation = J$.N(711865, 'ListCollectionsOperation', J$.T(711857, ListCollectionsOperation, 12, false), true, false, false);
                            J$.F(710233, J$.R(710209, '_inherits', _inherits, false, true), false)(J$.R(710217, 'ListCollectionsOperation', ListCollectionsOperation, false, false), J$.R(710225, '_CommandOperationV', _CommandOperationV, false, false));
                            J$.F(711809, J$.R(710601, '_createClass', _createClass, false, true), false)(J$.R(710609, 'ListCollectionsOperation', ListCollectionsOperation, false, false), J$.T(711801, [J$.T(711793, {
                                    key: J$.T(710617, 'execute', 21, false),
                                    value: J$.T(711785, function execute(server, callback) {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(711721, arguments.callee, this, arguments);
                                                    arguments = J$.N(711729, 'arguments', arguments, true, false, false);
                                                    server = J$.N(711737, 'server', server, true, false, false);
                                                    callback = J$.N(711745, 'callback', callback, true, false, false);
                                                    J$.N(711753, 'filter', filter, false, false, false);
                                                    J$.N(711761, 'databaseName', databaseName, false, false, false);
                                                    J$.N(711769, 'transforms', transforms, false, false, false);
                                                    J$.N(711777, 'command', command, false, false, false);
                                                    if (J$.C(45840, J$.B(44298, '<', J$.F(710641, J$.R(710625, 'maxWireVersion', maxWireVersion, false, true), false)(J$.R(710633, 'server', server, false, false)), J$.R(710649, 'LIST_COLLECTIONS_WIRE_VERSION', LIST_COLLECTIONS_WIRE_VERSION, false, true)))) {
                                                        var filter = J$.W(710673, 'filter', J$.G(710665, J$.R(710657, 'this', this, false, false), 'filter'), filter, false, false);
                                                        var databaseName = J$.W(710721, 'databaseName', J$.G(710713, J$.G(710705, J$.G(710697, J$.G(710689, J$.R(710681, 'this', this, false, false), 'db'), 's'), 'namespace'), 'db'), databaseName, false, false);
                                                        if (J$.C(45776, J$.C(45768, J$.B(44314, '===', J$.U(44306, 'typeof', J$.G(710737, J$.R(710729, 'filter', filter, false, false), 'name')), J$.T(710745, 'string', 21, false))) ? J$.U(44338, '!', J$.M(710809, J$.F(710785, J$.I(typeof RegExp === 'undefined' ? RegExp = J$.R(710753, 'RegExp', undefined, true, true) : RegExp = J$.R(710753, 'RegExp', RegExp, true, true)), true)(J$.B(44330, '+', J$.B(44322, '+', J$.T(710761, '^', 21, false), J$.R(710769, 'databaseName', databaseName, false, false)), J$.T(710777, '\\.', 21, false))), 'test', false)(J$.G(710801, J$.R(710793, 'filter', filter, false, false), 'name'))) : J$._())) {
                                                            filter = J$.W(710849, 'filter', J$.M(710841, J$.I(typeof Object === 'undefined' ? Object = J$.R(710817, 'Object', undefined, true, true) : Object = J$.R(710817, 'Object', Object, true, true)), 'assign', false)(J$.T(710825, {}, 11, false), J$.R(710833, 'filter', filter, false, false)), filter, false, false);
                                                            J$.P(710929, J$.R(710857, 'filter', filter, false, false), 'name', J$.M(710921, J$.M(710913, J$.G(710889, J$.G(710881, J$.G(710873, J$.R(710865, 'this', this, false, false), 'db'), 's'), 'namespace'), 'withCollection', false)(J$.G(710905, J$.R(710897, 'filter', filter, false, false), 'name')), 'toString', false)());
                                                        }
                                                        if (J$.C(45784, J$.B(44346, '==', J$.R(710937, 'filter', filter, false, false), J$.T(710945, null, 25, false)))) {
                                                            J$.P(710993, J$.R(710953, 'filter', filter, false, false), 'name', J$.M(710985, J$.T(710961, '/', 21, false), 'concat', false)(J$.R(710969, 'databaseName', databaseName, false, false), J$.T(710977, '/', 21, false)));
                                                        }
                                                        if (J$.C(45792, J$.G(711009, J$.R(711001, 'filter', filter, false, false), 'name'))) {
                                                            filter = J$.W(711073, 'filter', J$.T(711065, {
                                                                $and: J$.T(711057, [
                                                                    J$.T(711033, { name: J$.G(711025, J$.R(711017, 'filter', filter, false, false), 'name') }, 11, false),
                                                                    J$.T(711049, { name: J$.T(711041, /^((?!\$).)*$/, 14, false) }, 11, false)
                                                                ], 10, false)
                                                            }, 11, false), filter, false, false);
                                                        } else {
                                                            filter = J$.W(711097, 'filter', J$.T(711089, { name: J$.T(711081, /^((?!\$).)*$/, 14, false) }, 11, false), filter, false, false);
                                                        }
                                                        var transforms = J$.W(711129, 'transforms', J$.F(711121, J$.R(711105, 'listCollectionsTransforms', listCollectionsTransforms, false, true), false)(J$.R(711113, 'databaseName', databaseName, false, false)), transforms, false, false);
                                                        J$.M(711489, J$.R(711137, 'server', server, false, false), 'query', false)(J$.M(711193, J$.M(711169, J$.T(711145, '', 21, false), 'concat', false)(J$.R(711153, 'databaseName', databaseName, false, false), J$.T(711161, '.', 21, false)), 'concat', false)(J$.G(711185, J$.R(711177, 'CONSTANTS', CONSTANTS, false, true), 'SYSTEM_NAMESPACE_COLLECTION')), J$.T(711209, { query: J$.R(711201, 'filter', filter, false, false) }, 11, false), J$.T(711241, { batchSize: J$.C(45800, J$.G(711225, J$.R(711217, 'this', this, false, false), 'batchSize')) ? J$._() : J$.T(711233, 1000, 22, false) }, 11, false), J$.T(711249, {}, 11, false), J$.T(711481, function (err, result) {
                                                            jalangiLabel19:
                                                                while (true) {
                                                                    try {
                                                                        J$.Fe(711449, arguments.callee, this, arguments);
                                                                        arguments = J$.N(711457, 'arguments', arguments, true, false, false);
                                                                        err = J$.N(711465, 'err', err, true, false, false);
                                                                        result = J$.N(711473, 'result', result, true, false, false);
                                                                        if (J$.C(45832, J$.C(45824, J$.C(45816, J$.C(45808, J$.R(711257, 'result', result, false, false)) ? J$.G(711273, J$.R(711265, 'result', result, false, false), 'message') : J$._()) ? J$.G(711297, J$.G(711289, J$.R(711281, 'result', result, false, false), 'message'), 'documents') : J$._()) ? J$.M(711337, J$.I(typeof Array === 'undefined' ? Array = J$.R(711305, 'Array', undefined, true, true) : Array = J$.R(711305, 'Array', Array, true, true)), 'isArray', false)(J$.G(711329, J$.G(711321, J$.R(711313, 'result', result, false, false), 'message'), 'documents')) : J$._())) {
                                                                            J$.P(711409, J$.G(711353, J$.R(711345, 'result', result, false, false), 'message'), 'documents', J$.M(711401, J$.G(711377, J$.G(711369, J$.R(711361, 'result', result, false, false), 'message'), 'documents'), 'map', false)(J$.G(711393, J$.R(711385, 'transforms', transforms, false, false), 'doc')));
                                                                        }
                                                                        J$.F(711441, J$.R(711417, 'callback', callback, false, false), false)(J$.R(711425, 'err', err, false, false), J$.R(711433, 'result', result, false, false));
                                                                    } catch (J$e) {
                                                                        J$.Ex(712569, J$e);
                                                                    } finally {
                                                                        if (J$.Fr(712577))
                                                                            continue jalangiLabel19;
                                                                        else
                                                                            return J$.Ra();
                                                                    }
                                                                }
                                                        }, 12, false));
                                                        return J$.Rt(711497, undefined);
                                                    }
                                                    var command = J$.W(711601, 'command', J$.T(711593, {
                                                        listCollections: J$.T(711505, 1, 22, false),
                                                        filter: J$.G(711521, J$.R(711513, 'this', this, false, false), 'filter'),
                                                        cursor: J$.C(45848, J$.G(711537, J$.R(711529, 'this', this, false, false), 'batchSize')) ? J$.T(711561, { batchSize: J$.G(711553, J$.R(711545, 'this', this, false, false), 'batchSize') }, 11, false) : J$.T(711569, {}, 11, false),
                                                        nameOnly: J$.G(711585, J$.R(711577, 'this', this, false, false), 'nameOnly')
                                                    }, 11, false), command, false, false);
                                                    return J$.Rt(711713, J$.M(711705, J$.F(711665, J$.R(711609, '_get', _get, false, true), false)(J$.F(711641, J$.R(711617, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(711633, J$.R(711625, 'ListCollectionsOperation', ListCollectionsOperation, false, false), 'prototype')), J$.T(711649, 'executeCommand', 21, false), J$.R(711657, 'this', this, false, false)), 'call', false)(J$.R(711673, 'this', this, false, false), J$.R(711681, 'server', server, false, false), J$.R(711689, 'command', command, false, false), J$.R(711697, 'callback', callback, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(712585, J$e);
                                                } finally {
                                                    if (J$.Fr(712593))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(711825, J$.R(711817, 'ListCollectionsOperation', ListCollectionsOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(712601, J$e);
                        } finally {
                            if (J$.Fr(712609))
                                continue jalangiLabel21;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(711881, 'CommandOperationV2', CommandOperationV2, false, true)), ListCollectionsOperation, false, true);
            J$.F(711977, J$.R(711905, 'defineAspects', defineAspects, false, true), false)(J$.R(711913, 'ListCollectionsOperation', ListCollectionsOperation, false, true), J$.T(711969, [
                J$.G(711929, J$.R(711921, 'Aspect', Aspect, false, true), 'READ_OPERATION'),
                J$.G(711945, J$.R(711937, 'Aspect', Aspect, false, true), 'RETRYABLE'),
                J$.G(711961, J$.R(711953, 'Aspect', Aspect, false, true), 'EXECUTE_WITH_SELECTION')
            ], 10, false));
            J$.P(712001, J$.I(typeof module === 'undefined' ? module = J$.R(711985, 'module', undefined, true, true) : module = J$.R(711985, 'module', module, true, true)), 'exports', J$.R(711993, 'ListCollectionsOperation', ListCollectionsOperation, false, true));
        } catch (J$e) {
            J$.Ex(712617, J$e);
        } finally {
            if (J$.Sr(712625))
                continue jalangiLabel22;
            else
                break jalangiLabel22;
        }
    }
// JALANGI DO NOT INSTRUMENT
