// mongodb/lib/operations/rename.js
J$.noInstrEval = false;
jalangiLabel17:
    while (true) {
        try {
            J$.Se(748137, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/operations/rename.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(745137, arguments.callee, this, arguments);
                            arguments = J$.N(745145, 'arguments', arguments, true, false, false);
                            obj = J$.N(745153, 'obj', obj, true, false, false);
                            if (J$.C(48112, J$.C(48072, J$.B(46514, '===', J$.U(46506, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(744873, 'Symbol', undefined, true, true) : Symbol = J$.R(744873, 'Symbol', Symbol, true, true))), J$.T(744881, 'function', 21, false))) ? J$.B(46530, '===', J$.U(46522, 'typeof', J$.G(744897, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(744889, 'Symbol', undefined, true, true) : Symbol = J$.R(744889, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(744905, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(744961, '_typeof', J$.T(744953, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(744929, arguments.callee, this, arguments);
                                                arguments = J$.N(744937, 'arguments', arguments, true, false, false);
                                                obj = J$.N(744945, 'obj', obj, true, false, false);
                                                return J$.Rt(744921, J$.U(46538, 'typeof', J$.R(744913, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(748353, J$e);
                                            } finally {
                                                if (J$.Fr(748361))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(745097, '_typeof', J$.T(745089, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(745065, arguments.callee, this, arguments);
                                                arguments = J$.N(745073, 'arguments', arguments, true, false, false);
                                                obj = J$.N(745081, 'obj', obj, true, false, false);
                                                return J$.Rt(745057, J$.C(48104, J$.C(48096, J$.C(48088, J$.C(48080, J$.R(744969, 'obj', obj, false, false)) ? J$.B(46554, '===', J$.U(46546, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(744977, 'Symbol', undefined, true, true) : Symbol = J$.R(744977, 'Symbol', Symbol, true, true))), J$.T(744985, 'function', 21, false)) : J$._()) ? J$.B(46562, '===', J$.G(745001, J$.R(744993, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(745009, 'Symbol', undefined, true, true) : Symbol = J$.R(745009, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(46570, '!==', J$.R(745017, 'obj', obj, false, false), J$.G(745033, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(745025, 'Symbol', undefined, true, true) : Symbol = J$.R(745025, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(745041, 'symbol', 21, false) : J$.U(46578, 'typeof', J$.R(745049, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(748369, J$e);
                                            } finally {
                                                if (J$.Fr(748377))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(745129, J$.F(745121, J$.R(745105, '_typeof', _typeof, false, true), false)(J$.R(745113, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(748385, J$e);
                        } finally {
                            if (J$.Fr(748393))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(745201, arguments.callee, this, arguments);
                            arguments = J$.N(745209, 'arguments', arguments, true, false, false);
                            instance = J$.N(745217, 'instance', instance, true, false, false);
                            Constructor = J$.N(745225, 'Constructor', Constructor, true, false, false);
                            if (J$.C(48120, J$.U(46594, '!', J$.B(46586, 'instanceof', J$.R(745161, 'instance', instance, false, false), J$.R(745169, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(745193, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(745177, 'TypeError', undefined, true, true) : TypeError = J$.R(745177, 'TypeError', TypeError, true, true)), true)(J$.T(745185, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(748401, J$e);
                        } finally {
                            if (J$.Fr(748409))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(745473, arguments.callee, this, arguments);
                            arguments = J$.N(745481, 'arguments', arguments, true, false, false);
                            target = J$.N(745489, 'target', target, true, false, false);
                            props = J$.N(745497, 'props', props, true, false, false);
                            J$.N(745505, 'i', i, false, false, false);
                            J$.N(745513, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(745241, 'i', J$.T(745233, 0, 22, false), i, false, false); J$.C(48144, J$.B(46602, '<', J$.R(745249, 'i', i, false, false), J$.G(745265, J$.R(745257, 'props', props, false, false), 'length'))); J$.B(46626, '-', i = J$.W(745281, 'i', J$.B(46618, '+', J$.U(46610, '+', J$.R(745273, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(745313, 'descriptor', J$.G(745305, J$.R(745289, 'props', props, false, false), J$.R(745297, 'i', i, false, false)), descriptor, false, false);
                                J$.P(745353, J$.R(745321, 'descriptor', descriptor, false, false), 'enumerable', J$.C(48128, J$.G(745337, J$.R(745329, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(745345, false, 23, false));
                                J$.P(745377, J$.R(745361, 'descriptor', descriptor, false, false), 'configurable', J$.T(745369, true, 23, false));
                                if (J$.C(48136, J$.B(46634, 'in', J$.T(745385, 'value', 21, false), J$.R(745393, 'descriptor', descriptor, false, false))))
                                    J$.P(745417, J$.R(745401, 'descriptor', descriptor, false, false), 'writable', J$.T(745409, true, 23, false));
                                J$.M(745465, J$.I(typeof Object === 'undefined' ? Object = J$.R(745425, 'Object', undefined, true, true) : Object = J$.R(745425, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(745433, 'target', target, false, false), J$.G(745449, J$.R(745441, 'descriptor', descriptor, false, false), 'key'), J$.R(745457, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(748417, J$e);
                        } finally {
                            if (J$.Fr(748425))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(745625, arguments.callee, this, arguments);
                            arguments = J$.N(745633, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(745641, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(745649, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(745657, 'staticProps', staticProps, true, false, false);
                            if (J$.C(48152, J$.R(745521, 'protoProps', protoProps, false, false)))
                                J$.F(745561, J$.R(745529, '_defineProperties', _defineProperties, false, true), false)(J$.G(745545, J$.R(745537, 'Constructor', Constructor, false, false), 'prototype'), J$.R(745553, 'protoProps', protoProps, false, false));
                            if (J$.C(48160, J$.R(745569, 'staticProps', staticProps, false, false)))
                                J$.F(745601, J$.R(745577, '_defineProperties', _defineProperties, false, true), false)(J$.R(745585, 'Constructor', Constructor, false, false), J$.R(745593, 'staticProps', staticProps, false, false));
                            return J$.Rt(745617, J$.R(745609, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(748433, J$e);
                        } finally {
                            if (J$.Fr(748441))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(745769, arguments.callee, this, arguments);
                            arguments = J$.N(745777, 'arguments', arguments, true, false, false);
                            self = J$.N(745785, 'self', self, true, false, false);
                            call = J$.N(745793, 'call', call, true, false, false);
                            if (J$.C(48184, J$.C(48176, J$.R(745665, 'call', call, false, false)) ? J$.C(48168, J$.B(46642, '===', J$.F(745689, J$.R(745673, '_typeof', _typeof, false, true), false)(J$.R(745681, 'call', call, false, false)), J$.T(745697, 'object', 21, false))) ? J$._() : J$.B(46658, '===', J$.U(46650, 'typeof', J$.R(745705, 'call', call, false, false)), J$.T(745713, 'function', 21, false)) : J$._())) {
                                return J$.Rt(745729, J$.R(745721, 'call', call, false, false));
                            }
                            return J$.Rt(745761, J$.F(745753, J$.R(745737, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(745745, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(748449, J$e);
                        } finally {
                            if (J$.Fr(748457))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(745857, arguments.callee, this, arguments);
                            arguments = J$.N(745865, 'arguments', arguments, true, false, false);
                            self = J$.N(745873, 'self', self, true, false, false);
                            if (J$.C(48192, J$.B(46666, '===', J$.R(745801, 'self', self, false, false), void J$.T(745809, 0, 22, false)))) {
                                throw J$.F(745833, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(745817, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(745817, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(745825, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(745849, J$.R(745841, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(748465, J$e);
                        } finally {
                            if (J$.Fr(748473))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(746033, arguments.callee, this, arguments);
                            arguments = J$.N(746041, 'arguments', arguments, true, false, false);
                            o = J$.N(746049, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(745993, '_getPrototypeOf', J$.C(48208, J$.G(745889, J$.I(typeof Object === 'undefined' ? Object = J$.R(745881, 'Object', undefined, true, true) : Object = J$.R(745881, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(745905, J$.I(typeof Object === 'undefined' ? Object = J$.R(745897, 'Object', undefined, true, true) : Object = J$.R(745897, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(745985, function _getPrototypeOf(o) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(745961, arguments.callee, this, arguments);
                                            arguments = J$.N(745969, 'arguments', arguments, true, false, false);
                                            o = J$.N(745977, 'o', o, true, false, false);
                                            return J$.Rt(745953, J$.C(48200, J$.G(745921, J$.R(745913, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(745945, J$.I(typeof Object === 'undefined' ? Object = J$.R(745929, 'Object', undefined, true, true) : Object = J$.R(745929, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(745937, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(748481, J$e);
                                        } finally {
                                            if (J$.Fr(748489))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(746025, J$.F(746017, J$.R(746001, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(746009, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(748497, J$e);
                        } finally {
                            if (J$.Fr(748505))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(746249, arguments.callee, this, arguments);
                            arguments = J$.N(746257, 'arguments', arguments, true, false, false);
                            subClass = J$.N(746265, 'subClass', subClass, true, false, false);
                            superClass = J$.N(746273, 'superClass', superClass, true, false, false);
                            if (J$.C(48224, J$.C(48216, J$.B(46682, '!==', J$.U(46674, 'typeof', J$.R(746057, 'superClass', superClass, false, false)), J$.T(746065, 'function', 21, false))) ? J$.B(46690, '!==', J$.R(746073, 'superClass', superClass, false, false), J$.T(746081, null, 25, false)) : J$._())) {
                                throw J$.F(746105, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(746089, 'TypeError', undefined, true, true) : TypeError = J$.R(746089, 'TypeError', TypeError, true, true)), true)(J$.T(746097, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(746201, J$.R(746113, 'subClass', subClass, false, false), 'prototype', J$.M(746193, J$.I(typeof Object === 'undefined' ? Object = J$.R(746121, 'Object', undefined, true, true) : Object = J$.R(746121, 'Object', Object, true, true)), 'create', false)(J$.C(48232, J$.R(746129, 'superClass', superClass, false, false)) ? J$.G(746145, J$.R(746137, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(746185, {
                                constructor: J$.T(746177, {
                                    value: J$.R(746153, 'subClass', subClass, false, false),
                                    writable: J$.T(746161, true, 23, false),
                                    configurable: J$.T(746169, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(48240, J$.R(746209, 'superClass', superClass, false, false)))
                                J$.F(746241, J$.R(746217, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(746225, 'subClass', subClass, false, false), J$.R(746233, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(748513, J$e);
                        } finally {
                            if (J$.Fr(748521))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(746425, arguments.callee, this, arguments);
                            arguments = J$.N(746433, 'arguments', arguments, true, false, false);
                            o = J$.N(746441, 'o', o, true, false, false);
                            p = J$.N(746449, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(746377, '_setPrototypeOf', J$.C(48248, J$.G(746289, J$.I(typeof Object === 'undefined' ? Object = J$.R(746281, 'Object', undefined, true, true) : Object = J$.R(746281, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(746369, function _setPrototypeOf(o, p) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(746337, arguments.callee, this, arguments);
                                            arguments = J$.N(746345, 'arguments', arguments, true, false, false);
                                            o = J$.N(746353, 'o', o, true, false, false);
                                            p = J$.N(746361, 'p', p, true, false, false);
                                            J$.P(746313, J$.R(746297, 'o', o, false, false), '__proto__', J$.R(746305, 'p', p, false, false));
                                            return J$.Rt(746329, J$.R(746321, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(748529, J$e);
                                        } finally {
                                            if (J$.Fr(748537))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(746417, J$.F(746409, J$.R(746385, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(746393, 'o', o, false, false), J$.R(746401, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(748545, J$e);
                        } finally {
                            if (J$.Fr(748553))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(748153, '_typeof', J$.T(748145, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(748169, '_classCallCheck', J$.T(748161, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(748185, '_defineProperties', J$.T(748177, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(748201, '_createClass', J$.T(748193, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(748217, '_possibleConstructorReturn', J$.T(748209, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(748233, '_assertThisInitialized', J$.T(748225, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(748249, '_getPrototypeOf', J$.T(748241, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(748265, '_inherits', J$.T(748257, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(748281, '_setPrototypeOf', J$.T(748273, _setPrototypeOf, 12, false), true, false, false);
            J$.N(748289, 'OperationBase', OperationBase, false, false, false);
            J$.N(748297, 'applyWriteConcern', applyWriteConcern, false, false, false);
            J$.N(748305, 'checkCollectionName', checkCollectionName, false, false, false);
            J$.N(748313, 'executeDbAdminCommand', executeDbAdminCommand, false, false, false);
            J$.N(748321, 'handleCallback', handleCallback, false, false, false);
            J$.N(748329, 'loadCollection', loadCollection, false, false, false);
            J$.N(748337, 'toError', toError, false, false, false);
            J$.N(748345, 'RenameOperation', RenameOperation, false, false, false);
            J$.T(744865, 'use strict', 21, false);
            var OperationBase = J$.W(746489, 'OperationBase', J$.G(746481, J$.F(746473, J$.I(typeof require === 'undefined' ? require = J$.R(746457, 'require', undefined, true, true) : require = J$.R(746457, 'require', require, true, true)), false)(J$.T(746465, './operation', 21, false)), 'OperationBase'), OperationBase, false, true);
            var applyWriteConcern = J$.W(746529, 'applyWriteConcern', J$.G(746521, J$.F(746513, J$.I(typeof require === 'undefined' ? require = J$.R(746497, 'require', undefined, true, true) : require = J$.R(746497, 'require', require, true, true)), false)(J$.T(746505, '../utils', 21, false)), 'applyWriteConcern'), applyWriteConcern, false, true);
            var checkCollectionName = J$.W(746569, 'checkCollectionName', J$.G(746561, J$.F(746553, J$.I(typeof require === 'undefined' ? require = J$.R(746537, 'require', undefined, true, true) : require = J$.R(746537, 'require', require, true, true)), false)(J$.T(746545, '../utils', 21, false)), 'checkCollectionName'), checkCollectionName, false, true);
            var executeDbAdminCommand = J$.W(746609, 'executeDbAdminCommand', J$.G(746601, J$.F(746593, J$.I(typeof require === 'undefined' ? require = J$.R(746577, 'require', undefined, true, true) : require = J$.R(746577, 'require', require, true, true)), false)(J$.T(746585, './db_ops', 21, false)), 'executeDbAdminCommand'), executeDbAdminCommand, false, true);
            var handleCallback = J$.W(746649, 'handleCallback', J$.G(746641, J$.F(746633, J$.I(typeof require === 'undefined' ? require = J$.R(746617, 'require', undefined, true, true) : require = J$.R(746617, 'require', require, true, true)), false)(J$.T(746625, '../utils', 21, false)), 'handleCallback'), handleCallback, false, true);
            var loadCollection = J$.W(746689, 'loadCollection', J$.G(746681, J$.F(746673, J$.I(typeof require === 'undefined' ? require = J$.R(746657, 'require', undefined, true, true) : require = J$.R(746657, 'require', require, true, true)), false)(J$.T(746665, '../dynamic_loaders', 21, false)), 'loadCollection'), loadCollection, false, true);
            var toError = J$.W(746729, 'toError', J$.G(746721, J$.F(746713, J$.I(typeof require === 'undefined' ? require = J$.R(746697, 'require', undefined, true, true) : require = J$.R(746697, 'require', require, true, true)), false)(J$.T(746705, '../utils', 21, false)), 'toError'), toError, false, true);
            var RenameOperation = J$.W(748105, 'RenameOperation', J$.F(748097, J$.T(748081, function (_OperationBase) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(748041, arguments.callee, this, arguments);
                            function RenameOperation(collection, newName, options) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(746945, arguments.callee, this, arguments);
                                            arguments = J$.N(746953, 'arguments', arguments, true, false, false);
                                            collection = J$.N(746961, 'collection', collection, true, false, false);
                                            newName = J$.N(746969, 'newName', newName, true, false, false);
                                            options = J$.N(746977, 'options', options, true, false, false);
                                            J$.N(746985, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(746793, J$.R(746769, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(746777, 'this', this, false, false), J$.R(746785, 'RenameOperation', RenameOperation, false, false));
                                            _this = J$.W(746873, '_this', J$.F(746865, J$.R(746801, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(746809, 'this', this, false, false), J$.M(746857, J$.F(746833, J$.R(746817, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(746825, 'RenameOperation', RenameOperation, false, false)), 'call', false)(J$.R(746841, 'this', this, false, false), J$.R(746849, 'options', options, false, false))), _this, false, false);
                                            J$.P(746897, J$.R(746881, '_this', _this, false, false), 'collection', J$.R(746889, 'collection', collection, false, false));
                                            J$.P(746921, J$.R(746905, '_this', _this, false, false), 'newName', J$.R(746913, 'newName', newName, false, false));
                                            return J$.Rt(746937, J$.R(746929, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(748561, J$e);
                                        } finally {
                                            if (J$.Fr(748569))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(748049, 'arguments', arguments, true, false, false);
                            _OperationBase = J$.N(748057, '_OperationBase', _OperationBase, true, false, false);
                            RenameOperation = J$.N(748073, 'RenameOperation', J$.T(748065, RenameOperation, 12, false), true, false, false);
                            J$.F(746761, J$.R(746737, '_inherits', _inherits, false, true), false)(J$.R(746745, 'RenameOperation', RenameOperation, false, false), J$.R(746753, '_OperationBase', _OperationBase, false, false));
                            J$.F(748017, J$.R(746993, '_createClass', _createClass, false, true), false)(J$.R(747001, 'RenameOperation', RenameOperation, false, false), J$.T(748009, [J$.T(748001, {
                                    key: J$.T(747009, 'execute', 21, false),
                                    value: J$.T(747993, function execute(callback) {
                                        jalangiLabel15:
                                            while (true) {
                                                try {
                                                    J$.Fe(747905, arguments.callee, this, arguments);
                                                    arguments = J$.N(747913, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(747921, 'callback', callback, true, false, false);
                                                    J$.N(747929, 'coll', coll, false, false, false);
                                                    J$.N(747937, 'newName', newName, false, false, false);
                                                    J$.N(747945, 'options', options, false, false, false);
                                                    J$.N(747953, 'Collection', Collection, false, false, false);
                                                    J$.N(747961, 'renameCollection', renameCollection, false, false, false);
                                                    J$.N(747969, 'toCollection', toCollection, false, false, false);
                                                    J$.N(747977, 'dropTarget', dropTarget, false, false, false);
                                                    J$.N(747985, 'cmd', cmd, false, false, false);
                                                    var coll = J$.W(747033, 'coll', J$.G(747025, J$.R(747017, 'this', this, false, false), 'collection'), coll, false, false);
                                                    var newName = J$.W(747057, 'newName', J$.G(747049, J$.R(747041, 'this', this, false, false), 'newName'), newName, false, false);
                                                    var options = J$.W(747081, 'options', J$.G(747073, J$.R(747065, 'this', this, false, false), 'options'), options, false, false);
                                                    var Collection = J$.W(747105, 'Collection', J$.F(747097, J$.R(747089, 'loadCollection', loadCollection, false, true), false)(), Collection, false, false);
                                                    J$.F(747129, J$.R(747113, 'checkCollectionName', checkCollectionName, false, true), false)(J$.R(747121, 'newName', newName, false, false));
                                                    var renameCollection = J$.W(747153, 'renameCollection', J$.G(747145, J$.R(747137, 'coll', coll, false, false), 'namespace'), renameCollection, false, false);
                                                    var toCollection = J$.W(747209, 'toCollection', J$.M(747201, J$.M(747193, J$.G(747177, J$.G(747169, J$.R(747161, 'coll', coll, false, false), 's'), 'namespace'), 'withCollection', false)(J$.R(747185, 'newName', newName, false, false)), 'toString', false)(), toCollection, false, false);
                                                    var dropTarget = J$.W(747265, 'dropTarget', J$.C(48256, J$.B(46706, '===', J$.U(46698, 'typeof', J$.G(747225, J$.R(747217, 'options', options, false, false), 'dropTarget')), J$.T(747233, 'boolean', 21, false))) ? J$.G(747249, J$.R(747241, 'options', options, false, false), 'dropTarget') : J$.T(747257, false, 23, false), dropTarget, false, false);
                                                    var cmd = J$.W(747305, 'cmd', J$.T(747297, {
                                                        renameCollection: J$.R(747273, 'renameCollection', renameCollection, false, false),
                                                        to: J$.R(747281, 'toCollection', toCollection, false, false),
                                                        dropTarget: J$.R(747289, 'dropTarget', dropTarget, false, false)
                                                    }, 11, false), cmd, false, false);
                                                    J$.F(747377, J$.R(747313, 'applyWriteConcern', applyWriteConcern, false, true), false)(J$.R(747321, 'cmd', cmd, false, false), J$.T(747361, {
                                                        db: J$.G(747345, J$.G(747337, J$.R(747329, 'coll', coll, false, false), 's'), 'db'),
                                                        collection: J$.R(747353, 'coll', coll, false, false)
                                                    }, 11, false), J$.R(747369, 'options', options, false, false));
                                                    J$.F(747897, J$.R(747385, 'executeDbAdminCommand', executeDbAdminCommand, false, true), false)(J$.G(747433, J$.G(747425, J$.M(747417, J$.G(747409, J$.G(747401, J$.R(747393, 'coll', coll, false, false), 's'), 'db'), 'admin', false)(), 's'), 'db'), J$.R(747441, 'cmd', cmd, false, false), J$.R(747449, 'options', options, false, false), J$.T(747889, function (err, doc) {
                                                        jalangiLabel14:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(747857, arguments.callee, this, arguments);
                                                                    arguments = J$.N(747865, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(747873, 'err', err, true, false, false);
                                                                    doc = J$.N(747881, 'doc', doc, true, false, false);
                                                                    if (J$.C(48264, J$.R(747457, 'err', err, false, false)))
                                                                        return J$.Rt(747505, J$.F(747497, J$.R(747465, 'handleCallback', handleCallback, false, true), false)(J$.R(747473, 'callback', callback, false, false), J$.R(747481, 'err', err, false, false), J$.T(747489, null, 25, false)));
                                                                    if (J$.C(48272, J$.G(747521, J$.R(747513, 'doc', doc, false, false), 'errmsg')))
                                                                        return J$.Rt(747585, J$.F(747577, J$.R(747529, 'handleCallback', handleCallback, false, true), false)(J$.R(747537, 'callback', callback, false, false), J$.F(747561, J$.R(747545, 'toError', toError, false, true), false)(J$.R(747553, 'doc', doc, false, false)), J$.T(747569, null, 25, false)));
                                                                    try {
                                                                        return J$.Rt(747777, J$.F(747769, J$.R(747593, 'handleCallback', handleCallback, false, true), false)(J$.R(747601, 'callback', callback, false, false), J$.T(747609, null, 25, false), J$.F(747761, J$.R(747617, 'Collection', Collection, false, false), true)(J$.G(747641, J$.G(747633, J$.R(747625, 'coll', coll, false, false), 's'), 'db'), J$.G(747665, J$.G(747657, J$.R(747649, 'coll', coll, false, false), 's'), 'topology'), J$.G(747697, J$.G(747689, J$.G(747681, J$.R(747673, 'coll', coll, false, false), 's'), 'namespace'), 'db'), J$.R(747705, 'newName', newName, false, false), J$.G(747729, J$.G(747721, J$.R(747713, 'coll', coll, false, false), 's'), 'pkFactory'), J$.G(747753, J$.G(747745, J$.R(747737, 'coll', coll, false, false), 's'), 'options'))));
                                                                    } catch (err) {
                                                                        J$.N(747849, 'err', err, false, false, true);
                                                                        return J$.Rt(747841, J$.F(747833, J$.R(747785, 'handleCallback', handleCallback, false, true), false)(J$.R(747793, 'callback', callback, false, false), J$.F(747817, J$.R(747801, 'toError', toError, false, true), false)(J$.R(747809, 'err', err, false, false)), J$.T(747825, null, 25, false)));
                                                                    }
                                                                } catch (J$e) {
                                                                    J$.Ex(748577, J$e);
                                                                } finally {
                                                                    if (J$.Fr(748585))
                                                                        continue jalangiLabel14;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(748593, J$e);
                                                } finally {
                                                    if (J$.Fr(748601))
                                                        continue jalangiLabel15;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(748033, J$.R(748025, 'RenameOperation', RenameOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(748609, J$e);
                        } finally {
                            if (J$.Fr(748617))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(748089, 'OperationBase', OperationBase, false, true)), RenameOperation, false, true);
            J$.P(748129, J$.I(typeof module === 'undefined' ? module = J$.R(748113, 'module', undefined, true, true) : module = J$.R(748113, 'module', module, true, true)), 'exports', J$.R(748121, 'RenameOperation', RenameOperation, false, true));
        } catch (J$e) {
            J$.Ex(748625, J$e);
        } finally {
            if (J$.Sr(748633))
                continue jalangiLabel17;
            else
                break jalangiLabel17;
        }
    }
// JALANGI DO NOT INSTRUMENT
