// bungle/lib/pipes/input-names.js
import crypto from 'crypto';

import { BasePipe } from './base';


export class InputNamesPipe extends BasePipe {
    constructor() {
        super(...arguments);
    }

    mtime(/*file*/) {
        const mtime = crypto.createHmac('md5', '');
        for (let name of Object.keys(this.state.remoteFiles)) {
            mtime.update(name);
        }
        return mtime.digest('hex');
    }

    async start() {
        if (!this.exists(this.config.filename)) {
            await this.fileAdd(this.config.filename);
            await this.fileChange(this.config.filename);
        } else {
            const file = this.state.localFiles[this.config.filename];
            if (file.mtime != this.mtime()) {
                await this.fileChange(this.config.filename);
            }
        }
        return super.start();
    }

    async remoteAdd(file) {
        file.addCount++;

        if (this.started) {
            await this.fileChange(this.config.filename);
        }

        if (this.config.passthrough) {
            return super.remoteAdd(file);
        } else {
            return file;
        }
    }

    async remoteChange(file) {
        if (this.config.passthrough) {
            return super.remoteChange(file);
        } else {
            return file;
        }
    }

    async remoteUnlink(file) {
        file.addCount--;

        if (this.started) {
            await this.fileChange(this.config.filename);
        }

        if (this.config.passthrough) {
            return super.remoteUnlink(file);
        } else {
            return file;
        }
    }
}
