// colors/themes/generic-logging.js
J$.noInstrEval = false;
jalangiLabel0:
    while (true) {
        try {
            J$.Se(7075553, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/colors/themes/generic-logging.js');
            J$.P(7075545, J$.I(typeof module === 'undefined' ? module = J$.R(7075441, 'module', undefined, true, true) : module = J$.R(7075441, 'module', module, true, true)), J$.T(7075449, 'exports', 21, false), J$.T(7075537, {
                silly: J$.T(7075457, 'rainbow', 21, false),
                input: J$.T(7075465, 'grey', 21, false),
                verbose: J$.T(7075473, 'cyan', 21, false),
                prompt: J$.T(7075481, 'grey', 21, false),
                info: J$.T(7075489, 'green', 21, false),
                data: J$.T(7075497, 'grey', 21, false),
                help: J$.T(7075505, 'cyan', 21, false),
                warn: J$.T(7075513, 'yellow', 21, false),
                debug: J$.T(7075521, 'blue', 21, false),
                error: J$.T(7075529, 'red', 21, false)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(7075561, J$e);
        } finally {
            if (J$.Sr(7075569))
                continue jalangiLabel0;
            else
                break jalangiLabel0;
        }
    }
// JALANGI DO NOT INSTRUMENT
