// crc/index.js
"use strict";

require("core-js/modules/es.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "crc1", {
  enumerable: true,
  get: function get() {
    return _crc["default"];
  }
});
Object.defineProperty(exports, "crc8", {
  enumerable: true,
  get: function get() {
    return _crc2["default"];
  }
});
Object.defineProperty(exports, "crc81wire", {
  enumerable: true,
  get: function get() {
    return _crc81wire["default"];
  }
});
Object.defineProperty(exports, "crc16", {
  enumerable: true,
  get: function get() {
    return _crc3["default"];
  }
});
Object.defineProperty(exports, "crc16ccitt", {
  enumerable: true,
  get: function get() {
    return _crc16ccitt["default"];
  }
});
Object.defineProperty(exports, "crc16modbus", {
  enumerable: true,
  get: function get() {
    return _crc16modbus["default"];
  }
});
Object.defineProperty(exports, "crc16xmodem", {
  enumerable: true,
  get: function get() {
    return _crc16xmodem["default"];
  }
});
Object.defineProperty(exports, "crc16kermit", {
  enumerable: true,
  get: function get() {
    return _crc16kermit["default"];
  }
});
Object.defineProperty(exports, "crc24", {
  enumerable: true,
  get: function get() {
    return _crc4["default"];
  }
});
Object.defineProperty(exports, "crc32", {
  enumerable: true,
  get: function get() {
    return _crc5["default"];
  }
});
Object.defineProperty(exports, "crcjam", {
  enumerable: true,
  get: function get() {
    return _crcjam["default"];
  }
});
exports["default"] = void 0;

var _crc = _interopRequireDefault(require("./crc1"));

var _crc2 = _interopRequireDefault(require("./crc8"));

var _crc81wire = _interopRequireDefault(require("./crc81wire"));

var _crc3 = _interopRequireDefault(require("./crc16"));

var _crc16ccitt = _interopRequireDefault(require("./crc16ccitt"));

var _crc16modbus = _interopRequireDefault(require("./crc16modbus"));

var _crc16xmodem = _interopRequireDefault(require("./crc16xmodem"));

var _crc16kermit = _interopRequireDefault(require("./crc16kermit"));

var _crc4 = _interopRequireDefault(require("./crc24"));

var _crc5 = _interopRequireDefault(require("./crc32"));

var _crcjam = _interopRequireDefault(require("./crcjam"));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : {
    "default": obj
  };
}

var _default = {
  crc1: _crc["default"],
  crc8: _crc2["default"],
  crc81wire: _crc81wire["default"],
  crc16: _crc3["default"],
  crc16ccitt: _crc16ccitt["default"],
  crc16modbus: _crc16modbus["default"],
  crc16xmodem: _crc16xmodem["default"],
  crc16kermit: _crc16kermit["default"],
  crc24: _crc4["default"],
  crc32: _crc5["default"],
  crcjam: _crcjam["default"]
};
exports["default"] = _default;
