// uri-js/rollup.config.js
J$.noInstrEval = false;
jalangiLabel1:
    while (true) {
        try {
            J$.Se(3909105, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestRequest/node_modules/uri-js/rollup.config.js');
            function _interopRequireDefault(obj) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(3908729, arguments.callee, this, arguments);
                            arguments = J$.N(3908737, 'arguments', arguments, true, false, false);
                            obj = J$.N(3908745, 'obj', obj, true, false, false);
                            return J$.Rt(3908721, J$.C(175976, J$.C(175968, J$.R(3908673, 'obj', obj, false, false)) ? J$.G(3908689, J$.R(3908681, 'obj', obj, false, false), '__esModule') : J$._()) ? J$.R(3908697, 'obj', obj, false, false) : J$.T(3908713, { 'default': J$.R(3908705, 'obj', obj, false, false) }, 11, false));
                        } catch (J$e) {
                            J$.Ex(3909161, J$e);
                        } finally {
                            if (J$.Fr(3909169))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(3909113, '_rollupPluginNodeResolve', _rollupPluginNodeResolve, false, false, false);
            J$.N(3909121, '_rollupPluginBabel', _rollupPluginBabel, false, false, false);
            _interopRequireDefault = J$.N(3909137, '_interopRequireDefault', J$.T(3909129, _interopRequireDefault, 12, false), true, false, false);
            J$.N(3909145, 'packageJson', packageJson, false, false, false);
            J$.N(3909153, '_default', _default, false, false, false);
            J$.T(3908489, 'use strict', 21, false);
            J$.M(3908537, J$.I(typeof Object === 'undefined' ? Object = J$.R(3908497, 'Object', undefined, true, true) : Object = J$.R(3908497, 'Object', Object, true, true)), 'defineProperty', false)(J$.I(typeof exports === 'undefined' ? exports = J$.R(3908505, 'exports', undefined, true, true) : exports = J$.R(3908505, 'exports', exports, true, true)), J$.T(3908513, '__esModule', 21, false), J$.T(3908529, { value: J$.T(3908521, true, 23, false) }, 11, false));
            J$.P(3908569, J$.I(typeof exports === 'undefined' ? exports = J$.R(3908545, 'exports', undefined, true, true) : exports = J$.R(3908545, 'exports', exports, true, true)), J$.T(3908553, 'default', 21, false), void J$.T(3908561, 0, 22, false));
            var _rollupPluginNodeResolve = J$.W(3908617, '_rollupPluginNodeResolve', J$.F(3908609, J$.R(3908577, '_interopRequireDefault', _interopRequireDefault, false, true), false)(J$.F(3908601, J$.I(typeof require === 'undefined' ? require = J$.R(3908585, 'require', undefined, true, true) : require = J$.R(3908585, 'require', require, true, true)), false)(J$.T(3908593, 'rollup-plugin-node-resolve', 21, false))), _rollupPluginNodeResolve, false, true);
            var _rollupPluginBabel = J$.W(3908665, '_rollupPluginBabel', J$.F(3908657, J$.R(3908625, '_interopRequireDefault', _interopRequireDefault, false, true), false)(J$.F(3908649, J$.I(typeof require === 'undefined' ? require = J$.R(3908633, 'require', undefined, true, true) : require = J$.R(3908633, 'require', require, true, true)), false)(J$.T(3908641, 'rollup-plugin-babel', 21, false))), _rollupPluginBabel, false, true);
            var packageJson = J$.W(3908777, 'packageJson', J$.F(3908769, J$.I(typeof require === 'undefined' ? require = J$.R(3908753, 'require', undefined, true, true) : require = J$.R(3908753, 'require', require, true, true)), false)(J$.T(3908761, './package.json', 21, false)), packageJson, false, true);
            var _default = J$.W(3909065, '_default', J$.T(3909057, {
                entry: J$.T(3908785, 'dist/esnext/index.js', 21, false),
                format: J$.T(3908793, 'umd', 21, false),
                moduleName: J$.T(3908801, 'URI', 21, false),
                plugins: J$.T(3909001, [
                    J$.F(3908873, (J$.T(3908809, 0, 22, false), J$.G(3908833, J$.R(3908817, '_rollupPluginNodeResolve', _rollupPluginNodeResolve, false, true), J$.T(3908825, 'default', 21, false))), false)(J$.T(3908865, {
                        module: J$.T(3908841, true, 23, false),
                        jsnext: J$.T(3908849, true, 23, false),
                        preferBuiltins: J$.T(3908857, false, 23, false)
                    }, 11, false)),
                    J$.F(3908993, (J$.T(3908881, 0, 22, false), J$.G(3908905, J$.R(3908889, '_rollupPluginBabel', _rollupPluginBabel, false, true), J$.T(3908897, 'default', 21, false))), false)(J$.T(3908985, {
                        'presets': J$.T(3908953, [J$.T(3908945, [
                                J$.T(3908913, 'latest', 21, false),
                                J$.T(3908937, { 'es2015': J$.T(3908929, { 'modules': J$.T(3908921, false, 23, false) }, 11, false) }, 11, false)
                            ], 10, false)], 10, false),
                        'plugins': J$.T(3908969, [J$.T(3908961, 'external-helpers', 21, false)], 10, false),
                        'externalHelpers': J$.T(3908977, false, 23, false)
                    }, 11, false))
                ], 10, false),
                dest: J$.T(3909009, 'dist/es5/uri.all.js', 21, false),
                sourceMap: J$.T(3909017, true, 23, false),
                banner: J$.B(540946, '+', J$.B(540938, '+', J$.T(3909025, '/** @license URI.js v', 21, false), J$.G(3909041, J$.R(3909033, 'packageJson', packageJson, false, true), 'version')), J$.T(3909049, ' (c) 2011 Gary Court. License: http://github.com/garycourt/uri-js */', 21, false))
            }, 11, false), _default, false, true);
            J$.P(3909097, J$.I(typeof exports === 'undefined' ? exports = J$.R(3909073, 'exports', undefined, true, true) : exports = J$.R(3909073, 'exports', exports, true, true)), J$.T(3909081, 'default', 21, false), J$.R(3909089, '_default', _default, false, true));
        } catch (J$e) {
            J$.Ex(3909177, J$e);
        } finally {
            if (J$.Sr(3909185))
                continue jalangiLabel1;
            else
                break jalangiLabel1;
        }
    }
// JALANGI DO NOT INSTRUMENT
