// winston/dist/winston/common.js
J$.noInstrEval = false;
jalangiLabel11:
    while (true) {
        try {
            J$.Se(5790321, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/winston/dist/winston/common.js');
            J$.N(5790329, '_require', _require, false, false, false);
            J$.N(5790337, 'format', format, false, false, false);
            J$.T(5789337, 'use strict', 21, false);
            var _require = J$.W(5789385, '_require', J$.F(5789361, J$.I(typeof require === 'undefined' ? require = J$.R(5789345, 'require', undefined, true, true) : require = J$.R(5789345, 'require', require, true, true)), false)(J$.T(5789353, 'util', 21, false)), _require, false, true), format = J$.W(5789393, 'format', J$.G(5789377, J$.R(5789369, '_require', _require, false, true), 'format'), format, false, true);
            J$.P(5790313, J$.I(typeof exports === 'undefined' ? exports = J$.R(5789401, 'exports', undefined, true, true) : exports = J$.R(5789401, 'exports', exports, true, true)), 'warn', J$.T(5790305, {
                deprecated: J$.T(5789513, function deprecated(prop) {
                    jalangiLabel1:
                        while (true) {
                            try {
                                J$.Fe(5789489, arguments.callee, this, arguments);
                                arguments = J$.N(5789497, 'arguments', arguments, true, false, false);
                                prop = J$.N(5789505, 'prop', prop, true, false, false);
                                return J$.Rt(5789481, J$.T(5789473, function () {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(5789457, arguments.callee, this, arguments);
                                                arguments = J$.N(5789465, 'arguments', arguments, true, false, false);
                                                throw J$.F(5789449, J$.I(typeof Error === 'undefined' ? Error = J$.R(5789409, 'Error', undefined, true, true) : Error = J$.R(5789409, 'Error', Error, true, true)), true)(J$.F(5789441, J$.R(5789417, 'format', format, false, true), false)(J$.T(5789425, '{ %s } was removed in winston@3.0.0.', 21, false), J$.R(5789433, 'prop', prop, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(5790345, J$e);
                                            } finally {
                                                if (J$.Fr(5790353))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            } catch (J$e) {
                                J$.Ex(5790361, J$e);
                            } finally {
                                if (J$.Fr(5790369))
                                    continue jalangiLabel1;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                useFormat: J$.T(5789657, function useFormat(prop) {
                    jalangiLabel3:
                        while (true) {
                            try {
                                J$.Fe(5789633, arguments.callee, this, arguments);
                                arguments = J$.N(5789641, 'arguments', arguments, true, false, false);
                                prop = J$.N(5789649, 'prop', prop, true, false, false);
                                return J$.Rt(5789625, J$.T(5789617, function () {
                                    jalangiLabel2:
                                        while (true) {
                                            try {
                                                J$.Fe(5789601, arguments.callee, this, arguments);
                                                arguments = J$.N(5789609, 'arguments', arguments, true, false, false);
                                                throw J$.F(5789593, J$.I(typeof Error === 'undefined' ? Error = J$.R(5789521, 'Error', undefined, true, true) : Error = J$.R(5789521, 'Error', Error, true, true)), true)(J$.M(5789585, J$.T(5789569, [
                                                    J$.F(5789553, J$.R(5789529, 'format', format, false, true), false)(J$.T(5789537, '{ %s } was removed in winston@3.0.0.', 21, false), J$.R(5789545, 'prop', prop, false, false)),
                                                    J$.T(5789561, 'Use a custom winston.format = winston.format(function) instead.', 21, false)
                                                ], 10, false), 'join', false)(J$.T(5789577, '\n', 21, false)));
                                            } catch (J$e) {
                                                J$.Ex(5790377, J$e);
                                            } finally {
                                                if (J$.Fr(5790385))
                                                    continue jalangiLabel2;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            } catch (J$e) {
                                J$.Ex(5790393, J$e);
                            } finally {
                                if (J$.Fr(5790401))
                                    continue jalangiLabel3;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                forFunctions: J$.T(5789817, function forFunctions(obj, type, props) {
                    jalangiLabel5:
                        while (true) {
                            try {
                                J$.Fe(5789777, arguments.callee, this, arguments);
                                arguments = J$.N(5789785, 'arguments', arguments, true, false, false);
                                obj = J$.N(5789793, 'obj', obj, true, false, false);
                                type = J$.N(5789801, 'type', type, true, false, false);
                                props = J$.N(5789809, 'props', props, true, false, false);
                                J$.M(5789769, J$.R(5789665, 'props', props, false, false), 'forEach', false)(J$.T(5789761, function (prop) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(5789737, arguments.callee, this, arguments);
                                                arguments = J$.N(5789745, 'arguments', arguments, true, false, false);
                                                prop = J$.N(5789753, 'prop', prop, true, false, false);
                                                J$.P(5789729, J$.R(5789673, 'obj', obj, false, false), J$.R(5789681, 'prop', prop, false, false), J$.M(5789721, J$.G(5789697, J$.I(typeof exports === 'undefined' ? exports = J$.R(5789689, 'exports', undefined, true, true) : exports = J$.R(5789689, 'exports', exports, true, true)), 'warn'), J$.R(5789705, 'type', type, false, false), false)(J$.R(5789713, 'prop', prop, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(5790409, J$e);
                                            } finally {
                                                if (J$.Fr(5790417))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            } catch (J$e) {
                                J$.Ex(5790425, J$e);
                            } finally {
                                if (J$.Fr(5790433))
                                    continue jalangiLabel5;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                moved: J$.T(5790089, function moved(obj, movedTo, prop) {
                    jalangiLabel8:
                        while (true) {
                            try {
                                J$.Fe(5790033, arguments.callee, this, arguments);
                                function movedNotice() {
                                    jalangiLabel7:
                                        while (true) {
                                            try {
                                                J$.Fe(5789961, arguments.callee, this, arguments);
                                                arguments = J$.N(5789969, 'arguments', arguments, true, false, false);
                                                return J$.Rt(5789953, J$.T(5789945, function () {
                                                    jalangiLabel6:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(5789929, arguments.callee, this, arguments);
                                                                arguments = J$.N(5789937, 'arguments', arguments, true, false, false);
                                                                throw J$.F(5789921, J$.I(typeof Error === 'undefined' ? Error = J$.R(5789825, 'Error', undefined, true, true) : Error = J$.R(5789825, 'Error', Error, true, true)), true)(J$.M(5789913, J$.T(5789897, [
                                                                    J$.F(5789857, J$.R(5789833, 'format', format, false, true), false)(J$.T(5789841, 'winston.%s was moved in winston@3.0.0.', 21, false), J$.R(5789849, 'prop', prop, false, false)),
                                                                    J$.F(5789889, J$.R(5789865, 'format', format, false, true), false)(J$.T(5789873, 'Use a winston.%s instead.', 21, false), J$.R(5789881, 'movedTo', movedTo, false, false))
                                                                ], 10, false), 'join', false)(J$.T(5789905, '\n', 21, false)));
                                                            } catch (J$e) {
                                                                J$.Ex(5790441, J$e);
                                                            } finally {
                                                                if (J$.Fr(5790449))
                                                                    continue jalangiLabel6;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false));
                                            } catch (J$e) {
                                                J$.Ex(5790457, J$e);
                                            } finally {
                                                if (J$.Fr(5790465))
                                                    continue jalangiLabel7;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }
                                arguments = J$.N(5790041, 'arguments', arguments, true, false, false);
                                obj = J$.N(5790049, 'obj', obj, true, false, false);
                                movedTo = J$.N(5790057, 'movedTo', movedTo, true, false, false);
                                prop = J$.N(5790065, 'prop', prop, true, false, false);
                                movedNotice = J$.N(5790081, 'movedNotice', J$.T(5790073, movedNotice, 12, false), true, false, false);
                                J$.M(5790025, J$.I(typeof Object === 'undefined' ? Object = J$.R(5789977, 'Object', undefined, true, true) : Object = J$.R(5789977, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(5789985, 'obj', obj, false, false), J$.R(5789993, 'prop', prop, false, false), J$.T(5790017, {
                                    get: J$.R(5790001, 'movedNotice', movedNotice, false, false),
                                    set: J$.R(5790009, 'movedNotice', movedNotice, false, false)
                                }, 11, false));
                            } catch (J$e) {
                                J$.Ex(5790473, J$e);
                            } finally {
                                if (J$.Fr(5790481))
                                    continue jalangiLabel8;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                forProperties: J$.T(5790297, function forProperties(obj, type, props) {
                    jalangiLabel10:
                        while (true) {
                            try {
                                J$.Fe(5790257, arguments.callee, this, arguments);
                                arguments = J$.N(5790265, 'arguments', arguments, true, false, false);
                                obj = J$.N(5790273, 'obj', obj, true, false, false);
                                type = J$.N(5790281, 'type', type, true, false, false);
                                props = J$.N(5790289, 'props', props, true, false, false);
                                J$.M(5790249, J$.R(5790097, 'props', props, false, false), 'forEach', false)(J$.T(5790241, function (prop) {
                                    jalangiLabel9:
                                        while (true) {
                                            try {
                                                J$.Fe(5790209, arguments.callee, this, arguments);
                                                arguments = J$.N(5790217, 'arguments', arguments, true, false, false);
                                                prop = J$.N(5790225, 'prop', prop, true, false, false);
                                                J$.N(5790233, 'notice', notice, false, false, false);
                                                var notice = J$.W(5790145, 'notice', J$.M(5790137, J$.G(5790113, J$.I(typeof exports === 'undefined' ? exports = J$.R(5790105, 'exports', undefined, true, true) : exports = J$.R(5790105, 'exports', exports, true, true)), 'warn'), J$.R(5790121, 'type', type, false, false), false)(J$.R(5790129, 'prop', prop, false, false)), notice, false, false);
                                                J$.M(5790201, J$.I(typeof Object === 'undefined' ? Object = J$.R(5790153, 'Object', undefined, true, true) : Object = J$.R(5790153, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(5790161, 'obj', obj, false, false), J$.R(5790169, 'prop', prop, false, false), J$.T(5790193, {
                                                    get: J$.R(5790177, 'notice', notice, false, false),
                                                    set: J$.R(5790185, 'notice', notice, false, false)
                                                }, 11, false));
                                            } catch (J$e) {
                                                J$.Ex(5790489, J$e);
                                            } finally {
                                                if (J$.Fr(5790497))
                                                    continue jalangiLabel9;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            } catch (J$e) {
                                J$.Ex(5790505, J$e);
                            } finally {
                                if (J$.Fr(5790513))
                                    continue jalangiLabel10;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(5790521, J$e);
        } finally {
            if (J$.Sr(5790529))
                continue jalangiLabel11;
            else
                break jalangiLabel11;
        }
    }
// JALANGI DO NOT INSTRUMENT
