// aws-sign2/index.js
J$.noInstrEval = false;
jalangiLabel9:
    while (true) {
        try {
            J$.Se(6842633, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/aws-sign2/index.js');
            function authorization(options) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(6840817, arguments.callee, this, arguments);
                            arguments = J$.N(6840825, 'arguments', arguments, true, false, false);
                            options = J$.N(6840833, 'options', options, true, false, false);
                            return J$.Rt(6840809, J$.B(700114, '+', J$.B(700106, '+', J$.B(700098, '+', J$.T(6840753, 'AWS ', 21, false), J$.G(6840769, J$.R(6840761, 'options', options, false, false), 'key')), J$.T(6840777, ':', 21, false)), J$.F(6840801, J$.R(6840785, 'sign', sign, false, true), false)(J$.R(6840793, 'options', options, false, false))));
                        } catch (J$e) {
                            J$.Ex(6842793, J$e);
                        } finally {
                            if (J$.Fr(6842801))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function hmacSha1(options) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(6840985, arguments.callee, this, arguments);
                            arguments = J$.N(6840993, 'arguments', arguments, true, false, false);
                            options = J$.N(6841001, 'options', options, true, false, false);
                            return J$.Rt(6840977, J$.M(6840969, J$.M(6840953, J$.M(6840929, J$.R(6840897, 'crypto', crypto, false, true), 'createHmac', false)(J$.T(6840905, 'sha1', 21, false), J$.G(6840921, J$.R(6840913, 'options', options, false, false), 'secret')), 'update', false)(J$.G(6840945, J$.R(6840937, 'options', options, false, false), 'message')), 'digest', false)(J$.T(6840961, 'base64', 21, false)));
                        } catch (J$e) {
                            J$.Ex(6842809, J$e);
                        } finally {
                            if (J$.Fr(6842817))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function sign(options) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(6841113, arguments.callee, this, arguments);
                            arguments = J$.N(6841121, 'arguments', arguments, true, false, false);
                            options = J$.N(6841129, 'options', options, true, false, false);
                            J$.P(6841073, J$.R(6841041, 'options', options, false, false), 'message', J$.F(6841065, J$.R(6841049, 'stringToSign', stringToSign, false, true), false)(J$.R(6841057, 'options', options, false, false)));
                            return J$.Rt(6841105, J$.F(6841097, J$.R(6841081, 'hmacSha1', hmacSha1, false, true), false)(J$.R(6841089, 'options', options, false, false)));
                        } catch (J$e) {
                            J$.Ex(6842825, J$e);
                        } finally {
                            if (J$.Fr(6842833))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function signQuery(options) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(6841241, arguments.callee, this, arguments);
                            arguments = J$.N(6841249, 'arguments', arguments, true, false, false);
                            options = J$.N(6841257, 'options', options, true, false, false);
                            J$.P(6841201, J$.R(6841169, 'options', options, false, false), 'message', J$.F(6841193, J$.R(6841177, 'queryStringToSign', queryStringToSign, false, true), false)(J$.R(6841185, 'options', options, false, false)));
                            return J$.Rt(6841233, J$.F(6841225, J$.R(6841209, 'hmacSha1', hmacSha1, false, true), false)(J$.R(6841217, 'options', options, false, false)));
                        } catch (J$e) {
                            J$.Ex(6842841, J$e);
                        } finally {
                            if (J$.Fr(6842849))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function stringToSign(options) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(6841529, arguments.callee, this, arguments);
                            arguments = J$.N(6841537, 'arguments', arguments, true, false, false);
                            options = J$.N(6841545, 'options', options, true, false, false);
                            J$.N(6841553, 'headers', headers, false, false, false);
                            J$.N(6841561, 'r', r, false, false, false);
                            var headers = J$.W(6841321, 'headers', J$.C(370616, J$.G(6841305, J$.R(6841297, 'options', options, false, false), 'amazonHeaders')) ? J$._() : J$.T(6841313, '', 21, false), headers, false, false);
                            if (J$.C(370624, J$.R(6841329, 'headers', headers, false, false)))
                                headers = J$.W(6841353, 'headers', J$.B(700122, '+', J$.R(6841345, 'headers', headers, false, false), J$.T(6841337, '\n', 21, false)), headers, false, false);
                            var r = J$.W(6841489, 'r', J$.T(6841481, [
                                J$.G(6841369, J$.R(6841361, 'options', options, false, false), 'verb'),
                                J$.G(6841385, J$.R(6841377, 'options', options, false, false), 'md5'),
                                J$.G(6841401, J$.R(6841393, 'options', options, false, false), 'contentType'),
                                J$.C(370632, J$.G(6841417, J$.R(6841409, 'options', options, false, false), 'date')) ? J$.M(6841441, J$.G(6841433, J$.R(6841425, 'options', options, false, false), 'date'), 'toUTCString', false)() : J$.T(6841449, '', 21, false),
                                J$.B(700130, '+', J$.R(6841457, 'headers', headers, false, false), J$.G(6841473, J$.R(6841465, 'options', options, false, false), 'resource'))
                            ], 10, false), r, false, false);
                            return J$.Rt(6841521, J$.M(6841513, J$.R(6841497, 'r', r, false, false), 'join', false)(J$.T(6841505, '\n', 21, false)));
                        } catch (J$e) {
                            J$.Ex(6842857, J$e);
                        } finally {
                            if (J$.Fr(6842865))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function queryStringToSign(options) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(6841657, arguments.callee, this, arguments);
                            arguments = J$.N(6841665, 'arguments', arguments, true, false, false);
                            options = J$.N(6841673, 'options', options, true, false, false);
                            return J$.Rt(6841649, J$.B(700154, '+', J$.B(700146, '+', J$.B(700138, '+', J$.T(6841601, 'GET\n\n\n', 21, false), J$.G(6841617, J$.R(6841609, 'options', options, false, false), 'date')), J$.T(6841625, '\n', 21, false)), J$.G(6841641, J$.R(6841633, 'options', options, false, false), 'resource')));
                        } catch (J$e) {
                            J$.Ex(6842873, J$e);
                        } finally {
                            if (J$.Fr(6842881))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function canonicalizeHeaders(headers) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(6842033, arguments.callee, this, arguments);
                            arguments = J$.N(6842041, 'arguments', arguments, true, false, false);
                            headers = J$.N(6842049, 'headers', headers, true, false, false);
                            J$.N(6842057, 'buf', buf, false, false, false);
                            J$.N(6842065, 'fields', fields, false, false, false);
                            J$.N(6842073, 'i', i, false, false, false);
                            J$.N(6842081, 'len', len, false, false, false);
                            J$.N(6842089, 'field', field, false, false, false);
                            J$.N(6842097, 'val', val, false, false, false);
                            var buf = J$.W(6841745, 'buf', J$.T(6841713, [], 10, false), buf, false, false), fields = J$.W(6841753, 'fields', J$.M(6841737, J$.I(typeof Object === 'undefined' ? Object = J$.R(6841721, 'Object', undefined, true, true) : Object = J$.R(6841721, 'Object', Object, true, true)), 'keys', false)(J$.R(6841729, 'headers', headers, false, false)), fields, false, false);
                            for (var i = J$.W(6841785, 'i', J$.T(6841761, 0, 22, false), i, false, false), len = J$.W(6841793, 'len', J$.G(6841777, J$.R(6841769, 'fields', fields, false, false), 'length'), len, false, false); J$.C(370648, J$.B(700162, '<', J$.R(6841801, 'i', i, false, false), J$.R(6841809, 'len', len, false, false))); i = J$.W(6841825, 'i', J$.B(700178, '+', J$.U(700170, '+', J$.R(6841817, 'i', i, false, false)), 1), i, false, false)) {
                                var field = J$.W(6841897, 'field', J$.G(6841849, J$.R(6841833, 'fields', fields, false, false), J$.R(6841841, 'i', i, false, false)), field, false, false), val = J$.W(6841905, 'val', J$.G(6841873, J$.R(6841857, 'headers', headers, false, false), J$.R(6841865, 'field', field, false, false)), val, false, false), field = J$.W(6841913, 'field', J$.M(6841889, J$.R(6841881, 'field', field, false, false), 'toLowerCase', false)(), field, false, false);
                                if (J$.C(370640, J$.B(700186, '!==', J$.T(6841921, 0, 22, false), J$.M(6841945, J$.R(6841929, 'field', field, false, false), 'indexOf', false)(J$.T(6841937, 'x-amz', 21, false)))))
                                    continue;
                                J$.M(6841985, J$.R(6841953, 'buf', buf, false, false), 'push', false)(J$.B(700202, '+', J$.B(700194, '+', J$.R(6841961, 'field', field, false, false), J$.T(6841969, ':', 21, false)), J$.R(6841977, 'val', val, false, false)));
                            }
                            return J$.Rt(6842025, J$.M(6842017, J$.M(6842001, J$.R(6841993, 'buf', buf, false, false), 'sort', false)(), 'join', false)(J$.T(6842009, '\n', 21, false)));
                        } catch (J$e) {
                            J$.Ex(6842889, J$e);
                        } finally {
                            if (J$.Fr(6842897))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function canonicalizeResource(resource) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(6842553, arguments.callee, this, arguments);
                            arguments = J$.N(6842561, 'arguments', arguments, true, false, false);
                            resource = J$.N(6842569, 'resource', resource, true, false, false);
                            J$.N(6842577, 'url', url, false, false, false);
                            J$.N(6842585, 'path', path, false, false, false);
                            J$.N(6842593, 'buf', buf, false, false, false);
                            var url = J$.W(6842193, 'url', J$.F(6842161, J$.R(6842137, 'parse', parse, false, true), false)(J$.R(6842145, 'resource', resource, false, false), J$.T(6842153, true, 23, false)), url, false, false), path = J$.W(6842201, 'path', J$.G(6842177, J$.R(6842169, 'url', url, false, false), 'pathname'), path, false, false), buf = J$.W(6842209, 'buf', J$.T(6842185, [], 10, false), buf, false, false);
                            J$.M(6842465, J$.M(6842241, J$.I(typeof Object === 'undefined' ? Object = J$.R(6842217, 'Object', undefined, true, true) : Object = J$.R(6842217, 'Object', Object, true, true)), 'keys', false)(J$.G(6842233, J$.R(6842225, 'url', url, false, false), 'query')), 'forEach', false)(J$.T(6842457, function (key) {
                                jalangiLabel7:
                                    while (true) {
                                        try {
                                            J$.Fe(6842425, arguments.callee, this, arguments);
                                            arguments = J$.N(6842433, 'arguments', arguments, true, false, false);
                                            key = J$.N(6842441, 'key', key, true, false, false);
                                            J$.N(6842449, 'val', val, false, false, false);
                                            if (J$.C(370656, J$.U(700218, '!', J$.U(700210, '~', J$.M(6842265, J$.R(6842249, 'keys', keys, false, true), 'indexOf', false)(J$.R(6842257, 'key', key, false, false))))))
                                                return J$.Rt(6842273, undefined);
                                            var val = J$.W(6842385, 'val', J$.C(370664, J$.B(700226, '==', J$.T(6842281, '', 21, false), J$.G(6842313, J$.G(6842297, J$.R(6842289, 'url', url, false, false), 'query'), J$.R(6842305, 'key', key, false, false)))) ? J$.T(6842321, '', 21, false) : J$.B(700234, '+', J$.T(6842329, '=', 21, false), J$.F(6842377, J$.I(typeof encodeURIComponent === 'undefined' ? encodeURIComponent = J$.R(6842337, 'encodeURIComponent', undefined, true, true) : encodeURIComponent = J$.R(6842337, 'encodeURIComponent', encodeURIComponent, true, true)), false)(J$.G(6842369, J$.G(6842353, J$.R(6842345, 'url', url, false, false), 'query'), J$.R(6842361, 'key', key, false, false)))), val, false, false);
                                            J$.M(6842417, J$.R(6842393, 'buf', buf, false, false), 'push', false)(J$.B(700242, '+', J$.R(6842401, 'key', key, false, false), J$.R(6842409, 'val', val, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(6842905, J$e);
                                        } finally {
                                            if (J$.Fr(6842913))
                                                continue jalangiLabel7;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            return J$.Rt(6842545, J$.B(700258, '+', J$.R(6842473, 'path', path, false, false), J$.C(370672, J$.G(6842489, J$.R(6842481, 'buf', buf, false, false), 'length')) ? J$.B(700250, '+', J$.T(6842497, '?', 21, false), J$.M(6842529, J$.M(6842513, J$.R(6842505, 'buf', buf, false, false), 'sort', false)(), 'join', false)(J$.T(6842521, '&', 21, false))) : J$.T(6842537, '', 21, false)));
                        } catch (J$e) {
                            J$.Ex(6842921, J$e);
                        } finally {
                            if (J$.Fr(6842929))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(6842641, 'crypto', crypto, false, false, false);
            J$.N(6842649, 'parse', parse, false, false, false);
            J$.N(6842657, 'keys', keys, false, false, false);
            authorization = J$.N(6842673, 'authorization', J$.T(6842665, authorization, 12, false), true, false, false);
            hmacSha1 = J$.N(6842689, 'hmacSha1', J$.T(6842681, hmacSha1, 12, false), true, false, false);
            sign = J$.N(6842705, 'sign', J$.T(6842697, sign, 12, false), true, false, false);
            signQuery = J$.N(6842721, 'signQuery', J$.T(6842713, signQuery, 12, false), true, false, false);
            stringToSign = J$.N(6842737, 'stringToSign', J$.T(6842729, stringToSign, 12, false), true, false, false);
            queryStringToSign = J$.N(6842753, 'queryStringToSign', J$.T(6842745, queryStringToSign, 12, false), true, false, false);
            canonicalizeHeaders = J$.N(6842769, 'canonicalizeHeaders', J$.T(6842761, canonicalizeHeaders, 12, false), true, false, false);
            canonicalizeResource = J$.N(6842785, 'canonicalizeResource', J$.T(6842777, canonicalizeResource, 12, false), true, false, false);
            var crypto = J$.W(6840609, 'crypto', J$.F(6840569, J$.I(typeof require === 'undefined' ? require = J$.R(6840553, 'require', undefined, true, true) : require = J$.R(6840553, 'require', require, true, true)), false)(J$.T(6840561, 'crypto', 21, false)), crypto, false, true), parse = J$.W(6840617, 'parse', J$.G(6840601, J$.F(6840593, J$.I(typeof require === 'undefined' ? require = J$.R(6840577, 'require', undefined, true, true) : require = J$.R(6840577, 'require', require, true, true)), false)(J$.T(6840585, 'url', 21, false)), 'parse'), parse, false, true);
            var keys = J$.W(6840745, 'keys', J$.T(6840737, [
                J$.T(6840625, 'acl', 21, false),
                J$.T(6840633, 'location', 21, false),
                J$.T(6840641, 'logging', 21, false),
                J$.T(6840649, 'notification', 21, false),
                J$.T(6840657, 'partNumber', 21, false),
                J$.T(6840665, 'policy', 21, false),
                J$.T(6840673, 'requestPayment', 21, false),
                J$.T(6840681, 'torrent', 21, false),
                J$.T(6840689, 'uploadId', 21, false),
                J$.T(6840697, 'uploads', 21, false),
                J$.T(6840705, 'versionId', 21, false),
                J$.T(6840713, 'versioning', 21, false),
                J$.T(6840721, 'versions', 21, false),
                J$.T(6840729, 'website', 21, false)
            ], 10, false), keys, false, true);
            J$.P(6840857, J$.I(typeof module === 'undefined' ? module = J$.R(6840841, 'module', undefined, true, true) : module = J$.R(6840841, 'module', module, true, true)), 'exports', J$.R(6840849, 'authorization', authorization, false, true));
            J$.P(6840889, J$.G(6840873, J$.I(typeof module === 'undefined' ? module = J$.R(6840865, 'module', undefined, true, true) : module = J$.R(6840865, 'module', module, true, true)), 'exports'), 'authorization', J$.R(6840881, 'authorization', authorization, false, true));
            J$.P(6841033, J$.G(6841017, J$.I(typeof module === 'undefined' ? module = J$.R(6841009, 'module', undefined, true, true) : module = J$.R(6841009, 'module', module, true, true)), 'exports'), 'hmacSha1', J$.R(6841025, 'hmacSha1', hmacSha1, false, true));
            J$.P(6841161, J$.G(6841145, J$.I(typeof module === 'undefined' ? module = J$.R(6841137, 'module', undefined, true, true) : module = J$.R(6841137, 'module', module, true, true)), 'exports'), 'sign', J$.R(6841153, 'sign', sign, false, true));
            J$.P(6841289, J$.G(6841273, J$.I(typeof module === 'undefined' ? module = J$.R(6841265, 'module', undefined, true, true) : module = J$.R(6841265, 'module', module, true, true)), 'exports'), 'signQuery', J$.R(6841281, 'signQuery', signQuery, false, true));
            J$.P(6841593, J$.G(6841577, J$.I(typeof module === 'undefined' ? module = J$.R(6841569, 'module', undefined, true, true) : module = J$.R(6841569, 'module', module, true, true)), 'exports'), 'stringToSign', J$.R(6841585, 'stringToSign', stringToSign, false, true));
            J$.P(6841705, J$.G(6841689, J$.I(typeof module === 'undefined' ? module = J$.R(6841681, 'module', undefined, true, true) : module = J$.R(6841681, 'module', module, true, true)), 'exports'), 'queryStringToSign', J$.R(6841697, 'queryStringToSign', queryStringToSign, false, true));
            J$.P(6842129, J$.G(6842113, J$.I(typeof module === 'undefined' ? module = J$.R(6842105, 'module', undefined, true, true) : module = J$.R(6842105, 'module', module, true, true)), 'exports'), 'canonicalizeHeaders', J$.R(6842121, 'canonicalizeHeaders', canonicalizeHeaders, false, true));
            J$.P(6842625, J$.G(6842609, J$.I(typeof module === 'undefined' ? module = J$.R(6842601, 'module', undefined, true, true) : module = J$.R(6842601, 'module', module, true, true)), 'exports'), 'canonicalizeResource', J$.R(6842617, 'canonicalizeResource', canonicalizeResource, false, true));
        } catch (J$e) {
            J$.Ex(6842937, J$e);
        } finally {
            if (J$.Sr(6842945))
                continue jalangiLabel9;
            else
                break jalangiLabel9;
        }
    }
// JALANGI DO NOT INSTRUMENT
