// js-yaml/lib/js-yaml/type/js/undefined_jalangi_.js
J$.iids = {"9":[1,1,1,13],"10":[19,10,19,23],"17":[1,1,1,14],"18":[19,10,19,39],"25":[3,12,3,19],"33":[3,20,3,32],"41":[3,12,3,33],"49":[3,12,3,33],"57":[3,12,3,33],"65":[6,10,6,14],"73":[6,10,6,14],"81":[6,3,6,15],"89":[5,1,7,2],"97":[5,1,7,2],"105":[11,10,11,19],"113":[11,10,11,19],"121":[11,3,11,20],"129":[9,1,12,2],"137":[9,1,12,2],"145":[15,10,15,12],"153":[15,10,15,12],"161":[15,3,15,13],"169":[14,1,16,2],"177":[14,1,16,2],"185":[19,17,19,23],"193":[19,28,19,39],"201":[19,10,19,39],"209":[19,3,19,40],"217":[18,1,20,2],"225":[18,1,20,2],"233":[18,1,20,2],"241":[22,1,22,7],"249":[22,22,22,26],"257":[22,27,22,59],"265":[23,9,23,17],"273":[24,12,24,38],"281":[25,14,25,42],"289":[26,14,26,25],"297":[27,14,27,42],"305":[22,61,28,2],"313":[22,18,28,3],"321":[22,1,28,3],"329":[22,1,28,4],"337":[1,1,29,1],"345":[1,1,29,1],"353":[5,1,7,2],"361":[1,1,29,1],"369":[9,1,12,2],"377":[1,1,29,1],"385":[14,1,16,2],"393":[1,1,29,1],"401":[18,1,20,2],"409":[1,1,29,1],"417":[5,1,7,2],"425":[5,1,7,2],"433":[9,1,12,2],"441":[9,1,12,2],"449":[14,1,16,2],"457":[14,1,16,2],"465":[18,1,20,2],"473":[18,1,20,2],"481":[1,1,29,1],"489":[1,1,29,1],"nBranches":0,"originalCodeFileName":"/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/js/undefined.js","instrumentedCodeFileName":"/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/js/undefined_jalangi_.js","code":"'use strict';\n\nvar Type = require('../../type');\n\nfunction resolveJavascriptUndefined() {\n  return true;\n}\n\nfunction constructJavascriptUndefined() {\n  /*eslint-disable no-undefined*/\n  return undefined;\n}\n\nfunction representJavascriptUndefined() {\n  return '';\n}\n\nfunction isUndefined(object) {\n  return typeof object === 'undefined';\n}\n\nmodule.exports = new Type('tag:yaml.org,2002:js/undefined', {\n  kind: 'scalar',\n  resolve: resolveJavascriptUndefined,\n  construct: constructJavascriptUndefined,\n  predicate: isUndefined,\n  represent: representJavascriptUndefined\n});\n"};
jalangiLabel152:
    while (true) {
        try {
            J$.Se(337, '/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/js/undefined_jalangi_.js', '/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/js/undefined.js');
            function resolveJavascriptUndefined() {
                jalangiLabel148:
                    while (true) {
                        try {
                            J$.Fe(89, arguments.callee, this, arguments);
                            arguments = J$.N(97, 'arguments', arguments, 4);
                            return J$.X1(81, J$.Rt(73, J$.T(65, true, 23, false)));
                        } catch (J$e) {
                            J$.Ex(417, J$e);
                        } finally {
                            if (J$.Fr(425))
                                continue jalangiLabel148;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function constructJavascriptUndefined() {
                jalangiLabel149:
                    while (true) {
                        try {
                            J$.Fe(129, arguments.callee, this, arguments);
                            arguments = J$.N(137, 'arguments', arguments, 4);
                            return J$.X1(121, J$.Rt(113, J$.T(105, undefined, 24, false)));
                        } catch (J$e) {
                            J$.Ex(433, J$e);
                        } finally {
                            if (J$.Fr(441))
                                continue jalangiLabel149;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function representJavascriptUndefined() {
                jalangiLabel150:
                    while (true) {
                        try {
                            J$.Fe(169, arguments.callee, this, arguments);
                            arguments = J$.N(177, 'arguments', arguments, 4);
                            return J$.X1(161, J$.Rt(153, J$.T(145, '', 21, false)));
                        } catch (J$e) {
                            J$.Ex(449, J$e);
                        } finally {
                            if (J$.Fr(457))
                                continue jalangiLabel150;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isUndefined(object) {
                jalangiLabel151:
                    while (true) {
                        try {
                            J$.Fe(217, arguments.callee, this, arguments);
                            arguments = J$.N(225, 'arguments', arguments, 4);
                            object = J$.N(233, 'object', object, 4);
                            return J$.X1(209, J$.Rt(201, J$.B(18, '===', J$.U(10, 'typeof', J$.R(185, 'object', object, 0)), J$.T(193, 'undefined', 21, false), 0)));
                        } catch (J$e) {
                            J$.Ex(465, J$e);
                        } finally {
                            if (J$.Fr(473))
                                continue jalangiLabel151;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(345, 'Type', Type, 0);
            resolveJavascriptUndefined = J$.N(361, 'resolveJavascriptUndefined', J$.T(353, resolveJavascriptUndefined, 12, false, 89), 0);
            constructJavascriptUndefined = J$.N(377, 'constructJavascriptUndefined', J$.T(369, constructJavascriptUndefined, 12, false, 129), 0);
            representJavascriptUndefined = J$.N(393, 'representJavascriptUndefined', J$.T(385, representJavascriptUndefined, 12, false, 169), 0);
            isUndefined = J$.N(409, 'isUndefined', J$.T(401, isUndefined, 12, false, 217), 0);
            J$.X1(17, J$.T(9, 'use strict', 21, false));
            var Type = J$.X1(57, J$.W(49, 'Type', J$.F(41, J$.R(25, 'require', require, 2), 0)(J$.T(33, '../../type', 21, false)), Type, 3));
            J$.X1(329, J$.P(321, J$.R(241, 'module', module, 2), 'exports', J$.F(313, J$.R(249, 'Type', Type, 1), 1)(J$.T(257, 'tag:yaml.org,2002:js/undefined', 21, false), J$.T(305, {
                kind: J$.T(265, 'scalar', 21, false),
                resolve: J$.R(273, 'resolveJavascriptUndefined', resolveJavascriptUndefined, 1),
                construct: J$.R(281, 'constructJavascriptUndefined', constructJavascriptUndefined, 1),
                predicate: J$.R(289, 'isUndefined', isUndefined, 1),
                represent: J$.R(297, 'representJavascriptUndefined', representJavascriptUndefined, 1)
            }, 11, false)), 0));
        } catch (J$e) {
            J$.Ex(481, J$e);
        } finally {
            if (J$.Sr(489)) {
                J$.L();
                continue jalangiLabel152;
            } else {
                J$.L();
                break jalangiLabel152;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
