// js-yaml/lib/js-yaml/type/js/undefined_jalangi_.js
J$.iids = {"9":[2,12,2,19],"10":[18,10,18,23],"17":[2,20,2,32],"18":[18,10,18,39],"25":[2,12,2,33],"33":[2,12,2,33],"41":[2,12,2,33],"49":[5,10,5,14],"57":[5,10,5,14],"65":[5,3,5,15],"73":[4,1,6,2],"81":[4,1,6,2],"89":[10,10,10,19],"97":[10,10,10,19],"105":[10,3,10,20],"113":[8,1,11,2],"121":[8,1,11,2],"129":[14,10,14,12],"137":[14,10,14,12],"145":[14,3,14,13],"153":[13,1,15,2],"161":[13,1,15,2],"169":[18,17,18,23],"177":[18,28,18,39],"185":[18,10,18,39],"193":[18,3,18,40],"201":[17,1,19,2],"209":[17,1,19,2],"217":[17,1,19,2],"225":[21,1,21,7],"233":[21,22,21,26],"241":[21,27,21,59],"249":[22,9,22,17],"257":[23,12,23,38],"265":[24,14,24,42],"273":[25,14,25,25],"281":[26,14,26,42],"289":[21,61,26,44],"297":[21,18,26,45],"305":[21,1,26,45],"313":[21,1,26,46],"321":[1,1,26,46],"329":[1,1,26,46],"337":[4,1,6,2],"345":[1,1,26,46],"353":[8,1,11,2],"361":[1,1,26,46],"369":[13,1,15,2],"377":[1,1,26,46],"385":[17,1,19,2],"393":[1,1,26,46],"401":[4,1,6,2],"409":[4,1,6,2],"417":[8,1,11,2],"425":[8,1,11,2],"433":[13,1,15,2],"441":[13,1,15,2],"449":[17,1,19,2],"457":[17,1,19,2],"465":[1,1,26,46],"473":[1,1,26,46],"nBranches":0,"originalCodeFileName":"/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/js/undefined.js","instrumentedCodeFileName":"/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/js/undefined_jalangi_.js","code":"'use strict';\n\nvar Type = require('../../type');\n\nfunction resolveJavascriptUndefined() {\n  return true;\n}\n\nfunction constructJavascriptUndefined() {\n  /*eslint-disable no-undefined*/\n  return undefined;\n}\n\nfunction representJavascriptUndefined() {\n  return '';\n}\n\nfunction isUndefined(object) {\n  return typeof object === 'undefined';\n}\n\nmodule.exports = new Type('tag:yaml.org,2002:js/undefined', {\n  kind: 'scalar',\n  resolve: resolveJavascriptUndefined,\n  construct: constructJavascriptUndefined,\n  predicate: isUndefined,\n  represent: representJavascriptUndefined\n});\n"};
jalangiLabel156:
    while (true) {
        try {
            J$.Se(321, '/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/js/undefined_jalangi_.js', '/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/js/undefined.js');
            function resolveJavascriptUndefined() {
                jalangiLabel152:
                    while (true) {
                        try {
                            J$.Fe(73, arguments.callee, this, arguments);
                            arguments = J$.N(81, 'arguments', arguments, 4);
                            return J$.X1(65, J$.Rt(57, J$.T(49, true, 23, false)));
                        } catch (J$e) {
                            J$.Ex(401, J$e);
                        } finally {
                            if (J$.Fr(409))
                                continue jalangiLabel152;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function constructJavascriptUndefined() {
                jalangiLabel153:
                    while (true) {
                        try {
                            J$.Fe(113, arguments.callee, this, arguments);
                            arguments = J$.N(121, 'arguments', arguments, 4);
                            return J$.X1(105, J$.Rt(97, J$.T(89, undefined, 24, false)));
                        } catch (J$e) {
                            J$.Ex(417, J$e);
                        } finally {
                            if (J$.Fr(425))
                                continue jalangiLabel153;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function representJavascriptUndefined() {
                jalangiLabel154:
                    while (true) {
                        try {
                            J$.Fe(153, arguments.callee, this, arguments);
                            arguments = J$.N(161, 'arguments', arguments, 4);
                            return J$.X1(145, J$.Rt(137, J$.T(129, '', 21, false)));
                        } catch (J$e) {
                            J$.Ex(433, J$e);
                        } finally {
                            if (J$.Fr(441))
                                continue jalangiLabel154;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isUndefined(object) {
                jalangiLabel155:
                    while (true) {
                        try {
                            J$.Fe(201, arguments.callee, this, arguments);
                            arguments = J$.N(209, 'arguments', arguments, 4);
                            object = J$.N(217, 'object', object, 4);
                            return J$.X1(193, J$.Rt(185, J$.B(18, '===', J$.U(10, 'typeof', J$.R(169, 'object', object, 0)), J$.T(177, 'undefined', 21, false), 0)));
                        } catch (J$e) {
                            J$.Ex(449, J$e);
                        } finally {
                            if (J$.Fr(457))
                                continue jalangiLabel155;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(329, 'Type', Type, 0);
            resolveJavascriptUndefined = J$.N(345, 'resolveJavascriptUndefined', J$.T(337, resolveJavascriptUndefined, 12, false, 73), 0);
            constructJavascriptUndefined = J$.N(361, 'constructJavascriptUndefined', J$.T(353, constructJavascriptUndefined, 12, false, 113), 0);
            representJavascriptUndefined = J$.N(377, 'representJavascriptUndefined', J$.T(369, representJavascriptUndefined, 12, false, 153), 0);
            isUndefined = J$.N(393, 'isUndefined', J$.T(385, isUndefined, 12, false, 201), 0);
            var Type = J$.X1(41, J$.W(33, 'Type', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, '../../type', 21, false)), Type, 3));
            J$.X1(313, J$.P(305, J$.R(225, 'module', module, 2), 'exports', J$.F(297, J$.R(233, 'Type', Type, 1), 1)(J$.T(241, 'tag:yaml.org,2002:js/undefined', 21, false), J$.T(289, {
                kind: J$.T(249, 'scalar', 21, false),
                resolve: J$.R(257, 'resolveJavascriptUndefined', resolveJavascriptUndefined, 1),
                construct: J$.R(265, 'constructJavascriptUndefined', constructJavascriptUndefined, 1),
                predicate: J$.R(273, 'isUndefined', isUndefined, 1),
                represent: J$.R(281, 'representJavascriptUndefined', representJavascriptUndefined, 1)
            }, 11, false)), 0));
        } catch (J$e) {
            J$.Ex(465, J$e);
        } finally {
            if (J$.Sr(473)) {
                J$.L();
                continue jalangiLabel156;
            } else {
                J$.L();
                break jalangiLabel156;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
