// mongodb/lib/core/sdam/server.js
J$.noInstrEval = false;
jalangiLabel46:
    while (true) {
        try {
            J$.Se(239209, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/core/sdam/server.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(229361, arguments.callee, this, arguments);
                            arguments = J$.N(229369, 'arguments', arguments, true, false, false);
                            obj = J$.N(229377, 'obj', obj, true, false, false);
                            if (J$.C(13232, J$.C(13192, J$.B(14930, '===', J$.U(14922, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(229097, 'Symbol', undefined, true, true) : Symbol = J$.R(229097, 'Symbol', Symbol, true, true))), J$.T(229105, 'function', 21, false))) ? J$.B(14946, '===', J$.U(14938, 'typeof', J$.G(229121, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(229113, 'Symbol', undefined, true, true) : Symbol = J$.R(229113, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(229129, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(229185, '_typeof', J$.T(229177, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(229153, arguments.callee, this, arguments);
                                                arguments = J$.N(229161, 'arguments', arguments, true, false, false);
                                                obj = J$.N(229169, 'obj', obj, true, false, false);
                                                return J$.Rt(229145, J$.U(14954, 'typeof', J$.R(229137, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(239633, J$e);
                                            } finally {
                                                if (J$.Fr(239641))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(229321, '_typeof', J$.T(229313, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(229289, arguments.callee, this, arguments);
                                                arguments = J$.N(229297, 'arguments', arguments, true, false, false);
                                                obj = J$.N(229305, 'obj', obj, true, false, false);
                                                return J$.Rt(229281, J$.C(13224, J$.C(13216, J$.C(13208, J$.C(13200, J$.R(229193, 'obj', obj, false, false)) ? J$.B(14970, '===', J$.U(14962, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(229201, 'Symbol', undefined, true, true) : Symbol = J$.R(229201, 'Symbol', Symbol, true, true))), J$.T(229209, 'function', 21, false)) : J$._()) ? J$.B(14978, '===', J$.G(229225, J$.R(229217, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(229233, 'Symbol', undefined, true, true) : Symbol = J$.R(229233, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(14986, '!==', J$.R(229241, 'obj', obj, false, false), J$.G(229257, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(229249, 'Symbol', undefined, true, true) : Symbol = J$.R(229249, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(229265, 'symbol', 21, false) : J$.U(14994, 'typeof', J$.R(229273, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(239649, J$e);
                                            } finally {
                                                if (J$.Fr(239657))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(229353, J$.F(229345, J$.R(229329, '_typeof', _typeof, false, true), false)(J$.R(229337, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(239665, J$e);
                        } finally {
                            if (J$.Fr(239673))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(229425, arguments.callee, this, arguments);
                            arguments = J$.N(229433, 'arguments', arguments, true, false, false);
                            instance = J$.N(229441, 'instance', instance, true, false, false);
                            Constructor = J$.N(229449, 'Constructor', Constructor, true, false, false);
                            if (J$.C(13240, J$.U(15010, '!', J$.B(15002, 'instanceof', J$.R(229385, 'instance', instance, false, false), J$.R(229393, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(229417, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(229401, 'TypeError', undefined, true, true) : TypeError = J$.R(229401, 'TypeError', TypeError, true, true)), true)(J$.T(229409, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(239681, J$e);
                        } finally {
                            if (J$.Fr(239689))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(229697, arguments.callee, this, arguments);
                            arguments = J$.N(229705, 'arguments', arguments, true, false, false);
                            target = J$.N(229713, 'target', target, true, false, false);
                            props = J$.N(229721, 'props', props, true, false, false);
                            J$.N(229729, 'i', i, false, false, false);
                            J$.N(229737, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(229465, 'i', J$.T(229457, 0, 22, false), i, false, false); J$.C(13264, J$.B(15018, '<', J$.R(229473, 'i', i, false, false), J$.G(229489, J$.R(229481, 'props', props, false, false), 'length'))); J$.B(15042, '-', i = J$.W(229505, 'i', J$.B(15034, '+', J$.U(15026, '+', J$.R(229497, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(229537, 'descriptor', J$.G(229529, J$.R(229513, 'props', props, false, false), J$.R(229521, 'i', i, false, false)), descriptor, false, false);
                                J$.P(229577, J$.R(229545, 'descriptor', descriptor, false, false), 'enumerable', J$.C(13248, J$.G(229561, J$.R(229553, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(229569, false, 23, false));
                                J$.P(229601, J$.R(229585, 'descriptor', descriptor, false, false), 'configurable', J$.T(229593, true, 23, false));
                                if (J$.C(13256, J$.B(15050, 'in', J$.T(229609, 'value', 21, false), J$.R(229617, 'descriptor', descriptor, false, false))))
                                    J$.P(229641, J$.R(229625, 'descriptor', descriptor, false, false), 'writable', J$.T(229633, true, 23, false));
                                J$.M(229689, J$.I(typeof Object === 'undefined' ? Object = J$.R(229649, 'Object', undefined, true, true) : Object = J$.R(229649, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(229657, 'target', target, false, false), J$.G(229673, J$.R(229665, 'descriptor', descriptor, false, false), 'key'), J$.R(229681, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(239697, J$e);
                        } finally {
                            if (J$.Fr(239705))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(229849, arguments.callee, this, arguments);
                            arguments = J$.N(229857, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(229865, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(229873, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(229881, 'staticProps', staticProps, true, false, false);
                            if (J$.C(13272, J$.R(229745, 'protoProps', protoProps, false, false)))
                                J$.F(229785, J$.R(229753, '_defineProperties', _defineProperties, false, true), false)(J$.G(229769, J$.R(229761, 'Constructor', Constructor, false, false), 'prototype'), J$.R(229777, 'protoProps', protoProps, false, false));
                            if (J$.C(13280, J$.R(229793, 'staticProps', staticProps, false, false)))
                                J$.F(229825, J$.R(229801, '_defineProperties', _defineProperties, false, true), false)(J$.R(229809, 'Constructor', Constructor, false, false), J$.R(229817, 'staticProps', staticProps, false, false));
                            return J$.Rt(229841, J$.R(229833, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(239713, J$e);
                        } finally {
                            if (J$.Fr(239721))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(229993, arguments.callee, this, arguments);
                            arguments = J$.N(230001, 'arguments', arguments, true, false, false);
                            self = J$.N(230009, 'self', self, true, false, false);
                            call = J$.N(230017, 'call', call, true, false, false);
                            if (J$.C(13304, J$.C(13296, J$.R(229889, 'call', call, false, false)) ? J$.C(13288, J$.B(15058, '===', J$.F(229913, J$.R(229897, '_typeof', _typeof, false, true), false)(J$.R(229905, 'call', call, false, false)), J$.T(229921, 'object', 21, false))) ? J$._() : J$.B(15074, '===', J$.U(15066, 'typeof', J$.R(229929, 'call', call, false, false)), J$.T(229937, 'function', 21, false)) : J$._())) {
                                return J$.Rt(229953, J$.R(229945, 'call', call, false, false));
                            }
                            return J$.Rt(229985, J$.F(229977, J$.R(229961, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(229969, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(239729, J$e);
                        } finally {
                            if (J$.Fr(239737))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(230081, arguments.callee, this, arguments);
                            arguments = J$.N(230089, 'arguments', arguments, true, false, false);
                            self = J$.N(230097, 'self', self, true, false, false);
                            if (J$.C(13312, J$.B(15082, '===', J$.R(230025, 'self', self, false, false), void J$.T(230033, 0, 22, false)))) {
                                throw J$.F(230057, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(230041, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(230041, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(230049, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(230073, J$.R(230065, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(239745, J$e);
                        } finally {
                            if (J$.Fr(239753))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(230257, arguments.callee, this, arguments);
                            arguments = J$.N(230265, 'arguments', arguments, true, false, false);
                            o = J$.N(230273, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(230217, '_getPrototypeOf', J$.C(13328, J$.G(230113, J$.I(typeof Object === 'undefined' ? Object = J$.R(230105, 'Object', undefined, true, true) : Object = J$.R(230105, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(230129, J$.I(typeof Object === 'undefined' ? Object = J$.R(230121, 'Object', undefined, true, true) : Object = J$.R(230121, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(230209, function _getPrototypeOf(o) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(230185, arguments.callee, this, arguments);
                                            arguments = J$.N(230193, 'arguments', arguments, true, false, false);
                                            o = J$.N(230201, 'o', o, true, false, false);
                                            return J$.Rt(230177, J$.C(13320, J$.G(230145, J$.R(230137, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(230169, J$.I(typeof Object === 'undefined' ? Object = J$.R(230153, 'Object', undefined, true, true) : Object = J$.R(230153, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(230161, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(239761, J$e);
                                        } finally {
                                            if (J$.Fr(239769))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(230249, J$.F(230241, J$.R(230225, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(230233, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(239777, J$e);
                        } finally {
                            if (J$.Fr(239785))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(230473, arguments.callee, this, arguments);
                            arguments = J$.N(230481, 'arguments', arguments, true, false, false);
                            subClass = J$.N(230489, 'subClass', subClass, true, false, false);
                            superClass = J$.N(230497, 'superClass', superClass, true, false, false);
                            if (J$.C(13344, J$.C(13336, J$.B(15098, '!==', J$.U(15090, 'typeof', J$.R(230281, 'superClass', superClass, false, false)), J$.T(230289, 'function', 21, false))) ? J$.B(15106, '!==', J$.R(230297, 'superClass', superClass, false, false), J$.T(230305, null, 25, false)) : J$._())) {
                                throw J$.F(230329, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(230313, 'TypeError', undefined, true, true) : TypeError = J$.R(230313, 'TypeError', TypeError, true, true)), true)(J$.T(230321, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(230425, J$.R(230337, 'subClass', subClass, false, false), 'prototype', J$.M(230417, J$.I(typeof Object === 'undefined' ? Object = J$.R(230345, 'Object', undefined, true, true) : Object = J$.R(230345, 'Object', Object, true, true)), 'create', false)(J$.C(13352, J$.R(230353, 'superClass', superClass, false, false)) ? J$.G(230369, J$.R(230361, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(230409, {
                                constructor: J$.T(230401, {
                                    value: J$.R(230377, 'subClass', subClass, false, false),
                                    writable: J$.T(230385, true, 23, false),
                                    configurable: J$.T(230393, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(13360, J$.R(230433, 'superClass', superClass, false, false)))
                                J$.F(230465, J$.R(230441, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(230449, 'subClass', subClass, false, false), J$.R(230457, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(239793, J$e);
                        } finally {
                            if (J$.Fr(239801))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(230649, arguments.callee, this, arguments);
                            arguments = J$.N(230657, 'arguments', arguments, true, false, false);
                            o = J$.N(230665, 'o', o, true, false, false);
                            p = J$.N(230673, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(230601, '_setPrototypeOf', J$.C(13368, J$.G(230513, J$.I(typeof Object === 'undefined' ? Object = J$.R(230505, 'Object', undefined, true, true) : Object = J$.R(230505, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(230593, function _setPrototypeOf(o, p) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(230561, arguments.callee, this, arguments);
                                            arguments = J$.N(230569, 'arguments', arguments, true, false, false);
                                            o = J$.N(230577, 'o', o, true, false, false);
                                            p = J$.N(230585, 'p', p, true, false, false);
                                            J$.P(230537, J$.R(230521, 'o', o, false, false), '__proto__', J$.R(230529, 'p', p, false, false));
                                            return J$.Rt(230553, J$.R(230545, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(239809, J$e);
                                        } finally {
                                            if (J$.Fr(239817))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(230641, J$.F(230633, J$.R(230609, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(230617, 'o', o, false, false), J$.R(230625, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(239825, J$e);
                        } finally {
                            if (J$.Fr(239833))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function basicWriteValidations(server) {
                jalangiLabel36:
                    while (true) {
                        try {
                            J$.Fe(237249, arguments.callee, this, arguments);
                            arguments = J$.N(237257, 'arguments', arguments, true, false, false);
                            server = J$.N(237265, 'server', server, true, false, false);
                            if (J$.C(13672, J$.U(15234, '!', J$.G(237129, J$.G(237121, J$.R(237113, 'server', server, false, false), 's'), 'pool')))) {
                                return J$.Rt(237161, J$.F(237153, J$.R(237137, 'MongoError', MongoError, false, true), true)(J$.T(237145, 'server instance is not connected', 21, false)));
                            }
                            if (J$.C(13680, J$.M(237193, J$.G(237185, J$.G(237177, J$.R(237169, 'server', server, false, false), 's'), 'pool'), 'isDestroyed', false)())) {
                                return J$.Rt(237225, J$.F(237217, J$.R(237201, 'MongoError', MongoError, false, true), true)(J$.T(237209, 'server instance pool was destroyed', 21, false)));
                            }
                            return J$.Rt(237241, J$.T(237233, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(240209, J$e);
                        } finally {
                            if (J$.Fr(240217))
                                continue jalangiLabel36;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function basicReadValidations(server, options) {
                jalangiLabel37:
                    while (true) {
                        try {
                            J$.Fe(237409, arguments.callee, this, arguments);
                            arguments = J$.N(237417, 'arguments', arguments, true, false, false);
                            server = J$.N(237425, 'server', server, true, false, false);
                            options = J$.N(237433, 'options', options, true, false, false);
                            J$.N(237441, 'error', error, false, false, false);
                            var error = J$.W(237305, 'error', J$.F(237297, J$.R(237273, 'basicWriteValidations', basicWriteValidations, false, true), false)(J$.R(237281, 'server', server, false, false), J$.R(237289, 'options', options, false, false)), error, false, false);
                            if (J$.C(13688, J$.R(237313, 'error', error, false, false))) {
                                return J$.Rt(237329, J$.R(237321, 'error', error, false, false));
                            }
                            if (J$.C(13704, J$.C(13696, J$.G(237345, J$.R(237337, 'options', options, false, false), 'readPreference')) ? J$.U(15250, '!', J$.B(15242, 'instanceof', J$.G(237361, J$.R(237353, 'options', options, false, false), 'readPreference'), J$.R(237369, 'ReadPreference', ReadPreference, false, true))) : J$._())) {
                                return J$.Rt(237401, J$.F(237393, J$.R(237377, 'MongoError', MongoError, false, true), true)(J$.T(237385, 'readPreference must be an instance of ReadPreference', 21, false)));
                            }
                        } catch (J$e) {
                            J$.Ex(240225, J$e);
                        } finally {
                            if (J$.Fr(240233))
                                continue jalangiLabel37;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function executeWriteOperation(args, options, callback) {
                jalangiLabel39:
                    while (true) {
                        try {
                            J$.Fe(238153, arguments.callee, this, arguments);
                            arguments = J$.N(238161, 'arguments', arguments, true, false, false);
                            args = J$.N(238169, 'args', args, true, false, false);
                            options = J$.N(238177, 'options', options, true, false, false);
                            callback = J$.N(238185, 'callback', callback, true, false, false);
                            J$.N(238193, 'server', server, false, false, false);
                            J$.N(238201, 'op', op, false, false, false);
                            J$.N(238209, 'ns', ns, false, false, false);
                            J$.N(238217, 'ops', ops, false, false, false);
                            J$.N(238225, 'error', error, false, false, false);
                            if (J$.C(13712, J$.B(15266, '===', J$.U(15258, 'typeof', J$.R(237449, 'options', options, false, false)), J$.T(237457, 'function', 21, false))))
                                callback = J$.W(237473, 'callback', J$.R(237465, 'options', options, false, false), callback, false, false), options = J$.W(237489, 'options', J$.T(237481, {}, 11, false), options, false, false);
                            options = J$.W(237513, 'options', J$.C(13720, J$.R(237497, 'options', options, false, false)) ? J$._() : J$.T(237505, {}, 11, false), options, false, false);
                            var server = J$.W(237537, 'server', J$.G(237529, J$.R(237521, 'args', args, false, false), 'server'), server, false, false);
                            var op = J$.W(237561, 'op', J$.G(237553, J$.R(237545, 'args', args, false, false), 'op'), op, false, false);
                            var ns = J$.W(237585, 'ns', J$.G(237577, J$.R(237569, 'args', args, false, false), 'ns'), ns, false, false);
                            var ops = J$.W(237665, 'ops', J$.C(13728, J$.M(237617, J$.I(typeof Array === 'undefined' ? Array = J$.R(237593, 'Array', undefined, true, true) : Array = J$.R(237593, 'Array', Array, true, true)), 'isArray', false)(J$.G(237609, J$.R(237601, 'args', args, false, false), 'ops'))) ? J$.G(237633, J$.R(237625, 'args', args, false, false), 'ops') : J$.T(237657, [J$.G(237649, J$.R(237641, 'args', args, false, false), 'ops')], 10, false), ops, false, false);
                            var error = J$.W(237705, 'error', J$.F(237697, J$.R(237673, 'basicWriteValidations', basicWriteValidations, false, true), false)(J$.R(237681, 'server', server, false, false), J$.R(237689, 'options', options, false, false)), error, false, false);
                            if (J$.C(13736, J$.R(237713, 'error', error, false, false))) {
                                J$.F(237745, J$.R(237721, 'callback', callback, false, false), false)(J$.R(237729, 'error', error, false, false), J$.T(237737, null, 25, false));
                                return J$.Rt(237753, undefined);
                            }
                            if (J$.C(13744, J$.F(237785, J$.R(237761, 'collationNotSupported', collationNotSupported, false, true), false)(J$.R(237769, 'server', server, false, false), J$.R(237777, 'options', options, false, false)))) {
                                J$.F(237857, J$.R(237793, 'callback', callback, false, false), false)(J$.F(237849, J$.R(237801, 'MongoError', MongoError, false, true), true)(J$.M(237841, J$.T(237809, 'server ', 21, false), 'concat', false)(J$.G(237825, J$.R(237817, 'server', server, false, false), 'name'), J$.T(237833, ' does not support collation', 21, false))));
                                return J$.Rt(237865, undefined);
                            }
                            return J$.Rt(238145, J$.M(238137, J$.R(237873, 'wireProtocol', wireProtocol, false, true), J$.R(237881, 'op', op, false, false), false)(J$.R(237889, 'server', server, false, false), J$.R(237897, 'ns', ns, false, false), J$.R(237905, 'ops', ops, false, false), J$.R(237913, 'options', options, false, false), J$.T(238129, function (err, result) {
                                jalangiLabel38:
                                    while (true) {
                                        try {
                                            J$.Fe(238097, arguments.callee, this, arguments);
                                            arguments = J$.N(238105, 'arguments', arguments, true, false, false);
                                            err = J$.N(238113, 'err', err, true, false, false);
                                            result = J$.N(238121, 'result', result, true, false, false);
                                            if (J$.C(13776, J$.R(237921, 'err', err, false, false))) {
                                                if (J$.C(13760, J$.C(13752, J$.G(237937, J$.R(237929, 'options', options, false, false), 'session')) ? J$.B(15274, 'instanceof', J$.R(237945, 'err', err, false, false), J$.R(237953, 'MongoNetworkError', MongoNetworkError, false, true)) : J$._())) {
                                                    J$.P(237993, J$.G(237977, J$.G(237969, J$.R(237961, 'options', options, false, false), 'session'), 'serverSession'), 'isDirty', J$.T(237985, true, 23, false));
                                                }
                                                if (J$.C(13768, J$.F(238025, J$.R(238001, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true), false)(J$.R(238009, 'err', err, false, false), J$.R(238017, 'server', server, false, false)))) {
                                                    J$.M(238057, J$.R(238033, 'server', server, false, false), 'emit', false)(J$.T(238041, 'error', 21, false), J$.R(238049, 'err', err, false, false));
                                                }
                                            }
                                            J$.F(238089, J$.R(238065, 'callback', callback, false, false), false)(J$.R(238073, 'err', err, false, false), J$.R(238081, 'result', result, false, false));
                                        } catch (J$e) {
                                            J$.Ex(240241, J$e);
                                        } finally {
                                            if (J$.Fr(240249))
                                                continue jalangiLabel38;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(240257, J$e);
                        } finally {
                            if (J$.Fr(240265))
                                continue jalangiLabel39;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function connectEventHandler(server) {
                jalangiLabel41:
                    while (true) {
                        try {
                            J$.Fe(238873, arguments.callee, this, arguments);
                            arguments = J$.N(238881, 'arguments', arguments, true, false, false);
                            server = J$.N(238889, 'server', server, true, false, false);
                            return J$.Rt(238865, J$.T(238857, function (pool, conn) {
                                jalangiLabel40:
                                    while (true) {
                                        try {
                                            J$.Fe(238809, arguments.callee, this, arguments);
                                            arguments = J$.N(238817, 'arguments', arguments, true, false, false);
                                            pool = J$.N(238825, 'pool', pool, true, false, false);
                                            conn = J$.N(238833, 'conn', conn, true, false, false);
                                            J$.N(238841, 'ismaster', ismaster, false, false, false);
                                            J$.N(238849, '$clusterTime', $clusterTime, false, false, false);
                                            var ismaster = J$.W(238249, 'ismaster', J$.G(238241, J$.R(238233, 'conn', conn, false, false), 'ismaster'), ismaster, false, false);
                                            J$.P(238289, J$.G(238265, J$.R(238257, 'server', server, false, false), 's'), 'lastIsMasterMS', J$.G(238281, J$.R(238273, 'conn', conn, false, false), 'lastIsMasterMS'));
                                            if (J$.C(13784, J$.G(238305, J$.R(238297, 'conn', conn, false, false), 'agreedCompressor'))) {
                                                J$.P(238361, J$.G(238337, J$.G(238329, J$.G(238321, J$.R(238313, 'server', server, false, false), 's'), 'pool'), 'options'), 'agreedCompressor', J$.G(238353, J$.R(238345, 'conn', conn, false, false), 'agreedCompressor'));
                                            }
                                            if (J$.C(13792, J$.G(238377, J$.R(238369, 'conn', conn, false, false), 'zlibCompressionLevel'))) {
                                                J$.P(238433, J$.G(238409, J$.G(238401, J$.G(238393, J$.R(238385, 'server', server, false, false), 's'), 'pool'), 'options'), 'zlibCompressionLevel', J$.G(238425, J$.R(238417, 'conn', conn, false, false), 'zlibCompressionLevel'));
                                            }
                                            if (J$.C(13800, J$.G(238457, J$.G(238449, J$.R(238441, 'conn', conn, false, false), 'ismaster'), '$clusterTime'))) {
                                                var $clusterTime = J$.W(238489, '$clusterTime', J$.G(238481, J$.G(238473, J$.R(238465, 'conn', conn, false, false), 'ismaster'), '$clusterTime'), $clusterTime, false, false);
                                                J$.P(238521, J$.G(238505, J$.R(238497, 'server', server, false, false), 's'), 'sclusterTime', J$.R(238513, '$clusterTime', $clusterTime, false, false));
                                            }
                                            if (J$.C(13808, J$.M(238553, J$.G(238545, J$.G(238537, J$.R(238529, 'server', server, false, false), 's'), 'logger'), 'isInfo', false)())) {
                                                J$.M(238665, J$.G(238577, J$.G(238569, J$.R(238561, 'server', server, false, false), 's'), 'logger'), 'info', false)(J$.M(238657, J$.M(238617, J$.T(238585, 'server ', 21, false), 'concat', false)(J$.G(238601, J$.R(238593, 'server', server, false, false), 'name'), J$.T(238609, ' connected with ismaster [', 21, false)), 'concat', false)(J$.M(238641, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(238625, 'JSON', undefined, true, true) : JSON = J$.R(238625, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(238633, 'ismaster', ismaster, false, false)), J$.T(238649, ']', 21, false)));
                                            }
                                            J$.M(238737, J$.R(238673, 'server', server, false, false), 'emit', false)(J$.T(238681, 'descriptionReceived', 21, false), J$.F(238729, J$.R(238689, 'ServerDescription', ServerDescription, false, true), true)(J$.G(238713, J$.G(238705, J$.R(238697, 'server', server, false, false), 'description'), 'address'), J$.R(238721, 'ismaster', ismaster, false, false)));
                                            J$.P(238769, J$.G(238753, J$.R(238745, 'server', server, false, false), 's'), 'state', J$.R(238761, 'STATE_CONNECTED', STATE_CONNECTED, false, true));
                                            J$.M(238801, J$.R(238777, 'server', server, false, false), 'emit', false)(J$.T(238785, 'connect', 21, false), J$.R(238793, 'server', server, false, false));
                                        } catch (J$e) {
                                            J$.Ex(240273, J$e);
                                        } finally {
                                            if (J$.Fr(240281))
                                                continue jalangiLabel40;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(240289, J$e);
                        } finally {
                            if (J$.Fr(240297))
                                continue jalangiLabel41;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function errorEventHandler(server) {
                jalangiLabel43:
                    while (true) {
                        try {
                            J$.Fe(239017, arguments.callee, this, arguments);
                            arguments = J$.N(239025, 'arguments', arguments, true, false, false);
                            server = J$.N(239033, 'server', server, true, false, false);
                            return J$.Rt(239009, J$.T(239001, function (err) {
                                jalangiLabel42:
                                    while (true) {
                                        try {
                                            J$.Fe(238977, arguments.callee, this, arguments);
                                            arguments = J$.N(238985, 'arguments', arguments, true, false, false);
                                            err = J$.N(238993, 'err', err, true, false, false);
                                            if (J$.C(13816, J$.R(238897, 'err', err, false, false))) {
                                                J$.M(238945, J$.R(238905, 'server', server, false, false), 'emit', false)(J$.T(238913, 'error', 21, false), J$.F(238937, J$.R(238921, 'MongoNetworkError', MongoNetworkError, false, true), true)(J$.R(238929, 'err', err, false, false)));
                                            }
                                            J$.M(238969, J$.R(238953, 'server', server, false, false), 'emit', false)(J$.T(238961, 'close', 21, false));
                                        } catch (J$e) {
                                            J$.Ex(240305, J$e);
                                        } finally {
                                            if (J$.Fr(240313))
                                                continue jalangiLabel42;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(240321, J$e);
                        } finally {
                            if (J$.Fr(240329))
                                continue jalangiLabel43;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function parseErrorEventHandler(server) {
                jalangiLabel45:
                    while (true) {
                        try {
                            J$.Fe(239161, arguments.callee, this, arguments);
                            arguments = J$.N(239169, 'arguments', arguments, true, false, false);
                            server = J$.N(239177, 'server', server, true, false, false);
                            return J$.Rt(239153, J$.T(239145, function (err) {
                                jalangiLabel44:
                                    while (true) {
                                        try {
                                            J$.Fe(239121, arguments.callee, this, arguments);
                                            arguments = J$.N(239129, 'arguments', arguments, true, false, false);
                                            err = J$.N(239137, 'err', err, true, false, false);
                                            J$.P(239065, J$.G(239049, J$.R(239041, 'server', server, false, false), 's'), 'state', J$.R(239057, 'STATE_DISCONNECTED', STATE_DISCONNECTED, false, true));
                                            J$.M(239113, J$.R(239073, 'server', server, false, false), 'emit', false)(J$.T(239081, 'error', 21, false), J$.F(239105, J$.R(239089, 'MongoParseError', MongoParseError, false, true), true)(J$.R(239097, 'err', err, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(240337, J$e);
                                        } finally {
                                            if (J$.Fr(240345))
                                                continue jalangiLabel44;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(240353, J$e);
                        } finally {
                            if (J$.Fr(240361))
                                continue jalangiLabel45;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(239225, '_typeof', J$.T(239217, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(239241, '_classCallCheck', J$.T(239233, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(239257, '_defineProperties', J$.T(239249, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(239273, '_createClass', J$.T(239265, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(239289, '_possibleConstructorReturn', J$.T(239281, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(239305, '_assertThisInitialized', J$.T(239297, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(239321, '_getPrototypeOf', J$.T(239313, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(239337, '_inherits', J$.T(239329, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(239353, '_setPrototypeOf', J$.T(239345, _setPrototypeOf, 12, false), true, false, false);
            J$.N(239361, 'EventEmitter', EventEmitter, false, false, false);
            J$.N(239369, 'MongoError', MongoError, false, false, false);
            J$.N(239377, 'Pool', Pool, false, false, false);
            J$.N(239385, 'relayEvents', relayEvents, false, false, false);
            J$.N(239393, 'wireProtocol', wireProtocol, false, false, false);
            J$.N(239401, 'BSON', BSON, false, false, false);
            J$.N(239409, 'createClientInfo', createClientInfo, false, false, false);
            J$.N(239417, 'Logger', Logger, false, false, false);
            J$.N(239425, 'ServerDescription', ServerDescription, false, false, false);
            J$.N(239433, 'ReadPreference', ReadPreference, false, false, false);
            J$.N(239441, 'monitorServer', monitorServer, false, false, false);
            J$.N(239449, 'MongoParseError', MongoParseError, false, false, false);
            J$.N(239457, 'MongoNetworkError', MongoNetworkError, false, false, false);
            J$.N(239465, 'collationNotSupported', collationNotSupported, false, false, false);
            J$.N(239473, 'debugOptions', debugOptions, false, false, false);
            J$.N(239481, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, false, false);
            J$.N(239489, 'DEBUG_FIELDS', DEBUG_FIELDS, false, false, false);
            J$.N(239497, 'STATE_DISCONNECTING', STATE_DISCONNECTING, false, false, false);
            J$.N(239505, 'STATE_DISCONNECTED', STATE_DISCONNECTED, false, false, false);
            J$.N(239513, 'STATE_CONNECTING', STATE_CONNECTING, false, false, false);
            J$.N(239521, 'STATE_CONNECTED', STATE_CONNECTED, false, false, false);
            J$.N(239529, 'Server', Server, false, false, false);
            basicWriteValidations = J$.N(239545, 'basicWriteValidations', J$.T(239537, basicWriteValidations, 12, false), true, false, false);
            basicReadValidations = J$.N(239561, 'basicReadValidations', J$.T(239553, basicReadValidations, 12, false), true, false, false);
            executeWriteOperation = J$.N(239577, 'executeWriteOperation', J$.T(239569, executeWriteOperation, 12, false), true, false, false);
            connectEventHandler = J$.N(239593, 'connectEventHandler', J$.T(239585, connectEventHandler, 12, false), true, false, false);
            errorEventHandler = J$.N(239609, 'errorEventHandler', J$.T(239601, errorEventHandler, 12, false), true, false, false);
            parseErrorEventHandler = J$.N(239625, 'parseErrorEventHandler', J$.T(239617, parseErrorEventHandler, 12, false), true, false, false);
            J$.T(229089, 'use strict', 21, false);
            var EventEmitter = J$.W(230705, 'EventEmitter', J$.F(230697, J$.I(typeof require === 'undefined' ? require = J$.R(230681, 'require', undefined, true, true) : require = J$.R(230681, 'require', require, true, true)), false)(J$.T(230689, 'events', 21, false)), EventEmitter, false, true);
            var MongoError = J$.W(230745, 'MongoError', J$.G(230737, J$.F(230729, J$.I(typeof require === 'undefined' ? require = J$.R(230713, 'require', undefined, true, true) : require = J$.R(230713, 'require', require, true, true)), false)(J$.T(230721, '../error', 21, false)), 'MongoError'), MongoError, false, true);
            var Pool = J$.W(230777, 'Pool', J$.F(230769, J$.I(typeof require === 'undefined' ? require = J$.R(230753, 'require', undefined, true, true) : require = J$.R(230753, 'require', require, true, true)), false)(J$.T(230761, '../connection/pool', 21, false)), Pool, false, true);
            var relayEvents = J$.W(230817, 'relayEvents', J$.G(230809, J$.F(230801, J$.I(typeof require === 'undefined' ? require = J$.R(230785, 'require', undefined, true, true) : require = J$.R(230785, 'require', require, true, true)), false)(J$.T(230793, '../utils', 21, false)), 'relayEvents'), relayEvents, false, true);
            var wireProtocol = J$.W(230849, 'wireProtocol', J$.F(230841, J$.I(typeof require === 'undefined' ? require = J$.R(230825, 'require', undefined, true, true) : require = J$.R(230825, 'require', require, true, true)), false)(J$.T(230833, '../wireprotocol', 21, false)), wireProtocol, false, true);
            var BSON = J$.W(230889, 'BSON', J$.M(230881, J$.F(230873, J$.I(typeof require === 'undefined' ? require = J$.R(230857, 'require', undefined, true, true) : require = J$.R(230857, 'require', require, true, true)), false)(J$.T(230865, '../connection/utils', 21, false)), 'retrieveBSON', false)(), BSON, false, true);
            var createClientInfo = J$.W(230929, 'createClientInfo', J$.G(230921, J$.F(230913, J$.I(typeof require === 'undefined' ? require = J$.R(230897, 'require', undefined, true, true) : require = J$.R(230897, 'require', require, true, true)), false)(J$.T(230905, '../topologies/shared', 21, false)), 'createClientInfo'), createClientInfo, false, true);
            var Logger = J$.W(230961, 'Logger', J$.F(230953, J$.I(typeof require === 'undefined' ? require = J$.R(230937, 'require', undefined, true, true) : require = J$.R(230937, 'require', require, true, true)), false)(J$.T(230945, '../connection/logger', 21, false)), Logger, false, true);
            var ServerDescription = J$.W(231001, 'ServerDescription', J$.G(230993, J$.F(230985, J$.I(typeof require === 'undefined' ? require = J$.R(230969, 'require', undefined, true, true) : require = J$.R(230969, 'require', require, true, true)), false)(J$.T(230977, './server_description', 21, false)), 'ServerDescription'), ServerDescription, false, true);
            var ReadPreference = J$.W(231033, 'ReadPreference', J$.F(231025, J$.I(typeof require === 'undefined' ? require = J$.R(231009, 'require', undefined, true, true) : require = J$.R(231009, 'require', require, true, true)), false)(J$.T(231017, '../topologies/read_preference', 21, false)), ReadPreference, false, true);
            var monitorServer = J$.W(231073, 'monitorServer', J$.G(231065, J$.F(231057, J$.I(typeof require === 'undefined' ? require = J$.R(231041, 'require', undefined, true, true) : require = J$.R(231041, 'require', require, true, true)), false)(J$.T(231049, './monitoring', 21, false)), 'monitorServer'), monitorServer, false, true);
            var MongoParseError = J$.W(231113, 'MongoParseError', J$.G(231105, J$.F(231097, J$.I(typeof require === 'undefined' ? require = J$.R(231081, 'require', undefined, true, true) : require = J$.R(231081, 'require', require, true, true)), false)(J$.T(231089, '../error', 21, false)), 'MongoParseError'), MongoParseError, false, true);
            var MongoNetworkError = J$.W(231153, 'MongoNetworkError', J$.G(231145, J$.F(231137, J$.I(typeof require === 'undefined' ? require = J$.R(231121, 'require', undefined, true, true) : require = J$.R(231121, 'require', require, true, true)), false)(J$.T(231129, '../error', 21, false)), 'MongoNetworkError'), MongoNetworkError, false, true);
            var collationNotSupported = J$.W(231193, 'collationNotSupported', J$.G(231185, J$.F(231177, J$.I(typeof require === 'undefined' ? require = J$.R(231161, 'require', undefined, true, true) : require = J$.R(231161, 'require', require, true, true)), false)(J$.T(231169, '../utils', 21, false)), 'collationNotSupported'), collationNotSupported, false, true);
            var debugOptions = J$.W(231233, 'debugOptions', J$.G(231225, J$.F(231217, J$.I(typeof require === 'undefined' ? require = J$.R(231201, 'require', undefined, true, true) : require = J$.R(231201, 'require', require, true, true)), false)(J$.T(231209, '../connection/utils', 21, false)), 'debugOptions'), debugOptions, false, true);
            var isSDAMUnrecoverableError = J$.W(231273, 'isSDAMUnrecoverableError', J$.G(231265, J$.F(231257, J$.I(typeof require === 'undefined' ? require = J$.R(231241, 'require', undefined, true, true) : require = J$.R(231241, 'require', require, true, true)), false)(J$.T(231249, '../error', 21, false)), 'isSDAMUnrecoverableError'), isSDAMUnrecoverableError, false, true);
            var DEBUG_FIELDS = J$.W(231481, 'DEBUG_FIELDS', J$.T(231473, [
                J$.T(231281, 'reconnect', 21, false),
                J$.T(231289, 'reconnectTries', 21, false),
                J$.T(231297, 'reconnectInterval', 21, false),
                J$.T(231305, 'emitError', 21, false),
                J$.T(231313, 'cursorFactory', 21, false),
                J$.T(231321, 'host', 21, false),
                J$.T(231329, 'port', 21, false),
                J$.T(231337, 'size', 21, false),
                J$.T(231345, 'keepAlive', 21, false),
                J$.T(231353, 'keepAliveInitialDelay', 21, false),
                J$.T(231361, 'noDelay', 21, false),
                J$.T(231369, 'connectionTimeout', 21, false),
                J$.T(231377, 'checkServerIdentity', 21, false),
                J$.T(231385, 'socketTimeout', 21, false),
                J$.T(231393, 'ssl', 21, false),
                J$.T(231401, 'ca', 21, false),
                J$.T(231409, 'crl', 21, false),
                J$.T(231417, 'cert', 21, false),
                J$.T(231425, 'key', 21, false),
                J$.T(231433, 'rejectUnauthorized', 21, false),
                J$.T(231441, 'promoteLongs', 21, false),
                J$.T(231449, 'promoteValues', 21, false),
                J$.T(231457, 'promoteBuffers', 21, false),
                J$.T(231465, 'servername', 21, false)
            ], 10, false), DEBUG_FIELDS, false, true);
            var STATE_DISCONNECTING = J$.W(231497, 'STATE_DISCONNECTING', J$.T(231489, 'disconnecting', 21, false), STATE_DISCONNECTING, false, true);
            var STATE_DISCONNECTED = J$.W(231513, 'STATE_DISCONNECTED', J$.T(231505, 'disconnected', 21, false), STATE_DISCONNECTED, false, true);
            var STATE_CONNECTING = J$.W(231529, 'STATE_CONNECTING', J$.T(231521, 'connecting', 21, false), STATE_CONNECTING, false, true);
            var STATE_CONNECTED = J$.W(231545, 'STATE_CONNECTED', J$.T(231537, 'connected', 21, false), STATE_CONNECTED, false, true);
            var Server = J$.W(236921, 'Server', J$.F(236913, J$.T(236897, function (_EventEmitter) {
                jalangiLabel33:
                    while (true) {
                        try {
                            J$.Fe(236857, arguments.callee, this, arguments);
                            function Server(description, options, topology) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(232137, arguments.callee, this, arguments);
                                            arguments = J$.N(232145, 'arguments', arguments, true, false, false);
                                            description = J$.N(232153, 'description', description, true, false, false);
                                            options = J$.N(232161, 'options', options, true, false, false);
                                            topology = J$.N(232169, 'topology', topology, true, false, false);
                                            J$.N(232177, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(231609, J$.R(231585, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(231593, 'this', this, false, false), J$.R(231601, 'Server', Server, false, false));
                                            _this = J$.W(231681, '_this', J$.F(231673, J$.R(231617, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(231625, 'this', this, false, false), J$.M(231665, J$.F(231649, J$.R(231633, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(231641, 'Server', Server, false, false)), 'call', false)(J$.R(231657, 'this', this, false, false))), _this, false, false);
                                            J$.P(232113, J$.R(231689, '_this', _this, false, false), 's', J$.T(232105, {
                                                description: J$.R(231697, 'description', description, false, false),
                                                options: J$.R(231705, 'options', options, false, false),
                                                logger: J$.F(231737, J$.R(231713, 'Logger', Logger, false, true), false)(J$.T(231721, 'Server', 21, false), J$.R(231729, 'options', options, false, false)),
                                                bson: J$.C(13376, J$.G(231753, J$.R(231745, 'options', options, false, false), 'bson')) ? J$._() : J$.F(232001, J$.R(231761, 'BSON', BSON, false, true), true)(J$.T(231993, [
                                                    J$.G(231777, J$.R(231769, 'BSON', BSON, false, true), 'Binary'),
                                                    J$.G(231793, J$.R(231785, 'BSON', BSON, false, true), 'Code'),
                                                    J$.G(231809, J$.R(231801, 'BSON', BSON, false, true), 'DBRef'),
                                                    J$.G(231825, J$.R(231817, 'BSON', BSON, false, true), 'Decimal128'),
                                                    J$.G(231841, J$.R(231833, 'BSON', BSON, false, true), 'Double'),
                                                    J$.G(231857, J$.R(231849, 'BSON', BSON, false, true), 'Int32'),
                                                    J$.G(231873, J$.R(231865, 'BSON', BSON, false, true), 'Long'),
                                                    J$.G(231889, J$.R(231881, 'BSON', BSON, false, true), 'Map'),
                                                    J$.G(231905, J$.R(231897, 'BSON', BSON, false, true), 'MaxKey'),
                                                    J$.G(231921, J$.R(231913, 'BSON', BSON, false, true), 'MinKey'),
                                                    J$.G(231937, J$.R(231929, 'BSON', BSON, false, true), 'ObjectId'),
                                                    J$.G(231953, J$.R(231945, 'BSON', BSON, false, true), 'BSONRegExp'),
                                                    J$.G(231969, J$.R(231961, 'BSON', BSON, false, true), 'Symbol'),
                                                    J$.G(231985, J$.R(231977, 'BSON', BSON, false, true), 'Timestamp')
                                                ], 10, false)),
                                                clientInfo: J$.F(232025, J$.R(232009, 'createClientInfo', createClientInfo, false, true), false)(J$.R(232017, 'options', options, false, false)),
                                                monitoring: J$.T(232033, false, 23, false),
                                                monitorFunction: J$.C(13384, J$.G(232049, J$.R(232041, 'options', options, false, false), 'monitorFunction')) ? J$._() : J$.R(232057, 'monitorServer', monitorServer, false, true),
                                                pool: J$.T(232065, null, 25, false),
                                                state: J$.R(232073, 'STATE_DISCONNECTED', STATE_DISCONNECTED, false, true),
                                                credentials: J$.G(232089, J$.R(232081, 'options', options, false, false), 'credentials'),
                                                topology: J$.R(232097, 'topology', topology, false, false)
                                            }, 11, false));
                                            return J$.Rt(232129, J$.R(232121, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(239841, J$e);
                                        } finally {
                                            if (J$.Fr(239849))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(236865, 'arguments', arguments, true, false, false);
                            _EventEmitter = J$.N(236873, '_EventEmitter', _EventEmitter, true, false, false);
                            Server = J$.N(236889, 'Server', J$.T(236881, Server, 12, false), true, false, false);
                            J$.F(231577, J$.R(231553, '_inherits', _inherits, false, true), false)(J$.R(231561, 'Server', Server, false, false), J$.R(231569, '_EventEmitter', _EventEmitter, false, false));
                            J$.F(236833, J$.R(232185, '_createClass', _createClass, false, true), false)(J$.R(232193, 'Server', Server, false, false), J$.T(236825, [
                                J$.T(233257, {
                                    key: J$.T(232201, 'connect', 21, false),
                                    value: J$.T(233249, function connect(options) {
                                        jalangiLabel14:
                                            while (true) {
                                                try {
                                                    J$.Fe(233209, arguments.callee, this, arguments);
                                                    arguments = J$.N(233217, 'arguments', arguments, true, false, false);
                                                    options = J$.N(233225, 'options', options, true, false, false);
                                                    J$.N(233233, 'addressParts', addressParts, false, false, false);
                                                    J$.N(233241, 'poolOptions', poolOptions, false, false, false);
                                                    options = J$.W(232225, 'options', J$.C(13392, J$.R(232209, 'options', options, false, false)) ? J$._() : J$.T(232217, {}, 11, false), options, false, false);
                                                    if (J$.C(13416, J$.C(13408, J$.C(13400, J$.G(232249, J$.G(232241, J$.R(232233, 'this', this, false, false), 's'), 'pool')) ? J$.U(15114, '!', J$.M(232281, J$.G(232273, J$.G(232265, J$.R(232257, 'this', this, false, false), 's'), 'pool'), 'isDisconnected', false)()) : J$._()) ? J$.U(15122, '!', J$.M(232313, J$.G(232305, J$.G(232297, J$.R(232289, 'this', this, false, false), 's'), 'pool'), 'isDestroyed', false)()) : J$._())) {
                                                        throw J$.F(232377, J$.R(232321, 'MongoError', MongoError, false, true), true)(J$.M(232369, J$.T(232329, 'Server instance in invalid state ', 21, false), 'concat', false)(J$.G(232361, J$.G(232353, J$.G(232345, J$.R(232337, 'this', this, false, false), 's'), 'pool'), 'state')));
                                                    }
                                                    var addressParts = J$.W(232425, 'addressParts', J$.M(232417, J$.G(232401, J$.G(232393, J$.R(232385, 'this', this, false, false), 'description'), 'address'), 'split', false)(J$.T(232409, ':', 21, false)), addressParts, false, false);
                                                    var poolOptions = J$.W(232593, 'poolOptions', J$.M(232585, J$.I(typeof Object === 'undefined' ? Object = J$.R(232433, 'Object', undefined, true, true) : Object = J$.R(232433, 'Object', Object, true, true)), 'assign', false)(J$.T(232513, {
                                                        host: J$.G(232457, J$.R(232441, 'addressParts', addressParts, false, false), J$.T(232449, 0, 22, false)),
                                                        port: J$.F(232505, J$.I(typeof parseInt === 'undefined' ? parseInt = J$.R(232465, 'parseInt', undefined, true, true) : parseInt = J$.R(232465, 'parseInt', parseInt, true, true)), false)(J$.G(232489, J$.R(232473, 'addressParts', addressParts, false, false), J$.T(232481, 1, 22, false)), J$.T(232497, 10, 22, false))
                                                    }, 11, false), J$.G(232537, J$.G(232529, J$.R(232521, 'this', this, false, false), 's'), 'options'), J$.R(232545, 'options', options, false, false), J$.T(232577, { bson: J$.G(232569, J$.G(232561, J$.R(232553, 'this', this, false, false), 's'), 'bson') }, 11, false)), poolOptions, false, false);
                                                    J$.P(232617, J$.R(232601, 'poolOptions', poolOptions, false, false), 'reconnect', J$.T(232609, true, 23, false));
                                                    J$.P(232641, J$.R(232625, 'poolOptions', poolOptions, false, false), 'legacyCompatMode', J$.T(232633, false, 23, false));
                                                    J$.P(232697, J$.G(232657, J$.R(232649, 'this', this, false, false), 's'), 'pool', J$.F(232689, J$.R(232665, 'Pool', Pool, false, true), true)(J$.R(232673, 'this', this, false, false), J$.R(232681, 'poolOptions', poolOptions, false, false)));
                                                    J$.M(232761, J$.G(232721, J$.G(232713, J$.R(232705, 'this', this, false, false), 's'), 'pool'), 'on', false)(J$.T(232729, 'connect', 21, false), J$.F(232753, J$.R(232737, 'connectEventHandler', connectEventHandler, false, true), false)(J$.R(232745, 'this', this, false, false)));
                                                    J$.M(232825, J$.G(232785, J$.G(232777, J$.R(232769, 'this', this, false, false), 's'), 'pool'), 'on', false)(J$.T(232793, 'close', 21, false), J$.F(232817, J$.R(232801, 'errorEventHandler', errorEventHandler, false, true), false)(J$.R(232809, 'this', this, false, false)));
                                                    J$.M(232889, J$.G(232849, J$.G(232841, J$.R(232833, 'this', this, false, false), 's'), 'pool'), 'on', false)(J$.T(232857, 'error', 21, false), J$.F(232881, J$.R(232865, 'errorEventHandler', errorEventHandler, false, true), false)(J$.R(232873, 'this', this, false, false)));
                                                    J$.M(232953, J$.G(232913, J$.G(232905, J$.R(232897, 'this', this, false, false), 's'), 'pool'), 'on', false)(J$.T(232921, 'parseError', 21, false), J$.F(232945, J$.R(232929, 'parseErrorEventHandler', parseErrorEventHandler, false, true), false)(J$.R(232937, 'this', this, false, false)));
                                                    J$.F(233033, J$.R(232961, 'relayEvents', relayEvents, false, true), false)(J$.G(232985, J$.G(232977, J$.R(232969, 'this', this, false, false), 's'), 'pool'), J$.R(232993, 'this', this, false, false), J$.T(233025, [
                                                        J$.T(233001, 'commandStarted', 21, false),
                                                        J$.T(233009, 'commandSucceeded', 21, false),
                                                        J$.T(233017, 'commandFailed', 21, false)
                                                    ], 10, false));
                                                    J$.P(233065, J$.G(233049, J$.R(233041, 'this', this, false, false), 's'), 'state', J$.R(233057, 'STATE_CONNECTING', STATE_CONNECTING, false, true));
                                                    if (J$.C(13424, J$.G(233081, J$.R(233073, 'options', options, false, false), 'auth'))) {
                                                        J$.M(233161, J$.G(233113, J$.G(233105, J$.G(233097, J$.R(233089, 'this', this, false, false), 's'), 'pool'), 'connect'), 'apply', false)(J$.G(233137, J$.G(233129, J$.R(233121, 'this', this, false, false), 's'), 'pool'), J$.G(233153, J$.R(233145, 'options', options, false, false), 'auth'));
                                                        return J$.Rt(233169, undefined);
                                                    }
                                                    J$.M(233201, J$.G(233193, J$.G(233185, J$.R(233177, 'this', this, false, false), 's'), 'pool'), 'connect', false)();
                                                } catch (J$e) {
                                                    J$.Ex(239857, J$e);
                                                } finally {
                                                    if (J$.Fr(239865))
                                                        continue jalangiLabel14;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(233921, {
                                    key: J$.T(233265, 'destroy', 21, false),
                                    value: J$.T(233913, function destroy(options, callback) {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(233865, arguments.callee, this, arguments);
                                                    arguments = J$.N(233873, 'arguments', arguments, true, false, false);
                                                    options = J$.N(233881, 'options', options, true, false, false);
                                                    callback = J$.N(233889, 'callback', callback, true, false, false);
                                                    J$.N(233897, '_this2', _this2, false, false, false);
                                                    J$.N(233905, 'done', done, false, false, false);
                                                    var _this2 = J$.W(233281, '_this2', J$.R(233273, 'this', this, false, false), _this2, false, false);
                                                    if (J$.C(13432, J$.B(15138, '===', J$.U(15130, 'typeof', J$.R(233289, 'options', options, false, false)), J$.T(233297, 'function', 21, false))))
                                                        callback = J$.W(233313, 'callback', J$.R(233305, 'options', options, false, false), callback, false, false), options = J$.W(233329, 'options', J$.T(233321, {}, 11, false), options, false, false);
                                                    options = J$.W(233385, 'options', J$.M(233377, J$.I(typeof Object === 'undefined' ? Object = J$.R(233337, 'Object', undefined, true, true) : Object = J$.R(233337, 'Object', Object, true, true)), 'assign', false)(J$.T(233345, {}, 11, false), J$.T(233361, { force: J$.T(233353, false, 23, false) }, 11, false), J$.R(233369, 'options', options, false, false)), options, false, false);
                                                    J$.P(233417, J$.G(233401, J$.R(233393, 'this', this, false, false), 's'), 'state', J$.R(233409, 'STATE_DISCONNECTING', STATE_DISCONNECTING, false, true));
                                                    var done = J$.W(233561, 'done', J$.T(233553, function done(err) {
                                                        jalangiLabel15:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(233529, arguments.callee, this, arguments);
                                                                    arguments = J$.N(233537, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(233545, 'err', err, true, false, false);
                                                                    J$.M(233441, J$.R(233425, '_this2', _this2, false, false), 'emit', false)(J$.T(233433, 'closed', 21, false));
                                                                    J$.P(233473, J$.G(233457, J$.R(233449, '_this2', _this2, false, false), 's'), 'state', J$.R(233465, 'STATE_DISCONNECTED', STATE_DISCONNECTED, false, true));
                                                                    if (J$.C(13440, J$.B(15154, '===', J$.U(15146, 'typeof', J$.R(233481, 'callback', callback, false, false)), J$.T(233489, 'function', 21, false)))) {
                                                                        J$.F(233521, J$.R(233497, 'callback', callback, false, false), false)(J$.R(233505, 'err', err, false, false), J$.T(233513, null, 25, false));
                                                                    }
                                                                } catch (J$e) {
                                                                    J$.Ex(239873, J$e);
                                                                } finally {
                                                                    if (J$.Fr(239881))
                                                                        continue jalangiLabel15;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false), done, false, false);
                                                    if (J$.C(13448, J$.U(15162, '!', J$.G(233585, J$.G(233577, J$.R(233569, 'this', this, false, false), 's'), 'pool')))) {
                                                        return J$.Rt(233609, J$.F(233601, J$.R(233593, 'done', done, false, false), false)());
                                                    }
                                                    J$.M(233737, J$.T(233657, [
                                                        J$.T(233617, 'close', 21, false),
                                                        J$.T(233625, 'error', 21, false),
                                                        J$.T(233633, 'timeout', 21, false),
                                                        J$.T(233641, 'parseError', 21, false),
                                                        J$.T(233649, 'connect', 21, false)
                                                    ], 10, false), 'forEach', false)(J$.T(233729, function (event) {
                                                        jalangiLabel16:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(233705, arguments.callee, this, arguments);
                                                                    arguments = J$.N(233713, 'arguments', arguments, true, false, false);
                                                                    event = J$.N(233721, 'event', event, true, false, false);
                                                                    J$.M(233697, J$.G(233681, J$.G(233673, J$.R(233665, '_this2', _this2, false, false), 's'), 'pool'), 'removeAllListeners', false)(J$.R(233689, 'event', event, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(239889, J$e);
                                                                } finally {
                                                                    if (J$.Fr(239897))
                                                                        continue jalangiLabel16;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                    if (J$.C(13456, J$.G(233761, J$.G(233753, J$.R(233745, 'this', this, false, false), 's'), 'monitorId'))) {
                                                        J$.F(233801, J$.I(typeof clearTimeout === 'undefined' ? clearTimeout = J$.R(233769, 'clearTimeout', undefined, true, true) : clearTimeout = J$.R(233769, 'clearTimeout', clearTimeout, true, true)), false)(J$.G(233793, J$.G(233785, J$.R(233777, 'this', this, false, false), 's'), 'monitorId'));
                                                    }
                                                    J$.M(233857, J$.G(233825, J$.G(233817, J$.R(233809, 'this', this, false, false), 's'), 'pool'), 'destroy', false)(J$.G(233841, J$.R(233833, 'options', options, false, false), 'force'), J$.R(233849, 'done', done, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(239905, J$e);
                                                } finally {
                                                    if (J$.Fr(239913))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(234161, {
                                    key: J$.T(233929, 'monitor', 21, false),
                                    value: J$.T(234153, function monitor(options) {
                                        jalangiLabel18:
                                            while (true) {
                                                try {
                                                    J$.Fe(234129, arguments.callee, this, arguments);
                                                    arguments = J$.N(234137, 'arguments', arguments, true, false, false);
                                                    options = J$.N(234145, 'options', options, true, false, false);
                                                    options = J$.W(233953, 'options', J$.C(13464, J$.R(233937, 'options', options, false, false)) ? J$._() : J$.T(233945, {}, 11, false), options, false, false);
                                                    if (J$.C(13480, J$.C(13472, J$.B(15170, '!==', J$.G(233977, J$.G(233969, J$.R(233961, 'this', this, false, false), 's'), 'state'), J$.R(233985, 'STATE_CONNECTED', STATE_CONNECTED, false, true))) ? J$._() : J$.G(234009, J$.G(234001, J$.R(233993, 'this', this, false, false), 's'), 'monitoring')))
                                                        return J$.Rt(234017, undefined);
                                                    if (J$.C(13488, J$.G(234041, J$.G(234033, J$.R(234025, 'this', this, false, false), 's'), 'monitorId')))
                                                        J$.F(234081, J$.I(typeof clearTimeout === 'undefined' ? clearTimeout = J$.R(234049, 'clearTimeout', undefined, true, true) : clearTimeout = J$.R(234049, 'clearTimeout', clearTimeout, true, true)), false)(J$.G(234073, J$.G(234065, J$.R(234057, 'this', this, false, false), 's'), 'monitorId'));
                                                    J$.M(234121, J$.G(234097, J$.R(234089, 'this', this, false, false), 's'), 'monitorFunction', false)(J$.R(234105, 'this', this, false, false), J$.R(234113, 'options', options, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(239921, J$e);
                                                } finally {
                                                    if (J$.Fr(239929))
                                                        continue jalangiLabel18;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(235041, {
                                    key: J$.T(234169, 'command', 21, false),
                                    value: J$.T(235033, function command(ns, cmd, options, callback) {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(234969, arguments.callee, this, arguments);
                                                    arguments = J$.N(234977, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(234985, 'ns', ns, true, false, false);
                                                    cmd = J$.N(234993, 'cmd', cmd, true, false, false);
                                                    options = J$.N(235001, 'options', options, true, false, false);
                                                    callback = J$.N(235009, 'callback', callback, true, false, false);
                                                    J$.N(235017, '_this3', _this3, false, false, false);
                                                    J$.N(235025, 'error', error, false, false, false);
                                                    var _this3 = J$.W(234185, '_this3', J$.R(234177, 'this', this, false, false), _this3, false, false);
                                                    if (J$.C(13504, J$.B(15186, '===', J$.U(15178, 'typeof', J$.R(234193, 'options', options, false, false)), J$.T(234201, 'function', 21, false)))) {
                                                        callback = J$.W(234217, 'callback', J$.R(234209, 'options', options, false, false), callback, false, false), options = J$.W(234233, 'options', J$.T(234225, {}, 11, false), options, false, false), options = J$.W(234257, 'options', J$.C(13496, J$.R(234241, 'options', options, false, false)) ? J$._() : J$.T(234249, {}, 11, false), options, false, false);
                                                    }
                                                    var error = J$.W(234297, 'error', J$.F(234289, J$.R(234265, 'basicReadValidations', basicReadValidations, false, true), false)(J$.R(234273, 'this', this, false, false), J$.R(234281, 'options', options, false, false)), error, false, false);
                                                    if (J$.C(13512, J$.R(234305, 'error', error, false, false))) {
                                                        return J$.Rt(234345, J$.F(234337, J$.R(234313, 'callback', callback, false, false), false)(J$.R(234321, 'error', error, false, false), J$.T(234329, null, 25, false)));
                                                    }
                                                    options = J$.W(234401, 'options', J$.M(234393, J$.I(typeof Object === 'undefined' ? Object = J$.R(234353, 'Object', undefined, true, true) : Object = J$.R(234353, 'Object', Object, true, true)), 'assign', false)(J$.T(234361, {}, 11, false), J$.R(234369, 'options', options, false, false), J$.T(234385, { wireProtocolCommand: J$.T(234377, false, 23, false) }, 11, false)), options, false, false);
                                                    if (J$.C(13520, J$.M(234433, J$.G(234425, J$.G(234417, J$.R(234409, 'this', this, false, false), 's'), 'logger'), 'isDebug', false)())) {
                                                        J$.M(234585, J$.G(234457, J$.G(234449, J$.R(234441, 'this', this, false, false), 's'), 'logger'), 'debug', false)(J$.M(234577, J$.M(234553, J$.T(234465, 'executing command [', 21, false), 'concat', false)(J$.M(234537, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(234473, 'JSON', undefined, true, true) : JSON = J$.R(234473, 'JSON', JSON, true, true)), 'stringify', false)(J$.T(234529, {
                                                            ns: J$.R(234481, 'ns', ns, false, false),
                                                            cmd: J$.R(234489, 'cmd', cmd, false, false),
                                                            options: J$.F(234521, J$.R(234497, 'debugOptions', debugOptions, false, true), false)(J$.R(234505, 'DEBUG_FIELDS', DEBUG_FIELDS, false, true), J$.R(234513, 'options', options, false, false))
                                                        }, 11, false)), J$.T(234545, '] against ', 21, false)), 'concat', false)(J$.G(234569, J$.R(234561, 'this', this, false, false), 'name')));
                                                    }
                                                    if (J$.C(13528, J$.F(234617, J$.R(234593, 'collationNotSupported', collationNotSupported, false, true), false)(J$.R(234601, 'this', this, false, false), J$.R(234609, 'cmd', cmd, false, false)))) {
                                                        J$.F(234689, J$.R(234625, 'callback', callback, false, false), false)(J$.F(234681, J$.R(234633, 'MongoError', MongoError, false, true), true)(J$.M(234673, J$.T(234641, 'server ', 21, false), 'concat', false)(J$.G(234657, J$.R(234649, 'this', this, false, false), 'name'), J$.T(234665, ' does not support collation', 21, false))));
                                                        return J$.Rt(234697, undefined);
                                                    }
                                                    J$.M(234961, J$.R(234705, 'wireProtocol', wireProtocol, false, true), 'command', false)(J$.R(234713, 'this', this, false, false), J$.R(234721, 'ns', ns, false, false), J$.R(234729, 'cmd', cmd, false, false), J$.R(234737, 'options', options, false, false), J$.T(234953, function (err, result) {
                                                        jalangiLabel19:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(234921, arguments.callee, this, arguments);
                                                                    arguments = J$.N(234929, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(234937, 'err', err, true, false, false);
                                                                    result = J$.N(234945, 'result', result, true, false, false);
                                                                    if (J$.C(13560, J$.R(234745, 'err', err, false, false))) {
                                                                        if (J$.C(13544, J$.C(13536, J$.G(234761, J$.R(234753, 'options', options, false, false), 'session')) ? J$.B(15194, 'instanceof', J$.R(234769, 'err', err, false, false), J$.R(234777, 'MongoNetworkError', MongoNetworkError, false, true)) : J$._())) {
                                                                            J$.P(234817, J$.G(234801, J$.G(234793, J$.R(234785, 'options', options, false, false), 'session'), 'serverSession'), 'isDirty', J$.T(234809, true, 23, false));
                                                                        }
                                                                        if (J$.C(13552, J$.F(234849, J$.R(234825, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true), false)(J$.R(234833, 'err', err, false, false), J$.R(234841, '_this3', _this3, false, false)))) {
                                                                            J$.M(234881, J$.R(234857, '_this3', _this3, false, false), 'emit', false)(J$.T(234865, 'error', 21, false), J$.R(234873, 'err', err, false, false));
                                                                        }
                                                                    }
                                                                    J$.F(234913, J$.R(234889, 'callback', callback, false, false), false)(J$.R(234897, 'err', err, false, false), J$.R(234905, 'result', result, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(239937, J$e);
                                                                } finally {
                                                                    if (J$.Fr(239945))
                                                                        continue jalangiLabel19;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(239953, J$e);
                                                } finally {
                                                    if (J$.Fr(239961))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(235417, {
                                    key: J$.T(235049, 'query', 21, false),
                                    value: J$.T(235409, function query(ns, cmd, cursorState, options, callback) {
                                        jalangiLabel22:
                                            while (true) {
                                                try {
                                                    J$.Fe(235345, arguments.callee, this, arguments);
                                                    arguments = J$.N(235353, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(235361, 'ns', ns, true, false, false);
                                                    cmd = J$.N(235369, 'cmd', cmd, true, false, false);
                                                    cursorState = J$.N(235377, 'cursorState', cursorState, true, false, false);
                                                    options = J$.N(235385, 'options', options, true, false, false);
                                                    callback = J$.N(235393, 'callback', callback, true, false, false);
                                                    J$.N(235401, '_this4', _this4, false, false, false);
                                                    var _this4 = J$.W(235065, '_this4', J$.R(235057, 'this', this, false, false), _this4, false, false);
                                                    J$.M(235337, J$.R(235073, 'wireProtocol', wireProtocol, false, true), 'query', false)(J$.R(235081, 'this', this, false, false), J$.R(235089, 'ns', ns, false, false), J$.R(235097, 'cmd', cmd, false, false), J$.R(235105, 'cursorState', cursorState, false, false), J$.R(235113, 'options', options, false, false), J$.T(235329, function (err, result) {
                                                        jalangiLabel21:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(235297, arguments.callee, this, arguments);
                                                                    arguments = J$.N(235305, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(235313, 'err', err, true, false, false);
                                                                    result = J$.N(235321, 'result', result, true, false, false);
                                                                    if (J$.C(13592, J$.R(235121, 'err', err, false, false))) {
                                                                        if (J$.C(13576, J$.C(13568, J$.G(235137, J$.R(235129, 'options', options, false, false), 'session')) ? J$.B(15202, 'instanceof', J$.R(235145, 'err', err, false, false), J$.R(235153, 'MongoNetworkError', MongoNetworkError, false, true)) : J$._())) {
                                                                            J$.P(235193, J$.G(235177, J$.G(235169, J$.R(235161, 'options', options, false, false), 'session'), 'serverSession'), 'isDirty', J$.T(235185, true, 23, false));
                                                                        }
                                                                        if (J$.C(13584, J$.F(235225, J$.R(235201, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true), false)(J$.R(235209, 'err', err, false, false), J$.R(235217, '_this4', _this4, false, false)))) {
                                                                            J$.M(235257, J$.R(235233, '_this4', _this4, false, false), 'emit', false)(J$.T(235241, 'error', 21, false), J$.R(235249, 'err', err, false, false));
                                                                        }
                                                                    }
                                                                    J$.F(235289, J$.R(235265, 'callback', callback, false, false), false)(J$.R(235273, 'err', err, false, false), J$.R(235281, 'result', result, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(239969, J$e);
                                                                } finally {
                                                                    if (J$.Fr(239977))
                                                                        continue jalangiLabel21;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(239985, J$e);
                                                } finally {
                                                    if (J$.Fr(239993))
                                                        continue jalangiLabel22;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(235793, {
                                    key: J$.T(235425, 'getMore', 21, false),
                                    value: J$.T(235785, function getMore(ns, cursorState, batchSize, options, callback) {
                                        jalangiLabel24:
                                            while (true) {
                                                try {
                                                    J$.Fe(235721, arguments.callee, this, arguments);
                                                    arguments = J$.N(235729, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(235737, 'ns', ns, true, false, false);
                                                    cursorState = J$.N(235745, 'cursorState', cursorState, true, false, false);
                                                    batchSize = J$.N(235753, 'batchSize', batchSize, true, false, false);
                                                    options = J$.N(235761, 'options', options, true, false, false);
                                                    callback = J$.N(235769, 'callback', callback, true, false, false);
                                                    J$.N(235777, '_this5', _this5, false, false, false);
                                                    var _this5 = J$.W(235441, '_this5', J$.R(235433, 'this', this, false, false), _this5, false, false);
                                                    J$.M(235713, J$.R(235449, 'wireProtocol', wireProtocol, false, true), 'getMore', false)(J$.R(235457, 'this', this, false, false), J$.R(235465, 'ns', ns, false, false), J$.R(235473, 'cursorState', cursorState, false, false), J$.R(235481, 'batchSize', batchSize, false, false), J$.R(235489, 'options', options, false, false), J$.T(235705, function (err, result) {
                                                        jalangiLabel23:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(235673, arguments.callee, this, arguments);
                                                                    arguments = J$.N(235681, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(235689, 'err', err, true, false, false);
                                                                    result = J$.N(235697, 'result', result, true, false, false);
                                                                    if (J$.C(13624, J$.R(235497, 'err', err, false, false))) {
                                                                        if (J$.C(13608, J$.C(13600, J$.G(235513, J$.R(235505, 'options', options, false, false), 'session')) ? J$.B(15210, 'instanceof', J$.R(235521, 'err', err, false, false), J$.R(235529, 'MongoNetworkError', MongoNetworkError, false, true)) : J$._())) {
                                                                            J$.P(235569, J$.G(235553, J$.G(235545, J$.R(235537, 'options', options, false, false), 'session'), 'serverSession'), 'isDirty', J$.T(235561, true, 23, false));
                                                                        }
                                                                        if (J$.C(13616, J$.F(235601, J$.R(235577, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true), false)(J$.R(235585, 'err', err, false, false), J$.R(235593, '_this5', _this5, false, false)))) {
                                                                            J$.M(235633, J$.R(235609, '_this5', _this5, false, false), 'emit', false)(J$.T(235617, 'error', 21, false), J$.R(235625, 'err', err, false, false));
                                                                        }
                                                                    }
                                                                    J$.F(235665, J$.R(235641, 'callback', callback, false, false), false)(J$.R(235649, 'err', err, false, false), J$.R(235657, 'result', result, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(240001, J$e);
                                                                } finally {
                                                                    if (J$.Fr(240009))
                                                                        continue jalangiLabel23;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(240017, J$e);
                                                } finally {
                                                    if (J$.Fr(240025))
                                                        continue jalangiLabel24;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(236081, {
                                    key: J$.T(235801, 'killCursors', 21, false),
                                    value: J$.T(236073, function killCursors(ns, cursorState, callback) {
                                        jalangiLabel26:
                                            while (true) {
                                                try {
                                                    J$.Fe(236025, arguments.callee, this, arguments);
                                                    arguments = J$.N(236033, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(236041, 'ns', ns, true, false, false);
                                                    cursorState = J$.N(236049, 'cursorState', cursorState, true, false, false);
                                                    callback = J$.N(236057, 'callback', callback, true, false, false);
                                                    J$.N(236065, '_this6', _this6, false, false, false);
                                                    var _this6 = J$.W(235817, '_this6', J$.R(235809, 'this', this, false, false), _this6, false, false);
                                                    J$.M(236017, J$.R(235825, 'wireProtocol', wireProtocol, false, true), 'killCursors', false)(J$.R(235833, 'this', this, false, false), J$.R(235841, 'ns', ns, false, false), J$.R(235849, 'cursorState', cursorState, false, false), J$.T(236009, function (err, result) {
                                                        jalangiLabel25:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(235977, arguments.callee, this, arguments);
                                                                    arguments = J$.N(235985, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(235993, 'err', err, true, false, false);
                                                                    result = J$.N(236001, 'result', result, true, false, false);
                                                                    if (J$.C(13640, J$.C(13632, J$.R(235857, 'err', err, false, false)) ? J$.F(235889, J$.R(235865, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true), false)(J$.R(235873, 'err', err, false, false), J$.R(235881, '_this6', _this6, false, false)) : J$._())) {
                                                                        J$.M(235921, J$.R(235897, '_this6', _this6, false, false), 'emit', false)(J$.T(235905, 'error', 21, false), J$.R(235913, 'err', err, false, false));
                                                                    }
                                                                    if (J$.C(13648, J$.B(15226, '===', J$.U(15218, 'typeof', J$.R(235929, 'callback', callback, false, false)), J$.T(235937, 'function', 21, false)))) {
                                                                        J$.F(235969, J$.R(235945, 'callback', callback, false, false), false)(J$.R(235953, 'err', err, false, false), J$.R(235961, 'result', result, false, false));
                                                                    }
                                                                } catch (J$e) {
                                                                    J$.Ex(240033, J$e);
                                                                } finally {
                                                                    if (J$.Fr(240041))
                                                                        continue jalangiLabel25;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(240049, J$e);
                                                } finally {
                                                    if (J$.Fr(240057))
                                                        continue jalangiLabel26;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(236225, {
                                    key: J$.T(236089, 'insert', 21, false),
                                    value: J$.T(236217, function insert(ns, ops, options, callback) {
                                        jalangiLabel27:
                                            while (true) {
                                                try {
                                                    J$.Fe(236169, arguments.callee, this, arguments);
                                                    arguments = J$.N(236177, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(236185, 'ns', ns, true, false, false);
                                                    ops = J$.N(236193, 'ops', ops, true, false, false);
                                                    options = J$.N(236201, 'options', options, true, false, false);
                                                    callback = J$.N(236209, 'callback', callback, true, false, false);
                                                    J$.F(236161, J$.R(236097, 'executeWriteOperation', executeWriteOperation, false, true), false)(J$.T(236137, {
                                                        server: J$.R(236105, 'this', this, false, false),
                                                        op: J$.T(236113, 'insert', 21, false),
                                                        ns: J$.R(236121, 'ns', ns, false, false),
                                                        ops: J$.R(236129, 'ops', ops, false, false)
                                                    }, 11, false), J$.R(236145, 'options', options, false, false), J$.R(236153, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(240065, J$e);
                                                } finally {
                                                    if (J$.Fr(240073))
                                                        continue jalangiLabel27;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(236369, {
                                    key: J$.T(236233, 'update', 21, false),
                                    value: J$.T(236361, function update(ns, ops, options, callback) {
                                        jalangiLabel28:
                                            while (true) {
                                                try {
                                                    J$.Fe(236313, arguments.callee, this, arguments);
                                                    arguments = J$.N(236321, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(236329, 'ns', ns, true, false, false);
                                                    ops = J$.N(236337, 'ops', ops, true, false, false);
                                                    options = J$.N(236345, 'options', options, true, false, false);
                                                    callback = J$.N(236353, 'callback', callback, true, false, false);
                                                    J$.F(236305, J$.R(236241, 'executeWriteOperation', executeWriteOperation, false, true), false)(J$.T(236281, {
                                                        server: J$.R(236249, 'this', this, false, false),
                                                        op: J$.T(236257, 'update', 21, false),
                                                        ns: J$.R(236265, 'ns', ns, false, false),
                                                        ops: J$.R(236273, 'ops', ops, false, false)
                                                    }, 11, false), J$.R(236289, 'options', options, false, false), J$.R(236297, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(240081, J$e);
                                                } finally {
                                                    if (J$.Fr(240089))
                                                        continue jalangiLabel28;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(236513, {
                                    key: J$.T(236377, 'remove', 21, false),
                                    value: J$.T(236505, function remove(ns, ops, options, callback) {
                                        jalangiLabel29:
                                            while (true) {
                                                try {
                                                    J$.Fe(236457, arguments.callee, this, arguments);
                                                    arguments = J$.N(236465, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(236473, 'ns', ns, true, false, false);
                                                    ops = J$.N(236481, 'ops', ops, true, false, false);
                                                    options = J$.N(236489, 'options', options, true, false, false);
                                                    callback = J$.N(236497, 'callback', callback, true, false, false);
                                                    J$.F(236449, J$.R(236385, 'executeWriteOperation', executeWriteOperation, false, true), false)(J$.T(236425, {
                                                        server: J$.R(236393, 'this', this, false, false),
                                                        op: J$.T(236401, 'remove', 21, false),
                                                        ns: J$.R(236409, 'ns', ns, false, false),
                                                        ops: J$.R(236417, 'ops', ops, false, false)
                                                    }, 11, false), J$.R(236433, 'options', options, false, false), J$.R(236441, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(240097, J$e);
                                                } finally {
                                                    if (J$.Fr(240105))
                                                        continue jalangiLabel29;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(236585, {
                                    key: J$.T(236521, 'description', 21, false),
                                    get: J$.T(236577, function get() {
                                        jalangiLabel30:
                                            while (true) {
                                                try {
                                                    J$.Fe(236561, arguments.callee, this, arguments);
                                                    arguments = J$.N(236569, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(236553, J$.G(236545, J$.G(236537, J$.R(236529, 'this', this, false, false), 's'), 'description'));
                                                } catch (J$e) {
                                                    J$.Ex(240113, J$e);
                                                } finally {
                                                    if (J$.Fr(240121))
                                                        continue jalangiLabel30;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(236665, {
                                    key: J$.T(236593, 'name', 21, false),
                                    get: J$.T(236657, function get() {
                                        jalangiLabel31:
                                            while (true) {
                                                try {
                                                    J$.Fe(236641, arguments.callee, this, arguments);
                                                    arguments = J$.N(236649, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(236633, J$.G(236625, J$.G(236617, J$.G(236609, J$.R(236601, 'this', this, false, false), 's'), 'description'), 'address'));
                                                } catch (J$e) {
                                                    J$.Ex(240129, J$e);
                                                } finally {
                                                    if (J$.Fr(240137))
                                                        continue jalangiLabel31;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(236817, {
                                    key: J$.T(236673, 'autoEncrypter', 21, false),
                                    get: J$.T(236809, function get() {
                                        jalangiLabel32:
                                            while (true) {
                                                try {
                                                    J$.Fe(236793, arguments.callee, this, arguments);
                                                    arguments = J$.N(236801, 'arguments', arguments, true, false, false);
                                                    if (J$.C(13664, J$.C(13656, J$.G(236697, J$.G(236689, J$.R(236681, 'this', this, false, false), 's'), 'options')) ? J$.G(236729, J$.G(236721, J$.G(236713, J$.R(236705, 'this', this, false, false), 's'), 'options'), 'autoEncrypter') : J$._())) {
                                                        return J$.Rt(236769, J$.G(236761, J$.G(236753, J$.G(236745, J$.R(236737, 'this', this, false, false), 's'), 'options'), 'autoEncrypter'));
                                                    }
                                                    return J$.Rt(236785, J$.T(236777, null, 25, false));
                                                } catch (J$e) {
                                                    J$.Ex(240145, J$e);
                                                } finally {
                                                    if (J$.Fr(240153))
                                                        continue jalangiLabel32;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(236849, J$.R(236841, 'Server', Server, false, false));
                        } catch (J$e) {
                            J$.Ex(240161, J$e);
                        } finally {
                            if (J$.Fr(240169))
                                continue jalangiLabel33;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(236905, 'EventEmitter', EventEmitter, false, true)), Server, false, true);
            J$.M(237105, J$.I(typeof Object === 'undefined' ? Object = J$.R(236929, 'Object', undefined, true, true) : Object = J$.R(236929, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(236945, J$.R(236937, 'Server', Server, false, true), 'prototype'), J$.T(236953, 'clusterTime', 21, false), J$.T(237097, {
                get: J$.T(237017, function get() {
                    jalangiLabel34:
                        while (true) {
                            try {
                                J$.Fe(237001, arguments.callee, this, arguments);
                                arguments = J$.N(237009, 'arguments', arguments, true, false, false);
                                return J$.Rt(236993, J$.G(236985, J$.G(236977, J$.G(236969, J$.R(236961, 'this', this, false, false), 's'), 'topology'), 'clusterTime'));
                            } catch (J$e) {
                                J$.Ex(240177, J$e);
                            } finally {
                                if (J$.Fr(240185))
                                    continue jalangiLabel34;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                set: J$.T(237089, function set(clusterTime) {
                    jalangiLabel35:
                        while (true) {
                            try {
                                J$.Fe(237065, arguments.callee, this, arguments);
                                arguments = J$.N(237073, 'arguments', arguments, true, false, false);
                                clusterTime = J$.N(237081, 'clusterTime', clusterTime, true, false, false);
                                J$.P(237057, J$.G(237041, J$.G(237033, J$.R(237025, 'this', this, false, false), 's'), 'topology'), 'clusterTime', J$.R(237049, 'clusterTime', clusterTime, false, false));
                            } catch (J$e) {
                                J$.Ex(240193, J$e);
                            } finally {
                                if (J$.Fr(240201))
                                    continue jalangiLabel35;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.P(239201, J$.I(typeof module === 'undefined' ? module = J$.R(239185, 'module', undefined, true, true) : module = J$.R(239185, 'module', module, true, true)), 'exports', J$.R(239193, 'Server', Server, false, true));
        } catch (J$e) {
            J$.Ex(240369, J$e);
        } finally {
            if (J$.Sr(240377))
                continue jalangiLabel46;
            else
                break jalangiLabel46;
        }
    }
// JALANGI DO NOT INSTRUMENT
