// mongodb/lib/core/sdam/srv_polling.js
J$.noInstrEval = false;
jalangiLabel32:
    while (true) {
        try {
            J$.Se(253161, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/core/sdam/srv_polling.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(249025, arguments.callee, this, arguments);
                            arguments = J$.N(249033, 'arguments', arguments, true, false, false);
                            obj = J$.N(249041, 'obj', obj, true, false, false);
                            if (J$.C(14392, J$.C(14352, J$.B(15890, '===', J$.U(15882, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(248761, 'Symbol', undefined, true, true) : Symbol = J$.R(248761, 'Symbol', Symbol, true, true))), J$.T(248769, 'function', 21, false))) ? J$.B(15906, '===', J$.U(15898, 'typeof', J$.G(248785, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(248777, 'Symbol', undefined, true, true) : Symbol = J$.R(248777, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(248793, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(248849, '_typeof', J$.T(248841, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(248817, arguments.callee, this, arguments);
                                                arguments = J$.N(248825, 'arguments', arguments, true, false, false);
                                                obj = J$.N(248833, 'obj', obj, true, false, false);
                                                return J$.Rt(248809, J$.U(15914, 'typeof', J$.R(248801, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(253369, J$e);
                                            } finally {
                                                if (J$.Fr(253377))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(248985, '_typeof', J$.T(248977, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(248953, arguments.callee, this, arguments);
                                                arguments = J$.N(248961, 'arguments', arguments, true, false, false);
                                                obj = J$.N(248969, 'obj', obj, true, false, false);
                                                return J$.Rt(248945, J$.C(14384, J$.C(14376, J$.C(14368, J$.C(14360, J$.R(248857, 'obj', obj, false, false)) ? J$.B(15930, '===', J$.U(15922, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(248865, 'Symbol', undefined, true, true) : Symbol = J$.R(248865, 'Symbol', Symbol, true, true))), J$.T(248873, 'function', 21, false)) : J$._()) ? J$.B(15938, '===', J$.G(248889, J$.R(248881, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(248897, 'Symbol', undefined, true, true) : Symbol = J$.R(248897, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(15946, '!==', J$.R(248905, 'obj', obj, false, false), J$.G(248921, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(248913, 'Symbol', undefined, true, true) : Symbol = J$.R(248913, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(248929, 'symbol', 21, false) : J$.U(15954, 'typeof', J$.R(248937, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(253385, J$e);
                                            } finally {
                                                if (J$.Fr(253393))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(249017, J$.F(249009, J$.R(248993, '_typeof', _typeof, false, true), false)(J$.R(249001, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(253401, J$e);
                        } finally {
                            if (J$.Fr(253409))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(249153, arguments.callee, this, arguments);
                            arguments = J$.N(249161, 'arguments', arguments, true, false, false);
                            self = J$.N(249169, 'self', self, true, false, false);
                            call = J$.N(249177, 'call', call, true, false, false);
                            if (J$.C(14416, J$.C(14408, J$.R(249049, 'call', call, false, false)) ? J$.C(14400, J$.B(15962, '===', J$.F(249073, J$.R(249057, '_typeof', _typeof, false, true), false)(J$.R(249065, 'call', call, false, false)), J$.T(249081, 'object', 21, false))) ? J$._() : J$.B(15978, '===', J$.U(15970, 'typeof', J$.R(249089, 'call', call, false, false)), J$.T(249097, 'function', 21, false)) : J$._())) {
                                return J$.Rt(249113, J$.R(249105, 'call', call, false, false));
                            }
                            return J$.Rt(249145, J$.F(249137, J$.R(249121, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(249129, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(253417, J$e);
                        } finally {
                            if (J$.Fr(253425))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(249241, arguments.callee, this, arguments);
                            arguments = J$.N(249249, 'arguments', arguments, true, false, false);
                            self = J$.N(249257, 'self', self, true, false, false);
                            if (J$.C(14424, J$.B(15986, '===', J$.R(249185, 'self', self, false, false), void J$.T(249193, 0, 22, false)))) {
                                throw J$.F(249217, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(249201, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(249201, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(249209, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(249233, J$.R(249225, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(253433, J$e);
                        } finally {
                            if (J$.Fr(253441))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(249417, arguments.callee, this, arguments);
                            arguments = J$.N(249425, 'arguments', arguments, true, false, false);
                            o = J$.N(249433, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(249377, '_getPrototypeOf', J$.C(14440, J$.G(249273, J$.I(typeof Object === 'undefined' ? Object = J$.R(249265, 'Object', undefined, true, true) : Object = J$.R(249265, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(249289, J$.I(typeof Object === 'undefined' ? Object = J$.R(249281, 'Object', undefined, true, true) : Object = J$.R(249281, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(249369, function _getPrototypeOf(o) {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(249345, arguments.callee, this, arguments);
                                            arguments = J$.N(249353, 'arguments', arguments, true, false, false);
                                            o = J$.N(249361, 'o', o, true, false, false);
                                            return J$.Rt(249337, J$.C(14432, J$.G(249305, J$.R(249297, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(249329, J$.I(typeof Object === 'undefined' ? Object = J$.R(249313, 'Object', undefined, true, true) : Object = J$.R(249313, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(249321, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(253449, J$e);
                                        } finally {
                                            if (J$.Fr(253457))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(249409, J$.F(249401, J$.R(249385, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(249393, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(253465, J$e);
                        } finally {
                            if (J$.Fr(253473))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(249633, arguments.callee, this, arguments);
                            arguments = J$.N(249641, 'arguments', arguments, true, false, false);
                            subClass = J$.N(249649, 'subClass', subClass, true, false, false);
                            superClass = J$.N(249657, 'superClass', superClass, true, false, false);
                            if (J$.C(14456, J$.C(14448, J$.B(16002, '!==', J$.U(15994, 'typeof', J$.R(249441, 'superClass', superClass, false, false)), J$.T(249449, 'function', 21, false))) ? J$.B(16010, '!==', J$.R(249457, 'superClass', superClass, false, false), J$.T(249465, null, 25, false)) : J$._())) {
                                throw J$.F(249489, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(249473, 'TypeError', undefined, true, true) : TypeError = J$.R(249473, 'TypeError', TypeError, true, true)), true)(J$.T(249481, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(249585, J$.R(249497, 'subClass', subClass, false, false), 'prototype', J$.M(249577, J$.I(typeof Object === 'undefined' ? Object = J$.R(249505, 'Object', undefined, true, true) : Object = J$.R(249505, 'Object', Object, true, true)), 'create', false)(J$.C(14464, J$.R(249513, 'superClass', superClass, false, false)) ? J$.G(249529, J$.R(249521, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(249569, {
                                constructor: J$.T(249561, {
                                    value: J$.R(249537, 'subClass', subClass, false, false),
                                    writable: J$.T(249545, true, 23, false),
                                    configurable: J$.T(249553, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(14472, J$.R(249593, 'superClass', superClass, false, false)))
                                J$.F(249625, J$.R(249601, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(249609, 'subClass', subClass, false, false), J$.R(249617, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(253481, J$e);
                        } finally {
                            if (J$.Fr(253489))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(249809, arguments.callee, this, arguments);
                            arguments = J$.N(249817, 'arguments', arguments, true, false, false);
                            o = J$.N(249825, 'o', o, true, false, false);
                            p = J$.N(249833, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(249761, '_setPrototypeOf', J$.C(14480, J$.G(249673, J$.I(typeof Object === 'undefined' ? Object = J$.R(249665, 'Object', undefined, true, true) : Object = J$.R(249665, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(249753, function _setPrototypeOf(o, p) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(249721, arguments.callee, this, arguments);
                                            arguments = J$.N(249729, 'arguments', arguments, true, false, false);
                                            o = J$.N(249737, 'o', o, true, false, false);
                                            p = J$.N(249745, 'p', p, true, false, false);
                                            J$.P(249697, J$.R(249681, 'o', o, false, false), '__proto__', J$.R(249689, 'p', p, false, false));
                                            return J$.Rt(249713, J$.R(249705, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(253497, J$e);
                                        } finally {
                                            if (J$.Fr(253505))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(249801, J$.F(249793, J$.R(249769, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(249777, 'o', o, false, false), J$.R(249785, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(253513, J$e);
                        } finally {
                            if (J$.Fr(253521))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(249881, arguments.callee, this, arguments);
                            arguments = J$.N(249889, 'arguments', arguments, true, false, false);
                            instance = J$.N(249897, 'instance', instance, true, false, false);
                            Constructor = J$.N(249905, 'Constructor', Constructor, true, false, false);
                            if (J$.C(14488, J$.U(16026, '!', J$.B(16018, 'instanceof', J$.R(249841, 'instance', instance, false, false), J$.R(249849, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(249873, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(249857, 'TypeError', undefined, true, true) : TypeError = J$.R(249857, 'TypeError', TypeError, true, true)), true)(J$.T(249865, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(253529, J$e);
                        } finally {
                            if (J$.Fr(253537))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(250153, arguments.callee, this, arguments);
                            arguments = J$.N(250161, 'arguments', arguments, true, false, false);
                            target = J$.N(250169, 'target', target, true, false, false);
                            props = J$.N(250177, 'props', props, true, false, false);
                            J$.N(250185, 'i', i, false, false, false);
                            J$.N(250193, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(249921, 'i', J$.T(249913, 0, 22, false), i, false, false); J$.C(14512, J$.B(16034, '<', J$.R(249929, 'i', i, false, false), J$.G(249945, J$.R(249937, 'props', props, false, false), 'length'))); J$.B(16058, '-', i = J$.W(249961, 'i', J$.B(16050, '+', J$.U(16042, '+', J$.R(249953, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(249993, 'descriptor', J$.G(249985, J$.R(249969, 'props', props, false, false), J$.R(249977, 'i', i, false, false)), descriptor, false, false);
                                J$.P(250033, J$.R(250001, 'descriptor', descriptor, false, false), 'enumerable', J$.C(14496, J$.G(250017, J$.R(250009, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(250025, false, 23, false));
                                J$.P(250057, J$.R(250041, 'descriptor', descriptor, false, false), 'configurable', J$.T(250049, true, 23, false));
                                if (J$.C(14504, J$.B(16066, 'in', J$.T(250065, 'value', 21, false), J$.R(250073, 'descriptor', descriptor, false, false))))
                                    J$.P(250097, J$.R(250081, 'descriptor', descriptor, false, false), 'writable', J$.T(250089, true, 23, false));
                                J$.M(250145, J$.I(typeof Object === 'undefined' ? Object = J$.R(250105, 'Object', undefined, true, true) : Object = J$.R(250105, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(250113, 'target', target, false, false), J$.G(250129, J$.R(250121, 'descriptor', descriptor, false, false), 'key'), J$.R(250137, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(253545, J$e);
                        } finally {
                            if (J$.Fr(253553))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(250305, arguments.callee, this, arguments);
                            arguments = J$.N(250313, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(250321, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(250329, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(250337, 'staticProps', staticProps, true, false, false);
                            if (J$.C(14520, J$.R(250201, 'protoProps', protoProps, false, false)))
                                J$.F(250241, J$.R(250209, '_defineProperties', _defineProperties, false, true), false)(J$.G(250225, J$.R(250217, 'Constructor', Constructor, false, false), 'prototype'), J$.R(250233, 'protoProps', protoProps, false, false));
                            if (J$.C(14528, J$.R(250249, 'staticProps', staticProps, false, false)))
                                J$.F(250281, J$.R(250257, '_defineProperties', _defineProperties, false, true), false)(J$.R(250265, 'Constructor', Constructor, false, false), J$.R(250273, 'staticProps', staticProps, false, false));
                            return J$.Rt(250297, J$.R(250289, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(253561, J$e);
                        } finally {
                            if (J$.Fr(253569))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function matchesParentDomain(srvAddress, parentDomain) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(250609, arguments.callee, this, arguments);
                            arguments = J$.N(250617, 'arguments', arguments, true, false, false);
                            srvAddress = J$.N(250625, 'srvAddress', srvAddress, true, false, false);
                            parentDomain = J$.N(250633, 'parentDomain', parentDomain, true, false, false);
                            J$.N(250641, 'regex', regex, false, false, false);
                            J$.N(250649, 'srv', srv, false, false, false);
                            J$.N(250657, 'parent', parent, false, false, false);
                            var regex = J$.W(250457, 'regex', J$.T(250449, /^.*?\./, 14, false), regex, false, false);
                            var srv = J$.W(250513, 'srv', J$.M(250505, J$.T(250465, '.', 21, false), 'concat', false)(J$.M(250497, J$.R(250473, 'srvAddress', srvAddress, false, false), 'replace', false)(J$.R(250481, 'regex', regex, false, false), J$.T(250489, '', 21, false))), srv, false, false);
                            var parent = J$.W(250569, 'parent', J$.M(250561, J$.T(250521, '.', 21, false), 'concat', false)(J$.M(250553, J$.R(250529, 'parentDomain', parentDomain, false, false), 'replace', false)(J$.R(250537, 'regex', regex, false, false), J$.T(250545, '', 21, false))), parent, false, false);
                            return J$.Rt(250601, J$.M(250593, J$.R(250577, 'srv', srv, false, false), 'endsWith', false)(J$.R(250585, 'parent', parent, false, false)));
                        } catch (J$e) {
                            J$.Ex(253577, J$e);
                        } finally {
                            if (J$.Fr(253585))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(253177, '_typeof', J$.T(253169, _typeof, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(253193, '_possibleConstructorReturn', J$.T(253185, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(253209, '_assertThisInitialized', J$.T(253201, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(253225, '_getPrototypeOf', J$.T(253217, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(253241, '_inherits', J$.T(253233, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(253257, '_setPrototypeOf', J$.T(253249, _setPrototypeOf, 12, false), true, false, false);
            _classCallCheck = J$.N(253273, '_classCallCheck', J$.T(253265, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(253289, '_defineProperties', J$.T(253281, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(253305, '_createClass', J$.T(253297, _createClass, 12, false), true, false, false);
            J$.N(253313, 'Logger', Logger, false, false, false);
            J$.N(253321, 'EventEmitter', EventEmitter, false, false, false);
            J$.N(253329, 'dns', dns, false, false, false);
            matchesParentDomain = J$.N(253345, 'matchesParentDomain', J$.T(253337, matchesParentDomain, 12, false), true, false, false);
            J$.N(253353, 'SrvPollingEvent', SrvPollingEvent, false, false, false);
            J$.N(253361, 'SrvPoller', SrvPoller, false, false, false);
            J$.T(248753, 'use strict', 21, false);
            var Logger = J$.W(250369, 'Logger', J$.F(250361, J$.I(typeof require === 'undefined' ? require = J$.R(250345, 'require', undefined, true, true) : require = J$.R(250345, 'require', require, true, true)), false)(J$.T(250353, '../connection/logger', 21, false)), Logger, false, true);
            var EventEmitter = J$.W(250409, 'EventEmitter', J$.G(250401, J$.F(250393, J$.I(typeof require === 'undefined' ? require = J$.R(250377, 'require', undefined, true, true) : require = J$.R(250377, 'require', require, true, true)), false)(J$.T(250385, 'events', 21, false)), 'EventEmitter'), EventEmitter, false, true);
            var dns = J$.W(250441, 'dns', J$.F(250433, J$.I(typeof require === 'undefined' ? require = J$.R(250417, 'require', undefined, true, true) : require = J$.R(250417, 'require', require, true, true)), false)(J$.T(250425, 'dns', 21, false)), dns, false, true);
            var SrvPollingEvent = J$.W(251033, 'SrvPollingEvent', J$.F(251025, J$.T(251017, function () {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(250985, arguments.callee, this, arguments);
                            function SrvPollingEvent(srvRecords) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(250721, arguments.callee, this, arguments);
                                            arguments = J$.N(250729, 'arguments', arguments, true, false, false);
                                            srvRecords = J$.N(250737, 'srvRecords', srvRecords, true, false, false);
                                            J$.F(250689, J$.R(250665, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(250673, 'this', this, false, false), J$.R(250681, 'SrvPollingEvent', SrvPollingEvent, false, false));
                                            J$.P(250713, J$.R(250697, 'this', this, false, false), 'srvRecords', J$.R(250705, 'srvRecords', srvRecords, false, false));
                                        } catch (J$e) {
                                            J$.Ex(253593, J$e);
                                        } finally {
                                            if (J$.Fr(253601))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(250993, 'arguments', arguments, true, false, false);
                            SrvPollingEvent = J$.N(251009, 'SrvPollingEvent', J$.T(251001, SrvPollingEvent, 12, false), true, false, false);
                            J$.F(250961, J$.R(250745, '_createClass', _createClass, false, true), false)(J$.R(250753, 'SrvPollingEvent', SrvPollingEvent, false, false), J$.T(250953, [J$.T(250945, {
                                    key: J$.T(250761, 'addresses', 21, false),
                                    value: J$.T(250937, function addresses() {
                                        jalangiLabel16:
                                            while (true) {
                                                try {
                                                    J$.Fe(250921, arguments.callee, this, arguments);
                                                    arguments = J$.N(250929, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(250913, J$.F(250905, J$.I(typeof Set === 'undefined' ? Set = J$.R(250769, 'Set', undefined, true, true) : Set = J$.R(250769, 'Set', Set, true, true)), true)(J$.M(250897, J$.G(250785, J$.R(250777, 'this', this, false, false), 'srvRecords'), 'map', false)(J$.T(250889, function (record) {
                                                        jalangiLabel15:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(250865, arguments.callee, this, arguments);
                                                                    arguments = J$.N(250873, 'arguments', arguments, true, false, false);
                                                                    record = J$.N(250881, 'record', record, true, false, false);
                                                                    return J$.Rt(250857, J$.M(250849, J$.M(250825, J$.T(250793, '', 21, false), 'concat', false)(J$.G(250809, J$.R(250801, 'record', record, false, false), 'name'), J$.T(250817, ':', 21, false)), 'concat', false)(J$.G(250841, J$.R(250833, 'record', record, false, false), 'port')));
                                                                } catch (J$e) {
                                                                    J$.Ex(253609, J$e);
                                                                } finally {
                                                                    if (J$.Fr(253617))
                                                                        continue jalangiLabel15;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false))));
                                                } catch (J$e) {
                                                    J$.Ex(253625, J$e);
                                                } finally {
                                                    if (J$.Fr(253633))
                                                        continue jalangiLabel16;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(250977, J$.R(250969, 'SrvPollingEvent', SrvPollingEvent, false, false));
                        } catch (J$e) {
                            J$.Ex(253641, J$e);
                        } finally {
                            if (J$.Fr(253649))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), SrvPollingEvent, false, true);
            var SrvPoller = J$.W(253089, 'SrvPoller', J$.F(253081, J$.T(253065, function (_EventEmitter) {
                jalangiLabel31:
                    while (true) {
                        try {
                            J$.Fe(253025, arguments.callee, this, arguments);
                            function SrvPoller(options) {
                                jalangiLabel18:
                                    while (true) {
                                        try {
                                            J$.Fe(251457, arguments.callee, this, arguments);
                                            arguments = J$.N(251465, 'arguments', arguments, true, false, false);
                                            options = J$.N(251473, 'options', options, true, false, false);
                                            J$.N(251481, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(251097, J$.R(251073, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(251081, 'this', this, false, false), J$.R(251089, 'SrvPoller', SrvPoller, false, false));
                                            _this = J$.W(251169, '_this', J$.F(251161, J$.R(251105, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(251113, 'this', this, false, false), J$.M(251153, J$.F(251137, J$.R(251121, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(251129, 'SrvPoller', SrvPoller, false, false)), 'call', false)(J$.R(251145, 'this', this, false, false))), _this, false, false);
                                            if (J$.C(14544, J$.C(14536, J$.U(16074, '!', J$.R(251177, 'options', options, false, false))) ? J$._() : J$.U(16082, '!', J$.G(251193, J$.R(251185, 'options', options, false, false), 'srvHost')))) {
                                                throw J$.F(251217, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(251201, 'TypeError', undefined, true, true) : TypeError = J$.R(251201, 'TypeError', TypeError, true, true)), true)(J$.T(251209, 'options for SrvPoller must exist and include srvHost', 21, false));
                                            }
                                            J$.P(251249, J$.R(251225, '_this', _this, false, false), 'srvHost', J$.G(251241, J$.R(251233, 'options', options, false, false), 'srvHost'));
                                            J$.P(251273, J$.R(251257, '_this', _this, false, false), 'rescanSrvIntervalMS', J$.T(251265, 60000, 22, false));
                                            J$.P(251313, J$.R(251281, '_this', _this, false, false), 'heartbeatFrequencyMS', J$.C(14552, J$.G(251297, J$.R(251289, 'options', options, false, false), 'heartbeatFrequencyMS')) ? J$._() : J$.T(251305, 10000, 22, false));
                                            J$.P(251361, J$.R(251321, '_this', _this, false, false), 'logger', J$.F(251353, J$.R(251329, 'Logger', Logger, false, true), false)(J$.T(251337, 'srvPoller', 21, false), J$.R(251345, 'options', options, false, false)));
                                            J$.P(251385, J$.R(251369, '_this', _this, false, false), 'haMode', J$.T(251377, false, 23, false));
                                            J$.P(251409, J$.R(251393, '_this', _this, false, false), 'generation', J$.T(251401, 0, 22, false));
                                            J$.P(251433, J$.R(251417, '_this', _this, false, false), '_timeout', J$.T(251425, null, 25, false));
                                            return J$.Rt(251449, J$.R(251441, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(253657, J$e);
                                        } finally {
                                            if (J$.Fr(253665))
                                                continue jalangiLabel18;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(253033, 'arguments', arguments, true, false, false);
                            _EventEmitter = J$.N(253041, '_EventEmitter', _EventEmitter, true, false, false);
                            SrvPoller = J$.N(253057, 'SrvPoller', J$.T(253049, SrvPoller, 12, false), true, false, false);
                            J$.F(251065, J$.R(251041, '_inherits', _inherits, false, true), false)(J$.R(251049, 'SrvPoller', SrvPoller, false, false), J$.R(251057, '_EventEmitter', _EventEmitter, false, false));
                            J$.F(253001, J$.R(251489, '_createClass', _createClass, false, true), false)(J$.R(251497, 'SrvPoller', SrvPoller, false, false), J$.T(252993, [
                                J$.T(251569, {
                                    key: J$.T(251505, 'start', 21, false),
                                    value: J$.T(251561, function start() {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(251545, arguments.callee, this, arguments);
                                                    arguments = J$.N(251553, 'arguments', arguments, true, false, false);
                                                    if (J$.C(14560, J$.U(16090, '!', J$.G(251521, J$.R(251513, 'this', this, false, false), '_timeout')))) {
                                                        J$.M(251537, J$.R(251529, 'this', this, false, false), 'schedule', false)();
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(253673, J$e);
                                                } finally {
                                                    if (J$.Fr(253681))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(251705, {
                                    key: J$.T(251577, 'stop', 21, false),
                                    value: J$.T(251697, function stop() {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(251681, arguments.callee, this, arguments);
                                                    arguments = J$.N(251689, 'arguments', arguments, true, false, false);
                                                    if (J$.C(14568, J$.G(251593, J$.R(251585, 'this', this, false, false), '_timeout'))) {
                                                        J$.F(251625, J$.I(typeof clearTimeout === 'undefined' ? clearTimeout = J$.R(251601, 'clearTimeout', undefined, true, true) : clearTimeout = J$.R(251601, 'clearTimeout', clearTimeout, true, true)), false)(J$.G(251617, J$.R(251609, 'this', this, false, false), '_timeout'));
                                                        J$.A(251649, J$.R(251633, 'this', this, false, false), 'generation', '+')(J$.T(251641, 1, 22, false));
                                                        J$.P(251673, J$.R(251657, 'this', this, false, false), '_timeout', J$.T(251665, null, 25, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(253689, J$e);
                                                } finally {
                                                    if (J$.Fr(253697))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(251897, {
                                    key: J$.T(251713, 'schedule', 21, false),
                                    value: J$.T(251889, function schedule() {
                                        jalangiLabel22:
                                            while (true) {
                                                try {
                                                    J$.Fe(251865, arguments.callee, this, arguments);
                                                    arguments = J$.N(251873, 'arguments', arguments, true, false, false);
                                                    J$.N(251881, '_this2', _this2, false, false, false);
                                                    var _this2 = J$.W(251729, '_this2', J$.R(251721, 'this', this, false, false), _this2, false, false);
                                                    J$.F(251761, J$.I(typeof clearTimeout === 'undefined' ? clearTimeout = J$.R(251737, 'clearTimeout', undefined, true, true) : clearTimeout = J$.R(251737, 'clearTimeout', clearTimeout, true, true)), false)(J$.G(251753, J$.R(251745, 'this', this, false, false), '_timeout'));
                                                    J$.P(251857, J$.R(251769, 'this', this, false, false), '_timeout', J$.F(251849, J$.I(typeof setTimeout === 'undefined' ? setTimeout = J$.R(251777, 'setTimeout', undefined, true, true) : setTimeout = J$.R(251777, 'setTimeout', setTimeout, true, true)), false)(J$.T(251825, function () {
                                                        jalangiLabel21:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(251809, arguments.callee, this, arguments);
                                                                    arguments = J$.N(251817, 'arguments', arguments, true, false, false);
                                                                    return J$.Rt(251801, J$.M(251793, J$.R(251785, '_this2', _this2, false, false), '_poll', false)());
                                                                } catch (J$e) {
                                                                    J$.Ex(253705, J$e);
                                                                } finally {
                                                                    if (J$.Fr(253713))
                                                                        continue jalangiLabel21;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false), J$.G(251841, J$.R(251833, 'this', this, false, false), 'intervalMS')));
                                                } catch (J$e) {
                                                    J$.Ex(253721, J$e);
                                                } finally {
                                                    if (J$.Fr(253729))
                                                        continue jalangiLabel22;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(252033, {
                                    key: J$.T(251905, 'success', 21, false),
                                    value: J$.T(252025, function success(srvRecords) {
                                        jalangiLabel23:
                                            while (true) {
                                                try {
                                                    J$.Fe(252001, arguments.callee, this, arguments);
                                                    arguments = J$.N(252009, 'arguments', arguments, true, false, false);
                                                    srvRecords = J$.N(252017, 'srvRecords', srvRecords, true, false, false);
                                                    J$.P(251929, J$.R(251913, 'this', this, false, false), 'haMode', J$.T(251921, false, 23, false));
                                                    J$.M(251945, J$.R(251937, 'this', this, false, false), 'schedule', false)();
                                                    J$.M(251993, J$.R(251953, 'this', this, false, false), 'emit', false)(J$.T(251961, 'srvRecordDiscovery', 21, false), J$.F(251985, J$.R(251969, 'SrvPollingEvent', SrvPollingEvent, false, true), true)(J$.R(251977, 'srvRecords', srvRecords, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(253737, J$e);
                                                } finally {
                                                    if (J$.Fr(253745))
                                                        continue jalangiLabel23;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(252169, {
                                    key: J$.T(252041, 'failure', 21, false),
                                    value: J$.T(252161, function failure(message, obj) {
                                        jalangiLabel24:
                                            while (true) {
                                                try {
                                                    J$.Fe(252129, arguments.callee, this, arguments);
                                                    arguments = J$.N(252137, 'arguments', arguments, true, false, false);
                                                    message = J$.N(252145, 'message', message, true, false, false);
                                                    obj = J$.N(252153, 'obj', obj, true, false, false);
                                                    J$.M(252081, J$.G(252057, J$.R(252049, 'this', this, false, false), 'logger'), 'warn', false)(J$.R(252065, 'message', message, false, false), J$.R(252073, 'obj', obj, false, false));
                                                    J$.P(252105, J$.R(252089, 'this', this, false, false), 'haMode', J$.T(252097, true, 23, false));
                                                    J$.M(252121, J$.R(252113, 'this', this, false, false), 'schedule', false)();
                                                } catch (J$e) {
                                                    J$.Ex(253753, J$e);
                                                } finally {
                                                    if (J$.Fr(253761))
                                                        continue jalangiLabel24;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(252321, {
                                    key: J$.T(252177, 'parentDomainMismatch', 21, false),
                                    value: J$.T(252313, function parentDomainMismatch(srvRecord) {
                                        jalangiLabel25:
                                            while (true) {
                                                try {
                                                    J$.Fe(252289, arguments.callee, this, arguments);
                                                    arguments = J$.N(252297, 'arguments', arguments, true, false, false);
                                                    srvRecord = J$.N(252305, 'srvRecord', srvRecord, true, false, false);
                                                    J$.M(252281, J$.G(252193, J$.R(252185, 'this', this, false, false), 'logger'), 'warn', false)(J$.M(252265, J$.M(252233, J$.T(252201, 'parent domain mismatch on SRV record (', 21, false), 'concat', false)(J$.G(252217, J$.R(252209, 'srvRecord', srvRecord, false, false), 'name'), J$.T(252225, ':', 21, false)), 'concat', false)(J$.G(252249, J$.R(252241, 'srvRecord', srvRecord, false, false), 'port'), J$.T(252257, ')', 21, false)), J$.R(252273, 'srvRecord', srvRecord, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(253769, J$e);
                                                } finally {
                                                    if (J$.Fr(253777))
                                                        continue jalangiLabel25;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(252809, {
                                    key: J$.T(252329, '_poll', 21, false),
                                    value: J$.T(252801, function _poll() {
                                        jalangiLabel28:
                                            while (true) {
                                                try {
                                                    J$.Fe(252769, arguments.callee, this, arguments);
                                                    arguments = J$.N(252777, 'arguments', arguments, true, false, false);
                                                    J$.N(252785, '_this3', _this3, false, false, false);
                                                    J$.N(252793, 'generation', generation, false, false, false);
                                                    var _this3 = J$.W(252345, '_this3', J$.R(252337, 'this', this, false, false), _this3, false, false);
                                                    var generation = J$.W(252369, 'generation', J$.G(252361, J$.R(252353, 'this', this, false, false), 'generation'), generation, false, false);
                                                    J$.M(252761, J$.R(252377, 'dns', dns, false, true), 'resolveSrv', false)(J$.G(252393, J$.R(252385, 'this', this, false, false), 'srvAddress'), J$.T(252753, function (err, srvRecords) {
                                                        jalangiLabel27:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(252713, arguments.callee, this, arguments);
                                                                    arguments = J$.N(252721, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(252729, 'err', err, true, false, false);
                                                                    srvRecords = J$.N(252737, 'srvRecords', srvRecords, true, false, false);
                                                                    J$.N(252745, 'finalAddresses', finalAddresses, false, false, false);
                                                                    if (J$.C(14576, J$.B(16098, '!==', J$.R(252401, 'generation', generation, false, false), J$.G(252417, J$.R(252409, '_this3', _this3, false, false), 'generation')))) {
                                                                        return J$.Rt(252425, undefined);
                                                                    }
                                                                    if (J$.C(14584, J$.R(252433, 'err', err, false, false))) {
                                                                        J$.M(252465, J$.R(252441, '_this3', _this3, false, false), 'failure', false)(J$.T(252449, 'DNS error', 21, false), J$.R(252457, 'err', err, false, false));
                                                                        return J$.Rt(252473, undefined);
                                                                    }
                                                                    var finalAddresses = J$.W(252489, 'finalAddresses', J$.T(252481, [], 10, false), finalAddresses, false, false);
                                                                    J$.M(252633, J$.R(252497, 'srvRecords', srvRecords, false, false), 'forEach', false)(J$.T(252625, function (record) {
                                                                        jalangiLabel26:
                                                                            while (true) {
                                                                                try {
                                                                                    J$.Fe(252601, arguments.callee, this, arguments);
                                                                                    arguments = J$.N(252609, 'arguments', arguments, true, false, false);
                                                                                    record = J$.N(252617, 'record', record, true, false, false);
                                                                                    if (J$.C(14592, J$.F(252545, J$.R(252505, 'matchesParentDomain', matchesParentDomain, false, true), false)(J$.G(252521, J$.R(252513, 'record', record, false, false), 'name'), J$.G(252537, J$.R(252529, '_this3', _this3, false, false), 'srvHost')))) {
                                                                                        J$.M(252569, J$.R(252553, 'finalAddresses', finalAddresses, false, false), 'push', false)(J$.R(252561, 'record', record, false, false));
                                                                                    } else {
                                                                                        J$.M(252593, J$.R(252577, '_this3', _this3, false, false), 'parentDomainMismatch', false)(J$.R(252585, 'record', record, false, false));
                                                                                    }
                                                                                } catch (J$e) {
                                                                                    J$.Ex(253785, J$e);
                                                                                } finally {
                                                                                    if (J$.Fr(253793))
                                                                                        continue jalangiLabel26;
                                                                                    else
                                                                                        return J$.Ra();
                                                                                }
                                                                            }
                                                                    }, 12, false));
                                                                    if (J$.C(14600, J$.U(16106, '!', J$.G(252649, J$.R(252641, 'finalAddresses', finalAddresses, false, false), 'length')))) {
                                                                        J$.M(252673, J$.R(252657, '_this3', _this3, false, false), 'failure', false)(J$.T(252665, 'No valid addresses found at host', 21, false));
                                                                        return J$.Rt(252681, undefined);
                                                                    }
                                                                    J$.M(252705, J$.R(252689, '_this3', _this3, false, false), 'success', false)(J$.R(252697, 'finalAddresses', finalAddresses, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(253801, J$e);
                                                                } finally {
                                                                    if (J$.Fr(253809))
                                                                        continue jalangiLabel27;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(253817, J$e);
                                                } finally {
                                                    if (J$.Fr(253825))
                                                        continue jalangiLabel28;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(252889, {
                                    key: J$.T(252817, 'srvAddress', 21, false),
                                    get: J$.T(252881, function get() {
                                        jalangiLabel29:
                                            while (true) {
                                                try {
                                                    J$.Fe(252865, arguments.callee, this, arguments);
                                                    arguments = J$.N(252873, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(252857, J$.M(252849, J$.T(252825, '_mongodb._tcp.', 21, false), 'concat', false)(J$.G(252841, J$.R(252833, 'this', this, false, false), 'srvHost')));
                                                } catch (J$e) {
                                                    J$.Ex(253833, J$e);
                                                } finally {
                                                    if (J$.Fr(253841))
                                                        continue jalangiLabel29;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(252985, {
                                    key: J$.T(252897, 'intervalMS', 21, false),
                                    get: J$.T(252977, function get() {
                                        jalangiLabel30:
                                            while (true) {
                                                try {
                                                    J$.Fe(252961, arguments.callee, this, arguments);
                                                    arguments = J$.N(252969, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(252953, J$.C(14608, J$.G(252913, J$.R(252905, 'this', this, false, false), 'haMode')) ? J$.G(252929, J$.R(252921, 'this', this, false, false), 'heartbeatFrequencyMS') : J$.G(252945, J$.R(252937, 'this', this, false, false), 'rescanSrvIntervalMs'));
                                                } catch (J$e) {
                                                    J$.Ex(253849, J$e);
                                                } finally {
                                                    if (J$.Fr(253857))
                                                        continue jalangiLabel30;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(253017, J$.R(253009, 'SrvPoller', SrvPoller, false, false));
                        } catch (J$e) {
                            J$.Ex(253865, J$e);
                        } finally {
                            if (J$.Fr(253873))
                                continue jalangiLabel31;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(253073, 'EventEmitter', EventEmitter, false, true)), SrvPoller, false, true);
            J$.P(253121, J$.G(253105, J$.I(typeof module === 'undefined' ? module = J$.R(253097, 'module', undefined, true, true) : module = J$.R(253097, 'module', module, true, true)), 'exports'), 'SrvPollingEvent', J$.R(253113, 'SrvPollingEvent', SrvPollingEvent, false, true));
            J$.P(253153, J$.G(253137, J$.I(typeof module === 'undefined' ? module = J$.R(253129, 'module', undefined, true, true) : module = J$.R(253129, 'module', module, true, true)), 'exports'), 'SrvPoller', J$.R(253145, 'SrvPoller', SrvPoller, false, true));
        } catch (J$e) {
            J$.Ex(253881, J$e);
        } finally {
            if (J$.Sr(253889))
                continue jalangiLabel32;
            else
                break jalangiLabel32;
        }
    }
// JALANGI DO NOT INSTRUMENT
