// mongo-parse/DotNotationPointers.js
J$.noInstrEval = false;
jalangiLabel12:
    while (true) {
        try {
            J$.Se(7057, '../target_tmp/node_modules/mongo-parse/DotNotationPointers.js');
            function createPointers(rootObject, propertyParts) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(5681, arguments.callee, this, arguments);
                            arguments = J$.N(5689, 'arguments', arguments, true, false, false);
                            rootObject = J$.N(5697, 'rootObject', rootObject, true, false, false);
                            propertyParts = J$.N(5705, 'propertyParts', propertyParts, true, false, false);
                            J$.N(5713, 'initialObject', initialObject, false, false, false);
                            J$.N(5721, 'curInfoObjects', curInfoObjects, false, false, false);
                            var initialObject = J$.W(4889, 'initialObject', J$.T(4881, { dummy: J$.R(4873, 'rootObject', rootObject, false, false) }, 11, false), initialObject, false, false);
                            var curInfoObjects = J$.W(4937, 'curInfoObjects', J$.T(4929, [J$.T(4921, {
                                    obj: J$.R(4897, 'initialObject', initialObject, false, false),
                                    last: J$.T(4905, 'dummy', 21, false),
                                    propertyPath: J$.T(4913, [], 10, false)
                                }, 11, false)], 10, false), curInfoObjects, false, false);
                            J$.M(5409, J$.R(4945, 'propertyParts', propertyParts, false, false), 'forEach', false)(J$.T(5401, function (part) {
                                jalangiLabel3:
                                    while (true) {
                                        try {
                                            J$.Fe(5369, arguments.callee, this, arguments);
                                            arguments = J$.N(5377, 'arguments', arguments, true, false, false);
                                            part = J$.N(5385, 'part', part, true, false, false);
                                            J$.N(5393, 'nextInfoObjects', nextInfoObjects, false, false, false);
                                            var nextInfoObjects = J$.W(4961, 'nextInfoObjects', J$.T(4953, [], 10, false), nextInfoObjects, false, false);
                                            J$.M(5345, J$.R(4969, 'curInfoObjects', curInfoObjects, false, false), 'forEach', false)(J$.T(5337, function (current) {
                                                jalangiLabel2:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(5305, arguments.callee, this, arguments);
                                                            arguments = J$.N(5313, 'arguments', arguments, true, false, false);
                                                            current = J$.N(5321, 'current', current, true, false, false);
                                                            J$.N(5329, 'curValue', curValue, false, false, false);
                                                            var curValue = J$.W(5025, 'curValue', J$.F(5017, J$.R(4977, 'getValue', getValue, false, true), false)(J$.G(4993, J$.R(4985, 'current', current, false, false), 'obj'), J$.G(5009, J$.R(5001, 'current', current, false, false), 'last')), curValue, false, false);
                                                            if (J$.C(184, J$.C(176, J$.B(330, 'instanceof', J$.R(5033, 'curValue', curValue, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(5041, 'Array', undefined, true, true) : Array = J$.R(5041, 'Array', Array, true, true)))) ? J$.U(338, '!', J$.F(5065, J$.R(5049, 'isInteger', isInteger, false, true), false)(J$.R(5057, 'part', part, false, false))) : J$._())) {
                                                                J$.M(5225, J$.R(5073, 'curValue', curValue, false, false), 'forEach', false)(J$.T(5217, function (property, index) {
                                                                    jalangiLabel1:
                                                                        while (true) {
                                                                            try {
                                                                                J$.Fe(5185, arguments.callee, this, arguments);
                                                                                arguments = J$.N(5193, 'arguments', arguments, true, false, false);
                                                                                property = J$.N(5201, 'property', property, true, false, false);
                                                                                index = J$.N(5209, 'index', index, true, false, false);
                                                                                J$.M(5177, J$.R(5081, 'nextInfoObjects', nextInfoObjects, false, false), 'push', false)(J$.T(5169, {
                                                                                    obj: J$.F(5113, J$.R(5089, 'getValue', getValue, false, true), false)(J$.R(5097, 'curValue', curValue, false, false), J$.R(5105, 'index', index, false, false)),
                                                                                    propertyPath: J$.M(5153, J$.G(5129, J$.R(5121, 'current', current, false, false), 'propertyPath'), 'concat', false)(J$.R(5137, 'index', index, false, false), J$.R(5145, 'part', part, false, false)),
                                                                                    last: J$.R(5161, 'part', part, false, false)
                                                                                }, 11, false));
                                                                            } catch (J$e) {
                                                                                J$.Ex(7161, J$e);
                                                                            } finally {
                                                                                if (J$.Fr(7169))
                                                                                    continue jalangiLabel1;
                                                                                else
                                                                                    return J$.Ra();
                                                                            }
                                                                        }
                                                                }, 12, false));
                                                            } else {
                                                                J$.M(5297, J$.R(5233, 'nextInfoObjects', nextInfoObjects, false, false), 'push', false)(J$.T(5289, {
                                                                    obj: J$.R(5241, 'curValue', curValue, false, false),
                                                                    propertyPath: J$.M(5273, J$.G(5257, J$.R(5249, 'current', current, false, false), 'propertyPath'), 'concat', false)(J$.R(5265, 'part', part, false, false)),
                                                                    last: J$.R(5281, 'part', part, false, false)
                                                                }, 11, false));
                                                            }
                                                        } catch (J$e) {
                                                            J$.Ex(7177, J$e);
                                                        } finally {
                                                            if (J$.Fr(7185))
                                                                continue jalangiLabel2;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                            curInfoObjects = J$.W(5361, 'curInfoObjects', J$.R(5353, 'nextInfoObjects', nextInfoObjects, false, false), curInfoObjects, false, false);
                                        } catch (J$e) {
                                            J$.Ex(7193, J$e);
                                        } finally {
                                            if (J$.Fr(7201))
                                                continue jalangiLabel3;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            return J$.Rt(5673, J$.M(5665, J$.R(5417, 'curInfoObjects', curInfoObjects, false, false), 'map', false)(J$.T(5657, function (current) {
                                jalangiLabel4:
                                    while (true) {
                                        try {
                                            J$.Fe(5617, arguments.callee, this, arguments);
                                            arguments = J$.N(5625, 'arguments', arguments, true, false, false);
                                            current = J$.N(5633, 'current', current, true, false, false);
                                            J$.N(5641, 'obj', obj, false, false, false);
                                            J$.N(5649, 'last', last, false, false, false);
                                            if (J$.C(192, J$.B(346, '===', J$.G(5433, J$.R(5425, 'current', current, false, false), 'obj'), J$.R(5441, 'initialObject', initialObject, false, false)))) {
                                                var obj = J$.W(5473, 'obj', J$.G(5465, J$.G(5457, J$.R(5449, 'current', current, false, false), 'obj'), 'dummy'), obj, false, false);
                                                var last = J$.W(5489, 'last', J$.T(5481, undefined, 24, false), last, false, false);
                                            } else {
                                                var obj = J$.W(5513, 'obj', J$.G(5505, J$.R(5497, 'current', current, false, false), 'obj'), obj, false, false);
                                                var last = J$.W(5537, 'last', J$.G(5529, J$.R(5521, 'current', current, false, false), 'last'), last, false, false);
                                            }
                                            return J$.Rt(5609, J$.F(5601, J$.R(5545, 'DotNotationPointer', DotNotationPointer, false, true), true)(J$.R(5553, 'rootObject', rootObject, false, false), J$.G(5569, J$.R(5561, 'current', current, false, false), 'propertyPath'), J$.T(5593, {
                                                obj: J$.R(5577, 'obj', obj, false, false),
                                                last: J$.R(5585, 'last', last, false, false)
                                            }, 11, false)));
                                        } catch (J$e) {
                                            J$.Ex(7209, J$e);
                                        } finally {
                                            if (J$.Fr(7217))
                                                continue jalangiLabel4;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(7225, J$e);
                        } finally {
                            if (J$.Fr(7233))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function getValue(object, key) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(5793, arguments.callee, this, arguments);
                            arguments = J$.N(5801, 'arguments', arguments, true, false, false);
                            object = J$.N(5809, 'object', object, true, false, false);
                            key = J$.N(5817, 'key', key, true, false, false);
                            if (J$.C(200, J$.B(354, '===', J$.R(5729, 'object', object, false, false), J$.T(5737, undefined, 24, false))))
                                return J$.Rt(5753, J$.T(5745, undefined, 24, false));
                            else
                                return J$.Rt(5785, J$.G(5777, J$.R(5761, 'object', object, false, false), J$.R(5769, 'key', key, false, false)));
                        } catch (J$e) {
                            J$.Ex(7241, J$e);
                        } finally {
                            if (J$.Fr(7249))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function createProperty(that) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(6897, arguments.callee, this, arguments);
                            arguments = J$.N(6905, 'arguments', arguments, true, false, false);
                            that = J$.N(6913, 'that', that, true, false, false);
                            J$.N(6921, 'result', result, false, false, false);
                            J$.N(6929, 'lastIndex', lastIndex, false, false, false);
                            J$.N(6937, 'n', n, false, false, false);
                            J$.N(6945, 'value', value, false, false, false);
                            J$.N(6953, 'newValue', newValue, false, false, false);
                            var result = J$.W(6497, 'result', J$.G(6489, J$.R(6481, 'that', that, false, false), 'root'), result, false, false);
                            var lastIndex = J$.W(6537, 'lastIndex', J$.B(410, '-', J$.G(6521, J$.G(6513, J$.R(6505, 'that', that, false, false), 'property'), 'length'), J$.T(6529, 1, 22, false)), lastIndex, false, false);
                            for (var n = J$.W(6553, 'n', J$.T(6545, 0, 22, false), n, false, false); J$.C(272, J$.B(418, '<', J$.R(6561, 'n', n, false, false), J$.R(6569, 'lastIndex', lastIndex, false, false))); J$.B(442, '-', n = J$.W(6585, 'n', J$.B(434, '+', J$.U(426, '+', J$.R(6577, 'n', n, false, false)), 1), n, false, false), 1)) {
                                var value = J$.W(6641, 'value', J$.G(6633, J$.R(6593, 'result', result, false, false), J$.G(6625, J$.G(6609, J$.R(6601, 'that', that, false, false), 'property'), J$.R(6617, 'n', n, false, false))), value, false, false);
                                if (J$.C(264, J$.B(450, '===', J$.R(6649, 'value', value, false, false), J$.T(6657, undefined, 24, false)))) {
                                    if (J$.C(256, J$.F(6713, J$.R(6665, 'isInteger', isInteger, false, true), false)(J$.G(6705, J$.G(6681, J$.R(6673, 'that', that, false, false), 'property'), J$.B(458, '+', J$.R(6689, 'n', n, false, false), J$.T(6697, 1, 22, false))))))
                                        var newValue = J$.W(6729, 'newValue', J$.T(6721, [], 10, false), newValue, false, false);
                                    else
                                        var newValue = J$.W(6745, 'newValue', J$.T(6737, {}, 11, false), newValue, false, false);
                                    value = J$.W(6809, 'value', J$.P(6801, J$.R(6753, 'result', result, false, false), J$.G(6785, J$.G(6769, J$.R(6761, 'that', that, false, false), 'property'), J$.R(6777, 'n', n, false, false)), J$.R(6793, 'newValue', newValue, false, false)), value, false, false);
                                }
                                result = J$.W(6825, 'result', J$.R(6817, 'value', value, false, false), result, false, false);
                            }
                            J$.P(6889, J$.R(6833, 'that', that, false, false), 'propertyInfo', J$.T(6881, {
                                obj: J$.R(6841, 'result', result, false, false),
                                last: J$.G(6873, J$.G(6857, J$.R(6849, 'that', that, false, false), 'property'), J$.R(6865, 'lastIndex', lastIndex, false, false))
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(7305, J$e);
                        } finally {
                            if (J$.Fr(7313))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isInteger(v) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(7025, arguments.callee, this, arguments);
                            arguments = J$.N(7033, 'arguments', arguments, true, false, false);
                            v = J$.N(7041, 'v', v, true, false, false);
                            J$.N(7049, 'number', number, false, false, false);
                            var number = J$.W(6985, 'number', J$.F(6977, J$.I(typeof parseInt === 'undefined' ? parseInt = J$.R(6961, 'parseInt', undefined, true, true) : parseInt = J$.R(6961, 'parseInt', parseInt, true, true)), false)(J$.R(6969, 'v', v, false, false)), number, false, false);
                            return J$.Rt(7017, J$.U(466, '!', J$.F(7009, J$.I(typeof isNaN === 'undefined' ? isNaN = J$.R(6993, 'isNaN', undefined, true, true) : isNaN = J$.R(6993, 'isNaN', isNaN, true, true)), false)(J$.R(7001, 'number', number, false, false))));
                        } catch (J$e) {
                            J$.Ex(7321, J$e);
                        } finally {
                            if (J$.Fr(7329))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(7065, 'DotNotationPointers', DotNotationPointers, false, false, false);
            createPointers = J$.N(7081, 'createPointers', J$.T(7073, createPointers, 12, false), true, false, false);
            getValue = J$.N(7097, 'getValue', J$.T(7089, getValue, 12, false), true, false, false);
            J$.N(7105, 'DotNotationPointer', DotNotationPointer, false, false, false);
            createProperty = J$.N(7121, 'createProperty', J$.T(7113, createProperty, 12, false), true, false, false);
            isInteger = J$.N(7137, 'isInteger', J$.T(7129, isInteger, 12, false), true, false, false);
            var DotNotationPointers = J$.W(4865, 'DotNotationPointers', J$.P(4857, J$.I(typeof module === 'undefined' ? module = J$.R(4689, 'module', undefined, true, true) : module = J$.R(4689, 'module', module, true, true)), 'exports', J$.T(4849, function (rootObject, property) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(4817, arguments.callee, this, arguments);
                            arguments = J$.N(4825, 'arguments', arguments, true, false, false);
                            rootObject = J$.N(4833, 'rootObject', rootObject, true, false, false);
                            property = J$.N(4841, 'property', property, true, false, false);
                            if (J$.C(168, J$.B(306, '===', J$.R(4697, 'property', property, false, false), J$.T(4705, undefined, 24, false)))) {
                                property = J$.W(4721, 'property', J$.T(4713, [], 10, false), property, false, false);
                            } else if (J$.C(160, J$.U(322, '!', J$.B(314, 'instanceof', J$.R(4729, 'property', property, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(4737, 'Array', undefined, true, true) : Array = J$.R(4737, 'Array', Array, true, true)))))) {
                                property = J$.W(4769, 'property', J$.M(4761, J$.R(4745, 'property', property, false, false), 'split', false)(J$.T(4753, '.', 21, false)), property, false, false);
                            }
                            return J$.Rt(4809, J$.F(4801, J$.R(4777, 'createPointers', createPointers, false, true), false)(J$.R(4785, 'rootObject', rootObject, false, false), J$.R(4793, 'property', property, false, false)));
                        } catch (J$e) {
                            J$.Ex(7145, J$e);
                        } finally {
                            if (J$.Fr(7153))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), DotNotationPointers, false, true);
            var DotNotationPointer = J$.W(6057, 'DotNotationPointer', J$.T(6049, function (rootObject, property, propertyInfo) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(6009, arguments.callee, this, arguments);
                            arguments = J$.N(6017, 'arguments', arguments, true, false, false);
                            rootObject = J$.N(6025, 'rootObject', rootObject, true, false, false);
                            property = J$.N(6033, 'property', property, true, false, false);
                            propertyInfo = J$.N(6041, 'propertyInfo', propertyInfo, true, false, false);
                            J$.P(5841, J$.R(5825, 'this', this, false, false), 'root', J$.R(5833, 'rootObject', rootObject, false, false));
                            if (J$.C(216, J$.B(362, '===', J$.R(5849, 'property', property, false, false), J$.T(5857, undefined, 24, false)))) {
                                J$.P(5881, J$.R(5865, 'this', this, false, false), 'property', J$.T(5873, [], 10, false));
                            } else if (J$.C(208, J$.B(370, 'instanceof', J$.R(5889, 'property', property, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(5897, 'Array', undefined, true, true) : Array = J$.R(5897, 'Array', Array, true, true))))) {
                                J$.P(5921, J$.R(5905, 'this', this, false, false), 'property', J$.R(5913, 'property', property, false, false));
                            } else {
                                J$.P(5961, J$.R(5929, 'this', this, false, false), 'property', J$.M(5953, J$.R(5937, 'property', property, false, false), 'split', false)(J$.T(5945, '.', 21, false)));
                            }
                            if (J$.C(224, J$.B(378, '!==', J$.R(5969, 'propertyInfo', propertyInfo, false, false), J$.T(5977, undefined, 24, false)))) {
                                J$.P(6001, J$.R(5985, 'this', this, false, false), 'propertyInfo', J$.R(5993, 'propertyInfo', propertyInfo, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(7257, J$e);
                        } finally {
                            if (J$.Fr(7265))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), DotNotationPointer, false, true);
            J$.P(6081, J$.R(6065, 'DotNotationPointer', DotNotationPointer, false, true), 'prototype', J$.T(6073, {}, 11, false));
            J$.M(6473, J$.I(typeof Object === 'undefined' ? Object = J$.R(6089, 'Object', undefined, true, true) : Object = J$.R(6089, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(6105, J$.R(6097, 'DotNotationPointer', DotNotationPointer, false, true), 'prototype'), J$.T(6113, 'val', 21, false), J$.T(6465, {
                get: J$.T(6305, function () {
                    jalangiLabel8:
                        while (true) {
                            try {
                                J$.Fe(6281, arguments.callee, this, arguments);
                                arguments = J$.N(6289, 'arguments', arguments, true, false, false);
                                J$.N(6297, 'info', info, false, false, false);
                                var info = J$.W(6137, 'info', J$.G(6129, J$.R(6121, 'this', this, false, false), 'propertyInfo'), info, false, false);
                                if (J$.C(240, J$.B(386, '===', J$.G(6153, J$.R(6145, 'info', info, false, false), 'obj'), J$.T(6161, undefined, 24, false)))) {
                                    return J$.Rt(6177, J$.T(6169, undefined, 24, false));
                                } else {
                                    if (J$.C(232, J$.B(394, '!==', J$.G(6193, J$.R(6185, 'info', info, false, false), 'last'), J$.T(6201, undefined, 24, false)))) {
                                        return J$.Rt(6249, J$.G(6241, J$.G(6217, J$.R(6209, 'info', info, false, false), 'obj'), J$.G(6233, J$.R(6225, 'info', info, false, false), 'last')));
                                    } else {
                                        return J$.Rt(6273, J$.G(6265, J$.R(6257, 'info', info, false, false), 'obj'));
                                    }
                                }
                            } catch (J$e) {
                                J$.Ex(7273, J$e);
                            } finally {
                                if (J$.Fr(7281))
                                    continue jalangiLabel8;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                set: J$.T(6457, function (value) {
                    jalangiLabel9:
                        while (true) {
                            try {
                                J$.Fe(6433, arguments.callee, this, arguments);
                                arguments = J$.N(6441, 'arguments', arguments, true, false, false);
                                value = J$.N(6449, 'value', value, true, false, false);
                                if (J$.C(248, J$.B(402, '===', J$.G(6329, J$.G(6321, J$.R(6313, 'this', this, false, false), 'propertyInfo'), 'obj'), J$.T(6337, undefined, 24, false)))) {
                                    J$.F(6361, J$.R(6345, 'createProperty', createProperty, false, true), false)(J$.R(6353, 'this', this, false, false));
                                }
                                J$.P(6425, J$.G(6385, J$.G(6377, J$.R(6369, 'this', this, false, false), 'propertyInfo'), 'obj'), J$.G(6409, J$.G(6401, J$.R(6393, 'this', this, false, false), 'propertyInfo'), 'last'), J$.R(6417, 'value', value, false, false));
                            } catch (J$e) {
                                J$.Ex(7289, J$e);
                            } finally {
                                if (J$.Fr(7297))
                                    continue jalangiLabel9;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(7337, J$e);
        } finally {
            if (J$.Sr(7345))
                continue jalangiLabel12;
            else
                break jalangiLabel12;
        }
    }
// JALANGI DO NOT INSTRUMENT
