// mongo-parse/matches.js
J$.noInstrEval = false;
jalangiLabel22:
    while (true) {
        try {
            J$.Se(10729, '../target_tmp/node_modules/mongo-parse/matches.js');
            function partMatches(part, document) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(9217, arguments.callee, this, arguments);
                            arguments = J$.N(9225, 'arguments', arguments, true, false, false);
                            part = J$.N(9233, 'part', part, true, false, false);
                            document = J$.N(9241, 'document', document, true, false, false);
                            J$.N(9249, 'pointers', pointers, false, false, false);
                            J$.N(9257, 'p', p, false, false, false);
                            J$.N(9265, 'pointer', pointer, false, false, false);
                            J$.N(9273, 'test', test, false, false, false);
                            J$.N(9281, 'documentField', documentField, false, false, false);
                            J$.N(9289, 'n', n, false, false, false);
                            J$.N(9297, 'anyMatched', anyMatched, false, false, false);
                            J$.N(9305, 'documentToMatch', documentToMatch, false, false, false);
                            J$.N(9313, 'objectContext', objectContext, false, false, false);
                            var pointers = J$.W(7937, 'pointers', J$.F(7929, J$.R(7897, 'DotNotationPointers', DotNotationPointers, false, true), false)(J$.R(7905, 'document', document, false, false), J$.G(7921, J$.R(7913, 'part', part, false, false), 'field')), pointers, false, false);
                            for (var p = J$.W(7953, 'p', J$.T(7945, 0, 22, false), p, false, false); J$.C(544, J$.B(690, '<', J$.R(7961, 'p', p, false, false), J$.G(7977, J$.R(7969, 'pointers', pointers, false, false), 'length'))); J$.B(714, '-', p = J$.W(7993, 'p', J$.B(706, '+', J$.U(698, '+', J$.R(7985, 'p', p, false, false)), 1), p, false, false), 1)) {
                                var pointer = J$.W(8025, 'pointer', J$.G(8017, J$.R(8001, 'pointers', pointers, false, false), J$.R(8009, 'p', p, false, false)), pointer, false, false);
                                if (J$.C(536, J$.B(722, '===', J$.G(8041, J$.R(8033, 'part', part, false, false), 'operator'), J$.T(8049, undefined, 24, false)))) {
                                    if (J$.C(352, J$.U(730, '!', J$.F(8105, J$.R(8057, 'valueTest', valueTest, false, true), false)(J$.G(8073, J$.R(8065, 'pointer', pointer, false, false), 'val'), J$.G(8089, J$.R(8081, 'part', part, false, false), 'operand'), J$.R(8097, 'mongoEqual', mongoEqual, false, true))))) {
                                        continue;
                                    }
                                } else if (J$.C(528, J$.B(738, 'in', J$.G(8121, J$.R(8113, 'part', part, false, false), 'operator'), J$.R(8129, 'simpleComparators', simpleComparators, false, true)))) {
                                    var test = J$.W(8217, 'test', J$.F(8209, J$.R(8137, 'valueTest', valueTest, false, true), false)(J$.G(8153, J$.R(8145, 'pointer', pointer, false, false), 'val'), J$.G(8169, J$.R(8161, 'part', part, false, false), 'operand'), J$.G(8201, J$.R(8177, 'simpleComparators', simpleComparators, false, true), J$.G(8193, J$.R(8185, 'part', part, false, false), 'operator'))), test, false, false);
                                    if (J$.C(360, J$.U(746, '!', J$.R(8225, 'test', test, false, false))))
                                        continue;
                                } else if (J$.C(520, J$.B(754, 'in', J$.G(8241, J$.R(8233, 'part', part, false, false), 'operator'), J$.R(8249, 'compoundOperatorComparators', compoundOperatorComparators, false, true)))) {
                                    if (J$.C(368, J$.U(762, '!', J$.M(8305, J$.R(8257, 'compoundOperatorComparators', compoundOperatorComparators, false, true), J$.G(8273, J$.R(8265, 'part', part, false, false), 'operator'), false)(J$.R(8281, 'document', document, false, false), J$.G(8297, J$.R(8289, 'part', part, false, false), 'parts'))))) {
                                        continue;
                                    }
                                } else if (J$.C(512, J$.B(770, '===', J$.G(8321, J$.R(8313, 'part', part, false, false), 'operator'), J$.T(8329, '$not', 21, false)))) {
                                    if (J$.C(392, J$.B(778, '>', J$.G(8353, J$.G(8345, J$.R(8337, 'part', part, false, false), 'parts'), 'length'), J$.T(8361, 0, 22, false)))) {
                                        if (J$.C(376, J$.F(8401, J$.R(8369, 'matches', matches, false, true), false)(J$.G(8385, J$.R(8377, 'part', part, false, false), 'parts'), J$.R(8393, 'document', document, false, false)))) {
                                            continue;
                                        }
                                    } else {
                                        if (J$.C(384, J$.B(786, '===', J$.F(8457, J$.R(8409, 'valueTest', valueTest, false, true), false)(J$.G(8425, J$.R(8417, 'pointer', pointer, false, false), 'val'), J$.G(8441, J$.R(8433, 'part', part, false, false), 'operand'), J$.R(8449, 'mongoEqual', mongoEqual, false, true)), J$.T(8465, true, 23, false))))
                                            continue;
                                    }
                                } else if (J$.C(504, J$.B(794, '===', J$.G(8481, J$.R(8473, 'part', part, false, false), 'operator'), J$.T(8489, '$size', 21, false)))) {
                                    return J$.Rt(8561, J$.C(400, J$.B(802, 'instanceof', J$.G(8505, J$.R(8497, 'pointer', pointer, false, false), 'val'), J$.I(typeof Array === 'undefined' ? Array = J$.R(8513, 'Array', undefined, true, true) : Array = J$.R(8513, 'Array', Array, true, true)))) ? J$.B(810, '===', J$.G(8537, J$.G(8529, J$.R(8521, 'pointer', pointer, false, false), 'val'), 'length'), J$.G(8553, J$.R(8545, 'part', part, false, false), 'operand')) : J$._());
                                } else if (J$.C(496, J$.B(818, '===', J$.G(8577, J$.R(8569, 'part', part, false, false), 'operator'), J$.T(8585, '$elemMatch', 21, false)))) {
                                    var documentField = J$.W(8609, 'documentField', J$.G(8601, J$.R(8593, 'pointer', pointer, false, false), 'val'), documentField, false, false);
                                    if (J$.C(408, J$.B(826, '===', J$.R(8617, 'documentField', documentField, false, false), J$.T(8625, undefined, 24, false))))
                                        continue;
                                    if (J$.C(424, J$.G(8641, J$.R(8633, 'part', part, false, false), 'implicitField'))) {
                                        for (var n = J$.W(8657, 'n', J$.T(8649, 0, 22, false), n, false, false); J$.C(416, J$.B(834, '<', J$.R(8665, 'n', n, false, false), J$.G(8689, J$.G(8681, J$.R(8673, 'part', part, false, false), 'parts'), 'length'))); J$.B(858, '-', n = J$.W(8705, 'n', J$.B(850, '+', J$.U(842, '+', J$.R(8697, 'n', n, false, false)), 1), n, false, false), 1)) {
                                            J$.P(8753, J$.G(8737, J$.G(8721, J$.R(8713, 'part', part, false, false), 'parts'), J$.R(8729, 'n', n, false, false)), 'field', J$.T(8745, 'x', 21, false));
                                        }
                                    }
                                    var anyMatched = J$.W(8769, 'anyMatched', J$.T(8761, false, 23, false), anyMatched, false, false);
                                    for (var n = J$.W(8785, 'n', J$.T(8777, 0, 22, false), n, false, false); J$.C(448, J$.B(866, '<', J$.R(8793, 'n', n, false, false), J$.G(8809, J$.R(8801, 'documentField', documentField, false, false), 'length'))); J$.B(890, '-', n = J$.W(8825, 'n', J$.B(882, '+', J$.U(874, '+', J$.R(8817, 'n', n, false, false)), 1), n, false, false), 1)) {
                                        if (J$.C(432, J$.G(8841, J$.R(8833, 'part', part, false, false), 'implicitField'))) {
                                            var documentToMatch = J$.W(8881, 'documentToMatch', J$.T(8873, { x: J$.G(8865, J$.R(8849, 'documentField', documentField, false, false), J$.R(8857, 'n', n, false, false)) }, 11, false), documentToMatch, false, false);
                                        } else {
                                            var documentToMatch = J$.W(8913, 'documentToMatch', J$.G(8905, J$.R(8889, 'documentField', documentField, false, false), J$.R(8897, 'n', n, false, false)), documentToMatch, false, false);
                                        }
                                        if (J$.C(440, J$.F(8953, J$.R(8921, 'matches', matches, false, true), false)(J$.G(8937, J$.R(8929, 'part', part, false, false), 'parts'), J$.R(8945, 'documentToMatch', documentToMatch, false, false)))) {
                                            anyMatched = J$.W(8969, 'anyMatched', J$.T(8961, true, 23, false), anyMatched, false, false);
                                            break;
                                        }
                                    }
                                    if (J$.C(456, J$.U(898, '!', J$.R(8977, 'anyMatched', anyMatched, false, false))))
                                        continue;
                                } else if (J$.C(488, J$.B(906, '===', J$.G(8993, J$.R(8985, 'part', part, false, false), 'operator'), J$.T(9001, '$where', 21, false)))) {
                                    if (J$.C(464, J$.B(914, '!==', J$.G(9017, J$.R(9009, 'part', part, false, false), 'field'), J$.T(9025, undefined, 24, false)))) {
                                        var objectContext = J$.W(9049, 'objectContext', J$.G(9041, J$.R(9033, 'pointer', pointer, false, false), 'val'), objectContext, false, false);
                                    } else {
                                        var objectContext = J$.W(9065, 'objectContext', J$.R(9057, 'document', document, false, false), objectContext, false, false);
                                    }
                                    if (J$.C(472, J$.U(922, '!', J$.M(9097, J$.G(9081, J$.R(9073, 'part', part, false, false), 'operand'), 'call', false)(J$.R(9089, 'objectContext', objectContext, false, false)))))
                                        continue;
                                } else if (J$.C(480, J$.B(930, '===', J$.G(9113, J$.R(9105, 'part', part, false, false), 'operator'), J$.T(9121, '$comment', 21, false)))) {
                                    return J$.Rt(9137, J$.T(9129, true, 23, false));
                                } else {
                                    throw J$.F(9177, J$.I(typeof Error === 'undefined' ? Error = J$.R(9145, 'Error', undefined, true, true) : Error = J$.R(9145, 'Error', Error, true, true)), true)(J$.B(938, '+', J$.T(9153, 'Unsupported operator: ', 21, false), J$.G(9169, J$.I(typeof parts === 'undefined' ? parts = J$.R(9161, 'parts', undefined, true, true) : parts = J$.R(9161, 'parts', parts, true, true)), 'operator')));
                                }
                                return J$.Rt(9193, J$.T(9185, true, 23, false));
                            }
                            return J$.Rt(9209, J$.T(9201, false, 23, false));
                        } catch (J$e) {
                            J$.Ex(11089, J$e);
                        } finally {
                            if (J$.Fr(11097))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function valueTest(documentValue, queryOperand, compare) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(9601, arguments.callee, this, arguments);
                            arguments = J$.N(9609, 'arguments', arguments, true, false, false);
                            documentValue = J$.N(9617, 'documentValue', documentValue, true, false, false);
                            queryOperand = J$.N(9625, 'queryOperand', queryOperand, true, false, false);
                            compare = J$.N(9633, 'compare', compare, true, false, false);
                            J$.N(9641, 'test', test, false, false, false);
                            if (J$.C(584, J$.B(946, 'instanceof', J$.R(9321, 'documentValue', documentValue, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(9329, 'Array', undefined, true, true) : Array = J$.R(9329, 'Array', Array, true, true))))) {
                                if (J$.C(576, J$.B(954, 'instanceof', J$.R(9337, 'queryOperand', queryOperand, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(9345, 'Array', undefined, true, true) : Array = J$.R(9345, 'Array', Array, true, true))))) {
                                    if (J$.C(552, J$.U(962, '!', J$.F(9377, J$.R(9353, 'compare', compare, false, false), false)(J$.R(9361, 'documentValue', documentValue, false, false), J$.R(9369, 'queryOperand', queryOperand, false, false)))))
                                        return J$.Rt(9393, J$.T(9385, false, 23, false));
                                } else {
                                    var test = J$.W(9513, 'test', J$.M(9505, J$.R(9401, 'documentValue', documentValue, false, false), 'reduce', false)(J$.T(9489, function (last, cur) {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(9457, arguments.callee, this, arguments);
                                                    arguments = J$.N(9465, 'arguments', arguments, true, false, false);
                                                    last = J$.N(9473, 'last', last, true, false, false);
                                                    cur = J$.N(9481, 'cur', cur, true, false, false);
                                                    return J$.Rt(9449, J$.C(560, J$.R(9409, 'last', last, false, false)) ? J$._() : J$.F(9441, J$.R(9417, 'compare', compare, false, false), false)(J$.R(9425, 'cur', cur, false, false), J$.R(9433, 'queryOperand', queryOperand, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(11105, J$e);
                                                } finally {
                                                    if (J$.Fr(11113))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false), J$.T(9497, false, 23, false)), test, false, false);
                                    if (J$.C(568, J$.U(970, '!', J$.R(9521, 'test', test, false, false))))
                                        return J$.Rt(9537, J$.T(9529, false, 23, false));
                                }
                            } else {
                                return J$.Rt(9577, J$.F(9569, J$.R(9545, 'compare', compare, false, false), false)(J$.R(9553, 'documentValue', documentValue, false, false), J$.R(9561, 'queryOperand', queryOperand, false, false)));
                            }
                            return J$.Rt(9593, J$.T(9585, true, 23, false));
                        } catch (J$e) {
                            J$.Ex(11121, J$e);
                        } finally {
                            if (J$.Fr(11129))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function mongoEqual(documentValue, queryOperand) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(10409, arguments.callee, this, arguments);
                            arguments = J$.N(10417, 'arguments', arguments, true, false, false);
                            documentValue = J$.N(10425, 'documentValue', documentValue, true, false, false);
                            queryOperand = J$.N(10433, 'queryOperand', queryOperand, true, false, false);
                            J$.N(10441, 'aKeys', aKeys, false, false, false);
                            J$.N(10449, 'bKeys', bKeys, false, false, false);
                            J$.N(10457, 'n', n, false, false, false);
                            J$.N(10465, 'key', key, false, false, false);
                            J$.N(10473, 'aVal', aVal, false, false, false);
                            J$.N(10481, 'bVal', bVal, false, false, false);
                            if (J$.C(680, J$.B(978, 'instanceof', J$.R(9649, 'documentValue', documentValue, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(9657, 'Array', undefined, true, true) : Array = J$.R(9657, 'Array', Array, true, true))))) {
                                if (J$.C(592, J$.U(994, '!', J$.B(986, 'instanceof', J$.R(9665, 'queryOperand', queryOperand, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(9673, 'Array', undefined, true, true) : Array = J$.R(9673, 'Array', Array, true, true))))))
                                    return J$.Rt(9689, J$.T(9681, false, 23, false));
                                if (J$.C(608, J$.B(1002, '!==', J$.G(9705, J$.R(9697, 'documentValue', documentValue, false, false), 'length'), J$.G(9721, J$.R(9713, 'queryOperand', queryOperand, false, false), 'length')))) {
                                    return J$.Rt(9737, J$.T(9729, false, 23, false));
                                } else {
                                    return J$.Rt(9881, J$.M(9873, J$.R(9745, 'documentValue', documentValue, false, false), 'reduce', false)(J$.T(9857, function (previousValue, currentValue, index) {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(9817, arguments.callee, this, arguments);
                                                    arguments = J$.N(9825, 'arguments', arguments, true, false, false);
                                                    previousValue = J$.N(9833, 'previousValue', previousValue, true, false, false);
                                                    currentValue = J$.N(9841, 'currentValue', currentValue, true, false, false);
                                                    index = J$.N(9849, 'index', index, true, false, false);
                                                    return J$.Rt(9809, J$.C(600, J$.R(9753, 'previousValue', previousValue, false, false)) ? J$.F(9801, J$.R(9761, 'mongoEqual', mongoEqual, false, true), false)(J$.R(9769, 'currentValue', currentValue, false, false), J$.G(9793, J$.R(9777, 'queryOperand', queryOperand, false, false), J$.R(9785, 'index', index, false, false))) : J$._());
                                                } catch (J$e) {
                                                    J$.Ex(11137, J$e);
                                                } finally {
                                                    if (J$.Fr(11145))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false), J$.T(9865, true, 23, false)));
                                }
                            } else if (J$.C(672, J$.B(1010, 'instanceof', J$.R(9889, 'documentValue', documentValue, false, false), J$.I(typeof Object === 'undefined' ? Object = J$.R(9897, 'Object', undefined, true, true) : Object = J$.R(9897, 'Object', Object, true, true))))) {
                                if (J$.C(616, J$.U(1026, '!', J$.B(1018, 'instanceof', J$.R(9905, 'queryOperand', queryOperand, false, false), J$.I(typeof Object === 'undefined' ? Object = J$.R(9913, 'Object', undefined, true, true) : Object = J$.R(9913, 'Object', Object, true, true))))))
                                    return J$.Rt(9929, J$.T(9921, false, 23, false));
                                var aKeys = J$.W(9961, 'aKeys', J$.M(9953, J$.I(typeof Object === 'undefined' ? Object = J$.R(9937, 'Object', undefined, true, true) : Object = J$.R(9937, 'Object', Object, true, true)), 'keys', false)(J$.R(9945, 'documentValue', documentValue, false, false)), aKeys, false, false);
                                var bKeys = J$.W(9993, 'bKeys', J$.M(9985, J$.I(typeof Object === 'undefined' ? Object = J$.R(9969, 'Object', undefined, true, true) : Object = J$.R(9969, 'Object', Object, true, true)), 'keys', false)(J$.R(9977, 'queryOperand', queryOperand, false, false)), bKeys, false, false);
                                if (J$.C(648, J$.B(1034, '!==', J$.G(10009, J$.R(10001, 'aKeys', aKeys, false, false), 'length'), J$.G(10025, J$.R(10017, 'bKeys', bKeys, false, false), 'length')))) {
                                    return J$.Rt(10041, J$.T(10033, false, 23, false));
                                } else {
                                    for (var n = J$.W(10057, 'n', J$.T(10049, 0, 22, false), n, false, false); J$.C(640, J$.B(1042, '<', J$.R(10065, 'n', n, false, false), J$.G(10081, J$.R(10073, 'aKeys', aKeys, false, false), 'length'))); J$.B(1066, '-', n = J$.W(10097, 'n', J$.B(1058, '+', J$.U(1050, '+', J$.R(10089, 'n', n, false, false)), 1), n, false, false), 1)) {
                                        if (J$.C(624, J$.B(1074, '!==', J$.G(10121, J$.R(10105, 'aKeys', aKeys, false, false), J$.R(10113, 'n', n, false, false)), J$.G(10145, J$.R(10129, 'bKeys', bKeys, false, false), J$.R(10137, 'n', n, false, false)))))
                                            return J$.Rt(10161, J$.T(10153, false, 23, false));
                                        var key = J$.W(10193, 'key', J$.G(10185, J$.R(10169, 'aKeys', aKeys, false, false), J$.R(10177, 'n', n, false, false)), key, false, false);
                                        var aVal = J$.W(10225, 'aVal', J$.G(10217, J$.R(10201, 'documentValue', documentValue, false, false), J$.R(10209, 'key', key, false, false)), aVal, false, false);
                                        var bVal = J$.W(10257, 'bVal', J$.G(10249, J$.R(10233, 'queryOperand', queryOperand, false, false), J$.R(10241, 'key', key, false, false)), bVal, false, false);
                                        if (J$.C(632, J$.U(1082, '!', J$.F(10289, J$.R(10265, 'mongoEqual', mongoEqual, false, true), false)(J$.R(10273, 'aVal', aVal, false, false), J$.R(10281, 'bVal', bVal, false, false))))) {
                                            return J$.Rt(10305, J$.T(10297, false, 23, false));
                                        }
                                    }
                                    return J$.Rt(10321, J$.T(10313, true, 23, false));
                                }
                            } else {
                                if (J$.C(664, J$.B(1090, '===', J$.R(10329, 'queryOperand', queryOperand, false, false), J$.T(10337, null, 25, false)))) {
                                    return J$.Rt(10377, J$.C(656, J$.B(1098, '===', J$.R(10345, 'documentValue', documentValue, false, false), J$.T(10353, undefined, 24, false))) ? J$._() : J$.B(1106, '===', J$.R(10361, 'documentValue', documentValue, false, false), J$.T(10369, null, 25, false)));
                                } else {
                                    return J$.Rt(10401, J$.B(1114, '===', J$.R(10385, 'documentValue', documentValue, false, false), J$.R(10393, 'queryOperand', queryOperand, false, false)));
                                }
                            }
                        } catch (J$e) {
                            J$.Ex(11153, J$e);
                        } finally {
                            if (J$.Fr(11161))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function validateDocumentObject(document) {
                jalangiLabel21:
                    while (true) {
                        try {
                            J$.Fe(10697, arguments.callee, this, arguments);
                            arguments = J$.N(10705, 'arguments', arguments, true, false, false);
                            document = J$.N(10713, 'document', document, true, false, false);
                            J$.N(10721, 'key', key, false, false, false);
                            for (var key in J$.H(10681, J$.R(10489, 'document', document, false, false))) {
                                J$.N(10689, 'key', key, false, true, false);
                                {
                                    {
                                        if (J$.C(704, J$.B(1122, '===', J$.G(10513, J$.R(10497, 'key', key, false, false), J$.T(10505, 0, 22, false)), J$.T(10521, '$', 21, false))))
                                            throw J$.F(10545, J$.I(typeof Error === 'undefined' ? Error = J$.R(10529, 'Error', undefined, true, true) : Error = J$.R(10529, 'Error', Error, true, true)), true)(J$.T(10537, 'Field names can\'t start with $', 21, false));
                                        else if (J$.C(696, J$.B(1138, '!==', J$.M(10569, J$.R(10553, 'key', key, false, false), 'indexOf', false)(J$.T(10561, '.', 21, false)), J$.U(1130, '-', J$.T(10577, 1, 22, false)))))
                                            throw J$.F(10601, J$.I(typeof Error === 'undefined' ? Error = J$.R(10585, 'Error', undefined, true, true) : Error = J$.R(10585, 'Error', Error, true, true)), true)(J$.T(10593, 'Field names can\'t contain .', 21, false));
                                        else if (J$.C(688, J$.B(1146, 'instanceof', J$.G(10625, J$.R(10609, 'document', document, false, false), J$.R(10617, 'key', key, false, false)), J$.I(typeof Object === 'undefined' ? Object = J$.R(10633, 'Object', undefined, true, true) : Object = J$.R(10633, 'Object', Object, true, true))))) {
                                            J$.F(10673, J$.R(10641, 'validateDocumentObject', validateDocumentObject, false, true), false)(J$.G(10665, J$.R(10649, 'document', document, false, false), J$.R(10657, 'key', key, false, false)));
                                        }
                                    }
                                }
                            }
                        } catch (J$e) {
                            J$.Ex(11169, J$e);
                        } finally {
                            if (J$.Fr(11177))
                                continue jalangiLabel21;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(10737, 'DotNotationPointers', DotNotationPointers, false, false, false);
            J$.N(10745, 'simpleComparators', simpleComparators, false, false, false);
            J$.N(10753, 'compoundOperatorComparators', compoundOperatorComparators, false, false, false);
            J$.N(10761, 'matches', matches, false, false, false);
            partMatches = J$.N(10777, 'partMatches', J$.T(10769, partMatches, 12, false), true, false, false);
            valueTest = J$.N(10793, 'valueTest', J$.T(10785, valueTest, 12, false), true, false, false);
            mongoEqual = J$.N(10809, 'mongoEqual', J$.T(10801, mongoEqual, 12, false), true, false, false);
            validateDocumentObject = J$.N(10825, 'validateDocumentObject', J$.T(10817, validateDocumentObject, 12, false), true, false, false);
            var DotNotationPointers = J$.W(6081, 'DotNotationPointers', J$.F(6073, J$.I(typeof require === 'undefined' ? require = J$.R(6057, 'require', undefined, true, true) : require = J$.R(6057, 'require', require, true, true)), false)(J$.T(6065, './DotNotationPointers', 21, false)), DotNotationPointers, false, true);
            var simpleComparators = J$.W(7041, 'simpleComparators', J$.T(7033, {
                $gt: J$.T(6145, function (a, b) {
                    jalangiLabel0:
                        while (true) {
                            try {
                                J$.Fe(6113, arguments.callee, this, arguments);
                                arguments = J$.N(6121, 'arguments', arguments, true, false, false);
                                a = J$.N(6129, 'a', a, true, false, false);
                                b = J$.N(6137, 'b', b, true, false, false);
                                return J$.Rt(6105, J$.B(394, '>', J$.R(6089, 'a', a, false, false), J$.R(6097, 'b', b, false, false)));
                            } catch (J$e) {
                                J$.Ex(10833, J$e);
                            } finally {
                                if (J$.Fr(10841))
                                    continue jalangiLabel0;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $gte: J$.T(6209, function (a, b) {
                    jalangiLabel1:
                        while (true) {
                            try {
                                J$.Fe(6177, arguments.callee, this, arguments);
                                arguments = J$.N(6185, 'arguments', arguments, true, false, false);
                                a = J$.N(6193, 'a', a, true, false, false);
                                b = J$.N(6201, 'b', b, true, false, false);
                                return J$.Rt(6169, J$.B(402, '>=', J$.R(6153, 'a', a, false, false), J$.R(6161, 'b', b, false, false)));
                            } catch (J$e) {
                                J$.Ex(10849, J$e);
                            } finally {
                                if (J$.Fr(10857))
                                    continue jalangiLabel1;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $lt: J$.T(6273, function (a, b) {
                    jalangiLabel2:
                        while (true) {
                            try {
                                J$.Fe(6241, arguments.callee, this, arguments);
                                arguments = J$.N(6249, 'arguments', arguments, true, false, false);
                                a = J$.N(6257, 'a', a, true, false, false);
                                b = J$.N(6265, 'b', b, true, false, false);
                                return J$.Rt(6233, J$.B(410, '<', J$.R(6217, 'a', a, false, false), J$.R(6225, 'b', b, false, false)));
                            } catch (J$e) {
                                J$.Ex(10865, J$e);
                            } finally {
                                if (J$.Fr(10873))
                                    continue jalangiLabel2;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $lte: J$.T(6337, function (a, b) {
                    jalangiLabel3:
                        while (true) {
                            try {
                                J$.Fe(6305, arguments.callee, this, arguments);
                                arguments = J$.N(6313, 'arguments', arguments, true, false, false);
                                a = J$.N(6321, 'a', a, true, false, false);
                                b = J$.N(6329, 'b', b, true, false, false);
                                return J$.Rt(6297, J$.B(418, '<=', J$.R(6281, 'a', a, false, false), J$.R(6289, 'b', b, false, false)));
                            } catch (J$e) {
                                J$.Ex(10881, J$e);
                            } finally {
                                if (J$.Fr(10889))
                                    continue jalangiLabel3;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $ne: J$.T(6401, function (a, b) {
                    jalangiLabel4:
                        while (true) {
                            try {
                                J$.Fe(6369, arguments.callee, this, arguments);
                                arguments = J$.N(6377, 'arguments', arguments, true, false, false);
                                a = J$.N(6385, 'a', a, true, false, false);
                                b = J$.N(6393, 'b', b, true, false, false);
                                return J$.Rt(6361, J$.B(426, '!==', J$.R(6345, 'a', a, false, false), J$.R(6353, 'b', b, false, false)));
                            } catch (J$e) {
                                J$.Ex(10897, J$e);
                            } finally {
                                if (J$.Fr(10905))
                                    continue jalangiLabel4;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $mod: J$.T(6505, function (docValue, operand) {
                    jalangiLabel5:
                        while (true) {
                            try {
                                J$.Fe(6473, arguments.callee, this, arguments);
                                arguments = J$.N(6481, 'arguments', arguments, true, false, false);
                                docValue = J$.N(6489, 'docValue', docValue, true, false, false);
                                operand = J$.N(6497, 'operand', operand, true, false, false);
                                return J$.Rt(6465, J$.B(442, '===', J$.B(434, '%', J$.R(6409, 'docValue', docValue, false, false), J$.G(6433, J$.R(6417, 'operand', operand, false, false), J$.T(6425, 0, 22, false))), J$.G(6457, J$.R(6441, 'operand', operand, false, false), J$.T(6449, 1, 22, false))));
                            } catch (J$e) {
                                J$.Ex(10913, J$e);
                            } finally {
                                if (J$.Fr(10921))
                                    continue jalangiLabel5;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $regex: J$.T(6617, function (docValue, operand) {
                    jalangiLabel6:
                        while (true) {
                            try {
                                J$.Fe(6585, arguments.callee, this, arguments);
                                arguments = J$.N(6593, 'arguments', arguments, true, false, false);
                                docValue = J$.N(6601, 'docValue', docValue, true, false, false);
                                operand = J$.N(6609, 'operand', operand, true, false, false);
                                return J$.Rt(6577, J$.C(264, J$.B(458, '===', J$.U(450, 'typeof', J$.R(6513, 'docValue', docValue, false, false)), J$.T(6521, 'string', 21, false))) ? J$.B(466, '!==', J$.M(6561, J$.R(6529, 'docValue', docValue, false, false), 'match', false)(J$.F(6553, J$.I(typeof RegExp === 'undefined' ? RegExp = J$.R(6537, 'RegExp', undefined, true, true) : RegExp = J$.R(6537, 'RegExp', RegExp, true, true)), false)(J$.R(6545, 'operand', operand, false, false))), J$.T(6569, null, 25, false)) : J$._());
                            } catch (J$e) {
                                J$.Ex(10929, J$e);
                            } finally {
                                if (J$.Fr(10937))
                                    continue jalangiLabel6;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $exists: J$.T(6689, function (docValue, operand) {
                    jalangiLabel7:
                        while (true) {
                            try {
                                J$.Fe(6657, arguments.callee, this, arguments);
                                arguments = J$.N(6665, 'arguments', arguments, true, false, false);
                                docValue = J$.N(6673, 'docValue', docValue, true, false, false);
                                operand = J$.N(6681, 'operand', operand, true, false, false);
                                return J$.Rt(6649, J$.B(482, '===', J$.B(474, '!==', J$.R(6625, 'docValue', docValue, false, false), J$.T(6633, undefined, 24, false)), J$.R(6641, 'operand', operand, false, false)));
                            } catch (J$e) {
                                J$.Ex(10945, J$e);
                            } finally {
                                if (J$.Fr(10953))
                                    continue jalangiLabel7;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $in: J$.T(6769, function (docVal, operand) {
                    jalangiLabel8:
                        while (true) {
                            try {
                                J$.Fe(6737, arguments.callee, this, arguments);
                                arguments = J$.N(6745, 'arguments', arguments, true, false, false);
                                docVal = J$.N(6753, 'docVal', docVal, true, false, false);
                                operand = J$.N(6761, 'operand', operand, true, false, false);
                                return J$.Rt(6729, J$.B(498, '!==', J$.M(6713, J$.R(6697, 'operand', operand, false, false), 'indexOf', false)(J$.R(6705, 'docVal', docVal, false, false)), J$.U(490, '-', J$.T(6721, 1, 22, false))));
                            } catch (J$e) {
                                J$.Ex(10961, J$e);
                            } finally {
                                if (J$.Fr(10969))
                                    continue jalangiLabel8;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $nin: J$.T(6849, function (docVal, operand) {
                    jalangiLabel9:
                        while (true) {
                            try {
                                J$.Fe(6817, arguments.callee, this, arguments);
                                arguments = J$.N(6825, 'arguments', arguments, true, false, false);
                                docVal = J$.N(6833, 'docVal', docVal, true, false, false);
                                operand = J$.N(6841, 'operand', operand, true, false, false);
                                return J$.Rt(6809, J$.B(514, '===', J$.M(6793, J$.R(6777, 'operand', operand, false, false), 'indexOf', false)(J$.R(6785, 'docVal', docVal, false, false)), J$.U(506, '-', J$.T(6801, 1, 22, false))));
                            } catch (J$e) {
                                J$.Ex(10977, J$e);
                            } finally {
                                if (J$.Fr(10985))
                                    continue jalangiLabel9;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $all: J$.T(7025, function (docVal, operand) {
                    jalangiLabel11:
                        while (true) {
                            try {
                                J$.Fe(6993, arguments.callee, this, arguments);
                                arguments = J$.N(7001, 'arguments', arguments, true, false, false);
                                docVal = J$.N(7009, 'docVal', docVal, true, false, false);
                                operand = J$.N(7017, 'operand', operand, true, false, false);
                                return J$.Rt(6985, J$.C(280, J$.B(522, 'instanceof', J$.R(6857, 'docVal', docVal, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(6865, 'Array', undefined, true, true) : Array = J$.R(6865, 'Array', Array, true, true)))) ? J$.M(6977, J$.R(6873, 'docVal', docVal, false, false), 'reduce', false)(J$.T(6961, function (last, cur) {
                                    jalangiLabel10:
                                        while (true) {
                                            try {
                                                J$.Fe(6929, arguments.callee, this, arguments);
                                                arguments = J$.N(6937, 'arguments', arguments, true, false, false);
                                                last = J$.N(6945, 'last', last, true, false, false);
                                                cur = J$.N(6953, 'cur', cur, true, false, false);
                                                return J$.Rt(6921, J$.C(272, J$.R(6881, 'last', last, false, false)) ? J$.B(538, '!==', J$.M(6905, J$.R(6889, 'operand', operand, false, false), 'indexOf', false)(J$.R(6897, 'cur', cur, false, false)), J$.U(530, '-', J$.T(6913, 1, 22, false))) : J$._());
                                            } catch (J$e) {
                                                J$.Ex(10993, J$e);
                                            } finally {
                                                if (J$.Fr(11001))
                                                    continue jalangiLabel10;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), J$.T(6969, true, 23, false)) : J$._());
                            } catch (J$e) {
                                J$.Ex(11009, J$e);
                            } finally {
                                if (J$.Fr(11017))
                                    continue jalangiLabel11;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false), simpleComparators, false, true);
            var compoundOperatorComparators = J$.W(7633, 'compoundOperatorComparators', J$.T(7625, {
                $and: J$.T(7233, function (document, parts) {
                    jalangiLabel12:
                        while (true) {
                            try {
                                J$.Fe(7193, arguments.callee, this, arguments);
                                arguments = J$.N(7201, 'arguments', arguments, true, false, false);
                                document = J$.N(7209, 'document', document, true, false, false);
                                parts = J$.N(7217, 'parts', parts, true, false, false);
                                J$.N(7225, 'n', n, false, false, false);
                                for (var n = J$.W(7057, 'n', J$.T(7049, 0, 22, false), n, false, false); J$.C(296, J$.B(546, '<', J$.R(7065, 'n', n, false, false), J$.G(7081, J$.R(7073, 'parts', parts, false, false), 'length'))); J$.B(570, '-', n = J$.W(7097, 'n', J$.B(562, '+', J$.U(554, '+', J$.R(7089, 'n', n, false, false)), 1), n, false, false), 1)) {
                                    if (J$.C(288, J$.U(578, '!', J$.F(7153, J$.R(7105, 'matches', matches, false, true), false)(J$.G(7137, J$.G(7129, J$.R(7113, 'parts', parts, false, false), J$.R(7121, 'n', n, false, false)), 'parts'), J$.R(7145, 'document', document, false, false))))) {
                                        return J$.Rt(7169, J$.T(7161, false, 23, false));
                                    }
                                }
                                return J$.Rt(7185, J$.T(7177, true, 23, false));
                            } catch (J$e) {
                                J$.Ex(11025, J$e);
                            } finally {
                                if (J$.Fr(11033))
                                    continue jalangiLabel12;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $or: J$.T(7425, function (document, parts) {
                    jalangiLabel13:
                        while (true) {
                            try {
                                J$.Fe(7385, arguments.callee, this, arguments);
                                arguments = J$.N(7393, 'arguments', arguments, true, false, false);
                                document = J$.N(7401, 'document', document, true, false, false);
                                parts = J$.N(7409, 'parts', parts, true, false, false);
                                J$.N(7417, 'n', n, false, false, false);
                                for (var n = J$.W(7249, 'n', J$.T(7241, 0, 22, false), n, false, false); J$.C(312, J$.B(586, '<', J$.R(7257, 'n', n, false, false), J$.G(7273, J$.R(7265, 'parts', parts, false, false), 'length'))); J$.B(610, '-', n = J$.W(7289, 'n', J$.B(602, '+', J$.U(594, '+', J$.R(7281, 'n', n, false, false)), 1), n, false, false), 1)) {
                                    if (J$.C(304, J$.F(7345, J$.R(7297, 'matches', matches, false, true), false)(J$.G(7329, J$.G(7321, J$.R(7305, 'parts', parts, false, false), J$.R(7313, 'n', n, false, false)), 'parts'), J$.R(7337, 'document', document, false, false)))) {
                                        return J$.Rt(7361, J$.T(7353, true, 23, false));
                                    }
                                }
                                return J$.Rt(7377, J$.T(7369, false, 23, false));
                            } catch (J$e) {
                                J$.Ex(11041, J$e);
                            } finally {
                                if (J$.Fr(11049))
                                    continue jalangiLabel13;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $nor: J$.T(7617, function (document, parts) {
                    jalangiLabel14:
                        while (true) {
                            try {
                                J$.Fe(7577, arguments.callee, this, arguments);
                                arguments = J$.N(7585, 'arguments', arguments, true, false, false);
                                document = J$.N(7593, 'document', document, true, false, false);
                                parts = J$.N(7601, 'parts', parts, true, false, false);
                                J$.N(7609, 'n', n, false, false, false);
                                for (var n = J$.W(7441, 'n', J$.T(7433, 0, 22, false), n, false, false); J$.C(328, J$.B(618, '<', J$.R(7449, 'n', n, false, false), J$.G(7465, J$.R(7457, 'parts', parts, false, false), 'length'))); J$.B(642, '-', n = J$.W(7481, 'n', J$.B(634, '+', J$.U(626, '+', J$.R(7473, 'n', n, false, false)), 1), n, false, false), 1)) {
                                    if (J$.C(320, J$.F(7537, J$.R(7489, 'matches', matches, false, true), false)(J$.G(7521, J$.G(7513, J$.R(7497, 'parts', parts, false, false), J$.R(7505, 'n', n, false, false)), 'parts'), J$.R(7529, 'document', document, false, false)))) {
                                        return J$.Rt(7553, J$.T(7545, false, 23, false));
                                    }
                                }
                                return J$.Rt(7569, J$.T(7561, true, 23, false));
                            } catch (J$e) {
                                J$.Ex(11057, J$e);
                            } finally {
                                if (J$.Fr(11065))
                                    continue jalangiLabel14;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false), compoundOperatorComparators, false, true);
            var matches = J$.W(7889, 'matches', J$.P(7881, J$.I(typeof module === 'undefined' ? module = J$.R(7641, 'module', undefined, true, true) : module = J$.R(7641, 'module', module, true, true)), 'exports', J$.T(7873, function (parts, document) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(7825, arguments.callee, this, arguments);
                            arguments = J$.N(7833, 'arguments', arguments, true, false, false);
                            parts = J$.N(7841, 'parts', parts, true, false, false);
                            document = J$.N(7849, 'document', document, true, false, false);
                            J$.N(7857, 'n', n, false, false, false);
                            J$.N(7865, 'part', part, false, false, false);
                            J$.F(7665, J$.R(7649, 'validateDocumentObject', validateDocumentObject, false, true), false)(J$.R(7657, 'document', document, false, false));
                            for (var n = J$.W(7681, 'n', J$.T(7673, 0, 22, false), n, false, false); J$.C(344, J$.B(650, '<', J$.R(7689, 'n', n, false, false), J$.G(7705, J$.R(7697, 'parts', parts, false, false), 'length'))); J$.B(674, '-', n = J$.W(7721, 'n', J$.B(666, '+', J$.U(658, '+', J$.R(7713, 'n', n, false, false)), 1), n, false, false), 1)) {
                                var part = J$.W(7753, 'part', J$.G(7745, J$.R(7729, 'parts', parts, false, false), J$.R(7737, 'n', n, false, false)), part, false, false);
                                if (J$.C(336, J$.U(682, '!', J$.F(7785, J$.R(7761, 'partMatches', partMatches, false, true), false)(J$.R(7769, 'part', part, false, false), J$.R(7777, 'document', document, false, false)))))
                                    return J$.Rt(7801, J$.T(7793, false, 23, false));
                            }
                            return J$.Rt(7817, J$.T(7809, true, 23, false));
                        } catch (J$e) {
                            J$.Ex(11073, J$e);
                        } finally {
                            if (J$.Fr(11081))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), matches, false, true);
        } catch (J$e) {
            J$.Ex(11185, J$e);
        } finally {
            if (J$.Sr(11193))
                continue jalangiLabel22;
            else
                break jalangiLabel22;
        }
    }
// JALANGI DO NOT INSTRUMENT
