// mongodb/lib/operations/add_user.js
J$.noInstrEval = false;
jalangiLabel24:
    while (true) {
        try {
            J$.Se(1064225, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/operations/add_user.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(1060177, arguments.callee, this, arguments);
                            arguments = J$.N(1060185, 'arguments', arguments, true, false, false);
                            obj = J$.N(1060193, 'obj', obj, true, false, false);
                            if (J$.C(64440, J$.C(64400, J$.B(107770, '===', J$.U(107762, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1059913, 'Symbol', undefined, true, true) : Symbol = J$.R(1059913, 'Symbol', Symbol, true, true))), J$.T(1059921, 'function', 21, false))) ? J$.B(107786, '===', J$.U(107778, 'typeof', J$.G(1059937, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1059929, 'Symbol', undefined, true, true) : Symbol = J$.R(1059929, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1059945, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(1060001, '_typeof2', J$.T(1059993, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(1059969, arguments.callee, this, arguments);
                                                arguments = J$.N(1059977, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1059985, 'obj', obj, true, false, false);
                                                return J$.Rt(1059961, J$.U(107794, 'typeof', J$.R(1059953, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1064481, J$e);
                                            } finally {
                                                if (J$.Fr(1064489))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(1060137, '_typeof2', J$.T(1060129, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(1060105, arguments.callee, this, arguments);
                                                arguments = J$.N(1060113, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1060121, 'obj', obj, true, false, false);
                                                return J$.Rt(1060097, J$.C(64432, J$.C(64424, J$.C(64416, J$.C(64408, J$.R(1060009, 'obj', obj, false, false)) ? J$.B(107810, '===', J$.U(107802, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1060017, 'Symbol', undefined, true, true) : Symbol = J$.R(1060017, 'Symbol', Symbol, true, true))), J$.T(1060025, 'function', 21, false)) : J$._()) ? J$.B(107818, '===', J$.G(1060041, J$.R(1060033, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1060049, 'Symbol', undefined, true, true) : Symbol = J$.R(1060049, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(107826, '!==', J$.R(1060057, 'obj', obj, false, false), J$.G(1060073, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1060065, 'Symbol', undefined, true, true) : Symbol = J$.R(1060065, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1060081, 'symbol', 21, false) : J$.U(107834, 'typeof', J$.R(1060089, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1064497, J$e);
                                            } finally {
                                                if (J$.Fr(1064505))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(1060169, J$.F(1060161, J$.R(1060145, '_typeof2', _typeof2, false, true), false)(J$.R(1060153, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1064513, J$e);
                        } finally {
                            if (J$.Fr(1064521))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(1060513, arguments.callee, this, arguments);
                            arguments = J$.N(1060521, 'arguments', arguments, true, false, false);
                            obj = J$.N(1060529, 'obj', obj, true, false, false);
                            if (J$.C(64488, J$.C(64448, J$.B(107850, '===', J$.U(107842, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1060201, 'Symbol', undefined, true, true) : Symbol = J$.R(1060201, 'Symbol', Symbol, true, true))), J$.T(1060209, 'function', 21, false))) ? J$.B(107858, '===', J$.F(1060241, J$.R(1060217, '_typeof2', _typeof2, false, true), false)(J$.G(1060233, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1060225, 'Symbol', undefined, true, true) : Symbol = J$.R(1060225, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1060249, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(1060321, '_typeof', J$.T(1060313, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(1060289, arguments.callee, this, arguments);
                                                arguments = J$.N(1060297, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1060305, 'obj', obj, true, false, false);
                                                return J$.Rt(1060281, J$.F(1060273, J$.R(1060257, '_typeof2', _typeof2, false, true), false)(J$.R(1060265, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1064529, J$e);
                                            } finally {
                                                if (J$.Fr(1064537))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(1060473, '_typeof', J$.T(1060465, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(1060441, arguments.callee, this, arguments);
                                                arguments = J$.N(1060449, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1060457, 'obj', obj, true, false, false);
                                                return J$.Rt(1060433, J$.C(64480, J$.C(64472, J$.C(64464, J$.C(64456, J$.R(1060329, 'obj', obj, false, false)) ? J$.B(107874, '===', J$.U(107866, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1060337, 'Symbol', undefined, true, true) : Symbol = J$.R(1060337, 'Symbol', Symbol, true, true))), J$.T(1060345, 'function', 21, false)) : J$._()) ? J$.B(107882, '===', J$.G(1060361, J$.R(1060353, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1060369, 'Symbol', undefined, true, true) : Symbol = J$.R(1060369, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(107890, '!==', J$.R(1060377, 'obj', obj, false, false), J$.G(1060393, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1060385, 'Symbol', undefined, true, true) : Symbol = J$.R(1060385, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1060401, 'symbol', 21, false) : J$.F(1060425, J$.R(1060409, '_typeof2', _typeof2, false, true), false)(J$.R(1060417, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1064545, J$e);
                                            } finally {
                                                if (J$.Fr(1064553))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(1060505, J$.F(1060497, J$.R(1060481, '_typeof', _typeof, false, true), false)(J$.R(1060489, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1064561, J$e);
                        } finally {
                            if (J$.Fr(1064569))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(1060577, arguments.callee, this, arguments);
                            arguments = J$.N(1060585, 'arguments', arguments, true, false, false);
                            instance = J$.N(1060593, 'instance', instance, true, false, false);
                            Constructor = J$.N(1060601, 'Constructor', Constructor, true, false, false);
                            if (J$.C(64496, J$.U(107906, '!', J$.B(107898, 'instanceof', J$.R(1060537, 'instance', instance, false, false), J$.R(1060545, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(1060569, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1060553, 'TypeError', undefined, true, true) : TypeError = J$.R(1060553, 'TypeError', TypeError, true, true)), true)(J$.T(1060561, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1064577, J$e);
                        } finally {
                            if (J$.Fr(1064585))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(1060849, arguments.callee, this, arguments);
                            arguments = J$.N(1060857, 'arguments', arguments, true, false, false);
                            target = J$.N(1060865, 'target', target, true, false, false);
                            props = J$.N(1060873, 'props', props, true, false, false);
                            J$.N(1060881, 'i', i, false, false, false);
                            J$.N(1060889, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(1060617, 'i', J$.T(1060609, 0, 22, false), i, false, false); J$.C(64520, J$.B(107914, '<', J$.R(1060625, 'i', i, false, false), J$.G(1060641, J$.R(1060633, 'props', props, false, false), 'length'))); J$.B(107938, '-', i = J$.W(1060657, 'i', J$.B(107930, '+', J$.U(107922, '+', J$.R(1060649, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(1060689, 'descriptor', J$.G(1060681, J$.R(1060665, 'props', props, false, false), J$.R(1060673, 'i', i, false, false)), descriptor, false, false);
                                J$.P(1060729, J$.R(1060697, 'descriptor', descriptor, false, false), 'enumerable', J$.C(64504, J$.G(1060713, J$.R(1060705, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(1060721, false, 23, false));
                                J$.P(1060753, J$.R(1060737, 'descriptor', descriptor, false, false), 'configurable', J$.T(1060745, true, 23, false));
                                if (J$.C(64512, J$.B(107946, 'in', J$.T(1060761, 'value', 21, false), J$.R(1060769, 'descriptor', descriptor, false, false))))
                                    J$.P(1060793, J$.R(1060777, 'descriptor', descriptor, false, false), 'writable', J$.T(1060785, true, 23, false));
                                J$.M(1060841, J$.I(typeof Object === 'undefined' ? Object = J$.R(1060801, 'Object', undefined, true, true) : Object = J$.R(1060801, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(1060809, 'target', target, false, false), J$.G(1060825, J$.R(1060817, 'descriptor', descriptor, false, false), 'key'), J$.R(1060833, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1064593, J$e);
                        } finally {
                            if (J$.Fr(1064601))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(1061001, arguments.callee, this, arguments);
                            arguments = J$.N(1061009, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(1061017, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(1061025, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(1061033, 'staticProps', staticProps, true, false, false);
                            if (J$.C(64528, J$.R(1060897, 'protoProps', protoProps, false, false)))
                                J$.F(1060937, J$.R(1060905, '_defineProperties', _defineProperties, false, true), false)(J$.G(1060921, J$.R(1060913, 'Constructor', Constructor, false, false), 'prototype'), J$.R(1060929, 'protoProps', protoProps, false, false));
                            if (J$.C(64536, J$.R(1060945, 'staticProps', staticProps, false, false)))
                                J$.F(1060977, J$.R(1060953, '_defineProperties', _defineProperties, false, true), false)(J$.R(1060961, 'Constructor', Constructor, false, false), J$.R(1060969, 'staticProps', staticProps, false, false));
                            return J$.Rt(1060993, J$.R(1060985, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(1064609, J$e);
                        } finally {
                            if (J$.Fr(1064617))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(1061145, arguments.callee, this, arguments);
                            arguments = J$.N(1061153, 'arguments', arguments, true, false, false);
                            self = J$.N(1061161, 'self', self, true, false, false);
                            call = J$.N(1061169, 'call', call, true, false, false);
                            if (J$.C(64560, J$.C(64552, J$.R(1061041, 'call', call, false, false)) ? J$.C(64544, J$.B(107954, '===', J$.F(1061065, J$.R(1061049, '_typeof', _typeof, false, true), false)(J$.R(1061057, 'call', call, false, false)), J$.T(1061073, 'object', 21, false))) ? J$._() : J$.B(107970, '===', J$.U(107962, 'typeof', J$.R(1061081, 'call', call, false, false)), J$.T(1061089, 'function', 21, false)) : J$._())) {
                                return J$.Rt(1061105, J$.R(1061097, 'call', call, false, false));
                            }
                            return J$.Rt(1061137, J$.F(1061129, J$.R(1061113, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(1061121, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(1064625, J$e);
                        } finally {
                            if (J$.Fr(1064633))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(1061233, arguments.callee, this, arguments);
                            arguments = J$.N(1061241, 'arguments', arguments, true, false, false);
                            self = J$.N(1061249, 'self', self, true, false, false);
                            if (J$.C(64568, J$.B(107978, '===', J$.R(1061177, 'self', self, false, false), void J$.T(1061185, 0, 22, false)))) {
                                throw J$.F(1061209, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(1061193, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(1061193, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(1061201, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(1061225, J$.R(1061217, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(1064641, J$e);
                        } finally {
                            if (J$.Fr(1064649))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(1061617, arguments.callee, this, arguments);
                            arguments = J$.N(1061625, 'arguments', arguments, true, false, false);
                            target = J$.N(1061633, 'target', target, true, false, false);
                            property = J$.N(1061641, 'property', property, true, false, false);
                            receiver = J$.N(1061649, 'receiver', receiver, true, false, false);
                            if (J$.C(64600, J$.C(64576, J$.B(107994, '!==', J$.U(107986, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1061257, 'Reflect', undefined, true, true) : Reflect = J$.R(1061257, 'Reflect', Reflect, true, true))), J$.T(1061265, 'undefined', 21, false))) ? J$.G(1061281, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1061273, 'Reflect', undefined, true, true) : Reflect = J$.R(1061273, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(1061305, '_get', J$.G(1061297, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1061289, 'Reflect', undefined, true, true) : Reflect = J$.R(1061289, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(1061553, '_get', J$.T(1061545, function _get(target, property, receiver) {
                                    jalangiLabel11:
                                        while (true) {
                                            try {
                                                J$.Fe(1061489, arguments.callee, this, arguments);
                                                arguments = J$.N(1061497, 'arguments', arguments, true, false, false);
                                                target = J$.N(1061505, 'target', target, true, false, false);
                                                property = J$.N(1061513, 'property', property, true, false, false);
                                                receiver = J$.N(1061521, 'receiver', receiver, true, false, false);
                                                J$.N(1061529, 'base', base, false, false, false);
                                                J$.N(1061537, 'desc', desc, false, false, false);
                                                var base = J$.W(1061345, 'base', J$.F(1061337, J$.R(1061313, '_superPropBase', _superPropBase, false, true), false)(J$.R(1061321, 'target', target, false, false), J$.R(1061329, 'property', property, false, false)), base, false, false);
                                                if (J$.C(64584, J$.U(108002, '!', J$.R(1061353, 'base', base, false, false))))
                                                    return J$.Rt(1061361, undefined);
                                                var desc = J$.W(1061401, 'desc', J$.M(1061393, J$.I(typeof Object === 'undefined' ? Object = J$.R(1061369, 'Object', undefined, true, true) : Object = J$.R(1061369, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(1061377, 'base', base, false, false), J$.R(1061385, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(64592, J$.G(1061417, J$.R(1061409, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(1061457, J$.M(1061449, J$.G(1061433, J$.R(1061425, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(1061441, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(1061481, J$.G(1061473, J$.R(1061465, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(1064657, J$e);
                                            } finally {
                                                if (J$.Fr(1064665))
                                                    continue jalangiLabel11;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(1061609, J$.F(1061601, J$.R(1061561, '_get', _get, false, true), false)(J$.R(1061569, 'target', target, false, false), J$.R(1061577, 'property', property, false, false), J$.C(64608, J$.R(1061585, 'receiver', receiver, false, false)) ? J$._() : J$.R(1061593, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(1064673, J$e);
                        } finally {
                            if (J$.Fr(1064681))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(1061769, arguments.callee, this, arguments);
                            arguments = J$.N(1061777, 'arguments', arguments, true, false, false);
                            object = J$.N(1061785, 'object', object, true, false, false);
                            property = J$.N(1061793, 'property', property, true, false, false);
                            while (J$.C(64624, J$.U(108010, '!', J$.M(1061697, J$.G(1061673, J$.G(1061665, J$.I(typeof Object === 'undefined' ? Object = J$.R(1061657, 'Object', undefined, true, true) : Object = J$.R(1061657, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(1061681, 'object', object, false, false), J$.R(1061689, 'property', property, false, false))))) {
                                object = J$.W(1061729, 'object', J$.F(1061721, J$.R(1061705, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1061713, 'object', object, false, false)), object, false, false);
                                if (J$.C(64616, J$.B(108018, '===', J$.R(1061737, 'object', object, false, false), J$.T(1061745, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(1061761, J$.R(1061753, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(1064689, J$e);
                        } finally {
                            if (J$.Fr(1064697))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(1061953, arguments.callee, this, arguments);
                            arguments = J$.N(1061961, 'arguments', arguments, true, false, false);
                            o = J$.N(1061969, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(1061913, '_getPrototypeOf', J$.C(64640, J$.G(1061809, J$.I(typeof Object === 'undefined' ? Object = J$.R(1061801, 'Object', undefined, true, true) : Object = J$.R(1061801, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(1061825, J$.I(typeof Object === 'undefined' ? Object = J$.R(1061817, 'Object', undefined, true, true) : Object = J$.R(1061817, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(1061905, function _getPrototypeOf(o) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(1061881, arguments.callee, this, arguments);
                                            arguments = J$.N(1061889, 'arguments', arguments, true, false, false);
                                            o = J$.N(1061897, 'o', o, true, false, false);
                                            return J$.Rt(1061873, J$.C(64632, J$.G(1061841, J$.R(1061833, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(1061865, J$.I(typeof Object === 'undefined' ? Object = J$.R(1061849, 'Object', undefined, true, true) : Object = J$.R(1061849, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(1061857, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(1064705, J$e);
                                        } finally {
                                            if (J$.Fr(1064713))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(1061945, J$.F(1061937, J$.R(1061921, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1061929, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(1064721, J$e);
                        } finally {
                            if (J$.Fr(1064729))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(1062169, arguments.callee, this, arguments);
                            arguments = J$.N(1062177, 'arguments', arguments, true, false, false);
                            subClass = J$.N(1062185, 'subClass', subClass, true, false, false);
                            superClass = J$.N(1062193, 'superClass', superClass, true, false, false);
                            if (J$.C(64656, J$.C(64648, J$.B(108034, '!==', J$.U(108026, 'typeof', J$.R(1061977, 'superClass', superClass, false, false)), J$.T(1061985, 'function', 21, false))) ? J$.B(108042, '!==', J$.R(1061993, 'superClass', superClass, false, false), J$.T(1062001, null, 25, false)) : J$._())) {
                                throw J$.F(1062025, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1062009, 'TypeError', undefined, true, true) : TypeError = J$.R(1062009, 'TypeError', TypeError, true, true)), true)(J$.T(1062017, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(1062121, J$.R(1062033, 'subClass', subClass, false, false), 'prototype', J$.M(1062113, J$.I(typeof Object === 'undefined' ? Object = J$.R(1062041, 'Object', undefined, true, true) : Object = J$.R(1062041, 'Object', Object, true, true)), 'create', false)(J$.C(64664, J$.R(1062049, 'superClass', superClass, false, false)) ? J$.G(1062065, J$.R(1062057, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(1062105, {
                                constructor: J$.T(1062097, {
                                    value: J$.R(1062073, 'subClass', subClass, false, false),
                                    writable: J$.T(1062081, true, 23, false),
                                    configurable: J$.T(1062089, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(64672, J$.R(1062129, 'superClass', superClass, false, false)))
                                J$.F(1062161, J$.R(1062137, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1062145, 'subClass', subClass, false, false), J$.R(1062153, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(1064737, J$e);
                        } finally {
                            if (J$.Fr(1064745))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(1062345, arguments.callee, this, arguments);
                            arguments = J$.N(1062353, 'arguments', arguments, true, false, false);
                            o = J$.N(1062361, 'o', o, true, false, false);
                            p = J$.N(1062369, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(1062297, '_setPrototypeOf', J$.C(64680, J$.G(1062209, J$.I(typeof Object === 'undefined' ? Object = J$.R(1062201, 'Object', undefined, true, true) : Object = J$.R(1062201, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(1062289, function _setPrototypeOf(o, p) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(1062257, arguments.callee, this, arguments);
                                            arguments = J$.N(1062265, 'arguments', arguments, true, false, false);
                                            o = J$.N(1062273, 'o', o, true, false, false);
                                            p = J$.N(1062281, 'p', p, true, false, false);
                                            J$.P(1062233, J$.R(1062217, 'o', o, false, false), '__proto__', J$.R(1062225, 'p', p, false, false));
                                            return J$.Rt(1062249, J$.R(1062241, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1064753, J$e);
                                        } finally {
                                            if (J$.Fr(1064761))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(1062337, J$.F(1062329, J$.R(1062305, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1062313, 'o', o, false, false), J$.R(1062321, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(1064769, J$e);
                        } finally {
                            if (J$.Fr(1064777))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(1064241, '_typeof2', J$.T(1064233, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(1064257, '_typeof', J$.T(1064249, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(1064273, '_classCallCheck', J$.T(1064265, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(1064289, '_defineProperties', J$.T(1064281, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(1064305, '_createClass', J$.T(1064297, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(1064321, '_possibleConstructorReturn', J$.T(1064313, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(1064337, '_assertThisInitialized', J$.T(1064329, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(1064353, '_get', J$.T(1064345, _get, 12, false), true, false, false);
            _superPropBase = J$.N(1064369, '_superPropBase', J$.T(1064361, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(1064385, '_getPrototypeOf', J$.T(1064377, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(1064401, '_inherits', J$.T(1064393, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(1064417, '_setPrototypeOf', J$.T(1064409, _setPrototypeOf, 12, false), true, false, false);
            J$.N(1064425, 'Aspect', Aspect, false, false, false);
            J$.N(1064433, 'CommandOperation', CommandOperation, false, false, false);
            J$.N(1064441, 'defineAspects', defineAspects, false, false, false);
            J$.N(1064449, 'crypto', crypto, false, false, false);
            J$.N(1064457, 'handleCallback', handleCallback, false, false, false);
            J$.N(1064465, 'toError', toError, false, false, false);
            J$.N(1064473, 'AddUserOperation', AddUserOperation, false, false, false);
            J$.T(1059905, 'use strict', 21, false);
            var Aspect = J$.W(1062409, 'Aspect', J$.G(1062401, J$.F(1062393, J$.I(typeof require === 'undefined' ? require = J$.R(1062377, 'require', undefined, true, true) : require = J$.R(1062377, 'require', require, true, true)), false)(J$.T(1062385, './operation', 21, false)), 'Aspect'), Aspect, false, true);
            var CommandOperation = J$.W(1062441, 'CommandOperation', J$.F(1062433, J$.I(typeof require === 'undefined' ? require = J$.R(1062417, 'require', undefined, true, true) : require = J$.R(1062417, 'require', require, true, true)), false)(J$.T(1062425, './command', 21, false)), CommandOperation, false, true);
            var defineAspects = J$.W(1062481, 'defineAspects', J$.G(1062473, J$.F(1062465, J$.I(typeof require === 'undefined' ? require = J$.R(1062449, 'require', undefined, true, true) : require = J$.R(1062449, 'require', require, true, true)), false)(J$.T(1062457, './operation', 21, false)), 'defineAspects'), defineAspects, false, true);
            var crypto = J$.W(1062513, 'crypto', J$.F(1062505, J$.I(typeof require === 'undefined' ? require = J$.R(1062489, 'require', undefined, true, true) : require = J$.R(1062489, 'require', require, true, true)), false)(J$.T(1062497, 'crypto', 21, false)), crypto, false, true);
            var handleCallback = J$.W(1062553, 'handleCallback', J$.G(1062545, J$.F(1062537, J$.I(typeof require === 'undefined' ? require = J$.R(1062521, 'require', undefined, true, true) : require = J$.R(1062521, 'require', require, true, true)), false)(J$.T(1062529, '../utils', 21, false)), 'handleCallback'), handleCallback, false, true);
            var toError = J$.W(1062593, 'toError', J$.G(1062585, J$.F(1062577, J$.I(typeof require === 'undefined' ? require = J$.R(1062561, 'require', undefined, true, true) : require = J$.R(1062561, 'require', require, true, true)), false)(J$.T(1062569, '../utils', 21, false)), 'toError'), toError, false, true);
            var AddUserOperation = J$.W(1064153, 'AddUserOperation', J$.F(1064145, J$.T(1064129, function (_CommandOperation) {
                jalangiLabel23:
                    while (true) {
                        try {
                            J$.Fe(1064089, arguments.callee, this, arguments);
                            function AddUserOperation(db, username, password, options) {
                                jalangiLabel19:
                                    while (true) {
                                        try {
                                            J$.Fe(1062817, arguments.callee, this, arguments);
                                            arguments = J$.N(1062825, 'arguments', arguments, true, false, false);
                                            db = J$.N(1062833, 'db', db, true, false, false);
                                            username = J$.N(1062841, 'username', username, true, false, false);
                                            password = J$.N(1062849, 'password', password, true, false, false);
                                            options = J$.N(1062857, 'options', options, true, false, false);
                                            J$.N(1062865, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(1062657, J$.R(1062633, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(1062641, 'this', this, false, false), J$.R(1062649, 'AddUserOperation', AddUserOperation, false, false));
                                            _this = J$.W(1062745, '_this', J$.F(1062737, J$.R(1062665, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(1062673, 'this', this, false, false), J$.M(1062729, J$.F(1062697, J$.R(1062681, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1062689, 'AddUserOperation', AddUserOperation, false, false)), 'call', false)(J$.R(1062705, 'this', this, false, false), J$.R(1062713, 'db', db, false, false), J$.R(1062721, 'options', options, false, false))), _this, false, false);
                                            J$.P(1062769, J$.R(1062753, '_this', _this, false, false), 'username', J$.R(1062761, 'username', username, false, false));
                                            J$.P(1062793, J$.R(1062777, '_this', _this, false, false), 'password', J$.R(1062785, 'password', password, false, false));
                                            return J$.Rt(1062809, J$.R(1062801, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1064785, J$e);
                                        } finally {
                                            if (J$.Fr(1064793))
                                                continue jalangiLabel19;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(1064097, 'arguments', arguments, true, false, false);
                            _CommandOperation = J$.N(1064105, '_CommandOperation', _CommandOperation, true, false, false);
                            AddUserOperation = J$.N(1064121, 'AddUserOperation', J$.T(1064113, AddUserOperation, 12, false), true, false, false);
                            J$.F(1062625, J$.R(1062601, '_inherits', _inherits, false, true), false)(J$.R(1062609, 'AddUserOperation', AddUserOperation, false, false), J$.R(1062617, '_CommandOperation', _CommandOperation, false, false));
                            J$.F(1064065, J$.R(1062873, '_createClass', _createClass, false, true), false)(J$.R(1062881, 'AddUserOperation', AddUserOperation, false, false), J$.T(1064057, [
                                J$.T(1063673, {
                                    key: J$.T(1062889, '_buildCommand', 21, false),
                                    value: J$.T(1063665, function _buildCommand() {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(1063577, arguments.callee, this, arguments);
                                                    arguments = J$.N(1063585, 'arguments', arguments, true, false, false);
                                                    J$.N(1063593, 'db', db, false, false, false);
                                                    J$.N(1063601, 'username', username, false, false, false);
                                                    J$.N(1063609, 'password', password, false, false, false);
                                                    J$.N(1063617, 'options', options, false, false, false);
                                                    J$.N(1063625, 'roles', roles, false, false, false);
                                                    J$.N(1063633, 'digestPassword', digestPassword, false, false, false);
                                                    J$.N(1063641, 'userPassword', userPassword, false, false, false);
                                                    J$.N(1063649, 'md5', md5, false, false, false);
                                                    J$.N(1063657, 'command', command, false, false, false);
                                                    var db = J$.W(1062913, 'db', J$.G(1062905, J$.R(1062897, 'this', this, false, false), 'db'), db, false, false);
                                                    var username = J$.W(1062937, 'username', J$.G(1062929, J$.R(1062921, 'this', this, false, false), 'username'), username, false, false);
                                                    var password = J$.W(1062961, 'password', J$.G(1062953, J$.R(1062945, 'this', this, false, false), 'password'), password, false, false);
                                                    var options = J$.W(1062985, 'options', J$.G(1062977, J$.R(1062969, 'this', this, false, false), 'options'), options, false, false);
                                                    var roles = J$.W(1063049, 'roles', J$.C(64688, J$.M(1063017, J$.I(typeof Array === 'undefined' ? Array = J$.R(1062993, 'Array', undefined, true, true) : Array = J$.R(1062993, 'Array', Array, true, true)), 'isArray', false)(J$.G(1063009, J$.R(1063001, 'options', options, false, false), 'roles'))) ? J$.G(1063033, J$.R(1063025, 'options', options, false, false), 'roles') : J$.T(1063041, [], 10, false), roles, false, false);
                                                    if (J$.C(64696, J$.B(108050, '===', J$.G(1063065, J$.R(1063057, 'roles', roles, false, false), 'length'), J$.T(1063073, 0, 22, false)))) {
                                                        J$.M(1063097, J$.I(typeof console === 'undefined' ? console = J$.R(1063081, 'console', undefined, true, true) : console = J$.R(1063081, 'console', console, true, true)), 'log', false)(J$.T(1063089, 'Creating a user without roles is deprecated in MongoDB >= 2.6', 21, false));
                                                    }
                                                    if (J$.C(64728, J$.C(64712, J$.C(64704, J$.B(108058, '===', J$.M(1063121, J$.G(1063113, J$.R(1063105, 'db', db, false, false), 'databaseName'), 'toLowerCase', false)(), J$.T(1063129, 'admin', 21, false))) ? J$._() : J$.B(108066, '===', J$.G(1063145, J$.R(1063137, 'options', options, false, false), 'dbName'), J$.T(1063153, 'admin', 21, false))) ? J$.U(108074, '!', J$.M(1063185, J$.I(typeof Array === 'undefined' ? Array = J$.R(1063161, 'Array', undefined, true, true) : Array = J$.R(1063161, 'Array', Array, true, true)), 'isArray', false)(J$.G(1063177, J$.R(1063169, 'options', options, false, false), 'roles'))) : J$._())) {
                                                        roles = J$.W(1063209, 'roles', J$.T(1063201, [J$.T(1063193, 'root', 21, false)], 10, false), roles, false, false);
                                                    } else if (J$.C(64720, J$.U(108082, '!', J$.M(1063241, J$.I(typeof Array === 'undefined' ? Array = J$.R(1063217, 'Array', undefined, true, true) : Array = J$.R(1063217, 'Array', Array, true, true)), 'isArray', false)(J$.G(1063233, J$.R(1063225, 'options', options, false, false), 'roles'))))) {
                                                        roles = J$.W(1063265, 'roles', J$.T(1063257, [J$.T(1063249, 'dbOwner', 21, false)], 10, false), roles, false, false);
                                                    }
                                                    var digestPassword = J$.W(1063321, 'digestPassword', J$.B(108090, '>=', J$.G(1063305, J$.M(1063297, J$.G(1063289, J$.G(1063281, J$.R(1063273, 'db', db, false, false), 's'), 'topology'), 'lastIsMaster', false)(), 'maxWireVersion'), J$.T(1063313, 7, 22, false)), digestPassword, false, false);
                                                    var userPassword = J$.W(1063337, 'userPassword', J$.R(1063329, 'password', password, false, false), userPassword, false, false);
                                                    if (J$.C(64736, J$.U(108098, '!', J$.R(1063345, 'digestPassword', digestPassword, false, false)))) {
                                                        var md5 = J$.W(1063377, 'md5', J$.M(1063369, J$.R(1063353, 'crypto', crypto, false, true), 'createHash', false)(J$.T(1063361, 'md5', 21, false)), md5, false, false);
                                                        J$.M(1063417, J$.R(1063385, 'md5', md5, false, false), 'update', false)(J$.B(108114, '+', J$.B(108106, '+', J$.R(1063393, 'username', username, false, false), J$.T(1063401, ':mongo:', 21, false)), J$.R(1063409, 'password', password, false, false)));
                                                        userPassword = J$.W(1063449, 'userPassword', J$.M(1063441, J$.R(1063425, 'md5', md5, false, false), 'digest', false)(J$.T(1063433, 'hex', 21, false)), userPassword, false, false);
                                                    }
                                                    var command = J$.W(1063513, 'command', J$.T(1063505, {
                                                        createUser: J$.R(1063457, 'username', username, false, false),
                                                        customData: J$.C(64744, J$.G(1063473, J$.R(1063465, 'options', options, false, false), 'customData')) ? J$._() : J$.T(1063481, {}, 11, false),
                                                        roles: J$.R(1063489, 'roles', roles, false, false),
                                                        digestPassword: J$.R(1063497, 'digestPassword', digestPassword, false, false)
                                                    }, 11, false), command, false, false);
                                                    if (J$.C(64752, J$.B(108130, '===', J$.U(108122, 'typeof', J$.R(1063521, 'password', password, false, false)), J$.T(1063529, 'string', 21, false)))) {
                                                        J$.P(1063553, J$.R(1063537, 'command', command, false, false), 'pwd', J$.R(1063545, 'userPassword', userPassword, false, false));
                                                    }
                                                    return J$.Rt(1063569, J$.R(1063561, 'command', command, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1064801, J$e);
                                                } finally {
                                                    if (J$.Fr(1064809))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1064049, {
                                    key: J$.T(1063681, 'execute', 21, false),
                                    value: J$.T(1064041, function execute(callback) {
                                        jalangiLabel22:
                                            while (true) {
                                                try {
                                                    J$.Fe(1064009, arguments.callee, this, arguments);
                                                    arguments = J$.N(1064017, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(1064025, 'callback', callback, true, false, false);
                                                    J$.N(1064033, 'options', options, false, false, false);
                                                    var options = J$.W(1063705, 'options', J$.G(1063697, J$.R(1063689, 'this', this, false, false), 'options'), options, false, false);
                                                    if (J$.C(64760, J$.B(108138, '!=', J$.G(1063721, J$.R(1063713, 'options', options, false, false), 'digestPassword'), J$.T(1063729, null, 25, false)))) {
                                                        return J$.Rt(1063777, J$.F(1063769, J$.R(1063737, 'callback', callback, false, false), false)(J$.F(1063761, J$.R(1063745, 'toError', toError, false, true), false)(J$.T(1063753, 'The digestPassword option is not supported via add_user. Please use db.command(\'createUser\', ...) instead for this option.', 21, false))));
                                                    }
                                                    J$.M(1064001, J$.F(1063841, J$.R(1063785, '_get', _get, false, true), false)(J$.F(1063817, J$.R(1063793, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(1063809, J$.R(1063801, 'AddUserOperation', AddUserOperation, false, false), 'prototype')), J$.T(1063825, 'execute', 21, false), J$.R(1063833, 'this', this, false, false)), 'call', false)(J$.R(1063849, 'this', this, false, false), J$.T(1063993, function (err, r) {
                                                        jalangiLabel21:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(1063961, arguments.callee, this, arguments);
                                                                    arguments = J$.N(1063969, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(1063977, 'err', err, true, false, false);
                                                                    r = J$.N(1063985, 'r', r, true, false, false);
                                                                    if (J$.C(64768, J$.U(108146, '!', J$.R(1063857, 'err', err, false, false)))) {
                                                                        return J$.Rt(1063905, J$.F(1063897, J$.R(1063865, 'handleCallback', handleCallback, false, true), false)(J$.R(1063873, 'callback', callback, false, false), J$.R(1063881, 'err', err, false, false), J$.R(1063889, 'r', r, false, false)));
                                                                    }
                                                                    return J$.Rt(1063953, J$.F(1063945, J$.R(1063913, 'handleCallback', handleCallback, false, true), false)(J$.R(1063921, 'callback', callback, false, false), J$.R(1063929, 'err', err, false, false), J$.T(1063937, null, 25, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(1064817, J$e);
                                                                } finally {
                                                                    if (J$.Fr(1064825))
                                                                        continue jalangiLabel21;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(1064833, J$e);
                                                } finally {
                                                    if (J$.Fr(1064841))
                                                        continue jalangiLabel22;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(1064081, J$.R(1064073, 'AddUserOperation', AddUserOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(1064849, J$e);
                        } finally {
                            if (J$.Fr(1064857))
                                continue jalangiLabel23;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(1064137, 'CommandOperation', CommandOperation, false, true)), AddUserOperation, false, true);
            J$.F(1064193, J$.R(1064161, 'defineAspects', defineAspects, false, true), false)(J$.R(1064169, 'AddUserOperation', AddUserOperation, false, true), J$.G(1064185, J$.R(1064177, 'Aspect', Aspect, false, true), 'WRITE_OPERATION'));
            J$.P(1064217, J$.I(typeof module === 'undefined' ? module = J$.R(1064201, 'module', undefined, true, true) : module = J$.R(1064201, 'module', module, true, true)), 'exports', J$.R(1064209, 'AddUserOperation', AddUserOperation, false, true));
        } catch (J$e) {
            J$.Ex(1064865, J$e);
        } finally {
            if (J$.Sr(1064873))
                continue jalangiLabel24;
            else
                break jalangiLabel24;
        }
    }
// JALANGI DO NOT INSTRUMENT
