// mongodb/lib/operations/aggregate.js
J$.noInstrEval = false;
jalangiLabel24:
    while (true) {
        try {
            J$.Se(1071081, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/operations/aggregate.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(1066617, arguments.callee, this, arguments);
                            arguments = J$.N(1066625, 'arguments', arguments, true, false, false);
                            obj = J$.N(1066633, 'obj', obj, true, false, false);
                            if (J$.C(64904, J$.C(64864, J$.B(108282, '===', J$.U(108274, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1066353, 'Symbol', undefined, true, true) : Symbol = J$.R(1066353, 'Symbol', Symbol, true, true))), J$.T(1066361, 'function', 21, false))) ? J$.B(108298, '===', J$.U(108290, 'typeof', J$.G(1066377, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1066369, 'Symbol', undefined, true, true) : Symbol = J$.R(1066369, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1066385, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(1066441, '_typeof2', J$.T(1066433, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(1066409, arguments.callee, this, arguments);
                                                arguments = J$.N(1066417, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1066425, 'obj', obj, true, false, false);
                                                return J$.Rt(1066401, J$.U(108306, 'typeof', J$.R(1066393, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1071353, J$e);
                                            } finally {
                                                if (J$.Fr(1071361))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(1066577, '_typeof2', J$.T(1066569, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(1066545, arguments.callee, this, arguments);
                                                arguments = J$.N(1066553, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1066561, 'obj', obj, true, false, false);
                                                return J$.Rt(1066537, J$.C(64896, J$.C(64888, J$.C(64880, J$.C(64872, J$.R(1066449, 'obj', obj, false, false)) ? J$.B(108322, '===', J$.U(108314, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1066457, 'Symbol', undefined, true, true) : Symbol = J$.R(1066457, 'Symbol', Symbol, true, true))), J$.T(1066465, 'function', 21, false)) : J$._()) ? J$.B(108330, '===', J$.G(1066481, J$.R(1066473, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1066489, 'Symbol', undefined, true, true) : Symbol = J$.R(1066489, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(108338, '!==', J$.R(1066497, 'obj', obj, false, false), J$.G(1066513, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1066505, 'Symbol', undefined, true, true) : Symbol = J$.R(1066505, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1066521, 'symbol', 21, false) : J$.U(108346, 'typeof', J$.R(1066529, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1071369, J$e);
                                            } finally {
                                                if (J$.Fr(1071377))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(1066609, J$.F(1066601, J$.R(1066585, '_typeof2', _typeof2, false, true), false)(J$.R(1066593, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1071385, J$e);
                        } finally {
                            if (J$.Fr(1071393))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(1066953, arguments.callee, this, arguments);
                            arguments = J$.N(1066961, 'arguments', arguments, true, false, false);
                            obj = J$.N(1066969, 'obj', obj, true, false, false);
                            if (J$.C(64952, J$.C(64912, J$.B(108362, '===', J$.U(108354, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1066641, 'Symbol', undefined, true, true) : Symbol = J$.R(1066641, 'Symbol', Symbol, true, true))), J$.T(1066649, 'function', 21, false))) ? J$.B(108370, '===', J$.F(1066681, J$.R(1066657, '_typeof2', _typeof2, false, true), false)(J$.G(1066673, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1066665, 'Symbol', undefined, true, true) : Symbol = J$.R(1066665, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1066689, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(1066761, '_typeof', J$.T(1066753, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(1066729, arguments.callee, this, arguments);
                                                arguments = J$.N(1066737, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1066745, 'obj', obj, true, false, false);
                                                return J$.Rt(1066721, J$.F(1066713, J$.R(1066697, '_typeof2', _typeof2, false, true), false)(J$.R(1066705, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1071401, J$e);
                                            } finally {
                                                if (J$.Fr(1071409))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(1066913, '_typeof', J$.T(1066905, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(1066881, arguments.callee, this, arguments);
                                                arguments = J$.N(1066889, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1066897, 'obj', obj, true, false, false);
                                                return J$.Rt(1066873, J$.C(64944, J$.C(64936, J$.C(64928, J$.C(64920, J$.R(1066769, 'obj', obj, false, false)) ? J$.B(108386, '===', J$.U(108378, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1066777, 'Symbol', undefined, true, true) : Symbol = J$.R(1066777, 'Symbol', Symbol, true, true))), J$.T(1066785, 'function', 21, false)) : J$._()) ? J$.B(108394, '===', J$.G(1066801, J$.R(1066793, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1066809, 'Symbol', undefined, true, true) : Symbol = J$.R(1066809, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(108402, '!==', J$.R(1066817, 'obj', obj, false, false), J$.G(1066833, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1066825, 'Symbol', undefined, true, true) : Symbol = J$.R(1066825, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1066841, 'symbol', 21, false) : J$.F(1066865, J$.R(1066849, '_typeof2', _typeof2, false, true), false)(J$.R(1066857, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1071417, J$e);
                                            } finally {
                                                if (J$.Fr(1071425))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(1066945, J$.F(1066937, J$.R(1066921, '_typeof', _typeof, false, true), false)(J$.R(1066929, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1071433, J$e);
                        } finally {
                            if (J$.Fr(1071441))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(1067017, arguments.callee, this, arguments);
                            arguments = J$.N(1067025, 'arguments', arguments, true, false, false);
                            instance = J$.N(1067033, 'instance', instance, true, false, false);
                            Constructor = J$.N(1067041, 'Constructor', Constructor, true, false, false);
                            if (J$.C(64960, J$.U(108418, '!', J$.B(108410, 'instanceof', J$.R(1066977, 'instance', instance, false, false), J$.R(1066985, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(1067009, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1066993, 'TypeError', undefined, true, true) : TypeError = J$.R(1066993, 'TypeError', TypeError, true, true)), true)(J$.T(1067001, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1071449, J$e);
                        } finally {
                            if (J$.Fr(1071457))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(1067289, arguments.callee, this, arguments);
                            arguments = J$.N(1067297, 'arguments', arguments, true, false, false);
                            target = J$.N(1067305, 'target', target, true, false, false);
                            props = J$.N(1067313, 'props', props, true, false, false);
                            J$.N(1067321, 'i', i, false, false, false);
                            J$.N(1067329, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(1067057, 'i', J$.T(1067049, 0, 22, false), i, false, false); J$.C(64984, J$.B(108426, '<', J$.R(1067065, 'i', i, false, false), J$.G(1067081, J$.R(1067073, 'props', props, false, false), 'length'))); J$.B(108450, '-', i = J$.W(1067097, 'i', J$.B(108442, '+', J$.U(108434, '+', J$.R(1067089, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(1067129, 'descriptor', J$.G(1067121, J$.R(1067105, 'props', props, false, false), J$.R(1067113, 'i', i, false, false)), descriptor, false, false);
                                J$.P(1067169, J$.R(1067137, 'descriptor', descriptor, false, false), 'enumerable', J$.C(64968, J$.G(1067153, J$.R(1067145, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(1067161, false, 23, false));
                                J$.P(1067193, J$.R(1067177, 'descriptor', descriptor, false, false), 'configurable', J$.T(1067185, true, 23, false));
                                if (J$.C(64976, J$.B(108458, 'in', J$.T(1067201, 'value', 21, false), J$.R(1067209, 'descriptor', descriptor, false, false))))
                                    J$.P(1067233, J$.R(1067217, 'descriptor', descriptor, false, false), 'writable', J$.T(1067225, true, 23, false));
                                J$.M(1067281, J$.I(typeof Object === 'undefined' ? Object = J$.R(1067241, 'Object', undefined, true, true) : Object = J$.R(1067241, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(1067249, 'target', target, false, false), J$.G(1067265, J$.R(1067257, 'descriptor', descriptor, false, false), 'key'), J$.R(1067273, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1071465, J$e);
                        } finally {
                            if (J$.Fr(1071473))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(1067441, arguments.callee, this, arguments);
                            arguments = J$.N(1067449, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(1067457, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(1067465, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(1067473, 'staticProps', staticProps, true, false, false);
                            if (J$.C(64992, J$.R(1067337, 'protoProps', protoProps, false, false)))
                                J$.F(1067377, J$.R(1067345, '_defineProperties', _defineProperties, false, true), false)(J$.G(1067361, J$.R(1067353, 'Constructor', Constructor, false, false), 'prototype'), J$.R(1067369, 'protoProps', protoProps, false, false));
                            if (J$.C(65000, J$.R(1067385, 'staticProps', staticProps, false, false)))
                                J$.F(1067417, J$.R(1067393, '_defineProperties', _defineProperties, false, true), false)(J$.R(1067401, 'Constructor', Constructor, false, false), J$.R(1067409, 'staticProps', staticProps, false, false));
                            return J$.Rt(1067433, J$.R(1067425, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(1071481, J$e);
                        } finally {
                            if (J$.Fr(1071489))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(1067585, arguments.callee, this, arguments);
                            arguments = J$.N(1067593, 'arguments', arguments, true, false, false);
                            self = J$.N(1067601, 'self', self, true, false, false);
                            call = J$.N(1067609, 'call', call, true, false, false);
                            if (J$.C(65024, J$.C(65016, J$.R(1067481, 'call', call, false, false)) ? J$.C(65008, J$.B(108466, '===', J$.F(1067505, J$.R(1067489, '_typeof', _typeof, false, true), false)(J$.R(1067497, 'call', call, false, false)), J$.T(1067513, 'object', 21, false))) ? J$._() : J$.B(108482, '===', J$.U(108474, 'typeof', J$.R(1067521, 'call', call, false, false)), J$.T(1067529, 'function', 21, false)) : J$._())) {
                                return J$.Rt(1067545, J$.R(1067537, 'call', call, false, false));
                            }
                            return J$.Rt(1067577, J$.F(1067569, J$.R(1067553, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(1067561, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(1071497, J$e);
                        } finally {
                            if (J$.Fr(1071505))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(1067673, arguments.callee, this, arguments);
                            arguments = J$.N(1067681, 'arguments', arguments, true, false, false);
                            self = J$.N(1067689, 'self', self, true, false, false);
                            if (J$.C(65032, J$.B(108490, '===', J$.R(1067617, 'self', self, false, false), void J$.T(1067625, 0, 22, false)))) {
                                throw J$.F(1067649, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(1067633, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(1067633, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(1067641, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(1067665, J$.R(1067657, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(1071513, J$e);
                        } finally {
                            if (J$.Fr(1071521))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(1068057, arguments.callee, this, arguments);
                            arguments = J$.N(1068065, 'arguments', arguments, true, false, false);
                            target = J$.N(1068073, 'target', target, true, false, false);
                            property = J$.N(1068081, 'property', property, true, false, false);
                            receiver = J$.N(1068089, 'receiver', receiver, true, false, false);
                            if (J$.C(65064, J$.C(65040, J$.B(108506, '!==', J$.U(108498, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1067697, 'Reflect', undefined, true, true) : Reflect = J$.R(1067697, 'Reflect', Reflect, true, true))), J$.T(1067705, 'undefined', 21, false))) ? J$.G(1067721, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1067713, 'Reflect', undefined, true, true) : Reflect = J$.R(1067713, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(1067745, '_get', J$.G(1067737, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1067729, 'Reflect', undefined, true, true) : Reflect = J$.R(1067729, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(1067993, '_get', J$.T(1067985, function _get(target, property, receiver) {
                                    jalangiLabel11:
                                        while (true) {
                                            try {
                                                J$.Fe(1067929, arguments.callee, this, arguments);
                                                arguments = J$.N(1067937, 'arguments', arguments, true, false, false);
                                                target = J$.N(1067945, 'target', target, true, false, false);
                                                property = J$.N(1067953, 'property', property, true, false, false);
                                                receiver = J$.N(1067961, 'receiver', receiver, true, false, false);
                                                J$.N(1067969, 'base', base, false, false, false);
                                                J$.N(1067977, 'desc', desc, false, false, false);
                                                var base = J$.W(1067785, 'base', J$.F(1067777, J$.R(1067753, '_superPropBase', _superPropBase, false, true), false)(J$.R(1067761, 'target', target, false, false), J$.R(1067769, 'property', property, false, false)), base, false, false);
                                                if (J$.C(65048, J$.U(108514, '!', J$.R(1067793, 'base', base, false, false))))
                                                    return J$.Rt(1067801, undefined);
                                                var desc = J$.W(1067841, 'desc', J$.M(1067833, J$.I(typeof Object === 'undefined' ? Object = J$.R(1067809, 'Object', undefined, true, true) : Object = J$.R(1067809, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(1067817, 'base', base, false, false), J$.R(1067825, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(65056, J$.G(1067857, J$.R(1067849, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(1067897, J$.M(1067889, J$.G(1067873, J$.R(1067865, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(1067881, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(1067921, J$.G(1067913, J$.R(1067905, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(1071529, J$e);
                                            } finally {
                                                if (J$.Fr(1071537))
                                                    continue jalangiLabel11;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(1068049, J$.F(1068041, J$.R(1068001, '_get', _get, false, true), false)(J$.R(1068009, 'target', target, false, false), J$.R(1068017, 'property', property, false, false), J$.C(65072, J$.R(1068025, 'receiver', receiver, false, false)) ? J$._() : J$.R(1068033, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(1071545, J$e);
                        } finally {
                            if (J$.Fr(1071553))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(1068209, arguments.callee, this, arguments);
                            arguments = J$.N(1068217, 'arguments', arguments, true, false, false);
                            object = J$.N(1068225, 'object', object, true, false, false);
                            property = J$.N(1068233, 'property', property, true, false, false);
                            while (J$.C(65088, J$.U(108522, '!', J$.M(1068137, J$.G(1068113, J$.G(1068105, J$.I(typeof Object === 'undefined' ? Object = J$.R(1068097, 'Object', undefined, true, true) : Object = J$.R(1068097, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(1068121, 'object', object, false, false), J$.R(1068129, 'property', property, false, false))))) {
                                object = J$.W(1068169, 'object', J$.F(1068161, J$.R(1068145, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1068153, 'object', object, false, false)), object, false, false);
                                if (J$.C(65080, J$.B(108530, '===', J$.R(1068177, 'object', object, false, false), J$.T(1068185, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(1068201, J$.R(1068193, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(1071561, J$e);
                        } finally {
                            if (J$.Fr(1071569))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(1068393, arguments.callee, this, arguments);
                            arguments = J$.N(1068401, 'arguments', arguments, true, false, false);
                            o = J$.N(1068409, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(1068353, '_getPrototypeOf', J$.C(65104, J$.G(1068249, J$.I(typeof Object === 'undefined' ? Object = J$.R(1068241, 'Object', undefined, true, true) : Object = J$.R(1068241, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(1068265, J$.I(typeof Object === 'undefined' ? Object = J$.R(1068257, 'Object', undefined, true, true) : Object = J$.R(1068257, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(1068345, function _getPrototypeOf(o) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(1068321, arguments.callee, this, arguments);
                                            arguments = J$.N(1068329, 'arguments', arguments, true, false, false);
                                            o = J$.N(1068337, 'o', o, true, false, false);
                                            return J$.Rt(1068313, J$.C(65096, J$.G(1068281, J$.R(1068273, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(1068305, J$.I(typeof Object === 'undefined' ? Object = J$.R(1068289, 'Object', undefined, true, true) : Object = J$.R(1068289, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(1068297, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(1071577, J$e);
                                        } finally {
                                            if (J$.Fr(1071585))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(1068385, J$.F(1068377, J$.R(1068361, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1068369, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(1071593, J$e);
                        } finally {
                            if (J$.Fr(1071601))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(1068609, arguments.callee, this, arguments);
                            arguments = J$.N(1068617, 'arguments', arguments, true, false, false);
                            subClass = J$.N(1068625, 'subClass', subClass, true, false, false);
                            superClass = J$.N(1068633, 'superClass', superClass, true, false, false);
                            if (J$.C(65120, J$.C(65112, J$.B(108546, '!==', J$.U(108538, 'typeof', J$.R(1068417, 'superClass', superClass, false, false)), J$.T(1068425, 'function', 21, false))) ? J$.B(108554, '!==', J$.R(1068433, 'superClass', superClass, false, false), J$.T(1068441, null, 25, false)) : J$._())) {
                                throw J$.F(1068465, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1068449, 'TypeError', undefined, true, true) : TypeError = J$.R(1068449, 'TypeError', TypeError, true, true)), true)(J$.T(1068457, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(1068561, J$.R(1068473, 'subClass', subClass, false, false), 'prototype', J$.M(1068553, J$.I(typeof Object === 'undefined' ? Object = J$.R(1068481, 'Object', undefined, true, true) : Object = J$.R(1068481, 'Object', Object, true, true)), 'create', false)(J$.C(65128, J$.R(1068489, 'superClass', superClass, false, false)) ? J$.G(1068505, J$.R(1068497, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(1068545, {
                                constructor: J$.T(1068537, {
                                    value: J$.R(1068513, 'subClass', subClass, false, false),
                                    writable: J$.T(1068521, true, 23, false),
                                    configurable: J$.T(1068529, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(65136, J$.R(1068569, 'superClass', superClass, false, false)))
                                J$.F(1068601, J$.R(1068577, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1068585, 'subClass', subClass, false, false), J$.R(1068593, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(1071609, J$e);
                        } finally {
                            if (J$.Fr(1071617))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(1068785, arguments.callee, this, arguments);
                            arguments = J$.N(1068793, 'arguments', arguments, true, false, false);
                            o = J$.N(1068801, 'o', o, true, false, false);
                            p = J$.N(1068809, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(1068737, '_setPrototypeOf', J$.C(65144, J$.G(1068649, J$.I(typeof Object === 'undefined' ? Object = J$.R(1068641, 'Object', undefined, true, true) : Object = J$.R(1068641, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(1068729, function _setPrototypeOf(o, p) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(1068697, arguments.callee, this, arguments);
                                            arguments = J$.N(1068705, 'arguments', arguments, true, false, false);
                                            o = J$.N(1068713, 'o', o, true, false, false);
                                            p = J$.N(1068721, 'p', p, true, false, false);
                                            J$.P(1068673, J$.R(1068657, 'o', o, false, false), '__proto__', J$.R(1068665, 'p', p, false, false));
                                            return J$.Rt(1068689, J$.R(1068681, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1071625, J$e);
                                        } finally {
                                            if (J$.Fr(1071633))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(1068777, J$.F(1068769, J$.R(1068745, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1068753, 'o', o, false, false), J$.R(1068761, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(1071641, J$e);
                        } finally {
                            if (J$.Fr(1071649))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(1071097, '_typeof2', J$.T(1071089, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(1071113, '_typeof', J$.T(1071105, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(1071129, '_classCallCheck', J$.T(1071121, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(1071145, '_defineProperties', J$.T(1071137, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(1071161, '_createClass', J$.T(1071153, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(1071177, '_possibleConstructorReturn', J$.T(1071169, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(1071193, '_assertThisInitialized', J$.T(1071185, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(1071209, '_get', J$.T(1071201, _get, 12, false), true, false, false);
            _superPropBase = J$.N(1071225, '_superPropBase', J$.T(1071217, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(1071241, '_getPrototypeOf', J$.T(1071233, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(1071257, '_inherits', J$.T(1071249, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(1071273, '_setPrototypeOf', J$.T(1071265, _setPrototypeOf, 12, false), true, false, false);
            J$.N(1071281, 'CommandOperationV2', CommandOperationV2, false, false, false);
            J$.N(1071289, 'MongoError', MongoError, false, false, false);
            J$.N(1071297, 'maxWireVersion', maxWireVersion, false, false, false);
            J$.N(1071305, 'ReadPreference', ReadPreference, false, false, false);
            J$.N(1071313, 'Aspect', Aspect, false, false, false);
            J$.N(1071321, 'defineAspects', defineAspects, false, false, false);
            J$.N(1071329, 'DB_AGGREGATE_COLLECTION', DB_AGGREGATE_COLLECTION, false, false, false);
            J$.N(1071337, 'MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT', MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT, false, false, false);
            J$.N(1071345, 'AggregateOperation', AggregateOperation, false, false, false);
            J$.T(1066345, 'use strict', 21, false);
            var CommandOperationV2 = J$.W(1068841, 'CommandOperationV2', J$.F(1068833, J$.I(typeof require === 'undefined' ? require = J$.R(1068817, 'require', undefined, true, true) : require = J$.R(1068817, 'require', require, true, true)), false)(J$.T(1068825, './command_v2', 21, false)), CommandOperationV2, false, true);
            var MongoError = J$.W(1068881, 'MongoError', J$.G(1068873, J$.F(1068865, J$.I(typeof require === 'undefined' ? require = J$.R(1068849, 'require', undefined, true, true) : require = J$.R(1068849, 'require', require, true, true)), false)(J$.T(1068857, '../core', 21, false)), 'MongoError'), MongoError, false, true);
            var maxWireVersion = J$.W(1068921, 'maxWireVersion', J$.G(1068913, J$.F(1068905, J$.I(typeof require === 'undefined' ? require = J$.R(1068889, 'require', undefined, true, true) : require = J$.R(1068889, 'require', require, true, true)), false)(J$.T(1068897, '../core/utils', 21, false)), 'maxWireVersion'), maxWireVersion, false, true);
            var ReadPreference = J$.W(1068961, 'ReadPreference', J$.G(1068953, J$.F(1068945, J$.I(typeof require === 'undefined' ? require = J$.R(1068929, 'require', undefined, true, true) : require = J$.R(1068929, 'require', require, true, true)), false)(J$.T(1068937, '../core', 21, false)), 'ReadPreference'), ReadPreference, false, true);
            var Aspect = J$.W(1069001, 'Aspect', J$.G(1068993, J$.F(1068985, J$.I(typeof require === 'undefined' ? require = J$.R(1068969, 'require', undefined, true, true) : require = J$.R(1068969, 'require', require, true, true)), false)(J$.T(1068977, './operation', 21, false)), 'Aspect'), Aspect, false, true);
            var defineAspects = J$.W(1069041, 'defineAspects', J$.G(1069033, J$.F(1069025, J$.I(typeof require === 'undefined' ? require = J$.R(1069009, 'require', undefined, true, true) : require = J$.R(1069009, 'require', require, true, true)), false)(J$.T(1069017, './operation', 21, false)), 'defineAspects'), defineAspects, false, true);
            var DB_AGGREGATE_COLLECTION = J$.W(1069057, 'DB_AGGREGATE_COLLECTION', J$.T(1069049, 1, 22, false), DB_AGGREGATE_COLLECTION, false, true);
            var MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT = J$.W(1069073, 'MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT', J$.T(1069065, 8, 22, false), MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT, false, true);
            var AggregateOperation = J$.W(1070969, 'AggregateOperation', J$.F(1070961, J$.T(1070945, function (_CommandOperationV) {
                jalangiLabel23:
                    while (true) {
                        try {
                            J$.Fe(1070905, arguments.callee, this, arguments);
                            function AggregateOperation(parent, pipeline, options) {
                                jalangiLabel19:
                                    while (true) {
                                        try {
                                            J$.Fe(1069873, arguments.callee, this, arguments);
                                            arguments = J$.N(1069881, 'arguments', arguments, true, false, false);
                                            parent = J$.N(1069889, 'parent', parent, true, false, false);
                                            pipeline = J$.N(1069897, 'pipeline', pipeline, true, false, false);
                                            options = J$.N(1069905, 'options', options, true, false, false);
                                            J$.N(1069913, '_this', _this, false, false, false);
                                            J$.N(1069921, 'finalStage', finalStage, false, false, false);
                                            var _this;
                                            J$.F(1069137, J$.R(1069113, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(1069121, 'this', this, false, false), J$.R(1069129, 'AggregateOperation', AggregateOperation, false, false));
                                            _this = J$.W(1069241, '_this', J$.F(1069233, J$.R(1069145, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(1069153, 'this', this, false, false), J$.M(1069225, J$.F(1069177, J$.R(1069161, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1069169, 'AggregateOperation', AggregateOperation, false, false)), 'call', false)(J$.R(1069185, 'this', this, false, false), J$.R(1069193, 'parent', parent, false, false), J$.R(1069201, 'options', options, false, false), J$.T(1069217, { fullResponse: J$.T(1069209, true, 23, false) }, 11, false))), _this, false, false);
                                            J$.P(1069353, J$.R(1069249, '_this', _this, false, false), 'target', J$.C(65160, J$.C(65152, J$.G(1069273, J$.G(1069265, J$.R(1069257, 'parent', parent, false, false), 's'), 'namespace')) ? J$.G(1069305, J$.G(1069297, J$.G(1069289, J$.R(1069281, 'parent', parent, false, false), 's'), 'namespace'), 'collection') : J$._()) ? J$.G(1069337, J$.G(1069329, J$.G(1069321, J$.R(1069313, 'parent', parent, false, false), 's'), 'namespace'), 'collection') : J$.R(1069345, 'DB_AGGREGATE_COLLECTION', DB_AGGREGATE_COLLECTION, false, true));
                                            J$.P(1069377, J$.R(1069361, '_this', _this, false, false), 'pipeline', J$.R(1069369, 'pipeline', pipeline, false, false));
                                            J$.P(1069401, J$.R(1069385, '_this', _this, false, false), 'hasWriteStage', J$.T(1069393, false, 23, false));
                                            if (J$.C(65192, J$.B(108570, '===', J$.U(108562, 'typeof', J$.G(1069417, J$.R(1069409, 'options', options, false, false), 'out')), J$.T(1069425, 'string', 21, false)))) {
                                                J$.P(1069489, J$.R(1069433, '_this', _this, false, false), 'pipeline', J$.M(1069481, J$.G(1069449, J$.R(1069441, '_this', _this, false, false), 'pipeline'), 'concat', false)(J$.T(1069473, { $out: J$.G(1069465, J$.R(1069457, 'options', options, false, false), 'out') }, 11, false)));
                                                J$.P(1069513, J$.R(1069497, '_this', _this, false, false), 'hasWriteStage', J$.T(1069505, true, 23, false));
                                            } else if (J$.C(65184, J$.B(108578, '>', J$.G(1069529, J$.R(1069521, 'pipeline', pipeline, false, false), 'length'), J$.T(1069537, 0, 22, false)))) {
                                                var finalStage = J$.W(1069585, 'finalStage', J$.G(1069577, J$.R(1069545, 'pipeline', pipeline, false, false), J$.B(108586, '-', J$.G(1069561, J$.R(1069553, 'pipeline', pipeline, false, false), 'length'), J$.T(1069569, 1, 22, false))), finalStage, false, false);
                                                if (J$.C(65176, J$.C(65168, J$.G(1069601, J$.R(1069593, 'finalStage', finalStage, false, false), '$out')) ? J$._() : J$.G(1069617, J$.R(1069609, 'finalStage', finalStage, false, false), '$merge'))) {
                                                    J$.P(1069641, J$.R(1069625, '_this', _this, false, false), 'hasWriteStage', J$.T(1069633, true, 23, false));
                                                }
                                            }
                                            if (J$.C(65200, J$.G(1069657, J$.R(1069649, '_this', _this, false, false), 'hasWriteStage'))) {
                                                J$.P(1069689, J$.R(1069665, '_this', _this, false, false), 'readPreference', J$.G(1069681, J$.R(1069673, 'ReadPreference', ReadPreference, false, true), 'primary'));
                                            }
                                            if (J$.C(65224, J$.C(65216, J$.G(1069705, J$.R(1069697, 'options', options, false, false), 'explain')) ? J$.C(65208, J$.G(1069721, J$.R(1069713, '_this', _this, false, false), 'readConcern')) ? J$._() : J$.G(1069737, J$.R(1069729, '_this', _this, false, false), 'writeConcern') : J$._())) {
                                                throw J$.F(1069761, J$.R(1069745, 'MongoError', MongoError, false, true), true)(J$.T(1069753, '"explain" cannot be used on an aggregate call with readConcern/writeConcern', 21, false));
                                            }
                                            if (J$.C(65240, J$.C(65232, J$.B(108594, '!=', J$.G(1069777, J$.R(1069769, 'options', options, false, false), 'cursor'), J$.T(1069785, null, 25, false))) ? J$.B(108602, '!==', J$.F(1069817, J$.R(1069793, '_typeof', _typeof, false, true), false)(J$.G(1069809, J$.R(1069801, 'options', options, false, false), 'cursor')), J$.T(1069825, 'object', 21, false)) : J$._())) {
                                                throw J$.F(1069849, J$.R(1069833, 'MongoError', MongoError, false, true), true)(J$.T(1069841, 'cursor options must be an object', 21, false));
                                            }
                                            return J$.Rt(1069865, J$.R(1069857, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1071657, J$e);
                                        } finally {
                                            if (J$.Fr(1071665))
                                                continue jalangiLabel19;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(1070913, 'arguments', arguments, true, false, false);
                            _CommandOperationV = J$.N(1070921, '_CommandOperationV', _CommandOperationV, true, false, false);
                            AggregateOperation = J$.N(1070937, 'AggregateOperation', J$.T(1070929, AggregateOperation, 12, false), true, false, false);
                            J$.F(1069105, J$.R(1069081, '_inherits', _inherits, false, true), false)(J$.R(1069089, 'AggregateOperation', AggregateOperation, false, false), J$.R(1069097, '_CommandOperationV', _CommandOperationV, false, false));
                            J$.F(1070881, J$.R(1069929, '_createClass', _createClass, false, true), false)(J$.R(1069937, 'AggregateOperation', AggregateOperation, false, false), J$.T(1070873, [
                                J$.T(1070017, {
                                    key: J$.T(1069945, 'addToPipeline', 21, false),
                                    value: J$.T(1070009, function addToPipeline(stage) {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(1069985, arguments.callee, this, arguments);
                                                    arguments = J$.N(1069993, 'arguments', arguments, true, false, false);
                                                    stage = J$.N(1070001, 'stage', stage, true, false, false);
                                                    J$.M(1069977, J$.G(1069961, J$.R(1069953, 'this', this, false, false), 'pipeline'), 'push', false)(J$.R(1069969, 'stage', stage, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1071673, J$e);
                                                } finally {
                                                    if (J$.Fr(1071681))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1070801, {
                                    key: J$.T(1070025, 'execute', 21, false),
                                    value: J$.T(1070793, function execute(server, callback) {
                                        jalangiLabel21:
                                            while (true) {
                                                try {
                                                    J$.Fe(1070737, arguments.callee, this, arguments);
                                                    arguments = J$.N(1070745, 'arguments', arguments, true, false, false);
                                                    server = J$.N(1070753, 'server', server, true, false, false);
                                                    callback = J$.N(1070761, 'callback', callback, true, false, false);
                                                    J$.N(1070769, 'options', options, false, false, false);
                                                    J$.N(1070777, 'serverWireVersion', serverWireVersion, false, false, false);
                                                    J$.N(1070785, 'command', command, false, false, false);
                                                    var options = J$.W(1070049, 'options', J$.G(1070041, J$.R(1070033, 'this', this, false, false), 'options'), options, false, false);
                                                    var serverWireVersion = J$.W(1070081, 'serverWireVersion', J$.F(1070073, J$.R(1070057, 'maxWireVersion', maxWireVersion, false, true), false)(J$.R(1070065, 'server', server, false, false)), serverWireVersion, false, false);
                                                    var command = J$.W(1070129, 'command', J$.T(1070121, {
                                                        aggregate: J$.G(1070097, J$.R(1070089, 'this', this, false, false), 'target'),
                                                        pipeline: J$.G(1070113, J$.R(1070105, 'this', this, false, false), 'pipeline')
                                                    }, 11, false), command, false, false);
                                                    if (J$.C(65256, J$.C(65248, J$.G(1070145, J$.R(1070137, 'this', this, false, false), 'hasWriteStage')) ? J$.B(108610, '<', J$.R(1070153, 'serverWireVersion', serverWireVersion, false, false), J$.R(1070161, 'MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT', MIN_WIRE_VERSION_$OUT_READ_CONCERN_SUPPORT, false, true)) : J$._())) {
                                                        J$.P(1070185, J$.R(1070169, 'this', this, false, false), 'readConcern', J$.T(1070177, null, 25, false));
                                                    }
                                                    if (J$.C(65280, J$.B(108618, '>=', J$.R(1070193, 'serverWireVersion', serverWireVersion, false, false), J$.T(1070201, 5, 22, false)))) {
                                                        if (J$.C(65272, J$.C(65264, J$.G(1070217, J$.R(1070209, 'this', this, false, false), 'hasWriteStage')) ? J$.G(1070233, J$.R(1070225, 'this', this, false, false), 'writeConcern') : J$._())) {
                                                            J$.M(1070281, J$.I(typeof Object === 'undefined' ? Object = J$.R(1070241, 'Object', undefined, true, true) : Object = J$.R(1070241, 'Object', Object, true, true)), 'assign', false)(J$.R(1070249, 'command', command, false, false), J$.T(1070273, { writeConcern: J$.G(1070265, J$.R(1070257, 'this', this, false, false), 'writeConcern') }, 11, false));
                                                        }
                                                    }
                                                    if (J$.C(65288, J$.B(108626, '===', J$.G(1070297, J$.R(1070289, 'options', options, false, false), 'bypassDocumentValidation'), J$.T(1070305, true, 23, false)))) {
                                                        J$.P(1070337, J$.R(1070313, 'command', command, false, false), 'bypassDocumentValidation', J$.G(1070329, J$.R(1070321, 'options', options, false, false), 'bypassDocumentValidation'));
                                                    }
                                                    if (J$.C(65296, J$.B(108642, '===', J$.U(108634, 'typeof', J$.G(1070353, J$.R(1070345, 'options', options, false, false), 'allowDiskUse')), J$.T(1070361, 'boolean', 21, false)))) {
                                                        J$.P(1070393, J$.R(1070369, 'command', command, false, false), 'allowDiskUse', J$.G(1070385, J$.R(1070377, 'options', options, false, false), 'allowDiskUse'));
                                                    }
                                                    if (J$.C(65304, J$.G(1070409, J$.R(1070401, 'options', options, false, false), 'hint'))) {
                                                        J$.P(1070441, J$.R(1070417, 'command', command, false, false), 'hint', J$.G(1070433, J$.R(1070425, 'options', options, false, false), 'hint'));
                                                    }
                                                    if (J$.C(65312, J$.G(1070457, J$.R(1070449, 'options', options, false, false), 'explain'))) {
                                                        J$.P(1070481, J$.R(1070465, 'options', options, false, false), 'full', J$.T(1070473, false, 23, false));
                                                        J$.P(1070513, J$.R(1070489, 'command', command, false, false), 'explain', J$.G(1070505, J$.R(1070497, 'options', options, false, false), 'explain'));
                                                    }
                                                    J$.P(1070553, J$.R(1070521, 'command', command, false, false), 'cursor', J$.C(65320, J$.G(1070537, J$.R(1070529, 'options', options, false, false), 'cursor')) ? J$._() : J$.T(1070545, {}, 11, false));
                                                    if (J$.C(65336, J$.C(65328, J$.G(1070569, J$.R(1070561, 'options', options, false, false), 'batchSize')) ? J$.U(108650, '!', J$.G(1070585, J$.R(1070577, 'this', this, false, false), 'hasWriteStage')) : J$._())) {
                                                        J$.P(1070625, J$.G(1070601, J$.R(1070593, 'command', command, false, false), 'cursor'), 'batchSize', J$.G(1070617, J$.R(1070609, 'options', options, false, false), 'batchSize'));
                                                    }
                                                    J$.M(1070729, J$.F(1070689, J$.R(1070633, '_get', _get, false, true), false)(J$.F(1070665, J$.R(1070641, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(1070657, J$.R(1070649, 'AggregateOperation', AggregateOperation, false, false), 'prototype')), J$.T(1070673, 'executeCommand', 21, false), J$.R(1070681, 'this', this, false, false)), 'call', false)(J$.R(1070697, 'this', this, false, false), J$.R(1070705, 'server', server, false, false), J$.R(1070713, 'command', command, false, false), J$.R(1070721, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1071689, J$e);
                                                } finally {
                                                    if (J$.Fr(1071697))
                                                        continue jalangiLabel21;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1070865, {
                                    key: J$.T(1070809, 'canRetryRead', 21, false),
                                    get: J$.T(1070857, function get() {
                                        jalangiLabel22:
                                            while (true) {
                                                try {
                                                    J$.Fe(1070841, arguments.callee, this, arguments);
                                                    arguments = J$.N(1070849, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(1070833, J$.U(108658, '!', J$.G(1070825, J$.R(1070817, 'this', this, false, false), 'hasWriteStage')));
                                                } catch (J$e) {
                                                    J$.Ex(1071705, J$e);
                                                } finally {
                                                    if (J$.Fr(1071713))
                                                        continue jalangiLabel22;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(1070897, J$.R(1070889, 'AggregateOperation', AggregateOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(1071721, J$e);
                        } finally {
                            if (J$.Fr(1071729))
                                continue jalangiLabel23;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(1070953, 'CommandOperationV2', CommandOperationV2, false, true)), AggregateOperation, false, true);
            J$.F(1071049, J$.R(1070977, 'defineAspects', defineAspects, false, true), false)(J$.R(1070985, 'AggregateOperation', AggregateOperation, false, true), J$.T(1071041, [
                J$.G(1071001, J$.R(1070993, 'Aspect', Aspect, false, true), 'READ_OPERATION'),
                J$.G(1071017, J$.R(1071009, 'Aspect', Aspect, false, true), 'RETRYABLE'),
                J$.G(1071033, J$.R(1071025, 'Aspect', Aspect, false, true), 'EXECUTE_WITH_SELECTION')
            ], 10, false));
            J$.P(1071073, J$.I(typeof module === 'undefined' ? module = J$.R(1071057, 'module', undefined, true, true) : module = J$.R(1071057, 'module', module, true, true)), 'exports', J$.R(1071065, 'AggregateOperation', AggregateOperation, false, true));
        } catch (J$e) {
            J$.Ex(1071737, J$e);
        } finally {
            if (J$.Sr(1071745))
                continue jalangiLabel24;
            else
                break jalangiLabel24;
        }
    }
// JALANGI DO NOT INSTRUMENT
