// mongodb/lib/operations/create_index.js
J$.noInstrEval = false;
jalangiLabel24:
    while (true) {
        try {
            J$.Se(1144737, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/operations/create_index.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(1140617, arguments.callee, this, arguments);
                            arguments = J$.N(1140625, 'arguments', arguments, true, false, false);
                            obj = J$.N(1140633, 'obj', obj, true, false, false);
                            if (J$.C(70184, J$.C(70144, J$.B(112906, '===', J$.U(112898, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1140353, 'Symbol', undefined, true, true) : Symbol = J$.R(1140353, 'Symbol', Symbol, true, true))), J$.T(1140361, 'function', 21, false))) ? J$.B(112922, '===', J$.U(112914, 'typeof', J$.G(1140377, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1140369, 'Symbol', undefined, true, true) : Symbol = J$.R(1140369, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1140385, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(1140441, '_typeof2', J$.T(1140433, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(1140409, arguments.callee, this, arguments);
                                                arguments = J$.N(1140417, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1140425, 'obj', obj, true, false, false);
                                                return J$.Rt(1140401, J$.U(112930, 'typeof', J$.R(1140393, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1145001, J$e);
                                            } finally {
                                                if (J$.Fr(1145009))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(1140577, '_typeof2', J$.T(1140569, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(1140545, arguments.callee, this, arguments);
                                                arguments = J$.N(1140553, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1140561, 'obj', obj, true, false, false);
                                                return J$.Rt(1140537, J$.C(70176, J$.C(70168, J$.C(70160, J$.C(70152, J$.R(1140449, 'obj', obj, false, false)) ? J$.B(112946, '===', J$.U(112938, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1140457, 'Symbol', undefined, true, true) : Symbol = J$.R(1140457, 'Symbol', Symbol, true, true))), J$.T(1140465, 'function', 21, false)) : J$._()) ? J$.B(112954, '===', J$.G(1140481, J$.R(1140473, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1140489, 'Symbol', undefined, true, true) : Symbol = J$.R(1140489, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(112962, '!==', J$.R(1140497, 'obj', obj, false, false), J$.G(1140513, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1140505, 'Symbol', undefined, true, true) : Symbol = J$.R(1140505, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1140521, 'symbol', 21, false) : J$.U(112970, 'typeof', J$.R(1140529, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1145017, J$e);
                                            } finally {
                                                if (J$.Fr(1145025))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(1140609, J$.F(1140601, J$.R(1140585, '_typeof2', _typeof2, false, true), false)(J$.R(1140593, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1145033, J$e);
                        } finally {
                            if (J$.Fr(1145041))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(1140953, arguments.callee, this, arguments);
                            arguments = J$.N(1140961, 'arguments', arguments, true, false, false);
                            obj = J$.N(1140969, 'obj', obj, true, false, false);
                            if (J$.C(70232, J$.C(70192, J$.B(112986, '===', J$.U(112978, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1140641, 'Symbol', undefined, true, true) : Symbol = J$.R(1140641, 'Symbol', Symbol, true, true))), J$.T(1140649, 'function', 21, false))) ? J$.B(112994, '===', J$.F(1140681, J$.R(1140657, '_typeof2', _typeof2, false, true), false)(J$.G(1140673, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1140665, 'Symbol', undefined, true, true) : Symbol = J$.R(1140665, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1140689, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(1140761, '_typeof', J$.T(1140753, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(1140729, arguments.callee, this, arguments);
                                                arguments = J$.N(1140737, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1140745, 'obj', obj, true, false, false);
                                                return J$.Rt(1140721, J$.F(1140713, J$.R(1140697, '_typeof2', _typeof2, false, true), false)(J$.R(1140705, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1145049, J$e);
                                            } finally {
                                                if (J$.Fr(1145057))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(1140913, '_typeof', J$.T(1140905, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(1140881, arguments.callee, this, arguments);
                                                arguments = J$.N(1140889, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1140897, 'obj', obj, true, false, false);
                                                return J$.Rt(1140873, J$.C(70224, J$.C(70216, J$.C(70208, J$.C(70200, J$.R(1140769, 'obj', obj, false, false)) ? J$.B(113010, '===', J$.U(113002, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1140777, 'Symbol', undefined, true, true) : Symbol = J$.R(1140777, 'Symbol', Symbol, true, true))), J$.T(1140785, 'function', 21, false)) : J$._()) ? J$.B(113018, '===', J$.G(1140801, J$.R(1140793, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1140809, 'Symbol', undefined, true, true) : Symbol = J$.R(1140809, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(113026, '!==', J$.R(1140817, 'obj', obj, false, false), J$.G(1140833, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1140825, 'Symbol', undefined, true, true) : Symbol = J$.R(1140825, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1140841, 'symbol', 21, false) : J$.F(1140865, J$.R(1140849, '_typeof2', _typeof2, false, true), false)(J$.R(1140857, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1145065, J$e);
                                            } finally {
                                                if (J$.Fr(1145073))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(1140945, J$.F(1140937, J$.R(1140921, '_typeof', _typeof, false, true), false)(J$.R(1140929, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1145081, J$e);
                        } finally {
                            if (J$.Fr(1145089))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(1141017, arguments.callee, this, arguments);
                            arguments = J$.N(1141025, 'arguments', arguments, true, false, false);
                            instance = J$.N(1141033, 'instance', instance, true, false, false);
                            Constructor = J$.N(1141041, 'Constructor', Constructor, true, false, false);
                            if (J$.C(70240, J$.U(113042, '!', J$.B(113034, 'instanceof', J$.R(1140977, 'instance', instance, false, false), J$.R(1140985, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(1141009, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1140993, 'TypeError', undefined, true, true) : TypeError = J$.R(1140993, 'TypeError', TypeError, true, true)), true)(J$.T(1141001, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1145097, J$e);
                        } finally {
                            if (J$.Fr(1145105))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(1141289, arguments.callee, this, arguments);
                            arguments = J$.N(1141297, 'arguments', arguments, true, false, false);
                            target = J$.N(1141305, 'target', target, true, false, false);
                            props = J$.N(1141313, 'props', props, true, false, false);
                            J$.N(1141321, 'i', i, false, false, false);
                            J$.N(1141329, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(1141057, 'i', J$.T(1141049, 0, 22, false), i, false, false); J$.C(70264, J$.B(113050, '<', J$.R(1141065, 'i', i, false, false), J$.G(1141081, J$.R(1141073, 'props', props, false, false), 'length'))); J$.B(113074, '-', i = J$.W(1141097, 'i', J$.B(113066, '+', J$.U(113058, '+', J$.R(1141089, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(1141129, 'descriptor', J$.G(1141121, J$.R(1141105, 'props', props, false, false), J$.R(1141113, 'i', i, false, false)), descriptor, false, false);
                                J$.P(1141169, J$.R(1141137, 'descriptor', descriptor, false, false), 'enumerable', J$.C(70248, J$.G(1141153, J$.R(1141145, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(1141161, false, 23, false));
                                J$.P(1141193, J$.R(1141177, 'descriptor', descriptor, false, false), 'configurable', J$.T(1141185, true, 23, false));
                                if (J$.C(70256, J$.B(113082, 'in', J$.T(1141201, 'value', 21, false), J$.R(1141209, 'descriptor', descriptor, false, false))))
                                    J$.P(1141233, J$.R(1141217, 'descriptor', descriptor, false, false), 'writable', J$.T(1141225, true, 23, false));
                                J$.M(1141281, J$.I(typeof Object === 'undefined' ? Object = J$.R(1141241, 'Object', undefined, true, true) : Object = J$.R(1141241, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(1141249, 'target', target, false, false), J$.G(1141265, J$.R(1141257, 'descriptor', descriptor, false, false), 'key'), J$.R(1141273, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1145113, J$e);
                        } finally {
                            if (J$.Fr(1145121))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(1141441, arguments.callee, this, arguments);
                            arguments = J$.N(1141449, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(1141457, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(1141465, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(1141473, 'staticProps', staticProps, true, false, false);
                            if (J$.C(70272, J$.R(1141337, 'protoProps', protoProps, false, false)))
                                J$.F(1141377, J$.R(1141345, '_defineProperties', _defineProperties, false, true), false)(J$.G(1141361, J$.R(1141353, 'Constructor', Constructor, false, false), 'prototype'), J$.R(1141369, 'protoProps', protoProps, false, false));
                            if (J$.C(70280, J$.R(1141385, 'staticProps', staticProps, false, false)))
                                J$.F(1141417, J$.R(1141393, '_defineProperties', _defineProperties, false, true), false)(J$.R(1141401, 'Constructor', Constructor, false, false), J$.R(1141409, 'staticProps', staticProps, false, false));
                            return J$.Rt(1141433, J$.R(1141425, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(1145129, J$e);
                        } finally {
                            if (J$.Fr(1145137))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(1141585, arguments.callee, this, arguments);
                            arguments = J$.N(1141593, 'arguments', arguments, true, false, false);
                            self = J$.N(1141601, 'self', self, true, false, false);
                            call = J$.N(1141609, 'call', call, true, false, false);
                            if (J$.C(70304, J$.C(70296, J$.R(1141481, 'call', call, false, false)) ? J$.C(70288, J$.B(113090, '===', J$.F(1141505, J$.R(1141489, '_typeof', _typeof, false, true), false)(J$.R(1141497, 'call', call, false, false)), J$.T(1141513, 'object', 21, false))) ? J$._() : J$.B(113106, '===', J$.U(113098, 'typeof', J$.R(1141521, 'call', call, false, false)), J$.T(1141529, 'function', 21, false)) : J$._())) {
                                return J$.Rt(1141545, J$.R(1141537, 'call', call, false, false));
                            }
                            return J$.Rt(1141577, J$.F(1141569, J$.R(1141553, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(1141561, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(1145145, J$e);
                        } finally {
                            if (J$.Fr(1145153))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(1141673, arguments.callee, this, arguments);
                            arguments = J$.N(1141681, 'arguments', arguments, true, false, false);
                            self = J$.N(1141689, 'self', self, true, false, false);
                            if (J$.C(70312, J$.B(113114, '===', J$.R(1141617, 'self', self, false, false), void J$.T(1141625, 0, 22, false)))) {
                                throw J$.F(1141649, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(1141633, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(1141633, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(1141641, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(1141665, J$.R(1141657, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(1145161, J$e);
                        } finally {
                            if (J$.Fr(1145169))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(1142057, arguments.callee, this, arguments);
                            arguments = J$.N(1142065, 'arguments', arguments, true, false, false);
                            target = J$.N(1142073, 'target', target, true, false, false);
                            property = J$.N(1142081, 'property', property, true, false, false);
                            receiver = J$.N(1142089, 'receiver', receiver, true, false, false);
                            if (J$.C(70344, J$.C(70320, J$.B(113130, '!==', J$.U(113122, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1141697, 'Reflect', undefined, true, true) : Reflect = J$.R(1141697, 'Reflect', Reflect, true, true))), J$.T(1141705, 'undefined', 21, false))) ? J$.G(1141721, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1141713, 'Reflect', undefined, true, true) : Reflect = J$.R(1141713, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(1141745, '_get', J$.G(1141737, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1141729, 'Reflect', undefined, true, true) : Reflect = J$.R(1141729, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(1141993, '_get', J$.T(1141985, function _get(target, property, receiver) {
                                    jalangiLabel11:
                                        while (true) {
                                            try {
                                                J$.Fe(1141929, arguments.callee, this, arguments);
                                                arguments = J$.N(1141937, 'arguments', arguments, true, false, false);
                                                target = J$.N(1141945, 'target', target, true, false, false);
                                                property = J$.N(1141953, 'property', property, true, false, false);
                                                receiver = J$.N(1141961, 'receiver', receiver, true, false, false);
                                                J$.N(1141969, 'base', base, false, false, false);
                                                J$.N(1141977, 'desc', desc, false, false, false);
                                                var base = J$.W(1141785, 'base', J$.F(1141777, J$.R(1141753, '_superPropBase', _superPropBase, false, true), false)(J$.R(1141761, 'target', target, false, false), J$.R(1141769, 'property', property, false, false)), base, false, false);
                                                if (J$.C(70328, J$.U(113138, '!', J$.R(1141793, 'base', base, false, false))))
                                                    return J$.Rt(1141801, undefined);
                                                var desc = J$.W(1141841, 'desc', J$.M(1141833, J$.I(typeof Object === 'undefined' ? Object = J$.R(1141809, 'Object', undefined, true, true) : Object = J$.R(1141809, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(1141817, 'base', base, false, false), J$.R(1141825, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(70336, J$.G(1141857, J$.R(1141849, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(1141897, J$.M(1141889, J$.G(1141873, J$.R(1141865, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(1141881, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(1141921, J$.G(1141913, J$.R(1141905, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(1145177, J$e);
                                            } finally {
                                                if (J$.Fr(1145185))
                                                    continue jalangiLabel11;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(1142049, J$.F(1142041, J$.R(1142001, '_get', _get, false, true), false)(J$.R(1142009, 'target', target, false, false), J$.R(1142017, 'property', property, false, false), J$.C(70352, J$.R(1142025, 'receiver', receiver, false, false)) ? J$._() : J$.R(1142033, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(1145193, J$e);
                        } finally {
                            if (J$.Fr(1145201))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(1142209, arguments.callee, this, arguments);
                            arguments = J$.N(1142217, 'arguments', arguments, true, false, false);
                            object = J$.N(1142225, 'object', object, true, false, false);
                            property = J$.N(1142233, 'property', property, true, false, false);
                            while (J$.C(70368, J$.U(113146, '!', J$.M(1142137, J$.G(1142113, J$.G(1142105, J$.I(typeof Object === 'undefined' ? Object = J$.R(1142097, 'Object', undefined, true, true) : Object = J$.R(1142097, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(1142121, 'object', object, false, false), J$.R(1142129, 'property', property, false, false))))) {
                                object = J$.W(1142169, 'object', J$.F(1142161, J$.R(1142145, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1142153, 'object', object, false, false)), object, false, false);
                                if (J$.C(70360, J$.B(113154, '===', J$.R(1142177, 'object', object, false, false), J$.T(1142185, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(1142201, J$.R(1142193, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(1145209, J$e);
                        } finally {
                            if (J$.Fr(1145217))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(1142393, arguments.callee, this, arguments);
                            arguments = J$.N(1142401, 'arguments', arguments, true, false, false);
                            o = J$.N(1142409, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(1142353, '_getPrototypeOf', J$.C(70384, J$.G(1142249, J$.I(typeof Object === 'undefined' ? Object = J$.R(1142241, 'Object', undefined, true, true) : Object = J$.R(1142241, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(1142265, J$.I(typeof Object === 'undefined' ? Object = J$.R(1142257, 'Object', undefined, true, true) : Object = J$.R(1142257, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(1142345, function _getPrototypeOf(o) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(1142321, arguments.callee, this, arguments);
                                            arguments = J$.N(1142329, 'arguments', arguments, true, false, false);
                                            o = J$.N(1142337, 'o', o, true, false, false);
                                            return J$.Rt(1142313, J$.C(70376, J$.G(1142281, J$.R(1142273, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(1142305, J$.I(typeof Object === 'undefined' ? Object = J$.R(1142289, 'Object', undefined, true, true) : Object = J$.R(1142289, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(1142297, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(1145225, J$e);
                                        } finally {
                                            if (J$.Fr(1145233))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(1142385, J$.F(1142377, J$.R(1142361, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1142369, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(1145241, J$e);
                        } finally {
                            if (J$.Fr(1145249))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(1142609, arguments.callee, this, arguments);
                            arguments = J$.N(1142617, 'arguments', arguments, true, false, false);
                            subClass = J$.N(1142625, 'subClass', subClass, true, false, false);
                            superClass = J$.N(1142633, 'superClass', superClass, true, false, false);
                            if (J$.C(70400, J$.C(70392, J$.B(113170, '!==', J$.U(113162, 'typeof', J$.R(1142417, 'superClass', superClass, false, false)), J$.T(1142425, 'function', 21, false))) ? J$.B(113178, '!==', J$.R(1142433, 'superClass', superClass, false, false), J$.T(1142441, null, 25, false)) : J$._())) {
                                throw J$.F(1142465, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1142449, 'TypeError', undefined, true, true) : TypeError = J$.R(1142449, 'TypeError', TypeError, true, true)), true)(J$.T(1142457, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(1142561, J$.R(1142473, 'subClass', subClass, false, false), 'prototype', J$.M(1142553, J$.I(typeof Object === 'undefined' ? Object = J$.R(1142481, 'Object', undefined, true, true) : Object = J$.R(1142481, 'Object', Object, true, true)), 'create', false)(J$.C(70408, J$.R(1142489, 'superClass', superClass, false, false)) ? J$.G(1142505, J$.R(1142497, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(1142545, {
                                constructor: J$.T(1142537, {
                                    value: J$.R(1142513, 'subClass', subClass, false, false),
                                    writable: J$.T(1142521, true, 23, false),
                                    configurable: J$.T(1142529, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(70416, J$.R(1142569, 'superClass', superClass, false, false)))
                                J$.F(1142601, J$.R(1142577, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1142585, 'subClass', subClass, false, false), J$.R(1142593, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(1145257, J$e);
                        } finally {
                            if (J$.Fr(1145265))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(1142785, arguments.callee, this, arguments);
                            arguments = J$.N(1142793, 'arguments', arguments, true, false, false);
                            o = J$.N(1142801, 'o', o, true, false, false);
                            p = J$.N(1142809, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(1142737, '_setPrototypeOf', J$.C(70424, J$.G(1142649, J$.I(typeof Object === 'undefined' ? Object = J$.R(1142641, 'Object', undefined, true, true) : Object = J$.R(1142641, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(1142729, function _setPrototypeOf(o, p) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(1142697, arguments.callee, this, arguments);
                                            arguments = J$.N(1142705, 'arguments', arguments, true, false, false);
                                            o = J$.N(1142713, 'o', o, true, false, false);
                                            p = J$.N(1142721, 'p', p, true, false, false);
                                            J$.P(1142673, J$.R(1142657, 'o', o, false, false), '__proto__', J$.R(1142665, 'p', p, false, false));
                                            return J$.Rt(1142689, J$.R(1142681, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1145273, J$e);
                                        } finally {
                                            if (J$.Fr(1145281))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(1142777, J$.F(1142769, J$.R(1142745, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1142753, 'o', o, false, false), J$.R(1142761, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(1145289, J$e);
                        } finally {
                            if (J$.Fr(1145297))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(1144753, '_typeof2', J$.T(1144745, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(1144769, '_typeof', J$.T(1144761, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(1144785, '_classCallCheck', J$.T(1144777, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(1144801, '_defineProperties', J$.T(1144793, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(1144817, '_createClass', J$.T(1144809, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(1144833, '_possibleConstructorReturn', J$.T(1144825, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(1144849, '_assertThisInitialized', J$.T(1144841, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(1144865, '_get', J$.T(1144857, _get, 12, false), true, false, false);
            _superPropBase = J$.N(1144881, '_superPropBase', J$.T(1144873, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(1144897, '_getPrototypeOf', J$.T(1144889, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(1144913, '_inherits', J$.T(1144905, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(1144929, '_setPrototypeOf', J$.T(1144921, _setPrototypeOf, 12, false), true, false, false);
            J$.N(1144937, 'Aspect', Aspect, false, false, false);
            J$.N(1144945, 'CommandOperation', CommandOperation, false, false, false);
            J$.N(1144953, 'defineAspects', defineAspects, false, false, false);
            J$.N(1144961, 'handleCallback', handleCallback, false, false, false);
            J$.N(1144969, 'MongoError', MongoError, false, false, false);
            J$.N(1144977, 'parseIndexOptions', parseIndexOptions, false, false, false);
            J$.N(1144985, 'keysToOmit', keysToOmit, false, false, false);
            J$.N(1144993, 'CreateIndexOperation', CreateIndexOperation, false, false, false);
            J$.T(1140345, 'use strict', 21, false);
            var Aspect = J$.W(1142849, 'Aspect', J$.G(1142841, J$.F(1142833, J$.I(typeof require === 'undefined' ? require = J$.R(1142817, 'require', undefined, true, true) : require = J$.R(1142817, 'require', require, true, true)), false)(J$.T(1142825, './operation', 21, false)), 'Aspect'), Aspect, false, true);
            var CommandOperation = J$.W(1142881, 'CommandOperation', J$.F(1142873, J$.I(typeof require === 'undefined' ? require = J$.R(1142857, 'require', undefined, true, true) : require = J$.R(1142857, 'require', require, true, true)), false)(J$.T(1142865, './command', 21, false)), CommandOperation, false, true);
            var defineAspects = J$.W(1142921, 'defineAspects', J$.G(1142913, J$.F(1142905, J$.I(typeof require === 'undefined' ? require = J$.R(1142889, 'require', undefined, true, true) : require = J$.R(1142889, 'require', require, true, true)), false)(J$.T(1142897, './operation', 21, false)), 'defineAspects'), defineAspects, false, true);
            var handleCallback = J$.W(1142961, 'handleCallback', J$.G(1142953, J$.F(1142945, J$.I(typeof require === 'undefined' ? require = J$.R(1142929, 'require', undefined, true, true) : require = J$.R(1142929, 'require', require, true, true)), false)(J$.T(1142937, '../utils', 21, false)), 'handleCallback'), handleCallback, false, true);
            var MongoError = J$.W(1143001, 'MongoError', J$.G(1142993, J$.F(1142985, J$.I(typeof require === 'undefined' ? require = J$.R(1142969, 'require', undefined, true, true) : require = J$.R(1142969, 'require', require, true, true)), false)(J$.T(1142977, '../core', 21, false)), 'MongoError'), MongoError, false, true);
            var parseIndexOptions = J$.W(1143041, 'parseIndexOptions', J$.G(1143033, J$.F(1143025, J$.I(typeof require === 'undefined' ? require = J$.R(1143009, 'require', undefined, true, true) : require = J$.R(1143009, 'require', require, true, true)), false)(J$.T(1143017, '../utils', 21, false)), 'parseIndexOptions'), parseIndexOptions, false, true);
            var keysToOmit = J$.W(1143145, 'keysToOmit', J$.F(1143137, J$.I(typeof Set === 'undefined' ? Set = J$.R(1143049, 'Set', undefined, true, true) : Set = J$.R(1143049, 'Set', Set, true, true)), true)(J$.T(1143129, [
                J$.T(1143057, 'name', 21, false),
                J$.T(1143065, 'key', 21, false),
                J$.T(1143073, 'writeConcern', 21, false),
                J$.T(1143081, 'w', 21, false),
                J$.T(1143089, 'wtimeout', 21, false),
                J$.T(1143097, 'j', 21, false),
                J$.T(1143105, 'fsync', 21, false),
                J$.T(1143113, 'readPreference', 21, false),
                J$.T(1143121, 'session', 21, false)
            ], 10, false)), keysToOmit, false, true);
            var CreateIndexOperation = J$.W(1144665, 'CreateIndexOperation', J$.F(1144657, J$.T(1144641, function (_CommandOperation) {
                jalangiLabel23:
                    while (true) {
                        try {
                            J$.Fe(1144601, arguments.callee, this, arguments);
                            function CreateIndexOperation(db, name, fieldOrSpec, options) {
                                jalangiLabel19:
                                    while (true) {
                                        try {
                                            J$.Fe(1143529, arguments.callee, this, arguments);
                                            arguments = J$.N(1143537, 'arguments', arguments, true, false, false);
                                            db = J$.N(1143545, 'db', db, true, false, false);
                                            name = J$.N(1143553, 'name', name, true, false, false);
                                            fieldOrSpec = J$.N(1143561, 'fieldOrSpec', fieldOrSpec, true, false, false);
                                            options = J$.N(1143569, 'options', options, true, false, false);
                                            J$.N(1143577, '_this', _this, false, false, false);
                                            J$.N(1143585, 'indexParameters', indexParameters, false, false, false);
                                            J$.N(1143593, 'indexName', indexName, false, false, false);
                                            J$.N(1143601, 'indexesObject', indexesObject, false, false, false);
                                            var _this;
                                            J$.F(1143209, J$.R(1143185, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(1143193, 'this', this, false, false), J$.R(1143201, 'CreateIndexOperation', CreateIndexOperation, false, false));
                                            _this = J$.W(1143297, '_this', J$.F(1143289, J$.R(1143217, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(1143225, 'this', this, false, false), J$.M(1143281, J$.F(1143249, J$.R(1143233, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1143241, 'CreateIndexOperation', CreateIndexOperation, false, false)), 'call', false)(J$.R(1143257, 'this', this, false, false), J$.R(1143265, 'db', db, false, false), J$.R(1143273, 'options', options, false, false))), _this, false, false);
                                            var indexParameters = J$.W(1143329, 'indexParameters', J$.F(1143321, J$.R(1143305, 'parseIndexOptions', parseIndexOptions, false, true), false)(J$.R(1143313, 'fieldOrSpec', fieldOrSpec, false, false)), indexParameters, false, false);
                                            var indexName = J$.W(1143393, 'indexName', J$.C(70432, J$.B(113194, '===', J$.U(113186, 'typeof', J$.G(1143345, J$.R(1143337, 'options', options, false, false), 'name')), J$.T(1143353, 'string', 21, false))) ? J$.G(1143369, J$.R(1143361, 'options', options, false, false), 'name') : J$.G(1143385, J$.R(1143377, 'indexParameters', indexParameters, false, false), 'name'), indexName, false, false);
                                            var indexesObject = J$.W(1143433, 'indexesObject', J$.T(1143425, {
                                                name: J$.R(1143401, 'indexName', indexName, false, false),
                                                key: J$.G(1143417, J$.R(1143409, 'indexParameters', indexParameters, false, false), 'fieldHash')
                                            }, 11, false), indexesObject, false, false);
                                            J$.P(1143457, J$.R(1143441, '_this', _this, false, false), 'name', J$.R(1143449, 'name', name, false, false));
                                            J$.P(1143481, J$.R(1143465, '_this', _this, false, false), 'fieldOrSpec', J$.R(1143473, 'fieldOrSpec', fieldOrSpec, false, false));
                                            J$.P(1143505, J$.R(1143489, '_this', _this, false, false), 'indexes', J$.R(1143497, 'indexesObject', indexesObject, false, false));
                                            return J$.Rt(1143521, J$.R(1143513, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1145305, J$e);
                                        } finally {
                                            if (J$.Fr(1145313))
                                                continue jalangiLabel19;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(1144609, 'arguments', arguments, true, false, false);
                            _CommandOperation = J$.N(1144617, '_CommandOperation', _CommandOperation, true, false, false);
                            CreateIndexOperation = J$.N(1144633, 'CreateIndexOperation', J$.T(1144625, CreateIndexOperation, 12, false), true, false, false);
                            J$.F(1143177, J$.R(1143153, '_inherits', _inherits, false, true), false)(J$.R(1143161, 'CreateIndexOperation', CreateIndexOperation, false, false), J$.R(1143169, '_CommandOperation', _CommandOperation, false, false));
                            J$.F(1144577, J$.R(1143609, '_createClass', _createClass, false, true), false)(J$.R(1143617, 'CreateIndexOperation', CreateIndexOperation, false, false), J$.T(1144569, [
                                J$.T(1143921, {
                                    key: J$.T(1143625, '_buildCommand', 21, false),
                                    value: J$.T(1143913, function _buildCommand() {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(1143857, arguments.callee, this, arguments);
                                                    arguments = J$.N(1143865, 'arguments', arguments, true, false, false);
                                                    J$.N(1143873, 'options', options, false, false, false);
                                                    J$.N(1143881, 'name', name, false, false, false);
                                                    J$.N(1143889, 'indexes', indexes, false, false, false);
                                                    J$.N(1143897, 'optionName', optionName, false, false, false);
                                                    J$.N(1143905, 'cmd', cmd, false, false, false);
                                                    var options = J$.W(1143649, 'options', J$.G(1143641, J$.R(1143633, 'this', this, false, false), 'options'), options, false, false);
                                                    var name = J$.W(1143673, 'name', J$.G(1143665, J$.R(1143657, 'this', this, false, false), 'name'), name, false, false);
                                                    var indexes = J$.W(1143697, 'indexes', J$.G(1143689, J$.R(1143681, 'this', this, false, false), 'indexes'), indexes, false, false);
                                                    for (var optionName in J$.H(1143785, J$.R(1143705, 'options', options, false, false))) {
                                                        J$.N(1143793, 'optionName', optionName, false, true, false);
                                                        {
                                                            {
                                                                if (J$.C(70440, J$.U(113202, '!', J$.M(1143729, J$.R(1143713, 'keysToOmit', keysToOmit, false, true), 'has', false)(J$.R(1143721, 'optionName', optionName, false, false))))) {
                                                                    J$.P(1143777, J$.R(1143737, 'indexes', indexes, false, false), J$.R(1143745, 'optionName', optionName, false, false), J$.G(1143769, J$.R(1143753, 'options', options, false, false), J$.R(1143761, 'optionName', optionName, false, false)));
                                                                }
                                                            }
                                                        }
                                                    }
                                                    var cmd = J$.W(1143833, 'cmd', J$.T(1143825, {
                                                        createIndexes: J$.R(1143801, 'name', name, false, false),
                                                        indexes: J$.T(1143817, [J$.R(1143809, 'indexes', indexes, false, false)], 10, false)
                                                    }, 11, false), cmd, false, false);
                                                    return J$.Rt(1143849, J$.R(1143841, 'cmd', cmd, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1145321, J$e);
                                                } finally {
                                                    if (J$.Fr(1145329))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1144561, {
                                    key: J$.T(1143929, 'execute', 21, false),
                                    value: J$.T(1144553, function execute(callback) {
                                        jalangiLabel22:
                                            while (true) {
                                                try {
                                                    J$.Fe(1144489, arguments.callee, this, arguments);
                                                    arguments = J$.N(1144497, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(1144505, 'callback', callback, true, false, false);
                                                    J$.N(1144513, 'db', db, false, false, false);
                                                    J$.N(1144521, 'options', options, false, false, false);
                                                    J$.N(1144529, 'indexes', indexes, false, false, false);
                                                    J$.N(1144537, 'capabilities', capabilities, false, false, false);
                                                    J$.N(1144545, 'error', error, false, false, false);
                                                    var db = J$.W(1143953, 'db', J$.G(1143945, J$.R(1143937, 'this', this, false, false), 'db'), db, false, false);
                                                    var options = J$.W(1143977, 'options', J$.G(1143969, J$.R(1143961, 'this', this, false, false), 'options'), options, false, false);
                                                    var indexes = J$.W(1144001, 'indexes', J$.G(1143993, J$.R(1143985, 'this', this, false, false), 'indexes'), indexes, false, false);
                                                    var capabilities = J$.W(1144041, 'capabilities', J$.M(1144033, J$.G(1144025, J$.G(1144017, J$.R(1144009, 'db', db, false, false), 's'), 'topology'), 'capabilities', false)(), capabilities, false, false);
                                                    if (J$.C(70464, J$.C(70456, J$.C(70448, J$.G(1144057, J$.R(1144049, 'options', options, false, false), 'collation')) ? J$.R(1144065, 'capabilities', capabilities, false, false) : J$._()) ? J$.U(113210, '!', J$.G(1144081, J$.R(1144073, 'capabilities', capabilities, false, false), 'commandsTakeCollation')) : J$._())) {
                                                        var error = J$.W(1144113, 'error', J$.F(1144105, J$.R(1144089, 'MongoError', MongoError, false, true), true)(J$.T(1144097, 'server/primary/mongos does not support collation', 21, false)), error, false, false);
                                                        J$.P(1144137, J$.R(1144121, 'error', error, false, false), 'code', J$.T(1144129, 67, 22, false));
                                                        return J$.Rt(1144169, J$.F(1144161, J$.R(1144145, 'callback', callback, false, false), false)(J$.R(1144153, 'error', error, false, false)));
                                                    }
                                                    if (J$.C(70480, J$.C(70472, J$.G(1144185, J$.R(1144177, 'options', options, false, false), 'writeConcern')) ? J$.B(113226, '!==', J$.U(113218, 'typeof', J$.R(1144193, 'callback', callback, false, false)), J$.T(1144201, 'function', 21, false)) : J$._())) {
                                                        throw J$.M(1144241, J$.R(1144209, 'MongoError', MongoError, false, true), 'create', false)(J$.T(1144233, {
                                                            message: J$.T(1144217, 'Cannot use a writeConcern without a provided callback', 21, false),
                                                            driver: J$.T(1144225, true, 23, false)
                                                        }, 11, false));
                                                    }
                                                    J$.M(1144481, J$.F(1144305, J$.R(1144249, '_get', _get, false, true), false)(J$.F(1144281, J$.R(1144257, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(1144273, J$.R(1144265, 'CreateIndexOperation', CreateIndexOperation, false, false), 'prototype')), J$.T(1144289, 'execute', 21, false), J$.R(1144297, 'this', this, false, false)), 'call', false)(J$.R(1144313, 'this', this, false, false), J$.T(1144473, function (err, result) {
                                                        jalangiLabel21:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(1144441, arguments.callee, this, arguments);
                                                                    arguments = J$.N(1144449, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(1144457, 'err', err, true, false, false);
                                                                    result = J$.N(1144465, 'result', result, true, false, false);
                                                                    if (J$.C(70488, J$.B(113234, '==', J$.R(1144321, 'err', err, false, false), J$.T(1144329, null, 25, false))))
                                                                        return J$.Rt(1144385, J$.F(1144377, J$.R(1144337, 'handleCallback', handleCallback, false, true), false)(J$.R(1144345, 'callback', callback, false, false), J$.R(1144353, 'err', err, false, false), J$.G(1144369, J$.R(1144361, 'indexes', indexes, false, false), 'name')));
                                                                    return J$.Rt(1144433, J$.F(1144425, J$.R(1144393, 'handleCallback', handleCallback, false, true), false)(J$.R(1144401, 'callback', callback, false, false), J$.R(1144409, 'err', err, false, false), J$.R(1144417, 'result', result, false, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(1145337, J$e);
                                                                } finally {
                                                                    if (J$.Fr(1145345))
                                                                        continue jalangiLabel21;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(1145353, J$e);
                                                } finally {
                                                    if (J$.Fr(1145361))
                                                        continue jalangiLabel22;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(1144593, J$.R(1144585, 'CreateIndexOperation', CreateIndexOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(1145369, J$e);
                        } finally {
                            if (J$.Fr(1145377))
                                continue jalangiLabel23;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(1144649, 'CommandOperation', CommandOperation, false, true)), CreateIndexOperation, false, true);
            J$.F(1144705, J$.R(1144673, 'defineAspects', defineAspects, false, true), false)(J$.R(1144681, 'CreateIndexOperation', CreateIndexOperation, false, true), J$.G(1144697, J$.R(1144689, 'Aspect', Aspect, false, true), 'WRITE_OPERATION'));
            J$.P(1144729, J$.I(typeof module === 'undefined' ? module = J$.R(1144713, 'module', undefined, true, true) : module = J$.R(1144713, 'module', module, true, true)), 'exports', J$.R(1144721, 'CreateIndexOperation', CreateIndexOperation, false, true));
        } catch (J$e) {
            J$.Ex(1145385, J$e);
        } finally {
            if (J$.Sr(1145393))
                continue jalangiLabel24;
            else
                break jalangiLabel24;
        }
    }
// JALANGI DO NOT INSTRUMENT
