// mongodb/lib/operations/drop_index.js
J$.noInstrEval = false;
jalangiLabel24:
    while (true) {
        try {
            J$.Se(1190321, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/operations/drop_index.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(1186889, arguments.callee, this, arguments);
                            arguments = J$.N(1186897, 'arguments', arguments, true, false, false);
                            obj = J$.N(1186905, 'obj', obj, true, false, false);
                            if (J$.C(73408, J$.C(73368, J$.B(115922, '===', J$.U(115914, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1186625, 'Symbol', undefined, true, true) : Symbol = J$.R(1186625, 'Symbol', Symbol, true, true))), J$.T(1186633, 'function', 21, false))) ? J$.B(115938, '===', J$.U(115930, 'typeof', J$.G(1186649, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1186641, 'Symbol', undefined, true, true) : Symbol = J$.R(1186641, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1186657, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(1186713, '_typeof2', J$.T(1186705, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(1186681, arguments.callee, this, arguments);
                                                arguments = J$.N(1186689, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1186697, 'obj', obj, true, false, false);
                                                return J$.Rt(1186673, J$.U(115946, 'typeof', J$.R(1186665, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1190569, J$e);
                                            } finally {
                                                if (J$.Fr(1190577))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(1186849, '_typeof2', J$.T(1186841, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(1186817, arguments.callee, this, arguments);
                                                arguments = J$.N(1186825, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1186833, 'obj', obj, true, false, false);
                                                return J$.Rt(1186809, J$.C(73400, J$.C(73392, J$.C(73384, J$.C(73376, J$.R(1186721, 'obj', obj, false, false)) ? J$.B(115962, '===', J$.U(115954, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1186729, 'Symbol', undefined, true, true) : Symbol = J$.R(1186729, 'Symbol', Symbol, true, true))), J$.T(1186737, 'function', 21, false)) : J$._()) ? J$.B(115970, '===', J$.G(1186753, J$.R(1186745, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1186761, 'Symbol', undefined, true, true) : Symbol = J$.R(1186761, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(115978, '!==', J$.R(1186769, 'obj', obj, false, false), J$.G(1186785, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1186777, 'Symbol', undefined, true, true) : Symbol = J$.R(1186777, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1186793, 'symbol', 21, false) : J$.U(115986, 'typeof', J$.R(1186801, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1190585, J$e);
                                            } finally {
                                                if (J$.Fr(1190593))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(1186881, J$.F(1186873, J$.R(1186857, '_typeof2', _typeof2, false, true), false)(J$.R(1186865, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1190601, J$e);
                        } finally {
                            if (J$.Fr(1190609))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(1187225, arguments.callee, this, arguments);
                            arguments = J$.N(1187233, 'arguments', arguments, true, false, false);
                            obj = J$.N(1187241, 'obj', obj, true, false, false);
                            if (J$.C(73456, J$.C(73416, J$.B(116002, '===', J$.U(115994, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1186913, 'Symbol', undefined, true, true) : Symbol = J$.R(1186913, 'Symbol', Symbol, true, true))), J$.T(1186921, 'function', 21, false))) ? J$.B(116010, '===', J$.F(1186953, J$.R(1186929, '_typeof2', _typeof2, false, true), false)(J$.G(1186945, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1186937, 'Symbol', undefined, true, true) : Symbol = J$.R(1186937, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1186961, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(1187033, '_typeof', J$.T(1187025, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(1187001, arguments.callee, this, arguments);
                                                arguments = J$.N(1187009, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1187017, 'obj', obj, true, false, false);
                                                return J$.Rt(1186993, J$.F(1186985, J$.R(1186969, '_typeof2', _typeof2, false, true), false)(J$.R(1186977, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1190617, J$e);
                                            } finally {
                                                if (J$.Fr(1190625))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(1187185, '_typeof', J$.T(1187177, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(1187153, arguments.callee, this, arguments);
                                                arguments = J$.N(1187161, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1187169, 'obj', obj, true, false, false);
                                                return J$.Rt(1187145, J$.C(73448, J$.C(73440, J$.C(73432, J$.C(73424, J$.R(1187041, 'obj', obj, false, false)) ? J$.B(116026, '===', J$.U(116018, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1187049, 'Symbol', undefined, true, true) : Symbol = J$.R(1187049, 'Symbol', Symbol, true, true))), J$.T(1187057, 'function', 21, false)) : J$._()) ? J$.B(116034, '===', J$.G(1187073, J$.R(1187065, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1187081, 'Symbol', undefined, true, true) : Symbol = J$.R(1187081, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(116042, '!==', J$.R(1187089, 'obj', obj, false, false), J$.G(1187105, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1187097, 'Symbol', undefined, true, true) : Symbol = J$.R(1187097, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1187113, 'symbol', 21, false) : J$.F(1187137, J$.R(1187121, '_typeof2', _typeof2, false, true), false)(J$.R(1187129, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1190633, J$e);
                                            } finally {
                                                if (J$.Fr(1190641))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(1187217, J$.F(1187209, J$.R(1187193, '_typeof', _typeof, false, true), false)(J$.R(1187201, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1190649, J$e);
                        } finally {
                            if (J$.Fr(1190657))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(1187289, arguments.callee, this, arguments);
                            arguments = J$.N(1187297, 'arguments', arguments, true, false, false);
                            instance = J$.N(1187305, 'instance', instance, true, false, false);
                            Constructor = J$.N(1187313, 'Constructor', Constructor, true, false, false);
                            if (J$.C(73464, J$.U(116058, '!', J$.B(116050, 'instanceof', J$.R(1187249, 'instance', instance, false, false), J$.R(1187257, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(1187281, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1187265, 'TypeError', undefined, true, true) : TypeError = J$.R(1187265, 'TypeError', TypeError, true, true)), true)(J$.T(1187273, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1190665, J$e);
                        } finally {
                            if (J$.Fr(1190673))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(1187561, arguments.callee, this, arguments);
                            arguments = J$.N(1187569, 'arguments', arguments, true, false, false);
                            target = J$.N(1187577, 'target', target, true, false, false);
                            props = J$.N(1187585, 'props', props, true, false, false);
                            J$.N(1187593, 'i', i, false, false, false);
                            J$.N(1187601, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(1187329, 'i', J$.T(1187321, 0, 22, false), i, false, false); J$.C(73488, J$.B(116066, '<', J$.R(1187337, 'i', i, false, false), J$.G(1187353, J$.R(1187345, 'props', props, false, false), 'length'))); J$.B(116090, '-', i = J$.W(1187369, 'i', J$.B(116082, '+', J$.U(116074, '+', J$.R(1187361, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(1187401, 'descriptor', J$.G(1187393, J$.R(1187377, 'props', props, false, false), J$.R(1187385, 'i', i, false, false)), descriptor, false, false);
                                J$.P(1187441, J$.R(1187409, 'descriptor', descriptor, false, false), 'enumerable', J$.C(73472, J$.G(1187425, J$.R(1187417, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(1187433, false, 23, false));
                                J$.P(1187465, J$.R(1187449, 'descriptor', descriptor, false, false), 'configurable', J$.T(1187457, true, 23, false));
                                if (J$.C(73480, J$.B(116098, 'in', J$.T(1187473, 'value', 21, false), J$.R(1187481, 'descriptor', descriptor, false, false))))
                                    J$.P(1187505, J$.R(1187489, 'descriptor', descriptor, false, false), 'writable', J$.T(1187497, true, 23, false));
                                J$.M(1187553, J$.I(typeof Object === 'undefined' ? Object = J$.R(1187513, 'Object', undefined, true, true) : Object = J$.R(1187513, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(1187521, 'target', target, false, false), J$.G(1187537, J$.R(1187529, 'descriptor', descriptor, false, false), 'key'), J$.R(1187545, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1190681, J$e);
                        } finally {
                            if (J$.Fr(1190689))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(1187713, arguments.callee, this, arguments);
                            arguments = J$.N(1187721, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(1187729, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(1187737, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(1187745, 'staticProps', staticProps, true, false, false);
                            if (J$.C(73496, J$.R(1187609, 'protoProps', protoProps, false, false)))
                                J$.F(1187649, J$.R(1187617, '_defineProperties', _defineProperties, false, true), false)(J$.G(1187633, J$.R(1187625, 'Constructor', Constructor, false, false), 'prototype'), J$.R(1187641, 'protoProps', protoProps, false, false));
                            if (J$.C(73504, J$.R(1187657, 'staticProps', staticProps, false, false)))
                                J$.F(1187689, J$.R(1187665, '_defineProperties', _defineProperties, false, true), false)(J$.R(1187673, 'Constructor', Constructor, false, false), J$.R(1187681, 'staticProps', staticProps, false, false));
                            return J$.Rt(1187705, J$.R(1187697, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(1190697, J$e);
                        } finally {
                            if (J$.Fr(1190705))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(1187857, arguments.callee, this, arguments);
                            arguments = J$.N(1187865, 'arguments', arguments, true, false, false);
                            self = J$.N(1187873, 'self', self, true, false, false);
                            call = J$.N(1187881, 'call', call, true, false, false);
                            if (J$.C(73528, J$.C(73520, J$.R(1187753, 'call', call, false, false)) ? J$.C(73512, J$.B(116106, '===', J$.F(1187777, J$.R(1187761, '_typeof', _typeof, false, true), false)(J$.R(1187769, 'call', call, false, false)), J$.T(1187785, 'object', 21, false))) ? J$._() : J$.B(116122, '===', J$.U(116114, 'typeof', J$.R(1187793, 'call', call, false, false)), J$.T(1187801, 'function', 21, false)) : J$._())) {
                                return J$.Rt(1187817, J$.R(1187809, 'call', call, false, false));
                            }
                            return J$.Rt(1187849, J$.F(1187841, J$.R(1187825, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(1187833, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(1190713, J$e);
                        } finally {
                            if (J$.Fr(1190721))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(1187945, arguments.callee, this, arguments);
                            arguments = J$.N(1187953, 'arguments', arguments, true, false, false);
                            self = J$.N(1187961, 'self', self, true, false, false);
                            if (J$.C(73536, J$.B(116130, '===', J$.R(1187889, 'self', self, false, false), void J$.T(1187897, 0, 22, false)))) {
                                throw J$.F(1187921, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(1187905, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(1187905, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(1187913, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(1187937, J$.R(1187929, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(1190729, J$e);
                        } finally {
                            if (J$.Fr(1190737))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(1188329, arguments.callee, this, arguments);
                            arguments = J$.N(1188337, 'arguments', arguments, true, false, false);
                            target = J$.N(1188345, 'target', target, true, false, false);
                            property = J$.N(1188353, 'property', property, true, false, false);
                            receiver = J$.N(1188361, 'receiver', receiver, true, false, false);
                            if (J$.C(73568, J$.C(73544, J$.B(116146, '!==', J$.U(116138, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1187969, 'Reflect', undefined, true, true) : Reflect = J$.R(1187969, 'Reflect', Reflect, true, true))), J$.T(1187977, 'undefined', 21, false))) ? J$.G(1187993, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1187985, 'Reflect', undefined, true, true) : Reflect = J$.R(1187985, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(1188017, '_get', J$.G(1188009, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1188001, 'Reflect', undefined, true, true) : Reflect = J$.R(1188001, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(1188265, '_get', J$.T(1188257, function _get(target, property, receiver) {
                                    jalangiLabel11:
                                        while (true) {
                                            try {
                                                J$.Fe(1188201, arguments.callee, this, arguments);
                                                arguments = J$.N(1188209, 'arguments', arguments, true, false, false);
                                                target = J$.N(1188217, 'target', target, true, false, false);
                                                property = J$.N(1188225, 'property', property, true, false, false);
                                                receiver = J$.N(1188233, 'receiver', receiver, true, false, false);
                                                J$.N(1188241, 'base', base, false, false, false);
                                                J$.N(1188249, 'desc', desc, false, false, false);
                                                var base = J$.W(1188057, 'base', J$.F(1188049, J$.R(1188025, '_superPropBase', _superPropBase, false, true), false)(J$.R(1188033, 'target', target, false, false), J$.R(1188041, 'property', property, false, false)), base, false, false);
                                                if (J$.C(73552, J$.U(116154, '!', J$.R(1188065, 'base', base, false, false))))
                                                    return J$.Rt(1188073, undefined);
                                                var desc = J$.W(1188113, 'desc', J$.M(1188105, J$.I(typeof Object === 'undefined' ? Object = J$.R(1188081, 'Object', undefined, true, true) : Object = J$.R(1188081, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(1188089, 'base', base, false, false), J$.R(1188097, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(73560, J$.G(1188129, J$.R(1188121, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(1188169, J$.M(1188161, J$.G(1188145, J$.R(1188137, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(1188153, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(1188193, J$.G(1188185, J$.R(1188177, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(1190745, J$e);
                                            } finally {
                                                if (J$.Fr(1190753))
                                                    continue jalangiLabel11;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(1188321, J$.F(1188313, J$.R(1188273, '_get', _get, false, true), false)(J$.R(1188281, 'target', target, false, false), J$.R(1188289, 'property', property, false, false), J$.C(73576, J$.R(1188297, 'receiver', receiver, false, false)) ? J$._() : J$.R(1188305, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(1190761, J$e);
                        } finally {
                            if (J$.Fr(1190769))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(1188481, arguments.callee, this, arguments);
                            arguments = J$.N(1188489, 'arguments', arguments, true, false, false);
                            object = J$.N(1188497, 'object', object, true, false, false);
                            property = J$.N(1188505, 'property', property, true, false, false);
                            while (J$.C(73592, J$.U(116162, '!', J$.M(1188409, J$.G(1188385, J$.G(1188377, J$.I(typeof Object === 'undefined' ? Object = J$.R(1188369, 'Object', undefined, true, true) : Object = J$.R(1188369, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(1188393, 'object', object, false, false), J$.R(1188401, 'property', property, false, false))))) {
                                object = J$.W(1188441, 'object', J$.F(1188433, J$.R(1188417, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1188425, 'object', object, false, false)), object, false, false);
                                if (J$.C(73584, J$.B(116170, '===', J$.R(1188449, 'object', object, false, false), J$.T(1188457, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(1188473, J$.R(1188465, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(1190777, J$e);
                        } finally {
                            if (J$.Fr(1190785))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(1188665, arguments.callee, this, arguments);
                            arguments = J$.N(1188673, 'arguments', arguments, true, false, false);
                            o = J$.N(1188681, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(1188625, '_getPrototypeOf', J$.C(73608, J$.G(1188521, J$.I(typeof Object === 'undefined' ? Object = J$.R(1188513, 'Object', undefined, true, true) : Object = J$.R(1188513, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(1188537, J$.I(typeof Object === 'undefined' ? Object = J$.R(1188529, 'Object', undefined, true, true) : Object = J$.R(1188529, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(1188617, function _getPrototypeOf(o) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(1188593, arguments.callee, this, arguments);
                                            arguments = J$.N(1188601, 'arguments', arguments, true, false, false);
                                            o = J$.N(1188609, 'o', o, true, false, false);
                                            return J$.Rt(1188585, J$.C(73600, J$.G(1188553, J$.R(1188545, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(1188577, J$.I(typeof Object === 'undefined' ? Object = J$.R(1188561, 'Object', undefined, true, true) : Object = J$.R(1188561, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(1188569, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(1190793, J$e);
                                        } finally {
                                            if (J$.Fr(1190801))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(1188657, J$.F(1188649, J$.R(1188633, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1188641, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(1190809, J$e);
                        } finally {
                            if (J$.Fr(1190817))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(1188881, arguments.callee, this, arguments);
                            arguments = J$.N(1188889, 'arguments', arguments, true, false, false);
                            subClass = J$.N(1188897, 'subClass', subClass, true, false, false);
                            superClass = J$.N(1188905, 'superClass', superClass, true, false, false);
                            if (J$.C(73624, J$.C(73616, J$.B(116186, '!==', J$.U(116178, 'typeof', J$.R(1188689, 'superClass', superClass, false, false)), J$.T(1188697, 'function', 21, false))) ? J$.B(116194, '!==', J$.R(1188705, 'superClass', superClass, false, false), J$.T(1188713, null, 25, false)) : J$._())) {
                                throw J$.F(1188737, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1188721, 'TypeError', undefined, true, true) : TypeError = J$.R(1188721, 'TypeError', TypeError, true, true)), true)(J$.T(1188729, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(1188833, J$.R(1188745, 'subClass', subClass, false, false), 'prototype', J$.M(1188825, J$.I(typeof Object === 'undefined' ? Object = J$.R(1188753, 'Object', undefined, true, true) : Object = J$.R(1188753, 'Object', Object, true, true)), 'create', false)(J$.C(73632, J$.R(1188761, 'superClass', superClass, false, false)) ? J$.G(1188777, J$.R(1188769, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(1188817, {
                                constructor: J$.T(1188809, {
                                    value: J$.R(1188785, 'subClass', subClass, false, false),
                                    writable: J$.T(1188793, true, 23, false),
                                    configurable: J$.T(1188801, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(73640, J$.R(1188841, 'superClass', superClass, false, false)))
                                J$.F(1188873, J$.R(1188849, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1188857, 'subClass', subClass, false, false), J$.R(1188865, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(1190825, J$e);
                        } finally {
                            if (J$.Fr(1190833))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(1189057, arguments.callee, this, arguments);
                            arguments = J$.N(1189065, 'arguments', arguments, true, false, false);
                            o = J$.N(1189073, 'o', o, true, false, false);
                            p = J$.N(1189081, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(1189009, '_setPrototypeOf', J$.C(73648, J$.G(1188921, J$.I(typeof Object === 'undefined' ? Object = J$.R(1188913, 'Object', undefined, true, true) : Object = J$.R(1188913, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(1189001, function _setPrototypeOf(o, p) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(1188969, arguments.callee, this, arguments);
                                            arguments = J$.N(1188977, 'arguments', arguments, true, false, false);
                                            o = J$.N(1188985, 'o', o, true, false, false);
                                            p = J$.N(1188993, 'p', p, true, false, false);
                                            J$.P(1188945, J$.R(1188929, 'o', o, false, false), '__proto__', J$.R(1188937, 'p', p, false, false));
                                            return J$.Rt(1188961, J$.R(1188953, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1190841, J$e);
                                        } finally {
                                            if (J$.Fr(1190849))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(1189049, J$.F(1189041, J$.R(1189017, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1189025, 'o', o, false, false), J$.R(1189033, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(1190857, J$e);
                        } finally {
                            if (J$.Fr(1190865))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(1190337, '_typeof2', J$.T(1190329, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(1190353, '_typeof', J$.T(1190345, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(1190369, '_classCallCheck', J$.T(1190361, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(1190385, '_defineProperties', J$.T(1190377, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(1190401, '_createClass', J$.T(1190393, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(1190417, '_possibleConstructorReturn', J$.T(1190409, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(1190433, '_assertThisInitialized', J$.T(1190425, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(1190449, '_get', J$.T(1190441, _get, 12, false), true, false, false);
            _superPropBase = J$.N(1190465, '_superPropBase', J$.T(1190457, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(1190481, '_getPrototypeOf', J$.T(1190473, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(1190497, '_inherits', J$.T(1190489, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(1190513, '_setPrototypeOf', J$.T(1190505, _setPrototypeOf, 12, false), true, false, false);
            J$.N(1190521, 'Aspect', Aspect, false, false, false);
            J$.N(1190529, 'defineAspects', defineAspects, false, false, false);
            J$.N(1190537, 'CommandOperation', CommandOperation, false, false, false);
            J$.N(1190545, 'applyWriteConcern', applyWriteConcern, false, false, false);
            J$.N(1190553, 'handleCallback', handleCallback, false, false, false);
            J$.N(1190561, 'DropIndexOperation', DropIndexOperation, false, false, false);
            J$.T(1186617, 'use strict', 21, false);
            var Aspect = J$.W(1189121, 'Aspect', J$.G(1189113, J$.F(1189105, J$.I(typeof require === 'undefined' ? require = J$.R(1189089, 'require', undefined, true, true) : require = J$.R(1189089, 'require', require, true, true)), false)(J$.T(1189097, './operation', 21, false)), 'Aspect'), Aspect, false, true);
            var defineAspects = J$.W(1189161, 'defineAspects', J$.G(1189153, J$.F(1189145, J$.I(typeof require === 'undefined' ? require = J$.R(1189129, 'require', undefined, true, true) : require = J$.R(1189129, 'require', require, true, true)), false)(J$.T(1189137, './operation', 21, false)), 'defineAspects'), defineAspects, false, true);
            var CommandOperation = J$.W(1189193, 'CommandOperation', J$.F(1189185, J$.I(typeof require === 'undefined' ? require = J$.R(1189169, 'require', undefined, true, true) : require = J$.R(1189169, 'require', require, true, true)), false)(J$.T(1189177, './command', 21, false)), CommandOperation, false, true);
            var applyWriteConcern = J$.W(1189233, 'applyWriteConcern', J$.G(1189225, J$.F(1189217, J$.I(typeof require === 'undefined' ? require = J$.R(1189201, 'require', undefined, true, true) : require = J$.R(1189201, 'require', require, true, true)), false)(J$.T(1189209, '../utils', 21, false)), 'applyWriteConcern'), applyWriteConcern, false, true);
            var handleCallback = J$.W(1189273, 'handleCallback', J$.G(1189265, J$.F(1189257, J$.I(typeof require === 'undefined' ? require = J$.R(1189241, 'require', undefined, true, true) : require = J$.R(1189241, 'require', require, true, true)), false)(J$.T(1189249, '../utils', 21, false)), 'handleCallback'), handleCallback, false, true);
            var DropIndexOperation = J$.W(1190249, 'DropIndexOperation', J$.F(1190241, J$.T(1190225, function (_CommandOperation) {
                jalangiLabel23:
                    while (true) {
                        try {
                            J$.Fe(1190185, arguments.callee, this, arguments);
                            function DropIndexOperation(collection, indexName, options) {
                                jalangiLabel19:
                                    while (true) {
                                        try {
                                            J$.Fe(1189521, arguments.callee, this, arguments);
                                            arguments = J$.N(1189529, 'arguments', arguments, true, false, false);
                                            collection = J$.N(1189537, 'collection', collection, true, false, false);
                                            indexName = J$.N(1189545, 'indexName', indexName, true, false, false);
                                            options = J$.N(1189553, 'options', options, true, false, false);
                                            J$.N(1189561, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(1189337, J$.R(1189313, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(1189321, 'this', this, false, false), J$.R(1189329, 'DropIndexOperation', DropIndexOperation, false, false));
                                            _this = J$.W(1189449, '_this', J$.F(1189441, J$.R(1189345, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(1189353, 'this', this, false, false), J$.M(1189433, J$.F(1189377, J$.R(1189361, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1189369, 'DropIndexOperation', DropIndexOperation, false, false)), 'call', false)(J$.R(1189385, 'this', this, false, false), J$.G(1189409, J$.G(1189401, J$.R(1189393, 'collection', collection, false, false), 's'), 'db'), J$.R(1189417, 'options', options, false, false), J$.R(1189425, 'collection', collection, false, false))), _this, false, false);
                                            J$.P(1189473, J$.R(1189457, '_this', _this, false, false), 'collection', J$.R(1189465, 'collection', collection, false, false));
                                            J$.P(1189497, J$.R(1189481, '_this', _this, false, false), 'indexName', J$.R(1189489, 'indexName', indexName, false, false));
                                            return J$.Rt(1189513, J$.R(1189505, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1190873, J$e);
                                        } finally {
                                            if (J$.Fr(1190881))
                                                continue jalangiLabel19;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(1190193, 'arguments', arguments, true, false, false);
                            _CommandOperation = J$.N(1190201, '_CommandOperation', _CommandOperation, true, false, false);
                            DropIndexOperation = J$.N(1190217, 'DropIndexOperation', J$.T(1190209, DropIndexOperation, 12, false), true, false, false);
                            J$.F(1189305, J$.R(1189281, '_inherits', _inherits, false, true), false)(J$.R(1189289, 'DropIndexOperation', DropIndexOperation, false, false), J$.R(1189297, '_CommandOperation', _CommandOperation, false, false));
                            J$.F(1190161, J$.R(1189569, '_createClass', _createClass, false, true), false)(J$.R(1189577, 'DropIndexOperation', DropIndexOperation, false, false), J$.T(1190153, [
                                J$.T(1189857, {
                                    key: J$.T(1189585, '_buildCommand', 21, false),
                                    value: J$.T(1189849, function _buildCommand() {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(1189801, arguments.callee, this, arguments);
                                                    arguments = J$.N(1189809, 'arguments', arguments, true, false, false);
                                                    J$.N(1189817, 'collection', collection, false, false, false);
                                                    J$.N(1189825, 'indexName', indexName, false, false, false);
                                                    J$.N(1189833, 'options', options, false, false, false);
                                                    J$.N(1189841, 'cmd', cmd, false, false, false);
                                                    var collection = J$.W(1189609, 'collection', J$.G(1189601, J$.R(1189593, 'this', this, false, false), 'collection'), collection, false, false);
                                                    var indexName = J$.W(1189633, 'indexName', J$.G(1189625, J$.R(1189617, 'this', this, false, false), 'indexName'), indexName, false, false);
                                                    var options = J$.W(1189657, 'options', J$.G(1189649, J$.R(1189641, 'this', this, false, false), 'options'), options, false, false);
                                                    var cmd = J$.W(1189697, 'cmd', J$.T(1189689, {
                                                        dropIndexes: J$.G(1189673, J$.R(1189665, 'collection', collection, false, false), 'collectionName'),
                                                        index: J$.R(1189681, 'indexName', indexName, false, false)
                                                    }, 11, false), cmd, false, false);
                                                    cmd = J$.W(1189777, 'cmd', J$.F(1189769, J$.R(1189705, 'applyWriteConcern', applyWriteConcern, false, true), false)(J$.R(1189713, 'cmd', cmd, false, false), J$.T(1189753, {
                                                        db: J$.G(1189737, J$.G(1189729, J$.R(1189721, 'collection', collection, false, false), 's'), 'db'),
                                                        collection: J$.R(1189745, 'collection', collection, false, false)
                                                    }, 11, false), J$.R(1189761, 'options', options, false, false)), cmd, false, false);
                                                    return J$.Rt(1189793, J$.R(1189785, 'cmd', cmd, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1190889, J$e);
                                                } finally {
                                                    if (J$.Fr(1190897))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1190145, {
                                    key: J$.T(1189865, 'execute', 21, false),
                                    value: J$.T(1190137, function execute(callback) {
                                        jalangiLabel22:
                                            while (true) {
                                                try {
                                                    J$.Fe(1190113, arguments.callee, this, arguments);
                                                    arguments = J$.N(1190121, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(1190129, 'callback', callback, true, false, false);
                                                    J$.M(1190105, J$.F(1189929, J$.R(1189873, '_get', _get, false, true), false)(J$.F(1189905, J$.R(1189881, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(1189897, J$.R(1189889, 'DropIndexOperation', DropIndexOperation, false, false), 'prototype')), J$.T(1189913, 'execute', 21, false), J$.R(1189921, 'this', this, false, false)), 'call', false)(J$.R(1189937, 'this', this, false, false), J$.T(1190097, function (err, result) {
                                                        jalangiLabel21:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(1190065, arguments.callee, this, arguments);
                                                                    arguments = J$.N(1190073, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(1190081, 'err', err, true, false, false);
                                                                    result = J$.N(1190089, 'result', result, true, false, false);
                                                                    if (J$.C(73656, J$.B(116210, '!==', J$.U(116202, 'typeof', J$.R(1189945, 'callback', callback, false, false)), J$.T(1189953, 'function', 21, false))))
                                                                        return J$.Rt(1189961, undefined);
                                                                    if (J$.C(73664, J$.R(1189969, 'err', err, false, false)))
                                                                        return J$.Rt(1190017, J$.F(1190009, J$.R(1189977, 'handleCallback', handleCallback, false, true), false)(J$.R(1189985, 'callback', callback, false, false), J$.R(1189993, 'err', err, false, false), J$.T(1190001, null, 25, false)));
                                                                    J$.F(1190057, J$.R(1190025, 'handleCallback', handleCallback, false, true), false)(J$.R(1190033, 'callback', callback, false, false), J$.T(1190041, null, 25, false), J$.R(1190049, 'result', result, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(1190905, J$e);
                                                                } finally {
                                                                    if (J$.Fr(1190913))
                                                                        continue jalangiLabel21;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(1190921, J$e);
                                                } finally {
                                                    if (J$.Fr(1190929))
                                                        continue jalangiLabel22;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(1190177, J$.R(1190169, 'DropIndexOperation', DropIndexOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(1190937, J$e);
                        } finally {
                            if (J$.Fr(1190945))
                                continue jalangiLabel23;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(1190233, 'CommandOperation', CommandOperation, false, true)), DropIndexOperation, false, true);
            J$.F(1190289, J$.R(1190257, 'defineAspects', defineAspects, false, true), false)(J$.R(1190265, 'DropIndexOperation', DropIndexOperation, false, true), J$.G(1190281, J$.R(1190273, 'Aspect', Aspect, false, true), 'WRITE_OPERATION'));
            J$.P(1190313, J$.I(typeof module === 'undefined' ? module = J$.R(1190297, 'module', undefined, true, true) : module = J$.R(1190297, 'module', module, true, true)), 'exports', J$.R(1190305, 'DropIndexOperation', DropIndexOperation, false, true));
        } catch (J$e) {
            J$.Ex(1190953, J$e);
        } finally {
            if (J$.Sr(1190961))
                continue jalangiLabel24;
            else
                break jalangiLabel24;
        }
    }
// JALANGI DO NOT INSTRUMENT
