// mongodb/lib/operations/insert_many.js
J$.noInstrEval = false;
jalangiLabel21:
    while (true) {
        try {
            J$.Se(1249081, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/operations/insert_many.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(1246065, arguments.callee, this, arguments);
                            arguments = J$.N(1246073, 'arguments', arguments, true, false, false);
                            obj = J$.N(1246081, 'obj', obj, true, false, false);
                            if (J$.C(77816, J$.C(77776, J$.B(120298, '===', J$.U(120290, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1245801, 'Symbol', undefined, true, true) : Symbol = J$.R(1245801, 'Symbol', Symbol, true, true))), J$.T(1245809, 'function', 21, false))) ? J$.B(120314, '===', J$.U(120306, 'typeof', J$.G(1245825, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1245817, 'Symbol', undefined, true, true) : Symbol = J$.R(1245817, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1245833, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(1245889, '_typeof2', J$.T(1245881, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(1245857, arguments.callee, this, arguments);
                                                arguments = J$.N(1245865, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1245873, 'obj', obj, true, false, false);
                                                return J$.Rt(1245849, J$.U(120322, 'typeof', J$.R(1245841, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1249305, J$e);
                                            } finally {
                                                if (J$.Fr(1249313))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(1246025, '_typeof2', J$.T(1246017, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(1245993, arguments.callee, this, arguments);
                                                arguments = J$.N(1246001, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1246009, 'obj', obj, true, false, false);
                                                return J$.Rt(1245985, J$.C(77808, J$.C(77800, J$.C(77792, J$.C(77784, J$.R(1245897, 'obj', obj, false, false)) ? J$.B(120338, '===', J$.U(120330, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1245905, 'Symbol', undefined, true, true) : Symbol = J$.R(1245905, 'Symbol', Symbol, true, true))), J$.T(1245913, 'function', 21, false)) : J$._()) ? J$.B(120346, '===', J$.G(1245929, J$.R(1245921, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1245937, 'Symbol', undefined, true, true) : Symbol = J$.R(1245937, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(120354, '!==', J$.R(1245945, 'obj', obj, false, false), J$.G(1245961, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1245953, 'Symbol', undefined, true, true) : Symbol = J$.R(1245953, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1245969, 'symbol', 21, false) : J$.U(120362, 'typeof', J$.R(1245977, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1249321, J$e);
                                            } finally {
                                                if (J$.Fr(1249329))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(1246057, J$.F(1246049, J$.R(1246033, '_typeof2', _typeof2, false, true), false)(J$.R(1246041, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1249337, J$e);
                        } finally {
                            if (J$.Fr(1249345))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(1246401, arguments.callee, this, arguments);
                            arguments = J$.N(1246409, 'arguments', arguments, true, false, false);
                            obj = J$.N(1246417, 'obj', obj, true, false, false);
                            if (J$.C(77864, J$.C(77824, J$.B(120378, '===', J$.U(120370, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1246089, 'Symbol', undefined, true, true) : Symbol = J$.R(1246089, 'Symbol', Symbol, true, true))), J$.T(1246097, 'function', 21, false))) ? J$.B(120386, '===', J$.F(1246129, J$.R(1246105, '_typeof2', _typeof2, false, true), false)(J$.G(1246121, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1246113, 'Symbol', undefined, true, true) : Symbol = J$.R(1246113, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1246137, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(1246209, '_typeof', J$.T(1246201, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(1246177, arguments.callee, this, arguments);
                                                arguments = J$.N(1246185, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1246193, 'obj', obj, true, false, false);
                                                return J$.Rt(1246169, J$.F(1246161, J$.R(1246145, '_typeof2', _typeof2, false, true), false)(J$.R(1246153, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1249353, J$e);
                                            } finally {
                                                if (J$.Fr(1249361))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(1246361, '_typeof', J$.T(1246353, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(1246329, arguments.callee, this, arguments);
                                                arguments = J$.N(1246337, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1246345, 'obj', obj, true, false, false);
                                                return J$.Rt(1246321, J$.C(77856, J$.C(77848, J$.C(77840, J$.C(77832, J$.R(1246217, 'obj', obj, false, false)) ? J$.B(120402, '===', J$.U(120394, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1246225, 'Symbol', undefined, true, true) : Symbol = J$.R(1246225, 'Symbol', Symbol, true, true))), J$.T(1246233, 'function', 21, false)) : J$._()) ? J$.B(120410, '===', J$.G(1246249, J$.R(1246241, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1246257, 'Symbol', undefined, true, true) : Symbol = J$.R(1246257, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(120418, '!==', J$.R(1246265, 'obj', obj, false, false), J$.G(1246281, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1246273, 'Symbol', undefined, true, true) : Symbol = J$.R(1246273, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1246289, 'symbol', 21, false) : J$.F(1246313, J$.R(1246297, '_typeof2', _typeof2, false, true), false)(J$.R(1246305, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1249369, J$e);
                                            } finally {
                                                if (J$.Fr(1249377))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(1246393, J$.F(1246385, J$.R(1246369, '_typeof', _typeof, false, true), false)(J$.R(1246377, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1249385, J$e);
                        } finally {
                            if (J$.Fr(1249393))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(1246465, arguments.callee, this, arguments);
                            arguments = J$.N(1246473, 'arguments', arguments, true, false, false);
                            instance = J$.N(1246481, 'instance', instance, true, false, false);
                            Constructor = J$.N(1246489, 'Constructor', Constructor, true, false, false);
                            if (J$.C(77872, J$.U(120434, '!', J$.B(120426, 'instanceof', J$.R(1246425, 'instance', instance, false, false), J$.R(1246433, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(1246457, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1246441, 'TypeError', undefined, true, true) : TypeError = J$.R(1246441, 'TypeError', TypeError, true, true)), true)(J$.T(1246449, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1249401, J$e);
                        } finally {
                            if (J$.Fr(1249409))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(1246737, arguments.callee, this, arguments);
                            arguments = J$.N(1246745, 'arguments', arguments, true, false, false);
                            target = J$.N(1246753, 'target', target, true, false, false);
                            props = J$.N(1246761, 'props', props, true, false, false);
                            J$.N(1246769, 'i', i, false, false, false);
                            J$.N(1246777, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(1246505, 'i', J$.T(1246497, 0, 22, false), i, false, false); J$.C(77896, J$.B(120442, '<', J$.R(1246513, 'i', i, false, false), J$.G(1246529, J$.R(1246521, 'props', props, false, false), 'length'))); J$.B(120466, '-', i = J$.W(1246545, 'i', J$.B(120458, '+', J$.U(120450, '+', J$.R(1246537, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(1246577, 'descriptor', J$.G(1246569, J$.R(1246553, 'props', props, false, false), J$.R(1246561, 'i', i, false, false)), descriptor, false, false);
                                J$.P(1246617, J$.R(1246585, 'descriptor', descriptor, false, false), 'enumerable', J$.C(77880, J$.G(1246601, J$.R(1246593, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(1246609, false, 23, false));
                                J$.P(1246641, J$.R(1246625, 'descriptor', descriptor, false, false), 'configurable', J$.T(1246633, true, 23, false));
                                if (J$.C(77888, J$.B(120474, 'in', J$.T(1246649, 'value', 21, false), J$.R(1246657, 'descriptor', descriptor, false, false))))
                                    J$.P(1246681, J$.R(1246665, 'descriptor', descriptor, false, false), 'writable', J$.T(1246673, true, 23, false));
                                J$.M(1246729, J$.I(typeof Object === 'undefined' ? Object = J$.R(1246689, 'Object', undefined, true, true) : Object = J$.R(1246689, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(1246697, 'target', target, false, false), J$.G(1246713, J$.R(1246705, 'descriptor', descriptor, false, false), 'key'), J$.R(1246721, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1249417, J$e);
                        } finally {
                            if (J$.Fr(1249425))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(1246889, arguments.callee, this, arguments);
                            arguments = J$.N(1246897, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(1246905, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(1246913, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(1246921, 'staticProps', staticProps, true, false, false);
                            if (J$.C(77904, J$.R(1246785, 'protoProps', protoProps, false, false)))
                                J$.F(1246825, J$.R(1246793, '_defineProperties', _defineProperties, false, true), false)(J$.G(1246809, J$.R(1246801, 'Constructor', Constructor, false, false), 'prototype'), J$.R(1246817, 'protoProps', protoProps, false, false));
                            if (J$.C(77912, J$.R(1246833, 'staticProps', staticProps, false, false)))
                                J$.F(1246865, J$.R(1246841, '_defineProperties', _defineProperties, false, true), false)(J$.R(1246849, 'Constructor', Constructor, false, false), J$.R(1246857, 'staticProps', staticProps, false, false));
                            return J$.Rt(1246881, J$.R(1246873, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(1249433, J$e);
                        } finally {
                            if (J$.Fr(1249441))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(1247033, arguments.callee, this, arguments);
                            arguments = J$.N(1247041, 'arguments', arguments, true, false, false);
                            self = J$.N(1247049, 'self', self, true, false, false);
                            call = J$.N(1247057, 'call', call, true, false, false);
                            if (J$.C(77936, J$.C(77928, J$.R(1246929, 'call', call, false, false)) ? J$.C(77920, J$.B(120482, '===', J$.F(1246953, J$.R(1246937, '_typeof', _typeof, false, true), false)(J$.R(1246945, 'call', call, false, false)), J$.T(1246961, 'object', 21, false))) ? J$._() : J$.B(120498, '===', J$.U(120490, 'typeof', J$.R(1246969, 'call', call, false, false)), J$.T(1246977, 'function', 21, false)) : J$._())) {
                                return J$.Rt(1246993, J$.R(1246985, 'call', call, false, false));
                            }
                            return J$.Rt(1247025, J$.F(1247017, J$.R(1247001, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(1247009, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(1249449, J$e);
                        } finally {
                            if (J$.Fr(1249457))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(1247121, arguments.callee, this, arguments);
                            arguments = J$.N(1247129, 'arguments', arguments, true, false, false);
                            self = J$.N(1247137, 'self', self, true, false, false);
                            if (J$.C(77944, J$.B(120506, '===', J$.R(1247065, 'self', self, false, false), void J$.T(1247073, 0, 22, false)))) {
                                throw J$.F(1247097, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(1247081, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(1247081, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(1247089, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(1247113, J$.R(1247105, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(1249465, J$e);
                        } finally {
                            if (J$.Fr(1249473))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(1247297, arguments.callee, this, arguments);
                            arguments = J$.N(1247305, 'arguments', arguments, true, false, false);
                            o = J$.N(1247313, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(1247257, '_getPrototypeOf', J$.C(77960, J$.G(1247153, J$.I(typeof Object === 'undefined' ? Object = J$.R(1247145, 'Object', undefined, true, true) : Object = J$.R(1247145, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(1247169, J$.I(typeof Object === 'undefined' ? Object = J$.R(1247161, 'Object', undefined, true, true) : Object = J$.R(1247161, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(1247249, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(1247225, arguments.callee, this, arguments);
                                            arguments = J$.N(1247233, 'arguments', arguments, true, false, false);
                                            o = J$.N(1247241, 'o', o, true, false, false);
                                            return J$.Rt(1247217, J$.C(77952, J$.G(1247185, J$.R(1247177, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(1247209, J$.I(typeof Object === 'undefined' ? Object = J$.R(1247193, 'Object', undefined, true, true) : Object = J$.R(1247193, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(1247201, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(1249481, J$e);
                                        } finally {
                                            if (J$.Fr(1249489))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(1247289, J$.F(1247281, J$.R(1247265, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1247273, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(1249497, J$e);
                        } finally {
                            if (J$.Fr(1249505))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(1247513, arguments.callee, this, arguments);
                            arguments = J$.N(1247521, 'arguments', arguments, true, false, false);
                            subClass = J$.N(1247529, 'subClass', subClass, true, false, false);
                            superClass = J$.N(1247537, 'superClass', superClass, true, false, false);
                            if (J$.C(77976, J$.C(77968, J$.B(120522, '!==', J$.U(120514, 'typeof', J$.R(1247321, 'superClass', superClass, false, false)), J$.T(1247329, 'function', 21, false))) ? J$.B(120530, '!==', J$.R(1247337, 'superClass', superClass, false, false), J$.T(1247345, null, 25, false)) : J$._())) {
                                throw J$.F(1247369, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1247353, 'TypeError', undefined, true, true) : TypeError = J$.R(1247353, 'TypeError', TypeError, true, true)), true)(J$.T(1247361, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(1247465, J$.R(1247377, 'subClass', subClass, false, false), 'prototype', J$.M(1247457, J$.I(typeof Object === 'undefined' ? Object = J$.R(1247385, 'Object', undefined, true, true) : Object = J$.R(1247385, 'Object', Object, true, true)), 'create', false)(J$.C(77984, J$.R(1247393, 'superClass', superClass, false, false)) ? J$.G(1247409, J$.R(1247401, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(1247449, {
                                constructor: J$.T(1247441, {
                                    value: J$.R(1247417, 'subClass', subClass, false, false),
                                    writable: J$.T(1247425, true, 23, false),
                                    configurable: J$.T(1247433, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(77992, J$.R(1247473, 'superClass', superClass, false, false)))
                                J$.F(1247505, J$.R(1247481, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1247489, 'subClass', subClass, false, false), J$.R(1247497, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(1249513, J$e);
                        } finally {
                            if (J$.Fr(1249521))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(1247689, arguments.callee, this, arguments);
                            arguments = J$.N(1247697, 'arguments', arguments, true, false, false);
                            o = J$.N(1247705, 'o', o, true, false, false);
                            p = J$.N(1247713, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(1247641, '_setPrototypeOf', J$.C(78000, J$.G(1247553, J$.I(typeof Object === 'undefined' ? Object = J$.R(1247545, 'Object', undefined, true, true) : Object = J$.R(1247545, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(1247633, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(1247601, arguments.callee, this, arguments);
                                            arguments = J$.N(1247609, 'arguments', arguments, true, false, false);
                                            o = J$.N(1247617, 'o', o, true, false, false);
                                            p = J$.N(1247625, 'p', p, true, false, false);
                                            J$.P(1247577, J$.R(1247561, 'o', o, false, false), '__proto__', J$.R(1247569, 'p', p, false, false));
                                            return J$.Rt(1247593, J$.R(1247585, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1249529, J$e);
                                        } finally {
                                            if (J$.Fr(1249537))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(1247681, J$.F(1247673, J$.R(1247649, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1247657, 'o', o, false, false), J$.R(1247665, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(1249545, J$e);
                        } finally {
                            if (J$.Fr(1249553))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function mapInsertManyResults(docs, r) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(1249017, arguments.callee, this, arguments);
                            arguments = J$.N(1249025, 'arguments', arguments, true, false, false);
                            docs = J$.N(1249033, 'docs', docs, true, false, false);
                            r = J$.N(1249041, 'r', r, true, false, false);
                            J$.N(1249049, 'finalResult', finalResult, false, false, false);
                            var finalResult = J$.W(1248937, 'finalResult', J$.T(1248929, {
                                result: J$.T(1248881, {
                                    ok: J$.T(1248857, 1, 22, false),
                                    n: J$.G(1248873, J$.R(1248865, 'r', r, false, false), 'insertedCount')
                                }, 11, false),
                                ops: J$.R(1248889, 'docs', docs, false, false),
                                insertedCount: J$.G(1248905, J$.R(1248897, 'r', r, false, false), 'insertedCount'),
                                insertedIds: J$.G(1248921, J$.R(1248913, 'r', r, false, false), 'insertedIds')
                            }, 11, false), finalResult, false, false);
                            if (J$.C(78032, J$.M(1248953, J$.R(1248945, 'r', r, false, false), 'getLastOp', false)())) {
                                J$.P(1248993, J$.G(1248969, J$.R(1248961, 'finalResult', finalResult, false, false), 'result'), 'opTime', J$.M(1248985, J$.R(1248977, 'r', r, false, false), 'getLastOp', false)());
                            }
                            return J$.Rt(1249009, J$.R(1249001, 'finalResult', finalResult, false, false));
                        } catch (J$e) {
                            J$.Ex(1249625, J$e);
                        } finally {
                            if (J$.Fr(1249633))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(1249097, '_typeof2', J$.T(1249089, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(1249113, '_typeof', J$.T(1249105, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(1249129, '_classCallCheck', J$.T(1249121, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(1249145, '_defineProperties', J$.T(1249137, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(1249161, '_createClass', J$.T(1249153, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(1249177, '_possibleConstructorReturn', J$.T(1249169, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(1249193, '_assertThisInitialized', J$.T(1249185, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(1249209, '_getPrototypeOf', J$.T(1249201, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(1249225, '_inherits', J$.T(1249217, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(1249241, '_setPrototypeOf', J$.T(1249233, _setPrototypeOf, 12, false), true, false, false);
            J$.N(1249249, 'OperationBase', OperationBase, false, false, false);
            J$.N(1249257, 'BulkWriteOperation', BulkWriteOperation, false, false, false);
            J$.N(1249265, 'MongoError', MongoError, false, false, false);
            J$.N(1249273, 'prepareDocs', prepareDocs, false, false, false);
            J$.N(1249281, 'InsertManyOperation', InsertManyOperation, false, false, false);
            mapInsertManyResults = J$.N(1249297, 'mapInsertManyResults', J$.T(1249289, mapInsertManyResults, 12, false), true, false, false);
            J$.T(1245793, 'use strict', 21, false);
            var OperationBase = J$.W(1247753, 'OperationBase', J$.G(1247745, J$.F(1247737, J$.I(typeof require === 'undefined' ? require = J$.R(1247721, 'require', undefined, true, true) : require = J$.R(1247721, 'require', require, true, true)), false)(J$.T(1247729, './operation', 21, false)), 'OperationBase'), OperationBase, false, true);
            var BulkWriteOperation = J$.W(1247785, 'BulkWriteOperation', J$.F(1247777, J$.I(typeof require === 'undefined' ? require = J$.R(1247761, 'require', undefined, true, true) : require = J$.R(1247761, 'require', require, true, true)), false)(J$.T(1247769, './bulk_write', 21, false)), BulkWriteOperation, false, true);
            var MongoError = J$.W(1247825, 'MongoError', J$.G(1247817, J$.F(1247809, J$.I(typeof require === 'undefined' ? require = J$.R(1247793, 'require', undefined, true, true) : require = J$.R(1247793, 'require', require, true, true)), false)(J$.T(1247801, '../core', 21, false)), 'MongoError'), MongoError, false, true);
            var prepareDocs = J$.W(1247865, 'prepareDocs', J$.G(1247857, J$.F(1247849, J$.I(typeof require === 'undefined' ? require = J$.R(1247833, 'require', undefined, true, true) : require = J$.R(1247833, 'require', require, true, true)), false)(J$.T(1247841, './common_functions', 21, false)), 'prepareDocs'), prepareDocs, false, true);
            var InsertManyOperation = J$.W(1248849, 'InsertManyOperation', J$.F(1248841, J$.T(1248825, function (_OperationBase) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(1248785, arguments.callee, this, arguments);
                            function InsertManyOperation(collection, docs, options) {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(1248081, arguments.callee, this, arguments);
                                            arguments = J$.N(1248089, 'arguments', arguments, true, false, false);
                                            collection = J$.N(1248097, 'collection', collection, true, false, false);
                                            docs = J$.N(1248105, 'docs', docs, true, false, false);
                                            options = J$.N(1248113, 'options', options, true, false, false);
                                            J$.N(1248121, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(1247929, J$.R(1247905, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(1247913, 'this', this, false, false), J$.R(1247921, 'InsertManyOperation', InsertManyOperation, false, false));
                                            _this = J$.W(1248009, '_this', J$.F(1248001, J$.R(1247937, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(1247945, 'this', this, false, false), J$.M(1247993, J$.F(1247969, J$.R(1247953, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1247961, 'InsertManyOperation', InsertManyOperation, false, false)), 'call', false)(J$.R(1247977, 'this', this, false, false), J$.R(1247985, 'options', options, false, false))), _this, false, false);
                                            J$.P(1248033, J$.R(1248017, '_this', _this, false, false), 'collection', J$.R(1248025, 'collection', collection, false, false));
                                            J$.P(1248057, J$.R(1248041, '_this', _this, false, false), 'docs', J$.R(1248049, 'docs', docs, false, false));
                                            return J$.Rt(1248073, J$.R(1248065, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1249561, J$e);
                                        } finally {
                                            if (J$.Fr(1249569))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(1248793, 'arguments', arguments, true, false, false);
                            _OperationBase = J$.N(1248801, '_OperationBase', _OperationBase, true, false, false);
                            InsertManyOperation = J$.N(1248817, 'InsertManyOperation', J$.T(1248809, InsertManyOperation, 12, false), true, false, false);
                            J$.F(1247897, J$.R(1247873, '_inherits', _inherits, false, true), false)(J$.R(1247881, 'InsertManyOperation', InsertManyOperation, false, false), J$.R(1247889, '_OperationBase', _OperationBase, false, false));
                            J$.F(1248761, J$.R(1248129, '_createClass', _createClass, false, true), false)(J$.R(1248137, 'InsertManyOperation', InsertManyOperation, false, false), J$.T(1248753, [J$.T(1248745, {
                                    key: J$.T(1248145, 'execute', 21, false),
                                    value: J$.T(1248737, function execute(callback) {
                                        jalangiLabel18:
                                            while (true) {
                                                try {
                                                    J$.Fe(1248673, arguments.callee, this, arguments);
                                                    arguments = J$.N(1248681, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(1248689, 'callback', callback, true, false, false);
                                                    J$.N(1248697, 'coll', coll, false, false, false);
                                                    J$.N(1248705, 'docs', docs, false, false, false);
                                                    J$.N(1248713, 'options', options, false, false, false);
                                                    J$.N(1248721, 'operations', operations, false, false, false);
                                                    J$.N(1248729, 'bulkWriteOperation', bulkWriteOperation, false, false, false);
                                                    var coll = J$.W(1248169, 'coll', J$.G(1248161, J$.R(1248153, 'this', this, false, false), 'collection'), coll, false, false);
                                                    var docs = J$.W(1248193, 'docs', J$.G(1248185, J$.R(1248177, 'this', this, false, false), 'docs'), docs, false, false);
                                                    var options = J$.W(1248217, 'options', J$.G(1248209, J$.R(1248201, 'this', this, false, false), 'options'), options, false, false);
                                                    if (J$.C(78008, J$.U(120538, '!', J$.M(1248241, J$.I(typeof Array === 'undefined' ? Array = J$.R(1248225, 'Array', undefined, true, true) : Array = J$.R(1248225, 'Array', Array, true, true)), 'isArray', false)(J$.R(1248233, 'docs', docs, false, false))))) {
                                                        return J$.Rt(1248305, J$.F(1248297, J$.R(1248249, 'callback', callback, false, false), false)(J$.M(1248289, J$.R(1248257, 'MongoError', MongoError, false, true), 'create', false)(J$.T(1248281, {
                                                            message: J$.T(1248265, 'docs parameter must be an array of documents', 21, false),
                                                            driver: J$.T(1248273, true, 23, false)
                                                        }, 11, false))));
                                                    }
                                                    J$.P(1248377, J$.R(1248313, 'options', options, false, false), J$.T(1248321, 'serializeFunctions', 21, false), J$.C(78016, J$.G(1248345, J$.R(1248329, 'options', options, false, false), J$.T(1248337, 'serializeFunctions', 21, false))) ? J$._() : J$.G(1248369, J$.G(1248361, J$.R(1248353, 'coll', coll, false, false), 's'), 'serializeFunctions'));
                                                    docs = J$.W(1248425, 'docs', J$.F(1248417, J$.R(1248385, 'prepareDocs', prepareDocs, false, true), false)(J$.R(1248393, 'coll', coll, false, false), J$.R(1248401, 'docs', docs, false, false), J$.R(1248409, 'options', options, false, false)), docs, false, false);
                                                    var operations = J$.W(1248457, 'operations', J$.T(1248449, [J$.T(1248441, { insertMany: J$.R(1248433, 'docs', docs, false, false) }, 11, false)], 10, false), operations, false, false);
                                                    var bulkWriteOperation = J$.W(1248505, 'bulkWriteOperation', J$.F(1248497, J$.R(1248465, 'BulkWriteOperation', BulkWriteOperation, false, true), true)(J$.R(1248473, 'coll', coll, false, false), J$.R(1248481, 'operations', operations, false, false), J$.R(1248489, 'options', options, false, false)), bulkWriteOperation, false, false);
                                                    J$.M(1248665, J$.R(1248513, 'bulkWriteOperation', bulkWriteOperation, false, false), 'execute', false)(J$.T(1248657, function (err, result) {
                                                        jalangiLabel17:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(1248625, arguments.callee, this, arguments);
                                                                    arguments = J$.N(1248633, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(1248641, 'err', err, true, false, false);
                                                                    result = J$.N(1248649, 'result', result, true, false, false);
                                                                    if (J$.C(78024, J$.R(1248521, 'err', err, false, false)))
                                                                        return J$.Rt(1248561, J$.F(1248553, J$.R(1248529, 'callback', callback, false, false), false)(J$.R(1248537, 'err', err, false, false), J$.T(1248545, null, 25, false)));
                                                                    J$.F(1248617, J$.R(1248569, 'callback', callback, false, false), false)(J$.T(1248577, null, 25, false), J$.F(1248609, J$.R(1248585, 'mapInsertManyResults', mapInsertManyResults, false, true), false)(J$.R(1248593, 'docs', docs, false, false), J$.R(1248601, 'result', result, false, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(1249577, J$e);
                                                                } finally {
                                                                    if (J$.Fr(1249585))
                                                                        continue jalangiLabel17;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(1249593, J$e);
                                                } finally {
                                                    if (J$.Fr(1249601))
                                                        continue jalangiLabel18;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(1248777, J$.R(1248769, 'InsertManyOperation', InsertManyOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(1249609, J$e);
                        } finally {
                            if (J$.Fr(1249617))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(1248833, 'OperationBase', OperationBase, false, true)), InsertManyOperation, false, true);
            J$.P(1249073, J$.I(typeof module === 'undefined' ? module = J$.R(1249057, 'module', undefined, true, true) : module = J$.R(1249057, 'module', module, true, true)), 'exports', J$.R(1249065, 'InsertManyOperation', InsertManyOperation, false, true));
        } catch (J$e) {
            J$.Ex(1249641, J$e);
        } finally {
            if (J$.Sr(1249649))
                continue jalangiLabel21;
            else
                break jalangiLabel21;
        }
    }
// JALANGI DO NOT INSTRUMENT
