// mongodb/lib/operations/list_collections.js
J$.noInstrEval = false;
jalangiLabel25:
    while (true) {
        try {
            J$.Se(1261721, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/operations/list_collections.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(1257241, arguments.callee, this, arguments);
                            arguments = J$.N(1257249, 'arguments', arguments, true, false, false);
                            obj = J$.N(1257257, 'obj', obj, true, false, false);
                            if (J$.C(78664, J$.C(78624, J$.B(121122, '===', J$.U(121114, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1256977, 'Symbol', undefined, true, true) : Symbol = J$.R(1256977, 'Symbol', Symbol, true, true))), J$.T(1256985, 'function', 21, false))) ? J$.B(121138, '===', J$.U(121130, 'typeof', J$.G(1257001, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1256993, 'Symbol', undefined, true, true) : Symbol = J$.R(1256993, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1257009, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(1257065, '_typeof2', J$.T(1257057, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(1257033, arguments.callee, this, arguments);
                                                arguments = J$.N(1257041, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1257049, 'obj', obj, true, false, false);
                                                return J$.Rt(1257025, J$.U(121146, 'typeof', J$.R(1257017, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1261993, J$e);
                                            } finally {
                                                if (J$.Fr(1262001))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(1257201, '_typeof2', J$.T(1257193, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(1257169, arguments.callee, this, arguments);
                                                arguments = J$.N(1257177, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1257185, 'obj', obj, true, false, false);
                                                return J$.Rt(1257161, J$.C(78656, J$.C(78648, J$.C(78640, J$.C(78632, J$.R(1257073, 'obj', obj, false, false)) ? J$.B(121162, '===', J$.U(121154, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1257081, 'Symbol', undefined, true, true) : Symbol = J$.R(1257081, 'Symbol', Symbol, true, true))), J$.T(1257089, 'function', 21, false)) : J$._()) ? J$.B(121170, '===', J$.G(1257105, J$.R(1257097, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1257113, 'Symbol', undefined, true, true) : Symbol = J$.R(1257113, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(121178, '!==', J$.R(1257121, 'obj', obj, false, false), J$.G(1257137, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1257129, 'Symbol', undefined, true, true) : Symbol = J$.R(1257129, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1257145, 'symbol', 21, false) : J$.U(121186, 'typeof', J$.R(1257153, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1262009, J$e);
                                            } finally {
                                                if (J$.Fr(1262017))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(1257233, J$.F(1257225, J$.R(1257209, '_typeof2', _typeof2, false, true), false)(J$.R(1257217, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1262025, J$e);
                        } finally {
                            if (J$.Fr(1262033))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(1257577, arguments.callee, this, arguments);
                            arguments = J$.N(1257585, 'arguments', arguments, true, false, false);
                            obj = J$.N(1257593, 'obj', obj, true, false, false);
                            if (J$.C(78712, J$.C(78672, J$.B(121202, '===', J$.U(121194, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1257265, 'Symbol', undefined, true, true) : Symbol = J$.R(1257265, 'Symbol', Symbol, true, true))), J$.T(1257273, 'function', 21, false))) ? J$.B(121210, '===', J$.F(1257305, J$.R(1257281, '_typeof2', _typeof2, false, true), false)(J$.G(1257297, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1257289, 'Symbol', undefined, true, true) : Symbol = J$.R(1257289, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1257313, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(1257385, '_typeof', J$.T(1257377, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(1257353, arguments.callee, this, arguments);
                                                arguments = J$.N(1257361, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1257369, 'obj', obj, true, false, false);
                                                return J$.Rt(1257345, J$.F(1257337, J$.R(1257321, '_typeof2', _typeof2, false, true), false)(J$.R(1257329, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1262041, J$e);
                                            } finally {
                                                if (J$.Fr(1262049))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(1257537, '_typeof', J$.T(1257529, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(1257505, arguments.callee, this, arguments);
                                                arguments = J$.N(1257513, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1257521, 'obj', obj, true, false, false);
                                                return J$.Rt(1257497, J$.C(78704, J$.C(78696, J$.C(78688, J$.C(78680, J$.R(1257393, 'obj', obj, false, false)) ? J$.B(121226, '===', J$.U(121218, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1257401, 'Symbol', undefined, true, true) : Symbol = J$.R(1257401, 'Symbol', Symbol, true, true))), J$.T(1257409, 'function', 21, false)) : J$._()) ? J$.B(121234, '===', J$.G(1257425, J$.R(1257417, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1257433, 'Symbol', undefined, true, true) : Symbol = J$.R(1257433, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(121242, '!==', J$.R(1257441, 'obj', obj, false, false), J$.G(1257457, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1257449, 'Symbol', undefined, true, true) : Symbol = J$.R(1257449, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1257465, 'symbol', 21, false) : J$.F(1257489, J$.R(1257473, '_typeof2', _typeof2, false, true), false)(J$.R(1257481, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1262057, J$e);
                                            } finally {
                                                if (J$.Fr(1262065))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(1257569, J$.F(1257561, J$.R(1257545, '_typeof', _typeof, false, true), false)(J$.R(1257553, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1262073, J$e);
                        } finally {
                            if (J$.Fr(1262081))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(1257641, arguments.callee, this, arguments);
                            arguments = J$.N(1257649, 'arguments', arguments, true, false, false);
                            instance = J$.N(1257657, 'instance', instance, true, false, false);
                            Constructor = J$.N(1257665, 'Constructor', Constructor, true, false, false);
                            if (J$.C(78720, J$.U(121258, '!', J$.B(121250, 'instanceof', J$.R(1257601, 'instance', instance, false, false), J$.R(1257609, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(1257633, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1257617, 'TypeError', undefined, true, true) : TypeError = J$.R(1257617, 'TypeError', TypeError, true, true)), true)(J$.T(1257625, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1262089, J$e);
                        } finally {
                            if (J$.Fr(1262097))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(1257913, arguments.callee, this, arguments);
                            arguments = J$.N(1257921, 'arguments', arguments, true, false, false);
                            target = J$.N(1257929, 'target', target, true, false, false);
                            props = J$.N(1257937, 'props', props, true, false, false);
                            J$.N(1257945, 'i', i, false, false, false);
                            J$.N(1257953, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(1257681, 'i', J$.T(1257673, 0, 22, false), i, false, false); J$.C(78744, J$.B(121266, '<', J$.R(1257689, 'i', i, false, false), J$.G(1257705, J$.R(1257697, 'props', props, false, false), 'length'))); J$.B(121290, '-', i = J$.W(1257721, 'i', J$.B(121282, '+', J$.U(121274, '+', J$.R(1257713, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(1257753, 'descriptor', J$.G(1257745, J$.R(1257729, 'props', props, false, false), J$.R(1257737, 'i', i, false, false)), descriptor, false, false);
                                J$.P(1257793, J$.R(1257761, 'descriptor', descriptor, false, false), 'enumerable', J$.C(78728, J$.G(1257777, J$.R(1257769, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(1257785, false, 23, false));
                                J$.P(1257817, J$.R(1257801, 'descriptor', descriptor, false, false), 'configurable', J$.T(1257809, true, 23, false));
                                if (J$.C(78736, J$.B(121298, 'in', J$.T(1257825, 'value', 21, false), J$.R(1257833, 'descriptor', descriptor, false, false))))
                                    J$.P(1257857, J$.R(1257841, 'descriptor', descriptor, false, false), 'writable', J$.T(1257849, true, 23, false));
                                J$.M(1257905, J$.I(typeof Object === 'undefined' ? Object = J$.R(1257865, 'Object', undefined, true, true) : Object = J$.R(1257865, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(1257873, 'target', target, false, false), J$.G(1257889, J$.R(1257881, 'descriptor', descriptor, false, false), 'key'), J$.R(1257897, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1262105, J$e);
                        } finally {
                            if (J$.Fr(1262113))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(1258065, arguments.callee, this, arguments);
                            arguments = J$.N(1258073, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(1258081, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(1258089, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(1258097, 'staticProps', staticProps, true, false, false);
                            if (J$.C(78752, J$.R(1257961, 'protoProps', protoProps, false, false)))
                                J$.F(1258001, J$.R(1257969, '_defineProperties', _defineProperties, false, true), false)(J$.G(1257985, J$.R(1257977, 'Constructor', Constructor, false, false), 'prototype'), J$.R(1257993, 'protoProps', protoProps, false, false));
                            if (J$.C(78760, J$.R(1258009, 'staticProps', staticProps, false, false)))
                                J$.F(1258041, J$.R(1258017, '_defineProperties', _defineProperties, false, true), false)(J$.R(1258025, 'Constructor', Constructor, false, false), J$.R(1258033, 'staticProps', staticProps, false, false));
                            return J$.Rt(1258057, J$.R(1258049, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(1262121, J$e);
                        } finally {
                            if (J$.Fr(1262129))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(1258209, arguments.callee, this, arguments);
                            arguments = J$.N(1258217, 'arguments', arguments, true, false, false);
                            self = J$.N(1258225, 'self', self, true, false, false);
                            call = J$.N(1258233, 'call', call, true, false, false);
                            if (J$.C(78784, J$.C(78776, J$.R(1258105, 'call', call, false, false)) ? J$.C(78768, J$.B(121306, '===', J$.F(1258129, J$.R(1258113, '_typeof', _typeof, false, true), false)(J$.R(1258121, 'call', call, false, false)), J$.T(1258137, 'object', 21, false))) ? J$._() : J$.B(121322, '===', J$.U(121314, 'typeof', J$.R(1258145, 'call', call, false, false)), J$.T(1258153, 'function', 21, false)) : J$._())) {
                                return J$.Rt(1258169, J$.R(1258161, 'call', call, false, false));
                            }
                            return J$.Rt(1258201, J$.F(1258193, J$.R(1258177, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(1258185, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(1262137, J$e);
                        } finally {
                            if (J$.Fr(1262145))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(1258297, arguments.callee, this, arguments);
                            arguments = J$.N(1258305, 'arguments', arguments, true, false, false);
                            self = J$.N(1258313, 'self', self, true, false, false);
                            if (J$.C(78792, J$.B(121330, '===', J$.R(1258241, 'self', self, false, false), void J$.T(1258249, 0, 22, false)))) {
                                throw J$.F(1258273, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(1258257, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(1258257, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(1258265, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(1258289, J$.R(1258281, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(1262153, J$e);
                        } finally {
                            if (J$.Fr(1262161))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(1258681, arguments.callee, this, arguments);
                            arguments = J$.N(1258689, 'arguments', arguments, true, false, false);
                            target = J$.N(1258697, 'target', target, true, false, false);
                            property = J$.N(1258705, 'property', property, true, false, false);
                            receiver = J$.N(1258713, 'receiver', receiver, true, false, false);
                            if (J$.C(78824, J$.C(78800, J$.B(121346, '!==', J$.U(121338, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1258321, 'Reflect', undefined, true, true) : Reflect = J$.R(1258321, 'Reflect', Reflect, true, true))), J$.T(1258329, 'undefined', 21, false))) ? J$.G(1258345, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1258337, 'Reflect', undefined, true, true) : Reflect = J$.R(1258337, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(1258369, '_get', J$.G(1258361, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1258353, 'Reflect', undefined, true, true) : Reflect = J$.R(1258353, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(1258617, '_get', J$.T(1258609, function _get(target, property, receiver) {
                                    jalangiLabel11:
                                        while (true) {
                                            try {
                                                J$.Fe(1258553, arguments.callee, this, arguments);
                                                arguments = J$.N(1258561, 'arguments', arguments, true, false, false);
                                                target = J$.N(1258569, 'target', target, true, false, false);
                                                property = J$.N(1258577, 'property', property, true, false, false);
                                                receiver = J$.N(1258585, 'receiver', receiver, true, false, false);
                                                J$.N(1258593, 'base', base, false, false, false);
                                                J$.N(1258601, 'desc', desc, false, false, false);
                                                var base = J$.W(1258409, 'base', J$.F(1258401, J$.R(1258377, '_superPropBase', _superPropBase, false, true), false)(J$.R(1258385, 'target', target, false, false), J$.R(1258393, 'property', property, false, false)), base, false, false);
                                                if (J$.C(78808, J$.U(121354, '!', J$.R(1258417, 'base', base, false, false))))
                                                    return J$.Rt(1258425, undefined);
                                                var desc = J$.W(1258465, 'desc', J$.M(1258457, J$.I(typeof Object === 'undefined' ? Object = J$.R(1258433, 'Object', undefined, true, true) : Object = J$.R(1258433, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(1258441, 'base', base, false, false), J$.R(1258449, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(78816, J$.G(1258481, J$.R(1258473, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(1258521, J$.M(1258513, J$.G(1258497, J$.R(1258489, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(1258505, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(1258545, J$.G(1258537, J$.R(1258529, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(1262169, J$e);
                                            } finally {
                                                if (J$.Fr(1262177))
                                                    continue jalangiLabel11;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(1258673, J$.F(1258665, J$.R(1258625, '_get', _get, false, true), false)(J$.R(1258633, 'target', target, false, false), J$.R(1258641, 'property', property, false, false), J$.C(78832, J$.R(1258649, 'receiver', receiver, false, false)) ? J$._() : J$.R(1258657, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(1262185, J$e);
                        } finally {
                            if (J$.Fr(1262193))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(1258833, arguments.callee, this, arguments);
                            arguments = J$.N(1258841, 'arguments', arguments, true, false, false);
                            object = J$.N(1258849, 'object', object, true, false, false);
                            property = J$.N(1258857, 'property', property, true, false, false);
                            while (J$.C(78848, J$.U(121362, '!', J$.M(1258761, J$.G(1258737, J$.G(1258729, J$.I(typeof Object === 'undefined' ? Object = J$.R(1258721, 'Object', undefined, true, true) : Object = J$.R(1258721, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(1258745, 'object', object, false, false), J$.R(1258753, 'property', property, false, false))))) {
                                object = J$.W(1258793, 'object', J$.F(1258785, J$.R(1258769, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1258777, 'object', object, false, false)), object, false, false);
                                if (J$.C(78840, J$.B(121370, '===', J$.R(1258801, 'object', object, false, false), J$.T(1258809, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(1258825, J$.R(1258817, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(1262201, J$e);
                        } finally {
                            if (J$.Fr(1262209))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(1259017, arguments.callee, this, arguments);
                            arguments = J$.N(1259025, 'arguments', arguments, true, false, false);
                            o = J$.N(1259033, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(1258977, '_getPrototypeOf', J$.C(78864, J$.G(1258873, J$.I(typeof Object === 'undefined' ? Object = J$.R(1258865, 'Object', undefined, true, true) : Object = J$.R(1258865, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(1258889, J$.I(typeof Object === 'undefined' ? Object = J$.R(1258881, 'Object', undefined, true, true) : Object = J$.R(1258881, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(1258969, function _getPrototypeOf(o) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(1258945, arguments.callee, this, arguments);
                                            arguments = J$.N(1258953, 'arguments', arguments, true, false, false);
                                            o = J$.N(1258961, 'o', o, true, false, false);
                                            return J$.Rt(1258937, J$.C(78856, J$.G(1258905, J$.R(1258897, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(1258929, J$.I(typeof Object === 'undefined' ? Object = J$.R(1258913, 'Object', undefined, true, true) : Object = J$.R(1258913, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(1258921, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(1262217, J$e);
                                        } finally {
                                            if (J$.Fr(1262225))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(1259009, J$.F(1259001, J$.R(1258985, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1258993, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(1262233, J$e);
                        } finally {
                            if (J$.Fr(1262241))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(1259233, arguments.callee, this, arguments);
                            arguments = J$.N(1259241, 'arguments', arguments, true, false, false);
                            subClass = J$.N(1259249, 'subClass', subClass, true, false, false);
                            superClass = J$.N(1259257, 'superClass', superClass, true, false, false);
                            if (J$.C(78880, J$.C(78872, J$.B(121386, '!==', J$.U(121378, 'typeof', J$.R(1259041, 'superClass', superClass, false, false)), J$.T(1259049, 'function', 21, false))) ? J$.B(121394, '!==', J$.R(1259057, 'superClass', superClass, false, false), J$.T(1259065, null, 25, false)) : J$._())) {
                                throw J$.F(1259089, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1259073, 'TypeError', undefined, true, true) : TypeError = J$.R(1259073, 'TypeError', TypeError, true, true)), true)(J$.T(1259081, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(1259185, J$.R(1259097, 'subClass', subClass, false, false), 'prototype', J$.M(1259177, J$.I(typeof Object === 'undefined' ? Object = J$.R(1259105, 'Object', undefined, true, true) : Object = J$.R(1259105, 'Object', Object, true, true)), 'create', false)(J$.C(78888, J$.R(1259113, 'superClass', superClass, false, false)) ? J$.G(1259129, J$.R(1259121, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(1259169, {
                                constructor: J$.T(1259161, {
                                    value: J$.R(1259137, 'subClass', subClass, false, false),
                                    writable: J$.T(1259145, true, 23, false),
                                    configurable: J$.T(1259153, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(78896, J$.R(1259193, 'superClass', superClass, false, false)))
                                J$.F(1259225, J$.R(1259201, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1259209, 'subClass', subClass, false, false), J$.R(1259217, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(1262249, J$e);
                        } finally {
                            if (J$.Fr(1262257))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(1259409, arguments.callee, this, arguments);
                            arguments = J$.N(1259417, 'arguments', arguments, true, false, false);
                            o = J$.N(1259425, 'o', o, true, false, false);
                            p = J$.N(1259433, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(1259361, '_setPrototypeOf', J$.C(78904, J$.G(1259273, J$.I(typeof Object === 'undefined' ? Object = J$.R(1259265, 'Object', undefined, true, true) : Object = J$.R(1259265, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(1259353, function _setPrototypeOf(o, p) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(1259321, arguments.callee, this, arguments);
                                            arguments = J$.N(1259329, 'arguments', arguments, true, false, false);
                                            o = J$.N(1259337, 'o', o, true, false, false);
                                            p = J$.N(1259345, 'p', p, true, false, false);
                                            J$.P(1259297, J$.R(1259281, 'o', o, false, false), '__proto__', J$.R(1259289, 'p', p, false, false));
                                            return J$.Rt(1259313, J$.R(1259305, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1262265, J$e);
                                        } finally {
                                            if (J$.Fr(1262273))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(1259401, J$.F(1259393, J$.R(1259369, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1259377, 'o', o, false, false), J$.R(1259385, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(1262281, J$e);
                        } finally {
                            if (J$.Fr(1262289))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function listCollectionsTransforms(databaseName) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(1259889, arguments.callee, this, arguments);
                            arguments = J$.N(1259897, 'arguments', arguments, true, false, false);
                            databaseName = J$.N(1259905, 'databaseName', databaseName, true, false, false);
                            J$.N(1259913, 'matching', matching, false, false, false);
                            var matching = J$.W(1259673, 'matching', J$.M(1259665, J$.T(1259641, '', 21, false), 'concat', false)(J$.R(1259649, 'databaseName', databaseName, false, false), J$.T(1259657, '.', 21, false)), matching, false, false);
                            return J$.Rt(1259881, J$.T(1259873, {
                                doc: J$.T(1259865, function doc(_doc) {
                                    jalangiLabel19:
                                        while (true) {
                                            try {
                                                J$.Fe(1259833, arguments.callee, this, arguments);
                                                arguments = J$.N(1259841, 'arguments', arguments, true, false, false);
                                                _doc = J$.N(1259849, '_doc', _doc, true, false, false);
                                                J$.N(1259857, 'index', index, false, false, false);
                                                var index = J$.W(1259713, 'index', J$.M(1259705, J$.G(1259689, J$.R(1259681, '_doc', _doc, false, false), 'name'), 'indexOf', false)(J$.R(1259697, 'matching', matching, false, false)), index, false, false);
                                                if (J$.C(78920, J$.C(78912, J$.G(1259729, J$.R(1259721, '_doc', _doc, false, false), 'name')) ? J$.B(121402, '===', J$.R(1259737, 'index', index, false, false), J$.T(1259745, 0, 22, false)) : J$._())) {
                                                    J$.P(1259809, J$.R(1259753, '_doc', _doc, false, false), 'name', J$.M(1259801, J$.G(1259769, J$.R(1259761, '_doc', _doc, false, false), 'name'), 'substr', false)(J$.B(121410, '+', J$.R(1259777, 'index', index, false, false), J$.G(1259793, J$.R(1259785, 'matching', matching, false, false), 'length'))));
                                                }
                                                return J$.Rt(1259825, J$.R(1259817, '_doc', _doc, false, false));
                                            } catch (J$e) {
                                                J$.Ex(1262297, J$e);
                                            } finally {
                                                if (J$.Fr(1262305))
                                                    continue jalangiLabel19;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(1262313, J$e);
                        } finally {
                            if (J$.Fr(1262321))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(1261737, '_typeof2', J$.T(1261729, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(1261753, '_typeof', J$.T(1261745, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(1261769, '_classCallCheck', J$.T(1261761, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(1261785, '_defineProperties', J$.T(1261777, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(1261801, '_createClass', J$.T(1261793, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(1261817, '_possibleConstructorReturn', J$.T(1261809, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(1261833, '_assertThisInitialized', J$.T(1261825, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(1261849, '_get', J$.T(1261841, _get, 12, false), true, false, false);
            _superPropBase = J$.N(1261865, '_superPropBase', J$.T(1261857, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(1261881, '_getPrototypeOf', J$.T(1261873, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(1261897, '_inherits', J$.T(1261889, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(1261913, '_setPrototypeOf', J$.T(1261905, _setPrototypeOf, 12, false), true, false, false);
            J$.N(1261921, 'CommandOperationV2', CommandOperationV2, false, false, false);
            J$.N(1261929, 'Aspect', Aspect, false, false, false);
            J$.N(1261937, 'defineAspects', defineAspects, false, false, false);
            J$.N(1261945, 'maxWireVersion', maxWireVersion, false, false, false);
            J$.N(1261953, 'CONSTANTS', CONSTANTS, false, false, false);
            J$.N(1261961, 'LIST_COLLECTIONS_WIRE_VERSION', LIST_COLLECTIONS_WIRE_VERSION, false, false, false);
            listCollectionsTransforms = J$.N(1261977, 'listCollectionsTransforms', J$.T(1261969, listCollectionsTransforms, 12, false), true, false, false);
            J$.N(1261985, 'ListCollectionsOperation', ListCollectionsOperation, false, false, false);
            J$.T(1256969, 'use strict', 21, false);
            var CommandOperationV2 = J$.W(1259465, 'CommandOperationV2', J$.F(1259457, J$.I(typeof require === 'undefined' ? require = J$.R(1259441, 'require', undefined, true, true) : require = J$.R(1259441, 'require', require, true, true)), false)(J$.T(1259449, './command_v2', 21, false)), CommandOperationV2, false, true);
            var Aspect = J$.W(1259505, 'Aspect', J$.G(1259497, J$.F(1259489, J$.I(typeof require === 'undefined' ? require = J$.R(1259473, 'require', undefined, true, true) : require = J$.R(1259473, 'require', require, true, true)), false)(J$.T(1259481, './operation', 21, false)), 'Aspect'), Aspect, false, true);
            var defineAspects = J$.W(1259545, 'defineAspects', J$.G(1259537, J$.F(1259529, J$.I(typeof require === 'undefined' ? require = J$.R(1259513, 'require', undefined, true, true) : require = J$.R(1259513, 'require', require, true, true)), false)(J$.T(1259521, './operation', 21, false)), 'defineAspects'), defineAspects, false, true);
            var maxWireVersion = J$.W(1259585, 'maxWireVersion', J$.G(1259577, J$.F(1259569, J$.I(typeof require === 'undefined' ? require = J$.R(1259553, 'require', undefined, true, true) : require = J$.R(1259553, 'require', require, true, true)), false)(J$.T(1259561, '../core/utils', 21, false)), 'maxWireVersion'), maxWireVersion, false, true);
            var CONSTANTS = J$.W(1259617, 'CONSTANTS', J$.F(1259609, J$.I(typeof require === 'undefined' ? require = J$.R(1259593, 'require', undefined, true, true) : require = J$.R(1259593, 'require', require, true, true)), false)(J$.T(1259601, '../constants', 21, false)), CONSTANTS, false, true);
            var LIST_COLLECTIONS_WIRE_VERSION = J$.W(1259633, 'LIST_COLLECTIONS_WIRE_VERSION', J$.T(1259625, 3, 22, false), LIST_COLLECTIONS_WIRE_VERSION, false, true);
            var ListCollectionsOperation = J$.W(1261609, 'ListCollectionsOperation', J$.F(1261601, J$.T(1261585, function (_CommandOperationV) {
                jalangiLabel24:
                    while (true) {
                        try {
                            J$.Fe(1261545, arguments.callee, this, arguments);
                            function ListCollectionsOperation(db, filter, options) {
                                jalangiLabel21:
                                    while (true) {
                                        try {
                                            J$.Fe(1260265, arguments.callee, this, arguments);
                                            arguments = J$.N(1260273, 'arguments', arguments, true, false, false);
                                            db = J$.N(1260281, 'db', db, true, false, false);
                                            filter = J$.N(1260289, 'filter', filter, true, false, false);
                                            options = J$.N(1260297, 'options', options, true, false, false);
                                            J$.N(1260305, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(1259977, J$.R(1259953, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(1259961, 'this', this, false, false), J$.R(1259969, 'ListCollectionsOperation', ListCollectionsOperation, false, false));
                                            _this = J$.W(1260081, '_this', J$.F(1260073, J$.R(1259985, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(1259993, 'this', this, false, false), J$.M(1260065, J$.F(1260017, J$.R(1260001, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1260009, 'ListCollectionsOperation', ListCollectionsOperation, false, false)), 'call', false)(J$.R(1260025, 'this', this, false, false), J$.R(1260033, 'db', db, false, false), J$.R(1260041, 'options', options, false, false), J$.T(1260057, { fullResponse: J$.T(1260049, true, 23, false) }, 11, false))), _this, false, false);
                                            J$.P(1260105, J$.R(1260089, '_this', _this, false, false), 'db', J$.R(1260097, 'db', db, false, false));
                                            J$.P(1260129, J$.R(1260113, '_this', _this, false, false), 'filter', J$.R(1260121, 'filter', filter, false, false));
                                            J$.P(1260169, J$.R(1260137, '_this', _this, false, false), 'nameOnly', J$.U(121426, '!', J$.U(121418, '!', J$.G(1260161, J$.G(1260153, J$.R(1260145, '_this', _this, false, false), 'options'), 'nameOnly'))));
                                            if (J$.C(78928, J$.B(121442, '===', J$.U(121434, 'typeof', J$.G(1260193, J$.G(1260185, J$.R(1260177, '_this', _this, false, false), 'options'), 'batchSize')), J$.T(1260201, 'number', 21, false)))) {
                                                J$.P(1260241, J$.R(1260209, '_this', _this, false, false), 'batchSize', J$.G(1260233, J$.G(1260225, J$.R(1260217, '_this', _this, false, false), 'options'), 'batchSize'));
                                            }
                                            return J$.Rt(1260257, J$.R(1260249, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1262329, J$e);
                                        } finally {
                                            if (J$.Fr(1262337))
                                                continue jalangiLabel21;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(1261553, 'arguments', arguments, true, false, false);
                            _CommandOperationV = J$.N(1261561, '_CommandOperationV', _CommandOperationV, true, false, false);
                            ListCollectionsOperation = J$.N(1261577, 'ListCollectionsOperation', J$.T(1261569, ListCollectionsOperation, 12, false), true, false, false);
                            J$.F(1259945, J$.R(1259921, '_inherits', _inherits, false, true), false)(J$.R(1259929, 'ListCollectionsOperation', ListCollectionsOperation, false, false), J$.R(1259937, '_CommandOperationV', _CommandOperationV, false, false));
                            J$.F(1261521, J$.R(1260313, '_createClass', _createClass, false, true), false)(J$.R(1260321, 'ListCollectionsOperation', ListCollectionsOperation, false, false), J$.T(1261513, [J$.T(1261505, {
                                    key: J$.T(1260329, 'execute', 21, false),
                                    value: J$.T(1261497, function execute(server, callback) {
                                        jalangiLabel23:
                                            while (true) {
                                                try {
                                                    J$.Fe(1261433, arguments.callee, this, arguments);
                                                    arguments = J$.N(1261441, 'arguments', arguments, true, false, false);
                                                    server = J$.N(1261449, 'server', server, true, false, false);
                                                    callback = J$.N(1261457, 'callback', callback, true, false, false);
                                                    J$.N(1261465, 'filter', filter, false, false, false);
                                                    J$.N(1261473, 'databaseName', databaseName, false, false, false);
                                                    J$.N(1261481, 'transforms', transforms, false, false, false);
                                                    J$.N(1261489, 'command', command, false, false, false);
                                                    if (J$.C(79008, J$.B(121450, '<', J$.F(1260353, J$.R(1260337, 'maxWireVersion', maxWireVersion, false, true), false)(J$.R(1260345, 'server', server, false, false)), J$.R(1260361, 'LIST_COLLECTIONS_WIRE_VERSION', LIST_COLLECTIONS_WIRE_VERSION, false, true)))) {
                                                        var filter = J$.W(1260385, 'filter', J$.G(1260377, J$.R(1260369, 'this', this, false, false), 'filter'), filter, false, false);
                                                        var databaseName = J$.W(1260433, 'databaseName', J$.G(1260425, J$.G(1260417, J$.G(1260409, J$.G(1260401, J$.R(1260393, 'this', this, false, false), 'db'), 's'), 'namespace'), 'db'), databaseName, false, false);
                                                        if (J$.C(78944, J$.C(78936, J$.B(121466, '===', J$.U(121458, 'typeof', J$.G(1260449, J$.R(1260441, 'filter', filter, false, false), 'name')), J$.T(1260457, 'string', 21, false))) ? J$.U(121490, '!', J$.M(1260521, J$.F(1260497, J$.I(typeof RegExp === 'undefined' ? RegExp = J$.R(1260465, 'RegExp', undefined, true, true) : RegExp = J$.R(1260465, 'RegExp', RegExp, true, true)), true)(J$.B(121482, '+', J$.B(121474, '+', J$.T(1260473, '^', 21, false), J$.R(1260481, 'databaseName', databaseName, false, false)), J$.T(1260489, '\\.', 21, false))), 'test', false)(J$.G(1260513, J$.R(1260505, 'filter', filter, false, false), 'name'))) : J$._())) {
                                                            filter = J$.W(1260561, 'filter', J$.M(1260553, J$.I(typeof Object === 'undefined' ? Object = J$.R(1260529, 'Object', undefined, true, true) : Object = J$.R(1260529, 'Object', Object, true, true)), 'assign', false)(J$.T(1260537, {}, 11, false), J$.R(1260545, 'filter', filter, false, false)), filter, false, false);
                                                            J$.P(1260641, J$.R(1260569, 'filter', filter, false, false), 'name', J$.M(1260633, J$.M(1260625, J$.G(1260601, J$.G(1260593, J$.G(1260585, J$.R(1260577, 'this', this, false, false), 'db'), 's'), 'namespace'), 'withCollection', false)(J$.G(1260617, J$.R(1260609, 'filter', filter, false, false), 'name')), 'toString', false)());
                                                        }
                                                        if (J$.C(78952, J$.B(121498, '==', J$.R(1260649, 'filter', filter, false, false), J$.T(1260657, null, 25, false)))) {
                                                            J$.P(1260705, J$.R(1260665, 'filter', filter, false, false), 'name', J$.M(1260697, J$.T(1260673, '/', 21, false), 'concat', false)(J$.R(1260681, 'databaseName', databaseName, false, false), J$.T(1260689, '/', 21, false)));
                                                        }
                                                        if (J$.C(78960, J$.G(1260721, J$.R(1260713, 'filter', filter, false, false), 'name'))) {
                                                            filter = J$.W(1260785, 'filter', J$.T(1260777, {
                                                                $and: J$.T(1260769, [
                                                                    J$.T(1260745, { name: J$.G(1260737, J$.R(1260729, 'filter', filter, false, false), 'name') }, 11, false),
                                                                    J$.T(1260761, { name: J$.T(1260753, /^((?!\$).)*$/, 14, false) }, 11, false)
                                                                ], 10, false)
                                                            }, 11, false), filter, false, false);
                                                        } else {
                                                            filter = J$.W(1260809, 'filter', J$.T(1260801, { name: J$.T(1260793, /^((?!\$).)*$/, 14, false) }, 11, false), filter, false, false);
                                                        }
                                                        var transforms = J$.W(1260841, 'transforms', J$.F(1260833, J$.R(1260817, 'listCollectionsTransforms', listCollectionsTransforms, false, true), false)(J$.R(1260825, 'databaseName', databaseName, false, false)), transforms, false, false);
                                                        J$.M(1261201, J$.R(1260849, 'server', server, false, false), 'query', false)(J$.M(1260905, J$.M(1260881, J$.T(1260857, '', 21, false), 'concat', false)(J$.R(1260865, 'databaseName', databaseName, false, false), J$.T(1260873, '.', 21, false)), 'concat', false)(J$.G(1260897, J$.R(1260889, 'CONSTANTS', CONSTANTS, false, true), 'SYSTEM_NAMESPACE_COLLECTION')), J$.T(1260921, { query: J$.R(1260913, 'filter', filter, false, false) }, 11, false), J$.T(1260953, { batchSize: J$.C(78968, J$.G(1260937, J$.R(1260929, 'this', this, false, false), 'batchSize')) ? J$._() : J$.T(1260945, 1000, 22, false) }, 11, false), J$.T(1260961, {}, 11, false), J$.T(1261193, function (err, result) {
                                                            jalangiLabel22:
                                                                while (true) {
                                                                    try {
                                                                        J$.Fe(1261161, arguments.callee, this, arguments);
                                                                        arguments = J$.N(1261169, 'arguments', arguments, true, false, false);
                                                                        err = J$.N(1261177, 'err', err, true, false, false);
                                                                        result = J$.N(1261185, 'result', result, true, false, false);
                                                                        if (J$.C(79000, J$.C(78992, J$.C(78984, J$.C(78976, J$.R(1260969, 'result', result, false, false)) ? J$.G(1260985, J$.R(1260977, 'result', result, false, false), 'message') : J$._()) ? J$.G(1261009, J$.G(1261001, J$.R(1260993, 'result', result, false, false), 'message'), 'documents') : J$._()) ? J$.M(1261049, J$.I(typeof Array === 'undefined' ? Array = J$.R(1261017, 'Array', undefined, true, true) : Array = J$.R(1261017, 'Array', Array, true, true)), 'isArray', false)(J$.G(1261041, J$.G(1261033, J$.R(1261025, 'result', result, false, false), 'message'), 'documents')) : J$._())) {
                                                                            J$.P(1261121, J$.G(1261065, J$.R(1261057, 'result', result, false, false), 'message'), 'documents', J$.M(1261113, J$.G(1261089, J$.G(1261081, J$.R(1261073, 'result', result, false, false), 'message'), 'documents'), 'map', false)(J$.G(1261105, J$.R(1261097, 'transforms', transforms, false, false), 'doc')));
                                                                        }
                                                                        J$.F(1261153, J$.R(1261129, 'callback', callback, false, false), false)(J$.R(1261137, 'err', err, false, false), J$.R(1261145, 'result', result, false, false));
                                                                    } catch (J$e) {
                                                                        J$.Ex(1262345, J$e);
                                                                    } finally {
                                                                        if (J$.Fr(1262353))
                                                                            continue jalangiLabel22;
                                                                        else
                                                                            return J$.Ra();
                                                                    }
                                                                }
                                                        }, 12, false));
                                                        return J$.Rt(1261209, undefined);
                                                    }
                                                    var command = J$.W(1261313, 'command', J$.T(1261305, {
                                                        listCollections: J$.T(1261217, 1, 22, false),
                                                        filter: J$.G(1261233, J$.R(1261225, 'this', this, false, false), 'filter'),
                                                        cursor: J$.C(79016, J$.G(1261249, J$.R(1261241, 'this', this, false, false), 'batchSize')) ? J$.T(1261273, { batchSize: J$.G(1261265, J$.R(1261257, 'this', this, false, false), 'batchSize') }, 11, false) : J$.T(1261281, {}, 11, false),
                                                        nameOnly: J$.G(1261297, J$.R(1261289, 'this', this, false, false), 'nameOnly')
                                                    }, 11, false), command, false, false);
                                                    return J$.Rt(1261425, J$.M(1261417, J$.F(1261377, J$.R(1261321, '_get', _get, false, true), false)(J$.F(1261353, J$.R(1261329, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(1261345, J$.R(1261337, 'ListCollectionsOperation', ListCollectionsOperation, false, false), 'prototype')), J$.T(1261361, 'executeCommand', 21, false), J$.R(1261369, 'this', this, false, false)), 'call', false)(J$.R(1261385, 'this', this, false, false), J$.R(1261393, 'server', server, false, false), J$.R(1261401, 'command', command, false, false), J$.R(1261409, 'callback', callback, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(1262361, J$e);
                                                } finally {
                                                    if (J$.Fr(1262369))
                                                        continue jalangiLabel23;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(1261537, J$.R(1261529, 'ListCollectionsOperation', ListCollectionsOperation, false, false));
                        } catch (J$e) {
                            J$.Ex(1262377, J$e);
                        } finally {
                            if (J$.Fr(1262385))
                                continue jalangiLabel24;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(1261593, 'CommandOperationV2', CommandOperationV2, false, true)), ListCollectionsOperation, false, true);
            J$.F(1261689, J$.R(1261617, 'defineAspects', defineAspects, false, true), false)(J$.R(1261625, 'ListCollectionsOperation', ListCollectionsOperation, false, true), J$.T(1261681, [
                J$.G(1261641, J$.R(1261633, 'Aspect', Aspect, false, true), 'READ_OPERATION'),
                J$.G(1261657, J$.R(1261649, 'Aspect', Aspect, false, true), 'RETRYABLE'),
                J$.G(1261673, J$.R(1261665, 'Aspect', Aspect, false, true), 'EXECUTE_WITH_SELECTION')
            ], 10, false));
            J$.P(1261713, J$.I(typeof module === 'undefined' ? module = J$.R(1261697, 'module', undefined, true, true) : module = J$.R(1261697, 'module', module, true, true)), 'exports', J$.R(1261705, 'ListCollectionsOperation', ListCollectionsOperation, false, true));
        } catch (J$e) {
            J$.Ex(1262393, J$e);
        } finally {
            if (J$.Sr(1262401))
                continue jalangiLabel25;
            else
                break jalangiLabel25;
        }
    }
// JALANGI DO NOT INSTRUMENT
