// kuler/index.js
J$.noInstrEval = false;
jalangiLabel6:
    while (true) {
        try {
            J$.Se(7574393, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/kuler/index.js');
            function Kuler(text, color) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(7572729, arguments.callee, this, arguments);
                            arguments = J$.N(7572737, 'arguments', arguments, true, false, false);
                            text = J$.N(7572745, 'text', text, true, false, false);
                            color = J$.N(7572753, 'color', color, true, false, false);
                            if (J$.C(402472, J$.R(7572601, 'color', color, false, false)))
                                return J$.Rt(7572649, J$.M(7572641, J$.F(7572625, J$.R(7572609, 'Kuler', Kuler, false, true), true)(J$.R(7572617, 'text', text, false, false)), 'style', false)(J$.R(7572633, 'color', color, false, false)));
                            if (J$.C(402480, J$.U(755754, '!', J$.B(755746, 'instanceof', J$.R(7572657, 'this', this, false, false), J$.R(7572665, 'Kuler', Kuler, false, true)))))
                                return J$.Rt(7572697, J$.F(7572689, J$.R(7572673, 'Kuler', Kuler, false, true), true)(J$.R(7572681, 'text', text, false, false)));
                            J$.P(7572721, J$.R(7572705, 'this', this, false, false), 'text', J$.R(7572713, 'text', text, false, false));
                        } catch (J$e) {
                            J$.Ex(7574425, J$e);
                        } finally {
                            if (J$.Fr(7574433))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(7574401, 'colornames', colornames, false, false, false);
            Kuler = J$.N(7574417, 'Kuler', J$.T(7574409, Kuler, 12, false), true, false, false);
            J$.T(7572561, 'use strict', 21, false);
            var colornames = J$.W(7572593, 'colornames', J$.F(7572585, J$.I(typeof require === 'undefined' ? require = J$.R(7572569, 'require', undefined, true, true) : require = J$.R(7572569, 'require', require, true, true)), false)(J$.T(7572577, 'colornames', 21, false)), colornames, false, true);
            J$.P(7572785, J$.G(7572769, J$.R(7572761, 'Kuler', Kuler, false, true), 'prototype'), 'prefix', J$.T(7572777, '\x1B[', 21, false));
            J$.P(7572817, J$.G(7572801, J$.R(7572793, 'Kuler', Kuler, false, true), 'prototype'), 'suffix', J$.T(7572809, 'm', 21, false));
            J$.P(7573529, J$.G(7572833, J$.R(7572825, 'Kuler', Kuler, false, true), 'prototype'), 'hex', J$.T(7573521, function hex(color) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7573473, arguments.callee, this, arguments);
                            arguments = J$.N(7573481, 'arguments', arguments, true, false, false);
                            color = J$.N(7573489, 'color', color, true, false, false);
                            J$.N(7573497, 'r', r, false, false, false);
                            J$.N(7573505, 'g', g, false, false, false);
                            J$.N(7573513, 'b', b, false, false, false);
                            color = J$.W(7572905, 'color', J$.C(402488, J$.B(755762, '===', J$.G(7572857, J$.R(7572841, 'color', color, false, false), J$.T(7572849, 0, 22, false)), J$.T(7572865, '#', 21, false))) ? J$.M(7572889, J$.R(7572873, 'color', color, false, false), 'substring', false)(J$.T(7572881, 1, 22, false)) : J$.R(7572897, 'color', color, false, false), color, false, false);
                            if (J$.C(402496, J$.B(755770, '===', J$.G(7572921, J$.R(7572913, 'color', color, false, false), 'length'), J$.T(7572929, 3, 22, false)))) {
                                color = J$.W(7572961, 'color', J$.M(7572953, J$.R(7572937, 'color', color, false, false), 'split', false)(J$.T(7572945, '', 21, false)), color, false, false);
                                J$.P(7573009, J$.R(7572969, 'color', color, false, false), J$.T(7572977, 5, 22, false), J$.G(7573001, J$.R(7572985, 'color', color, false, false), J$.T(7572993, 2, 22, false)));
                                J$.P(7573057, J$.R(7573017, 'color', color, false, false), J$.T(7573025, 4, 22, false), J$.G(7573049, J$.R(7573033, 'color', color, false, false), J$.T(7573041, 2, 22, false)));
                                J$.P(7573105, J$.R(7573065, 'color', color, false, false), J$.T(7573073, 3, 22, false), J$.G(7573097, J$.R(7573081, 'color', color, false, false), J$.T(7573089, 1, 22, false)));
                                J$.P(7573153, J$.R(7573113, 'color', color, false, false), J$.T(7573121, 2, 22, false), J$.G(7573145, J$.R(7573129, 'color', color, false, false), J$.T(7573137, 1, 22, false)));
                                J$.P(7573201, J$.R(7573161, 'color', color, false, false), J$.T(7573169, 1, 22, false), J$.G(7573193, J$.R(7573177, 'color', color, false, false), J$.T(7573185, 0, 22, false)));
                                color = J$.W(7573233, 'color', J$.M(7573225, J$.R(7573209, 'color', color, false, false), 'join', false)(J$.T(7573217, '', 21, false)), color, false, false);
                            }
                            var r = J$.W(7573337, 'r', J$.M(7573265, J$.R(7573241, 'color', color, false, false), 'substring', false)(J$.T(7573249, 0, 22, false), J$.T(7573257, 2, 22, false)), r, false, false), g = J$.W(7573345, 'g', J$.M(7573297, J$.R(7573273, 'color', color, false, false), 'substring', false)(J$.T(7573281, 2, 22, false), J$.T(7573289, 4, 22, false)), g, false, false), b = J$.W(7573353, 'b', J$.M(7573329, J$.R(7573305, 'color', color, false, false), 'substring', false)(J$.T(7573313, 4, 22, false), J$.T(7573321, 6, 22, false)), b, false, false);
                            return J$.Rt(7573465, J$.T(7573457, [
                                J$.F(7573385, J$.I(typeof parseInt === 'undefined' ? parseInt = J$.R(7573361, 'parseInt', undefined, true, true) : parseInt = J$.R(7573361, 'parseInt', parseInt, true, true)), false)(J$.R(7573369, 'r', r, false, false), J$.T(7573377, 16, 22, false)),
                                J$.F(7573417, J$.I(typeof parseInt === 'undefined' ? parseInt = J$.R(7573393, 'parseInt', undefined, true, true) : parseInt = J$.R(7573393, 'parseInt', parseInt, true, true)), false)(J$.R(7573401, 'g', g, false, false), J$.T(7573409, 16, 22, false)),
                                J$.F(7573449, J$.I(typeof parseInt === 'undefined' ? parseInt = J$.R(7573425, 'parseInt', undefined, true, true) : parseInt = J$.R(7573425, 'parseInt', parseInt, true, true)), false)(J$.R(7573433, 'b', b, false, false), J$.T(7573441, 16, 22, false))
                            ], 10, false));
                        } catch (J$e) {
                            J$.Ex(7574441, J$e);
                        } finally {
                            if (J$.Fr(7574449))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7573769, J$.G(7573545, J$.R(7573537, 'Kuler', Kuler, false, true), 'prototype'), 'rgb', J$.T(7573761, function rgb(r, g, b) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(7573697, arguments.callee, this, arguments);
                            arguments = J$.N(7573705, 'arguments', arguments, true, false, false);
                            r = J$.N(7573713, 'r', r, true, false, false);
                            g = J$.N(7573721, 'g', g, true, false, false);
                            b = J$.N(7573729, 'b', b, true, false, false);
                            J$.N(7573737, 'red', red, false, false, false);
                            J$.N(7573745, 'green', green, false, false, false);
                            J$.N(7573753, 'blue', blue, false, false, false);
                            var red = J$.W(7573625, 'red', J$.B(755786, '*', J$.B(755778, '/', J$.R(7573553, 'r', r, false, false), J$.T(7573561, 255, 22, false)), J$.T(7573569, 5, 22, false)), red, false, false), green = J$.W(7573633, 'green', J$.B(755802, '*', J$.B(755794, '/', J$.R(7573577, 'g', g, false, false), J$.T(7573585, 255, 22, false)), J$.T(7573593, 5, 22, false)), green, false, false), blue = J$.W(7573641, 'blue', J$.B(755818, '*', J$.B(755810, '/', J$.R(7573601, 'b', b, false, false), J$.T(7573609, 255, 22, false)), J$.T(7573617, 5, 22, false)), blue, false, false);
                            return J$.Rt(7573689, J$.M(7573681, J$.R(7573649, 'this', this, false, false), 'ansi', false)(J$.R(7573657, 'red', red, false, false), J$.R(7573665, 'green', green, false, false), J$.R(7573673, 'blue', blue, false, false)));
                        } catch (J$e) {
                            J$.Ex(7574457, J$e);
                        } finally {
                            if (J$.Fr(7574465))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7574017, J$.G(7573785, J$.R(7573777, 'Kuler', Kuler, false, true), 'prototype'), 'ansi', J$.T(7574009, function ansi(r, g, b) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(7573945, arguments.callee, this, arguments);
                            arguments = J$.N(7573953, 'arguments', arguments, true, false, false);
                            r = J$.N(7573961, 'r', r, true, false, false);
                            g = J$.N(7573969, 'g', g, true, false, false);
                            b = J$.N(7573977, 'b', b, true, false, false);
                            J$.N(7573985, 'red', red, false, false, false);
                            J$.N(7573993, 'green', green, false, false, false);
                            J$.N(7574001, 'blue', blue, false, false, false);
                            var red = J$.W(7573865, 'red', J$.M(7573809, J$.I(typeof Math === 'undefined' ? Math = J$.R(7573793, 'Math', undefined, true, true) : Math = J$.R(7573793, 'Math', Math, true, true)), 'round', false)(J$.R(7573801, 'r', r, false, false)), red, false, false), green = J$.W(7573873, 'green', J$.M(7573833, J$.I(typeof Math === 'undefined' ? Math = J$.R(7573817, 'Math', undefined, true, true) : Math = J$.R(7573817, 'Math', Math, true, true)), 'round', false)(J$.R(7573825, 'g', g, false, false)), green, false, false), blue = J$.W(7573881, 'blue', J$.M(7573857, J$.I(typeof Math === 'undefined' ? Math = J$.R(7573841, 'Math', undefined, true, true) : Math = J$.R(7573841, 'Math', Math, true, true)), 'round', false)(J$.R(7573849, 'b', b, false, false)), blue, false, false);
                            return J$.Rt(7573937, J$.B(755858, '+', J$.B(755850, '+', J$.B(755834, '+', J$.T(7573889, 16, 22, false), J$.B(755826, '*', J$.R(7573897, 'red', red, false, false), J$.T(7573905, 36, 22, false))), J$.B(755842, '*', J$.R(7573913, 'green', green, false, false), J$.T(7573921, 6, 22, false))), J$.R(7573929, 'blue', blue, false, false)));
                        } catch (J$e) {
                            J$.Ex(7574473, J$e);
                        } finally {
                            if (J$.Fr(7574481))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7574113, J$.G(7574033, J$.R(7574025, 'Kuler', Kuler, false, true), 'prototype'), 'reset', J$.T(7574105, function reset() {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7574089, arguments.callee, this, arguments);
                            arguments = J$.N(7574097, 'arguments', arguments, true, false, false);
                            return J$.Rt(7574081, J$.B(755874, '+', J$.B(755866, '+', J$.G(7574049, J$.R(7574041, 'this', this, false, false), 'prefix'), J$.T(7574057, '39;49', 21, false)), J$.G(7574073, J$.R(7574065, 'this', this, false, false), 'suffix')));
                        } catch (J$e) {
                            J$.Ex(7574489, J$e);
                        } finally {
                            if (J$.Fr(7574497))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7574361, J$.G(7574129, J$.R(7574121, 'Kuler', Kuler, false, true), 'prototype'), 'style', J$.T(7574353, function style(color) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(7574329, arguments.callee, this, arguments);
                            arguments = J$.N(7574337, 'arguments', arguments, true, false, false);
                            color = J$.N(7574345, 'color', color, true, false, false);
                            if (J$.C(402504, J$.U(755882, '!', J$.M(7574153, J$.T(7574137, /^#?(?:[0-9a-fA-F]{3}){1,2}$/, 14, false), 'test', false)(J$.R(7574145, 'color', color, false, false))))) {
                                color = J$.W(7574185, 'color', J$.F(7574177, J$.R(7574161, 'colornames', colornames, false, true), false)(J$.R(7574169, 'color', color, false, false)), color, false, false);
                            }
                            return J$.Rt(7574321, J$.B(755922, '+', J$.B(755914, '+', J$.B(755906, '+', J$.B(755898, '+', J$.B(755890, '+', J$.G(7574201, J$.R(7574193, 'this', this, false, false), 'prefix'), J$.T(7574209, '38;5;', 21, false)), J$.M(7574265, J$.G(7574225, J$.R(7574217, 'this', this, false, false), 'rgb'), 'apply', false)(J$.R(7574233, 'this', this, false, false), J$.M(7574257, J$.R(7574241, 'this', this, false, false), 'hex', false)(J$.R(7574249, 'color', color, false, false)))), J$.G(7574281, J$.R(7574273, 'this', this, false, false), 'suffix')), J$.G(7574297, J$.R(7574289, 'this', this, false, false), 'text')), J$.M(7574313, J$.R(7574305, 'this', this, false, false), 'reset', false)()));
                        } catch (J$e) {
                            J$.Ex(7574505, J$e);
                        } finally {
                            if (J$.Fr(7574513))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7574385, J$.I(typeof module === 'undefined' ? module = J$.R(7574369, 'module', undefined, true, true) : module = J$.R(7574369, 'module', module, true, true)), 'exports', J$.R(7574377, 'Kuler', Kuler, false, true));
        } catch (J$e) {
            J$.Ex(7574521, J$e);
        } finally {
            if (J$.Sr(7574529))
                continue jalangiLabel6;
            else
                break jalangiLabel6;
        }
    }
// JALANGI DO NOT INSTRUMENT
