// coinbase-pro/lib/clients/websocket.js
J$.noInstrEval = false;
jalangiLabel24:
    while (true) {
        try {
            J$.Se(756281, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCoinBasePro/node_modules/coinbase-pro/lib/clients/websocket.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(752097, arguments.callee, this, arguments);
                            arguments = J$.N(752105, 'arguments', arguments, true, false, false);
                            obj = J$.N(752113, 'obj', obj, true, false, false);
                            if (J$.C(48816, J$.C(48776, J$.B(139610, '===', J$.U(139602, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(751833, 'Symbol', undefined, true, true) : Symbol = J$.R(751833, 'Symbol', Symbol, true, true))), J$.T(751841, 'function', 21, false))) ? J$.B(139626, '===', J$.U(139618, 'typeof', J$.G(751857, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(751849, 'Symbol', undefined, true, true) : Symbol = J$.R(751849, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(751865, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(751921, '_typeof', J$.T(751913, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(751889, arguments.callee, this, arguments);
                                                arguments = J$.N(751897, 'arguments', arguments, true, false, false);
                                                obj = J$.N(751905, 'obj', obj, true, false, false);
                                                return J$.Rt(751881, J$.U(139634, 'typeof', J$.R(751873, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(756489, J$e);
                                            } finally {
                                                if (J$.Fr(756497))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(752057, '_typeof', J$.T(752049, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(752025, arguments.callee, this, arguments);
                                                arguments = J$.N(752033, 'arguments', arguments, true, false, false);
                                                obj = J$.N(752041, 'obj', obj, true, false, false);
                                                return J$.Rt(752017, J$.C(48808, J$.C(48800, J$.C(48792, J$.C(48784, J$.R(751929, 'obj', obj, false, false)) ? J$.B(139650, '===', J$.U(139642, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(751937, 'Symbol', undefined, true, true) : Symbol = J$.R(751937, 'Symbol', Symbol, true, true))), J$.T(751945, 'function', 21, false)) : J$._()) ? J$.B(139658, '===', J$.G(751961, J$.R(751953, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(751969, 'Symbol', undefined, true, true) : Symbol = J$.R(751969, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(139666, '!==', J$.R(751977, 'obj', obj, false, false), J$.G(751993, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(751985, 'Symbol', undefined, true, true) : Symbol = J$.R(751985, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(752001, 'symbol', 21, false) : J$.U(139674, 'typeof', J$.R(752009, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(756505, J$e);
                                            } finally {
                                                if (J$.Fr(756513))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(752089, J$.F(752081, J$.R(752065, '_typeof', _typeof, false, true), false)(J$.R(752073, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(756521, J$e);
                        } finally {
                            if (J$.Fr(756529))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(752161, arguments.callee, this, arguments);
                            arguments = J$.N(752169, 'arguments', arguments, true, false, false);
                            instance = J$.N(752177, 'instance', instance, true, false, false);
                            Constructor = J$.N(752185, 'Constructor', Constructor, true, false, false);
                            if (J$.C(48824, J$.U(139690, '!', J$.B(139682, 'instanceof', J$.R(752121, 'instance', instance, false, false), J$.R(752129, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(752153, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(752137, 'TypeError', undefined, true, true) : TypeError = J$.R(752137, 'TypeError', TypeError, true, true)), true)(J$.T(752145, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(756537, J$e);
                        } finally {
                            if (J$.Fr(756545))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(752433, arguments.callee, this, arguments);
                            arguments = J$.N(752441, 'arguments', arguments, true, false, false);
                            target = J$.N(752449, 'target', target, true, false, false);
                            props = J$.N(752457, 'props', props, true, false, false);
                            J$.N(752465, 'i', i, false, false, false);
                            J$.N(752473, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(752201, 'i', J$.T(752193, 0, 22, false), i, false, false); J$.C(48848, J$.B(139698, '<', J$.R(752209, 'i', i, false, false), J$.G(752225, J$.R(752217, 'props', props, false, false), 'length'))); J$.B(139722, '-', i = J$.W(752241, 'i', J$.B(139714, '+', J$.U(139706, '+', J$.R(752233, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(752273, 'descriptor', J$.G(752265, J$.R(752249, 'props', props, false, false), J$.R(752257, 'i', i, false, false)), descriptor, false, false);
                                J$.P(752313, J$.R(752281, 'descriptor', descriptor, false, false), 'enumerable', J$.C(48832, J$.G(752297, J$.R(752289, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(752305, false, 23, false));
                                J$.P(752337, J$.R(752321, 'descriptor', descriptor, false, false), 'configurable', J$.T(752329, true, 23, false));
                                if (J$.C(48840, J$.B(139730, 'in', J$.T(752345, 'value', 21, false), J$.R(752353, 'descriptor', descriptor, false, false))))
                                    J$.P(752377, J$.R(752361, 'descriptor', descriptor, false, false), 'writable', J$.T(752369, true, 23, false));
                                J$.M(752425, J$.I(typeof Object === 'undefined' ? Object = J$.R(752385, 'Object', undefined, true, true) : Object = J$.R(752385, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(752393, 'target', target, false, false), J$.G(752409, J$.R(752401, 'descriptor', descriptor, false, false), 'key'), J$.R(752417, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(756553, J$e);
                        } finally {
                            if (J$.Fr(756561))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(752585, arguments.callee, this, arguments);
                            arguments = J$.N(752593, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(752601, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(752609, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(752617, 'staticProps', staticProps, true, false, false);
                            if (J$.C(48856, J$.R(752481, 'protoProps', protoProps, false, false)))
                                J$.F(752521, J$.R(752489, '_defineProperties', _defineProperties, false, true), false)(J$.G(752505, J$.R(752497, 'Constructor', Constructor, false, false), 'prototype'), J$.R(752513, 'protoProps', protoProps, false, false));
                            if (J$.C(48864, J$.R(752529, 'staticProps', staticProps, false, false)))
                                J$.F(752561, J$.R(752537, '_defineProperties', _defineProperties, false, true), false)(J$.R(752545, 'Constructor', Constructor, false, false), J$.R(752553, 'staticProps', staticProps, false, false));
                            return J$.Rt(752577, J$.R(752569, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(756569, J$e);
                        } finally {
                            if (J$.Fr(756577))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(752729, arguments.callee, this, arguments);
                            arguments = J$.N(752737, 'arguments', arguments, true, false, false);
                            self = J$.N(752745, 'self', self, true, false, false);
                            call = J$.N(752753, 'call', call, true, false, false);
                            if (J$.C(48888, J$.C(48880, J$.R(752625, 'call', call, false, false)) ? J$.C(48872, J$.B(139738, '===', J$.F(752649, J$.R(752633, '_typeof', _typeof, false, true), false)(J$.R(752641, 'call', call, false, false)), J$.T(752657, 'object', 21, false))) ? J$._() : J$.B(139754, '===', J$.U(139746, 'typeof', J$.R(752665, 'call', call, false, false)), J$.T(752673, 'function', 21, false)) : J$._())) {
                                return J$.Rt(752689, J$.R(752681, 'call', call, false, false));
                            }
                            return J$.Rt(752721, J$.F(752713, J$.R(752697, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(752705, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(756585, J$e);
                        } finally {
                            if (J$.Fr(756593))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(752817, arguments.callee, this, arguments);
                            arguments = J$.N(752825, 'arguments', arguments, true, false, false);
                            self = J$.N(752833, 'self', self, true, false, false);
                            if (J$.C(48896, J$.B(139762, '===', J$.R(752761, 'self', self, false, false), void J$.T(752769, 0, 22, false)))) {
                                throw J$.F(752793, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(752777, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(752777, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(752785, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(752809, J$.R(752801, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(756601, J$e);
                        } finally {
                            if (J$.Fr(756609))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(752993, arguments.callee, this, arguments);
                            arguments = J$.N(753001, 'arguments', arguments, true, false, false);
                            o = J$.N(753009, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(752953, '_getPrototypeOf', J$.C(48912, J$.G(752849, J$.I(typeof Object === 'undefined' ? Object = J$.R(752841, 'Object', undefined, true, true) : Object = J$.R(752841, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(752865, J$.I(typeof Object === 'undefined' ? Object = J$.R(752857, 'Object', undefined, true, true) : Object = J$.R(752857, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(752945, function _getPrototypeOf(o) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(752921, arguments.callee, this, arguments);
                                            arguments = J$.N(752929, 'arguments', arguments, true, false, false);
                                            o = J$.N(752937, 'o', o, true, false, false);
                                            return J$.Rt(752913, J$.C(48904, J$.G(752881, J$.R(752873, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(752905, J$.I(typeof Object === 'undefined' ? Object = J$.R(752889, 'Object', undefined, true, true) : Object = J$.R(752889, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(752897, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(756617, J$e);
                                        } finally {
                                            if (J$.Fr(756625))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(752985, J$.F(752977, J$.R(752961, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(752969, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(756633, J$e);
                        } finally {
                            if (J$.Fr(756641))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(753209, arguments.callee, this, arguments);
                            arguments = J$.N(753217, 'arguments', arguments, true, false, false);
                            subClass = J$.N(753225, 'subClass', subClass, true, false, false);
                            superClass = J$.N(753233, 'superClass', superClass, true, false, false);
                            if (J$.C(48928, J$.C(48920, J$.B(139778, '!==', J$.U(139770, 'typeof', J$.R(753017, 'superClass', superClass, false, false)), J$.T(753025, 'function', 21, false))) ? J$.B(139786, '!==', J$.R(753033, 'superClass', superClass, false, false), J$.T(753041, null, 25, false)) : J$._())) {
                                throw J$.F(753065, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(753049, 'TypeError', undefined, true, true) : TypeError = J$.R(753049, 'TypeError', TypeError, true, true)), true)(J$.T(753057, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(753161, J$.R(753073, 'subClass', subClass, false, false), 'prototype', J$.M(753153, J$.I(typeof Object === 'undefined' ? Object = J$.R(753081, 'Object', undefined, true, true) : Object = J$.R(753081, 'Object', Object, true, true)), 'create', false)(J$.C(48936, J$.R(753089, 'superClass', superClass, false, false)) ? J$.G(753105, J$.R(753097, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(753145, {
                                constructor: J$.T(753137, {
                                    value: J$.R(753113, 'subClass', subClass, false, false),
                                    writable: J$.T(753121, true, 23, false),
                                    configurable: J$.T(753129, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(48944, J$.R(753169, 'superClass', superClass, false, false)))
                                J$.F(753201, J$.R(753177, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(753185, 'subClass', subClass, false, false), J$.R(753193, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(756649, J$e);
                        } finally {
                            if (J$.Fr(756657))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(753385, arguments.callee, this, arguments);
                            arguments = J$.N(753393, 'arguments', arguments, true, false, false);
                            o = J$.N(753401, 'o', o, true, false, false);
                            p = J$.N(753409, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(753337, '_setPrototypeOf', J$.C(48952, J$.G(753249, J$.I(typeof Object === 'undefined' ? Object = J$.R(753241, 'Object', undefined, true, true) : Object = J$.R(753241, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(753329, function _setPrototypeOf(o, p) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(753297, arguments.callee, this, arguments);
                                            arguments = J$.N(753305, 'arguments', arguments, true, false, false);
                                            o = J$.N(753313, 'o', o, true, false, false);
                                            p = J$.N(753321, 'p', p, true, false, false);
                                            J$.P(753273, J$.R(753257, 'o', o, false, false), '__proto__', J$.R(753265, 'p', p, false, false));
                                            return J$.Rt(753289, J$.R(753281, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(756665, J$e);
                                        } finally {
                                            if (J$.Fr(756673))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(753377, J$.F(753369, J$.R(753345, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(753353, 'o', o, false, false), J$.R(753361, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(756681, J$e);
                        } finally {
                            if (J$.Fr(756689))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(756297, '_typeof', J$.T(756289, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(756313, '_classCallCheck', J$.T(756305, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(756329, '_defineProperties', J$.T(756321, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(756345, '_createClass', J$.T(756337, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(756361, '_possibleConstructorReturn', J$.T(756353, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(756377, '_assertThisInitialized', J$.T(756369, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(756393, '_getPrototypeOf', J$.T(756385, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(756409, '_inherits', J$.T(756401, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(756425, '_setPrototypeOf', J$.T(756417, _setPrototypeOf, 12, false), true, false, false);
            J$.N(756433, '_require', _require, false, false, false);
            J$.N(756441, 'EventEmitter', EventEmitter, false, false, false);
            J$.N(756449, 'Websocket', Websocket, false, false, false);
            J$.N(756457, 'Utils', Utils, false, false, false);
            J$.N(756465, '_require2', _require2, false, false, false);
            J$.N(756473, 'signRequest', signRequest, false, false, false);
            J$.N(756481, 'WebsocketClient', WebsocketClient, false, false, false);
            J$.T(751465, 'use strict', 21, false);
            J$.F(751489, J$.I(typeof require === 'undefined' ? require = J$.R(751473, 'require', undefined, true, true) : require = J$.R(751473, 'require', require, true, true)), false)(J$.T(751481, 'core-js/modules/es.symbol', 21, false));
            J$.F(751513, J$.I(typeof require === 'undefined' ? require = J$.R(751497, 'require', undefined, true, true) : require = J$.R(751497, 'require', require, true, true)), false)(J$.T(751505, 'core-js/modules/es.symbol.description', 21, false));
            J$.F(751537, J$.I(typeof require === 'undefined' ? require = J$.R(751521, 'require', undefined, true, true) : require = J$.R(751521, 'require', require, true, true)), false)(J$.T(751529, 'core-js/modules/es.symbol.iterator', 21, false));
            J$.F(751561, J$.I(typeof require === 'undefined' ? require = J$.R(751545, 'require', undefined, true, true) : require = J$.R(751545, 'require', require, true, true)), false)(J$.T(751553, 'core-js/modules/es.array.includes', 21, false));
            J$.F(751585, J$.I(typeof require === 'undefined' ? require = J$.R(751569, 'require', undefined, true, true) : require = J$.R(751569, 'require', require, true, true)), false)(J$.T(751577, 'core-js/modules/es.array.iterator', 21, false));
            J$.F(751609, J$.I(typeof require === 'undefined' ? require = J$.R(751593, 'require', undefined, true, true) : require = J$.R(751593, 'require', require, true, true)), false)(J$.T(751601, 'core-js/modules/es.function.bind', 21, false));
            J$.F(751633, J$.I(typeof require === 'undefined' ? require = J$.R(751617, 'require', undefined, true, true) : require = J$.R(751617, 'require', require, true, true)), false)(J$.T(751625, 'core-js/modules/es.object.assign', 21, false));
            J$.F(751657, J$.I(typeof require === 'undefined' ? require = J$.R(751641, 'require', undefined, true, true) : require = J$.R(751641, 'require', require, true, true)), false)(J$.T(751649, 'core-js/modules/es.object.create', 21, false));
            J$.F(751681, J$.I(typeof require === 'undefined' ? require = J$.R(751665, 'require', undefined, true, true) : require = J$.R(751665, 'require', require, true, true)), false)(J$.T(751673, 'core-js/modules/es.object.define-property', 21, false));
            J$.F(751705, J$.I(typeof require === 'undefined' ? require = J$.R(751689, 'require', undefined, true, true) : require = J$.R(751689, 'require', require, true, true)), false)(J$.T(751697, 'core-js/modules/es.object.get-prototype-of', 21, false));
            J$.F(751729, J$.I(typeof require === 'undefined' ? require = J$.R(751713, 'require', undefined, true, true) : require = J$.R(751713, 'require', require, true, true)), false)(J$.T(751721, 'core-js/modules/es.object.set-prototype-of', 21, false));
            J$.F(751753, J$.I(typeof require === 'undefined' ? require = J$.R(751737, 'require', undefined, true, true) : require = J$.R(751737, 'require', require, true, true)), false)(J$.T(751745, 'core-js/modules/es.object.to-string', 21, false));
            J$.F(751777, J$.I(typeof require === 'undefined' ? require = J$.R(751761, 'require', undefined, true, true) : require = J$.R(751761, 'require', require, true, true)), false)(J$.T(751769, 'core-js/modules/es.string.includes', 21, false));
            J$.F(751801, J$.I(typeof require === 'undefined' ? require = J$.R(751785, 'require', undefined, true, true) : require = J$.R(751785, 'require', require, true, true)), false)(J$.T(751793, 'core-js/modules/es.string.iterator', 21, false));
            J$.F(751825, J$.I(typeof require === 'undefined' ? require = J$.R(751809, 'require', undefined, true, true) : require = J$.R(751809, 'require', require, true, true)), false)(J$.T(751817, 'core-js/modules/web.dom-collections.iterator', 21, false));
            var _require = J$.W(753457, '_require', J$.F(753433, J$.I(typeof require === 'undefined' ? require = J$.R(753417, 'require', undefined, true, true) : require = J$.R(753417, 'require', require, true, true)), false)(J$.T(753425, 'events', 21, false)), _require, false, true), EventEmitter = J$.W(753465, 'EventEmitter', J$.G(753449, J$.R(753441, '_require', _require, false, true), 'EventEmitter'), EventEmitter, false, true);
            var Websocket = J$.W(753497, 'Websocket', J$.F(753489, J$.I(typeof require === 'undefined' ? require = J$.R(753473, 'require', undefined, true, true) : require = J$.R(753473, 'require', require, true, true)), false)(J$.T(753481, 'ws', 21, false)), Websocket, false, true);
            var Utils = J$.W(753529, 'Utils', J$.F(753521, J$.I(typeof require === 'undefined' ? require = J$.R(753505, 'require', undefined, true, true) : require = J$.R(753505, 'require', require, true, true)), false)(J$.T(753513, '../utilities.js', 21, false)), Utils, false, true);
            var _require2 = J$.W(753577, '_require2', J$.F(753553, J$.I(typeof require === 'undefined' ? require = J$.R(753537, 'require', undefined, true, true) : require = J$.R(753537, 'require', require, true, true)), false)(J$.T(753545, '../../lib/request_signer', 21, false)), _require2, false, true), signRequest = J$.W(753585, 'signRequest', J$.G(753569, J$.R(753561, '_require2', _require2, false, true), 'signRequest'), signRequest, false, true);
            var WebsocketClient = J$.W(756241, 'WebsocketClient', J$.F(756233, J$.T(756217, function (_EventEmitter) {
                jalangiLabel23:
                    while (true) {
                        try {
                            J$.Fe(756177, arguments.callee, this, arguments);
                            function WebsocketClient(productIDs) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(754321, arguments.callee, this, arguments);
                                            arguments = J$.N(754329, 'arguments', arguments, true, false, false);
                                            productIDs = J$.N(754337, 'productIDs', productIDs, true, false, false);
                                            J$.N(754345, '_this', _this, false, false, false);
                                            J$.N(754353, 'websocketURI', websocketURI, false, false, false);
                                            J$.N(754361, 'auth', auth, false, false, false);
                                            J$.N(754369, '_ref', _ref, false, false, false);
                                            J$.N(754377, '_ref$channels', _ref$channels, false, false, false);
                                            J$.N(754385, 'channels', channels, false, false, false);
                                            var _this;
                                            var websocketURI = J$.W(753713, 'websocketURI', J$.C(48968, J$.C(48960, J$.B(139794, '>', J$.G(753633, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(753625, 'arguments', undefined, true, true) : arguments = J$.R(753625, 'arguments', arguments, true, true)), 'length'), J$.T(753641, 1, 22, false))) ? J$.B(139802, '!==', J$.G(753665, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(753649, 'arguments', undefined, true, true) : arguments = J$.R(753649, 'arguments', arguments, true, true)), J$.T(753657, 1, 22, false)), J$.T(753673, undefined, 24, false)) : J$._()) ? J$.G(753697, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(753681, 'arguments', undefined, true, true) : arguments = J$.R(753681, 'arguments', arguments, true, true)), J$.T(753689, 1, 22, false)) : J$.T(753705, 'wss://ws-feed.pro.coinbase.com', 21, false), websocketURI, false, false);
                                            var auth = J$.W(753809, 'auth', J$.C(48984, J$.C(48976, J$.B(139810, '>', J$.G(753729, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(753721, 'arguments', undefined, true, true) : arguments = J$.R(753721, 'arguments', arguments, true, true)), 'length'), J$.T(753737, 2, 22, false))) ? J$.B(139818, '!==', J$.G(753761, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(753745, 'arguments', undefined, true, true) : arguments = J$.R(753745, 'arguments', arguments, true, true)), J$.T(753753, 2, 22, false)), J$.T(753769, undefined, 24, false)) : J$._()) ? J$.G(753793, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(753777, 'arguments', undefined, true, true) : arguments = J$.R(753777, 'arguments', arguments, true, true)), J$.T(753785, 2, 22, false)) : J$.T(753801, null, 25, false), auth, false, false);
                                            var _ref = J$.W(753953, '_ref', J$.C(49000, J$.C(48992, J$.B(139826, '>', J$.G(753825, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(753817, 'arguments', undefined, true, true) : arguments = J$.R(753817, 'arguments', arguments, true, true)), 'length'), J$.T(753833, 3, 22, false))) ? J$.B(139834, '!==', J$.G(753857, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(753841, 'arguments', undefined, true, true) : arguments = J$.R(753841, 'arguments', arguments, true, true)), J$.T(753849, 3, 22, false)), J$.T(753865, undefined, 24, false)) : J$._()) ? J$.G(753889, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(753873, 'arguments', undefined, true, true) : arguments = J$.R(753873, 'arguments', arguments, true, true)), J$.T(753881, 3, 22, false)) : J$.T(753897, {}, 11, false), _ref, false, false), _ref$channels = J$.W(753961, '_ref$channels', J$.G(753913, J$.R(753905, '_ref', _ref, false, false), 'channels'), _ref$channels, false, false), channels = J$.W(753969, 'channels', J$.C(49008, J$.B(139842, '===', J$.R(753921, '_ref$channels', _ref$channels, false, false), void J$.T(753929, 0, 22, false))) ? J$.T(753937, null, 25, false) : J$.R(753945, '_ref$channels', _ref$channels, false, false), channels, false, false);
                                            J$.F(754001, J$.R(753977, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(753985, 'this', this, false, false), J$.R(753993, 'WebsocketClient', WebsocketClient, false, false));
                                            _this = J$.W(754073, '_this', J$.F(754065, J$.R(754009, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(754017, 'this', this, false, false), J$.M(754057, J$.F(754041, J$.R(754025, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(754033, 'WebsocketClient', WebsocketClient, false, false)), 'call', false)(J$.R(754049, 'this', this, false, false))), _this, false, false);
                                            J$.P(754113, J$.R(754081, '_this', _this, false, false), 'productIDs', J$.M(754105, J$.R(754089, 'Utils', Utils, false, true), 'determineProductIDs', false)(J$.R(754097, 'productIDs', productIDs, false, false)));
                                            J$.P(754137, J$.R(754121, '_this', _this, false, false), 'websocketURI', J$.R(754129, 'websocketURI', websocketURI, false, false));
                                            J$.P(754177, J$.R(754145, '_this', _this, false, false), 'auth', J$.M(754169, J$.R(754153, 'Utils', Utils, false, true), 'checkAuth', false)(J$.R(754161, 'auth', auth, false, false)));
                                            J$.P(754217, J$.R(754185, '_this', _this, false, false), 'channels', J$.C(49016, J$.R(754193, 'channels', channels, false, false)) ? J$._() : J$.T(754209, [J$.T(754201, 'full', 21, false)], 10, false));
                                            if (J$.C(49024, J$.U(139850, '!', J$.M(754249, J$.G(754233, J$.R(754225, '_this', _this, false, false), 'channels'), 'includes', false)(J$.T(754241, 'heartbeat', 21, false))))) {
                                                J$.M(754281, J$.G(754265, J$.R(754257, '_this', _this, false, false), 'channels'), 'push', false)(J$.T(754273, 'heartbeat', 21, false));
                                            }
                                            J$.M(754297, J$.R(754289, '_this', _this, false, false), 'connect', false)();
                                            return J$.Rt(754313, J$.R(754305, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(756697, J$e);
                                        } finally {
                                            if (J$.Fr(756705))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(756185, 'arguments', arguments, true, false, false);
                            _EventEmitter = J$.N(756193, '_EventEmitter', _EventEmitter, true, false, false);
                            WebsocketClient = J$.N(756209, 'WebsocketClient', J$.T(756201, WebsocketClient, 12, false), true, false, false);
                            J$.F(753617, J$.R(753593, '_inherits', _inherits, false, true), false)(J$.R(753601, 'WebsocketClient', WebsocketClient, false, false), J$.R(753609, '_EventEmitter', _EventEmitter, false, false));
                            J$.F(756153, J$.R(754393, '_createClass', _createClass, false, true), false)(J$.R(754401, 'WebsocketClient', WebsocketClient, false, false), J$.T(756145, [
                                J$.T(754785, {
                                    key: J$.T(754409, 'connect', 21, false),
                                    value: J$.T(754777, function connect() {
                                        jalangiLabel14:
                                            while (true) {
                                                try {
                                                    J$.Fe(754761, arguments.callee, this, arguments);
                                                    arguments = J$.N(754769, 'arguments', arguments, true, false, false);
                                                    if (J$.C(49032, J$.G(754425, J$.R(754417, 'this', this, false, false), 'socket'))) {
                                                        J$.M(754449, J$.G(754441, J$.R(754433, 'this', this, false, false), 'socket'), 'close', false)();
                                                    }
                                                    J$.P(754497, J$.R(754457, 'this', this, false, false), 'socket', J$.F(754489, J$.R(754465, 'Websocket', Websocket, false, true), true)(J$.G(754481, J$.R(754473, 'this', this, false, false), 'websocketURI')));
                                                    J$.M(754561, J$.G(754513, J$.R(754505, 'this', this, false, false), 'socket'), 'on', false)(J$.T(754521, 'message', 21, false), J$.M(754553, J$.G(754537, J$.R(754529, 'this', this, false, false), 'onMessage'), 'bind', false)(J$.R(754545, 'this', this, false, false)));
                                                    J$.M(754625, J$.G(754577, J$.R(754569, 'this', this, false, false), 'socket'), 'on', false)(J$.T(754585, 'open', 21, false), J$.M(754617, J$.G(754601, J$.R(754593, 'this', this, false, false), 'onOpen'), 'bind', false)(J$.R(754609, 'this', this, false, false)));
                                                    J$.M(754689, J$.G(754641, J$.R(754633, 'this', this, false, false), 'socket'), 'on', false)(J$.T(754649, 'close', 21, false), J$.M(754681, J$.G(754665, J$.R(754657, 'this', this, false, false), 'onClose'), 'bind', false)(J$.R(754673, 'this', this, false, false)));
                                                    J$.M(754753, J$.G(754705, J$.R(754697, 'this', this, false, false), 'socket'), 'on', false)(J$.T(754713, 'error', 21, false), J$.M(754745, J$.G(754729, J$.R(754721, 'this', this, false, false), 'onError'), 'bind', false)(J$.R(754737, 'this', this, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(756713, J$e);
                                                } finally {
                                                    if (J$.Fr(756721))
                                                        continue jalangiLabel14;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(754913, {
                                    key: J$.T(754793, 'disconnect', 21, false),
                                    value: J$.T(754905, function disconnect() {
                                        jalangiLabel15:
                                            while (true) {
                                                try {
                                                    J$.Fe(754889, arguments.callee, this, arguments);
                                                    arguments = J$.N(754897, 'arguments', arguments, true, false, false);
                                                    if (J$.C(49040, J$.U(139858, '!', J$.G(754809, J$.R(754801, 'this', this, false, false), 'socket')))) {
                                                        throw J$.F(754833, J$.I(typeof Error === 'undefined' ? Error = J$.R(754817, 'Error', undefined, true, true) : Error = J$.R(754817, 'Error', Error, true, true)), true)(J$.T(754825, 'Could not disconnect (not connected)', 21, false));
                                                    }
                                                    J$.M(754857, J$.G(754849, J$.R(754841, 'this', this, false, false), 'socket'), 'close', false)();
                                                    J$.P(754881, J$.R(754865, 'this', this, false, false), 'socket', J$.T(754873, null, 25, false));
                                                } catch (J$e) {
                                                    J$.Ex(756729, J$e);
                                                } finally {
                                                    if (J$.Fr(756737))
                                                        continue jalangiLabel15;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(755297, {
                                    key: J$.T(754921, '_sendSubscription', 21, false),
                                    value: J$.T(755289, function _sendSubscription(type, _ref2) {
                                        jalangiLabel16:
                                            while (true) {
                                                try {
                                                    J$.Fe(755225, arguments.callee, this, arguments);
                                                    arguments = J$.N(755233, 'arguments', arguments, true, false, false);
                                                    type = J$.N(755241, 'type', type, true, false, false);
                                                    _ref2 = J$.N(755249, '_ref2', _ref2, true, false, false);
                                                    J$.N(755257, 'product_ids', product_ids, false, false, false);
                                                    J$.N(755265, 'channels', channels, false, false, false);
                                                    J$.N(755273, 'message', message, false, false, false);
                                                    J$.N(755281, 'sig', sig, false, false, false);
                                                    var product_ids = J$.W(754961, 'product_ids', J$.G(754937, J$.R(754929, '_ref2', _ref2, false, false), 'product_ids'), product_ids, false, false), channels = J$.W(754969, 'channels', J$.G(754953, J$.R(754945, '_ref2', _ref2, false, false), 'channels'), channels, false, false);
                                                    var message = J$.W(754993, 'message', J$.T(754985, { type: J$.R(754977, 'type', type, false, false) }, 11, false), message, false, false);
                                                    if (J$.C(49048, J$.R(755001, 'channels', channels, false, false))) {
                                                        J$.P(755025, J$.R(755009, 'message', message, false, false), 'channels', J$.R(755017, 'channels', channels, false, false));
                                                    }
                                                    if (J$.C(49056, J$.R(755033, 'product_ids', product_ids, false, false))) {
                                                        J$.P(755057, J$.R(755041, 'message', message, false, false), 'product_ids', J$.R(755049, 'product_ids', product_ids, false, false));
                                                    }
                                                    if (J$.C(49064, J$.G(755081, J$.G(755073, J$.R(755065, 'this', this, false, false), 'auth'), 'secret'))) {
                                                        var sig = J$.W(755137, 'sig', J$.F(755129, J$.R(755089, 'signRequest', signRequest, false, true), false)(J$.G(755105, J$.R(755097, 'this', this, false, false), 'auth'), J$.T(755113, 'GET', 21, false), J$.T(755121, '/users/self/verify', 21, false)), sig, false, false);
                                                        J$.M(755169, J$.I(typeof Object === 'undefined' ? Object = J$.R(755145, 'Object', undefined, true, true) : Object = J$.R(755145, 'Object', Object, true, true)), 'assign', false)(J$.R(755153, 'message', message, false, false), J$.R(755161, 'sig', sig, false, false));
                                                    }
                                                    J$.M(755217, J$.G(755185, J$.R(755177, 'this', this, false, false), 'socket'), 'send', false)(J$.M(755209, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(755193, 'JSON', undefined, true, true) : JSON = J$.R(755193, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(755201, 'message', message, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(756745, J$e);
                                                } finally {
                                                    if (J$.Fr(756753))
                                                        continue jalangiLabel16;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(755457, {
                                    key: J$.T(755305, 'subscribe', 21, false),
                                    value: J$.T(755449, function subscribe(_ref3) {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(755409, arguments.callee, this, arguments);
                                                    arguments = J$.N(755417, 'arguments', arguments, true, false, false);
                                                    _ref3 = J$.N(755425, '_ref3', _ref3, true, false, false);
                                                    J$.N(755433, 'product_ids', product_ids, false, false, false);
                                                    J$.N(755441, 'channels', channels, false, false, false);
                                                    var product_ids = J$.W(755345, 'product_ids', J$.G(755321, J$.R(755313, '_ref3', _ref3, false, false), 'product_ids'), product_ids, false, false), channels = J$.W(755353, 'channels', J$.G(755337, J$.R(755329, '_ref3', _ref3, false, false), 'channels'), channels, false, false);
                                                    J$.M(755401, J$.R(755361, 'this', this, false, false), '_sendSubscription', false)(J$.T(755369, 'subscribe', 21, false), J$.T(755393, {
                                                        product_ids: J$.R(755377, 'product_ids', product_ids, false, false),
                                                        channels: J$.R(755385, 'channels', channels, false, false)
                                                    }, 11, false));
                                                } catch (J$e) {
                                                    J$.Ex(756761, J$e);
                                                } finally {
                                                    if (J$.Fr(756769))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(755617, {
                                    key: J$.T(755465, 'unsubscribe', 21, false),
                                    value: J$.T(755609, function unsubscribe(_ref4) {
                                        jalangiLabel18:
                                            while (true) {
                                                try {
                                                    J$.Fe(755569, arguments.callee, this, arguments);
                                                    arguments = J$.N(755577, 'arguments', arguments, true, false, false);
                                                    _ref4 = J$.N(755585, '_ref4', _ref4, true, false, false);
                                                    J$.N(755593, 'product_ids', product_ids, false, false, false);
                                                    J$.N(755601, 'channels', channels, false, false, false);
                                                    var product_ids = J$.W(755505, 'product_ids', J$.G(755481, J$.R(755473, '_ref4', _ref4, false, false), 'product_ids'), product_ids, false, false), channels = J$.W(755513, 'channels', J$.G(755497, J$.R(755489, '_ref4', _ref4, false, false), 'channels'), channels, false, false);
                                                    J$.M(755561, J$.R(755521, 'this', this, false, false), '_sendSubscription', false)(J$.T(755529, 'unsubscribe', 21, false), J$.T(755553, {
                                                        product_ids: J$.R(755537, 'product_ids', product_ids, false, false),
                                                        channels: J$.R(755545, 'channels', channels, false, false)
                                                    }, 11, false));
                                                } catch (J$e) {
                                                    J$.Ex(756777, J$e);
                                                } finally {
                                                    if (J$.Fr(756785))
                                                        continue jalangiLabel18;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(755737, {
                                    key: J$.T(755625, 'onOpen', 21, false),
                                    value: J$.T(755729, function onOpen() {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(755713, arguments.callee, this, arguments);
                                                    arguments = J$.N(755721, 'arguments', arguments, true, false, false);
                                                    J$.M(755649, J$.R(755633, 'this', this, false, false), 'emit', false)(J$.T(755641, 'open', 21, false));
                                                    J$.M(755705, J$.R(755657, 'this', this, false, false), 'subscribe', false)(J$.T(755697, {
                                                        product_ids: J$.G(755673, J$.R(755665, 'this', this, false, false), 'productIDs'),
                                                        channels: J$.G(755689, J$.R(755681, 'this', this, false, false), 'channels')
                                                    }, 11, false));
                                                } catch (J$e) {
                                                    J$.Ex(756793, J$e);
                                                } finally {
                                                    if (J$.Fr(756801))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(755825, {
                                    key: J$.T(755745, 'onClose', 21, false),
                                    value: J$.T(755817, function onClose() {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(755801, arguments.callee, this, arguments);
                                                    arguments = J$.N(755809, 'arguments', arguments, true, false, false);
                                                    J$.P(755769, J$.R(755753, 'this', this, false, false), 'socket', J$.T(755761, null, 25, false));
                                                    J$.M(755793, J$.R(755777, 'this', this, false, false), 'emit', false)(J$.T(755785, 'close', 21, false));
                                                } catch (J$e) {
                                                    J$.Ex(756809, J$e);
                                                } finally {
                                                    if (J$.Fr(756817))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(755993, {
                                    key: J$.T(755833, 'onMessage', 21, false),
                                    value: J$.T(755985, function onMessage(data) {
                                        jalangiLabel21:
                                            while (true) {
                                                try {
                                                    J$.Fe(755953, arguments.callee, this, arguments);
                                                    arguments = J$.N(755961, 'arguments', arguments, true, false, false);
                                                    data = J$.N(755969, 'data', data, true, false, false);
                                                    J$.N(755977, 'message', message, false, false, false);
                                                    var message = J$.W(755865, 'message', J$.M(755857, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(755841, 'JSON', undefined, true, true) : JSON = J$.R(755841, 'JSON', JSON, true, true)), 'parse', false)(J$.R(755849, 'data', data, false, false)), message, false, false);
                                                    if (J$.C(49072, J$.B(139866, '===', J$.G(755881, J$.R(755873, 'message', message, false, false), 'type'), J$.T(755889, 'error', 21, false)))) {
                                                        J$.M(755913, J$.R(755897, 'this', this, false, false), 'onError', false)(J$.R(755905, 'message', message, false, false));
                                                    } else {
                                                        J$.M(755945, J$.R(755921, 'this', this, false, false), 'emit', false)(J$.T(755929, 'message', 21, false), J$.R(755937, 'message', message, false, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(756825, J$e);
                                                } finally {
                                                    if (J$.Fr(756833))
                                                        continue jalangiLabel21;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(756137, {
                                    key: J$.T(756001, 'onError', 21, false),
                                    value: J$.T(756129, function onError(err) {
                                        jalangiLabel22:
                                            while (true) {
                                                try {
                                                    J$.Fe(756105, arguments.callee, this, arguments);
                                                    arguments = J$.N(756113, 'arguments', arguments, true, false, false);
                                                    err = J$.N(756121, 'err', err, true, false, false);
                                                    if (J$.C(49080, J$.U(139874, '!', J$.R(756009, 'err', err, false, false)))) {
                                                        return J$.Rt(756017, undefined);
                                                    }
                                                    if (J$.C(49088, J$.B(139882, '===', J$.G(756033, J$.R(756025, 'err', err, false, false), 'message'), J$.T(756041, 'unexpected server response (429)', 21, false)))) {
                                                        throw J$.F(756065, J$.I(typeof Error === 'undefined' ? Error = J$.R(756049, 'Error', undefined, true, true) : Error = J$.R(756049, 'Error', Error, true, true)), true)(J$.T(756057, 'You are connecting too fast and are being throttled! Make sure you subscribe to multiple books on one connection.', 21, false));
                                                    }
                                                    J$.M(756097, J$.R(756073, 'this', this, false, false), 'emit', false)(J$.T(756081, 'error', 21, false), J$.R(756089, 'err', err, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(756841, J$e);
                                                } finally {
                                                    if (J$.Fr(756849))
                                                        continue jalangiLabel22;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(756169, J$.R(756161, 'WebsocketClient', WebsocketClient, false, false));
                        } catch (J$e) {
                            J$.Ex(756857, J$e);
                        } finally {
                            if (J$.Fr(756865))
                                continue jalangiLabel23;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(756225, 'EventEmitter', EventEmitter, false, true)), WebsocketClient, false, true);
            J$.P(756273, J$.I(typeof module === 'undefined' ? module = J$.R(756249, 'module', undefined, true, true) : module = J$.R(756249, 'module', module, true, true)), 'exports', exports = J$.W(756265, 'exports', J$.R(756257, 'WebsocketClient', WebsocketClient, false, true), J$.I(typeof exports === 'undefined' ? undefined : exports), true, true));
        } catch (J$e) {
            J$.Ex(756873, J$e);
        } finally {
            if (J$.Sr(756881))
                continue jalangiLabel24;
            else
                break jalangiLabel24;
        }
    }
// JALANGI DO NOT INSTRUMENT
