// ecc-jsbn/lib/sec.js
J$.noInstrEval = false;
jalangiLabel14:
    while (true) {
        try {
            J$.Se(2784129, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestRequest/node_modules/ecc-jsbn/lib/sec.js');
            function X9ECParameters(curve, g, n, h) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(2780713, arguments.callee, this, arguments);
                            arguments = J$.N(2780721, 'arguments', arguments, true, false, false);
                            curve = J$.N(2780729, 'curve', curve, true, false, false);
                            g = J$.N(2780737, 'g', g, true, false, false);
                            n = J$.N(2780745, 'n', n, true, false, false);
                            h = J$.N(2780753, 'h', h, true, false, false);
                            J$.P(2780633, J$.R(2780617, 'this', this, false, false), 'curve', J$.R(2780625, 'curve', curve, false, false));
                            J$.P(2780657, J$.R(2780641, 'this', this, false, false), 'g', J$.R(2780649, 'g', g, false, false));
                            J$.P(2780681, J$.R(2780665, 'this', this, false, false), 'n', J$.R(2780673, 'n', n, false, false));
                            J$.P(2780705, J$.R(2780689, 'this', this, false, false), 'h', J$.R(2780697, 'h', h, false, false));
                        } catch (J$e) {
                            J$.Ex(2784377, J$e);
                        } finally {
                            if (J$.Fr(2784385))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function x9getCurve() {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(2780785, arguments.callee, this, arguments);
                            arguments = J$.N(2780793, 'arguments', arguments, true, false, false);
                            return J$.Rt(2780777, J$.G(2780769, J$.R(2780761, 'this', this, false, false), 'curve'));
                        } catch (J$e) {
                            J$.Ex(2784393, J$e);
                        } finally {
                            if (J$.Fr(2784401))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function x9getG() {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(2780825, arguments.callee, this, arguments);
                            arguments = J$.N(2780833, 'arguments', arguments, true, false, false);
                            return J$.Rt(2780817, J$.G(2780809, J$.R(2780801, 'this', this, false, false), 'g'));
                        } catch (J$e) {
                            J$.Ex(2784409, J$e);
                        } finally {
                            if (J$.Fr(2784417))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function x9getN() {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(2780865, arguments.callee, this, arguments);
                            arguments = J$.N(2780873, 'arguments', arguments, true, false, false);
                            return J$.Rt(2780857, J$.G(2780849, J$.R(2780841, 'this', this, false, false), 'n'));
                        } catch (J$e) {
                            J$.Ex(2784425, J$e);
                        } finally {
                            if (J$.Fr(2784433))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function x9getH() {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(2780905, arguments.callee, this, arguments);
                            arguments = J$.N(2780913, 'arguments', arguments, true, false, false);
                            return J$.Rt(2780897, J$.G(2780889, J$.R(2780881, 'this', this, false, false), 'h'));
                        } catch (J$e) {
                            J$.Ex(2784441, J$e);
                        } finally {
                            if (J$.Fr(2784449))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function fromHex(s) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(2781089, arguments.callee, this, arguments);
                            arguments = J$.N(2781097, 'arguments', arguments, true, false, false);
                            s = J$.N(2781105, 's', s, true, false, false);
                            return J$.Rt(2781081, J$.F(2781073, J$.R(2781049, 'BigInteger', BigInteger, false, true), true)(J$.R(2781057, 's', s, false, false), J$.T(2781065, 16, 22, false)));
                        } catch (J$e) {
                            J$.Ex(2784457, J$e);
                        } finally {
                            if (J$.Fr(2784465))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp128r1() {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(2781417, arguments.callee, this, arguments);
                            arguments = J$.N(2781425, 'arguments', arguments, true, false, false);
                            J$.N(2781433, 'p', p, false, false, false);
                            J$.N(2781441, 'a', a, false, false, false);
                            J$.N(2781449, 'b', b, false, false, false);
                            J$.N(2781457, 'n', n, false, false, false);
                            J$.N(2781465, 'h', h, false, false, false);
                            J$.N(2781473, 'curve', curve, false, false, false);
                            J$.N(2781481, 'G', G, false, false, false);
                            var p = J$.W(2781137, 'p', J$.F(2781129, J$.R(2781113, 'fromHex', fromHex, false, true), false)(J$.T(2781121, 'FFFFFFFDFFFFFFFFFFFFFFFFFFFFFFFF', 21, false)), p, false, false);
                            var a = J$.W(2781169, 'a', J$.F(2781161, J$.R(2781145, 'fromHex', fromHex, false, true), false)(J$.T(2781153, 'FFFFFFFDFFFFFFFFFFFFFFFFFFFFFFFC', 21, false)), a, false, false);
                            var b = J$.W(2781201, 'b', J$.F(2781193, J$.R(2781177, 'fromHex', fromHex, false, true), false)(J$.T(2781185, 'E87579C11079F43DD824993C2CEE5ED3', 21, false)), b, false, false);
                            var n = J$.W(2781233, 'n', J$.F(2781225, J$.R(2781209, 'fromHex', fromHex, false, true), false)(J$.T(2781217, 'FFFFFFFE0000000075A30D1B9038A115', 21, false)), n, false, false);
                            var h = J$.W(2781257, 'h', J$.G(2781249, J$.R(2781241, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2781305, 'curve', J$.F(2781297, J$.R(2781265, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2781273, 'p', p, false, false), J$.R(2781281, 'a', a, false, false), J$.R(2781289, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2781353, 'G', J$.M(2781345, J$.R(2781313, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(410138, '+', J$.B(410130, '+', J$.T(2781321, '04', 21, false), J$.T(2781329, '161FF7528B899B2D0C28607CA52C5B86', 21, false)), J$.T(2781337, 'CF5AC8395BAFEB13C02DA292DDED7A83', 21, false))), G, false, false);
                            return J$.Rt(2781409, J$.F(2781401, J$.R(2781361, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2781369, 'curve', curve, false, false), J$.R(2781377, 'G', G, false, false), J$.R(2781385, 'n', n, false, false), J$.R(2781393, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2784473, J$e);
                        } finally {
                            if (J$.Fr(2784481))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp160k1() {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(2781785, arguments.callee, this, arguments);
                            arguments = J$.N(2781793, 'arguments', arguments, true, false, false);
                            J$.N(2781801, 'p', p, false, false, false);
                            J$.N(2781809, 'a', a, false, false, false);
                            J$.N(2781817, 'b', b, false, false, false);
                            J$.N(2781825, 'n', n, false, false, false);
                            J$.N(2781833, 'h', h, false, false, false);
                            J$.N(2781841, 'curve', curve, false, false, false);
                            J$.N(2781849, 'G', G, false, false, false);
                            var p = J$.W(2781513, 'p', J$.F(2781505, J$.R(2781489, 'fromHex', fromHex, false, true), false)(J$.T(2781497, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFAC73', 21, false)), p, false, false);
                            var a = J$.W(2781537, 'a', J$.G(2781529, J$.R(2781521, 'BigInteger', BigInteger, false, true), 'ZERO'), a, false, false);
                            var b = J$.W(2781569, 'b', J$.F(2781561, J$.R(2781545, 'fromHex', fromHex, false, true), false)(J$.T(2781553, '7', 21, false)), b, false, false);
                            var n = J$.W(2781601, 'n', J$.F(2781593, J$.R(2781577, 'fromHex', fromHex, false, true), false)(J$.T(2781585, '0100000000000000000001B8FA16DFAB9ACA16B6B3', 21, false)), n, false, false);
                            var h = J$.W(2781625, 'h', J$.G(2781617, J$.R(2781609, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2781673, 'curve', J$.F(2781665, J$.R(2781633, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2781641, 'p', p, false, false), J$.R(2781649, 'a', a, false, false), J$.R(2781657, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2781721, 'G', J$.M(2781713, J$.R(2781681, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(410154, '+', J$.B(410146, '+', J$.T(2781689, '04', 21, false), J$.T(2781697, '3B4C382CE37AA192A4019E763036F4F5DD4D7EBB', 21, false)), J$.T(2781705, '938CF935318FDCED6BC28286531733C3F03C4FEE', 21, false))), G, false, false);
                            return J$.Rt(2781777, J$.F(2781769, J$.R(2781729, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2781737, 'curve', curve, false, false), J$.R(2781745, 'G', G, false, false), J$.R(2781753, 'n', n, false, false), J$.R(2781761, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2784489, J$e);
                        } finally {
                            if (J$.Fr(2784497))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp160r1() {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(2782161, arguments.callee, this, arguments);
                            arguments = J$.N(2782169, 'arguments', arguments, true, false, false);
                            J$.N(2782177, 'p', p, false, false, false);
                            J$.N(2782185, 'a', a, false, false, false);
                            J$.N(2782193, 'b', b, false, false, false);
                            J$.N(2782201, 'n', n, false, false, false);
                            J$.N(2782209, 'h', h, false, false, false);
                            J$.N(2782217, 'curve', curve, false, false, false);
                            J$.N(2782225, 'G', G, false, false, false);
                            var p = J$.W(2781881, 'p', J$.F(2781873, J$.R(2781857, 'fromHex', fromHex, false, true), false)(J$.T(2781865, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF7FFFFFFF', 21, false)), p, false, false);
                            var a = J$.W(2781913, 'a', J$.F(2781905, J$.R(2781889, 'fromHex', fromHex, false, true), false)(J$.T(2781897, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF7FFFFFFC', 21, false)), a, false, false);
                            var b = J$.W(2781945, 'b', J$.F(2781937, J$.R(2781921, 'fromHex', fromHex, false, true), false)(J$.T(2781929, '1C97BEFC54BD7A8B65ACF89F81D4D4ADC565FA45', 21, false)), b, false, false);
                            var n = J$.W(2781977, 'n', J$.F(2781969, J$.R(2781953, 'fromHex', fromHex, false, true), false)(J$.T(2781961, '0100000000000000000001F4C8F927AED3CA752257', 21, false)), n, false, false);
                            var h = J$.W(2782001, 'h', J$.G(2781993, J$.R(2781985, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2782049, 'curve', J$.F(2782041, J$.R(2782009, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2782017, 'p', p, false, false), J$.R(2782025, 'a', a, false, false), J$.R(2782033, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2782097, 'G', J$.M(2782089, J$.R(2782057, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(410170, '+', J$.B(410162, '+', J$.T(2782065, '04', 21, false), J$.T(2782073, '4A96B5688EF573284664698968C38BB913CBFC82', 21, false)), J$.T(2782081, '23A628553168947D59DCC912042351377AC5FB32', 21, false))), G, false, false);
                            return J$.Rt(2782153, J$.F(2782145, J$.R(2782105, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2782113, 'curve', curve, false, false), J$.R(2782121, 'G', G, false, false), J$.R(2782129, 'n', n, false, false), J$.R(2782137, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2784505, J$e);
                        } finally {
                            if (J$.Fr(2784513))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp192k1() {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(2782529, arguments.callee, this, arguments);
                            arguments = J$.N(2782537, 'arguments', arguments, true, false, false);
                            J$.N(2782545, 'p', p, false, false, false);
                            J$.N(2782553, 'a', a, false, false, false);
                            J$.N(2782561, 'b', b, false, false, false);
                            J$.N(2782569, 'n', n, false, false, false);
                            J$.N(2782577, 'h', h, false, false, false);
                            J$.N(2782585, 'curve', curve, false, false, false);
                            J$.N(2782593, 'G', G, false, false, false);
                            var p = J$.W(2782257, 'p', J$.F(2782249, J$.R(2782233, 'fromHex', fromHex, false, true), false)(J$.T(2782241, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFEE37', 21, false)), p, false, false);
                            var a = J$.W(2782281, 'a', J$.G(2782273, J$.R(2782265, 'BigInteger', BigInteger, false, true), 'ZERO'), a, false, false);
                            var b = J$.W(2782313, 'b', J$.F(2782305, J$.R(2782289, 'fromHex', fromHex, false, true), false)(J$.T(2782297, '3', 21, false)), b, false, false);
                            var n = J$.W(2782345, 'n', J$.F(2782337, J$.R(2782321, 'fromHex', fromHex, false, true), false)(J$.T(2782329, 'FFFFFFFFFFFFFFFFFFFFFFFE26F2FC170F69466A74DEFD8D', 21, false)), n, false, false);
                            var h = J$.W(2782369, 'h', J$.G(2782361, J$.R(2782353, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2782417, 'curve', J$.F(2782409, J$.R(2782377, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2782385, 'p', p, false, false), J$.R(2782393, 'a', a, false, false), J$.R(2782401, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2782465, 'G', J$.M(2782457, J$.R(2782425, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(410186, '+', J$.B(410178, '+', J$.T(2782433, '04', 21, false), J$.T(2782441, 'DB4FF10EC057E9AE26B07D0280B7F4341DA5D1B1EAE06C7D', 21, false)), J$.T(2782449, '9B2F2F6D9C5628A7844163D015BE86344082AA88D95E2F9D', 21, false))), G, false, false);
                            return J$.Rt(2782521, J$.F(2782513, J$.R(2782473, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2782481, 'curve', curve, false, false), J$.R(2782489, 'G', G, false, false), J$.R(2782497, 'n', n, false, false), J$.R(2782505, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2784521, J$e);
                        } finally {
                            if (J$.Fr(2784529))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp192r1() {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(2782905, arguments.callee, this, arguments);
                            arguments = J$.N(2782913, 'arguments', arguments, true, false, false);
                            J$.N(2782921, 'p', p, false, false, false);
                            J$.N(2782929, 'a', a, false, false, false);
                            J$.N(2782937, 'b', b, false, false, false);
                            J$.N(2782945, 'n', n, false, false, false);
                            J$.N(2782953, 'h', h, false, false, false);
                            J$.N(2782961, 'curve', curve, false, false, false);
                            J$.N(2782969, 'G', G, false, false, false);
                            var p = J$.W(2782625, 'p', J$.F(2782617, J$.R(2782601, 'fromHex', fromHex, false, true), false)(J$.T(2782609, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFFFFFFFFFFFF', 21, false)), p, false, false);
                            var a = J$.W(2782657, 'a', J$.F(2782649, J$.R(2782633, 'fromHex', fromHex, false, true), false)(J$.T(2782641, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFFFFFFFFFFFC', 21, false)), a, false, false);
                            var b = J$.W(2782689, 'b', J$.F(2782681, J$.R(2782665, 'fromHex', fromHex, false, true), false)(J$.T(2782673, '64210519E59C80E70FA7E9AB72243049FEB8DEECC146B9B1', 21, false)), b, false, false);
                            var n = J$.W(2782721, 'n', J$.F(2782713, J$.R(2782697, 'fromHex', fromHex, false, true), false)(J$.T(2782705, 'FFFFFFFFFFFFFFFFFFFFFFFF99DEF836146BC9B1B4D22831', 21, false)), n, false, false);
                            var h = J$.W(2782745, 'h', J$.G(2782737, J$.R(2782729, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2782793, 'curve', J$.F(2782785, J$.R(2782753, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2782761, 'p', p, false, false), J$.R(2782769, 'a', a, false, false), J$.R(2782777, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2782841, 'G', J$.M(2782833, J$.R(2782801, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(410202, '+', J$.B(410194, '+', J$.T(2782809, '04', 21, false), J$.T(2782817, '188DA80EB03090F67CBF20EB43A18800F4FF0AFD82FF1012', 21, false)), J$.T(2782825, '07192B95FFC8DA78631011ED6B24CDD573F977A11E794811', 21, false))), G, false, false);
                            return J$.Rt(2782897, J$.F(2782889, J$.R(2782849, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2782857, 'curve', curve, false, false), J$.R(2782865, 'G', G, false, false), J$.R(2782873, 'n', n, false, false), J$.R(2782881, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2784537, J$e);
                        } finally {
                            if (J$.Fr(2784545))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp224r1() {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(2783281, arguments.callee, this, arguments);
                            arguments = J$.N(2783289, 'arguments', arguments, true, false, false);
                            J$.N(2783297, 'p', p, false, false, false);
                            J$.N(2783305, 'a', a, false, false, false);
                            J$.N(2783313, 'b', b, false, false, false);
                            J$.N(2783321, 'n', n, false, false, false);
                            J$.N(2783329, 'h', h, false, false, false);
                            J$.N(2783337, 'curve', curve, false, false, false);
                            J$.N(2783345, 'G', G, false, false, false);
                            var p = J$.W(2783001, 'p', J$.F(2782993, J$.R(2782977, 'fromHex', fromHex, false, true), false)(J$.T(2782985, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF000000000000000000000001', 21, false)), p, false, false);
                            var a = J$.W(2783033, 'a', J$.F(2783025, J$.R(2783009, 'fromHex', fromHex, false, true), false)(J$.T(2783017, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFFFFFFFFFFFFFFFFFFFE', 21, false)), a, false, false);
                            var b = J$.W(2783065, 'b', J$.F(2783057, J$.R(2783041, 'fromHex', fromHex, false, true), false)(J$.T(2783049, 'B4050A850C04B3ABF54132565044B0B7D7BFD8BA270B39432355FFB4', 21, false)), b, false, false);
                            var n = J$.W(2783097, 'n', J$.F(2783089, J$.R(2783073, 'fromHex', fromHex, false, true), false)(J$.T(2783081, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFF16A2E0B8F03E13DD29455C5C2A3D', 21, false)), n, false, false);
                            var h = J$.W(2783121, 'h', J$.G(2783113, J$.R(2783105, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2783169, 'curve', J$.F(2783161, J$.R(2783129, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2783137, 'p', p, false, false), J$.R(2783145, 'a', a, false, false), J$.R(2783153, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2783217, 'G', J$.M(2783209, J$.R(2783177, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(410218, '+', J$.B(410210, '+', J$.T(2783185, '04', 21, false), J$.T(2783193, 'B70E0CBD6BB4BF7F321390B94A03C1D356C21122343280D6115C1D21', 21, false)), J$.T(2783201, 'BD376388B5F723FB4C22DFE6CD4375A05A07476444D5819985007E34', 21, false))), G, false, false);
                            return J$.Rt(2783273, J$.F(2783265, J$.R(2783225, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2783233, 'curve', curve, false, false), J$.R(2783241, 'G', G, false, false), J$.R(2783249, 'n', n, false, false), J$.R(2783257, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2784553, J$e);
                        } finally {
                            if (J$.Fr(2784561))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp256r1() {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(2783657, arguments.callee, this, arguments);
                            arguments = J$.N(2783665, 'arguments', arguments, true, false, false);
                            J$.N(2783673, 'p', p, false, false, false);
                            J$.N(2783681, 'a', a, false, false, false);
                            J$.N(2783689, 'b', b, false, false, false);
                            J$.N(2783697, 'n', n, false, false, false);
                            J$.N(2783705, 'h', h, false, false, false);
                            J$.N(2783713, 'curve', curve, false, false, false);
                            J$.N(2783721, 'G', G, false, false, false);
                            var p = J$.W(2783377, 'p', J$.F(2783369, J$.R(2783353, 'fromHex', fromHex, false, true), false)(J$.T(2783361, 'FFFFFFFF00000001000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFF', 21, false)), p, false, false);
                            var a = J$.W(2783409, 'a', J$.F(2783401, J$.R(2783385, 'fromHex', fromHex, false, true), false)(J$.T(2783393, 'FFFFFFFF00000001000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFC', 21, false)), a, false, false);
                            var b = J$.W(2783441, 'b', J$.F(2783433, J$.R(2783417, 'fromHex', fromHex, false, true), false)(J$.T(2783425, '5AC635D8AA3A93E7B3EBBD55769886BC651D06B0CC53B0F63BCE3C3E27D2604B', 21, false)), b, false, false);
                            var n = J$.W(2783473, 'n', J$.F(2783465, J$.R(2783449, 'fromHex', fromHex, false, true), false)(J$.T(2783457, 'FFFFFFFF00000000FFFFFFFFFFFFFFFFBCE6FAADA7179E84F3B9CAC2FC632551', 21, false)), n, false, false);
                            var h = J$.W(2783497, 'h', J$.G(2783489, J$.R(2783481, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2783545, 'curve', J$.F(2783537, J$.R(2783505, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2783513, 'p', p, false, false), J$.R(2783521, 'a', a, false, false), J$.R(2783529, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2783593, 'G', J$.M(2783585, J$.R(2783553, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(410234, '+', J$.B(410226, '+', J$.T(2783561, '04', 21, false), J$.T(2783569, '6B17D1F2E12C4247F8BCE6E563A440F277037D812DEB33A0F4A13945D898C296', 21, false)), J$.T(2783577, '4FE342E2FE1A7F9B8EE7EB4A7C0F9E162BCE33576B315ECECBB6406837BF51F5', 21, false))), G, false, false);
                            return J$.Rt(2783649, J$.F(2783641, J$.R(2783601, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2783609, 'curve', curve, false, false), J$.R(2783617, 'G', G, false, false), J$.R(2783625, 'n', n, false, false), J$.R(2783633, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2784569, J$e);
                        } finally {
                            if (J$.Fr(2784577))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function getSECCurveByName(name) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(2784025, arguments.callee, this, arguments);
                            arguments = J$.N(2784033, 'arguments', arguments, true, false, false);
                            name = J$.N(2784041, 'name', name, true, false, false);
                            if (J$.C(133944, J$.B(410242, '==', J$.R(2783729, 'name', name, false, false), J$.T(2783737, 'secp128r1', 21, false))))
                                return J$.Rt(2783761, J$.F(2783753, J$.R(2783745, 'secp128r1', secp128r1, false, true), false)());
                            if (J$.C(133952, J$.B(410250, '==', J$.R(2783769, 'name', name, false, false), J$.T(2783777, 'secp160k1', 21, false))))
                                return J$.Rt(2783801, J$.F(2783793, J$.R(2783785, 'secp160k1', secp160k1, false, true), false)());
                            if (J$.C(133960, J$.B(410258, '==', J$.R(2783809, 'name', name, false, false), J$.T(2783817, 'secp160r1', 21, false))))
                                return J$.Rt(2783841, J$.F(2783833, J$.R(2783825, 'secp160r1', secp160r1, false, true), false)());
                            if (J$.C(133968, J$.B(410266, '==', J$.R(2783849, 'name', name, false, false), J$.T(2783857, 'secp192k1', 21, false))))
                                return J$.Rt(2783881, J$.F(2783873, J$.R(2783865, 'secp192k1', secp192k1, false, true), false)());
                            if (J$.C(133976, J$.B(410274, '==', J$.R(2783889, 'name', name, false, false), J$.T(2783897, 'secp192r1', 21, false))))
                                return J$.Rt(2783921, J$.F(2783913, J$.R(2783905, 'secp192r1', secp192r1, false, true), false)());
                            if (J$.C(133984, J$.B(410282, '==', J$.R(2783929, 'name', name, false, false), J$.T(2783937, 'secp224r1', 21, false))))
                                return J$.Rt(2783961, J$.F(2783953, J$.R(2783945, 'secp224r1', secp224r1, false, true), false)());
                            if (J$.C(133992, J$.B(410290, '==', J$.R(2783969, 'name', name, false, false), J$.T(2783977, 'secp256r1', 21, false))))
                                return J$.Rt(2784001, J$.F(2783993, J$.R(2783985, 'secp256r1', secp256r1, false, true), false)());
                            return J$.Rt(2784017, J$.T(2784009, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(2784585, J$e);
                        } finally {
                            if (J$.Fr(2784593))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(2784137, 'BigInteger', BigInteger, false, false, false);
            J$.N(2784145, 'ECCurveFp', ECCurveFp, false, false, false);
            X9ECParameters = J$.N(2784161, 'X9ECParameters', J$.T(2784153, X9ECParameters, 12, false), true, false, false);
            x9getCurve = J$.N(2784177, 'x9getCurve', J$.T(2784169, x9getCurve, 12, false), true, false, false);
            x9getG = J$.N(2784193, 'x9getG', J$.T(2784185, x9getG, 12, false), true, false, false);
            x9getN = J$.N(2784209, 'x9getN', J$.T(2784201, x9getN, 12, false), true, false, false);
            x9getH = J$.N(2784225, 'x9getH', J$.T(2784217, x9getH, 12, false), true, false, false);
            fromHex = J$.N(2784241, 'fromHex', J$.T(2784233, fromHex, 12, false), true, false, false);
            secp128r1 = J$.N(2784257, 'secp128r1', J$.T(2784249, secp128r1, 12, false), true, false, false);
            secp160k1 = J$.N(2784273, 'secp160k1', J$.T(2784265, secp160k1, 12, false), true, false, false);
            secp160r1 = J$.N(2784289, 'secp160r1', J$.T(2784281, secp160r1, 12, false), true, false, false);
            secp192k1 = J$.N(2784305, 'secp192k1', J$.T(2784297, secp192k1, 12, false), true, false, false);
            secp192r1 = J$.N(2784321, 'secp192r1', J$.T(2784313, secp192r1, 12, false), true, false, false);
            secp224r1 = J$.N(2784337, 'secp224r1', J$.T(2784329, secp224r1, 12, false), true, false, false);
            secp256r1 = J$.N(2784353, 'secp256r1', J$.T(2784345, secp256r1, 12, false), true, false, false);
            getSECCurveByName = J$.N(2784369, 'getSECCurveByName', J$.T(2784361, getSECCurveByName, 12, false), true, false, false);
            J$.T(2780529, 'use strict', 21, false);
            var BigInteger = J$.W(2780569, 'BigInteger', J$.G(2780561, J$.F(2780553, J$.I(typeof require === 'undefined' ? require = J$.R(2780537, 'require', undefined, true, true) : require = J$.R(2780537, 'require', require, true, true)), false)(J$.T(2780545, 'jsbn', 21, false)), 'BigInteger'), BigInteger, false, true);
            var ECCurveFp = J$.W(2780609, 'ECCurveFp', J$.G(2780601, J$.F(2780593, J$.I(typeof require === 'undefined' ? require = J$.R(2780577, 'require', undefined, true, true) : require = J$.R(2780577, 'require', require, true, true)), false)(J$.T(2780585, './ec.js', 21, false)), 'ECCurveFp'), ECCurveFp, false, true);
            J$.P(2780945, J$.G(2780929, J$.R(2780921, 'X9ECParameters', X9ECParameters, false, true), 'prototype'), 'getCurve', J$.R(2780937, 'x9getCurve', x9getCurve, false, true));
            J$.P(2780977, J$.G(2780961, J$.R(2780953, 'X9ECParameters', X9ECParameters, false, true), 'prototype'), 'getG', J$.R(2780969, 'x9getG', x9getG, false, true));
            J$.P(2781009, J$.G(2780993, J$.R(2780985, 'X9ECParameters', X9ECParameters, false, true), 'prototype'), 'getN', J$.R(2781001, 'x9getN', x9getN, false, true));
            J$.P(2781041, J$.G(2781025, J$.R(2781017, 'X9ECParameters', X9ECParameters, false, true), 'prototype'), 'getH', J$.R(2781033, 'x9getH', x9getH, false, true));
            J$.P(2784121, J$.I(typeof module === 'undefined' ? module = J$.R(2784049, 'module', undefined, true, true) : module = J$.R(2784049, 'module', module, true, true)), 'exports', J$.T(2784113, {
                'secp128r1': J$.R(2784057, 'secp128r1', secp128r1, false, true),
                'secp160k1': J$.R(2784065, 'secp160k1', secp160k1, false, true),
                'secp160r1': J$.R(2784073, 'secp160r1', secp160r1, false, true),
                'secp192k1': J$.R(2784081, 'secp192k1', secp192k1, false, true),
                'secp192r1': J$.R(2784089, 'secp192r1', secp192r1, false, true),
                'secp224r1': J$.R(2784097, 'secp224r1', secp224r1, false, true),
                'secp256r1': J$.R(2784105, 'secp256r1', secp256r1, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(2784601, J$e);
        } finally {
            if (J$.Sr(2784609))
                continue jalangiLabel14;
            else
                break jalangiLabel14;
        }
    }
// JALANGI DO NOT INSTRUMENT
