// uri-js/rollup.config.js
J$.noInstrEval = false;
jalangiLabel1:
    while (true) {
        try {
            J$.Se(1703041, '/mnt/data/fxiao/HiPar/outputs/target_cache/TestAjv/node_modules/uri-js/rollup.config.js');
            function _interopRequireDefault(obj) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(1702665, arguments.callee, this, arguments);
                            arguments = J$.N(1702673, 'arguments', arguments, true, false, false);
                            obj = J$.N(1702681, 'obj', obj, true, false, false);
                            return J$.Rt(1702657, J$.C(101856, J$.C(101848, J$.R(1702609, 'obj', obj, false, false)) ? J$.G(1702625, J$.R(1702617, 'obj', obj, false, false), '__esModule') : J$._()) ? J$.R(1702633, 'obj', obj, false, false) : J$.T(1702649, { 'default': J$.R(1702641, 'obj', obj, false, false) }, 11, false));
                        } catch (J$e) {
                            J$.Ex(1703097, J$e);
                        } finally {
                            if (J$.Fr(1703105))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(1703049, '_rollupPluginNodeResolve', _rollupPluginNodeResolve, false, false, false);
            J$.N(1703057, '_rollupPluginBabel', _rollupPluginBabel, false, false, false);
            _interopRequireDefault = J$.N(1703073, '_interopRequireDefault', J$.T(1703065, _interopRequireDefault, 12, false), true, false, false);
            J$.N(1703081, 'packageJson', packageJson, false, false, false);
            J$.N(1703089, '_default', _default, false, false, false);
            J$.T(1702425, 'use strict', 21, false);
            J$.M(1702473, J$.I(typeof Object === 'undefined' ? Object = J$.R(1702433, 'Object', undefined, true, true) : Object = J$.R(1702433, 'Object', Object, true, true)), 'defineProperty', false)(J$.I(typeof exports === 'undefined' ? exports = J$.R(1702441, 'exports', undefined, true, true) : exports = J$.R(1702441, 'exports', exports, true, true)), J$.T(1702449, '__esModule', 21, false), J$.T(1702465, { value: J$.T(1702457, true, 23, false) }, 11, false));
            J$.P(1702505, J$.I(typeof exports === 'undefined' ? exports = J$.R(1702481, 'exports', undefined, true, true) : exports = J$.R(1702481, 'exports', exports, true, true)), J$.T(1702489, 'default', 21, false), void J$.T(1702497, 0, 22, false));
            var _rollupPluginNodeResolve = J$.W(1702553, '_rollupPluginNodeResolve', J$.F(1702545, J$.R(1702513, '_interopRequireDefault', _interopRequireDefault, false, true), false)(J$.F(1702537, J$.I(typeof require === 'undefined' ? require = J$.R(1702521, 'require', undefined, true, true) : require = J$.R(1702521, 'require', require, true, true)), false)(J$.T(1702529, 'rollup-plugin-node-resolve', 21, false))), _rollupPluginNodeResolve, false, true);
            var _rollupPluginBabel = J$.W(1702601, '_rollupPluginBabel', J$.F(1702593, J$.R(1702561, '_interopRequireDefault', _interopRequireDefault, false, true), false)(J$.F(1702585, J$.I(typeof require === 'undefined' ? require = J$.R(1702569, 'require', undefined, true, true) : require = J$.R(1702569, 'require', require, true, true)), false)(J$.T(1702577, 'rollup-plugin-babel', 21, false))), _rollupPluginBabel, false, true);
            var packageJson = J$.W(1702713, 'packageJson', J$.F(1702705, J$.I(typeof require === 'undefined' ? require = J$.R(1702689, 'require', undefined, true, true) : require = J$.R(1702689, 'require', require, true, true)), false)(J$.T(1702697, './package.json', 21, false)), packageJson, false, true);
            var _default = J$.W(1703001, '_default', J$.T(1702993, {
                entry: J$.T(1702721, 'dist/esnext/index.js', 21, false),
                format: J$.T(1702729, 'umd', 21, false),
                moduleName: J$.T(1702737, 'URI', 21, false),
                plugins: J$.T(1702937, [
                    J$.F(1702809, (J$.T(1702745, 0, 22, false), J$.G(1702769, J$.R(1702753, '_rollupPluginNodeResolve', _rollupPluginNodeResolve, false, true), J$.T(1702761, 'default', 21, false))), false)(J$.T(1702801, {
                        module: J$.T(1702777, true, 23, false),
                        jsnext: J$.T(1702785, true, 23, false),
                        preferBuiltins: J$.T(1702793, false, 23, false)
                    }, 11, false)),
                    J$.F(1702929, (J$.T(1702817, 0, 22, false), J$.G(1702841, J$.R(1702825, '_rollupPluginBabel', _rollupPluginBabel, false, true), J$.T(1702833, 'default', 21, false))), false)(J$.T(1702921, {
                        'presets': J$.T(1702889, [J$.T(1702881, [
                                J$.T(1702849, 'latest', 21, false),
                                J$.T(1702873, { 'es2015': J$.T(1702865, { 'modules': J$.T(1702857, false, 23, false) }, 11, false) }, 11, false)
                            ], 10, false)], 10, false),
                        'plugins': J$.T(1702905, [J$.T(1702897, 'external-helpers', 21, false)], 10, false),
                        'externalHelpers': J$.T(1702913, false, 23, false)
                    }, 11, false))
                ], 10, false),
                dest: J$.T(1702945, 'dist/es5/uri.all.js', 21, false),
                sourceMap: J$.T(1702953, true, 23, false),
                banner: J$.B(235434, '+', J$.B(235426, '+', J$.T(1702961, '/** @license URI.js v', 21, false), J$.G(1702977, J$.R(1702969, 'packageJson', packageJson, false, true), 'version')), J$.T(1702985, ' (c) 2011 Gary Court. License: http://github.com/garycourt/uri-js */', 21, false))
            }, 11, false), _default, false, true);
            J$.P(1703033, J$.I(typeof exports === 'undefined' ? exports = J$.R(1703009, 'exports', undefined, true, true) : exports = J$.R(1703009, 'exports', exports, true, true)), J$.T(1703017, 'default', 21, false), J$.R(1703025, '_default', _default, false, true));
        } catch (J$e) {
            J$.Ex(1703113, J$e);
        } finally {
            if (J$.Sr(1703121))
                continue jalangiLabel1;
            else
                break jalangiLabel1;
        }
    }
// JALANGI DO NOT INSTRUMENT
