// http-signature/lib/utils.js
J$.noInstrEval = false;
jalangiLabel9:
    while (true) {
        try {
            J$.Se(928561, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCoinBasePro/node_modules/http-signature/lib/utils.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(927105, arguments.callee, this, arguments);
                            arguments = J$.N(927113, 'arguments', arguments, true, false, false);
                            obj = J$.N(927121, 'obj', obj, true, false, false);
                            if (J$.C(55704, J$.C(55664, J$.B(148034, '===', J$.U(148026, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(926841, 'Symbol', undefined, true, true) : Symbol = J$.R(926841, 'Symbol', Symbol, true, true))), J$.T(926849, 'function', 21, false))) ? J$.B(148050, '===', J$.U(148042, 'typeof', J$.G(926865, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(926857, 'Symbol', undefined, true, true) : Symbol = J$.R(926857, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(926873, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(926929, '_typeof', J$.T(926921, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(926897, arguments.callee, this, arguments);
                                                arguments = J$.N(926905, 'arguments', arguments, true, false, false);
                                                obj = J$.N(926913, 'obj', obj, true, false, false);
                                                return J$.Rt(926889, J$.U(148058, 'typeof', J$.R(926881, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(928673, J$e);
                                            } finally {
                                                if (J$.Fr(928681))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(927065, '_typeof', J$.T(927057, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(927033, arguments.callee, this, arguments);
                                                arguments = J$.N(927041, 'arguments', arguments, true, false, false);
                                                obj = J$.N(927049, 'obj', obj, true, false, false);
                                                return J$.Rt(927025, J$.C(55696, J$.C(55688, J$.C(55680, J$.C(55672, J$.R(926937, 'obj', obj, false, false)) ? J$.B(148074, '===', J$.U(148066, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(926945, 'Symbol', undefined, true, true) : Symbol = J$.R(926945, 'Symbol', Symbol, true, true))), J$.T(926953, 'function', 21, false)) : J$._()) ? J$.B(148082, '===', J$.G(926969, J$.R(926961, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(926977, 'Symbol', undefined, true, true) : Symbol = J$.R(926977, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(148090, '!==', J$.R(926985, 'obj', obj, false, false), J$.G(927001, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(926993, 'Symbol', undefined, true, true) : Symbol = J$.R(926993, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(927009, 'symbol', 21, false) : J$.U(148098, 'typeof', J$.R(927017, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(928689, J$e);
                                            } finally {
                                                if (J$.Fr(928697))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(927097, J$.F(927089, J$.R(927073, '_typeof', _typeof, false, true), false)(J$.R(927081, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(928705, J$e);
                        } finally {
                            if (J$.Fr(928713))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function HttpSignatureError(message, caller) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(927417, arguments.callee, this, arguments);
                            arguments = J$.N(927425, 'arguments', arguments, true, false, false);
                            message = J$.N(927433, 'message', message, true, false, false);
                            caller = J$.N(927441, 'caller', caller, true, false, false);
                            if (J$.C(55720, J$.G(927313, J$.I(typeof Error === 'undefined' ? Error = J$.R(927305, 'Error', undefined, true, true) : Error = J$.R(927305, 'Error', Error, true, true)), 'captureStackTrace')))
                                J$.M(927353, J$.I(typeof Error === 'undefined' ? Error = J$.R(927321, 'Error', undefined, true, true) : Error = J$.R(927321, 'Error', Error, true, true)), 'captureStackTrace', false)(J$.R(927329, 'this', this, false, false), J$.C(55712, J$.R(927337, 'caller', caller, false, false)) ? J$._() : J$.R(927345, 'HttpSignatureError', HttpSignatureError, false, true));
                            J$.P(927377, J$.R(927361, 'this', this, false, false), 'message', J$.R(927369, 'message', message, false, false));
                            J$.P(927409, J$.R(927385, 'this', this, false, false), 'name', J$.G(927401, J$.R(927393, 'caller', caller, false, false), 'name'));
                        } catch (J$e) {
                            J$.Ex(928721, J$e);
                        } finally {
                            if (J$.Fr(928729))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function InvalidAlgorithmError(message) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(927521, arguments.callee, this, arguments);
                            arguments = J$.N(927529, 'arguments', arguments, true, false, false);
                            message = J$.N(927537, 'message', message, true, false, false);
                            J$.M(927513, J$.R(927481, 'HttpSignatureError', HttpSignatureError, false, true), 'call', false)(J$.R(927489, 'this', this, false, false), J$.R(927497, 'message', message, false, false), J$.R(927505, 'InvalidAlgorithmError', InvalidAlgorithmError, false, true));
                        } catch (J$e) {
                            J$.Ex(928737, J$e);
                        } finally {
                            if (J$.Fr(928745))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function validateAlgorithm(algorithm) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(927961, arguments.callee, this, arguments);
                            arguments = J$.N(927969, 'arguments', arguments, true, false, false);
                            algorithm = J$.N(927977, 'algorithm', algorithm, true, false, false);
                            J$.N(927985, 'alg', alg, false, false, false);
                            var alg = J$.W(927609, 'alg', J$.M(927601, J$.M(927585, J$.R(927577, 'algorithm', algorithm, false, false), 'toLowerCase', false)(), 'split', false)(J$.T(927593, '-', 21, false)), alg, false, false);
                            if (J$.C(55728, J$.B(148106, '!==', J$.G(927625, J$.R(927617, 'alg', alg, false, false), 'length'), J$.T(927633, 2, 22, false)))) {
                                throw J$.F(927697, J$.R(927641, 'InvalidAlgorithmError', InvalidAlgorithmError, false, true), true)(J$.B(148122, '+', J$.B(148114, '+', J$.M(927673, J$.G(927665, J$.R(927649, 'alg', alg, false, false), J$.T(927657, 0, 22, false)), 'toUpperCase', false)(), J$.T(927681, ' is not a ', 21, false)), J$.T(927689, 'valid algorithm', 21, false)));
                            }
                            if (J$.C(55744, J$.C(55736, J$.B(148130, '!==', J$.G(927721, J$.R(927705, 'alg', alg, false, false), J$.T(927713, 0, 22, false)), J$.T(927729, 'hmac', 21, false))) ? J$.U(148138, '!', J$.G(927769, J$.R(927737, 'PK_ALGOS', PK_ALGOS, false, true), J$.G(927761, J$.R(927745, 'alg', alg, false, false), J$.T(927753, 0, 22, false)))) : J$._())) {
                                throw J$.F(927833, J$.R(927777, 'InvalidAlgorithmError', InvalidAlgorithmError, false, true), true)(J$.B(148154, '+', J$.B(148146, '+', J$.M(927809, J$.G(927801, J$.R(927785, 'alg', alg, false, false), J$.T(927793, 0, 22, false)), 'toUpperCase', false)(), J$.T(927817, ' type keys ', 21, false)), J$.T(927825, 'are not supported', 21, false)));
                            }
                            if (J$.C(55752, J$.U(148162, '!', J$.G(927873, J$.R(927841, 'HASH_ALGOS', HASH_ALGOS, false, true), J$.G(927865, J$.R(927849, 'alg', alg, false, false), J$.T(927857, 1, 22, false)))))) {
                                throw J$.F(927937, J$.R(927881, 'InvalidAlgorithmError', InvalidAlgorithmError, false, true), true)(J$.B(148178, '+', J$.B(148170, '+', J$.M(927913, J$.G(927905, J$.R(927889, 'alg', alg, false, false), J$.T(927897, 1, 22, false)), 'toUpperCase', false)(), J$.T(927921, ' is not a ', 21, false)), J$.T(927929, 'supported hash algorithm', 21, false)));
                            }
                            return J$.Rt(927953, J$.R(927945, 'alg', alg, false, false));
                        } catch (J$e) {
                            J$.Ex(928753, J$e);
                        } finally {
                            if (J$.Fr(928761))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(928577, '_typeof', J$.T(928569, _typeof, 12, false), true, false, false);
            J$.N(928585, 'assert', assert, false, false, false);
            J$.N(928593, 'sshpk', sshpk, false, false, false);
            J$.N(928601, 'util', util, false, false, false);
            J$.N(928609, 'HASH_ALGOS', HASH_ALGOS, false, false, false);
            J$.N(928617, 'PK_ALGOS', PK_ALGOS, false, false, false);
            HttpSignatureError = J$.N(928633, 'HttpSignatureError', J$.T(928625, HttpSignatureError, 12, false), true, false, false);
            InvalidAlgorithmError = J$.N(928649, 'InvalidAlgorithmError', J$.T(928641, InvalidAlgorithmError, 12, false), true, false, false);
            validateAlgorithm = J$.N(928665, 'validateAlgorithm', J$.T(928657, validateAlgorithm, 12, false), true, false, false);
            J$.T(926545, 'use strict', 21, false);
            J$.F(926569, J$.I(typeof require === 'undefined' ? require = J$.R(926553, 'require', undefined, true, true) : require = J$.R(926553, 'require', require, true, true)), false)(J$.T(926561, 'core-js/modules/es.symbol', 21, false));
            J$.F(926593, J$.I(typeof require === 'undefined' ? require = J$.R(926577, 'require', undefined, true, true) : require = J$.R(926577, 'require', require, true, true)), false)(J$.T(926585, 'core-js/modules/es.symbol.description', 21, false));
            J$.F(926617, J$.I(typeof require === 'undefined' ? require = J$.R(926601, 'require', undefined, true, true) : require = J$.R(926601, 'require', require, true, true)), false)(J$.T(926609, 'core-js/modules/es.symbol.iterator', 21, false));
            J$.F(926641, J$.I(typeof require === 'undefined' ? require = J$.R(926625, 'require', undefined, true, true) : require = J$.R(926625, 'require', require, true, true)), false)(J$.T(926633, 'core-js/modules/es.array.iterator', 21, false));
            J$.F(926665, J$.I(typeof require === 'undefined' ? require = J$.R(926649, 'require', undefined, true, true) : require = J$.R(926649, 'require', require, true, true)), false)(J$.T(926657, 'core-js/modules/es.date.to-string', 21, false));
            J$.F(926689, J$.I(typeof require === 'undefined' ? require = J$.R(926673, 'require', undefined, true, true) : require = J$.R(926673, 'require', require, true, true)), false)(J$.T(926681, 'core-js/modules/es.function.name', 21, false));
            J$.F(926713, J$.I(typeof require === 'undefined' ? require = J$.R(926697, 'require', undefined, true, true) : require = J$.R(926697, 'require', require, true, true)), false)(J$.T(926705, 'core-js/modules/es.object.to-string', 21, false));
            J$.F(926737, J$.I(typeof require === 'undefined' ? require = J$.R(926721, 'require', undefined, true, true) : require = J$.R(926721, 'require', require, true, true)), false)(J$.T(926729, 'core-js/modules/es.regexp.exec', 21, false));
            J$.F(926761, J$.I(typeof require === 'undefined' ? require = J$.R(926745, 'require', undefined, true, true) : require = J$.R(926745, 'require', require, true, true)), false)(J$.T(926753, 'core-js/modules/es.regexp.to-string', 21, false));
            J$.F(926785, J$.I(typeof require === 'undefined' ? require = J$.R(926769, 'require', undefined, true, true) : require = J$.R(926769, 'require', require, true, true)), false)(J$.T(926777, 'core-js/modules/es.string.iterator', 21, false));
            J$.F(926809, J$.I(typeof require === 'undefined' ? require = J$.R(926793, 'require', undefined, true, true) : require = J$.R(926793, 'require', require, true, true)), false)(J$.T(926801, 'core-js/modules/es.string.split', 21, false));
            J$.F(926833, J$.I(typeof require === 'undefined' ? require = J$.R(926817, 'require', undefined, true, true) : require = J$.R(926817, 'require', require, true, true)), false)(J$.T(926825, 'core-js/modules/web.dom-collections.iterator', 21, false));
            var assert = J$.W(927153, 'assert', J$.F(927145, J$.I(typeof require === 'undefined' ? require = J$.R(927129, 'require', undefined, true, true) : require = J$.R(927129, 'require', require, true, true)), false)(J$.T(927137, 'assert-plus', 21, false)), assert, false, true);
            var sshpk = J$.W(927185, 'sshpk', J$.F(927177, J$.I(typeof require === 'undefined' ? require = J$.R(927161, 'require', undefined, true, true) : require = J$.R(927161, 'require', require, true, true)), false)(J$.T(927169, 'sshpk', 21, false)), sshpk, false, true);
            var util = J$.W(927217, 'util', J$.F(927209, J$.I(typeof require === 'undefined' ? require = J$.R(927193, 'require', undefined, true, true) : require = J$.R(927193, 'require', require, true, true)), false)(J$.T(927201, 'util', 21, false)), util, false, true);
            var HASH_ALGOS = J$.W(927257, 'HASH_ALGOS', J$.T(927249, {
                'sha1': J$.T(927225, true, 23, false),
                'sha256': J$.T(927233, true, 23, false),
                'sha512': J$.T(927241, true, 23, false)
            }, 11, false), HASH_ALGOS, false, true);
            var PK_ALGOS = J$.W(927297, 'PK_ALGOS', J$.T(927289, {
                'rsa': J$.T(927265, true, 23, false),
                'dsa': J$.T(927273, true, 23, false),
                'ecdsa': J$.T(927281, true, 23, false)
            }, 11, false), PK_ALGOS, false, true);
            J$.M(927473, J$.R(927449, 'util', util, false, true), 'inherits', false)(J$.R(927457, 'HttpSignatureError', HttpSignatureError, false, true), J$.I(typeof Error === 'undefined' ? Error = J$.R(927465, 'Error', undefined, true, true) : Error = J$.R(927465, 'Error', Error, true, true)));
            J$.M(927569, J$.R(927545, 'util', util, false, true), 'inherits', false)(J$.R(927553, 'InvalidAlgorithmError', InvalidAlgorithmError, false, true), J$.R(927561, 'HttpSignatureError', HttpSignatureError, false, true));
            J$.P(928553, J$.I(typeof module === 'undefined' ? module = J$.R(927993, 'module', undefined, true, true) : module = J$.R(927993, 'module', module, true, true)), 'exports', J$.T(928545, {
                HASH_ALGOS: J$.R(928001, 'HASH_ALGOS', HASH_ALGOS, false, true),
                PK_ALGOS: J$.R(928009, 'PK_ALGOS', PK_ALGOS, false, true),
                HttpSignatureError: J$.R(928017, 'HttpSignatureError', HttpSignatureError, false, true),
                InvalidAlgorithmError: J$.R(928025, 'InvalidAlgorithmError', InvalidAlgorithmError, false, true),
                validateAlgorithm: J$.R(928033, 'validateAlgorithm', validateAlgorithm, false, true),
                sshKeyToPEM: J$.T(928177, function sshKeyToPEM(key) {
                    jalangiLabel6:
                        while (true) {
                            try {
                                J$.Fe(928145, arguments.callee, this, arguments);
                                arguments = J$.N(928153, 'arguments', arguments, true, false, false);
                                key = J$.N(928161, 'key', key, true, false, false);
                                J$.N(928169, 'k', k, false, false, false);
                                J$.M(928065, J$.R(928041, 'assert', assert, false, true), 'string', false)(J$.R(928049, 'key', key, false, false), J$.T(928057, 'ssh_key', 21, false));
                                var k = J$.W(928105, 'k', J$.M(928097, J$.R(928073, 'sshpk', sshpk, false, true), 'parseKey', false)(J$.R(928081, 'key', key, false, false), J$.T(928089, 'ssh', 21, false)), k, false, false);
                                return J$.Rt(928137, J$.M(928129, J$.R(928113, 'k', k, false, false), 'toString', false)(J$.T(928121, 'pem', 21, false)));
                            } catch (J$e) {
                                J$.Ex(928769, J$e);
                            } finally {
                                if (J$.Fr(928777))
                                    continue jalangiLabel6;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                fingerprint: J$.T(928337, function fingerprint(key) {
                    jalangiLabel7:
                        while (true) {
                            try {
                                J$.Fe(928305, arguments.callee, this, arguments);
                                arguments = J$.N(928313, 'arguments', arguments, true, false, false);
                                key = J$.N(928321, 'key', key, true, false, false);
                                J$.N(928329, 'k', k, false, false, false);
                                J$.M(928209, J$.R(928185, 'assert', assert, false, true), 'string', false)(J$.R(928193, 'key', key, false, false), J$.T(928201, 'ssh_key', 21, false));
                                var k = J$.W(928249, 'k', J$.M(928241, J$.R(928217, 'sshpk', sshpk, false, true), 'parseKey', false)(J$.R(928225, 'key', key, false, false), J$.T(928233, 'ssh', 21, false)), k, false, false);
                                return J$.Rt(928297, J$.M(928289, J$.M(928273, J$.R(928257, 'k', k, false, false), 'fingerprint', false)(J$.T(928265, 'md5', 21, false)), 'toString', false)(J$.T(928281, 'hex', 21, false)));
                            } catch (J$e) {
                                J$.Ex(928785, J$e);
                            } finally {
                                if (J$.Fr(928793))
                                    continue jalangiLabel7;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                pemToRsaSSHKey: J$.T(928537, function pemToRsaSSHKey(pem, comment) {
                    jalangiLabel8:
                        while (true) {
                            try {
                                J$.Fe(928497, arguments.callee, this, arguments);
                                arguments = J$.N(928505, 'arguments', arguments, true, false, false);
                                pem = J$.N(928513, 'pem', pem, true, false, false);
                                comment = J$.N(928521, 'comment', comment, true, false, false);
                                J$.N(928529, 'k', k, false, false, false);
                                J$.M(928393, J$.R(928345, 'assert', assert, false, true), 'equal', false)(J$.T(928353, 'string', 21, false), J$.F(928377, J$.R(928361, '_typeof', _typeof, false, true), false)(J$.R(928369, 'pem', pem, false, false)), J$.T(928385, 'typeof pem', 21, false));
                                var k = J$.W(928433, 'k', J$.M(928425, J$.R(928401, 'sshpk', sshpk, false, true), 'parseKey', false)(J$.R(928409, 'pem', pem, false, false), J$.T(928417, 'pem', 21, false)), k, false, false);
                                J$.P(928457, J$.R(928441, 'k', k, false, false), 'comment', J$.R(928449, 'comment', comment, false, false));
                                return J$.Rt(928489, J$.M(928481, J$.R(928465, 'k', k, false, false), 'toString', false)(J$.T(928473, 'ssh', 21, false)));
                            } catch (J$e) {
                                J$.Ex(928801, J$e);
                            } finally {
                                if (J$.Fr(928809))
                                    continue jalangiLabel8;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(928817, J$e);
        } finally {
            if (J$.Sr(928825))
                continue jalangiLabel9;
            else
                break jalangiLabel9;
        }
    }
// JALANGI DO NOT INSTRUMENT
