// indicative-rules/build/src/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var date_fns_1 = require("date-fns");

var months = {
  years: function years(unit) {
    return unit * 12;
  },
  quarters: function quarters(unit) {
    return unit * 3;
  },
  months: function months(unit) {
    return unit;
  }
};
var days = {
  weeks: function weeks(unit) {
    return unit * 7;
  },
  days: function days(unit) {
    return unit;
  }
};
var milliseconds = {
  hours: function hours(unit) {
    return unit * 3600000;
  },
  minutes: function minutes(unit) {
    return unit * 60000;
  },
  seconds: function seconds(unit) {
    return unit * 1000;
  },
  milliseconds: function milliseconds(unit) {
    return unit;
  }
};

exports.calcUnits = function (diffUnit, key, operator) {
  if (months[key]) {
    return date_fns_1.addMonths(new Date(), operator === '-' ? -months[key](diffUnit) : months[key](diffUnit));
  }

  if (days[key]) {
    return date_fns_1.addDays(new Date(), operator === '-' ? -days[key](diffUnit) : days[key](diffUnit));
  }

  if (milliseconds[key]) {
    return date_fns_1.addMilliseconds(new Date(), operator === '-' ? -milliseconds[key](diffUnit) : milliseconds[key](diffUnit));
  }
};

exports.dateOffsetKeys = ['years', 'quarters', 'months', 'weeks', 'days', 'hours', 'minutes', 'seconds', 'milliseconds'];
