// datalize/lib/result.js
J$.noInstrEval = false;
jalangiLabel11:
    while (true) {
        try {
            J$.Se(4857649, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestDatalize/node_modules/datalize/lib/result.js');
            function _classCallCheck(instance, Constructor) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(4855937, arguments.callee, this, arguments);
                            arguments = J$.N(4855945, 'arguments', arguments, true, false, false);
                            instance = J$.N(4855953, 'instance', instance, true, false, false);
                            Constructor = J$.N(4855961, 'Constructor', Constructor, true, false, false);
                            if (J$.C(258320, J$.U(489090, '!', J$.B(489082, 'instanceof', J$.R(4855897, 'instance', instance, false, false), J$.R(4855905, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(4855929, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(4855913, 'TypeError', undefined, true, true) : TypeError = J$.R(4855913, 'TypeError', TypeError, true, true)), true)(J$.T(4855921, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(4857745, J$e);
                        } finally {
                            if (J$.Fr(4857753))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(4856209, arguments.callee, this, arguments);
                            arguments = J$.N(4856217, 'arguments', arguments, true, false, false);
                            target = J$.N(4856225, 'target', target, true, false, false);
                            props = J$.N(4856233, 'props', props, true, false, false);
                            J$.N(4856241, 'i', i, false, false, false);
                            J$.N(4856249, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(4855977, 'i', J$.T(4855969, 0, 22, false), i, false, false); J$.C(258344, J$.B(489098, '<', J$.R(4855985, 'i', i, false, false), J$.G(4856001, J$.R(4855993, 'props', props, false, false), 'length'))); J$.B(489122, '-', i = J$.W(4856017, 'i', J$.B(489114, '+', J$.U(489106, '+', J$.R(4856009, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(4856049, 'descriptor', J$.G(4856041, J$.R(4856025, 'props', props, false, false), J$.R(4856033, 'i', i, false, false)), descriptor, false, false);
                                J$.P(4856089, J$.R(4856057, 'descriptor', descriptor, false, false), 'enumerable', J$.C(258328, J$.G(4856073, J$.R(4856065, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(4856081, false, 23, false));
                                J$.P(4856113, J$.R(4856097, 'descriptor', descriptor, false, false), 'configurable', J$.T(4856105, true, 23, false));
                                if (J$.C(258336, J$.B(489130, 'in', J$.T(4856121, 'value', 21, false), J$.R(4856129, 'descriptor', descriptor, false, false))))
                                    J$.P(4856153, J$.R(4856137, 'descriptor', descriptor, false, false), 'writable', J$.T(4856145, true, 23, false));
                                J$.M(4856201, J$.I(typeof Object === 'undefined' ? Object = J$.R(4856161, 'Object', undefined, true, true) : Object = J$.R(4856161, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(4856169, 'target', target, false, false), J$.G(4856185, J$.R(4856177, 'descriptor', descriptor, false, false), 'key'), J$.R(4856193, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(4857761, J$e);
                        } finally {
                            if (J$.Fr(4857769))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(4856361, arguments.callee, this, arguments);
                            arguments = J$.N(4856369, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(4856377, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(4856385, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(4856393, 'staticProps', staticProps, true, false, false);
                            if (J$.C(258352, J$.R(4856257, 'protoProps', protoProps, false, false)))
                                J$.F(4856297, J$.R(4856265, '_defineProperties', _defineProperties, false, true), false)(J$.G(4856281, J$.R(4856273, 'Constructor', Constructor, false, false), 'prototype'), J$.R(4856289, 'protoProps', protoProps, false, false));
                            if (J$.C(258360, J$.R(4856305, 'staticProps', staticProps, false, false)))
                                J$.F(4856337, J$.R(4856313, '_defineProperties', _defineProperties, false, true), false)(J$.R(4856321, 'Constructor', Constructor, false, false), J$.R(4856329, 'staticProps', staticProps, false, false));
                            return J$.Rt(4856353, J$.R(4856345, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(4857777, J$e);
                        } finally {
                            if (J$.Fr(4857785))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _classCallCheck = J$.N(4857665, '_classCallCheck', J$.T(4857657, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(4857681, '_defineProperties', J$.T(4857673, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(4857697, '_createClass', J$.T(4857689, _createClass, 12, false), true, false, false);
            J$.N(4857705, 'get', get, false, false, false);
            J$.N(4857713, 'set', set, false, false, false);
            J$.N(4857721, 'unset', unset, false, false, false);
            J$.N(4857729, 'Field', Field, false, false, false);
            J$.N(4857737, 'Result', Result, false, false, false);
            J$.T(4855625, 'use strict', 21, false);
            J$.F(4855649, J$.I(typeof require === 'undefined' ? require = J$.R(4855633, 'require', undefined, true, true) : require = J$.R(4855633, 'require', require, true, true)), false)(J$.T(4855641, 'core-js/modules/es.object.define-properties', 21, false));
            J$.F(4855673, J$.I(typeof require === 'undefined' ? require = J$.R(4855657, 'require', undefined, true, true) : require = J$.R(4855657, 'require', require, true, true)), false)(J$.T(4855665, 'core-js/modules/es.object.define-property', 21, false));
            J$.F(4855697, J$.I(typeof require === 'undefined' ? require = J$.R(4855681, 'require', undefined, true, true) : require = J$.R(4855681, 'require', require, true, true)), false)(J$.T(4855689, 'core-js/modules/es.object.keys', 21, false));
            J$.F(4855721, J$.I(typeof require === 'undefined' ? require = J$.R(4855705, 'require', undefined, true, true) : require = J$.R(4855705, 'require', require, true, true)), false)(J$.T(4855713, 'core-js/modules/es.object.to-string', 21, false));
            J$.F(4855745, J$.I(typeof require === 'undefined' ? require = J$.R(4855729, 'require', undefined, true, true) : require = J$.R(4855729, 'require', require, true, true)), false)(J$.T(4855737, 'core-js/modules/es.promise', 21, false));
            J$.F(4855769, J$.I(typeof require === 'undefined' ? require = J$.R(4855753, 'require', undefined, true, true) : require = J$.R(4855753, 'require', require, true, true)), false)(J$.T(4855761, 'core-js/modules/es.object.define-properties', 21, false));
            J$.F(4855793, J$.I(typeof require === 'undefined' ? require = J$.R(4855777, 'require', undefined, true, true) : require = J$.R(4855777, 'require', require, true, true)), false)(J$.T(4855785, 'core-js/modules/es.object.define-property', 21, false));
            J$.F(4855817, J$.I(typeof require === 'undefined' ? require = J$.R(4855801, 'require', undefined, true, true) : require = J$.R(4855801, 'require', require, true, true)), false)(J$.T(4855809, 'core-js/modules/es.object.keys', 21, false));
            J$.F(4855841, J$.I(typeof require === 'undefined' ? require = J$.R(4855825, 'require', undefined, true, true) : require = J$.R(4855825, 'require', require, true, true)), false)(J$.T(4855833, 'core-js/modules/es.object.to-string', 21, false));
            J$.F(4855865, J$.I(typeof require === 'undefined' ? require = J$.R(4855849, 'require', undefined, true, true) : require = J$.R(4855849, 'require', require, true, true)), false)(J$.T(4855857, 'core-js/modules/es.promise', 21, false));
            J$.F(4855889, J$.I(typeof require === 'undefined' ? require = J$.R(4855873, 'require', undefined, true, true) : require = J$.R(4855873, 'require', require, true, true)), false)(J$.T(4855881, 'regenerator-runtime/runtime', 21, false));
            var get = J$.W(4856425, 'get', J$.F(4856417, J$.I(typeof require === 'undefined' ? require = J$.R(4856401, 'require', undefined, true, true) : require = J$.R(4856401, 'require', require, true, true)), false)(J$.T(4856409, 'lodash/get', 21, false)), get, false, true);
            var set = J$.W(4856457, 'set', J$.F(4856449, J$.I(typeof require === 'undefined' ? require = J$.R(4856433, 'require', undefined, true, true) : require = J$.R(4856433, 'require', require, true, true)), false)(J$.T(4856441, 'lodash/set', 21, false)), set, false, true);
            var unset = J$.W(4856489, 'unset', J$.F(4856481, J$.I(typeof require === 'undefined' ? require = J$.R(4856465, 'require', undefined, true, true) : require = J$.R(4856465, 'require', require, true, true)), false)(J$.T(4856473, 'lodash/unset', 21, false)), unset, false, true);
            var Field = J$.W(4856521, 'Field', J$.F(4856513, J$.I(typeof require === 'undefined' ? require = J$.R(4856497, 'require', undefined, true, true) : require = J$.R(4856497, 'require', require, true, true)), false)(J$.T(4856505, './field', 21, false)), Field, false, true);
            var Result = J$.W(4857617, 'Result', J$.F(4857609, J$.T(4857601, function () {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(4857569, arguments.callee, this, arguments);
                            function Result() {
                                jalangiLabel3:
                                    while (true) {
                                        try {
                                            J$.Fe(4856617, arguments.callee, this, arguments);
                                            arguments = J$.N(4856625, 'arguments', arguments, true, false, false);
                                            J$.F(4856553, J$.R(4856529, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(4856537, 'this', this, false, false), J$.R(4856545, 'Result', Result, false, false));
                                            J$.M(4856609, J$.I(typeof Object === 'undefined' ? Object = J$.R(4856561, 'Object', undefined, true, true) : Object = J$.R(4856561, 'Object', Object, true, true)), 'defineProperties', false)(J$.R(4856569, 'this', this, false, false), J$.T(4856601, {
                                                errors: J$.T(4856593, {
                                                    value: J$.T(4856577, {}, 11, false),
                                                    enumerable: J$.T(4856585, false, 23, false)
                                                }, 11, false)
                                            }, 11, false));
                                        } catch (J$e) {
                                            J$.Ex(4857793, J$e);
                                        } finally {
                                            if (J$.Fr(4857801))
                                                continue jalangiLabel3;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(4857577, 'arguments', arguments, true, false, false);
                            Result = J$.N(4857593, 'Result', J$.T(4857585, Result, 12, false), true, false, false);
                            J$.F(4857545, J$.R(4856633, '_createClass', _createClass, false, true), false)(J$.R(4856641, 'Result', Result, false, false), J$.T(4857537, [
                                J$.T(4856841, {
                                    key: J$.T(4856649, 'addField', 21, false),
                                    value: J$.T(4856833, function addField(field, value, errors) {
                                        jalangiLabel4:
                                            while (true) {
                                                try {
                                                    J$.Fe(4856793, arguments.callee, this, arguments);
                                                    arguments = J$.N(4856801, 'arguments', arguments, true, false, false);
                                                    field = J$.N(4856809, 'field', field, true, false, false);
                                                    value = J$.N(4856817, 'value', value, true, false, false);
                                                    errors = J$.N(4856825, 'errors', errors, true, false, false);
                                                    J$.F(4856697, J$.R(4856657, 'set', set, false, true), false)(J$.R(4856665, 'this', this, false, false), J$.G(4856681, J$.R(4856673, 'field', field, false, false), 'path'), J$.R(4856689, 'value', value, false, false));
                                                    if (J$.C(258368, J$.U(489138, '!', J$.M(4856721, J$.R(4856705, 'Field', Field, false, true), 'hasError', false)(J$.R(4856713, 'errors', errors, false, false))))) {
                                                        return J$.Rt(4856729, undefined);
                                                    }
                                                    J$.F(4856785, J$.R(4856737, 'set', set, false, true), false)(J$.G(4856753, J$.R(4856745, 'this', this, false, false), 'errors'), J$.G(4856769, J$.R(4856761, 'field', field, false, false), 'path'), J$.R(4856777, 'errors', errors, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(4857809, J$e);
                                                } finally {
                                                    if (J$.Fr(4857817))
                                                        continue jalangiLabel4;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(4856929, {
                                    key: J$.T(4856849, 'removeField', 21, false),
                                    value: J$.T(4856921, function removeField(field) {
                                        jalangiLabel5:
                                            while (true) {
                                                try {
                                                    J$.Fe(4856897, arguments.callee, this, arguments);
                                                    arguments = J$.N(4856905, 'arguments', arguments, true, false, false);
                                                    field = J$.N(4856913, 'field', field, true, false, false);
                                                    J$.F(4856889, J$.R(4856857, 'unset', unset, false, true), false)(J$.R(4856865, 'this', this, false, false), J$.G(4856881, J$.R(4856873, 'field', field, false, false), 'path'));
                                                } catch (J$e) {
                                                    J$.Ex(4857825, J$e);
                                                } finally {
                                                    if (J$.Fr(4857833))
                                                        continue jalangiLabel5;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(4857369, {
                                    key: J$.T(4856937, 'getValue', 21, false),
                                    value: J$.T(4857361, function getValue(name) {
                                        jalangiLabel7:
                                            while (true) {
                                                try {
                                                    J$.Fe(4857329, arguments.callee, this, arguments);
                                                    arguments = J$.N(4857337, 'arguments', arguments, true, false, false);
                                                    name = J$.N(4857345, 'name', name, true, false, false);
                                                    J$.N(4857353, 'value', value, false, false, false);
                                                    var value;
                                                    return J$.Rt(4857321, J$.M(4857313, J$.I(typeof regeneratorRuntime === 'undefined' ? regeneratorRuntime = J$.R(4856945, 'regeneratorRuntime', undefined, true, true) : regeneratorRuntime = J$.R(4856945, 'regeneratorRuntime', regeneratorRuntime, true, true)), 'async', false)(J$.T(4857289, function getValue$(_context) {
                                                        jalangiLabel6:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(4857265, arguments.callee, this, arguments);
                                                                    arguments = J$.N(4857273, 'arguments', arguments, true, false, false);
                                                                    _context = J$.N(4857281, '_context', _context, true, false, false);
                                                                    while (J$.C(258432, J$.T(4856953, 1, 22, false))) {
                                                                        switch (J$.C1(258384, J$.P(4856985, J$.R(4856961, '_context', _context, false, false), 'prev', J$.G(4856977, J$.R(4856969, '_context', _context, false, false), 'next')))) {
                                                                        case J$.C2(258392, J$.T(4857129, 0, 22, false)):
                                                                            value = J$.W(4857025, 'value', J$.F(4857017, J$.R(4856993, 'get', get, false, true), false)(J$.R(4857001, 'this', this, false, false), J$.R(4857009, 'name', name, false, false)), value, false, false);
                                                                            if (J$.C(258376, J$.U(489154, '!', J$.B(489146, 'instanceof', J$.R(4857033, 'value', value, false, false), J$.I(typeof Promise === 'undefined' ? Promise = J$.R(4857041, 'Promise', undefined, true, true) : Promise = J$.R(4857041, 'Promise', Promise, true, true)))))) {
                                                                                J$.P(4857065, J$.R(4857049, '_context', _context, false, false), 'next', J$.T(4857057, 5, 22, false));
                                                                                break;
                                                                            }
                                                                            J$.P(4857089, J$.R(4857073, '_context', _context, false, false), 'next', J$.T(4857081, 4, 22, false));
                                                                            return J$.Rt(4857121, J$.M(4857113, J$.I(typeof regeneratorRuntime === 'undefined' ? regeneratorRuntime = J$.R(4857097, 'regeneratorRuntime', undefined, true, true) : regeneratorRuntime = J$.R(4857097, 'regeneratorRuntime', regeneratorRuntime, true, true)), 'awrap', false)(J$.R(4857105, 'value', value, false, false)));
                                                                        case J$.C2(258400, J$.T(4857169, 4, 22, false)):
                                                                            value = J$.W(4857161, 'value', J$.G(4857153, J$.G(4857145, J$.R(4857137, '_context', _context, false, false), 'sent'), 'value'), value, false, false);
                                                                        case J$.C2(258408, J$.T(4857217, 5, 22, false)):
                                                                            return J$.Rt(4857209, J$.M(4857201, J$.R(4857177, '_context', _context, false, false), 'abrupt', false)(J$.T(4857185, 'return', 21, false), J$.R(4857193, 'value', value, false, false)));
                                                                        case J$.C2(258416, J$.T(4857225, 6, 22, false)):
                                                                        case J$.C2(258424, J$.T(4857257, 'end', 21, false)):
                                                                            return J$.Rt(4857249, J$.M(4857241, J$.R(4857233, '_context', _context, false, false), 'stop', false)());
                                                                        }
                                                                    }
                                                                } catch (J$e) {
                                                                    J$.Ex(4857841, J$e);
                                                                } finally {
                                                                    if (J$.Fr(4857849))
                                                                        continue jalangiLabel6;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false), J$.T(4857297, null, 25, false), J$.R(4857305, 'this', this, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(4857857, J$e);
                                                } finally {
                                                    if (J$.Fr(4857865))
                                                        continue jalangiLabel7;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(4857433, {
                                    key: J$.T(4857377, 'createContainer', 21, false),
                                    value: J$.T(4857425, function createContainer() {
                                        jalangiLabel8:
                                            while (true) {
                                                try {
                                                    J$.Fe(4857409, arguments.callee, this, arguments);
                                                    arguments = J$.N(4857417, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(4857401, J$.F(4857393, J$.R(4857385, 'Result', Result, false, false), true)());
                                                } catch (J$e) {
                                                    J$.Ex(4857873, J$e);
                                                } finally {
                                                    if (J$.Fr(4857881))
                                                        continue jalangiLabel8;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(4857529, {
                                    key: J$.T(4857441, 'isValid', 21, false),
                                    get: J$.T(4857521, function get() {
                                        jalangiLabel9:
                                            while (true) {
                                                try {
                                                    J$.Fe(4857505, arguments.callee, this, arguments);
                                                    arguments = J$.N(4857513, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(4857497, J$.B(489162, '===', J$.G(4857481, J$.M(4857473, J$.I(typeof Object === 'undefined' ? Object = J$.R(4857449, 'Object', undefined, true, true) : Object = J$.R(4857449, 'Object', Object, true, true)), 'keys', false)(J$.G(4857465, J$.R(4857457, 'this', this, false, false), 'errors')), 'length'), J$.T(4857489, 0, 22, false)));
                                                } catch (J$e) {
                                                    J$.Ex(4857889, J$e);
                                                } finally {
                                                    if (J$.Fr(4857897))
                                                        continue jalangiLabel9;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(4857561, J$.R(4857553, 'Result', Result, false, false));
                        } catch (J$e) {
                            J$.Ex(4857905, J$e);
                        } finally {
                            if (J$.Fr(4857913))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), Result, false, true);
            J$.P(4857641, J$.I(typeof module === 'undefined' ? module = J$.R(4857625, 'module', undefined, true, true) : module = J$.R(4857625, 'module', module, true, true)), 'exports', J$.R(4857633, 'Result', Result, false, true));
        } catch (J$e) {
            J$.Ex(4857921, J$e);
        } finally {
            if (J$.Sr(4857929))
                continue jalangiLabel11;
            else
                break jalangiLabel11;
        }
    }
// JALANGI DO NOT INSTRUMENT
