// npmlog/log.js
J$.noInstrEval = false;
jalangiLabel32:
    while (true) {
        try {
            J$.Se(4585681, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/npmlog/log.js');
            J$.N(4585689, 'Progress', Progress, false, false, false);
            J$.N(4585697, 'Gauge', Gauge, false, false, false);
            J$.N(4585705, 'EE', EE, false, false, false);
            J$.N(4585713, 'log', log, false, false, false);
            J$.N(4585721, 'util', util, false, false, false);
            J$.N(4585729, 'setBlocking', setBlocking, false, false, false);
            J$.N(4585737, 'consoleControl', consoleControl, false, false, false);
            J$.N(4585745, 'stream', stream, false, false, false);
            J$.N(4585753, 'colorEnabled', colorEnabled, false, false, false);
            J$.N(4585761, 'unicodeEnabled', unicodeEnabled, false, false, false);
            J$.N(4585769, 'trackerConstructors', trackerConstructors, false, false, false);
            J$.N(4585777, 'mixinLog', mixinLog, false, false, false);
            J$.N(4585785, 'id', id, false, false, false);
            J$.T(4577649, 'use strict', 21, false);
            var Progress = J$.W(4577681, 'Progress', J$.F(4577673, J$.I(typeof require === 'undefined' ? require = J$.R(4577657, 'require', undefined, true, true) : require = J$.R(4577657, 'require', require, true, true)), false)(J$.T(4577665, 'are-we-there-yet', 21, false)), Progress, false, true);
            var Gauge = J$.W(4577713, 'Gauge', J$.F(4577705, J$.I(typeof require === 'undefined' ? require = J$.R(4577689, 'require', undefined, true, true) : require = J$.R(4577689, 'require', require, true, true)), false)(J$.T(4577697, 'gauge', 21, false)), Gauge, false, true);
            var EE = J$.W(4577753, 'EE', J$.G(4577745, J$.F(4577737, J$.I(typeof require === 'undefined' ? require = J$.R(4577721, 'require', undefined, true, true) : require = J$.R(4577721, 'require', require, true, true)), false)(J$.T(4577729, 'events', 21, false)), 'EventEmitter'), EE, false, true);
            var log = J$.W(4577801, 'log', exports = J$.W(4577793, 'exports', J$.P(4577785, J$.I(typeof module === 'undefined' ? module = J$.R(4577761, 'module', undefined, true, true) : module = J$.R(4577761, 'module', module, true, true)), 'exports', J$.F(4577777, J$.R(4577769, 'EE', EE, false, true), true)()), J$.I(typeof exports === 'undefined' ? undefined : exports), true, true), log, false, true);
            var util = J$.W(4577833, 'util', J$.F(4577825, J$.I(typeof require === 'undefined' ? require = J$.R(4577809, 'require', undefined, true, true) : require = J$.R(4577809, 'require', require, true, true)), false)(J$.T(4577817, 'util', 21, false)), util, false, true);
            var setBlocking = J$.W(4577865, 'setBlocking', J$.F(4577857, J$.I(typeof require === 'undefined' ? require = J$.R(4577841, 'require', undefined, true, true) : require = J$.R(4577841, 'require', require, true, true)), false)(J$.T(4577849, 'set-blocking', 21, false)), setBlocking, false, true);
            var consoleControl = J$.W(4577897, 'consoleControl', J$.F(4577889, J$.I(typeof require === 'undefined' ? require = J$.R(4577873, 'require', undefined, true, true) : require = J$.R(4577873, 'require', require, true, true)), false)(J$.T(4577881, 'console-control-strings', 21, false)), consoleControl, false, true);
            J$.F(4577921, J$.R(4577905, 'setBlocking', setBlocking, false, true), false)(J$.T(4577913, true, 23, false));
            var stream = J$.W(4577945, 'stream', J$.G(4577937, J$.I(typeof process === 'undefined' ? process = J$.R(4577929, 'process', undefined, true, true) : process = J$.R(4577929, 'process', process, true, true)), 'stderr'), stream, false, true);
            J$.M(4578129, J$.I(typeof Object === 'undefined' ? Object = J$.R(4577953, 'Object', undefined, true, true) : Object = J$.R(4577953, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(4577961, 'log', log, false, true), J$.T(4577969, 'stream', 21, false), J$.T(4578121, {
                set: J$.T(4578073, function (newStream) {
                    jalangiLabel0:
                        while (true) {
                            try {
                                J$.Fe(4578049, arguments.callee, this, arguments);
                                arguments = J$.N(4578057, 'arguments', arguments, true, false, false);
                                newStream = J$.N(4578065, 'newStream', newStream, true, false, false);
                                stream = J$.W(4577985, 'stream', J$.R(4577977, 'newStream', newStream, false, false), stream, false, true);
                                if (J$.C(252536, J$.G(4578001, J$.R(4577993, 'this', this, false, false), 'gauge')))
                                    J$.M(4578041, J$.G(4578017, J$.R(4578009, 'this', this, false, false), 'gauge'), 'setWriteTo', false)(J$.R(4578025, 'stream', stream, false, true), J$.R(4578033, 'stream', stream, false, true));
                            } catch (J$e) {
                                J$.Ex(4585793, J$e);
                            } finally {
                                if (J$.Fr(4585801))
                                    continue jalangiLabel0;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                get: J$.T(4578113, function () {
                    jalangiLabel1:
                        while (true) {
                            try {
                                J$.Fe(4578097, arguments.callee, this, arguments);
                                arguments = J$.N(4578105, 'arguments', arguments, true, false, false);
                                return J$.Rt(4578089, J$.R(4578081, 'stream', stream, false, true));
                            } catch (J$e) {
                                J$.Ex(4585809, J$e);
                            } finally {
                                if (J$.Fr(4585817))
                                    continue jalangiLabel1;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            var colorEnabled;
            J$.P(4578217, J$.R(4578137, 'log', log, false, true), 'useColor', J$.T(4578209, function () {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(4578193, arguments.callee, this, arguments);
                            arguments = J$.N(4578201, 'arguments', arguments, true, false, false);
                            return J$.Rt(4578185, J$.C(252544, J$.B(556634, '!=', J$.R(4578145, 'colorEnabled', colorEnabled, false, true), J$.T(4578153, null, 25, false))) ? J$.R(4578161, 'colorEnabled', colorEnabled, false, true) : J$.G(4578177, J$.R(4578169, 'stream', stream, false, true), 'isTTY'));
                        } catch (J$e) {
                            J$.Ex(4585825, J$e);
                        } finally {
                            if (J$.Fr(4585833))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4578321, J$.R(4578225, 'log', log, false, true), 'enableColor', J$.T(4578313, function () {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(4578297, arguments.callee, this, arguments);
                            arguments = J$.N(4578305, 'arguments', arguments, true, false, false);
                            colorEnabled = J$.W(4578241, 'colorEnabled', J$.T(4578233, true, 23, false), colorEnabled, false, true);
                            J$.M(4578289, J$.G(4578257, J$.R(4578249, 'this', this, false, false), 'gauge'), 'setTheme', false)(J$.T(4578281, {
                                hasColor: J$.R(4578265, 'colorEnabled', colorEnabled, false, true),
                                hasUnicode: J$.R(4578273, 'unicodeEnabled', unicodeEnabled, false, true)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(4585841, J$e);
                        } finally {
                            if (J$.Fr(4585849))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4578425, J$.R(4578329, 'log', log, false, true), 'disableColor', J$.T(4578417, function () {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(4578401, arguments.callee, this, arguments);
                            arguments = J$.N(4578409, 'arguments', arguments, true, false, false);
                            colorEnabled = J$.W(4578345, 'colorEnabled', J$.T(4578337, false, 23, false), colorEnabled, false, true);
                            J$.M(4578393, J$.G(4578361, J$.R(4578353, 'this', this, false, false), 'gauge'), 'setTheme', false)(J$.T(4578385, {
                                hasColor: J$.R(4578369, 'colorEnabled', colorEnabled, false, true),
                                hasUnicode: J$.R(4578377, 'unicodeEnabled', unicodeEnabled, false, true)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(4585857, J$e);
                        } finally {
                            if (J$.Fr(4585865))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4578449, J$.R(4578433, 'log', log, false, true), 'level', J$.T(4578441, 'info', 21, false));
            J$.P(4578657, J$.R(4578457, 'log', log, false, true), 'gauge', J$.F(4578649, J$.R(4578465, 'Gauge', Gauge, false, true), true)(J$.R(4578473, 'stream', stream, false, true), J$.T(4578641, {
                enabled: J$.T(4578481, false, 23, false),
                theme: J$.T(4578505, { hasColor: J$.M(4578497, J$.R(4578489, 'log', log, false, true), 'useColor', false)() }, 11, false),
                template: J$.T(4578633, [
                    J$.T(4578529, {
                        type: J$.T(4578513, 'progressbar', 21, false),
                        length: J$.T(4578521, 20, 22, false)
                    }, 11, false),
                    J$.T(4578561, {
                        type: J$.T(4578537, 'activityIndicator', 21, false),
                        kerning: J$.T(4578545, 1, 22, false),
                        length: J$.T(4578553, 1, 22, false)
                    }, 11, false),
                    J$.T(4578585, {
                        type: J$.T(4578569, 'section', 21, false),
                        default: J$.T(4578577, '', 21, false)
                    }, 11, false),
                    J$.T(4578593, ':', 21, false),
                    J$.T(4578625, {
                        type: J$.T(4578601, 'logline', 21, false),
                        kerning: J$.T(4578609, 1, 22, false),
                        default: J$.T(4578617, '', 21, false)
                    }, 11, false)
                ], 10, false)
            }, 11, false)));
            J$.P(4578689, J$.R(4578665, 'log', log, false, true), 'tracker', J$.M(4578681, J$.R(4578673, 'Progress', Progress, false, true), 'TrackerGroup', true)());
            J$.P(4578729, J$.R(4578697, 'log', log, false, true), 'progressEnabled', J$.M(4578721, J$.G(4578713, J$.R(4578705, 'log', log, false, true), 'gauge'), 'isEnabled', false)());
            var unicodeEnabled;
            J$.P(4578841, J$.R(4578737, 'log', log, false, true), 'enableUnicode', J$.T(4578833, function () {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(4578817, arguments.callee, this, arguments);
                            arguments = J$.N(4578825, 'arguments', arguments, true, false, false);
                            unicodeEnabled = J$.W(4578753, 'unicodeEnabled', J$.T(4578745, true, 23, false), unicodeEnabled, false, true);
                            J$.M(4578809, J$.G(4578769, J$.R(4578761, 'this', this, false, false), 'gauge'), 'setTheme', false)(J$.T(4578801, {
                                hasColor: J$.M(4578785, J$.R(4578777, 'this', this, false, false), 'useColor', false)(),
                                hasUnicode: J$.R(4578793, 'unicodeEnabled', unicodeEnabled, false, true)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(4585873, J$e);
                        } finally {
                            if (J$.Fr(4585881))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4578953, J$.R(4578849, 'log', log, false, true), 'disableUnicode', J$.T(4578945, function () {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(4578929, arguments.callee, this, arguments);
                            arguments = J$.N(4578937, 'arguments', arguments, true, false, false);
                            unicodeEnabled = J$.W(4578865, 'unicodeEnabled', J$.T(4578857, false, 23, false), unicodeEnabled, false, true);
                            J$.M(4578921, J$.G(4578881, J$.R(4578873, 'this', this, false, false), 'gauge'), 'setTheme', false)(J$.T(4578913, {
                                hasColor: J$.M(4578897, J$.R(4578889, 'this', this, false, false), 'useColor', false)(),
                                hasUnicode: J$.R(4578905, 'unicodeEnabled', unicodeEnabled, false, true)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(4585889, J$e);
                        } finally {
                            if (J$.Fr(4585897))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4579033, J$.R(4578961, 'log', log, false, true), 'setGaugeThemeset', J$.T(4579025, function (themes) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(4579001, arguments.callee, this, arguments);
                            arguments = J$.N(4579009, 'arguments', arguments, true, false, false);
                            themes = J$.N(4579017, 'themes', themes, true, false, false);
                            J$.M(4578993, J$.G(4578977, J$.R(4578969, 'this', this, false, false), 'gauge'), 'setThemeset', false)(J$.R(4578985, 'themes', themes, false, false));
                        } catch (J$e) {
                            J$.Ex(4585905, J$e);
                        } finally {
                            if (J$.Fr(4585913))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4579113, J$.R(4579041, 'log', log, false, true), 'setGaugeTemplate', J$.T(4579105, function (template) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(4579081, arguments.callee, this, arguments);
                            arguments = J$.N(4579089, 'arguments', arguments, true, false, false);
                            template = J$.N(4579097, 'template', template, true, false, false);
                            J$.M(4579073, J$.G(4579057, J$.R(4579049, 'this', this, false, false), 'gauge'), 'setTemplate', false)(J$.R(4579065, 'template', template, false, false));
                        } catch (J$e) {
                            J$.Ex(4585921, J$e);
                        } finally {
                            if (J$.Fr(4585929))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4579297, J$.R(4579121, 'log', log, false, true), 'enableProgress', J$.T(4579289, function () {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(4579273, arguments.callee, this, arguments);
                            arguments = J$.N(4579281, 'arguments', arguments, true, false, false);
                            if (J$.C(252552, J$.G(4579137, J$.R(4579129, 'this', this, false, false), 'progressEnabled')))
                                return J$.Rt(4579145, undefined);
                            J$.P(4579169, J$.R(4579153, 'this', this, false, false), 'progressEnabled', J$.T(4579161, true, 23, false));
                            J$.M(4579217, J$.G(4579185, J$.R(4579177, 'this', this, false, false), 'tracker'), 'on', false)(J$.T(4579193, 'change', 21, false), J$.G(4579209, J$.R(4579201, 'this', this, false, false), 'showProgress'));
                            if (J$.C(252560, J$.G(4579233, J$.R(4579225, 'this', this, false, false), '_pause')))
                                return J$.Rt(4579241, undefined);
                            J$.M(4579265, J$.G(4579257, J$.R(4579249, 'this', this, false, false), 'gauge'), 'enable', false)();
                        } catch (J$e) {
                            J$.Ex(4585937, J$e);
                        } finally {
                            if (J$.Fr(4585945))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4579457, J$.R(4579305, 'log', log, false, true), 'disableProgress', J$.T(4579449, function () {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(4579433, arguments.callee, this, arguments);
                            arguments = J$.N(4579441, 'arguments', arguments, true, false, false);
                            if (J$.C(252568, J$.U(556642, '!', J$.G(4579321, J$.R(4579313, 'this', this, false, false), 'progressEnabled'))))
                                return J$.Rt(4579329, undefined);
                            J$.P(4579353, J$.R(4579337, 'this', this, false, false), 'progressEnabled', J$.T(4579345, false, 23, false));
                            J$.M(4579401, J$.G(4579369, J$.R(4579361, 'this', this, false, false), 'tracker'), 'removeListener', false)(J$.T(4579377, 'change', 21, false), J$.G(4579393, J$.R(4579385, 'this', this, false, false), 'showProgress'));
                            J$.M(4579425, J$.G(4579417, J$.R(4579409, 'this', this, false, false), 'gauge'), 'disable', false)();
                        } catch (J$e) {
                            J$.Ex(4585953, J$e);
                        } finally {
                            if (J$.Fr(4585961))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var trackerConstructors = J$.W(4579497, 'trackerConstructors', J$.T(4579489, [
                J$.T(4579465, 'newGroup', 21, false),
                J$.T(4579473, 'newItem', 21, false),
                J$.T(4579481, 'newStream', 21, false)
            ], 10, false), trackerConstructors, false, true);
            var mixinLog = J$.W(4580161, 'mixinLog', J$.T(4580153, function (tracker) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(4580129, arguments.callee, this, arguments);
                            arguments = J$.N(4580137, 'arguments', arguments, true, false, false);
                            tracker = J$.N(4580145, 'tracker', tracker, true, false, false);
                            J$.M(4579889, J$.M(4579521, J$.I(typeof Object === 'undefined' ? Object = J$.R(4579505, 'Object', undefined, true, true) : Object = J$.R(4579505, 'Object', Object, true, true)), 'keys', false)(J$.R(4579513, 'log', log, false, true)), 'forEach', false)(J$.T(4579881, function (P) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(4579849, arguments.callee, this, arguments);
                                            arguments = J$.N(4579857, 'arguments', arguments, true, false, false);
                                            P = J$.N(4579865, 'P', P, true, false, false);
                                            J$.N(4579873, 'func', func, false, false, false);
                                            if (J$.C(252576, J$.B(556650, '===', J$.G(4579545, J$.R(4579529, 'P', P, false, false), J$.T(4579537, 0, 22, false)), J$.T(4579553, '_', 21, false))))
                                                return J$.Rt(4579561, undefined);
                                            if (J$.C(252584, J$.G(4579641, J$.M(4579633, J$.R(4579569, 'trackerConstructors', trackerConstructors, false, true), 'filter', false)(J$.T(4579625, function (C) {
                                                    jalangiLabel11:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(4579601, arguments.callee, this, arguments);
                                                                arguments = J$.N(4579609, 'arguments', arguments, true, false, false);
                                                                C = J$.N(4579617, 'C', C, true, false, false);
                                                                return J$.Rt(4579593, J$.B(556658, '===', J$.R(4579577, 'C', C, false, false), J$.R(4579585, 'P', P, false, false)));
                                                            } catch (J$e) {
                                                                J$.Ex(4585969, J$e);
                                                            } finally {
                                                                if (J$.Fr(4585977))
                                                                    continue jalangiLabel11;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false)), 'length')))
                                                return J$.Rt(4579649, undefined);
                                            if (J$.C(252592, J$.G(4579673, J$.R(4579657, 'tracker', tracker, false, false), J$.R(4579665, 'P', P, false, false))))
                                                return J$.Rt(4579681, undefined);
                                            if (J$.C(252600, J$.B(556674, '!==', J$.U(556666, 'typeof', J$.G(4579705, J$.R(4579689, 'log', log, false, true), J$.R(4579697, 'P', P, false, false))), J$.T(4579713, 'function', 21, false))))
                                                return J$.Rt(4579721, undefined);
                                            var func = J$.W(4579753, 'func', J$.G(4579745, J$.R(4579729, 'log', log, false, true), J$.R(4579737, 'P', P, false, false)), func, false, false);
                                            J$.P(4579841, J$.R(4579761, 'tracker', tracker, false, false), J$.R(4579769, 'P', P, false, false), J$.T(4579833, function () {
                                                jalangiLabel12:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(4579817, arguments.callee, this, arguments);
                                                            arguments = J$.N(4579825, 'arguments', arguments, true, false, false);
                                                            return J$.Rt(4579809, J$.M(4579801, J$.R(4579777, 'func', func, false, false), 'apply', false)(J$.R(4579785, 'log', log, false, true), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4579793, 'arguments', undefined, true, true) : arguments = J$.R(4579793, 'arguments', arguments, true, true))));
                                                        } catch (J$e) {
                                                            J$.Ex(4585985, J$e);
                                                        } finally {
                                                            if (J$.Fr(4585993))
                                                                continue jalangiLabel12;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(4586001, J$e);
                                        } finally {
                                            if (J$.Fr(4586009))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            if (J$.C(252608, J$.B(556682, 'instanceof', J$.R(4579897, 'tracker', tracker, false, false), J$.G(4579913, J$.R(4579905, 'Progress', Progress, false, true), 'TrackerGroup')))) {
                                J$.M(4580105, J$.R(4579921, 'trackerConstructors', trackerConstructors, false, true), 'forEach', false)(J$.T(4580097, function (C) {
                                    jalangiLabel15:
                                        while (true) {
                                            try {
                                                J$.Fe(4580065, arguments.callee, this, arguments);
                                                arguments = J$.N(4580073, 'arguments', arguments, true, false, false);
                                                C = J$.N(4580081, 'C', C, true, false, false);
                                                J$.N(4580089, 'func', func, false, false, false);
                                                var func = J$.W(4579953, 'func', J$.G(4579945, J$.R(4579929, 'tracker', tracker, false, false), J$.R(4579937, 'C', C, false, false)), func, false, false);
                                                J$.P(4580057, J$.R(4579961, 'tracker', tracker, false, false), J$.R(4579969, 'C', C, false, false), J$.T(4580049, function () {
                                                    jalangiLabel14:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(4580033, arguments.callee, this, arguments);
                                                                arguments = J$.N(4580041, 'arguments', arguments, true, false, false);
                                                                return J$.Rt(4580025, J$.F(4580017, J$.R(4579977, 'mixinLog', mixinLog, false, true), false)(J$.M(4580009, J$.R(4579985, 'func', func, false, false), 'apply', false)(J$.R(4579993, 'tracker', tracker, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4580001, 'arguments', undefined, true, true) : arguments = J$.R(4580001, 'arguments', arguments, true, true)))));
                                                            } catch (J$e) {
                                                                J$.Ex(4586017, J$e);
                                                            } finally {
                                                                if (J$.Fr(4586025))
                                                                    continue jalangiLabel14;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false));
                                            } catch (J$e) {
                                                J$.Ex(4586033, J$e);
                                            } finally {
                                                if (J$.Fr(4586041))
                                                    continue jalangiLabel15;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            }
                            return J$.Rt(4580121, J$.R(4580113, 'tracker', tracker, false, false));
                        } catch (J$e) {
                            J$.Ex(4586049, J$e);
                        } finally {
                            if (J$.Fr(4586057))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), mixinLog, false, true);
            J$.M(4580345, J$.R(4580169, 'trackerConstructors', trackerConstructors, false, true), 'forEach', false)(J$.T(4580337, function (C) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(4580313, arguments.callee, this, arguments);
                            arguments = J$.N(4580321, 'arguments', arguments, true, false, false);
                            C = J$.N(4580329, 'C', C, true, false, false);
                            J$.P(4580305, J$.R(4580177, 'log', log, false, true), J$.R(4580185, 'C', C, false, false), J$.T(4580297, function () {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(4580281, arguments.callee, this, arguments);
                                            arguments = J$.N(4580289, 'arguments', arguments, true, false, false);
                                            return J$.Rt(4580273, J$.F(4580265, J$.R(4580193, 'mixinLog', mixinLog, false, true), false)(J$.M(4580257, J$.G(4580225, J$.G(4580209, J$.R(4580201, 'this', this, false, false), 'tracker'), J$.R(4580217, 'C', C, false, false)), 'apply', false)(J$.G(4580241, J$.R(4580233, 'this', this, false, false), 'tracker'), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4580249, 'arguments', undefined, true, true) : arguments = J$.R(4580249, 'arguments', arguments, true, true)))));
                                        } catch (J$e) {
                                            J$.Ex(4586065, J$e);
                                        } finally {
                                            if (J$.Fr(4586073))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(4586081, J$e);
                        } finally {
                            if (J$.Fr(4586089))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4580481, J$.R(4580353, 'log', log, false, true), 'clearProgress', J$.T(4580473, function (cb) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(4580449, arguments.callee, this, arguments);
                            arguments = J$.N(4580457, 'arguments', arguments, true, false, false);
                            cb = J$.N(4580465, 'cb', cb, true, false, false);
                            if (J$.C(252624, J$.U(556690, '!', J$.G(4580369, J$.R(4580361, 'this', this, false, false), 'progressEnabled'))))
                                return J$.Rt(4580409, J$.C(252616, J$.R(4580377, 'cb', cb, false, false)) ? J$.M(4580401, J$.I(typeof process === 'undefined' ? process = J$.R(4580385, 'process', undefined, true, true) : process = J$.R(4580385, 'process', process, true, true)), 'nextTick', false)(J$.R(4580393, 'cb', cb, false, false)) : J$._());
                            J$.M(4580441, J$.G(4580425, J$.R(4580417, 'this', this, false, false), 'gauge'), 'hide', false)(J$.R(4580433, 'cb', cb, false, false));
                        } catch (J$e) {
                            J$.Ex(4586097, J$e);
                        } finally {
                            if (J$.Fr(4586105))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4581161, J$.R(4580489, 'log', log, false, true), 'showProgress', J$.M(4581153, J$.T(4581137, function (name, completed) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(4581073, arguments.callee, this, arguments);
                            arguments = J$.N(4581081, 'arguments', arguments, true, false, false);
                            name = J$.N(4581089, 'name', name, true, false, false);
                            completed = J$.N(4581097, 'completed', completed, true, false, false);
                            J$.N(4581105, 'values', values, false, false, false);
                            J$.N(4581113, 'last', last, false, false, false);
                            J$.N(4581121, 'disp', disp, false, false, false);
                            J$.N(4581129, 'logline', logline, false, false, false);
                            if (J$.C(252632, J$.U(556698, '!', J$.G(4580505, J$.R(4580497, 'this', this, false, false), 'progressEnabled'))))
                                return J$.Rt(4580513, undefined);
                            var values = J$.W(4580529, 'values', J$.T(4580521, {}, 11, false), values, false, false);
                            if (J$.C(252640, J$.R(4580537, 'name', name, false, false)))
                                J$.P(4580561, J$.R(4580545, 'values', values, false, false), 'section', J$.R(4580553, 'name', name, false, false));
                            var last = J$.W(4580625, 'last', J$.G(4580617, J$.G(4580577, J$.R(4580569, 'log', log, false, true), 'record'), J$.B(556706, '-', J$.G(4580601, J$.G(4580593, J$.R(4580585, 'log', log, false, true), 'record'), 'length'), J$.T(4580609, 1, 22, false))), last, false, false);
                            if (J$.C(252664, J$.R(4580633, 'last', last, false, false))) {
                                J$.P(4580665, J$.R(4580641, 'values', values, false, false), 'subsection', J$.G(4580657, J$.R(4580649, 'last', last, false, false), 'prefix'));
                                var disp = J$.W(4580729, 'disp', J$.C(252648, J$.G(4580705, J$.G(4580681, J$.R(4580673, 'log', log, false, true), 'disp'), J$.G(4580697, J$.R(4580689, 'last', last, false, false), 'level'))) ? J$._() : J$.G(4580721, J$.R(4580713, 'last', last, false, false), 'level'), disp, false, false);
                                var logline = J$.W(4580801, 'logline', J$.M(4580793, J$.R(4580737, 'this', this, false, false), '_format', false)(J$.R(4580745, 'disp', disp, false, false), J$.G(4580785, J$.G(4580761, J$.R(4580753, 'log', log, false, true), 'style'), J$.G(4580777, J$.R(4580769, 'last', last, false, false), 'level'))), logline, false, false);
                                if (J$.C(252656, J$.G(4580817, J$.R(4580809, 'last', last, false, false), 'prefix')))
                                    logline = J$.W(4580889, 'logline', J$.B(556722, '+', J$.R(4580881, 'logline', logline, false, false), J$.B(556714, '+', J$.T(4580825, ' ', 21, false), J$.M(4580873, J$.R(4580833, 'this', this, false, false), '_format', false)(J$.G(4580849, J$.R(4580841, 'last', last, false, false), 'prefix'), J$.G(4580865, J$.R(4580857, 'this', this, false, false), 'prefixStyle')))), logline, false, false);
                                logline = J$.W(4580961, 'logline', J$.B(556738, '+', J$.R(4580953, 'logline', logline, false, false), J$.B(556730, '+', J$.T(4580897, ' ', 21, false), J$.G(4580945, J$.M(4580929, J$.G(4580913, J$.R(4580905, 'last', last, false, false), 'message'), 'split', false)(J$.T(4580921, /\r?\n/, 14, false)), J$.T(4580937, 0, 22, false)))), logline, false, false);
                                J$.P(4580985, J$.R(4580969, 'values', values, false, false), 'logline', J$.R(4580977, 'logline', logline, false, false));
                            }
                            J$.P(4581033, J$.R(4580993, 'values', values, false, false), 'completed', J$.C(252672, J$.R(4581001, 'completed', completed, false, false)) ? J$._() : J$.M(4581025, J$.G(4581017, J$.R(4581009, 'this', this, false, false), 'tracker'), 'completed', false)());
                            J$.M(4581065, J$.G(4581049, J$.R(4581041, 'this', this, false, false), 'gauge'), 'show', false)(J$.R(4581057, 'values', values, false, false));
                        } catch (J$e) {
                            J$.Ex(4586113, J$e);
                        } finally {
                            if (J$.Fr(4586121))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), 'bind', false)(J$.R(4581145, 'log', log, false, true)));
            J$.P(4581265, J$.R(4581169, 'log', log, false, true), 'pause', J$.T(4581257, function () {
                jalangiLabel21:
                    while (true) {
                        try {
                            J$.Fe(4581241, arguments.callee, this, arguments);
                            arguments = J$.N(4581249, 'arguments', arguments, true, false, false);
                            J$.P(4581193, J$.R(4581177, 'this', this, false, false), '_paused', J$.T(4581185, true, 23, false));
                            if (J$.C(252680, J$.G(4581209, J$.R(4581201, 'this', this, false, false), 'progressEnabled')))
                                J$.M(4581233, J$.G(4581225, J$.R(4581217, 'this', this, false, false), 'gauge'), 'disable', false)();
                        } catch (J$e) {
                            J$.Ex(4586129, J$e);
                        } finally {
                            if (J$.Fr(4586137))
                                continue jalangiLabel21;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4581529, J$.R(4581273, 'log', log, false, true), 'resume', J$.T(4581521, function () {
                jalangiLabel23:
                    while (true) {
                        try {
                            J$.Fe(4581497, arguments.callee, this, arguments);
                            arguments = J$.N(4581505, 'arguments', arguments, true, false, false);
                            J$.N(4581513, 'b', b, false, false, false);
                            if (J$.C(252688, J$.U(556746, '!', J$.G(4581289, J$.R(4581281, 'this', this, false, false), '_paused'))))
                                return J$.Rt(4581297, undefined);
                            J$.P(4581321, J$.R(4581305, 'this', this, false, false), '_paused', J$.T(4581313, false, 23, false));
                            var b = J$.W(4581345, 'b', J$.G(4581337, J$.R(4581329, 'this', this, false, false), '_buffer'), b, false, false);
                            J$.P(4581369, J$.R(4581353, 'this', this, false, false), '_buffer', J$.T(4581361, [], 10, false));
                            J$.M(4581449, J$.R(4581377, 'b', b, false, false), 'forEach', false)(J$.T(4581433, function (m) {
                                jalangiLabel22:
                                    while (true) {
                                        try {
                                            J$.Fe(4581409, arguments.callee, this, arguments);
                                            arguments = J$.N(4581417, 'arguments', arguments, true, false, false);
                                            m = J$.N(4581425, 'm', m, true, false, false);
                                            J$.M(4581401, J$.R(4581385, 'this', this, false, false), 'emitLog', false)(J$.R(4581393, 'm', m, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4586145, J$e);
                                        } finally {
                                            if (J$.Fr(4586153))
                                                continue jalangiLabel22;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), J$.R(4581441, 'this', this, false, false));
                            if (J$.C(252696, J$.G(4581465, J$.R(4581457, 'this', this, false, false), 'progressEnabled')))
                                J$.M(4581489, J$.G(4581481, J$.R(4581473, 'this', this, false, false), 'gauge'), 'enable', false)();
                        } catch (J$e) {
                            J$.Ex(4586161, J$e);
                        } finally {
                            if (J$.Fr(4586169))
                                continue jalangiLabel23;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4581553, J$.R(4581537, 'log', log, false, true), '_buffer', J$.T(4581545, [], 10, false));
            var id = J$.W(4581569, 'id', J$.T(4581561, 0, 22, false), id, false, true);
            J$.P(4581593, J$.R(4581577, 'log', log, false, true), 'record', J$.T(4581585, [], 10, false));
            J$.P(4581617, J$.R(4581601, 'log', log, false, true), 'maxRecordSize', J$.T(4581609, 10000, 22, false));
            J$.P(4582777, J$.R(4581625, 'log', log, false, true), 'log', J$.M(4582769, J$.T(4582753, function (lvl, prefix, message) {
                jalangiLabel24:
                    while (true) {
                        try {
                            J$.Fe(4582641, arguments.callee, this, arguments);
                            arguments = J$.N(4582649, 'arguments', arguments, true, false, false);
                            lvl = J$.N(4582657, 'lvl', lvl, true, false, false);
                            prefix = J$.N(4582665, 'prefix', prefix, true, false, false);
                            message = J$.N(4582673, 'message', message, true, false, false);
                            J$.N(4582681, 'l', l, false, false, false);
                            J$.N(4582689, 'a', a, false, false, false);
                            J$.N(4582697, 'stack', stack, false, false, false);
                            J$.N(4582705, 'i', i, false, false, false);
                            J$.N(4582713, 'arg', arg, false, false, false);
                            J$.N(4582721, 'm', m, false, false, false);
                            J$.N(4582729, 'mrs', mrs, false, false, false);
                            J$.N(4582737, 'n', n, false, false, false);
                            J$.N(4582745, 'newSize', newSize, false, false, false);
                            var l = J$.W(4581665, 'l', J$.G(4581657, J$.G(4581641, J$.R(4581633, 'this', this, false, false), 'levels'), J$.R(4581649, 'lvl', lvl, false, false)), l, false, false);
                            if (J$.C(252704, J$.B(556754, '===', J$.R(4581673, 'l', l, false, false), J$.T(4581681, undefined, 24, false)))) {
                                return J$.Rt(4581761, J$.M(4581753, J$.R(4581689, 'this', this, false, false), 'emit', false)(J$.T(4581697, 'error', 21, false), J$.F(4581745, J$.I(typeof Error === 'undefined' ? Error = J$.R(4581705, 'Error', undefined, true, true) : Error = J$.R(4581705, 'Error', Error, true, true)), true)(J$.M(4581737, J$.R(4581713, 'util', util, false, true), 'format', false)(J$.T(4581721, 'Undefined log level: %j', 21, false), J$.R(4581729, 'lvl', lvl, false, false)))));
                            }
                            var a = J$.W(4581809, 'a', J$.F(4581801, J$.I(typeof Array === 'undefined' ? Array = J$.R(4581769, 'Array', undefined, true, true) : Array = J$.R(4581769, 'Array', Array, true, true)), true)(J$.B(556762, '-', J$.G(4581785, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4581777, 'arguments', undefined, true, true) : arguments = J$.R(4581777, 'arguments', arguments, true, true)), 'length'), J$.T(4581793, 2, 22, false))), a, false, false);
                            var stack = J$.W(4581825, 'stack', J$.T(4581817, null, 25, false), stack, false, false);
                            for (var i = J$.W(4581841, 'i', J$.T(4581833, 2, 22, false), i, false, false); J$.C(252744, J$.B(556770, '<', J$.R(4581849, 'i', i, false, false), J$.G(4581865, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4581857, 'arguments', undefined, true, true) : arguments = J$.R(4581857, 'arguments', arguments, true, true)), 'length'))); J$.B(556794, '-', i = J$.W(4581881, 'i', J$.B(556786, '+', J$.U(556778, '+', J$.R(4581873, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var arg = J$.W(4581945, 'arg', J$.P(4581937, J$.R(4581889, 'a', a, false, false), J$.B(556802, '-', J$.R(4581897, 'i', i, false, false), J$.T(4581905, 2, 22, false)), J$.G(4581929, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4581913, 'arguments', undefined, true, true) : arguments = J$.R(4581913, 'arguments', arguments, true, true)), J$.R(4581921, 'i', i, false, false))), arg, false, false);
                                if (J$.C(252736, J$.C(252728, J$.C(252720, J$.C(252712, J$.B(556818, '===', J$.U(556810, 'typeof', J$.R(4581953, 'arg', arg, false, false)), J$.T(4581961, 'object', 21, false))) ? J$.R(4581969, 'arg', arg, false, false) : J$._()) ? J$.B(556826, 'instanceof', J$.R(4581977, 'arg', arg, false, false), J$.I(typeof Error === 'undefined' ? Error = J$.R(4581985, 'Error', undefined, true, true) : Error = J$.R(4581985, 'Error', Error, true, true))) : J$._()) ? J$.G(4582001, J$.R(4581993, 'arg', arg, false, false), 'stack') : J$._())) {
                                    J$.M(4582089, J$.I(typeof Object === 'undefined' ? Object = J$.R(4582009, 'Object', undefined, true, true) : Object = J$.R(4582009, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(4582017, 'arg', arg, false, false), J$.T(4582025, 'stack', 21, false), J$.T(4582081, {
                                        value: stack = J$.W(4582057, 'stack', J$.B(556834, '+', J$.G(4582041, J$.R(4582033, 'arg', arg, false, false), 'stack'), J$.T(4582049, '', 21, false)), stack, false, false),
                                        enumerable: J$.T(4582065, true, 23, false),
                                        writable: J$.T(4582073, true, 23, false)
                                    }, 11, false));
                                }
                            }
                            if (J$.C(252752, J$.R(4582097, 'stack', stack, false, false)))
                                J$.M(4582129, J$.R(4582105, 'a', a, false, false), 'unshift', false)(J$.B(556842, '+', J$.R(4582113, 'stack', stack, false, false), J$.T(4582121, '\n', 21, false)));
                            message = J$.W(4582177, 'message', J$.M(4582169, J$.G(4582145, J$.R(4582137, 'util', util, false, true), 'format'), 'apply', false)(J$.R(4582153, 'util', util, false, true), J$.R(4582161, 'a', a, false, false)), message, false, false);
                            var m = J$.W(4582265, 'm', J$.T(4582257, {
                                id: J$.B(556866, '-', id = J$.W(4582193, 'id', J$.B(556858, '+', J$.U(556850, '+', J$.R(4582185, 'id', id, false, true)), 1), id, false, true), 1),
                                level: J$.R(4582201, 'lvl', lvl, false, false),
                                prefix: J$.F(4582233, J$.I(typeof String === 'undefined' ? String = J$.R(4582209, 'String', undefined, true, true) : String = J$.R(4582209, 'String', String, true, true)), false)(J$.C(252760, J$.R(4582217, 'prefix', prefix, false, false)) ? J$._() : J$.T(4582225, '', 21, false)),
                                message: J$.R(4582241, 'message', message, false, false),
                                messageRaw: J$.R(4582249, 'a', a, false, false)
                            }, 11, false), m, false, false);
                            J$.M(4582297, J$.R(4582273, 'this', this, false, false), 'emit', false)(J$.T(4582281, 'log', 21, false), J$.R(4582289, 'm', m, false, false));
                            J$.M(4582337, J$.R(4582305, 'this', this, false, false), 'emit', false)(J$.B(556874, '+', J$.T(4582313, 'log.', 21, false), J$.R(4582321, 'lvl', lvl, false, false)), J$.R(4582329, 'm', m, false, false));
                            if (J$.C(252768, J$.G(4582353, J$.R(4582345, 'm', m, false, false), 'prefix')))
                                J$.M(4582393, J$.R(4582361, 'this', this, false, false), 'emit', false)(J$.G(4582377, J$.R(4582369, 'm', m, false, false), 'prefix'), J$.R(4582385, 'm', m, false, false));
                            J$.M(4582425, J$.G(4582409, J$.R(4582401, 'this', this, false, false), 'record'), 'push', false)(J$.R(4582417, 'm', m, false, false));
                            var mrs = J$.W(4582449, 'mrs', J$.G(4582441, J$.R(4582433, 'this', this, false, false), 'maxRecordSize'), mrs, false, false);
                            var n = J$.W(4582489, 'n', J$.B(556882, '-', J$.G(4582473, J$.G(4582465, J$.R(4582457, 'this', this, false, false), 'record'), 'length'), J$.R(4582481, 'mrs', mrs, false, false)), n, false, false);
                            if (J$.C(252776, J$.B(556898, '>', J$.R(4582497, 'n', n, false, false), J$.B(556890, '/', J$.R(4582505, 'mrs', mrs, false, false), J$.T(4582513, 10, 22, false))))) {
                                var newSize = J$.W(4582553, 'newSize', J$.M(4582545, J$.I(typeof Math === 'undefined' ? Math = J$.R(4582521, 'Math', undefined, true, true) : Math = J$.R(4582521, 'Math', Math, true, true)), 'floor', false)(J$.B(556906, '*', J$.R(4582529, 'mrs', mrs, false, false), J$.T(4582537, 0.9, 22, false))), newSize, false, false);
                                J$.P(4582609, J$.R(4582561, 'this', this, false, false), 'record', J$.M(4582601, J$.G(4582577, J$.R(4582569, 'this', this, false, false), 'record'), 'slice', false)(J$.B(556922, '*', J$.U(556914, '-', J$.T(4582585, 1, 22, false)), J$.R(4582593, 'newSize', newSize, false, false))));
                            }
                            J$.M(4582633, J$.R(4582617, 'this', this, false, false), 'emitLog', false)(J$.R(4582625, 'm', m, false, false));
                        } catch (J$e) {
                            J$.Ex(4586177, J$e);
                        } finally {
                            if (J$.Fr(4586185))
                                continue jalangiLabel24;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), 'bind', false)(J$.R(4582761, 'log', log, false, true)));
            J$.P(4583657, J$.R(4582785, 'log', log, false, true), 'emitLog', J$.T(4583649, function (m) {
                jalangiLabel26:
                    while (true) {
                        try {
                            J$.Fe(4583609, arguments.callee, this, arguments);
                            arguments = J$.N(4583617, 'arguments', arguments, true, false, false);
                            m = J$.N(4583625, 'm', m, true, false, false);
                            J$.N(4583633, 'l', l, false, false, false);
                            J$.N(4583641, 'disp', disp, false, false, false);
                            if (J$.C(252784, J$.G(4582801, J$.R(4582793, 'this', this, false, false), '_paused'))) {
                                J$.M(4582833, J$.G(4582817, J$.R(4582809, 'this', this, false, false), '_buffer'), 'push', false)(J$.R(4582825, 'm', m, false, false));
                                return J$.Rt(4582841, undefined);
                            }
                            if (J$.C(252792, J$.G(4582857, J$.R(4582849, 'this', this, false, false), 'progressEnabled')))
                                J$.M(4582897, J$.G(4582873, J$.R(4582865, 'this', this, false, false), 'gauge'), 'pulse', false)(J$.G(4582889, J$.R(4582881, 'm', m, false, false), 'prefix'));
                            var l = J$.W(4582945, 'l', J$.G(4582937, J$.G(4582913, J$.R(4582905, 'this', this, false, false), 'levels'), J$.G(4582929, J$.R(4582921, 'm', m, false, false), 'level')), l, false, false);
                            if (J$.C(252800, J$.B(556930, '===', J$.R(4582953, 'l', l, false, false), J$.T(4582961, undefined, 24, false))))
                                return J$.Rt(4582969, undefined);
                            if (J$.C(252808, J$.B(556938, '<', J$.R(4582977, 'l', l, false, false), J$.G(4583017, J$.G(4582993, J$.R(4582985, 'this', this, false, false), 'levels'), J$.G(4583009, J$.R(4583001, 'this', this, false, false), 'level')))))
                                return J$.Rt(4583025, undefined);
                            if (J$.C(252824, J$.C(252816, J$.B(556946, '>', J$.R(4583033, 'l', l, false, false), J$.T(4583041, 0, 22, false))) ? J$.U(556954, '!', J$.F(4583065, J$.I(typeof isFinite === 'undefined' ? isFinite = J$.R(4583049, 'isFinite', undefined, true, true) : isFinite = J$.R(4583049, 'isFinite', isFinite, true, true)), false)(J$.R(4583057, 'l', l, false, false))) : J$._()))
                                return J$.Rt(4583073, undefined);
                            var disp = J$.W(4583185, 'disp', J$.C(252832, J$.B(556962, '!=', J$.G(4583113, J$.G(4583089, J$.R(4583081, 'log', log, false, true), 'disp'), J$.G(4583105, J$.R(4583097, 'm', m, false, false), 'level')), J$.T(4583121, null, 25, false))) ? J$.G(4583161, J$.G(4583137, J$.R(4583129, 'log', log, false, true), 'disp'), J$.G(4583153, J$.R(4583145, 'm', m, false, false), 'level')) : J$.G(4583177, J$.R(4583169, 'm', m, false, false), 'level'), disp, false, false);
                            J$.M(4583201, J$.R(4583193, 'this', this, false, false), 'clearProgress', false)();
                            J$.M(4583585, J$.M(4583233, J$.G(4583217, J$.R(4583209, 'm', m, false, false), 'message'), 'split', false)(J$.T(4583225, /\r?\n/, 14, false)), 'forEach', false)(J$.T(4583569, function (line) {
                                jalangiLabel25:
                                    while (true) {
                                        try {
                                            J$.Fe(4583537, arguments.callee, this, arguments);
                                            arguments = J$.N(4583545, 'arguments', arguments, true, false, false);
                                            line = J$.N(4583553, 'line', line, true, false, false);
                                            J$.N(4583561, 'p', p, false, false, false);
                                            if (J$.C(252840, J$.G(4583249, J$.R(4583241, 'this', this, false, false), 'heading'))) {
                                                J$.M(4583297, J$.R(4583257, 'this', this, false, false), 'write', false)(J$.G(4583273, J$.R(4583265, 'this', this, false, false), 'heading'), J$.G(4583289, J$.R(4583281, 'this', this, false, false), 'headingStyle'));
                                                J$.M(4583321, J$.R(4583305, 'this', this, false, false), 'write', false)(J$.T(4583313, ' ', 21, false));
                                            }
                                            J$.M(4583385, J$.R(4583329, 'this', this, false, false), 'write', false)(J$.R(4583337, 'disp', disp, false, false), J$.G(4583377, J$.G(4583353, J$.R(4583345, 'log', log, false, true), 'style'), J$.G(4583369, J$.R(4583361, 'm', m, false, false), 'level')));
                                            var p = J$.W(4583417, 'p', J$.C(252848, J$.G(4583401, J$.R(4583393, 'm', m, false, false), 'prefix')) ? J$._() : J$.T(4583409, '', 21, false), p, false, false);
                                            if (J$.C(252856, J$.R(4583425, 'p', p, false, false)))
                                                J$.M(4583449, J$.R(4583433, 'this', this, false, false), 'write', false)(J$.T(4583441, ' ', 21, false));
                                            J$.M(4583489, J$.R(4583457, 'this', this, false, false), 'write', false)(J$.R(4583465, 'p', p, false, false), J$.G(4583481, J$.R(4583473, 'this', this, false, false), 'prefixStyle'));
                                            J$.M(4583529, J$.R(4583497, 'this', this, false, false), 'write', false)(J$.B(556978, '+', J$.B(556970, '+', J$.T(4583505, ' ', 21, false), J$.R(4583513, 'line', line, false, false)), J$.T(4583521, '\n', 21, false)));
                                        } catch (J$e) {
                                            J$.Ex(4586193, J$e);
                                        } finally {
                                            if (J$.Fr(4586201))
                                                continue jalangiLabel25;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), J$.R(4583577, 'this', this, false, false));
                            J$.M(4583601, J$.R(4583593, 'this', this, false, false), 'showProgress', false)();
                        } catch (J$e) {
                            J$.Ex(4586209, J$e);
                        } finally {
                            if (J$.Fr(4586217))
                                continue jalangiLabel26;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4584297, J$.R(4583665, 'log', log, false, true), '_format', J$.T(4584289, function (msg, style) {
                jalangiLabel27:
                    while (true) {
                        try {
                            J$.Fe(4584241, arguments.callee, this, arguments);
                            arguments = J$.N(4584249, 'arguments', arguments, true, false, false);
                            msg = J$.N(4584257, 'msg', msg, true, false, false);
                            style = J$.N(4584265, 'style', style, true, false, false);
                            J$.N(4584273, 'output', output, false, false, false);
                            J$.N(4584281, 'settings', settings, false, false, false);
                            if (J$.C(252864, J$.U(556986, '!', J$.R(4583673, 'stream', stream, false, true))))
                                return J$.Rt(4583681, undefined);
                            var output = J$.W(4583697, 'output', J$.T(4583689, '', 21, false), output, false, false);
                            if (J$.C(252936, J$.M(4583713, J$.R(4583705, 'this', this, false, false), 'useColor', false)())) {
                                style = J$.W(4583737, 'style', J$.C(252872, J$.R(4583721, 'style', style, false, false)) ? J$._() : J$.T(4583729, {}, 11, false), style, false, false);
                                var settings = J$.W(4583753, 'settings', J$.T(4583745, [], 10, false), settings, false, false);
                                if (J$.C(252880, J$.G(4583769, J$.R(4583761, 'style', style, false, false), 'fg')))
                                    J$.M(4583801, J$.R(4583777, 'settings', settings, false, false), 'push', false)(J$.G(4583793, J$.R(4583785, 'style', style, false, false), 'fg'));
                                if (J$.C(252888, J$.G(4583817, J$.R(4583809, 'style', style, false, false), 'bg')))
                                    J$.M(4583913, J$.R(4583825, 'settings', settings, false, false), 'push', false)(J$.B(557002, '+', J$.B(556994, '+', J$.T(4583833, 'bg', 21, false), J$.M(4583873, J$.G(4583865, J$.G(4583849, J$.R(4583841, 'style', style, false, false), 'bg'), J$.T(4583857, 0, 22, false)), 'toUpperCase', false)()), J$.M(4583905, J$.G(4583889, J$.R(4583881, 'style', style, false, false), 'bg'), 'slice', false)(J$.T(4583897, 1, 22, false))));
                                if (J$.C(252896, J$.G(4583929, J$.R(4583921, 'style', style, false, false), 'bold')))
                                    J$.M(4583953, J$.R(4583937, 'settings', settings, false, false), 'push', false)(J$.T(4583945, 'bold', 21, false));
                                if (J$.C(252904, J$.G(4583969, J$.R(4583961, 'style', style, false, false), 'underline')))
                                    J$.M(4583993, J$.R(4583977, 'settings', settings, false, false), 'push', false)(J$.T(4583985, 'underline', 21, false));
                                if (J$.C(252912, J$.G(4584009, J$.R(4584001, 'style', style, false, false), 'inverse')))
                                    J$.M(4584033, J$.R(4584017, 'settings', settings, false, false), 'push', false)(J$.T(4584025, 'inverse', 21, false));
                                if (J$.C(252920, J$.G(4584049, J$.R(4584041, 'settings', settings, false, false), 'length')))
                                    output = J$.W(4584089, 'output', J$.B(557010, '+', J$.R(4584081, 'output', output, false, false), J$.M(4584073, J$.R(4584057, 'consoleControl', consoleControl, false, true), 'color', false)(J$.R(4584065, 'settings', settings, false, false))), output, false, false);
                                if (J$.C(252928, J$.G(4584105, J$.R(4584097, 'style', style, false, false), 'beep')))
                                    output = J$.W(4584137, 'output', J$.B(557018, '+', J$.R(4584129, 'output', output, false, false), J$.M(4584121, J$.R(4584113, 'consoleControl', consoleControl, false, true), 'beep', false)()), output, false, false);
                            }
                            output = J$.W(4584161, 'output', J$.B(557026, '+', J$.R(4584153, 'output', output, false, false), J$.R(4584145, 'msg', msg, false, false)), output, false, false);
                            if (J$.C(252944, J$.M(4584177, J$.R(4584169, 'this', this, false, false), 'useColor', false)())) {
                                output = J$.W(4584217, 'output', J$.B(557034, '+', J$.R(4584209, 'output', output, false, false), J$.M(4584201, J$.R(4584185, 'consoleControl', consoleControl, false, true), 'color', false)(J$.T(4584193, 'reset', 21, false))), output, false, false);
                            }
                            return J$.Rt(4584233, J$.R(4584225, 'output', output, false, false));
                        } catch (J$e) {
                            J$.Ex(4586225, J$e);
                        } finally {
                            if (J$.Fr(4586233))
                                continue jalangiLabel27;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4584417, J$.R(4584305, 'log', log, false, true), 'write', J$.T(4584409, function (msg, style) {
                jalangiLabel28:
                    while (true) {
                        try {
                            J$.Fe(4584377, arguments.callee, this, arguments);
                            arguments = J$.N(4584385, 'arguments', arguments, true, false, false);
                            msg = J$.N(4584393, 'msg', msg, true, false, false);
                            style = J$.N(4584401, 'style', style, true, false, false);
                            if (J$.C(252952, J$.U(557042, '!', J$.R(4584313, 'stream', stream, false, true))))
                                return J$.Rt(4584321, undefined);
                            J$.M(4584369, J$.R(4584329, 'stream', stream, false, true), 'write', false)(J$.M(4584361, J$.R(4584337, 'this', this, false, false), '_format', false)(J$.R(4584345, 'msg', msg, false, false), J$.R(4584353, 'style', style, false, false)));
                        } catch (J$e) {
                            J$.Ex(4586241, J$e);
                        } finally {
                            if (J$.Fr(4586249))
                                continue jalangiLabel28;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4584985, J$.R(4584425, 'log', log, false, true), 'addLevel', J$.T(4584977, function (lvl, n, style, disp) {
                jalangiLabel30:
                    while (true) {
                        try {
                            J$.Fe(4584929, arguments.callee, this, arguments);
                            arguments = J$.N(4584937, 'arguments', arguments, true, false, false);
                            lvl = J$.N(4584945, 'lvl', lvl, true, false, false);
                            n = J$.N(4584953, 'n', n, true, false, false);
                            style = J$.N(4584961, 'style', style, true, false, false);
                            disp = J$.N(4584969, 'disp', disp, true, false, false);
                            if (J$.C(252960, J$.B(557050, '==', J$.R(4584433, 'disp', disp, false, false), J$.T(4584441, null, 25, false))))
                                disp = J$.W(4584457, 'disp', J$.R(4584449, 'lvl', lvl, false, false), disp, false, false);
                            J$.P(4584497, J$.G(4584473, J$.R(4584465, 'this', this, false, false), 'levels'), J$.R(4584481, 'lvl', lvl, false, false), J$.R(4584489, 'n', n, false, false));
                            J$.P(4584537, J$.G(4584513, J$.R(4584505, 'this', this, false, false), 'style'), J$.R(4584521, 'lvl', lvl, false, false), J$.R(4584529, 'style', style, false, false));
                            if (J$.C(252976, J$.U(557058, '!', J$.G(4584561, J$.R(4584545, 'this', this, false, false), J$.R(4584553, 'lvl', lvl, false, false))))) {
                                J$.P(4584881, J$.R(4584569, 'this', this, false, false), J$.R(4584577, 'lvl', lvl, false, false), J$.M(4584873, J$.T(4584857, function () {
                                    jalangiLabel29:
                                        while (true) {
                                            try {
                                                J$.Fe(4584825, arguments.callee, this, arguments);
                                                arguments = J$.N(4584833, 'arguments', arguments, true, false, false);
                                                J$.N(4584841, 'a', a, false, false, false);
                                                J$.N(4584849, 'i', i, false, false, false);
                                                var a = J$.W(4584625, 'a', J$.F(4584617, J$.I(typeof Array === 'undefined' ? Array = J$.R(4584585, 'Array', undefined, true, true) : Array = J$.R(4584585, 'Array', Array, true, true)), true)(J$.B(557066, '+', J$.G(4584601, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4584593, 'arguments', undefined, true, true) : arguments = J$.R(4584593, 'arguments', arguments, true, true)), 'length'), J$.T(4584609, 1, 22, false))), a, false, false);
                                                J$.P(4584657, J$.R(4584633, 'a', a, false, false), J$.T(4584641, 0, 22, false), J$.R(4584649, 'lvl', lvl, false, false));
                                                for (var i = J$.W(4584673, 'i', J$.T(4584665, 0, 22, false), i, false, false); J$.C(252968, J$.B(557074, '<', J$.R(4584681, 'i', i, false, false), J$.G(4584697, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4584689, 'arguments', undefined, true, true) : arguments = J$.R(4584689, 'arguments', arguments, true, true)), 'length'))); J$.B(557098, '-', i = J$.W(4584713, 'i', J$.B(557090, '+', J$.U(557082, '+', J$.R(4584705, 'i', i, false, false)), 1), i, false, false), 1)) {
                                                    J$.P(4584769, J$.R(4584721, 'a', a, false, false), J$.B(557106, '+', J$.R(4584729, 'i', i, false, false), J$.T(4584737, 1, 22, false)), J$.G(4584761, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4584745, 'arguments', undefined, true, true) : arguments = J$.R(4584745, 'arguments', arguments, true, true)), J$.R(4584753, 'i', i, false, false)));
                                                }
                                                return J$.Rt(4584817, J$.M(4584809, J$.G(4584785, J$.R(4584777, 'this', this, false, false), 'log'), 'apply', false)(J$.R(4584793, 'this', this, false, false), J$.R(4584801, 'a', a, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(4586257, J$e);
                                            } finally {
                                                if (J$.Fr(4586265))
                                                    continue jalangiLabel29;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), 'bind', false)(J$.R(4584865, 'this', this, false, false)));
                            }
                            J$.P(4584921, J$.G(4584897, J$.R(4584889, 'this', this, false, false), 'disp'), J$.R(4584905, 'lvl', lvl, false, false), J$.R(4584913, 'disp', disp, false, false));
                        } catch (J$e) {
                            J$.Ex(4586273, J$e);
                        } finally {
                            if (J$.Fr(4586281))
                                continue jalangiLabel30;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4585017, J$.R(4584993, 'log', log, false, true), 'prefixStyle', J$.T(4585009, { fg: J$.T(4585001, 'magenta', 21, false) }, 11, false));
            J$.P(4585057, J$.R(4585025, 'log', log, false, true), 'headingStyle', J$.T(4585049, {
                fg: J$.T(4585033, 'white', 21, false),
                bg: J$.T(4585041, 'black', 21, false)
            }, 11, false));
            J$.P(4585081, J$.R(4585065, 'log', log, false, true), 'style', J$.T(4585073, {}, 11, false));
            J$.P(4585105, J$.R(4585089, 'log', log, false, true), 'levels', J$.T(4585097, {}, 11, false));
            J$.P(4585129, J$.R(4585113, 'log', log, false, true), 'disp', J$.T(4585121, {}, 11, false));
            J$.M(4585185, J$.R(4585137, 'log', log, false, true), 'addLevel', false)(J$.T(4585145, 'silly', 21, false), J$.U(557114, '-', J$.T(4585153, Infinity, 22, false)), J$.T(4585169, { inverse: J$.T(4585161, true, 23, false) }, 11, false), J$.T(4585177, 'sill', 21, false));
            J$.M(4585249, J$.R(4585193, 'log', log, false, true), 'addLevel', false)(J$.T(4585201, 'verbose', 21, false), J$.T(4585209, 1000, 22, false), J$.T(4585233, {
                fg: J$.T(4585217, 'blue', 21, false),
                bg: J$.T(4585225, 'black', 21, false)
            }, 11, false), J$.T(4585241, 'verb', 21, false));
            J$.M(4585297, J$.R(4585257, 'log', log, false, true), 'addLevel', false)(J$.T(4585265, 'info', 21, false), J$.T(4585273, 2000, 22, false), J$.T(4585289, { fg: J$.T(4585281, 'green', 21, false) }, 11, false));
            J$.M(4585353, J$.R(4585305, 'log', log, false, true), 'addLevel', false)(J$.T(4585313, 'timing', 21, false), J$.T(4585321, 2500, 22, false), J$.T(4585345, {
                fg: J$.T(4585329, 'green', 21, false),
                bg: J$.T(4585337, 'black', 21, false)
            }, 11, false));
            J$.M(4585409, J$.R(4585361, 'log', log, false, true), 'addLevel', false)(J$.T(4585369, 'http', 21, false), J$.T(4585377, 3000, 22, false), J$.T(4585401, {
                fg: J$.T(4585385, 'green', 21, false),
                bg: J$.T(4585393, 'black', 21, false)
            }, 11, false));
            J$.M(4585465, J$.R(4585417, 'log', log, false, true), 'addLevel', false)(J$.T(4585425, 'notice', 21, false), J$.T(4585433, 3500, 22, false), J$.T(4585457, {
                fg: J$.T(4585441, 'blue', 21, false),
                bg: J$.T(4585449, 'black', 21, false)
            }, 11, false));
            J$.M(4585529, J$.R(4585473, 'log', log, false, true), 'addLevel', false)(J$.T(4585481, 'warn', 21, false), J$.T(4585489, 4000, 22, false), J$.T(4585513, {
                fg: J$.T(4585497, 'black', 21, false),
                bg: J$.T(4585505, 'yellow', 21, false)
            }, 11, false), J$.T(4585521, 'WARN', 21, false));
            J$.M(4585593, J$.R(4585537, 'log', log, false, true), 'addLevel', false)(J$.T(4585545, 'error', 21, false), J$.T(4585553, 5000, 22, false), J$.T(4585577, {
                fg: J$.T(4585561, 'red', 21, false),
                bg: J$.T(4585569, 'black', 21, false)
            }, 11, false), J$.T(4585585, 'ERR!', 21, false));
            J$.M(4585625, J$.R(4585601, 'log', log, false, true), 'addLevel', false)(J$.T(4585609, 'silent', 21, false), J$.T(4585617, Infinity, 22, false));
            J$.M(4585673, J$.R(4585633, 'log', log, false, true), 'on', false)(J$.T(4585641, 'error', 21, false), J$.T(4585665, function () {
                jalangiLabel31:
                    while (true) {
                        try {
                            J$.Fe(4585649, arguments.callee, this, arguments);
                            arguments = J$.N(4585657, 'arguments', arguments, true, false, false);
                        } catch (J$e) {
                            J$.Ex(4586289, J$e);
                        } finally {
                            if (J$.Fr(4586297))
                                continue jalangiLabel31;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(4586305, J$e);
        } finally {
            if (J$.Sr(4586313))
                continue jalangiLabel32;
            else
                break jalangiLabel32;
        }
    }
// JALANGI DO NOT INSTRUMENT
