// extend/index.js
J$.noInstrEval = false;
jalangiLabel5:
    while (true) {
        try {
            J$.Se(7358689, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/extend/index.js');
            J$.N(7358697, 'hasOwn', hasOwn, false, false, false);
            J$.N(7358705, 'toStr', toStr, false, false, false);
            J$.N(7358713, 'defineProperty', defineProperty, false, false, false);
            J$.N(7358721, 'gOPD', gOPD, false, false, false);
            J$.N(7358729, 'isArray', isArray, false, false, false);
            J$.N(7358737, 'isPlainObject', isPlainObject, false, false, false);
            J$.N(7358745, 'setProperty', setProperty, false, false, false);
            J$.N(7358753, 'getProperty', getProperty, false, false, false);
            J$.T(7356705, 'use strict', 21, false);
            var hasOwn = J$.W(7356737, 'hasOwn', J$.G(7356729, J$.G(7356721, J$.I(typeof Object === 'undefined' ? Object = J$.R(7356713, 'Object', undefined, true, true) : Object = J$.R(7356713, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), hasOwn, false, true);
            var toStr = J$.W(7356769, 'toStr', J$.G(7356761, J$.G(7356753, J$.I(typeof Object === 'undefined' ? Object = J$.R(7356745, 'Object', undefined, true, true) : Object = J$.R(7356745, 'Object', Object, true, true)), 'prototype'), 'toString'), toStr, false, true);
            var defineProperty = J$.W(7356793, 'defineProperty', J$.G(7356785, J$.I(typeof Object === 'undefined' ? Object = J$.R(7356777, 'Object', undefined, true, true) : Object = J$.R(7356777, 'Object', Object, true, true)), 'defineProperty'), defineProperty, false, true);
            var gOPD = J$.W(7356817, 'gOPD', J$.G(7356809, J$.I(typeof Object === 'undefined' ? Object = J$.R(7356801, 'Object', undefined, true, true) : Object = J$.R(7356801, 'Object', Object, true, true)), 'getOwnPropertyDescriptor'), gOPD, false, true);
            var isArray = J$.W(7356953, 'isArray', J$.T(7356945, function isArray(arr) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(7356921, arguments.callee, this, arguments);
                            arguments = J$.N(7356929, 'arguments', arguments, true, false, false);
                            arr = J$.N(7356937, 'arr', arr, true, false, false);
                            if (J$.C(391608, J$.B(736490, '===', J$.U(736482, 'typeof', J$.G(7356833, J$.I(typeof Array === 'undefined' ? Array = J$.R(7356825, 'Array', undefined, true, true) : Array = J$.R(7356825, 'Array', Array, true, true)), 'isArray')), J$.T(7356841, 'function', 21, false)))) {
                                return J$.Rt(7356873, J$.M(7356865, J$.I(typeof Array === 'undefined' ? Array = J$.R(7356849, 'Array', undefined, true, true) : Array = J$.R(7356849, 'Array', Array, true, true)), 'isArray', false)(J$.R(7356857, 'arr', arr, false, false)));
                            }
                            return J$.Rt(7356913, J$.B(736498, '===', J$.M(7356897, J$.R(7356881, 'toStr', toStr, false, true), 'call', false)(J$.R(7356889, 'arr', arr, false, false)), J$.T(7356905, '[object Array]', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7358761, J$e);
                        } finally {
                            if (J$.Fr(7358769))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), isArray, false, true);
            var isPlainObject = J$.W(7357337, 'isPlainObject', J$.T(7357329, function isPlainObject(obj) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7357281, arguments.callee, this, arguments);
                            arguments = J$.N(7357289, 'arguments', arguments, true, false, false);
                            obj = J$.N(7357297, 'obj', obj, true, false, false);
                            J$.N(7357305, 'hasOwnConstructor', hasOwnConstructor, false, false, false);
                            J$.N(7357313, 'hasIsPrototypeOf', hasIsPrototypeOf, false, false, false);
                            J$.N(7357321, 'key', key, false, false, false);
                            if (J$.C(391624, J$.C(391616, J$.U(736506, '!', J$.R(7356961, 'obj', obj, false, false))) ? J$._() : J$.B(736514, '!==', J$.M(7356985, J$.R(7356969, 'toStr', toStr, false, true), 'call', false)(J$.R(7356977, 'obj', obj, false, false)), J$.T(7356993, '[object Object]', 21, false)))) {
                                return J$.Rt(7357009, J$.T(7357001, false, 23, false));
                            }
                            var hasOwnConstructor = J$.W(7357049, 'hasOwnConstructor', J$.M(7357041, J$.R(7357017, 'hasOwn', hasOwn, false, true), 'call', false)(J$.R(7357025, 'obj', obj, false, false), J$.T(7357033, 'constructor', 21, false)), hasOwnConstructor, false, false);
                            var hasIsPrototypeOf = J$.W(7357145, 'hasIsPrototypeOf', J$.C(391640, J$.C(391632, J$.G(7357065, J$.R(7357057, 'obj', obj, false, false), 'constructor')) ? J$.G(7357089, J$.G(7357081, J$.R(7357073, 'obj', obj, false, false), 'constructor'), 'prototype') : J$._()) ? J$.M(7357137, J$.R(7357097, 'hasOwn', hasOwn, false, true), 'call', false)(J$.G(7357121, J$.G(7357113, J$.R(7357105, 'obj', obj, false, false), 'constructor'), 'prototype'), J$.T(7357129, 'isPrototypeOf', 21, false)) : J$._(), hasIsPrototypeOf, false, false);
                            if (J$.C(391664, J$.C(391656, J$.C(391648, J$.G(7357161, J$.R(7357153, 'obj', obj, false, false), 'constructor')) ? J$.U(736522, '!', J$.R(7357169, 'hasOwnConstructor', hasOwnConstructor, false, false)) : J$._()) ? J$.U(736530, '!', J$.R(7357177, 'hasIsPrototypeOf', hasIsPrototypeOf, false, false)) : J$._())) {
                                return J$.Rt(7357193, J$.T(7357185, false, 23, false));
                            }
                            var key;
                            for (key in J$.H(7357209, J$.R(7357201, 'obj', obj, false, false))) {
                                J$.N(7357217, 'key', key, false, true, false);
                                {
                                    {
                                    }
                                }
                            }
                            return J$.Rt(7357273, J$.C(391672, J$.B(736546, '===', J$.U(736538, 'typeof', J$.R(7357225, 'key', key, false, false)), J$.T(7357233, 'undefined', 21, false))) ? J$._() : J$.M(7357265, J$.R(7357241, 'hasOwn', hasOwn, false, true), 'call', false)(J$.R(7357249, 'obj', obj, false, false), J$.R(7357257, 'key', key, false, false)));
                        } catch (J$e) {
                            J$.Ex(7358777, J$e);
                        } finally {
                            if (J$.Fr(7358785))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), isPlainObject, false, true);
            var setProperty = J$.W(7357553, 'setProperty', J$.T(7357545, function setProperty(target, options) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(7357513, arguments.callee, this, arguments);
                            arguments = J$.N(7357521, 'arguments', arguments, true, false, false);
                            target = J$.N(7357529, 'target', target, true, false, false);
                            options = J$.N(7357537, 'options', options, true, false, false);
                            if (J$.C(391688, J$.C(391680, J$.R(7357345, 'defineProperty', defineProperty, false, true)) ? J$.B(736554, '===', J$.G(7357361, J$.R(7357353, 'options', options, false, false), 'name'), J$.T(7357369, '__proto__', 21, false)) : J$._())) {
                                J$.F(7357457, J$.R(7357377, 'defineProperty', defineProperty, false, true), false)(J$.R(7357385, 'target', target, false, false), J$.G(7357401, J$.R(7357393, 'options', options, false, false), 'name'), J$.T(7357449, {
                                    enumerable: J$.T(7357409, true, 23, false),
                                    configurable: J$.T(7357417, true, 23, false),
                                    value: J$.G(7357433, J$.R(7357425, 'options', options, false, false), 'newValue'),
                                    writable: J$.T(7357441, true, 23, false)
                                }, 11, false));
                            } else {
                                J$.P(7357505, J$.R(7357465, 'target', target, false, false), J$.G(7357481, J$.R(7357473, 'options', options, false, false), 'name'), J$.G(7357497, J$.R(7357489, 'options', options, false, false), 'newValue'));
                            }
                        } catch (J$e) {
                            J$.Ex(7358793, J$e);
                        } finally {
                            if (J$.Fr(7358801))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), setProperty, false, true);
            var getProperty = J$.W(7357753, 'getProperty', J$.T(7357745, function getProperty(obj, name) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(7357713, arguments.callee, this, arguments);
                            arguments = J$.N(7357721, 'arguments', arguments, true, false, false);
                            obj = J$.N(7357729, 'obj', obj, true, false, false);
                            name = J$.N(7357737, 'name', name, true, false, false);
                            if (J$.C(391712, J$.B(736562, '===', J$.R(7357561, 'name', name, false, false), J$.T(7357569, '__proto__', 21, false)))) {
                                if (J$.C(391704, J$.U(736570, '!', J$.M(7357601, J$.R(7357577, 'hasOwn', hasOwn, false, true), 'call', false)(J$.R(7357585, 'obj', obj, false, false), J$.R(7357593, 'name', name, false, false))))) {
                                    return J$.Rt(7357617, void J$.T(7357609, 0, 22, false));
                                } else if (J$.C(391696, J$.R(7357625, 'gOPD', gOPD, false, true))) {
                                    return J$.Rt(7357673, J$.G(7357665, J$.F(7357657, J$.R(7357633, 'gOPD', gOPD, false, true), false)(J$.R(7357641, 'obj', obj, false, false), J$.R(7357649, 'name', name, false, false)), 'value'));
                                }
                            }
                            return J$.Rt(7357705, J$.G(7357697, J$.R(7357681, 'obj', obj, false, false), J$.R(7357689, 'name', name, false, false)));
                        } catch (J$e) {
                            J$.Ex(7358809, J$e);
                        } finally {
                            if (J$.Fr(7358817))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), getProperty, false, true);
            J$.P(7358681, J$.I(typeof module === 'undefined' ? module = J$.R(7357761, 'module', undefined, true, true) : module = J$.R(7357761, 'module', module, true, true)), 'exports', J$.T(7358673, function extend() {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7358577, arguments.callee, this, arguments);
                            arguments = J$.N(7358585, 'arguments', arguments, true, false, false);
                            J$.N(7358593, 'options', options, false, false, false);
                            J$.N(7358601, 'name', name, false, false, false);
                            J$.N(7358609, 'src', src, false, false, false);
                            J$.N(7358617, 'copy', copy, false, false, false);
                            J$.N(7358625, 'copyIsArray', copyIsArray, false, false, false);
                            J$.N(7358633, 'clone', clone, false, false, false);
                            J$.N(7358641, 'target', target, false, false, false);
                            J$.N(7358649, 'i', i, false, false, false);
                            J$.N(7358657, 'length', length, false, false, false);
                            J$.N(7358665, 'deep', deep, false, false, false);
                            var options, name, src, copy, copyIsArray, clone;
                            var target = J$.W(7357793, 'target', J$.G(7357785, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7357769, 'arguments', undefined, true, true) : arguments = J$.R(7357769, 'arguments', arguments, true, true)), J$.T(7357777, 0, 22, false)), target, false, false);
                            var i = J$.W(7357809, 'i', J$.T(7357801, 1, 22, false), i, false, false);
                            var length = J$.W(7357833, 'length', J$.G(7357825, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7357817, 'arguments', undefined, true, true) : arguments = J$.R(7357817, 'arguments', arguments, true, true)), 'length'), length, false, false);
                            var deep = J$.W(7357849, 'deep', J$.T(7357841, false, 23, false), deep, false, false);
                            if (J$.C(391728, J$.B(736586, '===', J$.U(736578, 'typeof', J$.R(7357857, 'target', target, false, false)), J$.T(7357865, 'boolean', 21, false)))) {
                                deep = J$.W(7357881, 'deep', J$.R(7357873, 'target', target, false, false), deep, false, false);
                                target = J$.W(7357921, 'target', J$.C(391720, J$.G(7357905, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7357889, 'arguments', undefined, true, true) : arguments = J$.R(7357889, 'arguments', arguments, true, true)), J$.T(7357897, 1, 22, false))) ? J$._() : J$.T(7357913, {}, 11, false), target, false, false);
                                i = J$.W(7357937, 'i', J$.T(7357929, 2, 22, false), i, false, false);
                            }
                            if (J$.C(391752, J$.C(391744, J$.B(736594, '==', J$.R(7357945, 'target', target, false, false), J$.T(7357953, null, 25, false))) ? J$._() : J$.C(391736, J$.B(736610, '!==', J$.U(736602, 'typeof', J$.R(7357961, 'target', target, false, false)), J$.T(7357969, 'object', 21, false))) ? J$.B(736626, '!==', J$.U(736618, 'typeof', J$.R(7357977, 'target', target, false, false)), J$.T(7357985, 'function', 21, false)) : J$._())) {
                                target = J$.W(7358001, 'target', J$.T(7357993, {}, 11, false), target, false, false);
                            }
                            for (; J$.C(391856, J$.B(736634, '<', J$.R(7358009, 'i', i, false, false), J$.R(7358017, 'length', length, false, false))); i = J$.W(7358033, 'i', J$.B(736650, '+', J$.U(736642, '+', J$.R(7358025, 'i', i, false, false)), 1), i, false, false)) {
                                options = J$.W(7358065, 'options', J$.G(7358057, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7358041, 'arguments', undefined, true, true) : arguments = J$.R(7358041, 'arguments', arguments, true, true)), J$.R(7358049, 'i', i, false, false)), options, false, false);
                                if (J$.C(391848, J$.B(736658, '!=', J$.R(7358073, 'options', options, false, false), J$.T(7358081, null, 25, false)))) {
                                    for (name in J$.H(7358545, J$.R(7358089, 'options', options, false, false))) {
                                        J$.N(7358553, 'name', name, false, true, false);
                                        {
                                            {
                                                src = J$.W(7358129, 'src', J$.F(7358121, J$.R(7358097, 'getProperty', getProperty, false, true), false)(J$.R(7358105, 'target', target, false, false), J$.R(7358113, 'name', name, false, false)), src, false, false);
                                                copy = J$.W(7358169, 'copy', J$.F(7358161, J$.R(7358137, 'getProperty', getProperty, false, true), false)(J$.R(7358145, 'options', options, false, false), J$.R(7358153, 'name', name, false, false)), copy, false, false);
                                                if (J$.C(391840, J$.B(736666, '!==', J$.R(7358177, 'target', target, false, false), J$.R(7358185, 'copy', copy, false, false)))) {
                                                    if (J$.C(391832, J$.C(391776, J$.C(391760, J$.R(7358193, 'deep', deep, false, false)) ? J$.R(7358201, 'copy', copy, false, false) : J$._()) ? J$.C(391768, J$.F(7358225, J$.R(7358209, 'isPlainObject', isPlainObject, false, true), false)(J$.R(7358217, 'copy', copy, false, false))) ? J$._() : copyIsArray = J$.W(7358257, 'copyIsArray', J$.F(7358249, J$.R(7358233, 'isArray', isArray, false, true), false)(J$.R(7358241, 'copy', copy, false, false)), copyIsArray, false, false) : J$._())) {
                                                        if (J$.C(391816, J$.R(7358265, 'copyIsArray', copyIsArray, false, false))) {
                                                            copyIsArray = J$.W(7358281, 'copyIsArray', J$.T(7358273, false, 23, false), copyIsArray, false, false);
                                                            clone = J$.W(7358337, 'clone', J$.C(391792, J$.C(391784, J$.R(7358289, 'src', src, false, false)) ? J$.F(7358313, J$.R(7358297, 'isArray', isArray, false, true), false)(J$.R(7358305, 'src', src, false, false)) : J$._()) ? J$.R(7358321, 'src', src, false, false) : J$.T(7358329, [], 10, false), clone, false, false);
                                                        } else {
                                                            clone = J$.W(7358393, 'clone', J$.C(391808, J$.C(391800, J$.R(7358345, 'src', src, false, false)) ? J$.F(7358369, J$.R(7358353, 'isPlainObject', isPlainObject, false, true), false)(J$.R(7358361, 'src', src, false, false)) : J$._()) ? J$.R(7358377, 'src', src, false, false) : J$.T(7358385, {}, 11, false), clone, false, false);
                                                        }
                                                        J$.F(7358473, J$.R(7358401, 'setProperty', setProperty, false, true), false)(J$.R(7358409, 'target', target, false, false), J$.T(7358465, {
                                                            name: J$.R(7358417, 'name', name, false, false),
                                                            newValue: J$.F(7358457, J$.R(7358425, 'extend', extend, false, false), false)(J$.R(7358433, 'deep', deep, false, false), J$.R(7358441, 'clone', clone, false, false), J$.R(7358449, 'copy', copy, false, false))
                                                        }, 11, false));
                                                    } else if (J$.C(391824, J$.B(736682, '!==', J$.U(736674, 'typeof', J$.R(7358481, 'copy', copy, false, false)), J$.T(7358489, 'undefined', 21, false)))) {
                                                        J$.F(7358537, J$.R(7358497, 'setProperty', setProperty, false, true), false)(J$.R(7358505, 'target', target, false, false), J$.T(7358529, {
                                                            name: J$.R(7358513, 'name', name, false, false),
                                                            newValue: J$.R(7358521, 'copy', copy, false, false)
                                                        }, 11, false));
                                                    }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                            return J$.Rt(7358569, J$.R(7358561, 'target', target, false, false));
                        } catch (J$e) {
                            J$.Ex(7358825, J$e);
                        } finally {
                            if (J$.Fr(7358833))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(7358841, J$e);
        } finally {
            if (J$.Sr(7358849))
                continue jalangiLabel5;
            else
                break jalangiLabel5;
        }
    }
// JALANGI DO NOT INSTRUMENT
