// mongodb/lib/core/auth/auth_provider.js
J$.noInstrEval = false;
jalangiLabel17:
    while (true) {
        try {
            J$.Se(625809, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/auth/auth_provider.js');
            function _classCallCheck(instance, Constructor) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(623329, arguments.callee, this, arguments);
                            arguments = J$.N(623337, 'arguments', arguments, true, false, false);
                            instance = J$.N(623345, 'instance', instance, true, false, false);
                            Constructor = J$.N(623353, 'Constructor', Constructor, true, false, false);
                            if (J$.C(35848, J$.U(79002, '!', J$.B(78994, 'instanceof', J$.R(623289, 'instance', instance, false, false), J$.R(623297, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(623321, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(623305, 'TypeError', undefined, true, true) : TypeError = J$.R(623305, 'TypeError', TypeError, true, true)), true)(J$.T(623313, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(625881, J$e);
                        } finally {
                            if (J$.Fr(625889))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(623601, arguments.callee, this, arguments);
                            arguments = J$.N(623609, 'arguments', arguments, true, false, false);
                            target = J$.N(623617, 'target', target, true, false, false);
                            props = J$.N(623625, 'props', props, true, false, false);
                            J$.N(623633, 'i', i, false, false, false);
                            J$.N(623641, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(623369, 'i', J$.T(623361, 0, 22, false), i, false, false); J$.C(35872, J$.B(79010, '<', J$.R(623377, 'i', i, false, false), J$.G(623393, J$.R(623385, 'props', props, false, false), 'length'))); J$.B(79034, '-', i = J$.W(623409, 'i', J$.B(79026, '+', J$.U(79018, '+', J$.R(623401, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(623441, 'descriptor', J$.G(623433, J$.R(623417, 'props', props, false, false), J$.R(623425, 'i', i, false, false)), descriptor, false, false);
                                J$.P(623481, J$.R(623449, 'descriptor', descriptor, false, false), 'enumerable', J$.C(35856, J$.G(623465, J$.R(623457, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(623473, false, 23, false));
                                J$.P(623505, J$.R(623489, 'descriptor', descriptor, false, false), 'configurable', J$.T(623497, true, 23, false));
                                if (J$.C(35864, J$.B(79042, 'in', J$.T(623513, 'value', 21, false), J$.R(623521, 'descriptor', descriptor, false, false))))
                                    J$.P(623545, J$.R(623529, 'descriptor', descriptor, false, false), 'writable', J$.T(623537, true, 23, false));
                                J$.M(623593, J$.I(typeof Object === 'undefined' ? Object = J$.R(623553, 'Object', undefined, true, true) : Object = J$.R(623553, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(623561, 'target', target, false, false), J$.G(623577, J$.R(623569, 'descriptor', descriptor, false, false), 'key'), J$.R(623585, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(625897, J$e);
                        } finally {
                            if (J$.Fr(625905))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(623753, arguments.callee, this, arguments);
                            arguments = J$.N(623761, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(623769, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(623777, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(623785, 'staticProps', staticProps, true, false, false);
                            if (J$.C(35880, J$.R(623649, 'protoProps', protoProps, false, false)))
                                J$.F(623689, J$.R(623657, '_defineProperties', _defineProperties, false, true), false)(J$.G(623673, J$.R(623665, 'Constructor', Constructor, false, false), 'prototype'), J$.R(623681, 'protoProps', protoProps, false, false));
                            if (J$.C(35888, J$.R(623697, 'staticProps', staticProps, false, false)))
                                J$.F(623729, J$.R(623705, '_defineProperties', _defineProperties, false, true), false)(J$.R(623713, 'Constructor', Constructor, false, false), J$.R(623721, 'staticProps', staticProps, false, false));
                            return J$.Rt(623745, J$.R(623737, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(625913, J$e);
                        } finally {
                            if (J$.Fr(625921))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _classCallCheck = J$.N(625825, '_classCallCheck', J$.T(625817, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(625841, '_defineProperties', J$.T(625833, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(625857, '_createClass', J$.T(625849, _createClass, 12, false), true, false, false);
            J$.N(625865, 'MongoError', MongoError, false, false, false);
            J$.N(625873, 'AuthProvider', AuthProvider, false, false, false);
            J$.T(623281, 'use strict', 21, false);
            var MongoError = J$.W(623825, 'MongoError', J$.G(623817, J$.F(623809, J$.I(typeof require === 'undefined' ? require = J$.R(623793, 'require', undefined, true, true) : require = J$.R(623793, 'require', require, true, true)), false)(J$.T(623801, '../error', 21, false)), 'MongoError'), MongoError, false, true);
            var AuthProvider = J$.W(625769, 'AuthProvider', J$.F(625761, J$.T(625753, function () {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(625721, arguments.callee, this, arguments);
                            function AuthProvider(bson) {
                                jalangiLabel3:
                                    while (true) {
                                        try {
                                            J$.Fe(623913, arguments.callee, this, arguments);
                                            arguments = J$.N(623921, 'arguments', arguments, true, false, false);
                                            bson = J$.N(623929, 'bson', bson, true, false, false);
                                            J$.F(623857, J$.R(623833, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(623841, 'this', this, false, false), J$.R(623849, 'AuthProvider', AuthProvider, false, false));
                                            J$.P(623881, J$.R(623865, 'this', this, false, false), 'bson', J$.R(623873, 'bson', bson, false, false));
                                            J$.P(623905, J$.R(623889, 'this', this, false, false), 'authStore', J$.T(623897, [], 10, false));
                                        } catch (J$e) {
                                            J$.Ex(625929, J$e);
                                        } finally {
                                            if (J$.Fr(625937))
                                                continue jalangiLabel3;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(625729, 'arguments', arguments, true, false, false);
                            AuthProvider = J$.N(625745, 'AuthProvider', J$.T(625737, AuthProvider, 12, false), true, false, false);
                            J$.F(625697, J$.R(623937, '_createClass', _createClass, false, true), false)(J$.R(623945, 'AuthProvider', AuthProvider, false, false), J$.T(625689, [
                                J$.T(624849, {
                                    key: J$.T(623953, 'auth', 21, false),
                                    value: J$.T(624841, function auth(sendAuthCommand, connections, credentials, callback) {
                                        jalangiLabel8:
                                            while (true) {
                                                try {
                                                    J$.Fe(624745, arguments.callee, this, arguments);
                                                    arguments = J$.N(624753, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(624761, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connections = J$.N(624769, 'connections', connections, true, false, false);
                                                    credentials = J$.N(624777, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(624785, 'callback', callback, true, false, false);
                                                    J$.N(624793, '_this', _this, false, false, false);
                                                    J$.N(624801, 'count', count, false, false, false);
                                                    J$.N(624809, 'numberOfValidConnections', numberOfValidConnections, false, false, false);
                                                    J$.N(624817, 'errorObject', errorObject, false, false, false);
                                                    J$.N(624825, 'execute', execute, false, false, false);
                                                    J$.N(624833, 'executeInNextTick', executeInNextTick, false, false, false);
                                                    var _this = J$.W(623969, '_this', J$.R(623961, 'this', this, false, false), _this, false, false);
                                                    var count = J$.W(623993, 'count', J$.G(623985, J$.R(623977, 'connections', connections, false, false), 'length'), count, false, false);
                                                    if (J$.C(35896, J$.B(79050, '===', J$.R(624001, 'count', count, false, false), J$.T(624009, 0, 22, false)))) {
                                                        J$.F(624041, J$.R(624017, 'callback', callback, false, false), false)(J$.T(624025, null, 25, false), J$.T(624033, null, 25, false));
                                                        return J$.Rt(624049, undefined);
                                                    }
                                                    var numberOfValidConnections = J$.W(624065, 'numberOfValidConnections', J$.T(624057, 0, 22, false), numberOfValidConnections, false, false);
                                                    var errorObject = J$.W(624081, 'errorObject', J$.T(624073, null, 25, false), errorObject, false, false);
                                                    var execute = J$.W(624561, 'execute', J$.T(624553, function execute(connection) {
                                                        jalangiLabel5:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(624529, arguments.callee, this, arguments);
                                                                    arguments = J$.N(624537, 'arguments', arguments, true, false, false);
                                                                    connection = J$.N(624545, 'connection', connection, true, false, false);
                                                                    J$.M(624521, J$.R(624089, '_this', _this, false, false), '_authenticateSingleConnection', false)(J$.R(624097, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(624105, 'connection', connection, false, false), J$.R(624113, 'credentials', credentials, false, false), J$.T(624513, function (err, r) {
                                                                        jalangiLabel4:
                                                                            while (true) {
                                                                                try {
                                                                                    J$.Fe(624481, arguments.callee, this, arguments);
                                                                                    arguments = J$.N(624489, 'arguments', arguments, true, false, false);
                                                                                    err = J$.N(624497, 'err', err, true, false, false);
                                                                                    r = J$.N(624505, 'r', r, true, false, false);
                                                                                    count = J$.W(624137, 'count', J$.B(79058, '-', J$.R(624121, 'count', count, false, false), J$.T(624129, 1, 22, false)), count, false, false);
                                                                                    if (J$.C(35928, J$.R(624145, 'err', err, false, false))) {
                                                                                        errorObject = J$.W(624177, 'errorObject', J$.F(624169, J$.R(624153, 'MongoError', MongoError, false, true), true)(J$.R(624161, 'err', err, false, false)), errorObject, false, false);
                                                                                    } else if (J$.C(35920, J$.C(35912, J$.R(624185, 'r', r, false, false)) ? J$.C(35904, J$.G(624201, J$.R(624193, 'r', r, false, false), '$err')) ? J$._() : J$.G(624217, J$.R(624209, 'r', r, false, false), 'errmsg') : J$._())) {
                                                                                        errorObject = J$.W(624249, 'errorObject', J$.F(624241, J$.R(624225, 'MongoError', MongoError, false, true), true)(J$.R(624233, 'r', r, false, false)), errorObject, false, false);
                                                                                    } else {
                                                                                        numberOfValidConnections = J$.W(624273, 'numberOfValidConnections', J$.B(79066, '+', J$.R(624257, 'numberOfValidConnections', numberOfValidConnections, false, false), J$.T(624265, 1, 22, false)), numberOfValidConnections, false, false);
                                                                                    }
                                                                                    if (J$.C(35936, J$.B(79074, '!==', J$.R(624281, 'count', count, false, false), J$.T(624289, 0, 22, false)))) {
                                                                                        return J$.Rt(624297, undefined);
                                                                                    }
                                                                                    if (J$.C(35952, J$.B(79082, '>', J$.R(624305, 'numberOfValidConnections', numberOfValidConnections, false, false), J$.T(624313, 0, 22, false)))) {
                                                                                        J$.M(624337, J$.R(624321, '_this', _this, false, false), 'addCredentials', false)(J$.R(624329, 'credentials', credentials, false, false));
                                                                                        J$.F(624369, J$.R(624345, 'callback', callback, false, false), false)(J$.T(624353, null, 25, false), J$.T(624361, true, 23, false));
                                                                                    } else {
                                                                                        if (J$.C(35944, J$.B(79090, '==', J$.R(624377, 'errorObject', errorObject, false, false), J$.T(624385, null, 25, false)))) {
                                                                                            errorObject = J$.W(624441, 'errorObject', J$.F(624433, J$.R(624393, 'MongoError', MongoError, false, true), true)(J$.M(624425, J$.T(624401, 'failed to authenticate using ', 21, false), 'concat', false)(J$.G(624417, J$.R(624409, 'credentials', credentials, false, false), 'mechanism'))), errorObject, false, false);
                                                                                        }
                                                                                        J$.F(624473, J$.R(624449, 'callback', callback, false, false), false)(J$.R(624457, 'errorObject', errorObject, false, false), J$.T(624465, false, 23, false));
                                                                                    }
                                                                                } catch (J$e) {
                                                                                    J$.Ex(625945, J$e);
                                                                                } finally {
                                                                                    if (J$.Fr(625953))
                                                                                        continue jalangiLabel4;
                                                                                    else
                                                                                        return J$.Ra();
                                                                                }
                                                                            }
                                                                    }, 12, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(625961, J$e);
                                                                } finally {
                                                                    if (J$.Fr(625969))
                                                                        continue jalangiLabel5;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false), execute, false, false);
                                                    var executeInNextTick = J$.W(624681, 'executeInNextTick', J$.T(624673, function executeInNextTick(_connection) {
                                                        jalangiLabel7:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(624649, arguments.callee, this, arguments);
                                                                    arguments = J$.N(624657, 'arguments', arguments, true, false, false);
                                                                    _connection = J$.N(624665, '_connection', _connection, true, false, false);
                                                                    return J$.Rt(624641, J$.M(624633, J$.I(typeof process === 'undefined' ? process = J$.R(624569, 'process', undefined, true, true) : process = J$.R(624569, 'process', process, true, true)), 'nextTick', false)(J$.T(624625, function () {
                                                                        jalangiLabel6:
                                                                            while (true) {
                                                                                try {
                                                                                    J$.Fe(624609, arguments.callee, this, arguments);
                                                                                    arguments = J$.N(624617, 'arguments', arguments, true, false, false);
                                                                                    return J$.Rt(624601, J$.F(624593, J$.R(624577, 'execute', execute, false, false), false)(J$.R(624585, '_connection', _connection, false, false)));
                                                                                } catch (J$e) {
                                                                                    J$.Ex(625977, J$e);
                                                                                } finally {
                                                                                    if (J$.Fr(625985))
                                                                                        continue jalangiLabel6;
                                                                                    else
                                                                                        return J$.Ra();
                                                                                }
                                                                            }
                                                                    }, 12, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(625993, J$e);
                                                                } finally {
                                                                    if (J$.Fr(626001))
                                                                        continue jalangiLabel7;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false), executeInNextTick, false, false);
                                                    while (J$.C(35960, J$.B(79098, '>', J$.G(624697, J$.R(624689, 'connections', connections, false, false), 'length'), J$.T(624705, 0, 22, false)))) {
                                                        J$.F(624737, J$.R(624713, 'executeInNextTick', executeInNextTick, false, false), false)(J$.M(624729, J$.R(624721, 'connections', connections, false, false), 'shift', false)());
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(626009, J$e);
                                                } finally {
                                                    if (J$.Fr(626017))
                                                        continue jalangiLabel8;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(624913, {
                                    key: J$.T(624857, '_authenticateSingleConnection', 21, false),
                                    value: J$.T(624905, function _authenticateSingleConnection() {
                                        jalangiLabel9:
                                            while (true) {
                                                try {
                                                    J$.Fe(624889, arguments.callee, this, arguments);
                                                    arguments = J$.N(624897, 'arguments', arguments, true, false, false);
                                                    throw J$.F(624881, J$.I(typeof Error === 'undefined' ? Error = J$.R(624865, 'Error', undefined, true, true) : Error = J$.R(624865, 'Error', Error, true, true)), true)(J$.T(624873, '_authenticateSingleConnection must be overridden', 21, false));
                                                } catch (J$e) {
                                                    J$.Ex(626025, J$e);
                                                } finally {
                                                    if (J$.Fr(626033))
                                                        continue jalangiLabel9;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(625105, {
                                    key: J$.T(624921, 'addCredentials', 21, false),
                                    value: J$.T(625097, function addCredentials(credentials) {
                                        jalangiLabel11:
                                            while (true) {
                                                try {
                                                    J$.Fe(625065, arguments.callee, this, arguments);
                                                    arguments = J$.N(625073, 'arguments', arguments, true, false, false);
                                                    credentials = J$.N(625081, 'credentials', credentials, true, false, false);
                                                    J$.N(625089, 'found', found, false, false, false);
                                                    var found = J$.W(625017, 'found', J$.M(625009, J$.G(624937, J$.R(624929, 'this', this, false, false), 'authStore'), 'some', false)(J$.T(625001, function (cred) {
                                                        jalangiLabel10:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(624977, arguments.callee, this, arguments);
                                                                    arguments = J$.N(624985, 'arguments', arguments, true, false, false);
                                                                    cred = J$.N(624993, 'cred', cred, true, false, false);
                                                                    return J$.Rt(624969, J$.M(624961, J$.R(624945, 'cred', cred, false, false), 'equals', false)(J$.R(624953, 'credentials', credentials, false, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(626041, J$e);
                                                                } finally {
                                                                    if (J$.Fr(626049))
                                                                        continue jalangiLabel10;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false)), found, false, false);
                                                    if (J$.C(35968, J$.U(79106, '!', J$.R(625025, 'found', found, false, false)))) {
                                                        J$.M(625057, J$.G(625041, J$.R(625033, 'this', this, false, false), 'authStore'), 'push', false)(J$.R(625049, 'credentials', credentials, false, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(626057, J$e);
                                                } finally {
                                                    if (J$.Fr(626065))
                                                        continue jalangiLabel11;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(625529, {
                                    key: J$.T(625113, 'reauthenticate', 21, false),
                                    value: J$.T(625521, function reauthenticate(sendAuthCommand, connections, callback) {
                                        jalangiLabel13:
                                            while (true) {
                                                try {
                                                    J$.Fe(625457, arguments.callee, this, arguments);
                                                    arguments = J$.N(625465, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(625473, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connections = J$.N(625481, 'connections', connections, true, false, false);
                                                    callback = J$.N(625489, 'callback', callback, true, false, false);
                                                    J$.N(625497, 'authStore', authStore, false, false, false);
                                                    J$.N(625505, 'count', count, false, false, false);
                                                    J$.N(625513, 'i', i, false, false, false);
                                                    var authStore = J$.W(625153, 'authStore', J$.M(625145, J$.G(625129, J$.R(625121, 'this', this, false, false), 'authStore'), 'slice', false)(J$.T(625137, 0, 22, false)), authStore, false, false);
                                                    var count = J$.W(625177, 'count', J$.G(625169, J$.R(625161, 'authStore', authStore, false, false), 'length'), count, false, false);
                                                    if (J$.C(35976, J$.B(79114, '===', J$.R(625185, 'count', count, false, false), J$.T(625193, 0, 22, false)))) {
                                                        return J$.Rt(625233, J$.F(625225, J$.R(625201, 'callback', callback, false, false), false)(J$.T(625209, null, 25, false), J$.T(625217, null, 25, false)));
                                                    }
                                                    for (var i = J$.W(625249, 'i', J$.T(625241, 0, 22, false), i, false, false); J$.C(35992, J$.B(79122, '<', J$.R(625257, 'i', i, false, false), J$.G(625273, J$.R(625265, 'authStore', authStore, false, false), 'length'))); J$.B(79146, '-', i = J$.W(625289, 'i', J$.B(79138, '+', J$.U(79130, '+', J$.R(625281, 'i', i, false, false)), 1), i, false, false), 1)) {
                                                        J$.M(625449, J$.R(625297, 'this', this, false, false), 'auth', false)(J$.R(625305, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(625313, 'connections', connections, false, false), J$.G(625337, J$.R(625321, 'authStore', authStore, false, false), J$.R(625329, 'i', i, false, false)), J$.T(625441, function (err) {
                                                            jalangiLabel12:
                                                                while (true) {
                                                                    try {
                                                                        J$.Fe(625417, arguments.callee, this, arguments);
                                                                        arguments = J$.N(625425, 'arguments', arguments, true, false, false);
                                                                        err = J$.N(625433, 'err', err, true, false, false);
                                                                        count = J$.W(625361, 'count', J$.B(79154, '-', J$.R(625345, 'count', count, false, false), J$.T(625353, 1, 22, false)), count, false, false);
                                                                        if (J$.C(35984, J$.B(79162, '===', J$.R(625369, 'count', count, false, false), J$.T(625377, 0, 22, false)))) {
                                                                            J$.F(625409, J$.R(625385, 'callback', callback, false, false), false)(J$.R(625393, 'err', err, false, false), J$.T(625401, null, 25, false));
                                                                        }
                                                                    } catch (J$e) {
                                                                        J$.Ex(626073, J$e);
                                                                    } finally {
                                                                        if (J$.Fr(626081))
                                                                            continue jalangiLabel12;
                                                                        else
                                                                            return J$.Ra();
                                                                    }
                                                                }
                                                        }, 12, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(626089, J$e);
                                                } finally {
                                                    if (J$.Fr(626097))
                                                        continue jalangiLabel13;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(625681, {
                                    key: J$.T(625537, 'logout', 21, false),
                                    value: J$.T(625673, function logout(source) {
                                        jalangiLabel15:
                                            while (true) {
                                                try {
                                                    J$.Fe(625649, arguments.callee, this, arguments);
                                                    arguments = J$.N(625657, 'arguments', arguments, true, false, false);
                                                    source = J$.N(625665, 'source', source, true, false, false);
                                                    J$.P(625641, J$.R(625545, 'this', this, false, false), 'authStore', J$.M(625633, J$.G(625561, J$.R(625553, 'this', this, false, false), 'authStore'), 'filter', false)(J$.T(625625, function (credentials) {
                                                        jalangiLabel14:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(625601, arguments.callee, this, arguments);
                                                                    arguments = J$.N(625609, 'arguments', arguments, true, false, false);
                                                                    credentials = J$.N(625617, 'credentials', credentials, true, false, false);
                                                                    return J$.Rt(625593, J$.B(79170, '!==', J$.G(625577, J$.R(625569, 'credentials', credentials, false, false), 'source'), J$.R(625585, 'source', source, false, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(626105, J$e);
                                                                } finally {
                                                                    if (J$.Fr(626113))
                                                                        continue jalangiLabel14;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false)));
                                                } catch (J$e) {
                                                    J$.Ex(626121, J$e);
                                                } finally {
                                                    if (J$.Fr(626129))
                                                        continue jalangiLabel15;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(625713, J$.R(625705, 'AuthProvider', AuthProvider, false, false));
                        } catch (J$e) {
                            J$.Ex(626137, J$e);
                        } finally {
                            if (J$.Fr(626145))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), AuthProvider, false, true);
            J$.P(625801, J$.I(typeof module === 'undefined' ? module = J$.R(625777, 'module', undefined, true, true) : module = J$.R(625777, 'module', module, true, true)), 'exports', J$.T(625793, { AuthProvider: J$.R(625785, 'AuthProvider', AuthProvider, false, true) }, 11, false));
        } catch (J$e) {
            J$.Ex(626153, J$e);
        } finally {
            if (J$.Sr(626161))
                continue jalangiLabel17;
            else
                break jalangiLabel17;
        }
    }
// JALANGI DO NOT INSTRUMENT
