// mongodb/lib/core/auth/defaultAuthProviders.js
J$.noInstrEval = false;
jalangiLabel1:
    while (true) {
        try {
            J$.Se(626657, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/auth/defaultAuthProviders.js');
            function defaultAuthProviders(bson) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(626601, arguments.callee, this, arguments);
                            arguments = J$.N(626609, 'arguments', arguments, true, false, false);
                            bson = J$.N(626617, 'bson', bson, true, false, false);
                            return J$.Rt(626593, J$.T(626585, {
                                mongocr: J$.F(626433, J$.R(626417, 'MongoCR', MongoCR, false, true), true)(J$.R(626425, 'bson', bson, false, false)),
                                x509: J$.F(626457, J$.R(626441, 'X509', X509, false, true), true)(J$.R(626449, 'bson', bson, false, false)),
                                plain: J$.F(626481, J$.R(626465, 'Plain', Plain, false, true), true)(J$.R(626473, 'bson', bson, false, false)),
                                gssapi: J$.F(626505, J$.R(626489, 'GSSAPI', GSSAPI, false, true), true)(J$.R(626497, 'bson', bson, false, false)),
                                sspi: J$.F(626529, J$.R(626513, 'SSPI', SSPI, false, true), true)(J$.R(626521, 'bson', bson, false, false)),
                                'scram-sha-1': J$.F(626553, J$.R(626537, 'ScramSHA1', ScramSHA1, false, true), true)(J$.R(626545, 'bson', bson, false, false)),
                                'scram-sha-256': J$.F(626577, J$.R(626561, 'ScramSHA256', ScramSHA256, false, true), true)(J$.R(626569, 'bson', bson, false, false))
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(626737, J$e);
                        } finally {
                            if (J$.Fr(626745))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(626665, 'MongoCR', MongoCR, false, false, false);
            J$.N(626673, 'X509', X509, false, false, false);
            J$.N(626681, 'Plain', Plain, false, false, false);
            J$.N(626689, 'GSSAPI', GSSAPI, false, false, false);
            J$.N(626697, 'SSPI', SSPI, false, false, false);
            J$.N(626705, 'ScramSHA1', ScramSHA1, false, false, false);
            J$.N(626713, 'ScramSHA256', ScramSHA256, false, false, false);
            defaultAuthProviders = J$.N(626729, 'defaultAuthProviders', J$.T(626721, defaultAuthProviders, 12, false), true, false, false);
            J$.T(626169, 'use strict', 21, false);
            var MongoCR = J$.W(626201, 'MongoCR', J$.F(626193, J$.I(typeof require === 'undefined' ? require = J$.R(626177, 'require', undefined, true, true) : require = J$.R(626177, 'require', require, true, true)), false)(J$.T(626185, './mongocr', 21, false)), MongoCR, false, true);
            var X509 = J$.W(626233, 'X509', J$.F(626225, J$.I(typeof require === 'undefined' ? require = J$.R(626209, 'require', undefined, true, true) : require = J$.R(626209, 'require', require, true, true)), false)(J$.T(626217, './x509', 21, false)), X509, false, true);
            var Plain = J$.W(626265, 'Plain', J$.F(626257, J$.I(typeof require === 'undefined' ? require = J$.R(626241, 'require', undefined, true, true) : require = J$.R(626241, 'require', require, true, true)), false)(J$.T(626249, './plain', 21, false)), Plain, false, true);
            var GSSAPI = J$.W(626297, 'GSSAPI', J$.F(626289, J$.I(typeof require === 'undefined' ? require = J$.R(626273, 'require', undefined, true, true) : require = J$.R(626273, 'require', require, true, true)), false)(J$.T(626281, './gssapi', 21, false)), GSSAPI, false, true);
            var SSPI = J$.W(626329, 'SSPI', J$.F(626321, J$.I(typeof require === 'undefined' ? require = J$.R(626305, 'require', undefined, true, true) : require = J$.R(626305, 'require', require, true, true)), false)(J$.T(626313, './sspi', 21, false)), SSPI, false, true);
            var ScramSHA1 = J$.W(626369, 'ScramSHA1', J$.G(626361, J$.F(626353, J$.I(typeof require === 'undefined' ? require = J$.R(626337, 'require', undefined, true, true) : require = J$.R(626337, 'require', require, true, true)), false)(J$.T(626345, './scram', 21, false)), 'ScramSHA1'), ScramSHA1, false, true);
            var ScramSHA256 = J$.W(626409, 'ScramSHA256', J$.G(626401, J$.F(626393, J$.I(typeof require === 'undefined' ? require = J$.R(626377, 'require', undefined, true, true) : require = J$.R(626377, 'require', require, true, true)), false)(J$.T(626385, './scram', 21, false)), 'ScramSHA256'), ScramSHA256, false, true);
            J$.P(626649, J$.I(typeof module === 'undefined' ? module = J$.R(626625, 'module', undefined, true, true) : module = J$.R(626625, 'module', module, true, true)), 'exports', J$.T(626641, { defaultAuthProviders: J$.R(626633, 'defaultAuthProviders', defaultAuthProviders, false, true) }, 11, false));
        } catch (J$e) {
            J$.Ex(626753, J$e);
        } finally {
            if (J$.Sr(626761))
                continue jalangiLabel1;
            else
                break jalangiLabel1;
        }
    }
// JALANGI DO NOT INSTRUMENT
