// mongodb/lib/core/auth/gssapi.js
J$.noInstrEval = false;
jalangiLabel35:
    while (true) {
        try {
            J$.Se(632297, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/auth/gssapi.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(627041, arguments.callee, this, arguments);
                            arguments = J$.N(627049, 'arguments', arguments, true, false, false);
                            obj = J$.N(627057, 'obj', obj, true, false, false);
                            if (J$.C(36040, J$.C(36000, J$.B(79186, '===', J$.U(79178, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(626777, 'Symbol', undefined, true, true) : Symbol = J$.R(626777, 'Symbol', Symbol, true, true))), J$.T(626785, 'function', 21, false))) ? J$.B(79202, '===', J$.U(79194, 'typeof', J$.G(626801, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(626793, 'Symbol', undefined, true, true) : Symbol = J$.R(626793, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(626809, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(626865, '_typeof2', J$.T(626857, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(626833, arguments.callee, this, arguments);
                                                arguments = J$.N(626841, 'arguments', arguments, true, false, false);
                                                obj = J$.N(626849, 'obj', obj, true, false, false);
                                                return J$.Rt(626825, J$.U(79210, 'typeof', J$.R(626817, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(632561, J$e);
                                            } finally {
                                                if (J$.Fr(632569))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(627001, '_typeof2', J$.T(626993, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(626969, arguments.callee, this, arguments);
                                                arguments = J$.N(626977, 'arguments', arguments, true, false, false);
                                                obj = J$.N(626985, 'obj', obj, true, false, false);
                                                return J$.Rt(626961, J$.C(36032, J$.C(36024, J$.C(36016, J$.C(36008, J$.R(626873, 'obj', obj, false, false)) ? J$.B(79226, '===', J$.U(79218, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(626881, 'Symbol', undefined, true, true) : Symbol = J$.R(626881, 'Symbol', Symbol, true, true))), J$.T(626889, 'function', 21, false)) : J$._()) ? J$.B(79234, '===', J$.G(626905, J$.R(626897, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(626913, 'Symbol', undefined, true, true) : Symbol = J$.R(626913, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(79242, '!==', J$.R(626921, 'obj', obj, false, false), J$.G(626937, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(626929, 'Symbol', undefined, true, true) : Symbol = J$.R(626929, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(626945, 'symbol', 21, false) : J$.U(79250, 'typeof', J$.R(626953, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(632577, J$e);
                                            } finally {
                                                if (J$.Fr(632585))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(627033, J$.F(627025, J$.R(627009, '_typeof2', _typeof2, false, true), false)(J$.R(627017, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(632593, J$e);
                        } finally {
                            if (J$.Fr(632601))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(627377, arguments.callee, this, arguments);
                            arguments = J$.N(627385, 'arguments', arguments, true, false, false);
                            obj = J$.N(627393, 'obj', obj, true, false, false);
                            if (J$.C(36088, J$.C(36048, J$.B(79266, '===', J$.U(79258, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(627065, 'Symbol', undefined, true, true) : Symbol = J$.R(627065, 'Symbol', Symbol, true, true))), J$.T(627073, 'function', 21, false))) ? J$.B(79274, '===', J$.F(627105, J$.R(627081, '_typeof2', _typeof2, false, true), false)(J$.G(627097, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(627089, 'Symbol', undefined, true, true) : Symbol = J$.R(627089, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(627113, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(627185, '_typeof', J$.T(627177, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(627153, arguments.callee, this, arguments);
                                                arguments = J$.N(627161, 'arguments', arguments, true, false, false);
                                                obj = J$.N(627169, 'obj', obj, true, false, false);
                                                return J$.Rt(627145, J$.F(627137, J$.R(627121, '_typeof2', _typeof2, false, true), false)(J$.R(627129, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(632609, J$e);
                                            } finally {
                                                if (J$.Fr(632617))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(627337, '_typeof', J$.T(627329, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(627305, arguments.callee, this, arguments);
                                                arguments = J$.N(627313, 'arguments', arguments, true, false, false);
                                                obj = J$.N(627321, 'obj', obj, true, false, false);
                                                return J$.Rt(627297, J$.C(36080, J$.C(36072, J$.C(36064, J$.C(36056, J$.R(627193, 'obj', obj, false, false)) ? J$.B(79290, '===', J$.U(79282, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(627201, 'Symbol', undefined, true, true) : Symbol = J$.R(627201, 'Symbol', Symbol, true, true))), J$.T(627209, 'function', 21, false)) : J$._()) ? J$.B(79298, '===', J$.G(627225, J$.R(627217, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(627233, 'Symbol', undefined, true, true) : Symbol = J$.R(627233, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(79306, '!==', J$.R(627241, 'obj', obj, false, false), J$.G(627257, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(627249, 'Symbol', undefined, true, true) : Symbol = J$.R(627249, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(627265, 'symbol', 21, false) : J$.F(627289, J$.R(627273, '_typeof2', _typeof2, false, true), false)(J$.R(627281, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(632625, J$e);
                                            } finally {
                                                if (J$.Fr(632633))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(627369, J$.F(627361, J$.R(627345, '_typeof', _typeof, false, true), false)(J$.R(627353, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(632641, J$e);
                        } finally {
                            if (J$.Fr(632649))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(627441, arguments.callee, this, arguments);
                            arguments = J$.N(627449, 'arguments', arguments, true, false, false);
                            instance = J$.N(627457, 'instance', instance, true, false, false);
                            Constructor = J$.N(627465, 'Constructor', Constructor, true, false, false);
                            if (J$.C(36096, J$.U(79322, '!', J$.B(79314, 'instanceof', J$.R(627401, 'instance', instance, false, false), J$.R(627409, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(627433, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(627417, 'TypeError', undefined, true, true) : TypeError = J$.R(627417, 'TypeError', TypeError, true, true)), true)(J$.T(627425, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(632657, J$e);
                        } finally {
                            if (J$.Fr(632665))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(627713, arguments.callee, this, arguments);
                            arguments = J$.N(627721, 'arguments', arguments, true, false, false);
                            target = J$.N(627729, 'target', target, true, false, false);
                            props = J$.N(627737, 'props', props, true, false, false);
                            J$.N(627745, 'i', i, false, false, false);
                            J$.N(627753, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(627481, 'i', J$.T(627473, 0, 22, false), i, false, false); J$.C(36120, J$.B(79330, '<', J$.R(627489, 'i', i, false, false), J$.G(627505, J$.R(627497, 'props', props, false, false), 'length'))); J$.B(79354, '-', i = J$.W(627521, 'i', J$.B(79346, '+', J$.U(79338, '+', J$.R(627513, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(627553, 'descriptor', J$.G(627545, J$.R(627529, 'props', props, false, false), J$.R(627537, 'i', i, false, false)), descriptor, false, false);
                                J$.P(627593, J$.R(627561, 'descriptor', descriptor, false, false), 'enumerable', J$.C(36104, J$.G(627577, J$.R(627569, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(627585, false, 23, false));
                                J$.P(627617, J$.R(627601, 'descriptor', descriptor, false, false), 'configurable', J$.T(627609, true, 23, false));
                                if (J$.C(36112, J$.B(79362, 'in', J$.T(627625, 'value', 21, false), J$.R(627633, 'descriptor', descriptor, false, false))))
                                    J$.P(627657, J$.R(627641, 'descriptor', descriptor, false, false), 'writable', J$.T(627649, true, 23, false));
                                J$.M(627705, J$.I(typeof Object === 'undefined' ? Object = J$.R(627665, 'Object', undefined, true, true) : Object = J$.R(627665, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(627673, 'target', target, false, false), J$.G(627689, J$.R(627681, 'descriptor', descriptor, false, false), 'key'), J$.R(627697, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(632673, J$e);
                        } finally {
                            if (J$.Fr(632681))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(627865, arguments.callee, this, arguments);
                            arguments = J$.N(627873, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(627881, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(627889, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(627897, 'staticProps', staticProps, true, false, false);
                            if (J$.C(36128, J$.R(627761, 'protoProps', protoProps, false, false)))
                                J$.F(627801, J$.R(627769, '_defineProperties', _defineProperties, false, true), false)(J$.G(627785, J$.R(627777, 'Constructor', Constructor, false, false), 'prototype'), J$.R(627793, 'protoProps', protoProps, false, false));
                            if (J$.C(36136, J$.R(627809, 'staticProps', staticProps, false, false)))
                                J$.F(627841, J$.R(627817, '_defineProperties', _defineProperties, false, true), false)(J$.R(627825, 'Constructor', Constructor, false, false), J$.R(627833, 'staticProps', staticProps, false, false));
                            return J$.Rt(627857, J$.R(627849, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(632689, J$e);
                        } finally {
                            if (J$.Fr(632697))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(628009, arguments.callee, this, arguments);
                            arguments = J$.N(628017, 'arguments', arguments, true, false, false);
                            self = J$.N(628025, 'self', self, true, false, false);
                            call = J$.N(628033, 'call', call, true, false, false);
                            if (J$.C(36160, J$.C(36152, J$.R(627905, 'call', call, false, false)) ? J$.C(36144, J$.B(79370, '===', J$.F(627929, J$.R(627913, '_typeof', _typeof, false, true), false)(J$.R(627921, 'call', call, false, false)), J$.T(627937, 'object', 21, false))) ? J$._() : J$.B(79386, '===', J$.U(79378, 'typeof', J$.R(627945, 'call', call, false, false)), J$.T(627953, 'function', 21, false)) : J$._())) {
                                return J$.Rt(627969, J$.R(627961, 'call', call, false, false));
                            }
                            return J$.Rt(628001, J$.F(627993, J$.R(627977, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(627985, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(632705, J$e);
                        } finally {
                            if (J$.Fr(632713))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(628097, arguments.callee, this, arguments);
                            arguments = J$.N(628105, 'arguments', arguments, true, false, false);
                            self = J$.N(628113, 'self', self, true, false, false);
                            if (J$.C(36168, J$.B(79394, '===', J$.R(628041, 'self', self, false, false), void J$.T(628049, 0, 22, false)))) {
                                throw J$.F(628073, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(628057, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(628057, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(628065, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(628089, J$.R(628081, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(632721, J$e);
                        } finally {
                            if (J$.Fr(632729))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(628481, arguments.callee, this, arguments);
                            arguments = J$.N(628489, 'arguments', arguments, true, false, false);
                            target = J$.N(628497, 'target', target, true, false, false);
                            property = J$.N(628505, 'property', property, true, false, false);
                            receiver = J$.N(628513, 'receiver', receiver, true, false, false);
                            if (J$.C(36200, J$.C(36176, J$.B(79410, '!==', J$.U(79402, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(628121, 'Reflect', undefined, true, true) : Reflect = J$.R(628121, 'Reflect', Reflect, true, true))), J$.T(628129, 'undefined', 21, false))) ? J$.G(628145, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(628137, 'Reflect', undefined, true, true) : Reflect = J$.R(628137, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(628169, '_get', J$.G(628161, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(628153, 'Reflect', undefined, true, true) : Reflect = J$.R(628153, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(628417, '_get', J$.T(628409, function _get(target, property, receiver) {
                                    jalangiLabel11:
                                        while (true) {
                                            try {
                                                J$.Fe(628353, arguments.callee, this, arguments);
                                                arguments = J$.N(628361, 'arguments', arguments, true, false, false);
                                                target = J$.N(628369, 'target', target, true, false, false);
                                                property = J$.N(628377, 'property', property, true, false, false);
                                                receiver = J$.N(628385, 'receiver', receiver, true, false, false);
                                                J$.N(628393, 'base', base, false, false, false);
                                                J$.N(628401, 'desc', desc, false, false, false);
                                                var base = J$.W(628209, 'base', J$.F(628201, J$.R(628177, '_superPropBase', _superPropBase, false, true), false)(J$.R(628185, 'target', target, false, false), J$.R(628193, 'property', property, false, false)), base, false, false);
                                                if (J$.C(36184, J$.U(79418, '!', J$.R(628217, 'base', base, false, false))))
                                                    return J$.Rt(628225, undefined);
                                                var desc = J$.W(628265, 'desc', J$.M(628257, J$.I(typeof Object === 'undefined' ? Object = J$.R(628233, 'Object', undefined, true, true) : Object = J$.R(628233, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(628241, 'base', base, false, false), J$.R(628249, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(36192, J$.G(628281, J$.R(628273, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(628321, J$.M(628313, J$.G(628297, J$.R(628289, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(628305, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(628345, J$.G(628337, J$.R(628329, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(632737, J$e);
                                            } finally {
                                                if (J$.Fr(632745))
                                                    continue jalangiLabel11;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(628473, J$.F(628465, J$.R(628425, '_get', _get, false, true), false)(J$.R(628433, 'target', target, false, false), J$.R(628441, 'property', property, false, false), J$.C(36208, J$.R(628449, 'receiver', receiver, false, false)) ? J$._() : J$.R(628457, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(632753, J$e);
                        } finally {
                            if (J$.Fr(632761))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(628633, arguments.callee, this, arguments);
                            arguments = J$.N(628641, 'arguments', arguments, true, false, false);
                            object = J$.N(628649, 'object', object, true, false, false);
                            property = J$.N(628657, 'property', property, true, false, false);
                            while (J$.C(36224, J$.U(79426, '!', J$.M(628561, J$.G(628537, J$.G(628529, J$.I(typeof Object === 'undefined' ? Object = J$.R(628521, 'Object', undefined, true, true) : Object = J$.R(628521, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(628545, 'object', object, false, false), J$.R(628553, 'property', property, false, false))))) {
                                object = J$.W(628593, 'object', J$.F(628585, J$.R(628569, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(628577, 'object', object, false, false)), object, false, false);
                                if (J$.C(36216, J$.B(79434, '===', J$.R(628601, 'object', object, false, false), J$.T(628609, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(628625, J$.R(628617, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(632769, J$e);
                        } finally {
                            if (J$.Fr(632777))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(628817, arguments.callee, this, arguments);
                            arguments = J$.N(628825, 'arguments', arguments, true, false, false);
                            o = J$.N(628833, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(628777, '_getPrototypeOf', J$.C(36240, J$.G(628673, J$.I(typeof Object === 'undefined' ? Object = J$.R(628665, 'Object', undefined, true, true) : Object = J$.R(628665, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(628689, J$.I(typeof Object === 'undefined' ? Object = J$.R(628681, 'Object', undefined, true, true) : Object = J$.R(628681, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(628769, function _getPrototypeOf(o) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(628745, arguments.callee, this, arguments);
                                            arguments = J$.N(628753, 'arguments', arguments, true, false, false);
                                            o = J$.N(628761, 'o', o, true, false, false);
                                            return J$.Rt(628737, J$.C(36232, J$.G(628705, J$.R(628697, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(628729, J$.I(typeof Object === 'undefined' ? Object = J$.R(628713, 'Object', undefined, true, true) : Object = J$.R(628713, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(628721, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(632785, J$e);
                                        } finally {
                                            if (J$.Fr(632793))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(628809, J$.F(628801, J$.R(628785, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(628793, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(632801, J$e);
                        } finally {
                            if (J$.Fr(632809))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(629033, arguments.callee, this, arguments);
                            arguments = J$.N(629041, 'arguments', arguments, true, false, false);
                            subClass = J$.N(629049, 'subClass', subClass, true, false, false);
                            superClass = J$.N(629057, 'superClass', superClass, true, false, false);
                            if (J$.C(36256, J$.C(36248, J$.B(79450, '!==', J$.U(79442, 'typeof', J$.R(628841, 'superClass', superClass, false, false)), J$.T(628849, 'function', 21, false))) ? J$.B(79458, '!==', J$.R(628857, 'superClass', superClass, false, false), J$.T(628865, null, 25, false)) : J$._())) {
                                throw J$.F(628889, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(628873, 'TypeError', undefined, true, true) : TypeError = J$.R(628873, 'TypeError', TypeError, true, true)), true)(J$.T(628881, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(628985, J$.R(628897, 'subClass', subClass, false, false), 'prototype', J$.M(628977, J$.I(typeof Object === 'undefined' ? Object = J$.R(628905, 'Object', undefined, true, true) : Object = J$.R(628905, 'Object', Object, true, true)), 'create', false)(J$.C(36264, J$.R(628913, 'superClass', superClass, false, false)) ? J$.G(628929, J$.R(628921, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(628969, {
                                constructor: J$.T(628961, {
                                    value: J$.R(628937, 'subClass', subClass, false, false),
                                    writable: J$.T(628945, true, 23, false),
                                    configurable: J$.T(628953, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(36272, J$.R(628993, 'superClass', superClass, false, false)))
                                J$.F(629025, J$.R(629001, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(629009, 'subClass', subClass, false, false), J$.R(629017, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(632817, J$e);
                        } finally {
                            if (J$.Fr(632825))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(629209, arguments.callee, this, arguments);
                            arguments = J$.N(629217, 'arguments', arguments, true, false, false);
                            o = J$.N(629225, 'o', o, true, false, false);
                            p = J$.N(629233, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(629161, '_setPrototypeOf', J$.C(36280, J$.G(629073, J$.I(typeof Object === 'undefined' ? Object = J$.R(629065, 'Object', undefined, true, true) : Object = J$.R(629065, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(629153, function _setPrototypeOf(o, p) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(629121, arguments.callee, this, arguments);
                                            arguments = J$.N(629129, 'arguments', arguments, true, false, false);
                                            o = J$.N(629137, 'o', o, true, false, false);
                                            p = J$.N(629145, 'p', p, true, false, false);
                                            J$.P(629097, J$.R(629081, 'o', o, false, false), '__proto__', J$.R(629089, 'p', p, false, false));
                                            return J$.Rt(629113, J$.R(629105, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(632833, J$e);
                                        } finally {
                                            if (J$.Fr(632841))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(629201, J$.F(629193, J$.R(629169, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(629177, 'o', o, false, false), J$.R(629185, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(632849, J$e);
                        } finally {
                            if (J$.Fr(632857))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(632313, '_typeof2', J$.T(632305, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(632329, '_typeof', J$.T(632321, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(632345, '_classCallCheck', J$.T(632337, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(632361, '_defineProperties', J$.T(632353, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(632377, '_createClass', J$.T(632369, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(632393, '_possibleConstructorReturn', J$.T(632385, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(632409, '_assertThisInitialized', J$.T(632401, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(632425, '_get', J$.T(632417, _get, 12, false), true, false, false);
            _superPropBase = J$.N(632441, '_superPropBase', J$.T(632433, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(632457, '_getPrototypeOf', J$.T(632449, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(632473, '_inherits', J$.T(632465, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(632489, '_setPrototypeOf', J$.T(632481, _setPrototypeOf, 12, false), true, false, false);
            J$.N(632497, 'AuthProvider', AuthProvider, false, false, false);
            J$.N(632505, 'retrieveKerberos', retrieveKerberos, false, false, false);
            J$.N(632513, 'kerberos', kerberos, false, false, false);
            J$.N(632521, 'GSSAPI', GSSAPI, false, false, false);
            J$.N(632529, 'GSSAPIInitialize', GSSAPIInitialize, false, false, false);
            J$.N(632537, 'MongoDBGSSAPIFirstStep', MongoDBGSSAPIFirstStep, false, false, false);
            J$.N(632545, 'MongoDBGSSAPISecondStep', MongoDBGSSAPISecondStep, false, false, false);
            J$.N(632553, 'MongoDBGSSAPIThirdStep', MongoDBGSSAPIThirdStep, false, false, false);
            J$.T(626769, 'use strict', 21, false);
            var AuthProvider = J$.W(629273, 'AuthProvider', J$.G(629265, J$.F(629257, J$.I(typeof require === 'undefined' ? require = J$.R(629241, 'require', undefined, true, true) : require = J$.R(629241, 'require', require, true, true)), false)(J$.T(629249, './auth_provider', 21, false)), 'AuthProvider'), AuthProvider, false, true);
            var retrieveKerberos = J$.W(629313, 'retrieveKerberos', J$.G(629305, J$.F(629297, J$.I(typeof require === 'undefined' ? require = J$.R(629281, 'require', undefined, true, true) : require = J$.R(629281, 'require', require, true, true)), false)(J$.T(629289, '../utils', 21, false)), 'retrieveKerberos'), retrieveKerberos, false, true);
            var kerberos;
            var GSSAPI = J$.W(630257, 'GSSAPI', J$.F(630249, J$.T(630233, function (_AuthProvider) {
                jalangiLabel22:
                    while (true) {
                        try {
                            J$.Fe(630193, arguments.callee, this, arguments);
                            function GSSAPI() {
                                jalangiLabel19:
                                    while (true) {
                                        try {
                                            J$.Fe(629465, arguments.callee, this, arguments);
                                            arguments = J$.N(629473, 'arguments', arguments, true, false, false);
                                            J$.F(629377, J$.R(629353, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(629361, 'this', this, false, false), J$.R(629369, 'GSSAPI', GSSAPI, false, false));
                                            return J$.Rt(629457, J$.F(629449, J$.R(629385, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(629393, 'this', this, false, false), J$.M(629441, J$.F(629417, J$.R(629401, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(629409, 'GSSAPI', GSSAPI, false, false)), 'apply', false)(J$.R(629425, 'this', this, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(629433, 'arguments', undefined, true, true) : arguments = J$.R(629433, 'arguments', arguments, true, true)))));
                                        } catch (J$e) {
                                            J$.Ex(632865, J$e);
                                        } finally {
                                            if (J$.Fr(632873))
                                                continue jalangiLabel19;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(630201, 'arguments', arguments, true, false, false);
                            _AuthProvider = J$.N(630209, '_AuthProvider', _AuthProvider, true, false, false);
                            GSSAPI = J$.N(630225, 'GSSAPI', J$.T(630217, GSSAPI, 12, false), true, false, false);
                            J$.F(629345, J$.R(629321, '_inherits', _inherits, false, true), false)(J$.R(629329, 'GSSAPI', GSSAPI, false, false), J$.R(629337, '_AuthProvider', _AuthProvider, false, false));
                            J$.F(630169, J$.R(629481, '_createClass', _createClass, false, true), false)(J$.R(629489, 'GSSAPI', GSSAPI, false, false), J$.T(630161, [
                                J$.T(629881, {
                                    key: J$.T(629497, '_authenticateSingleConnection', 21, false),
                                    value: J$.T(629873, function _authenticateSingleConnection(sendAuthCommand, connection, credentials, callback) {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(629785, arguments.callee, this, arguments);
                                                    arguments = J$.N(629793, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(629801, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connection = J$.N(629809, 'connection', connection, true, false, false);
                                                    credentials = J$.N(629817, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(629825, 'callback', callback, true, false, false);
                                                    J$.N(629833, 'source', source, false, false, false);
                                                    J$.N(629841, 'username', username, false, false, false);
                                                    J$.N(629849, 'password', password, false, false, false);
                                                    J$.N(629857, 'mechanismProperties', mechanismProperties, false, false, false);
                                                    J$.N(629865, 'gssapiServiceName', gssapiServiceName, false, false, false);
                                                    var source = J$.W(629521, 'source', J$.G(629513, J$.R(629505, 'credentials', credentials, false, false), 'source'), source, false, false);
                                                    var username = J$.W(629545, 'username', J$.G(629537, J$.R(629529, 'credentials', credentials, false, false), 'username'), username, false, false);
                                                    var password = J$.W(629569, 'password', J$.G(629561, J$.R(629553, 'credentials', credentials, false, false), 'password'), password, false, false);
                                                    var mechanismProperties = J$.W(629593, 'mechanismProperties', J$.G(629585, J$.R(629577, 'credentials', credentials, false, false), 'mechanismProperties'), mechanismProperties, false, false);
                                                    var gssapiServiceName = J$.W(629657, 'gssapiServiceName', J$.C(36296, J$.C(36288, J$.G(629617, J$.R(629601, 'mechanismProperties', mechanismProperties, false, false), J$.T(629609, 'gssapiservicename', 21, false))) ? J$._() : J$.G(629641, J$.R(629625, 'mechanismProperties', mechanismProperties, false, false), J$.T(629633, 'gssapiServiceName', 21, false))) ? J$._() : J$.T(629649, 'mongodb', 21, false), gssapiServiceName, false, false);
                                                    J$.F(629777, J$.R(629665, 'GSSAPIInitialize', GSSAPIInitialize, false, true), false)(J$.R(629673, 'this', this, false, false), J$.G(629697, J$.G(629689, J$.R(629681, 'kerberos', kerberos, false, true), 'processes'), 'MongoAuthProcess'), J$.R(629705, 'source', source, false, false), J$.R(629713, 'username', username, false, false), J$.R(629721, 'password', password, false, false), J$.R(629729, 'source', source, false, false), J$.R(629737, 'gssapiServiceName', gssapiServiceName, false, false), J$.R(629745, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(629753, 'connection', connection, false, false), J$.R(629761, 'mechanismProperties', mechanismProperties, false, false), J$.R(629769, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(632881, J$e);
                                                } finally {
                                                    if (J$.Fr(632889))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(630153, {
                                    key: J$.T(629889, 'auth', 21, false),
                                    value: J$.T(630145, function auth(sendAuthCommand, connections, credentials, callback) {
                                        jalangiLabel21:
                                            while (true) {
                                                try {
                                                    J$.Fe(630097, arguments.callee, this, arguments);
                                                    arguments = J$.N(630105, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(630113, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connections = J$.N(630121, 'connections', connections, true, false, false);
                                                    credentials = J$.N(630129, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(630137, 'callback', callback, true, false, false);
                                                    if (J$.C(36304, J$.B(79466, '==', J$.R(629897, 'kerberos', kerberos, false, true), J$.T(629905, null, 25, false)))) {
                                                        try {
                                                            kerberos = J$.W(629929, 'kerberos', J$.F(629921, J$.R(629913, 'retrieveKerberos', retrieveKerberos, false, true), false)(), kerberos, false, true);
                                                        } catch (e) {
                                                            J$.N(629977, 'e', e, false, false, true);
                                                            return J$.Rt(629969, J$.F(629961, J$.R(629937, 'callback', callback, false, false), false)(J$.R(629945, 'e', e, false, false), J$.T(629953, null, 25, false)));
                                                        }
                                                    }
                                                    J$.M(630089, J$.F(630041, J$.R(629985, '_get', _get, false, true), false)(J$.F(630017, J$.R(629993, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(630009, J$.R(630001, 'GSSAPI', GSSAPI, false, false), 'prototype')), J$.T(630025, 'auth', 21, false), J$.R(630033, 'this', this, false, false)), 'call', false)(J$.R(630049, 'this', this, false, false), J$.R(630057, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(630065, 'connections', connections, false, false), J$.R(630073, 'credentials', credentials, false, false), J$.R(630081, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(632897, J$e);
                                                } finally {
                                                    if (J$.Fr(632905))
                                                        continue jalangiLabel21;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(630185, J$.R(630177, 'GSSAPI', GSSAPI, false, false));
                        } catch (J$e) {
                            J$.Ex(632913, J$e);
                        } finally {
                            if (J$.Fr(632921))
                                continue jalangiLabel22;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(630241, 'AuthProvider', AuthProvider, false, true)), GSSAPI, false, true);
            var GSSAPIInitialize = J$.W(630777, 'GSSAPIInitialize', J$.T(630769, function GSSAPIInitialize(self, MongoAuthProcess, db, username, password, authdb, gssapiServiceName, sendAuthCommand, connection, options, callback) {
                jalangiLabel25:
                    while (true) {
                        try {
                            J$.Fe(630657, arguments.callee, this, arguments);
                            arguments = J$.N(630665, 'arguments', arguments, true, false, false);
                            self = J$.N(630673, 'self', self, true, false, false);
                            MongoAuthProcess = J$.N(630681, 'MongoAuthProcess', MongoAuthProcess, true, false, false);
                            db = J$.N(630689, 'db', db, true, false, false);
                            username = J$.N(630697, 'username', username, true, false, false);
                            password = J$.N(630705, 'password', password, true, false, false);
                            authdb = J$.N(630713, 'authdb', authdb, true, false, false);
                            gssapiServiceName = J$.N(630721, 'gssapiServiceName', gssapiServiceName, true, false, false);
                            sendAuthCommand = J$.N(630729, 'sendAuthCommand', sendAuthCommand, true, false, false);
                            connection = J$.N(630737, 'connection', connection, true, false, false);
                            options = J$.N(630745, 'options', options, true, false, false);
                            callback = J$.N(630753, 'callback', callback, true, false, false);
                            J$.N(630761, 'mongo_auth_process', mongo_auth_process, false, false, false);
                            var mongo_auth_process = J$.W(630329, 'mongo_auth_process', J$.F(630321, J$.R(630265, 'MongoAuthProcess', MongoAuthProcess, false, false), true)(J$.G(630281, J$.R(630273, 'connection', connection, false, false), 'host'), J$.G(630297, J$.R(630289, 'connection', connection, false, false), 'port'), J$.R(630305, 'gssapiServiceName', gssapiServiceName, false, false), J$.R(630313, 'options', options, false, false)), mongo_auth_process, false, false);
                            J$.M(630649, J$.R(630337, 'mongo_auth_process', mongo_auth_process, false, false), 'init', false)(J$.R(630345, 'username', username, false, false), J$.R(630353, 'password', password, false, false), J$.T(630641, function (err) {
                                jalangiLabel24:
                                    while (true) {
                                        try {
                                            J$.Fe(630617, arguments.callee, this, arguments);
                                            arguments = J$.N(630625, 'arguments', arguments, true, false, false);
                                            err = J$.N(630633, 'err', err, true, false, false);
                                            if (J$.C(36312, J$.R(630361, 'err', err, false, false)))
                                                return J$.Rt(630401, J$.F(630393, J$.R(630369, 'callback', callback, false, false), false)(J$.R(630377, 'err', err, false, false), J$.T(630385, false, 23, false)));
                                            J$.M(630609, J$.R(630409, 'mongo_auth_process', mongo_auth_process, false, false), 'transition', false)(J$.T(630417, '', 21, false), J$.T(630601, function (err, payload) {
                                                jalangiLabel23:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(630569, arguments.callee, this, arguments);
                                                            arguments = J$.N(630577, 'arguments', arguments, true, false, false);
                                                            err = J$.N(630585, 'err', err, true, false, false);
                                                            payload = J$.N(630593, 'payload', payload, true, false, false);
                                                            if (J$.C(36320, J$.R(630425, 'err', err, false, false)))
                                                                return J$.Rt(630465, J$.F(630457, J$.R(630433, 'callback', callback, false, false), false)(J$.R(630441, 'err', err, false, false), J$.T(630449, false, 23, false)));
                                                            J$.F(630561, J$.R(630473, 'MongoDBGSSAPIFirstStep', MongoDBGSSAPIFirstStep, false, true), false)(J$.R(630481, 'self', self, false, false), J$.R(630489, 'mongo_auth_process', mongo_auth_process, false, false), J$.R(630497, 'payload', payload, false, false), J$.R(630505, 'db', db, false, false), J$.R(630513, 'username', username, false, false), J$.R(630521, 'password', password, false, false), J$.R(630529, 'authdb', authdb, false, false), J$.R(630537, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(630545, 'connection', connection, false, false), J$.R(630553, 'callback', callback, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(632929, J$e);
                                                        } finally {
                                                            if (J$.Fr(632937))
                                                                continue jalangiLabel23;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(632945, J$e);
                                        } finally {
                                            if (J$.Fr(632953))
                                                continue jalangiLabel24;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(632961, J$e);
                        } finally {
                            if (J$.Fr(632969))
                                continue jalangiLabel25;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), GSSAPIInitialize, false, true);
            var MongoDBGSSAPIFirstStep = J$.W(631297, 'MongoDBGSSAPIFirstStep', J$.T(631289, function MongoDBGSSAPIFirstStep(self, mongo_auth_process, payload, db, username, password, authdb, sendAuthCommand, connection, callback) {
                jalangiLabel28:
                    while (true) {
                        try {
                            J$.Fe(631185, arguments.callee, this, arguments);
                            arguments = J$.N(631193, 'arguments', arguments, true, false, false);
                            self = J$.N(631201, 'self', self, true, false, false);
                            mongo_auth_process = J$.N(631209, 'mongo_auth_process', mongo_auth_process, true, false, false);
                            payload = J$.N(631217, 'payload', payload, true, false, false);
                            db = J$.N(631225, 'db', db, true, false, false);
                            username = J$.N(631233, 'username', username, true, false, false);
                            password = J$.N(631241, 'password', password, true, false, false);
                            authdb = J$.N(631249, 'authdb', authdb, true, false, false);
                            sendAuthCommand = J$.N(631257, 'sendAuthCommand', sendAuthCommand, true, false, false);
                            connection = J$.N(631265, 'connection', connection, true, false, false);
                            callback = J$.N(631273, 'callback', callback, true, false, false);
                            J$.N(631281, 'command', command, false, false, false);
                            var command = J$.W(630825, 'command', J$.T(630817, {
                                saslStart: J$.T(630785, 1, 22, false),
                                mechanism: J$.T(630793, 'GSSAPI', 21, false),
                                payload: J$.R(630801, 'payload', payload, false, false),
                                autoAuthorize: J$.T(630809, 1, 22, false)
                            }, 11, false), command, false, false);
                            J$.F(631177, J$.R(630833, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(630841, 'connection', connection, false, false), J$.T(630849, '$external.$cmd', 21, false), J$.R(630857, 'command', command, false, false), J$.T(631169, function (err, doc) {
                                jalangiLabel27:
                                    while (true) {
                                        try {
                                            J$.Fe(631137, arguments.callee, this, arguments);
                                            arguments = J$.N(631145, 'arguments', arguments, true, false, false);
                                            err = J$.N(631153, 'err', err, true, false, false);
                                            doc = J$.N(631161, 'doc', doc, true, false, false);
                                            if (J$.C(36328, J$.R(630865, 'err', err, false, false)))
                                                return J$.Rt(630905, J$.F(630897, J$.R(630873, 'callback', callback, false, false), false)(J$.R(630881, 'err', err, false, false), J$.T(630889, false, 23, false)));
                                            J$.M(631129, J$.R(630913, 'mongo_auth_process', mongo_auth_process, false, false), 'transition', false)(J$.G(630929, J$.R(630921, 'doc', doc, false, false), 'payload'), J$.T(631121, function (err, payload) {
                                                jalangiLabel26:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(631089, arguments.callee, this, arguments);
                                                            arguments = J$.N(631097, 'arguments', arguments, true, false, false);
                                                            err = J$.N(631105, 'err', err, true, false, false);
                                                            payload = J$.N(631113, 'payload', payload, true, false, false);
                                                            if (J$.C(36336, J$.R(630937, 'err', err, false, false)))
                                                                return J$.Rt(630977, J$.F(630969, J$.R(630945, 'callback', callback, false, false), false)(J$.R(630953, 'err', err, false, false), J$.T(630961, false, 23, false)));
                                                            J$.F(631081, J$.R(630985, 'MongoDBGSSAPISecondStep', MongoDBGSSAPISecondStep, false, true), false)(J$.R(630993, 'self', self, false, false), J$.R(631001, 'mongo_auth_process', mongo_auth_process, false, false), J$.R(631009, 'payload', payload, false, false), J$.R(631017, 'doc', doc, false, false), J$.R(631025, 'db', db, false, false), J$.R(631033, 'username', username, false, false), J$.R(631041, 'password', password, false, false), J$.R(631049, 'authdb', authdb, false, false), J$.R(631057, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(631065, 'connection', connection, false, false), J$.R(631073, 'callback', callback, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(632977, J$e);
                                                        } finally {
                                                            if (J$.Fr(632985))
                                                                continue jalangiLabel26;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(632993, J$e);
                                        } finally {
                                            if (J$.Fr(633001))
                                                continue jalangiLabel27;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(633009, J$e);
                        } finally {
                            if (J$.Fr(633017))
                                continue jalangiLabel28;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), MongoDBGSSAPIFirstStep, false, true);
            var MongoDBGSSAPISecondStep = J$.W(631825, 'MongoDBGSSAPISecondStep', J$.T(631817, function MongoDBGSSAPISecondStep(self, mongo_auth_process, payload, doc, db, username, password, authdb, sendAuthCommand, connection, callback) {
                jalangiLabel31:
                    while (true) {
                        try {
                            J$.Fe(631705, arguments.callee, this, arguments);
                            arguments = J$.N(631713, 'arguments', arguments, true, false, false);
                            self = J$.N(631721, 'self', self, true, false, false);
                            mongo_auth_process = J$.N(631729, 'mongo_auth_process', mongo_auth_process, true, false, false);
                            payload = J$.N(631737, 'payload', payload, true, false, false);
                            doc = J$.N(631745, 'doc', doc, true, false, false);
                            db = J$.N(631753, 'db', db, true, false, false);
                            username = J$.N(631761, 'username', username, true, false, false);
                            password = J$.N(631769, 'password', password, true, false, false);
                            authdb = J$.N(631777, 'authdb', authdb, true, false, false);
                            sendAuthCommand = J$.N(631785, 'sendAuthCommand', sendAuthCommand, true, false, false);
                            connection = J$.N(631793, 'connection', connection, true, false, false);
                            callback = J$.N(631801, 'callback', callback, true, false, false);
                            J$.N(631809, 'command', command, false, false, false);
                            var command = J$.W(631345, 'command', J$.T(631337, {
                                saslContinue: J$.T(631305, 1, 22, false),
                                conversationId: J$.G(631321, J$.R(631313, 'doc', doc, false, false), 'conversationId'),
                                payload: J$.R(631329, 'payload', payload, false, false)
                            }, 11, false), command, false, false);
                            J$.F(631697, J$.R(631353, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(631361, 'connection', connection, false, false), J$.T(631369, '$external.$cmd', 21, false), J$.R(631377, 'command', command, false, false), J$.T(631689, function (err, doc) {
                                jalangiLabel30:
                                    while (true) {
                                        try {
                                            J$.Fe(631657, arguments.callee, this, arguments);
                                            arguments = J$.N(631665, 'arguments', arguments, true, false, false);
                                            err = J$.N(631673, 'err', err, true, false, false);
                                            doc = J$.N(631681, 'doc', doc, true, false, false);
                                            if (J$.C(36344, J$.R(631385, 'err', err, false, false)))
                                                return J$.Rt(631425, J$.F(631417, J$.R(631393, 'callback', callback, false, false), false)(J$.R(631401, 'err', err, false, false), J$.T(631409, false, 23, false)));
                                            J$.M(631649, J$.R(631433, 'mongo_auth_process', mongo_auth_process, false, false), 'transition', false)(J$.G(631449, J$.R(631441, 'doc', doc, false, false), 'payload'), J$.T(631641, function (err, payload) {
                                                jalangiLabel29:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(631609, arguments.callee, this, arguments);
                                                            arguments = J$.N(631617, 'arguments', arguments, true, false, false);
                                                            err = J$.N(631625, 'err', err, true, false, false);
                                                            payload = J$.N(631633, 'payload', payload, true, false, false);
                                                            if (J$.C(36352, J$.R(631457, 'err', err, false, false)))
                                                                return J$.Rt(631497, J$.F(631489, J$.R(631465, 'callback', callback, false, false), false)(J$.R(631473, 'err', err, false, false), J$.T(631481, false, 23, false)));
                                                            J$.F(631601, J$.R(631505, 'MongoDBGSSAPIThirdStep', MongoDBGSSAPIThirdStep, false, true), false)(J$.R(631513, 'self', self, false, false), J$.R(631521, 'mongo_auth_process', mongo_auth_process, false, false), J$.R(631529, 'payload', payload, false, false), J$.R(631537, 'doc', doc, false, false), J$.R(631545, 'db', db, false, false), J$.R(631553, 'username', username, false, false), J$.R(631561, 'password', password, false, false), J$.R(631569, 'authdb', authdb, false, false), J$.R(631577, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(631585, 'connection', connection, false, false), J$.R(631593, 'callback', callback, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(633025, J$e);
                                                        } finally {
                                                            if (J$.Fr(633033))
                                                                continue jalangiLabel29;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(633041, J$e);
                                        } finally {
                                            if (J$.Fr(633049))
                                                continue jalangiLabel30;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(633057, J$e);
                        } finally {
                            if (J$.Fr(633065))
                                continue jalangiLabel31;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), MongoDBGSSAPISecondStep, false, true);
            var MongoDBGSSAPIThirdStep = J$.W(632265, 'MongoDBGSSAPIThirdStep', J$.T(632257, function MongoDBGSSAPIThirdStep(self, mongo_auth_process, payload, doc, db, username, password, authdb, sendAuthCommand, connection, callback) {
                jalangiLabel34:
                    while (true) {
                        try {
                            J$.Fe(632145, arguments.callee, this, arguments);
                            arguments = J$.N(632153, 'arguments', arguments, true, false, false);
                            self = J$.N(632161, 'self', self, true, false, false);
                            mongo_auth_process = J$.N(632169, 'mongo_auth_process', mongo_auth_process, true, false, false);
                            payload = J$.N(632177, 'payload', payload, true, false, false);
                            doc = J$.N(632185, 'doc', doc, true, false, false);
                            db = J$.N(632193, 'db', db, true, false, false);
                            username = J$.N(632201, 'username', username, true, false, false);
                            password = J$.N(632209, 'password', password, true, false, false);
                            authdb = J$.N(632217, 'authdb', authdb, true, false, false);
                            sendAuthCommand = J$.N(632225, 'sendAuthCommand', sendAuthCommand, true, false, false);
                            connection = J$.N(632233, 'connection', connection, true, false, false);
                            callback = J$.N(632241, 'callback', callback, true, false, false);
                            J$.N(632249, 'command', command, false, false, false);
                            var command = J$.W(631873, 'command', J$.T(631865, {
                                saslContinue: J$.T(631833, 1, 22, false),
                                conversationId: J$.G(631849, J$.R(631841, 'doc', doc, false, false), 'conversationId'),
                                payload: J$.R(631857, 'payload', payload, false, false)
                            }, 11, false), command, false, false);
                            J$.F(632137, J$.R(631881, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(631889, 'connection', connection, false, false), J$.T(631897, '$external.$cmd', 21, false), J$.R(631905, 'command', command, false, false), J$.T(632129, function (err, r) {
                                jalangiLabel33:
                                    while (true) {
                                        try {
                                            J$.Fe(632097, arguments.callee, this, arguments);
                                            arguments = J$.N(632105, 'arguments', arguments, true, false, false);
                                            err = J$.N(632113, 'err', err, true, false, false);
                                            r = J$.N(632121, 'r', r, true, false, false);
                                            if (J$.C(36360, J$.R(631913, 'err', err, false, false)))
                                                return J$.Rt(631953, J$.F(631945, J$.R(631921, 'callback', callback, false, false), false)(J$.R(631929, 'err', err, false, false), J$.T(631937, false, 23, false)));
                                            J$.M(632089, J$.R(631961, 'mongo_auth_process', mongo_auth_process, false, false), 'transition', false)(J$.T(631969, null, 25, false), J$.T(632081, function (err) {
                                                jalangiLabel32:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(632057, arguments.callee, this, arguments);
                                                            arguments = J$.N(632065, 'arguments', arguments, true, false, false);
                                                            err = J$.N(632073, 'err', err, true, false, false);
                                                            if (J$.C(36368, J$.R(631977, 'err', err, false, false)))
                                                                return J$.Rt(632017, J$.F(632009, J$.R(631985, 'callback', callback, false, false), false)(J$.R(631993, 'err', err, false, false), J$.T(632001, null, 25, false)));
                                                            J$.F(632049, J$.R(632025, 'callback', callback, false, false), false)(J$.T(632033, null, 25, false), J$.R(632041, 'r', r, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(633073, J$e);
                                                        } finally {
                                                            if (J$.Fr(633081))
                                                                continue jalangiLabel32;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(633089, J$e);
                                        } finally {
                                            if (J$.Fr(633097))
                                                continue jalangiLabel33;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(633105, J$e);
                        } finally {
                            if (J$.Fr(633113))
                                continue jalangiLabel34;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), MongoDBGSSAPIThirdStep, false, true);
            J$.P(632289, J$.I(typeof module === 'undefined' ? module = J$.R(632273, 'module', undefined, true, true) : module = J$.R(632273, 'module', module, true, true)), 'exports', J$.R(632281, 'GSSAPI', GSSAPI, false, true));
        } catch (J$e) {
            J$.Ex(633121, J$e);
        } finally {
            if (J$.Sr(633129))
                continue jalangiLabel35;
            else
                break jalangiLabel35;
        }
    }
// JALANGI DO NOT INSTRUMENT
