// mongodb/lib/core/auth/plain.js
J$.noInstrEval = false;
jalangiLabel19:
    while (true) {
        try {
            J$.Se(641009, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/auth/plain.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(638601, arguments.callee, this, arguments);
                            arguments = J$.N(638609, 'arguments', arguments, true, false, false);
                            obj = J$.N(638617, 'obj', obj, true, false, false);
                            if (J$.C(36792, J$.C(36752, J$.B(79882, '===', J$.U(79874, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(638337, 'Symbol', undefined, true, true) : Symbol = J$.R(638337, 'Symbol', Symbol, true, true))), J$.T(638345, 'function', 21, false))) ? J$.B(79898, '===', J$.U(79890, 'typeof', J$.G(638361, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(638353, 'Symbol', undefined, true, true) : Symbol = J$.R(638353, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(638369, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(638425, '_typeof2', J$.T(638417, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(638393, arguments.callee, this, arguments);
                                                arguments = J$.N(638401, 'arguments', arguments, true, false, false);
                                                obj = J$.N(638409, 'obj', obj, true, false, false);
                                                return J$.Rt(638385, J$.U(79906, 'typeof', J$.R(638377, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(641217, J$e);
                                            } finally {
                                                if (J$.Fr(641225))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(638561, '_typeof2', J$.T(638553, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(638529, arguments.callee, this, arguments);
                                                arguments = J$.N(638537, 'arguments', arguments, true, false, false);
                                                obj = J$.N(638545, 'obj', obj, true, false, false);
                                                return J$.Rt(638521, J$.C(36784, J$.C(36776, J$.C(36768, J$.C(36760, J$.R(638433, 'obj', obj, false, false)) ? J$.B(79922, '===', J$.U(79914, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(638441, 'Symbol', undefined, true, true) : Symbol = J$.R(638441, 'Symbol', Symbol, true, true))), J$.T(638449, 'function', 21, false)) : J$._()) ? J$.B(79930, '===', J$.G(638465, J$.R(638457, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(638473, 'Symbol', undefined, true, true) : Symbol = J$.R(638473, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(79938, '!==', J$.R(638481, 'obj', obj, false, false), J$.G(638497, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(638489, 'Symbol', undefined, true, true) : Symbol = J$.R(638489, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(638505, 'symbol', 21, false) : J$.U(79946, 'typeof', J$.R(638513, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(641233, J$e);
                                            } finally {
                                                if (J$.Fr(641241))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(638593, J$.F(638585, J$.R(638569, '_typeof2', _typeof2, false, true), false)(J$.R(638577, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(641249, J$e);
                        } finally {
                            if (J$.Fr(641257))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(638937, arguments.callee, this, arguments);
                            arguments = J$.N(638945, 'arguments', arguments, true, false, false);
                            obj = J$.N(638953, 'obj', obj, true, false, false);
                            if (J$.C(36840, J$.C(36800, J$.B(79962, '===', J$.U(79954, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(638625, 'Symbol', undefined, true, true) : Symbol = J$.R(638625, 'Symbol', Symbol, true, true))), J$.T(638633, 'function', 21, false))) ? J$.B(79970, '===', J$.F(638665, J$.R(638641, '_typeof2', _typeof2, false, true), false)(J$.G(638657, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(638649, 'Symbol', undefined, true, true) : Symbol = J$.R(638649, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(638673, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(638745, '_typeof', J$.T(638737, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(638713, arguments.callee, this, arguments);
                                                arguments = J$.N(638721, 'arguments', arguments, true, false, false);
                                                obj = J$.N(638729, 'obj', obj, true, false, false);
                                                return J$.Rt(638705, J$.F(638697, J$.R(638681, '_typeof2', _typeof2, false, true), false)(J$.R(638689, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(641265, J$e);
                                            } finally {
                                                if (J$.Fr(641273))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(638897, '_typeof', J$.T(638889, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(638865, arguments.callee, this, arguments);
                                                arguments = J$.N(638873, 'arguments', arguments, true, false, false);
                                                obj = J$.N(638881, 'obj', obj, true, false, false);
                                                return J$.Rt(638857, J$.C(36832, J$.C(36824, J$.C(36816, J$.C(36808, J$.R(638753, 'obj', obj, false, false)) ? J$.B(79986, '===', J$.U(79978, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(638761, 'Symbol', undefined, true, true) : Symbol = J$.R(638761, 'Symbol', Symbol, true, true))), J$.T(638769, 'function', 21, false)) : J$._()) ? J$.B(79994, '===', J$.G(638785, J$.R(638777, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(638793, 'Symbol', undefined, true, true) : Symbol = J$.R(638793, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(80002, '!==', J$.R(638801, 'obj', obj, false, false), J$.G(638817, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(638809, 'Symbol', undefined, true, true) : Symbol = J$.R(638809, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(638825, 'symbol', 21, false) : J$.F(638849, J$.R(638833, '_typeof2', _typeof2, false, true), false)(J$.R(638841, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(641281, J$e);
                                            } finally {
                                                if (J$.Fr(641289))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(638929, J$.F(638921, J$.R(638905, '_typeof', _typeof, false, true), false)(J$.R(638913, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(641297, J$e);
                        } finally {
                            if (J$.Fr(641305))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(639001, arguments.callee, this, arguments);
                            arguments = J$.N(639009, 'arguments', arguments, true, false, false);
                            instance = J$.N(639017, 'instance', instance, true, false, false);
                            Constructor = J$.N(639025, 'Constructor', Constructor, true, false, false);
                            if (J$.C(36848, J$.U(80018, '!', J$.B(80010, 'instanceof', J$.R(638961, 'instance', instance, false, false), J$.R(638969, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(638993, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(638977, 'TypeError', undefined, true, true) : TypeError = J$.R(638977, 'TypeError', TypeError, true, true)), true)(J$.T(638985, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(641313, J$e);
                        } finally {
                            if (J$.Fr(641321))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(639273, arguments.callee, this, arguments);
                            arguments = J$.N(639281, 'arguments', arguments, true, false, false);
                            target = J$.N(639289, 'target', target, true, false, false);
                            props = J$.N(639297, 'props', props, true, false, false);
                            J$.N(639305, 'i', i, false, false, false);
                            J$.N(639313, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(639041, 'i', J$.T(639033, 0, 22, false), i, false, false); J$.C(36872, J$.B(80026, '<', J$.R(639049, 'i', i, false, false), J$.G(639065, J$.R(639057, 'props', props, false, false), 'length'))); J$.B(80050, '-', i = J$.W(639081, 'i', J$.B(80042, '+', J$.U(80034, '+', J$.R(639073, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(639113, 'descriptor', J$.G(639105, J$.R(639089, 'props', props, false, false), J$.R(639097, 'i', i, false, false)), descriptor, false, false);
                                J$.P(639153, J$.R(639121, 'descriptor', descriptor, false, false), 'enumerable', J$.C(36856, J$.G(639137, J$.R(639129, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(639145, false, 23, false));
                                J$.P(639177, J$.R(639161, 'descriptor', descriptor, false, false), 'configurable', J$.T(639169, true, 23, false));
                                if (J$.C(36864, J$.B(80058, 'in', J$.T(639185, 'value', 21, false), J$.R(639193, 'descriptor', descriptor, false, false))))
                                    J$.P(639217, J$.R(639201, 'descriptor', descriptor, false, false), 'writable', J$.T(639209, true, 23, false));
                                J$.M(639265, J$.I(typeof Object === 'undefined' ? Object = J$.R(639225, 'Object', undefined, true, true) : Object = J$.R(639225, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(639233, 'target', target, false, false), J$.G(639249, J$.R(639241, 'descriptor', descriptor, false, false), 'key'), J$.R(639257, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(641329, J$e);
                        } finally {
                            if (J$.Fr(641337))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(639425, arguments.callee, this, arguments);
                            arguments = J$.N(639433, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(639441, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(639449, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(639457, 'staticProps', staticProps, true, false, false);
                            if (J$.C(36880, J$.R(639321, 'protoProps', protoProps, false, false)))
                                J$.F(639361, J$.R(639329, '_defineProperties', _defineProperties, false, true), false)(J$.G(639345, J$.R(639337, 'Constructor', Constructor, false, false), 'prototype'), J$.R(639353, 'protoProps', protoProps, false, false));
                            if (J$.C(36888, J$.R(639369, 'staticProps', staticProps, false, false)))
                                J$.F(639401, J$.R(639377, '_defineProperties', _defineProperties, false, true), false)(J$.R(639385, 'Constructor', Constructor, false, false), J$.R(639393, 'staticProps', staticProps, false, false));
                            return J$.Rt(639417, J$.R(639409, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(641345, J$e);
                        } finally {
                            if (J$.Fr(641353))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(639569, arguments.callee, this, arguments);
                            arguments = J$.N(639577, 'arguments', arguments, true, false, false);
                            self = J$.N(639585, 'self', self, true, false, false);
                            call = J$.N(639593, 'call', call, true, false, false);
                            if (J$.C(36912, J$.C(36904, J$.R(639465, 'call', call, false, false)) ? J$.C(36896, J$.B(80066, '===', J$.F(639489, J$.R(639473, '_typeof', _typeof, false, true), false)(J$.R(639481, 'call', call, false, false)), J$.T(639497, 'object', 21, false))) ? J$._() : J$.B(80082, '===', J$.U(80074, 'typeof', J$.R(639505, 'call', call, false, false)), J$.T(639513, 'function', 21, false)) : J$._())) {
                                return J$.Rt(639529, J$.R(639521, 'call', call, false, false));
                            }
                            return J$.Rt(639561, J$.F(639553, J$.R(639537, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(639545, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(641361, J$e);
                        } finally {
                            if (J$.Fr(641369))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(639657, arguments.callee, this, arguments);
                            arguments = J$.N(639665, 'arguments', arguments, true, false, false);
                            self = J$.N(639673, 'self', self, true, false, false);
                            if (J$.C(36920, J$.B(80090, '===', J$.R(639601, 'self', self, false, false), void J$.T(639609, 0, 22, false)))) {
                                throw J$.F(639633, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(639617, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(639617, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(639625, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(639649, J$.R(639641, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(641377, J$e);
                        } finally {
                            if (J$.Fr(641385))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(639833, arguments.callee, this, arguments);
                            arguments = J$.N(639841, 'arguments', arguments, true, false, false);
                            o = J$.N(639849, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(639793, '_getPrototypeOf', J$.C(36936, J$.G(639689, J$.I(typeof Object === 'undefined' ? Object = J$.R(639681, 'Object', undefined, true, true) : Object = J$.R(639681, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(639705, J$.I(typeof Object === 'undefined' ? Object = J$.R(639697, 'Object', undefined, true, true) : Object = J$.R(639697, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(639785, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(639761, arguments.callee, this, arguments);
                                            arguments = J$.N(639769, 'arguments', arguments, true, false, false);
                                            o = J$.N(639777, 'o', o, true, false, false);
                                            return J$.Rt(639753, J$.C(36928, J$.G(639721, J$.R(639713, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(639745, J$.I(typeof Object === 'undefined' ? Object = J$.R(639729, 'Object', undefined, true, true) : Object = J$.R(639729, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(639737, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(641393, J$e);
                                        } finally {
                                            if (J$.Fr(641401))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(639825, J$.F(639817, J$.R(639801, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(639809, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(641409, J$e);
                        } finally {
                            if (J$.Fr(641417))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(640049, arguments.callee, this, arguments);
                            arguments = J$.N(640057, 'arguments', arguments, true, false, false);
                            subClass = J$.N(640065, 'subClass', subClass, true, false, false);
                            superClass = J$.N(640073, 'superClass', superClass, true, false, false);
                            if (J$.C(36952, J$.C(36944, J$.B(80106, '!==', J$.U(80098, 'typeof', J$.R(639857, 'superClass', superClass, false, false)), J$.T(639865, 'function', 21, false))) ? J$.B(80114, '!==', J$.R(639873, 'superClass', superClass, false, false), J$.T(639881, null, 25, false)) : J$._())) {
                                throw J$.F(639905, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(639889, 'TypeError', undefined, true, true) : TypeError = J$.R(639889, 'TypeError', TypeError, true, true)), true)(J$.T(639897, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(640001, J$.R(639913, 'subClass', subClass, false, false), 'prototype', J$.M(639993, J$.I(typeof Object === 'undefined' ? Object = J$.R(639921, 'Object', undefined, true, true) : Object = J$.R(639921, 'Object', Object, true, true)), 'create', false)(J$.C(36960, J$.R(639929, 'superClass', superClass, false, false)) ? J$.G(639945, J$.R(639937, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(639985, {
                                constructor: J$.T(639977, {
                                    value: J$.R(639953, 'subClass', subClass, false, false),
                                    writable: J$.T(639961, true, 23, false),
                                    configurable: J$.T(639969, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(36968, J$.R(640009, 'superClass', superClass, false, false)))
                                J$.F(640041, J$.R(640017, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(640025, 'subClass', subClass, false, false), J$.R(640033, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(641425, J$e);
                        } finally {
                            if (J$.Fr(641433))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(640225, arguments.callee, this, arguments);
                            arguments = J$.N(640233, 'arguments', arguments, true, false, false);
                            o = J$.N(640241, 'o', o, true, false, false);
                            p = J$.N(640249, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(640177, '_setPrototypeOf', J$.C(36976, J$.G(640089, J$.I(typeof Object === 'undefined' ? Object = J$.R(640081, 'Object', undefined, true, true) : Object = J$.R(640081, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(640169, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(640137, arguments.callee, this, arguments);
                                            arguments = J$.N(640145, 'arguments', arguments, true, false, false);
                                            o = J$.N(640153, 'o', o, true, false, false);
                                            p = J$.N(640161, 'p', p, true, false, false);
                                            J$.P(640113, J$.R(640097, 'o', o, false, false), '__proto__', J$.R(640105, 'p', p, false, false));
                                            return J$.Rt(640129, J$.R(640121, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(641441, J$e);
                                        } finally {
                                            if (J$.Fr(641449))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(640217, J$.F(640209, J$.R(640185, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(640193, 'o', o, false, false), J$.R(640201, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(641457, J$e);
                        } finally {
                            if (J$.Fr(641465))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(641025, '_typeof2', J$.T(641017, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(641041, '_typeof', J$.T(641033, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(641057, '_classCallCheck', J$.T(641049, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(641073, '_defineProperties', J$.T(641065, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(641089, '_createClass', J$.T(641081, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(641105, '_possibleConstructorReturn', J$.T(641097, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(641121, '_assertThisInitialized', J$.T(641113, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(641137, '_getPrototypeOf', J$.T(641129, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(641153, '_inherits', J$.T(641145, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(641169, '_setPrototypeOf', J$.T(641161, _setPrototypeOf, 12, false), true, false, false);
            J$.N(641177, 'retrieveBSON', retrieveBSON, false, false, false);
            J$.N(641185, 'AuthProvider', AuthProvider, false, false, false);
            J$.N(641193, 'BSON', BSON, false, false, false);
            J$.N(641201, 'Binary', Binary, false, false, false);
            J$.N(641209, 'Plain', Plain, false, false, false);
            J$.T(638329, 'use strict', 21, false);
            var retrieveBSON = J$.W(640289, 'retrieveBSON', J$.G(640281, J$.F(640273, J$.I(typeof require === 'undefined' ? require = J$.R(640257, 'require', undefined, true, true) : require = J$.R(640257, 'require', require, true, true)), false)(J$.T(640265, '../connection/utils', 21, false)), 'retrieveBSON'), retrieveBSON, false, true);
            var AuthProvider = J$.W(640329, 'AuthProvider', J$.G(640321, J$.F(640313, J$.I(typeof require === 'undefined' ? require = J$.R(640297, 'require', undefined, true, true) : require = J$.R(640297, 'require', require, true, true)), false)(J$.T(640305, './auth_provider', 21, false)), 'AuthProvider'), AuthProvider, false, true);
            var BSON = J$.W(640353, 'BSON', J$.F(640345, J$.R(640337, 'retrieveBSON', retrieveBSON, false, true), false)(), BSON, false, true);
            var Binary = J$.W(640377, 'Binary', J$.G(640369, J$.R(640361, 'BSON', BSON, false, true), 'Binary'), Binary, false, true);
            var Plain = J$.W(640977, 'Plain', J$.F(640969, J$.T(640953, function (_AuthProvider) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(640913, arguments.callee, this, arguments);
                            function Plain() {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(640529, arguments.callee, this, arguments);
                                            arguments = J$.N(640537, 'arguments', arguments, true, false, false);
                                            J$.F(640441, J$.R(640417, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(640425, 'this', this, false, false), J$.R(640433, 'Plain', Plain, false, false));
                                            return J$.Rt(640521, J$.F(640513, J$.R(640449, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(640457, 'this', this, false, false), J$.M(640505, J$.F(640481, J$.R(640465, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(640473, 'Plain', Plain, false, false)), 'apply', false)(J$.R(640489, 'this', this, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(640497, 'arguments', undefined, true, true) : arguments = J$.R(640497, 'arguments', arguments, true, true)))));
                                        } catch (J$e) {
                                            J$.Ex(641473, J$e);
                                        } finally {
                                            if (J$.Fr(641481))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(640921, 'arguments', arguments, true, false, false);
                            _AuthProvider = J$.N(640929, '_AuthProvider', _AuthProvider, true, false, false);
                            Plain = J$.N(640945, 'Plain', J$.T(640937, Plain, 12, false), true, false, false);
                            J$.F(640409, J$.R(640385, '_inherits', _inherits, false, true), false)(J$.R(640393, 'Plain', Plain, false, false), J$.R(640401, '_AuthProvider', _AuthProvider, false, false));
                            J$.F(640889, J$.R(640545, '_createClass', _createClass, false, true), false)(J$.R(640553, 'Plain', Plain, false, false), J$.T(640881, [J$.T(640873, {
                                    key: J$.T(640561, '_authenticateSingleConnection', 21, false),
                                    value: J$.T(640865, function _authenticateSingleConnection(sendAuthCommand, connection, credentials, callback) {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(640785, arguments.callee, this, arguments);
                                                    arguments = J$.N(640793, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(640801, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connection = J$.N(640809, 'connection', connection, true, false, false);
                                                    credentials = J$.N(640817, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(640825, 'callback', callback, true, false, false);
                                                    J$.N(640833, 'username', username, false, false, false);
                                                    J$.N(640841, 'password', password, false, false, false);
                                                    J$.N(640849, 'payload', payload, false, false, false);
                                                    J$.N(640857, 'command', command, false, false, false);
                                                    var username = J$.W(640585, 'username', J$.G(640577, J$.R(640569, 'credentials', credentials, false, false), 'username'), username, false, false);
                                                    var password = J$.W(640609, 'password', J$.G(640601, J$.R(640593, 'credentials', credentials, false, false), 'password'), password, false, false);
                                                    var payload = J$.W(640681, 'payload', J$.F(640673, J$.R(640617, 'Binary', Binary, false, true), true)(J$.M(640665, J$.M(640649, J$.T(640625, '\0', 21, false), 'concat', false)(J$.R(640633, 'username', username, false, false), J$.T(640641, '\0', 21, false)), 'concat', false)(J$.R(640657, 'password', password, false, false))), payload, false, false);
                                                    var command = J$.W(640729, 'command', J$.T(640721, {
                                                        saslStart: J$.T(640689, 1, 22, false),
                                                        mechanism: J$.T(640697, 'PLAIN', 21, false),
                                                        payload: J$.R(640705, 'payload', payload, false, false),
                                                        autoAuthorize: J$.T(640713, 1, 22, false)
                                                    }, 11, false), command, false, false);
                                                    J$.F(640777, J$.R(640737, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(640745, 'connection', connection, false, false), J$.T(640753, '$external.$cmd', 21, false), J$.R(640761, 'command', command, false, false), J$.R(640769, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(641489, J$e);
                                                } finally {
                                                    if (J$.Fr(641497))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(640905, J$.R(640897, 'Plain', Plain, false, false));
                        } catch (J$e) {
                            J$.Ex(641505, J$e);
                        } finally {
                            if (J$.Fr(641513))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(640961, 'AuthProvider', AuthProvider, false, true)), Plain, false, true);
            J$.P(641001, J$.I(typeof module === 'undefined' ? module = J$.R(640985, 'module', undefined, true, true) : module = J$.R(640985, 'module', module, true, true)), 'exports', J$.R(640993, 'Plain', Plain, false, true));
        } catch (J$e) {
            J$.Ex(641521, J$e);
        } finally {
            if (J$.Sr(641529))
                continue jalangiLabel19;
            else
                break jalangiLabel19;
        }
    }
// JALANGI DO NOT INSTRUMENT
