// mongodb/lib/core/auth/scram.js
J$.noInstrEval = false;
jalangiLabel40:
    while (true) {
        try {
            J$.Se(649929, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/auth/scram.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(641809, arguments.callee, this, arguments);
                            arguments = J$.N(641817, 'arguments', arguments, true, false, false);
                            obj = J$.N(641825, 'obj', obj, true, false, false);
                            if (J$.C(37024, J$.C(36984, J$.B(80130, '===', J$.U(80122, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(641545, 'Symbol', undefined, true, true) : Symbol = J$.R(641545, 'Symbol', Symbol, true, true))), J$.T(641553, 'function', 21, false))) ? J$.B(80146, '===', J$.U(80138, 'typeof', J$.G(641569, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(641561, 'Symbol', undefined, true, true) : Symbol = J$.R(641561, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(641577, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(641633, '_typeof2', J$.T(641625, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(641601, arguments.callee, this, arguments);
                                                arguments = J$.N(641609, 'arguments', arguments, true, false, false);
                                                obj = J$.N(641617, 'obj', obj, true, false, false);
                                                return J$.Rt(641593, J$.U(80154, 'typeof', J$.R(641585, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(650329, J$e);
                                            } finally {
                                                if (J$.Fr(650337))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(641769, '_typeof2', J$.T(641761, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(641737, arguments.callee, this, arguments);
                                                arguments = J$.N(641745, 'arguments', arguments, true, false, false);
                                                obj = J$.N(641753, 'obj', obj, true, false, false);
                                                return J$.Rt(641729, J$.C(37016, J$.C(37008, J$.C(37000, J$.C(36992, J$.R(641641, 'obj', obj, false, false)) ? J$.B(80170, '===', J$.U(80162, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(641649, 'Symbol', undefined, true, true) : Symbol = J$.R(641649, 'Symbol', Symbol, true, true))), J$.T(641657, 'function', 21, false)) : J$._()) ? J$.B(80178, '===', J$.G(641673, J$.R(641665, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(641681, 'Symbol', undefined, true, true) : Symbol = J$.R(641681, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(80186, '!==', J$.R(641689, 'obj', obj, false, false), J$.G(641705, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(641697, 'Symbol', undefined, true, true) : Symbol = J$.R(641697, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(641713, 'symbol', 21, false) : J$.U(80194, 'typeof', J$.R(641721, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(650345, J$e);
                                            } finally {
                                                if (J$.Fr(650353))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(641801, J$.F(641793, J$.R(641777, '_typeof2', _typeof2, false, true), false)(J$.R(641785, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(650361, J$e);
                        } finally {
                            if (J$.Fr(650369))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(642145, arguments.callee, this, arguments);
                            arguments = J$.N(642153, 'arguments', arguments, true, false, false);
                            obj = J$.N(642161, 'obj', obj, true, false, false);
                            if (J$.C(37072, J$.C(37032, J$.B(80210, '===', J$.U(80202, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(641833, 'Symbol', undefined, true, true) : Symbol = J$.R(641833, 'Symbol', Symbol, true, true))), J$.T(641841, 'function', 21, false))) ? J$.B(80218, '===', J$.F(641873, J$.R(641849, '_typeof2', _typeof2, false, true), false)(J$.G(641865, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(641857, 'Symbol', undefined, true, true) : Symbol = J$.R(641857, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(641881, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(641953, '_typeof', J$.T(641945, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(641921, arguments.callee, this, arguments);
                                                arguments = J$.N(641929, 'arguments', arguments, true, false, false);
                                                obj = J$.N(641937, 'obj', obj, true, false, false);
                                                return J$.Rt(641913, J$.F(641905, J$.R(641889, '_typeof2', _typeof2, false, true), false)(J$.R(641897, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(650377, J$e);
                                            } finally {
                                                if (J$.Fr(650385))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(642105, '_typeof', J$.T(642097, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(642073, arguments.callee, this, arguments);
                                                arguments = J$.N(642081, 'arguments', arguments, true, false, false);
                                                obj = J$.N(642089, 'obj', obj, true, false, false);
                                                return J$.Rt(642065, J$.C(37064, J$.C(37056, J$.C(37048, J$.C(37040, J$.R(641961, 'obj', obj, false, false)) ? J$.B(80234, '===', J$.U(80226, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(641969, 'Symbol', undefined, true, true) : Symbol = J$.R(641969, 'Symbol', Symbol, true, true))), J$.T(641977, 'function', 21, false)) : J$._()) ? J$.B(80242, '===', J$.G(641993, J$.R(641985, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(642001, 'Symbol', undefined, true, true) : Symbol = J$.R(642001, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(80250, '!==', J$.R(642009, 'obj', obj, false, false), J$.G(642025, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(642017, 'Symbol', undefined, true, true) : Symbol = J$.R(642017, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(642033, 'symbol', 21, false) : J$.F(642057, J$.R(642041, '_typeof2', _typeof2, false, true), false)(J$.R(642049, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(650393, J$e);
                                            } finally {
                                                if (J$.Fr(650401))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(642137, J$.F(642129, J$.R(642113, '_typeof', _typeof, false, true), false)(J$.R(642121, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(650409, J$e);
                        } finally {
                            if (J$.Fr(650417))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(642209, arguments.callee, this, arguments);
                            arguments = J$.N(642217, 'arguments', arguments, true, false, false);
                            instance = J$.N(642225, 'instance', instance, true, false, false);
                            Constructor = J$.N(642233, 'Constructor', Constructor, true, false, false);
                            if (J$.C(37080, J$.U(80266, '!', J$.B(80258, 'instanceof', J$.R(642169, 'instance', instance, false, false), J$.R(642177, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(642201, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(642185, 'TypeError', undefined, true, true) : TypeError = J$.R(642185, 'TypeError', TypeError, true, true)), true)(J$.T(642193, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(650425, J$e);
                        } finally {
                            if (J$.Fr(650433))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(642481, arguments.callee, this, arguments);
                            arguments = J$.N(642489, 'arguments', arguments, true, false, false);
                            target = J$.N(642497, 'target', target, true, false, false);
                            props = J$.N(642505, 'props', props, true, false, false);
                            J$.N(642513, 'i', i, false, false, false);
                            J$.N(642521, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(642249, 'i', J$.T(642241, 0, 22, false), i, false, false); J$.C(37104, J$.B(80274, '<', J$.R(642257, 'i', i, false, false), J$.G(642273, J$.R(642265, 'props', props, false, false), 'length'))); J$.B(80298, '-', i = J$.W(642289, 'i', J$.B(80290, '+', J$.U(80282, '+', J$.R(642281, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(642321, 'descriptor', J$.G(642313, J$.R(642297, 'props', props, false, false), J$.R(642305, 'i', i, false, false)), descriptor, false, false);
                                J$.P(642361, J$.R(642329, 'descriptor', descriptor, false, false), 'enumerable', J$.C(37088, J$.G(642345, J$.R(642337, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(642353, false, 23, false));
                                J$.P(642385, J$.R(642369, 'descriptor', descriptor, false, false), 'configurable', J$.T(642377, true, 23, false));
                                if (J$.C(37096, J$.B(80306, 'in', J$.T(642393, 'value', 21, false), J$.R(642401, 'descriptor', descriptor, false, false))))
                                    J$.P(642425, J$.R(642409, 'descriptor', descriptor, false, false), 'writable', J$.T(642417, true, 23, false));
                                J$.M(642473, J$.I(typeof Object === 'undefined' ? Object = J$.R(642433, 'Object', undefined, true, true) : Object = J$.R(642433, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(642441, 'target', target, false, false), J$.G(642457, J$.R(642449, 'descriptor', descriptor, false, false), 'key'), J$.R(642465, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(650441, J$e);
                        } finally {
                            if (J$.Fr(650449))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(642633, arguments.callee, this, arguments);
                            arguments = J$.N(642641, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(642649, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(642657, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(642665, 'staticProps', staticProps, true, false, false);
                            if (J$.C(37112, J$.R(642529, 'protoProps', protoProps, false, false)))
                                J$.F(642569, J$.R(642537, '_defineProperties', _defineProperties, false, true), false)(J$.G(642553, J$.R(642545, 'Constructor', Constructor, false, false), 'prototype'), J$.R(642561, 'protoProps', protoProps, false, false));
                            if (J$.C(37120, J$.R(642577, 'staticProps', staticProps, false, false)))
                                J$.F(642609, J$.R(642585, '_defineProperties', _defineProperties, false, true), false)(J$.R(642593, 'Constructor', Constructor, false, false), J$.R(642601, 'staticProps', staticProps, false, false));
                            return J$.Rt(642625, J$.R(642617, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(650457, J$e);
                        } finally {
                            if (J$.Fr(650465))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(642777, arguments.callee, this, arguments);
                            arguments = J$.N(642785, 'arguments', arguments, true, false, false);
                            self = J$.N(642793, 'self', self, true, false, false);
                            call = J$.N(642801, 'call', call, true, false, false);
                            if (J$.C(37144, J$.C(37136, J$.R(642673, 'call', call, false, false)) ? J$.C(37128, J$.B(80314, '===', J$.F(642697, J$.R(642681, '_typeof', _typeof, false, true), false)(J$.R(642689, 'call', call, false, false)), J$.T(642705, 'object', 21, false))) ? J$._() : J$.B(80330, '===', J$.U(80322, 'typeof', J$.R(642713, 'call', call, false, false)), J$.T(642721, 'function', 21, false)) : J$._())) {
                                return J$.Rt(642737, J$.R(642729, 'call', call, false, false));
                            }
                            return J$.Rt(642769, J$.F(642761, J$.R(642745, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(642753, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(650473, J$e);
                        } finally {
                            if (J$.Fr(650481))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(642865, arguments.callee, this, arguments);
                            arguments = J$.N(642873, 'arguments', arguments, true, false, false);
                            self = J$.N(642881, 'self', self, true, false, false);
                            if (J$.C(37152, J$.B(80338, '===', J$.R(642809, 'self', self, false, false), void J$.T(642817, 0, 22, false)))) {
                                throw J$.F(642841, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(642825, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(642825, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(642833, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(642857, J$.R(642849, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(650489, J$e);
                        } finally {
                            if (J$.Fr(650497))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(643249, arguments.callee, this, arguments);
                            arguments = J$.N(643257, 'arguments', arguments, true, false, false);
                            target = J$.N(643265, 'target', target, true, false, false);
                            property = J$.N(643273, 'property', property, true, false, false);
                            receiver = J$.N(643281, 'receiver', receiver, true, false, false);
                            if (J$.C(37184, J$.C(37160, J$.B(80354, '!==', J$.U(80346, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(642889, 'Reflect', undefined, true, true) : Reflect = J$.R(642889, 'Reflect', Reflect, true, true))), J$.T(642897, 'undefined', 21, false))) ? J$.G(642913, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(642905, 'Reflect', undefined, true, true) : Reflect = J$.R(642905, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(642937, '_get', J$.G(642929, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(642921, 'Reflect', undefined, true, true) : Reflect = J$.R(642921, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(643185, '_get', J$.T(643177, function _get(target, property, receiver) {
                                    jalangiLabel11:
                                        while (true) {
                                            try {
                                                J$.Fe(643121, arguments.callee, this, arguments);
                                                arguments = J$.N(643129, 'arguments', arguments, true, false, false);
                                                target = J$.N(643137, 'target', target, true, false, false);
                                                property = J$.N(643145, 'property', property, true, false, false);
                                                receiver = J$.N(643153, 'receiver', receiver, true, false, false);
                                                J$.N(643161, 'base', base, false, false, false);
                                                J$.N(643169, 'desc', desc, false, false, false);
                                                var base = J$.W(642977, 'base', J$.F(642969, J$.R(642945, '_superPropBase', _superPropBase, false, true), false)(J$.R(642953, 'target', target, false, false), J$.R(642961, 'property', property, false, false)), base, false, false);
                                                if (J$.C(37168, J$.U(80362, '!', J$.R(642985, 'base', base, false, false))))
                                                    return J$.Rt(642993, undefined);
                                                var desc = J$.W(643033, 'desc', J$.M(643025, J$.I(typeof Object === 'undefined' ? Object = J$.R(643001, 'Object', undefined, true, true) : Object = J$.R(643001, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(643009, 'base', base, false, false), J$.R(643017, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(37176, J$.G(643049, J$.R(643041, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(643089, J$.M(643081, J$.G(643065, J$.R(643057, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(643073, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(643113, J$.G(643105, J$.R(643097, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(650505, J$e);
                                            } finally {
                                                if (J$.Fr(650513))
                                                    continue jalangiLabel11;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(643241, J$.F(643233, J$.R(643193, '_get', _get, false, true), false)(J$.R(643201, 'target', target, false, false), J$.R(643209, 'property', property, false, false), J$.C(37192, J$.R(643217, 'receiver', receiver, false, false)) ? J$._() : J$.R(643225, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(650521, J$e);
                        } finally {
                            if (J$.Fr(650529))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(643401, arguments.callee, this, arguments);
                            arguments = J$.N(643409, 'arguments', arguments, true, false, false);
                            object = J$.N(643417, 'object', object, true, false, false);
                            property = J$.N(643425, 'property', property, true, false, false);
                            while (J$.C(37208, J$.U(80370, '!', J$.M(643329, J$.G(643305, J$.G(643297, J$.I(typeof Object === 'undefined' ? Object = J$.R(643289, 'Object', undefined, true, true) : Object = J$.R(643289, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(643313, 'object', object, false, false), J$.R(643321, 'property', property, false, false))))) {
                                object = J$.W(643361, 'object', J$.F(643353, J$.R(643337, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(643345, 'object', object, false, false)), object, false, false);
                                if (J$.C(37200, J$.B(80378, '===', J$.R(643369, 'object', object, false, false), J$.T(643377, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(643393, J$.R(643385, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(650537, J$e);
                        } finally {
                            if (J$.Fr(650545))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(643585, arguments.callee, this, arguments);
                            arguments = J$.N(643593, 'arguments', arguments, true, false, false);
                            o = J$.N(643601, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(643545, '_getPrototypeOf', J$.C(37224, J$.G(643441, J$.I(typeof Object === 'undefined' ? Object = J$.R(643433, 'Object', undefined, true, true) : Object = J$.R(643433, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(643457, J$.I(typeof Object === 'undefined' ? Object = J$.R(643449, 'Object', undefined, true, true) : Object = J$.R(643449, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(643537, function _getPrototypeOf(o) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(643513, arguments.callee, this, arguments);
                                            arguments = J$.N(643521, 'arguments', arguments, true, false, false);
                                            o = J$.N(643529, 'o', o, true, false, false);
                                            return J$.Rt(643505, J$.C(37216, J$.G(643473, J$.R(643465, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(643497, J$.I(typeof Object === 'undefined' ? Object = J$.R(643481, 'Object', undefined, true, true) : Object = J$.R(643481, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(643489, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(650553, J$e);
                                        } finally {
                                            if (J$.Fr(650561))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(643577, J$.F(643569, J$.R(643553, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(643561, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(650569, J$e);
                        } finally {
                            if (J$.Fr(650577))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(643801, arguments.callee, this, arguments);
                            arguments = J$.N(643809, 'arguments', arguments, true, false, false);
                            subClass = J$.N(643817, 'subClass', subClass, true, false, false);
                            superClass = J$.N(643825, 'superClass', superClass, true, false, false);
                            if (J$.C(37240, J$.C(37232, J$.B(80394, '!==', J$.U(80386, 'typeof', J$.R(643609, 'superClass', superClass, false, false)), J$.T(643617, 'function', 21, false))) ? J$.B(80402, '!==', J$.R(643625, 'superClass', superClass, false, false), J$.T(643633, null, 25, false)) : J$._())) {
                                throw J$.F(643657, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(643641, 'TypeError', undefined, true, true) : TypeError = J$.R(643641, 'TypeError', TypeError, true, true)), true)(J$.T(643649, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(643753, J$.R(643665, 'subClass', subClass, false, false), 'prototype', J$.M(643745, J$.I(typeof Object === 'undefined' ? Object = J$.R(643673, 'Object', undefined, true, true) : Object = J$.R(643673, 'Object', Object, true, true)), 'create', false)(J$.C(37248, J$.R(643681, 'superClass', superClass, false, false)) ? J$.G(643697, J$.R(643689, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(643737, {
                                constructor: J$.T(643729, {
                                    value: J$.R(643705, 'subClass', subClass, false, false),
                                    writable: J$.T(643713, true, 23, false),
                                    configurable: J$.T(643721, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(37256, J$.R(643761, 'superClass', superClass, false, false)))
                                J$.F(643793, J$.R(643769, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(643777, 'subClass', subClass, false, false), J$.R(643785, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(650585, J$e);
                        } finally {
                            if (J$.Fr(650593))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(643977, arguments.callee, this, arguments);
                            arguments = J$.N(643985, 'arguments', arguments, true, false, false);
                            o = J$.N(643993, 'o', o, true, false, false);
                            p = J$.N(644001, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(643929, '_setPrototypeOf', J$.C(37264, J$.G(643841, J$.I(typeof Object === 'undefined' ? Object = J$.R(643833, 'Object', undefined, true, true) : Object = J$.R(643833, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(643921, function _setPrototypeOf(o, p) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(643889, arguments.callee, this, arguments);
                                            arguments = J$.N(643897, 'arguments', arguments, true, false, false);
                                            o = J$.N(643905, 'o', o, true, false, false);
                                            p = J$.N(643913, 'p', p, true, false, false);
                                            J$.P(643865, J$.R(643849, 'o', o, false, false), '__proto__', J$.R(643857, 'p', p, false, false));
                                            return J$.Rt(643881, J$.R(643873, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(650601, J$e);
                                        } finally {
                                            if (J$.Fr(650609))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(643969, J$.F(643961, J$.R(643937, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(643945, 'o', o, false, false), J$.R(643953, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(650617, J$e);
                        } finally {
                            if (J$.Fr(650625))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function xor(a, b) {
                jalangiLabel21:
                    while (true) {
                        try {
                            J$.Fe(645241, arguments.callee, this, arguments);
                            arguments = J$.N(645249, 'arguments', arguments, true, false, false);
                            a = J$.N(645257, 'a', a, true, false, false);
                            b = J$.N(645265, 'b', b, true, false, false);
                            J$.N(645273, 'length', length, false, false, false);
                            J$.N(645281, 'res', res, false, false, false);
                            J$.N(645289, 'i', i, false, false, false);
                            if (J$.C(37304, J$.U(80498, '!', J$.M(644905, J$.R(644889, 'Buffer', Buffer, false, true), 'isBuffer', false)(J$.R(644897, 'a', a, false, false)))))
                                a = J$.W(644937, 'a', J$.M(644929, J$.R(644913, 'Buffer', Buffer, false, true), 'from', false)(J$.R(644921, 'a', a, false, false)), a, false, false);
                            if (J$.C(37312, J$.U(80506, '!', J$.M(644961, J$.R(644945, 'Buffer', Buffer, false, true), 'isBuffer', false)(J$.R(644953, 'b', b, false, false)))))
                                b = J$.W(644993, 'b', J$.M(644985, J$.R(644969, 'Buffer', Buffer, false, true), 'from', false)(J$.R(644977, 'b', b, false, false)), b, false, false);
                            var length = J$.W(645049, 'length', J$.M(645041, J$.I(typeof Math === 'undefined' ? Math = J$.R(645001, 'Math', undefined, true, true) : Math = J$.R(645001, 'Math', Math, true, true)), 'max', false)(J$.G(645017, J$.R(645009, 'a', a, false, false), 'length'), J$.G(645033, J$.R(645025, 'b', b, false, false), 'length')), length, false, false);
                            var res = J$.W(645065, 'res', J$.T(645057, [], 10, false), res, false, false);
                            for (var i = J$.W(645081, 'i', J$.T(645073, 0, 22, false), i, false, false); J$.C(37320, J$.B(80514, '<', J$.R(645089, 'i', i, false, false), J$.R(645097, 'length', length, false, false))); i = J$.W(645121, 'i', J$.B(80522, '+', J$.R(645113, 'i', i, false, false), J$.T(645105, 1, 22, false)), i, false, false)) {
                                J$.M(645185, J$.R(645129, 'res', res, false, false), 'push', false)(J$.B(80530, '^', J$.G(645153, J$.R(645137, 'a', a, false, false), J$.R(645145, 'i', i, false, false)), J$.G(645177, J$.R(645161, 'b', b, false, false), J$.R(645169, 'i', i, false, false))));
                            }
                            return J$.Rt(645233, J$.M(645225, J$.M(645209, J$.R(645193, 'Buffer', Buffer, false, true), 'from', false)(J$.R(645201, 'res', res, false, false)), 'toString', false)(J$.T(645217, 'base64', 21, false)));
                        } catch (J$e) {
                            J$.Ex(650665, J$e);
                        } finally {
                            if (J$.Fr(650673))
                                continue jalangiLabel21;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function H(method, text) {
                jalangiLabel22:
                    while (true) {
                        try {
                            J$.Fe(645353, arguments.callee, this, arguments);
                            arguments = J$.N(645361, 'arguments', arguments, true, false, false);
                            method = J$.N(645369, 'method', method, true, false, false);
                            text = J$.N(645377, 'text', text, true, false, false);
                            return J$.Rt(645345, J$.M(645337, J$.M(645329, J$.M(645313, J$.R(645297, 'crypto', crypto, false, true), 'createHash', false)(J$.R(645305, 'method', method, false, false)), 'update', false)(J$.R(645321, 'text', text, false, false)), 'digest', false)());
                        } catch (J$e) {
                            J$.Ex(650681, J$e);
                        } finally {
                            if (J$.Fr(650689))
                                continue jalangiLabel22;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function HMAC(method, key, text) {
                jalangiLabel23:
                    while (true) {
                        try {
                            J$.Fe(645449, arguments.callee, this, arguments);
                            arguments = J$.N(645457, 'arguments', arguments, true, false, false);
                            method = J$.N(645465, 'method', method, true, false, false);
                            key = J$.N(645473, 'key', key, true, false, false);
                            text = J$.N(645481, 'text', text, true, false, false);
                            return J$.Rt(645441, J$.M(645433, J$.M(645425, J$.M(645409, J$.R(645385, 'crypto', crypto, false, true), 'createHmac', false)(J$.R(645393, 'method', method, false, false), J$.R(645401, 'key', key, false, false)), 'update', false)(J$.R(645417, 'text', text, false, false)), 'digest', false)());
                        } catch (J$e) {
                            J$.Ex(650697, J$e);
                        } finally {
                            if (J$.Fr(650705))
                                continue jalangiLabel23;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function HI(data, salt, iterations, cryptoMethod) {
                jalangiLabel25:
                    while (true) {
                        try {
                            J$.Fe(645937, arguments.callee, this, arguments);
                            arguments = J$.N(645945, 'arguments', arguments, true, false, false);
                            data = J$.N(645953, 'data', data, true, false, false);
                            salt = J$.N(645961, 'salt', salt, true, false, false);
                            iterations = J$.N(645969, 'iterations', iterations, true, false, false);
                            cryptoMethod = J$.N(645977, 'cryptoMethod', cryptoMethod, true, false, false);
                            J$.N(645985, 'key', key, false, false, false);
                            J$.N(645993, 'saltedData', saltedData, false, false, false);
                            var key = J$.W(645681, 'key', J$.M(645673, J$.T(645657, [
                                J$.R(645617, 'data', data, false, false),
                                J$.M(645641, J$.R(645625, 'salt', salt, false, false), 'toString', false)(J$.T(645633, 'base64', 21, false)),
                                J$.R(645649, 'iterations', iterations, false, false)
                            ], 10, false), 'join', false)(J$.T(645665, '_', 21, false)), key, false, false);
                            if (J$.C(37328, J$.B(80538, '!==', J$.G(645705, J$.R(645689, '_hiCache', _hiCache, false, true), J$.R(645697, 'key', key, false, false)), J$.T(645713, undefined, 24, false)))) {
                                return J$.Rt(645745, J$.G(645737, J$.R(645721, '_hiCache', _hiCache, false, true), J$.R(645729, 'key', key, false, false)));
                            }
                            var saltedData = J$.W(645825, 'saltedData', J$.M(645817, J$.R(645753, 'crypto', crypto, false, true), 'pbkdf2Sync', false)(J$.R(645761, 'data', data, false, false), J$.R(645769, 'salt', salt, false, false), J$.R(645777, 'iterations', iterations, false, false), J$.G(645801, J$.R(645785, 'hiLengthMap', hiLengthMap, false, true), J$.R(645793, 'cryptoMethod', cryptoMethod, false, false)), J$.R(645809, 'cryptoMethod', cryptoMethod, false, false)), saltedData, false, false);
                            if (J$.C(37336, J$.B(80546, '>=', J$.R(645833, '_hiCacheCount', _hiCacheCount, false, true), J$.T(645841, 200, 22, false)))) {
                                J$.F(645857, J$.R(645849, '_hiCachePurge', _hiCachePurge, false, true), false)();
                            }
                            J$.P(645889, J$.R(645865, '_hiCache', _hiCache, false, true), J$.R(645873, 'key', key, false, false), J$.R(645881, 'saltedData', saltedData, false, false));
                            _hiCacheCount = J$.W(645913, '_hiCacheCount', J$.B(80554, '+', J$.R(645905, '_hiCacheCount', _hiCacheCount, false, true), J$.T(645897, 1, 22, false)), _hiCacheCount, false, true);
                            return J$.Rt(645929, J$.R(645921, 'saltedData', saltedData, false, false));
                        } catch (J$e) {
                            J$.Ex(650729, J$e);
                        } finally {
                            if (J$.Fr(650737))
                                continue jalangiLabel25;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(649945, '_typeof2', J$.T(649937, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(649961, '_typeof', J$.T(649953, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(649977, '_classCallCheck', J$.T(649969, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(649993, '_defineProperties', J$.T(649985, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(650009, '_createClass', J$.T(650001, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(650025, '_possibleConstructorReturn', J$.T(650017, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(650041, '_assertThisInitialized', J$.T(650033, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(650057, '_get', J$.T(650049, _get, 12, false), true, false, false);
            _superPropBase = J$.N(650073, '_superPropBase', J$.T(650065, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(650089, '_getPrototypeOf', J$.T(650081, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(650105, '_inherits', J$.T(650097, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(650121, '_setPrototypeOf', J$.T(650113, _setPrototypeOf, 12, false), true, false, false);
            J$.N(650129, 'crypto', crypto, false, false, false);
            J$.N(650137, 'Buffer', Buffer, false, false, false);
            J$.N(650145, 'retrieveBSON', retrieveBSON, false, false, false);
            J$.N(650153, 'MongoError', MongoError, false, false, false);
            J$.N(650161, 'AuthProvider', AuthProvider, false, false, false);
            J$.N(650169, 'BSON', BSON, false, false, false);
            J$.N(650177, 'Binary', Binary, false, false, false);
            J$.N(650185, 'saslprep', saslprep, false, false, false);
            J$.N(650193, 'parsePayload', parsePayload, false, false, false);
            J$.N(650201, 'passwordDigest', passwordDigest, false, false, false);
            xor = J$.N(650217, 'xor', J$.T(650209, xor, 12, false), true, false, false);
            H = J$.N(650233, 'H', J$.T(650225, H, 12, false), true, false, false);
            HMAC = J$.N(650249, 'HMAC', J$.T(650241, HMAC, 12, false), true, false, false);
            J$.N(650257, '_hiCache', _hiCache, false, false, false);
            J$.N(650265, '_hiCacheCount', _hiCacheCount, false, false, false);
            J$.N(650273, '_hiCachePurge', _hiCachePurge, false, false, false);
            J$.N(650281, 'hiLengthMap', hiLengthMap, false, false, false);
            HI = J$.N(650297, 'HI', J$.T(650289, HI, 12, false), true, false, false);
            J$.N(650305, 'ScramSHA', ScramSHA, false, false, false);
            J$.N(650313, 'ScramSHA1', ScramSHA1, false, false, false);
            J$.N(650321, 'ScramSHA256', ScramSHA256, false, false, false);
            J$.T(641537, 'use strict', 21, false);
            var crypto = J$.W(644033, 'crypto', J$.F(644025, J$.I(typeof require === 'undefined' ? require = J$.R(644009, 'require', undefined, true, true) : require = J$.R(644009, 'require', require, true, true)), false)(J$.T(644017, 'crypto', 21, false)), crypto, false, true);
            var Buffer = J$.W(644073, 'Buffer', J$.G(644065, J$.F(644057, J$.I(typeof require === 'undefined' ? require = J$.R(644041, 'require', undefined, true, true) : require = J$.R(644041, 'require', require, true, true)), false)(J$.T(644049, 'safe-buffer', 21, false)), 'Buffer'), Buffer, false, true);
            var retrieveBSON = J$.W(644113, 'retrieveBSON', J$.G(644105, J$.F(644097, J$.I(typeof require === 'undefined' ? require = J$.R(644081, 'require', undefined, true, true) : require = J$.R(644081, 'require', require, true, true)), false)(J$.T(644089, '../connection/utils', 21, false)), 'retrieveBSON'), retrieveBSON, false, true);
            var MongoError = J$.W(644153, 'MongoError', J$.G(644145, J$.F(644137, J$.I(typeof require === 'undefined' ? require = J$.R(644121, 'require', undefined, true, true) : require = J$.R(644121, 'require', require, true, true)), false)(J$.T(644129, '../error', 21, false)), 'MongoError'), MongoError, false, true);
            var AuthProvider = J$.W(644193, 'AuthProvider', J$.G(644185, J$.F(644177, J$.I(typeof require === 'undefined' ? require = J$.R(644161, 'require', undefined, true, true) : require = J$.R(644161, 'require', require, true, true)), false)(J$.T(644169, './auth_provider', 21, false)), 'AuthProvider'), AuthProvider, false, true);
            var BSON = J$.W(644217, 'BSON', J$.F(644209, J$.R(644201, 'retrieveBSON', retrieveBSON, false, true), false)(), BSON, false, true);
            var Binary = J$.W(644241, 'Binary', J$.G(644233, J$.R(644225, 'BSON', BSON, false, true), 'Binary'), Binary, false, true);
            var saslprep;
            try {
                saslprep = J$.W(644273, 'saslprep', J$.F(644265, J$.I(typeof require === 'undefined' ? require = J$.R(644249, 'require', undefined, true, true) : require = J$.R(644249, 'require', require, true, true)), false)(J$.T(644257, 'saslprep', 21, false)), saslprep, false, true);
            } catch (e) {
                J$.N(644281, 'e', e, false, false, true);
            }
            var parsePayload = J$.W(644585, 'parsePayload', J$.T(644577, function parsePayload(payload) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(644521, arguments.callee, this, arguments);
                            arguments = J$.N(644529, 'arguments', arguments, true, false, false);
                            payload = J$.N(644537, 'payload', payload, true, false, false);
                            J$.N(644545, 'dict', dict, false, false, false);
                            J$.N(644553, 'parts', parts, false, false, false);
                            J$.N(644561, 'i', i, false, false, false);
                            J$.N(644569, 'valueParts', valueParts, false, false, false);
                            var dict = J$.W(644297, 'dict', J$.T(644289, {}, 11, false), dict, false, false);
                            var parts = J$.W(644329, 'parts', J$.M(644321, J$.R(644305, 'payload', payload, false, false), 'split', false)(J$.T(644313, ',', 21, false)), parts, false, false);
                            for (var i = J$.W(644345, 'i', J$.T(644337, 0, 22, false), i, false, false); J$.C(37272, J$.B(80410, '<', J$.R(644353, 'i', i, false, false), J$.G(644369, J$.R(644361, 'parts', parts, false, false), 'length'))); J$.B(80434, '-', i = J$.W(644385, 'i', J$.B(80426, '+', J$.U(80418, '+', J$.R(644377, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var valueParts = J$.W(644433, 'valueParts', J$.M(644425, J$.G(644409, J$.R(644393, 'parts', parts, false, false), J$.R(644401, 'i', i, false, false)), 'split', false)(J$.T(644417, '=', 21, false)), valueParts, false, false);
                                J$.P(644497, J$.R(644441, 'dict', dict, false, false), J$.G(644465, J$.R(644449, 'valueParts', valueParts, false, false), J$.T(644457, 0, 22, false)), J$.G(644489, J$.R(644473, 'valueParts', valueParts, false, false), J$.T(644481, 1, 22, false)));
                            }
                            return J$.Rt(644513, J$.R(644505, 'dict', dict, false, false));
                        } catch (J$e) {
                            J$.Ex(650633, J$e);
                        } finally {
                            if (J$.Fr(650641))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), parsePayload, false, true);
            var passwordDigest = J$.W(644881, 'passwordDigest', J$.T(644873, function passwordDigest(username, password) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(644833, arguments.callee, this, arguments);
                            arguments = J$.N(644841, 'arguments', arguments, true, false, false);
                            username = J$.N(644849, 'username', username, true, false, false);
                            password = J$.N(644857, 'password', password, true, false, false);
                            J$.N(644865, 'md5', md5, false, false, false);
                            if (J$.C(37280, J$.B(80450, '!==', J$.U(80442, 'typeof', J$.R(644593, 'username', username, false, false)), J$.T(644601, 'string', 21, false))))
                                throw J$.F(644625, J$.R(644609, 'MongoError', MongoError, false, true), true)(J$.T(644617, 'username must be a string', 21, false));
                            if (J$.C(37288, J$.B(80466, '!==', J$.U(80458, 'typeof', J$.R(644633, 'password', password, false, false)), J$.T(644641, 'string', 21, false))))
                                throw J$.F(644665, J$.R(644649, 'MongoError', MongoError, false, true), true)(J$.T(644657, 'password must be a string', 21, false));
                            if (J$.C(37296, J$.B(80474, '===', J$.G(644681, J$.R(644673, 'password', password, false, false), 'length'), J$.T(644689, 0, 22, false))))
                                throw J$.F(644713, J$.R(644697, 'MongoError', MongoError, false, true), true)(J$.T(644705, 'password cannot be empty', 21, false));
                            var md5 = J$.W(644745, 'md5', J$.M(644737, J$.R(644721, 'crypto', crypto, false, true), 'createHash', false)(J$.T(644729, 'md5', 21, false)), md5, false, false);
                            J$.M(644793, J$.R(644753, 'md5', md5, false, false), 'update', false)(J$.B(80490, '+', J$.B(80482, '+', J$.R(644761, 'username', username, false, false), J$.T(644769, ':mongo:', 21, false)), J$.R(644777, 'password', password, false, false)), J$.T(644785, 'utf8', 21, false));
                            return J$.Rt(644825, J$.M(644817, J$.R(644801, 'md5', md5, false, false), 'digest', false)(J$.T(644809, 'hex', 21, false)));
                        } catch (J$e) {
                            J$.Ex(650649, J$e);
                        } finally {
                            if (J$.Fr(650657))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), passwordDigest, false, true);
            var _hiCache = J$.W(645497, '_hiCache', J$.T(645489, {}, 11, false), _hiCache, false, true);
            var _hiCacheCount = J$.W(645513, '_hiCacheCount', J$.T(645505, 0, 22, false), _hiCacheCount, false, true);
            var _hiCachePurge = J$.W(645577, '_hiCachePurge', J$.T(645569, function _hiCachePurge() {
                jalangiLabel24:
                    while (true) {
                        try {
                            J$.Fe(645553, arguments.callee, this, arguments);
                            arguments = J$.N(645561, 'arguments', arguments, true, false, false);
                            _hiCache = J$.W(645529, '_hiCache', J$.T(645521, {}, 11, false), _hiCache, false, true);
                            _hiCacheCount = J$.W(645545, '_hiCacheCount', J$.T(645537, 0, 22, false), _hiCacheCount, false, true);
                        } catch (J$e) {
                            J$.Ex(650713, J$e);
                        } finally {
                            if (J$.Fr(650721))
                                continue jalangiLabel24;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), _hiCachePurge, false, true);
            var hiLengthMap = J$.W(645609, 'hiLengthMap', J$.T(645601, {
                sha256: J$.T(645585, 32, 22, false),
                sha1: J$.T(645593, 20, 22, false)
            }, 11, false), hiLengthMap, false, true);
            var ScramSHA = J$.W(649353, 'ScramSHA', J$.F(649345, J$.T(649329, function (_AuthProvider) {
                jalangiLabel35:
                    while (true) {
                        try {
                            J$.Fe(649289, arguments.callee, this, arguments);
                            function ScramSHA(bson, cryptoMethod) {
                                jalangiLabel26:
                                    while (true) {
                                        try {
                                            J$.Fe(646193, arguments.callee, this, arguments);
                                            arguments = J$.N(646201, 'arguments', arguments, true, false, false);
                                            bson = J$.N(646209, 'bson', bson, true, false, false);
                                            cryptoMethod = J$.N(646217, 'cryptoMethod', cryptoMethod, true, false, false);
                                            J$.N(646225, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(646057, J$.R(646033, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(646041, 'this', this, false, false), J$.R(646049, 'ScramSHA', ScramSHA, false, false));
                                            _this = J$.W(646137, '_this', J$.F(646129, J$.R(646065, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(646073, 'this', this, false, false), J$.M(646121, J$.F(646097, J$.R(646081, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(646089, 'ScramSHA', ScramSHA, false, false)), 'call', false)(J$.R(646105, 'this', this, false, false), J$.R(646113, 'bson', bson, false, false))), _this, false, false);
                                            J$.P(646169, J$.R(646145, '_this', _this, false, false), 'cryptoMethod', J$.C(37344, J$.R(646153, 'cryptoMethod', cryptoMethod, false, false)) ? J$._() : J$.T(646161, 'sha1', 21, false));
                                            return J$.Rt(646185, J$.R(646177, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(650745, J$e);
                                        } finally {
                                            if (J$.Fr(650753))
                                                continue jalangiLabel26;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(649297, 'arguments', arguments, true, false, false);
                            _AuthProvider = J$.N(649305, '_AuthProvider', _AuthProvider, true, false, false);
                            ScramSHA = J$.N(649321, 'ScramSHA', J$.T(649313, ScramSHA, 12, false), true, false, false);
                            J$.F(646025, J$.R(646001, '_inherits', _inherits, false, true), false)(J$.R(646009, 'ScramSHA', ScramSHA, false, false), J$.R(646017, '_AuthProvider', _AuthProvider, false, false));
                            J$.F(649265, J$.R(646233, '_createClass', _createClass, false, true), false)(J$.R(646241, 'ScramSHA', ScramSHA, false, false), J$.T(649105, [
                                J$.T(648489, {
                                    key: J$.T(646249, '_executeScram', 21, false),
                                    value: J$.T(648481, function _executeScram(sendAuthCommand, connection, credentials, nonce, callback) {
                                        jalangiLabel29:
                                            while (true) {
                                                try {
                                                    J$.Fe(648361, arguments.callee, this, arguments);
                                                    arguments = J$.N(648369, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(648377, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connection = J$.N(648385, 'connection', connection, true, false, false);
                                                    credentials = J$.N(648393, 'credentials', credentials, true, false, false);
                                                    nonce = J$.N(648401, 'nonce', nonce, true, false, false);
                                                    callback = J$.N(648409, 'callback', callback, true, false, false);
                                                    J$.N(648417, 'username', username, false, false, false);
                                                    J$.N(648425, 'password', password, false, false, false);
                                                    J$.N(648433, 'db', db, false, false, false);
                                                    J$.N(648441, 'cryptoMethod', cryptoMethod, false, false, false);
                                                    J$.N(648449, 'mechanism', mechanism, false, false, false);
                                                    J$.N(648457, 'processedPassword', processedPassword, false, false, false);
                                                    J$.N(648465, 'firstBare', firstBare, false, false, false);
                                                    J$.N(648473, 'saslStartCmd', saslStartCmd, false, false, false);
                                                    var username = J$.W(646273, 'username', J$.G(646265, J$.R(646257, 'credentials', credentials, false, false), 'username'), username, false, false);
                                                    var password = J$.W(646297, 'password', J$.G(646289, J$.R(646281, 'credentials', credentials, false, false), 'password'), password, false, false);
                                                    var db = J$.W(646321, 'db', J$.G(646313, J$.R(646305, 'credentials', credentials, false, false), 'source'), db, false, false);
                                                    var cryptoMethod = J$.W(646345, 'cryptoMethod', J$.G(646337, J$.R(646329, 'this', this, false, false), 'cryptoMethod'), cryptoMethod, false, false);
                                                    var mechanism = J$.W(646361, 'mechanism', J$.T(646353, 'SCRAM-SHA-1', 21, false), mechanism, false, false);
                                                    var processedPassword;
                                                    if (J$.C(37360, J$.B(80562, '===', J$.R(646369, 'cryptoMethod', cryptoMethod, false, false), J$.T(646377, 'sha256', 21, false)))) {
                                                        mechanism = J$.W(646393, 'mechanism', J$.T(646385, 'SCRAM-SHA-256', 21, false), mechanism, false, false);
                                                        processedPassword = J$.W(646441, 'processedPassword', J$.C(37352, J$.R(646401, 'saslprep', saslprep, false, true)) ? J$.F(646425, J$.R(646409, 'saslprep', saslprep, false, true), false)(J$.R(646417, 'password', password, false, false)) : J$.R(646433, 'password', password, false, false), processedPassword, false, false);
                                                    } else {
                                                        try {
                                                            processedPassword = J$.W(646481, 'processedPassword', J$.F(646473, J$.R(646449, 'passwordDigest', passwordDigest, false, true), false)(J$.R(646457, 'username', username, false, false), J$.R(646465, 'password', password, false, false)), processedPassword, false, false);
                                                        } catch (e) {
                                                            J$.N(646521, 'e', e, false, false, true);
                                                            return J$.Rt(646513, J$.F(646505, J$.R(646489, 'callback', callback, false, false), false)(J$.R(646497, 'e', e, false, false)));
                                                        }
                                                    }
                                                    username = J$.W(646585, 'username', J$.M(646577, J$.M(646553, J$.R(646529, 'username', username, false, false), 'replace', false)(J$.T(646537, '=', 21, false), J$.T(646545, '=3D', 21, false)), 'replace', false)(J$.T(646561, ',', 21, false), J$.T(646569, '=2C', 21, false)), username, false, false);
                                                    var firstBare = J$.W(646745, 'firstBare', J$.M(646737, J$.R(646593, 'Buffer', Buffer, false, true), 'concat', false)(J$.T(646729, [
                                                        J$.M(646625, J$.R(646601, 'Buffer', Buffer, false, true), 'from', false)(J$.T(646609, 'n=', 21, false), J$.T(646617, 'utf8', 21, false)),
                                                        J$.M(646657, J$.R(646633, 'Buffer', Buffer, false, true), 'from', false)(J$.R(646641, 'username', username, false, false), J$.T(646649, 'utf8', 21, false)),
                                                        J$.M(646689, J$.R(646665, 'Buffer', Buffer, false, true), 'from', false)(J$.T(646673, ',r=', 21, false), J$.T(646681, 'utf8', 21, false)),
                                                        J$.M(646721, J$.R(646697, 'Buffer', Buffer, false, true), 'from', false)(J$.R(646705, 'nonce', nonce, false, false), J$.T(646713, 'utf8', 21, false))
                                                    ], 10, false)), firstBare, false, false);
                                                    var saslStartCmd = J$.W(646865, 'saslStartCmd', J$.T(646857, {
                                                        saslStart: J$.T(646753, 1, 22, false),
                                                        mechanism: J$.R(646761, 'mechanism', mechanism, false, false),
                                                        payload: J$.F(646841, J$.R(646769, 'Binary', Binary, false, true), true)(J$.M(646833, J$.R(646777, 'Buffer', Buffer, false, true), 'concat', false)(J$.T(646825, [
                                                            J$.M(646809, J$.R(646785, 'Buffer', Buffer, false, true), 'from', false)(J$.T(646793, 'n,,', 21, false), J$.T(646801, 'utf8', 21, false)),
                                                            J$.R(646817, 'firstBare', firstBare, false, false)
                                                        ], 10, false))),
                                                        autoAuthorize: J$.T(646849, 1, 22, false)
                                                    }, 11, false), saslStartCmd, false, false);
                                                    J$.F(648353, J$.R(646873, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(646881, 'connection', connection, false, false), J$.M(646913, J$.T(646889, '', 21, false), 'concat', false)(J$.R(646897, 'db', db, false, false), J$.T(646905, '.$cmd', 21, false)), J$.R(646921, 'saslStartCmd', saslStartCmd, false, false), J$.T(648345, function (err, r) {
                                                        jalangiLabel28:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(648185, arguments.callee, this, arguments);
                                                                    arguments = J$.N(648193, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(648201, 'err', err, true, false, false);
                                                                    r = J$.N(648209, 'r', r, true, false, false);
                                                                    J$.N(648217, 'tmpError', tmpError, false, false, false);
                                                                    J$.N(648225, 'payload', payload, false, false, false);
                                                                    J$.N(648233, 'dict', dict, false, false, false);
                                                                    J$.N(648241, 'iterations', iterations, false, false, false);
                                                                    J$.N(648249, 'salt', salt, false, false, false);
                                                                    J$.N(648257, 'rnonce', rnonce, false, false, false);
                                                                    J$.N(648265, 'withoutProof', withoutProof, false, false, false);
                                                                    J$.N(648273, 'saltedPassword', saltedPassword, false, false, false);
                                                                    J$.N(648281, 'error', error, false, false, false);
                                                                    J$.N(648289, 'clientKey', clientKey, false, false, false);
                                                                    J$.N(648297, 'storedKey', storedKey, false, false, false);
                                                                    J$.N(648305, 'authMessage', authMessage, false, false, false);
                                                                    J$.N(648313, 'clientSignature', clientSignature, false, false, false);
                                                                    J$.N(648321, 'clientProof', clientProof, false, false, false);
                                                                    J$.N(648329, 'clientFinal', clientFinal, false, false, false);
                                                                    J$.N(648337, 'saslContinueCmd', saslContinueCmd, false, false, false);
                                                                    var tmpError = J$.W(646961, 'tmpError', J$.M(646953, J$.R(646929, 'ScramSHA', ScramSHA, false, false), '_getError', false)(J$.R(646937, 'err', err, false, false), J$.R(646945, 'r', r, false, false)), tmpError, false, false);
                                                                    if (J$.C(37368, J$.R(646969, 'tmpError', tmpError, false, false))) {
                                                                        return J$.Rt(647009, J$.F(647001, J$.R(646977, 'callback', callback, false, false), false)(J$.R(646985, 'tmpError', tmpError, false, false), J$.T(646993, null, 25, false)));
                                                                    }
                                                                    var payload = J$.W(647097, 'payload', J$.C(37376, J$.M(647041, J$.R(647017, 'Buffer', Buffer, false, true), 'isBuffer', false)(J$.G(647033, J$.R(647025, 'r', r, false, false), 'payload'))) ? J$.F(647073, J$.R(647049, 'Binary', Binary, false, true), true)(J$.G(647065, J$.R(647057, 'r', r, false, false), 'payload')) : J$.G(647089, J$.R(647081, 'r', r, false, false), 'payload'), payload, false, false);
                                                                    var dict = J$.W(647137, 'dict', J$.F(647129, J$.R(647105, 'parsePayload', parsePayload, false, true), false)(J$.M(647121, J$.R(647113, 'payload', payload, false, false), 'value', false)()), dict, false, false);
                                                                    var iterations = J$.W(647185, 'iterations', J$.F(647177, J$.I(typeof parseInt === 'undefined' ? parseInt = J$.R(647145, 'parseInt', undefined, true, true) : parseInt = J$.R(647145, 'parseInt', parseInt, true, true)), false)(J$.G(647161, J$.R(647153, 'dict', dict, false, false), 'i'), J$.T(647169, 10, 22, false)), iterations, false, false);
                                                                    var salt = J$.W(647209, 'salt', J$.G(647201, J$.R(647193, 'dict', dict, false, false), 's'), salt, false, false);
                                                                    var rnonce = J$.W(647233, 'rnonce', J$.G(647225, J$.R(647217, 'dict', dict, false, false), 'r'), rnonce, false, false);
                                                                    var withoutProof = J$.W(647265, 'withoutProof', J$.M(647257, J$.T(647241, 'c=biws,r=', 21, false), 'concat', false)(J$.R(647249, 'rnonce', rnonce, false, false)), withoutProof, false, false);
                                                                    var saltedPassword = J$.W(647345, 'saltedPassword', J$.F(647337, J$.R(647273, 'HI', HI, false, true), false)(J$.R(647281, 'processedPassword', processedPassword, false, false), J$.M(647313, J$.R(647289, 'Buffer', Buffer, false, true), 'from', false)(J$.R(647297, 'salt', salt, false, false), J$.T(647305, 'base64', 21, false)), J$.R(647321, 'iterations', iterations, false, false), J$.R(647329, 'cryptoMethod', cryptoMethod, false, false)), saltedPassword, false, false);
                                                                    if (J$.C(37392, J$.C(37384, J$.R(647353, 'iterations', iterations, false, false)) ? J$.B(80570, '<', J$.R(647361, 'iterations', iterations, false, false), J$.T(647369, 4096, 22, false)) : J$._())) {
                                                                        var error = J$.W(647417, 'error', J$.F(647409, J$.R(647377, 'MongoError', MongoError, false, true), true)(J$.M(647401, J$.T(647385, 'Server returned an invalid iteration count ', 21, false), 'concat', false)(J$.R(647393, 'iterations', iterations, false, false))), error, false, false);
                                                                        return J$.Rt(647457, J$.F(647449, J$.R(647425, 'callback', callback, false, false), false)(J$.R(647433, 'error', error, false, false), J$.T(647441, false, 23, false)));
                                                                    }
                                                                    var clientKey = J$.W(647505, 'clientKey', J$.F(647497, J$.R(647465, 'HMAC', HMAC, false, true), false)(J$.R(647473, 'cryptoMethod', cryptoMethod, false, false), J$.R(647481, 'saltedPassword', saltedPassword, false, false), J$.T(647489, 'Client Key', 21, false)), clientKey, false, false);
                                                                    var storedKey = J$.W(647545, 'storedKey', J$.F(647537, J$.R(647513, 'H', H, false, true), false)(J$.R(647521, 'cryptoMethod', cryptoMethod, false, false), J$.R(647529, 'clientKey', clientKey, false, false)), storedKey, false, false);
                                                                    var authMessage = J$.W(647625, 'authMessage', J$.M(647617, J$.T(647601, [
                                                                        J$.R(647553, 'firstBare', firstBare, false, false),
                                                                        J$.M(647585, J$.M(647569, J$.R(647561, 'payload', payload, false, false), 'value', false)(), 'toString', false)(J$.T(647577, 'base64', 21, false)),
                                                                        J$.R(647593, 'withoutProof', withoutProof, false, false)
                                                                    ], 10, false), 'join', false)(J$.T(647609, ',', 21, false)), authMessage, false, false);
                                                                    var clientSignature = J$.W(647673, 'clientSignature', J$.F(647665, J$.R(647633, 'HMAC', HMAC, false, true), false)(J$.R(647641, 'cryptoMethod', cryptoMethod, false, false), J$.R(647649, 'storedKey', storedKey, false, false), J$.R(647657, 'authMessage', authMessage, false, false)), clientSignature, false, false);
                                                                    var clientProof = J$.W(647729, 'clientProof', J$.M(647721, J$.T(647681, 'p=', 21, false), 'concat', false)(J$.F(647713, J$.R(647689, 'xor', xor, false, true), false)(J$.R(647697, 'clientKey', clientKey, false, false), J$.R(647705, 'clientSignature', clientSignature, false, false))), clientProof, false, false);
                                                                    var clientFinal = J$.W(647777, 'clientFinal', J$.M(647769, J$.T(647753, [
                                                                        J$.R(647737, 'withoutProof', withoutProof, false, false),
                                                                        J$.R(647745, 'clientProof', clientProof, false, false)
                                                                    ], 10, false), 'join', false)(J$.T(647761, ',', 21, false)), clientFinal, false, false);
                                                                    var saslContinueCmd = J$.W(647857, 'saslContinueCmd', J$.T(647849, {
                                                                        saslContinue: J$.T(647785, 1, 22, false),
                                                                        conversationId: J$.G(647801, J$.R(647793, 'r', r, false, false), 'conversationId'),
                                                                        payload: J$.F(647841, J$.R(647809, 'Binary', Binary, false, true), true)(J$.M(647833, J$.R(647817, 'Buffer', Buffer, false, true), 'from', false)(J$.R(647825, 'clientFinal', clientFinal, false, false)))
                                                                    }, 11, false), saslContinueCmd, false, false);
                                                                    J$.F(648177, J$.R(647865, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(647873, 'connection', connection, false, false), J$.M(647905, J$.T(647881, '', 21, false), 'concat', false)(J$.R(647889, 'db', db, false, false), J$.T(647897, '.$cmd', 21, false)), J$.R(647913, 'saslContinueCmd', saslContinueCmd, false, false), J$.T(648169, function (err, r) {
                                                                        jalangiLabel27:
                                                                            while (true) {
                                                                                try {
                                                                                    J$.Fe(648129, arguments.callee, this, arguments);
                                                                                    arguments = J$.N(648137, 'arguments', arguments, true, false, false);
                                                                                    err = J$.N(648145, 'err', err, true, false, false);
                                                                                    r = J$.N(648153, 'r', r, true, false, false);
                                                                                    J$.N(648161, 'retrySaslContinueCmd', retrySaslContinueCmd, false, false, false);
                                                                                    if (J$.C(37408, J$.C(37400, J$.U(80578, '!', J$.R(647921, 'r', r, false, false))) ? J$._() : J$.B(80586, '!==', J$.G(647937, J$.R(647929, 'r', r, false, false), 'done'), J$.T(647945, false, 23, false)))) {
                                                                                        return J$.Rt(647985, J$.F(647977, J$.R(647953, 'callback', callback, false, false), false)(J$.R(647961, 'err', err, false, false), J$.R(647969, 'r', r, false, false)));
                                                                                    }
                                                                                    var retrySaslContinueCmd = J$.W(648049, 'retrySaslContinueCmd', J$.T(648041, {
                                                                                        saslContinue: J$.T(647993, 1, 22, false),
                                                                                        conversationId: J$.G(648009, J$.R(648001, 'r', r, false, false), 'conversationId'),
                                                                                        payload: J$.M(648033, J$.R(648017, 'Buffer', Buffer, false, true), 'alloc', false)(J$.T(648025, 0, 22, false))
                                                                                    }, 11, false), retrySaslContinueCmd, false, false);
                                                                                    J$.F(648121, J$.R(648057, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(648065, 'connection', connection, false, false), J$.M(648097, J$.T(648073, '', 21, false), 'concat', false)(J$.R(648081, 'db', db, false, false), J$.T(648089, '.$cmd', 21, false)), J$.R(648105, 'retrySaslContinueCmd', retrySaslContinueCmd, false, false), J$.R(648113, 'callback', callback, false, false));
                                                                                } catch (J$e) {
                                                                                    J$.Ex(650761, J$e);
                                                                                } finally {
                                                                                    if (J$.Fr(650769))
                                                                                        continue jalangiLabel27;
                                                                                    else
                                                                                        return J$.Ra();
                                                                                }
                                                                            }
                                                                    }, 12, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(650777, J$e);
                                                                } finally {
                                                                    if (J$.Fr(650785))
                                                                        continue jalangiLabel28;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(650793, J$e);
                                                } finally {
                                                    if (J$.Fr(650801))
                                                        continue jalangiLabel29;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(648777, {
                                    key: J$.T(648497, '_authenticateSingleConnection', 21, false),
                                    value: J$.T(648769, function _authenticateSingleConnection(sendAuthCommand, connection, credentials, callback) {
                                        jalangiLabel31:
                                            while (true) {
                                                try {
                                                    J$.Fe(648713, arguments.callee, this, arguments);
                                                    arguments = J$.N(648721, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(648729, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connection = J$.N(648737, 'connection', connection, true, false, false);
                                                    credentials = J$.N(648745, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(648753, 'callback', callback, true, false, false);
                                                    J$.N(648761, '_this2', _this2, false, false, false);
                                                    var _this2 = J$.W(648513, '_this2', J$.R(648505, 'this', this, false, false), _this2, false, false);
                                                    J$.M(648705, J$.R(648521, 'crypto', crypto, false, true), 'randomBytes', false)(J$.T(648529, 24, 22, false), J$.T(648697, function (err, buff) {
                                                        jalangiLabel30:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(648665, arguments.callee, this, arguments);
                                                                    arguments = J$.N(648673, 'arguments', arguments, true, false, false);
                                                                    err = J$.N(648681, 'err', err, true, false, false);
                                                                    buff = J$.N(648689, 'buff', buff, true, false, false);
                                                                    if (J$.C(37416, J$.R(648537, 'err', err, false, false))) {
                                                                        return J$.Rt(648577, J$.F(648569, J$.R(648545, 'callback', callback, false, false), false)(J$.R(648553, 'err', err, false, false), J$.T(648561, null, 25, false)));
                                                                    }
                                                                    return J$.Rt(648657, J$.M(648649, J$.R(648585, '_this2', _this2, false, false), '_executeScram', false)(J$.R(648593, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(648601, 'connection', connection, false, false), J$.R(648609, 'credentials', credentials, false, false), J$.M(648633, J$.R(648617, 'buff', buff, false, false), 'toString', false)(J$.T(648625, 'base64', 21, false)), J$.R(648641, 'callback', callback, false, false)));
                                                                } catch (J$e) {
                                                                    J$.Ex(650809, J$e);
                                                                } finally {
                                                                    if (J$.Fr(650817))
                                                                        continue jalangiLabel30;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                } catch (J$e) {
                                                    J$.Ex(650825, J$e);
                                                } finally {
                                                    if (J$.Fr(650833))
                                                        continue jalangiLabel31;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(648977, {
                                    key: J$.T(648785, 'auth', 21, false),
                                    value: J$.T(648969, function auth(sendAuthCommand, connections, credentials, callback) {
                                        jalangiLabel32:
                                            while (true) {
                                                try {
                                                    J$.Fe(648921, arguments.callee, this, arguments);
                                                    arguments = J$.N(648929, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(648937, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connections = J$.N(648945, 'connections', connections, true, false, false);
                                                    credentials = J$.N(648953, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(648961, 'callback', callback, true, false, false);
                                                    J$.M(648801, J$.R(648793, 'this', this, false, false), '_checkSaslprep', false)();
                                                    J$.M(648913, J$.F(648865, J$.R(648809, '_get', _get, false, true), false)(J$.F(648841, J$.R(648817, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(648833, J$.R(648825, 'ScramSHA', ScramSHA, false, false), 'prototype')), J$.T(648849, 'auth', 21, false), J$.R(648857, 'this', this, false, false)), 'call', false)(J$.R(648873, 'this', this, false, false), J$.R(648881, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(648889, 'connections', connections, false, false), J$.R(648897, 'credentials', credentials, false, false), J$.R(648905, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(650841, J$e);
                                                } finally {
                                                    if (J$.Fr(650849))
                                                        continue jalangiLabel32;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(649097, {
                                    key: J$.T(648985, '_checkSaslprep', 21, false),
                                    value: J$.T(649089, function _checkSaslprep() {
                                        jalangiLabel33:
                                            while (true) {
                                                try {
                                                    J$.Fe(649065, arguments.callee, this, arguments);
                                                    arguments = J$.N(649073, 'arguments', arguments, true, false, false);
                                                    J$.N(649081, 'cryptoMethod', cryptoMethod, false, false, false);
                                                    var cryptoMethod = J$.W(649009, 'cryptoMethod', J$.G(649001, J$.R(648993, 'this', this, false, false), 'cryptoMethod'), cryptoMethod, false, false);
                                                    if (J$.C(37432, J$.B(80594, '===', J$.R(649017, 'cryptoMethod', cryptoMethod, false, false), J$.T(649025, 'sha256', 21, false)))) {
                                                        if (J$.C(37424, J$.U(80602, '!', J$.R(649033, 'saslprep', saslprep, false, true)))) {
                                                            J$.M(649057, J$.I(typeof console === 'undefined' ? console = J$.R(649041, 'console', undefined, true, true) : console = J$.R(649041, 'console', console, true, true)), 'warn', false)(J$.T(649049, 'Warning: no saslprep library specified. Passwords will not be sanitized', 21, false));
                                                        }
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(650857, J$e);
                                                } finally {
                                                    if (J$.Fr(650865))
                                                        continue jalangiLabel33;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false), J$.T(649257, [J$.T(649249, {
                                    key: J$.T(649113, '_getError', 21, false),
                                    value: J$.T(649241, function _getError(err, r) {
                                        jalangiLabel34:
                                            while (true) {
                                                try {
                                                    J$.Fe(649209, arguments.callee, this, arguments);
                                                    arguments = J$.N(649217, 'arguments', arguments, true, false, false);
                                                    err = J$.N(649225, 'err', err, true, false, false);
                                                    r = J$.N(649233, 'r', r, true, false, false);
                                                    if (J$.C(37440, J$.R(649121, 'err', err, false, false))) {
                                                        return J$.Rt(649137, J$.R(649129, 'err', err, false, false));
                                                    }
                                                    if (J$.C(37456, J$.C(37448, J$.G(649153, J$.R(649145, 'r', r, false, false), '$err')) ? J$._() : J$.G(649169, J$.R(649161, 'r', r, false, false), 'errmsg'))) {
                                                        return J$.Rt(649201, J$.F(649193, J$.R(649177, 'MongoError', MongoError, false, true), true)(J$.R(649185, 'r', r, false, false)));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(650873, J$e);
                                                } finally {
                                                    if (J$.Fr(650881))
                                                        continue jalangiLabel34;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(649281, J$.R(649273, 'ScramSHA', ScramSHA, false, false));
                        } catch (J$e) {
                            J$.Ex(650889, J$e);
                        } finally {
                            if (J$.Fr(650897))
                                continue jalangiLabel35;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(649337, 'AuthProvider', AuthProvider, false, true)), ScramSHA, false, true);
            var ScramSHA1 = J$.W(649617, 'ScramSHA1', J$.F(649609, J$.T(649593, function (_ScramSHA) {
                jalangiLabel37:
                    while (true) {
                        try {
                            J$.Fe(649553, arguments.callee, this, arguments);
                            function ScramSHA1(bson) {
                                jalangiLabel36:
                                    while (true) {
                                        try {
                                            J$.Fe(649513, arguments.callee, this, arguments);
                                            arguments = J$.N(649521, 'arguments', arguments, true, false, false);
                                            bson = J$.N(649529, 'bson', bson, true, false, false);
                                            J$.F(649417, J$.R(649393, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(649401, 'this', this, false, false), J$.R(649409, 'ScramSHA1', ScramSHA1, false, false));
                                            return J$.Rt(649505, J$.F(649497, J$.R(649425, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(649433, 'this', this, false, false), J$.M(649489, J$.F(649457, J$.R(649441, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(649449, 'ScramSHA1', ScramSHA1, false, false)), 'call', false)(J$.R(649465, 'this', this, false, false), J$.R(649473, 'bson', bson, false, false), J$.T(649481, 'sha1', 21, false))));
                                        } catch (J$e) {
                                            J$.Ex(650905, J$e);
                                        } finally {
                                            if (J$.Fr(650913))
                                                continue jalangiLabel36;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(649561, 'arguments', arguments, true, false, false);
                            _ScramSHA = J$.N(649569, '_ScramSHA', _ScramSHA, true, false, false);
                            ScramSHA1 = J$.N(649585, 'ScramSHA1', J$.T(649577, ScramSHA1, 12, false), true, false, false);
                            J$.F(649385, J$.R(649361, '_inherits', _inherits, false, true), false)(J$.R(649369, 'ScramSHA1', ScramSHA1, false, false), J$.R(649377, '_ScramSHA', _ScramSHA, false, false));
                            return J$.Rt(649545, J$.R(649537, 'ScramSHA1', ScramSHA1, false, false));
                        } catch (J$e) {
                            J$.Ex(650921, J$e);
                        } finally {
                            if (J$.Fr(650929))
                                continue jalangiLabel37;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(649601, 'ScramSHA', ScramSHA, false, true)), ScramSHA1, false, true);
            var ScramSHA256 = J$.W(649881, 'ScramSHA256', J$.F(649873, J$.T(649857, function (_ScramSHA2) {
                jalangiLabel39:
                    while (true) {
                        try {
                            J$.Fe(649817, arguments.callee, this, arguments);
                            function ScramSHA256(bson) {
                                jalangiLabel38:
                                    while (true) {
                                        try {
                                            J$.Fe(649777, arguments.callee, this, arguments);
                                            arguments = J$.N(649785, 'arguments', arguments, true, false, false);
                                            bson = J$.N(649793, 'bson', bson, true, false, false);
                                            J$.F(649681, J$.R(649657, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(649665, 'this', this, false, false), J$.R(649673, 'ScramSHA256', ScramSHA256, false, false));
                                            return J$.Rt(649769, J$.F(649761, J$.R(649689, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(649697, 'this', this, false, false), J$.M(649753, J$.F(649721, J$.R(649705, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(649713, 'ScramSHA256', ScramSHA256, false, false)), 'call', false)(J$.R(649729, 'this', this, false, false), J$.R(649737, 'bson', bson, false, false), J$.T(649745, 'sha256', 21, false))));
                                        } catch (J$e) {
                                            J$.Ex(650937, J$e);
                                        } finally {
                                            if (J$.Fr(650945))
                                                continue jalangiLabel38;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(649825, 'arguments', arguments, true, false, false);
                            _ScramSHA2 = J$.N(649833, '_ScramSHA2', _ScramSHA2, true, false, false);
                            ScramSHA256 = J$.N(649849, 'ScramSHA256', J$.T(649841, ScramSHA256, 12, false), true, false, false);
                            J$.F(649649, J$.R(649625, '_inherits', _inherits, false, true), false)(J$.R(649633, 'ScramSHA256', ScramSHA256, false, false), J$.R(649641, '_ScramSHA2', _ScramSHA2, false, false));
                            return J$.Rt(649809, J$.R(649801, 'ScramSHA256', ScramSHA256, false, false));
                        } catch (J$e) {
                            J$.Ex(650953, J$e);
                        } finally {
                            if (J$.Fr(650961))
                                continue jalangiLabel39;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(649865, 'ScramSHA', ScramSHA, false, true)), ScramSHA256, false, true);
            J$.P(649921, J$.I(typeof module === 'undefined' ? module = J$.R(649889, 'module', undefined, true, true) : module = J$.R(649889, 'module', module, true, true)), 'exports', J$.T(649913, {
                ScramSHA1: J$.R(649897, 'ScramSHA1', ScramSHA1, false, true),
                ScramSHA256: J$.R(649905, 'ScramSHA256', ScramSHA256, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(650969, J$e);
        } finally {
            if (J$.Sr(650977))
                continue jalangiLabel40;
            else
                break jalangiLabel40;
        }
    }
// JALANGI DO NOT INSTRUMENT
