// mongodb/lib/core/auth/sspi.js
J$.noInstrEval = false;
jalangiLabel33:
    while (true) {
        try {
            J$.Se(655825, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/auth/sspi.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(651257, arguments.callee, this, arguments);
                            arguments = J$.N(651265, 'arguments', arguments, true, false, false);
                            obj = J$.N(651273, 'obj', obj, true, false, false);
                            if (J$.C(37504, J$.C(37464, J$.B(80618, '===', J$.U(80610, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(650993, 'Symbol', undefined, true, true) : Symbol = J$.R(650993, 'Symbol', Symbol, true, true))), J$.T(651001, 'function', 21, false))) ? J$.B(80634, '===', J$.U(80626, 'typeof', J$.G(651017, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(651009, 'Symbol', undefined, true, true) : Symbol = J$.R(651009, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(651025, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(651081, '_typeof2', J$.T(651073, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(651049, arguments.callee, this, arguments);
                                                arguments = J$.N(651057, 'arguments', arguments, true, false, false);
                                                obj = J$.N(651065, 'obj', obj, true, false, false);
                                                return J$.Rt(651041, J$.U(80642, 'typeof', J$.R(651033, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(656073, J$e);
                                            } finally {
                                                if (J$.Fr(656081))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(651217, '_typeof2', J$.T(651209, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(651185, arguments.callee, this, arguments);
                                                arguments = J$.N(651193, 'arguments', arguments, true, false, false);
                                                obj = J$.N(651201, 'obj', obj, true, false, false);
                                                return J$.Rt(651177, J$.C(37496, J$.C(37488, J$.C(37480, J$.C(37472, J$.R(651089, 'obj', obj, false, false)) ? J$.B(80658, '===', J$.U(80650, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(651097, 'Symbol', undefined, true, true) : Symbol = J$.R(651097, 'Symbol', Symbol, true, true))), J$.T(651105, 'function', 21, false)) : J$._()) ? J$.B(80666, '===', J$.G(651121, J$.R(651113, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(651129, 'Symbol', undefined, true, true) : Symbol = J$.R(651129, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(80674, '!==', J$.R(651137, 'obj', obj, false, false), J$.G(651153, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(651145, 'Symbol', undefined, true, true) : Symbol = J$.R(651145, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(651161, 'symbol', 21, false) : J$.U(80682, 'typeof', J$.R(651169, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(656089, J$e);
                                            } finally {
                                                if (J$.Fr(656097))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(651249, J$.F(651241, J$.R(651225, '_typeof2', _typeof2, false, true), false)(J$.R(651233, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(656105, J$e);
                        } finally {
                            if (J$.Fr(656113))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(651593, arguments.callee, this, arguments);
                            arguments = J$.N(651601, 'arguments', arguments, true, false, false);
                            obj = J$.N(651609, 'obj', obj, true, false, false);
                            if (J$.C(37552, J$.C(37512, J$.B(80698, '===', J$.U(80690, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(651281, 'Symbol', undefined, true, true) : Symbol = J$.R(651281, 'Symbol', Symbol, true, true))), J$.T(651289, 'function', 21, false))) ? J$.B(80706, '===', J$.F(651321, J$.R(651297, '_typeof2', _typeof2, false, true), false)(J$.G(651313, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(651305, 'Symbol', undefined, true, true) : Symbol = J$.R(651305, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(651329, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(651401, '_typeof', J$.T(651393, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(651369, arguments.callee, this, arguments);
                                                arguments = J$.N(651377, 'arguments', arguments, true, false, false);
                                                obj = J$.N(651385, 'obj', obj, true, false, false);
                                                return J$.Rt(651361, J$.F(651353, J$.R(651337, '_typeof2', _typeof2, false, true), false)(J$.R(651345, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(656121, J$e);
                                            } finally {
                                                if (J$.Fr(656129))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(651553, '_typeof', J$.T(651545, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(651521, arguments.callee, this, arguments);
                                                arguments = J$.N(651529, 'arguments', arguments, true, false, false);
                                                obj = J$.N(651537, 'obj', obj, true, false, false);
                                                return J$.Rt(651513, J$.C(37544, J$.C(37536, J$.C(37528, J$.C(37520, J$.R(651409, 'obj', obj, false, false)) ? J$.B(80722, '===', J$.U(80714, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(651417, 'Symbol', undefined, true, true) : Symbol = J$.R(651417, 'Symbol', Symbol, true, true))), J$.T(651425, 'function', 21, false)) : J$._()) ? J$.B(80730, '===', J$.G(651441, J$.R(651433, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(651449, 'Symbol', undefined, true, true) : Symbol = J$.R(651449, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(80738, '!==', J$.R(651457, 'obj', obj, false, false), J$.G(651473, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(651465, 'Symbol', undefined, true, true) : Symbol = J$.R(651465, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(651481, 'symbol', 21, false) : J$.F(651505, J$.R(651489, '_typeof2', _typeof2, false, true), false)(J$.R(651497, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(656137, J$e);
                                            } finally {
                                                if (J$.Fr(656145))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(651585, J$.F(651577, J$.R(651561, '_typeof', _typeof, false, true), false)(J$.R(651569, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(656153, J$e);
                        } finally {
                            if (J$.Fr(656161))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(651657, arguments.callee, this, arguments);
                            arguments = J$.N(651665, 'arguments', arguments, true, false, false);
                            instance = J$.N(651673, 'instance', instance, true, false, false);
                            Constructor = J$.N(651681, 'Constructor', Constructor, true, false, false);
                            if (J$.C(37560, J$.U(80754, '!', J$.B(80746, 'instanceof', J$.R(651617, 'instance', instance, false, false), J$.R(651625, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(651649, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(651633, 'TypeError', undefined, true, true) : TypeError = J$.R(651633, 'TypeError', TypeError, true, true)), true)(J$.T(651641, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(656169, J$e);
                        } finally {
                            if (J$.Fr(656177))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(651929, arguments.callee, this, arguments);
                            arguments = J$.N(651937, 'arguments', arguments, true, false, false);
                            target = J$.N(651945, 'target', target, true, false, false);
                            props = J$.N(651953, 'props', props, true, false, false);
                            J$.N(651961, 'i', i, false, false, false);
                            J$.N(651969, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(651697, 'i', J$.T(651689, 0, 22, false), i, false, false); J$.C(37584, J$.B(80762, '<', J$.R(651705, 'i', i, false, false), J$.G(651721, J$.R(651713, 'props', props, false, false), 'length'))); J$.B(80786, '-', i = J$.W(651737, 'i', J$.B(80778, '+', J$.U(80770, '+', J$.R(651729, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(651769, 'descriptor', J$.G(651761, J$.R(651745, 'props', props, false, false), J$.R(651753, 'i', i, false, false)), descriptor, false, false);
                                J$.P(651809, J$.R(651777, 'descriptor', descriptor, false, false), 'enumerable', J$.C(37568, J$.G(651793, J$.R(651785, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(651801, false, 23, false));
                                J$.P(651833, J$.R(651817, 'descriptor', descriptor, false, false), 'configurable', J$.T(651825, true, 23, false));
                                if (J$.C(37576, J$.B(80794, 'in', J$.T(651841, 'value', 21, false), J$.R(651849, 'descriptor', descriptor, false, false))))
                                    J$.P(651873, J$.R(651857, 'descriptor', descriptor, false, false), 'writable', J$.T(651865, true, 23, false));
                                J$.M(651921, J$.I(typeof Object === 'undefined' ? Object = J$.R(651881, 'Object', undefined, true, true) : Object = J$.R(651881, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(651889, 'target', target, false, false), J$.G(651905, J$.R(651897, 'descriptor', descriptor, false, false), 'key'), J$.R(651913, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(656185, J$e);
                        } finally {
                            if (J$.Fr(656193))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(652081, arguments.callee, this, arguments);
                            arguments = J$.N(652089, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(652097, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(652105, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(652113, 'staticProps', staticProps, true, false, false);
                            if (J$.C(37592, J$.R(651977, 'protoProps', protoProps, false, false)))
                                J$.F(652017, J$.R(651985, '_defineProperties', _defineProperties, false, true), false)(J$.G(652001, J$.R(651993, 'Constructor', Constructor, false, false), 'prototype'), J$.R(652009, 'protoProps', protoProps, false, false));
                            if (J$.C(37600, J$.R(652025, 'staticProps', staticProps, false, false)))
                                J$.F(652057, J$.R(652033, '_defineProperties', _defineProperties, false, true), false)(J$.R(652041, 'Constructor', Constructor, false, false), J$.R(652049, 'staticProps', staticProps, false, false));
                            return J$.Rt(652073, J$.R(652065, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(656201, J$e);
                        } finally {
                            if (J$.Fr(656209))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(652225, arguments.callee, this, arguments);
                            arguments = J$.N(652233, 'arguments', arguments, true, false, false);
                            self = J$.N(652241, 'self', self, true, false, false);
                            call = J$.N(652249, 'call', call, true, false, false);
                            if (J$.C(37624, J$.C(37616, J$.R(652121, 'call', call, false, false)) ? J$.C(37608, J$.B(80802, '===', J$.F(652145, J$.R(652129, '_typeof', _typeof, false, true), false)(J$.R(652137, 'call', call, false, false)), J$.T(652153, 'object', 21, false))) ? J$._() : J$.B(80818, '===', J$.U(80810, 'typeof', J$.R(652161, 'call', call, false, false)), J$.T(652169, 'function', 21, false)) : J$._())) {
                                return J$.Rt(652185, J$.R(652177, 'call', call, false, false));
                            }
                            return J$.Rt(652217, J$.F(652209, J$.R(652193, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(652201, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(656217, J$e);
                        } finally {
                            if (J$.Fr(656225))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(652313, arguments.callee, this, arguments);
                            arguments = J$.N(652321, 'arguments', arguments, true, false, false);
                            self = J$.N(652329, 'self', self, true, false, false);
                            if (J$.C(37632, J$.B(80826, '===', J$.R(652257, 'self', self, false, false), void J$.T(652265, 0, 22, false)))) {
                                throw J$.F(652289, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(652273, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(652273, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(652281, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(652305, J$.R(652297, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(656233, J$e);
                        } finally {
                            if (J$.Fr(656241))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(652697, arguments.callee, this, arguments);
                            arguments = J$.N(652705, 'arguments', arguments, true, false, false);
                            target = J$.N(652713, 'target', target, true, false, false);
                            property = J$.N(652721, 'property', property, true, false, false);
                            receiver = J$.N(652729, 'receiver', receiver, true, false, false);
                            if (J$.C(37664, J$.C(37640, J$.B(80842, '!==', J$.U(80834, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(652337, 'Reflect', undefined, true, true) : Reflect = J$.R(652337, 'Reflect', Reflect, true, true))), J$.T(652345, 'undefined', 21, false))) ? J$.G(652361, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(652353, 'Reflect', undefined, true, true) : Reflect = J$.R(652353, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(652385, '_get', J$.G(652377, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(652369, 'Reflect', undefined, true, true) : Reflect = J$.R(652369, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(652633, '_get', J$.T(652625, function _get(target, property, receiver) {
                                    jalangiLabel11:
                                        while (true) {
                                            try {
                                                J$.Fe(652569, arguments.callee, this, arguments);
                                                arguments = J$.N(652577, 'arguments', arguments, true, false, false);
                                                target = J$.N(652585, 'target', target, true, false, false);
                                                property = J$.N(652593, 'property', property, true, false, false);
                                                receiver = J$.N(652601, 'receiver', receiver, true, false, false);
                                                J$.N(652609, 'base', base, false, false, false);
                                                J$.N(652617, 'desc', desc, false, false, false);
                                                var base = J$.W(652425, 'base', J$.F(652417, J$.R(652393, '_superPropBase', _superPropBase, false, true), false)(J$.R(652401, 'target', target, false, false), J$.R(652409, 'property', property, false, false)), base, false, false);
                                                if (J$.C(37648, J$.U(80850, '!', J$.R(652433, 'base', base, false, false))))
                                                    return J$.Rt(652441, undefined);
                                                var desc = J$.W(652481, 'desc', J$.M(652473, J$.I(typeof Object === 'undefined' ? Object = J$.R(652449, 'Object', undefined, true, true) : Object = J$.R(652449, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(652457, 'base', base, false, false), J$.R(652465, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(37656, J$.G(652497, J$.R(652489, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(652537, J$.M(652529, J$.G(652513, J$.R(652505, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(652521, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(652561, J$.G(652553, J$.R(652545, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(656249, J$e);
                                            } finally {
                                                if (J$.Fr(656257))
                                                    continue jalangiLabel11;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(652689, J$.F(652681, J$.R(652641, '_get', _get, false, true), false)(J$.R(652649, 'target', target, false, false), J$.R(652657, 'property', property, false, false), J$.C(37672, J$.R(652665, 'receiver', receiver, false, false)) ? J$._() : J$.R(652673, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(656265, J$e);
                        } finally {
                            if (J$.Fr(656273))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(652849, arguments.callee, this, arguments);
                            arguments = J$.N(652857, 'arguments', arguments, true, false, false);
                            object = J$.N(652865, 'object', object, true, false, false);
                            property = J$.N(652873, 'property', property, true, false, false);
                            while (J$.C(37688, J$.U(80858, '!', J$.M(652777, J$.G(652753, J$.G(652745, J$.I(typeof Object === 'undefined' ? Object = J$.R(652737, 'Object', undefined, true, true) : Object = J$.R(652737, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(652761, 'object', object, false, false), J$.R(652769, 'property', property, false, false))))) {
                                object = J$.W(652809, 'object', J$.F(652801, J$.R(652785, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(652793, 'object', object, false, false)), object, false, false);
                                if (J$.C(37680, J$.B(80866, '===', J$.R(652817, 'object', object, false, false), J$.T(652825, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(652841, J$.R(652833, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(656281, J$e);
                        } finally {
                            if (J$.Fr(656289))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(653033, arguments.callee, this, arguments);
                            arguments = J$.N(653041, 'arguments', arguments, true, false, false);
                            o = J$.N(653049, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(652993, '_getPrototypeOf', J$.C(37704, J$.G(652889, J$.I(typeof Object === 'undefined' ? Object = J$.R(652881, 'Object', undefined, true, true) : Object = J$.R(652881, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(652905, J$.I(typeof Object === 'undefined' ? Object = J$.R(652897, 'Object', undefined, true, true) : Object = J$.R(652897, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(652985, function _getPrototypeOf(o) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(652961, arguments.callee, this, arguments);
                                            arguments = J$.N(652969, 'arguments', arguments, true, false, false);
                                            o = J$.N(652977, 'o', o, true, false, false);
                                            return J$.Rt(652953, J$.C(37696, J$.G(652921, J$.R(652913, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(652945, J$.I(typeof Object === 'undefined' ? Object = J$.R(652929, 'Object', undefined, true, true) : Object = J$.R(652929, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(652937, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(656297, J$e);
                                        } finally {
                                            if (J$.Fr(656305))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(653025, J$.F(653017, J$.R(653001, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(653009, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(656313, J$e);
                        } finally {
                            if (J$.Fr(656321))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(653249, arguments.callee, this, arguments);
                            arguments = J$.N(653257, 'arguments', arguments, true, false, false);
                            subClass = J$.N(653265, 'subClass', subClass, true, false, false);
                            superClass = J$.N(653273, 'superClass', superClass, true, false, false);
                            if (J$.C(37720, J$.C(37712, J$.B(80882, '!==', J$.U(80874, 'typeof', J$.R(653057, 'superClass', superClass, false, false)), J$.T(653065, 'function', 21, false))) ? J$.B(80890, '!==', J$.R(653073, 'superClass', superClass, false, false), J$.T(653081, null, 25, false)) : J$._())) {
                                throw J$.F(653105, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(653089, 'TypeError', undefined, true, true) : TypeError = J$.R(653089, 'TypeError', TypeError, true, true)), true)(J$.T(653097, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(653201, J$.R(653113, 'subClass', subClass, false, false), 'prototype', J$.M(653193, J$.I(typeof Object === 'undefined' ? Object = J$.R(653121, 'Object', undefined, true, true) : Object = J$.R(653121, 'Object', Object, true, true)), 'create', false)(J$.C(37728, J$.R(653129, 'superClass', superClass, false, false)) ? J$.G(653145, J$.R(653137, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(653185, {
                                constructor: J$.T(653177, {
                                    value: J$.R(653153, 'subClass', subClass, false, false),
                                    writable: J$.T(653161, true, 23, false),
                                    configurable: J$.T(653169, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(37736, J$.R(653209, 'superClass', superClass, false, false)))
                                J$.F(653241, J$.R(653217, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(653225, 'subClass', subClass, false, false), J$.R(653233, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(656329, J$e);
                        } finally {
                            if (J$.Fr(656337))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(653425, arguments.callee, this, arguments);
                            arguments = J$.N(653433, 'arguments', arguments, true, false, false);
                            o = J$.N(653441, 'o', o, true, false, false);
                            p = J$.N(653449, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(653377, '_setPrototypeOf', J$.C(37744, J$.G(653289, J$.I(typeof Object === 'undefined' ? Object = J$.R(653281, 'Object', undefined, true, true) : Object = J$.R(653281, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(653369, function _setPrototypeOf(o, p) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(653337, arguments.callee, this, arguments);
                                            arguments = J$.N(653345, 'arguments', arguments, true, false, false);
                                            o = J$.N(653353, 'o', o, true, false, false);
                                            p = J$.N(653361, 'p', p, true, false, false);
                                            J$.P(653313, J$.R(653297, 'o', o, false, false), '__proto__', J$.R(653305, 'p', p, false, false));
                                            return J$.Rt(653329, J$.R(653321, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(656345, J$e);
                                        } finally {
                                            if (J$.Fr(656353))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(653417, J$.F(653409, J$.R(653385, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(653393, 'o', o, false, false), J$.R(653401, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(656361, J$e);
                        } finally {
                            if (J$.Fr(656369))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function SSIPAuthenticate(self, MongoAuthProcess, username, password, gssapiServiceName, sendAuthCommand, connection, options, callback) {
                jalangiLabel32:
                    while (true) {
                        try {
                            J$.Fe(655689, arguments.callee, this, arguments);
                            function authCommand(command, authCb) {
                                jalangiLabel23:
                                    while (true) {
                                        try {
                                            J$.Fe(654553, arguments.callee, this, arguments);
                                            arguments = J$.N(654561, 'arguments', arguments, true, false, false);
                                            command = J$.N(654569, 'command', command, true, false, false);
                                            authCb = J$.N(654577, 'authCb', authCb, true, false, false);
                                            J$.F(654545, J$.R(654505, 'sendAuthCommand', sendAuthCommand, false, false), false)(J$.R(654513, 'connection', connection, false, false), J$.T(654521, '$external.$cmd', 21, false), J$.R(654529, 'command', command, false, false), J$.R(654537, 'authCb', authCb, false, false));
                                        } catch (J$e) {
                                            J$.Ex(656441, J$e);
                                        } finally {
                                            if (J$.Fr(656449))
                                                continue jalangiLabel23;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(655697, 'arguments', arguments, true, false, false);
                            self = J$.N(655705, 'self', self, true, false, false);
                            MongoAuthProcess = J$.N(655713, 'MongoAuthProcess', MongoAuthProcess, true, false, false);
                            username = J$.N(655721, 'username', username, true, false, false);
                            password = J$.N(655729, 'password', password, true, false, false);
                            gssapiServiceName = J$.N(655737, 'gssapiServiceName', gssapiServiceName, true, false, false);
                            sendAuthCommand = J$.N(655745, 'sendAuthCommand', sendAuthCommand, true, false, false);
                            connection = J$.N(655753, 'connection', connection, true, false, false);
                            options = J$.N(655761, 'options', options, true, false, false);
                            callback = J$.N(655769, 'callback', callback, true, false, false);
                            J$.N(655777, 'authProcess', authProcess, false, false, false);
                            authCommand = J$.N(655793, 'authCommand', J$.T(655785, authCommand, 12, false), true, false, false);
                            var authProcess = J$.W(654497, 'authProcess', J$.F(654489, J$.R(654433, 'MongoAuthProcess', MongoAuthProcess, false, false), true)(J$.G(654449, J$.R(654441, 'connection', connection, false, false), 'host'), J$.G(654465, J$.R(654457, 'connection', connection, false, false), 'port'), J$.R(654473, 'gssapiServiceName', gssapiServiceName, false, false), J$.R(654481, 'options', options, false, false)), authProcess, false, false);
                            J$.M(655681, J$.R(654585, 'authProcess', authProcess, false, false), 'init', false)(J$.R(654593, 'username', username, false, false), J$.R(654601, 'password', password, false, false), J$.T(655673, function (err) {
                                jalangiLabel31:
                                    while (true) {
                                        try {
                                            J$.Fe(655649, arguments.callee, this, arguments);
                                            arguments = J$.N(655657, 'arguments', arguments, true, false, false);
                                            err = J$.N(655665, 'err', err, true, false, false);
                                            if (J$.C(37776, J$.R(654609, 'err', err, false, false)))
                                                return J$.Rt(654649, J$.F(654641, J$.R(654617, 'callback', callback, false, false), false)(J$.R(654625, 'err', err, false, false), J$.T(654633, false, 23, false)));
                                            J$.M(655641, J$.R(654657, 'authProcess', authProcess, false, false), 'transition', false)(J$.T(654665, '', 21, false), J$.T(655633, function (err, payload) {
                                                jalangiLabel30:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(655593, arguments.callee, this, arguments);
                                                            arguments = J$.N(655601, 'arguments', arguments, true, false, false);
                                                            err = J$.N(655609, 'err', err, true, false, false);
                                                            payload = J$.N(655617, 'payload', payload, true, false, false);
                                                            J$.N(655625, 'command', command, false, false, false);
                                                            if (J$.C(37784, J$.R(654673, 'err', err, false, false)))
                                                                return J$.Rt(654713, J$.F(654705, J$.R(654681, 'callback', callback, false, false), false)(J$.R(654689, 'err', err, false, false), J$.T(654697, false, 23, false)));
                                                            var command = J$.W(654761, 'command', J$.T(654753, {
                                                                saslStart: J$.T(654721, 1, 22, false),
                                                                mechanism: J$.T(654729, 'GSSAPI', 21, false),
                                                                payload: J$.R(654737, 'payload', payload, false, false),
                                                                autoAuthorize: J$.T(654745, 1, 22, false)
                                                            }, 11, false), command, false, false);
                                                            J$.F(655585, J$.R(654769, 'authCommand', authCommand, false, false), false)(J$.R(654777, 'command', command, false, false), J$.T(655577, function (err, doc) {
                                                                jalangiLabel29:
                                                                    while (true) {
                                                                        try {
                                                                            J$.Fe(655545, arguments.callee, this, arguments);
                                                                            arguments = J$.N(655553, 'arguments', arguments, true, false, false);
                                                                            err = J$.N(655561, 'err', err, true, false, false);
                                                                            doc = J$.N(655569, 'doc', doc, true, false, false);
                                                                            if (J$.C(37792, J$.R(654785, 'err', err, false, false)))
                                                                                return J$.Rt(654825, J$.F(654817, J$.R(654793, 'callback', callback, false, false), false)(J$.R(654801, 'err', err, false, false), J$.T(654809, false, 23, false)));
                                                                            J$.M(655537, J$.R(654833, 'authProcess', authProcess, false, false), 'transition', false)(J$.G(654849, J$.R(654841, 'doc', doc, false, false), 'payload'), J$.T(655529, function (err, payload) {
                                                                                jalangiLabel28:
                                                                                    while (true) {
                                                                                        try {
                                                                                            J$.Fe(655489, arguments.callee, this, arguments);
                                                                                            arguments = J$.N(655497, 'arguments', arguments, true, false, false);
                                                                                            err = J$.N(655505, 'err', err, true, false, false);
                                                                                            payload = J$.N(655513, 'payload', payload, true, false, false);
                                                                                            J$.N(655521, 'command', command, false, false, false);
                                                                                            if (J$.C(37800, J$.R(654857, 'err', err, false, false)))
                                                                                                return J$.Rt(654897, J$.F(654889, J$.R(654865, 'callback', callback, false, false), false)(J$.R(654873, 'err', err, false, false), J$.T(654881, false, 23, false)));
                                                                                            var command = J$.W(654945, 'command', J$.T(654937, {
                                                                                                saslContinue: J$.T(654905, 1, 22, false),
                                                                                                conversationId: J$.G(654921, J$.R(654913, 'doc', doc, false, false), 'conversationId'),
                                                                                                payload: J$.R(654929, 'payload', payload, false, false)
                                                                                            }, 11, false), command, false, false);
                                                                                            J$.F(655481, J$.R(654953, 'authCommand', authCommand, false, false), false)(J$.R(654961, 'command', command, false, false), J$.T(655473, function (err, doc) {
                                                                                                jalangiLabel27:
                                                                                                    while (true) {
                                                                                                        try {
                                                                                                            J$.Fe(655441, arguments.callee, this, arguments);
                                                                                                            arguments = J$.N(655449, 'arguments', arguments, true, false, false);
                                                                                                            err = J$.N(655457, 'err', err, true, false, false);
                                                                                                            doc = J$.N(655465, 'doc', doc, true, false, false);
                                                                                                            if (J$.C(37808, J$.R(654969, 'err', err, false, false)))
                                                                                                                return J$.Rt(655009, J$.F(655001, J$.R(654977, 'callback', callback, false, false), false)(J$.R(654985, 'err', err, false, false), J$.T(654993, false, 23, false)));
                                                                                                            J$.M(655433, J$.R(655017, 'authProcess', authProcess, false, false), 'transition', false)(J$.G(655033, J$.R(655025, 'doc', doc, false, false), 'payload'), J$.T(655425, function (err, payload) {
                                                                                                                jalangiLabel26:
                                                                                                                    while (true) {
                                                                                                                        try {
                                                                                                                            J$.Fe(655385, arguments.callee, this, arguments);
                                                                                                                            arguments = J$.N(655393, 'arguments', arguments, true, false, false);
                                                                                                                            err = J$.N(655401, 'err', err, true, false, false);
                                                                                                                            payload = J$.N(655409, 'payload', payload, true, false, false);
                                                                                                                            J$.N(655417, 'command', command, false, false, false);
                                                                                                                            if (J$.C(37816, J$.R(655041, 'err', err, false, false)))
                                                                                                                                return J$.Rt(655081, J$.F(655073, J$.R(655049, 'callback', callback, false, false), false)(J$.R(655057, 'err', err, false, false), J$.T(655065, false, 23, false)));
                                                                                                                            var command = J$.W(655129, 'command', J$.T(655121, {
                                                                                                                                saslContinue: J$.T(655089, 1, 22, false),
                                                                                                                                conversationId: J$.G(655105, J$.R(655097, 'doc', doc, false, false), 'conversationId'),
                                                                                                                                payload: J$.R(655113, 'payload', payload, false, false)
                                                                                                                            }, 11, false), command, false, false);
                                                                                                                            J$.F(655377, J$.R(655137, 'authCommand', authCommand, false, false), false)(J$.R(655145, 'command', command, false, false), J$.T(655369, function (err, response) {
                                                                                                                                jalangiLabel25:
                                                                                                                                    while (true) {
                                                                                                                                        try {
                                                                                                                                            J$.Fe(655337, arguments.callee, this, arguments);
                                                                                                                                            arguments = J$.N(655345, 'arguments', arguments, true, false, false);
                                                                                                                                            err = J$.N(655353, 'err', err, true, false, false);
                                                                                                                                            response = J$.N(655361, 'response', response, true, false, false);
                                                                                                                                            if (J$.C(37824, J$.R(655153, 'err', err, false, false)))
                                                                                                                                                return J$.Rt(655193, J$.F(655185, J$.R(655161, 'callback', callback, false, false), false)(J$.R(655169, 'err', err, false, false), J$.T(655177, false, 23, false)));
                                                                                                                                            J$.M(655329, J$.R(655201, 'authProcess', authProcess, false, false), 'transition', false)(J$.T(655209, null, 25, false), J$.T(655321, function (err) {
                                                                                                                                                jalangiLabel24:
                                                                                                                                                    while (true) {
                                                                                                                                                        try {
                                                                                                                                                            J$.Fe(655297, arguments.callee, this, arguments);
                                                                                                                                                            arguments = J$.N(655305, 'arguments', arguments, true, false, false);
                                                                                                                                                            err = J$.N(655313, 'err', err, true, false, false);
                                                                                                                                                            if (J$.C(37832, J$.R(655217, 'err', err, false, false)))
                                                                                                                                                                return J$.Rt(655257, J$.F(655249, J$.R(655225, 'callback', callback, false, false), false)(J$.R(655233, 'err', err, false, false), J$.T(655241, null, 25, false)));
                                                                                                                                                            J$.F(655289, J$.R(655265, 'callback', callback, false, false), false)(J$.T(655273, null, 25, false), J$.R(655281, 'response', response, false, false));
                                                                                                                                                        } catch (J$e) {
                                                                                                                                                            J$.Ex(656457, J$e);
                                                                                                                                                        } finally {
                                                                                                                                                            if (J$.Fr(656465))
                                                                                                                                                                continue jalangiLabel24;
                                                                                                                                                            else
                                                                                                                                                                return J$.Ra();
                                                                                                                                                        }
                                                                                                                                                    }
                                                                                                                                            }, 12, false));
                                                                                                                                        } catch (J$e) {
                                                                                                                                            J$.Ex(656473, J$e);
                                                                                                                                        } finally {
                                                                                                                                            if (J$.Fr(656481))
                                                                                                                                                continue jalangiLabel25;
                                                                                                                                            else
                                                                                                                                                return J$.Ra();
                                                                                                                                        }
                                                                                                                                    }
                                                                                                                            }, 12, false));
                                                                                                                        } catch (J$e) {
                                                                                                                            J$.Ex(656489, J$e);
                                                                                                                        } finally {
                                                                                                                            if (J$.Fr(656497))
                                                                                                                                continue jalangiLabel26;
                                                                                                                            else
                                                                                                                                return J$.Ra();
                                                                                                                        }
                                                                                                                    }
                                                                                                            }, 12, false));
                                                                                                        } catch (J$e) {
                                                                                                            J$.Ex(656505, J$e);
                                                                                                        } finally {
                                                                                                            if (J$.Fr(656513))
                                                                                                                continue jalangiLabel27;
                                                                                                            else
                                                                                                                return J$.Ra();
                                                                                                        }
                                                                                                    }
                                                                                            }, 12, false));
                                                                                        } catch (J$e) {
                                                                                            J$.Ex(656521, J$e);
                                                                                        } finally {
                                                                                            if (J$.Fr(656529))
                                                                                                continue jalangiLabel28;
                                                                                            else
                                                                                                return J$.Ra();
                                                                                        }
                                                                                    }
                                                                            }, 12, false));
                                                                        } catch (J$e) {
                                                                            J$.Ex(656537, J$e);
                                                                        } finally {
                                                                            if (J$.Fr(656545))
                                                                                continue jalangiLabel29;
                                                                            else
                                                                                return J$.Ra();
                                                                        }
                                                                    }
                                                            }, 12, false));
                                                        } catch (J$e) {
                                                            J$.Ex(656553, J$e);
                                                        } finally {
                                                            if (J$.Fr(656561))
                                                                continue jalangiLabel30;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(656569, J$e);
                                        } finally {
                                            if (J$.Fr(656577))
                                                continue jalangiLabel31;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(656585, J$e);
                        } finally {
                            if (J$.Fr(656593))
                                continue jalangiLabel32;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(655841, '_typeof2', J$.T(655833, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(655857, '_typeof', J$.T(655849, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(655873, '_classCallCheck', J$.T(655865, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(655889, '_defineProperties', J$.T(655881, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(655905, '_createClass', J$.T(655897, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(655921, '_possibleConstructorReturn', J$.T(655913, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(655937, '_assertThisInitialized', J$.T(655929, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(655953, '_get', J$.T(655945, _get, 12, false), true, false, false);
            _superPropBase = J$.N(655969, '_superPropBase', J$.T(655961, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(655985, '_getPrototypeOf', J$.T(655977, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(656001, '_inherits', J$.T(655993, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(656017, '_setPrototypeOf', J$.T(656009, _setPrototypeOf, 12, false), true, false, false);
            J$.N(656025, 'AuthProvider', AuthProvider, false, false, false);
            J$.N(656033, 'retrieveKerberos', retrieveKerberos, false, false, false);
            J$.N(656041, 'kerberos', kerberos, false, false, false);
            J$.N(656049, 'SSPI', SSPI, false, false, false);
            SSIPAuthenticate = J$.N(656065, 'SSIPAuthenticate', J$.T(656057, SSIPAuthenticate, 12, false), true, false, false);
            J$.T(650985, 'use strict', 21, false);
            var AuthProvider = J$.W(653489, 'AuthProvider', J$.G(653481, J$.F(653473, J$.I(typeof require === 'undefined' ? require = J$.R(653457, 'require', undefined, true, true) : require = J$.R(653457, 'require', require, true, true)), false)(J$.T(653465, './auth_provider', 21, false)), 'AuthProvider'), AuthProvider, false, true);
            var retrieveKerberos = J$.W(653529, 'retrieveKerberos', J$.G(653521, J$.F(653513, J$.I(typeof require === 'undefined' ? require = J$.R(653497, 'require', undefined, true, true) : require = J$.R(653497, 'require', require, true, true)), false)(J$.T(653505, '../utils', 21, false)), 'retrieveKerberos'), retrieveKerberos, false, true);
            var kerberos;
            var SSPI = J$.W(654425, 'SSPI', J$.F(654417, J$.T(654401, function (_AuthProvider) {
                jalangiLabel22:
                    while (true) {
                        try {
                            J$.Fe(654361, arguments.callee, this, arguments);
                            function SSPI() {
                                jalangiLabel19:
                                    while (true) {
                                        try {
                                            J$.Fe(653681, arguments.callee, this, arguments);
                                            arguments = J$.N(653689, 'arguments', arguments, true, false, false);
                                            J$.F(653593, J$.R(653569, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(653577, 'this', this, false, false), J$.R(653585, 'SSPI', SSPI, false, false));
                                            return J$.Rt(653673, J$.F(653665, J$.R(653601, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(653609, 'this', this, false, false), J$.M(653657, J$.F(653633, J$.R(653617, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(653625, 'SSPI', SSPI, false, false)), 'apply', false)(J$.R(653641, 'this', this, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(653649, 'arguments', undefined, true, true) : arguments = J$.R(653649, 'arguments', arguments, true, true)))));
                                        } catch (J$e) {
                                            J$.Ex(656377, J$e);
                                        } finally {
                                            if (J$.Fr(656385))
                                                continue jalangiLabel19;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(654369, 'arguments', arguments, true, false, false);
                            _AuthProvider = J$.N(654377, '_AuthProvider', _AuthProvider, true, false, false);
                            SSPI = J$.N(654393, 'SSPI', J$.T(654385, SSPI, 12, false), true, false, false);
                            J$.F(653561, J$.R(653537, '_inherits', _inherits, false, true), false)(J$.R(653545, 'SSPI', SSPI, false, false), J$.R(653553, '_AuthProvider', _AuthProvider, false, false));
                            J$.F(654337, J$.R(653697, '_createClass', _createClass, false, true), false)(J$.R(653705, 'SSPI', SSPI, false, false), J$.T(654329, [
                                J$.T(654049, {
                                    key: J$.T(653713, '_authenticateSingleConnection', 21, false),
                                    value: J$.T(654041, function _authenticateSingleConnection(sendAuthCommand, connection, credentials, callback) {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(653961, arguments.callee, this, arguments);
                                                    arguments = J$.N(653969, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(653977, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connection = J$.N(653985, 'connection', connection, true, false, false);
                                                    credentials = J$.N(653993, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(654001, 'callback', callback, true, false, false);
                                                    J$.N(654009, 'username', username, false, false, false);
                                                    J$.N(654017, 'password', password, false, false, false);
                                                    J$.N(654025, 'mechanismProperties', mechanismProperties, false, false, false);
                                                    J$.N(654033, 'gssapiServiceName', gssapiServiceName, false, false, false);
                                                    var username = J$.W(653737, 'username', J$.G(653729, J$.R(653721, 'credentials', credentials, false, false), 'username'), username, false, false);
                                                    var password = J$.W(653761, 'password', J$.G(653753, J$.R(653745, 'credentials', credentials, false, false), 'password'), password, false, false);
                                                    var mechanismProperties = J$.W(653785, 'mechanismProperties', J$.G(653777, J$.R(653769, 'credentials', credentials, false, false), 'mechanismProperties'), mechanismProperties, false, false);
                                                    var gssapiServiceName = J$.W(653849, 'gssapiServiceName', J$.C(37760, J$.C(37752, J$.G(653809, J$.R(653793, 'mechanismProperties', mechanismProperties, false, false), J$.T(653801, 'gssapiservicename', 21, false))) ? J$._() : J$.G(653833, J$.R(653817, 'mechanismProperties', mechanismProperties, false, false), J$.T(653825, 'gssapiServiceName', 21, false))) ? J$._() : J$.T(653841, 'mongodb', 21, false), gssapiServiceName, false, false);
                                                    J$.F(653953, J$.R(653857, 'SSIPAuthenticate', SSIPAuthenticate, false, true), false)(J$.R(653865, 'this', this, false, false), J$.G(653889, J$.G(653881, J$.R(653873, 'kerberos', kerberos, false, true), 'processes'), 'MongoAuthProcess'), J$.R(653897, 'username', username, false, false), J$.R(653905, 'password', password, false, false), J$.R(653913, 'gssapiServiceName', gssapiServiceName, false, false), J$.R(653921, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(653929, 'connection', connection, false, false), J$.R(653937, 'mechanismProperties', mechanismProperties, false, false), J$.R(653945, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(656393, J$e);
                                                } finally {
                                                    if (J$.Fr(656401))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(654321, {
                                    key: J$.T(654057, 'auth', 21, false),
                                    value: J$.T(654313, function auth(sendAuthCommand, connections, credentials, callback) {
                                        jalangiLabel21:
                                            while (true) {
                                                try {
                                                    J$.Fe(654265, arguments.callee, this, arguments);
                                                    arguments = J$.N(654273, 'arguments', arguments, true, false, false);
                                                    sendAuthCommand = J$.N(654281, 'sendAuthCommand', sendAuthCommand, true, false, false);
                                                    connections = J$.N(654289, 'connections', connections, true, false, false);
                                                    credentials = J$.N(654297, 'credentials', credentials, true, false, false);
                                                    callback = J$.N(654305, 'callback', callback, true, false, false);
                                                    if (J$.C(37768, J$.B(80898, '==', J$.R(654065, 'kerberos', kerberos, false, true), J$.T(654073, null, 25, false)))) {
                                                        try {
                                                            kerberos = J$.W(654097, 'kerberos', J$.F(654089, J$.R(654081, 'retrieveKerberos', retrieveKerberos, false, true), false)(), kerberos, false, true);
                                                        } catch (e) {
                                                            J$.N(654145, 'e', e, false, false, true);
                                                            return J$.Rt(654137, J$.F(654129, J$.R(654105, 'callback', callback, false, false), false)(J$.R(654113, 'e', e, false, false), J$.T(654121, null, 25, false)));
                                                        }
                                                    }
                                                    J$.M(654257, J$.F(654209, J$.R(654153, '_get', _get, false, true), false)(J$.F(654185, J$.R(654161, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(654177, J$.R(654169, 'SSPI', SSPI, false, false), 'prototype')), J$.T(654193, 'auth', 21, false), J$.R(654201, 'this', this, false, false)), 'call', false)(J$.R(654217, 'this', this, false, false), J$.R(654225, 'sendAuthCommand', sendAuthCommand, false, false), J$.R(654233, 'connections', connections, false, false), J$.R(654241, 'credentials', credentials, false, false), J$.R(654249, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(656409, J$e);
                                                } finally {
                                                    if (J$.Fr(656417))
                                                        continue jalangiLabel21;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(654353, J$.R(654345, 'SSPI', SSPI, false, false));
                        } catch (J$e) {
                            J$.Ex(656425, J$e);
                        } finally {
                            if (J$.Fr(656433))
                                continue jalangiLabel22;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(654409, 'AuthProvider', AuthProvider, false, true)), SSPI, false, true);
            J$.P(655817, J$.I(typeof module === 'undefined' ? module = J$.R(655801, 'module', undefined, true, true) : module = J$.R(655801, 'module', module, true, true)), 'exports', J$.R(655809, 'SSPI', SSPI, false, true));
        } catch (J$e) {
            J$.Ex(656601, J$e);
        } finally {
            if (J$.Sr(656609))
                continue jalangiLabel33;
            else
                break jalangiLabel33;
        }
    }
// JALANGI DO NOT INSTRUMENT
