// codemirror/src/util/browser.js
"use strict";

require("core-js/modules/es.number.constructor");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.string.match");

require("core-js/modules/es.number.constructor");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.string.match");

require("core-js/modules/es.number.constructor");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.string.match");

require("core-js/modules/es.number.constructor");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.string.match");

require("core-js/modules/es.number.constructor");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.string.match");

require("core-js/modules/es.number.constructor");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.string.match");

require("core-js/modules/es.number.constructor");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.string.match");

require("core-js/modules/es.number.constructor");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.string.match");

require("core-js/modules/es.number.constructor");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.string.match");

require("core-js/modules/es.number.constructor");

require("core-js/modules/es.regexp.exec");

require("core-js/modules/es.string.match");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.captureRightClick = exports.flipCtrlCmd = exports.windows = exports.chromeOS = exports.mac = exports.mobile = exports.android = exports.ios = exports.phantom = exports.mac_geMountainLion = exports.safari = exports.presto = exports.chrome = exports.webkit = exports.ie_version = exports.ie = exports.gecko = void 0; // Kludges for bugs and behavior differences that can't be feature
// detected are enabled based on userAgent etc sniffing.

var userAgent = navigator.userAgent;
var platform = navigator.platform;
var gecko = /gecko\/\d/i.test(userAgent);
exports.gecko = gecko;
var ie_upto10 = /MSIE \d/.test(userAgent);
var ie_11up = /Trident\/(?:[7-9]|\d{2,})\..*rv:(\d+)/.exec(userAgent);
var edge = /Edge\/(\d+)/.exec(userAgent);
var ie = ie_upto10 || ie_11up || edge;
exports.ie = ie;
var ie_version = ie && (ie_upto10 ? document.documentMode || 6 : +(edge || ie_11up)[1]);
exports.ie_version = ie_version;
var webkit = !edge && /WebKit\//.test(userAgent);
exports.webkit = webkit;
var qtwebkit = webkit && /Qt\/\d+\.\d+/.test(userAgent);
var chrome = !edge && /Chrome\//.test(userAgent);
exports.chrome = chrome;
var presto = /Opera\//.test(userAgent);
exports.presto = presto;
var safari = /Apple Computer/.test(navigator.vendor);
exports.safari = safari;
var mac_geMountainLion = /Mac OS X 1\d\D([8-9]|\d\d)\D/.test(userAgent);
exports.mac_geMountainLion = mac_geMountainLion;
var phantom = /PhantomJS/.test(userAgent);
exports.phantom = phantom;
var ios = !edge && /AppleWebKit/.test(userAgent) && /Mobile\/\w+/.test(userAgent);
exports.ios = ios;
var android = /Android/.test(userAgent); // This is woefully incomplete. Suggestions for alternative methods welcome.

exports.android = android;
var mobile = ios || android || /webOS|BlackBerry|Opera Mini|Opera Mobi|IEMobile/i.test(userAgent);
exports.mobile = mobile;
var mac = ios || /Mac/.test(platform);
exports.mac = mac;
var chromeOS = /\bCrOS\b/.test(userAgent);
exports.chromeOS = chromeOS;
var windows = /win/i.test(platform);
exports.windows = windows;
var presto_version = presto && userAgent.match(/Version\/(\d*\.\d*)/);
if (presto_version) presto_version = Number(presto_version[1]);

if (presto_version && presto_version >= 15) {
  exports.presto = presto = false;
  exports.webkit = webkit = true;
} // Some browsers use the wrong event properties to signal cmd/ctrl on OS X


var flipCtrlCmd = mac && (qtwebkit || presto && (presto_version == null || presto_version < 12.11));
exports.flipCtrlCmd = flipCtrlCmd;
var captureRightClick = gecko || ie && ie_version >= 9;
exports.captureRightClick = captureRightClick;
