// js-yaml/lib/js-yaml/common.js
J$.noInstrEval = false;
jalangiLabel6:
    while (true) {
        try {
            J$.Se(3543401, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/common.js');
            function isNothing(subject) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(3542473, arguments.callee, this, arguments);
                            arguments = J$.N(3542481, 'arguments', arguments, true, false, false);
                            subject = J$.N(3542489, 'subject', subject, true, false, false);
                            return J$.Rt(3542465, J$.C(184064, J$.B(389290, '===', J$.U(389282, 'typeof', J$.R(3542433, 'subject', subject, false, false)), J$.T(3542441, 'undefined', 21, false))) ? J$._() : J$.B(389298, '===', J$.R(3542449, 'subject', subject, false, false), J$.T(3542457, null, 25, false)));
                        } catch (J$e) {
                            J$.Ex(3543505, J$e);
                        } finally {
                            if (J$.Fr(3543513))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isObject(subject) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(3542537, arguments.callee, this, arguments);
                            arguments = J$.N(3542545, 'arguments', arguments, true, false, false);
                            subject = J$.N(3542553, 'subject', subject, true, false, false);
                            return J$.Rt(3542529, J$.C(184072, J$.B(389314, '===', J$.U(389306, 'typeof', J$.R(3542497, 'subject', subject, false, false)), J$.T(3542505, 'object', 21, false))) ? J$.B(389322, '!==', J$.R(3542513, 'subject', subject, false, false), J$.T(3542521, null, 25, false)) : J$._());
                        } catch (J$e) {
                            J$.Ex(3543521, J$e);
                        } finally {
                            if (J$.Fr(3543529))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function toArray(sequence) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(3542665, arguments.callee, this, arguments);
                            arguments = J$.N(3542673, 'arguments', arguments, true, false, false);
                            sequence = J$.N(3542681, 'sequence', sequence, true, false, false);
                            if (J$.C(184088, J$.M(3542577, J$.I(typeof Array === 'undefined' ? Array = J$.R(3542561, 'Array', undefined, true, true) : Array = J$.R(3542561, 'Array', Array, true, true)), 'isArray', false)(J$.R(3542569, 'sequence', sequence, false, false))))
                                return J$.Rt(3542593, J$.R(3542585, 'sequence', sequence, false, false));
                            else if (J$.C(184080, J$.F(3542617, J$.R(3542601, 'isNothing', isNothing, false, true), false)(J$.R(3542609, 'sequence', sequence, false, false))))
                                return J$.Rt(3542633, J$.T(3542625, [], 10, false));
                            return J$.Rt(3542657, J$.T(3542649, [J$.R(3542641, 'sequence', sequence, false, false)], 10, false));
                        } catch (J$e) {
                            J$.Ex(3543537, J$e);
                        } finally {
                            if (J$.Fr(3543545))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function extend(target, source) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(3542905, arguments.callee, this, arguments);
                            arguments = J$.N(3542913, 'arguments', arguments, true, false, false);
                            target = J$.N(3542921, 'target', target, true, false, false);
                            source = J$.N(3542929, 'source', source, true, false, false);
                            J$.N(3542937, 'index', index, false, false, false);
                            J$.N(3542945, 'length', length, false, false, false);
                            J$.N(3542953, 'key', key, false, false, false);
                            J$.N(3542961, 'sourceKeys', sourceKeys, false, false, false);
                            var index, length, key, sourceKeys;
                            if (J$.C(184104, J$.R(3542689, 'source', source, false, false))) {
                                sourceKeys = J$.W(3542721, 'sourceKeys', J$.M(3542713, J$.I(typeof Object === 'undefined' ? Object = J$.R(3542697, 'Object', undefined, true, true) : Object = J$.R(3542697, 'Object', Object, true, true)), 'keys', false)(J$.R(3542705, 'source', source, false, false)), sourceKeys, false, false);
                                for (index = J$.W(3542737, 'index', J$.T(3542729, 0, 22, false), index, false, false), length = J$.W(3542761, 'length', J$.G(3542753, J$.R(3542745, 'sourceKeys', sourceKeys, false, false), 'length'), length, false, false); J$.C(184096, J$.B(389330, '<', J$.R(3542769, 'index', index, false, false), J$.R(3542777, 'length', length, false, false))); index = J$.W(3542801, 'index', J$.B(389338, '+', J$.R(3542793, 'index', index, false, false), J$.T(3542785, 1, 22, false)), index, false, false)) {
                                    key = J$.W(3542833, 'key', J$.G(3542825, J$.R(3542809, 'sourceKeys', sourceKeys, false, false), J$.R(3542817, 'index', index, false, false)), key, false, false);
                                    J$.P(3542881, J$.R(3542841, 'target', target, false, false), J$.R(3542849, 'key', key, false, false), J$.G(3542873, J$.R(3542857, 'source', source, false, false), J$.R(3542865, 'key', key, false, false)));
                                }
                            }
                            return J$.Rt(3542897, J$.R(3542889, 'target', target, false, false));
                        } catch (J$e) {
                            J$.Ex(3543553, J$e);
                        } finally {
                            if (J$.Fr(3543561))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function repeat(string, count) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(3543081, arguments.callee, this, arguments);
                            arguments = J$.N(3543089, 'arguments', arguments, true, false, false);
                            string = J$.N(3543097, 'string', string, true, false, false);
                            count = J$.N(3543105, 'count', count, true, false, false);
                            J$.N(3543113, 'result', result, false, false, false);
                            J$.N(3543121, 'cycle', cycle, false, false, false);
                            var result = J$.W(3542977, 'result', J$.T(3542969, '', 21, false), result, false, false), cycle;
                            for (cycle = J$.W(3542993, 'cycle', J$.T(3542985, 0, 22, false), cycle, false, false); J$.C(184112, J$.B(389346, '<', J$.R(3543001, 'cycle', cycle, false, false), J$.R(3543009, 'count', count, false, false))); cycle = J$.W(3543033, 'cycle', J$.B(389354, '+', J$.R(3543025, 'cycle', cycle, false, false), J$.T(3543017, 1, 22, false)), cycle, false, false)) {
                                result = J$.W(3543057, 'result', J$.B(389362, '+', J$.R(3543049, 'result', result, false, false), J$.R(3543041, 'string', string, false, false)), result, false, false);
                            }
                            return J$.Rt(3543073, J$.R(3543065, 'result', result, false, false));
                        } catch (J$e) {
                            J$.Ex(3543569, J$e);
                        } finally {
                            if (J$.Fr(3543577))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isNegativeZero(number) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(3543185, arguments.callee, this, arguments);
                            arguments = J$.N(3543193, 'arguments', arguments, true, false, false);
                            number = J$.N(3543201, 'number', number, true, false, false);
                            return J$.Rt(3543177, J$.C(184120, J$.B(389370, '===', J$.R(3543129, 'number', number, false, false), J$.T(3543137, 0, 22, false))) ? J$.B(389386, '===', J$.G(3543153, J$.I(typeof Number === 'undefined' ? Number = J$.R(3543145, 'Number', undefined, true, true) : Number = J$.R(3543145, 'Number', Number, true, true)), 'NEGATIVE_INFINITY'), J$.B(389378, '/', J$.T(3543161, 1, 22, false), J$.R(3543169, 'number', number, false, false))) : J$._());
                        } catch (J$e) {
                            J$.Ex(3543585, J$e);
                        } finally {
                            if (J$.Fr(3543593))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            isNothing = J$.N(3543417, 'isNothing', J$.T(3543409, isNothing, 12, false), true, false, false);
            isObject = J$.N(3543433, 'isObject', J$.T(3543425, isObject, 12, false), true, false, false);
            toArray = J$.N(3543449, 'toArray', J$.T(3543441, toArray, 12, false), true, false, false);
            extend = J$.N(3543465, 'extend', J$.T(3543457, extend, 12, false), true, false, false);
            repeat = J$.N(3543481, 'repeat', J$.T(3543473, repeat, 12, false), true, false, false);
            isNegativeZero = J$.N(3543497, 'isNegativeZero', J$.T(3543489, isNegativeZero, 12, false), true, false, false);
            J$.T(3542425, 'use strict', 21, false);
            J$.P(3543233, J$.G(3543217, J$.I(typeof module === 'undefined' ? module = J$.R(3543209, 'module', undefined, true, true) : module = J$.R(3543209, 'module', module, true, true)), 'exports'), 'isNothing', J$.R(3543225, 'isNothing', isNothing, false, true));
            J$.P(3543265, J$.G(3543249, J$.I(typeof module === 'undefined' ? module = J$.R(3543241, 'module', undefined, true, true) : module = J$.R(3543241, 'module', module, true, true)), 'exports'), 'isObject', J$.R(3543257, 'isObject', isObject, false, true));
            J$.P(3543297, J$.G(3543281, J$.I(typeof module === 'undefined' ? module = J$.R(3543273, 'module', undefined, true, true) : module = J$.R(3543273, 'module', module, true, true)), 'exports'), 'toArray', J$.R(3543289, 'toArray', toArray, false, true));
            J$.P(3543329, J$.G(3543313, J$.I(typeof module === 'undefined' ? module = J$.R(3543305, 'module', undefined, true, true) : module = J$.R(3543305, 'module', module, true, true)), 'exports'), 'repeat', J$.R(3543321, 'repeat', repeat, false, true));
            J$.P(3543361, J$.G(3543345, J$.I(typeof module === 'undefined' ? module = J$.R(3543337, 'module', undefined, true, true) : module = J$.R(3543337, 'module', module, true, true)), 'exports'), 'isNegativeZero', J$.R(3543353, 'isNegativeZero', isNegativeZero, false, true));
            J$.P(3543393, J$.G(3543377, J$.I(typeof module === 'undefined' ? module = J$.R(3543369, 'module', undefined, true, true) : module = J$.R(3543369, 'module', module, true, true)), 'exports'), 'extend', J$.R(3543385, 'extend', extend, false, true));
        } catch (J$e) {
            J$.Ex(3543601, J$e);
        } finally {
            if (J$.Sr(3543609))
                continue jalangiLabel6;
            else
                break jalangiLabel6;
        }
    }
// JALANGI DO NOT INSTRUMENT
