// bungle/lib/logger[old].js
import chalk from 'chalk';
import growl from 'growl';

const levels = {
    debug: 0,
    verbose: 1000,
    info: 2000,
    warn: 3000,
    error: 4000
};

const formatedLevel = {
    debug: chalk.green('debug'),
    verbose: chalk.green('verbose'),
    info: chalk.green('info'),
    warn: chalk.red('warn'),
    error: chalk.bgRed.white('error')
};

const ansiRx =
/[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g;

class Logger {
    constructor(config) {
        this.level = {
            console: levels[config.console || 'info'],
            notify: levels[config.notify || 'info']
        };
    }

    log(level, prefix, message) {
        if (this.level.console <= levels[level]) {
            console.log( /*eslint no-console:0*/
                chalk.blue('bungle'),
                formatedLevel[level],
                chalk.yellow(prefix),
                message
            );
        }

        if (this.level.notify <= levels[level]) {
            growl(message.replace(ansiRx, ''), {
                title: 'Bungle ' + level + ' ' + prefix
            });
        }
    }

}

export { Logger };
