// codemirror/rollup.config.js
"use strict";

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _rollupPluginBuble = _interopRequireDefault(require("rollup-plugin-buble"));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : {
    "default": obj
  };
}

var _default = {
  input: "src/codemirror.js",
  output: {
    banner: "// CodeMirror, copyright (c) by Marijn Haverbeke and others\n// Distributed under an MIT license: https://codemirror.net/LICENSE\n\n// This is CodeMirror (https://codemirror.net), a code editor\n// implemented in JavaScript on top of the browser's DOM.\n//\n// You can find some technical background for some of the code below\n// at http://marijnhaverbeke.nl/blog/#cm-internals .\n",
    format: "umd",
    file: "lib/codemirror.js",
    name: "CodeMirror"
  },
  plugins: [(0, _rollupPluginBuble["default"])({
    namedFunctionExpressions: false
  })]
};
exports["default"] = _default;
