// js-yaml/lib/js-yaml/type/js/undefined.js
J$.noInstrEval = false;
jalangiLabel4:
    while (true) {
        try {
            J$.Se(3608273, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/js/undefined.js');
            function resolveJavascriptUndefined() {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(3608057, arguments.callee, this, arguments);
                            arguments = J$.N(3608065, 'arguments', arguments, true, false, false);
                            return J$.Rt(3608049, J$.T(3608041, true, 23, false));
                        } catch (J$e) {
                            J$.Ex(3608353, J$e);
                        } finally {
                            if (J$.Fr(3608361))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function constructJavascriptUndefined() {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(3608089, arguments.callee, this, arguments);
                            arguments = J$.N(3608097, 'arguments', arguments, true, false, false);
                            return J$.Rt(3608081, J$.T(3608073, undefined, 24, false));
                        } catch (J$e) {
                            J$.Ex(3608369, J$e);
                        } finally {
                            if (J$.Fr(3608377))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function representJavascriptUndefined() {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(3608121, arguments.callee, this, arguments);
                            arguments = J$.N(3608129, 'arguments', arguments, true, false, false);
                            return J$.Rt(3608113, J$.T(3608105, '', 21, false));
                        } catch (J$e) {
                            J$.Ex(3608385, J$e);
                        } finally {
                            if (J$.Fr(3608393))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isUndefined(object) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(3608161, arguments.callee, this, arguments);
                            arguments = J$.N(3608169, 'arguments', arguments, true, false, false);
                            object = J$.N(3608177, 'object', object, true, false, false);
                            return J$.Rt(3608153, J$.B(397546, '===', J$.U(397538, 'typeof', J$.R(3608137, 'object', object, false, false)), J$.T(3608145, 'undefined', 21, false)));
                        } catch (J$e) {
                            J$.Ex(3608401, J$e);
                        } finally {
                            if (J$.Fr(3608409))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(3608281, 'Type', Type, false, false, false);
            resolveJavascriptUndefined = J$.N(3608297, 'resolveJavascriptUndefined', J$.T(3608289, resolveJavascriptUndefined, 12, false), true, false, false);
            constructJavascriptUndefined = J$.N(3608313, 'constructJavascriptUndefined', J$.T(3608305, constructJavascriptUndefined, 12, false), true, false, false);
            representJavascriptUndefined = J$.N(3608329, 'representJavascriptUndefined', J$.T(3608321, representJavascriptUndefined, 12, false), true, false, false);
            isUndefined = J$.N(3608345, 'isUndefined', J$.T(3608337, isUndefined, 12, false), true, false, false);
            J$.T(3608001, 'use strict', 21, false);
            var Type = J$.W(3608033, 'Type', J$.F(3608025, J$.I(typeof require === 'undefined' ? require = J$.R(3608009, 'require', undefined, true, true) : require = J$.R(3608009, 'require', require, true, true)), false)(J$.T(3608017, '../../type', 21, false)), Type, false, true);
            J$.P(3608265, J$.I(typeof module === 'undefined' ? module = J$.R(3608185, 'module', undefined, true, true) : module = J$.R(3608185, 'module', module, true, true)), 'exports', J$.F(3608257, J$.R(3608193, 'Type', Type, false, true), true)(J$.T(3608201, 'tag:yaml.org,2002:js/undefined', 21, false), J$.T(3608249, {
                kind: J$.T(3608209, 'scalar', 21, false),
                resolve: J$.R(3608217, 'resolveJavascriptUndefined', resolveJavascriptUndefined, false, true),
                construct: J$.R(3608225, 'constructJavascriptUndefined', constructJavascriptUndefined, false, true),
                predicate: J$.R(3608233, 'isUndefined', isUndefined, false, true),
                represent: J$.R(3608241, 'representJavascriptUndefined', representJavascriptUndefined, false, true)
            }, 11, false)));
        } catch (J$e) {
            J$.Ex(3608417, J$e);
        } finally {
            if (J$.Sr(3608425))
                continue jalangiLabel4;
            else
                break jalangiLabel4;
        }
    }
// JALANGI DO NOT INSTRUMENT
