// mongodb/lib/core/error.js
J$.noInstrEval = false;
jalangiLabel42:
    while (true) {
        try {
            J$.Se(753217, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/error.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(746441, arguments.callee, this, arguments);
                            arguments = J$.N(746449, 'arguments', arguments, true, false, false);
                            obj = J$.N(746457, 'obj', obj, true, false, false);
                            if (J$.C(43384, J$.C(43344, J$.B(88746, '===', J$.U(88738, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(746177, 'Symbol', undefined, true, true) : Symbol = J$.R(746177, 'Symbol', Symbol, true, true))), J$.T(746185, 'function', 21, false))) ? J$.B(88762, '===', J$.U(88754, 'typeof', J$.G(746201, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(746193, 'Symbol', undefined, true, true) : Symbol = J$.R(746193, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(746209, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(746265, '_typeof2', J$.T(746257, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(746233, arguments.callee, this, arguments);
                                                arguments = J$.N(746241, 'arguments', arguments, true, false, false);
                                                obj = J$.N(746249, 'obj', obj, true, false, false);
                                                return J$.Rt(746225, J$.U(88770, 'typeof', J$.R(746217, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(753633, J$e);
                                            } finally {
                                                if (J$.Fr(753641))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(746401, '_typeof2', J$.T(746393, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(746369, arguments.callee, this, arguments);
                                                arguments = J$.N(746377, 'arguments', arguments, true, false, false);
                                                obj = J$.N(746385, 'obj', obj, true, false, false);
                                                return J$.Rt(746361, J$.C(43376, J$.C(43368, J$.C(43360, J$.C(43352, J$.R(746273, 'obj', obj, false, false)) ? J$.B(88786, '===', J$.U(88778, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(746281, 'Symbol', undefined, true, true) : Symbol = J$.R(746281, 'Symbol', Symbol, true, true))), J$.T(746289, 'function', 21, false)) : J$._()) ? J$.B(88794, '===', J$.G(746305, J$.R(746297, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(746313, 'Symbol', undefined, true, true) : Symbol = J$.R(746313, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(88802, '!==', J$.R(746321, 'obj', obj, false, false), J$.G(746337, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(746329, 'Symbol', undefined, true, true) : Symbol = J$.R(746329, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(746345, 'symbol', 21, false) : J$.U(88810, 'typeof', J$.R(746353, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(753649, J$e);
                                            } finally {
                                                if (J$.Fr(753657))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(746433, J$.F(746425, J$.R(746409, '_typeof2', _typeof2, false, true), false)(J$.R(746417, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(753665, J$e);
                        } finally {
                            if (J$.Fr(753673))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(746777, arguments.callee, this, arguments);
                            arguments = J$.N(746785, 'arguments', arguments, true, false, false);
                            obj = J$.N(746793, 'obj', obj, true, false, false);
                            if (J$.C(43432, J$.C(43392, J$.B(88826, '===', J$.U(88818, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(746465, 'Symbol', undefined, true, true) : Symbol = J$.R(746465, 'Symbol', Symbol, true, true))), J$.T(746473, 'function', 21, false))) ? J$.B(88834, '===', J$.F(746505, J$.R(746481, '_typeof2', _typeof2, false, true), false)(J$.G(746497, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(746489, 'Symbol', undefined, true, true) : Symbol = J$.R(746489, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(746513, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(746585, '_typeof', J$.T(746577, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(746553, arguments.callee, this, arguments);
                                                arguments = J$.N(746561, 'arguments', arguments, true, false, false);
                                                obj = J$.N(746569, 'obj', obj, true, false, false);
                                                return J$.Rt(746545, J$.F(746537, J$.R(746521, '_typeof2', _typeof2, false, true), false)(J$.R(746529, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(753681, J$e);
                                            } finally {
                                                if (J$.Fr(753689))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(746737, '_typeof', J$.T(746729, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(746705, arguments.callee, this, arguments);
                                                arguments = J$.N(746713, 'arguments', arguments, true, false, false);
                                                obj = J$.N(746721, 'obj', obj, true, false, false);
                                                return J$.Rt(746697, J$.C(43424, J$.C(43416, J$.C(43408, J$.C(43400, J$.R(746593, 'obj', obj, false, false)) ? J$.B(88850, '===', J$.U(88842, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(746601, 'Symbol', undefined, true, true) : Symbol = J$.R(746601, 'Symbol', Symbol, true, true))), J$.T(746609, 'function', 21, false)) : J$._()) ? J$.B(88858, '===', J$.G(746625, J$.R(746617, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(746633, 'Symbol', undefined, true, true) : Symbol = J$.R(746633, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(88866, '!==', J$.R(746641, 'obj', obj, false, false), J$.G(746657, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(746649, 'Symbol', undefined, true, true) : Symbol = J$.R(746649, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(746665, 'symbol', 21, false) : J$.F(746689, J$.R(746673, '_typeof2', _typeof2, false, true), false)(J$.R(746681, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(753697, J$e);
                                            } finally {
                                                if (J$.Fr(753705))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(746769, J$.F(746761, J$.R(746745, '_typeof', _typeof, false, true), false)(J$.R(746753, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(753713, J$e);
                        } finally {
                            if (J$.Fr(753721))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(746841, arguments.callee, this, arguments);
                            arguments = J$.N(746849, 'arguments', arguments, true, false, false);
                            instance = J$.N(746857, 'instance', instance, true, false, false);
                            Constructor = J$.N(746865, 'Constructor', Constructor, true, false, false);
                            if (J$.C(43440, J$.U(88882, '!', J$.B(88874, 'instanceof', J$.R(746801, 'instance', instance, false, false), J$.R(746809, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(746833, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(746817, 'TypeError', undefined, true, true) : TypeError = J$.R(746817, 'TypeError', TypeError, true, true)), true)(J$.T(746825, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(753729, J$e);
                        } finally {
                            if (J$.Fr(753737))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(747113, arguments.callee, this, arguments);
                            arguments = J$.N(747121, 'arguments', arguments, true, false, false);
                            target = J$.N(747129, 'target', target, true, false, false);
                            props = J$.N(747137, 'props', props, true, false, false);
                            J$.N(747145, 'i', i, false, false, false);
                            J$.N(747153, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(746881, 'i', J$.T(746873, 0, 22, false), i, false, false); J$.C(43464, J$.B(88890, '<', J$.R(746889, 'i', i, false, false), J$.G(746905, J$.R(746897, 'props', props, false, false), 'length'))); J$.B(88914, '-', i = J$.W(746921, 'i', J$.B(88906, '+', J$.U(88898, '+', J$.R(746913, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(746953, 'descriptor', J$.G(746945, J$.R(746929, 'props', props, false, false), J$.R(746937, 'i', i, false, false)), descriptor, false, false);
                                J$.P(746993, J$.R(746961, 'descriptor', descriptor, false, false), 'enumerable', J$.C(43448, J$.G(746977, J$.R(746969, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(746985, false, 23, false));
                                J$.P(747017, J$.R(747001, 'descriptor', descriptor, false, false), 'configurable', J$.T(747009, true, 23, false));
                                if (J$.C(43456, J$.B(88922, 'in', J$.T(747025, 'value', 21, false), J$.R(747033, 'descriptor', descriptor, false, false))))
                                    J$.P(747057, J$.R(747041, 'descriptor', descriptor, false, false), 'writable', J$.T(747049, true, 23, false));
                                J$.M(747105, J$.I(typeof Object === 'undefined' ? Object = J$.R(747065, 'Object', undefined, true, true) : Object = J$.R(747065, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(747073, 'target', target, false, false), J$.G(747089, J$.R(747081, 'descriptor', descriptor, false, false), 'key'), J$.R(747097, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(753745, J$e);
                        } finally {
                            if (J$.Fr(753753))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(747265, arguments.callee, this, arguments);
                            arguments = J$.N(747273, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(747281, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(747289, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(747297, 'staticProps', staticProps, true, false, false);
                            if (J$.C(43472, J$.R(747161, 'protoProps', protoProps, false, false)))
                                J$.F(747201, J$.R(747169, '_defineProperties', _defineProperties, false, true), false)(J$.G(747185, J$.R(747177, 'Constructor', Constructor, false, false), 'prototype'), J$.R(747193, 'protoProps', protoProps, false, false));
                            if (J$.C(43480, J$.R(747209, 'staticProps', staticProps, false, false)))
                                J$.F(747241, J$.R(747217, '_defineProperties', _defineProperties, false, true), false)(J$.R(747225, 'Constructor', Constructor, false, false), J$.R(747233, 'staticProps', staticProps, false, false));
                            return J$.Rt(747257, J$.R(747249, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(753761, J$e);
                        } finally {
                            if (J$.Fr(753769))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(747409, arguments.callee, this, arguments);
                            arguments = J$.N(747417, 'arguments', arguments, true, false, false);
                            self = J$.N(747425, 'self', self, true, false, false);
                            call = J$.N(747433, 'call', call, true, false, false);
                            if (J$.C(43504, J$.C(43496, J$.R(747305, 'call', call, false, false)) ? J$.C(43488, J$.B(88930, '===', J$.F(747329, J$.R(747313, '_typeof', _typeof, false, true), false)(J$.R(747321, 'call', call, false, false)), J$.T(747337, 'object', 21, false))) ? J$._() : J$.B(88946, '===', J$.U(88938, 'typeof', J$.R(747345, 'call', call, false, false)), J$.T(747353, 'function', 21, false)) : J$._())) {
                                return J$.Rt(747369, J$.R(747361, 'call', call, false, false));
                            }
                            return J$.Rt(747401, J$.F(747393, J$.R(747377, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(747385, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(753777, J$e);
                        } finally {
                            if (J$.Fr(753785))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(747497, arguments.callee, this, arguments);
                            arguments = J$.N(747505, 'arguments', arguments, true, false, false);
                            self = J$.N(747513, 'self', self, true, false, false);
                            if (J$.C(43512, J$.B(88954, '===', J$.R(747441, 'self', self, false, false), void J$.T(747449, 0, 22, false)))) {
                                throw J$.F(747473, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(747457, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(747457, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(747465, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(747489, J$.R(747481, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(753793, J$e);
                        } finally {
                            if (J$.Fr(753801))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(747713, arguments.callee, this, arguments);
                            arguments = J$.N(747721, 'arguments', arguments, true, false, false);
                            subClass = J$.N(747729, 'subClass', subClass, true, false, false);
                            superClass = J$.N(747737, 'superClass', superClass, true, false, false);
                            if (J$.C(43528, J$.C(43520, J$.B(88970, '!==', J$.U(88962, 'typeof', J$.R(747521, 'superClass', superClass, false, false)), J$.T(747529, 'function', 21, false))) ? J$.B(88978, '!==', J$.R(747537, 'superClass', superClass, false, false), J$.T(747545, null, 25, false)) : J$._())) {
                                throw J$.F(747569, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(747553, 'TypeError', undefined, true, true) : TypeError = J$.R(747553, 'TypeError', TypeError, true, true)), true)(J$.T(747561, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(747665, J$.R(747577, 'subClass', subClass, false, false), 'prototype', J$.M(747657, J$.I(typeof Object === 'undefined' ? Object = J$.R(747585, 'Object', undefined, true, true) : Object = J$.R(747585, 'Object', Object, true, true)), 'create', false)(J$.C(43536, J$.R(747593, 'superClass', superClass, false, false)) ? J$.G(747609, J$.R(747601, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(747649, {
                                constructor: J$.T(747641, {
                                    value: J$.R(747617, 'subClass', subClass, false, false),
                                    writable: J$.T(747625, true, 23, false),
                                    configurable: J$.T(747633, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(43544, J$.R(747673, 'superClass', superClass, false, false)))
                                J$.F(747705, J$.R(747681, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(747689, 'subClass', subClass, false, false), J$.R(747697, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(753809, J$e);
                        } finally {
                            if (J$.Fr(753817))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _wrapNativeSuper(Class) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(748305, arguments.callee, this, arguments);
                            arguments = J$.N(748313, 'arguments', arguments, true, false, false);
                            Class = J$.N(748321, 'Class', Class, true, false, false);
                            J$.N(748329, '_cache', _cache, false, false, false);
                            var _cache = J$.W(747785, '_cache', J$.C(43552, J$.B(88994, '===', J$.U(88986, 'typeof', J$.I(typeof Map === 'undefined' ? Map = J$.R(747745, 'Map', undefined, true, true) : Map = J$.R(747745, 'Map', Map, true, true))), J$.T(747753, 'function', 21, false))) ? J$.F(747769, J$.I(typeof Map === 'undefined' ? Map = J$.R(747761, 'Map', undefined, true, true) : Map = J$.R(747761, 'Map', Map, true, true)), true)() : J$.T(747777, undefined, 24, false), _cache, false, false);
                            _wrapNativeSuper = J$.W(748265, '_wrapNativeSuper', J$.T(748257, function _wrapNativeSuper(Class) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(748217, arguments.callee, this, arguments);
                                            function Wrapper() {
                                                jalangiLabel12:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(748065, arguments.callee, this, arguments);
                                                            arguments = J$.N(748073, 'arguments', arguments, true, false, false);
                                                            return J$.Rt(748057, J$.F(748049, J$.R(747993, '_construct', _construct, false, true), false)(J$.R(748001, 'Class', Class, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(748009, 'arguments', undefined, true, true) : arguments = J$.R(748009, 'arguments', arguments, true, true)), J$.G(748041, J$.F(748033, J$.R(748017, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(748025, 'this', this, false, false)), 'constructor')));
                                                        } catch (J$e) {
                                                            J$.Ex(753825, J$e);
                                                        } finally {
                                                            if (J$.Fr(753833))
                                                                continue jalangiLabel12;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }
                                            arguments = J$.N(748225, 'arguments', arguments, true, false, false);
                                            Class = J$.N(748233, 'Class', Class, true, false, false);
                                            Wrapper = J$.N(748249, 'Wrapper', J$.T(748241, Wrapper, 12, false), true, false, false);
                                            if (J$.C(43568, J$.C(43560, J$.B(89002, '===', J$.R(747793, 'Class', Class, false, false), J$.T(747801, null, 25, false))) ? J$._() : J$.U(89010, '!', J$.F(747825, J$.R(747809, '_isNativeFunction', _isNativeFunction, false, true), false)(J$.R(747817, 'Class', Class, false, false)))))
                                                return J$.Rt(747841, J$.R(747833, 'Class', Class, false, false));
                                            if (J$.C(43576, J$.B(89026, '!==', J$.U(89018, 'typeof', J$.R(747849, 'Class', Class, false, false)), J$.T(747857, 'function', 21, false)))) {
                                                throw J$.F(747881, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(747865, 'TypeError', undefined, true, true) : TypeError = J$.R(747865, 'TypeError', TypeError, true, true)), true)(J$.T(747873, 'Super expression must either be null or a function', 21, false));
                                            }
                                            if (J$.C(43592, J$.B(89042, '!==', J$.U(89034, 'typeof', J$.R(747889, '_cache', _cache, false, false)), J$.T(747897, 'undefined', 21, false)))) {
                                                if (J$.C(43584, J$.M(747921, J$.R(747905, '_cache', _cache, false, false), 'has', false)(J$.R(747913, 'Class', Class, false, false))))
                                                    return J$.Rt(747953, J$.M(747945, J$.R(747929, '_cache', _cache, false, false), 'get', false)(J$.R(747937, 'Class', Class, false, false)));
                                                J$.M(747985, J$.R(747961, '_cache', _cache, false, false), 'set', false)(J$.R(747969, 'Class', Class, false, false), J$.R(747977, 'Wrapper', Wrapper, false, false));
                                            }
                                            J$.P(748169, J$.R(748081, 'Wrapper', Wrapper, false, false), 'prototype', J$.M(748161, J$.I(typeof Object === 'undefined' ? Object = J$.R(748089, 'Object', undefined, true, true) : Object = J$.R(748089, 'Object', Object, true, true)), 'create', false)(J$.G(748105, J$.R(748097, 'Class', Class, false, false), 'prototype'), J$.T(748153, {
                                                constructor: J$.T(748145, {
                                                    value: J$.R(748113, 'Wrapper', Wrapper, false, false),
                                                    enumerable: J$.T(748121, false, 23, false),
                                                    writable: J$.T(748129, true, 23, false),
                                                    configurable: J$.T(748137, true, 23, false)
                                                }, 11, false)
                                            }, 11, false)));
                                            return J$.Rt(748209, J$.F(748201, J$.R(748177, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(748185, 'Wrapper', Wrapper, false, false), J$.R(748193, 'Class', Class, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(753841, J$e);
                                        } finally {
                                            if (J$.Fr(753849))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _wrapNativeSuper, false, true);
                            return J$.Rt(748297, J$.F(748289, J$.R(748273, '_wrapNativeSuper', _wrapNativeSuper, false, true), false)(J$.R(748281, 'Class', Class, false, false)));
                        } catch (J$e) {
                            J$.Ex(753857, J$e);
                        } finally {
                            if (J$.Fr(753865))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isNativeReflectConstruct() {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(748585, arguments.callee, this, arguments);
                            arguments = J$.N(748593, 'arguments', arguments, true, false, false);
                            if (J$.C(43608, J$.C(43600, J$.B(89058, '===', J$.U(89050, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(748337, 'Reflect', undefined, true, true) : Reflect = J$.R(748337, 'Reflect', Reflect, true, true))), J$.T(748345, 'undefined', 21, false))) ? J$._() : J$.U(89066, '!', J$.G(748361, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(748353, 'Reflect', undefined, true, true) : Reflect = J$.R(748353, 'Reflect', Reflect, true, true)), 'construct'))))
                                return J$.Rt(748377, J$.T(748369, false, 23, false));
                            if (J$.C(43616, J$.G(748401, J$.G(748393, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(748385, 'Reflect', undefined, true, true) : Reflect = J$.R(748385, 'Reflect', Reflect, true, true)), 'construct'), 'sham')))
                                return J$.Rt(748417, J$.T(748409, false, 23, false));
                            if (J$.C(43624, J$.B(89082, '===', J$.U(89074, 'typeof', J$.I(typeof Proxy === 'undefined' ? Proxy = J$.R(748425, 'Proxy', undefined, true, true) : Proxy = J$.R(748425, 'Proxy', Proxy, true, true))), J$.T(748433, 'function', 21, false))))
                                return J$.Rt(748449, J$.T(748441, true, 23, false));
                            try {
                                J$.M(748537, J$.G(748473, J$.G(748465, J$.I(typeof Date === 'undefined' ? Date = J$.R(748457, 'Date', undefined, true, true) : Date = J$.R(748457, 'Date', Date, true, true)), 'prototype'), 'toString'), 'call', false)(J$.M(748529, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(748481, 'Reflect', undefined, true, true) : Reflect = J$.R(748481, 'Reflect', Reflect, true, true)), 'construct', false)(J$.I(typeof Date === 'undefined' ? Date = J$.R(748489, 'Date', undefined, true, true) : Date = J$.R(748489, 'Date', Date, true, true)), J$.T(748497, [], 10, false), J$.T(748521, function () {
                                    jalangiLabel15:
                                        while (true) {
                                            try {
                                                J$.Fe(748505, arguments.callee, this, arguments);
                                                arguments = J$.N(748513, 'arguments', arguments, true, false, false);
                                            } catch (J$e) {
                                                J$.Ex(753873, J$e);
                                            } finally {
                                                if (J$.Fr(753881))
                                                    continue jalangiLabel15;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)));
                                return J$.Rt(748553, J$.T(748545, true, 23, false));
                            } catch (e) {
                                J$.N(748577, 'e', e, false, false, true);
                                return J$.Rt(748569, J$.T(748561, false, 23, false));
                            }
                        } catch (J$e) {
                            J$.Ex(753889, J$e);
                        } finally {
                            if (J$.Fr(753897))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _construct(Parent, args, Class) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(748961, arguments.callee, this, arguments);
                            arguments = J$.N(748969, 'arguments', arguments, true, false, false);
                            Parent = J$.N(748977, 'Parent', Parent, true, false, false);
                            args = J$.N(748985, 'args', args, true, false, false);
                            Class = J$.N(748993, 'Class', Class, true, false, false);
                            if (J$.C(43640, J$.F(748609, J$.R(748601, 'isNativeReflectConstruct', isNativeReflectConstruct, false, true), false)())) {
                                _construct = J$.W(748633, '_construct', J$.G(748625, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(748617, 'Reflect', undefined, true, true) : Reflect = J$.R(748617, 'Reflect', Reflect, true, true)), 'construct'), _construct, false, true);
                            } else {
                                _construct = J$.W(748913, '_construct', J$.T(748905, function _construct(Parent, args, Class) {
                                    jalangiLabel17:
                                        while (true) {
                                            try {
                                                J$.Fe(748841, arguments.callee, this, arguments);
                                                arguments = J$.N(748849, 'arguments', arguments, true, false, false);
                                                Parent = J$.N(748857, 'Parent', Parent, true, false, false);
                                                args = J$.N(748865, 'args', args, true, false, false);
                                                Class = J$.N(748873, 'Class', Class, true, false, false);
                                                J$.N(748881, 'a', a, false, false, false);
                                                J$.N(748889, 'Constructor', Constructor, false, false, false);
                                                J$.N(748897, 'instance', instance, false, false, false);
                                                var a = J$.W(748657, 'a', J$.T(748649, [J$.T(748641, null, 25, false)], 10, false), a, false, false);
                                                J$.M(748697, J$.G(748673, J$.R(748665, 'a', a, false, false), 'push'), 'apply', false)(J$.R(748681, 'a', a, false, false), J$.R(748689, 'args', args, false, false));
                                                var Constructor = J$.W(748745, 'Constructor', J$.M(748737, J$.G(748713, J$.I(typeof Function === 'undefined' ? Function = J$.R(748705, 'Function', undefined, true, true) : Function = J$.R(748705, 'Function', Function, true, true)), 'bind'), 'apply', false)(J$.R(748721, 'Parent', Parent, false, false), J$.R(748729, 'a', a, false, false)), Constructor, false, false);
                                                var instance = J$.W(748769, 'instance', J$.F(748761, J$.R(748753, 'Constructor', Constructor, false, false), true)(), instance, false, false);
                                                if (J$.C(43632, J$.R(748777, 'Class', Class, false, false)))
                                                    J$.F(748817, J$.R(748785, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(748793, 'instance', instance, false, false), J$.G(748809, J$.R(748801, 'Class', Class, false, false), 'prototype'));
                                                return J$.Rt(748833, J$.R(748825, 'instance', instance, false, false));
                                            } catch (J$e) {
                                                J$.Ex(753905, J$e);
                                            } finally {
                                                if (J$.Fr(753913))
                                                    continue jalangiLabel17;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _construct, false, true);
                            }
                            return J$.Rt(748953, J$.M(748945, J$.R(748921, '_construct', _construct, false, true), 'apply', false)(J$.T(748929, null, 25, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(748937, 'arguments', undefined, true, true) : arguments = J$.R(748937, 'arguments', arguments, true, true))));
                        } catch (J$e) {
                            J$.Ex(753921, J$e);
                        } finally {
                            if (J$.Fr(753929))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _isNativeFunction(fn) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(749065, arguments.callee, this, arguments);
                            arguments = J$.N(749073, 'arguments', arguments, true, false, false);
                            fn = J$.N(749081, 'fn', fn, true, false, false);
                            return J$.Rt(749057, J$.B(89098, '!==', J$.M(749041, J$.M(749025, J$.G(749009, J$.I(typeof Function === 'undefined' ? Function = J$.R(749001, 'Function', undefined, true, true) : Function = J$.R(749001, 'Function', Function, true, true)), 'toString'), 'call', false)(J$.R(749017, 'fn', fn, false, false)), 'indexOf', false)(J$.T(749033, '[native code]', 21, false)), J$.U(89090, '-', J$.T(749049, 1, 22, false))));
                        } catch (J$e) {
                            J$.Ex(753937, J$e);
                        } finally {
                            if (J$.Fr(753945))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel21:
                    while (true) {
                        try {
                            J$.Fe(749233, arguments.callee, this, arguments);
                            arguments = J$.N(749241, 'arguments', arguments, true, false, false);
                            o = J$.N(749249, 'o', o, true, false, false);
                            p = J$.N(749257, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(749185, '_setPrototypeOf', J$.C(43648, J$.G(749097, J$.I(typeof Object === 'undefined' ? Object = J$.R(749089, 'Object', undefined, true, true) : Object = J$.R(749089, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(749177, function _setPrototypeOf(o, p) {
                                jalangiLabel20:
                                    while (true) {
                                        try {
                                            J$.Fe(749145, arguments.callee, this, arguments);
                                            arguments = J$.N(749153, 'arguments', arguments, true, false, false);
                                            o = J$.N(749161, 'o', o, true, false, false);
                                            p = J$.N(749169, 'p', p, true, false, false);
                                            J$.P(749121, J$.R(749105, 'o', o, false, false), '__proto__', J$.R(749113, 'p', p, false, false));
                                            return J$.Rt(749137, J$.R(749129, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(753953, J$e);
                                        } finally {
                                            if (J$.Fr(753961))
                                                continue jalangiLabel20;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(749225, J$.F(749217, J$.R(749193, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(749201, 'o', o, false, false), J$.R(749209, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(753969, J$e);
                        } finally {
                            if (J$.Fr(753977))
                                continue jalangiLabel21;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel23:
                    while (true) {
                        try {
                            J$.Fe(749417, arguments.callee, this, arguments);
                            arguments = J$.N(749425, 'arguments', arguments, true, false, false);
                            o = J$.N(749433, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(749377, '_getPrototypeOf', J$.C(43664, J$.G(749273, J$.I(typeof Object === 'undefined' ? Object = J$.R(749265, 'Object', undefined, true, true) : Object = J$.R(749265, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(749289, J$.I(typeof Object === 'undefined' ? Object = J$.R(749281, 'Object', undefined, true, true) : Object = J$.R(749281, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(749369, function _getPrototypeOf(o) {
                                jalangiLabel22:
                                    while (true) {
                                        try {
                                            J$.Fe(749345, arguments.callee, this, arguments);
                                            arguments = J$.N(749353, 'arguments', arguments, true, false, false);
                                            o = J$.N(749361, 'o', o, true, false, false);
                                            return J$.Rt(749337, J$.C(43656, J$.G(749305, J$.R(749297, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(749329, J$.I(typeof Object === 'undefined' ? Object = J$.R(749313, 'Object', undefined, true, true) : Object = J$.R(749313, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(749321, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(753985, J$e);
                                        } finally {
                                            if (J$.Fr(753993))
                                                continue jalangiLabel22;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(749409, J$.F(749401, J$.R(749385, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(749393, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(754001, J$e);
                        } finally {
                            if (J$.Fr(754009))
                                continue jalangiLabel23;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function makeWriteConcernResultObject(input) {
                jalangiLabel34:
                    while (true) {
                        try {
                            J$.Fe(751673, arguments.callee, this, arguments);
                            arguments = J$.N(751681, 'arguments', arguments, true, false, false);
                            input = J$.N(751689, 'input', input, true, false, false);
                            J$.N(751697, 'output', output, false, false, false);
                            var output = J$.W(751577, 'output', J$.M(751569, J$.I(typeof Object === 'undefined' ? Object = J$.R(751545, 'Object', undefined, true, true) : Object = J$.R(751545, 'Object', Object, true, true)), 'assign', false)(J$.T(751553, {}, 11, false), J$.R(751561, 'input', input, false, false)), output, false, false);
                            if (J$.C(43736, J$.B(89154, '===', J$.G(751593, J$.R(751585, 'output', output, false, false), 'ok'), J$.T(751601, 0, 22, false)))) {
                                J$.P(751625, J$.R(751609, 'output', output, false, false), 'ok', J$.T(751617, 1, 22, false));
                                J$.B(89162, 'delete', J$.R(751633, 'output', output, false, false), 'errmsg');
                                J$.B(89170, 'delete', J$.R(751641, 'output', output, false, false), 'code');
                                J$.B(89178, 'delete', J$.R(751649, 'output', output, false, false), 'codeName');
                            }
                            return J$.Rt(751665, J$.R(751657, 'output', output, false, false));
                        } catch (J$e) {
                            J$.Ex(754177, J$e);
                        } finally {
                            if (J$.Fr(754185))
                                continue jalangiLabel34;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isRetryableError(error) {
                jalangiLabel37:
                    while (true) {
                        try {
                            J$.Fe(752313, arguments.callee, this, arguments);
                            arguments = J$.N(752321, 'arguments', arguments, true, false, false);
                            error = J$.N(752329, 'error', error, true, false, false);
                            return J$.Rt(752305, J$.C(43768, J$.C(43760, J$.C(43752, J$.M(752217, J$.R(752193, 'RETRYABLE_ERROR_CODES', RETRYABLE_ERROR_CODES, false, true), 'has', false)(J$.G(752209, J$.R(752201, 'error', error, false, false), 'code'))) ? J$._() : J$.B(89194, 'instanceof', J$.R(752225, 'error', error, false, false), J$.R(752233, 'MongoNetworkError', MongoNetworkError, false, true))) ? J$._() : J$.M(752265, J$.G(752249, J$.R(752241, 'error', error, false, false), 'message'), 'match', false)(J$.T(752257, /not master/, 14, false))) ? J$._() : J$.M(752297, J$.G(752281, J$.R(752273, 'error', error, false, false), 'message'), 'match', false)(J$.T(752289, /node is recovering/, 14, false)));
                        } catch (J$e) {
                            J$.Ex(754225, J$e);
                        } finally {
                            if (J$.Fr(754233))
                                continue jalangiLabel37;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isRecoveringError(err) {
                jalangiLabel38:
                    while (true) {
                        try {
                            J$.Fe(752641, arguments.callee, this, arguments);
                            arguments = J$.N(752649, 'arguments', arguments, true, false, false);
                            err = J$.N(752657, 'err', err, true, false, false);
                            if (J$.C(43784, J$.C(43776, J$.G(752513, J$.R(752505, 'err', err, false, false), 'code')) ? J$.M(752545, J$.R(752521, 'SDAM_RECOVERING_CODES', SDAM_RECOVERING_CODES, false, true), 'has', false)(J$.G(752537, J$.R(752529, 'err', err, false, false), 'code')) : J$._())) {
                                return J$.Rt(752561, J$.T(752553, true, 23, false));
                            }
                            return J$.Rt(752633, J$.C(43792, J$.M(752593, J$.G(752577, J$.R(752569, 'err', err, false, false), 'message'), 'match', false)(J$.T(752585, /not master or secondary/, 14, false))) ? J$._() : J$.M(752625, J$.G(752609, J$.R(752601, 'err', err, false, false), 'message'), 'match', false)(J$.T(752617, /node is recovering/, 14, false)));
                        } catch (J$e) {
                            J$.Ex(754241, J$e);
                        } finally {
                            if (J$.Fr(754249))
                                continue jalangiLabel38;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isNotMasterError(err) {
                jalangiLabel39:
                    while (true) {
                        try {
                            J$.Fe(752809, arguments.callee, this, arguments);
                            arguments = J$.N(752817, 'arguments', arguments, true, false, false);
                            err = J$.N(752825, 'err', err, true, false, false);
                            if (J$.C(43808, J$.C(43800, J$.G(752673, J$.R(752665, 'err', err, false, false), 'code')) ? J$.M(752705, J$.R(752681, 'SDAM_NOTMASTER_CODES', SDAM_NOTMASTER_CODES, false, true), 'has', false)(J$.G(752697, J$.R(752689, 'err', err, false, false), 'code')) : J$._())) {
                                return J$.Rt(752721, J$.T(752713, true, 23, false));
                            }
                            if (J$.C(43816, J$.F(752745, J$.R(752729, 'isRecoveringError', isRecoveringError, false, true), false)(J$.R(752737, 'err', err, false, false)))) {
                                return J$.Rt(752761, J$.T(752753, false, 23, false));
                            }
                            return J$.Rt(752801, J$.M(752793, J$.G(752777, J$.R(752769, 'err', err, false, false), 'message'), 'match', false)(J$.T(752785, /not master/, 14, false)));
                        } catch (J$e) {
                            J$.Ex(754257, J$e);
                        } finally {
                            if (J$.Fr(754265))
                                continue jalangiLabel39;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isNodeShuttingDownError(err) {
                jalangiLabel40:
                    while (true) {
                        try {
                            J$.Fe(752889, arguments.callee, this, arguments);
                            arguments = J$.N(752897, 'arguments', arguments, true, false, false);
                            err = J$.N(752905, 'err', err, true, false, false);
                            return J$.Rt(752881, J$.C(43824, J$.G(752841, J$.R(752833, 'err', err, false, false), 'code')) ? J$.M(752873, J$.R(752849, 'SDAM_NODE_SHUTTING_DOWN_ERROR_CODES', SDAM_NODE_SHUTTING_DOWN_ERROR_CODES, false, true), 'has', false)(J$.G(752865, J$.R(752857, 'err', err, false, false), 'code')) : J$._());
                        } catch (J$e) {
                            J$.Ex(754273, J$e);
                        } finally {
                            if (J$.Fr(754281))
                                continue jalangiLabel40;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isSDAMUnrecoverableError(error, server) {
                jalangiLabel41:
                    while (true) {
                        try {
                            J$.Fe(753097, arguments.callee, this, arguments);
                            arguments = J$.N(753105, 'arguments', arguments, true, false, false);
                            error = J$.N(753113, 'error', error, true, false, false);
                            server = J$.N(753121, 'server', server, true, false, false);
                            if (J$.C(43832, J$.B(89202, 'instanceof', J$.R(752913, 'error', error, false, false), J$.R(752921, 'MongoParseError', MongoParseError, false, true)))) {
                                return J$.Rt(752937, J$.T(752929, true, 23, false));
                            }
                            if (J$.C(43864, J$.C(43840, J$.F(752961, J$.R(752945, 'isRecoveringError', isRecoveringError, false, true), false)(J$.R(752953, 'error', error, false, false))) ? J$._() : J$.F(752985, J$.R(752969, 'isNotMasterError', isNotMasterError, false, true), false)(J$.R(752977, 'error', error, false, false)))) {
                                if (J$.C(43856, J$.C(43848, J$.B(89210, '>=', J$.F(753009, J$.R(752993, 'maxWireVersion', maxWireVersion, false, true), false)(J$.R(753001, 'server', server, false, false)), J$.T(753017, 8, 22, false))) ? J$.U(89218, '!', J$.F(753041, J$.R(753025, 'isNodeShuttingDownError', isNodeShuttingDownError, false, true), false)(J$.R(753033, 'error', error, false, false))) : J$._())) {
                                    return J$.Rt(753057, J$.T(753049, false, 23, false));
                                }
                                return J$.Rt(753073, J$.T(753065, true, 23, false));
                            }
                            return J$.Rt(753089, J$.T(753081, false, 23, false));
                        } catch (J$e) {
                            J$.Ex(754289, J$e);
                        } finally {
                            if (J$.Fr(754297))
                                continue jalangiLabel41;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(753233, '_typeof2', J$.T(753225, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(753249, '_typeof', J$.T(753241, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(753265, '_classCallCheck', J$.T(753257, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(753281, '_defineProperties', J$.T(753273, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(753297, '_createClass', J$.T(753289, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(753313, '_possibleConstructorReturn', J$.T(753305, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(753329, '_assertThisInitialized', J$.T(753321, _assertThisInitialized, 12, false), true, false, false);
            _inherits = J$.N(753345, '_inherits', J$.T(753337, _inherits, 12, false), true, false, false);
            _wrapNativeSuper = J$.N(753361, '_wrapNativeSuper', J$.T(753353, _wrapNativeSuper, 12, false), true, false, false);
            isNativeReflectConstruct = J$.N(753377, 'isNativeReflectConstruct', J$.T(753369, isNativeReflectConstruct, 12, false), true, false, false);
            _construct = J$.N(753393, '_construct', J$.T(753385, _construct, 12, false), true, false, false);
            _isNativeFunction = J$.N(753409, '_isNativeFunction', J$.T(753401, _isNativeFunction, 12, false), true, false, false);
            _setPrototypeOf = J$.N(753425, '_setPrototypeOf', J$.T(753417, _setPrototypeOf, 12, false), true, false, false);
            _getPrototypeOf = J$.N(753441, '_getPrototypeOf', J$.T(753433, _getPrototypeOf, 12, false), true, false, false);
            J$.N(753449, 'mongoErrorContextSymbol', mongoErrorContextSymbol, false, false, false);
            J$.N(753457, 'maxWireVersion', maxWireVersion, false, false, false);
            J$.N(753465, 'MongoError', MongoError, false, false, false);
            J$.N(753473, 'MongoNetworkError', MongoNetworkError, false, false, false);
            J$.N(753481, 'MongoParseError', MongoParseError, false, false, false);
            J$.N(753489, 'MongoTimeoutError', MongoTimeoutError, false, false, false);
            makeWriteConcernResultObject = J$.N(753505, 'makeWriteConcernResultObject', J$.T(753497, makeWriteConcernResultObject, 12, false), true, false, false);
            J$.N(753513, 'MongoWriteConcernError', MongoWriteConcernError, false, false, false);
            J$.N(753521, 'RETRYABLE_ERROR_CODES', RETRYABLE_ERROR_CODES, false, false, false);
            isRetryableError = J$.N(753537, 'isRetryableError', J$.T(753529, isRetryableError, 12, false), true, false, false);
            J$.N(753545, 'SDAM_RECOVERING_CODES', SDAM_RECOVERING_CODES, false, false, false);
            J$.N(753553, 'SDAM_NOTMASTER_CODES', SDAM_NOTMASTER_CODES, false, false, false);
            J$.N(753561, 'SDAM_NODE_SHUTTING_DOWN_ERROR_CODES', SDAM_NODE_SHUTTING_DOWN_ERROR_CODES, false, false, false);
            isRecoveringError = J$.N(753577, 'isRecoveringError', J$.T(753569, isRecoveringError, 12, false), true, false, false);
            isNotMasterError = J$.N(753593, 'isNotMasterError', J$.T(753585, isNotMasterError, 12, false), true, false, false);
            isNodeShuttingDownError = J$.N(753609, 'isNodeShuttingDownError', J$.T(753601, isNodeShuttingDownError, 12, false), true, false, false);
            isSDAMUnrecoverableError = J$.N(753625, 'isSDAMUnrecoverableError', J$.T(753617, isSDAMUnrecoverableError, 12, false), true, false, false);
            J$.T(746169, 'use strict', 21, false);
            var mongoErrorContextSymbol = J$.W(749465, 'mongoErrorContextSymbol', J$.F(749457, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(749441, 'Symbol', undefined, true, true) : Symbol = J$.R(749441, 'Symbol', Symbol, true, true)), false)(J$.T(749449, 'mongoErrorContextSymbol', 21, false)), mongoErrorContextSymbol, false, true);
            var maxWireVersion = J$.W(749505, 'maxWireVersion', J$.G(749497, J$.F(749489, J$.I(typeof require === 'undefined' ? require = J$.R(749473, 'require', undefined, true, true) : require = J$.R(749473, 'require', require, true, true)), false)(J$.T(749481, './utils', 21, false)), 'maxWireVersion'), maxWireVersion, false, true);
            var MongoError = J$.W(750545, 'MongoError', J$.F(750537, J$.T(750505, function (_Error) {
                jalangiLabel27:
                    while (true) {
                        try {
                            J$.Fe(750465, arguments.callee, this, arguments);
                            function MongoError(message) {
                                jalangiLabel24:
                                    while (true) {
                                        try {
                                            J$.Fe(750177, arguments.callee, this, arguments);
                                            arguments = J$.N(750185, 'arguments', arguments, true, false, false);
                                            message = J$.N(750193, 'message', message, true, false, false);
                                            J$.N(750201, '_this', _this, false, false, false);
                                            J$.N(750209, 'name', name, false, false, false);
                                            var _this;
                                            J$.F(749569, J$.R(749545, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(749553, 'this', this, false, false), J$.R(749561, 'MongoError', MongoError, false, false));
                                            if (J$.C(43704, J$.B(89106, 'instanceof', J$.R(749577, 'message', message, false, false), J$.I(typeof Error === 'undefined' ? Error = J$.R(749585, 'Error', undefined, true, true) : Error = J$.R(749585, 'Error', Error, true, true))))) {
                                                _this = J$.W(749673, '_this', J$.F(749665, J$.R(749593, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(749601, 'this', this, false, false), J$.M(749657, J$.F(749625, J$.R(749609, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(749617, 'MongoError', MongoError, false, false)), 'call', false)(J$.R(749633, 'this', this, false, false), J$.G(749649, J$.R(749641, 'message', message, false, false), 'message'))), _this, false, false);
                                                J$.P(749705, J$.R(749681, '_this', _this, false, false), 'stack', J$.G(749697, J$.R(749689, 'message', message, false, false), 'stack'));
                                            } else {
                                                if (J$.C(43696, J$.B(89122, '===', J$.U(89114, 'typeof', J$.R(749713, 'message', message, false, false)), J$.T(749721, 'string', 21, false)))) {
                                                    _this = J$.W(749801, '_this', J$.F(749793, J$.R(749729, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(749737, 'this', this, false, false), J$.M(749785, J$.F(749761, J$.R(749745, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(749753, 'MongoError', MongoError, false, false)), 'call', false)(J$.R(749769, 'this', this, false, false), J$.R(749777, 'message', message, false, false))), _this, false, false);
                                                } else {
                                                    _this = J$.W(749929, '_this', J$.F(749921, J$.R(749809, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(749817, 'this', this, false, false), J$.M(749913, J$.F(749841, J$.R(749825, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(749833, 'MongoError', MongoError, false, false)), 'call', false)(J$.R(749849, 'this', this, false, false), J$.C(43688, J$.C(43680, J$.C(43672, J$.G(749865, J$.R(749857, 'message', message, false, false), 'message')) ? J$._() : J$.G(749881, J$.R(749873, 'message', message, false, false), 'errmsg')) ? J$._() : J$.G(749897, J$.R(749889, 'message', message, false, false), '$err')) ? J$._() : J$.T(749905, 'n/a', 21, false))), _this, false, false);
                                                    for (var name in J$.H(749993, J$.R(749937, 'message', message, false, false))) {
                                                        J$.N(750001, 'name', name, false, true, false);
                                                        {
                                                            {
                                                                J$.P(749985, J$.R(749945, '_this', _this, false, false), J$.R(749953, 'name', name, false, false), J$.G(749977, J$.R(749961, 'message', message, false, false), J$.R(749969, 'name', name, false, false)));
                                                            }
                                                        }
                                                    }
                                                }
                                                J$.M(750057, J$.I(typeof Error === 'undefined' ? Error = J$.R(750009, 'Error', undefined, true, true) : Error = J$.R(750009, 'Error', Error, true, true)), 'captureStackTrace', false)(J$.F(750033, J$.R(750017, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(750025, '_this', _this, false, false)), J$.G(750049, J$.R(750041, '_this', _this, false, false), 'constructor'));
                                            }
                                            J$.P(750081, J$.R(750065, '_this', _this, false, false), 'name', J$.T(750073, 'MongoError', 21, false));
                                            J$.P(750137, J$.R(750089, '_this', _this, false, false), J$.R(750097, 'mongoErrorContextSymbol', mongoErrorContextSymbol, false, true), J$.C(43712, J$.G(750121, J$.R(750105, '_this', _this, false, false), J$.R(750113, 'mongoErrorContextSymbol', mongoErrorContextSymbol, false, true))) ? J$._() : J$.T(750129, {}, 11, false));
                                            return J$.Rt(750169, J$.F(750161, J$.R(750145, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(750153, '_this', _this, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(754017, J$e);
                                        } finally {
                                            if (J$.Fr(754025))
                                                continue jalangiLabel24;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(750473, 'arguments', arguments, true, false, false);
                            _Error = J$.N(750481, '_Error', _Error, true, false, false);
                            MongoError = J$.N(750497, 'MongoError', J$.T(750489, MongoError, 12, false), true, false, false);
                            J$.F(749537, J$.R(749513, '_inherits', _inherits, false, true), false)(J$.R(749521, 'MongoError', MongoError, false, false), J$.R(749529, '_Error', _Error, false, false));
                            J$.F(750441, J$.R(750217, '_createClass', _createClass, false, true), false)(J$.R(750225, 'MongoError', MongoError, false, false), J$.T(750345, [J$.T(750337, {
                                    key: J$.T(750233, 'hasErrorLabel', 21, false),
                                    value: J$.T(750329, function hasErrorLabel(label) {
                                        jalangiLabel25:
                                            while (true) {
                                                try {
                                                    J$.Fe(750305, arguments.callee, this, arguments);
                                                    arguments = J$.N(750313, 'arguments', arguments, true, false, false);
                                                    label = J$.N(750321, 'label', label, true, false, false);
                                                    return J$.Rt(750297, J$.C(43720, J$.G(750249, J$.R(750241, 'this', this, false, false), 'errorLabels')) ? J$.B(89138, '!==', J$.M(750281, J$.G(750265, J$.R(750257, 'this', this, false, false), 'errorLabels'), 'indexOf', false)(J$.R(750273, 'label', label, false, false)), J$.U(89130, '-', J$.T(750289, 1, 22, false))) : J$._());
                                                } catch (J$e) {
                                                    J$.Ex(754033, J$e);
                                                } finally {
                                                    if (J$.Fr(754041))
                                                        continue jalangiLabel25;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false), J$.T(750433, [J$.T(750425, {
                                    key: J$.T(750353, 'create', 21, false),
                                    value: J$.T(750417, function create(options) {
                                        jalangiLabel26:
                                            while (true) {
                                                try {
                                                    J$.Fe(750393, arguments.callee, this, arguments);
                                                    arguments = J$.N(750401, 'arguments', arguments, true, false, false);
                                                    options = J$.N(750409, 'options', options, true, false, false);
                                                    return J$.Rt(750385, J$.F(750377, J$.R(750361, 'MongoError', MongoError, false, false), true)(J$.R(750369, 'options', options, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(754049, J$e);
                                                } finally {
                                                    if (J$.Fr(754057))
                                                        continue jalangiLabel26;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(750457, J$.R(750449, 'MongoError', MongoError, false, false));
                        } catch (J$e) {
                            J$.Ex(754065, J$e);
                        } finally {
                            if (J$.Fr(754073))
                                continue jalangiLabel27;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.F(750529, J$.R(750513, '_wrapNativeSuper', _wrapNativeSuper, false, true), false)(J$.I(typeof Error === 'undefined' ? Error = J$.R(750521, 'Error', undefined, true, true) : Error = J$.R(750521, 'Error', Error, true, true)))), MongoError, false, true);
            var MongoNetworkError = J$.W(750881, 'MongoNetworkError', J$.F(750873, J$.T(750857, function (_MongoError) {
                jalangiLabel29:
                    while (true) {
                        try {
                            J$.Fe(750817, arguments.callee, this, arguments);
                            function MongoNetworkError(message) {
                                jalangiLabel28:
                                    while (true) {
                                        try {
                                            J$.Fe(750769, arguments.callee, this, arguments);
                                            arguments = J$.N(750777, 'arguments', arguments, true, false, false);
                                            message = J$.N(750785, 'message', message, true, false, false);
                                            J$.N(750793, '_this2', _this2, false, false, false);
                                            var _this2;
                                            J$.F(750609, J$.R(750585, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(750593, 'this', this, false, false), J$.R(750601, 'MongoNetworkError', MongoNetworkError, false, false));
                                            _this2 = J$.W(750689, '_this2', J$.F(750681, J$.R(750617, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(750625, 'this', this, false, false), J$.M(750673, J$.F(750649, J$.R(750633, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(750641, 'MongoNetworkError', MongoNetworkError, false, false)), 'call', false)(J$.R(750657, 'this', this, false, false), J$.R(750665, 'message', message, false, false))), _this2, false, false);
                                            J$.P(750713, J$.R(750697, '_this2', _this2, false, false), 'name', J$.T(750705, 'MongoNetworkError', 21, false));
                                            J$.P(750745, J$.R(750721, '_this2', _this2, false, false), 'errorLabels', J$.T(750737, [J$.T(750729, 'TransientTransactionError', 21, false)], 10, false));
                                            return J$.Rt(750761, J$.R(750753, '_this2', _this2, false, false));
                                        } catch (J$e) {
                                            J$.Ex(754081, J$e);
                                        } finally {
                                            if (J$.Fr(754089))
                                                continue jalangiLabel28;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(750825, 'arguments', arguments, true, false, false);
                            _MongoError = J$.N(750833, '_MongoError', _MongoError, true, false, false);
                            MongoNetworkError = J$.N(750849, 'MongoNetworkError', J$.T(750841, MongoNetworkError, 12, false), true, false, false);
                            J$.F(750577, J$.R(750553, '_inherits', _inherits, false, true), false)(J$.R(750561, 'MongoNetworkError', MongoNetworkError, false, false), J$.R(750569, '_MongoError', _MongoError, false, false));
                            return J$.Rt(750809, J$.R(750801, 'MongoNetworkError', MongoNetworkError, false, false));
                        } catch (J$e) {
                            J$.Ex(754097, J$e);
                        } finally {
                            if (J$.Fr(754105))
                                continue jalangiLabel29;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(750865, 'MongoError', MongoError, false, true)), MongoNetworkError, false, true);
            var MongoParseError = J$.W(751185, 'MongoParseError', J$.F(751177, J$.T(751161, function (_MongoError2) {
                jalangiLabel31:
                    while (true) {
                        try {
                            J$.Fe(751121, arguments.callee, this, arguments);
                            function MongoParseError(message) {
                                jalangiLabel30:
                                    while (true) {
                                        try {
                                            J$.Fe(751073, arguments.callee, this, arguments);
                                            arguments = J$.N(751081, 'arguments', arguments, true, false, false);
                                            message = J$.N(751089, 'message', message, true, false, false);
                                            J$.N(751097, '_this3', _this3, false, false, false);
                                            var _this3;
                                            J$.F(750945, J$.R(750921, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(750929, 'this', this, false, false), J$.R(750937, 'MongoParseError', MongoParseError, false, false));
                                            _this3 = J$.W(751025, '_this3', J$.F(751017, J$.R(750953, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(750961, 'this', this, false, false), J$.M(751009, J$.F(750985, J$.R(750969, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(750977, 'MongoParseError', MongoParseError, false, false)), 'call', false)(J$.R(750993, 'this', this, false, false), J$.R(751001, 'message', message, false, false))), _this3, false, false);
                                            J$.P(751049, J$.R(751033, '_this3', _this3, false, false), 'name', J$.T(751041, 'MongoParseError', 21, false));
                                            return J$.Rt(751065, J$.R(751057, '_this3', _this3, false, false));
                                        } catch (J$e) {
                                            J$.Ex(754113, J$e);
                                        } finally {
                                            if (J$.Fr(754121))
                                                continue jalangiLabel30;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(751129, 'arguments', arguments, true, false, false);
                            _MongoError2 = J$.N(751137, '_MongoError2', _MongoError2, true, false, false);
                            MongoParseError = J$.N(751153, 'MongoParseError', J$.T(751145, MongoParseError, 12, false), true, false, false);
                            J$.F(750913, J$.R(750889, '_inherits', _inherits, false, true), false)(J$.R(750897, 'MongoParseError', MongoParseError, false, false), J$.R(750905, '_MongoError2', _MongoError2, false, false));
                            return J$.Rt(751113, J$.R(751105, 'MongoParseError', MongoParseError, false, false));
                        } catch (J$e) {
                            J$.Ex(754129, J$e);
                        } finally {
                            if (J$.Fr(754137))
                                continue jalangiLabel31;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(751169, 'MongoError', MongoError, false, true)), MongoParseError, false, true);
            var MongoTimeoutError = J$.W(751537, 'MongoTimeoutError', J$.F(751529, J$.T(751513, function (_MongoError3) {
                jalangiLabel33:
                    while (true) {
                        try {
                            J$.Fe(751473, arguments.callee, this, arguments);
                            function MongoTimeoutError(message, reason) {
                                jalangiLabel32:
                                    while (true) {
                                        try {
                                            J$.Fe(751417, arguments.callee, this, arguments);
                                            arguments = J$.N(751425, 'arguments', arguments, true, false, false);
                                            message = J$.N(751433, 'message', message, true, false, false);
                                            reason = J$.N(751441, 'reason', reason, true, false, false);
                                            J$.N(751449, '_this4', _this4, false, false, false);
                                            var _this4;
                                            J$.F(751249, J$.R(751225, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(751233, 'this', this, false, false), J$.R(751241, 'MongoTimeoutError', MongoTimeoutError, false, false));
                                            _this4 = J$.W(751329, '_this4', J$.F(751321, J$.R(751257, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(751265, 'this', this, false, false), J$.M(751313, J$.F(751289, J$.R(751273, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(751281, 'MongoTimeoutError', MongoTimeoutError, false, false)), 'call', false)(J$.R(751297, 'this', this, false, false), J$.R(751305, 'message', message, false, false))), _this4, false, false);
                                            J$.P(751353, J$.R(751337, '_this4', _this4, false, false), 'name', J$.T(751345, 'MongoTimeoutError', 21, false));
                                            if (J$.C(43728, J$.B(89146, '!=', J$.R(751361, 'reason', reason, false, false), J$.T(751369, null, 25, false)))) {
                                                J$.P(751393, J$.R(751377, '_this4', _this4, false, false), 'reason', J$.R(751385, 'reason', reason, false, false));
                                            }
                                            return J$.Rt(751409, J$.R(751401, '_this4', _this4, false, false));
                                        } catch (J$e) {
                                            J$.Ex(754145, J$e);
                                        } finally {
                                            if (J$.Fr(754153))
                                                continue jalangiLabel32;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(751481, 'arguments', arguments, true, false, false);
                            _MongoError3 = J$.N(751489, '_MongoError3', _MongoError3, true, false, false);
                            MongoTimeoutError = J$.N(751505, 'MongoTimeoutError', J$.T(751497, MongoTimeoutError, 12, false), true, false, false);
                            J$.F(751217, J$.R(751193, '_inherits', _inherits, false, true), false)(J$.R(751201, 'MongoTimeoutError', MongoTimeoutError, false, false), J$.R(751209, '_MongoError3', _MongoError3, false, false));
                            return J$.Rt(751465, J$.R(751457, 'MongoTimeoutError', MongoTimeoutError, false, false));
                        } catch (J$e) {
                            J$.Ex(754161, J$e);
                        } finally {
                            if (J$.Fr(754169))
                                continue jalangiLabel33;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(751521, 'MongoError', MongoError, false, true)), MongoTimeoutError, false, true);
            var MongoWriteConcernError = J$.W(752065, 'MongoWriteConcernError', J$.F(752057, J$.T(752041, function (_MongoError4) {
                jalangiLabel36:
                    while (true) {
                        try {
                            J$.Fe(752001, arguments.callee, this, arguments);
                            function MongoWriteConcernError(message, result) {
                                jalangiLabel35:
                                    while (true) {
                                        try {
                                            J$.Fe(751945, arguments.callee, this, arguments);
                                            arguments = J$.N(751953, 'arguments', arguments, true, false, false);
                                            message = J$.N(751961, 'message', message, true, false, false);
                                            result = J$.N(751969, 'result', result, true, false, false);
                                            J$.N(751977, '_this5', _this5, false, false, false);
                                            var _this5;
                                            J$.F(751761, J$.R(751737, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(751745, 'this', this, false, false), J$.R(751753, 'MongoWriteConcernError', MongoWriteConcernError, false, false));
                                            _this5 = J$.W(751841, '_this5', J$.F(751833, J$.R(751769, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(751777, 'this', this, false, false), J$.M(751825, J$.F(751801, J$.R(751785, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(751793, 'MongoWriteConcernError', MongoWriteConcernError, false, false)), 'call', false)(J$.R(751809, 'this', this, false, false), J$.R(751817, 'message', message, false, false))), _this5, false, false);
                                            J$.P(751865, J$.R(751849, '_this5', _this5, false, false), 'name', J$.T(751857, 'MongoWriteConcernError', 21, false));
                                            if (J$.C(43744, J$.B(89186, '!=', J$.R(751873, 'result', result, false, false), J$.T(751881, null, 25, false)))) {
                                                J$.P(751921, J$.R(751889, '_this5', _this5, false, false), 'result', J$.F(751913, J$.R(751897, 'makeWriteConcernResultObject', makeWriteConcernResultObject, false, true), false)(J$.R(751905, 'result', result, false, false)));
                                            }
                                            return J$.Rt(751937, J$.R(751929, '_this5', _this5, false, false));
                                        } catch (J$e) {
                                            J$.Ex(754193, J$e);
                                        } finally {
                                            if (J$.Fr(754201))
                                                continue jalangiLabel35;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(752009, 'arguments', arguments, true, false, false);
                            _MongoError4 = J$.N(752017, '_MongoError4', _MongoError4, true, false, false);
                            MongoWriteConcernError = J$.N(752033, 'MongoWriteConcernError', J$.T(752025, MongoWriteConcernError, 12, false), true, false, false);
                            J$.F(751729, J$.R(751705, '_inherits', _inherits, false, true), false)(J$.R(751713, 'MongoWriteConcernError', MongoWriteConcernError, false, false), J$.R(751721, '_MongoError4', _MongoError4, false, false));
                            return J$.Rt(751993, J$.R(751985, 'MongoWriteConcernError', MongoWriteConcernError, false, false));
                        } catch (J$e) {
                            J$.Ex(754209, J$e);
                        } finally {
                            if (J$.Fr(754217))
                                continue jalangiLabel36;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(752049, 'MongoError', MongoError, false, true)), MongoWriteConcernError, false, true);
            var RETRYABLE_ERROR_CODES = J$.W(752185, 'RETRYABLE_ERROR_CODES', J$.F(752177, J$.I(typeof Set === 'undefined' ? Set = J$.R(752073, 'Set', undefined, true, true) : Set = J$.R(752073, 'Set', Set, true, true)), true)(J$.T(752169, [
                J$.T(752081, 6, 22, false),
                J$.T(752089, 7, 22, false),
                J$.T(752097, 89, 22, false),
                J$.T(752105, 91, 22, false),
                J$.T(752113, 189, 22, false),
                J$.T(752121, 9001, 22, false),
                J$.T(752129, 10107, 22, false),
                J$.T(752137, 11600, 22, false),
                J$.T(752145, 11602, 22, false),
                J$.T(752153, 13435, 22, false),
                J$.T(752161, 13436, 22, false)
            ], 10, false)), RETRYABLE_ERROR_CODES, false, true);
            var SDAM_RECOVERING_CODES = J$.W(752401, 'SDAM_RECOVERING_CODES', J$.F(752393, J$.I(typeof Set === 'undefined' ? Set = J$.R(752337, 'Set', undefined, true, true) : Set = J$.R(752337, 'Set', Set, true, true)), true)(J$.T(752385, [
                J$.T(752345, 91, 22, false),
                J$.T(752353, 189, 22, false),
                J$.T(752361, 11600, 22, false),
                J$.T(752369, 11602, 22, false),
                J$.T(752377, 13436, 22, false)
            ], 10, false)), SDAM_RECOVERING_CODES, false, true);
            var SDAM_NOTMASTER_CODES = J$.W(752449, 'SDAM_NOTMASTER_CODES', J$.F(752441, J$.I(typeof Set === 'undefined' ? Set = J$.R(752409, 'Set', undefined, true, true) : Set = J$.R(752409, 'Set', Set, true, true)), true)(J$.T(752433, [
                J$.T(752417, 10107, 22, false),
                J$.T(752425, 13435, 22, false)
            ], 10, false)), SDAM_NOTMASTER_CODES, false, true);
            var SDAM_NODE_SHUTTING_DOWN_ERROR_CODES = J$.W(752497, 'SDAM_NODE_SHUTTING_DOWN_ERROR_CODES', J$.F(752489, J$.I(typeof Set === 'undefined' ? Set = J$.R(752457, 'Set', undefined, true, true) : Set = J$.R(752457, 'Set', Set, true, true)), true)(J$.T(752481, [
                J$.T(752465, 11600, 22, false),
                J$.T(752473, 91, 22, false)
            ], 10, false)), SDAM_NODE_SHUTTING_DOWN_ERROR_CODES, false, true);
            J$.P(753209, J$.I(typeof module === 'undefined' ? module = J$.R(753129, 'module', undefined, true, true) : module = J$.R(753129, 'module', module, true, true)), 'exports', J$.T(753201, {
                MongoError: J$.R(753137, 'MongoError', MongoError, false, true),
                MongoNetworkError: J$.R(753145, 'MongoNetworkError', MongoNetworkError, false, true),
                MongoParseError: J$.R(753153, 'MongoParseError', MongoParseError, false, true),
                MongoTimeoutError: J$.R(753161, 'MongoTimeoutError', MongoTimeoutError, false, true),
                MongoWriteConcernError: J$.R(753169, 'MongoWriteConcernError', MongoWriteConcernError, false, true),
                mongoErrorContextSymbol: J$.R(753177, 'mongoErrorContextSymbol', mongoErrorContextSymbol, false, true),
                isRetryableError: J$.R(753185, 'isRetryableError', isRetryableError, false, true),
                isSDAMUnrecoverableError: J$.R(753193, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(754305, J$e);
        } finally {
            if (J$.Sr(754313))
                continue jalangiLabel42;
            else
                break jalangiLabel42;
        }
    }
// JALANGI DO NOT INSTRUMENT
