// mongodb/lib/core/sessions.js
J$.noInstrEval = false;
jalangiLabel64:
    while (true) {
        try {
            J$.Se(831201, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/sessions.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(818121, arguments.callee, this, arguments);
                            arguments = J$.N(818129, 'arguments', arguments, true, false, false);
                            obj = J$.N(818137, 'obj', obj, true, false, false);
                            if (J$.C(47496, J$.C(47456, J$.B(92266, '===', J$.U(92258, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(817857, 'Symbol', undefined, true, true) : Symbol = J$.R(817857, 'Symbol', Symbol, true, true))), J$.T(817865, 'function', 21, false))) ? J$.B(92282, '===', J$.U(92274, 'typeof', J$.G(817881, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(817873, 'Symbol', undefined, true, true) : Symbol = J$.R(817873, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(817889, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(817945, '_typeof2', J$.T(817937, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(817913, arguments.callee, this, arguments);
                                                arguments = J$.N(817921, 'arguments', arguments, true, false, false);
                                                obj = J$.N(817929, 'obj', obj, true, false, false);
                                                return J$.Rt(817905, J$.U(92290, 'typeof', J$.R(817897, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(831793, J$e);
                                            } finally {
                                                if (J$.Fr(831801))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(818081, '_typeof2', J$.T(818073, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(818049, arguments.callee, this, arguments);
                                                arguments = J$.N(818057, 'arguments', arguments, true, false, false);
                                                obj = J$.N(818065, 'obj', obj, true, false, false);
                                                return J$.Rt(818041, J$.C(47488, J$.C(47480, J$.C(47472, J$.C(47464, J$.R(817953, 'obj', obj, false, false)) ? J$.B(92306, '===', J$.U(92298, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(817961, 'Symbol', undefined, true, true) : Symbol = J$.R(817961, 'Symbol', Symbol, true, true))), J$.T(817969, 'function', 21, false)) : J$._()) ? J$.B(92314, '===', J$.G(817985, J$.R(817977, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(817993, 'Symbol', undefined, true, true) : Symbol = J$.R(817993, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(92322, '!==', J$.R(818001, 'obj', obj, false, false), J$.G(818017, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(818009, 'Symbol', undefined, true, true) : Symbol = J$.R(818009, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(818025, 'symbol', 21, false) : J$.U(92330, 'typeof', J$.R(818033, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(831809, J$e);
                                            } finally {
                                                if (J$.Fr(831817))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(818113, J$.F(818105, J$.R(818089, '_typeof2', _typeof2, false, true), false)(J$.R(818097, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(831825, J$e);
                        } finally {
                            if (J$.Fr(831833))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperty(obj, key, value) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(818281, arguments.callee, this, arguments);
                            arguments = J$.N(818289, 'arguments', arguments, true, false, false);
                            obj = J$.N(818297, 'obj', obj, true, false, false);
                            key = J$.N(818305, 'key', key, true, false, false);
                            value = J$.N(818313, 'value', value, true, false, false);
                            if (J$.C(47504, J$.B(92338, 'in', J$.R(818145, 'key', key, false, false), J$.R(818153, 'obj', obj, false, false)))) {
                                J$.M(818225, J$.I(typeof Object === 'undefined' ? Object = J$.R(818161, 'Object', undefined, true, true) : Object = J$.R(818161, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(818169, 'obj', obj, false, false), J$.R(818177, 'key', key, false, false), J$.T(818217, {
                                    value: J$.R(818185, 'value', value, false, false),
                                    enumerable: J$.T(818193, true, 23, false),
                                    configurable: J$.T(818201, true, 23, false),
                                    writable: J$.T(818209, true, 23, false)
                                }, 11, false));
                            } else {
                                J$.P(818257, J$.R(818233, 'obj', obj, false, false), J$.R(818241, 'key', key, false, false), J$.R(818249, 'value', value, false, false));
                            }
                            return J$.Rt(818273, J$.R(818265, 'obj', obj, false, false));
                        } catch (J$e) {
                            J$.Ex(831841, J$e);
                        } finally {
                            if (J$.Fr(831849))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(818633, arguments.callee, this, arguments);
                            arguments = J$.N(818641, 'arguments', arguments, true, false, false);
                            obj = J$.N(818649, 'obj', obj, true, false, false);
                            if (J$.C(47552, J$.C(47512, J$.B(92354, '===', J$.U(92346, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(818321, 'Symbol', undefined, true, true) : Symbol = J$.R(818321, 'Symbol', Symbol, true, true))), J$.T(818329, 'function', 21, false))) ? J$.B(92362, '===', J$.F(818361, J$.R(818337, '_typeof2', _typeof2, false, true), false)(J$.G(818353, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(818345, 'Symbol', undefined, true, true) : Symbol = J$.R(818345, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(818369, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(818441, '_typeof', J$.T(818433, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(818409, arguments.callee, this, arguments);
                                                arguments = J$.N(818417, 'arguments', arguments, true, false, false);
                                                obj = J$.N(818425, 'obj', obj, true, false, false);
                                                return J$.Rt(818401, J$.F(818393, J$.R(818377, '_typeof2', _typeof2, false, true), false)(J$.R(818385, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(831857, J$e);
                                            } finally {
                                                if (J$.Fr(831865))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(818593, '_typeof', J$.T(818585, function _typeof(obj) {
                                    jalangiLabel5:
                                        while (true) {
                                            try {
                                                J$.Fe(818561, arguments.callee, this, arguments);
                                                arguments = J$.N(818569, 'arguments', arguments, true, false, false);
                                                obj = J$.N(818577, 'obj', obj, true, false, false);
                                                return J$.Rt(818553, J$.C(47544, J$.C(47536, J$.C(47528, J$.C(47520, J$.R(818449, 'obj', obj, false, false)) ? J$.B(92378, '===', J$.U(92370, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(818457, 'Symbol', undefined, true, true) : Symbol = J$.R(818457, 'Symbol', Symbol, true, true))), J$.T(818465, 'function', 21, false)) : J$._()) ? J$.B(92386, '===', J$.G(818481, J$.R(818473, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(818489, 'Symbol', undefined, true, true) : Symbol = J$.R(818489, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(92394, '!==', J$.R(818497, 'obj', obj, false, false), J$.G(818513, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(818505, 'Symbol', undefined, true, true) : Symbol = J$.R(818505, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(818521, 'symbol', 21, false) : J$.F(818545, J$.R(818529, '_typeof2', _typeof2, false, true), false)(J$.R(818537, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(831873, J$e);
                                            } finally {
                                                if (J$.Fr(831881))
                                                    continue jalangiLabel5;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(818625, J$.F(818617, J$.R(818601, '_typeof', _typeof, false, true), false)(J$.R(818609, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(831889, J$e);
                        } finally {
                            if (J$.Fr(831897))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(818697, arguments.callee, this, arguments);
                            arguments = J$.N(818705, 'arguments', arguments, true, false, false);
                            instance = J$.N(818713, 'instance', instance, true, false, false);
                            Constructor = J$.N(818721, 'Constructor', Constructor, true, false, false);
                            if (J$.C(47560, J$.U(92410, '!', J$.B(92402, 'instanceof', J$.R(818657, 'instance', instance, false, false), J$.R(818665, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(818689, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(818673, 'TypeError', undefined, true, true) : TypeError = J$.R(818673, 'TypeError', TypeError, true, true)), true)(J$.T(818681, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(831905, J$e);
                        } finally {
                            if (J$.Fr(831913))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(818969, arguments.callee, this, arguments);
                            arguments = J$.N(818977, 'arguments', arguments, true, false, false);
                            target = J$.N(818985, 'target', target, true, false, false);
                            props = J$.N(818993, 'props', props, true, false, false);
                            J$.N(819001, 'i', i, false, false, false);
                            J$.N(819009, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(818737, 'i', J$.T(818729, 0, 22, false), i, false, false); J$.C(47584, J$.B(92418, '<', J$.R(818745, 'i', i, false, false), J$.G(818761, J$.R(818753, 'props', props, false, false), 'length'))); J$.B(92442, '-', i = J$.W(818777, 'i', J$.B(92434, '+', J$.U(92426, '+', J$.R(818769, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(818809, 'descriptor', J$.G(818801, J$.R(818785, 'props', props, false, false), J$.R(818793, 'i', i, false, false)), descriptor, false, false);
                                J$.P(818849, J$.R(818817, 'descriptor', descriptor, false, false), 'enumerable', J$.C(47568, J$.G(818833, J$.R(818825, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(818841, false, 23, false));
                                J$.P(818873, J$.R(818857, 'descriptor', descriptor, false, false), 'configurable', J$.T(818865, true, 23, false));
                                if (J$.C(47576, J$.B(92450, 'in', J$.T(818881, 'value', 21, false), J$.R(818889, 'descriptor', descriptor, false, false))))
                                    J$.P(818913, J$.R(818897, 'descriptor', descriptor, false, false), 'writable', J$.T(818905, true, 23, false));
                                J$.M(818961, J$.I(typeof Object === 'undefined' ? Object = J$.R(818921, 'Object', undefined, true, true) : Object = J$.R(818921, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(818929, 'target', target, false, false), J$.G(818945, J$.R(818937, 'descriptor', descriptor, false, false), 'key'), J$.R(818953, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(831921, J$e);
                        } finally {
                            if (J$.Fr(831929))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(819121, arguments.callee, this, arguments);
                            arguments = J$.N(819129, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(819137, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(819145, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(819153, 'staticProps', staticProps, true, false, false);
                            if (J$.C(47592, J$.R(819017, 'protoProps', protoProps, false, false)))
                                J$.F(819057, J$.R(819025, '_defineProperties', _defineProperties, false, true), false)(J$.G(819041, J$.R(819033, 'Constructor', Constructor, false, false), 'prototype'), J$.R(819049, 'protoProps', protoProps, false, false));
                            if (J$.C(47600, J$.R(819065, 'staticProps', staticProps, false, false)))
                                J$.F(819097, J$.R(819073, '_defineProperties', _defineProperties, false, true), false)(J$.R(819081, 'Constructor', Constructor, false, false), J$.R(819089, 'staticProps', staticProps, false, false));
                            return J$.Rt(819113, J$.R(819105, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(831937, J$e);
                        } finally {
                            if (J$.Fr(831945))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(819265, arguments.callee, this, arguments);
                            arguments = J$.N(819273, 'arguments', arguments, true, false, false);
                            self = J$.N(819281, 'self', self, true, false, false);
                            call = J$.N(819289, 'call', call, true, false, false);
                            if (J$.C(47624, J$.C(47616, J$.R(819161, 'call', call, false, false)) ? J$.C(47608, J$.B(92458, '===', J$.F(819185, J$.R(819169, '_typeof', _typeof, false, true), false)(J$.R(819177, 'call', call, false, false)), J$.T(819193, 'object', 21, false))) ? J$._() : J$.B(92474, '===', J$.U(92466, 'typeof', J$.R(819201, 'call', call, false, false)), J$.T(819209, 'function', 21, false)) : J$._())) {
                                return J$.Rt(819225, J$.R(819217, 'call', call, false, false));
                            }
                            return J$.Rt(819257, J$.F(819249, J$.R(819233, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(819241, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(831953, J$e);
                        } finally {
                            if (J$.Fr(831961))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(819353, arguments.callee, this, arguments);
                            arguments = J$.N(819361, 'arguments', arguments, true, false, false);
                            self = J$.N(819369, 'self', self, true, false, false);
                            if (J$.C(47632, J$.B(92482, '===', J$.R(819297, 'self', self, false, false), void J$.T(819305, 0, 22, false)))) {
                                throw J$.F(819329, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(819313, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(819313, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(819321, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(819345, J$.R(819337, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(831969, J$e);
                        } finally {
                            if (J$.Fr(831977))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(819529, arguments.callee, this, arguments);
                            arguments = J$.N(819537, 'arguments', arguments, true, false, false);
                            o = J$.N(819545, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(819489, '_getPrototypeOf', J$.C(47648, J$.G(819385, J$.I(typeof Object === 'undefined' ? Object = J$.R(819377, 'Object', undefined, true, true) : Object = J$.R(819377, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(819401, J$.I(typeof Object === 'undefined' ? Object = J$.R(819393, 'Object', undefined, true, true) : Object = J$.R(819393, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(819481, function _getPrototypeOf(o) {
                                jalangiLabel12:
                                    while (true) {
                                        try {
                                            J$.Fe(819457, arguments.callee, this, arguments);
                                            arguments = J$.N(819465, 'arguments', arguments, true, false, false);
                                            o = J$.N(819473, 'o', o, true, false, false);
                                            return J$.Rt(819449, J$.C(47640, J$.G(819417, J$.R(819409, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(819441, J$.I(typeof Object === 'undefined' ? Object = J$.R(819425, 'Object', undefined, true, true) : Object = J$.R(819425, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(819433, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(831985, J$e);
                                        } finally {
                                            if (J$.Fr(831993))
                                                continue jalangiLabel12;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(819521, J$.F(819513, J$.R(819497, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(819505, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(832001, J$e);
                        } finally {
                            if (J$.Fr(832009))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(819745, arguments.callee, this, arguments);
                            arguments = J$.N(819753, 'arguments', arguments, true, false, false);
                            subClass = J$.N(819761, 'subClass', subClass, true, false, false);
                            superClass = J$.N(819769, 'superClass', superClass, true, false, false);
                            if (J$.C(47664, J$.C(47656, J$.B(92498, '!==', J$.U(92490, 'typeof', J$.R(819553, 'superClass', superClass, false, false)), J$.T(819561, 'function', 21, false))) ? J$.B(92506, '!==', J$.R(819569, 'superClass', superClass, false, false), J$.T(819577, null, 25, false)) : J$._())) {
                                throw J$.F(819601, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(819585, 'TypeError', undefined, true, true) : TypeError = J$.R(819585, 'TypeError', TypeError, true, true)), true)(J$.T(819593, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(819697, J$.R(819609, 'subClass', subClass, false, false), 'prototype', J$.M(819689, J$.I(typeof Object === 'undefined' ? Object = J$.R(819617, 'Object', undefined, true, true) : Object = J$.R(819617, 'Object', Object, true, true)), 'create', false)(J$.C(47672, J$.R(819625, 'superClass', superClass, false, false)) ? J$.G(819641, J$.R(819633, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(819681, {
                                constructor: J$.T(819673, {
                                    value: J$.R(819649, 'subClass', subClass, false, false),
                                    writable: J$.T(819657, true, 23, false),
                                    configurable: J$.T(819665, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(47680, J$.R(819705, 'superClass', superClass, false, false)))
                                J$.F(819737, J$.R(819713, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(819721, 'subClass', subClass, false, false), J$.R(819729, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(832017, J$e);
                        } finally {
                            if (J$.Fr(832025))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(819921, arguments.callee, this, arguments);
                            arguments = J$.N(819929, 'arguments', arguments, true, false, false);
                            o = J$.N(819937, 'o', o, true, false, false);
                            p = J$.N(819945, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(819873, '_setPrototypeOf', J$.C(47688, J$.G(819785, J$.I(typeof Object === 'undefined' ? Object = J$.R(819777, 'Object', undefined, true, true) : Object = J$.R(819777, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(819865, function _setPrototypeOf(o, p) {
                                jalangiLabel15:
                                    while (true) {
                                        try {
                                            J$.Fe(819833, arguments.callee, this, arguments);
                                            arguments = J$.N(819841, 'arguments', arguments, true, false, false);
                                            o = J$.N(819849, 'o', o, true, false, false);
                                            p = J$.N(819857, 'p', p, true, false, false);
                                            J$.P(819809, J$.R(819793, 'o', o, false, false), '__proto__', J$.R(819801, 'p', p, false, false));
                                            return J$.Rt(819825, J$.R(819817, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(832033, J$e);
                                        } finally {
                                            if (J$.Fr(832041))
                                                continue jalangiLabel15;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(819913, J$.F(819905, J$.R(819881, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(819889, 'o', o, false, false), J$.R(819897, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(832049, J$e);
                        } finally {
                            if (J$.Fr(832057))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function assertAlive(session, callback) {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(820745, arguments.callee, this, arguments);
                            arguments = J$.N(820753, 'arguments', arguments, true, false, false);
                            session = J$.N(820761, 'session', session, true, false, false);
                            callback = J$.N(820769, 'callback', callback, true, false, false);
                            J$.N(820777, 'error', error, false, false, false);
                            if (J$.C(47704, J$.B(92514, '==', J$.G(820609, J$.R(820601, 'session', session, false, false), 'serverSession'), J$.T(820617, null, 25, false)))) {
                                var error = J$.W(820649, 'error', J$.F(820641, J$.R(820625, 'MongoError', MongoError, false, true), true)(J$.T(820633, 'Cannot use a session that has ended', 21, false)), error, false, false);
                                if (J$.C(47696, J$.B(92530, '===', J$.U(92522, 'typeof', J$.R(820657, 'callback', callback, false, false)), J$.T(820665, 'function', 21, false)))) {
                                    J$.F(820697, J$.R(820673, 'callback', callback, false, false), false)(J$.R(820681, 'error', error, false, false), J$.T(820689, null, 25, false));
                                    return J$.Rt(820713, J$.T(820705, false, 23, false));
                                }
                                throw J$.R(820721, 'error', error, false, false);
                            }
                            return J$.Rt(820737, J$.T(820729, true, 23, false));
                        } catch (J$e) {
                            J$.Ex(832065, J$e);
                        } finally {
                            if (J$.Fr(832073))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function hasNotTimedOut(startTime, max) {
                jalangiLabel35:
                    while (true) {
                        try {
                            J$.Fe(824049, arguments.callee, this, arguments);
                            arguments = J$.N(824057, 'arguments', arguments, true, false, false);
                            startTime = J$.N(824065, 'startTime', startTime, true, false, false);
                            max = J$.N(824073, 'max', max, true, false, false);
                            return J$.Rt(824041, J$.B(92738, '<', J$.B(92730, '-', J$.M(824017, J$.I(typeof Date === 'undefined' ? Date = J$.R(824009, 'Date', undefined, true, true) : Date = J$.R(824009, 'Date', Date, true, true)), 'now', false)(), J$.R(824025, 'startTime', startTime, false, false)), J$.R(824033, 'max', max, false, false)));
                        } catch (J$e) {
                            J$.Ex(832353, J$e);
                        } finally {
                            if (J$.Fr(832361))
                                continue jalangiLabel35;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isUnknownTransactionCommitResult(err) {
                jalangiLabel36:
                    while (true) {
                        try {
                            J$.Fe(824193, arguments.callee, this, arguments);
                            arguments = J$.N(824201, 'arguments', arguments, true, false, false);
                            err = J$.N(824209, 'err', err, true, false, false);
                            return J$.Rt(824185, J$.C(47928, J$.F(824097, J$.R(824081, 'isMaxTimeMSExpiredError', isMaxTimeMSExpiredError, false, true), false)(J$.R(824089, 'err', err, false, false))) ? J$._() : J$.C(47920, J$.C(47912, J$.U(92746, '!', J$.M(824129, J$.R(824105, 'NON_DETERMINISTIC_WRITE_CONCERN_ERRORS', NON_DETERMINISTIC_WRITE_CONCERN_ERRORS, false, true), 'has', false)(J$.G(824121, J$.R(824113, 'err', err, false, false), 'codeName')))) ? J$.B(92754, '!==', J$.G(824145, J$.R(824137, 'err', err, false, false), 'code'), J$.R(824153, 'UNSATISFIABLE_WRITE_CONCERN_CODE', UNSATISFIABLE_WRITE_CONCERN_CODE, false, true)) : J$._()) ? J$.B(92762, '!==', J$.G(824169, J$.R(824161, 'err', err, false, false), 'code'), J$.R(824177, 'UNKNOWN_REPL_WRITE_CONCERN_CODE', UNKNOWN_REPL_WRITE_CONCERN_CODE, false, true)) : J$._());
                        } catch (J$e) {
                            J$.Ex(832369, J$e);
                        } finally {
                            if (J$.Fr(832377))
                                continue jalangiLabel36;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isMaxTimeMSExpiredError(err) {
                jalangiLabel37:
                    while (true) {
                        try {
                            J$.Fe(824297, arguments.callee, this, arguments);
                            arguments = J$.N(824305, 'arguments', arguments, true, false, false);
                            err = J$.N(824313, 'err', err, true, false, false);
                            return J$.Rt(824289, J$.C(47944, J$.B(92770, '===', J$.G(824225, J$.R(824217, 'err', err, false, false), 'code'), J$.R(824233, 'MAX_TIME_MS_EXPIRED_CODE', MAX_TIME_MS_EXPIRED_CODE, false, true))) ? J$._() : J$.C(47936, J$.G(824249, J$.R(824241, 'err', err, false, false), 'writeConcernError')) ? J$.B(92778, '===', J$.G(824273, J$.G(824265, J$.R(824257, 'err', err, false, false), 'writeConcernError'), 'code'), J$.R(824281, 'MAX_TIME_MS_EXPIRED_CODE', MAX_TIME_MS_EXPIRED_CODE, false, true)) : J$._());
                        } catch (J$e) {
                            J$.Ex(832385, J$e);
                        } finally {
                            if (J$.Fr(832393))
                                continue jalangiLabel37;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function attemptTransactionCommit(session, startTime, fn, options) {
                jalangiLabel39:
                    while (true) {
                        try {
                            J$.Fe(824633, arguments.callee, this, arguments);
                            arguments = J$.N(824641, 'arguments', arguments, true, false, false);
                            session = J$.N(824649, 'session', session, true, false, false);
                            startTime = J$.N(824657, 'startTime', startTime, true, false, false);
                            fn = J$.N(824665, 'fn', fn, true, false, false);
                            options = J$.N(824673, 'options', options, true, false, false);
                            return J$.Rt(824625, J$.M(824617, J$.M(824329, J$.R(824321, 'session', session, false, false), 'commitTransaction', false)(), J$.T(824337, 'catch', 21, false), false)(J$.T(824609, function (err) {
                                jalangiLabel38:
                                    while (true) {
                                        try {
                                            J$.Fe(824585, arguments.callee, this, arguments);
                                            arguments = J$.N(824593, 'arguments', arguments, true, false, false);
                                            err = J$.N(824601, 'err', err, true, false, false);
                                            if (J$.C(47984, J$.C(47960, J$.C(47952, J$.B(92786, 'instanceof', J$.R(824345, 'err', err, false, false), J$.R(824353, 'MongoError', MongoError, false, true))) ? J$.F(824385, J$.R(824361, 'hasNotTimedOut', hasNotTimedOut, false, true), false)(J$.R(824369, 'startTime', startTime, false, false), J$.R(824377, 'MAX_WITH_TRANSACTION_TIMEOUT', MAX_WITH_TRANSACTION_TIMEOUT, false, true)) : J$._()) ? J$.U(92794, '!', J$.F(824409, J$.R(824393, 'isMaxTimeMSExpiredError', isMaxTimeMSExpiredError, false, true), false)(J$.R(824401, 'err', err, false, false))) : J$._())) {
                                                if (J$.C(47968, J$.M(824433, J$.R(824417, 'err', err, false, false), 'hasErrorLabel', false)(J$.T(824425, 'UnknownTransactionCommitResult', 21, false)))) {
                                                    return J$.Rt(824489, J$.F(824481, J$.R(824441, 'attemptTransactionCommit', attemptTransactionCommit, false, true), false)(J$.R(824449, 'session', session, false, false), J$.R(824457, 'startTime', startTime, false, false), J$.R(824465, 'fn', fn, false, false), J$.R(824473, 'options', options, false, false)));
                                                }
                                                if (J$.C(47976, J$.M(824513, J$.R(824497, 'err', err, false, false), 'hasErrorLabel', false)(J$.T(824505, 'TransientTransactionError', 21, false)))) {
                                                    return J$.Rt(824569, J$.F(824561, J$.R(824521, 'attemptTransaction', attemptTransaction, false, true), false)(J$.R(824529, 'session', session, false, false), J$.R(824537, 'startTime', startTime, false, false), J$.R(824545, 'fn', fn, false, false), J$.R(824553, 'options', options, false, false)));
                                                }
                                            }
                                            throw J$.R(824577, 'err', err, false, false);
                                        } catch (J$e) {
                                            J$.Ex(832401, J$e);
                                        } finally {
                                            if (J$.Fr(832409))
                                                continue jalangiLabel38;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(832417, J$e);
                        } finally {
                            if (J$.Fr(832425))
                                continue jalangiLabel39;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function userExplicitlyEndedTransaction(session) {
                jalangiLabel40:
                    while (true) {
                        try {
                            J$.Fe(824809, arguments.callee, this, arguments);
                            arguments = J$.N(824817, 'arguments', arguments, true, false, false);
                            session = J$.N(824825, 'session', session, true, false, false);
                            return J$.Rt(824801, J$.M(824793, J$.R(824761, 'USER_EXPLICIT_TXN_END_STATES', USER_EXPLICIT_TXN_END_STATES, false, true), 'has', false)(J$.G(824785, J$.G(824777, J$.R(824769, 'session', session, false, false), 'transaction'), 'state')));
                        } catch (J$e) {
                            J$.Ex(832433, J$e);
                        } finally {
                            if (J$.Fr(832441))
                                continue jalangiLabel40;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function attemptTransaction(session, startTime, fn, options) {
                jalangiLabel45:
                    while (true) {
                        try {
                            J$.Fe(825601, arguments.callee, this, arguments);
                            arguments = J$.N(825609, 'arguments', arguments, true, false, false);
                            session = J$.N(825617, 'session', session, true, false, false);
                            startTime = J$.N(825625, 'startTime', startTime, true, false, false);
                            fn = J$.N(825633, 'fn', fn, true, false, false);
                            options = J$.N(825641, 'options', options, true, false, false);
                            J$.N(825649, 'promise', promise, false, false, false);
                            J$.M(824849, J$.R(824833, 'session', session, false, false), 'startTransaction', false)(J$.R(824841, 'options', options, false, false));
                            var promise;
                            try {
                                promise = J$.W(824881, 'promise', J$.F(824873, J$.R(824857, 'fn', fn, false, false), false)(J$.R(824865, 'session', session, false, false)), promise, false, false);
                            } catch (err) {
                                J$.N(824921, 'err', err, false, false, true);
                                promise = J$.W(824913, 'promise', J$.M(824905, J$.I(typeof Promise === 'undefined' ? Promise = J$.R(824889, 'Promise', undefined, true, true) : Promise = J$.R(824889, 'Promise', Promise, true, true)), 'reject', false)(J$.R(824897, 'err', err, false, false)), promise, false, false);
                            }
                            if (J$.C(47992, J$.U(92802, '!', J$.F(824945, J$.R(824929, 'isPromiseLike', isPromiseLike, false, true), false)(J$.R(824937, 'promise', promise, false, false))))) {
                                J$.M(824961, J$.R(824953, 'session', session, false, false), 'abortTransaction', false)();
                                throw J$.F(824985, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(824969, 'TypeError', undefined, true, true) : TypeError = J$.R(824969, 'TypeError', TypeError, true, true)), true)(J$.T(824977, 'Function provided to `withTransaction` must return a Promise', 21, false));
                            }
                            return J$.Rt(825593, J$.M(825585, J$.M(825113, J$.R(824993, 'promise', promise, false, false), 'then', false)(J$.T(825105, function () {
                                jalangiLabel41:
                                    while (true) {
                                        try {
                                            J$.Fe(825089, arguments.callee, this, arguments);
                                            arguments = J$.N(825097, 'arguments', arguments, true, false, false);
                                            if (J$.C(48000, J$.F(825017, J$.R(825001, 'userExplicitlyEndedTransaction', userExplicitlyEndedTransaction, false, true), false)(J$.R(825009, 'session', session, false, false)))) {
                                                return J$.Rt(825025, undefined);
                                            }
                                            return J$.Rt(825081, J$.F(825073, J$.R(825033, 'attemptTransactionCommit', attemptTransactionCommit, false, true), false)(J$.R(825041, 'session', session, false, false), J$.R(825049, 'startTime', startTime, false, false), J$.R(825057, 'fn', fn, false, false), J$.R(825065, 'options', options, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(832449, J$e);
                                        } finally {
                                            if (J$.Fr(832457))
                                                continue jalangiLabel41;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), J$.T(825121, 'catch', 21, false), false)(J$.T(825577, function (err) {
                                jalangiLabel44:
                                    while (true) {
                                        try {
                                            J$.Fe(825537, arguments.callee, this, arguments);
                                            function maybeRetryOrThrow(err) {
                                                jalangiLabel42:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(825369, arguments.callee, this, arguments);
                                                            arguments = J$.N(825377, 'arguments', arguments, true, false, false);
                                                            err = J$.N(825385, 'err', err, true, false, false);
                                                            if (J$.C(48024, J$.C(48016, J$.C(48008, J$.B(92810, 'instanceof', J$.R(825129, 'err', err, false, false), J$.R(825137, 'MongoError', MongoError, false, true))) ? J$.M(825161, J$.R(825145, 'err', err, false, false), 'hasErrorLabel', false)(J$.T(825153, 'TransientTransactionError', 21, false)) : J$._()) ? J$.F(825193, J$.R(825169, 'hasNotTimedOut', hasNotTimedOut, false, true), false)(J$.R(825177, 'startTime', startTime, false, false), J$.R(825185, 'MAX_WITH_TRANSACTION_TIMEOUT', MAX_WITH_TRANSACTION_TIMEOUT, false, true)) : J$._())) {
                                                                return J$.Rt(825249, J$.F(825241, J$.R(825201, 'attemptTransaction', attemptTransaction, false, true), false)(J$.R(825209, 'session', session, false, false), J$.R(825217, 'startTime', startTime, false, false), J$.R(825225, 'fn', fn, false, false), J$.R(825233, 'options', options, false, false)));
                                                            }
                                                            if (J$.C(48040, J$.F(825273, J$.R(825257, 'isMaxTimeMSExpiredError', isMaxTimeMSExpiredError, false, true), false)(J$.R(825265, 'err', err, false, false)))) {
                                                                if (J$.C(48032, J$.B(92818, '==', J$.G(825289, J$.R(825281, 'err', err, false, false), 'errorLabels'), J$.T(825297, null, 25, false)))) {
                                                                    J$.P(825321, J$.R(825305, 'err', err, false, false), 'errorLabels', J$.T(825313, [], 10, false));
                                                                }
                                                                J$.M(825353, J$.G(825337, J$.R(825329, 'err', err, false, false), 'errorLabels'), 'push', false)(J$.T(825345, 'UnknownTransactionCommitResult', 21, false));
                                                            }
                                                            throw J$.R(825361, 'err', err, false, false);
                                                        } catch (J$e) {
                                                            J$.Ex(832465, J$e);
                                                        } finally {
                                                            if (J$.Fr(832473))
                                                                continue jalangiLabel42;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }
                                            arguments = J$.N(825545, 'arguments', arguments, true, false, false);
                                            err = J$.N(825553, 'err', err, true, false, false);
                                            maybeRetryOrThrow = J$.N(825569, 'maybeRetryOrThrow', J$.T(825561, maybeRetryOrThrow, 12, false), true, false, false);
                                            if (J$.C(48048, J$.G(825409, J$.G(825401, J$.R(825393, 'session', session, false, false), 'transaction'), 'isActive'))) {
                                                return J$.Rt(825497, J$.M(825489, J$.M(825425, J$.R(825417, 'session', session, false, false), 'abortTransaction', false)(), 'then', false)(J$.T(825481, function () {
                                                    jalangiLabel43:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(825465, arguments.callee, this, arguments);
                                                                arguments = J$.N(825473, 'arguments', arguments, true, false, false);
                                                                return J$.Rt(825457, J$.F(825449, J$.R(825433, 'maybeRetryOrThrow', maybeRetryOrThrow, false, false), false)(J$.R(825441, 'err', err, false, false)));
                                                            } catch (J$e) {
                                                                J$.Ex(832481, J$e);
                                                            } finally {
                                                                if (J$.Fr(832489))
                                                                    continue jalangiLabel43;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false)));
                                            }
                                            return J$.Rt(825529, J$.F(825521, J$.R(825505, 'maybeRetryOrThrow', maybeRetryOrThrow, false, false), false)(J$.R(825513, 'err', err, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(832497, J$e);
                                        } finally {
                                            if (J$.Fr(832505))
                                                continue jalangiLabel44;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(832513, J$e);
                        } finally {
                            if (J$.Fr(832521))
                                continue jalangiLabel45;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function endTransaction(session, commandName, callback) {
                jalangiLabel50:
                    while (true) {
                        try {
                            J$.Fe(827841, arguments.callee, this, arguments);
                            function commandHandler(e, r) {
                                jalangiLabel46:
                                    while (true) {
                                        try {
                                            J$.Fe(827193, arguments.callee, this, arguments);
                                            arguments = J$.N(827201, 'arguments', arguments, true, false, false);
                                            e = J$.N(827209, 'e', e, true, false, false);
                                            r = J$.N(827217, 'r', r, true, false, false);
                                            J$.N(827225, 'idx', idx, false, false, false);
                                            if (J$.C(48256, J$.B(92922, '===', J$.R(826761, 'commandName', commandName, false, false), J$.T(826769, 'commitTransaction', 21, false)))) {
                                                J$.M(826809, J$.G(826785, J$.R(826777, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(826801, J$.R(826793, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED'));
                                                if (J$.C(48248, J$.C(48216, J$.R(826817, 'e', e, false, false)) ? J$.C(48208, J$.C(48200, J$.C(48192, J$.B(92930, 'instanceof', J$.R(826825, 'e', e, false, false), J$.R(826833, 'MongoNetworkError', MongoNetworkError, false, true))) ? J$._() : J$.B(92938, 'instanceof', J$.R(826841, 'e', e, false, false), J$.R(826849, 'MongoWriteConcernError', MongoWriteConcernError, false, true))) ? J$._() : J$.F(826873, J$.R(826857, 'isRetryableError', isRetryableError, false, true), false)(J$.R(826865, 'e', e, false, false))) ? J$._() : J$.F(826897, J$.R(826881, 'isMaxTimeMSExpiredError', isMaxTimeMSExpiredError, false, true), false)(J$.R(826889, 'e', e, false, false)) : J$._())) {
                                                    if (J$.C(48232, J$.G(826913, J$.R(826905, 'e', e, false, false), 'errorLabels'))) {
                                                        var idx = J$.W(826953, 'idx', J$.M(826945, J$.G(826929, J$.R(826921, 'e', e, false, false), 'errorLabels'), 'indexOf', false)(J$.T(826937, 'TransientTransactionError', 21, false)), idx, false, false);
                                                        if (J$.C(48224, J$.B(92954, '!==', J$.R(826961, 'idx', idx, false, false), J$.U(92946, '-', J$.T(826969, 1, 22, false))))) {
                                                            J$.M(827009, J$.G(826985, J$.R(826977, 'e', e, false, false), 'errorLabels'), 'splice', false)(J$.R(826993, 'idx', idx, false, false), J$.T(827001, 1, 22, false));
                                                        }
                                                    } else {
                                                        J$.P(827033, J$.R(827017, 'e', e, false, false), 'errorLabels', J$.T(827025, [], 10, false));
                                                    }
                                                    if (J$.C(48240, J$.F(827057, J$.R(827041, 'isUnknownTransactionCommitResult', isUnknownTransactionCommitResult, false, true), false)(J$.R(827049, 'e', e, false, false)))) {
                                                        J$.M(827089, J$.G(827073, J$.R(827065, 'e', e, false, false), 'errorLabels'), 'push', false)(J$.T(827081, 'UnknownTransactionCommitResult', 21, false));
                                                        J$.M(827113, J$.G(827105, J$.R(827097, 'session', session, false, false), 'transaction'), 'unpinServer', false)();
                                                    }
                                                }
                                            } else {
                                                J$.M(827153, J$.G(827129, J$.R(827121, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(827145, J$.R(827137, 'TxnState', TxnState, false, true), 'TRANSACTION_ABORTED'));
                                            }
                                            J$.F(827185, J$.R(827161, 'callback', callback, false, false), false)(J$.R(827169, 'e', e, false, false), J$.R(827177, 'r', r, false, false));
                                        } catch (J$e) {
                                            J$.Ex(832529, J$e);
                                        } finally {
                                            if (J$.Fr(832537))
                                                continue jalangiLabel46;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            function transactionError(err) {
                                jalangiLabel47:
                                    while (true) {
                                        try {
                                            J$.Fe(827273, arguments.callee, this, arguments);
                                            arguments = J$.N(827281, 'arguments', arguments, true, false, false);
                                            err = J$.N(827289, 'err', err, true, false, false);
                                            return J$.Rt(827265, J$.C(48264, J$.B(92962, '===', J$.R(827233, 'commandName', commandName, false, false), J$.T(827241, 'commitTransaction', 21, false))) ? J$.R(827249, 'err', err, false, false) : J$.T(827257, null, 25, false));
                                        } catch (J$e) {
                                            J$.Ex(832545, J$e);
                                        } finally {
                                            if (J$.Fr(832553))
                                                continue jalangiLabel47;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(827849, 'arguments', arguments, true, false, false);
                            session = J$.N(827857, 'session', session, true, false, false);
                            commandName = J$.N(827865, 'commandName', commandName, true, false, false);
                            callback = J$.N(827873, 'callback', callback, true, false, false);
                            J$.N(827881, 'txnState', txnState, false, false, false);
                            J$.N(827889, 'command', command, false, false, false);
                            J$.N(827897, 'writeConcern', writeConcern, false, false, false);
                            commandHandler = J$.N(827913, 'commandHandler', J$.T(827905, commandHandler, 12, false), true, false, false);
                            transactionError = J$.N(827929, 'transactionError', J$.T(827921, transactionError, 12, false), true, false, false);
                            if (J$.C(48056, J$.U(92826, '!', J$.F(825681, J$.R(825657, 'assertAlive', assertAlive, false, true), false)(J$.R(825665, 'session', session, false, false), J$.R(825673, 'callback', callback, false, false))))) {
                                return J$.Rt(825689, undefined);
                            }
                            var txnState = J$.W(825721, 'txnState', J$.G(825713, J$.G(825705, J$.R(825697, 'session', session, false, false), 'transaction'), 'state'), txnState, false, false);
                            if (J$.C(48064, J$.B(92834, '===', J$.R(825729, 'txnState', txnState, false, false), J$.G(825745, J$.R(825737, 'TxnState', TxnState, false, true), 'NO_TRANSACTION')))) {
                                J$.F(825785, J$.R(825753, 'callback', callback, false, false), false)(J$.F(825777, J$.R(825761, 'MongoError', MongoError, false, true), true)(J$.T(825769, 'No transaction started', 21, false)));
                                return J$.Rt(825793, undefined);
                            }
                            if (J$.C(48128, J$.B(92842, '===', J$.R(825801, 'commandName', commandName, false, false), J$.T(825809, 'commitTransaction', 21, false)))) {
                                if (J$.C(48080, J$.C(48072, J$.B(92850, '===', J$.R(825817, 'txnState', txnState, false, false), J$.G(825833, J$.R(825825, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION'))) ? J$._() : J$.B(92858, '===', J$.R(825841, 'txnState', txnState, false, false), J$.G(825857, J$.R(825849, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED_EMPTY')))) {
                                    J$.M(825897, J$.G(825873, J$.R(825865, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(825889, J$.R(825881, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED_EMPTY'));
                                    J$.F(825929, J$.R(825905, 'callback', callback, false, false), false)(J$.T(825913, null, 25, false), J$.T(825921, null, 25, false));
                                    return J$.Rt(825937, undefined);
                                }
                                if (J$.C(48088, J$.B(92866, '===', J$.R(825945, 'txnState', txnState, false, false), J$.G(825961, J$.R(825953, 'TxnState', TxnState, false, true), 'TRANSACTION_ABORTED')))) {
                                    J$.F(826001, J$.R(825969, 'callback', callback, false, false), false)(J$.F(825993, J$.R(825977, 'MongoError', MongoError, false, true), true)(J$.T(825985, 'Cannot call commitTransaction after calling abortTransaction', 21, false)));
                                    return J$.Rt(826009, undefined);
                                }
                            } else {
                                if (J$.C(48096, J$.B(92874, '===', J$.R(826017, 'txnState', txnState, false, false), J$.G(826033, J$.R(826025, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION')))) {
                                    J$.M(826073, J$.G(826049, J$.R(826041, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(826065, J$.R(826057, 'TxnState', TxnState, false, true), 'TRANSACTION_ABORTED'));
                                    J$.F(826105, J$.R(826081, 'callback', callback, false, false), false)(J$.T(826089, null, 25, false), J$.T(826097, null, 25, false));
                                    return J$.Rt(826113, undefined);
                                }
                                if (J$.C(48104, J$.B(92882, '===', J$.R(826121, 'txnState', txnState, false, false), J$.G(826137, J$.R(826129, 'TxnState', TxnState, false, true), 'TRANSACTION_ABORTED')))) {
                                    J$.F(826177, J$.R(826145, 'callback', callback, false, false), false)(J$.F(826169, J$.R(826153, 'MongoError', MongoError, false, true), true)(J$.T(826161, 'Cannot call abortTransaction twice', 21, false)));
                                    return J$.Rt(826185, undefined);
                                }
                                if (J$.C(48120, J$.C(48112, J$.B(92890, '===', J$.R(826193, 'txnState', txnState, false, false), J$.G(826209, J$.R(826201, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED'))) ? J$._() : J$.B(92898, '===', J$.R(826217, 'txnState', txnState, false, false), J$.G(826233, J$.R(826225, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED_EMPTY')))) {
                                    J$.F(826273, J$.R(826241, 'callback', callback, false, false), false)(J$.F(826265, J$.R(826249, 'MongoError', MongoError, false, true), true)(J$.T(826257, 'Cannot call abortTransaction after calling commitTransaction', 21, false)));
                                    return J$.Rt(826281, undefined);
                                }
                            }
                            var command = J$.W(826329, 'command', J$.F(826321, J$.R(826289, '_defineProperty', _defineProperty, false, true), false)(J$.T(826297, {}, 11, false), J$.R(826305, 'commandName', commandName, false, false), J$.T(826313, 1, 22, false)), command, false, false);
                            var writeConcern;
                            if (J$.C(48152, J$.G(826361, J$.G(826353, J$.G(826345, J$.R(826337, 'session', session, false, false), 'transaction'), 'options'), 'writeConcern'))) {
                                writeConcern = J$.W(826425, 'writeConcern', J$.M(826417, J$.I(typeof Object === 'undefined' ? Object = J$.R(826369, 'Object', undefined, true, true) : Object = J$.R(826369, 'Object', Object, true, true)), 'assign', false)(J$.T(826377, {}, 11, false), J$.G(826409, J$.G(826401, J$.G(826393, J$.R(826385, 'session', session, false, false), 'transaction'), 'options'), 'writeConcern')), writeConcern, false, false);
                            } else if (J$.C(48144, J$.C(48136, J$.G(826441, J$.R(826433, 'session', session, false, false), 'clientOptions')) ? J$.G(826465, J$.G(826457, J$.R(826449, 'session', session, false, false), 'clientOptions'), 'w') : J$._())) {
                                writeConcern = J$.W(826505, 'writeConcern', J$.T(826497, { w: J$.G(826489, J$.G(826481, J$.R(826473, 'session', session, false, false), 'clientOptions'), 'w') }, 11, false), writeConcern, false, false);
                            }
                            if (J$.C(48160, J$.B(92906, '===', J$.R(826513, 'txnState', txnState, false, false), J$.G(826529, J$.R(826521, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED')))) {
                                writeConcern = J$.W(826593, 'writeConcern', J$.M(826585, J$.I(typeof Object === 'undefined' ? Object = J$.R(826537, 'Object', undefined, true, true) : Object = J$.R(826537, 'Object', Object, true, true)), 'assign', false)(J$.T(826553, { wtimeout: J$.T(826545, 10000, 22, false) }, 11, false), J$.R(826561, 'writeConcern', writeConcern, false, false), J$.T(826577, { w: J$.T(826569, 'majority', 21, false) }, 11, false)), writeConcern, false, false);
                            }
                            if (J$.C(48168, J$.R(826601, 'writeConcern', writeConcern, false, false))) {
                                J$.M(826641, J$.I(typeof Object === 'undefined' ? Object = J$.R(826609, 'Object', undefined, true, true) : Object = J$.R(826609, 'Object', Object, true, true)), 'assign', false)(J$.R(826617, 'command', command, false, false), J$.T(826633, { writeConcern: J$.R(826625, 'writeConcern', writeConcern, false, false) }, 11, false));
                            }
                            if (J$.C(48184, J$.C(48176, J$.B(92914, '===', J$.R(826649, 'commandName', commandName, false, false), J$.T(826657, 'commitTransaction', 21, false))) ? J$.G(826689, J$.G(826681, J$.G(826673, J$.R(826665, 'session', session, false, false), 'transaction'), 'options'), 'maxTimeMS') : J$._())) {
                                J$.M(826753, J$.I(typeof Object === 'undefined' ? Object = J$.R(826697, 'Object', undefined, true, true) : Object = J$.R(826697, 'Object', Object, true, true)), 'assign', false)(J$.R(826705, 'command', command, false, false), J$.T(826745, { maxTimeMS: J$.G(826737, J$.G(826729, J$.G(826721, J$.R(826713, 'session', session, false, false), 'transaction'), 'options'), 'maxTimeMS') }, 11, false));
                            }
                            if (J$.C(48280, J$.C(48272, J$.G(827313, J$.G(827305, J$.R(827297, 'session', session, false, false), 'transaction'), 'recoveryToken')) ? J$.F(827337, J$.R(827321, 'supportsRecoveryToken', supportsRecoveryToken, false, true), false)(J$.R(827329, 'session', session, false, false)) : J$._())) {
                                J$.P(827377, J$.R(827345, 'command', command, false, false), 'recoveryToken', J$.G(827369, J$.G(827361, J$.R(827353, 'session', session, false, false), 'transaction'), 'recoveryToken'));
                            }
                            J$.M(827833, J$.G(827393, J$.R(827385, 'session', session, false, false), 'topology'), 'command', false)(J$.T(827401, 'admin.$cmd', 21, false), J$.R(827409, 'command', command, false, false), J$.T(827425, { session: J$.R(827417, 'session', session, false, false) }, 11, false), J$.T(827825, function (err, reply) {
                                jalangiLabel49:
                                    while (true) {
                                        try {
                                            J$.Fe(827793, arguments.callee, this, arguments);
                                            arguments = J$.N(827801, 'arguments', arguments, true, false, false);
                                            err = J$.N(827809, 'err', err, true, false, false);
                                            reply = J$.N(827817, 'reply', reply, true, false, false);
                                            if (J$.C(48304, J$.C(48288, J$.R(827433, 'err', err, false, false)) ? J$.F(827457, J$.R(827441, 'isRetryableError', isRetryableError, false, true), false)(J$.R(827449, 'err', err, false, false)) : J$._())) {
                                                if (J$.C(48296, J$.G(827473, J$.R(827465, 'command', command, false, false), 'commitTransaction'))) {
                                                    J$.M(827497, J$.G(827489, J$.R(827481, 'session', session, false, false), 'transaction'), 'unpinServer', false)();
                                                    J$.P(827577, J$.R(827505, 'command', command, false, false), 'writeConcern', J$.M(827569, J$.I(typeof Object === 'undefined' ? Object = J$.R(827513, 'Object', undefined, true, true) : Object = J$.R(827513, 'Object', Object, true, true)), 'assign', false)(J$.T(827529, { wtimeout: J$.T(827521, 10000, 22, false) }, 11, false), J$.G(827545, J$.R(827537, 'command', command, false, false), 'writeConcern'), J$.T(827561, { w: J$.T(827553, 'majority', 21, false) }, 11, false)));
                                                }
                                                return J$.Rt(827737, J$.M(827729, J$.G(827593, J$.R(827585, 'session', session, false, false), 'topology'), 'command', false)(J$.T(827601, 'admin.$cmd', 21, false), J$.R(827609, 'command', command, false, false), J$.T(827625, { session: J$.R(827617, 'session', session, false, false) }, 11, false), J$.T(827721, function (_err, _reply) {
                                                    jalangiLabel48:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(827689, arguments.callee, this, arguments);
                                                                arguments = J$.N(827697, 'arguments', arguments, true, false, false);
                                                                _err = J$.N(827705, '_err', _err, true, false, false);
                                                                _reply = J$.N(827713, '_reply', _reply, true, false, false);
                                                                return J$.Rt(827681, J$.F(827673, J$.R(827633, 'commandHandler', commandHandler, false, false), false)(J$.F(827657, J$.R(827641, 'transactionError', transactionError, false, false), false)(J$.R(827649, '_err', _err, false, false)), J$.R(827665, '_reply', _reply, false, false)));
                                                            } catch (J$e) {
                                                                J$.Ex(832561, J$e);
                                                            } finally {
                                                                if (J$.Fr(832569))
                                                                    continue jalangiLabel48;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false)));
                                            }
                                            J$.F(827785, J$.R(827745, 'commandHandler', commandHandler, false, false), false)(J$.F(827769, J$.R(827753, 'transactionError', transactionError, false, false), false)(J$.R(827761, 'err', err, false, false)), J$.R(827777, 'reply', reply, false, false));
                                        } catch (J$e) {
                                            J$.Ex(832577, J$e);
                                        } finally {
                                            if (J$.Fr(832585))
                                                continue jalangiLabel49;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(832593, J$e);
                        } finally {
                            if (J$.Fr(832601))
                                continue jalangiLabel50;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function supportsRecoveryToken(session) {
                jalangiLabel51:
                    while (true) {
                        try {
                            J$.Fe(828001, arguments.callee, this, arguments);
                            arguments = J$.N(828009, 'arguments', arguments, true, false, false);
                            session = J$.N(828017, 'session', session, true, false, false);
                            J$.N(828025, 'topology', topology, false, false, false);
                            var topology = J$.W(827953, 'topology', J$.G(827945, J$.R(827937, 'session', session, false, false), 'topology'), topology, false, false);
                            return J$.Rt(827993, J$.U(92978, '!', J$.U(92970, '!', J$.G(827985, J$.G(827977, J$.G(827969, J$.R(827961, 'topology', topology, false, false), 's'), 'options'), 'useRecoveryToken'))));
                        } catch (J$e) {
                            J$.Ex(832609, J$e);
                        } finally {
                            if (J$.Fr(832617))
                                continue jalangiLabel51;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function commandSupportsReadConcern(command, options) {
                jalangiLabel61:
                    while (true) {
                        try {
                            J$.Fe(829713, arguments.callee, this, arguments);
                            arguments = J$.N(829721, 'arguments', arguments, true, false, false);
                            command = J$.N(829729, 'command', command, true, false, false);
                            options = J$.N(829737, 'options', options, true, false, false);
                            if (J$.C(48424, J$.C(48416, J$.C(48408, J$.C(48400, J$.C(48392, J$.C(48384, J$.C(48376, J$.G(829473, J$.R(829465, 'command', command, false, false), 'aggregate')) ? J$._() : J$.G(829489, J$.R(829481, 'command', command, false, false), 'count')) ? J$._() : J$.G(829505, J$.R(829497, 'command', command, false, false), 'distinct')) ? J$._() : J$.G(829521, J$.R(829513, 'command', command, false, false), 'find')) ? J$._() : J$.G(829537, J$.R(829529, 'command', command, false, false), 'parallelCollectionScan')) ? J$._() : J$.G(829553, J$.R(829545, 'command', command, false, false), 'geoNear')) ? J$._() : J$.G(829569, J$.R(829561, 'command', command, false, false), 'geoSearch'))) {
                                return J$.Rt(829585, J$.T(829577, true, 23, false));
                            }
                            if (J$.C(48456, J$.C(48448, J$.C(48432, J$.G(829601, J$.R(829593, 'command', command, false, false), 'mapReduce')) ? J$.G(829617, J$.R(829609, 'options', options, false, false), 'out') : J$._()) ? J$.C(48440, J$.B(93066, '===', J$.G(829641, J$.G(829633, J$.R(829625, 'options', options, false, false), 'out'), 'inline'), J$.T(829649, 1, 22, false))) ? J$._() : J$.B(93074, '===', J$.G(829665, J$.R(829657, 'options', options, false, false), 'out'), J$.T(829673, 'inline', 21, false)) : J$._())) {
                                return J$.Rt(829689, J$.T(829681, true, 23, false));
                            }
                            return J$.Rt(829705, J$.T(829697, false, 23, false));
                        } catch (J$e) {
                            J$.Ex(832769, J$e);
                        } finally {
                            if (J$.Fr(832777))
                                continue jalangiLabel61;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function applySession(session, command, options) {
                jalangiLabel62:
                    while (true) {
                        try {
                            J$.Fe(830793, arguments.callee, this, arguments);
                            arguments = J$.N(830801, 'arguments', arguments, true, false, false);
                            session = J$.N(830809, 'session', session, true, false, false);
                            command = J$.N(830817, 'command', command, true, false, false);
                            options = J$.N(830825, 'options', options, true, false, false);
                            J$.N(830833, 'serverSession', serverSession, false, false, false);
                            J$.N(830841, 'inTransaction', inTransaction, false, false, false);
                            J$.N(830849, 'isRetryableWrite', isRetryableWrite, false, false, false);
                            J$.N(830857, 'shouldApplyReadConcern', shouldApplyReadConcern, false, false, false);
                            J$.N(830865, 'readConcern', readConcern, false, false, false);
                            var serverSession = J$.W(829761, 'serverSession', J$.G(829753, J$.R(829745, 'session', session, false, false), 'serverSession'), serverSession, false, false);
                            if (J$.C(48464, J$.B(93082, '==', J$.R(829769, 'serverSession', serverSession, false, false), J$.T(829777, null, 25, false)))) {
                                return J$.Rt(829809, J$.F(829801, J$.R(829785, 'MongoError', MongoError, false, true), true)(J$.T(829793, 'Cannot use a session that has ended', 21, false)));
                            }
                            J$.P(829841, J$.R(829817, 'serverSession', serverSession, false, false), 'lastUse', J$.M(829833, J$.I(typeof Date === 'undefined' ? Date = J$.R(829825, 'Date', undefined, true, true) : Date = J$.R(829825, 'Date', Date, true, true)), 'now', false)());
                            J$.P(829873, J$.R(829849, 'command', command, false, false), 'lsid', J$.G(829865, J$.R(829857, 'serverSession', serverSession, false, false), 'id'));
                            var inTransaction = J$.W(829921, 'inTransaction', J$.C(48472, J$.M(829889, J$.R(829881, 'session', session, false, false), 'inTransaction', false)()) ? J$._() : J$.F(829913, J$.R(829897, 'isTransactionCommand', isTransactionCommand, false, true), false)(J$.R(829905, 'command', command, false, false)), inTransaction, false, false);
                            var isRetryableWrite = J$.W(829945, 'isRetryableWrite', J$.G(829937, J$.R(829929, 'options', options, false, false), 'willRetryWrite'), isRetryableWrite, false, false);
                            var shouldApplyReadConcern = J$.W(829977, 'shouldApplyReadConcern', J$.F(829969, J$.R(829953, 'commandSupportsReadConcern', commandSupportsReadConcern, false, true), false)(J$.R(829961, 'command', command, false, false)), shouldApplyReadConcern, false, false);
                            if (J$.C(48496, J$.C(48488, J$.G(829993, J$.R(829985, 'serverSession', serverSession, false, false), 'txnNumber')) ? J$.C(48480, J$.R(830001, 'isRetryableWrite', isRetryableWrite, false, false)) ? J$._() : J$.R(830009, 'inTransaction', inTransaction, false, false) : J$._())) {
                                J$.P(830065, J$.R(830017, 'command', command, false, false), 'txnNumber', J$.M(830057, J$.G(830033, J$.R(830025, 'BSON', BSON, false, true), 'Long'), 'fromNumber', false)(J$.G(830049, J$.R(830041, 'serverSession', serverSession, false, false), 'txnNumber')));
                            }
                            if (J$.C(48544, J$.U(93090, '!', J$.R(830073, 'inTransaction', inTransaction, false, false)))) {
                                if (J$.C(48504, J$.B(93098, '!==', J$.G(830097, J$.G(830089, J$.R(830081, 'session', session, false, false), 'transaction'), 'state'), J$.G(830113, J$.R(830105, 'TxnState', TxnState, false, true), 'NO_TRANSACTION')))) {
                                    J$.M(830153, J$.G(830129, J$.R(830121, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(830145, J$.R(830137, 'TxnState', TxnState, false, true), 'NO_TRANSACTION'));
                                }
                                if (J$.C(48536, J$.C(48520, J$.C(48512, J$.G(830177, J$.G(830169, J$.R(830161, 'session', session, false, false), 'supports'), 'causalConsistency')) ? J$.G(830193, J$.R(830185, 'session', session, false, false), 'operationTime') : J$._()) ? J$.R(830201, 'shouldApplyReadConcern', shouldApplyReadConcern, false, false) : J$._())) {
                                    J$.P(830241, J$.R(830209, 'command', command, false, false), 'readConcern', J$.C(48528, J$.G(830225, J$.R(830217, 'command', command, false, false), 'readConcern')) ? J$._() : J$.T(830233, {}, 11, false));
                                    J$.M(830297, J$.I(typeof Object === 'undefined' ? Object = J$.R(830249, 'Object', undefined, true, true) : Object = J$.R(830249, 'Object', Object, true, true)), 'assign', false)(J$.G(830265, J$.R(830257, 'command', command, false, false), 'readConcern'), J$.T(830289, { afterClusterTime: J$.G(830281, J$.R(830273, 'session', session, false, false), 'operationTime') }, 11, false));
                                }
                                return J$.Rt(830305, undefined);
                            }
                            if (J$.C(48560, J$.C(48552, J$.G(830321, J$.R(830313, 'options', options, false, false), 'readPreference')) ? J$.U(93106, '!', J$.M(830361, J$.G(830337, J$.R(830329, 'options', options, false, false), 'readPreference'), 'equals', false)(J$.G(830353, J$.R(830345, 'ReadPreference', ReadPreference, false, true), 'primary'))) : J$._())) {
                                return J$.Rt(830425, J$.F(830417, J$.R(830369, 'MongoError', MongoError, false, true), true)(J$.M(830409, J$.T(830377, 'Read preference in a transaction must be primary, not: ', 21, false), 'concat', false)(J$.G(830401, J$.G(830393, J$.R(830385, 'options', options, false, false), 'readPreference'), 'mode'))));
                            }
                            J$.P(830449, J$.R(830433, 'command', command, false, false), 'autocommit', J$.T(830441, false, 23, false));
                            if (J$.C(48608, J$.B(93114, '===', J$.G(830473, J$.G(830465, J$.R(830457, 'session', session, false, false), 'transaction'), 'state'), J$.G(830489, J$.R(830481, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION')))) {
                                J$.M(830529, J$.G(830505, J$.R(830497, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(830521, J$.R(830513, 'TxnState', TxnState, false, true), 'TRANSACTION_IN_PROGRESS'));
                                J$.P(830553, J$.R(830537, 'command', command, false, false), 'startTransaction', J$.T(830545, true, 23, false));
                                var readConcern = J$.W(830617, 'readConcern', J$.C(48568, J$.G(830585, J$.G(830577, J$.G(830569, J$.R(830561, 'session', session, false, false), 'transaction'), 'options'), 'readConcern')) ? J$._() : J$.G(830609, J$.G(830601, J$.R(830593, 'session', session, false, false), 'clientOptions'), 'readConcern'), readConcern, false, false);
                                if (J$.C(48576, J$.R(830625, 'readConcern', readConcern, false, false))) {
                                    J$.P(830649, J$.R(830633, 'command', command, false, false), 'readConcern', J$.R(830641, 'readConcern', readConcern, false, false));
                                }
                                if (J$.C(48600, J$.C(48584, J$.G(830673, J$.G(830665, J$.R(830657, 'session', session, false, false), 'supports'), 'causalConsistency')) ? J$.G(830689, J$.R(830681, 'session', session, false, false), 'operationTime') : J$._())) {
                                    J$.P(830729, J$.R(830697, 'command', command, false, false), 'readConcern', J$.C(48592, J$.G(830713, J$.R(830705, 'command', command, false, false), 'readConcern')) ? J$._() : J$.T(830721, {}, 11, false));
                                    J$.M(830785, J$.I(typeof Object === 'undefined' ? Object = J$.R(830737, 'Object', undefined, true, true) : Object = J$.R(830737, 'Object', Object, true, true)), 'assign', false)(J$.G(830753, J$.R(830745, 'command', command, false, false), 'readConcern'), J$.T(830777, { afterClusterTime: J$.G(830769, J$.R(830761, 'session', session, false, false), 'operationTime') }, 11, false));
                                }
                            }
                        } catch (J$e) {
                            J$.Ex(832785, J$e);
                        } finally {
                            if (J$.Fr(832793))
                                continue jalangiLabel62;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function updateSessionFromResponse(session, document) {
                jalangiLabel63:
                    while (true) {
                        try {
                            J$.Fe(831089, arguments.callee, this, arguments);
                            arguments = J$.N(831097, 'arguments', arguments, true, false, false);
                            session = J$.N(831105, 'session', session, true, false, false);
                            document = J$.N(831113, 'document', document, true, false, false);
                            if (J$.C(48616, J$.G(830881, J$.R(830873, 'document', document, false, false), '$clusterTime'))) {
                                J$.F(830921, J$.R(830889, 'resolveClusterTime', resolveClusterTime, false, true), false)(J$.R(830897, 'session', session, false, false), J$.G(830913, J$.R(830905, 'document', document, false, false), '$clusterTime'));
                            }
                            if (J$.C(48640, J$.C(48632, J$.C(48624, J$.G(830937, J$.R(830929, 'document', document, false, false), 'operationTime')) ? J$.R(830945, 'session', session, false, false) : J$._()) ? J$.G(830969, J$.G(830961, J$.R(830953, 'session', session, false, false), 'supports'), 'causalConsistency') : J$._())) {
                                J$.M(831001, J$.R(830977, 'session', session, false, false), 'advanceOperationTime', false)(J$.G(830993, J$.R(830985, 'document', document, false, false), 'operationTime'));
                            }
                            if (J$.C(48664, J$.C(48656, J$.C(48648, J$.G(831017, J$.R(831009, 'document', document, false, false), 'recoveryToken')) ? J$.R(831025, 'session', session, false, false) : J$._()) ? J$.M(831041, J$.R(831033, 'session', session, false, false), 'inTransaction', false)() : J$._())) {
                                J$.P(831081, J$.G(831057, J$.R(831049, 'session', session, false, false), 'transaction'), '_recoveryToken', J$.G(831073, J$.R(831065, 'document', document, false, false), 'recoveryToken'));
                            }
                        } catch (J$e) {
                            J$.Ex(832801, J$e);
                        } finally {
                            if (J$.Fr(832809))
                                continue jalangiLabel63;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(831217, '_typeof2', J$.T(831209, _typeof2, 12, false), true, false, false);
            _defineProperty = J$.N(831233, '_defineProperty', J$.T(831225, _defineProperty, 12, false), true, false, false);
            _typeof = J$.N(831249, '_typeof', J$.T(831241, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(831265, '_classCallCheck', J$.T(831257, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(831281, '_defineProperties', J$.T(831273, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(831297, '_createClass', J$.T(831289, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(831313, '_possibleConstructorReturn', J$.T(831305, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(831329, '_assertThisInitialized', J$.T(831321, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(831345, '_getPrototypeOf', J$.T(831337, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(831361, '_inherits', J$.T(831353, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(831377, '_setPrototypeOf', J$.T(831369, _setPrototypeOf, 12, false), true, false, false);
            J$.N(831385, 'retrieveBSON', retrieveBSON, false, false, false);
            J$.N(831393, 'EventEmitter', EventEmitter, false, false, false);
            J$.N(831401, 'BSON', BSON, false, false, false);
            J$.N(831409, 'Binary', Binary, false, false, false);
            J$.N(831417, 'uuidV4', uuidV4, false, false, false);
            J$.N(831425, 'MongoError', MongoError, false, false, false);
            J$.N(831433, 'isRetryableError', isRetryableError, false, false, false);
            J$.N(831441, 'MongoNetworkError', MongoNetworkError, false, false, false);
            J$.N(831449, 'MongoWriteConcernError', MongoWriteConcernError, false, false, false);
            J$.N(831457, 'Transaction', Transaction, false, false, false);
            J$.N(831465, 'TxnState', TxnState, false, false, false);
            J$.N(831473, 'isPromiseLike', isPromiseLike, false, false, false);
            J$.N(831481, 'ReadPreference', ReadPreference, false, false, false);
            J$.N(831489, 'isTransactionCommand', isTransactionCommand, false, false, false);
            J$.N(831497, 'resolveClusterTime', resolveClusterTime, false, false, false);
            J$.N(831505, 'isSharded', isSharded, false, false, false);
            J$.N(831513, 'maxWireVersion', maxWireVersion, false, false, false);
            J$.N(831521, 'minWireVersionForShardedTransactions', minWireVersionForShardedTransactions, false, false, false);
            assertAlive = J$.N(831537, 'assertAlive', J$.T(831529, assertAlive, 12, false), true, false, false);
            J$.N(831545, 'ClientSession', ClientSession, false, false, false);
            J$.N(831553, 'MAX_WITH_TRANSACTION_TIMEOUT', MAX_WITH_TRANSACTION_TIMEOUT, false, false, false);
            J$.N(831561, 'UNSATISFIABLE_WRITE_CONCERN_CODE', UNSATISFIABLE_WRITE_CONCERN_CODE, false, false, false);
            J$.N(831569, 'UNKNOWN_REPL_WRITE_CONCERN_CODE', UNKNOWN_REPL_WRITE_CONCERN_CODE, false, false, false);
            J$.N(831577, 'MAX_TIME_MS_EXPIRED_CODE', MAX_TIME_MS_EXPIRED_CODE, false, false, false);
            J$.N(831585, 'NON_DETERMINISTIC_WRITE_CONCERN_ERRORS', NON_DETERMINISTIC_WRITE_CONCERN_ERRORS, false, false, false);
            hasNotTimedOut = J$.N(831601, 'hasNotTimedOut', J$.T(831593, hasNotTimedOut, 12, false), true, false, false);
            isUnknownTransactionCommitResult = J$.N(831617, 'isUnknownTransactionCommitResult', J$.T(831609, isUnknownTransactionCommitResult, 12, false), true, false, false);
            isMaxTimeMSExpiredError = J$.N(831633, 'isMaxTimeMSExpiredError', J$.T(831625, isMaxTimeMSExpiredError, 12, false), true, false, false);
            attemptTransactionCommit = J$.N(831649, 'attemptTransactionCommit', J$.T(831641, attemptTransactionCommit, 12, false), true, false, false);
            J$.N(831657, 'USER_EXPLICIT_TXN_END_STATES', USER_EXPLICIT_TXN_END_STATES, false, false, false);
            userExplicitlyEndedTransaction = J$.N(831673, 'userExplicitlyEndedTransaction', J$.T(831665, userExplicitlyEndedTransaction, 12, false), true, false, false);
            attemptTransaction = J$.N(831689, 'attemptTransaction', J$.T(831681, attemptTransaction, 12, false), true, false, false);
            endTransaction = J$.N(831705, 'endTransaction', J$.T(831697, endTransaction, 12, false), true, false, false);
            supportsRecoveryToken = J$.N(831721, 'supportsRecoveryToken', J$.T(831713, supportsRecoveryToken, 12, false), true, false, false);
            J$.N(831729, 'ServerSession', ServerSession, false, false, false);
            J$.N(831737, 'ServerSessionPool', ServerSessionPool, false, false, false);
            commandSupportsReadConcern = J$.N(831753, 'commandSupportsReadConcern', J$.T(831745, commandSupportsReadConcern, 12, false), true, false, false);
            applySession = J$.N(831769, 'applySession', J$.T(831761, applySession, 12, false), true, false, false);
            updateSessionFromResponse = J$.N(831785, 'updateSessionFromResponse', J$.T(831777, updateSessionFromResponse, 12, false), true, false, false);
            J$.T(817849, 'use strict', 21, false);
            var retrieveBSON = J$.W(819985, 'retrieveBSON', J$.G(819977, J$.F(819969, J$.I(typeof require === 'undefined' ? require = J$.R(819953, 'require', undefined, true, true) : require = J$.R(819953, 'require', require, true, true)), false)(J$.T(819961, './connection/utils', 21, false)), 'retrieveBSON'), retrieveBSON, false, true);
            var EventEmitter = J$.W(820017, 'EventEmitter', J$.F(820009, J$.I(typeof require === 'undefined' ? require = J$.R(819993, 'require', undefined, true, true) : require = J$.R(819993, 'require', require, true, true)), false)(J$.T(820001, 'events', 21, false)), EventEmitter, false, true);
            var BSON = J$.W(820041, 'BSON', J$.F(820033, J$.R(820025, 'retrieveBSON', retrieveBSON, false, true), false)(), BSON, false, true);
            var Binary = J$.W(820065, 'Binary', J$.G(820057, J$.R(820049, 'BSON', BSON, false, true), 'Binary'), Binary, false, true);
            var uuidV4 = J$.W(820105, 'uuidV4', J$.G(820097, J$.F(820089, J$.I(typeof require === 'undefined' ? require = J$.R(820073, 'require', undefined, true, true) : require = J$.R(820073, 'require', require, true, true)), false)(J$.T(820081, './utils', 21, false)), 'uuidV4'), uuidV4, false, true);
            var MongoError = J$.W(820145, 'MongoError', J$.G(820137, J$.F(820129, J$.I(typeof require === 'undefined' ? require = J$.R(820113, 'require', undefined, true, true) : require = J$.R(820113, 'require', require, true, true)), false)(J$.T(820121, './error', 21, false)), 'MongoError'), MongoError, false, true);
            var isRetryableError = J$.W(820185, 'isRetryableError', J$.G(820177, J$.F(820169, J$.I(typeof require === 'undefined' ? require = J$.R(820153, 'require', undefined, true, true) : require = J$.R(820153, 'require', require, true, true)), false)(J$.T(820161, '././error', 21, false)), 'isRetryableError'), isRetryableError, false, true);
            var MongoNetworkError = J$.W(820225, 'MongoNetworkError', J$.G(820217, J$.F(820209, J$.I(typeof require === 'undefined' ? require = J$.R(820193, 'require', undefined, true, true) : require = J$.R(820193, 'require', require, true, true)), false)(J$.T(820201, './error', 21, false)), 'MongoNetworkError'), MongoNetworkError, false, true);
            var MongoWriteConcernError = J$.W(820265, 'MongoWriteConcernError', J$.G(820257, J$.F(820249, J$.I(typeof require === 'undefined' ? require = J$.R(820233, 'require', undefined, true, true) : require = J$.R(820233, 'require', require, true, true)), false)(J$.T(820241, './error', 21, false)), 'MongoWriteConcernError'), MongoWriteConcernError, false, true);
            var Transaction = J$.W(820305, 'Transaction', J$.G(820297, J$.F(820289, J$.I(typeof require === 'undefined' ? require = J$.R(820273, 'require', undefined, true, true) : require = J$.R(820273, 'require', require, true, true)), false)(J$.T(820281, './transactions', 21, false)), 'Transaction'), Transaction, false, true);
            var TxnState = J$.W(820345, 'TxnState', J$.G(820337, J$.F(820329, J$.I(typeof require === 'undefined' ? require = J$.R(820313, 'require', undefined, true, true) : require = J$.R(820313, 'require', require, true, true)), false)(J$.T(820321, './transactions', 21, false)), 'TxnState'), TxnState, false, true);
            var isPromiseLike = J$.W(820385, 'isPromiseLike', J$.G(820377, J$.F(820369, J$.I(typeof require === 'undefined' ? require = J$.R(820353, 'require', undefined, true, true) : require = J$.R(820353, 'require', require, true, true)), false)(J$.T(820361, './utils', 21, false)), 'isPromiseLike'), isPromiseLike, false, true);
            var ReadPreference = J$.W(820417, 'ReadPreference', J$.F(820409, J$.I(typeof require === 'undefined' ? require = J$.R(820393, 'require', undefined, true, true) : require = J$.R(820393, 'require', require, true, true)), false)(J$.T(820401, './topologies/read_preference', 21, false)), ReadPreference, false, true);
            var isTransactionCommand = J$.W(820457, 'isTransactionCommand', J$.G(820449, J$.F(820441, J$.I(typeof require === 'undefined' ? require = J$.R(820425, 'require', undefined, true, true) : require = J$.R(820425, 'require', require, true, true)), false)(J$.T(820433, './transactions', 21, false)), 'isTransactionCommand'), isTransactionCommand, false, true);
            var resolveClusterTime = J$.W(820497, 'resolveClusterTime', J$.G(820489, J$.F(820481, J$.I(typeof require === 'undefined' ? require = J$.R(820465, 'require', undefined, true, true) : require = J$.R(820465, 'require', require, true, true)), false)(J$.T(820473, './topologies/shared', 21, false)), 'resolveClusterTime'), resolveClusterTime, false, true);
            var isSharded = J$.W(820537, 'isSharded', J$.G(820529, J$.F(820521, J$.I(typeof require === 'undefined' ? require = J$.R(820505, 'require', undefined, true, true) : require = J$.R(820505, 'require', require, true, true)), false)(J$.T(820513, './wireprotocol/shared', 21, false)), 'isSharded'), isSharded, false, true);
            var maxWireVersion = J$.W(820577, 'maxWireVersion', J$.G(820569, J$.F(820561, J$.I(typeof require === 'undefined' ? require = J$.R(820545, 'require', undefined, true, true) : require = J$.R(820545, 'require', require, true, true)), false)(J$.T(820553, './utils', 21, false)), 'maxWireVersion'), maxWireVersion, false, true);
            var minWireVersionForShardedTransactions = J$.W(820593, 'minWireVersionForShardedTransactions', J$.T(820585, 8, 22, false), minWireVersionForShardedTransactions, false, true);
            var ClientSession = J$.W(823881, 'ClientSession', J$.F(823873, J$.T(823857, function (_EventEmitter) {
                jalangiLabel34:
                    while (true) {
                        try {
                            J$.Fe(823817, arguments.callee, this, arguments);
                            function ClientSession(topology, sessionPool, options, clientOptions) {
                                jalangiLabel18:
                                    while (true) {
                                        try {
                                            J$.Fe(821489, arguments.callee, this, arguments);
                                            arguments = J$.N(821497, 'arguments', arguments, true, false, false);
                                            topology = J$.N(821505, 'topology', topology, true, false, false);
                                            sessionPool = J$.N(821513, 'sessionPool', sessionPool, true, false, false);
                                            options = J$.N(821521, 'options', options, true, false, false);
                                            clientOptions = J$.N(821529, 'clientOptions', clientOptions, true, false, false);
                                            J$.N(821537, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(820841, J$.R(820817, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(820825, 'this', this, false, false), J$.R(820833, 'ClientSession', ClientSession, false, false));
                                            _this = J$.W(820913, '_this', J$.F(820905, J$.R(820849, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(820857, 'this', this, false, false), J$.M(820897, J$.F(820881, J$.R(820865, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(820873, 'ClientSession', ClientSession, false, false)), 'call', false)(J$.R(820889, 'this', this, false, false))), _this, false, false);
                                            if (J$.C(47712, J$.B(92538, '==', J$.R(820921, 'topology', topology, false, false), J$.T(820929, null, 25, false)))) {
                                                throw J$.F(820953, J$.I(typeof Error === 'undefined' ? Error = J$.R(820937, 'Error', undefined, true, true) : Error = J$.R(820937, 'Error', Error, true, true)), true)(J$.T(820945, 'ClientSession requires a topology', 21, false));
                                            }
                                            if (J$.C(47728, J$.C(47720, J$.B(92546, '==', J$.R(820961, 'sessionPool', sessionPool, false, false), J$.T(820969, null, 25, false))) ? J$._() : J$.U(92562, '!', J$.B(92554, 'instanceof', J$.R(820977, 'sessionPool', sessionPool, false, false), J$.R(820985, 'ServerSessionPool', ServerSessionPool, false, true))))) {
                                                throw J$.F(821009, J$.I(typeof Error === 'undefined' ? Error = J$.R(820993, 'Error', undefined, true, true) : Error = J$.R(820993, 'Error', Error, true, true)), true)(J$.T(821001, 'ClientSession requires a ServerSessionPool', 21, false));
                                            }
                                            options = J$.W(821033, 'options', J$.C(47736, J$.R(821017, 'options', options, false, false)) ? J$._() : J$.T(821025, {}, 11, false), options, false, false);
                                            clientOptions = J$.W(821057, 'clientOptions', J$.C(47744, J$.R(821041, 'clientOptions', clientOptions, false, false)) ? J$._() : J$.T(821049, {}, 11, false), clientOptions, false, false);
                                            J$.P(821081, J$.R(821065, '_this', _this, false, false), 'topology', J$.R(821073, 'topology', topology, false, false));
                                            J$.P(821105, J$.R(821089, '_this', _this, false, false), 'sessionPool', J$.R(821097, 'sessionPool', sessionPool, false, false));
                                            J$.P(821129, J$.R(821113, '_this', _this, false, false), 'hasEnded', J$.T(821121, false, 23, false));
                                            J$.P(821161, J$.R(821137, '_this', _this, false, false), 'serverSession', J$.M(821153, J$.R(821145, 'sessionPool', sessionPool, false, false), 'acquire', false)());
                                            J$.P(821185, J$.R(821169, '_this', _this, false, false), 'clientOptions', J$.R(821177, 'clientOptions', clientOptions, false, false));
                                            J$.P(821257, J$.R(821193, '_this', _this, false, false), 'supports', J$.T(821249, { causalConsistency: J$.C(47752, J$.B(92578, '!==', J$.U(92570, 'typeof', J$.G(821209, J$.R(821201, 'options', options, false, false), 'causalConsistency')), J$.T(821217, 'undefined', 21, false))) ? J$.G(821233, J$.R(821225, 'options', options, false, false), 'causalConsistency') : J$.T(821241, true, 23, false) }, 11, false));
                                            J$.P(821289, J$.R(821265, '_this', _this, false, false), 'clusterTime', J$.G(821281, J$.R(821273, 'options', options, false, false), 'initialClusterTime'));
                                            J$.P(821313, J$.R(821297, '_this', _this, false, false), 'operationTime', J$.T(821305, null, 25, false));
                                            J$.P(821345, J$.R(821321, '_this', _this, false, false), 'explicit', J$.U(92594, '!', J$.U(92586, '!', J$.G(821337, J$.R(821329, 'options', options, false, false), 'explicit'))));
                                            J$.P(821377, J$.R(821353, '_this', _this, false, false), 'owner', J$.G(821369, J$.R(821361, 'options', options, false, false), 'owner'));
                                            J$.P(821433, J$.R(821385, '_this', _this, false, false), 'defaultTransactionOptions', J$.M(821425, J$.I(typeof Object === 'undefined' ? Object = J$.R(821393, 'Object', undefined, true, true) : Object = J$.R(821393, 'Object', Object, true, true)), 'assign', false)(J$.T(821401, {}, 11, false), J$.G(821417, J$.R(821409, 'options', options, false, false), 'defaultTransactionOptions')));
                                            J$.P(821465, J$.R(821441, '_this', _this, false, false), 'transaction', J$.F(821457, J$.R(821449, 'Transaction', Transaction, false, true), true)());
                                            return J$.Rt(821481, J$.R(821473, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(832081, J$e);
                                        } finally {
                                            if (J$.Fr(832089))
                                                continue jalangiLabel18;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(823825, 'arguments', arguments, true, false, false);
                            _EventEmitter = J$.N(823833, '_EventEmitter', _EventEmitter, true, false, false);
                            ClientSession = J$.N(823849, 'ClientSession', J$.T(823841, ClientSession, 12, false), true, false, false);
                            J$.F(820809, J$.R(820785, '_inherits', _inherits, false, true), false)(J$.R(820793, 'ClientSession', ClientSession, false, false), J$.R(820801, '_EventEmitter', _EventEmitter, false, false));
                            J$.F(823793, J$.R(821545, '_createClass', _createClass, false, true), false)(J$.R(821553, 'ClientSession', ClientSession, false, false), J$.T(823785, [
                                J$.T(821969, {
                                    key: J$.T(821561, 'endSession', 21, false),
                                    value: J$.T(821961, function endSession(options, callback) {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(821929, arguments.callee, this, arguments);
                                                    arguments = J$.N(821937, 'arguments', arguments, true, false, false);
                                                    options = J$.N(821945, 'options', options, true, false, false);
                                                    callback = J$.N(821953, 'callback', callback, true, false, false);
                                                    if (J$.C(47760, J$.B(92610, '===', J$.U(92602, 'typeof', J$.R(821569, 'options', options, false, false)), J$.T(821577, 'function', 21, false))))
                                                        callback = J$.W(821593, 'callback', J$.R(821585, 'options', options, false, false), callback, false, false), options = J$.W(821609, 'options', J$.T(821601, {}, 11, false), options, false, false);
                                                    options = J$.W(821633, 'options', J$.C(47768, J$.R(821617, 'options', options, false, false)) ? J$._() : J$.T(821625, {}, 11, false), options, false, false);
                                                    if (J$.C(47784, J$.G(821649, J$.R(821641, 'this', this, false, false), 'hasEnded'))) {
                                                        if (J$.C(47776, J$.B(92626, '===', J$.U(92618, 'typeof', J$.R(821657, 'callback', callback, false, false)), J$.T(821665, 'function', 21, false))))
                                                            J$.F(821697, J$.R(821673, 'callback', callback, false, false), false)(J$.T(821681, null, 25, false), J$.T(821689, null, 25, false));
                                                        return J$.Rt(821705, undefined);
                                                    }
                                                    if (J$.C(47800, J$.C(47792, J$.G(821721, J$.R(821713, 'this', this, false, false), 'serverSession')) ? J$.M(821737, J$.R(821729, 'this', this, false, false), 'inTransaction', false)() : J$._())) {
                                                        J$.M(821753, J$.R(821745, 'this', this, false, false), 'abortTransaction', false)();
                                                    }
                                                    J$.P(821777, J$.R(821761, 'this', this, false, false), 'hasEnded', J$.T(821769, true, 23, false));
                                                    J$.M(821809, J$.R(821785, 'this', this, false, false), 'emit', false)(J$.T(821793, 'ended', 21, false), J$.R(821801, 'this', this, false, false));
                                                    J$.M(821849, J$.G(821825, J$.R(821817, 'this', this, false, false), 'sessionPool'), 'release', false)(J$.G(821841, J$.R(821833, 'this', this, false, false), 'serverSession'));
                                                    J$.P(821873, J$.R(821857, 'this', this, false, false), 'serverSession', J$.T(821865, null, 25, false));
                                                    if (J$.C(47808, J$.B(92642, '===', J$.U(92634, 'typeof', J$.R(821881, 'callback', callback, false, false)), J$.T(821889, 'function', 21, false))))
                                                        J$.F(821921, J$.R(821897, 'callback', callback, false, false), false)(J$.T(821905, null, 25, false), J$.T(821913, null, 25, false));
                                                } catch (J$e) {
                                                    J$.Ex(832097, J$e);
                                                } finally {
                                                    if (J$.Fr(832105))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(822129, {
                                    key: J$.T(821977, 'advanceOperationTime', 21, false),
                                    value: J$.T(822121, function advanceOperationTime(operationTime) {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(822097, arguments.callee, this, arguments);
                                                    arguments = J$.N(822105, 'arguments', arguments, true, false, false);
                                                    operationTime = J$.N(822113, 'operationTime', operationTime, true, false, false);
                                                    if (J$.C(47816, J$.B(92650, '==', J$.G(821993, J$.R(821985, 'this', this, false, false), 'operationTime'), J$.T(822001, null, 25, false)))) {
                                                        J$.P(822025, J$.R(822009, 'this', this, false, false), 'operationTime', J$.R(822017, 'operationTime', operationTime, false, false));
                                                        return J$.Rt(822033, undefined);
                                                    }
                                                    if (J$.C(47824, J$.M(822065, J$.R(822041, 'operationTime', operationTime, false, false), 'greaterThan', false)(J$.G(822057, J$.R(822049, 'this', this, false, false), 'operationTime')))) {
                                                        J$.P(822089, J$.R(822073, 'this', this, false, false), 'operationTime', J$.R(822081, 'operationTime', operationTime, false, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(832113, J$e);
                                                } finally {
                                                    if (J$.Fr(832121))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(822289, {
                                    key: J$.T(822137, 'equals', 21, false),
                                    value: J$.T(822281, function equals(session) {
                                        jalangiLabel21:
                                            while (true) {
                                                try {
                                                    J$.Fe(822257, arguments.callee, this, arguments);
                                                    arguments = J$.N(822265, 'arguments', arguments, true, false, false);
                                                    session = J$.N(822273, 'session', session, true, false, false);
                                                    if (J$.C(47832, J$.U(92666, '!', J$.B(92658, 'instanceof', J$.R(822145, 'session', session, false, false), J$.R(822153, 'ClientSession', ClientSession, false, false))))) {
                                                        return J$.Rt(822169, J$.T(822161, false, 23, false));
                                                    }
                                                    return J$.Rt(822249, J$.M(822241, J$.G(822201, J$.G(822193, J$.G(822185, J$.R(822177, 'this', this, false, false), 'id'), 'id'), 'buffer'), 'equals', false)(J$.G(822233, J$.G(822225, J$.G(822217, J$.R(822209, 'session', session, false, false), 'id'), 'id'), 'buffer')));
                                                } catch (J$e) {
                                                    J$.Ex(832129, J$e);
                                                } finally {
                                                    if (J$.Fr(832137))
                                                        continue jalangiLabel21;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(822353, {
                                    key: J$.T(822297, 'incrementTransactionNumber', 21, false),
                                    value: J$.T(822345, function incrementTransactionNumber() {
                                        jalangiLabel22:
                                            while (true) {
                                                try {
                                                    J$.Fe(822329, arguments.callee, this, arguments);
                                                    arguments = J$.N(822337, 'arguments', arguments, true, false, false);
                                                    J$.B(92674, '-', J$.A(822321, J$.G(822313, J$.R(822305, 'this', this, false, false), 'serverSession'), 'txnNumber', '+')(1), 1);
                                                } catch (J$e) {
                                                    J$.Ex(832145, J$e);
                                                } finally {
                                                    if (J$.Fr(832153))
                                                        continue jalangiLabel22;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(822425, {
                                    key: J$.T(822361, 'inTransaction', 21, false),
                                    value: J$.T(822417, function inTransaction() {
                                        jalangiLabel23:
                                            while (true) {
                                                try {
                                                    J$.Fe(822401, arguments.callee, this, arguments);
                                                    arguments = J$.N(822409, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(822393, J$.G(822385, J$.G(822377, J$.R(822369, 'this', this, false, false), 'transaction'), 'isActive'));
                                                } catch (J$e) {
                                                    J$.Ex(832161, J$e);
                                                } finally {
                                                    if (J$.Fr(832169))
                                                        continue jalangiLabel23;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(822825, {
                                    key: J$.T(822433, 'startTransaction', 21, false),
                                    value: J$.T(822817, function startTransaction(options) {
                                        jalangiLabel24:
                                            while (true) {
                                                try {
                                                    J$.Fe(822785, arguments.callee, this, arguments);
                                                    arguments = J$.N(822793, 'arguments', arguments, true, false, false);
                                                    options = J$.N(822801, 'options', options, true, false, false);
                                                    J$.N(822809, 'topologyMaxWireVersion', topologyMaxWireVersion, false, false, false);
                                                    J$.F(822457, J$.R(822441, 'assertAlive', assertAlive, false, true), false)(J$.R(822449, 'this', this, false, false));
                                                    if (J$.C(47840, J$.M(822473, J$.R(822465, 'this', this, false, false), 'inTransaction', false)())) {
                                                        throw J$.F(822497, J$.R(822481, 'MongoError', MongoError, false, true), true)(J$.T(822489, 'Transaction already in progress', 21, false));
                                                    }
                                                    var topologyMaxWireVersion = J$.W(822537, 'topologyMaxWireVersion', J$.F(822529, J$.R(822505, 'maxWireVersion', maxWireVersion, false, true), false)(J$.G(822521, J$.R(822513, 'this', this, false, false), 'topology')), topologyMaxWireVersion, false, false);
                                                    if (J$.C(47864, J$.C(47856, J$.C(47848, J$.F(822569, J$.R(822545, 'isSharded', isSharded, false, true), false)(J$.G(822561, J$.R(822553, 'this', this, false, false), 'topology'))) ? J$.B(92682, '!=', J$.R(822577, 'topologyMaxWireVersion', topologyMaxWireVersion, false, false), J$.T(822585, null, 25, false)) : J$._()) ? J$.B(92690, '<', J$.R(822593, 'topologyMaxWireVersion', topologyMaxWireVersion, false, false), J$.R(822601, 'minWireVersionForShardedTransactions', minWireVersionForShardedTransactions, false, true)) : J$._())) {
                                                        throw J$.F(822625, J$.R(822609, 'MongoError', MongoError, false, true), true)(J$.T(822617, 'Transactions are not supported on sharded clusters in MongoDB < 4.2.', 21, false));
                                                    }
                                                    J$.M(822641, J$.R(822633, 'this', this, false, false), 'incrementTransactionNumber', false)();
                                                    J$.P(822737, J$.R(822649, 'this', this, false, false), 'transaction', J$.F(822729, J$.R(822657, 'Transaction', Transaction, false, true), true)(J$.M(822721, J$.I(typeof Object === 'undefined' ? Object = J$.R(822665, 'Object', undefined, true, true) : Object = J$.R(822665, 'Object', Object, true, true)), 'assign', false)(J$.T(822673, {}, 11, false), J$.G(822689, J$.R(822681, 'this', this, false, false), 'clientOptions'), J$.C(47872, J$.R(822697, 'options', options, false, false)) ? J$._() : J$.G(822713, J$.R(822705, 'this', this, false, false), 'defaultTransactionOptions'))));
                                                    J$.M(822777, J$.G(822753, J$.R(822745, 'this', this, false, false), 'transaction'), 'transition', false)(J$.G(822769, J$.R(822761, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION'));
                                                } catch (J$e) {
                                                    J$.Ex(832177, J$e);
                                                } finally {
                                                    if (J$.Fr(832185))
                                                        continue jalangiLabel24;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(823161, {
                                    key: J$.T(822833, 'commitTransaction', 21, false),
                                    value: J$.T(823153, function commitTransaction(callback) {
                                        jalangiLabel27:
                                            while (true) {
                                                try {
                                                    J$.Fe(823121, arguments.callee, this, arguments);
                                                    arguments = J$.N(823129, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(823137, 'callback', callback, true, false, false);
                                                    J$.N(823145, '_this2', _this2, false, false, false);
                                                    var _this2 = J$.W(822849, '_this2', J$.R(822841, 'this', this, false, false), _this2, false, false);
                                                    if (J$.C(47880, J$.B(92706, '===', J$.U(92698, 'typeof', J$.R(822857, 'callback', callback, false, false)), J$.T(822865, 'function', 21, false)))) {
                                                        J$.F(822905, J$.R(822873, 'endTransaction', endTransaction, false, true), false)(J$.R(822881, 'this', this, false, false), J$.T(822889, 'commitTransaction', 21, false), J$.R(822897, 'callback', callback, false, false));
                                                        return J$.Rt(822913, undefined);
                                                    }
                                                    return J$.Rt(823113, J$.F(823105, J$.I(typeof Promise === 'undefined' ? Promise = J$.R(822921, 'Promise', undefined, true, true) : Promise = J$.R(822921, 'Promise', Promise, true, true)), true)(J$.T(823097, function (resolve, reject) {
                                                        jalangiLabel26:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(823065, arguments.callee, this, arguments);
                                                                    arguments = J$.N(823073, 'arguments', arguments, true, false, false);
                                                                    resolve = J$.N(823081, 'resolve', resolve, true, false, false);
                                                                    reject = J$.N(823089, 'reject', reject, true, false, false);
                                                                    J$.F(823057, J$.R(822929, 'endTransaction', endTransaction, false, true), false)(J$.R(822937, '_this2', _this2, false, false), J$.T(822945, 'commitTransaction', 21, false), J$.T(823049, function (err, reply) {
                                                                        jalangiLabel25:
                                                                            while (true) {
                                                                                try {
                                                                                    J$.Fe(823017, arguments.callee, this, arguments);
                                                                                    arguments = J$.N(823025, 'arguments', arguments, true, false, false);
                                                                                    err = J$.N(823033, 'err', err, true, false, false);
                                                                                    reply = J$.N(823041, 'reply', reply, true, false, false);
                                                                                    return J$.Rt(823009, J$.C(47888, J$.R(822953, 'err', err, false, false)) ? J$.F(822977, J$.R(822961, 'reject', reject, false, false), false)(J$.R(822969, 'err', err, false, false)) : J$.F(823001, J$.R(822985, 'resolve', resolve, false, false), false)(J$.R(822993, 'reply', reply, false, false)));
                                                                                } catch (J$e) {
                                                                                    J$.Ex(832193, J$e);
                                                                                } finally {
                                                                                    if (J$.Fr(832201))
                                                                                        continue jalangiLabel25;
                                                                                    else
                                                                                        return J$.Ra();
                                                                                }
                                                                            }
                                                                    }, 12, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(832209, J$e);
                                                                } finally {
                                                                    if (J$.Fr(832217))
                                                                        continue jalangiLabel26;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false)));
                                                } catch (J$e) {
                                                    J$.Ex(832225, J$e);
                                                } finally {
                                                    if (J$.Fr(832233))
                                                        continue jalangiLabel27;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(823497, {
                                    key: J$.T(823169, 'abortTransaction', 21, false),
                                    value: J$.T(823489, function abortTransaction(callback) {
                                        jalangiLabel30:
                                            while (true) {
                                                try {
                                                    J$.Fe(823457, arguments.callee, this, arguments);
                                                    arguments = J$.N(823465, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(823473, 'callback', callback, true, false, false);
                                                    J$.N(823481, '_this3', _this3, false, false, false);
                                                    var _this3 = J$.W(823185, '_this3', J$.R(823177, 'this', this, false, false), _this3, false, false);
                                                    if (J$.C(47896, J$.B(92722, '===', J$.U(92714, 'typeof', J$.R(823193, 'callback', callback, false, false)), J$.T(823201, 'function', 21, false)))) {
                                                        J$.F(823241, J$.R(823209, 'endTransaction', endTransaction, false, true), false)(J$.R(823217, 'this', this, false, false), J$.T(823225, 'abortTransaction', 21, false), J$.R(823233, 'callback', callback, false, false));
                                                        return J$.Rt(823249, undefined);
                                                    }
                                                    return J$.Rt(823449, J$.F(823441, J$.I(typeof Promise === 'undefined' ? Promise = J$.R(823257, 'Promise', undefined, true, true) : Promise = J$.R(823257, 'Promise', Promise, true, true)), true)(J$.T(823433, function (resolve, reject) {
                                                        jalangiLabel29:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(823401, arguments.callee, this, arguments);
                                                                    arguments = J$.N(823409, 'arguments', arguments, true, false, false);
                                                                    resolve = J$.N(823417, 'resolve', resolve, true, false, false);
                                                                    reject = J$.N(823425, 'reject', reject, true, false, false);
                                                                    J$.F(823393, J$.R(823265, 'endTransaction', endTransaction, false, true), false)(J$.R(823273, '_this3', _this3, false, false), J$.T(823281, 'abortTransaction', 21, false), J$.T(823385, function (err, reply) {
                                                                        jalangiLabel28:
                                                                            while (true) {
                                                                                try {
                                                                                    J$.Fe(823353, arguments.callee, this, arguments);
                                                                                    arguments = J$.N(823361, 'arguments', arguments, true, false, false);
                                                                                    err = J$.N(823369, 'err', err, true, false, false);
                                                                                    reply = J$.N(823377, 'reply', reply, true, false, false);
                                                                                    return J$.Rt(823345, J$.C(47904, J$.R(823289, 'err', err, false, false)) ? J$.F(823313, J$.R(823297, 'reject', reject, false, false), false)(J$.R(823305, 'err', err, false, false)) : J$.F(823337, J$.R(823321, 'resolve', resolve, false, false), false)(J$.R(823329, 'reply', reply, false, false)));
                                                                                } catch (J$e) {
                                                                                    J$.Ex(832241, J$e);
                                                                                } finally {
                                                                                    if (J$.Fr(832249))
                                                                                        continue jalangiLabel28;
                                                                                    else
                                                                                        return J$.Ra();
                                                                                }
                                                                            }
                                                                    }, 12, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(832257, J$e);
                                                                } finally {
                                                                    if (J$.Fr(832265))
                                                                        continue jalangiLabel29;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false)));
                                                } catch (J$e) {
                                                    J$.Ex(832273, J$e);
                                                } finally {
                                                    if (J$.Fr(832281))
                                                        continue jalangiLabel30;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(823561, {
                                    key: J$.T(823505, 'toBSON', 21, false),
                                    value: J$.T(823553, function toBSON() {
                                        jalangiLabel31:
                                            while (true) {
                                                try {
                                                    J$.Fe(823537, arguments.callee, this, arguments);
                                                    arguments = J$.N(823545, 'arguments', arguments, true, false, false);
                                                    throw J$.F(823529, J$.I(typeof Error === 'undefined' ? Error = J$.R(823513, 'Error', undefined, true, true) : Error = J$.R(823513, 'Error', Error, true, true)), true)(J$.T(823521, 'ClientSession cannot be serialized to BSON.', 21, false));
                                                } catch (J$e) {
                                                    J$.Ex(832289, J$e);
                                                } finally {
                                                    if (J$.Fr(832297))
                                                        continue jalangiLabel31;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(823705, {
                                    key: J$.T(823569, 'withTransaction', 21, false),
                                    value: J$.T(823697, function withTransaction(fn, options) {
                                        jalangiLabel32:
                                            while (true) {
                                                try {
                                                    J$.Fe(823657, arguments.callee, this, arguments);
                                                    arguments = J$.N(823665, 'arguments', arguments, true, false, false);
                                                    fn = J$.N(823673, 'fn', fn, true, false, false);
                                                    options = J$.N(823681, 'options', options, true, false, false);
                                                    J$.N(823689, 'startTime', startTime, false, false, false);
                                                    var startTime = J$.W(823593, 'startTime', J$.M(823585, J$.I(typeof Date === 'undefined' ? Date = J$.R(823577, 'Date', undefined, true, true) : Date = J$.R(823577, 'Date', Date, true, true)), 'now', false)(), startTime, false, false);
                                                    return J$.Rt(823649, J$.F(823641, J$.R(823601, 'attemptTransaction', attemptTransaction, false, true), false)(J$.R(823609, 'this', this, false, false), J$.R(823617, 'startTime', startTime, false, false), J$.R(823625, 'fn', fn, false, false), J$.R(823633, 'options', options, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(832305, J$e);
                                                } finally {
                                                    if (J$.Fr(832313))
                                                        continue jalangiLabel32;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(823777, {
                                    key: J$.T(823713, 'id', 21, false),
                                    get: J$.T(823769, function get() {
                                        jalangiLabel33:
                                            while (true) {
                                                try {
                                                    J$.Fe(823753, arguments.callee, this, arguments);
                                                    arguments = J$.N(823761, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(823745, J$.G(823737, J$.G(823729, J$.R(823721, 'this', this, false, false), 'serverSession'), 'id'));
                                                } catch (J$e) {
                                                    J$.Ex(832321, J$e);
                                                } finally {
                                                    if (J$.Fr(832329))
                                                        continue jalangiLabel33;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(823809, J$.R(823801, 'ClientSession', ClientSession, false, false));
                        } catch (J$e) {
                            J$.Ex(832337, J$e);
                        } finally {
                            if (J$.Fr(832345))
                                continue jalangiLabel34;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(823865, 'EventEmitter', EventEmitter, false, true)), ClientSession, false, true);
            var MAX_WITH_TRANSACTION_TIMEOUT = J$.W(823897, 'MAX_WITH_TRANSACTION_TIMEOUT', J$.T(823889, 120000, 22, false), MAX_WITH_TRANSACTION_TIMEOUT, false, true);
            var UNSATISFIABLE_WRITE_CONCERN_CODE = J$.W(823913, 'UNSATISFIABLE_WRITE_CONCERN_CODE', J$.T(823905, 100, 22, false), UNSATISFIABLE_WRITE_CONCERN_CODE, false, true);
            var UNKNOWN_REPL_WRITE_CONCERN_CODE = J$.W(823929, 'UNKNOWN_REPL_WRITE_CONCERN_CODE', J$.T(823921, 79, 22, false), UNKNOWN_REPL_WRITE_CONCERN_CODE, false, true);
            var MAX_TIME_MS_EXPIRED_CODE = J$.W(823945, 'MAX_TIME_MS_EXPIRED_CODE', J$.T(823937, 50, 22, false), MAX_TIME_MS_EXPIRED_CODE, false, true);
            var NON_DETERMINISTIC_WRITE_CONCERN_ERRORS = J$.W(824001, 'NON_DETERMINISTIC_WRITE_CONCERN_ERRORS', J$.F(823993, J$.I(typeof Set === 'undefined' ? Set = J$.R(823953, 'Set', undefined, true, true) : Set = J$.R(823953, 'Set', Set, true, true)), true)(J$.T(823985, [
                J$.T(823961, 'CannotSatisfyWriteConcern', 21, false),
                J$.T(823969, 'UnknownReplWriteConcern', 21, false),
                J$.T(823977, 'UnsatisfiableWriteConcern', 21, false)
            ], 10, false)), NON_DETERMINISTIC_WRITE_CONCERN_ERRORS, false, true);
            var USER_EXPLICIT_TXN_END_STATES = J$.W(824753, 'USER_EXPLICIT_TXN_END_STATES', J$.F(824745, J$.I(typeof Set === 'undefined' ? Set = J$.R(824681, 'Set', undefined, true, true) : Set = J$.R(824681, 'Set', Set, true, true)), true)(J$.T(824737, [
                J$.G(824697, J$.R(824689, 'TxnState', TxnState, false, true), 'NO_TRANSACTION'),
                J$.G(824713, J$.R(824705, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED'),
                J$.G(824729, J$.R(824721, 'TxnState', TxnState, false, true), 'TRANSACTION_ABORTED')
            ], 10, false)), USER_EXPLICIT_TXN_END_STATES, false, true);
            var ServerSession = J$.W(828497, 'ServerSession', J$.F(828489, J$.T(828481, function () {
                jalangiLabel54:
                    while (true) {
                        try {
                            J$.Fe(828449, arguments.callee, this, arguments);
                            function ServerSession() {
                                jalangiLabel52:
                                    while (true) {
                                        try {
                                            J$.Fe(828217, arguments.callee, this, arguments);
                                            arguments = J$.N(828225, 'arguments', arguments, true, false, false);
                                            J$.F(828057, J$.R(828033, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(828041, 'this', this, false, false), J$.R(828049, 'ServerSession', ServerSession, false, false));
                                            J$.P(828129, J$.R(828065, 'this', this, false, false), 'id', J$.T(828121, { id: J$.F(828113, J$.R(828073, 'Binary', Binary, false, true), true)(J$.F(828089, J$.R(828081, 'uuidV4', uuidV4, false, true), false)(), J$.G(828105, J$.R(828097, 'Binary', Binary, false, true), 'SUBTYPE_UUID')) }, 11, false));
                                            J$.P(828161, J$.R(828137, 'this', this, false, false), 'lastUse', J$.M(828153, J$.I(typeof Date === 'undefined' ? Date = J$.R(828145, 'Date', undefined, true, true) : Date = J$.R(828145, 'Date', Date, true, true)), 'now', false)());
                                            J$.P(828185, J$.R(828169, 'this', this, false, false), 'txnNumber', J$.T(828177, 0, 22, false));
                                            J$.P(828209, J$.R(828193, 'this', this, false, false), 'isDirty', J$.T(828201, false, 23, false));
                                        } catch (J$e) {
                                            J$.Ex(832625, J$e);
                                        } finally {
                                            if (J$.Fr(832633))
                                                continue jalangiLabel52;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(828457, 'arguments', arguments, true, false, false);
                            ServerSession = J$.N(828473, 'ServerSession', J$.T(828465, ServerSession, 12, false), true, false, false);
                            J$.F(828425, J$.R(828233, '_createClass', _createClass, false, true), false)(J$.R(828241, 'ServerSession', ServerSession, false, false), J$.T(828417, [J$.T(828409, {
                                    key: J$.T(828249, 'hasTimedOut', 21, false),
                                    value: J$.T(828401, function hasTimedOut(sessionTimeoutMinutes) {
                                        jalangiLabel53:
                                            while (true) {
                                                try {
                                                    J$.Fe(828369, arguments.callee, this, arguments);
                                                    arguments = J$.N(828377, 'arguments', arguments, true, false, false);
                                                    sessionTimeoutMinutes = J$.N(828385, 'sessionTimeoutMinutes', sessionTimeoutMinutes, true, false, false);
                                                    J$.N(828393, 'idleTimeMinutes', idleTimeMinutes, false, false, false);
                                                    var idleTimeMinutes = J$.W(828329, 'idleTimeMinutes', J$.M(828321, J$.I(typeof Math === 'undefined' ? Math = J$.R(828257, 'Math', undefined, true, true) : Math = J$.R(828257, 'Math', Math, true, true)), 'round', false)(J$.B(93010, '/', J$.B(93002, '%', J$.B(92994, '%', J$.B(92986, '-', J$.M(828273, J$.I(typeof Date === 'undefined' ? Date = J$.R(828265, 'Date', undefined, true, true) : Date = J$.R(828265, 'Date', Date, true, true)), 'now', false)(), J$.G(828289, J$.R(828281, 'this', this, false, false), 'lastUse')), J$.T(828297, 86400000, 22, false)), J$.T(828305, 3600000, 22, false)), J$.T(828313, 60000, 22, false))), idleTimeMinutes, false, false);
                                                    return J$.Rt(828361, J$.B(93026, '>', J$.R(828337, 'idleTimeMinutes', idleTimeMinutes, false, false), J$.B(93018, '-', J$.R(828345, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false), J$.T(828353, 1, 22, false))));
                                                } catch (J$e) {
                                                    J$.Ex(832641, J$e);
                                                } finally {
                                                    if (J$.Fr(832649))
                                                        continue jalangiLabel53;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(828441, J$.R(828433, 'ServerSession', ServerSession, false, false));
                        } catch (J$e) {
                            J$.Ex(832657, J$e);
                        } finally {
                            if (J$.Fr(832665))
                                continue jalangiLabel54;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), ServerSession, false, true);
            var ServerSessionPool = J$.W(829457, 'ServerSessionPool', J$.F(829449, J$.T(829441, function () {
                jalangiLabel60:
                    while (true) {
                        try {
                            J$.Fe(829409, arguments.callee, this, arguments);
                            function ServerSessionPool(topology) {
                                jalangiLabel55:
                                    while (true) {
                                        try {
                                            J$.Fe(828625, arguments.callee, this, arguments);
                                            arguments = J$.N(828633, 'arguments', arguments, true, false, false);
                                            topology = J$.N(828641, 'topology', topology, true, false, false);
                                            J$.F(828529, J$.R(828505, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(828513, 'this', this, false, false), J$.R(828521, 'ServerSessionPool', ServerSessionPool, false, false));
                                            if (J$.C(48312, J$.B(93034, '==', J$.R(828537, 'topology', topology, false, false), J$.T(828545, null, 25, false)))) {
                                                throw J$.F(828569, J$.I(typeof Error === 'undefined' ? Error = J$.R(828553, 'Error', undefined, true, true) : Error = J$.R(828553, 'Error', Error, true, true)), true)(J$.T(828561, 'ServerSessionPool requires a topology', 21, false));
                                            }
                                            J$.P(828593, J$.R(828577, 'this', this, false, false), 'topology', J$.R(828585, 'topology', topology, false, false));
                                            J$.P(828617, J$.R(828601, 'this', this, false, false), 'sessions', J$.T(828609, [], 10, false));
                                        } catch (J$e) {
                                            J$.Ex(832673, J$e);
                                        } finally {
                                            if (J$.Fr(832681))
                                                continue jalangiLabel55;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(829417, 'arguments', arguments, true, false, false);
                            ServerSessionPool = J$.N(829433, 'ServerSessionPool', J$.T(829425, ServerSessionPool, 12, false), true, false, false);
                            J$.F(829385, J$.R(828649, '_createClass', _createClass, false, true), false)(J$.R(828657, 'ServerSessionPool', ServerSessionPool, false, false), J$.T(829377, [
                                J$.T(828849, {
                                    key: J$.T(828665, 'endAllPooledSessions', 21, false),
                                    value: J$.T(828841, function endAllPooledSessions() {
                                        jalangiLabel57:
                                            while (true) {
                                                try {
                                                    J$.Fe(828825, arguments.callee, this, arguments);
                                                    arguments = J$.N(828833, 'arguments', arguments, true, false, false);
                                                    if (J$.C(48320, J$.G(828689, J$.G(828681, J$.R(828673, 'this', this, false, false), 'sessions'), 'length'))) {
                                                        J$.M(828793, J$.G(828705, J$.R(828697, 'this', this, false, false), 'topology'), 'endSessions', false)(J$.M(828785, J$.G(828721, J$.R(828713, 'this', this, false, false), 'sessions'), 'map', false)(J$.T(828777, function (session) {
                                                            jalangiLabel56:
                                                                while (true) {
                                                                    try {
                                                                        J$.Fe(828753, arguments.callee, this, arguments);
                                                                        arguments = J$.N(828761, 'arguments', arguments, true, false, false);
                                                                        session = J$.N(828769, 'session', session, true, false, false);
                                                                        return J$.Rt(828745, J$.G(828737, J$.R(828729, 'session', session, false, false), 'id'));
                                                                    } catch (J$e) {
                                                                        J$.Ex(832689, J$e);
                                                                    } finally {
                                                                        if (J$.Fr(832697))
                                                                            continue jalangiLabel56;
                                                                        else
                                                                            return J$.Ra();
                                                                    }
                                                                }
                                                        }, 12, false)));
                                                        J$.P(828817, J$.R(828801, 'this', this, false, false), 'sessions', J$.T(828809, [], 10, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(832705, J$e);
                                                } finally {
                                                    if (J$.Fr(832713))
                                                        continue jalangiLabel57;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(829057, {
                                    key: J$.T(828857, 'acquire', 21, false),
                                    value: J$.T(829049, function acquire() {
                                        jalangiLabel58:
                                            while (true) {
                                                try {
                                                    J$.Fe(829017, arguments.callee, this, arguments);
                                                    arguments = J$.N(829025, 'arguments', arguments, true, false, false);
                                                    J$.N(829033, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false, false);
                                                    J$.N(829041, 'session', session, false, false, false);
                                                    var sessionTimeoutMinutes = J$.W(828889, 'sessionTimeoutMinutes', J$.G(828881, J$.G(828873, J$.R(828865, 'this', this, false, false), 'topology'), 'logicalSessionTimeoutMinutes'), sessionTimeoutMinutes, false, false);
                                                    while (J$.C(48336, J$.G(828913, J$.G(828905, J$.R(828897, 'this', this, false, false), 'sessions'), 'length'))) {
                                                        var session = J$.W(828945, 'session', J$.M(828937, J$.G(828929, J$.R(828921, 'this', this, false, false), 'sessions'), 'shift', false)(), session, false, false);
                                                        if (J$.C(48328, J$.U(93042, '!', J$.M(828969, J$.R(828953, 'session', session, false, false), 'hasTimedOut', false)(J$.R(828961, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false))))) {
                                                            return J$.Rt(828985, J$.R(828977, 'session', session, false, false));
                                                        }
                                                    }
                                                    return J$.Rt(829009, J$.F(829001, J$.R(828993, 'ServerSession', ServerSession, false, true), true)());
                                                } catch (J$e) {
                                                    J$.Ex(832721, J$e);
                                                } finally {
                                                    if (J$.Fr(832729))
                                                        continue jalangiLabel58;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(829369, {
                                    key: J$.T(829065, 'release', 21, false),
                                    value: J$.T(829361, function release(session) {
                                        jalangiLabel59:
                                            while (true) {
                                                try {
                                                    J$.Fe(829321, arguments.callee, this, arguments);
                                                    arguments = J$.N(829329, 'arguments', arguments, true, false, false);
                                                    session = J$.N(829337, 'session', session, true, false, false);
                                                    J$.N(829345, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false, false);
                                                    J$.N(829353, 'pooledSession', pooledSession, false, false, false);
                                                    var sessionTimeoutMinutes = J$.W(829097, 'sessionTimeoutMinutes', J$.G(829089, J$.G(829081, J$.R(829073, 'this', this, false, false), 'topology'), 'logicalSessionTimeoutMinutes'), sessionTimeoutMinutes, false, false);
                                                    while (J$.C(48352, J$.G(829121, J$.G(829113, J$.R(829105, 'this', this, false, false), 'sessions'), 'length'))) {
                                                        var pooledSession = J$.W(829185, 'pooledSession', J$.G(829177, J$.G(829137, J$.R(829129, 'this', this, false, false), 'sessions'), J$.B(93050, '-', J$.G(829161, J$.G(829153, J$.R(829145, 'this', this, false, false), 'sessions'), 'length'), J$.T(829169, 1, 22, false))), pooledSession, false, false);
                                                        if (J$.C(48344, J$.M(829209, J$.R(829193, 'pooledSession', pooledSession, false, false), 'hasTimedOut', false)(J$.R(829201, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false)))) {
                                                            J$.M(829233, J$.G(829225, J$.R(829217, 'this', this, false, false), 'sessions'), 'pop', false)();
                                                        } else {
                                                            break;
                                                        }
                                                    }
                                                    if (J$.C(48368, J$.U(93058, '!', J$.M(829257, J$.R(829241, 'session', session, false, false), 'hasTimedOut', false)(J$.R(829249, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false))))) {
                                                        if (J$.C(48360, J$.G(829273, J$.R(829265, 'session', session, false, false), 'isDirty'))) {
                                                            return J$.Rt(829281, undefined);
                                                        }
                                                        J$.M(829313, J$.G(829297, J$.R(829289, 'this', this, false, false), 'sessions'), 'unshift', false)(J$.R(829305, 'session', session, false, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(832737, J$e);
                                                } finally {
                                                    if (J$.Fr(832745))
                                                        continue jalangiLabel59;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(829401, J$.R(829393, 'ServerSessionPool', ServerSessionPool, false, false));
                        } catch (J$e) {
                            J$.Ex(832753, J$e);
                        } finally {
                            if (J$.Fr(832761))
                                continue jalangiLabel60;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), ServerSessionPool, false, true);
            J$.P(831193, J$.I(typeof module === 'undefined' ? module = J$.R(831121, 'module', undefined, true, true) : module = J$.R(831121, 'module', module, true, true)), 'exports', J$.T(831185, {
                ClientSession: J$.R(831129, 'ClientSession', ClientSession, false, true),
                ServerSession: J$.R(831137, 'ServerSession', ServerSession, false, true),
                ServerSessionPool: J$.R(831145, 'ServerSessionPool', ServerSessionPool, false, true),
                TxnState: J$.R(831153, 'TxnState', TxnState, false, true),
                applySession: J$.R(831161, 'applySession', applySession, false, true),
                updateSessionFromResponse: J$.R(831169, 'updateSessionFromResponse', updateSessionFromResponse, false, true),
                commandSupportsReadConcern: J$.R(831177, 'commandSupportsReadConcern', commandSupportsReadConcern, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(832817, J$e);
        } finally {
            if (J$.Sr(832825))
                continue jalangiLabel64;
            else
                break jalangiLabel64;
        }
    }
// JALANGI DO NOT INSTRUMENT
