// mongodb/lib/core/transactions.js
J$.noInstrEval = false;
jalangiLabel15:
    while (true) {
        try {
            J$.Se(929089, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/transactions.js');
            function _classCallCheck(instance, Constructor) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(926161, arguments.callee, this, arguments);
                            arguments = J$.N(926169, 'arguments', arguments, true, false, false);
                            instance = J$.N(926177, 'instance', instance, true, false, false);
                            Constructor = J$.N(926185, 'Constructor', Constructor, true, false, false);
                            if (J$.C(55376, J$.U(99186, '!', J$.B(99178, 'instanceof', J$.R(926121, 'instance', instance, false, false), J$.R(926129, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(926153, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(926137, 'TypeError', undefined, true, true) : TypeError = J$.R(926137, 'TypeError', TypeError, true, true)), true)(J$.T(926145, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(929209, J$e);
                        } finally {
                            if (J$.Fr(929217))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(926433, arguments.callee, this, arguments);
                            arguments = J$.N(926441, 'arguments', arguments, true, false, false);
                            target = J$.N(926449, 'target', target, true, false, false);
                            props = J$.N(926457, 'props', props, true, false, false);
                            J$.N(926465, 'i', i, false, false, false);
                            J$.N(926473, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(926201, 'i', J$.T(926193, 0, 22, false), i, false, false); J$.C(55400, J$.B(99194, '<', J$.R(926209, 'i', i, false, false), J$.G(926225, J$.R(926217, 'props', props, false, false), 'length'))); J$.B(99218, '-', i = J$.W(926241, 'i', J$.B(99210, '+', J$.U(99202, '+', J$.R(926233, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(926273, 'descriptor', J$.G(926265, J$.R(926249, 'props', props, false, false), J$.R(926257, 'i', i, false, false)), descriptor, false, false);
                                J$.P(926313, J$.R(926281, 'descriptor', descriptor, false, false), 'enumerable', J$.C(55384, J$.G(926297, J$.R(926289, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(926305, false, 23, false));
                                J$.P(926337, J$.R(926321, 'descriptor', descriptor, false, false), 'configurable', J$.T(926329, true, 23, false));
                                if (J$.C(55392, J$.B(99226, 'in', J$.T(926345, 'value', 21, false), J$.R(926353, 'descriptor', descriptor, false, false))))
                                    J$.P(926377, J$.R(926361, 'descriptor', descriptor, false, false), 'writable', J$.T(926369, true, 23, false));
                                J$.M(926425, J$.I(typeof Object === 'undefined' ? Object = J$.R(926385, 'Object', undefined, true, true) : Object = J$.R(926385, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(926393, 'target', target, false, false), J$.G(926409, J$.R(926401, 'descriptor', descriptor, false, false), 'key'), J$.R(926417, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(929225, J$e);
                        } finally {
                            if (J$.Fr(929233))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(926585, arguments.callee, this, arguments);
                            arguments = J$.N(926593, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(926601, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(926609, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(926617, 'staticProps', staticProps, true, false, false);
                            if (J$.C(55408, J$.R(926481, 'protoProps', protoProps, false, false)))
                                J$.F(926521, J$.R(926489, '_defineProperties', _defineProperties, false, true), false)(J$.G(926505, J$.R(926497, 'Constructor', Constructor, false, false), 'prototype'), J$.R(926513, 'protoProps', protoProps, false, false));
                            if (J$.C(55416, J$.R(926529, 'staticProps', staticProps, false, false)))
                                J$.F(926561, J$.R(926537, '_defineProperties', _defineProperties, false, true), false)(J$.R(926545, 'Constructor', Constructor, false, false), J$.R(926553, 'staticProps', staticProps, false, false));
                            return J$.Rt(926577, J$.R(926569, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(929241, J$e);
                        } finally {
                            if (J$.Fr(929249))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperty(obj, key, value) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(926761, arguments.callee, this, arguments);
                            arguments = J$.N(926769, 'arguments', arguments, true, false, false);
                            obj = J$.N(926777, 'obj', obj, true, false, false);
                            key = J$.N(926785, 'key', key, true, false, false);
                            value = J$.N(926793, 'value', value, true, false, false);
                            if (J$.C(55424, J$.B(99234, 'in', J$.R(926625, 'key', key, false, false), J$.R(926633, 'obj', obj, false, false)))) {
                                J$.M(926705, J$.I(typeof Object === 'undefined' ? Object = J$.R(926641, 'Object', undefined, true, true) : Object = J$.R(926641, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(926649, 'obj', obj, false, false), J$.R(926657, 'key', key, false, false), J$.T(926697, {
                                    value: J$.R(926665, 'value', value, false, false),
                                    enumerable: J$.T(926673, true, 23, false),
                                    configurable: J$.T(926681, true, 23, false),
                                    writable: J$.T(926689, true, 23, false)
                                }, 11, false));
                            } else {
                                J$.P(926737, J$.R(926713, 'obj', obj, false, false), J$.R(926721, 'key', key, false, false), J$.R(926729, 'value', value, false, false));
                            }
                            return J$.Rt(926753, J$.R(926745, 'obj', obj, false, false));
                        } catch (J$e) {
                            J$.Ex(929257, J$e);
                        } finally {
                            if (J$.Fr(929265))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isTransactionCommand(command) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(929017, arguments.callee, this, arguments);
                            arguments = J$.N(929025, 'arguments', arguments, true, false, false);
                            command = J$.N(929033, 'command', command, true, false, false);
                            return J$.Rt(929009, J$.U(99338, '!', J$.U(99330, '!', J$.C(55544, J$.G(928985, J$.R(928977, 'command', command, false, false), 'commitTransaction')) ? J$._() : J$.G(929001, J$.R(928993, 'command', command, false, false), 'abortTransaction'))));
                        } catch (J$e) {
                            J$.Ex(929433, J$e);
                        } finally {
                            if (J$.Fr(929441))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _classCallCheck = J$.N(929105, '_classCallCheck', J$.T(929097, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(929121, '_defineProperties', J$.T(929113, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(929137, '_createClass', J$.T(929129, _createClass, 12, false), true, false, false);
            _defineProperty = J$.N(929153, '_defineProperty', J$.T(929145, _defineProperty, 12, false), true, false, false);
            J$.N(929161, 'MongoError', MongoError, false, false, false);
            J$.N(929169, 'TxnState', TxnState, false, false, false);
            J$.N(929177, 'stateMachine', stateMachine, false, false, false);
            J$.N(929185, 'Transaction', Transaction, false, false, false);
            isTransactionCommand = J$.N(929201, 'isTransactionCommand', J$.T(929193, isTransactionCommand, 12, false), true, false, false);
            J$.T(926113, 'use strict', 21, false);
            var MongoError = J$.W(926833, 'MongoError', J$.G(926825, J$.F(926817, J$.I(typeof require === 'undefined' ? require = J$.R(926801, 'require', undefined, true, true) : require = J$.R(926801, 'require', require, true, true)), false)(J$.T(926809, './error', 21, false)), 'MongoError'), MongoError, false, true);
            var TxnState;
            var stateMachine;
            J$.F(927489, J$.T(927481, function () {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(927409, arguments.callee, this, arguments);
                            arguments = J$.N(927417, 'arguments', arguments, true, false, false);
                            J$.N(927425, '_stateMachine', _stateMachine, false, false, false);
                            J$.N(927433, 'NO_TRANSACTION', NO_TRANSACTION, false, false, false);
                            J$.N(927441, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false, false);
                            J$.N(927449, 'TRANSACTION_IN_PROGRESS', TRANSACTION_IN_PROGRESS, false, false, false);
                            J$.N(927457, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false, false);
                            J$.N(927465, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false, false);
                            J$.N(927473, 'TRANSACTION_ABORTED', TRANSACTION_ABORTED, false, false, false);
                            var _stateMachine;
                            var NO_TRANSACTION = J$.W(926849, 'NO_TRANSACTION', J$.T(926841, 'NO_TRANSACTION', 21, false), NO_TRANSACTION, false, false);
                            var STARTING_TRANSACTION = J$.W(926865, 'STARTING_TRANSACTION', J$.T(926857, 'STARTING_TRANSACTION', 21, false), STARTING_TRANSACTION, false, false);
                            var TRANSACTION_IN_PROGRESS = J$.W(926881, 'TRANSACTION_IN_PROGRESS', J$.T(926873, 'TRANSACTION_IN_PROGRESS', 21, false), TRANSACTION_IN_PROGRESS, false, false);
                            var TRANSACTION_COMMITTED = J$.W(926897, 'TRANSACTION_COMMITTED', J$.T(926889, 'TRANSACTION_COMMITTED', 21, false), TRANSACTION_COMMITTED, false, false);
                            var TRANSACTION_COMMITTED_EMPTY = J$.W(926913, 'TRANSACTION_COMMITTED_EMPTY', J$.T(926905, 'TRANSACTION_COMMITTED_EMPTY', 21, false), TRANSACTION_COMMITTED_EMPTY, false, false);
                            var TRANSACTION_ABORTED = J$.W(926929, 'TRANSACTION_ABORTED', J$.T(926921, 'TRANSACTION_ABORTED', 21, false), TRANSACTION_ABORTED, false, false);
                            TxnState = J$.W(926993, 'TxnState', J$.T(926985, {
                                NO_TRANSACTION: J$.R(926937, 'NO_TRANSACTION', NO_TRANSACTION, false, false),
                                STARTING_TRANSACTION: J$.R(926945, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false),
                                TRANSACTION_IN_PROGRESS: J$.R(926953, 'TRANSACTION_IN_PROGRESS', TRANSACTION_IN_PROGRESS, false, false),
                                TRANSACTION_COMMITTED: J$.R(926961, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false),
                                TRANSACTION_COMMITTED_EMPTY: J$.R(926969, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false),
                                TRANSACTION_ABORTED: J$.R(926977, 'TRANSACTION_ABORTED', TRANSACTION_ABORTED, false, false)
                            }, 11, false), TxnState, false, true);
                            stateMachine = J$.W(927401, 'stateMachine', (_stateMachine = J$.W(927009, '_stateMachine', J$.T(927001, {}, 11, false), _stateMachine, false, false), J$.F(927065, J$.R(927017, '_defineProperty', _defineProperty, false, true), false)(J$.R(927025, '_stateMachine', _stateMachine, false, false), J$.R(927033, 'NO_TRANSACTION', NO_TRANSACTION, false, false), J$.T(927057, [
                                J$.R(927041, 'NO_TRANSACTION', NO_TRANSACTION, false, false),
                                J$.R(927049, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false)
                            ], 10, false)), J$.F(927137, J$.R(927073, '_defineProperty', _defineProperty, false, true), false)(J$.R(927081, '_stateMachine', _stateMachine, false, false), J$.R(927089, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false), J$.T(927129, [
                                J$.R(927097, 'TRANSACTION_IN_PROGRESS', TRANSACTION_IN_PROGRESS, false, false),
                                J$.R(927105, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false),
                                J$.R(927113, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false),
                                J$.R(927121, 'TRANSACTION_ABORTED', TRANSACTION_ABORTED, false, false)
                            ], 10, false)), J$.F(927201, J$.R(927145, '_defineProperty', _defineProperty, false, true), false)(J$.R(927153, '_stateMachine', _stateMachine, false, false), J$.R(927161, 'TRANSACTION_IN_PROGRESS', TRANSACTION_IN_PROGRESS, false, false), J$.T(927193, [
                                J$.R(927169, 'TRANSACTION_IN_PROGRESS', TRANSACTION_IN_PROGRESS, false, false),
                                J$.R(927177, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false),
                                J$.R(927185, 'TRANSACTION_ABORTED', TRANSACTION_ABORTED, false, false)
                            ], 10, false)), J$.F(927273, J$.R(927209, '_defineProperty', _defineProperty, false, true), false)(J$.R(927217, '_stateMachine', _stateMachine, false, false), J$.R(927225, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false), J$.T(927265, [
                                J$.R(927233, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false),
                                J$.R(927241, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false),
                                J$.R(927249, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false),
                                J$.R(927257, 'NO_TRANSACTION', NO_TRANSACTION, false, false)
                            ], 10, false)), J$.F(927329, J$.R(927281, '_defineProperty', _defineProperty, false, true), false)(J$.R(927289, '_stateMachine', _stateMachine, false, false), J$.R(927297, 'TRANSACTION_ABORTED', TRANSACTION_ABORTED, false, false), J$.T(927321, [
                                J$.R(927305, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false),
                                J$.R(927313, 'NO_TRANSACTION', NO_TRANSACTION, false, false)
                            ], 10, false)), J$.F(927385, J$.R(927337, '_defineProperty', _defineProperty, false, true), false)(J$.R(927345, '_stateMachine', _stateMachine, false, false), J$.R(927353, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false), J$.T(927377, [
                                J$.R(927361, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false),
                                J$.R(927369, 'NO_TRANSACTION', NO_TRANSACTION, false, false)
                            ], 10, false)), J$.R(927393, '_stateMachine', _stateMachine, false, false)), stateMachine, false, true);
                        } catch (J$e) {
                            J$.Ex(929273, J$e);
                        } finally {
                            if (J$.Fr(929281))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)();
            var Transaction = J$.W(928969, 'Transaction', J$.F(928961, J$.T(928953, function () {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(928921, arguments.callee, this, arguments);
                            function Transaction(options) {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(928049, arguments.callee, this, arguments);
                                            arguments = J$.N(928057, 'arguments', arguments, true, false, false);
                                            options = J$.N(928065, 'options', options, true, false, false);
                                            J$.N(928073, 'w', w, false, false, false);
                                            J$.F(927521, J$.R(927497, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(927505, 'this', this, false, false), J$.R(927513, 'Transaction', Transaction, false, false));
                                            options = J$.W(927545, 'options', J$.C(55432, J$.R(927529, 'options', options, false, false)) ? J$._() : J$.T(927537, {}, 11, false), options, false, false);
                                            J$.P(927577, J$.R(927553, 'this', this, false, false), 'state', J$.G(927569, J$.R(927561, 'TxnState', TxnState, false, true), 'NO_TRANSACTION'));
                                            J$.P(927601, J$.R(927585, 'this', this, false, false), 'options', J$.T(927593, {}, 11, false));
                                            if (J$.C(55472, J$.C(55440, J$.G(927617, J$.R(927609, 'options', options, false, false), 'writeConcern')) ? J$._() : J$.B(99250, '!==', J$.U(99242, 'typeof', J$.G(927633, J$.R(927625, 'options', options, false, false), 'w')), J$.T(927641, 'undefined', 21, false)))) {
                                                var w = J$.W(927705, 'w', J$.C(55448, J$.G(927657, J$.R(927649, 'options', options, false, false), 'writeConcern')) ? J$.G(927681, J$.G(927673, J$.R(927665, 'options', options, false, false), 'writeConcern'), 'w') : J$.G(927697, J$.R(927689, 'options', options, false, false), 'w'), w, false, false);
                                                if (J$.C(55456, J$.B(99258, '<=', J$.R(927713, 'w', w, false, false), J$.T(927721, 0, 22, false)))) {
                                                    throw J$.F(927745, J$.R(927729, 'MongoError', MongoError, false, true), true)(J$.T(927737, 'Transactions do not support unacknowledged write concern', 21, false));
                                                }
                                                J$.P(927825, J$.G(927761, J$.R(927753, 'this', this, false, false), 'options'), 'writeConcern', J$.C(55464, J$.G(927777, J$.R(927769, 'options', options, false, false), 'writeConcern')) ? J$.G(927793, J$.R(927785, 'options', options, false, false), 'writeConcern') : J$.T(927817, { w: J$.G(927809, J$.R(927801, 'options', options, false, false), 'w') }, 11, false));
                                            }
                                            if (J$.C(55480, J$.G(927841, J$.R(927833, 'options', options, false, false), 'readConcern')))
                                                J$.P(927881, J$.G(927857, J$.R(927849, 'this', this, false, false), 'options'), 'readConcern', J$.G(927873, J$.R(927865, 'options', options, false, false), 'readConcern'));
                                            if (J$.C(55488, J$.G(927897, J$.R(927889, 'options', options, false, false), 'readPreference')))
                                                J$.P(927937, J$.G(927913, J$.R(927905, 'this', this, false, false), 'options'), 'readPreference', J$.G(927929, J$.R(927921, 'options', options, false, false), 'readPreference'));
                                            if (J$.C(55496, J$.G(927953, J$.R(927945, 'options', options, false, false), 'maxCommitTimeMS')))
                                                J$.P(927993, J$.G(927969, J$.R(927961, 'this', this, false, false), 'options'), 'maxTimeMS', J$.G(927985, J$.R(927977, 'options', options, false, false), 'maxCommitTimeMS'));
                                            J$.P(928017, J$.R(928001, 'this', this, false, false), '_pinnedServer', J$.T(928009, undefined, 24, false));
                                            J$.P(928041, J$.R(928025, 'this', this, false, false), '_recoveryToken', J$.T(928033, undefined, 24, false));
                                        } catch (J$e) {
                                            J$.Ex(929289, J$e);
                                        } finally {
                                            if (J$.Fr(929297))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(928929, 'arguments', arguments, true, false, false);
                            Transaction = J$.N(928945, 'Transaction', J$.T(928937, Transaction, 12, false), true, false, false);
                            J$.F(928897, J$.R(928081, '_createClass', _createClass, false, true), false)(J$.R(928089, 'Transaction', Transaction, false, false), J$.T(928889, [
                                J$.T(928417, {
                                    key: J$.T(928097, 'transition', 21, false),
                                    value: J$.T(928409, function transition(nextState) {
                                        jalangiLabel6:
                                            while (true) {
                                                try {
                                                    J$.Fe(928377, arguments.callee, this, arguments);
                                                    arguments = J$.N(928385, 'arguments', arguments, true, false, false);
                                                    nextState = J$.N(928393, 'nextState', nextState, true, false, false);
                                                    J$.N(928401, 'nextStates', nextStates, false, false, false);
                                                    var nextStates = J$.W(928137, 'nextStates', J$.G(928129, J$.R(928105, 'stateMachine', stateMachine, false, true), J$.G(928121, J$.R(928113, 'this', this, false, false), 'state')), nextStates, false, false);
                                                    if (J$.C(55528, J$.C(55504, J$.R(928145, 'nextStates', nextStates, false, false)) ? J$.B(99274, '!==', J$.M(928169, J$.R(928153, 'nextStates', nextStates, false, false), 'indexOf', false)(J$.R(928161, 'nextState', nextState, false, false)), J$.U(99266, '-', J$.T(928177, 1, 22, false))) : J$._())) {
                                                        J$.P(928201, J$.R(928185, 'this', this, false, false), 'state', J$.R(928193, 'nextState', nextState, false, false));
                                                        if (J$.C(55520, J$.C(55512, J$.B(99282, '===', J$.G(928217, J$.R(928209, 'this', this, false, false), 'state'), J$.G(928233, J$.R(928225, 'TxnState', TxnState, false, true), 'NO_TRANSACTION'))) ? J$._() : J$.B(99290, '===', J$.G(928249, J$.R(928241, 'this', this, false, false), 'state'), J$.G(928265, J$.R(928257, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION')))) {
                                                            J$.M(928281, J$.R(928273, 'this', this, false, false), 'unpinServer', false)();
                                                        }
                                                        return J$.Rt(928289, undefined);
                                                    }
                                                    throw J$.F(928369, J$.R(928297, 'MongoError', MongoError, false, true), true)(J$.M(928361, J$.M(928337, J$.T(928305, 'Attempted illegal state transition from [', 21, false), 'concat', false)(J$.G(928321, J$.R(928313, 'this', this, false, false), 'state'), J$.T(928329, '] to [', 21, false)), 'concat', false)(J$.R(928345, 'nextState', nextState, false, false), J$.T(928353, ']', 21, false)));
                                                } catch (J$e) {
                                                    J$.Ex(929305, J$e);
                                                } finally {
                                                    if (J$.Fr(929313))
                                                        continue jalangiLabel6;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(928505, {
                                    key: J$.T(928425, 'pinServer', 21, false),
                                    value: J$.T(928497, function pinServer(server) {
                                        jalangiLabel7:
                                            while (true) {
                                                try {
                                                    J$.Fe(928473, arguments.callee, this, arguments);
                                                    arguments = J$.N(928481, 'arguments', arguments, true, false, false);
                                                    server = J$.N(928489, 'server', server, true, false, false);
                                                    if (J$.C(55536, J$.G(928441, J$.R(928433, 'this', this, false, false), 'isActive'))) {
                                                        J$.P(928465, J$.R(928449, 'this', this, false, false), '_pinnedServer', J$.R(928457, 'server', server, false, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(929321, J$e);
                                                } finally {
                                                    if (J$.Fr(929329))
                                                        continue jalangiLabel7;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(928569, {
                                    key: J$.T(928513, 'unpinServer', 21, false),
                                    value: J$.T(928561, function unpinServer() {
                                        jalangiLabel8:
                                            while (true) {
                                                try {
                                                    J$.Fe(928545, arguments.callee, this, arguments);
                                                    arguments = J$.N(928553, 'arguments', arguments, true, false, false);
                                                    J$.P(928537, J$.R(928521, 'this', this, false, false), '_pinnedServer', J$.T(928529, undefined, 24, false));
                                                } catch (J$e) {
                                                    J$.Ex(929337, J$e);
                                                } finally {
                                                    if (J$.Fr(929345))
                                                        continue jalangiLabel8;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(928633, {
                                    key: J$.T(928577, 'server', 21, false),
                                    get: J$.T(928625, function get() {
                                        jalangiLabel9:
                                            while (true) {
                                                try {
                                                    J$.Fe(928609, arguments.callee, this, arguments);
                                                    arguments = J$.N(928617, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(928601, J$.G(928593, J$.R(928585, 'this', this, false, false), '_pinnedServer'));
                                                } catch (J$e) {
                                                    J$.Ex(929353, J$e);
                                                } finally {
                                                    if (J$.Fr(929361))
                                                        continue jalangiLabel9;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(928697, {
                                    key: J$.T(928641, 'recoveryToken', 21, false),
                                    get: J$.T(928689, function get() {
                                        jalangiLabel10:
                                            while (true) {
                                                try {
                                                    J$.Fe(928673, arguments.callee, this, arguments);
                                                    arguments = J$.N(928681, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(928665, J$.G(928657, J$.R(928649, 'this', this, false, false), '_recoveryToken'));
                                                } catch (J$e) {
                                                    J$.Ex(929369, J$e);
                                                } finally {
                                                    if (J$.Fr(929377))
                                                        continue jalangiLabel10;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(928761, {
                                    key: J$.T(928705, 'isPinned', 21, false),
                                    get: J$.T(928753, function get() {
                                        jalangiLabel11:
                                            while (true) {
                                                try {
                                                    J$.Fe(928737, arguments.callee, this, arguments);
                                                    arguments = J$.N(928745, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(928729, J$.U(99306, '!', J$.U(99298, '!', J$.G(928721, J$.R(928713, 'this', this, false, false), 'server'))));
                                                } catch (J$e) {
                                                    J$.Ex(929385, J$e);
                                                } finally {
                                                    if (J$.Fr(929393))
                                                        continue jalangiLabel11;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(928881, {
                                    key: J$.T(928769, 'isActive', 21, false),
                                    get: J$.T(928873, function get() {
                                        jalangiLabel12:
                                            while (true) {
                                                try {
                                                    J$.Fe(928857, arguments.callee, this, arguments);
                                                    arguments = J$.N(928865, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(928849, J$.B(99322, '!==', J$.M(928833, J$.T(928809, [
                                                        J$.G(928785, J$.R(928777, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION'),
                                                        J$.G(928801, J$.R(928793, 'TxnState', TxnState, false, true), 'TRANSACTION_IN_PROGRESS')
                                                    ], 10, false), 'indexOf', false)(J$.G(928825, J$.R(928817, 'this', this, false, false), 'state')), J$.U(99314, '-', J$.T(928841, 1, 22, false))));
                                                } catch (J$e) {
                                                    J$.Ex(929401, J$e);
                                                } finally {
                                                    if (J$.Fr(929409))
                                                        continue jalangiLabel12;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(928913, J$.R(928905, 'Transaction', Transaction, false, false));
                        } catch (J$e) {
                            J$.Ex(929417, J$e);
                        } finally {
                            if (J$.Fr(929425))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), Transaction, false, true);
            J$.P(929081, J$.I(typeof module === 'undefined' ? module = J$.R(929041, 'module', undefined, true, true) : module = J$.R(929041, 'module', module, true, true)), 'exports', J$.T(929073, {
                TxnState: J$.R(929049, 'TxnState', TxnState, false, true),
                Transaction: J$.R(929057, 'Transaction', Transaction, false, true),
                isTransactionCommand: J$.R(929065, 'isTransactionCommand', isTransactionCommand, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(929449, J$e);
        } finally {
            if (J$.Sr(929457))
                continue jalangiLabel15;
            else
                break jalangiLabel15;
        }
    }
// JALANGI DO NOT INSTRUMENT
