// js-yaml/lib/js-yaml/common_jalangi_.js
J$.iids = {"8":[5,10,5,64],"9":[1,1,1,13],"10":[5,11,5,25],"16":[10,10,10,61],"17":[1,1,1,14],"18":[5,11,5,41],"24":[16,12,16,31],"25":[5,18,5,25],"26":[5,47,5,63],"32":[15,7,15,30],"33":[5,30,5,41],"34":[10,11,10,25],"40":[28,49,28,63],"41":[5,47,5,54],"42":[10,11,10,38],"48":[25,7,25,13],"49":[5,59,5,63],"50":[10,44,10,60],"56":[41,19,41,32],"57":[5,10,5,64],"58":[28,49,28,63],"64":[50,10,50,69],"65":[5,3,5,65],"66":[28,74,28,75],"73":[4,1,6,2],"74":[41,19,41,32],"81":[4,1,6,2],"82":[41,43,41,44],"89":[4,1,6,2],"90":[42,15,42,21],"97":[10,18,10,25],"98":[50,11,50,23],"105":[10,30,10,38],"106":[50,58,50,68],"113":[10,44,10,51],"114":[50,29,50,68],"121":[10,56,10,60],"129":[10,10,10,61],"137":[10,3,10,62],"145":[9,1,11,2],"153":[9,1,11,2],"161":[9,1,11,2],"169":[15,7,15,12],"177":[15,21,15,29],"185":[15,7,15,30],"187":[15,7,15,20],"193":[15,39,15,47],"201":[15,39,15,47],"209":[15,32,15,48],"217":[16,12,16,21],"225":[16,22,16,30],"233":[16,12,16,31],"241":[16,40,16,42],"249":[16,40,16,42],"257":[16,33,16,43],"265":[18,12,18,20],"273":[18,10,18,22],"281":[18,10,18,22],"289":[18,3,18,23],"297":[14,1,19,2],"305":[14,1,19,2],"313":[14,1,19,2],"321":[25,7,25,13],"329":[26,18,26,24],"337":[26,30,26,36],"345":[26,18,26,37],"347":[26,18,26,29],"353":[26,18,26,37],"361":[26,5,26,38],"369":[28,18,28,19],"377":[28,18,28,19],"385":[28,30,28,40],"393":[28,30,28,47],"401":[28,30,28,47],"409":[28,10,28,19],"417":[28,49,28,54],"425":[28,57,28,63],"433":[28,74,28,75],"441":[28,65,28,70],"449":[28,65,28,75],"457":[29,13,29,23],"465":[29,24,29,29],"473":[29,13,29,30],"481":[29,13,29,30],"489":[29,7,29,31],"497":[30,7,30,13],"505":[30,14,30,17],"513":[30,21,30,27],"521":[30,28,30,31],"529":[30,21,30,32],"537":[30,7,30,32],"545":[30,7,30,33],"553":[34,10,34,16],"561":[34,10,34,16],"569":[34,3,34,17],"577":[22,1,35,2],"585":[22,1,35,2],"593":[22,1,35,2],"601":[22,1,35,2],"609":[22,1,35,2],"617":[22,1,35,2],"625":[22,1,35,2],"633":[22,1,35,2],"641":[39,16,39,18],"649":[39,16,39,18],"657":[39,16,39,18],"665":[41,16,41,17],"673":[41,16,41,17],"681":[41,19,41,24],"689":[41,27,41,32],"697":[41,43,41,44],"705":[41,34,41,39],"713":[41,34,41,44],"721":[42,15,42,21],"729":[42,5,42,11],"737":[42,5,42,21],"745":[42,5,42,22],"753":[45,10,45,16],"761":[45,10,45,16],"769":[45,3,45,17],"777":[38,1,46,2],"785":[38,1,46,2],"793":[38,1,46,2],"801":[38,1,46,2],"809":[38,1,46,2],"817":[38,1,46,2],"825":[50,11,50,17],"833":[50,22,50,23],"841":[50,29,50,35],"849":[50,29,50,53],"857":[50,58,50,59],"865":[50,62,50,68],"873":[50,10,50,69],"881":[50,3,50,70],"889":[49,1,51,2],"897":[49,1,51,2],"905":[49,1,51,2],"913":[54,1,54,7],"921":[54,1,54,15],"929":[54,33,54,42],"937":[54,1,54,42],"945":[54,1,54,43],"953":[55,1,55,7],"961":[55,1,55,15],"969":[55,33,55,41],"977":[55,1,55,41],"985":[55,1,55,42],"993":[56,1,56,7],"1001":[56,1,56,15],"1009":[56,33,56,40],"1017":[56,1,56,40],"1025":[56,1,56,41],"1033":[57,1,57,7],"1041":[57,1,57,15],"1049":[57,33,57,39],"1057":[57,1,57,39],"1065":[57,1,57,40],"1073":[58,1,58,7],"1081":[58,1,58,15],"1089":[58,33,58,47],"1097":[58,1,58,47],"1105":[58,1,58,48],"1113":[59,1,59,7],"1121":[59,1,59,15],"1129":[59,33,59,39],"1137":[59,1,59,39],"1145":[59,1,59,40],"1153":[1,1,60,1],"1161":[4,1,6,2],"1169":[1,1,60,1],"1177":[9,1,11,2],"1185":[1,1,60,1],"1193":[14,1,19,2],"1201":[1,1,60,1],"1209":[22,1,35,2],"1217":[1,1,60,1],"1225":[38,1,46,2],"1233":[1,1,60,1],"1241":[49,1,51,2],"1249":[1,1,60,1],"1257":[4,1,6,2],"1265":[4,1,6,2],"1273":[9,1,11,2],"1281":[9,1,11,2],"1289":[16,8,16,43],"1297":[15,3,16,43],"1305":[14,1,19,2],"1313":[14,1,19,2],"1321":[28,5,31,6],"1329":[28,5,31,6],"1337":[28,5,31,6],"1345":[25,3,32,4],"1353":[22,1,35,2],"1361":[22,1,35,2],"1369":[41,3,43,4],"1377":[41,3,43,4],"1385":[41,3,43,4],"1393":[38,1,46,2],"1401":[38,1,46,2],"1409":[49,1,51,2],"1417":[49,1,51,2],"1425":[1,1,60,1],"1433":[1,1,60,1],"nBranches":16,"originalCodeFileName":"/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/common.js","instrumentedCodeFileName":"/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/common_jalangi_.js","code":"'use strict';\n\n\nfunction isNothing(subject) {\n  return (typeof subject === 'undefined') || (subject === null);\n}\n\n\nfunction isObject(subject) {\n  return (typeof subject === 'object') && (subject !== null);\n}\n\n\nfunction toArray(sequence) {\n  if (Array.isArray(sequence)) return sequence;\n  else if (isNothing(sequence)) return [];\n\n  return [ sequence ];\n}\n\n\nfunction extend(target, source) {\n  var index, length, key, sourceKeys;\n\n  if (source) {\n    sourceKeys = Object.keys(source);\n\n    for (index = 0, length = sourceKeys.length; index < length; index += 1) {\n      key = sourceKeys[index];\n      target[key] = source[key];\n    }\n  }\n\n  return target;\n}\n\n\nfunction repeat(string, count) {\n  var result = '', cycle;\n\n  for (cycle = 0; cycle < count; cycle += 1) {\n    result += string;\n  }\n\n  return result;\n}\n\n\nfunction isNegativeZero(number) {\n  return (number === 0) && (Number.NEGATIVE_INFINITY === 1 / number);\n}\n\n\nmodule.exports.isNothing      = isNothing;\nmodule.exports.isObject       = isObject;\nmodule.exports.toArray        = toArray;\nmodule.exports.repeat         = repeat;\nmodule.exports.isNegativeZero = isNegativeZero;\nmodule.exports.extend         = extend;\n"};
jalangiLabel53:
    while (true) {
        try {
            J$.Se(1153, '/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/common_jalangi_.js', '/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/common.js');
            function isNothing(subject) {
                jalangiLabel47:
                    while (true) {
                        try {
                            J$.Fe(73, arguments.callee, this, arguments);
                            arguments = J$.N(81, 'arguments', arguments, 4);
                            subject = J$.N(89, 'subject', subject, 4);
                            return J$.X1(65, J$.Rt(57, J$.C(8, J$.B(18, '===', J$.U(10, 'typeof', J$.R(25, 'subject', subject, 0)), J$.T(33, 'undefined', 21, false), 0)) ? J$._() : J$.B(26, '===', J$.R(41, 'subject', subject, 0), J$.T(49, null, 25, false), 0)));
                        } catch (J$e) {
                            J$.Ex(1257, J$e);
                        } finally {
                            if (J$.Fr(1265))
                                continue jalangiLabel47;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isObject(subject) {
                jalangiLabel48:
                    while (true) {
                        try {
                            J$.Fe(145, arguments.callee, this, arguments);
                            arguments = J$.N(153, 'arguments', arguments, 4);
                            subject = J$.N(161, 'subject', subject, 4);
                            return J$.X1(137, J$.Rt(129, J$.C(16, J$.B(42, '===', J$.U(34, 'typeof', J$.R(97, 'subject', subject, 0)), J$.T(105, 'object', 21, false), 0)) ? J$.B(50, '!==', J$.R(113, 'subject', subject, 0), J$.T(121, null, 25, false), 0) : J$._()));
                        } catch (J$e) {
                            J$.Ex(1273, J$e);
                        } finally {
                            if (J$.Fr(1281))
                                continue jalangiLabel48;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function toArray(sequence) {
                jalangiLabel49:
                    while (true) {
                        try {
                            J$.Fe(297, arguments.callee, this, arguments);
                            arguments = J$.N(305, 'arguments', arguments, 4);
                            sequence = J$.N(313, 'sequence', sequence, 4);
                            if (J$.X1(1297, J$.C(32, J$.M(185, J$.R(169, 'Array', Array, 2), 'isArray', 0)(J$.R(177, 'sequence', sequence, 0)))))
                                return J$.X1(209, J$.Rt(201, J$.R(193, 'sequence', sequence, 0)));
                            else if (J$.X1(1289, J$.C(24, J$.F(233, J$.R(217, 'isNothing', isNothing, 1), 0)(J$.R(225, 'sequence', sequence, 0)))))
                                return J$.X1(257, J$.Rt(249, J$.T(241, [], 10, false)));
                            return J$.X1(289, J$.Rt(281, J$.T(273, [J$.R(265, 'sequence', sequence, 0)], 10, false)));
                        } catch (J$e) {
                            J$.Ex(1305, J$e);
                        } finally {
                            if (J$.Fr(1313))
                                continue jalangiLabel49;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function extend(target, source) {
                jalangiLabel50:
                    while (true) {
                        try {
                            J$.Fe(577, arguments.callee, this, arguments);
                            arguments = J$.N(585, 'arguments', arguments, 4);
                            target = J$.N(593, 'target', target, 4);
                            source = J$.N(601, 'source', source, 4);
                            J$.N(609, 'index', index, 0);
                            J$.N(617, 'length', length, 0);
                            J$.N(625, 'key', key, 0);
                            J$.N(633, 'sourceKeys', sourceKeys, 0);
                            var index, length, key, sourceKeys;
                            if (J$.X1(1345, J$.C(48, J$.R(321, 'source', source, 0)))) {
                                J$.X1(361, sourceKeys = J$.W(353, 'sourceKeys', J$.M(345, J$.R(329, 'Object', Object, 2), 'keys', 0)(J$.R(337, 'source', source, 0)), sourceKeys, 0));
                                for (J$.X1(1329, (J$.X1(409, index = J$.W(377, 'index', J$.T(369, 0, 22, false), index, 0)), length = J$.W(401, 'length', J$.G(393, J$.R(385, 'sourceKeys', sourceKeys, 0), 'length', 0), length, 0))); J$.X1(1321, J$.C(40, J$.B(58, '<', J$.R(417, 'index', index, 0), J$.R(425, 'length', length, 0), 0))); J$.X1(1337, index = J$.W(449, 'index', J$.B(66, '+', J$.R(441, 'index', index, 0), J$.T(433, 1, 22, false), 0), index, 0))) {
                                    J$.X1(489, key = J$.W(481, 'key', J$.G(473, J$.R(457, 'sourceKeys', sourceKeys, 0), J$.R(465, 'index', index, 0), 4), key, 0));
                                    J$.X1(545, J$.P(537, J$.R(497, 'target', target, 0), J$.R(505, 'key', key, 0), J$.G(529, J$.R(513, 'source', source, 0), J$.R(521, 'key', key, 0), 4), 2));
                                }
                            }
                            return J$.X1(569, J$.Rt(561, J$.R(553, 'target', target, 0)));
                        } catch (J$e) {
                            J$.Ex(1353, J$e);
                        } finally {
                            if (J$.Fr(1361))
                                continue jalangiLabel50;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function repeat(string, count) {
                jalangiLabel51:
                    while (true) {
                        try {
                            J$.Fe(777, arguments.callee, this, arguments);
                            arguments = J$.N(785, 'arguments', arguments, 4);
                            string = J$.N(793, 'string', string, 4);
                            count = J$.N(801, 'count', count, 4);
                            J$.N(809, 'result', result, 0);
                            J$.N(817, 'cycle', cycle, 0);
                            var result = J$.X1(657, J$.W(649, 'result', J$.T(641, '', 21, false), result, 1)), cycle;
                            for (J$.X1(1377, cycle = J$.W(673, 'cycle', J$.T(665, 0, 22, false), cycle, 0)); J$.X1(1369, J$.C(56, J$.B(74, '<', J$.R(681, 'cycle', cycle, 0), J$.R(689, 'count', count, 0), 0))); J$.X1(1385, cycle = J$.W(713, 'cycle', J$.B(82, '+', J$.R(705, 'cycle', cycle, 0), J$.T(697, 1, 22, false), 0), cycle, 0))) {
                                J$.X1(745, result = J$.W(737, 'result', J$.B(90, '+', J$.R(729, 'result', result, 0), J$.R(721, 'string', string, 0), 0), result, 0));
                            }
                            return J$.X1(769, J$.Rt(761, J$.R(753, 'result', result, 0)));
                        } catch (J$e) {
                            J$.Ex(1393, J$e);
                        } finally {
                            if (J$.Fr(1401))
                                continue jalangiLabel51;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isNegativeZero(number) {
                jalangiLabel52:
                    while (true) {
                        try {
                            J$.Fe(889, arguments.callee, this, arguments);
                            arguments = J$.N(897, 'arguments', arguments, 4);
                            number = J$.N(905, 'number', number, 4);
                            return J$.X1(881, J$.Rt(873, J$.C(64, J$.B(98, '===', J$.R(825, 'number', number, 0), J$.T(833, 0, 22, false), 0)) ? J$.B(114, '===', J$.G(849, J$.R(841, 'Number', Number, 2), 'NEGATIVE_INFINITY', 0), J$.B(106, '/', J$.T(857, 1, 22, false), J$.R(865, 'number', number, 0), 0), 0) : J$._()));
                        } catch (J$e) {
                            J$.Ex(1409, J$e);
                        } finally {
                            if (J$.Fr(1417))
                                continue jalangiLabel52;
                            else
                                return J$.Ra();
                        }
                    }
            }
            isNothing = J$.N(1169, 'isNothing', J$.T(1161, isNothing, 12, false, 73), 0);
            isObject = J$.N(1185, 'isObject', J$.T(1177, isObject, 12, false, 145), 0);
            toArray = J$.N(1201, 'toArray', J$.T(1193, toArray, 12, false, 297), 0);
            extend = J$.N(1217, 'extend', J$.T(1209, extend, 12, false, 577), 0);
            repeat = J$.N(1233, 'repeat', J$.T(1225, repeat, 12, false, 777), 0);
            isNegativeZero = J$.N(1249, 'isNegativeZero', J$.T(1241, isNegativeZero, 12, false, 889), 0);
            J$.X1(17, J$.T(9, 'use strict', 21, false));
            J$.X1(945, J$.P(937, J$.G(921, J$.R(913, 'module', module, 2), 'exports', 0), 'isNothing', J$.R(929, 'isNothing', isNothing, 1), 0));
            J$.X1(985, J$.P(977, J$.G(961, J$.R(953, 'module', module, 2), 'exports', 0), 'isObject', J$.R(969, 'isObject', isObject, 1), 0));
            J$.X1(1025, J$.P(1017, J$.G(1001, J$.R(993, 'module', module, 2), 'exports', 0), 'toArray', J$.R(1009, 'toArray', toArray, 1), 0));
            J$.X1(1065, J$.P(1057, J$.G(1041, J$.R(1033, 'module', module, 2), 'exports', 0), 'repeat', J$.R(1049, 'repeat', repeat, 1), 0));
            J$.X1(1105, J$.P(1097, J$.G(1081, J$.R(1073, 'module', module, 2), 'exports', 0), 'isNegativeZero', J$.R(1089, 'isNegativeZero', isNegativeZero, 1), 0));
            J$.X1(1145, J$.P(1137, J$.G(1121, J$.R(1113, 'module', module, 2), 'exports', 0), 'extend', J$.R(1129, 'extend', extend, 1), 0));
        } catch (J$e) {
            J$.Ex(1425, J$e);
        } finally {
            if (J$.Sr(1433)) {
                J$.L();
                continue jalangiLabel53;
            } else {
                J$.L();
                break jalangiLabel53;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
