// mongodb/lib/core/connection/apm.js
J$.noInstrEval = false;
jalangiLabel15:
    while (true) {
        try {
            J$.Se(134681, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/core/connection/apm.js');
            function _classCallCheck(instance, Constructor) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(131089, arguments.callee, this, arguments);
                            arguments = J$.N(131097, 'arguments', arguments, true, false, false);
                            instance = J$.N(131105, 'instance', instance, true, false, false);
                            Constructor = J$.N(131113, 'Constructor', Constructor, true, false, false);
                            if (J$.C(7456, J$.U(6978, '!', J$.B(6970, 'instanceof', J$.R(131049, 'instance', instance, false, false), J$.R(131057, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(131081, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(131065, 'TypeError', undefined, true, true) : TypeError = J$.R(131065, 'TypeError', TypeError, true, true)), true)(J$.T(131073, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(134857, J$e);
                        } finally {
                            if (J$.Fr(134865))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _classCallCheck = J$.N(134697, '_classCallCheck', J$.T(134689, _classCallCheck, 12, false), true, false, false);
            J$.N(134705, 'Msg', Msg, false, false, false);
            J$.N(134713, 'KillCursor', KillCursor, false, false, false);
            J$.N(134721, 'GetMore', GetMore, false, false, false);
            J$.N(134729, 'calculateDurationInMs', calculateDurationInMs, false, false, false);
            J$.N(134737, 'SENSITIVE_COMMANDS', SENSITIVE_COMMANDS, false, false, false);
            J$.N(134745, 'extractCommandName', extractCommandName, false, false, false);
            J$.N(134753, 'namespace', namespace, false, false, false);
            J$.N(134761, 'databaseName', databaseName, false, false, false);
            J$.N(134769, 'collectionName', collectionName, false, false, false);
            J$.N(134777, 'generateConnectionId', generateConnectionId, false, false, false);
            J$.N(134785, 'maybeRedact', maybeRedact, false, false, false);
            J$.N(134793, 'LEGACY_FIND_QUERY_MAP', LEGACY_FIND_QUERY_MAP, false, false, false);
            J$.N(134801, 'LEGACY_FIND_OPTIONS_MAP', LEGACY_FIND_OPTIONS_MAP, false, false, false);
            J$.N(134809, 'OP_QUERY_KEYS', OP_QUERY_KEYS, false, false, false);
            J$.N(134817, 'extractCommand', extractCommand, false, false, false);
            J$.N(134825, 'extractReply', extractReply, false, false, false);
            J$.N(134833, 'CommandStartedEvent', CommandStartedEvent, false, false, false);
            J$.N(134841, 'CommandSucceededEvent', CommandSucceededEvent, false, false, false);
            J$.N(134849, 'CommandFailedEvent', CommandFailedEvent, false, false, false);
            J$.T(131041, 'use strict', 21, false);
            var Msg = J$.W(131153, 'Msg', J$.G(131145, J$.F(131137, J$.I(typeof require === 'undefined' ? require = J$.R(131121, 'require', undefined, true, true) : require = J$.R(131121, 'require', require, true, true)), false)(J$.T(131129, '../connection/msg', 21, false)), 'Msg'), Msg, false, true);
            var KillCursor = J$.W(131193, 'KillCursor', J$.G(131185, J$.F(131177, J$.I(typeof require === 'undefined' ? require = J$.R(131161, 'require', undefined, true, true) : require = J$.R(131161, 'require', require, true, true)), false)(J$.T(131169, '../connection/commands', 21, false)), 'KillCursor'), KillCursor, false, true);
            var GetMore = J$.W(131233, 'GetMore', J$.G(131225, J$.F(131217, J$.I(typeof require === 'undefined' ? require = J$.R(131201, 'require', undefined, true, true) : require = J$.R(131201, 'require', require, true, true)), false)(J$.T(131209, '../connection/commands', 21, false)), 'GetMore'), GetMore, false, true);
            var calculateDurationInMs = J$.W(131273, 'calculateDurationInMs', J$.G(131265, J$.F(131257, J$.I(typeof require === 'undefined' ? require = J$.R(131241, 'require', undefined, true, true) : require = J$.R(131241, 'require', require, true, true)), false)(J$.T(131249, '../utils', 21, false)), 'calculateDurationInMs'), calculateDurationInMs, false, true);
            var SENSITIVE_COMMANDS = J$.W(131377, 'SENSITIVE_COMMANDS', J$.F(131369, J$.I(typeof Set === 'undefined' ? Set = J$.R(131281, 'Set', undefined, true, true) : Set = J$.R(131281, 'Set', Set, true, true)), true)(J$.T(131361, [
                J$.T(131289, 'authenticate', 21, false),
                J$.T(131297, 'saslStart', 21, false),
                J$.T(131305, 'saslContinue', 21, false),
                J$.T(131313, 'getnonce', 21, false),
                J$.T(131321, 'createUser', 21, false),
                J$.T(131329, 'updateUser', 21, false),
                J$.T(131337, 'copydbgetnonce', 21, false),
                J$.T(131345, 'copydbsaslstart', 21, false),
                J$.T(131353, 'copydb', 21, false)
            ], 10, false)), SENSITIVE_COMMANDS, false, true);
            var extractCommandName = J$.W(131465, 'extractCommandName', J$.T(131457, function extractCommandName(commandDoc) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(131433, arguments.callee, this, arguments);
                            arguments = J$.N(131441, 'arguments', arguments, true, false, false);
                            commandDoc = J$.N(131449, 'commandDoc', commandDoc, true, false, false);
                            return J$.Rt(131425, J$.G(131417, J$.M(131401, J$.I(typeof Object === 'undefined' ? Object = J$.R(131385, 'Object', undefined, true, true) : Object = J$.R(131385, 'Object', Object, true, true)), 'keys', false)(J$.R(131393, 'commandDoc', commandDoc, false, false)), J$.T(131409, 0, 22, false)));
                        } catch (J$e) {
                            J$.Ex(134873, J$e);
                        } finally {
                            if (J$.Fr(134881))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), extractCommandName, false, true);
            var namespace = J$.W(131529, 'namespace', J$.T(131521, function namespace(command) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(131497, arguments.callee, this, arguments);
                            arguments = J$.N(131505, 'arguments', arguments, true, false, false);
                            command = J$.N(131513, 'command', command, true, false, false);
                            return J$.Rt(131489, J$.G(131481, J$.R(131473, 'command', command, false, false), 'ns'));
                        } catch (J$e) {
                            J$.Ex(134889, J$e);
                        } finally {
                            if (J$.Fr(134897))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), namespace, false, true);
            var databaseName = J$.W(131625, 'databaseName', J$.T(131617, function databaseName(command) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(131593, arguments.callee, this, arguments);
                            arguments = J$.N(131601, 'arguments', arguments, true, false, false);
                            command = J$.N(131609, 'command', command, true, false, false);
                            return J$.Rt(131585, J$.G(131577, J$.M(131561, J$.G(131545, J$.R(131537, 'command', command, false, false), 'ns'), 'split', false)(J$.T(131553, '.', 21, false)), J$.T(131569, 0, 22, false)));
                        } catch (J$e) {
                            J$.Ex(134905, J$e);
                        } finally {
                            if (J$.Fr(134913))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), databaseName, false, true);
            var collectionName = J$.W(131721, 'collectionName', J$.T(131713, function collectionName(command) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(131689, arguments.callee, this, arguments);
                            arguments = J$.N(131697, 'arguments', arguments, true, false, false);
                            command = J$.N(131705, 'command', command, true, false, false);
                            return J$.Rt(131681, J$.G(131673, J$.M(131657, J$.G(131641, J$.R(131633, 'command', command, false, false), 'ns'), 'split', false)(J$.T(131649, '.', 21, false)), J$.T(131665, 1, 22, false)));
                        } catch (J$e) {
                            J$.Ex(134921, J$e);
                        } finally {
                            if (J$.Fr(134929))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), collectionName, false, true);
            var generateConnectionId = J$.W(131849, 'generateConnectionId', J$.T(131841, function generateConnectionId(pool) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(131817, arguments.callee, this, arguments);
                            arguments = J$.N(131825, 'arguments', arguments, true, false, false);
                            pool = J$.N(131833, 'pool', pool, true, false, false);
                            return J$.Rt(131809, J$.M(131801, J$.M(131769, J$.T(131729, '', 21, false), 'concat', false)(J$.G(131753, J$.G(131745, J$.R(131737, 'pool', pool, false, false), 'options'), 'host'), J$.T(131761, ':', 21, false)), 'concat', false)(J$.G(131793, J$.G(131785, J$.R(131777, 'pool', pool, false, false), 'options'), 'port')));
                        } catch (J$e) {
                            J$.Ex(134937, J$e);
                        } finally {
                            if (J$.Fr(134945))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), generateConnectionId, false, true);
            var maybeRedact = J$.W(131945, 'maybeRedact', J$.T(131937, function maybeRedact(commandName, result) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(131905, arguments.callee, this, arguments);
                            arguments = J$.N(131913, 'arguments', arguments, true, false, false);
                            commandName = J$.N(131921, 'commandName', commandName, true, false, false);
                            result = J$.N(131929, 'result', result, true, false, false);
                            return J$.Rt(131897, J$.C(7464, J$.M(131873, J$.R(131857, 'SENSITIVE_COMMANDS', SENSITIVE_COMMANDS, false, true), 'has', false)(J$.R(131865, 'commandName', commandName, false, false))) ? J$.T(131881, {}, 11, false) : J$.R(131889, 'result', result, false, false));
                        } catch (J$e) {
                            J$.Ex(134953, J$e);
                        } finally {
                            if (J$.Fr(134961))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), maybeRedact, false, true);
            var LEGACY_FIND_QUERY_MAP = J$.W(132049, 'LEGACY_FIND_QUERY_MAP', J$.T(132041, {
                $query: J$.T(131953, 'filter', 21, false),
                $orderby: J$.T(131961, 'sort', 21, false),
                $hint: J$.T(131969, 'hint', 21, false),
                $comment: J$.T(131977, 'comment', 21, false),
                $maxScan: J$.T(131985, 'maxScan', 21, false),
                $max: J$.T(131993, 'max', 21, false),
                $min: J$.T(132001, 'min', 21, false),
                $returnKey: J$.T(132009, 'returnKey', 21, false),
                $showDiskLoc: J$.T(132017, 'showRecordId', 21, false),
                $maxTimeMS: J$.T(132025, 'maxTimeMS', 21, false),
                $snapshot: J$.T(132033, 'snapshot', 21, false)
            }, 11, false), LEGACY_FIND_QUERY_MAP, false, true);
            var LEGACY_FIND_OPTIONS_MAP = J$.W(132089, 'LEGACY_FIND_OPTIONS_MAP', J$.T(132081, {
                numberToSkip: J$.T(132057, 'skip', 21, false),
                numberToReturn: J$.T(132065, 'batchSize', 21, false),
                returnFieldsSelector: J$.T(132073, 'projection', 21, false)
            }, 11, false), LEGACY_FIND_OPTIONS_MAP, false, true);
            var OP_QUERY_KEYS = J$.W(132153, 'OP_QUERY_KEYS', J$.T(132145, [
                J$.T(132097, 'tailable', 21, false),
                J$.T(132105, 'oplogReplay', 21, false),
                J$.T(132113, 'noCursorTimeout', 21, false),
                J$.T(132121, 'awaitData', 21, false),
                J$.T(132129, 'partial', 21, false),
                J$.T(132137, 'exhaust', 21, false)
            ], 10, false), OP_QUERY_KEYS, false, true);
            var extractCommand = J$.W(133185, 'extractCommand', J$.T(133177, function extractCommand(command) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(133145, arguments.callee, this, arguments);
                            arguments = J$.N(133153, 'arguments', arguments, true, false, false);
                            command = J$.N(133161, 'command', command, true, false, false);
                            J$.N(133169, 'result', result, false, false, false);
                            if (J$.C(7472, J$.B(6986, 'instanceof', J$.R(132161, 'command', command, false, false), J$.R(132169, 'GetMore', GetMore, false, true)))) {
                                return J$.Rt(132241, J$.T(132233, {
                                    getMore: J$.G(132185, J$.R(132177, 'command', command, false, false), 'cursorId'),
                                    collection: J$.F(132209, J$.R(132193, 'collectionName', collectionName, false, true), false)(J$.R(132201, 'command', command, false, false)),
                                    batchSize: J$.G(132225, J$.R(132217, 'command', command, false, false), 'numberToReturn')
                                }, 11, false));
                            }
                            if (J$.C(7480, J$.B(6994, 'instanceof', J$.R(132249, 'command', command, false, false), J$.R(132257, 'KillCursor', KillCursor, false, true)))) {
                                return J$.Rt(132313, J$.T(132305, {
                                    killCursors: J$.F(132281, J$.R(132265, 'collectionName', collectionName, false, true), false)(J$.R(132273, 'command', command, false, false)),
                                    cursors: J$.G(132297, J$.R(132289, 'command', command, false, false), 'cursorIds')
                                }, 11, false));
                            }
                            if (J$.C(7488, J$.B(7002, 'instanceof', J$.R(132321, 'command', command, false, false), J$.R(132329, 'Msg', Msg, false, true)))) {
                                return J$.Rt(132353, J$.G(132345, J$.R(132337, 'command', command, false, false), 'command'));
                            }
                            if (J$.C(7552, J$.C(7496, J$.G(132369, J$.R(132361, 'command', command, false, false), 'query')) ? J$.G(132393, J$.G(132385, J$.R(132377, 'command', command, false, false), 'query'), '$query') : J$._())) {
                                var result;
                                if (J$.C(7512, J$.B(7010, '===', J$.G(132409, J$.R(132401, 'command', command, false, false), 'ns'), J$.T(132417, 'admin.$cmd', 21, false)))) {
                                    result = J$.W(132473, 'result', J$.M(132465, J$.I(typeof Object === 'undefined' ? Object = J$.R(132425, 'Object', undefined, true, true) : Object = J$.R(132425, 'Object', Object, true, true)), 'assign', false)(J$.T(132433, {}, 11, false), J$.G(132457, J$.G(132449, J$.R(132441, 'command', command, false, false), 'query'), '$query')), result, false, false);
                                } else {
                                    result = J$.W(132513, 'result', J$.T(132505, { find: J$.F(132497, J$.R(132481, 'collectionName', collectionName, false, true), false)(J$.R(132489, 'command', command, false, false)) }, 11, false), result, false, false);
                                    J$.M(132689, J$.M(132537, J$.I(typeof Object === 'undefined' ? Object = J$.R(132521, 'Object', undefined, true, true) : Object = J$.R(132521, 'Object', Object, true, true)), 'keys', false)(J$.R(132529, 'LEGACY_FIND_QUERY_MAP', LEGACY_FIND_QUERY_MAP, false, true)), 'forEach', false)(J$.T(132681, function (key) {
                                        jalangiLabel7:
                                            while (true) {
                                                try {
                                                    J$.Fe(132657, arguments.callee, this, arguments);
                                                    arguments = J$.N(132665, 'arguments', arguments, true, false, false);
                                                    key = J$.N(132673, 'key', key, true, false, false);
                                                    if (J$.C(7504, J$.B(7026, '!==', J$.U(7018, 'typeof', J$.G(132569, J$.G(132553, J$.R(132545, 'command', command, false, false), 'query'), J$.R(132561, 'key', key, false, false))), J$.T(132577, 'undefined', 21, false))))
                                                        J$.P(132649, J$.R(132585, 'result', result, false, false), J$.G(132609, J$.R(132593, 'LEGACY_FIND_QUERY_MAP', LEGACY_FIND_QUERY_MAP, false, true), J$.R(132601, 'key', key, false, false)), J$.G(132641, J$.G(132625, J$.R(132617, 'command', command, false, false), 'query'), J$.R(132633, 'key', key, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(134969, J$e);
                                                } finally {
                                                    if (J$.Fr(134977))
                                                        continue jalangiLabel7;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false));
                                }
                                J$.M(132849, J$.M(132713, J$.I(typeof Object === 'undefined' ? Object = J$.R(132697, 'Object', undefined, true, true) : Object = J$.R(132697, 'Object', Object, true, true)), 'keys', false)(J$.R(132705, 'LEGACY_FIND_OPTIONS_MAP', LEGACY_FIND_OPTIONS_MAP, false, true)), 'forEach', false)(J$.T(132841, function (key) {
                                    jalangiLabel8:
                                        while (true) {
                                            try {
                                                J$.Fe(132817, arguments.callee, this, arguments);
                                                arguments = J$.N(132825, 'arguments', arguments, true, false, false);
                                                key = J$.N(132833, 'key', key, true, false, false);
                                                if (J$.C(7520, J$.B(7042, '!==', J$.U(7034, 'typeof', J$.G(132737, J$.R(132721, 'command', command, false, false), J$.R(132729, 'key', key, false, false))), J$.T(132745, 'undefined', 21, false))))
                                                    J$.P(132809, J$.R(132753, 'result', result, false, false), J$.G(132777, J$.R(132761, 'LEGACY_FIND_OPTIONS_MAP', LEGACY_FIND_OPTIONS_MAP, false, true), J$.R(132769, 'key', key, false, false)), J$.G(132801, J$.R(132785, 'command', command, false, false), J$.R(132793, 'key', key, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(134985, J$e);
                                            } finally {
                                                if (J$.Fr(134993))
                                                    continue jalangiLabel8;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                                J$.M(132969, J$.R(132857, 'OP_QUERY_KEYS', OP_QUERY_KEYS, false, true), 'forEach', false)(J$.T(132961, function (key) {
                                    jalangiLabel9:
                                        while (true) {
                                            try {
                                                J$.Fe(132937, arguments.callee, this, arguments);
                                                arguments = J$.N(132945, 'arguments', arguments, true, false, false);
                                                key = J$.N(132953, 'key', key, true, false, false);
                                                if (J$.C(7528, J$.G(132881, J$.R(132865, 'command', command, false, false), J$.R(132873, 'key', key, false, false))))
                                                    J$.P(132929, J$.R(132889, 'result', result, false, false), J$.R(132897, 'key', key, false, false), J$.G(132921, J$.R(132905, 'command', command, false, false), J$.R(132913, 'key', key, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(135001, J$e);
                                            } finally {
                                                if (J$.Fr(135009))
                                                    continue jalangiLabel9;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                                if (J$.C(7536, J$.B(7058, '!==', J$.U(7050, 'typeof', J$.G(132985, J$.R(132977, 'command', command, false, false), 'pre32Limit')), J$.T(132993, 'undefined', 21, false)))) {
                                    J$.P(133025, J$.R(133001, 'result', result, false, false), 'limit', J$.G(133017, J$.R(133009, 'command', command, false, false), 'pre32Limit'));
                                }
                                if (J$.C(7544, J$.G(133049, J$.G(133041, J$.R(133033, 'command', command, false, false), 'query'), '$explain'))) {
                                    return J$.Rt(133073, J$.T(133065, { explain: J$.R(133057, 'result', result, false, false) }, 11, false));
                                }
                                return J$.Rt(133089, J$.R(133081, 'result', result, false, false));
                            }
                            return J$.Rt(133137, J$.C(7560, J$.G(133105, J$.R(133097, 'command', command, false, false), 'query')) ? J$.G(133121, J$.R(133113, 'command', command, false, false), 'query') : J$.R(133129, 'command', command, false, false));
                        } catch (J$e) {
                            J$.Ex(135017, J$e);
                        } finally {
                            if (J$.Fr(135025))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), extractCommand, false, true);
            var extractReply = J$.W(133617, 'extractReply', J$.T(133609, function extractReply(command, reply) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(133577, arguments.callee, this, arguments);
                            arguments = J$.N(133585, 'arguments', arguments, true, false, false);
                            command = J$.N(133593, 'command', command, true, false, false);
                            reply = J$.N(133601, 'reply', reply, true, false, false);
                            if (J$.C(7568, J$.B(7066, 'instanceof', J$.R(133193, 'command', command, false, false), J$.R(133201, 'GetMore', GetMore, false, true)))) {
                                return J$.Rt(133305, J$.T(133297, {
                                    ok: J$.T(133209, 1, 22, false),
                                    cursor: J$.T(133289, {
                                        id: J$.G(133233, J$.G(133225, J$.R(133217, 'reply', reply, false, false), 'message'), 'cursorId'),
                                        ns: J$.F(133257, J$.R(133241, 'namespace', namespace, false, true), false)(J$.R(133249, 'command', command, false, false)),
                                        nextBatch: J$.G(133281, J$.G(133273, J$.R(133265, 'reply', reply, false, false), 'message'), 'documents')
                                    }, 11, false)
                                }, 11, false));
                            }
                            if (J$.C(7576, J$.B(7074, 'instanceof', J$.R(133313, 'command', command, false, false), J$.R(133321, 'KillCursor', KillCursor, false, true)))) {
                                return J$.Rt(133361, J$.T(133353, {
                                    ok: J$.T(133329, 1, 22, false),
                                    cursorsUnknown: J$.G(133345, J$.R(133337, 'command', command, false, false), 'cursorIds')
                                }, 11, false));
                            }
                            if (J$.C(7592, J$.C(7584, J$.G(133377, J$.R(133369, 'command', command, false, false), 'query')) ? J$.B(7090, '!==', J$.U(7082, 'typeof', J$.G(133401, J$.G(133393, J$.R(133385, 'command', command, false, false), 'query'), '$query')), J$.T(133409, 'undefined', 21, false)) : J$._())) {
                                return J$.Rt(133513, J$.T(133505, {
                                    ok: J$.T(133417, 1, 22, false),
                                    cursor: J$.T(133497, {
                                        id: J$.G(133441, J$.G(133433, J$.R(133425, 'reply', reply, false, false), 'message'), 'cursorId'),
                                        ns: J$.F(133465, J$.R(133449, 'namespace', namespace, false, true), false)(J$.R(133457, 'command', command, false, false)),
                                        firstBatch: J$.G(133489, J$.G(133481, J$.R(133473, 'reply', reply, false, false), 'message'), 'documents')
                                    }, 11, false)
                                }, 11, false));
                            }
                            if (J$.C(7600, J$.B(7098, '===', J$.R(133521, 'reply', reply, false, false), J$.T(133529, null, 25, false))))
                                return J$.Rt(133545, J$.R(133537, 'reply', reply, false, false));
                            return J$.Rt(133569, J$.G(133561, J$.R(133553, 'reply', reply, false, false), 'result'));
                        } catch (J$e) {
                            J$.Ex(135033, J$e);
                        } finally {
                            if (J$.Fr(135041))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), extractReply, false, true);
            var CommandStartedEvent = J$.W(133977, 'CommandStartedEvent', J$.T(133969, function CommandStartedEvent(pool, command) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(133921, arguments.callee, this, arguments);
                            arguments = J$.N(133929, 'arguments', arguments, true, false, false);
                            pool = J$.N(133937, 'pool', pool, true, false, false);
                            command = J$.N(133945, 'command', command, true, false, false);
                            J$.N(133953, 'cmd', cmd, false, false, false);
                            J$.N(133961, 'commandName', commandName, false, false, false);
                            J$.F(133649, J$.R(133625, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(133633, 'this', this, false, false), J$.R(133641, 'CommandStartedEvent', CommandStartedEvent, false, false));
                            var cmd = J$.W(133681, 'cmd', J$.F(133673, J$.R(133657, 'extractCommand', extractCommand, false, true), false)(J$.R(133665, 'command', command, false, false)), cmd, false, false);
                            var commandName = J$.W(133713, 'commandName', J$.F(133705, J$.R(133689, 'extractCommandName', extractCommandName, false, true), false)(J$.R(133697, 'cmd', cmd, false, false)), commandName, false, false);
                            if (J$.C(7608, J$.M(133737, J$.R(133721, 'SENSITIVE_COMMANDS', SENSITIVE_COMMANDS, false, true), 'has', false)(J$.R(133729, 'commandName', commandName, false, false)))) {
                                J$.P(133761, J$.R(133745, 'this', this, false, false), 'commandObj', J$.T(133753, {}, 11, false));
                                J$.P(133801, J$.G(133777, J$.R(133769, 'this', this, false, false), 'commandObj'), J$.R(133785, 'commandName', commandName, false, false), J$.T(133793, true, 23, false));
                            }
                            J$.M(133913, J$.I(typeof Object === 'undefined' ? Object = J$.R(133809, 'Object', undefined, true, true) : Object = J$.R(133809, 'Object', Object, true, true)), 'assign', false)(J$.R(133817, 'this', this, false, false), J$.T(133905, {
                                command: J$.R(133825, 'cmd', cmd, false, false),
                                databaseName: J$.F(133849, J$.R(133833, 'databaseName', databaseName, false, true), false)(J$.R(133841, 'command', command, false, false)),
                                commandName: J$.R(133857, 'commandName', commandName, false, false),
                                requestId: J$.G(133873, J$.R(133865, 'command', command, false, false), 'requestId'),
                                connectionId: J$.F(133897, J$.R(133881, 'generateConnectionId', generateConnectionId, false, true), false)(J$.R(133889, 'pool', pool, false, false))
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(135049, J$e);
                        } finally {
                            if (J$.Fr(135057))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), CommandStartedEvent, false, true);
            var CommandSucceededEvent = J$.W(134313, 'CommandSucceededEvent', J$.T(134305, function CommandSucceededEvent(pool, command, reply, started) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(134241, arguments.callee, this, arguments);
                            arguments = J$.N(134249, 'arguments', arguments, true, false, false);
                            pool = J$.N(134257, 'pool', pool, true, false, false);
                            command = J$.N(134265, 'command', command, true, false, false);
                            reply = J$.N(134273, 'reply', reply, true, false, false);
                            started = J$.N(134281, 'started', started, true, false, false);
                            J$.N(134289, 'cmd', cmd, false, false, false);
                            J$.N(134297, 'commandName', commandName, false, false, false);
                            J$.F(134009, J$.R(133985, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(133993, 'this', this, false, false), J$.R(134001, 'CommandSucceededEvent', CommandSucceededEvent, false, false));
                            var cmd = J$.W(134041, 'cmd', J$.F(134033, J$.R(134017, 'extractCommand', extractCommand, false, true), false)(J$.R(134025, 'command', command, false, false)), cmd, false, false);
                            var commandName = J$.W(134073, 'commandName', J$.F(134065, J$.R(134049, 'extractCommandName', extractCommandName, false, true), false)(J$.R(134057, 'cmd', cmd, false, false)), commandName, false, false);
                            J$.M(134233, J$.I(typeof Object === 'undefined' ? Object = J$.R(134081, 'Object', undefined, true, true) : Object = J$.R(134081, 'Object', Object, true, true)), 'assign', false)(J$.R(134089, 'this', this, false, false), J$.T(134225, {
                                duration: J$.F(134113, J$.R(134097, 'calculateDurationInMs', calculateDurationInMs, false, true), false)(J$.R(134105, 'started', started, false, false)),
                                commandName: J$.R(134121, 'commandName', commandName, false, false),
                                reply: J$.F(134177, J$.R(134129, 'maybeRedact', maybeRedact, false, true), false)(J$.R(134137, 'commandName', commandName, false, false), J$.F(134169, J$.R(134145, 'extractReply', extractReply, false, true), false)(J$.R(134153, 'command', command, false, false), J$.R(134161, 'reply', reply, false, false))),
                                requestId: J$.G(134193, J$.R(134185, 'command', command, false, false), 'requestId'),
                                connectionId: J$.F(134217, J$.R(134201, 'generateConnectionId', generateConnectionId, false, true), false)(J$.R(134209, 'pool', pool, false, false))
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(135065, J$e);
                        } finally {
                            if (J$.Fr(135073))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), CommandSucceededEvent, false, true);
            var CommandFailedEvent = J$.W(134625, 'CommandFailedEvent', J$.T(134617, function CommandFailedEvent(pool, command, error, started) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(134553, arguments.callee, this, arguments);
                            arguments = J$.N(134561, 'arguments', arguments, true, false, false);
                            pool = J$.N(134569, 'pool', pool, true, false, false);
                            command = J$.N(134577, 'command', command, true, false, false);
                            error = J$.N(134585, 'error', error, true, false, false);
                            started = J$.N(134593, 'started', started, true, false, false);
                            J$.N(134601, 'cmd', cmd, false, false, false);
                            J$.N(134609, 'commandName', commandName, false, false, false);
                            J$.F(134345, J$.R(134321, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(134329, 'this', this, false, false), J$.R(134337, 'CommandFailedEvent', CommandFailedEvent, false, false));
                            var cmd = J$.W(134377, 'cmd', J$.F(134369, J$.R(134353, 'extractCommand', extractCommand, false, true), false)(J$.R(134361, 'command', command, false, false)), cmd, false, false);
                            var commandName = J$.W(134409, 'commandName', J$.F(134401, J$.R(134385, 'extractCommandName', extractCommandName, false, true), false)(J$.R(134393, 'cmd', cmd, false, false)), commandName, false, false);
                            J$.M(134545, J$.I(typeof Object === 'undefined' ? Object = J$.R(134417, 'Object', undefined, true, true) : Object = J$.R(134417, 'Object', Object, true, true)), 'assign', false)(J$.R(134425, 'this', this, false, false), J$.T(134537, {
                                duration: J$.F(134449, J$.R(134433, 'calculateDurationInMs', calculateDurationInMs, false, true), false)(J$.R(134441, 'started', started, false, false)),
                                commandName: J$.R(134457, 'commandName', commandName, false, false),
                                failure: J$.F(134489, J$.R(134465, 'maybeRedact', maybeRedact, false, true), false)(J$.R(134473, 'commandName', commandName, false, false), J$.R(134481, 'error', error, false, false)),
                                requestId: J$.G(134505, J$.R(134497, 'command', command, false, false), 'requestId'),
                                connectionId: J$.F(134529, J$.R(134513, 'generateConnectionId', generateConnectionId, false, true), false)(J$.R(134521, 'pool', pool, false, false))
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(135081, J$e);
                        } finally {
                            if (J$.Fr(135089))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), CommandFailedEvent, false, true);
            J$.P(134673, J$.I(typeof module === 'undefined' ? module = J$.R(134633, 'module', undefined, true, true) : module = J$.R(134633, 'module', module, true, true)), 'exports', J$.T(134665, {
                CommandStartedEvent: J$.R(134641, 'CommandStartedEvent', CommandStartedEvent, false, true),
                CommandSucceededEvent: J$.R(134649, 'CommandSucceededEvent', CommandSucceededEvent, false, true),
                CommandFailedEvent: J$.R(134657, 'CommandFailedEvent', CommandFailedEvent, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(135097, J$e);
        } finally {
            if (J$.Sr(135105))
                continue jalangiLabel15;
            else
                break jalangiLabel15;
        }
    }
// JALANGI DO NOT INSTRUMENT
