// mongodb/lib/core/connection/logger.js
J$.noInstrEval = false;
jalangiLabel15:
    while (true) {
        try {
            J$.Se(172289, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/core/connection/logger.js');
            J$.N(172297, 'f', f, false, false, false);
            J$.N(172305, 'MongoError', MongoError, false, false, false);
            J$.N(172313, 'classFilters', classFilters, false, false, false);
            J$.N(172321, 'filteredClasses', filteredClasses, false, false, false);
            J$.N(172329, 'level', level, false, false, false);
            J$.N(172337, 'pid', pid, false, false, false);
            J$.N(172345, 'currentLogger', currentLogger, false, false, false);
            J$.N(172353, 'Logger', Logger, false, false, false);
            J$.T(168793, 'use strict', 21, false);
            var f = J$.W(168865, 'f', J$.G(168825, J$.F(168817, J$.I(typeof require === 'undefined' ? require = J$.R(168801, 'require', undefined, true, true) : require = J$.R(168801, 'require', require, true, true)), false)(J$.T(168809, 'util', 21, false)), 'format'), f, false, true), MongoError = J$.W(168873, 'MongoError', J$.G(168857, J$.F(168849, J$.I(typeof require === 'undefined' ? require = J$.R(168833, 'require', undefined, true, true) : require = J$.R(168833, 'require', require, true, true)), false)(J$.T(168841, '../error', 21, false)), 'MongoError'), MongoError, false, true);
            var classFilters = J$.W(168889, 'classFilters', J$.T(168881, {}, 11, false), classFilters, false, true);
            var filteredClasses = J$.W(168905, 'filteredClasses', J$.T(168897, {}, 11, false), filteredClasses, false, true);
            var level = J$.W(168921, 'level', J$.T(168913, null, 25, false), level, false, true);
            var pid = J$.W(168945, 'pid', J$.G(168937, J$.I(typeof process === 'undefined' ? process = J$.R(168929, 'process', undefined, true, true) : process = J$.R(168929, 'process', process, true, true)), 'pid'), pid, false, true);
            var currentLogger = J$.W(168961, 'currentLogger', J$.T(168953, null, 25, false), currentLogger, false, true);
            var Logger = J$.W(169321, 'Logger', J$.T(169313, function Logger(className, options) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(169281, arguments.callee, this, arguments);
                            arguments = J$.N(169289, 'arguments', arguments, true, false, false);
                            className = J$.N(169297, 'className', className, true, false, false);
                            options = J$.N(169305, 'options', options, true, false, false);
                            if (J$.C(9160, J$.U(11234, '!', J$.B(11226, 'instanceof', J$.R(168969, 'this', this, false, false), J$.R(168977, 'Logger', Logger, false, false)))))
                                return J$.Rt(169017, J$.F(169009, J$.R(168985, 'Logger', Logger, false, false), true)(J$.R(168993, 'className', className, false, false), J$.R(169001, 'options', options, false, false)));
                            options = J$.W(169041, 'options', J$.C(9168, J$.R(169025, 'options', options, false, false)) ? J$._() : J$.T(169033, {}, 11, false), options, false, false);
                            J$.P(169065, J$.R(169049, 'this', this, false, false), 'className', J$.R(169057, 'className', className, false, false));
                            if (J$.C(9184, J$.G(169081, J$.R(169073, 'options', options, false, false), 'logger'))) {
                                currentLogger = J$.W(169105, 'currentLogger', J$.G(169097, J$.R(169089, 'options', options, false, false), 'logger'), currentLogger, false, true);
                            } else if (J$.C(9176, J$.B(11242, '==', J$.R(169113, 'currentLogger', currentLogger, false, true), J$.T(169121, null, 25, false)))) {
                                currentLogger = J$.W(169145, 'currentLogger', J$.G(169137, J$.I(typeof console === 'undefined' ? console = J$.R(169129, 'console', undefined, true, true) : console = J$.R(169129, 'console', console, true, true)), 'log'), currentLogger, false, true);
                            }
                            if (J$.C(9200, J$.G(169161, J$.R(169153, 'options', options, false, false), 'loggerLevel'))) {
                                level = J$.W(169193, 'level', J$.C(9192, J$.G(169177, J$.R(169169, 'options', options, false, false), 'loggerLevel')) ? J$._() : J$.T(169185, 'error', 21, false), level, false, true);
                            }
                            if (J$.C(9208, J$.B(11250, '==', J$.G(169225, J$.R(169201, 'filteredClasses', filteredClasses, false, true), J$.G(169217, J$.R(169209, 'this', this, false, false), 'className')), J$.T(169233, null, 25, false))))
                                J$.P(169273, J$.R(169241, 'classFilters', classFilters, false, true), J$.G(169257, J$.R(169249, 'this', this, false, false), 'className'), J$.T(169265, true, 23, false));
                        } catch (J$e) {
                            J$.Ex(172361, J$e);
                        } finally {
                            if (J$.Fr(172369))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), Logger, false, true);
            J$.P(169809, J$.G(169337, J$.R(169329, 'Logger', Logger, false, true), 'prototype'), 'debug', J$.T(169801, function (message, object) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(169745, arguments.callee, this, arguments);
                            arguments = J$.N(169753, 'arguments', arguments, true, false, false);
                            message = J$.N(169761, 'message', message, true, false, false);
                            object = J$.N(169769, 'object', object, true, false, false);
                            J$.N(169777, 'dateTime', dateTime, false, false, false);
                            J$.N(169785, 'msg', msg, false, false, false);
                            J$.N(169793, 'state', state, false, false, false);
                            if (J$.C(9256, J$.C(9240, J$.M(169353, J$.R(169345, 'this', this, false, false), 'isDebug', false)()) ? J$.C(9232, J$.C(9216, J$.B(11258, '>', J$.G(169385, J$.M(169377, J$.I(typeof Object === 'undefined' ? Object = J$.R(169361, 'Object', undefined, true, true) : Object = J$.R(169361, 'Object', Object, true, true)), 'keys', false)(J$.R(169369, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(169393, 0, 22, false))) ? J$.G(169425, J$.R(169401, 'filteredClasses', filteredClasses, false, true), J$.G(169417, J$.R(169409, 'this', this, false, false), 'className')) : J$._()) ? J$._() : J$.C(9224, J$.B(11266, '===', J$.G(169457, J$.M(169449, J$.I(typeof Object === 'undefined' ? Object = J$.R(169433, 'Object', undefined, true, true) : Object = J$.R(169433, 'Object', Object, true, true)), 'keys', false)(J$.R(169441, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(169465, 0, 22, false))) ? J$.G(169497, J$.R(169473, 'classFilters', classFilters, false, true), J$.G(169489, J$.R(169481, 'this', this, false, false), 'className')) : J$._() : J$._())) {
                                var dateTime = J$.W(169529, 'dateTime', J$.M(169521, J$.F(169513, J$.I(typeof Date === 'undefined' ? Date = J$.R(169505, 'Date', undefined, true, true) : Date = J$.R(169505, 'Date', Date, true, true)), true)(), 'getTime', false)(), dateTime, false, false);
                                var msg = J$.W(169609, 'msg', J$.F(169601, J$.R(169537, 'f', f, false, true), false)(J$.T(169545, '[%s-%s:%s] %s %s', 21, false), J$.T(169553, 'DEBUG', 21, false), J$.G(169569, J$.R(169561, 'this', this, false, false), 'className'), J$.R(169577, 'pid', pid, false, true), J$.R(169585, 'dateTime', dateTime, false, false), J$.R(169593, 'message', message, false, false)), msg, false, false);
                                var state = J$.W(169673, 'state', J$.T(169665, {
                                    type: J$.T(169617, 'debug', 21, false),
                                    message: J$.R(169625, 'message', message, false, false),
                                    className: J$.G(169641, J$.R(169633, 'this', this, false, false), 'className'),
                                    pid: J$.R(169649, 'pid', pid, false, true),
                                    date: J$.R(169657, 'dateTime', dateTime, false, false)
                                }, 11, false), state, false, false);
                                if (J$.C(9248, J$.R(169681, 'object', object, false, false)))
                                    J$.P(169705, J$.R(169689, 'state', state, false, false), 'meta', J$.R(169697, 'object', object, false, false));
                                J$.F(169737, J$.R(169713, 'currentLogger', currentLogger, false, true), false)(J$.R(169721, 'msg', msg, false, false), J$.R(169729, 'state', state, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(172377, J$e);
                        } finally {
                            if (J$.Fr(172385))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(170297, J$.G(169825, J$.R(169817, 'Logger', Logger, false, true), 'prototype'), 'warn', J$.T(170289, function (message, object) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(170233, arguments.callee, this, arguments);
                            arguments = J$.N(170241, 'arguments', arguments, true, false, false);
                            message = J$.N(170249, 'message', message, true, false, false);
                            object = J$.N(170257, 'object', object, true, false, false);
                            J$.N(170265, 'dateTime', dateTime, false, false, false);
                            J$.N(170273, 'msg', msg, false, false, false);
                            J$.N(170281, 'state', state, false, false, false);
                            if (J$.C(9304, J$.C(9288, J$.M(169841, J$.R(169833, 'this', this, false, false), 'isWarn', false)()) ? J$.C(9280, J$.C(9264, J$.B(11274, '>', J$.G(169873, J$.M(169865, J$.I(typeof Object === 'undefined' ? Object = J$.R(169849, 'Object', undefined, true, true) : Object = J$.R(169849, 'Object', Object, true, true)), 'keys', false)(J$.R(169857, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(169881, 0, 22, false))) ? J$.G(169913, J$.R(169889, 'filteredClasses', filteredClasses, false, true), J$.G(169905, J$.R(169897, 'this', this, false, false), 'className')) : J$._()) ? J$._() : J$.C(9272, J$.B(11282, '===', J$.G(169945, J$.M(169937, J$.I(typeof Object === 'undefined' ? Object = J$.R(169921, 'Object', undefined, true, true) : Object = J$.R(169921, 'Object', Object, true, true)), 'keys', false)(J$.R(169929, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(169953, 0, 22, false))) ? J$.G(169985, J$.R(169961, 'classFilters', classFilters, false, true), J$.G(169977, J$.R(169969, 'this', this, false, false), 'className')) : J$._() : J$._())) {
                                var dateTime = J$.W(170017, 'dateTime', J$.M(170009, J$.F(170001, J$.I(typeof Date === 'undefined' ? Date = J$.R(169993, 'Date', undefined, true, true) : Date = J$.R(169993, 'Date', Date, true, true)), true)(), 'getTime', false)(), dateTime, false, false);
                                var msg = J$.W(170097, 'msg', J$.F(170089, J$.R(170025, 'f', f, false, true), false)(J$.T(170033, '[%s-%s:%s] %s %s', 21, false), J$.T(170041, 'WARN', 21, false), J$.G(170057, J$.R(170049, 'this', this, false, false), 'className'), J$.R(170065, 'pid', pid, false, true), J$.R(170073, 'dateTime', dateTime, false, false), J$.R(170081, 'message', message, false, false)), msg, false, false);
                                var state = J$.W(170161, 'state', J$.T(170153, {
                                    type: J$.T(170105, 'warn', 21, false),
                                    message: J$.R(170113, 'message', message, false, false),
                                    className: J$.G(170129, J$.R(170121, 'this', this, false, false), 'className'),
                                    pid: J$.R(170137, 'pid', pid, false, true),
                                    date: J$.R(170145, 'dateTime', dateTime, false, false)
                                }, 11, false), state, false, false);
                                if (J$.C(9296, J$.R(170169, 'object', object, false, false)))
                                    J$.P(170193, J$.R(170177, 'state', state, false, false), 'meta', J$.R(170185, 'object', object, false, false));
                                J$.F(170225, J$.R(170201, 'currentLogger', currentLogger, false, true), false)(J$.R(170209, 'msg', msg, false, false), J$.R(170217, 'state', state, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(172393, J$e);
                        } finally {
                            if (J$.Fr(172401))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(170785, J$.G(170313, J$.R(170305, 'Logger', Logger, false, true), 'prototype'), 'info', J$.T(170777, function (message, object) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(170721, arguments.callee, this, arguments);
                            arguments = J$.N(170729, 'arguments', arguments, true, false, false);
                            message = J$.N(170737, 'message', message, true, false, false);
                            object = J$.N(170745, 'object', object, true, false, false);
                            J$.N(170753, 'dateTime', dateTime, false, false, false);
                            J$.N(170761, 'msg', msg, false, false, false);
                            J$.N(170769, 'state', state, false, false, false);
                            if (J$.C(9352, J$.C(9336, J$.M(170329, J$.R(170321, 'this', this, false, false), 'isInfo', false)()) ? J$.C(9328, J$.C(9312, J$.B(11290, '>', J$.G(170361, J$.M(170353, J$.I(typeof Object === 'undefined' ? Object = J$.R(170337, 'Object', undefined, true, true) : Object = J$.R(170337, 'Object', Object, true, true)), 'keys', false)(J$.R(170345, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(170369, 0, 22, false))) ? J$.G(170401, J$.R(170377, 'filteredClasses', filteredClasses, false, true), J$.G(170393, J$.R(170385, 'this', this, false, false), 'className')) : J$._()) ? J$._() : J$.C(9320, J$.B(11298, '===', J$.G(170433, J$.M(170425, J$.I(typeof Object === 'undefined' ? Object = J$.R(170409, 'Object', undefined, true, true) : Object = J$.R(170409, 'Object', Object, true, true)), 'keys', false)(J$.R(170417, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(170441, 0, 22, false))) ? J$.G(170473, J$.R(170449, 'classFilters', classFilters, false, true), J$.G(170465, J$.R(170457, 'this', this, false, false), 'className')) : J$._() : J$._())) {
                                var dateTime = J$.W(170505, 'dateTime', J$.M(170497, J$.F(170489, J$.I(typeof Date === 'undefined' ? Date = J$.R(170481, 'Date', undefined, true, true) : Date = J$.R(170481, 'Date', Date, true, true)), true)(), 'getTime', false)(), dateTime, false, false);
                                var msg = J$.W(170585, 'msg', J$.F(170577, J$.R(170513, 'f', f, false, true), false)(J$.T(170521, '[%s-%s:%s] %s %s', 21, false), J$.T(170529, 'INFO', 21, false), J$.G(170545, J$.R(170537, 'this', this, false, false), 'className'), J$.R(170553, 'pid', pid, false, true), J$.R(170561, 'dateTime', dateTime, false, false), J$.R(170569, 'message', message, false, false)), msg, false, false);
                                var state = J$.W(170649, 'state', J$.T(170641, {
                                    type: J$.T(170593, 'info', 21, false),
                                    message: J$.R(170601, 'message', message, false, false),
                                    className: J$.G(170617, J$.R(170609, 'this', this, false, false), 'className'),
                                    pid: J$.R(170625, 'pid', pid, false, true),
                                    date: J$.R(170633, 'dateTime', dateTime, false, false)
                                }, 11, false), state, false, false);
                                if (J$.C(9344, J$.R(170657, 'object', object, false, false)))
                                    J$.P(170681, J$.R(170665, 'state', state, false, false), 'meta', J$.R(170673, 'object', object, false, false));
                                J$.F(170713, J$.R(170689, 'currentLogger', currentLogger, false, true), false)(J$.R(170697, 'msg', msg, false, false), J$.R(170705, 'state', state, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(172409, J$e);
                        } finally {
                            if (J$.Fr(172417))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(171273, J$.G(170801, J$.R(170793, 'Logger', Logger, false, true), 'prototype'), 'error', J$.T(171265, function (message, object) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(171209, arguments.callee, this, arguments);
                            arguments = J$.N(171217, 'arguments', arguments, true, false, false);
                            message = J$.N(171225, 'message', message, true, false, false);
                            object = J$.N(171233, 'object', object, true, false, false);
                            J$.N(171241, 'dateTime', dateTime, false, false, false);
                            J$.N(171249, 'msg', msg, false, false, false);
                            J$.N(171257, 'state', state, false, false, false);
                            if (J$.C(9400, J$.C(9384, J$.M(170817, J$.R(170809, 'this', this, false, false), 'isError', false)()) ? J$.C(9376, J$.C(9360, J$.B(11306, '>', J$.G(170849, J$.M(170841, J$.I(typeof Object === 'undefined' ? Object = J$.R(170825, 'Object', undefined, true, true) : Object = J$.R(170825, 'Object', Object, true, true)), 'keys', false)(J$.R(170833, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(170857, 0, 22, false))) ? J$.G(170889, J$.R(170865, 'filteredClasses', filteredClasses, false, true), J$.G(170881, J$.R(170873, 'this', this, false, false), 'className')) : J$._()) ? J$._() : J$.C(9368, J$.B(11314, '===', J$.G(170921, J$.M(170913, J$.I(typeof Object === 'undefined' ? Object = J$.R(170897, 'Object', undefined, true, true) : Object = J$.R(170897, 'Object', Object, true, true)), 'keys', false)(J$.R(170905, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(170929, 0, 22, false))) ? J$.G(170961, J$.R(170937, 'classFilters', classFilters, false, true), J$.G(170953, J$.R(170945, 'this', this, false, false), 'className')) : J$._() : J$._())) {
                                var dateTime = J$.W(170993, 'dateTime', J$.M(170985, J$.F(170977, J$.I(typeof Date === 'undefined' ? Date = J$.R(170969, 'Date', undefined, true, true) : Date = J$.R(170969, 'Date', Date, true, true)), true)(), 'getTime', false)(), dateTime, false, false);
                                var msg = J$.W(171073, 'msg', J$.F(171065, J$.R(171001, 'f', f, false, true), false)(J$.T(171009, '[%s-%s:%s] %s %s', 21, false), J$.T(171017, 'ERROR', 21, false), J$.G(171033, J$.R(171025, 'this', this, false, false), 'className'), J$.R(171041, 'pid', pid, false, true), J$.R(171049, 'dateTime', dateTime, false, false), J$.R(171057, 'message', message, false, false)), msg, false, false);
                                var state = J$.W(171137, 'state', J$.T(171129, {
                                    type: J$.T(171081, 'error', 21, false),
                                    message: J$.R(171089, 'message', message, false, false),
                                    className: J$.G(171105, J$.R(171097, 'this', this, false, false), 'className'),
                                    pid: J$.R(171113, 'pid', pid, false, true),
                                    date: J$.R(171121, 'dateTime', dateTime, false, false)
                                }, 11, false), state, false, false);
                                if (J$.C(9392, J$.R(171145, 'object', object, false, false)))
                                    J$.P(171169, J$.R(171153, 'state', state, false, false), 'meta', J$.R(171161, 'object', object, false, false));
                                J$.F(171201, J$.R(171177, 'currentLogger', currentLogger, false, true), false)(J$.R(171185, 'msg', msg, false, false), J$.R(171193, 'state', state, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(172425, J$e);
                        } finally {
                            if (J$.Fr(172433))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(171361, J$.G(171289, J$.R(171281, 'Logger', Logger, false, true), 'prototype'), 'isInfo', J$.T(171353, function () {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(171337, arguments.callee, this, arguments);
                            arguments = J$.N(171345, 'arguments', arguments, true, false, false);
                            return J$.Rt(171329, J$.C(9408, J$.B(11322, '===', J$.R(171297, 'level', level, false, true), J$.T(171305, 'info', 21, false))) ? J$._() : J$.B(11330, '===', J$.R(171313, 'level', level, false, true), J$.T(171321, 'debug', 21, false)));
                        } catch (J$e) {
                            J$.Ex(172441, J$e);
                        } finally {
                            if (J$.Fr(172449))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(171465, J$.G(171377, J$.R(171369, 'Logger', Logger, false, true), 'prototype'), 'isError', J$.T(171457, function () {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(171441, arguments.callee, this, arguments);
                            arguments = J$.N(171449, 'arguments', arguments, true, false, false);
                            return J$.Rt(171433, J$.C(9424, J$.C(9416, J$.B(11338, '===', J$.R(171385, 'level', level, false, true), J$.T(171393, 'error', 21, false))) ? J$._() : J$.B(11346, '===', J$.R(171401, 'level', level, false, true), J$.T(171409, 'info', 21, false))) ? J$._() : J$.B(11354, '===', J$.R(171417, 'level', level, false, true), J$.T(171425, 'debug', 21, false)));
                        } catch (J$e) {
                            J$.Ex(172457, J$e);
                        } finally {
                            if (J$.Fr(172465))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(171585, J$.G(171481, J$.R(171473, 'Logger', Logger, false, true), 'prototype'), 'isWarn', J$.T(171577, function () {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(171561, arguments.callee, this, arguments);
                            arguments = J$.N(171569, 'arguments', arguments, true, false, false);
                            return J$.Rt(171553, J$.C(9448, J$.C(9440, J$.C(9432, J$.B(11362, '===', J$.R(171489, 'level', level, false, true), J$.T(171497, 'error', 21, false))) ? J$._() : J$.B(11370, '===', J$.R(171505, 'level', level, false, true), J$.T(171513, 'warn', 21, false))) ? J$._() : J$.B(11378, '===', J$.R(171521, 'level', level, false, true), J$.T(171529, 'info', 21, false))) ? J$._() : J$.B(11386, '===', J$.R(171537, 'level', level, false, true), J$.T(171545, 'debug', 21, false)));
                        } catch (J$e) {
                            J$.Ex(172473, J$e);
                        } finally {
                            if (J$.Fr(172481))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(171657, J$.G(171601, J$.R(171593, 'Logger', Logger, false, true), 'prototype'), 'isDebug', J$.T(171649, function () {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(171633, arguments.callee, this, arguments);
                            arguments = J$.N(171641, 'arguments', arguments, true, false, false);
                            return J$.Rt(171625, J$.B(11394, '===', J$.R(171609, 'level', level, false, true), J$.T(171617, 'debug', 21, false)));
                        } catch (J$e) {
                            J$.Ex(172489, J$e);
                        } finally {
                            if (J$.Fr(172497))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(171729, J$.R(171665, 'Logger', Logger, false, true), 'reset', J$.T(171721, function () {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(171705, arguments.callee, this, arguments);
                            arguments = J$.N(171713, 'arguments', arguments, true, false, false);
                            level = J$.W(171681, 'level', J$.T(171673, 'error', 21, false), level, false, true);
                            filteredClasses = J$.W(171697, 'filteredClasses', J$.T(171689, {}, 11, false), filteredClasses, false, true);
                        } catch (J$e) {
                            J$.Ex(172505, J$e);
                        } finally {
                            if (J$.Fr(172513))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(171785, J$.R(171737, 'Logger', Logger, false, true), 'currentLogger', J$.T(171777, function () {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(171761, arguments.callee, this, arguments);
                            arguments = J$.N(171769, 'arguments', arguments, true, false, false);
                            return J$.Rt(171753, J$.R(171745, 'currentLogger', currentLogger, false, true));
                        } catch (J$e) {
                            J$.Ex(172521, J$e);
                        } finally {
                            if (J$.Fr(172529))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(171889, J$.R(171793, 'Logger', Logger, false, true), 'setCurrentLogger', J$.T(171881, function (logger) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(171857, arguments.callee, this, arguments);
                            arguments = J$.N(171865, 'arguments', arguments, true, false, false);
                            logger = J$.N(171873, 'logger', logger, true, false, false);
                            if (J$.C(9456, J$.B(11410, '!==', J$.U(11402, 'typeof', J$.R(171801, 'logger', logger, false, false)), J$.T(171809, 'function', 21, false))))
                                throw J$.F(171833, J$.R(171817, 'MongoError', MongoError, false, true), true)(J$.T(171825, 'current logger must be a function', 21, false));
                            currentLogger = J$.W(171849, 'currentLogger', J$.R(171841, 'logger', logger, false, false), currentLogger, false, true);
                        } catch (J$e) {
                            J$.Ex(172537, J$e);
                        } finally {
                            if (J$.Fr(172545))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(172081, J$.R(171897, 'Logger', Logger, false, true), 'filter', J$.T(172073, function (type, values) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(172041, arguments.callee, this, arguments);
                            arguments = J$.N(172049, 'arguments', arguments, true, false, false);
                            type = J$.N(172057, 'type', type, true, false, false);
                            values = J$.N(172065, 'values', values, true, false, false);
                            if (J$.C(9472, J$.C(9464, J$.B(11418, '===', J$.R(171905, 'type', type, false, false), J$.T(171913, 'class', 21, false))) ? J$.M(171937, J$.I(typeof Array === 'undefined' ? Array = J$.R(171921, 'Array', undefined, true, true) : Array = J$.R(171921, 'Array', Array, true, true)), 'isArray', false)(J$.R(171929, 'values', values, false, false)) : J$._())) {
                                filteredClasses = J$.W(171953, 'filteredClasses', J$.T(171945, {}, 11, false), filteredClasses, false, true);
                                J$.M(172033, J$.R(171961, 'values', values, false, false), 'forEach', false)(J$.T(172025, function (x) {
                                    jalangiLabel12:
                                        while (true) {
                                            try {
                                                J$.Fe(172001, arguments.callee, this, arguments);
                                                arguments = J$.N(172009, 'arguments', arguments, true, false, false);
                                                x = J$.N(172017, 'x', x, true, false, false);
                                                J$.P(171993, J$.R(171969, 'filteredClasses', filteredClasses, false, true), J$.R(171977, 'x', x, false, false), J$.T(171985, true, 23, false));
                                            } catch (J$e) {
                                                J$.Ex(172553, J$e);
                                            } finally {
                                                if (J$.Fr(172561))
                                                    continue jalangiLabel12;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            }
                        } catch (J$e) {
                            J$.Ex(172569, J$e);
                        } finally {
                            if (J$.Fr(172577))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(172257, J$.R(172089, 'Logger', Logger, false, true), 'setLevel', J$.T(172249, function (_level) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(172225, arguments.callee, this, arguments);
                            arguments = J$.N(172233, 'arguments', arguments, true, false, false);
                            _level = J$.N(172241, '_level', _level, true, false, false);
                            if (J$.C(9504, J$.C(9496, J$.C(9488, J$.C(9480, J$.B(11426, '!==', J$.R(172097, '_level', _level, false, false), J$.T(172105, 'info', 21, false))) ? J$.B(11434, '!==', J$.R(172113, '_level', _level, false, false), J$.T(172121, 'error', 21, false)) : J$._()) ? J$.B(11442, '!==', J$.R(172129, '_level', _level, false, false), J$.T(172137, 'debug', 21, false)) : J$._()) ? J$.B(11450, '!==', J$.R(172145, '_level', _level, false, false), J$.T(172153, 'warn', 21, false)) : J$._())) {
                                throw J$.F(172201, J$.I(typeof Error === 'undefined' ? Error = J$.R(172161, 'Error', undefined, true, true) : Error = J$.R(172161, 'Error', Error, true, true)), true)(J$.F(172193, J$.R(172169, 'f', f, false, true), false)(J$.T(172177, '%s is an illegal logging level', 21, false), J$.R(172185, '_level', _level, false, false)));
                            }
                            level = J$.W(172217, 'level', J$.R(172209, '_level', _level, false, false), level, false, true);
                        } catch (J$e) {
                            J$.Ex(172585, J$e);
                        } finally {
                            if (J$.Fr(172593))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(172281, J$.I(typeof module === 'undefined' ? module = J$.R(172265, 'module', undefined, true, true) : module = J$.R(172265, 'module', module, true, true)), 'exports', J$.R(172273, 'Logger', Logger, false, true));
        } catch (J$e) {
            J$.Ex(172601, J$e);
        } finally {
            if (J$.Sr(172609))
                continue jalangiLabel15;
            else
                break jalangiLabel15;
        }
    }
// JALANGI DO NOT INSTRUMENT
