// sift/webpack.config.js
var execSync = require('child_process').execSync;
execSync("echo '" + __filename + "' >> /tmp/file_paths");
var execSync = require('child_process').execSync;
execSync("echo '" + __filename + "' >> /tmp/file_paths");
var execSync = require('child_process').execSync;
execSync("echo '" + __filename + "' >> /tmp/file_paths");
var execSync = require('child_process').execSync;
execSync("echo '" + __filename + "' >> /tmp/file_paths");
var execSync = require('child_process').execSync;
execSync("echo '" + __filename + "' >> /tmp/file_paths");
var execSync = require('child_process').execSync;
execSync("echo '" + __filename + "' >> /tmp/file_paths");
var execSync = require('child_process').execSync;
execSync("echo '" + __filename + "' >> /tmp/file_paths");
var execSync = require('child_process').execSync;
execSync("echo '" + __filename + "' >> /tmp/file_paths");
var execSync = require('child_process').execSync;
execSync("echo '" + __filename + "' >> /tmp/file_paths");
const {resolve} = require('path');
const fs = require('fs');

module.exports = {
  devtool: 'none',
  mode: 'production',
  entry: {
    index: [__dirname + '/lib/index.js']
  },
  output: {
    path: __dirname,
    library: 'sift',
    libraryTarget: 'umd',
    filename: 'sift.min.js'
  },
  resolve: {
    extensions: ['.js']
  },
  module: {
    rules: [
    ]
  }
};
