// tar-stream/pack.js
J$.noInstrEval = false;
jalangiLabel21:
    while (true) {
        try {
            J$.Se(5301281, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/tar-stream/pack.js');
            function modeToType(mode) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(5296809, arguments.callee, this, arguments);
                            arguments = J$.N(5296817, 'arguments', arguments, true, false, false);
                            mode = J$.N(5296825, 'mode', mode, true, false, false);
                            switch (J$.C1(284536, J$.B(601178, '&', J$.R(5296609, 'mode', mode, false, false), J$.G(5296625, J$.R(5296617, 'constants', constants, false, true), 'S_IFMT')))) {
                            case J$.C2(284544, J$.G(5296657, J$.R(5296649, 'constants', constants, false, true), 'S_IFBLK')):
                                return J$.Rt(5296641, J$.T(5296633, 'block-device', 21, false));
                            case J$.C2(284552, J$.G(5296689, J$.R(5296681, 'constants', constants, false, true), 'S_IFCHR')):
                                return J$.Rt(5296673, J$.T(5296665, 'character-device', 21, false));
                            case J$.C2(284560, J$.G(5296721, J$.R(5296713, 'constants', constants, false, true), 'S_IFDIR')):
                                return J$.Rt(5296705, J$.T(5296697, 'directory', 21, false));
                            case J$.C2(284568, J$.G(5296753, J$.R(5296745, 'constants', constants, false, true), 'S_IFIFO')):
                                return J$.Rt(5296737, J$.T(5296729, 'fifo', 21, false));
                            case J$.C2(284576, J$.G(5296785, J$.R(5296777, 'constants', constants, false, true), 'S_IFLNK')):
                                return J$.Rt(5296769, J$.T(5296761, 'symlink', 21, false));
                            }
                            return J$.Rt(5296801, J$.T(5296793, 'file', 21, false));
                        } catch (J$e) {
                            J$.Ex(5301473, J$e);
                        } finally {
                            if (J$.Fr(5301481))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(5301289, 'constants', constants, false, false, false);
            J$.N(5301297, 'eos', eos, false, false, false);
            J$.N(5301305, 'inherits', inherits, false, false, false);
            J$.N(5301313, 'alloc', alloc, false, false, false);
            J$.N(5301321, 'Readable', Readable, false, false, false);
            J$.N(5301329, 'Writable', Writable, false, false, false);
            J$.N(5301337, 'StringDecoder', StringDecoder, false, false, false);
            J$.N(5301345, 'headers', headers, false, false, false);
            J$.N(5301353, 'DMODE', DMODE, false, false, false);
            J$.N(5301361, 'FMODE', FMODE, false, false, false);
            J$.N(5301369, 'END_OF_TAR', END_OF_TAR, false, false, false);
            J$.N(5301377, 'noop', noop, false, false, false);
            J$.N(5301385, 'overflow', overflow, false, false, false);
            modeToType = J$.N(5301401, 'modeToType', J$.T(5301393, modeToType, 12, false), true, false, false);
            J$.N(5301409, 'Sink', Sink, false, false, false);
            J$.N(5301417, 'LinkSink', LinkSink, false, false, false);
            J$.N(5301425, 'Void', Void, false, false, false);
            J$.N(5301433, 'Pack', Pack, false, false, false);
            var constants = J$.W(5296081, 'constants', J$.F(5296073, J$.I(typeof require === 'undefined' ? require = J$.R(5296057, 'require', undefined, true, true) : require = J$.R(5296057, 'require', require, true, true)), false)(J$.T(5296065, 'fs-constants', 21, false)), constants, false, true);
            var eos = J$.W(5296113, 'eos', J$.F(5296105, J$.I(typeof require === 'undefined' ? require = J$.R(5296089, 'require', undefined, true, true) : require = J$.R(5296089, 'require', require, true, true)), false)(J$.T(5296097, 'end-of-stream', 21, false)), eos, false, true);
            var inherits = J$.W(5296145, 'inherits', J$.F(5296137, J$.I(typeof require === 'undefined' ? require = J$.R(5296121, 'require', undefined, true, true) : require = J$.R(5296121, 'require', require, true, true)), false)(J$.T(5296129, 'inherits', 21, false)), inherits, false, true);
            var alloc = J$.W(5296169, 'alloc', J$.G(5296161, J$.I(typeof Buffer === 'undefined' ? Buffer = J$.R(5296153, 'Buffer', undefined, true, true) : Buffer = J$.R(5296153, 'Buffer', Buffer, true, true)), 'alloc'), alloc, false, true);
            var Readable = J$.W(5296209, 'Readable', J$.G(5296201, J$.F(5296193, J$.I(typeof require === 'undefined' ? require = J$.R(5296177, 'require', undefined, true, true) : require = J$.R(5296177, 'require', require, true, true)), false)(J$.T(5296185, 'readable-stream', 21, false)), 'Readable'), Readable, false, true);
            var Writable = J$.W(5296249, 'Writable', J$.G(5296241, J$.F(5296233, J$.I(typeof require === 'undefined' ? require = J$.R(5296217, 'require', undefined, true, true) : require = J$.R(5296217, 'require', require, true, true)), false)(J$.T(5296225, 'readable-stream', 21, false)), 'Writable'), Writable, false, true);
            var StringDecoder = J$.W(5296289, 'StringDecoder', J$.G(5296281, J$.F(5296273, J$.I(typeof require === 'undefined' ? require = J$.R(5296257, 'require', undefined, true, true) : require = J$.R(5296257, 'require', require, true, true)), false)(J$.T(5296265, 'string_decoder', 21, false)), 'StringDecoder'), StringDecoder, false, true);
            var headers = J$.W(5296321, 'headers', J$.F(5296313, J$.I(typeof require === 'undefined' ? require = J$.R(5296297, 'require', undefined, true, true) : require = J$.R(5296297, 'require', require, true, true)), false)(J$.T(5296305, './headers', 21, false)), headers, false, true);
            var DMODE = J$.W(5296361, 'DMODE', J$.F(5296353, J$.I(typeof parseInt === 'undefined' ? parseInt = J$.R(5296329, 'parseInt', undefined, true, true) : parseInt = J$.R(5296329, 'parseInt', parseInt, true, true)), false)(J$.T(5296337, '755', 21, false), J$.T(5296345, 8, 22, false)), DMODE, false, true);
            var FMODE = J$.W(5296401, 'FMODE', J$.F(5296393, J$.I(typeof parseInt === 'undefined' ? parseInt = J$.R(5296369, 'parseInt', undefined, true, true) : parseInt = J$.R(5296369, 'parseInt', parseInt, true, true)), false)(J$.T(5296377, '644', 21, false), J$.T(5296385, 8, 22, false)), FMODE, false, true);
            var END_OF_TAR = J$.W(5296433, 'END_OF_TAR', J$.F(5296425, J$.R(5296409, 'alloc', alloc, false, true), false)(J$.T(5296417, 1024, 22, false)), END_OF_TAR, false, true);
            var noop = J$.W(5296465, 'noop', J$.T(5296457, function () {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(5296441, arguments.callee, this, arguments);
                            arguments = J$.N(5296449, 'arguments', arguments, true, false, false);
                        } catch (J$e) {
                            J$.Ex(5301441, J$e);
                        } finally {
                            if (J$.Fr(5301449))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), noop, false, true);
            var overflow = J$.W(5296601, 'overflow', J$.T(5296593, function (self, size) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(5296561, arguments.callee, this, arguments);
                            arguments = J$.N(5296569, 'arguments', arguments, true, false, false);
                            self = J$.N(5296577, 'self', self, true, false, false);
                            size = J$.N(5296585, 'size', size, true, false, false);
                            size = J$.W(5296489, 'size', J$.B(601162, '&', J$.R(5296481, 'size', size, false, false), J$.T(5296473, 511, 22, false)), size, false, false);
                            if (J$.C(284528, J$.R(5296497, 'size', size, false, false)))
                                J$.M(5296553, J$.R(5296505, 'self', self, false, false), 'push', false)(J$.M(5296545, J$.R(5296513, 'END_OF_TAR', END_OF_TAR, false, true), 'slice', false)(J$.T(5296521, 0, 22, false), J$.B(601170, '-', J$.T(5296529, 512, 22, false), J$.R(5296537, 'size', size, false, false))));
                        } catch (J$e) {
                            J$.Ex(5301457, J$e);
                        } finally {
                            if (J$.Fr(5301465))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), overflow, false, true);
            var Sink = J$.W(5296961, 'Sink', J$.T(5296953, function (to) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(5296929, arguments.callee, this, arguments);
                            arguments = J$.N(5296937, 'arguments', arguments, true, false, false);
                            to = J$.N(5296945, 'to', to, true, false, false);
                            J$.M(5296849, J$.R(5296833, 'Writable', Writable, false, true), 'call', false)(J$.R(5296841, 'this', this, false, false));
                            J$.P(5296873, J$.R(5296857, 'this', this, false, false), 'written', J$.T(5296865, 0, 22, false));
                            J$.P(5296897, J$.R(5296881, 'this', this, false, false), '_to', J$.R(5296889, 'to', to, false, false));
                            J$.P(5296921, J$.R(5296905, 'this', this, false, false), '_destroyed', J$.T(5296913, false, 23, false));
                        } catch (J$e) {
                            J$.Ex(5301489, J$e);
                        } finally {
                            if (J$.Fr(5301497))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), Sink, false, true);
            J$.F(5296993, J$.R(5296969, 'inherits', inherits, false, true), false)(J$.R(5296977, 'Sink', Sink, false, true), J$.R(5296985, 'Writable', Writable, false, true));
            J$.P(5297185, J$.G(5297009, J$.R(5297001, 'Sink', Sink, false, true), 'prototype'), '_write', J$.T(5297177, function (data, enc, cb) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(5297137, arguments.callee, this, arguments);
                            arguments = J$.N(5297145, 'arguments', arguments, true, false, false);
                            data = J$.N(5297153, 'data', data, true, false, false);
                            enc = J$.N(5297161, 'enc', enc, true, false, false);
                            cb = J$.N(5297169, 'cb', cb, true, false, false);
                            J$.A(5297041, J$.R(5297017, 'this', this, false, false), 'written', '+')(J$.G(5297033, J$.R(5297025, 'data', data, false, false), 'length'));
                            if (J$.C(284584, J$.M(5297073, J$.G(5297057, J$.R(5297049, 'this', this, false, false), '_to'), 'push', false)(J$.R(5297065, 'data', data, false, false))))
                                return J$.Rt(5297097, J$.F(5297089, J$.R(5297081, 'cb', cb, false, false), false)());
                            J$.P(5297129, J$.G(5297113, J$.R(5297105, 'this', this, false, false), '_to'), '_drain', J$.R(5297121, 'cb', cb, false, false));
                        } catch (J$e) {
                            J$.Ex(5301505, J$e);
                        } finally {
                            if (J$.Fr(5301513))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(5297305, J$.G(5297201, J$.R(5297193, 'Sink', Sink, false, true), 'prototype'), 'destroy', J$.T(5297297, function () {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(5297281, arguments.callee, this, arguments);
                            arguments = J$.N(5297289, 'arguments', arguments, true, false, false);
                            if (J$.C(284592, J$.G(5297217, J$.R(5297209, 'this', this, false, false), '_destroyed')))
                                return J$.Rt(5297225, undefined);
                            J$.P(5297249, J$.R(5297233, 'this', this, false, false), '_destroyed', J$.T(5297241, true, 23, false));
                            J$.M(5297273, J$.R(5297257, 'this', this, false, false), 'emit', false)(J$.T(5297265, 'close', 21, false));
                        } catch (J$e) {
                            J$.Ex(5301521, J$e);
                        } finally {
                            if (J$.Fr(5301529))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var LinkSink = J$.W(5297449, 'LinkSink', J$.T(5297441, function () {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(5297425, arguments.callee, this, arguments);
                            arguments = J$.N(5297433, 'arguments', arguments, true, false, false);
                            J$.M(5297329, J$.R(5297313, 'Writable', Writable, false, true), 'call', false)(J$.R(5297321, 'this', this, false, false));
                            J$.P(5297353, J$.R(5297337, 'this', this, false, false), 'linkname', J$.T(5297345, '', 21, false));
                            J$.P(5297393, J$.R(5297361, 'this', this, false, false), '_decoder', J$.F(5297385, J$.R(5297369, 'StringDecoder', StringDecoder, false, true), true)(J$.T(5297377, 'utf-8', 21, false)));
                            J$.P(5297417, J$.R(5297401, 'this', this, false, false), '_destroyed', J$.T(5297409, false, 23, false));
                        } catch (J$e) {
                            J$.Ex(5301537, J$e);
                        } finally {
                            if (J$.Fr(5301545))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), LinkSink, false, true);
            J$.F(5297481, J$.R(5297457, 'inherits', inherits, false, true), false)(J$.R(5297465, 'LinkSink', LinkSink, false, true), J$.R(5297473, 'Writable', Writable, false, true));
            J$.P(5297617, J$.G(5297497, J$.R(5297489, 'LinkSink', LinkSink, false, true), 'prototype'), '_write', J$.T(5297609, function (data, enc, cb) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(5297569, arguments.callee, this, arguments);
                            arguments = J$.N(5297577, 'arguments', arguments, true, false, false);
                            data = J$.N(5297585, 'data', data, true, false, false);
                            enc = J$.N(5297593, 'enc', enc, true, false, false);
                            cb = J$.N(5297601, 'cb', cb, true, false, false);
                            J$.A(5297545, J$.R(5297505, 'this', this, false, false), 'linkname', '+')(J$.M(5297537, J$.G(5297521, J$.R(5297513, 'this', this, false, false), '_decoder'), 'write', false)(J$.R(5297529, 'data', data, false, false)));
                            J$.F(5297561, J$.R(5297553, 'cb', cb, false, false), false)();
                        } catch (J$e) {
                            J$.Ex(5301553, J$e);
                        } finally {
                            if (J$.Fr(5301561))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(5297737, J$.G(5297633, J$.R(5297625, 'LinkSink', LinkSink, false, true), 'prototype'), 'destroy', J$.T(5297729, function () {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(5297713, arguments.callee, this, arguments);
                            arguments = J$.N(5297721, 'arguments', arguments, true, false, false);
                            if (J$.C(284600, J$.G(5297649, J$.R(5297641, 'this', this, false, false), '_destroyed')))
                                return J$.Rt(5297657, undefined);
                            J$.P(5297681, J$.R(5297665, 'this', this, false, false), '_destroyed', J$.T(5297673, true, 23, false));
                            J$.M(5297705, J$.R(5297689, 'this', this, false, false), 'emit', false)(J$.T(5297697, 'close', 21, false));
                        } catch (J$e) {
                            J$.Ex(5301569, J$e);
                        } finally {
                            if (J$.Fr(5301577))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var Void = J$.W(5297817, 'Void', J$.T(5297809, function () {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(5297793, arguments.callee, this, arguments);
                            arguments = J$.N(5297801, 'arguments', arguments, true, false, false);
                            J$.M(5297761, J$.R(5297745, 'Writable', Writable, false, true), 'call', false)(J$.R(5297753, 'this', this, false, false));
                            J$.P(5297785, J$.R(5297769, 'this', this, false, false), '_destroyed', J$.T(5297777, false, 23, false));
                        } catch (J$e) {
                            J$.Ex(5301585, J$e);
                        } finally {
                            if (J$.Fr(5301593))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), Void, false, true);
            J$.F(5297849, J$.R(5297825, 'inherits', inherits, false, true), false)(J$.R(5297833, 'Void', Void, false, true), J$.R(5297841, 'Writable', Writable, false, true));
            J$.P(5297961, J$.G(5297865, J$.R(5297857, 'Void', Void, false, true), 'prototype'), '_write', J$.T(5297953, function (data, enc, cb) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(5297913, arguments.callee, this, arguments);
                            arguments = J$.N(5297921, 'arguments', arguments, true, false, false);
                            data = J$.N(5297929, 'data', data, true, false, false);
                            enc = J$.N(5297937, 'enc', enc, true, false, false);
                            cb = J$.N(5297945, 'cb', cb, true, false, false);
                            J$.F(5297905, J$.R(5297873, 'cb', cb, false, false), false)(J$.F(5297897, J$.I(typeof Error === 'undefined' ? Error = J$.R(5297881, 'Error', undefined, true, true) : Error = J$.R(5297881, 'Error', Error, true, true)), true)(J$.T(5297889, 'No body allowed for this entry', 21, false)));
                        } catch (J$e) {
                            J$.Ex(5301601, J$e);
                        } finally {
                            if (J$.Fr(5301609))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(5298081, J$.G(5297977, J$.R(5297969, 'Void', Void, false, true), 'prototype'), 'destroy', J$.T(5298073, function () {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(5298057, arguments.callee, this, arguments);
                            arguments = J$.N(5298065, 'arguments', arguments, true, false, false);
                            if (J$.C(284608, J$.G(5297993, J$.R(5297985, 'this', this, false, false), '_destroyed')))
                                return J$.Rt(5298001, undefined);
                            J$.P(5298025, J$.R(5298009, 'this', this, false, false), '_destroyed', J$.T(5298017, true, 23, false));
                            J$.M(5298049, J$.R(5298033, 'this', this, false, false), 'emit', false)(J$.T(5298041, 'close', 21, false));
                        } catch (J$e) {
                            J$.Ex(5301617, J$e);
                        } finally {
                            if (J$.Fr(5301625))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var Pack = J$.W(5298321, 'Pack', J$.T(5298313, function (opts) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(5298289, arguments.callee, this, arguments);
                            arguments = J$.N(5298297, 'arguments', arguments, true, false, false);
                            opts = J$.N(5298305, 'opts', opts, true, false, false);
                            if (J$.C(284616, J$.U(601194, '!', J$.B(601186, 'instanceof', J$.R(5298089, 'this', this, false, false), J$.R(5298097, 'Pack', Pack, false, true)))))
                                return J$.Rt(5298129, J$.F(5298121, J$.R(5298105, 'Pack', Pack, false, true), true)(J$.R(5298113, 'opts', opts, false, false)));
                            J$.M(5298161, J$.R(5298137, 'Readable', Readable, false, true), 'call', false)(J$.R(5298145, 'this', this, false, false), J$.R(5298153, 'opts', opts, false, false));
                            J$.P(5298185, J$.R(5298169, 'this', this, false, false), '_drain', J$.R(5298177, 'noop', noop, false, true));
                            J$.P(5298209, J$.R(5298193, 'this', this, false, false), '_finalized', J$.T(5298201, false, 23, false));
                            J$.P(5298233, J$.R(5298217, 'this', this, false, false), '_finalizing', J$.T(5298225, false, 23, false));
                            J$.P(5298257, J$.R(5298241, 'this', this, false, false), '_destroyed', J$.T(5298249, false, 23, false));
                            J$.P(5298281, J$.R(5298265, 'this', this, false, false), '_stream', J$.T(5298273, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(5301633, J$e);
                        } finally {
                            if (J$.Fr(5301641))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), Pack, false, true);
            J$.F(5298353, J$.R(5298329, 'inherits', inherits, false, true), false)(J$.R(5298337, 'Pack', Pack, false, true), J$.R(5298345, 'Readable', Readable, false, true));
            J$.P(5299961, J$.G(5298369, J$.R(5298361, 'Pack', Pack, false, true), 'prototype'), 'entry', J$.T(5299953, function (header, buffer, callback) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(5299889, arguments.callee, this, arguments);
                            arguments = J$.N(5299897, 'arguments', arguments, true, false, false);
                            header = J$.N(5299905, 'header', header, true, false, false);
                            buffer = J$.N(5299913, 'buffer', buffer, true, false, false);
                            callback = J$.N(5299921, 'callback', callback, true, false, false);
                            J$.N(5299929, 'self', self, false, false, false);
                            J$.N(5299937, 'linkSink', linkSink, false, false, false);
                            J$.N(5299945, 'sink', sink, false, false, false);
                            if (J$.C(284624, J$.G(5298385, J$.R(5298377, 'this', this, false, false), '_stream')))
                                throw J$.F(5298409, J$.I(typeof Error === 'undefined' ? Error = J$.R(5298393, 'Error', undefined, true, true) : Error = J$.R(5298393, 'Error', Error, true, true)), true)(J$.T(5298401, 'already piping an entry', 21, false));
                            if (J$.C(284640, J$.C(284632, J$.G(5298425, J$.R(5298417, 'this', this, false, false), '_finalized')) ? J$._() : J$.G(5298441, J$.R(5298433, 'this', this, false, false), '_destroyed')))
                                return J$.Rt(5298449, undefined);
                            if (J$.C(284648, J$.B(601210, '===', J$.U(601202, 'typeof', J$.R(5298457, 'buffer', buffer, false, false)), J$.T(5298465, 'function', 21, false)))) {
                                callback = J$.W(5298481, 'callback', J$.R(5298473, 'buffer', buffer, false, false), callback, false, false);
                                buffer = J$.W(5298497, 'buffer', J$.T(5298489, null, 25, false), buffer, false, false);
                            }
                            if (J$.C(284656, J$.U(601218, '!', J$.R(5298505, 'callback', callback, false, false))))
                                callback = J$.W(5298521, 'callback', J$.R(5298513, 'noop', noop, false, true), callback, false, false);
                            var self = J$.W(5298537, 'self', J$.R(5298529, 'this', this, false, false), self, false, false);
                            if (J$.C(284672, J$.C(284664, J$.U(601226, '!', J$.G(5298553, J$.R(5298545, 'header', header, false, false), 'size'))) ? J$._() : J$.B(601234, '===', J$.G(5298569, J$.R(5298561, 'header', header, false, false), 'type'), J$.T(5298577, 'symlink', 21, false))))
                                J$.P(5298601, J$.R(5298585, 'header', header, false, false), 'size', J$.T(5298593, 0, 22, false));
                            if (J$.C(284680, J$.U(601242, '!', J$.G(5298617, J$.R(5298609, 'header', header, false, false), 'type'))))
                                J$.P(5298665, J$.R(5298625, 'header', header, false, false), 'type', J$.F(5298657, J$.R(5298633, 'modeToType', modeToType, false, true), false)(J$.G(5298649, J$.R(5298641, 'header', header, false, false), 'mode')));
                            if (J$.C(284696, J$.U(601250, '!', J$.G(5298681, J$.R(5298673, 'header', header, false, false), 'mode'))))
                                J$.P(5298737, J$.R(5298689, 'header', header, false, false), 'mode', J$.C(284688, J$.B(601258, '===', J$.G(5298705, J$.R(5298697, 'header', header, false, false), 'type'), J$.T(5298713, 'directory', 21, false))) ? J$.R(5298721, 'DMODE', DMODE, false, true) : J$.R(5298729, 'FMODE', FMODE, false, true));
                            if (J$.C(284704, J$.U(601266, '!', J$.G(5298753, J$.R(5298745, 'header', header, false, false), 'uid'))))
                                J$.P(5298777, J$.R(5298761, 'header', header, false, false), 'uid', J$.T(5298769, 0, 22, false));
                            if (J$.C(284712, J$.U(601274, '!', J$.G(5298793, J$.R(5298785, 'header', header, false, false), 'gid'))))
                                J$.P(5298817, J$.R(5298801, 'header', header, false, false), 'gid', J$.T(5298809, 0, 22, false));
                            if (J$.C(284720, J$.U(601282, '!', J$.G(5298833, J$.R(5298825, 'header', header, false, false), 'mtime'))))
                                J$.P(5298865, J$.R(5298841, 'header', header, false, false), 'mtime', J$.F(5298857, J$.I(typeof Date === 'undefined' ? Date = J$.R(5298849, 'Date', undefined, true, true) : Date = J$.R(5298849, 'Date', Date, true, true)), true)());
                            if (J$.C(284728, J$.B(601298, '===', J$.U(601290, 'typeof', J$.R(5298873, 'buffer', buffer, false, false)), J$.T(5298881, 'string', 21, false))))
                                buffer = J$.W(5298913, 'buffer', J$.M(5298905, J$.I(typeof Buffer === 'undefined' ? Buffer = J$.R(5298889, 'Buffer', undefined, true, true) : Buffer = J$.R(5298889, 'Buffer', Buffer, true, true)), 'from', false)(J$.R(5298897, 'buffer', buffer, false, false)), buffer, false, false);
                            if (J$.C(284736, J$.M(5298937, J$.I(typeof Buffer === 'undefined' ? Buffer = J$.R(5298921, 'Buffer', undefined, true, true) : Buffer = J$.R(5298921, 'Buffer', Buffer, true, true)), 'isBuffer', false)(J$.R(5298929, 'buffer', buffer, false, false)))) {
                                J$.P(5298969, J$.R(5298945, 'header', header, false, false), 'size', J$.G(5298961, J$.R(5298953, 'buffer', buffer, false, false), 'length'));
                                J$.M(5298993, J$.R(5298977, 'this', this, false, false), '_encode', false)(J$.R(5298985, 'header', header, false, false));
                                J$.M(5299017, J$.R(5299001, 'this', this, false, false), 'push', false)(J$.R(5299009, 'buffer', buffer, false, false));
                                J$.F(5299057, J$.R(5299025, 'overflow', overflow, false, true), false)(J$.R(5299033, 'self', self, false, false), J$.G(5299049, J$.R(5299041, 'header', header, false, false), 'size'));
                                J$.M(5299081, J$.I(typeof process === 'undefined' ? process = J$.R(5299065, 'process', undefined, true, true) : process = J$.R(5299065, 'process', process, true, true)), 'nextTick', false)(J$.R(5299073, 'callback', callback, false, false));
                                return J$.Rt(5299105, J$.F(5299097, J$.R(5299089, 'Void', Void, false, true), true)());
                            }
                            if (J$.C(284760, J$.C(284744, J$.B(601306, '===', J$.G(5299121, J$.R(5299113, 'header', header, false, false), 'type'), J$.T(5299129, 'symlink', 21, false))) ? J$.U(601314, '!', J$.G(5299145, J$.R(5299137, 'header', header, false, false), 'linkname')) : J$._())) {
                                var linkSink = J$.W(5299169, 'linkSink', J$.F(5299161, J$.R(5299153, 'LinkSink', LinkSink, false, true), true)(), linkSink, false, false);
                                J$.F(5299353, J$.R(5299177, 'eos', eos, false, true), false)(J$.R(5299185, 'linkSink', linkSink, false, false), J$.T(5299345, function (err) {
                                    jalangiLabel13:
                                        while (true) {
                                            try {
                                                J$.Fe(5299321, arguments.callee, this, arguments);
                                                arguments = J$.N(5299329, 'arguments', arguments, true, false, false);
                                                err = J$.N(5299337, 'err', err, true, false, false);
                                                if (J$.C(284752, J$.R(5299193, 'err', err, false, false))) {
                                                    J$.M(5299209, J$.R(5299201, 'self', self, false, false), 'destroy', false)();
                                                    return J$.Rt(5299241, J$.F(5299233, J$.R(5299217, 'callback', callback, false, false), false)(J$.R(5299225, 'err', err, false, false)));
                                                }
                                                J$.P(5299273, J$.R(5299249, 'header', header, false, false), 'linkname', J$.G(5299265, J$.R(5299257, 'linkSink', linkSink, false, false), 'linkname'));
                                                J$.M(5299297, J$.R(5299281, 'self', self, false, false), '_encode', false)(J$.R(5299289, 'header', header, false, false));
                                                J$.F(5299313, J$.R(5299305, 'callback', callback, false, false), false)();
                                            } catch (J$e) {
                                                J$.Ex(5301649, J$e);
                                            } finally {
                                                if (J$.Fr(5301657))
                                                    continue jalangiLabel13;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                                return J$.Rt(5299369, J$.R(5299361, 'linkSink', linkSink, false, false));
                            }
                            J$.M(5299393, J$.R(5299377, 'this', this, false, false), '_encode', false)(J$.R(5299385, 'header', header, false, false));
                            if (J$.C(284776, J$.C(284768, J$.B(601322, '!==', J$.G(5299409, J$.R(5299401, 'header', header, false, false), 'type'), J$.T(5299417, 'file', 21, false))) ? J$.B(601330, '!==', J$.G(5299433, J$.R(5299425, 'header', header, false, false), 'type'), J$.T(5299441, 'contiguous-file', 21, false)) : J$._())) {
                                J$.M(5299465, J$.I(typeof process === 'undefined' ? process = J$.R(5299449, 'process', undefined, true, true) : process = J$.R(5299449, 'process', process, true, true)), 'nextTick', false)(J$.R(5299457, 'callback', callback, false, false));
                                return J$.Rt(5299489, J$.F(5299481, J$.R(5299473, 'Void', Void, false, true), true)());
                            }
                            var sink = J$.W(5299521, 'sink', J$.F(5299513, J$.R(5299497, 'Sink', Sink, false, true), true)(J$.R(5299505, 'this', this, false, false)), sink, false, false);
                            J$.P(5299545, J$.R(5299529, 'this', this, false, false), '_stream', J$.R(5299537, 'sink', sink, false, false));
                            J$.F(5299865, J$.R(5299553, 'eos', eos, false, true), false)(J$.R(5299561, 'sink', sink, false, false), J$.T(5299857, function (err) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(5299833, arguments.callee, this, arguments);
                                            arguments = J$.N(5299841, 'arguments', arguments, true, false, false);
                                            err = J$.N(5299849, 'err', err, true, false, false);
                                            J$.P(5299585, J$.R(5299569, 'self', self, false, false), '_stream', J$.T(5299577, null, 25, false));
                                            if (J$.C(284784, J$.R(5299593, 'err', err, false, false))) {
                                                J$.M(5299609, J$.R(5299601, 'self', self, false, false), 'destroy', false)();
                                                return J$.Rt(5299641, J$.F(5299633, J$.R(5299617, 'callback', callback, false, false), false)(J$.R(5299625, 'err', err, false, false)));
                                            }
                                            if (J$.C(284792, J$.B(601338, '!==', J$.G(5299657, J$.R(5299649, 'sink', sink, false, false), 'written'), J$.G(5299673, J$.R(5299665, 'header', header, false, false), 'size')))) {
                                                J$.M(5299689, J$.R(5299681, 'self', self, false, false), 'destroy', false)();
                                                return J$.Rt(5299737, J$.F(5299729, J$.R(5299697, 'callback', callback, false, false), false)(J$.F(5299721, J$.I(typeof Error === 'undefined' ? Error = J$.R(5299705, 'Error', undefined, true, true) : Error = J$.R(5299705, 'Error', Error, true, true)), true)(J$.T(5299713, 'size mismatch', 21, false))));
                                            }
                                            J$.F(5299777, J$.R(5299745, 'overflow', overflow, false, true), false)(J$.R(5299753, 'self', self, false, false), J$.G(5299769, J$.R(5299761, 'header', header, false, false), 'size'));
                                            if (J$.C(284800, J$.G(5299793, J$.R(5299785, 'self', self, false, false), '_finalizing')))
                                                J$.M(5299809, J$.R(5299801, 'self', self, false, false), 'finalize', false)();
                                            J$.F(5299825, J$.R(5299817, 'callback', callback, false, false), false)();
                                        } catch (J$e) {
                                            J$.Ex(5301665, J$e);
                                        } finally {
                                            if (J$.Fr(5301673))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            return J$.Rt(5299881, J$.R(5299873, 'sink', sink, false, false));
                        } catch (J$e) {
                            J$.Ex(5301681, J$e);
                        } finally {
                            if (J$.Fr(5301689))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(5300153, J$.G(5299977, J$.R(5299969, 'Pack', Pack, false, true), 'prototype'), 'finalize', J$.T(5300145, function () {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(5300129, arguments.callee, this, arguments);
                            arguments = J$.N(5300137, 'arguments', arguments, true, false, false);
                            if (J$.C(284808, J$.G(5299993, J$.R(5299985, 'this', this, false, false), '_stream'))) {
                                J$.P(5300017, J$.R(5300001, 'this', this, false, false), '_finalizing', J$.T(5300009, true, 23, false));
                                return J$.Rt(5300025, undefined);
                            }
                            if (J$.C(284816, J$.G(5300041, J$.R(5300033, 'this', this, false, false), '_finalized')))
                                return J$.Rt(5300049, undefined);
                            J$.P(5300073, J$.R(5300057, 'this', this, false, false), '_finalized', J$.T(5300065, true, 23, false));
                            J$.M(5300097, J$.R(5300081, 'this', this, false, false), 'push', false)(J$.R(5300089, 'END_OF_TAR', END_OF_TAR, false, true));
                            J$.M(5300121, J$.R(5300105, 'this', this, false, false), 'push', false)(J$.T(5300113, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(5301697, J$e);
                        } finally {
                            if (J$.Fr(5301705))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(5300385, J$.G(5300169, J$.R(5300161, 'Pack', Pack, false, true), 'prototype'), 'destroy', J$.T(5300377, function (err) {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(5300353, arguments.callee, this, arguments);
                            arguments = J$.N(5300361, 'arguments', arguments, true, false, false);
                            err = J$.N(5300369, 'err', err, true, false, false);
                            if (J$.C(284824, J$.G(5300185, J$.R(5300177, 'this', this, false, false), '_destroyed')))
                                return J$.Rt(5300193, undefined);
                            J$.P(5300217, J$.R(5300201, 'this', this, false, false), '_destroyed', J$.T(5300209, true, 23, false));
                            if (J$.C(284832, J$.R(5300225, 'err', err, false, false)))
                                J$.M(5300257, J$.R(5300233, 'this', this, false, false), 'emit', false)(J$.T(5300241, 'error', 21, false), J$.R(5300249, 'err', err, false, false));
                            J$.M(5300281, J$.R(5300265, 'this', this, false, false), 'emit', false)(J$.T(5300273, 'close', 21, false));
                            if (J$.C(284848, J$.C(284840, J$.G(5300297, J$.R(5300289, 'this', this, false, false), '_stream')) ? J$.G(5300321, J$.G(5300313, J$.R(5300305, 'this', this, false, false), '_stream'), 'destroy') : J$._()))
                                J$.M(5300345, J$.G(5300337, J$.R(5300329, 'this', this, false, false), '_stream'), 'destroy', false)();
                        } catch (J$e) {
                            J$.Ex(5301713, J$e);
                        } finally {
                            if (J$.Fr(5301721))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(5300561, J$.G(5300401, J$.R(5300393, 'Pack', Pack, false, true), 'prototype'), '_encode', J$.T(5300553, function (header) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(5300521, arguments.callee, this, arguments);
                            arguments = J$.N(5300529, 'arguments', arguments, true, false, false);
                            header = J$.N(5300537, 'header', header, true, false, false);
                            J$.N(5300545, 'buf', buf, false, false, false);
                            if (J$.C(284864, J$.U(601346, '!', J$.G(5300417, J$.R(5300409, 'header', header, false, false), 'pax')))) {
                                var buf = J$.W(5300449, 'buf', J$.M(5300441, J$.R(5300425, 'headers', headers, false, true), 'encode', false)(J$.R(5300433, 'header', header, false, false)), buf, false, false);
                                if (J$.C(284856, J$.R(5300457, 'buf', buf, false, false))) {
                                    J$.M(5300481, J$.R(5300465, 'this', this, false, false), 'push', false)(J$.R(5300473, 'buf', buf, false, false));
                                    return J$.Rt(5300489, undefined);
                                }
                            }
                            J$.M(5300513, J$.R(5300497, 'this', this, false, false), '_encodePax', false)(J$.R(5300505, 'header', header, false, false));
                        } catch (J$e) {
                            J$.Ex(5301729, J$e);
                        } finally {
                            if (J$.Fr(5301737))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(5301121, J$.G(5300577, J$.R(5300569, 'Pack', Pack, false, true), 'prototype'), '_encodePax', J$.T(5301113, function (header) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(5301073, arguments.callee, this, arguments);
                            arguments = J$.N(5301081, 'arguments', arguments, true, false, false);
                            header = J$.N(5301089, 'header', header, true, false, false);
                            J$.N(5301097, 'paxHeader', paxHeader, false, false, false);
                            J$.N(5301105, 'newHeader', newHeader, false, false, false);
                            var paxHeader = J$.W(5300657, 'paxHeader', J$.M(5300649, J$.R(5300585, 'headers', headers, false, true), 'encodePax', false)(J$.T(5300641, {
                                name: J$.G(5300601, J$.R(5300593, 'header', header, false, false), 'name'),
                                linkname: J$.G(5300617, J$.R(5300609, 'header', header, false, false), 'linkname'),
                                pax: J$.G(5300633, J$.R(5300625, 'header', header, false, false), 'pax')
                            }, 11, false)), paxHeader, false, false);
                            var newHeader = J$.W(5300857, 'newHeader', J$.T(5300849, {
                                name: J$.T(5300665, 'PaxHeader', 21, false),
                                mode: J$.G(5300681, J$.R(5300673, 'header', header, false, false), 'mode'),
                                uid: J$.G(5300697, J$.R(5300689, 'header', header, false, false), 'uid'),
                                gid: J$.G(5300713, J$.R(5300705, 'header', header, false, false), 'gid'),
                                size: J$.G(5300729, J$.R(5300721, 'paxHeader', paxHeader, false, false), 'length'),
                                mtime: J$.G(5300745, J$.R(5300737, 'header', header, false, false), 'mtime'),
                                type: J$.T(5300753, 'pax-header', 21, false),
                                linkname: J$.C(284872, J$.G(5300769, J$.R(5300761, 'header', header, false, false), 'linkname')) ? J$.T(5300777, 'PaxHeader', 21, false) : J$._(),
                                uname: J$.G(5300793, J$.R(5300785, 'header', header, false, false), 'uname'),
                                gname: J$.G(5300809, J$.R(5300801, 'header', header, false, false), 'gname'),
                                devmajor: J$.G(5300825, J$.R(5300817, 'header', header, false, false), 'devmajor'),
                                devminor: J$.G(5300841, J$.R(5300833, 'header', header, false, false), 'devminor')
                            }, 11, false), newHeader, false, false);
                            J$.M(5300897, J$.R(5300865, 'this', this, false, false), 'push', false)(J$.M(5300889, J$.R(5300873, 'headers', headers, false, true), 'encode', false)(J$.R(5300881, 'newHeader', newHeader, false, false)));
                            J$.M(5300921, J$.R(5300905, 'this', this, false, false), 'push', false)(J$.R(5300913, 'paxHeader', paxHeader, false, false));
                            J$.F(5300961, J$.R(5300929, 'overflow', overflow, false, true), false)(J$.R(5300937, 'this', this, false, false), J$.G(5300953, J$.R(5300945, 'paxHeader', paxHeader, false, false), 'length'));
                            J$.P(5300993, J$.R(5300969, 'newHeader', newHeader, false, false), 'size', J$.G(5300985, J$.R(5300977, 'header', header, false, false), 'size'));
                            J$.P(5301025, J$.R(5301001, 'newHeader', newHeader, false, false), 'type', J$.G(5301017, J$.R(5301009, 'header', header, false, false), 'type'));
                            J$.M(5301065, J$.R(5301033, 'this', this, false, false), 'push', false)(J$.M(5301057, J$.R(5301041, 'headers', headers, false, true), 'encode', false)(J$.R(5301049, 'newHeader', newHeader, false, false)));
                        } catch (J$e) {
                            J$.Ex(5301745, J$e);
                        } finally {
                            if (J$.Fr(5301753))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(5301249, J$.G(5301137, J$.R(5301129, 'Pack', Pack, false, true), 'prototype'), '_read', J$.T(5301241, function (n) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(5301209, arguments.callee, this, arguments);
                            arguments = J$.N(5301217, 'arguments', arguments, true, false, false);
                            n = J$.N(5301225, 'n', n, true, false, false);
                            J$.N(5301233, 'drain', drain, false, false, false);
                            var drain = J$.W(5301161, 'drain', J$.G(5301153, J$.R(5301145, 'this', this, false, false), '_drain'), drain, false, false);
                            J$.P(5301185, J$.R(5301169, 'this', this, false, false), '_drain', J$.R(5301177, 'noop', noop, false, true));
                            J$.F(5301201, J$.R(5301193, 'drain', drain, false, false), false)();
                        } catch (J$e) {
                            J$.Ex(5301761, J$e);
                        } finally {
                            if (J$.Fr(5301769))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(5301273, J$.I(typeof module === 'undefined' ? module = J$.R(5301257, 'module', undefined, true, true) : module = J$.R(5301257, 'module', module, true, true)), 'exports', J$.R(5301265, 'Pack', Pack, false, true));
        } catch (J$e) {
            J$.Ex(5301777, J$e);
        } finally {
            if (J$.Sr(5301785))
                continue jalangiLabel21;
            else
                break jalangiLabel21;
        }
    }
// JALANGI DO NOT INSTRUMENT
