// mongodb/lib/topologies/native_topology.js
J$.noInstrEval = false;
jalangiLabel23:
    while (true) {
        try {
            J$.Se(786825, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/topologies/native_topology.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(782977, arguments.callee, this, arguments);
                            arguments = J$.N(782985, 'arguments', arguments, true, false, false);
                            obj = J$.N(782993, 'obj', obj, true, false, false);
                            if (J$.C(50520, J$.C(50480, J$.B(48954, '===', J$.U(48946, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(782713, 'Symbol', undefined, true, true) : Symbol = J$.R(782713, 'Symbol', Symbol, true, true))), J$.T(782721, 'function', 21, false))) ? J$.B(48970, '===', J$.U(48962, 'typeof', J$.G(782737, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(782729, 'Symbol', undefined, true, true) : Symbol = J$.R(782729, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(782745, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(782801, '_typeof', J$.T(782793, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(782769, arguments.callee, this, arguments);
                                                arguments = J$.N(782777, 'arguments', arguments, true, false, false);
                                                obj = J$.N(782785, 'obj', obj, true, false, false);
                                                return J$.Rt(782761, J$.U(48978, 'typeof', J$.R(782753, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(787049, J$e);
                                            } finally {
                                                if (J$.Fr(787057))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(782937, '_typeof', J$.T(782929, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(782905, arguments.callee, this, arguments);
                                                arguments = J$.N(782913, 'arguments', arguments, true, false, false);
                                                obj = J$.N(782921, 'obj', obj, true, false, false);
                                                return J$.Rt(782897, J$.C(50512, J$.C(50504, J$.C(50496, J$.C(50488, J$.R(782809, 'obj', obj, false, false)) ? J$.B(48994, '===', J$.U(48986, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(782817, 'Symbol', undefined, true, true) : Symbol = J$.R(782817, 'Symbol', Symbol, true, true))), J$.T(782825, 'function', 21, false)) : J$._()) ? J$.B(49002, '===', J$.G(782841, J$.R(782833, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(782849, 'Symbol', undefined, true, true) : Symbol = J$.R(782849, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(49010, '!==', J$.R(782857, 'obj', obj, false, false), J$.G(782873, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(782865, 'Symbol', undefined, true, true) : Symbol = J$.R(782865, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(782881, 'symbol', 21, false) : J$.U(49018, 'typeof', J$.R(782889, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(787065, J$e);
                                            } finally {
                                                if (J$.Fr(787073))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(782969, J$.F(782961, J$.R(782945, '_typeof', _typeof, false, true), false)(J$.R(782953, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(787081, J$e);
                        } finally {
                            if (J$.Fr(787089))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(783041, arguments.callee, this, arguments);
                            arguments = J$.N(783049, 'arguments', arguments, true, false, false);
                            instance = J$.N(783057, 'instance', instance, true, false, false);
                            Constructor = J$.N(783065, 'Constructor', Constructor, true, false, false);
                            if (J$.C(50528, J$.U(49034, '!', J$.B(49026, 'instanceof', J$.R(783001, 'instance', instance, false, false), J$.R(783009, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(783033, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(783017, 'TypeError', undefined, true, true) : TypeError = J$.R(783017, 'TypeError', TypeError, true, true)), true)(J$.T(783025, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(787097, J$e);
                        } finally {
                            if (J$.Fr(787105))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(783313, arguments.callee, this, arguments);
                            arguments = J$.N(783321, 'arguments', arguments, true, false, false);
                            target = J$.N(783329, 'target', target, true, false, false);
                            props = J$.N(783337, 'props', props, true, false, false);
                            J$.N(783345, 'i', i, false, false, false);
                            J$.N(783353, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(783081, 'i', J$.T(783073, 0, 22, false), i, false, false); J$.C(50552, J$.B(49042, '<', J$.R(783089, 'i', i, false, false), J$.G(783105, J$.R(783097, 'props', props, false, false), 'length'))); J$.B(49066, '-', i = J$.W(783121, 'i', J$.B(49058, '+', J$.U(49050, '+', J$.R(783113, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(783153, 'descriptor', J$.G(783145, J$.R(783129, 'props', props, false, false), J$.R(783137, 'i', i, false, false)), descriptor, false, false);
                                J$.P(783193, J$.R(783161, 'descriptor', descriptor, false, false), 'enumerable', J$.C(50536, J$.G(783177, J$.R(783169, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(783185, false, 23, false));
                                J$.P(783217, J$.R(783201, 'descriptor', descriptor, false, false), 'configurable', J$.T(783209, true, 23, false));
                                if (J$.C(50544, J$.B(49074, 'in', J$.T(783225, 'value', 21, false), J$.R(783233, 'descriptor', descriptor, false, false))))
                                    J$.P(783257, J$.R(783241, 'descriptor', descriptor, false, false), 'writable', J$.T(783249, true, 23, false));
                                J$.M(783305, J$.I(typeof Object === 'undefined' ? Object = J$.R(783265, 'Object', undefined, true, true) : Object = J$.R(783265, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(783273, 'target', target, false, false), J$.G(783289, J$.R(783281, 'descriptor', descriptor, false, false), 'key'), J$.R(783297, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(787113, J$e);
                        } finally {
                            if (J$.Fr(787121))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(783465, arguments.callee, this, arguments);
                            arguments = J$.N(783473, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(783481, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(783489, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(783497, 'staticProps', staticProps, true, false, false);
                            if (J$.C(50560, J$.R(783361, 'protoProps', protoProps, false, false)))
                                J$.F(783401, J$.R(783369, '_defineProperties', _defineProperties, false, true), false)(J$.G(783385, J$.R(783377, 'Constructor', Constructor, false, false), 'prototype'), J$.R(783393, 'protoProps', protoProps, false, false));
                            if (J$.C(50568, J$.R(783409, 'staticProps', staticProps, false, false)))
                                J$.F(783441, J$.R(783417, '_defineProperties', _defineProperties, false, true), false)(J$.R(783425, 'Constructor', Constructor, false, false), J$.R(783433, 'staticProps', staticProps, false, false));
                            return J$.Rt(783457, J$.R(783449, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(787129, J$e);
                        } finally {
                            if (J$.Fr(787137))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(783609, arguments.callee, this, arguments);
                            arguments = J$.N(783617, 'arguments', arguments, true, false, false);
                            self = J$.N(783625, 'self', self, true, false, false);
                            call = J$.N(783633, 'call', call, true, false, false);
                            if (J$.C(50592, J$.C(50584, J$.R(783505, 'call', call, false, false)) ? J$.C(50576, J$.B(49082, '===', J$.F(783529, J$.R(783513, '_typeof', _typeof, false, true), false)(J$.R(783521, 'call', call, false, false)), J$.T(783537, 'object', 21, false))) ? J$._() : J$.B(49098, '===', J$.U(49090, 'typeof', J$.R(783545, 'call', call, false, false)), J$.T(783553, 'function', 21, false)) : J$._())) {
                                return J$.Rt(783569, J$.R(783561, 'call', call, false, false));
                            }
                            return J$.Rt(783601, J$.F(783593, J$.R(783577, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(783585, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(787145, J$e);
                        } finally {
                            if (J$.Fr(787153))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(783697, arguments.callee, this, arguments);
                            arguments = J$.N(783705, 'arguments', arguments, true, false, false);
                            self = J$.N(783713, 'self', self, true, false, false);
                            if (J$.C(50600, J$.B(49106, '===', J$.R(783641, 'self', self, false, false), void J$.T(783649, 0, 22, false)))) {
                                throw J$.F(783673, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(783657, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(783657, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(783665, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(783689, J$.R(783681, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(787161, J$e);
                        } finally {
                            if (J$.Fr(787169))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(784081, arguments.callee, this, arguments);
                            arguments = J$.N(784089, 'arguments', arguments, true, false, false);
                            target = J$.N(784097, 'target', target, true, false, false);
                            property = J$.N(784105, 'property', property, true, false, false);
                            receiver = J$.N(784113, 'receiver', receiver, true, false, false);
                            if (J$.C(50632, J$.C(50608, J$.B(49122, '!==', J$.U(49114, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(783721, 'Reflect', undefined, true, true) : Reflect = J$.R(783721, 'Reflect', Reflect, true, true))), J$.T(783729, 'undefined', 21, false))) ? J$.G(783745, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(783737, 'Reflect', undefined, true, true) : Reflect = J$.R(783737, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(783769, '_get', J$.G(783761, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(783753, 'Reflect', undefined, true, true) : Reflect = J$.R(783753, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(784017, '_get', J$.T(784009, function _get(target, property, receiver) {
                                    jalangiLabel8:
                                        while (true) {
                                            try {
                                                J$.Fe(783953, arguments.callee, this, arguments);
                                                arguments = J$.N(783961, 'arguments', arguments, true, false, false);
                                                target = J$.N(783969, 'target', target, true, false, false);
                                                property = J$.N(783977, 'property', property, true, false, false);
                                                receiver = J$.N(783985, 'receiver', receiver, true, false, false);
                                                J$.N(783993, 'base', base, false, false, false);
                                                J$.N(784001, 'desc', desc, false, false, false);
                                                var base = J$.W(783809, 'base', J$.F(783801, J$.R(783777, '_superPropBase', _superPropBase, false, true), false)(J$.R(783785, 'target', target, false, false), J$.R(783793, 'property', property, false, false)), base, false, false);
                                                if (J$.C(50616, J$.U(49130, '!', J$.R(783817, 'base', base, false, false))))
                                                    return J$.Rt(783825, undefined);
                                                var desc = J$.W(783865, 'desc', J$.M(783857, J$.I(typeof Object === 'undefined' ? Object = J$.R(783833, 'Object', undefined, true, true) : Object = J$.R(783833, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(783841, 'base', base, false, false), J$.R(783849, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(50624, J$.G(783881, J$.R(783873, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(783921, J$.M(783913, J$.G(783897, J$.R(783889, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(783905, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(783945, J$.G(783937, J$.R(783929, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(787177, J$e);
                                            } finally {
                                                if (J$.Fr(787185))
                                                    continue jalangiLabel8;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(784073, J$.F(784065, J$.R(784025, '_get', _get, false, true), false)(J$.R(784033, 'target', target, false, false), J$.R(784041, 'property', property, false, false), J$.C(50640, J$.R(784049, 'receiver', receiver, false, false)) ? J$._() : J$.R(784057, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(787193, J$e);
                        } finally {
                            if (J$.Fr(787201))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(784233, arguments.callee, this, arguments);
                            arguments = J$.N(784241, 'arguments', arguments, true, false, false);
                            object = J$.N(784249, 'object', object, true, false, false);
                            property = J$.N(784257, 'property', property, true, false, false);
                            while (J$.C(50656, J$.U(49138, '!', J$.M(784161, J$.G(784137, J$.G(784129, J$.I(typeof Object === 'undefined' ? Object = J$.R(784121, 'Object', undefined, true, true) : Object = J$.R(784121, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(784145, 'object', object, false, false), J$.R(784153, 'property', property, false, false))))) {
                                object = J$.W(784193, 'object', J$.F(784185, J$.R(784169, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(784177, 'object', object, false, false)), object, false, false);
                                if (J$.C(50648, J$.B(49146, '===', J$.R(784201, 'object', object, false, false), J$.T(784209, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(784225, J$.R(784217, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(787209, J$e);
                        } finally {
                            if (J$.Fr(787217))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(784417, arguments.callee, this, arguments);
                            arguments = J$.N(784425, 'arguments', arguments, true, false, false);
                            o = J$.N(784433, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(784377, '_getPrototypeOf', J$.C(50672, J$.G(784273, J$.I(typeof Object === 'undefined' ? Object = J$.R(784265, 'Object', undefined, true, true) : Object = J$.R(784265, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(784289, J$.I(typeof Object === 'undefined' ? Object = J$.R(784281, 'Object', undefined, true, true) : Object = J$.R(784281, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(784369, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(784345, arguments.callee, this, arguments);
                                            arguments = J$.N(784353, 'arguments', arguments, true, false, false);
                                            o = J$.N(784361, 'o', o, true, false, false);
                                            return J$.Rt(784337, J$.C(50664, J$.G(784305, J$.R(784297, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(784329, J$.I(typeof Object === 'undefined' ? Object = J$.R(784313, 'Object', undefined, true, true) : Object = J$.R(784313, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(784321, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(787225, J$e);
                                        } finally {
                                            if (J$.Fr(787233))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(784409, J$.F(784401, J$.R(784385, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(784393, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(787241, J$e);
                        } finally {
                            if (J$.Fr(787249))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(784633, arguments.callee, this, arguments);
                            arguments = J$.N(784641, 'arguments', arguments, true, false, false);
                            subClass = J$.N(784649, 'subClass', subClass, true, false, false);
                            superClass = J$.N(784657, 'superClass', superClass, true, false, false);
                            if (J$.C(50688, J$.C(50680, J$.B(49162, '!==', J$.U(49154, 'typeof', J$.R(784441, 'superClass', superClass, false, false)), J$.T(784449, 'function', 21, false))) ? J$.B(49170, '!==', J$.R(784457, 'superClass', superClass, false, false), J$.T(784465, null, 25, false)) : J$._())) {
                                throw J$.F(784489, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(784473, 'TypeError', undefined, true, true) : TypeError = J$.R(784473, 'TypeError', TypeError, true, true)), true)(J$.T(784481, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(784585, J$.R(784497, 'subClass', subClass, false, false), 'prototype', J$.M(784577, J$.I(typeof Object === 'undefined' ? Object = J$.R(784505, 'Object', undefined, true, true) : Object = J$.R(784505, 'Object', Object, true, true)), 'create', false)(J$.C(50696, J$.R(784513, 'superClass', superClass, false, false)) ? J$.G(784529, J$.R(784521, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(784569, {
                                constructor: J$.T(784561, {
                                    value: J$.R(784537, 'subClass', subClass, false, false),
                                    writable: J$.T(784545, true, 23, false),
                                    configurable: J$.T(784553, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(50704, J$.R(784593, 'superClass', superClass, false, false)))
                                J$.F(784625, J$.R(784601, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(784609, 'subClass', subClass, false, false), J$.R(784617, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(787257, J$e);
                        } finally {
                            if (J$.Fr(787265))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(784809, arguments.callee, this, arguments);
                            arguments = J$.N(784817, 'arguments', arguments, true, false, false);
                            o = J$.N(784825, 'o', o, true, false, false);
                            p = J$.N(784833, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(784761, '_setPrototypeOf', J$.C(50712, J$.G(784673, J$.I(typeof Object === 'undefined' ? Object = J$.R(784665, 'Object', undefined, true, true) : Object = J$.R(784665, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(784753, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(784721, arguments.callee, this, arguments);
                                            arguments = J$.N(784729, 'arguments', arguments, true, false, false);
                                            o = J$.N(784737, 'o', o, true, false, false);
                                            p = J$.N(784745, 'p', p, true, false, false);
                                            J$.P(784697, J$.R(784681, 'o', o, false, false), '__proto__', J$.R(784689, 'p', p, false, false));
                                            return J$.Rt(784713, J$.R(784705, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(787273, J$e);
                                        } finally {
                                            if (J$.Fr(787281))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(784801, J$.F(784793, J$.R(784769, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(784777, 'o', o, false, false), J$.R(784785, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(787289, J$e);
                        } finally {
                            if (J$.Fr(787297))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(786841, '_typeof', J$.T(786833, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(786857, '_classCallCheck', J$.T(786849, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(786873, '_defineProperties', J$.T(786865, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(786889, '_createClass', J$.T(786881, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(786905, '_possibleConstructorReturn', J$.T(786897, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(786921, '_assertThisInitialized', J$.T(786913, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(786937, '_get', J$.T(786929, _get, 12, false), true, false, false);
            _superPropBase = J$.N(786953, '_superPropBase', J$.T(786945, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(786969, '_getPrototypeOf', J$.T(786961, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(786985, '_inherits', J$.T(786977, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(787001, '_setPrototypeOf', J$.T(786993, _setPrototypeOf, 12, false), true, false, false);
            J$.N(787009, 'Topology', Topology, false, false, false);
            J$.N(787017, 'ServerCapabilities', ServerCapabilities, false, false, false);
            J$.N(787025, 'Cursor', Cursor, false, false, false);
            J$.N(787033, 'translateOptions', translateOptions, false, false, false);
            J$.N(787041, 'NativeTopology', NativeTopology, false, false, false);
            J$.T(782705, 'use strict', 21, false);
            var Topology = J$.W(784873, 'Topology', J$.G(784865, J$.F(784857, J$.I(typeof require === 'undefined' ? require = J$.R(784841, 'require', undefined, true, true) : require = J$.R(784841, 'require', require, true, true)), false)(J$.T(784849, '../core', 21, false)), 'Topology'), Topology, false, true);
            var ServerCapabilities = J$.W(784913, 'ServerCapabilities', J$.G(784905, J$.F(784897, J$.I(typeof require === 'undefined' ? require = J$.R(784881, 'require', undefined, true, true) : require = J$.R(784881, 'require', require, true, true)), false)(J$.T(784889, './topology_base', 21, false)), 'ServerCapabilities'), ServerCapabilities, false, true);
            var Cursor = J$.W(784945, 'Cursor', J$.F(784937, J$.I(typeof require === 'undefined' ? require = J$.R(784921, 'require', undefined, true, true) : require = J$.R(784921, 'require', require, true, true)), false)(J$.T(784929, '../cursor', 21, false)), Cursor, false, true);
            var translateOptions = J$.W(784985, 'translateOptions', J$.G(784977, J$.F(784969, J$.I(typeof require === 'undefined' ? require = J$.R(784953, 'require', undefined, true, true) : require = J$.R(784953, 'require', require, true, true)), false)(J$.T(784961, '../utils', 21, false)), 'translateOptions'), translateOptions, false, true);
            var NativeTopology = J$.W(786793, 'NativeTopology', J$.F(786785, J$.T(786769, function (_Topology) {
                jalangiLabel22:
                    while (true) {
                        try {
                            J$.Fe(786729, arguments.callee, this, arguments);
                            function NativeTopology(servers, options) {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(785633, arguments.callee, this, arguments);
                                            arguments = J$.N(785641, 'arguments', arguments, true, false, false);
                                            servers = J$.N(785649, 'servers', servers, true, false, false);
                                            options = J$.N(785657, 'options', options, true, false, false);
                                            J$.N(785665, '_this', _this, false, false, false);
                                            J$.N(785673, 'clonedOptions', clonedOptions, false, false, false);
                                            J$.N(785681, 'socketOptions', socketOptions, false, false, false);
                                            var _this;
                                            J$.F(785049, J$.R(785025, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(785033, 'this', this, false, false), J$.R(785041, 'NativeTopology', NativeTopology, false, false));
                                            options = J$.W(785073, 'options', J$.C(50720, J$.R(785057, 'options', options, false, false)) ? J$._() : J$.T(785065, {}, 11, false), options, false, false);
                                            var clonedOptions = J$.W(785273, 'clonedOptions', J$.M(785265, J$.I(typeof Object === 'undefined' ? Object = J$.R(785081, 'Object', undefined, true, true) : Object = J$.R(785081, 'Object', Object, true, true)), 'assign', false)(J$.T(785089, {}, 11, false), J$.T(785257, {
                                                cursorFactory: J$.R(785097, 'Cursor', Cursor, false, true),
                                                reconnect: J$.T(785105, false, 23, false),
                                                emitError: J$.C(50728, J$.B(49186, '===', J$.U(49178, 'typeof', J$.G(785121, J$.R(785113, 'options', options, false, false), 'emitError')), J$.T(785129, 'boolean', 21, false))) ? J$.G(785145, J$.R(785137, 'options', options, false, false), 'emitError') : J$.T(785153, true, 23, false),
                                                size: J$.C(50736, J$.B(49202, '===', J$.U(49194, 'typeof', J$.G(785169, J$.R(785161, 'options', options, false, false), 'poolSize')), J$.T(785177, 'number', 21, false))) ? J$.G(785193, J$.R(785185, 'options', options, false, false), 'poolSize') : J$.T(785201, 5, 22, false),
                                                monitorCommands: J$.C(50744, J$.B(49218, '===', J$.U(49210, 'typeof', J$.G(785217, J$.R(785209, 'options', options, false, false), 'monitorCommands')), J$.T(785225, 'boolean', 21, false))) ? J$.G(785241, J$.R(785233, 'options', options, false, false), 'monitorCommands') : J$.T(785249, false, 23, false)
                                            }, 11, false)), clonedOptions, false, false);
                                            clonedOptions = J$.W(785313, 'clonedOptions', J$.F(785305, J$.R(785281, 'translateOptions', translateOptions, false, true), false)(J$.R(785289, 'clonedOptions', clonedOptions, false, false), J$.R(785297, 'options', options, false, false)), clonedOptions, false, false);
                                            var socketOptions = J$.W(785409, 'socketOptions', J$.C(50760, J$.C(50752, J$.G(785329, J$.R(785321, 'options', options, false, false), 'socketOptions')) ? J$.B(49226, '>', J$.G(785369, J$.M(785361, J$.I(typeof Object === 'undefined' ? Object = J$.R(785337, 'Object', undefined, true, true) : Object = J$.R(785337, 'Object', Object, true, true)), 'keys', false)(J$.G(785353, J$.R(785345, 'options', options, false, false), 'socketOptions')), 'length'), J$.T(785377, 0, 22, false)) : J$._()) ? J$.G(785393, J$.R(785385, 'options', options, false, false), 'socketOptions') : J$.R(785401, 'options', options, false, false), socketOptions, false, false);
                                            clonedOptions = J$.W(785449, 'clonedOptions', J$.F(785441, J$.R(785417, 'translateOptions', translateOptions, false, true), false)(J$.R(785425, 'clonedOptions', clonedOptions, false, false), J$.R(785433, 'socketOptions', socketOptions, false, false)), clonedOptions, false, false);
                                            _this = J$.W(785537, '_this', J$.F(785529, J$.R(785457, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(785465, 'this', this, false, false), J$.M(785521, J$.F(785489, J$.R(785473, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(785481, 'NativeTopology', NativeTopology, false, false)), 'call', false)(J$.R(785497, 'this', this, false, false), J$.R(785505, 'servers', servers, false, false), J$.R(785513, 'clonedOptions', clonedOptions, false, false))), _this, false, false);
                                            if (J$.C(50768, J$.G(785553, J$.R(785545, 'options', options, false, false), 'appname'))) {
                                                J$.P(785609, J$.G(785577, J$.G(785569, J$.R(785561, '_this', _this, false, false), 's'), 'clientInfo'), 'application', J$.T(785601, { name: J$.G(785593, J$.R(785585, 'options', options, false, false), 'appname') }, 11, false));
                                            }
                                            return J$.Rt(785625, J$.R(785617, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(787305, J$e);
                                        } finally {
                                            if (J$.Fr(787313))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(786737, 'arguments', arguments, true, false, false);
                            _Topology = J$.N(786745, '_Topology', _Topology, true, false, false);
                            NativeTopology = J$.N(786761, 'NativeTopology', J$.T(786753, NativeTopology, 12, false), true, false, false);
                            J$.F(785017, J$.R(784993, '_inherits', _inherits, false, true), false)(J$.R(785001, 'NativeTopology', NativeTopology, false, false), J$.R(785009, '_Topology', _Topology, false, false));
                            J$.F(786705, J$.R(785689, '_createClass', _createClass, false, true), false)(J$.R(785697, 'NativeTopology', NativeTopology, false, false), J$.T(786697, [
                                J$.T(785921, {
                                    key: J$.T(785705, 'capabilities', 21, false),
                                    value: J$.T(785913, function capabilities() {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(785897, arguments.callee, this, arguments);
                                                    arguments = J$.N(785905, 'arguments', arguments, true, false, false);
                                                    if (J$.C(50776, J$.G(785729, J$.G(785721, J$.R(785713, 'this', this, false, false), 's'), 'sCapabilities')))
                                                        return J$.Rt(785761, J$.G(785753, J$.G(785745, J$.R(785737, 'this', this, false, false), 's'), 'sCapabilities'));
                                                    if (J$.C(50784, J$.B(49234, '==', J$.M(785777, J$.R(785769, 'this', this, false, false), 'lastIsMaster', false)(), J$.T(785785, null, 25, false))))
                                                        return J$.Rt(785801, J$.T(785793, null, 25, false));
                                                    J$.P(785857, J$.G(785817, J$.R(785809, 'this', this, false, false), 's'), 'sCapabilities', J$.F(785849, J$.R(785825, 'ServerCapabilities', ServerCapabilities, false, true), true)(J$.M(785841, J$.R(785833, 'this', this, false, false), 'lastIsMaster', false)()));
                                                    return J$.Rt(785889, J$.G(785881, J$.G(785873, J$.R(785865, 'this', this, false, false), 's'), 'sCapabilities'));
                                                } catch (J$e) {
                                                    J$.Ex(787321, J$e);
                                                } finally {
                                                    if (J$.Fr(787329))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(786113, {
                                    key: J$.T(785929, 'command', 21, false),
                                    value: J$.T(786105, function command(ns, cmd, options, callback) {
                                        jalangiLabel18:
                                            while (true) {
                                                try {
                                                    J$.Fe(786057, arguments.callee, this, arguments);
                                                    arguments = J$.N(786065, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(786073, 'ns', ns, true, false, false);
                                                    cmd = J$.N(786081, 'cmd', cmd, true, false, false);
                                                    options = J$.N(786089, 'options', options, true, false, false);
                                                    callback = J$.N(786097, 'callback', callback, true, false, false);
                                                    J$.M(786049, J$.F(785993, J$.R(785937, '_get', _get, false, true), false)(J$.F(785969, J$.R(785945, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(785961, J$.R(785953, 'NativeTopology', NativeTopology, false, false), 'prototype')), J$.T(785977, 'command', 21, false), J$.R(785985, 'this', this, false, false)), 'call', false)(J$.R(786001, 'this', this, false, false), J$.M(786017, J$.R(786009, 'ns', ns, false, false), 'toString', false)(), J$.R(786025, 'cmd', cmd, false, false), J$.R(786033, 'options', options, false, false), J$.R(786041, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(787337, J$e);
                                                } finally {
                                                    if (J$.Fr(787345))
                                                        continue jalangiLabel18;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(786305, {
                                    key: J$.T(786121, 'insert', 21, false),
                                    value: J$.T(786297, function insert(ns, ops, options, callback) {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(786249, arguments.callee, this, arguments);
                                                    arguments = J$.N(786257, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(786265, 'ns', ns, true, false, false);
                                                    ops = J$.N(786273, 'ops', ops, true, false, false);
                                                    options = J$.N(786281, 'options', options, true, false, false);
                                                    callback = J$.N(786289, 'callback', callback, true, false, false);
                                                    J$.M(786241, J$.F(786185, J$.R(786129, '_get', _get, false, true), false)(J$.F(786161, J$.R(786137, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(786153, J$.R(786145, 'NativeTopology', NativeTopology, false, false), 'prototype')), J$.T(786169, 'insert', 21, false), J$.R(786177, 'this', this, false, false)), 'call', false)(J$.R(786193, 'this', this, false, false), J$.M(786209, J$.R(786201, 'ns', ns, false, false), 'toString', false)(), J$.R(786217, 'ops', ops, false, false), J$.R(786225, 'options', options, false, false), J$.R(786233, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(787353, J$e);
                                                } finally {
                                                    if (J$.Fr(787361))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(786497, {
                                    key: J$.T(786313, 'update', 21, false),
                                    value: J$.T(786489, function update(ns, ops, options, callback) {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(786441, arguments.callee, this, arguments);
                                                    arguments = J$.N(786449, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(786457, 'ns', ns, true, false, false);
                                                    ops = J$.N(786465, 'ops', ops, true, false, false);
                                                    options = J$.N(786473, 'options', options, true, false, false);
                                                    callback = J$.N(786481, 'callback', callback, true, false, false);
                                                    J$.M(786433, J$.F(786377, J$.R(786321, '_get', _get, false, true), false)(J$.F(786353, J$.R(786329, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(786345, J$.R(786337, 'NativeTopology', NativeTopology, false, false), 'prototype')), J$.T(786361, 'update', 21, false), J$.R(786369, 'this', this, false, false)), 'call', false)(J$.R(786385, 'this', this, false, false), J$.M(786401, J$.R(786393, 'ns', ns, false, false), 'toString', false)(), J$.R(786409, 'ops', ops, false, false), J$.R(786417, 'options', options, false, false), J$.R(786425, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(787369, J$e);
                                                } finally {
                                                    if (J$.Fr(787377))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(786689, {
                                    key: J$.T(786505, 'remove', 21, false),
                                    value: J$.T(786681, function remove(ns, ops, options, callback) {
                                        jalangiLabel21:
                                            while (true) {
                                                try {
                                                    J$.Fe(786633, arguments.callee, this, arguments);
                                                    arguments = J$.N(786641, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(786649, 'ns', ns, true, false, false);
                                                    ops = J$.N(786657, 'ops', ops, true, false, false);
                                                    options = J$.N(786665, 'options', options, true, false, false);
                                                    callback = J$.N(786673, 'callback', callback, true, false, false);
                                                    J$.M(786625, J$.F(786569, J$.R(786513, '_get', _get, false, true), false)(J$.F(786545, J$.R(786521, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(786537, J$.R(786529, 'NativeTopology', NativeTopology, false, false), 'prototype')), J$.T(786553, 'remove', 21, false), J$.R(786561, 'this', this, false, false)), 'call', false)(J$.R(786577, 'this', this, false, false), J$.M(786593, J$.R(786585, 'ns', ns, false, false), 'toString', false)(), J$.R(786601, 'ops', ops, false, false), J$.R(786609, 'options', options, false, false), J$.R(786617, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(787385, J$e);
                                                } finally {
                                                    if (J$.Fr(787393))
                                                        continue jalangiLabel21;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(786721, J$.R(786713, 'NativeTopology', NativeTopology, false, false));
                        } catch (J$e) {
                            J$.Ex(787401, J$e);
                        } finally {
                            if (J$.Fr(787409))
                                continue jalangiLabel22;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(786777, 'Topology', Topology, false, true)), NativeTopology, false, true);
            J$.P(786817, J$.I(typeof module === 'undefined' ? module = J$.R(786801, 'module', undefined, true, true) : module = J$.R(786801, 'module', module, true, true)), 'exports', J$.R(786809, 'NativeTopology', NativeTopology, false, true));
        } catch (J$e) {
            J$.Ex(787417, J$e);
        } finally {
            if (J$.Sr(787425))
                continue jalangiLabel23;
            else
                break jalangiLabel23;
        }
    }
// JALANGI DO NOT INSTRUMENT
