// mongodb/lib/topologies/topology_base.js
J$.noInstrEval = false;
jalangiLabel47:
    while (true) {
        try {
            J$.Se(815929, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/topologies/topology_base.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(805753, arguments.callee, this, arguments);
                            arguments = J$.N(805761, 'arguments', arguments, true, false, false);
                            obj = J$.N(805769, 'obj', obj, true, false, false);
                            if (J$.C(51584, J$.C(51544, J$.B(50122, '===', J$.U(50114, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(805489, 'Symbol', undefined, true, true) : Symbol = J$.R(805489, 'Symbol', Symbol, true, true))), J$.T(805497, 'function', 21, false))) ? J$.B(50138, '===', J$.U(50130, 'typeof', J$.G(805513, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(805505, 'Symbol', undefined, true, true) : Symbol = J$.R(805505, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(805521, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(805577, '_typeof', J$.T(805569, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(805545, arguments.callee, this, arguments);
                                                arguments = J$.N(805553, 'arguments', arguments, true, false, false);
                                                obj = J$.N(805561, 'obj', obj, true, false, false);
                                                return J$.Rt(805537, J$.U(50146, 'typeof', J$.R(805529, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(816217, J$e);
                                            } finally {
                                                if (J$.Fr(816225))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(805713, '_typeof', J$.T(805705, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(805681, arguments.callee, this, arguments);
                                                arguments = J$.N(805689, 'arguments', arguments, true, false, false);
                                                obj = J$.N(805697, 'obj', obj, true, false, false);
                                                return J$.Rt(805673, J$.C(51576, J$.C(51568, J$.C(51560, J$.C(51552, J$.R(805585, 'obj', obj, false, false)) ? J$.B(50162, '===', J$.U(50154, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(805593, 'Symbol', undefined, true, true) : Symbol = J$.R(805593, 'Symbol', Symbol, true, true))), J$.T(805601, 'function', 21, false)) : J$._()) ? J$.B(50170, '===', J$.G(805617, J$.R(805609, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(805625, 'Symbol', undefined, true, true) : Symbol = J$.R(805625, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(50178, '!==', J$.R(805633, 'obj', obj, false, false), J$.G(805649, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(805641, 'Symbol', undefined, true, true) : Symbol = J$.R(805641, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(805657, 'symbol', 21, false) : J$.U(50186, 'typeof', J$.R(805665, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(816233, J$e);
                                            } finally {
                                                if (J$.Fr(816241))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(805745, J$.F(805737, J$.R(805721, '_typeof', _typeof, false, true), false)(J$.R(805729, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(816249, J$e);
                        } finally {
                            if (J$.Fr(816257))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(805817, arguments.callee, this, arguments);
                            arguments = J$.N(805825, 'arguments', arguments, true, false, false);
                            instance = J$.N(805833, 'instance', instance, true, false, false);
                            Constructor = J$.N(805841, 'Constructor', Constructor, true, false, false);
                            if (J$.C(51592, J$.U(50202, '!', J$.B(50194, 'instanceof', J$.R(805777, 'instance', instance, false, false), J$.R(805785, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(805809, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(805793, 'TypeError', undefined, true, true) : TypeError = J$.R(805793, 'TypeError', TypeError, true, true)), true)(J$.T(805801, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(816265, J$e);
                        } finally {
                            if (J$.Fr(816273))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(806089, arguments.callee, this, arguments);
                            arguments = J$.N(806097, 'arguments', arguments, true, false, false);
                            target = J$.N(806105, 'target', target, true, false, false);
                            props = J$.N(806113, 'props', props, true, false, false);
                            J$.N(806121, 'i', i, false, false, false);
                            J$.N(806129, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(805857, 'i', J$.T(805849, 0, 22, false), i, false, false); J$.C(51616, J$.B(50210, '<', J$.R(805865, 'i', i, false, false), J$.G(805881, J$.R(805873, 'props', props, false, false), 'length'))); J$.B(50234, '-', i = J$.W(805897, 'i', J$.B(50226, '+', J$.U(50218, '+', J$.R(805889, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(805929, 'descriptor', J$.G(805921, J$.R(805905, 'props', props, false, false), J$.R(805913, 'i', i, false, false)), descriptor, false, false);
                                J$.P(805969, J$.R(805937, 'descriptor', descriptor, false, false), 'enumerable', J$.C(51600, J$.G(805953, J$.R(805945, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(805961, false, 23, false));
                                J$.P(805993, J$.R(805977, 'descriptor', descriptor, false, false), 'configurable', J$.T(805985, true, 23, false));
                                if (J$.C(51608, J$.B(50242, 'in', J$.T(806001, 'value', 21, false), J$.R(806009, 'descriptor', descriptor, false, false))))
                                    J$.P(806033, J$.R(806017, 'descriptor', descriptor, false, false), 'writable', J$.T(806025, true, 23, false));
                                J$.M(806081, J$.I(typeof Object === 'undefined' ? Object = J$.R(806041, 'Object', undefined, true, true) : Object = J$.R(806041, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(806049, 'target', target, false, false), J$.G(806065, J$.R(806057, 'descriptor', descriptor, false, false), 'key'), J$.R(806073, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(816281, J$e);
                        } finally {
                            if (J$.Fr(816289))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(806241, arguments.callee, this, arguments);
                            arguments = J$.N(806249, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(806257, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(806265, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(806273, 'staticProps', staticProps, true, false, false);
                            if (J$.C(51624, J$.R(806137, 'protoProps', protoProps, false, false)))
                                J$.F(806177, J$.R(806145, '_defineProperties', _defineProperties, false, true), false)(J$.G(806161, J$.R(806153, 'Constructor', Constructor, false, false), 'prototype'), J$.R(806169, 'protoProps', protoProps, false, false));
                            if (J$.C(51632, J$.R(806185, 'staticProps', staticProps, false, false)))
                                J$.F(806217, J$.R(806193, '_defineProperties', _defineProperties, false, true), false)(J$.R(806201, 'Constructor', Constructor, false, false), J$.R(806209, 'staticProps', staticProps, false, false));
                            return J$.Rt(806233, J$.R(806225, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(816297, J$e);
                        } finally {
                            if (J$.Fr(816305))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(806385, arguments.callee, this, arguments);
                            arguments = J$.N(806393, 'arguments', arguments, true, false, false);
                            self = J$.N(806401, 'self', self, true, false, false);
                            call = J$.N(806409, 'call', call, true, false, false);
                            if (J$.C(51656, J$.C(51648, J$.R(806281, 'call', call, false, false)) ? J$.C(51640, J$.B(50250, '===', J$.F(806305, J$.R(806289, '_typeof', _typeof, false, true), false)(J$.R(806297, 'call', call, false, false)), J$.T(806313, 'object', 21, false))) ? J$._() : J$.B(50266, '===', J$.U(50258, 'typeof', J$.R(806321, 'call', call, false, false)), J$.T(806329, 'function', 21, false)) : J$._())) {
                                return J$.Rt(806345, J$.R(806337, 'call', call, false, false));
                            }
                            return J$.Rt(806377, J$.F(806369, J$.R(806353, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(806361, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(816313, J$e);
                        } finally {
                            if (J$.Fr(816321))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(806473, arguments.callee, this, arguments);
                            arguments = J$.N(806481, 'arguments', arguments, true, false, false);
                            self = J$.N(806489, 'self', self, true, false, false);
                            if (J$.C(51664, J$.B(50274, '===', J$.R(806417, 'self', self, false, false), void J$.T(806425, 0, 22, false)))) {
                                throw J$.F(806449, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(806433, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(806433, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(806441, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(806465, J$.R(806457, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(816329, J$e);
                        } finally {
                            if (J$.Fr(816337))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(806649, arguments.callee, this, arguments);
                            arguments = J$.N(806657, 'arguments', arguments, true, false, false);
                            o = J$.N(806665, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(806609, '_getPrototypeOf', J$.C(51680, J$.G(806505, J$.I(typeof Object === 'undefined' ? Object = J$.R(806497, 'Object', undefined, true, true) : Object = J$.R(806497, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(806521, J$.I(typeof Object === 'undefined' ? Object = J$.R(806513, 'Object', undefined, true, true) : Object = J$.R(806513, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(806601, function _getPrototypeOf(o) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(806577, arguments.callee, this, arguments);
                                            arguments = J$.N(806585, 'arguments', arguments, true, false, false);
                                            o = J$.N(806593, 'o', o, true, false, false);
                                            return J$.Rt(806569, J$.C(51672, J$.G(806537, J$.R(806529, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(806561, J$.I(typeof Object === 'undefined' ? Object = J$.R(806545, 'Object', undefined, true, true) : Object = J$.R(806545, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(806553, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(816345, J$e);
                                        } finally {
                                            if (J$.Fr(816353))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(806641, J$.F(806633, J$.R(806617, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(806625, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(816361, J$e);
                        } finally {
                            if (J$.Fr(816369))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(806865, arguments.callee, this, arguments);
                            arguments = J$.N(806873, 'arguments', arguments, true, false, false);
                            subClass = J$.N(806881, 'subClass', subClass, true, false, false);
                            superClass = J$.N(806889, 'superClass', superClass, true, false, false);
                            if (J$.C(51696, J$.C(51688, J$.B(50290, '!==', J$.U(50282, 'typeof', J$.R(806673, 'superClass', superClass, false, false)), J$.T(806681, 'function', 21, false))) ? J$.B(50298, '!==', J$.R(806689, 'superClass', superClass, false, false), J$.T(806697, null, 25, false)) : J$._())) {
                                throw J$.F(806721, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(806705, 'TypeError', undefined, true, true) : TypeError = J$.R(806705, 'TypeError', TypeError, true, true)), true)(J$.T(806713, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(806817, J$.R(806729, 'subClass', subClass, false, false), 'prototype', J$.M(806809, J$.I(typeof Object === 'undefined' ? Object = J$.R(806737, 'Object', undefined, true, true) : Object = J$.R(806737, 'Object', Object, true, true)), 'create', false)(J$.C(51704, J$.R(806745, 'superClass', superClass, false, false)) ? J$.G(806761, J$.R(806753, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(806801, {
                                constructor: J$.T(806793, {
                                    value: J$.R(806769, 'subClass', subClass, false, false),
                                    writable: J$.T(806777, true, 23, false),
                                    configurable: J$.T(806785, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(51712, J$.R(806825, 'superClass', superClass, false, false)))
                                J$.F(806857, J$.R(806833, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(806841, 'subClass', subClass, false, false), J$.R(806849, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(816377, J$e);
                        } finally {
                            if (J$.Fr(816385))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(807041, arguments.callee, this, arguments);
                            arguments = J$.N(807049, 'arguments', arguments, true, false, false);
                            o = J$.N(807057, 'o', o, true, false, false);
                            p = J$.N(807065, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(806993, '_setPrototypeOf', J$.C(51720, J$.G(806905, J$.I(typeof Object === 'undefined' ? Object = J$.R(806897, 'Object', undefined, true, true) : Object = J$.R(806897, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(806985, function _setPrototypeOf(o, p) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(806953, arguments.callee, this, arguments);
                                            arguments = J$.N(806961, 'arguments', arguments, true, false, false);
                                            o = J$.N(806969, 'o', o, true, false, false);
                                            p = J$.N(806977, 'p', p, true, false, false);
                                            J$.P(806929, J$.R(806913, 'o', o, false, false), '__proto__', J$.R(806921, 'p', p, false, false));
                                            return J$.Rt(806945, J$.R(806937, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(816393, J$e);
                                        } finally {
                                            if (J$.Fr(816401))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(807033, J$.F(807025, J$.R(807001, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(807009, 'o', o, false, false), J$.R(807017, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(816409, J$e);
                        } finally {
                            if (J$.Fr(816417))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(815945, '_typeof', J$.T(815937, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(815961, '_classCallCheck', J$.T(815953, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(815977, '_defineProperties', J$.T(815969, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(815993, '_createClass', J$.T(815985, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(816009, '_possibleConstructorReturn', J$.T(816001, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(816025, '_assertThisInitialized', J$.T(816017, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(816041, '_getPrototypeOf', J$.T(816033, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(816057, '_inherits', J$.T(816049, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(816073, '_setPrototypeOf', J$.T(816065, _setPrototypeOf, 12, false), true, false, false);
            J$.N(816081, 'EventEmitter', EventEmitter, false, false, false);
            J$.N(816089, 'MongoError', MongoError, false, false, false);
            J$.N(816097, 'f', f, false, false, false);
            J$.N(816105, 'os', os, false, false, false);
            J$.N(816113, 'translateReadPreference', translateReadPreference, false, false, false);
            J$.N(816121, 'ClientSession', ClientSession, false, false, false);
            J$.N(816129, 'Store', Store, false, false, false);
            J$.N(816137, 'primaryOptions', primaryOptions, false, false, false);
            J$.N(816145, 'secondaryOptions', secondaryOptions, false, false, false);
            J$.N(816153, 'ServerCapabilities', ServerCapabilities, false, false, false);
            J$.N(816161, 'driverVersion', driverVersion, false, false, false);
            J$.N(816169, 'nodejsversion', nodejsversion, false, false, false);
            J$.N(816177, 'type', type, false, false, false);
            J$.N(816185, 'name', name, false, false, false);
            J$.N(816193, 'architecture', architecture, false, false, false);
            J$.N(816201, 'release', release, false, false, false);
            J$.N(816209, 'TopologyBase', TopologyBase, false, false, false);
            J$.T(805481, 'use strict', 21, false);
            var EventEmitter = J$.W(807257, 'EventEmitter', J$.F(807089, J$.I(typeof require === 'undefined' ? require = J$.R(807073, 'require', undefined, true, true) : require = J$.R(807073, 'require', require, true, true)), false)(J$.T(807081, 'events', 21, false)), EventEmitter, false, true), MongoError = J$.W(807265, 'MongoError', J$.G(807121, J$.F(807113, J$.I(typeof require === 'undefined' ? require = J$.R(807097, 'require', undefined, true, true) : require = J$.R(807097, 'require', require, true, true)), false)(J$.T(807105, '../core', 21, false)), 'MongoError'), MongoError, false, true), f = J$.W(807273, 'f', J$.G(807153, J$.F(807145, J$.I(typeof require === 'undefined' ? require = J$.R(807129, 'require', undefined, true, true) : require = J$.R(807129, 'require', require, true, true)), false)(J$.T(807137, 'util', 21, false)), 'format'), f, false, true), os = J$.W(807281, 'os', J$.F(807177, J$.I(typeof require === 'undefined' ? require = J$.R(807161, 'require', undefined, true, true) : require = J$.R(807161, 'require', require, true, true)), false)(J$.T(807169, 'os', 21, false)), os, false, true), translateReadPreference = J$.W(807289, 'translateReadPreference', J$.G(807209, J$.F(807201, J$.I(typeof require === 'undefined' ? require = J$.R(807185, 'require', undefined, true, true) : require = J$.R(807185, 'require', require, true, true)), false)(J$.T(807193, '../utils', 21, false)), 'translateReadPreference'), translateReadPreference, false, true), ClientSession = J$.W(807297, 'ClientSession', J$.G(807249, J$.G(807241, J$.F(807233, J$.I(typeof require === 'undefined' ? require = J$.R(807217, 'require', undefined, true, true) : require = J$.R(807217, 'require', require, true, true)), false)(J$.T(807225, '../core', 21, false)), 'Sessions'), 'ClientSession'), ClientSession, false, true);
            var Store = J$.W(807593, 'Store', J$.T(807585, function Store(topology, storeOptions) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(807537, arguments.callee, this, arguments);
                            arguments = J$.N(807545, 'arguments', arguments, true, false, false);
                            topology = J$.N(807553, 'topology', topology, true, false, false);
                            storeOptions = J$.N(807561, 'storeOptions', storeOptions, true, false, false);
                            J$.N(807569, 'self', self, false, false, false);
                            J$.N(807577, 'storedOps', storedOps, false, false, false);
                            var self = J$.W(807313, 'self', J$.R(807305, 'this', this, false, false), self, false, false);
                            var storedOps = J$.W(807329, 'storedOps', J$.T(807321, [], 10, false), storedOps, false, false);
                            storeOptions = J$.W(807369, 'storeOptions', J$.C(51728, J$.R(807337, 'storeOptions', storeOptions, false, false)) ? J$._() : J$.T(807361, {
                                force: J$.T(807345, false, 23, false),
                                bufferMaxEntries: J$.U(50306, '-', J$.T(807353, 1, 22, false))
                            }, 11, false), storeOptions, false, false);
                            J$.P(807417, J$.R(807377, 'this', this, false, false), 's', J$.T(807409, {
                                storedOps: J$.R(807385, 'storedOps', storedOps, false, false),
                                storeOptions: J$.R(807393, 'storeOptions', storeOptions, false, false),
                                topology: J$.R(807401, 'topology', topology, false, false)
                            }, 11, false));
                            J$.M(807529, J$.I(typeof Object === 'undefined' ? Object = J$.R(807425, 'Object', undefined, true, true) : Object = J$.R(807425, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(807433, 'this', this, false, false), J$.T(807441, 'length', 21, false), J$.T(807521, {
                                enumerable: J$.T(807449, true, 23, false),
                                get: J$.T(807513, function get() {
                                    jalangiLabel13:
                                        while (true) {
                                            try {
                                                J$.Fe(807497, arguments.callee, this, arguments);
                                                arguments = J$.N(807505, 'arguments', arguments, true, false, false);
                                                return J$.Rt(807489, J$.G(807481, J$.G(807473, J$.G(807465, J$.R(807457, 'self', self, false, false), 's'), 'storedOps'), 'length'));
                                            } catch (J$e) {
                                                J$.Ex(816425, J$e);
                                            } finally {
                                                if (J$.Fr(816433))
                                                    continue jalangiLabel13;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(816441, J$e);
                        } finally {
                            if (J$.Fr(816449))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), Store, false, true);
            J$.P(808313, J$.G(807609, J$.R(807601, 'Store', Store, false, true), 'prototype'), 'add', J$.T(808305, function (opType, ns, ops, options, callback) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(808241, arguments.callee, this, arguments);
                            arguments = J$.N(808249, 'arguments', arguments, true, false, false);
                            opType = J$.N(808257, 'opType', opType, true, false, false);
                            ns = J$.N(808265, 'ns', ns, true, false, false);
                            ops = J$.N(808273, 'ops', ops, true, false, false);
                            options = J$.N(808281, 'options', options, true, false, false);
                            callback = J$.N(808289, 'callback', callback, true, false, false);
                            J$.N(808297, 'op', op, false, false, false);
                            if (J$.C(51736, J$.G(807641, J$.G(807633, J$.G(807625, J$.R(807617, 'this', this, false, false), 's'), 'storeOptions'), 'force'))) {
                                return J$.Rt(807705, J$.F(807697, J$.R(807649, 'callback', callback, false, false), false)(J$.M(807689, J$.R(807657, 'MongoError', MongoError, false, true), 'create', false)(J$.T(807681, {
                                    message: J$.T(807665, 'db closed by application', 21, false),
                                    driver: J$.T(807673, true, 23, false)
                                }, 11, false))));
                            }
                            if (J$.C(51744, J$.B(50314, '===', J$.G(807737, J$.G(807729, J$.G(807721, J$.R(807713, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries'), J$.T(807745, 0, 22, false)))) {
                                return J$.Rt(807857, J$.F(807849, J$.R(807753, 'callback', callback, false, false), false)(J$.M(807841, J$.R(807761, 'MongoError', MongoError, false, true), 'create', false)(J$.T(807833, {
                                    message: J$.F(807817, J$.R(807769, 'f', f, false, true), false)(J$.T(807777, 'no connection available for operation and number of stored operation > %s', 21, false), J$.G(807809, J$.G(807801, J$.G(807793, J$.R(807785, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')),
                                    driver: J$.T(807825, true, 23, false)
                                }, 11, false))));
                            }
                            if (J$.C(51768, J$.C(51752, J$.B(50322, '>', J$.G(807889, J$.G(807881, J$.G(807873, J$.R(807865, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries'), J$.T(807897, 0, 22, false))) ? J$.B(50330, '>', J$.G(807929, J$.G(807921, J$.G(807913, J$.R(807905, 'this', this, false, false), 's'), 'storedOps'), 'length'), J$.G(807961, J$.G(807953, J$.G(807945, J$.R(807937, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')) : J$._())) {
                                while (J$.C(51760, J$.B(50338, '>', J$.G(807993, J$.G(807985, J$.G(807977, J$.R(807969, 'this', this, false, false), 's'), 'storedOps'), 'length'), J$.T(808001, 0, 22, false)))) {
                                    var op = J$.W(808041, 'op', J$.M(808033, J$.G(808025, J$.G(808017, J$.R(808009, 'this', this, false, false), 's'), 'storedOps'), 'shift', false)(), op, false, false);
                                    J$.M(808145, J$.R(808049, 'op', op, false, false), 'c', false)(J$.M(808137, J$.R(808057, 'MongoError', MongoError, false, true), 'create', false)(J$.T(808129, {
                                        message: J$.F(808113, J$.R(808065, 'f', f, false, true), false)(J$.T(808073, 'no connection available for operation and number of stored operation > %s', 21, false), J$.G(808105, J$.G(808097, J$.G(808089, J$.R(808081, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')),
                                        driver: J$.T(808121, true, 23, false)
                                    }, 11, false)));
                                }
                                return J$.Rt(808153, undefined);
                            }
                            J$.M(808233, J$.G(808177, J$.G(808169, J$.R(808161, 'this', this, false, false), 's'), 'storedOps'), 'push', false)(J$.T(808225, {
                                t: J$.R(808185, 'opType', opType, false, false),
                                n: J$.R(808193, 'ns', ns, false, false),
                                o: J$.R(808201, 'ops', ops, false, false),
                                op: J$.R(808209, 'options', options, false, false),
                                c: J$.R(808217, 'callback', callback, false, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(816457, J$e);
                        } finally {
                            if (J$.Fr(816465))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(809033, J$.G(808329, J$.R(808321, 'Store', Store, false, true), 'prototype'), 'addObjectAndMethod', J$.T(809025, function (opType, object, method, params, callback) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(808961, arguments.callee, this, arguments);
                            arguments = J$.N(808969, 'arguments', arguments, true, false, false);
                            opType = J$.N(808977, 'opType', opType, true, false, false);
                            object = J$.N(808985, 'object', object, true, false, false);
                            method = J$.N(808993, 'method', method, true, false, false);
                            params = J$.N(809001, 'params', params, true, false, false);
                            callback = J$.N(809009, 'callback', callback, true, false, false);
                            J$.N(809017, 'op', op, false, false, false);
                            if (J$.C(51776, J$.G(808361, J$.G(808353, J$.G(808345, J$.R(808337, 'this', this, false, false), 's'), 'storeOptions'), 'force'))) {
                                return J$.Rt(808425, J$.F(808417, J$.R(808369, 'callback', callback, false, false), false)(J$.M(808409, J$.R(808377, 'MongoError', MongoError, false, true), 'create', false)(J$.T(808401, {
                                    message: J$.T(808385, 'db closed by application', 21, false),
                                    driver: J$.T(808393, true, 23, false)
                                }, 11, false))));
                            }
                            if (J$.C(51784, J$.B(50346, '===', J$.G(808457, J$.G(808449, J$.G(808441, J$.R(808433, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries'), J$.T(808465, 0, 22, false)))) {
                                return J$.Rt(808577, J$.F(808569, J$.R(808473, 'callback', callback, false, false), false)(J$.M(808561, J$.R(808481, 'MongoError', MongoError, false, true), 'create', false)(J$.T(808553, {
                                    message: J$.F(808537, J$.R(808489, 'f', f, false, true), false)(J$.T(808497, 'no connection available for operation and number of stored operation > %s', 21, false), J$.G(808529, J$.G(808521, J$.G(808513, J$.R(808505, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')),
                                    driver: J$.T(808545, true, 23, false)
                                }, 11, false))));
                            }
                            if (J$.C(51808, J$.C(51792, J$.B(50354, '>', J$.G(808609, J$.G(808601, J$.G(808593, J$.R(808585, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries'), J$.T(808617, 0, 22, false))) ? J$.B(50362, '>', J$.G(808649, J$.G(808641, J$.G(808633, J$.R(808625, 'this', this, false, false), 's'), 'storedOps'), 'length'), J$.G(808681, J$.G(808673, J$.G(808665, J$.R(808657, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')) : J$._())) {
                                while (J$.C(51800, J$.B(50370, '>', J$.G(808713, J$.G(808705, J$.G(808697, J$.R(808689, 'this', this, false, false), 's'), 'storedOps'), 'length'), J$.T(808721, 0, 22, false)))) {
                                    var op = J$.W(808761, 'op', J$.M(808753, J$.G(808745, J$.G(808737, J$.R(808729, 'this', this, false, false), 's'), 'storedOps'), 'shift', false)(), op, false, false);
                                    J$.M(808865, J$.R(808769, 'op', op, false, false), 'c', false)(J$.M(808857, J$.R(808777, 'MongoError', MongoError, false, true), 'create', false)(J$.T(808849, {
                                        message: J$.F(808833, J$.R(808785, 'f', f, false, true), false)(J$.T(808793, 'no connection available for operation and number of stored operation > %s', 21, false), J$.G(808825, J$.G(808817, J$.G(808809, J$.R(808801, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')),
                                        driver: J$.T(808841, true, 23, false)
                                    }, 11, false)));
                                }
                                return J$.Rt(808873, undefined);
                            }
                            J$.M(808953, J$.G(808897, J$.G(808889, J$.R(808881, 'this', this, false, false), 's'), 'storedOps'), 'push', false)(J$.T(808945, {
                                t: J$.R(808905, 'opType', opType, false, false),
                                m: J$.R(808913, 'method', method, false, false),
                                o: J$.R(808921, 'object', object, false, false),
                                p: J$.R(808929, 'params', params, false, false),
                                c: J$.R(808937, 'callback', callback, false, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(816473, J$e);
                        } finally {
                            if (J$.Fr(816481))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(809233, J$.G(809049, J$.R(809041, 'Store', Store, false, true), 'prototype'), 'flush', J$.T(809225, function (err) {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(809201, arguments.callee, this, arguments);
                            arguments = J$.N(809209, 'arguments', arguments, true, false, false);
                            err = J$.N(809217, 'err', err, true, false, false);
                            while (J$.C(51824, J$.B(50378, '>', J$.G(809081, J$.G(809073, J$.G(809065, J$.R(809057, 'this', this, false, false), 's'), 'storedOps'), 'length'), J$.T(809089, 0, 22, false)))) {
                                J$.M(809193, J$.M(809121, J$.G(809113, J$.G(809105, J$.R(809097, 'this', this, false, false), 's'), 'storedOps'), 'shift', false)(), 'c', false)(J$.C(51816, J$.R(809129, 'err', err, false, false)) ? J$._() : J$.M(809185, J$.R(809137, 'MongoError', MongoError, false, true), 'create', false)(J$.T(809177, {
                                    message: J$.F(809161, J$.R(809145, 'f', f, false, true), false)(J$.T(809153, 'no connection available for operation', 21, false)),
                                    driver: J$.T(809169, true, 23, false)
                                }, 11, false)));
                            }
                        } catch (J$e) {
                            J$.Ex(816489, J$e);
                        } finally {
                            if (J$.Fr(816497))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var primaryOptions = J$.W(809281, 'primaryOptions', J$.T(809273, [
                J$.T(809241, 'primary', 21, false),
                J$.T(809249, 'primaryPreferred', 21, false),
                J$.T(809257, 'nearest', 21, false),
                J$.T(809265, 'secondaryPreferred', 21, false)
            ], 10, false), primaryOptions, false, true);
            var secondaryOptions = J$.W(809313, 'secondaryOptions', J$.T(809305, [
                J$.T(809289, 'secondary', 21, false),
                J$.T(809297, 'secondaryPreferred', 21, false)
            ], 10, false), secondaryOptions, false, true);
            J$.P(810881, J$.G(809329, J$.R(809321, 'Store', Store, false, true), 'prototype'), 'execute', J$.T(810873, function (options) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(810817, arguments.callee, this, arguments);
                            arguments = J$.N(810825, 'arguments', arguments, true, false, false);
                            options = J$.N(810833, 'options', options, true, false, false);
                            J$.N(810841, 'ops', ops, false, false, false);
                            J$.N(810849, 'executePrimary', executePrimary, false, false, false);
                            J$.N(810857, 'executeSecondary', executeSecondary, false, false, false);
                            J$.N(810865, 'op', op, false, false, false);
                            options = J$.W(809353, 'options', J$.C(51832, J$.R(809337, 'options', options, false, false)) ? J$._() : J$.T(809345, {}, 11, false), options, false, false);
                            var ops = J$.W(809385, 'ops', J$.G(809377, J$.G(809369, J$.R(809361, 'this', this, false, false), 's'), 'storedOps'), ops, false, false);
                            J$.P(809417, J$.G(809401, J$.R(809393, 'this', this, false, false), 's'), 'storedOps', J$.T(809409, [], 10, false));
                            var executePrimary = J$.W(809473, 'executePrimary', J$.C(51840, J$.B(50394, '===', J$.U(50386, 'typeof', J$.G(809433, J$.R(809425, 'options', options, false, false), 'executePrimary')), J$.T(809441, 'boolean', 21, false))) ? J$.G(809457, J$.R(809449, 'options', options, false, false), 'executePrimary') : J$.T(809465, true, 23, false), executePrimary, false, false);
                            var executeSecondary = J$.W(809529, 'executeSecondary', J$.C(51848, J$.B(50410, '===', J$.U(50402, 'typeof', J$.G(809489, J$.R(809481, 'options', options, false, false), 'executeSecondary')), J$.T(809497, 'boolean', 21, false))) ? J$.G(809513, J$.R(809505, 'options', options, false, false), 'executeSecondary') : J$.T(809521, true, 23, false), executeSecondary, false, false);
                            while (J$.C(52048, J$.B(50418, '>', J$.G(809545, J$.R(809537, 'ops', ops, false, false), 'length'), J$.T(809553, 0, 22, false)))) {
                                var op = J$.W(809577, 'op', J$.M(809569, J$.R(809561, 'ops', ops, false, false), 'shift', false)(), op, false, false);
                                if (J$.C(52040, J$.B(50426, '===', J$.G(809593, J$.R(809585, 'op', op, false, false), 't'), J$.T(809601, 'cursor', 21, false)))) {
                                    if (J$.C(51936, J$.C(51856, J$.R(809609, 'executePrimary', executePrimary, false, false)) ? J$.R(809617, 'executeSecondary', executeSecondary, false, false) : J$._())) {
                                        J$.M(809697, J$.G(809657, J$.G(809633, J$.R(809625, 'op', op, false, false), 'o'), J$.G(809649, J$.R(809641, 'op', op, false, false), 'm')), 'apply', false)(J$.G(809673, J$.R(809665, 'op', op, false, false), 'o'), J$.G(809689, J$.R(809681, 'op', op, false, false), 'p'));
                                    } else if (J$.C(51928, J$.C(51880, J$.C(51872, J$.C(51864, J$.R(809705, 'executePrimary', executePrimary, false, false)) ? J$.G(809729, J$.G(809721, J$.R(809713, 'op', op, false, false), 'o'), 'options') : J$._()) ? J$.G(809761, J$.G(809753, J$.G(809745, J$.R(809737, 'op', op, false, false), 'o'), 'options'), 'readPreference') : J$._()) ? J$.B(50442, '!==', J$.M(809817, J$.R(809769, 'primaryOptions', primaryOptions, false, true), 'indexOf', false)(J$.G(809809, J$.G(809801, J$.G(809793, J$.G(809785, J$.R(809777, 'op', op, false, false), 'o'), 'options'), 'readPreference'), 'mode')), J$.U(50434, '-', J$.T(809825, 1, 22, false))) : J$._())) {
                                        J$.M(809905, J$.G(809865, J$.G(809841, J$.R(809833, 'op', op, false, false), 'o'), J$.G(809857, J$.R(809849, 'op', op, false, false), 'm')), 'apply', false)(J$.G(809881, J$.R(809873, 'op', op, false, false), 'o'), J$.G(809897, J$.R(809889, 'op', op, false, false), 'p'));
                                    } else if (J$.C(51920, J$.C(51912, J$.C(51904, J$.C(51896, J$.C(51888, J$.U(50450, '!', J$.R(809913, 'executePrimary', executePrimary, false, false))) ? J$.R(809921, 'executeSecondary', executeSecondary, false, false) : J$._()) ? J$.G(809945, J$.G(809937, J$.R(809929, 'op', op, false, false), 'o'), 'options') : J$._()) ? J$.G(809977, J$.G(809969, J$.G(809961, J$.R(809953, 'op', op, false, false), 'o'), 'options'), 'readPreference') : J$._()) ? J$.B(50466, '!==', J$.M(810033, J$.R(809985, 'secondaryOptions', secondaryOptions, false, true), 'indexOf', false)(J$.G(810025, J$.G(810017, J$.G(810009, J$.G(810001, J$.R(809993, 'op', op, false, false), 'o'), 'options'), 'readPreference'), 'mode')), J$.U(50458, '-', J$.T(810041, 1, 22, false))) : J$._())) {
                                        J$.M(810121, J$.G(810081, J$.G(810057, J$.R(810049, 'op', op, false, false), 'o'), J$.G(810073, J$.R(810065, 'op', op, false, false), 'm')), 'apply', false)(J$.G(810097, J$.R(810089, 'op', op, false, false), 'o'), J$.G(810113, J$.R(810105, 'op', op, false, false), 'p'));
                                    }
                                } else if (J$.C(52032, J$.B(50474, '===', J$.G(810137, J$.R(810129, 'op', op, false, false), 't'), J$.T(810145, 'auth', 21, false)))) {
                                    J$.M(810241, J$.G(810193, J$.G(810169, J$.G(810161, J$.R(810153, 'this', this, false, false), 's'), 'topology'), J$.G(810185, J$.R(810177, 'op', op, false, false), 't')), 'apply', false)(J$.G(810217, J$.G(810209, J$.R(810201, 'this', this, false, false), 's'), 'topology'), J$.G(810233, J$.R(810225, 'op', op, false, false), 'o'));
                                } else {
                                    if (J$.C(52024, J$.C(51944, J$.R(810249, 'executePrimary', executePrimary, false, false)) ? J$.R(810257, 'executeSecondary', executeSecondary, false, false) : J$._())) {
                                        J$.M(810369, J$.G(810281, J$.G(810273, J$.R(810265, 'this', this, false, false), 's'), 'topology'), J$.G(810297, J$.R(810289, 'op', op, false, false), 't'), false)(J$.G(810313, J$.R(810305, 'op', op, false, false), 'n'), J$.G(810329, J$.R(810321, 'op', op, false, false), 'o'), J$.G(810345, J$.R(810337, 'op', op, false, false), 'op'), J$.G(810361, J$.R(810353, 'op', op, false, false), 'c'));
                                    } else if (J$.C(52016, J$.C(51968, J$.C(51960, J$.C(51952, J$.R(810377, 'executePrimary', executePrimary, false, false)) ? J$.G(810393, J$.R(810385, 'op', op, false, false), 'op') : J$._()) ? J$.G(810417, J$.G(810409, J$.R(810401, 'op', op, false, false), 'op'), 'readPreference') : J$._()) ? J$.B(50490, '!==', J$.M(810465, J$.R(810425, 'primaryOptions', primaryOptions, false, true), 'indexOf', false)(J$.G(810457, J$.G(810449, J$.G(810441, J$.R(810433, 'op', op, false, false), 'op'), 'readPreference'), 'mode')), J$.U(50482, '-', J$.T(810473, 1, 22, false))) : J$._())) {
                                        J$.M(810585, J$.G(810497, J$.G(810489, J$.R(810481, 'this', this, false, false), 's'), 'topology'), J$.G(810513, J$.R(810505, 'op', op, false, false), 't'), false)(J$.G(810529, J$.R(810521, 'op', op, false, false), 'n'), J$.G(810545, J$.R(810537, 'op', op, false, false), 'o'), J$.G(810561, J$.R(810553, 'op', op, false, false), 'op'), J$.G(810577, J$.R(810569, 'op', op, false, false), 'c'));
                                    } else if (J$.C(52008, J$.C(52000, J$.C(51992, J$.C(51984, J$.C(51976, J$.U(50498, '!', J$.R(810593, 'executePrimary', executePrimary, false, false))) ? J$.R(810601, 'executeSecondary', executeSecondary, false, false) : J$._()) ? J$.G(810617, J$.R(810609, 'op', op, false, false), 'op') : J$._()) ? J$.G(810641, J$.G(810633, J$.R(810625, 'op', op, false, false), 'op'), 'readPreference') : J$._()) ? J$.B(50514, '!==', J$.M(810689, J$.R(810649, 'secondaryOptions', secondaryOptions, false, true), 'indexOf', false)(J$.G(810681, J$.G(810673, J$.G(810665, J$.R(810657, 'op', op, false, false), 'op'), 'readPreference'), 'mode')), J$.U(50506, '-', J$.T(810697, 1, 22, false))) : J$._())) {
                                        J$.M(810809, J$.G(810721, J$.G(810713, J$.R(810705, 'this', this, false, false), 's'), 'topology'), J$.G(810737, J$.R(810729, 'op', op, false, false), 't'), false)(J$.G(810753, J$.R(810745, 'op', op, false, false), 'n'), J$.G(810769, J$.R(810761, 'op', op, false, false), 'o'), J$.G(810785, J$.R(810777, 'op', op, false, false), 'op'), J$.G(810801, J$.R(810793, 'op', op, false, false), 'c'));
                                    }
                                }
                            }
                        } catch (J$e) {
                            J$.Ex(816505, J$e);
                        } finally {
                            if (J$.Fr(816513))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(810961, J$.G(810897, J$.R(810889, 'Store', Store, false, true), 'prototype'), 'all', J$.T(810953, function () {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(810937, arguments.callee, this, arguments);
                            arguments = J$.N(810945, 'arguments', arguments, true, false, false);
                            return J$.Rt(810929, J$.G(810921, J$.G(810913, J$.R(810905, 'this', this, false, false), 's'), 'storedOps'));
                        } catch (J$e) {
                            J$.Ex(816521, J$e);
                        } finally {
                            if (J$.Fr(816529))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var ServerCapabilities = J$.W(812185, 'ServerCapabilities', J$.T(812177, function ServerCapabilities(ismaster) {
                jalangiLabel22:
                    while (true) {
                        try {
                            J$.Fe(812073, arguments.callee, this, arguments);
                            arguments = J$.N(812081, 'arguments', arguments, true, false, false);
                            ismaster = J$.N(812089, 'ismaster', ismaster, true, false, false);
                            J$.N(812097, 'setup_get_property', setup_get_property, false, false, false);
                            J$.N(812105, 'aggregationCursor', aggregationCursor, false, false, false);
                            J$.N(812113, 'writeCommands', writeCommands, false, false, false);
                            J$.N(812121, 'textSearch', textSearch, false, false, false);
                            J$.N(812129, 'authCommands', authCommands, false, false, false);
                            J$.N(812137, 'listCollections', listCollections, false, false, false);
                            J$.N(812145, 'listIndexes', listIndexes, false, false, false);
                            J$.N(812153, 'maxNumberOfDocsInBatch', maxNumberOfDocsInBatch, false, false, false);
                            J$.N(812161, 'commandsTakeWriteConcern', commandsTakeWriteConcern, false, false, false);
                            J$.N(812169, 'commandsTakeCollation', commandsTakeCollation, false, false, false);
                            var setup_get_property = J$.W(811105, 'setup_get_property', J$.T(811097, function setup_get_property(object, name, value) {
                                jalangiLabel21:
                                    while (true) {
                                        try {
                                            J$.Fe(811057, arguments.callee, this, arguments);
                                            arguments = J$.N(811065, 'arguments', arguments, true, false, false);
                                            object = J$.N(811073, 'object', object, true, false, false);
                                            name = J$.N(811081, 'name', name, true, false, false);
                                            value = J$.N(811089, 'value', value, true, false, false);
                                            J$.M(811049, J$.I(typeof Object === 'undefined' ? Object = J$.R(810969, 'Object', undefined, true, true) : Object = J$.R(810969, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(810977, 'object', object, false, false), J$.R(810985, 'name', name, false, false), J$.T(811041, {
                                                enumerable: J$.T(810993, true, 23, false),
                                                get: J$.T(811033, function get() {
                                                    jalangiLabel20:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(811017, arguments.callee, this, arguments);
                                                                arguments = J$.N(811025, 'arguments', arguments, true, false, false);
                                                                return J$.Rt(811009, J$.R(811001, 'value', value, false, false));
                                                            } catch (J$e) {
                                                                J$.Ex(816537, J$e);
                                                            } finally {
                                                                if (J$.Fr(816545))
                                                                    continue jalangiLabel20;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false)
                                            }, 11, false));
                                        } catch (J$e) {
                                            J$.Ex(816553, J$e);
                                        } finally {
                                            if (J$.Fr(816561))
                                                continue jalangiLabel21;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), setup_get_property, false, false);
                            var aggregationCursor = J$.W(811121, 'aggregationCursor', J$.T(811113, false, 23, false), aggregationCursor, false, false);
                            var writeCommands = J$.W(811137, 'writeCommands', J$.T(811129, false, 23, false), writeCommands, false, false);
                            var textSearch = J$.W(811153, 'textSearch', J$.T(811145, false, 23, false), textSearch, false, false);
                            var authCommands = J$.W(811169, 'authCommands', J$.T(811161, false, 23, false), authCommands, false, false);
                            var listCollections = J$.W(811185, 'listCollections', J$.T(811177, false, 23, false), listCollections, false, false);
                            var listIndexes = J$.W(811201, 'listIndexes', J$.T(811193, false, 23, false), listIndexes, false, false);
                            var maxNumberOfDocsInBatch = J$.W(811233, 'maxNumberOfDocsInBatch', J$.C(52056, J$.G(811217, J$.R(811209, 'ismaster', ismaster, false, false), 'maxWriteBatchSize')) ? J$._() : J$.T(811225, 1000, 22, false), maxNumberOfDocsInBatch, false, false);
                            var commandsTakeWriteConcern = J$.W(811249, 'commandsTakeWriteConcern', J$.T(811241, false, 23, false), commandsTakeWriteConcern, false, false);
                            var commandsTakeCollation = J$.W(811265, 'commandsTakeCollation', J$.T(811257, false, 23, false), commandsTakeCollation, false, false);
                            if (J$.C(52064, J$.B(50522, '>=', J$.G(811281, J$.R(811273, 'ismaster', ismaster, false, false), 'minWireVersion'), J$.T(811289, 0, 22, false)))) {
                                textSearch = J$.W(811305, 'textSearch', J$.T(811297, true, 23, false), textSearch, false, false);
                            }
                            if (J$.C(52072, J$.B(50530, '>=', J$.G(811321, J$.R(811313, 'ismaster', ismaster, false, false), 'maxWireVersion'), J$.T(811329, 1, 22, false)))) {
                                aggregationCursor = J$.W(811345, 'aggregationCursor', J$.T(811337, true, 23, false), aggregationCursor, false, false);
                                authCommands = J$.W(811361, 'authCommands', J$.T(811353, true, 23, false), authCommands, false, false);
                            }
                            if (J$.C(52080, J$.B(50538, '>=', J$.G(811377, J$.R(811369, 'ismaster', ismaster, false, false), 'maxWireVersion'), J$.T(811385, 2, 22, false)))) {
                                writeCommands = J$.W(811401, 'writeCommands', J$.T(811393, true, 23, false), writeCommands, false, false);
                            }
                            if (J$.C(52088, J$.B(50546, '>=', J$.G(811417, J$.R(811409, 'ismaster', ismaster, false, false), 'maxWireVersion'), J$.T(811425, 3, 22, false)))) {
                                listCollections = J$.W(811441, 'listCollections', J$.T(811433, true, 23, false), listCollections, false, false);
                                listIndexes = J$.W(811457, 'listIndexes', J$.T(811449, true, 23, false), listIndexes, false, false);
                            }
                            if (J$.C(52096, J$.B(50554, '>=', J$.G(811473, J$.R(811465, 'ismaster', ismaster, false, false), 'maxWireVersion'), J$.T(811481, 5, 22, false)))) {
                                commandsTakeWriteConcern = J$.W(811497, 'commandsTakeWriteConcern', J$.T(811489, true, 23, false), commandsTakeWriteConcern, false, false);
                                commandsTakeCollation = J$.W(811513, 'commandsTakeCollation', J$.T(811505, true, 23, false), commandsTakeCollation, false, false);
                            }
                            if (J$.C(52104, J$.B(50562, '==', J$.G(811529, J$.R(811521, 'ismaster', ismaster, false, false), 'minWireVersion'), J$.T(811537, null, 25, false)))) {
                                J$.P(811561, J$.R(811545, 'ismaster', ismaster, false, false), 'minWireVersion', J$.T(811553, 0, 22, false));
                            }
                            if (J$.C(52112, J$.B(50570, '==', J$.G(811577, J$.R(811569, 'ismaster', ismaster, false, false), 'maxWireVersion'), J$.T(811585, null, 25, false)))) {
                                J$.P(811609, J$.R(811593, 'ismaster', ismaster, false, false), 'maxWireVersion', J$.T(811601, 0, 22, false));
                            }
                            J$.F(811649, J$.R(811617, 'setup_get_property', setup_get_property, false, false), false)(J$.R(811625, 'this', this, false, false), J$.T(811633, 'hasAggregationCursor', 21, false), J$.R(811641, 'aggregationCursor', aggregationCursor, false, false));
                            J$.F(811689, J$.R(811657, 'setup_get_property', setup_get_property, false, false), false)(J$.R(811665, 'this', this, false, false), J$.T(811673, 'hasWriteCommands', 21, false), J$.R(811681, 'writeCommands', writeCommands, false, false));
                            J$.F(811729, J$.R(811697, 'setup_get_property', setup_get_property, false, false), false)(J$.R(811705, 'this', this, false, false), J$.T(811713, 'hasTextSearch', 21, false), J$.R(811721, 'textSearch', textSearch, false, false));
                            J$.F(811769, J$.R(811737, 'setup_get_property', setup_get_property, false, false), false)(J$.R(811745, 'this', this, false, false), J$.T(811753, 'hasAuthCommands', 21, false), J$.R(811761, 'authCommands', authCommands, false, false));
                            J$.F(811809, J$.R(811777, 'setup_get_property', setup_get_property, false, false), false)(J$.R(811785, 'this', this, false, false), J$.T(811793, 'hasListCollectionsCommand', 21, false), J$.R(811801, 'listCollections', listCollections, false, false));
                            J$.F(811849, J$.R(811817, 'setup_get_property', setup_get_property, false, false), false)(J$.R(811825, 'this', this, false, false), J$.T(811833, 'hasListIndexesCommand', 21, false), J$.R(811841, 'listIndexes', listIndexes, false, false));
                            J$.F(811897, J$.R(811857, 'setup_get_property', setup_get_property, false, false), false)(J$.R(811865, 'this', this, false, false), J$.T(811873, 'minWireVersion', 21, false), J$.G(811889, J$.R(811881, 'ismaster', ismaster, false, false), 'minWireVersion'));
                            J$.F(811945, J$.R(811905, 'setup_get_property', setup_get_property, false, false), false)(J$.R(811913, 'this', this, false, false), J$.T(811921, 'maxWireVersion', 21, false), J$.G(811937, J$.R(811929, 'ismaster', ismaster, false, false), 'maxWireVersion'));
                            J$.F(811985, J$.R(811953, 'setup_get_property', setup_get_property, false, false), false)(J$.R(811961, 'this', this, false, false), J$.T(811969, 'maxNumberOfDocsInBatch', 21, false), J$.R(811977, 'maxNumberOfDocsInBatch', maxNumberOfDocsInBatch, false, false));
                            J$.F(812025, J$.R(811993, 'setup_get_property', setup_get_property, false, false), false)(J$.R(812001, 'this', this, false, false), J$.T(812009, 'commandsTakeWriteConcern', 21, false), J$.R(812017, 'commandsTakeWriteConcern', commandsTakeWriteConcern, false, false));
                            J$.F(812065, J$.R(812033, 'setup_get_property', setup_get_property, false, false), false)(J$.R(812041, 'this', this, false, false), J$.T(812049, 'commandsTakeCollation', 21, false), J$.R(812057, 'commandsTakeCollation', commandsTakeCollation, false, false));
                        } catch (J$e) {
                            J$.Ex(816569, J$e);
                        } finally {
                            if (J$.Fr(816577))
                                continue jalangiLabel22;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), ServerCapabilities, false, true);
            var driverVersion = J$.W(812345, 'driverVersion', J$.G(812217, J$.F(812209, J$.I(typeof require === 'undefined' ? require = J$.R(812193, 'require', undefined, true, true) : require = J$.R(812193, 'require', require, true, true)), false)(J$.T(812201, '../../package.json', 21, false)), 'version'), driverVersion, false, true), nodejsversion = J$.W(812353, 'nodejsversion', J$.F(812273, J$.R(812225, 'f', f, false, true), false)(J$.T(812233, 'Node.js %s, %s', 21, false), J$.G(812249, J$.I(typeof process === 'undefined' ? process = J$.R(812241, 'process', undefined, true, true) : process = J$.R(812241, 'process', process, true, true)), 'version'), J$.M(812265, J$.R(812257, 'os', os, false, true), 'endianness', false)()), nodejsversion, false, true), type = J$.W(812361, 'type', J$.M(812289, J$.R(812281, 'os', os, false, true), 'type', false)(), type, false, true), name = J$.W(812369, 'name', J$.G(812305, J$.I(typeof process === 'undefined' ? process = J$.R(812297, 'process', undefined, true, true) : process = J$.R(812297, 'process', process, true, true)), 'platform'), name, false, true), architecture = J$.W(812377, 'architecture', J$.G(812321, J$.I(typeof process === 'undefined' ? process = J$.R(812313, 'process', undefined, true, true) : process = J$.R(812313, 'process', process, true, true)), 'arch'), architecture, false, true), release = J$.W(812385, 'release', J$.M(812337, J$.R(812329, 'os', os, false, true), 'release', false)(), release, false, true);
            var TopologyBase = J$.W(815361, 'TopologyBase', J$.F(815353, J$.T(815337, function (_EventEmitter) {
                jalangiLabel42:
                    while (true) {
                        try {
                            J$.Fe(815297, arguments.callee, this, arguments);
                            function TopologyBase() {
                                jalangiLabel23:
                                    while (true) {
                                        try {
                                            J$.Fe(812665, arguments.callee, this, arguments);
                                            arguments = J$.N(812673, 'arguments', arguments, true, false, false);
                                            J$.N(812681, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(812449, J$.R(812425, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(812433, 'this', this, false, false), J$.R(812441, 'TopologyBase', TopologyBase, false, false));
                                            _this = J$.W(812521, '_this', J$.F(812513, J$.R(812457, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(812465, 'this', this, false, false), J$.M(812505, J$.F(812489, J$.R(812473, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(812481, 'TopologyBase', TopologyBase, false, false)), 'call', false)(J$.R(812497, 'this', this, false, false))), _this, false, false);
                                            J$.P(812617, J$.R(812529, '_this', _this, false, false), 'clientInfo', J$.T(812609, {
                                                driver: J$.T(812553, {
                                                    name: J$.T(812537, 'nodejs', 21, false),
                                                    version: J$.R(812545, 'driverVersion', driverVersion, false, true)
                                                }, 11, false),
                                                os: J$.T(812593, {
                                                    type: J$.R(812561, 'type', type, false, true),
                                                    name: J$.R(812569, 'name', name, false, true),
                                                    architecture: J$.R(812577, 'architecture', architecture, false, true),
                                                    version: J$.R(812585, 'release', release, false, true)
                                                }, 11, false),
                                                platform: J$.R(812601, 'nodejsversion', nodejsversion, false, true)
                                            }, 11, false));
                                            J$.M(812641, J$.R(812625, '_this', _this, false, false), 'setMaxListeners', false)(J$.T(812633, Infinity, 22, false));
                                            return J$.Rt(812657, J$.R(812649, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(816585, J$e);
                                        } finally {
                                            if (J$.Fr(816593))
                                                continue jalangiLabel23;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(815305, 'arguments', arguments, true, false, false);
                            _EventEmitter = J$.N(815313, '_EventEmitter', _EventEmitter, true, false, false);
                            TopologyBase = J$.N(815329, 'TopologyBase', J$.T(815321, TopologyBase, 12, false), true, false, false);
                            J$.F(812417, J$.R(812393, '_inherits', _inherits, false, true), false)(J$.R(812401, 'TopologyBase', TopologyBase, false, false), J$.R(812409, '_EventEmitter', _EventEmitter, false, false));
                            J$.F(815273, J$.R(812689, '_createClass', _createClass, false, true), false)(J$.R(812697, 'TopologyBase', TopologyBase, false, false), J$.T(815265, [
                                J$.T(812769, {
                                    key: J$.T(812705, 'hasSessionSupport', 21, false),
                                    value: J$.T(812761, function hasSessionSupport() {
                                        jalangiLabel24:
                                            while (true) {
                                                try {
                                                    J$.Fe(812745, arguments.callee, this, arguments);
                                                    arguments = J$.N(812753, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(812737, J$.B(50578, '!=', J$.G(812721, J$.R(812713, 'this', this, false, false), 'logicalSessionTimeoutMinutes'), J$.T(812729, null, 25, false)));
                                                } catch (J$e) {
                                                    J$.Ex(816601, J$e);
                                                } finally {
                                                    if (J$.Fr(816609))
                                                        continue jalangiLabel24;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(813081, {
                                    key: J$.T(812777, 'startSession', 21, false),
                                    value: J$.T(813073, function startSession(options, clientOptions) {
                                        jalangiLabel26:
                                            while (true) {
                                                try {
                                                    J$.Fe(813025, arguments.callee, this, arguments);
                                                    arguments = J$.N(813033, 'arguments', arguments, true, false, false);
                                                    options = J$.N(813041, 'options', options, true, false, false);
                                                    clientOptions = J$.N(813049, 'clientOptions', clientOptions, true, false, false);
                                                    J$.N(813057, '_this2', _this2, false, false, false);
                                                    J$.N(813065, 'session', session, false, false, false);
                                                    var _this2 = J$.W(812793, '_this2', J$.R(812785, 'this', this, false, false), _this2, false, false);
                                                    var session = J$.W(812865, 'session', J$.F(812857, J$.R(812801, 'ClientSession', ClientSession, false, true), true)(J$.R(812809, 'this', this, false, false), J$.G(812833, J$.G(812825, J$.R(812817, 'this', this, false, false), 's'), 'sessionPool'), J$.R(812841, 'options', options, false, false), J$.R(812849, 'clientOptions', clientOptions, false, false)), session, false, false);
                                                    J$.M(812961, J$.R(812873, 'session', session, false, false), 'once', false)(J$.T(812881, 'ended', 21, false), J$.T(812953, function () {
                                                        jalangiLabel25:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(812937, arguments.callee, this, arguments);
                                                                    arguments = J$.N(812945, 'arguments', arguments, true, false, false);
                                                                    J$.M(812929, J$.G(812905, J$.G(812897, J$.R(812889, '_this2', _this2, false, false), 's'), 'sessions'), J$.T(812913, 'delete', 21, false), false)(J$.R(812921, 'session', session, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(816617, J$e);
                                                                } finally {
                                                                    if (J$.Fr(816625))
                                                                        continue jalangiLabel25;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                    J$.M(813001, J$.G(812985, J$.G(812977, J$.R(812969, 'this', this, false, false), 's'), 'sessions'), 'add', false)(J$.R(812993, 'session', session, false, false));
                                                    return J$.Rt(813017, J$.R(813009, 'session', session, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(816633, J$e);
                                                } finally {
                                                    if (J$.Fr(816641))
                                                        continue jalangiLabel26;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(813193, {
                                    key: J$.T(813089, 'endSessions', 21, false),
                                    value: J$.T(813185, function endSessions(sessions, callback) {
                                        jalangiLabel27:
                                            while (true) {
                                                try {
                                                    J$.Fe(813153, arguments.callee, this, arguments);
                                                    arguments = J$.N(813161, 'arguments', arguments, true, false, false);
                                                    sessions = J$.N(813169, 'sessions', sessions, true, false, false);
                                                    callback = J$.N(813177, 'callback', callback, true, false, false);
                                                    return J$.Rt(813145, J$.M(813137, J$.G(813113, J$.G(813105, J$.R(813097, 'this', this, false, false), 's'), 'coreTopology'), 'endSessions', false)(J$.R(813121, 'sessions', sessions, false, false), J$.R(813129, 'callback', callback, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(816649, J$e);
                                                } finally {
                                                    if (J$.Fr(816657))
                                                        continue jalangiLabel27;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(813449, {
                                    key: J$.T(813201, 'capabilities', 21, false),
                                    value: J$.T(813441, function capabilities() {
                                        jalangiLabel28:
                                            while (true) {
                                                try {
                                                    J$.Fe(813425, arguments.callee, this, arguments);
                                                    arguments = J$.N(813433, 'arguments', arguments, true, false, false);
                                                    if (J$.C(52120, J$.G(813225, J$.G(813217, J$.R(813209, 'this', this, false, false), 's'), 'sCapabilities')))
                                                        return J$.Rt(813257, J$.G(813249, J$.G(813241, J$.R(813233, 'this', this, false, false), 's'), 'sCapabilities'));
                                                    if (J$.C(52128, J$.B(50586, '==', J$.M(813289, J$.G(813281, J$.G(813273, J$.R(813265, 'this', this, false, false), 's'), 'coreTopology'), 'lastIsMaster', false)(), J$.T(813297, null, 25, false))))
                                                        return J$.Rt(813313, J$.T(813305, null, 25, false));
                                                    J$.P(813385, J$.G(813329, J$.R(813321, 'this', this, false, false), 's'), 'sCapabilities', J$.F(813377, J$.R(813337, 'ServerCapabilities', ServerCapabilities, false, true), true)(J$.M(813369, J$.G(813361, J$.G(813353, J$.R(813345, 'this', this, false, false), 's'), 'coreTopology'), 'lastIsMaster', false)()));
                                                    return J$.Rt(813417, J$.G(813409, J$.G(813401, J$.R(813393, 'this', this, false, false), 's'), 'sCapabilities'));
                                                } catch (J$e) {
                                                    J$.Ex(816665, J$e);
                                                } finally {
                                                    if (J$.Fr(816673))
                                                        continue jalangiLabel28;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(813609, {
                                    key: J$.T(813457, 'command', 21, false),
                                    value: J$.T(813601, function command(ns, cmd, options, callback) {
                                        jalangiLabel29:
                                            while (true) {
                                                try {
                                                    J$.Fe(813553, arguments.callee, this, arguments);
                                                    arguments = J$.N(813561, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(813569, 'ns', ns, true, false, false);
                                                    cmd = J$.N(813577, 'cmd', cmd, true, false, false);
                                                    options = J$.N(813585, 'options', options, true, false, false);
                                                    callback = J$.N(813593, 'callback', callback, true, false, false);
                                                    J$.M(813545, J$.G(813481, J$.G(813473, J$.R(813465, 'this', this, false, false), 's'), 'coreTopology'), 'command', false)(J$.M(813497, J$.R(813489, 'ns', ns, false, false), 'toString', false)(), J$.R(813505, 'cmd', cmd, false, false), J$.F(813529, J$.R(813513, 'translateReadPreference', translateReadPreference, false, true), false)(J$.R(813521, 'options', options, false, false)), J$.R(813537, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(816681, J$e);
                                                } finally {
                                                    if (J$.Fr(816689))
                                                        continue jalangiLabel29;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(813753, {
                                    key: J$.T(813617, 'insert', 21, false),
                                    value: J$.T(813745, function insert(ns, ops, options, callback) {
                                        jalangiLabel30:
                                            while (true) {
                                                try {
                                                    J$.Fe(813697, arguments.callee, this, arguments);
                                                    arguments = J$.N(813705, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(813713, 'ns', ns, true, false, false);
                                                    ops = J$.N(813721, 'ops', ops, true, false, false);
                                                    options = J$.N(813729, 'options', options, true, false, false);
                                                    callback = J$.N(813737, 'callback', callback, true, false, false);
                                                    J$.M(813689, J$.G(813641, J$.G(813633, J$.R(813625, 'this', this, false, false), 's'), 'coreTopology'), 'insert', false)(J$.M(813657, J$.R(813649, 'ns', ns, false, false), 'toString', false)(), J$.R(813665, 'ops', ops, false, false), J$.R(813673, 'options', options, false, false), J$.R(813681, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(816697, J$e);
                                                } finally {
                                                    if (J$.Fr(816705))
                                                        continue jalangiLabel30;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(813897, {
                                    key: J$.T(813761, 'update', 21, false),
                                    value: J$.T(813889, function update(ns, ops, options, callback) {
                                        jalangiLabel31:
                                            while (true) {
                                                try {
                                                    J$.Fe(813841, arguments.callee, this, arguments);
                                                    arguments = J$.N(813849, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(813857, 'ns', ns, true, false, false);
                                                    ops = J$.N(813865, 'ops', ops, true, false, false);
                                                    options = J$.N(813873, 'options', options, true, false, false);
                                                    callback = J$.N(813881, 'callback', callback, true, false, false);
                                                    J$.M(813833, J$.G(813785, J$.G(813777, J$.R(813769, 'this', this, false, false), 's'), 'coreTopology'), 'update', false)(J$.M(813801, J$.R(813793, 'ns', ns, false, false), 'toString', false)(), J$.R(813809, 'ops', ops, false, false), J$.R(813817, 'options', options, false, false), J$.R(813825, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(816713, J$e);
                                                } finally {
                                                    if (J$.Fr(816721))
                                                        continue jalangiLabel31;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(814041, {
                                    key: J$.T(813905, 'remove', 21, false),
                                    value: J$.T(814033, function remove(ns, ops, options, callback) {
                                        jalangiLabel32:
                                            while (true) {
                                                try {
                                                    J$.Fe(813985, arguments.callee, this, arguments);
                                                    arguments = J$.N(813993, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(814001, 'ns', ns, true, false, false);
                                                    ops = J$.N(814009, 'ops', ops, true, false, false);
                                                    options = J$.N(814017, 'options', options, true, false, false);
                                                    callback = J$.N(814025, 'callback', callback, true, false, false);
                                                    J$.M(813977, J$.G(813929, J$.G(813921, J$.R(813913, 'this', this, false, false), 's'), 'coreTopology'), 'remove', false)(J$.M(813945, J$.R(813937, 'ns', ns, false, false), 'toString', false)(), J$.R(813953, 'ops', ops, false, false), J$.R(813961, 'options', options, false, false), J$.R(813969, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(816729, J$e);
                                                } finally {
                                                    if (J$.Fr(816737))
                                                        continue jalangiLabel32;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(814193, {
                                    key: J$.T(814049, 'isConnected', 21, false),
                                    value: J$.T(814185, function isConnected(options) {
                                        jalangiLabel33:
                                            while (true) {
                                                try {
                                                    J$.Fe(814161, arguments.callee, this, arguments);
                                                    arguments = J$.N(814169, 'arguments', arguments, true, false, false);
                                                    options = J$.N(814177, 'options', options, true, false, false);
                                                    options = J$.W(814073, 'options', J$.C(52136, J$.R(814057, 'options', options, false, false)) ? J$._() : J$.T(814065, {}, 11, false), options, false, false);
                                                    options = J$.W(814105, 'options', J$.F(814097, J$.R(814081, 'translateReadPreference', translateReadPreference, false, true), false)(J$.R(814089, 'options', options, false, false)), options, false, false);
                                                    return J$.Rt(814153, J$.M(814145, J$.G(814129, J$.G(814121, J$.R(814113, 'this', this, false, false), 's'), 'coreTopology'), 'isConnected', false)(J$.R(814137, 'options', options, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(816745, J$e);
                                                } finally {
                                                    if (J$.Fr(816753))
                                                        continue jalangiLabel33;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(814273, {
                                    key: J$.T(814201, 'isDestroyed', 21, false),
                                    value: J$.T(814265, function isDestroyed() {
                                        jalangiLabel34:
                                            while (true) {
                                                try {
                                                    J$.Fe(814249, arguments.callee, this, arguments);
                                                    arguments = J$.N(814257, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(814241, J$.M(814233, J$.G(814225, J$.G(814217, J$.R(814209, 'this', this, false, false), 's'), 'coreTopology'), 'isDestroyed', false)());
                                                } catch (J$e) {
                                                    J$.Ex(816761, J$e);
                                                } finally {
                                                    if (J$.Fr(816769))
                                                        continue jalangiLabel34;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(814521, {
                                    key: J$.T(814281, 'cursor', 21, false),
                                    value: J$.T(814513, function cursor(ns, cmd, options) {
                                        jalangiLabel35:
                                            while (true) {
                                                try {
                                                    J$.Fe(814473, arguments.callee, this, arguments);
                                                    arguments = J$.N(814481, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(814489, 'ns', ns, true, false, false);
                                                    cmd = J$.N(814497, 'cmd', cmd, true, false, false);
                                                    options = J$.N(814505, 'options', options, true, false, false);
                                                    options = J$.W(814305, 'options', J$.C(52144, J$.R(814289, 'options', options, false, false)) ? J$._() : J$.T(814297, {}, 11, false), options, false, false);
                                                    options = J$.W(814337, 'options', J$.F(814329, J$.R(814313, 'translateReadPreference', translateReadPreference, false, true), false)(J$.R(814321, 'options', options, false, false)), options, false, false);
                                                    J$.P(814377, J$.R(814345, 'options', options, false, false), 'disconnectHandler', J$.G(814369, J$.G(814361, J$.R(814353, 'this', this, false, false), 's'), 'store'));
                                                    J$.P(814401, J$.R(814385, 'options', options, false, false), 'topology', J$.R(814393, 'this', this, false, false));
                                                    return J$.Rt(814465, J$.M(814457, J$.G(814425, J$.G(814417, J$.R(814409, 'this', this, false, false), 's'), 'coreTopology'), 'cursor', false)(J$.R(814433, 'ns', ns, false, false), J$.R(814441, 'cmd', cmd, false, false), J$.R(814449, 'options', options, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(816777, J$e);
                                                } finally {
                                                    if (J$.Fr(816785))
                                                        continue jalangiLabel35;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(814601, {
                                    key: J$.T(814529, 'lastIsMaster', 21, false),
                                    value: J$.T(814593, function lastIsMaster() {
                                        jalangiLabel36:
                                            while (true) {
                                                try {
                                                    J$.Fe(814577, arguments.callee, this, arguments);
                                                    arguments = J$.N(814585, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(814569, J$.M(814561, J$.G(814553, J$.G(814545, J$.R(814537, 'this', this, false, false), 's'), 'coreTopology'), 'lastIsMaster', false)());
                                                } catch (J$e) {
                                                    J$.Ex(816793, J$e);
                                                } finally {
                                                    if (J$.Fr(816801))
                                                        continue jalangiLabel36;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(814729, {
                                    key: J$.T(814609, 'selectServer', 21, false),
                                    value: J$.T(814721, function selectServer(selector, options, callback) {
                                        jalangiLabel37:
                                            while (true) {
                                                try {
                                                    J$.Fe(814681, arguments.callee, this, arguments);
                                                    arguments = J$.N(814689, 'arguments', arguments, true, false, false);
                                                    selector = J$.N(814697, 'selector', selector, true, false, false);
                                                    options = J$.N(814705, 'options', options, true, false, false);
                                                    callback = J$.N(814713, 'callback', callback, true, false, false);
                                                    return J$.Rt(814673, J$.M(814665, J$.G(814633, J$.G(814625, J$.R(814617, 'this', this, false, false), 's'), 'coreTopology'), 'selectServer', false)(J$.R(814641, 'selector', selector, false, false), J$.R(814649, 'options', options, false, false), J$.R(814657, 'callback', callback, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(816809, J$e);
                                                } finally {
                                                    if (J$.Fr(816817))
                                                        continue jalangiLabel37;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(814809, {
                                    key: J$.T(814737, 'unref', 21, false),
                                    value: J$.T(814801, function unref() {
                                        jalangiLabel38:
                                            while (true) {
                                                try {
                                                    J$.Fe(814785, arguments.callee, this, arguments);
                                                    arguments = J$.N(814793, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(814777, J$.M(814769, J$.G(814761, J$.G(814753, J$.R(814745, 'this', this, false, false), 's'), 'coreTopology'), 'unref', false)());
                                                } catch (J$e) {
                                                    J$.Ex(816825, J$e);
                                                } finally {
                                                    if (J$.Fr(816833))
                                                        continue jalangiLabel38;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(814889, {
                                    key: J$.T(814817, 'connections', 21, false),
                                    value: J$.T(814881, function connections() {
                                        jalangiLabel39:
                                            while (true) {
                                                try {
                                                    J$.Fe(814865, arguments.callee, this, arguments);
                                                    arguments = J$.N(814873, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(814857, J$.M(814849, J$.G(814841, J$.G(814833, J$.R(814825, 'this', this, false, false), 's'), 'coreTopology'), 'connections', false)());
                                                } catch (J$e) {
                                                    J$.Ex(816841, J$e);
                                                } finally {
                                                    if (J$.Fr(816849))
                                                        continue jalangiLabel39;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(815257, {
                                    key: J$.T(814897, 'close', 21, false),
                                    value: J$.T(815249, function close(forceClosed, callback) {
                                        jalangiLabel41:
                                            while (true) {
                                                try {
                                                    J$.Fe(815217, arguments.callee, this, arguments);
                                                    arguments = J$.N(815225, 'arguments', arguments, true, false, false);
                                                    forceClosed = J$.N(815233, 'forceClosed', forceClosed, true, false, false);
                                                    callback = J$.N(815241, 'callback', callback, true, false, false);
                                                    J$.M(814985, J$.G(814921, J$.G(814913, J$.R(814905, 'this', this, false, false), 's'), 'sessions'), 'forEach', false)(J$.T(814977, function (session) {
                                                        jalangiLabel40:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(814953, arguments.callee, this, arguments);
                                                                    arguments = J$.N(814961, 'arguments', arguments, true, false, false);
                                                                    session = J$.N(814969, 'session', session, true, false, false);
                                                                    return J$.Rt(814945, J$.M(814937, J$.R(814929, 'session', session, false, false), 'endSession', false)());
                                                                } catch (J$e) {
                                                                    J$.Ex(816857, J$e);
                                                                } finally {
                                                                    if (J$.Fr(816865))
                                                                        continue jalangiLabel40;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                    if (J$.C(52152, J$.G(815009, J$.G(815001, J$.R(814993, 'this', this, false, false), 's'), 'sessionPool'))) {
                                                        J$.M(815041, J$.G(815033, J$.G(815025, J$.R(815017, 'this', this, false, false), 's'), 'sessionPool'), 'endAllPooledSessions', false)();
                                                    }
                                                    if (J$.C(52160, J$.B(50594, '===', J$.R(815049, 'forceClosed', forceClosed, false, false), J$.T(815057, true, 23, false)))) {
                                                        J$.P(815097, J$.G(815081, J$.G(815073, J$.R(815065, 'this', this, false, false), 's'), 'storeOptions'), 'force', J$.R(815089, 'forceClosed', forceClosed, false, false));
                                                        J$.M(815129, J$.G(815121, J$.G(815113, J$.R(815105, 'this', this, false, false), 's'), 'store'), 'flush', false)();
                                                    }
                                                    J$.M(815209, J$.G(815153, J$.G(815145, J$.R(815137, 'this', this, false, false), 's'), 'coreTopology'), 'destroy', false)(J$.T(815193, { force: J$.C(52168, J$.B(50610, '===', J$.U(50602, 'typeof', J$.R(815161, 'forceClosed', forceClosed, false, false)), J$.T(815169, 'boolean', 21, false))) ? J$.R(815177, 'forceClosed', forceClosed, false, false) : J$.T(815185, false, 23, false) }, 11, false), J$.R(815201, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(816873, J$e);
                                                } finally {
                                                    if (J$.Fr(816881))
                                                        continue jalangiLabel41;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(815289, J$.R(815281, 'TopologyBase', TopologyBase, false, false));
                        } catch (J$e) {
                            J$.Ex(816889, J$e);
                        } finally {
                            if (J$.Fr(816897))
                                continue jalangiLabel42;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(815345, 'EventEmitter', EventEmitter, false, true)), TopologyBase, false, true);
            J$.M(815489, J$.I(typeof Object === 'undefined' ? Object = J$.R(815369, 'Object', undefined, true, true) : Object = J$.R(815369, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(815385, J$.R(815377, 'TopologyBase', TopologyBase, false, true), 'prototype'), J$.T(815393, 'bson', 21, false), J$.T(815481, {
                enumerable: J$.T(815401, true, 23, false),
                get: J$.T(815473, function get() {
                    jalangiLabel43:
                        while (true) {
                            try {
                                J$.Fe(815457, arguments.callee, this, arguments);
                                arguments = J$.N(815465, 'arguments', arguments, true, false, false);
                                return J$.Rt(815449, J$.G(815441, J$.G(815433, J$.G(815425, J$.G(815417, J$.R(815409, 'this', this, false, false), 's'), 'coreTopology'), 's'), 'bson'));
                            } catch (J$e) {
                                J$.Ex(816905, J$e);
                            } finally {
                                if (J$.Fr(816913))
                                    continue jalangiLabel43;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.M(815609, J$.I(typeof Object === 'undefined' ? Object = J$.R(815497, 'Object', undefined, true, true) : Object = J$.R(815497, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(815513, J$.R(815505, 'TopologyBase', TopologyBase, false, true), 'prototype'), J$.T(815521, 'parserType', 21, false), J$.T(815601, {
                enumerable: J$.T(815529, true, 23, false),
                get: J$.T(815593, function get() {
                    jalangiLabel44:
                        while (true) {
                            try {
                                J$.Fe(815577, arguments.callee, this, arguments);
                                arguments = J$.N(815585, 'arguments', arguments, true, false, false);
                                return J$.Rt(815569, J$.G(815561, J$.G(815553, J$.G(815545, J$.R(815537, 'this', this, false, false), 's'), 'coreTopology'), 'parserType'));
                            } catch (J$e) {
                                J$.Ex(816921, J$e);
                            } finally {
                                if (J$.Fr(816929))
                                    continue jalangiLabel44;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.M(815729, J$.I(typeof Object === 'undefined' ? Object = J$.R(815617, 'Object', undefined, true, true) : Object = J$.R(815617, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(815633, J$.R(815625, 'TopologyBase', TopologyBase, false, true), 'prototype'), J$.T(815641, 'logicalSessionTimeoutMinutes', 21, false), J$.T(815721, {
                enumerable: J$.T(815649, true, 23, false),
                get: J$.T(815713, function get() {
                    jalangiLabel45:
                        while (true) {
                            try {
                                J$.Fe(815697, arguments.callee, this, arguments);
                                arguments = J$.N(815705, 'arguments', arguments, true, false, false);
                                return J$.Rt(815689, J$.G(815681, J$.G(815673, J$.G(815665, J$.R(815657, 'this', this, false, false), 's'), 'coreTopology'), 'logicalSessionTimeoutMinutes'));
                            } catch (J$e) {
                                J$.Ex(816937, J$e);
                            } finally {
                                if (J$.Fr(816945))
                                    continue jalangiLabel45;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.M(815849, J$.I(typeof Object === 'undefined' ? Object = J$.R(815737, 'Object', undefined, true, true) : Object = J$.R(815737, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(815753, J$.R(815745, 'TopologyBase', TopologyBase, false, true), 'prototype'), J$.T(815761, 'type', 21, false), J$.T(815841, {
                enumerable: J$.T(815769, true, 23, false),
                get: J$.T(815833, function get() {
                    jalangiLabel46:
                        while (true) {
                            try {
                                J$.Fe(815817, arguments.callee, this, arguments);
                                arguments = J$.N(815825, 'arguments', arguments, true, false, false);
                                return J$.Rt(815809, J$.G(815801, J$.G(815793, J$.G(815785, J$.R(815777, 'this', this, false, false), 's'), 'coreTopology'), 'type'));
                            } catch (J$e) {
                                J$.Ex(816953, J$e);
                            } finally {
                                if (J$.Fr(816961))
                                    continue jalangiLabel46;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.P(815873, J$.I(typeof exports === 'undefined' ? exports = J$.R(815857, 'exports', undefined, true, true) : exports = J$.R(815857, 'exports', exports, true, true)), 'Store', J$.R(815865, 'Store', Store, false, true));
            J$.P(815897, J$.I(typeof exports === 'undefined' ? exports = J$.R(815881, 'exports', undefined, true, true) : exports = J$.R(815881, 'exports', exports, true, true)), 'ServerCapabilities', J$.R(815889, 'ServerCapabilities', ServerCapabilities, false, true));
            J$.P(815921, J$.I(typeof exports === 'undefined' ? exports = J$.R(815905, 'exports', undefined, true, true) : exports = J$.R(815905, 'exports', exports, true, true)), 'TopologyBase', J$.R(815913, 'TopologyBase', TopologyBase, false, true));
        } catch (J$e) {
            J$.Ex(816969, J$e);
        } finally {
            if (J$.Sr(816977))
                continue jalangiLabel47;
            else
                break jalangiLabel47;
        }
    }
// JALANGI DO NOT INSTRUMENT
