// jpv/expressions.js
J$.noInstrEval = false;
jalangiLabel0:
    while (true) {
        try {
            J$.Se(2960993, '/mnt/data/fxiao/HiPar/outputs/target_cache/TestJpv/node_modules/jpv/expressions.js');
            J$.T(2960857, 'use strict', 21, false);
            J$.P(2960985, J$.I(typeof module === 'undefined' ? module = J$.R(2960865, 'module', undefined, true, true) : module = J$.R(2960865, 'module', module, true, true)), 'exports', J$.T(2960977, {
                exist: J$.T(2960873, /.*/, 14, false),
                empty: J$.T(2960881, /^$/, 14, false),
                boolean: J$.T(2960889, /^(true|false)$/i, 14, false),
                double: J$.T(2960897, /^-?\d*(\.\d+)?$/, 14, false),
                float: J$.T(2960905, /^-?\d*(\.\d+)?$/, 14, false),
                naturalNumber: J$.T(2960913, /^[1-9][0-9]*$/, 14, false),
                number: J$.T(2960921, /^[0-9]+$/i, 14, false),
                integer: J$.T(2960929, /^\d+$/, 14, false),
                url: J$.T(2960937, /^https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)$/i, 14, false),
                alphaNumeric: J$.T(2960945, /^([a-zA-Z0-9_])+$/i, 14, false),
                email: J$.T(2960953, /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/i, 14, false),
                date: J$.T(2960961, /^\d{4}\-(0?[1-9]|1[012])\-(0?[1-9]|[12][0-9]|3[01])$/, 14, false),
                datetime: J$.T(2960969, /^([\+-]?\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00)([\.,]\d+(?!:))?)?(\17[0-5]\d([\.,]\d+)?)?([zZ]|([\+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$/, 14, false)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(2961001, J$e);
        } finally {
            if (J$.Sr(2961009))
                continue jalangiLabel0;
            else
                break jalangiLabel0;
        }
    }
// JALANGI DO NOT INSTRUMENT
