// node-abi/index.js
J$.noInstrEval = false;
jalangiLabel6:
    while (true) {
        try {
            J$.Se(4569281, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/node-abi/index.js');
            function getNextTarget(runtime, targets) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(4565817, arguments.callee, this, arguments);
                            arguments = J$.N(4565825, 'arguments', arguments, true, false, false);
                            runtime = J$.N(4565833, 'runtime', runtime, true, false, false);
                            targets = J$.N(4565841, 'targets', targets, true, false, false);
                            J$.N(4565849, 'latest', latest, false, false, false);
                            J$.N(4565857, 'increment', increment, false, false, false);
                            if (J$.C(252360, J$.B(556274, '==', J$.R(4565577, 'targets', targets, false, false), J$.T(4565585, null, 25, false))))
                                targets = J$.W(4565601, 'targets', J$.R(4565593, 'allTargets', allTargets, false, true), targets, false, false);
                            var latest = J$.W(4565721, 'latest', J$.G(4565713, J$.M(4565697, J$.M(4565681, J$.R(4565609, 'targets', targets, false, false), 'filter', false)(J$.T(4565673, function (t) {
                                jalangiLabel0:
                                    while (true) {
                                        try {
                                            J$.Fe(4565649, arguments.callee, this, arguments);
                                            arguments = J$.N(4565657, 'arguments', arguments, true, false, false);
                                            t = J$.N(4565665, 't', t, true, false, false);
                                            return J$.Rt(4565641, J$.B(556282, '===', J$.G(4565625, J$.R(4565617, 't', t, false, false), 'runtime'), J$.R(4565633, 'runtime', runtime, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(4569385, J$e);
                                        } finally {
                                            if (J$.Fr(4569393))
                                                continue jalangiLabel0;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), 'slice', false)(J$.U(556290, '-', J$.T(4565689, 1, 22, false))), J$.T(4565705, 0, 22, false)), latest, false, false);
                            var increment = J$.W(4565761, 'increment', J$.C(252368, J$.B(556298, '===', J$.R(4565729, 'runtime', runtime, false, false), J$.T(4565737, 'electron', 21, false))) ? J$.T(4565745, 'minor', 21, false) : J$.T(4565753, 'major', 21, false), increment, false, false);
                            return J$.Rt(4565809, J$.M(4565801, J$.R(4565769, 'semver', semver, false, true), 'inc', false)(J$.G(4565785, J$.R(4565777, 'latest', latest, false, false), 'target'), J$.R(4565793, 'increment', increment, false, false)));
                        } catch (J$e) {
                            J$.Ex(4569401, J$e);
                        } finally {
                            if (J$.Fr(4569409))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function getAbi(target, runtime) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(4566433, arguments.callee, this, arguments);
                            arguments = J$.N(4566441, 'arguments', arguments, true, false, false);
                            target = J$.N(4566449, 'target', target, true, false, false);
                            runtime = J$.N(4566457, 'runtime', runtime, true, false, false);
                            J$.N(4566465, 'abi', abi, false, false, false);
                            J$.N(4566473, 'i', i, false, false, false);
                            J$.N(4566481, 't', t, false, false, false);
                            if (J$.C(252376, J$.B(556306, '===', J$.R(4565865, 'target', target, false, false), J$.F(4565905, J$.I(typeof String === 'undefined' ? String = J$.R(4565873, 'String', undefined, true, true) : String = J$.R(4565873, 'String', String, true, true)), false)(J$.F(4565897, J$.I(typeof Number === 'undefined' ? Number = J$.R(4565881, 'Number', undefined, true, true) : Number = J$.R(4565881, 'Number', Number, true, true)), false)(J$.R(4565889, 'target', target, false, false))))))
                                return J$.Rt(4565921, J$.R(4565913, 'target', target, false, false));
                            if (J$.C(252384, J$.R(4565929, 'target', target, false, false)))
                                target = J$.W(4565969, 'target', J$.M(4565961, J$.R(4565937, 'target', target, false, false), 'replace', false)(J$.T(4565945, /^v/, 14, false), J$.T(4565953, '', 21, false)), target, false, false);
                            if (J$.C(252392, J$.U(556314, '!', J$.R(4565977, 'runtime', runtime, false, false))))
                                runtime = J$.W(4565993, 'runtime', J$.T(4565985, 'node', 21, false), runtime, false, false);
                            if (J$.C(252416, J$.B(556322, '===', J$.R(4566001, 'runtime', runtime, false, false), J$.T(4566009, 'node', 21, false)))) {
                                if (J$.C(252400, J$.U(556330, '!', J$.R(4566017, 'target', target, false, false))))
                                    return J$.Rt(4566049, J$.G(4566041, J$.G(4566033, J$.I(typeof process === 'undefined' ? process = J$.R(4566025, 'process', undefined, true, true) : process = J$.R(4566025, 'process', process, true, true)), 'versions'), 'modules'));
                                if (J$.C(252408, J$.B(556338, '===', J$.R(4566057, 'target', target, false, false), J$.G(4566081, J$.G(4566073, J$.I(typeof process === 'undefined' ? process = J$.R(4566065, 'process', undefined, true, true) : process = J$.R(4566065, 'process', process, true, true)), 'versions'), 'node'))))
                                    return J$.Rt(4566113, J$.G(4566105, J$.G(4566097, J$.I(typeof process === 'undefined' ? process = J$.R(4566089, 'process', undefined, true, true) : process = J$.R(4566089, 'process', process, true, true)), 'versions'), 'modules'));
                            }
                            var abi;
                            for (var i = J$.W(4566129, 'i', J$.T(4566121, 0, 22, false), i, false, false); J$.C(252440, J$.B(556346, '<', J$.R(4566137, 'i', i, false, false), J$.G(4566153, J$.R(4566145, 'allTargets', allTargets, false, true), 'length'))); J$.B(556370, '-', i = J$.W(4566169, 'i', J$.B(556362, '+', J$.U(556354, '+', J$.R(4566161, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var t = J$.W(4566201, 't', J$.G(4566193, J$.R(4566177, 'allTargets', allTargets, false, true), J$.R(4566185, 'i', i, false, false)), t, false, false);
                                if (J$.C(252424, J$.B(556378, '!==', J$.G(4566217, J$.R(4566209, 't', t, false, false), 'runtime'), J$.R(4566225, 'runtime', runtime, false, false))))
                                    continue;
                                if (J$.C(252432, J$.M(4566265, J$.R(4566233, 'semver', semver, false, true), 'lte', false)(J$.G(4566249, J$.R(4566241, 't', t, false, false), 'target'), J$.R(4566257, 'target', target, false, false))))
                                    abi = J$.W(4566289, 'abi', J$.G(4566281, J$.R(4566273, 't', t, false, false), 'abi'), abi, false, false);
                                else
                                    break;
                            }
                            if (J$.C(252456, J$.C(252448, J$.R(4566297, 'abi', abi, false, false)) ? J$.M(4566345, J$.R(4566305, 'semver', semver, false, true), 'lt', false)(J$.R(4566313, 'target', target, false, false), J$.F(4566337, J$.R(4566321, 'getNextTarget', getNextTarget, false, true), false)(J$.R(4566329, 'runtime', runtime, false, false))) : J$._()))
                                return J$.Rt(4566361, J$.R(4566353, 'abi', abi, false, false));
                            throw J$.F(4566425, J$.I(typeof Error === 'undefined' ? Error = J$.R(4566369, 'Error', undefined, true, true) : Error = J$.R(4566369, 'Error', Error, true, true)), true)(J$.B(556418, '+', J$.B(556410, '+', J$.B(556402, '+', J$.B(556394, '+', J$.B(556386, '+', J$.T(4566377, 'Could not detect abi for version ', 21, false), J$.R(4566385, 'target', target, false, false)), J$.T(4566393, ' and runtime ', 21, false)), J$.R(4566401, 'runtime', runtime, false, false)), J$.T(4566409, '.  Updating "node-abi" might help solve this issue if it is a new release of ', 21, false)), J$.R(4566417, 'runtime', runtime, false, false)));
                        } catch (J$e) {
                            J$.Ex(4569417, J$e);
                        } finally {
                            if (J$.Fr(4569425))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function getTarget(abi, runtime) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(4566913, arguments.callee, this, arguments);
                            arguments = J$.N(4566921, 'arguments', arguments, true, false, false);
                            abi = J$.N(4566929, 'abi', abi, true, false, false);
                            runtime = J$.N(4566937, 'runtime', runtime, true, false, false);
                            J$.N(4566945, 'match', match, false, false, false);
                            if (J$.C(252472, J$.C(252464, J$.R(4566489, 'abi', abi, false, false)) ? J$.B(556426, '!==', J$.R(4566497, 'abi', abi, false, false), J$.F(4566537, J$.I(typeof String === 'undefined' ? String = J$.R(4566505, 'String', undefined, true, true) : String = J$.R(4566505, 'String', String, true, true)), false)(J$.F(4566529, J$.I(typeof Number === 'undefined' ? Number = J$.R(4566513, 'Number', undefined, true, true) : Number = J$.R(4566513, 'Number', Number, true, true)), false)(J$.R(4566521, 'abi', abi, false, false)))) : J$._()))
                                return J$.Rt(4566553, J$.R(4566545, 'abi', abi, false, false));
                            if (J$.C(252480, J$.U(556434, '!', J$.R(4566561, 'runtime', runtime, false, false))))
                                runtime = J$.W(4566577, 'runtime', J$.T(4566569, 'node', 21, false), runtime, false, false);
                            if (J$.C(252496, J$.C(252488, J$.B(556442, '===', J$.R(4566585, 'runtime', runtime, false, false), J$.T(4566593, 'node', 21, false))) ? J$.U(556450, '!', J$.R(4566601, 'abi', abi, false, false)) : J$._()))
                                return J$.Rt(4566633, J$.G(4566625, J$.G(4566617, J$.I(typeof process === 'undefined' ? process = J$.R(4566609, 'process', undefined, true, true) : process = J$.R(4566609, 'process', process, true, true)), 'versions'), 'node'));
                            var match = J$.W(4566809, 'match', J$.M(4566801, J$.M(4566737, J$.R(4566641, 'allTargets', allTargets, false, true), 'filter', false)(J$.T(4566729, function (t) {
                                jalangiLabel3:
                                    while (true) {
                                        try {
                                            J$.Fe(4566705, arguments.callee, this, arguments);
                                            arguments = J$.N(4566713, 'arguments', arguments, true, false, false);
                                            t = J$.N(4566721, 't', t, true, false, false);
                                            return J$.Rt(4566697, J$.C(252504, J$.B(556458, '===', J$.G(4566657, J$.R(4566649, 't', t, false, false), 'abi'), J$.R(4566665, 'abi', abi, false, false))) ? J$.B(556466, '===', J$.G(4566681, J$.R(4566673, 't', t, false, false), 'runtime'), J$.R(4566689, 'runtime', runtime, false, false)) : J$._());
                                        } catch (J$e) {
                                            J$.Ex(4569433, J$e);
                                        } finally {
                                            if (J$.Fr(4569441))
                                                continue jalangiLabel3;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), 'map', false)(J$.T(4566793, function (t) {
                                jalangiLabel4:
                                    while (true) {
                                        try {
                                            J$.Fe(4566769, arguments.callee, this, arguments);
                                            arguments = J$.N(4566777, 'arguments', arguments, true, false, false);
                                            t = J$.N(4566785, 't', t, true, false, false);
                                            return J$.Rt(4566761, J$.G(4566753, J$.R(4566745, 't', t, false, false), 'target'));
                                        } catch (J$e) {
                                            J$.Ex(4569449, J$e);
                                        } finally {
                                            if (J$.Fr(4569457))
                                                continue jalangiLabel4;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), match, false, false);
                            if (J$.C(252512, J$.G(4566825, J$.R(4566817, 'match', match, false, false), 'length')))
                                return J$.Rt(4566857, J$.G(4566849, J$.R(4566833, 'match', match, false, false), J$.T(4566841, 0, 22, false)));
                            throw J$.F(4566905, J$.I(typeof Error === 'undefined' ? Error = J$.R(4566865, 'Error', undefined, true, true) : Error = J$.R(4566865, 'Error', Error, true, true)), true)(J$.B(556490, '+', J$.B(556482, '+', J$.B(556474, '+', J$.T(4566873, 'Could not detect target for abi ', 21, false), J$.R(4566881, 'abi', abi, false, false)), J$.T(4566889, ' and runtime ', 21, false)), J$.R(4566897, 'runtime', runtime, false, false)));
                        } catch (J$e) {
                            J$.Ex(4569465, J$e);
                        } finally {
                            if (J$.Fr(4569473))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(4569289, 'semver', semver, false, false, false);
            getNextTarget = J$.N(4569305, 'getNextTarget', J$.T(4569297, getNextTarget, 12, false), true, false, false);
            getAbi = J$.N(4569321, 'getAbi', J$.T(4569313, getAbi, 12, false), true, false, false);
            getTarget = J$.N(4569337, 'getTarget', J$.T(4569329, getTarget, 12, false), true, false, false);
            J$.N(4569345, 'supportedTargets', supportedTargets, false, false, false);
            J$.N(4569353, 'additionalTargets', additionalTargets, false, false, false);
            J$.N(4569361, 'deprecatedTargets', deprecatedTargets, false, false, false);
            J$.N(4569369, 'futureTargets', futureTargets, false, false, false);
            J$.N(4569377, 'allTargets', allTargets, false, false, false);
            var semver = J$.W(4565569, 'semver', J$.F(4565561, J$.I(typeof require === 'undefined' ? require = J$.R(4565545, 'require', undefined, true, true) : require = J$.R(4565545, 'require', require, true, true)), false)(J$.T(4565553, 'semver', 21, false)), semver, false, true);
            var supportedTargets = J$.W(4568129, 'supportedTargets', J$.T(4568121, [
                J$.T(4566985, {
                    runtime: J$.T(4566953, 'node', 21, false),
                    target: J$.T(4566961, '5.0.0', 21, false),
                    abi: J$.T(4566969, '47', 21, false),
                    lts: J$.T(4566977, false, 23, false)
                }, 11, false),
                J$.T(4567025, {
                    runtime: J$.T(4566993, 'node', 21, false),
                    target: J$.T(4567001, '6.0.0', 21, false),
                    abi: J$.T(4567009, '48', 21, false),
                    lts: J$.T(4567017, false, 23, false)
                }, 11, false),
                J$.T(4567065, {
                    runtime: J$.T(4567033, 'node', 21, false),
                    target: J$.T(4567041, '7.0.0', 21, false),
                    abi: J$.T(4567049, '51', 21, false),
                    lts: J$.T(4567057, false, 23, false)
                }, 11, false),
                J$.T(4567105, {
                    runtime: J$.T(4567073, 'node', 21, false),
                    target: J$.T(4567081, '8.0.0', 21, false),
                    abi: J$.T(4567089, '57', 21, false),
                    lts: J$.T(4567097, false, 23, false)
                }, 11, false),
                J$.T(4567145, {
                    runtime: J$.T(4567113, 'node', 21, false),
                    target: J$.T(4567121, '9.0.0', 21, false),
                    abi: J$.T(4567129, '59', 21, false),
                    lts: J$.T(4567137, false, 23, false)
                }, 11, false),
                J$.T(4567289, {
                    runtime: J$.T(4567153, 'node', 21, false),
                    target: J$.T(4567161, '10.0.0', 21, false),
                    abi: J$.T(4567169, '64', 21, false),
                    lts: J$.C(252520, J$.B(556498, '<', J$.F(4567209, J$.I(typeof Date === 'undefined' ? Date = J$.R(4567177, 'Date', undefined, true, true) : Date = J$.R(4567177, 'Date', Date, true, true)), true)(J$.T(4567185, 2018, 22, false), J$.T(4567193, 10, 22, false), J$.T(4567201, 1, 22, false)), J$.F(4567225, J$.I(typeof Date === 'undefined' ? Date = J$.R(4567217, 'Date', undefined, true, true) : Date = J$.R(4567217, 'Date', Date, true, true)), true)())) ? J$.B(556506, '<', J$.F(4567241, J$.I(typeof Date === 'undefined' ? Date = J$.R(4567233, 'Date', undefined, true, true) : Date = J$.R(4567233, 'Date', Date, true, true)), true)(), J$.F(4567281, J$.I(typeof Date === 'undefined' ? Date = J$.R(4567249, 'Date', undefined, true, true) : Date = J$.R(4567249, 'Date', Date, true, true)), true)(J$.T(4567257, 2020, 22, false), J$.T(4567265, 4, 22, false), J$.T(4567273, 31, 22, false))) : J$._()
                }, 11, false),
                J$.T(4567329, {
                    runtime: J$.T(4567297, 'node', 21, false),
                    target: J$.T(4567305, '11.0.0', 21, false),
                    abi: J$.T(4567313, '67', 21, false),
                    lts: J$.T(4567321, false, 23, false)
                }, 11, false),
                J$.T(4567473, {
                    runtime: J$.T(4567337, 'node', 21, false),
                    target: J$.T(4567345, '12.0.0', 21, false),
                    abi: J$.T(4567353, '72', 21, false),
                    lts: J$.C(252528, J$.B(556514, '<', J$.F(4567393, J$.I(typeof Date === 'undefined' ? Date = J$.R(4567361, 'Date', undefined, true, true) : Date = J$.R(4567361, 'Date', Date, true, true)), true)(J$.T(4567369, 2019, 22, false), J$.T(4567377, 9, 22, false), J$.T(4567385, 21, 22, false)), J$.F(4567409, J$.I(typeof Date === 'undefined' ? Date = J$.R(4567401, 'Date', undefined, true, true) : Date = J$.R(4567401, 'Date', Date, true, true)), true)())) ? J$.B(556522, '<', J$.F(4567425, J$.I(typeof Date === 'undefined' ? Date = J$.R(4567417, 'Date', undefined, true, true) : Date = J$.R(4567417, 'Date', Date, true, true)), true)(), J$.F(4567465, J$.I(typeof Date === 'undefined' ? Date = J$.R(4567433, 'Date', undefined, true, true) : Date = J$.R(4567433, 'Date', Date, true, true)), true)(J$.T(4567441, 2020, 22, false), J$.T(4567449, 9, 22, false), J$.T(4567457, 31, 22, false))) : J$._()
                }, 11, false),
                J$.T(4567513, {
                    runtime: J$.T(4567481, 'node', 21, false),
                    target: J$.T(4567489, '13.0.0', 21, false),
                    abi: J$.T(4567497, '79', 21, false),
                    lts: J$.T(4567505, false, 23, false)
                }, 11, false),
                J$.T(4567553, {
                    runtime: J$.T(4567521, 'electron', 21, false),
                    target: J$.T(4567529, '0.36.0', 21, false),
                    abi: J$.T(4567537, '47', 21, false),
                    lts: J$.T(4567545, false, 23, false)
                }, 11, false),
                J$.T(4567593, {
                    runtime: J$.T(4567561, 'electron', 21, false),
                    target: J$.T(4567569, '1.1.0', 21, false),
                    abi: J$.T(4567577, '48', 21, false),
                    lts: J$.T(4567585, false, 23, false)
                }, 11, false),
                J$.T(4567633, {
                    runtime: J$.T(4567601, 'electron', 21, false),
                    target: J$.T(4567609, '1.3.0', 21, false),
                    abi: J$.T(4567617, '49', 21, false),
                    lts: J$.T(4567625, false, 23, false)
                }, 11, false),
                J$.T(4567673, {
                    runtime: J$.T(4567641, 'electron', 21, false),
                    target: J$.T(4567649, '1.4.0', 21, false),
                    abi: J$.T(4567657, '50', 21, false),
                    lts: J$.T(4567665, false, 23, false)
                }, 11, false),
                J$.T(4567713, {
                    runtime: J$.T(4567681, 'electron', 21, false),
                    target: J$.T(4567689, '1.5.0', 21, false),
                    abi: J$.T(4567697, '51', 21, false),
                    lts: J$.T(4567705, false, 23, false)
                }, 11, false),
                J$.T(4567753, {
                    runtime: J$.T(4567721, 'electron', 21, false),
                    target: J$.T(4567729, '1.6.0', 21, false),
                    abi: J$.T(4567737, '53', 21, false),
                    lts: J$.T(4567745, false, 23, false)
                }, 11, false),
                J$.T(4567793, {
                    runtime: J$.T(4567761, 'electron', 21, false),
                    target: J$.T(4567769, '1.7.0', 21, false),
                    abi: J$.T(4567777, '54', 21, false),
                    lts: J$.T(4567785, false, 23, false)
                }, 11, false),
                J$.T(4567833, {
                    runtime: J$.T(4567801, 'electron', 21, false),
                    target: J$.T(4567809, '1.8.0', 21, false),
                    abi: J$.T(4567817, '57', 21, false),
                    lts: J$.T(4567825, false, 23, false)
                }, 11, false),
                J$.T(4567873, {
                    runtime: J$.T(4567841, 'electron', 21, false),
                    target: J$.T(4567849, '2.0.0', 21, false),
                    abi: J$.T(4567857, '57', 21, false),
                    lts: J$.T(4567865, false, 23, false)
                }, 11, false),
                J$.T(4567913, {
                    runtime: J$.T(4567881, 'electron', 21, false),
                    target: J$.T(4567889, '3.0.0', 21, false),
                    abi: J$.T(4567897, '64', 21, false),
                    lts: J$.T(4567905, false, 23, false)
                }, 11, false),
                J$.T(4567953, {
                    runtime: J$.T(4567921, 'electron', 21, false),
                    target: J$.T(4567929, '4.0.0', 21, false),
                    abi: J$.T(4567937, '64', 21, false),
                    lts: J$.T(4567945, false, 23, false)
                }, 11, false),
                J$.T(4567993, {
                    runtime: J$.T(4567961, 'electron', 21, false),
                    target: J$.T(4567969, '4.0.4', 21, false),
                    abi: J$.T(4567977, '69', 21, false),
                    lts: J$.T(4567985, false, 23, false)
                }, 11, false),
                J$.T(4568033, {
                    runtime: J$.T(4568001, 'electron', 21, false),
                    target: J$.T(4568009, '5.0.0', 21, false),
                    abi: J$.T(4568017, '70', 21, false),
                    lts: J$.T(4568025, false, 23, false)
                }, 11, false),
                J$.T(4568073, {
                    runtime: J$.T(4568041, 'electron', 21, false),
                    target: J$.T(4568049, '6.0.0', 21, false),
                    abi: J$.T(4568057, '73', 21, false),
                    lts: J$.T(4568065, false, 23, false)
                }, 11, false),
                J$.T(4568113, {
                    runtime: J$.T(4568081, 'electron', 21, false),
                    target: J$.T(4568089, '7.0.0', 21, false),
                    abi: J$.T(4568097, '75', 21, false),
                    lts: J$.T(4568105, false, 23, false)
                }, 11, false)
            ], 10, false), supportedTargets, false, true);
            var additionalTargets = J$.W(4568345, 'additionalTargets', J$.T(4568337, [
                J$.T(4568169, {
                    runtime: J$.T(4568137, 'node-webkit', 21, false),
                    target: J$.T(4568145, '0.13.0', 21, false),
                    abi: J$.T(4568153, '47', 21, false),
                    lts: J$.T(4568161, false, 23, false)
                }, 11, false),
                J$.T(4568209, {
                    runtime: J$.T(4568177, 'node-webkit', 21, false),
                    target: J$.T(4568185, '0.15.0', 21, false),
                    abi: J$.T(4568193, '48', 21, false),
                    lts: J$.T(4568201, false, 23, false)
                }, 11, false),
                J$.T(4568249, {
                    runtime: J$.T(4568217, 'node-webkit', 21, false),
                    target: J$.T(4568225, '0.18.3', 21, false),
                    abi: J$.T(4568233, '51', 21, false),
                    lts: J$.T(4568241, false, 23, false)
                }, 11, false),
                J$.T(4568289, {
                    runtime: J$.T(4568257, 'node-webkit', 21, false),
                    target: J$.T(4568265, '0.23.0', 21, false),
                    abi: J$.T(4568273, '57', 21, false),
                    lts: J$.T(4568281, false, 23, false)
                }, 11, false),
                J$.T(4568329, {
                    runtime: J$.T(4568297, 'node-webkit', 21, false),
                    target: J$.T(4568305, '0.26.5', 21, false),
                    abi: J$.T(4568313, '59', 21, false),
                    lts: J$.T(4568321, false, 23, false)
                }, 11, false)
            ], 10, false), additionalTargets, false, true);
            var deprecatedTargets = J$.W(4568961, 'deprecatedTargets', J$.T(4568953, [
                J$.T(4568385, {
                    runtime: J$.T(4568353, 'node', 21, false),
                    target: J$.T(4568361, '0.2.0', 21, false),
                    abi: J$.T(4568369, '1', 21, false),
                    lts: J$.T(4568377, false, 23, false)
                }, 11, false),
                J$.T(4568425, {
                    runtime: J$.T(4568393, 'node', 21, false),
                    target: J$.T(4568401, '0.9.1', 21, false),
                    abi: J$.T(4568409, '0x000A', 21, false),
                    lts: J$.T(4568417, false, 23, false)
                }, 11, false),
                J$.T(4568465, {
                    runtime: J$.T(4568433, 'node', 21, false),
                    target: J$.T(4568441, '0.9.9', 21, false),
                    abi: J$.T(4568449, '0x000B', 21, false),
                    lts: J$.T(4568457, false, 23, false)
                }, 11, false),
                J$.T(4568505, {
                    runtime: J$.T(4568473, 'node', 21, false),
                    target: J$.T(4568481, '0.10.4', 21, false),
                    abi: J$.T(4568489, '11', 21, false),
                    lts: J$.T(4568497, false, 23, false)
                }, 11, false),
                J$.T(4568545, {
                    runtime: J$.T(4568513, 'node', 21, false),
                    target: J$.T(4568521, '0.11.0', 21, false),
                    abi: J$.T(4568529, '0x000C', 21, false),
                    lts: J$.T(4568537, false, 23, false)
                }, 11, false),
                J$.T(4568585, {
                    runtime: J$.T(4568553, 'node', 21, false),
                    target: J$.T(4568561, '0.11.8', 21, false),
                    abi: J$.T(4568569, '13', 21, false),
                    lts: J$.T(4568577, false, 23, false)
                }, 11, false),
                J$.T(4568625, {
                    runtime: J$.T(4568593, 'node', 21, false),
                    target: J$.T(4568601, '0.11.11', 21, false),
                    abi: J$.T(4568609, '14', 21, false),
                    lts: J$.T(4568617, false, 23, false)
                }, 11, false),
                J$.T(4568665, {
                    runtime: J$.T(4568633, 'node', 21, false),
                    target: J$.T(4568641, '1.0.0', 21, false),
                    abi: J$.T(4568649, '42', 21, false),
                    lts: J$.T(4568657, false, 23, false)
                }, 11, false),
                J$.T(4568705, {
                    runtime: J$.T(4568673, 'node', 21, false),
                    target: J$.T(4568681, '1.1.0', 21, false),
                    abi: J$.T(4568689, '43', 21, false),
                    lts: J$.T(4568697, false, 23, false)
                }, 11, false),
                J$.T(4568745, {
                    runtime: J$.T(4568713, 'node', 21, false),
                    target: J$.T(4568721, '2.0.0', 21, false),
                    abi: J$.T(4568729, '44', 21, false),
                    lts: J$.T(4568737, false, 23, false)
                }, 11, false),
                J$.T(4568785, {
                    runtime: J$.T(4568753, 'node', 21, false),
                    target: J$.T(4568761, '3.0.0', 21, false),
                    abi: J$.T(4568769, '45', 21, false),
                    lts: J$.T(4568777, false, 23, false)
                }, 11, false),
                J$.T(4568825, {
                    runtime: J$.T(4568793, 'node', 21, false),
                    target: J$.T(4568801, '4.0.0', 21, false),
                    abi: J$.T(4568809, '46', 21, false),
                    lts: J$.T(4568817, false, 23, false)
                }, 11, false),
                J$.T(4568865, {
                    runtime: J$.T(4568833, 'electron', 21, false),
                    target: J$.T(4568841, '0.30.0', 21, false),
                    abi: J$.T(4568849, '44', 21, false),
                    lts: J$.T(4568857, false, 23, false)
                }, 11, false),
                J$.T(4568905, {
                    runtime: J$.T(4568873, 'electron', 21, false),
                    target: J$.T(4568881, '0.31.0', 21, false),
                    abi: J$.T(4568889, '45', 21, false),
                    lts: J$.T(4568897, false, 23, false)
                }, 11, false),
                J$.T(4568945, {
                    runtime: J$.T(4568913, 'electron', 21, false),
                    target: J$.T(4568921, '0.33.0', 21, false),
                    abi: J$.T(4568929, '46', 21, false),
                    lts: J$.T(4568937, false, 23, false)
                }, 11, false)
            ], 10, false), deprecatedTargets, false, true);
            var futureTargets = J$.W(4569017, 'futureTargets', J$.T(4569009, [J$.T(4569001, {
                    runtime: J$.T(4568969, 'electron', 21, false),
                    target: J$.T(4568977, '8.0.0-beta.0', 21, false),
                    abi: J$.T(4568985, '76', 21, false),
                    lts: J$.T(4568993, false, 23, false)
                }, 11, false)], 10, false), futureTargets, false, true);
            var allTargets = J$.W(4569081, 'allTargets', J$.M(4569073, J$.M(4569057, J$.M(4569041, J$.R(4569025, 'deprecatedTargets', deprecatedTargets, false, true), 'concat', false)(J$.R(4569033, 'supportedTargets', supportedTargets, false, true)), 'concat', false)(J$.R(4569049, 'additionalTargets', additionalTargets, false, true)), 'concat', false)(J$.R(4569065, 'futureTargets', futureTargets, false, true)), allTargets, false, true);
            J$.P(4569105, J$.I(typeof exports === 'undefined' ? exports = J$.R(4569089, 'exports', undefined, true, true) : exports = J$.R(4569089, 'exports', exports, true, true)), 'getAbi', J$.R(4569097, 'getAbi', getAbi, false, true));
            J$.P(4569129, J$.I(typeof exports === 'undefined' ? exports = J$.R(4569113, 'exports', undefined, true, true) : exports = J$.R(4569113, 'exports', exports, true, true)), 'getTarget', J$.R(4569121, 'getTarget', getTarget, false, true));
            J$.P(4569153, J$.I(typeof exports === 'undefined' ? exports = J$.R(4569137, 'exports', undefined, true, true) : exports = J$.R(4569137, 'exports', exports, true, true)), 'deprecatedTargets', J$.R(4569145, 'deprecatedTargets', deprecatedTargets, false, true));
            J$.P(4569177, J$.I(typeof exports === 'undefined' ? exports = J$.R(4569161, 'exports', undefined, true, true) : exports = J$.R(4569161, 'exports', exports, true, true)), 'supportedTargets', J$.R(4569169, 'supportedTargets', supportedTargets, false, true));
            J$.P(4569201, J$.I(typeof exports === 'undefined' ? exports = J$.R(4569185, 'exports', undefined, true, true) : exports = J$.R(4569185, 'exports', exports, true, true)), 'additionalTargets', J$.R(4569193, 'additionalTargets', additionalTargets, false, true));
            J$.P(4569225, J$.I(typeof exports === 'undefined' ? exports = J$.R(4569209, 'exports', undefined, true, true) : exports = J$.R(4569209, 'exports', exports, true, true)), 'futureTargets', J$.R(4569217, 'futureTargets', futureTargets, false, true));
            J$.P(4569249, J$.I(typeof exports === 'undefined' ? exports = J$.R(4569233, 'exports', undefined, true, true) : exports = J$.R(4569233, 'exports', exports, true, true)), 'allTargets', J$.R(4569241, 'allTargets', allTargets, false, true));
            J$.P(4569273, J$.I(typeof exports === 'undefined' ? exports = J$.R(4569257, 'exports', undefined, true, true) : exports = J$.R(4569257, 'exports', exports, true, true)), '_getNextTarget', J$.R(4569265, 'getNextTarget', getNextTarget, false, true));
        } catch (J$e) {
            J$.Ex(4569481, J$e);
        } finally {
            if (J$.Sr(4569489))
                continue jalangiLabel6;
            else
                break jalangiLabel6;
        }
    }
// JALANGI DO NOT INSTRUMENT
