// prebuild-install/asset.js
J$.noInstrEval = false;
jalangiLabel5:
    while (true) {
        try {
            J$.Se(4596281, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/prebuild-install/asset.js');
            function findAssetId(opts, cb) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(4596185, arguments.callee, this, arguments);
                            arguments = J$.N(4596193, 'arguments', arguments, true, false, false);
                            opts = J$.N(4596201, 'opts', opts, true, false, false);
                            cb = J$.N(4596209, 'cb', cb, true, false, false);
                            J$.N(4596217, 'downloadUrl', downloadUrl, false, false, false);
                            J$.N(4596225, 'apiUrl', apiUrl, false, false, false);
                            J$.N(4596233, 'log', log, false, false, false);
                            J$.N(4596241, 'reqOpts', reqOpts, false, false, false);
                            J$.N(4596249, 'req', req, false, false, false);
                            var downloadUrl = J$.W(4595473, 'downloadUrl', J$.M(4595465, J$.R(4595449, 'util', util, false, true), 'getDownloadUrl', false)(J$.R(4595457, 'opts', opts, false, false)), downloadUrl, false, false);
                            var apiUrl = J$.W(4595505, 'apiUrl', J$.M(4595497, J$.R(4595481, 'util', util, false, true), 'getApiUrl', false)(J$.R(4595489, 'opts', opts, false, false)), apiUrl, false, false);
                            var log = J$.W(4595537, 'log', J$.C(253368, J$.G(4595521, J$.R(4595513, 'opts', opts, false, false), 'log')) ? J$._() : J$.R(4595529, 'noop', noop, false, true), log, false, false);
                            J$.M(4595577, J$.R(4595545, 'log', log, false, false), 'http', false)(J$.T(4595553, 'request', 21, false), J$.B(557506, '+', J$.T(4595561, 'GET ', 21, false), J$.R(4595569, 'apiUrl', apiUrl, false, false)));
                            var reqOpts = J$.W(4595673, 'reqOpts', J$.F(4595665, J$.R(4595585, 'proxy', proxy, false, true), false)(J$.T(4595649, {
                                url: J$.R(4595593, 'apiUrl', apiUrl, false, false),
                                json: J$.T(4595601, true, 23, false),
                                headers: J$.T(4595641, {
                                    'User-Agent': J$.T(4595609, 'simple-get', 21, false),
                                    Authorization: J$.B(557514, '+', J$.T(4595617, 'token ', 21, false), J$.G(4595633, J$.R(4595625, 'opts', opts, false, false), 'token'))
                                }, 11, false)
                            }, 11, false), J$.R(4595657, 'opts', opts, false, false)), reqOpts, false, false);
                            var req = J$.W(4596105, 'req', J$.M(4596097, J$.R(4595681, 'get', get, false, true), 'concat', false)(J$.R(4595689, 'reqOpts', reqOpts, false, false), J$.T(4596089, function (err, res, data) {
                                jalangiLabel2:
                                    while (true) {
                                        try {
                                            J$.Fe(4596033, arguments.callee, this, arguments);
                                            arguments = J$.N(4596041, 'arguments', arguments, true, false, false);
                                            err = J$.N(4596049, 'err', err, true, false, false);
                                            res = J$.N(4596057, 'res', res, true, false, false);
                                            data = J$.N(4596065, 'data', data, true, false, false);
                                            J$.N(4596073, 'release', release, false, false, false);
                                            J$.N(4596081, 'asset', asset, false, false, false);
                                            if (J$.C(253376, J$.R(4595697, 'err', err, false, false)))
                                                return J$.Rt(4595729, J$.F(4595721, J$.R(4595705, 'cb', cb, false, false), false)(J$.R(4595713, 'err', err, false, false)));
                                            J$.M(4595769, J$.R(4595737, 'log', log, false, false), 'http', false)(J$.G(4595753, J$.R(4595745, 'res', res, false, false), 'statusCode'), J$.R(4595761, 'apiUrl', apiUrl, false, false));
                                            if (J$.C(253384, J$.B(557522, '!==', J$.G(4595785, J$.R(4595777, 'res', res, false, false), 'statusCode'), J$.T(4595793, 200, 22, false))))
                                                return J$.Rt(4595825, J$.F(4595817, J$.R(4595801, 'cb', cb, false, false), false)(J$.R(4595809, 'err', err, false, false)));
                                            for (var release of J$.R(4595833, 'data', data, false, false)) {
                                                if (J$.C(253400, J$.B(557538, '===', J$.G(4595849, J$.R(4595841, 'release', release, false, false), 'tag_name'), J$.B(557530, '+', J$.G(4595873, J$.R(4595857, 'opts', opts, false, false), J$.T(4595865, 'tag-prefix', 21, false)), J$.G(4595897, J$.G(4595889, J$.R(4595881, 'opts', opts, false, false), 'pkg'), 'version'))))) {
                                                    for (var asset of J$.G(4595913, J$.R(4595905, 'release', release, false, false), 'assets')) {
                                                        if (J$.C(253392, J$.B(557546, '===', J$.G(4595929, J$.R(4595921, 'asset', asset, false, false), 'browser_download_url'), J$.R(4595937, 'downloadUrl', downloadUrl, false, false)))) {
                                                            return J$.Rt(4595985, J$.F(4595977, J$.R(4595945, 'cb', cb, false, false), false)(J$.T(4595953, null, 25, false), J$.G(4595969, J$.R(4595961, 'asset', asset, false, false), 'id')));
                                                        }
                                                    }
                                                }
                                            }
                                            J$.F(4596025, J$.R(4595993, 'cb', cb, false, false), false)(J$.F(4596017, J$.I(typeof Error === 'undefined' ? Error = J$.R(4596001, 'Error', undefined, true, true) : Error = J$.R(4596001, 'Error', Error, true, true)), true)(J$.T(4596009, 'Could not find GitHub release for version', 21, false)));
                                        } catch (J$e) {
                                            J$.Ex(4596369, J$e);
                                        } finally {
                                            if (J$.Fr(4596377))
                                                continue jalangiLabel2;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), req, false, false);
                            J$.M(4596177, J$.R(4596113, 'req', req, false, false), 'setTimeout', false)(J$.B(557554, '*', J$.T(4596121, 30, 22, false), J$.T(4596129, 1000, 22, false)), J$.T(4596169, function () {
                                jalangiLabel3:
                                    while (true) {
                                        try {
                                            J$.Fe(4596153, arguments.callee, this, arguments);
                                            arguments = J$.N(4596161, 'arguments', arguments, true, false, false);
                                            J$.M(4596145, J$.R(4596137, 'req', req, false, false), 'abort', false)();
                                        } catch (J$e) {
                                            J$.Ex(4596385, J$e);
                                        } finally {
                                            if (J$.Fr(4596393))
                                                continue jalangiLabel3;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(4596401, J$e);
                        } finally {
                            if (J$.Fr(4596409))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(4596289, 'get', get, false, false, false);
            J$.N(4596297, 'util', util, false, false, false);
            J$.N(4596305, 'proxy', proxy, false, false, false);
            J$.N(4596313, 'noop', noop, false, false, false);
            findAssetId = J$.N(4596329, 'findAssetId', J$.T(4596321, findAssetId, 12, false), true, false, false);
            var get = J$.W(4595273, 'get', J$.F(4595265, J$.I(typeof require === 'undefined' ? require = J$.R(4595249, 'require', undefined, true, true) : require = J$.R(4595249, 'require', require, true, true)), false)(J$.T(4595257, 'simple-get', 21, false)), get, false, true);
            var util = J$.W(4595305, 'util', J$.F(4595297, J$.I(typeof require === 'undefined' ? require = J$.R(4595281, 'require', undefined, true, true) : require = J$.R(4595281, 'require', require, true, true)), false)(J$.T(4595289, './util', 21, false)), util, false, true);
            var proxy = J$.W(4595337, 'proxy', J$.F(4595329, J$.I(typeof require === 'undefined' ? require = J$.R(4595313, 'require', undefined, true, true) : require = J$.R(4595313, 'require', require, true, true)), false)(J$.T(4595321, './proxy', 21, false)), proxy, false, true);
            var noop = J$.W(4595441, 'noop', J$.M(4595433, J$.I(typeof Object === 'undefined' ? Object = J$.R(4595345, 'Object', undefined, true, true) : Object = J$.R(4595345, 'Object', Object, true, true)), 'assign', false)(J$.T(4595401, {
                http: J$.T(4595369, function () {
                    jalangiLabel0:
                        while (true) {
                            try {
                                J$.Fe(4595353, arguments.callee, this, arguments);
                                arguments = J$.N(4595361, 'arguments', arguments, true, false, false);
                            } catch (J$e) {
                                J$.Ex(4596337, J$e);
                            } finally {
                                if (J$.Fr(4596345))
                                    continue jalangiLabel0;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                silly: J$.T(4595393, function () {
                    jalangiLabel1:
                        while (true) {
                            try {
                                J$.Fe(4595377, arguments.callee, this, arguments);
                                arguments = J$.N(4595385, 'arguments', arguments, true, false, false);
                            } catch (J$e) {
                                J$.Ex(4596353, J$e);
                            } finally {
                                if (J$.Fr(4596361))
                                    continue jalangiLabel1;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false), J$.F(4595425, J$.I(typeof require === 'undefined' ? require = J$.R(4595409, 'require', undefined, true, true) : require = J$.R(4595409, 'require', require, true, true)), false)(J$.T(4595417, 'noop-logger', 21, false))), noop, false, true);
            J$.P(4596273, J$.I(typeof module === 'undefined' ? module = J$.R(4596257, 'module', undefined, true, true) : module = J$.R(4596257, 'module', module, true, true)), 'exports', J$.R(4596265, 'findAssetId', findAssetId, false, true));
        } catch (J$e) {
            J$.Ex(4596417, J$e);
        } finally {
            if (J$.Sr(4596425))
                continue jalangiLabel5;
            else
                break jalangiLabel5;
        }
    }
// JALANGI DO NOT INSTRUMENT
