// prebuild-install/util.js
J$.noInstrEval = false;
jalangiLabel11:
    while (true) {
        try {
            J$.Se(4608361, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/prebuild-install/util.js');
            function getDownloadUrl(opts) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(4606601, arguments.callee, this, arguments);
                            arguments = J$.N(4606609, 'arguments', arguments, true, false, false);
                            opts = J$.N(4606617, 'opts', opts, true, false, false);
                            J$.N(4606625, 'pkgName', pkgName, false, false, false);
                            var pkgName = J$.W(4605929, 'pkgName', J$.M(4605921, J$.G(4605897, J$.G(4605889, J$.R(4605881, 'opts', opts, false, false), 'pkg'), 'name'), 'replace', false)(J$.T(4605905, /^@\w+\//, 14, false), J$.T(4605913, '', 21, false)), pkgName, false, false);
                            return J$.Rt(4606593, J$.F(4606585, J$.R(4605937, 'expandTemplate', expandTemplate, false, true), false)(J$.F(4605961, J$.R(4605945, 'urlTemplate', urlTemplate, false, true), false)(J$.R(4605953, 'opts', opts, false, false)), J$.T(4606577, {
                                name: J$.R(4605969, 'pkgName', pkgName, false, false),
                                package_name: J$.R(4605977, 'pkgName', pkgName, false, false),
                                version: J$.G(4606001, J$.G(4605993, J$.R(4605985, 'opts', opts, false, false), 'pkg'), 'version'),
                                major: J$.G(4606057, J$.M(4606041, J$.G(4606025, J$.G(4606017, J$.R(4606009, 'opts', opts, false, false), 'pkg'), 'version'), 'split', false)(J$.T(4606033, '.', 21, false)), J$.T(4606049, 0, 22, false)),
                                minor: J$.G(4606113, J$.M(4606097, J$.G(4606081, J$.G(4606073, J$.R(4606065, 'opts', opts, false, false), 'pkg'), 'version'), 'split', false)(J$.T(4606089, '.', 21, false)), J$.T(4606105, 1, 22, false)),
                                patch: J$.G(4606169, J$.M(4606153, J$.G(4606137, J$.G(4606129, J$.R(4606121, 'opts', opts, false, false), 'pkg'), 'version'), 'split', false)(J$.T(4606145, '.', 21, false)), J$.T(4606161, 2, 22, false)),
                                prerelease: J$.G(4606225, J$.M(4606209, J$.G(4606193, J$.G(4606185, J$.R(4606177, 'opts', opts, false, false), 'pkg'), 'version'), 'split', false)(J$.T(4606201, '-', 21, false)), J$.T(4606217, 1, 22, false)),
                                build: J$.G(4606281, J$.M(4606265, J$.G(4606249, J$.G(4606241, J$.R(4606233, 'opts', opts, false, false), 'pkg'), 'version'), 'split', false)(J$.T(4606257, '+', 21, false)), J$.T(4606273, 1, 22, false)),
                                abi: J$.C(253992, J$.G(4606297, J$.R(4606289, 'opts', opts, false, false), 'abi')) ? J$._() : J$.G(4606321, J$.G(4606313, J$.I(typeof process === 'undefined' ? process = J$.R(4606305, 'process', undefined, true, true) : process = J$.R(4606305, 'process', process, true, true)), 'versions'), 'modules'),
                                node_abi: J$.G(4606345, J$.G(4606337, J$.I(typeof process === 'undefined' ? process = J$.R(4606329, 'process', undefined, true, true) : process = J$.R(4606329, 'process', process, true, true)), 'versions'), 'modules'),
                                runtime: J$.C(254000, J$.G(4606361, J$.R(4606353, 'opts', opts, false, false), 'runtime')) ? J$._() : J$.T(4606369, 'node', 21, false),
                                platform: J$.G(4606385, J$.R(4606377, 'opts', opts, false, false), 'platform'),
                                arch: J$.G(4606401, J$.R(4606393, 'opts', opts, false, false), 'arch'),
                                libc: J$.C(254016, J$.C(254008, J$.G(4606417, J$.R(4606409, 'opts', opts, false, false), 'libc')) ? J$._() : J$.G(4606441, J$.G(4606433, J$.I(typeof process === 'undefined' ? process = J$.R(4606425, 'process', undefined, true, true) : process = J$.R(4606425, 'process', process, true, true)), 'env'), 'LIBC')) ? J$._() : J$.T(4606449, '', 21, false),
                                configuration: J$.C(254024, J$.G(4606465, J$.R(4606457, 'opts', opts, false, false), 'debug')) ? J$.T(4606473, 'Debug', 21, false) : J$.T(4606481, 'Release', 21, false),
                                module_name: J$.C(254032, J$.G(4606505, J$.G(4606497, J$.R(4606489, 'opts', opts, false, false), 'pkg'), 'binary')) ? J$.G(4606537, J$.G(4606529, J$.G(4606521, J$.R(4606513, 'opts', opts, false, false), 'pkg'), 'binary'), 'module_name') : J$._(),
                                tag_prefix: J$.C(254040, J$.G(4606561, J$.R(4606545, 'opts', opts, false, false), J$.T(4606553, 'tag-prefix', 21, false))) ? J$._() : J$.T(4606569, 'v', 21, false)
                            }, 11, false)));
                        } catch (J$e) {
                            J$.Ex(4608569, J$e);
                        } finally {
                            if (J$.Fr(4608577))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function getApiUrl(opts) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(4606705, arguments.callee, this, arguments);
                            arguments = J$.N(4606713, 'arguments', arguments, true, false, false);
                            opts = J$.N(4606721, 'opts', opts, true, false, false);
                            return J$.Rt(4606697, J$.B(558098, '+', J$.M(4606681, J$.F(4606657, J$.R(4606633, 'github', github, false, true), false)(J$.G(4606649, J$.R(4606641, 'opts', opts, false, false), 'pkg')), 'replace', false)(J$.T(4606665, 'github.com', 21, false), J$.T(4606673, 'api.github.com/repos', 21, false)), J$.T(4606689, '/releases', 21, false)));
                        } catch (J$e) {
                            J$.Ex(4608585, J$e);
                        } finally {
                            if (J$.Fr(4608593))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function getAssetUrl(opts, assetId) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(4606777, arguments.callee, this, arguments);
                            arguments = J$.N(4606785, 'arguments', arguments, true, false, false);
                            opts = J$.N(4606793, 'opts', opts, true, false, false);
                            assetId = J$.N(4606801, 'assetId', assetId, true, false, false);
                            return J$.Rt(4606769, J$.B(558114, '+', J$.B(558106, '+', J$.F(4606745, J$.R(4606729, 'getApiUrl', getApiUrl, false, true), false)(J$.R(4606737, 'opts', opts, false, false)), J$.T(4606753, '/assets/', 21, false)), J$.R(4606761, 'assetId', assetId, false, false)));
                        } catch (J$e) {
                            J$.Ex(4608601, J$e);
                        } finally {
                            if (J$.Fr(4608609))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function urlTemplate(opts) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(4607281, arguments.callee, this, arguments);
                            arguments = J$.N(4607289, 'arguments', arguments, true, false, false);
                            opts = J$.N(4607297, 'opts', opts, true, false, false);
                            J$.N(4607305, 'packageName', packageName, false, false, false);
                            J$.N(4607313, 'hostMirrorUrl', hostMirrorUrl, false, false, false);
                            if (J$.C(254048, J$.B(558130, '===', J$.U(558122, 'typeof', J$.G(4606817, J$.R(4606809, 'opts', opts, false, false), 'download')), J$.T(4606825, 'string', 21, false)))) {
                                return J$.Rt(4606849, J$.G(4606841, J$.R(4606833, 'opts', opts, false, false), 'download'));
                            }
                            var packageName = J$.W(4606865, 'packageName', J$.T(4606857, '{name}-v{version}-{runtime}-v{abi}-{platform}{libc}-{arch}.tar.gz', 21, false), packageName, false, false);
                            var hostMirrorUrl = J$.W(4606897, 'hostMirrorUrl', J$.F(4606889, J$.R(4606873, 'getHostMirrorUrl', getHostMirrorUrl, false, true), false)(J$.R(4606881, 'opts', opts, false, false)), hostMirrorUrl, false, false);
                            if (J$.C(254056, J$.R(4606905, 'hostMirrorUrl', hostMirrorUrl, false, false))) {
                                return J$.Rt(4606937, J$.B(558146, '+', J$.B(558138, '+', J$.R(4606913, 'hostMirrorUrl', hostMirrorUrl, false, false), J$.T(4606921, '/{tag_prefix}{version}/', 21, false)), J$.R(4606929, 'packageName', packageName, false, false)));
                            }
                            if (J$.C(254080, J$.C(254064, J$.G(4606961, J$.G(4606953, J$.R(4606945, 'opts', opts, false, false), 'pkg'), 'binary')) ? J$.G(4606993, J$.G(4606985, J$.G(4606977, J$.R(4606969, 'opts', opts, false, false), 'pkg'), 'binary'), 'host') : J$._())) {
                                return J$.Rt(4607217, J$.M(4607209, J$.M(4607193, J$.M(4607177, J$.T(4607105, [
                                    J$.G(4607025, J$.G(4607017, J$.G(4607009, J$.R(4607001, 'opts', opts, false, false), 'pkg'), 'binary'), 'host'),
                                    J$.G(4607057, J$.G(4607049, J$.G(4607041, J$.R(4607033, 'opts', opts, false, false), 'pkg'), 'binary'), 'remote_path'),
                                    J$.C(254072, J$.G(4607089, J$.G(4607081, J$.G(4607073, J$.R(4607065, 'opts', opts, false, false), 'pkg'), 'binary'), 'package_name')) ? J$._() : J$.R(4607097, 'packageName', packageName, false, false)
                                ], 10, false), 'map', false)(J$.T(4607169, function (path) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(4607145, arguments.callee, this, arguments);
                                                arguments = J$.N(4607153, 'arguments', arguments, true, false, false);
                                                path = J$.N(4607161, 'path', path, true, false, false);
                                                return J$.Rt(4607137, J$.F(4607129, J$.R(4607113, 'trimSlashes', trimSlashes, false, true), false)(J$.R(4607121, 'path', path, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(4608617, J$e);
                                            } finally {
                                                if (J$.Fr(4608625))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)), 'filter', false)(J$.I(typeof Boolean === 'undefined' ? Boolean = J$.R(4607185, 'Boolean', undefined, true, true) : Boolean = J$.R(4607185, 'Boolean', Boolean, true, true))), 'join', false)(J$.T(4607201, '/', 21, false)));
                            }
                            return J$.Rt(4607273, J$.B(558162, '+', J$.B(558154, '+', J$.F(4607249, J$.R(4607225, 'github', github, false, true), false)(J$.G(4607241, J$.R(4607233, 'opts', opts, false, false), 'pkg')), J$.T(4607257, '/releases/download/{tag_prefix}{version}/', 21, false)), J$.R(4607265, 'packageName', packageName, false, false)));
                        } catch (J$e) {
                            J$.Ex(4608633, J$e);
                        } finally {
                            if (J$.Fr(4608641))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function getHostMirrorUrl(opts) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(4607481, arguments.callee, this, arguments);
                            arguments = J$.N(4607489, 'arguments', arguments, true, false, false);
                            opts = J$.N(4607497, 'opts', opts, true, false, false);
                            J$.N(4607505, 'propName', propName, false, false, false);
                            var propName = J$.W(4607393, 'propName', J$.B(558178, '+', J$.B(558170, '+', J$.T(4607321, 'npm_config_', 21, false), J$.M(4607377, J$.C(254088, J$.G(4607345, J$.G(4607337, J$.R(4607329, 'opts', opts, false, false), 'pkg'), 'name')) ? J$._() : J$.T(4607353, '', 21, false), 'replace', false)(J$.T(4607361, /[^a-zA-Z0-9]/g, 14, false), J$.T(4607369, '_', 21, false))), J$.T(4607385, '_binary_host', 21, false)), propName, false, false);
                            return J$.Rt(4607473, J$.C(254096, J$.G(4607425, J$.G(4607409, J$.I(typeof process === 'undefined' ? process = J$.R(4607401, 'process', undefined, true, true) : process = J$.R(4607401, 'process', process, true, true)), 'env'), J$.R(4607417, 'propName', propName, false, false))) ? J$._() : J$.G(4607465, J$.G(4607441, J$.I(typeof process === 'undefined' ? process = J$.R(4607433, 'process', undefined, true, true) : process = J$.R(4607433, 'process', process, true, true)), 'env'), J$.B(558186, '+', J$.R(4607449, 'propName', propName, false, false), J$.T(4607457, '_mirror', 21, false))));
                        } catch (J$e) {
                            J$.Ex(4608649, J$e);
                        } finally {
                            if (J$.Fr(4608657))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function trimSlashes(str) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(4607561, arguments.callee, this, arguments);
                            arguments = J$.N(4607569, 'arguments', arguments, true, false, false);
                            str = J$.N(4607577, 'str', str, true, false, false);
                            if (J$.C(254104, J$.R(4607513, 'str', str, false, false)))
                                return J$.Rt(4607553, J$.M(4607545, J$.R(4607521, 'str', str, false, false), 'replace', false)(J$.T(4607529, /^\.\/|^\/|\/$/g, 14, false), J$.T(4607537, '', 21, false)));
                        } catch (J$e) {
                            J$.Ex(4608665, J$e);
                        } finally {
                            if (J$.Fr(4608673))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function cachedPrebuild(url) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(4607777, arguments.callee, this, arguments);
                            arguments = J$.N(4607785, 'arguments', arguments, true, false, false);
                            url = J$.N(4607793, 'url', url, true, false, false);
                            J$.N(4607801, 'digest', digest, false, false, false);
                            var digest = J$.W(4607665, 'digest', J$.M(4607657, J$.M(4607633, J$.M(4607617, J$.M(4607601, J$.R(4607585, 'crypto', crypto, false, true), 'createHash', false)(J$.T(4607593, 'md5', 21, false)), 'update', false)(J$.R(4607609, 'url', url, false, false)), 'digest', false)(J$.T(4607625, 'hex', 21, false)), 'slice', false)(J$.T(4607641, 0, 22, false), J$.T(4607649, 6, 22, false)), digest, false, false);
                            return J$.Rt(4607769, J$.M(4607761, J$.R(4607673, 'path', path, false, true), 'join', false)(J$.F(4607689, J$.R(4607681, 'prebuildCache', prebuildCache, false, true), false)(), J$.B(558202, '+', J$.B(558194, '+', J$.R(4607697, 'digest', digest, false, false), J$.T(4607705, '-', 21, false)), J$.M(4607753, J$.M(4607729, J$.R(4607713, 'path', path, false, true), 'basename', false)(J$.R(4607721, 'url', url, false, false)), 'replace', false)(J$.T(4607737, /[^a-zA-Z0-9.]+/g, 14, false), J$.T(4607745, '-', 21, false)))));
                        } catch (J$e) {
                            J$.Ex(4608681, J$e);
                        } finally {
                            if (J$.Fr(4608689))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function npmCache() {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(4607953, arguments.callee, this, arguments);
                            arguments = J$.N(4607961, 'arguments', arguments, true, false, false);
                            J$.N(4607969, 'env', env, false, false, false);
                            var env = J$.W(4607825, 'env', J$.G(4607817, J$.I(typeof process === 'undefined' ? process = J$.R(4607809, 'process', undefined, true, true) : process = J$.R(4607809, 'process', process, true, true)), 'env'), env, false, false);
                            return J$.Rt(4607945, J$.C(254120, J$.G(4607841, J$.R(4607833, 'env', env, false, false), 'npm_config_cache')) ? J$._() : J$.C(254112, J$.G(4607857, J$.R(4607849, 'env', env, false, false), 'APPDATA')) ? J$.M(4607897, J$.R(4607865, 'path', path, false, true), 'join', false)(J$.G(4607881, J$.R(4607873, 'env', env, false, false), 'APPDATA'), J$.T(4607889, 'npm-cache', 21, false)) : J$.M(4607937, J$.R(4607905, 'path', path, false, true), 'join', false)(J$.F(4607921, J$.R(4607913, 'home', home, false, true), false)(), J$.T(4607929, '.npm', 21, false)));
                        } catch (J$e) {
                            J$.Ex(4608697, J$e);
                        } finally {
                            if (J$.Fr(4608705))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function prebuildCache() {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(4608025, arguments.callee, this, arguments);
                            arguments = J$.N(4608033, 'arguments', arguments, true, false, false);
                            return J$.Rt(4608017, J$.M(4608009, J$.R(4607977, 'path', path, false, true), 'join', false)(J$.F(4607993, J$.R(4607985, 'npmCache', npmCache, false, true), false)(), J$.T(4608001, '_prebuilds', 21, false)));
                        } catch (J$e) {
                            J$.Ex(4608713, J$e);
                        } finally {
                            if (J$.Fr(4608721))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function tempFile(cached) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(4608145, arguments.callee, this, arguments);
                            arguments = J$.N(4608153, 'arguments', arguments, true, false, false);
                            cached = J$.N(4608161, 'cached', cached, true, false, false);
                            return J$.Rt(4608137, J$.B(558242, '+', J$.B(558234, '+', J$.B(558226, '+', J$.B(558218, '+', J$.B(558210, '+', J$.R(4608041, 'cached', cached, false, false), J$.T(4608049, '.', 21, false)), J$.G(4608065, J$.I(typeof process === 'undefined' ? process = J$.R(4608057, 'process', undefined, true, true) : process = J$.R(4608057, 'process', process, true, true)), 'pid')), J$.T(4608073, '-', 21, false)), J$.M(4608121, J$.M(4608105, J$.M(4608089, J$.I(typeof Math === 'undefined' ? Math = J$.R(4608081, 'Math', undefined, true, true) : Math = J$.R(4608081, 'Math', Math, true, true)), 'random', false)(), 'toString', false)(J$.T(4608097, 16, 22, false)), 'slice', false)(J$.T(4608113, 2, 22, false))), J$.T(4608129, '.tmp', 21, false)));
                        } catch (J$e) {
                            J$.Ex(4608729, J$e);
                        } finally {
                            if (J$.Fr(4608737))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(4608369, 'path', path, false, false, false);
            J$.N(4608377, 'github', github, false, false, false);
            J$.N(4608385, 'home', home, false, false, false);
            J$.N(4608393, 'crypto', crypto, false, false, false);
            J$.N(4608401, 'expandTemplate', expandTemplate, false, false, false);
            getDownloadUrl = J$.N(4608417, 'getDownloadUrl', J$.T(4608409, getDownloadUrl, 12, false), true, false, false);
            getApiUrl = J$.N(4608433, 'getApiUrl', J$.T(4608425, getApiUrl, 12, false), true, false, false);
            getAssetUrl = J$.N(4608449, 'getAssetUrl', J$.T(4608441, getAssetUrl, 12, false), true, false, false);
            urlTemplate = J$.N(4608465, 'urlTemplate', J$.T(4608457, urlTemplate, 12, false), true, false, false);
            getHostMirrorUrl = J$.N(4608481, 'getHostMirrorUrl', J$.T(4608473, getHostMirrorUrl, 12, false), true, false, false);
            trimSlashes = J$.N(4608497, 'trimSlashes', J$.T(4608489, trimSlashes, 12, false), true, false, false);
            cachedPrebuild = J$.N(4608513, 'cachedPrebuild', J$.T(4608505, cachedPrebuild, 12, false), true, false, false);
            npmCache = J$.N(4608529, 'npmCache', J$.T(4608521, npmCache, 12, false), true, false, false);
            prebuildCache = J$.N(4608545, 'prebuildCache', J$.T(4608537, prebuildCache, 12, false), true, false, false);
            tempFile = J$.N(4608561, 'tempFile', J$.T(4608553, tempFile, 12, false), true, false, false);
            var path = J$.W(4605729, 'path', J$.F(4605721, J$.I(typeof require === 'undefined' ? require = J$.R(4605705, 'require', undefined, true, true) : require = J$.R(4605705, 'require', require, true, true)), false)(J$.T(4605713, 'path', 21, false)), path, false, true);
            var github = J$.W(4605761, 'github', J$.F(4605753, J$.I(typeof require === 'undefined' ? require = J$.R(4605737, 'require', undefined, true, true) : require = J$.R(4605737, 'require', require, true, true)), false)(J$.T(4605745, 'github-from-package', 21, false)), github, false, true);
            var home = J$.W(4605801, 'home', J$.G(4605793, J$.F(4605785, J$.I(typeof require === 'undefined' ? require = J$.R(4605769, 'require', undefined, true, true) : require = J$.R(4605769, 'require', require, true, true)), false)(J$.T(4605777, 'os', 21, false)), 'homedir'), home, false, true);
            var crypto = J$.W(4605833, 'crypto', J$.F(4605825, J$.I(typeof require === 'undefined' ? require = J$.R(4605809, 'require', undefined, true, true) : require = J$.R(4605809, 'require', require, true, true)), false)(J$.T(4605817, 'crypto', 21, false)), crypto, false, true);
            var expandTemplate = J$.W(4605873, 'expandTemplate', J$.F(4605865, J$.F(4605857, J$.I(typeof require === 'undefined' ? require = J$.R(4605841, 'require', undefined, true, true) : require = J$.R(4605841, 'require', require, true, true)), false)(J$.T(4605849, 'expand-template', 21, false)), false)(), expandTemplate, false, true);
            J$.P(4608185, J$.I(typeof exports === 'undefined' ? exports = J$.R(4608169, 'exports', undefined, true, true) : exports = J$.R(4608169, 'exports', exports, true, true)), 'getDownloadUrl', J$.R(4608177, 'getDownloadUrl', getDownloadUrl, false, true));
            J$.P(4608209, J$.I(typeof exports === 'undefined' ? exports = J$.R(4608193, 'exports', undefined, true, true) : exports = J$.R(4608193, 'exports', exports, true, true)), 'getApiUrl', J$.R(4608201, 'getApiUrl', getApiUrl, false, true));
            J$.P(4608233, J$.I(typeof exports === 'undefined' ? exports = J$.R(4608217, 'exports', undefined, true, true) : exports = J$.R(4608217, 'exports', exports, true, true)), 'getAssetUrl', J$.R(4608225, 'getAssetUrl', getAssetUrl, false, true));
            J$.P(4608257, J$.I(typeof exports === 'undefined' ? exports = J$.R(4608241, 'exports', undefined, true, true) : exports = J$.R(4608241, 'exports', exports, true, true)), 'urlTemplate', J$.R(4608249, 'urlTemplate', urlTemplate, false, true));
            J$.P(4608281, J$.I(typeof exports === 'undefined' ? exports = J$.R(4608265, 'exports', undefined, true, true) : exports = J$.R(4608265, 'exports', exports, true, true)), 'cachedPrebuild', J$.R(4608273, 'cachedPrebuild', cachedPrebuild, false, true));
            J$.P(4608305, J$.I(typeof exports === 'undefined' ? exports = J$.R(4608289, 'exports', undefined, true, true) : exports = J$.R(4608289, 'exports', exports, true, true)), 'prebuildCache', J$.R(4608297, 'prebuildCache', prebuildCache, false, true));
            J$.P(4608329, J$.I(typeof exports === 'undefined' ? exports = J$.R(4608313, 'exports', undefined, true, true) : exports = J$.R(4608313, 'exports', exports, true, true)), 'npmCache', J$.R(4608321, 'npmCache', npmCache, false, true));
            J$.P(4608353, J$.I(typeof exports === 'undefined' ? exports = J$.R(4608337, 'exports', undefined, true, true) : exports = J$.R(4608337, 'exports', exports, true, true)), 'tempFile', J$.R(4608345, 'tempFile', tempFile, false, true));
        } catch (J$e) {
            J$.Ex(4608745, J$e);
        } finally {
            if (J$.Sr(4608753))
                continue jalangiLabel11;
            else
                break jalangiLabel11;
        }
    }
// JALANGI DO NOT INSTRUMENT
