// request/lib/getProxyFromURI.js
J$.noInstrEval = false;
jalangiLabel5:
    while (true) {
        try {
            J$.Se(4917177, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/request/lib/getProxyFromURI.js');
            function formatHostname(hostname) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(4915945, arguments.callee, this, arguments);
                            arguments = J$.N(4915953, 'arguments', arguments, true, false, false);
                            hostname = J$.N(4915961, 'hostname', hostname, true, false, false);
                            return J$.Rt(4915937, J$.M(4915929, J$.M(4915921, J$.R(4915897, 'hostname', hostname, false, false), 'replace', false)(J$.T(4915905, /^\.*/, 14, false), J$.T(4915913, '.', 21, false)), 'toLowerCase', false)());
                        } catch (J$e) {
                            J$.Ex(4917249, J$e);
                        } finally {
                            if (J$.Fr(4917257))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function parseNoProxyZone(zone) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(4916201, arguments.callee, this, arguments);
                            arguments = J$.N(4916209, 'arguments', arguments, true, false, false);
                            zone = J$.N(4916217, 'zone', zone, true, false, false);
                            J$.N(4916225, 'zoneParts', zoneParts, false, false, false);
                            J$.N(4916233, 'zoneHost', zoneHost, false, false, false);
                            J$.N(4916241, 'zonePort', zonePort, false, false, false);
                            J$.N(4916249, 'hasPort', hasPort, false, false, false);
                            zone = J$.W(4915993, 'zone', J$.M(4915985, J$.M(4915977, J$.R(4915969, 'zone', zone, false, false), 'trim', false)(), 'toLowerCase', false)(), zone, false, false);
                            var zoneParts = J$.W(4916033, 'zoneParts', J$.M(4916025, J$.R(4916001, 'zone', zone, false, false), 'split', false)(J$.T(4916009, ':', 21, false), J$.T(4916017, 2, 22, false)), zoneParts, false, false);
                            var zoneHost = J$.W(4916081, 'zoneHost', J$.F(4916073, J$.R(4916041, 'formatHostname', formatHostname, false, true), false)(J$.G(4916065, J$.R(4916049, 'zoneParts', zoneParts, false, false), J$.T(4916057, 0, 22, false))), zoneHost, false, false);
                            var zonePort = J$.W(4916113, 'zonePort', J$.G(4916105, J$.R(4916089, 'zoneParts', zoneParts, false, false), J$.T(4916097, 1, 22, false)), zonePort, false, false);
                            var hasPort = J$.W(4916153, 'hasPort', J$.B(572714, '>', J$.M(4916137, J$.R(4916121, 'zone', zone, false, false), 'indexOf', false)(J$.T(4916129, ':', 21, false)), J$.U(572706, '-', J$.T(4916145, 1, 22, false))), hasPort, false, false);
                            return J$.Rt(4916193, J$.T(4916185, {
                                hostname: J$.R(4916161, 'zoneHost', zoneHost, false, false),
                                port: J$.R(4916169, 'zonePort', zonePort, false, false),
                                hasPort: J$.R(4916177, 'hasPort', hasPort, false, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(4917265, J$e);
                        } finally {
                            if (J$.Fr(4917273))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function uriInNoProxy(uri, noProxy) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(4916665, arguments.callee, this, arguments);
                            arguments = J$.N(4916673, 'arguments', arguments, true, false, false);
                            uri = J$.N(4916681, 'uri', uri, true, false, false);
                            noProxy = J$.N(4916689, 'noProxy', noProxy, true, false, false);
                            J$.N(4916697, 'port', port, false, false, false);
                            J$.N(4916705, 'hostname', hostname, false, false, false);
                            J$.N(4916713, 'noProxyList', noProxyList, false, false, false);
                            var port = J$.W(4916313, 'port', J$.C(263392, J$.G(4916265, J$.R(4916257, 'uri', uri, false, false), 'port')) ? J$._() : J$.C(263384, J$.B(572722, '===', J$.G(4916281, J$.R(4916273, 'uri', uri, false, false), 'protocol'), J$.T(4916289, 'https:', 21, false))) ? J$.T(4916297, '443', 21, false) : J$.T(4916305, '80', 21, false), port, false, false);
                            var hostname = J$.W(4916353, 'hostname', J$.F(4916345, J$.R(4916321, 'formatHostname', formatHostname, false, true), false)(J$.G(4916337, J$.R(4916329, 'uri', uri, false, false), 'hostname')), hostname, false, false);
                            var noProxyList = J$.W(4916385, 'noProxyList', J$.M(4916377, J$.R(4916361, 'noProxy', noProxy, false, false), 'split', false)(J$.T(4916369, ',', 21, false)), noProxyList, false, false);
                            return J$.Rt(4916657, J$.M(4916649, J$.M(4916409, J$.R(4916393, 'noProxyList', noProxyList, false, false), 'map', false)(J$.R(4916401, 'parseNoProxyZone', parseNoProxyZone, false, true)), 'some', false)(J$.T(4916641, function (noProxyZone) {
                                jalangiLabel2:
                                    while (true) {
                                        try {
                                            J$.Fe(4916601, arguments.callee, this, arguments);
                                            arguments = J$.N(4916609, 'arguments', arguments, true, false, false);
                                            noProxyZone = J$.N(4916617, 'noProxyZone', noProxyZone, true, false, false);
                                            J$.N(4916625, 'isMatchedAt', isMatchedAt, false, false, false);
                                            J$.N(4916633, 'hostnameMatched', hostnameMatched, false, false, false);
                                            var isMatchedAt = J$.W(4916449, 'isMatchedAt', J$.M(4916441, J$.R(4916417, 'hostname', hostname, false, false), 'indexOf', false)(J$.G(4916433, J$.R(4916425, 'noProxyZone', noProxyZone, false, false), 'hostname')), isMatchedAt, false, false);
                                            var hostnameMatched = J$.W(4916521, 'hostnameMatched', J$.C(263400, J$.B(572738, '>', J$.R(4916457, 'isMatchedAt', isMatchedAt, false, false), J$.U(572730, '-', J$.T(4916465, 1, 22, false)))) ? J$.B(572754, '===', J$.R(4916473, 'isMatchedAt', isMatchedAt, false, false), J$.B(572746, '-', J$.G(4916489, J$.R(4916481, 'hostname', hostname, false, false), 'length'), J$.G(4916513, J$.G(4916505, J$.R(4916497, 'noProxyZone', noProxyZone, false, false), 'hostname'), 'length'))) : J$._(), hostnameMatched, false, false);
                                            if (J$.C(263416, J$.G(4916537, J$.R(4916529, 'noProxyZone', noProxyZone, false, false), 'hasPort'))) {
                                                return J$.Rt(4916577, J$.C(263408, J$.B(572762, '===', J$.R(4916545, 'port', port, false, false), J$.G(4916561, J$.R(4916553, 'noProxyZone', noProxyZone, false, false), 'port'))) ? J$.R(4916569, 'hostnameMatched', hostnameMatched, false, false) : J$._());
                                            }
                                            return J$.Rt(4916593, J$.R(4916585, 'hostnameMatched', hostnameMatched, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4917281, J$e);
                                        } finally {
                                            if (J$.Fr(4917289))
                                                continue jalangiLabel2;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(4917297, J$e);
                        } finally {
                            if (J$.Fr(4917305))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function getProxyFromURI(uri) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(4917121, arguments.callee, this, arguments);
                            arguments = J$.N(4917129, 'arguments', arguments, true, false, false);
                            uri = J$.N(4917137, 'uri', uri, true, false, false);
                            J$.N(4917145, 'noProxy', noProxy, false, false, false);
                            var noProxy = J$.W(4916777, 'noProxy', J$.C(263432, J$.C(263424, J$.G(4916737, J$.G(4916729, J$.I(typeof process === 'undefined' ? process = J$.R(4916721, 'process', undefined, true, true) : process = J$.R(4916721, 'process', process, true, true)), 'env'), 'NO_PROXY')) ? J$._() : J$.G(4916761, J$.G(4916753, J$.I(typeof process === 'undefined' ? process = J$.R(4916745, 'process', undefined, true, true) : process = J$.R(4916745, 'process', process, true, true)), 'env'), 'no_proxy')) ? J$._() : J$.T(4916769, '', 21, false), noProxy, false, false);
                            if (J$.C(263440, J$.B(572770, '===', J$.R(4916785, 'noProxy', noProxy, false, false), J$.T(4916793, '*', 21, false)))) {
                                return J$.Rt(4916809, J$.T(4916801, null, 25, false));
                            }
                            if (J$.C(263456, J$.C(263448, J$.B(572778, '!==', J$.R(4916817, 'noProxy', noProxy, false, false), J$.T(4916825, '', 21, false))) ? J$.F(4916857, J$.R(4916833, 'uriInNoProxy', uriInNoProxy, false, true), false)(J$.R(4916841, 'uri', uri, false, false), J$.R(4916849, 'noProxy', noProxy, false, false)) : J$._())) {
                                return J$.Rt(4916873, J$.T(4916865, null, 25, false));
                            }
                            if (J$.C(263480, J$.B(572786, '===', J$.G(4916889, J$.R(4916881, 'uri', uri, false, false), 'protocol'), J$.T(4916897, 'http:', 21, false)))) {
                                return J$.Rt(4916961, J$.C(263472, J$.C(263464, J$.G(4916921, J$.G(4916913, J$.I(typeof process === 'undefined' ? process = J$.R(4916905, 'process', undefined, true, true) : process = J$.R(4916905, 'process', process, true, true)), 'env'), 'HTTP_PROXY')) ? J$._() : J$.G(4916945, J$.G(4916937, J$.I(typeof process === 'undefined' ? process = J$.R(4916929, 'process', undefined, true, true) : process = J$.R(4916929, 'process', process, true, true)), 'env'), 'http_proxy')) ? J$._() : J$.T(4916953, null, 25, false));
                            }
                            if (J$.C(263520, J$.B(572794, '===', J$.G(4916977, J$.R(4916969, 'uri', uri, false, false), 'protocol'), J$.T(4916985, 'https:', 21, false)))) {
                                return J$.Rt(4917097, J$.C(263512, J$.C(263504, J$.C(263496, J$.C(263488, J$.G(4917009, J$.G(4917001, J$.I(typeof process === 'undefined' ? process = J$.R(4916993, 'process', undefined, true, true) : process = J$.R(4916993, 'process', process, true, true)), 'env'), 'HTTPS_PROXY')) ? J$._() : J$.G(4917033, J$.G(4917025, J$.I(typeof process === 'undefined' ? process = J$.R(4917017, 'process', undefined, true, true) : process = J$.R(4917017, 'process', process, true, true)), 'env'), 'https_proxy')) ? J$._() : J$.G(4917057, J$.G(4917049, J$.I(typeof process === 'undefined' ? process = J$.R(4917041, 'process', undefined, true, true) : process = J$.R(4917041, 'process', process, true, true)), 'env'), 'HTTP_PROXY')) ? J$._() : J$.G(4917081, J$.G(4917073, J$.I(typeof process === 'undefined' ? process = J$.R(4917065, 'process', undefined, true, true) : process = J$.R(4917065, 'process', process, true, true)), 'env'), 'http_proxy')) ? J$._() : J$.T(4917089, null, 25, false));
                            }
                            return J$.Rt(4917113, J$.T(4917105, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(4917313, J$e);
                        } finally {
                            if (J$.Fr(4917321))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            formatHostname = J$.N(4917193, 'formatHostname', J$.T(4917185, formatHostname, 12, false), true, false, false);
            parseNoProxyZone = J$.N(4917209, 'parseNoProxyZone', J$.T(4917201, parseNoProxyZone, 12, false), true, false, false);
            uriInNoProxy = J$.N(4917225, 'uriInNoProxy', J$.T(4917217, uriInNoProxy, 12, false), true, false, false);
            getProxyFromURI = J$.N(4917241, 'getProxyFromURI', J$.T(4917233, getProxyFromURI, 12, false), true, false, false);
            J$.T(4915889, 'use strict', 21, false);
            J$.P(4917169, J$.I(typeof module === 'undefined' ? module = J$.R(4917153, 'module', undefined, true, true) : module = J$.R(4917153, 'module', module, true, true)), 'exports', J$.R(4917161, 'getProxyFromURI', getProxyFromURI, false, true));
        } catch (J$e) {
            J$.Ex(4917329, J$e);
        } finally {
            if (J$.Sr(4917337))
                continue jalangiLabel5;
            else
                break jalangiLabel5;
        }
    }
// JALANGI DO NOT INSTRUMENT
