// synchronous-promise/index.js
J$.noInstrEval = false;
jalangiLabel56:
    while (true) {
        try {
            J$.Se(4637225, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestYup/node_modules/synchronous-promise/index.js');
            function makeArrayFrom(obj) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(4630761, arguments.callee, this, arguments);
                            arguments = J$.N(4630769, 'arguments', arguments, true, false, false);
                            obj = J$.N(4630777, 'obj', obj, true, false, false);
                            return J$.Rt(4630753, J$.M(4630745, J$.G(4630729, J$.G(4630721, J$.I(typeof Array === 'undefined' ? Array = J$.R(4630713, 'Array', undefined, true, true) : Array = J$.R(4630713, 'Array', Array, true, true)), 'prototype'), 'slice'), 'apply', false)(J$.R(4630737, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(4637329, J$e);
                        } finally {
                            if (J$.Fr(4637337))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function SynchronousPromise(handler) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(4631025, arguments.callee, this, arguments);
                            arguments = J$.N(4631033, 'arguments', arguments, true, false, false);
                            handler = J$.N(4631041, 'handler', handler, true, false, false);
                            J$.P(4630849, J$.R(4630833, 'this', this, false, false), 'status', J$.R(4630841, 'PENDING', PENDING, false, true));
                            J$.P(4630873, J$.R(4630857, 'this', this, false, false), '_continuations', J$.T(4630865, [], 10, false));
                            J$.P(4630897, J$.R(4630881, 'this', this, false, false), '_parent', J$.T(4630889, null, 25, false));
                            J$.P(4630921, J$.R(4630905, 'this', this, false, false), '_paused', J$.T(4630913, false, 23, false));
                            if (J$.C(247920, J$.R(4630929, 'handler', handler, false, false))) {
                                J$.M(4631017, J$.R(4630937, 'handler', handler, false, false), 'call', false)(J$.R(4630945, 'this', this, false, false), J$.M(4630977, J$.G(4630961, J$.R(4630953, 'this', this, false, false), '_continueWith'), 'bind', false)(J$.R(4630969, 'this', this, false, false)), J$.M(4631009, J$.G(4630993, J$.R(4630985, 'this', this, false, false), '_failWith'), 'bind', false)(J$.R(4631001, 'this', this, false, false)));
                            }
                        } catch (J$e) {
                            J$.Ex(4637345, J$e);
                        } finally {
                            if (J$.Fr(4637353))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function looksLikeAPromise(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(4631089, arguments.callee, this, arguments);
                            arguments = J$.N(4631097, 'arguments', arguments, true, false, false);
                            obj = J$.N(4631105, 'obj', obj, true, false, false);
                            return J$.Rt(4631081, J$.C(247928, J$.R(4631049, 'obj', obj, false, false)) ? J$.B(478714, '===', J$.U(478706, 'typeof', J$.G(4631065, J$.R(4631057, 'obj', obj, false, false), 'then')), J$.T(4631073, 'function', 21, false)) : J$._());
                        } catch (J$e) {
                            J$.Ex(4637361, J$e);
                        } finally {
                            if (J$.Fr(4637369))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function patchAwaiterIfRequired(__awaiter) {
                jalangiLabel55:
                    while (true) {
                        try {
                            J$.Fe(4637161, arguments.callee, this, arguments);
                            arguments = J$.N(4637169, 'arguments', arguments, true, false, false);
                            __awaiter = J$.N(4637177, '__awaiter', __awaiter, true, false, false);
                            J$.N(4637185, 'originalAwaiter', originalAwaiter, false, false, false);
                            if (J$.C(248232, J$.C(248224, J$.B(478842, '===', J$.U(478834, 'typeof', J$.R(4636953, '__awaiter', __awaiter, false, false)), J$.T(4636961, 'undefined', 21, false))) ? J$._() : J$.G(4636977, J$.R(4636969, '__awaiter', __awaiter, false, false), '__patched'))) {
                                return J$.Rt(4636993, J$.R(4636985, '__awaiter', __awaiter, false, false));
                            }
                            var originalAwaiter = J$.W(4637009, 'originalAwaiter', J$.R(4637001, '__awaiter', __awaiter, false, false), originalAwaiter, false, false);
                            __awaiter = J$.W(4637113, '__awaiter', J$.T(4637105, function __awaiter() {
                                jalangiLabel54:
                                    while (true) {
                                        try {
                                            J$.Fe(4637081, arguments.callee, this, arguments);
                                            arguments = J$.N(4637089, 'arguments', arguments, true, false, false);
                                            J$.N(4637097, 'Promise', Promise, false, false, false);
                                            var Promise = J$.W(4637025, 'Promise', J$.R(4637017, 'RealPromise', RealPromise, false, true), Promise, false, false);
                                            J$.M(4637073, J$.R(4637033, 'originalAwaiter', originalAwaiter, false, false), 'apply', false)(J$.R(4637041, 'this', this, false, false), J$.F(4637065, J$.R(4637049, 'makeArrayFrom', makeArrayFrom, false, true), false)(J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4637057, 'arguments', undefined, true, true) : arguments = J$.R(4637057, 'arguments', arguments, true, true))));
                                        } catch (J$e) {
                                            J$.Ex(4638193, J$e);
                                        } finally {
                                            if (J$.Fr(4638201))
                                                continue jalangiLabel54;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), __awaiter, false, false);
                            J$.P(4637137, J$.R(4637121, '__awaiter', __awaiter, false, false), '__patched', J$.T(4637129, true, 23, false));
                            return J$.Rt(4637153, J$.R(4637145, '__awaiter', __awaiter, false, false));
                        } catch (J$e) {
                            J$.Ex(4638209, J$e);
                        } finally {
                            if (J$.Fr(4638217))
                                continue jalangiLabel55;
                            else
                                return J$.Ra();
                        }
                    }
            }
            makeArrayFrom = J$.N(4637241, 'makeArrayFrom', J$.T(4637233, makeArrayFrom, 12, false), true, false, false);
            J$.N(4637249, 'PENDING', PENDING, false, false, false);
            J$.N(4637257, 'RESOLVED', RESOLVED, false, false, false);
            J$.N(4637265, 'REJECTED', REJECTED, false, false, false);
            SynchronousPromise = J$.N(4637281, 'SynchronousPromise', J$.T(4637273, SynchronousPromise, 12, false), true, false, false);
            looksLikeAPromise = J$.N(4637297, 'looksLikeAPromise', J$.T(4637289, looksLikeAPromise, 12, false), true, false, false);
            J$.N(4637305, 'RealPromise', RealPromise, false, false, false);
            patchAwaiterIfRequired = J$.N(4637321, 'patchAwaiterIfRequired', J$.T(4637313, patchAwaiterIfRequired, 12, false), true, false, false);
            J$.T(4630489, 'use strict', 21, false);
            J$.F(4630513, J$.I(typeof require === 'undefined' ? require = J$.R(4630497, 'require', undefined, true, true) : require = J$.R(4630497, 'require', require, true, true)), false)(J$.T(4630505, 'core-js/modules/es.array.for-each', 21, false));
            J$.F(4630537, J$.I(typeof require === 'undefined' ? require = J$.R(4630521, 'require', undefined, true, true) : require = J$.R(4630521, 'require', require, true, true)), false)(J$.T(4630529, 'core-js/modules/es.array.is-array', 21, false));
            J$.F(4630561, J$.I(typeof require === 'undefined' ? require = J$.R(4630545, 'require', undefined, true, true) : require = J$.R(4630545, 'require', require, true, true)), false)(J$.T(4630553, 'core-js/modules/es.array.reduce', 21, false));
            J$.F(4630585, J$.I(typeof require === 'undefined' ? require = J$.R(4630569, 'require', undefined, true, true) : require = J$.R(4630569, 'require', require, true, true)), false)(J$.T(4630577, 'core-js/modules/es.array.slice', 21, false));
            J$.F(4630609, J$.I(typeof require === 'undefined' ? require = J$.R(4630593, 'require', undefined, true, true) : require = J$.R(4630593, 'require', require, true, true)), false)(J$.T(4630601, 'core-js/modules/es.array.splice', 21, false));
            J$.F(4630633, J$.I(typeof require === 'undefined' ? require = J$.R(4630617, 'require', undefined, true, true) : require = J$.R(4630617, 'require', require, true, true)), false)(J$.T(4630625, 'core-js/modules/es.function.bind', 21, false));
            J$.F(4630657, J$.I(typeof require === 'undefined' ? require = J$.R(4630641, 'require', undefined, true, true) : require = J$.R(4630641, 'require', require, true, true)), false)(J$.T(4630649, 'core-js/modules/es.object.to-string', 21, false));
            J$.F(4630681, J$.I(typeof require === 'undefined' ? require = J$.R(4630665, 'require', undefined, true, true) : require = J$.R(4630665, 'require', require, true, true)), false)(J$.T(4630673, 'core-js/modules/es.promise', 21, false));
            J$.F(4630705, J$.I(typeof require === 'undefined' ? require = J$.R(4630689, 'require', undefined, true, true) : require = J$.R(4630689, 'require', require, true, true)), false)(J$.T(4630697, 'core-js/modules/web.dom-collections.for-each', 21, false));
            var PENDING = J$.W(4630809, 'PENDING', J$.T(4630785, 'pending', 21, false), PENDING, false, true), RESOLVED = J$.W(4630817, 'RESOLVED', J$.T(4630793, 'resolved', 21, false), RESOLVED, false, true), REJECTED = J$.W(4630825, 'REJECTED', J$.T(4630801, 'rejected', 21, false), REJECTED, false, true);
            J$.P(4635281, J$.R(4631113, 'SynchronousPromise', SynchronousPromise, false, true), 'prototype', J$.T(4635273, {
                then: J$.T(4631681, function then(nextFn, catchFn) {
                    jalangiLabel3:
                        while (true) {
                            try {
                                J$.Fe(4631633, arguments.callee, this, arguments);
                                arguments = J$.N(4631641, 'arguments', arguments, true, false, false);
                                nextFn = J$.N(4631649, 'nextFn', nextFn, true, false, false);
                                catchFn = J$.N(4631657, 'catchFn', catchFn, true, false, false);
                                J$.N(4631665, 'next', next, false, false, false);
                                J$.N(4631673, 'catchResult', catchResult, false, false, false);
                                var next = J$.W(4631153, 'next', J$.M(4631145, J$.M(4631129, J$.R(4631121, 'SynchronousPromise', SynchronousPromise, false, true), 'unresolved', false)(), '_setParent', false)(J$.R(4631137, 'this', this, false, false)), next, false, false);
                                if (J$.C(247960, J$.M(4631169, J$.R(4631161, 'this', this, false, false), '_isRejected', false)())) {
                                    if (J$.C(247936, J$.G(4631185, J$.R(4631177, 'this', this, false, false), '_paused'))) {
                                        J$.M(4631241, J$.G(4631201, J$.R(4631193, 'this', this, false, false), '_continuations'), 'push', false)(J$.T(4631233, {
                                            promise: J$.R(4631209, 'next', next, false, false),
                                            nextFn: J$.R(4631217, 'nextFn', nextFn, false, false),
                                            catchFn: J$.R(4631225, 'catchFn', catchFn, false, false)
                                        }, 11, false));
                                        return J$.Rt(4631257, J$.R(4631249, 'next', next, false, false));
                                    }
                                    if (J$.C(247952, J$.R(4631265, 'catchFn', catchFn, false, false))) {
                                        try {
                                            var catchResult = J$.W(4631305, 'catchResult', J$.F(4631297, J$.R(4631273, 'catchFn', catchFn, false, false), false)(J$.G(4631289, J$.R(4631281, 'this', this, false, false), '_error')), catchResult, false, false);
                                            if (J$.C(247944, J$.F(4631329, J$.R(4631313, 'looksLikeAPromise', looksLikeAPromise, false, true), false)(J$.R(4631321, 'catchResult', catchResult, false, false)))) {
                                                J$.M(4631361, J$.R(4631337, 'this', this, false, false), '_chainPromiseData', false)(J$.R(4631345, 'catchResult', catchResult, false, false), J$.R(4631353, 'next', next, false, false));
                                                return J$.Rt(4631377, J$.R(4631369, 'next', next, false, false));
                                            } else {
                                                return J$.Rt(4631425, J$.M(4631417, J$.M(4631401, J$.R(4631385, 'SynchronousPromise', SynchronousPromise, false, true), 'resolve', false)(J$.R(4631393, 'catchResult', catchResult, false, false)), '_setParent', false)(J$.R(4631409, 'this', this, false, false)));
                                            }
                                        } catch (e) {
                                            J$.N(4631481, 'e', e, false, false, true);
                                            return J$.Rt(4631473, J$.M(4631465, J$.M(4631449, J$.R(4631433, 'SynchronousPromise', SynchronousPromise, false, true), 'reject', false)(J$.R(4631441, 'e', e, false, false)), '_setParent', false)(J$.R(4631457, 'this', this, false, false)));
                                        }
                                    }
                                    return J$.Rt(4631537, J$.M(4631529, J$.M(4631513, J$.R(4631489, 'SynchronousPromise', SynchronousPromise, false, true), 'reject', false)(J$.G(4631505, J$.R(4631497, 'this', this, false, false), '_error')), '_setParent', false)(J$.R(4631521, 'this', this, false, false)));
                                }
                                J$.M(4631593, J$.G(4631553, J$.R(4631545, 'this', this, false, false), '_continuations'), 'push', false)(J$.T(4631585, {
                                    promise: J$.R(4631561, 'next', next, false, false),
                                    nextFn: J$.R(4631569, 'nextFn', nextFn, false, false),
                                    catchFn: J$.R(4631577, 'catchFn', catchFn, false, false)
                                }, 11, false));
                                J$.M(4631609, J$.R(4631601, 'this', this, false, false), '_runResolutions', false)();
                                return J$.Rt(4631625, J$.R(4631617, 'next', next, false, false));
                            } catch (J$e) {
                                J$.Ex(4637377, J$e);
                            } finally {
                                if (J$.Fr(4637385))
                                    continue jalangiLabel3;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                'catch': J$.T(4631913, function _catch(handler) {
                    jalangiLabel4:
                        while (true) {
                            try {
                                J$.Fe(4631881, arguments.callee, this, arguments);
                                arguments = J$.N(4631889, 'arguments', arguments, true, false, false);
                                handler = J$.N(4631897, 'handler', handler, true, false, false);
                                J$.N(4631905, 'next', next, false, false, false);
                                if (J$.C(247968, J$.M(4631697, J$.R(4631689, 'this', this, false, false), '_isResolved', false)())) {
                                    return J$.Rt(4631753, J$.M(4631745, J$.M(4631729, J$.R(4631705, 'SynchronousPromise', SynchronousPromise, false, true), 'resolve', false)(J$.G(4631721, J$.R(4631713, 'this', this, false, false), '_data')), '_setParent', false)(J$.R(4631737, 'this', this, false, false)));
                                }
                                var next = J$.W(4631793, 'next', J$.M(4631785, J$.M(4631769, J$.R(4631761, 'SynchronousPromise', SynchronousPromise, false, true), 'unresolved', false)(), '_setParent', false)(J$.R(4631777, 'this', this, false, false)), next, false, false);
                                J$.M(4631841, J$.G(4631809, J$.R(4631801, 'this', this, false, false), '_continuations'), 'push', false)(J$.T(4631833, {
                                    promise: J$.R(4631817, 'next', next, false, false),
                                    catchFn: J$.R(4631825, 'handler', handler, false, false)
                                }, 11, false));
                                J$.M(4631857, J$.R(4631849, 'this', this, false, false), '_runRejections', false)();
                                return J$.Rt(4631873, J$.R(4631865, 'next', next, false, false));
                            } catch (J$e) {
                                J$.Ex(4637393, J$e);
                            } finally {
                                if (J$.Fr(4637401))
                                    continue jalangiLabel4;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                'finally': J$.T(4632105, function _finally(callback) {
                    jalangiLabel6:
                        while (true) {
                            try {
                                J$.Fe(4632057, arguments.callee, this, arguments);
                                function runFinally() {
                                    jalangiLabel5:
                                        while (true) {
                                            try {
                                                J$.Fe(4631985, arguments.callee, this, arguments);
                                                arguments = J$.N(4631993, 'arguments', arguments, true, false, false);
                                                if (J$.C(247976, J$.U(478722, '!', J$.R(4631937, 'ran', ran, false, false)))) {
                                                    ran = J$.W(4631953, 'ran', J$.T(4631945, true, 23, false), ran, false, false);
                                                    return J$.Rt(4631977, J$.F(4631969, J$.R(4631961, 'callback', callback, false, false), false)());
                                                }
                                            } catch (J$e) {
                                                J$.Ex(4637409, J$e);
                                            } finally {
                                                if (J$.Fr(4637417))
                                                    continue jalangiLabel5;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }
                                arguments = J$.N(4632065, 'arguments', arguments, true, false, false);
                                callback = J$.N(4632073, 'callback', callback, true, false, false);
                                J$.N(4632081, 'ran', ran, false, false, false);
                                runFinally = J$.N(4632097, 'runFinally', J$.T(4632089, runFinally, 12, false), true, false, false);
                                var ran = J$.W(4631929, 'ran', J$.T(4631921, false, 23, false), ran, false, false);
                                return J$.Rt(4632049, J$.M(4632041, J$.M(4632017, J$.R(4632001, 'this', this, false, false), 'then', false)(J$.R(4632009, 'runFinally', runFinally, false, false)), J$.T(4632025, 'catch', 21, false), false)(J$.R(4632033, 'runFinally', runFinally, false, false)));
                            } catch (J$e) {
                                J$.Ex(4637425, J$e);
                            } finally {
                                if (J$.Fr(4637433))
                                    continue jalangiLabel6;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                pause: J$.T(4632169, function pause() {
                    jalangiLabel7:
                        while (true) {
                            try {
                                J$.Fe(4632153, arguments.callee, this, arguments);
                                arguments = J$.N(4632161, 'arguments', arguments, true, false, false);
                                J$.P(4632129, J$.R(4632113, 'this', this, false, false), '_paused', J$.T(4632121, true, 23, false));
                                return J$.Rt(4632145, J$.R(4632137, 'this', this, false, false));
                            } catch (J$e) {
                                J$.Ex(4637441, J$e);
                            } finally {
                                if (J$.Fr(4637449))
                                    continue jalangiLabel7;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                resume: J$.T(4632305, function resume() {
                    jalangiLabel8:
                        while (true) {
                            try {
                                J$.Fe(4632281, arguments.callee, this, arguments);
                                arguments = J$.N(4632289, 'arguments', arguments, true, false, false);
                                J$.N(4632297, 'firstPaused', firstPaused, false, false, false);
                                var firstPaused = J$.W(4632193, 'firstPaused', J$.M(4632185, J$.R(4632177, 'this', this, false, false), '_findFirstPaused', false)(), firstPaused, false, false);
                                if (J$.C(247984, J$.R(4632201, 'firstPaused', firstPaused, false, false))) {
                                    J$.P(4632225, J$.R(4632209, 'firstPaused', firstPaused, false, false), '_paused', J$.T(4632217, false, 23, false));
                                    J$.M(4632241, J$.R(4632233, 'firstPaused', firstPaused, false, false), '_runResolutions', false)();
                                    J$.M(4632257, J$.R(4632249, 'firstPaused', firstPaused, false, false), '_runRejections', false)();
                                }
                                return J$.Rt(4632273, J$.R(4632265, 'this', this, false, false));
                            } catch (J$e) {
                                J$.Ex(4637457, J$e);
                            } finally {
                                if (J$.Fr(4637465))
                                    continue jalangiLabel8;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _findAncestry: J$.T(4632529, function _findAncestry() {
                    jalangiLabel10:
                        while (true) {
                            try {
                                J$.Fe(4632513, arguments.callee, this, arguments);
                                arguments = J$.N(4632521, 'arguments', arguments, true, false, false);
                                return J$.Rt(4632505, J$.M(4632497, J$.G(4632321, J$.R(4632313, 'this', this, false, false), '_continuations'), 'reduce', false)(J$.T(4632481, function (acc, cur) {
                                    jalangiLabel9:
                                        while (true) {
                                            try {
                                                J$.Fe(4632441, arguments.callee, this, arguments);
                                                arguments = J$.N(4632449, 'arguments', arguments, true, false, false);
                                                acc = J$.N(4632457, 'acc', acc, true, false, false);
                                                cur = J$.N(4632465, 'cur', cur, true, false, false);
                                                J$.N(4632473, 'node', node, false, false, false);
                                                if (J$.C(247992, J$.G(4632337, J$.R(4632329, 'cur', cur, false, false), 'promise'))) {
                                                    var node = J$.W(4632393, 'node', J$.T(4632385, {
                                                        promise: J$.G(4632353, J$.R(4632345, 'cur', cur, false, false), 'promise'),
                                                        children: J$.M(4632377, J$.G(4632369, J$.R(4632361, 'cur', cur, false, false), 'promise'), '_findAncestry', false)()
                                                    }, 11, false), node, false, false);
                                                    J$.M(4632417, J$.R(4632401, 'acc', acc, false, false), 'push', false)(J$.R(4632409, 'node', node, false, false));
                                                }
                                                return J$.Rt(4632433, J$.R(4632425, 'acc', acc, false, false));
                                            } catch (J$e) {
                                                J$.Ex(4637473, J$e);
                                            } finally {
                                                if (J$.Fr(4637481))
                                                    continue jalangiLabel9;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), J$.T(4632489, [], 10, false)));
                            } catch (J$e) {
                                J$.Ex(4637489, J$e);
                            } finally {
                                if (J$.Fr(4637497))
                                    continue jalangiLabel10;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _setParent: J$.T(4632641, function _setParent(parent) {
                    jalangiLabel11:
                        while (true) {
                            try {
                                J$.Fe(4632617, arguments.callee, this, arguments);
                                arguments = J$.N(4632625, 'arguments', arguments, true, false, false);
                                parent = J$.N(4632633, 'parent', parent, true, false, false);
                                if (J$.C(248000, J$.G(4632545, J$.R(4632537, 'this', this, false, false), '_parent'))) {
                                    throw J$.F(4632569, J$.I(typeof Error === 'undefined' ? Error = J$.R(4632553, 'Error', undefined, true, true) : Error = J$.R(4632553, 'Error', Error, true, true)), true)(J$.T(4632561, 'parent already set', 21, false));
                                }
                                J$.P(4632593, J$.R(4632577, 'this', this, false, false), '_parent', J$.R(4632585, 'parent', parent, false, false));
                                return J$.Rt(4632609, J$.R(4632601, 'this', this, false, false));
                            } catch (J$e) {
                                J$.Ex(4637505, J$e);
                            } finally {
                                if (J$.Fr(4637513))
                                    continue jalangiLabel11;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _continueWith: J$.T(4632753, function _continueWith(data) {
                    jalangiLabel12:
                        while (true) {
                            try {
                                J$.Fe(4632721, arguments.callee, this, arguments);
                                arguments = J$.N(4632729, 'arguments', arguments, true, false, false);
                                data = J$.N(4632737, 'data', data, true, false, false);
                                J$.N(4632745, 'firstPending', firstPending, false, false, false);
                                var firstPending = J$.W(4632665, 'firstPending', J$.M(4632657, J$.R(4632649, 'this', this, false, false), '_findFirstPending', false)(), firstPending, false, false);
                                if (J$.C(248008, J$.R(4632673, 'firstPending', firstPending, false, false))) {
                                    J$.P(4632697, J$.R(4632681, 'firstPending', firstPending, false, false), '_data', J$.R(4632689, 'data', data, false, false));
                                    J$.M(4632713, J$.R(4632705, 'firstPending', firstPending, false, false), '_setResolved', false)();
                                }
                            } catch (J$e) {
                                J$.Ex(4637521, J$e);
                            } finally {
                                if (J$.Fr(4637529))
                                    continue jalangiLabel12;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _findFirstPending: J$.T(4632873, function _findFirstPending() {
                    jalangiLabel14:
                        while (true) {
                            try {
                                J$.Fe(4632857, arguments.callee, this, arguments);
                                arguments = J$.N(4632865, 'arguments', arguments, true, false, false);
                                return J$.Rt(4632849, J$.M(4632841, J$.R(4632761, 'this', this, false, false), '_findFirstAncestor', false)(J$.T(4632833, function (test) {
                                    jalangiLabel13:
                                        while (true) {
                                            try {
                                                J$.Fe(4632809, arguments.callee, this, arguments);
                                                arguments = J$.N(4632817, 'arguments', arguments, true, false, false);
                                                test = J$.N(4632825, 'test', test, true, false, false);
                                                return J$.Rt(4632801, J$.C(248016, J$.G(4632777, J$.R(4632769, 'test', test, false, false), '_isPending')) ? J$.M(4632793, J$.R(4632785, 'test', test, false, false), '_isPending', false)() : J$._());
                                            } catch (J$e) {
                                                J$.Ex(4637537, J$e);
                                            } finally {
                                                if (J$.Fr(4637545))
                                                    continue jalangiLabel13;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)));
                            } catch (J$e) {
                                J$.Ex(4637553, J$e);
                            } finally {
                                if (J$.Fr(4637561))
                                    continue jalangiLabel14;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _findFirstPaused: J$.T(4632977, function _findFirstPaused() {
                    jalangiLabel16:
                        while (true) {
                            try {
                                J$.Fe(4632961, arguments.callee, this, arguments);
                                arguments = J$.N(4632969, 'arguments', arguments, true, false, false);
                                return J$.Rt(4632953, J$.M(4632945, J$.R(4632881, 'this', this, false, false), '_findFirstAncestor', false)(J$.T(4632937, function (test) {
                                    jalangiLabel15:
                                        while (true) {
                                            try {
                                                J$.Fe(4632913, arguments.callee, this, arguments);
                                                arguments = J$.N(4632921, 'arguments', arguments, true, false, false);
                                                test = J$.N(4632929, 'test', test, true, false, false);
                                                return J$.Rt(4632905, J$.G(4632897, J$.R(4632889, 'test', test, false, false), '_paused'));
                                            } catch (J$e) {
                                                J$.Ex(4637569, J$e);
                                            } finally {
                                                if (J$.Fr(4637577))
                                                    continue jalangiLabel15;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)));
                            } catch (J$e) {
                                J$.Ex(4637585, J$e);
                            } finally {
                                if (J$.Fr(4637593))
                                    continue jalangiLabel16;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _findFirstAncestor: J$.T(4633129, function _findFirstAncestor(matching) {
                    jalangiLabel17:
                        while (true) {
                            try {
                                J$.Fe(4633089, arguments.callee, this, arguments);
                                arguments = J$.N(4633097, 'arguments', arguments, true, false, false);
                                matching = J$.N(4633105, 'matching', matching, true, false, false);
                                J$.N(4633113, 'test', test, false, false, false);
                                J$.N(4633121, 'result', result, false, false, false);
                                var test = J$.W(4632993, 'test', J$.R(4632985, 'this', this, false, false), test, false, false);
                                var result;
                                while (J$.C(248032, J$.R(4633001, 'test', test, false, false))) {
                                    if (J$.C(248024, J$.F(4633025, J$.R(4633009, 'matching', matching, false, false), false)(J$.R(4633017, 'test', test, false, false)))) {
                                        result = J$.W(4633041, 'result', J$.R(4633033, 'test', test, false, false), result, false, false);
                                    }
                                    test = J$.W(4633065, 'test', J$.G(4633057, J$.R(4633049, 'test', test, false, false), '_parent'), test, false, false);
                                }
                                return J$.Rt(4633081, J$.R(4633073, 'result', result, false, false));
                            } catch (J$e) {
                                J$.Ex(4637601, J$e);
                            } finally {
                                if (J$.Fr(4637609))
                                    continue jalangiLabel17;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _failWith: J$.T(4633241, function _failWith(error) {
                    jalangiLabel18:
                        while (true) {
                            try {
                                J$.Fe(4633209, arguments.callee, this, arguments);
                                arguments = J$.N(4633217, 'arguments', arguments, true, false, false);
                                error = J$.N(4633225, 'error', error, true, false, false);
                                J$.N(4633233, 'firstRejected', firstRejected, false, false, false);
                                var firstRejected = J$.W(4633153, 'firstRejected', J$.M(4633145, J$.R(4633137, 'this', this, false, false), '_findFirstPending', false)(), firstRejected, false, false);
                                if (J$.C(248040, J$.R(4633161, 'firstRejected', firstRejected, false, false))) {
                                    J$.P(4633185, J$.R(4633169, 'firstRejected', firstRejected, false, false), '_error', J$.R(4633177, 'error', error, false, false));
                                    J$.M(4633201, J$.R(4633193, 'firstRejected', firstRejected, false, false), '_setRejected', false)();
                                }
                            } catch (J$e) {
                                J$.Ex(4637617, J$e);
                            } finally {
                                if (J$.Fr(4637625))
                                    continue jalangiLabel18;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _takeContinuations: J$.T(4633329, function _takeContinuations() {
                    jalangiLabel19:
                        while (true) {
                            try {
                                J$.Fe(4633313, arguments.callee, this, arguments);
                                arguments = J$.N(4633321, 'arguments', arguments, true, false, false);
                                return J$.Rt(4633305, J$.M(4633297, J$.G(4633257, J$.R(4633249, 'this', this, false, false), '_continuations'), 'splice', false)(J$.T(4633265, 0, 22, false), J$.G(4633289, J$.G(4633281, J$.R(4633273, 'this', this, false, false), '_continuations'), 'length')));
                            } catch (J$e) {
                                J$.Ex(4637633, J$e);
                            } finally {
                                if (J$.Fr(4637641))
                                    continue jalangiLabel19;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _runRejections: J$.T(4633729, function _runRejections() {
                    jalangiLabel21:
                        while (true) {
                            try {
                                J$.Fe(4633689, arguments.callee, this, arguments);
                                arguments = J$.N(4633697, 'arguments', arguments, true, false, false);
                                J$.N(4633705, 'error', error, false, false, false);
                                J$.N(4633713, 'continuations', continuations, false, false, false);
                                J$.N(4633721, 'self', self, false, false, false);
                                if (J$.C(248056, J$.C(248048, J$.G(4633345, J$.R(4633337, 'this', this, false, false), '_paused')) ? J$._() : J$.U(478730, '!', J$.M(4633361, J$.R(4633353, 'this', this, false, false), '_isRejected', false)()))) {
                                    return J$.Rt(4633369, undefined);
                                }
                                var error = J$.W(4633417, 'error', J$.G(4633385, J$.R(4633377, 'this', this, false, false), '_error'), error, false, false), continuations = J$.W(4633425, 'continuations', J$.M(4633401, J$.R(4633393, 'this', this, false, false), '_takeContinuations', false)(), continuations, false, false), self = J$.W(4633433, 'self', J$.R(4633409, 'this', this, false, false), self, false, false);
                                J$.M(4633681, J$.R(4633441, 'continuations', continuations, false, false), 'forEach', false)(J$.T(4633673, function (cont) {
                                    jalangiLabel20:
                                        while (true) {
                                            try {
                                                J$.Fe(4633633, arguments.callee, this, arguments);
                                                arguments = J$.N(4633641, 'arguments', arguments, true, false, false);
                                                cont = J$.N(4633649, 'cont', cont, true, false, false);
                                                J$.N(4633657, 'catchResult', catchResult, false, false, false);
                                                J$.N(4633665, 'message', message, false, false, false);
                                                if (J$.C(248064, J$.G(4633457, J$.R(4633449, 'cont', cont, false, false), 'catchFn'))) {
                                                    try {
                                                        var catchResult = J$.W(4633489, 'catchResult', J$.M(4633481, J$.R(4633465, 'cont', cont, false, false), 'catchFn', false)(J$.R(4633473, 'error', error, false, false)), catchResult, false, false);
                                                        J$.M(4633529, J$.R(4633497, 'self', self, false, false), '_handleUserFunctionResult', false)(J$.R(4633505, 'catchResult', catchResult, false, false), J$.G(4633521, J$.R(4633513, 'cont', cont, false, false), 'promise'));
                                                    } catch (e) {
                                                        J$.N(4633593, 'e', e, false, false, true);
                                                        var message = J$.W(4633553, 'message', J$.G(4633545, J$.R(4633537, 'e', e, false, false), 'message'), message, false, false);
                                                        J$.M(4633585, J$.G(4633569, J$.R(4633561, 'cont', cont, false, false), 'promise'), 'reject', false)(J$.R(4633577, 'e', e, false, false));
                                                    }
                                                } else {
                                                    J$.M(4633625, J$.G(4633609, J$.R(4633601, 'cont', cont, false, false), 'promise'), 'reject', false)(J$.R(4633617, 'error', error, false, false));
                                                }
                                            } catch (J$e) {
                                                J$.Ex(4637649, J$e);
                                            } finally {
                                                if (J$.Fr(4637657))
                                                    continue jalangiLabel20;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            } catch (J$e) {
                                J$.Ex(4637665, J$e);
                            } finally {
                                if (J$.Fr(4637673))
                                    continue jalangiLabel21;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _runResolutions: J$.T(4634201, function _runResolutions() {
                    jalangiLabel23:
                        while (true) {
                            try {
                                J$.Fe(4634161, arguments.callee, this, arguments);
                                arguments = J$.N(4634169, 'arguments', arguments, true, false, false);
                                J$.N(4634177, 'continuations', continuations, false, false, false);
                                J$.N(4634185, 'data', data, false, false, false);
                                J$.N(4634193, 'self', self, false, false, false);
                                if (J$.C(248088, J$.C(248080, J$.C(248072, J$.G(4633745, J$.R(4633737, 'this', this, false, false), '_paused')) ? J$._() : J$.U(478738, '!', J$.M(4633761, J$.R(4633753, 'this', this, false, false), '_isResolved', false)())) ? J$._() : J$.M(4633777, J$.R(4633769, 'this', this, false, false), '_isPending', false)())) {
                                    return J$.Rt(4633785, undefined);
                                }
                                var continuations = J$.W(4633809, 'continuations', J$.M(4633801, J$.R(4633793, 'this', this, false, false), '_takeContinuations', false)(), continuations, false, false);
                                if (J$.C(248096, J$.F(4633841, J$.R(4633817, 'looksLikeAPromise', looksLikeAPromise, false, true), false)(J$.G(4633833, J$.R(4633825, 'this', this, false, false), '_data')))) {
                                    return J$.Rt(4633881, J$.M(4633873, J$.R(4633849, 'this', this, false, false), '_handleWhenResolvedDataIsPromise', false)(J$.G(4633865, J$.R(4633857, 'this', this, false, false), '_data')));
                                }
                                var data = J$.W(4633905, 'data', J$.G(4633897, J$.R(4633889, 'this', this, false, false), '_data'), data, false, false);
                                var self = J$.W(4633921, 'self', J$.R(4633913, 'this', this, false, false), self, false, false);
                                J$.M(4634153, J$.R(4633929, 'continuations', continuations, false, false), 'forEach', false)(J$.T(4634145, function (cont) {
                                    jalangiLabel22:
                                        while (true) {
                                            try {
                                                J$.Fe(4634113, arguments.callee, this, arguments);
                                                arguments = J$.N(4634121, 'arguments', arguments, true, false, false);
                                                cont = J$.N(4634129, 'cont', cont, true, false, false);
                                                J$.N(4634137, 'result', result, false, false, false);
                                                if (J$.C(248112, J$.G(4633945, J$.R(4633937, 'cont', cont, false, false), 'nextFn'))) {
                                                    try {
                                                        var result = J$.W(4633977, 'result', J$.M(4633969, J$.R(4633953, 'cont', cont, false, false), 'nextFn', false)(J$.R(4633961, 'data', data, false, false)), result, false, false);
                                                        J$.M(4634017, J$.R(4633985, 'self', self, false, false), '_handleUserFunctionResult', false)(J$.R(4633993, 'result', result, false, false), J$.G(4634009, J$.R(4634001, 'cont', cont, false, false), 'promise'));
                                                    } catch (e) {
                                                        J$.N(4634057, 'e', e, false, false, true);
                                                        J$.M(4634049, J$.R(4634025, 'self', self, false, false), '_handleResolutionError', false)(J$.R(4634033, 'e', e, false, false), J$.R(4634041, 'cont', cont, false, false));
                                                    }
                                                } else if (J$.C(248104, J$.G(4634073, J$.R(4634065, 'cont', cont, false, false), 'promise'))) {
                                                    J$.M(4634105, J$.G(4634089, J$.R(4634081, 'cont', cont, false, false), 'promise'), 'resolve', false)(J$.R(4634097, 'data', data, false, false));
                                                }
                                            } catch (J$e) {
                                                J$.Ex(4637681, J$e);
                                            } finally {
                                                if (J$.Fr(4637689))
                                                    continue jalangiLabel22;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            } catch (J$e) {
                                J$.Ex(4637697, J$e);
                            } finally {
                                if (J$.Fr(4637705))
                                    continue jalangiLabel23;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _handleResolutionError: J$.T(4634377, function _handleResolutionError(e, continuation) {
                    jalangiLabel24:
                        while (true) {
                            try {
                                J$.Fe(4634345, arguments.callee, this, arguments);
                                arguments = J$.N(4634353, 'arguments', arguments, true, false, false);
                                e = J$.N(4634361, 'e', e, true, false, false);
                                continuation = J$.N(4634369, 'continuation', continuation, true, false, false);
                                J$.M(4634217, J$.R(4634209, 'this', this, false, false), '_setRejected', false)();
                                if (J$.C(248120, J$.G(4634233, J$.R(4634225, 'continuation', continuation, false, false), 'catchFn'))) {
                                    try {
                                        J$.M(4634257, J$.R(4634241, 'continuation', continuation, false, false), 'catchFn', false)(J$.R(4634249, 'e', e, false, false));
                                        return J$.Rt(4634265, undefined);
                                    } catch (e2) {
                                        J$.N(4634289, 'e2', e2, false, false, true);
                                        e = J$.W(4634281, 'e', J$.R(4634273, 'e2', e2, false, false), e, false, false);
                                    }
                                }
                                if (J$.C(248128, J$.G(4634305, J$.R(4634297, 'continuation', continuation, false, false), 'promise'))) {
                                    J$.M(4634337, J$.G(4634321, J$.R(4634313, 'continuation', continuation, false, false), 'promise'), 'reject', false)(J$.R(4634329, 'e', e, false, false));
                                }
                            } catch (J$e) {
                                J$.Ex(4637713, J$e);
                            } finally {
                                if (J$.Fr(4637721))
                                    continue jalangiLabel24;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _handleWhenResolvedDataIsPromise: J$.T(4634633, function _handleWhenResolvedDataIsPromise(data) {
                    jalangiLabel27:
                        while (true) {
                            try {
                                J$.Fe(4634601, arguments.callee, this, arguments);
                                arguments = J$.N(4634609, 'arguments', arguments, true, false, false);
                                data = J$.N(4634617, 'data', data, true, false, false);
                                J$.N(4634625, 'self', self, false, false, false);
                                var self = J$.W(4634393, 'self', J$.R(4634385, 'this', this, false, false), self, false, false);
                                return J$.Rt(4634593, J$.M(4634585, J$.M(4634481, J$.R(4634401, 'data', data, false, false), 'then', false)(J$.T(4634473, function (result) {
                                    jalangiLabel25:
                                        while (true) {
                                            try {
                                                J$.Fe(4634449, arguments.callee, this, arguments);
                                                arguments = J$.N(4634457, 'arguments', arguments, true, false, false);
                                                result = J$.N(4634465, 'result', result, true, false, false);
                                                J$.P(4634425, J$.R(4634409, 'self', self, false, false), '_data', J$.R(4634417, 'result', result, false, false));
                                                J$.M(4634441, J$.R(4634433, 'self', self, false, false), '_runResolutions', false)();
                                            } catch (J$e) {
                                                J$.Ex(4637729, J$e);
                                            } finally {
                                                if (J$.Fr(4637737))
                                                    continue jalangiLabel25;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)), J$.T(4634489, 'catch', 21, false), false)(J$.T(4634577, function (error) {
                                    jalangiLabel26:
                                        while (true) {
                                            try {
                                                J$.Fe(4634553, arguments.callee, this, arguments);
                                                arguments = J$.N(4634561, 'arguments', arguments, true, false, false);
                                                error = J$.N(4634569, 'error', error, true, false, false);
                                                J$.P(4634513, J$.R(4634497, 'self', self, false, false), '_error', J$.R(4634505, 'error', error, false, false));
                                                J$.M(4634529, J$.R(4634521, 'self', self, false, false), '_setRejected', false)();
                                                J$.M(4634545, J$.R(4634537, 'self', self, false, false), '_runRejections', false)();
                                            } catch (J$e) {
                                                J$.Ex(4637745, J$e);
                                            } finally {
                                                if (J$.Fr(4637753))
                                                    continue jalangiLabel26;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)));
                            } catch (J$e) {
                                J$.Ex(4637761, J$e);
                            } finally {
                                if (J$.Fr(4637769))
                                    continue jalangiLabel27;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _handleUserFunctionResult: J$.T(4634753, function _handleUserFunctionResult(data, nextSynchronousPromise) {
                    jalangiLabel28:
                        while (true) {
                            try {
                                J$.Fe(4634721, arguments.callee, this, arguments);
                                arguments = J$.N(4634729, 'arguments', arguments, true, false, false);
                                data = J$.N(4634737, 'data', data, true, false, false);
                                nextSynchronousPromise = J$.N(4634745, 'nextSynchronousPromise', nextSynchronousPromise, true, false, false);
                                if (J$.C(248136, J$.F(4634657, J$.R(4634641, 'looksLikeAPromise', looksLikeAPromise, false, true), false)(J$.R(4634649, 'data', data, false, false)))) {
                                    J$.M(4634689, J$.R(4634665, 'this', this, false, false), '_chainPromiseData', false)(J$.R(4634673, 'data', data, false, false), J$.R(4634681, 'nextSynchronousPromise', nextSynchronousPromise, false, false));
                                } else {
                                    J$.M(4634713, J$.R(4634697, 'nextSynchronousPromise', nextSynchronousPromise, false, false), 'resolve', false)(J$.R(4634705, 'data', data, false, false));
                                }
                            } catch (J$e) {
                                J$.Ex(4637777, J$e);
                            } finally {
                                if (J$.Fr(4637785))
                                    continue jalangiLabel28;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _chainPromiseData: J$.T(4634937, function _chainPromiseData(promiseData, nextSynchronousPromise) {
                    jalangiLabel31:
                        while (true) {
                            try {
                                J$.Fe(4634905, arguments.callee, this, arguments);
                                arguments = J$.N(4634913, 'arguments', arguments, true, false, false);
                                promiseData = J$.N(4634921, 'promiseData', promiseData, true, false, false);
                                nextSynchronousPromise = J$.N(4634929, 'nextSynchronousPromise', nextSynchronousPromise, true, false, false);
                                J$.M(4634897, J$.M(4634825, J$.R(4634761, 'promiseData', promiseData, false, false), 'then', false)(J$.T(4634817, function (newData) {
                                    jalangiLabel29:
                                        while (true) {
                                            try {
                                                J$.Fe(4634793, arguments.callee, this, arguments);
                                                arguments = J$.N(4634801, 'arguments', arguments, true, false, false);
                                                newData = J$.N(4634809, 'newData', newData, true, false, false);
                                                J$.M(4634785, J$.R(4634769, 'nextSynchronousPromise', nextSynchronousPromise, false, false), 'resolve', false)(J$.R(4634777, 'newData', newData, false, false));
                                            } catch (J$e) {
                                                J$.Ex(4637793, J$e);
                                            } finally {
                                                if (J$.Fr(4637801))
                                                    continue jalangiLabel29;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)), J$.T(4634833, 'catch', 21, false), false)(J$.T(4634889, function (newError) {
                                    jalangiLabel30:
                                        while (true) {
                                            try {
                                                J$.Fe(4634865, arguments.callee, this, arguments);
                                                arguments = J$.N(4634873, 'arguments', arguments, true, false, false);
                                                newError = J$.N(4634881, 'newError', newError, true, false, false);
                                                J$.M(4634857, J$.R(4634841, 'nextSynchronousPromise', nextSynchronousPromise, false, false), 'reject', false)(J$.R(4634849, 'newError', newError, false, false));
                                            } catch (J$e) {
                                                J$.Ex(4637809, J$e);
                                            } finally {
                                                if (J$.Fr(4637817))
                                                    continue jalangiLabel30;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            } catch (J$e) {
                                J$.Ex(4637825, J$e);
                            } finally {
                                if (J$.Fr(4637833))
                                    continue jalangiLabel31;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _setResolved: J$.T(4635017, function _setResolved() {
                    jalangiLabel32:
                        while (true) {
                            try {
                                J$.Fe(4635001, arguments.callee, this, arguments);
                                arguments = J$.N(4635009, 'arguments', arguments, true, false, false);
                                J$.P(4634961, J$.R(4634945, 'this', this, false, false), 'status', J$.R(4634953, 'RESOLVED', RESOLVED, false, true));
                                if (J$.C(248144, J$.U(478746, '!', J$.G(4634977, J$.R(4634969, 'this', this, false, false), '_paused')))) {
                                    J$.M(4634993, J$.R(4634985, 'this', this, false, false), '_runResolutions', false)();
                                }
                            } catch (J$e) {
                                J$.Ex(4637841, J$e);
                            } finally {
                                if (J$.Fr(4637849))
                                    continue jalangiLabel32;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _setRejected: J$.T(4635097, function _setRejected() {
                    jalangiLabel33:
                        while (true) {
                            try {
                                J$.Fe(4635081, arguments.callee, this, arguments);
                                arguments = J$.N(4635089, 'arguments', arguments, true, false, false);
                                J$.P(4635041, J$.R(4635025, 'this', this, false, false), 'status', J$.R(4635033, 'REJECTED', REJECTED, false, true));
                                if (J$.C(248152, J$.U(478754, '!', J$.G(4635057, J$.R(4635049, 'this', this, false, false), '_paused')))) {
                                    J$.M(4635073, J$.R(4635065, 'this', this, false, false), '_runRejections', false)();
                                }
                            } catch (J$e) {
                                J$.Ex(4637857, J$e);
                            } finally {
                                if (J$.Fr(4637865))
                                    continue jalangiLabel33;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _isPending: J$.T(4635153, function _isPending() {
                    jalangiLabel34:
                        while (true) {
                            try {
                                J$.Fe(4635137, arguments.callee, this, arguments);
                                arguments = J$.N(4635145, 'arguments', arguments, true, false, false);
                                return J$.Rt(4635129, J$.B(478762, '===', J$.G(4635113, J$.R(4635105, 'this', this, false, false), 'status'), J$.R(4635121, 'PENDING', PENDING, false, true)));
                            } catch (J$e) {
                                J$.Ex(4637873, J$e);
                            } finally {
                                if (J$.Fr(4637881))
                                    continue jalangiLabel34;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _isResolved: J$.T(4635209, function _isResolved() {
                    jalangiLabel35:
                        while (true) {
                            try {
                                J$.Fe(4635193, arguments.callee, this, arguments);
                                arguments = J$.N(4635201, 'arguments', arguments, true, false, false);
                                return J$.Rt(4635185, J$.B(478770, '===', J$.G(4635169, J$.R(4635161, 'this', this, false, false), 'status'), J$.R(4635177, 'RESOLVED', RESOLVED, false, true)));
                            } catch (J$e) {
                                J$.Ex(4637889, J$e);
                            } finally {
                                if (J$.Fr(4637897))
                                    continue jalangiLabel35;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                _isRejected: J$.T(4635265, function _isRejected() {
                    jalangiLabel36:
                        while (true) {
                            try {
                                J$.Fe(4635249, arguments.callee, this, arguments);
                                arguments = J$.N(4635257, 'arguments', arguments, true, false, false);
                                return J$.Rt(4635241, J$.B(478778, '===', J$.G(4635225, J$.R(4635217, 'this', this, false, false), 'status'), J$.R(4635233, 'REJECTED', REJECTED, false, true)));
                            } catch (J$e) {
                                J$.Ex(4637905, J$e);
                            } finally {
                                if (J$.Fr(4637913))
                                    continue jalangiLabel36;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.P(4635585, J$.R(4635289, 'SynchronousPromise', SynchronousPromise, false, true), 'resolve', J$.T(4635577, function (result) {
                jalangiLabel40:
                    while (true) {
                        try {
                            J$.Fe(4635553, arguments.callee, this, arguments);
                            arguments = J$.N(4635561, 'arguments', arguments, true, false, false);
                            result = J$.N(4635569, 'result', result, true, false, false);
                            return J$.Rt(4635545, J$.F(4635537, J$.R(4635297, 'SynchronousPromise', SynchronousPromise, false, true), true)(J$.T(4635529, function (resolve, reject) {
                                jalangiLabel39:
                                    while (true) {
                                        try {
                                            J$.Fe(4635497, arguments.callee, this, arguments);
                                            arguments = J$.N(4635505, 'arguments', arguments, true, false, false);
                                            resolve = J$.N(4635513, 'resolve', resolve, true, false, false);
                                            reject = J$.N(4635521, 'reject', reject, true, false, false);
                                            if (J$.C(248160, J$.F(4635321, J$.R(4635305, 'looksLikeAPromise', looksLikeAPromise, false, true), false)(J$.R(4635313, 'result', result, false, false)))) {
                                                J$.M(4635465, J$.M(4635393, J$.R(4635329, 'result', result, false, false), 'then', false)(J$.T(4635385, function (newResult) {
                                                    jalangiLabel37:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(4635361, arguments.callee, this, arguments);
                                                                arguments = J$.N(4635369, 'arguments', arguments, true, false, false);
                                                                newResult = J$.N(4635377, 'newResult', newResult, true, false, false);
                                                                J$.F(4635353, J$.R(4635337, 'resolve', resolve, false, false), false)(J$.R(4635345, 'newResult', newResult, false, false));
                                                            } catch (J$e) {
                                                                J$.Ex(4637921, J$e);
                                                            } finally {
                                                                if (J$.Fr(4637929))
                                                                    continue jalangiLabel37;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false)), J$.T(4635401, 'catch', 21, false), false)(J$.T(4635457, function (error) {
                                                    jalangiLabel38:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(4635433, arguments.callee, this, arguments);
                                                                arguments = J$.N(4635441, 'arguments', arguments, true, false, false);
                                                                error = J$.N(4635449, 'error', error, true, false, false);
                                                                J$.F(4635425, J$.R(4635409, 'reject', reject, false, false), false)(J$.R(4635417, 'error', error, false, false));
                                                            } catch (J$e) {
                                                                J$.Ex(4637937, J$e);
                                                            } finally {
                                                                if (J$.Fr(4637945))
                                                                    continue jalangiLabel38;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false));
                                            } else {
                                                J$.F(4635489, J$.R(4635473, 'resolve', resolve, false, false), false)(J$.R(4635481, 'result', result, false, false));
                                            }
                                        } catch (J$e) {
                                            J$.Ex(4637953, J$e);
                                        } finally {
                                            if (J$.Fr(4637961))
                                                continue jalangiLabel39;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(4637969, J$e);
                        } finally {
                            if (J$.Fr(4637977))
                                continue jalangiLabel40;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4635721, J$.R(4635593, 'SynchronousPromise', SynchronousPromise, false, true), 'reject', J$.T(4635713, function (result) {
                jalangiLabel42:
                    while (true) {
                        try {
                            J$.Fe(4635689, arguments.callee, this, arguments);
                            arguments = J$.N(4635697, 'arguments', arguments, true, false, false);
                            result = J$.N(4635705, 'result', result, true, false, false);
                            return J$.Rt(4635681, J$.F(4635673, J$.R(4635601, 'SynchronousPromise', SynchronousPromise, false, true), true)(J$.T(4635665, function (resolve, reject) {
                                jalangiLabel41:
                                    while (true) {
                                        try {
                                            J$.Fe(4635633, arguments.callee, this, arguments);
                                            arguments = J$.N(4635641, 'arguments', arguments, true, false, false);
                                            resolve = J$.N(4635649, 'resolve', resolve, true, false, false);
                                            reject = J$.N(4635657, 'reject', reject, true, false, false);
                                            J$.F(4635625, J$.R(4635609, 'reject', reject, false, false), false)(J$.R(4635617, 'result', result, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4637985, J$e);
                                        } finally {
                                            if (J$.Fr(4637993))
                                                continue jalangiLabel41;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(4638001, J$e);
                        } finally {
                            if (J$.Fr(4638009))
                                continue jalangiLabel42;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4635873, J$.R(4635729, 'SynchronousPromise', SynchronousPromise, false, true), 'unresolved', J$.T(4635865, function () {
                jalangiLabel44:
                    while (true) {
                        try {
                            J$.Fe(4635849, arguments.callee, this, arguments);
                            arguments = J$.N(4635857, 'arguments', arguments, true, false, false);
                            return J$.Rt(4635841, J$.F(4635833, J$.R(4635737, 'SynchronousPromise', SynchronousPromise, false, true), true)(J$.T(4635825, function (resolve, reject) {
                                jalangiLabel43:
                                    while (true) {
                                        try {
                                            J$.Fe(4635793, arguments.callee, this, arguments);
                                            arguments = J$.N(4635801, 'arguments', arguments, true, false, false);
                                            resolve = J$.N(4635809, 'resolve', resolve, true, false, false);
                                            reject = J$.N(4635817, 'reject', reject, true, false, false);
                                            J$.P(4635761, J$.R(4635745, 'this', this, false, false), 'resolve', J$.R(4635753, 'resolve', resolve, false, false));
                                            J$.P(4635785, J$.R(4635769, 'this', this, false, false), 'reject', J$.R(4635777, 'reject', reject, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4638017, J$e);
                                        } finally {
                                            if (J$.Fr(4638025))
                                                continue jalangiLabel43;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(4638033, J$e);
                        } finally {
                            if (J$.Fr(4638041))
                                continue jalangiLabel44;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4636665, J$.R(4635881, 'SynchronousPromise', SynchronousPromise, false, true), 'all', J$.T(4636657, function () {
                jalangiLabel51:
                    while (true) {
                        try {
                            J$.Fe(4636633, arguments.callee, this, arguments);
                            arguments = J$.N(4636641, 'arguments', arguments, true, false, false);
                            J$.N(4636649, 'args', args, false, false, false);
                            var args = J$.W(4635913, 'args', J$.F(4635905, J$.R(4635889, 'makeArrayFrom', makeArrayFrom, false, true), false)(J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4635897, 'arguments', undefined, true, true) : arguments = J$.R(4635897, 'arguments', arguments, true, true))), args, false, false);
                            if (J$.C(248168, J$.M(4635953, J$.I(typeof Array === 'undefined' ? Array = J$.R(4635921, 'Array', undefined, true, true) : Array = J$.R(4635921, 'Array', Array, true, true)), 'isArray', false)(J$.G(4635945, J$.R(4635929, 'args', args, false, false), J$.T(4635937, 0, 22, false))))) {
                                args = J$.W(4635985, 'args', J$.G(4635977, J$.R(4635961, 'args', args, false, false), J$.T(4635969, 0, 22, false)), args, false, false);
                            }
                            if (J$.C(248176, J$.U(478786, '!', J$.G(4636001, J$.R(4635993, 'args', args, false, false), 'length')))) {
                                return J$.Rt(4636033, J$.M(4636025, J$.R(4636009, 'SynchronousPromise', SynchronousPromise, false, true), 'resolve', false)(J$.T(4636017, [], 10, false)));
                            }
                            return J$.Rt(4636625, J$.F(4636617, J$.R(4636041, 'SynchronousPromise', SynchronousPromise, false, true), true)(J$.T(4636609, function (resolve, reject) {
                                jalangiLabel50:
                                    while (true) {
                                        try {
                                            J$.Fe(4636537, arguments.callee, this, arguments);
                                            arguments = J$.N(4636545, 'arguments', arguments, true, false, false);
                                            resolve = J$.N(4636553, 'resolve', resolve, true, false, false);
                                            reject = J$.N(4636561, 'reject', reject, true, false, false);
                                            J$.N(4636569, 'allData', allData, false, false, false);
                                            J$.N(4636577, 'numResolved', numResolved, false, false, false);
                                            J$.N(4636585, 'doResolve', doResolve, false, false, false);
                                            J$.N(4636593, 'rejected', rejected, false, false, false);
                                            J$.N(4636601, 'doReject', doReject, false, false, false);
                                            var allData = J$.W(4636233, 'allData', J$.T(4636049, [], 10, false), allData, false, false), numResolved = J$.W(4636241, 'numResolved', J$.T(4636057, 0, 22, false), numResolved, false, false), doResolve = J$.W(4636249, 'doResolve', J$.T(4636129, function doResolve() {
                                                    jalangiLabel45:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(4636113, arguments.callee, this, arguments);
                                                                arguments = J$.N(4636121, 'arguments', arguments, true, false, false);
                                                                if (J$.C(248184, J$.B(478794, '===', J$.R(4636065, 'numResolved', numResolved, false, false), J$.G(4636081, J$.R(4636073, 'args', args, false, false), 'length')))) {
                                                                    J$.F(4636105, J$.R(4636089, 'resolve', resolve, false, false), false)(J$.R(4636097, 'allData', allData, false, false));
                                                                }
                                                            } catch (J$e) {
                                                                J$.Ex(4638049, J$e);
                                                            } finally {
                                                                if (J$.Fr(4638057))
                                                                    continue jalangiLabel45;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false), doResolve, false, false), rejected = J$.W(4636257, 'rejected', J$.T(4636137, false, 23, false), rejected, false, false), doReject = J$.W(4636265, 'doReject', J$.T(4636225, function doReject(err) {
                                                    jalangiLabel46:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(4636201, arguments.callee, this, arguments);
                                                                arguments = J$.N(4636209, 'arguments', arguments, true, false, false);
                                                                err = J$.N(4636217, 'err', err, true, false, false);
                                                                if (J$.C(248192, J$.R(4636145, 'rejected', rejected, false, false))) {
                                                                    return J$.Rt(4636153, undefined);
                                                                }
                                                                rejected = J$.W(4636169, 'rejected', J$.T(4636161, true, 23, false), rejected, false, false);
                                                                J$.F(4636193, J$.R(4636177, 'reject', reject, false, false), false)(J$.R(4636185, 'err', err, false, false));
                                                            } catch (J$e) {
                                                                J$.Ex(4638065, J$e);
                                                            } finally {
                                                                if (J$.Fr(4638073))
                                                                    continue jalangiLabel46;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false), doReject, false, false);
                                            J$.M(4636529, J$.R(4636273, 'args', args, false, false), 'forEach', false)(J$.T(4636521, function (arg, idx) {
                                                jalangiLabel49:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(4636489, arguments.callee, this, arguments);
                                                            arguments = J$.N(4636497, 'arguments', arguments, true, false, false);
                                                            arg = J$.N(4636505, 'arg', arg, true, false, false);
                                                            idx = J$.N(4636513, 'idx', idx, true, false, false);
                                                            J$.M(4636481, J$.M(4636409, J$.M(4636297, J$.R(4636281, 'SynchronousPromise', SynchronousPromise, false, true), 'resolve', false)(J$.R(4636289, 'arg', arg, false, false)), 'then', false)(J$.T(4636401, function (thisResult) {
                                                                jalangiLabel47:
                                                                    while (true) {
                                                                        try {
                                                                            J$.Fe(4636377, arguments.callee, this, arguments);
                                                                            arguments = J$.N(4636385, 'arguments', arguments, true, false, false);
                                                                            thisResult = J$.N(4636393, 'thisResult', thisResult, true, false, false);
                                                                            J$.P(4636329, J$.R(4636305, 'allData', allData, false, false), J$.R(4636313, 'idx', idx, false, false), J$.R(4636321, 'thisResult', thisResult, false, false));
                                                                            numResolved = J$.W(4636353, 'numResolved', J$.B(478802, '+', J$.R(4636345, 'numResolved', numResolved, false, false), J$.T(4636337, 1, 22, false)), numResolved, false, false);
                                                                            J$.F(4636369, J$.R(4636361, 'doResolve', doResolve, false, false), false)();
                                                                        } catch (J$e) {
                                                                            J$.Ex(4638081, J$e);
                                                                        } finally {
                                                                            if (J$.Fr(4638089))
                                                                                continue jalangiLabel47;
                                                                            else
                                                                                return J$.Ra();
                                                                        }
                                                                    }
                                                            }, 12, false)), J$.T(4636417, 'catch', 21, false), false)(J$.T(4636473, function (err) {
                                                                jalangiLabel48:
                                                                    while (true) {
                                                                        try {
                                                                            J$.Fe(4636449, arguments.callee, this, arguments);
                                                                            arguments = J$.N(4636457, 'arguments', arguments, true, false, false);
                                                                            err = J$.N(4636465, 'err', err, true, false, false);
                                                                            J$.F(4636441, J$.R(4636425, 'doReject', doReject, false, false), false)(J$.R(4636433, 'err', err, false, false));
                                                                        } catch (J$e) {
                                                                            J$.Ex(4638097, J$e);
                                                                        } finally {
                                                                            if (J$.Fr(4638105))
                                                                                continue jalangiLabel48;
                                                                            else
                                                                                return J$.Ra();
                                                                        }
                                                                    }
                                                            }, 12, false));
                                                        } catch (J$e) {
                                                            J$.Ex(4638113, J$e);
                                                        } finally {
                                                            if (J$.Fr(4638121))
                                                                continue jalangiLabel49;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(4638129, J$e);
                                        } finally {
                                            if (J$.Fr(4638137))
                                                continue jalangiLabel50;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(4638145, J$e);
                        } finally {
                            if (J$.Fr(4638153))
                                continue jalangiLabel51;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            if (J$.C(248200, J$.B(478810, '===', J$.I(typeof Promise === 'undefined' ? Promise = J$.R(4636673, 'Promise', undefined, true, true) : Promise = J$.R(4636673, 'Promise', Promise, true, true)), J$.R(4636681, 'SynchronousPromise', SynchronousPromise, false, true)))) {
                throw J$.F(4636705, J$.I(typeof Error === 'undefined' ? Error = J$.R(4636689, 'Error', undefined, true, true) : Error = J$.R(4636689, 'Error', Error, true, true)), true)(J$.T(4636697, 'Please use SynchronousPromise.installGlobally() to install globally', 21, false));
            }
            var RealPromise = J$.W(4636721, 'RealPromise', J$.I(typeof Promise === 'undefined' ? Promise = J$.R(4636713, 'Promise', undefined, true, true) : Promise = J$.R(4636713, 'Promise', Promise, true, true)), RealPromise, false, true);
            J$.P(4636873, J$.R(4636729, 'SynchronousPromise', SynchronousPromise, false, true), 'installGlobally', J$.T(4636865, function (__awaiter) {
                jalangiLabel52:
                    while (true) {
                        try {
                            J$.Fe(4636833, arguments.callee, this, arguments);
                            arguments = J$.N(4636841, 'arguments', arguments, true, false, false);
                            __awaiter = J$.N(4636849, '__awaiter', __awaiter, true, false, false);
                            J$.N(4636857, 'result', result, false, false, false);
                            if (J$.C(248208, J$.B(478818, '===', J$.I(typeof Promise === 'undefined' ? Promise = J$.R(4636737, 'Promise', undefined, true, true) : Promise = J$.R(4636737, 'Promise', Promise, true, true)), J$.R(4636745, 'SynchronousPromise', SynchronousPromise, false, true)))) {
                                return J$.Rt(4636761, J$.R(4636753, '__awaiter', __awaiter, false, false));
                            }
                            var result = J$.W(4636793, 'result', J$.F(4636785, J$.R(4636769, 'patchAwaiterIfRequired', patchAwaiterIfRequired, false, true), false)(J$.R(4636777, '__awaiter', __awaiter, false, false)), result, false, false);
                            Promise = J$.W(4636809, 'Promise', J$.R(4636801, 'SynchronousPromise', SynchronousPromise, false, true), J$.I(typeof Promise === 'undefined' ? undefined : Promise), true, true);
                            return J$.Rt(4636825, J$.R(4636817, 'result', result, false, false));
                        } catch (J$e) {
                            J$.Ex(4638161, J$e);
                        } finally {
                            if (J$.Fr(4638169))
                                continue jalangiLabel52;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4636945, J$.R(4636881, 'SynchronousPromise', SynchronousPromise, false, true), 'uninstallGlobally', J$.T(4636937, function () {
                jalangiLabel53:
                    while (true) {
                        try {
                            J$.Fe(4636921, arguments.callee, this, arguments);
                            arguments = J$.N(4636929, 'arguments', arguments, true, false, false);
                            if (J$.C(248216, J$.B(478826, '===', J$.I(typeof Promise === 'undefined' ? Promise = J$.R(4636889, 'Promise', undefined, true, true) : Promise = J$.R(4636889, 'Promise', Promise, true, true)), J$.R(4636897, 'SynchronousPromise', SynchronousPromise, false, true)))) {
                                Promise = J$.W(4636913, 'Promise', J$.R(4636905, 'RealPromise', RealPromise, false, true), J$.I(typeof Promise === 'undefined' ? undefined : Promise), true, true);
                            }
                        } catch (J$e) {
                            J$.Ex(4638177, J$e);
                        } finally {
                            if (J$.Fr(4638185))
                                continue jalangiLabel53;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4637217, J$.I(typeof module === 'undefined' ? module = J$.R(4637193, 'module', undefined, true, true) : module = J$.R(4637193, 'module', module, true, true)), 'exports', J$.T(4637209, { SynchronousPromise: J$.R(4637201, 'SynchronousPromise', SynchronousPromise, false, true) }, 11, false));
        } catch (J$e) {
            J$.Ex(4638225, J$e);
        } finally {
            if (J$.Sr(4638233))
                continue jalangiLabel56;
            else
                break jalangiLabel56;
        }
    }
// JALANGI DO NOT INSTRUMENT
