// colors/lib/custom/trap.js
J$.noInstrEval = false;
jalangiLabel2:
    while (true) {
        try {
            J$.Se(7064265, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/colors/lib/custom/trap.js');
            J$.P(7064257, J$.I(typeof module === 'undefined' ? module = J$.R(7062785, 'module', undefined, true, true) : module = J$.R(7062785, 'module', module, true, true)), J$.T(7062793, 'exports', 21, false), J$.T(7064249, function runTheTrap(text, options) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7064201, arguments.callee, this, arguments);
                            arguments = J$.N(7064209, 'arguments', arguments, true, false, false);
                            text = J$.N(7064217, 'text', text, true, false, false);
                            options = J$.N(7064225, 'options', options, true, false, false);
                            J$.N(7064233, 'result', result, false, false, false);
                            J$.N(7064241, 'trap', trap, false, false, false);
                            var result = J$.W(7062809, 'result', J$.T(7062801, '', 21, false), result, false, false);
                            text = J$.W(7062833, 'text', J$.C(378912, J$.R(7062817, 'text', text, false, false)) ? J$._() : J$.T(7062825, 'Run the trap, drop the bass', 21, false), text, false, false);
                            text = J$.W(7062865, 'text', J$.M(7062857, J$.R(7062841, 'text', text, false, false), 'split', false)(J$.T(7062849, '', 21, false)), text, false, false);
                            var trap = J$.W(7063873, 'trap', J$.T(7063865, {
                                a: J$.T(7062929, [
                                    J$.T(7062873, '@', 21, false),
                                    J$.T(7062881, 'Ą', 21, false),
                                    J$.T(7062889, 'Ⱥ', 21, false),
                                    J$.T(7062897, 'Ʌ', 21, false),
                                    J$.T(7062905, 'Δ', 21, false),
                                    J$.T(7062913, 'Λ', 21, false),
                                    J$.T(7062921, 'Д', 21, false)
                                ], 10, false),
                                b: J$.T(7062985, [
                                    J$.T(7062937, 'ß', 21, false),
                                    J$.T(7062945, 'Ɓ', 21, false),
                                    J$.T(7062953, 'Ƀ', 21, false),
                                    J$.T(7062961, 'ɮ', 21, false),
                                    J$.T(7062969, 'β', 21, false),
                                    J$.T(7062977, '\u0E3F', 21, false)
                                ], 10, false),
                                c: J$.T(7063017, [
                                    J$.T(7062993, '\xA9', 21, false),
                                    J$.T(7063001, 'Ȼ', 21, false),
                                    J$.T(7063009, 'Ͼ', 21, false)
                                ], 10, false),
                                d: J$.T(7063073, [
                                    J$.T(7063025, 'Ð', 21, false),
                                    J$.T(7063033, 'Ɗ', 21, false),
                                    J$.T(7063041, 'Ԁ', 21, false),
                                    J$.T(7063049, 'ԁ', 21, false),
                                    J$.T(7063057, 'Ԃ', 21, false),
                                    J$.T(7063065, 'ԃ', 21, false)
                                ], 10, false),
                                e: J$.T(7063145, [
                                    J$.T(7063081, 'Ë', 21, false),
                                    J$.T(7063089, 'ĕ', 21, false),
                                    J$.T(7063097, 'Ǝ', 21, false),
                                    J$.T(7063105, 'ɘ', 21, false),
                                    J$.T(7063113, 'Σ', 21, false),
                                    J$.T(7063121, 'ξ', 21, false),
                                    J$.T(7063129, 'Ҽ', 21, false),
                                    J$.T(7063137, '੬', 21, false)
                                ], 10, false),
                                f: J$.T(7063161, [J$.T(7063153, 'Ӻ', 21, false)], 10, false),
                                g: J$.T(7063177, [J$.T(7063169, 'ɢ', 21, false)], 10, false),
                                h: J$.T(7063233, [
                                    J$.T(7063185, 'Ħ', 21, false),
                                    J$.T(7063193, 'ƕ', 21, false),
                                    J$.T(7063201, 'Ң', 21, false),
                                    J$.T(7063209, 'Һ', 21, false),
                                    J$.T(7063217, 'Ӈ', 21, false),
                                    J$.T(7063225, 'Ԋ', 21, false)
                                ], 10, false),
                                i: J$.T(7063249, [J$.T(7063241, '\u0F0F', 21, false)], 10, false),
                                j: J$.T(7063265, [J$.T(7063257, 'Ĵ', 21, false)], 10, false),
                                k: J$.T(7063305, [
                                    J$.T(7063273, 'ĸ', 21, false),
                                    J$.T(7063281, 'Ҡ', 21, false),
                                    J$.T(7063289, 'Ӄ', 21, false),
                                    J$.T(7063297, 'Ԟ', 21, false)
                                ], 10, false),
                                l: J$.T(7063321, [J$.T(7063313, 'Ĺ', 21, false)], 10, false),
                                m: J$.T(7063377, [
                                    J$.T(7063329, 'ʍ', 21, false),
                                    J$.T(7063337, 'Ӎ', 21, false),
                                    J$.T(7063345, 'ӎ', 21, false),
                                    J$.T(7063353, 'Ԡ', 21, false),
                                    J$.T(7063361, 'ԡ', 21, false),
                                    J$.T(7063369, '൩', 21, false)
                                ], 10, false),
                                n: J$.T(7063433, [
                                    J$.T(7063385, 'Ñ', 21, false),
                                    J$.T(7063393, 'ŋ', 21, false),
                                    J$.T(7063401, 'Ɲ', 21, false),
                                    J$.T(7063409, 'Ͷ', 21, false),
                                    J$.T(7063417, 'Π', 21, false),
                                    J$.T(7063425, 'Ҋ', 21, false)
                                ], 10, false),
                                o: J$.T(7063513, [
                                    J$.T(7063441, 'Ø', 21, false),
                                    J$.T(7063449, 'õ', 21, false),
                                    J$.T(7063457, 'ø', 21, false),
                                    J$.T(7063465, 'Ǿ', 21, false),
                                    J$.T(7063473, 'ʘ', 21, false),
                                    J$.T(7063481, 'Ѻ', 21, false),
                                    J$.T(7063489, 'ם', 21, false),
                                    J$.T(7063497, '\u06DD', 21, false),
                                    J$.T(7063505, '\u0E4F', 21, false)
                                ], 10, false),
                                p: J$.T(7063537, [
                                    J$.T(7063521, 'Ƿ', 21, false),
                                    J$.T(7063529, 'Ҏ', 21, false)
                                ], 10, false),
                                q: J$.T(7063553, [J$.T(7063545, '্', 21, false)], 10, false),
                                r: J$.T(7063609, [
                                    J$.T(7063561, '\xAE', 21, false),
                                    J$.T(7063569, 'Ʀ', 21, false),
                                    J$.T(7063577, 'Ȑ', 21, false),
                                    J$.T(7063585, 'Ɍ', 21, false),
                                    J$.T(7063593, 'ʀ', 21, false),
                                    J$.T(7063601, 'Я', 21, false)
                                ], 10, false),
                                s: J$.T(7063649, [
                                    J$.T(7063617, '\xA7', 21, false),
                                    J$.T(7063625, 'Ϟ', 21, false),
                                    J$.T(7063633, 'ϟ', 21, false),
                                    J$.T(7063641, 'Ϩ', 21, false)
                                ], 10, false),
                                t: J$.T(7063681, [
                                    J$.T(7063657, 'Ł', 21, false),
                                    J$.T(7063665, 'Ŧ', 21, false),
                                    J$.T(7063673, 'ͳ', 21, false)
                                ], 10, false),
                                u: J$.T(7063705, [
                                    J$.T(7063689, 'Ʊ', 21, false),
                                    J$.T(7063697, 'Ս', 21, false)
                                ], 10, false),
                                v: J$.T(7063721, [J$.T(7063713, 'ט', 21, false)], 10, false),
                                w: J$.T(7063761, [
                                    J$.T(7063729, 'Ш', 21, false),
                                    J$.T(7063737, 'Ѡ', 21, false),
                                    J$.T(7063745, 'Ѽ', 21, false),
                                    J$.T(7063753, '\u0D70', 21, false)
                                ], 10, false),
                                x: J$.T(7063801, [
                                    J$.T(7063769, 'Ҳ', 21, false),
                                    J$.T(7063777, 'Ӿ', 21, false),
                                    J$.T(7063785, 'Ӽ', 21, false),
                                    J$.T(7063793, 'ӽ', 21, false)
                                ], 10, false),
                                y: J$.T(7063833, [
                                    J$.T(7063809, '\xA5', 21, false),
                                    J$.T(7063817, 'Ұ', 21, false),
                                    J$.T(7063825, 'Ӌ', 21, false)
                                ], 10, false),
                                z: J$.T(7063857, [
                                    J$.T(7063841, 'Ƶ', 21, false),
                                    J$.T(7063849, 'ɀ', 21, false)
                                ], 10, false)
                            }, 11, false), trap, false, false);
                            J$.M(7064177, J$.R(7063881, 'text', text, false, false), 'forEach', false)(J$.T(7064169, function (c) {
                                jalangiLabel0:
                                    while (true) {
                                        try {
                                            J$.Fe(7064129, arguments.callee, this, arguments);
                                            arguments = J$.N(7064137, 'arguments', arguments, true, false, false);
                                            c = J$.N(7064145, 'c', c, true, false, false);
                                            J$.N(7064153, 'chars', chars, false, false, false);
                                            J$.N(7064161, 'rand', rand, false, false, false);
                                            c = J$.W(7063905, 'c', J$.M(7063897, J$.R(7063889, 'c', c, false, false), 'toLowerCase', false)(), c, false, false);
                                            var chars = J$.W(7063953, 'chars', J$.C(378920, J$.G(7063929, J$.R(7063913, 'trap', trap, false, false), J$.R(7063921, 'c', c, false, false))) ? J$._() : J$.T(7063945, [J$.T(7063937, ' ', 21, false)], 10, false), chars, false, false);
                                            var rand = J$.W(7064009, 'rand', J$.M(7064001, J$.I(typeof Math === 'undefined' ? Math = J$.R(7063961, 'Math', undefined, true, true) : Math = J$.R(7063961, 'Math', Math, true, true)), 'floor', false)(J$.B(716890, '*', J$.M(7063977, J$.I(typeof Math === 'undefined' ? Math = J$.R(7063969, 'Math', undefined, true, true) : Math = J$.R(7063969, 'Math', Math, true, true)), 'random', false)(), J$.G(7063993, J$.R(7063985, 'chars', chars, false, false), 'length'))), rand, false, false);
                                            if (J$.C(378928, J$.B(716906, '!==', J$.U(716898, 'typeof', J$.G(7064033, J$.R(7064017, 'trap', trap, false, false), J$.R(7064025, 'c', c, false, false))), J$.T(7064041, 'undefined', 21, false)))) {
                                                result = J$.W(7064097, 'result', J$.B(716914, '+', J$.R(7064089, 'result', result, false, false), J$.G(7064081, J$.G(7064065, J$.R(7064049, 'trap', trap, false, false), J$.R(7064057, 'c', c, false, false)), J$.R(7064073, 'rand', rand, false, false))), result, false, false);
                                            } else {
                                                result = J$.W(7064121, 'result', J$.B(716922, '+', J$.R(7064113, 'result', result, false, false), J$.R(7064105, 'c', c, false, false)), result, false, false);
                                            }
                                        } catch (J$e) {
                                            J$.Ex(7064273, J$e);
                                        } finally {
                                            if (J$.Fr(7064281))
                                                continue jalangiLabel0;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            return J$.Rt(7064193, J$.R(7064185, 'result', result, false, false));
                        } catch (J$e) {
                            J$.Ex(7064289, J$e);
                        } finally {
                            if (J$.Fr(7064297))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(7064305, J$e);
        } finally {
            if (J$.Sr(7064313))
                continue jalangiLabel2;
            else
                break jalangiLabel2;
        }
    }
// JALANGI DO NOT INSTRUMENT
