// mongodb/lib/constants.js
J$.noInstrEval = false;
jalangiLabel0:
    while (true) {
        try {
            J$.Se(97081, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/constants.js');
            J$.T(97001, 'use strict', 21, false);
            J$.P(97073, J$.I(typeof module === 'undefined' ? module = J$.R(97009, 'module', undefined, true, true) : module = J$.R(97009, 'module', module, true, true)), 'exports', J$.T(97065, {
                SYSTEM_NAMESPACE_COLLECTION: J$.T(97017, 'system.namespaces', 21, false),
                SYSTEM_INDEX_COLLECTION: J$.T(97025, 'system.indexes', 21, false),
                SYSTEM_PROFILE_COLLECTION: J$.T(97033, 'system.profile', 21, false),
                SYSTEM_USER_COLLECTION: J$.T(97041, 'system.users', 21, false),
                SYSTEM_COMMAND_COLLECTION: J$.T(97049, '$cmd', 21, false),
                SYSTEM_JS_COLLECTION: J$.T(97057, 'system.js', 21, false)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(97089, J$e);
        } finally {
            if (J$.Sr(97097))
                continue jalangiLabel0;
            else
                break jalangiLabel0;
        }
    }
// JALANGI DO NOT INSTRUMENT
