// mongodb/lib/core/error.js
J$.noInstrEval = false;
jalangiLabel39:
    while (true) {
        try {
            J$.Se(223505, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/core/error.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(217065, arguments.callee, this, arguments);
                            arguments = J$.N(217073, 'arguments', arguments, true, false, false);
                            obj = J$.N(217081, 'obj', obj, true, false, false);
                            if (J$.C(12664, J$.C(12624, J$.B(14450, '===', J$.U(14442, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(216801, 'Symbol', undefined, true, true) : Symbol = J$.R(216801, 'Symbol', Symbol, true, true))), J$.T(216809, 'function', 21, false))) ? J$.B(14466, '===', J$.U(14458, 'typeof', J$.G(216825, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(216817, 'Symbol', undefined, true, true) : Symbol = J$.R(216817, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(216833, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(216889, '_typeof', J$.T(216881, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(216857, arguments.callee, this, arguments);
                                                arguments = J$.N(216865, 'arguments', arguments, true, false, false);
                                                obj = J$.N(216873, 'obj', obj, true, false, false);
                                                return J$.Rt(216849, J$.U(14474, 'typeof', J$.R(216841, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(223905, J$e);
                                            } finally {
                                                if (J$.Fr(223913))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(217025, '_typeof', J$.T(217017, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(216993, arguments.callee, this, arguments);
                                                arguments = J$.N(217001, 'arguments', arguments, true, false, false);
                                                obj = J$.N(217009, 'obj', obj, true, false, false);
                                                return J$.Rt(216985, J$.C(12656, J$.C(12648, J$.C(12640, J$.C(12632, J$.R(216897, 'obj', obj, false, false)) ? J$.B(14490, '===', J$.U(14482, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(216905, 'Symbol', undefined, true, true) : Symbol = J$.R(216905, 'Symbol', Symbol, true, true))), J$.T(216913, 'function', 21, false)) : J$._()) ? J$.B(14498, '===', J$.G(216929, J$.R(216921, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(216937, 'Symbol', undefined, true, true) : Symbol = J$.R(216937, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(14506, '!==', J$.R(216945, 'obj', obj, false, false), J$.G(216961, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(216953, 'Symbol', undefined, true, true) : Symbol = J$.R(216953, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(216969, 'symbol', 21, false) : J$.U(14514, 'typeof', J$.R(216977, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(223921, J$e);
                                            } finally {
                                                if (J$.Fr(223929))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(217057, J$.F(217049, J$.R(217033, '_typeof', _typeof, false, true), false)(J$.R(217041, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(223937, J$e);
                        } finally {
                            if (J$.Fr(223945))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(217129, arguments.callee, this, arguments);
                            arguments = J$.N(217137, 'arguments', arguments, true, false, false);
                            instance = J$.N(217145, 'instance', instance, true, false, false);
                            Constructor = J$.N(217153, 'Constructor', Constructor, true, false, false);
                            if (J$.C(12672, J$.U(14530, '!', J$.B(14522, 'instanceof', J$.R(217089, 'instance', instance, false, false), J$.R(217097, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(217121, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(217105, 'TypeError', undefined, true, true) : TypeError = J$.R(217105, 'TypeError', TypeError, true, true)), true)(J$.T(217113, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(223953, J$e);
                        } finally {
                            if (J$.Fr(223961))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(217401, arguments.callee, this, arguments);
                            arguments = J$.N(217409, 'arguments', arguments, true, false, false);
                            target = J$.N(217417, 'target', target, true, false, false);
                            props = J$.N(217425, 'props', props, true, false, false);
                            J$.N(217433, 'i', i, false, false, false);
                            J$.N(217441, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(217169, 'i', J$.T(217161, 0, 22, false), i, false, false); J$.C(12696, J$.B(14538, '<', J$.R(217177, 'i', i, false, false), J$.G(217193, J$.R(217185, 'props', props, false, false), 'length'))); J$.B(14562, '-', i = J$.W(217209, 'i', J$.B(14554, '+', J$.U(14546, '+', J$.R(217201, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(217241, 'descriptor', J$.G(217233, J$.R(217217, 'props', props, false, false), J$.R(217225, 'i', i, false, false)), descriptor, false, false);
                                J$.P(217281, J$.R(217249, 'descriptor', descriptor, false, false), 'enumerable', J$.C(12680, J$.G(217265, J$.R(217257, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(217273, false, 23, false));
                                J$.P(217305, J$.R(217289, 'descriptor', descriptor, false, false), 'configurable', J$.T(217297, true, 23, false));
                                if (J$.C(12688, J$.B(14570, 'in', J$.T(217313, 'value', 21, false), J$.R(217321, 'descriptor', descriptor, false, false))))
                                    J$.P(217345, J$.R(217329, 'descriptor', descriptor, false, false), 'writable', J$.T(217337, true, 23, false));
                                J$.M(217393, J$.I(typeof Object === 'undefined' ? Object = J$.R(217353, 'Object', undefined, true, true) : Object = J$.R(217353, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(217361, 'target', target, false, false), J$.G(217377, J$.R(217369, 'descriptor', descriptor, false, false), 'key'), J$.R(217385, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(223969, J$e);
                        } finally {
                            if (J$.Fr(223977))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(217553, arguments.callee, this, arguments);
                            arguments = J$.N(217561, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(217569, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(217577, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(217585, 'staticProps', staticProps, true, false, false);
                            if (J$.C(12704, J$.R(217449, 'protoProps', protoProps, false, false)))
                                J$.F(217489, J$.R(217457, '_defineProperties', _defineProperties, false, true), false)(J$.G(217473, J$.R(217465, 'Constructor', Constructor, false, false), 'prototype'), J$.R(217481, 'protoProps', protoProps, false, false));
                            if (J$.C(12712, J$.R(217497, 'staticProps', staticProps, false, false)))
                                J$.F(217529, J$.R(217505, '_defineProperties', _defineProperties, false, true), false)(J$.R(217513, 'Constructor', Constructor, false, false), J$.R(217521, 'staticProps', staticProps, false, false));
                            return J$.Rt(217545, J$.R(217537, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(223985, J$e);
                        } finally {
                            if (J$.Fr(223993))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(217697, arguments.callee, this, arguments);
                            arguments = J$.N(217705, 'arguments', arguments, true, false, false);
                            self = J$.N(217713, 'self', self, true, false, false);
                            call = J$.N(217721, 'call', call, true, false, false);
                            if (J$.C(12736, J$.C(12728, J$.R(217593, 'call', call, false, false)) ? J$.C(12720, J$.B(14578, '===', J$.F(217617, J$.R(217601, '_typeof', _typeof, false, true), false)(J$.R(217609, 'call', call, false, false)), J$.T(217625, 'object', 21, false))) ? J$._() : J$.B(14594, '===', J$.U(14586, 'typeof', J$.R(217633, 'call', call, false, false)), J$.T(217641, 'function', 21, false)) : J$._())) {
                                return J$.Rt(217657, J$.R(217649, 'call', call, false, false));
                            }
                            return J$.Rt(217689, J$.F(217681, J$.R(217665, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(217673, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(224001, J$e);
                        } finally {
                            if (J$.Fr(224009))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(217785, arguments.callee, this, arguments);
                            arguments = J$.N(217793, 'arguments', arguments, true, false, false);
                            self = J$.N(217801, 'self', self, true, false, false);
                            if (J$.C(12744, J$.B(14602, '===', J$.R(217729, 'self', self, false, false), void J$.T(217737, 0, 22, false)))) {
                                throw J$.F(217761, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(217745, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(217745, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(217753, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(217777, J$.R(217769, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(224017, J$e);
                        } finally {
                            if (J$.Fr(224025))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(218001, arguments.callee, this, arguments);
                            arguments = J$.N(218009, 'arguments', arguments, true, false, false);
                            subClass = J$.N(218017, 'subClass', subClass, true, false, false);
                            superClass = J$.N(218025, 'superClass', superClass, true, false, false);
                            if (J$.C(12760, J$.C(12752, J$.B(14618, '!==', J$.U(14610, 'typeof', J$.R(217809, 'superClass', superClass, false, false)), J$.T(217817, 'function', 21, false))) ? J$.B(14626, '!==', J$.R(217825, 'superClass', superClass, false, false), J$.T(217833, null, 25, false)) : J$._())) {
                                throw J$.F(217857, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(217841, 'TypeError', undefined, true, true) : TypeError = J$.R(217841, 'TypeError', TypeError, true, true)), true)(J$.T(217849, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(217953, J$.R(217865, 'subClass', subClass, false, false), 'prototype', J$.M(217945, J$.I(typeof Object === 'undefined' ? Object = J$.R(217873, 'Object', undefined, true, true) : Object = J$.R(217873, 'Object', Object, true, true)), 'create', false)(J$.C(12768, J$.R(217881, 'superClass', superClass, false, false)) ? J$.G(217897, J$.R(217889, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(217937, {
                                constructor: J$.T(217929, {
                                    value: J$.R(217905, 'subClass', subClass, false, false),
                                    writable: J$.T(217913, true, 23, false),
                                    configurable: J$.T(217921, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(12776, J$.R(217961, 'superClass', superClass, false, false)))
                                J$.F(217993, J$.R(217969, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(217977, 'subClass', subClass, false, false), J$.R(217985, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(224033, J$e);
                        } finally {
                            if (J$.Fr(224041))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _wrapNativeSuper(Class) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(218593, arguments.callee, this, arguments);
                            arguments = J$.N(218601, 'arguments', arguments, true, false, false);
                            Class = J$.N(218609, 'Class', Class, true, false, false);
                            J$.N(218617, '_cache', _cache, false, false, false);
                            var _cache = J$.W(218073, '_cache', J$.C(12784, J$.B(14642, '===', J$.U(14634, 'typeof', J$.I(typeof Map === 'undefined' ? Map = J$.R(218033, 'Map', undefined, true, true) : Map = J$.R(218033, 'Map', Map, true, true))), J$.T(218041, 'function', 21, false))) ? J$.F(218057, J$.I(typeof Map === 'undefined' ? Map = J$.R(218049, 'Map', undefined, true, true) : Map = J$.R(218049, 'Map', Map, true, true)), true)() : J$.T(218065, undefined, 24, false), _cache, false, false);
                            _wrapNativeSuper = J$.W(218553, '_wrapNativeSuper', J$.T(218545, function _wrapNativeSuper(Class) {
                                jalangiLabel10:
                                    while (true) {
                                        try {
                                            J$.Fe(218505, arguments.callee, this, arguments);
                                            function Wrapper() {
                                                jalangiLabel9:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(218353, arguments.callee, this, arguments);
                                                            arguments = J$.N(218361, 'arguments', arguments, true, false, false);
                                                            return J$.Rt(218345, J$.F(218337, J$.R(218281, '_construct', _construct, false, true), false)(J$.R(218289, 'Class', Class, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(218297, 'arguments', undefined, true, true) : arguments = J$.R(218297, 'arguments', arguments, true, true)), J$.G(218329, J$.F(218321, J$.R(218305, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(218313, 'this', this, false, false)), 'constructor')));
                                                        } catch (J$e) {
                                                            J$.Ex(224049, J$e);
                                                        } finally {
                                                            if (J$.Fr(224057))
                                                                continue jalangiLabel9;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }
                                            arguments = J$.N(218513, 'arguments', arguments, true, false, false);
                                            Class = J$.N(218521, 'Class', Class, true, false, false);
                                            Wrapper = J$.N(218537, 'Wrapper', J$.T(218529, Wrapper, 12, false), true, false, false);
                                            if (J$.C(12800, J$.C(12792, J$.B(14650, '===', J$.R(218081, 'Class', Class, false, false), J$.T(218089, null, 25, false))) ? J$._() : J$.U(14658, '!', J$.F(218113, J$.R(218097, '_isNativeFunction', _isNativeFunction, false, true), false)(J$.R(218105, 'Class', Class, false, false)))))
                                                return J$.Rt(218129, J$.R(218121, 'Class', Class, false, false));
                                            if (J$.C(12808, J$.B(14674, '!==', J$.U(14666, 'typeof', J$.R(218137, 'Class', Class, false, false)), J$.T(218145, 'function', 21, false)))) {
                                                throw J$.F(218169, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(218153, 'TypeError', undefined, true, true) : TypeError = J$.R(218153, 'TypeError', TypeError, true, true)), true)(J$.T(218161, 'Super expression must either be null or a function', 21, false));
                                            }
                                            if (J$.C(12824, J$.B(14690, '!==', J$.U(14682, 'typeof', J$.R(218177, '_cache', _cache, false, false)), J$.T(218185, 'undefined', 21, false)))) {
                                                if (J$.C(12816, J$.M(218209, J$.R(218193, '_cache', _cache, false, false), 'has', false)(J$.R(218201, 'Class', Class, false, false))))
                                                    return J$.Rt(218241, J$.M(218233, J$.R(218217, '_cache', _cache, false, false), 'get', false)(J$.R(218225, 'Class', Class, false, false)));
                                                J$.M(218273, J$.R(218249, '_cache', _cache, false, false), 'set', false)(J$.R(218257, 'Class', Class, false, false), J$.R(218265, 'Wrapper', Wrapper, false, false));
                                            }
                                            J$.P(218457, J$.R(218369, 'Wrapper', Wrapper, false, false), 'prototype', J$.M(218449, J$.I(typeof Object === 'undefined' ? Object = J$.R(218377, 'Object', undefined, true, true) : Object = J$.R(218377, 'Object', Object, true, true)), 'create', false)(J$.G(218393, J$.R(218385, 'Class', Class, false, false), 'prototype'), J$.T(218441, {
                                                constructor: J$.T(218433, {
                                                    value: J$.R(218401, 'Wrapper', Wrapper, false, false),
                                                    enumerable: J$.T(218409, false, 23, false),
                                                    writable: J$.T(218417, true, 23, false),
                                                    configurable: J$.T(218425, true, 23, false)
                                                }, 11, false)
                                            }, 11, false)));
                                            return J$.Rt(218497, J$.F(218489, J$.R(218465, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(218473, 'Wrapper', Wrapper, false, false), J$.R(218481, 'Class', Class, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(224065, J$e);
                                        } finally {
                                            if (J$.Fr(224073))
                                                continue jalangiLabel10;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _wrapNativeSuper, false, true);
                            return J$.Rt(218585, J$.F(218577, J$.R(218561, '_wrapNativeSuper', _wrapNativeSuper, false, true), false)(J$.R(218569, 'Class', Class, false, false)));
                        } catch (J$e) {
                            J$.Ex(224081, J$e);
                        } finally {
                            if (J$.Fr(224089))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isNativeReflectConstruct() {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(218873, arguments.callee, this, arguments);
                            arguments = J$.N(218881, 'arguments', arguments, true, false, false);
                            if (J$.C(12840, J$.C(12832, J$.B(14706, '===', J$.U(14698, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(218625, 'Reflect', undefined, true, true) : Reflect = J$.R(218625, 'Reflect', Reflect, true, true))), J$.T(218633, 'undefined', 21, false))) ? J$._() : J$.U(14714, '!', J$.G(218649, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(218641, 'Reflect', undefined, true, true) : Reflect = J$.R(218641, 'Reflect', Reflect, true, true)), 'construct'))))
                                return J$.Rt(218665, J$.T(218657, false, 23, false));
                            if (J$.C(12848, J$.G(218689, J$.G(218681, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(218673, 'Reflect', undefined, true, true) : Reflect = J$.R(218673, 'Reflect', Reflect, true, true)), 'construct'), 'sham')))
                                return J$.Rt(218705, J$.T(218697, false, 23, false));
                            if (J$.C(12856, J$.B(14730, '===', J$.U(14722, 'typeof', J$.I(typeof Proxy === 'undefined' ? Proxy = J$.R(218713, 'Proxy', undefined, true, true) : Proxy = J$.R(218713, 'Proxy', Proxy, true, true))), J$.T(218721, 'function', 21, false))))
                                return J$.Rt(218737, J$.T(218729, true, 23, false));
                            try {
                                J$.M(218825, J$.G(218761, J$.G(218753, J$.I(typeof Date === 'undefined' ? Date = J$.R(218745, 'Date', undefined, true, true) : Date = J$.R(218745, 'Date', Date, true, true)), 'prototype'), 'toString'), 'call', false)(J$.M(218817, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(218769, 'Reflect', undefined, true, true) : Reflect = J$.R(218769, 'Reflect', Reflect, true, true)), 'construct', false)(J$.I(typeof Date === 'undefined' ? Date = J$.R(218777, 'Date', undefined, true, true) : Date = J$.R(218777, 'Date', Date, true, true)), J$.T(218785, [], 10, false), J$.T(218809, function () {
                                    jalangiLabel12:
                                        while (true) {
                                            try {
                                                J$.Fe(218793, arguments.callee, this, arguments);
                                                arguments = J$.N(218801, 'arguments', arguments, true, false, false);
                                            } catch (J$e) {
                                                J$.Ex(224097, J$e);
                                            } finally {
                                                if (J$.Fr(224105))
                                                    continue jalangiLabel12;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)));
                                return J$.Rt(218841, J$.T(218833, true, 23, false));
                            } catch (e) {
                                J$.N(218865, 'e', e, false, false, true);
                                return J$.Rt(218857, J$.T(218849, false, 23, false));
                            }
                        } catch (J$e) {
                            J$.Ex(224113, J$e);
                        } finally {
                            if (J$.Fr(224121))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _construct(Parent, args, Class) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(219249, arguments.callee, this, arguments);
                            arguments = J$.N(219257, 'arguments', arguments, true, false, false);
                            Parent = J$.N(219265, 'Parent', Parent, true, false, false);
                            args = J$.N(219273, 'args', args, true, false, false);
                            Class = J$.N(219281, 'Class', Class, true, false, false);
                            if (J$.C(12872, J$.F(218897, J$.R(218889, 'isNativeReflectConstruct', isNativeReflectConstruct, false, true), false)())) {
                                _construct = J$.W(218921, '_construct', J$.G(218913, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(218905, 'Reflect', undefined, true, true) : Reflect = J$.R(218905, 'Reflect', Reflect, true, true)), 'construct'), _construct, false, true);
                            } else {
                                _construct = J$.W(219201, '_construct', J$.T(219193, function _construct(Parent, args, Class) {
                                    jalangiLabel14:
                                        while (true) {
                                            try {
                                                J$.Fe(219129, arguments.callee, this, arguments);
                                                arguments = J$.N(219137, 'arguments', arguments, true, false, false);
                                                Parent = J$.N(219145, 'Parent', Parent, true, false, false);
                                                args = J$.N(219153, 'args', args, true, false, false);
                                                Class = J$.N(219161, 'Class', Class, true, false, false);
                                                J$.N(219169, 'a', a, false, false, false);
                                                J$.N(219177, 'Constructor', Constructor, false, false, false);
                                                J$.N(219185, 'instance', instance, false, false, false);
                                                var a = J$.W(218945, 'a', J$.T(218937, [J$.T(218929, null, 25, false)], 10, false), a, false, false);
                                                J$.M(218985, J$.G(218961, J$.R(218953, 'a', a, false, false), 'push'), 'apply', false)(J$.R(218969, 'a', a, false, false), J$.R(218977, 'args', args, false, false));
                                                var Constructor = J$.W(219033, 'Constructor', J$.M(219025, J$.G(219001, J$.I(typeof Function === 'undefined' ? Function = J$.R(218993, 'Function', undefined, true, true) : Function = J$.R(218993, 'Function', Function, true, true)), 'bind'), 'apply', false)(J$.R(219009, 'Parent', Parent, false, false), J$.R(219017, 'a', a, false, false)), Constructor, false, false);
                                                var instance = J$.W(219057, 'instance', J$.F(219049, J$.R(219041, 'Constructor', Constructor, false, false), true)(), instance, false, false);
                                                if (J$.C(12864, J$.R(219065, 'Class', Class, false, false)))
                                                    J$.F(219105, J$.R(219073, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(219081, 'instance', instance, false, false), J$.G(219097, J$.R(219089, 'Class', Class, false, false), 'prototype'));
                                                return J$.Rt(219121, J$.R(219113, 'instance', instance, false, false));
                                            } catch (J$e) {
                                                J$.Ex(224129, J$e);
                                            } finally {
                                                if (J$.Fr(224137))
                                                    continue jalangiLabel14;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _construct, false, true);
                            }
                            return J$.Rt(219241, J$.M(219233, J$.R(219209, '_construct', _construct, false, true), 'apply', false)(J$.T(219217, null, 25, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(219225, 'arguments', undefined, true, true) : arguments = J$.R(219225, 'arguments', arguments, true, true))));
                        } catch (J$e) {
                            J$.Ex(224145, J$e);
                        } finally {
                            if (J$.Fr(224153))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _isNativeFunction(fn) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(219353, arguments.callee, this, arguments);
                            arguments = J$.N(219361, 'arguments', arguments, true, false, false);
                            fn = J$.N(219369, 'fn', fn, true, false, false);
                            return J$.Rt(219345, J$.B(14746, '!==', J$.M(219329, J$.M(219313, J$.G(219297, J$.I(typeof Function === 'undefined' ? Function = J$.R(219289, 'Function', undefined, true, true) : Function = J$.R(219289, 'Function', Function, true, true)), 'toString'), 'call', false)(J$.R(219305, 'fn', fn, false, false)), 'indexOf', false)(J$.T(219321, '[native code]', 21, false)), J$.U(14738, '-', J$.T(219337, 1, 22, false))));
                        } catch (J$e) {
                            J$.Ex(224161, J$e);
                        } finally {
                            if (J$.Fr(224169))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(219521, arguments.callee, this, arguments);
                            arguments = J$.N(219529, 'arguments', arguments, true, false, false);
                            o = J$.N(219537, 'o', o, true, false, false);
                            p = J$.N(219545, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(219473, '_setPrototypeOf', J$.C(12880, J$.G(219385, J$.I(typeof Object === 'undefined' ? Object = J$.R(219377, 'Object', undefined, true, true) : Object = J$.R(219377, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(219465, function _setPrototypeOf(o, p) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(219433, arguments.callee, this, arguments);
                                            arguments = J$.N(219441, 'arguments', arguments, true, false, false);
                                            o = J$.N(219449, 'o', o, true, false, false);
                                            p = J$.N(219457, 'p', p, true, false, false);
                                            J$.P(219409, J$.R(219393, 'o', o, false, false), '__proto__', J$.R(219401, 'p', p, false, false));
                                            return J$.Rt(219425, J$.R(219417, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(224177, J$e);
                                        } finally {
                                            if (J$.Fr(224185))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(219513, J$.F(219505, J$.R(219481, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(219489, 'o', o, false, false), J$.R(219497, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(224193, J$e);
                        } finally {
                            if (J$.Fr(224201))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(219705, arguments.callee, this, arguments);
                            arguments = J$.N(219713, 'arguments', arguments, true, false, false);
                            o = J$.N(219721, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(219665, '_getPrototypeOf', J$.C(12896, J$.G(219561, J$.I(typeof Object === 'undefined' ? Object = J$.R(219553, 'Object', undefined, true, true) : Object = J$.R(219553, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(219577, J$.I(typeof Object === 'undefined' ? Object = J$.R(219569, 'Object', undefined, true, true) : Object = J$.R(219569, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(219657, function _getPrototypeOf(o) {
                                jalangiLabel19:
                                    while (true) {
                                        try {
                                            J$.Fe(219633, arguments.callee, this, arguments);
                                            arguments = J$.N(219641, 'arguments', arguments, true, false, false);
                                            o = J$.N(219649, 'o', o, true, false, false);
                                            return J$.Rt(219625, J$.C(12888, J$.G(219593, J$.R(219585, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(219617, J$.I(typeof Object === 'undefined' ? Object = J$.R(219601, 'Object', undefined, true, true) : Object = J$.R(219601, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(219609, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(224209, J$e);
                                        } finally {
                                            if (J$.Fr(224217))
                                                continue jalangiLabel19;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(219697, J$.F(219689, J$.R(219673, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(219681, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(224225, J$e);
                        } finally {
                            if (J$.Fr(224233))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function makeWriteConcernResultObject(input) {
                jalangiLabel31:
                    while (true) {
                        try {
                            J$.Fe(221961, arguments.callee, this, arguments);
                            arguments = J$.N(221969, 'arguments', arguments, true, false, false);
                            input = J$.N(221977, 'input', input, true, false, false);
                            J$.N(221985, 'output', output, false, false, false);
                            var output = J$.W(221865, 'output', J$.M(221857, J$.I(typeof Object === 'undefined' ? Object = J$.R(221833, 'Object', undefined, true, true) : Object = J$.R(221833, 'Object', Object, true, true)), 'assign', false)(J$.T(221841, {}, 11, false), J$.R(221849, 'input', input, false, false)), output, false, false);
                            if (J$.C(12968, J$.B(14802, '===', J$.G(221881, J$.R(221873, 'output', output, false, false), 'ok'), J$.T(221889, 0, 22, false)))) {
                                J$.P(221913, J$.R(221897, 'output', output, false, false), 'ok', J$.T(221905, 1, 22, false));
                                J$.B(14810, 'delete', J$.R(221921, 'output', output, false, false), 'errmsg');
                                J$.B(14818, 'delete', J$.R(221929, 'output', output, false, false), 'code');
                                J$.B(14826, 'delete', J$.R(221937, 'output', output, false, false), 'codeName');
                            }
                            return J$.Rt(221953, J$.R(221945, 'output', output, false, false));
                        } catch (J$e) {
                            J$.Ex(224401, J$e);
                        } finally {
                            if (J$.Fr(224409))
                                continue jalangiLabel31;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isRetryableError(error) {
                jalangiLabel34:
                    while (true) {
                        try {
                            J$.Fe(222601, arguments.callee, this, arguments);
                            arguments = J$.N(222609, 'arguments', arguments, true, false, false);
                            error = J$.N(222617, 'error', error, true, false, false);
                            return J$.Rt(222593, J$.C(13000, J$.C(12992, J$.C(12984, J$.M(222505, J$.R(222481, 'RETRYABLE_ERROR_CODES', RETRYABLE_ERROR_CODES, false, true), 'has', false)(J$.G(222497, J$.R(222489, 'error', error, false, false), 'code'))) ? J$._() : J$.B(14842, 'instanceof', J$.R(222513, 'error', error, false, false), J$.R(222521, 'MongoNetworkError', MongoNetworkError, false, true))) ? J$._() : J$.M(222553, J$.G(222537, J$.R(222529, 'error', error, false, false), 'message'), 'match', false)(J$.T(222545, /not master/, 14, false))) ? J$._() : J$.M(222585, J$.G(222569, J$.R(222561, 'error', error, false, false), 'message'), 'match', false)(J$.T(222577, /node is recovering/, 14, false)));
                        } catch (J$e) {
                            J$.Ex(224449, J$e);
                        } finally {
                            if (J$.Fr(224457))
                                continue jalangiLabel34;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isRecoveringError(err) {
                jalangiLabel35:
                    while (true) {
                        try {
                            J$.Fe(222929, arguments.callee, this, arguments);
                            arguments = J$.N(222937, 'arguments', arguments, true, false, false);
                            err = J$.N(222945, 'err', err, true, false, false);
                            if (J$.C(13016, J$.C(13008, J$.G(222801, J$.R(222793, 'err', err, false, false), 'code')) ? J$.M(222833, J$.R(222809, 'SDAM_RECOVERING_CODES', SDAM_RECOVERING_CODES, false, true), 'has', false)(J$.G(222825, J$.R(222817, 'err', err, false, false), 'code')) : J$._())) {
                                return J$.Rt(222849, J$.T(222841, true, 23, false));
                            }
                            return J$.Rt(222921, J$.C(13024, J$.M(222881, J$.G(222865, J$.R(222857, 'err', err, false, false), 'message'), 'match', false)(J$.T(222873, /not master or secondary/, 14, false))) ? J$._() : J$.M(222913, J$.G(222897, J$.R(222889, 'err', err, false, false), 'message'), 'match', false)(J$.T(222905, /node is recovering/, 14, false)));
                        } catch (J$e) {
                            J$.Ex(224465, J$e);
                        } finally {
                            if (J$.Fr(224473))
                                continue jalangiLabel35;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isNotMasterError(err) {
                jalangiLabel36:
                    while (true) {
                        try {
                            J$.Fe(223097, arguments.callee, this, arguments);
                            arguments = J$.N(223105, 'arguments', arguments, true, false, false);
                            err = J$.N(223113, 'err', err, true, false, false);
                            if (J$.C(13040, J$.C(13032, J$.G(222961, J$.R(222953, 'err', err, false, false), 'code')) ? J$.M(222993, J$.R(222969, 'SDAM_NOTMASTER_CODES', SDAM_NOTMASTER_CODES, false, true), 'has', false)(J$.G(222985, J$.R(222977, 'err', err, false, false), 'code')) : J$._())) {
                                return J$.Rt(223009, J$.T(223001, true, 23, false));
                            }
                            if (J$.C(13048, J$.F(223033, J$.R(223017, 'isRecoveringError', isRecoveringError, false, true), false)(J$.R(223025, 'err', err, false, false)))) {
                                return J$.Rt(223049, J$.T(223041, false, 23, false));
                            }
                            return J$.Rt(223089, J$.M(223081, J$.G(223065, J$.R(223057, 'err', err, false, false), 'message'), 'match', false)(J$.T(223073, /not master/, 14, false)));
                        } catch (J$e) {
                            J$.Ex(224481, J$e);
                        } finally {
                            if (J$.Fr(224489))
                                continue jalangiLabel36;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isNodeShuttingDownError(err) {
                jalangiLabel37:
                    while (true) {
                        try {
                            J$.Fe(223177, arguments.callee, this, arguments);
                            arguments = J$.N(223185, 'arguments', arguments, true, false, false);
                            err = J$.N(223193, 'err', err, true, false, false);
                            return J$.Rt(223169, J$.C(13056, J$.G(223129, J$.R(223121, 'err', err, false, false), 'code')) ? J$.M(223161, J$.R(223137, 'SDAM_NODE_SHUTTING_DOWN_ERROR_CODES', SDAM_NODE_SHUTTING_DOWN_ERROR_CODES, false, true), 'has', false)(J$.G(223153, J$.R(223145, 'err', err, false, false), 'code')) : J$._());
                        } catch (J$e) {
                            J$.Ex(224497, J$e);
                        } finally {
                            if (J$.Fr(224505))
                                continue jalangiLabel37;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isSDAMUnrecoverableError(error, server) {
                jalangiLabel38:
                    while (true) {
                        try {
                            J$.Fe(223385, arguments.callee, this, arguments);
                            arguments = J$.N(223393, 'arguments', arguments, true, false, false);
                            error = J$.N(223401, 'error', error, true, false, false);
                            server = J$.N(223409, 'server', server, true, false, false);
                            if (J$.C(13064, J$.B(14850, 'instanceof', J$.R(223201, 'error', error, false, false), J$.R(223209, 'MongoParseError', MongoParseError, false, true)))) {
                                return J$.Rt(223225, J$.T(223217, true, 23, false));
                            }
                            if (J$.C(13096, J$.C(13072, J$.F(223249, J$.R(223233, 'isRecoveringError', isRecoveringError, false, true), false)(J$.R(223241, 'error', error, false, false))) ? J$._() : J$.F(223273, J$.R(223257, 'isNotMasterError', isNotMasterError, false, true), false)(J$.R(223265, 'error', error, false, false)))) {
                                if (J$.C(13088, J$.C(13080, J$.B(14858, '>=', J$.F(223297, J$.R(223281, 'maxWireVersion', maxWireVersion, false, true), false)(J$.R(223289, 'server', server, false, false)), J$.T(223305, 8, 22, false))) ? J$.U(14866, '!', J$.F(223329, J$.R(223313, 'isNodeShuttingDownError', isNodeShuttingDownError, false, true), false)(J$.R(223321, 'error', error, false, false))) : J$._())) {
                                    return J$.Rt(223345, J$.T(223337, false, 23, false));
                                }
                                return J$.Rt(223361, J$.T(223353, true, 23, false));
                            }
                            return J$.Rt(223377, J$.T(223369, false, 23, false));
                        } catch (J$e) {
                            J$.Ex(224513, J$e);
                        } finally {
                            if (J$.Fr(224521))
                                continue jalangiLabel38;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(223521, '_typeof', J$.T(223513, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(223537, '_classCallCheck', J$.T(223529, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(223553, '_defineProperties', J$.T(223545, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(223569, '_createClass', J$.T(223561, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(223585, '_possibleConstructorReturn', J$.T(223577, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(223601, '_assertThisInitialized', J$.T(223593, _assertThisInitialized, 12, false), true, false, false);
            _inherits = J$.N(223617, '_inherits', J$.T(223609, _inherits, 12, false), true, false, false);
            _wrapNativeSuper = J$.N(223633, '_wrapNativeSuper', J$.T(223625, _wrapNativeSuper, 12, false), true, false, false);
            isNativeReflectConstruct = J$.N(223649, 'isNativeReflectConstruct', J$.T(223641, isNativeReflectConstruct, 12, false), true, false, false);
            _construct = J$.N(223665, '_construct', J$.T(223657, _construct, 12, false), true, false, false);
            _isNativeFunction = J$.N(223681, '_isNativeFunction', J$.T(223673, _isNativeFunction, 12, false), true, false, false);
            _setPrototypeOf = J$.N(223697, '_setPrototypeOf', J$.T(223689, _setPrototypeOf, 12, false), true, false, false);
            _getPrototypeOf = J$.N(223713, '_getPrototypeOf', J$.T(223705, _getPrototypeOf, 12, false), true, false, false);
            J$.N(223721, 'mongoErrorContextSymbol', mongoErrorContextSymbol, false, false, false);
            J$.N(223729, 'maxWireVersion', maxWireVersion, false, false, false);
            J$.N(223737, 'MongoError', MongoError, false, false, false);
            J$.N(223745, 'MongoNetworkError', MongoNetworkError, false, false, false);
            J$.N(223753, 'MongoParseError', MongoParseError, false, false, false);
            J$.N(223761, 'MongoTimeoutError', MongoTimeoutError, false, false, false);
            makeWriteConcernResultObject = J$.N(223777, 'makeWriteConcernResultObject', J$.T(223769, makeWriteConcernResultObject, 12, false), true, false, false);
            J$.N(223785, 'MongoWriteConcernError', MongoWriteConcernError, false, false, false);
            J$.N(223793, 'RETRYABLE_ERROR_CODES', RETRYABLE_ERROR_CODES, false, false, false);
            isRetryableError = J$.N(223809, 'isRetryableError', J$.T(223801, isRetryableError, 12, false), true, false, false);
            J$.N(223817, 'SDAM_RECOVERING_CODES', SDAM_RECOVERING_CODES, false, false, false);
            J$.N(223825, 'SDAM_NOTMASTER_CODES', SDAM_NOTMASTER_CODES, false, false, false);
            J$.N(223833, 'SDAM_NODE_SHUTTING_DOWN_ERROR_CODES', SDAM_NODE_SHUTTING_DOWN_ERROR_CODES, false, false, false);
            isRecoveringError = J$.N(223849, 'isRecoveringError', J$.T(223841, isRecoveringError, 12, false), true, false, false);
            isNotMasterError = J$.N(223865, 'isNotMasterError', J$.T(223857, isNotMasterError, 12, false), true, false, false);
            isNodeShuttingDownError = J$.N(223881, 'isNodeShuttingDownError', J$.T(223873, isNodeShuttingDownError, 12, false), true, false, false);
            isSDAMUnrecoverableError = J$.N(223897, 'isSDAMUnrecoverableError', J$.T(223889, isSDAMUnrecoverableError, 12, false), true, false, false);
            J$.T(216793, 'use strict', 21, false);
            var mongoErrorContextSymbol = J$.W(219753, 'mongoErrorContextSymbol', J$.F(219745, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(219729, 'Symbol', undefined, true, true) : Symbol = J$.R(219729, 'Symbol', Symbol, true, true)), false)(J$.T(219737, 'mongoErrorContextSymbol', 21, false)), mongoErrorContextSymbol, false, true);
            var maxWireVersion = J$.W(219793, 'maxWireVersion', J$.G(219785, J$.F(219777, J$.I(typeof require === 'undefined' ? require = J$.R(219761, 'require', undefined, true, true) : require = J$.R(219761, 'require', require, true, true)), false)(J$.T(219769, './utils', 21, false)), 'maxWireVersion'), maxWireVersion, false, true);
            var MongoError = J$.W(220833, 'MongoError', J$.F(220825, J$.T(220793, function (_Error) {
                jalangiLabel24:
                    while (true) {
                        try {
                            J$.Fe(220753, arguments.callee, this, arguments);
                            function MongoError(message) {
                                jalangiLabel21:
                                    while (true) {
                                        try {
                                            J$.Fe(220465, arguments.callee, this, arguments);
                                            arguments = J$.N(220473, 'arguments', arguments, true, false, false);
                                            message = J$.N(220481, 'message', message, true, false, false);
                                            J$.N(220489, '_this', _this, false, false, false);
                                            J$.N(220497, 'name', name, false, false, false);
                                            var _this;
                                            J$.F(219857, J$.R(219833, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(219841, 'this', this, false, false), J$.R(219849, 'MongoError', MongoError, false, false));
                                            if (J$.C(12936, J$.B(14754, 'instanceof', J$.R(219865, 'message', message, false, false), J$.I(typeof Error === 'undefined' ? Error = J$.R(219873, 'Error', undefined, true, true) : Error = J$.R(219873, 'Error', Error, true, true))))) {
                                                _this = J$.W(219961, '_this', J$.F(219953, J$.R(219881, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(219889, 'this', this, false, false), J$.M(219945, J$.F(219913, J$.R(219897, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(219905, 'MongoError', MongoError, false, false)), 'call', false)(J$.R(219921, 'this', this, false, false), J$.G(219937, J$.R(219929, 'message', message, false, false), 'message'))), _this, false, false);
                                                J$.P(219993, J$.R(219969, '_this', _this, false, false), 'stack', J$.G(219985, J$.R(219977, 'message', message, false, false), 'stack'));
                                            } else {
                                                if (J$.C(12928, J$.B(14770, '===', J$.U(14762, 'typeof', J$.R(220001, 'message', message, false, false)), J$.T(220009, 'string', 21, false)))) {
                                                    _this = J$.W(220089, '_this', J$.F(220081, J$.R(220017, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(220025, 'this', this, false, false), J$.M(220073, J$.F(220049, J$.R(220033, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(220041, 'MongoError', MongoError, false, false)), 'call', false)(J$.R(220057, 'this', this, false, false), J$.R(220065, 'message', message, false, false))), _this, false, false);
                                                } else {
                                                    _this = J$.W(220217, '_this', J$.F(220209, J$.R(220097, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(220105, 'this', this, false, false), J$.M(220201, J$.F(220129, J$.R(220113, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(220121, 'MongoError', MongoError, false, false)), 'call', false)(J$.R(220137, 'this', this, false, false), J$.C(12920, J$.C(12912, J$.C(12904, J$.G(220153, J$.R(220145, 'message', message, false, false), 'message')) ? J$._() : J$.G(220169, J$.R(220161, 'message', message, false, false), 'errmsg')) ? J$._() : J$.G(220185, J$.R(220177, 'message', message, false, false), '$err')) ? J$._() : J$.T(220193, 'n/a', 21, false))), _this, false, false);
                                                    for (var name in J$.H(220281, J$.R(220225, 'message', message, false, false))) {
                                                        J$.N(220289, 'name', name, false, true, false);
                                                        {
                                                            {
                                                                J$.P(220273, J$.R(220233, '_this', _this, false, false), J$.R(220241, 'name', name, false, false), J$.G(220265, J$.R(220249, 'message', message, false, false), J$.R(220257, 'name', name, false, false)));
                                                            }
                                                        }
                                                    }
                                                }
                                                J$.M(220345, J$.I(typeof Error === 'undefined' ? Error = J$.R(220297, 'Error', undefined, true, true) : Error = J$.R(220297, 'Error', Error, true, true)), 'captureStackTrace', false)(J$.F(220321, J$.R(220305, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(220313, '_this', _this, false, false)), J$.G(220337, J$.R(220329, '_this', _this, false, false), 'constructor'));
                                            }
                                            J$.P(220369, J$.R(220353, '_this', _this, false, false), 'name', J$.T(220361, 'MongoError', 21, false));
                                            J$.P(220425, J$.R(220377, '_this', _this, false, false), J$.R(220385, 'mongoErrorContextSymbol', mongoErrorContextSymbol, false, true), J$.C(12944, J$.G(220409, J$.R(220393, '_this', _this, false, false), J$.R(220401, 'mongoErrorContextSymbol', mongoErrorContextSymbol, false, true))) ? J$._() : J$.T(220417, {}, 11, false));
                                            return J$.Rt(220457, J$.F(220449, J$.R(220433, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(220441, '_this', _this, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(224241, J$e);
                                        } finally {
                                            if (J$.Fr(224249))
                                                continue jalangiLabel21;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(220761, 'arguments', arguments, true, false, false);
                            _Error = J$.N(220769, '_Error', _Error, true, false, false);
                            MongoError = J$.N(220785, 'MongoError', J$.T(220777, MongoError, 12, false), true, false, false);
                            J$.F(219825, J$.R(219801, '_inherits', _inherits, false, true), false)(J$.R(219809, 'MongoError', MongoError, false, false), J$.R(219817, '_Error', _Error, false, false));
                            J$.F(220729, J$.R(220505, '_createClass', _createClass, false, true), false)(J$.R(220513, 'MongoError', MongoError, false, false), J$.T(220633, [J$.T(220625, {
                                    key: J$.T(220521, 'hasErrorLabel', 21, false),
                                    value: J$.T(220617, function hasErrorLabel(label) {
                                        jalangiLabel22:
                                            while (true) {
                                                try {
                                                    J$.Fe(220593, arguments.callee, this, arguments);
                                                    arguments = J$.N(220601, 'arguments', arguments, true, false, false);
                                                    label = J$.N(220609, 'label', label, true, false, false);
                                                    return J$.Rt(220585, J$.C(12952, J$.G(220537, J$.R(220529, 'this', this, false, false), 'errorLabels')) ? J$.B(14786, '!==', J$.M(220569, J$.G(220553, J$.R(220545, 'this', this, false, false), 'errorLabels'), 'indexOf', false)(J$.R(220561, 'label', label, false, false)), J$.U(14778, '-', J$.T(220577, 1, 22, false))) : J$._());
                                                } catch (J$e) {
                                                    J$.Ex(224257, J$e);
                                                } finally {
                                                    if (J$.Fr(224265))
                                                        continue jalangiLabel22;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false), J$.T(220721, [J$.T(220713, {
                                    key: J$.T(220641, 'create', 21, false),
                                    value: J$.T(220705, function create(options) {
                                        jalangiLabel23:
                                            while (true) {
                                                try {
                                                    J$.Fe(220681, arguments.callee, this, arguments);
                                                    arguments = J$.N(220689, 'arguments', arguments, true, false, false);
                                                    options = J$.N(220697, 'options', options, true, false, false);
                                                    return J$.Rt(220673, J$.F(220665, J$.R(220649, 'MongoError', MongoError, false, false), true)(J$.R(220657, 'options', options, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(224273, J$e);
                                                } finally {
                                                    if (J$.Fr(224281))
                                                        continue jalangiLabel23;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(220745, J$.R(220737, 'MongoError', MongoError, false, false));
                        } catch (J$e) {
                            J$.Ex(224289, J$e);
                        } finally {
                            if (J$.Fr(224297))
                                continue jalangiLabel24;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.F(220817, J$.R(220801, '_wrapNativeSuper', _wrapNativeSuper, false, true), false)(J$.I(typeof Error === 'undefined' ? Error = J$.R(220809, 'Error', undefined, true, true) : Error = J$.R(220809, 'Error', Error, true, true)))), MongoError, false, true);
            var MongoNetworkError = J$.W(221169, 'MongoNetworkError', J$.F(221161, J$.T(221145, function (_MongoError) {
                jalangiLabel26:
                    while (true) {
                        try {
                            J$.Fe(221105, arguments.callee, this, arguments);
                            function MongoNetworkError(message) {
                                jalangiLabel25:
                                    while (true) {
                                        try {
                                            J$.Fe(221057, arguments.callee, this, arguments);
                                            arguments = J$.N(221065, 'arguments', arguments, true, false, false);
                                            message = J$.N(221073, 'message', message, true, false, false);
                                            J$.N(221081, '_this2', _this2, false, false, false);
                                            var _this2;
                                            J$.F(220897, J$.R(220873, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(220881, 'this', this, false, false), J$.R(220889, 'MongoNetworkError', MongoNetworkError, false, false));
                                            _this2 = J$.W(220977, '_this2', J$.F(220969, J$.R(220905, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(220913, 'this', this, false, false), J$.M(220961, J$.F(220937, J$.R(220921, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(220929, 'MongoNetworkError', MongoNetworkError, false, false)), 'call', false)(J$.R(220945, 'this', this, false, false), J$.R(220953, 'message', message, false, false))), _this2, false, false);
                                            J$.P(221001, J$.R(220985, '_this2', _this2, false, false), 'name', J$.T(220993, 'MongoNetworkError', 21, false));
                                            J$.P(221033, J$.R(221009, '_this2', _this2, false, false), 'errorLabels', J$.T(221025, [J$.T(221017, 'TransientTransactionError', 21, false)], 10, false));
                                            return J$.Rt(221049, J$.R(221041, '_this2', _this2, false, false));
                                        } catch (J$e) {
                                            J$.Ex(224305, J$e);
                                        } finally {
                                            if (J$.Fr(224313))
                                                continue jalangiLabel25;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(221113, 'arguments', arguments, true, false, false);
                            _MongoError = J$.N(221121, '_MongoError', _MongoError, true, false, false);
                            MongoNetworkError = J$.N(221137, 'MongoNetworkError', J$.T(221129, MongoNetworkError, 12, false), true, false, false);
                            J$.F(220865, J$.R(220841, '_inherits', _inherits, false, true), false)(J$.R(220849, 'MongoNetworkError', MongoNetworkError, false, false), J$.R(220857, '_MongoError', _MongoError, false, false));
                            return J$.Rt(221097, J$.R(221089, 'MongoNetworkError', MongoNetworkError, false, false));
                        } catch (J$e) {
                            J$.Ex(224321, J$e);
                        } finally {
                            if (J$.Fr(224329))
                                continue jalangiLabel26;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(221153, 'MongoError', MongoError, false, true)), MongoNetworkError, false, true);
            var MongoParseError = J$.W(221473, 'MongoParseError', J$.F(221465, J$.T(221449, function (_MongoError2) {
                jalangiLabel28:
                    while (true) {
                        try {
                            J$.Fe(221409, arguments.callee, this, arguments);
                            function MongoParseError(message) {
                                jalangiLabel27:
                                    while (true) {
                                        try {
                                            J$.Fe(221361, arguments.callee, this, arguments);
                                            arguments = J$.N(221369, 'arguments', arguments, true, false, false);
                                            message = J$.N(221377, 'message', message, true, false, false);
                                            J$.N(221385, '_this3', _this3, false, false, false);
                                            var _this3;
                                            J$.F(221233, J$.R(221209, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(221217, 'this', this, false, false), J$.R(221225, 'MongoParseError', MongoParseError, false, false));
                                            _this3 = J$.W(221313, '_this3', J$.F(221305, J$.R(221241, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(221249, 'this', this, false, false), J$.M(221297, J$.F(221273, J$.R(221257, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(221265, 'MongoParseError', MongoParseError, false, false)), 'call', false)(J$.R(221281, 'this', this, false, false), J$.R(221289, 'message', message, false, false))), _this3, false, false);
                                            J$.P(221337, J$.R(221321, '_this3', _this3, false, false), 'name', J$.T(221329, 'MongoParseError', 21, false));
                                            return J$.Rt(221353, J$.R(221345, '_this3', _this3, false, false));
                                        } catch (J$e) {
                                            J$.Ex(224337, J$e);
                                        } finally {
                                            if (J$.Fr(224345))
                                                continue jalangiLabel27;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(221417, 'arguments', arguments, true, false, false);
                            _MongoError2 = J$.N(221425, '_MongoError2', _MongoError2, true, false, false);
                            MongoParseError = J$.N(221441, 'MongoParseError', J$.T(221433, MongoParseError, 12, false), true, false, false);
                            J$.F(221201, J$.R(221177, '_inherits', _inherits, false, true), false)(J$.R(221185, 'MongoParseError', MongoParseError, false, false), J$.R(221193, '_MongoError2', _MongoError2, false, false));
                            return J$.Rt(221401, J$.R(221393, 'MongoParseError', MongoParseError, false, false));
                        } catch (J$e) {
                            J$.Ex(224353, J$e);
                        } finally {
                            if (J$.Fr(224361))
                                continue jalangiLabel28;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(221457, 'MongoError', MongoError, false, true)), MongoParseError, false, true);
            var MongoTimeoutError = J$.W(221825, 'MongoTimeoutError', J$.F(221817, J$.T(221801, function (_MongoError3) {
                jalangiLabel30:
                    while (true) {
                        try {
                            J$.Fe(221761, arguments.callee, this, arguments);
                            function MongoTimeoutError(message, reason) {
                                jalangiLabel29:
                                    while (true) {
                                        try {
                                            J$.Fe(221705, arguments.callee, this, arguments);
                                            arguments = J$.N(221713, 'arguments', arguments, true, false, false);
                                            message = J$.N(221721, 'message', message, true, false, false);
                                            reason = J$.N(221729, 'reason', reason, true, false, false);
                                            J$.N(221737, '_this4', _this4, false, false, false);
                                            var _this4;
                                            J$.F(221537, J$.R(221513, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(221521, 'this', this, false, false), J$.R(221529, 'MongoTimeoutError', MongoTimeoutError, false, false));
                                            _this4 = J$.W(221617, '_this4', J$.F(221609, J$.R(221545, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(221553, 'this', this, false, false), J$.M(221601, J$.F(221577, J$.R(221561, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(221569, 'MongoTimeoutError', MongoTimeoutError, false, false)), 'call', false)(J$.R(221585, 'this', this, false, false), J$.R(221593, 'message', message, false, false))), _this4, false, false);
                                            J$.P(221641, J$.R(221625, '_this4', _this4, false, false), 'name', J$.T(221633, 'MongoTimeoutError', 21, false));
                                            if (J$.C(12960, J$.B(14794, '!=', J$.R(221649, 'reason', reason, false, false), J$.T(221657, null, 25, false)))) {
                                                J$.P(221681, J$.R(221665, '_this4', _this4, false, false), 'reason', J$.R(221673, 'reason', reason, false, false));
                                            }
                                            return J$.Rt(221697, J$.R(221689, '_this4', _this4, false, false));
                                        } catch (J$e) {
                                            J$.Ex(224369, J$e);
                                        } finally {
                                            if (J$.Fr(224377))
                                                continue jalangiLabel29;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(221769, 'arguments', arguments, true, false, false);
                            _MongoError3 = J$.N(221777, '_MongoError3', _MongoError3, true, false, false);
                            MongoTimeoutError = J$.N(221793, 'MongoTimeoutError', J$.T(221785, MongoTimeoutError, 12, false), true, false, false);
                            J$.F(221505, J$.R(221481, '_inherits', _inherits, false, true), false)(J$.R(221489, 'MongoTimeoutError', MongoTimeoutError, false, false), J$.R(221497, '_MongoError3', _MongoError3, false, false));
                            return J$.Rt(221753, J$.R(221745, 'MongoTimeoutError', MongoTimeoutError, false, false));
                        } catch (J$e) {
                            J$.Ex(224385, J$e);
                        } finally {
                            if (J$.Fr(224393))
                                continue jalangiLabel30;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(221809, 'MongoError', MongoError, false, true)), MongoTimeoutError, false, true);
            var MongoWriteConcernError = J$.W(222353, 'MongoWriteConcernError', J$.F(222345, J$.T(222329, function (_MongoError4) {
                jalangiLabel33:
                    while (true) {
                        try {
                            J$.Fe(222289, arguments.callee, this, arguments);
                            function MongoWriteConcernError(message, result) {
                                jalangiLabel32:
                                    while (true) {
                                        try {
                                            J$.Fe(222233, arguments.callee, this, arguments);
                                            arguments = J$.N(222241, 'arguments', arguments, true, false, false);
                                            message = J$.N(222249, 'message', message, true, false, false);
                                            result = J$.N(222257, 'result', result, true, false, false);
                                            J$.N(222265, '_this5', _this5, false, false, false);
                                            var _this5;
                                            J$.F(222049, J$.R(222025, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(222033, 'this', this, false, false), J$.R(222041, 'MongoWriteConcernError', MongoWriteConcernError, false, false));
                                            _this5 = J$.W(222129, '_this5', J$.F(222121, J$.R(222057, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(222065, 'this', this, false, false), J$.M(222113, J$.F(222089, J$.R(222073, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(222081, 'MongoWriteConcernError', MongoWriteConcernError, false, false)), 'call', false)(J$.R(222097, 'this', this, false, false), J$.R(222105, 'message', message, false, false))), _this5, false, false);
                                            J$.P(222153, J$.R(222137, '_this5', _this5, false, false), 'name', J$.T(222145, 'MongoWriteConcernError', 21, false));
                                            if (J$.C(12976, J$.B(14834, '!=', J$.R(222161, 'result', result, false, false), J$.T(222169, null, 25, false)))) {
                                                J$.P(222209, J$.R(222177, '_this5', _this5, false, false), 'result', J$.F(222201, J$.R(222185, 'makeWriteConcernResultObject', makeWriteConcernResultObject, false, true), false)(J$.R(222193, 'result', result, false, false)));
                                            }
                                            return J$.Rt(222225, J$.R(222217, '_this5', _this5, false, false));
                                        } catch (J$e) {
                                            J$.Ex(224417, J$e);
                                        } finally {
                                            if (J$.Fr(224425))
                                                continue jalangiLabel32;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(222297, 'arguments', arguments, true, false, false);
                            _MongoError4 = J$.N(222305, '_MongoError4', _MongoError4, true, false, false);
                            MongoWriteConcernError = J$.N(222321, 'MongoWriteConcernError', J$.T(222313, MongoWriteConcernError, 12, false), true, false, false);
                            J$.F(222017, J$.R(221993, '_inherits', _inherits, false, true), false)(J$.R(222001, 'MongoWriteConcernError', MongoWriteConcernError, false, false), J$.R(222009, '_MongoError4', _MongoError4, false, false));
                            return J$.Rt(222281, J$.R(222273, 'MongoWriteConcernError', MongoWriteConcernError, false, false));
                        } catch (J$e) {
                            J$.Ex(224433, J$e);
                        } finally {
                            if (J$.Fr(224441))
                                continue jalangiLabel33;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(222337, 'MongoError', MongoError, false, true)), MongoWriteConcernError, false, true);
            var RETRYABLE_ERROR_CODES = J$.W(222473, 'RETRYABLE_ERROR_CODES', J$.F(222465, J$.I(typeof Set === 'undefined' ? Set = J$.R(222361, 'Set', undefined, true, true) : Set = J$.R(222361, 'Set', Set, true, true)), true)(J$.T(222457, [
                J$.T(222369, 6, 22, false),
                J$.T(222377, 7, 22, false),
                J$.T(222385, 89, 22, false),
                J$.T(222393, 91, 22, false),
                J$.T(222401, 189, 22, false),
                J$.T(222409, 9001, 22, false),
                J$.T(222417, 10107, 22, false),
                J$.T(222425, 11600, 22, false),
                J$.T(222433, 11602, 22, false),
                J$.T(222441, 13435, 22, false),
                J$.T(222449, 13436, 22, false)
            ], 10, false)), RETRYABLE_ERROR_CODES, false, true);
            var SDAM_RECOVERING_CODES = J$.W(222689, 'SDAM_RECOVERING_CODES', J$.F(222681, J$.I(typeof Set === 'undefined' ? Set = J$.R(222625, 'Set', undefined, true, true) : Set = J$.R(222625, 'Set', Set, true, true)), true)(J$.T(222673, [
                J$.T(222633, 91, 22, false),
                J$.T(222641, 189, 22, false),
                J$.T(222649, 11600, 22, false),
                J$.T(222657, 11602, 22, false),
                J$.T(222665, 13436, 22, false)
            ], 10, false)), SDAM_RECOVERING_CODES, false, true);
            var SDAM_NOTMASTER_CODES = J$.W(222737, 'SDAM_NOTMASTER_CODES', J$.F(222729, J$.I(typeof Set === 'undefined' ? Set = J$.R(222697, 'Set', undefined, true, true) : Set = J$.R(222697, 'Set', Set, true, true)), true)(J$.T(222721, [
                J$.T(222705, 10107, 22, false),
                J$.T(222713, 13435, 22, false)
            ], 10, false)), SDAM_NOTMASTER_CODES, false, true);
            var SDAM_NODE_SHUTTING_DOWN_ERROR_CODES = J$.W(222785, 'SDAM_NODE_SHUTTING_DOWN_ERROR_CODES', J$.F(222777, J$.I(typeof Set === 'undefined' ? Set = J$.R(222745, 'Set', undefined, true, true) : Set = J$.R(222745, 'Set', Set, true, true)), true)(J$.T(222769, [
                J$.T(222753, 11600, 22, false),
                J$.T(222761, 91, 22, false)
            ], 10, false)), SDAM_NODE_SHUTTING_DOWN_ERROR_CODES, false, true);
            J$.P(223497, J$.I(typeof module === 'undefined' ? module = J$.R(223417, 'module', undefined, true, true) : module = J$.R(223417, 'module', module, true, true)), 'exports', J$.T(223489, {
                MongoError: J$.R(223425, 'MongoError', MongoError, false, true),
                MongoNetworkError: J$.R(223433, 'MongoNetworkError', MongoNetworkError, false, true),
                MongoParseError: J$.R(223441, 'MongoParseError', MongoParseError, false, true),
                MongoTimeoutError: J$.R(223449, 'MongoTimeoutError', MongoTimeoutError, false, true),
                MongoWriteConcernError: J$.R(223457, 'MongoWriteConcernError', MongoWriteConcernError, false, true),
                mongoErrorContextSymbol: J$.R(223465, 'mongoErrorContextSymbol', mongoErrorContextSymbol, false, true),
                isRetryableError: J$.R(223473, 'isRetryableError', isRetryableError, false, true),
                isSDAMUnrecoverableError: J$.R(223481, 'isSDAMUnrecoverableError', isSDAMUnrecoverableError, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(224529, J$e);
        } finally {
            if (J$.Sr(224537))
                continue jalangiLabel39;
            else
                break jalangiLabel39;
        }
    }
// JALANGI DO NOT INSTRUMENT
