// mongodb/lib/core/sessions.js
J$.noInstrEval = false;
jalangiLabel61:
    while (true) {
        try {
            J$.Se(299489, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/core/sessions.js');
            function _defineProperty(obj, key, value) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(286617, arguments.callee, this, arguments);
                            arguments = J$.N(286625, 'arguments', arguments, true, false, false);
                            obj = J$.N(286633, 'obj', obj, true, false, false);
                            key = J$.N(286641, 'key', key, true, false, false);
                            value = J$.N(286649, 'value', value, true, false, false);
                            if (J$.C(16496, J$.B(17682, 'in', J$.R(286481, 'key', key, false, false), J$.R(286489, 'obj', obj, false, false)))) {
                                J$.M(286561, J$.I(typeof Object === 'undefined' ? Object = J$.R(286497, 'Object', undefined, true, true) : Object = J$.R(286497, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(286505, 'obj', obj, false, false), J$.R(286513, 'key', key, false, false), J$.T(286553, {
                                    value: J$.R(286521, 'value', value, false, false),
                                    enumerable: J$.T(286529, true, 23, false),
                                    configurable: J$.T(286537, true, 23, false),
                                    writable: J$.T(286545, true, 23, false)
                                }, 11, false));
                            } else {
                                J$.P(286593, J$.R(286569, 'obj', obj, false, false), J$.R(286577, 'key', key, false, false), J$.R(286585, 'value', value, false, false));
                            }
                            return J$.Rt(286609, J$.R(286601, 'obj', obj, false, false));
                        } catch (J$e) {
                            J$.Ex(300065, J$e);
                        } finally {
                            if (J$.Fr(300073))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(286921, arguments.callee, this, arguments);
                            arguments = J$.N(286929, 'arguments', arguments, true, false, false);
                            obj = J$.N(286937, 'obj', obj, true, false, false);
                            if (J$.C(16544, J$.C(16504, J$.B(17698, '===', J$.U(17690, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(286657, 'Symbol', undefined, true, true) : Symbol = J$.R(286657, 'Symbol', Symbol, true, true))), J$.T(286665, 'function', 21, false))) ? J$.B(17714, '===', J$.U(17706, 'typeof', J$.G(286681, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(286673, 'Symbol', undefined, true, true) : Symbol = J$.R(286673, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(286689, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(286745, '_typeof', J$.T(286737, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(286713, arguments.callee, this, arguments);
                                                arguments = J$.N(286721, 'arguments', arguments, true, false, false);
                                                obj = J$.N(286729, 'obj', obj, true, false, false);
                                                return J$.Rt(286705, J$.U(17722, 'typeof', J$.R(286697, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(300081, J$e);
                                            } finally {
                                                if (J$.Fr(300089))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(286881, '_typeof', J$.T(286873, function _typeof(obj) {
                                    jalangiLabel2:
                                        while (true) {
                                            try {
                                                J$.Fe(286849, arguments.callee, this, arguments);
                                                arguments = J$.N(286857, 'arguments', arguments, true, false, false);
                                                obj = J$.N(286865, 'obj', obj, true, false, false);
                                                return J$.Rt(286841, J$.C(16536, J$.C(16528, J$.C(16520, J$.C(16512, J$.R(286753, 'obj', obj, false, false)) ? J$.B(17738, '===', J$.U(17730, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(286761, 'Symbol', undefined, true, true) : Symbol = J$.R(286761, 'Symbol', Symbol, true, true))), J$.T(286769, 'function', 21, false)) : J$._()) ? J$.B(17746, '===', J$.G(286785, J$.R(286777, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(286793, 'Symbol', undefined, true, true) : Symbol = J$.R(286793, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(17754, '!==', J$.R(286801, 'obj', obj, false, false), J$.G(286817, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(286809, 'Symbol', undefined, true, true) : Symbol = J$.R(286809, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(286825, 'symbol', 21, false) : J$.U(17762, 'typeof', J$.R(286833, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(300097, J$e);
                                            } finally {
                                                if (J$.Fr(300105))
                                                    continue jalangiLabel2;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(286913, J$.F(286905, J$.R(286889, '_typeof', _typeof, false, true), false)(J$.R(286897, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(300113, J$e);
                        } finally {
                            if (J$.Fr(300121))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(286985, arguments.callee, this, arguments);
                            arguments = J$.N(286993, 'arguments', arguments, true, false, false);
                            instance = J$.N(287001, 'instance', instance, true, false, false);
                            Constructor = J$.N(287009, 'Constructor', Constructor, true, false, false);
                            if (J$.C(16552, J$.U(17778, '!', J$.B(17770, 'instanceof', J$.R(286945, 'instance', instance, false, false), J$.R(286953, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(286977, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(286961, 'TypeError', undefined, true, true) : TypeError = J$.R(286961, 'TypeError', TypeError, true, true)), true)(J$.T(286969, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(300129, J$e);
                        } finally {
                            if (J$.Fr(300137))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(287257, arguments.callee, this, arguments);
                            arguments = J$.N(287265, 'arguments', arguments, true, false, false);
                            target = J$.N(287273, 'target', target, true, false, false);
                            props = J$.N(287281, 'props', props, true, false, false);
                            J$.N(287289, 'i', i, false, false, false);
                            J$.N(287297, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(287025, 'i', J$.T(287017, 0, 22, false), i, false, false); J$.C(16576, J$.B(17786, '<', J$.R(287033, 'i', i, false, false), J$.G(287049, J$.R(287041, 'props', props, false, false), 'length'))); J$.B(17810, '-', i = J$.W(287065, 'i', J$.B(17802, '+', J$.U(17794, '+', J$.R(287057, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(287097, 'descriptor', J$.G(287089, J$.R(287073, 'props', props, false, false), J$.R(287081, 'i', i, false, false)), descriptor, false, false);
                                J$.P(287137, J$.R(287105, 'descriptor', descriptor, false, false), 'enumerable', J$.C(16560, J$.G(287121, J$.R(287113, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(287129, false, 23, false));
                                J$.P(287161, J$.R(287145, 'descriptor', descriptor, false, false), 'configurable', J$.T(287153, true, 23, false));
                                if (J$.C(16568, J$.B(17818, 'in', J$.T(287169, 'value', 21, false), J$.R(287177, 'descriptor', descriptor, false, false))))
                                    J$.P(287201, J$.R(287185, 'descriptor', descriptor, false, false), 'writable', J$.T(287193, true, 23, false));
                                J$.M(287249, J$.I(typeof Object === 'undefined' ? Object = J$.R(287209, 'Object', undefined, true, true) : Object = J$.R(287209, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(287217, 'target', target, false, false), J$.G(287233, J$.R(287225, 'descriptor', descriptor, false, false), 'key'), J$.R(287241, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(300145, J$e);
                        } finally {
                            if (J$.Fr(300153))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(287409, arguments.callee, this, arguments);
                            arguments = J$.N(287417, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(287425, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(287433, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(287441, 'staticProps', staticProps, true, false, false);
                            if (J$.C(16584, J$.R(287305, 'protoProps', protoProps, false, false)))
                                J$.F(287345, J$.R(287313, '_defineProperties', _defineProperties, false, true), false)(J$.G(287329, J$.R(287321, 'Constructor', Constructor, false, false), 'prototype'), J$.R(287337, 'protoProps', protoProps, false, false));
                            if (J$.C(16592, J$.R(287353, 'staticProps', staticProps, false, false)))
                                J$.F(287385, J$.R(287361, '_defineProperties', _defineProperties, false, true), false)(J$.R(287369, 'Constructor', Constructor, false, false), J$.R(287377, 'staticProps', staticProps, false, false));
                            return J$.Rt(287401, J$.R(287393, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(300161, J$e);
                        } finally {
                            if (J$.Fr(300169))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(287553, arguments.callee, this, arguments);
                            arguments = J$.N(287561, 'arguments', arguments, true, false, false);
                            self = J$.N(287569, 'self', self, true, false, false);
                            call = J$.N(287577, 'call', call, true, false, false);
                            if (J$.C(16616, J$.C(16608, J$.R(287449, 'call', call, false, false)) ? J$.C(16600, J$.B(17826, '===', J$.F(287473, J$.R(287457, '_typeof', _typeof, false, true), false)(J$.R(287465, 'call', call, false, false)), J$.T(287481, 'object', 21, false))) ? J$._() : J$.B(17842, '===', J$.U(17834, 'typeof', J$.R(287489, 'call', call, false, false)), J$.T(287497, 'function', 21, false)) : J$._())) {
                                return J$.Rt(287513, J$.R(287505, 'call', call, false, false));
                            }
                            return J$.Rt(287545, J$.F(287537, J$.R(287521, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(287529, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(300177, J$e);
                        } finally {
                            if (J$.Fr(300185))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(287641, arguments.callee, this, arguments);
                            arguments = J$.N(287649, 'arguments', arguments, true, false, false);
                            self = J$.N(287657, 'self', self, true, false, false);
                            if (J$.C(16624, J$.B(17850, '===', J$.R(287585, 'self', self, false, false), void J$.T(287593, 0, 22, false)))) {
                                throw J$.F(287617, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(287601, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(287601, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(287609, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(287633, J$.R(287625, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(300193, J$e);
                        } finally {
                            if (J$.Fr(300201))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(287817, arguments.callee, this, arguments);
                            arguments = J$.N(287825, 'arguments', arguments, true, false, false);
                            o = J$.N(287833, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(287777, '_getPrototypeOf', J$.C(16640, J$.G(287673, J$.I(typeof Object === 'undefined' ? Object = J$.R(287665, 'Object', undefined, true, true) : Object = J$.R(287665, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(287689, J$.I(typeof Object === 'undefined' ? Object = J$.R(287681, 'Object', undefined, true, true) : Object = J$.R(287681, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(287769, function _getPrototypeOf(o) {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(287745, arguments.callee, this, arguments);
                                            arguments = J$.N(287753, 'arguments', arguments, true, false, false);
                                            o = J$.N(287761, 'o', o, true, false, false);
                                            return J$.Rt(287737, J$.C(16632, J$.G(287705, J$.R(287697, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(287729, J$.I(typeof Object === 'undefined' ? Object = J$.R(287713, 'Object', undefined, true, true) : Object = J$.R(287713, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(287721, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(300209, J$e);
                                        } finally {
                                            if (J$.Fr(300217))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(287809, J$.F(287801, J$.R(287785, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(287793, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(300225, J$e);
                        } finally {
                            if (J$.Fr(300233))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(288033, arguments.callee, this, arguments);
                            arguments = J$.N(288041, 'arguments', arguments, true, false, false);
                            subClass = J$.N(288049, 'subClass', subClass, true, false, false);
                            superClass = J$.N(288057, 'superClass', superClass, true, false, false);
                            if (J$.C(16656, J$.C(16648, J$.B(17866, '!==', J$.U(17858, 'typeof', J$.R(287841, 'superClass', superClass, false, false)), J$.T(287849, 'function', 21, false))) ? J$.B(17874, '!==', J$.R(287857, 'superClass', superClass, false, false), J$.T(287865, null, 25, false)) : J$._())) {
                                throw J$.F(287889, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(287873, 'TypeError', undefined, true, true) : TypeError = J$.R(287873, 'TypeError', TypeError, true, true)), true)(J$.T(287881, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(287985, J$.R(287897, 'subClass', subClass, false, false), 'prototype', J$.M(287977, J$.I(typeof Object === 'undefined' ? Object = J$.R(287905, 'Object', undefined, true, true) : Object = J$.R(287905, 'Object', Object, true, true)), 'create', false)(J$.C(16664, J$.R(287913, 'superClass', superClass, false, false)) ? J$.G(287929, J$.R(287921, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(287969, {
                                constructor: J$.T(287961, {
                                    value: J$.R(287937, 'subClass', subClass, false, false),
                                    writable: J$.T(287945, true, 23, false),
                                    configurable: J$.T(287953, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(16672, J$.R(287993, 'superClass', superClass, false, false)))
                                J$.F(288025, J$.R(288001, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(288009, 'subClass', subClass, false, false), J$.R(288017, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(300241, J$e);
                        } finally {
                            if (J$.Fr(300249))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(288209, arguments.callee, this, arguments);
                            arguments = J$.N(288217, 'arguments', arguments, true, false, false);
                            o = J$.N(288225, 'o', o, true, false, false);
                            p = J$.N(288233, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(288161, '_setPrototypeOf', J$.C(16680, J$.G(288073, J$.I(typeof Object === 'undefined' ? Object = J$.R(288065, 'Object', undefined, true, true) : Object = J$.R(288065, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(288153, function _setPrototypeOf(o, p) {
                                jalangiLabel12:
                                    while (true) {
                                        try {
                                            J$.Fe(288121, arguments.callee, this, arguments);
                                            arguments = J$.N(288129, 'arguments', arguments, true, false, false);
                                            o = J$.N(288137, 'o', o, true, false, false);
                                            p = J$.N(288145, 'p', p, true, false, false);
                                            J$.P(288097, J$.R(288081, 'o', o, false, false), '__proto__', J$.R(288089, 'p', p, false, false));
                                            return J$.Rt(288113, J$.R(288105, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(300257, J$e);
                                        } finally {
                                            if (J$.Fr(300265))
                                                continue jalangiLabel12;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(288201, J$.F(288193, J$.R(288169, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(288177, 'o', o, false, false), J$.R(288185, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(300273, J$e);
                        } finally {
                            if (J$.Fr(300281))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function assertAlive(session, callback) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(289033, arguments.callee, this, arguments);
                            arguments = J$.N(289041, 'arguments', arguments, true, false, false);
                            session = J$.N(289049, 'session', session, true, false, false);
                            callback = J$.N(289057, 'callback', callback, true, false, false);
                            J$.N(289065, 'error', error, false, false, false);
                            if (J$.C(16696, J$.B(17882, '==', J$.G(288897, J$.R(288889, 'session', session, false, false), 'serverSession'), J$.T(288905, null, 25, false)))) {
                                var error = J$.W(288937, 'error', J$.F(288929, J$.R(288913, 'MongoError', MongoError, false, true), true)(J$.T(288921, 'Cannot use a session that has ended', 21, false)), error, false, false);
                                if (J$.C(16688, J$.B(17898, '===', J$.U(17890, 'typeof', J$.R(288945, 'callback', callback, false, false)), J$.T(288953, 'function', 21, false)))) {
                                    J$.F(288985, J$.R(288961, 'callback', callback, false, false), false)(J$.R(288969, 'error', error, false, false), J$.T(288977, null, 25, false));
                                    return J$.Rt(289001, J$.T(288993, false, 23, false));
                                }
                                throw J$.R(289009, 'error', error, false, false);
                            }
                            return J$.Rt(289025, J$.T(289017, true, 23, false));
                        } catch (J$e) {
                            J$.Ex(300289, J$e);
                        } finally {
                            if (J$.Fr(300297))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function hasNotTimedOut(startTime, max) {
                jalangiLabel32:
                    while (true) {
                        try {
                            J$.Fe(292337, arguments.callee, this, arguments);
                            arguments = J$.N(292345, 'arguments', arguments, true, false, false);
                            startTime = J$.N(292353, 'startTime', startTime, true, false, false);
                            max = J$.N(292361, 'max', max, true, false, false);
                            return J$.Rt(292329, J$.B(18106, '<', J$.B(18098, '-', J$.M(292305, J$.I(typeof Date === 'undefined' ? Date = J$.R(292297, 'Date', undefined, true, true) : Date = J$.R(292297, 'Date', Date, true, true)), 'now', false)(), J$.R(292313, 'startTime', startTime, false, false)), J$.R(292321, 'max', max, false, false)));
                        } catch (J$e) {
                            J$.Ex(300577, J$e);
                        } finally {
                            if (J$.Fr(300585))
                                continue jalangiLabel32;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isUnknownTransactionCommitResult(err) {
                jalangiLabel33:
                    while (true) {
                        try {
                            J$.Fe(292481, arguments.callee, this, arguments);
                            arguments = J$.N(292489, 'arguments', arguments, true, false, false);
                            err = J$.N(292497, 'err', err, true, false, false);
                            return J$.Rt(292473, J$.C(16920, J$.F(292385, J$.R(292369, 'isMaxTimeMSExpiredError', isMaxTimeMSExpiredError, false, true), false)(J$.R(292377, 'err', err, false, false))) ? J$._() : J$.C(16912, J$.C(16904, J$.U(18114, '!', J$.M(292417, J$.R(292393, 'NON_DETERMINISTIC_WRITE_CONCERN_ERRORS', NON_DETERMINISTIC_WRITE_CONCERN_ERRORS, false, true), 'has', false)(J$.G(292409, J$.R(292401, 'err', err, false, false), 'codeName')))) ? J$.B(18122, '!==', J$.G(292433, J$.R(292425, 'err', err, false, false), 'code'), J$.R(292441, 'UNSATISFIABLE_WRITE_CONCERN_CODE', UNSATISFIABLE_WRITE_CONCERN_CODE, false, true)) : J$._()) ? J$.B(18130, '!==', J$.G(292457, J$.R(292449, 'err', err, false, false), 'code'), J$.R(292465, 'UNKNOWN_REPL_WRITE_CONCERN_CODE', UNKNOWN_REPL_WRITE_CONCERN_CODE, false, true)) : J$._());
                        } catch (J$e) {
                            J$.Ex(300593, J$e);
                        } finally {
                            if (J$.Fr(300601))
                                continue jalangiLabel33;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isMaxTimeMSExpiredError(err) {
                jalangiLabel34:
                    while (true) {
                        try {
                            J$.Fe(292585, arguments.callee, this, arguments);
                            arguments = J$.N(292593, 'arguments', arguments, true, false, false);
                            err = J$.N(292601, 'err', err, true, false, false);
                            return J$.Rt(292577, J$.C(16936, J$.B(18138, '===', J$.G(292513, J$.R(292505, 'err', err, false, false), 'code'), J$.R(292521, 'MAX_TIME_MS_EXPIRED_CODE', MAX_TIME_MS_EXPIRED_CODE, false, true))) ? J$._() : J$.C(16928, J$.G(292537, J$.R(292529, 'err', err, false, false), 'writeConcernError')) ? J$.B(18146, '===', J$.G(292561, J$.G(292553, J$.R(292545, 'err', err, false, false), 'writeConcernError'), 'code'), J$.R(292569, 'MAX_TIME_MS_EXPIRED_CODE', MAX_TIME_MS_EXPIRED_CODE, false, true)) : J$._());
                        } catch (J$e) {
                            J$.Ex(300609, J$e);
                        } finally {
                            if (J$.Fr(300617))
                                continue jalangiLabel34;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function attemptTransactionCommit(session, startTime, fn, options) {
                jalangiLabel36:
                    while (true) {
                        try {
                            J$.Fe(292921, arguments.callee, this, arguments);
                            arguments = J$.N(292929, 'arguments', arguments, true, false, false);
                            session = J$.N(292937, 'session', session, true, false, false);
                            startTime = J$.N(292945, 'startTime', startTime, true, false, false);
                            fn = J$.N(292953, 'fn', fn, true, false, false);
                            options = J$.N(292961, 'options', options, true, false, false);
                            return J$.Rt(292913, J$.M(292905, J$.M(292617, J$.R(292609, 'session', session, false, false), 'commitTransaction', false)(), J$.T(292625, 'catch', 21, false), false)(J$.T(292897, function (err) {
                                jalangiLabel35:
                                    while (true) {
                                        try {
                                            J$.Fe(292873, arguments.callee, this, arguments);
                                            arguments = J$.N(292881, 'arguments', arguments, true, false, false);
                                            err = J$.N(292889, 'err', err, true, false, false);
                                            if (J$.C(16976, J$.C(16952, J$.C(16944, J$.B(18154, 'instanceof', J$.R(292633, 'err', err, false, false), J$.R(292641, 'MongoError', MongoError, false, true))) ? J$.F(292673, J$.R(292649, 'hasNotTimedOut', hasNotTimedOut, false, true), false)(J$.R(292657, 'startTime', startTime, false, false), J$.R(292665, 'MAX_WITH_TRANSACTION_TIMEOUT', MAX_WITH_TRANSACTION_TIMEOUT, false, true)) : J$._()) ? J$.U(18162, '!', J$.F(292697, J$.R(292681, 'isMaxTimeMSExpiredError', isMaxTimeMSExpiredError, false, true), false)(J$.R(292689, 'err', err, false, false))) : J$._())) {
                                                if (J$.C(16960, J$.M(292721, J$.R(292705, 'err', err, false, false), 'hasErrorLabel', false)(J$.T(292713, 'UnknownTransactionCommitResult', 21, false)))) {
                                                    return J$.Rt(292777, J$.F(292769, J$.R(292729, 'attemptTransactionCommit', attemptTransactionCommit, false, true), false)(J$.R(292737, 'session', session, false, false), J$.R(292745, 'startTime', startTime, false, false), J$.R(292753, 'fn', fn, false, false), J$.R(292761, 'options', options, false, false)));
                                                }
                                                if (J$.C(16968, J$.M(292801, J$.R(292785, 'err', err, false, false), 'hasErrorLabel', false)(J$.T(292793, 'TransientTransactionError', 21, false)))) {
                                                    return J$.Rt(292857, J$.F(292849, J$.R(292809, 'attemptTransaction', attemptTransaction, false, true), false)(J$.R(292817, 'session', session, false, false), J$.R(292825, 'startTime', startTime, false, false), J$.R(292833, 'fn', fn, false, false), J$.R(292841, 'options', options, false, false)));
                                                }
                                            }
                                            throw J$.R(292865, 'err', err, false, false);
                                        } catch (J$e) {
                                            J$.Ex(300625, J$e);
                                        } finally {
                                            if (J$.Fr(300633))
                                                continue jalangiLabel35;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(300641, J$e);
                        } finally {
                            if (J$.Fr(300649))
                                continue jalangiLabel36;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function userExplicitlyEndedTransaction(session) {
                jalangiLabel37:
                    while (true) {
                        try {
                            J$.Fe(293097, arguments.callee, this, arguments);
                            arguments = J$.N(293105, 'arguments', arguments, true, false, false);
                            session = J$.N(293113, 'session', session, true, false, false);
                            return J$.Rt(293089, J$.M(293081, J$.R(293049, 'USER_EXPLICIT_TXN_END_STATES', USER_EXPLICIT_TXN_END_STATES, false, true), 'has', false)(J$.G(293073, J$.G(293065, J$.R(293057, 'session', session, false, false), 'transaction'), 'state')));
                        } catch (J$e) {
                            J$.Ex(300657, J$e);
                        } finally {
                            if (J$.Fr(300665))
                                continue jalangiLabel37;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function attemptTransaction(session, startTime, fn, options) {
                jalangiLabel42:
                    while (true) {
                        try {
                            J$.Fe(293889, arguments.callee, this, arguments);
                            arguments = J$.N(293897, 'arguments', arguments, true, false, false);
                            session = J$.N(293905, 'session', session, true, false, false);
                            startTime = J$.N(293913, 'startTime', startTime, true, false, false);
                            fn = J$.N(293921, 'fn', fn, true, false, false);
                            options = J$.N(293929, 'options', options, true, false, false);
                            J$.N(293937, 'promise', promise, false, false, false);
                            J$.M(293137, J$.R(293121, 'session', session, false, false), 'startTransaction', false)(J$.R(293129, 'options', options, false, false));
                            var promise;
                            try {
                                promise = J$.W(293169, 'promise', J$.F(293161, J$.R(293145, 'fn', fn, false, false), false)(J$.R(293153, 'session', session, false, false)), promise, false, false);
                            } catch (err) {
                                J$.N(293209, 'err', err, false, false, true);
                                promise = J$.W(293201, 'promise', J$.M(293193, J$.I(typeof Promise === 'undefined' ? Promise = J$.R(293177, 'Promise', undefined, true, true) : Promise = J$.R(293177, 'Promise', Promise, true, true)), 'reject', false)(J$.R(293185, 'err', err, false, false)), promise, false, false);
                            }
                            if (J$.C(16984, J$.U(18170, '!', J$.F(293233, J$.R(293217, 'isPromiseLike', isPromiseLike, false, true), false)(J$.R(293225, 'promise', promise, false, false))))) {
                                J$.M(293249, J$.R(293241, 'session', session, false, false), 'abortTransaction', false)();
                                throw J$.F(293273, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(293257, 'TypeError', undefined, true, true) : TypeError = J$.R(293257, 'TypeError', TypeError, true, true)), true)(J$.T(293265, 'Function provided to `withTransaction` must return a Promise', 21, false));
                            }
                            return J$.Rt(293881, J$.M(293873, J$.M(293401, J$.R(293281, 'promise', promise, false, false), 'then', false)(J$.T(293393, function () {
                                jalangiLabel38:
                                    while (true) {
                                        try {
                                            J$.Fe(293377, arguments.callee, this, arguments);
                                            arguments = J$.N(293385, 'arguments', arguments, true, false, false);
                                            if (J$.C(16992, J$.F(293305, J$.R(293289, 'userExplicitlyEndedTransaction', userExplicitlyEndedTransaction, false, true), false)(J$.R(293297, 'session', session, false, false)))) {
                                                return J$.Rt(293313, undefined);
                                            }
                                            return J$.Rt(293369, J$.F(293361, J$.R(293321, 'attemptTransactionCommit', attemptTransactionCommit, false, true), false)(J$.R(293329, 'session', session, false, false), J$.R(293337, 'startTime', startTime, false, false), J$.R(293345, 'fn', fn, false, false), J$.R(293353, 'options', options, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(300673, J$e);
                                        } finally {
                                            if (J$.Fr(300681))
                                                continue jalangiLabel38;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), J$.T(293409, 'catch', 21, false), false)(J$.T(293865, function (err) {
                                jalangiLabel41:
                                    while (true) {
                                        try {
                                            J$.Fe(293825, arguments.callee, this, arguments);
                                            function maybeRetryOrThrow(err) {
                                                jalangiLabel39:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(293657, arguments.callee, this, arguments);
                                                            arguments = J$.N(293665, 'arguments', arguments, true, false, false);
                                                            err = J$.N(293673, 'err', err, true, false, false);
                                                            if (J$.C(17016, J$.C(17008, J$.C(17000, J$.B(18178, 'instanceof', J$.R(293417, 'err', err, false, false), J$.R(293425, 'MongoError', MongoError, false, true))) ? J$.M(293449, J$.R(293433, 'err', err, false, false), 'hasErrorLabel', false)(J$.T(293441, 'TransientTransactionError', 21, false)) : J$._()) ? J$.F(293481, J$.R(293457, 'hasNotTimedOut', hasNotTimedOut, false, true), false)(J$.R(293465, 'startTime', startTime, false, false), J$.R(293473, 'MAX_WITH_TRANSACTION_TIMEOUT', MAX_WITH_TRANSACTION_TIMEOUT, false, true)) : J$._())) {
                                                                return J$.Rt(293537, J$.F(293529, J$.R(293489, 'attemptTransaction', attemptTransaction, false, true), false)(J$.R(293497, 'session', session, false, false), J$.R(293505, 'startTime', startTime, false, false), J$.R(293513, 'fn', fn, false, false), J$.R(293521, 'options', options, false, false)));
                                                            }
                                                            if (J$.C(17032, J$.F(293561, J$.R(293545, 'isMaxTimeMSExpiredError', isMaxTimeMSExpiredError, false, true), false)(J$.R(293553, 'err', err, false, false)))) {
                                                                if (J$.C(17024, J$.B(18186, '==', J$.G(293577, J$.R(293569, 'err', err, false, false), 'errorLabels'), J$.T(293585, null, 25, false)))) {
                                                                    J$.P(293609, J$.R(293593, 'err', err, false, false), 'errorLabels', J$.T(293601, [], 10, false));
                                                                }
                                                                J$.M(293641, J$.G(293625, J$.R(293617, 'err', err, false, false), 'errorLabels'), 'push', false)(J$.T(293633, 'UnknownTransactionCommitResult', 21, false));
                                                            }
                                                            throw J$.R(293649, 'err', err, false, false);
                                                        } catch (J$e) {
                                                            J$.Ex(300689, J$e);
                                                        } finally {
                                                            if (J$.Fr(300697))
                                                                continue jalangiLabel39;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }
                                            arguments = J$.N(293833, 'arguments', arguments, true, false, false);
                                            err = J$.N(293841, 'err', err, true, false, false);
                                            maybeRetryOrThrow = J$.N(293857, 'maybeRetryOrThrow', J$.T(293849, maybeRetryOrThrow, 12, false), true, false, false);
                                            if (J$.C(17040, J$.G(293697, J$.G(293689, J$.R(293681, 'session', session, false, false), 'transaction'), 'isActive'))) {
                                                return J$.Rt(293785, J$.M(293777, J$.M(293713, J$.R(293705, 'session', session, false, false), 'abortTransaction', false)(), 'then', false)(J$.T(293769, function () {
                                                    jalangiLabel40:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(293753, arguments.callee, this, arguments);
                                                                arguments = J$.N(293761, 'arguments', arguments, true, false, false);
                                                                return J$.Rt(293745, J$.F(293737, J$.R(293721, 'maybeRetryOrThrow', maybeRetryOrThrow, false, false), false)(J$.R(293729, 'err', err, false, false)));
                                                            } catch (J$e) {
                                                                J$.Ex(300705, J$e);
                                                            } finally {
                                                                if (J$.Fr(300713))
                                                                    continue jalangiLabel40;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false)));
                                            }
                                            return J$.Rt(293817, J$.F(293809, J$.R(293793, 'maybeRetryOrThrow', maybeRetryOrThrow, false, false), false)(J$.R(293801, 'err', err, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(300721, J$e);
                                        } finally {
                                            if (J$.Fr(300729))
                                                continue jalangiLabel41;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                        } catch (J$e) {
                            J$.Ex(300737, J$e);
                        } finally {
                            if (J$.Fr(300745))
                                continue jalangiLabel42;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function endTransaction(session, commandName, callback) {
                jalangiLabel47:
                    while (true) {
                        try {
                            J$.Fe(296129, arguments.callee, this, arguments);
                            function commandHandler(e, r) {
                                jalangiLabel43:
                                    while (true) {
                                        try {
                                            J$.Fe(295481, arguments.callee, this, arguments);
                                            arguments = J$.N(295489, 'arguments', arguments, true, false, false);
                                            e = J$.N(295497, 'e', e, true, false, false);
                                            r = J$.N(295505, 'r', r, true, false, false);
                                            J$.N(295513, 'idx', idx, false, false, false);
                                            if (J$.C(17248, J$.B(18290, '===', J$.R(295049, 'commandName', commandName, false, false), J$.T(295057, 'commitTransaction', 21, false)))) {
                                                J$.M(295097, J$.G(295073, J$.R(295065, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(295089, J$.R(295081, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED'));
                                                if (J$.C(17240, J$.C(17208, J$.R(295105, 'e', e, false, false)) ? J$.C(17200, J$.C(17192, J$.C(17184, J$.B(18298, 'instanceof', J$.R(295113, 'e', e, false, false), J$.R(295121, 'MongoNetworkError', MongoNetworkError, false, true))) ? J$._() : J$.B(18306, 'instanceof', J$.R(295129, 'e', e, false, false), J$.R(295137, 'MongoWriteConcernError', MongoWriteConcernError, false, true))) ? J$._() : J$.F(295161, J$.R(295145, 'isRetryableError', isRetryableError, false, true), false)(J$.R(295153, 'e', e, false, false))) ? J$._() : J$.F(295185, J$.R(295169, 'isMaxTimeMSExpiredError', isMaxTimeMSExpiredError, false, true), false)(J$.R(295177, 'e', e, false, false)) : J$._())) {
                                                    if (J$.C(17224, J$.G(295201, J$.R(295193, 'e', e, false, false), 'errorLabels'))) {
                                                        var idx = J$.W(295241, 'idx', J$.M(295233, J$.G(295217, J$.R(295209, 'e', e, false, false), 'errorLabels'), 'indexOf', false)(J$.T(295225, 'TransientTransactionError', 21, false)), idx, false, false);
                                                        if (J$.C(17216, J$.B(18322, '!==', J$.R(295249, 'idx', idx, false, false), J$.U(18314, '-', J$.T(295257, 1, 22, false))))) {
                                                            J$.M(295297, J$.G(295273, J$.R(295265, 'e', e, false, false), 'errorLabels'), 'splice', false)(J$.R(295281, 'idx', idx, false, false), J$.T(295289, 1, 22, false));
                                                        }
                                                    } else {
                                                        J$.P(295321, J$.R(295305, 'e', e, false, false), 'errorLabels', J$.T(295313, [], 10, false));
                                                    }
                                                    if (J$.C(17232, J$.F(295345, J$.R(295329, 'isUnknownTransactionCommitResult', isUnknownTransactionCommitResult, false, true), false)(J$.R(295337, 'e', e, false, false)))) {
                                                        J$.M(295377, J$.G(295361, J$.R(295353, 'e', e, false, false), 'errorLabels'), 'push', false)(J$.T(295369, 'UnknownTransactionCommitResult', 21, false));
                                                        J$.M(295401, J$.G(295393, J$.R(295385, 'session', session, false, false), 'transaction'), 'unpinServer', false)();
                                                    }
                                                }
                                            } else {
                                                J$.M(295441, J$.G(295417, J$.R(295409, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(295433, J$.R(295425, 'TxnState', TxnState, false, true), 'TRANSACTION_ABORTED'));
                                            }
                                            J$.F(295473, J$.R(295449, 'callback', callback, false, false), false)(J$.R(295457, 'e', e, false, false), J$.R(295465, 'r', r, false, false));
                                        } catch (J$e) {
                                            J$.Ex(300753, J$e);
                                        } finally {
                                            if (J$.Fr(300761))
                                                continue jalangiLabel43;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            function transactionError(err) {
                                jalangiLabel44:
                                    while (true) {
                                        try {
                                            J$.Fe(295561, arguments.callee, this, arguments);
                                            arguments = J$.N(295569, 'arguments', arguments, true, false, false);
                                            err = J$.N(295577, 'err', err, true, false, false);
                                            return J$.Rt(295553, J$.C(17256, J$.B(18330, '===', J$.R(295521, 'commandName', commandName, false, false), J$.T(295529, 'commitTransaction', 21, false))) ? J$.R(295537, 'err', err, false, false) : J$.T(295545, null, 25, false));
                                        } catch (J$e) {
                                            J$.Ex(300769, J$e);
                                        } finally {
                                            if (J$.Fr(300777))
                                                continue jalangiLabel44;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(296137, 'arguments', arguments, true, false, false);
                            session = J$.N(296145, 'session', session, true, false, false);
                            commandName = J$.N(296153, 'commandName', commandName, true, false, false);
                            callback = J$.N(296161, 'callback', callback, true, false, false);
                            J$.N(296169, 'txnState', txnState, false, false, false);
                            J$.N(296177, 'command', command, false, false, false);
                            J$.N(296185, 'writeConcern', writeConcern, false, false, false);
                            commandHandler = J$.N(296201, 'commandHandler', J$.T(296193, commandHandler, 12, false), true, false, false);
                            transactionError = J$.N(296217, 'transactionError', J$.T(296209, transactionError, 12, false), true, false, false);
                            if (J$.C(17048, J$.U(18194, '!', J$.F(293969, J$.R(293945, 'assertAlive', assertAlive, false, true), false)(J$.R(293953, 'session', session, false, false), J$.R(293961, 'callback', callback, false, false))))) {
                                return J$.Rt(293977, undefined);
                            }
                            var txnState = J$.W(294009, 'txnState', J$.G(294001, J$.G(293993, J$.R(293985, 'session', session, false, false), 'transaction'), 'state'), txnState, false, false);
                            if (J$.C(17056, J$.B(18202, '===', J$.R(294017, 'txnState', txnState, false, false), J$.G(294033, J$.R(294025, 'TxnState', TxnState, false, true), 'NO_TRANSACTION')))) {
                                J$.F(294073, J$.R(294041, 'callback', callback, false, false), false)(J$.F(294065, J$.R(294049, 'MongoError', MongoError, false, true), true)(J$.T(294057, 'No transaction started', 21, false)));
                                return J$.Rt(294081, undefined);
                            }
                            if (J$.C(17120, J$.B(18210, '===', J$.R(294089, 'commandName', commandName, false, false), J$.T(294097, 'commitTransaction', 21, false)))) {
                                if (J$.C(17072, J$.C(17064, J$.B(18218, '===', J$.R(294105, 'txnState', txnState, false, false), J$.G(294121, J$.R(294113, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION'))) ? J$._() : J$.B(18226, '===', J$.R(294129, 'txnState', txnState, false, false), J$.G(294145, J$.R(294137, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED_EMPTY')))) {
                                    J$.M(294185, J$.G(294161, J$.R(294153, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(294177, J$.R(294169, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED_EMPTY'));
                                    J$.F(294217, J$.R(294193, 'callback', callback, false, false), false)(J$.T(294201, null, 25, false), J$.T(294209, null, 25, false));
                                    return J$.Rt(294225, undefined);
                                }
                                if (J$.C(17080, J$.B(18234, '===', J$.R(294233, 'txnState', txnState, false, false), J$.G(294249, J$.R(294241, 'TxnState', TxnState, false, true), 'TRANSACTION_ABORTED')))) {
                                    J$.F(294289, J$.R(294257, 'callback', callback, false, false), false)(J$.F(294281, J$.R(294265, 'MongoError', MongoError, false, true), true)(J$.T(294273, 'Cannot call commitTransaction after calling abortTransaction', 21, false)));
                                    return J$.Rt(294297, undefined);
                                }
                            } else {
                                if (J$.C(17088, J$.B(18242, '===', J$.R(294305, 'txnState', txnState, false, false), J$.G(294321, J$.R(294313, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION')))) {
                                    J$.M(294361, J$.G(294337, J$.R(294329, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(294353, J$.R(294345, 'TxnState', TxnState, false, true), 'TRANSACTION_ABORTED'));
                                    J$.F(294393, J$.R(294369, 'callback', callback, false, false), false)(J$.T(294377, null, 25, false), J$.T(294385, null, 25, false));
                                    return J$.Rt(294401, undefined);
                                }
                                if (J$.C(17096, J$.B(18250, '===', J$.R(294409, 'txnState', txnState, false, false), J$.G(294425, J$.R(294417, 'TxnState', TxnState, false, true), 'TRANSACTION_ABORTED')))) {
                                    J$.F(294465, J$.R(294433, 'callback', callback, false, false), false)(J$.F(294457, J$.R(294441, 'MongoError', MongoError, false, true), true)(J$.T(294449, 'Cannot call abortTransaction twice', 21, false)));
                                    return J$.Rt(294473, undefined);
                                }
                                if (J$.C(17112, J$.C(17104, J$.B(18258, '===', J$.R(294481, 'txnState', txnState, false, false), J$.G(294497, J$.R(294489, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED'))) ? J$._() : J$.B(18266, '===', J$.R(294505, 'txnState', txnState, false, false), J$.G(294521, J$.R(294513, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED_EMPTY')))) {
                                    J$.F(294561, J$.R(294529, 'callback', callback, false, false), false)(J$.F(294553, J$.R(294537, 'MongoError', MongoError, false, true), true)(J$.T(294545, 'Cannot call abortTransaction after calling commitTransaction', 21, false)));
                                    return J$.Rt(294569, undefined);
                                }
                            }
                            var command = J$.W(294617, 'command', J$.F(294609, J$.R(294577, '_defineProperty', _defineProperty, false, true), false)(J$.T(294585, {}, 11, false), J$.R(294593, 'commandName', commandName, false, false), J$.T(294601, 1, 22, false)), command, false, false);
                            var writeConcern;
                            if (J$.C(17144, J$.G(294649, J$.G(294641, J$.G(294633, J$.R(294625, 'session', session, false, false), 'transaction'), 'options'), 'writeConcern'))) {
                                writeConcern = J$.W(294713, 'writeConcern', J$.M(294705, J$.I(typeof Object === 'undefined' ? Object = J$.R(294657, 'Object', undefined, true, true) : Object = J$.R(294657, 'Object', Object, true, true)), 'assign', false)(J$.T(294665, {}, 11, false), J$.G(294697, J$.G(294689, J$.G(294681, J$.R(294673, 'session', session, false, false), 'transaction'), 'options'), 'writeConcern')), writeConcern, false, false);
                            } else if (J$.C(17136, J$.C(17128, J$.G(294729, J$.R(294721, 'session', session, false, false), 'clientOptions')) ? J$.G(294753, J$.G(294745, J$.R(294737, 'session', session, false, false), 'clientOptions'), 'w') : J$._())) {
                                writeConcern = J$.W(294793, 'writeConcern', J$.T(294785, { w: J$.G(294777, J$.G(294769, J$.R(294761, 'session', session, false, false), 'clientOptions'), 'w') }, 11, false), writeConcern, false, false);
                            }
                            if (J$.C(17152, J$.B(18274, '===', J$.R(294801, 'txnState', txnState, false, false), J$.G(294817, J$.R(294809, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED')))) {
                                writeConcern = J$.W(294881, 'writeConcern', J$.M(294873, J$.I(typeof Object === 'undefined' ? Object = J$.R(294825, 'Object', undefined, true, true) : Object = J$.R(294825, 'Object', Object, true, true)), 'assign', false)(J$.T(294841, { wtimeout: J$.T(294833, 10000, 22, false) }, 11, false), J$.R(294849, 'writeConcern', writeConcern, false, false), J$.T(294865, { w: J$.T(294857, 'majority', 21, false) }, 11, false)), writeConcern, false, false);
                            }
                            if (J$.C(17160, J$.R(294889, 'writeConcern', writeConcern, false, false))) {
                                J$.M(294929, J$.I(typeof Object === 'undefined' ? Object = J$.R(294897, 'Object', undefined, true, true) : Object = J$.R(294897, 'Object', Object, true, true)), 'assign', false)(J$.R(294905, 'command', command, false, false), J$.T(294921, { writeConcern: J$.R(294913, 'writeConcern', writeConcern, false, false) }, 11, false));
                            }
                            if (J$.C(17176, J$.C(17168, J$.B(18282, '===', J$.R(294937, 'commandName', commandName, false, false), J$.T(294945, 'commitTransaction', 21, false))) ? J$.G(294977, J$.G(294969, J$.G(294961, J$.R(294953, 'session', session, false, false), 'transaction'), 'options'), 'maxTimeMS') : J$._())) {
                                J$.M(295041, J$.I(typeof Object === 'undefined' ? Object = J$.R(294985, 'Object', undefined, true, true) : Object = J$.R(294985, 'Object', Object, true, true)), 'assign', false)(J$.R(294993, 'command', command, false, false), J$.T(295033, { maxTimeMS: J$.G(295025, J$.G(295017, J$.G(295009, J$.R(295001, 'session', session, false, false), 'transaction'), 'options'), 'maxTimeMS') }, 11, false));
                            }
                            if (J$.C(17272, J$.C(17264, J$.G(295601, J$.G(295593, J$.R(295585, 'session', session, false, false), 'transaction'), 'recoveryToken')) ? J$.F(295625, J$.R(295609, 'supportsRecoveryToken', supportsRecoveryToken, false, true), false)(J$.R(295617, 'session', session, false, false)) : J$._())) {
                                J$.P(295665, J$.R(295633, 'command', command, false, false), 'recoveryToken', J$.G(295657, J$.G(295649, J$.R(295641, 'session', session, false, false), 'transaction'), 'recoveryToken'));
                            }
                            J$.M(296121, J$.G(295681, J$.R(295673, 'session', session, false, false), 'topology'), 'command', false)(J$.T(295689, 'admin.$cmd', 21, false), J$.R(295697, 'command', command, false, false), J$.T(295713, { session: J$.R(295705, 'session', session, false, false) }, 11, false), J$.T(296113, function (err, reply) {
                                jalangiLabel46:
                                    while (true) {
                                        try {
                                            J$.Fe(296081, arguments.callee, this, arguments);
                                            arguments = J$.N(296089, 'arguments', arguments, true, false, false);
                                            err = J$.N(296097, 'err', err, true, false, false);
                                            reply = J$.N(296105, 'reply', reply, true, false, false);
                                            if (J$.C(17296, J$.C(17280, J$.R(295721, 'err', err, false, false)) ? J$.F(295745, J$.R(295729, 'isRetryableError', isRetryableError, false, true), false)(J$.R(295737, 'err', err, false, false)) : J$._())) {
                                                if (J$.C(17288, J$.G(295761, J$.R(295753, 'command', command, false, false), 'commitTransaction'))) {
                                                    J$.M(295785, J$.G(295777, J$.R(295769, 'session', session, false, false), 'transaction'), 'unpinServer', false)();
                                                    J$.P(295865, J$.R(295793, 'command', command, false, false), 'writeConcern', J$.M(295857, J$.I(typeof Object === 'undefined' ? Object = J$.R(295801, 'Object', undefined, true, true) : Object = J$.R(295801, 'Object', Object, true, true)), 'assign', false)(J$.T(295817, { wtimeout: J$.T(295809, 10000, 22, false) }, 11, false), J$.G(295833, J$.R(295825, 'command', command, false, false), 'writeConcern'), J$.T(295849, { w: J$.T(295841, 'majority', 21, false) }, 11, false)));
                                                }
                                                return J$.Rt(296025, J$.M(296017, J$.G(295881, J$.R(295873, 'session', session, false, false), 'topology'), 'command', false)(J$.T(295889, 'admin.$cmd', 21, false), J$.R(295897, 'command', command, false, false), J$.T(295913, { session: J$.R(295905, 'session', session, false, false) }, 11, false), J$.T(296009, function (_err, _reply) {
                                                    jalangiLabel45:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(295977, arguments.callee, this, arguments);
                                                                arguments = J$.N(295985, 'arguments', arguments, true, false, false);
                                                                _err = J$.N(295993, '_err', _err, true, false, false);
                                                                _reply = J$.N(296001, '_reply', _reply, true, false, false);
                                                                return J$.Rt(295969, J$.F(295961, J$.R(295921, 'commandHandler', commandHandler, false, false), false)(J$.F(295945, J$.R(295929, 'transactionError', transactionError, false, false), false)(J$.R(295937, '_err', _err, false, false)), J$.R(295953, '_reply', _reply, false, false)));
                                                            } catch (J$e) {
                                                                J$.Ex(300785, J$e);
                                                            } finally {
                                                                if (J$.Fr(300793))
                                                                    continue jalangiLabel45;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false)));
                                            }
                                            J$.F(296073, J$.R(296033, 'commandHandler', commandHandler, false, false), false)(J$.F(296057, J$.R(296041, 'transactionError', transactionError, false, false), false)(J$.R(296049, 'err', err, false, false)), J$.R(296065, 'reply', reply, false, false));
                                        } catch (J$e) {
                                            J$.Ex(300801, J$e);
                                        } finally {
                                            if (J$.Fr(300809))
                                                continue jalangiLabel46;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(300817, J$e);
                        } finally {
                            if (J$.Fr(300825))
                                continue jalangiLabel47;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function supportsRecoveryToken(session) {
                jalangiLabel48:
                    while (true) {
                        try {
                            J$.Fe(296289, arguments.callee, this, arguments);
                            arguments = J$.N(296297, 'arguments', arguments, true, false, false);
                            session = J$.N(296305, 'session', session, true, false, false);
                            J$.N(296313, 'topology', topology, false, false, false);
                            var topology = J$.W(296241, 'topology', J$.G(296233, J$.R(296225, 'session', session, false, false), 'topology'), topology, false, false);
                            return J$.Rt(296281, J$.U(18346, '!', J$.U(18338, '!', J$.G(296273, J$.G(296265, J$.G(296257, J$.R(296249, 'topology', topology, false, false), 's'), 'options'), 'useRecoveryToken'))));
                        } catch (J$e) {
                            J$.Ex(300833, J$e);
                        } finally {
                            if (J$.Fr(300841))
                                continue jalangiLabel48;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function commandSupportsReadConcern(command, options) {
                jalangiLabel58:
                    while (true) {
                        try {
                            J$.Fe(298001, arguments.callee, this, arguments);
                            arguments = J$.N(298009, 'arguments', arguments, true, false, false);
                            command = J$.N(298017, 'command', command, true, false, false);
                            options = J$.N(298025, 'options', options, true, false, false);
                            if (J$.C(17416, J$.C(17408, J$.C(17400, J$.C(17392, J$.C(17384, J$.C(17376, J$.C(17368, J$.G(297761, J$.R(297753, 'command', command, false, false), 'aggregate')) ? J$._() : J$.G(297777, J$.R(297769, 'command', command, false, false), 'count')) ? J$._() : J$.G(297793, J$.R(297785, 'command', command, false, false), 'distinct')) ? J$._() : J$.G(297809, J$.R(297801, 'command', command, false, false), 'find')) ? J$._() : J$.G(297825, J$.R(297817, 'command', command, false, false), 'parallelCollectionScan')) ? J$._() : J$.G(297841, J$.R(297833, 'command', command, false, false), 'geoNear')) ? J$._() : J$.G(297857, J$.R(297849, 'command', command, false, false), 'geoSearch'))) {
                                return J$.Rt(297873, J$.T(297865, true, 23, false));
                            }
                            if (J$.C(17448, J$.C(17440, J$.C(17424, J$.G(297889, J$.R(297881, 'command', command, false, false), 'mapReduce')) ? J$.G(297905, J$.R(297897, 'options', options, false, false), 'out') : J$._()) ? J$.C(17432, J$.B(18434, '===', J$.G(297929, J$.G(297921, J$.R(297913, 'options', options, false, false), 'out'), 'inline'), J$.T(297937, 1, 22, false))) ? J$._() : J$.B(18442, '===', J$.G(297953, J$.R(297945, 'options', options, false, false), 'out'), J$.T(297961, 'inline', 21, false)) : J$._())) {
                                return J$.Rt(297977, J$.T(297969, true, 23, false));
                            }
                            return J$.Rt(297993, J$.T(297985, false, 23, false));
                        } catch (J$e) {
                            J$.Ex(300993, J$e);
                        } finally {
                            if (J$.Fr(301001))
                                continue jalangiLabel58;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function applySession(session, command, options) {
                jalangiLabel59:
                    while (true) {
                        try {
                            J$.Fe(299081, arguments.callee, this, arguments);
                            arguments = J$.N(299089, 'arguments', arguments, true, false, false);
                            session = J$.N(299097, 'session', session, true, false, false);
                            command = J$.N(299105, 'command', command, true, false, false);
                            options = J$.N(299113, 'options', options, true, false, false);
                            J$.N(299121, 'serverSession', serverSession, false, false, false);
                            J$.N(299129, 'inTransaction', inTransaction, false, false, false);
                            J$.N(299137, 'isRetryableWrite', isRetryableWrite, false, false, false);
                            J$.N(299145, 'shouldApplyReadConcern', shouldApplyReadConcern, false, false, false);
                            J$.N(299153, 'readConcern', readConcern, false, false, false);
                            var serverSession = J$.W(298049, 'serverSession', J$.G(298041, J$.R(298033, 'session', session, false, false), 'serverSession'), serverSession, false, false);
                            if (J$.C(17456, J$.B(18450, '==', J$.R(298057, 'serverSession', serverSession, false, false), J$.T(298065, null, 25, false)))) {
                                return J$.Rt(298097, J$.F(298089, J$.R(298073, 'MongoError', MongoError, false, true), true)(J$.T(298081, 'Cannot use a session that has ended', 21, false)));
                            }
                            J$.P(298129, J$.R(298105, 'serverSession', serverSession, false, false), 'lastUse', J$.M(298121, J$.I(typeof Date === 'undefined' ? Date = J$.R(298113, 'Date', undefined, true, true) : Date = J$.R(298113, 'Date', Date, true, true)), 'now', false)());
                            J$.P(298161, J$.R(298137, 'command', command, false, false), 'lsid', J$.G(298153, J$.R(298145, 'serverSession', serverSession, false, false), 'id'));
                            var inTransaction = J$.W(298209, 'inTransaction', J$.C(17464, J$.M(298177, J$.R(298169, 'session', session, false, false), 'inTransaction', false)()) ? J$._() : J$.F(298201, J$.R(298185, 'isTransactionCommand', isTransactionCommand, false, true), false)(J$.R(298193, 'command', command, false, false)), inTransaction, false, false);
                            var isRetryableWrite = J$.W(298233, 'isRetryableWrite', J$.G(298225, J$.R(298217, 'options', options, false, false), 'willRetryWrite'), isRetryableWrite, false, false);
                            var shouldApplyReadConcern = J$.W(298265, 'shouldApplyReadConcern', J$.F(298257, J$.R(298241, 'commandSupportsReadConcern', commandSupportsReadConcern, false, true), false)(J$.R(298249, 'command', command, false, false)), shouldApplyReadConcern, false, false);
                            if (J$.C(17488, J$.C(17480, J$.G(298281, J$.R(298273, 'serverSession', serverSession, false, false), 'txnNumber')) ? J$.C(17472, J$.R(298289, 'isRetryableWrite', isRetryableWrite, false, false)) ? J$._() : J$.R(298297, 'inTransaction', inTransaction, false, false) : J$._())) {
                                J$.P(298353, J$.R(298305, 'command', command, false, false), 'txnNumber', J$.M(298345, J$.G(298321, J$.R(298313, 'BSON', BSON, false, true), 'Long'), 'fromNumber', false)(J$.G(298337, J$.R(298329, 'serverSession', serverSession, false, false), 'txnNumber')));
                            }
                            if (J$.C(17536, J$.U(18458, '!', J$.R(298361, 'inTransaction', inTransaction, false, false)))) {
                                if (J$.C(17496, J$.B(18466, '!==', J$.G(298385, J$.G(298377, J$.R(298369, 'session', session, false, false), 'transaction'), 'state'), J$.G(298401, J$.R(298393, 'TxnState', TxnState, false, true), 'NO_TRANSACTION')))) {
                                    J$.M(298441, J$.G(298417, J$.R(298409, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(298433, J$.R(298425, 'TxnState', TxnState, false, true), 'NO_TRANSACTION'));
                                }
                                if (J$.C(17528, J$.C(17512, J$.C(17504, J$.G(298465, J$.G(298457, J$.R(298449, 'session', session, false, false), 'supports'), 'causalConsistency')) ? J$.G(298481, J$.R(298473, 'session', session, false, false), 'operationTime') : J$._()) ? J$.R(298489, 'shouldApplyReadConcern', shouldApplyReadConcern, false, false) : J$._())) {
                                    J$.P(298529, J$.R(298497, 'command', command, false, false), 'readConcern', J$.C(17520, J$.G(298513, J$.R(298505, 'command', command, false, false), 'readConcern')) ? J$._() : J$.T(298521, {}, 11, false));
                                    J$.M(298585, J$.I(typeof Object === 'undefined' ? Object = J$.R(298537, 'Object', undefined, true, true) : Object = J$.R(298537, 'Object', Object, true, true)), 'assign', false)(J$.G(298553, J$.R(298545, 'command', command, false, false), 'readConcern'), J$.T(298577, { afterClusterTime: J$.G(298569, J$.R(298561, 'session', session, false, false), 'operationTime') }, 11, false));
                                }
                                return J$.Rt(298593, undefined);
                            }
                            if (J$.C(17552, J$.C(17544, J$.G(298609, J$.R(298601, 'options', options, false, false), 'readPreference')) ? J$.U(18474, '!', J$.M(298649, J$.G(298625, J$.R(298617, 'options', options, false, false), 'readPreference'), 'equals', false)(J$.G(298641, J$.R(298633, 'ReadPreference', ReadPreference, false, true), 'primary'))) : J$._())) {
                                return J$.Rt(298713, J$.F(298705, J$.R(298657, 'MongoError', MongoError, false, true), true)(J$.M(298697, J$.T(298665, 'Read preference in a transaction must be primary, not: ', 21, false), 'concat', false)(J$.G(298689, J$.G(298681, J$.R(298673, 'options', options, false, false), 'readPreference'), 'mode'))));
                            }
                            J$.P(298737, J$.R(298721, 'command', command, false, false), 'autocommit', J$.T(298729, false, 23, false));
                            if (J$.C(17600, J$.B(18482, '===', J$.G(298761, J$.G(298753, J$.R(298745, 'session', session, false, false), 'transaction'), 'state'), J$.G(298777, J$.R(298769, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION')))) {
                                J$.M(298817, J$.G(298793, J$.R(298785, 'session', session, false, false), 'transaction'), 'transition', false)(J$.G(298809, J$.R(298801, 'TxnState', TxnState, false, true), 'TRANSACTION_IN_PROGRESS'));
                                J$.P(298841, J$.R(298825, 'command', command, false, false), 'startTransaction', J$.T(298833, true, 23, false));
                                var readConcern = J$.W(298905, 'readConcern', J$.C(17560, J$.G(298873, J$.G(298865, J$.G(298857, J$.R(298849, 'session', session, false, false), 'transaction'), 'options'), 'readConcern')) ? J$._() : J$.G(298897, J$.G(298889, J$.R(298881, 'session', session, false, false), 'clientOptions'), 'readConcern'), readConcern, false, false);
                                if (J$.C(17568, J$.R(298913, 'readConcern', readConcern, false, false))) {
                                    J$.P(298937, J$.R(298921, 'command', command, false, false), 'readConcern', J$.R(298929, 'readConcern', readConcern, false, false));
                                }
                                if (J$.C(17592, J$.C(17576, J$.G(298961, J$.G(298953, J$.R(298945, 'session', session, false, false), 'supports'), 'causalConsistency')) ? J$.G(298977, J$.R(298969, 'session', session, false, false), 'operationTime') : J$._())) {
                                    J$.P(299017, J$.R(298985, 'command', command, false, false), 'readConcern', J$.C(17584, J$.G(299001, J$.R(298993, 'command', command, false, false), 'readConcern')) ? J$._() : J$.T(299009, {}, 11, false));
                                    J$.M(299073, J$.I(typeof Object === 'undefined' ? Object = J$.R(299025, 'Object', undefined, true, true) : Object = J$.R(299025, 'Object', Object, true, true)), 'assign', false)(J$.G(299041, J$.R(299033, 'command', command, false, false), 'readConcern'), J$.T(299065, { afterClusterTime: J$.G(299057, J$.R(299049, 'session', session, false, false), 'operationTime') }, 11, false));
                                }
                            }
                        } catch (J$e) {
                            J$.Ex(301009, J$e);
                        } finally {
                            if (J$.Fr(301017))
                                continue jalangiLabel59;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function updateSessionFromResponse(session, document) {
                jalangiLabel60:
                    while (true) {
                        try {
                            J$.Fe(299377, arguments.callee, this, arguments);
                            arguments = J$.N(299385, 'arguments', arguments, true, false, false);
                            session = J$.N(299393, 'session', session, true, false, false);
                            document = J$.N(299401, 'document', document, true, false, false);
                            if (J$.C(17608, J$.G(299169, J$.R(299161, 'document', document, false, false), '$clusterTime'))) {
                                J$.F(299209, J$.R(299177, 'resolveClusterTime', resolveClusterTime, false, true), false)(J$.R(299185, 'session', session, false, false), J$.G(299201, J$.R(299193, 'document', document, false, false), '$clusterTime'));
                            }
                            if (J$.C(17632, J$.C(17624, J$.C(17616, J$.G(299225, J$.R(299217, 'document', document, false, false), 'operationTime')) ? J$.R(299233, 'session', session, false, false) : J$._()) ? J$.G(299257, J$.G(299249, J$.R(299241, 'session', session, false, false), 'supports'), 'causalConsistency') : J$._())) {
                                J$.M(299289, J$.R(299265, 'session', session, false, false), 'advanceOperationTime', false)(J$.G(299281, J$.R(299273, 'document', document, false, false), 'operationTime'));
                            }
                            if (J$.C(17656, J$.C(17648, J$.C(17640, J$.G(299305, J$.R(299297, 'document', document, false, false), 'recoveryToken')) ? J$.R(299313, 'session', session, false, false) : J$._()) ? J$.M(299329, J$.R(299321, 'session', session, false, false), 'inTransaction', false)() : J$._())) {
                                J$.P(299369, J$.G(299345, J$.R(299337, 'session', session, false, false), 'transaction'), '_recoveryToken', J$.G(299361, J$.R(299353, 'document', document, false, false), 'recoveryToken'));
                            }
                        } catch (J$e) {
                            J$.Ex(301025, J$e);
                        } finally {
                            if (J$.Fr(301033))
                                continue jalangiLabel60;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _defineProperty = J$.N(299505, '_defineProperty', J$.T(299497, _defineProperty, 12, false), true, false, false);
            _typeof = J$.N(299521, '_typeof', J$.T(299513, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(299537, '_classCallCheck', J$.T(299529, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(299553, '_defineProperties', J$.T(299545, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(299569, '_createClass', J$.T(299561, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(299585, '_possibleConstructorReturn', J$.T(299577, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(299601, '_assertThisInitialized', J$.T(299593, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(299617, '_getPrototypeOf', J$.T(299609, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(299633, '_inherits', J$.T(299625, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(299649, '_setPrototypeOf', J$.T(299641, _setPrototypeOf, 12, false), true, false, false);
            J$.N(299657, 'retrieveBSON', retrieveBSON, false, false, false);
            J$.N(299665, 'EventEmitter', EventEmitter, false, false, false);
            J$.N(299673, 'BSON', BSON, false, false, false);
            J$.N(299681, 'Binary', Binary, false, false, false);
            J$.N(299689, 'uuidV4', uuidV4, false, false, false);
            J$.N(299697, 'MongoError', MongoError, false, false, false);
            J$.N(299705, 'isRetryableError', isRetryableError, false, false, false);
            J$.N(299713, 'MongoNetworkError', MongoNetworkError, false, false, false);
            J$.N(299721, 'MongoWriteConcernError', MongoWriteConcernError, false, false, false);
            J$.N(299729, 'Transaction', Transaction, false, false, false);
            J$.N(299737, 'TxnState', TxnState, false, false, false);
            J$.N(299745, 'isPromiseLike', isPromiseLike, false, false, false);
            J$.N(299753, 'ReadPreference', ReadPreference, false, false, false);
            J$.N(299761, 'isTransactionCommand', isTransactionCommand, false, false, false);
            J$.N(299769, 'resolveClusterTime', resolveClusterTime, false, false, false);
            J$.N(299777, 'isSharded', isSharded, false, false, false);
            J$.N(299785, 'maxWireVersion', maxWireVersion, false, false, false);
            J$.N(299793, 'minWireVersionForShardedTransactions', minWireVersionForShardedTransactions, false, false, false);
            assertAlive = J$.N(299809, 'assertAlive', J$.T(299801, assertAlive, 12, false), true, false, false);
            J$.N(299817, 'ClientSession', ClientSession, false, false, false);
            J$.N(299825, 'MAX_WITH_TRANSACTION_TIMEOUT', MAX_WITH_TRANSACTION_TIMEOUT, false, false, false);
            J$.N(299833, 'UNSATISFIABLE_WRITE_CONCERN_CODE', UNSATISFIABLE_WRITE_CONCERN_CODE, false, false, false);
            J$.N(299841, 'UNKNOWN_REPL_WRITE_CONCERN_CODE', UNKNOWN_REPL_WRITE_CONCERN_CODE, false, false, false);
            J$.N(299849, 'MAX_TIME_MS_EXPIRED_CODE', MAX_TIME_MS_EXPIRED_CODE, false, false, false);
            J$.N(299857, 'NON_DETERMINISTIC_WRITE_CONCERN_ERRORS', NON_DETERMINISTIC_WRITE_CONCERN_ERRORS, false, false, false);
            hasNotTimedOut = J$.N(299873, 'hasNotTimedOut', J$.T(299865, hasNotTimedOut, 12, false), true, false, false);
            isUnknownTransactionCommitResult = J$.N(299889, 'isUnknownTransactionCommitResult', J$.T(299881, isUnknownTransactionCommitResult, 12, false), true, false, false);
            isMaxTimeMSExpiredError = J$.N(299905, 'isMaxTimeMSExpiredError', J$.T(299897, isMaxTimeMSExpiredError, 12, false), true, false, false);
            attemptTransactionCommit = J$.N(299921, 'attemptTransactionCommit', J$.T(299913, attemptTransactionCommit, 12, false), true, false, false);
            J$.N(299929, 'USER_EXPLICIT_TXN_END_STATES', USER_EXPLICIT_TXN_END_STATES, false, false, false);
            userExplicitlyEndedTransaction = J$.N(299945, 'userExplicitlyEndedTransaction', J$.T(299937, userExplicitlyEndedTransaction, 12, false), true, false, false);
            attemptTransaction = J$.N(299961, 'attemptTransaction', J$.T(299953, attemptTransaction, 12, false), true, false, false);
            endTransaction = J$.N(299977, 'endTransaction', J$.T(299969, endTransaction, 12, false), true, false, false);
            supportsRecoveryToken = J$.N(299993, 'supportsRecoveryToken', J$.T(299985, supportsRecoveryToken, 12, false), true, false, false);
            J$.N(300001, 'ServerSession', ServerSession, false, false, false);
            J$.N(300009, 'ServerSessionPool', ServerSessionPool, false, false, false);
            commandSupportsReadConcern = J$.N(300025, 'commandSupportsReadConcern', J$.T(300017, commandSupportsReadConcern, 12, false), true, false, false);
            applySession = J$.N(300041, 'applySession', J$.T(300033, applySession, 12, false), true, false, false);
            updateSessionFromResponse = J$.N(300057, 'updateSessionFromResponse', J$.T(300049, updateSessionFromResponse, 12, false), true, false, false);
            J$.T(286473, 'use strict', 21, false);
            var retrieveBSON = J$.W(288273, 'retrieveBSON', J$.G(288265, J$.F(288257, J$.I(typeof require === 'undefined' ? require = J$.R(288241, 'require', undefined, true, true) : require = J$.R(288241, 'require', require, true, true)), false)(J$.T(288249, './connection/utils', 21, false)), 'retrieveBSON'), retrieveBSON, false, true);
            var EventEmitter = J$.W(288305, 'EventEmitter', J$.F(288297, J$.I(typeof require === 'undefined' ? require = J$.R(288281, 'require', undefined, true, true) : require = J$.R(288281, 'require', require, true, true)), false)(J$.T(288289, 'events', 21, false)), EventEmitter, false, true);
            var BSON = J$.W(288329, 'BSON', J$.F(288321, J$.R(288313, 'retrieveBSON', retrieveBSON, false, true), false)(), BSON, false, true);
            var Binary = J$.W(288353, 'Binary', J$.G(288345, J$.R(288337, 'BSON', BSON, false, true), 'Binary'), Binary, false, true);
            var uuidV4 = J$.W(288393, 'uuidV4', J$.G(288385, J$.F(288377, J$.I(typeof require === 'undefined' ? require = J$.R(288361, 'require', undefined, true, true) : require = J$.R(288361, 'require', require, true, true)), false)(J$.T(288369, './utils', 21, false)), 'uuidV4'), uuidV4, false, true);
            var MongoError = J$.W(288433, 'MongoError', J$.G(288425, J$.F(288417, J$.I(typeof require === 'undefined' ? require = J$.R(288401, 'require', undefined, true, true) : require = J$.R(288401, 'require', require, true, true)), false)(J$.T(288409, './error', 21, false)), 'MongoError'), MongoError, false, true);
            var isRetryableError = J$.W(288473, 'isRetryableError', J$.G(288465, J$.F(288457, J$.I(typeof require === 'undefined' ? require = J$.R(288441, 'require', undefined, true, true) : require = J$.R(288441, 'require', require, true, true)), false)(J$.T(288449, '././error', 21, false)), 'isRetryableError'), isRetryableError, false, true);
            var MongoNetworkError = J$.W(288513, 'MongoNetworkError', J$.G(288505, J$.F(288497, J$.I(typeof require === 'undefined' ? require = J$.R(288481, 'require', undefined, true, true) : require = J$.R(288481, 'require', require, true, true)), false)(J$.T(288489, './error', 21, false)), 'MongoNetworkError'), MongoNetworkError, false, true);
            var MongoWriteConcernError = J$.W(288553, 'MongoWriteConcernError', J$.G(288545, J$.F(288537, J$.I(typeof require === 'undefined' ? require = J$.R(288521, 'require', undefined, true, true) : require = J$.R(288521, 'require', require, true, true)), false)(J$.T(288529, './error', 21, false)), 'MongoWriteConcernError'), MongoWriteConcernError, false, true);
            var Transaction = J$.W(288593, 'Transaction', J$.G(288585, J$.F(288577, J$.I(typeof require === 'undefined' ? require = J$.R(288561, 'require', undefined, true, true) : require = J$.R(288561, 'require', require, true, true)), false)(J$.T(288569, './transactions', 21, false)), 'Transaction'), Transaction, false, true);
            var TxnState = J$.W(288633, 'TxnState', J$.G(288625, J$.F(288617, J$.I(typeof require === 'undefined' ? require = J$.R(288601, 'require', undefined, true, true) : require = J$.R(288601, 'require', require, true, true)), false)(J$.T(288609, './transactions', 21, false)), 'TxnState'), TxnState, false, true);
            var isPromiseLike = J$.W(288673, 'isPromiseLike', J$.G(288665, J$.F(288657, J$.I(typeof require === 'undefined' ? require = J$.R(288641, 'require', undefined, true, true) : require = J$.R(288641, 'require', require, true, true)), false)(J$.T(288649, './utils', 21, false)), 'isPromiseLike'), isPromiseLike, false, true);
            var ReadPreference = J$.W(288705, 'ReadPreference', J$.F(288697, J$.I(typeof require === 'undefined' ? require = J$.R(288681, 'require', undefined, true, true) : require = J$.R(288681, 'require', require, true, true)), false)(J$.T(288689, './topologies/read_preference', 21, false)), ReadPreference, false, true);
            var isTransactionCommand = J$.W(288745, 'isTransactionCommand', J$.G(288737, J$.F(288729, J$.I(typeof require === 'undefined' ? require = J$.R(288713, 'require', undefined, true, true) : require = J$.R(288713, 'require', require, true, true)), false)(J$.T(288721, './transactions', 21, false)), 'isTransactionCommand'), isTransactionCommand, false, true);
            var resolveClusterTime = J$.W(288785, 'resolveClusterTime', J$.G(288777, J$.F(288769, J$.I(typeof require === 'undefined' ? require = J$.R(288753, 'require', undefined, true, true) : require = J$.R(288753, 'require', require, true, true)), false)(J$.T(288761, './topologies/shared', 21, false)), 'resolveClusterTime'), resolveClusterTime, false, true);
            var isSharded = J$.W(288825, 'isSharded', J$.G(288817, J$.F(288809, J$.I(typeof require === 'undefined' ? require = J$.R(288793, 'require', undefined, true, true) : require = J$.R(288793, 'require', require, true, true)), false)(J$.T(288801, './wireprotocol/shared', 21, false)), 'isSharded'), isSharded, false, true);
            var maxWireVersion = J$.W(288865, 'maxWireVersion', J$.G(288857, J$.F(288849, J$.I(typeof require === 'undefined' ? require = J$.R(288833, 'require', undefined, true, true) : require = J$.R(288833, 'require', require, true, true)), false)(J$.T(288841, './utils', 21, false)), 'maxWireVersion'), maxWireVersion, false, true);
            var minWireVersionForShardedTransactions = J$.W(288881, 'minWireVersionForShardedTransactions', J$.T(288873, 8, 22, false), minWireVersionForShardedTransactions, false, true);
            var ClientSession = J$.W(292169, 'ClientSession', J$.F(292161, J$.T(292145, function (_EventEmitter) {
                jalangiLabel31:
                    while (true) {
                        try {
                            J$.Fe(292105, arguments.callee, this, arguments);
                            function ClientSession(topology, sessionPool, options, clientOptions) {
                                jalangiLabel15:
                                    while (true) {
                                        try {
                                            J$.Fe(289777, arguments.callee, this, arguments);
                                            arguments = J$.N(289785, 'arguments', arguments, true, false, false);
                                            topology = J$.N(289793, 'topology', topology, true, false, false);
                                            sessionPool = J$.N(289801, 'sessionPool', sessionPool, true, false, false);
                                            options = J$.N(289809, 'options', options, true, false, false);
                                            clientOptions = J$.N(289817, 'clientOptions', clientOptions, true, false, false);
                                            J$.N(289825, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(289129, J$.R(289105, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(289113, 'this', this, false, false), J$.R(289121, 'ClientSession', ClientSession, false, false));
                                            _this = J$.W(289201, '_this', J$.F(289193, J$.R(289137, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(289145, 'this', this, false, false), J$.M(289185, J$.F(289169, J$.R(289153, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(289161, 'ClientSession', ClientSession, false, false)), 'call', false)(J$.R(289177, 'this', this, false, false))), _this, false, false);
                                            if (J$.C(16704, J$.B(17906, '==', J$.R(289209, 'topology', topology, false, false), J$.T(289217, null, 25, false)))) {
                                                throw J$.F(289241, J$.I(typeof Error === 'undefined' ? Error = J$.R(289225, 'Error', undefined, true, true) : Error = J$.R(289225, 'Error', Error, true, true)), true)(J$.T(289233, 'ClientSession requires a topology', 21, false));
                                            }
                                            if (J$.C(16720, J$.C(16712, J$.B(17914, '==', J$.R(289249, 'sessionPool', sessionPool, false, false), J$.T(289257, null, 25, false))) ? J$._() : J$.U(17930, '!', J$.B(17922, 'instanceof', J$.R(289265, 'sessionPool', sessionPool, false, false), J$.R(289273, 'ServerSessionPool', ServerSessionPool, false, true))))) {
                                                throw J$.F(289297, J$.I(typeof Error === 'undefined' ? Error = J$.R(289281, 'Error', undefined, true, true) : Error = J$.R(289281, 'Error', Error, true, true)), true)(J$.T(289289, 'ClientSession requires a ServerSessionPool', 21, false));
                                            }
                                            options = J$.W(289321, 'options', J$.C(16728, J$.R(289305, 'options', options, false, false)) ? J$._() : J$.T(289313, {}, 11, false), options, false, false);
                                            clientOptions = J$.W(289345, 'clientOptions', J$.C(16736, J$.R(289329, 'clientOptions', clientOptions, false, false)) ? J$._() : J$.T(289337, {}, 11, false), clientOptions, false, false);
                                            J$.P(289369, J$.R(289353, '_this', _this, false, false), 'topology', J$.R(289361, 'topology', topology, false, false));
                                            J$.P(289393, J$.R(289377, '_this', _this, false, false), 'sessionPool', J$.R(289385, 'sessionPool', sessionPool, false, false));
                                            J$.P(289417, J$.R(289401, '_this', _this, false, false), 'hasEnded', J$.T(289409, false, 23, false));
                                            J$.P(289449, J$.R(289425, '_this', _this, false, false), 'serverSession', J$.M(289441, J$.R(289433, 'sessionPool', sessionPool, false, false), 'acquire', false)());
                                            J$.P(289473, J$.R(289457, '_this', _this, false, false), 'clientOptions', J$.R(289465, 'clientOptions', clientOptions, false, false));
                                            J$.P(289545, J$.R(289481, '_this', _this, false, false), 'supports', J$.T(289537, { causalConsistency: J$.C(16744, J$.B(17946, '!==', J$.U(17938, 'typeof', J$.G(289497, J$.R(289489, 'options', options, false, false), 'causalConsistency')), J$.T(289505, 'undefined', 21, false))) ? J$.G(289521, J$.R(289513, 'options', options, false, false), 'causalConsistency') : J$.T(289529, true, 23, false) }, 11, false));
                                            J$.P(289577, J$.R(289553, '_this', _this, false, false), 'clusterTime', J$.G(289569, J$.R(289561, 'options', options, false, false), 'initialClusterTime'));
                                            J$.P(289601, J$.R(289585, '_this', _this, false, false), 'operationTime', J$.T(289593, null, 25, false));
                                            J$.P(289633, J$.R(289609, '_this', _this, false, false), 'explicit', J$.U(17962, '!', J$.U(17954, '!', J$.G(289625, J$.R(289617, 'options', options, false, false), 'explicit'))));
                                            J$.P(289665, J$.R(289641, '_this', _this, false, false), 'owner', J$.G(289657, J$.R(289649, 'options', options, false, false), 'owner'));
                                            J$.P(289721, J$.R(289673, '_this', _this, false, false), 'defaultTransactionOptions', J$.M(289713, J$.I(typeof Object === 'undefined' ? Object = J$.R(289681, 'Object', undefined, true, true) : Object = J$.R(289681, 'Object', Object, true, true)), 'assign', false)(J$.T(289689, {}, 11, false), J$.G(289705, J$.R(289697, 'options', options, false, false), 'defaultTransactionOptions')));
                                            J$.P(289753, J$.R(289729, '_this', _this, false, false), 'transaction', J$.F(289745, J$.R(289737, 'Transaction', Transaction, false, true), true)());
                                            return J$.Rt(289769, J$.R(289761, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(300305, J$e);
                                        } finally {
                                            if (J$.Fr(300313))
                                                continue jalangiLabel15;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(292113, 'arguments', arguments, true, false, false);
                            _EventEmitter = J$.N(292121, '_EventEmitter', _EventEmitter, true, false, false);
                            ClientSession = J$.N(292137, 'ClientSession', J$.T(292129, ClientSession, 12, false), true, false, false);
                            J$.F(289097, J$.R(289073, '_inherits', _inherits, false, true), false)(J$.R(289081, 'ClientSession', ClientSession, false, false), J$.R(289089, '_EventEmitter', _EventEmitter, false, false));
                            J$.F(292081, J$.R(289833, '_createClass', _createClass, false, true), false)(J$.R(289841, 'ClientSession', ClientSession, false, false), J$.T(292073, [
                                J$.T(290257, {
                                    key: J$.T(289849, 'endSession', 21, false),
                                    value: J$.T(290249, function endSession(options, callback) {
                                        jalangiLabel16:
                                            while (true) {
                                                try {
                                                    J$.Fe(290217, arguments.callee, this, arguments);
                                                    arguments = J$.N(290225, 'arguments', arguments, true, false, false);
                                                    options = J$.N(290233, 'options', options, true, false, false);
                                                    callback = J$.N(290241, 'callback', callback, true, false, false);
                                                    if (J$.C(16752, J$.B(17978, '===', J$.U(17970, 'typeof', J$.R(289857, 'options', options, false, false)), J$.T(289865, 'function', 21, false))))
                                                        callback = J$.W(289881, 'callback', J$.R(289873, 'options', options, false, false), callback, false, false), options = J$.W(289897, 'options', J$.T(289889, {}, 11, false), options, false, false);
                                                    options = J$.W(289921, 'options', J$.C(16760, J$.R(289905, 'options', options, false, false)) ? J$._() : J$.T(289913, {}, 11, false), options, false, false);
                                                    if (J$.C(16776, J$.G(289937, J$.R(289929, 'this', this, false, false), 'hasEnded'))) {
                                                        if (J$.C(16768, J$.B(17994, '===', J$.U(17986, 'typeof', J$.R(289945, 'callback', callback, false, false)), J$.T(289953, 'function', 21, false))))
                                                            J$.F(289985, J$.R(289961, 'callback', callback, false, false), false)(J$.T(289969, null, 25, false), J$.T(289977, null, 25, false));
                                                        return J$.Rt(289993, undefined);
                                                    }
                                                    if (J$.C(16792, J$.C(16784, J$.G(290009, J$.R(290001, 'this', this, false, false), 'serverSession')) ? J$.M(290025, J$.R(290017, 'this', this, false, false), 'inTransaction', false)() : J$._())) {
                                                        J$.M(290041, J$.R(290033, 'this', this, false, false), 'abortTransaction', false)();
                                                    }
                                                    J$.P(290065, J$.R(290049, 'this', this, false, false), 'hasEnded', J$.T(290057, true, 23, false));
                                                    J$.M(290097, J$.R(290073, 'this', this, false, false), 'emit', false)(J$.T(290081, 'ended', 21, false), J$.R(290089, 'this', this, false, false));
                                                    J$.M(290137, J$.G(290113, J$.R(290105, 'this', this, false, false), 'sessionPool'), 'release', false)(J$.G(290129, J$.R(290121, 'this', this, false, false), 'serverSession'));
                                                    J$.P(290161, J$.R(290145, 'this', this, false, false), 'serverSession', J$.T(290153, null, 25, false));
                                                    if (J$.C(16800, J$.B(18010, '===', J$.U(18002, 'typeof', J$.R(290169, 'callback', callback, false, false)), J$.T(290177, 'function', 21, false))))
                                                        J$.F(290209, J$.R(290185, 'callback', callback, false, false), false)(J$.T(290193, null, 25, false), J$.T(290201, null, 25, false));
                                                } catch (J$e) {
                                                    J$.Ex(300321, J$e);
                                                } finally {
                                                    if (J$.Fr(300329))
                                                        continue jalangiLabel16;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(290417, {
                                    key: J$.T(290265, 'advanceOperationTime', 21, false),
                                    value: J$.T(290409, function advanceOperationTime(operationTime) {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(290385, arguments.callee, this, arguments);
                                                    arguments = J$.N(290393, 'arguments', arguments, true, false, false);
                                                    operationTime = J$.N(290401, 'operationTime', operationTime, true, false, false);
                                                    if (J$.C(16808, J$.B(18018, '==', J$.G(290281, J$.R(290273, 'this', this, false, false), 'operationTime'), J$.T(290289, null, 25, false)))) {
                                                        J$.P(290313, J$.R(290297, 'this', this, false, false), 'operationTime', J$.R(290305, 'operationTime', operationTime, false, false));
                                                        return J$.Rt(290321, undefined);
                                                    }
                                                    if (J$.C(16816, J$.M(290353, J$.R(290329, 'operationTime', operationTime, false, false), 'greaterThan', false)(J$.G(290345, J$.R(290337, 'this', this, false, false), 'operationTime')))) {
                                                        J$.P(290377, J$.R(290361, 'this', this, false, false), 'operationTime', J$.R(290369, 'operationTime', operationTime, false, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(300337, J$e);
                                                } finally {
                                                    if (J$.Fr(300345))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(290577, {
                                    key: J$.T(290425, 'equals', 21, false),
                                    value: J$.T(290569, function equals(session) {
                                        jalangiLabel18:
                                            while (true) {
                                                try {
                                                    J$.Fe(290545, arguments.callee, this, arguments);
                                                    arguments = J$.N(290553, 'arguments', arguments, true, false, false);
                                                    session = J$.N(290561, 'session', session, true, false, false);
                                                    if (J$.C(16824, J$.U(18034, '!', J$.B(18026, 'instanceof', J$.R(290433, 'session', session, false, false), J$.R(290441, 'ClientSession', ClientSession, false, false))))) {
                                                        return J$.Rt(290457, J$.T(290449, false, 23, false));
                                                    }
                                                    return J$.Rt(290537, J$.M(290529, J$.G(290489, J$.G(290481, J$.G(290473, J$.R(290465, 'this', this, false, false), 'id'), 'id'), 'buffer'), 'equals', false)(J$.G(290521, J$.G(290513, J$.G(290505, J$.R(290497, 'session', session, false, false), 'id'), 'id'), 'buffer')));
                                                } catch (J$e) {
                                                    J$.Ex(300353, J$e);
                                                } finally {
                                                    if (J$.Fr(300361))
                                                        continue jalangiLabel18;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(290641, {
                                    key: J$.T(290585, 'incrementTransactionNumber', 21, false),
                                    value: J$.T(290633, function incrementTransactionNumber() {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(290617, arguments.callee, this, arguments);
                                                    arguments = J$.N(290625, 'arguments', arguments, true, false, false);
                                                    J$.B(18042, '-', J$.A(290609, J$.G(290601, J$.R(290593, 'this', this, false, false), 'serverSession'), 'txnNumber', '+')(1), 1);
                                                } catch (J$e) {
                                                    J$.Ex(300369, J$e);
                                                } finally {
                                                    if (J$.Fr(300377))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(290713, {
                                    key: J$.T(290649, 'inTransaction', 21, false),
                                    value: J$.T(290705, function inTransaction() {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(290689, arguments.callee, this, arguments);
                                                    arguments = J$.N(290697, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(290681, J$.G(290673, J$.G(290665, J$.R(290657, 'this', this, false, false), 'transaction'), 'isActive'));
                                                } catch (J$e) {
                                                    J$.Ex(300385, J$e);
                                                } finally {
                                                    if (J$.Fr(300393))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(291113, {
                                    key: J$.T(290721, 'startTransaction', 21, false),
                                    value: J$.T(291105, function startTransaction(options) {
                                        jalangiLabel21:
                                            while (true) {
                                                try {
                                                    J$.Fe(291073, arguments.callee, this, arguments);
                                                    arguments = J$.N(291081, 'arguments', arguments, true, false, false);
                                                    options = J$.N(291089, 'options', options, true, false, false);
                                                    J$.N(291097, 'topologyMaxWireVersion', topologyMaxWireVersion, false, false, false);
                                                    J$.F(290745, J$.R(290729, 'assertAlive', assertAlive, false, true), false)(J$.R(290737, 'this', this, false, false));
                                                    if (J$.C(16832, J$.M(290761, J$.R(290753, 'this', this, false, false), 'inTransaction', false)())) {
                                                        throw J$.F(290785, J$.R(290769, 'MongoError', MongoError, false, true), true)(J$.T(290777, 'Transaction already in progress', 21, false));
                                                    }
                                                    var topologyMaxWireVersion = J$.W(290825, 'topologyMaxWireVersion', J$.F(290817, J$.R(290793, 'maxWireVersion', maxWireVersion, false, true), false)(J$.G(290809, J$.R(290801, 'this', this, false, false), 'topology')), topologyMaxWireVersion, false, false);
                                                    if (J$.C(16856, J$.C(16848, J$.C(16840, J$.F(290857, J$.R(290833, 'isSharded', isSharded, false, true), false)(J$.G(290849, J$.R(290841, 'this', this, false, false), 'topology'))) ? J$.B(18050, '!=', J$.R(290865, 'topologyMaxWireVersion', topologyMaxWireVersion, false, false), J$.T(290873, null, 25, false)) : J$._()) ? J$.B(18058, '<', J$.R(290881, 'topologyMaxWireVersion', topologyMaxWireVersion, false, false), J$.R(290889, 'minWireVersionForShardedTransactions', minWireVersionForShardedTransactions, false, true)) : J$._())) {
                                                        throw J$.F(290913, J$.R(290897, 'MongoError', MongoError, false, true), true)(J$.T(290905, 'Transactions are not supported on sharded clusters in MongoDB < 4.2.', 21, false));
                                                    }
                                                    J$.M(290929, J$.R(290921, 'this', this, false, false), 'incrementTransactionNumber', false)();
                                                    J$.P(291025, J$.R(290937, 'this', this, false, false), 'transaction', J$.F(291017, J$.R(290945, 'Transaction', Transaction, false, true), true)(J$.M(291009, J$.I(typeof Object === 'undefined' ? Object = J$.R(290953, 'Object', undefined, true, true) : Object = J$.R(290953, 'Object', Object, true, true)), 'assign', false)(J$.T(290961, {}, 11, false), J$.G(290977, J$.R(290969, 'this', this, false, false), 'clientOptions'), J$.C(16864, J$.R(290985, 'options', options, false, false)) ? J$._() : J$.G(291001, J$.R(290993, 'this', this, false, false), 'defaultTransactionOptions'))));
                                                    J$.M(291065, J$.G(291041, J$.R(291033, 'this', this, false, false), 'transaction'), 'transition', false)(J$.G(291057, J$.R(291049, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION'));
                                                } catch (J$e) {
                                                    J$.Ex(300401, J$e);
                                                } finally {
                                                    if (J$.Fr(300409))
                                                        continue jalangiLabel21;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(291449, {
                                    key: J$.T(291121, 'commitTransaction', 21, false),
                                    value: J$.T(291441, function commitTransaction(callback) {
                                        jalangiLabel24:
                                            while (true) {
                                                try {
                                                    J$.Fe(291409, arguments.callee, this, arguments);
                                                    arguments = J$.N(291417, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(291425, 'callback', callback, true, false, false);
                                                    J$.N(291433, '_this2', _this2, false, false, false);
                                                    var _this2 = J$.W(291137, '_this2', J$.R(291129, 'this', this, false, false), _this2, false, false);
                                                    if (J$.C(16872, J$.B(18074, '===', J$.U(18066, 'typeof', J$.R(291145, 'callback', callback, false, false)), J$.T(291153, 'function', 21, false)))) {
                                                        J$.F(291193, J$.R(291161, 'endTransaction', endTransaction, false, true), false)(J$.R(291169, 'this', this, false, false), J$.T(291177, 'commitTransaction', 21, false), J$.R(291185, 'callback', callback, false, false));
                                                        return J$.Rt(291201, undefined);
                                                    }
                                                    return J$.Rt(291401, J$.F(291393, J$.I(typeof Promise === 'undefined' ? Promise = J$.R(291209, 'Promise', undefined, true, true) : Promise = J$.R(291209, 'Promise', Promise, true, true)), true)(J$.T(291385, function (resolve, reject) {
                                                        jalangiLabel23:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(291353, arguments.callee, this, arguments);
                                                                    arguments = J$.N(291361, 'arguments', arguments, true, false, false);
                                                                    resolve = J$.N(291369, 'resolve', resolve, true, false, false);
                                                                    reject = J$.N(291377, 'reject', reject, true, false, false);
                                                                    J$.F(291345, J$.R(291217, 'endTransaction', endTransaction, false, true), false)(J$.R(291225, '_this2', _this2, false, false), J$.T(291233, 'commitTransaction', 21, false), J$.T(291337, function (err, reply) {
                                                                        jalangiLabel22:
                                                                            while (true) {
                                                                                try {
                                                                                    J$.Fe(291305, arguments.callee, this, arguments);
                                                                                    arguments = J$.N(291313, 'arguments', arguments, true, false, false);
                                                                                    err = J$.N(291321, 'err', err, true, false, false);
                                                                                    reply = J$.N(291329, 'reply', reply, true, false, false);
                                                                                    return J$.Rt(291297, J$.C(16880, J$.R(291241, 'err', err, false, false)) ? J$.F(291265, J$.R(291249, 'reject', reject, false, false), false)(J$.R(291257, 'err', err, false, false)) : J$.F(291289, J$.R(291273, 'resolve', resolve, false, false), false)(J$.R(291281, 'reply', reply, false, false)));
                                                                                } catch (J$e) {
                                                                                    J$.Ex(300417, J$e);
                                                                                } finally {
                                                                                    if (J$.Fr(300425))
                                                                                        continue jalangiLabel22;
                                                                                    else
                                                                                        return J$.Ra();
                                                                                }
                                                                            }
                                                                    }, 12, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(300433, J$e);
                                                                } finally {
                                                                    if (J$.Fr(300441))
                                                                        continue jalangiLabel23;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false)));
                                                } catch (J$e) {
                                                    J$.Ex(300449, J$e);
                                                } finally {
                                                    if (J$.Fr(300457))
                                                        continue jalangiLabel24;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(291785, {
                                    key: J$.T(291457, 'abortTransaction', 21, false),
                                    value: J$.T(291777, function abortTransaction(callback) {
                                        jalangiLabel27:
                                            while (true) {
                                                try {
                                                    J$.Fe(291745, arguments.callee, this, arguments);
                                                    arguments = J$.N(291753, 'arguments', arguments, true, false, false);
                                                    callback = J$.N(291761, 'callback', callback, true, false, false);
                                                    J$.N(291769, '_this3', _this3, false, false, false);
                                                    var _this3 = J$.W(291473, '_this3', J$.R(291465, 'this', this, false, false), _this3, false, false);
                                                    if (J$.C(16888, J$.B(18090, '===', J$.U(18082, 'typeof', J$.R(291481, 'callback', callback, false, false)), J$.T(291489, 'function', 21, false)))) {
                                                        J$.F(291529, J$.R(291497, 'endTransaction', endTransaction, false, true), false)(J$.R(291505, 'this', this, false, false), J$.T(291513, 'abortTransaction', 21, false), J$.R(291521, 'callback', callback, false, false));
                                                        return J$.Rt(291537, undefined);
                                                    }
                                                    return J$.Rt(291737, J$.F(291729, J$.I(typeof Promise === 'undefined' ? Promise = J$.R(291545, 'Promise', undefined, true, true) : Promise = J$.R(291545, 'Promise', Promise, true, true)), true)(J$.T(291721, function (resolve, reject) {
                                                        jalangiLabel26:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(291689, arguments.callee, this, arguments);
                                                                    arguments = J$.N(291697, 'arguments', arguments, true, false, false);
                                                                    resolve = J$.N(291705, 'resolve', resolve, true, false, false);
                                                                    reject = J$.N(291713, 'reject', reject, true, false, false);
                                                                    J$.F(291681, J$.R(291553, 'endTransaction', endTransaction, false, true), false)(J$.R(291561, '_this3', _this3, false, false), J$.T(291569, 'abortTransaction', 21, false), J$.T(291673, function (err, reply) {
                                                                        jalangiLabel25:
                                                                            while (true) {
                                                                                try {
                                                                                    J$.Fe(291641, arguments.callee, this, arguments);
                                                                                    arguments = J$.N(291649, 'arguments', arguments, true, false, false);
                                                                                    err = J$.N(291657, 'err', err, true, false, false);
                                                                                    reply = J$.N(291665, 'reply', reply, true, false, false);
                                                                                    return J$.Rt(291633, J$.C(16896, J$.R(291577, 'err', err, false, false)) ? J$.F(291601, J$.R(291585, 'reject', reject, false, false), false)(J$.R(291593, 'err', err, false, false)) : J$.F(291625, J$.R(291609, 'resolve', resolve, false, false), false)(J$.R(291617, 'reply', reply, false, false)));
                                                                                } catch (J$e) {
                                                                                    J$.Ex(300465, J$e);
                                                                                } finally {
                                                                                    if (J$.Fr(300473))
                                                                                        continue jalangiLabel25;
                                                                                    else
                                                                                        return J$.Ra();
                                                                                }
                                                                            }
                                                                    }, 12, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(300481, J$e);
                                                                } finally {
                                                                    if (J$.Fr(300489))
                                                                        continue jalangiLabel26;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false)));
                                                } catch (J$e) {
                                                    J$.Ex(300497, J$e);
                                                } finally {
                                                    if (J$.Fr(300505))
                                                        continue jalangiLabel27;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(291849, {
                                    key: J$.T(291793, 'toBSON', 21, false),
                                    value: J$.T(291841, function toBSON() {
                                        jalangiLabel28:
                                            while (true) {
                                                try {
                                                    J$.Fe(291825, arguments.callee, this, arguments);
                                                    arguments = J$.N(291833, 'arguments', arguments, true, false, false);
                                                    throw J$.F(291817, J$.I(typeof Error === 'undefined' ? Error = J$.R(291801, 'Error', undefined, true, true) : Error = J$.R(291801, 'Error', Error, true, true)), true)(J$.T(291809, 'ClientSession cannot be serialized to BSON.', 21, false));
                                                } catch (J$e) {
                                                    J$.Ex(300513, J$e);
                                                } finally {
                                                    if (J$.Fr(300521))
                                                        continue jalangiLabel28;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(291993, {
                                    key: J$.T(291857, 'withTransaction', 21, false),
                                    value: J$.T(291985, function withTransaction(fn, options) {
                                        jalangiLabel29:
                                            while (true) {
                                                try {
                                                    J$.Fe(291945, arguments.callee, this, arguments);
                                                    arguments = J$.N(291953, 'arguments', arguments, true, false, false);
                                                    fn = J$.N(291961, 'fn', fn, true, false, false);
                                                    options = J$.N(291969, 'options', options, true, false, false);
                                                    J$.N(291977, 'startTime', startTime, false, false, false);
                                                    var startTime = J$.W(291881, 'startTime', J$.M(291873, J$.I(typeof Date === 'undefined' ? Date = J$.R(291865, 'Date', undefined, true, true) : Date = J$.R(291865, 'Date', Date, true, true)), 'now', false)(), startTime, false, false);
                                                    return J$.Rt(291937, J$.F(291929, J$.R(291889, 'attemptTransaction', attemptTransaction, false, true), false)(J$.R(291897, 'this', this, false, false), J$.R(291905, 'startTime', startTime, false, false), J$.R(291913, 'fn', fn, false, false), J$.R(291921, 'options', options, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(300529, J$e);
                                                } finally {
                                                    if (J$.Fr(300537))
                                                        continue jalangiLabel29;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(292065, {
                                    key: J$.T(292001, 'id', 21, false),
                                    get: J$.T(292057, function get() {
                                        jalangiLabel30:
                                            while (true) {
                                                try {
                                                    J$.Fe(292041, arguments.callee, this, arguments);
                                                    arguments = J$.N(292049, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(292033, J$.G(292025, J$.G(292017, J$.R(292009, 'this', this, false, false), 'serverSession'), 'id'));
                                                } catch (J$e) {
                                                    J$.Ex(300545, J$e);
                                                } finally {
                                                    if (J$.Fr(300553))
                                                        continue jalangiLabel30;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(292097, J$.R(292089, 'ClientSession', ClientSession, false, false));
                        } catch (J$e) {
                            J$.Ex(300561, J$e);
                        } finally {
                            if (J$.Fr(300569))
                                continue jalangiLabel31;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(292153, 'EventEmitter', EventEmitter, false, true)), ClientSession, false, true);
            var MAX_WITH_TRANSACTION_TIMEOUT = J$.W(292185, 'MAX_WITH_TRANSACTION_TIMEOUT', J$.T(292177, 120000, 22, false), MAX_WITH_TRANSACTION_TIMEOUT, false, true);
            var UNSATISFIABLE_WRITE_CONCERN_CODE = J$.W(292201, 'UNSATISFIABLE_WRITE_CONCERN_CODE', J$.T(292193, 100, 22, false), UNSATISFIABLE_WRITE_CONCERN_CODE, false, true);
            var UNKNOWN_REPL_WRITE_CONCERN_CODE = J$.W(292217, 'UNKNOWN_REPL_WRITE_CONCERN_CODE', J$.T(292209, 79, 22, false), UNKNOWN_REPL_WRITE_CONCERN_CODE, false, true);
            var MAX_TIME_MS_EXPIRED_CODE = J$.W(292233, 'MAX_TIME_MS_EXPIRED_CODE', J$.T(292225, 50, 22, false), MAX_TIME_MS_EXPIRED_CODE, false, true);
            var NON_DETERMINISTIC_WRITE_CONCERN_ERRORS = J$.W(292289, 'NON_DETERMINISTIC_WRITE_CONCERN_ERRORS', J$.F(292281, J$.I(typeof Set === 'undefined' ? Set = J$.R(292241, 'Set', undefined, true, true) : Set = J$.R(292241, 'Set', Set, true, true)), true)(J$.T(292273, [
                J$.T(292249, 'CannotSatisfyWriteConcern', 21, false),
                J$.T(292257, 'UnknownReplWriteConcern', 21, false),
                J$.T(292265, 'UnsatisfiableWriteConcern', 21, false)
            ], 10, false)), NON_DETERMINISTIC_WRITE_CONCERN_ERRORS, false, true);
            var USER_EXPLICIT_TXN_END_STATES = J$.W(293041, 'USER_EXPLICIT_TXN_END_STATES', J$.F(293033, J$.I(typeof Set === 'undefined' ? Set = J$.R(292969, 'Set', undefined, true, true) : Set = J$.R(292969, 'Set', Set, true, true)), true)(J$.T(293025, [
                J$.G(292985, J$.R(292977, 'TxnState', TxnState, false, true), 'NO_TRANSACTION'),
                J$.G(293001, J$.R(292993, 'TxnState', TxnState, false, true), 'TRANSACTION_COMMITTED'),
                J$.G(293017, J$.R(293009, 'TxnState', TxnState, false, true), 'TRANSACTION_ABORTED')
            ], 10, false)), USER_EXPLICIT_TXN_END_STATES, false, true);
            var ServerSession = J$.W(296785, 'ServerSession', J$.F(296777, J$.T(296769, function () {
                jalangiLabel51:
                    while (true) {
                        try {
                            J$.Fe(296737, arguments.callee, this, arguments);
                            function ServerSession() {
                                jalangiLabel49:
                                    while (true) {
                                        try {
                                            J$.Fe(296505, arguments.callee, this, arguments);
                                            arguments = J$.N(296513, 'arguments', arguments, true, false, false);
                                            J$.F(296345, J$.R(296321, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(296329, 'this', this, false, false), J$.R(296337, 'ServerSession', ServerSession, false, false));
                                            J$.P(296417, J$.R(296353, 'this', this, false, false), 'id', J$.T(296409, { id: J$.F(296401, J$.R(296361, 'Binary', Binary, false, true), true)(J$.F(296377, J$.R(296369, 'uuidV4', uuidV4, false, true), false)(), J$.G(296393, J$.R(296385, 'Binary', Binary, false, true), 'SUBTYPE_UUID')) }, 11, false));
                                            J$.P(296449, J$.R(296425, 'this', this, false, false), 'lastUse', J$.M(296441, J$.I(typeof Date === 'undefined' ? Date = J$.R(296433, 'Date', undefined, true, true) : Date = J$.R(296433, 'Date', Date, true, true)), 'now', false)());
                                            J$.P(296473, J$.R(296457, 'this', this, false, false), 'txnNumber', J$.T(296465, 0, 22, false));
                                            J$.P(296497, J$.R(296481, 'this', this, false, false), 'isDirty', J$.T(296489, false, 23, false));
                                        } catch (J$e) {
                                            J$.Ex(300849, J$e);
                                        } finally {
                                            if (J$.Fr(300857))
                                                continue jalangiLabel49;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(296745, 'arguments', arguments, true, false, false);
                            ServerSession = J$.N(296761, 'ServerSession', J$.T(296753, ServerSession, 12, false), true, false, false);
                            J$.F(296713, J$.R(296521, '_createClass', _createClass, false, true), false)(J$.R(296529, 'ServerSession', ServerSession, false, false), J$.T(296705, [J$.T(296697, {
                                    key: J$.T(296537, 'hasTimedOut', 21, false),
                                    value: J$.T(296689, function hasTimedOut(sessionTimeoutMinutes) {
                                        jalangiLabel50:
                                            while (true) {
                                                try {
                                                    J$.Fe(296657, arguments.callee, this, arguments);
                                                    arguments = J$.N(296665, 'arguments', arguments, true, false, false);
                                                    sessionTimeoutMinutes = J$.N(296673, 'sessionTimeoutMinutes', sessionTimeoutMinutes, true, false, false);
                                                    J$.N(296681, 'idleTimeMinutes', idleTimeMinutes, false, false, false);
                                                    var idleTimeMinutes = J$.W(296617, 'idleTimeMinutes', J$.M(296609, J$.I(typeof Math === 'undefined' ? Math = J$.R(296545, 'Math', undefined, true, true) : Math = J$.R(296545, 'Math', Math, true, true)), 'round', false)(J$.B(18378, '/', J$.B(18370, '%', J$.B(18362, '%', J$.B(18354, '-', J$.M(296561, J$.I(typeof Date === 'undefined' ? Date = J$.R(296553, 'Date', undefined, true, true) : Date = J$.R(296553, 'Date', Date, true, true)), 'now', false)(), J$.G(296577, J$.R(296569, 'this', this, false, false), 'lastUse')), J$.T(296585, 86400000, 22, false)), J$.T(296593, 3600000, 22, false)), J$.T(296601, 60000, 22, false))), idleTimeMinutes, false, false);
                                                    return J$.Rt(296649, J$.B(18394, '>', J$.R(296625, 'idleTimeMinutes', idleTimeMinutes, false, false), J$.B(18386, '-', J$.R(296633, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false), J$.T(296641, 1, 22, false))));
                                                } catch (J$e) {
                                                    J$.Ex(300865, J$e);
                                                } finally {
                                                    if (J$.Fr(300873))
                                                        continue jalangiLabel50;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false));
                            return J$.Rt(296729, J$.R(296721, 'ServerSession', ServerSession, false, false));
                        } catch (J$e) {
                            J$.Ex(300881, J$e);
                        } finally {
                            if (J$.Fr(300889))
                                continue jalangiLabel51;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), ServerSession, false, true);
            var ServerSessionPool = J$.W(297745, 'ServerSessionPool', J$.F(297737, J$.T(297729, function () {
                jalangiLabel57:
                    while (true) {
                        try {
                            J$.Fe(297697, arguments.callee, this, arguments);
                            function ServerSessionPool(topology) {
                                jalangiLabel52:
                                    while (true) {
                                        try {
                                            J$.Fe(296913, arguments.callee, this, arguments);
                                            arguments = J$.N(296921, 'arguments', arguments, true, false, false);
                                            topology = J$.N(296929, 'topology', topology, true, false, false);
                                            J$.F(296817, J$.R(296793, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(296801, 'this', this, false, false), J$.R(296809, 'ServerSessionPool', ServerSessionPool, false, false));
                                            if (J$.C(17304, J$.B(18402, '==', J$.R(296825, 'topology', topology, false, false), J$.T(296833, null, 25, false)))) {
                                                throw J$.F(296857, J$.I(typeof Error === 'undefined' ? Error = J$.R(296841, 'Error', undefined, true, true) : Error = J$.R(296841, 'Error', Error, true, true)), true)(J$.T(296849, 'ServerSessionPool requires a topology', 21, false));
                                            }
                                            J$.P(296881, J$.R(296865, 'this', this, false, false), 'topology', J$.R(296873, 'topology', topology, false, false));
                                            J$.P(296905, J$.R(296889, 'this', this, false, false), 'sessions', J$.T(296897, [], 10, false));
                                        } catch (J$e) {
                                            J$.Ex(300897, J$e);
                                        } finally {
                                            if (J$.Fr(300905))
                                                continue jalangiLabel52;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(297705, 'arguments', arguments, true, false, false);
                            ServerSessionPool = J$.N(297721, 'ServerSessionPool', J$.T(297713, ServerSessionPool, 12, false), true, false, false);
                            J$.F(297673, J$.R(296937, '_createClass', _createClass, false, true), false)(J$.R(296945, 'ServerSessionPool', ServerSessionPool, false, false), J$.T(297665, [
                                J$.T(297137, {
                                    key: J$.T(296953, 'endAllPooledSessions', 21, false),
                                    value: J$.T(297129, function endAllPooledSessions() {
                                        jalangiLabel54:
                                            while (true) {
                                                try {
                                                    J$.Fe(297113, arguments.callee, this, arguments);
                                                    arguments = J$.N(297121, 'arguments', arguments, true, false, false);
                                                    if (J$.C(17312, J$.G(296977, J$.G(296969, J$.R(296961, 'this', this, false, false), 'sessions'), 'length'))) {
                                                        J$.M(297081, J$.G(296993, J$.R(296985, 'this', this, false, false), 'topology'), 'endSessions', false)(J$.M(297073, J$.G(297009, J$.R(297001, 'this', this, false, false), 'sessions'), 'map', false)(J$.T(297065, function (session) {
                                                            jalangiLabel53:
                                                                while (true) {
                                                                    try {
                                                                        J$.Fe(297041, arguments.callee, this, arguments);
                                                                        arguments = J$.N(297049, 'arguments', arguments, true, false, false);
                                                                        session = J$.N(297057, 'session', session, true, false, false);
                                                                        return J$.Rt(297033, J$.G(297025, J$.R(297017, 'session', session, false, false), 'id'));
                                                                    } catch (J$e) {
                                                                        J$.Ex(300913, J$e);
                                                                    } finally {
                                                                        if (J$.Fr(300921))
                                                                            continue jalangiLabel53;
                                                                        else
                                                                            return J$.Ra();
                                                                    }
                                                                }
                                                        }, 12, false)));
                                                        J$.P(297105, J$.R(297089, 'this', this, false, false), 'sessions', J$.T(297097, [], 10, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(300929, J$e);
                                                } finally {
                                                    if (J$.Fr(300937))
                                                        continue jalangiLabel54;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(297345, {
                                    key: J$.T(297145, 'acquire', 21, false),
                                    value: J$.T(297337, function acquire() {
                                        jalangiLabel55:
                                            while (true) {
                                                try {
                                                    J$.Fe(297305, arguments.callee, this, arguments);
                                                    arguments = J$.N(297313, 'arguments', arguments, true, false, false);
                                                    J$.N(297321, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false, false);
                                                    J$.N(297329, 'session', session, false, false, false);
                                                    var sessionTimeoutMinutes = J$.W(297177, 'sessionTimeoutMinutes', J$.G(297169, J$.G(297161, J$.R(297153, 'this', this, false, false), 'topology'), 'logicalSessionTimeoutMinutes'), sessionTimeoutMinutes, false, false);
                                                    while (J$.C(17328, J$.G(297201, J$.G(297193, J$.R(297185, 'this', this, false, false), 'sessions'), 'length'))) {
                                                        var session = J$.W(297233, 'session', J$.M(297225, J$.G(297217, J$.R(297209, 'this', this, false, false), 'sessions'), 'shift', false)(), session, false, false);
                                                        if (J$.C(17320, J$.U(18410, '!', J$.M(297257, J$.R(297241, 'session', session, false, false), 'hasTimedOut', false)(J$.R(297249, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false))))) {
                                                            return J$.Rt(297273, J$.R(297265, 'session', session, false, false));
                                                        }
                                                    }
                                                    return J$.Rt(297297, J$.F(297289, J$.R(297281, 'ServerSession', ServerSession, false, true), true)());
                                                } catch (J$e) {
                                                    J$.Ex(300945, J$e);
                                                } finally {
                                                    if (J$.Fr(300953))
                                                        continue jalangiLabel55;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(297657, {
                                    key: J$.T(297353, 'release', 21, false),
                                    value: J$.T(297649, function release(session) {
                                        jalangiLabel56:
                                            while (true) {
                                                try {
                                                    J$.Fe(297609, arguments.callee, this, arguments);
                                                    arguments = J$.N(297617, 'arguments', arguments, true, false, false);
                                                    session = J$.N(297625, 'session', session, true, false, false);
                                                    J$.N(297633, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false, false);
                                                    J$.N(297641, 'pooledSession', pooledSession, false, false, false);
                                                    var sessionTimeoutMinutes = J$.W(297385, 'sessionTimeoutMinutes', J$.G(297377, J$.G(297369, J$.R(297361, 'this', this, false, false), 'topology'), 'logicalSessionTimeoutMinutes'), sessionTimeoutMinutes, false, false);
                                                    while (J$.C(17344, J$.G(297409, J$.G(297401, J$.R(297393, 'this', this, false, false), 'sessions'), 'length'))) {
                                                        var pooledSession = J$.W(297473, 'pooledSession', J$.G(297465, J$.G(297425, J$.R(297417, 'this', this, false, false), 'sessions'), J$.B(18418, '-', J$.G(297449, J$.G(297441, J$.R(297433, 'this', this, false, false), 'sessions'), 'length'), J$.T(297457, 1, 22, false))), pooledSession, false, false);
                                                        if (J$.C(17336, J$.M(297497, J$.R(297481, 'pooledSession', pooledSession, false, false), 'hasTimedOut', false)(J$.R(297489, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false)))) {
                                                            J$.M(297521, J$.G(297513, J$.R(297505, 'this', this, false, false), 'sessions'), 'pop', false)();
                                                        } else {
                                                            break;
                                                        }
                                                    }
                                                    if (J$.C(17360, J$.U(18426, '!', J$.M(297545, J$.R(297529, 'session', session, false, false), 'hasTimedOut', false)(J$.R(297537, 'sessionTimeoutMinutes', sessionTimeoutMinutes, false, false))))) {
                                                        if (J$.C(17352, J$.G(297561, J$.R(297553, 'session', session, false, false), 'isDirty'))) {
                                                            return J$.Rt(297569, undefined);
                                                        }
                                                        J$.M(297601, J$.G(297585, J$.R(297577, 'this', this, false, false), 'sessions'), 'unshift', false)(J$.R(297593, 'session', session, false, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(300961, J$e);
                                                } finally {
                                                    if (J$.Fr(300969))
                                                        continue jalangiLabel56;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(297689, J$.R(297681, 'ServerSessionPool', ServerSessionPool, false, false));
                        } catch (J$e) {
                            J$.Ex(300977, J$e);
                        } finally {
                            if (J$.Fr(300985))
                                continue jalangiLabel57;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), ServerSessionPool, false, true);
            J$.P(299481, J$.I(typeof module === 'undefined' ? module = J$.R(299409, 'module', undefined, true, true) : module = J$.R(299409, 'module', module, true, true)), 'exports', J$.T(299473, {
                ClientSession: J$.R(299417, 'ClientSession', ClientSession, false, true),
                ServerSession: J$.R(299425, 'ServerSession', ServerSession, false, true),
                ServerSessionPool: J$.R(299433, 'ServerSessionPool', ServerSessionPool, false, true),
                TxnState: J$.R(299441, 'TxnState', TxnState, false, true),
                applySession: J$.R(299449, 'applySession', applySession, false, true),
                updateSessionFromResponse: J$.R(299457, 'updateSessionFromResponse', updateSessionFromResponse, false, true),
                commandSupportsReadConcern: J$.R(299465, 'commandSupportsReadConcern', commandSupportsReadConcern, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(301041, J$e);
        } finally {
            if (J$.Sr(301049))
                continue jalangiLabel61;
            else
                break jalangiLabel61;
        }
    }
// JALANGI DO NOT INSTRUMENT
