// mongodb/lib/core/transactions.js
J$.noInstrEval = false;
jalangiLabel15:
    while (true) {
        try {
            J$.Se(396913, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongodb/lib/core/transactions.js');
            function _classCallCheck(instance, Constructor) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(393985, arguments.callee, this, arguments);
                            arguments = J$.N(393993, 'arguments', arguments, true, false, false);
                            instance = J$.N(394001, 'instance', instance, true, false, false);
                            Constructor = J$.N(394009, 'Constructor', Constructor, true, false, false);
                            if (J$.C(24320, J$.U(24498, '!', J$.B(24490, 'instanceof', J$.R(393945, 'instance', instance, false, false), J$.R(393953, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(393977, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(393961, 'TypeError', undefined, true, true) : TypeError = J$.R(393961, 'TypeError', TypeError, true, true)), true)(J$.T(393969, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(397033, J$e);
                        } finally {
                            if (J$.Fr(397041))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(394257, arguments.callee, this, arguments);
                            arguments = J$.N(394265, 'arguments', arguments, true, false, false);
                            target = J$.N(394273, 'target', target, true, false, false);
                            props = J$.N(394281, 'props', props, true, false, false);
                            J$.N(394289, 'i', i, false, false, false);
                            J$.N(394297, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(394025, 'i', J$.T(394017, 0, 22, false), i, false, false); J$.C(24344, J$.B(24506, '<', J$.R(394033, 'i', i, false, false), J$.G(394049, J$.R(394041, 'props', props, false, false), 'length'))); J$.B(24530, '-', i = J$.W(394065, 'i', J$.B(24522, '+', J$.U(24514, '+', J$.R(394057, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(394097, 'descriptor', J$.G(394089, J$.R(394073, 'props', props, false, false), J$.R(394081, 'i', i, false, false)), descriptor, false, false);
                                J$.P(394137, J$.R(394105, 'descriptor', descriptor, false, false), 'enumerable', J$.C(24328, J$.G(394121, J$.R(394113, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(394129, false, 23, false));
                                J$.P(394161, J$.R(394145, 'descriptor', descriptor, false, false), 'configurable', J$.T(394153, true, 23, false));
                                if (J$.C(24336, J$.B(24538, 'in', J$.T(394169, 'value', 21, false), J$.R(394177, 'descriptor', descriptor, false, false))))
                                    J$.P(394201, J$.R(394185, 'descriptor', descriptor, false, false), 'writable', J$.T(394193, true, 23, false));
                                J$.M(394249, J$.I(typeof Object === 'undefined' ? Object = J$.R(394209, 'Object', undefined, true, true) : Object = J$.R(394209, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(394217, 'target', target, false, false), J$.G(394233, J$.R(394225, 'descriptor', descriptor, false, false), 'key'), J$.R(394241, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(397049, J$e);
                        } finally {
                            if (J$.Fr(397057))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(394409, arguments.callee, this, arguments);
                            arguments = J$.N(394417, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(394425, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(394433, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(394441, 'staticProps', staticProps, true, false, false);
                            if (J$.C(24352, J$.R(394305, 'protoProps', protoProps, false, false)))
                                J$.F(394345, J$.R(394313, '_defineProperties', _defineProperties, false, true), false)(J$.G(394329, J$.R(394321, 'Constructor', Constructor, false, false), 'prototype'), J$.R(394337, 'protoProps', protoProps, false, false));
                            if (J$.C(24360, J$.R(394353, 'staticProps', staticProps, false, false)))
                                J$.F(394385, J$.R(394361, '_defineProperties', _defineProperties, false, true), false)(J$.R(394369, 'Constructor', Constructor, false, false), J$.R(394377, 'staticProps', staticProps, false, false));
                            return J$.Rt(394401, J$.R(394393, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(397065, J$e);
                        } finally {
                            if (J$.Fr(397073))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperty(obj, key, value) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(394585, arguments.callee, this, arguments);
                            arguments = J$.N(394593, 'arguments', arguments, true, false, false);
                            obj = J$.N(394601, 'obj', obj, true, false, false);
                            key = J$.N(394609, 'key', key, true, false, false);
                            value = J$.N(394617, 'value', value, true, false, false);
                            if (J$.C(24368, J$.B(24546, 'in', J$.R(394449, 'key', key, false, false), J$.R(394457, 'obj', obj, false, false)))) {
                                J$.M(394529, J$.I(typeof Object === 'undefined' ? Object = J$.R(394465, 'Object', undefined, true, true) : Object = J$.R(394465, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(394473, 'obj', obj, false, false), J$.R(394481, 'key', key, false, false), J$.T(394521, {
                                    value: J$.R(394489, 'value', value, false, false),
                                    enumerable: J$.T(394497, true, 23, false),
                                    configurable: J$.T(394505, true, 23, false),
                                    writable: J$.T(394513, true, 23, false)
                                }, 11, false));
                            } else {
                                J$.P(394561, J$.R(394537, 'obj', obj, false, false), J$.R(394545, 'key', key, false, false), J$.R(394553, 'value', value, false, false));
                            }
                            return J$.Rt(394577, J$.R(394569, 'obj', obj, false, false));
                        } catch (J$e) {
                            J$.Ex(397081, J$e);
                        } finally {
                            if (J$.Fr(397089))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isTransactionCommand(command) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(396841, arguments.callee, this, arguments);
                            arguments = J$.N(396849, 'arguments', arguments, true, false, false);
                            command = J$.N(396857, 'command', command, true, false, false);
                            return J$.Rt(396833, J$.U(24650, '!', J$.U(24642, '!', J$.C(24488, J$.G(396809, J$.R(396801, 'command', command, false, false), 'commitTransaction')) ? J$._() : J$.G(396825, J$.R(396817, 'command', command, false, false), 'abortTransaction'))));
                        } catch (J$e) {
                            J$.Ex(397257, J$e);
                        } finally {
                            if (J$.Fr(397265))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _classCallCheck = J$.N(396929, '_classCallCheck', J$.T(396921, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(396945, '_defineProperties', J$.T(396937, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(396961, '_createClass', J$.T(396953, _createClass, 12, false), true, false, false);
            _defineProperty = J$.N(396977, '_defineProperty', J$.T(396969, _defineProperty, 12, false), true, false, false);
            J$.N(396985, 'MongoError', MongoError, false, false, false);
            J$.N(396993, 'TxnState', TxnState, false, false, false);
            J$.N(397001, 'stateMachine', stateMachine, false, false, false);
            J$.N(397009, 'Transaction', Transaction, false, false, false);
            isTransactionCommand = J$.N(397025, 'isTransactionCommand', J$.T(397017, isTransactionCommand, 12, false), true, false, false);
            J$.T(393937, 'use strict', 21, false);
            var MongoError = J$.W(394657, 'MongoError', J$.G(394649, J$.F(394641, J$.I(typeof require === 'undefined' ? require = J$.R(394625, 'require', undefined, true, true) : require = J$.R(394625, 'require', require, true, true)), false)(J$.T(394633, './error', 21, false)), 'MongoError'), MongoError, false, true);
            var TxnState;
            var stateMachine;
            J$.F(395313, J$.T(395305, function () {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(395233, arguments.callee, this, arguments);
                            arguments = J$.N(395241, 'arguments', arguments, true, false, false);
                            J$.N(395249, '_stateMachine', _stateMachine, false, false, false);
                            J$.N(395257, 'NO_TRANSACTION', NO_TRANSACTION, false, false, false);
                            J$.N(395265, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false, false);
                            J$.N(395273, 'TRANSACTION_IN_PROGRESS', TRANSACTION_IN_PROGRESS, false, false, false);
                            J$.N(395281, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false, false);
                            J$.N(395289, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false, false);
                            J$.N(395297, 'TRANSACTION_ABORTED', TRANSACTION_ABORTED, false, false, false);
                            var _stateMachine;
                            var NO_TRANSACTION = J$.W(394673, 'NO_TRANSACTION', J$.T(394665, 'NO_TRANSACTION', 21, false), NO_TRANSACTION, false, false);
                            var STARTING_TRANSACTION = J$.W(394689, 'STARTING_TRANSACTION', J$.T(394681, 'STARTING_TRANSACTION', 21, false), STARTING_TRANSACTION, false, false);
                            var TRANSACTION_IN_PROGRESS = J$.W(394705, 'TRANSACTION_IN_PROGRESS', J$.T(394697, 'TRANSACTION_IN_PROGRESS', 21, false), TRANSACTION_IN_PROGRESS, false, false);
                            var TRANSACTION_COMMITTED = J$.W(394721, 'TRANSACTION_COMMITTED', J$.T(394713, 'TRANSACTION_COMMITTED', 21, false), TRANSACTION_COMMITTED, false, false);
                            var TRANSACTION_COMMITTED_EMPTY = J$.W(394737, 'TRANSACTION_COMMITTED_EMPTY', J$.T(394729, 'TRANSACTION_COMMITTED_EMPTY', 21, false), TRANSACTION_COMMITTED_EMPTY, false, false);
                            var TRANSACTION_ABORTED = J$.W(394753, 'TRANSACTION_ABORTED', J$.T(394745, 'TRANSACTION_ABORTED', 21, false), TRANSACTION_ABORTED, false, false);
                            TxnState = J$.W(394817, 'TxnState', J$.T(394809, {
                                NO_TRANSACTION: J$.R(394761, 'NO_TRANSACTION', NO_TRANSACTION, false, false),
                                STARTING_TRANSACTION: J$.R(394769, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false),
                                TRANSACTION_IN_PROGRESS: J$.R(394777, 'TRANSACTION_IN_PROGRESS', TRANSACTION_IN_PROGRESS, false, false),
                                TRANSACTION_COMMITTED: J$.R(394785, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false),
                                TRANSACTION_COMMITTED_EMPTY: J$.R(394793, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false),
                                TRANSACTION_ABORTED: J$.R(394801, 'TRANSACTION_ABORTED', TRANSACTION_ABORTED, false, false)
                            }, 11, false), TxnState, false, true);
                            stateMachine = J$.W(395225, 'stateMachine', (_stateMachine = J$.W(394833, '_stateMachine', J$.T(394825, {}, 11, false), _stateMachine, false, false), J$.F(394889, J$.R(394841, '_defineProperty', _defineProperty, false, true), false)(J$.R(394849, '_stateMachine', _stateMachine, false, false), J$.R(394857, 'NO_TRANSACTION', NO_TRANSACTION, false, false), J$.T(394881, [
                                J$.R(394865, 'NO_TRANSACTION', NO_TRANSACTION, false, false),
                                J$.R(394873, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false)
                            ], 10, false)), J$.F(394961, J$.R(394897, '_defineProperty', _defineProperty, false, true), false)(J$.R(394905, '_stateMachine', _stateMachine, false, false), J$.R(394913, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false), J$.T(394953, [
                                J$.R(394921, 'TRANSACTION_IN_PROGRESS', TRANSACTION_IN_PROGRESS, false, false),
                                J$.R(394929, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false),
                                J$.R(394937, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false),
                                J$.R(394945, 'TRANSACTION_ABORTED', TRANSACTION_ABORTED, false, false)
                            ], 10, false)), J$.F(395025, J$.R(394969, '_defineProperty', _defineProperty, false, true), false)(J$.R(394977, '_stateMachine', _stateMachine, false, false), J$.R(394985, 'TRANSACTION_IN_PROGRESS', TRANSACTION_IN_PROGRESS, false, false), J$.T(395017, [
                                J$.R(394993, 'TRANSACTION_IN_PROGRESS', TRANSACTION_IN_PROGRESS, false, false),
                                J$.R(395001, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false),
                                J$.R(395009, 'TRANSACTION_ABORTED', TRANSACTION_ABORTED, false, false)
                            ], 10, false)), J$.F(395097, J$.R(395033, '_defineProperty', _defineProperty, false, true), false)(J$.R(395041, '_stateMachine', _stateMachine, false, false), J$.R(395049, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false), J$.T(395089, [
                                J$.R(395057, 'TRANSACTION_COMMITTED', TRANSACTION_COMMITTED, false, false),
                                J$.R(395065, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false),
                                J$.R(395073, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false),
                                J$.R(395081, 'NO_TRANSACTION', NO_TRANSACTION, false, false)
                            ], 10, false)), J$.F(395153, J$.R(395105, '_defineProperty', _defineProperty, false, true), false)(J$.R(395113, '_stateMachine', _stateMachine, false, false), J$.R(395121, 'TRANSACTION_ABORTED', TRANSACTION_ABORTED, false, false), J$.T(395145, [
                                J$.R(395129, 'STARTING_TRANSACTION', STARTING_TRANSACTION, false, false),
                                J$.R(395137, 'NO_TRANSACTION', NO_TRANSACTION, false, false)
                            ], 10, false)), J$.F(395209, J$.R(395161, '_defineProperty', _defineProperty, false, true), false)(J$.R(395169, '_stateMachine', _stateMachine, false, false), J$.R(395177, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false), J$.T(395201, [
                                J$.R(395185, 'TRANSACTION_COMMITTED_EMPTY', TRANSACTION_COMMITTED_EMPTY, false, false),
                                J$.R(395193, 'NO_TRANSACTION', NO_TRANSACTION, false, false)
                            ], 10, false)), J$.R(395217, '_stateMachine', _stateMachine, false, false)), stateMachine, false, true);
                        } catch (J$e) {
                            J$.Ex(397097, J$e);
                        } finally {
                            if (J$.Fr(397105))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)();
            var Transaction = J$.W(396793, 'Transaction', J$.F(396785, J$.T(396777, function () {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(396745, arguments.callee, this, arguments);
                            function Transaction(options) {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(395873, arguments.callee, this, arguments);
                                            arguments = J$.N(395881, 'arguments', arguments, true, false, false);
                                            options = J$.N(395889, 'options', options, true, false, false);
                                            J$.N(395897, 'w', w, false, false, false);
                                            J$.F(395345, J$.R(395321, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(395329, 'this', this, false, false), J$.R(395337, 'Transaction', Transaction, false, false));
                                            options = J$.W(395369, 'options', J$.C(24376, J$.R(395353, 'options', options, false, false)) ? J$._() : J$.T(395361, {}, 11, false), options, false, false);
                                            J$.P(395401, J$.R(395377, 'this', this, false, false), 'state', J$.G(395393, J$.R(395385, 'TxnState', TxnState, false, true), 'NO_TRANSACTION'));
                                            J$.P(395425, J$.R(395409, 'this', this, false, false), 'options', J$.T(395417, {}, 11, false));
                                            if (J$.C(24416, J$.C(24384, J$.G(395441, J$.R(395433, 'options', options, false, false), 'writeConcern')) ? J$._() : J$.B(24562, '!==', J$.U(24554, 'typeof', J$.G(395457, J$.R(395449, 'options', options, false, false), 'w')), J$.T(395465, 'undefined', 21, false)))) {
                                                var w = J$.W(395529, 'w', J$.C(24392, J$.G(395481, J$.R(395473, 'options', options, false, false), 'writeConcern')) ? J$.G(395505, J$.G(395497, J$.R(395489, 'options', options, false, false), 'writeConcern'), 'w') : J$.G(395521, J$.R(395513, 'options', options, false, false), 'w'), w, false, false);
                                                if (J$.C(24400, J$.B(24570, '<=', J$.R(395537, 'w', w, false, false), J$.T(395545, 0, 22, false)))) {
                                                    throw J$.F(395569, J$.R(395553, 'MongoError', MongoError, false, true), true)(J$.T(395561, 'Transactions do not support unacknowledged write concern', 21, false));
                                                }
                                                J$.P(395649, J$.G(395585, J$.R(395577, 'this', this, false, false), 'options'), 'writeConcern', J$.C(24408, J$.G(395601, J$.R(395593, 'options', options, false, false), 'writeConcern')) ? J$.G(395617, J$.R(395609, 'options', options, false, false), 'writeConcern') : J$.T(395641, { w: J$.G(395633, J$.R(395625, 'options', options, false, false), 'w') }, 11, false));
                                            }
                                            if (J$.C(24424, J$.G(395665, J$.R(395657, 'options', options, false, false), 'readConcern')))
                                                J$.P(395705, J$.G(395681, J$.R(395673, 'this', this, false, false), 'options'), 'readConcern', J$.G(395697, J$.R(395689, 'options', options, false, false), 'readConcern'));
                                            if (J$.C(24432, J$.G(395721, J$.R(395713, 'options', options, false, false), 'readPreference')))
                                                J$.P(395761, J$.G(395737, J$.R(395729, 'this', this, false, false), 'options'), 'readPreference', J$.G(395753, J$.R(395745, 'options', options, false, false), 'readPreference'));
                                            if (J$.C(24440, J$.G(395777, J$.R(395769, 'options', options, false, false), 'maxCommitTimeMS')))
                                                J$.P(395817, J$.G(395793, J$.R(395785, 'this', this, false, false), 'options'), 'maxTimeMS', J$.G(395809, J$.R(395801, 'options', options, false, false), 'maxCommitTimeMS'));
                                            J$.P(395841, J$.R(395825, 'this', this, false, false), '_pinnedServer', J$.T(395833, undefined, 24, false));
                                            J$.P(395865, J$.R(395849, 'this', this, false, false), '_recoveryToken', J$.T(395857, undefined, 24, false));
                                        } catch (J$e) {
                                            J$.Ex(397113, J$e);
                                        } finally {
                                            if (J$.Fr(397121))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(396753, 'arguments', arguments, true, false, false);
                            Transaction = J$.N(396769, 'Transaction', J$.T(396761, Transaction, 12, false), true, false, false);
                            J$.F(396721, J$.R(395905, '_createClass', _createClass, false, true), false)(J$.R(395913, 'Transaction', Transaction, false, false), J$.T(396713, [
                                J$.T(396241, {
                                    key: J$.T(395921, 'transition', 21, false),
                                    value: J$.T(396233, function transition(nextState) {
                                        jalangiLabel6:
                                            while (true) {
                                                try {
                                                    J$.Fe(396201, arguments.callee, this, arguments);
                                                    arguments = J$.N(396209, 'arguments', arguments, true, false, false);
                                                    nextState = J$.N(396217, 'nextState', nextState, true, false, false);
                                                    J$.N(396225, 'nextStates', nextStates, false, false, false);
                                                    var nextStates = J$.W(395961, 'nextStates', J$.G(395953, J$.R(395929, 'stateMachine', stateMachine, false, true), J$.G(395945, J$.R(395937, 'this', this, false, false), 'state')), nextStates, false, false);
                                                    if (J$.C(24472, J$.C(24448, J$.R(395969, 'nextStates', nextStates, false, false)) ? J$.B(24586, '!==', J$.M(395993, J$.R(395977, 'nextStates', nextStates, false, false), 'indexOf', false)(J$.R(395985, 'nextState', nextState, false, false)), J$.U(24578, '-', J$.T(396001, 1, 22, false))) : J$._())) {
                                                        J$.P(396025, J$.R(396009, 'this', this, false, false), 'state', J$.R(396017, 'nextState', nextState, false, false));
                                                        if (J$.C(24464, J$.C(24456, J$.B(24594, '===', J$.G(396041, J$.R(396033, 'this', this, false, false), 'state'), J$.G(396057, J$.R(396049, 'TxnState', TxnState, false, true), 'NO_TRANSACTION'))) ? J$._() : J$.B(24602, '===', J$.G(396073, J$.R(396065, 'this', this, false, false), 'state'), J$.G(396089, J$.R(396081, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION')))) {
                                                            J$.M(396105, J$.R(396097, 'this', this, false, false), 'unpinServer', false)();
                                                        }
                                                        return J$.Rt(396113, undefined);
                                                    }
                                                    throw J$.F(396193, J$.R(396121, 'MongoError', MongoError, false, true), true)(J$.M(396185, J$.M(396161, J$.T(396129, 'Attempted illegal state transition from [', 21, false), 'concat', false)(J$.G(396145, J$.R(396137, 'this', this, false, false), 'state'), J$.T(396153, '] to [', 21, false)), 'concat', false)(J$.R(396169, 'nextState', nextState, false, false), J$.T(396177, ']', 21, false)));
                                                } catch (J$e) {
                                                    J$.Ex(397129, J$e);
                                                } finally {
                                                    if (J$.Fr(397137))
                                                        continue jalangiLabel6;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(396329, {
                                    key: J$.T(396249, 'pinServer', 21, false),
                                    value: J$.T(396321, function pinServer(server) {
                                        jalangiLabel7:
                                            while (true) {
                                                try {
                                                    J$.Fe(396297, arguments.callee, this, arguments);
                                                    arguments = J$.N(396305, 'arguments', arguments, true, false, false);
                                                    server = J$.N(396313, 'server', server, true, false, false);
                                                    if (J$.C(24480, J$.G(396265, J$.R(396257, 'this', this, false, false), 'isActive'))) {
                                                        J$.P(396289, J$.R(396273, 'this', this, false, false), '_pinnedServer', J$.R(396281, 'server', server, false, false));
                                                    }
                                                } catch (J$e) {
                                                    J$.Ex(397145, J$e);
                                                } finally {
                                                    if (J$.Fr(397153))
                                                        continue jalangiLabel7;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(396393, {
                                    key: J$.T(396337, 'unpinServer', 21, false),
                                    value: J$.T(396385, function unpinServer() {
                                        jalangiLabel8:
                                            while (true) {
                                                try {
                                                    J$.Fe(396369, arguments.callee, this, arguments);
                                                    arguments = J$.N(396377, 'arguments', arguments, true, false, false);
                                                    J$.P(396361, J$.R(396345, 'this', this, false, false), '_pinnedServer', J$.T(396353, undefined, 24, false));
                                                } catch (J$e) {
                                                    J$.Ex(397161, J$e);
                                                } finally {
                                                    if (J$.Fr(397169))
                                                        continue jalangiLabel8;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(396457, {
                                    key: J$.T(396401, 'server', 21, false),
                                    get: J$.T(396449, function get() {
                                        jalangiLabel9:
                                            while (true) {
                                                try {
                                                    J$.Fe(396433, arguments.callee, this, arguments);
                                                    arguments = J$.N(396441, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(396425, J$.G(396417, J$.R(396409, 'this', this, false, false), '_pinnedServer'));
                                                } catch (J$e) {
                                                    J$.Ex(397177, J$e);
                                                } finally {
                                                    if (J$.Fr(397185))
                                                        continue jalangiLabel9;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(396521, {
                                    key: J$.T(396465, 'recoveryToken', 21, false),
                                    get: J$.T(396513, function get() {
                                        jalangiLabel10:
                                            while (true) {
                                                try {
                                                    J$.Fe(396497, arguments.callee, this, arguments);
                                                    arguments = J$.N(396505, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(396489, J$.G(396481, J$.R(396473, 'this', this, false, false), '_recoveryToken'));
                                                } catch (J$e) {
                                                    J$.Ex(397193, J$e);
                                                } finally {
                                                    if (J$.Fr(397201))
                                                        continue jalangiLabel10;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(396585, {
                                    key: J$.T(396529, 'isPinned', 21, false),
                                    get: J$.T(396577, function get() {
                                        jalangiLabel11:
                                            while (true) {
                                                try {
                                                    J$.Fe(396561, arguments.callee, this, arguments);
                                                    arguments = J$.N(396569, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(396553, J$.U(24618, '!', J$.U(24610, '!', J$.G(396545, J$.R(396537, 'this', this, false, false), 'server'))));
                                                } catch (J$e) {
                                                    J$.Ex(397209, J$e);
                                                } finally {
                                                    if (J$.Fr(397217))
                                                        continue jalangiLabel11;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(396705, {
                                    key: J$.T(396593, 'isActive', 21, false),
                                    get: J$.T(396697, function get() {
                                        jalangiLabel12:
                                            while (true) {
                                                try {
                                                    J$.Fe(396681, arguments.callee, this, arguments);
                                                    arguments = J$.N(396689, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(396673, J$.B(24634, '!==', J$.M(396657, J$.T(396633, [
                                                        J$.G(396609, J$.R(396601, 'TxnState', TxnState, false, true), 'STARTING_TRANSACTION'),
                                                        J$.G(396625, J$.R(396617, 'TxnState', TxnState, false, true), 'TRANSACTION_IN_PROGRESS')
                                                    ], 10, false), 'indexOf', false)(J$.G(396649, J$.R(396641, 'this', this, false, false), 'state')), J$.U(24626, '-', J$.T(396665, 1, 22, false))));
                                                } catch (J$e) {
                                                    J$.Ex(397225, J$e);
                                                } finally {
                                                    if (J$.Fr(397233))
                                                        continue jalangiLabel12;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(396737, J$.R(396729, 'Transaction', Transaction, false, false));
                        } catch (J$e) {
                            J$.Ex(397241, J$e);
                        } finally {
                            if (J$.Fr(397249))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), Transaction, false, true);
            J$.P(396905, J$.I(typeof module === 'undefined' ? module = J$.R(396865, 'module', undefined, true, true) : module = J$.R(396865, 'module', module, true, true)), 'exports', J$.T(396897, {
                TxnState: J$.R(396873, 'TxnState', TxnState, false, true),
                Transaction: J$.R(396881, 'Transaction', Transaction, false, true),
                isTransactionCommand: J$.R(396889, 'isTransactionCommand', isTransactionCommand, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(397273, J$e);
        } finally {
            if (J$.Sr(397281))
                continue jalangiLabel15;
            else
                break jalangiLabel15;
        }
    }
// JALANGI DO NOT INSTRUMENT
