// verror/lib/verror.js
J$.noInstrEval = false;
jalangiLabel23:
    while (true) {
        try {
            J$.Se(3984777, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestRequest/node_modules/verror/lib/verror.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(3978697, arguments.callee, this, arguments);
                            arguments = J$.N(3978705, 'arguments', arguments, true, false, false);
                            obj = J$.N(3978713, 'obj', obj, true, false, false);
                            if (J$.C(178160, J$.C(178120, J$.B(547138, '===', J$.U(547130, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(3978433, 'Symbol', undefined, true, true) : Symbol = J$.R(3978433, 'Symbol', Symbol, true, true))), J$.T(3978441, 'function', 21, false))) ? J$.B(547154, '===', J$.U(547146, 'typeof', J$.G(3978457, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(3978449, 'Symbol', undefined, true, true) : Symbol = J$.R(3978449, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(3978465, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(3978521, '_typeof', J$.T(3978513, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(3978489, arguments.callee, this, arguments);
                                                arguments = J$.N(3978497, 'arguments', arguments, true, false, false);
                                                obj = J$.N(3978505, 'obj', obj, true, false, false);
                                                return J$.Rt(3978481, J$.U(547162, 'typeof', J$.R(3978473, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(3984921, J$e);
                                            } finally {
                                                if (J$.Fr(3984929))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(3978657, '_typeof', J$.T(3978649, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(3978625, arguments.callee, this, arguments);
                                                arguments = J$.N(3978633, 'arguments', arguments, true, false, false);
                                                obj = J$.N(3978641, 'obj', obj, true, false, false);
                                                return J$.Rt(3978617, J$.C(178152, J$.C(178144, J$.C(178136, J$.C(178128, J$.R(3978529, 'obj', obj, false, false)) ? J$.B(547178, '===', J$.U(547170, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(3978537, 'Symbol', undefined, true, true) : Symbol = J$.R(3978537, 'Symbol', Symbol, true, true))), J$.T(3978545, 'function', 21, false)) : J$._()) ? J$.B(547186, '===', J$.G(3978561, J$.R(3978553, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(3978569, 'Symbol', undefined, true, true) : Symbol = J$.R(3978569, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(547194, '!==', J$.R(3978577, 'obj', obj, false, false), J$.G(3978593, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(3978585, 'Symbol', undefined, true, true) : Symbol = J$.R(3978585, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(3978601, 'symbol', 21, false) : J$.U(547202, 'typeof', J$.R(3978609, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(3984937, J$e);
                                            } finally {
                                                if (J$.Fr(3984945))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(3978689, J$.F(3978681, J$.R(3978665, '_typeof', _typeof, false, true), false)(J$.R(3978673, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(3984953, J$e);
                        } finally {
                            if (J$.Fr(3984961))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function parseConstructorArguments(args) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(3979881, arguments.callee, this, arguments);
                            arguments = J$.N(3979889, 'arguments', arguments, true, false, false);
                            args = J$.N(3979897, 'args', args, true, false, false);
                            J$.N(3979905, 'argv', argv, false, false, false);
                            J$.N(3979913, 'options', options, false, false, false);
                            J$.N(3979921, 'sprintf_args', sprintf_args, false, false, false);
                            J$.N(3979929, 'shortmessage', shortmessage, false, false, false);
                            J$.N(3979937, 'k', k, false, false, false);
                            var argv, options, sprintf_args, shortmessage, k;
                            J$.M(3979025, J$.R(3979001, 'mod_assertplus', mod_assertplus, false, true), 'object', false)(J$.R(3979009, 'args', args, false, false), J$.T(3979017, 'args', 21, false));
                            J$.M(3979065, J$.R(3979033, 'mod_assertplus', mod_assertplus, false, true), 'bool', false)(J$.G(3979049, J$.R(3979041, 'args', args, false, false), 'strict'), J$.T(3979057, 'args.strict', 21, false));
                            J$.M(3979105, J$.R(3979073, 'mod_assertplus', mod_assertplus, false, true), 'array', false)(J$.G(3979089, J$.R(3979081, 'args', args, false, false), 'argv'), J$.T(3979097, 'args.argv', 21, false));
                            argv = J$.W(3979129, 'argv', J$.G(3979121, J$.R(3979113, 'args', args, false, false), 'argv'), argv, false, false);
                            if (J$.C(178184, J$.B(547210, '===', J$.G(3979145, J$.R(3979137, 'argv', argv, false, false), 'length'), J$.T(3979153, 0, 22, false)))) {
                                options = J$.W(3979169, 'options', J$.T(3979161, {}, 11, false), options, false, false);
                                sprintf_args = J$.W(3979185, 'sprintf_args', J$.T(3979177, [], 10, false), sprintf_args, false, false);
                            } else if (J$.C(178176, J$.F(3979225, J$.R(3979193, 'mod_isError', mod_isError, false, true), false)(J$.G(3979217, J$.R(3979201, 'argv', argv, false, false), J$.T(3979209, 0, 22, false))))) {
                                options = J$.W(3979265, 'options', J$.T(3979257, { 'cause': J$.G(3979249, J$.R(3979233, 'argv', argv, false, false), J$.T(3979241, 0, 22, false)) }, 11, false), options, false, false);
                                sprintf_args = J$.W(3979297, 'sprintf_args', J$.M(3979289, J$.R(3979273, 'argv', argv, false, false), 'slice', false)(J$.T(3979281, 1, 22, false)), sprintf_args, false, false);
                            } else if (J$.C(178168, J$.B(547218, '===', J$.F(3979337, J$.R(3979305, '_typeof', _typeof, false, true), false)(J$.G(3979329, J$.R(3979313, 'argv', argv, false, false), J$.T(3979321, 0, 22, false))), J$.T(3979345, 'object', 21, false)))) {
                                options = J$.W(3979361, 'options', J$.T(3979353, {}, 11, false), options, false, false);
                                for (k in J$.H(3979457, J$.G(3979385, J$.R(3979369, 'argv', argv, false, false), J$.T(3979377, 0, 22, false)))) {
                                    J$.N(3979465, 'k', k, false, true, false);
                                    {
                                        {
                                            J$.P(3979449, J$.R(3979393, 'options', options, false, false), J$.R(3979401, 'k', k, false, false), J$.G(3979441, J$.G(3979425, J$.R(3979409, 'argv', argv, false, false), J$.T(3979417, 0, 22, false)), J$.R(3979433, 'k', k, false, false)));
                                        }
                                    }
                                }
                                sprintf_args = J$.W(3979497, 'sprintf_args', J$.M(3979489, J$.R(3979473, 'argv', argv, false, false), 'slice', false)(J$.T(3979481, 1, 22, false)), sprintf_args, false, false);
                            } else {
                                J$.M(3979553, J$.R(3979505, 'mod_assertplus', mod_assertplus, false, true), 'string', false)(J$.G(3979529, J$.R(3979513, 'argv', argv, false, false), J$.T(3979521, 0, 22, false)), J$.B(547226, '+', J$.T(3979537, 'first argument to VError, SError, or WError ', 21, false), J$.T(3979545, 'constructor must be a string, object, or Error', 21, false)));
                                options = J$.W(3979569, 'options', J$.T(3979561, {}, 11, false), options, false, false);
                                sprintf_args = J$.W(3979585, 'sprintf_args', J$.R(3979577, 'argv', argv, false, false), sprintf_args, false, false);
                            }
                            J$.M(3979609, J$.R(3979593, 'mod_assertplus', mod_assertplus, false, true), 'object', false)(J$.R(3979601, 'options', options, false, false));
                            if (J$.C(178216, J$.C(178192, J$.U(547234, '!', J$.G(3979625, J$.R(3979617, 'options', options, false, false), 'strict'))) ? J$.U(547242, '!', J$.G(3979641, J$.R(3979633, 'args', args, false, false), 'strict')) : J$._())) {
                                sprintf_args = J$.W(3979761, 'sprintf_args', J$.M(3979753, J$.R(3979649, 'sprintf_args', sprintf_args, false, false), 'map', false)(J$.T(3979745, function (a) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(3979721, arguments.callee, this, arguments);
                                                arguments = J$.N(3979729, 'arguments', arguments, true, false, false);
                                                a = J$.N(3979737, 'a', a, true, false, false);
                                                return J$.Rt(3979713, J$.C(178208, J$.B(547250, '===', J$.R(3979657, 'a', a, false, false), J$.T(3979665, null, 25, false))) ? J$.T(3979673, 'null', 21, false) : J$.C(178200, J$.B(547258, '===', J$.R(3979681, 'a', a, false, false), J$.T(3979689, undefined, 24, false))) ? J$.T(3979697, 'undefined', 21, false) : J$.R(3979705, 'a', a, false, false));
                                            } catch (J$e) {
                                                J$.Ex(3984969, J$e);
                                            } finally {
                                                if (J$.Fr(3984977))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)), sprintf_args, false, false);
                            }
                            if (J$.C(178224, J$.B(547266, '===', J$.G(3979777, J$.R(3979769, 'sprintf_args', sprintf_args, false, false), 'length'), J$.T(3979785, 0, 22, false)))) {
                                shortmessage = J$.W(3979801, 'shortmessage', J$.T(3979793, '', 21, false), shortmessage, false, false);
                            } else {
                                shortmessage = J$.W(3979841, 'shortmessage', J$.M(3979833, J$.R(3979809, 'sprintf', sprintf, false, true), 'apply', false)(J$.T(3979817, null, 25, false), J$.R(3979825, 'sprintf_args', sprintf_args, false, false)), shortmessage, false, false);
                            }
                            return J$.Rt(3979873, J$.T(3979865, {
                                'options': J$.R(3979849, 'options', options, false, false),
                                'shortmessage': J$.R(3979857, 'shortmessage', shortmessage, false, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(3984985, J$e);
                        } finally {
                            if (J$.Fr(3984993))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function VError() {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(3980825, arguments.callee, this, arguments);
                            arguments = J$.N(3980833, 'arguments', arguments, true, false, false);
                            J$.N(3980841, 'args', args, false, false, false);
                            J$.N(3980849, 'obj', obj, false, false, false);
                            J$.N(3980857, 'parsed', parsed, false, false, false);
                            J$.N(3980865, 'cause', cause, false, false, false);
                            J$.N(3980873, 'ctor', ctor, false, false, false);
                            J$.N(3980881, 'message', message, false, false, false);
                            J$.N(3980889, 'k', k, false, false, false);
                            var args, obj, parsed, cause, ctor, message, k;
                            args = J$.W(3979993, 'args', J$.M(3979985, J$.G(3979961, J$.G(3979953, J$.I(typeof Array === 'undefined' ? Array = J$.R(3979945, 'Array', undefined, true, true) : Array = J$.R(3979945, 'Array', Array, true, true)), 'prototype'), 'slice'), 'call', false)(J$.I(typeof arguments === 'undefined' ? arguments = J$.R(3979969, 'arguments', undefined, true, true) : arguments = J$.R(3979969, 'arguments', arguments, true, true)), J$.T(3979977, 0, 22, false)), args, false, false);
                            if (J$.C(178232, J$.U(547282, '!', J$.B(547274, 'instanceof', J$.R(3980001, 'this', this, false, false), J$.R(3980009, 'VError', VError, false, true))))) {
                                obj = J$.W(3980049, 'obj', J$.M(3980041, J$.I(typeof Object === 'undefined' ? Object = J$.R(3980017, 'Object', undefined, true, true) : Object = J$.R(3980017, 'Object', Object, true, true)), 'create', false)(J$.G(3980033, J$.R(3980025, 'VError', VError, false, true), 'prototype')), obj, false, false);
                                J$.M(3980081, J$.R(3980057, 'VError', VError, false, true), 'apply', false)(J$.R(3980065, 'obj', obj, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(3980073, 'arguments', undefined, true, true) : arguments = J$.R(3980073, 'arguments', arguments, true, true)));
                                return J$.Rt(3980097, J$.R(3980089, 'obj', obj, false, false));
                            }
                            parsed = J$.W(3980145, 'parsed', J$.F(3980137, J$.R(3980105, 'parseConstructorArguments', parseConstructorArguments, false, true), false)(J$.T(3980129, {
                                'argv': J$.R(3980113, 'args', args, false, false),
                                'strict': J$.T(3980121, false, 23, false)
                            }, 11, false)), parsed, false, false);
                            if (J$.C(178240, J$.G(3980169, J$.G(3980161, J$.R(3980153, 'parsed', parsed, false, false), 'options'), 'name'))) {
                                J$.M(3980217, J$.R(3980177, 'mod_assertplus', mod_assertplus, false, true), 'string', false)(J$.G(3980201, J$.G(3980193, J$.R(3980185, 'parsed', parsed, false, false), 'options'), 'name'), J$.T(3980209, 'error\'s "name" must be a string', 21, false));
                                J$.P(3980257, J$.R(3980225, 'this', this, false, false), 'name', J$.G(3980249, J$.G(3980241, J$.R(3980233, 'parsed', parsed, false, false), 'options'), 'name'));
                            }
                            J$.P(3980289, J$.R(3980265, 'this', this, false, false), 'jse_shortmsg', J$.G(3980281, J$.R(3980273, 'parsed', parsed, false, false), 'shortmessage'));
                            message = J$.W(3980313, 'message', J$.G(3980305, J$.R(3980297, 'parsed', parsed, false, false), 'shortmessage'), message, false, false);
                            cause = J$.W(3980345, 'cause', J$.G(3980337, J$.G(3980329, J$.R(3980321, 'parsed', parsed, false, false), 'options'), 'cause'), cause, false, false);
                            if (J$.C(178256, J$.R(3980353, 'cause', cause, false, false))) {
                                J$.M(3980401, J$.R(3980361, 'mod_assertplus', mod_assertplus, false, true), 'ok', false)(J$.F(3980385, J$.R(3980369, 'mod_isError', mod_isError, false, true), false)(J$.R(3980377, 'cause', cause, false, false)), J$.T(3980393, 'cause is not an Error', 21, false));
                                J$.P(3980425, J$.R(3980409, 'this', this, false, false), 'jse_cause', J$.R(3980417, 'cause', cause, false, false));
                                if (J$.C(178248, J$.U(547290, '!', J$.G(3980449, J$.G(3980441, J$.R(3980433, 'parsed', parsed, false, false), 'options'), 'skipCauseMessage')))) {
                                    message = J$.W(3980489, 'message', J$.B(547306, '+', J$.R(3980481, 'message', message, false, false), J$.B(547298, '+', J$.T(3980457, ': ', 21, false), J$.G(3980473, J$.R(3980465, 'cause', cause, false, false), 'message'))), message, false, false);
                                }
                            }
                            J$.P(3980513, J$.R(3980497, 'this', this, false, false), 'jse_info', J$.T(3980505, {}, 11, false));
                            if (J$.C(178264, J$.G(3980537, J$.G(3980529, J$.R(3980521, 'parsed', parsed, false, false), 'options'), 'info'))) {
                                for (k in J$.H(3980641, J$.G(3980561, J$.G(3980553, J$.R(3980545, 'parsed', parsed, false, false), 'options'), 'info'))) {
                                    J$.N(3980649, 'k', k, false, true, false);
                                    {
                                        {
                                            J$.P(3980633, J$.G(3980577, J$.R(3980569, 'this', this, false, false), 'jse_info'), J$.R(3980585, 'k', k, false, false), J$.G(3980625, J$.G(3980609, J$.G(3980601, J$.R(3980593, 'parsed', parsed, false, false), 'options'), 'info'), J$.R(3980617, 'k', k, false, false)));
                                        }
                                    }
                                }
                            }
                            J$.P(3980673, J$.R(3980657, 'this', this, false, false), 'message', J$.R(3980665, 'message', message, false, false));
                            J$.M(3980705, J$.I(typeof Error === 'undefined' ? Error = J$.R(3980681, 'Error', undefined, true, true) : Error = J$.R(3980681, 'Error', Error, true, true)), 'call', false)(J$.R(3980689, 'this', this, false, false), J$.R(3980697, 'message', message, false, false));
                            if (J$.C(178280, J$.G(3980721, J$.I(typeof Error === 'undefined' ? Error = J$.R(3980713, 'Error', undefined, true, true) : Error = J$.R(3980713, 'Error', Error, true, true)), 'captureStackTrace'))) {
                                ctor = J$.W(3980769, 'ctor', J$.C(178272, J$.G(3980745, J$.G(3980737, J$.R(3980729, 'parsed', parsed, false, false), 'options'), 'constructorOpt')) ? J$._() : J$.G(3980761, J$.R(3980753, 'this', this, false, false), 'constructor'), ctor, false, false);
                                J$.M(3980801, J$.I(typeof Error === 'undefined' ? Error = J$.R(3980777, 'Error', undefined, true, true) : Error = J$.R(3980777, 'Error', Error, true, true)), 'captureStackTrace', false)(J$.R(3980785, 'this', this, false, false), J$.R(3980793, 'ctor', ctor, false, false));
                            }
                            return J$.Rt(3980817, J$.R(3980809, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(3985001, J$e);
                        } finally {
                            if (J$.Fr(3985009))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function SError() {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(3983401, arguments.callee, this, arguments);
                            arguments = J$.N(3983409, 'arguments', arguments, true, false, false);
                            J$.N(3983417, 'args', args, false, false, false);
                            J$.N(3983425, 'obj', obj, false, false, false);
                            J$.N(3983433, 'parsed', parsed, false, false, false);
                            J$.N(3983441, 'options', options, false, false, false);
                            var args, obj, parsed, options;
                            args = J$.W(3983145, 'args', J$.M(3983137, J$.G(3983113, J$.G(3983105, J$.I(typeof Array === 'undefined' ? Array = J$.R(3983097, 'Array', undefined, true, true) : Array = J$.R(3983097, 'Array', Array, true, true)), 'prototype'), 'slice'), 'call', false)(J$.I(typeof arguments === 'undefined' ? arguments = J$.R(3983121, 'arguments', undefined, true, true) : arguments = J$.R(3983121, 'arguments', arguments, true, true)), J$.T(3983129, 0, 22, false)), args, false, false);
                            if (J$.C(178408, J$.U(547442, '!', J$.B(547434, 'instanceof', J$.R(3983153, 'this', this, false, false), J$.R(3983161, 'SError', SError, false, true))))) {
                                obj = J$.W(3983201, 'obj', J$.M(3983193, J$.I(typeof Object === 'undefined' ? Object = J$.R(3983169, 'Object', undefined, true, true) : Object = J$.R(3983169, 'Object', Object, true, true)), 'create', false)(J$.G(3983185, J$.R(3983177, 'SError', SError, false, true), 'prototype')), obj, false, false);
                                J$.M(3983233, J$.R(3983209, 'SError', SError, false, true), 'apply', false)(J$.R(3983217, 'obj', obj, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(3983225, 'arguments', undefined, true, true) : arguments = J$.R(3983225, 'arguments', arguments, true, true)));
                                return J$.Rt(3983249, J$.R(3983241, 'obj', obj, false, false));
                            }
                            parsed = J$.W(3983297, 'parsed', J$.F(3983289, J$.R(3983257, 'parseConstructorArguments', parseConstructorArguments, false, true), false)(J$.T(3983281, {
                                'argv': J$.R(3983265, 'args', args, false, false),
                                'strict': J$.T(3983273, true, 23, false)
                            }, 11, false)), parsed, false, false);
                            options = J$.W(3983321, 'options', J$.G(3983313, J$.R(3983305, 'parsed', parsed, false, false), 'options'), options, false, false);
                            J$.M(3983377, J$.R(3983329, 'VError', VError, false, true), 'call', false)(J$.R(3983337, 'this', this, false, false), J$.R(3983345, 'options', options, false, false), J$.T(3983353, '%s', 21, false), J$.G(3983369, J$.R(3983361, 'parsed', parsed, false, false), 'shortmessage'));
                            return J$.Rt(3983393, J$.R(3983385, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(3985193, J$e);
                        } finally {
                            if (J$.Fr(3985201))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function MultiError(errors) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(3983705, arguments.callee, this, arguments);
                            arguments = J$.N(3983713, 'arguments', arguments, true, false, false);
                            errors = J$.N(3983721, 'errors', errors, true, false, false);
                            J$.M(3983505, J$.R(3983481, 'mod_assertplus', mod_assertplus, false, true), 'array', false)(J$.R(3983489, 'errors', errors, false, false), J$.T(3983497, 'list of errors', 21, false));
                            J$.M(3983553, J$.R(3983513, 'mod_assertplus', mod_assertplus, false, true), 'ok', false)(J$.B(547450, '>', J$.G(3983529, J$.R(3983521, 'errors', errors, false, false), 'length'), J$.T(3983537, 0, 22, false)), J$.T(3983545, 'must be at least one error', 21, false));
                            J$.P(3983577, J$.R(3983561, 'this', this, false, false), 'ase_errors', J$.R(3983569, 'errors', errors, false, false));
                            J$.M(3983697, J$.R(3983585, 'VError', VError, false, true), 'call', false)(J$.R(3983593, 'this', this, false, false), J$.T(3983625, { 'cause': J$.G(3983617, J$.R(3983601, 'errors', errors, false, false), J$.T(3983609, 0, 22, false)) }, 11, false), J$.T(3983633, 'first of %d error%s', 21, false), J$.G(3983649, J$.R(3983641, 'errors', errors, false, false), 'length'), J$.C(178416, J$.B(547458, '==', J$.G(3983665, J$.R(3983657, 'errors', errors, false, false), 'length'), J$.T(3983673, 1, 22, false))) ? J$.T(3983681, '', 21, false) : J$.T(3983689, 's', 21, false));
                        } catch (J$e) {
                            J$.Ex(3985209, J$e);
                        } finally {
                            if (J$.Fr(3985217))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function WError() {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(3984217, arguments.callee, this, arguments);
                            arguments = J$.N(3984225, 'arguments', arguments, true, false, false);
                            J$.N(3984233, 'args', args, false, false, false);
                            J$.N(3984241, 'obj', obj, false, false, false);
                            J$.N(3984249, 'parsed', parsed, false, false, false);
                            J$.N(3984257, 'options', options, false, false, false);
                            var args, obj, parsed, options;
                            args = J$.W(3983929, 'args', J$.M(3983921, J$.G(3983897, J$.G(3983889, J$.I(typeof Array === 'undefined' ? Array = J$.R(3983881, 'Array', undefined, true, true) : Array = J$.R(3983881, 'Array', Array, true, true)), 'prototype'), 'slice'), 'call', false)(J$.I(typeof arguments === 'undefined' ? arguments = J$.R(3983905, 'arguments', undefined, true, true) : arguments = J$.R(3983905, 'arguments', arguments, true, true)), J$.T(3983913, 0, 22, false)), args, false, false);
                            if (J$.C(178424, J$.U(547474, '!', J$.B(547466, 'instanceof', J$.R(3983937, 'this', this, false, false), J$.R(3983945, 'WError', WError, false, true))))) {
                                obj = J$.W(3983985, 'obj', J$.M(3983977, J$.I(typeof Object === 'undefined' ? Object = J$.R(3983953, 'Object', undefined, true, true) : Object = J$.R(3983953, 'Object', Object, true, true)), 'create', false)(J$.G(3983969, J$.R(3983961, 'WError', WError, false, true), 'prototype')), obj, false, false);
                                J$.M(3984017, J$.R(3983993, 'WError', WError, false, true), 'apply', false)(J$.R(3984001, 'obj', obj, false, false), J$.R(3984009, 'args', args, false, false));
                                return J$.Rt(3984033, J$.R(3984025, 'obj', obj, false, false));
                            }
                            parsed = J$.W(3984081, 'parsed', J$.F(3984073, J$.R(3984041, 'parseConstructorArguments', parseConstructorArguments, false, true), false)(J$.T(3984065, {
                                'argv': J$.R(3984049, 'args', args, false, false),
                                'strict': J$.T(3984057, false, 23, false)
                            }, 11, false)), parsed, false, false);
                            options = J$.W(3984105, 'options', J$.G(3984097, J$.R(3984089, 'parsed', parsed, false, false), 'options'), options, false, false);
                            J$.P(3984137, J$.R(3984113, 'options', options, false, false), J$.T(3984121, 'skipCauseMessage', 21, false), J$.T(3984129, true, 23, false));
                            J$.M(3984193, J$.R(3984145, 'VError', VError, false, true), 'call', false)(J$.R(3984153, 'this', this, false, false), J$.R(3984161, 'options', options, false, false), J$.T(3984169, '%s', 21, false), J$.G(3984185, J$.R(3984177, 'parsed', parsed, false, false), 'shortmessage'));
                            return J$.Rt(3984209, J$.R(3984201, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(3985241, J$e);
                        } finally {
                            if (J$.Fr(3985249))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(3984793, '_typeof', J$.T(3984785, _typeof, 12, false), true, false, false);
            J$.N(3984801, 'mod_assertplus', mod_assertplus, false, false, false);
            J$.N(3984809, 'mod_util', mod_util, false, false, false);
            J$.N(3984817, 'mod_extsprintf', mod_extsprintf, false, false, false);
            J$.N(3984825, 'mod_isError', mod_isError, false, false, false);
            J$.N(3984833, 'sprintf', sprintf, false, false, false);
            parseConstructorArguments = J$.N(3984849, 'parseConstructorArguments', J$.T(3984841, parseConstructorArguments, 12, false), true, false, false);
            VError = J$.N(3984865, 'VError', J$.T(3984857, VError, 12, false), true, false, false);
            SError = J$.N(3984881, 'SError', J$.T(3984873, SError, 12, false), true, false, false);
            MultiError = J$.N(3984897, 'MultiError', J$.T(3984889, MultiError, 12, false), true, false, false);
            WError = J$.N(3984913, 'WError', J$.T(3984905, WError, 12, false), true, false, false);
            J$.T(3978065, 'use strict', 21, false);
            J$.F(3978089, J$.I(typeof require === 'undefined' ? require = J$.R(3978073, 'require', undefined, true, true) : require = J$.R(3978073, 'require', require, true, true)), false)(J$.T(3978081, 'core-js/modules/es.symbol', 21, false));
            J$.F(3978113, J$.I(typeof require === 'undefined' ? require = J$.R(3978097, 'require', undefined, true, true) : require = J$.R(3978097, 'require', require, true, true)), false)(J$.T(3978105, 'core-js/modules/es.symbol.description', 21, false));
            J$.F(3978137, J$.I(typeof require === 'undefined' ? require = J$.R(3978121, 'require', undefined, true, true) : require = J$.R(3978121, 'require', require, true, true)), false)(J$.T(3978129, 'core-js/modules/es.symbol.iterator', 21, false));
            J$.F(3978161, J$.I(typeof require === 'undefined' ? require = J$.R(3978145, 'require', undefined, true, true) : require = J$.R(3978145, 'require', require, true, true)), false)(J$.T(3978153, 'core-js/modules/es.array.for-each', 21, false));
            J$.F(3978185, J$.I(typeof require === 'undefined' ? require = J$.R(3978169, 'require', undefined, true, true) : require = J$.R(3978169, 'require', require, true, true)), false)(J$.T(3978177, 'core-js/modules/es.array.iterator', 21, false));
            J$.F(3978209, J$.I(typeof require === 'undefined' ? require = J$.R(3978193, 'require', undefined, true, true) : require = J$.R(3978193, 'require', require, true, true)), false)(J$.T(3978201, 'core-js/modules/es.array.map', 21, false));
            J$.F(3978233, J$.I(typeof require === 'undefined' ? require = J$.R(3978217, 'require', undefined, true, true) : require = J$.R(3978217, 'require', require, true, true)), false)(J$.T(3978225, 'core-js/modules/es.array.slice', 21, false));
            J$.F(3978257, J$.I(typeof require === 'undefined' ? require = J$.R(3978241, 'require', undefined, true, true) : require = J$.R(3978241, 'require', require, true, true)), false)(J$.T(3978249, 'core-js/modules/es.date.to-string', 21, false));
            J$.F(3978281, J$.I(typeof require === 'undefined' ? require = J$.R(3978265, 'require', undefined, true, true) : require = J$.R(3978265, 'require', require, true, true)), false)(J$.T(3978273, 'core-js/modules/es.function.name', 21, false));
            J$.F(3978305, J$.I(typeof require === 'undefined' ? require = J$.R(3978289, 'require', undefined, true, true) : require = J$.R(3978289, 'require', require, true, true)), false)(J$.T(3978297, 'core-js/modules/es.object.create', 21, false));
            J$.F(3978329, J$.I(typeof require === 'undefined' ? require = J$.R(3978313, 'require', undefined, true, true) : require = J$.R(3978313, 'require', require, true, true)), false)(J$.T(3978321, 'core-js/modules/es.object.to-string', 21, false));
            J$.F(3978353, J$.I(typeof require === 'undefined' ? require = J$.R(3978337, 'require', undefined, true, true) : require = J$.R(3978337, 'require', require, true, true)), false)(J$.T(3978345, 'core-js/modules/es.regexp.to-string', 21, false));
            J$.F(3978377, J$.I(typeof require === 'undefined' ? require = J$.R(3978361, 'require', undefined, true, true) : require = J$.R(3978361, 'require', require, true, true)), false)(J$.T(3978369, 'core-js/modules/es.string.iterator', 21, false));
            J$.F(3978401, J$.I(typeof require === 'undefined' ? require = J$.R(3978385, 'require', undefined, true, true) : require = J$.R(3978385, 'require', require, true, true)), false)(J$.T(3978393, 'core-js/modules/web.dom-collections.for-each', 21, false));
            J$.F(3978425, J$.I(typeof require === 'undefined' ? require = J$.R(3978409, 'require', undefined, true, true) : require = J$.R(3978409, 'require', require, true, true)), false)(J$.T(3978417, 'core-js/modules/web.dom-collections.iterator', 21, false));
            var mod_assertplus = J$.W(3978745, 'mod_assertplus', J$.F(3978737, J$.I(typeof require === 'undefined' ? require = J$.R(3978721, 'require', undefined, true, true) : require = J$.R(3978721, 'require', require, true, true)), false)(J$.T(3978729, 'assert-plus', 21, false)), mod_assertplus, false, true);
            var mod_util = J$.W(3978777, 'mod_util', J$.F(3978769, J$.I(typeof require === 'undefined' ? require = J$.R(3978753, 'require', undefined, true, true) : require = J$.R(3978753, 'require', require, true, true)), false)(J$.T(3978761, 'util', 21, false)), mod_util, false, true);
            var mod_extsprintf = J$.W(3978809, 'mod_extsprintf', J$.F(3978801, J$.I(typeof require === 'undefined' ? require = J$.R(3978785, 'require', undefined, true, true) : require = J$.R(3978785, 'require', require, true, true)), false)(J$.T(3978793, 'extsprintf', 21, false)), mod_extsprintf, false, true);
            var mod_isError = J$.W(3978849, 'mod_isError', J$.G(3978841, J$.F(3978833, J$.I(typeof require === 'undefined' ? require = J$.R(3978817, 'require', undefined, true, true) : require = J$.R(3978817, 'require', require, true, true)), false)(J$.T(3978825, 'core-util-is', 21, false)), 'isError'), mod_isError, false, true);
            var sprintf = J$.W(3978873, 'sprintf', J$.G(3978865, J$.R(3978857, 'mod_extsprintf', mod_extsprintf, false, true), 'sprintf'), sprintf, false, true);
            J$.P(3978897, J$.I(typeof module === 'undefined' ? module = J$.R(3978881, 'module', undefined, true, true) : module = J$.R(3978881, 'module', module, true, true)), 'exports', J$.R(3978889, 'VError', VError, false, true));
            J$.P(3978921, J$.R(3978905, 'VError', VError, false, true), 'VError', J$.R(3978913, 'VError', VError, false, true));
            J$.P(3978945, J$.R(3978929, 'VError', VError, false, true), 'SError', J$.R(3978937, 'SError', SError, false, true));
            J$.P(3978969, J$.R(3978953, 'VError', VError, false, true), 'WError', J$.R(3978961, 'WError', WError, false, true));
            J$.P(3978993, J$.R(3978977, 'VError', VError, false, true), 'MultiError', J$.R(3978985, 'MultiError', MultiError, false, true));
            J$.M(3980921, J$.R(3980897, 'mod_util', mod_util, false, true), 'inherits', false)(J$.R(3980905, 'VError', VError, false, true), J$.I(typeof Error === 'undefined' ? Error = J$.R(3980913, 'Error', undefined, true, true) : Error = J$.R(3980913, 'Error', Error, true, true)));
            J$.P(3980953, J$.G(3980937, J$.R(3980929, 'VError', VError, false, true), 'prototype'), 'name', J$.T(3980945, 'VError', 21, false));
            J$.P(3981185, J$.G(3980969, J$.R(3980961, 'VError', VError, false, true), 'prototype'), 'toString', J$.T(3981177, function ve_toString() {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(3981153, arguments.callee, this, arguments);
                            arguments = J$.N(3981161, 'arguments', arguments, true, false, false);
                            J$.N(3981169, 'str', str, false, false, false);
                            var str = J$.W(3981073, 'str', J$.C(178304, J$.C(178296, J$.C(178288, J$.M(3980993, J$.R(3980977, 'this', this, false, false), 'hasOwnProperty', false)(J$.T(3980985, 'name', 21, false))) ? J$.G(3981009, J$.R(3981001, 'this', this, false, false), 'name') : J$._()) ? J$._() : J$.G(3981033, J$.G(3981025, J$.R(3981017, 'this', this, false, false), 'constructor'), 'name')) ? J$._() : J$.G(3981065, J$.G(3981057, J$.G(3981049, J$.R(3981041, 'this', this, false, false), 'constructor'), 'prototype'), 'name'), str, false, false);
                            if (J$.C(178312, J$.G(3981089, J$.R(3981081, 'this', this, false, false), 'message')))
                                str = J$.W(3981129, 'str', J$.B(547322, '+', J$.R(3981121, 'str', str, false, false), J$.B(547314, '+', J$.T(3981097, ': ', 21, false), J$.G(3981113, J$.R(3981105, 'this', this, false, false), 'message'))), str, false, false);
                            return J$.Rt(3981145, J$.R(3981137, 'str', str, false, false));
                        } catch (J$e) {
                            J$.Ex(3985017, J$e);
                        } finally {
                            if (J$.Fr(3985025))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(3981313, J$.G(3981201, J$.R(3981193, 'VError', VError, false, true), 'prototype'), 'cause', J$.T(3981305, function ve_cause() {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(3981281, arguments.callee, this, arguments);
                            arguments = J$.N(3981289, 'arguments', arguments, true, false, false);
                            J$.N(3981297, 'cause', cause, false, false, false);
                            var cause = J$.W(3981233, 'cause', J$.M(3981225, J$.R(3981209, 'VError', VError, false, true), 'cause', false)(J$.R(3981217, 'this', this, false, false)), cause, false, false);
                            return J$.Rt(3981273, J$.C(178320, J$.B(547330, '===', J$.R(3981241, 'cause', cause, false, false), J$.T(3981249, null, 25, false))) ? J$.T(3981257, undefined, 24, false) : J$.R(3981265, 'cause', cause, false, false));
                        } catch (J$e) {
                            J$.Ex(3985033, J$e);
                        } finally {
                            if (J$.Fr(3985041))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(3981473, J$.R(3981321, 'VError', VError, false, true), 'cause', J$.T(3981465, function (err) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(3981441, arguments.callee, this, arguments);
                            arguments = J$.N(3981449, 'arguments', arguments, true, false, false);
                            err = J$.N(3981457, 'err', err, true, false, false);
                            J$.M(3981369, J$.R(3981329, 'mod_assertplus', mod_assertplus, false, true), 'ok', false)(J$.F(3981353, J$.R(3981337, 'mod_isError', mod_isError, false, true), false)(J$.R(3981345, 'err', err, false, false)), J$.T(3981361, 'err must be an Error', 21, false));
                            return J$.Rt(3981433, J$.C(178328, J$.F(3981401, J$.R(3981377, 'mod_isError', mod_isError, false, true), false)(J$.G(3981393, J$.R(3981385, 'err', err, false, false), 'jse_cause'))) ? J$.G(3981417, J$.R(3981409, 'err', err, false, false), 'jse_cause') : J$.T(3981425, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(3985049, J$e);
                        } finally {
                            if (J$.Fr(3985057))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(3981857, J$.R(3981481, 'VError', VError, false, true), 'info', J$.T(3981849, function (err) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(3981801, arguments.callee, this, arguments);
                            arguments = J$.N(3981809, 'arguments', arguments, true, false, false);
                            err = J$.N(3981817, 'err', err, true, false, false);
                            J$.N(3981825, 'rv', rv, false, false, false);
                            J$.N(3981833, 'cause', cause, false, false, false);
                            J$.N(3981841, 'k', k, false, false, false);
                            var rv, cause, k;
                            J$.M(3981529, J$.R(3981489, 'mod_assertplus', mod_assertplus, false, true), 'ok', false)(J$.F(3981513, J$.R(3981497, 'mod_isError', mod_isError, false, true), false)(J$.R(3981505, 'err', err, false, false)), J$.T(3981521, 'err must be an Error', 21, false));
                            cause = J$.W(3981561, 'cause', J$.M(3981553, J$.R(3981537, 'VError', VError, false, true), 'cause', false)(J$.R(3981545, 'err', err, false, false)), cause, false, false);
                            if (J$.C(178336, J$.B(547338, '!==', J$.R(3981569, 'cause', cause, false, false), J$.T(3981577, null, 25, false)))) {
                                rv = J$.W(3981609, 'rv', J$.M(3981601, J$.R(3981585, 'VError', VError, false, true), 'info', false)(J$.R(3981593, 'cause', cause, false, false)), rv, false, false);
                            } else {
                                rv = J$.W(3981625, 'rv', J$.T(3981617, {}, 11, false), rv, false, false);
                            }
                            if (J$.C(178352, J$.C(178344, J$.B(547346, '==', J$.F(3981657, J$.R(3981633, '_typeof', _typeof, false, true), false)(J$.G(3981649, J$.R(3981641, 'err', err, false, false), 'jse_info')), J$.T(3981665, 'object', 21, false))) ? J$.B(547354, '!==', J$.G(3981681, J$.R(3981673, 'err', err, false, false), 'jse_info'), J$.T(3981689, null, 25, false)) : J$._())) {
                                for (k in J$.H(3981769, J$.G(3981705, J$.R(3981697, 'err', err, false, false), 'jse_info'))) {
                                    J$.N(3981777, 'k', k, false, true, false);
                                    {
                                        {
                                            J$.P(3981761, J$.R(3981713, 'rv', rv, false, false), J$.R(3981721, 'k', k, false, false), J$.G(3981753, J$.G(3981737, J$.R(3981729, 'err', err, false, false), 'jse_info'), J$.R(3981745, 'k', k, false, false)));
                                        }
                                    }
                                }
                            }
                            return J$.Rt(3981793, J$.R(3981785, 'rv', rv, false, false));
                        } catch (J$e) {
                            J$.Ex(3985065, J$e);
                        } finally {
                            if (J$.Fr(3985073))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(3982209, J$.R(3981865, 'VError', VError, false, true), 'findCauseByName', J$.T(3982201, function (err, name) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(3982161, arguments.callee, this, arguments);
                            arguments = J$.N(3982169, 'arguments', arguments, true, false, false);
                            err = J$.N(3982177, 'err', err, true, false, false);
                            name = J$.N(3982185, 'name', name, true, false, false);
                            J$.N(3982193, 'cause', cause, false, false, false);
                            var cause;
                            J$.M(3981913, J$.R(3981873, 'mod_assertplus', mod_assertplus, false, true), 'ok', false)(J$.F(3981897, J$.R(3981881, 'mod_isError', mod_isError, false, true), false)(J$.R(3981889, 'err', err, false, false)), J$.T(3981905, 'err must be an Error', 21, false));
                            J$.M(3981945, J$.R(3981921, 'mod_assertplus', mod_assertplus, false, true), 'string', false)(J$.R(3981929, 'name', name, false, false), J$.T(3981937, 'name', 21, false));
                            J$.M(3981993, J$.R(3981953, 'mod_assertplus', mod_assertplus, false, true), 'ok', false)(J$.B(547362, '>', J$.G(3981969, J$.R(3981961, 'name', name, false, false), 'length'), J$.T(3981977, 0, 22, false)), J$.T(3981985, 'name cannot be empty', 21, false));
                            for (cause = J$.W(3982009, 'cause', J$.R(3982001, 'err', err, false, false), cause, false, false); J$.C(178368, J$.B(547370, '!==', J$.R(3982017, 'cause', cause, false, false), J$.T(3982025, null, 25, false))); cause = J$.W(3982057, 'cause', J$.M(3982049, J$.R(3982033, 'VError', VError, false, true), 'cause', false)(J$.R(3982041, 'cause', cause, false, false)), cause, false, false)) {
                                J$.M(3982097, J$.R(3982065, 'mod_assertplus', mod_assertplus, false, true), 'ok', false)(J$.F(3982089, J$.R(3982073, 'mod_isError', mod_isError, false, true), false)(J$.R(3982081, 'cause', cause, false, false)));
                                if (J$.C(178360, J$.B(547378, '==', J$.G(3982113, J$.R(3982105, 'cause', cause, false, false), 'name'), J$.R(3982121, 'name', name, false, false)))) {
                                    return J$.Rt(3982137, J$.R(3982129, 'cause', cause, false, false));
                                }
                            }
                            return J$.Rt(3982153, J$.T(3982145, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(3985081, J$e);
                        } finally {
                            if (J$.Fr(3985089))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(3982313, J$.R(3982217, 'VError', VError, false, true), 'hasCauseWithName', J$.T(3982305, function (err, name) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(3982273, arguments.callee, this, arguments);
                            arguments = J$.N(3982281, 'arguments', arguments, true, false, false);
                            err = J$.N(3982289, 'err', err, true, false, false);
                            name = J$.N(3982297, 'name', name, true, false, false);
                            return J$.Rt(3982265, J$.B(547386, '!==', J$.M(3982249, J$.R(3982225, 'VError', VError, false, true), 'findCauseByName', false)(J$.R(3982233, 'err', err, false, false), J$.R(3982241, 'name', name, false, false)), J$.T(3982257, null, 25, false)));
                        } catch (J$e) {
                            J$.Ex(3985097, J$e);
                        } finally {
                            if (J$.Fr(3985105))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(3982537, J$.R(3982321, 'VError', VError, false, true), 'fullStack', J$.T(3982529, function (err) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(3982497, arguments.callee, this, arguments);
                            arguments = J$.N(3982505, 'arguments', arguments, true, false, false);
                            err = J$.N(3982513, 'err', err, true, false, false);
                            J$.N(3982521, 'cause', cause, false, false, false);
                            J$.M(3982369, J$.R(3982329, 'mod_assertplus', mod_assertplus, false, true), 'ok', false)(J$.F(3982353, J$.R(3982337, 'mod_isError', mod_isError, false, true), false)(J$.R(3982345, 'err', err, false, false)), J$.T(3982361, 'err must be an Error', 21, false));
                            var cause = J$.W(3982401, 'cause', J$.M(3982393, J$.R(3982377, 'VError', VError, false, true), 'cause', false)(J$.R(3982385, 'err', err, false, false)), cause, false, false);
                            if (J$.C(178376, J$.R(3982409, 'cause', cause, false, false))) {
                                return J$.Rt(3982465, J$.B(547402, '+', J$.B(547394, '+', J$.G(3982425, J$.R(3982417, 'err', err, false, false), 'stack'), J$.T(3982433, '\ncaused by: ', 21, false)), J$.M(3982457, J$.R(3982441, 'VError', VError, false, true), 'fullStack', false)(J$.R(3982449, 'cause', cause, false, false))));
                            }
                            return J$.Rt(3982489, J$.G(3982481, J$.R(3982473, 'err', err, false, false), 'stack'));
                        } catch (J$e) {
                            J$.Ex(3985113, J$e);
                        } finally {
                            if (J$.Fr(3985121))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(3982833, J$.R(3982545, 'VError', VError, false, true), 'errorFromList', J$.T(3982825, function (errors) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(3982801, arguments.callee, this, arguments);
                            arguments = J$.N(3982809, 'arguments', arguments, true, false, false);
                            errors = J$.N(3982817, 'errors', errors, true, false, false);
                            J$.M(3982577, J$.R(3982553, 'mod_assertplus', mod_assertplus, false, true), 'arrayOfObject', false)(J$.R(3982561, 'errors', errors, false, false), J$.T(3982569, 'errors', 21, false));
                            if (J$.C(178384, J$.B(547410, '===', J$.G(3982593, J$.R(3982585, 'errors', errors, false, false), 'length'), J$.T(3982601, 0, 22, false)))) {
                                return J$.Rt(3982617, J$.T(3982609, null, 25, false));
                            }
                            J$.M(3982705, J$.R(3982625, 'errors', errors, false, false), 'forEach', false)(J$.T(3982697, function (e) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(3982673, arguments.callee, this, arguments);
                                            arguments = J$.N(3982681, 'arguments', arguments, true, false, false);
                                            e = J$.N(3982689, 'e', e, true, false, false);
                                            J$.M(3982665, J$.R(3982633, 'mod_assertplus', mod_assertplus, false, true), 'ok', false)(J$.F(3982657, J$.R(3982641, 'mod_isError', mod_isError, false, true), false)(J$.R(3982649, 'e', e, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(3985129, J$e);
                                        } finally {
                                            if (J$.Fr(3985137))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            if (J$.C(178392, J$.B(547418, '==', J$.G(3982721, J$.R(3982713, 'errors', errors, false, false), 'length'), J$.T(3982729, 1, 22, false)))) {
                                return J$.Rt(3982761, J$.G(3982753, J$.R(3982737, 'errors', errors, false, false), J$.T(3982745, 0, 22, false)));
                            }
                            return J$.Rt(3982793, J$.F(3982785, J$.R(3982769, 'MultiError', MultiError, false, true), true)(J$.R(3982777, 'errors', errors, false, false)));
                        } catch (J$e) {
                            J$.Ex(3985145, J$e);
                        } finally {
                            if (J$.Fr(3985153))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(3983089, J$.R(3982841, 'VError', VError, false, true), 'errorForEach', J$.T(3983081, function (err, func) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(3983049, arguments.callee, this, arguments);
                            arguments = J$.N(3983057, 'arguments', arguments, true, false, false);
                            err = J$.N(3983065, 'err', err, true, false, false);
                            func = J$.N(3983073, 'func', func, true, false, false);
                            J$.M(3982889, J$.R(3982849, 'mod_assertplus', mod_assertplus, false, true), 'ok', false)(J$.F(3982873, J$.R(3982857, 'mod_isError', mod_isError, false, true), false)(J$.R(3982865, 'err', err, false, false)), J$.T(3982881, 'err must be an Error', 21, false));
                            J$.M(3982921, J$.R(3982897, 'mod_assertplus', mod_assertplus, false, true), 'func', false)(J$.R(3982905, 'func', func, false, false), J$.T(3982913, 'func', 21, false));
                            if (J$.C(178400, J$.B(547426, 'instanceof', J$.R(3982929, 'err', err, false, false), J$.R(3982937, 'MultiError', MultiError, false, true)))) {
                                J$.M(3983017, J$.M(3982953, J$.R(3982945, 'err', err, false, false), 'errors', false)(), 'forEach', false)(J$.T(3983009, function iterError(e) {
                                    jalangiLabel15:
                                        while (true) {
                                            try {
                                                J$.Fe(3982985, arguments.callee, this, arguments);
                                                arguments = J$.N(3982993, 'arguments', arguments, true, false, false);
                                                e = J$.N(3983001, 'e', e, true, false, false);
                                                J$.F(3982977, J$.R(3982961, 'func', func, false, false), false)(J$.R(3982969, 'e', e, false, false));
                                            } catch (J$e) {
                                                J$.Ex(3985161, J$e);
                                            } finally {
                                                if (J$.Fr(3985169))
                                                    continue jalangiLabel15;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            } else {
                                J$.F(3983041, J$.R(3983025, 'func', func, false, false), false)(J$.R(3983033, 'err', err, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(3985177, J$e);
                        } finally {
                            if (J$.Fr(3985185))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.M(3983473, J$.R(3983449, 'mod_util', mod_util, false, true), 'inherits', false)(J$.R(3983457, 'SError', SError, false, true), J$.R(3983465, 'VError', VError, false, true));
            J$.M(3983753, J$.R(3983729, 'mod_util', mod_util, false, true), 'inherits', false)(J$.R(3983737, 'MultiError', MultiError, false, true), J$.R(3983745, 'VError', VError, false, true));
            J$.P(3983785, J$.G(3983769, J$.R(3983761, 'MultiError', MultiError, false, true), 'prototype'), 'name', J$.T(3983777, 'MultiError', 21, false));
            J$.P(3983873, J$.G(3983801, J$.R(3983793, 'MultiError', MultiError, false, true), 'prototype'), 'errors', J$.T(3983865, function me_errors() {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(3983849, arguments.callee, this, arguments);
                            arguments = J$.N(3983857, 'arguments', arguments, true, false, false);
                            return J$.Rt(3983841, J$.M(3983833, J$.G(3983817, J$.R(3983809, 'this', this, false, false), 'ase_errors'), 'slice', false)(J$.T(3983825, 0, 22, false)));
                        } catch (J$e) {
                            J$.Ex(3985225, J$e);
                        } finally {
                            if (J$.Fr(3985233))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.M(3984289, J$.R(3984265, 'mod_util', mod_util, false, true), 'inherits', false)(J$.R(3984273, 'WError', WError, false, true), J$.R(3984281, 'VError', VError, false, true));
            J$.P(3984321, J$.G(3984305, J$.R(3984297, 'WError', WError, false, true), 'prototype'), 'name', J$.T(3984313, 'WError', 21, false));
            J$.P(3984641, J$.G(3984337, J$.R(3984329, 'WError', WError, false, true), 'prototype'), 'toString', J$.T(3984633, function we_toString() {
                jalangiLabel21:
                    while (true) {
                        try {
                            J$.Fe(3984609, arguments.callee, this, arguments);
                            arguments = J$.N(3984617, 'arguments', arguments, true, false, false);
                            J$.N(3984625, 'str', str, false, false, false);
                            var str = J$.W(3984441, 'str', J$.C(178448, J$.C(178440, J$.C(178432, J$.M(3984361, J$.R(3984345, 'this', this, false, false), 'hasOwnProperty', false)(J$.T(3984353, 'name', 21, false))) ? J$.G(3984377, J$.R(3984369, 'this', this, false, false), 'name') : J$._()) ? J$._() : J$.G(3984401, J$.G(3984393, J$.R(3984385, 'this', this, false, false), 'constructor'), 'name')) ? J$._() : J$.G(3984433, J$.G(3984425, J$.G(3984417, J$.R(3984409, 'this', this, false, false), 'constructor'), 'prototype'), 'name'), str, false, false);
                            if (J$.C(178456, J$.G(3984457, J$.R(3984449, 'this', this, false, false), 'message')))
                                str = J$.W(3984497, 'str', J$.B(547490, '+', J$.R(3984489, 'str', str, false, false), J$.B(547482, '+', J$.T(3984465, ': ', 21, false), J$.G(3984481, J$.R(3984473, 'this', this, false, false), 'message'))), str, false, false);
                            if (J$.C(178472, J$.C(178464, J$.G(3984513, J$.R(3984505, 'this', this, false, false), 'jse_cause')) ? J$.G(3984537, J$.G(3984529, J$.R(3984521, 'this', this, false, false), 'jse_cause'), 'message') : J$._()))
                                str = J$.W(3984585, 'str', J$.B(547506, '+', J$.R(3984577, 'str', str, false, false), J$.B(547498, '+', J$.T(3984545, '; caused by ', 21, false), J$.M(3984569, J$.G(3984561, J$.R(3984553, 'this', this, false, false), 'jse_cause'), 'toString', false)())), str, false, false);
                            return J$.Rt(3984601, J$.R(3984593, 'str', str, false, false));
                        } catch (J$e) {
                            J$.Ex(3985257, J$e);
                        } finally {
                            if (J$.Fr(3985265))
                                continue jalangiLabel21;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(3984769, J$.G(3984657, J$.R(3984649, 'WError', WError, false, true), 'prototype'), 'cause', J$.T(3984761, function we_cause(c) {
                jalangiLabel22:
                    while (true) {
                        try {
                            J$.Fe(3984737, arguments.callee, this, arguments);
                            arguments = J$.N(3984745, 'arguments', arguments, true, false, false);
                            c = J$.N(3984753, 'c', c, true, false, false);
                            if (J$.C(178480, J$.F(3984681, J$.R(3984665, 'mod_isError', mod_isError, false, true), false)(J$.R(3984673, 'c', c, false, false))))
                                J$.P(3984705, J$.R(3984689, 'this', this, false, false), 'jse_cause', J$.R(3984697, 'c', c, false, false));
                            return J$.Rt(3984729, J$.G(3984721, J$.R(3984713, 'this', this, false, false), 'jse_cause'));
                        } catch (J$e) {
                            J$.Ex(3985273, J$e);
                        } finally {
                            if (J$.Fr(3985281))
                                continue jalangiLabel22;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(3985289, J$e);
        } finally {
            if (J$.Sr(3985297))
                continue jalangiLabel23;
            else
                break jalangiLabel23;
        }
    }
// JALANGI DO NOT INSTRUMENT
