// oauth-sign/index.js
J$.noInstrEval = false;
jalangiLabel17:
    while (true) {
        try {
            J$.Se(2952817, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestRequest/node_modules/oauth-sign/index.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(2949857, arguments.callee, this, arguments);
                            arguments = J$.N(2949865, 'arguments', arguments, true, false, false);
                            obj = J$.N(2949873, 'obj', obj, true, false, false);
                            if (J$.C(142296, J$.C(142256, J$.B(426362, '===', J$.U(426354, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(2949593, 'Symbol', undefined, true, true) : Symbol = J$.R(2949593, 'Symbol', Symbol, true, true))), J$.T(2949601, 'function', 21, false))) ? J$.B(426378, '===', J$.U(426370, 'typeof', J$.G(2949617, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(2949609, 'Symbol', undefined, true, true) : Symbol = J$.R(2949609, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(2949625, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(2949681, '_typeof', J$.T(2949673, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(2949649, arguments.callee, this, arguments);
                                                arguments = J$.N(2949657, 'arguments', arguments, true, false, false);
                                                obj = J$.N(2949665, 'obj', obj, true, false, false);
                                                return J$.Rt(2949641, J$.U(426386, 'typeof', J$.R(2949633, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(2953025, J$e);
                                            } finally {
                                                if (J$.Fr(2953033))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(2949817, '_typeof', J$.T(2949809, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(2949785, arguments.callee, this, arguments);
                                                arguments = J$.N(2949793, 'arguments', arguments, true, false, false);
                                                obj = J$.N(2949801, 'obj', obj, true, false, false);
                                                return J$.Rt(2949777, J$.C(142288, J$.C(142280, J$.C(142272, J$.C(142264, J$.R(2949689, 'obj', obj, false, false)) ? J$.B(426402, '===', J$.U(426394, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(2949697, 'Symbol', undefined, true, true) : Symbol = J$.R(2949697, 'Symbol', Symbol, true, true))), J$.T(2949705, 'function', 21, false)) : J$._()) ? J$.B(426410, '===', J$.G(2949721, J$.R(2949713, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(2949729, 'Symbol', undefined, true, true) : Symbol = J$.R(2949729, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(426418, '!==', J$.R(2949737, 'obj', obj, false, false), J$.G(2949753, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(2949745, 'Symbol', undefined, true, true) : Symbol = J$.R(2949745, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(2949761, 'symbol', 21, false) : J$.U(426426, 'typeof', J$.R(2949769, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(2953041, J$e);
                                            } finally {
                                                if (J$.Fr(2953049))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(2949849, J$.F(2949841, J$.R(2949825, '_typeof', _typeof, false, true), false)(J$.R(2949833, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(2953057, J$e);
                        } finally {
                            if (J$.Fr(2953065))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function sha(key, body, algorithm) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(2949985, arguments.callee, this, arguments);
                            arguments = J$.N(2949993, 'arguments', arguments, true, false, false);
                            key = J$.N(2950001, 'key', key, true, false, false);
                            body = J$.N(2950009, 'body', body, true, false, false);
                            algorithm = J$.N(2950017, 'algorithm', algorithm, true, false, false);
                            return J$.Rt(2949977, J$.M(2949969, J$.M(2949953, J$.M(2949937, J$.R(2949913, 'crypto', crypto, false, true), 'createHmac', false)(J$.R(2949921, 'algorithm', algorithm, false, false), J$.R(2949929, 'key', key, false, false)), 'update', false)(J$.R(2949945, 'body', body, false, false)), 'digest', false)(J$.T(2949961, 'base64', 21, false)));
                        } catch (J$e) {
                            J$.Ex(2953073, J$e);
                        } finally {
                            if (J$.Fr(2953081))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function rsa(key, body) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(2950097, arguments.callee, this, arguments);
                            arguments = J$.N(2950105, 'arguments', arguments, true, false, false);
                            key = J$.N(2950113, 'key', key, true, false, false);
                            body = J$.N(2950121, 'body', body, true, false, false);
                            return J$.Rt(2950089, J$.M(2950081, J$.M(2950057, J$.M(2950041, J$.R(2950025, 'crypto', crypto, false, true), 'createSign', false)(J$.T(2950033, 'RSA-SHA1', 21, false)), 'update', false)(J$.R(2950049, 'body', body, false, false)), 'sign', false)(J$.R(2950065, 'key', key, false, false), J$.T(2950073, 'base64', 21, false)));
                        } catch (J$e) {
                            J$.Ex(2953089, J$e);
                        } finally {
                            if (J$.Fr(2953097))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function rfc3986(str) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(2950281, arguments.callee, this, arguments);
                            arguments = J$.N(2950289, 'arguments', arguments, true, false, false);
                            str = J$.N(2950297, 'str', str, true, false, false);
                            return J$.Rt(2950273, J$.M(2950265, J$.M(2950241, J$.M(2950217, J$.M(2950193, J$.M(2950169, J$.F(2950145, J$.I(typeof encodeURIComponent === 'undefined' ? encodeURIComponent = J$.R(2950129, 'encodeURIComponent', undefined, true, true) : encodeURIComponent = J$.R(2950129, 'encodeURIComponent', encodeURIComponent, true, true)), false)(J$.R(2950137, 'str', str, false, false)), 'replace', false)(J$.T(2950153, /!/g, 14, false), J$.T(2950161, '%21', 21, false)), 'replace', false)(J$.T(2950177, /\*/g, 14, false), J$.T(2950185, '%2A', 21, false)), 'replace', false)(J$.T(2950201, /\(/g, 14, false), J$.T(2950209, '%28', 21, false)), 'replace', false)(J$.T(2950225, /\)/g, 14, false), J$.T(2950233, '%29', 21, false)), 'replace', false)(J$.T(2950249, /'/g, 14, false), J$.T(2950257, '%27', 21, false)));
                        } catch (J$e) {
                            J$.Ex(2953105, J$e);
                        } finally {
                            if (J$.Fr(2953113))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function map(obj) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(2950705, arguments.callee, this, arguments);
                            arguments = J$.N(2950713, 'arguments', arguments, true, false, false);
                            obj = J$.N(2950721, 'obj', obj, true, false, false);
                            J$.N(2950729, 'key', key, false, false, false);
                            J$.N(2950737, 'val', val, false, false, false);
                            J$.N(2950745, 'arr', arr, false, false, false);
                            J$.N(2950753, 'i', i, false, false, false);
                            J$.N(2950761, 'prop', prop, false, false, false);
                            var key, val, arr = J$.W(2950313, 'arr', J$.T(2950305, [], 10, false), arr, false, false);
                            for (key in J$.H(2950673, J$.R(2950321, 'obj', obj, false, false))) {
                                J$.N(2950681, 'key', key, false, true, false);
                                {
                                    {
                                        val = J$.W(2950353, 'val', J$.G(2950345, J$.R(2950329, 'obj', obj, false, false), J$.R(2950337, 'key', key, false, false)), val, false, false);
                                        if (J$.C(142320, J$.M(2950377, J$.I(typeof Array === 'undefined' ? Array = J$.R(2950361, 'Array', undefined, true, true) : Array = J$.R(2950361, 'Array', Array, true, true)), 'isArray', false)(J$.R(2950369, 'val', val, false, false))))
                                            for (var i = J$.W(2950393, 'i', J$.T(2950385, 0, 22, false), i, false, false); J$.C(142304, J$.B(426434, '<', J$.R(2950401, 'i', i, false, false), J$.G(2950417, J$.R(2950409, 'val', val, false, false), 'length'))); J$.B(426458, '-', i = J$.W(2950433, 'i', J$.B(426450, '+', J$.U(426442, '+', J$.R(2950425, 'i', i, false, false)), 1), i, false, false), 1)) {
                                                J$.M(2950489, J$.R(2950441, 'arr', arr, false, false), 'push', false)(J$.T(2950481, [
                                                    J$.R(2950449, 'key', key, false, false),
                                                    J$.G(2950473, J$.R(2950457, 'val', val, false, false), J$.R(2950465, 'i', i, false, false))
                                                ], 10, false));
                                            }
                                        else if (J$.C(142312, J$.B(426466, '===', J$.F(2950513, J$.R(2950497, '_typeof', _typeof, false, true), false)(J$.R(2950505, 'val', val, false, false)), J$.T(2950521, 'object', 21, false))))
                                            for (var prop in J$.H(2950617, J$.R(2950529, 'val', val, false, false))) {
                                                J$.N(2950625, 'prop', prop, false, true, false);
                                                {
                                                    {
                                                        J$.M(2950609, J$.R(2950537, 'arr', arr, false, false), 'push', false)(J$.T(2950601, [
                                                            J$.B(426490, '+', J$.B(426482, '+', J$.B(426474, '+', J$.R(2950545, 'key', key, false, false), J$.T(2950553, '[', 21, false)), J$.R(2950561, 'prop', prop, false, false)), J$.T(2950569, ']', 21, false)),
                                                            J$.G(2950593, J$.R(2950577, 'val', val, false, false), J$.R(2950585, 'prop', prop, false, false))
                                                        ], 10, false));
                                                    }
                                                }
                                            }
                                        else
                                            J$.M(2950665, J$.R(2950633, 'arr', arr, false, false), 'push', false)(J$.T(2950657, [
                                                J$.R(2950641, 'key', key, false, false),
                                                J$.R(2950649, 'val', val, false, false)
                                            ], 10, false));
                                    }
                                }
                            }
                            return J$.Rt(2950697, J$.R(2950689, 'arr', arr, false, false));
                        } catch (J$e) {
                            J$.Ex(2953121, J$e);
                        } finally {
                            if (J$.Fr(2953129))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function compare(a, b) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(2950833, arguments.callee, this, arguments);
                            arguments = J$.N(2950841, 'arguments', arguments, true, false, false);
                            a = J$.N(2950849, 'a', a, true, false, false);
                            b = J$.N(2950857, 'b', b, true, false, false);
                            return J$.Rt(2950825, J$.C(142336, J$.B(426498, '>', J$.R(2950769, 'a', a, false, false), J$.R(2950777, 'b', b, false, false))) ? J$.T(2950785, 1, 22, false) : J$.C(142328, J$.B(426506, '<', J$.R(2950793, 'a', a, false, false), J$.R(2950801, 'b', b, false, false))) ? J$.U(426514, '-', J$.T(2950809, 1, 22, false)) : J$.T(2950817, 0, 22, false));
                        } catch (J$e) {
                            J$.Ex(2953137, J$e);
                        } finally {
                            if (J$.Fr(2953145))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function generateBase(httpMethod, base_uri, params) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(2951457, arguments.callee, this, arguments);
                            arguments = J$.N(2951465, 'arguments', arguments, true, false, false);
                            httpMethod = J$.N(2951473, 'httpMethod', httpMethod, true, false, false);
                            base_uri = J$.N(2951481, 'base_uri', base_uri, true, false, false);
                            params = J$.N(2951489, 'params', params, true, false, false);
                            J$.N(2951497, 'normalized', normalized, false, false, false);
                            J$.N(2951505, 'base', base, false, false, false);
                            var normalized = J$.W(2951305, 'normalized', J$.M(2951297, J$.M(2951281, J$.M(2951209, J$.M(2951025, J$.F(2950881, J$.R(2950865, 'map', map, false, true), false)(J$.R(2950873, 'params', params, false, false)), 'map', false)(J$.T(2951017, function (p) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(2950993, arguments.callee, this, arguments);
                                            arguments = J$.N(2951001, 'arguments', arguments, true, false, false);
                                            p = J$.N(2951009, 'p', p, true, false, false);
                                            return J$.Rt(2950985, J$.T(2950977, [
                                                J$.F(2950921, J$.R(2950889, 'rfc3986', rfc3986, false, true), false)(J$.G(2950913, J$.R(2950897, 'p', p, false, false), J$.T(2950905, 0, 22, false))),
                                                J$.F(2950969, J$.R(2950929, 'rfc3986', rfc3986, false, true), false)(J$.C(142344, J$.G(2950953, J$.R(2950937, 'p', p, false, false), J$.T(2950945, 1, 22, false))) ? J$._() : J$.T(2950961, '', 21, false))
                                            ], 10, false));
                                        } catch (J$e) {
                                            J$.Ex(2953153, J$e);
                                        } finally {
                                            if (J$.Fr(2953161))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), 'sort', false)(J$.T(2951201, function (a, b) {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(2951169, arguments.callee, this, arguments);
                                            arguments = J$.N(2951177, 'arguments', arguments, true, false, false);
                                            a = J$.N(2951185, 'a', a, true, false, false);
                                            b = J$.N(2951193, 'b', b, true, false, false);
                                            return J$.Rt(2951161, J$.C(142352, J$.F(2951089, J$.R(2951033, 'compare', compare, false, true), false)(J$.G(2951057, J$.R(2951041, 'a', a, false, false), J$.T(2951049, 0, 22, false)), J$.G(2951081, J$.R(2951065, 'b', b, false, false), J$.T(2951073, 0, 22, false)))) ? J$._() : J$.F(2951153, J$.R(2951097, 'compare', compare, false, true), false)(J$.G(2951121, J$.R(2951105, 'a', a, false, false), J$.T(2951113, 1, 22, false)), J$.G(2951145, J$.R(2951129, 'b', b, false, false), J$.T(2951137, 1, 22, false))));
                                        } catch (J$e) {
                                            J$.Ex(2953169, J$e);
                                        } finally {
                                            if (J$.Fr(2953177))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), 'map', false)(J$.T(2951273, function (p) {
                                jalangiLabel10:
                                    while (true) {
                                        try {
                                            J$.Fe(2951249, arguments.callee, this, arguments);
                                            arguments = J$.N(2951257, 'arguments', arguments, true, false, false);
                                            p = J$.N(2951265, 'p', p, true, false, false);
                                            return J$.Rt(2951241, J$.M(2951233, J$.R(2951217, 'p', p, false, false), 'join', false)(J$.T(2951225, '=', 21, false)));
                                        } catch (J$e) {
                                            J$.Ex(2953185, J$e);
                                        } finally {
                                            if (J$.Fr(2953193))
                                                continue jalangiLabel10;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), 'join', false)(J$.T(2951289, '&', 21, false)), normalized, false, false);
                            var base = J$.W(2951433, 'base', J$.M(2951425, J$.T(2951409, [
                                J$.F(2951353, J$.R(2951313, 'rfc3986', rfc3986, false, true), false)(J$.C(142360, J$.R(2951321, 'httpMethod', httpMethod, false, false)) ? J$.M(2951337, J$.R(2951329, 'httpMethod', httpMethod, false, false), 'toUpperCase', false)() : J$.T(2951345, 'GET', 21, false)),
                                J$.F(2951377, J$.R(2951361, 'rfc3986', rfc3986, false, true), false)(J$.R(2951369, 'base_uri', base_uri, false, false)),
                                J$.F(2951401, J$.R(2951385, 'rfc3986', rfc3986, false, true), false)(J$.R(2951393, 'normalized', normalized, false, false))
                            ], 10, false), 'join', false)(J$.T(2951417, '&', 21, false)), base, false, false);
                            return J$.Rt(2951449, J$.R(2951441, 'base', base, false, false));
                        } catch (J$e) {
                            J$.Ex(2953201, J$e);
                        } finally {
                            if (J$.Fr(2953209))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function hmacsign(httpMethod, base_uri, params, consumer_secret, token_secret) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(2951689, arguments.callee, this, arguments);
                            arguments = J$.N(2951697, 'arguments', arguments, true, false, false);
                            httpMethod = J$.N(2951705, 'httpMethod', httpMethod, true, false, false);
                            base_uri = J$.N(2951713, 'base_uri', base_uri, true, false, false);
                            params = J$.N(2951721, 'params', params, true, false, false);
                            consumer_secret = J$.N(2951729, 'consumer_secret', consumer_secret, true, false, false);
                            token_secret = J$.N(2951737, 'token_secret', token_secret, true, false, false);
                            J$.N(2951745, 'base', base, false, false, false);
                            J$.N(2951753, 'key', key, false, false, false);
                            var base = J$.W(2951553, 'base', J$.F(2951545, J$.R(2951513, 'generateBase', generateBase, false, true), false)(J$.R(2951521, 'httpMethod', httpMethod, false, false), J$.R(2951529, 'base_uri', base_uri, false, false), J$.R(2951537, 'params', params, false, false)), base, false, false);
                            var key = J$.W(2951633, 'key', J$.M(2951625, J$.M(2951609, J$.T(2951593, [
                                J$.C(142368, J$.R(2951561, 'consumer_secret', consumer_secret, false, false)) ? J$._() : J$.T(2951569, '', 21, false),
                                J$.C(142376, J$.R(2951577, 'token_secret', token_secret, false, false)) ? J$._() : J$.T(2951585, '', 21, false)
                            ], 10, false), 'map', false)(J$.R(2951601, 'rfc3986', rfc3986, false, true)), 'join', false)(J$.T(2951617, '&', 21, false)), key, false, false);
                            return J$.Rt(2951681, J$.F(2951673, J$.R(2951641, 'sha', sha, false, true), false)(J$.R(2951649, 'key', key, false, false), J$.R(2951657, 'base', base, false, false), J$.T(2951665, 'sha1', 21, false)));
                        } catch (J$e) {
                            J$.Ex(2953217, J$e);
                        } finally {
                            if (J$.Fr(2953225))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function hmacsign256(httpMethod, base_uri, params, consumer_secret, token_secret) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(2951937, arguments.callee, this, arguments);
                            arguments = J$.N(2951945, 'arguments', arguments, true, false, false);
                            httpMethod = J$.N(2951953, 'httpMethod', httpMethod, true, false, false);
                            base_uri = J$.N(2951961, 'base_uri', base_uri, true, false, false);
                            params = J$.N(2951969, 'params', params, true, false, false);
                            consumer_secret = J$.N(2951977, 'consumer_secret', consumer_secret, true, false, false);
                            token_secret = J$.N(2951985, 'token_secret', token_secret, true, false, false);
                            J$.N(2951993, 'base', base, false, false, false);
                            J$.N(2952001, 'key', key, false, false, false);
                            var base = J$.W(2951801, 'base', J$.F(2951793, J$.R(2951761, 'generateBase', generateBase, false, true), false)(J$.R(2951769, 'httpMethod', httpMethod, false, false), J$.R(2951777, 'base_uri', base_uri, false, false), J$.R(2951785, 'params', params, false, false)), base, false, false);
                            var key = J$.W(2951881, 'key', J$.M(2951873, J$.M(2951857, J$.T(2951841, [
                                J$.C(142384, J$.R(2951809, 'consumer_secret', consumer_secret, false, false)) ? J$._() : J$.T(2951817, '', 21, false),
                                J$.C(142392, J$.R(2951825, 'token_secret', token_secret, false, false)) ? J$._() : J$.T(2951833, '', 21, false)
                            ], 10, false), 'map', false)(J$.R(2951849, 'rfc3986', rfc3986, false, true)), 'join', false)(J$.T(2951865, '&', 21, false)), key, false, false);
                            return J$.Rt(2951929, J$.F(2951921, J$.R(2951889, 'sha', sha, false, true), false)(J$.R(2951897, 'key', key, false, false), J$.R(2951905, 'base', base, false, false), J$.T(2951913, 'sha256', 21, false)));
                        } catch (J$e) {
                            J$.Ex(2953233, J$e);
                        } finally {
                            if (J$.Fr(2953241))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function rsasign(httpMethod, base_uri, params, private_key, token_secret) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(2952121, arguments.callee, this, arguments);
                            arguments = J$.N(2952129, 'arguments', arguments, true, false, false);
                            httpMethod = J$.N(2952137, 'httpMethod', httpMethod, true, false, false);
                            base_uri = J$.N(2952145, 'base_uri', base_uri, true, false, false);
                            params = J$.N(2952153, 'params', params, true, false, false);
                            private_key = J$.N(2952161, 'private_key', private_key, true, false, false);
                            token_secret = J$.N(2952169, 'token_secret', token_secret, true, false, false);
                            J$.N(2952177, 'base', base, false, false, false);
                            J$.N(2952185, 'key', key, false, false, false);
                            var base = J$.W(2952049, 'base', J$.F(2952041, J$.R(2952009, 'generateBase', generateBase, false, true), false)(J$.R(2952017, 'httpMethod', httpMethod, false, false), J$.R(2952025, 'base_uri', base_uri, false, false), J$.R(2952033, 'params', params, false, false)), base, false, false);
                            var key = J$.W(2952073, 'key', J$.C(142400, J$.R(2952057, 'private_key', private_key, false, false)) ? J$._() : J$.T(2952065, '', 21, false), key, false, false);
                            return J$.Rt(2952113, J$.F(2952105, J$.R(2952081, 'rsa', rsa, false, true), false)(J$.R(2952089, 'key', key, false, false), J$.R(2952097, 'base', base, false, false)));
                        } catch (J$e) {
                            J$.Ex(2953249, J$e);
                        } finally {
                            if (J$.Fr(2953257))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function plaintext(consumer_secret, token_secret) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(2952289, arguments.callee, this, arguments);
                            arguments = J$.N(2952297, 'arguments', arguments, true, false, false);
                            consumer_secret = J$.N(2952305, 'consumer_secret', consumer_secret, true, false, false);
                            token_secret = J$.N(2952313, 'token_secret', token_secret, true, false, false);
                            J$.N(2952321, 'key', key, false, false, false);
                            var key = J$.W(2952265, 'key', J$.M(2952257, J$.M(2952241, J$.T(2952225, [
                                J$.C(142408, J$.R(2952193, 'consumer_secret', consumer_secret, false, false)) ? J$._() : J$.T(2952201, '', 21, false),
                                J$.C(142416, J$.R(2952209, 'token_secret', token_secret, false, false)) ? J$._() : J$.T(2952217, '', 21, false)
                            ], 10, false), 'map', false)(J$.R(2952233, 'rfc3986', rfc3986, false, true)), 'join', false)(J$.T(2952249, '&', 21, false)), key, false, false);
                            return J$.Rt(2952281, J$.R(2952273, 'key', key, false, false));
                        } catch (J$e) {
                            J$.Ex(2953265, J$e);
                        } finally {
                            if (J$.Fr(2953273))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function sign(signMethod, httpMethod, base_uri, params, consumer_secret, token_secret) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(2952569, arguments.callee, this, arguments);
                            arguments = J$.N(2952577, 'arguments', arguments, true, false, false);
                            signMethod = J$.N(2952585, 'signMethod', signMethod, true, false, false);
                            httpMethod = J$.N(2952593, 'httpMethod', httpMethod, true, false, false);
                            base_uri = J$.N(2952601, 'base_uri', base_uri, true, false, false);
                            params = J$.N(2952609, 'params', params, true, false, false);
                            consumer_secret = J$.N(2952617, 'consumer_secret', consumer_secret, true, false, false);
                            token_secret = J$.N(2952625, 'token_secret', token_secret, true, false, false);
                            J$.N(2952633, 'method', method, false, false, false);
                            J$.N(2952641, 'skipArgs', skipArgs, false, false, false);
                            var method;
                            var skipArgs = J$.W(2952337, 'skipArgs', J$.T(2952329, 1, 22, false), skipArgs, false, false);
                            switch (J$.C1(142424, J$.R(2952345, 'signMethod', signMethod, false, false))) {
                            case J$.C2(142432, J$.T(2952369, 'RSA-SHA1', 21, false)):
                                method = J$.W(2952361, 'method', J$.R(2952353, 'rsasign', rsasign, false, true), method, false, false);
                                break;
                            case J$.C2(142440, J$.T(2952393, 'HMAC-SHA1', 21, false)):
                                method = J$.W(2952385, 'method', J$.R(2952377, 'hmacsign', hmacsign, false, true), method, false, false);
                                break;
                            case J$.C2(142448, J$.T(2952417, 'HMAC-SHA256', 21, false)):
                                method = J$.W(2952409, 'method', J$.R(2952401, 'hmacsign256', hmacsign256, false, true), method, false, false);
                                break;
                            case J$.C2(142456, J$.T(2952457, 'PLAINTEXT', 21, false)):
                                method = J$.W(2952433, 'method', J$.R(2952425, 'plaintext', plaintext, false, true), method, false, false);
                                skipArgs = J$.W(2952449, 'skipArgs', J$.T(2952441, 4, 22, false), skipArgs, false, false);
                                break;
                            default:
                                throw J$.F(2952489, J$.I(typeof Error === 'undefined' ? Error = J$.R(2952465, 'Error', undefined, true, true) : Error = J$.R(2952465, 'Error', Error, true, true)), true)(J$.B(426522, '+', J$.T(2952473, 'Signature method not supported: ', 21, false), J$.R(2952481, 'signMethod', signMethod, false, false)));
                            }
                            return J$.Rt(2952561, J$.M(2952553, J$.R(2952497, 'method', method, false, false), 'apply', false)(J$.T(2952505, null, 25, false), J$.M(2952545, J$.G(2952521, J$.T(2952513, [], 10, false), 'slice'), 'call', false)(J$.I(typeof arguments === 'undefined' ? arguments = J$.R(2952529, 'arguments', undefined, true, true) : arguments = J$.R(2952529, 'arguments', arguments, true, true)), J$.R(2952537, 'skipArgs', skipArgs, false, false))));
                        } catch (J$e) {
                            J$.Ex(2953281, J$e);
                        } finally {
                            if (J$.Fr(2953289))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(2952833, '_typeof', J$.T(2952825, _typeof, 12, false), true, false, false);
            J$.N(2952841, 'crypto', crypto, false, false, false);
            sha = J$.N(2952857, 'sha', J$.T(2952849, sha, 12, false), true, false, false);
            rsa = J$.N(2952873, 'rsa', J$.T(2952865, rsa, 12, false), true, false, false);
            rfc3986 = J$.N(2952889, 'rfc3986', J$.T(2952881, rfc3986, 12, false), true, false, false);
            map = J$.N(2952905, 'map', J$.T(2952897, map, 12, false), true, false, false);
            compare = J$.N(2952921, 'compare', J$.T(2952913, compare, 12, false), true, false, false);
            generateBase = J$.N(2952937, 'generateBase', J$.T(2952929, generateBase, 12, false), true, false, false);
            hmacsign = J$.N(2952953, 'hmacsign', J$.T(2952945, hmacsign, 12, false), true, false, false);
            hmacsign256 = J$.N(2952969, 'hmacsign256', J$.T(2952961, hmacsign256, 12, false), true, false, false);
            rsasign = J$.N(2952985, 'rsasign', J$.T(2952977, rsasign, 12, false), true, false, false);
            plaintext = J$.N(2953001, 'plaintext', J$.T(2952993, plaintext, 12, false), true, false, false);
            sign = J$.N(2953017, 'sign', J$.T(2953009, sign, 12, false), true, false, false);
            J$.T(2949249, 'use strict', 21, false);
            J$.F(2949273, J$.I(typeof require === 'undefined' ? require = J$.R(2949257, 'require', undefined, true, true) : require = J$.R(2949257, 'require', require, true, true)), false)(J$.T(2949265, 'core-js/modules/es.symbol', 21, false));
            J$.F(2949297, J$.I(typeof require === 'undefined' ? require = J$.R(2949281, 'require', undefined, true, true) : require = J$.R(2949281, 'require', require, true, true)), false)(J$.T(2949289, 'core-js/modules/es.symbol.description', 21, false));
            J$.F(2949321, J$.I(typeof require === 'undefined' ? require = J$.R(2949305, 'require', undefined, true, true) : require = J$.R(2949305, 'require', require, true, true)), false)(J$.T(2949313, 'core-js/modules/es.symbol.iterator', 21, false));
            J$.F(2949345, J$.I(typeof require === 'undefined' ? require = J$.R(2949329, 'require', undefined, true, true) : require = J$.R(2949329, 'require', require, true, true)), false)(J$.T(2949337, 'core-js/modules/es.array.is-array', 21, false));
            J$.F(2949369, J$.I(typeof require === 'undefined' ? require = J$.R(2949353, 'require', undefined, true, true) : require = J$.R(2949353, 'require', require, true, true)), false)(J$.T(2949361, 'core-js/modules/es.array.iterator', 21, false));
            J$.F(2949393, J$.I(typeof require === 'undefined' ? require = J$.R(2949377, 'require', undefined, true, true) : require = J$.R(2949377, 'require', require, true, true)), false)(J$.T(2949385, 'core-js/modules/es.array.join', 21, false));
            J$.F(2949417, J$.I(typeof require === 'undefined' ? require = J$.R(2949401, 'require', undefined, true, true) : require = J$.R(2949401, 'require', require, true, true)), false)(J$.T(2949409, 'core-js/modules/es.array.map', 21, false));
            J$.F(2949441, J$.I(typeof require === 'undefined' ? require = J$.R(2949425, 'require', undefined, true, true) : require = J$.R(2949425, 'require', require, true, true)), false)(J$.T(2949433, 'core-js/modules/es.array.slice', 21, false));
            J$.F(2949465, J$.I(typeof require === 'undefined' ? require = J$.R(2949449, 'require', undefined, true, true) : require = J$.R(2949449, 'require', require, true, true)), false)(J$.T(2949457, 'core-js/modules/es.array.sort', 21, false));
            J$.F(2949489, J$.I(typeof require === 'undefined' ? require = J$.R(2949473, 'require', undefined, true, true) : require = J$.R(2949473, 'require', require, true, true)), false)(J$.T(2949481, 'core-js/modules/es.object.to-string', 21, false));
            J$.F(2949513, J$.I(typeof require === 'undefined' ? require = J$.R(2949497, 'require', undefined, true, true) : require = J$.R(2949497, 'require', require, true, true)), false)(J$.T(2949505, 'core-js/modules/es.regexp.exec', 21, false));
            J$.F(2949537, J$.I(typeof require === 'undefined' ? require = J$.R(2949521, 'require', undefined, true, true) : require = J$.R(2949521, 'require', require, true, true)), false)(J$.T(2949529, 'core-js/modules/es.string.iterator', 21, false));
            J$.F(2949561, J$.I(typeof require === 'undefined' ? require = J$.R(2949545, 'require', undefined, true, true) : require = J$.R(2949545, 'require', require, true, true)), false)(J$.T(2949553, 'core-js/modules/es.string.replace', 21, false));
            J$.F(2949585, J$.I(typeof require === 'undefined' ? require = J$.R(2949569, 'require', undefined, true, true) : require = J$.R(2949569, 'require', require, true, true)), false)(J$.T(2949577, 'core-js/modules/web.dom-collections.iterator', 21, false));
            var crypto = J$.W(2949905, 'crypto', J$.F(2949897, J$.I(typeof require === 'undefined' ? require = J$.R(2949881, 'require', undefined, true, true) : require = J$.R(2949881, 'require', require, true, true)), false)(J$.T(2949889, 'crypto', 21, false)), crypto, false, true);
            J$.P(2952665, J$.I(typeof exports === 'undefined' ? exports = J$.R(2952649, 'exports', undefined, true, true) : exports = J$.R(2952649, 'exports', exports, true, true)), 'hmacsign', J$.R(2952657, 'hmacsign', hmacsign, false, true));
            J$.P(2952689, J$.I(typeof exports === 'undefined' ? exports = J$.R(2952673, 'exports', undefined, true, true) : exports = J$.R(2952673, 'exports', exports, true, true)), 'hmacsign256', J$.R(2952681, 'hmacsign256', hmacsign256, false, true));
            J$.P(2952713, J$.I(typeof exports === 'undefined' ? exports = J$.R(2952697, 'exports', undefined, true, true) : exports = J$.R(2952697, 'exports', exports, true, true)), 'rsasign', J$.R(2952705, 'rsasign', rsasign, false, true));
            J$.P(2952737, J$.I(typeof exports === 'undefined' ? exports = J$.R(2952721, 'exports', undefined, true, true) : exports = J$.R(2952721, 'exports', exports, true, true)), 'plaintext', J$.R(2952729, 'plaintext', plaintext, false, true));
            J$.P(2952761, J$.I(typeof exports === 'undefined' ? exports = J$.R(2952745, 'exports', undefined, true, true) : exports = J$.R(2952745, 'exports', exports, true, true)), 'sign', J$.R(2952753, 'sign', sign, false, true));
            J$.P(2952785, J$.I(typeof exports === 'undefined' ? exports = J$.R(2952769, 'exports', undefined, true, true) : exports = J$.R(2952769, 'exports', exports, true, true)), 'rfc3986', J$.R(2952777, 'rfc3986', rfc3986, false, true));
            J$.P(2952809, J$.I(typeof exports === 'undefined' ? exports = J$.R(2952793, 'exports', undefined, true, true) : exports = J$.R(2952793, 'exports', exports, true, true)), 'generateBase', J$.R(2952801, 'generateBase', generateBase, false, true));
        } catch (J$e) {
            J$.Ex(2953297, J$e);
        } finally {
            if (J$.Sr(2953305))
                continue jalangiLabel17;
            else
                break jalangiLabel17;
        }
    }
// JALANGI DO NOT INSTRUMENT
