// console-control-strings/index.js
J$.noInstrEval = false;
jalangiLabel16:
    while (true) {
        try {
            J$.Se(7082673, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/console-control-strings/index.js');
            function colorNameToCode(color) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(7082649, arguments.callee, this, arguments);
                            arguments = J$.N(7082657, 'arguments', arguments, true, false, false);
                            color = J$.N(7082665, 'color', color, true, false, false);
                            if (J$.C(379792, J$.B(718106, '!=', J$.G(7082569, J$.R(7082553, 'colors', colors, false, true), J$.R(7082561, 'color', color, false, false)), J$.T(7082577, null, 25, false))))
                                return J$.Rt(7082609, J$.G(7082601, J$.R(7082585, 'colors', colors, false, true), J$.R(7082593, 'color', color, false, false)));
                            throw J$.F(7082641, J$.I(typeof Error === 'undefined' ? Error = J$.R(7082617, 'Error', undefined, true, true) : Error = J$.R(7082617, 'Error', Error, true, true)), true)(J$.B(718114, '+', J$.T(7082625, 'Unknown color or style name: ', 21, false), J$.R(7082633, 'color', color, false, false)));
                        } catch (J$e) {
                            J$.Ex(7082953, J$e);
                        } finally {
                            if (J$.Fr(7082961))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(7082681, 'prefix', prefix, false, false, false);
            J$.N(7082689, 'colors', colors, false, false, false);
            colorNameToCode = J$.N(7082705, 'colorNameToCode', J$.T(7082697, colorNameToCode, 12, false), true, false, false);
            J$.T(7080841, 'use strict', 21, false);
            var prefix = J$.W(7080857, 'prefix', J$.T(7080849, '\x1B[', 21, false), prefix, false, true);
            J$.P(7080945, J$.I(typeof exports === 'undefined' ? exports = J$.R(7080865, 'exports', undefined, true, true) : exports = J$.R(7080865, 'exports', exports, true, true)), 'up', J$.T(7080937, function up(num) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(7080913, arguments.callee, this, arguments);
                            arguments = J$.N(7080921, 'arguments', arguments, true, false, false);
                            num = J$.N(7080929, 'num', num, true, false, false);
                            return J$.Rt(7080905, J$.B(717898, '+', J$.B(717890, '+', J$.R(7080873, 'prefix', prefix, false, true), J$.C(379720, J$.R(7080881, 'num', num, false, false)) ? J$._() : J$.T(7080889, '', 21, false)), J$.T(7080897, 'A', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082713, J$e);
                        } finally {
                            if (J$.Fr(7082721))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081033, J$.I(typeof exports === 'undefined' ? exports = J$.R(7080953, 'exports', undefined, true, true) : exports = J$.R(7080953, 'exports', exports, true, true)), 'down', J$.T(7081025, function down(num) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7081001, arguments.callee, this, arguments);
                            arguments = J$.N(7081009, 'arguments', arguments, true, false, false);
                            num = J$.N(7081017, 'num', num, true, false, false);
                            return J$.Rt(7080993, J$.B(717914, '+', J$.B(717906, '+', J$.R(7080961, 'prefix', prefix, false, true), J$.C(379728, J$.R(7080969, 'num', num, false, false)) ? J$._() : J$.T(7080977, '', 21, false)), J$.T(7080985, 'B', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082729, J$e);
                        } finally {
                            if (J$.Fr(7082737))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081121, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081041, 'exports', undefined, true, true) : exports = J$.R(7081041, 'exports', exports, true, true)), 'forward', J$.T(7081113, function forward(num) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(7081089, arguments.callee, this, arguments);
                            arguments = J$.N(7081097, 'arguments', arguments, true, false, false);
                            num = J$.N(7081105, 'num', num, true, false, false);
                            return J$.Rt(7081081, J$.B(717930, '+', J$.B(717922, '+', J$.R(7081049, 'prefix', prefix, false, true), J$.C(379736, J$.R(7081057, 'num', num, false, false)) ? J$._() : J$.T(7081065, '', 21, false)), J$.T(7081073, 'C', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082745, J$e);
                        } finally {
                            if (J$.Fr(7082753))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081209, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081129, 'exports', undefined, true, true) : exports = J$.R(7081129, 'exports', exports, true, true)), 'back', J$.T(7081201, function back(num) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(7081177, arguments.callee, this, arguments);
                            arguments = J$.N(7081185, 'arguments', arguments, true, false, false);
                            num = J$.N(7081193, 'num', num, true, false, false);
                            return J$.Rt(7081169, J$.B(717946, '+', J$.B(717938, '+', J$.R(7081137, 'prefix', prefix, false, true), J$.C(379744, J$.R(7081145, 'num', num, false, false)) ? J$._() : J$.T(7081153, '', 21, false)), J$.T(7081161, 'D', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082761, J$e);
                        } finally {
                            if (J$.Fr(7082769))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081297, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081217, 'exports', undefined, true, true) : exports = J$.R(7081217, 'exports', exports, true, true)), 'nextLine', J$.T(7081289, function nextLine(num) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7081265, arguments.callee, this, arguments);
                            arguments = J$.N(7081273, 'arguments', arguments, true, false, false);
                            num = J$.N(7081281, 'num', num, true, false, false);
                            return J$.Rt(7081257, J$.B(717962, '+', J$.B(717954, '+', J$.R(7081225, 'prefix', prefix, false, true), J$.C(379752, J$.R(7081233, 'num', num, false, false)) ? J$._() : J$.T(7081241, '', 21, false)), J$.T(7081249, 'E', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082777, J$e);
                        } finally {
                            if (J$.Fr(7082785))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081385, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081305, 'exports', undefined, true, true) : exports = J$.R(7081305, 'exports', exports, true, true)), 'previousLine', J$.T(7081377, function previousLine(num) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(7081353, arguments.callee, this, arguments);
                            arguments = J$.N(7081361, 'arguments', arguments, true, false, false);
                            num = J$.N(7081369, 'num', num, true, false, false);
                            return J$.Rt(7081345, J$.B(717978, '+', J$.B(717970, '+', J$.R(7081313, 'prefix', prefix, false, true), J$.C(379760, J$.R(7081321, 'num', num, false, false)) ? J$._() : J$.T(7081329, '', 21, false)), J$.T(7081337, 'F', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082793, J$e);
                        } finally {
                            if (J$.Fr(7082801))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081505, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081393, 'exports', undefined, true, true) : exports = J$.R(7081393, 'exports', exports, true, true)), 'horizontalAbsolute', J$.T(7081497, function horizontalAbsolute(num) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(7081473, arguments.callee, this, arguments);
                            arguments = J$.N(7081481, 'arguments', arguments, true, false, false);
                            num = J$.N(7081489, 'num', num, true, false, false);
                            if (J$.C(379768, J$.B(717986, '==', J$.R(7081401, 'num', num, false, false), J$.T(7081409, null, 25, false))))
                                throw J$.F(7081433, J$.I(typeof Error === 'undefined' ? Error = J$.R(7081417, 'Error', undefined, true, true) : Error = J$.R(7081417, 'Error', Error, true, true)), true)(J$.T(7081425, 'horizontalAboslute requires a column to position to', 21, false));
                            return J$.Rt(7081465, J$.B(718002, '+', J$.B(717994, '+', J$.R(7081441, 'prefix', prefix, false, true), J$.R(7081449, 'num', num, false, false)), J$.T(7081457, 'G', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082809, J$e);
                        } finally {
                            if (J$.Fr(7082817))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081569, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081513, 'exports', undefined, true, true) : exports = J$.R(7081513, 'exports', exports, true, true)), 'eraseData', J$.T(7081561, function eraseData() {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(7081545, arguments.callee, this, arguments);
                            arguments = J$.N(7081553, 'arguments', arguments, true, false, false);
                            return J$.Rt(7081537, J$.B(718010, '+', J$.R(7081521, 'prefix', prefix, false, true), J$.T(7081529, 'J', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082825, J$e);
                        } finally {
                            if (J$.Fr(7082833))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081633, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081577, 'exports', undefined, true, true) : exports = J$.R(7081577, 'exports', exports, true, true)), 'eraseLine', J$.T(7081625, function eraseLine() {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(7081609, arguments.callee, this, arguments);
                            arguments = J$.N(7081617, 'arguments', arguments, true, false, false);
                            return J$.Rt(7081601, J$.B(718018, '+', J$.R(7081585, 'prefix', prefix, false, true), J$.T(7081593, 'K', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082841, J$e);
                        } finally {
                            if (J$.Fr(7082849))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081737, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081641, 'exports', undefined, true, true) : exports = J$.R(7081641, 'exports', exports, true, true)), 'goto', J$.T(7081729, function (x, y) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(7081697, arguments.callee, this, arguments);
                            arguments = J$.N(7081705, 'arguments', arguments, true, false, false);
                            x = J$.N(7081713, 'x', x, true, false, false);
                            y = J$.N(7081721, 'y', y, true, false, false);
                            return J$.Rt(7081689, J$.B(718050, '+', J$.B(718042, '+', J$.B(718034, '+', J$.B(718026, '+', J$.R(7081649, 'prefix', prefix, false, true), J$.R(7081657, 'y', y, false, false)), J$.T(7081665, ';', 21, false)), J$.R(7081673, 'x', x, false, false)), J$.T(7081681, 'H', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082857, J$e);
                        } finally {
                            if (J$.Fr(7082865))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081793, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081745, 'exports', undefined, true, true) : exports = J$.R(7081745, 'exports', exports, true, true)), 'gotoSOL', J$.T(7081785, function () {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(7081769, arguments.callee, this, arguments);
                            arguments = J$.N(7081777, 'arguments', arguments, true, false, false);
                            return J$.Rt(7081761, J$.T(7081753, '\r', 21, false));
                        } catch (J$e) {
                            J$.Ex(7082873, J$e);
                        } finally {
                            if (J$.Fr(7082881))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081849, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081801, 'exports', undefined, true, true) : exports = J$.R(7081801, 'exports', exports, true, true)), 'beep', J$.T(7081841, function () {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(7081825, arguments.callee, this, arguments);
                            arguments = J$.N(7081833, 'arguments', arguments, true, false, false);
                            return J$.Rt(7081817, J$.T(7081809, '\x07', 21, false));
                        } catch (J$e) {
                            J$.Ex(7082889, J$e);
                        } finally {
                            if (J$.Fr(7082897))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081913, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081857, 'exports', undefined, true, true) : exports = J$.R(7081857, 'exports', exports, true, true)), 'hideCursor', J$.T(7081905, function hideCursor() {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(7081889, arguments.callee, this, arguments);
                            arguments = J$.N(7081897, 'arguments', arguments, true, false, false);
                            return J$.Rt(7081881, J$.B(718058, '+', J$.R(7081865, 'prefix', prefix, false, true), J$.T(7081873, '?25l', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082905, J$e);
                        } finally {
                            if (J$.Fr(7082913))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7081977, J$.I(typeof exports === 'undefined' ? exports = J$.R(7081921, 'exports', undefined, true, true) : exports = J$.R(7081921, 'exports', exports, true, true)), 'showCursor', J$.T(7081969, function showCursor() {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(7081953, arguments.callee, this, arguments);
                            arguments = J$.N(7081961, 'arguments', arguments, true, false, false);
                            return J$.Rt(7081945, J$.B(718066, '+', J$.R(7081929, 'prefix', prefix, false, true), J$.T(7081937, '?25h', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082921, J$e);
                        } finally {
                            if (J$.Fr(7082929))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var colors = J$.W(7082337, 'colors', J$.T(7082329, {
                reset: J$.T(7081985, 0, 22, false),
                bold: J$.T(7081993, 1, 22, false),
                italic: J$.T(7082001, 3, 22, false),
                underline: J$.T(7082009, 4, 22, false),
                inverse: J$.T(7082017, 7, 22, false),
                stopBold: J$.T(7082025, 22, 22, false),
                stopItalic: J$.T(7082033, 23, 22, false),
                stopUnderline: J$.T(7082041, 24, 22, false),
                stopInverse: J$.T(7082049, 27, 22, false),
                white: J$.T(7082057, 37, 22, false),
                black: J$.T(7082065, 30, 22, false),
                blue: J$.T(7082073, 34, 22, false),
                cyan: J$.T(7082081, 36, 22, false),
                green: J$.T(7082089, 32, 22, false),
                magenta: J$.T(7082097, 35, 22, false),
                red: J$.T(7082105, 31, 22, false),
                yellow: J$.T(7082113, 33, 22, false),
                bgWhite: J$.T(7082121, 47, 22, false),
                bgBlack: J$.T(7082129, 40, 22, false),
                bgBlue: J$.T(7082137, 44, 22, false),
                bgCyan: J$.T(7082145, 46, 22, false),
                bgGreen: J$.T(7082153, 42, 22, false),
                bgMagenta: J$.T(7082161, 45, 22, false),
                bgRed: J$.T(7082169, 41, 22, false),
                bgYellow: J$.T(7082177, 43, 22, false),
                grey: J$.T(7082185, 90, 22, false),
                brightBlack: J$.T(7082193, 90, 22, false),
                brightRed: J$.T(7082201, 91, 22, false),
                brightGreen: J$.T(7082209, 92, 22, false),
                brightYellow: J$.T(7082217, 93, 22, false),
                brightBlue: J$.T(7082225, 94, 22, false),
                brightMagenta: J$.T(7082233, 95, 22, false),
                brightCyan: J$.T(7082241, 96, 22, false),
                brightWhite: J$.T(7082249, 97, 22, false),
                bgGrey: J$.T(7082257, 100, 22, false),
                bgBrightBlack: J$.T(7082265, 100, 22, false),
                bgBrightRed: J$.T(7082273, 101, 22, false),
                bgBrightGreen: J$.T(7082281, 102, 22, false),
                bgBrightYellow: J$.T(7082289, 103, 22, false),
                bgBrightBlue: J$.T(7082297, 104, 22, false),
                bgBrightMagenta: J$.T(7082305, 105, 22, false),
                bgBrightCyan: J$.T(7082313, 106, 22, false),
                bgBrightWhite: J$.T(7082321, 107, 22, false)
            }, 11, false), colors, false, true);
            J$.P(7082545, J$.I(typeof exports === 'undefined' ? exports = J$.R(7082345, 'exports', undefined, true, true) : exports = J$.R(7082345, 'exports', exports, true, true)), 'color', J$.T(7082537, function color(colorWith) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(7082513, arguments.callee, this, arguments);
                            arguments = J$.N(7082521, 'arguments', arguments, true, false, false);
                            colorWith = J$.N(7082529, 'colorWith', colorWith, true, false, false);
                            if (J$.C(379784, J$.C(379776, J$.B(718074, '!==', J$.G(7082361, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7082353, 'arguments', undefined, true, true) : arguments = J$.R(7082353, 'arguments', arguments, true, true)), 'length'), J$.T(7082369, 1, 22, false))) ? J$._() : J$.U(718082, '!', J$.M(7082393, J$.I(typeof Array === 'undefined' ? Array = J$.R(7082377, 'Array', undefined, true, true) : Array = J$.R(7082377, 'Array', Array, true, true)), 'isArray', false)(J$.R(7082385, 'colorWith', colorWith, false, false))))) {
                                colorWith = J$.W(7082441, 'colorWith', J$.M(7082433, J$.G(7082417, J$.G(7082409, J$.I(typeof Array === 'undefined' ? Array = J$.R(7082401, 'Array', undefined, true, true) : Array = J$.R(7082401, 'Array', Array, true, true)), 'prototype'), 'slice'), 'call', false)(J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7082425, 'arguments', undefined, true, true) : arguments = J$.R(7082425, 'arguments', arguments, true, true))), colorWith, false, false);
                            }
                            return J$.Rt(7082505, J$.B(718098, '+', J$.B(718090, '+', J$.R(7082449, 'prefix', prefix, false, true), J$.M(7082489, J$.M(7082473, J$.R(7082457, 'colorWith', colorWith, false, false), 'map', false)(J$.R(7082465, 'colorNameToCode', colorNameToCode, false, true)), 'join', false)(J$.T(7082481, ';', 21, false))), J$.T(7082497, 'm', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7082937, J$e);
                        } finally {
                            if (J$.Fr(7082945))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(7082969, J$e);
        } finally {
            if (J$.Sr(7082977))
                continue jalangiLabel16;
            else
                break jalangiLabel16;
        }
    }
// JALANGI DO NOT INSTRUMENT
