// logform/dist/format.js
J$.noInstrEval = false;
jalangiLabel24:
    while (true) {
        try {
            J$.Se(4469337, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/logform/dist/format.js');
            function _typeof(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(4466281, arguments.callee, this, arguments);
                            arguments = J$.N(4466289, 'arguments', arguments, true, false, false);
                            obj = J$.N(4466297, 'obj', obj, true, false, false);
                            if (J$.C(248328, J$.C(248288, J$.B(551594, '===', J$.U(551586, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(4466017, 'Symbol', undefined, true, true) : Symbol = J$.R(4466017, 'Symbol', Symbol, true, true))), J$.T(4466025, 'function', 21, false))) ? J$.B(551610, '===', J$.U(551602, 'typeof', J$.G(4466041, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(4466033, 'Symbol', undefined, true, true) : Symbol = J$.R(4466033, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(4466049, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(4466105, '_typeof', J$.T(4466097, function _typeof(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(4466073, arguments.callee, this, arguments);
                                                arguments = J$.N(4466081, 'arguments', arguments, true, false, false);
                                                obj = J$.N(4466089, 'obj', obj, true, false, false);
                                                return J$.Rt(4466065, J$.U(551618, 'typeof', J$.R(4466057, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(4469529, J$e);
                                            } finally {
                                                if (J$.Fr(4469537))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(4466241, '_typeof', J$.T(4466233, function _typeof(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(4466209, arguments.callee, this, arguments);
                                                arguments = J$.N(4466217, 'arguments', arguments, true, false, false);
                                                obj = J$.N(4466225, 'obj', obj, true, false, false);
                                                return J$.Rt(4466201, J$.C(248320, J$.C(248312, J$.C(248304, J$.C(248296, J$.R(4466113, 'obj', obj, false, false)) ? J$.B(551634, '===', J$.U(551626, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(4466121, 'Symbol', undefined, true, true) : Symbol = J$.R(4466121, 'Symbol', Symbol, true, true))), J$.T(4466129, 'function', 21, false)) : J$._()) ? J$.B(551642, '===', J$.G(4466145, J$.R(4466137, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(4466153, 'Symbol', undefined, true, true) : Symbol = J$.R(4466153, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(551650, '!==', J$.R(4466161, 'obj', obj, false, false), J$.G(4466177, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(4466169, 'Symbol', undefined, true, true) : Symbol = J$.R(4466169, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(4466185, 'symbol', 21, false) : J$.U(551658, 'typeof', J$.R(4466193, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(4469545, J$e);
                                            } finally {
                                                if (J$.Fr(4469553))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(4466273, J$.F(4466265, J$.R(4466249, '_typeof', _typeof, false, true), false)(J$.R(4466257, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(4469561, J$e);
                        } finally {
                            if (J$.Fr(4469569))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(4466345, arguments.callee, this, arguments);
                            arguments = J$.N(4466353, 'arguments', arguments, true, false, false);
                            instance = J$.N(4466361, 'instance', instance, true, false, false);
                            Constructor = J$.N(4466369, 'Constructor', Constructor, true, false, false);
                            if (J$.C(248336, J$.U(551674, '!', J$.B(551666, 'instanceof', J$.R(4466305, 'instance', instance, false, false), J$.R(4466313, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(4466337, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(4466321, 'TypeError', undefined, true, true) : TypeError = J$.R(4466321, 'TypeError', TypeError, true, true)), true)(J$.T(4466329, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(4469577, J$e);
                        } finally {
                            if (J$.Fr(4469585))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(4466481, arguments.callee, this, arguments);
                            arguments = J$.N(4466489, 'arguments', arguments, true, false, false);
                            self = J$.N(4466497, 'self', self, true, false, false);
                            call = J$.N(4466505, 'call', call, true, false, false);
                            if (J$.C(248360, J$.C(248352, J$.R(4466377, 'call', call, false, false)) ? J$.C(248344, J$.B(551682, '===', J$.F(4466401, J$.R(4466385, '_typeof', _typeof, false, true), false)(J$.R(4466393, 'call', call, false, false)), J$.T(4466409, 'object', 21, false))) ? J$._() : J$.B(551698, '===', J$.U(551690, 'typeof', J$.R(4466417, 'call', call, false, false)), J$.T(4466425, 'function', 21, false)) : J$._())) {
                                return J$.Rt(4466441, J$.R(4466433, 'call', call, false, false));
                            }
                            return J$.Rt(4466473, J$.F(4466465, J$.R(4466449, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(4466457, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(4469593, J$e);
                        } finally {
                            if (J$.Fr(4469601))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(4466705, arguments.callee, this, arguments);
                            arguments = J$.N(4466713, 'arguments', arguments, true, false, false);
                            subClass = J$.N(4466721, 'subClass', subClass, true, false, false);
                            superClass = J$.N(4466729, 'superClass', superClass, true, false, false);
                            if (J$.C(248376, J$.C(248368, J$.B(551714, '!==', J$.U(551706, 'typeof', J$.R(4466513, 'superClass', superClass, false, false)), J$.T(4466521, 'function', 21, false))) ? J$.B(551722, '!==', J$.R(4466529, 'superClass', superClass, false, false), J$.T(4466537, null, 25, false)) : J$._())) {
                                throw J$.F(4466561, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(4466545, 'TypeError', undefined, true, true) : TypeError = J$.R(4466545, 'TypeError', TypeError, true, true)), true)(J$.T(4466553, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(4466657, J$.R(4466569, 'subClass', subClass, false, false), 'prototype', J$.M(4466649, J$.I(typeof Object === 'undefined' ? Object = J$.R(4466577, 'Object', undefined, true, true) : Object = J$.R(4466577, 'Object', Object, true, true)), 'create', false)(J$.C(248384, J$.R(4466585, 'superClass', superClass, false, false)) ? J$.G(4466601, J$.R(4466593, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(4466641, {
                                constructor: J$.T(4466633, {
                                    value: J$.R(4466609, 'subClass', subClass, false, false),
                                    writable: J$.T(4466617, true, 23, false),
                                    configurable: J$.T(4466625, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(248392, J$.R(4466665, 'superClass', superClass, false, false)))
                                J$.F(4466697, J$.R(4466673, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(4466681, 'subClass', subClass, false, false), J$.R(4466689, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(4469609, J$e);
                        } finally {
                            if (J$.Fr(4469617))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(4466793, arguments.callee, this, arguments);
                            arguments = J$.N(4466801, 'arguments', arguments, true, false, false);
                            self = J$.N(4466809, 'self', self, true, false, false);
                            if (J$.C(248400, J$.B(551730, '===', J$.R(4466737, 'self', self, false, false), void J$.T(4466745, 0, 22, false)))) {
                                throw J$.F(4466769, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(4466753, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(4466753, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(4466761, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(4466785, J$.R(4466777, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(4469625, J$e);
                        } finally {
                            if (J$.Fr(4469633))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _wrapNativeSuper(Class) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(4467377, arguments.callee, this, arguments);
                            arguments = J$.N(4467385, 'arguments', arguments, true, false, false);
                            Class = J$.N(4467393, 'Class', Class, true, false, false);
                            J$.N(4467401, '_cache', _cache, false, false, false);
                            var _cache = J$.W(4466857, '_cache', J$.C(248408, J$.B(551746, '===', J$.U(551738, 'typeof', J$.I(typeof Map === 'undefined' ? Map = J$.R(4466817, 'Map', undefined, true, true) : Map = J$.R(4466817, 'Map', Map, true, true))), J$.T(4466825, 'function', 21, false))) ? J$.F(4466841, J$.I(typeof Map === 'undefined' ? Map = J$.R(4466833, 'Map', undefined, true, true) : Map = J$.R(4466833, 'Map', Map, true, true)), true)() : J$.T(4466849, undefined, 24, false), _cache, false, false);
                            _wrapNativeSuper = J$.W(4467337, '_wrapNativeSuper', J$.T(4467329, function _wrapNativeSuper(Class) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(4467289, arguments.callee, this, arguments);
                                            function Wrapper() {
                                                jalangiLabel7:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(4467137, arguments.callee, this, arguments);
                                                            arguments = J$.N(4467145, 'arguments', arguments, true, false, false);
                                                            return J$.Rt(4467129, J$.F(4467121, J$.R(4467065, '_construct', _construct, false, true), false)(J$.R(4467073, 'Class', Class, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4467081, 'arguments', undefined, true, true) : arguments = J$.R(4467081, 'arguments', arguments, true, true)), J$.G(4467113, J$.F(4467105, J$.R(4467089, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(4467097, 'this', this, false, false)), 'constructor')));
                                                        } catch (J$e) {
                                                            J$.Ex(4469641, J$e);
                                                        } finally {
                                                            if (J$.Fr(4469649))
                                                                continue jalangiLabel7;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }
                                            arguments = J$.N(4467297, 'arguments', arguments, true, false, false);
                                            Class = J$.N(4467305, 'Class', Class, true, false, false);
                                            Wrapper = J$.N(4467321, 'Wrapper', J$.T(4467313, Wrapper, 12, false), true, false, false);
                                            if (J$.C(248424, J$.C(248416, J$.B(551754, '===', J$.R(4466865, 'Class', Class, false, false), J$.T(4466873, null, 25, false))) ? J$._() : J$.U(551762, '!', J$.F(4466897, J$.R(4466881, '_isNativeFunction', _isNativeFunction, false, true), false)(J$.R(4466889, 'Class', Class, false, false)))))
                                                return J$.Rt(4466913, J$.R(4466905, 'Class', Class, false, false));
                                            if (J$.C(248432, J$.B(551778, '!==', J$.U(551770, 'typeof', J$.R(4466921, 'Class', Class, false, false)), J$.T(4466929, 'function', 21, false)))) {
                                                throw J$.F(4466953, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(4466937, 'TypeError', undefined, true, true) : TypeError = J$.R(4466937, 'TypeError', TypeError, true, true)), true)(J$.T(4466945, 'Super expression must either be null or a function', 21, false));
                                            }
                                            if (J$.C(248448, J$.B(551794, '!==', J$.U(551786, 'typeof', J$.R(4466961, '_cache', _cache, false, false)), J$.T(4466969, 'undefined', 21, false)))) {
                                                if (J$.C(248440, J$.M(4466993, J$.R(4466977, '_cache', _cache, false, false), 'has', false)(J$.R(4466985, 'Class', Class, false, false))))
                                                    return J$.Rt(4467025, J$.M(4467017, J$.R(4467001, '_cache', _cache, false, false), 'get', false)(J$.R(4467009, 'Class', Class, false, false)));
                                                J$.M(4467057, J$.R(4467033, '_cache', _cache, false, false), 'set', false)(J$.R(4467041, 'Class', Class, false, false), J$.R(4467049, 'Wrapper', Wrapper, false, false));
                                            }
                                            J$.P(4467241, J$.R(4467153, 'Wrapper', Wrapper, false, false), 'prototype', J$.M(4467233, J$.I(typeof Object === 'undefined' ? Object = J$.R(4467161, 'Object', undefined, true, true) : Object = J$.R(4467161, 'Object', Object, true, true)), 'create', false)(J$.G(4467177, J$.R(4467169, 'Class', Class, false, false), 'prototype'), J$.T(4467225, {
                                                constructor: J$.T(4467217, {
                                                    value: J$.R(4467185, 'Wrapper', Wrapper, false, false),
                                                    enumerable: J$.T(4467193, false, 23, false),
                                                    writable: J$.T(4467201, true, 23, false),
                                                    configurable: J$.T(4467209, true, 23, false)
                                                }, 11, false)
                                            }, 11, false)));
                                            return J$.Rt(4467281, J$.F(4467273, J$.R(4467249, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(4467257, 'Wrapper', Wrapper, false, false), J$.R(4467265, 'Class', Class, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(4469657, J$e);
                                        } finally {
                                            if (J$.Fr(4469665))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _wrapNativeSuper, false, true);
                            return J$.Rt(4467369, J$.F(4467361, J$.R(4467345, '_wrapNativeSuper', _wrapNativeSuper, false, true), false)(J$.R(4467353, 'Class', Class, false, false)));
                        } catch (J$e) {
                            J$.Ex(4469673, J$e);
                        } finally {
                            if (J$.Fr(4469681))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isNativeReflectConstruct() {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(4467657, arguments.callee, this, arguments);
                            arguments = J$.N(4467665, 'arguments', arguments, true, false, false);
                            if (J$.C(248464, J$.C(248456, J$.B(551810, '===', J$.U(551802, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(4467409, 'Reflect', undefined, true, true) : Reflect = J$.R(4467409, 'Reflect', Reflect, true, true))), J$.T(4467417, 'undefined', 21, false))) ? J$._() : J$.U(551818, '!', J$.G(4467433, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(4467425, 'Reflect', undefined, true, true) : Reflect = J$.R(4467425, 'Reflect', Reflect, true, true)), 'construct'))))
                                return J$.Rt(4467449, J$.T(4467441, false, 23, false));
                            if (J$.C(248472, J$.G(4467473, J$.G(4467465, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(4467457, 'Reflect', undefined, true, true) : Reflect = J$.R(4467457, 'Reflect', Reflect, true, true)), 'construct'), 'sham')))
                                return J$.Rt(4467489, J$.T(4467481, false, 23, false));
                            if (J$.C(248480, J$.B(551834, '===', J$.U(551826, 'typeof', J$.I(typeof Proxy === 'undefined' ? Proxy = J$.R(4467497, 'Proxy', undefined, true, true) : Proxy = J$.R(4467497, 'Proxy', Proxy, true, true))), J$.T(4467505, 'function', 21, false))))
                                return J$.Rt(4467521, J$.T(4467513, true, 23, false));
                            try {
                                J$.M(4467609, J$.G(4467545, J$.G(4467537, J$.I(typeof Date === 'undefined' ? Date = J$.R(4467529, 'Date', undefined, true, true) : Date = J$.R(4467529, 'Date', Date, true, true)), 'prototype'), 'toString'), 'call', false)(J$.M(4467601, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(4467553, 'Reflect', undefined, true, true) : Reflect = J$.R(4467553, 'Reflect', Reflect, true, true)), 'construct', false)(J$.I(typeof Date === 'undefined' ? Date = J$.R(4467561, 'Date', undefined, true, true) : Date = J$.R(4467561, 'Date', Date, true, true)), J$.T(4467569, [], 10, false), J$.T(4467593, function () {
                                    jalangiLabel10:
                                        while (true) {
                                            try {
                                                J$.Fe(4467577, arguments.callee, this, arguments);
                                                arguments = J$.N(4467585, 'arguments', arguments, true, false, false);
                                            } catch (J$e) {
                                                J$.Ex(4469689, J$e);
                                            } finally {
                                                if (J$.Fr(4469697))
                                                    continue jalangiLabel10;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)));
                                return J$.Rt(4467625, J$.T(4467617, true, 23, false));
                            } catch (e) {
                                J$.N(4467649, 'e', e, false, false, true);
                                return J$.Rt(4467641, J$.T(4467633, false, 23, false));
                            }
                        } catch (J$e) {
                            J$.Ex(4469705, J$e);
                        } finally {
                            if (J$.Fr(4469713))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _construct(Parent, args, Class) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(4468033, arguments.callee, this, arguments);
                            arguments = J$.N(4468041, 'arguments', arguments, true, false, false);
                            Parent = J$.N(4468049, 'Parent', Parent, true, false, false);
                            args = J$.N(4468057, 'args', args, true, false, false);
                            Class = J$.N(4468065, 'Class', Class, true, false, false);
                            if (J$.C(248496, J$.F(4467681, J$.R(4467673, 'isNativeReflectConstruct', isNativeReflectConstruct, false, true), false)())) {
                                _construct = J$.W(4467705, '_construct', J$.G(4467697, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(4467689, 'Reflect', undefined, true, true) : Reflect = J$.R(4467689, 'Reflect', Reflect, true, true)), 'construct'), _construct, false, true);
                            } else {
                                _construct = J$.W(4467985, '_construct', J$.T(4467977, function _construct(Parent, args, Class) {
                                    jalangiLabel12:
                                        while (true) {
                                            try {
                                                J$.Fe(4467913, arguments.callee, this, arguments);
                                                arguments = J$.N(4467921, 'arguments', arguments, true, false, false);
                                                Parent = J$.N(4467929, 'Parent', Parent, true, false, false);
                                                args = J$.N(4467937, 'args', args, true, false, false);
                                                Class = J$.N(4467945, 'Class', Class, true, false, false);
                                                J$.N(4467953, 'a', a, false, false, false);
                                                J$.N(4467961, 'Constructor', Constructor, false, false, false);
                                                J$.N(4467969, 'instance', instance, false, false, false);
                                                var a = J$.W(4467729, 'a', J$.T(4467721, [J$.T(4467713, null, 25, false)], 10, false), a, false, false);
                                                J$.M(4467769, J$.G(4467745, J$.R(4467737, 'a', a, false, false), 'push'), 'apply', false)(J$.R(4467753, 'a', a, false, false), J$.R(4467761, 'args', args, false, false));
                                                var Constructor = J$.W(4467817, 'Constructor', J$.M(4467809, J$.G(4467785, J$.I(typeof Function === 'undefined' ? Function = J$.R(4467777, 'Function', undefined, true, true) : Function = J$.R(4467777, 'Function', Function, true, true)), 'bind'), 'apply', false)(J$.R(4467793, 'Parent', Parent, false, false), J$.R(4467801, 'a', a, false, false)), Constructor, false, false);
                                                var instance = J$.W(4467841, 'instance', J$.F(4467833, J$.R(4467825, 'Constructor', Constructor, false, false), true)(), instance, false, false);
                                                if (J$.C(248488, J$.R(4467849, 'Class', Class, false, false)))
                                                    J$.F(4467889, J$.R(4467857, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(4467865, 'instance', instance, false, false), J$.G(4467881, J$.R(4467873, 'Class', Class, false, false), 'prototype'));
                                                return J$.Rt(4467905, J$.R(4467897, 'instance', instance, false, false));
                                            } catch (J$e) {
                                                J$.Ex(4469721, J$e);
                                            } finally {
                                                if (J$.Fr(4469729))
                                                    continue jalangiLabel12;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _construct, false, true);
                            }
                            return J$.Rt(4468025, J$.M(4468017, J$.R(4467993, '_construct', _construct, false, true), 'apply', false)(J$.T(4468001, null, 25, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4468009, 'arguments', undefined, true, true) : arguments = J$.R(4468009, 'arguments', arguments, true, true))));
                        } catch (J$e) {
                            J$.Ex(4469737, J$e);
                        } finally {
                            if (J$.Fr(4469745))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _isNativeFunction(fn) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(4468137, arguments.callee, this, arguments);
                            arguments = J$.N(4468145, 'arguments', arguments, true, false, false);
                            fn = J$.N(4468153, 'fn', fn, true, false, false);
                            return J$.Rt(4468129, J$.B(551850, '!==', J$.M(4468113, J$.M(4468097, J$.G(4468081, J$.I(typeof Function === 'undefined' ? Function = J$.R(4468073, 'Function', undefined, true, true) : Function = J$.R(4468073, 'Function', Function, true, true)), 'toString'), 'call', false)(J$.R(4468089, 'fn', fn, false, false)), 'indexOf', false)(J$.T(4468105, '[native code]', 21, false)), J$.U(551842, '-', J$.T(4468121, 1, 22, false))));
                        } catch (J$e) {
                            J$.Ex(4469753, J$e);
                        } finally {
                            if (J$.Fr(4469761))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(4468305, arguments.callee, this, arguments);
                            arguments = J$.N(4468313, 'arguments', arguments, true, false, false);
                            o = J$.N(4468321, 'o', o, true, false, false);
                            p = J$.N(4468329, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(4468257, '_setPrototypeOf', J$.C(248504, J$.G(4468169, J$.I(typeof Object === 'undefined' ? Object = J$.R(4468161, 'Object', undefined, true, true) : Object = J$.R(4468161, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(4468249, function _setPrototypeOf(o, p) {
                                jalangiLabel15:
                                    while (true) {
                                        try {
                                            J$.Fe(4468217, arguments.callee, this, arguments);
                                            arguments = J$.N(4468225, 'arguments', arguments, true, false, false);
                                            o = J$.N(4468233, 'o', o, true, false, false);
                                            p = J$.N(4468241, 'p', p, true, false, false);
                                            J$.P(4468193, J$.R(4468177, 'o', o, false, false), '__proto__', J$.R(4468185, 'p', p, false, false));
                                            return J$.Rt(4468209, J$.R(4468201, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4469769, J$e);
                                        } finally {
                                            if (J$.Fr(4469777))
                                                continue jalangiLabel15;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(4468297, J$.F(4468289, J$.R(4468265, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(4468273, 'o', o, false, false), J$.R(4468281, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(4469785, J$e);
                        } finally {
                            if (J$.Fr(4469793))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(4468489, arguments.callee, this, arguments);
                            arguments = J$.N(4468497, 'arguments', arguments, true, false, false);
                            o = J$.N(4468505, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(4468449, '_getPrototypeOf', J$.C(248520, J$.G(4468345, J$.I(typeof Object === 'undefined' ? Object = J$.R(4468337, 'Object', undefined, true, true) : Object = J$.R(4468337, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(4468361, J$.I(typeof Object === 'undefined' ? Object = J$.R(4468353, 'Object', undefined, true, true) : Object = J$.R(4468353, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(4468441, function _getPrototypeOf(o) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(4468417, arguments.callee, this, arguments);
                                            arguments = J$.N(4468425, 'arguments', arguments, true, false, false);
                                            o = J$.N(4468433, 'o', o, true, false, false);
                                            return J$.Rt(4468409, J$.C(248512, J$.G(4468377, J$.R(4468369, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(4468401, J$.I(typeof Object === 'undefined' ? Object = J$.R(4468385, 'Object', undefined, true, true) : Object = J$.R(4468385, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(4468393, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(4469801, J$e);
                                        } finally {
                                            if (J$.Fr(4469809))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(4468481, J$.F(4468473, J$.R(4468457, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(4468465, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(4469817, J$e);
                        } finally {
                            if (J$.Fr(4469825))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof = J$.N(4469353, '_typeof', J$.T(4469345, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(4469369, '_classCallCheck', J$.T(4469361, _classCallCheck, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(4469385, '_possibleConstructorReturn', J$.T(4469377, _possibleConstructorReturn, 12, false), true, false, false);
            _inherits = J$.N(4469401, '_inherits', J$.T(4469393, _inherits, 12, false), true, false, false);
            _assertThisInitialized = J$.N(4469417, '_assertThisInitialized', J$.T(4469409, _assertThisInitialized, 12, false), true, false, false);
            _wrapNativeSuper = J$.N(4469433, '_wrapNativeSuper', J$.T(4469425, _wrapNativeSuper, 12, false), true, false, false);
            isNativeReflectConstruct = J$.N(4469449, 'isNativeReflectConstruct', J$.T(4469441, isNativeReflectConstruct, 12, false), true, false, false);
            _construct = J$.N(4469465, '_construct', J$.T(4469457, _construct, 12, false), true, false, false);
            _isNativeFunction = J$.N(4469481, '_isNativeFunction', J$.T(4469473, _isNativeFunction, 12, false), true, false, false);
            _setPrototypeOf = J$.N(4469497, '_setPrototypeOf', J$.T(4469489, _setPrototypeOf, 12, false), true, false, false);
            _getPrototypeOf = J$.N(4469513, '_getPrototypeOf', J$.T(4469505, _getPrototypeOf, 12, false), true, false, false);
            J$.N(4469521, 'InvalidFormatError', InvalidFormatError, false, false, false);
            J$.T(4466009, 'use strict', 21, false);
            var InvalidFormatError = J$.W(4468929, 'InvalidFormatError', J$.F(4468921, J$.T(4468889, function (_Error) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(4468849, arguments.callee, this, arguments);
                            function InvalidFormatError(formatFn) {
                                jalangiLabel19:
                                    while (true) {
                                        try {
                                            J$.Fe(4468801, arguments.callee, this, arguments);
                                            arguments = J$.N(4468809, 'arguments', arguments, true, false, false);
                                            formatFn = J$.N(4468817, 'formatFn', formatFn, true, false, false);
                                            J$.N(4468825, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(4468569, J$.R(4468545, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(4468553, 'this', this, false, false), J$.R(4468561, 'InvalidFormatError', InvalidFormatError, false, false));
                                            _this = J$.W(4468713, '_this', J$.F(4468705, J$.R(4468577, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(4468585, 'this', this, false, false), J$.M(4468697, J$.F(4468609, J$.R(4468593, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(4468601, 'InvalidFormatError', InvalidFormatError, false, false)), 'call', false)(J$.R(4468617, 'this', this, false, false), J$.M(4468689, J$.T(4468625, 'Format functions must be synchronous taking a two arguments: (info, opts)\nFound: ', 21, false), 'concat', false)(J$.G(4468673, J$.M(4468657, J$.M(4468641, J$.R(4468633, 'formatFn', formatFn, false, false), 'toString', false)(), 'split', false)(J$.T(4468649, '\n', 21, false)), J$.T(4468665, 0, 22, false)), J$.T(4468681, '\n', 21, false)))), _this, false, false);
                                            J$.M(4468777, J$.I(typeof Error === 'undefined' ? Error = J$.R(4468721, 'Error', undefined, true, true) : Error = J$.R(4468721, 'Error', Error, true, true)), 'captureStackTrace', false)(J$.F(4468761, J$.R(4468729, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.F(4468753, J$.R(4468737, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(4468745, '_this', _this, false, false))), J$.R(4468769, 'InvalidFormatError', InvalidFormatError, false, false));
                                            return J$.Rt(4468793, J$.R(4468785, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4469833, J$e);
                                        } finally {
                                            if (J$.Fr(4469841))
                                                continue jalangiLabel19;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(4468857, 'arguments', arguments, true, false, false);
                            _Error = J$.N(4468865, '_Error', _Error, true, false, false);
                            InvalidFormatError = J$.N(4468881, 'InvalidFormatError', J$.T(4468873, InvalidFormatError, 12, false), true, false, false);
                            J$.F(4468537, J$.R(4468513, '_inherits', _inherits, false, true), false)(J$.R(4468521, 'InvalidFormatError', InvalidFormatError, false, false), J$.R(4468529, '_Error', _Error, false, false));
                            return J$.Rt(4468841, J$.R(4468833, 'InvalidFormatError', InvalidFormatError, false, false));
                        } catch (J$e) {
                            J$.Ex(4469849, J$e);
                        } finally {
                            if (J$.Fr(4469857))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.F(4468913, J$.R(4468897, '_wrapNativeSuper', _wrapNativeSuper, false, true), false)(J$.I(typeof Error === 'undefined' ? Error = J$.R(4468905, 'Error', undefined, true, true) : Error = J$.R(4468905, 'Error', Error, true, true)))), InvalidFormatError, false, true);
            J$.P(4469329, J$.I(typeof module === 'undefined' ? module = J$.R(4468937, 'module', undefined, true, true) : module = J$.R(4468937, 'module', module, true, true)), 'exports', J$.T(4469321, function (formatFn) {
                jalangiLabel23:
                    while (true) {
                        try {
                            J$.Fe(4469265, arguments.callee, this, arguments);
                            function Format() {
                                jalangiLabel21:
                                    while (true) {
                                        try {
                                            J$.Fe(4469113, arguments.callee, this, arguments);
                                            arguments = J$.N(4469121, 'arguments', arguments, true, false, false);
                                            J$.N(4469129, 'options', options, false, false, false);
                                            var options = J$.W(4469081, 'options', J$.C(248544, J$.C(248536, J$.B(551866, '>', J$.G(4469001, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4468993, 'arguments', undefined, true, true) : arguments = J$.R(4468993, 'arguments', arguments, true, true)), 'length'), J$.T(4469009, 0, 22, false))) ? J$.B(551874, '!==', J$.G(4469033, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4469017, 'arguments', undefined, true, true) : arguments = J$.R(4469017, 'arguments', arguments, true, true)), J$.T(4469025, 0, 22, false)), J$.T(4469041, undefined, 24, false)) : J$._()) ? J$.G(4469065, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4469049, 'arguments', undefined, true, true) : arguments = J$.R(4469049, 'arguments', arguments, true, true)), J$.T(4469057, 0, 22, false)) : J$.T(4469073, {}, 11, false), options, false, false);
                                            J$.P(4469105, J$.R(4469089, 'this', this, false, false), 'options', J$.R(4469097, 'options', options, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4469865, J$e);
                                        } finally {
                                            if (J$.Fr(4469873))
                                                continue jalangiLabel21;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            function createFormatWrap(opts) {
                                jalangiLabel22:
                                    while (true) {
                                        try {
                                            J$.Fe(4469201, arguments.callee, this, arguments);
                                            arguments = J$.N(4469209, 'arguments', arguments, true, false, false);
                                            opts = J$.N(4469217, 'opts', opts, true, false, false);
                                            return J$.Rt(4469193, J$.F(4469185, J$.R(4469169, 'Format', Format, false, false), true)(J$.R(4469177, 'opts', opts, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(4469881, J$e);
                                        } finally {
                                            if (J$.Fr(4469889))
                                                continue jalangiLabel22;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(4469273, 'arguments', arguments, true, false, false);
                            formatFn = J$.N(4469281, 'formatFn', formatFn, true, false, false);
                            Format = J$.N(4469297, 'Format', J$.T(4469289, Format, 12, false), true, false, false);
                            createFormatWrap = J$.N(4469313, 'createFormatWrap', J$.T(4469305, createFormatWrap, 12, false), true, false, false);
                            if (J$.C(248528, J$.B(551858, '>', J$.G(4468953, J$.R(4468945, 'formatFn', formatFn, false, false), 'length'), J$.T(4468961, 2, 22, false)))) {
                                throw J$.F(4468985, J$.R(4468969, 'InvalidFormatError', InvalidFormatError, false, true), true)(J$.R(4468977, 'formatFn', formatFn, false, false));
                            }
                            J$.P(4469161, J$.G(4469145, J$.R(4469137, 'Format', Format, false, false), 'prototype'), 'transform', J$.R(4469153, 'formatFn', formatFn, false, false));
                            J$.P(4469241, J$.R(4469225, 'createFormatWrap', createFormatWrap, false, false), 'Format', J$.R(4469233, 'Format', Format, false, false));
                            return J$.Rt(4469257, J$.R(4469249, 'createFormatWrap', createFormatWrap, false, false));
                        } catch (J$e) {
                            J$.Ex(4469897, J$e);
                        } finally {
                            if (J$.Fr(4469905))
                                continue jalangiLabel23;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(4469913, J$e);
        } finally {
            if (J$.Sr(4469921))
                continue jalangiLabel24;
            else
                break jalangiLabel24;
        }
    }
// JALANGI DO NOT INSTRUMENT
