// logform/dist/pad-levels.js
J$.noInstrEval = false;
jalangiLabel16:
    while (true) {
        try {
            J$.Se(4477201, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/logform/dist/pad-levels.js');
            function _toConsumableArray(arr) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(4474473, arguments.callee, this, arguments);
                            arguments = J$.N(4474481, 'arguments', arguments, true, false, false);
                            arr = J$.N(4474489, 'arr', arr, true, false, false);
                            return J$.Rt(4474465, J$.C(248712, J$.C(248704, J$.F(4474417, J$.R(4474401, '_arrayWithoutHoles', _arrayWithoutHoles, false, true), false)(J$.R(4474409, 'arr', arr, false, false))) ? J$._() : J$.F(4474441, J$.R(4474425, '_iterableToArray', _iterableToArray, false, true), false)(J$.R(4474433, 'arr', arr, false, false))) ? J$._() : J$.F(4474457, J$.R(4474449, '_nonIterableSpread', _nonIterableSpread, false, true), false)());
                        } catch (J$e) {
                            J$.Ex(4477361, J$e);
                        } finally {
                            if (J$.Fr(4477369))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _nonIterableSpread() {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(4474521, arguments.callee, this, arguments);
                            arguments = J$.N(4474529, 'arguments', arguments, true, false, false);
                            throw J$.F(4474513, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(4474497, 'TypeError', undefined, true, true) : TypeError = J$.R(4474497, 'TypeError', TypeError, true, true)), true)(J$.T(4474505, 'Invalid attempt to spread non-iterable instance', 21, false));
                        } catch (J$e) {
                            J$.Ex(4477377, J$e);
                        } finally {
                            if (J$.Fr(4477385))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _iterableToArray(iter) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(4474657, arguments.callee, this, arguments);
                            arguments = J$.N(4474665, 'arguments', arguments, true, false, false);
                            iter = J$.N(4474673, 'iter', iter, true, false, false);
                            if (J$.C(248728, J$.C(248720, J$.B(552010, 'in', J$.G(4474545, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(4474537, 'Symbol', undefined, true, true) : Symbol = J$.R(4474537, 'Symbol', Symbol, true, true)), 'iterator'), J$.F(4474569, J$.I(typeof Object === 'undefined' ? Object = J$.R(4474553, 'Object', undefined, true, true) : Object = J$.R(4474553, 'Object', Object, true, true)), false)(J$.R(4474561, 'iter', iter, false, false)))) ? J$._() : J$.B(552018, '===', J$.M(4474609, J$.G(4474593, J$.G(4474585, J$.I(typeof Object === 'undefined' ? Object = J$.R(4474577, 'Object', undefined, true, true) : Object = J$.R(4474577, 'Object', Object, true, true)), 'prototype'), 'toString'), 'call', false)(J$.R(4474601, 'iter', iter, false, false)), J$.T(4474617, '[object Arguments]', 21, false))))
                                return J$.Rt(4474649, J$.M(4474641, J$.I(typeof Array === 'undefined' ? Array = J$.R(4474625, 'Array', undefined, true, true) : Array = J$.R(4474625, 'Array', Array, true, true)), 'from', false)(J$.R(4474633, 'iter', iter, false, false)));
                        } catch (J$e) {
                            J$.Ex(4477393, J$e);
                        } finally {
                            if (J$.Fr(4477401))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _arrayWithoutHoles(arr) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(4474865, arguments.callee, this, arguments);
                            arguments = J$.N(4474873, 'arguments', arguments, true, false, false);
                            arr = J$.N(4474881, 'arr', arr, true, false, false);
                            J$.N(4474889, 'i', i, false, false, false);
                            J$.N(4474897, 'arr2', arr2, false, false, false);
                            if (J$.C(248744, J$.M(4474697, J$.I(typeof Array === 'undefined' ? Array = J$.R(4474681, 'Array', undefined, true, true) : Array = J$.R(4474681, 'Array', Array, true, true)), 'isArray', false)(J$.R(4474689, 'arr', arr, false, false)))) {
                                for (var i = J$.W(4474745, 'i', J$.T(4474705, 0, 22, false), i, false, false), arr2 = J$.W(4474753, 'arr2', J$.F(4474737, J$.I(typeof Array === 'undefined' ? Array = J$.R(4474713, 'Array', undefined, true, true) : Array = J$.R(4474713, 'Array', Array, true, true)), true)(J$.G(4474729, J$.R(4474721, 'arr', arr, false, false), 'length')), arr2, false, false); J$.C(248736, J$.B(552026, '<', J$.R(4474761, 'i', i, false, false), J$.G(4474777, J$.R(4474769, 'arr', arr, false, false), 'length'))); J$.B(552050, '-', i = J$.W(4474793, 'i', J$.B(552042, '+', J$.U(552034, '+', J$.R(4474785, 'i', i, false, false)), 1), i, false, false), 1)) {
                                    J$.P(4474841, J$.R(4474801, 'arr2', arr2, false, false), J$.R(4474809, 'i', i, false, false), J$.G(4474833, J$.R(4474817, 'arr', arr, false, false), J$.R(4474825, 'i', i, false, false)));
                                }
                                return J$.Rt(4474857, J$.R(4474849, 'arr2', arr2, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(4477409, J$e);
                        } finally {
                            if (J$.Fr(4477417))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(4474945, arguments.callee, this, arguments);
                            arguments = J$.N(4474953, 'arguments', arguments, true, false, false);
                            instance = J$.N(4474961, 'instance', instance, true, false, false);
                            Constructor = J$.N(4474969, 'Constructor', Constructor, true, false, false);
                            if (J$.C(248752, J$.U(552066, '!', J$.B(552058, 'instanceof', J$.R(4474905, 'instance', instance, false, false), J$.R(4474913, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(4474937, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(4474921, 'TypeError', undefined, true, true) : TypeError = J$.R(4474921, 'TypeError', TypeError, true, true)), true)(J$.T(4474929, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(4477425, J$e);
                        } finally {
                            if (J$.Fr(4477433))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(4475217, arguments.callee, this, arguments);
                            arguments = J$.N(4475225, 'arguments', arguments, true, false, false);
                            target = J$.N(4475233, 'target', target, true, false, false);
                            props = J$.N(4475241, 'props', props, true, false, false);
                            J$.N(4475249, 'i', i, false, false, false);
                            J$.N(4475257, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(4474985, 'i', J$.T(4474977, 0, 22, false), i, false, false); J$.C(248776, J$.B(552074, '<', J$.R(4474993, 'i', i, false, false), J$.G(4475009, J$.R(4475001, 'props', props, false, false), 'length'))); J$.B(552098, '-', i = J$.W(4475025, 'i', J$.B(552090, '+', J$.U(552082, '+', J$.R(4475017, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(4475057, 'descriptor', J$.G(4475049, J$.R(4475033, 'props', props, false, false), J$.R(4475041, 'i', i, false, false)), descriptor, false, false);
                                J$.P(4475097, J$.R(4475065, 'descriptor', descriptor, false, false), 'enumerable', J$.C(248760, J$.G(4475081, J$.R(4475073, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(4475089, false, 23, false));
                                J$.P(4475121, J$.R(4475105, 'descriptor', descriptor, false, false), 'configurable', J$.T(4475113, true, 23, false));
                                if (J$.C(248768, J$.B(552106, 'in', J$.T(4475129, 'value', 21, false), J$.R(4475137, 'descriptor', descriptor, false, false))))
                                    J$.P(4475161, J$.R(4475145, 'descriptor', descriptor, false, false), 'writable', J$.T(4475153, true, 23, false));
                                J$.M(4475209, J$.I(typeof Object === 'undefined' ? Object = J$.R(4475169, 'Object', undefined, true, true) : Object = J$.R(4475169, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(4475177, 'target', target, false, false), J$.G(4475193, J$.R(4475185, 'descriptor', descriptor, false, false), 'key'), J$.R(4475201, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(4477441, J$e);
                        } finally {
                            if (J$.Fr(4477449))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(4475369, arguments.callee, this, arguments);
                            arguments = J$.N(4475377, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(4475385, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(4475393, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(4475401, 'staticProps', staticProps, true, false, false);
                            if (J$.C(248784, J$.R(4475265, 'protoProps', protoProps, false, false)))
                                J$.F(4475305, J$.R(4475273, '_defineProperties', _defineProperties, false, true), false)(J$.G(4475289, J$.R(4475281, 'Constructor', Constructor, false, false), 'prototype'), J$.R(4475297, 'protoProps', protoProps, false, false));
                            if (J$.C(248792, J$.R(4475313, 'staticProps', staticProps, false, false)))
                                J$.F(4475345, J$.R(4475321, '_defineProperties', _defineProperties, false, true), false)(J$.R(4475329, 'Constructor', Constructor, false, false), J$.R(4475337, 'staticProps', staticProps, false, false));
                            return J$.Rt(4475361, J$.R(4475353, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(4477457, J$e);
                        } finally {
                            if (J$.Fr(4477465))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _toConsumableArray = J$.N(4477217, '_toConsumableArray', J$.T(4477209, _toConsumableArray, 12, false), true, false, false);
            _nonIterableSpread = J$.N(4477233, '_nonIterableSpread', J$.T(4477225, _nonIterableSpread, 12, false), true, false, false);
            _iterableToArray = J$.N(4477249, '_iterableToArray', J$.T(4477241, _iterableToArray, 12, false), true, false, false);
            _arrayWithoutHoles = J$.N(4477265, '_arrayWithoutHoles', J$.T(4477257, _arrayWithoutHoles, 12, false), true, false, false);
            _classCallCheck = J$.N(4477281, '_classCallCheck', J$.T(4477273, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(4477297, '_defineProperties', J$.T(4477289, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(4477313, '_createClass', J$.T(4477305, _createClass, 12, false), true, false, false);
            J$.N(4477321, '_require', _require, false, false, false);
            J$.N(4477329, 'configs', configs, false, false, false);
            J$.N(4477337, 'LEVEL', LEVEL, false, false, false);
            J$.N(4477345, 'MESSAGE', MESSAGE, false, false, false);
            J$.N(4477353, 'Padder', Padder, false, false, false);
            J$.T(4474393, 'use strict', 21, false);
            var _require = J$.W(4475481, '_require', J$.F(4475425, J$.I(typeof require === 'undefined' ? require = J$.R(4475409, 'require', undefined, true, true) : require = J$.R(4475409, 'require', require, true, true)), false)(J$.T(4475417, 'triple-beam', 21, false)), _require, false, true), configs = J$.W(4475489, 'configs', J$.G(4475441, J$.R(4475433, '_require', _require, false, true), 'configs'), configs, false, true), LEVEL = J$.W(4475497, 'LEVEL', J$.G(4475457, J$.R(4475449, '_require', _require, false, true), 'LEVEL'), LEVEL, false, true), MESSAGE = J$.W(4475505, 'MESSAGE', J$.G(4475473, J$.R(4475465, '_require', _require, false, true), 'MESSAGE'), MESSAGE, false, true);
            var Padder = J$.W(4477057, 'Padder', J$.F(4477049, J$.T(4477041, function () {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(4477009, arguments.callee, this, arguments);
                            function Padder() {
                                jalangiLabel7:
                                    while (true) {
                                        try {
                                            J$.Fe(4475753, arguments.callee, this, arguments);
                                            arguments = J$.N(4475761, 'arguments', arguments, true, false, false);
                                            J$.N(4475769, 'opts', opts, false, false, false);
                                            var opts = J$.W(4475625, 'opts', J$.C(248808, J$.C(248800, J$.B(552114, '>', J$.G(4475521, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4475513, 'arguments', undefined, true, true) : arguments = J$.R(4475513, 'arguments', arguments, true, true)), 'length'), J$.T(4475529, 0, 22, false))) ? J$.B(552122, '!==', J$.G(4475553, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4475537, 'arguments', undefined, true, true) : arguments = J$.R(4475537, 'arguments', arguments, true, true)), J$.T(4475545, 0, 22, false)), J$.T(4475561, undefined, 24, false)) : J$._()) ? J$.G(4475585, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4475569, 'arguments', undefined, true, true) : arguments = J$.R(4475569, 'arguments', arguments, true, true)), J$.T(4475577, 0, 22, false)) : J$.T(4475617, { levels: J$.G(4475609, J$.G(4475601, J$.R(4475593, 'configs', configs, false, true), 'npm'), 'levels') }, 11, false), opts, false, false);
                                            J$.F(4475657, J$.R(4475633, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(4475641, 'this', this, false, false), J$.R(4475649, 'Padder', Padder, false, false));
                                            J$.P(4475721, J$.R(4475665, 'this', this, false, false), 'paddings', J$.M(4475713, J$.R(4475673, 'Padder', Padder, false, false), 'paddingForLevels', false)(J$.G(4475689, J$.R(4475681, 'opts', opts, false, false), 'levels'), J$.G(4475705, J$.R(4475697, 'opts', opts, false, false), 'filler')));
                                            J$.P(4475745, J$.R(4475729, 'this', this, false, false), 'options', J$.R(4475737, 'opts', opts, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4477473, J$e);
                                        } finally {
                                            if (J$.Fr(4477481))
                                                continue jalangiLabel7;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(4477017, 'arguments', arguments, true, false, false);
                            Padder = J$.N(4477033, 'Padder', J$.T(4477025, Padder, 12, false), true, false, false);
                            J$.F(4476985, J$.R(4475777, '_createClass', _createClass, false, true), false)(J$.R(4475785, 'Padder', Padder, false, false), J$.T(4476113, [J$.T(4476105, {
                                    key: J$.T(4475793, 'transform', 21, false),
                                    value: J$.T(4476097, function transform(info, opts) {
                                        jalangiLabel8:
                                            while (true) {
                                                try {
                                                    J$.Fe(4476065, arguments.callee, this, arguments);
                                                    arguments = J$.N(4476073, 'arguments', arguments, true, false, false);
                                                    info = J$.N(4476081, 'info', info, true, false, false);
                                                    opts = J$.N(4476089, 'opts', opts, true, false, false);
                                                    J$.P(4475897, J$.R(4475801, 'info', info, false, false), 'message', J$.M(4475889, J$.M(4475865, J$.T(4475809, '', 21, false), 'concat', false)(J$.G(4475857, J$.G(4475825, J$.R(4475817, 'this', this, false, false), 'paddings'), J$.G(4475849, J$.R(4475833, 'info', info, false, false), J$.R(4475841, 'LEVEL', LEVEL, false, true)))), 'concat', false)(J$.G(4475881, J$.R(4475873, 'info', info, false, false), 'message')));
                                                    if (J$.C(248816, J$.G(4475921, J$.R(4475905, 'info', info, false, false), J$.R(4475913, 'MESSAGE', MESSAGE, false, true)))) {
                                                        J$.P(4476041, J$.R(4475929, 'info', info, false, false), J$.R(4475937, 'MESSAGE', MESSAGE, false, true), J$.M(4476033, J$.M(4476001, J$.T(4475945, '', 21, false), 'concat', false)(J$.G(4475993, J$.G(4475961, J$.R(4475953, 'this', this, false, false), 'paddings'), J$.G(4475985, J$.R(4475969, 'info', info, false, false), J$.R(4475977, 'LEVEL', LEVEL, false, true)))), 'concat', false)(J$.G(4476025, J$.R(4476009, 'info', info, false, false), J$.R(4476017, 'MESSAGE', MESSAGE, false, true))));
                                                    }
                                                    return J$.Rt(4476057, J$.R(4476049, 'info', info, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(4477489, J$e);
                                                } finally {
                                                    if (J$.Fr(4477497))
                                                        continue jalangiLabel8;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)], 10, false), J$.T(4476977, [
                                J$.T(4476329, {
                                    key: J$.T(4476121, 'getLongestLevel', 21, false),
                                    value: J$.T(4476321, function getLongestLevel(levels) {
                                        jalangiLabel10:
                                            while (true) {
                                                try {
                                                    J$.Fe(4476289, arguments.callee, this, arguments);
                                                    arguments = J$.N(4476297, 'arguments', arguments, true, false, false);
                                                    levels = J$.N(4476305, 'levels', levels, true, false, false);
                                                    J$.N(4476313, 'lvls', lvls, false, false, false);
                                                    var lvls = J$.W(4476217, 'lvls', J$.M(4476209, J$.M(4476145, J$.I(typeof Object === 'undefined' ? Object = J$.R(4476129, 'Object', undefined, true, true) : Object = J$.R(4476129, 'Object', Object, true, true)), 'keys', false)(J$.R(4476137, 'levels', levels, false, false)), 'map', false)(J$.T(4476201, function (level) {
                                                        jalangiLabel9:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(4476177, arguments.callee, this, arguments);
                                                                    arguments = J$.N(4476185, 'arguments', arguments, true, false, false);
                                                                    level = J$.N(4476193, 'level', level, true, false, false);
                                                                    return J$.Rt(4476169, J$.G(4476161, J$.R(4476153, 'level', level, false, false), 'length'));
                                                                } catch (J$e) {
                                                                    J$.Ex(4477505, J$e);
                                                                } finally {
                                                                    if (J$.Fr(4477513))
                                                                        continue jalangiLabel9;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false)), lvls, false, false);
                                                    return J$.Rt(4476281, J$.M(4476273, J$.G(4476233, J$.I(typeof Math === 'undefined' ? Math = J$.R(4476225, 'Math', undefined, true, true) : Math = J$.R(4476225, 'Math', Math, true, true)), 'max'), 'apply', false)(J$.I(typeof Math === 'undefined' ? Math = J$.R(4476241, 'Math', undefined, true, true) : Math = J$.R(4476241, 'Math', Math, true, true)), J$.F(4476265, J$.R(4476249, '_toConsumableArray', _toConsumableArray, false, true), false)(J$.R(4476257, 'lvls', lvls, false, false))));
                                                } catch (J$e) {
                                                    J$.Ex(4477521, J$e);
                                                } finally {
                                                    if (J$.Fr(4477529))
                                                        continue jalangiLabel10;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(4476609, {
                                    key: J$.T(4476337, 'paddingForLevel', 21, false),
                                    value: J$.T(4476601, function paddingForLevel(level, filler, maxLength) {
                                        jalangiLabel11:
                                            while (true) {
                                                try {
                                                    J$.Fe(4476537, arguments.callee, this, arguments);
                                                    arguments = J$.N(4476545, 'arguments', arguments, true, false, false);
                                                    level = J$.N(4476553, 'level', level, true, false, false);
                                                    filler = J$.N(4476561, 'filler', filler, true, false, false);
                                                    maxLength = J$.N(4476569, 'maxLength', maxLength, true, false, false);
                                                    J$.N(4476577, 'targetLen', targetLen, false, false, false);
                                                    J$.N(4476585, 'rep', rep, false, false, false);
                                                    J$.N(4476593, 'padding', padding, false, false, false);
                                                    var targetLen = J$.W(4476377, 'targetLen', J$.B(552138, '-', J$.B(552130, '+', J$.R(4476345, 'maxLength', maxLength, false, false), J$.T(4476353, 1, 22, false)), J$.G(4476369, J$.R(4476361, 'level', level, false, false), 'length')), targetLen, false, false);
                                                    var rep = J$.W(4476425, 'rep', J$.M(4476417, J$.I(typeof Math === 'undefined' ? Math = J$.R(4476385, 'Math', undefined, true, true) : Math = J$.R(4476385, 'Math', Math, true, true)), 'floor', false)(J$.B(552146, '/', J$.R(4476393, 'targetLen', targetLen, false, false), J$.G(4476409, J$.R(4476401, 'filler', filler, false, false), 'length'))), rep, false, false);
                                                    var padding = J$.W(4476489, 'padding', J$.M(4476481, J$.M(4476449, J$.T(4476433, '', 21, false), 'concat', false)(J$.R(4476441, 'filler', filler, false, false)), 'concat', false)(J$.M(4476473, J$.R(4476457, 'filler', filler, false, false), 'repeat', false)(J$.R(4476465, 'rep', rep, false, false))), padding, false, false);
                                                    return J$.Rt(4476529, J$.M(4476521, J$.R(4476497, 'padding', padding, false, false), 'slice', false)(J$.T(4476505, 0, 22, false), J$.R(4476513, 'targetLen', targetLen, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(4477537, J$e);
                                                } finally {
                                                    if (J$.Fr(4477545))
                                                        continue jalangiLabel11;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(4476969, {
                                    key: J$.T(4476617, 'paddingForLevels', 21, false),
                                    value: J$.T(4476961, function paddingForLevels(levels) {
                                        jalangiLabel13:
                                            while (true) {
                                                try {
                                                    J$.Fe(4476921, arguments.callee, this, arguments);
                                                    arguments = J$.N(4476929, 'arguments', arguments, true, false, false);
                                                    levels = J$.N(4476937, 'levels', levels, true, false, false);
                                                    J$.N(4476945, 'filler', filler, false, false, false);
                                                    J$.N(4476953, 'maxLength', maxLength, false, false, false);
                                                    var filler = J$.W(4476713, 'filler', J$.C(248832, J$.C(248824, J$.B(552154, '>', J$.G(4476633, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4476625, 'arguments', undefined, true, true) : arguments = J$.R(4476625, 'arguments', arguments, true, true)), 'length'), J$.T(4476641, 1, 22, false))) ? J$.B(552162, '!==', J$.G(4476665, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4476649, 'arguments', undefined, true, true) : arguments = J$.R(4476649, 'arguments', arguments, true, true)), J$.T(4476657, 1, 22, false)), J$.T(4476673, undefined, 24, false)) : J$._()) ? J$.G(4476697, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4476681, 'arguments', undefined, true, true) : arguments = J$.R(4476681, 'arguments', arguments, true, true)), J$.T(4476689, 1, 22, false)) : J$.T(4476705, ' ', 21, false), filler, false, false);
                                                    var maxLength = J$.W(4476745, 'maxLength', J$.M(4476737, J$.R(4476721, 'Padder', Padder, false, false), 'getLongestLevel', false)(J$.R(4476729, 'levels', levels, false, false)), maxLength, false, false);
                                                    return J$.Rt(4476913, J$.M(4476905, J$.M(4476769, J$.I(typeof Object === 'undefined' ? Object = J$.R(4476753, 'Object', undefined, true, true) : Object = J$.R(4476753, 'Object', Object, true, true)), 'keys', false)(J$.R(4476761, 'levels', levels, false, false)), 'reduce', false)(J$.T(4476889, function (acc, level) {
                                                        jalangiLabel12:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(4476857, arguments.callee, this, arguments);
                                                                    arguments = J$.N(4476865, 'arguments', arguments, true, false, false);
                                                                    acc = J$.N(4476873, 'acc', acc, true, false, false);
                                                                    level = J$.N(4476881, 'level', level, true, false, false);
                                                                    J$.P(4476833, J$.R(4476777, 'acc', acc, false, false), J$.R(4476785, 'level', level, false, false), J$.M(4476825, J$.R(4476793, 'Padder', Padder, false, false), 'paddingForLevel', false)(J$.R(4476801, 'level', level, false, false), J$.R(4476809, 'filler', filler, false, false), J$.R(4476817, 'maxLength', maxLength, false, false)));
                                                                    return J$.Rt(4476849, J$.R(4476841, 'acc', acc, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(4477553, J$e);
                                                                } finally {
                                                                    if (J$.Fr(4477561))
                                                                        continue jalangiLabel12;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false), J$.T(4476897, {}, 11, false)));
                                                } catch (J$e) {
                                                    J$.Ex(4477569, J$e);
                                                } finally {
                                                    if (J$.Fr(4477577))
                                                        continue jalangiLabel13;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(4477001, J$.R(4476993, 'Padder', Padder, false, false));
                        } catch (J$e) {
                            J$.Ex(4477585, J$e);
                        } finally {
                            if (J$.Fr(4477593))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), Padder, false, true);
            J$.P(4477137, J$.I(typeof module === 'undefined' ? module = J$.R(4477065, 'module', undefined, true, true) : module = J$.R(4477065, 'module', module, true, true)), 'exports', J$.T(4477129, function (opts) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(4477105, arguments.callee, this, arguments);
                            arguments = J$.N(4477113, 'arguments', arguments, true, false, false);
                            opts = J$.N(4477121, 'opts', opts, true, false, false);
                            return J$.Rt(4477097, J$.F(4477089, J$.R(4477073, 'Padder', Padder, false, true), true)(J$.R(4477081, 'opts', opts, false, false)));
                        } catch (J$e) {
                            J$.Ex(4477601, J$e);
                        } finally {
                            if (J$.Fr(4477609))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(4477193, J$.G(4477153, J$.I(typeof module === 'undefined' ? module = J$.R(4477145, 'module', undefined, true, true) : module = J$.R(4477145, 'module', module, true, true)), 'exports'), 'Padder', J$.P(4477185, J$.G(4477169, J$.I(typeof module === 'undefined' ? module = J$.R(4477161, 'module', undefined, true, true) : module = J$.R(4477161, 'module', module, true, true)), 'exports'), 'Format', J$.R(4477177, 'Padder', Padder, false, true)));
        } catch (J$e) {
            J$.Ex(4477617, J$e);
        } finally {
            if (J$.Sr(4477625))
                continue jalangiLabel16;
            else
                break jalangiLabel16;
        }
    }
// JALANGI DO NOT INSTRUMENT
