// ecc-jsbn/lib/sec.js
J$.noInstrEval = false;
jalangiLabel14:
    while (true) {
        try {
            J$.Se(2272481, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchDb/node_modules/ecc-jsbn/lib/sec.js');
            function X9ECParameters(curve, g, n, h) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(2269065, arguments.callee, this, arguments);
                            arguments = J$.N(2269073, 'arguments', arguments, true, false, false);
                            curve = J$.N(2269081, 'curve', curve, true, false, false);
                            g = J$.N(2269089, 'g', g, true, false, false);
                            n = J$.N(2269097, 'n', n, true, false, false);
                            h = J$.N(2269105, 'h', h, true, false, false);
                            J$.P(2268985, J$.R(2268969, 'this', this, false, false), 'curve', J$.R(2268977, 'curve', curve, false, false));
                            J$.P(2269009, J$.R(2268993, 'this', this, false, false), 'g', J$.R(2269001, 'g', g, false, false));
                            J$.P(2269033, J$.R(2269017, 'this', this, false, false), 'n', J$.R(2269025, 'n', n, false, false));
                            J$.P(2269057, J$.R(2269041, 'this', this, false, false), 'h', J$.R(2269049, 'h', h, false, false));
                        } catch (J$e) {
                            J$.Ex(2272729, J$e);
                        } finally {
                            if (J$.Fr(2272737))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function x9getCurve() {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(2269137, arguments.callee, this, arguments);
                            arguments = J$.N(2269145, 'arguments', arguments, true, false, false);
                            return J$.Rt(2269129, J$.G(2269121, J$.R(2269113, 'this', this, false, false), 'curve'));
                        } catch (J$e) {
                            J$.Ex(2272745, J$e);
                        } finally {
                            if (J$.Fr(2272753))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function x9getG() {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(2269177, arguments.callee, this, arguments);
                            arguments = J$.N(2269185, 'arguments', arguments, true, false, false);
                            return J$.Rt(2269169, J$.G(2269161, J$.R(2269153, 'this', this, false, false), 'g'));
                        } catch (J$e) {
                            J$.Ex(2272761, J$e);
                        } finally {
                            if (J$.Fr(2272769))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function x9getN() {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(2269217, arguments.callee, this, arguments);
                            arguments = J$.N(2269225, 'arguments', arguments, true, false, false);
                            return J$.Rt(2269209, J$.G(2269201, J$.R(2269193, 'this', this, false, false), 'n'));
                        } catch (J$e) {
                            J$.Ex(2272777, J$e);
                        } finally {
                            if (J$.Fr(2272785))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function x9getH() {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(2269257, arguments.callee, this, arguments);
                            arguments = J$.N(2269265, 'arguments', arguments, true, false, false);
                            return J$.Rt(2269249, J$.G(2269241, J$.R(2269233, 'this', this, false, false), 'h'));
                        } catch (J$e) {
                            J$.Ex(2272793, J$e);
                        } finally {
                            if (J$.Fr(2272801))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function fromHex(s) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(2269441, arguments.callee, this, arguments);
                            arguments = J$.N(2269449, 'arguments', arguments, true, false, false);
                            s = J$.N(2269457, 's', s, true, false, false);
                            return J$.Rt(2269433, J$.F(2269425, J$.R(2269401, 'BigInteger', BigInteger, false, true), true)(J$.R(2269409, 's', s, false, false), J$.T(2269417, 16, 22, false)));
                        } catch (J$e) {
                            J$.Ex(2272809, J$e);
                        } finally {
                            if (J$.Fr(2272817))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp128r1() {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(2269769, arguments.callee, this, arguments);
                            arguments = J$.N(2269777, 'arguments', arguments, true, false, false);
                            J$.N(2269785, 'p', p, false, false, false);
                            J$.N(2269793, 'a', a, false, false, false);
                            J$.N(2269801, 'b', b, false, false, false);
                            J$.N(2269809, 'n', n, false, false, false);
                            J$.N(2269817, 'h', h, false, false, false);
                            J$.N(2269825, 'curve', curve, false, false, false);
                            J$.N(2269833, 'G', G, false, false, false);
                            var p = J$.W(2269489, 'p', J$.F(2269481, J$.R(2269465, 'fromHex', fromHex, false, true), false)(J$.T(2269473, 'FFFFFFFDFFFFFFFFFFFFFFFFFFFFFFFF', 21, false)), p, false, false);
                            var a = J$.W(2269521, 'a', J$.F(2269513, J$.R(2269497, 'fromHex', fromHex, false, true), false)(J$.T(2269505, 'FFFFFFFDFFFFFFFFFFFFFFFFFFFFFFFC', 21, false)), a, false, false);
                            var b = J$.W(2269553, 'b', J$.F(2269545, J$.R(2269529, 'fromHex', fromHex, false, true), false)(J$.T(2269537, 'E87579C11079F43DD824993C2CEE5ED3', 21, false)), b, false, false);
                            var n = J$.W(2269585, 'n', J$.F(2269577, J$.R(2269561, 'fromHex', fromHex, false, true), false)(J$.T(2269569, 'FFFFFFFE0000000075A30D1B9038A115', 21, false)), n, false, false);
                            var h = J$.W(2269609, 'h', J$.G(2269601, J$.R(2269593, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2269657, 'curve', J$.F(2269649, J$.R(2269617, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2269625, 'p', p, false, false), J$.R(2269633, 'a', a, false, false), J$.R(2269641, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2269705, 'G', J$.M(2269697, J$.R(2269665, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(270506, '+', J$.B(270498, '+', J$.T(2269673, '04', 21, false), J$.T(2269681, '161FF7528B899B2D0C28607CA52C5B86', 21, false)), J$.T(2269689, 'CF5AC8395BAFEB13C02DA292DDED7A83', 21, false))), G, false, false);
                            return J$.Rt(2269761, J$.F(2269753, J$.R(2269713, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2269721, 'curve', curve, false, false), J$.R(2269729, 'G', G, false, false), J$.R(2269737, 'n', n, false, false), J$.R(2269745, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2272825, J$e);
                        } finally {
                            if (J$.Fr(2272833))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp160k1() {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(2270137, arguments.callee, this, arguments);
                            arguments = J$.N(2270145, 'arguments', arguments, true, false, false);
                            J$.N(2270153, 'p', p, false, false, false);
                            J$.N(2270161, 'a', a, false, false, false);
                            J$.N(2270169, 'b', b, false, false, false);
                            J$.N(2270177, 'n', n, false, false, false);
                            J$.N(2270185, 'h', h, false, false, false);
                            J$.N(2270193, 'curve', curve, false, false, false);
                            J$.N(2270201, 'G', G, false, false, false);
                            var p = J$.W(2269865, 'p', J$.F(2269857, J$.R(2269841, 'fromHex', fromHex, false, true), false)(J$.T(2269849, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFAC73', 21, false)), p, false, false);
                            var a = J$.W(2269889, 'a', J$.G(2269881, J$.R(2269873, 'BigInteger', BigInteger, false, true), 'ZERO'), a, false, false);
                            var b = J$.W(2269921, 'b', J$.F(2269913, J$.R(2269897, 'fromHex', fromHex, false, true), false)(J$.T(2269905, '7', 21, false)), b, false, false);
                            var n = J$.W(2269953, 'n', J$.F(2269945, J$.R(2269929, 'fromHex', fromHex, false, true), false)(J$.T(2269937, '0100000000000000000001B8FA16DFAB9ACA16B6B3', 21, false)), n, false, false);
                            var h = J$.W(2269977, 'h', J$.G(2269969, J$.R(2269961, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2270025, 'curve', J$.F(2270017, J$.R(2269985, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2269993, 'p', p, false, false), J$.R(2270001, 'a', a, false, false), J$.R(2270009, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2270073, 'G', J$.M(2270065, J$.R(2270033, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(270522, '+', J$.B(270514, '+', J$.T(2270041, '04', 21, false), J$.T(2270049, '3B4C382CE37AA192A4019E763036F4F5DD4D7EBB', 21, false)), J$.T(2270057, '938CF935318FDCED6BC28286531733C3F03C4FEE', 21, false))), G, false, false);
                            return J$.Rt(2270129, J$.F(2270121, J$.R(2270081, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2270089, 'curve', curve, false, false), J$.R(2270097, 'G', G, false, false), J$.R(2270105, 'n', n, false, false), J$.R(2270113, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2272841, J$e);
                        } finally {
                            if (J$.Fr(2272849))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp160r1() {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(2270513, arguments.callee, this, arguments);
                            arguments = J$.N(2270521, 'arguments', arguments, true, false, false);
                            J$.N(2270529, 'p', p, false, false, false);
                            J$.N(2270537, 'a', a, false, false, false);
                            J$.N(2270545, 'b', b, false, false, false);
                            J$.N(2270553, 'n', n, false, false, false);
                            J$.N(2270561, 'h', h, false, false, false);
                            J$.N(2270569, 'curve', curve, false, false, false);
                            J$.N(2270577, 'G', G, false, false, false);
                            var p = J$.W(2270233, 'p', J$.F(2270225, J$.R(2270209, 'fromHex', fromHex, false, true), false)(J$.T(2270217, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF7FFFFFFF', 21, false)), p, false, false);
                            var a = J$.W(2270265, 'a', J$.F(2270257, J$.R(2270241, 'fromHex', fromHex, false, true), false)(J$.T(2270249, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF7FFFFFFC', 21, false)), a, false, false);
                            var b = J$.W(2270297, 'b', J$.F(2270289, J$.R(2270273, 'fromHex', fromHex, false, true), false)(J$.T(2270281, '1C97BEFC54BD7A8B65ACF89F81D4D4ADC565FA45', 21, false)), b, false, false);
                            var n = J$.W(2270329, 'n', J$.F(2270321, J$.R(2270305, 'fromHex', fromHex, false, true), false)(J$.T(2270313, '0100000000000000000001F4C8F927AED3CA752257', 21, false)), n, false, false);
                            var h = J$.W(2270353, 'h', J$.G(2270345, J$.R(2270337, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2270401, 'curve', J$.F(2270393, J$.R(2270361, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2270369, 'p', p, false, false), J$.R(2270377, 'a', a, false, false), J$.R(2270385, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2270449, 'G', J$.M(2270441, J$.R(2270409, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(270538, '+', J$.B(270530, '+', J$.T(2270417, '04', 21, false), J$.T(2270425, '4A96B5688EF573284664698968C38BB913CBFC82', 21, false)), J$.T(2270433, '23A628553168947D59DCC912042351377AC5FB32', 21, false))), G, false, false);
                            return J$.Rt(2270505, J$.F(2270497, J$.R(2270457, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2270465, 'curve', curve, false, false), J$.R(2270473, 'G', G, false, false), J$.R(2270481, 'n', n, false, false), J$.R(2270489, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2272857, J$e);
                        } finally {
                            if (J$.Fr(2272865))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp192k1() {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(2270881, arguments.callee, this, arguments);
                            arguments = J$.N(2270889, 'arguments', arguments, true, false, false);
                            J$.N(2270897, 'p', p, false, false, false);
                            J$.N(2270905, 'a', a, false, false, false);
                            J$.N(2270913, 'b', b, false, false, false);
                            J$.N(2270921, 'n', n, false, false, false);
                            J$.N(2270929, 'h', h, false, false, false);
                            J$.N(2270937, 'curve', curve, false, false, false);
                            J$.N(2270945, 'G', G, false, false, false);
                            var p = J$.W(2270609, 'p', J$.F(2270601, J$.R(2270585, 'fromHex', fromHex, false, true), false)(J$.T(2270593, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFEE37', 21, false)), p, false, false);
                            var a = J$.W(2270633, 'a', J$.G(2270625, J$.R(2270617, 'BigInteger', BigInteger, false, true), 'ZERO'), a, false, false);
                            var b = J$.W(2270665, 'b', J$.F(2270657, J$.R(2270641, 'fromHex', fromHex, false, true), false)(J$.T(2270649, '3', 21, false)), b, false, false);
                            var n = J$.W(2270697, 'n', J$.F(2270689, J$.R(2270673, 'fromHex', fromHex, false, true), false)(J$.T(2270681, 'FFFFFFFFFFFFFFFFFFFFFFFE26F2FC170F69466A74DEFD8D', 21, false)), n, false, false);
                            var h = J$.W(2270721, 'h', J$.G(2270713, J$.R(2270705, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2270769, 'curve', J$.F(2270761, J$.R(2270729, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2270737, 'p', p, false, false), J$.R(2270745, 'a', a, false, false), J$.R(2270753, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2270817, 'G', J$.M(2270809, J$.R(2270777, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(270554, '+', J$.B(270546, '+', J$.T(2270785, '04', 21, false), J$.T(2270793, 'DB4FF10EC057E9AE26B07D0280B7F4341DA5D1B1EAE06C7D', 21, false)), J$.T(2270801, '9B2F2F6D9C5628A7844163D015BE86344082AA88D95E2F9D', 21, false))), G, false, false);
                            return J$.Rt(2270873, J$.F(2270865, J$.R(2270825, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2270833, 'curve', curve, false, false), J$.R(2270841, 'G', G, false, false), J$.R(2270849, 'n', n, false, false), J$.R(2270857, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2272873, J$e);
                        } finally {
                            if (J$.Fr(2272881))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp192r1() {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(2271257, arguments.callee, this, arguments);
                            arguments = J$.N(2271265, 'arguments', arguments, true, false, false);
                            J$.N(2271273, 'p', p, false, false, false);
                            J$.N(2271281, 'a', a, false, false, false);
                            J$.N(2271289, 'b', b, false, false, false);
                            J$.N(2271297, 'n', n, false, false, false);
                            J$.N(2271305, 'h', h, false, false, false);
                            J$.N(2271313, 'curve', curve, false, false, false);
                            J$.N(2271321, 'G', G, false, false, false);
                            var p = J$.W(2270977, 'p', J$.F(2270969, J$.R(2270953, 'fromHex', fromHex, false, true), false)(J$.T(2270961, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFFFFFFFFFFFF', 21, false)), p, false, false);
                            var a = J$.W(2271009, 'a', J$.F(2271001, J$.R(2270985, 'fromHex', fromHex, false, true), false)(J$.T(2270993, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFFFFFFFFFFFC', 21, false)), a, false, false);
                            var b = J$.W(2271041, 'b', J$.F(2271033, J$.R(2271017, 'fromHex', fromHex, false, true), false)(J$.T(2271025, '64210519E59C80E70FA7E9AB72243049FEB8DEECC146B9B1', 21, false)), b, false, false);
                            var n = J$.W(2271073, 'n', J$.F(2271065, J$.R(2271049, 'fromHex', fromHex, false, true), false)(J$.T(2271057, 'FFFFFFFFFFFFFFFFFFFFFFFF99DEF836146BC9B1B4D22831', 21, false)), n, false, false);
                            var h = J$.W(2271097, 'h', J$.G(2271089, J$.R(2271081, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2271145, 'curve', J$.F(2271137, J$.R(2271105, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2271113, 'p', p, false, false), J$.R(2271121, 'a', a, false, false), J$.R(2271129, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2271193, 'G', J$.M(2271185, J$.R(2271153, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(270570, '+', J$.B(270562, '+', J$.T(2271161, '04', 21, false), J$.T(2271169, '188DA80EB03090F67CBF20EB43A18800F4FF0AFD82FF1012', 21, false)), J$.T(2271177, '07192B95FFC8DA78631011ED6B24CDD573F977A11E794811', 21, false))), G, false, false);
                            return J$.Rt(2271249, J$.F(2271241, J$.R(2271201, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2271209, 'curve', curve, false, false), J$.R(2271217, 'G', G, false, false), J$.R(2271225, 'n', n, false, false), J$.R(2271233, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2272889, J$e);
                        } finally {
                            if (J$.Fr(2272897))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp224r1() {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(2271633, arguments.callee, this, arguments);
                            arguments = J$.N(2271641, 'arguments', arguments, true, false, false);
                            J$.N(2271649, 'p', p, false, false, false);
                            J$.N(2271657, 'a', a, false, false, false);
                            J$.N(2271665, 'b', b, false, false, false);
                            J$.N(2271673, 'n', n, false, false, false);
                            J$.N(2271681, 'h', h, false, false, false);
                            J$.N(2271689, 'curve', curve, false, false, false);
                            J$.N(2271697, 'G', G, false, false, false);
                            var p = J$.W(2271353, 'p', J$.F(2271345, J$.R(2271329, 'fromHex', fromHex, false, true), false)(J$.T(2271337, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF000000000000000000000001', 21, false)), p, false, false);
                            var a = J$.W(2271385, 'a', J$.F(2271377, J$.R(2271361, 'fromHex', fromHex, false, true), false)(J$.T(2271369, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFFFFFFFFFFFFFFFFFFFE', 21, false)), a, false, false);
                            var b = J$.W(2271417, 'b', J$.F(2271409, J$.R(2271393, 'fromHex', fromHex, false, true), false)(J$.T(2271401, 'B4050A850C04B3ABF54132565044B0B7D7BFD8BA270B39432355FFB4', 21, false)), b, false, false);
                            var n = J$.W(2271449, 'n', J$.F(2271441, J$.R(2271425, 'fromHex', fromHex, false, true), false)(J$.T(2271433, 'FFFFFFFFFFFFFFFFFFFFFFFFFFFF16A2E0B8F03E13DD29455C5C2A3D', 21, false)), n, false, false);
                            var h = J$.W(2271473, 'h', J$.G(2271465, J$.R(2271457, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2271521, 'curve', J$.F(2271513, J$.R(2271481, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2271489, 'p', p, false, false), J$.R(2271497, 'a', a, false, false), J$.R(2271505, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2271569, 'G', J$.M(2271561, J$.R(2271529, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(270586, '+', J$.B(270578, '+', J$.T(2271537, '04', 21, false), J$.T(2271545, 'B70E0CBD6BB4BF7F321390B94A03C1D356C21122343280D6115C1D21', 21, false)), J$.T(2271553, 'BD376388B5F723FB4C22DFE6CD4375A05A07476444D5819985007E34', 21, false))), G, false, false);
                            return J$.Rt(2271625, J$.F(2271617, J$.R(2271577, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2271585, 'curve', curve, false, false), J$.R(2271593, 'G', G, false, false), J$.R(2271601, 'n', n, false, false), J$.R(2271609, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2272905, J$e);
                        } finally {
                            if (J$.Fr(2272913))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function secp256r1() {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(2272009, arguments.callee, this, arguments);
                            arguments = J$.N(2272017, 'arguments', arguments, true, false, false);
                            J$.N(2272025, 'p', p, false, false, false);
                            J$.N(2272033, 'a', a, false, false, false);
                            J$.N(2272041, 'b', b, false, false, false);
                            J$.N(2272049, 'n', n, false, false, false);
                            J$.N(2272057, 'h', h, false, false, false);
                            J$.N(2272065, 'curve', curve, false, false, false);
                            J$.N(2272073, 'G', G, false, false, false);
                            var p = J$.W(2271729, 'p', J$.F(2271721, J$.R(2271705, 'fromHex', fromHex, false, true), false)(J$.T(2271713, 'FFFFFFFF00000001000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFF', 21, false)), p, false, false);
                            var a = J$.W(2271761, 'a', J$.F(2271753, J$.R(2271737, 'fromHex', fromHex, false, true), false)(J$.T(2271745, 'FFFFFFFF00000001000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFC', 21, false)), a, false, false);
                            var b = J$.W(2271793, 'b', J$.F(2271785, J$.R(2271769, 'fromHex', fromHex, false, true), false)(J$.T(2271777, '5AC635D8AA3A93E7B3EBBD55769886BC651D06B0CC53B0F63BCE3C3E27D2604B', 21, false)), b, false, false);
                            var n = J$.W(2271825, 'n', J$.F(2271817, J$.R(2271801, 'fromHex', fromHex, false, true), false)(J$.T(2271809, 'FFFFFFFF00000000FFFFFFFFFFFFFFFFBCE6FAADA7179E84F3B9CAC2FC632551', 21, false)), n, false, false);
                            var h = J$.W(2271849, 'h', J$.G(2271841, J$.R(2271833, 'BigInteger', BigInteger, false, true), 'ONE'), h, false, false);
                            var curve = J$.W(2271897, 'curve', J$.F(2271889, J$.R(2271857, 'ECCurveFp', ECCurveFp, false, true), true)(J$.R(2271865, 'p', p, false, false), J$.R(2271873, 'a', a, false, false), J$.R(2271881, 'b', b, false, false)), curve, false, false);
                            var G = J$.W(2271945, 'G', J$.M(2271937, J$.R(2271905, 'curve', curve, false, false), 'decodePointHex', false)(J$.B(270602, '+', J$.B(270594, '+', J$.T(2271913, '04', 21, false), J$.T(2271921, '6B17D1F2E12C4247F8BCE6E563A440F277037D812DEB33A0F4A13945D898C296', 21, false)), J$.T(2271929, '4FE342E2FE1A7F9B8EE7EB4A7C0F9E162BCE33576B315ECECBB6406837BF51F5', 21, false))), G, false, false);
                            return J$.Rt(2272001, J$.F(2271993, J$.R(2271953, 'X9ECParameters', X9ECParameters, false, true), true)(J$.R(2271961, 'curve', curve, false, false), J$.R(2271969, 'G', G, false, false), J$.R(2271977, 'n', n, false, false), J$.R(2271985, 'h', h, false, false)));
                        } catch (J$e) {
                            J$.Ex(2272921, J$e);
                        } finally {
                            if (J$.Fr(2272929))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function getSECCurveByName(name) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(2272377, arguments.callee, this, arguments);
                            arguments = J$.N(2272385, 'arguments', arguments, true, false, false);
                            name = J$.N(2272393, 'name', name, true, false, false);
                            if (J$.C(140256, J$.B(270610, '==', J$.R(2272081, 'name', name, false, false), J$.T(2272089, 'secp128r1', 21, false))))
                                return J$.Rt(2272113, J$.F(2272105, J$.R(2272097, 'secp128r1', secp128r1, false, true), false)());
                            if (J$.C(140264, J$.B(270618, '==', J$.R(2272121, 'name', name, false, false), J$.T(2272129, 'secp160k1', 21, false))))
                                return J$.Rt(2272153, J$.F(2272145, J$.R(2272137, 'secp160k1', secp160k1, false, true), false)());
                            if (J$.C(140272, J$.B(270626, '==', J$.R(2272161, 'name', name, false, false), J$.T(2272169, 'secp160r1', 21, false))))
                                return J$.Rt(2272193, J$.F(2272185, J$.R(2272177, 'secp160r1', secp160r1, false, true), false)());
                            if (J$.C(140280, J$.B(270634, '==', J$.R(2272201, 'name', name, false, false), J$.T(2272209, 'secp192k1', 21, false))))
                                return J$.Rt(2272233, J$.F(2272225, J$.R(2272217, 'secp192k1', secp192k1, false, true), false)());
                            if (J$.C(140288, J$.B(270642, '==', J$.R(2272241, 'name', name, false, false), J$.T(2272249, 'secp192r1', 21, false))))
                                return J$.Rt(2272273, J$.F(2272265, J$.R(2272257, 'secp192r1', secp192r1, false, true), false)());
                            if (J$.C(140296, J$.B(270650, '==', J$.R(2272281, 'name', name, false, false), J$.T(2272289, 'secp224r1', 21, false))))
                                return J$.Rt(2272313, J$.F(2272305, J$.R(2272297, 'secp224r1', secp224r1, false, true), false)());
                            if (J$.C(140304, J$.B(270658, '==', J$.R(2272321, 'name', name, false, false), J$.T(2272329, 'secp256r1', 21, false))))
                                return J$.Rt(2272353, J$.F(2272345, J$.R(2272337, 'secp256r1', secp256r1, false, true), false)());
                            return J$.Rt(2272369, J$.T(2272361, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(2272937, J$e);
                        } finally {
                            if (J$.Fr(2272945))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(2272489, 'BigInteger', BigInteger, false, false, false);
            J$.N(2272497, 'ECCurveFp', ECCurveFp, false, false, false);
            X9ECParameters = J$.N(2272513, 'X9ECParameters', J$.T(2272505, X9ECParameters, 12, false), true, false, false);
            x9getCurve = J$.N(2272529, 'x9getCurve', J$.T(2272521, x9getCurve, 12, false), true, false, false);
            x9getG = J$.N(2272545, 'x9getG', J$.T(2272537, x9getG, 12, false), true, false, false);
            x9getN = J$.N(2272561, 'x9getN', J$.T(2272553, x9getN, 12, false), true, false, false);
            x9getH = J$.N(2272577, 'x9getH', J$.T(2272569, x9getH, 12, false), true, false, false);
            fromHex = J$.N(2272593, 'fromHex', J$.T(2272585, fromHex, 12, false), true, false, false);
            secp128r1 = J$.N(2272609, 'secp128r1', J$.T(2272601, secp128r1, 12, false), true, false, false);
            secp160k1 = J$.N(2272625, 'secp160k1', J$.T(2272617, secp160k1, 12, false), true, false, false);
            secp160r1 = J$.N(2272641, 'secp160r1', J$.T(2272633, secp160r1, 12, false), true, false, false);
            secp192k1 = J$.N(2272657, 'secp192k1', J$.T(2272649, secp192k1, 12, false), true, false, false);
            secp192r1 = J$.N(2272673, 'secp192r1', J$.T(2272665, secp192r1, 12, false), true, false, false);
            secp224r1 = J$.N(2272689, 'secp224r1', J$.T(2272681, secp224r1, 12, false), true, false, false);
            secp256r1 = J$.N(2272705, 'secp256r1', J$.T(2272697, secp256r1, 12, false), true, false, false);
            getSECCurveByName = J$.N(2272721, 'getSECCurveByName', J$.T(2272713, getSECCurveByName, 12, false), true, false, false);
            J$.T(2268881, 'use strict', 21, false);
            var BigInteger = J$.W(2268921, 'BigInteger', J$.G(2268913, J$.F(2268905, J$.I(typeof require === 'undefined' ? require = J$.R(2268889, 'require', undefined, true, true) : require = J$.R(2268889, 'require', require, true, true)), false)(J$.T(2268897, 'jsbn', 21, false)), 'BigInteger'), BigInteger, false, true);
            var ECCurveFp = J$.W(2268961, 'ECCurveFp', J$.G(2268953, J$.F(2268945, J$.I(typeof require === 'undefined' ? require = J$.R(2268929, 'require', undefined, true, true) : require = J$.R(2268929, 'require', require, true, true)), false)(J$.T(2268937, './ec.js', 21, false)), 'ECCurveFp'), ECCurveFp, false, true);
            J$.P(2269297, J$.G(2269281, J$.R(2269273, 'X9ECParameters', X9ECParameters, false, true), 'prototype'), 'getCurve', J$.R(2269289, 'x9getCurve', x9getCurve, false, true));
            J$.P(2269329, J$.G(2269313, J$.R(2269305, 'X9ECParameters', X9ECParameters, false, true), 'prototype'), 'getG', J$.R(2269321, 'x9getG', x9getG, false, true));
            J$.P(2269361, J$.G(2269345, J$.R(2269337, 'X9ECParameters', X9ECParameters, false, true), 'prototype'), 'getN', J$.R(2269353, 'x9getN', x9getN, false, true));
            J$.P(2269393, J$.G(2269377, J$.R(2269369, 'X9ECParameters', X9ECParameters, false, true), 'prototype'), 'getH', J$.R(2269385, 'x9getH', x9getH, false, true));
            J$.P(2272473, J$.I(typeof module === 'undefined' ? module = J$.R(2272401, 'module', undefined, true, true) : module = J$.R(2272401, 'module', module, true, true)), 'exports', J$.T(2272465, {
                'secp128r1': J$.R(2272409, 'secp128r1', secp128r1, false, true),
                'secp160k1': J$.R(2272417, 'secp160k1', secp160k1, false, true),
                'secp160r1': J$.R(2272425, 'secp160r1', secp160r1, false, true),
                'secp192k1': J$.R(2272433, 'secp192k1', secp192k1, false, true),
                'secp192r1': J$.R(2272441, 'secp192r1', secp192r1, false, true),
                'secp224r1': J$.R(2272449, 'secp224r1', secp224r1, false, true),
                'secp256r1': J$.R(2272457, 'secp256r1', secp256r1, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(2272953, J$e);
        } finally {
            if (J$.Sr(2272961))
                continue jalangiLabel14;
            else
                break jalangiLabel14;
        }
    }
// JALANGI DO NOT INSTRUMENT
