// mongo-parse/matches.js
J$.noInstrEval = false;
jalangiLabel22:
    while (true) {
        try {
            J$.Se(19185, '/Users/ex1t/Desktop/nodeAnalysis/Goldie/tests/target/cache/node_modules/mongo-parse/matches.js');
            function partMatches(part, document) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(17673, arguments.callee, this, arguments);
                            arguments = J$.N(17681, 'arguments', arguments, true, false, false);
                            part = J$.N(17689, 'part', part, true, false, false);
                            document = J$.N(17697, 'document', document, true, false, false);
                            J$.N(17705, 'pointers', pointers, false, false, false);
                            J$.N(17713, 'p', p, false, false, false);
                            J$.N(17721, 'pointer', pointer, false, false, false);
                            J$.N(17729, 'test', test, false, false, false);
                            J$.N(17737, 'documentField', documentField, false, false, false);
                            J$.N(17745, 'n', n, false, false, false);
                            J$.N(17753, 'anyMatched', anyMatched, false, false, false);
                            J$.N(17761, 'documentToMatch', documentToMatch, false, false, false);
                            J$.N(17769, 'objectContext', objectContext, false, false, false);
                            var pointers = J$.W(16393, 'pointers', J$.F(16385, J$.R(16353, 'DotNotationPointers', DotNotationPointers, false, true), false)(J$.R(16361, 'document', document, false, false), J$.G(16377, J$.R(16369, 'part', part, false, false), 'field')), pointers, false, false);
                            for (var p = J$.W(16409, 'p', J$.T(16401, 0, 22, false), p, false, false); J$.C(528, J$.B(690, '<', J$.R(16417, 'p', p, false, false), J$.G(16433, J$.R(16425, 'pointers', pointers, false, false), 'length'))); J$.B(714, '-', p = J$.W(16449, 'p', J$.B(706, '+', J$.U(698, '+', J$.R(16441, 'p', p, false, false)), 1), p, false, false), 1)) {
                                var pointer = J$.W(16481, 'pointer', J$.G(16473, J$.R(16457, 'pointers', pointers, false, false), J$.R(16465, 'p', p, false, false)), pointer, false, false);
                                if (J$.C(520, J$.B(722, '===', J$.G(16497, J$.R(16489, 'part', part, false, false), 'operator'), J$.T(16505, undefined, 24, false)))) {
                                    if (J$.C(336, J$.U(730, '!', J$.F(16561, J$.R(16513, 'valueTest', valueTest, false, true), false)(J$.G(16529, J$.R(16521, 'pointer', pointer, false, false), 'val'), J$.G(16545, J$.R(16537, 'part', part, false, false), 'operand'), J$.R(16553, 'mongoEqual', mongoEqual, false, true))))) {
                                        continue;
                                    }
                                } else if (J$.C(512, J$.B(738, 'in', J$.G(16577, J$.R(16569, 'part', part, false, false), 'operator'), J$.R(16585, 'simpleComparators', simpleComparators, false, true)))) {
                                    var test = J$.W(16673, 'test', J$.F(16665, J$.R(16593, 'valueTest', valueTest, false, true), false)(J$.G(16609, J$.R(16601, 'pointer', pointer, false, false), 'val'), J$.G(16625, J$.R(16617, 'part', part, false, false), 'operand'), J$.G(16657, J$.R(16633, 'simpleComparators', simpleComparators, false, true), J$.G(16649, J$.R(16641, 'part', part, false, false), 'operator'))), test, false, false);
                                    if (J$.C(344, J$.U(746, '!', J$.R(16681, 'test', test, false, false))))
                                        continue;
                                } else if (J$.C(504, J$.B(754, 'in', J$.G(16697, J$.R(16689, 'part', part, false, false), 'operator'), J$.R(16705, 'compoundOperatorComparators', compoundOperatorComparators, false, true)))) {
                                    if (J$.C(352, J$.U(762, '!', J$.M(16761, J$.R(16713, 'compoundOperatorComparators', compoundOperatorComparators, false, true), J$.G(16729, J$.R(16721, 'part', part, false, false), 'operator'), false)(J$.R(16737, 'document', document, false, false), J$.G(16753, J$.R(16745, 'part', part, false, false), 'parts'))))) {
                                        continue;
                                    }
                                } else if (J$.C(496, J$.B(770, '===', J$.G(16777, J$.R(16769, 'part', part, false, false), 'operator'), J$.T(16785, '$not', 21, false)))) {
                                    if (J$.C(376, J$.B(778, '>', J$.G(16809, J$.G(16801, J$.R(16793, 'part', part, false, false), 'parts'), 'length'), J$.T(16817, 0, 22, false)))) {
                                        if (J$.C(360, J$.F(16857, J$.R(16825, 'matches', matches, false, true), false)(J$.G(16841, J$.R(16833, 'part', part, false, false), 'parts'), J$.R(16849, 'document', document, false, false)))) {
                                            continue;
                                        }
                                    } else {
                                        if (J$.C(368, J$.B(786, '===', J$.F(16913, J$.R(16865, 'valueTest', valueTest, false, true), false)(J$.G(16881, J$.R(16873, 'pointer', pointer, false, false), 'val'), J$.G(16897, J$.R(16889, 'part', part, false, false), 'operand'), J$.R(16905, 'mongoEqual', mongoEqual, false, true)), J$.T(16921, true, 23, false))))
                                            continue;
                                    }
                                } else if (J$.C(488, J$.B(794, '===', J$.G(16937, J$.R(16929, 'part', part, false, false), 'operator'), J$.T(16945, '$size', 21, false)))) {
                                    return J$.Rt(17017, J$.C(384, J$.B(802, 'instanceof', J$.G(16961, J$.R(16953, 'pointer', pointer, false, false), 'val'), J$.I(typeof Array === 'undefined' ? Array = J$.R(16969, 'Array', undefined, true, true) : Array = J$.R(16969, 'Array', Array, true, true)))) ? J$.B(810, '===', J$.G(16993, J$.G(16985, J$.R(16977, 'pointer', pointer, false, false), 'val'), 'length'), J$.G(17009, J$.R(17001, 'part', part, false, false), 'operand')) : J$._());
                                } else if (J$.C(480, J$.B(818, '===', J$.G(17033, J$.R(17025, 'part', part, false, false), 'operator'), J$.T(17041, '$elemMatch', 21, false)))) {
                                    var documentField = J$.W(17065, 'documentField', J$.G(17057, J$.R(17049, 'pointer', pointer, false, false), 'val'), documentField, false, false);
                                    if (J$.C(392, J$.B(826, '===', J$.R(17073, 'documentField', documentField, false, false), J$.T(17081, undefined, 24, false))))
                                        continue;
                                    if (J$.C(408, J$.G(17097, J$.R(17089, 'part', part, false, false), 'implicitField'))) {
                                        for (var n = J$.W(17113, 'n', J$.T(17105, 0, 22, false), n, false, false); J$.C(400, J$.B(834, '<', J$.R(17121, 'n', n, false, false), J$.G(17145, J$.G(17137, J$.R(17129, 'part', part, false, false), 'parts'), 'length'))); J$.B(858, '-', n = J$.W(17161, 'n', J$.B(850, '+', J$.U(842, '+', J$.R(17153, 'n', n, false, false)), 1), n, false, false), 1)) {
                                            J$.P(17209, J$.G(17193, J$.G(17177, J$.R(17169, 'part', part, false, false), 'parts'), J$.R(17185, 'n', n, false, false)), 'field', J$.T(17201, 'x', 21, false));
                                        }
                                    }
                                    var anyMatched = J$.W(17225, 'anyMatched', J$.T(17217, false, 23, false), anyMatched, false, false);
                                    for (var n = J$.W(17241, 'n', J$.T(17233, 0, 22, false), n, false, false); J$.C(432, J$.B(866, '<', J$.R(17249, 'n', n, false, false), J$.G(17265, J$.R(17257, 'documentField', documentField, false, false), 'length'))); J$.B(890, '-', n = J$.W(17281, 'n', J$.B(882, '+', J$.U(874, '+', J$.R(17273, 'n', n, false, false)), 1), n, false, false), 1)) {
                                        if (J$.C(416, J$.G(17297, J$.R(17289, 'part', part, false, false), 'implicitField'))) {
                                            var documentToMatch = J$.W(17337, 'documentToMatch', J$.T(17329, { x: J$.G(17321, J$.R(17305, 'documentField', documentField, false, false), J$.R(17313, 'n', n, false, false)) }, 11, false), documentToMatch, false, false);
                                        } else {
                                            var documentToMatch = J$.W(17369, 'documentToMatch', J$.G(17361, J$.R(17345, 'documentField', documentField, false, false), J$.R(17353, 'n', n, false, false)), documentToMatch, false, false);
                                        }
                                        if (J$.C(424, J$.F(17409, J$.R(17377, 'matches', matches, false, true), false)(J$.G(17393, J$.R(17385, 'part', part, false, false), 'parts'), J$.R(17401, 'documentToMatch', documentToMatch, false, false)))) {
                                            anyMatched = J$.W(17425, 'anyMatched', J$.T(17417, true, 23, false), anyMatched, false, false);
                                            break;
                                        }
                                    }
                                    if (J$.C(440, J$.U(898, '!', J$.R(17433, 'anyMatched', anyMatched, false, false))))
                                        continue;
                                } else if (J$.C(472, J$.B(906, '===', J$.G(17449, J$.R(17441, 'part', part, false, false), 'operator'), J$.T(17457, '$where', 21, false)))) {
                                    if (J$.C(448, J$.B(914, '!==', J$.G(17473, J$.R(17465, 'part', part, false, false), 'field'), J$.T(17481, undefined, 24, false)))) {
                                        var objectContext = J$.W(17505, 'objectContext', J$.G(17497, J$.R(17489, 'pointer', pointer, false, false), 'val'), objectContext, false, false);
                                    } else {
                                        var objectContext = J$.W(17521, 'objectContext', J$.R(17513, 'document', document, false, false), objectContext, false, false);
                                    }
                                    if (J$.C(456, J$.U(922, '!', J$.M(17553, J$.G(17537, J$.R(17529, 'part', part, false, false), 'operand'), 'call', false)(J$.R(17545, 'objectContext', objectContext, false, false)))))
                                        continue;
                                } else if (J$.C(464, J$.B(930, '===', J$.G(17569, J$.R(17561, 'part', part, false, false), 'operator'), J$.T(17577, '$comment', 21, false)))) {
                                    return J$.Rt(17593, J$.T(17585, true, 23, false));
                                } else {
                                    throw J$.F(17633, J$.I(typeof Error === 'undefined' ? Error = J$.R(17601, 'Error', undefined, true, true) : Error = J$.R(17601, 'Error', Error, true, true)), true)(J$.B(938, '+', J$.T(17609, 'Unsupported operator: ', 21, false), J$.G(17625, J$.I(typeof parts === 'undefined' ? parts = J$.R(17617, 'parts', undefined, true, true) : parts = J$.R(17617, 'parts', parts, true, true)), 'operator')));
                                }
                                return J$.Rt(17649, J$.T(17641, true, 23, false));
                            }
                            return J$.Rt(17665, J$.T(17657, false, 23, false));
                        } catch (J$e) {
                            J$.Ex(19545, J$e);
                        } finally {
                            if (J$.Fr(19553))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function valueTest(documentValue, queryOperand, compare) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(18057, arguments.callee, this, arguments);
                            arguments = J$.N(18065, 'arguments', arguments, true, false, false);
                            documentValue = J$.N(18073, 'documentValue', documentValue, true, false, false);
                            queryOperand = J$.N(18081, 'queryOperand', queryOperand, true, false, false);
                            compare = J$.N(18089, 'compare', compare, true, false, false);
                            J$.N(18097, 'test', test, false, false, false);
                            if (J$.C(568, J$.B(946, 'instanceof', J$.R(17777, 'documentValue', documentValue, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(17785, 'Array', undefined, true, true) : Array = J$.R(17785, 'Array', Array, true, true))))) {
                                if (J$.C(560, J$.B(954, 'instanceof', J$.R(17793, 'queryOperand', queryOperand, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(17801, 'Array', undefined, true, true) : Array = J$.R(17801, 'Array', Array, true, true))))) {
                                    if (J$.C(536, J$.U(962, '!', J$.F(17833, J$.R(17809, 'compare', compare, false, false), false)(J$.R(17817, 'documentValue', documentValue, false, false), J$.R(17825, 'queryOperand', queryOperand, false, false)))))
                                        return J$.Rt(17849, J$.T(17841, false, 23, false));
                                } else {
                                    var test = J$.W(17969, 'test', J$.M(17961, J$.R(17857, 'documentValue', documentValue, false, false), 'reduce', false)(J$.T(17945, function (last, cur) {
                                        jalangiLabel17:
                                            while (true) {
                                                try {
                                                    J$.Fe(17913, arguments.callee, this, arguments);
                                                    arguments = J$.N(17921, 'arguments', arguments, true, false, false);
                                                    last = J$.N(17929, 'last', last, true, false, false);
                                                    cur = J$.N(17937, 'cur', cur, true, false, false);
                                                    return J$.Rt(17905, J$.C(544, J$.R(17865, 'last', last, false, false)) ? J$._() : J$.F(17897, J$.R(17873, 'compare', compare, false, false), false)(J$.R(17881, 'cur', cur, false, false), J$.R(17889, 'queryOperand', queryOperand, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(19561, J$e);
                                                } finally {
                                                    if (J$.Fr(19569))
                                                        continue jalangiLabel17;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false), J$.T(17953, false, 23, false)), test, false, false);
                                    if (J$.C(552, J$.U(970, '!', J$.R(17977, 'test', test, false, false))))
                                        return J$.Rt(17993, J$.T(17985, false, 23, false));
                                }
                            } else {
                                return J$.Rt(18033, J$.F(18025, J$.R(18001, 'compare', compare, false, false), false)(J$.R(18009, 'documentValue', documentValue, false, false), J$.R(18017, 'queryOperand', queryOperand, false, false)));
                            }
                            return J$.Rt(18049, J$.T(18041, true, 23, false));
                        } catch (J$e) {
                            J$.Ex(19577, J$e);
                        } finally {
                            if (J$.Fr(19585))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function mongoEqual(documentValue, queryOperand) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(18865, arguments.callee, this, arguments);
                            arguments = J$.N(18873, 'arguments', arguments, true, false, false);
                            documentValue = J$.N(18881, 'documentValue', documentValue, true, false, false);
                            queryOperand = J$.N(18889, 'queryOperand', queryOperand, true, false, false);
                            J$.N(18897, 'aKeys', aKeys, false, false, false);
                            J$.N(18905, 'bKeys', bKeys, false, false, false);
                            J$.N(18913, 'n', n, false, false, false);
                            J$.N(18921, 'key', key, false, false, false);
                            J$.N(18929, 'aVal', aVal, false, false, false);
                            J$.N(18937, 'bVal', bVal, false, false, false);
                            if (J$.C(664, J$.B(978, 'instanceof', J$.R(18105, 'documentValue', documentValue, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(18113, 'Array', undefined, true, true) : Array = J$.R(18113, 'Array', Array, true, true))))) {
                                if (J$.C(576, J$.U(994, '!', J$.B(986, 'instanceof', J$.R(18121, 'queryOperand', queryOperand, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(18129, 'Array', undefined, true, true) : Array = J$.R(18129, 'Array', Array, true, true))))))
                                    return J$.Rt(18145, J$.T(18137, false, 23, false));
                                if (J$.C(592, J$.B(1002, '!==', J$.G(18161, J$.R(18153, 'documentValue', documentValue, false, false), 'length'), J$.G(18177, J$.R(18169, 'queryOperand', queryOperand, false, false), 'length')))) {
                                    return J$.Rt(18193, J$.T(18185, false, 23, false));
                                } else {
                                    return J$.Rt(18337, J$.M(18329, J$.R(18201, 'documentValue', documentValue, false, false), 'reduce', false)(J$.T(18313, function (previousValue, currentValue, index) {
                                        jalangiLabel19:
                                            while (true) {
                                                try {
                                                    J$.Fe(18273, arguments.callee, this, arguments);
                                                    arguments = J$.N(18281, 'arguments', arguments, true, false, false);
                                                    previousValue = J$.N(18289, 'previousValue', previousValue, true, false, false);
                                                    currentValue = J$.N(18297, 'currentValue', currentValue, true, false, false);
                                                    index = J$.N(18305, 'index', index, true, false, false);
                                                    return J$.Rt(18265, J$.C(584, J$.R(18209, 'previousValue', previousValue, false, false)) ? J$.F(18257, J$.R(18217, 'mongoEqual', mongoEqual, false, true), false)(J$.R(18225, 'currentValue', currentValue, false, false), J$.G(18249, J$.R(18233, 'queryOperand', queryOperand, false, false), J$.R(18241, 'index', index, false, false))) : J$._());
                                                } catch (J$e) {
                                                    J$.Ex(19593, J$e);
                                                } finally {
                                                    if (J$.Fr(19601))
                                                        continue jalangiLabel19;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false), J$.T(18321, true, 23, false)));
                                }
                            } else if (J$.C(656, J$.B(1010, 'instanceof', J$.R(18345, 'documentValue', documentValue, false, false), J$.I(typeof Object === 'undefined' ? Object = J$.R(18353, 'Object', undefined, true, true) : Object = J$.R(18353, 'Object', Object, true, true))))) {
                                if (J$.C(600, J$.U(1026, '!', J$.B(1018, 'instanceof', J$.R(18361, 'queryOperand', queryOperand, false, false), J$.I(typeof Object === 'undefined' ? Object = J$.R(18369, 'Object', undefined, true, true) : Object = J$.R(18369, 'Object', Object, true, true))))))
                                    return J$.Rt(18385, J$.T(18377, false, 23, false));
                                var aKeys = J$.W(18417, 'aKeys', J$.M(18409, J$.I(typeof Object === 'undefined' ? Object = J$.R(18393, 'Object', undefined, true, true) : Object = J$.R(18393, 'Object', Object, true, true)), 'keys', false)(J$.R(18401, 'documentValue', documentValue, false, false)), aKeys, false, false);
                                var bKeys = J$.W(18449, 'bKeys', J$.M(18441, J$.I(typeof Object === 'undefined' ? Object = J$.R(18425, 'Object', undefined, true, true) : Object = J$.R(18425, 'Object', Object, true, true)), 'keys', false)(J$.R(18433, 'queryOperand', queryOperand, false, false)), bKeys, false, false);
                                if (J$.C(632, J$.B(1034, '!==', J$.G(18465, J$.R(18457, 'aKeys', aKeys, false, false), 'length'), J$.G(18481, J$.R(18473, 'bKeys', bKeys, false, false), 'length')))) {
                                    return J$.Rt(18497, J$.T(18489, false, 23, false));
                                } else {
                                    for (var n = J$.W(18513, 'n', J$.T(18505, 0, 22, false), n, false, false); J$.C(624, J$.B(1042, '<', J$.R(18521, 'n', n, false, false), J$.G(18537, J$.R(18529, 'aKeys', aKeys, false, false), 'length'))); J$.B(1066, '-', n = J$.W(18553, 'n', J$.B(1058, '+', J$.U(1050, '+', J$.R(18545, 'n', n, false, false)), 1), n, false, false), 1)) {
                                        if (J$.C(608, J$.B(1074, '!==', J$.G(18577, J$.R(18561, 'aKeys', aKeys, false, false), J$.R(18569, 'n', n, false, false)), J$.G(18601, J$.R(18585, 'bKeys', bKeys, false, false), J$.R(18593, 'n', n, false, false)))))
                                            return J$.Rt(18617, J$.T(18609, false, 23, false));
                                        var key = J$.W(18649, 'key', J$.G(18641, J$.R(18625, 'aKeys', aKeys, false, false), J$.R(18633, 'n', n, false, false)), key, false, false);
                                        var aVal = J$.W(18681, 'aVal', J$.G(18673, J$.R(18657, 'documentValue', documentValue, false, false), J$.R(18665, 'key', key, false, false)), aVal, false, false);
                                        var bVal = J$.W(18713, 'bVal', J$.G(18705, J$.R(18689, 'queryOperand', queryOperand, false, false), J$.R(18697, 'key', key, false, false)), bVal, false, false);
                                        if (J$.C(616, J$.U(1082, '!', J$.F(18745, J$.R(18721, 'mongoEqual', mongoEqual, false, true), false)(J$.R(18729, 'aVal', aVal, false, false), J$.R(18737, 'bVal', bVal, false, false))))) {
                                            return J$.Rt(18761, J$.T(18753, false, 23, false));
                                        }
                                    }
                                    return J$.Rt(18777, J$.T(18769, true, 23, false));
                                }
                            } else {
                                if (J$.C(648, J$.B(1090, '===', J$.R(18785, 'queryOperand', queryOperand, false, false), J$.T(18793, null, 25, false)))) {
                                    return J$.Rt(18833, J$.C(640, J$.B(1098, '===', J$.R(18801, 'documentValue', documentValue, false, false), J$.T(18809, undefined, 24, false))) ? J$._() : J$.B(1106, '===', J$.R(18817, 'documentValue', documentValue, false, false), J$.T(18825, null, 25, false)));
                                } else {
                                    return J$.Rt(18857, J$.B(1114, '===', J$.R(18841, 'documentValue', documentValue, false, false), J$.R(18849, 'queryOperand', queryOperand, false, false)));
                                }
                            }
                        } catch (J$e) {
                            J$.Ex(19609, J$e);
                        } finally {
                            if (J$.Fr(19617))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function validateDocumentObject(document) {
                jalangiLabel21:
                    while (true) {
                        try {
                            J$.Fe(19153, arguments.callee, this, arguments);
                            arguments = J$.N(19161, 'arguments', arguments, true, false, false);
                            document = J$.N(19169, 'document', document, true, false, false);
                            J$.N(19177, 'key', key, false, false, false);
                            for (var key in J$.H(19137, J$.R(18945, 'document', document, false, false))) {
                                J$.N(19145, 'key', key, false, true, false);
                                {
                                    {
                                        if (J$.C(688, J$.B(1122, '===', J$.G(18969, J$.R(18953, 'key', key, false, false), J$.T(18961, 0, 22, false)), J$.T(18977, '$', 21, false))))
                                            throw J$.F(19001, J$.I(typeof Error === 'undefined' ? Error = J$.R(18985, 'Error', undefined, true, true) : Error = J$.R(18985, 'Error', Error, true, true)), true)(J$.T(18993, 'Field names can\'t start with $', 21, false));
                                        else if (J$.C(680, J$.B(1138, '!==', J$.M(19025, J$.R(19009, 'key', key, false, false), 'indexOf', false)(J$.T(19017, '.', 21, false)), J$.U(1130, '-', J$.T(19033, 1, 22, false)))))
                                            throw J$.F(19057, J$.I(typeof Error === 'undefined' ? Error = J$.R(19041, 'Error', undefined, true, true) : Error = J$.R(19041, 'Error', Error, true, true)), true)(J$.T(19049, 'Field names can\'t contain .', 21, false));
                                        else if (J$.C(672, J$.B(1146, 'instanceof', J$.G(19081, J$.R(19065, 'document', document, false, false), J$.R(19073, 'key', key, false, false)), J$.I(typeof Object === 'undefined' ? Object = J$.R(19089, 'Object', undefined, true, true) : Object = J$.R(19089, 'Object', Object, true, true))))) {
                                            J$.F(19129, J$.R(19097, 'validateDocumentObject', validateDocumentObject, false, true), false)(J$.G(19121, J$.R(19105, 'document', document, false, false), J$.R(19113, 'key', key, false, false)));
                                        }
                                    }
                                }
                            }
                        } catch (J$e) {
                            J$.Ex(19625, J$e);
                        } finally {
                            if (J$.Fr(19633))
                                continue jalangiLabel21;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(19193, 'DotNotationPointers', DotNotationPointers, false, false, false);
            J$.N(19201, 'simpleComparators', simpleComparators, false, false, false);
            J$.N(19209, 'compoundOperatorComparators', compoundOperatorComparators, false, false, false);
            J$.N(19217, 'matches', matches, false, false, false);
            partMatches = J$.N(19233, 'partMatches', J$.T(19225, partMatches, 12, false), true, false, false);
            valueTest = J$.N(19249, 'valueTest', J$.T(19241, valueTest, 12, false), true, false, false);
            mongoEqual = J$.N(19265, 'mongoEqual', J$.T(19257, mongoEqual, 12, false), true, false, false);
            validateDocumentObject = J$.N(19281, 'validateDocumentObject', J$.T(19273, validateDocumentObject, 12, false), true, false, false);
            var DotNotationPointers = J$.W(14537, 'DotNotationPointers', J$.F(14529, J$.I(typeof require === 'undefined' ? require = J$.R(14513, 'require', undefined, true, true) : require = J$.R(14513, 'require', require, true, true)), false)(J$.T(14521, './DotNotationPointers', 21, false)), DotNotationPointers, false, true);
            var simpleComparators = J$.W(15497, 'simpleComparators', J$.T(15489, {
                $gt: J$.T(14601, function (a, b) {
                    jalangiLabel0:
                        while (true) {
                            try {
                                J$.Fe(14569, arguments.callee, this, arguments);
                                arguments = J$.N(14577, 'arguments', arguments, true, false, false);
                                a = J$.N(14585, 'a', a, true, false, false);
                                b = J$.N(14593, 'b', b, true, false, false);
                                return J$.Rt(14561, J$.B(394, '>', J$.R(14545, 'a', a, false, false), J$.R(14553, 'b', b, false, false)));
                            } catch (J$e) {
                                J$.Ex(19289, J$e);
                            } finally {
                                if (J$.Fr(19297))
                                    continue jalangiLabel0;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $gte: J$.T(14665, function (a, b) {
                    jalangiLabel1:
                        while (true) {
                            try {
                                J$.Fe(14633, arguments.callee, this, arguments);
                                arguments = J$.N(14641, 'arguments', arguments, true, false, false);
                                a = J$.N(14649, 'a', a, true, false, false);
                                b = J$.N(14657, 'b', b, true, false, false);
                                return J$.Rt(14625, J$.B(402, '>=', J$.R(14609, 'a', a, false, false), J$.R(14617, 'b', b, false, false)));
                            } catch (J$e) {
                                J$.Ex(19305, J$e);
                            } finally {
                                if (J$.Fr(19313))
                                    continue jalangiLabel1;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $lt: J$.T(14729, function (a, b) {
                    jalangiLabel2:
                        while (true) {
                            try {
                                J$.Fe(14697, arguments.callee, this, arguments);
                                arguments = J$.N(14705, 'arguments', arguments, true, false, false);
                                a = J$.N(14713, 'a', a, true, false, false);
                                b = J$.N(14721, 'b', b, true, false, false);
                                return J$.Rt(14689, J$.B(410, '<', J$.R(14673, 'a', a, false, false), J$.R(14681, 'b', b, false, false)));
                            } catch (J$e) {
                                J$.Ex(19321, J$e);
                            } finally {
                                if (J$.Fr(19329))
                                    continue jalangiLabel2;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $lte: J$.T(14793, function (a, b) {
                    jalangiLabel3:
                        while (true) {
                            try {
                                J$.Fe(14761, arguments.callee, this, arguments);
                                arguments = J$.N(14769, 'arguments', arguments, true, false, false);
                                a = J$.N(14777, 'a', a, true, false, false);
                                b = J$.N(14785, 'b', b, true, false, false);
                                return J$.Rt(14753, J$.B(418, '<=', J$.R(14737, 'a', a, false, false), J$.R(14745, 'b', b, false, false)));
                            } catch (J$e) {
                                J$.Ex(19337, J$e);
                            } finally {
                                if (J$.Fr(19345))
                                    continue jalangiLabel3;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $ne: J$.T(14857, function (a, b) {
                    jalangiLabel4:
                        while (true) {
                            try {
                                J$.Fe(14825, arguments.callee, this, arguments);
                                arguments = J$.N(14833, 'arguments', arguments, true, false, false);
                                a = J$.N(14841, 'a', a, true, false, false);
                                b = J$.N(14849, 'b', b, true, false, false);
                                return J$.Rt(14817, J$.B(426, '!==', J$.R(14801, 'a', a, false, false), J$.R(14809, 'b', b, false, false)));
                            } catch (J$e) {
                                J$.Ex(19353, J$e);
                            } finally {
                                if (J$.Fr(19361))
                                    continue jalangiLabel4;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $mod: J$.T(14961, function (docValue, operand) {
                    jalangiLabel5:
                        while (true) {
                            try {
                                J$.Fe(14929, arguments.callee, this, arguments);
                                arguments = J$.N(14937, 'arguments', arguments, true, false, false);
                                docValue = J$.N(14945, 'docValue', docValue, true, false, false);
                                operand = J$.N(14953, 'operand', operand, true, false, false);
                                return J$.Rt(14921, J$.B(442, '===', J$.B(434, '%', J$.R(14865, 'docValue', docValue, false, false), J$.G(14889, J$.R(14873, 'operand', operand, false, false), J$.T(14881, 0, 22, false))), J$.G(14913, J$.R(14897, 'operand', operand, false, false), J$.T(14905, 1, 22, false))));
                            } catch (J$e) {
                                J$.Ex(19369, J$e);
                            } finally {
                                if (J$.Fr(19377))
                                    continue jalangiLabel5;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $regex: J$.T(15073, function (docValue, operand) {
                    jalangiLabel6:
                        while (true) {
                            try {
                                J$.Fe(15041, arguments.callee, this, arguments);
                                arguments = J$.N(15049, 'arguments', arguments, true, false, false);
                                docValue = J$.N(15057, 'docValue', docValue, true, false, false);
                                operand = J$.N(15065, 'operand', operand, true, false, false);
                                return J$.Rt(15033, J$.C(248, J$.B(458, '===', J$.U(450, 'typeof', J$.R(14969, 'docValue', docValue, false, false)), J$.T(14977, 'string', 21, false))) ? J$.B(466, '!==', J$.M(15017, J$.R(14985, 'docValue', docValue, false, false), 'match', false)(J$.F(15009, J$.I(typeof RegExp === 'undefined' ? RegExp = J$.R(14993, 'RegExp', undefined, true, true) : RegExp = J$.R(14993, 'RegExp', RegExp, true, true)), false)(J$.R(15001, 'operand', operand, false, false))), J$.T(15025, null, 25, false)) : J$._());
                            } catch (J$e) {
                                J$.Ex(19385, J$e);
                            } finally {
                                if (J$.Fr(19393))
                                    continue jalangiLabel6;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $exists: J$.T(15145, function (docValue, operand) {
                    jalangiLabel7:
                        while (true) {
                            try {
                                J$.Fe(15113, arguments.callee, this, arguments);
                                arguments = J$.N(15121, 'arguments', arguments, true, false, false);
                                docValue = J$.N(15129, 'docValue', docValue, true, false, false);
                                operand = J$.N(15137, 'operand', operand, true, false, false);
                                return J$.Rt(15105, J$.B(482, '===', J$.B(474, '!==', J$.R(15081, 'docValue', docValue, false, false), J$.T(15089, undefined, 24, false)), J$.R(15097, 'operand', operand, false, false)));
                            } catch (J$e) {
                                J$.Ex(19401, J$e);
                            } finally {
                                if (J$.Fr(19409))
                                    continue jalangiLabel7;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $in: J$.T(15225, function (docVal, operand) {
                    jalangiLabel8:
                        while (true) {
                            try {
                                J$.Fe(15193, arguments.callee, this, arguments);
                                arguments = J$.N(15201, 'arguments', arguments, true, false, false);
                                docVal = J$.N(15209, 'docVal', docVal, true, false, false);
                                operand = J$.N(15217, 'operand', operand, true, false, false);
                                return J$.Rt(15185, J$.B(498, '!==', J$.M(15169, J$.R(15153, 'operand', operand, false, false), 'indexOf', false)(J$.R(15161, 'docVal', docVal, false, false)), J$.U(490, '-', J$.T(15177, 1, 22, false))));
                            } catch (J$e) {
                                J$.Ex(19417, J$e);
                            } finally {
                                if (J$.Fr(19425))
                                    continue jalangiLabel8;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $nin: J$.T(15305, function (docVal, operand) {
                    jalangiLabel9:
                        while (true) {
                            try {
                                J$.Fe(15273, arguments.callee, this, arguments);
                                arguments = J$.N(15281, 'arguments', arguments, true, false, false);
                                docVal = J$.N(15289, 'docVal', docVal, true, false, false);
                                operand = J$.N(15297, 'operand', operand, true, false, false);
                                return J$.Rt(15265, J$.B(514, '===', J$.M(15249, J$.R(15233, 'operand', operand, false, false), 'indexOf', false)(J$.R(15241, 'docVal', docVal, false, false)), J$.U(506, '-', J$.T(15257, 1, 22, false))));
                            } catch (J$e) {
                                J$.Ex(19433, J$e);
                            } finally {
                                if (J$.Fr(19441))
                                    continue jalangiLabel9;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $all: J$.T(15481, function (docVal, operand) {
                    jalangiLabel11:
                        while (true) {
                            try {
                                J$.Fe(15449, arguments.callee, this, arguments);
                                arguments = J$.N(15457, 'arguments', arguments, true, false, false);
                                docVal = J$.N(15465, 'docVal', docVal, true, false, false);
                                operand = J$.N(15473, 'operand', operand, true, false, false);
                                return J$.Rt(15441, J$.C(264, J$.B(522, 'instanceof', J$.R(15313, 'docVal', docVal, false, false), J$.I(typeof Array === 'undefined' ? Array = J$.R(15321, 'Array', undefined, true, true) : Array = J$.R(15321, 'Array', Array, true, true)))) ? J$.M(15433, J$.R(15329, 'docVal', docVal, false, false), 'reduce', false)(J$.T(15417, function (last, cur) {
                                    jalangiLabel10:
                                        while (true) {
                                            try {
                                                J$.Fe(15385, arguments.callee, this, arguments);
                                                arguments = J$.N(15393, 'arguments', arguments, true, false, false);
                                                last = J$.N(15401, 'last', last, true, false, false);
                                                cur = J$.N(15409, 'cur', cur, true, false, false);
                                                return J$.Rt(15377, J$.C(256, J$.R(15337, 'last', last, false, false)) ? J$.B(538, '!==', J$.M(15361, J$.R(15345, 'operand', operand, false, false), 'indexOf', false)(J$.R(15353, 'cur', cur, false, false)), J$.U(530, '-', J$.T(15369, 1, 22, false))) : J$._());
                                            } catch (J$e) {
                                                J$.Ex(19449, J$e);
                                            } finally {
                                                if (J$.Fr(19457))
                                                    continue jalangiLabel10;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), J$.T(15425, true, 23, false)) : J$._());
                            } catch (J$e) {
                                J$.Ex(19465, J$e);
                            } finally {
                                if (J$.Fr(19473))
                                    continue jalangiLabel11;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false), simpleComparators, false, true);
            var compoundOperatorComparators = J$.W(16089, 'compoundOperatorComparators', J$.T(16081, {
                $and: J$.T(15689, function (document, parts) {
                    jalangiLabel12:
                        while (true) {
                            try {
                                J$.Fe(15649, arguments.callee, this, arguments);
                                arguments = J$.N(15657, 'arguments', arguments, true, false, false);
                                document = J$.N(15665, 'document', document, true, false, false);
                                parts = J$.N(15673, 'parts', parts, true, false, false);
                                J$.N(15681, 'n', n, false, false, false);
                                for (var n = J$.W(15513, 'n', J$.T(15505, 0, 22, false), n, false, false); J$.C(280, J$.B(546, '<', J$.R(15521, 'n', n, false, false), J$.G(15537, J$.R(15529, 'parts', parts, false, false), 'length'))); J$.B(570, '-', n = J$.W(15553, 'n', J$.B(562, '+', J$.U(554, '+', J$.R(15545, 'n', n, false, false)), 1), n, false, false), 1)) {
                                    if (J$.C(272, J$.U(578, '!', J$.F(15609, J$.R(15561, 'matches', matches, false, true), false)(J$.G(15593, J$.G(15585, J$.R(15569, 'parts', parts, false, false), J$.R(15577, 'n', n, false, false)), 'parts'), J$.R(15601, 'document', document, false, false))))) {
                                        return J$.Rt(15625, J$.T(15617, false, 23, false));
                                    }
                                }
                                return J$.Rt(15641, J$.T(15633, true, 23, false));
                            } catch (J$e) {
                                J$.Ex(19481, J$e);
                            } finally {
                                if (J$.Fr(19489))
                                    continue jalangiLabel12;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $or: J$.T(15881, function (document, parts) {
                    jalangiLabel13:
                        while (true) {
                            try {
                                J$.Fe(15841, arguments.callee, this, arguments);
                                arguments = J$.N(15849, 'arguments', arguments, true, false, false);
                                document = J$.N(15857, 'document', document, true, false, false);
                                parts = J$.N(15865, 'parts', parts, true, false, false);
                                J$.N(15873, 'n', n, false, false, false);
                                for (var n = J$.W(15705, 'n', J$.T(15697, 0, 22, false), n, false, false); J$.C(296, J$.B(586, '<', J$.R(15713, 'n', n, false, false), J$.G(15729, J$.R(15721, 'parts', parts, false, false), 'length'))); J$.B(610, '-', n = J$.W(15745, 'n', J$.B(602, '+', J$.U(594, '+', J$.R(15737, 'n', n, false, false)), 1), n, false, false), 1)) {
                                    if (J$.C(288, J$.F(15801, J$.R(15753, 'matches', matches, false, true), false)(J$.G(15785, J$.G(15777, J$.R(15761, 'parts', parts, false, false), J$.R(15769, 'n', n, false, false)), 'parts'), J$.R(15793, 'document', document, false, false)))) {
                                        return J$.Rt(15817, J$.T(15809, true, 23, false));
                                    }
                                }
                                return J$.Rt(15833, J$.T(15825, false, 23, false));
                            } catch (J$e) {
                                J$.Ex(19497, J$e);
                            } finally {
                                if (J$.Fr(19505))
                                    continue jalangiLabel13;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                $nor: J$.T(16073, function (document, parts) {
                    jalangiLabel14:
                        while (true) {
                            try {
                                J$.Fe(16033, arguments.callee, this, arguments);
                                arguments = J$.N(16041, 'arguments', arguments, true, false, false);
                                document = J$.N(16049, 'document', document, true, false, false);
                                parts = J$.N(16057, 'parts', parts, true, false, false);
                                J$.N(16065, 'n', n, false, false, false);
                                for (var n = J$.W(15897, 'n', J$.T(15889, 0, 22, false), n, false, false); J$.C(312, J$.B(618, '<', J$.R(15905, 'n', n, false, false), J$.G(15921, J$.R(15913, 'parts', parts, false, false), 'length'))); J$.B(642, '-', n = J$.W(15937, 'n', J$.B(634, '+', J$.U(626, '+', J$.R(15929, 'n', n, false, false)), 1), n, false, false), 1)) {
                                    if (J$.C(304, J$.F(15993, J$.R(15945, 'matches', matches, false, true), false)(J$.G(15977, J$.G(15969, J$.R(15953, 'parts', parts, false, false), J$.R(15961, 'n', n, false, false)), 'parts'), J$.R(15985, 'document', document, false, false)))) {
                                        return J$.Rt(16009, J$.T(16001, false, 23, false));
                                    }
                                }
                                return J$.Rt(16025, J$.T(16017, true, 23, false));
                            } catch (J$e) {
                                J$.Ex(19513, J$e);
                            } finally {
                                if (J$.Fr(19521))
                                    continue jalangiLabel14;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false), compoundOperatorComparators, false, true);
            var matches = J$.W(16345, 'matches', J$.P(16337, J$.I(typeof module === 'undefined' ? module = J$.R(16097, 'module', undefined, true, true) : module = J$.R(16097, 'module', module, true, true)), 'exports', J$.T(16329, function (parts, document) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(16281, arguments.callee, this, arguments);
                            arguments = J$.N(16289, 'arguments', arguments, true, false, false);
                            parts = J$.N(16297, 'parts', parts, true, false, false);
                            document = J$.N(16305, 'document', document, true, false, false);
                            J$.N(16313, 'n', n, false, false, false);
                            J$.N(16321, 'part', part, false, false, false);
                            J$.F(16121, J$.R(16105, 'validateDocumentObject', validateDocumentObject, false, true), false)(J$.R(16113, 'document', document, false, false));
                            for (var n = J$.W(16137, 'n', J$.T(16129, 0, 22, false), n, false, false); J$.C(328, J$.B(650, '<', J$.R(16145, 'n', n, false, false), J$.G(16161, J$.R(16153, 'parts', parts, false, false), 'length'))); J$.B(674, '-', n = J$.W(16177, 'n', J$.B(666, '+', J$.U(658, '+', J$.R(16169, 'n', n, false, false)), 1), n, false, false), 1)) {
                                var part = J$.W(16209, 'part', J$.G(16201, J$.R(16185, 'parts', parts, false, false), J$.R(16193, 'n', n, false, false)), part, false, false);
                                if (J$.C(320, J$.U(682, '!', J$.F(16241, J$.R(16217, 'partMatches', partMatches, false, true), false)(J$.R(16225, 'part', part, false, false), J$.R(16233, 'document', document, false, false)))))
                                    return J$.Rt(16257, J$.T(16249, false, 23, false));
                            }
                            return J$.Rt(16273, J$.T(16265, true, 23, false));
                        } catch (J$e) {
                            J$.Ex(19529, J$e);
                        } finally {
                            if (J$.Fr(19537))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), matches, false, true);
        } catch (J$e) {
            J$.Ex(19641, J$e);
        } finally {
            if (J$.Sr(19649))
                continue jalangiLabel22;
            else
                break jalangiLabel22;
        }
    }
// JALANGI DO NOT INSTRUMENT
