// pino/lib/symbols.js
'use strict';

require("core-js/modules/es.symbol");

require("core-js/modules/es.symbol.description");

require("core-js/modules/es.object.to-string");

var setLevelSym = Symbol('pino.setLevel');
var getLevelSym = Symbol('pino.getLevel');
var levelValSym = Symbol('pino.levelVal');
var useLevelLabelsSym = Symbol('pino.useLevelLabels');
var changeLevelNameSym = Symbol('pino.changeLevelName');
var useOnlyCustomLevelsSym = Symbol('pino.useOnlyCustomLevels');
var lsCacheSym = Symbol('pino.lsCache');
var chindingsSym = Symbol('pino.chindings');
var parsedChindingsSym = Symbol('pino.parsedChindings');
var asJsonSym = Symbol('pino.asJson');
var writeSym = Symbol('pino.write');
var redactFmtSym = Symbol('pino.redactFmt');
var timeSym = Symbol('pino.time');
var timeSliceIndexSym = Symbol('pino.timeSliceIndex');
var streamSym = Symbol('pino.stream');
var stringifySym = Symbol('pino.stringify');
var stringifiersSym = Symbol('pino.stringifiers');
var endSym = Symbol('pino.end');
var formatOptsSym = Symbol('pino.formatOpts');
var messageKeySym = Symbol('pino.messageKey');
var wildcardFirstSym = Symbol('pino.wildcardFirst'); // public symbols, no need to use the same pino
// version for these

var serializersSym = Symbol["for"]('pino.serializers');
var wildcardGsym = Symbol["for"]('pino.*');
var needsMetadataGsym = Symbol["for"]('pino.metadata');
module.exports = {
  setLevelSym: setLevelSym,
  getLevelSym: getLevelSym,
  levelValSym: levelValSym,
  useLevelLabelsSym: useLevelLabelsSym,
  lsCacheSym: lsCacheSym,
  chindingsSym: chindingsSym,
  parsedChindingsSym: parsedChindingsSym,
  asJsonSym: asJsonSym,
  writeSym: writeSym,
  serializersSym: serializersSym,
  redactFmtSym: redactFmtSym,
  timeSym: timeSym,
  timeSliceIndexSym: timeSliceIndexSym,
  streamSym: streamSym,
  stringifySym: stringifySym,
  stringifiersSym: stringifiersSym,
  endSym: endSym,
  formatOptsSym: formatOptsSym,
  messageKeySym: messageKeySym,
  wildcardFirstSym: wildcardFirstSym,
  changeLevelNameSym: changeLevelNameSym,
  wildcardGsym: wildcardGsym,
  needsMetadataGsym: needsMetadataGsym,
  useOnlyCustomLevelsSym: useOnlyCustomLevelsSym
};
