// debug/karma.conf.js
J$.noInstrEval = false;
jalangiLabel1:
    while (true) {
        try {
            J$.Se(4858193, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestDatalize/node_modules/debug/karma.conf.js');
            J$.T(4857937, 'use strict', 21, false);
            J$.P(4858185, J$.I(typeof module === 'undefined' ? module = J$.R(4857945, 'module', undefined, true, true) : module = J$.R(4857945, 'module', module, true, true)), 'exports', J$.T(4858177, function (config) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(4858153, arguments.callee, this, arguments);
                            arguments = J$.N(4858161, 'arguments', arguments, true, false, false);
                            config = J$.N(4858169, 'config', config, true, false, false);
                            J$.M(4858145, J$.R(4857953, 'config', config, false, false), 'set', false)(J$.T(4858137, {
                                basePath: J$.T(4857961, '', 21, false),
                                frameworks: J$.T(4857993, [
                                    J$.T(4857969, 'mocha', 21, false),
                                    J$.T(4857977, 'chai', 21, false),
                                    J$.T(4857985, 'sinon', 21, false)
                                ], 10, false),
                                files: J$.T(4858017, [
                                    J$.T(4858001, 'dist/debug.js', 21, false),
                                    J$.T(4858009, 'test/*spec.js', 21, false)
                                ], 10, false),
                                exclude: J$.T(4858033, [J$.T(4858025, 'src/node.js', 21, false)], 10, false),
                                preprocessors: J$.T(4858041, {}, 11, false),
                                reporters: J$.T(4858057, [J$.T(4858049, 'progress', 21, false)], 10, false),
                                port: J$.T(4858065, 9876, 22, false),
                                colors: J$.T(4858073, true, 23, false),
                                logLevel: J$.G(4858089, J$.R(4858081, 'config', config, false, false), 'LOG_INFO'),
                                autoWatch: J$.T(4858097, true, 23, false),
                                browsers: J$.T(4858113, [J$.T(4858105, 'PhantomJS', 21, false)], 10, false),
                                singleRun: J$.T(4858121, false, 23, false),
                                concurrency: J$.T(4858129, Infinity, 22, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(4858201, J$e);
                        } finally {
                            if (J$.Fr(4858209))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(4858217, J$e);
        } finally {
            if (J$.Sr(4858225))
                continue jalangiLabel1;
            else
                break jalangiLabel1;
        }
    }
// JALANGI DO NOT INSTRUMENT
