// colors/lib/colors.js
J$.noInstrEval = false;
jalangiLabel23:
    while (true) {
        try {
            J$.Se(7062249, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/colors/lib/colors.js');
            function build(_styles) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(7059585, arguments.callee, this, arguments);
                            arguments = J$.N(7059593, 'arguments', arguments, true, false, false);
                            _styles = J$.N(7059601, '_styles', _styles, true, false, false);
                            J$.N(7059609, 'builder', builder, false, false, false);
                            var builder = J$.W(7059513, 'builder', J$.T(7059505, function builder() {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(7059489, arguments.callee, this, arguments);
                                            arguments = J$.N(7059497, 'arguments', arguments, true, false, false);
                                            return J$.Rt(7059481, J$.M(7059473, J$.R(7059449, 'applyStyle', applyStyle, false, true), 'apply', false)(J$.R(7059457, 'builder', builder, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7059465, 'arguments', undefined, true, true) : arguments = J$.R(7059465, 'arguments', arguments, true, true))));
                                        } catch (J$e) {
                                            J$.Ex(7062481, J$e);
                                        } finally {
                                            if (J$.Fr(7062489))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), builder, false, false);
                            J$.P(7059537, J$.R(7059521, 'builder', builder, false, false), '_styles', J$.R(7059529, '_styles', _styles, false, false));
                            J$.P(7059561, J$.R(7059545, 'builder', builder, false, false), '__proto__', J$.R(7059553, 'proto', proto, false, true));
                            return J$.Rt(7059577, J$.R(7059569, 'builder', builder, false, false));
                        } catch (J$e) {
                            J$.Ex(7062497, J$e);
                        } finally {
                            if (J$.Fr(7062505))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function applyStyle() {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(7060689, arguments.callee, this, arguments);
                            arguments = J$.N(7060697, 'arguments', arguments, true, false, false);
                            J$.N(7060705, 'args', args, false, false, false);
                            J$.N(7060713, 'str', str, false, false, false);
                            J$.N(7060721, 'newLinesPresent', newLinesPresent, false, false, false);
                            J$.N(7060729, 'nestedStyles', nestedStyles, false, false, false);
                            J$.N(7060737, 'i', i, false, false, false);
                            J$.N(7060745, 'code', code, false, false, false);
                            var args = J$.W(7060105, 'args', J$.M(7060097, J$.G(7060081, J$.G(7060073, J$.I(typeof Array === 'undefined' ? Array = J$.R(7060065, 'Array', undefined, true, true) : Array = J$.R(7060065, 'Array', Array, true, true)), 'prototype'), 'slice'), 'call', false)(J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7060089, 'arguments', undefined, true, true) : arguments = J$.R(7060089, 'arguments', arguments, true, true))), args, false, false);
                            var str = J$.W(7060265, 'str', J$.M(7060257, J$.M(7060241, J$.R(7060113, 'args', args, false, false), 'map', false)(J$.T(7060233, function (arg) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(7060209, arguments.callee, this, arguments);
                                            arguments = J$.N(7060217, 'arguments', arguments, true, false, false);
                                            arg = J$.N(7060225, 'arg', arg, true, false, false);
                                            if (J$.C(378856, J$.C(378848, J$.B(716706, '!=', J$.R(7060121, 'arg', arg, false, false), J$.T(7060129, null, 25, false))) ? J$.B(716714, '===', J$.G(7060145, J$.R(7060137, 'arg', arg, false, false), 'constructor'), J$.I(typeof String === 'undefined' ? String = J$.R(7060153, 'String', undefined, true, true) : String = J$.R(7060153, 'String', String, true, true))) : J$._())) {
                                                return J$.Rt(7060169, J$.R(7060161, 'arg', arg, false, false));
                                            } else {
                                                return J$.Rt(7060201, J$.M(7060193, J$.R(7060177, 'util', util, false, true), 'inspect', false)(J$.R(7060185, 'arg', arg, false, false)));
                                            }
                                        } catch (J$e) {
                                            J$.Ex(7062577, J$e);
                                        } finally {
                                            if (J$.Fr(7062585))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), 'join', false)(J$.T(7060249, ' ', 21, false)), str, false, false);
                            if (J$.C(378872, J$.C(378864, J$.U(716722, '!', J$.G(7060281, J$.R(7060273, 'colors', colors, false, true), 'enabled'))) ? J$._() : J$.U(716730, '!', J$.R(7060289, 'str', str, false, false)))) {
                                return J$.Rt(7060305, J$.R(7060297, 'str', str, false, false));
                            }
                            var newLinesPresent = J$.W(7060345, 'newLinesPresent', J$.B(716746, '!=', J$.M(7060329, J$.R(7060313, 'str', str, false, false), 'indexOf', false)(J$.T(7060321, '\n', 21, false)), J$.U(716738, '-', J$.T(7060337, 1, 22, false))), newLinesPresent, false, false);
                            var nestedStyles = J$.W(7060369, 'nestedStyles', J$.G(7060361, J$.R(7060353, 'this', this, false, false), '_styles'), nestedStyles, false, false);
                            var i = J$.W(7060393, 'i', J$.G(7060385, J$.R(7060377, 'nestedStyles', nestedStyles, false, false), 'length'), i, false, false);
                            while (J$.C(378888, J$.B(716770, '+', i = J$.W(7060409, 'i', J$.B(716762, '-', J$.U(716754, '+', J$.R(7060401, 'i', i, false, false)), 1), i, false, false), 1))) {
                                var code = J$.W(7060457, 'code', J$.G(7060449, J$.R(7060417, 'ansiStyles', ansiStyles, false, true), J$.G(7060441, J$.R(7060425, 'nestedStyles', nestedStyles, false, false), J$.R(7060433, 'i', i, false, false))), code, false, false);
                                str = J$.W(7060545, 'str', J$.B(716786, '+', J$.B(716778, '+', J$.G(7060473, J$.R(7060465, 'code', code, false, false), 'open'), J$.M(7060521, J$.R(7060481, 'str', str, false, false), 'replace', false)(J$.G(7060497, J$.R(7060489, 'code', code, false, false), 'closeRe'), J$.G(7060513, J$.R(7060505, 'code', code, false, false), 'open'))), J$.G(7060537, J$.R(7060529, 'code', code, false, false), 'close')), str, false, false);
                                if (J$.C(378880, J$.R(7060553, 'newLinesPresent', newLinesPresent, false, false))) {
                                    str = J$.W(7060665, 'str', J$.M(7060657, J$.R(7060561, 'str', str, false, false), 'replace', false)(J$.R(7060569, 'newLineRegex', newLineRegex, false, true), J$.T(7060649, function (match) {
                                        jalangiLabel12:
                                            while (true) {
                                                try {
                                                    J$.Fe(7060625, arguments.callee, this, arguments);
                                                    arguments = J$.N(7060633, 'arguments', arguments, true, false, false);
                                                    match = J$.N(7060641, 'match', match, true, false, false);
                                                    return J$.Rt(7060617, J$.B(716802, '+', J$.B(716794, '+', J$.G(7060585, J$.R(7060577, 'code', code, false, false), 'close'), J$.R(7060593, 'match', match, false, false)), J$.G(7060609, J$.R(7060601, 'code', code, false, false), 'open')));
                                                } catch (J$e) {
                                                    J$.Ex(7062593, J$e);
                                                } finally {
                                                    if (J$.Fr(7062601))
                                                        continue jalangiLabel12;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)), str, false, false);
                                }
                            }
                            return J$.Rt(7060681, J$.R(7060673, 'str', str, false, false));
                        } catch (J$e) {
                            J$.Ex(7062609, J$e);
                        } finally {
                            if (J$.Fr(7062617))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function init() {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(7061473, arguments.callee, this, arguments);
                            arguments = J$.N(7061481, 'arguments', arguments, true, false, false);
                            J$.N(7061489, 'ret', ret, false, false, false);
                            var ret = J$.W(7061289, 'ret', J$.T(7061281, {}, 11, false), ret, false, false);
                            J$.M(7061449, J$.M(7061313, J$.I(typeof Object === 'undefined' ? Object = J$.R(7061297, 'Object', undefined, true, true) : Object = J$.R(7061297, 'Object', Object, true, true)), 'keys', false)(J$.R(7061305, 'styles', styles, false, true)), 'forEach', false)(J$.T(7061441, function (name) {
                                jalangiLabel18:
                                    while (true) {
                                        try {
                                            J$.Fe(7061417, arguments.callee, this, arguments);
                                            arguments = J$.N(7061425, 'arguments', arguments, true, false, false);
                                            name = J$.N(7061433, 'name', name, true, false, false);
                                            J$.P(7061409, J$.R(7061321, 'ret', ret, false, false), J$.R(7061329, 'name', name, false, false), J$.T(7061401, {
                                                get: J$.T(7061393, function () {
                                                    jalangiLabel17:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(7061377, arguments.callee, this, arguments);
                                                                arguments = J$.N(7061385, 'arguments', arguments, true, false, false);
                                                                return J$.Rt(7061369, J$.F(7061361, J$.R(7061337, 'build', build, false, true), false)(J$.T(7061353, [J$.R(7061345, 'name', name, false, false)], 10, false)));
                                                            } catch (J$e) {
                                                                J$.Ex(7062673, J$e);
                                                            } finally {
                                                                if (J$.Fr(7062681))
                                                                    continue jalangiLabel17;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false)
                                            }, 11, false));
                                        } catch (J$e) {
                                            J$.Ex(7062689, J$e);
                                        } finally {
                                            if (J$.Fr(7062697))
                                                continue jalangiLabel18;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            return J$.Rt(7061465, J$.R(7061457, 'ret', ret, false, false));
                        } catch (J$e) {
                            J$.Ex(7062705, J$e);
                        } finally {
                            if (J$.Fr(7062713))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(7062257, 'colors', colors, false, false, false);
            J$.N(7062265, 'util', util, false, false, false);
            J$.N(7062273, 'ansiStyles', ansiStyles, false, false, false);
            J$.N(7062281, 'defineProps', defineProps, false, false, false);
            J$.N(7062289, 'newLineRegex', newLineRegex, false, false, false);
            J$.N(7062297, 'stylize', stylize, false, false, false);
            J$.N(7062305, 'matchOperatorsRe', matchOperatorsRe, false, false, false);
            J$.N(7062313, 'escapeStringRegexp', escapeStringRegexp, false, false, false);
            build = J$.N(7062329, 'build', J$.T(7062321, build, 12, false), true, false, false);
            J$.N(7062337, 'styles', styles, false, false, false);
            J$.N(7062345, 'proto', proto, false, false, false);
            applyStyle = J$.N(7062361, 'applyStyle', J$.T(7062353, applyStyle, 12, false), true, false, false);
            init = J$.N(7062377, 'init', J$.T(7062369, init, 12, false), true, false, false);
            J$.N(7062385, 'sequencer', sequencer, false, false, false);
            J$.N(7062393, 'map', map, false, false, false);
            var colors = J$.W(7058505, 'colors', J$.T(7058497, {}, 11, false), colors, false, true);
            J$.P(7058537, J$.I(typeof module === 'undefined' ? module = J$.R(7058513, 'module', undefined, true, true) : module = J$.R(7058513, 'module', module, true, true)), J$.T(7058521, 'exports', 21, false), J$.R(7058529, 'colors', colors, false, true));
            J$.P(7058561, J$.R(7058545, 'colors', colors, false, true), 'themes', J$.T(7058553, {}, 11, false));
            var util = J$.W(7058593, 'util', J$.F(7058585, J$.I(typeof require === 'undefined' ? require = J$.R(7058569, 'require', undefined, true, true) : require = J$.R(7058569, 'require', require, true, true)), false)(J$.T(7058577, 'util', 21, false)), util, false, true);
            var ansiStyles = J$.W(7058641, 'ansiStyles', J$.P(7058633, J$.R(7058601, 'colors', colors, false, true), 'styles', J$.F(7058625, J$.I(typeof require === 'undefined' ? require = J$.R(7058609, 'require', undefined, true, true) : require = J$.R(7058609, 'require', require, true, true)), false)(J$.T(7058617, './styles', 21, false))), ansiStyles, false, true);
            var defineProps = J$.W(7058665, 'defineProps', J$.G(7058657, J$.I(typeof Object === 'undefined' ? Object = J$.R(7058649, 'Object', undefined, true, true) : Object = J$.R(7058649, 'Object', Object, true, true)), 'defineProperties'), defineProps, false, true);
            var newLineRegex = J$.W(7058697, 'newLineRegex', J$.F(7058689, J$.I(typeof RegExp === 'undefined' ? RegExp = J$.R(7058673, 'RegExp', undefined, true, true) : RegExp = J$.R(7058673, 'RegExp', RegExp, true, true)), true)(J$.T(7058681, /[\r\n]+/g, 14, false)), newLineRegex, false, true);
            J$.P(7058745, J$.R(7058705, 'colors', colors, false, true), 'supportsColor', J$.G(7058737, J$.F(7058729, J$.I(typeof require === 'undefined' ? require = J$.R(7058713, 'require', undefined, true, true) : require = J$.R(7058713, 'require', require, true, true)), false)(J$.T(7058721, './system/supports-colors', 21, false)), 'supportsColor'));
            if (J$.C(378808, J$.B(716618, '===', J$.U(716610, 'typeof', J$.G(7058761, J$.R(7058753, 'colors', colors, false, true), 'enabled')), J$.T(7058769, 'undefined', 21, false)))) {
                J$.P(7058809, J$.R(7058777, 'colors', colors, false, true), 'enabled', J$.B(716626, '!==', J$.M(7058793, J$.R(7058785, 'colors', colors, false, true), 'supportsColor', false)(), J$.T(7058801, false, 23, false)));
            }
            J$.P(7058873, J$.R(7058817, 'colors', colors, false, true), 'enable', J$.T(7058865, function () {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(7058849, arguments.callee, this, arguments);
                            arguments = J$.N(7058857, 'arguments', arguments, true, false, false);
                            J$.P(7058841, J$.R(7058825, 'colors', colors, false, true), 'enabled', J$.T(7058833, true, 23, false));
                        } catch (J$e) {
                            J$.Ex(7062401, J$e);
                        } finally {
                            if (J$.Fr(7062409))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7058937, J$.R(7058881, 'colors', colors, false, true), 'disable', J$.T(7058929, function () {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7058913, arguments.callee, this, arguments);
                            arguments = J$.N(7058921, 'arguments', arguments, true, false, false);
                            J$.P(7058905, J$.R(7058889, 'colors', colors, false, true), 'enabled', J$.T(7058897, false, 23, false));
                        } catch (J$e) {
                            J$.Ex(7062417, J$e);
                        } finally {
                            if (J$.Fr(7062425))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7059049, J$.R(7058945, 'colors', colors, false, true), 'stripColors', J$.P(7059041, J$.R(7058953, 'colors', colors, false, true), 'strip', J$.T(7059033, function (str) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(7059009, arguments.callee, this, arguments);
                            arguments = J$.N(7059017, 'arguments', arguments, true, false, false);
                            str = J$.N(7059025, 'str', str, true, false, false);
                            return J$.Rt(7059001, J$.M(7058993, J$.B(716634, '+', J$.T(7058961, '', 21, false), J$.R(7058969, 'str', str, false, false)), 'replace', false)(J$.T(7058977, /\x1B\[\d+m/g, 14, false), J$.T(7058985, '', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7062433, J$e);
                        } finally {
                            if (J$.Fr(7062441))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)));
            var stylize = J$.W(7059305, 'stylize', J$.P(7059297, J$.R(7059057, 'colors', colors, false, true), 'stylize', J$.T(7059289, function stylize(str, style) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(7059249, arguments.callee, this, arguments);
                            arguments = J$.N(7059257, 'arguments', arguments, true, false, false);
                            str = J$.N(7059265, 'str', str, true, false, false);
                            style = J$.N(7059273, 'style', style, true, false, false);
                            J$.N(7059281, 'styleMap', styleMap, false, false, false);
                            if (J$.C(378816, J$.U(716642, '!', J$.G(7059073, J$.R(7059065, 'colors', colors, false, true), 'enabled')))) {
                                return J$.Rt(7059097, J$.B(716650, '+', J$.R(7059081, 'str', str, false, false), J$.T(7059089, '', 21, false)));
                            }
                            var styleMap = J$.W(7059129, 'styleMap', J$.G(7059121, J$.R(7059105, 'ansiStyles', ansiStyles, false, true), J$.R(7059113, 'style', style, false, false)), styleMap, false, false);
                            if (J$.C(378832, J$.C(378824, J$.U(716658, '!', J$.R(7059137, 'styleMap', styleMap, false, false))) ? J$.B(716666, 'in', J$.R(7059145, 'style', style, false, false), J$.R(7059153, 'colors', colors, false, true)) : J$._())) {
                                return J$.Rt(7059193, J$.M(7059185, J$.R(7059161, 'colors', colors, false, true), J$.R(7059169, 'style', style, false, false), false)(J$.R(7059177, 'str', str, false, false)));
                            }
                            return J$.Rt(7059241, J$.B(716682, '+', J$.B(716674, '+', J$.G(7059209, J$.R(7059201, 'styleMap', styleMap, false, false), 'open'), J$.R(7059217, 'str', str, false, false)), J$.G(7059233, J$.R(7059225, 'styleMap', styleMap, false, false), 'close')));
                        } catch (J$e) {
                            J$.Ex(7062449, J$e);
                        } finally {
                            if (J$.Fr(7062457))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), stylize, false, true);
            var matchOperatorsRe = J$.W(7059321, 'matchOperatorsRe', J$.T(7059313, /[|\\{}()[\]^$+*?.]/g, 14, false), matchOperatorsRe, false, true);
            var escapeStringRegexp = J$.W(7059441, 'escapeStringRegexp', J$.T(7059433, function (str) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7059409, arguments.callee, this, arguments);
                            arguments = J$.N(7059417, 'arguments', arguments, true, false, false);
                            str = J$.N(7059425, 'str', str, true, false, false);
                            if (J$.C(378840, J$.B(716698, '!==', J$.U(716690, 'typeof', J$.R(7059329, 'str', str, false, false)), J$.T(7059337, 'string', 21, false)))) {
                                throw J$.F(7059361, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(7059345, 'TypeError', undefined, true, true) : TypeError = J$.R(7059345, 'TypeError', TypeError, true, true)), true)(J$.T(7059353, 'Expected a string', 21, false));
                            }
                            return J$.Rt(7059401, J$.M(7059393, J$.R(7059369, 'str', str, false, false), 'replace', false)(J$.R(7059377, 'matchOperatorsRe', matchOperatorsRe, false, true), J$.T(7059385, '\\$&', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7062465, J$e);
                        } finally {
                            if (J$.Fr(7062473))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), escapeStringRegexp, false, true);
            var styles = J$.W(7060001, 'styles', J$.F(7059993, J$.T(7059985, function () {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(7059961, arguments.callee, this, arguments);
                            arguments = J$.N(7059969, 'arguments', arguments, true, false, false);
                            J$.N(7059977, 'ret', ret, false, false, false);
                            var ret = J$.W(7059625, 'ret', J$.T(7059617, {}, 11, false), ret, false, false);
                            J$.P(7059657, J$.R(7059633, 'ansiStyles', ansiStyles, false, true), 'grey', J$.G(7059649, J$.R(7059641, 'ansiStyles', ansiStyles, false, true), 'gray'));
                            J$.M(7059937, J$.M(7059681, J$.I(typeof Object === 'undefined' ? Object = J$.R(7059665, 'Object', undefined, true, true) : Object = J$.R(7059665, 'Object', Object, true, true)), 'keys', false)(J$.R(7059673, 'ansiStyles', ansiStyles, false, true)), 'forEach', false)(J$.T(7059929, function (key) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(7059905, arguments.callee, this, arguments);
                                            arguments = J$.N(7059913, 'arguments', arguments, true, false, false);
                                            key = J$.N(7059921, 'key', key, true, false, false);
                                            J$.P(7059785, J$.G(7059705, J$.R(7059689, 'ansiStyles', ansiStyles, false, true), J$.R(7059697, 'key', key, false, false)), 'closeRe', J$.F(7059777, J$.I(typeof RegExp === 'undefined' ? RegExp = J$.R(7059713, 'RegExp', undefined, true, true) : RegExp = J$.R(7059713, 'RegExp', RegExp, true, true)), true)(J$.F(7059761, J$.R(7059721, 'escapeStringRegexp', escapeStringRegexp, false, true), false)(J$.G(7059753, J$.G(7059745, J$.R(7059729, 'ansiStyles', ansiStyles, false, true), J$.R(7059737, 'key', key, false, false)), 'close')), J$.T(7059769, 'g', 21, false)));
                                            J$.P(7059897, J$.R(7059793, 'ret', ret, false, false), J$.R(7059801, 'key', key, false, false), J$.T(7059889, {
                                                get: J$.T(7059881, function () {
                                                    jalangiLabel7:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(7059865, arguments.callee, this, arguments);
                                                                arguments = J$.N(7059873, 'arguments', arguments, true, false, false);
                                                                return J$.Rt(7059857, J$.F(7059849, J$.R(7059809, 'build', build, false, true), false)(J$.M(7059841, J$.G(7059825, J$.R(7059817, 'this', this, false, false), '_styles'), 'concat', false)(J$.R(7059833, 'key', key, false, false))));
                                                            } catch (J$e) {
                                                                J$.Ex(7062513, J$e);
                                                            } finally {
                                                                if (J$.Fr(7062521))
                                                                    continue jalangiLabel7;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false)
                                            }, 11, false));
                                        } catch (J$e) {
                                            J$.Ex(7062529, J$e);
                                        } finally {
                                            if (J$.Fr(7062537))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            return J$.Rt(7059953, J$.R(7059945, 'ret', ret, false, false));
                        } catch (J$e) {
                            J$.Ex(7062545, J$e);
                        } finally {
                            if (J$.Fr(7062553))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(), styles, false, true);
            var proto = J$.W(7060057, 'proto', J$.F(7060049, J$.R(7060009, 'defineProps', defineProps, false, true), false)(J$.T(7060033, function colors() {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(7060017, arguments.callee, this, arguments);
                            arguments = J$.N(7060025, 'arguments', arguments, true, false, false);
                        } catch (J$e) {
                            J$.Ex(7062561, J$e);
                        } finally {
                            if (J$.Fr(7062569))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), J$.R(7060041, 'styles', styles, false, true)), proto, false, true);
            J$.P(7061273, J$.R(7060753, 'colors', colors, false, true), 'setTheme', J$.T(7061265, function (theme) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(7061233, arguments.callee, this, arguments);
                            arguments = J$.N(7061241, 'arguments', arguments, true, false, false);
                            theme = J$.N(7061249, 'theme', theme, true, false, false);
                            J$.N(7061257, 'style', style, false, false, false);
                            if (J$.C(378896, J$.B(716818, '===', J$.U(716810, 'typeof', J$.R(7060761, 'theme', theme, false, false)), J$.T(7060769, 'string', 21, false)))) {
                                J$.M(7060841, J$.I(typeof console === 'undefined' ? console = J$.R(7060777, 'console', undefined, true, true) : console = J$.R(7060777, 'console', console, true, true)), 'log', false)(J$.B(716866, '+', J$.B(716858, '+', J$.B(716850, '+', J$.B(716842, '+', J$.B(716834, '+', J$.B(716826, '+', J$.T(7060785, 'colors.setTheme now only accepts an object, not a string.  ', 21, false), J$.T(7060793, 'If you are trying to set a theme from a file, it is now your (the ', 21, false)), J$.T(7060801, 'caller\'s) responsibility to require the file.  The old syntax ', 21, false)), J$.T(7060809, 'looked like colors.setTheme(__dirname + ', 21, false)), J$.T(7060817, '\'/../themes/generic-logging.js\'); The new syntax looks like ', 21, false)), J$.T(7060825, 'colors.setTheme(require(__dirname + ', 21, false)), J$.T(7060833, '\'/../themes/generic-logging.js\'));', 21, false)));
                                return J$.Rt(7060849, undefined);
                            }
                            for (var style in J$.H(7061217, J$.R(7060857, 'theme', theme, false, false))) {
                                J$.N(7061225, 'style', style, false, true, false);
                                {
                                    {
                                        J$.F(7061209, J$.T(7061193, function (style) {
                                            jalangiLabel15:
                                                while (true) {
                                                    try {
                                                        J$.Fe(7061169, arguments.callee, this, arguments);
                                                        arguments = J$.N(7061177, 'arguments', arguments, true, false, false);
                                                        style = J$.N(7061185, 'style', style, true, false, false);
                                                        J$.P(7061161, J$.R(7060865, 'colors', colors, false, true), J$.R(7060873, 'style', style, false, false), J$.T(7061153, function (str) {
                                                            jalangiLabel14:
                                                                while (true) {
                                                                    try {
                                                                        J$.Fe(7061113, arguments.callee, this, arguments);
                                                                        arguments = J$.N(7061121, 'arguments', arguments, true, false, false);
                                                                        str = J$.N(7061129, 'str', str, true, false, false);
                                                                        J$.N(7061137, 'out', out, false, false, false);
                                                                        J$.N(7061145, 'i', i, false, false, false);
                                                                        if (J$.C(378904, J$.B(716882, '===', J$.U(716874, 'typeof', J$.G(7060897, J$.R(7060881, 'theme', theme, false, false), J$.R(7060889, 'style', style, false, false))), J$.T(7060905, 'object', 21, false)))) {
                                                                            var out = J$.W(7060921, 'out', J$.R(7060913, 'str', str, false, false), out, false, false);
                                                                            for (var i in J$.H(7061025, J$.G(7060945, J$.R(7060929, 'theme', theme, false, false), J$.R(7060937, 'style', style, false, false)))) {
                                                                                J$.N(7061033, 'i', i, false, true, false);
                                                                                {
                                                                                    {
                                                                                        out = J$.W(7061017, 'out', J$.M(7061009, J$.R(7060953, 'colors', colors, false, true), J$.G(7060993, J$.G(7060977, J$.R(7060961, 'theme', theme, false, false), J$.R(7060969, 'style', style, false, false)), J$.R(7060985, 'i', i, false, false)), false)(J$.R(7061001, 'out', out, false, false)), out, false, false);
                                                                                    }
                                                                                }
                                                                            }
                                                                            return J$.Rt(7061049, J$.R(7061041, 'out', out, false, false));
                                                                        }
                                                                        return J$.Rt(7061105, J$.M(7061097, J$.R(7061057, 'colors', colors, false, true), J$.G(7061081, J$.R(7061065, 'theme', theme, false, false), J$.R(7061073, 'style', style, false, false)), false)(J$.R(7061089, 'str', str, false, false)));
                                                                    } catch (J$e) {
                                                                        J$.Ex(7062625, J$e);
                                                                    } finally {
                                                                        if (J$.Fr(7062633))
                                                                            continue jalangiLabel14;
                                                                        else
                                                                            return J$.Ra();
                                                                    }
                                                                }
                                                        }, 12, false));
                                                    } catch (J$e) {
                                                        J$.Ex(7062641, J$e);
                                                    } finally {
                                                        if (J$.Fr(7062649))
                                                            continue jalangiLabel15;
                                                        else
                                                            return J$.Ra();
                                                    }
                                                }
                                        }, 12, false), false)(J$.R(7061201, 'style', style, false, false));
                                    }
                                }
                            }
                        } catch (J$e) {
                            J$.Ex(7062657, J$e);
                        } finally {
                            if (J$.Fr(7062665))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var sequencer = J$.W(7061641, 'sequencer', J$.T(7061633, function sequencer(map, str) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(7061593, arguments.callee, this, arguments);
                            arguments = J$.N(7061601, 'arguments', arguments, true, false, false);
                            map = J$.N(7061609, 'map', map, true, false, false);
                            str = J$.N(7061617, 'str', str, true, false, false);
                            J$.N(7061625, 'exploded', exploded, false, false, false);
                            var exploded = J$.W(7061521, 'exploded', J$.M(7061513, J$.R(7061497, 'str', str, false, false), 'split', false)(J$.T(7061505, '', 21, false)), exploded, false, false);
                            exploded = J$.W(7061553, 'exploded', J$.M(7061545, J$.R(7061529, 'exploded', exploded, false, false), 'map', false)(J$.R(7061537, 'map', map, false, false)), exploded, false, false);
                            return J$.Rt(7061585, J$.M(7061577, J$.R(7061561, 'exploded', exploded, false, false), 'join', false)(J$.T(7061569, '', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7062721, J$e);
                        } finally {
                            if (J$.Fr(7062729))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), sequencer, false, true);
            J$.P(7061681, J$.R(7061649, 'colors', colors, false, true), 'trap', J$.F(7061673, J$.I(typeof require === 'undefined' ? require = J$.R(7061657, 'require', undefined, true, true) : require = J$.R(7061657, 'require', require, true, true)), false)(J$.T(7061665, './custom/trap', 21, false)));
            J$.P(7061721, J$.R(7061689, 'colors', colors, false, true), 'zalgo', J$.F(7061713, J$.I(typeof require === 'undefined' ? require = J$.R(7061697, 'require', undefined, true, true) : require = J$.R(7061697, 'require', require, true, true)), false)(J$.T(7061705, './custom/zalgo', 21, false)));
            J$.P(7061745, J$.R(7061729, 'colors', colors, false, true), 'maps', J$.T(7061737, {}, 11, false));
            J$.P(7061809, J$.G(7061761, J$.R(7061753, 'colors', colors, false, true), 'maps'), 'america', J$.F(7061801, J$.F(7061785, J$.I(typeof require === 'undefined' ? require = J$.R(7061769, 'require', undefined, true, true) : require = J$.R(7061769, 'require', require, true, true)), false)(J$.T(7061777, './maps/america', 21, false)), false)(J$.R(7061793, 'colors', colors, false, true)));
            J$.P(7061873, J$.G(7061825, J$.R(7061817, 'colors', colors, false, true), 'maps'), 'zebra', J$.F(7061865, J$.F(7061849, J$.I(typeof require === 'undefined' ? require = J$.R(7061833, 'require', undefined, true, true) : require = J$.R(7061833, 'require', require, true, true)), false)(J$.T(7061841, './maps/zebra', 21, false)), false)(J$.R(7061857, 'colors', colors, false, true)));
            J$.P(7061937, J$.G(7061889, J$.R(7061881, 'colors', colors, false, true), 'maps'), 'rainbow', J$.F(7061929, J$.F(7061913, J$.I(typeof require === 'undefined' ? require = J$.R(7061897, 'require', undefined, true, true) : require = J$.R(7061897, 'require', require, true, true)), false)(J$.T(7061905, './maps/rainbow', 21, false)), false)(J$.R(7061921, 'colors', colors, false, true)));
            J$.P(7062001, J$.G(7061953, J$.R(7061945, 'colors', colors, false, true), 'maps'), 'random', J$.F(7061993, J$.F(7061977, J$.I(typeof require === 'undefined' ? require = J$.R(7061961, 'require', undefined, true, true) : require = J$.R(7061961, 'require', require, true, true)), false)(J$.T(7061969, './maps/random', 21, false)), false)(J$.R(7061985, 'colors', colors, false, true)));
            for (var map in J$.H(7062193, J$.G(7062017, J$.R(7062009, 'colors', colors, false, true), 'maps'))) {
                J$.N(7062201, 'map', map, false, true, false);
                {
                    {
                        J$.F(7062185, J$.T(7062169, function (map) {
                            jalangiLabel22:
                                while (true) {
                                    try {
                                        J$.Fe(7062145, arguments.callee, this, arguments);
                                        arguments = J$.N(7062153, 'arguments', arguments, true, false, false);
                                        map = J$.N(7062161, 'map', map, true, false, false);
                                        J$.P(7062137, J$.R(7062025, 'colors', colors, false, true), J$.R(7062033, 'map', map, false, false), J$.T(7062129, function (str) {
                                            jalangiLabel21:
                                                while (true) {
                                                    try {
                                                        J$.Fe(7062105, arguments.callee, this, arguments);
                                                        arguments = J$.N(7062113, 'arguments', arguments, true, false, false);
                                                        str = J$.N(7062121, 'str', str, true, false, false);
                                                        return J$.Rt(7062097, J$.F(7062089, J$.R(7062041, 'sequencer', sequencer, false, true), false)(J$.G(7062073, J$.G(7062057, J$.R(7062049, 'colors', colors, false, true), 'maps'), J$.R(7062065, 'map', map, false, false)), J$.R(7062081, 'str', str, false, false)));
                                                    } catch (J$e) {
                                                        J$.Ex(7062737, J$e);
                                                    } finally {
                                                        if (J$.Fr(7062745))
                                                            continue jalangiLabel21;
                                                        else
                                                            return J$.Ra();
                                                    }
                                                }
                                        }, 12, false));
                                    } catch (J$e) {
                                        J$.Ex(7062753, J$e);
                                    } finally {
                                        if (J$.Fr(7062761))
                                            continue jalangiLabel22;
                                        else
                                            return J$.Ra();
                                    }
                                }
                        }, 12, false), false)(J$.R(7062177, 'map', map, false, true));
                    }
                }
            }
            J$.F(7062241, J$.R(7062209, 'defineProps', defineProps, false, true), false)(J$.R(7062217, 'colors', colors, false, true), J$.F(7062233, J$.R(7062225, 'init', init, false, true), false)());
        } catch (J$e) {
            J$.Ex(7062769, J$e);
        } finally {
            if (J$.Sr(7062777))
                continue jalangiLabel23;
            else
                break jalangiLabel23;
        }
    }
// JALANGI DO NOT INSTRUMENT
