// colors/lib/styles.js
J$.noInstrEval = false;
jalangiLabel1:
    while (true) {
        try {
            J$.Se(7072777, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/colors/lib/styles.js');
            J$.N(7072785, 'styles', styles, false, false, false);
            J$.N(7072793, 'codes', codes, false, false, false);
            var styles = J$.W(7071257, 'styles', J$.T(7071249, {}, 11, false), styles, false, true);
            J$.P(7071289, J$.I(typeof module === 'undefined' ? module = J$.R(7071265, 'module', undefined, true, true) : module = J$.R(7071265, 'module', module, true, true)), J$.T(7071273, 'exports', 21, false), J$.R(7071281, 'styles', styles, false, true));
            var codes = J$.W(7072505, 'codes', J$.T(7072497, {
                reset: J$.T(7071313, [
                    J$.T(7071297, 0, 22, false),
                    J$.T(7071305, 0, 22, false)
                ], 10, false),
                bold: J$.T(7071337, [
                    J$.T(7071321, 1, 22, false),
                    J$.T(7071329, 22, 22, false)
                ], 10, false),
                dim: J$.T(7071361, [
                    J$.T(7071345, 2, 22, false),
                    J$.T(7071353, 22, 22, false)
                ], 10, false),
                italic: J$.T(7071385, [
                    J$.T(7071369, 3, 22, false),
                    J$.T(7071377, 23, 22, false)
                ], 10, false),
                underline: J$.T(7071409, [
                    J$.T(7071393, 4, 22, false),
                    J$.T(7071401, 24, 22, false)
                ], 10, false),
                inverse: J$.T(7071433, [
                    J$.T(7071417, 7, 22, false),
                    J$.T(7071425, 27, 22, false)
                ], 10, false),
                hidden: J$.T(7071457, [
                    J$.T(7071441, 8, 22, false),
                    J$.T(7071449, 28, 22, false)
                ], 10, false),
                strikethrough: J$.T(7071481, [
                    J$.T(7071465, 9, 22, false),
                    J$.T(7071473, 29, 22, false)
                ], 10, false),
                black: J$.T(7071505, [
                    J$.T(7071489, 30, 22, false),
                    J$.T(7071497, 39, 22, false)
                ], 10, false),
                red: J$.T(7071529, [
                    J$.T(7071513, 31, 22, false),
                    J$.T(7071521, 39, 22, false)
                ], 10, false),
                green: J$.T(7071553, [
                    J$.T(7071537, 32, 22, false),
                    J$.T(7071545, 39, 22, false)
                ], 10, false),
                yellow: J$.T(7071577, [
                    J$.T(7071561, 33, 22, false),
                    J$.T(7071569, 39, 22, false)
                ], 10, false),
                blue: J$.T(7071601, [
                    J$.T(7071585, 34, 22, false),
                    J$.T(7071593, 39, 22, false)
                ], 10, false),
                magenta: J$.T(7071625, [
                    J$.T(7071609, 35, 22, false),
                    J$.T(7071617, 39, 22, false)
                ], 10, false),
                cyan: J$.T(7071649, [
                    J$.T(7071633, 36, 22, false),
                    J$.T(7071641, 39, 22, false)
                ], 10, false),
                white: J$.T(7071673, [
                    J$.T(7071657, 37, 22, false),
                    J$.T(7071665, 39, 22, false)
                ], 10, false),
                gray: J$.T(7071697, [
                    J$.T(7071681, 90, 22, false),
                    J$.T(7071689, 39, 22, false)
                ], 10, false),
                grey: J$.T(7071721, [
                    J$.T(7071705, 90, 22, false),
                    J$.T(7071713, 39, 22, false)
                ], 10, false),
                brightRed: J$.T(7071745, [
                    J$.T(7071729, 91, 22, false),
                    J$.T(7071737, 39, 22, false)
                ], 10, false),
                brightGreen: J$.T(7071769, [
                    J$.T(7071753, 92, 22, false),
                    J$.T(7071761, 39, 22, false)
                ], 10, false),
                brightYellow: J$.T(7071793, [
                    J$.T(7071777, 93, 22, false),
                    J$.T(7071785, 39, 22, false)
                ], 10, false),
                brightBlue: J$.T(7071817, [
                    J$.T(7071801, 94, 22, false),
                    J$.T(7071809, 39, 22, false)
                ], 10, false),
                brightMagenta: J$.T(7071841, [
                    J$.T(7071825, 95, 22, false),
                    J$.T(7071833, 39, 22, false)
                ], 10, false),
                brightCyan: J$.T(7071865, [
                    J$.T(7071849, 96, 22, false),
                    J$.T(7071857, 39, 22, false)
                ], 10, false),
                brightWhite: J$.T(7071889, [
                    J$.T(7071873, 97, 22, false),
                    J$.T(7071881, 39, 22, false)
                ], 10, false),
                bgBlack: J$.T(7071913, [
                    J$.T(7071897, 40, 22, false),
                    J$.T(7071905, 49, 22, false)
                ], 10, false),
                bgRed: J$.T(7071937, [
                    J$.T(7071921, 41, 22, false),
                    J$.T(7071929, 49, 22, false)
                ], 10, false),
                bgGreen: J$.T(7071961, [
                    J$.T(7071945, 42, 22, false),
                    J$.T(7071953, 49, 22, false)
                ], 10, false),
                bgYellow: J$.T(7071985, [
                    J$.T(7071969, 43, 22, false),
                    J$.T(7071977, 49, 22, false)
                ], 10, false),
                bgBlue: J$.T(7072009, [
                    J$.T(7071993, 44, 22, false),
                    J$.T(7072001, 49, 22, false)
                ], 10, false),
                bgMagenta: J$.T(7072033, [
                    J$.T(7072017, 45, 22, false),
                    J$.T(7072025, 49, 22, false)
                ], 10, false),
                bgCyan: J$.T(7072057, [
                    J$.T(7072041, 46, 22, false),
                    J$.T(7072049, 49, 22, false)
                ], 10, false),
                bgWhite: J$.T(7072081, [
                    J$.T(7072065, 47, 22, false),
                    J$.T(7072073, 49, 22, false)
                ], 10, false),
                bgGray: J$.T(7072105, [
                    J$.T(7072089, 100, 22, false),
                    J$.T(7072097, 49, 22, false)
                ], 10, false),
                bgGrey: J$.T(7072129, [
                    J$.T(7072113, 100, 22, false),
                    J$.T(7072121, 49, 22, false)
                ], 10, false),
                bgBrightRed: J$.T(7072153, [
                    J$.T(7072137, 101, 22, false),
                    J$.T(7072145, 49, 22, false)
                ], 10, false),
                bgBrightGreen: J$.T(7072177, [
                    J$.T(7072161, 102, 22, false),
                    J$.T(7072169, 49, 22, false)
                ], 10, false),
                bgBrightYellow: J$.T(7072201, [
                    J$.T(7072185, 103, 22, false),
                    J$.T(7072193, 49, 22, false)
                ], 10, false),
                bgBrightBlue: J$.T(7072225, [
                    J$.T(7072209, 104, 22, false),
                    J$.T(7072217, 49, 22, false)
                ], 10, false),
                bgBrightMagenta: J$.T(7072249, [
                    J$.T(7072233, 105, 22, false),
                    J$.T(7072241, 49, 22, false)
                ], 10, false),
                bgBrightCyan: J$.T(7072273, [
                    J$.T(7072257, 106, 22, false),
                    J$.T(7072265, 49, 22, false)
                ], 10, false),
                bgBrightWhite: J$.T(7072297, [
                    J$.T(7072281, 107, 22, false),
                    J$.T(7072289, 49, 22, false)
                ], 10, false),
                blackBG: J$.T(7072321, [
                    J$.T(7072305, 40, 22, false),
                    J$.T(7072313, 49, 22, false)
                ], 10, false),
                redBG: J$.T(7072345, [
                    J$.T(7072329, 41, 22, false),
                    J$.T(7072337, 49, 22, false)
                ], 10, false),
                greenBG: J$.T(7072369, [
                    J$.T(7072353, 42, 22, false),
                    J$.T(7072361, 49, 22, false)
                ], 10, false),
                yellowBG: J$.T(7072393, [
                    J$.T(7072377, 43, 22, false),
                    J$.T(7072385, 49, 22, false)
                ], 10, false),
                blueBG: J$.T(7072417, [
                    J$.T(7072401, 44, 22, false),
                    J$.T(7072409, 49, 22, false)
                ], 10, false),
                magentaBG: J$.T(7072441, [
                    J$.T(7072425, 45, 22, false),
                    J$.T(7072433, 49, 22, false)
                ], 10, false),
                cyanBG: J$.T(7072465, [
                    J$.T(7072449, 46, 22, false),
                    J$.T(7072457, 49, 22, false)
                ], 10, false),
                whiteBG: J$.T(7072489, [
                    J$.T(7072473, 47, 22, false),
                    J$.T(7072481, 49, 22, false)
                ], 10, false)
            }, 11, false), codes, false, true);
            J$.M(7072769, J$.M(7072529, J$.I(typeof Object === 'undefined' ? Object = J$.R(7072513, 'Object', undefined, true, true) : Object = J$.R(7072513, 'Object', Object, true, true)), 'keys', false)(J$.R(7072521, 'codes', codes, false, true)), 'forEach', false)(J$.T(7072761, function (key) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(7072721, arguments.callee, this, arguments);
                            arguments = J$.N(7072729, 'arguments', arguments, true, false, false);
                            key = J$.N(7072737, 'key', key, true, false, false);
                            J$.N(7072745, 'val', val, false, false, false);
                            J$.N(7072753, 'style', style, false, false, false);
                            var val = J$.W(7072561, 'val', J$.G(7072553, J$.R(7072537, 'codes', codes, false, true), J$.R(7072545, 'key', key, false, false)), val, false, false);
                            var style = J$.W(7072601, 'style', J$.P(7072593, J$.R(7072569, 'styles', styles, false, true), J$.R(7072577, 'key', key, false, false), J$.T(7072585, [], 10, false)), style, false, false);
                            J$.P(7072657, J$.R(7072609, 'style', style, false, false), 'open', J$.B(717370, '+', J$.B(717362, '+', J$.T(7072617, '\x1B[', 21, false), J$.G(7072641, J$.R(7072625, 'val', val, false, false), J$.T(7072633, 0, 22, false))), J$.T(7072649, 'm', 21, false)));
                            J$.P(7072713, J$.R(7072665, 'style', style, false, false), 'close', J$.B(717386, '+', J$.B(717378, '+', J$.T(7072673, '\x1B[', 21, false), J$.G(7072697, J$.R(7072681, 'val', val, false, false), J$.T(7072689, 1, 22, false))), J$.T(7072705, 'm', 21, false)));
                        } catch (J$e) {
                            J$.Ex(7072801, J$e);
                        } finally {
                            if (J$.Fr(7072809))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(7072817, J$e);
        } finally {
            if (J$.Sr(7072825))
                continue jalangiLabel1;
            else
                break jalangiLabel1;
        }
    }
// JALANGI DO NOT INSTRUMENT
