// regre_tester/index.js
J$.noInstrEval = false;
jalangiLabel15:
    while (true) {
        try {
            J$.Se(3726849, '/mnt/data/fxiao/HiPar/outputs/target_cache/TestRegression/node_modules/regre_tester/index.js');
            function do_test(user_input) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(3725497, arguments.callee, this, arguments);
                            arguments = J$.N(3725505, 'arguments', arguments, true, false, false);
                            user_input = J$.N(3725513, 'user_input', user_input, true, false, false);
                            J$.F(3725329, J$.R(3725313, 'one_layer_hipar', one_layer_hipar, false, true), false)(J$.R(3725321, 'user_input', user_input, false, false));
                            J$.F(3725353, J$.R(3725337, 'multi_layer_hipar', multi_layer_hipar, false, true), false)(J$.R(3725345, 'user_input', user_input, false, false));
                            J$.F(3725377, J$.R(3725361, 'mix_hipar', mix_hipar, false, true), false)(J$.R(3725369, 'user_input', user_input, false, false));
                            J$.F(3725401, J$.R(3725385, 'proto_hipar', proto_hipar, false, true), false)(J$.R(3725393, 'user_input', user_input, false, false));
                            J$.F(3725433, J$.R(3725409, 'child_hipar', child_hipar, false, true), false)(J$.T(3725425, { 'root': J$.R(3725417, 'user_input', user_input, false, false) }, 11, false));
                            J$.F(3725465, J$.R(3725441, 'time_hipar', time_hipar, false, true), false)(J$.T(3725457, { 'root': J$.R(3725449, 'user_input', user_input, false, false) }, 11, false));
                            J$.F(3725489, J$.R(3725473, 'func_var_nested_hipar', func_var_nested_hipar, false, true), false)(J$.R(3725481, 'user_input', user_input, false, false));
                        } catch (J$e) {
                            J$.Ex(3726993, J$e);
                        } finally {
                            if (J$.Fr(3727001))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function one_layer_hipar(obj) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(3725617, arguments.callee, this, arguments);
                            arguments = J$.N(3725625, 'arguments', arguments, true, false, false);
                            obj = J$.N(3725633, 'obj', obj, true, false, false);
                            J$.N(3725641, 'b', b, false, false, false);
                            if (J$.C(200960, J$.G(3725561, J$.R(3725545, 'obj', obj, false, false), J$.T(3725553, 'explicit_one_layer_hipar', 21, false)))) {
                                return J$.Rt(3725569, undefined);
                            }
                            if (J$.C(200968, J$.T(3725577, false, 23, false))) {
                                var b = J$.W(3725609, 'b', J$.G(3725601, J$.R(3725585, 'obj', obj, false, false), J$.T(3725593, 'implicit_one_layer_hipar', 21, false)), b, false, false);
                            }
                        } catch (J$e) {
                            J$.Ex(3727009, J$e);
                        } finally {
                            if (J$.Fr(3727017))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function multi_layer_hipar(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(3725745, arguments.callee, this, arguments);
                            arguments = J$.N(3725753, 'arguments', arguments, true, false, false);
                            obj = J$.N(3725761, 'obj', obj, true, false, false);
                            J$.N(3725769, 'cmd', cmd, false, false, false);
                            J$.P(3725665, J$.R(3725649, 'obj', obj, false, false), 'a', J$.T(3725657, {}, 11, false));
                            J$.P(3725697, J$.G(3725681, J$.R(3725673, 'obj', obj, false, false), 'a'), 'b', J$.T(3725689, {}, 11, false));
                            var cmd = J$.W(3725737, 'cmd', J$.G(3725729, J$.G(3725721, J$.G(3725713, J$.R(3725705, 'obj', obj, false, false), 'a'), 'b'), 'multi_layer_hipar'), cmd, false, false);
                        } catch (J$e) {
                            J$.Ex(3727025, J$e);
                        } finally {
                            if (J$.Fr(3727033))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function mix_hipar(obj) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(3725929, arguments.callee, this, arguments);
                            arguments = J$.N(3725937, 'arguments', arguments, true, false, false);
                            obj = J$.N(3725945, 'obj', obj, true, false, false);
                            J$.N(3725953, 'cmd', cmd, false, false, false);
                            J$.P(3725793, J$.R(3725777, 'obj', obj, false, false), 'a', J$.T(3725785, {}, 11, false));
                            J$.P(3725825, J$.G(3725809, J$.R(3725801, 'obj', obj, false, false), 'a'), 'b', J$.T(3725817, {}, 11, false));
                            J$.P(3725865, J$.G(3725849, J$.G(3725841, J$.R(3725833, 'obj', obj, false, false), 'a'), 'b'), 'c', J$.T(3725857, {}, 11, false));
                            var cmd = J$.W(3725921, 'cmd', J$.G(3725913, J$.G(3725905, J$.G(3725897, J$.G(3725881, J$.R(3725873, 'obj', obj, false, false), 'a'), J$.T(3725889, 'b', 21, false)), 'c'), 'mix_hipar'), cmd, false, false);
                        } catch (J$e) {
                            J$.Ex(3727041, J$e);
                        } finally {
                            if (J$.Fr(3727049))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function proto_hipar(obj) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(3726241, arguments.callee, this, arguments);
                            arguments = J$.N(3726249, 'arguments', arguments, true, false, false);
                            obj = J$.N(3726257, 'obj', obj, true, false, false);
                            J$.N(3726265, 'person', person, false, false, false);
                            J$.N(3726273, 'overwritten_obj', overwritten_obj, false, false, false);
                            J$.N(3726281, 'cmd1', cmd1, false, false, false);
                            J$.N(3726289, 'cmd2', cmd2, false, false, false);
                            J$.N(3726297, 'cmd3', cmd3, false, false, false);
                            var person = J$.W(3726089, 'person', J$.T(3726081, function person(firstname, lastname) {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(3726049, arguments.callee, this, arguments);
                                            arguments = J$.N(3726057, 'arguments', arguments, true, false, false);
                                            firstname = J$.N(3726065, 'firstname', firstname, true, false, false);
                                            lastname = J$.N(3726073, 'lastname', lastname, true, false, false);
                                            J$.P(3725977, J$.R(3725961, 'this', this, false, false), 'normal_hipar', J$.R(3725969, 'firstname', firstname, false, false)), J$.P(3726041, J$.R(3725985, 'this', this, false, false), 'func_hipar', J$.T(3726033, function () {
                                                jalangiLabel4:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(3726017, arguments.callee, this, arguments);
                                                            arguments = J$.N(3726025, 'arguments', arguments, true, false, false);
                                                            J$.M(3726009, J$.I(typeof console === 'undefined' ? console = J$.R(3725993, 'console', undefined, true, true) : console = J$.R(3725993, 'console', console, true, true)), 'log', false)(J$.T(3726001, 'this is function', 21, false));
                                                        } catch (J$e) {
                                                            J$.Ex(3727057, J$e);
                                                        } finally {
                                                            if (J$.Fr(3727065))
                                                                continue jalangiLabel4;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(3727073, J$e);
                                        } finally {
                                            if (J$.Fr(3727081))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), person, false, false);
                            var overwritten_obj = J$.W(3726129, 'overwritten_obj', J$.F(3726121, J$.R(3726097, 'person', person, false, false), true)(J$.T(3726105, 'a', 21, false), J$.T(3726113, 'b', 21, false)), overwritten_obj, false, false);
                            J$.M(3726161, J$.I(typeof Object === 'undefined' ? Object = J$.R(3726137, 'Object', undefined, true, true) : Object = J$.R(3726137, 'Object', Object, true, true)), 'assign', false)(J$.R(3726145, 'overwritten_obj', overwritten_obj, false, false), J$.R(3726153, 'obj', obj, false, false));
                            var cmd1 = J$.W(3726185, 'cmd1', J$.G(3726177, J$.R(3726169, 'overwritten_obj', overwritten_obj, false, false), 'normal_hipar'), cmd1, false, false);
                            var cmd2 = J$.W(3726209, 'cmd2', J$.G(3726201, J$.R(3726193, 'overwritten_obj', overwritten_obj, false, false), 'constructor'), cmd2, false, false);
                            var cmd3 = J$.W(3726233, 'cmd3', J$.M(3726225, J$.R(3726217, 'overwritten_obj', overwritten_obj, false, false), 'func_hipar', false)(), cmd3, false, false);
                        } catch (J$e) {
                            J$.Ex(3727089, J$e);
                        } finally {
                            if (J$.Fr(3727097))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function child_hipar(obj) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(3726337, arguments.callee, this, arguments);
                            arguments = J$.N(3726345, 'arguments', arguments, true, false, false);
                            obj = J$.N(3726353, 'obj', obj, true, false, false);
                            J$.N(3726361, 'cmd', cmd, false, false, false);
                            var cmd = J$.W(3726329, 'cmd', J$.G(3726321, J$.G(3726313, J$.R(3726305, 'obj', obj, false, false), 'root'), 'child_hipar'), cmd, false, false);
                        } catch (J$e) {
                            J$.Ex(3727105, J$e);
                        } finally {
                            if (J$.Fr(3727113))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function time_hipar(obj) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(3726601, arguments.callee, this, arguments);
                            arguments = J$.N(3726609, 'arguments', arguments, true, false, false);
                            obj = J$.N(3726617, 'obj', obj, true, false, false);
                            J$.N(3726625, 'sleep', sleep, false, false, false);
                            var sleep = J$.W(3726497, 'sleep', J$.T(3726489, function sleep(milliseconds) {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(3726465, arguments.callee, this, arguments);
                                            arguments = J$.N(3726473, 'arguments', arguments, true, false, false);
                                            milliseconds = J$.N(3726481, 'milliseconds', milliseconds, true, false, false);
                                            return J$.Rt(3726457, J$.F(3726449, J$.I(typeof Promise === 'undefined' ? Promise = J$.R(3726369, 'Promise', undefined, true, true) : Promise = J$.R(3726369, 'Promise', Promise, true, true)), true)(J$.T(3726441, function (resolve) {
                                                jalangiLabel8:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(3726417, arguments.callee, this, arguments);
                                                            arguments = J$.N(3726425, 'arguments', arguments, true, false, false);
                                                            resolve = J$.N(3726433, 'resolve', resolve, true, false, false);
                                                            return J$.Rt(3726409, J$.F(3726401, J$.I(typeof setTimeout === 'undefined' ? setTimeout = J$.R(3726377, 'setTimeout', undefined, true, true) : setTimeout = J$.R(3726377, 'setTimeout', setTimeout, true, true)), false)(J$.R(3726385, 'resolve', resolve, false, false), J$.R(3726393, 'milliseconds', milliseconds, false, false)));
                                                        } catch (J$e) {
                                                            J$.Ex(3727121, J$e);
                                                        } finally {
                                                            if (J$.Fr(3727129))
                                                                continue jalangiLabel8;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)));
                                        } catch (J$e) {
                                            J$.Ex(3727137, J$e);
                                        } finally {
                                            if (J$.Fr(3727145))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), sleep, false, false);
                            J$.M(3726593, J$.F(3726521, J$.R(3726505, 'sleep', sleep, false, false), false)(J$.T(3726513, 1000, 22, false)), 'then', false)(J$.T(3726585, function () {
                                jalangiLabel10:
                                    while (true) {
                                        try {
                                            J$.Fe(3726561, arguments.callee, this, arguments);
                                            arguments = J$.N(3726569, 'arguments', arguments, true, false, false);
                                            J$.N(3726577, 'cmd', cmd, false, false, false);
                                            var cmd = J$.W(3726553, 'cmd', J$.G(3726545, J$.G(3726537, J$.R(3726529, 'obj', obj, false, false), 'root'), 'timing_hipar'), cmd, false, false);
                                        } catch (J$e) {
                                            J$.Ex(3727153, J$e);
                                        } finally {
                                            if (J$.Fr(3727161))
                                                continue jalangiLabel10;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(3727169, J$e);
                        } finally {
                            if (J$.Fr(3727177))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function func_var_nested_hipar(obj) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(3726793, arguments.callee, this, arguments);
                            function nest_func(n_obj) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(3726737, arguments.callee, this, arguments);
                                            arguments = J$.N(3726745, 'arguments', arguments, true, false, false);
                                            n_obj = J$.N(3726753, 'n_obj', n_obj, true, false, false);
                                            J$.N(3726761, 'cmd', cmd, false, false, false);
                                            var cmd = J$.W(3726729, 'cmd', J$.G(3726721, J$.R(3726713, 'n_obj', n_obj, false, false), 'func_func_hipar'), cmd, false, false);
                                        } catch (J$e) {
                                            J$.Ex(3727201, J$e);
                                        } finally {
                                            if (J$.Fr(3727209))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(3726801, 'arguments', arguments, true, false, false);
                            obj = J$.N(3726809, 'obj', obj, true, false, false);
                            nest_func = J$.N(3726825, 'nest_func', J$.T(3726817, nest_func, 12, false), true, false, false);
                            J$.F(3726705, J$.T(3726689, function (anon) {
                                jalangiLabel12:
                                    while (true) {
                                        try {
                                            J$.Fe(3726657, arguments.callee, this, arguments);
                                            arguments = J$.N(3726665, 'arguments', arguments, true, false, false);
                                            anon = J$.N(3726673, 'anon', anon, true, false, false);
                                            J$.N(3726681, 'cmd', cmd, false, false, false);
                                            var cmd = J$.W(3726649, 'cmd', J$.G(3726641, J$.R(3726633, 'anon', anon, false, false), 'func_anon_hipar'), cmd, false, false);
                                        } catch (J$e) {
                                            J$.Ex(3727185, J$e);
                                        } finally {
                                            if (J$.Fr(3727193))
                                                continue jalangiLabel12;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), false)(J$.R(3726697, 'obj', obj, false, false));
                            J$.F(3726785, J$.R(3726769, 'nest_func', nest_func, false, false), false)(J$.R(3726777, 'obj', obj, false, false));
                        } catch (J$e) {
                            J$.Ex(3727217, J$e);
                        } finally {
                            if (J$.Fr(3727225))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }
            do_test = J$.N(3726865, 'do_test', J$.T(3726857, do_test, 12, false), true, false, false);
            one_layer_hipar = J$.N(3726881, 'one_layer_hipar', J$.T(3726873, one_layer_hipar, 12, false), true, false, false);
            multi_layer_hipar = J$.N(3726897, 'multi_layer_hipar', J$.T(3726889, multi_layer_hipar, 12, false), true, false, false);
            mix_hipar = J$.N(3726913, 'mix_hipar', J$.T(3726905, mix_hipar, 12, false), true, false, false);
            proto_hipar = J$.N(3726929, 'proto_hipar', J$.T(3726921, proto_hipar, 12, false), true, false, false);
            child_hipar = J$.N(3726945, 'child_hipar', J$.T(3726937, child_hipar, 12, false), true, false, false);
            time_hipar = J$.N(3726961, 'time_hipar', J$.T(3726953, time_hipar, 12, false), true, false, false);
            func_var_nested_hipar = J$.N(3726977, 'func_var_nested_hipar', J$.T(3726969, func_var_nested_hipar, 12, false), true, false, false);
            J$.N(3726985, 'obj', obj, false, false, false);
            J$.T(3725305, 'use strict', 21, false);
            J$.P(3725537, J$.I(typeof module === 'undefined' ? module = J$.R(3725521, 'module', undefined, true, true) : module = J$.R(3725521, 'module', module, true, true)), 'exports', J$.R(3725529, 'do_test', do_test, false, true));
            var obj = J$.W(3726841, 'obj', J$.T(3726833, {}, 11, false), obj, false, true);
        } catch (J$e) {
            J$.Ex(3727233, J$e);
        } finally {
            if (J$.Sr(3727241))
                continue jalangiLabel15;
            else
                break jalangiLabel15;
        }
    }
// JALANGI DO NOT INSTRUMENT
