// regre_tester/index.js
J$.noInstrEval = false;
jalangiLabel16:
    while (true) {
        try {
            J$.Se(3917265, '/home/james/nodejs/HiPar/outputs/target_cache/TestRegression/node_modules/regre_tester/index.js');
            function do_test(user_input) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(3915857, arguments.callee, this, arguments);
                            arguments = J$.N(3915865, 'arguments', arguments, true, false, false);
                            user_input = J$.N(3915873, 'user_input', user_input, true, false, false);
                            J$.F(3915665, J$.R(3915649, 'one_layer_hipar', one_layer_hipar, false, true), false)(J$.R(3915657, 'user_input', user_input, false, false));
                            J$.F(3915689, J$.R(3915673, 'multi_layer_hipar', multi_layer_hipar, false, true), false)(J$.R(3915681, 'user_input', user_input, false, false));
                            J$.F(3915713, J$.R(3915697, 'mix_hipar', mix_hipar, false, true), false)(J$.R(3915705, 'user_input', user_input, false, false));
                            J$.F(3915737, J$.R(3915721, 'proto_hipar', proto_hipar, false, true), false)(J$.R(3915729, 'user_input', user_input, false, false));
                            J$.F(3915769, J$.R(3915745, 'child_hipar', child_hipar, false, true), false)(J$.T(3915761, { 'root': J$.R(3915753, 'user_input', user_input, false, false) }, 11, false));
                            J$.F(3915801, J$.R(3915777, 'time_hipar', time_hipar, false, true), false)(J$.T(3915793, { 'root': J$.R(3915785, 'user_input', user_input, false, false) }, 11, false));
                            J$.F(3915825, J$.R(3915809, 'func_var_nested_hipar', func_var_nested_hipar, false, true), false)(J$.R(3915817, 'user_input', user_input, false, false));
                            J$.F(3915849, J$.R(3915833, 'expr_func_nested_hipar', expr_func_nested_hipar, false, true), false)(J$.R(3915841, 'user_input', user_input, false, false));
                        } catch (J$e) {
                            J$.Ex(3917425, J$e);
                        } finally {
                            if (J$.Fr(3917433))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function one_layer_hipar(obj) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(3915977, arguments.callee, this, arguments);
                            arguments = J$.N(3915985, 'arguments', arguments, true, false, false);
                            obj = J$.N(3915993, 'obj', obj, true, false, false);
                            J$.N(3916001, 'b', b, false, false, false);
                            if (J$.C(206136, J$.G(3915921, J$.R(3915905, 'obj', obj, false, false), J$.T(3915913, 'explicit_one_layer_hipar', 21, false)))) {
                                return J$.Rt(3915929, undefined);
                            }
                            if (J$.C(206144, J$.T(3915937, false, 23, false))) {
                                var b = J$.W(3915969, 'b', J$.G(3915961, J$.R(3915945, 'obj', obj, false, false), J$.T(3915953, 'implicit_one_layer_hipar', 21, false)), b, false, false);
                            }
                        } catch (J$e) {
                            J$.Ex(3917441, J$e);
                        } finally {
                            if (J$.Fr(3917449))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function multi_layer_hipar(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(3916105, arguments.callee, this, arguments);
                            arguments = J$.N(3916113, 'arguments', arguments, true, false, false);
                            obj = J$.N(3916121, 'obj', obj, true, false, false);
                            J$.N(3916129, 'cmd', cmd, false, false, false);
                            J$.P(3916025, J$.R(3916009, 'obj', obj, false, false), 'a', J$.T(3916017, {}, 11, false));
                            J$.P(3916057, J$.G(3916041, J$.R(3916033, 'obj', obj, false, false), 'a'), 'b', J$.T(3916049, {}, 11, false));
                            var cmd = J$.W(3916097, 'cmd', J$.G(3916089, J$.G(3916081, J$.G(3916073, J$.R(3916065, 'obj', obj, false, false), 'a'), 'b'), 'multi_layer_hipar'), cmd, false, false);
                        } catch (J$e) {
                            J$.Ex(3917457, J$e);
                        } finally {
                            if (J$.Fr(3917465))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function mix_hipar(obj) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(3916289, arguments.callee, this, arguments);
                            arguments = J$.N(3916297, 'arguments', arguments, true, false, false);
                            obj = J$.N(3916305, 'obj', obj, true, false, false);
                            J$.N(3916313, 'cmd', cmd, false, false, false);
                            J$.P(3916153, J$.R(3916137, 'obj', obj, false, false), 'a', J$.T(3916145, {}, 11, false));
                            J$.P(3916185, J$.G(3916169, J$.R(3916161, 'obj', obj, false, false), 'a'), 'b', J$.T(3916177, {}, 11, false));
                            J$.P(3916225, J$.G(3916209, J$.G(3916201, J$.R(3916193, 'obj', obj, false, false), 'a'), 'b'), 'c', J$.T(3916217, {}, 11, false));
                            var cmd = J$.W(3916281, 'cmd', J$.G(3916273, J$.G(3916265, J$.G(3916257, J$.G(3916241, J$.R(3916233, 'obj', obj, false, false), 'a'), J$.T(3916249, 'b', 21, false)), 'c'), 'mix_hipar'), cmd, false, false);
                        } catch (J$e) {
                            J$.Ex(3917473, J$e);
                        } finally {
                            if (J$.Fr(3917481))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function proto_hipar(obj) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(3916585, arguments.callee, this, arguments);
                            arguments = J$.N(3916593, 'arguments', arguments, true, false, false);
                            obj = J$.N(3916601, 'obj', obj, true, false, false);
                            J$.N(3916609, 'person', person, false, false, false);
                            J$.N(3916617, 'overwritten_obj', overwritten_obj, false, false, false);
                            J$.N(3916625, 'cmd1', cmd1, false, false, false);
                            J$.N(3916633, 'cmd2', cmd2, false, false, false);
                            J$.N(3916641, 'cmd3', cmd3, false, false, false);
                            var person = J$.W(3916433, 'person', J$.T(3916425, function person(firstname, lastname) {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(3916393, arguments.callee, this, arguments);
                                            arguments = J$.N(3916401, 'arguments', arguments, true, false, false);
                                            firstname = J$.N(3916409, 'firstname', firstname, true, false, false);
                                            lastname = J$.N(3916417, 'lastname', lastname, true, false, false);
                                            J$.P(3916337, J$.R(3916321, 'this', this, false, false), 'normal_hipar', J$.R(3916329, 'firstname', firstname, false, false)), J$.P(3916385, J$.R(3916345, 'this', this, false, false), 'func_hipar', J$.T(3916377, function () {
                                                jalangiLabel4:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(3916361, arguments.callee, this, arguments);
                                                            arguments = J$.N(3916369, 'arguments', arguments, true, false, false);
                                                            return J$.Rt(3916353, undefined);
                                                        } catch (J$e) {
                                                            J$.Ex(3917489, J$e);
                                                        } finally {
                                                            if (J$.Fr(3917497))
                                                                continue jalangiLabel4;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(3917505, J$e);
                                        } finally {
                                            if (J$.Fr(3917513))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), person, false, false);
                            var overwritten_obj = J$.W(3916473, 'overwritten_obj', J$.F(3916465, J$.R(3916441, 'person', person, false, false), true)(J$.T(3916449, 'a', 21, false), J$.T(3916457, 'b', 21, false)), overwritten_obj, false, false);
                            J$.M(3916505, J$.I(typeof Object === 'undefined' ? Object = J$.R(3916481, 'Object', undefined, true, true) : Object = J$.R(3916481, 'Object', Object, true, true)), 'assign', false)(J$.R(3916489, 'overwritten_obj', overwritten_obj, false, false), J$.R(3916497, 'obj', obj, false, false));
                            var cmd1 = J$.W(3916529, 'cmd1', J$.G(3916521, J$.R(3916513, 'overwritten_obj', overwritten_obj, false, false), 'normal_hipar'), cmd1, false, false);
                            var cmd2 = J$.W(3916553, 'cmd2', J$.G(3916545, J$.R(3916537, 'overwritten_obj', overwritten_obj, false, false), 'constructor'), cmd2, false, false);
                            var cmd3 = J$.W(3916577, 'cmd3', J$.M(3916569, J$.R(3916561, 'overwritten_obj', overwritten_obj, false, false), 'func_hipar', false)(), cmd3, false, false);
                        } catch (J$e) {
                            J$.Ex(3917521, J$e);
                        } finally {
                            if (J$.Fr(3917529))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function child_hipar(obj) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(3916681, arguments.callee, this, arguments);
                            arguments = J$.N(3916689, 'arguments', arguments, true, false, false);
                            obj = J$.N(3916697, 'obj', obj, true, false, false);
                            J$.N(3916705, 'cmd', cmd, false, false, false);
                            var cmd = J$.W(3916673, 'cmd', J$.G(3916665, J$.G(3916657, J$.R(3916649, 'obj', obj, false, false), 'root'), 'child_hipar'), cmd, false, false);
                        } catch (J$e) {
                            J$.Ex(3917537, J$e);
                        } finally {
                            if (J$.Fr(3917545))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function time_hipar(obj) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(3916945, arguments.callee, this, arguments);
                            arguments = J$.N(3916953, 'arguments', arguments, true, false, false);
                            obj = J$.N(3916961, 'obj', obj, true, false, false);
                            J$.N(3916969, 'sleep', sleep, false, false, false);
                            var sleep = J$.W(3916841, 'sleep', J$.T(3916833, function sleep(milliseconds) {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(3916809, arguments.callee, this, arguments);
                                            arguments = J$.N(3916817, 'arguments', arguments, true, false, false);
                                            milliseconds = J$.N(3916825, 'milliseconds', milliseconds, true, false, false);
                                            return J$.Rt(3916801, J$.F(3916793, J$.I(typeof Promise === 'undefined' ? Promise = J$.R(3916713, 'Promise', undefined, true, true) : Promise = J$.R(3916713, 'Promise', Promise, true, true)), true)(J$.T(3916785, function (resolve) {
                                                jalangiLabel8:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(3916761, arguments.callee, this, arguments);
                                                            arguments = J$.N(3916769, 'arguments', arguments, true, false, false);
                                                            resolve = J$.N(3916777, 'resolve', resolve, true, false, false);
                                                            return J$.Rt(3916753, J$.F(3916745, J$.I(typeof setTimeout === 'undefined' ? setTimeout = J$.R(3916721, 'setTimeout', undefined, true, true) : setTimeout = J$.R(3916721, 'setTimeout', setTimeout, true, true)), false)(J$.R(3916729, 'resolve', resolve, false, false), J$.R(3916737, 'milliseconds', milliseconds, false, false)));
                                                        } catch (J$e) {
                                                            J$.Ex(3917553, J$e);
                                                        } finally {
                                                            if (J$.Fr(3917561))
                                                                continue jalangiLabel8;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)));
                                        } catch (J$e) {
                                            J$.Ex(3917569, J$e);
                                        } finally {
                                            if (J$.Fr(3917577))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), sleep, false, false);
                            J$.M(3916937, J$.F(3916865, J$.R(3916849, 'sleep', sleep, false, false), false)(J$.T(3916857, 1000, 22, false)), 'then', false)(J$.T(3916929, function () {
                                jalangiLabel10:
                                    while (true) {
                                        try {
                                            J$.Fe(3916905, arguments.callee, this, arguments);
                                            arguments = J$.N(3916913, 'arguments', arguments, true, false, false);
                                            J$.N(3916921, 'cmd', cmd, false, false, false);
                                            var cmd = J$.W(3916897, 'cmd', J$.G(3916889, J$.G(3916881, J$.R(3916873, 'obj', obj, false, false), 'root'), 'timing_hipar'), cmd, false, false);
                                        } catch (J$e) {
                                            J$.Ex(3917585, J$e);
                                        } finally {
                                            if (J$.Fr(3917593))
                                                continue jalangiLabel10;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(3917601, J$e);
                        } finally {
                            if (J$.Fr(3917609))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function func_var_nested_hipar(obj) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(3917137, arguments.callee, this, arguments);
                            function nest_func(n_obj) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(3917081, arguments.callee, this, arguments);
                                            arguments = J$.N(3917089, 'arguments', arguments, true, false, false);
                                            n_obj = J$.N(3917097, 'n_obj', n_obj, true, false, false);
                                            J$.N(3917105, 'cmd', cmd, false, false, false);
                                            var cmd = J$.W(3917073, 'cmd', J$.G(3917065, J$.R(3917057, 'n_obj', n_obj, false, false), 'func_func_hipar'), cmd, false, false);
                                        } catch (J$e) {
                                            J$.Ex(3917633, J$e);
                                        } finally {
                                            if (J$.Fr(3917641))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(3917145, 'arguments', arguments, true, false, false);
                            obj = J$.N(3917153, 'obj', obj, true, false, false);
                            nest_func = J$.N(3917169, 'nest_func', J$.T(3917161, nest_func, 12, false), true, false, false);
                            J$.F(3917049, J$.T(3917033, function (anon) {
                                jalangiLabel12:
                                    while (true) {
                                        try {
                                            J$.Fe(3917001, arguments.callee, this, arguments);
                                            arguments = J$.N(3917009, 'arguments', arguments, true, false, false);
                                            anon = J$.N(3917017, 'anon', anon, true, false, false);
                                            J$.N(3917025, 'cmd', cmd, false, false, false);
                                            var cmd = J$.W(3916993, 'cmd', J$.G(3916985, J$.R(3916977, 'anon', anon, false, false), 'func_anon_hipar'), cmd, false, false);
                                        } catch (J$e) {
                                            J$.Ex(3917617, J$e);
                                        } finally {
                                            if (J$.Fr(3917625))
                                                continue jalangiLabel12;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), false)(J$.R(3917041, 'obj', obj, false, false));
                            J$.F(3917129, J$.R(3917113, 'nest_func', nest_func, false, false), false)(J$.R(3917121, 'obj', obj, false, false));
                        } catch (J$e) {
                            J$.Ex(3917649, J$e);
                        } finally {
                            if (J$.Fr(3917657))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function expr_func_nested_hipar(obj) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(3917225, arguments.callee, this, arguments);
                            arguments = J$.N(3917233, 'arguments', arguments, true, false, false);
                            obj = J$.N(3917241, 'obj', obj, true, false, false);
                            J$.G(3917217, J$.M(3917209, J$.T(3917193, [J$.G(3917185, J$.R(3917177, 'obj', obj, false, false), 'expr_f_hipar')], 10, false), 'concat', false)(J$.T(3917201, '1', 21, false)), 'length');
                        } catch (J$e) {
                            J$.Ex(3917665, J$e);
                        } finally {
                            if (J$.Fr(3917673))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            do_test = J$.N(3917281, 'do_test', J$.T(3917273, do_test, 12, false), true, false, false);
            one_layer_hipar = J$.N(3917297, 'one_layer_hipar', J$.T(3917289, one_layer_hipar, 12, false), true, false, false);
            multi_layer_hipar = J$.N(3917313, 'multi_layer_hipar', J$.T(3917305, multi_layer_hipar, 12, false), true, false, false);
            mix_hipar = J$.N(3917329, 'mix_hipar', J$.T(3917321, mix_hipar, 12, false), true, false, false);
            proto_hipar = J$.N(3917345, 'proto_hipar', J$.T(3917337, proto_hipar, 12, false), true, false, false);
            child_hipar = J$.N(3917361, 'child_hipar', J$.T(3917353, child_hipar, 12, false), true, false, false);
            time_hipar = J$.N(3917377, 'time_hipar', J$.T(3917369, time_hipar, 12, false), true, false, false);
            func_var_nested_hipar = J$.N(3917393, 'func_var_nested_hipar', J$.T(3917385, func_var_nested_hipar, 12, false), true, false, false);
            expr_func_nested_hipar = J$.N(3917409, 'expr_func_nested_hipar', J$.T(3917401, expr_func_nested_hipar, 12, false), true, false, false);
            J$.N(3917417, 'obj', obj, false, false, false);
            J$.T(3915641, 'use strict', 21, false);
            J$.P(3915897, J$.I(typeof module === 'undefined' ? module = J$.R(3915881, 'module', undefined, true, true) : module = J$.R(3915881, 'module', module, true, true)), 'exports', J$.R(3915889, 'do_test', do_test, false, true));
            var obj = J$.W(3917257, 'obj', J$.T(3917249, {}, 11, false), obj, false, true);
        } catch (J$e) {
            J$.Ex(3917681, J$e);
        } finally {
            if (J$.Sr(3917689))
                continue jalangiLabel16;
            else
                break jalangiLabel16;
        }
    }
// JALANGI DO NOT INSTRUMENT
