// mongodb/lib/core/connection/apm.js
J$.noInstrEval = false;
jalangiLabel15:
    while (true) {
        try {
            J$.Se(663257, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/connection/apm.js');
            function _classCallCheck(instance, Constructor) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(659665, arguments.callee, this, arguments);
                            arguments = J$.N(659673, 'arguments', arguments, true, false, false);
                            instance = J$.N(659681, 'instance', instance, true, false, false);
                            Constructor = J$.N(659689, 'Constructor', Constructor, true, false, false);
                            if (J$.C(38080, J$.U(81162, '!', J$.B(81154, 'instanceof', J$.R(659625, 'instance', instance, false, false), J$.R(659633, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(659657, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(659641, 'TypeError', undefined, true, true) : TypeError = J$.R(659641, 'TypeError', TypeError, true, true)), true)(J$.T(659649, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(663433, J$e);
                        } finally {
                            if (J$.Fr(663441))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _classCallCheck = J$.N(663273, '_classCallCheck', J$.T(663265, _classCallCheck, 12, false), true, false, false);
            J$.N(663281, 'Msg', Msg, false, false, false);
            J$.N(663289, 'KillCursor', KillCursor, false, false, false);
            J$.N(663297, 'GetMore', GetMore, false, false, false);
            J$.N(663305, 'calculateDurationInMs', calculateDurationInMs, false, false, false);
            J$.N(663313, 'SENSITIVE_COMMANDS', SENSITIVE_COMMANDS, false, false, false);
            J$.N(663321, 'extractCommandName', extractCommandName, false, false, false);
            J$.N(663329, 'namespace', namespace, false, false, false);
            J$.N(663337, 'databaseName', databaseName, false, false, false);
            J$.N(663345, 'collectionName', collectionName, false, false, false);
            J$.N(663353, 'generateConnectionId', generateConnectionId, false, false, false);
            J$.N(663361, 'maybeRedact', maybeRedact, false, false, false);
            J$.N(663369, 'LEGACY_FIND_QUERY_MAP', LEGACY_FIND_QUERY_MAP, false, false, false);
            J$.N(663377, 'LEGACY_FIND_OPTIONS_MAP', LEGACY_FIND_OPTIONS_MAP, false, false, false);
            J$.N(663385, 'OP_QUERY_KEYS', OP_QUERY_KEYS, false, false, false);
            J$.N(663393, 'extractCommand', extractCommand, false, false, false);
            J$.N(663401, 'extractReply', extractReply, false, false, false);
            J$.N(663409, 'CommandStartedEvent', CommandStartedEvent, false, false, false);
            J$.N(663417, 'CommandSucceededEvent', CommandSucceededEvent, false, false, false);
            J$.N(663425, 'CommandFailedEvent', CommandFailedEvent, false, false, false);
            J$.T(659617, 'use strict', 21, false);
            var Msg = J$.W(659729, 'Msg', J$.G(659721, J$.F(659713, J$.I(typeof require === 'undefined' ? require = J$.R(659697, 'require', undefined, true, true) : require = J$.R(659697, 'require', require, true, true)), false)(J$.T(659705, '../connection/msg', 21, false)), 'Msg'), Msg, false, true);
            var KillCursor = J$.W(659769, 'KillCursor', J$.G(659761, J$.F(659753, J$.I(typeof require === 'undefined' ? require = J$.R(659737, 'require', undefined, true, true) : require = J$.R(659737, 'require', require, true, true)), false)(J$.T(659745, '../connection/commands', 21, false)), 'KillCursor'), KillCursor, false, true);
            var GetMore = J$.W(659809, 'GetMore', J$.G(659801, J$.F(659793, J$.I(typeof require === 'undefined' ? require = J$.R(659777, 'require', undefined, true, true) : require = J$.R(659777, 'require', require, true, true)), false)(J$.T(659785, '../connection/commands', 21, false)), 'GetMore'), GetMore, false, true);
            var calculateDurationInMs = J$.W(659849, 'calculateDurationInMs', J$.G(659841, J$.F(659833, J$.I(typeof require === 'undefined' ? require = J$.R(659817, 'require', undefined, true, true) : require = J$.R(659817, 'require', require, true, true)), false)(J$.T(659825, '../utils', 21, false)), 'calculateDurationInMs'), calculateDurationInMs, false, true);
            var SENSITIVE_COMMANDS = J$.W(659953, 'SENSITIVE_COMMANDS', J$.F(659945, J$.I(typeof Set === 'undefined' ? Set = J$.R(659857, 'Set', undefined, true, true) : Set = J$.R(659857, 'Set', Set, true, true)), true)(J$.T(659937, [
                J$.T(659865, 'authenticate', 21, false),
                J$.T(659873, 'saslStart', 21, false),
                J$.T(659881, 'saslContinue', 21, false),
                J$.T(659889, 'getnonce', 21, false),
                J$.T(659897, 'createUser', 21, false),
                J$.T(659905, 'updateUser', 21, false),
                J$.T(659913, 'copydbgetnonce', 21, false),
                J$.T(659921, 'copydbsaslstart', 21, false),
                J$.T(659929, 'copydb', 21, false)
            ], 10, false)), SENSITIVE_COMMANDS, false, true);
            var extractCommandName = J$.W(660041, 'extractCommandName', J$.T(660033, function extractCommandName(commandDoc) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(660009, arguments.callee, this, arguments);
                            arguments = J$.N(660017, 'arguments', arguments, true, false, false);
                            commandDoc = J$.N(660025, 'commandDoc', commandDoc, true, false, false);
                            return J$.Rt(660001, J$.G(659993, J$.M(659977, J$.I(typeof Object === 'undefined' ? Object = J$.R(659961, 'Object', undefined, true, true) : Object = J$.R(659961, 'Object', Object, true, true)), 'keys', false)(J$.R(659969, 'commandDoc', commandDoc, false, false)), J$.T(659985, 0, 22, false)));
                        } catch (J$e) {
                            J$.Ex(663449, J$e);
                        } finally {
                            if (J$.Fr(663457))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), extractCommandName, false, true);
            var namespace = J$.W(660105, 'namespace', J$.T(660097, function namespace(command) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(660073, arguments.callee, this, arguments);
                            arguments = J$.N(660081, 'arguments', arguments, true, false, false);
                            command = J$.N(660089, 'command', command, true, false, false);
                            return J$.Rt(660065, J$.G(660057, J$.R(660049, 'command', command, false, false), 'ns'));
                        } catch (J$e) {
                            J$.Ex(663465, J$e);
                        } finally {
                            if (J$.Fr(663473))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), namespace, false, true);
            var databaseName = J$.W(660201, 'databaseName', J$.T(660193, function databaseName(command) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(660169, arguments.callee, this, arguments);
                            arguments = J$.N(660177, 'arguments', arguments, true, false, false);
                            command = J$.N(660185, 'command', command, true, false, false);
                            return J$.Rt(660161, J$.G(660153, J$.M(660137, J$.G(660121, J$.R(660113, 'command', command, false, false), 'ns'), 'split', false)(J$.T(660129, '.', 21, false)), J$.T(660145, 0, 22, false)));
                        } catch (J$e) {
                            J$.Ex(663481, J$e);
                        } finally {
                            if (J$.Fr(663489))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), databaseName, false, true);
            var collectionName = J$.W(660297, 'collectionName', J$.T(660289, function collectionName(command) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(660265, arguments.callee, this, arguments);
                            arguments = J$.N(660273, 'arguments', arguments, true, false, false);
                            command = J$.N(660281, 'command', command, true, false, false);
                            return J$.Rt(660257, J$.G(660249, J$.M(660233, J$.G(660217, J$.R(660209, 'command', command, false, false), 'ns'), 'split', false)(J$.T(660225, '.', 21, false)), J$.T(660241, 1, 22, false)));
                        } catch (J$e) {
                            J$.Ex(663497, J$e);
                        } finally {
                            if (J$.Fr(663505))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), collectionName, false, true);
            var generateConnectionId = J$.W(660425, 'generateConnectionId', J$.T(660417, function generateConnectionId(pool) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(660393, arguments.callee, this, arguments);
                            arguments = J$.N(660401, 'arguments', arguments, true, false, false);
                            pool = J$.N(660409, 'pool', pool, true, false, false);
                            return J$.Rt(660385, J$.M(660377, J$.M(660345, J$.T(660305, '', 21, false), 'concat', false)(J$.G(660329, J$.G(660321, J$.R(660313, 'pool', pool, false, false), 'options'), 'host'), J$.T(660337, ':', 21, false)), 'concat', false)(J$.G(660369, J$.G(660361, J$.R(660353, 'pool', pool, false, false), 'options'), 'port')));
                        } catch (J$e) {
                            J$.Ex(663513, J$e);
                        } finally {
                            if (J$.Fr(663521))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), generateConnectionId, false, true);
            var maybeRedact = J$.W(660521, 'maybeRedact', J$.T(660513, function maybeRedact(commandName, result) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(660481, arguments.callee, this, arguments);
                            arguments = J$.N(660489, 'arguments', arguments, true, false, false);
                            commandName = J$.N(660497, 'commandName', commandName, true, false, false);
                            result = J$.N(660505, 'result', result, true, false, false);
                            return J$.Rt(660473, J$.C(38088, J$.M(660449, J$.R(660433, 'SENSITIVE_COMMANDS', SENSITIVE_COMMANDS, false, true), 'has', false)(J$.R(660441, 'commandName', commandName, false, false))) ? J$.T(660457, {}, 11, false) : J$.R(660465, 'result', result, false, false));
                        } catch (J$e) {
                            J$.Ex(663529, J$e);
                        } finally {
                            if (J$.Fr(663537))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), maybeRedact, false, true);
            var LEGACY_FIND_QUERY_MAP = J$.W(660625, 'LEGACY_FIND_QUERY_MAP', J$.T(660617, {
                $query: J$.T(660529, 'filter', 21, false),
                $orderby: J$.T(660537, 'sort', 21, false),
                $hint: J$.T(660545, 'hint', 21, false),
                $comment: J$.T(660553, 'comment', 21, false),
                $maxScan: J$.T(660561, 'maxScan', 21, false),
                $max: J$.T(660569, 'max', 21, false),
                $min: J$.T(660577, 'min', 21, false),
                $returnKey: J$.T(660585, 'returnKey', 21, false),
                $showDiskLoc: J$.T(660593, 'showRecordId', 21, false),
                $maxTimeMS: J$.T(660601, 'maxTimeMS', 21, false),
                $snapshot: J$.T(660609, 'snapshot', 21, false)
            }, 11, false), LEGACY_FIND_QUERY_MAP, false, true);
            var LEGACY_FIND_OPTIONS_MAP = J$.W(660665, 'LEGACY_FIND_OPTIONS_MAP', J$.T(660657, {
                numberToSkip: J$.T(660633, 'skip', 21, false),
                numberToReturn: J$.T(660641, 'batchSize', 21, false),
                returnFieldsSelector: J$.T(660649, 'projection', 21, false)
            }, 11, false), LEGACY_FIND_OPTIONS_MAP, false, true);
            var OP_QUERY_KEYS = J$.W(660729, 'OP_QUERY_KEYS', J$.T(660721, [
                J$.T(660673, 'tailable', 21, false),
                J$.T(660681, 'oplogReplay', 21, false),
                J$.T(660689, 'noCursorTimeout', 21, false),
                J$.T(660697, 'awaitData', 21, false),
                J$.T(660705, 'partial', 21, false),
                J$.T(660713, 'exhaust', 21, false)
            ], 10, false), OP_QUERY_KEYS, false, true);
            var extractCommand = J$.W(661761, 'extractCommand', J$.T(661753, function extractCommand(command) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(661721, arguments.callee, this, arguments);
                            arguments = J$.N(661729, 'arguments', arguments, true, false, false);
                            command = J$.N(661737, 'command', command, true, false, false);
                            J$.N(661745, 'result', result, false, false, false);
                            if (J$.C(38096, J$.B(81170, 'instanceof', J$.R(660737, 'command', command, false, false), J$.R(660745, 'GetMore', GetMore, false, true)))) {
                                return J$.Rt(660817, J$.T(660809, {
                                    getMore: J$.G(660761, J$.R(660753, 'command', command, false, false), 'cursorId'),
                                    collection: J$.F(660785, J$.R(660769, 'collectionName', collectionName, false, true), false)(J$.R(660777, 'command', command, false, false)),
                                    batchSize: J$.G(660801, J$.R(660793, 'command', command, false, false), 'numberToReturn')
                                }, 11, false));
                            }
                            if (J$.C(38104, J$.B(81178, 'instanceof', J$.R(660825, 'command', command, false, false), J$.R(660833, 'KillCursor', KillCursor, false, true)))) {
                                return J$.Rt(660889, J$.T(660881, {
                                    killCursors: J$.F(660857, J$.R(660841, 'collectionName', collectionName, false, true), false)(J$.R(660849, 'command', command, false, false)),
                                    cursors: J$.G(660873, J$.R(660865, 'command', command, false, false), 'cursorIds')
                                }, 11, false));
                            }
                            if (J$.C(38112, J$.B(81186, 'instanceof', J$.R(660897, 'command', command, false, false), J$.R(660905, 'Msg', Msg, false, true)))) {
                                return J$.Rt(660929, J$.G(660921, J$.R(660913, 'command', command, false, false), 'command'));
                            }
                            if (J$.C(38176, J$.C(38120, J$.G(660945, J$.R(660937, 'command', command, false, false), 'query')) ? J$.G(660969, J$.G(660961, J$.R(660953, 'command', command, false, false), 'query'), '$query') : J$._())) {
                                var result;
                                if (J$.C(38136, J$.B(81194, '===', J$.G(660985, J$.R(660977, 'command', command, false, false), 'ns'), J$.T(660993, 'admin.$cmd', 21, false)))) {
                                    result = J$.W(661049, 'result', J$.M(661041, J$.I(typeof Object === 'undefined' ? Object = J$.R(661001, 'Object', undefined, true, true) : Object = J$.R(661001, 'Object', Object, true, true)), 'assign', false)(J$.T(661009, {}, 11, false), J$.G(661033, J$.G(661025, J$.R(661017, 'command', command, false, false), 'query'), '$query')), result, false, false);
                                } else {
                                    result = J$.W(661089, 'result', J$.T(661081, { find: J$.F(661073, J$.R(661057, 'collectionName', collectionName, false, true), false)(J$.R(661065, 'command', command, false, false)) }, 11, false), result, false, false);
                                    J$.M(661265, J$.M(661113, J$.I(typeof Object === 'undefined' ? Object = J$.R(661097, 'Object', undefined, true, true) : Object = J$.R(661097, 'Object', Object, true, true)), 'keys', false)(J$.R(661105, 'LEGACY_FIND_QUERY_MAP', LEGACY_FIND_QUERY_MAP, false, true)), 'forEach', false)(J$.T(661257, function (key) {
                                        jalangiLabel7:
                                            while (true) {
                                                try {
                                                    J$.Fe(661233, arguments.callee, this, arguments);
                                                    arguments = J$.N(661241, 'arguments', arguments, true, false, false);
                                                    key = J$.N(661249, 'key', key, true, false, false);
                                                    if (J$.C(38128, J$.B(81210, '!==', J$.U(81202, 'typeof', J$.G(661145, J$.G(661129, J$.R(661121, 'command', command, false, false), 'query'), J$.R(661137, 'key', key, false, false))), J$.T(661153, 'undefined', 21, false))))
                                                        J$.P(661225, J$.R(661161, 'result', result, false, false), J$.G(661185, J$.R(661169, 'LEGACY_FIND_QUERY_MAP', LEGACY_FIND_QUERY_MAP, false, true), J$.R(661177, 'key', key, false, false)), J$.G(661217, J$.G(661201, J$.R(661193, 'command', command, false, false), 'query'), J$.R(661209, 'key', key, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(663545, J$e);
                                                } finally {
                                                    if (J$.Fr(663553))
                                                        continue jalangiLabel7;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false));
                                }
                                J$.M(661425, J$.M(661289, J$.I(typeof Object === 'undefined' ? Object = J$.R(661273, 'Object', undefined, true, true) : Object = J$.R(661273, 'Object', Object, true, true)), 'keys', false)(J$.R(661281, 'LEGACY_FIND_OPTIONS_MAP', LEGACY_FIND_OPTIONS_MAP, false, true)), 'forEach', false)(J$.T(661417, function (key) {
                                    jalangiLabel8:
                                        while (true) {
                                            try {
                                                J$.Fe(661393, arguments.callee, this, arguments);
                                                arguments = J$.N(661401, 'arguments', arguments, true, false, false);
                                                key = J$.N(661409, 'key', key, true, false, false);
                                                if (J$.C(38144, J$.B(81226, '!==', J$.U(81218, 'typeof', J$.G(661313, J$.R(661297, 'command', command, false, false), J$.R(661305, 'key', key, false, false))), J$.T(661321, 'undefined', 21, false))))
                                                    J$.P(661385, J$.R(661329, 'result', result, false, false), J$.G(661353, J$.R(661337, 'LEGACY_FIND_OPTIONS_MAP', LEGACY_FIND_OPTIONS_MAP, false, true), J$.R(661345, 'key', key, false, false)), J$.G(661377, J$.R(661361, 'command', command, false, false), J$.R(661369, 'key', key, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(663561, J$e);
                                            } finally {
                                                if (J$.Fr(663569))
                                                    continue jalangiLabel8;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                                J$.M(661545, J$.R(661433, 'OP_QUERY_KEYS', OP_QUERY_KEYS, false, true), 'forEach', false)(J$.T(661537, function (key) {
                                    jalangiLabel9:
                                        while (true) {
                                            try {
                                                J$.Fe(661513, arguments.callee, this, arguments);
                                                arguments = J$.N(661521, 'arguments', arguments, true, false, false);
                                                key = J$.N(661529, 'key', key, true, false, false);
                                                if (J$.C(38152, J$.G(661457, J$.R(661441, 'command', command, false, false), J$.R(661449, 'key', key, false, false))))
                                                    J$.P(661505, J$.R(661465, 'result', result, false, false), J$.R(661473, 'key', key, false, false), J$.G(661497, J$.R(661481, 'command', command, false, false), J$.R(661489, 'key', key, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(663577, J$e);
                                            } finally {
                                                if (J$.Fr(663585))
                                                    continue jalangiLabel9;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                                if (J$.C(38160, J$.B(81242, '!==', J$.U(81234, 'typeof', J$.G(661561, J$.R(661553, 'command', command, false, false), 'pre32Limit')), J$.T(661569, 'undefined', 21, false)))) {
                                    J$.P(661601, J$.R(661577, 'result', result, false, false), 'limit', J$.G(661593, J$.R(661585, 'command', command, false, false), 'pre32Limit'));
                                }
                                if (J$.C(38168, J$.G(661625, J$.G(661617, J$.R(661609, 'command', command, false, false), 'query'), '$explain'))) {
                                    return J$.Rt(661649, J$.T(661641, { explain: J$.R(661633, 'result', result, false, false) }, 11, false));
                                }
                                return J$.Rt(661665, J$.R(661657, 'result', result, false, false));
                            }
                            return J$.Rt(661713, J$.C(38184, J$.G(661681, J$.R(661673, 'command', command, false, false), 'query')) ? J$.G(661697, J$.R(661689, 'command', command, false, false), 'query') : J$.R(661705, 'command', command, false, false));
                        } catch (J$e) {
                            J$.Ex(663593, J$e);
                        } finally {
                            if (J$.Fr(663601))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), extractCommand, false, true);
            var extractReply = J$.W(662193, 'extractReply', J$.T(662185, function extractReply(command, reply) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(662153, arguments.callee, this, arguments);
                            arguments = J$.N(662161, 'arguments', arguments, true, false, false);
                            command = J$.N(662169, 'command', command, true, false, false);
                            reply = J$.N(662177, 'reply', reply, true, false, false);
                            if (J$.C(38192, J$.B(81250, 'instanceof', J$.R(661769, 'command', command, false, false), J$.R(661777, 'GetMore', GetMore, false, true)))) {
                                return J$.Rt(661881, J$.T(661873, {
                                    ok: J$.T(661785, 1, 22, false),
                                    cursor: J$.T(661865, {
                                        id: J$.G(661809, J$.G(661801, J$.R(661793, 'reply', reply, false, false), 'message'), 'cursorId'),
                                        ns: J$.F(661833, J$.R(661817, 'namespace', namespace, false, true), false)(J$.R(661825, 'command', command, false, false)),
                                        nextBatch: J$.G(661857, J$.G(661849, J$.R(661841, 'reply', reply, false, false), 'message'), 'documents')
                                    }, 11, false)
                                }, 11, false));
                            }
                            if (J$.C(38200, J$.B(81258, 'instanceof', J$.R(661889, 'command', command, false, false), J$.R(661897, 'KillCursor', KillCursor, false, true)))) {
                                return J$.Rt(661937, J$.T(661929, {
                                    ok: J$.T(661905, 1, 22, false),
                                    cursorsUnknown: J$.G(661921, J$.R(661913, 'command', command, false, false), 'cursorIds')
                                }, 11, false));
                            }
                            if (J$.C(38216, J$.C(38208, J$.G(661953, J$.R(661945, 'command', command, false, false), 'query')) ? J$.B(81274, '!==', J$.U(81266, 'typeof', J$.G(661977, J$.G(661969, J$.R(661961, 'command', command, false, false), 'query'), '$query')), J$.T(661985, 'undefined', 21, false)) : J$._())) {
                                return J$.Rt(662089, J$.T(662081, {
                                    ok: J$.T(661993, 1, 22, false),
                                    cursor: J$.T(662073, {
                                        id: J$.G(662017, J$.G(662009, J$.R(662001, 'reply', reply, false, false), 'message'), 'cursorId'),
                                        ns: J$.F(662041, J$.R(662025, 'namespace', namespace, false, true), false)(J$.R(662033, 'command', command, false, false)),
                                        firstBatch: J$.G(662065, J$.G(662057, J$.R(662049, 'reply', reply, false, false), 'message'), 'documents')
                                    }, 11, false)
                                }, 11, false));
                            }
                            if (J$.C(38224, J$.B(81282, '===', J$.R(662097, 'reply', reply, false, false), J$.T(662105, null, 25, false))))
                                return J$.Rt(662121, J$.R(662113, 'reply', reply, false, false));
                            return J$.Rt(662145, J$.G(662137, J$.R(662129, 'reply', reply, false, false), 'result'));
                        } catch (J$e) {
                            J$.Ex(663609, J$e);
                        } finally {
                            if (J$.Fr(663617))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), extractReply, false, true);
            var CommandStartedEvent = J$.W(662553, 'CommandStartedEvent', J$.T(662545, function CommandStartedEvent(pool, command) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(662497, arguments.callee, this, arguments);
                            arguments = J$.N(662505, 'arguments', arguments, true, false, false);
                            pool = J$.N(662513, 'pool', pool, true, false, false);
                            command = J$.N(662521, 'command', command, true, false, false);
                            J$.N(662529, 'cmd', cmd, false, false, false);
                            J$.N(662537, 'commandName', commandName, false, false, false);
                            J$.F(662225, J$.R(662201, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(662209, 'this', this, false, false), J$.R(662217, 'CommandStartedEvent', CommandStartedEvent, false, false));
                            var cmd = J$.W(662257, 'cmd', J$.F(662249, J$.R(662233, 'extractCommand', extractCommand, false, true), false)(J$.R(662241, 'command', command, false, false)), cmd, false, false);
                            var commandName = J$.W(662289, 'commandName', J$.F(662281, J$.R(662265, 'extractCommandName', extractCommandName, false, true), false)(J$.R(662273, 'cmd', cmd, false, false)), commandName, false, false);
                            if (J$.C(38232, J$.M(662313, J$.R(662297, 'SENSITIVE_COMMANDS', SENSITIVE_COMMANDS, false, true), 'has', false)(J$.R(662305, 'commandName', commandName, false, false)))) {
                                J$.P(662337, J$.R(662321, 'this', this, false, false), 'commandObj', J$.T(662329, {}, 11, false));
                                J$.P(662377, J$.G(662353, J$.R(662345, 'this', this, false, false), 'commandObj'), J$.R(662361, 'commandName', commandName, false, false), J$.T(662369, true, 23, false));
                            }
                            J$.M(662489, J$.I(typeof Object === 'undefined' ? Object = J$.R(662385, 'Object', undefined, true, true) : Object = J$.R(662385, 'Object', Object, true, true)), 'assign', false)(J$.R(662393, 'this', this, false, false), J$.T(662481, {
                                command: J$.R(662401, 'cmd', cmd, false, false),
                                databaseName: J$.F(662425, J$.R(662409, 'databaseName', databaseName, false, true), false)(J$.R(662417, 'command', command, false, false)),
                                commandName: J$.R(662433, 'commandName', commandName, false, false),
                                requestId: J$.G(662449, J$.R(662441, 'command', command, false, false), 'requestId'),
                                connectionId: J$.F(662473, J$.R(662457, 'generateConnectionId', generateConnectionId, false, true), false)(J$.R(662465, 'pool', pool, false, false))
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(663625, J$e);
                        } finally {
                            if (J$.Fr(663633))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), CommandStartedEvent, false, true);
            var CommandSucceededEvent = J$.W(662889, 'CommandSucceededEvent', J$.T(662881, function CommandSucceededEvent(pool, command, reply, started) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(662817, arguments.callee, this, arguments);
                            arguments = J$.N(662825, 'arguments', arguments, true, false, false);
                            pool = J$.N(662833, 'pool', pool, true, false, false);
                            command = J$.N(662841, 'command', command, true, false, false);
                            reply = J$.N(662849, 'reply', reply, true, false, false);
                            started = J$.N(662857, 'started', started, true, false, false);
                            J$.N(662865, 'cmd', cmd, false, false, false);
                            J$.N(662873, 'commandName', commandName, false, false, false);
                            J$.F(662585, J$.R(662561, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(662569, 'this', this, false, false), J$.R(662577, 'CommandSucceededEvent', CommandSucceededEvent, false, false));
                            var cmd = J$.W(662617, 'cmd', J$.F(662609, J$.R(662593, 'extractCommand', extractCommand, false, true), false)(J$.R(662601, 'command', command, false, false)), cmd, false, false);
                            var commandName = J$.W(662649, 'commandName', J$.F(662641, J$.R(662625, 'extractCommandName', extractCommandName, false, true), false)(J$.R(662633, 'cmd', cmd, false, false)), commandName, false, false);
                            J$.M(662809, J$.I(typeof Object === 'undefined' ? Object = J$.R(662657, 'Object', undefined, true, true) : Object = J$.R(662657, 'Object', Object, true, true)), 'assign', false)(J$.R(662665, 'this', this, false, false), J$.T(662801, {
                                duration: J$.F(662689, J$.R(662673, 'calculateDurationInMs', calculateDurationInMs, false, true), false)(J$.R(662681, 'started', started, false, false)),
                                commandName: J$.R(662697, 'commandName', commandName, false, false),
                                reply: J$.F(662753, J$.R(662705, 'maybeRedact', maybeRedact, false, true), false)(J$.R(662713, 'commandName', commandName, false, false), J$.F(662745, J$.R(662721, 'extractReply', extractReply, false, true), false)(J$.R(662729, 'command', command, false, false), J$.R(662737, 'reply', reply, false, false))),
                                requestId: J$.G(662769, J$.R(662761, 'command', command, false, false), 'requestId'),
                                connectionId: J$.F(662793, J$.R(662777, 'generateConnectionId', generateConnectionId, false, true), false)(J$.R(662785, 'pool', pool, false, false))
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(663641, J$e);
                        } finally {
                            if (J$.Fr(663649))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), CommandSucceededEvent, false, true);
            var CommandFailedEvent = J$.W(663201, 'CommandFailedEvent', J$.T(663193, function CommandFailedEvent(pool, command, error, started) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(663129, arguments.callee, this, arguments);
                            arguments = J$.N(663137, 'arguments', arguments, true, false, false);
                            pool = J$.N(663145, 'pool', pool, true, false, false);
                            command = J$.N(663153, 'command', command, true, false, false);
                            error = J$.N(663161, 'error', error, true, false, false);
                            started = J$.N(663169, 'started', started, true, false, false);
                            J$.N(663177, 'cmd', cmd, false, false, false);
                            J$.N(663185, 'commandName', commandName, false, false, false);
                            J$.F(662921, J$.R(662897, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(662905, 'this', this, false, false), J$.R(662913, 'CommandFailedEvent', CommandFailedEvent, false, false));
                            var cmd = J$.W(662953, 'cmd', J$.F(662945, J$.R(662929, 'extractCommand', extractCommand, false, true), false)(J$.R(662937, 'command', command, false, false)), cmd, false, false);
                            var commandName = J$.W(662985, 'commandName', J$.F(662977, J$.R(662961, 'extractCommandName', extractCommandName, false, true), false)(J$.R(662969, 'cmd', cmd, false, false)), commandName, false, false);
                            J$.M(663121, J$.I(typeof Object === 'undefined' ? Object = J$.R(662993, 'Object', undefined, true, true) : Object = J$.R(662993, 'Object', Object, true, true)), 'assign', false)(J$.R(663001, 'this', this, false, false), J$.T(663113, {
                                duration: J$.F(663025, J$.R(663009, 'calculateDurationInMs', calculateDurationInMs, false, true), false)(J$.R(663017, 'started', started, false, false)),
                                commandName: J$.R(663033, 'commandName', commandName, false, false),
                                failure: J$.F(663065, J$.R(663041, 'maybeRedact', maybeRedact, false, true), false)(J$.R(663049, 'commandName', commandName, false, false), J$.R(663057, 'error', error, false, false)),
                                requestId: J$.G(663081, J$.R(663073, 'command', command, false, false), 'requestId'),
                                connectionId: J$.F(663105, J$.R(663089, 'generateConnectionId', generateConnectionId, false, true), false)(J$.R(663097, 'pool', pool, false, false))
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(663657, J$e);
                        } finally {
                            if (J$.Fr(663665))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), CommandFailedEvent, false, true);
            J$.P(663249, J$.I(typeof module === 'undefined' ? module = J$.R(663209, 'module', undefined, true, true) : module = J$.R(663209, 'module', module, true, true)), 'exports', J$.T(663241, {
                CommandStartedEvent: J$.R(663217, 'CommandStartedEvent', CommandStartedEvent, false, true),
                CommandSucceededEvent: J$.R(663225, 'CommandSucceededEvent', CommandSucceededEvent, false, true),
                CommandFailedEvent: J$.R(663233, 'CommandFailedEvent', CommandFailedEvent, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(663673, J$e);
        } finally {
            if (J$.Sr(663681))
                continue jalangiLabel15;
            else
                break jalangiLabel15;
        }
    }
// JALANGI DO NOT INSTRUMENT
