// mongodb/lib/core/connection/logger.js
J$.noInstrEval = false;
jalangiLabel15:
    while (true) {
        try {
            J$.Se(701265, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/core/connection/logger.js');
            J$.N(701273, 'f', f, false, false, false);
            J$.N(701281, 'MongoError', MongoError, false, false, false);
            J$.N(701289, 'classFilters', classFilters, false, false, false);
            J$.N(701297, 'filteredClasses', filteredClasses, false, false, false);
            J$.N(701305, 'level', level, false, false, false);
            J$.N(701313, 'pid', pid, false, false, false);
            J$.N(701321, 'currentLogger', currentLogger, false, false, false);
            J$.N(701329, 'Logger', Logger, false, false, false);
            J$.T(697769, 'use strict', 21, false);
            var f = J$.W(697841, 'f', J$.G(697801, J$.F(697793, J$.I(typeof require === 'undefined' ? require = J$.R(697777, 'require', undefined, true, true) : require = J$.R(697777, 'require', require, true, true)), false)(J$.T(697785, 'util', 21, false)), 'format'), f, false, true), MongoError = J$.W(697849, 'MongoError', J$.G(697833, J$.F(697825, J$.I(typeof require === 'undefined' ? require = J$.R(697809, 'require', undefined, true, true) : require = J$.R(697809, 'require', require, true, true)), false)(J$.T(697817, '../error', 21, false)), 'MongoError'), MongoError, false, true);
            var classFilters = J$.W(697865, 'classFilters', J$.T(697857, {}, 11, false), classFilters, false, true);
            var filteredClasses = J$.W(697881, 'filteredClasses', J$.T(697873, {}, 11, false), filteredClasses, false, true);
            var level = J$.W(697897, 'level', J$.T(697889, null, 25, false), level, false, true);
            var pid = J$.W(697921, 'pid', J$.G(697913, J$.I(typeof process === 'undefined' ? process = J$.R(697905, 'process', undefined, true, true) : process = J$.R(697905, 'process', process, true, true)), 'pid'), pid, false, true);
            var currentLogger = J$.W(697937, 'currentLogger', J$.T(697929, null, 25, false), currentLogger, false, true);
            var Logger = J$.W(698297, 'Logger', J$.T(698289, function Logger(className, options) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(698257, arguments.callee, this, arguments);
                            arguments = J$.N(698265, 'arguments', arguments, true, false, false);
                            className = J$.N(698273, 'className', className, true, false, false);
                            options = J$.N(698281, 'options', options, true, false, false);
                            if (J$.C(39832, J$.U(85474, '!', J$.B(85466, 'instanceof', J$.R(697945, 'this', this, false, false), J$.R(697953, 'Logger', Logger, false, false)))))
                                return J$.Rt(697993, J$.F(697985, J$.R(697961, 'Logger', Logger, false, false), true)(J$.R(697969, 'className', className, false, false), J$.R(697977, 'options', options, false, false)));
                            options = J$.W(698017, 'options', J$.C(39840, J$.R(698001, 'options', options, false, false)) ? J$._() : J$.T(698009, {}, 11, false), options, false, false);
                            J$.P(698041, J$.R(698025, 'this', this, false, false), 'className', J$.R(698033, 'className', className, false, false));
                            if (J$.C(39856, J$.G(698057, J$.R(698049, 'options', options, false, false), 'logger'))) {
                                currentLogger = J$.W(698081, 'currentLogger', J$.G(698073, J$.R(698065, 'options', options, false, false), 'logger'), currentLogger, false, true);
                            } else if (J$.C(39848, J$.B(85482, '==', J$.R(698089, 'currentLogger', currentLogger, false, true), J$.T(698097, null, 25, false)))) {
                                currentLogger = J$.W(698121, 'currentLogger', J$.G(698113, J$.I(typeof console === 'undefined' ? console = J$.R(698105, 'console', undefined, true, true) : console = J$.R(698105, 'console', console, true, true)), 'log'), currentLogger, false, true);
                            }
                            if (J$.C(39872, J$.G(698137, J$.R(698129, 'options', options, false, false), 'loggerLevel'))) {
                                level = J$.W(698169, 'level', J$.C(39864, J$.G(698153, J$.R(698145, 'options', options, false, false), 'loggerLevel')) ? J$._() : J$.T(698161, 'error', 21, false), level, false, true);
                            }
                            if (J$.C(39880, J$.B(85490, '==', J$.G(698201, J$.R(698177, 'filteredClasses', filteredClasses, false, true), J$.G(698193, J$.R(698185, 'this', this, false, false), 'className')), J$.T(698209, null, 25, false))))
                                J$.P(698249, J$.R(698217, 'classFilters', classFilters, false, true), J$.G(698233, J$.R(698225, 'this', this, false, false), 'className'), J$.T(698241, true, 23, false));
                        } catch (J$e) {
                            J$.Ex(701337, J$e);
                        } finally {
                            if (J$.Fr(701345))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), Logger, false, true);
            J$.P(698785, J$.G(698313, J$.R(698305, 'Logger', Logger, false, true), 'prototype'), 'debug', J$.T(698777, function (message, object) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(698721, arguments.callee, this, arguments);
                            arguments = J$.N(698729, 'arguments', arguments, true, false, false);
                            message = J$.N(698737, 'message', message, true, false, false);
                            object = J$.N(698745, 'object', object, true, false, false);
                            J$.N(698753, 'dateTime', dateTime, false, false, false);
                            J$.N(698761, 'msg', msg, false, false, false);
                            J$.N(698769, 'state', state, false, false, false);
                            if (J$.C(39928, J$.C(39912, J$.M(698329, J$.R(698321, 'this', this, false, false), 'isDebug', false)()) ? J$.C(39904, J$.C(39888, J$.B(85498, '>', J$.G(698361, J$.M(698353, J$.I(typeof Object === 'undefined' ? Object = J$.R(698337, 'Object', undefined, true, true) : Object = J$.R(698337, 'Object', Object, true, true)), 'keys', false)(J$.R(698345, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(698369, 0, 22, false))) ? J$.G(698401, J$.R(698377, 'filteredClasses', filteredClasses, false, true), J$.G(698393, J$.R(698385, 'this', this, false, false), 'className')) : J$._()) ? J$._() : J$.C(39896, J$.B(85506, '===', J$.G(698433, J$.M(698425, J$.I(typeof Object === 'undefined' ? Object = J$.R(698409, 'Object', undefined, true, true) : Object = J$.R(698409, 'Object', Object, true, true)), 'keys', false)(J$.R(698417, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(698441, 0, 22, false))) ? J$.G(698473, J$.R(698449, 'classFilters', classFilters, false, true), J$.G(698465, J$.R(698457, 'this', this, false, false), 'className')) : J$._() : J$._())) {
                                var dateTime = J$.W(698505, 'dateTime', J$.M(698497, J$.F(698489, J$.I(typeof Date === 'undefined' ? Date = J$.R(698481, 'Date', undefined, true, true) : Date = J$.R(698481, 'Date', Date, true, true)), true)(), 'getTime', false)(), dateTime, false, false);
                                var msg = J$.W(698585, 'msg', J$.F(698577, J$.R(698513, 'f', f, false, true), false)(J$.T(698521, '[%s-%s:%s] %s %s', 21, false), J$.T(698529, 'DEBUG', 21, false), J$.G(698545, J$.R(698537, 'this', this, false, false), 'className'), J$.R(698553, 'pid', pid, false, true), J$.R(698561, 'dateTime', dateTime, false, false), J$.R(698569, 'message', message, false, false)), msg, false, false);
                                var state = J$.W(698649, 'state', J$.T(698641, {
                                    type: J$.T(698593, 'debug', 21, false),
                                    message: J$.R(698601, 'message', message, false, false),
                                    className: J$.G(698617, J$.R(698609, 'this', this, false, false), 'className'),
                                    pid: J$.R(698625, 'pid', pid, false, true),
                                    date: J$.R(698633, 'dateTime', dateTime, false, false)
                                }, 11, false), state, false, false);
                                if (J$.C(39920, J$.R(698657, 'object', object, false, false)))
                                    J$.P(698681, J$.R(698665, 'state', state, false, false), 'meta', J$.R(698673, 'object', object, false, false));
                                J$.F(698713, J$.R(698689, 'currentLogger', currentLogger, false, true), false)(J$.R(698697, 'msg', msg, false, false), J$.R(698705, 'state', state, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(701353, J$e);
                        } finally {
                            if (J$.Fr(701361))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(699273, J$.G(698801, J$.R(698793, 'Logger', Logger, false, true), 'prototype'), 'warn', J$.T(699265, function (message, object) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(699209, arguments.callee, this, arguments);
                            arguments = J$.N(699217, 'arguments', arguments, true, false, false);
                            message = J$.N(699225, 'message', message, true, false, false);
                            object = J$.N(699233, 'object', object, true, false, false);
                            J$.N(699241, 'dateTime', dateTime, false, false, false);
                            J$.N(699249, 'msg', msg, false, false, false);
                            J$.N(699257, 'state', state, false, false, false);
                            if (J$.C(39976, J$.C(39960, J$.M(698817, J$.R(698809, 'this', this, false, false), 'isWarn', false)()) ? J$.C(39952, J$.C(39936, J$.B(85514, '>', J$.G(698849, J$.M(698841, J$.I(typeof Object === 'undefined' ? Object = J$.R(698825, 'Object', undefined, true, true) : Object = J$.R(698825, 'Object', Object, true, true)), 'keys', false)(J$.R(698833, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(698857, 0, 22, false))) ? J$.G(698889, J$.R(698865, 'filteredClasses', filteredClasses, false, true), J$.G(698881, J$.R(698873, 'this', this, false, false), 'className')) : J$._()) ? J$._() : J$.C(39944, J$.B(85522, '===', J$.G(698921, J$.M(698913, J$.I(typeof Object === 'undefined' ? Object = J$.R(698897, 'Object', undefined, true, true) : Object = J$.R(698897, 'Object', Object, true, true)), 'keys', false)(J$.R(698905, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(698929, 0, 22, false))) ? J$.G(698961, J$.R(698937, 'classFilters', classFilters, false, true), J$.G(698953, J$.R(698945, 'this', this, false, false), 'className')) : J$._() : J$._())) {
                                var dateTime = J$.W(698993, 'dateTime', J$.M(698985, J$.F(698977, J$.I(typeof Date === 'undefined' ? Date = J$.R(698969, 'Date', undefined, true, true) : Date = J$.R(698969, 'Date', Date, true, true)), true)(), 'getTime', false)(), dateTime, false, false);
                                var msg = J$.W(699073, 'msg', J$.F(699065, J$.R(699001, 'f', f, false, true), false)(J$.T(699009, '[%s-%s:%s] %s %s', 21, false), J$.T(699017, 'WARN', 21, false), J$.G(699033, J$.R(699025, 'this', this, false, false), 'className'), J$.R(699041, 'pid', pid, false, true), J$.R(699049, 'dateTime', dateTime, false, false), J$.R(699057, 'message', message, false, false)), msg, false, false);
                                var state = J$.W(699137, 'state', J$.T(699129, {
                                    type: J$.T(699081, 'warn', 21, false),
                                    message: J$.R(699089, 'message', message, false, false),
                                    className: J$.G(699105, J$.R(699097, 'this', this, false, false), 'className'),
                                    pid: J$.R(699113, 'pid', pid, false, true),
                                    date: J$.R(699121, 'dateTime', dateTime, false, false)
                                }, 11, false), state, false, false);
                                if (J$.C(39968, J$.R(699145, 'object', object, false, false)))
                                    J$.P(699169, J$.R(699153, 'state', state, false, false), 'meta', J$.R(699161, 'object', object, false, false));
                                J$.F(699201, J$.R(699177, 'currentLogger', currentLogger, false, true), false)(J$.R(699185, 'msg', msg, false, false), J$.R(699193, 'state', state, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(701369, J$e);
                        } finally {
                            if (J$.Fr(701377))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(699761, J$.G(699289, J$.R(699281, 'Logger', Logger, false, true), 'prototype'), 'info', J$.T(699753, function (message, object) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(699697, arguments.callee, this, arguments);
                            arguments = J$.N(699705, 'arguments', arguments, true, false, false);
                            message = J$.N(699713, 'message', message, true, false, false);
                            object = J$.N(699721, 'object', object, true, false, false);
                            J$.N(699729, 'dateTime', dateTime, false, false, false);
                            J$.N(699737, 'msg', msg, false, false, false);
                            J$.N(699745, 'state', state, false, false, false);
                            if (J$.C(40024, J$.C(40008, J$.M(699305, J$.R(699297, 'this', this, false, false), 'isInfo', false)()) ? J$.C(40000, J$.C(39984, J$.B(85530, '>', J$.G(699337, J$.M(699329, J$.I(typeof Object === 'undefined' ? Object = J$.R(699313, 'Object', undefined, true, true) : Object = J$.R(699313, 'Object', Object, true, true)), 'keys', false)(J$.R(699321, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(699345, 0, 22, false))) ? J$.G(699377, J$.R(699353, 'filteredClasses', filteredClasses, false, true), J$.G(699369, J$.R(699361, 'this', this, false, false), 'className')) : J$._()) ? J$._() : J$.C(39992, J$.B(85538, '===', J$.G(699409, J$.M(699401, J$.I(typeof Object === 'undefined' ? Object = J$.R(699385, 'Object', undefined, true, true) : Object = J$.R(699385, 'Object', Object, true, true)), 'keys', false)(J$.R(699393, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(699417, 0, 22, false))) ? J$.G(699449, J$.R(699425, 'classFilters', classFilters, false, true), J$.G(699441, J$.R(699433, 'this', this, false, false), 'className')) : J$._() : J$._())) {
                                var dateTime = J$.W(699481, 'dateTime', J$.M(699473, J$.F(699465, J$.I(typeof Date === 'undefined' ? Date = J$.R(699457, 'Date', undefined, true, true) : Date = J$.R(699457, 'Date', Date, true, true)), true)(), 'getTime', false)(), dateTime, false, false);
                                var msg = J$.W(699561, 'msg', J$.F(699553, J$.R(699489, 'f', f, false, true), false)(J$.T(699497, '[%s-%s:%s] %s %s', 21, false), J$.T(699505, 'INFO', 21, false), J$.G(699521, J$.R(699513, 'this', this, false, false), 'className'), J$.R(699529, 'pid', pid, false, true), J$.R(699537, 'dateTime', dateTime, false, false), J$.R(699545, 'message', message, false, false)), msg, false, false);
                                var state = J$.W(699625, 'state', J$.T(699617, {
                                    type: J$.T(699569, 'info', 21, false),
                                    message: J$.R(699577, 'message', message, false, false),
                                    className: J$.G(699593, J$.R(699585, 'this', this, false, false), 'className'),
                                    pid: J$.R(699601, 'pid', pid, false, true),
                                    date: J$.R(699609, 'dateTime', dateTime, false, false)
                                }, 11, false), state, false, false);
                                if (J$.C(40016, J$.R(699633, 'object', object, false, false)))
                                    J$.P(699657, J$.R(699641, 'state', state, false, false), 'meta', J$.R(699649, 'object', object, false, false));
                                J$.F(699689, J$.R(699665, 'currentLogger', currentLogger, false, true), false)(J$.R(699673, 'msg', msg, false, false), J$.R(699681, 'state', state, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(701385, J$e);
                        } finally {
                            if (J$.Fr(701393))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(700249, J$.G(699777, J$.R(699769, 'Logger', Logger, false, true), 'prototype'), 'error', J$.T(700241, function (message, object) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(700185, arguments.callee, this, arguments);
                            arguments = J$.N(700193, 'arguments', arguments, true, false, false);
                            message = J$.N(700201, 'message', message, true, false, false);
                            object = J$.N(700209, 'object', object, true, false, false);
                            J$.N(700217, 'dateTime', dateTime, false, false, false);
                            J$.N(700225, 'msg', msg, false, false, false);
                            J$.N(700233, 'state', state, false, false, false);
                            if (J$.C(40072, J$.C(40056, J$.M(699793, J$.R(699785, 'this', this, false, false), 'isError', false)()) ? J$.C(40048, J$.C(40032, J$.B(85546, '>', J$.G(699825, J$.M(699817, J$.I(typeof Object === 'undefined' ? Object = J$.R(699801, 'Object', undefined, true, true) : Object = J$.R(699801, 'Object', Object, true, true)), 'keys', false)(J$.R(699809, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(699833, 0, 22, false))) ? J$.G(699865, J$.R(699841, 'filteredClasses', filteredClasses, false, true), J$.G(699857, J$.R(699849, 'this', this, false, false), 'className')) : J$._()) ? J$._() : J$.C(40040, J$.B(85554, '===', J$.G(699897, J$.M(699889, J$.I(typeof Object === 'undefined' ? Object = J$.R(699873, 'Object', undefined, true, true) : Object = J$.R(699873, 'Object', Object, true, true)), 'keys', false)(J$.R(699881, 'filteredClasses', filteredClasses, false, true)), 'length'), J$.T(699905, 0, 22, false))) ? J$.G(699937, J$.R(699913, 'classFilters', classFilters, false, true), J$.G(699929, J$.R(699921, 'this', this, false, false), 'className')) : J$._() : J$._())) {
                                var dateTime = J$.W(699969, 'dateTime', J$.M(699961, J$.F(699953, J$.I(typeof Date === 'undefined' ? Date = J$.R(699945, 'Date', undefined, true, true) : Date = J$.R(699945, 'Date', Date, true, true)), true)(), 'getTime', false)(), dateTime, false, false);
                                var msg = J$.W(700049, 'msg', J$.F(700041, J$.R(699977, 'f', f, false, true), false)(J$.T(699985, '[%s-%s:%s] %s %s', 21, false), J$.T(699993, 'ERROR', 21, false), J$.G(700009, J$.R(700001, 'this', this, false, false), 'className'), J$.R(700017, 'pid', pid, false, true), J$.R(700025, 'dateTime', dateTime, false, false), J$.R(700033, 'message', message, false, false)), msg, false, false);
                                var state = J$.W(700113, 'state', J$.T(700105, {
                                    type: J$.T(700057, 'error', 21, false),
                                    message: J$.R(700065, 'message', message, false, false),
                                    className: J$.G(700081, J$.R(700073, 'this', this, false, false), 'className'),
                                    pid: J$.R(700089, 'pid', pid, false, true),
                                    date: J$.R(700097, 'dateTime', dateTime, false, false)
                                }, 11, false), state, false, false);
                                if (J$.C(40064, J$.R(700121, 'object', object, false, false)))
                                    J$.P(700145, J$.R(700129, 'state', state, false, false), 'meta', J$.R(700137, 'object', object, false, false));
                                J$.F(700177, J$.R(700153, 'currentLogger', currentLogger, false, true), false)(J$.R(700161, 'msg', msg, false, false), J$.R(700169, 'state', state, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(701401, J$e);
                        } finally {
                            if (J$.Fr(701409))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(700337, J$.G(700265, J$.R(700257, 'Logger', Logger, false, true), 'prototype'), 'isInfo', J$.T(700329, function () {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(700313, arguments.callee, this, arguments);
                            arguments = J$.N(700321, 'arguments', arguments, true, false, false);
                            return J$.Rt(700305, J$.C(40080, J$.B(85562, '===', J$.R(700273, 'level', level, false, true), J$.T(700281, 'info', 21, false))) ? J$._() : J$.B(85570, '===', J$.R(700289, 'level', level, false, true), J$.T(700297, 'debug', 21, false)));
                        } catch (J$e) {
                            J$.Ex(701417, J$e);
                        } finally {
                            if (J$.Fr(701425))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(700441, J$.G(700353, J$.R(700345, 'Logger', Logger, false, true), 'prototype'), 'isError', J$.T(700433, function () {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(700417, arguments.callee, this, arguments);
                            arguments = J$.N(700425, 'arguments', arguments, true, false, false);
                            return J$.Rt(700409, J$.C(40096, J$.C(40088, J$.B(85578, '===', J$.R(700361, 'level', level, false, true), J$.T(700369, 'error', 21, false))) ? J$._() : J$.B(85586, '===', J$.R(700377, 'level', level, false, true), J$.T(700385, 'info', 21, false))) ? J$._() : J$.B(85594, '===', J$.R(700393, 'level', level, false, true), J$.T(700401, 'debug', 21, false)));
                        } catch (J$e) {
                            J$.Ex(701433, J$e);
                        } finally {
                            if (J$.Fr(701441))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(700561, J$.G(700457, J$.R(700449, 'Logger', Logger, false, true), 'prototype'), 'isWarn', J$.T(700553, function () {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(700537, arguments.callee, this, arguments);
                            arguments = J$.N(700545, 'arguments', arguments, true, false, false);
                            return J$.Rt(700529, J$.C(40120, J$.C(40112, J$.C(40104, J$.B(85602, '===', J$.R(700465, 'level', level, false, true), J$.T(700473, 'error', 21, false))) ? J$._() : J$.B(85610, '===', J$.R(700481, 'level', level, false, true), J$.T(700489, 'warn', 21, false))) ? J$._() : J$.B(85618, '===', J$.R(700497, 'level', level, false, true), J$.T(700505, 'info', 21, false))) ? J$._() : J$.B(85626, '===', J$.R(700513, 'level', level, false, true), J$.T(700521, 'debug', 21, false)));
                        } catch (J$e) {
                            J$.Ex(701449, J$e);
                        } finally {
                            if (J$.Fr(701457))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), J$.P(700633, J$.G(700577, J$.R(700569, 'Logger', Logger, false, true), 'prototype'), 'isDebug', J$.T(700625, function () {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(700609, arguments.callee, this, arguments);
                            arguments = J$.N(700617, 'arguments', arguments, true, false, false);
                            return J$.Rt(700601, J$.B(85634, '===', J$.R(700585, 'level', level, false, true), J$.T(700593, 'debug', 21, false)));
                        } catch (J$e) {
                            J$.Ex(701465, J$e);
                        } finally {
                            if (J$.Fr(701473))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(700705, J$.R(700641, 'Logger', Logger, false, true), 'reset', J$.T(700697, function () {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(700681, arguments.callee, this, arguments);
                            arguments = J$.N(700689, 'arguments', arguments, true, false, false);
                            level = J$.W(700657, 'level', J$.T(700649, 'error', 21, false), level, false, true);
                            filteredClasses = J$.W(700673, 'filteredClasses', J$.T(700665, {}, 11, false), filteredClasses, false, true);
                        } catch (J$e) {
                            J$.Ex(701481, J$e);
                        } finally {
                            if (J$.Fr(701489))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(700761, J$.R(700713, 'Logger', Logger, false, true), 'currentLogger', J$.T(700753, function () {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(700737, arguments.callee, this, arguments);
                            arguments = J$.N(700745, 'arguments', arguments, true, false, false);
                            return J$.Rt(700729, J$.R(700721, 'currentLogger', currentLogger, false, true));
                        } catch (J$e) {
                            J$.Ex(701497, J$e);
                        } finally {
                            if (J$.Fr(701505))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(700865, J$.R(700769, 'Logger', Logger, false, true), 'setCurrentLogger', J$.T(700857, function (logger) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(700833, arguments.callee, this, arguments);
                            arguments = J$.N(700841, 'arguments', arguments, true, false, false);
                            logger = J$.N(700849, 'logger', logger, true, false, false);
                            if (J$.C(40128, J$.B(85650, '!==', J$.U(85642, 'typeof', J$.R(700777, 'logger', logger, false, false)), J$.T(700785, 'function', 21, false))))
                                throw J$.F(700809, J$.R(700793, 'MongoError', MongoError, false, true), true)(J$.T(700801, 'current logger must be a function', 21, false));
                            currentLogger = J$.W(700825, 'currentLogger', J$.R(700817, 'logger', logger, false, false), currentLogger, false, true);
                        } catch (J$e) {
                            J$.Ex(701513, J$e);
                        } finally {
                            if (J$.Fr(701521))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(701057, J$.R(700873, 'Logger', Logger, false, true), 'filter', J$.T(701049, function (type, values) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(701017, arguments.callee, this, arguments);
                            arguments = J$.N(701025, 'arguments', arguments, true, false, false);
                            type = J$.N(701033, 'type', type, true, false, false);
                            values = J$.N(701041, 'values', values, true, false, false);
                            if (J$.C(40144, J$.C(40136, J$.B(85658, '===', J$.R(700881, 'type', type, false, false), J$.T(700889, 'class', 21, false))) ? J$.M(700913, J$.I(typeof Array === 'undefined' ? Array = J$.R(700897, 'Array', undefined, true, true) : Array = J$.R(700897, 'Array', Array, true, true)), 'isArray', false)(J$.R(700905, 'values', values, false, false)) : J$._())) {
                                filteredClasses = J$.W(700929, 'filteredClasses', J$.T(700921, {}, 11, false), filteredClasses, false, true);
                                J$.M(701009, J$.R(700937, 'values', values, false, false), 'forEach', false)(J$.T(701001, function (x) {
                                    jalangiLabel12:
                                        while (true) {
                                            try {
                                                J$.Fe(700977, arguments.callee, this, arguments);
                                                arguments = J$.N(700985, 'arguments', arguments, true, false, false);
                                                x = J$.N(700993, 'x', x, true, false, false);
                                                J$.P(700969, J$.R(700945, 'filteredClasses', filteredClasses, false, true), J$.R(700953, 'x', x, false, false), J$.T(700961, true, 23, false));
                                            } catch (J$e) {
                                                J$.Ex(701529, J$e);
                                            } finally {
                                                if (J$.Fr(701537))
                                                    continue jalangiLabel12;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            }
                        } catch (J$e) {
                            J$.Ex(701545, J$e);
                        } finally {
                            if (J$.Fr(701553))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(701233, J$.R(701065, 'Logger', Logger, false, true), 'setLevel', J$.T(701225, function (_level) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(701201, arguments.callee, this, arguments);
                            arguments = J$.N(701209, 'arguments', arguments, true, false, false);
                            _level = J$.N(701217, '_level', _level, true, false, false);
                            if (J$.C(40176, J$.C(40168, J$.C(40160, J$.C(40152, J$.B(85666, '!==', J$.R(701073, '_level', _level, false, false), J$.T(701081, 'info', 21, false))) ? J$.B(85674, '!==', J$.R(701089, '_level', _level, false, false), J$.T(701097, 'error', 21, false)) : J$._()) ? J$.B(85682, '!==', J$.R(701105, '_level', _level, false, false), J$.T(701113, 'debug', 21, false)) : J$._()) ? J$.B(85690, '!==', J$.R(701121, '_level', _level, false, false), J$.T(701129, 'warn', 21, false)) : J$._())) {
                                throw J$.F(701177, J$.I(typeof Error === 'undefined' ? Error = J$.R(701137, 'Error', undefined, true, true) : Error = J$.R(701137, 'Error', Error, true, true)), true)(J$.F(701169, J$.R(701145, 'f', f, false, true), false)(J$.T(701153, '%s is an illegal logging level', 21, false), J$.R(701161, '_level', _level, false, false)));
                            }
                            level = J$.W(701193, 'level', J$.R(701185, '_level', _level, false, false), level, false, true);
                        } catch (J$e) {
                            J$.Ex(701561, J$e);
                        } finally {
                            if (J$.Fr(701569))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(701257, J$.I(typeof module === 'undefined' ? module = J$.R(701241, 'module', undefined, true, true) : module = J$.R(701241, 'module', module, true, true)), 'exports', J$.R(701249, 'Logger', Logger, false, true));
        } catch (J$e) {
            J$.Ex(701577, J$e);
        } finally {
            if (J$.Sr(701585))
                continue jalangiLabel15;
            else
                break jalangiLabel15;
        }
    }
// JALANGI DO NOT INSTRUMENT
