// generate-function/index.js
J$.noInstrEval = false;
jalangiLabel15:
    while (true) {
        try {
            J$.Se(4086881, '/home/james/nodejs/HiPar/outputs/target_cache/TestIsMyJsonValid/node_modules/generate-function/index.js');
            J$.N(4086889, 'util', util, false, false, false);
            J$.N(4086897, 'isProperty', isProperty, false, false, false);
            J$.N(4086905, 'INDENT_START', INDENT_START, false, false, false);
            J$.N(4086913, 'INDENT_END', INDENT_END, false, false, false);
            J$.N(4086921, 'RESERVED', RESERVED, false, false, false);
            J$.N(4086929, 'RESERVED_MAP', RESERVED_MAP, false, false, false);
            J$.N(4086937, 'i', i, false, false, false);
            J$.N(4086945, 'isVariable', isVariable, false, false, false);
            J$.N(4086953, 'formats', formats, false, false, false);
            J$.N(4086961, 'genfun', genfun, false, false, false);
            J$.T(4083265, 'use strict', 21, false);
            J$.F(4083289, J$.I(typeof require === 'undefined' ? require = J$.R(4083273, 'require', undefined, true, true) : require = J$.R(4083273, 'require', require, true, true)), false)(J$.T(4083281, 'core-js/modules/es.array.concat', 21, false));
            J$.F(4083313, J$.I(typeof require === 'undefined' ? require = J$.R(4083297, 'require', undefined, true, true) : require = J$.R(4083297, 'require', require, true, true)), false)(J$.T(4083305, 'core-js/modules/es.array.for-each', 21, false));
            J$.F(4083337, J$.I(typeof require === 'undefined' ? require = J$.R(4083321, 'require', undefined, true, true) : require = J$.R(4083321, 'require', require, true, true)), false)(J$.T(4083329, 'core-js/modules/es.array.index-of', 21, false));
            J$.F(4083361, J$.I(typeof require === 'undefined' ? require = J$.R(4083345, 'require', undefined, true, true) : require = J$.R(4083345, 'require', require, true, true)), false)(J$.T(4083353, 'core-js/modules/es.array.join', 21, false));
            J$.F(4083385, J$.I(typeof require === 'undefined' ? require = J$.R(4083369, 'require', undefined, true, true) : require = J$.R(4083369, 'require', require, true, true)), false)(J$.T(4083377, 'core-js/modules/es.array.map', 21, false));
            J$.F(4083409, J$.I(typeof require === 'undefined' ? require = J$.R(4083393, 'require', undefined, true, true) : require = J$.R(4083393, 'require', require, true, true)), false)(J$.T(4083401, 'core-js/modules/es.date.to-string', 21, false));
            J$.F(4083433, J$.I(typeof require === 'undefined' ? require = J$.R(4083417, 'require', undefined, true, true) : require = J$.R(4083417, 'require', require, true, true)), false)(J$.T(4083425, 'core-js/modules/es.number.constructor', 21, false));
            J$.F(4083457, J$.I(typeof require === 'undefined' ? require = J$.R(4083441, 'require', undefined, true, true) : require = J$.R(4083441, 'require', require, true, true)), false)(J$.T(4083449, 'core-js/modules/es.object.keys', 21, false));
            J$.F(4083481, J$.I(typeof require === 'undefined' ? require = J$.R(4083465, 'require', undefined, true, true) : require = J$.R(4083465, 'require', require, true, true)), false)(J$.T(4083473, 'core-js/modules/es.object.to-string', 21, false));
            J$.F(4083505, J$.I(typeof require === 'undefined' ? require = J$.R(4083489, 'require', undefined, true, true) : require = J$.R(4083489, 'require', require, true, true)), false)(J$.T(4083497, 'core-js/modules/es.regexp.exec', 21, false));
            J$.F(4083529, J$.I(typeof require === 'undefined' ? require = J$.R(4083513, 'require', undefined, true, true) : require = J$.R(4083513, 'require', require, true, true)), false)(J$.T(4083521, 'core-js/modules/es.regexp.to-string', 21, false));
            J$.F(4083553, J$.I(typeof require === 'undefined' ? require = J$.R(4083537, 'require', undefined, true, true) : require = J$.R(4083537, 'require', require, true, true)), false)(J$.T(4083545, 'core-js/modules/es.string.split', 21, false));
            J$.F(4083577, J$.I(typeof require === 'undefined' ? require = J$.R(4083561, 'require', undefined, true, true) : require = J$.R(4083561, 'require', require, true, true)), false)(J$.T(4083569, 'core-js/modules/es.string.trim', 21, false));
            J$.F(4083601, J$.I(typeof require === 'undefined' ? require = J$.R(4083585, 'require', undefined, true, true) : require = J$.R(4083585, 'require', require, true, true)), false)(J$.T(4083593, 'core-js/modules/web.dom-collections.for-each', 21, false));
            var util = J$.W(4083633, 'util', J$.F(4083625, J$.I(typeof require === 'undefined' ? require = J$.R(4083609, 'require', undefined, true, true) : require = J$.R(4083609, 'require', require, true, true)), false)(J$.T(4083617, 'util', 21, false)), util, false, true);
            var isProperty = J$.W(4083665, 'isProperty', J$.F(4083657, J$.I(typeof require === 'undefined' ? require = J$.R(4083641, 'require', undefined, true, true) : require = J$.R(4083641, 'require', require, true, true)), false)(J$.T(4083649, 'is-property', 21, false)), isProperty, false, true);
            var INDENT_START = J$.W(4083681, 'INDENT_START', J$.T(4083673, /[\{\[]/, 14, false), INDENT_START, false, true);
            var INDENT_END = J$.W(4083697, 'INDENT_END', J$.T(4083689, /[\}\]]/, 14, false), INDENT_END, false, true);
            var RESERVED = J$.W(4084113, 'RESERVED', J$.T(4084105, [
                J$.T(4083705, 'do', 21, false),
                J$.T(4083713, 'if', 21, false),
                J$.T(4083721, 'in', 21, false),
                J$.T(4083729, 'for', 21, false),
                J$.T(4083737, 'let', 21, false),
                J$.T(4083745, 'new', 21, false),
                J$.T(4083753, 'try', 21, false),
                J$.T(4083761, 'var', 21, false),
                J$.T(4083769, 'case', 21, false),
                J$.T(4083777, 'else', 21, false),
                J$.T(4083785, 'enum', 21, false),
                J$.T(4083793, 'eval', 21, false),
                J$.T(4083801, 'null', 21, false),
                J$.T(4083809, 'this', 21, false),
                J$.T(4083817, 'true', 21, false),
                J$.T(4083825, 'void', 21, false),
                J$.T(4083833, 'with', 21, false),
                J$.T(4083841, 'await', 21, false),
                J$.T(4083849, 'break', 21, false),
                J$.T(4083857, 'catch', 21, false),
                J$.T(4083865, 'class', 21, false),
                J$.T(4083873, 'const', 21, false),
                J$.T(4083881, 'false', 21, false),
                J$.T(4083889, 'super', 21, false),
                J$.T(4083897, 'throw', 21, false),
                J$.T(4083905, 'while', 21, false),
                J$.T(4083913, 'yield', 21, false),
                J$.T(4083921, 'delete', 21, false),
                J$.T(4083929, 'export', 21, false),
                J$.T(4083937, 'import', 21, false),
                J$.T(4083945, 'public', 21, false),
                J$.T(4083953, 'return', 21, false),
                J$.T(4083961, 'static', 21, false),
                J$.T(4083969, 'switch', 21, false),
                J$.T(4083977, 'typeof', 21, false),
                J$.T(4083985, 'default', 21, false),
                J$.T(4083993, 'extends', 21, false),
                J$.T(4084001, 'finally', 21, false),
                J$.T(4084009, 'package', 21, false),
                J$.T(4084017, 'private', 21, false),
                J$.T(4084025, 'continue', 21, false),
                J$.T(4084033, 'debugger', 21, false),
                J$.T(4084041, 'function', 21, false),
                J$.T(4084049, 'arguments', 21, false),
                J$.T(4084057, 'interface', 21, false),
                J$.T(4084065, 'protected', 21, false),
                J$.T(4084073, 'implements', 21, false),
                J$.T(4084081, 'instanceof', 21, false),
                J$.T(4084089, 'NaN', 21, false),
                J$.T(4084097, 'undefined', 21, false)
            ], 10, false), RESERVED, false, true);
            var RESERVED_MAP = J$.W(4084129, 'RESERVED_MAP', J$.T(4084121, {}, 11, false), RESERVED_MAP, false, true);
            for (var i = J$.W(4084145, 'i', J$.T(4084137, 0, 22, false), i, false, true); J$.C(216360, J$.B(396482, '<', J$.R(4084153, 'i', i, false, true), J$.G(4084169, J$.R(4084161, 'RESERVED', RESERVED, false, true), 'length'))); J$.B(396506, '-', i = J$.W(4084185, 'i', J$.B(396498, '+', J$.U(396490, '+', J$.R(4084177, 'i', i, false, true)), 1), i, false, true), 1)) {
                J$.P(4084233, J$.R(4084193, 'RESERVED_MAP', RESERVED_MAP, false, true), J$.G(4084217, J$.R(4084201, 'RESERVED', RESERVED, false, true), J$.R(4084209, 'i', i, false, true)), J$.T(4084225, true, 23, false));
            }
            var isVariable = J$.W(4084329, 'isVariable', J$.T(4084321, function isVariable(name) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(4084297, arguments.callee, this, arguments);
                            arguments = J$.N(4084305, 'arguments', arguments, true, false, false);
                            name = J$.N(4084313, 'name', name, true, false, false);
                            return J$.Rt(4084289, J$.C(216368, J$.F(4084257, J$.R(4084241, 'isProperty', isProperty, false, true), false)(J$.R(4084249, 'name', name, false, false))) ? J$.U(396514, '!', J$.M(4084281, J$.R(4084265, 'RESERVED_MAP', RESERVED_MAP, false, true), 'hasOwnProperty', false)(J$.R(4084273, 'name', name, false, false))) : J$._());
                        } catch (J$e) {
                            J$.Ex(4086969, J$e);
                        } finally {
                            if (J$.Fr(4086977))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), isVariable, false, true);
            var formats = J$.W(4084537, 'formats', J$.T(4084529, {
                s: J$.T(4084385, function s(_s) {
                    jalangiLabel1:
                        while (true) {
                            try {
                                J$.Fe(4084361, arguments.callee, this, arguments);
                                arguments = J$.N(4084369, 'arguments', arguments, true, false, false);
                                _s = J$.N(4084377, '_s', _s, true, false, false);
                                return J$.Rt(4084353, J$.B(396522, '+', J$.T(4084337, '', 21, false), J$.R(4084345, '_s', _s, false, false)));
                            } catch (J$e) {
                                J$.Ex(4086985, J$e);
                            } finally {
                                if (J$.Fr(4086993))
                                    continue jalangiLabel1;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                d: J$.T(4084457, function d(_d) {
                    jalangiLabel2:
                        while (true) {
                            try {
                                J$.Fe(4084433, arguments.callee, this, arguments);
                                arguments = J$.N(4084441, 'arguments', arguments, true, false, false);
                                _d = J$.N(4084449, '_d', _d, true, false, false);
                                return J$.Rt(4084425, J$.B(396530, '+', J$.T(4084393, '', 21, false), J$.F(4084417, J$.I(typeof Number === 'undefined' ? Number = J$.R(4084401, 'Number', undefined, true, true) : Number = J$.R(4084401, 'Number', Number, true, true)), false)(J$.R(4084409, '_d', _d, false, false))));
                            } catch (J$e) {
                                J$.Ex(4087001, J$e);
                            } finally {
                                if (J$.Fr(4087009))
                                    continue jalangiLabel2;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                o: J$.T(4084521, function o(_o) {
                    jalangiLabel3:
                        while (true) {
                            try {
                                J$.Fe(4084497, arguments.callee, this, arguments);
                                arguments = J$.N(4084505, 'arguments', arguments, true, false, false);
                                _o = J$.N(4084513, '_o', _o, true, false, false);
                                return J$.Rt(4084489, J$.M(4084481, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(4084465, 'JSON', undefined, true, true) : JSON = J$.R(4084465, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(4084473, '_o', _o, false, false)));
                            } catch (J$e) {
                                J$.Ex(4087017, J$e);
                            } finally {
                                if (J$.Fr(4087025))
                                    continue jalangiLabel3;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false), formats, false, true);
            var genfun = J$.W(4086825, 'genfun', J$.T(4086817, function genfun() {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(4086753, arguments.callee, this, arguments);
                            arguments = J$.N(4086761, 'arguments', arguments, true, false, false);
                            J$.N(4086769, 'lines', lines, false, false, false);
                            J$.N(4086777, 'indent', indent, false, false, false);
                            J$.N(4086785, 'vars', vars, false, false, false);
                            J$.N(4086793, 'push', push, false, false, false);
                            J$.N(4086801, 'pushLine', pushLine, false, false, false);
                            J$.N(4086809, 'line', line, false, false, false);
                            var lines = J$.W(4084553, 'lines', J$.T(4084545, [], 10, false), lines, false, false);
                            var indent = J$.W(4084569, 'indent', J$.T(4084561, 0, 22, false), indent, false, false);
                            var vars = J$.W(4084585, 'vars', J$.T(4084577, {}, 11, false), vars, false, false);
                            var push = J$.W(4084737, 'push', J$.T(4084729, function push(str) {
                                jalangiLabel4:
                                    while (true) {
                                        try {
                                            J$.Fe(4084697, arguments.callee, this, arguments);
                                            arguments = J$.N(4084705, 'arguments', arguments, true, false, false);
                                            str = J$.N(4084713, 'str', str, true, false, false);
                                            J$.N(4084721, 'spaces', spaces, false, false, false);
                                            var spaces = J$.W(4084601, 'spaces', J$.T(4084593, '', 21, false), spaces, false, false);
                                            while (J$.C(216376, J$.B(396546, '<', J$.G(4084617, J$.R(4084609, 'spaces', spaces, false, false), 'length'), J$.B(396538, '*', J$.R(4084625, 'indent', indent, false, false), J$.T(4084633, 2, 22, false))))) {
                                                spaces = J$.W(4084657, 'spaces', J$.B(396554, '+', J$.R(4084649, 'spaces', spaces, false, false), J$.T(4084641, '  ', 21, false)), spaces, false, false);
                                            }
                                            J$.M(4084689, J$.R(4084665, 'lines', lines, false, false), 'push', false)(J$.B(396562, '+', J$.R(4084673, 'spaces', spaces, false, false), J$.R(4084681, 'str', str, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(4087033, J$e);
                                        } finally {
                                            if (J$.Fr(4087041))
                                                continue jalangiLabel4;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), push, false, false);
                            var pushLine = J$.W(4085169, 'pushLine', J$.T(4085161, function pushLine(line) {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(4085137, arguments.callee, this, arguments);
                                            arguments = J$.N(4085145, 'arguments', arguments, true, false, false);
                                            line = J$.N(4085153, 'line', line, true, false, false);
                                            if (J$.C(216392, J$.C(216384, J$.M(4084785, J$.R(4084745, 'INDENT_END', INDENT_END, false, true), 'test', false)(J$.G(4084777, J$.M(4084761, J$.R(4084753, 'line', line, false, false), 'trim', false)(), J$.T(4084769, 0, 22, false)))) ? J$.M(4084841, J$.R(4084793, 'INDENT_START', INDENT_START, false, true), 'test', false)(J$.G(4084833, J$.R(4084801, 'line', line, false, false), J$.B(396570, '-', J$.G(4084817, J$.R(4084809, 'line', line, false, false), 'length'), J$.T(4084825, 1, 22, false)))) : J$._())) {
                                                J$.B(396594, '+', indent = J$.W(4084857, 'indent', J$.B(396586, '-', J$.U(396578, '+', J$.R(4084849, 'indent', indent, false, false)), 1), indent, false, false), 1);
                                                J$.F(4084881, J$.R(4084865, 'push', push, false, false), false)(J$.R(4084873, 'line', line, false, false));
                                                J$.B(396618, '-', indent = J$.W(4084897, 'indent', J$.B(396610, '+', J$.U(396602, '+', J$.R(4084889, 'indent', indent, false, false)), 1), indent, false, false), 1);
                                                return J$.Rt(4084905, undefined);
                                            }
                                            if (J$.C(216400, J$.M(4084961, J$.R(4084913, 'INDENT_START', INDENT_START, false, true), 'test', false)(J$.G(4084953, J$.R(4084921, 'line', line, false, false), J$.B(396626, '-', J$.G(4084937, J$.R(4084929, 'line', line, false, false), 'length'), J$.T(4084945, 1, 22, false)))))) {
                                                J$.F(4084985, J$.R(4084969, 'push', push, false, false), false)(J$.R(4084977, 'line', line, false, false));
                                                J$.B(396650, '-', indent = J$.W(4085001, 'indent', J$.B(396642, '+', J$.U(396634, '+', J$.R(4084993, 'indent', indent, false, false)), 1), indent, false, false), 1);
                                                return J$.Rt(4085009, undefined);
                                            }
                                            if (J$.C(216408, J$.M(4085057, J$.R(4085017, 'INDENT_END', INDENT_END, false, true), 'test', false)(J$.G(4085049, J$.M(4085033, J$.R(4085025, 'line', line, false, false), 'trim', false)(), J$.T(4085041, 0, 22, false))))) {
                                                J$.B(396674, '+', indent = J$.W(4085073, 'indent', J$.B(396666, '-', J$.U(396658, '+', J$.R(4085065, 'indent', indent, false, false)), 1), indent, false, false), 1);
                                                J$.F(4085097, J$.R(4085081, 'push', push, false, false), false)(J$.R(4085089, 'line', line, false, false));
                                                return J$.Rt(4085105, undefined);
                                            }
                                            J$.F(4085129, J$.R(4085113, 'push', push, false, false), false)(J$.R(4085121, 'line', line, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4087049, J$e);
                                        } finally {
                                            if (J$.Fr(4087057))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), pushLine, false, false);
                            var line = J$.W(4085521, 'line', J$.T(4085513, function line(fmt) {
                                jalangiLabel6:
                                    while (true) {
                                        try {
                                            J$.Fe(4085473, arguments.callee, this, arguments);
                                            arguments = J$.N(4085481, 'arguments', arguments, true, false, false);
                                            fmt = J$.N(4085489, 'fmt', fmt, true, false, false);
                                            J$.N(4085497, 'lines', lines, false, false, false);
                                            J$.N(4085505, 'i', i, false, false, false);
                                            if (J$.C(216416, J$.U(396682, '!', J$.R(4085177, 'fmt', fmt, false, false))))
                                                return J$.Rt(4085193, J$.R(4085185, 'line', line, false, false));
                                            if (J$.C(216440, J$.C(216424, J$.B(396690, '===', J$.G(4085209, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4085201, 'arguments', undefined, true, true) : arguments = J$.R(4085201, 'arguments', arguments, true, true)), 'length'), J$.T(4085217, 1, 22, false))) ? J$.B(396706, '>', J$.M(4085241, J$.R(4085225, 'fmt', fmt, false, false), 'indexOf', false)(J$.T(4085233, '\n', 21, false)), J$.U(396698, '-', J$.T(4085249, 1, 22, false))) : J$._())) {
                                                var lines = J$.W(4085289, 'lines', J$.M(4085281, J$.M(4085265, J$.R(4085257, 'fmt', fmt, false, false), 'trim', false)(), 'split', false)(J$.T(4085273, '\n', 21, false)), lines, false, false);
                                                for (var i = J$.W(4085305, 'i', J$.T(4085297, 0, 22, false), i, false, false); J$.C(216432, J$.B(396714, '<', J$.R(4085313, 'i', i, false, false), J$.G(4085329, J$.R(4085321, 'lines', lines, false, false), 'length'))); J$.B(396738, '-', i = J$.W(4085345, 'i', J$.B(396730, '+', J$.U(396722, '+', J$.R(4085337, 'i', i, false, false)), 1), i, false, false), 1)) {
                                                    J$.F(4085393, J$.R(4085353, 'pushLine', pushLine, false, false), false)(J$.M(4085385, J$.G(4085377, J$.R(4085361, 'lines', lines, false, false), J$.R(4085369, 'i', i, false, false)), 'trim', false)());
                                                }
                                            } else {
                                                J$.F(4085449, J$.R(4085401, 'pushLine', pushLine, false, false), false)(J$.M(4085441, J$.G(4085417, J$.R(4085409, 'util', util, false, true), 'format'), 'apply', false)(J$.R(4085425, 'util', util, false, true), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4085433, 'arguments', undefined, true, true) : arguments = J$.R(4085433, 'arguments', arguments, true, true))));
                                            }
                                            return J$.Rt(4085465, J$.R(4085457, 'line', line, false, false));
                                        } catch (J$e) {
                                            J$.Ex(4087065, J$e);
                                        } finally {
                                            if (J$.Fr(4087073))
                                                continue jalangiLabel6;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), line, false, false);
                            J$.P(4085545, J$.R(4085529, 'line', line, false, false), 'scope', J$.T(4085537, {}, 11, false));
                            J$.P(4085569, J$.R(4085553, 'line', line, false, false), 'formats', J$.R(4085561, 'formats', formats, false, true));
                            J$.P(4085769, J$.R(4085577, 'line', line, false, false), 'sym', J$.T(4085761, function (name) {
                                jalangiLabel7:
                                    while (true) {
                                        try {
                                            J$.Fe(4085737, arguments.callee, this, arguments);
                                            arguments = J$.N(4085745, 'arguments', arguments, true, false, false);
                                            name = J$.N(4085753, 'name', name, true, false, false);
                                            if (J$.C(216456, J$.C(216448, J$.U(396746, '!', J$.R(4085585, 'name', name, false, false))) ? J$._() : J$.U(396754, '!', J$.F(4085609, J$.R(4085593, 'isVariable', isVariable, false, true), false)(J$.R(4085601, 'name', name, false, false)))))
                                                name = J$.W(4085625, 'name', J$.T(4085617, 'tmp', 21, false), name, false, false);
                                            if (J$.C(216464, J$.U(396762, '!', J$.G(4085649, J$.R(4085633, 'vars', vars, false, false), J$.R(4085641, 'name', name, false, false)))))
                                                J$.P(4085681, J$.R(4085657, 'vars', vars, false, false), J$.R(4085665, 'name', name, false, false), J$.T(4085673, 0, 22, false));
                                            return J$.Rt(4085729, J$.B(396778, '+', J$.R(4085689, 'name', name, false, false), J$.C(216472, J$.B(396770, '-', J$.A(4085713, J$.R(4085697, 'vars', vars, false, false), J$.R(4085705, 'name', name, false, false), '+')(1), 1)) ? J$._() : J$.T(4085721, '', 21, false)));
                                        } catch (J$e) {
                                            J$.Ex(4087081, J$e);
                                        } finally {
                                            if (J$.Fr(4087089))
                                                continue jalangiLabel7;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.P(4086065, J$.R(4085777, 'line', line, false, false), 'property', J$.T(4086057, function (obj, name) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(4086025, arguments.callee, this, arguments);
                                            arguments = J$.N(4086033, 'arguments', arguments, true, false, false);
                                            obj = J$.N(4086041, 'obj', obj, true, false, false);
                                            name = J$.N(4086049, 'name', name, true, false, false);
                                            if (J$.C(216480, J$.B(396786, '===', J$.G(4085793, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4085785, 'arguments', undefined, true, true) : arguments = J$.R(4085785, 'arguments', arguments, true, true)), 'length'), J$.T(4085801, 1, 22, false)))) {
                                                name = J$.W(4085817, 'name', J$.R(4085809, 'obj', obj, false, false), name, false, false);
                                                obj = J$.W(4085833, 'obj', J$.T(4085825, '', 21, false), obj, false, false);
                                            }
                                            name = J$.W(4085857, 'name', J$.B(396794, '+', J$.R(4085841, 'name', name, false, false), J$.T(4085849, '', 21, false)), name, false, false);
                                            if (J$.C(216496, J$.F(4085881, J$.R(4085865, 'isProperty', isProperty, false, true), false)(J$.R(4085873, 'name', name, false, false))))
                                                return J$.Rt(4085929, J$.C(216488, J$.R(4085889, 'obj', obj, false, false)) ? J$.B(396810, '+', J$.B(396802, '+', J$.R(4085897, 'obj', obj, false, false), J$.T(4085905, '.', 21, false)), J$.R(4085913, 'name', name, false, false)) : J$.R(4085921, 'name', name, false, false));
                                            return J$.Rt(4086017, J$.C(216504, J$.R(4085937, 'obj', obj, false, false)) ? J$.B(396834, '+', J$.B(396826, '+', J$.B(396818, '+', J$.R(4085945, 'obj', obj, false, false), J$.T(4085953, '[', 21, false)), J$.M(4085977, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(4085961, 'JSON', undefined, true, true) : JSON = J$.R(4085961, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(4085969, 'name', name, false, false))), J$.T(4085985, ']', 21, false)) : J$.M(4086009, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(4085993, 'JSON', undefined, true, true) : JSON = J$.R(4085993, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(4086001, 'name', name, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(4087097, J$e);
                                        } finally {
                                            if (J$.Fr(4087105))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.P(4086137, J$.R(4086073, 'line', line, false, false), 'toString', J$.T(4086129, function () {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(4086113, arguments.callee, this, arguments);
                                            arguments = J$.N(4086121, 'arguments', arguments, true, false, false);
                                            return J$.Rt(4086105, J$.M(4086097, J$.R(4086081, 'lines', lines, false, false), 'join', false)(J$.T(4086089, '\n', 21, false)));
                                        } catch (J$e) {
                                            J$.Ex(4087113, J$e);
                                        } finally {
                                            if (J$.Fr(4087121))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.P(4086681, J$.R(4086145, 'line', line, false, false), 'toFunction', J$.T(4086673, function (scope) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(4086625, arguments.callee, this, arguments);
                                            arguments = J$.N(4086633, 'arguments', arguments, true, false, false);
                                            scope = J$.N(4086641, 'scope', scope, true, false, false);
                                            J$.N(4086649, 'src', src, false, false, false);
                                            J$.N(4086657, 'keys', keys, false, false, false);
                                            J$.N(4086665, 'vals', vals, false, false, false);
                                            if (J$.C(216512, J$.U(396842, '!', J$.R(4086153, 'scope', scope, false, false))))
                                                scope = J$.W(4086169, 'scope', J$.T(4086161, {}, 11, false), scope, false, false);
                                            var src = J$.W(4086209, 'src', J$.B(396858, '+', J$.B(396850, '+', J$.T(4086177, 'return (', 21, false), J$.M(4086193, J$.R(4086185, 'line', line, false, false), 'toString', false)()), J$.T(4086201, ')', 21, false)), src, false, false);
                                            J$.M(4086361, J$.M(4086241, J$.I(typeof Object === 'undefined' ? Object = J$.R(4086217, 'Object', undefined, true, true) : Object = J$.R(4086217, 'Object', Object, true, true)), 'keys', false)(J$.G(4086233, J$.R(4086225, 'line', line, false, false), 'scope')), 'forEach', false)(J$.T(4086353, function (key) {
                                                jalangiLabel10:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(4086329, arguments.callee, this, arguments);
                                                            arguments = J$.N(4086337, 'arguments', arguments, true, false, false);
                                                            key = J$.N(4086345, 'key', key, true, false, false);
                                                            if (J$.C(216520, J$.U(396866, '!', J$.G(4086265, J$.R(4086249, 'scope', scope, false, false), J$.R(4086257, 'key', key, false, false)))))
                                                                J$.P(4086321, J$.R(4086273, 'scope', scope, false, false), J$.R(4086281, 'key', key, false, false), J$.G(4086313, J$.G(4086297, J$.R(4086289, 'line', line, false, false), 'scope'), J$.R(4086305, 'key', key, false, false)));
                                                        } catch (J$e) {
                                                            J$.Ex(4087129, J$e);
                                                        } finally {
                                                            if (J$.Fr(4087137))
                                                                continue jalangiLabel10;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                            var keys = J$.W(4086449, 'keys', J$.M(4086441, J$.M(4086385, J$.I(typeof Object === 'undefined' ? Object = J$.R(4086369, 'Object', undefined, true, true) : Object = J$.R(4086369, 'Object', Object, true, true)), 'keys', false)(J$.R(4086377, 'scope', scope, false, false)), 'map', false)(J$.T(4086433, function (key) {
                                                jalangiLabel11:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(4086409, arguments.callee, this, arguments);
                                                            arguments = J$.N(4086417, 'arguments', arguments, true, false, false);
                                                            key = J$.N(4086425, 'key', key, true, false, false);
                                                            return J$.Rt(4086401, J$.R(4086393, 'key', key, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(4087145, J$e);
                                                        } finally {
                                                            if (J$.Fr(4087153))
                                                                continue jalangiLabel11;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)), keys, false, false);
                                            var vals = J$.W(4086537, 'vals', J$.M(4086529, J$.R(4086457, 'keys', keys, false, false), 'map', false)(J$.T(4086521, function (key) {
                                                jalangiLabel12:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(4086497, arguments.callee, this, arguments);
                                                            arguments = J$.N(4086505, 'arguments', arguments, true, false, false);
                                                            key = J$.N(4086513, 'key', key, true, false, false);
                                                            return J$.Rt(4086489, J$.G(4086481, J$.R(4086465, 'scope', scope, false, false), J$.R(4086473, 'key', key, false, false)));
                                                        } catch (J$e) {
                                                            J$.Ex(4087161, J$e);
                                                        } finally {
                                                            if (J$.Fr(4087169))
                                                                continue jalangiLabel12;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)), vals, false, false);
                                            return J$.Rt(4086617, J$.M(4086609, J$.M(4086585, J$.I(typeof Function === 'undefined' ? Function = J$.R(4086545, 'Function', undefined, true, true) : Function = J$.R(4086545, 'Function', Function, true, true)), 'apply', false)(J$.T(4086553, null, 25, false), J$.M(4086577, J$.R(4086561, 'keys', keys, false, false), 'concat', false)(J$.R(4086569, 'src', src, false, false))), 'apply', false)(J$.T(4086593, null, 25, false), J$.R(4086601, 'vals', vals, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(4087177, J$e);
                                        } finally {
                                            if (J$.Fr(4087185))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            if (J$.C(216528, J$.G(4086697, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4086689, 'arguments', undefined, true, true) : arguments = J$.R(4086689, 'arguments', arguments, true, true)), 'length')))
                                J$.M(4086729, J$.R(4086705, 'line', line, false, false), 'apply', false)(J$.T(4086713, null, 25, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(4086721, 'arguments', undefined, true, true) : arguments = J$.R(4086721, 'arguments', arguments, true, true)));
                            return J$.Rt(4086745, J$.R(4086737, 'line', line, false, false));
                        } catch (J$e) {
                            J$.Ex(4087193, J$e);
                        } finally {
                            if (J$.Fr(4087201))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), genfun, false, true);
            J$.P(4086849, J$.R(4086833, 'genfun', genfun, false, true), 'formats', J$.R(4086841, 'formats', formats, false, true));
            J$.P(4086873, J$.I(typeof module === 'undefined' ? module = J$.R(4086857, 'module', undefined, true, true) : module = J$.R(4086857, 'module', module, true, true)), 'exports', J$.R(4086865, 'genfun', genfun, false, true));
        } catch (J$e) {
            J$.Ex(4087209, J$e);
        } finally {
            if (J$.Sr(4087217))
                continue jalangiLabel15;
            else
                break jalangiLabel15;
        }
    }
// JALANGI DO NOT INSTRUMENT
