// generate-function/index.js
J$.noInstrEval = false;
jalangiLabel15:
    while (true) {
        try {
            J$.Se(171249, '/mnt/data/fxiao/HiPar/outputs/target_cache/TestIsMyJsonValid/node_modules/generate-function/index.js');
            J$.N(171257, 'util', util, false, false, false);
            J$.N(171265, 'isProperty', isProperty, false, false, false);
            J$.N(171273, 'INDENT_START', INDENT_START, false, false, false);
            J$.N(171281, 'INDENT_END', INDENT_END, false, false, false);
            J$.N(171289, 'RESERVED', RESERVED, false, false, false);
            J$.N(171297, 'RESERVED_MAP', RESERVED_MAP, false, false, false);
            J$.N(171305, 'i', i, false, false, false);
            J$.N(171313, 'isVariable', isVariable, false, false, false);
            J$.N(171321, 'formats', formats, false, false, false);
            J$.N(171329, 'genfun', genfun, false, false, false);
            J$.T(167633, 'use strict', 21, false);
            J$.F(167657, J$.I(typeof require === 'undefined' ? require = J$.R(167641, 'require', undefined, true, true) : require = J$.R(167641, 'require', require, true, true)), false)(J$.T(167649, 'core-js/modules/es.array.concat', 21, false));
            J$.F(167681, J$.I(typeof require === 'undefined' ? require = J$.R(167665, 'require', undefined, true, true) : require = J$.R(167665, 'require', require, true, true)), false)(J$.T(167673, 'core-js/modules/es.array.for-each', 21, false));
            J$.F(167705, J$.I(typeof require === 'undefined' ? require = J$.R(167689, 'require', undefined, true, true) : require = J$.R(167689, 'require', require, true, true)), false)(J$.T(167697, 'core-js/modules/es.array.index-of', 21, false));
            J$.F(167729, J$.I(typeof require === 'undefined' ? require = J$.R(167713, 'require', undefined, true, true) : require = J$.R(167713, 'require', require, true, true)), false)(J$.T(167721, 'core-js/modules/es.array.join', 21, false));
            J$.F(167753, J$.I(typeof require === 'undefined' ? require = J$.R(167737, 'require', undefined, true, true) : require = J$.R(167737, 'require', require, true, true)), false)(J$.T(167745, 'core-js/modules/es.array.map', 21, false));
            J$.F(167777, J$.I(typeof require === 'undefined' ? require = J$.R(167761, 'require', undefined, true, true) : require = J$.R(167761, 'require', require, true, true)), false)(J$.T(167769, 'core-js/modules/es.date.to-string', 21, false));
            J$.F(167801, J$.I(typeof require === 'undefined' ? require = J$.R(167785, 'require', undefined, true, true) : require = J$.R(167785, 'require', require, true, true)), false)(J$.T(167793, 'core-js/modules/es.number.constructor', 21, false));
            J$.F(167825, J$.I(typeof require === 'undefined' ? require = J$.R(167809, 'require', undefined, true, true) : require = J$.R(167809, 'require', require, true, true)), false)(J$.T(167817, 'core-js/modules/es.object.keys', 21, false));
            J$.F(167849, J$.I(typeof require === 'undefined' ? require = J$.R(167833, 'require', undefined, true, true) : require = J$.R(167833, 'require', require, true, true)), false)(J$.T(167841, 'core-js/modules/es.object.to-string', 21, false));
            J$.F(167873, J$.I(typeof require === 'undefined' ? require = J$.R(167857, 'require', undefined, true, true) : require = J$.R(167857, 'require', require, true, true)), false)(J$.T(167865, 'core-js/modules/es.regexp.exec', 21, false));
            J$.F(167897, J$.I(typeof require === 'undefined' ? require = J$.R(167881, 'require', undefined, true, true) : require = J$.R(167881, 'require', require, true, true)), false)(J$.T(167889, 'core-js/modules/es.regexp.to-string', 21, false));
            J$.F(167921, J$.I(typeof require === 'undefined' ? require = J$.R(167905, 'require', undefined, true, true) : require = J$.R(167905, 'require', require, true, true)), false)(J$.T(167913, 'core-js/modules/es.string.split', 21, false));
            J$.F(167945, J$.I(typeof require === 'undefined' ? require = J$.R(167929, 'require', undefined, true, true) : require = J$.R(167929, 'require', require, true, true)), false)(J$.T(167937, 'core-js/modules/es.string.trim', 21, false));
            J$.F(167969, J$.I(typeof require === 'undefined' ? require = J$.R(167953, 'require', undefined, true, true) : require = J$.R(167953, 'require', require, true, true)), false)(J$.T(167961, 'core-js/modules/web.dom-collections.for-each', 21, false));
            var util = J$.W(168001, 'util', J$.F(167993, J$.I(typeof require === 'undefined' ? require = J$.R(167977, 'require', undefined, true, true) : require = J$.R(167977, 'require', require, true, true)), false)(J$.T(167985, 'util', 21, false)), util, false, true);
            var isProperty = J$.W(168033, 'isProperty', J$.F(168025, J$.I(typeof require === 'undefined' ? require = J$.R(168009, 'require', undefined, true, true) : require = J$.R(168009, 'require', require, true, true)), false)(J$.T(168017, 'is-property', 21, false)), isProperty, false, true);
            var INDENT_START = J$.W(168049, 'INDENT_START', J$.T(168041, /[\{\[]/, 14, false), INDENT_START, false, true);
            var INDENT_END = J$.W(168065, 'INDENT_END', J$.T(168057, /[\}\]]/, 14, false), INDENT_END, false, true);
            var RESERVED = J$.W(168481, 'RESERVED', J$.T(168473, [
                J$.T(168073, 'do', 21, false),
                J$.T(168081, 'if', 21, false),
                J$.T(168089, 'in', 21, false),
                J$.T(168097, 'for', 21, false),
                J$.T(168105, 'let', 21, false),
                J$.T(168113, 'new', 21, false),
                J$.T(168121, 'try', 21, false),
                J$.T(168129, 'var', 21, false),
                J$.T(168137, 'case', 21, false),
                J$.T(168145, 'else', 21, false),
                J$.T(168153, 'enum', 21, false),
                J$.T(168161, 'eval', 21, false),
                J$.T(168169, 'null', 21, false),
                J$.T(168177, 'this', 21, false),
                J$.T(168185, 'true', 21, false),
                J$.T(168193, 'void', 21, false),
                J$.T(168201, 'with', 21, false),
                J$.T(168209, 'await', 21, false),
                J$.T(168217, 'break', 21, false),
                J$.T(168225, 'catch', 21, false),
                J$.T(168233, 'class', 21, false),
                J$.T(168241, 'const', 21, false),
                J$.T(168249, 'false', 21, false),
                J$.T(168257, 'super', 21, false),
                J$.T(168265, 'throw', 21, false),
                J$.T(168273, 'while', 21, false),
                J$.T(168281, 'yield', 21, false),
                J$.T(168289, 'delete', 21, false),
                J$.T(168297, 'export', 21, false),
                J$.T(168305, 'import', 21, false),
                J$.T(168313, 'public', 21, false),
                J$.T(168321, 'return', 21, false),
                J$.T(168329, 'static', 21, false),
                J$.T(168337, 'switch', 21, false),
                J$.T(168345, 'typeof', 21, false),
                J$.T(168353, 'default', 21, false),
                J$.T(168361, 'extends', 21, false),
                J$.T(168369, 'finally', 21, false),
                J$.T(168377, 'package', 21, false),
                J$.T(168385, 'private', 21, false),
                J$.T(168393, 'continue', 21, false),
                J$.T(168401, 'debugger', 21, false),
                J$.T(168409, 'function', 21, false),
                J$.T(168417, 'arguments', 21, false),
                J$.T(168425, 'interface', 21, false),
                J$.T(168433, 'protected', 21, false),
                J$.T(168441, 'implements', 21, false),
                J$.T(168449, 'instanceof', 21, false),
                J$.T(168457, 'NaN', 21, false),
                J$.T(168465, 'undefined', 21, false)
            ], 10, false), RESERVED, false, true);
            var RESERVED_MAP = J$.W(168497, 'RESERVED_MAP', J$.T(168489, {}, 11, false), RESERVED_MAP, false, true);
            for (var i = J$.W(168513, 'i', J$.T(168505, 0, 22, false), i, false, true); J$.C(10232, J$.B(13554, '<', J$.R(168521, 'i', i, false, true), J$.G(168537, J$.R(168529, 'RESERVED', RESERVED, false, true), 'length'))); J$.B(13578, '-', i = J$.W(168553, 'i', J$.B(13570, '+', J$.U(13562, '+', J$.R(168545, 'i', i, false, true)), 1), i, false, true), 1)) {
                J$.P(168601, J$.R(168561, 'RESERVED_MAP', RESERVED_MAP, false, true), J$.G(168585, J$.R(168569, 'RESERVED', RESERVED, false, true), J$.R(168577, 'i', i, false, true)), J$.T(168593, true, 23, false));
            }
            var isVariable = J$.W(168697, 'isVariable', J$.T(168689, function isVariable(name) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(168665, arguments.callee, this, arguments);
                            arguments = J$.N(168673, 'arguments', arguments, true, false, false);
                            name = J$.N(168681, 'name', name, true, false, false);
                            return J$.Rt(168657, J$.C(10240, J$.F(168625, J$.R(168609, 'isProperty', isProperty, false, true), false)(J$.R(168617, 'name', name, false, false))) ? J$.U(13586, '!', J$.M(168649, J$.R(168633, 'RESERVED_MAP', RESERVED_MAP, false, true), 'hasOwnProperty', false)(J$.R(168641, 'name', name, false, false))) : J$._());
                        } catch (J$e) {
                            J$.Ex(171337, J$e);
                        } finally {
                            if (J$.Fr(171345))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), isVariable, false, true);
            var formats = J$.W(168905, 'formats', J$.T(168897, {
                s: J$.T(168753, function s(_s) {
                    jalangiLabel1:
                        while (true) {
                            try {
                                J$.Fe(168729, arguments.callee, this, arguments);
                                arguments = J$.N(168737, 'arguments', arguments, true, false, false);
                                _s = J$.N(168745, '_s', _s, true, false, false);
                                return J$.Rt(168721, J$.B(13594, '+', J$.T(168705, '', 21, false), J$.R(168713, '_s', _s, false, false)));
                            } catch (J$e) {
                                J$.Ex(171353, J$e);
                            } finally {
                                if (J$.Fr(171361))
                                    continue jalangiLabel1;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                d: J$.T(168825, function d(_d) {
                    jalangiLabel2:
                        while (true) {
                            try {
                                J$.Fe(168801, arguments.callee, this, arguments);
                                arguments = J$.N(168809, 'arguments', arguments, true, false, false);
                                _d = J$.N(168817, '_d', _d, true, false, false);
                                return J$.Rt(168793, J$.B(13602, '+', J$.T(168761, '', 21, false), J$.F(168785, J$.I(typeof Number === 'undefined' ? Number = J$.R(168769, 'Number', undefined, true, true) : Number = J$.R(168769, 'Number', Number, true, true)), false)(J$.R(168777, '_d', _d, false, false))));
                            } catch (J$e) {
                                J$.Ex(171369, J$e);
                            } finally {
                                if (J$.Fr(171377))
                                    continue jalangiLabel2;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                o: J$.T(168889, function o(_o) {
                    jalangiLabel3:
                        while (true) {
                            try {
                                J$.Fe(168865, arguments.callee, this, arguments);
                                arguments = J$.N(168873, 'arguments', arguments, true, false, false);
                                _o = J$.N(168881, '_o', _o, true, false, false);
                                return J$.Rt(168857, J$.M(168849, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(168833, 'JSON', undefined, true, true) : JSON = J$.R(168833, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(168841, '_o', _o, false, false)));
                            } catch (J$e) {
                                J$.Ex(171385, J$e);
                            } finally {
                                if (J$.Fr(171393))
                                    continue jalangiLabel3;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false), formats, false, true);
            var genfun = J$.W(171193, 'genfun', J$.T(171185, function genfun() {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(171121, arguments.callee, this, arguments);
                            arguments = J$.N(171129, 'arguments', arguments, true, false, false);
                            J$.N(171137, 'lines', lines, false, false, false);
                            J$.N(171145, 'indent', indent, false, false, false);
                            J$.N(171153, 'vars', vars, false, false, false);
                            J$.N(171161, 'push', push, false, false, false);
                            J$.N(171169, 'pushLine', pushLine, false, false, false);
                            J$.N(171177, 'line', line, false, false, false);
                            var lines = J$.W(168921, 'lines', J$.T(168913, [], 10, false), lines, false, false);
                            var indent = J$.W(168937, 'indent', J$.T(168929, 0, 22, false), indent, false, false);
                            var vars = J$.W(168953, 'vars', J$.T(168945, {}, 11, false), vars, false, false);
                            var push = J$.W(169105, 'push', J$.T(169097, function push(str) {
                                jalangiLabel4:
                                    while (true) {
                                        try {
                                            J$.Fe(169065, arguments.callee, this, arguments);
                                            arguments = J$.N(169073, 'arguments', arguments, true, false, false);
                                            str = J$.N(169081, 'str', str, true, false, false);
                                            J$.N(169089, 'spaces', spaces, false, false, false);
                                            var spaces = J$.W(168969, 'spaces', J$.T(168961, '', 21, false), spaces, false, false);
                                            while (J$.C(10248, J$.B(13618, '<', J$.G(168985, J$.R(168977, 'spaces', spaces, false, false), 'length'), J$.B(13610, '*', J$.R(168993, 'indent', indent, false, false), J$.T(169001, 2, 22, false))))) {
                                                spaces = J$.W(169025, 'spaces', J$.B(13626, '+', J$.R(169017, 'spaces', spaces, false, false), J$.T(169009, '  ', 21, false)), spaces, false, false);
                                            }
                                            J$.M(169057, J$.R(169033, 'lines', lines, false, false), 'push', false)(J$.B(13634, '+', J$.R(169041, 'spaces', spaces, false, false), J$.R(169049, 'str', str, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(171401, J$e);
                                        } finally {
                                            if (J$.Fr(171409))
                                                continue jalangiLabel4;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), push, false, false);
                            var pushLine = J$.W(169537, 'pushLine', J$.T(169529, function pushLine(line) {
                                jalangiLabel5:
                                    while (true) {
                                        try {
                                            J$.Fe(169505, arguments.callee, this, arguments);
                                            arguments = J$.N(169513, 'arguments', arguments, true, false, false);
                                            line = J$.N(169521, 'line', line, true, false, false);
                                            if (J$.C(10264, J$.C(10256, J$.M(169153, J$.R(169113, 'INDENT_END', INDENT_END, false, true), 'test', false)(J$.G(169145, J$.M(169129, J$.R(169121, 'line', line, false, false), 'trim', false)(), J$.T(169137, 0, 22, false)))) ? J$.M(169209, J$.R(169161, 'INDENT_START', INDENT_START, false, true), 'test', false)(J$.G(169201, J$.R(169169, 'line', line, false, false), J$.B(13642, '-', J$.G(169185, J$.R(169177, 'line', line, false, false), 'length'), J$.T(169193, 1, 22, false)))) : J$._())) {
                                                J$.B(13666, '+', indent = J$.W(169225, 'indent', J$.B(13658, '-', J$.U(13650, '+', J$.R(169217, 'indent', indent, false, false)), 1), indent, false, false), 1);
                                                J$.F(169249, J$.R(169233, 'push', push, false, false), false)(J$.R(169241, 'line', line, false, false));
                                                J$.B(13690, '-', indent = J$.W(169265, 'indent', J$.B(13682, '+', J$.U(13674, '+', J$.R(169257, 'indent', indent, false, false)), 1), indent, false, false), 1);
                                                return J$.Rt(169273, undefined);
                                            }
                                            if (J$.C(10272, J$.M(169329, J$.R(169281, 'INDENT_START', INDENT_START, false, true), 'test', false)(J$.G(169321, J$.R(169289, 'line', line, false, false), J$.B(13698, '-', J$.G(169305, J$.R(169297, 'line', line, false, false), 'length'), J$.T(169313, 1, 22, false)))))) {
                                                J$.F(169353, J$.R(169337, 'push', push, false, false), false)(J$.R(169345, 'line', line, false, false));
                                                J$.B(13722, '-', indent = J$.W(169369, 'indent', J$.B(13714, '+', J$.U(13706, '+', J$.R(169361, 'indent', indent, false, false)), 1), indent, false, false), 1);
                                                return J$.Rt(169377, undefined);
                                            }
                                            if (J$.C(10280, J$.M(169425, J$.R(169385, 'INDENT_END', INDENT_END, false, true), 'test', false)(J$.G(169417, J$.M(169401, J$.R(169393, 'line', line, false, false), 'trim', false)(), J$.T(169409, 0, 22, false))))) {
                                                J$.B(13746, '+', indent = J$.W(169441, 'indent', J$.B(13738, '-', J$.U(13730, '+', J$.R(169433, 'indent', indent, false, false)), 1), indent, false, false), 1);
                                                J$.F(169465, J$.R(169449, 'push', push, false, false), false)(J$.R(169457, 'line', line, false, false));
                                                return J$.Rt(169473, undefined);
                                            }
                                            J$.F(169497, J$.R(169481, 'push', push, false, false), false)(J$.R(169489, 'line', line, false, false));
                                        } catch (J$e) {
                                            J$.Ex(171417, J$e);
                                        } finally {
                                            if (J$.Fr(171425))
                                                continue jalangiLabel5;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), pushLine, false, false);
                            var line = J$.W(169889, 'line', J$.T(169881, function line(fmt) {
                                jalangiLabel6:
                                    while (true) {
                                        try {
                                            J$.Fe(169841, arguments.callee, this, arguments);
                                            arguments = J$.N(169849, 'arguments', arguments, true, false, false);
                                            fmt = J$.N(169857, 'fmt', fmt, true, false, false);
                                            J$.N(169865, 'lines', lines, false, false, false);
                                            J$.N(169873, 'i', i, false, false, false);
                                            if (J$.C(10288, J$.U(13754, '!', J$.R(169545, 'fmt', fmt, false, false))))
                                                return J$.Rt(169561, J$.R(169553, 'line', line, false, false));
                                            if (J$.C(10312, J$.C(10296, J$.B(13762, '===', J$.G(169577, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(169569, 'arguments', undefined, true, true) : arguments = J$.R(169569, 'arguments', arguments, true, true)), 'length'), J$.T(169585, 1, 22, false))) ? J$.B(13778, '>', J$.M(169609, J$.R(169593, 'fmt', fmt, false, false), 'indexOf', false)(J$.T(169601, '\n', 21, false)), J$.U(13770, '-', J$.T(169617, 1, 22, false))) : J$._())) {
                                                var lines = J$.W(169657, 'lines', J$.M(169649, J$.M(169633, J$.R(169625, 'fmt', fmt, false, false), 'trim', false)(), 'split', false)(J$.T(169641, '\n', 21, false)), lines, false, false);
                                                for (var i = J$.W(169673, 'i', J$.T(169665, 0, 22, false), i, false, false); J$.C(10304, J$.B(13786, '<', J$.R(169681, 'i', i, false, false), J$.G(169697, J$.R(169689, 'lines', lines, false, false), 'length'))); J$.B(13810, '-', i = J$.W(169713, 'i', J$.B(13802, '+', J$.U(13794, '+', J$.R(169705, 'i', i, false, false)), 1), i, false, false), 1)) {
                                                    J$.F(169761, J$.R(169721, 'pushLine', pushLine, false, false), false)(J$.M(169753, J$.G(169745, J$.R(169729, 'lines', lines, false, false), J$.R(169737, 'i', i, false, false)), 'trim', false)());
                                                }
                                            } else {
                                                J$.F(169817, J$.R(169769, 'pushLine', pushLine, false, false), false)(J$.M(169809, J$.G(169785, J$.R(169777, 'util', util, false, true), 'format'), 'apply', false)(J$.R(169793, 'util', util, false, true), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(169801, 'arguments', undefined, true, true) : arguments = J$.R(169801, 'arguments', arguments, true, true))));
                                            }
                                            return J$.Rt(169833, J$.R(169825, 'line', line, false, false));
                                        } catch (J$e) {
                                            J$.Ex(171433, J$e);
                                        } finally {
                                            if (J$.Fr(171441))
                                                continue jalangiLabel6;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), line, false, false);
                            J$.P(169913, J$.R(169897, 'line', line, false, false), 'scope', J$.T(169905, {}, 11, false));
                            J$.P(169937, J$.R(169921, 'line', line, false, false), 'formats', J$.R(169929, 'formats', formats, false, true));
                            J$.P(170137, J$.R(169945, 'line', line, false, false), 'sym', J$.T(170129, function (name) {
                                jalangiLabel7:
                                    while (true) {
                                        try {
                                            J$.Fe(170105, arguments.callee, this, arguments);
                                            arguments = J$.N(170113, 'arguments', arguments, true, false, false);
                                            name = J$.N(170121, 'name', name, true, false, false);
                                            if (J$.C(10328, J$.C(10320, J$.U(13818, '!', J$.R(169953, 'name', name, false, false))) ? J$._() : J$.U(13826, '!', J$.F(169977, J$.R(169961, 'isVariable', isVariable, false, true), false)(J$.R(169969, 'name', name, false, false)))))
                                                name = J$.W(169993, 'name', J$.T(169985, 'tmp', 21, false), name, false, false);
                                            if (J$.C(10336, J$.U(13834, '!', J$.G(170017, J$.R(170001, 'vars', vars, false, false), J$.R(170009, 'name', name, false, false)))))
                                                J$.P(170049, J$.R(170025, 'vars', vars, false, false), J$.R(170033, 'name', name, false, false), J$.T(170041, 0, 22, false));
                                            return J$.Rt(170097, J$.B(13850, '+', J$.R(170057, 'name', name, false, false), J$.C(10344, J$.B(13842, '-', J$.A(170081, J$.R(170065, 'vars', vars, false, false), J$.R(170073, 'name', name, false, false), '+')(1), 1)) ? J$._() : J$.T(170089, '', 21, false)));
                                        } catch (J$e) {
                                            J$.Ex(171449, J$e);
                                        } finally {
                                            if (J$.Fr(171457))
                                                continue jalangiLabel7;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.P(170433, J$.R(170145, 'line', line, false, false), 'property', J$.T(170425, function (obj, name) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(170393, arguments.callee, this, arguments);
                                            arguments = J$.N(170401, 'arguments', arguments, true, false, false);
                                            obj = J$.N(170409, 'obj', obj, true, false, false);
                                            name = J$.N(170417, 'name', name, true, false, false);
                                            if (J$.C(10352, J$.B(13858, '===', J$.G(170161, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(170153, 'arguments', undefined, true, true) : arguments = J$.R(170153, 'arguments', arguments, true, true)), 'length'), J$.T(170169, 1, 22, false)))) {
                                                name = J$.W(170185, 'name', J$.R(170177, 'obj', obj, false, false), name, false, false);
                                                obj = J$.W(170201, 'obj', J$.T(170193, '', 21, false), obj, false, false);
                                            }
                                            name = J$.W(170225, 'name', J$.B(13866, '+', J$.R(170209, 'name', name, false, false), J$.T(170217, '', 21, false)), name, false, false);
                                            if (J$.C(10368, J$.F(170249, J$.R(170233, 'isProperty', isProperty, false, true), false)(J$.R(170241, 'name', name, false, false))))
                                                return J$.Rt(170297, J$.C(10360, J$.R(170257, 'obj', obj, false, false)) ? J$.B(13882, '+', J$.B(13874, '+', J$.R(170265, 'obj', obj, false, false), J$.T(170273, '.', 21, false)), J$.R(170281, 'name', name, false, false)) : J$.R(170289, 'name', name, false, false));
                                            return J$.Rt(170385, J$.C(10376, J$.R(170305, 'obj', obj, false, false)) ? J$.B(13906, '+', J$.B(13898, '+', J$.B(13890, '+', J$.R(170313, 'obj', obj, false, false), J$.T(170321, '[', 21, false)), J$.M(170345, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(170329, 'JSON', undefined, true, true) : JSON = J$.R(170329, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(170337, 'name', name, false, false))), J$.T(170353, ']', 21, false)) : J$.M(170377, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(170361, 'JSON', undefined, true, true) : JSON = J$.R(170361, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(170369, 'name', name, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(171465, J$e);
                                        } finally {
                                            if (J$.Fr(171473))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.P(170505, J$.R(170441, 'line', line, false, false), 'toString', J$.T(170497, function () {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(170481, arguments.callee, this, arguments);
                                            arguments = J$.N(170489, 'arguments', arguments, true, false, false);
                                            return J$.Rt(170473, J$.M(170465, J$.R(170449, 'lines', lines, false, false), 'join', false)(J$.T(170457, '\n', 21, false)));
                                        } catch (J$e) {
                                            J$.Ex(171481, J$e);
                                        } finally {
                                            if (J$.Fr(171489))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.P(171049, J$.R(170513, 'line', line, false, false), 'toFunction', J$.T(171041, function (scope) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(170993, arguments.callee, this, arguments);
                                            arguments = J$.N(171001, 'arguments', arguments, true, false, false);
                                            scope = J$.N(171009, 'scope', scope, true, false, false);
                                            J$.N(171017, 'src', src, false, false, false);
                                            J$.N(171025, 'keys', keys, false, false, false);
                                            J$.N(171033, 'vals', vals, false, false, false);
                                            if (J$.C(10384, J$.U(13914, '!', J$.R(170521, 'scope', scope, false, false))))
                                                scope = J$.W(170537, 'scope', J$.T(170529, {}, 11, false), scope, false, false);
                                            var src = J$.W(170577, 'src', J$.B(13930, '+', J$.B(13922, '+', J$.T(170545, 'return (', 21, false), J$.M(170561, J$.R(170553, 'line', line, false, false), 'toString', false)()), J$.T(170569, ')', 21, false)), src, false, false);
                                            J$.M(170729, J$.M(170609, J$.I(typeof Object === 'undefined' ? Object = J$.R(170585, 'Object', undefined, true, true) : Object = J$.R(170585, 'Object', Object, true, true)), 'keys', false)(J$.G(170601, J$.R(170593, 'line', line, false, false), 'scope')), 'forEach', false)(J$.T(170721, function (key) {
                                                jalangiLabel10:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(170697, arguments.callee, this, arguments);
                                                            arguments = J$.N(170705, 'arguments', arguments, true, false, false);
                                                            key = J$.N(170713, 'key', key, true, false, false);
                                                            if (J$.C(10392, J$.U(13938, '!', J$.G(170633, J$.R(170617, 'scope', scope, false, false), J$.R(170625, 'key', key, false, false)))))
                                                                J$.P(170689, J$.R(170641, 'scope', scope, false, false), J$.R(170649, 'key', key, false, false), J$.G(170681, J$.G(170665, J$.R(170657, 'line', line, false, false), 'scope'), J$.R(170673, 'key', key, false, false)));
                                                        } catch (J$e) {
                                                            J$.Ex(171497, J$e);
                                                        } finally {
                                                            if (J$.Fr(171505))
                                                                continue jalangiLabel10;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                            var keys = J$.W(170817, 'keys', J$.M(170809, J$.M(170753, J$.I(typeof Object === 'undefined' ? Object = J$.R(170737, 'Object', undefined, true, true) : Object = J$.R(170737, 'Object', Object, true, true)), 'keys', false)(J$.R(170745, 'scope', scope, false, false)), 'map', false)(J$.T(170801, function (key) {
                                                jalangiLabel11:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(170777, arguments.callee, this, arguments);
                                                            arguments = J$.N(170785, 'arguments', arguments, true, false, false);
                                                            key = J$.N(170793, 'key', key, true, false, false);
                                                            return J$.Rt(170769, J$.R(170761, 'key', key, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(171513, J$e);
                                                        } finally {
                                                            if (J$.Fr(171521))
                                                                continue jalangiLabel11;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)), keys, false, false);
                                            var vals = J$.W(170905, 'vals', J$.M(170897, J$.R(170825, 'keys', keys, false, false), 'map', false)(J$.T(170889, function (key) {
                                                jalangiLabel12:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(170865, arguments.callee, this, arguments);
                                                            arguments = J$.N(170873, 'arguments', arguments, true, false, false);
                                                            key = J$.N(170881, 'key', key, true, false, false);
                                                            return J$.Rt(170857, J$.G(170849, J$.R(170833, 'scope', scope, false, false), J$.R(170841, 'key', key, false, false)));
                                                        } catch (J$e) {
                                                            J$.Ex(171529, J$e);
                                                        } finally {
                                                            if (J$.Fr(171537))
                                                                continue jalangiLabel12;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)), vals, false, false);
                                            return J$.Rt(170985, J$.M(170977, J$.M(170953, J$.I(typeof Function === 'undefined' ? Function = J$.R(170913, 'Function', undefined, true, true) : Function = J$.R(170913, 'Function', Function, true, true)), 'apply', false)(J$.T(170921, null, 25, false), J$.M(170945, J$.R(170929, 'keys', keys, false, false), 'concat', false)(J$.R(170937, 'src', src, false, false))), 'apply', false)(J$.T(170961, null, 25, false), J$.R(170969, 'vals', vals, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(171545, J$e);
                                        } finally {
                                            if (J$.Fr(171553))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            if (J$.C(10400, J$.G(171065, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(171057, 'arguments', undefined, true, true) : arguments = J$.R(171057, 'arguments', arguments, true, true)), 'length')))
                                J$.M(171097, J$.R(171073, 'line', line, false, false), 'apply', false)(J$.T(171081, null, 25, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(171089, 'arguments', undefined, true, true) : arguments = J$.R(171089, 'arguments', arguments, true, true)));
                            return J$.Rt(171113, J$.R(171105, 'line', line, false, false));
                        } catch (J$e) {
                            J$.Ex(171561, J$e);
                        } finally {
                            if (J$.Fr(171569))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), genfun, false, true);
            J$.P(171217, J$.R(171201, 'genfun', genfun, false, true), 'formats', J$.R(171209, 'formats', formats, false, true));
            J$.P(171241, J$.I(typeof module === 'undefined' ? module = J$.R(171225, 'module', undefined, true, true) : module = J$.R(171225, 'module', module, true, true)), 'exports', J$.R(171233, 'genfun', genfun, false, true));
        } catch (J$e) {
            J$.Ex(171577, J$e);
        } finally {
            if (J$.Sr(171585))
                continue jalangiLabel15;
            else
                break jalangiLabel15;
        }
    }
// JALANGI DO NOT INSTRUMENT
