// codemirror/src/line/utils_line.js
"use strict";

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.array.slice");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.array.slice");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.array.slice");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.array.slice");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.array.slice");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.array.slice");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.array.slice");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.array.slice");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.array.index-of");

require("core-js/modules/es.array.slice");

require("core-js/modules/es.object.define-property");

require("core-js/modules/es.array.slice");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getLine = getLine;
exports.getBetween = getBetween;
exports.getLines = getLines;
exports.updateLineHeight = updateLineHeight;
exports.lineNo = lineNo;
exports.lineAtHeight = lineAtHeight;
exports.isLine = isLine;
exports.lineNumberFor = lineNumberFor;

var _misc = require("../util/misc.js");

function getLine(doc, n) {
  n -= doc.first;
  if (n < 0 || n >= doc.size) throw new Error("There is no line " + (n + doc.first) + " in the document.");
  var chunk = doc;

  while (!chunk.lines) {
    for (var i = 0;; ++i) {
      var child = chunk.children[i],
          sz = child.chunkSize();

      if (n < sz) {
        chunk = child;
        break;
      }

      n -= sz;
    }
  }

  return chunk.lines[n];
} // Get the part of a document between two positions, as an array of
// strings.


function getBetween(doc, start, end) {
  var out = [],
      n = start.line;
  doc.iter(start.line, end.line + 1, function (line) {
    var text = line.text;
    if (n == end.line) text = text.slice(0, end.ch);
    if (n == start.line) text = text.slice(start.ch);
    out.push(text);
    ++n;
  });
  return out;
} // Get the lines between from and to, as array of strings.


function getLines(doc, from, to) {
  var out = [];
  doc.iter(from, to, function (line) {
    out.push(line.text);
  }); // iter aborts when callback returns truthy value

  return out;
} // Update the height of a line, propagating the height change
// upwards to parent nodes.


function updateLineHeight(line, height) {
  var diff = height - line.height;
  if (diff) for (var n = line; n; n = n.parent) {
    n.height += diff;
  }
} // Given a line object, find its line number by walking up through
// its parent links.


function lineNo(line) {
  if (line.parent == null) return null;
  var cur = line.parent,
      no = (0, _misc.indexOf)(cur.lines, line);

  for (var chunk = cur.parent; chunk; cur = chunk, chunk = chunk.parent) {
    for (var i = 0;; ++i) {
      if (chunk.children[i] == cur) break;
      no += chunk.children[i].chunkSize();
    }
  }

  return no + cur.first;
} // Find the line at the given vertical position, using the height
// information in the document tree.


function lineAtHeight(chunk, h) {
  var n = chunk.first;

  outer: do {
    for (var _i = 0; _i < chunk.children.length; ++_i) {
      var child = chunk.children[_i],
          ch = child.height;

      if (h < ch) {
        chunk = child;
        continue outer;
      }

      h -= ch;
      n += child.chunkSize();
    }

    return n;
  } while (!chunk.lines);

  var i = 0;

  for (; i < chunk.lines.length; ++i) {
    var line = chunk.lines[i],
        lh = line.height;
    if (h < lh) break;
    h -= lh;
  }

  return n + i;
}

function isLine(doc, l) {
  return l >= doc.first && l < doc.first + doc.size;
}

function lineNumberFor(options, i) {
  return String(options.lineNumberFormatter(i + options.firstLineNumber));
}
