// aerospike/examples/get.js
#!/usr/bin/env node
// *****************************************************************************
// Copyright 2013-2018 Aerospike, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License")
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// *****************************************************************************
//
"use strict";

var Aerospike = require('aerospike');

var shared = require('./shared');

shared.runner();

function get(client, argv) {
  var key, record;
  return regeneratorRuntime.async(function get$(_context) {
    while (1) {
      switch (_context.prev = _context.next) {
        case 0:
          key = new Aerospike.Key(argv.namespace, argv.set, argv.key);

          if (!argv.bins) {
            _context.next = 7;
            break;
          }

          _context.next = 4;
          return regeneratorRuntime.awrap(client.select(key, argv.bins));

        case 4:
          record = _context.sent;
          _context.next = 10;
          break;

        case 7:
          _context.next = 9;
          return regeneratorRuntime.awrap(client.get(key));

        case 9:
          record = _context.sent;

        case 10:
          console.info(record);

        case 11:
        case "end":
          return _context.stop();
      }
    }
  });
}

exports.command = 'get <key>';
exports.describe = 'Fetch a record from the database';
exports.handler = shared.run(get);
exports.builder = {
  bins: {
    desc: 'Select specific bins to fetch',
    type: 'array'
  }
};
