// color/index.js
J$.noInstrEval = false;
jalangiLabel46:
    while (true) {
        try {
            J$.Se(6998961, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/color/index.js');
            function Color(obj, model) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(6989897, arguments.callee, this, arguments);
                            arguments = J$.N(6989905, 'arguments', arguments, true, false, false);
                            obj = J$.N(6989913, 'obj', obj, true, false, false);
                            model = J$.N(6989921, 'model', model, true, false, false);
                            J$.N(6989929, 'i', i, false, false, false);
                            J$.N(6989937, 'channels', channels, false, false, false);
                            J$.N(6989945, 'result', result, false, false, false);
                            J$.N(6989953, 'newArr', newArr, false, false, false);
                            J$.N(6989961, 'keys', keys, false, false, false);
                            J$.N(6989969, 'hashedKeys', hashedKeys, false, false, false);
                            J$.N(6989977, 'labels', labels, false, false, false);
                            J$.N(6989985, 'color', color, false, false, false);
                            J$.N(6989993, 'limit', limit, false, false, false);
                            if (J$.C(377032, J$.U(708914, '!', J$.B(708906, 'instanceof', J$.R(6987809, 'this', this, false, false), J$.R(6987817, 'Color', Color, false, true))))) {
                                return J$.Rt(6987857, J$.F(6987849, J$.R(6987825, 'Color', Color, false, true), true)(J$.R(6987833, 'obj', obj, false, false), J$.R(6987841, 'model', model, false, false)));
                            }
                            if (J$.C(377048, J$.C(377040, J$.R(6987865, 'model', model, false, false)) ? J$.B(708922, 'in', J$.R(6987873, 'model', model, false, false), J$.R(6987881, 'skippedModels', skippedModels, false, true)) : J$._())) {
                                model = J$.W(6987897, 'model', J$.T(6987889, null, 25, false), model, false, false);
                            }
                            if (J$.C(377064, J$.C(377056, J$.R(6987905, 'model', model, false, false)) ? J$.U(708938, '!', J$.B(708930, 'in', J$.R(6987913, 'model', model, false, false), J$.R(6987921, 'convert', convert, false, true))) : J$._())) {
                                throw J$.F(6987953, J$.I(typeof Error === 'undefined' ? Error = J$.R(6987929, 'Error', undefined, true, true) : Error = J$.R(6987929, 'Error', Error, true, true)), true)(J$.B(708946, '+', J$.T(6987937, 'Unknown model: ', 21, false), J$.R(6987945, 'model', model, false, false)));
                            }
                            var i;
                            var channels;
                            if (J$.C(377168, J$.U(708954, '!', J$.R(6987961, 'obj', obj, false, false)))) {
                                J$.P(6987985, J$.R(6987969, 'this', this, false, false), 'model', J$.T(6987977, 'rgb', 21, false));
                                J$.P(6988033, J$.R(6987993, 'this', this, false, false), 'color', J$.T(6988025, [
                                    J$.T(6988001, 0, 22, false),
                                    J$.T(6988009, 0, 22, false),
                                    J$.T(6988017, 0, 22, false)
                                ], 10, false));
                                J$.P(6988057, J$.R(6988041, 'this', this, false, false), 'valpha', J$.T(6988049, 1, 22, false));
                            } else if (J$.C(377160, J$.B(708962, 'instanceof', J$.R(6988065, 'obj', obj, false, false), J$.R(6988073, 'Color', Color, false, true)))) {
                                J$.P(6988105, J$.R(6988081, 'this', this, false, false), 'model', J$.G(6988097, J$.R(6988089, 'obj', obj, false, false), 'model'));
                                J$.P(6988145, J$.R(6988113, 'this', this, false, false), 'color', J$.M(6988137, J$.G(6988129, J$.R(6988121, 'obj', obj, false, false), 'color'), 'slice', false)());
                                J$.P(6988177, J$.R(6988153, 'this', this, false, false), 'valpha', J$.G(6988169, J$.R(6988161, 'obj', obj, false, false), 'valpha'));
                            } else if (J$.C(377152, J$.B(708978, '===', J$.U(708970, 'typeof', J$.R(6988185, 'obj', obj, false, false)), J$.T(6988193, 'string', 21, false)))) {
                                var result = J$.W(6988225, 'result', J$.M(6988217, J$.R(6988201, 'colorString', colorString, false, true), 'get', false)(J$.R(6988209, 'obj', obj, false, false)), result, false, false);
                                if (J$.C(377072, J$.B(708986, '===', J$.R(6988233, 'result', result, false, false), J$.T(6988241, null, 25, false)))) {
                                    throw J$.F(6988273, J$.I(typeof Error === 'undefined' ? Error = J$.R(6988249, 'Error', undefined, true, true) : Error = J$.R(6988249, 'Error', Error, true, true)), true)(J$.B(708994, '+', J$.T(6988257, 'Unable to parse color from string: ', 21, false), J$.R(6988265, 'obj', obj, false, false)));
                                }
                                J$.P(6988305, J$.R(6988281, 'this', this, false, false), 'model', J$.G(6988297, J$.R(6988289, 'result', result, false, false), 'model'));
                                channels = J$.W(6988353, 'channels', J$.G(6988345, J$.G(6988337, J$.R(6988313, 'convert', convert, false, true), J$.G(6988329, J$.R(6988321, 'this', this, false, false), 'model')), 'channels'), channels, false, false);
                                J$.P(6988409, J$.R(6988361, 'this', this, false, false), 'color', J$.M(6988401, J$.G(6988377, J$.R(6988369, 'result', result, false, false), 'value'), 'slice', false)(J$.T(6988385, 0, 22, false), J$.R(6988393, 'channels', channels, false, false)));
                                J$.P(6988505, J$.R(6988417, 'this', this, false, false), 'valpha', J$.C(377080, J$.B(709010, '===', J$.U(709002, 'typeof', J$.G(6988449, J$.G(6988433, J$.R(6988425, 'result', result, false, false), 'value'), J$.R(6988441, 'channels', channels, false, false))), J$.T(6988457, 'number', 21, false))) ? J$.G(6988489, J$.G(6988473, J$.R(6988465, 'result', result, false, false), 'value'), J$.R(6988481, 'channels', channels, false, false)) : J$.T(6988497, 1, 22, false));
                            } else if (J$.C(377144, J$.G(6988521, J$.R(6988513, 'obj', obj, false, false), 'length'))) {
                                J$.P(6988553, J$.R(6988529, 'this', this, false, false), 'model', J$.C(377088, J$.R(6988537, 'model', model, false, false)) ? J$._() : J$.T(6988545, 'rgb', 21, false));
                                channels = J$.W(6988601, 'channels', J$.G(6988593, J$.G(6988585, J$.R(6988561, 'convert', convert, false, true), J$.G(6988577, J$.R(6988569, 'this', this, false, false), 'model')), 'channels'), channels, false, false);
                                var newArr = J$.W(6988649, 'newArr', J$.M(6988641, J$.R(6988609, '_slice', _slice, false, true), 'call', false)(J$.R(6988617, 'obj', obj, false, false), J$.T(6988625, 0, 22, false), J$.R(6988633, 'channels', channels, false, false)), newArr, false, false);
                                J$.P(6988697, J$.R(6988657, 'this', this, false, false), 'color', J$.F(6988689, J$.R(6988665, 'zeroArray', zeroArray, false, true), false)(J$.R(6988673, 'newArr', newArr, false, false), J$.R(6988681, 'channels', channels, false, false)));
                                J$.P(6988777, J$.R(6988705, 'this', this, false, false), 'valpha', J$.C(377096, J$.B(709026, '===', J$.U(709018, 'typeof', J$.G(6988729, J$.R(6988713, 'obj', obj, false, false), J$.R(6988721, 'channels', channels, false, false))), J$.T(6988737, 'number', 21, false))) ? J$.G(6988761, J$.R(6988745, 'obj', obj, false, false), J$.R(6988753, 'channels', channels, false, false)) : J$.T(6988769, 1, 22, false));
                            } else if (J$.C(377136, J$.B(709042, '===', J$.U(709034, 'typeof', J$.R(6988785, 'obj', obj, false, false)), J$.T(6988793, 'number', 21, false)))) {
                                obj = J$.W(6988817, 'obj', J$.B(709050, '&', J$.R(6988809, 'obj', obj, false, false), J$.T(6988801, 16777215, 22, false)), obj, false, false);
                                J$.P(6988841, J$.R(6988825, 'this', this, false, false), 'model', J$.T(6988833, 'rgb', 21, false));
                                J$.P(6988929, J$.R(6988849, 'this', this, false, false), 'color', J$.T(6988921, [
                                    J$.B(709066, '&', J$.B(709058, '>>', J$.R(6988857, 'obj', obj, false, false), J$.T(6988865, 16, 22, false)), J$.T(6988873, 255, 22, false)),
                                    J$.B(709082, '&', J$.B(709074, '>>', J$.R(6988881, 'obj', obj, false, false), J$.T(6988889, 8, 22, false)), J$.T(6988897, 255, 22, false)),
                                    J$.B(709090, '&', J$.R(6988905, 'obj', obj, false, false), J$.T(6988913, 255, 22, false))
                                ], 10, false));
                                J$.P(6988953, J$.R(6988937, 'this', this, false, false), 'valpha', J$.T(6988945, 1, 22, false));
                            } else {
                                J$.P(6988977, J$.R(6988961, 'this', this, false, false), 'valpha', J$.T(6988969, 1, 22, false));
                                var keys = J$.W(6989009, 'keys', J$.M(6989001, J$.I(typeof Object === 'undefined' ? Object = J$.R(6988985, 'Object', undefined, true, true) : Object = J$.R(6988985, 'Object', Object, true, true)), 'keys', false)(J$.R(6988993, 'obj', obj, false, false)), keys, false, false);
                                if (J$.C(377112, J$.B(709098, 'in', J$.T(6989017, 'alpha', 21, false), J$.R(6989025, 'obj', obj, false, false)))) {
                                    J$.M(6989073, J$.R(6989033, 'keys', keys, false, false), 'splice', false)(J$.M(6989057, J$.R(6989041, 'keys', keys, false, false), 'indexOf', false)(J$.T(6989049, 'alpha', 21, false)), J$.T(6989065, 1, 22, false));
                                    J$.P(6989137, J$.R(6989081, 'this', this, false, false), 'valpha', J$.C(377104, J$.B(709114, '===', J$.U(709106, 'typeof', J$.G(6989097, J$.R(6989089, 'obj', obj, false, false), 'alpha')), J$.T(6989105, 'number', 21, false))) ? J$.G(6989121, J$.R(6989113, 'obj', obj, false, false), 'alpha') : J$.T(6989129, 0, 22, false));
                                }
                                var hashedKeys = J$.W(6989177, 'hashedKeys', J$.M(6989169, J$.M(6989153, J$.R(6989145, 'keys', keys, false, false), 'sort', false)(), 'join', false)(J$.T(6989161, '', 21, false)), hashedKeys, false, false);
                                if (J$.C(377120, J$.U(709130, '!', J$.B(709122, 'in', J$.R(6989185, 'hashedKeys', hashedKeys, false, false), J$.R(6989193, 'hashedModelKeys', hashedModelKeys, false, true))))) {
                                    throw J$.F(6989241, J$.I(typeof Error === 'undefined' ? Error = J$.R(6989201, 'Error', undefined, true, true) : Error = J$.R(6989201, 'Error', Error, true, true)), true)(J$.B(709138, '+', J$.T(6989209, 'Unable to parse color from object: ', 21, false), J$.M(6989233, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(6989217, 'JSON', undefined, true, true) : JSON = J$.R(6989217, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(6989225, 'obj', obj, false, false))));
                                }
                                J$.P(6989281, J$.R(6989249, 'this', this, false, false), 'model', J$.G(6989273, J$.R(6989257, 'hashedModelKeys', hashedModelKeys, false, true), J$.R(6989265, 'hashedKeys', hashedKeys, false, false)));
                                var labels = J$.W(6989329, 'labels', J$.G(6989321, J$.G(6989313, J$.R(6989289, 'convert', convert, false, true), J$.G(6989305, J$.R(6989297, 'this', this, false, false), 'model')), 'labels'), labels, false, false);
                                var color = J$.W(6989345, 'color', J$.T(6989337, [], 10, false), color, false, false);
                                for (i = J$.W(6989361, 'i', J$.T(6989353, 0, 22, false), i, false, false); J$.C(377128, J$.B(709146, '<', J$.R(6989369, 'i', i, false, false), J$.G(6989385, J$.R(6989377, 'labels', labels, false, false), 'length'))); J$.B(709170, '-', i = J$.W(6989401, 'i', J$.B(709162, '+', J$.U(709154, '+', J$.R(6989393, 'i', i, false, false)), 1), i, false, false), 1)) {
                                    J$.M(6989457, J$.R(6989409, 'color', color, false, false), 'push', false)(J$.G(6989449, J$.R(6989417, 'obj', obj, false, false), J$.G(6989441, J$.R(6989425, 'labels', labels, false, false), J$.R(6989433, 'i', i, false, false))));
                                }
                                J$.P(6989497, J$.R(6989465, 'this', this, false, false), 'color', J$.F(6989489, J$.R(6989473, 'zeroArray', zeroArray, false, true), false)(J$.R(6989481, 'color', color, false, false)));
                            }
                            if (J$.C(377192, J$.G(6989529, J$.R(6989505, 'limiters', limiters, false, true), J$.G(6989521, J$.R(6989513, 'this', this, false, false), 'model')))) {
                                channels = J$.W(6989577, 'channels', J$.G(6989569, J$.G(6989561, J$.R(6989537, 'convert', convert, false, true), J$.G(6989553, J$.R(6989545, 'this', this, false, false), 'model')), 'channels'), channels, false, false);
                                for (i = J$.W(6989593, 'i', J$.T(6989585, 0, 22, false), i, false, false); J$.C(377184, J$.B(709178, '<', J$.R(6989601, 'i', i, false, false), J$.R(6989609, 'channels', channels, false, false))); J$.B(709202, '-', i = J$.W(6989625, 'i', J$.B(709194, '+', J$.U(709186, '+', J$.R(6989617, 'i', i, false, false)), 1), i, false, false), 1)) {
                                    var limit = J$.W(6989681, 'limit', J$.G(6989673, J$.G(6989657, J$.R(6989633, 'limiters', limiters, false, true), J$.G(6989649, J$.R(6989641, 'this', this, false, false), 'model')), J$.R(6989665, 'i', i, false, false)), limit, false, false);
                                    if (J$.C(377176, J$.R(6989689, 'limit', limit, false, false))) {
                                        J$.P(6989769, J$.G(6989705, J$.R(6989697, 'this', this, false, false), 'color'), J$.R(6989713, 'i', i, false, false), J$.F(6989761, J$.R(6989721, 'limit', limit, false, false), false)(J$.G(6989753, J$.G(6989737, J$.R(6989729, 'this', this, false, false), 'color'), J$.R(6989745, 'i', i, false, false))));
                                    }
                                }
                            }
                            J$.P(6989849, J$.R(6989777, 'this', this, false, false), 'valpha', J$.M(6989841, J$.I(typeof Math === 'undefined' ? Math = J$.R(6989785, 'Math', undefined, true, true) : Math = J$.R(6989785, 'Math', Math, true, true)), 'max', false)(J$.T(6989793, 0, 22, false), J$.M(6989833, J$.I(typeof Math === 'undefined' ? Math = J$.R(6989801, 'Math', undefined, true, true) : Math = J$.R(6989801, 'Math', Math, true, true)), 'min', false)(J$.T(6989809, 1, 22, false), J$.G(6989825, J$.R(6989817, 'this', this, false, false), 'valpha'))));
                            if (J$.C(377200, J$.G(6989865, J$.I(typeof Object === 'undefined' ? Object = J$.R(6989857, 'Object', undefined, true, true) : Object = J$.R(6989857, 'Object', Object, true, true)), 'freeze'))) {
                                J$.M(6989889, J$.I(typeof Object === 'undefined' ? Object = J$.R(6989873, 'Object', undefined, true, true) : Object = J$.R(6989873, 'Object', Object, true, true)), 'freeze', false)(J$.R(6989881, 'this', this, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(6999145, J$e);
                        } finally {
                            if (J$.Fr(6999153))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function roundTo(num, places) {
                jalangiLabel36:
                    while (true) {
                        try {
                            J$.Fe(6997865, arguments.callee, this, arguments);
                            arguments = J$.N(6997873, 'arguments', arguments, true, false, false);
                            num = J$.N(6997881, 'num', num, true, false, false);
                            places = J$.N(6997889, 'places', places, true, false, false);
                            return J$.Rt(6997857, J$.F(6997849, J$.I(typeof Number === 'undefined' ? Number = J$.R(6997817, 'Number', undefined, true, true) : Number = J$.R(6997817, 'Number', Number, true, true)), false)(J$.M(6997841, J$.R(6997825, 'num', num, false, false), 'toFixed', false)(J$.R(6997833, 'places', places, false, false))));
                        } catch (J$e) {
                            J$.Ex(6999705, J$e);
                        } finally {
                            if (J$.Fr(6999713))
                                continue jalangiLabel36;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function roundToPlace(places) {
                jalangiLabel38:
                    while (true) {
                        try {
                            J$.Fe(6997977, arguments.callee, this, arguments);
                            arguments = J$.N(6997985, 'arguments', arguments, true, false, false);
                            places = J$.N(6997993, 'places', places, true, false, false);
                            return J$.Rt(6997969, J$.T(6997961, function (num) {
                                jalangiLabel37:
                                    while (true) {
                                        try {
                                            J$.Fe(6997937, arguments.callee, this, arguments);
                                            arguments = J$.N(6997945, 'arguments', arguments, true, false, false);
                                            num = J$.N(6997953, 'num', num, true, false, false);
                                            return J$.Rt(6997929, J$.F(6997921, J$.R(6997897, 'roundTo', roundTo, false, true), false)(J$.R(6997905, 'num', num, false, false), J$.R(6997913, 'places', places, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(6999721, J$e);
                                        } finally {
                                            if (J$.Fr(6999729))
                                                continue jalangiLabel37;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(6999737, J$e);
                        } finally {
                            if (J$.Fr(6999745))
                                continue jalangiLabel38;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function getset(model, channel, modifier) {
                jalangiLabel41:
                    while (true) {
                        try {
                            J$.Fe(6998521, arguments.callee, this, arguments);
                            arguments = J$.N(6998529, 'arguments', arguments, true, false, false);
                            model = J$.N(6998537, 'model', model, true, false, false);
                            channel = J$.N(6998545, 'channel', channel, true, false, false);
                            modifier = J$.N(6998553, 'modifier', modifier, true, false, false);
                            model = J$.W(6998049, 'model', J$.C(377432, J$.M(6998017, J$.I(typeof Array === 'undefined' ? Array = J$.R(6998001, 'Array', undefined, true, true) : Array = J$.R(6998001, 'Array', Array, true, true)), 'isArray', false)(J$.R(6998009, 'model', model, false, false))) ? J$.R(6998025, 'model', model, false, false) : J$.T(6998041, [J$.R(6998033, 'model', model, false, false)], 10, false), model, false, false);
                            J$.M(6998177, J$.R(6998057, 'model', model, false, false), 'forEach', false)(J$.T(6998169, function (m) {
                                jalangiLabel39:
                                    while (true) {
                                        try {
                                            J$.Fe(6998145, arguments.callee, this, arguments);
                                            arguments = J$.N(6998153, 'arguments', arguments, true, false, false);
                                            m = J$.N(6998161, 'm', m, true, false, false);
                                            J$.P(6998137, J$.C(377440, J$.G(6998081, J$.R(6998065, 'limiters', limiters, false, true), J$.R(6998073, 'm', m, false, false))) ? J$._() : J$.P(6998113, J$.R(6998089, 'limiters', limiters, false, true), J$.R(6998097, 'm', m, false, false), J$.T(6998105, [], 10, false)), J$.R(6998121, 'channel', channel, false, false), J$.R(6998129, 'modifier', modifier, false, false));
                                        } catch (J$e) {
                                            J$.Ex(6999753, J$e);
                                        } finally {
                                            if (J$.Fr(6999761))
                                                continue jalangiLabel39;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            model = J$.W(6998209, 'model', J$.G(6998201, J$.R(6998185, 'model', model, false, false), J$.T(6998193, 0, 22, false)), model, false, false);
                            return J$.Rt(6998513, J$.T(6998505, function (val) {
                                jalangiLabel40:
                                    while (true) {
                                        try {
                                            J$.Fe(6998473, arguments.callee, this, arguments);
                                            arguments = J$.N(6998481, 'arguments', arguments, true, false, false);
                                            val = J$.N(6998489, 'val', val, true, false, false);
                                            J$.N(6998497, 'result', result, false, false, false);
                                            var result;
                                            if (J$.C(377456, J$.G(6998225, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(6998217, 'arguments', undefined, true, true) : arguments = J$.R(6998217, 'arguments', arguments, true, true)), 'length'))) {
                                                if (J$.C(377448, J$.R(6998233, 'modifier', modifier, false, false))) {
                                                    val = J$.W(6998265, 'val', J$.F(6998257, J$.R(6998241, 'modifier', modifier, false, false), false)(J$.R(6998249, 'val', val, false, false)), val, false, false);
                                                }
                                                result = J$.W(6998297, 'result', J$.M(6998289, J$.R(6998273, 'this', this, false, false), J$.R(6998281, 'model', model, false, false), false)(), result, false, false);
                                                J$.P(6998337, J$.G(6998313, J$.R(6998305, 'result', result, false, false), 'color'), J$.R(6998321, 'channel', channel, false, false), J$.R(6998329, 'val', val, false, false));
                                                return J$.Rt(6998353, J$.R(6998345, 'result', result, false, false));
                                            }
                                            result = J$.W(6998409, 'result', J$.G(6998401, J$.G(6998385, J$.M(6998377, J$.R(6998361, 'this', this, false, false), J$.R(6998369, 'model', model, false, false), false)(), 'color'), J$.R(6998393, 'channel', channel, false, false)), result, false, false);
                                            if (J$.C(377464, J$.R(6998417, 'modifier', modifier, false, false))) {
                                                result = J$.W(6998449, 'result', J$.F(6998441, J$.R(6998425, 'modifier', modifier, false, false), false)(J$.R(6998433, 'result', result, false, false)), result, false, false);
                                            }
                                            return J$.Rt(6998465, J$.R(6998457, 'result', result, false, false));
                                        } catch (J$e) {
                                            J$.Ex(6999769, J$e);
                                        } finally {
                                            if (J$.Fr(6999777))
                                                continue jalangiLabel40;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(6999785, J$e);
                        } finally {
                            if (J$.Fr(6999793))
                                continue jalangiLabel41;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function maxfn(max) {
                jalangiLabel43:
                    while (true) {
                        try {
                            J$.Fe(6998665, arguments.callee, this, arguments);
                            arguments = J$.N(6998673, 'arguments', arguments, true, false, false);
                            max = J$.N(6998681, 'max', max, true, false, false);
                            return J$.Rt(6998657, J$.T(6998649, function (v) {
                                jalangiLabel42:
                                    while (true) {
                                        try {
                                            J$.Fe(6998625, arguments.callee, this, arguments);
                                            arguments = J$.N(6998633, 'arguments', arguments, true, false, false);
                                            v = J$.N(6998641, 'v', v, true, false, false);
                                            return J$.Rt(6998617, J$.M(6998609, J$.I(typeof Math === 'undefined' ? Math = J$.R(6998561, 'Math', undefined, true, true) : Math = J$.R(6998561, 'Math', Math, true, true)), 'max', false)(J$.T(6998569, 0, 22, false), J$.M(6998601, J$.I(typeof Math === 'undefined' ? Math = J$.R(6998577, 'Math', undefined, true, true) : Math = J$.R(6998577, 'Math', Math, true, true)), 'min', false)(J$.R(6998585, 'max', max, false, false), J$.R(6998593, 'v', v, false, false))));
                                        } catch (J$e) {
                                            J$.Ex(6999801, J$e);
                                        } finally {
                                            if (J$.Fr(6999809))
                                                continue jalangiLabel42;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(6999817, J$e);
                        } finally {
                            if (J$.Fr(6999825))
                                continue jalangiLabel43;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function assertArray(val) {
                jalangiLabel44:
                    while (true) {
                        try {
                            J$.Fe(6998745, arguments.callee, this, arguments);
                            arguments = J$.N(6998753, 'arguments', arguments, true, false, false);
                            val = J$.N(6998761, 'val', val, true, false, false);
                            return J$.Rt(6998737, J$.C(377472, J$.M(6998705, J$.I(typeof Array === 'undefined' ? Array = J$.R(6998689, 'Array', undefined, true, true) : Array = J$.R(6998689, 'Array', Array, true, true)), 'isArray', false)(J$.R(6998697, 'val', val, false, false))) ? J$.R(6998713, 'val', val, false, false) : J$.T(6998729, [J$.R(6998721, 'val', val, false, false)], 10, false));
                        } catch (J$e) {
                            J$.Ex(6999833, J$e);
                        } finally {
                            if (J$.Fr(6999841))
                                continue jalangiLabel44;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function zeroArray(arr, length) {
                jalangiLabel45:
                    while (true) {
                        try {
                            J$.Fe(6998897, arguments.callee, this, arguments);
                            arguments = J$.N(6998905, 'arguments', arguments, true, false, false);
                            arr = J$.N(6998913, 'arr', arr, true, false, false);
                            length = J$.N(6998921, 'length', length, true, false, false);
                            J$.N(6998929, 'i', i, false, false, false);
                            for (var i = J$.W(6998777, 'i', J$.T(6998769, 0, 22, false), i, false, false); J$.C(377488, J$.B(710122, '<', J$.R(6998785, 'i', i, false, false), J$.R(6998793, 'length', length, false, false))); J$.B(710146, '-', i = J$.W(6998809, 'i', J$.B(710138, '+', J$.U(710130, '+', J$.R(6998801, 'i', i, false, false)), 1), i, false, false), 1)) {
                                if (J$.C(377480, J$.B(710162, '!==', J$.U(710154, 'typeof', J$.G(6998833, J$.R(6998817, 'arr', arr, false, false), J$.R(6998825, 'i', i, false, false))), J$.T(6998841, 'number', 21, false)))) {
                                    J$.P(6998873, J$.R(6998849, 'arr', arr, false, false), J$.R(6998857, 'i', i, false, false), J$.T(6998865, 0, 22, false));
                                }
                            }
                            return J$.Rt(6998889, J$.R(6998881, 'arr', arr, false, false));
                        } catch (J$e) {
                            J$.Ex(6999849, J$e);
                        } finally {
                            if (J$.Fr(6999857))
                                continue jalangiLabel45;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(6998969, 'colorString', colorString, false, false, false);
            J$.N(6998977, 'convert', convert, false, false, false);
            J$.N(6998985, '_slice', _slice, false, false, false);
            J$.N(6998993, 'skippedModels', skippedModels, false, false, false);
            J$.N(6999001, 'hashedModelKeys', hashedModelKeys, false, false, false);
            J$.N(6999009, 'limiters', limiters, false, false, false);
            Color = J$.N(6999025, 'Color', J$.T(6999017, Color, 12, false), true, false, false);
            roundTo = J$.N(6999041, 'roundTo', J$.T(6999033, roundTo, 12, false), true, false, false);
            roundToPlace = J$.N(6999057, 'roundToPlace', J$.T(6999049, roundToPlace, 12, false), true, false, false);
            getset = J$.N(6999073, 'getset', J$.T(6999065, getset, 12, false), true, false, false);
            maxfn = J$.N(6999089, 'maxfn', J$.T(6999081, maxfn, 12, false), true, false, false);
            assertArray = J$.N(6999105, 'assertArray', J$.T(6999097, assertArray, 12, false), true, false, false);
            zeroArray = J$.N(6999121, 'zeroArray', J$.T(6999113, zeroArray, 12, false), true, false, false);
            J$.T(6987481, 'use strict', 21, false);
            var colorString = J$.W(6987513, 'colorString', J$.F(6987505, J$.I(typeof require === 'undefined' ? require = J$.R(6987489, 'require', undefined, true, true) : require = J$.R(6987489, 'require', require, true, true)), false)(J$.T(6987497, 'color-string', 21, false)), colorString, false, true);
            var convert = J$.W(6987545, 'convert', J$.F(6987537, J$.I(typeof require === 'undefined' ? require = J$.R(6987521, 'require', undefined, true, true) : require = J$.R(6987521, 'require', require, true, true)), false)(J$.T(6987529, 'color-convert', 21, false)), convert, false, true);
            var _slice = J$.W(6987569, '_slice', J$.G(6987561, J$.T(6987553, [], 10, false), 'slice'), _slice, false, true);
            var skippedModels = J$.W(6987609, 'skippedModels', J$.T(6987601, [
                J$.T(6987577, 'keyword', 21, false),
                J$.T(6987585, 'gray', 21, false),
                J$.T(6987593, 'hex', 21, false)
            ], 10, false), skippedModels, false, true);
            var hashedModelKeys = J$.W(6987625, 'hashedModelKeys', J$.T(6987617, {}, 11, false), hashedModelKeys, false, true);
            J$.M(6987785, J$.M(6987649, J$.I(typeof Object === 'undefined' ? Object = J$.R(6987633, 'Object', undefined, true, true) : Object = J$.R(6987633, 'Object', Object, true, true)), 'keys', false)(J$.R(6987641, 'convert', convert, false, true)), 'forEach', false)(J$.T(6987777, function (model) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(6987753, arguments.callee, this, arguments);
                            arguments = J$.N(6987761, 'arguments', arguments, true, false, false);
                            model = J$.N(6987769, 'model', model, true, false, false);
                            J$.P(6987745, J$.R(6987657, 'hashedModelKeys', hashedModelKeys, false, true), J$.M(6987729, J$.M(6987713, J$.M(6987705, J$.R(6987665, '_slice', _slice, false, true), 'call', false)(J$.G(6987697, J$.G(6987689, J$.R(6987673, 'convert', convert, false, true), J$.R(6987681, 'model', model, false, false)), 'labels')), 'sort', false)(), 'join', false)(J$.T(6987721, '', 21, false)), J$.R(6987737, 'model', model, false, false));
                        } catch (J$e) {
                            J$.Ex(6999129, J$e);
                        } finally {
                            if (J$.Fr(6999137))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var limiters = J$.W(6987801, 'limiters', J$.T(6987793, {}, 11, false), limiters, false, true);
            J$.P(6997113, J$.R(6990001, 'Color', Color, false, true), 'prototype', J$.T(6997105, {
                toString: J$.T(6990049, function () {
                    jalangiLabel2:
                        while (true) {
                            try {
                                J$.Fe(6990033, arguments.callee, this, arguments);
                                arguments = J$.N(6990041, 'arguments', arguments, true, false, false);
                                return J$.Rt(6990025, J$.M(6990017, J$.R(6990009, 'this', this, false, false), 'string', false)());
                            } catch (J$e) {
                                J$.Ex(6999161, J$e);
                            } finally {
                                if (J$.Fr(6999169))
                                    continue jalangiLabel2;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                toJSON: J$.T(6990113, function () {
                    jalangiLabel3:
                        while (true) {
                            try {
                                J$.Fe(6990097, arguments.callee, this, arguments);
                                arguments = J$.N(6990105, 'arguments', arguments, true, false, false);
                                return J$.Rt(6990089, J$.M(6990081, J$.R(6990057, 'this', this, false, false), J$.G(6990073, J$.R(6990065, 'this', this, false, false), 'model'), false)());
                            } catch (J$e) {
                                J$.Ex(6999177, J$e);
                            } finally {
                                if (J$.Fr(6999185))
                                    continue jalangiLabel3;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                string: J$.T(6990425, function (places) {
                    jalangiLabel4:
                        while (true) {
                            try {
                                J$.Fe(6990385, arguments.callee, this, arguments);
                                arguments = J$.N(6990393, 'arguments', arguments, true, false, false);
                                places = J$.N(6990401, 'places', places, true, false, false);
                                J$.N(6990409, 'self', self, false, false, false);
                                J$.N(6990417, 'args', args, false, false, false);
                                var self = J$.W(6990177, 'self', J$.C(377208, J$.B(709210, 'in', J$.G(6990129, J$.R(6990121, 'this', this, false, false), 'model'), J$.G(6990145, J$.R(6990137, 'colorString', colorString, false, true), 'to'))) ? J$.R(6990153, 'this', this, false, false) : J$.M(6990169, J$.R(6990161, 'this', this, false, false), 'rgb', false)(), self, false, false);
                                self = J$.W(6990233, 'self', J$.M(6990225, J$.R(6990185, 'self', self, false, false), 'round', false)(J$.C(377216, J$.B(709226, '===', J$.U(709218, 'typeof', J$.R(6990193, 'places', places, false, false)), J$.T(6990201, 'number', 21, false))) ? J$.R(6990209, 'places', places, false, false) : J$.T(6990217, 1, 22, false)), self, false, false);
                                var args = J$.W(6990321, 'args', J$.C(377224, J$.B(709234, '===', J$.G(6990249, J$.R(6990241, 'self', self, false, false), 'valpha'), J$.T(6990257, 1, 22, false))) ? J$.G(6990273, J$.R(6990265, 'self', self, false, false), 'color') : J$.M(6990313, J$.G(6990289, J$.R(6990281, 'self', self, false, false), 'color'), 'concat', false)(J$.G(6990305, J$.R(6990297, 'this', this, false, false), 'valpha')), args, false, false);
                                return J$.Rt(6990377, J$.M(6990369, J$.G(6990337, J$.R(6990329, 'colorString', colorString, false, true), 'to'), J$.G(6990353, J$.R(6990345, 'self', self, false, false), 'model'), false)(J$.R(6990361, 'args', args, false, false)));
                            } catch (J$e) {
                                J$.Ex(6999193, J$e);
                            } finally {
                                if (J$.Fr(6999201))
                                    continue jalangiLabel4;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                percentString: J$.T(6990673, function (places) {
                    jalangiLabel5:
                        while (true) {
                            try {
                                J$.Fe(6990633, arguments.callee, this, arguments);
                                arguments = J$.N(6990641, 'arguments', arguments, true, false, false);
                                places = J$.N(6990649, 'places', places, true, false, false);
                                J$.N(6990657, 'self', self, false, false, false);
                                J$.N(6990665, 'args', args, false, false, false);
                                var self = J$.W(6990489, 'self', J$.M(6990481, J$.M(6990441, J$.R(6990433, 'this', this, false, false), 'rgb', false)(), 'round', false)(J$.C(377232, J$.B(709250, '===', J$.U(709242, 'typeof', J$.R(6990449, 'places', places, false, false)), J$.T(6990457, 'number', 21, false))) ? J$.R(6990465, 'places', places, false, false) : J$.T(6990473, 1, 22, false)), self, false, false);
                                var args = J$.W(6990577, 'args', J$.C(377240, J$.B(709258, '===', J$.G(6990505, J$.R(6990497, 'self', self, false, false), 'valpha'), J$.T(6990513, 1, 22, false))) ? J$.G(6990529, J$.R(6990521, 'self', self, false, false), 'color') : J$.M(6990569, J$.G(6990545, J$.R(6990537, 'self', self, false, false), 'color'), 'concat', false)(J$.G(6990561, J$.R(6990553, 'this', this, false, false), 'valpha')), args, false, false);
                                return J$.Rt(6990625, J$.M(6990617, J$.G(6990601, J$.G(6990593, J$.R(6990585, 'colorString', colorString, false, true), 'to'), 'rgb'), 'percent', false)(J$.R(6990609, 'args', args, false, false)));
                            } catch (J$e) {
                                J$.Ex(6999209, J$e);
                            } finally {
                                if (J$.Fr(6999217))
                                    continue jalangiLabel5;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                array: J$.T(6990793, function () {
                    jalangiLabel6:
                        while (true) {
                            try {
                                J$.Fe(6990777, arguments.callee, this, arguments);
                                arguments = J$.N(6990785, 'arguments', arguments, true, false, false);
                                return J$.Rt(6990769, J$.C(377248, J$.B(709266, '===', J$.G(6990689, J$.R(6990681, 'this', this, false, false), 'valpha'), J$.T(6990697, 1, 22, false))) ? J$.M(6990721, J$.G(6990713, J$.R(6990705, 'this', this, false, false), 'color'), 'slice', false)() : J$.M(6990761, J$.G(6990737, J$.R(6990729, 'this', this, false, false), 'color'), 'concat', false)(J$.G(6990753, J$.R(6990745, 'this', this, false, false), 'valpha')));
                            } catch (J$e) {
                                J$.Ex(6999225, J$e);
                            } finally {
                                if (J$.Fr(6999233))
                                    continue jalangiLabel6;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                object: J$.T(6991153, function () {
                    jalangiLabel7:
                        while (true) {
                            try {
                                J$.Fe(6991105, arguments.callee, this, arguments);
                                arguments = J$.N(6991113, 'arguments', arguments, true, false, false);
                                J$.N(6991121, 'result', result, false, false, false);
                                J$.N(6991129, 'channels', channels, false, false, false);
                                J$.N(6991137, 'labels', labels, false, false, false);
                                J$.N(6991145, 'i', i, false, false, false);
                                var result = J$.W(6990809, 'result', J$.T(6990801, {}, 11, false), result, false, false);
                                var channels = J$.W(6990857, 'channels', J$.G(6990849, J$.G(6990841, J$.R(6990817, 'convert', convert, false, true), J$.G(6990833, J$.R(6990825, 'this', this, false, false), 'model')), 'channels'), channels, false, false);
                                var labels = J$.W(6990905, 'labels', J$.G(6990897, J$.G(6990889, J$.R(6990865, 'convert', convert, false, true), J$.G(6990881, J$.R(6990873, 'this', this, false, false), 'model')), 'labels'), labels, false, false);
                                for (var i = J$.W(6990921, 'i', J$.T(6990913, 0, 22, false), i, false, false); J$.C(377256, J$.B(709274, '<', J$.R(6990929, 'i', i, false, false), J$.R(6990937, 'channels', channels, false, false))); J$.B(709298, '-', i = J$.W(6990953, 'i', J$.B(709290, '+', J$.U(709282, '+', J$.R(6990945, 'i', i, false, false)), 1), i, false, false), 1)) {
                                    J$.P(6991025, J$.R(6990961, 'result', result, false, false), J$.G(6990985, J$.R(6990969, 'labels', labels, false, false), J$.R(6990977, 'i', i, false, false)), J$.G(6991017, J$.G(6991001, J$.R(6990993, 'this', this, false, false), 'color'), J$.R(6991009, 'i', i, false, false)));
                                }
                                if (J$.C(377264, J$.B(709306, '!==', J$.G(6991041, J$.R(6991033, 'this', this, false, false), 'valpha'), J$.T(6991049, 1, 22, false)))) {
                                    J$.P(6991081, J$.R(6991057, 'result', result, false, false), 'alpha', J$.G(6991073, J$.R(6991065, 'this', this, false, false), 'valpha'));
                                }
                                return J$.Rt(6991097, J$.R(6991089, 'result', result, false, false));
                            } catch (J$e) {
                                J$.Ex(6999241, J$e);
                            } finally {
                                if (J$.Fr(6999249))
                                    continue jalangiLabel7;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                unitArray: J$.T(6991385, function () {
                    jalangiLabel8:
                        while (true) {
                            try {
                                J$.Fe(6991361, arguments.callee, this, arguments);
                                arguments = J$.N(6991369, 'arguments', arguments, true, false, false);
                                J$.N(6991377, 'rgb', rgb, false, false, false);
                                var rgb = J$.W(6991185, 'rgb', J$.G(6991177, J$.M(6991169, J$.R(6991161, 'this', this, false, false), 'rgb', false)(), 'color'), rgb, false, false);
                                J$.A(6991217, J$.R(6991193, 'rgb', rgb, false, false), J$.T(6991201, 0, 22, false), '/')(J$.T(6991209, 255, 22, false));
                                J$.A(6991249, J$.R(6991225, 'rgb', rgb, false, false), J$.T(6991233, 1, 22, false), '/')(J$.T(6991241, 255, 22, false));
                                J$.A(6991281, J$.R(6991257, 'rgb', rgb, false, false), J$.T(6991265, 2, 22, false), '/')(J$.T(6991273, 255, 22, false));
                                if (J$.C(377272, J$.B(709314, '!==', J$.G(6991297, J$.R(6991289, 'this', this, false, false), 'valpha'), J$.T(6991305, 1, 22, false)))) {
                                    J$.M(6991337, J$.R(6991313, 'rgb', rgb, false, false), 'push', false)(J$.G(6991329, J$.R(6991321, 'this', this, false, false), 'valpha'));
                                }
                                return J$.Rt(6991353, J$.R(6991345, 'rgb', rgb, false, false));
                            } catch (J$e) {
                                J$.Ex(6999257, J$e);
                            } finally {
                                if (J$.Fr(6999265))
                                    continue jalangiLabel8;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                unitObject: J$.T(6991593, function () {
                    jalangiLabel9:
                        while (true) {
                            try {
                                J$.Fe(6991569, arguments.callee, this, arguments);
                                arguments = J$.N(6991577, 'arguments', arguments, true, false, false);
                                J$.N(6991585, 'rgb', rgb, false, false, false);
                                var rgb = J$.W(6991417, 'rgb', J$.M(6991409, J$.M(6991401, J$.R(6991393, 'this', this, false, false), 'rgb', false)(), 'object', false)(), rgb, false, false);
                                J$.A(6991441, J$.R(6991425, 'rgb', rgb, false, false), 'r', '/')(J$.T(6991433, 255, 22, false));
                                J$.A(6991465, J$.R(6991449, 'rgb', rgb, false, false), 'g', '/')(J$.T(6991457, 255, 22, false));
                                J$.A(6991489, J$.R(6991473, 'rgb', rgb, false, false), 'b', '/')(J$.T(6991481, 255, 22, false));
                                if (J$.C(377280, J$.B(709322, '!==', J$.G(6991505, J$.R(6991497, 'this', this, false, false), 'valpha'), J$.T(6991513, 1, 22, false)))) {
                                    J$.P(6991545, J$.R(6991521, 'rgb', rgb, false, false), 'alpha', J$.G(6991537, J$.R(6991529, 'this', this, false, false), 'valpha'));
                                }
                                return J$.Rt(6991561, J$.R(6991553, 'rgb', rgb, false, false));
                            } catch (J$e) {
                                J$.Ex(6999273, J$e);
                            } finally {
                                if (J$.Fr(6999281))
                                    continue jalangiLabel9;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                round: J$.T(6991785, function (places) {
                    jalangiLabel10:
                        while (true) {
                            try {
                                J$.Fe(6991761, arguments.callee, this, arguments);
                                arguments = J$.N(6991769, 'arguments', arguments, true, false, false);
                                places = J$.N(6991777, 'places', places, true, false, false);
                                places = J$.W(6991641, 'places', J$.M(6991633, J$.I(typeof Math === 'undefined' ? Math = J$.R(6991601, 'Math', undefined, true, true) : Math = J$.R(6991601, 'Math', Math, true, true)), 'max', false)(J$.C(377288, J$.R(6991609, 'places', places, false, false)) ? J$._() : J$.T(6991617, 0, 22, false), J$.T(6991625, 0, 22, false)), places, false, false);
                                return J$.Rt(6991753, J$.F(6991745, J$.R(6991649, 'Color', Color, false, true), true)(J$.M(6991721, J$.M(6991697, J$.G(6991665, J$.R(6991657, 'this', this, false, false), 'color'), 'map', false)(J$.F(6991689, J$.R(6991673, 'roundToPlace', roundToPlace, false, true), false)(J$.R(6991681, 'places', places, false, false))), 'concat', false)(J$.G(6991713, J$.R(6991705, 'this', this, false, false), 'valpha')), J$.G(6991737, J$.R(6991729, 'this', this, false, false), 'model')));
                            } catch (J$e) {
                                J$.Ex(6999289, J$e);
                            } finally {
                                if (J$.Fr(6999297))
                                    continue jalangiLabel10;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                alpha: J$.T(6991977, function (val) {
                    jalangiLabel11:
                        while (true) {
                            try {
                                J$.Fe(6991953, arguments.callee, this, arguments);
                                arguments = J$.N(6991961, 'arguments', arguments, true, false, false);
                                val = J$.N(6991969, 'val', val, true, false, false);
                                if (J$.C(377296, J$.G(6991801, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(6991793, 'arguments', undefined, true, true) : arguments = J$.R(6991793, 'arguments', arguments, true, true)), 'length'))) {
                                    return J$.Rt(6991921, J$.F(6991913, J$.R(6991809, 'Color', Color, false, true), true)(J$.M(6991889, J$.G(6991825, J$.R(6991817, 'this', this, false, false), 'color'), 'concat', false)(J$.M(6991881, J$.I(typeof Math === 'undefined' ? Math = J$.R(6991833, 'Math', undefined, true, true) : Math = J$.R(6991833, 'Math', Math, true, true)), 'max', false)(J$.T(6991841, 0, 22, false), J$.M(6991873, J$.I(typeof Math === 'undefined' ? Math = J$.R(6991849, 'Math', undefined, true, true) : Math = J$.R(6991849, 'Math', Math, true, true)), 'min', false)(J$.T(6991857, 1, 22, false), J$.R(6991865, 'val', val, false, false)))), J$.G(6991905, J$.R(6991897, 'this', this, false, false), 'model')));
                                }
                                return J$.Rt(6991945, J$.G(6991937, J$.R(6991929, 'this', this, false, false), 'valpha'));
                            } catch (J$e) {
                                J$.Ex(6999305, J$e);
                            } finally {
                                if (J$.Fr(6999313))
                                    continue jalangiLabel11;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                red: J$.F(6992033, J$.R(6991985, 'getset', getset, false, true), false)(J$.T(6991993, 'rgb', 21, false), J$.T(6992001, 0, 22, false), J$.F(6992025, J$.R(6992009, 'maxfn', maxfn, false, true), false)(J$.T(6992017, 255, 22, false))),
                green: J$.F(6992089, J$.R(6992041, 'getset', getset, false, true), false)(J$.T(6992049, 'rgb', 21, false), J$.T(6992057, 1, 22, false), J$.F(6992081, J$.R(6992065, 'maxfn', maxfn, false, true), false)(J$.T(6992073, 255, 22, false))),
                blue: J$.F(6992145, J$.R(6992097, 'getset', getset, false, true), false)(J$.T(6992105, 'rgb', 21, false), J$.T(6992113, 2, 22, false), J$.F(6992137, J$.R(6992121, 'maxfn', maxfn, false, true), false)(J$.T(6992129, 255, 22, false))),
                hue: J$.F(6992289, J$.R(6992153, 'getset', getset, false, true), false)(J$.T(6992201, [
                    J$.T(6992161, 'hsl', 21, false),
                    J$.T(6992169, 'hsv', 21, false),
                    J$.T(6992177, 'hsl', 21, false),
                    J$.T(6992185, 'hwb', 21, false),
                    J$.T(6992193, 'hcg', 21, false)
                ], 10, false), J$.T(6992209, 0, 22, false), J$.T(6992281, function (val) {
                    jalangiLabel12:
                        while (true) {
                            try {
                                J$.Fe(6992257, arguments.callee, this, arguments);
                                arguments = J$.N(6992265, 'arguments', arguments, true, false, false);
                                val = J$.N(6992273, 'val', val, true, false, false);
                                return J$.Rt(6992249, J$.B(709346, '%', J$.B(709338, '+', J$.B(709330, '%', J$.R(6992217, 'val', val, false, false), J$.T(6992225, 360, 22, false)), J$.T(6992233, 360, 22, false)), J$.T(6992241, 360, 22, false)));
                            } catch (J$e) {
                                J$.Ex(6999321, J$e);
                            } finally {
                                if (J$.Fr(6999329))
                                    continue jalangiLabel12;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)),
                saturationl: J$.F(6992345, J$.R(6992297, 'getset', getset, false, true), false)(J$.T(6992305, 'hsl', 21, false), J$.T(6992313, 1, 22, false), J$.F(6992337, J$.R(6992321, 'maxfn', maxfn, false, true), false)(J$.T(6992329, 100, 22, false))),
                lightness: J$.F(6992401, J$.R(6992353, 'getset', getset, false, true), false)(J$.T(6992361, 'hsl', 21, false), J$.T(6992369, 2, 22, false), J$.F(6992393, J$.R(6992377, 'maxfn', maxfn, false, true), false)(J$.T(6992385, 100, 22, false))),
                saturationv: J$.F(6992457, J$.R(6992409, 'getset', getset, false, true), false)(J$.T(6992417, 'hsv', 21, false), J$.T(6992425, 1, 22, false), J$.F(6992449, J$.R(6992433, 'maxfn', maxfn, false, true), false)(J$.T(6992441, 100, 22, false))),
                value: J$.F(6992513, J$.R(6992465, 'getset', getset, false, true), false)(J$.T(6992473, 'hsv', 21, false), J$.T(6992481, 2, 22, false), J$.F(6992505, J$.R(6992489, 'maxfn', maxfn, false, true), false)(J$.T(6992497, 100, 22, false))),
                chroma: J$.F(6992569, J$.R(6992521, 'getset', getset, false, true), false)(J$.T(6992529, 'hcg', 21, false), J$.T(6992537, 1, 22, false), J$.F(6992561, J$.R(6992545, 'maxfn', maxfn, false, true), false)(J$.T(6992553, 100, 22, false))),
                gray: J$.F(6992625, J$.R(6992577, 'getset', getset, false, true), false)(J$.T(6992585, 'hcg', 21, false), J$.T(6992593, 2, 22, false), J$.F(6992617, J$.R(6992601, 'maxfn', maxfn, false, true), false)(J$.T(6992609, 100, 22, false))),
                white: J$.F(6992681, J$.R(6992633, 'getset', getset, false, true), false)(J$.T(6992641, 'hwb', 21, false), J$.T(6992649, 1, 22, false), J$.F(6992673, J$.R(6992657, 'maxfn', maxfn, false, true), false)(J$.T(6992665, 100, 22, false))),
                wblack: J$.F(6992737, J$.R(6992689, 'getset', getset, false, true), false)(J$.T(6992697, 'hwb', 21, false), J$.T(6992705, 2, 22, false), J$.F(6992729, J$.R(6992713, 'maxfn', maxfn, false, true), false)(J$.T(6992721, 100, 22, false))),
                cyan: J$.F(6992793, J$.R(6992745, 'getset', getset, false, true), false)(J$.T(6992753, 'cmyk', 21, false), J$.T(6992761, 0, 22, false), J$.F(6992785, J$.R(6992769, 'maxfn', maxfn, false, true), false)(J$.T(6992777, 100, 22, false))),
                magenta: J$.F(6992849, J$.R(6992801, 'getset', getset, false, true), false)(J$.T(6992809, 'cmyk', 21, false), J$.T(6992817, 1, 22, false), J$.F(6992841, J$.R(6992825, 'maxfn', maxfn, false, true), false)(J$.T(6992833, 100, 22, false))),
                yellow: J$.F(6992905, J$.R(6992857, 'getset', getset, false, true), false)(J$.T(6992865, 'cmyk', 21, false), J$.T(6992873, 2, 22, false), J$.F(6992897, J$.R(6992881, 'maxfn', maxfn, false, true), false)(J$.T(6992889, 100, 22, false))),
                black: J$.F(6992961, J$.R(6992913, 'getset', getset, false, true), false)(J$.T(6992921, 'cmyk', 21, false), J$.T(6992929, 3, 22, false), J$.F(6992953, J$.R(6992937, 'maxfn', maxfn, false, true), false)(J$.T(6992945, 100, 22, false))),
                x: J$.F(6993017, J$.R(6992969, 'getset', getset, false, true), false)(J$.T(6992977, 'xyz', 21, false), J$.T(6992985, 0, 22, false), J$.F(6993009, J$.R(6992993, 'maxfn', maxfn, false, true), false)(J$.T(6993001, 100, 22, false))),
                y: J$.F(6993073, J$.R(6993025, 'getset', getset, false, true), false)(J$.T(6993033, 'xyz', 21, false), J$.T(6993041, 1, 22, false), J$.F(6993065, J$.R(6993049, 'maxfn', maxfn, false, true), false)(J$.T(6993057, 100, 22, false))),
                z: J$.F(6993129, J$.R(6993081, 'getset', getset, false, true), false)(J$.T(6993089, 'xyz', 21, false), J$.T(6993097, 2, 22, false), J$.F(6993121, J$.R(6993105, 'maxfn', maxfn, false, true), false)(J$.T(6993113, 100, 22, false))),
                l: J$.F(6993185, J$.R(6993137, 'getset', getset, false, true), false)(J$.T(6993145, 'lab', 21, false), J$.T(6993153, 0, 22, false), J$.F(6993177, J$.R(6993161, 'maxfn', maxfn, false, true), false)(J$.T(6993169, 100, 22, false))),
                a: J$.F(6993217, J$.R(6993193, 'getset', getset, false, true), false)(J$.T(6993201, 'lab', 21, false), J$.T(6993209, 1, 22, false)),
                b: J$.F(6993249, J$.R(6993225, 'getset', getset, false, true), false)(J$.T(6993233, 'lab', 21, false), J$.T(6993241, 2, 22, false)),
                keyword: J$.T(6993393, function (val) {
                    jalangiLabel13:
                        while (true) {
                            try {
                                J$.Fe(6993369, arguments.callee, this, arguments);
                                arguments = J$.N(6993377, 'arguments', arguments, true, false, false);
                                val = J$.N(6993385, 'val', val, true, false, false);
                                if (J$.C(377304, J$.G(6993265, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(6993257, 'arguments', undefined, true, true) : arguments = J$.R(6993257, 'arguments', arguments, true, true)), 'length'))) {
                                    return J$.Rt(6993297, J$.F(6993289, J$.R(6993273, 'Color', Color, false, true), true)(J$.R(6993281, 'val', val, false, false)));
                                }
                                return J$.Rt(6993361, J$.M(6993353, J$.G(6993329, J$.R(6993305, 'convert', convert, false, true), J$.G(6993321, J$.R(6993313, 'this', this, false, false), 'model')), 'keyword', false)(J$.G(6993345, J$.R(6993337, 'this', this, false, false), 'color')));
                            } catch (J$e) {
                                J$.Ex(6999337, J$e);
                            } finally {
                                if (J$.Fr(6999345))
                                    continue jalangiLabel13;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                hex: J$.T(6993537, function (val) {
                    jalangiLabel14:
                        while (true) {
                            try {
                                J$.Fe(6993513, arguments.callee, this, arguments);
                                arguments = J$.N(6993521, 'arguments', arguments, true, false, false);
                                val = J$.N(6993529, 'val', val, true, false, false);
                                if (J$.C(377312, J$.G(6993409, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(6993401, 'arguments', undefined, true, true) : arguments = J$.R(6993401, 'arguments', arguments, true, true)), 'length'))) {
                                    return J$.Rt(6993441, J$.F(6993433, J$.R(6993417, 'Color', Color, false, true), true)(J$.R(6993425, 'val', val, false, false)));
                                }
                                return J$.Rt(6993505, J$.M(6993497, J$.G(6993457, J$.R(6993449, 'colorString', colorString, false, true), 'to'), 'hex', false)(J$.G(6993489, J$.M(6993481, J$.M(6993473, J$.R(6993465, 'this', this, false, false), 'rgb', false)(), 'round', false)(), 'color')));
                            } catch (J$e) {
                                J$.Ex(6999353, J$e);
                            } finally {
                                if (J$.Fr(6999361))
                                    continue jalangiLabel14;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                rgbNumber: J$.T(6993721, function () {
                    jalangiLabel15:
                        while (true) {
                            try {
                                J$.Fe(6993697, arguments.callee, this, arguments);
                                arguments = J$.N(6993705, 'arguments', arguments, true, false, false);
                                J$.N(6993713, 'rgb', rgb, false, false, false);
                                var rgb = J$.W(6993569, 'rgb', J$.G(6993561, J$.M(6993553, J$.R(6993545, 'this', this, false, false), 'rgb', false)(), 'color'), rgb, false, false);
                                return J$.Rt(6993689, J$.B(709402, '|', J$.B(709386, '|', J$.B(709362, '<<', J$.B(709354, '&', J$.G(6993593, J$.R(6993577, 'rgb', rgb, false, false), J$.T(6993585, 0, 22, false)), J$.T(6993601, 255, 22, false)), J$.T(6993609, 16, 22, false)), J$.B(709378, '<<', J$.B(709370, '&', J$.G(6993633, J$.R(6993617, 'rgb', rgb, false, false), J$.T(6993625, 1, 22, false)), J$.T(6993641, 255, 22, false)), J$.T(6993649, 8, 22, false))), J$.B(709394, '&', J$.G(6993673, J$.R(6993657, 'rgb', rgb, false, false), J$.T(6993665, 2, 22, false)), J$.T(6993681, 255, 22, false))));
                            } catch (J$e) {
                                J$.Ex(6999369, J$e);
                            } finally {
                                if (J$.Fr(6999377))
                                    continue jalangiLabel15;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                luminosity: J$.T(6994129, function () {
                    jalangiLabel16:
                        while (true) {
                            try {
                                J$.Fe(6994081, arguments.callee, this, arguments);
                                arguments = J$.N(6994089, 'arguments', arguments, true, false, false);
                                J$.N(6994097, 'rgb', rgb, false, false, false);
                                J$.N(6994105, 'lum', lum, false, false, false);
                                J$.N(6994113, 'i', i, false, false, false);
                                J$.N(6994121, 'chan', chan, false, false, false);
                                var rgb = J$.W(6993753, 'rgb', J$.G(6993745, J$.M(6993737, J$.R(6993729, 'this', this, false, false), 'rgb', false)(), 'color'), rgb, false, false);
                                var lum = J$.W(6993769, 'lum', J$.T(6993761, [], 10, false), lum, false, false);
                                for (var i = J$.W(6993785, 'i', J$.T(6993777, 0, 22, false), i, false, false); J$.C(377328, J$.B(709410, '<', J$.R(6993793, 'i', i, false, false), J$.G(6993809, J$.R(6993801, 'rgb', rgb, false, false), 'length'))); J$.B(709434, '-', i = J$.W(6993825, 'i', J$.B(709426, '+', J$.U(709418, '+', J$.R(6993817, 'i', i, false, false)), 1), i, false, false), 1)) {
                                    var chan = J$.W(6993865, 'chan', J$.B(709442, '/', J$.G(6993849, J$.R(6993833, 'rgb', rgb, false, false), J$.R(6993841, 'i', i, false, false)), J$.T(6993857, 255, 22, false)), chan, false, false);
                                    J$.P(6993969, J$.R(6993873, 'lum', lum, false, false), J$.R(6993881, 'i', i, false, false), J$.C(377320, J$.B(709450, '<=', J$.R(6993889, 'chan', chan, false, false), J$.T(6993897, 0.03928, 22, false))) ? J$.B(709458, '/', J$.R(6993905, 'chan', chan, false, false), J$.T(6993913, 12.92, 22, false)) : J$.M(6993961, J$.I(typeof Math === 'undefined' ? Math = J$.R(6993921, 'Math', undefined, true, true) : Math = J$.R(6993921, 'Math', Math, true, true)), 'pow', false)(J$.B(709474, '/', J$.B(709466, '+', J$.R(6993929, 'chan', chan, false, false), J$.T(6993937, 0.055, 22, false)), J$.T(6993945, 1.055, 22, false)), J$.T(6993953, 2.4, 22, false)));
                                }
                                return J$.Rt(6994073, J$.B(709514, '+', J$.B(709498, '+', J$.B(709482, '*', J$.T(6993977, 0.2126, 22, false), J$.G(6994001, J$.R(6993985, 'lum', lum, false, false), J$.T(6993993, 0, 22, false))), J$.B(709490, '*', J$.T(6994009, 0.7152, 22, false), J$.G(6994033, J$.R(6994017, 'lum', lum, false, false), J$.T(6994025, 1, 22, false)))), J$.B(709506, '*', J$.T(6994041, 0.0722, 22, false), J$.G(6994065, J$.R(6994049, 'lum', lum, false, false), J$.T(6994057, 2, 22, false)))));
                            } catch (J$e) {
                                J$.Ex(6999385, J$e);
                            } finally {
                                if (J$.Fr(6999393))
                                    continue jalangiLabel16;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                contrast: J$.T(6994321, function (color2) {
                    jalangiLabel17:
                        while (true) {
                            try {
                                J$.Fe(6994281, arguments.callee, this, arguments);
                                arguments = J$.N(6994289, 'arguments', arguments, true, false, false);
                                color2 = J$.N(6994297, 'color2', color2, true, false, false);
                                J$.N(6994305, 'lum1', lum1, false, false, false);
                                J$.N(6994313, 'lum2', lum2, false, false, false);
                                var lum1 = J$.W(6994153, 'lum1', J$.M(6994145, J$.R(6994137, 'this', this, false, false), 'luminosity', false)(), lum1, false, false);
                                var lum2 = J$.W(6994177, 'lum2', J$.M(6994169, J$.R(6994161, 'color2', color2, false, false), 'luminosity', false)(), lum2, false, false);
                                if (J$.C(377336, J$.B(709522, '>', J$.R(6994185, 'lum1', lum1, false, false), J$.R(6994193, 'lum2', lum2, false, false)))) {
                                    return J$.Rt(6994233, J$.B(709546, '/', J$.B(709530, '+', J$.R(6994201, 'lum1', lum1, false, false), J$.T(6994209, 0.05, 22, false)), J$.B(709538, '+', J$.R(6994217, 'lum2', lum2, false, false), J$.T(6994225, 0.05, 22, false))));
                                }
                                return J$.Rt(6994273, J$.B(709570, '/', J$.B(709554, '+', J$.R(6994241, 'lum2', lum2, false, false), J$.T(6994249, 0.05, 22, false)), J$.B(709562, '+', J$.R(6994257, 'lum1', lum1, false, false), J$.T(6994265, 0.05, 22, false))));
                            } catch (J$e) {
                                J$.Ex(6999401, J$e);
                            } finally {
                                if (J$.Fr(6999409))
                                    continue jalangiLabel17;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                level: J$.T(6994465, function (color2) {
                    jalangiLabel18:
                        while (true) {
                            try {
                                J$.Fe(6994433, arguments.callee, this, arguments);
                                arguments = J$.N(6994441, 'arguments', arguments, true, false, false);
                                color2 = J$.N(6994449, 'color2', color2, true, false, false);
                                J$.N(6994457, 'contrastRatio', contrastRatio, false, false, false);
                                var contrastRatio = J$.W(6994353, 'contrastRatio', J$.M(6994345, J$.R(6994329, 'this', this, false, false), 'contrast', false)(J$.R(6994337, 'color2', color2, false, false)), contrastRatio, false, false);
                                if (J$.C(377344, J$.B(709578, '>=', J$.R(6994361, 'contrastRatio', contrastRatio, false, false), J$.T(6994369, 7.1, 22, false)))) {
                                    return J$.Rt(6994385, J$.T(6994377, 'AAA', 21, false));
                                }
                                return J$.Rt(6994425, J$.C(377352, J$.B(709586, '>=', J$.R(6994393, 'contrastRatio', contrastRatio, false, false), J$.T(6994401, 4.5, 22, false))) ? J$.T(6994409, 'AA', 21, false) : J$.T(6994417, '', 21, false));
                            } catch (J$e) {
                                J$.Ex(6999417, J$e);
                            } finally {
                                if (J$.Fr(6999425))
                                    continue jalangiLabel18;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                isDark: J$.T(6994673, function () {
                    jalangiLabel19:
                        while (true) {
                            try {
                                J$.Fe(6994641, arguments.callee, this, arguments);
                                arguments = J$.N(6994649, 'arguments', arguments, true, false, false);
                                J$.N(6994657, 'rgb', rgb, false, false, false);
                                J$.N(6994665, 'yiq', yiq, false, false, false);
                                var rgb = J$.W(6994497, 'rgb', J$.G(6994489, J$.M(6994481, J$.R(6994473, 'this', this, false, false), 'rgb', false)(), 'color'), rgb, false, false);
                                var yiq = J$.W(6994609, 'yiq', J$.B(709634, '/', J$.B(709626, '+', J$.B(709610, '+', J$.B(709594, '*', J$.G(6994521, J$.R(6994505, 'rgb', rgb, false, false), J$.T(6994513, 0, 22, false)), J$.T(6994529, 299, 22, false)), J$.B(709602, '*', J$.G(6994553, J$.R(6994537, 'rgb', rgb, false, false), J$.T(6994545, 1, 22, false)), J$.T(6994561, 587, 22, false))), J$.B(709618, '*', J$.G(6994585, J$.R(6994569, 'rgb', rgb, false, false), J$.T(6994577, 2, 22, false)), J$.T(6994593, 114, 22, false))), J$.T(6994601, 1000, 22, false)), yiq, false, false);
                                return J$.Rt(6994633, J$.B(709642, '<', J$.R(6994617, 'yiq', yiq, false, false), J$.T(6994625, 128, 22, false)));
                            } catch (J$e) {
                                J$.Ex(6999433, J$e);
                            } finally {
                                if (J$.Fr(6999441))
                                    continue jalangiLabel19;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                isLight: J$.T(6994721, function () {
                    jalangiLabel20:
                        while (true) {
                            try {
                                J$.Fe(6994705, arguments.callee, this, arguments);
                                arguments = J$.N(6994713, 'arguments', arguments, true, false, false);
                                return J$.Rt(6994697, J$.U(709650, '!', J$.M(6994689, J$.R(6994681, 'this', this, false, false), 'isDark', false)()));
                            } catch (J$e) {
                                J$.Ex(6999449, J$e);
                            } finally {
                                if (J$.Fr(6999457))
                                    continue jalangiLabel20;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                negate: J$.T(6994921, function () {
                    jalangiLabel21:
                        while (true) {
                            try {
                                J$.Fe(6994889, arguments.callee, this, arguments);
                                arguments = J$.N(6994897, 'arguments', arguments, true, false, false);
                                J$.N(6994905, 'rgb', rgb, false, false, false);
                                J$.N(6994913, 'i', i, false, false, false);
                                var rgb = J$.W(6994745, 'rgb', J$.M(6994737, J$.R(6994729, 'this', this, false, false), 'rgb', false)(), rgb, false, false);
                                for (var i = J$.W(6994761, 'i', J$.T(6994753, 0, 22, false), i, false, false); J$.C(377360, J$.B(709658, '<', J$.R(6994769, 'i', i, false, false), J$.T(6994777, 3, 22, false))); J$.B(709682, '-', i = J$.W(6994793, 'i', J$.B(709674, '+', J$.U(709666, '+', J$.R(6994785, 'i', i, false, false)), 1), i, false, false), 1)) {
                                    J$.P(6994865, J$.G(6994809, J$.R(6994801, 'rgb', rgb, false, false), 'color'), J$.R(6994817, 'i', i, false, false), J$.B(709690, '-', J$.T(6994825, 255, 22, false), J$.G(6994857, J$.G(6994841, J$.R(6994833, 'rgb', rgb, false, false), 'color'), J$.R(6994849, 'i', i, false, false))));
                                }
                                return J$.Rt(6994881, J$.R(6994873, 'rgb', rgb, false, false));
                            } catch (J$e) {
                                J$.Ex(6999465, J$e);
                            } finally {
                                if (J$.Fr(6999473))
                                    continue jalangiLabel21;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                lighten: J$.T(6995073, function (ratio) {
                    jalangiLabel22:
                        while (true) {
                            try {
                                J$.Fe(6995041, arguments.callee, this, arguments);
                                arguments = J$.N(6995049, 'arguments', arguments, true, false, false);
                                ratio = J$.N(6995057, 'ratio', ratio, true, false, false);
                                J$.N(6995065, 'hsl', hsl, false, false, false);
                                var hsl = J$.W(6994945, 'hsl', J$.M(6994937, J$.R(6994929, 'this', this, false, false), 'hsl', false)(), hsl, false, false);
                                J$.A(6995017, J$.G(6994961, J$.R(6994953, 'hsl', hsl, false, false), 'color'), J$.T(6994969, 2, 22, false), '+')(J$.B(709698, '*', J$.G(6995001, J$.G(6994985, J$.R(6994977, 'hsl', hsl, false, false), 'color'), J$.T(6994993, 2, 22, false)), J$.R(6995009, 'ratio', ratio, false, false)));
                                return J$.Rt(6995033, J$.R(6995025, 'hsl', hsl, false, false));
                            } catch (J$e) {
                                J$.Ex(6999481, J$e);
                            } finally {
                                if (J$.Fr(6999489))
                                    continue jalangiLabel22;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                darken: J$.T(6995225, function (ratio) {
                    jalangiLabel23:
                        while (true) {
                            try {
                                J$.Fe(6995193, arguments.callee, this, arguments);
                                arguments = J$.N(6995201, 'arguments', arguments, true, false, false);
                                ratio = J$.N(6995209, 'ratio', ratio, true, false, false);
                                J$.N(6995217, 'hsl', hsl, false, false, false);
                                var hsl = J$.W(6995097, 'hsl', J$.M(6995089, J$.R(6995081, 'this', this, false, false), 'hsl', false)(), hsl, false, false);
                                J$.A(6995169, J$.G(6995113, J$.R(6995105, 'hsl', hsl, false, false), 'color'), J$.T(6995121, 2, 22, false), '-')(J$.B(709706, '*', J$.G(6995153, J$.G(6995137, J$.R(6995129, 'hsl', hsl, false, false), 'color'), J$.T(6995145, 2, 22, false)), J$.R(6995161, 'ratio', ratio, false, false)));
                                return J$.Rt(6995185, J$.R(6995177, 'hsl', hsl, false, false));
                            } catch (J$e) {
                                J$.Ex(6999497, J$e);
                            } finally {
                                if (J$.Fr(6999505))
                                    continue jalangiLabel23;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                saturate: J$.T(6995377, function (ratio) {
                    jalangiLabel24:
                        while (true) {
                            try {
                                J$.Fe(6995345, arguments.callee, this, arguments);
                                arguments = J$.N(6995353, 'arguments', arguments, true, false, false);
                                ratio = J$.N(6995361, 'ratio', ratio, true, false, false);
                                J$.N(6995369, 'hsl', hsl, false, false, false);
                                var hsl = J$.W(6995249, 'hsl', J$.M(6995241, J$.R(6995233, 'this', this, false, false), 'hsl', false)(), hsl, false, false);
                                J$.A(6995321, J$.G(6995265, J$.R(6995257, 'hsl', hsl, false, false), 'color'), J$.T(6995273, 1, 22, false), '+')(J$.B(709714, '*', J$.G(6995305, J$.G(6995289, J$.R(6995281, 'hsl', hsl, false, false), 'color'), J$.T(6995297, 1, 22, false)), J$.R(6995313, 'ratio', ratio, false, false)));
                                return J$.Rt(6995337, J$.R(6995329, 'hsl', hsl, false, false));
                            } catch (J$e) {
                                J$.Ex(6999513, J$e);
                            } finally {
                                if (J$.Fr(6999521))
                                    continue jalangiLabel24;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                desaturate: J$.T(6995529, function (ratio) {
                    jalangiLabel25:
                        while (true) {
                            try {
                                J$.Fe(6995497, arguments.callee, this, arguments);
                                arguments = J$.N(6995505, 'arguments', arguments, true, false, false);
                                ratio = J$.N(6995513, 'ratio', ratio, true, false, false);
                                J$.N(6995521, 'hsl', hsl, false, false, false);
                                var hsl = J$.W(6995401, 'hsl', J$.M(6995393, J$.R(6995385, 'this', this, false, false), 'hsl', false)(), hsl, false, false);
                                J$.A(6995473, J$.G(6995417, J$.R(6995409, 'hsl', hsl, false, false), 'color'), J$.T(6995425, 1, 22, false), '-')(J$.B(709722, '*', J$.G(6995457, J$.G(6995441, J$.R(6995433, 'hsl', hsl, false, false), 'color'), J$.T(6995449, 1, 22, false)), J$.R(6995465, 'ratio', ratio, false, false)));
                                return J$.Rt(6995489, J$.R(6995481, 'hsl', hsl, false, false));
                            } catch (J$e) {
                                J$.Ex(6999529, J$e);
                            } finally {
                                if (J$.Fr(6999537))
                                    continue jalangiLabel25;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                whiten: J$.T(6995681, function (ratio) {
                    jalangiLabel26:
                        while (true) {
                            try {
                                J$.Fe(6995649, arguments.callee, this, arguments);
                                arguments = J$.N(6995657, 'arguments', arguments, true, false, false);
                                ratio = J$.N(6995665, 'ratio', ratio, true, false, false);
                                J$.N(6995673, 'hwb', hwb, false, false, false);
                                var hwb = J$.W(6995553, 'hwb', J$.M(6995545, J$.R(6995537, 'this', this, false, false), 'hwb', false)(), hwb, false, false);
                                J$.A(6995625, J$.G(6995569, J$.R(6995561, 'hwb', hwb, false, false), 'color'), J$.T(6995577, 1, 22, false), '+')(J$.B(709730, '*', J$.G(6995609, J$.G(6995593, J$.R(6995585, 'hwb', hwb, false, false), 'color'), J$.T(6995601, 1, 22, false)), J$.R(6995617, 'ratio', ratio, false, false)));
                                return J$.Rt(6995641, J$.R(6995633, 'hwb', hwb, false, false));
                            } catch (J$e) {
                                J$.Ex(6999545, J$e);
                            } finally {
                                if (J$.Fr(6999553))
                                    continue jalangiLabel26;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                blacken: J$.T(6995833, function (ratio) {
                    jalangiLabel27:
                        while (true) {
                            try {
                                J$.Fe(6995801, arguments.callee, this, arguments);
                                arguments = J$.N(6995809, 'arguments', arguments, true, false, false);
                                ratio = J$.N(6995817, 'ratio', ratio, true, false, false);
                                J$.N(6995825, 'hwb', hwb, false, false, false);
                                var hwb = J$.W(6995705, 'hwb', J$.M(6995697, J$.R(6995689, 'this', this, false, false), 'hwb', false)(), hwb, false, false);
                                J$.A(6995777, J$.G(6995721, J$.R(6995713, 'hwb', hwb, false, false), 'color'), J$.T(6995729, 2, 22, false), '+')(J$.B(709738, '*', J$.G(6995761, J$.G(6995745, J$.R(6995737, 'hwb', hwb, false, false), 'color'), J$.T(6995753, 2, 22, false)), J$.R(6995769, 'ratio', ratio, false, false)));
                                return J$.Rt(6995793, J$.R(6995785, 'hwb', hwb, false, false));
                            } catch (J$e) {
                                J$.Ex(6999561, J$e);
                            } finally {
                                if (J$.Fr(6999569))
                                    continue jalangiLabel27;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                grayscale: J$.T(6996057, function () {
                    jalangiLabel28:
                        while (true) {
                            try {
                                J$.Fe(6996025, arguments.callee, this, arguments);
                                arguments = J$.N(6996033, 'arguments', arguments, true, false, false);
                                J$.N(6996041, 'rgb', rgb, false, false, false);
                                J$.N(6996049, 'val', val, false, false, false);
                                var rgb = J$.W(6995865, 'rgb', J$.G(6995857, J$.M(6995849, J$.R(6995841, 'this', this, false, false), 'rgb', false)(), 'color'), rgb, false, false);
                                var val = J$.W(6995969, 'val', J$.B(709778, '+', J$.B(709762, '+', J$.B(709746, '*', J$.G(6995889, J$.R(6995873, 'rgb', rgb, false, false), J$.T(6995881, 0, 22, false)), J$.T(6995897, 0.3, 22, false)), J$.B(709754, '*', J$.G(6995921, J$.R(6995905, 'rgb', rgb, false, false), J$.T(6995913, 1, 22, false)), J$.T(6995929, 0.59, 22, false))), J$.B(709770, '*', J$.G(6995953, J$.R(6995937, 'rgb', rgb, false, false), J$.T(6995945, 2, 22, false)), J$.T(6995961, 0.11, 22, false))), val, false, false);
                                return J$.Rt(6996017, J$.M(6996009, J$.R(6995977, 'Color', Color, false, true), 'rgb', false)(J$.R(6995985, 'val', val, false, false), J$.R(6995993, 'val', val, false, false), J$.R(6996001, 'val', val, false, false)));
                            } catch (J$e) {
                                J$.Ex(6999577, J$e);
                            } finally {
                                if (J$.Fr(6999585))
                                    continue jalangiLabel28;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                fade: J$.T(6996153, function (ratio) {
                    jalangiLabel29:
                        while (true) {
                            try {
                                J$.Fe(6996129, arguments.callee, this, arguments);
                                arguments = J$.N(6996137, 'arguments', arguments, true, false, false);
                                ratio = J$.N(6996145, 'ratio', ratio, true, false, false);
                                return J$.Rt(6996121, J$.M(6996113, J$.R(6996065, 'this', this, false, false), 'alpha', false)(J$.B(709794, '-', J$.G(6996081, J$.R(6996073, 'this', this, false, false), 'valpha'), J$.B(709786, '*', J$.G(6996097, J$.R(6996089, 'this', this, false, false), 'valpha'), J$.R(6996105, 'ratio', ratio, false, false)))));
                            } catch (J$e) {
                                J$.Ex(6999593, J$e);
                            } finally {
                                if (J$.Fr(6999601))
                                    continue jalangiLabel29;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                opaquer: J$.T(6996249, function (ratio) {
                    jalangiLabel30:
                        while (true) {
                            try {
                                J$.Fe(6996225, arguments.callee, this, arguments);
                                arguments = J$.N(6996233, 'arguments', arguments, true, false, false);
                                ratio = J$.N(6996241, 'ratio', ratio, true, false, false);
                                return J$.Rt(6996217, J$.M(6996209, J$.R(6996161, 'this', this, false, false), 'alpha', false)(J$.B(709810, '+', J$.G(6996177, J$.R(6996169, 'this', this, false, false), 'valpha'), J$.B(709802, '*', J$.G(6996193, J$.R(6996185, 'this', this, false, false), 'valpha'), J$.R(6996201, 'ratio', ratio, false, false)))));
                            } catch (J$e) {
                                J$.Ex(6999609, J$e);
                            } finally {
                                if (J$.Fr(6999617))
                                    continue jalangiLabel30;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                rotate: J$.T(6996497, function (degrees) {
                    jalangiLabel31:
                        while (true) {
                            try {
                                J$.Fe(6996457, arguments.callee, this, arguments);
                                arguments = J$.N(6996465, 'arguments', arguments, true, false, false);
                                degrees = J$.N(6996473, 'degrees', degrees, true, false, false);
                                J$.N(6996481, 'hsl', hsl, false, false, false);
                                J$.N(6996489, 'hue', hue, false, false, false);
                                var hsl = J$.W(6996273, 'hsl', J$.M(6996265, J$.R(6996257, 'this', this, false, false), 'hsl', false)(), hsl, false, false);
                                var hue = J$.W(6996313, 'hue', J$.G(6996305, J$.G(6996289, J$.R(6996281, 'hsl', hsl, false, false), 'color'), J$.T(6996297, 0, 22, false)), hue, false, false);
                                hue = J$.W(6996345, 'hue', J$.B(709826, '%', J$.B(709818, '+', J$.R(6996321, 'hue', hue, false, false), J$.R(6996329, 'degrees', degrees, false, false)), J$.T(6996337, 360, 22, false)), hue, false, false);
                                hue = J$.W(6996393, 'hue', J$.C(377368, J$.B(709834, '<', J$.R(6996353, 'hue', hue, false, false), J$.T(6996361, 0, 22, false))) ? J$.B(709842, '+', J$.T(6996369, 360, 22, false), J$.R(6996377, 'hue', hue, false, false)) : J$.R(6996385, 'hue', hue, false, false), hue, false, false);
                                J$.P(6996433, J$.G(6996409, J$.R(6996401, 'hsl', hsl, false, false), 'color'), J$.T(6996417, 0, 22, false), J$.R(6996425, 'hue', hue, false, false));
                                return J$.Rt(6996449, J$.R(6996441, 'hsl', hsl, false, false));
                            } catch (J$e) {
                                J$.Ex(6999625, J$e);
                            } finally {
                                if (J$.Fr(6999633))
                                    continue jalangiLabel31;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false),
                mix: J$.T(6997097, function (mixinColor, weight) {
                    jalangiLabel32:
                        while (true) {
                            try {
                                J$.Fe(6997009, arguments.callee, this, arguments);
                                arguments = J$.N(6997017, 'arguments', arguments, true, false, false);
                                mixinColor = J$.N(6997025, 'mixinColor', mixinColor, true, false, false);
                                weight = J$.N(6997033, 'weight', weight, true, false, false);
                                J$.N(6997041, 'color1', color1, false, false, false);
                                J$.N(6997049, 'color2', color2, false, false, false);
                                J$.N(6997057, 'p', p, false, false, false);
                                J$.N(6997065, 'w', w, false, false, false);
                                J$.N(6997073, 'a', a, false, false, false);
                                J$.N(6997081, 'w1', w1, false, false, false);
                                J$.N(6997089, 'w2', w2, false, false, false);
                                var color1 = J$.W(6996521, 'color1', J$.M(6996513, J$.R(6996505, 'mixinColor', mixinColor, false, false), 'rgb', false)(), color1, false, false);
                                var color2 = J$.W(6996545, 'color2', J$.M(6996537, J$.R(6996529, 'this', this, false, false), 'rgb', false)(), color2, false, false);
                                var p = J$.W(6996585, 'p', J$.C(377376, J$.B(709850, '===', J$.R(6996553, 'weight', weight, false, false), J$.T(6996561, undefined, 24, false))) ? J$.T(6996569, 0.5, 22, false) : J$.R(6996577, 'weight', weight, false, false), p, false, false);
                                var w = J$.W(6996617, 'w', J$.B(709866, '-', J$.B(709858, '*', J$.T(6996593, 2, 22, false), J$.R(6996601, 'p', p, false, false)), J$.T(6996609, 1, 22, false)), w, false, false);
                                var a = J$.W(6996657, 'a', J$.B(709874, '-', J$.M(6996633, J$.R(6996625, 'color1', color1, false, false), 'alpha', false)(), J$.M(6996649, J$.R(6996641, 'color2', color2, false, false), 'alpha', false)()), a, false, false);
                                var w1 = J$.W(6996753, 'w1', J$.B(709946, '/', J$.B(709938, '+', J$.C(377384, J$.B(709898, '===', J$.B(709882, '*', J$.R(6996665, 'w', w, false, false), J$.R(6996673, 'a', a, false, false)), J$.U(709890, '-', J$.T(6996681, 1, 22, false)))) ? J$.R(6996689, 'w', w, false, false) : J$.B(709930, '/', J$.B(709906, '+', J$.R(6996697, 'w', w, false, false), J$.R(6996705, 'a', a, false, false)), J$.B(709922, '+', J$.T(6996713, 1, 22, false), J$.B(709914, '*', J$.R(6996721, 'w', w, false, false), J$.R(6996729, 'a', a, false, false)))), J$.T(6996737, 1, 22, false)), J$.T(6996745, 2, 22, false)), w1, false, false);
                                var w2 = J$.W(6996777, 'w2', J$.B(709954, '-', J$.T(6996761, 1, 22, false), J$.R(6996769, 'w1', w1, false, false)), w2, false, false);
                                return J$.Rt(6997001, J$.M(6996993, J$.R(6996785, 'Color', Color, false, true), 'rgb', false)(J$.B(709978, '+', J$.B(709962, '*', J$.R(6996793, 'w1', w1, false, false), J$.M(6996809, J$.R(6996801, 'color1', color1, false, false), 'red', false)()), J$.B(709970, '*', J$.R(6996817, 'w2', w2, false, false), J$.M(6996833, J$.R(6996825, 'color2', color2, false, false), 'red', false)())), J$.B(710002, '+', J$.B(709986, '*', J$.R(6996841, 'w1', w1, false, false), J$.M(6996857, J$.R(6996849, 'color1', color1, false, false), 'green', false)()), J$.B(709994, '*', J$.R(6996865, 'w2', w2, false, false), J$.M(6996881, J$.R(6996873, 'color2', color2, false, false), 'green', false)())), J$.B(710026, '+', J$.B(710010, '*', J$.R(6996889, 'w1', w1, false, false), J$.M(6996905, J$.R(6996897, 'color1', color1, false, false), 'blue', false)()), J$.B(710018, '*', J$.R(6996913, 'w2', w2, false, false), J$.M(6996929, J$.R(6996921, 'color2', color2, false, false), 'blue', false)())), J$.B(710058, '+', J$.B(710034, '*', J$.M(6996945, J$.R(6996937, 'color1', color1, false, false), 'alpha', false)(), J$.R(6996953, 'p', p, false, false)), J$.B(710050, '*', J$.M(6996969, J$.R(6996961, 'color2', color2, false, false), 'alpha', false)(), J$.B(710042, '-', J$.T(6996977, 1, 22, false), J$.R(6996985, 'p', p, false, false))))));
                            } catch (J$e) {
                                J$.Ex(6999641, J$e);
                            } finally {
                                if (J$.Fr(6999649))
                                    continue jalangiLabel32;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.M(6997809, J$.M(6997137, J$.I(typeof Object === 'undefined' ? Object = J$.R(6997121, 'Object', undefined, true, true) : Object = J$.R(6997121, 'Object', Object, true, true)), 'keys', false)(J$.R(6997129, 'convert', convert, false, true)), 'forEach', false)(J$.T(6997801, function (model) {
                jalangiLabel35:
                    while (true) {
                        try {
                            J$.Fe(6997769, arguments.callee, this, arguments);
                            arguments = J$.N(6997777, 'arguments', arguments, true, false, false);
                            model = J$.N(6997785, 'model', model, true, false, false);
                            J$.N(6997793, 'channels', channels, false, false, false);
                            if (J$.C(377392, J$.B(710074, '!==', J$.M(6997161, J$.R(6997145, 'skippedModels', skippedModels, false, true), 'indexOf', false)(J$.R(6997153, 'model', model, false, false)), J$.U(710066, '-', J$.T(6997169, 1, 22, false))))) {
                                return J$.Rt(6997177, undefined);
                            }
                            var channels = J$.W(6997217, 'channels', J$.G(6997209, J$.G(6997201, J$.R(6997185, 'convert', convert, false, true), J$.R(6997193, 'model', model, false, false)), 'channels'), channels, false, false);
                            J$.P(6997593, J$.G(6997233, J$.R(6997225, 'Color', Color, false, true), 'prototype'), J$.R(6997241, 'model', model, false, false), J$.T(6997585, function () {
                                jalangiLabel33:
                                    while (true) {
                                        try {
                                            J$.Fe(6997561, arguments.callee, this, arguments);
                                            arguments = J$.N(6997569, 'arguments', arguments, true, false, false);
                                            J$.N(6997577, 'newAlpha', newAlpha, false, false, false);
                                            if (J$.C(377400, J$.B(710082, '===', J$.G(6997257, J$.R(6997249, 'this', this, false, false), 'model'), J$.R(6997265, 'model', model, false, false)))) {
                                                return J$.Rt(6997297, J$.F(6997289, J$.R(6997273, 'Color', Color, false, true), true)(J$.R(6997281, 'this', this, false, false)));
                                            }
                                            if (J$.C(377408, J$.G(6997313, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(6997305, 'arguments', undefined, true, true) : arguments = J$.R(6997305, 'arguments', arguments, true, true)), 'length'))) {
                                                return J$.Rt(6997353, J$.F(6997345, J$.R(6997321, 'Color', Color, false, true), true)(J$.I(typeof arguments === 'undefined' ? arguments = J$.R(6997329, 'arguments', undefined, true, true) : arguments = J$.R(6997329, 'arguments', arguments, true, true)), J$.R(6997337, 'model', model, false, false)));
                                            }
                                            var newAlpha = J$.W(6997417, 'newAlpha', J$.C(377416, J$.B(710098, '===', J$.U(710090, 'typeof', J$.G(6997377, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(6997361, 'arguments', undefined, true, true) : arguments = J$.R(6997361, 'arguments', arguments, true, true)), J$.R(6997369, 'channels', channels, false, false))), J$.T(6997385, 'number', 21, false))) ? J$.R(6997393, 'channels', channels, false, false) : J$.G(6997409, J$.R(6997401, 'this', this, false, false), 'valpha'), newAlpha, false, false);
                                            return J$.Rt(6997553, J$.F(6997545, J$.R(6997425, 'Color', Color, false, true), true)(J$.M(6997529, J$.F(6997513, J$.R(6997433, 'assertArray', assertArray, false, true), false)(J$.M(6997505, J$.G(6997481, J$.G(6997465, J$.R(6997441, 'convert', convert, false, true), J$.G(6997457, J$.R(6997449, 'this', this, false, false), 'model')), J$.R(6997473, 'model', model, false, false)), 'raw', false)(J$.G(6997497, J$.R(6997489, 'this', this, false, false), 'color'))), 'concat', false)(J$.R(6997521, 'newAlpha', newAlpha, false, false)), J$.R(6997537, 'model', model, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(6999657, J$e);
                                        } finally {
                                            if (J$.Fr(6999665))
                                                continue jalangiLabel33;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.P(6997761, J$.R(6997601, 'Color', Color, false, true), J$.R(6997609, 'model', model, false, false), J$.T(6997753, function (color) {
                                jalangiLabel34:
                                    while (true) {
                                        try {
                                            J$.Fe(6997729, arguments.callee, this, arguments);
                                            arguments = J$.N(6997737, 'arguments', arguments, true, false, false);
                                            color = J$.N(6997745, 'color', color, true, false, false);
                                            if (J$.C(377424, J$.B(710114, '===', J$.U(710106, 'typeof', J$.R(6997617, 'color', color, false, false)), J$.T(6997625, 'number', 21, false)))) {
                                                color = J$.W(6997681, 'color', J$.F(6997673, J$.R(6997633, 'zeroArray', zeroArray, false, true), false)(J$.M(6997657, J$.R(6997641, '_slice', _slice, false, true), 'call', false)(J$.I(typeof arguments === 'undefined' ? arguments = J$.R(6997649, 'arguments', undefined, true, true) : arguments = J$.R(6997649, 'arguments', arguments, true, true))), J$.R(6997665, 'channels', channels, false, false)), color, false, false);
                                            }
                                            return J$.Rt(6997721, J$.F(6997713, J$.R(6997689, 'Color', Color, false, true), true)(J$.R(6997697, 'color', color, false, false), J$.R(6997705, 'model', model, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(6999673, J$e);
                                        } finally {
                                            if (J$.Fr(6999681))
                                                continue jalangiLabel34;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(6999689, J$e);
                        } finally {
                            if (J$.Fr(6999697))
                                continue jalangiLabel35;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(6998953, J$.I(typeof module === 'undefined' ? module = J$.R(6998937, 'module', undefined, true, true) : module = J$.R(6998937, 'module', module, true, true)), 'exports', J$.R(6998945, 'Color', Color, false, true));
        } catch (J$e) {
            J$.Ex(6999865, J$e);
        } finally {
            if (J$.Sr(6999873))
                continue jalangiLabel46;
            else
                break jalangiLabel46;
        }
    }
// JALANGI DO NOT INSTRUMENT
