// triple-beam/config/cli.js
J$.noInstrEval = false;
jalangiLabel0:
    while (true) {
        try {
            J$.Se(5338001, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/triple-beam/config/cli.js');
            J$.T(5337785, 'use strict', 21, false);
            J$.P(5337889, J$.I(typeof exports === 'undefined' ? exports = J$.R(5337793, 'exports', undefined, true, true) : exports = J$.R(5337793, 'exports', exports, true, true)), 'levels', J$.T(5337881, {
                error: J$.T(5337801, 0, 22, false),
                warn: J$.T(5337809, 1, 22, false),
                help: J$.T(5337817, 2, 22, false),
                data: J$.T(5337825, 3, 22, false),
                info: J$.T(5337833, 4, 22, false),
                debug: J$.T(5337841, 5, 22, false),
                prompt: J$.T(5337849, 6, 22, false),
                verbose: J$.T(5337857, 7, 22, false),
                input: J$.T(5337865, 8, 22, false),
                silly: J$.T(5337873, 9, 22, false)
            }, 11, false));
            J$.P(5337993, J$.I(typeof exports === 'undefined' ? exports = J$.R(5337897, 'exports', undefined, true, true) : exports = J$.R(5337897, 'exports', exports, true, true)), 'colors', J$.T(5337985, {
                error: J$.T(5337905, 'red', 21, false),
                warn: J$.T(5337913, 'yellow', 21, false),
                help: J$.T(5337921, 'cyan', 21, false),
                data: J$.T(5337929, 'grey', 21, false),
                info: J$.T(5337937, 'green', 21, false),
                debug: J$.T(5337945, 'blue', 21, false),
                prompt: J$.T(5337953, 'grey', 21, false),
                verbose: J$.T(5337961, 'cyan', 21, false),
                input: J$.T(5337969, 'grey', 21, false),
                silly: J$.T(5337977, 'magenta', 21, false)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(5338009, J$e);
        } finally {
            if (J$.Sr(5338017))
                continue jalangiLabel0;
            else
                break jalangiLabel0;
        }
    }
// JALANGI DO NOT INSTRUMENT
