// triple-beam/config/npm.js
J$.noInstrEval = false;
jalangiLabel0:
    while (true) {
        try {
            J$.Se(5338417, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/triple-beam/config/npm.js');
            J$.T(5338249, 'use strict', 21, false);
            J$.P(5338329, J$.I(typeof exports === 'undefined' ? exports = J$.R(5338257, 'exports', undefined, true, true) : exports = J$.R(5338257, 'exports', exports, true, true)), 'levels', J$.T(5338321, {
                error: J$.T(5338265, 0, 22, false),
                warn: J$.T(5338273, 1, 22, false),
                info: J$.T(5338281, 2, 22, false),
                http: J$.T(5338289, 3, 22, false),
                verbose: J$.T(5338297, 4, 22, false),
                debug: J$.T(5338305, 5, 22, false),
                silly: J$.T(5338313, 6, 22, false)
            }, 11, false));
            J$.P(5338409, J$.I(typeof exports === 'undefined' ? exports = J$.R(5338337, 'exports', undefined, true, true) : exports = J$.R(5338337, 'exports', exports, true, true)), 'colors', J$.T(5338401, {
                error: J$.T(5338345, 'red', 21, false),
                warn: J$.T(5338353, 'yellow', 21, false),
                info: J$.T(5338361, 'green', 21, false),
                http: J$.T(5338369, 'green', 21, false),
                verbose: J$.T(5338377, 'cyan', 21, false),
                debug: J$.T(5338385, 'blue', 21, false),
                silly: J$.T(5338393, 'magenta', 21, false)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(5338425, J$e);
        } finally {
            if (J$.Sr(5338433))
                continue jalangiLabel0;
            else
                break jalangiLabel0;
        }
    }
// JALANGI DO NOT INSTRUMENT
