// triple-beam/config/syslog.js
J$.noInstrEval = false;
jalangiLabel0:
    while (true) {
        try {
            J$.Se(5338625, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/triple-beam/config/syslog.js');
            J$.T(5338441, 'use strict', 21, false);
            J$.P(5338529, J$.I(typeof exports === 'undefined' ? exports = J$.R(5338449, 'exports', undefined, true, true) : exports = J$.R(5338449, 'exports', exports, true, true)), 'levels', J$.T(5338521, {
                emerg: J$.T(5338457, 0, 22, false),
                alert: J$.T(5338465, 1, 22, false),
                crit: J$.T(5338473, 2, 22, false),
                error: J$.T(5338481, 3, 22, false),
                warning: J$.T(5338489, 4, 22, false),
                notice: J$.T(5338497, 5, 22, false),
                info: J$.T(5338505, 6, 22, false),
                debug: J$.T(5338513, 7, 22, false)
            }, 11, false));
            J$.P(5338617, J$.I(typeof exports === 'undefined' ? exports = J$.R(5338537, 'exports', undefined, true, true) : exports = J$.R(5338537, 'exports', exports, true, true)), 'colors', J$.T(5338609, {
                emerg: J$.T(5338545, 'red', 21, false),
                alert: J$.T(5338553, 'yellow', 21, false),
                crit: J$.T(5338561, 'red', 21, false),
                error: J$.T(5338569, 'red', 21, false),
                warning: J$.T(5338577, 'red', 21, false),
                notice: J$.T(5338585, 'yellow', 21, false),
                info: J$.T(5338593, 'green', 21, false),
                debug: J$.T(5338601, 'blue', 21, false)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(5338633, J$e);
        } finally {
            if (J$.Sr(5338641))
                continue jalangiLabel0;
            else
                break jalangiLabel0;
        }
    }
// JALANGI DO NOT INSTRUMENT
