// asynckit/lib/streamify.js
J$.noInstrEval = false;
jalangiLabel8:
    while (true) {
        try {
            J$.Se(2694761, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestRequest/node_modules/asynckit/lib/streamify.js');
            function wrapIterator(iterator) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(2693921, arguments.callee, this, arguments);
                            arguments = J$.N(2693929, 'arguments', arguments, true, false, false);
                            iterator = J$.N(2693937, 'iterator', iterator, true, false, false);
                            J$.N(2693945, 'stream', stream, false, false, false);
                            var stream = J$.W(2693609, 'stream', J$.R(2693601, 'this', this, false, false), stream, false, false);
                            return J$.Rt(2693913, J$.T(2693905, function (item, key, cb) {
                                jalangiLabel0:
                                    while (true) {
                                        try {
                                            J$.Fe(2693849, arguments.callee, this, arguments);
                                            arguments = J$.N(2693857, 'arguments', arguments, true, false, false);
                                            item = J$.N(2693865, 'item', item, true, false, false);
                                            key = J$.N(2693873, 'key', key, true, false, false);
                                            cb = J$.N(2693881, 'cb', cb, true, false, false);
                                            J$.N(2693889, 'aborter', aborter, false, false, false);
                                            J$.N(2693897, 'wrappedCb', wrappedCb, false, false, false);
                                            var aborter, wrappedCb = J$.W(2693673, 'wrappedCb', J$.F(2693665, J$.R(2693617, 'async', async, false, true), false)(J$.M(2693657, J$.R(2693625, 'wrapIteratorCallback', wrapIteratorCallback, false, true), 'call', false)(J$.R(2693633, 'stream', stream, false, false), J$.R(2693641, 'cb', cb, false, false), J$.R(2693649, 'key', key, false, false))), wrappedCb, false, false);
                                            J$.P(2693713, J$.G(2693689, J$.R(2693681, 'stream', stream, false, false), 'jobs'), J$.R(2693697, 'key', key, false, false), J$.R(2693705, 'wrappedCb', wrappedCb, false, false));
                                            if (J$.C(130176, J$.B(404762, '==', J$.G(2693729, J$.R(2693721, 'iterator', iterator, false, false), 'length'), J$.T(2693737, 2, 22, false)))) {
                                                aborter = J$.W(2693777, 'aborter', J$.F(2693769, J$.R(2693745, 'iterator', iterator, false, false), false)(J$.R(2693753, 'item', item, false, false), J$.R(2693761, 'wrappedCb', wrappedCb, false, false)), aborter, false, false);
                                            } else {
                                                aborter = J$.W(2693825, 'aborter', J$.F(2693817, J$.R(2693785, 'iterator', iterator, false, false), false)(J$.R(2693793, 'item', item, false, false), J$.R(2693801, 'key', key, false, false), J$.R(2693809, 'wrappedCb', wrappedCb, false, false)), aborter, false, false);
                                            }
                                            return J$.Rt(2693841, J$.R(2693833, 'aborter', aborter, false, false));
                                        } catch (J$e) {
                                            J$.Ex(2694857, J$e);
                                        } finally {
                                            if (J$.Fr(2694865))
                                                continue jalangiLabel0;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(2694873, J$e);
                        } finally {
                            if (J$.Fr(2694881))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function wrapCallback(callback) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(2694089, arguments.callee, this, arguments);
                            arguments = J$.N(2694097, 'arguments', arguments, true, false, false);
                            callback = J$.N(2694105, 'callback', callback, true, false, false);
                            J$.N(2694113, 'stream', stream, false, false, false);
                            J$.N(2694121, 'wrapped', wrapped, false, false, false);
                            var stream = J$.W(2693961, 'stream', J$.R(2693953, 'this', this, false, false), stream, false, false);
                            var wrapped = J$.W(2694065, 'wrapped', J$.T(2694057, function wrapped(error, result) {
                                jalangiLabel2:
                                    while (true) {
                                        try {
                                            J$.Fe(2694025, arguments.callee, this, arguments);
                                            arguments = J$.N(2694033, 'arguments', arguments, true, false, false);
                                            error = J$.N(2694041, 'error', error, true, false, false);
                                            result = J$.N(2694049, 'result', result, true, false, false);
                                            return J$.Rt(2694017, J$.M(2694009, J$.R(2693969, 'finisher', finisher, false, true), 'call', false)(J$.R(2693977, 'stream', stream, false, false), J$.R(2693985, 'error', error, false, false), J$.R(2693993, 'result', result, false, false), J$.R(2694001, 'callback', callback, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(2694889, J$e);
                                        } finally {
                                            if (J$.Fr(2694897))
                                                continue jalangiLabel2;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), wrapped, false, false);
                            return J$.Rt(2694081, J$.R(2694073, 'wrapped', wrapped, false, false));
                        } catch (J$e) {
                            J$.Ex(2694905, J$e);
                        } finally {
                            if (J$.Fr(2694913))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function wrapIteratorCallback(callback, key) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(2694353, arguments.callee, this, arguments);
                            arguments = J$.N(2694361, 'arguments', arguments, true, false, false);
                            callback = J$.N(2694369, 'callback', callback, true, false, false);
                            key = J$.N(2694377, 'key', key, true, false, false);
                            J$.N(2694385, 'stream', stream, false, false, false);
                            var stream = J$.W(2694137, 'stream', J$.R(2694129, 'this', this, false, false), stream, false, false);
                            return J$.Rt(2694345, J$.T(2694337, function (error, output) {
                                jalangiLabel4:
                                    while (true) {
                                        try {
                                            J$.Fe(2694305, arguments.callee, this, arguments);
                                            arguments = J$.N(2694313, 'arguments', arguments, true, false, false);
                                            error = J$.N(2694321, 'error', error, true, false, false);
                                            output = J$.N(2694329, 'output', output, true, false, false);
                                            if (J$.C(130184, J$.U(404778, '!', J$.B(404770, 'in', J$.R(2694145, 'key', key, false, false), J$.G(2694161, J$.R(2694153, 'stream', stream, false, false), 'jobs'))))) {
                                                J$.F(2694193, J$.R(2694169, 'callback', callback, false, false), false)(J$.R(2694177, 'error', error, false, false), J$.R(2694185, 'output', output, false, false));
                                                return J$.Rt(2694201, undefined);
                                            }
                                            J$.B(404786, 'delete', J$.G(2694217, J$.R(2694209, 'stream', stream, false, false), 'jobs'), J$.R(2694225, 'key', key, false, false));
                                            return J$.Rt(2694297, J$.M(2694289, J$.R(2694233, 'streamer', streamer, false, true), 'call', false)(J$.R(2694241, 'stream', stream, false, false), J$.R(2694249, 'error', error, false, false), J$.T(2694273, {
                                                key: J$.R(2694257, 'key', key, false, false),
                                                value: J$.R(2694265, 'output', output, false, false)
                                            }, 11, false), J$.R(2694281, 'callback', callback, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(2694921, J$e);
                                        } finally {
                                            if (J$.Fr(2694929))
                                                continue jalangiLabel4;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(2694937, J$e);
                        } finally {
                            if (J$.Fr(2694945))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function streamer(error, output, callback) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(2694617, arguments.callee, this, arguments);
                            arguments = J$.N(2694625, 'arguments', arguments, true, false, false);
                            error = J$.N(2694633, 'error', error, true, false, false);
                            output = J$.N(2694641, 'output', output, true, false, false);
                            callback = J$.N(2694649, 'callback', callback, true, false, false);
                            if (J$.C(130208, J$.C(130192, J$.R(2694393, 'error', error, false, false)) ? J$.U(404794, '!', J$.G(2694409, J$.R(2694401, 'this', this, false, false), 'error')) : J$._())) {
                                J$.P(2694433, J$.R(2694417, 'this', this, false, false), 'error', J$.R(2694425, 'error', error, false, false));
                                J$.M(2694449, J$.R(2694441, 'this', this, false, false), 'pause', false)();
                                J$.M(2694481, J$.R(2694457, 'this', this, false, false), 'emit', false)(J$.T(2694465, 'error', 21, false), J$.R(2694473, 'error', error, false, false));
                                J$.F(2694529, J$.R(2694489, 'callback', callback, false, false), false)(J$.R(2694497, 'error', error, false, false), J$.C(130200, J$.R(2694505, 'output', output, false, false)) ? J$.G(2694521, J$.R(2694513, 'output', output, false, false), 'value') : J$._());
                                return J$.Rt(2694537, undefined);
                            }
                            J$.M(2694561, J$.R(2694545, 'this', this, false, false), 'push', false)(J$.R(2694553, 'output', output, false, false));
                            J$.F(2694609, J$.R(2694569, 'callback', callback, false, false), false)(J$.R(2694577, 'error', error, false, false), J$.C(130216, J$.R(2694585, 'output', output, false, false)) ? J$.G(2694601, J$.R(2694593, 'output', output, false, false), 'value') : J$._());
                        } catch (J$e) {
                            J$.Ex(2694953, J$e);
                        } finally {
                            if (J$.Fr(2694961))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function finisher(error, output, callback) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(2694721, arguments.callee, this, arguments);
                            arguments = J$.N(2694729, 'arguments', arguments, true, false, false);
                            error = J$.N(2694737, 'error', error, true, false, false);
                            output = J$.N(2694745, 'output', output, true, false, false);
                            callback = J$.N(2694753, 'callback', callback, true, false, false);
                            if (J$.C(130224, J$.U(404802, '!', J$.R(2694657, 'error', error, false, false)))) {
                                J$.M(2694681, J$.R(2694665, 'this', this, false, false), 'push', false)(J$.T(2694673, null, 25, false));
                            }
                            J$.F(2694713, J$.R(2694689, 'callback', callback, false, false), false)(J$.R(2694697, 'error', error, false, false), J$.R(2694705, 'output', output, false, false));
                        } catch (J$e) {
                            J$.Ex(2694969, J$e);
                        } finally {
                            if (J$.Fr(2694977))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(2694769, 'async', async, false, false, false);
            wrapIterator = J$.N(2694785, 'wrapIterator', J$.T(2694777, wrapIterator, 12, false), true, false, false);
            wrapCallback = J$.N(2694801, 'wrapCallback', J$.T(2694793, wrapCallback, 12, false), true, false, false);
            wrapIteratorCallback = J$.N(2694817, 'wrapIteratorCallback', J$.T(2694809, wrapIteratorCallback, 12, false), true, false, false);
            streamer = J$.N(2694833, 'streamer', J$.T(2694825, streamer, 12, false), true, false, false);
            finisher = J$.N(2694849, 'finisher', J$.T(2694841, finisher, 12, false), true, false, false);
            J$.T(2693521, 'use strict', 21, false);
            var async = J$.W(2693553, 'async', J$.F(2693545, J$.I(typeof require === 'undefined' ? require = J$.R(2693529, 'require', undefined, true, true) : require = J$.R(2693529, 'require', require, true, true)), false)(J$.T(2693537, './async.js', 21, false)), async, false, true);
            J$.P(2693593, J$.I(typeof module === 'undefined' ? module = J$.R(2693561, 'module', undefined, true, true) : module = J$.R(2693561, 'module', module, true, true)), 'exports', J$.T(2693585, {
                iterator: J$.R(2693569, 'wrapIterator', wrapIterator, false, true),
                callback: J$.R(2693577, 'wrapCallback', wrapCallback, false, true)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(2694985, J$e);
        } finally {
            if (J$.Sr(2694993))
                continue jalangiLabel8;
            else
                break jalangiLabel8;
        }
    }
// JALANGI DO NOT INSTRUMENT
