// js-yaml/lib/js-yaml/type/bool_jalangi_.js
J$.iids = {"8":[5,7,5,20],"9":[2,12,2,19],"10":[5,7,5,20],"16":[9,24,9,58],"17":[2,20,2,29],"18":[9,10,9,19],"24":[9,24,9,77],"25":[2,12,2,30],"26":[9,24,9,39],"32":[9,10,9,78],"33":[2,12,2,30],"34":[9,43,9,58],"40":[10,17,10,53],"41":[2,12,2,30],"42":[9,62,9,77],"48":[10,17,10,73],"49":[5,7,5,11],"50":[10,3,10,12],"56":[10,3,10,74],"57":[5,16,5,20],"58":[10,17,10,33],"64":[9,10,10,74],"65":[5,29,5,34],"66":[10,37,10,53],"72":[14,10,15,18],"73":[5,29,5,34],"74":[10,57,10,73],"80":[14,10,16,18],"81":[5,22,5,35],"82":[14,10,14,25],"88":[29,51,29,57],"89":[7,13,7,17],"90":[15,3,15,18],"96":[30,51,30,57],"97":[7,13,7,24],"98":[16,3,16,18],"104":[31,51,31,57],"105":[7,13,7,24],"106":[20,10,20,71],"113":[7,13,7,24],"121":[9,10,9,13],"129":[9,18,9,19],"137":[9,24,9,28],"145":[9,33,9,39],"153":[9,43,9,47],"161":[9,52,9,58],"169":[9,62,9,66],"177":[9,71,9,77],"185":[10,3,10,6],"193":[10,11,10,12],"201":[10,17,10,21],"209":[10,26,10,33],"217":[10,37,10,41],"225":[10,46,10,53],"233":[10,57,10,61],"241":[10,66,10,73],"249":[9,10,10,74],"257":[9,3,10,75],"265":[4,1,11,2],"273":[4,1,11,2],"281":[4,1,11,2],"289":[4,1,11,2],"297":[14,10,14,14],"305":[14,19,14,25],"313":[15,3,15,7],"321":[15,12,15,18],"329":[16,3,16,7],"337":[16,12,16,18],"345":[14,10,16,18],"353":[14,3,16,19],"361":[13,1,17,2],"369":[13,1,17,2],"377":[13,1,17,2],"385":[20,10,20,16],"393":[20,10,20,26],"401":[20,10,20,35],"409":[20,41,20,47],"417":[20,10,20,48],"419":[20,10,20,40],"425":[20,53,20,71],"433":[20,10,20,71],"441":[20,3,20,72],"449":[19,1,21,2],"457":[19,1,21,2],"465":[19,1,21,2],"473":[23,1,23,7],"481":[23,22,23,26],"489":[23,27,23,51],"497":[24,9,24,17],"505":[25,12,25,30],"513":[26,14,26,34],"521":[27,14,27,23],"529":[29,51,29,57],"537":[29,60,29,66],"545":[29,69,29,76],"553":[29,51,29,76],"561":[29,44,29,77],"569":[29,16,29,78],"577":[29,16,29,78],"585":[29,16,29,78],"593":[29,16,29,78],"601":[29,16,29,78],"609":[30,51,30,57],"617":[30,60,30,66],"625":[30,69,30,76],"633":[30,51,30,76],"641":[30,44,30,77],"649":[30,16,30,78],"657":[30,16,30,78],"665":[30,16,30,78],"673":[30,16,30,78],"681":[30,16,30,78],"689":[31,51,31,57],"697":[31,60,31,66],"705":[31,69,31,76],"713":[31,51,31,76],"721":[31,44,31,77],"729":[31,16,31,78],"737":[31,16,31,78],"745":[31,16,31,78],"753":[31,16,31,78],"761":[31,16,31,78],"769":[28,14,31,80],"777":[33,17,33,28],"785":[23,53,33,30],"793":[23,18,33,31],"801":[23,1,33,31],"809":[23,1,33,32],"817":[1,1,33,32],"825":[1,1,33,32],"833":[4,1,11,2],"841":[1,1,33,32],"849":[13,1,17,2],"857":[1,1,33,32],"865":[19,1,21,2],"873":[1,1,33,32],"881":[5,3,5,35],"889":[4,1,11,2],"897":[4,1,11,2],"905":[13,1,17,2],"913":[13,1,17,2],"921":[19,1,21,2],"929":[19,1,21,2],"937":[29,51,29,76],"945":[29,16,29,78],"953":[29,16,29,78],"961":[30,51,30,76],"969":[30,16,30,78],"977":[30,16,30,78],"985":[31,51,31,76],"993":[31,16,31,78],"1001":[31,16,31,78],"1009":[1,1,33,32],"1017":[1,1,33,32],"nBranches":26,"originalCodeFileName":"/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/bool.js","instrumentedCodeFileName":"/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/bool_jalangi_.js","code":"'use strict';\n\nvar Type = require('../type');\n\nfunction resolveYamlBoolean(data) {\n  if (data === null) return false;\n\n  var max = data.length;\n\n  return (max === 4 && (data === 'true' || data === 'True' || data === 'TRUE')) ||\n         (max === 5 && (data === 'false' || data === 'False' || data === 'FALSE'));\n}\n\nfunction constructYamlBoolean(data) {\n  return data === 'true' ||\n         data === 'True' ||\n         data === 'TRUE';\n}\n\nfunction isBoolean(object) {\n  return Object.prototype.toString.call(object) === '[object Boolean]';\n}\n\nmodule.exports = new Type('tag:yaml.org,2002:bool', {\n  kind: 'scalar',\n  resolve: resolveYamlBoolean,\n  construct: constructYamlBoolean,\n  predicate: isBoolean,\n  represent: {\n    lowercase: function (object) { return object ? 'true' : 'false'; },\n    uppercase: function (object) { return object ? 'TRUE' : 'FALSE'; },\n    camelcase: function (object) { return object ? 'True' : 'False'; }\n  },\n  defaultStyle: 'lowercase'\n});\n"};
jalangiLabel110:
    while (true) {
        try {
            J$.Se(817, '/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/bool_jalangi_.js', '/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/bool.js');
            function resolveYamlBoolean(data) {
                jalangiLabel104:
                    while (true) {
                        try {
                            J$.Fe(265, arguments.callee, this, arguments);
                            arguments = J$.N(273, 'arguments', arguments, 4);
                            data = J$.N(281, 'data', data, 4);
                            J$.N(289, 'max', max, 0);
                            if (J$.X1(881, J$.C(8, J$.B(10, '===', J$.R(49, 'data', data, 0), J$.T(57, null, 25, false), 0))))
                                return J$.X1(81, J$.Rt(73, J$.T(65, false, 23, false)));
                            var max = J$.X1(113, J$.W(105, 'max', J$.G(97, J$.R(89, 'data', data, 0), 'length', 0), max, 1));
                            return J$.X1(257, J$.Rt(249, J$.C(64, J$.C(32, J$.B(18, '===', J$.R(121, 'max', max, 0), J$.T(129, 4, 22, false), 0)) ? J$.C(24, J$.C(16, J$.B(26, '===', J$.R(137, 'data', data, 0), J$.T(145, 'true', 21, false), 0)) ? J$._() : J$.B(34, '===', J$.R(153, 'data', data, 0), J$.T(161, 'True', 21, false), 0)) ? J$._() : J$.B(42, '===', J$.R(169, 'data', data, 0), J$.T(177, 'TRUE', 21, false), 0) : J$._()) ? J$._() : J$.C(56, J$.B(50, '===', J$.R(185, 'max', max, 0), J$.T(193, 5, 22, false), 0)) ? J$.C(48, J$.C(40, J$.B(58, '===', J$.R(201, 'data', data, 0), J$.T(209, 'false', 21, false), 0)) ? J$._() : J$.B(66, '===', J$.R(217, 'data', data, 0), J$.T(225, 'False', 21, false), 0)) ? J$._() : J$.B(74, '===', J$.R(233, 'data', data, 0), J$.T(241, 'FALSE', 21, false), 0) : J$._()));
                        } catch (J$e) {
                            J$.Ex(889, J$e);
                        } finally {
                            if (J$.Fr(897))
                                continue jalangiLabel104;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function constructYamlBoolean(data) {
                jalangiLabel105:
                    while (true) {
                        try {
                            J$.Fe(361, arguments.callee, this, arguments);
                            arguments = J$.N(369, 'arguments', arguments, 4);
                            data = J$.N(377, 'data', data, 4);
                            return J$.X1(353, J$.Rt(345, J$.C(80, J$.C(72, J$.B(82, '===', J$.R(297, 'data', data, 0), J$.T(305, 'true', 21, false), 0)) ? J$._() : J$.B(90, '===', J$.R(313, 'data', data, 0), J$.T(321, 'True', 21, false), 0)) ? J$._() : J$.B(98, '===', J$.R(329, 'data', data, 0), J$.T(337, 'TRUE', 21, false), 0)));
                        } catch (J$e) {
                            J$.Ex(905, J$e);
                        } finally {
                            if (J$.Fr(913))
                                continue jalangiLabel105;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isBoolean(object) {
                jalangiLabel106:
                    while (true) {
                        try {
                            J$.Fe(449, arguments.callee, this, arguments);
                            arguments = J$.N(457, 'arguments', arguments, 4);
                            object = J$.N(465, 'object', object, 4);
                            return J$.X1(441, J$.Rt(433, J$.B(106, '===', J$.M(417, J$.G(401, J$.G(393, J$.R(385, 'Object', Object, 2), 'prototype', 0), 'toString', 0), 'call', 0)(J$.R(409, 'object', object, 0)), J$.T(425, '[object Boolean]', 21, false), 0)));
                        } catch (J$e) {
                            J$.Ex(921, J$e);
                        } finally {
                            if (J$.Fr(929))
                                continue jalangiLabel106;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(825, 'Type', Type, 0);
            resolveYamlBoolean = J$.N(841, 'resolveYamlBoolean', J$.T(833, resolveYamlBoolean, 12, false, 265), 0);
            constructYamlBoolean = J$.N(857, 'constructYamlBoolean', J$.T(849, constructYamlBoolean, 12, false, 361), 0);
            isBoolean = J$.N(873, 'isBoolean', J$.T(865, isBoolean, 12, false, 449), 0);
            var Type = J$.X1(41, J$.W(33, 'Type', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, '../type', 21, false)), Type, 3));
            J$.X1(809, J$.P(801, J$.R(473, 'module', module, 2), 'exports', J$.F(793, J$.R(481, 'Type', Type, 1), 1)(J$.T(489, 'tag:yaml.org,2002:bool', 21, false), J$.T(785, {
                kind: J$.T(497, 'scalar', 21, false),
                resolve: J$.R(505, 'resolveYamlBoolean', resolveYamlBoolean, 1),
                construct: J$.R(513, 'constructYamlBoolean', constructYamlBoolean, 1),
                predicate: J$.R(521, 'isBoolean', isBoolean, 1),
                represent: J$.T(769, {
                    lowercase: J$.T(601, function lowercase(object) {
                        jalangiLabel107:
                            while (true) {
                                try {
                                    J$.Fe(569, arguments.callee, this, arguments);
                                    arguments = J$.N(577, 'arguments', arguments, 4);
                                    lowercase = J$.N(585, 'lowercase', lowercase, 0);
                                    object = J$.N(593, 'object', object, 4);
                                    return J$.X1(561, J$.Rt(553, J$.X1(937, J$.C(88, J$.R(529, 'object', object, 0))) ? J$.T(537, 'true', 21, false) : J$.T(545, 'false', 21, false)));
                                } catch (J$e) {
                                    J$.Ex(945, J$e);
                                } finally {
                                    if (J$.Fr(953))
                                        continue jalangiLabel107;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 569),
                    uppercase: J$.T(681, function uppercase(object) {
                        jalangiLabel108:
                            while (true) {
                                try {
                                    J$.Fe(649, arguments.callee, this, arguments);
                                    arguments = J$.N(657, 'arguments', arguments, 4);
                                    uppercase = J$.N(665, 'uppercase', uppercase, 0);
                                    object = J$.N(673, 'object', object, 4);
                                    return J$.X1(641, J$.Rt(633, J$.X1(961, J$.C(96, J$.R(609, 'object', object, 0))) ? J$.T(617, 'TRUE', 21, false) : J$.T(625, 'FALSE', 21, false)));
                                } catch (J$e) {
                                    J$.Ex(969, J$e);
                                } finally {
                                    if (J$.Fr(977))
                                        continue jalangiLabel108;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 649),
                    camelcase: J$.T(761, function camelcase(object) {
                        jalangiLabel109:
                            while (true) {
                                try {
                                    J$.Fe(729, arguments.callee, this, arguments);
                                    arguments = J$.N(737, 'arguments', arguments, 4);
                                    camelcase = J$.N(745, 'camelcase', camelcase, 0);
                                    object = J$.N(753, 'object', object, 4);
                                    return J$.X1(721, J$.Rt(713, J$.X1(985, J$.C(104, J$.R(689, 'object', object, 0))) ? J$.T(697, 'True', 21, false) : J$.T(705, 'False', 21, false)));
                                } catch (J$e) {
                                    J$.Ex(993, J$e);
                                } finally {
                                    if (J$.Fr(1001))
                                        continue jalangiLabel109;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 729)
                }, 11, false),
                defaultStyle: J$.T(777, 'lowercase', 21, false)
            }, 11, false)), 0));
        } catch (J$e) {
            J$.Ex(1009, J$e);
        } finally {
            if (J$.Sr(1017)) {
                J$.L();
                continue jalangiLabel110;
            } else {
                J$.L();
                break jalangiLabel110;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
