// js-yaml/lib/js-yaml/type/bool_jalangi_.js
J$.iids = {"8":[6,7,6,20],"9":[1,1,1,13],"10":[6,7,6,20],"16":[10,25,10,59],"17":[1,1,1,14],"18":[10,11,10,20],"24":[10,25,10,78],"25":[3,12,3,19],"26":[10,25,10,40],"32":[10,11,10,79],"33":[3,20,3,29],"34":[10,44,10,59],"40":[11,25,11,61],"41":[3,12,3,30],"42":[10,63,10,78],"48":[11,25,11,81],"49":[3,12,3,30],"50":[11,11,11,20],"56":[11,11,11,82],"57":[3,12,3,30],"58":[11,25,11,41],"64":[10,10,11,83],"65":[6,7,6,11],"66":[11,45,11,61],"72":[15,10,16,25],"73":[6,16,6,20],"74":[11,65,11,81],"80":[15,10,17,25],"81":[6,29,6,34],"82":[15,10,15,25],"88":[30,43,30,49],"89":[6,29,6,34],"90":[16,10,16,25],"96":[31,43,31,49],"97":[6,22,6,35],"98":[17,10,17,25],"104":[32,43,32,49],"105":[8,13,8,17],"106":[21,10,21,71],"113":[8,13,8,24],"121":[8,13,8,24],"129":[8,13,8,24],"137":[10,11,10,14],"145":[10,19,10,20],"153":[10,25,10,29],"161":[10,34,10,40],"169":[10,44,10,48],"177":[10,53,10,59],"185":[10,63,10,67],"193":[10,72,10,78],"201":[11,11,11,14],"209":[11,19,11,20],"217":[11,25,11,29],"225":[11,34,11,41],"233":[11,45,11,49],"241":[11,54,11,61],"249":[11,65,11,69],"257":[11,74,11,81],"265":[10,10,11,83],"273":[10,3,11,84],"281":[5,1,12,2],"289":[5,1,12,2],"297":[5,1,12,2],"305":[5,1,12,2],"313":[15,10,15,14],"321":[15,19,15,25],"329":[16,10,16,14],"337":[16,19,16,25],"345":[17,10,17,14],"353":[17,19,17,25],"361":[15,10,17,25],"369":[15,3,17,26],"377":[14,1,18,2],"385":[14,1,18,2],"393":[14,1,18,2],"401":[21,10,21,16],"409":[21,10,21,26],"417":[21,10,21,35],"425":[21,41,21,47],"433":[21,10,21,48],"435":[21,10,21,40],"441":[21,53,21,71],"449":[21,10,21,71],"457":[21,3,21,72],"465":[20,1,22,2],"473":[20,1,22,2],"481":[20,1,22,2],"489":[24,1,24,7],"497":[24,22,24,26],"505":[24,27,24,51],"513":[25,9,25,17],"521":[26,12,26,30],"529":[27,14,27,34],"537":[28,14,28,23],"545":[30,43,30,49],"553":[30,52,30,58],"561":[30,61,30,68],"569":[30,43,30,68],"577":[30,36,30,69],"585":[30,16,30,71],"593":[30,16,30,71],"601":[30,16,30,71],"609":[30,16,30,71],"617":[31,43,31,49],"625":[31,52,31,58],"633":[31,61,31,68],"641":[31,43,31,68],"649":[31,36,31,69],"657":[31,16,31,71],"665":[31,16,31,71],"673":[31,16,31,71],"681":[31,16,31,71],"689":[32,43,32,49],"697":[32,52,32,58],"705":[32,61,32,68],"713":[32,43,32,68],"721":[32,36,32,69],"729":[32,16,32,71],"737":[32,16,32,71],"745":[32,16,32,71],"753":[32,16,32,71],"761":[29,14,33,4],"769":[34,17,34,28],"777":[24,53,35,2],"785":[24,18,35,3],"793":[24,1,35,3],"801":[24,1,35,4],"809":[1,1,36,1],"817":[1,1,36,1],"825":[5,1,12,2],"833":[1,1,36,1],"841":[14,1,18,2],"849":[1,1,36,1],"857":[20,1,22,2],"865":[1,1,36,1],"873":[6,3,6,35],"881":[5,1,12,2],"889":[5,1,12,2],"897":[14,1,18,2],"905":[14,1,18,2],"913":[20,1,22,2],"921":[20,1,22,2],"929":[30,43,30,68],"937":[30,16,30,71],"945":[30,16,30,71],"953":[31,43,31,68],"961":[31,16,31,71],"969":[31,16,31,71],"977":[32,43,32,68],"985":[32,16,32,71],"993":[32,16,32,71],"1001":[1,1,36,1],"1009":[1,1,36,1],"nBranches":26,"originalCodeFileName":"/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/bool.js","instrumentedCodeFileName":"/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/bool_jalangi_.js","code":"'use strict';\n\nvar Type = require('../type');\n\nfunction resolveYamlBoolean(data) {\n  if (data === null) return false;\n\n  var max = data.length;\n\n  return (max === 4 && (data === 'true' || data === 'True' || data === 'TRUE')) ||\n         (max === 5 && (data === 'false' || data === 'False' || data === 'FALSE'));\n}\n\nfunction constructYamlBoolean(data) {\n  return data === 'true' ||\n         data === 'True' ||\n         data === 'TRUE';\n}\n\nfunction isBoolean(object) {\n  return Object.prototype.toString.call(object) === '[object Boolean]';\n}\n\nmodule.exports = new Type('tag:yaml.org,2002:bool', {\n  kind: 'scalar',\n  resolve: resolveYamlBoolean,\n  construct: constructYamlBoolean,\n  predicate: isBoolean,\n  represent: {\n    lowercase: function (object) { return object ? 'true' : 'false'; },\n    uppercase: function (object) { return object ? 'TRUE' : 'FALSE'; },\n    camelcase: function (object) { return object ? 'True' : 'False'; }\n  },\n  defaultStyle: 'lowercase'\n});\n"};
jalangiLabel106:
    while (true) {
        try {
            J$.Se(809, '/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/bool_jalangi_.js', '/home/james/nodejs/HiPar/tests/target/TestJs-yaml/node_modules/js-yaml/lib/js-yaml/type/bool.js');
            function resolveYamlBoolean(data) {
                jalangiLabel100:
                    while (true) {
                        try {
                            J$.Fe(281, arguments.callee, this, arguments);
                            arguments = J$.N(289, 'arguments', arguments, 4);
                            data = J$.N(297, 'data', data, 4);
                            J$.N(305, 'max', max, 0);
                            if (J$.X1(873, J$.C(8, J$.B(10, '===', J$.R(65, 'data', data, 0), J$.T(73, null, 25, false), 0))))
                                return J$.X1(97, J$.Rt(89, J$.T(81, false, 23, false)));
                            var max = J$.X1(129, J$.W(121, 'max', J$.G(113, J$.R(105, 'data', data, 0), 'length', 0), max, 1));
                            return J$.X1(273, J$.Rt(265, J$.C(64, J$.C(32, J$.B(18, '===', J$.R(137, 'max', max, 0), J$.T(145, 4, 22, false), 0)) ? J$.C(24, J$.C(16, J$.B(26, '===', J$.R(153, 'data', data, 0), J$.T(161, 'true', 21, false), 0)) ? J$._() : J$.B(34, '===', J$.R(169, 'data', data, 0), J$.T(177, 'True', 21, false), 0)) ? J$._() : J$.B(42, '===', J$.R(185, 'data', data, 0), J$.T(193, 'TRUE', 21, false), 0) : J$._()) ? J$._() : J$.C(56, J$.B(50, '===', J$.R(201, 'max', max, 0), J$.T(209, 5, 22, false), 0)) ? J$.C(48, J$.C(40, J$.B(58, '===', J$.R(217, 'data', data, 0), J$.T(225, 'false', 21, false), 0)) ? J$._() : J$.B(66, '===', J$.R(233, 'data', data, 0), J$.T(241, 'False', 21, false), 0)) ? J$._() : J$.B(74, '===', J$.R(249, 'data', data, 0), J$.T(257, 'FALSE', 21, false), 0) : J$._()));
                        } catch (J$e) {
                            J$.Ex(881, J$e);
                        } finally {
                            if (J$.Fr(889))
                                continue jalangiLabel100;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function constructYamlBoolean(data) {
                jalangiLabel101:
                    while (true) {
                        try {
                            J$.Fe(377, arguments.callee, this, arguments);
                            arguments = J$.N(385, 'arguments', arguments, 4);
                            data = J$.N(393, 'data', data, 4);
                            return J$.X1(369, J$.Rt(361, J$.C(80, J$.C(72, J$.B(82, '===', J$.R(313, 'data', data, 0), J$.T(321, 'true', 21, false), 0)) ? J$._() : J$.B(90, '===', J$.R(329, 'data', data, 0), J$.T(337, 'True', 21, false), 0)) ? J$._() : J$.B(98, '===', J$.R(345, 'data', data, 0), J$.T(353, 'TRUE', 21, false), 0)));
                        } catch (J$e) {
                            J$.Ex(897, J$e);
                        } finally {
                            if (J$.Fr(905))
                                continue jalangiLabel101;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isBoolean(object) {
                jalangiLabel102:
                    while (true) {
                        try {
                            J$.Fe(465, arguments.callee, this, arguments);
                            arguments = J$.N(473, 'arguments', arguments, 4);
                            object = J$.N(481, 'object', object, 4);
                            return J$.X1(457, J$.Rt(449, J$.B(106, '===', J$.M(433, J$.G(417, J$.G(409, J$.R(401, 'Object', Object, 2), 'prototype', 0), 'toString', 0), 'call', 0)(J$.R(425, 'object', object, 0)), J$.T(441, '[object Boolean]', 21, false), 0)));
                        } catch (J$e) {
                            J$.Ex(913, J$e);
                        } finally {
                            if (J$.Fr(921))
                                continue jalangiLabel102;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(817, 'Type', Type, 0);
            resolveYamlBoolean = J$.N(833, 'resolveYamlBoolean', J$.T(825, resolveYamlBoolean, 12, false, 281), 0);
            constructYamlBoolean = J$.N(849, 'constructYamlBoolean', J$.T(841, constructYamlBoolean, 12, false, 377), 0);
            isBoolean = J$.N(865, 'isBoolean', J$.T(857, isBoolean, 12, false, 465), 0);
            J$.X1(17, J$.T(9, 'use strict', 21, false));
            var Type = J$.X1(57, J$.W(49, 'Type', J$.F(41, J$.R(25, 'require', require, 2), 0)(J$.T(33, '../type', 21, false)), Type, 3));
            J$.X1(801, J$.P(793, J$.R(489, 'module', module, 2), 'exports', J$.F(785, J$.R(497, 'Type', Type, 1), 1)(J$.T(505, 'tag:yaml.org,2002:bool', 21, false), J$.T(777, {
                kind: J$.T(513, 'scalar', 21, false),
                resolve: J$.R(521, 'resolveYamlBoolean', resolveYamlBoolean, 1),
                construct: J$.R(529, 'constructYamlBoolean', constructYamlBoolean, 1),
                predicate: J$.R(537, 'isBoolean', isBoolean, 1),
                represent: J$.T(761, {
                    lowercase: J$.T(609, function (object) {
                        jalangiLabel103:
                            while (true) {
                                try {
                                    J$.Fe(585, arguments.callee, this, arguments);
                                    arguments = J$.N(593, 'arguments', arguments, 4);
                                    object = J$.N(601, 'object', object, 4);
                                    return J$.X1(577, J$.Rt(569, J$.X1(929, J$.C(88, J$.R(545, 'object', object, 0))) ? J$.T(553, 'true', 21, false) : J$.T(561, 'false', 21, false)));
                                } catch (J$e) {
                                    J$.Ex(937, J$e);
                                } finally {
                                    if (J$.Fr(945))
                                        continue jalangiLabel103;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 585),
                    uppercase: J$.T(681, function (object) {
                        jalangiLabel104:
                            while (true) {
                                try {
                                    J$.Fe(657, arguments.callee, this, arguments);
                                    arguments = J$.N(665, 'arguments', arguments, 4);
                                    object = J$.N(673, 'object', object, 4);
                                    return J$.X1(649, J$.Rt(641, J$.X1(953, J$.C(96, J$.R(617, 'object', object, 0))) ? J$.T(625, 'TRUE', 21, false) : J$.T(633, 'FALSE', 21, false)));
                                } catch (J$e) {
                                    J$.Ex(961, J$e);
                                } finally {
                                    if (J$.Fr(969))
                                        continue jalangiLabel104;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 657),
                    camelcase: J$.T(753, function (object) {
                        jalangiLabel105:
                            while (true) {
                                try {
                                    J$.Fe(729, arguments.callee, this, arguments);
                                    arguments = J$.N(737, 'arguments', arguments, 4);
                                    object = J$.N(745, 'object', object, 4);
                                    return J$.X1(721, J$.Rt(713, J$.X1(977, J$.C(104, J$.R(689, 'object', object, 0))) ? J$.T(697, 'True', 21, false) : J$.T(705, 'False', 21, false)));
                                } catch (J$e) {
                                    J$.Ex(985, J$e);
                                } finally {
                                    if (J$.Fr(993))
                                        continue jalangiLabel105;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 729)
                }, 11, false),
                defaultStyle: J$.T(769, 'lowercase', 21, false)
            }, 11, false)), 0));
        } catch (J$e) {
            J$.Ex(1001, J$e);
        } finally {
            if (J$.Sr(1009)) {
                J$.L();
                continue jalangiLabel106;
            } else {
                J$.L();
                break jalangiLabel106;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
