// couchbase/lib/bucketmgr.js
J$.noInstrEval = false;
jalangiLabel45:
    while (true) {
        try {
            J$.Se(7147241, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/couchbase/lib/bucketmgr.js');
            function _respRead(callback) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7137889, arguments.callee, this, arguments);
                            arguments = J$.N(7137897, 'arguments', arguments, true, false, false);
                            callback = J$.N(7137905, 'callback', callback, true, false, false);
                            return J$.Rt(7137881, J$.T(7137873, function (resp) {
                                jalangiLabel3:
                                    while (true) {
                                        try {
                                            J$.Fe(7137841, arguments.callee, this, arguments);
                                            arguments = J$.N(7137849, 'arguments', arguments, true, false, false);
                                            resp = J$.N(7137857, 'resp', resp, true, false, false);
                                            J$.N(7137865, 'strBuffer', strBuffer, false, false, false);
                                            J$.M(7137553, J$.R(7137537, 'resp', resp, false, false), 'setEncoding', false)(J$.T(7137545, 'utf8', 21, false));
                                            var strBuffer = J$.W(7137569, 'strBuffer', J$.T(7137561, '', 21, false), strBuffer, false, false);
                                            J$.M(7137649, J$.R(7137577, 'resp', resp, false, false), 'on', false)(J$.T(7137585, 'data', 21, false), J$.T(7137641, function (data) {
                                                jalangiLabel0:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7137617, arguments.callee, this, arguments);
                                                            arguments = J$.N(7137625, 'arguments', arguments, true, false, false);
                                                            data = J$.N(7137633, 'data', data, true, false, false);
                                                            strBuffer = J$.W(7137609, 'strBuffer', J$.B(721218, '+', J$.R(7137601, 'strBuffer', strBuffer, false, false), J$.R(7137593, 'data', data, false, false)), strBuffer, false, false);
                                                        } catch (J$e) {
                                                            J$.Ex(7147289, J$e);
                                                        } finally {
                                                            if (J$.Fr(7147297))
                                                                continue jalangiLabel0;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                            J$.M(7137737, J$.R(7137657, 'resp', resp, false, false), 'on', false)(J$.T(7137665, 'end', 21, false), J$.T(7137729, function () {
                                                jalangiLabel1:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7137713, arguments.callee, this, arguments);
                                                            arguments = J$.N(7137721, 'arguments', arguments, true, false, false);
                                                            J$.F(7137705, J$.R(7137673, 'callback', callback, false, false), false)(J$.T(7137681, null, 25, false), J$.R(7137689, 'resp', resp, false, false), J$.R(7137697, 'strBuffer', strBuffer, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7147305, J$e);
                                                        } finally {
                                                            if (J$.Fr(7147313))
                                                                continue jalangiLabel1;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                            J$.M(7137833, J$.R(7137745, 'resp', resp, false, false), 'on', false)(J$.T(7137753, 'error', 21, false), J$.T(7137825, function (err) {
                                                jalangiLabel2:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7137801, arguments.callee, this, arguments);
                                                            arguments = J$.N(7137809, 'arguments', arguments, true, false, false);
                                                            err = J$.N(7137817, 'err', err, true, false, false);
                                                            J$.F(7137793, J$.R(7137761, 'callback', callback, false, false), false)(J$.R(7137769, 'err', err, false, false), J$.R(7137777, 'resp', resp, false, false), J$.T(7137785, null, 25, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7147321, J$e);
                                                        } finally {
                                                            if (J$.Fr(7147329))
                                                                continue jalangiLabel2;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(7147337, J$e);
                                        } finally {
                                            if (J$.Fr(7147345))
                                                continue jalangiLabel3;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147353, J$e);
                        } finally {
                            if (J$.Fr(7147361))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function BucketManager(bucket) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(7137937, arguments.callee, this, arguments);
                            arguments = J$.N(7137945, 'arguments', arguments, true, false, false);
                            bucket = J$.N(7137953, 'bucket', bucket, true, false, false);
                            J$.P(7137929, J$.R(7137913, 'this', this, false, false), '_bucket', J$.R(7137921, 'bucket', bucket, false, false));
                        } catch (J$e) {
                            J$.Ex(7147369, J$e);
                        } finally {
                            if (J$.Fr(7147377))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(7147249, 'N1qlQuery', N1qlQuery, false, false, false);
            _respRead = J$.N(7147265, '_respRead', J$.T(7147257, _respRead, 12, false), true, false, false);
            BucketManager = J$.N(7147281, 'BucketManager', J$.T(7147273, BucketManager, 12, false), true, false, false);
            J$.T(7137497, 'use strict', 21, false);
            var N1qlQuery = J$.W(7137529, 'N1qlQuery', J$.F(7137521, J$.I(typeof require === 'undefined' ? require = J$.R(7137505, 'require', undefined, true, true) : require = J$.R(7137505, 'require', require, true, true)), false)(J$.T(7137513, './n1qlquery', 21, false)), N1qlQuery, false, true);
            J$.P(7138137, J$.G(7137969, J$.R(7137961, 'BucketManager', BucketManager, false, true), 'prototype'), '_mgmtRequest', J$.T(7138129, function (path, method, callback) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(7138081, arguments.callee, this, arguments);
                            arguments = J$.N(7138089, 'arguments', arguments, true, false, false);
                            path = J$.N(7138097, 'path', path, true, false, false);
                            method = J$.N(7138105, 'method', method, true, false, false);
                            callback = J$.N(7138113, 'callback', callback, true, false, false);
                            J$.N(7138121, 'b', b, false, false, false);
                            var b = J$.W(7137993, 'b', J$.G(7137985, J$.R(7137977, 'this', this, false, false), '_bucket'), b, false, false);
                            J$.M(7138073, J$.R(7138001, 'b', b, false, false), '_maybeInvoke', false)(J$.M(7138033, J$.G(7138017, J$.R(7138009, 'b', b, false, false), '_mgmtRequest'), 'bind', false)(J$.R(7138025, 'b', b, false, false)), J$.T(7138065, [
                                J$.R(7138041, 'path', path, false, false),
                                J$.R(7138049, 'method', method, false, false),
                                J$.R(7138057, 'callback', callback, false, false)
                            ], 10, false));
                        } catch (J$e) {
                            J$.Ex(7147385, J$e);
                        } finally {
                            if (J$.Fr(7147393))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7138321, J$.G(7138153, J$.R(7138145, 'BucketManager', BucketManager, false, true), 'prototype'), '_capiRequest', J$.T(7138313, function (path, method, callback) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(7138265, arguments.callee, this, arguments);
                            arguments = J$.N(7138273, 'arguments', arguments, true, false, false);
                            path = J$.N(7138281, 'path', path, true, false, false);
                            method = J$.N(7138289, 'method', method, true, false, false);
                            callback = J$.N(7138297, 'callback', callback, true, false, false);
                            J$.N(7138305, 'b', b, false, false, false);
                            var b = J$.W(7138177, 'b', J$.G(7138169, J$.R(7138161, 'this', this, false, false), '_bucket'), b, false, false);
                            J$.M(7138257, J$.R(7138185, 'b', b, false, false), '_maybeInvoke', false)(J$.M(7138217, J$.G(7138201, J$.R(7138193, 'b', b, false, false), '_capiRequest'), 'bind', false)(J$.R(7138209, 'b', b, false, false)), J$.T(7138249, [
                                J$.R(7138225, 'path', path, false, false),
                                J$.R(7138233, 'method', method, false, false),
                                J$.R(7138241, 'callback', callback, false, false)
                            ], 10, false));
                        } catch (J$e) {
                            J$.Ex(7147401, J$e);
                        } finally {
                            if (J$.Fr(7147409))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7139289, J$.G(7138337, J$.R(7138329, 'BucketManager', BucketManager, false, true), 'prototype'), 'getDesignDocuments', J$.T(7139281, function (callback) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(7139249, arguments.callee, this, arguments);
                            arguments = J$.N(7139257, 'arguments', arguments, true, false, false);
                            callback = J$.N(7139265, 'callback', callback, true, false, false);
                            J$.N(7139273, 'path', path, false, false, false);
                            var path = J$.W(7138385, 'path', J$.B(721234, '+', J$.B(721226, '+', J$.T(7138345, 'pools/default/buckets/', 21, false), J$.G(7138369, J$.G(7138361, J$.R(7138353, 'this', this, false, false), '_bucket'), '_name')), J$.T(7138377, '/ddocs', 21, false)), path, false, false);
                            J$.M(7139241, J$.R(7138393, 'this', this, false, false), '_mgmtRequest', false)(J$.R(7138401, 'path', path, false, false), J$.T(7138409, 'GET', 21, false), J$.T(7139233, function (err, httpReq) {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(7139201, arguments.callee, this, arguments);
                                            arguments = J$.N(7139209, 'arguments', arguments, true, false, false);
                                            err = J$.N(7139217, 'err', err, true, false, false);
                                            httpReq = J$.N(7139225, 'httpReq', httpReq, true, false, false);
                                            if (J$.C(382232, J$.R(7138417, 'err', err, false, false))) {
                                                return J$.Rt(7138457, J$.F(7138449, J$.R(7138425, 'callback', callback, false, false), false)(J$.R(7138433, 'err', err, false, false), J$.T(7138441, null, 25, false)));
                                            }
                                            J$.M(7138489, J$.R(7138465, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7138473, 'error', 21, false), J$.R(7138481, 'callback', callback, false, false));
                                            J$.M(7139177, J$.R(7138497, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7138505, 'response', 21, false), J$.F(7139169, J$.R(7138513, '_respRead', _respRead, false, true), false)(J$.T(7139161, function (err, resp, data) {
                                                jalangiLabel8:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7139073, arguments.callee, this, arguments);
                                                            arguments = J$.N(7139081, 'arguments', arguments, true, false, false);
                                                            err = J$.N(7139089, 'err', err, true, false, false);
                                                            resp = J$.N(7139097, 'resp', resp, true, false, false);
                                                            data = J$.N(7139105, 'data', data, true, false, false);
                                                            J$.N(7139113, 'errData', errData, false, false, false);
                                                            J$.N(7139121, 'ddocData', ddocData, false, false, false);
                                                            J$.N(7139129, 'ddocs', ddocs, false, false, false);
                                                            J$.N(7139137, 'i', i, false, false, false);
                                                            J$.N(7139145, 'ddoc', ddoc, false, false, false);
                                                            J$.N(7139153, 'ddocName', ddocName, false, false, false);
                                                            if (J$.C(382240, J$.R(7138521, 'err', err, false, false))) {
                                                                return J$.Rt(7138553, J$.F(7138545, J$.R(7138529, 'callback', callback, false, false), false)(J$.R(7138537, 'err', err, false, false)));
                                                            }
                                                            if (J$.C(382256, J$.B(721242, '!==', J$.G(7138569, J$.R(7138561, 'resp', resp, false, false), 'statusCode'), J$.T(7138577, 200, 22, false)))) {
                                                                var errData = J$.W(7138593, 'errData', J$.T(7138585, null, 25, false), errData, false, false);
                                                                try {
                                                                    errData = J$.W(7138625, 'errData', J$.M(7138617, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7138601, 'JSON', undefined, true, true) : JSON = J$.R(7138601, 'JSON', JSON, true, true)), 'parse', false)(J$.R(7138609, 'data', data, false, false)), errData, false, false);
                                                                } catch (e) {
                                                                    J$.N(7138633, 'e', e, false, false, true);
                                                                }
                                                                if (J$.C(382248, J$.U(721250, '!', J$.R(7138641, 'errData', errData, false, false)))) {
                                                                    J$.F(7138713, J$.R(7138649, 'callback', callback, false, false), false)(J$.F(7138697, J$.I(typeof Error === 'undefined' ? Error = J$.R(7138657, 'Error', undefined, true, true) : Error = J$.R(7138657, 'Error', Error, true, true)), true)(J$.B(721266, '+', J$.B(721258, '+', J$.T(7138665, 'operation failed (', 21, false), J$.G(7138681, J$.R(7138673, 'resp', resp, false, false), 'statusCode')), J$.T(7138689, ')', 21, false))), J$.T(7138705, null, 25, false));
                                                                    return J$.Rt(7138721, undefined);
                                                                }
                                                                J$.F(7138777, J$.R(7138729, 'callback', callback, false, false), false)(J$.F(7138761, J$.I(typeof Error === 'undefined' ? Error = J$.R(7138737, 'Error', undefined, true, true) : Error = J$.R(7138737, 'Error', Error, true, true)), true)(J$.G(7138753, J$.R(7138745, 'errData', errData, false, false), 'reason')), J$.T(7138769, null, 25, false));
                                                                return J$.Rt(7138785, undefined);
                                                            }
                                                            var ddocData = J$.W(7138817, 'ddocData', J$.M(7138809, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7138793, 'JSON', undefined, true, true) : JSON = J$.R(7138793, 'JSON', JSON, true, true)), 'parse', false)(J$.R(7138801, 'data', data, false, false)), ddocData, false, false);
                                                            var ddocs = J$.W(7138833, 'ddocs', J$.T(7138825, {}, 11, false), ddocs, false, false);
                                                            for (var i = J$.W(7138849, 'i', J$.T(7138841, 0, 22, false), i, false, false); J$.C(382264, J$.B(721274, '<', J$.R(7138857, 'i', i, false, false), J$.G(7138881, J$.G(7138873, J$.R(7138865, 'ddocData', ddocData, false, false), 'rows'), 'length'))); i = J$.W(7138897, 'i', J$.B(721290, '+', J$.U(721282, '+', J$.R(7138889, 'i', i, false, false)), 1), i, false, false)) {
                                                                var ddoc = J$.W(7138945, 'ddoc', J$.G(7138937, J$.G(7138929, J$.G(7138913, J$.R(7138905, 'ddocData', ddocData, false, false), 'rows'), J$.R(7138921, 'i', i, false, false)), 'doc'), ddoc, false, false);
                                                                var ddocName = J$.W(7138993, 'ddocName', J$.M(7138985, J$.G(7138969, J$.G(7138961, J$.R(7138953, 'ddoc', ddoc, false, false), 'meta'), 'id'), 'substr', false)(J$.T(7138977, 8, 22, false)), ddocName, false, false);
                                                                J$.P(7139033, J$.R(7139001, 'ddocs', ddocs, false, false), J$.R(7139009, 'ddocName', ddocName, false, false), J$.G(7139025, J$.R(7139017, 'ddoc', ddoc, false, false), 'json'));
                                                            }
                                                            J$.F(7139065, J$.R(7139041, 'callback', callback, false, false), false)(J$.T(7139049, null, 25, false), J$.R(7139057, 'ddocs', ddocs, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7147417, J$e);
                                                        } finally {
                                                            if (J$.Fr(7147425))
                                                                continue jalangiLabel8;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)));
                                            J$.M(7139193, J$.R(7139185, 'httpReq', httpReq, false, false), 'end', false)();
                                        } catch (J$e) {
                                            J$.Ex(7147433, J$e);
                                        } finally {
                                            if (J$.Fr(7147441))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147449, J$e);
                        } finally {
                            if (J$.Fr(7147457))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7139553, J$.G(7139305, J$.R(7139297, 'BucketManager', BucketManager, false, true), 'prototype'), 'insertDesignDocument', J$.T(7139545, function (name, data, callback) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(7139497, arguments.callee, this, arguments);
                            arguments = J$.N(7139505, 'arguments', arguments, true, false, false);
                            name = J$.N(7139513, 'name', name, true, false, false);
                            data = J$.N(7139521, 'data', data, true, false, false);
                            callback = J$.N(7139529, 'callback', callback, true, false, false);
                            J$.N(7139537, 'self', self, false, false, false);
                            var self = J$.W(7139321, 'self', J$.R(7139313, 'this', this, false, false), self, false, false);
                            J$.M(7139489, J$.R(7139329, 'this', this, false, false), 'getDesignDocument', false)(J$.R(7139337, 'name', name, false, false), J$.T(7139481, function (err, res) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(7139449, arguments.callee, this, arguments);
                                            arguments = J$.N(7139457, 'arguments', arguments, true, false, false);
                                            err = J$.N(7139465, 'err', err, true, false, false);
                                            res = J$.N(7139473, 'res', res, true, false, false);
                                            if (J$.C(382272, J$.U(721298, '!', J$.R(7139345, 'err', err, false, false)))) {
                                                return J$.Rt(7139401, J$.F(7139393, J$.R(7139353, 'callback', callback, false, false), false)(J$.F(7139377, J$.I(typeof Error === 'undefined' ? Error = J$.R(7139361, 'Error', undefined, true, true) : Error = J$.R(7139361, 'Error', Error, true, true)), true)(J$.T(7139369, 'design document already exists', 21, false)), J$.T(7139385, null, 25, false)));
                                            }
                                            J$.M(7139441, J$.R(7139409, 'self', self, false, false), 'upsertDesignDocument', false)(J$.R(7139417, 'name', name, false, false), J$.R(7139425, 'data', data, false, false), J$.R(7139433, 'callback', callback, false, false));
                                        } catch (J$e) {
                                            J$.Ex(7147465, J$e);
                                        } finally {
                                            if (J$.Fr(7147473))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147481, J$e);
                        } finally {
                            if (J$.Fr(7147489))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7140361, J$.G(7139569, J$.R(7139561, 'BucketManager', BucketManager, false, true), 'prototype'), 'upsertDesignDocument', J$.T(7140353, function (name, data, callback) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(7140305, arguments.callee, this, arguments);
                            arguments = J$.N(7140313, 'arguments', arguments, true, false, false);
                            name = J$.N(7140321, 'name', name, true, false, false);
                            data = J$.N(7140329, 'data', data, true, false, false);
                            callback = J$.N(7140337, 'callback', callback, true, false, false);
                            J$.N(7140345, 'path', path, false, false, false);
                            var path = J$.W(7139593, 'path', J$.B(721306, '+', J$.T(7139577, '_design/', 21, false), J$.R(7139585, 'name', name, false, false)), path, false, false);
                            J$.M(7140297, J$.R(7139601, 'this', this, false, false), '_capiRequest', false)(J$.R(7139609, 'path', path, false, false), J$.T(7139617, 'PUT', 21, false), J$.T(7140289, function (err, httpReq) {
                                jalangiLabel15:
                                    while (true) {
                                        try {
                                            J$.Fe(7140257, arguments.callee, this, arguments);
                                            arguments = J$.N(7140265, 'arguments', arguments, true, false, false);
                                            err = J$.N(7140273, 'err', err, true, false, false);
                                            httpReq = J$.N(7140281, 'httpReq', httpReq, true, false, false);
                                            if (J$.C(382280, J$.R(7139625, 'err', err, false, false))) {
                                                return J$.Rt(7139665, J$.F(7139657, J$.R(7139633, 'callback', callback, false, false), false)(J$.R(7139641, 'err', err, false, false), J$.T(7139649, null, 25, false)));
                                            }
                                            J$.M(7139697, J$.R(7139673, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7139681, 'error', 21, false), J$.R(7139689, 'callback', callback, false, false));
                                            J$.M(7140097, J$.R(7139705, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7139713, 'response', 21, false), J$.F(7140089, J$.R(7139721, '_respRead', _respRead, false, true), false)(J$.T(7140081, function (err, resp, data) {
                                                jalangiLabel13:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7140033, arguments.callee, this, arguments);
                                                            arguments = J$.N(7140041, 'arguments', arguments, true, false, false);
                                                            err = J$.N(7140049, 'err', err, true, false, false);
                                                            resp = J$.N(7140057, 'resp', resp, true, false, false);
                                                            data = J$.N(7140065, 'data', data, true, false, false);
                                                            J$.N(7140073, 'errData', errData, false, false, false);
                                                            if (J$.C(382288, J$.R(7139729, 'err', err, false, false))) {
                                                                return J$.Rt(7139761, J$.F(7139753, J$.R(7139737, 'callback', callback, false, false), false)(J$.R(7139745, 'err', err, false, false)));
                                                            }
                                                            if (J$.C(382304, J$.B(721314, '!==', J$.G(7139777, J$.R(7139769, 'resp', resp, false, false), 'statusCode'), J$.T(7139785, 201, 22, false)))) {
                                                                var errData = J$.W(7139801, 'errData', J$.T(7139793, null, 25, false), errData, false, false);
                                                                try {
                                                                    errData = J$.W(7139833, 'errData', J$.M(7139825, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7139809, 'JSON', undefined, true, true) : JSON = J$.R(7139809, 'JSON', JSON, true, true)), 'parse', false)(J$.R(7139817, 'data', data, false, false)), errData, false, false);
                                                                } catch (e) {
                                                                    J$.N(7139841, 'e', e, false, false, true);
                                                                }
                                                                if (J$.C(382296, J$.U(721322, '!', J$.R(7139849, 'errData', errData, false, false)))) {
                                                                    J$.F(7139921, J$.R(7139857, 'callback', callback, false, false), false)(J$.F(7139905, J$.I(typeof Error === 'undefined' ? Error = J$.R(7139865, 'Error', undefined, true, true) : Error = J$.R(7139865, 'Error', Error, true, true)), true)(J$.B(721338, '+', J$.B(721330, '+', J$.T(7139873, 'operation failed (', 21, false), J$.G(7139889, J$.R(7139881, 'resp', resp, false, false), 'statusCode')), J$.T(7139897, ')', 21, false))), J$.T(7139913, null, 25, false));
                                                                    return J$.Rt(7139929, undefined);
                                                                }
                                                                J$.F(7139985, J$.R(7139937, 'callback', callback, false, false), false)(J$.F(7139969, J$.I(typeof Error === 'undefined' ? Error = J$.R(7139945, 'Error', undefined, true, true) : Error = J$.R(7139945, 'Error', Error, true, true)), true)(J$.G(7139961, J$.R(7139953, 'errData', errData, false, false), 'reason')), J$.T(7139977, null, 25, false));
                                                                return J$.Rt(7139993, undefined);
                                                            }
                                                            J$.F(7140025, J$.R(7140001, 'callback', callback, false, false), false)(J$.T(7140009, null, 25, false), J$.T(7140017, true, 23, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7147497, J$e);
                                                        } finally {
                                                            if (J$.Fr(7147505))
                                                                continue jalangiLabel13;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)));
                                            J$.M(7140233, J$.R(7140105, 'httpReq', httpReq, false, false), 'write', false)(J$.M(7140225, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7140113, 'JSON', undefined, true, true) : JSON = J$.R(7140113, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(7140121, 'data', data, false, false), J$.T(7140217, function (key, value) {
                                                jalangiLabel14:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7140185, arguments.callee, this, arguments);
                                                            arguments = J$.N(7140193, 'arguments', arguments, true, false, false);
                                                            key = J$.N(7140201, 'key', key, true, false, false);
                                                            value = J$.N(7140209, 'value', value, true, false, false);
                                                            if (J$.C(382312, J$.B(721346, 'instanceof', J$.R(7140129, 'value', value, false, false), J$.I(typeof Function === 'undefined' ? Function = J$.R(7140137, 'Function', undefined, true, true) : Function = J$.R(7140137, 'Function', Function, true, true))))) {
                                                                return J$.Rt(7140161, J$.M(7140153, J$.R(7140145, 'value', value, false, false), 'toString', false)());
                                                            }
                                                            return J$.Rt(7140177, J$.R(7140169, 'value', value, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7147513, J$e);
                                                        } finally {
                                                            if (J$.Fr(7147521))
                                                                continue jalangiLabel14;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)));
                                            J$.M(7140249, J$.R(7140241, 'httpReq', httpReq, false, false), 'end', false)();
                                        } catch (J$e) {
                                            J$.Ex(7147529, J$e);
                                        } finally {
                                            if (J$.Fr(7147537))
                                                continue jalangiLabel15;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147545, J$e);
                        } finally {
                            if (J$.Fr(7147553))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7141065, J$.G(7140377, J$.R(7140369, 'BucketManager', BucketManager, false, true), 'prototype'), 'getDesignDocument', J$.T(7141057, function (name, callback) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(7141017, arguments.callee, this, arguments);
                            arguments = J$.N(7141025, 'arguments', arguments, true, false, false);
                            name = J$.N(7141033, 'name', name, true, false, false);
                            callback = J$.N(7141041, 'callback', callback, true, false, false);
                            J$.N(7141049, 'path', path, false, false, false);
                            var path = J$.W(7140401, 'path', J$.B(721354, '+', J$.T(7140385, '_design/', 21, false), J$.R(7140393, 'name', name, false, false)), path, false, false);
                            J$.M(7141009, J$.R(7140409, 'this', this, false, false), '_capiRequest', false)(J$.R(7140417, 'path', path, false, false), J$.T(7140425, 'GET', 21, false), J$.T(7141001, function (err, httpReq) {
                                jalangiLabel18:
                                    while (true) {
                                        try {
                                            J$.Fe(7140969, arguments.callee, this, arguments);
                                            arguments = J$.N(7140977, 'arguments', arguments, true, false, false);
                                            err = J$.N(7140985, 'err', err, true, false, false);
                                            httpReq = J$.N(7140993, 'httpReq', httpReq, true, false, false);
                                            if (J$.C(382320, J$.R(7140433, 'err', err, false, false))) {
                                                return J$.Rt(7140473, J$.F(7140465, J$.R(7140441, 'callback', callback, false, false), false)(J$.R(7140449, 'err', err, false, false), J$.T(7140457, null, 25, false)));
                                            }
                                            J$.M(7140505, J$.R(7140481, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7140489, 'error', 21, false), J$.R(7140497, 'callback', callback, false, false));
                                            J$.M(7140945, J$.R(7140513, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7140521, 'response', 21, false), J$.F(7140937, J$.R(7140529, '_respRead', _respRead, false, true), false)(J$.T(7140929, function (err, resp, data) {
                                                jalangiLabel17:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7140873, arguments.callee, this, arguments);
                                                            arguments = J$.N(7140881, 'arguments', arguments, true, false, false);
                                                            err = J$.N(7140889, 'err', err, true, false, false);
                                                            resp = J$.N(7140897, 'resp', resp, true, false, false);
                                                            data = J$.N(7140905, 'data', data, true, false, false);
                                                            J$.N(7140913, 'ddocData', ddocData, false, false, false);
                                                            J$.N(7140921, 'errData', errData, false, false, false);
                                                            if (J$.C(382328, J$.R(7140537, 'err', err, false, false))) {
                                                                return J$.Rt(7140569, J$.F(7140561, J$.R(7140545, 'callback', callback, false, false), false)(J$.R(7140553, 'err', err, false, false)));
                                                            }
                                                            var ddocData = J$.W(7140601, 'ddocData', J$.M(7140593, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7140577, 'JSON', undefined, true, true) : JSON = J$.R(7140577, 'JSON', JSON, true, true)), 'parse', false)(J$.R(7140585, 'data', data, false, false)), ddocData, false, false);
                                                            if (J$.C(382344, J$.B(721362, '!==', J$.G(7140617, J$.R(7140609, 'resp', resp, false, false), 'statusCode'), J$.T(7140625, 200, 22, false)))) {
                                                                var errData = J$.W(7140641, 'errData', J$.T(7140633, null, 25, false), errData, false, false);
                                                                try {
                                                                    errData = J$.W(7140673, 'errData', J$.M(7140665, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7140649, 'JSON', undefined, true, true) : JSON = J$.R(7140649, 'JSON', JSON, true, true)), 'parse', false)(J$.R(7140657, 'data', data, false, false)), errData, false, false);
                                                                } catch (e) {
                                                                    J$.N(7140681, 'e', e, false, false, true);
                                                                }
                                                                if (J$.C(382336, J$.U(721370, '!', J$.R(7140689, 'errData', errData, false, false)))) {
                                                                    J$.F(7140761, J$.R(7140697, 'callback', callback, false, false), false)(J$.F(7140745, J$.I(typeof Error === 'undefined' ? Error = J$.R(7140705, 'Error', undefined, true, true) : Error = J$.R(7140705, 'Error', Error, true, true)), true)(J$.B(721386, '+', J$.B(721378, '+', J$.T(7140713, 'operation failed (', 21, false), J$.G(7140729, J$.R(7140721, 'resp', resp, false, false), 'statusCode')), J$.T(7140737, ')', 21, false))), J$.T(7140753, null, 25, false));
                                                                    return J$.Rt(7140769, undefined);
                                                                }
                                                                J$.F(7140825, J$.R(7140777, 'callback', callback, false, false), false)(J$.F(7140809, J$.I(typeof Error === 'undefined' ? Error = J$.R(7140785, 'Error', undefined, true, true) : Error = J$.R(7140785, 'Error', Error, true, true)), true)(J$.G(7140801, J$.R(7140793, 'errData', errData, false, false), 'reason')), J$.T(7140817, null, 25, false));
                                                                return J$.Rt(7140833, undefined);
                                                            }
                                                            J$.F(7140865, J$.R(7140841, 'callback', callback, false, false), false)(J$.T(7140849, null, 25, false), J$.R(7140857, 'ddocData', ddocData, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7147561, J$e);
                                                        } finally {
                                                            if (J$.Fr(7147569))
                                                                continue jalangiLabel17;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)));
                                            J$.M(7140961, J$.R(7140953, 'httpReq', httpReq, false, false), 'end', false)();
                                        } catch (J$e) {
                                            J$.Ex(7147577, J$e);
                                        } finally {
                                            if (J$.Fr(7147585))
                                                continue jalangiLabel18;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147593, J$e);
                        } finally {
                            if (J$.Fr(7147601))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7141729, J$.G(7141081, J$.R(7141073, 'BucketManager', BucketManager, false, true), 'prototype'), 'removeDesignDocument', J$.T(7141721, function (name, callback) {
                jalangiLabel22:
                    while (true) {
                        try {
                            J$.Fe(7141681, arguments.callee, this, arguments);
                            arguments = J$.N(7141689, 'arguments', arguments, true, false, false);
                            name = J$.N(7141697, 'name', name, true, false, false);
                            callback = J$.N(7141705, 'callback', callback, true, false, false);
                            J$.N(7141713, 'path', path, false, false, false);
                            var path = J$.W(7141105, 'path', J$.B(721394, '+', J$.T(7141089, '_design/', 21, false), J$.R(7141097, 'name', name, false, false)), path, false, false);
                            J$.M(7141673, J$.R(7141113, 'this', this, false, false), '_capiRequest', false)(J$.R(7141121, 'path', path, false, false), J$.T(7141129, 'DELETE', 21, false), J$.T(7141665, function (err, httpReq) {
                                jalangiLabel21:
                                    while (true) {
                                        try {
                                            J$.Fe(7141633, arguments.callee, this, arguments);
                                            arguments = J$.N(7141641, 'arguments', arguments, true, false, false);
                                            err = J$.N(7141649, 'err', err, true, false, false);
                                            httpReq = J$.N(7141657, 'httpReq', httpReq, true, false, false);
                                            if (J$.C(382352, J$.R(7141137, 'err', err, false, false))) {
                                                return J$.Rt(7141177, J$.F(7141169, J$.R(7141145, 'callback', callback, false, false), false)(J$.R(7141153, 'err', err, false, false), J$.T(7141161, null, 25, false)));
                                            }
                                            J$.M(7141209, J$.R(7141185, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7141193, 'error', 21, false), J$.R(7141201, 'callback', callback, false, false));
                                            J$.M(7141609, J$.R(7141217, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7141225, 'response', 21, false), J$.F(7141601, J$.R(7141233, '_respRead', _respRead, false, true), false)(J$.T(7141593, function (err, resp, data) {
                                                jalangiLabel20:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7141545, arguments.callee, this, arguments);
                                                            arguments = J$.N(7141553, 'arguments', arguments, true, false, false);
                                                            err = J$.N(7141561, 'err', err, true, false, false);
                                                            resp = J$.N(7141569, 'resp', resp, true, false, false);
                                                            data = J$.N(7141577, 'data', data, true, false, false);
                                                            J$.N(7141585, 'errData', errData, false, false, false);
                                                            if (J$.C(382360, J$.R(7141241, 'err', err, false, false))) {
                                                                return J$.Rt(7141273, J$.F(7141265, J$.R(7141249, 'callback', callback, false, false), false)(J$.R(7141257, 'err', err, false, false)));
                                                            }
                                                            if (J$.C(382376, J$.B(721402, '!==', J$.G(7141289, J$.R(7141281, 'resp', resp, false, false), 'statusCode'), J$.T(7141297, 200, 22, false)))) {
                                                                var errData = J$.W(7141313, 'errData', J$.T(7141305, null, 25, false), errData, false, false);
                                                                try {
                                                                    errData = J$.W(7141345, 'errData', J$.M(7141337, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7141321, 'JSON', undefined, true, true) : JSON = J$.R(7141321, 'JSON', JSON, true, true)), 'parse', false)(J$.R(7141329, 'data', data, false, false)), errData, false, false);
                                                                } catch (e) {
                                                                    J$.N(7141353, 'e', e, false, false, true);
                                                                }
                                                                if (J$.C(382368, J$.U(721410, '!', J$.R(7141361, 'errData', errData, false, false)))) {
                                                                    J$.F(7141433, J$.R(7141369, 'callback', callback, false, false), false)(J$.F(7141417, J$.I(typeof Error === 'undefined' ? Error = J$.R(7141377, 'Error', undefined, true, true) : Error = J$.R(7141377, 'Error', Error, true, true)), true)(J$.B(721426, '+', J$.B(721418, '+', J$.T(7141385, 'operation failed (', 21, false), J$.G(7141401, J$.R(7141393, 'resp', resp, false, false), 'statusCode')), J$.T(7141409, ')', 21, false))), J$.T(7141425, null, 25, false));
                                                                    return J$.Rt(7141441, undefined);
                                                                }
                                                                J$.F(7141497, J$.R(7141449, 'callback', callback, false, false), false)(J$.F(7141481, J$.I(typeof Error === 'undefined' ? Error = J$.R(7141457, 'Error', undefined, true, true) : Error = J$.R(7141457, 'Error', Error, true, true)), true)(J$.G(7141473, J$.R(7141465, 'errData', errData, false, false), 'reason')), J$.T(7141489, null, 25, false));
                                                                return J$.Rt(7141505, undefined);
                                                            }
                                                            J$.F(7141537, J$.R(7141513, 'callback', callback, false, false), false)(J$.T(7141521, null, 25, false), J$.T(7141529, true, 23, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7147609, J$e);
                                                        } finally {
                                                            if (J$.Fr(7147617))
                                                                continue jalangiLabel20;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)));
                                            J$.M(7141625, J$.R(7141617, 'httpReq', httpReq, false, false), 'end', false)();
                                        } catch (J$e) {
                                            J$.Ex(7147625, J$e);
                                        } finally {
                                            if (J$.Fr(7147633))
                                                continue jalangiLabel21;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147641, J$e);
                        } finally {
                            if (J$.Fr(7147649))
                                continue jalangiLabel22;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7142409, J$.G(7141745, J$.R(7141737, 'BucketManager', BucketManager, false, true), 'prototype'), 'flush', J$.T(7142401, function (callback) {
                jalangiLabel25:
                    while (true) {
                        try {
                            J$.Fe(7142369, arguments.callee, this, arguments);
                            arguments = J$.N(7142377, 'arguments', arguments, true, false, false);
                            callback = J$.N(7142385, 'callback', callback, true, false, false);
                            J$.N(7142393, 'path', path, false, false, false);
                            var path = J$.W(7141793, 'path', J$.B(721442, '+', J$.B(721434, '+', J$.T(7141753, 'pools/default/buckets/', 21, false), J$.G(7141777, J$.G(7141769, J$.R(7141761, 'this', this, false, false), '_bucket'), '_name')), J$.T(7141785, '/controller/doFlush', 21, false)), path, false, false);
                            J$.M(7142361, J$.R(7141801, 'this', this, false, false), '_mgmtRequest', false)(J$.R(7141809, 'path', path, false, false), J$.T(7141817, 'POST', 21, false), J$.T(7142353, function (err, httpReq) {
                                jalangiLabel24:
                                    while (true) {
                                        try {
                                            J$.Fe(7142321, arguments.callee, this, arguments);
                                            arguments = J$.N(7142329, 'arguments', arguments, true, false, false);
                                            err = J$.N(7142337, 'err', err, true, false, false);
                                            httpReq = J$.N(7142345, 'httpReq', httpReq, true, false, false);
                                            if (J$.C(382384, J$.R(7141825, 'err', err, false, false))) {
                                                return J$.Rt(7141865, J$.F(7141857, J$.R(7141833, 'callback', callback, false, false), false)(J$.R(7141841, 'err', err, false, false), J$.T(7141849, null, 25, false)));
                                            }
                                            J$.M(7141897, J$.R(7141873, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7141881, 'error', 21, false), J$.R(7141889, 'callback', callback, false, false));
                                            J$.M(7142297, J$.R(7141905, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7141913, 'response', 21, false), J$.F(7142289, J$.R(7141921, '_respRead', _respRead, false, true), false)(J$.T(7142281, function (err, resp, data) {
                                                jalangiLabel23:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7142233, arguments.callee, this, arguments);
                                                            arguments = J$.N(7142241, 'arguments', arguments, true, false, false);
                                                            err = J$.N(7142249, 'err', err, true, false, false);
                                                            resp = J$.N(7142257, 'resp', resp, true, false, false);
                                                            data = J$.N(7142265, 'data', data, true, false, false);
                                                            J$.N(7142273, 'errData', errData, false, false, false);
                                                            if (J$.C(382392, J$.R(7141929, 'err', err, false, false))) {
                                                                return J$.Rt(7141961, J$.F(7141953, J$.R(7141937, 'callback', callback, false, false), false)(J$.R(7141945, 'err', err, false, false)));
                                                            }
                                                            if (J$.C(382408, J$.B(721450, '!==', J$.G(7141977, J$.R(7141969, 'resp', resp, false, false), 'statusCode'), J$.T(7141985, 200, 22, false)))) {
                                                                var errData = J$.W(7142001, 'errData', J$.T(7141993, null, 25, false), errData, false, false);
                                                                try {
                                                                    errData = J$.W(7142033, 'errData', J$.M(7142025, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7142009, 'JSON', undefined, true, true) : JSON = J$.R(7142009, 'JSON', JSON, true, true)), 'parse', false)(J$.R(7142017, 'data', data, false, false)), errData, false, false);
                                                                } catch (e) {
                                                                    J$.N(7142041, 'e', e, false, false, true);
                                                                }
                                                                if (J$.C(382400, J$.U(721458, '!', J$.R(7142049, 'errData', errData, false, false)))) {
                                                                    J$.F(7142121, J$.R(7142057, 'callback', callback, false, false), false)(J$.F(7142105, J$.I(typeof Error === 'undefined' ? Error = J$.R(7142065, 'Error', undefined, true, true) : Error = J$.R(7142065, 'Error', Error, true, true)), true)(J$.B(721474, '+', J$.B(721466, '+', J$.T(7142073, 'operation failed (', 21, false), J$.G(7142089, J$.R(7142081, 'resp', resp, false, false), 'statusCode')), J$.T(7142097, ')', 21, false))), J$.T(7142113, null, 25, false));
                                                                    return J$.Rt(7142129, undefined);
                                                                }
                                                                J$.F(7142185, J$.R(7142137, 'callback', callback, false, false), false)(J$.F(7142169, J$.I(typeof Error === 'undefined' ? Error = J$.R(7142145, 'Error', undefined, true, true) : Error = J$.R(7142145, 'Error', Error, true, true)), true)(J$.G(7142161, J$.R(7142153, 'errData', errData, false, false), 'reason')), J$.T(7142177, null, 25, false));
                                                                return J$.Rt(7142193, undefined);
                                                            }
                                                            J$.F(7142225, J$.R(7142201, 'callback', callback, false, false), false)(J$.T(7142209, null, 25, false), J$.T(7142217, true, 23, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7147657, J$e);
                                                        } finally {
                                                            if (J$.Fr(7147665))
                                                                continue jalangiLabel23;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false)));
                                            J$.M(7142313, J$.R(7142305, 'httpReq', httpReq, false, false), 'end', false)();
                                        } catch (J$e) {
                                            J$.Ex(7147673, J$e);
                                        } finally {
                                            if (J$.Fr(7147681))
                                                continue jalangiLabel24;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147689, J$e);
                        } finally {
                            if (J$.Fr(7147697))
                                continue jalangiLabel25;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7143281, J$.G(7142425, J$.R(7142417, 'BucketManager', BucketManager, false, true), 'prototype'), '_createIndex', J$.T(7143273, function (options, callback) {
                jalangiLabel27:
                    while (true) {
                        try {
                            J$.Fe(7143225, arguments.callee, this, arguments);
                            arguments = J$.N(7143233, 'arguments', arguments, true, false, false);
                            options = J$.N(7143241, 'options', options, true, false, false);
                            callback = J$.N(7143249, 'callback', callback, true, false, false);
                            J$.N(7143257, 'qs', qs, false, false, false);
                            J$.N(7143265, 'i', i, false, false, false);
                            if (J$.C(382416, J$.U(721482, '!', J$.G(7142441, J$.R(7142433, 'options', options, false, false), 'fields')))) {
                                J$.P(7142465, J$.R(7142449, 'options', options, false, false), 'fields', J$.T(7142457, [], 10, false));
                            }
                            var qs = J$.W(7142481, 'qs', J$.T(7142473, '', 21, false), qs, false, false);
                            if (J$.C(382424, J$.B(721490, '===', J$.G(7142505, J$.G(7142497, J$.R(7142489, 'options', options, false, false), 'fields'), 'length'), J$.T(7142513, 0, 22, false)))) {
                                qs = J$.W(7142537, 'qs', J$.B(721498, '+', J$.R(7142529, 'qs', qs, false, false), J$.T(7142521, 'CREATE PRIMARY INDEX', 21, false)), qs, false, false);
                            } else {
                                qs = J$.W(7142561, 'qs', J$.B(721506, '+', J$.R(7142553, 'qs', qs, false, false), J$.T(7142545, 'CREATE INDEX', 21, false)), qs, false, false);
                            }
                            if (J$.C(382432, J$.B(721514, '!==', J$.G(7142577, J$.R(7142569, 'options', options, false, false), 'name'), J$.T(7142585, '', 21, false)))) {
                                qs = J$.W(7142633, 'qs', J$.B(721538, '+', J$.R(7142625, 'qs', qs, false, false), J$.B(721530, '+', J$.B(721522, '+', J$.T(7142593, ' `', 21, false), J$.G(7142609, J$.R(7142601, 'options', options, false, false), 'name')), J$.T(7142617, '`', 21, false))), qs, false, false);
                            }
                            qs = J$.W(7142689, 'qs', J$.B(721562, '+', J$.R(7142681, 'qs', qs, false, false), J$.B(721554, '+', J$.B(721546, '+', J$.T(7142641, ' ON `', 21, false), J$.G(7142665, J$.G(7142657, J$.R(7142649, 'this', this, false, false), '_bucket'), '_name')), J$.T(7142673, '`', 21, false))), qs, false, false);
                            if (J$.C(382456, J$.B(721570, '>', J$.G(7142713, J$.G(7142705, J$.R(7142697, 'options', options, false, false), 'fields'), 'length'), J$.T(7142721, 0, 22, false)))) {
                                qs = J$.W(7142745, 'qs', J$.B(721578, '+', J$.R(7142737, 'qs', qs, false, false), J$.T(7142729, '(', 21, false)), qs, false, false);
                                for (var i = J$.W(7142761, 'i', J$.T(7142753, 0, 22, false), i, false, false); J$.C(382448, J$.B(721586, '<', J$.R(7142769, 'i', i, false, false), J$.G(7142793, J$.G(7142785, J$.R(7142777, 'options', options, false, false), 'fields'), 'length'))); i = J$.W(7142809, 'i', J$.B(721602, '+', J$.U(721594, '+', J$.R(7142801, 'i', i, false, false)), 1), i, false, false)) {
                                    if (J$.C(382440, J$.B(721610, '>', J$.R(7142817, 'i', i, false, false), J$.T(7142825, 0, 22, false)))) {
                                        qs = J$.W(7142849, 'qs', J$.B(721618, '+', J$.R(7142841, 'qs', qs, false, false), J$.T(7142833, ', ', 21, false)), qs, false, false);
                                    }
                                    qs = J$.W(7142913, 'qs', J$.B(721642, '+', J$.R(7142905, 'qs', qs, false, false), J$.B(721634, '+', J$.B(721626, '+', J$.T(7142857, '`', 21, false), J$.G(7142889, J$.G(7142873, J$.R(7142865, 'options', options, false, false), 'fields'), J$.R(7142881, 'i', i, false, false))), J$.T(7142897, '`', 21, false))), qs, false, false);
                                }
                                qs = J$.W(7142937, 'qs', J$.B(721650, '+', J$.R(7142929, 'qs', qs, false, false), J$.T(7142921, ')', 21, false)), qs, false, false);
                            }
                            if (J$.C(382464, J$.G(7142953, J$.R(7142945, 'options', options, false, false), 'deferred'))) {
                                qs = J$.W(7142977, 'qs', J$.B(721658, '+', J$.R(7142969, 'qs', qs, false, false), J$.T(7142961, ' WITH {"defer_build": true}', 21, false)), qs, false, false);
                            }
                            J$.M(7143217, J$.G(7142993, J$.R(7142985, 'this', this, false, false), '_bucket'), 'query', false)(J$.M(7143017, J$.R(7143001, 'N1qlQuery', N1qlQuery, false, true), 'fromString', false)(J$.R(7143009, 'qs', qs, false, false)), J$.T(7143209, function (err, rows) {
                                jalangiLabel26:
                                    while (true) {
                                        try {
                                            J$.Fe(7143177, arguments.callee, this, arguments);
                                            arguments = J$.N(7143185, 'arguments', arguments, true, false, false);
                                            err = J$.N(7143193, 'err', err, true, false, false);
                                            rows = J$.N(7143201, 'rows', rows, true, false, false);
                                            if (J$.C(382488, J$.R(7143025, 'err', err, false, false))) {
                                                if (J$.C(382480, J$.C(382472, J$.B(721674, '!==', J$.M(7143057, J$.G(7143041, J$.R(7143033, 'err', err, false, false), 'message'), 'indexOf', false)(J$.T(7143049, 'already exist', 21, false)), J$.U(721666, '-', J$.T(7143065, 1, 22, false)))) ? J$.G(7143081, J$.R(7143073, 'options', options, false, false), 'ignoreIfExists') : J$._())) {
                                                    J$.F(7143105, J$.R(7143089, 'callback', callback, false, false), false)(J$.T(7143097, null, 25, false));
                                                    return J$.Rt(7143113, undefined);
                                                }
                                                J$.F(7143137, J$.R(7143121, 'callback', callback, false, false), false)(J$.R(7143129, 'err', err, false, false));
                                                return J$.Rt(7143145, undefined);
                                            }
                                            J$.F(7143169, J$.R(7143153, 'callback', callback, false, false), false)(J$.T(7143161, null, 25, false));
                                        } catch (J$e) {
                                            J$.Ex(7147705, J$e);
                                        } finally {
                                            if (J$.Fr(7147713))
                                                continue jalangiLabel26;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147721, J$e);
                        } finally {
                            if (J$.Fr(7147729))
                                continue jalangiLabel27;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7143537, J$.G(7143297, J$.R(7143289, 'BucketManager', BucketManager, false, true), 'prototype'), 'createIndex', J$.T(7143529, function (indexName, fields, options, callback) {
                jalangiLabel28:
                    while (true) {
                        try {
                            J$.Fe(7143481, arguments.callee, this, arguments);
                            arguments = J$.N(7143489, 'arguments', arguments, true, false, false);
                            indexName = J$.N(7143497, 'indexName', indexName, true, false, false);
                            fields = J$.N(7143505, 'fields', fields, true, false, false);
                            options = J$.N(7143513, 'options', options, true, false, false);
                            callback = J$.N(7143521, 'callback', callback, true, false, false);
                            if (J$.C(382496, J$.B(721682, 'instanceof', J$.R(7143305, 'options', options, false, false), J$.I(typeof Function === 'undefined' ? Function = J$.R(7143313, 'Function', undefined, true, true) : Function = J$.R(7143313, 'Function', Function, true, true))))) {
                                callback = J$.W(7143345, 'callback', J$.G(7143337, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7143321, 'arguments', undefined, true, true) : arguments = J$.R(7143321, 'arguments', arguments, true, true)), J$.T(7143329, 2, 22, false)), callback, false, false);
                                options = J$.W(7143361, 'options', J$.T(7143353, undefined, 24, false), options, false, false);
                            }
                            if (J$.C(382504, J$.U(721690, '!', J$.R(7143369, 'options', options, false, false)))) {
                                options = J$.W(7143385, 'options', J$.T(7143377, {}, 11, false), options, false, false);
                            }
                            return J$.Rt(7143473, J$.M(7143465, J$.R(7143393, 'this', this, false, false), '_createIndex', false)(J$.T(7143449, {
                                name: J$.R(7143401, 'indexName', indexName, false, false),
                                fields: J$.R(7143409, 'fields', fields, false, false),
                                ignoreIfExists: J$.G(7143425, J$.R(7143417, 'options', options, false, false), 'ignoreIfExists'),
                                deferred: J$.G(7143441, J$.R(7143433, 'options', options, false, false), 'deferred')
                            }, 11, false), J$.R(7143457, 'callback', callback, false, false)));
                        } catch (J$e) {
                            J$.Ex(7147737, J$e);
                        } finally {
                            if (J$.Fr(7147745))
                                continue jalangiLabel28;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7143777, J$.G(7143553, J$.R(7143545, 'BucketManager', BucketManager, false, true), 'prototype'), 'createPrimaryIndex', J$.T(7143769, function (options, callback) {
                jalangiLabel29:
                    while (true) {
                        try {
                            J$.Fe(7143737, arguments.callee, this, arguments);
                            arguments = J$.N(7143745, 'arguments', arguments, true, false, false);
                            options = J$.N(7143753, 'options', options, true, false, false);
                            callback = J$.N(7143761, 'callback', callback, true, false, false);
                            if (J$.C(382512, J$.B(721698, 'instanceof', J$.R(7143561, 'options', options, false, false), J$.I(typeof Function === 'undefined' ? Function = J$.R(7143569, 'Function', undefined, true, true) : Function = J$.R(7143569, 'Function', Function, true, true))))) {
                                callback = J$.W(7143601, 'callback', J$.G(7143593, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7143577, 'arguments', undefined, true, true) : arguments = J$.R(7143577, 'arguments', arguments, true, true)), J$.T(7143585, 0, 22, false)), callback, false, false);
                                options = J$.W(7143617, 'options', J$.T(7143609, undefined, 24, false), options, false, false);
                            }
                            if (J$.C(382520, J$.U(721706, '!', J$.R(7143625, 'options', options, false, false)))) {
                                options = J$.W(7143641, 'options', J$.T(7143633, {}, 11, false), options, false, false);
                            }
                            return J$.Rt(7143729, J$.M(7143721, J$.R(7143649, 'this', this, false, false), '_createIndex', false)(J$.T(7143705, {
                                name: J$.G(7143665, J$.R(7143657, 'options', options, false, false), 'name'),
                                ignoreIfExists: J$.G(7143681, J$.R(7143673, 'options', options, false, false), 'ignoreIfExists'),
                                deferred: J$.G(7143697, J$.R(7143689, 'options', options, false, false), 'deferred')
                            }, 11, false), J$.R(7143713, 'callback', callback, false, false)));
                        } catch (J$e) {
                            J$.Ex(7147753, J$e);
                        } finally {
                            if (J$.Fr(7147761))
                                continue jalangiLabel29;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7144249, J$.G(7143793, J$.R(7143785, 'BucketManager', BucketManager, false, true), 'prototype'), '_dropIndex', J$.T(7144241, function (options, callback) {
                jalangiLabel31:
                    while (true) {
                        try {
                            J$.Fe(7144201, arguments.callee, this, arguments);
                            arguments = J$.N(7144209, 'arguments', arguments, true, false, false);
                            options = J$.N(7144217, 'options', options, true, false, false);
                            callback = J$.N(7144225, 'callback', callback, true, false, false);
                            J$.N(7144233, 'qs', qs, false, false, false);
                            var qs = J$.W(7143809, 'qs', J$.T(7143801, '', 21, false), qs, false, false);
                            if (J$.C(382528, J$.U(721714, '!', J$.G(7143825, J$.R(7143817, 'options', options, false, false), 'name')))) {
                                qs = J$.W(7143881, 'qs', J$.B(721738, '+', J$.R(7143873, 'qs', qs, false, false), J$.B(721730, '+', J$.B(721722, '+', J$.T(7143833, 'DROP PRIMARY INDEX `', 21, false), J$.G(7143857, J$.G(7143849, J$.R(7143841, 'this', this, false, false), '_bucket'), '_name')), J$.T(7143865, '`', 21, false))), qs, false, false);
                            } else {
                                qs = J$.W(7143961, 'qs', J$.B(721778, '+', J$.R(7143953, 'qs', qs, false, false), J$.B(721770, '+', J$.B(721762, '+', J$.B(721754, '+', J$.B(721746, '+', J$.T(7143889, 'DROP INDEX `', 21, false), J$.G(7143913, J$.G(7143905, J$.R(7143897, 'this', this, false, false), '_bucket'), '_name')), J$.T(7143921, '`.`', 21, false)), J$.G(7143937, J$.R(7143929, 'options', options, false, false), 'name')), J$.T(7143945, '`', 21, false))), qs, false, false);
                            }
                            J$.M(7144193, J$.G(7143977, J$.R(7143969, 'this', this, false, false), '_bucket'), 'query', false)(J$.M(7144001, J$.R(7143985, 'N1qlQuery', N1qlQuery, false, true), 'fromString', false)(J$.R(7143993, 'qs', qs, false, false)), J$.T(7144185, function (err) {
                                jalangiLabel30:
                                    while (true) {
                                        try {
                                            J$.Fe(7144161, arguments.callee, this, arguments);
                                            arguments = J$.N(7144169, 'arguments', arguments, true, false, false);
                                            err = J$.N(7144177, 'err', err, true, false, false);
                                            if (J$.C(382552, J$.R(7144009, 'err', err, false, false))) {
                                                if (J$.C(382544, J$.C(382536, J$.B(721794, '!==', J$.M(7144041, J$.G(7144025, J$.R(7144017, 'err', err, false, false), 'message'), 'indexOf', false)(J$.T(7144033, 'not found', 21, false)), J$.U(721786, '-', J$.T(7144049, 1, 22, false)))) ? J$.G(7144065, J$.R(7144057, 'options', options, false, false), 'ignoreIfNotExists') : J$._())) {
                                                    J$.F(7144089, J$.R(7144073, 'callback', callback, false, false), false)(J$.T(7144081, null, 25, false));
                                                    return J$.Rt(7144097, undefined);
                                                }
                                                J$.F(7144121, J$.R(7144105, 'callback', callback, false, false), false)(J$.R(7144113, 'err', err, false, false));
                                                return J$.Rt(7144129, undefined);
                                            }
                                            J$.F(7144153, J$.R(7144137, 'callback', callback, false, false), false)(J$.T(7144145, null, 25, false));
                                        } catch (J$e) {
                                            J$.Ex(7147769, J$e);
                                        } finally {
                                            if (J$.Fr(7147777))
                                                continue jalangiLabel30;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147785, J$e);
                        } finally {
                            if (J$.Fr(7147793))
                                continue jalangiLabel31;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7144473, J$.G(7144265, J$.R(7144257, 'BucketManager', BucketManager, false, true), 'prototype'), 'dropIndex', J$.T(7144465, function (indexName, options, callback) {
                jalangiLabel32:
                    while (true) {
                        try {
                            J$.Fe(7144425, arguments.callee, this, arguments);
                            arguments = J$.N(7144433, 'arguments', arguments, true, false, false);
                            indexName = J$.N(7144441, 'indexName', indexName, true, false, false);
                            options = J$.N(7144449, 'options', options, true, false, false);
                            callback = J$.N(7144457, 'callback', callback, true, false, false);
                            if (J$.C(382560, J$.B(721802, 'instanceof', J$.R(7144273, 'options', options, false, false), J$.I(typeof Function === 'undefined' ? Function = J$.R(7144281, 'Function', undefined, true, true) : Function = J$.R(7144281, 'Function', Function, true, true))))) {
                                callback = J$.W(7144313, 'callback', J$.G(7144305, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7144289, 'arguments', undefined, true, true) : arguments = J$.R(7144289, 'arguments', arguments, true, true)), J$.T(7144297, 1, 22, false)), callback, false, false);
                                options = J$.W(7144329, 'options', J$.T(7144321, undefined, 24, false), options, false, false);
                            }
                            if (J$.C(382568, J$.U(721810, '!', J$.R(7144337, 'options', options, false, false)))) {
                                options = J$.W(7144353, 'options', J$.T(7144345, {}, 11, false), options, false, false);
                            }
                            return J$.Rt(7144417, J$.M(7144409, J$.R(7144361, 'this', this, false, false), '_dropIndex', false)(J$.T(7144393, {
                                name: J$.R(7144369, 'indexName', indexName, false, false),
                                ignoreIfNotExists: J$.G(7144385, J$.R(7144377, 'options', options, false, false), 'ignoreIfNotExists')
                            }, 11, false), J$.R(7144401, 'callback', callback, false, false)));
                        } catch (J$e) {
                            J$.Ex(7147801, J$e);
                        } finally {
                            if (J$.Fr(7147809))
                                continue jalangiLabel32;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7144697, J$.G(7144489, J$.R(7144481, 'BucketManager', BucketManager, false, true), 'prototype'), 'dropPrimaryIndex', J$.T(7144689, function (options, callback) {
                jalangiLabel33:
                    while (true) {
                        try {
                            J$.Fe(7144657, arguments.callee, this, arguments);
                            arguments = J$.N(7144665, 'arguments', arguments, true, false, false);
                            options = J$.N(7144673, 'options', options, true, false, false);
                            callback = J$.N(7144681, 'callback', callback, true, false, false);
                            if (J$.C(382576, J$.B(721818, 'instanceof', J$.R(7144497, 'options', options, false, false), J$.I(typeof Function === 'undefined' ? Function = J$.R(7144505, 'Function', undefined, true, true) : Function = J$.R(7144505, 'Function', Function, true, true))))) {
                                callback = J$.W(7144537, 'callback', J$.G(7144529, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7144513, 'arguments', undefined, true, true) : arguments = J$.R(7144513, 'arguments', arguments, true, true)), J$.T(7144521, 0, 22, false)), callback, false, false);
                                options = J$.W(7144553, 'options', J$.T(7144545, undefined, 24, false), options, false, false);
                            }
                            if (J$.C(382584, J$.U(721826, '!', J$.R(7144561, 'options', options, false, false)))) {
                                options = J$.W(7144577, 'options', J$.T(7144569, {}, 11, false), options, false, false);
                            }
                            return J$.Rt(7144649, J$.M(7144641, J$.R(7144585, 'this', this, false, false), '_dropIndex', false)(J$.T(7144625, {
                                name: J$.G(7144601, J$.R(7144593, 'options', options, false, false), 'name'),
                                ignoreIfNotExists: J$.G(7144617, J$.R(7144609, 'options', options, false, false), 'ignoreIfNotExists')
                            }, 11, false), J$.R(7144633, 'callback', callback, false, false)));
                        } catch (J$e) {
                            J$.Ex(7147817, J$e);
                        } finally {
                            if (J$.Fr(7147825))
                                continue jalangiLabel33;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7145025, J$.G(7144713, J$.R(7144705, 'BucketManager', BucketManager, false, true), 'prototype'), 'getIndexes', J$.T(7145017, function (callback) {
                jalangiLabel35:
                    while (true) {
                        try {
                            J$.Fe(7144985, arguments.callee, this, arguments);
                            arguments = J$.N(7144993, 'arguments', arguments, true, false, false);
                            callback = J$.N(7145001, 'callback', callback, true, false, false);
                            J$.N(7145009, 'qs', qs, false, false, false);
                            var qs = J$.W(7144729, 'qs', J$.T(7144721, 'SELECT idx.* FROM system:indexes AS idx', 21, false), qs, false, false);
                            qs = J$.W(7144785, 'qs', J$.B(721850, '+', J$.R(7144777, 'qs', qs, false, false), J$.B(721842, '+', J$.B(721834, '+', J$.T(7144737, ' WHERE keyspace_id="', 21, false), J$.G(7144761, J$.G(7144753, J$.R(7144745, 'this', this, false, false), '_bucket'), '_name')), J$.T(7144769, '"', 21, false))), qs, false, false);
                            qs = J$.W(7144809, 'qs', J$.B(721858, '+', J$.R(7144801, 'qs', qs, false, false), J$.T(7144793, ' AND `using`="gsi" ORDER BY is_primary DESC, name ASC;', 21, false)), qs, false, false);
                            J$.M(7144977, J$.G(7144825, J$.R(7144817, 'this', this, false, false), '_bucket'), 'query', false)(J$.M(7144849, J$.R(7144833, 'N1qlQuery', N1qlQuery, false, true), 'fromString', false)(J$.R(7144841, 'qs', qs, false, false)), J$.T(7144969, function (err, rows) {
                                jalangiLabel34:
                                    while (true) {
                                        try {
                                            J$.Fe(7144937, arguments.callee, this, arguments);
                                            arguments = J$.N(7144945, 'arguments', arguments, true, false, false);
                                            err = J$.N(7144953, 'err', err, true, false, false);
                                            rows = J$.N(7144961, 'rows', rows, true, false, false);
                                            if (J$.C(382592, J$.R(7144857, 'err', err, false, false))) {
                                                J$.F(7144889, J$.R(7144865, 'callback', callback, false, false), false)(J$.R(7144873, 'err', err, false, false), J$.T(7144881, null, 25, false));
                                                return J$.Rt(7144897, undefined);
                                            }
                                            J$.F(7144929, J$.R(7144905, 'callback', callback, false, false), false)(J$.T(7144913, null, 25, false), J$.R(7144921, 'rows', rows, false, false));
                                        } catch (J$e) {
                                            J$.Ex(7147833, J$e);
                                        } finally {
                                            if (J$.Fr(7147841))
                                                continue jalangiLabel34;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147849, J$e);
                        } finally {
                            if (J$.Fr(7147857))
                                continue jalangiLabel35;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7145913, J$.G(7145041, J$.R(7145033, 'BucketManager', BucketManager, false, true), 'prototype'), 'buildDeferredIndexes', J$.T(7145905, function (callback) {
                jalangiLabel38:
                    while (true) {
                        try {
                            J$.Fe(7145873, arguments.callee, this, arguments);
                            arguments = J$.N(7145881, 'arguments', arguments, true, false, false);
                            callback = J$.N(7145889, 'callback', callback, true, false, false);
                            J$.N(7145897, 'self', self, false, false, false);
                            var self = J$.W(7145057, 'self', J$.R(7145049, 'this', this, false, false), self, false, false);
                            J$.M(7145865, J$.R(7145065, 'this', this, false, false), 'getIndexes', false)(J$.T(7145857, function (err, indexes) {
                                jalangiLabel37:
                                    while (true) {
                                        try {
                                            J$.Fe(7145793, arguments.callee, this, arguments);
                                            arguments = J$.N(7145801, 'arguments', arguments, true, false, false);
                                            err = J$.N(7145809, 'err', err, true, false, false);
                                            indexes = J$.N(7145817, 'indexes', indexes, true, false, false);
                                            J$.N(7145825, 'deferredList', deferredList, false, false, false);
                                            J$.N(7145833, 'i', i, false, false, false);
                                            J$.N(7145841, 'qs', qs, false, false, false);
                                            J$.N(7145849, 'j', j, false, false, false);
                                            if (J$.C(382600, J$.R(7145073, 'err', err, false, false))) {
                                                J$.F(7145105, J$.R(7145081, 'callback', callback, false, false), false)(J$.R(7145089, 'err', err, false, false), J$.T(7145097, null, 25, false));
                                                return J$.Rt(7145113, undefined);
                                            }
                                            var deferredList = J$.W(7145129, 'deferredList', J$.T(7145121, [], 10, false), deferredList, false, false);
                                            for (var i = J$.W(7145145, 'i', J$.T(7145137, 0, 22, false), i, false, false); J$.C(382624, J$.B(721866, '<', J$.R(7145153, 'i', i, false, false), J$.G(7145169, J$.R(7145161, 'indexes', indexes, false, false), 'length'))); i = J$.W(7145185, 'i', J$.B(721882, '+', J$.U(721874, '+', J$.R(7145177, 'i', i, false, false)), 1), i, false, false)) {
                                                if (J$.C(382616, J$.C(382608, J$.B(721890, '===', J$.G(7145217, J$.G(7145209, J$.R(7145193, 'indexes', indexes, false, false), J$.R(7145201, 'i', i, false, false)), 'state'), J$.T(7145225, 'deferred', 21, false))) ? J$._() : J$.B(721898, '===', J$.G(7145257, J$.G(7145249, J$.R(7145233, 'indexes', indexes, false, false), J$.R(7145241, 'i', i, false, false)), 'state'), J$.T(7145265, 'pending', 21, false)))) {
                                                    J$.M(7145313, J$.R(7145273, 'deferredList', deferredList, false, false), 'push', false)(J$.G(7145305, J$.G(7145297, J$.R(7145281, 'indexes', indexes, false, false), J$.R(7145289, 'i', i, false, false)), 'name'));
                                                }
                                            }
                                            if (J$.C(382632, J$.B(721906, '===', J$.G(7145329, J$.R(7145321, 'deferredList', deferredList, false, false), 'length'), J$.T(7145337, 0, 22, false)))) {
                                                J$.F(7145369, J$.R(7145345, 'callback', callback, false, false), false)(J$.T(7145353, null, 25, false), J$.R(7145361, 'deferredList', deferredList, false, false));
                                                return J$.Rt(7145377, undefined);
                                            }
                                            var qs = J$.W(7145393, 'qs', J$.T(7145385, '', 21, false), qs, false, false);
                                            qs = J$.W(7145449, 'qs', J$.B(721930, '+', J$.R(7145441, 'qs', qs, false, false), J$.B(721922, '+', J$.B(721914, '+', J$.T(7145401, 'BUILD INDEX ON `', 21, false), J$.G(7145425, J$.G(7145417, J$.R(7145409, 'self', self, false, false), '_bucket'), '_name')), J$.T(7145433, '` (', 21, false))), qs, false, false);
                                            for (var j = J$.W(7145465, 'j', J$.T(7145457, 0, 22, false), j, false, false); J$.C(382648, J$.B(721938, '<', J$.R(7145473, 'j', j, false, false), J$.G(7145489, J$.R(7145481, 'deferredList', deferredList, false, false), 'length'))); j = J$.W(7145505, 'j', J$.B(721954, '+', J$.U(721946, '+', J$.R(7145497, 'j', j, false, false)), 1), j, false, false)) {
                                                if (J$.C(382640, J$.B(721962, '>', J$.R(7145513, 'j', j, false, false), J$.T(7145521, 0, 22, false)))) {
                                                    qs = J$.W(7145545, 'qs', J$.B(721970, '+', J$.R(7145537, 'qs', qs, false, false), J$.T(7145529, ', ', 21, false)), qs, false, false);
                                                }
                                                qs = J$.W(7145601, 'qs', J$.B(721994, '+', J$.R(7145593, 'qs', qs, false, false), J$.B(721986, '+', J$.B(721978, '+', J$.T(7145553, '`', 21, false), J$.G(7145577, J$.R(7145561, 'deferredList', deferredList, false, false), J$.R(7145569, 'j', j, false, false))), J$.T(7145585, '`', 21, false))), qs, false, false);
                                            }
                                            qs = J$.W(7145625, 'qs', J$.B(722002, '+', J$.R(7145617, 'qs', qs, false, false), J$.T(7145609, ')', 21, false)), qs, false, false);
                                            J$.M(7145785, J$.G(7145641, J$.R(7145633, 'self', self, false, false), '_bucket'), 'query', false)(J$.M(7145665, J$.R(7145649, 'N1qlQuery', N1qlQuery, false, true), 'fromString', false)(J$.R(7145657, 'qs', qs, false, false)), J$.T(7145777, function (err) {
                                                jalangiLabel36:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7145753, arguments.callee, this, arguments);
                                                            arguments = J$.N(7145761, 'arguments', arguments, true, false, false);
                                                            err = J$.N(7145769, 'err', err, true, false, false);
                                                            if (J$.C(382656, J$.R(7145673, 'err', err, false, false))) {
                                                                J$.F(7145705, J$.R(7145681, 'callback', callback, false, false), false)(J$.R(7145689, 'err', err, false, false), J$.R(7145697, 'deferredList', deferredList, false, false));
                                                                return J$.Rt(7145713, undefined);
                                                            }
                                                            J$.F(7145745, J$.R(7145721, 'callback', callback, false, false), false)(J$.T(7145729, null, 25, false), J$.R(7145737, 'deferredList', deferredList, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7147865, J$e);
                                                        } finally {
                                                            if (J$.Fr(7147873))
                                                                continue jalangiLabel36;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(7147881, J$e);
                                        } finally {
                                            if (J$.Fr(7147889))
                                                continue jalangiLabel37;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147897, J$e);
                        } finally {
                            if (J$.Fr(7147905))
                                continue jalangiLabel38;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7146529, J$.G(7145929, J$.R(7145921, 'BucketManager', BucketManager, false, true), 'prototype'), '_checkIndexesActive', J$.T(7146521, function (checkList, callback) {
                jalangiLabel40:
                    while (true) {
                        try {
                            J$.Fe(7146489, arguments.callee, this, arguments);
                            arguments = J$.N(7146497, 'arguments', arguments, true, false, false);
                            checkList = J$.N(7146505, 'checkList', checkList, true, false, false);
                            callback = J$.N(7146513, 'callback', callback, true, false, false);
                            J$.M(7146481, J$.R(7145937, 'this', this, false, false), 'getIndexes', false)(J$.T(7146473, function (err, indexes) {
                                jalangiLabel39:
                                    while (true) {
                                        try {
                                            J$.Fe(7146417, arguments.callee, this, arguments);
                                            arguments = J$.N(7146425, 'arguments', arguments, true, false, false);
                                            err = J$.N(7146433, 'err', err, true, false, false);
                                            indexes = J$.N(7146441, 'indexes', indexes, true, false, false);
                                            J$.N(7146449, 'checkIndexes', checkIndexes, false, false, false);
                                            J$.N(7146457, 'i', i, false, false, false);
                                            J$.N(7146465, 'j', j, false, false, false);
                                            if (J$.C(382664, J$.R(7145945, 'err', err, false, false))) {
                                                J$.F(7145977, J$.R(7145953, 'callback', callback, false, false), false)(J$.R(7145961, 'err', err, false, false), J$.T(7145969, false, 23, false));
                                                return J$.Rt(7145985, undefined);
                                            }
                                            var checkIndexes = J$.W(7146001, 'checkIndexes', J$.T(7145993, [], 10, false), checkIndexes, false, false);
                                            for (var i = J$.W(7146017, 'i', J$.T(7146009, 0, 22, false), i, false, false); J$.C(382680, J$.B(722010, '<', J$.R(7146025, 'i', i, false, false), J$.G(7146041, J$.R(7146033, 'indexes', indexes, false, false), 'length'))); J$.B(722034, '-', i = J$.W(7146057, 'i', J$.B(722026, '+', J$.U(722018, '+', J$.R(7146049, 'i', i, false, false)), 1), i, false, false), 1)) {
                                                if (J$.C(382672, J$.B(722050, '!==', J$.M(7146105, J$.R(7146065, 'checkList', checkList, false, false), 'indexOf', false)(J$.G(7146097, J$.G(7146089, J$.R(7146073, 'indexes', indexes, false, false), J$.R(7146081, 'i', i, false, false)), 'name')), J$.U(722042, '-', J$.T(7146113, 1, 22, false))))) {
                                                    J$.M(7146153, J$.R(7146121, 'checkIndexes', checkIndexes, false, false), 'push', false)(J$.G(7146145, J$.R(7146129, 'indexes', indexes, false, false), J$.R(7146137, 'i', i, false, false)));
                                                }
                                            }
                                            if (J$.C(382688, J$.B(722058, '!==', J$.G(7146169, J$.R(7146161, 'checkIndexes', checkIndexes, false, false), 'length'), J$.G(7146185, J$.R(7146177, 'checkList', checkList, false, false), 'length')))) {
                                                J$.F(7146233, J$.R(7146193, 'callback', callback, false, false), false)(J$.F(7146217, J$.I(typeof Error === 'undefined' ? Error = J$.R(7146201, 'Error', undefined, true, true) : Error = J$.R(7146201, 'Error', Error, true, true)), true)(J$.T(7146209, 'index not found', 21, false)), J$.T(7146225, false, 23, false));
                                                return J$.Rt(7146241, undefined);
                                            }
                                            for (var j = J$.W(7146257, 'j', J$.T(7146249, 0, 22, false), j, false, false); J$.C(382704, J$.B(722066, '<', J$.R(7146265, 'j', j, false, false), J$.G(7146281, J$.R(7146273, 'checkIndexes', checkIndexes, false, false), 'length'))); j = J$.W(7146297, 'j', J$.B(722082, '+', J$.U(722074, '+', J$.R(7146289, 'j', j, false, false)), 1), j, false, false)) {
                                                if (J$.C(382696, J$.B(722090, '!==', J$.G(7146329, J$.G(7146321, J$.R(7146305, 'checkIndexes', checkIndexes, false, false), J$.R(7146313, 'j', j, false, false)), 'state'), J$.T(7146337, 'online', 21, false)))) {
                                                    J$.F(7146369, J$.R(7146345, 'callback', callback, false, false), false)(J$.T(7146353, null, 25, false), J$.T(7146361, false, 23, false));
                                                    return J$.Rt(7146377, undefined);
                                                }
                                            }
                                            J$.F(7146409, J$.R(7146385, 'callback', callback, false, false), false)(J$.T(7146393, null, 25, false), J$.T(7146401, true, 23, false));
                                        } catch (J$e) {
                                            J$.Ex(7147913, J$e);
                                        } finally {
                                            if (J$.Fr(7147921))
                                                continue jalangiLabel39;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7147929, J$e);
                        } finally {
                            if (J$.Fr(7147937))
                                continue jalangiLabel40;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7147209, J$.G(7146545, J$.R(7146537, 'BucketManager', BucketManager, false, true), 'prototype'), 'watchIndexes', J$.T(7147201, function (watchList, options, callback) {
                jalangiLabel44:
                    while (true) {
                        try {
                            J$.Fe(7147137, arguments.callee, this, arguments);
                            arguments = J$.N(7147145, 'arguments', arguments, true, false, false);
                            watchList = J$.N(7147153, 'watchList', watchList, true, false, false);
                            options = J$.N(7147161, 'options', options, true, false, false);
                            callback = J$.N(7147169, 'callback', callback, true, false, false);
                            J$.N(7147177, 'timeoutTime', timeoutTime, false, false, false);
                            J$.N(7147185, 'self', self, false, false, false);
                            J$.N(7147193, 'curInterval', curInterval, false, false, false);
                            if (J$.C(382712, J$.B(722098, 'instanceof', J$.R(7146553, 'options', options, false, false), J$.I(typeof Function === 'undefined' ? Function = J$.R(7146561, 'Function', undefined, true, true) : Function = J$.R(7146561, 'Function', Function, true, true))))) {
                                callback = J$.W(7146593, 'callback', J$.G(7146585, J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7146569, 'arguments', undefined, true, true) : arguments = J$.R(7146569, 'arguments', arguments, true, true)), J$.T(7146577, 1, 22, false)), callback, false, false);
                                options = J$.W(7146609, 'options', J$.T(7146601, undefined, 24, false), options, false, false);
                            }
                            if (J$.C(382720, J$.U(722106, '!', J$.R(7146617, 'options', options, false, false)))) {
                                options = J$.W(7146633, 'options', J$.T(7146625, {}, 11, false), options, false, false);
                            }
                            var timeoutTime = J$.W(7146649, 'timeoutTime', J$.T(7146641, 0, 22, false), timeoutTime, false, false);
                            if (J$.C(382728, J$.G(7146665, J$.R(7146657, 'options', options, false, false), 'timeout'))) {
                                timeoutTime = J$.W(7146705, 'timeoutTime', J$.B(722114, '+', J$.M(7146681, J$.I(typeof Date === 'undefined' ? Date = J$.R(7146673, 'Date', undefined, true, true) : Date = J$.R(7146673, 'Date', Date, true, true)), 'now', false)(), J$.G(7146697, J$.R(7146689, 'options', options, false, false), 'timeout')), timeoutTime, false, false);
                            }
                            var self = J$.W(7146721, 'self', J$.R(7146713, 'this', this, false, false), self, false, false);
                            var curInterval = J$.W(7146737, 'curInterval', J$.T(7146729, 50, 22, false), curInterval, false, false);
                            J$.F(7147129, J$.T(7147121, function _checkIndexes() {
                                jalangiLabel43:
                                    while (true) {
                                        try {
                                            J$.Fe(7147105, arguments.callee, this, arguments);
                                            arguments = J$.N(7147113, 'arguments', arguments, true, false, false);
                                            J$.M(7147097, J$.R(7146745, 'self', self, false, false), '_checkIndexesActive', false)(J$.R(7146753, 'watchList', watchList, false, false), J$.T(7147089, function (err, allActive) {
                                                jalangiLabel42:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7147057, arguments.callee, this, arguments);
                                                            arguments = J$.N(7147065, 'arguments', arguments, true, false, false);
                                                            err = J$.N(7147073, 'err', err, true, false, false);
                                                            allActive = J$.N(7147081, 'allActive', allActive, true, false, false);
                                                            if (J$.C(382736, J$.R(7146761, 'err', err, false, false))) {
                                                                J$.F(7146785, J$.R(7146769, 'callback', callback, false, false), false)(J$.R(7146777, 'err', err, false, false));
                                                                return J$.Rt(7146793, undefined);
                                                            }
                                                            if (J$.C(382744, J$.R(7146801, 'allActive', allActive, false, false))) {
                                                                J$.F(7146825, J$.R(7146809, 'callback', callback, false, false), false)(J$.T(7146817, null, 25, false));
                                                                return J$.Rt(7146833, undefined);
                                                            }
                                                            if (J$.C(382760, J$.C(382752, J$.B(722122, '>', J$.R(7146841, 'timeoutTime', timeoutTime, false, false), J$.T(7146849, 0, 22, false))) ? J$.B(722138, '>', J$.B(722130, '+', J$.M(7146865, J$.I(typeof Date === 'undefined' ? Date = J$.R(7146857, 'Date', undefined, true, true) : Date = J$.R(7146857, 'Date', Date, true, true)), 'now', false)(), J$.R(7146873, 'curInterval', curInterval, false, false)), J$.R(7146881, 'timeoutTime', timeoutTime, false, false)) : J$._())) {
                                                                J$.F(7146921, J$.R(7146889, 'callback', callback, false, false), false)(J$.F(7146913, J$.I(typeof Error === 'undefined' ? Error = J$.R(7146897, 'Error', undefined, true, true) : Error = J$.R(7146897, 'Error', Error, true, true)), true)(J$.T(7146905, 'timeout', 21, false)));
                                                                return J$.Rt(7146929, undefined);
                                                            }
                                                            J$.F(7147049, J$.I(typeof setTimeout === 'undefined' ? setTimeout = J$.R(7146937, 'setTimeout', undefined, true, true) : setTimeout = J$.R(7146937, 'setTimeout', setTimeout, true, true)), false)(J$.T(7147033, function () {
                                                                jalangiLabel41:
                                                                    while (true) {
                                                                        try {
                                                                            J$.Fe(7147017, arguments.callee, this, arguments);
                                                                            arguments = J$.N(7147025, 'arguments', arguments, true, false, false);
                                                                            curInterval = J$.W(7146961, 'curInterval', J$.B(722146, '+', J$.R(7146953, 'curInterval', curInterval, false, false), J$.T(7146945, 500, 22, false)), curInterval, false, false);
                                                                            if (J$.C(382768, J$.B(722154, '>', J$.R(7146969, 'curInterval', curInterval, false, false), J$.T(7146977, 1000, 22, false)))) {
                                                                                curInterval = J$.W(7146993, 'curInterval', J$.T(7146985, 1000, 22, false), curInterval, false, false);
                                                                            }
                                                                            J$.F(7147009, J$.R(7147001, '_checkIndexes', _checkIndexes, false, false), false)();
                                                                        } catch (J$e) {
                                                                            J$.Ex(7147945, J$e);
                                                                        } finally {
                                                                            if (J$.Fr(7147953))
                                                                                continue jalangiLabel41;
                                                                            else
                                                                                return J$.Ra();
                                                                        }
                                                                    }
                                                            }, 12, false), J$.R(7147041, 'curInterval', curInterval, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7147961, J$e);
                                                        } finally {
                                                            if (J$.Fr(7147969))
                                                                continue jalangiLabel42;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(7147977, J$e);
                                        } finally {
                                            if (J$.Fr(7147985))
                                                continue jalangiLabel43;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), false)();
                        } catch (J$e) {
                            J$.Ex(7147993, J$e);
                        } finally {
                            if (J$.Fr(7148001))
                                continue jalangiLabel44;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7147233, J$.I(typeof module === 'undefined' ? module = J$.R(7147217, 'module', undefined, true, true) : module = J$.R(7147217, 'module', module, true, true)), 'exports', J$.R(7147225, 'BucketManager', BucketManager, false, true));
        } catch (J$e) {
            J$.Ex(7148009, J$e);
        } finally {
            if (J$.Sr(7148017))
                continue jalangiLabel45;
            else
                break jalangiLabel45;
        }
    }
// JALANGI DO NOT INSTRUMENT
