// couchbase/lib/clustermgr.js
J$.noInstrEval = false;
jalangiLabel25:
    while (true) {
        try {
            J$.Se(7162665, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/couchbase/lib/clustermgr.js');
            function _respRead(callback) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7157161, arguments.callee, this, arguments);
                            arguments = J$.N(7157169, 'arguments', arguments, true, false, false);
                            callback = J$.N(7157177, 'callback', callback, true, false, false);
                            return J$.Rt(7157153, J$.T(7157145, function (resp) {
                                jalangiLabel3:
                                    while (true) {
                                        try {
                                            J$.Fe(7157113, arguments.callee, this, arguments);
                                            arguments = J$.N(7157121, 'arguments', arguments, true, false, false);
                                            resp = J$.N(7157129, 'resp', resp, true, false, false);
                                            J$.N(7157137, 'strBuffer', strBuffer, false, false, false);
                                            J$.M(7156825, J$.R(7156809, 'resp', resp, false, false), 'setEncoding', false)(J$.T(7156817, 'utf8', 21, false));
                                            var strBuffer = J$.W(7156841, 'strBuffer', J$.T(7156833, '', 21, false), strBuffer, false, false);
                                            J$.M(7156921, J$.R(7156849, 'resp', resp, false, false), 'on', false)(J$.T(7156857, 'data', 21, false), J$.T(7156913, function (data) {
                                                jalangiLabel0:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7156889, arguments.callee, this, arguments);
                                                            arguments = J$.N(7156897, 'arguments', arguments, true, false, false);
                                                            data = J$.N(7156905, 'data', data, true, false, false);
                                                            strBuffer = J$.W(7156881, 'strBuffer', J$.B(722722, '+', J$.R(7156873, 'strBuffer', strBuffer, false, false), J$.R(7156865, 'data', data, false, false)), strBuffer, false, false);
                                                        } catch (J$e) {
                                                            J$.Ex(7162745, J$e);
                                                        } finally {
                                                            if (J$.Fr(7162753))
                                                                continue jalangiLabel0;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                            J$.M(7157009, J$.R(7156929, 'resp', resp, false, false), 'on', false)(J$.T(7156937, 'end', 21, false), J$.T(7157001, function () {
                                                jalangiLabel1:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7156985, arguments.callee, this, arguments);
                                                            arguments = J$.N(7156993, 'arguments', arguments, true, false, false);
                                                            J$.F(7156977, J$.R(7156945, 'callback', callback, false, false), false)(J$.T(7156953, null, 25, false), J$.R(7156961, 'resp', resp, false, false), J$.R(7156969, 'strBuffer', strBuffer, false, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7162761, J$e);
                                                        } finally {
                                                            if (J$.Fr(7162769))
                                                                continue jalangiLabel1;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                            J$.M(7157105, J$.R(7157017, 'resp', resp, false, false), 'on', false)(J$.T(7157025, 'error', 21, false), J$.T(7157097, function (err) {
                                                jalangiLabel2:
                                                    while (true) {
                                                        try {
                                                            J$.Fe(7157073, arguments.callee, this, arguments);
                                                            arguments = J$.N(7157081, 'arguments', arguments, true, false, false);
                                                            err = J$.N(7157089, 'err', err, true, false, false);
                                                            J$.F(7157065, J$.R(7157033, 'callback', callback, false, false), false)(J$.R(7157041, 'err', err, false, false), J$.R(7157049, 'resp', resp, false, false), J$.T(7157057, null, 25, false));
                                                        } catch (J$e) {
                                                            J$.Ex(7162777, J$e);
                                                        } finally {
                                                            if (J$.Fr(7162785))
                                                                continue jalangiLabel2;
                                                            else
                                                                return J$.Ra();
                                                        }
                                                    }
                                            }, 12, false));
                                        } catch (J$e) {
                                            J$.Ex(7162793, J$e);
                                        } finally {
                                            if (J$.Fr(7162801))
                                                continue jalangiLabel3;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7162809, J$e);
                        } finally {
                            if (J$.Fr(7162817))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _respParseError(resp, data) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(7158017, arguments.callee, this, arguments);
                            arguments = J$.N(7158025, 'arguments', arguments, true, false, false);
                            resp = J$.N(7158033, 'resp', resp, true, false, false);
                            data = J$.N(7158041, 'data', data, true, false, false);
                            J$.N(7158049, 'errData', errData, false, false, false);
                            J$.N(7158057, 'dataStr', dataStr, false, false, false);
                            J$.N(7158065, 'strDataStr', strDataStr, false, false, false);
                            J$.N(7158073, 'errMessage', errMessage, false, false, false);
                            J$.N(7158081, 'errObj', errObj, false, false, false);
                            if (J$.C(383288, J$.C(383216, J$.B(722730, '<', J$.G(7157193, J$.R(7157185, 'resp', resp, false, false), 'statusCode'), J$.T(7157201, 200, 22, false))) ? J$._() : J$.B(722738, '>=', J$.G(7157217, J$.R(7157209, 'resp', resp, false, false), 'statusCode'), J$.T(7157225, 300, 22, false)))) {
                                var errData = J$.W(7157241, 'errData', J$.T(7157233, null, 25, false), errData, false, false);
                                try {
                                    errData = J$.W(7157273, 'errData', J$.M(7157265, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7157249, 'JSON', undefined, true, true) : JSON = J$.R(7157249, 'JSON', JSON, true, true)), 'parse', false)(J$.R(7157257, 'data', data, false, false)), errData, false, false);
                                } catch (e) {
                                    J$.N(7157281, 'e', e, false, false, true);
                                }
                                if (J$.C(383232, J$.U(722746, '!', J$.R(7157289, 'errData', errData, false, false)))) {
                                    var dataStr = J$.W(7157305, 'dataStr', J$.T(7157297, '', 21, false), dataStr, false, false);
                                    if (J$.C(383224, J$.R(7157313, 'data', data, false, false))) {
                                        dataStr = J$.W(7157337, 'dataStr', J$.B(722754, '+', J$.T(7157321, ', ', 21, false), J$.R(7157329, 'data', data, false, false)), dataStr, false, false);
                                    }
                                    return J$.Rt(7157401, J$.F(7157393, J$.I(typeof Error === 'undefined' ? Error = J$.R(7157345, 'Error', undefined, true, true) : Error = J$.R(7157345, 'Error', Error, true, true)), true)(J$.B(722778, '+', J$.B(722770, '+', J$.B(722762, '+', J$.T(7157353, 'operation failed (', 21, false), J$.G(7157369, J$.R(7157361, 'resp', resp, false, false), 'statusCode')), J$.R(7157377, 'dataStr', dataStr, false, false)), J$.T(7157385, ')', 21, false))));
                                }
                                if (J$.C(383248, J$.B(722794, '===', J$.U(722786, 'typeof', J$.R(7157409, 'errData', errData, false, false)), J$.T(7157417, 'string', 21, false)))) {
                                    var strDataStr = J$.W(7157433, 'strDataStr', J$.T(7157425, '', 21, false), strDataStr, false, false);
                                    if (J$.C(383240, J$.R(7157441, 'data', data, false, false))) {
                                        strDataStr = J$.W(7157465, 'strDataStr', J$.B(722802, '+', J$.T(7157449, ', ', 21, false), J$.R(7157457, 'errData', errData, false, false)), strDataStr, false, false);
                                    }
                                    return J$.Rt(7157529, J$.F(7157521, J$.I(typeof Error === 'undefined' ? Error = J$.R(7157473, 'Error', undefined, true, true) : Error = J$.R(7157473, 'Error', Error, true, true)), true)(J$.B(722826, '+', J$.B(722818, '+', J$.B(722810, '+', J$.T(7157481, 'operation failed (', 21, false), J$.G(7157497, J$.R(7157489, 'resp', resp, false, false), 'statusCode')), J$.R(7157505, 'strDataStr', strDataStr, false, false)), J$.T(7157513, ')', 21, false))));
                                }
                                var errMessage = J$.W(7157545, 'errMessage', J$.T(7157537, null, 25, false), errMessage, false, false);
                                if (J$.C(383280, J$.G(7157561, J$.R(7157553, 'errData', errData, false, false), '_'))) {
                                    errMessage = J$.W(7157585, 'errMessage', J$.G(7157577, J$.R(7157569, 'errData', errData, false, false), '_'), errMessage, false, false);
                                } else if (J$.C(383272, J$.G(7157601, J$.R(7157593, 'errData', errData, false, false), 'errors'))) {
                                    errMessage = J$.W(7157737, 'errMessage', J$.M(7157729, J$.M(7157713, J$.M(7157633, J$.I(typeof Object === 'undefined' ? Object = J$.R(7157609, 'Object', undefined, true, true) : Object = J$.R(7157609, 'Object', Object, true, true)), 'keys', false)(J$.G(7157625, J$.R(7157617, 'errData', errData, false, false), 'errors')), 'map', false)(J$.T(7157705, function (errKey) {
                                        jalangiLabel5:
                                            while (true) {
                                                try {
                                                    J$.Fe(7157681, arguments.callee, this, arguments);
                                                    arguments = J$.N(7157689, 'arguments', arguments, true, false, false);
                                                    errKey = J$.N(7157697, 'errKey', errKey, true, false, false);
                                                    return J$.Rt(7157673, J$.G(7157665, J$.G(7157649, J$.R(7157641, 'errData', errData, false, false), 'errors'), J$.R(7157657, 'errKey', errKey, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(7162825, J$e);
                                                } finally {
                                                    if (J$.Fr(7162833))
                                                        continue jalangiLabel5;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)), 'join', false)(J$.T(7157721, ' ', 21, false)), errMessage, false, false);
                                } else if (J$.C(383264, J$.C(383256, J$.B(722834, '!==', J$.R(7157745, 'errData', errData, false, false), J$.T(7157753, null, 25, false))) ? J$.B(722850, '===', J$.U(722842, 'typeof', J$.R(7157761, 'errData', errData, false, false)), J$.T(7157769, 'object', 21, false)) : J$._())) {
                                    errMessage = J$.W(7157889, 'errMessage', J$.M(7157881, J$.M(7157865, J$.M(7157793, J$.I(typeof Object === 'undefined' ? Object = J$.R(7157777, 'Object', undefined, true, true) : Object = J$.R(7157777, 'Object', Object, true, true)), 'keys', false)(J$.R(7157785, 'errData', errData, false, false)), 'map', false)(J$.T(7157857, function (errKey) {
                                        jalangiLabel6:
                                            while (true) {
                                                try {
                                                    J$.Fe(7157833, arguments.callee, this, arguments);
                                                    arguments = J$.N(7157841, 'arguments', arguments, true, false, false);
                                                    errKey = J$.N(7157849, 'errKey', errKey, true, false, false);
                                                    return J$.Rt(7157825, J$.G(7157817, J$.R(7157801, 'errData', errData, false, false), J$.R(7157809, 'errKey', errKey, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(7162841, J$e);
                                                } finally {
                                                    if (J$.Fr(7162849))
                                                        continue jalangiLabel6;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)), 'join', false)(J$.T(7157873, ' ', 21, false)), errMessage, false, false);
                                }
                                var errObj = J$.W(7157921, 'errObj', J$.F(7157913, J$.I(typeof Error === 'undefined' ? Error = J$.R(7157897, 'Error', undefined, true, true) : Error = J$.R(7157897, 'Error', Error, true, true)), true)(J$.R(7157905, 'errMessage', errMessage, false, false)), errObj, false, false);
                                J$.P(7157945, J$.R(7157929, 'errObj', errObj, false, false), 'response', J$.R(7157937, 'errData', errData, false, false));
                                J$.P(7157977, J$.R(7157953, 'errObj', errObj, false, false), 'statusCode', J$.G(7157969, J$.R(7157961, 'resp', resp, false, false), 'statusCode'));
                                return J$.Rt(7157993, J$.R(7157985, 'errObj', errObj, false, false));
                            }
                            return J$.Rt(7158009, J$.T(7158001, null, 25, false));
                        } catch (J$e) {
                            J$.Ex(7162857, J$e);
                        } finally {
                            if (J$.Fr(7162865))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function ClusterManager(cluster, username, password) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(7158161, arguments.callee, this, arguments);
                            arguments = J$.N(7158169, 'arguments', arguments, true, false, false);
                            cluster = J$.N(7158177, 'cluster', cluster, true, false, false);
                            username = J$.N(7158185, 'username', username, true, false, false);
                            password = J$.N(7158193, 'password', password, true, false, false);
                            J$.P(7158105, J$.R(7158089, 'this', this, false, false), '_cluster', J$.R(7158097, 'cluster', cluster, false, false));
                            J$.P(7158129, J$.R(7158113, 'this', this, false, false), '_username', J$.R(7158121, 'username', username, false, false));
                            J$.P(7158153, J$.R(7158137, 'this', this, false, false), '_password', J$.R(7158145, 'password', password, false, false));
                        } catch (J$e) {
                            J$.Ex(7162873, J$e);
                        } finally {
                            if (J$.Fr(7162881))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(7162673, 'http', http, false, false, false);
            J$.N(7162681, 'qs', qs, false, false, false);
            J$.N(7162689, 'SearchIndexManager', SearchIndexManager, false, false, false);
            _respRead = J$.N(7162705, '_respRead', J$.T(7162697, _respRead, 12, false), true, false, false);
            _respParseError = J$.N(7162721, '_respParseError', J$.T(7162713, _respParseError, 12, false), true, false, false);
            ClusterManager = J$.N(7162737, 'ClusterManager', J$.T(7162729, ClusterManager, 12, false), true, false, false);
            J$.T(7156705, 'use strict', 21, false);
            var http = J$.W(7156737, 'http', J$.F(7156729, J$.I(typeof require === 'undefined' ? require = J$.R(7156713, 'require', undefined, true, true) : require = J$.R(7156713, 'require', require, true, true)), false)(J$.T(7156721, 'http', 21, false)), http, false, true);
            var qs = J$.W(7156769, 'qs', J$.F(7156761, J$.I(typeof require === 'undefined' ? require = J$.R(7156745, 'require', undefined, true, true) : require = J$.R(7156745, 'require', require, true, true)), false)(J$.T(7156753, 'querystring', 21, false)), qs, false, true);
            var SearchIndexManager = J$.W(7156801, 'SearchIndexManager', J$.F(7156793, J$.I(typeof require === 'undefined' ? require = J$.R(7156777, 'require', undefined, true, true) : require = J$.R(7156777, 'require', require, true, true)), false)(J$.T(7156785, './searchindexmgr', 21, false)), SearchIndexManager, false, true);
            J$.P(7158657, J$.G(7158209, J$.R(7158201, 'ClusterManager', ClusterManager, false, true), 'prototype'), '_mgmtRequest', J$.T(7158649, function (path, method, uses_qs) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(7158585, arguments.callee, this, arguments);
                            arguments = J$.N(7158593, 'arguments', arguments, true, false, false);
                            path = J$.N(7158601, 'path', path, true, false, false);
                            method = J$.N(7158609, 'method', method, true, false, false);
                            uses_qs = J$.N(7158617, 'uses_qs', uses_qs, true, false, false);
                            J$.N(7158625, 'clusterHosts', clusterHosts, false, false, false);
                            J$.N(7158633, 'myHost', myHost, false, false, false);
                            J$.N(7158641, 'reqOpts', reqOpts, false, false, false);
                            var clusterHosts = J$.W(7158249, 'clusterHosts', J$.G(7158241, J$.G(7158233, J$.G(7158225, J$.R(7158217, 'this', this, false, false), '_cluster'), 'dsnObj'), 'hosts'), clusterHosts, false, false);
                            var myHost = J$.W(7158321, 'myHost', J$.G(7158313, J$.R(7158257, 'clusterHosts', clusterHosts, false, false), J$.M(7158305, J$.I(typeof Math === 'undefined' ? Math = J$.R(7158265, 'Math', undefined, true, true) : Math = J$.R(7158265, 'Math', Math, true, true)), 'floor', false)(J$.B(722858, '*', J$.M(7158281, J$.I(typeof Math === 'undefined' ? Math = J$.R(7158273, 'Math', undefined, true, true) : Math = J$.R(7158273, 'Math', Math, true, true)), 'random', false)(), J$.G(7158297, J$.R(7158289, 'clusterHosts', clusterHosts, false, false), 'length')))), myHost, false, false);
                            var reqOpts = J$.W(7158473, 'reqOpts', J$.T(7158465, {
                                hostname: J$.G(7158345, J$.R(7158329, 'myHost', myHost, false, false), J$.T(7158337, 0, 22, false)),
                                port: J$.C(383296, J$.G(7158369, J$.R(7158353, 'myHost', myHost, false, false), J$.T(7158361, 1, 22, false))) ? J$.G(7158393, J$.R(7158377, 'myHost', myHost, false, false), J$.T(7158385, 1, 22, false)) : J$.T(7158401, 8091, 22, false),
                                path: J$.B(722866, '+', J$.T(7158409, '/', 21, false), J$.R(7158417, 'path', path, false, false)),
                                method: J$.R(7158425, 'method', method, false, false),
                                headers: J$.T(7158457, { 'Content-Type': J$.C(383304, J$.R(7158433, 'uses_qs', uses_qs, false, false)) ? J$.T(7158441, 'application/x-www-form-urlencoded', 21, false) : J$.T(7158449, 'application/json', 21, false) }, 11, false)
                            }, 11, false), reqOpts, false, false);
                            if (J$.C(383312, J$.G(7158489, J$.R(7158481, 'this', this, false, false), '_password'))) {
                                J$.P(7158545, J$.R(7158497, 'reqOpts', reqOpts, false, false), 'auth', J$.B(722882, '+', J$.B(722874, '+', J$.G(7158513, J$.R(7158505, 'this', this, false, false), '_username'), J$.T(7158521, ':', 21, false)), J$.G(7158537, J$.R(7158529, 'this', this, false, false), '_password')));
                            }
                            return J$.Rt(7158577, J$.M(7158569, J$.R(7158553, 'http', http, false, true), 'request', false)(J$.R(7158561, 'reqOpts', reqOpts, false, false)));
                        } catch (J$e) {
                            J$.Ex(7162889, J$e);
                        } finally {
                            if (J$.Fr(7162897))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7159177, J$.G(7158673, J$.R(7158665, 'ClusterManager', ClusterManager, false, true), 'prototype'), 'listBuckets', J$.T(7159169, function (callback) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(7159129, arguments.callee, this, arguments);
                            arguments = J$.N(7159137, 'arguments', arguments, true, false, false);
                            callback = J$.N(7159145, 'callback', callback, true, false, false);
                            J$.N(7159153, 'path', path, false, false, false);
                            J$.N(7159161, 'httpReq', httpReq, false, false, false);
                            var path = J$.W(7158689, 'path', J$.T(7158681, 'pools/default/buckets', 21, false), path, false, false);
                            var httpReq = J$.W(7158729, 'httpReq', J$.M(7158721, J$.R(7158697, 'this', this, false, false), '_mgmtRequest', false)(J$.R(7158705, 'path', path, false, false), J$.T(7158713, 'GET', 21, false)), httpReq, false, false);
                            J$.M(7158761, J$.R(7158737, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7158745, 'error', 21, false), J$.R(7158753, 'callback', callback, false, false));
                            J$.M(7159105, J$.R(7158769, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7158777, 'response', 21, false), J$.F(7159097, J$.R(7158785, '_respRead', _respRead, false, true), false)(J$.T(7159089, function (err, resp, data) {
                                jalangiLabel10:
                                    while (true) {
                                        try {
                                            J$.Fe(7159041, arguments.callee, this, arguments);
                                            arguments = J$.N(7159049, 'arguments', arguments, true, false, false);
                                            err = J$.N(7159057, 'err', err, true, false, false);
                                            resp = J$.N(7159065, 'resp', resp, true, false, false);
                                            data = J$.N(7159073, 'data', data, true, false, false);
                                            J$.N(7159081, 'bucketInfo', bucketInfo, false, false, false);
                                            if (J$.C(383320, J$.R(7158793, 'err', err, false, false))) {
                                                J$.F(7158817, J$.R(7158801, 'callback', callback, false, false), false)(J$.R(7158809, 'err', err, false, false));
                                                return J$.Rt(7158825, undefined);
                                            }
                                            err = J$.W(7158865, 'err', J$.F(7158857, J$.R(7158833, '_respParseError', _respParseError, false, true), false)(J$.R(7158841, 'resp', resp, false, false), J$.R(7158849, 'data', data, false, false)), err, false, false);
                                            if (J$.C(383328, J$.R(7158873, 'err', err, false, false))) {
                                                J$.F(7158897, J$.R(7158881, 'callback', callback, false, false), false)(J$.R(7158889, 'err', err, false, false));
                                                return J$.Rt(7158905, undefined);
                                            }
                                            var bucketInfo;
                                            try {
                                                bucketInfo = J$.W(7158937, 'bucketInfo', J$.M(7158929, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7158913, 'JSON', undefined, true, true) : JSON = J$.R(7158913, 'JSON', JSON, true, true)), 'parse', false)(J$.R(7158921, 'data', data, false, false)), bucketInfo, false, false);
                                            } catch (e) {
                                                J$.N(7158961, 'e', e, false, false, true);
                                                err = J$.W(7158953, 'err', J$.R(7158945, 'e', e, false, false), err, false, false);
                                            }
                                            if (J$.C(383336, J$.R(7158969, 'err', err, false, false))) {
                                                J$.F(7158993, J$.R(7158977, 'callback', callback, false, false), false)(J$.R(7158985, 'err', err, false, false));
                                                return J$.Rt(7159001, undefined);
                                            }
                                            J$.F(7159033, J$.R(7159009, 'callback', callback, false, false), false)(J$.T(7159017, null, 25, false), J$.R(7159025, 'bucketInfo', bucketInfo, false, false));
                                        } catch (J$e) {
                                            J$.Ex(7162905, J$e);
                                        } finally {
                                            if (J$.Fr(7162913))
                                                continue jalangiLabel10;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                            J$.M(7159121, J$.R(7159113, 'httpReq', httpReq, false, false), 'end', false)();
                        } catch (J$e) {
                            J$.Ex(7162921, J$e);
                        } finally {
                            if (J$.Fr(7162929))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7159825, J$.G(7159193, J$.R(7159185, 'ClusterManager', ClusterManager, false, true), 'prototype'), 'createBucket', J$.T(7159817, function (name, opts, callback) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(7159745, arguments.callee, this, arguments);
                            arguments = J$.N(7159753, 'arguments', arguments, true, false, false);
                            name = J$.N(7159761, 'name', name, true, false, false);
                            opts = J$.N(7159769, 'opts', opts, true, false, false);
                            callback = J$.N(7159777, 'callback', callback, true, false, false);
                            J$.N(7159785, 'myOpts', myOpts, false, false, false);
                            J$.N(7159793, 'i', i, false, false, false);
                            J$.N(7159801, 'path', path, false, false, false);
                            J$.N(7159809, 'httpReq', httpReq, false, false, false);
                            var myOpts = J$.W(7159249, 'myOpts', J$.T(7159241, {
                                name: J$.R(7159201, 'name', name, false, false),
                                authType: J$.T(7159209, 'sasl', 21, false),
                                bucketType: J$.T(7159217, 'couchbase', 21, false),
                                ramQuotaMB: J$.T(7159225, 100, 22, false),
                                replicaNumber: J$.T(7159233, 1, 22, false)
                            }, 11, false), myOpts, false, false);
                            for (var i in J$.H(7159337, J$.R(7159257, 'opts', opts, false, false))) {
                                J$.N(7159345, 'i', i, false, true, false);
                                {
                                    {
                                        if (J$.C(383344, J$.M(7159281, J$.R(7159265, 'opts', opts, false, false), 'hasOwnProperty', false)(J$.R(7159273, 'i', i, false, false)))) {
                                            J$.P(7159329, J$.R(7159289, 'myOpts', myOpts, false, false), J$.R(7159297, 'i', i, false, false), J$.G(7159321, J$.R(7159305, 'opts', opts, false, false), J$.R(7159313, 'i', i, false, false)));
                                        }
                                    }
                                }
                            }
                            var path = J$.W(7159361, 'path', J$.T(7159353, 'pools/default/buckets', 21, false), path, false, false);
                            var httpReq = J$.W(7159409, 'httpReq', J$.M(7159401, J$.R(7159369, 'this', this, false, false), '_mgmtRequest', false)(J$.R(7159377, 'path', path, false, false), J$.T(7159385, 'POST', 21, false), J$.T(7159393, true, 23, false)), httpReq, false, false);
                            J$.M(7159441, J$.R(7159417, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7159425, 'error', 21, false), J$.R(7159433, 'callback', callback, false, false));
                            J$.M(7159681, J$.R(7159449, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7159457, 'response', 21, false), J$.F(7159673, J$.R(7159465, '_respRead', _respRead, false, true), false)(J$.T(7159665, function (err, resp, data) {
                                jalangiLabel12:
                                    while (true) {
                                        try {
                                            J$.Fe(7159625, arguments.callee, this, arguments);
                                            arguments = J$.N(7159633, 'arguments', arguments, true, false, false);
                                            err = J$.N(7159641, 'err', err, true, false, false);
                                            resp = J$.N(7159649, 'resp', resp, true, false, false);
                                            data = J$.N(7159657, 'data', data, true, false, false);
                                            if (J$.C(383352, J$.R(7159473, 'err', err, false, false))) {
                                                J$.F(7159497, J$.R(7159481, 'callback', callback, false, false), false)(J$.R(7159489, 'err', err, false, false));
                                                return J$.Rt(7159505, undefined);
                                            }
                                            err = J$.W(7159545, 'err', J$.F(7159537, J$.R(7159513, '_respParseError', _respParseError, false, true), false)(J$.R(7159521, 'resp', resp, false, false), J$.R(7159529, 'data', data, false, false)), err, false, false);
                                            if (J$.C(383360, J$.R(7159553, 'err', err, false, false))) {
                                                J$.F(7159577, J$.R(7159561, 'callback', callback, false, false), false)(J$.R(7159569, 'err', err, false, false));
                                                return J$.Rt(7159585, undefined);
                                            }
                                            J$.F(7159617, J$.R(7159593, 'callback', callback, false, false), false)(J$.T(7159601, null, 25, false), J$.T(7159609, true, 23, false));
                                        } catch (J$e) {
                                            J$.Ex(7162937, J$e);
                                        } finally {
                                            if (J$.Fr(7162945))
                                                continue jalangiLabel12;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                            J$.M(7159721, J$.R(7159689, 'httpReq', httpReq, false, false), 'write', false)(J$.M(7159713, J$.R(7159697, 'qs', qs, false, true), 'stringify', false)(J$.R(7159705, 'myOpts', myOpts, false, false)));
                            J$.M(7159737, J$.R(7159729, 'httpReq', httpReq, false, false), 'end', false)();
                        } catch (J$e) {
                            J$.Ex(7162953, J$e);
                        } finally {
                            if (J$.Fr(7162961))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7160257, J$.G(7159841, J$.R(7159833, 'ClusterManager', ClusterManager, false, true), 'prototype'), 'removeBucket', J$.T(7160249, function (name, callback) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(7160201, arguments.callee, this, arguments);
                            arguments = J$.N(7160209, 'arguments', arguments, true, false, false);
                            name = J$.N(7160217, 'name', name, true, false, false);
                            callback = J$.N(7160225, 'callback', callback, true, false, false);
                            J$.N(7160233, 'path', path, false, false, false);
                            J$.N(7160241, 'httpReq', httpReq, false, false, false);
                            var path = J$.W(7159865, 'path', J$.B(722890, '+', J$.T(7159849, 'pools/default/buckets/', 21, false), J$.R(7159857, 'name', name, false, false)), path, false, false);
                            var httpReq = J$.W(7159905, 'httpReq', J$.M(7159897, J$.R(7159873, 'this', this, false, false), '_mgmtRequest', false)(J$.R(7159881, 'path', path, false, false), J$.T(7159889, 'DELETE', 21, false)), httpReq, false, false);
                            J$.M(7159937, J$.R(7159913, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7159921, 'error', 21, false), J$.R(7159929, 'callback', callback, false, false));
                            J$.M(7160177, J$.R(7159945, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7159953, 'response', 21, false), J$.F(7160169, J$.R(7159961, '_respRead', _respRead, false, true), false)(J$.T(7160161, function (err, resp, data) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(7160121, arguments.callee, this, arguments);
                                            arguments = J$.N(7160129, 'arguments', arguments, true, false, false);
                                            err = J$.N(7160137, 'err', err, true, false, false);
                                            resp = J$.N(7160145, 'resp', resp, true, false, false);
                                            data = J$.N(7160153, 'data', data, true, false, false);
                                            if (J$.C(383368, J$.R(7159969, 'err', err, false, false))) {
                                                J$.F(7159993, J$.R(7159977, 'callback', callback, false, false), false)(J$.R(7159985, 'err', err, false, false));
                                                return J$.Rt(7160001, undefined);
                                            }
                                            err = J$.W(7160041, 'err', J$.F(7160033, J$.R(7160009, '_respParseError', _respParseError, false, true), false)(J$.R(7160017, 'resp', resp, false, false), J$.R(7160025, 'data', data, false, false)), err, false, false);
                                            if (J$.C(383376, J$.R(7160049, 'err', err, false, false))) {
                                                J$.F(7160073, J$.R(7160057, 'callback', callback, false, false), false)(J$.R(7160065, 'err', err, false, false));
                                                return J$.Rt(7160081, undefined);
                                            }
                                            J$.F(7160113, J$.R(7160089, 'callback', callback, false, false), false)(J$.T(7160097, null, 25, false), J$.T(7160105, true, 23, false));
                                        } catch (J$e) {
                                            J$.Ex(7162969, J$e);
                                        } finally {
                                            if (J$.Fr(7162977))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                            J$.M(7160193, J$.R(7160185, 'httpReq', httpReq, false, false), 'end', false)();
                        } catch (J$e) {
                            J$.Ex(7162985, J$e);
                        } finally {
                            if (J$.Fr(7162993))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7161057, J$.G(7160273, J$.R(7160265, 'ClusterManager', ClusterManager, false, true), 'prototype'), 'upsertUser', J$.T(7161049, function (domain, userid, settings, callback) {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(7160961, arguments.callee, this, arguments);
                            arguments = J$.N(7160969, 'arguments', arguments, true, false, false);
                            domain = J$.N(7160977, 'domain', domain, true, false, false);
                            userid = J$.N(7160985, 'userid', userid, true, false, false);
                            settings = J$.N(7160993, 'settings', settings, true, false, false);
                            callback = J$.N(7161001, 'callback', callback, true, false, false);
                            J$.N(7161009, 'httpReq', httpReq, false, false, false);
                            J$.N(7161017, 'qsroles', qsroles, false, false, false);
                            J$.N(7161025, 'i', i, false, false, false);
                            J$.N(7161033, 'role', role, false, false, false);
                            J$.N(7161041, 'qssettings', qssettings, false, false, false);
                            var httpReq = J$.W(7160337, 'httpReq', J$.M(7160329, J$.R(7160281, 'this', this, false, false), '_mgmtRequest', false)(J$.B(722914, '+', J$.B(722906, '+', J$.B(722898, '+', J$.T(7160289, '/settings/rbac/users/', 21, false), J$.R(7160297, 'domain', domain, false, false)), J$.T(7160305, '/', 21, false)), J$.R(7160313, 'userid', userid, false, false)), J$.T(7160321, 'PUT', 21, false)), httpReq, false, false);
                            J$.M(7160369, J$.R(7160345, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7160353, 'error', 21, false), J$.R(7160361, 'callback', callback, false, false));
                            J$.M(7160609, J$.R(7160377, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7160385, 'response', 21, false), J$.F(7160601, J$.R(7160393, '_respRead', _respRead, false, true), false)(J$.T(7160593, function (err, resp, data) {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(7160553, arguments.callee, this, arguments);
                                            arguments = J$.N(7160561, 'arguments', arguments, true, false, false);
                                            err = J$.N(7160569, 'err', err, true, false, false);
                                            resp = J$.N(7160577, 'resp', resp, true, false, false);
                                            data = J$.N(7160585, 'data', data, true, false, false);
                                            if (J$.C(383384, J$.R(7160401, 'err', err, false, false))) {
                                                J$.F(7160425, J$.R(7160409, 'callback', callback, false, false), false)(J$.R(7160417, 'err', err, false, false));
                                                return J$.Rt(7160433, undefined);
                                            }
                                            err = J$.W(7160473, 'err', J$.F(7160465, J$.R(7160441, '_respParseError', _respParseError, false, true), false)(J$.R(7160449, 'resp', resp, false, false), J$.R(7160457, 'data', data, false, false)), err, false, false);
                                            if (J$.C(383392, J$.R(7160481, 'err', err, false, false))) {
                                                J$.F(7160505, J$.R(7160489, 'callback', callback, false, false), false)(J$.R(7160497, 'err', err, false, false));
                                                return J$.Rt(7160513, undefined);
                                            }
                                            J$.F(7160545, J$.R(7160521, 'callback', callback, false, false), false)(J$.T(7160529, null, 25, false), J$.T(7160537, true, 23, false));
                                        } catch (J$e) {
                                            J$.Ex(7163001, J$e);
                                        } finally {
                                            if (J$.Fr(7163009))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                            var qsroles = J$.W(7160625, 'qsroles', J$.T(7160617, [], 10, false), qsroles, false, false);
                            for (var i = J$.W(7160641, 'i', J$.T(7160633, 0, 22, false), i, false, false); J$.C(383400, J$.B(722922, '<', J$.R(7160649, 'i', i, false, false), J$.G(7160673, J$.G(7160665, J$.R(7160657, 'settings', settings, false, false), 'roles'), 'length'))); i = J$.W(7160689, 'i', J$.B(722938, '+', J$.U(722930, '+', J$.R(7160681, 'i', i, false, false)), 1), i, false, false)) {
                                var role = J$.W(7160729, 'role', J$.G(7160721, J$.G(7160705, J$.R(7160697, 'settings', settings, false, false), 'roles'), J$.R(7160713, 'i', i, false, false)), role, false, false);
                                J$.M(7160793, J$.R(7160737, 'qsroles', qsroles, false, false), 'push', false)(J$.B(722962, '+', J$.B(722954, '+', J$.B(722946, '+', J$.G(7160753, J$.R(7160745, 'role', role, false, false), 'role'), J$.T(7160761, '[', 21, false)), J$.G(7160777, J$.R(7160769, 'role', role, false, false), 'bucket_name')), J$.T(7160785, ']', 21, false)));
                            }
                            var qssettings = J$.W(7160865, 'qssettings', J$.T(7160857, {
                                name: J$.G(7160809, J$.R(7160801, 'settings', settings, false, false), 'name'),
                                password: J$.G(7160825, J$.R(7160817, 'settings', settings, false, false), 'password'),
                                roles: J$.M(7160849, J$.R(7160833, 'qsroles', qsroles, false, false), 'join', false)(J$.T(7160841, ',', 21, false))
                            }, 11, false), qssettings, false, false);
                            J$.M(7160897, J$.R(7160873, 'httpReq', httpReq, false, false), 'setHeader', false)(J$.T(7160881, 'Content-Type', 21, false), J$.T(7160889, 'application/x-www-form-urlencoded', 21, false));
                            J$.M(7160937, J$.R(7160905, 'httpReq', httpReq, false, false), 'write', false)(J$.M(7160929, J$.R(7160913, 'qs', qs, false, true), 'stringify', false)(J$.R(7160921, 'qssettings', qssettings, false, false)));
                            J$.M(7160953, J$.R(7160945, 'httpReq', httpReq, false, false), 'end', false)();
                        } catch (J$e) {
                            J$.Ex(7163017, J$e);
                        } finally {
                            if (J$.Fr(7163025))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7161489, J$.G(7161073, J$.R(7161065, 'ClusterManager', ClusterManager, false, true), 'prototype'), 'removeUser', J$.T(7161481, function (domain, userid, callback) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(7161433, arguments.callee, this, arguments);
                            arguments = J$.N(7161441, 'arguments', arguments, true, false, false);
                            domain = J$.N(7161449, 'domain', domain, true, false, false);
                            userid = J$.N(7161457, 'userid', userid, true, false, false);
                            callback = J$.N(7161465, 'callback', callback, true, false, false);
                            J$.N(7161473, 'httpReq', httpReq, false, false, false);
                            var httpReq = J$.W(7161137, 'httpReq', J$.M(7161129, J$.R(7161081, 'this', this, false, false), '_mgmtRequest', false)(J$.B(722986, '+', J$.B(722978, '+', J$.B(722970, '+', J$.T(7161089, '/settings/rbac/users/', 21, false), J$.R(7161097, 'domain', domain, false, false)), J$.T(7161105, '/', 21, false)), J$.R(7161113, 'userid', userid, false, false)), J$.T(7161121, 'DELETE', 21, false)), httpReq, false, false);
                            J$.M(7161169, J$.R(7161145, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7161153, 'error', 21, false), J$.R(7161161, 'callback', callback, false, false));
                            J$.M(7161409, J$.R(7161177, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7161185, 'response', 21, false), J$.F(7161401, J$.R(7161193, '_respRead', _respRead, false, true), false)(J$.T(7161393, function (err, resp, data) {
                                jalangiLabel18:
                                    while (true) {
                                        try {
                                            J$.Fe(7161353, arguments.callee, this, arguments);
                                            arguments = J$.N(7161361, 'arguments', arguments, true, false, false);
                                            err = J$.N(7161369, 'err', err, true, false, false);
                                            resp = J$.N(7161377, 'resp', resp, true, false, false);
                                            data = J$.N(7161385, 'data', data, true, false, false);
                                            if (J$.C(383408, J$.R(7161201, 'err', err, false, false))) {
                                                J$.F(7161225, J$.R(7161209, 'callback', callback, false, false), false)(J$.R(7161217, 'err', err, false, false));
                                                return J$.Rt(7161233, undefined);
                                            }
                                            err = J$.W(7161273, 'err', J$.F(7161265, J$.R(7161241, '_respParseError', _respParseError, false, true), false)(J$.R(7161249, 'resp', resp, false, false), J$.R(7161257, 'data', data, false, false)), err, false, false);
                                            if (J$.C(383416, J$.R(7161281, 'err', err, false, false))) {
                                                J$.F(7161305, J$.R(7161289, 'callback', callback, false, false), false)(J$.R(7161297, 'err', err, false, false));
                                                return J$.Rt(7161313, undefined);
                                            }
                                            J$.F(7161345, J$.R(7161321, 'callback', callback, false, false), false)(J$.T(7161329, null, 25, false), J$.T(7161337, true, 23, false));
                                        } catch (J$e) {
                                            J$.Ex(7163033, J$e);
                                        } finally {
                                            if (J$.Fr(7163041))
                                                continue jalangiLabel18;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                            J$.M(7161425, J$.R(7161417, 'httpReq', httpReq, false, false), 'end', false)();
                        } catch (J$e) {
                            J$.Ex(7163049, J$e);
                        } finally {
                            if (J$.Fr(7163057))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7162009, J$.G(7161505, J$.R(7161497, 'ClusterManager', ClusterManager, false, true), 'prototype'), 'getUsers', J$.T(7162001, function (domain, callback) {
                jalangiLabel21:
                    while (true) {
                        try {
                            J$.Fe(7161961, arguments.callee, this, arguments);
                            arguments = J$.N(7161969, 'arguments', arguments, true, false, false);
                            domain = J$.N(7161977, 'domain', domain, true, false, false);
                            callback = J$.N(7161985, 'callback', callback, true, false, false);
                            J$.N(7161993, 'httpReq', httpReq, false, false, false);
                            var httpReq = J$.W(7161553, 'httpReq', J$.M(7161545, J$.R(7161513, 'this', this, false, false), '_mgmtRequest', false)(J$.B(722994, '+', J$.T(7161521, '/settings/rbac/users/', 21, false), J$.R(7161529, 'domain', domain, false, false)), J$.T(7161537, 'GET', 21, false)), httpReq, false, false);
                            J$.M(7161585, J$.R(7161561, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7161569, 'error', 21, false), J$.R(7161577, 'callback', callback, false, false));
                            J$.M(7161937, J$.R(7161593, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7161601, 'response', 21, false), J$.F(7161929, J$.R(7161609, '_respRead', _respRead, false, true), false)(J$.T(7161921, function (err, resp, data) {
                                jalangiLabel20:
                                    while (true) {
                                        try {
                                            J$.Fe(7161873, arguments.callee, this, arguments);
                                            arguments = J$.N(7161881, 'arguments', arguments, true, false, false);
                                            err = J$.N(7161889, 'err', err, true, false, false);
                                            resp = J$.N(7161897, 'resp', resp, true, false, false);
                                            data = J$.N(7161905, 'data', data, true, false, false);
                                            J$.N(7161913, 'users', users, false, false, false);
                                            if (J$.C(383424, J$.R(7161617, 'err', err, false, false))) {
                                                J$.F(7161641, J$.R(7161625, 'callback', callback, false, false), false)(J$.R(7161633, 'err', err, false, false));
                                                return J$.Rt(7161649, undefined);
                                            }
                                            err = J$.W(7161689, 'err', J$.F(7161681, J$.R(7161657, '_respParseError', _respParseError, false, true), false)(J$.R(7161665, 'resp', resp, false, false), J$.R(7161673, 'data', data, false, false)), err, false, false);
                                            if (J$.C(383432, J$.R(7161697, 'err', err, false, false))) {
                                                J$.F(7161721, J$.R(7161705, 'callback', callback, false, false), false)(J$.R(7161713, 'err', err, false, false));
                                                return J$.Rt(7161729, undefined);
                                            }
                                            var users;
                                            try {
                                                users = J$.W(7161761, 'users', J$.M(7161753, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7161737, 'JSON', undefined, true, true) : JSON = J$.R(7161737, 'JSON', JSON, true, true)), 'parse', false)(J$.R(7161745, 'data', data, false, false)), users, false, false);
                                            } catch (e) {
                                                J$.N(7161785, 'e', e, false, false, true);
                                                err = J$.W(7161777, 'err', J$.R(7161769, 'e', e, false, false), err, false, false);
                                            }
                                            if (J$.C(383440, J$.R(7161793, 'err', err, false, false))) {
                                                J$.F(7161825, J$.R(7161801, 'callback', callback, false, false), false)(J$.R(7161809, 'err', err, false, false), J$.T(7161817, null, 25, false));
                                                return J$.Rt(7161833, undefined);
                                            }
                                            J$.F(7161865, J$.R(7161841, 'callback', callback, false, false), false)(J$.T(7161849, null, 25, false), J$.R(7161857, 'users', users, false, false));
                                        } catch (J$e) {
                                            J$.Ex(7163065, J$e);
                                        } finally {
                                            if (J$.Fr(7163073))
                                                continue jalangiLabel20;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                            J$.M(7161953, J$.R(7161945, 'httpReq', httpReq, false, false), 'end', false)();
                        } catch (J$e) {
                            J$.Ex(7163081, J$e);
                        } finally {
                            if (J$.Fr(7163089))
                                continue jalangiLabel21;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7162553, J$.G(7162025, J$.R(7162017, 'ClusterManager', ClusterManager, false, true), 'prototype'), 'getUser', J$.T(7162545, function (domain, userid, callback) {
                jalangiLabel23:
                    while (true) {
                        try {
                            J$.Fe(7162497, arguments.callee, this, arguments);
                            arguments = J$.N(7162505, 'arguments', arguments, true, false, false);
                            domain = J$.N(7162513, 'domain', domain, true, false, false);
                            userid = J$.N(7162521, 'userid', userid, true, false, false);
                            callback = J$.N(7162529, 'callback', callback, true, false, false);
                            J$.N(7162537, 'httpReq', httpReq, false, false, false);
                            var httpReq = J$.W(7162089, 'httpReq', J$.M(7162081, J$.R(7162033, 'this', this, false, false), '_mgmtRequest', false)(J$.B(723018, '+', J$.B(723010, '+', J$.B(723002, '+', J$.T(7162041, '/settings/rbac/users/', 21, false), J$.R(7162049, 'domain', domain, false, false)), J$.T(7162057, '/', 21, false)), J$.R(7162065, 'userid', userid, false, false)), J$.T(7162073, 'GET', 21, false)), httpReq, false, false);
                            J$.M(7162121, J$.R(7162097, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7162105, 'error', 21, false), J$.R(7162113, 'callback', callback, false, false));
                            J$.M(7162473, J$.R(7162129, 'httpReq', httpReq, false, false), 'on', false)(J$.T(7162137, 'response', 21, false), J$.F(7162465, J$.R(7162145, '_respRead', _respRead, false, true), false)(J$.T(7162457, function (err, resp, data) {
                                jalangiLabel22:
                                    while (true) {
                                        try {
                                            J$.Fe(7162409, arguments.callee, this, arguments);
                                            arguments = J$.N(7162417, 'arguments', arguments, true, false, false);
                                            err = J$.N(7162425, 'err', err, true, false, false);
                                            resp = J$.N(7162433, 'resp', resp, true, false, false);
                                            data = J$.N(7162441, 'data', data, true, false, false);
                                            J$.N(7162449, 'users', users, false, false, false);
                                            if (J$.C(383448, J$.R(7162153, 'err', err, false, false))) {
                                                J$.F(7162177, J$.R(7162161, 'callback', callback, false, false), false)(J$.R(7162169, 'err', err, false, false));
                                                return J$.Rt(7162185, undefined);
                                            }
                                            err = J$.W(7162225, 'err', J$.F(7162217, J$.R(7162193, '_respParseError', _respParseError, false, true), false)(J$.R(7162201, 'resp', resp, false, false), J$.R(7162209, 'data', data, false, false)), err, false, false);
                                            if (J$.C(383456, J$.R(7162233, 'err', err, false, false))) {
                                                J$.F(7162257, J$.R(7162241, 'callback', callback, false, false), false)(J$.R(7162249, 'err', err, false, false));
                                                return J$.Rt(7162265, undefined);
                                            }
                                            var users;
                                            try {
                                                users = J$.W(7162297, 'users', J$.M(7162289, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7162273, 'JSON', undefined, true, true) : JSON = J$.R(7162273, 'JSON', JSON, true, true)), 'parse', false)(J$.R(7162281, 'data', data, false, false)), users, false, false);
                                            } catch (e) {
                                                J$.N(7162321, 'e', e, false, false, true);
                                                err = J$.W(7162313, 'err', J$.R(7162305, 'e', e, false, false), err, false, false);
                                            }
                                            if (J$.C(383464, J$.R(7162329, 'err', err, false, false))) {
                                                J$.F(7162361, J$.R(7162337, 'callback', callback, false, false), false)(J$.R(7162345, 'err', err, false, false), J$.T(7162353, null, 25, false));
                                                return J$.Rt(7162369, undefined);
                                            }
                                            J$.F(7162401, J$.R(7162377, 'callback', callback, false, false), false)(J$.T(7162385, null, 25, false), J$.R(7162393, 'users', users, false, false));
                                        } catch (J$e) {
                                            J$.Ex(7163097, J$e);
                                        } finally {
                                            if (J$.Fr(7163105))
                                                continue jalangiLabel22;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)));
                            J$.M(7162489, J$.R(7162481, 'httpReq', httpReq, false, false), 'end', false)();
                        } catch (J$e) {
                            J$.Ex(7163113, J$e);
                        } finally {
                            if (J$.Fr(7163121))
                                continue jalangiLabel23;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7162633, J$.G(7162569, J$.R(7162561, 'ClusterManager', ClusterManager, false, true), 'prototype'), 'searchIndexManager', J$.T(7162625, function () {
                jalangiLabel24:
                    while (true) {
                        try {
                            J$.Fe(7162609, arguments.callee, this, arguments);
                            arguments = J$.N(7162617, 'arguments', arguments, true, false, false);
                            return J$.Rt(7162601, J$.F(7162593, J$.R(7162577, 'SearchIndexManager', SearchIndexManager, false, true), true)(J$.R(7162585, 'this', this, false, false)));
                        } catch (J$e) {
                            J$.Ex(7163129, J$e);
                        } finally {
                            if (J$.Fr(7163137))
                                continue jalangiLabel24;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7162657, J$.I(typeof module === 'undefined' ? module = J$.R(7162641, 'module', undefined, true, true) : module = J$.R(7162641, 'module', module, true, true)), 'exports', J$.R(7162649, 'ClusterManager', ClusterManager, false, true));
        } catch (J$e) {
            J$.Ex(7163145, J$e);
        } finally {
            if (J$.Sr(7163153))
                continue jalangiLabel25;
            else
                break jalangiLabel25;
        }
    }
// JALANGI DO NOT INSTRUMENT
