// couchbase/lib/constants.js
J$.noInstrEval = false;
jalangiLabel0:
    while (true) {
        try {
            J$.Se(7166969, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/couchbase/lib/constants.js');
            J$.N(7166977, 'ServiceType', ServiceType, false, false, false);
            J$.T(7166865, 'use strict', 21, false);
            var ServiceType = J$.W(7166929, 'ServiceType', J$.T(7166921, {
                Config: J$.T(7166873, 'config', 21, false),
                KeyValue: J$.T(7166881, 'kv', 21, false),
                Views: J$.T(7166889, 'view', 21, false),
                Query: J$.T(7166897, 'query', 21, false),
                Search: J$.T(7166905, 'search', 21, false),
                Analytics: J$.T(7166913, 'analytics', 21, false)
            }, 11, false), ServiceType, false, true);
            J$.P(7166961, J$.G(7166945, J$.I(typeof module === 'undefined' ? module = J$.R(7166937, 'module', undefined, true, true) : module = J$.R(7166937, 'module', module, true, true)), 'exports'), 'ServiceType', J$.R(7166953, 'ServiceType', ServiceType, false, true));
        } catch (J$e) {
            J$.Ex(7166985, J$e);
        } finally {
            if (J$.Sr(7166993))
                continue jalangiLabel0;
            else
                break jalangiLabel0;
        }
    }
// JALANGI DO NOT INSTRUMENT
