// couchbase/lib/searchquery_sort.js
J$.noInstrEval = false;
jalangiLabel18:
    while (true) {
        try {
            J$.Se(7230361, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/couchbase/lib/searchquery_sort.js');
            function SearchSortBase() {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(7228529, arguments.callee, this, arguments);
                            arguments = J$.N(7228537, 'arguments', arguments, true, false, false);
                            J$.P(7228521, J$.R(7228505, 'this', this, false, false), 'data', J$.T(7228513, {}, 11, false));
                        } catch (J$e) {
                            J$.Ex(7230457, J$e);
                        } finally {
                            if (J$.Fr(7230465))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function ScoreSort() {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(7228689, arguments.callee, this, arguments);
                            arguments = J$.N(7228697, 'arguments', arguments, true, false, false);
                            J$.P(7228681, J$.R(7228657, 'this', this, false, false), 'data', J$.T(7228673, { by: J$.T(7228665, 'score', 21, false) }, 11, false));
                        } catch (J$e) {
                            J$.Ex(7230489, J$e);
                        } finally {
                            if (J$.Fr(7230497))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function IdSort() {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(7228977, arguments.callee, this, arguments);
                            arguments = J$.N(7228985, 'arguments', arguments, true, false, false);
                            J$.P(7228969, J$.R(7228945, 'this', this, false, false), 'data', J$.T(7228961, { by: J$.T(7228953, 'id', 21, false) }, 11, false));
                        } catch (J$e) {
                            J$.Ex(7230537, J$e);
                        } finally {
                            if (J$.Fr(7230545))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function FieldSort(field) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(7229273, arguments.callee, this, arguments);
                            arguments = J$.N(7229281, 'arguments', arguments, true, false, false);
                            field = J$.N(7229289, 'field', field, true, false, false);
                            J$.P(7229265, J$.R(7229233, 'this', this, false, false), 'data', J$.T(7229257, {
                                by: J$.T(7229241, 'field', 21, false),
                                field: J$.R(7229249, 'field', field, false, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(7230585, J$e);
                        } finally {
                            if (J$.Fr(7230593))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function GeoDistanceSort(field, lat, lon) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(7229929, arguments.callee, this, arguments);
                            arguments = J$.N(7229937, 'arguments', arguments, true, false, false);
                            field = J$.N(7229945, 'field', field, true, false, false);
                            lat = J$.N(7229953, 'lat', lat, true, false, false);
                            lon = J$.N(7229961, 'lon', lon, true, false, false);
                            J$.P(7229921, J$.R(7229865, 'this', this, false, false), 'data', J$.T(7229913, {
                                by: J$.T(7229873, 'geo_distance', 21, false),
                                field: J$.R(7229881, 'field', field, false, false),
                                location: J$.T(7229905, [
                                    J$.R(7229889, 'lon', lon, false, false),
                                    J$.R(7229897, 'lat', lat, false, false)
                                ], 10, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(7230681, J$e);
                        } finally {
                            if (J$.Fr(7230689))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }
            SearchSortBase = J$.N(7230377, 'SearchSortBase', J$.T(7230369, SearchSortBase, 12, false), true, false, false);
            J$.N(7230385, 'SearchSort', SearchSort, false, false, false);
            ScoreSort = J$.N(7230401, 'ScoreSort', J$.T(7230393, ScoreSort, 12, false), true, false, false);
            IdSort = J$.N(7230417, 'IdSort', J$.T(7230409, IdSort, 12, false), true, false, false);
            FieldSort = J$.N(7230433, 'FieldSort', J$.T(7230425, FieldSort, 12, false), true, false, false);
            GeoDistanceSort = J$.N(7230449, 'GeoDistanceSort', J$.T(7230441, GeoDistanceSort, 12, false), true, false, false);
            J$.T(7228497, 'use strict', 21, false);
            J$.P(7228609, J$.G(7228553, J$.R(7228545, 'SearchSortBase', SearchSortBase, false, true), 'prototype'), 'toJSON', J$.T(7228601, function () {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7228585, arguments.callee, this, arguments);
                            arguments = J$.N(7228593, 'arguments', arguments, true, false, false);
                            return J$.Rt(7228577, J$.G(7228569, J$.R(7228561, 'this', this, false, false), 'data'));
                        } catch (J$e) {
                            J$.Ex(7230473, J$e);
                        } finally {
                            if (J$.Fr(7230481))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var SearchSort = J$.W(7228625, 'SearchSort', J$.T(7228617, {}, 11, false), SearchSort, false, true);
            J$.P(7228649, J$.I(typeof module === 'undefined' ? module = J$.R(7228633, 'module', undefined, true, true) : module = J$.R(7228633, 'module', module, true, true)), 'exports', J$.R(7228641, 'SearchSort', SearchSort, false, true));
            J$.P(7228721, J$.R(7228705, 'SearchSort', SearchSort, false, true), 'ScoreSort', J$.R(7228713, 'ScoreSort', ScoreSort, false, true));
            J$.P(7228785, J$.R(7228729, 'SearchSort', SearchSort, false, true), 'score', J$.T(7228777, function () {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(7228761, arguments.callee, this, arguments);
                            arguments = J$.N(7228769, 'arguments', arguments, true, false, false);
                            return J$.Rt(7228753, J$.F(7228745, J$.R(7228737, 'ScoreSort', ScoreSort, false, true), true)());
                        } catch (J$e) {
                            J$.Ex(7230505, J$e);
                        } finally {
                            if (J$.Fr(7230513))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7228889, J$.G(7228801, J$.R(7228793, 'ScoreSort', ScoreSort, false, true), 'prototype'), 'descending', J$.T(7228881, function (descending) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7228857, arguments.callee, this, arguments);
                            arguments = J$.N(7228865, 'arguments', arguments, true, false, false);
                            descending = J$.N(7228873, 'descending', descending, true, false, false);
                            J$.P(7228833, J$.G(7228817, J$.R(7228809, 'this', this, false, false), 'data'), 'desc', J$.R(7228825, 'descending', descending, false, false));
                            return J$.Rt(7228849, J$.R(7228841, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7230521, J$e);
                        } finally {
                            if (J$.Fr(7230529))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7228937, J$.G(7228905, J$.R(7228897, 'ScoreSort', ScoreSort, false, true), 'prototype'), 'toJSON', J$.G(7228929, J$.G(7228921, J$.R(7228913, 'SearchSortBase', SearchSortBase, false, true), 'prototype'), 'toJSON'));
            J$.P(7229009, J$.R(7228993, 'SearchSort', SearchSort, false, true), 'IdSort', J$.R(7229001, 'IdSort', IdSort, false, true));
            J$.P(7229073, J$.R(7229017, 'SearchSort', SearchSort, false, true), 'id', J$.T(7229065, function () {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(7229049, arguments.callee, this, arguments);
                            arguments = J$.N(7229057, 'arguments', arguments, true, false, false);
                            return J$.Rt(7229041, J$.F(7229033, J$.R(7229025, 'IdSort', IdSort, false, true), true)());
                        } catch (J$e) {
                            J$.Ex(7230553, J$e);
                        } finally {
                            if (J$.Fr(7230561))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7229177, J$.G(7229089, J$.R(7229081, 'IdSort', IdSort, false, true), 'prototype'), 'descending', J$.T(7229169, function (descending) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(7229145, arguments.callee, this, arguments);
                            arguments = J$.N(7229153, 'arguments', arguments, true, false, false);
                            descending = J$.N(7229161, 'descending', descending, true, false, false);
                            J$.P(7229121, J$.G(7229105, J$.R(7229097, 'this', this, false, false), 'data'), 'desc', J$.R(7229113, 'descending', descending, false, false));
                            return J$.Rt(7229137, J$.R(7229129, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7230569, J$e);
                        } finally {
                            if (J$.Fr(7230577))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7229225, J$.G(7229193, J$.R(7229185, 'IdSort', IdSort, false, true), 'prototype'), 'toJSON', J$.G(7229217, J$.G(7229209, J$.R(7229201, 'SearchSortBase', SearchSortBase, false, true), 'prototype'), 'toJSON'));
            J$.P(7229313, J$.R(7229297, 'SearchSort', SearchSort, false, true), 'FieldSort', J$.R(7229305, 'FieldSort', FieldSort, false, true));
            J$.P(7229393, J$.R(7229321, 'SearchSort', SearchSort, false, true), 'field', J$.T(7229385, function (field) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(7229361, arguments.callee, this, arguments);
                            arguments = J$.N(7229369, 'arguments', arguments, true, false, false);
                            field = J$.N(7229377, 'field', field, true, false, false);
                            return J$.Rt(7229353, J$.F(7229345, J$.R(7229329, 'FieldSort', FieldSort, false, true), true)(J$.R(7229337, 'field', field, false, false)));
                        } catch (J$e) {
                            J$.Ex(7230601, J$e);
                        } finally {
                            if (J$.Fr(7230609))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7229497, J$.G(7229409, J$.R(7229401, 'FieldSort', FieldSort, false, true), 'prototype'), 'type', J$.T(7229489, function (type) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(7229465, arguments.callee, this, arguments);
                            arguments = J$.N(7229473, 'arguments', arguments, true, false, false);
                            type = J$.N(7229481, 'type', type, true, false, false);
                            J$.P(7229441, J$.G(7229425, J$.R(7229417, 'this', this, false, false), 'data'), 'type', J$.R(7229433, 'type', type, false, false));
                            return J$.Rt(7229457, J$.R(7229449, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7230617, J$e);
                        } finally {
                            if (J$.Fr(7230625))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7229601, J$.G(7229513, J$.R(7229505, 'FieldSort', FieldSort, false, true), 'prototype'), 'mode', J$.T(7229593, function (mode) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(7229569, arguments.callee, this, arguments);
                            arguments = J$.N(7229577, 'arguments', arguments, true, false, false);
                            mode = J$.N(7229585, 'mode', mode, true, false, false);
                            J$.P(7229545, J$.G(7229529, J$.R(7229521, 'this', this, false, false), 'data'), 'mode', J$.R(7229537, 'mode', mode, false, false));
                            return J$.Rt(7229561, J$.R(7229553, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7230633, J$e);
                        } finally {
                            if (J$.Fr(7230641))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7229705, J$.G(7229617, J$.R(7229609, 'FieldSort', FieldSort, false, true), 'prototype'), 'missing', J$.T(7229697, function (missing) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(7229673, arguments.callee, this, arguments);
                            arguments = J$.N(7229681, 'arguments', arguments, true, false, false);
                            missing = J$.N(7229689, 'missing', missing, true, false, false);
                            J$.P(7229649, J$.G(7229633, J$.R(7229625, 'this', this, false, false), 'data'), 'missing', J$.R(7229641, 'missing', missing, false, false));
                            return J$.Rt(7229665, J$.R(7229657, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7230649, J$e);
                        } finally {
                            if (J$.Fr(7230657))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7229809, J$.G(7229721, J$.R(7229713, 'FieldSort', FieldSort, false, true), 'prototype'), 'descending', J$.T(7229801, function (descending) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(7229777, arguments.callee, this, arguments);
                            arguments = J$.N(7229785, 'arguments', arguments, true, false, false);
                            descending = J$.N(7229793, 'descending', descending, true, false, false);
                            J$.P(7229753, J$.G(7229737, J$.R(7229729, 'this', this, false, false), 'data'), 'desc', J$.R(7229745, 'descending', descending, false, false));
                            return J$.Rt(7229769, J$.R(7229761, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7230665, J$e);
                        } finally {
                            if (J$.Fr(7230673))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7229857, J$.G(7229825, J$.R(7229817, 'FieldSort', FieldSort, false, true), 'prototype'), 'toJSON', J$.G(7229849, J$.G(7229841, J$.R(7229833, 'SearchSortBase', SearchSortBase, false, true), 'prototype'), 'toJSON'));
            J$.P(7229985, J$.R(7229969, 'SearchSort', SearchSort, false, true), 'GeoDistanceSort', J$.R(7229977, 'GeoDistanceSort', GeoDistanceSort, false, true));
            J$.P(7230097, J$.R(7229993, 'SearchSort', SearchSort, false, true), 'geoDistance', J$.T(7230089, function (field, lat, lon) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(7230049, arguments.callee, this, arguments);
                            arguments = J$.N(7230057, 'arguments', arguments, true, false, false);
                            field = J$.N(7230065, 'field', field, true, false, false);
                            lat = J$.N(7230073, 'lat', lat, true, false, false);
                            lon = J$.N(7230081, 'lon', lon, true, false, false);
                            return J$.Rt(7230041, J$.F(7230033, J$.R(7230001, 'GeoDistanceSort', GeoDistanceSort, false, true), true)(J$.R(7230009, 'field', field, false, false), J$.R(7230017, 'lat', lat, false, false), J$.R(7230025, 'lon', lon, false, false)));
                        } catch (J$e) {
                            J$.Ex(7230697, J$e);
                        } finally {
                            if (J$.Fr(7230705))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7230201, J$.G(7230113, J$.R(7230105, 'GeoDistanceSort', GeoDistanceSort, false, true), 'prototype'), 'unit', J$.T(7230193, function (unit) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(7230169, arguments.callee, this, arguments);
                            arguments = J$.N(7230177, 'arguments', arguments, true, false, false);
                            unit = J$.N(7230185, 'unit', unit, true, false, false);
                            J$.P(7230145, J$.G(7230129, J$.R(7230121, 'this', this, false, false), 'data'), 'unit', J$.R(7230137, 'unit', unit, false, false));
                            return J$.Rt(7230161, J$.R(7230153, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7230713, J$e);
                        } finally {
                            if (J$.Fr(7230721))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7230305, J$.G(7230217, J$.R(7230209, 'GeoDistanceSort', GeoDistanceSort, false, true), 'prototype'), 'descending', J$.T(7230297, function (descending) {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(7230273, arguments.callee, this, arguments);
                            arguments = J$.N(7230281, 'arguments', arguments, true, false, false);
                            descending = J$.N(7230289, 'descending', descending, true, false, false);
                            J$.P(7230249, J$.G(7230233, J$.R(7230225, 'this', this, false, false), 'data'), 'desc', J$.R(7230241, 'descending', descending, false, false));
                            return J$.Rt(7230265, J$.R(7230257, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7230729, J$e);
                        } finally {
                            if (J$.Fr(7230737))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7230353, J$.G(7230321, J$.R(7230313, 'GeoDistanceSort', GeoDistanceSort, false, true), 'prototype'), 'toJSON', J$.G(7230345, J$.G(7230337, J$.R(7230329, 'SearchSortBase', SearchSortBase, false, true), 'prototype'), 'toJSON'));
        } catch (J$e) {
            J$.Ex(7230745, J$e);
        } finally {
            if (J$.Sr(7230753))
                continue jalangiLabel18;
            else
                break jalangiLabel18;
        }
    }
// JALANGI DO NOT INSTRUMENT
