// couchbase/lib/searchquery.js
J$.noInstrEval = false;
jalangiLabel13:
    while (true) {
        try {
            J$.Se(7215753, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/couchbase/lib/searchquery.js');
            function SearchQuery(indexName, query) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(7213409, arguments.callee, this, arguments);
                            arguments = J$.N(7213417, 'arguments', arguments, true, false, false);
                            indexName = J$.N(7213425, 'indexName', indexName, true, false, false);
                            query = J$.N(7213433, 'query', query, true, false, false);
                            J$.P(7213401, J$.R(7213369, 'this', this, false, false), 'data', J$.T(7213393, {
                                indexName: J$.R(7213377, 'indexName', indexName, false, false),
                                query: J$.R(7213385, 'query', query, false, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(7215809, J$e);
                        } finally {
                            if (J$.Fr(7215817))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(7215761, 'util', util, false, false, false);
            J$.N(7215769, 'cbutils', cbutils, false, false, false);
            J$.N(7215777, 'queryProtos', queryProtos, false, false, false);
            SearchQuery = J$.N(7215793, 'SearchQuery', J$.T(7215785, SearchQuery, 12, false), true, false, false);
            J$.N(7215801, 'i', i, false, false, false);
            J$.T(7213265, 'use strict', 21, false);
            var util = J$.W(7213297, 'util', J$.F(7213289, J$.I(typeof require === 'undefined' ? require = J$.R(7213273, 'require', undefined, true, true) : require = J$.R(7213273, 'require', require, true, true)), false)(J$.T(7213281, 'util', 21, false)), util, false, true);
            var cbutils = J$.W(7213329, 'cbutils', J$.F(7213321, J$.I(typeof require === 'undefined' ? require = J$.R(7213305, 'require', undefined, true, true) : require = J$.R(7213305, 'require', require, true, true)), false)(J$.T(7213313, './utils', 21, false)), cbutils, false, true);
            var queryProtos = J$.W(7213361, 'queryProtos', J$.F(7213353, J$.I(typeof require === 'undefined' ? require = J$.R(7213337, 'require', undefined, true, true) : require = J$.R(7213337, 'require', require, true, true)), false)(J$.T(7213345, './searchquery_queries', 21, false)), queryProtos, false, true);
            J$.P(7213465, J$.R(7213441, 'SearchQuery', SearchQuery, false, true), 'Consistency', J$.T(7213457, { NOT_BOUNDED: J$.T(7213449, 1, 22, false) }, 11, false));
            J$.P(7213513, J$.R(7213473, 'SearchQuery', SearchQuery, false, true), 'HighlightStyle', J$.T(7213505, {
                DEFAULT: J$.T(7213481, undefined, 24, false),
                HTML: J$.T(7213489, 'html', 21, false),
                ANSI: J$.T(7213497, 'ansi', 21, false)
            }, 11, false));
            J$.P(7213617, J$.G(7213529, J$.R(7213521, 'SearchQuery', SearchQuery, false, true), 'prototype'), 'skip', J$.T(7213609, function (skip) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7213585, arguments.callee, this, arguments);
                            arguments = J$.N(7213593, 'arguments', arguments, true, false, false);
                            skip = J$.N(7213601, 'skip', skip, true, false, false);
                            J$.P(7213561, J$.G(7213545, J$.R(7213537, 'this', this, false, false), 'data'), 'from', J$.R(7213553, 'skip', skip, false, false));
                            return J$.Rt(7213577, J$.R(7213569, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7215825, J$e);
                        } finally {
                            if (J$.Fr(7215833))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7213721, J$.G(7213633, J$.R(7213625, 'SearchQuery', SearchQuery, false, true), 'prototype'), 'limit', J$.T(7213713, function (limit) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(7213689, arguments.callee, this, arguments);
                            arguments = J$.N(7213697, 'arguments', arguments, true, false, false);
                            limit = J$.N(7213705, 'limit', limit, true, false, false);
                            J$.P(7213665, J$.G(7213649, J$.R(7213641, 'this', this, false, false), 'data'), 'size', J$.R(7213657, 'limit', limit, false, false));
                            return J$.Rt(7213681, J$.R(7213673, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7215841, J$e);
                        } finally {
                            if (J$.Fr(7215849))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7213825, J$.G(7213737, J$.R(7213729, 'SearchQuery', SearchQuery, false, true), 'prototype'), 'explain', J$.T(7213817, function (explain) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(7213793, arguments.callee, this, arguments);
                            arguments = J$.N(7213801, 'arguments', arguments, true, false, false);
                            explain = J$.N(7213809, 'explain', explain, true, false, false);
                            J$.P(7213769, J$.G(7213753, J$.R(7213745, 'this', this, false, false), 'data'), 'explain', J$.R(7213761, 'explain', explain, false, false));
                            return J$.Rt(7213785, J$.R(7213777, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7215857, J$e);
                        } finally {
                            if (J$.Fr(7215865))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7214089, J$.G(7213841, J$.R(7213833, 'SearchQuery', SearchQuery, false, true), 'prototype'), 'highlight', J$.T(7214081, function (style, fields) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7214049, arguments.callee, this, arguments);
                            arguments = J$.N(7214057, 'arguments', arguments, true, false, false);
                            style = J$.N(7214065, 'style', style, true, false, false);
                            fields = J$.N(7214073, 'fields', fields, true, false, false);
                            fields = J$.W(7213889, 'fields', J$.M(7213881, J$.R(7213849, 'cbutils', cbutils, false, true), 'unpackArgs', false)(J$.R(7213857, 'fields', fields, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7213865, 'arguments', undefined, true, true) : arguments = J$.R(7213865, 'arguments', arguments, true, true)), J$.T(7213873, 1, 22, false)), fields, false, false);
                            if (J$.C(386456, J$.U(726626, '!', J$.G(7213913, J$.G(7213905, J$.R(7213897, 'this', this, false, false), 'data'), 'highlight')))) {
                                J$.P(7213945, J$.G(7213929, J$.R(7213921, 'this', this, false, false), 'data'), 'highlight', J$.T(7213937, {}, 11, false));
                            }
                            J$.P(7213985, J$.G(7213969, J$.G(7213961, J$.R(7213953, 'this', this, false, false), 'data'), 'highlight'), 'style', J$.R(7213977, 'style', style, false, false));
                            J$.P(7214025, J$.G(7214009, J$.G(7214001, J$.R(7213993, 'this', this, false, false), 'data'), 'highlight'), 'fields', J$.R(7214017, 'fields', fields, false, false));
                            return J$.Rt(7214041, J$.R(7214033, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7215873, J$e);
                        } finally {
                            if (J$.Fr(7215881))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7214233, J$.G(7214105, J$.R(7214097, 'SearchQuery', SearchQuery, false, true), 'prototype'), 'fields', J$.T(7214225, function (fields) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(7214201, arguments.callee, this, arguments);
                            arguments = J$.N(7214209, 'arguments', arguments, true, false, false);
                            fields = J$.N(7214217, 'fields', fields, true, false, false);
                            fields = J$.W(7214145, 'fields', J$.M(7214137, J$.R(7214113, 'cbutils', cbutils, false, true), 'unpackArgs', false)(J$.R(7214121, 'fields', fields, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7214129, 'arguments', undefined, true, true) : arguments = J$.R(7214129, 'arguments', arguments, true, true))), fields, false, false);
                            J$.P(7214177, J$.G(7214161, J$.R(7214153, 'this', this, false, false), 'data'), 'fields', J$.R(7214169, 'fields', fields, false, false));
                            return J$.Rt(7214193, J$.R(7214185, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7215889, J$e);
                        } finally {
                            if (J$.Fr(7215897))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7214377, J$.G(7214249, J$.R(7214241, 'SearchQuery', SearchQuery, false, true), 'prototype'), 'sort', J$.T(7214369, function (fields) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(7214345, arguments.callee, this, arguments);
                            arguments = J$.N(7214353, 'arguments', arguments, true, false, false);
                            fields = J$.N(7214361, 'fields', fields, true, false, false);
                            fields = J$.W(7214289, 'fields', J$.M(7214281, J$.R(7214257, 'cbutils', cbutils, false, true), 'unpackArgs', false)(J$.R(7214265, 'fields', fields, false, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7214273, 'arguments', undefined, true, true) : arguments = J$.R(7214273, 'arguments', arguments, true, true))), fields, false, false);
                            J$.P(7214321, J$.G(7214305, J$.R(7214297, 'this', this, false, false), 'data'), 'sort', J$.R(7214313, 'fields', fields, false, false));
                            return J$.Rt(7214337, J$.R(7214329, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7215905, J$e);
                        } finally {
                            if (J$.Fr(7215913))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7214561, J$.G(7214393, J$.R(7214385, 'SearchQuery', SearchQuery, false, true), 'prototype'), 'addFacet', J$.T(7214553, function (name, facet) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(7214521, arguments.callee, this, arguments);
                            arguments = J$.N(7214529, 'arguments', arguments, true, false, false);
                            name = J$.N(7214537, 'name', name, true, false, false);
                            facet = J$.N(7214545, 'facet', facet, true, false, false);
                            if (J$.C(386464, J$.U(726634, '!', J$.G(7214417, J$.G(7214409, J$.R(7214401, 'this', this, false, false), 'data'), 'facets')))) {
                                J$.P(7214449, J$.G(7214433, J$.R(7214425, 'this', this, false, false), 'data'), 'facets', J$.T(7214441, {}, 11, false));
                            }
                            J$.P(7214497, J$.G(7214473, J$.G(7214465, J$.R(7214457, 'this', this, false, false), 'data'), 'facets'), J$.R(7214481, 'name', name, false, false), J$.R(7214489, 'facet', facet, false, false));
                            return J$.Rt(7214513, J$.R(7214505, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7215921, J$e);
                        } finally {
                            if (J$.Fr(7215929))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7214937, J$.G(7214577, J$.R(7214569, 'SearchQuery', SearchQuery, false, true), 'prototype'), 'consistency', J$.T(7214929, function (val) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(7214905, arguments.callee, this, arguments);
                            arguments = J$.N(7214913, 'arguments', arguments, true, false, false);
                            val = J$.N(7214921, 'val', val, true, false, false);
                            if (J$.C(386472, J$.U(726642, '!', J$.G(7214601, J$.G(7214593, J$.R(7214585, 'this', this, false, false), 'data'), 'ctl')))) {
                                J$.P(7214633, J$.G(7214617, J$.R(7214609, 'this', this, false, false), 'data'), 'ctl', J$.T(7214625, {}, 11, false));
                            }
                            if (J$.C(386480, J$.U(726650, '!', J$.G(7214665, J$.G(7214657, J$.G(7214649, J$.R(7214641, 'this', this, false, false), 'data'), 'ctl'), 'consistency')))) {
                                J$.P(7214705, J$.G(7214689, J$.G(7214681, J$.R(7214673, 'this', this, false, false), 'data'), 'ctl'), 'consistency', J$.T(7214697, {}, 11, false));
                            }
                            if (J$.C(386488, J$.B(726658, '!==', J$.G(7214745, J$.G(7214737, J$.G(7214729, J$.G(7214721, J$.R(7214713, 'this', this, false, false), 'data'), 'ctl'), 'consistency'), 'level'), J$.T(7214753, undefined, 24, false)))) {
                                throw J$.F(7214777, J$.I(typeof Error === 'undefined' ? Error = J$.R(7214761, 'Error', undefined, true, true) : Error = J$.R(7214761, 'Error', Error, true, true)), true)(J$.T(7214769, 'consistency and consistentWith must be use exclusively.', 21, false));
                            }
                            if (J$.C(386496, J$.B(726666, '===', J$.R(7214785, 'val', val, false, false), J$.G(7214809, J$.G(7214801, J$.R(7214793, 'SearchQuery', SearchQuery, false, true), 'Consistency'), 'NOT_BOUNDED')))) {
                                J$.P(7214857, J$.G(7214841, J$.G(7214833, J$.G(7214825, J$.R(7214817, 'this', this, false, false), 'data'), 'ctl'), 'consistency'), 'level', J$.T(7214849, 'not_bounded', 21, false));
                            } else {
                                throw J$.F(7214881, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(7214865, 'TypeError', undefined, true, true) : TypeError = J$.R(7214865, 'TypeError', TypeError, true, true)), true)(J$.T(7214873, 'invalid option passed.', 21, false));
                            }
                            return J$.Rt(7214897, J$.R(7214889, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7215937, J$e);
                        } finally {
                            if (J$.Fr(7215945))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7215289, J$.G(7214953, J$.R(7214945, 'SearchQuery', SearchQuery, false, true), 'prototype'), 'consistentWith', J$.T(7215281, function (state) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(7215257, arguments.callee, this, arguments);
                            arguments = J$.N(7215265, 'arguments', arguments, true, false, false);
                            state = J$.N(7215273, 'state', state, true, false, false);
                            if (J$.C(386504, J$.U(726674, '!', J$.G(7214977, J$.G(7214969, J$.R(7214961, 'this', this, false, false), 'data'), 'ctl')))) {
                                J$.P(7215009, J$.G(7214993, J$.R(7214985, 'this', this, false, false), 'data'), 'ctl', J$.T(7215001, {}, 11, false));
                            }
                            if (J$.C(386512, J$.U(726682, '!', J$.G(7215041, J$.G(7215033, J$.G(7215025, J$.R(7215017, 'this', this, false, false), 'data'), 'ctl'), 'consistency')))) {
                                J$.P(7215081, J$.G(7215065, J$.G(7215057, J$.R(7215049, 'this', this, false, false), 'data'), 'ctl'), 'consistency', J$.T(7215073, {}, 11, false));
                            }
                            if (J$.C(386520, J$.B(726690, '!==', J$.G(7215121, J$.G(7215113, J$.G(7215105, J$.G(7215097, J$.R(7215089, 'this', this, false, false), 'data'), 'ctl'), 'consistency'), 'level'), J$.T(7215129, undefined, 24, false)))) {
                                throw J$.F(7215153, J$.I(typeof Error === 'undefined' ? Error = J$.R(7215137, 'Error', undefined, true, true) : Error = J$.R(7215137, 'Error', Error, true, true)), true)(J$.T(7215145, 'consistency and consistentWith must be use exclusively.', 21, false));
                            }
                            J$.P(7215201, J$.G(7215185, J$.G(7215177, J$.G(7215169, J$.R(7215161, 'this', this, false, false), 'data'), 'ctl'), 'consistency'), 'level', J$.T(7215193, 'at_plus', 21, false));
                            J$.P(7215249, J$.G(7215233, J$.G(7215225, J$.G(7215217, J$.R(7215209, 'this', this, false, false), 'data'), 'ctl'), 'consistency'), 'vectors', J$.R(7215241, 'state', state, false, false));
                        } catch (J$e) {
                            J$.Ex(7215953, J$e);
                        } finally {
                            if (J$.Fr(7215961))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7215457, J$.G(7215305, J$.R(7215297, 'SearchQuery', SearchQuery, false, true), 'prototype'), 'timeout', J$.T(7215449, function (timeout) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(7215425, arguments.callee, this, arguments);
                            arguments = J$.N(7215433, 'arguments', arguments, true, false, false);
                            timeout = J$.N(7215441, 'timeout', timeout, true, false, false);
                            if (J$.C(386528, J$.U(726698, '!', J$.G(7215329, J$.G(7215321, J$.R(7215313, 'this', this, false, false), 'data'), 'ctl')))) {
                                J$.P(7215361, J$.G(7215345, J$.R(7215337, 'this', this, false, false), 'data'), 'ctl', J$.T(7215353, {}, 11, false));
                            }
                            J$.P(7215401, J$.G(7215385, J$.G(7215377, J$.R(7215369, 'this', this, false, false), 'data'), 'ctl'), 'timeout', J$.R(7215393, 'timeout', timeout, false, false));
                            return J$.Rt(7215417, J$.R(7215409, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7215969, J$e);
                        } finally {
                            if (J$.Fr(7215977))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7215529, J$.G(7215473, J$.R(7215465, 'SearchQuery', SearchQuery, false, true), 'prototype'), 'toJSON', J$.T(7215521, function () {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(7215505, arguments.callee, this, arguments);
                            arguments = J$.N(7215513, 'arguments', arguments, true, false, false);
                            return J$.Rt(7215497, J$.G(7215489, J$.R(7215481, 'this', this, false, false), 'data'));
                        } catch (J$e) {
                            J$.Ex(7215985, J$e);
                        } finally {
                            if (J$.Fr(7215993))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7215625, J$.R(7215537, 'SearchQuery', SearchQuery, false, true), 'new', J$.T(7215617, function (indexName, query) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(7215585, arguments.callee, this, arguments);
                            arguments = J$.N(7215593, 'arguments', arguments, true, false, false);
                            indexName = J$.N(7215601, 'indexName', indexName, true, false, false);
                            query = J$.N(7215609, 'query', query, true, false, false);
                            return J$.Rt(7215577, J$.F(7215569, J$.R(7215545, 'SearchQuery', SearchQuery, false, true), true)(J$.R(7215553, 'indexName', indexName, false, false), J$.R(7215561, 'query', query, false, false)));
                        } catch (J$e) {
                            J$.Ex(7216001, J$e);
                        } finally {
                            if (J$.Fr(7216009))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            for (var i in J$.H(7215713, J$.R(7215633, 'queryProtos', queryProtos, false, true))) {
                J$.N(7215721, 'i', i, false, true, false);
                {
                    {
                        if (J$.C(386536, J$.M(7215657, J$.R(7215641, 'queryProtos', queryProtos, false, true), 'hasOwnProperty', false)(J$.R(7215649, 'i', i, false, true)))) {
                            J$.P(7215705, J$.R(7215665, 'SearchQuery', SearchQuery, false, true), J$.R(7215673, 'i', i, false, true), J$.G(7215697, J$.R(7215681, 'queryProtos', queryProtos, false, true), J$.R(7215689, 'i', i, false, true)));
                        }
                    }
                }
            }
            J$.P(7215745, J$.I(typeof module === 'undefined' ? module = J$.R(7215729, 'module', undefined, true, true) : module = J$.R(7215729, 'module', module, true, true)), 'exports', J$.R(7215737, 'SearchQuery', SearchQuery, false, true));
        } catch (J$e) {
            J$.Ex(7216017, J$e);
        } finally {
            if (J$.Sr(7216025))
                continue jalangiLabel13;
            else
                break jalangiLabel13;
        }
    }
// JALANGI DO NOT INSTRUMENT
