// couchbase/lib/searchquery_facets.js
J$.noInstrEval = false;
jalangiLabel8:
    while (true) {
        try {
            J$.Se(7217025, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/couchbase/lib/searchquery_facets.js');
            function TermFacet(field, size) {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(7216089, arguments.callee, this, arguments);
                            arguments = J$.N(7216097, 'arguments', arguments, true, false, false);
                            field = J$.N(7216105, 'field', field, true, false, false);
                            size = J$.N(7216113, 'size', size, true, false, false);
                            J$.P(7216057, J$.R(7216041, 'this', this, false, false), 'field', J$.R(7216049, 'field', field, false, false));
                            J$.P(7216081, J$.R(7216065, 'this', this, false, false), 'size', J$.R(7216073, 'size', size, false, false));
                        } catch (J$e) {
                            J$.Ex(7217081, J$e);
                        } finally {
                            if (J$.Fr(7217089))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function NumericFacet(field, size) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(7216329, arguments.callee, this, arguments);
                            arguments = J$.N(7216337, 'arguments', arguments, true, false, false);
                            field = J$.N(7216345, 'field', field, true, false, false);
                            size = J$.N(7216353, 'size', size, true, false, false);
                            J$.P(7216273, J$.R(7216257, 'this', this, false, false), 'field', J$.R(7216265, 'field', field, false, false));
                            J$.P(7216297, J$.R(7216281, 'this', this, false, false), 'size', J$.R(7216289, 'size', size, false, false));
                            J$.P(7216321, J$.R(7216305, 'this', this, false, false), 'numeric_ranges', J$.T(7216313, [], 10, false));
                        } catch (J$e) {
                            J$.Ex(7217113, J$e);
                        } finally {
                            if (J$.Fr(7217121))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function DateFacet(field, size) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(7216713, arguments.callee, this, arguments);
                            arguments = J$.N(7216721, 'arguments', arguments, true, false, false);
                            field = J$.N(7216729, 'field', field, true, false, false);
                            size = J$.N(7216737, 'size', size, true, false, false);
                            J$.P(7216657, J$.R(7216641, 'this', this, false, false), 'field', J$.R(7216649, 'field', field, false, false));
                            J$.P(7216681, J$.R(7216665, 'this', this, false, false), 'size', J$.R(7216673, 'size', size, false, false));
                            J$.P(7216705, J$.R(7216689, 'this', this, false, false), 'date_ranges', J$.T(7216697, [], 10, false));
                        } catch (J$e) {
                            J$.Ex(7217161, J$e);
                        } finally {
                            if (J$.Fr(7217169))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            TermFacet = J$.N(7217041, 'TermFacet', J$.T(7217033, TermFacet, 12, false), true, false, false);
            NumericFacet = J$.N(7217057, 'NumericFacet', J$.T(7217049, NumericFacet, 12, false), true, false, false);
            DateFacet = J$.N(7217073, 'DateFacet', J$.T(7217065, DateFacet, 12, false), true, false, false);
            J$.T(7216033, 'use strict', 21, false);
            J$.P(7216145, J$.G(7216129, J$.I(typeof module === 'undefined' ? module = J$.R(7216121, 'module', undefined, true, true) : module = J$.R(7216121, 'module', module, true, true)), 'exports'), 'TermFacet', J$.R(7216137, 'TermFacet', TermFacet, false, true));
            J$.P(7216249, J$.G(7216161, J$.I(typeof module === 'undefined' ? module = J$.R(7216153, 'module', undefined, true, true) : module = J$.R(7216153, 'module', module, true, true)), 'exports'), 'term', J$.T(7216241, function (field, size) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7216209, arguments.callee, this, arguments);
                            arguments = J$.N(7216217, 'arguments', arguments, true, false, false);
                            field = J$.N(7216225, 'field', field, true, false, false);
                            size = J$.N(7216233, 'size', size, true, false, false);
                            return J$.Rt(7216201, J$.F(7216193, J$.R(7216169, 'TermFacet', TermFacet, false, true), true)(J$.R(7216177, 'field', field, false, false), J$.R(7216185, 'size', size, false, false)));
                        } catch (J$e) {
                            J$.Ex(7217097, J$e);
                        } finally {
                            if (J$.Fr(7217105))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7216385, J$.G(7216369, J$.I(typeof module === 'undefined' ? module = J$.R(7216361, 'module', undefined, true, true) : module = J$.R(7216361, 'module', module, true, true)), 'exports'), 'NumericFacet', J$.R(7216377, 'NumericFacet', NumericFacet, false, true));
            J$.P(7216489, J$.G(7216401, J$.I(typeof module === 'undefined' ? module = J$.R(7216393, 'module', undefined, true, true) : module = J$.R(7216393, 'module', module, true, true)), 'exports'), 'numeric', J$.T(7216481, function (field, size) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(7216449, arguments.callee, this, arguments);
                            arguments = J$.N(7216457, 'arguments', arguments, true, false, false);
                            field = J$.N(7216465, 'field', field, true, false, false);
                            size = J$.N(7216473, 'size', size, true, false, false);
                            return J$.Rt(7216441, J$.F(7216433, J$.R(7216409, 'NumericFacet', NumericFacet, false, true), true)(J$.R(7216417, 'field', field, false, false), J$.R(7216425, 'size', size, false, false)));
                        } catch (J$e) {
                            J$.Ex(7217129, J$e);
                        } finally {
                            if (J$.Fr(7217137))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7216633, J$.G(7216505, J$.R(7216497, 'NumericFacet', NumericFacet, false, true), 'prototype'), 'addRange', J$.T(7216625, function (name, min, max) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7216585, arguments.callee, this, arguments);
                            arguments = J$.N(7216593, 'arguments', arguments, true, false, false);
                            name = J$.N(7216601, 'name', name, true, false, false);
                            min = J$.N(7216609, 'min', min, true, false, false);
                            max = J$.N(7216617, 'max', max, true, false, false);
                            J$.M(7216561, J$.G(7216521, J$.R(7216513, 'this', this, false, false), 'numeric_ranges'), 'push', false)(J$.T(7216553, {
                                name: J$.R(7216529, 'name', name, false, false),
                                min: J$.R(7216537, 'min', min, false, false),
                                max: J$.R(7216545, 'max', max, false, false)
                            }, 11, false));
                            return J$.Rt(7216577, J$.R(7216569, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7217145, J$e);
                        } finally {
                            if (J$.Fr(7217153))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7216769, J$.G(7216753, J$.I(typeof module === 'undefined' ? module = J$.R(7216745, 'module', undefined, true, true) : module = J$.R(7216745, 'module', module, true, true)), 'exports'), 'DateFacet', J$.R(7216761, 'DateFacet', DateFacet, false, true));
            J$.P(7216873, J$.G(7216785, J$.I(typeof module === 'undefined' ? module = J$.R(7216777, 'module', undefined, true, true) : module = J$.R(7216777, 'module', module, true, true)), 'exports'), 'date', J$.T(7216865, function (field, size) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(7216833, arguments.callee, this, arguments);
                            arguments = J$.N(7216841, 'arguments', arguments, true, false, false);
                            field = J$.N(7216849, 'field', field, true, false, false);
                            size = J$.N(7216857, 'size', size, true, false, false);
                            return J$.Rt(7216825, J$.F(7216817, J$.R(7216793, 'DateFacet', DateFacet, false, true), true)(J$.R(7216801, 'field', field, false, false), J$.R(7216809, 'size', size, false, false)));
                        } catch (J$e) {
                            J$.Ex(7217177, J$e);
                        } finally {
                            if (J$.Fr(7217185))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7217017, J$.G(7216889, J$.R(7216881, 'DateFacet', DateFacet, false, true), 'prototype'), 'addRange', J$.T(7217009, function (name, start, end) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(7216969, arguments.callee, this, arguments);
                            arguments = J$.N(7216977, 'arguments', arguments, true, false, false);
                            name = J$.N(7216985, 'name', name, true, false, false);
                            start = J$.N(7216993, 'start', start, true, false, false);
                            end = J$.N(7217001, 'end', end, true, false, false);
                            J$.M(7216945, J$.G(7216905, J$.R(7216897, 'this', this, false, false), 'date_ranges'), 'push', false)(J$.T(7216937, {
                                name: J$.R(7216913, 'name', name, false, false),
                                start: J$.R(7216921, 'start', start, false, false),
                                end: J$.R(7216929, 'end', end, false, false)
                            }, 11, false));
                            return J$.Rt(7216961, J$.R(7216953, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7217193, J$e);
                        } finally {
                            if (J$.Fr(7217201))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(7217209, J$e);
        } finally {
            if (J$.Sr(7217217))
                continue jalangiLabel8;
            else
                break jalangiLabel8;
        }
    }
// JALANGI DO NOT INSTRUMENT
