// couchbase/lib/viewquery.js
J$.noInstrEval = false;
jalangiLabel18:
    while (true) {
        try {
            J$.Se(7235745, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/couchbase/lib/viewquery.js');
            function ViewQuery() {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(7232641, arguments.callee, this, arguments);
                            arguments = J$.N(7232649, 'arguments', arguments, true, false, false);
                            J$.P(7232561, J$.R(7232545, 'this', this, false, false), 'ddoc', J$.T(7232553, null, 25, false));
                            J$.P(7232585, J$.R(7232569, 'this', this, false, false), 'name', J$.T(7232577, null, 25, false));
                            J$.P(7232609, J$.R(7232593, 'this', this, false, false), 'options', J$.T(7232601, {}, 11, false));
                            J$.P(7232633, J$.R(7232617, 'this', this, false, false), 'postoptions', J$.T(7232625, {}, 11, false));
                        } catch (J$e) {
                            J$.Ex(7235785, J$e);
                        } finally {
                            if (J$.Fr(7235793))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(7235753, 'util', util, false, false, false);
            ViewQuery = J$.N(7235769, 'ViewQuery', J$.T(7235761, ViewQuery, 12, false), true, false, false);
            J$.N(7235777, 'SpatialQuery', SpatialQuery, false, false, false);
            J$.T(7232505, 'use strict', 21, false);
            var util = J$.W(7232537, 'util', J$.F(7232529, J$.I(typeof require === 'undefined' ? require = J$.R(7232513, 'require', undefined, true, true) : require = J$.R(7232513, 'require', require, true, true)), false)(J$.T(7232521, 'util', 21, false)), util, false, true);
            J$.P(7232697, J$.R(7232657, 'ViewQuery', ViewQuery, false, true), 'Update', J$.T(7232689, {
                BEFORE: J$.T(7232665, 1, 22, false),
                NONE: J$.T(7232673, 2, 22, false),
                AFTER: J$.T(7232681, 3, 22, false)
            }, 11, false));
            J$.P(7232737, J$.R(7232705, 'ViewQuery', ViewQuery, false, true), 'Order', J$.T(7232729, {
                ASCENDING: J$.T(7232713, 1, 22, false),
                DESCENDING: J$.T(7232721, 2, 22, false)
            }, 11, false));
            J$.P(7232777, J$.R(7232745, 'ViewQuery', ViewQuery, false, true), 'ErrorMode', J$.T(7232769, {
                CONTINUE: J$.T(7232753, 1, 22, false),
                STOP: J$.T(7232761, 2, 22, false)
            }, 11, false));
            J$.P(7232905, J$.G(7232793, J$.R(7232785, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'from', J$.T(7232897, function (ddoc, name) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7232865, arguments.callee, this, arguments);
                            arguments = J$.N(7232873, 'arguments', arguments, true, false, false);
                            ddoc = J$.N(7232881, 'ddoc', ddoc, true, false, false);
                            name = J$.N(7232889, 'name', name, true, false, false);
                            J$.P(7232817, J$.R(7232801, 'this', this, false, false), 'ddoc', J$.R(7232809, 'ddoc', ddoc, false, false));
                            J$.P(7232841, J$.R(7232825, 'this', this, false, false), 'name', J$.R(7232833, 'name', name, false, false));
                            return J$.Rt(7232857, J$.R(7232849, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235801, J$e);
                        } finally {
                            if (J$.Fr(7235809))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7233193, J$.G(7232921, J$.R(7232913, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'stale', J$.T(7233185, function (stale) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(7233161, arguments.callee, this, arguments);
                            arguments = J$.N(7233169, 'arguments', arguments, true, false, false);
                            stale = J$.N(7233177, 'stale', stale, true, false, false);
                            if (J$.C(386752, J$.B(726962, '===', J$.R(7232929, 'stale', stale, false, false), J$.G(7232953, J$.G(7232945, J$.R(7232937, 'ViewQuery', ViewQuery, false, true), 'Update'), 'BEFORE')))) {
                                J$.P(7232985, J$.G(7232969, J$.R(7232961, 'this', this, false, false), 'options'), 'stale', J$.T(7232977, 'false', 21, false));
                            } else if (J$.C(386744, J$.B(726970, '===', J$.R(7232993, 'stale', stale, false, false), J$.G(7233017, J$.G(7233009, J$.R(7233001, 'ViewQuery', ViewQuery, false, true), 'Update'), 'NONE')))) {
                                J$.P(7233049, J$.G(7233033, J$.R(7233025, 'this', this, false, false), 'options'), 'stale', J$.T(7233041, 'ok', 21, false));
                            } else if (J$.C(386736, J$.B(726978, '===', J$.R(7233057, 'stale', stale, false, false), J$.G(7233081, J$.G(7233073, J$.R(7233065, 'ViewQuery', ViewQuery, false, true), 'Update'), 'AFTER')))) {
                                J$.P(7233113, J$.G(7233097, J$.R(7233089, 'this', this, false, false), 'options'), 'stale', J$.T(7233105, 'update_after', 21, false));
                            } else {
                                throw J$.F(7233137, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(7233121, 'TypeError', undefined, true, true) : TypeError = J$.R(7233121, 'TypeError', TypeError, true, true)), true)(J$.T(7233129, 'invalid option passed.', 21, false));
                            }
                            return J$.Rt(7233153, J$.R(7233145, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235817, J$e);
                        } finally {
                            if (J$.Fr(7235825))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7233297, J$.G(7233209, J$.R(7233201, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'skip', J$.T(7233289, function (skip) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(7233265, arguments.callee, this, arguments);
                            arguments = J$.N(7233273, 'arguments', arguments, true, false, false);
                            skip = J$.N(7233281, 'skip', skip, true, false, false);
                            J$.P(7233241, J$.G(7233225, J$.R(7233217, 'this', this, false, false), 'options'), 'skip', J$.R(7233233, 'skip', skip, false, false));
                            return J$.Rt(7233257, J$.R(7233249, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235833, J$e);
                        } finally {
                            if (J$.Fr(7235841))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7233401, J$.G(7233313, J$.R(7233305, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'limit', J$.T(7233393, function (limit) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7233369, arguments.callee, this, arguments);
                            arguments = J$.N(7233377, 'arguments', arguments, true, false, false);
                            limit = J$.N(7233385, 'limit', limit, true, false, false);
                            J$.P(7233345, J$.G(7233329, J$.R(7233321, 'this', this, false, false), 'options'), 'limit', J$.R(7233337, 'limit', limit, false, false));
                            return J$.Rt(7233361, J$.R(7233353, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235849, J$e);
                        } finally {
                            if (J$.Fr(7235857))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7233585, J$.G(7233417, J$.R(7233409, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'custom', J$.T(7233577, function (opts) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(7233545, arguments.callee, this, arguments);
                            arguments = J$.N(7233553, 'arguments', arguments, true, false, false);
                            opts = J$.N(7233561, 'opts', opts, true, false, false);
                            J$.N(7233569, 'i', i, false, false, false);
                            for (var i in J$.H(7233513, J$.R(7233425, 'opts', opts, false, false))) {
                                J$.N(7233521, 'i', i, false, true, false);
                                {
                                    {
                                        if (J$.C(386760, J$.M(7233449, J$.R(7233433, 'opts', opts, false, false), 'hasOwnProperty', false)(J$.R(7233441, 'i', i, false, false)))) {
                                            J$.P(7233505, J$.G(7233465, J$.R(7233457, 'this', this, false, false), 'options'), J$.R(7233473, 'i', i, false, false), J$.G(7233497, J$.R(7233481, 'opts', opts, false, false), J$.R(7233489, 'i', i, false, false)));
                                        }
                                    }
                                }
                            }
                            return J$.Rt(7233537, J$.R(7233529, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235865, J$e);
                        } finally {
                            if (J$.Fr(7235873))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7233809, J$.G(7233601, J$.R(7233593, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'order', J$.T(7233801, function (order) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(7233777, arguments.callee, this, arguments);
                            arguments = J$.N(7233785, 'arguments', arguments, true, false, false);
                            order = J$.N(7233793, 'order', order, true, false, false);
                            if (J$.C(386776, J$.B(726986, '===', J$.R(7233609, 'order', order, false, false), J$.G(7233633, J$.G(7233625, J$.R(7233617, 'ViewQuery', ViewQuery, false, true), 'Order'), 'ASCENDING')))) {
                                J$.P(7233665, J$.G(7233649, J$.R(7233641, 'this', this, false, false), 'options'), 'descending', J$.T(7233657, false, 23, false));
                            } else if (J$.C(386768, J$.B(726994, '===', J$.R(7233673, 'order', order, false, false), J$.G(7233697, J$.G(7233689, J$.R(7233681, 'ViewQuery', ViewQuery, false, true), 'Order'), 'DESCENDING')))) {
                                J$.P(7233729, J$.G(7233713, J$.R(7233705, 'this', this, false, false), 'options'), 'descending', J$.T(7233721, true, 23, false));
                            } else {
                                throw J$.F(7233753, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(7233737, 'TypeError', undefined, true, true) : TypeError = J$.R(7233737, 'TypeError', TypeError, true, true)), true)(J$.T(7233745, 'invalid option passed.', 21, false));
                            }
                            return J$.Rt(7233769, J$.R(7233761, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235881, J$e);
                        } finally {
                            if (J$.Fr(7235889))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7233913, J$.G(7233825, J$.R(7233817, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'reduce', J$.T(7233905, function (reduce) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(7233881, arguments.callee, this, arguments);
                            arguments = J$.N(7233889, 'arguments', arguments, true, false, false);
                            reduce = J$.N(7233897, 'reduce', reduce, true, false, false);
                            J$.P(7233857, J$.G(7233841, J$.R(7233833, 'this', this, false, false), 'options'), 'reduce', J$.R(7233849, 'reduce', reduce, false, false));
                            return J$.Rt(7233873, J$.R(7233865, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235897, J$e);
                        } finally {
                            if (J$.Fr(7235905))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7234241, J$.G(7233929, J$.R(7233921, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'group', J$.T(7234233, function (group) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(7234209, arguments.callee, this, arguments);
                            arguments = J$.N(7234217, 'arguments', arguments, true, false, false);
                            group = J$.N(7234225, 'group', group, true, false, false);
                            if (J$.C(386808, J$.C(386784, J$.B(727002, '===', J$.R(7233937, 'group', group, false, false), J$.T(7233945, undefined, 24, false))) ? J$._() : J$.B(727010, '===', J$.R(7233953, 'group', group, false, false), J$.T(7233961, true, 23, false)))) {
                                J$.P(7233993, J$.G(7233977, J$.R(7233969, 'this', this, false, false), 'options'), 'group', J$.T(7233985, true, 23, false));
                            } else if (J$.C(386800, J$.B(727018, '===', J$.R(7234001, 'group', group, false, false), J$.T(7234009, false, 23, false)))) {
                                J$.P(7234041, J$.G(7234025, J$.R(7234017, 'this', this, false, false), 'options'), 'group', J$.T(7234033, false, 23, false));
                            } else if (J$.C(386792, J$.B(727026, '>=', J$.R(7234049, 'group', group, false, false), J$.T(7234057, 0, 22, false)))) {
                                J$.P(7234089, J$.G(7234073, J$.R(7234065, 'this', this, false, false), 'options'), 'group', J$.T(7234081, false, 23, false));
                                J$.P(7234121, J$.G(7234105, J$.R(7234097, 'this', this, false, false), 'options'), 'group_level', J$.R(7234113, 'group', group, false, false));
                            } else {
                                J$.P(7234153, J$.G(7234137, J$.R(7234129, 'this', this, false, false), 'options'), 'group', J$.T(7234145, true, 23, false));
                                J$.P(7234185, J$.G(7234169, J$.R(7234161, 'this', this, false, false), 'options'), 'group_level', J$.T(7234177, 0, 22, false));
                            }
                            return J$.Rt(7234201, J$.R(7234193, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235913, J$e);
                        } finally {
                            if (J$.Fr(7235921))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7234377, J$.G(7234257, J$.R(7234249, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'group_level', J$.T(7234369, function (group_level) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(7234345, arguments.callee, this, arguments);
                            arguments = J$.N(7234353, 'arguments', arguments, true, false, false);
                            group_level = J$.N(7234361, 'group_level', group_level, true, false, false);
                            if (J$.C(386816, J$.B(727034, '!==', J$.R(7234265, 'group_level', group_level, false, false), J$.T(7234273, undefined, 24, false)))) {
                                J$.P(7234305, J$.G(7234289, J$.R(7234281, 'this', this, false, false), 'options'), 'group_level', J$.R(7234297, 'group_level', group_level, false, false));
                            } else {
                                J$.B(727042, 'delete', J$.G(7234321, J$.R(7234313, 'this', this, false, false), 'options'), 'group_level');
                            }
                            return J$.Rt(7234337, J$.R(7234329, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235929, J$e);
                        } finally {
                            if (J$.Fr(7235937))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7234497, J$.G(7234393, J$.R(7234385, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'key', J$.T(7234489, function (key) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(7234465, arguments.callee, this, arguments);
                            arguments = J$.N(7234473, 'arguments', arguments, true, false, false);
                            key = J$.N(7234481, 'key', key, true, false, false);
                            J$.P(7234441, J$.G(7234409, J$.R(7234401, 'this', this, false, false), 'options'), 'key', J$.M(7234433, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7234417, 'JSON', undefined, true, true) : JSON = J$.R(7234417, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(7234425, 'key', key, false, false)));
                            return J$.Rt(7234457, J$.R(7234449, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235945, J$e);
                        } finally {
                            if (J$.Fr(7235953))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7234601, J$.G(7234513, J$.R(7234505, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'keys', J$.T(7234593, function (keys) {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(7234569, arguments.callee, this, arguments);
                            arguments = J$.N(7234577, 'arguments', arguments, true, false, false);
                            keys = J$.N(7234585, 'keys', keys, true, false, false);
                            J$.P(7234545, J$.G(7234529, J$.R(7234521, 'this', this, false, false), 'postoptions'), 'keys', J$.R(7234537, 'keys', keys, false, false));
                            return J$.Rt(7234561, J$.R(7234553, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235961, J$e);
                        } finally {
                            if (J$.Fr(7235969))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7234841, J$.G(7234617, J$.R(7234609, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'range', J$.T(7234833, function (start, end, inclusive_end) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(7234793, arguments.callee, this, arguments);
                            arguments = J$.N(7234801, 'arguments', arguments, true, false, false);
                            start = J$.N(7234809, 'start', start, true, false, false);
                            end = J$.N(7234817, 'end', end, true, false, false);
                            inclusive_end = J$.N(7234825, 'inclusive_end', inclusive_end, true, false, false);
                            J$.P(7234665, J$.G(7234633, J$.R(7234625, 'this', this, false, false), 'options'), 'startkey', J$.M(7234657, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7234641, 'JSON', undefined, true, true) : JSON = J$.R(7234641, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(7234649, 'start', start, false, false)));
                            J$.P(7234713, J$.G(7234681, J$.R(7234673, 'this', this, false, false), 'options'), 'endkey', J$.M(7234705, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7234689, 'JSON', undefined, true, true) : JSON = J$.R(7234689, 'JSON', JSON, true, true)), 'stringify', false)(J$.R(7234697, 'end', end, false, false)));
                            if (J$.C(386824, J$.R(7234721, 'inclusive_end', inclusive_end, false, false))) {
                                J$.P(7234753, J$.G(7234737, J$.R(7234729, 'this', this, false, false), 'options'), 'inclusive_end', J$.T(7234745, 'true', 21, false));
                            } else {
                                J$.B(727050, 'delete', J$.G(7234769, J$.R(7234761, 'this', this, false, false), 'options'), 'inclusive_end');
                            }
                            return J$.Rt(7234785, J$.R(7234777, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235977, J$e);
                        } finally {
                            if (J$.Fr(7235985))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7234985, J$.G(7234857, J$.R(7234849, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'id_range', J$.T(7234977, function (start, end) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(7234945, arguments.callee, this, arguments);
                            arguments = J$.N(7234953, 'arguments', arguments, true, false, false);
                            start = J$.N(7234961, 'start', start, true, false, false);
                            end = J$.N(7234969, 'end', end, true, false, false);
                            J$.P(7234889, J$.G(7234873, J$.R(7234865, 'this', this, false, false), 'options'), 'startkey_docid', J$.R(7234881, 'start', start, false, false));
                            J$.P(7234921, J$.G(7234905, J$.R(7234897, 'this', this, false, false), 'options'), 'endkey_docid', J$.R(7234913, 'end', end, false, false));
                            return J$.Rt(7234937, J$.R(7234929, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7235993, J$e);
                        } finally {
                            if (J$.Fr(7236001))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7235129, J$.G(7235001, J$.R(7234993, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'include_docs', J$.T(7235121, function (include_docs) {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(7235097, arguments.callee, this, arguments);
                            arguments = J$.N(7235105, 'arguments', arguments, true, false, false);
                            include_docs = J$.N(7235113, 'include_docs', include_docs, true, false, false);
                            if (J$.C(386840, J$.C(386832, J$.B(727058, '===', J$.R(7235009, 'include_docs', include_docs, false, false), J$.T(7235017, undefined, 24, false))) ? J$._() : J$.R(7235025, 'include_docs', include_docs, false, false))) {
                                J$.P(7235057, J$.G(7235041, J$.R(7235033, 'this', this, false, false), 'options'), 'include_docs', J$.T(7235049, 'true', 21, false));
                            } else {
                                J$.B(727066, 'delete', J$.G(7235073, J$.R(7235065, 'this', this, false, false), 'options'), 'include_docs');
                            }
                            return J$.Rt(7235089, J$.R(7235081, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7236009, J$e);
                        } finally {
                            if (J$.Fr(7236017))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7235273, J$.G(7235145, J$.R(7235137, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'full_set', J$.T(7235265, function (full_set) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(7235241, arguments.callee, this, arguments);
                            arguments = J$.N(7235249, 'arguments', arguments, true, false, false);
                            full_set = J$.N(7235257, 'full_set', full_set, true, false, false);
                            if (J$.C(386856, J$.C(386848, J$.B(727074, '===', J$.R(7235153, 'full_set', full_set, false, false), J$.T(7235161, undefined, 24, false))) ? J$._() : J$.R(7235169, 'full_set', full_set, false, false))) {
                                J$.P(7235201, J$.G(7235185, J$.R(7235177, 'this', this, false, false), 'options'), 'full_set', J$.T(7235193, 'true', 21, false));
                            } else {
                                J$.B(727082, 'delete', J$.G(7235217, J$.R(7235209, 'this', this, false, false), 'options'), 'full_set');
                            }
                            return J$.Rt(7235233, J$.R(7235225, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7236025, J$e);
                        } finally {
                            if (J$.Fr(7236033))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7235497, J$.G(7235289, J$.R(7235281, 'ViewQuery', ViewQuery, false, true), 'prototype'), 'on_error', J$.T(7235489, function (mode) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(7235465, arguments.callee, this, arguments);
                            arguments = J$.N(7235473, 'arguments', arguments, true, false, false);
                            mode = J$.N(7235481, 'mode', mode, true, false, false);
                            if (J$.C(386872, J$.B(727090, '===', J$.R(7235297, 'mode', mode, false, false), J$.G(7235321, J$.G(7235313, J$.R(7235305, 'ViewQuery', ViewQuery, false, true), 'ErrorMode'), 'CONTINUE')))) {
                                J$.P(7235353, J$.G(7235337, J$.R(7235329, 'this', this, false, false), 'options'), 'on_error', J$.T(7235345, 'continue', 21, false));
                            } else if (J$.C(386864, J$.B(727098, '===', J$.R(7235361, 'mode', mode, false, false), J$.G(7235385, J$.G(7235377, J$.R(7235369, 'ViewQuery', ViewQuery, false, true), 'ErrorMode'), 'STOP')))) {
                                J$.P(7235417, J$.G(7235401, J$.R(7235393, 'this', this, false, false), 'options'), 'on_error', J$.T(7235409, 'stop', 21, false));
                            } else {
                                throw J$.F(7235441, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(7235425, 'TypeError', undefined, true, true) : TypeError = J$.R(7235425, 'TypeError', TypeError, true, true)), true)(J$.T(7235433, 'invalid option passed.', 21, false));
                            }
                            return J$.Rt(7235457, J$.R(7235449, 'this', this, false, false));
                        } catch (J$e) {
                            J$.Ex(7236041, J$e);
                        } finally {
                            if (J$.Fr(7236049))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7235601, J$.R(7235505, 'ViewQuery', ViewQuery, false, true), 'from', J$.T(7235593, function (ddoc, name) {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(7235561, arguments.callee, this, arguments);
                            arguments = J$.N(7235569, 'arguments', arguments, true, false, false);
                            ddoc = J$.N(7235577, 'ddoc', ddoc, true, false, false);
                            name = J$.N(7235585, 'name', name, true, false, false);
                            return J$.Rt(7235553, J$.M(7235545, J$.F(7235521, J$.R(7235513, 'ViewQuery', ViewQuery, false, true), true)(), 'from', false)(J$.R(7235529, 'ddoc', ddoc, false, false), J$.R(7235537, 'name', name, false, false)));
                        } catch (J$e) {
                            J$.Ex(7236057, J$e);
                        } finally {
                            if (J$.Fr(7236065))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var SpatialQuery = J$.W(7235633, 'SpatialQuery', J$.F(7235625, J$.I(typeof require === 'undefined' ? require = J$.R(7235609, 'require', undefined, true, true) : require = J$.R(7235609, 'require', require, true, true)), false)(J$.T(7235617, './spatialquery', 21, false)), SpatialQuery, false, true);
            J$.P(7235665, J$.R(7235641, 'ViewQuery', ViewQuery, false, true), 'fromSpatial', J$.G(7235657, J$.R(7235649, 'SpatialQuery', SpatialQuery, false, true), 'from'));
            J$.P(7235689, J$.R(7235673, 'ViewQuery', ViewQuery, false, true), 'Default', J$.R(7235681, 'ViewQuery', ViewQuery, false, true));
            J$.P(7235713, J$.R(7235697, 'ViewQuery', ViewQuery, false, true), 'Spatial', J$.R(7235705, 'SpatialQuery', SpatialQuery, false, true));
            J$.P(7235737, J$.I(typeof module === 'undefined' ? module = J$.R(7235721, 'module', undefined, true, true) : module = J$.R(7235721, 'module', module, true, true)), 'exports', J$.R(7235729, 'ViewQuery', ViewQuery, false, true));
        } catch (J$e) {
            J$.Ex(7236073, J$e);
        } finally {
            if (J$.Sr(7236081))
                continue jalangiLabel18;
            else
                break jalangiLabel18;
        }
    }
// JALANGI DO NOT INSTRUMENT
