// bungle/lib/plugins/auto-prefixer.js
import autoprefixer from 'autoprefixer';
import postcss from 'postcss';
import { ModPipe } from '../pipe';


export default class ExtPipe extends ModPipe {
    constructor() {
        super(...arguments);

        this.defaultConfig({
            pattern: '**/*.css',
            browsers: ['last 2 versions']
        });

        this.ap = postcss([
            autoprefixer({
                browsers: this.config.browsers
            })
        ]);
    }

    static schema() {
        return {
            description: 'Add vendor prefixes to css.',
            properties: {
                browsers: {
                    description: 'Browser versions to support ' +
                                 '(default: [\'last 2 versions\'])',
                    type: 'array',
                    items: {
                        type: 'string',
                        minItems: 1,
                        uniqueItems: true
                    }
                }
            }
        };
    }

    async generate(file) {
        const source = await file.source.content();
        return (await this.ap.process(source)).css;
    }
}
