// mongodb-core/lib/sdam/server_selectors.js
'use strict';

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

var ServerType = require('./server_description').ServerType;

var TopologyType = require('./topology_description').TopologyType;

var ReadPreference = require('../topologies/read_preference');

var MongoError = require('../error').MongoError; // max staleness constants


var IDLE_WRITE_PERIOD = 10000;
var SMALLEST_MAX_STALENESS_SECONDS = 90;

function writableServerSelector() {
  return function (topologyDescription, servers) {
    return latencyWindowReducer(topologyDescription, servers.filter(function (s) {
      return s.isWritable;
    }));
  };
} // reducers


function maxStalenessReducer(readPreference, topologyDescription, servers) {
  if (readPreference.maxStalenessSeconds == null || readPreference.maxStalenessSeconds < 0) {
    return servers;
  }

  var maxStaleness = readPreference.maxStalenessSeconds;
  var maxStalenessVariance = (topologyDescription.heartbeatFrequencyMS + IDLE_WRITE_PERIOD) / 1000;

  if (maxStaleness < maxStalenessVariance) {
    throw MongoError("maxStalenessSeconds must be at least ".concat(maxStalenessVariance, " seconds"));
  }

  if (maxStaleness < SMALLEST_MAX_STALENESS_SECONDS) {
    throw new MongoError("maxStalenessSeconds must be at least ".concat(SMALLEST_MAX_STALENESS_SECONDS, " seconds"));
  }

  if (topologyDescription.type === TopologyType.ReplicaSetWithPrimary) {
    var primary = servers.filter(primaryFilter)[0];
    return servers.reduce(function (result, server) {
      var stalenessMS = server.lastUpdateTime - server.lastWriteDate - (primary.lastUpdateTime - primary.lastWriteDate) + topologyDescription.heartbeatFrequencyMS;
      var staleness = stalenessMS / 1000;
      if (staleness <= readPreference.maxStalenessSeconds) result.push(server);
      return result;
    }, []);
  } else if (topologyDescription.type === TopologyType.ReplicaSetNoPrimary) {
    var sMax = servers.reduce(function (max, s) {
      return s.lastWriteDate > max.lastWriteDate ? s : max;
    });
    return servers.reduce(function (result, server) {
      var stalenessMS = sMax.lastWriteDate - server.lastWriteDate + topologyDescription.heartbeatFrequencyMS;
      var staleness = stalenessMS / 1000;
      if (staleness <= readPreference.maxStalenessSeconds) result.push(server);
      return result;
    }, []);
  }

  return servers;
}

function tagSetMatch(tagSet, serverTags) {
  var keys = Object.keys(tagSet);
  var serverTagKeys = Object.keys(serverTags);

  for (var i = 0; i < keys.length; ++i) {
    var key = keys[i];

    if (serverTagKeys.indexOf(key) === -1 || serverTags[key] !== tagSet[key]) {
      return false;
    }
  }

  return true;
}

function tagSetReducer(readPreference, servers) {
  if (readPreference.tags == null || Array.isArray(readPreference.tags) && readPreference.tags.length === 0) {
    return servers;
  }

  var _loop = function _loop(i) {
    var tagSet = readPreference.tags[i];
    var serversMatchingTagset = servers.reduce(function (matched, server) {
      if (tagSetMatch(tagSet, server.tags)) matched.push(server);
      return matched;
    }, []);

    if (serversMatchingTagset.length) {
      return {
        v: serversMatchingTagset
      };
    }
  };

  for (var i = 0; i < readPreference.tags.length; ++i) {
    var _ret = _loop(i);

    if (_typeof(_ret) === "object") return _ret.v;
  }

  return [];
}

function latencyWindowReducer(topologyDescription, servers) {
  var low = servers.reduce(function (min, server) {
    return min === -1 ? server.roundTripTime : Math.min(server.roundTripTime, min);
  }, -1);
  var high = low + topologyDescription.localThresholdMS;
  return servers.reduce(function (result, server) {
    if (server.roundTripTime <= high && server.roundTripTime >= low) result.push(server);
    return result;
  }, []);
} // filters


function primaryFilter(server) {
  return server.type === ServerType.RSPrimary;
}

function secondaryFilter(server) {
  return server.type === ServerType.RSSecondary;
}

function nearestFilter(server) {
  return server.type === ServerType.RSSecondary || server.type === ServerType.RSPrimary;
}

function knownFilter(server) {
  return server.type !== ServerType.Unknown;
}

function readPreferenceServerSelector(readPreference) {
  if (!readPreference.isValid()) {
    throw new TypeError('Invalid read preference specified');
  }

  return function (topologyDescription, servers) {
    var commonWireVersion = topologyDescription.commonWireVersion;

    if (commonWireVersion && readPreference.minWireVersion && readPreference.minWireVersion > commonWireVersion) {
      throw new MongoError("Minimum wire version '".concat(readPreference.minWireVersion, "' required, but found '").concat(commonWireVersion, "'"));
    }

    if (topologyDescription.type === TopologyType.Single || topologyDescription.type === TopologyType.Sharded) {
      return latencyWindowReducer(topologyDescription, servers.filter(knownFilter));
    }

    if (readPreference.mode === ReadPreference.PRIMARY) {
      return servers.filter(primaryFilter);
    }

    if (readPreference.mode === ReadPreference.SECONDARY) {
      return latencyWindowReducer(topologyDescription, tagSetReducer(readPreference, maxStalenessReducer(readPreference, topologyDescription, servers))).filter(secondaryFilter);
    } else if (readPreference.mode === ReadPreference.NEAREST) {
      return latencyWindowReducer(topologyDescription, tagSetReducer(readPreference, maxStalenessReducer(readPreference, topologyDescription, servers))).filter(nearestFilter);
    } else if (readPreference.mode === ReadPreference.SECONDARY_PREFERRED) {
      var result = latencyWindowReducer(topologyDescription, tagSetReducer(readPreference, maxStalenessReducer(readPreference, topologyDescription, servers))).filter(secondaryFilter);
      return result.length === 0 ? servers.filter(primaryFilter) : result;
    } else if (readPreference.mode === ReadPreference.PRIMARY_PREFERRED) {
      var _result = servers.filter(primaryFilter);

      if (_result.length) {
        return _result;
      }

      return latencyWindowReducer(topologyDescription, tagSetReducer(readPreference, maxStalenessReducer(readPreference, topologyDescription, servers))).filter(secondaryFilter);
    }
  };
}

module.exports = {
  writableServerSelector: writableServerSelector,
  readPreferenceServerSelector: readPreferenceServerSelector
};
