// pino/pino.js
'use strict';

var os = require('os');

var stdSerializers = require('pino-std-serializers');

var redaction = require('./lib/redaction');

var time = require('./lib/time');

var proto = require('./lib/proto');

var symbols = require('./lib/symbols');

var _require = require('./lib/levels'),
    assertDefaultLevelFound = _require.assertDefaultLevelFound,
    mappings = _require.mappings,
    genLsCache = _require.genLsCache;

var _require2 = require('./lib/tools'),
    createArgsNormalizer = _require2.createArgsNormalizer,
    asChindings = _require2.asChindings,
    final = _require2.final,
    stringify = _require2.stringify,
    buildSafeSonicBoom = _require2.buildSafeSonicBoom;

var _require3 = require('./lib/meta'),
    version = _require3.version,
    LOG_VERSION = _require3.LOG_VERSION;

var chindingsSym = symbols.chindingsSym,
    redactFmtSym = symbols.redactFmtSym,
    serializersSym = symbols.serializersSym,
    timeSym = symbols.timeSym,
    timeSliceIndexSym = symbols.timeSliceIndexSym,
    streamSym = symbols.streamSym,
    stringifySym = symbols.stringifySym,
    stringifiersSym = symbols.stringifiersSym,
    setLevelSym = symbols.setLevelSym,
    endSym = symbols.endSym,
    formatOptsSym = symbols.formatOptsSym,
    messageKeySym = symbols.messageKeySym,
    useLevelLabelsSym = symbols.useLevelLabelsSym,
    changeLevelNameSym = symbols.changeLevelNameSym,
    useOnlyCustomLevelsSym = symbols.useOnlyCustomLevelsSym;
var epochTime = time.epochTime,
    nullTime = time.nullTime;
var _process = process,
    pid = _process.pid;
var hostname = os.hostname();
var defaultErrorSerializer = stdSerializers.err;
var defaultOptions = {
  level: 'info',
  useLevelLabels: false,
  messageKey: 'msg',
  enabled: true,
  prettyPrint: false,
  base: {
    pid,
    hostname
  },
  serializers: Object.assign(Object.create(null), {
    err: defaultErrorSerializer
  }),
  timestamp: epochTime,
  name: undefined,
  redact: null,
  customLevels: null,
  changeLevelName: 'level',
  useOnlyCustomLevels: false
};
var normalize = createArgsNormalizer(defaultOptions);
var serializers = Object.assign(Object.create(null), stdSerializers);

function pino() {
  var _normalize = normalize(...arguments),
      opts = _normalize.opts,
      stream = _normalize.stream;

  var redact = opts.redact,
      crlf = opts.crlf,
      serializers = opts.serializers,
      timestamp = opts.timestamp,
      messageKey = opts.messageKey,
      base = opts.base,
      name = opts.name,
      level = opts.level,
      customLevels = opts.customLevels,
      useLevelLabels = opts.useLevelLabels,
      changeLevelName = opts.changeLevelName,
      useOnlyCustomLevels = opts.useOnlyCustomLevels;
  var stringifiers = redact ? redaction(redact, stringify) : {};
  var formatOpts = redact ? {
    stringify: stringifiers[redactFmtSym]
  } : {
    stringify
  };
  var end = ',"v":' + LOG_VERSION + '}' + (crlf ? '\r\n' : '\n');
  var coreChindings = asChindings.bind(null, {
    [chindingsSym]: '',
    [serializersSym]: serializers,
    [stringifiersSym]: stringifiers,
    [stringifySym]: stringify
  });
  var chindings = base === null ? '' : name === undefined ? coreChindings(base) : coreChindings(Object.assign({}, base, {
    name
  }));
  var time = timestamp instanceof Function ? timestamp : timestamp ? epochTime : nullTime;
  var timeSliceIndex = time().indexOf(':') + 1;
  if (useOnlyCustomLevels && !customLevels) throw Error('customLevels is required if useOnlyCustomLevels is set true');
  assertDefaultLevelFound(level, customLevels, useOnlyCustomLevels);
  var levels = mappings(customLevels, useOnlyCustomLevels);
  var instance = {
    levels,
    [useLevelLabelsSym]: useLevelLabels,
    [changeLevelNameSym]: changeLevelName,
    [useOnlyCustomLevelsSym]: useOnlyCustomLevels,
    [streamSym]: stream,
    [timeSym]: time,
    [timeSliceIndexSym]: timeSliceIndex,
    [stringifySym]: stringify,
    [stringifiersSym]: stringifiers,
    [endSym]: end,
    [formatOptsSym]: formatOpts,
    [messageKeySym]: messageKey,
    [serializersSym]: serializers,
    [chindingsSym]: chindings
  };
  Object.setPrototypeOf(instance, proto);
  if (customLevels || useLevelLabels || changeLevelName !== defaultOptions.changeLevelName) genLsCache(instance);
  instance[setLevelSym](level);
  return instance;
}

pino.extreme = function () {
  var dest = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : process.stdout.fd;
  return buildSafeSonicBoom(dest, 4096, false);
};

pino.destination = function () {
  var dest = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : process.stdout.fd;
  return buildSafeSonicBoom(dest, 0, true);
};

pino.final = final;
pino.levels = mappings();
pino.stdSerializers = serializers;
pino.stdTimeFunctions = Object.assign({}, time);
pino.symbols = symbols;
pino.version = version;
pino.LOG_VERSION = LOG_VERSION;
module.exports = pino;
