// gauge/index.js
J$.noInstrEval = false;
jalangiLabel21:
    while (true) {
        try {
            J$.Se(7426185, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/gauge/index.js');
            function callWith(obj, method) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7420577, arguments.callee, this, arguments);
                            arguments = J$.N(7420585, 'arguments', arguments, true, false, false);
                            obj = J$.N(7420593, 'obj', obj, true, false, false);
                            method = J$.N(7420601, 'method', method, true, false, false);
                            return J$.Rt(7420569, J$.T(7420561, function () {
                                jalangiLabel0:
                                    while (true) {
                                        try {
                                            J$.Fe(7420545, arguments.callee, this, arguments);
                                            arguments = J$.N(7420553, 'arguments', arguments, true, false, false);
                                            return J$.Rt(7420537, J$.M(7420529, J$.R(7420513, 'method', method, false, false), 'call', false)(J$.R(7420521, 'obj', obj, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(7426289, J$e);
                                        } finally {
                                            if (J$.Fr(7426297))
                                                continue jalangiLabel0;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7426305, J$e);
                        } finally {
                            if (J$.Fr(7426313))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function Gauge(arg1, arg2) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(7421905, arguments.callee, this, arguments);
                            arguments = J$.N(7421913, 'arguments', arguments, true, false, false);
                            arg1 = J$.N(7421921, 'arg1', arg1, true, false, false);
                            arg2 = J$.N(7421929, 'arg2', arg2, true, false, false);
                            J$.N(7421937, 'options', options, false, false, false);
                            J$.N(7421945, 'writeTo', writeTo, false, false, false);
                            J$.N(7421953, 'theme', theme, false, false, false);
                            J$.N(7421961, 'template', template, false, false, false);
                            J$.N(7421969, 'PlumbingClass', PlumbingClass, false, false, false);
                            var options, writeTo;
                            if (J$.C(394616, J$.C(394560, J$.R(7420609, 'arg1', arg1, false, false)) ? J$.G(7420625, J$.R(7420617, 'arg1', arg1, false, false), 'write') : J$._())) {
                                writeTo = J$.W(7420641, 'writeTo', J$.R(7420633, 'arg1', arg1, false, false), writeTo, false, false);
                                options = J$.W(7420665, 'options', J$.C(394568, J$.R(7420649, 'arg2', arg2, false, false)) ? J$._() : J$.T(7420657, {}, 11, false), options, false, false);
                            } else if (J$.C(394608, J$.C(394576, J$.R(7420673, 'arg2', arg2, false, false)) ? J$.G(7420689, J$.R(7420681, 'arg2', arg2, false, false), 'write') : J$._())) {
                                writeTo = J$.W(7420705, 'writeTo', J$.R(7420697, 'arg2', arg2, false, false), writeTo, false, false);
                                options = J$.W(7420729, 'options', J$.C(394584, J$.R(7420713, 'arg1', arg1, false, false)) ? J$._() : J$.T(7420721, {}, 11, false), options, false, false);
                            } else {
                                writeTo = J$.W(7420753, 'writeTo', J$.G(7420745, J$.R(7420737, 'process', process, false, true), 'stderr'), writeTo, false, false);
                                options = J$.W(7420785, 'options', J$.C(394600, J$.C(394592, J$.R(7420761, 'arg1', arg1, false, false)) ? J$._() : J$.R(7420769, 'arg2', arg2, false, false)) ? J$._() : J$.T(7420777, {}, 11, false), options, false, false);
                            }
                            J$.P(7420833, J$.R(7420793, 'this', this, false, false), '_status', J$.T(7420825, {
                                spun: J$.T(7420801, 0, 22, false),
                                section: J$.T(7420809, '', 21, false),
                                subsection: J$.T(7420817, '', 21, false)
                            }, 11, false));
                            J$.P(7420857, J$.R(7420841, 'this', this, false, false), '_paused', J$.T(7420849, false, 23, false));
                            J$.P(7420881, J$.R(7420865, 'this', this, false, false), '_disabled', J$.T(7420873, true, 23, false));
                            J$.P(7420905, J$.R(7420889, 'this', this, false, false), '_showing', J$.T(7420897, false, 23, false));
                            J$.P(7420929, J$.R(7420913, 'this', this, false, false), '_onScreen', J$.T(7420921, false, 23, false));
                            J$.P(7420953, J$.R(7420937, 'this', this, false, false), '_needsRedraw', J$.T(7420945, false, 23, false));
                            J$.P(7421017, J$.R(7420961, 'this', this, false, false), '_hideCursor', J$.C(394624, J$.B(740378, '==', J$.G(7420977, J$.R(7420969, 'options', options, false, false), 'hideCursor'), J$.T(7420985, null, 25, false))) ? J$.T(7420993, true, 23, false) : J$.G(7421009, J$.R(7421001, 'options', options, false, false), 'hideCursor'));
                            J$.P(7421105, J$.R(7421025, 'this', this, false, false), '_fixedFramerate', J$.C(394632, J$.B(740386, '==', J$.G(7421041, J$.R(7421033, 'options', options, false, false), 'fixedFramerate'), J$.T(7421049, null, 25, false))) ? J$.U(740394, '!', J$.M(7421081, J$.T(7421057, /^v0\.8\./, 14, false), 'test', false)(J$.G(7421073, J$.R(7421065, 'process', process, false, true), 'version'))) : J$.G(7421097, J$.R(7421089, 'options', options, false, false), 'fixedFramerate'));
                            J$.P(7421129, J$.R(7421113, 'this', this, false, false), '_lastUpdateAt', J$.T(7421121, null, 25, false));
                            J$.P(7421193, J$.R(7421137, 'this', this, false, false), '_updateInterval', J$.C(394640, J$.B(740402, '==', J$.G(7421153, J$.R(7421145, 'options', options, false, false), 'updateInterval'), J$.T(7421161, null, 25, false))) ? J$.T(7421169, 50, 22, false) : J$.G(7421185, J$.R(7421177, 'options', options, false, false), 'updateInterval'));
                            J$.P(7421233, J$.R(7421201, 'this', this, false, false), '_themes', J$.C(394648, J$.G(7421217, J$.R(7421209, 'options', options, false, false), 'themes')) ? J$._() : J$.R(7421225, 'defaultThemes', defaultThemes, false, true));
                            J$.P(7421265, J$.R(7421241, 'this', this, false, false), '_theme', J$.G(7421257, J$.R(7421249, 'options', options, false, false), 'theme'));
                            var theme = J$.W(7421305, 'theme', J$.M(7421297, J$.R(7421273, 'this', this, false, false), '_computeTheme', false)(J$.G(7421289, J$.R(7421281, 'options', options, false, false), 'theme')), theme, false, false);
                            var template = J$.W(7421457, 'template', J$.C(394656, J$.G(7421321, J$.R(7421313, 'options', options, false, false), 'template')) ? J$._() : J$.T(7421449, [
                                J$.T(7421345, {
                                    type: J$.T(7421329, 'progressbar', 21, false),
                                    length: J$.T(7421337, 20, 22, false)
                                }, 11, false),
                                J$.T(7421377, {
                                    type: J$.T(7421353, 'activityIndicator', 21, false),
                                    kerning: J$.T(7421361, 1, 22, false),
                                    length: J$.T(7421369, 1, 22, false)
                                }, 11, false),
                                J$.T(7421409, {
                                    type: J$.T(7421385, 'section', 21, false),
                                    kerning: J$.T(7421393, 1, 22, false),
                                    default: J$.T(7421401, '', 21, false)
                                }, 11, false),
                                J$.T(7421441, {
                                    type: J$.T(7421417, 'subsection', 21, false),
                                    kerning: J$.T(7421425, 1, 22, false),
                                    default: J$.T(7421433, '', 21, false)
                                }, 11, false)
                            ], 10, false), template, false, false);
                            J$.M(7421497, J$.R(7421465, 'this', this, false, false), 'setWriteTo', false)(J$.R(7421473, 'writeTo', writeTo, false, false), J$.G(7421489, J$.R(7421481, 'options', options, false, false), 'tty'));
                            var PlumbingClass = J$.W(7421529, 'PlumbingClass', J$.C(394664, J$.G(7421513, J$.R(7421505, 'options', options, false, false), 'Plumbing')) ? J$._() : J$.R(7421521, 'Plumbing', Plumbing, false, true), PlumbingClass, false, false);
                            J$.P(7421593, J$.R(7421537, 'this', this, false, false), '_gauge', J$.F(7421585, J$.R(7421545, 'PlumbingClass', PlumbingClass, false, false), true)(J$.R(7421553, 'theme', theme, false, false), J$.R(7421561, 'template', template, false, false), J$.M(7421577, J$.R(7421569, 'this', this, false, false), 'getWidth', false)()));
                            J$.P(7421649, J$.R(7421601, 'this', this, false, false), '_$$doRedraw', J$.F(7421641, J$.R(7421609, 'callWith', callWith, false, true), false)(J$.R(7421617, 'this', this, false, false), J$.G(7421633, J$.R(7421625, 'this', this, false, false), '_doRedraw')));
                            J$.P(7421705, J$.R(7421657, 'this', this, false, false), '_$$handleSizeChange', J$.F(7421697, J$.R(7421665, 'callWith', callWith, false, true), false)(J$.R(7421673, 'this', this, false, false), J$.G(7421689, J$.R(7421681, 'this', this, false, false), '_handleSizeChange')));
                            J$.P(7421761, J$.R(7421713, 'this', this, false, false), '_cleanupOnExit', J$.C(394672, J$.B(740410, '==', J$.G(7421729, J$.R(7421721, 'options', options, false, false), 'cleanupOnExit'), J$.T(7421737, null, 25, false))) ? J$._() : J$.G(7421753, J$.R(7421745, 'options', options, false, false), 'cleanupOnExit'));
                            J$.P(7421785, J$.R(7421769, 'this', this, false, false), '_removeOnExit', J$.T(7421777, null, 25, false));
                            if (J$.C(394704, J$.C(394696, J$.G(7421801, J$.R(7421793, 'options', options, false, false), 'enabled')) ? J$._() : J$.C(394688, J$.C(394680, J$.B(740418, '==', J$.G(7421817, J$.R(7421809, 'options', options, false, false), 'enabled'), J$.T(7421825, null, 25, false))) ? J$.G(7421841, J$.R(7421833, 'this', this, false, false), '_tty') : J$._()) ? J$.G(7421865, J$.G(7421857, J$.R(7421849, 'this', this, false, false), '_tty'), 'isTTY') : J$._())) {
                                J$.M(7421881, J$.R(7421873, 'this', this, false, false), 'enable', false)();
                            } else {
                                J$.M(7421897, J$.R(7421889, 'this', this, false, false), 'disable', false)();
                            }
                        } catch (J$e) {
                            J$.Ex(7426321, J$e);
                        } finally {
                            if (J$.Fr(7426329))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(7426193, 'Plumbing', Plumbing, false, false, false);
            J$.N(7426201, 'hasUnicode', hasUnicode, false, false, false);
            J$.N(7426209, 'hasColor', hasColor, false, false, false);
            J$.N(7426217, 'onExit', onExit, false, false, false);
            J$.N(7426225, 'defaultThemes', defaultThemes, false, false, false);
            J$.N(7426233, 'setInterval', setInterval, false, false, false);
            J$.N(7426241, 'process', process, false, false, false);
            J$.N(7426249, 'setImmediate', setImmediate, false, false, false);
            callWith = J$.N(7426265, 'callWith', J$.T(7426257, callWith, 12, false), true, false, false);
            Gauge = J$.N(7426281, 'Gauge', J$.T(7426273, Gauge, 12, false), true, false, false);
            J$.T(7420225, 'use strict', 21, false);
            var Plumbing = J$.W(7420257, 'Plumbing', J$.F(7420249, J$.I(typeof require === 'undefined' ? require = J$.R(7420233, 'require', undefined, true, true) : require = J$.R(7420233, 'require', require, true, true)), false)(J$.T(7420241, './plumbing.js', 21, false)), Plumbing, false, true);
            var hasUnicode = J$.W(7420289, 'hasUnicode', J$.F(7420281, J$.I(typeof require === 'undefined' ? require = J$.R(7420265, 'require', undefined, true, true) : require = J$.R(7420265, 'require', require, true, true)), false)(J$.T(7420273, 'has-unicode', 21, false)), hasUnicode, false, true);
            var hasColor = J$.W(7420321, 'hasColor', J$.F(7420313, J$.I(typeof require === 'undefined' ? require = J$.R(7420297, 'require', undefined, true, true) : require = J$.R(7420297, 'require', require, true, true)), false)(J$.T(7420305, './has-color.js', 21, false)), hasColor, false, true);
            var onExit = J$.W(7420353, 'onExit', J$.F(7420345, J$.I(typeof require === 'undefined' ? require = J$.R(7420329, 'require', undefined, true, true) : require = J$.R(7420329, 'require', require, true, true)), false)(J$.T(7420337, 'signal-exit', 21, false)), onExit, false, true);
            var defaultThemes = J$.W(7420385, 'defaultThemes', J$.F(7420377, J$.I(typeof require === 'undefined' ? require = J$.R(7420361, 'require', undefined, true, true) : require = J$.R(7420361, 'require', require, true, true)), false)(J$.T(7420369, './themes', 21, false)), defaultThemes, false, true);
            var setInterval = J$.W(7420417, 'setInterval', J$.F(7420409, J$.I(typeof require === 'undefined' ? require = J$.R(7420393, 'require', undefined, true, true) : require = J$.R(7420393, 'require', require, true, true)), false)(J$.T(7420401, './set-interval.js', 21, false)), setInterval, false, true);
            var process = J$.W(7420449, 'process', J$.F(7420441, J$.I(typeof require === 'undefined' ? require = J$.R(7420425, 'require', undefined, true, true) : require = J$.R(7420425, 'require', require, true, true)), false)(J$.T(7420433, './process.js', 21, false)), process, false, true);
            var setImmediate = J$.W(7420481, 'setImmediate', J$.F(7420473, J$.I(typeof require === 'undefined' ? require = J$.R(7420457, 'require', undefined, true, true) : require = J$.R(7420457, 'require', require, true, true)), false)(J$.T(7420465, './set-immediate', 21, false)), setImmediate, false, true);
            J$.P(7420505, J$.I(typeof module === 'undefined' ? module = J$.R(7420489, 'module', undefined, true, true) : module = J$.R(7420489, 'module', module, true, true)), 'exports', J$.R(7420497, 'Gauge', Gauge, false, true));
            J$.P(7421993, J$.R(7421977, 'Gauge', Gauge, false, true), 'prototype', J$.T(7421985, {}, 11, false));
            J$.P(7422065, J$.G(7422009, J$.R(7422001, 'Gauge', Gauge, false, true), 'prototype'), 'isEnabled', J$.T(7422057, function () {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(7422041, arguments.callee, this, arguments);
                            arguments = J$.N(7422049, 'arguments', arguments, true, false, false);
                            return J$.Rt(7422033, J$.U(740426, '!', J$.G(7422025, J$.R(7422017, 'this', this, false, false), '_disabled')));
                        } catch (J$e) {
                            J$.Ex(7426337, J$e);
                        } finally {
                            if (J$.Fr(7426345))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7422185, J$.G(7422081, J$.R(7422073, 'Gauge', Gauge, false, true), 'prototype'), 'setTemplate', J$.T(7422177, function (template) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7422153, arguments.callee, this, arguments);
                            arguments = J$.N(7422161, 'arguments', arguments, true, false, false);
                            template = J$.N(7422169, 'template', template, true, false, false);
                            J$.M(7422113, J$.G(7422097, J$.R(7422089, 'this', this, false, false), '_gauge'), 'setTemplate', false)(J$.R(7422105, 'template', template, false, false));
                            if (J$.C(394712, J$.G(7422129, J$.R(7422121, 'this', this, false, false), '_showing')))
                                J$.M(7422145, J$.R(7422137, 'this', this, false, false), '_requestRedraw', false)();
                        } catch (J$e) {
                            J$.Ex(7426353, J$e);
                        } finally {
                            if (J$.Fr(7426361))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7422641, J$.G(7422201, J$.R(7422193, 'Gauge', Gauge, false, true), 'prototype'), '_computeTheme', J$.T(7422633, function (theme) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(7422593, arguments.callee, this, arguments);
                            arguments = J$.N(7422601, 'arguments', arguments, true, false, false);
                            theme = J$.N(7422609, 'theme', theme, true, false, false);
                            J$.N(7422617, 'useUnicode', useUnicode, false, false, false);
                            J$.N(7422625, 'useColor', useColor, false, false, false);
                            if (J$.C(394720, J$.U(740434, '!', J$.R(7422209, 'theme', theme, false, false))))
                                theme = J$.W(7422225, 'theme', J$.T(7422217, {}, 11, false), theme, false, false);
                            if (J$.C(394776, J$.B(740450, '===', J$.U(740442, 'typeof', J$.R(7422233, 'theme', theme, false, false)), J$.T(7422241, 'string', 21, false)))) {
                                theme = J$.W(7422281, 'theme', J$.M(7422273, J$.G(7422257, J$.R(7422249, 'this', this, false, false), '_themes'), 'getTheme', false)(J$.R(7422265, 'theme', theme, false, false)), theme, false, false);
                            } else if (J$.C(394768, J$.C(394744, J$.R(7422289, 'theme', theme, false, false)) ? J$.C(394736, J$.C(394728, J$.B(740458, '===', J$.G(7422321, J$.M(7422313, J$.I(typeof Object === 'undefined' ? Object = J$.R(7422297, 'Object', undefined, true, true) : Object = J$.R(7422297, 'Object', Object, true, true)), 'keys', false)(J$.R(7422305, 'theme', theme, false, false)), 'length'), J$.T(7422329, 0, 22, false))) ? J$._() : J$.B(740466, '!=', J$.G(7422345, J$.R(7422337, 'theme', theme, false, false), 'hasUnicode'), J$.T(7422353, null, 25, false))) ? J$._() : J$.B(740474, '!=', J$.G(7422369, J$.R(7422361, 'theme', theme, false, false), 'hasColor'), J$.T(7422377, null, 25, false)) : J$._())) {
                                var useUnicode = J$.W(7422441, 'useUnicode', J$.C(394752, J$.B(740482, '==', J$.G(7422393, J$.R(7422385, 'theme', theme, false, false), 'hasUnicode'), J$.T(7422401, null, 25, false))) ? J$.F(7422417, J$.R(7422409, 'hasUnicode', hasUnicode, false, true), false)() : J$.G(7422433, J$.R(7422425, 'theme', theme, false, false), 'hasUnicode'), useUnicode, false, false);
                                var useColor = J$.W(7422497, 'useColor', J$.C(394760, J$.B(740490, '==', J$.G(7422457, J$.R(7422449, 'theme', theme, false, false), 'hasColor'), J$.T(7422465, null, 25, false))) ? J$.R(7422473, 'hasColor', hasColor, false, true) : J$.G(7422489, J$.R(7422481, 'theme', theme, false, false), 'hasColor'), useColor, false, false);
                                theme = J$.W(7422569, 'theme', J$.M(7422561, J$.G(7422513, J$.R(7422505, 'this', this, false, false), '_themes'), 'getDefault', false)(J$.T(7422553, {
                                    hasUnicode: J$.R(7422521, 'useUnicode', useUnicode, false, false),
                                    hasColor: J$.R(7422529, 'useColor', useColor, false, false),
                                    platform: J$.G(7422545, J$.R(7422537, 'theme', theme, false, false), 'platform')
                                }, 11, false)), theme, false, false);
                            }
                            return J$.Rt(7422585, J$.R(7422577, 'theme', theme, false, false));
                        } catch (J$e) {
                            J$.Ex(7426369, J$e);
                        } finally {
                            if (J$.Fr(7426377))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7422753, J$.G(7422657, J$.R(7422649, 'Gauge', Gauge, false, true), 'prototype'), 'setThemeset', J$.T(7422745, function (themes) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(7422721, arguments.callee, this, arguments);
                            arguments = J$.N(7422729, 'arguments', arguments, true, false, false);
                            themes = J$.N(7422737, 'themes', themes, true, false, false);
                            J$.P(7422681, J$.R(7422665, 'this', this, false, false), '_themes', J$.R(7422673, 'themes', themes, false, false));
                            J$.M(7422713, J$.R(7422689, 'this', this, false, false), 'setTheme', false)(J$.G(7422705, J$.R(7422697, 'this', this, false, false), '_theme'));
                        } catch (J$e) {
                            J$.Ex(7426385, J$e);
                        } finally {
                            if (J$.Fr(7426393))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7422913, J$.G(7422769, J$.R(7422761, 'Gauge', Gauge, false, true), 'prototype'), 'setTheme', J$.T(7422905, function (theme) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(7422881, arguments.callee, this, arguments);
                            arguments = J$.N(7422889, 'arguments', arguments, true, false, false);
                            theme = J$.N(7422897, 'theme', theme, true, false, false);
                            J$.M(7422817, J$.G(7422785, J$.R(7422777, 'this', this, false, false), '_gauge'), 'setTheme', false)(J$.M(7422809, J$.R(7422793, 'this', this, false, false), '_computeTheme', false)(J$.R(7422801, 'theme', theme, false, false)));
                            if (J$.C(394784, J$.G(7422833, J$.R(7422825, 'this', this, false, false), '_showing')))
                                J$.M(7422849, J$.R(7422841, 'this', this, false, false), '_requestRedraw', false)();
                            J$.P(7422873, J$.R(7422857, 'this', this, false, false), '_theme', J$.R(7422865, 'theme', theme, false, false));
                        } catch (J$e) {
                            J$.Ex(7426401, J$e);
                        } finally {
                            if (J$.Fr(7426409))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7423017, J$.G(7422929, J$.R(7422921, 'Gauge', Gauge, false, true), 'prototype'), '_requestRedraw', J$.T(7423009, function () {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(7422993, arguments.callee, this, arguments);
                            arguments = J$.N(7423001, 'arguments', arguments, true, false, false);
                            J$.P(7422953, J$.R(7422937, 'this', this, false, false), '_needsRedraw', J$.T(7422945, true, 23, false));
                            if (J$.C(394792, J$.U(740498, '!', J$.G(7422969, J$.R(7422961, 'this', this, false, false), '_fixedFramerate'))))
                                J$.M(7422985, J$.R(7422977, 'this', this, false, false), '_doRedraw', false)();
                        } catch (J$e) {
                            J$.Ex(7426417, J$e);
                        } finally {
                            if (J$.Fr(7426425))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7423129, J$.G(7423033, J$.R(7423025, 'Gauge', Gauge, false, true), 'prototype'), 'getWidth', J$.T(7423121, function () {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(7423105, arguments.callee, this, arguments);
                            arguments = J$.N(7423113, 'arguments', arguments, true, false, false);
                            return J$.Rt(7423097, J$.B(740506, '-', J$.C(394808, J$.C(394800, J$.G(7423049, J$.R(7423041, 'this', this, false, false), '_tty')) ? J$.G(7423073, J$.G(7423065, J$.R(7423057, 'this', this, false, false), '_tty'), 'columns') : J$._()) ? J$._() : J$.T(7423081, 80, 22, false), J$.T(7423089, 1, 22, false)));
                        } catch (J$e) {
                            J$.Ex(7426433, J$e);
                        } finally {
                            if (J$.Fr(7426441))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7423481, J$.G(7423145, J$.R(7423137, 'Gauge', Gauge, false, true), 'prototype'), 'setWriteTo', J$.T(7423473, function (writeTo, tty) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(7423433, arguments.callee, this, arguments);
                            arguments = J$.N(7423441, 'arguments', arguments, true, false, false);
                            writeTo = J$.N(7423449, 'writeTo', writeTo, true, false, false);
                            tty = J$.N(7423457, 'tty', tty, true, false, false);
                            J$.N(7423465, 'enabled', enabled, false, false, false);
                            var enabled = J$.W(7423169, 'enabled', J$.U(740514, '!', J$.G(7423161, J$.R(7423153, 'this', this, false, false), '_disabled')), enabled, false, false);
                            if (J$.C(394816, J$.R(7423177, 'enabled', enabled, false, false)))
                                J$.M(7423193, J$.R(7423185, 'this', this, false, false), 'disable', false)();
                            J$.P(7423217, J$.R(7423201, 'this', this, false, false), '_writeTo', J$.R(7423209, 'writeTo', writeTo, false, false));
                            J$.P(7423345, J$.R(7423225, 'this', this, false, false), '_tty', J$.C(394864, J$.C(394856, J$.C(394840, J$.R(7423233, 'tty', tty, false, false)) ? J$._() : J$.C(394832, J$.C(394824, J$.B(740522, '===', J$.R(7423241, 'writeTo', writeTo, false, false), J$.G(7423257, J$.R(7423249, 'process', process, false, true), 'stderr'))) ? J$.G(7423281, J$.G(7423273, J$.R(7423265, 'process', process, false, true), 'stdout'), 'isTTY') : J$._()) ? J$.G(7423297, J$.R(7423289, 'process', process, false, true), 'stdout') : J$._()) ? J$._() : J$.C(394848, J$.G(7423313, J$.R(7423305, 'writeTo', writeTo, false, false), 'isTTY')) ? J$.R(7423321, 'writeTo', writeTo, false, false) : J$._()) ? J$._() : J$.G(7423337, J$.R(7423329, 'this', this, false, false), '_tty'));
                            if (J$.C(394872, J$.G(7423361, J$.R(7423353, 'this', this, false, false), '_gauge')))
                                J$.M(7423401, J$.G(7423377, J$.R(7423369, 'this', this, false, false), '_gauge'), 'setWidth', false)(J$.M(7423393, J$.R(7423385, 'this', this, false, false), 'getWidth', false)());
                            if (J$.C(394880, J$.R(7423409, 'enabled', enabled, false, false)))
                                J$.M(7423425, J$.R(7423417, 'this', this, false, false), 'enable', false)();
                        } catch (J$e) {
                            J$.Ex(7426449, J$e);
                        } finally {
                            if (J$.Fr(7426457))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7423641, J$.G(7423497, J$.R(7423489, 'Gauge', Gauge, false, true), 'prototype'), 'enable', J$.T(7423633, function () {
                jalangiLabel11:
                    while (true) {
                        try {
                            J$.Fe(7423617, arguments.callee, this, arguments);
                            arguments = J$.N(7423625, 'arguments', arguments, true, false, false);
                            if (J$.C(394888, J$.U(740530, '!', J$.G(7423513, J$.R(7423505, 'this', this, false, false), '_disabled'))))
                                return J$.Rt(7423521, undefined);
                            J$.P(7423545, J$.R(7423529, 'this', this, false, false), '_disabled', J$.T(7423537, false, 23, false));
                            if (J$.C(394896, J$.G(7423561, J$.R(7423553, 'this', this, false, false), '_tty')))
                                J$.M(7423577, J$.R(7423569, 'this', this, false, false), '_enableEvents', false)();
                            if (J$.C(394904, J$.G(7423593, J$.R(7423585, 'this', this, false, false), '_showing')))
                                J$.M(7423609, J$.R(7423601, 'this', this, false, false), 'show', false)();
                        } catch (J$e) {
                            J$.Ex(7426465, J$e);
                        } finally {
                            if (J$.Fr(7426473))
                                continue jalangiLabel11;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7423873, J$.G(7423657, J$.R(7423649, 'Gauge', Gauge, false, true), 'prototype'), 'disable', J$.T(7423865, function () {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(7423849, arguments.callee, this, arguments);
                            arguments = J$.N(7423857, 'arguments', arguments, true, false, false);
                            if (J$.C(394912, J$.G(7423673, J$.R(7423665, 'this', this, false, false), '_disabled')))
                                return J$.Rt(7423681, undefined);
                            if (J$.C(394920, J$.G(7423697, J$.R(7423689, 'this', this, false, false), '_showing'))) {
                                J$.P(7423721, J$.R(7423705, 'this', this, false, false), '_lastUpdateAt', J$.T(7423713, null, 25, false));
                                J$.P(7423745, J$.R(7423729, 'this', this, false, false), '_showing', J$.T(7423737, false, 23, false));
                                J$.M(7423761, J$.R(7423753, 'this', this, false, false), '_doRedraw', false)();
                                J$.P(7423785, J$.R(7423769, 'this', this, false, false), '_showing', J$.T(7423777, true, 23, false));
                            }
                            J$.P(7423809, J$.R(7423793, 'this', this, false, false), '_disabled', J$.T(7423801, true, 23, false));
                            if (J$.C(394928, J$.G(7423825, J$.R(7423817, 'this', this, false, false), '_tty')))
                                J$.M(7423841, J$.R(7423833, 'this', this, false, false), '_disableEvents', false)();
                        } catch (J$e) {
                            J$.Ex(7426481, J$e);
                        } finally {
                            if (J$.Fr(7426489))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7424185, J$.G(7423889, J$.R(7423881, 'Gauge', Gauge, false, true), 'prototype'), '_enableEvents', J$.T(7424177, function () {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(7424161, arguments.callee, this, arguments);
                            arguments = J$.N(7424169, 'arguments', arguments, true, false, false);
                            if (J$.C(394936, J$.G(7423905, J$.R(7423897, 'this', this, false, false), '_cleanupOnExit'))) {
                                J$.P(7423977, J$.R(7423913, 'this', this, false, false), '_removeOnExit', J$.F(7423969, J$.R(7423921, 'onExit', onExit, false, true), false)(J$.F(7423961, J$.R(7423929, 'callWith', callWith, false, true), false)(J$.R(7423937, 'this', this, false, false), J$.G(7423953, J$.R(7423945, 'this', this, false, false), 'disable'))));
                            }
                            J$.M(7424025, J$.G(7423993, J$.R(7423985, 'this', this, false, false), '_tty'), 'on', false)(J$.T(7424001, 'resize', 21, false), J$.G(7424017, J$.R(7424009, 'this', this, false, false), '_$$handleSizeChange'));
                            if (J$.C(394952, J$.G(7424041, J$.R(7424033, 'this', this, false, false), '_fixedFramerate'))) {
                                J$.P(7424105, J$.R(7424049, 'this', this, false, false), 'redrawTracker', J$.F(7424097, J$.R(7424057, 'setInterval', setInterval, false, true), false)(J$.G(7424073, J$.R(7424065, 'this', this, false, false), '_$$doRedraw'), J$.G(7424089, J$.R(7424081, 'this', this, false, false), '_updateInterval')));
                                if (J$.C(394944, J$.G(7424129, J$.G(7424121, J$.R(7424113, 'this', this, false, false), 'redrawTracker'), 'unref')))
                                    J$.M(7424153, J$.G(7424145, J$.R(7424137, 'this', this, false, false), 'redrawTracker'), 'unref', false)();
                            }
                        } catch (J$e) {
                            J$.Ex(7426497, J$e);
                        } finally {
                            if (J$.Fr(7426505))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7424361, J$.G(7424201, J$.R(7424193, 'Gauge', Gauge, false, true), 'prototype'), '_disableEvents', J$.T(7424353, function () {
                jalangiLabel14:
                    while (true) {
                        try {
                            J$.Fe(7424337, arguments.callee, this, arguments);
                            arguments = J$.N(7424345, 'arguments', arguments, true, false, false);
                            J$.M(7424249, J$.G(7424217, J$.R(7424209, 'this', this, false, false), '_tty'), 'removeListener', false)(J$.T(7424225, 'resize', 21, false), J$.G(7424241, J$.R(7424233, 'this', this, false, false), '_$$handleSizeChange'));
                            if (J$.C(394960, J$.G(7424265, J$.R(7424257, 'this', this, false, false), '_fixedFramerate')))
                                J$.F(7424297, J$.I(typeof clearInterval === 'undefined' ? clearInterval = J$.R(7424273, 'clearInterval', undefined, true, true) : clearInterval = J$.R(7424273, 'clearInterval', clearInterval, true, true)), false)(J$.G(7424289, J$.R(7424281, 'this', this, false, false), 'redrawTracker'));
                            if (J$.C(394968, J$.G(7424313, J$.R(7424305, 'this', this, false, false), '_removeOnExit')))
                                J$.M(7424329, J$.R(7424321, 'this', this, false, false), '_removeOnExit', false)();
                        } catch (J$e) {
                            J$.Ex(7426513, J$e);
                        } finally {
                            if (J$.Fr(7426521))
                                continue jalangiLabel14;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7424601, J$.G(7424377, J$.R(7424369, 'Gauge', Gauge, false, true), 'prototype'), 'hide', J$.T(7424593, function (cb) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(7424569, arguments.callee, this, arguments);
                            arguments = J$.N(7424577, 'arguments', arguments, true, false, false);
                            cb = J$.N(7424585, 'cb', cb, true, false, false);
                            if (J$.C(394984, J$.G(7424393, J$.R(7424385, 'this', this, false, false), '_disabled')))
                                return J$.Rt(7424433, J$.C(394976, J$.R(7424401, 'cb', cb, false, false)) ? J$.M(7424425, J$.R(7424409, 'process', process, false, true), 'nextTick', false)(J$.R(7424417, 'cb', cb, false, false)) : J$._());
                            if (J$.C(395000, J$.U(740538, '!', J$.G(7424449, J$.R(7424441, 'this', this, false, false), '_showing'))))
                                return J$.Rt(7424489, J$.C(394992, J$.R(7424457, 'cb', cb, false, false)) ? J$.M(7424481, J$.R(7424465, 'process', process, false, true), 'nextTick', false)(J$.R(7424473, 'cb', cb, false, false)) : J$._());
                            J$.P(7424513, J$.R(7424497, 'this', this, false, false), '_showing', J$.T(7424505, false, 23, false));
                            J$.M(7424529, J$.R(7424521, 'this', this, false, false), '_doRedraw', false)();
                            J$.C(395008, J$.R(7424537, 'cb', cb, false, false)) ? J$.F(7424561, J$.R(7424545, 'setImmediate', setImmediate, false, true), false)(J$.R(7424553, 'cb', cb, false, false)) : J$._();
                        } catch (J$e) {
                            J$.Ex(7426529, J$e);
                        } finally {
                            if (J$.Fr(7426537))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7425041, J$.G(7424617, J$.R(7424609, 'Gauge', Gauge, false, true), 'prototype'), 'show', J$.T(7425033, function (section, completed) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(7424977, arguments.callee, this, arguments);
                            arguments = J$.N(7424985, 'arguments', arguments, true, false, false);
                            section = J$.N(7424993, 'section', section, true, false, false);
                            completed = J$.N(7425001, 'completed', completed, true, false, false);
                            J$.N(7425009, 'sectionKeys', sectionKeys, false, false, false);
                            J$.N(7425017, 'ii', ii, false, false, false);
                            J$.N(7425025, 'key', key, false, false, false);
                            J$.P(7424641, J$.R(7424625, 'this', this, false, false), '_showing', J$.T(7424633, true, 23, false));
                            if (J$.C(395032, J$.B(740554, '===', J$.U(740546, 'typeof', J$.R(7424649, 'section', section, false, false)), J$.T(7424657, 'string', 21, false)))) {
                                J$.P(7424689, J$.G(7424673, J$.R(7424665, 'this', this, false, false), '_status'), 'section', J$.R(7424681, 'section', section, false, false));
                            } else if (J$.C(395024, J$.B(740570, '===', J$.U(740562, 'typeof', J$.R(7424697, 'section', section, false, false)), J$.T(7424705, 'object', 21, false)))) {
                                var sectionKeys = J$.W(7424737, 'sectionKeys', J$.M(7424729, J$.I(typeof Object === 'undefined' ? Object = J$.R(7424713, 'Object', undefined, true, true) : Object = J$.R(7424713, 'Object', Object, true, true)), 'keys', false)(J$.R(7424721, 'section', section, false, false)), sectionKeys, false, false);
                                for (var ii = J$.W(7424753, 'ii', J$.T(7424745, 0, 22, false), ii, false, false); J$.C(395016, J$.B(740578, '<', J$.R(7424761, 'ii', ii, false, false), J$.G(7424777, J$.R(7424769, 'sectionKeys', sectionKeys, false, false), 'length'))); ii = J$.W(7424793, 'ii', J$.B(740594, '+', J$.U(740586, '+', J$.R(7424785, 'ii', ii, false, false)), 1), ii, false, false)) {
                                    var key = J$.W(7424825, 'key', J$.G(7424817, J$.R(7424801, 'sectionKeys', sectionKeys, false, false), J$.R(7424809, 'ii', ii, false, false)), key, false, false);
                                    J$.P(7424881, J$.G(7424841, J$.R(7424833, 'this', this, false, false), '_status'), J$.R(7424849, 'key', key, false, false), J$.G(7424873, J$.R(7424857, 'section', section, false, false), J$.R(7424865, 'key', key, false, false)));
                                }
                            }
                            if (J$.C(395040, J$.B(740602, '!=', J$.R(7424889, 'completed', completed, false, false), J$.T(7424897, null, 25, false))))
                                J$.P(7424929, J$.G(7424913, J$.R(7424905, 'this', this, false, false), '_status'), 'completed', J$.R(7424921, 'completed', completed, false, false));
                            if (J$.C(395048, J$.G(7424945, J$.R(7424937, 'this', this, false, false), '_disabled')))
                                return J$.Rt(7424953, undefined);
                            J$.M(7424969, J$.R(7424961, 'this', this, false, false), '_requestRedraw', false)();
                        } catch (J$e) {
                            J$.Ex(7426545, J$e);
                        } finally {
                            if (J$.Fr(7426553))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7425225, J$.G(7425057, J$.R(7425049, 'Gauge', Gauge, false, true), 'prototype'), 'pulse', J$.T(7425217, function (subsection) {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(7425193, arguments.callee, this, arguments);
                            arguments = J$.N(7425201, 'arguments', arguments, true, false, false);
                            subsection = J$.N(7425209, 'subsection', subsection, true, false, false);
                            J$.P(7425097, J$.G(7425073, J$.R(7425065, 'this', this, false, false), '_status'), 'subsection', J$.C(395056, J$.R(7425081, 'subsection', subsection, false, false)) ? J$._() : J$.T(7425089, '', 21, false));
                            J$.B(740610, '-', J$.A(7425121, J$.G(7425113, J$.R(7425105, 'this', this, false, false), '_status'), 'spun', '+')(1), 1);
                            if (J$.C(395064, J$.G(7425137, J$.R(7425129, 'this', this, false, false), '_disabled')))
                                return J$.Rt(7425145, undefined);
                            if (J$.C(395072, J$.U(740618, '!', J$.G(7425161, J$.R(7425153, 'this', this, false, false), '_showing'))))
                                return J$.Rt(7425169, undefined);
                            J$.M(7425185, J$.R(7425177, 'this', this, false, false), '_requestRedraw', false)();
                        } catch (J$e) {
                            J$.Ex(7426561, J$e);
                        } finally {
                            if (J$.Fr(7426569))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7425345, J$.G(7425241, J$.R(7425233, 'Gauge', Gauge, false, true), 'prototype'), '_handleSizeChange', J$.T(7425337, function () {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(7425321, arguments.callee, this, arguments);
                            arguments = J$.N(7425329, 'arguments', arguments, true, false, false);
                            J$.M(7425297, J$.G(7425257, J$.R(7425249, 'this', this, false, false), '_gauge'), 'setWidth', false)(J$.B(740626, '-', J$.G(7425281, J$.G(7425273, J$.R(7425265, 'this', this, false, false), '_tty'), 'columns'), J$.T(7425289, 1, 22, false)));
                            J$.M(7425313, J$.R(7425305, 'this', this, false, false), '_requestRedraw', false)();
                        } catch (J$e) {
                            J$.Ex(7426577, J$e);
                        } finally {
                            if (J$.Fr(7426585))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7426177, J$.G(7425361, J$.R(7425353, 'Gauge', Gauge, false, true), 'prototype'), '_doRedraw', J$.T(7426169, function () {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(7426137, arguments.callee, this, arguments);
                            arguments = J$.N(7426145, 'arguments', arguments, true, false, false);
                            J$.N(7426153, 'now', now, false, false, false);
                            J$.N(7426161, 'result', result, false, false, false);
                            if (J$.C(395088, J$.C(395080, J$.G(7425377, J$.R(7425369, 'this', this, false, false), '_disabled')) ? J$._() : J$.G(7425393, J$.R(7425385, 'this', this, false, false), '_paused')))
                                return J$.Rt(7425401, undefined);
                            if (J$.C(395112, J$.U(740634, '!', J$.G(7425417, J$.R(7425409, 'this', this, false, false), '_fixedFramerate')))) {
                                var now = J$.W(7425441, 'now', J$.M(7425433, J$.I(typeof Date === 'undefined' ? Date = J$.R(7425425, 'Date', undefined, true, true) : Date = J$.R(7425425, 'Date', Date, true, true)), 'now', false)(), now, false, false);
                                if (J$.C(395104, J$.C(395096, J$.G(7425457, J$.R(7425449, 'this', this, false, false), '_lastUpdateAt')) ? J$.B(740650, '<', J$.B(740642, '-', J$.R(7425465, 'now', now, false, false), J$.G(7425481, J$.R(7425473, 'this', this, false, false), '_lastUpdateAt')), J$.G(7425497, J$.R(7425489, 'this', this, false, false), '_updateInterval')) : J$._()))
                                    return J$.Rt(7425505, undefined);
                                J$.P(7425529, J$.R(7425513, 'this', this, false, false), '_lastUpdateAt', J$.R(7425521, 'now', now, false, false));
                            }
                            if (J$.C(395136, J$.C(395120, J$.U(740658, '!', J$.G(7425545, J$.R(7425537, 'this', this, false, false), '_showing'))) ? J$.G(7425561, J$.R(7425553, 'this', this, false, false), '_onScreen') : J$._())) {
                                J$.P(7425585, J$.R(7425569, 'this', this, false, false), '_onScreen', J$.T(7425577, false, 23, false));
                                var result = J$.W(7425617, 'result', J$.M(7425609, J$.G(7425601, J$.R(7425593, 'this', this, false, false), '_gauge'), 'hide', false)(), result, false, false);
                                if (J$.C(395128, J$.G(7425633, J$.R(7425625, 'this', this, false, false), '_hideCursor'))) {
                                    result = J$.W(7425673, 'result', J$.B(740666, '+', J$.R(7425665, 'result', result, false, false), J$.M(7425657, J$.G(7425649, J$.R(7425641, 'this', this, false, false), '_gauge'), 'showCursor', false)()), result, false, false);
                                }
                                return J$.Rt(7425713, J$.M(7425705, J$.G(7425689, J$.R(7425681, 'this', this, false, false), '_writeTo'), 'write', false)(J$.R(7425697, 'result', result, false, false)));
                            }
                            if (J$.C(395152, J$.C(395144, J$.U(740674, '!', J$.G(7425729, J$.R(7425721, 'this', this, false, false), '_showing'))) ? J$.U(740682, '!', J$.G(7425745, J$.R(7425737, 'this', this, false, false), '_onScreen')) : J$._()))
                                return J$.Rt(7425753, undefined);
                            if (J$.C(395176, J$.C(395160, J$.G(7425769, J$.R(7425761, 'this', this, false, false), '_showing')) ? J$.U(740690, '!', J$.G(7425785, J$.R(7425777, 'this', this, false, false), '_onScreen')) : J$._())) {
                                J$.P(7425809, J$.R(7425793, 'this', this, false, false), '_onScreen', J$.T(7425801, true, 23, false));
                                J$.P(7425833, J$.R(7425817, 'this', this, false, false), '_needsRedraw', J$.T(7425825, true, 23, false));
                                if (J$.C(395168, J$.G(7425849, J$.R(7425841, 'this', this, false, false), '_hideCursor'))) {
                                    J$.M(7425897, J$.G(7425865, J$.R(7425857, 'this', this, false, false), '_writeTo'), 'write', false)(J$.M(7425889, J$.G(7425881, J$.R(7425873, 'this', this, false, false), '_gauge'), 'hideCursor', false)());
                                }
                            }
                            if (J$.C(395184, J$.U(740698, '!', J$.G(7425913, J$.R(7425905, 'this', this, false, false), '_needsRedraw'))))
                                return J$.Rt(7425921, undefined);
                            if (J$.C(395192, J$.U(740706, '!', J$.M(7425985, J$.G(7425937, J$.R(7425929, 'this', this, false, false), '_writeTo'), 'write', false)(J$.M(7425977, J$.G(7425953, J$.R(7425945, 'this', this, false, false), '_gauge'), 'show', false)(J$.G(7425969, J$.R(7425961, 'this', this, false, false), '_status')))))) {
                                J$.P(7426009, J$.R(7425993, 'this', this, false, false), '_paused', J$.T(7426001, true, 23, false));
                                J$.M(7426129, J$.G(7426025, J$.R(7426017, 'this', this, false, false), '_writeTo'), 'on', false)(J$.T(7426033, 'drain', 21, false), J$.F(7426121, J$.R(7426041, 'callWith', callWith, false, true), false)(J$.R(7426049, 'this', this, false, false), J$.T(7426113, function () {
                                    jalangiLabel19:
                                        while (true) {
                                            try {
                                                J$.Fe(7426097, arguments.callee, this, arguments);
                                                arguments = J$.N(7426105, 'arguments', arguments, true, false, false);
                                                J$.P(7426073, J$.R(7426057, 'this', this, false, false), '_paused', J$.T(7426065, false, 23, false));
                                                J$.M(7426089, J$.R(7426081, 'this', this, false, false), '_doRedraw', false)();
                                            } catch (J$e) {
                                                J$.Ex(7426593, J$e);
                                            } finally {
                                                if (J$.Fr(7426601))
                                                    continue jalangiLabel19;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)));
                            }
                        } catch (J$e) {
                            J$.Ex(7426609, J$e);
                        } finally {
                            if (J$.Fr(7426617))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(7426625, J$e);
        } finally {
            if (J$.Sr(7426633))
                continue jalangiLabel21;
            else
                break jalangiLabel21;
        }
    }
// JALANGI DO NOT INSTRUMENT
