// gauge/render-template.js
J$.noInstrEval = false;
jalangiLabel20:
    while (true) {
        try {
            J$.Se(7434065, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/gauge/render-template.js');
            function renderValueWithValues(values) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7429345, arguments.callee, this, arguments);
                            arguments = J$.N(7429353, 'arguments', arguments, true, false, false);
                            values = J$.N(7429361, 'values', values, true, false, false);
                            return J$.Rt(7429337, J$.T(7429329, function (item) {
                                jalangiLabel0:
                                    while (true) {
                                        try {
                                            J$.Fe(7429305, arguments.callee, this, arguments);
                                            arguments = J$.N(7429313, 'arguments', arguments, true, false, false);
                                            item = J$.N(7429321, 'item', item, true, false, false);
                                            return J$.Rt(7429297, J$.F(7429289, J$.R(7429265, 'renderValue', renderValue, false, true), false)(J$.R(7429273, 'item', item, false, false), J$.R(7429281, 'values', values, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(7434257, J$e);
                                        } finally {
                                            if (J$.Fr(7434265))
                                                continue jalangiLabel0;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7434273, J$e);
                        } finally {
                            if (J$.Fr(7434281))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function preType(item) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(7429737, arguments.callee, this, arguments);
                            arguments = J$.N(7429745, 'arguments', arguments, true, false, false);
                            item = J$.N(7429753, 'item', item, true, false, false);
                            J$.N(7429761, 'cappedTypeName', cappedTypeName, false, false, false);
                            var cappedTypeName = J$.W(7429705, 'cappedTypeName', J$.B(740834, '+', J$.M(7429665, J$.G(7429657, J$.G(7429641, J$.R(7429633, 'item', item, false, false), 'type'), J$.T(7429649, 0, 22, false)), 'toUpperCase', false)(), J$.M(7429697, J$.G(7429681, J$.R(7429673, 'item', item, false, false), 'type'), 'slice', false)(J$.T(7429689, 1, 22, false))), cappedTypeName, false, false);
                            return J$.Rt(7429729, J$.B(740842, '+', J$.T(7429713, 'pre', 21, false), J$.R(7429721, 'cappedTypeName', cappedTypeName, false, false)));
                        } catch (J$e) {
                            J$.Ex(7434305, J$e);
                        } finally {
                            if (J$.Fr(7434313))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function postType(item) {
                jalangiLabel4:
                    while (true) {
                        try {
                            J$.Fe(7429873, arguments.callee, this, arguments);
                            arguments = J$.N(7429881, 'arguments', arguments, true, false, false);
                            item = J$.N(7429889, 'item', item, true, false, false);
                            J$.N(7429897, 'cappedTypeName', cappedTypeName, false, false, false);
                            var cappedTypeName = J$.W(7429841, 'cappedTypeName', J$.B(740850, '+', J$.M(7429801, J$.G(7429793, J$.G(7429777, J$.R(7429769, 'item', item, false, false), 'type'), J$.T(7429785, 0, 22, false)), 'toUpperCase', false)(), J$.M(7429833, J$.G(7429817, J$.R(7429809, 'item', item, false, false), 'type'), 'slice', false)(J$.T(7429825, 1, 22, false))), cappedTypeName, false, false);
                            return J$.Rt(7429865, J$.B(740858, '+', J$.T(7429849, 'post', 21, false), J$.R(7429857, 'cappedTypeName', cappedTypeName, false, false)));
                        } catch (J$e) {
                            J$.Ex(7434321, J$e);
                        } finally {
                            if (J$.Fr(7434329))
                                continue jalangiLabel4;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function hasPreOrPost(item, values) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(7430017, arguments.callee, this, arguments);
                            arguments = J$.N(7430025, 'arguments', arguments, true, false, false);
                            item = J$.N(7430033, 'item', item, true, false, false);
                            values = J$.N(7430041, 'values', values, true, false, false);
                            if (J$.C(395256, J$.U(740866, '!', J$.G(7429913, J$.R(7429905, 'item', item, false, false), 'type'))))
                                return J$.Rt(7429921, undefined);
                            return J$.Rt(7430009, J$.C(395264, J$.G(7429961, J$.R(7429929, 'values', values, false, false), J$.F(7429953, J$.R(7429937, 'preType', preType, false, true), false)(J$.R(7429945, 'item', item, false, false)))) ? J$._() : J$.G(7430001, J$.R(7429969, 'values', values, false, false), J$.F(7429993, J$.R(7429977, 'postType', postType, false, true), false)(J$.R(7429985, 'item', item, false, false))));
                        } catch (J$e) {
                            J$.Ex(7434337, J$e);
                        } finally {
                            if (J$.Fr(7434345))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function generatePreAndPost(baseItem, parentValues) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(7430673, arguments.callee, this, arguments);
                            arguments = J$.N(7430681, 'arguments', arguments, true, false, false);
                            baseItem = J$.N(7430689, 'baseItem', baseItem, true, false, false);
                            parentValues = J$.N(7430697, 'parentValues', parentValues, true, false, false);
                            J$.N(7430705, 'item', item, false, false, false);
                            J$.N(7430713, 'values', values, false, false, false);
                            J$.N(7430721, 'template', template, false, false, false);
                            J$.N(7430729, 'pre', pre, false, false, false);
                            J$.N(7430737, 'post', post, false, false, false);
                            var item = J$.W(7430081, 'item', J$.F(7430073, J$.R(7430049, 'objectAssign', objectAssign, false, true), false)(J$.T(7430057, {}, 11, false), J$.R(7430065, 'baseItem', baseItem, false, false)), item, false, false);
                            var values = J$.W(7430113, 'values', J$.M(7430105, J$.I(typeof Object === 'undefined' ? Object = J$.R(7430089, 'Object', undefined, true, true) : Object = J$.R(7430089, 'Object', Object, true, true)), 'create', false)(J$.R(7430097, 'parentValues', parentValues, false, false)), values, false, false);
                            var template = J$.W(7430129, 'template', J$.T(7430121, [], 10, false), template, false, false);
                            var pre = J$.W(7430161, 'pre', J$.F(7430153, J$.R(7430137, 'preType', preType, false, true), false)(J$.R(7430145, 'item', item, false, false)), pre, false, false);
                            var post = J$.W(7430193, 'post', J$.F(7430185, J$.R(7430169, 'postType', postType, false, true), false)(J$.R(7430177, 'item', item, false, false)), post, false, false);
                            if (J$.C(395272, J$.G(7430217, J$.R(7430201, 'values', values, false, false), J$.R(7430209, 'pre', pre, false, false)))) {
                                J$.M(7430265, J$.R(7430225, 'template', template, false, false), 'push', false)(J$.T(7430257, { value: J$.G(7430249, J$.R(7430233, 'values', values, false, false), J$.R(7430241, 'pre', pre, false, false)) }, 11, false));
                                J$.P(7430297, J$.R(7430273, 'values', values, false, false), J$.R(7430281, 'pre', pre, false, false), J$.T(7430289, null, 25, false));
                            }
                            J$.P(7430321, J$.R(7430305, 'item', item, false, false), 'minLength', J$.T(7430313, null, 25, false));
                            J$.P(7430345, J$.R(7430329, 'item', item, false, false), 'length', J$.T(7430337, null, 25, false));
                            J$.P(7430369, J$.R(7430353, 'item', item, false, false), 'maxLength', J$.T(7430361, null, 25, false));
                            J$.M(7430393, J$.R(7430377, 'template', template, false, false), 'push', false)(J$.R(7430385, 'item', item, false, false));
                            J$.P(7430457, J$.R(7430401, 'values', values, false, false), J$.G(7430417, J$.R(7430409, 'item', item, false, false), 'type'), J$.G(7430449, J$.R(7430425, 'values', values, false, false), J$.G(7430441, J$.R(7430433, 'item', item, false, false), 'type')));
                            if (J$.C(395280, J$.G(7430481, J$.R(7430465, 'values', values, false, false), J$.R(7430473, 'post', post, false, false)))) {
                                J$.M(7430529, J$.R(7430489, 'template', template, false, false), 'push', false)(J$.T(7430521, { value: J$.G(7430513, J$.R(7430497, 'values', values, false, false), J$.R(7430505, 'post', post, false, false)) }, 11, false));
                                J$.P(7430561, J$.R(7430537, 'values', values, false, false), J$.R(7430545, 'post', post, false, false), J$.T(7430553, null, 25, false));
                            }
                            return J$.Rt(7430665, J$.T(7430657, function ($1, $2, length) {
                                jalangiLabel6:
                                    while (true) {
                                        try {
                                            J$.Fe(7430617, arguments.callee, this, arguments);
                                            arguments = J$.N(7430625, 'arguments', arguments, true, false, false);
                                            $1 = J$.N(7430633, '$1', $1, true, false, false);
                                            $2 = J$.N(7430641, '$2', $2, true, false, false);
                                            length = J$.N(7430649, 'length', length, true, false, false);
                                            return J$.Rt(7430609, J$.F(7430601, J$.R(7430569, 'renderTemplate', renderTemplate, false, true), false)(J$.R(7430577, 'length', length, false, false), J$.R(7430585, 'template', template, false, false), J$.R(7430593, 'values', values, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(7434353, J$e);
                                        } finally {
                                            if (J$.Fr(7434361))
                                                continue jalangiLabel6;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                        } catch (J$e) {
                            J$.Ex(7434369, J$e);
                        } finally {
                            if (J$.Fr(7434377))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function prepareItems(width, template, values) {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(7433121, arguments.callee, this, arguments);
                            function cloneAndObjectify(item, index, arr) {
                                jalangiLabel8:
                                    while (true) {
                                        try {
                                            J$.Fe(7431241, arguments.callee, this, arguments);
                                            arguments = J$.N(7431249, 'arguments', arguments, true, false, false);
                                            item = J$.N(7431257, 'item', item, true, false, false);
                                            index = J$.N(7431265, 'index', index, true, false, false);
                                            arr = J$.N(7431273, 'arr', arr, true, false, false);
                                            J$.N(7431281, 'cloned', cloned, false, false, false);
                                            J$.N(7431289, 'type', type, false, false, false);
                                            var cloned = J$.W(7430777, 'cloned', J$.F(7430769, J$.R(7430745, 'TemplateItem', TemplateItem, false, true), true)(J$.R(7430753, 'item', item, false, false), J$.R(7430761, 'width', width, false, false)), cloned, false, false);
                                            var type = J$.W(7430801, 'type', J$.G(7430793, J$.R(7430785, 'cloned', cloned, false, false), 'type'), type, false, false);
                                            if (J$.C(395304, J$.B(740874, '==', J$.G(7430817, J$.R(7430809, 'cloned', cloned, false, false), 'value'), J$.T(7430825, null, 25, false)))) {
                                                if (J$.C(395296, J$.U(740890, '!', J$.B(740882, 'in', J$.R(7430833, 'type', type, false, false), J$.R(7430841, 'values', values, false, false))))) {
                                                    if (J$.C(395288, J$.B(740898, '==', J$.G(7430857, J$.R(7430849, 'cloned', cloned, false, false), 'default'), J$.T(7430865, null, 25, false)))) {
                                                        throw J$.M(7430897, J$.R(7430873, 'error', error, false, true), 'MissingTemplateValue', true)(J$.R(7430881, 'cloned', cloned, false, false), J$.R(7430889, 'values', values, false, false));
                                                    } else {
                                                        J$.P(7430929, J$.R(7430905, 'cloned', cloned, false, false), 'value', J$.G(7430921, J$.R(7430913, 'cloned', cloned, false, false), 'default'));
                                                    }
                                                } else {
                                                    J$.P(7430969, J$.R(7430937, 'cloned', cloned, false, false), 'value', J$.G(7430961, J$.R(7430945, 'values', values, false, false), J$.R(7430953, 'type', type, false, false)));
                                                }
                                            }
                                            if (J$.C(395320, J$.C(395312, J$.B(740906, '==', J$.G(7430985, J$.R(7430977, 'cloned', cloned, false, false), 'value'), J$.T(7430993, null, 25, false))) ? J$._() : J$.B(740914, '===', J$.G(7431009, J$.R(7431001, 'cloned', cloned, false, false), 'value'), J$.T(7431017, '', 21, false))))
                                                return J$.Rt(7431033, J$.T(7431025, null, 25, false));
                                            J$.P(7431057, J$.R(7431041, 'cloned', cloned, false, false), 'index', J$.R(7431049, 'index', index, false, false));
                                            J$.P(7431089, J$.R(7431065, 'cloned', cloned, false, false), 'first', J$.B(740922, '===', J$.R(7431073, 'index', index, false, false), J$.T(7431081, 0, 22, false)));
                                            J$.P(7431137, J$.R(7431097, 'cloned', cloned, false, false), 'last', J$.B(740938, '===', J$.R(7431105, 'index', index, false, false), J$.B(740930, '-', J$.G(7431121, J$.R(7431113, 'arr', arr, false, false), 'length'), J$.T(7431129, 1, 22, false))));
                                            if (J$.C(395328, J$.F(7431169, J$.R(7431145, 'hasPreOrPost', hasPreOrPost, false, true), false)(J$.R(7431153, 'cloned', cloned, false, false), J$.R(7431161, 'values', values, false, false))))
                                                J$.P(7431217, J$.R(7431177, 'cloned', cloned, false, false), 'value', J$.F(7431209, J$.R(7431185, 'generatePreAndPost', generatePreAndPost, false, true), false)(J$.R(7431193, 'cloned', cloned, false, false), J$.R(7431201, 'values', values, false, false)));
                                            return J$.Rt(7431233, J$.R(7431225, 'cloned', cloned, false, false));
                                        } catch (J$e) {
                                            J$.Ex(7434385, J$e);
                                        } finally {
                                            if (J$.Fr(7434393))
                                                continue jalangiLabel8;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            function consumeSpace(length) {
                                jalangiLabel10:
                                    while (true) {
                                        try {
                                            J$.Fe(7431529, arguments.callee, this, arguments);
                                            arguments = J$.N(7431537, 'arguments', arguments, true, false, false);
                                            length = J$.N(7431545, 'length', length, true, false, false);
                                            if (J$.C(395336, J$.B(740954, '>', J$.R(7431449, 'length', length, false, false), J$.R(7431457, 'remainingSpace', remainingSpace, false, false))))
                                                length = J$.W(7431473, 'length', J$.R(7431465, 'remainingSpace', remainingSpace, false, false), length, false, false);
                                            outputLength = J$.W(7431497, 'outputLength', J$.B(740962, '+', J$.R(7431489, 'outputLength', outputLength, false, false), J$.R(7431481, 'length', length, false, false)), outputLength, false, false);
                                            remainingSpace = J$.W(7431521, 'remainingSpace', J$.B(740970, '-', J$.R(7431513, 'remainingSpace', remainingSpace, false, false), J$.R(7431505, 'length', length, false, false)), remainingSpace, false, false);
                                        } catch (J$e) {
                                            J$.Ex(7434417, J$e);
                                        } finally {
                                            if (J$.Fr(7434425))
                                                continue jalangiLabel10;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            function finishSizing(item, length) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(7431897, arguments.callee, this, arguments);
                                            arguments = J$.N(7431905, 'arguments', arguments, true, false, false);
                                            item = J$.N(7431913, 'item', item, true, false, false);
                                            length = J$.N(7431921, 'length', length, true, false, false);
                                            if (J$.C(395344, J$.G(7431561, J$.R(7431553, 'item', item, false, false), 'finished')))
                                                throw J$.M(7431585, J$.R(7431569, 'error', error, false, true), 'Internal', true)(J$.T(7431577, 'Tried to finish template item that was already finished', 21, false));
                                            if (J$.C(395352, J$.B(740978, '===', J$.R(7431593, 'length', length, false, false), J$.T(7431601, Infinity, 22, false))))
                                                throw J$.M(7431625, J$.R(7431609, 'error', error, false, true), 'Internal', true)(J$.T(7431617, 'Length of template item cannot be infinity', 21, false));
                                            if (J$.C(395360, J$.B(740986, '!=', J$.R(7431633, 'length', length, false, false), J$.T(7431641, null, 25, false))))
                                                J$.P(7431665, J$.R(7431649, 'item', item, false, false), 'length', J$.R(7431657, 'length', length, false, false));
                                            J$.P(7431689, J$.R(7431673, 'item', item, false, false), 'minLength', J$.T(7431681, null, 25, false));
                                            J$.P(7431713, J$.R(7431697, 'item', item, false, false), 'maxLength', J$.T(7431705, null, 25, false));
                                            variableCount = J$.W(7431729, 'variableCount', J$.B(741002, '-', J$.U(740994, '+', J$.R(7431721, 'variableCount', variableCount, false, false)), 1), variableCount, false, false);
                                            J$.P(7431753, J$.R(7431737, 'item', item, false, false), 'finished', J$.T(7431745, true, 23, false));
                                            if (J$.C(395368, J$.B(741010, '==', J$.G(7431769, J$.R(7431761, 'item', item, false, false), 'length'), J$.T(7431777, null, 25, false))))
                                                J$.P(7431809, J$.R(7431785, 'item', item, false, false), 'length', J$.M(7431801, J$.R(7431793, 'item', item, false, false), 'getBaseLength', false)());
                                            if (J$.C(395376, J$.B(741018, '==', J$.G(7431825, J$.R(7431817, 'item', item, false, false), 'length'), J$.T(7431833, null, 25, false))))
                                                throw J$.M(7431857, J$.R(7431841, 'error', error, false, true), 'Internal', true)(J$.T(7431849, 'Finished template items must have a length', 21, false));
                                            J$.F(7431889, J$.R(7431865, 'consumeSpace', consumeSpace, false, false), false)(J$.M(7431881, J$.R(7431873, 'item', item, false, false), 'getLength', false)());
                                        } catch (J$e) {
                                            J$.Ex(7434433, J$e);
                                        } finally {
                                            if (J$.Fr(7434441))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(7433129, 'arguments', arguments, true, false, false);
                            width = J$.N(7433137, 'width', width, true, false, false);
                            template = J$.N(7433145, 'template', template, true, false, false);
                            values = J$.N(7433153, 'values', values, true, false, false);
                            cloneAndObjectify = J$.N(7433169, 'cloneAndObjectify', J$.T(7433161, cloneAndObjectify, 12, false), true, false, false);
                            J$.N(7433177, 'output', output, false, false, false);
                            J$.N(7433185, 'outputLength', outputLength, false, false, false);
                            J$.N(7433193, 'remainingSpace', remainingSpace, false, false, false);
                            J$.N(7433201, 'variableCount', variableCount, false, false, false);
                            consumeSpace = J$.N(7433217, 'consumeSpace', J$.T(7433209, consumeSpace, 12, false), true, false, false);
                            finishSizing = J$.N(7433233, 'finishSizing', J$.T(7433225, finishSizing, 12, false), true, false, false);
                            J$.N(7433241, 'resized', resized, false, false, false);
                            J$.N(7433249, 'resizing', resizing, false, false, false);
                            J$.N(7433257, 'hunkSize', hunkSize, false, false, false);
                            var output = J$.W(7431385, 'output', J$.M(7431377, J$.M(7431313, J$.R(7431297, 'template', template, false, false), 'map', false)(J$.R(7431305, 'cloneAndObjectify', cloneAndObjectify, false, false)), 'filter', false)(J$.T(7431369, function (item) {
                                jalangiLabel9:
                                    while (true) {
                                        try {
                                            J$.Fe(7431345, arguments.callee, this, arguments);
                                            arguments = J$.N(7431353, 'arguments', arguments, true, false, false);
                                            item = J$.N(7431361, 'item', item, true, false, false);
                                            return J$.Rt(7431337, J$.B(740946, '!=', J$.R(7431321, 'item', item, false, false), J$.T(7431329, null, 25, false)));
                                        } catch (J$e) {
                                            J$.Ex(7434401, J$e);
                                        } finally {
                                            if (J$.Fr(7434409))
                                                continue jalangiLabel9;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false)), output, false, false);
                            var outputLength = J$.W(7431401, 'outputLength', J$.T(7431393, 0, 22, false), outputLength, false, false);
                            var remainingSpace = J$.W(7431417, 'remainingSpace', J$.R(7431409, 'width', width, false, false), remainingSpace, false, false);
                            var variableCount = J$.W(7431441, 'variableCount', J$.G(7431433, J$.R(7431425, 'output', output, false, false), 'length'), variableCount, false, false);
                            J$.M(7432209, J$.R(7431929, 'output', output, false, false), 'forEach', false)(J$.T(7432201, function (item) {
                                jalangiLabel12:
                                    while (true) {
                                        try {
                                            J$.Fe(7432169, arguments.callee, this, arguments);
                                            arguments = J$.N(7432177, 'arguments', arguments, true, false, false);
                                            item = J$.N(7432185, 'item', item, true, false, false);
                                            J$.N(7432193, 'prevPadRight', prevPadRight, false, false, false);
                                            if (J$.C(395384, J$.U(741026, '!', J$.G(7431945, J$.R(7431937, 'item', item, false, false), 'kerning'))))
                                                return J$.Rt(7431953, undefined);
                                            var prevPadRight = J$.W(7432033, 'prevPadRight', J$.C(395392, J$.G(7431969, J$.R(7431961, 'item', item, false, false), 'first')) ? J$.T(7431977, 0, 22, false) : J$.G(7432025, J$.G(7432017, J$.R(7431985, 'output', output, false, false), J$.B(741034, '-', J$.G(7432001, J$.R(7431993, 'item', item, false, false), 'index'), J$.T(7432009, 1, 22, false))), 'padRight'), prevPadRight, false, false);
                                            if (J$.C(395408, J$.C(395400, J$.U(741042, '!', J$.G(7432049, J$.R(7432041, 'item', item, false, false), 'first'))) ? J$.B(741050, '<', J$.R(7432057, 'prevPadRight', prevPadRight, false, false), J$.G(7432073, J$.R(7432065, 'item', item, false, false), 'kerning')) : J$._()))
                                                J$.P(7432113, J$.R(7432081, 'item', item, false, false), 'padLeft', J$.B(741058, '-', J$.G(7432097, J$.R(7432089, 'item', item, false, false), 'kerning'), J$.R(7432105, 'prevPadRight', prevPadRight, false, false)));
                                            if (J$.C(395416, J$.U(741066, '!', J$.G(7432129, J$.R(7432121, 'item', item, false, false), 'last'))))
                                                J$.P(7432161, J$.R(7432137, 'item', item, false, false), 'padRight', J$.G(7432153, J$.R(7432145, 'item', item, false, false), 'kerning'));
                                        } catch (J$e) {
                                            J$.Ex(7434449, J$e);
                                        } finally {
                                            if (J$.Fr(7434457))
                                                continue jalangiLabel12;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.M(7432313, J$.R(7432217, 'output', output, false, false), 'forEach', false)(J$.T(7432305, function (item) {
                                jalangiLabel13:
                                    while (true) {
                                        try {
                                            J$.Fe(7432281, arguments.callee, this, arguments);
                                            arguments = J$.N(7432289, 'arguments', arguments, true, false, false);
                                            item = J$.N(7432297, 'item', item, true, false, false);
                                            if (J$.C(395424, J$.B(741074, '==', J$.M(7432233, J$.R(7432225, 'item', item, false, false), 'getBaseLength', false)(), J$.T(7432241, null, 25, false))))
                                                return J$.Rt(7432249, undefined);
                                            J$.F(7432273, J$.R(7432257, 'finishSizing', finishSizing, false, false), false)(J$.R(7432265, 'item', item, false, false));
                                        } catch (J$e) {
                                            J$.Ex(7434465, J$e);
                                        } finally {
                                            if (J$.Fr(7434473))
                                                continue jalangiLabel13;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            var resized = J$.W(7432329, 'resized', J$.T(7432321, 0, 22, false), resized, false, false);
                            var resizing;
                            var hunkSize;
                            do {
                                resizing = J$.W(7432345, 'resizing', J$.T(7432337, false, 23, false), resizing, false, false);
                                hunkSize = J$.W(7432385, 'hunkSize', J$.M(7432377, J$.I(typeof Math === 'undefined' ? Math = J$.R(7432353, 'Math', undefined, true, true) : Math = J$.R(7432353, 'Math', Math, true, true)), 'round', false)(J$.B(741082, '/', J$.R(7432361, 'remainingSpace', remainingSpace, false, false), J$.R(7432369, 'variableCount', variableCount, false, false))), hunkSize, false, false);
                                J$.M(7432561, J$.R(7432393, 'output', output, false, false), 'forEach', false)(J$.T(7432553, function (item) {
                                    jalangiLabel14:
                                        while (true) {
                                            try {
                                                J$.Fe(7432529, arguments.callee, this, arguments);
                                                arguments = J$.N(7432537, 'arguments', arguments, true, false, false);
                                                item = J$.N(7432545, 'item', item, true, false, false);
                                                if (J$.C(395432, J$.G(7432409, J$.R(7432401, 'item', item, false, false), 'finished')))
                                                    return J$.Rt(7432417, undefined);
                                                if (J$.C(395440, J$.U(741090, '!', J$.G(7432433, J$.R(7432425, 'item', item, false, false), 'maxLength'))))
                                                    return J$.Rt(7432441, undefined);
                                                if (J$.C(395448, J$.B(741098, '<', J$.M(7432457, J$.R(7432449, 'item', item, false, false), 'getMaxLength', false)(), J$.R(7432465, 'hunkSize', hunkSize, false, false)))) {
                                                    J$.F(7432505, J$.R(7432473, 'finishSizing', finishSizing, false, false), false)(J$.R(7432481, 'item', item, false, false), J$.G(7432497, J$.R(7432489, 'item', item, false, false), 'maxLength'));
                                                    resizing = J$.W(7432521, 'resizing', J$.T(7432513, true, 23, false), resizing, false, false);
                                                }
                                            } catch (J$e) {
                                                J$.Ex(7434481, J$e);
                                            } finally {
                                                if (J$.Fr(7434489))
                                                    continue jalangiLabel14;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            } while (J$.C(395464, J$.C(395456, J$.R(7432569, 'resizing', resizing, false, false)) ? J$.B(741130, '<', J$.B(741122, '-', resized = J$.W(7432585, 'resized', J$.B(741114, '+', J$.U(741106, '+', J$.R(7432577, 'resized', resized, false, false)), 1), resized, false, false), 1), J$.G(7432601, J$.R(7432593, 'output', output, false, false), 'length')) : J$._()));
                            if (J$.C(395472, J$.R(7432609, 'resizing', resizing, false, false)))
                                throw J$.M(7432633, J$.R(7432617, 'error', error, false, true), 'Internal', true)(J$.T(7432625, 'Resize loop iterated too many times while determining maxLength', 21, false));
                            resized = J$.W(7432649, 'resized', J$.T(7432641, 0, 22, false), resized, false, false);
                            do {
                                resizing = J$.W(7432665, 'resizing', J$.T(7432657, false, 23, false), resizing, false, false);
                                hunkSize = J$.W(7432705, 'hunkSize', J$.M(7432697, J$.I(typeof Math === 'undefined' ? Math = J$.R(7432673, 'Math', undefined, true, true) : Math = J$.R(7432673, 'Math', Math, true, true)), 'round', false)(J$.B(741138, '/', J$.R(7432681, 'remainingSpace', remainingSpace, false, false), J$.R(7432689, 'variableCount', variableCount, false, false))), hunkSize, false, false);
                                J$.M(7432881, J$.R(7432713, 'output', output, false, false), 'forEach', false)(J$.T(7432873, function (item) {
                                    jalangiLabel15:
                                        while (true) {
                                            try {
                                                J$.Fe(7432849, arguments.callee, this, arguments);
                                                arguments = J$.N(7432857, 'arguments', arguments, true, false, false);
                                                item = J$.N(7432865, 'item', item, true, false, false);
                                                if (J$.C(395480, J$.G(7432729, J$.R(7432721, 'item', item, false, false), 'finished')))
                                                    return J$.Rt(7432737, undefined);
                                                if (J$.C(395488, J$.U(741146, '!', J$.G(7432753, J$.R(7432745, 'item', item, false, false), 'minLength'))))
                                                    return J$.Rt(7432761, undefined);
                                                if (J$.C(395496, J$.B(741154, '>=', J$.M(7432777, J$.R(7432769, 'item', item, false, false), 'getMinLength', false)(), J$.R(7432785, 'hunkSize', hunkSize, false, false)))) {
                                                    J$.F(7432825, J$.R(7432793, 'finishSizing', finishSizing, false, false), false)(J$.R(7432801, 'item', item, false, false), J$.G(7432817, J$.R(7432809, 'item', item, false, false), 'minLength'));
                                                    resizing = J$.W(7432841, 'resizing', J$.T(7432833, true, 23, false), resizing, false, false);
                                                }
                                            } catch (J$e) {
                                                J$.Ex(7434497, J$e);
                                            } finally {
                                                if (J$.Fr(7434505))
                                                    continue jalangiLabel15;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false));
                            } while (J$.C(395512, J$.C(395504, J$.R(7432889, 'resizing', resizing, false, false)) ? J$.B(741186, '<', J$.B(741178, '-', resized = J$.W(7432905, 'resized', J$.B(741170, '+', J$.U(741162, '+', J$.R(7432897, 'resized', resized, false, false)), 1), resized, false, false), 1), J$.G(7432921, J$.R(7432913, 'output', output, false, false), 'length')) : J$._()));
                            if (J$.C(395520, J$.R(7432929, 'resizing', resizing, false, false)))
                                throw J$.M(7432953, J$.R(7432937, 'error', error, false, true), 'Internal', true)(J$.T(7432945, 'Resize loop iterated too many times while determining minLength', 21, false));
                            hunkSize = J$.W(7432993, 'hunkSize', J$.M(7432985, J$.I(typeof Math === 'undefined' ? Math = J$.R(7432961, 'Math', undefined, true, true) : Math = J$.R(7432961, 'Math', Math, true, true)), 'round', false)(J$.B(741194, '/', J$.R(7432969, 'remainingSpace', remainingSpace, false, false), J$.R(7432977, 'variableCount', variableCount, false, false))), hunkSize, false, false);
                            J$.M(7433097, J$.R(7433001, 'output', output, false, false), 'forEach', false)(J$.T(7433089, function (item) {
                                jalangiLabel16:
                                    while (true) {
                                        try {
                                            J$.Fe(7433065, arguments.callee, this, arguments);
                                            arguments = J$.N(7433073, 'arguments', arguments, true, false, false);
                                            item = J$.N(7433081, 'item', item, true, false, false);
                                            if (J$.C(395528, J$.G(7433017, J$.R(7433009, 'item', item, false, false), 'finished')))
                                                return J$.Rt(7433025, undefined);
                                            J$.F(7433057, J$.R(7433033, 'finishSizing', finishSizing, false, false), false)(J$.R(7433041, 'item', item, false, false), J$.R(7433049, 'hunkSize', hunkSize, false, false));
                                        } catch (J$e) {
                                            J$.Ex(7434513, J$e);
                                        } finally {
                                            if (J$.Fr(7434521))
                                                continue jalangiLabel16;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            return J$.Rt(7433113, J$.R(7433105, 'output', output, false, false));
                        } catch (J$e) {
                            J$.Ex(7434529, J$e);
                        } finally {
                            if (J$.Fr(7434537))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function renderFunction(item, values, length) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(7433449, arguments.callee, this, arguments);
                            arguments = J$.N(7433457, 'arguments', arguments, true, false, false);
                            item = J$.N(7433465, 'item', item, true, false, false);
                            values = J$.N(7433473, 'values', values, true, false, false);
                            length = J$.N(7433481, 'length', length, true, false, false);
                            J$.F(7433289, J$.R(7433265, 'validate', validate, false, true), false)(J$.T(7433273, 'OON', 21, false), J$.I(typeof arguments === 'undefined' ? arguments = J$.R(7433281, 'arguments', undefined, true, true) : arguments = J$.R(7433281, 'arguments', arguments, true, true)));
                            if (J$.C(395544, J$.G(7433305, J$.R(7433297, 'item', item, false, false), 'type'))) {
                                return J$.Rt(7433393, J$.M(7433385, J$.R(7433313, 'item', item, false, false), 'value', false)(J$.R(7433321, 'values', values, false, false), J$.C(395536, J$.G(7433361, J$.R(7433329, 'values', values, false, false), J$.B(741202, '+', J$.G(7433345, J$.R(7433337, 'item', item, false, false), 'type'), J$.T(7433353, 'Theme', 21, false)))) ? J$._() : J$.T(7433369, {}, 11, false), J$.R(7433377, 'length', length, false, false)));
                            } else {
                                return J$.Rt(7433441, J$.M(7433433, J$.R(7433401, 'item', item, false, false), 'value', false)(J$.R(7433409, 'values', values, false, false), J$.T(7433417, {}, 11, false), J$.R(7433425, 'length', length, false, false)));
                            }
                        } catch (J$e) {
                            J$.Ex(7434545, J$e);
                        } finally {
                            if (J$.Fr(7434553))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function renderValue(item, values) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(7433977, arguments.callee, this, arguments);
                            arguments = J$.N(7433985, 'arguments', arguments, true, false, false);
                            item = J$.N(7433993, 'item', item, true, false, false);
                            values = J$.N(7434001, 'values', values, true, false, false);
                            J$.N(7434009, 'length', length, false, false, false);
                            J$.N(7434017, 'value', value, false, false, false);
                            J$.N(7434025, 'alignWith', alignWith, false, false, false);
                            J$.N(7434033, 'leftPadding', leftPadding, false, false, false);
                            J$.N(7434041, 'rightPadding', rightPadding, false, false, false);
                            J$.N(7434049, 'truncated', truncated, false, false, false);
                            J$.N(7434057, 'aligned', aligned, false, false, false);
                            var length = J$.W(7433505, 'length', J$.M(7433497, J$.R(7433489, 'item', item, false, false), 'getBaseLength', false)(), length, false, false);
                            var value = J$.W(7433593, 'value', J$.C(395552, J$.B(741218, '===', J$.U(741210, 'typeof', J$.G(7433521, J$.R(7433513, 'item', item, false, false), 'value')), J$.T(7433529, 'function', 21, false))) ? J$.F(7433569, J$.R(7433537, 'renderFunction', renderFunction, false, true), false)(J$.R(7433545, 'item', item, false, false), J$.R(7433553, 'values', values, false, false), J$.R(7433561, 'length', length, false, false)) : J$.G(7433585, J$.R(7433577, 'item', item, false, false), 'value'), value, false, false);
                            if (J$.C(395568, J$.C(395560, J$.B(741226, '==', J$.R(7433601, 'value', value, false, false), J$.T(7433609, null, 25, false))) ? J$._() : J$.B(741234, '===', J$.R(7433617, 'value', value, false, false), J$.T(7433625, '', 21, false))))
                                return J$.Rt(7433641, J$.T(7433633, '', 21, false));
                            var alignWith = J$.W(7433697, 'alignWith', J$.C(395576, J$.G(7433673, J$.R(7433649, 'align', align, false, true), J$.G(7433665, J$.R(7433657, 'item', item, false, false), 'align'))) ? J$._() : J$.G(7433689, J$.R(7433681, 'align', align, false, true), 'left'), alignWith, false, false);
                            var leftPadding = J$.W(7433769, 'leftPadding', J$.C(395584, J$.G(7433713, J$.R(7433705, 'item', item, false, false), 'padLeft')) ? J$.M(7433753, J$.R(7433721, 'align', align, false, true), 'left', false)(J$.T(7433729, '', 21, false), J$.G(7433745, J$.R(7433737, 'item', item, false, false), 'padLeft')) : J$.T(7433761, '', 21, false), leftPadding, false, false);
                            var rightPadding = J$.W(7433841, 'rightPadding', J$.C(395592, J$.G(7433785, J$.R(7433777, 'item', item, false, false), 'padRight')) ? J$.M(7433825, J$.R(7433793, 'align', align, false, true), 'right', false)(J$.T(7433801, '', 21, false), J$.G(7433817, J$.R(7433809, 'item', item, false, false), 'padRight')) : J$.T(7433833, '', 21, false), rightPadding, false, false);
                            var truncated = J$.W(7433897, 'truncated', J$.F(7433889, J$.R(7433849, 'wideTruncate', wideTruncate, false, true), false)(J$.F(7433873, J$.I(typeof String === 'undefined' ? String = J$.R(7433857, 'String', undefined, true, true) : String = J$.R(7433857, 'String', String, true, true)), false)(J$.R(7433865, 'value', value, false, false)), J$.R(7433881, 'length', length, false, false)), truncated, false, false);
                            var aligned = J$.W(7433937, 'aligned', J$.F(7433929, J$.R(7433905, 'alignWith', alignWith, false, false), false)(J$.R(7433913, 'truncated', truncated, false, false), J$.R(7433921, 'length', length, false, false)), aligned, false, false);
                            return J$.Rt(7433969, J$.B(741250, '+', J$.B(741242, '+', J$.R(7433945, 'leftPadding', leftPadding, false, false), J$.R(7433953, 'aligned', aligned, false, false)), J$.R(7433961, 'rightPadding', rightPadding, false, false)));
                        } catch (J$e) {
                            J$.Ex(7434561, J$e);
                        } finally {
                            if (J$.Fr(7434569))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(7434073, 'align', align, false, false, false);
            J$.N(7434081, 'validate', validate, false, false, false);
            J$.N(7434089, 'objectAssign', objectAssign, false, false, false);
            J$.N(7434097, 'wideTruncate', wideTruncate, false, false, false);
            J$.N(7434105, 'error', error, false, false, false);
            J$.N(7434113, 'TemplateItem', TemplateItem, false, false, false);
            renderValueWithValues = J$.N(7434129, 'renderValueWithValues', J$.T(7434121, renderValueWithValues, 12, false), true, false, false);
            J$.N(7434137, 'renderTemplate', renderTemplate, false, false, false);
            preType = J$.N(7434153, 'preType', J$.T(7434145, preType, 12, false), true, false, false);
            postType = J$.N(7434169, 'postType', J$.T(7434161, postType, 12, false), true, false, false);
            hasPreOrPost = J$.N(7434185, 'hasPreOrPost', J$.T(7434177, hasPreOrPost, 12, false), true, false, false);
            generatePreAndPost = J$.N(7434201, 'generatePreAndPost', J$.T(7434193, generatePreAndPost, 12, false), true, false, false);
            prepareItems = J$.N(7434217, 'prepareItems', J$.T(7434209, prepareItems, 12, false), true, false, false);
            renderFunction = J$.N(7434233, 'renderFunction', J$.T(7434225, renderFunction, 12, false), true, false, false);
            renderValue = J$.N(7434249, 'renderValue', J$.T(7434241, renderValue, 12, false), true, false, false);
            J$.T(7429065, 'use strict', 21, false);
            var align = J$.W(7429097, 'align', J$.F(7429089, J$.I(typeof require === 'undefined' ? require = J$.R(7429073, 'require', undefined, true, true) : require = J$.R(7429073, 'require', require, true, true)), false)(J$.T(7429081, 'wide-align', 21, false)), align, false, true);
            var validate = J$.W(7429129, 'validate', J$.F(7429121, J$.I(typeof require === 'undefined' ? require = J$.R(7429105, 'require', undefined, true, true) : require = J$.R(7429105, 'require', require, true, true)), false)(J$.T(7429113, 'aproba', 21, false)), validate, false, true);
            var objectAssign = J$.W(7429161, 'objectAssign', J$.F(7429153, J$.I(typeof require === 'undefined' ? require = J$.R(7429137, 'require', undefined, true, true) : require = J$.R(7429137, 'require', require, true, true)), false)(J$.T(7429145, 'object-assign', 21, false)), objectAssign, false, true);
            var wideTruncate = J$.W(7429193, 'wideTruncate', J$.F(7429185, J$.I(typeof require === 'undefined' ? require = J$.R(7429169, 'require', undefined, true, true) : require = J$.R(7429169, 'require', require, true, true)), false)(J$.T(7429177, './wide-truncate', 21, false)), wideTruncate, false, true);
            var error = J$.W(7429225, 'error', J$.F(7429217, J$.I(typeof require === 'undefined' ? require = J$.R(7429201, 'require', undefined, true, true) : require = J$.R(7429201, 'require', require, true, true)), false)(J$.T(7429209, './error', 21, false)), error, false, true);
            var TemplateItem = J$.W(7429257, 'TemplateItem', J$.F(7429249, J$.I(typeof require === 'undefined' ? require = J$.R(7429233, 'require', undefined, true, true) : require = J$.R(7429233, 'require', require, true, true)), false)(J$.T(7429241, './template-item', 21, false)), TemplateItem, false, true);
            var renderTemplate = J$.W(7429625, 'renderTemplate', J$.P(7429617, J$.I(typeof module === 'undefined' ? module = J$.R(7429369, 'module', undefined, true, true) : module = J$.R(7429369, 'module', module, true, true)), 'exports', J$.T(7429609, function (width, template, values) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(7429553, arguments.callee, this, arguments);
                            arguments = J$.N(7429561, 'arguments', arguments, true, false, false);
                            width = J$.N(7429569, 'width', width, true, false, false);
                            template = J$.N(7429577, 'template', template, true, false, false);
                            values = J$.N(7429585, 'values', values, true, false, false);
                            J$.N(7429593, 'items', items, false, false, false);
                            J$.N(7429601, 'rendered', rendered, false, false, false);
                            var items = J$.W(7429417, 'items', J$.F(7429409, J$.R(7429377, 'prepareItems', prepareItems, false, true), false)(J$.R(7429385, 'width', width, false, false), J$.R(7429393, 'template', template, false, false), J$.R(7429401, 'values', values, false, false)), items, false, false);
                            var rendered = J$.W(7429481, 'rendered', J$.M(7429473, J$.M(7429457, J$.R(7429425, 'items', items, false, false), 'map', false)(J$.F(7429449, J$.R(7429433, 'renderValueWithValues', renderValueWithValues, false, true), false)(J$.R(7429441, 'values', values, false, false))), 'join', false)(J$.T(7429465, '', 21, false)), rendered, false, false);
                            return J$.Rt(7429545, J$.M(7429537, J$.R(7429489, 'align', align, false, true), 'left', false)(J$.F(7429521, J$.R(7429497, 'wideTruncate', wideTruncate, false, true), false)(J$.R(7429505, 'rendered', rendered, false, false), J$.R(7429513, 'width', width, false, false)), J$.R(7429529, 'width', width, false, false)));
                        } catch (J$e) {
                            J$.Ex(7434289, J$e);
                        } finally {
                            if (J$.Fr(7434297))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false)), renderTemplate, false, true);
        } catch (J$e) {
            J$.Ex(7434577, J$e);
        } finally {
            if (J$.Sr(7434585))
                continue jalangiLabel20;
            else
                break jalangiLabel20;
        }
    }
// JALANGI DO NOT INSTRUMENT
