// gauge/theme-set.js
J$.noInstrEval = false;
jalangiLabel13:
    while (true) {
        try {
            J$.Se(7440001, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCouchNode/node_modules/gauge/theme-set.js');
            J$.N(7440009, 'objectAssign', objectAssign, false, false, false);
            J$.N(7440017, 'ThemeSetProto', ThemeSetProto, false, false, false);
            J$.T(7436913, 'use strict', 21, false);
            var objectAssign = J$.W(7436945, 'objectAssign', J$.F(7436937, J$.I(typeof require === 'undefined' ? require = J$.R(7436921, 'require', undefined, true, true) : require = J$.R(7436921, 'require', require, true, true)), false)(J$.T(7436929, 'object-assign', 21, false)), objectAssign, false, true);
            J$.P(7437009, J$.I(typeof module === 'undefined' ? module = J$.R(7436953, 'module', undefined, true, true) : module = J$.R(7436953, 'module', module, true, true)), 'exports', J$.T(7437001, function () {
                jalangiLabel0:
                    while (true) {
                        try {
                            J$.Fe(7436985, arguments.callee, this, arguments);
                            arguments = J$.N(7436993, 'arguments', arguments, true, false, false);
                            return J$.Rt(7436977, J$.M(7436969, J$.R(7436961, 'ThemeSetProto', ThemeSetProto, false, true), 'newThemeSet', false)());
                        } catch (J$e) {
                            J$.Ex(7440025, J$e);
                        } finally {
                            if (J$.Fr(7440033))
                                continue jalangiLabel0;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var ThemeSetProto = J$.W(7437025, 'ThemeSetProto', J$.T(7437017, {}, 11, false), ThemeSetProto, false, true);
            J$.P(7437065, J$.R(7437033, 'ThemeSetProto', ThemeSetProto, false, true), 'baseTheme', J$.F(7437057, J$.I(typeof require === 'undefined' ? require = J$.R(7437041, 'require', undefined, true, true) : require = J$.R(7437041, 'require', require, true, true)), false)(J$.T(7437049, './base-theme.js', 21, false)));
            J$.P(7437217, J$.R(7437073, 'ThemeSetProto', ThemeSetProto, false, true), 'newTheme', J$.T(7437209, function (parent, theme) {
                jalangiLabel1:
                    while (true) {
                        try {
                            J$.Fe(7437177, arguments.callee, this, arguments);
                            arguments = J$.N(7437185, 'arguments', arguments, true, false, false);
                            parent = J$.N(7437193, 'parent', parent, true, false, false);
                            theme = J$.N(7437201, 'theme', theme, true, false, false);
                            if (J$.C(395696, J$.U(741466, '!', J$.R(7437081, 'theme', theme, false, false)))) {
                                theme = J$.W(7437097, 'theme', J$.R(7437089, 'parent', parent, false, false), theme, false, false);
                                parent = J$.W(7437121, 'parent', J$.G(7437113, J$.R(7437105, 'this', this, false, false), 'baseTheme'), parent, false, false);
                            }
                            return J$.Rt(7437169, J$.F(7437161, J$.R(7437129, 'objectAssign', objectAssign, false, true), false)(J$.T(7437137, {}, 11, false), J$.R(7437145, 'parent', parent, false, false), J$.R(7437153, 'theme', theme, false, false)));
                        } catch (J$e) {
                            J$.Ex(7440041, J$e);
                        } finally {
                            if (J$.Fr(7440049))
                                continue jalangiLabel1;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7437297, J$.R(7437225, 'ThemeSetProto', ThemeSetProto, false, true), 'getThemeNames', J$.T(7437289, function () {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(7437273, arguments.callee, this, arguments);
                            arguments = J$.N(7437281, 'arguments', arguments, true, false, false);
                            return J$.Rt(7437265, J$.M(7437257, J$.I(typeof Object === 'undefined' ? Object = J$.R(7437233, 'Object', undefined, true, true) : Object = J$.R(7437233, 'Object', Object, true, true)), 'keys', false)(J$.G(7437249, J$.R(7437241, 'this', this, false, false), 'themes')));
                        } catch (J$e) {
                            J$.Ex(7440057, J$e);
                        } finally {
                            if (J$.Fr(7440065))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7437425, J$.R(7437305, 'ThemeSetProto', ThemeSetProto, false, true), 'addTheme', J$.T(7437417, function (name, parent, theme) {
                jalangiLabel3:
                    while (true) {
                        try {
                            J$.Fe(7437377, arguments.callee, this, arguments);
                            arguments = J$.N(7437385, 'arguments', arguments, true, false, false);
                            name = J$.N(7437393, 'name', name, true, false, false);
                            parent = J$.N(7437401, 'parent', parent, true, false, false);
                            theme = J$.N(7437409, 'theme', theme, true, false, false);
                            J$.P(7437369, J$.G(7437321, J$.R(7437313, 'this', this, false, false), 'themes'), J$.R(7437329, 'name', name, false, false), J$.M(7437361, J$.R(7437337, 'this', this, false, false), 'newTheme', false)(J$.R(7437345, 'parent', parent, false, false), J$.R(7437353, 'theme', theme, false, false)));
                        } catch (J$e) {
                            J$.Ex(7440073, J$e);
                        } finally {
                            if (J$.Fr(7440081))
                                continue jalangiLabel3;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7437657, J$.R(7437433, 'ThemeSetProto', ThemeSetProto, false, true), 'addToAllThemes', J$.T(7437649, function (theme) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(7437617, arguments.callee, this, arguments);
                            arguments = J$.N(7437625, 'arguments', arguments, true, false, false);
                            theme = J$.N(7437633, 'theme', theme, true, false, false);
                            J$.N(7437641, 'themes', themes, false, false, false);
                            var themes = J$.W(7437457, 'themes', J$.G(7437449, J$.R(7437441, 'this', this, false, false), 'themes'), themes, false, false);
                            J$.M(7437569, J$.M(7437481, J$.I(typeof Object === 'undefined' ? Object = J$.R(7437465, 'Object', undefined, true, true) : Object = J$.R(7437465, 'Object', Object, true, true)), 'keys', false)(J$.R(7437473, 'themes', themes, false, false)), 'forEach', false)(J$.T(7437561, function (name) {
                                jalangiLabel4:
                                    while (true) {
                                        try {
                                            J$.Fe(7437537, arguments.callee, this, arguments);
                                            arguments = J$.N(7437545, 'arguments', arguments, true, false, false);
                                            name = J$.N(7437553, 'name', name, true, false, false);
                                            J$.F(7437529, J$.R(7437489, 'objectAssign', objectAssign, false, true), false)(J$.G(7437513, J$.R(7437497, 'themes', themes, false, false), J$.R(7437505, 'name', name, false, false)), J$.R(7437521, 'theme', theme, false, false));
                                        } catch (J$e) {
                                            J$.Ex(7440089, J$e);
                                        } finally {
                                            if (J$.Fr(7440097))
                                                continue jalangiLabel4;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false));
                            J$.F(7437609, J$.R(7437577, 'objectAssign', objectAssign, false, true), false)(J$.G(7437593, J$.R(7437585, 'this', this, false, false), 'baseTheme'), J$.R(7437601, 'theme', theme, false, false));
                        } catch (J$e) {
                            J$.Ex(7440105, J$e);
                        } finally {
                            if (J$.Fr(7440113))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7437801, J$.R(7437665, 'ThemeSetProto', ThemeSetProto, false, true), 'getTheme', J$.T(7437793, function (name) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(7437769, arguments.callee, this, arguments);
                            arguments = J$.N(7437777, 'arguments', arguments, true, false, false);
                            name = J$.N(7437785, 'name', name, true, false, false);
                            if (J$.C(395704, J$.U(741474, '!', J$.G(7437697, J$.G(7437681, J$.R(7437673, 'this', this, false, false), 'themes'), J$.R(7437689, 'name', name, false, false)))))
                                throw J$.M(7437721, J$.R(7437705, 'this', this, false, false), 'newMissingThemeError', false)(J$.R(7437713, 'name', name, false, false));
                            return J$.Rt(7437761, J$.G(7437753, J$.G(7437737, J$.R(7437729, 'this', this, false, false), 'themes'), J$.R(7437745, 'name', name, false, false)));
                        } catch (J$e) {
                            J$.Ex(7440121, J$e);
                        } finally {
                            if (J$.Fr(7440129))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7438193, J$.R(7437809, 'ThemeSetProto', ThemeSetProto, false, true), 'setDefault', J$.T(7438185, function (opts, name) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(7438129, arguments.callee, this, arguments);
                            arguments = J$.N(7438137, 'arguments', arguments, true, false, false);
                            opts = J$.N(7438145, 'opts', opts, true, false, false);
                            name = J$.N(7438153, 'name', name, true, false, false);
                            J$.N(7438161, 'platform', platform, false, false, false);
                            J$.N(7438169, 'hasUnicode', hasUnicode, false, false, false);
                            J$.N(7438177, 'hasColor', hasColor, false, false, false);
                            if (J$.C(395712, J$.B(741482, '==', J$.R(7437817, 'name', name, false, false), J$.T(7437825, null, 25, false)))) {
                                name = J$.W(7437841, 'name', J$.R(7437833, 'opts', opts, false, false), name, false, false);
                                opts = J$.W(7437857, 'opts', J$.T(7437849, {}, 11, false), opts, false, false);
                            }
                            var platform = J$.W(7437913, 'platform', J$.C(395720, J$.B(741490, '==', J$.G(7437873, J$.R(7437865, 'opts', opts, false, false), 'platform'), J$.T(7437881, null, 25, false))) ? J$.T(7437889, 'fallback', 21, false) : J$.G(7437905, J$.R(7437897, 'opts', opts, false, false), 'platform'), platform, false, false);
                            var hasUnicode = J$.W(7437937, 'hasUnicode', J$.U(741506, '!', J$.U(741498, '!', J$.G(7437929, J$.R(7437921, 'opts', opts, false, false), 'hasUnicode'))), hasUnicode, false, false);
                            var hasColor = J$.W(7437961, 'hasColor', J$.U(741522, '!', J$.U(741514, '!', J$.G(7437953, J$.R(7437945, 'opts', opts, false, false), 'hasColor'))), hasColor, false, false);
                            if (J$.C(395728, J$.U(741530, '!', J$.G(7437993, J$.G(7437977, J$.R(7437969, 'this', this, false, false), 'defaults'), J$.R(7437985, 'platform', platform, false, false)))))
                                J$.P(7438049, J$.G(7438009, J$.R(7438001, 'this', this, false, false), 'defaults'), J$.R(7438017, 'platform', platform, false, false), J$.T(7438041, {
                                    true: J$.T(7438025, {}, 11, false),
                                    false: J$.T(7438033, {}, 11, false)
                                }, 11, false));
                            J$.P(7438121, J$.G(7438097, J$.G(7438081, J$.G(7438065, J$.R(7438057, 'this', this, false, false), 'defaults'), J$.R(7438073, 'platform', platform, false, false)), J$.R(7438089, 'hasUnicode', hasUnicode, false, false)), J$.R(7438105, 'hasColor', hasColor, false, false), J$.R(7438113, 'name', name, false, false));
                        } catch (J$e) {
                            J$.Ex(7440137, J$e);
                        } finally {
                            if (J$.Fr(7440145))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7439161, J$.R(7438201, 'ThemeSetProto', ThemeSetProto, false, true), 'getDefault', J$.T(7439153, function (opts) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(7439097, arguments.callee, this, arguments);
                            arguments = J$.N(7439105, 'arguments', arguments, true, false, false);
                            opts = J$.N(7439113, 'opts', opts, true, false, false);
                            J$.N(7439121, 'platformName', platformName, false, false, false);
                            J$.N(7439129, 'platform', platform, false, false, false);
                            J$.N(7439137, 'hasUnicode', hasUnicode, false, false, false);
                            J$.N(7439145, 'hasColor', hasColor, false, false, false);
                            if (J$.C(395736, J$.U(741538, '!', J$.R(7438209, 'opts', opts, false, false))))
                                opts = J$.W(7438225, 'opts', J$.T(7438217, {}, 11, false), opts, false, false);
                            var platformName = J$.W(7438265, 'platformName', J$.C(395744, J$.G(7438241, J$.R(7438233, 'opts', opts, false, false), 'platform')) ? J$._() : J$.G(7438257, J$.I(typeof process === 'undefined' ? process = J$.R(7438249, 'process', undefined, true, true) : process = J$.R(7438249, 'process', process, true, true)), 'platform'), platformName, false, false);
                            var platform = J$.W(7438329, 'platform', J$.C(395752, J$.G(7438297, J$.G(7438281, J$.R(7438273, 'this', this, false, false), 'defaults'), J$.R(7438289, 'platformName', platformName, false, false))) ? J$._() : J$.G(7438321, J$.G(7438313, J$.R(7438305, 'this', this, false, false), 'defaults'), 'fallback'), platform, false, false);
                            var hasUnicode = J$.W(7438353, 'hasUnicode', J$.U(741554, '!', J$.U(741546, '!', J$.G(7438345, J$.R(7438337, 'opts', opts, false, false), 'hasUnicode'))), hasUnicode, false, false);
                            var hasColor = J$.W(7438377, 'hasColor', J$.U(741570, '!', J$.U(741562, '!', J$.G(7438369, J$.R(7438361, 'opts', opts, false, false), 'hasColor'))), hasColor, false, false);
                            if (J$.C(395760, J$.U(741578, '!', J$.R(7438385, 'platform', platform, false, false))))
                                throw J$.M(7438425, J$.R(7438393, 'this', this, false, false), 'newMissingDefaultThemeError', false)(J$.R(7438401, 'platformName', platformName, false, false), J$.R(7438409, 'hasUnicode', hasUnicode, false, false), J$.R(7438417, 'hasColor', hasColor, false, false));
                            if (J$.C(395896, J$.U(741586, '!', J$.G(7438465, J$.G(7438449, J$.R(7438433, 'platform', platform, false, false), J$.R(7438441, 'hasUnicode', hasUnicode, false, false)), J$.R(7438457, 'hasColor', hasColor, false, false))))) {
                                if (J$.C(395888, J$.C(395776, J$.C(395768, J$.R(7438473, 'hasUnicode', hasUnicode, false, false)) ? J$.R(7438481, 'hasColor', hasColor, false, false) : J$._()) ? J$.G(7438521, J$.G(7438505, J$.R(7438489, 'platform', platform, false, false), J$.U(741594, '!', J$.R(7438497, 'hasUnicode', hasUnicode, false, false))), J$.R(7438513, 'hasColor', hasColor, false, false)) : J$._())) {
                                    hasUnicode = J$.W(7438537, 'hasUnicode', J$.T(7438529, false, 23, false), hasUnicode, false, false);
                                } else if (J$.C(395880, J$.C(395792, J$.C(395784, J$.R(7438545, 'hasUnicode', hasUnicode, false, false)) ? J$.R(7438553, 'hasColor', hasColor, false, false) : J$._()) ? J$.G(7438593, J$.G(7438577, J$.R(7438561, 'platform', platform, false, false), J$.R(7438569, 'hasUnicode', hasUnicode, false, false)), J$.U(741602, '!', J$.R(7438585, 'hasColor', hasColor, false, false))) : J$._())) {
                                    hasColor = J$.W(7438609, 'hasColor', J$.T(7438601, false, 23, false), hasColor, false, false);
                                } else if (J$.C(395872, J$.C(395808, J$.C(395800, J$.R(7438617, 'hasUnicode', hasUnicode, false, false)) ? J$.R(7438625, 'hasColor', hasColor, false, false) : J$._()) ? J$.G(7438665, J$.G(7438649, J$.R(7438633, 'platform', platform, false, false), J$.U(741610, '!', J$.R(7438641, 'hasUnicode', hasUnicode, false, false))), J$.U(741618, '!', J$.R(7438657, 'hasColor', hasColor, false, false))) : J$._())) {
                                    hasUnicode = J$.W(7438681, 'hasUnicode', J$.T(7438673, false, 23, false), hasUnicode, false, false);
                                    hasColor = J$.W(7438697, 'hasColor', J$.T(7438689, false, 23, false), hasColor, false, false);
                                } else if (J$.C(395864, J$.C(395824, J$.C(395816, J$.R(7438705, 'hasUnicode', hasUnicode, false, false)) ? J$.U(741626, '!', J$.R(7438713, 'hasColor', hasColor, false, false)) : J$._()) ? J$.G(7438753, J$.G(7438737, J$.R(7438721, 'platform', platform, false, false), J$.U(741634, '!', J$.R(7438729, 'hasUnicode', hasUnicode, false, false))), J$.R(7438745, 'hasColor', hasColor, false, false)) : J$._())) {
                                    hasUnicode = J$.W(7438769, 'hasUnicode', J$.T(7438761, false, 23, false), hasUnicode, false, false);
                                } else if (J$.C(395856, J$.C(395840, J$.C(395832, J$.U(741642, '!', J$.R(7438777, 'hasUnicode', hasUnicode, false, false))) ? J$.R(7438785, 'hasColor', hasColor, false, false) : J$._()) ? J$.G(7438825, J$.G(7438809, J$.R(7438793, 'platform', platform, false, false), J$.R(7438801, 'hasUnicode', hasUnicode, false, false)), J$.U(741650, '!', J$.R(7438817, 'hasColor', hasColor, false, false))) : J$._())) {
                                    hasColor = J$.W(7438841, 'hasColor', J$.T(7438833, false, 23, false), hasColor, false, false);
                                } else if (J$.C(395848, J$.B(741658, '===', J$.R(7438849, 'platform', platform, false, false), J$.G(7438873, J$.G(7438865, J$.R(7438857, 'this', this, false, false), 'defaults'), 'fallback')))) {
                                    throw J$.M(7438913, J$.R(7438881, 'this', this, false, false), 'newMissingDefaultThemeError', false)(J$.R(7438889, 'platformName', platformName, false, false), J$.R(7438897, 'hasUnicode', hasUnicode, false, false), J$.R(7438905, 'hasColor', hasColor, false, false));
                                }
                            }
                            if (J$.C(395904, J$.G(7438953, J$.G(7438937, J$.R(7438921, 'platform', platform, false, false), J$.R(7438929, 'hasUnicode', hasUnicode, false, false)), J$.R(7438945, 'hasColor', hasColor, false, false)))) {
                                return J$.Rt(7439017, J$.M(7439009, J$.R(7438961, 'this', this, false, false), 'getTheme', false)(J$.G(7439001, J$.G(7438985, J$.R(7438969, 'platform', platform, false, false), J$.R(7438977, 'hasUnicode', hasUnicode, false, false)), J$.R(7438993, 'hasColor', hasColor, false, false))));
                            } else {
                                return J$.Rt(7439089, J$.M(7439081, J$.R(7439025, 'this', this, false, false), 'getDefault', false)(J$.F(7439073, J$.R(7439033, 'objectAssign', objectAssign, false, true), false)(J$.T(7439041, {}, 11, false), J$.R(7439049, 'opts', opts, false, false), J$.T(7439065, { platform: J$.T(7439057, 'fallback', 21, false) }, 11, false))));
                            }
                        } catch (J$e) {
                            J$.Ex(7440153, J$e);
                        } finally {
                            if (J$.Fr(7440161))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7439369, J$.R(7439169, 'ThemeSetProto', ThemeSetProto, false, true), 'newMissingThemeError', J$.T(7439361, function newMissingThemeError(name) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(7439329, arguments.callee, this, arguments);
                            arguments = J$.N(7439337, 'arguments', arguments, true, false, false);
                            name = J$.N(7439345, 'name', name, true, false, false);
                            J$.N(7439353, 'err', err, false, false, false);
                            var err = J$.W(7439217, 'err', J$.F(7439209, J$.I(typeof Error === 'undefined' ? Error = J$.R(7439177, 'Error', undefined, true, true) : Error = J$.R(7439177, 'Error', Error, true, true)), true)(J$.B(741674, '+', J$.B(741666, '+', J$.T(7439185, 'Could not find a gauge theme named "', 21, false), J$.R(7439193, 'name', name, false, false)), J$.T(7439201, '"', 21, false))), err, false, false);
                            J$.M(7439257, J$.G(7439233, J$.I(typeof Error === 'undefined' ? Error = J$.R(7439225, 'Error', undefined, true, true) : Error = J$.R(7439225, 'Error', Error, true, true)), 'captureStackTrace'), 'call', false)(J$.R(7439241, 'err', err, false, false), J$.R(7439249, 'newMissingThemeError', newMissingThemeError, false, false));
                            J$.P(7439281, J$.R(7439265, 'err', err, false, false), 'theme', J$.R(7439273, 'name', name, false, false));
                            J$.P(7439305, J$.R(7439289, 'err', err, false, false), 'code', J$.T(7439297, 'EMISSINGTHEME', 21, false));
                            return J$.Rt(7439321, J$.R(7439313, 'err', err, false, false));
                        } catch (J$e) {
                            J$.Ex(7440169, J$e);
                        } finally {
                            if (J$.Fr(7440177))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7439689, J$.R(7439377, 'ThemeSetProto', ThemeSetProto, false, true), 'newMissingDefaultThemeError', J$.T(7439681, function newMissingDefaultThemeError(platformName, hasUnicode, hasColor) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(7439633, arguments.callee, this, arguments);
                            arguments = J$.N(7439641, 'arguments', arguments, true, false, false);
                            platformName = J$.N(7439649, 'platformName', platformName, true, false, false);
                            hasUnicode = J$.N(7439657, 'hasUnicode', hasUnicode, true, false, false);
                            hasColor = J$.N(7439665, 'hasColor', hasColor, true, false, false);
                            J$.N(7439673, 'err', err, false, false, false);
                            var err = J$.W(7439473, 'err', J$.F(7439465, J$.I(typeof Error === 'undefined' ? Error = J$.R(7439385, 'Error', undefined, true, true) : Error = J$.R(7439385, 'Error', Error, true, true)), true)(J$.B(741738, '+', J$.B(741730, '+', J$.B(741722, '+', J$.B(741714, '+', J$.B(741706, '+', J$.B(741698, '+', J$.B(741690, '+', J$.B(741682, '+', J$.T(7439393, 'Could not find a gauge theme for your platform/unicode/color use combo:\n', 21, false), J$.T(7439401, '    platform = ', 21, false)), J$.R(7439409, 'platformName', platformName, false, false)), J$.T(7439417, '\n', 21, false)), J$.T(7439425, '    hasUnicode = ', 21, false)), J$.R(7439433, 'hasUnicode', hasUnicode, false, false)), J$.T(7439441, '\n', 21, false)), J$.T(7439449, '    hasColor = ', 21, false)), J$.R(7439457, 'hasColor', hasColor, false, false))), err, false, false);
                            J$.M(7439513, J$.G(7439489, J$.I(typeof Error === 'undefined' ? Error = J$.R(7439481, 'Error', undefined, true, true) : Error = J$.R(7439481, 'Error', Error, true, true)), 'captureStackTrace'), 'call', false)(J$.R(7439497, 'err', err, false, false), J$.R(7439505, 'newMissingDefaultThemeError', newMissingDefaultThemeError, false, false));
                            J$.P(7439537, J$.R(7439521, 'err', err, false, false), 'platform', J$.R(7439529, 'platformName', platformName, false, false));
                            J$.P(7439561, J$.R(7439545, 'err', err, false, false), 'hasUnicode', J$.R(7439553, 'hasUnicode', hasUnicode, false, false));
                            J$.P(7439585, J$.R(7439569, 'err', err, false, false), 'hasColor', J$.R(7439577, 'hasColor', hasColor, false, false));
                            J$.P(7439609, J$.R(7439593, 'err', err, false, false), 'code', J$.T(7439601, 'EMISSINGTHEME', 21, false));
                            return J$.Rt(7439625, J$.R(7439617, 'err', err, false, false));
                        } catch (J$e) {
                            J$.Ex(7440185, J$e);
                        } finally {
                            if (J$.Fr(7440193))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(7439993, J$.R(7439697, 'ThemeSetProto', ThemeSetProto, false, true), 'newThemeSet', J$.T(7439985, function () {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(7439961, arguments.callee, this, arguments);
                            arguments = J$.N(7439969, 'arguments', arguments, true, false, false);
                            J$.N(7439977, 'themeset', themeset, false, false, false);
                            var themeset = J$.W(7439769, 'themeset', J$.T(7439761, function (opts) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(7439737, arguments.callee, this, arguments);
                                            arguments = J$.N(7439745, 'arguments', arguments, true, false, false);
                                            opts = J$.N(7439753, 'opts', opts, true, false, false);
                                            return J$.Rt(7439729, J$.M(7439721, J$.R(7439705, 'themeset', themeset, false, false), 'getDefault', false)(J$.R(7439713, 'opts', opts, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(7440201, J$e);
                                        } finally {
                                            if (J$.Fr(7440209))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), themeset, false, false);
                            return J$.Rt(7439953, J$.F(7439945, J$.R(7439777, 'objectAssign', objectAssign, false, true), false)(J$.R(7439785, 'themeset', themeset, false, false), J$.R(7439793, 'ThemeSetProto', ThemeSetProto, false, true), J$.T(7439937, {
                                themes: J$.F(7439833, J$.R(7439801, 'objectAssign', objectAssign, false, true), false)(J$.T(7439809, {}, 11, false), J$.G(7439825, J$.R(7439817, 'this', this, false, false), 'themes')),
                                baseTheme: J$.F(7439873, J$.R(7439841, 'objectAssign', objectAssign, false, true), false)(J$.T(7439849, {}, 11, false), J$.G(7439865, J$.R(7439857, 'this', this, false, false), 'baseTheme')),
                                defaults: J$.M(7439929, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7439881, 'JSON', undefined, true, true) : JSON = J$.R(7439881, 'JSON', JSON, true, true)), 'parse', false)(J$.M(7439921, J$.I(typeof JSON === 'undefined' ? JSON = J$.R(7439889, 'JSON', undefined, true, true) : JSON = J$.R(7439889, 'JSON', JSON, true, true)), 'stringify', false)(J$.C(395912, J$.G(7439905, J$.R(7439897, 'this', this, false, false), 'defaults')) ? J$._() : J$.T(7439913, {}, 11, false)))
                            }, 11, false)));
                        } catch (J$e) {
                            J$.Ex(7440217, J$e);
                        } finally {
                            if (J$.Fr(7440225))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
        } catch (J$e) {
            J$.Ex(7440233, J$e);
        } finally {
            if (J$.Sr(7440241))
                continue jalangiLabel13;
            else
                break jalangiLabel13;
        }
    }
// JALANGI DO NOT INSTRUMENT
