// indicative/src/Sanitizer/sanitizations.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sanitizations = {
  escape: function escape() {
    return {
      name: 'escape',
      args: []
    };
  },
  lowerCase: function lowerCase(args) {
    return {
      name: 'lowerCase',
      args: args || []
    };
  },
  normalizeEmail: function normalizeEmail(args) {
    return {
      name: 'normalizeEmail',
      args: args || []
    };
  },
  plural: function plural() {
    return {
      name: 'plural',
      args: []
    };
  },
  singular: function singular() {
    return {
      name: 'singular',
      args: []
    };
  },
  slug: function slug() {
    return {
      name: 'slug',
      args: []
    };
  },
  stripLinks: function stripLinks() {
    return {
      name: 'stripLinks',
      args: []
    };
  },
  stripTags: function stripTags(args) {
    return {
      name: 'stripTags',
      args: args || []
    };
  },
  trim: function trim() {
    return {
      name: 'trim',
      args: []
    };
  },
  upperCase: function upperCase(args) {
    return {
      name: 'upperCase',
      args: args || []
    };
  }
};
