// indicative-formatters/build/src/VanillaFormatter.js
"use strict";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

Object.defineProperty(exports, "__esModule", {
  value: true
});

var VanillaFormatter =
/*#__PURE__*/
function () {
  function VanillaFormatter() {
    _classCallCheck(this, VanillaFormatter);

    this.errors = [];
  }

  _createClass(VanillaFormatter, [{
    key: "addError",
    value: function addError(error, field, rule) {
      var message = '';
      var validation = rule;

      if (error instanceof Error) {
        message = error.message;
        validation = 'ENGINE_EXCEPTION';
      } else {
        message = error;
      }

      this.errors.push({
        message: message,
        validation: validation,
        field: field
      });
    }
  }, {
    key: "toJSON",
    value: function toJSON() {
      return this.errors.length ? this.errors : null;
    }
  }]);

  return VanillaFormatter;
}();

exports.VanillaFormatter = VanillaFormatter;
