// asn1/lib/ber/types.js
J$.noInstrEval = false;
jalangiLabel0:
    while (true) {
        try {
            J$.Se(514505, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestCoinBasePro/node_modules/asn1/lib/ber/types.js');
            J$.T(514225, 'use strict', 21, false);
            J$.P(514497, J$.I(typeof module === 'undefined' ? module = J$.R(514233, 'module', undefined, true, true) : module = J$.R(514233, 'module', module, true, true)), 'exports', J$.T(514489, {
                EOC: J$.T(514241, 0, 22, false),
                Boolean: J$.T(514249, 1, 22, false),
                Integer: J$.T(514257, 2, 22, false),
                BitString: J$.T(514265, 3, 22, false),
                OctetString: J$.T(514273, 4, 22, false),
                Null: J$.T(514281, 5, 22, false),
                OID: J$.T(514289, 6, 22, false),
                ObjectDescriptor: J$.T(514297, 7, 22, false),
                External: J$.T(514305, 8, 22, false),
                Real: J$.T(514313, 9, 22, false),
                Enumeration: J$.T(514321, 10, 22, false),
                PDV: J$.T(514329, 11, 22, false),
                Utf8String: J$.T(514337, 12, 22, false),
                RelativeOID: J$.T(514345, 13, 22, false),
                Sequence: J$.T(514353, 16, 22, false),
                Set: J$.T(514361, 17, 22, false),
                NumericString: J$.T(514369, 18, 22, false),
                PrintableString: J$.T(514377, 19, 22, false),
                T61String: J$.T(514385, 20, 22, false),
                VideotexString: J$.T(514393, 21, 22, false),
                IA5String: J$.T(514401, 22, 22, false),
                UTCTime: J$.T(514409, 23, 22, false),
                GeneralizedTime: J$.T(514417, 24, 22, false),
                GraphicString: J$.T(514425, 25, 22, false),
                VisibleString: J$.T(514433, 26, 22, false),
                GeneralString: J$.T(514441, 28, 22, false),
                UniversalString: J$.T(514449, 29, 22, false),
                CharacterString: J$.T(514457, 30, 22, false),
                BMPString: J$.T(514465, 31, 22, false),
                Constructor: J$.T(514473, 32, 22, false),
                Context: J$.T(514481, 128, 22, false)
            }, 11, false));
        } catch (J$e) {
            J$.Ex(514513, J$e);
        } finally {
            if (J$.Sr(514521))
                continue jalangiLabel0;
            else
                break jalangiLabel0;
        }
    }
// JALANGI DO NOT INSTRUMENT
