// mongodb/lib/topologies/native_topology.js
J$.noInstrEval = false;
jalangiLabel26:
    while (true) {
        try {
            J$.Se(1343737, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/topologies/native_topology.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(1339553, arguments.callee, this, arguments);
                            arguments = J$.N(1339561, 'arguments', arguments, true, false, false);
                            obj = J$.N(1339569, 'obj', obj, true, false, false);
                            if (J$.C(84504, J$.C(84464, J$.B(127058, '===', J$.U(127050, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1339289, 'Symbol', undefined, true, true) : Symbol = J$.R(1339289, 'Symbol', Symbol, true, true))), J$.T(1339297, 'function', 21, false))) ? J$.B(127074, '===', J$.U(127066, 'typeof', J$.G(1339313, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1339305, 'Symbol', undefined, true, true) : Symbol = J$.R(1339305, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1339321, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(1339377, '_typeof2', J$.T(1339369, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(1339345, arguments.callee, this, arguments);
                                                arguments = J$.N(1339353, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1339361, 'obj', obj, true, false, false);
                                                return J$.Rt(1339337, J$.U(127082, 'typeof', J$.R(1339329, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1343977, J$e);
                                            } finally {
                                                if (J$.Fr(1343985))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(1339513, '_typeof2', J$.T(1339505, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(1339481, arguments.callee, this, arguments);
                                                arguments = J$.N(1339489, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1339497, 'obj', obj, true, false, false);
                                                return J$.Rt(1339473, J$.C(84496, J$.C(84488, J$.C(84480, J$.C(84472, J$.R(1339385, 'obj', obj, false, false)) ? J$.B(127098, '===', J$.U(127090, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1339393, 'Symbol', undefined, true, true) : Symbol = J$.R(1339393, 'Symbol', Symbol, true, true))), J$.T(1339401, 'function', 21, false)) : J$._()) ? J$.B(127106, '===', J$.G(1339417, J$.R(1339409, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1339425, 'Symbol', undefined, true, true) : Symbol = J$.R(1339425, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(127114, '!==', J$.R(1339433, 'obj', obj, false, false), J$.G(1339449, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1339441, 'Symbol', undefined, true, true) : Symbol = J$.R(1339441, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1339457, 'symbol', 21, false) : J$.U(127122, 'typeof', J$.R(1339465, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1343993, J$e);
                                            } finally {
                                                if (J$.Fr(1344001))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(1339545, J$.F(1339537, J$.R(1339521, '_typeof2', _typeof2, false, true), false)(J$.R(1339529, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1344009, J$e);
                        } finally {
                            if (J$.Fr(1344017))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(1339889, arguments.callee, this, arguments);
                            arguments = J$.N(1339897, 'arguments', arguments, true, false, false);
                            obj = J$.N(1339905, 'obj', obj, true, false, false);
                            if (J$.C(84552, J$.C(84512, J$.B(127138, '===', J$.U(127130, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1339577, 'Symbol', undefined, true, true) : Symbol = J$.R(1339577, 'Symbol', Symbol, true, true))), J$.T(1339585, 'function', 21, false))) ? J$.B(127146, '===', J$.F(1339617, J$.R(1339593, '_typeof2', _typeof2, false, true), false)(J$.G(1339609, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1339601, 'Symbol', undefined, true, true) : Symbol = J$.R(1339601, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1339625, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(1339697, '_typeof', J$.T(1339689, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(1339665, arguments.callee, this, arguments);
                                                arguments = J$.N(1339673, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1339681, 'obj', obj, true, false, false);
                                                return J$.Rt(1339657, J$.F(1339649, J$.R(1339633, '_typeof2', _typeof2, false, true), false)(J$.R(1339641, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1344025, J$e);
                                            } finally {
                                                if (J$.Fr(1344033))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(1339849, '_typeof', J$.T(1339841, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(1339817, arguments.callee, this, arguments);
                                                arguments = J$.N(1339825, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1339833, 'obj', obj, true, false, false);
                                                return J$.Rt(1339809, J$.C(84544, J$.C(84536, J$.C(84528, J$.C(84520, J$.R(1339705, 'obj', obj, false, false)) ? J$.B(127162, '===', J$.U(127154, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1339713, 'Symbol', undefined, true, true) : Symbol = J$.R(1339713, 'Symbol', Symbol, true, true))), J$.T(1339721, 'function', 21, false)) : J$._()) ? J$.B(127170, '===', J$.G(1339737, J$.R(1339729, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1339745, 'Symbol', undefined, true, true) : Symbol = J$.R(1339745, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(127178, '!==', J$.R(1339753, 'obj', obj, false, false), J$.G(1339769, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1339761, 'Symbol', undefined, true, true) : Symbol = J$.R(1339761, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1339777, 'symbol', 21, false) : J$.F(1339801, J$.R(1339785, '_typeof2', _typeof2, false, true), false)(J$.R(1339793, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1344041, J$e);
                                            } finally {
                                                if (J$.Fr(1344049))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(1339881, J$.F(1339873, J$.R(1339857, '_typeof', _typeof, false, true), false)(J$.R(1339865, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1344057, J$e);
                        } finally {
                            if (J$.Fr(1344065))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(1339953, arguments.callee, this, arguments);
                            arguments = J$.N(1339961, 'arguments', arguments, true, false, false);
                            instance = J$.N(1339969, 'instance', instance, true, false, false);
                            Constructor = J$.N(1339977, 'Constructor', Constructor, true, false, false);
                            if (J$.C(84560, J$.U(127194, '!', J$.B(127186, 'instanceof', J$.R(1339913, 'instance', instance, false, false), J$.R(1339921, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(1339945, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1339929, 'TypeError', undefined, true, true) : TypeError = J$.R(1339929, 'TypeError', TypeError, true, true)), true)(J$.T(1339937, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1344073, J$e);
                        } finally {
                            if (J$.Fr(1344081))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(1340225, arguments.callee, this, arguments);
                            arguments = J$.N(1340233, 'arguments', arguments, true, false, false);
                            target = J$.N(1340241, 'target', target, true, false, false);
                            props = J$.N(1340249, 'props', props, true, false, false);
                            J$.N(1340257, 'i', i, false, false, false);
                            J$.N(1340265, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(1339993, 'i', J$.T(1339985, 0, 22, false), i, false, false); J$.C(84584, J$.B(127202, '<', J$.R(1340001, 'i', i, false, false), J$.G(1340017, J$.R(1340009, 'props', props, false, false), 'length'))); J$.B(127226, '-', i = J$.W(1340033, 'i', J$.B(127218, '+', J$.U(127210, '+', J$.R(1340025, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(1340065, 'descriptor', J$.G(1340057, J$.R(1340041, 'props', props, false, false), J$.R(1340049, 'i', i, false, false)), descriptor, false, false);
                                J$.P(1340105, J$.R(1340073, 'descriptor', descriptor, false, false), 'enumerable', J$.C(84568, J$.G(1340089, J$.R(1340081, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(1340097, false, 23, false));
                                J$.P(1340129, J$.R(1340113, 'descriptor', descriptor, false, false), 'configurable', J$.T(1340121, true, 23, false));
                                if (J$.C(84576, J$.B(127234, 'in', J$.T(1340137, 'value', 21, false), J$.R(1340145, 'descriptor', descriptor, false, false))))
                                    J$.P(1340169, J$.R(1340153, 'descriptor', descriptor, false, false), 'writable', J$.T(1340161, true, 23, false));
                                J$.M(1340217, J$.I(typeof Object === 'undefined' ? Object = J$.R(1340177, 'Object', undefined, true, true) : Object = J$.R(1340177, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(1340185, 'target', target, false, false), J$.G(1340201, J$.R(1340193, 'descriptor', descriptor, false, false), 'key'), J$.R(1340209, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1344089, J$e);
                        } finally {
                            if (J$.Fr(1344097))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(1340377, arguments.callee, this, arguments);
                            arguments = J$.N(1340385, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(1340393, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(1340401, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(1340409, 'staticProps', staticProps, true, false, false);
                            if (J$.C(84592, J$.R(1340273, 'protoProps', protoProps, false, false)))
                                J$.F(1340313, J$.R(1340281, '_defineProperties', _defineProperties, false, true), false)(J$.G(1340297, J$.R(1340289, 'Constructor', Constructor, false, false), 'prototype'), J$.R(1340305, 'protoProps', protoProps, false, false));
                            if (J$.C(84600, J$.R(1340321, 'staticProps', staticProps, false, false)))
                                J$.F(1340353, J$.R(1340329, '_defineProperties', _defineProperties, false, true), false)(J$.R(1340337, 'Constructor', Constructor, false, false), J$.R(1340345, 'staticProps', staticProps, false, false));
                            return J$.Rt(1340369, J$.R(1340361, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(1344105, J$e);
                        } finally {
                            if (J$.Fr(1344113))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(1340521, arguments.callee, this, arguments);
                            arguments = J$.N(1340529, 'arguments', arguments, true, false, false);
                            self = J$.N(1340537, 'self', self, true, false, false);
                            call = J$.N(1340545, 'call', call, true, false, false);
                            if (J$.C(84624, J$.C(84616, J$.R(1340417, 'call', call, false, false)) ? J$.C(84608, J$.B(127242, '===', J$.F(1340441, J$.R(1340425, '_typeof', _typeof, false, true), false)(J$.R(1340433, 'call', call, false, false)), J$.T(1340449, 'object', 21, false))) ? J$._() : J$.B(127258, '===', J$.U(127250, 'typeof', J$.R(1340457, 'call', call, false, false)), J$.T(1340465, 'function', 21, false)) : J$._())) {
                                return J$.Rt(1340481, J$.R(1340473, 'call', call, false, false));
                            }
                            return J$.Rt(1340513, J$.F(1340505, J$.R(1340489, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(1340497, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(1344121, J$e);
                        } finally {
                            if (J$.Fr(1344129))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(1340609, arguments.callee, this, arguments);
                            arguments = J$.N(1340617, 'arguments', arguments, true, false, false);
                            self = J$.N(1340625, 'self', self, true, false, false);
                            if (J$.C(84632, J$.B(127266, '===', J$.R(1340553, 'self', self, false, false), void J$.T(1340561, 0, 22, false)))) {
                                throw J$.F(1340585, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(1340569, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(1340569, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(1340577, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(1340601, J$.R(1340593, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(1344137, J$e);
                        } finally {
                            if (J$.Fr(1344145))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _get(target, property, receiver) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(1340993, arguments.callee, this, arguments);
                            arguments = J$.N(1341001, 'arguments', arguments, true, false, false);
                            target = J$.N(1341009, 'target', target, true, false, false);
                            property = J$.N(1341017, 'property', property, true, false, false);
                            receiver = J$.N(1341025, 'receiver', receiver, true, false, false);
                            if (J$.C(84664, J$.C(84640, J$.B(127282, '!==', J$.U(127274, 'typeof', J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1340633, 'Reflect', undefined, true, true) : Reflect = J$.R(1340633, 'Reflect', Reflect, true, true))), J$.T(1340641, 'undefined', 21, false))) ? J$.G(1340657, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1340649, 'Reflect', undefined, true, true) : Reflect = J$.R(1340649, 'Reflect', Reflect, true, true)), 'get') : J$._())) {
                                _get = J$.W(1340681, '_get', J$.G(1340673, J$.I(typeof Reflect === 'undefined' ? Reflect = J$.R(1340665, 'Reflect', undefined, true, true) : Reflect = J$.R(1340665, 'Reflect', Reflect, true, true)), 'get'), _get, false, true);
                            } else {
                                _get = J$.W(1340929, '_get', J$.T(1340921, function _get(target, property, receiver) {
                                    jalangiLabel11:
                                        while (true) {
                                            try {
                                                J$.Fe(1340865, arguments.callee, this, arguments);
                                                arguments = J$.N(1340873, 'arguments', arguments, true, false, false);
                                                target = J$.N(1340881, 'target', target, true, false, false);
                                                property = J$.N(1340889, 'property', property, true, false, false);
                                                receiver = J$.N(1340897, 'receiver', receiver, true, false, false);
                                                J$.N(1340905, 'base', base, false, false, false);
                                                J$.N(1340913, 'desc', desc, false, false, false);
                                                var base = J$.W(1340721, 'base', J$.F(1340713, J$.R(1340689, '_superPropBase', _superPropBase, false, true), false)(J$.R(1340697, 'target', target, false, false), J$.R(1340705, 'property', property, false, false)), base, false, false);
                                                if (J$.C(84648, J$.U(127290, '!', J$.R(1340729, 'base', base, false, false))))
                                                    return J$.Rt(1340737, undefined);
                                                var desc = J$.W(1340777, 'desc', J$.M(1340769, J$.I(typeof Object === 'undefined' ? Object = J$.R(1340745, 'Object', undefined, true, true) : Object = J$.R(1340745, 'Object', Object, true, true)), 'getOwnPropertyDescriptor', false)(J$.R(1340753, 'base', base, false, false), J$.R(1340761, 'property', property, false, false)), desc, false, false);
                                                if (J$.C(84656, J$.G(1340793, J$.R(1340785, 'desc', desc, false, false), 'get'))) {
                                                    return J$.Rt(1340833, J$.M(1340825, J$.G(1340809, J$.R(1340801, 'desc', desc, false, false), 'get'), 'call', false)(J$.R(1340817, 'receiver', receiver, false, false)));
                                                }
                                                return J$.Rt(1340857, J$.G(1340849, J$.R(1340841, 'desc', desc, false, false), 'value'));
                                            } catch (J$e) {
                                                J$.Ex(1344153, J$e);
                                            } finally {
                                                if (J$.Fr(1344161))
                                                    continue jalangiLabel11;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _get, false, true);
                            }
                            return J$.Rt(1340985, J$.F(1340977, J$.R(1340937, '_get', _get, false, true), false)(J$.R(1340945, 'target', target, false, false), J$.R(1340953, 'property', property, false, false), J$.C(84672, J$.R(1340961, 'receiver', receiver, false, false)) ? J$._() : J$.R(1340969, 'target', target, false, false)));
                        } catch (J$e) {
                            J$.Ex(1344169, J$e);
                        } finally {
                            if (J$.Fr(1344177))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _superPropBase(object, property) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(1341145, arguments.callee, this, arguments);
                            arguments = J$.N(1341153, 'arguments', arguments, true, false, false);
                            object = J$.N(1341161, 'object', object, true, false, false);
                            property = J$.N(1341169, 'property', property, true, false, false);
                            while (J$.C(84688, J$.U(127298, '!', J$.M(1341073, J$.G(1341049, J$.G(1341041, J$.I(typeof Object === 'undefined' ? Object = J$.R(1341033, 'Object', undefined, true, true) : Object = J$.R(1341033, 'Object', Object, true, true)), 'prototype'), 'hasOwnProperty'), 'call', false)(J$.R(1341057, 'object', object, false, false), J$.R(1341065, 'property', property, false, false))))) {
                                object = J$.W(1341105, 'object', J$.F(1341097, J$.R(1341081, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1341089, 'object', object, false, false)), object, false, false);
                                if (J$.C(84680, J$.B(127306, '===', J$.R(1341113, 'object', object, false, false), J$.T(1341121, null, 25, false))))
                                    break;
                            }
                            return J$.Rt(1341137, J$.R(1341129, 'object', object, false, false));
                        } catch (J$e) {
                            J$.Ex(1344185, J$e);
                        } finally {
                            if (J$.Fr(1344193))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(1341329, arguments.callee, this, arguments);
                            arguments = J$.N(1341337, 'arguments', arguments, true, false, false);
                            o = J$.N(1341345, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(1341289, '_getPrototypeOf', J$.C(84704, J$.G(1341185, J$.I(typeof Object === 'undefined' ? Object = J$.R(1341177, 'Object', undefined, true, true) : Object = J$.R(1341177, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(1341201, J$.I(typeof Object === 'undefined' ? Object = J$.R(1341193, 'Object', undefined, true, true) : Object = J$.R(1341193, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(1341281, function _getPrototypeOf(o) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(1341257, arguments.callee, this, arguments);
                                            arguments = J$.N(1341265, 'arguments', arguments, true, false, false);
                                            o = J$.N(1341273, 'o', o, true, false, false);
                                            return J$.Rt(1341249, J$.C(84696, J$.G(1341217, J$.R(1341209, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(1341241, J$.I(typeof Object === 'undefined' ? Object = J$.R(1341225, 'Object', undefined, true, true) : Object = J$.R(1341225, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(1341233, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(1344201, J$e);
                                        } finally {
                                            if (J$.Fr(1344209))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(1341321, J$.F(1341313, J$.R(1341297, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1341305, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(1344217, J$e);
                        } finally {
                            if (J$.Fr(1344225))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel16:
                    while (true) {
                        try {
                            J$.Fe(1341545, arguments.callee, this, arguments);
                            arguments = J$.N(1341553, 'arguments', arguments, true, false, false);
                            subClass = J$.N(1341561, 'subClass', subClass, true, false, false);
                            superClass = J$.N(1341569, 'superClass', superClass, true, false, false);
                            if (J$.C(84720, J$.C(84712, J$.B(127322, '!==', J$.U(127314, 'typeof', J$.R(1341353, 'superClass', superClass, false, false)), J$.T(1341361, 'function', 21, false))) ? J$.B(127330, '!==', J$.R(1341369, 'superClass', superClass, false, false), J$.T(1341377, null, 25, false)) : J$._())) {
                                throw J$.F(1341401, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1341385, 'TypeError', undefined, true, true) : TypeError = J$.R(1341385, 'TypeError', TypeError, true, true)), true)(J$.T(1341393, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(1341497, J$.R(1341409, 'subClass', subClass, false, false), 'prototype', J$.M(1341489, J$.I(typeof Object === 'undefined' ? Object = J$.R(1341417, 'Object', undefined, true, true) : Object = J$.R(1341417, 'Object', Object, true, true)), 'create', false)(J$.C(84728, J$.R(1341425, 'superClass', superClass, false, false)) ? J$.G(1341441, J$.R(1341433, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(1341481, {
                                constructor: J$.T(1341473, {
                                    value: J$.R(1341449, 'subClass', subClass, false, false),
                                    writable: J$.T(1341457, true, 23, false),
                                    configurable: J$.T(1341465, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(84736, J$.R(1341505, 'superClass', superClass, false, false)))
                                J$.F(1341537, J$.R(1341513, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1341521, 'subClass', subClass, false, false), J$.R(1341529, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(1344233, J$e);
                        } finally {
                            if (J$.Fr(1344241))
                                continue jalangiLabel16;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(1341721, arguments.callee, this, arguments);
                            arguments = J$.N(1341729, 'arguments', arguments, true, false, false);
                            o = J$.N(1341737, 'o', o, true, false, false);
                            p = J$.N(1341745, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(1341673, '_setPrototypeOf', J$.C(84744, J$.G(1341585, J$.I(typeof Object === 'undefined' ? Object = J$.R(1341577, 'Object', undefined, true, true) : Object = J$.R(1341577, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(1341665, function _setPrototypeOf(o, p) {
                                jalangiLabel17:
                                    while (true) {
                                        try {
                                            J$.Fe(1341633, arguments.callee, this, arguments);
                                            arguments = J$.N(1341641, 'arguments', arguments, true, false, false);
                                            o = J$.N(1341649, 'o', o, true, false, false);
                                            p = J$.N(1341657, 'p', p, true, false, false);
                                            J$.P(1341609, J$.R(1341593, 'o', o, false, false), '__proto__', J$.R(1341601, 'p', p, false, false));
                                            return J$.Rt(1341625, J$.R(1341617, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1344249, J$e);
                                        } finally {
                                            if (J$.Fr(1344257))
                                                continue jalangiLabel17;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(1341713, J$.F(1341705, J$.R(1341681, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1341689, 'o', o, false, false), J$.R(1341697, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(1344265, J$e);
                        } finally {
                            if (J$.Fr(1344273))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(1343753, '_typeof2', J$.T(1343745, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(1343769, '_typeof', J$.T(1343761, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(1343785, '_classCallCheck', J$.T(1343777, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(1343801, '_defineProperties', J$.T(1343793, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(1343817, '_createClass', J$.T(1343809, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(1343833, '_possibleConstructorReturn', J$.T(1343825, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(1343849, '_assertThisInitialized', J$.T(1343841, _assertThisInitialized, 12, false), true, false, false);
            _get = J$.N(1343865, '_get', J$.T(1343857, _get, 12, false), true, false, false);
            _superPropBase = J$.N(1343881, '_superPropBase', J$.T(1343873, _superPropBase, 12, false), true, false, false);
            _getPrototypeOf = J$.N(1343897, '_getPrototypeOf', J$.T(1343889, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(1343913, '_inherits', J$.T(1343905, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(1343929, '_setPrototypeOf', J$.T(1343921, _setPrototypeOf, 12, false), true, false, false);
            J$.N(1343937, 'Topology', Topology, false, false, false);
            J$.N(1343945, 'ServerCapabilities', ServerCapabilities, false, false, false);
            J$.N(1343953, 'Cursor', Cursor, false, false, false);
            J$.N(1343961, 'translateOptions', translateOptions, false, false, false);
            J$.N(1343969, 'NativeTopology', NativeTopology, false, false, false);
            J$.T(1339281, 'use strict', 21, false);
            var Topology = J$.W(1341785, 'Topology', J$.G(1341777, J$.F(1341769, J$.I(typeof require === 'undefined' ? require = J$.R(1341753, 'require', undefined, true, true) : require = J$.R(1341753, 'require', require, true, true)), false)(J$.T(1341761, '../core', 21, false)), 'Topology'), Topology, false, true);
            var ServerCapabilities = J$.W(1341825, 'ServerCapabilities', J$.G(1341817, J$.F(1341809, J$.I(typeof require === 'undefined' ? require = J$.R(1341793, 'require', undefined, true, true) : require = J$.R(1341793, 'require', require, true, true)), false)(J$.T(1341801, './topology_base', 21, false)), 'ServerCapabilities'), ServerCapabilities, false, true);
            var Cursor = J$.W(1341857, 'Cursor', J$.F(1341849, J$.I(typeof require === 'undefined' ? require = J$.R(1341833, 'require', undefined, true, true) : require = J$.R(1341833, 'require', require, true, true)), false)(J$.T(1341841, '../cursor', 21, false)), Cursor, false, true);
            var translateOptions = J$.W(1341897, 'translateOptions', J$.G(1341889, J$.F(1341881, J$.I(typeof require === 'undefined' ? require = J$.R(1341865, 'require', undefined, true, true) : require = J$.R(1341865, 'require', require, true, true)), false)(J$.T(1341873, '../utils', 21, false)), 'translateOptions'), translateOptions, false, true);
            var NativeTopology = J$.W(1343705, 'NativeTopology', J$.F(1343697, J$.T(1343681, function (_Topology) {
                jalangiLabel25:
                    while (true) {
                        try {
                            J$.Fe(1343641, arguments.callee, this, arguments);
                            function NativeTopology(servers, options) {
                                jalangiLabel19:
                                    while (true) {
                                        try {
                                            J$.Fe(1342545, arguments.callee, this, arguments);
                                            arguments = J$.N(1342553, 'arguments', arguments, true, false, false);
                                            servers = J$.N(1342561, 'servers', servers, true, false, false);
                                            options = J$.N(1342569, 'options', options, true, false, false);
                                            J$.N(1342577, '_this', _this, false, false, false);
                                            J$.N(1342585, 'clonedOptions', clonedOptions, false, false, false);
                                            J$.N(1342593, 'socketOptions', socketOptions, false, false, false);
                                            var _this;
                                            J$.F(1341961, J$.R(1341937, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(1341945, 'this', this, false, false), J$.R(1341953, 'NativeTopology', NativeTopology, false, false));
                                            options = J$.W(1341985, 'options', J$.C(84752, J$.R(1341969, 'options', options, false, false)) ? J$._() : J$.T(1341977, {}, 11, false), options, false, false);
                                            var clonedOptions = J$.W(1342185, 'clonedOptions', J$.M(1342177, J$.I(typeof Object === 'undefined' ? Object = J$.R(1341993, 'Object', undefined, true, true) : Object = J$.R(1341993, 'Object', Object, true, true)), 'assign', false)(J$.T(1342001, {}, 11, false), J$.T(1342169, {
                                                cursorFactory: J$.R(1342009, 'Cursor', Cursor, false, true),
                                                reconnect: J$.T(1342017, false, 23, false),
                                                emitError: J$.C(84760, J$.B(127346, '===', J$.U(127338, 'typeof', J$.G(1342033, J$.R(1342025, 'options', options, false, false), 'emitError')), J$.T(1342041, 'boolean', 21, false))) ? J$.G(1342057, J$.R(1342049, 'options', options, false, false), 'emitError') : J$.T(1342065, true, 23, false),
                                                size: J$.C(84768, J$.B(127362, '===', J$.U(127354, 'typeof', J$.G(1342081, J$.R(1342073, 'options', options, false, false), 'poolSize')), J$.T(1342089, 'number', 21, false))) ? J$.G(1342105, J$.R(1342097, 'options', options, false, false), 'poolSize') : J$.T(1342113, 5, 22, false),
                                                monitorCommands: J$.C(84776, J$.B(127378, '===', J$.U(127370, 'typeof', J$.G(1342129, J$.R(1342121, 'options', options, false, false), 'monitorCommands')), J$.T(1342137, 'boolean', 21, false))) ? J$.G(1342153, J$.R(1342145, 'options', options, false, false), 'monitorCommands') : J$.T(1342161, false, 23, false)
                                            }, 11, false)), clonedOptions, false, false);
                                            clonedOptions = J$.W(1342225, 'clonedOptions', J$.F(1342217, J$.R(1342193, 'translateOptions', translateOptions, false, true), false)(J$.R(1342201, 'clonedOptions', clonedOptions, false, false), J$.R(1342209, 'options', options, false, false)), clonedOptions, false, false);
                                            var socketOptions = J$.W(1342321, 'socketOptions', J$.C(84792, J$.C(84784, J$.G(1342241, J$.R(1342233, 'options', options, false, false), 'socketOptions')) ? J$.B(127386, '>', J$.G(1342281, J$.M(1342273, J$.I(typeof Object === 'undefined' ? Object = J$.R(1342249, 'Object', undefined, true, true) : Object = J$.R(1342249, 'Object', Object, true, true)), 'keys', false)(J$.G(1342265, J$.R(1342257, 'options', options, false, false), 'socketOptions')), 'length'), J$.T(1342289, 0, 22, false)) : J$._()) ? J$.G(1342305, J$.R(1342297, 'options', options, false, false), 'socketOptions') : J$.R(1342313, 'options', options, false, false), socketOptions, false, false);
                                            clonedOptions = J$.W(1342361, 'clonedOptions', J$.F(1342353, J$.R(1342329, 'translateOptions', translateOptions, false, true), false)(J$.R(1342337, 'clonedOptions', clonedOptions, false, false), J$.R(1342345, 'socketOptions', socketOptions, false, false)), clonedOptions, false, false);
                                            _this = J$.W(1342449, '_this', J$.F(1342441, J$.R(1342369, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(1342377, 'this', this, false, false), J$.M(1342433, J$.F(1342401, J$.R(1342385, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1342393, 'NativeTopology', NativeTopology, false, false)), 'call', false)(J$.R(1342409, 'this', this, false, false), J$.R(1342417, 'servers', servers, false, false), J$.R(1342425, 'clonedOptions', clonedOptions, false, false))), _this, false, false);
                                            if (J$.C(84800, J$.G(1342465, J$.R(1342457, 'options', options, false, false), 'appname'))) {
                                                J$.P(1342521, J$.G(1342489, J$.G(1342481, J$.R(1342473, '_this', _this, false, false), 's'), 'clientInfo'), 'application', J$.T(1342513, { name: J$.G(1342505, J$.R(1342497, 'options', options, false, false), 'appname') }, 11, false));
                                            }
                                            return J$.Rt(1342537, J$.R(1342529, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1344281, J$e);
                                        } finally {
                                            if (J$.Fr(1344289))
                                                continue jalangiLabel19;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(1343649, 'arguments', arguments, true, false, false);
                            _Topology = J$.N(1343657, '_Topology', _Topology, true, false, false);
                            NativeTopology = J$.N(1343673, 'NativeTopology', J$.T(1343665, NativeTopology, 12, false), true, false, false);
                            J$.F(1341929, J$.R(1341905, '_inherits', _inherits, false, true), false)(J$.R(1341913, 'NativeTopology', NativeTopology, false, false), J$.R(1341921, '_Topology', _Topology, false, false));
                            J$.F(1343617, J$.R(1342601, '_createClass', _createClass, false, true), false)(J$.R(1342609, 'NativeTopology', NativeTopology, false, false), J$.T(1343609, [
                                J$.T(1342833, {
                                    key: J$.T(1342617, 'capabilities', 21, false),
                                    value: J$.T(1342825, function capabilities() {
                                        jalangiLabel20:
                                            while (true) {
                                                try {
                                                    J$.Fe(1342809, arguments.callee, this, arguments);
                                                    arguments = J$.N(1342817, 'arguments', arguments, true, false, false);
                                                    if (J$.C(84808, J$.G(1342641, J$.G(1342633, J$.R(1342625, 'this', this, false, false), 's'), 'sCapabilities')))
                                                        return J$.Rt(1342673, J$.G(1342665, J$.G(1342657, J$.R(1342649, 'this', this, false, false), 's'), 'sCapabilities'));
                                                    if (J$.C(84816, J$.B(127394, '==', J$.M(1342689, J$.R(1342681, 'this', this, false, false), 'lastIsMaster', false)(), J$.T(1342697, null, 25, false))))
                                                        return J$.Rt(1342713, J$.T(1342705, null, 25, false));
                                                    J$.P(1342769, J$.G(1342729, J$.R(1342721, 'this', this, false, false), 's'), 'sCapabilities', J$.F(1342761, J$.R(1342737, 'ServerCapabilities', ServerCapabilities, false, true), true)(J$.M(1342753, J$.R(1342745, 'this', this, false, false), 'lastIsMaster', false)()));
                                                    return J$.Rt(1342801, J$.G(1342793, J$.G(1342785, J$.R(1342777, 'this', this, false, false), 's'), 'sCapabilities'));
                                                } catch (J$e) {
                                                    J$.Ex(1344297, J$e);
                                                } finally {
                                                    if (J$.Fr(1344305))
                                                        continue jalangiLabel20;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1343025, {
                                    key: J$.T(1342841, 'command', 21, false),
                                    value: J$.T(1343017, function command(ns, cmd, options, callback) {
                                        jalangiLabel21:
                                            while (true) {
                                                try {
                                                    J$.Fe(1342969, arguments.callee, this, arguments);
                                                    arguments = J$.N(1342977, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(1342985, 'ns', ns, true, false, false);
                                                    cmd = J$.N(1342993, 'cmd', cmd, true, false, false);
                                                    options = J$.N(1343001, 'options', options, true, false, false);
                                                    callback = J$.N(1343009, 'callback', callback, true, false, false);
                                                    J$.M(1342961, J$.F(1342905, J$.R(1342849, '_get', _get, false, true), false)(J$.F(1342881, J$.R(1342857, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(1342873, J$.R(1342865, 'NativeTopology', NativeTopology, false, false), 'prototype')), J$.T(1342889, 'command', 21, false), J$.R(1342897, 'this', this, false, false)), 'call', false)(J$.R(1342913, 'this', this, false, false), J$.M(1342929, J$.R(1342921, 'ns', ns, false, false), 'toString', false)(), J$.R(1342937, 'cmd', cmd, false, false), J$.R(1342945, 'options', options, false, false), J$.R(1342953, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1344313, J$e);
                                                } finally {
                                                    if (J$.Fr(1344321))
                                                        continue jalangiLabel21;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1343217, {
                                    key: J$.T(1343033, 'insert', 21, false),
                                    value: J$.T(1343209, function insert(ns, ops, options, callback) {
                                        jalangiLabel22:
                                            while (true) {
                                                try {
                                                    J$.Fe(1343161, arguments.callee, this, arguments);
                                                    arguments = J$.N(1343169, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(1343177, 'ns', ns, true, false, false);
                                                    ops = J$.N(1343185, 'ops', ops, true, false, false);
                                                    options = J$.N(1343193, 'options', options, true, false, false);
                                                    callback = J$.N(1343201, 'callback', callback, true, false, false);
                                                    J$.M(1343153, J$.F(1343097, J$.R(1343041, '_get', _get, false, true), false)(J$.F(1343073, J$.R(1343049, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(1343065, J$.R(1343057, 'NativeTopology', NativeTopology, false, false), 'prototype')), J$.T(1343081, 'insert', 21, false), J$.R(1343089, 'this', this, false, false)), 'call', false)(J$.R(1343105, 'this', this, false, false), J$.M(1343121, J$.R(1343113, 'ns', ns, false, false), 'toString', false)(), J$.R(1343129, 'ops', ops, false, false), J$.R(1343137, 'options', options, false, false), J$.R(1343145, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1344329, J$e);
                                                } finally {
                                                    if (J$.Fr(1344337))
                                                        continue jalangiLabel22;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1343409, {
                                    key: J$.T(1343225, 'update', 21, false),
                                    value: J$.T(1343401, function update(ns, ops, options, callback) {
                                        jalangiLabel23:
                                            while (true) {
                                                try {
                                                    J$.Fe(1343353, arguments.callee, this, arguments);
                                                    arguments = J$.N(1343361, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(1343369, 'ns', ns, true, false, false);
                                                    ops = J$.N(1343377, 'ops', ops, true, false, false);
                                                    options = J$.N(1343385, 'options', options, true, false, false);
                                                    callback = J$.N(1343393, 'callback', callback, true, false, false);
                                                    J$.M(1343345, J$.F(1343289, J$.R(1343233, '_get', _get, false, true), false)(J$.F(1343265, J$.R(1343241, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(1343257, J$.R(1343249, 'NativeTopology', NativeTopology, false, false), 'prototype')), J$.T(1343273, 'update', 21, false), J$.R(1343281, 'this', this, false, false)), 'call', false)(J$.R(1343297, 'this', this, false, false), J$.M(1343313, J$.R(1343305, 'ns', ns, false, false), 'toString', false)(), J$.R(1343321, 'ops', ops, false, false), J$.R(1343329, 'options', options, false, false), J$.R(1343337, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1344345, J$e);
                                                } finally {
                                                    if (J$.Fr(1344353))
                                                        continue jalangiLabel23;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1343601, {
                                    key: J$.T(1343417, 'remove', 21, false),
                                    value: J$.T(1343593, function remove(ns, ops, options, callback) {
                                        jalangiLabel24:
                                            while (true) {
                                                try {
                                                    J$.Fe(1343545, arguments.callee, this, arguments);
                                                    arguments = J$.N(1343553, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(1343561, 'ns', ns, true, false, false);
                                                    ops = J$.N(1343569, 'ops', ops, true, false, false);
                                                    options = J$.N(1343577, 'options', options, true, false, false);
                                                    callback = J$.N(1343585, 'callback', callback, true, false, false);
                                                    J$.M(1343537, J$.F(1343481, J$.R(1343425, '_get', _get, false, true), false)(J$.F(1343457, J$.R(1343433, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.G(1343449, J$.R(1343441, 'NativeTopology', NativeTopology, false, false), 'prototype')), J$.T(1343465, 'remove', 21, false), J$.R(1343473, 'this', this, false, false)), 'call', false)(J$.R(1343489, 'this', this, false, false), J$.M(1343505, J$.R(1343497, 'ns', ns, false, false), 'toString', false)(), J$.R(1343513, 'ops', ops, false, false), J$.R(1343521, 'options', options, false, false), J$.R(1343529, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1344361, J$e);
                                                } finally {
                                                    if (J$.Fr(1344369))
                                                        continue jalangiLabel24;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(1343633, J$.R(1343625, 'NativeTopology', NativeTopology, false, false));
                        } catch (J$e) {
                            J$.Ex(1344377, J$e);
                        } finally {
                            if (J$.Fr(1344385))
                                continue jalangiLabel25;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(1343689, 'Topology', Topology, false, true)), NativeTopology, false, true);
            J$.P(1343729, J$.I(typeof module === 'undefined' ? module = J$.R(1343713, 'module', undefined, true, true) : module = J$.R(1343713, 'module', module, true, true)), 'exports', J$.R(1343721, 'NativeTopology', NativeTopology, false, true));
        } catch (J$e) {
            J$.Ex(1344393, J$e);
        } finally {
            if (J$.Sr(1344401))
                continue jalangiLabel26;
            else
                break jalangiLabel26;
        }
    }
// JALANGI DO NOT INSTRUMENT
