// mongodb/lib/topologies/topology_base.js
J$.noInstrEval = false;
jalangiLabel50:
    while (true) {
        try {
            J$.Se(1374041, '/Users/ex1t/Desktop/nodeAnalysis/HiPar/outputs/target_cache/TestMongoDb/node_modules/mongodb/lib/topologies/topology_base.js');
            function _typeof2(obj) {
                jalangiLabel2:
                    while (true) {
                        try {
                            J$.Fe(1363529, arguments.callee, this, arguments);
                            arguments = J$.N(1363537, 'arguments', arguments, true, false, false);
                            obj = J$.N(1363545, 'obj', obj, true, false, false);
                            if (J$.C(85712, J$.C(85672, J$.B(128394, '===', J$.U(128386, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1363265, 'Symbol', undefined, true, true) : Symbol = J$.R(1363265, 'Symbol', Symbol, true, true))), J$.T(1363273, 'function', 21, false))) ? J$.B(128410, '===', J$.U(128402, 'typeof', J$.G(1363289, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1363281, 'Symbol', undefined, true, true) : Symbol = J$.R(1363281, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1363297, 'symbol', 21, false)) : J$._())) {
                                _typeof2 = J$.W(1363353, '_typeof2', J$.T(1363345, function _typeof2(obj) {
                                    jalangiLabel0:
                                        while (true) {
                                            try {
                                                J$.Fe(1363321, arguments.callee, this, arguments);
                                                arguments = J$.N(1363329, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1363337, 'obj', obj, true, false, false);
                                                return J$.Rt(1363313, J$.U(128418, 'typeof', J$.R(1363305, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1374345, J$e);
                                            } finally {
                                                if (J$.Fr(1374353))
                                                    continue jalangiLabel0;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            } else {
                                _typeof2 = J$.W(1363489, '_typeof2', J$.T(1363481, function _typeof2(obj) {
                                    jalangiLabel1:
                                        while (true) {
                                            try {
                                                J$.Fe(1363457, arguments.callee, this, arguments);
                                                arguments = J$.N(1363465, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1363473, 'obj', obj, true, false, false);
                                                return J$.Rt(1363449, J$.C(85704, J$.C(85696, J$.C(85688, J$.C(85680, J$.R(1363361, 'obj', obj, false, false)) ? J$.B(128434, '===', J$.U(128426, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1363369, 'Symbol', undefined, true, true) : Symbol = J$.R(1363369, 'Symbol', Symbol, true, true))), J$.T(1363377, 'function', 21, false)) : J$._()) ? J$.B(128442, '===', J$.G(1363393, J$.R(1363385, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1363401, 'Symbol', undefined, true, true) : Symbol = J$.R(1363401, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(128450, '!==', J$.R(1363409, 'obj', obj, false, false), J$.G(1363425, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1363417, 'Symbol', undefined, true, true) : Symbol = J$.R(1363417, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1363433, 'symbol', 21, false) : J$.U(128458, 'typeof', J$.R(1363441, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1374361, J$e);
                                            } finally {
                                                if (J$.Fr(1374369))
                                                    continue jalangiLabel1;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof2, false, true);
                            }
                            return J$.Rt(1363521, J$.F(1363513, J$.R(1363497, '_typeof2', _typeof2, false, true), false)(J$.R(1363505, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1374377, J$e);
                        } finally {
                            if (J$.Fr(1374385))
                                continue jalangiLabel2;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _typeof(obj) {
                jalangiLabel5:
                    while (true) {
                        try {
                            J$.Fe(1363865, arguments.callee, this, arguments);
                            arguments = J$.N(1363873, 'arguments', arguments, true, false, false);
                            obj = J$.N(1363881, 'obj', obj, true, false, false);
                            if (J$.C(85760, J$.C(85720, J$.B(128474, '===', J$.U(128466, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1363553, 'Symbol', undefined, true, true) : Symbol = J$.R(1363553, 'Symbol', Symbol, true, true))), J$.T(1363561, 'function', 21, false))) ? J$.B(128482, '===', J$.F(1363593, J$.R(1363569, '_typeof2', _typeof2, false, true), false)(J$.G(1363585, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1363577, 'Symbol', undefined, true, true) : Symbol = J$.R(1363577, 'Symbol', Symbol, true, true)), 'iterator')), J$.T(1363601, 'symbol', 21, false)) : J$._())) {
                                _typeof = J$.W(1363673, '_typeof', J$.T(1363665, function _typeof(obj) {
                                    jalangiLabel3:
                                        while (true) {
                                            try {
                                                J$.Fe(1363641, arguments.callee, this, arguments);
                                                arguments = J$.N(1363649, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1363657, 'obj', obj, true, false, false);
                                                return J$.Rt(1363633, J$.F(1363625, J$.R(1363609, '_typeof2', _typeof2, false, true), false)(J$.R(1363617, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1374393, J$e);
                                            } finally {
                                                if (J$.Fr(1374401))
                                                    continue jalangiLabel3;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            } else {
                                _typeof = J$.W(1363825, '_typeof', J$.T(1363817, function _typeof(obj) {
                                    jalangiLabel4:
                                        while (true) {
                                            try {
                                                J$.Fe(1363793, arguments.callee, this, arguments);
                                                arguments = J$.N(1363801, 'arguments', arguments, true, false, false);
                                                obj = J$.N(1363809, 'obj', obj, true, false, false);
                                                return J$.Rt(1363785, J$.C(85752, J$.C(85744, J$.C(85736, J$.C(85728, J$.R(1363681, 'obj', obj, false, false)) ? J$.B(128498, '===', J$.U(128490, 'typeof', J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1363689, 'Symbol', undefined, true, true) : Symbol = J$.R(1363689, 'Symbol', Symbol, true, true))), J$.T(1363697, 'function', 21, false)) : J$._()) ? J$.B(128506, '===', J$.G(1363713, J$.R(1363705, 'obj', obj, false, false), 'constructor'), J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1363721, 'Symbol', undefined, true, true) : Symbol = J$.R(1363721, 'Symbol', Symbol, true, true))) : J$._()) ? J$.B(128514, '!==', J$.R(1363729, 'obj', obj, false, false), J$.G(1363745, J$.I(typeof Symbol === 'undefined' ? Symbol = J$.R(1363737, 'Symbol', undefined, true, true) : Symbol = J$.R(1363737, 'Symbol', Symbol, true, true)), 'prototype')) : J$._()) ? J$.T(1363753, 'symbol', 21, false) : J$.F(1363777, J$.R(1363761, '_typeof2', _typeof2, false, true), false)(J$.R(1363769, 'obj', obj, false, false)));
                                            } catch (J$e) {
                                                J$.Ex(1374409, J$e);
                                            } finally {
                                                if (J$.Fr(1374417))
                                                    continue jalangiLabel4;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false), _typeof, false, true);
                            }
                            return J$.Rt(1363857, J$.F(1363849, J$.R(1363833, '_typeof', _typeof, false, true), false)(J$.R(1363841, 'obj', obj, false, false)));
                        } catch (J$e) {
                            J$.Ex(1374425, J$e);
                        } finally {
                            if (J$.Fr(1374433))
                                continue jalangiLabel5;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _classCallCheck(instance, Constructor) {
                jalangiLabel6:
                    while (true) {
                        try {
                            J$.Fe(1363929, arguments.callee, this, arguments);
                            arguments = J$.N(1363937, 'arguments', arguments, true, false, false);
                            instance = J$.N(1363945, 'instance', instance, true, false, false);
                            Constructor = J$.N(1363953, 'Constructor', Constructor, true, false, false);
                            if (J$.C(85768, J$.U(128530, '!', J$.B(128522, 'instanceof', J$.R(1363889, 'instance', instance, false, false), J$.R(1363897, 'Constructor', Constructor, false, false))))) {
                                throw J$.F(1363921, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1363905, 'TypeError', undefined, true, true) : TypeError = J$.R(1363905, 'TypeError', TypeError, true, true)), true)(J$.T(1363913, 'Cannot call a class as a function', 21, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1374441, J$e);
                        } finally {
                            if (J$.Fr(1374449))
                                continue jalangiLabel6;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _defineProperties(target, props) {
                jalangiLabel7:
                    while (true) {
                        try {
                            J$.Fe(1364201, arguments.callee, this, arguments);
                            arguments = J$.N(1364209, 'arguments', arguments, true, false, false);
                            target = J$.N(1364217, 'target', target, true, false, false);
                            props = J$.N(1364225, 'props', props, true, false, false);
                            J$.N(1364233, 'i', i, false, false, false);
                            J$.N(1364241, 'descriptor', descriptor, false, false, false);
                            for (var i = J$.W(1363969, 'i', J$.T(1363961, 0, 22, false), i, false, false); J$.C(85792, J$.B(128538, '<', J$.R(1363977, 'i', i, false, false), J$.G(1363993, J$.R(1363985, 'props', props, false, false), 'length'))); J$.B(128562, '-', i = J$.W(1364009, 'i', J$.B(128554, '+', J$.U(128546, '+', J$.R(1364001, 'i', i, false, false)), 1), i, false, false), 1)) {
                                var descriptor = J$.W(1364041, 'descriptor', J$.G(1364033, J$.R(1364017, 'props', props, false, false), J$.R(1364025, 'i', i, false, false)), descriptor, false, false);
                                J$.P(1364081, J$.R(1364049, 'descriptor', descriptor, false, false), 'enumerable', J$.C(85776, J$.G(1364065, J$.R(1364057, 'descriptor', descriptor, false, false), 'enumerable')) ? J$._() : J$.T(1364073, false, 23, false));
                                J$.P(1364105, J$.R(1364089, 'descriptor', descriptor, false, false), 'configurable', J$.T(1364097, true, 23, false));
                                if (J$.C(85784, J$.B(128570, 'in', J$.T(1364113, 'value', 21, false), J$.R(1364121, 'descriptor', descriptor, false, false))))
                                    J$.P(1364145, J$.R(1364129, 'descriptor', descriptor, false, false), 'writable', J$.T(1364137, true, 23, false));
                                J$.M(1364193, J$.I(typeof Object === 'undefined' ? Object = J$.R(1364153, 'Object', undefined, true, true) : Object = J$.R(1364153, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(1364161, 'target', target, false, false), J$.G(1364177, J$.R(1364169, 'descriptor', descriptor, false, false), 'key'), J$.R(1364185, 'descriptor', descriptor, false, false));
                            }
                        } catch (J$e) {
                            J$.Ex(1374457, J$e);
                        } finally {
                            if (J$.Fr(1374465))
                                continue jalangiLabel7;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _createClass(Constructor, protoProps, staticProps) {
                jalangiLabel8:
                    while (true) {
                        try {
                            J$.Fe(1364353, arguments.callee, this, arguments);
                            arguments = J$.N(1364361, 'arguments', arguments, true, false, false);
                            Constructor = J$.N(1364369, 'Constructor', Constructor, true, false, false);
                            protoProps = J$.N(1364377, 'protoProps', protoProps, true, false, false);
                            staticProps = J$.N(1364385, 'staticProps', staticProps, true, false, false);
                            if (J$.C(85800, J$.R(1364249, 'protoProps', protoProps, false, false)))
                                J$.F(1364289, J$.R(1364257, '_defineProperties', _defineProperties, false, true), false)(J$.G(1364273, J$.R(1364265, 'Constructor', Constructor, false, false), 'prototype'), J$.R(1364281, 'protoProps', protoProps, false, false));
                            if (J$.C(85808, J$.R(1364297, 'staticProps', staticProps, false, false)))
                                J$.F(1364329, J$.R(1364305, '_defineProperties', _defineProperties, false, true), false)(J$.R(1364313, 'Constructor', Constructor, false, false), J$.R(1364321, 'staticProps', staticProps, false, false));
                            return J$.Rt(1364345, J$.R(1364337, 'Constructor', Constructor, false, false));
                        } catch (J$e) {
                            J$.Ex(1374473, J$e);
                        } finally {
                            if (J$.Fr(1374481))
                                continue jalangiLabel8;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _possibleConstructorReturn(self, call) {
                jalangiLabel9:
                    while (true) {
                        try {
                            J$.Fe(1364497, arguments.callee, this, arguments);
                            arguments = J$.N(1364505, 'arguments', arguments, true, false, false);
                            self = J$.N(1364513, 'self', self, true, false, false);
                            call = J$.N(1364521, 'call', call, true, false, false);
                            if (J$.C(85832, J$.C(85824, J$.R(1364393, 'call', call, false, false)) ? J$.C(85816, J$.B(128578, '===', J$.F(1364417, J$.R(1364401, '_typeof', _typeof, false, true), false)(J$.R(1364409, 'call', call, false, false)), J$.T(1364425, 'object', 21, false))) ? J$._() : J$.B(128594, '===', J$.U(128586, 'typeof', J$.R(1364433, 'call', call, false, false)), J$.T(1364441, 'function', 21, false)) : J$._())) {
                                return J$.Rt(1364457, J$.R(1364449, 'call', call, false, false));
                            }
                            return J$.Rt(1364489, J$.F(1364481, J$.R(1364465, '_assertThisInitialized', _assertThisInitialized, false, true), false)(J$.R(1364473, 'self', self, false, false)));
                        } catch (J$e) {
                            J$.Ex(1374489, J$e);
                        } finally {
                            if (J$.Fr(1374497))
                                continue jalangiLabel9;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _assertThisInitialized(self) {
                jalangiLabel10:
                    while (true) {
                        try {
                            J$.Fe(1364585, arguments.callee, this, arguments);
                            arguments = J$.N(1364593, 'arguments', arguments, true, false, false);
                            self = J$.N(1364601, 'self', self, true, false, false);
                            if (J$.C(85840, J$.B(128602, '===', J$.R(1364529, 'self', self, false, false), void J$.T(1364537, 0, 22, false)))) {
                                throw J$.F(1364561, J$.I(typeof ReferenceError === 'undefined' ? ReferenceError = J$.R(1364545, 'ReferenceError', undefined, true, true) : ReferenceError = J$.R(1364545, 'ReferenceError', ReferenceError, true, true)), true)(J$.T(1364553, 'this hasn\'t been initialised - super() hasn\'t been called', 21, false));
                            }
                            return J$.Rt(1364577, J$.R(1364569, 'self', self, false, false));
                        } catch (J$e) {
                            J$.Ex(1374505, J$e);
                        } finally {
                            if (J$.Fr(1374513))
                                continue jalangiLabel10;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _getPrototypeOf(o) {
                jalangiLabel12:
                    while (true) {
                        try {
                            J$.Fe(1364761, arguments.callee, this, arguments);
                            arguments = J$.N(1364769, 'arguments', arguments, true, false, false);
                            o = J$.N(1364777, 'o', o, true, false, false);
                            _getPrototypeOf = J$.W(1364721, '_getPrototypeOf', J$.C(85856, J$.G(1364617, J$.I(typeof Object === 'undefined' ? Object = J$.R(1364609, 'Object', undefined, true, true) : Object = J$.R(1364609, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$.G(1364633, J$.I(typeof Object === 'undefined' ? Object = J$.R(1364625, 'Object', undefined, true, true) : Object = J$.R(1364625, 'Object', Object, true, true)), 'getPrototypeOf') : J$.T(1364713, function _getPrototypeOf(o) {
                                jalangiLabel11:
                                    while (true) {
                                        try {
                                            J$.Fe(1364689, arguments.callee, this, arguments);
                                            arguments = J$.N(1364697, 'arguments', arguments, true, false, false);
                                            o = J$.N(1364705, 'o', o, true, false, false);
                                            return J$.Rt(1364681, J$.C(85848, J$.G(1364649, J$.R(1364641, 'o', o, false, false), '__proto__')) ? J$._() : J$.M(1364673, J$.I(typeof Object === 'undefined' ? Object = J$.R(1364657, 'Object', undefined, true, true) : Object = J$.R(1364657, 'Object', Object, true, true)), 'getPrototypeOf', false)(J$.R(1364665, 'o', o, false, false)));
                                        } catch (J$e) {
                                            J$.Ex(1374521, J$e);
                                        } finally {
                                            if (J$.Fr(1374529))
                                                continue jalangiLabel11;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _getPrototypeOf, false, true);
                            return J$.Rt(1364753, J$.F(1364745, J$.R(1364729, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1364737, 'o', o, false, false)));
                        } catch (J$e) {
                            J$.Ex(1374537, J$e);
                        } finally {
                            if (J$.Fr(1374545))
                                continue jalangiLabel12;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _inherits(subClass, superClass) {
                jalangiLabel13:
                    while (true) {
                        try {
                            J$.Fe(1364977, arguments.callee, this, arguments);
                            arguments = J$.N(1364985, 'arguments', arguments, true, false, false);
                            subClass = J$.N(1364993, 'subClass', subClass, true, false, false);
                            superClass = J$.N(1365001, 'superClass', superClass, true, false, false);
                            if (J$.C(85872, J$.C(85864, J$.B(128618, '!==', J$.U(128610, 'typeof', J$.R(1364785, 'superClass', superClass, false, false)), J$.T(1364793, 'function', 21, false))) ? J$.B(128626, '!==', J$.R(1364801, 'superClass', superClass, false, false), J$.T(1364809, null, 25, false)) : J$._())) {
                                throw J$.F(1364833, J$.I(typeof TypeError === 'undefined' ? TypeError = J$.R(1364817, 'TypeError', undefined, true, true) : TypeError = J$.R(1364817, 'TypeError', TypeError, true, true)), true)(J$.T(1364825, 'Super expression must either be null or a function', 21, false));
                            }
                            J$.P(1364929, J$.R(1364841, 'subClass', subClass, false, false), 'prototype', J$.M(1364921, J$.I(typeof Object === 'undefined' ? Object = J$.R(1364849, 'Object', undefined, true, true) : Object = J$.R(1364849, 'Object', Object, true, true)), 'create', false)(J$.C(85880, J$.R(1364857, 'superClass', superClass, false, false)) ? J$.G(1364873, J$.R(1364865, 'superClass', superClass, false, false), 'prototype') : J$._(), J$.T(1364913, {
                                constructor: J$.T(1364905, {
                                    value: J$.R(1364881, 'subClass', subClass, false, false),
                                    writable: J$.T(1364889, true, 23, false),
                                    configurable: J$.T(1364897, true, 23, false)
                                }, 11, false)
                            }, 11, false)));
                            if (J$.C(85888, J$.R(1364937, 'superClass', superClass, false, false)))
                                J$.F(1364969, J$.R(1364945, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1364953, 'subClass', subClass, false, false), J$.R(1364961, 'superClass', superClass, false, false));
                        } catch (J$e) {
                            J$.Ex(1374553, J$e);
                        } finally {
                            if (J$.Fr(1374561))
                                continue jalangiLabel13;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function _setPrototypeOf(o, p) {
                jalangiLabel15:
                    while (true) {
                        try {
                            J$.Fe(1365153, arguments.callee, this, arguments);
                            arguments = J$.N(1365161, 'arguments', arguments, true, false, false);
                            o = J$.N(1365169, 'o', o, true, false, false);
                            p = J$.N(1365177, 'p', p, true, false, false);
                            _setPrototypeOf = J$.W(1365105, '_setPrototypeOf', J$.C(85896, J$.G(1365017, J$.I(typeof Object === 'undefined' ? Object = J$.R(1365009, 'Object', undefined, true, true) : Object = J$.R(1365009, 'Object', Object, true, true)), 'setPrototypeOf')) ? J$._() : J$.T(1365097, function _setPrototypeOf(o, p) {
                                jalangiLabel14:
                                    while (true) {
                                        try {
                                            J$.Fe(1365065, arguments.callee, this, arguments);
                                            arguments = J$.N(1365073, 'arguments', arguments, true, false, false);
                                            o = J$.N(1365081, 'o', o, true, false, false);
                                            p = J$.N(1365089, 'p', p, true, false, false);
                                            J$.P(1365041, J$.R(1365025, 'o', o, false, false), '__proto__', J$.R(1365033, 'p', p, false, false));
                                            return J$.Rt(1365057, J$.R(1365049, 'o', o, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1374569, J$e);
                                        } finally {
                                            if (J$.Fr(1374577))
                                                continue jalangiLabel14;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), _setPrototypeOf, false, true);
                            return J$.Rt(1365145, J$.F(1365137, J$.R(1365113, '_setPrototypeOf', _setPrototypeOf, false, true), false)(J$.R(1365121, 'o', o, false, false), J$.R(1365129, 'p', p, false, false)));
                        } catch (J$e) {
                            J$.Ex(1374585, J$e);
                        } finally {
                            if (J$.Fr(1374593))
                                continue jalangiLabel15;
                            else
                                return J$.Ra();
                        }
                    }
            }
            _typeof2 = J$.N(1374057, '_typeof2', J$.T(1374049, _typeof2, 12, false), true, false, false);
            _typeof = J$.N(1374073, '_typeof', J$.T(1374065, _typeof, 12, false), true, false, false);
            _classCallCheck = J$.N(1374089, '_classCallCheck', J$.T(1374081, _classCallCheck, 12, false), true, false, false);
            _defineProperties = J$.N(1374105, '_defineProperties', J$.T(1374097, _defineProperties, 12, false), true, false, false);
            _createClass = J$.N(1374121, '_createClass', J$.T(1374113, _createClass, 12, false), true, false, false);
            _possibleConstructorReturn = J$.N(1374137, '_possibleConstructorReturn', J$.T(1374129, _possibleConstructorReturn, 12, false), true, false, false);
            _assertThisInitialized = J$.N(1374153, '_assertThisInitialized', J$.T(1374145, _assertThisInitialized, 12, false), true, false, false);
            _getPrototypeOf = J$.N(1374169, '_getPrototypeOf', J$.T(1374161, _getPrototypeOf, 12, false), true, false, false);
            _inherits = J$.N(1374185, '_inherits', J$.T(1374177, _inherits, 12, false), true, false, false);
            _setPrototypeOf = J$.N(1374201, '_setPrototypeOf', J$.T(1374193, _setPrototypeOf, 12, false), true, false, false);
            J$.N(1374209, 'EventEmitter', EventEmitter, false, false, false);
            J$.N(1374217, 'MongoError', MongoError, false, false, false);
            J$.N(1374225, 'f', f, false, false, false);
            J$.N(1374233, 'os', os, false, false, false);
            J$.N(1374241, 'translateReadPreference', translateReadPreference, false, false, false);
            J$.N(1374249, 'ClientSession', ClientSession, false, false, false);
            J$.N(1374257, 'Store', Store, false, false, false);
            J$.N(1374265, 'primaryOptions', primaryOptions, false, false, false);
            J$.N(1374273, 'secondaryOptions', secondaryOptions, false, false, false);
            J$.N(1374281, 'ServerCapabilities', ServerCapabilities, false, false, false);
            J$.N(1374289, 'driverVersion', driverVersion, false, false, false);
            J$.N(1374297, 'nodejsversion', nodejsversion, false, false, false);
            J$.N(1374305, 'type', type, false, false, false);
            J$.N(1374313, 'name', name, false, false, false);
            J$.N(1374321, 'architecture', architecture, false, false, false);
            J$.N(1374329, 'release', release, false, false, false);
            J$.N(1374337, 'TopologyBase', TopologyBase, false, false, false);
            J$.T(1363257, 'use strict', 21, false);
            var EventEmitter = J$.W(1365369, 'EventEmitter', J$.F(1365201, J$.I(typeof require === 'undefined' ? require = J$.R(1365185, 'require', undefined, true, true) : require = J$.R(1365185, 'require', require, true, true)), false)(J$.T(1365193, 'events', 21, false)), EventEmitter, false, true), MongoError = J$.W(1365377, 'MongoError', J$.G(1365233, J$.F(1365225, J$.I(typeof require === 'undefined' ? require = J$.R(1365209, 'require', undefined, true, true) : require = J$.R(1365209, 'require', require, true, true)), false)(J$.T(1365217, '../core', 21, false)), 'MongoError'), MongoError, false, true), f = J$.W(1365385, 'f', J$.G(1365265, J$.F(1365257, J$.I(typeof require === 'undefined' ? require = J$.R(1365241, 'require', undefined, true, true) : require = J$.R(1365241, 'require', require, true, true)), false)(J$.T(1365249, 'util', 21, false)), 'format'), f, false, true), os = J$.W(1365393, 'os', J$.F(1365289, J$.I(typeof require === 'undefined' ? require = J$.R(1365273, 'require', undefined, true, true) : require = J$.R(1365273, 'require', require, true, true)), false)(J$.T(1365281, 'os', 21, false)), os, false, true), translateReadPreference = J$.W(1365401, 'translateReadPreference', J$.G(1365321, J$.F(1365313, J$.I(typeof require === 'undefined' ? require = J$.R(1365297, 'require', undefined, true, true) : require = J$.R(1365297, 'require', require, true, true)), false)(J$.T(1365305, '../utils', 21, false)), 'translateReadPreference'), translateReadPreference, false, true), ClientSession = J$.W(1365409, 'ClientSession', J$.G(1365361, J$.G(1365353, J$.F(1365345, J$.I(typeof require === 'undefined' ? require = J$.R(1365329, 'require', undefined, true, true) : require = J$.R(1365329, 'require', require, true, true)), false)(J$.T(1365337, '../core', 21, false)), 'Sessions'), 'ClientSession'), ClientSession, false, true);
            var Store = J$.W(1365705, 'Store', J$.T(1365697, function Store(topology, storeOptions) {
                jalangiLabel17:
                    while (true) {
                        try {
                            J$.Fe(1365649, arguments.callee, this, arguments);
                            arguments = J$.N(1365657, 'arguments', arguments, true, false, false);
                            topology = J$.N(1365665, 'topology', topology, true, false, false);
                            storeOptions = J$.N(1365673, 'storeOptions', storeOptions, true, false, false);
                            J$.N(1365681, 'self', self, false, false, false);
                            J$.N(1365689, 'storedOps', storedOps, false, false, false);
                            var self = J$.W(1365425, 'self', J$.R(1365417, 'this', this, false, false), self, false, false);
                            var storedOps = J$.W(1365441, 'storedOps', J$.T(1365433, [], 10, false), storedOps, false, false);
                            storeOptions = J$.W(1365481, 'storeOptions', J$.C(85904, J$.R(1365449, 'storeOptions', storeOptions, false, false)) ? J$._() : J$.T(1365473, {
                                force: J$.T(1365457, false, 23, false),
                                bufferMaxEntries: J$.U(128634, '-', J$.T(1365465, 1, 22, false))
                            }, 11, false), storeOptions, false, false);
                            J$.P(1365529, J$.R(1365489, 'this', this, false, false), 's', J$.T(1365521, {
                                storedOps: J$.R(1365497, 'storedOps', storedOps, false, false),
                                storeOptions: J$.R(1365505, 'storeOptions', storeOptions, false, false),
                                topology: J$.R(1365513, 'topology', topology, false, false)
                            }, 11, false));
                            J$.M(1365641, J$.I(typeof Object === 'undefined' ? Object = J$.R(1365537, 'Object', undefined, true, true) : Object = J$.R(1365537, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(1365545, 'this', this, false, false), J$.T(1365553, 'length', 21, false), J$.T(1365633, {
                                enumerable: J$.T(1365561, true, 23, false),
                                get: J$.T(1365625, function get() {
                                    jalangiLabel16:
                                        while (true) {
                                            try {
                                                J$.Fe(1365609, arguments.callee, this, arguments);
                                                arguments = J$.N(1365617, 'arguments', arguments, true, false, false);
                                                return J$.Rt(1365601, J$.G(1365593, J$.G(1365585, J$.G(1365577, J$.R(1365569, 'self', self, false, false), 's'), 'storedOps'), 'length'));
                                            } catch (J$e) {
                                                J$.Ex(1374601, J$e);
                                            } finally {
                                                if (J$.Fr(1374609))
                                                    continue jalangiLabel16;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(1374617, J$e);
                        } finally {
                            if (J$.Fr(1374625))
                                continue jalangiLabel17;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), Store, false, true);
            J$.P(1366425, J$.G(1365721, J$.R(1365713, 'Store', Store, false, true), 'prototype'), 'add', J$.T(1366417, function (opType, ns, ops, options, callback) {
                jalangiLabel18:
                    while (true) {
                        try {
                            J$.Fe(1366353, arguments.callee, this, arguments);
                            arguments = J$.N(1366361, 'arguments', arguments, true, false, false);
                            opType = J$.N(1366369, 'opType', opType, true, false, false);
                            ns = J$.N(1366377, 'ns', ns, true, false, false);
                            ops = J$.N(1366385, 'ops', ops, true, false, false);
                            options = J$.N(1366393, 'options', options, true, false, false);
                            callback = J$.N(1366401, 'callback', callback, true, false, false);
                            J$.N(1366409, 'op', op, false, false, false);
                            if (J$.C(85912, J$.G(1365753, J$.G(1365745, J$.G(1365737, J$.R(1365729, 'this', this, false, false), 's'), 'storeOptions'), 'force'))) {
                                return J$.Rt(1365817, J$.F(1365809, J$.R(1365761, 'callback', callback, false, false), false)(J$.M(1365801, J$.R(1365769, 'MongoError', MongoError, false, true), 'create', false)(J$.T(1365793, {
                                    message: J$.T(1365777, 'db closed by application', 21, false),
                                    driver: J$.T(1365785, true, 23, false)
                                }, 11, false))));
                            }
                            if (J$.C(85920, J$.B(128642, '===', J$.G(1365849, J$.G(1365841, J$.G(1365833, J$.R(1365825, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries'), J$.T(1365857, 0, 22, false)))) {
                                return J$.Rt(1365969, J$.F(1365961, J$.R(1365865, 'callback', callback, false, false), false)(J$.M(1365953, J$.R(1365873, 'MongoError', MongoError, false, true), 'create', false)(J$.T(1365945, {
                                    message: J$.F(1365929, J$.R(1365881, 'f', f, false, true), false)(J$.T(1365889, 'no connection available for operation and number of stored operation > %s', 21, false), J$.G(1365921, J$.G(1365913, J$.G(1365905, J$.R(1365897, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')),
                                    driver: J$.T(1365937, true, 23, false)
                                }, 11, false))));
                            }
                            if (J$.C(85944, J$.C(85928, J$.B(128650, '>', J$.G(1366001, J$.G(1365993, J$.G(1365985, J$.R(1365977, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries'), J$.T(1366009, 0, 22, false))) ? J$.B(128658, '>', J$.G(1366041, J$.G(1366033, J$.G(1366025, J$.R(1366017, 'this', this, false, false), 's'), 'storedOps'), 'length'), J$.G(1366073, J$.G(1366065, J$.G(1366057, J$.R(1366049, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')) : J$._())) {
                                while (J$.C(85936, J$.B(128666, '>', J$.G(1366105, J$.G(1366097, J$.G(1366089, J$.R(1366081, 'this', this, false, false), 's'), 'storedOps'), 'length'), J$.T(1366113, 0, 22, false)))) {
                                    var op = J$.W(1366153, 'op', J$.M(1366145, J$.G(1366137, J$.G(1366129, J$.R(1366121, 'this', this, false, false), 's'), 'storedOps'), 'shift', false)(), op, false, false);
                                    J$.M(1366257, J$.R(1366161, 'op', op, false, false), 'c', false)(J$.M(1366249, J$.R(1366169, 'MongoError', MongoError, false, true), 'create', false)(J$.T(1366241, {
                                        message: J$.F(1366225, J$.R(1366177, 'f', f, false, true), false)(J$.T(1366185, 'no connection available for operation and number of stored operation > %s', 21, false), J$.G(1366217, J$.G(1366209, J$.G(1366201, J$.R(1366193, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')),
                                        driver: J$.T(1366233, true, 23, false)
                                    }, 11, false)));
                                }
                                return J$.Rt(1366265, undefined);
                            }
                            J$.M(1366345, J$.G(1366289, J$.G(1366281, J$.R(1366273, 'this', this, false, false), 's'), 'storedOps'), 'push', false)(J$.T(1366337, {
                                t: J$.R(1366297, 'opType', opType, false, false),
                                n: J$.R(1366305, 'ns', ns, false, false),
                                o: J$.R(1366313, 'ops', ops, false, false),
                                op: J$.R(1366321, 'options', options, false, false),
                                c: J$.R(1366329, 'callback', callback, false, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(1374633, J$e);
                        } finally {
                            if (J$.Fr(1374641))
                                continue jalangiLabel18;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(1367145, J$.G(1366441, J$.R(1366433, 'Store', Store, false, true), 'prototype'), 'addObjectAndMethod', J$.T(1367137, function (opType, object, method, params, callback) {
                jalangiLabel19:
                    while (true) {
                        try {
                            J$.Fe(1367073, arguments.callee, this, arguments);
                            arguments = J$.N(1367081, 'arguments', arguments, true, false, false);
                            opType = J$.N(1367089, 'opType', opType, true, false, false);
                            object = J$.N(1367097, 'object', object, true, false, false);
                            method = J$.N(1367105, 'method', method, true, false, false);
                            params = J$.N(1367113, 'params', params, true, false, false);
                            callback = J$.N(1367121, 'callback', callback, true, false, false);
                            J$.N(1367129, 'op', op, false, false, false);
                            if (J$.C(85952, J$.G(1366473, J$.G(1366465, J$.G(1366457, J$.R(1366449, 'this', this, false, false), 's'), 'storeOptions'), 'force'))) {
                                return J$.Rt(1366537, J$.F(1366529, J$.R(1366481, 'callback', callback, false, false), false)(J$.M(1366521, J$.R(1366489, 'MongoError', MongoError, false, true), 'create', false)(J$.T(1366513, {
                                    message: J$.T(1366497, 'db closed by application', 21, false),
                                    driver: J$.T(1366505, true, 23, false)
                                }, 11, false))));
                            }
                            if (J$.C(85960, J$.B(128674, '===', J$.G(1366569, J$.G(1366561, J$.G(1366553, J$.R(1366545, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries'), J$.T(1366577, 0, 22, false)))) {
                                return J$.Rt(1366689, J$.F(1366681, J$.R(1366585, 'callback', callback, false, false), false)(J$.M(1366673, J$.R(1366593, 'MongoError', MongoError, false, true), 'create', false)(J$.T(1366665, {
                                    message: J$.F(1366649, J$.R(1366601, 'f', f, false, true), false)(J$.T(1366609, 'no connection available for operation and number of stored operation > %s', 21, false), J$.G(1366641, J$.G(1366633, J$.G(1366625, J$.R(1366617, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')),
                                    driver: J$.T(1366657, true, 23, false)
                                }, 11, false))));
                            }
                            if (J$.C(85984, J$.C(85968, J$.B(128682, '>', J$.G(1366721, J$.G(1366713, J$.G(1366705, J$.R(1366697, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries'), J$.T(1366729, 0, 22, false))) ? J$.B(128690, '>', J$.G(1366761, J$.G(1366753, J$.G(1366745, J$.R(1366737, 'this', this, false, false), 's'), 'storedOps'), 'length'), J$.G(1366793, J$.G(1366785, J$.G(1366777, J$.R(1366769, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')) : J$._())) {
                                while (J$.C(85976, J$.B(128698, '>', J$.G(1366825, J$.G(1366817, J$.G(1366809, J$.R(1366801, 'this', this, false, false), 's'), 'storedOps'), 'length'), J$.T(1366833, 0, 22, false)))) {
                                    var op = J$.W(1366873, 'op', J$.M(1366865, J$.G(1366857, J$.G(1366849, J$.R(1366841, 'this', this, false, false), 's'), 'storedOps'), 'shift', false)(), op, false, false);
                                    J$.M(1366977, J$.R(1366881, 'op', op, false, false), 'c', false)(J$.M(1366969, J$.R(1366889, 'MongoError', MongoError, false, true), 'create', false)(J$.T(1366961, {
                                        message: J$.F(1366945, J$.R(1366897, 'f', f, false, true), false)(J$.T(1366905, 'no connection available for operation and number of stored operation > %s', 21, false), J$.G(1366937, J$.G(1366929, J$.G(1366921, J$.R(1366913, 'this', this, false, false), 's'), 'storeOptions'), 'bufferMaxEntries')),
                                        driver: J$.T(1366953, true, 23, false)
                                    }, 11, false)));
                                }
                                return J$.Rt(1366985, undefined);
                            }
                            J$.M(1367065, J$.G(1367009, J$.G(1367001, J$.R(1366993, 'this', this, false, false), 's'), 'storedOps'), 'push', false)(J$.T(1367057, {
                                t: J$.R(1367017, 'opType', opType, false, false),
                                m: J$.R(1367025, 'method', method, false, false),
                                o: J$.R(1367033, 'object', object, false, false),
                                p: J$.R(1367041, 'params', params, false, false),
                                c: J$.R(1367049, 'callback', callback, false, false)
                            }, 11, false));
                        } catch (J$e) {
                            J$.Ex(1374649, J$e);
                        } finally {
                            if (J$.Fr(1374657))
                                continue jalangiLabel19;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(1367345, J$.G(1367161, J$.R(1367153, 'Store', Store, false, true), 'prototype'), 'flush', J$.T(1367337, function (err) {
                jalangiLabel20:
                    while (true) {
                        try {
                            J$.Fe(1367313, arguments.callee, this, arguments);
                            arguments = J$.N(1367321, 'arguments', arguments, true, false, false);
                            err = J$.N(1367329, 'err', err, true, false, false);
                            while (J$.C(86000, J$.B(128706, '>', J$.G(1367193, J$.G(1367185, J$.G(1367177, J$.R(1367169, 'this', this, false, false), 's'), 'storedOps'), 'length'), J$.T(1367201, 0, 22, false)))) {
                                J$.M(1367305, J$.M(1367233, J$.G(1367225, J$.G(1367217, J$.R(1367209, 'this', this, false, false), 's'), 'storedOps'), 'shift', false)(), 'c', false)(J$.C(85992, J$.R(1367241, 'err', err, false, false)) ? J$._() : J$.M(1367297, J$.R(1367249, 'MongoError', MongoError, false, true), 'create', false)(J$.T(1367289, {
                                    message: J$.F(1367273, J$.R(1367257, 'f', f, false, true), false)(J$.T(1367265, 'no connection available for operation', 21, false)),
                                    driver: J$.T(1367281, true, 23, false)
                                }, 11, false)));
                            }
                        } catch (J$e) {
                            J$.Ex(1374665, J$e);
                        } finally {
                            if (J$.Fr(1374673))
                                continue jalangiLabel20;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var primaryOptions = J$.W(1367393, 'primaryOptions', J$.T(1367385, [
                J$.T(1367353, 'primary', 21, false),
                J$.T(1367361, 'primaryPreferred', 21, false),
                J$.T(1367369, 'nearest', 21, false),
                J$.T(1367377, 'secondaryPreferred', 21, false)
            ], 10, false), primaryOptions, false, true);
            var secondaryOptions = J$.W(1367425, 'secondaryOptions', J$.T(1367417, [
                J$.T(1367401, 'secondary', 21, false),
                J$.T(1367409, 'secondaryPreferred', 21, false)
            ], 10, false), secondaryOptions, false, true);
            J$.P(1368993, J$.G(1367441, J$.R(1367433, 'Store', Store, false, true), 'prototype'), 'execute', J$.T(1368985, function (options) {
                jalangiLabel21:
                    while (true) {
                        try {
                            J$.Fe(1368929, arguments.callee, this, arguments);
                            arguments = J$.N(1368937, 'arguments', arguments, true, false, false);
                            options = J$.N(1368945, 'options', options, true, false, false);
                            J$.N(1368953, 'ops', ops, false, false, false);
                            J$.N(1368961, 'executePrimary', executePrimary, false, false, false);
                            J$.N(1368969, 'executeSecondary', executeSecondary, false, false, false);
                            J$.N(1368977, 'op', op, false, false, false);
                            options = J$.W(1367465, 'options', J$.C(86008, J$.R(1367449, 'options', options, false, false)) ? J$._() : J$.T(1367457, {}, 11, false), options, false, false);
                            var ops = J$.W(1367497, 'ops', J$.G(1367489, J$.G(1367481, J$.R(1367473, 'this', this, false, false), 's'), 'storedOps'), ops, false, false);
                            J$.P(1367529, J$.G(1367513, J$.R(1367505, 'this', this, false, false), 's'), 'storedOps', J$.T(1367521, [], 10, false));
                            var executePrimary = J$.W(1367585, 'executePrimary', J$.C(86016, J$.B(128722, '===', J$.U(128714, 'typeof', J$.G(1367545, J$.R(1367537, 'options', options, false, false), 'executePrimary')), J$.T(1367553, 'boolean', 21, false))) ? J$.G(1367569, J$.R(1367561, 'options', options, false, false), 'executePrimary') : J$.T(1367577, true, 23, false), executePrimary, false, false);
                            var executeSecondary = J$.W(1367641, 'executeSecondary', J$.C(86024, J$.B(128738, '===', J$.U(128730, 'typeof', J$.G(1367601, J$.R(1367593, 'options', options, false, false), 'executeSecondary')), J$.T(1367609, 'boolean', 21, false))) ? J$.G(1367625, J$.R(1367617, 'options', options, false, false), 'executeSecondary') : J$.T(1367633, true, 23, false), executeSecondary, false, false);
                            while (J$.C(86224, J$.B(128746, '>', J$.G(1367657, J$.R(1367649, 'ops', ops, false, false), 'length'), J$.T(1367665, 0, 22, false)))) {
                                var op = J$.W(1367689, 'op', J$.M(1367681, J$.R(1367673, 'ops', ops, false, false), 'shift', false)(), op, false, false);
                                if (J$.C(86216, J$.B(128754, '===', J$.G(1367705, J$.R(1367697, 'op', op, false, false), 't'), J$.T(1367713, 'cursor', 21, false)))) {
                                    if (J$.C(86112, J$.C(86032, J$.R(1367721, 'executePrimary', executePrimary, false, false)) ? J$.R(1367729, 'executeSecondary', executeSecondary, false, false) : J$._())) {
                                        J$.M(1367809, J$.G(1367769, J$.G(1367745, J$.R(1367737, 'op', op, false, false), 'o'), J$.G(1367761, J$.R(1367753, 'op', op, false, false), 'm')), 'apply', false)(J$.G(1367785, J$.R(1367777, 'op', op, false, false), 'o'), J$.G(1367801, J$.R(1367793, 'op', op, false, false), 'p'));
                                    } else if (J$.C(86104, J$.C(86056, J$.C(86048, J$.C(86040, J$.R(1367817, 'executePrimary', executePrimary, false, false)) ? J$.G(1367841, J$.G(1367833, J$.R(1367825, 'op', op, false, false), 'o'), 'options') : J$._()) ? J$.G(1367873, J$.G(1367865, J$.G(1367857, J$.R(1367849, 'op', op, false, false), 'o'), 'options'), 'readPreference') : J$._()) ? J$.B(128770, '!==', J$.M(1367929, J$.R(1367881, 'primaryOptions', primaryOptions, false, true), 'indexOf', false)(J$.G(1367921, J$.G(1367913, J$.G(1367905, J$.G(1367897, J$.R(1367889, 'op', op, false, false), 'o'), 'options'), 'readPreference'), 'mode')), J$.U(128762, '-', J$.T(1367937, 1, 22, false))) : J$._())) {
                                        J$.M(1368017, J$.G(1367977, J$.G(1367953, J$.R(1367945, 'op', op, false, false), 'o'), J$.G(1367969, J$.R(1367961, 'op', op, false, false), 'm')), 'apply', false)(J$.G(1367993, J$.R(1367985, 'op', op, false, false), 'o'), J$.G(1368009, J$.R(1368001, 'op', op, false, false), 'p'));
                                    } else if (J$.C(86096, J$.C(86088, J$.C(86080, J$.C(86072, J$.C(86064, J$.U(128778, '!', J$.R(1368025, 'executePrimary', executePrimary, false, false))) ? J$.R(1368033, 'executeSecondary', executeSecondary, false, false) : J$._()) ? J$.G(1368057, J$.G(1368049, J$.R(1368041, 'op', op, false, false), 'o'), 'options') : J$._()) ? J$.G(1368089, J$.G(1368081, J$.G(1368073, J$.R(1368065, 'op', op, false, false), 'o'), 'options'), 'readPreference') : J$._()) ? J$.B(128794, '!==', J$.M(1368145, J$.R(1368097, 'secondaryOptions', secondaryOptions, false, true), 'indexOf', false)(J$.G(1368137, J$.G(1368129, J$.G(1368121, J$.G(1368113, J$.R(1368105, 'op', op, false, false), 'o'), 'options'), 'readPreference'), 'mode')), J$.U(128786, '-', J$.T(1368153, 1, 22, false))) : J$._())) {
                                        J$.M(1368233, J$.G(1368193, J$.G(1368169, J$.R(1368161, 'op', op, false, false), 'o'), J$.G(1368185, J$.R(1368177, 'op', op, false, false), 'm')), 'apply', false)(J$.G(1368209, J$.R(1368201, 'op', op, false, false), 'o'), J$.G(1368225, J$.R(1368217, 'op', op, false, false), 'p'));
                                    }
                                } else if (J$.C(86208, J$.B(128802, '===', J$.G(1368249, J$.R(1368241, 'op', op, false, false), 't'), J$.T(1368257, 'auth', 21, false)))) {
                                    J$.M(1368353, J$.G(1368305, J$.G(1368281, J$.G(1368273, J$.R(1368265, 'this', this, false, false), 's'), 'topology'), J$.G(1368297, J$.R(1368289, 'op', op, false, false), 't')), 'apply', false)(J$.G(1368329, J$.G(1368321, J$.R(1368313, 'this', this, false, false), 's'), 'topology'), J$.G(1368345, J$.R(1368337, 'op', op, false, false), 'o'));
                                } else {
                                    if (J$.C(86200, J$.C(86120, J$.R(1368361, 'executePrimary', executePrimary, false, false)) ? J$.R(1368369, 'executeSecondary', executeSecondary, false, false) : J$._())) {
                                        J$.M(1368481, J$.G(1368393, J$.G(1368385, J$.R(1368377, 'this', this, false, false), 's'), 'topology'), J$.G(1368409, J$.R(1368401, 'op', op, false, false), 't'), false)(J$.G(1368425, J$.R(1368417, 'op', op, false, false), 'n'), J$.G(1368441, J$.R(1368433, 'op', op, false, false), 'o'), J$.G(1368457, J$.R(1368449, 'op', op, false, false), 'op'), J$.G(1368473, J$.R(1368465, 'op', op, false, false), 'c'));
                                    } else if (J$.C(86192, J$.C(86144, J$.C(86136, J$.C(86128, J$.R(1368489, 'executePrimary', executePrimary, false, false)) ? J$.G(1368505, J$.R(1368497, 'op', op, false, false), 'op') : J$._()) ? J$.G(1368529, J$.G(1368521, J$.R(1368513, 'op', op, false, false), 'op'), 'readPreference') : J$._()) ? J$.B(128818, '!==', J$.M(1368577, J$.R(1368537, 'primaryOptions', primaryOptions, false, true), 'indexOf', false)(J$.G(1368569, J$.G(1368561, J$.G(1368553, J$.R(1368545, 'op', op, false, false), 'op'), 'readPreference'), 'mode')), J$.U(128810, '-', J$.T(1368585, 1, 22, false))) : J$._())) {
                                        J$.M(1368697, J$.G(1368609, J$.G(1368601, J$.R(1368593, 'this', this, false, false), 's'), 'topology'), J$.G(1368625, J$.R(1368617, 'op', op, false, false), 't'), false)(J$.G(1368641, J$.R(1368633, 'op', op, false, false), 'n'), J$.G(1368657, J$.R(1368649, 'op', op, false, false), 'o'), J$.G(1368673, J$.R(1368665, 'op', op, false, false), 'op'), J$.G(1368689, J$.R(1368681, 'op', op, false, false), 'c'));
                                    } else if (J$.C(86184, J$.C(86176, J$.C(86168, J$.C(86160, J$.C(86152, J$.U(128826, '!', J$.R(1368705, 'executePrimary', executePrimary, false, false))) ? J$.R(1368713, 'executeSecondary', executeSecondary, false, false) : J$._()) ? J$.G(1368729, J$.R(1368721, 'op', op, false, false), 'op') : J$._()) ? J$.G(1368753, J$.G(1368745, J$.R(1368737, 'op', op, false, false), 'op'), 'readPreference') : J$._()) ? J$.B(128842, '!==', J$.M(1368801, J$.R(1368761, 'secondaryOptions', secondaryOptions, false, true), 'indexOf', false)(J$.G(1368793, J$.G(1368785, J$.G(1368777, J$.R(1368769, 'op', op, false, false), 'op'), 'readPreference'), 'mode')), J$.U(128834, '-', J$.T(1368809, 1, 22, false))) : J$._())) {
                                        J$.M(1368921, J$.G(1368833, J$.G(1368825, J$.R(1368817, 'this', this, false, false), 's'), 'topology'), J$.G(1368849, J$.R(1368841, 'op', op, false, false), 't'), false)(J$.G(1368865, J$.R(1368857, 'op', op, false, false), 'n'), J$.G(1368881, J$.R(1368873, 'op', op, false, false), 'o'), J$.G(1368897, J$.R(1368889, 'op', op, false, false), 'op'), J$.G(1368913, J$.R(1368905, 'op', op, false, false), 'c'));
                                    }
                                }
                            }
                        } catch (J$e) {
                            J$.Ex(1374681, J$e);
                        } finally {
                            if (J$.Fr(1374689))
                                continue jalangiLabel21;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            J$.P(1369073, J$.G(1369009, J$.R(1369001, 'Store', Store, false, true), 'prototype'), 'all', J$.T(1369065, function () {
                jalangiLabel22:
                    while (true) {
                        try {
                            J$.Fe(1369049, arguments.callee, this, arguments);
                            arguments = J$.N(1369057, 'arguments', arguments, true, false, false);
                            return J$.Rt(1369041, J$.G(1369033, J$.G(1369025, J$.R(1369017, 'this', this, false, false), 's'), 'storedOps'));
                        } catch (J$e) {
                            J$.Ex(1374697, J$e);
                        } finally {
                            if (J$.Fr(1374705))
                                continue jalangiLabel22;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false));
            var ServerCapabilities = J$.W(1370297, 'ServerCapabilities', J$.T(1370289, function ServerCapabilities(ismaster) {
                jalangiLabel25:
                    while (true) {
                        try {
                            J$.Fe(1370185, arguments.callee, this, arguments);
                            arguments = J$.N(1370193, 'arguments', arguments, true, false, false);
                            ismaster = J$.N(1370201, 'ismaster', ismaster, true, false, false);
                            J$.N(1370209, 'setup_get_property', setup_get_property, false, false, false);
                            J$.N(1370217, 'aggregationCursor', aggregationCursor, false, false, false);
                            J$.N(1370225, 'writeCommands', writeCommands, false, false, false);
                            J$.N(1370233, 'textSearch', textSearch, false, false, false);
                            J$.N(1370241, 'authCommands', authCommands, false, false, false);
                            J$.N(1370249, 'listCollections', listCollections, false, false, false);
                            J$.N(1370257, 'listIndexes', listIndexes, false, false, false);
                            J$.N(1370265, 'maxNumberOfDocsInBatch', maxNumberOfDocsInBatch, false, false, false);
                            J$.N(1370273, 'commandsTakeWriteConcern', commandsTakeWriteConcern, false, false, false);
                            J$.N(1370281, 'commandsTakeCollation', commandsTakeCollation, false, false, false);
                            var setup_get_property = J$.W(1369217, 'setup_get_property', J$.T(1369209, function setup_get_property(object, name, value) {
                                jalangiLabel24:
                                    while (true) {
                                        try {
                                            J$.Fe(1369169, arguments.callee, this, arguments);
                                            arguments = J$.N(1369177, 'arguments', arguments, true, false, false);
                                            object = J$.N(1369185, 'object', object, true, false, false);
                                            name = J$.N(1369193, 'name', name, true, false, false);
                                            value = J$.N(1369201, 'value', value, true, false, false);
                                            J$.M(1369161, J$.I(typeof Object === 'undefined' ? Object = J$.R(1369081, 'Object', undefined, true, true) : Object = J$.R(1369081, 'Object', Object, true, true)), 'defineProperty', false)(J$.R(1369089, 'object', object, false, false), J$.R(1369097, 'name', name, false, false), J$.T(1369153, {
                                                enumerable: J$.T(1369105, true, 23, false),
                                                get: J$.T(1369145, function get() {
                                                    jalangiLabel23:
                                                        while (true) {
                                                            try {
                                                                J$.Fe(1369129, arguments.callee, this, arguments);
                                                                arguments = J$.N(1369137, 'arguments', arguments, true, false, false);
                                                                return J$.Rt(1369121, J$.R(1369113, 'value', value, false, false));
                                                            } catch (J$e) {
                                                                J$.Ex(1374713, J$e);
                                                            } finally {
                                                                if (J$.Fr(1374721))
                                                                    continue jalangiLabel23;
                                                                else
                                                                    return J$.Ra();
                                                            }
                                                        }
                                                }, 12, false)
                                            }, 11, false));
                                        } catch (J$e) {
                                            J$.Ex(1374729, J$e);
                                        } finally {
                                            if (J$.Fr(1374737))
                                                continue jalangiLabel24;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false), setup_get_property, false, false);
                            var aggregationCursor = J$.W(1369233, 'aggregationCursor', J$.T(1369225, false, 23, false), aggregationCursor, false, false);
                            var writeCommands = J$.W(1369249, 'writeCommands', J$.T(1369241, false, 23, false), writeCommands, false, false);
                            var textSearch = J$.W(1369265, 'textSearch', J$.T(1369257, false, 23, false), textSearch, false, false);
                            var authCommands = J$.W(1369281, 'authCommands', J$.T(1369273, false, 23, false), authCommands, false, false);
                            var listCollections = J$.W(1369297, 'listCollections', J$.T(1369289, false, 23, false), listCollections, false, false);
                            var listIndexes = J$.W(1369313, 'listIndexes', J$.T(1369305, false, 23, false), listIndexes, false, false);
                            var maxNumberOfDocsInBatch = J$.W(1369345, 'maxNumberOfDocsInBatch', J$.C(86232, J$.G(1369329, J$.R(1369321, 'ismaster', ismaster, false, false), 'maxWriteBatchSize')) ? J$._() : J$.T(1369337, 1000, 22, false), maxNumberOfDocsInBatch, false, false);
                            var commandsTakeWriteConcern = J$.W(1369361, 'commandsTakeWriteConcern', J$.T(1369353, false, 23, false), commandsTakeWriteConcern, false, false);
                            var commandsTakeCollation = J$.W(1369377, 'commandsTakeCollation', J$.T(1369369, false, 23, false), commandsTakeCollation, false, false);
                            if (J$.C(86240, J$.B(128850, '>=', J$.G(1369393, J$.R(1369385, 'ismaster', ismaster, false, false), 'minWireVersion'), J$.T(1369401, 0, 22, false)))) {
                                textSearch = J$.W(1369417, 'textSearch', J$.T(1369409, true, 23, false), textSearch, false, false);
                            }
                            if (J$.C(86248, J$.B(128858, '>=', J$.G(1369433, J$.R(1369425, 'ismaster', ismaster, false, false), 'maxWireVersion'), J$.T(1369441, 1, 22, false)))) {
                                aggregationCursor = J$.W(1369457, 'aggregationCursor', J$.T(1369449, true, 23, false), aggregationCursor, false, false);
                                authCommands = J$.W(1369473, 'authCommands', J$.T(1369465, true, 23, false), authCommands, false, false);
                            }
                            if (J$.C(86256, J$.B(128866, '>=', J$.G(1369489, J$.R(1369481, 'ismaster', ismaster, false, false), 'maxWireVersion'), J$.T(1369497, 2, 22, false)))) {
                                writeCommands = J$.W(1369513, 'writeCommands', J$.T(1369505, true, 23, false), writeCommands, false, false);
                            }
                            if (J$.C(86264, J$.B(128874, '>=', J$.G(1369529, J$.R(1369521, 'ismaster', ismaster, false, false), 'maxWireVersion'), J$.T(1369537, 3, 22, false)))) {
                                listCollections = J$.W(1369553, 'listCollections', J$.T(1369545, true, 23, false), listCollections, false, false);
                                listIndexes = J$.W(1369569, 'listIndexes', J$.T(1369561, true, 23, false), listIndexes, false, false);
                            }
                            if (J$.C(86272, J$.B(128882, '>=', J$.G(1369585, J$.R(1369577, 'ismaster', ismaster, false, false), 'maxWireVersion'), J$.T(1369593, 5, 22, false)))) {
                                commandsTakeWriteConcern = J$.W(1369609, 'commandsTakeWriteConcern', J$.T(1369601, true, 23, false), commandsTakeWriteConcern, false, false);
                                commandsTakeCollation = J$.W(1369625, 'commandsTakeCollation', J$.T(1369617, true, 23, false), commandsTakeCollation, false, false);
                            }
                            if (J$.C(86280, J$.B(128890, '==', J$.G(1369641, J$.R(1369633, 'ismaster', ismaster, false, false), 'minWireVersion'), J$.T(1369649, null, 25, false)))) {
                                J$.P(1369673, J$.R(1369657, 'ismaster', ismaster, false, false), 'minWireVersion', J$.T(1369665, 0, 22, false));
                            }
                            if (J$.C(86288, J$.B(128898, '==', J$.G(1369689, J$.R(1369681, 'ismaster', ismaster, false, false), 'maxWireVersion'), J$.T(1369697, null, 25, false)))) {
                                J$.P(1369721, J$.R(1369705, 'ismaster', ismaster, false, false), 'maxWireVersion', J$.T(1369713, 0, 22, false));
                            }
                            J$.F(1369761, J$.R(1369729, 'setup_get_property', setup_get_property, false, false), false)(J$.R(1369737, 'this', this, false, false), J$.T(1369745, 'hasAggregationCursor', 21, false), J$.R(1369753, 'aggregationCursor', aggregationCursor, false, false));
                            J$.F(1369801, J$.R(1369769, 'setup_get_property', setup_get_property, false, false), false)(J$.R(1369777, 'this', this, false, false), J$.T(1369785, 'hasWriteCommands', 21, false), J$.R(1369793, 'writeCommands', writeCommands, false, false));
                            J$.F(1369841, J$.R(1369809, 'setup_get_property', setup_get_property, false, false), false)(J$.R(1369817, 'this', this, false, false), J$.T(1369825, 'hasTextSearch', 21, false), J$.R(1369833, 'textSearch', textSearch, false, false));
                            J$.F(1369881, J$.R(1369849, 'setup_get_property', setup_get_property, false, false), false)(J$.R(1369857, 'this', this, false, false), J$.T(1369865, 'hasAuthCommands', 21, false), J$.R(1369873, 'authCommands', authCommands, false, false));
                            J$.F(1369921, J$.R(1369889, 'setup_get_property', setup_get_property, false, false), false)(J$.R(1369897, 'this', this, false, false), J$.T(1369905, 'hasListCollectionsCommand', 21, false), J$.R(1369913, 'listCollections', listCollections, false, false));
                            J$.F(1369961, J$.R(1369929, 'setup_get_property', setup_get_property, false, false), false)(J$.R(1369937, 'this', this, false, false), J$.T(1369945, 'hasListIndexesCommand', 21, false), J$.R(1369953, 'listIndexes', listIndexes, false, false));
                            J$.F(1370009, J$.R(1369969, 'setup_get_property', setup_get_property, false, false), false)(J$.R(1369977, 'this', this, false, false), J$.T(1369985, 'minWireVersion', 21, false), J$.G(1370001, J$.R(1369993, 'ismaster', ismaster, false, false), 'minWireVersion'));
                            J$.F(1370057, J$.R(1370017, 'setup_get_property', setup_get_property, false, false), false)(J$.R(1370025, 'this', this, false, false), J$.T(1370033, 'maxWireVersion', 21, false), J$.G(1370049, J$.R(1370041, 'ismaster', ismaster, false, false), 'maxWireVersion'));
                            J$.F(1370097, J$.R(1370065, 'setup_get_property', setup_get_property, false, false), false)(J$.R(1370073, 'this', this, false, false), J$.T(1370081, 'maxNumberOfDocsInBatch', 21, false), J$.R(1370089, 'maxNumberOfDocsInBatch', maxNumberOfDocsInBatch, false, false));
                            J$.F(1370137, J$.R(1370105, 'setup_get_property', setup_get_property, false, false), false)(J$.R(1370113, 'this', this, false, false), J$.T(1370121, 'commandsTakeWriteConcern', 21, false), J$.R(1370129, 'commandsTakeWriteConcern', commandsTakeWriteConcern, false, false));
                            J$.F(1370177, J$.R(1370145, 'setup_get_property', setup_get_property, false, false), false)(J$.R(1370153, 'this', this, false, false), J$.T(1370161, 'commandsTakeCollation', 21, false), J$.R(1370169, 'commandsTakeCollation', commandsTakeCollation, false, false));
                        } catch (J$e) {
                            J$.Ex(1374745, J$e);
                        } finally {
                            if (J$.Fr(1374753))
                                continue jalangiLabel25;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), ServerCapabilities, false, true);
            var driverVersion = J$.W(1370457, 'driverVersion', J$.G(1370329, J$.F(1370321, J$.I(typeof require === 'undefined' ? require = J$.R(1370305, 'require', undefined, true, true) : require = J$.R(1370305, 'require', require, true, true)), false)(J$.T(1370313, '../../package.json', 21, false)), 'version'), driverVersion, false, true), nodejsversion = J$.W(1370465, 'nodejsversion', J$.F(1370385, J$.R(1370337, 'f', f, false, true), false)(J$.T(1370345, 'Node.js %s, %s', 21, false), J$.G(1370361, J$.I(typeof process === 'undefined' ? process = J$.R(1370353, 'process', undefined, true, true) : process = J$.R(1370353, 'process', process, true, true)), 'version'), J$.M(1370377, J$.R(1370369, 'os', os, false, true), 'endianness', false)()), nodejsversion, false, true), type = J$.W(1370473, 'type', J$.M(1370401, J$.R(1370393, 'os', os, false, true), 'type', false)(), type, false, true), name = J$.W(1370481, 'name', J$.G(1370417, J$.I(typeof process === 'undefined' ? process = J$.R(1370409, 'process', undefined, true, true) : process = J$.R(1370409, 'process', process, true, true)), 'platform'), name, false, true), architecture = J$.W(1370489, 'architecture', J$.G(1370433, J$.I(typeof process === 'undefined' ? process = J$.R(1370425, 'process', undefined, true, true) : process = J$.R(1370425, 'process', process, true, true)), 'arch'), architecture, false, true), release = J$.W(1370497, 'release', J$.M(1370449, J$.R(1370441, 'os', os, false, true), 'release', false)(), release, false, true);
            var TopologyBase = J$.W(1373473, 'TopologyBase', J$.F(1373465, J$.T(1373449, function (_EventEmitter) {
                jalangiLabel45:
                    while (true) {
                        try {
                            J$.Fe(1373409, arguments.callee, this, arguments);
                            function TopologyBase() {
                                jalangiLabel26:
                                    while (true) {
                                        try {
                                            J$.Fe(1370777, arguments.callee, this, arguments);
                                            arguments = J$.N(1370785, 'arguments', arguments, true, false, false);
                                            J$.N(1370793, '_this', _this, false, false, false);
                                            var _this;
                                            J$.F(1370561, J$.R(1370537, '_classCallCheck', _classCallCheck, false, true), false)(J$.R(1370545, 'this', this, false, false), J$.R(1370553, 'TopologyBase', TopologyBase, false, false));
                                            _this = J$.W(1370633, '_this', J$.F(1370625, J$.R(1370569, '_possibleConstructorReturn', _possibleConstructorReturn, false, true), false)(J$.R(1370577, 'this', this, false, false), J$.M(1370617, J$.F(1370601, J$.R(1370585, '_getPrototypeOf', _getPrototypeOf, false, true), false)(J$.R(1370593, 'TopologyBase', TopologyBase, false, false)), 'call', false)(J$.R(1370609, 'this', this, false, false))), _this, false, false);
                                            J$.P(1370729, J$.R(1370641, '_this', _this, false, false), 'clientInfo', J$.T(1370721, {
                                                driver: J$.T(1370665, {
                                                    name: J$.T(1370649, 'nodejs', 21, false),
                                                    version: J$.R(1370657, 'driverVersion', driverVersion, false, true)
                                                }, 11, false),
                                                os: J$.T(1370705, {
                                                    type: J$.R(1370673, 'type', type, false, true),
                                                    name: J$.R(1370681, 'name', name, false, true),
                                                    architecture: J$.R(1370689, 'architecture', architecture, false, true),
                                                    version: J$.R(1370697, 'release', release, false, true)
                                                }, 11, false),
                                                platform: J$.R(1370713, 'nodejsversion', nodejsversion, false, true)
                                            }, 11, false));
                                            J$.M(1370753, J$.R(1370737, '_this', _this, false, false), 'setMaxListeners', false)(J$.T(1370745, Infinity, 22, false));
                                            return J$.Rt(1370769, J$.R(1370761, '_this', _this, false, false));
                                        } catch (J$e) {
                                            J$.Ex(1374761, J$e);
                                        } finally {
                                            if (J$.Fr(1374769))
                                                continue jalangiLabel26;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }
                            arguments = J$.N(1373417, 'arguments', arguments, true, false, false);
                            _EventEmitter = J$.N(1373425, '_EventEmitter', _EventEmitter, true, false, false);
                            TopologyBase = J$.N(1373441, 'TopologyBase', J$.T(1373433, TopologyBase, 12, false), true, false, false);
                            J$.F(1370529, J$.R(1370505, '_inherits', _inherits, false, true), false)(J$.R(1370513, 'TopologyBase', TopologyBase, false, false), J$.R(1370521, '_EventEmitter', _EventEmitter, false, false));
                            J$.F(1373385, J$.R(1370801, '_createClass', _createClass, false, true), false)(J$.R(1370809, 'TopologyBase', TopologyBase, false, false), J$.T(1373377, [
                                J$.T(1370881, {
                                    key: J$.T(1370817, 'hasSessionSupport', 21, false),
                                    value: J$.T(1370873, function hasSessionSupport() {
                                        jalangiLabel27:
                                            while (true) {
                                                try {
                                                    J$.Fe(1370857, arguments.callee, this, arguments);
                                                    arguments = J$.N(1370865, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(1370849, J$.B(128906, '!=', J$.G(1370833, J$.R(1370825, 'this', this, false, false), 'logicalSessionTimeoutMinutes'), J$.T(1370841, null, 25, false)));
                                                } catch (J$e) {
                                                    J$.Ex(1374777, J$e);
                                                } finally {
                                                    if (J$.Fr(1374785))
                                                        continue jalangiLabel27;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1371193, {
                                    key: J$.T(1370889, 'startSession', 21, false),
                                    value: J$.T(1371185, function startSession(options, clientOptions) {
                                        jalangiLabel29:
                                            while (true) {
                                                try {
                                                    J$.Fe(1371137, arguments.callee, this, arguments);
                                                    arguments = J$.N(1371145, 'arguments', arguments, true, false, false);
                                                    options = J$.N(1371153, 'options', options, true, false, false);
                                                    clientOptions = J$.N(1371161, 'clientOptions', clientOptions, true, false, false);
                                                    J$.N(1371169, '_this2', _this2, false, false, false);
                                                    J$.N(1371177, 'session', session, false, false, false);
                                                    var _this2 = J$.W(1370905, '_this2', J$.R(1370897, 'this', this, false, false), _this2, false, false);
                                                    var session = J$.W(1370977, 'session', J$.F(1370969, J$.R(1370913, 'ClientSession', ClientSession, false, true), true)(J$.R(1370921, 'this', this, false, false), J$.G(1370945, J$.G(1370937, J$.R(1370929, 'this', this, false, false), 's'), 'sessionPool'), J$.R(1370953, 'options', options, false, false), J$.R(1370961, 'clientOptions', clientOptions, false, false)), session, false, false);
                                                    J$.M(1371073, J$.R(1370985, 'session', session, false, false), 'once', false)(J$.T(1370993, 'ended', 21, false), J$.T(1371065, function () {
                                                        jalangiLabel28:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(1371049, arguments.callee, this, arguments);
                                                                    arguments = J$.N(1371057, 'arguments', arguments, true, false, false);
                                                                    J$.M(1371041, J$.G(1371017, J$.G(1371009, J$.R(1371001, '_this2', _this2, false, false), 's'), 'sessions'), J$.T(1371025, 'delete', 21, false), false)(J$.R(1371033, 'session', session, false, false));
                                                                } catch (J$e) {
                                                                    J$.Ex(1374793, J$e);
                                                                } finally {
                                                                    if (J$.Fr(1374801))
                                                                        continue jalangiLabel28;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                    J$.M(1371113, J$.G(1371097, J$.G(1371089, J$.R(1371081, 'this', this, false, false), 's'), 'sessions'), 'add', false)(J$.R(1371105, 'session', session, false, false));
                                                    return J$.Rt(1371129, J$.R(1371121, 'session', session, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1374809, J$e);
                                                } finally {
                                                    if (J$.Fr(1374817))
                                                        continue jalangiLabel29;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1371305, {
                                    key: J$.T(1371201, 'endSessions', 21, false),
                                    value: J$.T(1371297, function endSessions(sessions, callback) {
                                        jalangiLabel30:
                                            while (true) {
                                                try {
                                                    J$.Fe(1371265, arguments.callee, this, arguments);
                                                    arguments = J$.N(1371273, 'arguments', arguments, true, false, false);
                                                    sessions = J$.N(1371281, 'sessions', sessions, true, false, false);
                                                    callback = J$.N(1371289, 'callback', callback, true, false, false);
                                                    return J$.Rt(1371257, J$.M(1371249, J$.G(1371225, J$.G(1371217, J$.R(1371209, 'this', this, false, false), 's'), 'coreTopology'), 'endSessions', false)(J$.R(1371233, 'sessions', sessions, false, false), J$.R(1371241, 'callback', callback, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(1374825, J$e);
                                                } finally {
                                                    if (J$.Fr(1374833))
                                                        continue jalangiLabel30;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1371561, {
                                    key: J$.T(1371313, 'capabilities', 21, false),
                                    value: J$.T(1371553, function capabilities() {
                                        jalangiLabel31:
                                            while (true) {
                                                try {
                                                    J$.Fe(1371537, arguments.callee, this, arguments);
                                                    arguments = J$.N(1371545, 'arguments', arguments, true, false, false);
                                                    if (J$.C(86296, J$.G(1371337, J$.G(1371329, J$.R(1371321, 'this', this, false, false), 's'), 'sCapabilities')))
                                                        return J$.Rt(1371369, J$.G(1371361, J$.G(1371353, J$.R(1371345, 'this', this, false, false), 's'), 'sCapabilities'));
                                                    if (J$.C(86304, J$.B(128914, '==', J$.M(1371401, J$.G(1371393, J$.G(1371385, J$.R(1371377, 'this', this, false, false), 's'), 'coreTopology'), 'lastIsMaster', false)(), J$.T(1371409, null, 25, false))))
                                                        return J$.Rt(1371425, J$.T(1371417, null, 25, false));
                                                    J$.P(1371497, J$.G(1371441, J$.R(1371433, 'this', this, false, false), 's'), 'sCapabilities', J$.F(1371489, J$.R(1371449, 'ServerCapabilities', ServerCapabilities, false, true), true)(J$.M(1371481, J$.G(1371473, J$.G(1371465, J$.R(1371457, 'this', this, false, false), 's'), 'coreTopology'), 'lastIsMaster', false)()));
                                                    return J$.Rt(1371529, J$.G(1371521, J$.G(1371513, J$.R(1371505, 'this', this, false, false), 's'), 'sCapabilities'));
                                                } catch (J$e) {
                                                    J$.Ex(1374841, J$e);
                                                } finally {
                                                    if (J$.Fr(1374849))
                                                        continue jalangiLabel31;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1371721, {
                                    key: J$.T(1371569, 'command', 21, false),
                                    value: J$.T(1371713, function command(ns, cmd, options, callback) {
                                        jalangiLabel32:
                                            while (true) {
                                                try {
                                                    J$.Fe(1371665, arguments.callee, this, arguments);
                                                    arguments = J$.N(1371673, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(1371681, 'ns', ns, true, false, false);
                                                    cmd = J$.N(1371689, 'cmd', cmd, true, false, false);
                                                    options = J$.N(1371697, 'options', options, true, false, false);
                                                    callback = J$.N(1371705, 'callback', callback, true, false, false);
                                                    J$.M(1371657, J$.G(1371593, J$.G(1371585, J$.R(1371577, 'this', this, false, false), 's'), 'coreTopology'), 'command', false)(J$.M(1371609, J$.R(1371601, 'ns', ns, false, false), 'toString', false)(), J$.R(1371617, 'cmd', cmd, false, false), J$.F(1371641, J$.R(1371625, 'translateReadPreference', translateReadPreference, false, true), false)(J$.R(1371633, 'options', options, false, false)), J$.R(1371649, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1374857, J$e);
                                                } finally {
                                                    if (J$.Fr(1374865))
                                                        continue jalangiLabel32;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1371865, {
                                    key: J$.T(1371729, 'insert', 21, false),
                                    value: J$.T(1371857, function insert(ns, ops, options, callback) {
                                        jalangiLabel33:
                                            while (true) {
                                                try {
                                                    J$.Fe(1371809, arguments.callee, this, arguments);
                                                    arguments = J$.N(1371817, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(1371825, 'ns', ns, true, false, false);
                                                    ops = J$.N(1371833, 'ops', ops, true, false, false);
                                                    options = J$.N(1371841, 'options', options, true, false, false);
                                                    callback = J$.N(1371849, 'callback', callback, true, false, false);
                                                    J$.M(1371801, J$.G(1371753, J$.G(1371745, J$.R(1371737, 'this', this, false, false), 's'), 'coreTopology'), 'insert', false)(J$.M(1371769, J$.R(1371761, 'ns', ns, false, false), 'toString', false)(), J$.R(1371777, 'ops', ops, false, false), J$.R(1371785, 'options', options, false, false), J$.R(1371793, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1374873, J$e);
                                                } finally {
                                                    if (J$.Fr(1374881))
                                                        continue jalangiLabel33;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1372009, {
                                    key: J$.T(1371873, 'update', 21, false),
                                    value: J$.T(1372001, function update(ns, ops, options, callback) {
                                        jalangiLabel34:
                                            while (true) {
                                                try {
                                                    J$.Fe(1371953, arguments.callee, this, arguments);
                                                    arguments = J$.N(1371961, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(1371969, 'ns', ns, true, false, false);
                                                    ops = J$.N(1371977, 'ops', ops, true, false, false);
                                                    options = J$.N(1371985, 'options', options, true, false, false);
                                                    callback = J$.N(1371993, 'callback', callback, true, false, false);
                                                    J$.M(1371945, J$.G(1371897, J$.G(1371889, J$.R(1371881, 'this', this, false, false), 's'), 'coreTopology'), 'update', false)(J$.M(1371913, J$.R(1371905, 'ns', ns, false, false), 'toString', false)(), J$.R(1371921, 'ops', ops, false, false), J$.R(1371929, 'options', options, false, false), J$.R(1371937, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1374889, J$e);
                                                } finally {
                                                    if (J$.Fr(1374897))
                                                        continue jalangiLabel34;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1372153, {
                                    key: J$.T(1372017, 'remove', 21, false),
                                    value: J$.T(1372145, function remove(ns, ops, options, callback) {
                                        jalangiLabel35:
                                            while (true) {
                                                try {
                                                    J$.Fe(1372097, arguments.callee, this, arguments);
                                                    arguments = J$.N(1372105, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(1372113, 'ns', ns, true, false, false);
                                                    ops = J$.N(1372121, 'ops', ops, true, false, false);
                                                    options = J$.N(1372129, 'options', options, true, false, false);
                                                    callback = J$.N(1372137, 'callback', callback, true, false, false);
                                                    J$.M(1372089, J$.G(1372041, J$.G(1372033, J$.R(1372025, 'this', this, false, false), 's'), 'coreTopology'), 'remove', false)(J$.M(1372057, J$.R(1372049, 'ns', ns, false, false), 'toString', false)(), J$.R(1372065, 'ops', ops, false, false), J$.R(1372073, 'options', options, false, false), J$.R(1372081, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1374905, J$e);
                                                } finally {
                                                    if (J$.Fr(1374913))
                                                        continue jalangiLabel35;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1372305, {
                                    key: J$.T(1372161, 'isConnected', 21, false),
                                    value: J$.T(1372297, function isConnected(options) {
                                        jalangiLabel36:
                                            while (true) {
                                                try {
                                                    J$.Fe(1372273, arguments.callee, this, arguments);
                                                    arguments = J$.N(1372281, 'arguments', arguments, true, false, false);
                                                    options = J$.N(1372289, 'options', options, true, false, false);
                                                    options = J$.W(1372185, 'options', J$.C(86312, J$.R(1372169, 'options', options, false, false)) ? J$._() : J$.T(1372177, {}, 11, false), options, false, false);
                                                    options = J$.W(1372217, 'options', J$.F(1372209, J$.R(1372193, 'translateReadPreference', translateReadPreference, false, true), false)(J$.R(1372201, 'options', options, false, false)), options, false, false);
                                                    return J$.Rt(1372265, J$.M(1372257, J$.G(1372241, J$.G(1372233, J$.R(1372225, 'this', this, false, false), 's'), 'coreTopology'), 'isConnected', false)(J$.R(1372249, 'options', options, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(1374921, J$e);
                                                } finally {
                                                    if (J$.Fr(1374929))
                                                        continue jalangiLabel36;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1372385, {
                                    key: J$.T(1372313, 'isDestroyed', 21, false),
                                    value: J$.T(1372377, function isDestroyed() {
                                        jalangiLabel37:
                                            while (true) {
                                                try {
                                                    J$.Fe(1372361, arguments.callee, this, arguments);
                                                    arguments = J$.N(1372369, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(1372353, J$.M(1372345, J$.G(1372337, J$.G(1372329, J$.R(1372321, 'this', this, false, false), 's'), 'coreTopology'), 'isDestroyed', false)());
                                                } catch (J$e) {
                                                    J$.Ex(1374937, J$e);
                                                } finally {
                                                    if (J$.Fr(1374945))
                                                        continue jalangiLabel37;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1372633, {
                                    key: J$.T(1372393, 'cursor', 21, false),
                                    value: J$.T(1372625, function cursor(ns, cmd, options) {
                                        jalangiLabel38:
                                            while (true) {
                                                try {
                                                    J$.Fe(1372585, arguments.callee, this, arguments);
                                                    arguments = J$.N(1372593, 'arguments', arguments, true, false, false);
                                                    ns = J$.N(1372601, 'ns', ns, true, false, false);
                                                    cmd = J$.N(1372609, 'cmd', cmd, true, false, false);
                                                    options = J$.N(1372617, 'options', options, true, false, false);
                                                    options = J$.W(1372417, 'options', J$.C(86320, J$.R(1372401, 'options', options, false, false)) ? J$._() : J$.T(1372409, {}, 11, false), options, false, false);
                                                    options = J$.W(1372449, 'options', J$.F(1372441, J$.R(1372425, 'translateReadPreference', translateReadPreference, false, true), false)(J$.R(1372433, 'options', options, false, false)), options, false, false);
                                                    J$.P(1372489, J$.R(1372457, 'options', options, false, false), 'disconnectHandler', J$.G(1372481, J$.G(1372473, J$.R(1372465, 'this', this, false, false), 's'), 'store'));
                                                    J$.P(1372513, J$.R(1372497, 'options', options, false, false), 'topology', J$.R(1372505, 'this', this, false, false));
                                                    return J$.Rt(1372577, J$.M(1372569, J$.G(1372537, J$.G(1372529, J$.R(1372521, 'this', this, false, false), 's'), 'coreTopology'), 'cursor', false)(J$.R(1372545, 'ns', ns, false, false), J$.R(1372553, 'cmd', cmd, false, false), J$.R(1372561, 'options', options, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(1374953, J$e);
                                                } finally {
                                                    if (J$.Fr(1374961))
                                                        continue jalangiLabel38;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1372713, {
                                    key: J$.T(1372641, 'lastIsMaster', 21, false),
                                    value: J$.T(1372705, function lastIsMaster() {
                                        jalangiLabel39:
                                            while (true) {
                                                try {
                                                    J$.Fe(1372689, arguments.callee, this, arguments);
                                                    arguments = J$.N(1372697, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(1372681, J$.M(1372673, J$.G(1372665, J$.G(1372657, J$.R(1372649, 'this', this, false, false), 's'), 'coreTopology'), 'lastIsMaster', false)());
                                                } catch (J$e) {
                                                    J$.Ex(1374969, J$e);
                                                } finally {
                                                    if (J$.Fr(1374977))
                                                        continue jalangiLabel39;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1372841, {
                                    key: J$.T(1372721, 'selectServer', 21, false),
                                    value: J$.T(1372833, function selectServer(selector, options, callback) {
                                        jalangiLabel40:
                                            while (true) {
                                                try {
                                                    J$.Fe(1372793, arguments.callee, this, arguments);
                                                    arguments = J$.N(1372801, 'arguments', arguments, true, false, false);
                                                    selector = J$.N(1372809, 'selector', selector, true, false, false);
                                                    options = J$.N(1372817, 'options', options, true, false, false);
                                                    callback = J$.N(1372825, 'callback', callback, true, false, false);
                                                    return J$.Rt(1372785, J$.M(1372777, J$.G(1372745, J$.G(1372737, J$.R(1372729, 'this', this, false, false), 's'), 'coreTopology'), 'selectServer', false)(J$.R(1372753, 'selector', selector, false, false), J$.R(1372761, 'options', options, false, false), J$.R(1372769, 'callback', callback, false, false)));
                                                } catch (J$e) {
                                                    J$.Ex(1374985, J$e);
                                                } finally {
                                                    if (J$.Fr(1374993))
                                                        continue jalangiLabel40;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1372921, {
                                    key: J$.T(1372849, 'unref', 21, false),
                                    value: J$.T(1372913, function unref() {
                                        jalangiLabel41:
                                            while (true) {
                                                try {
                                                    J$.Fe(1372897, arguments.callee, this, arguments);
                                                    arguments = J$.N(1372905, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(1372889, J$.M(1372881, J$.G(1372873, J$.G(1372865, J$.R(1372857, 'this', this, false, false), 's'), 'coreTopology'), 'unref', false)());
                                                } catch (J$e) {
                                                    J$.Ex(1375001, J$e);
                                                } finally {
                                                    if (J$.Fr(1375009))
                                                        continue jalangiLabel41;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1373001, {
                                    key: J$.T(1372929, 'connections', 21, false),
                                    value: J$.T(1372993, function connections() {
                                        jalangiLabel42:
                                            while (true) {
                                                try {
                                                    J$.Fe(1372977, arguments.callee, this, arguments);
                                                    arguments = J$.N(1372985, 'arguments', arguments, true, false, false);
                                                    return J$.Rt(1372969, J$.M(1372961, J$.G(1372953, J$.G(1372945, J$.R(1372937, 'this', this, false, false), 's'), 'coreTopology'), 'connections', false)());
                                                } catch (J$e) {
                                                    J$.Ex(1375017, J$e);
                                                } finally {
                                                    if (J$.Fr(1375025))
                                                        continue jalangiLabel42;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false),
                                J$.T(1373369, {
                                    key: J$.T(1373009, 'close', 21, false),
                                    value: J$.T(1373361, function close(forceClosed, callback) {
                                        jalangiLabel44:
                                            while (true) {
                                                try {
                                                    J$.Fe(1373329, arguments.callee, this, arguments);
                                                    arguments = J$.N(1373337, 'arguments', arguments, true, false, false);
                                                    forceClosed = J$.N(1373345, 'forceClosed', forceClosed, true, false, false);
                                                    callback = J$.N(1373353, 'callback', callback, true, false, false);
                                                    J$.M(1373097, J$.G(1373033, J$.G(1373025, J$.R(1373017, 'this', this, false, false), 's'), 'sessions'), 'forEach', false)(J$.T(1373089, function (session) {
                                                        jalangiLabel43:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(1373065, arguments.callee, this, arguments);
                                                                    arguments = J$.N(1373073, 'arguments', arguments, true, false, false);
                                                                    session = J$.N(1373081, 'session', session, true, false, false);
                                                                    return J$.Rt(1373057, J$.M(1373049, J$.R(1373041, 'session', session, false, false), 'endSession', false)());
                                                                } catch (J$e) {
                                                                    J$.Ex(1375033, J$e);
                                                                } finally {
                                                                    if (J$.Fr(1375041))
                                                                        continue jalangiLabel43;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false));
                                                    if (J$.C(86328, J$.G(1373121, J$.G(1373113, J$.R(1373105, 'this', this, false, false), 's'), 'sessionPool'))) {
                                                        J$.M(1373153, J$.G(1373145, J$.G(1373137, J$.R(1373129, 'this', this, false, false), 's'), 'sessionPool'), 'endAllPooledSessions', false)();
                                                    }
                                                    if (J$.C(86336, J$.B(128922, '===', J$.R(1373161, 'forceClosed', forceClosed, false, false), J$.T(1373169, true, 23, false)))) {
                                                        J$.P(1373209, J$.G(1373193, J$.G(1373185, J$.R(1373177, 'this', this, false, false), 's'), 'storeOptions'), 'force', J$.R(1373201, 'forceClosed', forceClosed, false, false));
                                                        J$.M(1373241, J$.G(1373233, J$.G(1373225, J$.R(1373217, 'this', this, false, false), 's'), 'store'), 'flush', false)();
                                                    }
                                                    J$.M(1373321, J$.G(1373265, J$.G(1373257, J$.R(1373249, 'this', this, false, false), 's'), 'coreTopology'), 'destroy', false)(J$.T(1373305, { force: J$.C(86344, J$.B(128938, '===', J$.U(128930, 'typeof', J$.R(1373273, 'forceClosed', forceClosed, false, false)), J$.T(1373281, 'boolean', 21, false))) ? J$.R(1373289, 'forceClosed', forceClosed, false, false) : J$.T(1373297, false, 23, false) }, 11, false), J$.R(1373313, 'callback', callback, false, false));
                                                } catch (J$e) {
                                                    J$.Ex(1375049, J$e);
                                                } finally {
                                                    if (J$.Fr(1375057))
                                                        continue jalangiLabel44;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false)
                                }, 11, false)
                            ], 10, false));
                            return J$.Rt(1373401, J$.R(1373393, 'TopologyBase', TopologyBase, false, false));
                        } catch (J$e) {
                            J$.Ex(1375065, J$e);
                        } finally {
                            if (J$.Fr(1375073))
                                continue jalangiLabel45;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false), false)(J$.R(1373457, 'EventEmitter', EventEmitter, false, true)), TopologyBase, false, true);
            J$.M(1373601, J$.I(typeof Object === 'undefined' ? Object = J$.R(1373481, 'Object', undefined, true, true) : Object = J$.R(1373481, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(1373497, J$.R(1373489, 'TopologyBase', TopologyBase, false, true), 'prototype'), J$.T(1373505, 'bson', 21, false), J$.T(1373593, {
                enumerable: J$.T(1373513, true, 23, false),
                get: J$.T(1373585, function get() {
                    jalangiLabel46:
                        while (true) {
                            try {
                                J$.Fe(1373569, arguments.callee, this, arguments);
                                arguments = J$.N(1373577, 'arguments', arguments, true, false, false);
                                return J$.Rt(1373561, J$.G(1373553, J$.G(1373545, J$.G(1373537, J$.G(1373529, J$.R(1373521, 'this', this, false, false), 's'), 'coreTopology'), 's'), 'bson'));
                            } catch (J$e) {
                                J$.Ex(1375081, J$e);
                            } finally {
                                if (J$.Fr(1375089))
                                    continue jalangiLabel46;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.M(1373721, J$.I(typeof Object === 'undefined' ? Object = J$.R(1373609, 'Object', undefined, true, true) : Object = J$.R(1373609, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(1373625, J$.R(1373617, 'TopologyBase', TopologyBase, false, true), 'prototype'), J$.T(1373633, 'parserType', 21, false), J$.T(1373713, {
                enumerable: J$.T(1373641, true, 23, false),
                get: J$.T(1373705, function get() {
                    jalangiLabel47:
                        while (true) {
                            try {
                                J$.Fe(1373689, arguments.callee, this, arguments);
                                arguments = J$.N(1373697, 'arguments', arguments, true, false, false);
                                return J$.Rt(1373681, J$.G(1373673, J$.G(1373665, J$.G(1373657, J$.R(1373649, 'this', this, false, false), 's'), 'coreTopology'), 'parserType'));
                            } catch (J$e) {
                                J$.Ex(1375097, J$e);
                            } finally {
                                if (J$.Fr(1375105))
                                    continue jalangiLabel47;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.M(1373841, J$.I(typeof Object === 'undefined' ? Object = J$.R(1373729, 'Object', undefined, true, true) : Object = J$.R(1373729, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(1373745, J$.R(1373737, 'TopologyBase', TopologyBase, false, true), 'prototype'), J$.T(1373753, 'logicalSessionTimeoutMinutes', 21, false), J$.T(1373833, {
                enumerable: J$.T(1373761, true, 23, false),
                get: J$.T(1373825, function get() {
                    jalangiLabel48:
                        while (true) {
                            try {
                                J$.Fe(1373809, arguments.callee, this, arguments);
                                arguments = J$.N(1373817, 'arguments', arguments, true, false, false);
                                return J$.Rt(1373801, J$.G(1373793, J$.G(1373785, J$.G(1373777, J$.R(1373769, 'this', this, false, false), 's'), 'coreTopology'), 'logicalSessionTimeoutMinutes'));
                            } catch (J$e) {
                                J$.Ex(1375113, J$e);
                            } finally {
                                if (J$.Fr(1375121))
                                    continue jalangiLabel48;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.M(1373961, J$.I(typeof Object === 'undefined' ? Object = J$.R(1373849, 'Object', undefined, true, true) : Object = J$.R(1373849, 'Object', Object, true, true)), 'defineProperty', false)(J$.G(1373865, J$.R(1373857, 'TopologyBase', TopologyBase, false, true), 'prototype'), J$.T(1373873, 'type', 21, false), J$.T(1373953, {
                enumerable: J$.T(1373881, true, 23, false),
                get: J$.T(1373945, function get() {
                    jalangiLabel49:
                        while (true) {
                            try {
                                J$.Fe(1373929, arguments.callee, this, arguments);
                                arguments = J$.N(1373937, 'arguments', arguments, true, false, false);
                                return J$.Rt(1373921, J$.G(1373913, J$.G(1373905, J$.G(1373897, J$.R(1373889, 'this', this, false, false), 's'), 'coreTopology'), 'type'));
                            } catch (J$e) {
                                J$.Ex(1375129, J$e);
                            } finally {
                                if (J$.Fr(1375137))
                                    continue jalangiLabel49;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false)
            }, 11, false));
            J$.P(1373985, J$.I(typeof exports === 'undefined' ? exports = J$.R(1373969, 'exports', undefined, true, true) : exports = J$.R(1373969, 'exports', exports, true, true)), 'Store', J$.R(1373977, 'Store', Store, false, true));
            J$.P(1374009, J$.I(typeof exports === 'undefined' ? exports = J$.R(1373993, 'exports', undefined, true, true) : exports = J$.R(1373993, 'exports', exports, true, true)), 'ServerCapabilities', J$.R(1374001, 'ServerCapabilities', ServerCapabilities, false, true));
            J$.P(1374033, J$.I(typeof exports === 'undefined' ? exports = J$.R(1374017, 'exports', undefined, true, true) : exports = J$.R(1374017, 'exports', exports, true, true)), 'TopologyBase', J$.R(1374025, 'TopologyBase', TopologyBase, false, true));
        } catch (J$e) {
            J$.Ex(1375145, J$e);
        } finally {
            if (J$.Sr(1375153))
                continue jalangiLabel50;
            else
                break jalangiLabel50;
        }
    }
// JALANGI DO NOT INSTRUMENT
